<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<title>Untitled</title>
</head>
<body><div class="page"><p/>
<p>Structure
Determination
by X-ray
Crystallography
</p>
<p>Mark Ladd
Rex Palmer
</p>
<p>Analysis by X-rays and Neutrons
</p>
<p>Fifth Edition</p>
<p/>
</div>
<div class="page"><p/>
<p>Structure Determination by X-ray
Crystallography</p>
<p/>
</div>
<div class="page"><p/>
<p>Official contribution of the National Institute of Standards and Technology; not subject to copyright in the United States
</p>
<p>[Reproduced by courtesy of N. I. S. T.]</p>
<p/>
</div>
<div class="page"><p/>
<p>Mark Ladd &bull; Rex Palmer
</p>
<p>Structure
Determination by
X-ray Crystallography
</p>
<p>Analysis by X-rays and Neutrons
</p>
<p>Fifth Edition
</p>
<p>Celebrating the Centenary of
</p>
<p>X-ray Crystallography</p>
<p/>
</div>
<div class="page"><p/>
<p>Additional material to this book can be downloaded from http://extra.springer.com.
</p>
<p>ISBN 978-1-4614-3956-1 ISBN 978-1-4614-3954-7 (eBook)
DOI 10.1007/978-1-4614-3954-7
Springer New York Heidelberg Dordrecht London
</p>
<p>Library of Congress Control Number: 2012947362
</p>
<p># Springer Science+Business Media New York 1977, 1985, 1994, 2003, 2013
</p>
<p>This work is subject to copyright. All rights are reserved by the Publisher, whether the whole or
</p>
<p>part of the material is concerned, specifically the rights of translation, reprinting, reuse of
</p>
<p>illustrations, recitation, broadcasting, reproduction on microfilms or in any other physical way,
</p>
<p>and transmission or information storage and retrieval, electronic adaptation, computer software,
</p>
<p>or by similar or dissimilar methodology now known or hereafter developed. Exempted from this
</p>
<p>legal reservation are brief excerpts in connection with reviews or scholarly analysis or material
</p>
<p>supplied specifically for the purpose of being entered and executed on a computer system, for
</p>
<p>exclusive use by the purchaser of the work. Duplication of this publication or parts thereof is
</p>
<p>permitted only under the provisions of the Copyright Law of the Publisher&rsquo;s location, in its
</p>
<p>current version, and permission for use must always be obtained from Springer. Permissions for
</p>
<p>use may be obtained through RightsLink at the Copyright Clearance Center. Violations are liable
</p>
<p>to prosecution under the respective Copyright Law.
</p>
<p>The use of general descriptive names, registered names, trademarks, service marks, etc. in this
</p>
<p>publication does not imply, even in the absence of a specific statement, that such names are
</p>
<p>exempt from the relevant protective laws and regulations and therefore free for general use.
</p>
<p>While the advice and information in this book are believed to be true and accurate at the date of
</p>
<p>publication, neither the authors nor the editors nor the publisher can accept any legal
</p>
<p>responsibility for any errors or omissions that may be made. The publisher makes no warranty,
</p>
<p>express or implied, with respect to the material contained herein.
</p>
<p>Printed on acid-free paper
</p>
<p>Springer is part of Springer Science+Business Media (www.springer.com)
</p>
<p>Rex Palmer
Reader Emeritus in Structural Crystallography
Birkbeck College, University of London,
London, England
</p>
<p>Visiting Professor in X-ray Crystallography
</p>
<p>University of Greenwich, England
</p>
<p>Senior Visiting Research Fellow
</p>
<p>Christ Church University,
</p>
<p>Canterbury, England
</p>
<p>Mark Ladd
Formerly Head of Chemical Physics
University of Surrey
Guildford, England</p>
<p/>
</div>
<div class="page"><p/>
<p>When you can measure what you are speaking about and
express it in numbers, you know something about it; but when
you cannot express it in numbers, your knowledge is of a meagre
and unsatisfactory kind; it may be the beginning of knowledge,
but you have scarcely in your thoughts advanced to the state of
science, whatever the matter may be.
</p>
<p>Lord Kelvin</p>
<p/>
</div>
<div class="page"><p/>
</div>
<div class="page"><p/>
<p>To Valentia and Hilda</p>
<p/>
</div>
<div class="page"><p/>
</div>
<div class="page"><p/>
<p>Foreword
</p>
<p>I am privileged to write the Foreword to this fifth edition of Ladd and
</p>
<p>Palmer&rsquo;s Structure Determination by X-ray Crystallography, a textbook
</p>
<p>that is now world renowned and that has helped educate two generations of
</p>
<p>crystallographers in the theory and practice of modern crystallography,
</p>
<p>myself included. Indeed, a well-worn first edition of this venerable text
</p>
<p>remains on my shelves today, now somewhat battered and bruised from
</p>
<p>passage through the hands of successive students who have learned the
</p>
<p>fundamentals of crystallography from its pages.
</p>
<p>This new fifth edition is especially timely, marking as it does a century
</p>
<p>of discovery in which X-ray diffraction, and diffraction of other radiations,
</p>
<p>has opened a window to the atomic world. From fundamental knowledge of
</p>
<p>atomic interactions and chemical bonds in the simplest materials to the
</p>
<p>atomic resolution analysis of the molecular machines of the cell, crystallo-
</p>
<p>graphic science underpins much of our understanding of the world we live in
</p>
<p>today. In recent years, advances in diffraction theory, automated technolo-
</p>
<p>gies, and computational tools have helped move crystallography from a
</p>
<p>specialist discipline to a standard laboratory tool across many fields of
</p>
<p>science. In some fields, these advances have been so spectacularly successful
</p>
<p>that the solution of the crystal structures of all but the most challenging
</p>
<p>systems is now considered largely routine. At the same time, the develop-
</p>
<p>ment of a new generation of high powered synchrotron, neutron and, most
</p>
<p>recently, free electron laser facilities are pushing crystallographic science to
</p>
<p>new frontiers, aiming to provide diffraction from single molecules, to locate
</p>
<p>light atoms such as hydrogen in crystal structures, and to move beyond static
</p>
<p>crystal structures towards time-resolved analyses of structural dynamics at
</p>
<p>pico-second timescales.
</p>
<p>For the interdisciplinary students of today seeking a thorough and
</p>
<p>detailed understanding of the principles and methods of modern crystallog-
</p>
<p>raphy, Ladd and Palmer remains as essential and relevant today as when it
</p>
<p>first appeared some 35 years ago. Building from the fundamental concepts of
</p>
<p>crystallography, through crystal symmetry to the mathematical formalism of
</p>
<p>diffraction and on to the principle and practice of structure determination, the
</p>
<p>text provides an excellent introduction to the techniques and applications of
</p>
<p>crystallography, illustrated throughout by applications to real world pro-
</p>
<p>blems. The fifth edition is expanded and enhanced with updated examples
</p>
<p>and description of recent technical developments and achievements in X-ray
</p>
<p>crystallography and benefits from a completely new chapter that describes
</p>
<p>ix</p>
<p/>
</div>
<div class="page"><p/>
<p>the application of neutron crystallography in structural science. This is an
</p>
<p>important addition. Neutrons are scattered by atomic nuclei and have a
</p>
<p>magnetic moment. Hence, neutron diffraction can be used to determine
</p>
<p>accurate atomic and magnetic structures of materials. With a new generation
</p>
<p>of neutron sources and instruments now coming on-line, these properties will
</p>
<p>be increasingly exploited in fundamental studies of new inorganic, organic,
</p>
<p>and biological systems, of superconducting and magnetic materials, and for
</p>
<p>structure-function analysis of hydrogen atoms in macromolecules.
</p>
<p>Extending the scope of this classic text beyond the purely X-ray Crystal-
</p>
<p>lography of its title to include diffraction of other radiations acknowledges
</p>
<p>some of the new frontiers and ever-increasing impact of crystallographic
</p>
<p>analysis in structural sciences. As has been the case for the last 35 years,
</p>
<p>Ladd and Palmer is set to educate and equip the students of today to drive and
</p>
<p>inspire the developments of tomorrow!
</p>
<p>Neutron Sciences Directorate
</p>
<p>Oak Ridge National Laboratory, TN, USA Dean A.A. Myles
</p>
<p>x Foreword</p>
<p/>
</div>
<div class="page"><p/>
<p>Preface to the Fifth Edition
</p>
<p>We were honoured to be asked by Springer, New York to prepare a fifth
</p>
<p>edition of Structure Determination by X-ray Crystallography. First published
</p>
<p>in 1977 under the Plenum imprint, this book has received wide acclaim in
</p>
<p>both teaching and research in X-ray crystallography because of its extensive
</p>
<p>and detailed coverage of all aspects of the subject.
</p>
<p>As we prepare this new edition, we are entering the centenary of the
</p>
<p>discovery of X-ray diffraction in 1912, the beginning of X-ray crystallogra-
</p>
<p>phy as a science in its own right. Today, X-ray crystallography and the
</p>
<p>complementary technique of neutron diffraction together provide the most
</p>
<p>powerful tools for the investigation and elucidation of crystal and molecular
</p>
<p>structures. X-ray and neutron crystallography may be described as the sci-
</p>
<p>ence of the structure of materials, in the widest sense of the phrase, and their
</p>
<p>ramifications are evident across a broad spectrum of scientific endeavour.
</p>
<p>The power of computers and available software has unleashed an unprec-
</p>
<p>edented ability to carry out with speed the complicated calculations involved
</p>
<p>in crystal structure determination on a desktop PC. This is paralleled by the
</p>
<p>availability of powerful X-ray and neutron sources and low temperature
</p>
<p>devices for facilitating measurements at liquid nitrogen temperature or
</p>
<p>lower, which provide ever higher precision in the determination of crystal
</p>
<p>structures. However, a detailed knowledge of the theory underlying the
</p>
<p>process of crystal structure determination is still required in order both to
</p>
<p>ensure that the literature contains correct well-determined structures and to
</p>
<p>understand the complexities introduced by features such as disorder and
</p>
<p>twinning in crystals. There are many pitfalls in crystal structure determina-
</p>
<p>tion to trap the unwary.
</p>
<p>In this new edition, we have continued the approach that has been well
</p>
<p>reviewed in its earlier editions. We have always kept in mind that students
</p>
<p>meeting X-ray crystallography for the first time are encountering a new disci-
</p>
<p>pline, and not merely extending the range of a subject already studied. In
</p>
<p>consequence, we have chosen, for example, to discuss the geometry and sym-
</p>
<p>metry of crystals in rather more detail than is found in other books on this
</p>
<p>subject, for it is our experience that some of the difficulties that students meet in
</p>
<p>introductory X-ray crystallography lie in their unfamiliarity with a three-
</p>
<p>dimensional concept, whether they be final-year undergraduate or post-graduate
</p>
<p>students in chemistry, biochemistry, materials science, geology, bioinformatics,
</p>
<p>information technology, or physics. Both low molecular weight (small
</p>
<p>molecules) and macromolecular methods (proteins) are covered in detail.
</p>
<p>xi</p>
<p/>
</div>
<div class="page"><p/>
<p>As well as retaining and thoroughly revising the overall contents of the
</p>
<p>earlier editions, we have added a significant chapter on neutron diffraction
</p>
<p>studies, and sections introducing Molecular Modelling and Structure Predic-
</p>
<p>tion. In order to maintain a workable size for the book, a number of elabora-
</p>
<p>tions of mainly mathematical argument have been stored as Web Appendices
</p>
<p>on the website http://extras.springer.com.
</p>
<p>Although several novel methods have been added to the armoury of
</p>
<p>crystal structure determination, we limit our discussion principally to Patter-
</p>
<p>son interpretation, Direct Methods, Isomorphous and Molecular Replace-
</p>
<p>ment and Powder Crystallography, and developments from them. The basic
</p>
<p>problem remains the determination of the phases of X-ray reflections, and
</p>
<p>this problem is addressed in these techniques discussed herein. In order to
</p>
<p>simulate the actual process of structure determination, we are fortunate to be
</p>
<p>able to include the XRAY program package prepared by Dr. Neil Bailey and
</p>
<p>colleagues of the University of Sheffield, and we are grateful to him for
</p>
<p>permission to use it in the present context. It has been modified (M.L.) for PC
</p>
<p>operation and several enhancements made, including the presentation of
</p>
<p>Fourier contour maps on the monitor. Although this package uses two-
</p>
<p>dimensional data, much valuable insight into X-ray structure determination
</p>
<p>can be gained, and a number of sets of X-ray data are included.
</p>
<p>There are now numerous computer packages available for the many
</p>
<p>aspects of crystallography that are in current use. We have referred to them
</p>
<p>freely within the text, and they have been collected in an appendix together
</p>
<p>with a reference to a source for each so that they become readily available to
</p>
<p>the practising crystallographer. There are numerous references to each chap-
</p>
<p>ter including website addresses for topics of crystallographic importance.
</p>
<p>References among the text are given as &ldquo;Sect. 1.2.3,&rdquo; which refers to that
</p>
<p>section in Chap. 1, or as &ldquo;(3.4)&rdquo; which refers to that equation in Chap. 3.
</p>
<p>Each chapter contains a set of problems designed to assist the reader in the
</p>
<p>understanding of the textual material, and detailed tutorial solutions are
</p>
<p>provided. Some of these problems require computer assistance, and a set of
</p>
<p>programs has been designed and included with the Web material and dated 1
</p>
<p>January 2013 (Version 5.1). In this context, we are grateful to Dr. Jan Vissser
</p>
<p>of the Technisch Physische Dienst, Delft, Professor Armel Le Bail of Labor-
</p>
<p>atoire Fluorures, Université du Main, LeMans, and Professor A L Spek of the
</p>
<p>University of Utrecht for the continued incorporation of the programs ITO12,
</p>
<p>ESPOIR, and LEPAGE, respectively, in the Program Suite for this book.
</p>
<p>Finally we thank Springer Science + Business Media for inviting this edition
</p>
<p>and bringing it to a state of completion.
</p>
<p>University of Surrey
</p>
<p>Guildford, England Mark Ladd
</p>
<p>Birkbeck College
</p>
<p>London, England, London Rex Palmer
</p>
<p>xii Preface to the Fifth Edition</p>
<p/>
<div class="annotation"><a href="http://extra.springer.com">http://extra.springer.com</a></div>
</div>
<div class="page"><p/>
<p>Disclaimer
</p>
<p>Every effort has been made to ensure the correct functioning of the software
</p>
<p>associated with this book. However, the reader planning to use the software
</p>
<p>should note that, from the legal point of view, there is no warranty, expressed
</p>
<p>or implied, that the programs are free from error or will prove suitable for a
</p>
<p>particular application; by using the software the reader accepts full responsi-
</p>
<p>bility for all the results produced, and the authors and publisher disclaim all
</p>
<p>liability from any consequences arising from the use of the software. The
</p>
<p>software should not be relied upon for solving a problem, the incorrect
</p>
<p>solution of which could result in injury to a person or loss of property. If
</p>
<p>you do use the programs in such a manner, it is at your own risk. The authors
</p>
<p>and publisher disclaim all liability for direct or consequential damages
</p>
<p>resulting from your use of the programs.
</p>
<p>xiii</p>
<p/>
</div>
<div class="page"><p/>
</div>
<div class="page"><p/>
<p>Table of Contents
</p>
<p>Periodic Table . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . ii
</p>
<p>Physical Constants and Other Numerical Data . . . . . . . . . . . . . . . . . . . . . xxxi
</p>
<p>Notation . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .xxxiii
</p>
<p>1 Crystal Morphology and Crystal Symmetry . . . . . . . . . . . . . . . . . . . . 1
1.1 Brief Historical Introduction . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 1
</p>
<p>1.2 The Crystalline State . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 7
</p>
<p>1.2.1 Crystallographic Reference Axes . . . . . . . . . . . . . . . . . . . . 7
</p>
<p>1.2.2 Equation of a Plane . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 7
</p>
<p>1.2.3 Indices of Planes and the Law
</p>
<p>of Rational Intercepts . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 8
</p>
<p>1.2.4 Axial Ratios . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 12
</p>
<p>1.2.5 Zones . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 12
</p>
<p>1.3 Stereographic Projection: Brief Survey . . . . . . . . . . . . . . . . . . . . . . 15
</p>
<p>1.4 External Symmetry of Crystals . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 17
</p>
<p>1.4.1 Two-Dimensional Point Groups . . . . . . . . . . . . . . . . . . . . . . 19
</p>
<p>1.4.2 Three-Dimensional Point Groups . . . . . . . . . . . . . . . . . . . . 22
</p>
<p>1.4.3 Quasicrystals, Buckyballs, and Icosahedral
</p>
<p>Symmetry . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 32
</p>
<p>1.5 Problems . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 39
</p>
<p>References and Bibliography . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 49
</p>
<p>2 Lattices and Space-Group Theory . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 51
2.1 Introduction . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 51
</p>
<p>2.2 Lattices . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 51
</p>
<p>2.2.1 Two-Dimensional Lattices . . . . . . . . . . . . . . . . . . . . . . . . . . . . 52
</p>
<p>2.2.2 Choice of Unit Cell . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 53
</p>
<p>2.2.3 Three-Dimensional Lattices . . . . . . . . . . . . . . . . . . . . . . . . . . 54
</p>
<p>2.3 Families of Planes and Interplanar Spacings . . . . . . . . . . . . . . . . . 62
</p>
<p>2.4 Reciprocal Lattice: Geometrical Treatment . . . . . . . . . . . . . . . . . . 63
</p>
<p>2.5 Unit-Cell Transformations . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 65
</p>
<p>2.5.1 Bravais Unit-Cell Vectors . . . . . . . . . . . . . . . . . . . . . . . . . . . . 65
</p>
<p>2.5.2 Directions (Zone Symbols) . . . . . . . . . . . . . . . . . . . . . . . . . . . 66
</p>
<p>2.5.3 Coordinates of Sites in the Unit Cell . . . . . . . . . . . . . . . . . 67
</p>
<p>2.5.4 Miller Indices . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 67
</p>
<p>2.5.5 Reciprocal Unit-Cell Vectors . . . . . . . . . . . . . . . . . . . . . . . . . 68
</p>
<p>2.6 Rotational Symmetries of Lattices . . . . . . . . . . . . . . . . . . . . . . . . . . . 71
</p>
<p>xv</p>
<p/>
</div>
<div class="page"><p/>
<p>2.7 Space Groups . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 72
</p>
<p>2.7.1 Two-Dimensional Space Groups
</p>
<p>(Plane Groups) . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 73
</p>
<p>2.7.2 Plane Groups Related to 2mm . . . . . . . . . . . . . . . . . . . . . 79
</p>
<p>2.7.3 Three-Dimensional Space Groups . . . . . . . . . . . . . . . . . 81
</p>
<p>2.7.4 Screw Axes . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 84
</p>
<p>2.7.5 Glide Planes . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 87
</p>
<p>2.7.6 Analysis of the Space-Group Symbol . . . . . . . . . . . . . 90
</p>
<p>2.7.7 Orthorhombic Space Groups . . . . . . . . . . . . . . . . . . . . . . . 91
</p>
<p>2.7.8 Relative Orientations of Symmetry
</p>
<p>Elements in Space Groups . . . . . . . . . . . . . . . . . . . . . . . . . 93
</p>
<p>2.7.9 Tetragonal and Hexagonal Space Groups . . . . . . . . . 95
</p>
<p>2.8 Matrix Representation of Symmetry Operations . . . . . . . . . . . 98
</p>
<p>2.8.1 Matrices in Point-Group Symmetry . . . . . . . . . . . . . . . 98
</p>
<p>2.8.2 Matrices in Space-Group Symmetry . . . . . . . . . . . . . . . 100
</p>
<p>2.9 Diffraction Symbols . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 101
</p>
<p>2.10 Some Other Types of Symmetry . . . . . . . . . . . . . . . . . . . . . . . . . . . . 103
</p>
<p>2.10.1 Black-White Symmetry . . . . . . . . . . . . . . . . . . . . . . . . . . . . 103
</p>
<p>2.10.2 Color Symmetry . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 104
</p>
<p>2.11 Problems . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 106
</p>
<p>References . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 109
</p>
<p>3 X-Rays and X-Ray Diffraction . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 111
3.1 Generation and Properties of X-Rays . . . . . . . . . . . . . . . . . . . . . . . 111
</p>
<p>3.1.1 X-Rays and White Radiation . . . . . . . . . . . . . . . . . . . . . . . 111
</p>
<p>3.1.2 Characteristic Radiation . . . . . . . . . . . . . . . . . . . . . . . . . . . . 113
</p>
<p>3.1.3 Absorption of X-Rays . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 114
</p>
<p>3.1.4 Monochromatic Radiation . . . . . . . . . . . . . . . . . . . . . . . . . . 116
</p>
<p>3.1.5 Collimation . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 116
</p>
<p>3.1.6 Synchrotron Sources . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 118
</p>
<p>3.2 X-Ray Scattering . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 121
</p>
<p>3.2.1 Scattering by a Single Electron . . . . . . . . . . . . . . . . . . . . 122
</p>
<p>3.2.2 Scattering by Two or More Electrons . . . . . . . . . . . . . 122
</p>
<p>3.2.3 Waves and Wave Sums . . . . . . . . . . . . . . . . . . . . . . . . . . . . 123
</p>
<p>3.2.4 Coherent and Incoherent Scattering . . . . . . . . . . . . . . . 127
</p>
<p>3.2.5 Scattering by an Atom . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 128
</p>
<p>3.3 Scattering by Regular Arrays of Atoms . . . . . . . . . . . . . . . . . . . . 130
</p>
<p>3.3.1 Laue Equations . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 130
</p>
<p>3.3.2 Bragg Equation . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 132
</p>
<p>3.3.3 Equivalence of the Laue and Bragg
</p>
<p>Equations . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 134
</p>
<p>3.3.4 Further Analysis of the Path Difference . . . . . . . . . . . 135
</p>
<p>3.4 Reciprocal Lattice: Analytical Treatment . . . . . . . . . . . . . . . . . . 135
</p>
<p>3.4.1 Reciprocal Lattice Properties . . . . . . . . . . . . . . . . . . . . . . 137
</p>
<p>3.4.2 Reciprocal Lattice and Reflection
Condition: Ewald Sphere . . . . . . . . . . . . . . . . . . . . . . . . . . . 138
</p>
<p>3.5 Scattering by a Crystal Structure . . . . . . . . . . . . . . . . . . . . . . . . . . . . 139
</p>
<p>3.5.1 Structure Factor Equation . . . . . . . . . . . . . . . . . . . . . . . . . . . 140
</p>
<p>3.6 Using the Structure Factor Equation . . . . . . . . . . . . . . . . . . . . . . . . 140
</p>
<p>3.6.1 Friedel&rsquo;s Law . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 140
</p>
<p>3.6.2 Structure Factor for a Centrosymmetric Crystal . . . 141
</p>
<p>xvi Table of Contents</p>
<p/>
</div>
<div class="page"><p/>
<p>3.7 Limiting Conditions and Systematic Absences . . . . . . . . . . . . . . 142
</p>
<p>3.7.1 Body-Centered Unit Cell . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 142
</p>
<p>3.7.2 Screw Axes and Glide Planes . . . . . . . . . . . . . . . . . . . . . . . . 143
</p>
<p>3.8 Practical Determination of Space Groups
</p>
<p>from Diffraction Data . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 152
</p>
<p>3.8.1 Monoclinic Space Groups . . . . . . . . . . . . . . . . . . . . . . . . . . . . 153
</p>
<p>3.8.2 Orthorhombic Space Groups . . . . . . . . . . . . . . . . . . . . . . . . . 154
</p>
<p>3.8.3 Tetragonal Space Groups . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 155
</p>
<p>3.8.4 Hexagonal Space Groups . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 155
</p>
<p>3.9 Problems . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 155
</p>
<p>References . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 159
</p>
<p>4 Intensities and Intensity Statistics . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 161
4.1 Intensity Expressions and Factors Affecting
</p>
<p>Intensities . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 161
</p>
<p>4.1.1 Polarization and Lorentz Factors . . . . . . . . . . . . . . . . . . . . . 162
</p>
<p>4.1.2 Extinction . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 164
</p>
<p>4.1.3 Absorption Measurement and Correction . . . . . . . . . . . . 165
</p>
<p>4.1.4 Scaling . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 167
</p>
<p>4.1.5 Merging Equivalent Reflections . . . . . . . . . . . . . . . . . . . . . . 167
</p>
<p>4.1.6 Practical Intensity Expression
</p>
<p>and its Standard Deviation . . . . . . . . . . . . . . . . . . . . . . . . . . . . 168
</p>
<p>4.1.7 Scale Factor for Fo . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 169
</p>
<p>4.1.8 Thermal Vibrations and the Temperature
</p>
<p>Factor . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 169
</p>
<p>4.2 Intensity Statistics . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 172
</p>
<p>4.2.1 Determining Scale and Temperature
</p>
<p>Factors . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 172
</p>
<p>4.2.2 Other Aspects of the Wilson Plot . . . . . . . . . . . . . . . . . . . . 175
</p>
<p>4.2.3 Statistics of Reciprocal Space . . . . . . . . . . . . . . . . . . . . . . . . 175
</p>
<p>4.2.4 Acentric and Centric Distributions
</p>
<p>of Structure Factors . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 177
</p>
<p>4.2.5 Normalized Structure Factors . . . . . . . . . . . . . . . . . . . . . . . . 182
</p>
<p>4.3 Problems . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 185
</p>
<p>References . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 186
</p>
<p>5 Examination of Single Crystals: Optical and X-Ray
Diffraction Practice . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 187
5.1 Introduction . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 187
</p>
<p>5.2 Crystal Growing . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 187
</p>
<p>5.2.1 Growing Crystals for X-Ray Diffraction . . . . . . . . . . . . 187
</p>
<p>5.2.2 Crystallization from Solution . . . . . . . . . . . . . . . . . . . . . . . . . 188
</p>
<p>5.2.3 Crystallization by Diffusion . . . . . . . . . . . . . . . . . . . . . . . . . . 188
</p>
<p>5.2.4 Crystallization by Sublimation . . . . . . . . . . . . . . . . . . . . . . . 188
</p>
<p>5.2.5 Other Issues . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 188
</p>
<p>5.3 Optical Techniques . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 189
</p>
<p>5.3.1 Polarized Light . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 189
</p>
<p>5.3.2 Optical Classification of Crystals . . . . . . . . . . . . . . . . . . . . 190
</p>
<p>5.3.3 Uniaxial Crystals . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 190
</p>
<p>Table of Contents xvii</p>
<p/>
</div>
<div class="page"><p/>
<p>5.3.4 Birefringence . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 192
</p>
<p>5.3.5 Biaxial Crystals . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 194
</p>
<p>5.4 Single-Crystal X-Ray Diffraction Techniques:
</p>
<p>Intensity Data Collection . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 197
</p>
<p>5.4.1 Laue Method . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 197
</p>
<p>5.4.2 Symmetry in Laue Photographs . . . . . . . . . . . . . . . . . . . 200
</p>
<p>5.4.3 Laue Method and Synchrotron Radiation . . . . . . . . . 200
</p>
<p>5.4.4 Oscillation Method . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 205
</p>
<p>5.5 Measurement of the Intensities of Diffraction Data . . . . . . . . 208
</p>
<p>5.5.1 Single Counter or Serial Diffractometers . . . . . . . . . 209
</p>
<p>5.6 Single-Crystal X-Ray Diffractometry . . . . . . . . . . . . . . . . . . . . . . . 209
</p>
<p>5.6.1 Instrument Geometry . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 209
</p>
<p>5.6.2 Rotation of the Crystal into a
</p>
<p>Diffracting Position . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 210
</p>
<p>5.6.3 Transformation from Miller Indices
</p>
<p>to Diffractometer Angles . . . . . . . . . . . . . . . . . . . . . . . . . . . 211
</p>
<p>5.6.4 Data Collection . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 211
</p>
<p>5.6.5 Scanning Over a Peak: o/y
</p>
<p>Versus o Scans . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 212
</p>
<p>5.7 Area Detectors (Position-Sensitive Detectors) . . . . . . . . . . . . . 213
</p>
<p>5.7.1 Multiwire Proportional Counter . . . . . . . . . . . . . . . . . . . 213
</p>
<p>5.7.2 FAST Area Detector (Enraf&ndash;Nonius
</p>
<p>FAST) . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 215
</p>
<p>5.7.3 Image Plate . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 215
</p>
<p>5.7.4 Charge-Coupled Device Area Detectors . . . . . . . . . . 217
</p>
<p>5.7.5 The Tiled CCD . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 219
</p>
<p>5.7.6 Charge-Coupled Device Including Tiled
</p>
<p>CCD Versus Image Plate . . . . . . . . . . . . . . . . . . . . . . . . . . 219
</p>
<p>5.7.7 Data Collection Strategies . . . . . . . . . . . . . . . . . . . . . . . . . 219
</p>
<p>5.7.8 The CMOS Detector, Pilatus 1M Detector
</p>
<p>System, and Continuous Rotation . . . . . . . . . . . . . . . . . 221
</p>
<p>5.7.9 Data Processing Software . . . . . . . . . . . . . . . . . . . . . . . . . . 222
</p>
<p>5.7.10 Detectors and Diffractometers . . . . . . . . . . . . . . . . . . . . . 222
</p>
<p>5.7.11 Other Diffractometer Systems . . . . . . . . . . . . . . . . . . . . . 223
</p>
<p>5.8 Monochromators . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 223
</p>
<p>5.8.1 Single-Type Crystal Monochromators . . . . . . . . . . . . 224
</p>
<p>5.8.2 Double-Type Crystal Monochromators . . . . . . . . . . . 224
</p>
<p>5.8.3 Monochromators for Synchrotron
Radiation . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 225
</p>
<p>5.9 Focusing Mirrors . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 225
</p>
<p>5.10 Twinning . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 226
</p>
<p>5.10.1 Morphology of Twinning . . . . . . . . . . . . . . . . . . . . . . . . . . 226
</p>
<p>5.10.2 Twinning and X-Ray Diffraction . . . . . . . . . . . . . . . . . . 228
</p>
<p>5.11 Problems . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 229
</p>
<p>References . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 232
</p>
<p>xviii Table of Contents</p>
<p/>
</div>
<div class="page"><p/>
<p>6 Fourier Series and Fourier Transforms . . . . . . . . . . . . . . . . . . . . . . . . 235
6.1 Image Formation and Focusing . . . . . . . . . . . . . . . . . . . . . . . . . . . . 235
</p>
<p>6.2 Fourier Series . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 236
</p>
<p>6.2.1 Analysis of the Square Wave . . . . . . . . . . . . . . . . . . . . . . 238
</p>
<p>6.2.2 Exponential Forms of Fourier Series . . . . . . . . . . . . . . 240
</p>
<p>6.3 Fourier Series in X-Ray Crystallography . . . . . . . . . . . . . . . . . . 241
</p>
<p>6.3.1 One-Dimensional Function . . . . . . . . . . . . . . . . . . . . . . . . 241
</p>
<p>6.3.2 Two- and Three-Dimensional Functions . . . . . . . . . . 243
</p>
<p>6.3.3 Units of Electron Density . . . . . . . . . . . . . . . . . . . . . . . . . . 245
</p>
<p>6.4 Holes and Atoms . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 245
</p>
<p>6.5 Generalized Fourier Transform . . . . . . . . . . . . . . . . . . . . . . . . . . . . 246
</p>
<p>6.5.1 Fourier Transform of a Molecule . . . . . . . . . . . . . . . . . . 248
</p>
<p>6.5.2 Fourier Transform of a Unit Cell . . . . . . . . . . . . . . . . . . 248
</p>
<p>6.6 Practice with Transforms . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 249
</p>
<p>6.6.1 Optical Diffractometer . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 249
</p>
<p>6.6.2 Single Hole . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 249
</p>
<p>6.6.3 Two or More Holes . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 250
</p>
<p>6.6.4 Change of Origin . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 252
</p>
<p>6.6.5 Systematic Absences . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 252
</p>
<p>6.6.6 Reconstruction of the Image . . . . . . . . . . . . . . . . . . . . . . . 252
</p>
<p>6.6.7 Transforms and Inverse Transforms . . . . . . . . . . . . . . . 255
</p>
<p>6.6.8 Delta Function . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 258
</p>
<p>6.6.9 Weighted Reciprocal Lattice . . . . . . . . . . . . . . . . . . . . . . . 259
</p>
<p>6.7 Some General Properties of Transforms . . . . . . . . . . . . . . . . . . . 261
</p>
<p>6.8 Convolution . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 261
</p>
<p>6.8.1 Convolution and Diffraction . . . . . . . . . . . . . . . . . . . . . . . 261
</p>
<p>6.8.2 Convolution Integral . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 262
</p>
<p>6.8.3 Convolution and Crystal Structure . . . . . . . . . . . . . . . . 264
</p>
<p>6.9 Structure Solution in Brief . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 266
</p>
<p>6.9.1 Use of Heavy Atoms . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 266
</p>
<p>6.9.2 General Phase-Free Transform:
</p>
<p>Patterson Function . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 267
</p>
<p>6.9.3 Sign Relationships . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 268
</p>
<p>6.10 Problems . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 270
</p>
<p>References . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 272
</p>
<p>7 Fourier Techniques in X-Ray Structure
Determination . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 273
7.1 Introduction . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 273
</p>
<p>7.2 Analysis of the Unit-Cell Contents . . . . . . . . . . . . . . . . . . . . . . . . 273
</p>
<p>7.2.1 Papaverine Hydrochloride,
</p>
<p>C20H21NO4&middot;HCl . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 274
</p>
<p>7.2.2 Naphthalene, C10H8 . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 275
</p>
<p>7.2.3 Molecular Symmetry . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 275
</p>
<p>7.2.4 Special Positions . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 276
</p>
<p>7.2.5 Nickel Tungstate, NiWO4 . . . . . . . . . . . . . . . . . . . . . . . . . . 276
</p>
<p>Table of Contents xix</p>
<p/>
</div>
<div class="page"><p/>
<p>7.3 Interpretation of Electron Density Distributions . . . . . . . . . . . . . 278
</p>
<p>7.3.1 Peak Heights and Weights . . . . . . . . . . . . . . . . . . . . . . . . . . . . 279
</p>
<p>7.3.2 Computation and Display of Electron
</p>
<p>Density Distributions . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 279
</p>
<p>7.3.3 Projections . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 279
</p>
<p>7.4 Methods of Solving the Phase Problem . . . . . . . . . . . . . . . . . . . . . . 281
</p>
<p>7.4.1 Number of Reflections in the Data Set . . . . . . . . . . . . . . . 281
</p>
<p>7.4.2 The Patterson Function . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 282
</p>
<p>7.4.3 Positions and Weights of Peaks
</p>
<p>in the Patterson Function . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 285
</p>
<p>7.4.4 Sharpened Patterson Function . . . . . . . . . . . . . . . . . . . . . . . . 287
</p>
<p>7.4.5 Symmetry of the Patterson Function
</p>
<p>for a Crystal of Space Group Pm . . . . . . . . . . . . . . . . . . . . . 288
</p>
<p>7.4.6 Vector Interactions in Other Space
</p>
<p>Groups . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 289
</p>
<p>7.4.7 Examples of the Use of the Patterson Function
</p>
<p>in Solving the Phase Problem . . . . . . . . . . . . . . . . . . . . . . . . 289
</p>
<p>7.4.8 Determination of the Chlorine Atom Positions
</p>
<p>in Papaverine Hydrochloride . . . . . . . . . . . . . . . . . . . . . . . . . 296
</p>
<p>7.4.9 Determination of the Mercury Atom
</p>
<p>Positions in KHg2 . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 296
</p>
<p>7.5 Heavy-Atom Method and Partial Fourier Synthesis . . . . . . . . . 301
</p>
<p>7.5.1 Reliability Factor . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 303
</p>
<p>7.5.2 Pseudosymmetry in Electron Density Maps . . . . . . . . . 308
</p>
<p>7.5.3 Successive Fourier Refinement . . . . . . . . . . . . . . . . . . . . . . . 309
</p>
<p>7.5.4 Difference-Fourier Synthesis . . . . . . . . . . . . . . . . . . . . . . . . . 309
</p>
<p>7.5.5 Limitations of the Heavy-Atom Method . . . . . . . . . . . . . 310
</p>
<p>7.5.6 Patterson Selection . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 310
</p>
<p>7.5.7 Isomorphous Replacement . . . . . . . . . . . . . . . . . . . . . . . . . . . . 312
</p>
<p>7.5.8 Further Details of the Isomorphous
</p>
<p>Replacement Phasing Procedure . . . . . . . . . . . . . . . . . . . . . 319
</p>
<p>7.6 Anomalous Scattering . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 325
</p>
<p>7.6.1 The Flack x Parameter . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 326
</p>
<p>7.6.2 Effect of Anomalous Scattering
</p>
<p>on the Symmetry of Diffraction Patterns . . . . . . . . . . . . 330
</p>
<p>7.6.3 Form of the Structure Factor for a Structure
</p>
<p>Composed of Heavy-Atom Anomalous
</p>
<p>Scattering Species . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 332
</p>
<p>7.6.4 Phasing by Use of Anomalous Scattering . . . . . . . . . . . 334
</p>
<p>7.6.5 Resolution of the Phase Problem for Proteins
</p>
<p>Using Anomalous Scattering Measurements
</p>
<p>(SIRAS Method) . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 335
</p>
<p>7.6.6 Protein Phasing Using the Multiple-Wavelength
</p>
<p>Anomalous Dispersion Technique (MAD) with
</p>
<p>Synchrotron Radiation (SR) . . . . . . . . . . . . . . . . . . . . . . . . . . 337
</p>
<p>xx Table of Contents</p>
<p/>
</div>
<div class="page"><p/>
<p>7.7 Charge flipping . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 338
</p>
<p>7.8 Location of Hydrogen Atoms . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 339
</p>
<p>7.9 Problems . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 340
</p>
<p>References . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 347
</p>
<p>8 Direct Methods and Refinement . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 351
8.1 Introduction . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 351
</p>
<p>8.2 Direct Methods of Phase Determination . . . . . . . . . . . . . . . . . . . . . 351
</p>
<p>8.2.1 Normalized Structure Factors . . . . . . . . . . . . . . . . . . . . . . . 351
</p>
<p>8.2.2 Structure Invariants and Origin-Fixing
</p>
<p>Reflections . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 352
</p>
<p>8.2.3 Sign Determination: Centrosymmetric
</p>
<p>Crystals . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 355
</p>
<p>8.2.4 Amplitude Symmetry and Phase Symmetry . . . . . . . 358
</p>
<p>8.2.5
P
</p>
<p>2-Listing . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 358
</p>
<p>8.2.6 Symbolic-Addition Procedure: Example . . . . . . . . . . . 359
</p>
<p>8.2.7 Calculation of E Maps . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 360
</p>
<p>8.2.8 Phase Determination: Non-centrosymmetric
</p>
<p>Crystals . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 361
</p>
<p>8.2.9 Enantiomorph Selection . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 367
</p>
<p>8.2.10 Phase Determination in Space Group P21 . . . . . . . . . . 368
</p>
<p>8.2.11 Advantages and Disadvantages
</p>
<p>of Symbolic Addition . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 371
</p>
<p>8.2.12 Signs of Trouble, and Past Remedies
</p>
<p>When the Structure Failed to Solve . . . . . . . . . . . . . . . . . 372
</p>
<p>8.2.13 Triplets, Quartets, and the SHELX
</p>
<p>Program Strategy . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 372
</p>
<p>8.2.14 The SHELX Computer Program System . . . . . . . . . . . 374
</p>
<p>8.2.15 The WinGX Program System . . . . . . . . . . . . . . . . . . . . . . . 375
</p>
<p>8.2.16 Direct Methods in the Program SHELX-97
</p>
<p>for Small Molecules . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 375
</p>
<p>8.2.17 Example of a SHELX-97 Structure Solution:
</p>
<p>Crystal Code Name BW202W92(R) . . . . . . . . . . . . . . . . 377
</p>
<p>8.3 Patterson Search Methods . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 380
</p>
<p>8.3.1 General Comments for Small Molecules
</p>
<p>and Macromolecules . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 381
</p>
<p>8.3.2 Intramolecular Interatomic Vectors
</p>
<p>and Molecular Orientation . . . . . . . . . . . . . . . . . . . . . . . . . . 382
</p>
<p>8.3.3 Intermolecular Interatomic Vectors:
</p>
<p>Translation Stage of MR . . . . . . . . . . . . . . . . . . . . . . . . . . . . 384
</p>
<p>8.3.4 Crystal Packing and Refinement
</p>
<p>of the Structure . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 385
</p>
<p>8.3.5 Patterson Search Methods for Small
</p>
<p>Molecules . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 386
</p>
<p>8.3.6 The Program PATSEE . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 387
</p>
<p>8.3.7 Examples of Structure Solution
</p>
<p>Using PATSEE . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 388
</p>
<p>8.3.8 Shake and Bake . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 399
</p>
<p>Table of Contents xxi</p>
<p/>
</div>
<div class="page"><p/>
<p>8.4 Least-Squares Refinement . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 400
</p>
<p>8.4.1 Unit-Cell Dimensions . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 401
</p>
<p>8.4.2 Least-Squares Parameters . . . . . . . . . . . . . . . . . . . . . . . . . . 401
</p>
<p>8.4.3 Theory of Least-Squares Refinement
</p>
<p>and Strategies to Use . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 405
</p>
<p>8.4.4 Least-Squares Refinement Against Fo
2 . . . . . . . . . . . . 407
</p>
<p>8.4.5 Constraints and Restraints . . . . . . . . . . . . . . . . . . . . . . . . . 408
</p>
<p>8.5 Molecular Geometry . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 408
</p>
<p>8.5.1 Bond Lengths and Angles . . . . . . . . . . . . . . . . . . . . . . . . . 408
</p>
<p>8.5.2 Torsion Angles . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 411
</p>
<p>8.5.3 Conformational Analysis . . . . . . . . . . . . . . . . . . . . . . . . . . 412
</p>
<p>8.5.4 Mean Planes . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 414
</p>
<p>8.6 Precision . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 415
</p>
<p>8.7 Correctness of a Structure Analysis . . . . . . . . . . . . . . . . . . . . . . . . . 416
</p>
<p>8.7.1 Databases . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 417
</p>
<p>8.8 Limitations of X-Ray Structure Analysis . . . . . . . . . . . . . . . . . . . 419
</p>
<p>8.9 Disorder in Single Crystals . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 419
</p>
<p>8.10 Computer Prediction of Crystal Structures . . . . . . . . . . . . . . . . . 422
</p>
<p>8.10.1 Crystal Structure of 5-Azauracil . . . . . . . . . . . . . . . . . . . 422
</p>
<p>8.10.2 Developments in Computer Crystal Structure
</p>
<p>Prediction . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 425
</p>
<p>8.11 Blind Structure Prediction of the Flexible
</p>
<p>Molecule 1-Benzyl-1H-Tetrazole . . . . . . . . . . . . . . . . . . . . . . . . . . . 426
</p>
<p>8.12 Problems . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 433
</p>
<p>References . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 435
</p>
<p>9 Examples of Crystal Structure Determination . . . . . . . . . . . . . . . . . . 439
9.1 Introduction . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 439
</p>
<p>9.2 Crystal Structure of 2-Bromobenzo[b]
</p>
<p>Indeno[1,2-e] Pyran . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 439
</p>
<p>9.2.1 Preliminary Physical and X-Ray
</p>
<p>Measurements . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 439
</p>
<p>9.2.2 Intensity Measurement and Correction . . . . . . . . . . . . 444
</p>
<p>9.2.3 Structure Analysis in the xz Projection . . . . . . . . . . . . 446
</p>
<p>9.2.4 Three-Dimensional Structure
</p>
<p>Determination . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 447
</p>
<p>9.2.5 Refinement . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 449
</p>
<p>9.2.6 Molecular Geometry . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 451
</p>
<p>9.3 Crystal Structure of Potassium
</p>
<p>2-Hydroxy-3,4-Dioxocyclobut-1-ene-1-Olate
</p>
<p>Monohydrate (KHSQ) . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 455
</p>
<p>9.3.1 Preliminary X-Ray and Physical
</p>
<p>Measurements . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 455
</p>
<p>9.3.2 Intensity Measurement and Correction . . . . . . . . . . . . 456
</p>
<p>9.3.3
P
</p>
<p>2-Listing . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 456
</p>
<p>9.3.4 Specifying the Origin . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 457
</p>
<p>9.3.5 Sign Determination . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 458
</p>
<p>xxii Table of Contents</p>
<p/>
</div>
<div class="page"><p/>
<p>9.3.6 The E Map . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 459
</p>
<p>9.3.7 Completion and Refinement
</p>
<p>of the Structure . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 462
</p>
<p>9.4 Crystal and Molecular Structure and Absolute
</p>
<p>Configuration of 3b-Acetoxy-6,7-Epidithio-19-
</p>
<p>Norlanosta-5,7,9,11-Tetraene . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 465
</p>
<p>9.4.1 Preparation and Preliminary Optical
</p>
<p>and X-Ray Examinations . . . . . . . . . . . . . . . . . . . . . . . . . . 466
</p>
<p>9.4.2 X-Ray Measurement of the Unit-Cell
</p>
<p>Dimensions and Intensities . . . . . . . . . . . . . . . . . . . . . . . . 466
</p>
<p>9.4.3 Structure Determination and Refinement . . . . . . . . . 468
</p>
<p>9.4.4 Absolute Configuration . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 468
</p>
<p>9.5 Discussion of the Structure . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 468
</p>
<p>9.6 Some Remarks on X-Ray Structure
</p>
<p>Determination . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 470
</p>
<p>9.7 Biomolecular Modeling: Bioinformatics . . . . . . . . . . . . . . . . . . . 471
</p>
<p>9.8 Docking Oligomycin into ATP Synthase:
</p>
<p>Ligand and Receptor . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 471
</p>
<p>9.8.1 Why Modeling Studies? . . . . . . . . . . . . . . . . . . . . . . . . . . . 471
</p>
<p>9.9 X-Ray Structures and Absolute Configurations
</p>
<p>of the Antibiotics Oligomycins A,B, and C:
</p>
<p>Inhibitors of ATP Synthase . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 472
</p>
<p>9.9.1 Summary . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 473
</p>
<p>9.9.2 Background . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 474
</p>
<p>9.9.3 Experimental . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 474
</p>
<p>9.9.4 Structure Determination and Refinement . . . . . . . . . 475
</p>
<p>9.9.5 Results . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 475
</p>
<p>9.9.6 Discussion . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 475
</p>
<p>9.9.7 Conformational Variations
</p>
<p>in the Macrocyclic Structures . . . . . . . . . . . . . . . . . . . . . 478
</p>
<p>9.10 Structure of ATP Synthase (ATPase): The Receptor . . . . . . 480
</p>
<p>9.11 Docking Oligomycin into ATPase . . . . . . . . . . . . . . . . . . . . . . . . . . 481
</p>
<p>9.11.1 ATP Synthase FO Model . . . . . . . . . . . . . . . . . . . . . . . . . . 481
</p>
<p>9.11.2 Homology Modeling . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 482
</p>
<p>9.11.3 Refining the Model: Energy Minimization . . . . . . . 482
</p>
<p>9.11.4 Creation of a Pocket for Docking
</p>
<p>Oligomycin into the ATP Synthase FO . . . . . . . . . . . 483
</p>
<p>9.12 Problems . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 484
</p>
<p>References . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 487
</p>
<p>10 Proteins and Macromolecular X-Ray Analysis . . . . . . . . . . . . . . . . . 489
10.1 Introduction . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 489
</p>
<p>10.1.1 What Is a Protein? . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 489
</p>
<p>10.2 Crystallization of Proteins and Complexes
</p>
<p>for X-Ray Analysis . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 491
</p>
<p>10.2.1 Introduction . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 491
</p>
<p>10.2.2 Crystallization Conditions
</p>
<p>for Macromolecules . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 492
</p>
<p>Table of Contents xxiii</p>
<p/>
</div>
<div class="page"><p/>
<p>10.2.3 Properties of Protein Crystals . . . . . . . . . . . . . . . . . . . . 492
</p>
<p>10.2.4 Crystallization of Proteins . . . . . . . . . . . . . . . . . . . . . . . . 492
</p>
<p>10.2.5 Molecular Purity . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 493
</p>
<p>10.2.6 Practical Considerations . . . . . . . . . . . . . . . . . . . . . . . . . . 493
</p>
<p>10.2.7 Batch Crystallization . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 493
</p>
<p>10.2.8 Microbatch Screening . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 493
</p>
<p>10.2.9 Vapor Diffusion Techniques . . . . . . . . . . . . . . . . . . . . . 494
</p>
<p>10.2.10 Co-crystallization . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 496
</p>
<p>10.2.11 How to Improve the Crystals . . . . . . . . . . . . . . . . . . . . . 496
</p>
<p>10.2.12 Heavy-Atom Derivatives for MIR . . . . . . . . . . . . . . . 497
</p>
<p>10.2.13 Protein Complex Crystals with Small Molecules 498
</p>
<p>10.3 Crystal Mounting for X-Ray Data Collection . . . . . . . . . . . . . . 499
</p>
<p>10.3.1 Mounting at Room Temperature . . . . . . . . . . . . . . . . . 499
</p>
<p>10.3.2 Cryo-Crystallography . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 499
</p>
<p>10.4 Macromolecular Crystallography . . . . . . . . . . . . . . . . . . . . . . . . . . . 501
</p>
<p>10.4.1 Space Groups . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 501
</p>
<p>10.4.2 X-Ray Diffraction from Macromolecular
</p>
<p>Crystals . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 501
</p>
<p>10.4.3 Recording X-Ray Diffraction
</p>
<p>from Macromolecular Crystals . . . . . . . . . . . . . . . . . . . 503
</p>
<p>10.4.4 Measurement of X-Ray Diffraction
</p>
<p>from Macromolecular Crystals . . . . . . . . . . . . . . . . . . . 505
</p>
<p>10.4.5 Problems with Data Collection
</p>
<p>and Suggested Cures . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 508
</p>
<p>10.4.6 Preliminary Structure Determination:
</p>
<p>Unit Cell and Symmetry . . . . . . . . . . . . . . . . . . . . . . . . . . 509
</p>
<p>10.4.7 Ricin Agglutinin . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 509
</p>
<p>10.5 Types of Fourier Synthesis for Protein Analysis . . . . . . . . . . . 512
</p>
<p>10.5.1 Reconstruction of the Molecular Structure . . . . . . 512
</p>
<p>10.5.2 Difference Electron Density . . . . . . . . . . . . . . . . . . . . . . 513
</p>
<p>10.5.3 The 2Fo&eth;hkl&THORN;�jFc&eth;hkl&THORN;j Map . . . . . . . . . . . . . . . . . . . . . 514
10.6 Determination of the Phases for Protein Crystals . . . . . . . . . . 514
</p>
<p>10.6.1 Introduction . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 514
</p>
<p>10.6.2 Isomorphous Replacement (MIR) . . . . . . . . . . . . . . . . 514
</p>
<p>10.6.3 Preparation and Screening
</p>
<p>of Heavy-Atom Derivatives . . . . . . . . . . . . . . . . . . . . . . 515
</p>
<p>10.6.4 Molecular Replacement (MR) . . . . . . . . . . . . . . . . . . . . 516
</p>
<p>10.6.5 Example of a Self-Rotation Function:
</p>
<p>Ricin Agglutinin . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 520
</p>
<p>10.6.6 Molecular Replacement in Practice . . . . . . . . . . . . . . 521
</p>
<p>10.6.7 Application of the AmoRe Algorithms
</p>
<p>to Ricin Agglutinin . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 525
</p>
<p>10.7 SIRAS and MAD Phasing . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 526
</p>
<p>xxiv Table of Contents</p>
<p/>
</div>
<div class="page"><p/>
<p>10.8 Use of Phase Information and Density
</p>
<p>Modification . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 528
</p>
<p>10.8.1 Properties of r(xyz) for Proteins . . . . . . . . . . . . . . . . 528
</p>
<p>10.8.2 Programs for Density Modification . . . . . . . . . . . . . 528
</p>
<p>10.8.3 Preparing to Refine the Structure . . . . . . . . . . . . . . . 529
</p>
<p>10.9 Macromolecular Structure Refinement
</p>
<p>and Solvent and Ligand Fitting . . . . . . . . . . . . . . . . . . . . . . . . . . . . 531
</p>
<p>10.9.1 Refinement Techniques . . . . . . . . . . . . . . . . . . . . . . . . . . 531
</p>
<p>10.9.2 Simulated Annealing . . . . . . . . . . . . . . . . . . . . . . . . . . . . 533
</p>
<p>10.9.3 Least-Squares Refinement: Constrained,
Restrained and Other Protocols . . . . . . . . . . . . . . . . . 534
</p>
<p>10.10 Structure Validation: Final Checks . . . . . . . . . . . . . . . . . . . . . . . . 537
</p>
<p>10.10.1 R-Factors . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 537
</p>
<p>10.10.2 Evaluation of Errors . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 539
</p>
<p>10.11 Geometry Validation: Final Checks . . . . . . . . . . . . . . . . . . . . . . . 539
</p>
<p>10.11.1 Bond Lengths, Bond Angles,
</p>
<p>Planarity, and Chirality . . . . . . . . . . . . . . . . . . . . . . . . . . 539
</p>
<p>10.11.2 Conformation . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 540
</p>
<p>10.12 Humidity Control and the Use of Cryoprotectants
</p>
<p>in Protein Crystallography . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 545
</p>
<p>10.13 Problems . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 545
</p>
<p>References . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 546
</p>
<p>11 Neutron Diffraction from Single Crystals . . . . . . . . . . . . . . . . . . . . . . . 549
11.1 Introduction . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 549
</p>
<p>11.1.1 Refinement of Hydrogen Atom Positions . . . . . . 550
</p>
<p>11.2 Neutrons, Neutron Sources, and Data Collection . . . . . . . . . 551
</p>
<p>11.2.1 Neutrons . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 551
</p>
<p>11.2.2 Neutron Sources . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 551
</p>
<p>11.2.3 Neutron Data Collection . . . . . . . . . . . . . . . . . . . . . . . . . 551
</p>
<p>11.2.4 Thermal Neutrons . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 553
</p>
<p>11.3 Neutron Scattering . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 553
</p>
<p>11.3.1 Neutron Scattering Lengths . . . . . . . . . . . . . . . . . . . . . 554
</p>
<p>11.4 Experimental Neutron Diffraction Data Collection . . . . . . . 554
</p>
<p>11.4.1 LADI-III and VIVALDI at ILL, Grenoble . . . . . 555
</p>
<p>11.4.2 Oak Ridge National Laboratory
</p>
<p>(ORNL) . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 556
</p>
<p>11.4.3 Other Neutron Sources . . . . . . . . . . . . . . . . . . . . . . . . . . 559
</p>
<p>11.5 Deuteration and Perdeuteration . . . . . . . . . . . . . . . . . . . . . . . . . . . . 559
</p>
<p>11.6 Examples of Structure Determination
</p>
<p>by Neutron Crystallography . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 560
</p>
<p>11.7 X-Ray and Neutron Structure of 1,8-(3,6,9-
</p>
<p>Trioxaundecane-1,11-diyldioxy)-9,10-
</p>
<p>dihydro-10-10 dimethylanthracene-9-ol . . . . . . . . . . . . . . . . . . . 560
</p>
<p>11.7.1 Experimental . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 561
</p>
<p>11.7.2 Structure Analysis and Refinement . . . . . . . . . . . . . 564
</p>
<p>11.7.3 Discussion of the Structure . . . . . . . . . . . . . . . . . . . . . . 565
</p>
<p>11.7.4 Hydrogen Bonding . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 566
</p>
<p>11.8 The Pointless Program in CCP4 . . . . . . . . . . . . . . . . . . . . . . . . . . . 567
</p>
<p>Table of Contents xxv</p>
<p/>
</div>
<div class="page"><p/>
<p>11.9 Determination of the Positions of the Deuterium
</p>
<p>Atoms of the Bound Water Molecules in the Lectin
</p>
<p>Protein Concanavalin A by Neutron Laue
</p>
<p>Crystallography . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 567
</p>
<p>11.9.1 Introduction . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 567
</p>
<p>11.9.2 Deuteration of the Concanavalin
</p>
<p>A Crystals . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 568
</p>
<p>11.9.3 Data Collection and Analysis . . . . . . . . . . . . . . . . . . . 568
</p>
<p>11.9.4 X-Ray Model Refinement . . . . . . . . . . . . . . . . . . . . . . . 569
</p>
<p>11.9.5 Neutron Structure Refinement . . . . . . . . . . . . . . . . . . 569
</p>
<p>11.9.6 The Bound Water Structure . . . . . . . . . . . . . . . . . . . . . 570
</p>
<p>11.9.7 The Metal Sites . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 570
</p>
<p>11.9.8 The Saccharide Binding Site . . . . . . . . . . . . . . . . . . . . 571
</p>
<p>11.9.9 Conclusion . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 572
</p>
<p>11.10 The Neutron Structure of the Formyl Peptide
</p>
<p>Receptor Antagonist Cyclosporin H (CsH)
</p>
<p>Unambiguously Determines the Solvent
</p>
<p>and Hydrogen Bonding Structure for Crystal
</p>
<p>Form II . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 574
</p>
<p>11.10.1 Introduction . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 574
</p>
<p>11.10.2 Experimental . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 576
</p>
<p>11.10.3 Structure Refinement . . . . . . . . . . . . . . . . . . . . . . . . . . . . 576
</p>
<p>11.10.4 Description of the Neutron Structure
</p>
<p>and Comparison with the X-Ray Structure . . . . . 579
</p>
<p>11.10.5 Conclusion . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 580
</p>
<p>11.11 Problems . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 582
</p>
<p>References . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 583
</p>
<p>12 Powder Diffraction . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 585
12.1 Introduction . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 585
</p>
<p>12.1.1 Identification . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 585
</p>
<p>12.1.2 Crystallinity: Size and Strain Broadening . . . . . . 585
</p>
<p>12.1.3 Unit-Cell Parameters . . . . . . . . . . . . . . . . . . . . . . . . . . . . 586
</p>
<p>12.1.4 Expansion Properties . . . . . . . . . . . . . . . . . . . . . . . . . . . . 586
</p>
<p>12.1.5 Phase Transitions and Alloy Systems . . . . . . . . . . 586
</p>
<p>12.2 Crystal Structure Analysis with Powders . . . . . . . . . . . . . . . . . . 586
</p>
<p>12.2.1 Crystal Structure Determination Scheme . . . . . . . 586
</p>
<p>12.3 Basis of the Powder Method . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 588
</p>
<p>12.4 Data Collection . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 590
</p>
<p>12.4.1 Guinier-Type Cameras . . . . . . . . . . . . . . . . . . . . . . . . . . 590
</p>
<p>12.4.2 Image Plate Camera . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 592
</p>
<p>12.4.3 Powder Diffractometers . . . . . . . . . . . . . . . . . . . . . . . . . 593
</p>
<p>12.4.4 Diffractometry at a Neutron Source . . . . . . . . . . . . 594
</p>
<p>12.5 Indexing Powder Patterns . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 598
</p>
<p>12.5.1 General Indexing . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 599
</p>
<p>12.5.2 Reduced and Conventional Unit Cells . . . . . . . . . 601
</p>
<p>12.5.3 Computer Indexing of the
</p>
<p>Diffraction Pattern . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 602
</p>
<p>xxvi Table of Contents</p>
<p/>
</div>
<div class="page"><p/>
<p>12.6 Extracting Integrated Intensities from
</p>
<p>a Powder Pattern . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 605
</p>
<p>12.7 The Rietveld Procedure . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 605
</p>
<p>12.7.1 The Le Bail Method . . . . . . . . . . . . . . . . . . . . . . . . . . . . 607
</p>
<p>12.7.2 The Pawley Method . . . . . . . . . . . . . . . . . . . . . . . . . . . . 608
</p>
<p>12.8 Examples of Solved Structures . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 608
</p>
<p>12.8.1 Traditional Methods . . . . . . . . . . . . . . . . . . . . . . . . . . . . 609
</p>
<p>12.8.2 SIR Program System . . . . . . . . . . . . . . . . . . . . . . . . . . . 611
</p>
<p>12.8.3 EXPO Program System . . . . . . . . . . . . . . . . . . . . . . . . 612
</p>
<p>12.9 Direct-Space Methods . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 613
</p>
<p>12.9.1 Zeolites and the FOCUS Algorithm . . . . . . . . . . . 614
</p>
<p>12.9.2 Zinc&ndash;Silicate Complex VIP-9 . . . . . . . . . . . . . . . . . 614
</p>
<p>12.10 Monte Carlo Method . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 617
</p>
<p>12.10.1 Simulated Annealing . . . . . . . . . . . . . . . . . . . . . . . . . . . 621
</p>
<p>12.11 ESPOIR Program System . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 621
</p>
<p>12.12 Powder Diffraction with Proteins . . . . . . . . . . . . . . . . . . . . . . . . . . 623
</p>
<p>12.12.1 T3R3 Zinc&ndash;Insulin Complex . . . . . . . . . . . . . . . . . . . 623
</p>
<p>12.13 Maximum Entropy in Crystal Structure Analysis . . . . . . . . . 624
</p>
<p>12.13.1 Most Probable Distribution . . . . . . . . . . . . . . . . . . . . 624
</p>
<p>12.13.2 Electron Density Map . . . . . . . . . . . . . . . . . . . . . . . . . . 625
</p>
<p>12.14 Log-Likelihood Gain in the Phase Problem . . . . . . . . . . . . . . . 626
</p>
<p>12.14.1 Basis Set and Expansion of Reflections . . . . . . . 626
</p>
<p>12.14.2 Log-Likelihood Gain . . . . . . . . . . . . . . . . . . . . . . . . . . . 627
</p>
<p>12.14.3 Centroid Maps . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 627
</p>
<p>12.15 Genetic Algorithms . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 628
</p>
<p>12.16 Energy Minimization Techniques . . . . . . . . . . . . . . . . . . . . . . . . . . 628
</p>
<p>12.17 Concluding Remarks . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 629
</p>
<p>12.18 Problems . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 630
</p>
<p>References . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 632
</p>
<p>13 Computer-Aided Crystallography . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 635
13.1 Introduction . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 635
</p>
<p>13.1.1 Collaborative Computational Projects . . . . . . . . . 635
</p>
<p>13.1.2 Structure of the Web Program Packages . . . . . . 636
</p>
<p>13.2 Derivation of Point Groups (EULR) . . . . . . . . . . . . . . . . . . . . . . . 636
</p>
<p>13.3 Point-Group Recognition (SYMM) . . . . . . . . . . . . . . . . . . . . . . . . 637
</p>
<p>13.4 Structure Determination Simulation (XRAY) . . . . . . . . . . . . . 640
</p>
<p>13.4.1 Patterson Function . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 641
</p>
<p>13.4.2 Superposition Function . . . . . . . . . . . . . . . . . . . . . . . . . 642
</p>
<p>13.4.3 Structure Factor Calculation . . . . . . . . . . . . . . . . . . . 642
</p>
<p>13.4.4 Least-Squares Refinement . . . . . . . . . . . . . . . . . . . . . 642
</p>
<p>13.4.5 Electron Density Maps . . . . . . . . . . . . . . . . . . . . . . . . . 643
</p>
<p>13.4.6 Direct Methods: Calculation of jEj Values . . . 643
13.4.7 Calculation of E Maps . . . . . . . . . . . . . . . . . . . . . . . . . 644
</p>
<p>13.4.8 Bond Lengths and Bond Angles . . . . . . . . . . . . . . . 645
</p>
<p>13.4.9 Scale and Temperature Factors
</p>
<p>by Wilson&rsquo;s Method . . . . . . . . . . . . . . . . . . . . . . . . . . . . 645
</p>
<p>13.4.10 jEj Values Calculated from
the Structure . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 645
</p>
<p>Table of Contents xxvii</p>
<p/>
</div>
<div class="page"><p/>
<p>13.5 Crystal Structure Analysis Problems . . . . . . . . . . . . . . . . . . . . . . . . 646
</p>
<p>13.5.1 Ni o-Phenanthroline Complex (NIOP) . . . . . . . . . . . 647
</p>
<p>13.5.2 2-Amino-4,6-dichloropyrimidine (CL1P) . . . . . . . 648
</p>
<p>13.5.3 2-Amino-4-methyl-6-chloropyrimidine
</p>
<p>(CL2P) . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 648
</p>
<p>13.5.4 m-Tolidine Dihydrochloride (MTOL) . . . . . . . . . . . 649
</p>
<p>13.5.5 Nitroguanidine (NO2G) . . . . . . . . . . . . . . . . . . . . . . . . . . . 649
</p>
<p>13.5.6 Bis(6-sulfanyloxy-1,3,5-triazin-2(1H)-one)
</p>
<p>(BSTO) . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 650
</p>
<p>13.5.7 2-S-methylthiouracil (SMTX and SMTY) . . . . . . . 650
</p>
<p>13.6 General Crystal Structure and Other Programs . . . . . . . . . . . . . 650
</p>
<p>13.6.1 One-Dimensional Fourier Summation
</p>
<p>(FOUR1D) . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 650
</p>
<p>13.6.2 Two-Dimensional Fourier Summation
</p>
<p>(FOUR2D) . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 650
</p>
<p>13.6.3 One-Dimensional Fourier Transform
</p>
<p>(TRANS1) . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 651
</p>
<p>13.6.4 Reciprocal Unit Cell (RECIP) . . . . . . . . . . . . . . . . . . . . 651
</p>
<p>13.6.5 Molecular Geometry (MOLGOM) . . . . . . . . . . . . . . . 651
</p>
<p>13.6.6 Internal and Cartesian Coordinates
</p>
<p>(INTXYZ) . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 652
</p>
<p>13.6.7 Linear Least Squares (LSLI) . . . . . . . . . . . . . . . . . . . . . 653
</p>
<p>13.6.8 Matrix Operations (MATOPS) . . . . . . . . . . . . . . . . . . . 653
</p>
<p>13.6.9 Q Values (QVALS) . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 653
</p>
<p>13.6.10 Le Page Unit-Cell Reduction
</p>
<p>(LEPAGE) . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 654
</p>
<p>13.6.11 Zone symbols/Miller indices (ZONE) . . . . . . . . . . . 654
</p>
<p>13.7 Automatic Powder Indexing: ITO12 . . . . . . . . . . . . . . . . . . . . . . . . 654
</p>
<p>13.8 Automatic Powder Structure Solving: ESPOIR . . . . . . . . . . . . 655
</p>
<p>13.8.1 Aragonite . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 655
</p>
<p>13.8.2 a-Alumina (Corundum) . . . . . . . . . . . . . . . . . . . . . . . . . . . 656
</p>
<p>13.9 Problems . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 658
</p>
<p>References . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 658
</p>
<p>Appendix A: Stereoviews and Crystal Models . . . . . . . . . . . . . . . . . . . . . . . 659
A.1 Stereoviews . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 659
</p>
<p>A.2 Model of a Tetragonal Crystal . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 659
</p>
<p>Appendix B: Schönflies&rsquo; Symmetry Notation . . . . . . . . . . . . . . . . . . . . . . . . . 663
B.1 Alternating Axis of Symmetry . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 663
</p>
<p>B.2 Symmetry Notations . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 663
</p>
<p>Appendix C: Cartesian Coordinates . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 665
C.1 Cartesian to Crystallographic Transformation
</p>
<p>and Its Inverse . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 665
</p>
<p>Appendix D: Crystallographic Software . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 669
D.1 Single Crystal Suites . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 669
</p>
<p>D.2 Single Crystal Structure Solution Programs . . . . . . . . . . . . . . . 670
</p>
<p>xxviii Table of Contents</p>
<p/>
</div>
<div class="page"><p/>
<p>D.3 Single Crystal Twinning Software . . . . . . . . . . . . . . . . . . . . . . . . . . 670
</p>
<p>D.4 Freestanding Structure Visualization Software . . . . . . . . . . . . 670
</p>
<p>D.5 Powder Diffraction Data: Powder Indexing
</p>
<p>Suites (Dedicated and Other) . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 671
</p>
<p>D.6 Powder Pattern Decomposition . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 671
</p>
<p>D.7 Structure Solution from Powder
</p>
<p>Diffraction Data . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 671
</p>
<p>D.8 Software for Macromolecular
</p>
<p>Crystallography . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 672
</p>
<p>D.8.1 Data Processing . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 672
</p>
<p>D.8.2 Fourier and Structure
</p>
<p>Factor Calculations . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 672
</p>
<p>D.8.3 Molecular Replacement . . . . . . . . . . . . . . . . . . . . . . . . . . . 672
</p>
<p>D.8.4 Schematic Structure Plots . . . . . . . . . . . . . . . . . . . . . . . . . 673
</p>
<p>D.8.5 Software for Packing, Molecular Geometry,
</p>
<p>Validation and Deposition . . . . . . . . . . . . . . . . . . . . . . . . 673
</p>
<p>D.8.6 Software for Graphics
</p>
<p>and Model Building . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 673
</p>
<p>D.8.7 Software for Molecular Graphics
</p>
<p>and Display . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 673
</p>
<p>D.8.8 Software for Refinement . . . . . . . . . . . . . . . . . . . . . . . . . . 674
</p>
<p>D.8.9 Software for Molecular Dynamics
</p>
<p>and Energy Minimization . . . . . . . . . . . . . . . . . . . . . . . . . 674
</p>
<p>D.8.10 Data Bases . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 674
</p>
<p>D.8.11 Synchrotron Web Page . . . . . . . . . . . . . . . . . . . . . . . . . . . . 675
</p>
<p>D.9 Bioinformatics . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 675
</p>
<p>D.9.1 Molecular Modelling Software . . . . . . . . . . . . . . . . . . . 675
</p>
<p>D.9.2 External Links . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 676
</p>
<p>D.9.3 Useful Homepages . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 677
</p>
<p>Appendix E: Structure Invariants, Structure Seminvariants,
Origin and Enantiomorph Specifications . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 679
</p>
<p>E.1 Structure Invariants . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 679
</p>
<p>E.2 Structure Seminvariants . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 681
</p>
<p>E.2.1 Difference Between Structure
</p>
<p>Invariant and Structure Seminvariant . . . . . . . . . . . . 682
</p>
<p>E.3 Origin Specification . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 682
</p>
<p>E.4 Choice of Enantiomorph . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 682
</p>
<p>Tutorial Solutions . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 685
</p>
<p>Index . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 737
</p>
<p>Table of Contents xxix</p>
<p/>
</div>
<div class="page"><p/>
</div>
<div class="page"><p/>
<p>Physical Constants and Other
Numerical Data
</p>
<p>Atomic mass unit mu 1.6605 � 10�27 kg
Avogadro constant L 6.0221 � 1023 mol�1
Bohr radius for hydrogen a0 5.2918 � 10�11 m
Boltzmann constant k 1.3806 � 10�23 J K�1
Elementary charge e 1.6022 � 10�19 C
Permittivity of a vacuum e 8.8542 � 10�12 F m�1
Planck constant h 6.6261 � 10�34 J s
Rest mass of the electron me 9.1094 � 10�31 kg
Rest mass of the neutron mn 1.6749 � 10�27 kg
Rest mass of the proton mp 1.6726 � 10�27 kg
Speed of light in a vacuum c 2.9979 � 108 m s�1
</p>
<p>Conversions
</p>
<p>1 eV (electron-volt) &frac14; 1.6022 � 10�19 J
1 Å (Ångström unit) &frac14; 10�10 m &frac14; 0.1 nm
</p>
<p>Prefixes to Units
</p>
<p>femto pico nano micro milli centi deci kilo mega giga
</p>
<p>f p n m m c d k M G
</p>
<p>10�15 10�12 10�9 10�5 10�3 10�2 10�1 103 106 109
</p>
<p>xxxi</p>
<p/>
</div>
<div class="page"><p/>
<p>Projected Revision of SI Units
</p>
<p>The year 1960 saw the publication of Le Système international d&rsquo;unités (the SI)
</p>
<p>as a rational and coherent system of units for scientific research and communi-
</p>
<p>cation. A projected revised SI aims to eliminate certain infelicities in the current
</p>
<p>system, particularly in relation to the kilogram, kelvin, mole, and ampere. The
</p>
<p>standard kilogram, a Pt&ndash;Ir alloy, was adopted as a standard in 1889, but has very
</p>
<p>slowly lost material over the intervening years. In the case of the kelvin, the
</p>
<p>purity and isotopic composition of water need to be defined for a complete
</p>
<p>specification of its triple point, which is used in fixing the kelvin.
</p>
<p>The new SI scheme will define the values of certain constants exactly.
</p>
<p>Thus, it begins with the speed of light (c), which was set exactly as
</p>
<p>2.99792458 � 108 m s�1 in 1983. A re-definition of the other fundamental
SI units can then be projected. For example, the kilogram will be defined
</p>
<p>such that the Planck constant (h) is exactly 6.6260693 � 10�34 J s, then the
kilogram will be fixed, since hn &frac14; E &frac14; mc2 and the metre and second have
defined values.
</p>
<p>The metre is defined in terms of the speed of light, and the second as the
</p>
<p>distance travelled by light in a vacuum in 1/(2.99792458 � 108) s. The
second was given originally as 1/(8.6400 � 104) of the mean solar day, but
in 1967 it was re-defined as the duration of 9.192631770 � 109 periods of the
radiation corresponding to the transition between two hyperfine levels in the
</p>
<p>ground state of 133Cs at 0�K; these two units will be unaltered.
The mole hitherto based on the molar mass of 12C will be revised to that
</p>
<p>mass of the isotope which makes the Avogadro constant exactly
</p>
<p>6.0221415 � 1023 per mole. Changes have also been proposed for the
ampere, but the candela remains unaltered.
</p>
<p>Notwithstanding the value of the SI, certain traditional units are still in
</p>
<p>common use. Thus the Ångström (1 Å &frac14; 10�10 m) remains a very conve-
nient unit in crystallography for quoting interatomic distances and wave-
</p>
<p>lengths. Detailed accounts of the history, revisions, and proposed changes of the
</p>
<p>fundamental units in the system may be found in the published literature1,2.
</p>
<p>These changes in the fundamental units will not affect the numerical values
</p>
<p>involved in the text of this book or in its set problems.
</p>
<p>1 http://physics.nist.gov/cuu/Units/
2Mills IM, Mohr PJ, Quinn TJ, Taylon BN, Williams ER (2011) Phil Trans Roy Soc. 369:3907ff
</p>
<p>xxxii Physical Constants and Other Numerical Data</p>
<p/>
</div>
<div class="page"><p/>
<p>Notation
</p>
<p>These notes provide a key to the main symbols and constants used throughout
</p>
<p>the book. Inevitably, some symbols havemore than one use. This feature arises
</p>
<p>partly from general usage in X-ray crystallography, and partly from a desire to
</p>
<p>preserve a mnemonic character in the notation wherever possible. It is our
</p>
<p>belief that, in context, no confusion will arise. Where several symbols are
</p>
<p>closely linked, they are listed together under the first member of the set. Two
</p>
<p>or more applications of one and the same symbol are separated by a semicolon.
</p>
<p>A0(hkl) Components of the structure factor, measured along the real and
imaginary axes of an Argand diagramB
</p>
<p>0(hkl)
</p>
<p>A(hkl) Components of the geometrical structure factor, measured along the real
</p>
<p>and imaginary axes of an Argand diagramB(hkl)
</p>
<p>A A-face-centred unit cell; absorption correction factor
</p>
<p>Å Ångström unit
</p>
<p>a, b, c Unit-cell edges parallel to the x, y, and z axes, respectively, of a crystal;
</p>
<p>intercepts made by the parametral plane on the x, y, and z axes
</p>
<p>respectively; glide planes with translational components of a/2, b/2, and c/
</p>
<p>2, respectively
</p>
<p>a, b, c Unit-cell edge vectors parallel to the x, y, and z axes, respectively
</p>
<p>a*, b*, c* Reciprocal unit-cell edges associated with the x*, y*, and z* axes,
</p>
<p>respectively
</p>
<p>a*, b*, c* Reciprocal unit-cell vectors associated with the x*, y*, and z* axes,
respectively
</p>
<p>B B-face-centred unit cell; overall isotropic temperature factor
</p>
<p>Bj Isotropic temperature factor for the jth atom
</p>
<p>C C-face-centred unit cell
</p>
<p>C= Not constrained by symmetry to equal
</p>
<p>c Speed of light; as a subscript: calculated, as in jFcj
Dm Experimentally measured crystal density
</p>
<p>Dc Calculated crystal density
</p>
<p>d Interplanar spacing
</p>
<p>d(hkl) Interplanar spacing of the (hkl) family of planes
</p>
<p>d* Distance in reciprocal space
</p>
<p>d*(hkl) Distance from the chosen origin of the reciprocal lattice to the hklth
</p>
<p>reciprocal lattice point
</p>
<p>Da Dalton; equivalent to mu
</p>
<p>E Normalized structure factor (E value), including phase
</p>
<p>jEj Amplitude of normalized structure factor, E (an &ldquo;observed&rdquo; value)
E, E(hkl) Normalized structure factor in centrosymmetric crystals (an &ldquo;observed&rdquo;
</p>
<p>value)
</p>
<p>Ec Normalized structure factor calculated from the atomic positions in the
</p>
<p>unit cell
</p>
<p>E Total energy of the hklth diffracted beam from one unit cell
e Electron charge
</p>
<p>e, exp Exponential function
</p>
<p>esd Estimated standard deviation
</p>
<p>(continued)
</p>
<p>Physical Constants and Other Numerical Data xxxiii</p>
<p/>
</div>
<div class="page"><p/>
<p>F(hkl) Structure factor for the hkl spectrum referred to one unit cell, including
</p>
<p>phase
</p>
<p>F*(hkl) Conjugate of F(hkl), including phase
</p>
<p>jFj or F Modulus, or amplitude, of the structure factor F (excluding phase); jFj is
superfluous notation, but frequently used informally
</p>
<p>Fo Observed structure &ldquo;factor&rdquo; (only ever an amplitude); jFoj is superfluous
notation, but frequently used informally
</p>
<p>f / fj Atomic scattering factor / for the jth atom
</p>
<p>fj,y Atomic scattering factor for the jth atom at a given sin y/l
</p>
<p>g Glide line in two-dimensional space groups
</p>
<p>gj Atomic scattering factor for the jth atom, in a crystal, corrected for
</p>
<p>thermal vibrations
</p>
<p>H Hexagonal (triply primitive) unit cell
</p>
<p>(hkl)/(hkil) Miller / Miller&ndash;Bravais indices (of planes) associated with the x, y, and z
</p>
<p>axes or the x, y, u, and z axes, respectively&mdash;any single index containing
</p>
<p>two digits has a comma placed after such an index
</p>
<p>{hkl} Form of (hkl) planes
</p>
<p>hkl Reciprocal lattice point corresponding to the (hkl) family of planes
</p>
<p>h Vector with components h, k, l in reciprocal space
</p>
<p>h Miller index parallel to the x axis; Planck&rsquo;s constant
</p>
<p>I Body-centred unit cell; intensity of reflection
</p>
<p>Io(hkl) Observed intensity of reflection from the (hkl) planes referred to one unit
</p>
<p>cell
</p>
<p>i Imaginary axis on an Argand diagram
</p>
<p>i
ffiffiffiffiffiffiffi
�1
</p>
<p>p
; an operator that rotates a quantity on an Argand diagram through
</p>
<p>90� in a right-handed (counterclockwise) sense from the real axis
</p>
<p>K Scale factor for Fo(hkl) data
</p>
<p>k Miller index parallel to the y axis; Boltzmann constant
</p>
<p>l Miller index parallel to the z axis
</p>
<p>L Lorentz correction factor
</p>
<p>mu Atomic mass unit
</p>
<p>Mr Relative molecular mass (molecular &ldquo;weight&rdquo;)
</p>
<p>m Mirror plane
</p>
<p>N Number of atoms per unit cell
</p>
<p>n Glide plane, with translational component of (a + b)/2, (b + c)/2, or
</p>
<p>(c + a)/2
</p>
<p>n1, n2, n3 Principal refractive indices in a biaxial crystal
</p>
<p>o Subscript: observed, as in jFo(hkl)j
o Superscript, as in 25�C
</p>
<p>P Probability; Patterson function; Polarization correction factor
</p>
<p>P(uvw) Patterson function at the fractional coordinates u, v, w in the unit cell
</p>
<p>p Polarization correction factor
</p>
<p>pi Probability of the ith state of a system
</p>
<p>R Rhombohedral unit cell; rotation axis of degree R; reliability factor
</p>
<p>(several R parameters are in current use)
</p>
<p>R Inversion axis of degree R
</p>
<p>R Real axis on an Argand diagram
</p>
<p>rms Root mean square
</p>
<p>RU Reciprocal lattice unit
</p>
<p>S Statistical distribution parameter; 2 sin y=l
</p>
<p>s, s(hkl), s(h) Sign of a centric reflection, jFj or jEj
(continued)
</p>
<p>xxxiv Physical Constants and Other Numerical Data</p>
<p/>
</div>
<div class="page"><p/>
<p>Tj,y Thermal vibration parameter for the jth atom at a given sin y/l
</p>
<p>[U V W] Zone or direction symbol
</p>
<p>&lt;U V W&gt; Form of zone axes or directions
</p>
<p>(uvw) Components of a vector in Patterson space
</p>
<p>U2 Mean-square amplitude of vibration
</p>
<p>V Volume
</p>
<p>Vc Volume of a unit cell
</p>
<p>W Probability or number of arrangements of a system
</p>
<p>w Weight factor
</p>
<p>x, y, u, z Crystallographic reference axes descriptors
</p>
<p>X, Y, Z Spatial coordinates, in absolute measure, of a point with respect to the x, y,
</p>
<p>and z axes
</p>
<p>x, y, z Spatial fractional coordinates in a unit cell parallel to x, y, z, respectively
</p>
<p>xj, yj, zj Spatial fractional coordinates of the jth atom in a unit cell parallel to x, y,
</p>
<p>z, respectively
</p>
<p>[x, b, g] Line parallel to the x axis and intersecting the y and z axes at b and g,
</p>
<p>respectively
</p>
<p>(x, y, g) Plane normal to the z axis and intersecting it at g
</p>
<p>� fx; y; z; :::g x; y; z; x; y; z . . .
Z Number of formula entities of mass Mrr per unit cell
</p>
<p>Zj Atomic number of the jth atom in a unit cell
</p>
<p>a, b, g Angles between the pairs of unit-cell edges bc, ca, and ab, respectively
</p>
<p>a*, b*, g* Angles between the pairs of reciprocal unit-cell edges b*c*, c*a*, and a*b*,
</p>
<p>respectively
</p>
<p>d Path difference
</p>
<p>e, e(hkl) Statistical weight of a reflection (epsilon factor)
</p>
<p>e, o Principal refractive indices for a uniaxial crystal
</p>
<p>y Bragg angle
</p>
<p>k Reciprocal space constant
</p>
<p>l Wavelength
</p>
<p>m Linear absorption coefficient
</p>
<p>n Frequency
</p>
<p>r(xyz) Electron density at the point x, y, z (units are length�3)
</p>
<p>F Interfacial (internormal) angle
</p>
<p>&rsquo;(hkl), &rsquo;(h), &rsquo; Phase angle associated with a structure factor
</p>
<p>w, C, o (cos w, cos C, cos o) direction cosines of a line with respect to the x, y,
</p>
<p>and z axes
</p>
<p>o Angular frequency
</p>
<p>O Azimuthal angle in experimental methods; ohm
</p>
<p>X; &lt;X&gt; Average value of X
</p>
<p>Physical Constants and Other Numerical Data xxxv</p>
<p/>
</div>
<div class="page"><p/>
<p>Crystal Morphology and Crystal
Symmetry 1
</p>
<p>1.1 Brief Historical Introduction
</p>
<p>Crystals, with their plane faces, sharp angles, and color, have excited interest since the earliest times.
</p>
<p>Their color and decorative qualities are recorded in the Bible [1]: we need not start as far back as that, but
</p>
<p>will consider instead some of the highlights in the build-up of the science of Crystallography.
</p>
<p>The Swiss naturalist Conrad Gessner [2] observed in 1564 that &ldquo;one crystal differs from another in
</p>
<p>its angles, and consequently in its figure,&rdquo; and Pliny [3], Caesalpinus [4], and Buffon [5] also denied
</p>
<p>&ldquo;the fixity of crystals.&rdquo; Niels Stensen, a Dane, also known as Nicolaus Steno [6], showed in 1669 that
</p>
<p>crystals of quartz, although of varying edge lengths, nevertheless preserved constant angles between
</p>
<p>corresponding faces, an observation that led, after further work, to the law of constant interfacial
</p>
<p>angles (q.v.), sometimes called Steno&rsquo;s law. Some examples of Steno&rsquo;s figures for quartz are shown in
</p>
<p>Fig. 1.1. Pliny referred to quartz as krustallos (Gk. krnstallos &frac14; ice), believing it to be ice
permanently congealed by intense cold: &ldquo;rain-water and pure snow are needed for its formation.&rdquo;
</p>
<p>In the same year, Bartholinus [7] published his work on the Iceland Spar modification of calcium
</p>
<p>carbonate: he measured the interfacial angles (q.v.) of crystal fragments and found that &ldquo;the
</p>
<p>(cleavage) fragments have the same shape as the original crystal.&rdquo;
</p>
<p>JohannesKepler, best known for his research in astronomy,worked sometime for EmperorRudolph II
</p>
<p>of Austria, who was somewhat negligent in paying salaries. Kepler, who was without money with which
</p>
<p>to purchase aChristmas present for a friend in 1611,wrote a booklet entitledThe Six-Cornered Snowflake
</p>
<p>[8], Fig. 1.2. He had noticed that snow crystals of whatever shape always exhibited sixfold symmetry. He
</p>
<p>speculated on a relationship between the snow crystal and a hexagonal close packing of spheres: he
</p>
<p>recognized the genre of crystal symmetry, which was to be developed about two centuries years later.
</p>
<p>Jokingly, hepassedoff his &ldquo;NewYear&rsquo;sGift&rdquo; as amere &ldquo;nothing&rdquo; (like his finances),makinga playon the
</p>
<p>word nix, which means &ldquo;snowflake&rdquo; in Latin but &ldquo;nothing&rdquo; in Lower German, Kepler&rsquo;s language.
</p>
<p>Over the period 1688&ndash;1705, many investigations by the Italian physicist Giovanni Guglielmini [9]
</p>
<p>confirmed and extended the work of Steno, and during 1772&ndash;1783, about a century after Steno, the
</p>
<p>Frenchman Jean-Baptiste Romé de l&rsquo;Isle [10] carried out an extensive series of measurements that
</p>
<p>confirmed fully Steno&rsquo;s findings: he stated further, as a law, that &ldquo;the interfacial angles of a crystal
</p>
<p>material are characteristic of it.&rdquo; In his work, hemademuch use of the contact goniometer, developed by
</p>
<p>Arnould Carangeot [11], for measuring interfacial angles on crystals.
</p>
<p>His compatriot René Just (Abbé) Ha&euro;uy [12] published a treatise on mineralogy in 1801 in which he
</p>
<p>envisaged crystals built up by stacking identical blocks of structural material in ways that led to the
</p>
<p>shapes of crystals, thus explaining Steno&rsquo;s law. He was led to this view by the observation that when
</p>
<p>M. Ladd and R. Palmer, Structure Determination by X-ray Crystallography:
</p>
<p>Analysis by X-rays and Neutrons, DOI 10.1007/978-1-4614-3954-7_1,
# Springer Science+Business Media New York 2013
</p>
<p>1</p>
<p/>
</div>
<div class="page"><p/>
<p>Fig. 1.1 Examples of Steno&rsquo;s drawings of transverse sections of different crystal specimens of quartz, SiO2; the
interfacial angles are 120� or (120/2)� in each case
</p>
<p>Fig. 1.2 The cover (in translation) of Kepler&rsquo;s booklet on The Six-Cornered Snowflake; some of Kepler&rsquo;s many
drawings of snowflakes appear on the cover (reproduced by courtesy of Paul Dry Books)
</p>
<p>2 1 Crystal Morphology and Crystal Symmetry</p>
<p/>
</div>
<div class="page"><p/>
<p>crystals of calcite were broken, they always formed rhombohedral-shaped fragments whatever the
</p>
<p>shape of the original crystal, an observation similar to that made by Bartholinus. Other crystals cleaved
</p>
<p>into different shapes, such as cubes; Fig. 1.3 illustrates some of Ha&euro;uy&rsquo;s figures. He laid the foundation
</p>
<p>for the law of rational intercepts (q.v.), which demonstrated, albeit implicitly, a shape for the unit cell
</p>
<p>(q.v.) of a crystal. He described also different crystal shapes, or habits, obtained by the packing of
</p>
<p>different shaped blocks.
</p>
<p>In the period 1804&ndash;1815, the German scientists Christian Weiss [13] and Frederick Mohs [14],
</p>
<p>independently, identified the six &ldquo;major&rdquo; crystal systems (q.v.)&mdash;it seems that they did not distinguish
</p>
<p>between the two systems that we now recognize as trigonal and hexagonal. Another German scientist
</p>
<p>Johann Hessel [15] determined the 32 crystallographic point groups (q.v.) in 1835 and his compatriot
</p>
<p>Moritz Frankenheim [16] also described six crystal systems and, additionally, the 32 crystal classes in
</p>
<p>1842. Crystal classes are names for the general forms (q.v.) of crystals, of which we shall have more to
</p>
<p>say later, Sect. 1.4.2; Fig. 1.4 illustrates two examples of crystal class.
</p>
<p>A next landmark was that contained in the work of William Miller, 1839, who proposed an
</p>
<p>unambiguous notation for specifying the orientations of the faces of a crystal, leading to the Miller
</p>
<p>Indices [17] (q.v.), based on the scalar equation of a plane; the law of rational indices is clear from
</p>
<p>Miller&rsquo;s work&mdash;the descriptor &ldquo;law of rational ratios of intercepts&rdquo; is perhaps more apt.
</p>
<p>The regular arrangements of points in space were addressed by August Bravais [18] in 1846, who
</p>
<p>derived the 14 Bravais lattices. Actually, Frankenheim had derived 15 such lattices in 1842, but
</p>
<p>Bravais showed that two of them were identical. For his infelicity, Frankenheim&rsquo;s name is frequently
</p>
<p>omitted from a discussion of lattices, albeit the nature of his 15th lattice was never completely clear.
</p>
<p>Continuing the development of crystal symmetry, we have at the end of the nineteenth century,
</p>
<p>Yevgraf Fyodorov [19] (in Russia), Artur Sch&euro;onflies [20] (in Germany), and William Barlow [21]
</p>
<p>(in Britain), all independently, describing the 230 ordered spatial patterns, or space groups (q.v.), that
</p>
<p>represented the possible ways of arranging infinite arrays of points (atoms) regularly in space,
</p>
<p>commensurate with the 14 Bravais lattices.
</p>
<p>By that time, the work of mineralogists and mathematicians had led to a well-defined study of the
</p>
<p>external form (morphology) of crystals, as well as to predictions about their internal structure. The
</p>
<p>totality of mineralogical studies was collected in the extensive six-volume treatise ChemischeKris-
</p>
<p>tallographie of Paul von Groth, the first part of which was published in 1904. The time was right for a
</p>
<p>major breakthrough, and we arrive at the year 1912 and the inception of the study of crystals by X-ray
</p>
<p>diffraction methods: X-ray Crystallography was born.
</p>
<p>Fig. 1.3 Examples of Ha&euro;uy&rsquo;s figures. (a) Rhombic dodecahedron formed from stacked cubes, as in garnet (idealized
formula Ca3Al2Si3O12). (b) Pentagonal dodecahedron formed from stacked cubes, as in K2Mn2(SO4)3. (c) Scalenohe-
dron formed from stacked rhombohedra, as in calcite, CaCO3
</p>
<p>1.1 Brief Historical Introduction 3</p>
<p/>
</div>
<div class="page"><p/>
<p>The classic experiment [22], suggested by Laue,1 and performed by Paul Friedrich and Walter
</p>
<p>Knipping, demonstrated the diffraction of X-rays from a crystal of copper sulfate. This material was,
</p>
<p>perhaps, not the best choice because of its low (triclinic) symmetry. Nevertheless, the diffraction
</p>
<p>effects showed conclusively both that crystals were periodic in three dimensions and that X-rays
</p>
<p>possessed wave-like properties. Figure 1.5 shows X-ray diffraction patterns from early experiments on
</p>
<p>X-ray diffraction.
</p>
<p>The results from this work may be said to be a landmark in the development of modern science.
</p>
<p>The diffraction technique that was initiated by Laue was improved quickly by W. L. Bragg, in his
</p>
<p>work on crystals of the alkali-metal halides and other crystals. Barlow, one of those who had derived
</p>
<p>the space groups earlier, had also developed structure models for some metallic elements, and simple
</p>
<p>binary compounds such as sodium chloride, cesium chloride, and zinc blende. None of his results was
</p>
<p>proved at that time: all were speculative, but remarkably accurate, as it turned out.
</p>
<p>Bragg [23] investigated the X-ray diffraction patterns of sodium chloride and other alkali halides
</p>
<p>and zinc blende (ZnS). He found that the models suggested by Barlow were correct, and other models
</p>
<p>proposed by Barlow, such as cubic close-packed and hexagonal close-packed structures for metallic
</p>
<p>elements, were confirmed. The number of structure analyses grew very rapidly: Fig. 1.6 is a stereo-
</p>
<p>view of the structure of sodium chloride, NaCl. Several stereoviews are used in this book in order to
</p>
<p>Fig. 1.4 Examples
of crystal classes:
</p>
<p>(a) Rhombic disphenoid,
showing left-hand and
</p>
<p>right-hand enantiomorphs
</p>
<p>(q.v.). (b) Hex(akis)
octahedron
</p>
<p>1 In 1913, Laue&rsquo;s father was raised to the ranks of hereditary nobility; Laue then became von Laue.
</p>
<p>4 1 Crystal Morphology and Crystal Symmetry</p>
<p/>
</div>
<div class="page"><p/>
<p>demonstrate clearly the three-dimensional nature of crystal structures, and Appendix A describes the
</p>
<p>process of stereoviewing, and the construction of a simple stereoviewer.
</p>
<p>The early structure analyses were carried out with the aid of an X-ray ionization spectrometer, the
</p>
<p>forerunner of the modern single-crystal X-ray diffractometer, designed largely by W. H. Bragg [24].
</p>
<p>Generally, the name Bragg refers toW. L. Bragg, but his father (W. H.) also played a highly significant
</p>
<p>role at the very beginning of the technique of crystal structure analysis by X-ray diffraction.
</p>
<p>X-ray diffraction provides the most powerful technique for probing the internal structures of crystals
</p>
<p>and for determining with high precision the actual atomic arrangement in space. Figure 1.7 shows a
</p>
<p>three-dimensional contour map of the electron density in a medium-sized molecule, euphenyl iodoace-
</p>
<p>tate [25], C32H53O2I. The contour lines join points of equal electron density in the structure; hydrogen
</p>
<p>atoms are not revealed in this map because of their relatively small scattering power for X-rays.
</p>
<p>If we assume that the centers of atoms are located at the maxima in the electron density map, we can
</p>
<p>deduce the molecular model in Fig. 1.8a; the chemical structural formula is shown for comparison in
</p>
<p>Fig. 1.8b. The iodine atom is represented by the large number of contours at the extreme left of Fig. 1.7.
</p>
<p>The carbon and oxygen atoms are depicted by approximately equal numbers of contours. The atoms in the
</p>
<p>side chain, shown on the extreme right of the figure, have contours that are spaced further apart. Thermal
</p>
<p>Fig. 1.6 Stereoview of the face-centered cubic unit cell and its environs for the crystal structure of sodium chloride:
</p>
<p>&frac14; Cl�, &frac14; Na&thorn;
</p>
<p>Fig. 1.5 The first X-ray diffraction photograph by Friedrich et al. [22]. (a) Copper sulfate pentahydrate, CuSO4&middot;5H2O,
showing no symmetry. (b) Zinc blende (the first crystal structure to be fully determined), ZnS, showing fourfold symmetry
(Bragg WH, Bragg WL (1949) The crystalline state, vol 1. G. Bell and Sons)
</p>
<p>1.1 Brief Historical Introduction 5</p>
<p/>
</div>
<div class="page"><p/>
<p>vibrations of the atoms aremost severe in this portion of themolecule, and theyhave the effect of smearing
</p>
<p>out the electron density, so that its gradient, represented by the closeness of the contours, is less steep than
</p>
<p>in other parts of the molecule.
</p>
<p>Molecules of much greater complexity than that in this example are now being investigated;
</p>
<p>the structures of proteins, enzymes, and nucleic acids&mdash;the &ldquo;elements&rdquo; of life itself&mdash;are
</p>
<p>being revealed by powerful X-ray diffraction techniques [25a].
</p>
<p>Fig. 1.8 Euphenyl iodoacetate, C32H53O2I. (a) Molecular model, excluding hydrogen atoms. (b) Chemical structural
formula: the orientations at carbon atoms 13, 14, and 17 are a, b, and a, respectively (standard numbering)
</p>
<p>Fig. 1.7 Three-dimensional electron density contour map for euphenyl iodoacetate, as seen along the b direction of the
unit cell; the contours connect points of equal electron density: the electron density contours have been drawn on a set of
</p>
<p>perspex sheets and stacked normal to b
</p>
<p>6 1 Crystal Morphology and Crystal Symmetry</p>
<p/>
</div>
<div class="page"><p/>
<p>1.2 The Crystalline State
</p>
<p>A crystalline substance may be defined as a homogeneous solid having an ordered internal atomic
</p>
<p>arrangement2 and a definite overall chemical composition, albeit non-stoichiometric in some exam-
</p>
<p>ples. In addition to the more obvious manifestations of crystalline materials, like sugar and common
</p>
<p>salt, other substances, such as cellophane sheet and fibrous asbestos, which reveal different degrees of
</p>
<p>long-range order (extending over many atomic dimensions), may be described as crystalline.
</p>
<p>With the unaided eye, fragments of glass and of quartz look similar to each other, yet quartz is
</p>
<p>crystalline and glass is non-crystalline, oramorphous. Glass has an atomic arrangement that displays only
</p>
<p>very short-range order (extending over a few atomic dimensions). Figure 1.9 illustrates the structures of
</p>
<p>quartz and silica glass; both of them are based on the same atomic group, the tetrahedral SiO4 structural
</p>
<p>unit, but in quartz these groups are arranged regularly throughout three-dimensional space.
</p>
<p>A crystal may be defined as a substance that is crystalline and periodic in three dimensions and
</p>
<p>bounded by plane faces. We have now made the useful distinction that crystalline substances exhibit
</p>
<p>long-range order in three dimensions or less, whereas crystals have both this three-dimensional
</p>
<p>regularity and plane bounding faces; see also Sect. 1.4.3.
</p>
<p>1.2.1 Crystallographic Reference Axes
</p>
<p>In describing the external features of crystals, we make use of relationships in coordinate geometry. It
</p>
<p>is important to set up a system of reference axes, and three such axes are needed in the description of a
</p>
<p>crystal, Fig. 1.10. By convention, the x, y, and z crystallographic reference axes are set parallel to
</p>
<p>important directions in the crystal. We shall see later that these directions (crystal edges, or possible
</p>
<p>crystal edges) are related closely to the symmetry of the crystal; in some cases, a choice of non-
</p>
<p>orthogonal axes then will arise naturally.
</p>
<p>It is usual to work with right-handed axes. In Fig. 1.11, +y and +z are in the plane of the paper, as
</p>
<p>shown, and +x is directed forward; the succession +x ! +y ! +z simulates an anticlockwise screw
motion, which is one way of describing right-handed axes. Notice the selection of the interaxial angles a,
</p>
<p>b, and g and the mnemonic connection between their positions and the directions of the x, y, and z axes.
</p>
<p>1.2.2 Equation of a Plane
</p>
<p>The plane ABC shown in Fig. 1.12 intercepts the x, y, and z axes (which need not be orthogonal) at A,
</p>
<p>B, and C, respectively. ON is the perpendicular from the origin O to the plane; it has the length d, and
</p>
<p>its direction cosines (see Web Appendix WA1) are cos w, cos C, and cos o with respect to OA, OB,
</p>
<p>and OC, respectively, which have the lengths a, b, and c, and P is any point X, Y, Z in the plane ABC.
</p>
<p>Let PK be parallel toOC and meet the plane AOB atK, and let KM be parallel toOB and meetOA atM.
</p>
<p>Then the lengths of OM, MK, and KP are X, Y, and Z, respectively. Since ON is the projection of
</p>
<p>OP on to ON, it is equal to the sum of the projections OM, MK, and KP all on to ON. Hence,
</p>
<p>d &frac14; X cos w&thorn; Y cosC&thorn; Z coso (1.1)
</p>
<p>2 See also Sect. 8.9.
</p>
<p>1.2 The Crystalline State 7</p>
<p/>
</div>
<div class="page"><p/>
<p>In DOAN, d &frac14; OA cos w &frac14; a cos w. Similarly, d &frac14; b cos C &frac14; c cos o; dividing by d:
</p>
<p>&eth;X=a&THORN; &thorn; &eth;Y=b&THORN; &thorn; &eth;Z=c&THORN; &frac14; 1 (1.2)
</p>
<p>Equation (1.2) is the intercept form of the equation of the plane ABC.
</p>
<p>1.2.3 Indices of Planes and the Law of Rational Intercepts
</p>
<p>Miller Indices
The faces of a crystal are planes in three-dimensional space. Once the crystallographic axes are
</p>
<p>chosen, a parametral plane may be defined and any other plane described by reference to it in terms of
</p>
<p>three numbers h, k, and l. If the parametral plane is designated by integral values of h, k, and l,
</p>
<p>Fig. 1.9 Arrangements of
SiO4 structural units (the
</p>
<p>darker spheres represent
</p>
<p>Si). (a) a-Quartz. (b) Silica
glass (reproduced by
</p>
<p>courtesy of NPL)
</p>
<p>8 1 Crystal Morphology and Crystal Symmetry</p>
<p/>
</div>
<div class="page"><p/>
<p>normally (111), then the indices of all other crystal faces are small integer values, rarely more than 5.
</p>
<p>This result is known as the law of rational intercepts (indices) (q.v.) and has a basis in lattice theory.
</p>
<p>A notation for describing the faces of a crystal was introduced first by William Whewell in 1825 and
</p>
<p>developed fully by Miller [17] in 1839, and h, k, and l are the Miller indices.
</p>
<p>In Fig. 1.13, let the parametral plane (111) be ABC, making intercepts a, b, and c on the
</p>
<p>crystallographic axes x, y, and z, respectively. Another plane LMN makes corresponding intercepts
</p>
<p>of lengths a/h, b/k, and c/l. The Miller indices of plane LMN are expressed by the ratios of the
</p>
<p>intercepts of the parametral plane to those of the plane LMN. If in the figure, a/h &frac14; a/4, b/k &frac14; b/3,
and c/l &frac14; c/2, then LMN is (432), see also Sect. 2.3. If fractions occur in formulating h, k, or l, they are
</p>
<p>Fig. 1.11 Right-handed, general crystallographic axes x, y, and z, and the interaxial angles a, b, and g
</p>
<p>Fig. 1.10 Idealized tetragonal crystal with orthogonal (mutually perpendicular) axes x, y, and z drawn in
</p>
<p>1.2 The Crystalline State 9</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec6_2">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec6_2</a></div>
</div>
<div class="page"><p/>
<p>cleared by multiplication throughout by the lowest common denominator. Conditions of parallelism
</p>
<p>to axes and intercepts on the negative sides of the axes lead respectively to zero or negative values for
</p>
<p>h, k, and l. Thus, ABDE is the plane (110), BDFG is (010), and PBQ is &eth;�21�3&THORN;.3 It may be noted that it
has not been necessary to assign numerical values to either a, b, and c or a, b, and g in order to
</p>
<p>describe the crystal faces by their Miller indices. In the next chapter, we shall identify a, b, and c with
</p>
<p>the edges of the crystal unit cell in a lattice, but this relationship is not needed at present.
</p>
<p>The preferred choice of the parametral plane leads to small numerical values for the Miller indices
</p>
<p>of crystal faces. If LMN had been chosen as (111), then ABC would have been (346). Summarizing,
</p>
<p>we may say that the plane (hkl) makes intercepts a/h, b/k, and c/l along the crystallographic x, y, and z
</p>
<p>axes, respectively, where a, b, and c are the corresponding intercepts made by the parametral plane.
</p>
<p>The conventional choice of reference axes leads to special relationships between the intercepts a, b, c,
</p>
<p>and the (111) parametral plane, and between the interaxial angles a, b and g, in all crystals other than
</p>
<p>triclinic (q.v.).
</p>
<p>From (1.1) and (1.2), the intercept equation of the general plane (hkl) may be written as
</p>
<p>&eth;hX=a&THORN; &thorn; &eth;kY=b&THORN; &thorn; &eth;lZ=c&THORN; &frac14; 1 (1.3)
</p>
<p>The equation of the parallel plane passing through the origin is
</p>
<p>&eth;hX=a&THORN; &thorn; &eth;kY=b&THORN; &thorn; &eth;lZ=c&THORN; &frac14; 0 (1.4)
</p>
<p>it must satisfy the condition X &frac14; Y &frac14; Z &frac14; 0. It follows from (1.4) that the Miller indices of a crystal
plane cannot be determined if the origin is chosen on that plane.
</p>
<p>Fig. 1.12 Plane ABC in three-dimensional space; ON, of length d, is the normal to the plane from the origin O
</p>
<p>3Read as &ldquo;bar-two one bar-three,&rdquo; or &ldquo;two-bar one three-bar&rdquo; in the USA.
</p>
<p>10 1 Crystal Morphology and Crystal Symmetry</p>
<p/>
</div>
<div class="page"><p/>
<p>Miller&ndash;Bravais Indices
In crystals that exhibit sixfold or threefold symmetry referred to hexagonal axes, see Table 1.3, four
</p>
<p>axes of reference may be used, an extension of the Miller indices due to Bravais [18]. The axes are
</p>
<p>designated x, y, u, and z; the x, y, and u axes lie in one plane, at 120� to one another, and the z axis is
perpendicular to the x, y, u plane, Fig. 1.14; the sequence x, y, u, z is right-handed. Planes in these
</p>
<p>crystals are described by four numbers, the Miller&ndash;Bravais indices h, k, i, and l. The index i is not
</p>
<p>Fig. 1.13 Miller indices of planes: OA &frac14; a, OB &frac14; b, OC &frac14; c; ABC is the parametral plane (111), and LMN is the
plane (hkl)
</p>
<p>Fig. 1.14 Miller&ndash;Bravais indices (hkil). The crystallographic axes are labeled x, y, u, z, and the plane &eth;2354&THORN; is shown;
the parametral plane is &eth;1121&THORN;
</p>
<p>1.2 The Crystalline State 11</p>
<p/>
</div>
<div class="page"><p/>
<p>independent of h and k: thus, if the plane ABC in Fig. 1.14 intercepts the x and y axes at a/2 and b/3,
</p>
<p>for example, then the u axis is intercepted at�u/5. If also the z axis is intercepted at c/4, then the plane
is designated &eth;23�54&THORN;. From Problem 1.14, we show that, in general, i &frac14; �(h + k).
</p>
<p>Law of Rational Intercepts
Planes that appear as external crystal faces are those most densely populated by the units of
</p>
<p>structure. Consider a P orthorhombic lattice, Fig. 2.4. The (010) planes are more densely populated
</p>
<p>in proportion to 1/ac than are the (110) planes at 1=&eth;c
ffiffiffiffiffiffiffiffiffiffiffiffiffiffiffi
a2 &thorn; b2
</p>
<p>p
&THORN; which, in turn, are more densely
</p>
<p>populated than, say, the (210) at 1=&eth;c
ffiffiffiffiffiffiffiffiffiffiffiffiffiffiffiffiffi
a2 &thorn; 4b2
</p>
<p>p
&THORN;. The more densely populated planes are those of
</p>
<p>wider spacing in a given material: d is proportional to the reticular density or to 1/reticular area. The
</p>
<p>planes with the lower values of S2, where S2 &frac14; h2 + k2 + l2, lead to the more stable (lower energy)
crystal state. If we consider a cubic crystal, for example, then d &frac14; a=&eth;h2&thorn; k2 &thorn; l2&THORN;1=2 &frac14; a=S, from
Table 2.4, so that for the three cubic lattices, Fig. 2.4, we have:
</p>
<p>P
</p>
<p>hkl 100 110 111 210 211 221
</p>
<p>S2 1 2 3 5 6 9
</p>
<p>Showing a preference for hexahedral {100} forms, as in caesium chloride or sodium chlorate.
</p>
<p>I
</p>
<p>hkl 110 200 211 310 420 442
</p>
<p>S2 2 4 6 10 20 36
</p>
<p>Showing a preference for dodecahedral {110} forms, as in garnet structures.
</p>
<p>F
</p>
<p>hkl 111 200 220 310 420 422
</p>
<p>S2 3 4 8 10 20 246
</p>
<p>Showing a preference for octahedral {111} forms, as in diamond and calcium fluoride. Thus, on
</p>
<p>grounds of stability, planes of larger d-values (smaller values of h, k, and l) are to be expected.
</p>
<p>1.2.4 Axial Ratios
</p>
<p>If both sides of (1.4) are multiplied by b, we obtain
</p>
<p>hX
</p>
<p>a=b
&thorn; kY &thorn; lZ
</p>
<p>c=b
&frac14; 0 (1.5)
</p>
<p>The quantities a/b and c/b are termed axial ratios; they can be deduced from an analysis of the
</p>
<p>crystal morphology, but not the individual values of a, b, or c.
</p>
<p>1.2.5 Zones
</p>
<p>Most well formed crystals have their faces arranged in groups of two or more with respect to certain
</p>
<p>directions in the crystal. In other words, crystals exhibit symmetry; this feature is an external
</p>
<p>manifestation of the ordered arrangement of atoms in the crystal. Figure 1.15 illustrates zircon,
</p>
<p>12 1 Crystal Morphology and Crystal Symmetry</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Fig4_2">http://dx.doi.org/10.1007/978-1-4614-3954-7#Fig4_2</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Tab4_2">http://dx.doi.org/10.1007/978-1-4614-3954-7#Tab4_2</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Fig4_2">http://dx.doi.org/10.1007/978-1-4614-3954-7#Fig4_2</a></div>
</div>
<div class="page"><p/>
<p>ZrSiO4, an example of a highly symmetric crystal. It is evident that several faces have a given
</p>
<p>direction in common. Such faces are said to lie in a zone, and the common direction is called a zone
</p>
<p>axis. Any two faces, (h1k1l1) and (h2k2l2), define a zone. The zone axis is the line of intersection of the
</p>
<p>two planes and is given by the solution of the equations
</p>
<p>&eth;h1X=a&THORN; &thorn; &eth;k1Y=b&THORN; &thorn; &eth;l1Z=c&THORN; &frac14; 0
&eth;h2X=a&THORN; &thorn; &eth;k2Y=b&THORN; &thorn; &eth;l2Z=c&THORN; &frac14; 0
</p>
<p>(1.6)
</p>
<p>for the two planes passing through the origin (since we are concerned here only with the directional-
</p>
<p>ity). The solution is given by the line
</p>
<p>X
</p>
<p>a&eth;k1l2 � k2l1&THORN;
&frac14; Y
</p>
<p>b&eth;l1h2 � l2h1&THORN;
&frac14; Z
</p>
<p>c&eth;h1k2 � h2k1&THORN;
(1.7)
</p>
<p>which must also pass through the origin. It may be written as
</p>
<p>X=&eth;aU&THORN; &frac14; Y=&eth;bV&THORN; &frac14; Z=&eth;cW&THORN; (1.8)
</p>
<p>where [UVW] is called the zone symbol.
</p>
<p>If any other face (hkl) lies in the same zone as that defined by (h1k1l1) and (h2k2l2), then it follows
</p>
<p>from immediately from (1.4) and (1.8), that
</p>
<p>hU &thorn; kV &thorn; lW &frac14; 0 (1.9)
</p>
<p>which is an expression of the Weiss zone law [13]. For if planes (h1k1l1), (h2k2l2), and (hkl) lie in one
</p>
<p>and the same zone (tautozonal), then applying (1.6) to (hkl) and using (1.8) in the result leads to (1.9).
</p>
<p>We shall show this result in another way in Sect. 2.5.4.
</p>
<p>Fig. 1.15 A highly symmetric crystal (zircon, ZrSiO4), showing the Miller indices of some of its faces. What are the
Miller indices of the other faces on this crystal?
</p>
<p>1.2 The Crystalline State 13</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec12_2">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec12_2</a></div>
</div>
<div class="page"><p/>
<p>In the zircon crystal, the vertical (prism) faces lie in one zone. If the prism faces are indexed in the
</p>
<p>usual manner, as in Fig. 1.15, then, from (1.7) and (1.8), the corresponding zone symbol is [001]. The
</p>
<p>symbols [UVW] and &frac12; �U �V �W� refer to lines that are collinear, but of opposite sense. From (1.9), we see
that (110) and (100) are faces in the [001] zone, but (111) is not. Other relationships follow from (1.9) in a
</p>
<p>similar way. In the manipulation of these equations, it may be noted that a zone axis is described by
</p>
<p>[UVW], the simplest symbol; the axes that may be described as [nU, nV, nW] (n &frac14; 0, �1, �2,. . .) are
coincident with [UVW] in crystal morphology. From (1.6), a zone symbol cannot be determined from
</p>
<p>two faces (hkl) and &eth;�h �k �l&THORN;. Thus, although both the (110) and &eth;�1�10&THORN; planes lie in the [001] zone, this zone
symbol could not be determined from these two planes alone.
</p>
<p>Angle Between Two Directions (Zone Axes)
The angle y between two directions [UVW] and &frac12;U0V0W0� is given by
</p>
<p>y &frac14; r&frac12;UVW� � r&frac12;U
0V0W0�
</p>
<p>r&frac12;UVW�r&frac12;U0V0W0�
(1.10)
</p>
<p>the evaluation of which follows from (2.16).
</p>
<p>Fig. 1.16 Optical goniometry. (a) Principle of the reflecting goniometer; AB and BC represent two adjacent tautozonal
faces on the crystal. (b) Two-circle optical goniometer; the crystal rotates about the vertical circle (the zone axis
through O), and the telescope and collimator rotate about the horizontal circle
</p>
<p>14 1 Crystal Morphology and Crystal Symmetry</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Equ16_2">http://dx.doi.org/10.1007/978-1-4614-3954-7#Equ16_2</a></div>
</div>
<div class="page"><p/>
<p>Constancy of Interfacial Angles
The law of constant interfacial angles states that in all crystals of the same substance, angles between
</p>
<p>corresponding faces have a constant value. Interfacial angles may be measured by a goniometer, the
</p>
<p>first of which was that developed by Carangeot [11], as we noted earlier.
</p>
<p>The principle of the more precise reflecting goniometer is shown in Fig. 1.16a, which forms the basis
</p>
<p>of modern optical goniometry. A crystal is arranged to rotate about a zone axis O, which is set
</p>
<p>perpendicular to a plane containing the incident and crystal-reflected light beams. Parallel light reflected
</p>
<p>from the face AB is received by a telescope. If the crystal is rotated in a clockwise direction, a reflection
</p>
<p>from the face BC is received next when the crystal has been turned through the angle F; then, the
</p>
<p>interfacial angle is 180 � F�. Accurate goniometry brought a quantitative significance to observable
angular relationships in crystals. Figure 1.16b illustrates a simple two-circle optical goniometer.
</p>
<p>1.3 Stereographic Projection: Brief Survey
</p>
<p>A representation of the many faces of a crystal by means of a convenient two-dimensional illustration
</p>
<p>may be achieved by means of a stereographic projection, or stereogram. Figure 1.17 shows a cubic
</p>
<p>crystal set within a sphere of arbitrary radius, with the normals to the sets of crystal faces drawn to
</p>
<p>intersect the sphere. Lines are drawn from the intersections with the sphere to meet the horizontal
</p>
<p>plane, or primitive, which becomes the plane of projection. An intersection from the upper hemi-
</p>
<p>sphere on to the horizontal plane, a pole, is marked as ● and that from the lower hemisphere as ○;
</p>
<p>where the two coincide the notation used is the dot inside the circle. Thus, we obtain Fig. 1.18, which
</p>
<p>may be indexed for the given crystal as shown in Fig. 1.19. Later in this chapter, we shall introduce a
</p>
<p>modification for the notation of points on a stereogram, for reasons that will be there described.
</p>
<p>Fig. 1.17 Spherical
projection of the cubic
</p>
<p>crystal showing three forms
</p>
<p>of planes: cube&mdash;faces b, e,
</p>
<p>d, and parallel faces;
</p>
<p>octahedron&mdash;faces r, m, n,
</p>
<p>q, and parallel faces;
</p>
<p>rhombic dodecahedron&mdash;
</p>
<p>faces f, g, p, o, c, a, and
</p>
<p>parallel faces. The x, y, and
</p>
<p>z axes are chosen parallel to
</p>
<p>important (symmetry)
</p>
<p>directions in the crystal; the
</p>
<p>radius of the sphere is
</p>
<p>arbitrary. The inclined
</p>
<p>great circle, b, m, o0, q0,. . .,
projects as G3G
</p>
<p>0
3 in
</p>
<p>Fig. 1.18
</p>
<p>1.3 Stereographic Projection: Brief Survey 15</p>
<p/>
</div>
<div class="page"><p/>
<p>Fig. 1.18 Stereogram of
the crystal shown in
</p>
<p>Fig. 1.17; the zone circle
</p>
<p>(great circle) G1G
0
1, symbol
</p>
<p>[101], passes through e, q,
</p>
<p>a, n, e0, q0, a0, n0; the zone
circle G2G
</p>
<p>0
2 symbol &frac12;1�10�,
</p>
<p>passes through f, r, d, q0, f0,
r0, d0, q; the zone circle
G3G
</p>
<p>0
3, symbol [011], passes
</p>
<p>through b, m, o0, q0, b0, m0,
o, q
</p>
<p>Fig. 1.19 Stereogram in
Fig. 1.18 indexed, taking r
</p>
<p>as 111. The zone
</p>
<p>containing (100) and (111)
</p>
<p>is &frac12;0�11�, and that containing
(010) and (001) is [100].
</p>
<p>From (1.7), (1.8), and (1.9),
</p>
<p>it follows that the face
</p>
<p>p common to these two
</p>
<p>zones is (011)
</p>
<p>16 1 Crystal Morphology and Crystal Symmetry</p>
<p/>
</div>
<div class="page"><p/>
<p>The important feature of the stereogram for our purposes is that it preserves the interfacial angles
</p>
<p>of the crystal and, hence, displays faithfully the crystal symmetry. A more detailed description of the
</p>
<p>stereographic projection may be found in the Web Appendix WA2.
</p>
<p>1.4 External Symmetry of Crystals
</p>
<p>The existence of faces on a crystal in groups of two or more, in a similar orientation with respect to
</p>
<p>some line or plane in the crystal, is a manifestation of symmetry. The crystal drawing of zircon in
</p>
<p>Fig. 1.15 shows several sets of symmetrically arranged faces.
</p>
<p>Few of us have difficulty in recognizing symmetry in two-dimensional pictures of objects such as a
</p>
<p>dumbbell, the three-legged emblem of the Isle of Man, a Maltese cross, the five-petal Tudor rose, the
</p>
<p>six-pointed Star of David; see Problem 1.10(c). But it is a rather different matter when we are dealing
</p>
<p>with three-dimensional objects.
</p>
<p>The problem arises first from the fact that we can see all parts of a two-dimensional object
</p>
<p>simultaneously, and thus we take in the relation of the parts to the whole; but we cannot do that so
</p>
<p>easily with three-dimensional objects. Secondly, while some three-dimensional objects, such as
</p>
<p>flowers, pencils, and architectural columns, are simple enough for us to visualize and to rotate in
</p>
<p>our mind&rsquo;s eye, few of us have a natural gift for mentally perceiving and manipulating more complex
</p>
<p>three-dimensional objects. Nevertheless, the art of doing so can be developed with suitable aids and
</p>
<p>patience. If, initially, you have problems, take heart. You are not alone and, like many before you, you
</p>
<p>will be surprised at how swiftly the required facility can be acquired. Engineers, architects, and
</p>
<p>sculptors may be blessed with a natural three-dimensional visualization aptitude, but they have
</p>
<p>learned to develop it&mdash;particularly by making and handling models.
</p>
<p>Standard practice in the past was to reduce three-dimensional objects to one ormore two-dimensional
</p>
<p>drawings (projections and elevations): it was cheap, well suited to reproduction in books, and less
</p>
<p>cumbersome than handling three-dimensional models. In this book, we shall continue to use such two-
</p>
<p>dimensional representations where appropriate, but to rely on them exclusively only delays the acquisi-
</p>
<p>tion of a three-dimensional visualization ability. Fortunately, we can now use stereoscopic image pairs,
</p>
<p>such as that shown in Fig. 1.6. These illustrations are a great help, but, because they provide a view from
</p>
<p>only one standpoint, they are not always quite the equal of models that can be examined by hand.
</p>
<p>Symmetry
Symmetrymaybe defined as that spatial property of a body (or pattern) bywhich thebody (orpattern) can
</p>
<p>be brought from an initial state to another indistinguishable state by means of a certain operation&mdash;a
</p>
<p>symmetry operation. For our purposes, the operation will be considered to take place in n-dimensional
</p>
<p>space (n &frac14; 1, 2, or 3) and to represent an action with respect to a symmetry element.
</p>
<p>Symmetry Elements and Symmetry Operations
A symmetry element is a geometrical entity (point, line, or plane) in a body or assemblage, with which is
</p>
<p>associated an appropriate symmetry operation. The symmetry element is strictly conceptual, but it is
</p>
<p>convenient to accord it a sense of reality. The symmetry element connects all parts of the body or
</p>
<p>assemblage as a number of symmetrically related parts. The term assemblage is often useful because it
</p>
<p>describes more obviously a bundle of radiating face normals, Fig. 1.17, or a number of bonds emanating
</p>
<p>from a central atom in the case of a molecule or ion, Fig. 1.34, to which these symmetry concepts equally
</p>
<p>apply.
</p>
<p>The symmetry operation corresponding to a symmetry element, when applied to a body, converts
</p>
<p>it to a state that is indistinguishable from the initial state of that body, and thus the operation reveals
</p>
<p>1.4 External Symmetry of Crystals 17</p>
<p/>
</div>
<div class="page"><p/>
<p>the symmetry inherent in the body. In many cases, different symmetry operations can reveal one
</p>
<p>and the same symmetry element. Thus, 31 (&frac14;3), 32, and 33 (&frac14;3) may be regarded as either multiple
steps of 3, a threefold operation (q.v.), or single-step operations in their own, but all are contained
</p>
<p>within the same single symmetry element, 3. The latter idea is of particular importance in the study
</p>
<p>of group theory. Symmetry elements may occur singly in a body, as in Fig. 1.20, for example, or in
</p>
<p>certain combinations, as in the example of Fig. 1.22.
</p>
<p>Point Groups
A set of interacting symmetry operations in a finite body, or just one such element, is referred to as a point
</p>
<p>group. A point group may be defined as a set of symmetry operations the action of which leaves at least
</p>
<p>one point unmoved: this point is taken as the origin of the reference axes for the body, through which all
</p>
<p>symmetry elements pass. The assembly of points defining a rotation axis or a mirror plane is effectively
</p>
<p>unmoved by their operations.
</p>
<p>It can be contended that, in real objects, since they are imperfect, even if only on a microscopic
</p>
<p>scale, an indistinguishable second state can be obtained only by a rotation of 360� (or 0�); this
operation is identity, or &ldquo;doing nothing.&rdquo; For practical purposes, however, the effects of most
</p>
<p>Fig. 1.20 Some two-dimensional objects and their point-group symbols. The motifs are built up from the asymmetric
unit (a), by operating on it according to the point-group symmetry. Note that, except in (a), the symmetry element
intersects the asymmetric unit and lies at the center of each figure
</p>
<p>18 1 Crystal Morphology and Crystal Symmetry</p>
<p/>
</div>
<div class="page"><p/>
<p>imperfections are small, and although our discussion of symmetry will be set up in terms of ideal
</p>
<p>geometrical objects, the extension of the results to real situations is scientifically rewarding.
</p>
<p>The observable symmetry may depend upon the nature of the examining probe, and different
</p>
<p>results for a given material may arise in terms of its optical, magnetic, and photoelastic properties,
</p>
<p>and from neutron diffraction, Sect. 11.1. Here, we shall be concerned with the symmetry shown by
</p>
<p>directions in space, such as the normals to the faces on crystals, or the bond directions in chemical
</p>
<p>species. Such angular relationships can be presented conveniently on stereograms, Sect. 1.3, and we
</p>
<p>shall draw fully on this method of representation in the ensuing discussion.
</p>
<p>Several concepts in symmetry can be introduced conveniently with two-dimensional objects;
</p>
<p>subsequently, the third dimension can be introduced mainly as a geometrical extension of the two-
</p>
<p>dimensional ideas. There is a single one-dimensional point group; it is more difficult to grasp
</p>
<p>conceptually, and we shall not be particularly concerned with it in this book.
</p>
<p>1.4.1 Two-Dimensional Point Groups
</p>
<p>&ldquo;We proceeded straight from plane geometry to solid bodies in motion without considering solid
</p>
<p>bodies first on their own. The right thing is to proceed from second dimension to third, which brings us
</p>
<p>Fig. 1.21 Stereograms of the point groups of the objects in Fig. 1.20; the conventional graphic symbols for R (R &frac14; 1,
2, 3, 4, 6) and m are shown
</p>
<p>1.4 External Symmetry of Crystals 19</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec1_11">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec1_11</a></div>
</div>
<div class="page"><p/>
<p>Fig. 1.22 Further two-dimensional objects with their stereograms and point groups. (a) 2mm. (b) 3m. (c) 4mm.
(d) 6mm
</p>
<p>20 1 Crystal Morphology and Crystal Symmetry</p>
<p/>
</div>
<div class="page"><p/>
<p>to cubes and other three-dimensional figures&rdquo; [26]. If we examine the two-dimensional objects in
</p>
<p>Fig. 1.20, we can discover two types of symmetry elements that can bring an object from one state to
</p>
<p>another indistinguishable state: parts (a) to (e) of Fig. 1.20 depict rotational symmetry, whereas (f)
</p>
<p>shows reflection symmetry.
</p>
<p>Rotation Symmetry
A two-dimensional object possesses rotational symmetry of degreeR (orR-fold symmetry) about a point if
</p>
<p>it can be brought from one state to another indistinguishable state by each and every rotation of (360/R)�
</p>
<p>about that symmetry point. Figure 1.20a&ndash;e illustrate the rotational symmetry elementsR equal to 1, 2, 3, 4,
</p>
<p>and 6, respectively. The onefold element is the identity element and is crystallographically trivial; every
</p>
<p>object has onefold symmetry.
</p>
<p>Reflection Symmetry
A two-dimensional object possesses reflection symmetry, symbol m, if it can be brought from one
</p>
<p>state to another indistinguishable state by reflection across the symmetry line. The operation is not
</p>
<p>one that we can perform physically with an object, unlike rotation, but we can appreciate from the
</p>
<p>object itself (and its stereogram) that m symmetry is present. The m line divides the figure into its
</p>
<p>asymmetric unit, , and a mirror image or enantiomorph of this unit, , which situation (left-
</p>
<p>hand&ndash;right-hand relationship) is characteristic of reflection symmetry, Fig. 1.20f.
</p>
<p>Each of the objects in Fig. 1.20 has a symmetry pattern that can be described by a two-dimensional
</p>
<p>point group, and it is convenient to illustrate these point groups by stereograms. Figure 1.21 shows
</p>
<p>stereograms for the two-dimensional point groups 1, 2, 3, 4, 6, and m. It should be noted that in using
</p>
<p>stereogram-like drawings to illustrate two-dimensional symmetry, the representative points (poles)
</p>
<p>must fall on the perimeter; such situations may represent special forms (q.v.) on the stereograms of
</p>
<p>three-dimensional objects.
</p>
<p>Combinations of R and m lead to four more point groups; they are illustrated in Fig. 1.22. We have
</p>
<p>deliberately omittedpoint groups inwhichR &frac14; 5 andR � 7, for a reason thatwill be discussed inChap. 2.
It is convenient to allocate the ten two- dimensional point groups to two-dimensional systems and
</p>
<p>to choose reference axes for the objects in close relation to the directions of their symmetry elements.
</p>
<p>Table 1.1 lists these systems, together with the meanings of the positions in the point-group symbols.
</p>
<p>It should be noted that combinations of m with R (R &gt; 2) introduce additional reflection lines of a
</p>
<p>different crystallographic form. In the case of 3m, however, these additional m lines are coincident
</p>
<p>with the first set; the symbol 3mm is not meaningful.
</p>
<p>Table 1.1 Two-dimensional point groups and notation
</p>
<p>Symbol meaning, appropriate to position occupied
</p>
<p>System Point groups First position Second position Third position
</p>
<p>Oblique 1, 2 Rotation about a point &ndash; &ndash;
</p>
<p>Rectangular 1ma As above m &perp; x &ndash;
2mm As above m &perp; x m &perp; y
</p>
<p>Square As above &ndash; &ndash;
4mm As above m &perp; x, y m at 45� to x, y
</p>
<p>Hexagonal 3 As above &ndash; &ndash;
3m As above m &perp; x, y, u &ndash;
6 As above &ndash; &ndash;
6mm As above m &perp; x, y, u m at 30� to x, y, u
</p>
<p>aUsually written as m, but the full symbol is given here in order to clarify the positions of the symmetry elements in the
</p>
<p>symbol
</p>
<p>1.4 External Symmetry of Crystals 21</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_2">http://dx.doi.org/10.1007/978-1-4614-3954-7_2</a></div>
</div>
<div class="page"><p/>
<p>It is important to remember the relative orientations of the symmetry elements in the point groups
</p>
<p>and the variations in the meanings of the positions in the different systems. In the two-dimensional
</p>
<p>hexagonal system, three axes may be chosen in the x, y plane; this selection corresponds with the
</p>
<p>use of Miller&ndash;Bravais indices in three dimensions.
</p>
<p>1.4.2 Three-Dimensional Point Groups
</p>
<p>The symmetry elements encountered in three dimensions are rotation axes (R), inversion axes (R),
</p>
<p>and a reflection (mirror) plane (m). A center of symmetry can be present also, although this symmetry
</p>
<p>element may be included in a point group that contains the element R, as we shall see, and a check for
</p>
<p>this must be made.
</p>
<p>The operations of rotation and reflection are similar to those in two dimensions, except that the
</p>
<p>geometric extensions of the operations are now increased to rotation about a line and reflection across
</p>
<p>a plane, respectively.
</p>
<p>Inversion Axes
An object is said to possess an inversion (strictly, roto-inversion) axis4 R if it can be brought from one
</p>
<p>state to another indistinguishable state by the combined actions of rotation by (360/R)� about the axis
and inversion through a point on the axis that also serves as the origin point; the two actions comprise
</p>
<p>a single symmetry operation. Like mirror symmetry, Sect. 1.4.1, the inversion axis is a physically
</p>
<p>non-performable symmetry operation on a model, but it may be represented conveniently on a
</p>
<p>stereogram. It is a little more difficult to envisage this operation than those of rotation and reflection.
</p>
<p>Figure 1.23 illustrates a hypothetical molecule having a vertical �4 axis: the stereoscopic effect can be
</p>
<p>created by using a stereoviewer (see Appendix A for instructions for making a model with �4
</p>
<p>symmetry). In crystals, R in R, can take only the values 1, 2, 3, 4, and 6; this is sometimes referred
</p>
<p>to as the crystallograpic restriction theorem. A simple explanation for this restriction is that only
</p>
<p>figures that are based on these rotational symmetries can be stacked together to fill space completely
</p>
<p>in a periodic manner, as Fig. 1.27 shows; see also Sect. 1.4.3. A further discussion of these restrictions
</p>
<p>on R is given in Sect. 2.6.
</p>
<p>In pictorial representations of the three-dimensional point groups, it is helpful to indicate the third
</p>
<p>dimension on their stereograms and, in addition, to illustrate the change-of-hand relationship that
</p>
<p>Fig. 1.23 Stereoview of a
hypothetical C4 molecule;
</p>
<p>the �4 axis is in the vertical
direction
</p>
<p>4Read as &ldquo;bar-R&rdquo;, or &ldquo;R-bar&rdquo; in the United States.
</p>
<p>22 1 Crystal Morphology and Crystal Symmetry</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec14_2">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec14_2</a></div>
</div>
<div class="page"><p/>
<p>occurs with R (including m) symmetry operations. For example, referring to Fig. 1.24, the element
</p>
<p>2 lying in the plane of projection and the element �4 normal to the plane of projection, when acting on a
</p>
<p>point derived from the upper hemisphere (symbol ●), both move the point into the lower hemisphere
</p>
<p>region (symbol O). Both operations involve a reversal of the sign of the vertical coordinate, but only �4
</p>
<p>involves also a change of hand, and this distinction is not clear from the conventional stereogram
</p>
<p>notation. Consequently, we shall adopt a symbolism, shown in Fig. 1.25, that is common to three-
</p>
<p>dimensional space groups, and which will affect the necessary distinction.
</p>
<p>With the modified notation, a representative point in the upper, l-positive hemisphere will now be
</p>
<p>shown by O+, signifying, for example, the face (hkl), or its pole. A change of hemisphere to &eth;hk�l&THORN; will
be indicated by O�, and a change-of-hand on reflection or inversion by or Fig. 1.25. This
</p>
<p>notation may appear to nullify partially the purpose of a stereogram. However, although the
</p>
<p>stereogram is a two-dimensional diagram, it is helpful here to convey a three-dimensional impression
</p>
<p>clearly, and this notation is used as an aid to this end.
</p>
<p>Figure 1.26a shows a stereogram for point group m. The inverse diad &eth;�2&THORN; is lying normal to the m
plane. A consideration of the two operations in the given relative orientations shows that they produce
</p>
<p>equivalent actions. It is conventional to use the symbol m for this operation, although sometimes it is
</p>
<p>helpful to employ the symbol �2 instead; potassium tetrathionate, Fig. 1.26b, crystallizes in point groupm.
</p>
<p>Fig. 1.24 Stereograms of general forms. (a) Point group 2 (axis horizontal and in the plane of the stereogram).
(b) Point group �4 (axis normal to the plane of the stereogram). In (a), the point ● is rotated through 180� to O: (1) and
(2). In (b), the point ● is rotated through 90� and then inverted through the origin to O; this combined operation
generates, in all, four symmetry-equivalent points: (1) ! (4) ! (3) ! (2)
</p>
<p>Fig. 1.25 (a) Stereograms from Fig. 1.24 in the revised notation: the different natures of points (2) in (a) and (2) and
(4) in (b), all with respect to point (1), are now clear
</p>
<p>1.4 External Symmetry of Crystals 23</p>
<p/>
</div>
<div class="page"><p/>
<p>We shall not be concerned here to derive the crystallographic point groups&mdash;and there are several
</p>
<p>ways in which it can be done5&mdash;but to give, instead, a scheme which allows them to be worked
</p>
<p>through simply and adequately for present purposes. In addition, the program EULR, Sect. 13.2,
</p>
<p>shows how the combinations of symmetry operations based on R and R (R &frac14; 1, 2, 3, 4 and 6) lead
inter alia to the 32 crystallographic point groups. The symbols for rotation and reflection symmetry in
</p>
<p>three dimensions are similar to those already discussed; certain additional symbols are now required,
</p>
<p>and Table 1.2 lists them all.
</p>
<p>Crystal Classes
There are 32 crystal classes that describe the possible types of crystals that occur. Each class has a
</p>
<p>name that corresponds to the general form on the crystal, and each class is characterized by a point
</p>
<p>group. A crystal form is the set {hkl} of faces (hkl) related by the point group of the crystal: if it is
</p>
<p>&ldquo;general&rdquo; then none of the (hkl) faces lies on symmetry elements; if it is a special form, then the faces
</p>
<p>of the set lie on symmetry elements.
</p>
<p>Two crystal classes were illustrated in Fig. 1.4: the rhombic disphenoid class (a) belongs to the
</p>
<p>orthorhombic system (to be discussed next) and shows point group 222; it has four similar scalene
</p>
<p>triangle faces and can exist in enantiomorphic forms.An example isMozartite, CaMn(SiO4)(OH),which
was first noted in 1991, the 200th anniversary of the death of Mozart. Another, less exotic, example is
</p>
<p>Epsomite, MgSO4&middot;7H2O. The hexakisoctohedron (cl. Gk. hexakis&mdash;six times; aka hexoctahedron) of
</p>
<p>class (b) belongs to the cubic systemwith point groupm3m and is exhibited by some specimens of native
</p>
<p>silver; each face is a regular octahedron sub-divided into six equal triangular faces.
</p>
<p>Crystal Systems and Point-Group Scheme
Crystals are grouped into seven systems according to the characteristic symmetry listed in Table 1.3.
</p>
<p>The characteristic symmetry refers to that minimum necessary for classification of a crystal in a given
</p>
<p>system; a crystal of a given system may contain more than its characteristic symmetry.
</p>
<p>Fig. 1.26 Point group m. (a) Stereogram showing equivalence of m and �2 (the graphic symbol used here for �2 is
useful, albeit not conventional). (b) Crystal of potassium tetrathionate (K2S4O6), point group m
</p>
<p>5 See Bibliography (Ladd 1989).
</p>
<p>24 1 Crystal Morphology and Crystal Symmetry</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec4_13">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec4_13</a></div>
</div>
<div class="page"><p/>
<p>Fig. 1.27 Sections of three-dimensional figures and the rotational symmetries of their smallest structural units; (a&ndash;d)
Space-filling patterns. In (e) and (f) the v-marks represent voids in the pattern (see also Problem 1.10a)
</p>
<p>Table 1.2 Three-dimensional symmetry symbols
</p>
<p>Symbol Name Action for indistinguishability Graphic symbol
</p>
<p>1 Monad 360� (0�) rotation; identity None
</p>
<p>2 Diad 180� rotation projection, projection
</p>
<p>3 Triad 120� rotation or inclined to projection
</p>
<p>4 Tetrad 90� rotation projection, projection
</p>
<p>6 Hexad 60� rotation projection
�1 Inverse monad Inversion
</p>
<p>a
</p>
<p>�3 Inverse triad 120� rotation + inversion or inclined to projection
�4 Inverse tetrad 90� rotation + inversion projection, projection
�6 Inverse hexad 60� rotation + inversion projection
</p>
<p>m Mirror planeb Reflection across plane projection, projection
</p>
<p>a �R is equivalent to R plus �1 only where R is an odd number: �1 represents the center of symmetry, but �2, �4, and �6 are not
centrosymmetric point groups. For R even, R&thorn; �1 � R=m
bThe symmetry elements m and �2 produce an equivalent operation, with �2 oriented perpendicularly to the mirror plane</p>
<p/>
</div>
<div class="page"><p/>
<p>A crystallographic point-group scheme is given in Table 1.4, under the seven crystal systems as
</p>
<p>headings. Themain difficulty in understanding point groups lies not somuch in knowing the action of the
</p>
<p>individual symmetry elements, but in appreciating both the relative orientation of the different elements
</p>
<p>in a point-group symbol and the fact that this orientation changes among the crystal systems according to
</p>
<p>the principal symmetry axis, that is, the rotation axis R of highest degree. These orientations need to be
</p>
<p>learned: they form the key to point-group and space-group studies.
</p>
<p>Table 1.5 lists the meanings of the three positions in the three-dimensional point-group symbols.
</p>
<p>Tables 1.4 and 1.5 should be studied carefully in conjunction with Fig. 1.32. For example,
</p>
<p>consider carefully point groups 222 and 422, and note how and why the orientations repre-
</p>
<p>sented by the three positions in the symbol change their meanings. In 222, the three symmetry
</p>
<p>axes are along x, y, and z, respectively. In 422, 4 is taken along z, by convention; the first
</p>
<p>symbol 2 (second position in the symbol) represents both the x and y directions, because they
</p>
<p>are equivalent under fourfold symmetry. This combination of 4 and 2 introduces symmetry
</p>
<p>along [110] and &frac12;1�10�, so that the second symbol 2 represents this symmetry. Similar situations
exist among other point groups where the principal symmetry axis is of degree greater than 2.
</p>
<p>The reader should not be discouraged by the wealth of convention which surrounds this part of the
</p>
<p>subject. It arises for two main reasons. First, there are many different, equally correct ways of
</p>
<p>describing crystal geometry. For example, the unique axis in the monoclinic system could be chosen
</p>
<p>as x or z instead of y, or along some arbitrary direction. Secondly, a strict system of notation is
</p>
<p>Table 1.3 Crystal systems and their characteristics
</p>
<p>System
</p>
<p>Characteristic symmetry axes,
</p>
<p>with their orientation
</p>
<p>Parametral plane intercepts and interaxial angles,
</p>
<p>assuming the simplest indexing of facesa,b
</p>
<p>Triclinic None a 6C b 6C c; a 6C b 6C g 6C 90�, 120�
Monoclinic One 2 or �2 axisc along y a 6C b 6C c; a &frac14; g &frac14; 90�; b 6C 90�, 120�
Orthorhombic Three mutually perpendicular
</p>
<p>2 or �2 axes along x, y, and z
a 6C b 6C c; a &frac14; b &frac14; g &frac14; 90�
</p>
<p>Tetragonal One 4 or �4 axis along z a &frac14; b 6C c; a &frac14; b &frac14; g &frac14; 90�
Trigonald One 3 axis along z a &frac14; b 6C c; a &frac14; b &frac14; 90�; g &frac14; 120�
Hexagonal One 6 or �6 axis along z
</p>
<p>Cubic Four 3 axes inclined at 54.74�
</p>
<p>(cos�11=
ffiffiffi
3
</p>
<p>p
) to x, y, and z
</p>
<p>a &frac14; b &frac14; c; a &frac14; b &frac14; g &frac14; 90�
</p>
<p>aWe shall see in Chap. 2 that the same relationships apply to conventional unit cells in lattices
bThe special symbol 6C should be read as &ldquo;not constrained by symmetry to equal&rdquo;
cIt must be remembered that �2 is equivalent to an m plane normal to the �2 axis
dFor convenience, the trigonal system is referred to hexagonal axes (but see also Table 1.5)
</p>
<p>Table 1.4 Crystallographic point-group schemea
</p>
<p>Type Triclinic Monoclinic Trigonal Tetragonal Hexagonal Cubicb
</p>
<p>R 1 2 3 4 6 23
�R �1 m �3 �4 �6 m�3
R + center &mdash; 2/m &mdash; 4/m 6/m &mdash;
</p>
<p>Orthorhombic
R2 222 32 422 622 432
Rm mm2 3m 4mm 6mm
�Rm &mdash; �3m �42m �6m2 �43m
R2 + center mmm &mdash; 4
</p>
<p>m
mm
</p>
<p>6
</p>
<p>m
mm m�3m
</p>
<p>aThe reader should consider the implications of the spaces (marked &mdash;) in this table
bThe cubic system is characterized by its four threefold axes; R refers here to the element 2, �2, 4 or �4, but 3 is always
present along h111i
</p>
<p>26 1 Crystal Morphology and Crystal Symmetry</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_2">http://dx.doi.org/10.1007/978-1-4614-3954-7_2</a></div>
</div>
<div class="page"><p/>
<p>desirable for the purposes of concise and unambiguous communication of crystallographic material.
</p>
<p>With familiarity, the conventions cease to be a problem.
</p>
<p>We now consider two point groups in a little more detail in order to elaborate some of the topics
</p>
<p>discussed so far.
</p>
<p>Point Group mm2
</p>
<p>Once we fix the orientations of two of the symmetry elements in this point group, the third is
</p>
<p>introduced in a unique orientation. Referring to Fig. 1.28, we start with mm &eth;m?x and m?y&THORN; as
shown by the thick lines. Point (1), in a general position, is reflected across the m plane perpendicular
</p>
<p>to the x axis (mx) to give point (2). This point is now reflected across the second m plane (my) to (3).
</p>
<p>Then either (3) across mx or (1) across my produces (4). It is evident now that the points in each of the
</p>
<p>pairs (1), (3) and (2), (4) are related by the twofold rotation axis along z, which is, here, the line of
</p>
<p>intersection of the two m planes.
</p>
<p>Point Group 4mm
</p>
<p>Ifwe startwith the fourfold axis along z andm perpendicular to x, we see straightaway that anotherm plane
</p>
<p>(perpendicular to y) is required, Fig. 1.29a, b; the fourfold axis acts on all other symmetry elements in the
</p>
<p>Table 1.5 Three-dimensional point groups and Hermann&ndash;Mauguin notation
</p>
<p>Symbol meaning for each position
</p>
<p>System Point groupsa First position Second position Third position
</p>
<p>Triclinic 1, �1 All directions
in crystal
</p>
<p>&ndash; &ndash;
</p>
<p>Monoclinicb 2, m,
2
</p>
<p>m
</p>
<p>2 and/or �2
along y
</p>
<p>&ndash; &ndash;
</p>
<p>Orthorhombic 222, mm2,
</p>
<p>mmm
</p>
<p>2 and/or �2
along x
</p>
<p>2 and/or �2 along y 2 and/or �2 along z
</p>
<p>Tetragonal 4, �4,
4
</p>
<p>m
4 and/or �4
along z
</p>
<p>&ndash; &ndash;
</p>
<p>422, 4mm,
</p>
<p>�42m,
4
</p>
<p>m
mm
</p>
<p>4 and/or �4
along z
</p>
<p>2 and/or �2 along x, y 2 and/or �2 at 45� to x, y and in xy
plane, i.e., along h110i
</p>
<p>Cubicc 23, m�3 2 and/or �2
along x, y, z
</p>
<p>3 and/or �3 at 54�440d to x, y, z,
i.e., along h111i
</p>
<p>&ndash;
</p>
<p>432, �43m,
</p>
<p>m�3m
</p>
<p>4 and/or �4
along x, y, z
</p>
<p>3 and/or �3 at 54�440d to x, y, z,
i.e., along h111i
</p>
<p>2 and/or �2 at 45� to x, y, z, i.e., along
h110i
</p>
<p>Hexagonal 6, �6,
6
</p>
<p>m
6 and/or �6
along z
</p>
<p>&ndash; &ndash;
</p>
<p>622, 6mm,
</p>
<p>�6m2,
6
</p>
<p>m
mm
</p>
<p>6 and/or �6
along z
</p>
<p>2 and/or �2 along x, y, u 2 and/or �2 perpendicular to x, y, u
and in xy plane
</p>
<p>Trigonale 3, �3 3 and/or �3
along z
</p>
<p>&ndash; &ndash;
</p>
<p>32, 3m, �3m 3 and/or �3
along z
</p>
<p>2 and/or �2 along x, y, u &ndash;
</p>
<p>aR/m occupies a single position in a point-group symbol because only one direction is involved
bIn the monoclinic system, the y axis is taken as the unique 2 or �2 axis. Since �2 � m, then if �2 is along y, the m plane
represented by the same position in the point-group symbol is perpendicular to y. The latter comment applies mutatis
</p>
<p>mutandis in other crystal systems (it is best to specify the orientation of a plane by that of its normal)
cEarlier notation uses m3 and m3m for m�3 and m�3m, respectively
dActually cos�1&eth;1=
</p>
<p>ffiffiffi
3
</p>
<p>p
&THORN;
</p>
<p>eFor convenience; the trigonal system is referred to hexagonal axes; on the axes of a rhombohedral unit cell (q.v.), the
</p>
<p>orientations of the first and second positions of the symbol are [111] and h1�10i, respectively
</p>
<p>1.4 External Symmetry of Crystals 27</p>
<p/>
</div>
<div class="page"><p/>
<p>crystal aswell as on crystal faces.Ageneral point operated on by the symmetry 4mproduces eight points in
</p>
<p>all, Fig. 1.29c. The stereogram shows that a second formof (vertical)m planes, lying at 45� to the first set is
introduced, Fig. 1.29d.More concisely,wemay say that the normals to the two forms ofm planes lie at 45�
</p>
<p>to one another. No further points are introduced by the second set of m planes: a fourfold rotation
</p>
<p>(1) ! (2), followed by reflection across the mirror plane normal to the x axis, (2) ! (3), is equivalent
to reflectionof the original point across amirror at 45� to x, (1) ! (3).The reader should now refer again to
Table 1.5 for the relationship between the positions of the symmetry elements and the point-group
</p>
<p>symbols, particularly for the tetragonal and orthorhombic systems, from which these detailed examples
</p>
<p>have been drawn.
</p>
<p>In this discussion, we have used a general form {hkl} to illustrate the point group, and each
</p>
<p>symmetry-equivalent point lies in a general position, point-group symmetry 1, on the stereogram of
</p>
<p>the group. The crystal planes that coincide with symmetry planes or symmetry axes are special forms,
</p>
<p>and their poles lie on symmetry elements: the forms {110} and {010} in 4mm are examples of special
</p>
<p>forms. The need for the general form in a correct description of a point group is illustrated by Fig. 1.30.
</p>
<p>The poles of the faces on each of the two stereograms shown are identical, although theymay be derived
</p>
<p>from crystals in different classes, �42m and 4mm in this example (and also in
4
</p>
<p>m
mm).
</p>
<p>Figure 1.31 shows crystals of these two classes with the {110} form, among others, developed.
</p>
<p>In Fig. 1.31b, the presence of only special forms led originally to an incorrect deduction of the point
</p>
<p>group of this crystal.
</p>
<p>The stereograms for the 32 crystallographic point groups are shown in Fig. 1.32. The conventional
</p>
<p>crystallographic axes are drawn once for each system. Two comments on the notation are necessary at
</p>
<p>this stage.
</p>
<p>The symbol &minus; + indicates two points, O&thorn; and , immediately below it and related by a mirror
</p>
<p>plane in the plane of projection. In the cubic crystal system, the four points related by a fourfold
</p>
<p>lying axis in the plane of the stereogram lie on a stereographic small circle, which is a circle on the
</p>
<p>surface of the sphere that does not pass through the center of the sphere, Fig. 1.33. In general, two of
</p>
<p>the points are projected from the upper hemisphere and the other two points from the lower
</p>
<p>hemisphere. We can distinguish them readily by remembering that 2 is a subgroup (q.v.) of both
</p>
<p>4 and �4.
</p>
<p>The use of the program SYMM for assisting with point-group recognition is described in Sect. 13.3,
</p>
<p>and the reader may wish to refer forward at this stage. Appendix B discusses the Sch&euro;onflies symmetry
</p>
<p>Fig. 1.28 Stereogram, symmetry elements, and general form for point group mm2
</p>
<p>28 1 Crystal Morphology and Crystal Symmetry</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec5_13">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec5_13</a></div>
</div>
<div class="page"><p/>
<p>notation for point groups. Because this notation is also in use in certain contexts, we have written the
</p>
<p>equivalent Sch&euro;onflies symbols in Fig. 1.32, in parentheses, after the Hermann&ndash;Mauguin symbols.
</p>
<p>Subgroups, Laue Groups, Centrosymmetric Groups, and Projection Symmetry
Subgroups
</p>
<p>A subgroup of a given point group is a point group of lower symmetry than the given
</p>
<p>group, contained within it and capable of separate existence as a point group. For example, 32 is
</p>
<p>a subgroup of �3m, 622, �6m2,
6
</p>
<p>m
mm, 432 and m3m, whereas �4 is a subgroup of
</p>
<p>4
</p>
<p>m
, �42m,
</p>
<p>4
</p>
<p>m
mm, �43m,
</p>
<p>and m3m. The subgroup principle provides a rationale for some of the graphic symbols
</p>
<p>for symmetry elements. Thus, �4 is shown by a square (fourfold rotation), unshaded (to distinguish
</p>
<p>it from 4), and with a twofold rotation symbol inscribed (2 is a subgroup of �4).
</p>
<p>Fig. 1.29 Intersecting
symmetry elements.
</p>
<p>(a) One m plane
intersecting the fourfold
</p>
<p>axis is inconsistent.
</p>
<p>(b) Consistent group of
symmetry elements.
</p>
<p>(c) General form of
points generated by 4m.
</p>
<p>(d) Complete stereogram,
point group 4mm
</p>
<p>Fig. 1.30 The {110} form
in tetragonal point groups.
</p>
<p>(a) Point group �42m.
(b) Point group 4mm
</p>
<p>1.4 External Symmetry of Crystals 29</p>
<p/>
</div>
<div class="page"><p/>
<p>Laue Groups
</p>
<p>Point group �1 and point groups that have �1 as a subgroup are centrosymmetric. We shall see later that
</p>
<p>X-ray diffraction patterns are, in the absence of significant anomalous dispersion (q.v.), effectively
</p>
<p>centrosymmetric, so that the arrangement of spots on the X-ray diffraction photograph of a crystal can
</p>
<p>exhibit only the symmetry that would be found from a crystal having the corresponding centrosym-
</p>
<p>metric point group. In the case of a crystal belonging to a non-centrosymmetric point group, the
</p>
<p>corresponding centrosymmetric point group is simply the given group combined with a center of
</p>
<p>symmetry.
</p>
<p>There are 11 such point groups; they are called Laue groups,6 since symmetry is often investigated
</p>
<p>by the Laue X-ray method, Sect. 5.4.1ff. In Table 1.6, the point groups are classified according to their
</p>
<p>Laue group, and the symmetry of the Laue flat-plate film photograph is given for directions of the
</p>
<p>X-ray beam normal to the crystallographic forms listed.
</p>
<p>Laue-Projection Symmetry
</p>
<p>The Laue-projection symmetry corresponds to one of the ten two-dimensional points groups. What is
</p>
<p>the Laue-projection symmetry on {110} for a crystal of point group 4mm? This question can be
</p>
<p>answered with the stereogram of the corresponding Laue group,
4
</p>
<p>m
mm. Reference to the appropriate
</p>
<p>diagram in Fig. 1.32 shows that an X-ray beam traveling normal to {110} encounters 2mm symmetry.
</p>
<p>The entries in Table 1.6 can be deduced in this way. The reader should refer again to Table 1.5 and
</p>
<p>compare corresponding entries between Tables 1.5 and 1.6
</p>
<p>Fig. 1.31 Tetragonal
crystals showing, among
</p>
<p>others, the {110} form. (a)
Copper pyrites, point group
�42m. (b) Iodosuccinimide,
apparent point group 4mm;
</p>
<p>X-ray photographs
</p>
<p>revealed that the true point
</p>
<p>group is 4
</p>
<p>6 Strictly, the term Laue group should be Laue class, but the former is in general use.
</p>
<p>30 1 Crystal Morphology and Crystal Symmetry</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec15_5">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec15_5</a></div>
</div>
<div class="page"><p/>
<p>Point-Group Projection Symmetry
</p>
<p>Point-group projection symmetry is the symmetry of the projection of the general form of a point
</p>
<p>group on to a plane. Thus, the point-group projection symmetry of 4mm on {110} is m.
</p>
<p>Non-crystallographic Point Groups
There are species that exhibit symmetries other than those of the crystallographic point groups.
</p>
<p>Indeed, R could, in principle, take any integer value between one and infinity. The statement R &frac14; 1
implies cylindrical symmetry; the molecule of carbon monoxide has an 1 axis along the C&ndash;O bond,
if we assume spherical atoms.
</p>
<p>A fivefold symmetry axis is present in uranium heptafluoride, Fig. 1.34, and the point-group
</p>
<p>symbol may be written as 10m2, or
5
</p>
<p>m
m. The stereogram of this point group is shown in Fig. 1.35;
</p>
<p>the graphic symbol used here for
5
</p>
<p>m
is not standard.
</p>
<p>Other examples of non-crystallographic point groups will be encountered among chemical
</p>
<p>molecules, and a stereogram can always be used to represent the point-group symmetry. In every
</p>
<p>such example, however, the substance crystallizing in one of the seven crystal systems will normally
</p>
<p>belong to one of the 32 crystal classes.
</p>
<p>Fig. 1.32 Stereograms showing both the symmetry elements and the general form {hkl} in the 32 crystallographic
point groups. The arrangement is by system and common Laue group. The crystallographic axes are named once for
</p>
<p>each system and the z axis is chosen normal to the stereogram. The Sch&euro;onflies symbols are given in parentheses
</p>
<p>1.4 External Symmetry of Crystals 31</p>
<p/>
</div>
<div class="page"><p/>
<p>1.4.3 Quasicrystals, Buckyballs, and Icosahedral Symmetry
</p>
<p>Quasicrystals
Our main task in this chapter has been the description of classical crystallography, in which a crystal is
</p>
<p>defined as an ideally infinite three-dimensional periodic arrangement of atoms, the periodicities being
</p>
<p>denoted by the directions of three crystallographic reference axes. Thus, we obtain a crystal structure,
</p>
<p>with building blocks of atoms (unit cells, q.v.) aligned so as to fill space. Most crystal structures can be
</p>
<p>described in terms of one of the 230 space groups (q.v.), which show the different types of symmetry
</p>
<p>elements present in the structures.
</p>
<p>Until the year 1982, no scientist anywhere would believe in crystals with rotational symmetry
</p>
<p>degrees other than 1, 2, 3, 4, and 6, as we have discussed earlier: crystals had only these symmetries and
</p>
<p>were periodic in three dimensions. In that year, however, Professor Daniel Shechtman7 was
</p>
<p>Fig. 1.32 (continued)
</p>
<p>7Nobel Laureate in Chemistry, 2011.
</p>
<p>32 1 Crystal Morphology and Crystal Symmetry</p>
<p/>
</div>
<div class="page"><p/>
<p>experimenting with an alloy containing aluminum andmanganese, Al6Mn; he was concerned about the
</p>
<p>appearance of certain areas of the metal surface, Fig. 1.36a, and took a transmission electron micros-
</p>
<p>copy (TEM) photograph of that region of the surface. He observed diffraction spots in patterns of ten
</p>
<p>extending over the area of reciprocal space (q.v.) recorded, Fig. 1.36b, and he spent the next 2 years
</p>
<p>investigating an effect that he found hard to believe. Attempts to explain the results by a process
</p>
<p>of crystal twinning were unsuccessful, and after more experimentation he was forced to the conclusion
</p>
<p>of having discovered fivefold symmetry in a crystalline material. The crystal structure is aperiodic: it
</p>
<p>Fig. 1.32 (continued)
</p>
<p>1.4 External Symmetry of Crystals 33</p>
<p/>
</div>
<div class="page"><p/>
<p>Fig. 1.33 Stereogram notation for points related by a fourfold axis (y) lying in the plane of a stereogram. The � signs
refer to the z direction. (a) Vertical section normal to the y axis. (b) Corresponding stereogram; the pairs of points
(1)&ndash;(2) and (3)&ndash;(4) are related by twofold symmetry (subgroup of 4)
</p>
<p>Table 1.6 Laue groups and Laue-projection symmetry
</p>
<p>System Point groups Laue group Laue-projection symmetry normal to the given form
</p>
<p>{100} {010} {010}
</p>
<p>Triclinic 1, �1 �1 1 1 1
</p>
<p>Monoclinic 2, m, 2/m 2/m m 2 m
</p>
<p>Orthorhombic 222, mm2, mmm mmm 2mm 2mm 2mm
</p>
<p>{001} {100} {110}
</p>
<p>Tetragonal 4; �4 and 4=m
</p>
<p>422; 4mm;
</p>
<p>�42m;
4
</p>
<p>m
mm
</p>
<p>)
4/m 4 m m
</p>
<p>4
</p>
<p>m
mm 4mm 2mm 2mm
</p>
<p>{0001} f10�10g f11�20g
Trigonala 3, �3 �3 3 1 1
</p>
<p>32, 3m, �3m �3m 3m m 2
</p>
<p>Hexagonal 6; �6; 6=m
</p>
<p>622; 6mm;
</p>
<p>�6m2;
6
</p>
<p>m
mm
</p>
<p>)
6/m 6 m m
</p>
<p>6
</p>
<p>m
mm 6mm 2mm 2mm
</p>
<p>{100} {111} {110}
</p>
<p>Cubic 23, m3 m3 2mm 3 m
</p>
<p>432, �43m, m3m m3m 4mm 3m 2mm
</p>
<p>aReferred to hexagonal axes
</p>
<p>Fig. 1.34 Stereoview of the molecule of uranium heptafluoride, UF7</p>
<p/>
</div>
<div class="page"><p/>
<p>Fig. 1.35 Stereogram of the non-crystallographic point group 10m2 (D5h) showing the general form (20 poles), and a
special form of 5 poles lying on the m planes that can be used to represent the five F atoms in one plane in UF7. The
</p>
<p>poles for the remaining two F atoms lie at these center of the stereogram, on the 10 axis
</p>
<p>Fig. 1.36 Transmission electronmicroscopy (TEM) on anAl6Mnalloy. (a) Surface of theAl6Mnalloy, with irregularities
inviting further examination. (b) Copy of Original TEM photograph of the alloy surface taken by Professor Shechtman,
showing a tenfold spot pattern indicative of a crystalline nature, but without the periodicity of classical crystals (reproduced
</p>
<p>by courtesy of Professor Shechtman)</p>
<p/>
</div>
<div class="page"><p/>
<p>is space-filling, but with none of the three-dimensional periodicity that characterizes normal crystals.
</p>
<p>His first paper on these findings was rejected, but a detailed account was published in 1984 [27].
</p>
<p>This alloy structure is denoted a quasicrystal: it has icosahedral symmetry, point group 532;
</p>
<p>Fig. 1.37 is a simulated Laue pattern for X-ray diffraction from an icosahedral crystal, with the X-ray
</p>
<p>beam along a fivefold axis.
</p>
<p>Subsequently,many stable andmetastable quasicrystals have been discovered, frequently as binary or
</p>
<p>ternary intermetallic compounds containing aluminum as one of the constituents. Icosahedral quasicrys-
</p>
<p>tals form one group of quasicrystals and polygonal quasicrystals, two-dimensional quasicrystals with
</p>
<p>8-fold (Mn&ndash;Fe&ndash;Al), 10-fold (Al&ndash;Cu&ndash;Ni), or 12-fold (Al&ndash;Mn&ndash;Si) symmetry yet another. The former
</p>
<p>group is manifested in the occurrence of sharp diffraction spots and the latter by the presence of a non-
</p>
<p>crystallographic rotational symmetry. Figure 1.38 is an example of a simulated zero-layer X-ray
</p>
<p>precession photograph (q.v.) from a polygonal crystal showing 12-fold symmetry (decagonal symmetry)
</p>
<p>[28]. Two-dimensional quasicrystals symmetry have been reported for rapidly cooled samples
</p>
<p>corresponding to the compositions V3Ni2 and V15Ni10Si: TEM studies showed that they exhibit
</p>
<p>12-fold rotational symmetry but no long-range periodicity [29].
</p>
<p>Two-dimensional quasiperiodical structures have been said to occur in medieval mosques and other
</p>
<p>decorative tilings. Penrose [30] demonstrated the covering of plane space in a non-periodic manner by
</p>
<p>using two differently shaped units, or tiles, and Fig. 1.39 is an example of a Penrose tiling which shows
</p>
<p>two superimposed layers of fivefold symmetry. This tiling was extended to three dimensions and,
</p>
<p>subsequently, a similarity was discovered between Penrose three-dimensional tiling and icosahedral
</p>
<p>quasicrystals. Mackay has shown experimentally [31] that the diffraction pattern from a Penrose plane
</p>
<p>tiling has a two-dimensional Fourier transform consisting of sharp d-peaks arranged in a fivefold
</p>
<p>symmetry pattern.
</p>
<p>In 2009, naturally occurring quasicrystals were found in Russia [32]; in composition they were
</p>
<p>Cu&ndash;Al&ndash;Zn minerals with varying amounts of iron, including an Al63Cu24Fe13 phase; the quasicrystal
</p>
<p>grains were stated to be of high crystalline quality [33].
</p>
<p>Mathematically, the structures of quasicrystals are derivable by a general method that treats them
</p>
<p>as projections of lattices of higher dimensions. The icosahedral quasicrystals found by Shechtman
</p>
<p>were shown to be projections from a six-dimensional hypercubic lattice [34]. Whereas three integer
</p>
<p>Fig. 1.37 Simulated Laue photograph of an icosahedral quasicrystal with the X-ray beam along a fivefold axis
(reproduced by courtesy of Dr. Steffen Weber)
</p>
<p>36 1 Crystal Morphology and Crystal Symmetry</p>
<p/>
</div>
<div class="page"><p/>
<p>values, the Miller indices, are sufficient to label reflections in normal crystals, five linearly indepen-
</p>
<p>dent vectors are needed for polygonal quasicrystals and six for icosahedral quasicrystals.
</p>
<p>As a result of the findings on polygonal crystals and quasicrystals, the International Union of
</p>
<p>Crystallography revised the definition of &ldquo;crystal&rdquo; as given in Sect. 1.2 to a material capable of
</p>
<p>producing a clear-cut diffraction pattern, with ordering that is either periodic or aperiodic. The
</p>
<p>concept of an aperiodic crystal was introduced by Schr&euro;odinger [35]. He sought to explain how
</p>
<p>hereditary information is stored: molecules were deemed too small, amorphous solids were plainly
</p>
<p>chaotic, so it had to be a kind of crystal; and as a periodic structure could not encode information, it
</p>
<p>Fig. 1.39 Example of a Penrose tiling [30] that uses just two shapes of tile (reproduced by courtesy of Jeff Preshing)
</p>
<p>Fig. 1.38 Simulated zero-layer X-ray precession photograph of a polygonal crystal showing tenfold (decagonal)
symmetry (reproduced by courtesy of Dr. Steffen Weber)
</p>
<p>1.4 External Symmetry of Crystals 37</p>
<p/>
</div>
<div class="page"><p/>
<p>had to be aperiodic. Later, DNA was discovered and, although not crystalline, it possesses proper-
</p>
<p>ties predicted by Schr&euro;odinger&mdash;a regular but aperiodic molecule [36]. In this book, however, the
</p>
<p>word &lsquo;crystal&rsquo; is used in its traditional sense unless otherwise noted.
</p>
<p>Buckyballs and Icosahedral Symmetry
The discovery of a new form of a chemical element is a very rare event, but this occurred with the
</p>
<p>finding of a new allotropic form of carbon. Mass spectrometric examination of the products of a high-
</p>
<p>energy laser interaction with graphite in a helium atmosphere contained fragments with varying
</p>
<p>numbers of carbon atoms, their distribution depending upon the pressure of helium [37]. A very stable
</p>
<p>carbon atom cluster was found to be C60. Stability arises because a sheet of carbon atoms formed a
</p>
<p>ball, thereby satisfying fully the valence requirements of carbon.
</p>
<p>This situation was met by a structural formation similar to that of a geodesic dome, Fig. 1.40, but
</p>
<p>with interlocking pentagons and hexagons, and completed to a full sphere, Fig. 1.41. The C60
structure was named buckminsterfullerene [37], after Richard Buckminster Fuller, an American
</p>
<p>engineer, who described the geodesic dome in detail. The geodesic dome was actually invented in
</p>
<p>1922 by Walther Bauersfeld of the Zeiss optical company: it is a spherical or near-spherical lattice-
</p>
<p>type surface formed by a network of great circles, or geodesics, on a sphere. The geodesics intersect to
</p>
<p>form a rigid, stress-free triangular structure. Buckminster Fuller developed the mathematics of the
</p>
<p>dome and popularized it.
</p>
<p>For simplicity, near-spherical fullerenes are termed buckyballs, the simplest stable structure being
</p>
<p>buckminsterfullerene [37], C60. This buckyball structure has 32 faces: 20 hexagons and 12 pentagons,
</p>
<p>Fig. 1.41. Molecules that consist entirely of carbon atoms in the form of hollow spheres, ellipsoids,
</p>
<p>and tubes are known as fullerenes. Buckyballs and buckytubes are topics of intense research both in
</p>
<p>pure chemistry, in which fullerenes are manipulated to form compounds, and in technological
</p>
<p>applications, such as carbon nanotubes.
</p>
<p>The C60 fullerene exhibits icosahedral symmetry. As icosahedral symmetry is not compatible with
</p>
<p>translational symmetry, there are no associated crystallographic space groups. Nevertheless, icosa-
</p>
<p>hedral symmetry can be classified conveniently under the Sch&euro;onflies point group system:
</p>
<p>full icosahedral symmetry Ih comprises the following symmetry elements: E (identity), C5, C3, C2,
</p>
<p>i (center of symmetry), S10, S6, s. Thus, the order of the group, the total number of its symmetry
</p>
<p>elements, is 120. Figure 1.42 illustrates icosahedral symmetry; the similarity to C60 is clear in (b).
</p>
<p>Fig. 1.40 Example of a geodesic dome (reproduced by courtesy of Lotus Domes UK)
</p>
<p>38 1 Crystal Morphology and Crystal Symmetry</p>
<p/>
</div>
<div class="page"><p/>
<p>The icosahedral group of lower symmetry I has the elements E, C5, C3, and C2; it is of interest in some
</p>
<p>biological fields as it can represent a chiral structure.
</p>
<p>1.5 Problems
</p>
<p>1.1. The line AC, Fig. P1.1, may be indexed as (12) with respect to the rectangular two-dimensional
</p>
<p>axes x and y. What are the indices of the same line with respect to the axes x0 and y, where the
angle x0 Oy &frac14; 120�? PQ is the parametral line for both sets of axes, and OB/OA &frac14; 2.
</p>
<p>1.2. Write the Miller indices for planes that make the intercepts given below:
</p>
<p>(a) a/2, �b/2, kc.
(b) 2a, b/3, c/2.
</p>
<p>(c) ka, kb, �c.
</p>
<p>Fig. 1.42 Icosahedra. (a) Regular icosahedron of vertex figure 3.3.3.3.3: the vertex figure means that five triangles
meet at a vertex. (b) Truncated icosahedron of vertex figure 5.6.6: one pentagon and two hexagons meet at a vertex. The
symmetry is Ih in each case
</p>
<p>Fig. 1.41 Molecular structure of buckminsterfullerene, a C60 buckyball. There are two different bond lengths: 1.458 Å
for the bonds fusing 5- and 6-membered rings and 1.401 Å for bonds fusing the 6-membered rings [38] (Harrison P,
</p>
<p>McCaw C (2011) Educ Chem 48:113. Reproduced by permission of The Royal Society of Chemistry)
</p>
<p>1.5 Problems 39</p>
<p/>
</div>
<div class="page"><p/>
<p>(d) a, �b, 3c/4.
(e) ka, �b/4, c/3.
(f) �a/4, b/2, �c/3.
</p>
<p>1.3. Evaluate zone symbols for the pairs of planes given below:
</p>
<p>(a) (123), &eth;0�11&THORN;.
(b) &eth;20�3&THORN;, (111).
(c) &eth;41�5&THORN;, &eth;1�10&THORN;.
(d) &eth;�11�2&THORN;, (001).
</p>
<p>1.4. What are theMiller indices of the plane that lies in both of the zones [123] and &frac12;�11�1�?Why are there,
apparently, two answers to this problem and to each part of Problem 1.3?
</p>
<p>1.5. How many different, unique point groups can be obtained from the symbol 422 by replacing one
</p>
<p>or more of the rotation axes by roto-inversion axes of the same degree. Write the standard
</p>
<p>symbols for the unique point groups so derived.
</p>
<p>1.6. Take the cover of a matchbox, Fig. P1.2a.
</p>
<p>(a) Ignore the label, and write down its point group.
</p>
<p>(b) Squash it diagonally, Fig. P1.2b. What is the point group now?
</p>
<p>(c) In each case, what is the point group if the label is not ignored?
</p>
<p>1.7. Draw stereograms to show the general form in each of the point groups deduced in Problems 1.6a
</p>
<p>and b. Satisfy yourself that in 1.6a three, and in 1.6b two, symmetry operations carried out in
</p>
<p>sequence produce a resultant action that is equivalent to another operation in the group.
</p>
<p>1.8. How many planes are there in the forms {010}, f�110g, and f11�3g in each of the point groups 2/
m, �42m, and m3?
</p>
<p>Fig. P1.1 Line referred to rectangular and oblique axes
</p>
<p>Fig. P1.2 Matchbox: (a) normal; (b) squashed
</p>
<p>40 1 Crystal Morphology and Crystal Symmetry</p>
<p/>
</div>
<div class="page"><p/>
<p>1.9. What symmetry would be revealed by the Laue flat-film photographs where the X-ray beam is
</p>
<p>normal to a plane in the form given in each of the examples below?
</p>
<p>Point group Orientation
</p>
<p>(a) �1 {100}
</p>
<p>(b) mm2 {011}
</p>
<p>(c) m {010}
</p>
<p>(d) 422 {120}
</p>
<p>(e) 3 f10�10g
(f) 3m f11�20g
(g) �6 {0001}
</p>
<p>(h) �6m2 {0001}
</p>
<p>(i) 23 {111}
</p>
<p>(j) 432 {110}
</p>
<p>In some examples, it may help to draw stereograms.
</p>
<p>1.10. (a) What is the non-trivial symmetry of the figure obtained by packing a number of equivalent
</p>
<p>but irregular quadrilaterals in one plane?
</p>
<p>(b) What is the symmetry of the Dobermann in Fig. P1.3? This example illustrates how one can
</p>
<p>study symmetry by means of everyday objects.
</p>
<p>(c) What is the point group of each of the objects in Fig. P1.4a&ndash;e, assuming that they all have
</p>
<p>depth, normal to the plane of the diagram?
</p>
<p>1.11. Write the point-group symbol for the species (a) to (v) of molecule or ion in Fig. P1.5, in both the
</p>
<p>Hermann&ndash;Mauguin and Sch&euro;onflies notations. Use the program SYMM* with this question for
</p>
<p>species (a) to (j) and allocate model numbers as follow:
</p>
<p>(a) 90 (b) 49 (c) 3 (d) 18 (e) 42 (f) 91 (g) 30 (h) 16 (i) 65 (j) 71.
</p>
<p>Fig. P1.3 Vijentor seal of approval at Valmara, JW
</p>
<p>1.5 Problems 41</p>
<p/>
</div>
<div class="page"><p/>
<p>1.12. What is the point-group projection symmetry for each of the examples in Problem 1.9?
</p>
<p>1.13. What are the two-dimensional &ldquo;Miller Indices&rdquo; (hk) of the lines that form the perimeter of a
</p>
<p>rectangle? Are they the same for a parallelogram? (Choose appropriate x and y axes with the
</p>
<p>origin at the center of the figure.)
</p>
<p>1.14. Show from Fig. P1.6 that with Miller&ndash;Bravais axes i &frac14; �&eth;h&thorn; k&THORN;.
1.15. Twelve 1 O resistors are linked to form the edges of a cube, Fig. P1.7. Use the cubic symmetry
</p>
<p>to determine the effective resistance of the assembly of resistors to a current that is developed
</p>
<p>by connecting a battery across the ends of a threefold axis.
</p>
<p>Fig. P1.4 Six symmetrical objects. (a) Dumbbell. (b) Isle of Man style emblem (reproduced by courtesy of Mapsof-
World.com). (c) Maltese cross. (d) Flower pattern (reproduced by courtesy of Emeritus Professor John Huffman).
(e) Star of David
</p>
<p>42 1 Crystal Morphology and Crystal Symmetry</p>
<p/>
</div>
<div class="page"><p/>
<p>Fig. P1.5 (a) Carbonate ion; planar. (b) Tetrabromogold(III) ion; planar. (c) Hexachloroplatinum(IV) ion; all
Cl&ndash;Pt&ndash;Cl angles are 90�. (d) Methane all H&ndash;C&ndash;H angles are 109.47�. (e) Trichloromethane; pyramidal.
(f) Bromochlorofluoromethane. (g) Benzene; planar. (h) Monochlorobenzene; planar. (i) 1,4-Dichlorobenzene; planar.
(j) 1,2-Bromochlorobenzene; planar. (k) Hydrogen peroxide. (l) Orhophosphoric acid. (m) Dibenzyl.
(n) Hexanitronickelate(II) ion. (o) Dihydrogen phosphate ion; the hydrogen atoms are arranged statistically on
the four positions shown. (p) 1,2,4,-Trichlorobenzene; planar. (q) 2,4,6-Triazidotriazine; planar. (r) trans-1,2-Dichlo-
roethane; planar. (s) Cycloocta-1,5-diene. (t) Tetranitrodiamminocobaltate(III) ion. (u) Pentafluoroantimonate(III) ion.
(v) Thorium tetrabromide
</p>
<p>1.5 Problems 43</p>
<p/>
</div>
<div class="page"><p/>
<p>Fig. P1.5 (continued)
</p>
<p>44 1 Crystal Morphology and Crystal Symmetry</p>
<p/>
</div>
<div class="page"><p/>
<p>Fig. P1.5 (continued)
</p>
<p>1.5 Problems 45</p>
<p/>
</div>
<div class="page"><p/>
<p>Fig. P1.5 (continued)
</p>
<p>46 1 Crystal Morphology and Crystal Symmetry</p>
<p/>
</div>
<div class="page"><p/>
<p>Fig. P1.5 (continued)
</p>
<p>1.5 Problems 47</p>
<p/>
</div>
<div class="page"><p/>
<p>Fig. P1.5 (continued)
</p>
<p>48 1 Crystal Morphology and Crystal Symmetry</p>
<p/>
</div>
<div class="page"><p/>
<p>References and Bibliography
</p>
<p>1. Exodus 28:17&ndash;20; Revelation 21:19&ndash;20
</p>
<p>2. Gessner C (1564) De rerum Fossilium, Lapidum et Gemmarum, Figuris. Z&euro;urich
3. Pliny the Elder (77&ndash;79) Naturis histori&aelig;. Rome
</p>
<p>4. Caesalpinus A (1596) De metallicis libri tres. Rome
</p>
<p>Fig. P1.6 Equivalence of i
and �(h &thorn; k) with
Miller&ndash;Bravais axes
</p>
<p>Fig. P1.7 The cube shown
consists of twelve 1 O
</p>
<p>resistors; the current I
</p>
<p>follows a path from A to G
</p>
<p>References and Bibliography 49</p>
<p/>
</div>
<div class="page"><p/>
<p>5. Buffon G-LL (1807) Buffon&rsquo;s natural history. London
</p>
<p>6. Stensen N (Steno N) (1669) De solido intra solidum naturaliter contento dissertationis prodromus, book 3.
</p>
<p>Florence
</p>
<p>7. Bartholinus E (1669) Experimenta Crystallici Islandici. Copenhagen
</p>
<p>8. Kepler J (1611) Strena Seu de Nive Sexangula. Frankfurt
</p>
<p>9. Guglielmini D (1688) Riflessioni filosofiche dedotte dalle figure de Sali. Bologna
</p>
<p>10. Romé de l&rsquo;Isle J-BL (1783) Cristallographie. Paris
</p>
<p>11. Carangeot A (1782) Nouvelles de la Rèpublique des lettres et des arts, no. 14. Paris
</p>
<p>12. Ha&euro;uy RJ (Abbé Ha&euro;uy) (1801) Traité élémentaire de physique. Paris
13. Weiss CS (1815) Abhandlungen, der K&euro;oniglichen Akademie der Wissenschaften. Berlin
14. Mohs F (1822) Grund-Rib der Mineralogie. Dresden
</p>
<p>15. Hessel JFC (1830) Kristall. Gehlers Physikalische W&euro;orterbuch. Leipzig
16. Frankenheim ML (1842) Nova Acta Acad, vol 19. von Grass, Breslau
</p>
<p>17. Miller WH (1839) Treatise on crystallography. Cambridge
</p>
<p>18. Bravais A (1846) Mem Acad R Sci Inst France 9:255
</p>
<p>19. Federov Y (1891) Zap Mineral Obch 28:1
</p>
<p>20. Sch&euro;onflies A (1891) Kristallsysteme und Kristallstruktur. Leipzig
21. Barlow W (1894) Z Kristallogr 23:1
</p>
<p>22. Freidrich W, Knipping P, Laue M (1912) M&euro;unchener Ber:303; (1913) Ann Phys 41:917
23. Bragg WL (1913) Camb Philos Soc 17:43
</p>
<p>24. Bragg WH (1913) Proc R Soc 88A:428
</p>
<p>25. Carlisle CH, Ladd MFC (1966) Acta Crystallogr 21:689
</p>
<p>25a. http://proteopedia.org/wiki/index.php/Nobel_Prizes _for_3D_Molecular_Structure
</p>
<p>26. Plato (1955) The republic, book VII, circa 360 B. C. E. (trans: Lee HDP). Penguin Books, Harmondsworth
</p>
<p>27. Shechtman D et al (1984) Phys Rev Lett 53:1951
</p>
<p>28. &euro;Unal BV et al (2007) Phys Rev B 75:064205
29. Chen H et al (1988) Phys Rev Lett 60:1645
</p>
<p>30. Penrose R (1976) Sets of tiles for covering a surface. US Patent 413315
</p>
<p>31. Mackay AL (1982) Physica A114:609
</p>
<p>32. Bindi L et al (2009) Science 324:1306
</p>
<p>33. Steinhardt P et al (2011) Philos Mag 91:2421
</p>
<p>34. Kramer P et al (1984) Acta Crystallogr A40:580
</p>
<p>35. Schr&euro;odinger E (1944) What is life. Cambridge University Press, Cambridge
36. http://tc.engr.wisc.edu/uer/uer96/author9/index.html
</p>
<p>37. Kroto H et al (1985) Nature 318:162
</p>
<p>38. Hedberg K et al (1991) Science 254:410
</p>
<p>Bibliography: General and Historical Study of Crystallography
</p>
<p>Bragg WL (1949) A general survey, vol I, The crystalline state. Bell, London
</p>
<p>Ewald PP (ed) (1962) Fifty years of X-ray diffraction. Oosthoek, Utrecht
</p>
<p>Hauptmann HA (1990) Structural chemistry, vol 1. Springer, New York
</p>
<p>Lonsdale K (1948) Crystals and X-rays. Bell, London
</p>
<p>Crystal Morphology and Stereographic Projection
</p>
<p>Phillips FC (1971) An introduction to crystallography. Longmans, London
</p>
<p>Crystal Symmetry and Point Groups
</p>
<p>Hahn T (ed) (2002) International tables for crystallography, vol A, 5th edn. Kluwer Academic, Dordrecht
</p>
<p>Henry NFM, Lonsdale K (eds) (1965) International tables for X-ray crystallography, vol I. Kynoch Press, Birmingham.
</p>
<p>http//:it.iucr.org
</p>
<p>Ladd MFC (1989) Symmetry in molecules and crystals. Ellis Horwood, Chichester
</p>
<p>Ladd Mark (1998) Symmetry and group theory in chemistry. Horwood Publishing, Chichester
</p>
<p>50 1 Crystal Morphology and Crystal Symmetry</p>
<p/>
<div class="annotation"><a href="http://proteopedia.org/wiki/index.php/Nobel_Prizes_for_3D_Molecular_Structure">http://proteopedia.org/wiki/index.php/Nobel_Prizes_for_3D_Molecular_Structure</a></div>
<div class="annotation"><a href="http://tc.engr.wisc.edu/uer/uer96/author9/index.html">http://tc.engr.wisc.edu/uer/uer96/author9/index.html</a></div>
<div class="annotation"><a href="http://http//:it.iucr.org">http://http//:it.iucr.org</a></div>
</div>
<div class="page"><p/>
<p>Lattices and Space-Group Theory 2
</p>
<p>2.1 Introduction
</p>
<p>We continue our study of crystals by investigating the internal arrangements of crystalline materials.
</p>
<p>Crystals are characterized by periodicities in three dimensions.1 An atomic grouping, or pattern motif
</p>
<p>which, itself, may or may not be symmetrical, is repeated again and again by a symmetry mechanism,
</p>
<p>namely the space group of the crystal. There are 230 space groups, and each crystal substance belongs
</p>
<p>to one or other of them. In its simplest form, a space group may be derived from the repetition of a
</p>
<p>pattern motif by the translations of a lattice, as discussed below. It can be developed further by
</p>
<p>incorporating additional symmetry elements, as demonstrated through the following text and Problem
</p>
<p>2.1. We now enlarge on these ideas, starting with an examination of lattices.
</p>
<p>2.2 Lattices
</p>
<p>Every crystal has a lattice as its geometrical basis. A lattice may be described as a regular, infinite
</p>
<p>arrangement of points in space in which every point has exactly the same environment as any other
</p>
<p>point. This description is applicable, equally, in one-, two-, or three-dimensional space.
</p>
<p>Lattice geometry in three-dimensional space is described in relation to three noncoplanar basic
</p>
<p>repeat (translation) vectors a, b, and c. Any lattice point may be chosen as an origin, whence a vector
r to any other lattice point is given by
</p>
<p>r &frac14; Ua&thorn; Vb&thorn;Wc (2.1)
</p>
<p>where U, V, andW are positive or negative integers or zero, and represent the coordinates of the given
</p>
<p>lattice point. The direction (directed line) joining the origin to the points U, V, W; 2U, 2V, 2W; . . .;
</p>
<p>nU, nV, nW defines the row [UVW]. A set of such rows, or directions, related by the symmetry
</p>
<p>constitutes a form of directions hUVWi; compare with zone symbols, Sect. 1.2.5. The magnitude r can
be evaluated by (2.16) mutatis mutandis.2
</p>
<p>1We shall not be concerned here with the aperiodic crystalline materials discussed in Sect. 1.4.3.
2 &ldquo;The necessary changes having been made.&rdquo;
</p>
<p>M. Ladd and R. Palmer, Structure Determination by X-ray Crystallography:
</p>
<p>Analysis by X-rays and Neutrons, DOI 10.1007/978-1-4614-3954-7_2,
# Springer Science+Business Media New York 2013
</p>
<p>51</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Sec7_1">http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Sec7_1</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Sec12_1">http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Sec12_1</a></div>
</div>
<div class="page"><p/>
<p>We consider first lattices in two dimensions; the three-dimensional lattices then become an
</p>
<p>extension of the principles that evolve, rather like the symmetry operations discussed in the previous
</p>
<p>chapter.
</p>
<p>2.2.1 Two-Dimensional Lattices
</p>
<p>A two-dimensional lattice is called a net; it may be imagined as being formed by aligning, in a regular
</p>
<p>manner, one-dimensional rows of equally spaced points, Fig. 2.1a. The net (lattice) is the array of
</p>
<p>points; the connecting lines are a convenience, drawn to aid our appreciation of the lattice geometry.
</p>
<p>Since nets exhibit symmetry, they can be allocated to the two-dimensional systems, Sect. 1.4.1,
</p>
<p>Table 1.1. The most general net is shown in Fig. 2.1b. A sufficient and representative portion of the
</p>
<p>lattice is the unit cell, outlined by the vectors a and b; an infinite number of such unit cells stacked
side by side builds up the net.
</p>
<p>The net under consideration exhibits twofold rotational symmetry about each lattice point; conse-
</p>
<p>quently, it is placed in the oblique system. The chosen unit cell is primitive, symbol p, which implies that
</p>
<p>one lattice point is associated with the area of the unit cell: each point is shared equally by four adjacent
</p>
<p>unit cells. In the oblique unit cell, a 6C3 b, and g 6C 90 or 120�; angles of 90 or 120� in a lattice imply
symmetry higher than 2.
</p>
<p>Consider next the stacking of unit cells in which a 6C b but g &frac14; 90�, Fig. 2.2. The symmetry at
every point is 2mm, and this net belongs to the rectangular system. The net in Fig. 2.3 may be
</p>
<p>described by a unit cell in which a0 &frac14; b0 and g0 6C 90 or 120�. It may seem at first that such a net is
oblique, but careful inspection shows that each point has 2mm symmetry, and so this net, too, is
</p>
<p>allocated to the rectangular system.
</p>
<p>In order to display this fact clearly, a centered (symbol c) unit cell is chosen, shown in Fig. 2.3 by
</p>
<p>the vectors a and b. This cell has two lattice points per unit-cell area. It is left as an exercise to the
reader to show that a centered, oblique unit cell does not represent a net with a fundamentally
</p>
<p>different arrangement of points from that in Fig. 2.1b.
</p>
<p>Fig. 2.1 Formation of a net. (a) Row (a one-dimensional lattice) of equally spaced points. (b) Regular stack of rows
forming a net
</p>
<p>3The symbol 6C should be read as &ldquo;not constrained by symmetry to equal.&rdquo;
</p>
<p>52 2 Lattices and Space-Group Theory</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Sec10_1">http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Sec10_1</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Tab1_1">http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Tab1_1</a></div>
</div>
<div class="page"><p/>
<p>2.2.2 Choice of Unit Cell
</p>
<p>From the foregoing discussion, it will be evident that there is an infinity ofways inwhich a unit cellmight
</p>
<p>be chosen for a given lattice (and structure). However, we shall follow a universal crystallographic
</p>
<p>convention in choosing a unit cell: the unit cell is the smallest repeat unit for which its delineating vectors
</p>
<p>are parallel to, or coincide with, important symmetry directions in the lattice. Returning to Fig. 2.3, the
</p>
<p>centered cell is preferred because a and b coincide with the symmetry (m) lines in the net. The primitive
unit cell (a0, b0) is, of course, a possible unit cell, but it does not, in isolation, reveal the lattice symmetry
clearly. The symmetry is still there; it is invariant under choice of unit cell. The following equations show
</p>
<p>the necessary equivalence of a0 and b0:
</p>
<p>a0 2 &frac14; a2=4&thorn; b2=4 (2.2)
</p>
<p>b0 2 &frac14; a2=4&thorn; b2=4 (2.3)
</p>
<p>the value of g0 depends only on the ratio a/b.
Two other nets exist, governed by the unit-cell relationships a &frac14; b, g &frac14; 90� and a &frac14; b, g &frac14; 120�;
</p>
<p>their study constitutes the Problem 1.2 at the end of this chapter. The five two-dimensional lattices are
</p>
<p>summarized in Table 2.1. A lattice has the highest point-group symmetry of its system at each lattice
</p>
<p>point: compare Table 2.1 with Table 1.1 and Table 2.3 with Table 1.5.
</p>
<p>Fig. 2.3 Rectangular net with p and c unit cells drawn in; the c unit cell is the standard choice for this net
</p>
<p>Fig. 2.2 Rectangular net with a p unit cell drawn in
</p>
<p>2.2 Lattices 53</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Tab1_1">http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Tab1_1</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Tab5_1">http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Tab5_1</a></div>
</div>
<div class="page"><p/>
<p>2.2.3 Three-Dimensional Lattices
</p>
<p>The three-dimensional lattices, or Bravais lattices, may be imagined as being developed by the regular
</p>
<p>stacking of nets. There are 14 unique ways in which this can be done, and the corresponding Bravais
</p>
<p>lattices are distributed, unequally, among the seven crystal systems, as shown in Fig. 2.4. Each lattice is
</p>
<p>represented by a unit cell, outlined by three vectors a, b, and c. In accordance with convention, these
vectors are chosen so that they both form a parallelepipedon of smallest volume in the lattice and are
</p>
<p>parallel to, or coincide with, important symmetry directions in the lattice; thus, not all conventional unit
</p>
<p>cells are primitive. In three dimensions, we encounter unit cells centered on a pair of opposite faces,
</p>
<p>body-centered, or centered on all faces. Table 2.2 lists the unit-cell types and their notation.
</p>
<p>Fractional Coordinates
A fractional coordinate x is given by X/a, where X is that coordinate in absolute measure (Å or nm)
</p>
<p>and a is the unit-cell repeat distance in the same direction and in the same units. Thus, a position x at
</p>
<p>1.45 Å along a unit cell of edge of length 12.34 Å corresponds to a fractional coordinate of 0.1175.
</p>
<p>Triclinic Lattice
If oblique nets are stacked in a general and regular manner, a triclinic lattice is obtained, Fig. 2.5.
</p>
<p>The unit cell is characterized by �1 symmetry at each lattice point, with the conditions a 6C b 6C c and
a 6Cb 6C g 6C 90 or 120�. This unit cell is primitive (symbol P), which means that one lattice point is
associated with the unit-cell volume; each point is shared equally by eight adjacent unit cells in
</p>
<p>three dimensions; refer to Fig. 2.6 for this sharing principle. There is no symmetry direction to
</p>
<p>constrain the choice of the unit-cell vectors, and a parallelepipedon of smallest volume can always
</p>
<p>be chosen conventionally.
</p>
<p>Monoclinic Lattices
The monoclinic system is characterized by one diad (rotation or inversion), with the y axis (and b)
</p>
<p>chosen along or parallel to it. The conventional unit cell is specified by the conditions a 6C b 6C c,
a &frac14; g &frac14; 90�, and b 6C 90 or 120�. Figure 2.6 illustrates a stereoscopic pair of drawings of a monoclinic
lattice, showing eight P unit cells; according to convention, the b angle is chosen to be oblique.
</p>
<p>Reference to Fig. 2.4 shows that there are two conventional monoclinic lattices, symbolized by the
</p>
<p>unit-cell types P and C.
</p>
<p>A monoclinic unit cell centered on the A faces is equivalent to that described as C; the choice of the
</p>
<p>b axis4 is governedby symmetry:a and cmaybe interchanged, but thedirectionofbmust thenbe reversed
in order to preserve right-handed axes.
</p>
<p>Table 2.1 The five two-dimensional lattices
</p>
<p>System
</p>
<p>Unit-cell
</p>
<p>symbol(s)
</p>
<p>Symmetry at
</p>
<p>lattice points
</p>
<p>Unit-cell edges
</p>
<p>and angles
</p>
<p>Oblique p 2 a 6C b; g 6C 90�, 120�
Rectangular p, c 2mm a 6C b; g &frac14; 90�
Square p 4mm a &frac14; b ; g &frac14; 90�
Hexagonal p 6mm a &frac14; b ; g &frac14; 120�
</p>
<p>4We often speak of the b axis (meaning the y axis) because our attention is usually confined to the unit cell.
</p>
<p>54 2 Lattices and Space-Group Theory</p>
<p/>
</div>
<div class="page"><p/>
<p>Fig. 2.4 Unit cells of the 14 Bravais lattices; interaxial angles are 90� unless indicated otherwise by a numerical value
or symbol. (1) Triclinic P. (2) Monoclinic P. (3) Monoclinic C. (4) Orthorhombic P. (5) Orthorhombic C. (6)
</p>
<p>Orthorhombic I. (7) Orthorhombic F. (8) Tetragonal P. (9) Tetragonal I. (10) Cubic P. (11) Cubic I. (12) Cubic F.
</p>
<p>(13) Hexagonal P. (14) Trigonal R. Note that (13) shows three P hexagonal unit cells. A hexagon of lattice points
</p>
<p>without the central points in the basal planes shown does not lead to a lattice. Why?
</p>
<p>2.2 Lattices 55</p>
<p/>
</div>
<div class="page"><p/>
<p>The centering of the B faces is illustrated in Fig. 2.7. In this situation a new unit cell, a0, b0, c0, can
be defined by the following equations:
</p>
<p>a0 &frac14; a (2.4)
</p>
<p>b0 &frac14; b (2.5)
</p>
<p>c0 &frac14; a=2&thorn; c=2 (2.6)
</p>
<p>Table 2.2 Notation for conventional crystallographic unit cells
</p>
<p>Centering site(s) Symbol
</p>
<p>Miller indices of
</p>
<p>centred faces in the
</p>
<p>unit cell
</p>
<p>Fractional coordinates
</p>
<p>of centered sites
</p>
<p>in the unit cell
</p>
<p>None P &ndash; &ndash;
</p>
<p>bc faces A (100) 0; 1
2
; 1
2
</p>
<p>ca faces B (010) 1
2
; 0; 1
</p>
<p>2
</p>
<p>ab faces C (001) 1
2
; 1
2
; 0
</p>
<p>Body center I &ndash; 1
2
; 1
2
; 1
2
</p>
<p>All faces F (100), (010),
</p>
<p>(001)
0; 1
</p>
<p>2
; 1
2
</p>
<p>1
2
; 0; 1
</p>
<p>2
1
2
; 1
2
; 0
</p>
<p>8
&gt;&lt;
</p>
<p>&gt;:
</p>
<p>Fig. 2.5 Oblique nets stacked regularly at a vector spacing c to form a triclinic lattice
</p>
<p>56 2 Lattices and Space-Group Theory</p>
<p/>
</div>
<div class="page"><p/>
<p>If b is not very obtuse, an equivalent transformation c0 &frac14; � a/2 + c/2 can ensure that b0 is obtuse
(by convention). Since c0 lies in the ac plane, a0 &frac14; g0 &frac14; 90�, but b0 6C 90 or 120�. The new monoclinic
cell is primitive; symbolically we may write B � P. Similarly, it may be shown that I � F � C �
(A), Figs. 2.8 and 2.9.
</p>
<p>If the C unit cell, Fig. 2.10, is reduced to primitive as shown, it no longer displays in isolation the
</p>
<p>characteristic monoclinic symmetry clearly (see Table 2.3); neither a0 nor g0 is 90�. We may conclude
that there are two distinct monoclinic lattices, described by the unit-cell types P and C.
</p>
<p>Fig. 2.6 Stereoview showing eight adjacent P unit cells in a monoclinic lattice. The sharing of lattice points among the
unit cells can be seen readily by focusing attention on the central lattice point in the drawings. A similar sharing occurs
</p>
<p>with P unit cells of lattices in all systems
</p>
<p>Fig. 2.7 Monoclinic lattice showing that B � P; b is the angle between c and a, and b0 the angle between c0 and a0
</p>
<p>2.2 Lattices 57</p>
<p/>
</div>
<div class="page"><p/>
<p>It may be necessary to calculate the new dimensions of a transformed unit cell. Consider the
</p>
<p>transformation B ! P, (2.4)&ndash;(2.6). Clearly, a0 &frac14; a and b0 &frac14; b. Taking the scalar product5 of (2.6)
with itself, we obtain
</p>
<p>Fig. 2.8 Monoclinic lattice showing that I � C
</p>
<p>Fig. 2.9 Monoclinic lattice showing that F � C
</p>
<p>5The scalar (dot) product of two vectors p and q is denoted by p�q, and is equal to pq coscpq, wherecpq represents the
angle between the (positive) directions of p and q.
</p>
<p>58 2 Lattices and Space-Group Theory</p>
<p/>
</div>
<div class="page"><p/>
<p>c0 � c &frac14; &eth;a=2&thorn; c=2&THORN; � &eth;a=2&thorn; c=2&THORN; (2.7)
</p>
<p>Hence
</p>
<p>c02 &frac14; a2=4&thorn; c2=4&thorn; ac&eth;cos b&THORN;=2 (2.8)
</p>
<p>The new angle b0 is given by
</p>
<p>cos b0 &frac14; a0 � c0=a0c0 (2.9)
</p>
<p>In order to make b0 obtuse, it may be necessary to begin with �a/2 in (2.6).
Using (2.6) again and expanding, we obtain
</p>
<p>cos b0 &frac14; &frac12;�a=2&thorn; c&eth;cos b&THORN;=2�=c0 &frac14; &eth;�a&thorn; c cos b&THORN; &eth;2c0&THORN;= (2.10)
</p>
<p>Fig. 2.10 Monoclinic lattice showing that C ≢ P
</p>
<p>Table 2.3 The 14 Bravais lattices and their notation
</p>
<p>System Unit cell(s)
</p>
<p>Symmetry at
</p>
<p>lattice points Axial relationships
</p>
<p>Triclinic Pa 1 a 6C b 6C c; a 6C b 6C g 6C 90�; 120�
Monoclinic P, C 2/m a 6C b 6C c; a &frac14; g &frac14; 90�; b 6C 90�; 120�
Orthorhombic P, C, I, F mmm a 6C b 6C c; a &frac14; b &frac14; g &frac14; 90�
Tetragonal P, I 4
</p>
<p>m
mm
</p>
<p>a &frac14; b 6C c; a &frac14; b &frac14; g &frac14; 90�
</p>
<p>Cubic P, I, F m3m a &frac14; b &frac14; c; a &frac14; b &frac14; g &frac14; 90�
Hexagonal P 6
</p>
<p>m
mm
</p>
<p>a &frac14; b 6C c; a &frac14; b &frac14; 90�; g &frac14; 120�
</p>
<p>Trigonalb R or P 3m a &frac14; b &frac14; c; a &frac14; b &frac14; g 6C 90�; &lt;120�
aCapital letters are used for unit cells in three-dimensional lattices
bOn hexagonal axes, column 4 would be the same as for the hexagonal system, but the symmetry at each lattice point
</p>
<p>remains �3m. This table may be compared with Table 1.3
</p>
<p>2.2 Lattices 59</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Tab3_1">http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Tab3_1</a></div>
</div>
<div class="page"><p/>
<p>where c0 is given by (2.8). This type of calculation can be carried out in any crystal system, giving due
consideration to any nontrivial relationships between a, b, and c and between a, b, and g (see, for
</p>
<p>example, Problem 2.3).
</p>
<p>Orthorhombic Lattices
The monoclinic system was treated in some detail. It will not be necessary here to give such an
</p>
<p>extensive discussion for either the orthorhombic system or the remaining crystal systems. Remember
</p>
<p>always to think of the unit cell as a representative portion of its lattice and not as a finite body.
</p>
<p>The orthorhombic system is characterized by three mutually perpendicular diad axes (rotation and/or
</p>
<p>inversion); the unit-cell vectors are chosen to be parallel to, or to coincide with, these axes.
</p>
<p>The orthorhombic unit cell is specified by the relationships a 6C b 6C c and a &frac14; b &frac14; g &frac14; 90�. It will
not be difficult for the reader to verify that the descriptorsP,C, I, andF are necessary and sufficient in this
</p>
<p>system. One way in which this exercise may be carried out is as follows. After centering the P unit cell,
</p>
<p>four questions must be considered, in the following order:
</p>
<p>1. Does the centered unit cell represent a lattice?
</p>
<p>2. If so, is its symmetry, in isolation, different from that of the P unit cell?
</p>
<p>3. If the symmetry is unchanged, is the lattice different in type (arrangement of points) from the
</p>
<p>lattice or lattices already determined for the given system?
</p>
<p>4. Has the unit cell been chosen correctly?
</p>
<p>Notice that we answered these questions implicitly in discussing the monoclinic lattices.
</p>
<p>The descriptors A, B, and C do not all remain equivalent for orthorhombic space groups in the class
</p>
<p>mm2; it is necessary to distinguish C from A (or B). The reader may like to consider now, or later, why
</p>
<p>this distinction is necessary.
</p>
<p>Tetragonal Lattices
The tetragonal system is characterized by one tetrad (rotation or inversion) along z (c); the unit-cell
</p>
<p>conditions are a &frac14; b 6C c and a &frac14; b &frac14; g &frac14; 90�. There are two tetragonal lattices, specified by the
unit-cell symbols P and I, Fig. 2.4; C and F tetragonal unit cells may be transformed to P and I,
</p>
<p>respectively, see also Problem 2.4.
</p>
<p>Cubic Lattices
The symmetry of the cubic system is characterized by four triad axes at angles of cos�1&eth;1=3&THORN; to one
another, or cos �1&eth;1=
</p>
<p>ffiffiffi
3
</p>
<p>p
&THORN; to x, y, and z; they are the body diagonals h111i of a cube; the unit-cell
</p>
<p>conditions are a &frac14; b &frac14; c; a &frac14; b &frac14; g &frac14; 90�. The four threefold axes, in this orientation, introduce
twofold axes along h100i; fourfold axes exist in three of the five cubic classes. There are three cubic
Bravais lattices, Fig. 2.4, with conventional unit cells P, I, and F.
</p>
<p>Hexagonal Lattice
The basic feature of a hexagonal lattice is that it should be able to accommodate a 6- or 6-fold symmetry
</p>
<p>axis. This requirement is achieved by a lattice based on a P unit cell, with a &frac14; b 6C c, a &frac14; b &frac14; 90�, and
g &frac14; 120�, the c direction being taken along the unique axis in the lattice.
</p>
<p>Lattices in the Trigonal System
A two-dimensional unit cell in which a &frac14; b and g &frac14; 120� is compatible with either sixfold or
threefold symmetry; see Fig. 2.22, plane groups p6 and p3. For this reason, the hexagonal lattice (P
</p>
<p>unit cell) may be used for certain crystals which belong to the trigonal system. However, as shown in
</p>
<p>Fig. 2.11, the presence of two threefold axes within a unit cell, with x, y coordinates of 2
3
; 1
3
and 1
</p>
<p>3
; 2
3
, and
</p>
<p>parallel to the z axis, introduces the possibility of a lattice which, although belonging to the trigonal
</p>
<p>system, has a triply primitive unit cell Rhex, with lattice points at
2
3
; 1
3
; 1
3
, and 1
</p>
<p>3
; 2
3
; 2
3
(in addition to 0, 0, 0)
</p>
<p>in the unit cell. Thus, for some trigonal crystals the unit cell will be P, and for others it will be Rhex, the
</p>
<p>60 2 Lattices and Space-Group Theory</p>
<p/>
</div>
<div class="page"><p/>
<p>latter being distinguished by systematically absent X-ray reflections, Table 3.2. The Rhex cell can be
</p>
<p>transformed to a primitive rhombohedral unit cell R, with a &frac14; b &frac14; c and a &frac14; b &frac14; g 6C 90 and&lt;120�;
the threefold axis is then along [111]. The R cell may be thought of as a cube extended (or squashed)
</p>
<p>along one of its threefold axes.
</p>
<p>The lattice based on an R unit cell is the only truly exclusive trigonal lattice, the trigonal lattice
</p>
<p>based on a P unit cell being borrowed from the hexagonal system, Table 2.3.
</p>
<p>We note in passing that the symbols P, R, A, B,C, I, and F cannot apply, strictly, to lattices [1]; they
</p>
<p>are unit-cell symbols, and refer to the types of unit cells already chosen to represent their lattices.
</p>
<p>However, terminology such as &ldquo;P lattice&rdquo; is in general use and, as long as it is used with understanding,
</p>
<p>is perfectly acceptable.
</p>
<p>Fig. 2.11 Trigonal lattice;
the fractions refer to values
</p>
<p>of chex. (a) Rhombohedral
(R) unit cell in the obverse
</p>
<p>setting developed from a
</p>
<p>triply primitive hexagonal
</p>
<p>(Rhex) unit cell. In the
</p>
<p>reverse setting, the
</p>
<p>rhombohedral lattice and
</p>
<p>unit cell are rotated about
</p>
<p>[111] 60� clockwise with
respect to the Rhex axes.
</p>
<p>The ratio of the volumes
</p>
<p>of any two unit cells in one
</p>
<p>and the same lattice is
</p>
<p>equal to the ratio of the
</p>
<p>numbers of lattice points in
</p>
<p>the two unit-cell volumes.
</p>
<p>In the reverse, setting, the
</p>
<p>lattice points in the unit cell
</p>
<p>lie at 2
3
; 1
3
; 1
3
and 1
</p>
<p>3
; 2
3
; 2
3
.
</p>
<p>(b) Plan view of (a) as seen
along chex
</p>
<p>2.2 Lattices 61</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Tab2_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Tab2_3</a></div>
</div>
<div class="page"><p/>
<p>2.3 Families of Planes and Interplanar Spacings
</p>
<p>Figure 2.12 shows one unit cell of an orthorhombic lattice projected on to the a, b plane. The trace of the
</p>
<p>(110) plane nearest the originO is indicated by a dashed line, and the perpendicular distance of this plane
</p>
<p>from O is d(110). By repeating the operation of the translation �d(110) on the plane (110), a series, or
family, of parallel, equidistant planes is generated, as shown in Fig. 2.13. Miller indices, Sect. 1.2.3, are
</p>
<p>by definition prime to one another: in discussing X-ray diffraction effects, however, it is necessary to
</p>
<p>consider planes for which the indices h, k, and l may contain a common factor while still making
</p>
<p>intercepts a/h, b/k, and c/l on the x, y, and z axes, respectively, as required by the definition of Miller
</p>
<p>indices. It follows that the plane with indices (nh, nk, nl) makes intercepts a/nh, b/nk, and c/nl along x, y,
</p>
<p>and z, respectively, and that this plane is nearer to the origin by a factor of 1/n than is the plane (hkl). In
</p>
<p>other words, d(nh, nk, nl) &frac14; d(hkl)/n.
In general, we denote a family of planes as (hkl) where h, k, and l may contain a common factor.
</p>
<p>For example, the (220) family of planes is shown in Fig. 2.14 with interplanar spacing d(220) &frac14; d
(110)/2; alternate (220) planes therefore coincide with (110) planes. Note, that an external crystal face
</p>
<p>normal to d(hh0) would always be designated (110), since external observations reveal the shape but
</p>
<p>not the size for the unit cell.
</p>
<p>Fig. 2.12 One P unit cell in an orthorhombic lattice in projection on (001), showing the trace of the (110) plane
</p>
<p>Fig. 2.13 Family of (110) planes in an orthorhombic lattice, as seen in projection along c
</p>
<p>Fig. 2.14 Family of (220) planes in an orthorhombic lattice, as seen in projection along c
</p>
<p>62 2 Lattices and Space-Group Theory</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Sec5_1">http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Sec5_1</a></div>
</div>
<div class="page"><p/>
<p>2.4 Reciprocal Lattice: Geometrical Treatment
</p>
<p>Although we shall discuss the reciprocal lattice in detail in the next chapter, it is useful to introduce it
</p>
<p>here, because there exists a reciprocal lattice for each of the Bravais lattices. The reciprocal lattice, a
</p>
<p>lattice in reciprocal (diffraction) space, is derived here graphically from the Bravais lattice, a lattice in
</p>
<p>real (direct) space, and we choose the monoclinic system for an example.
</p>
<p>Figure 2.15a represents a monoclinic lattice as seen in projection along the y axis, the normal to the
</p>
<p>(010) plane in this example. From the origin O of a P unit cell, lines are drawn normal to families of
</p>
<p>planes (hkl) in real space. We note in passing that the normal to a plane (hkl) does not, in general,
</p>
<p>coincide with the direction [hkl]: see Sect. 2.2. However, there are special cases, such as [010] and the
</p>
<p>normal to (010) in the present example, in which the two directions do coincide.
</p>
<p>Along each line, reciprocal lattice points hkl (no parentheses) are marked off such that the distance
</p>
<p>from the origin to the first point in any line is inversely proportional to the corresponding interplanar
</p>
<p>spacing d(hkl).
</p>
<p>Fig. 2.15 Direct and reciprocal lattices. (a) Monoclinic P, as seen in projection along b, showing three families of
planes. (b) Corresponding reciprocal lattice showing the points representing these three and other families of planes
</p>
<p>2.4 Reciprocal Lattice: Geometrical Treatment 63</p>
<p/>
</div>
<div class="page"><p/>
<p>In three dimensions, we refer to d*(100), d*(010), and d*(001) as a*, b*, and c*, respectively, and so
</p>
<p>define a unit cell in the reciprocal lattice. In general,
</p>
<p>d�&eth;hkl&THORN; &frac14; k=d&eth;hkl&THORN; (2.11)
</p>
<p>where k is a constant. Hence, for the monoclinic system,
</p>
<p>a� &frac14; k=d&eth;100&THORN; &frac14; k=&eth;a sin b&THORN; (2.12)
</p>
<p>From Fig. 2.15a, the scalar product a � a� is given by
</p>
<p>a � a� &frac14; aa� cos&eth;b� 90&THORN; &frac14; ak cos&eth;b� 90&THORN;
a sin b
</p>
<p>&frac14; k (2.13)
</p>
<p>The mixed scalar products, such as a � c� are identically zero, because the angle between a and
c* is 90�.
</p>
<p>The reciprocal lattice points form a true lattice with a representative unit cell outlined by a*, b*, and c*
</p>
<p>which, therefore, involves six reciprocal unit-cell parameters in the most general case, three sides a*, b*,
</p>
<p>and c*, and three angles a*, b*, and g*. The size of the reciprocal unit cell is governed by the choice of the
</p>
<p>constant k. In practice, k may be taken as the wavelength l of the X-radiation used in an experiment, in
</p>
<p>which case reciprocal lattice units are dimensionless.Alternatively,kmaybe taken as unity, inwhich case
</p>
<p>reciprocal lattice units have the dimensions of length�1. The different situations where one or other
convention is used will become clear as we proceed.
</p>
<p>A reciprocal lattice row hkl; 2h,2k,2l; . . . may be considered to be derived from the families of
</p>
<p>planes (nh, nk, nl) with n &frac14; 1, 2, . . ., since d(nh, nk, nl) &frac14; d(hkl)/n. Hence,
</p>
<p>d�&eth;nh; nk; nl&THORN; &frac14; nd�&eth;hkl&THORN; (2.14)
</p>
<p>where d*(hkl) is the distance of the reciprocal lattice point hkl from the origin, expressed in the
</p>
<p>appropriate reciprocal lattice units (RU). Since h, k, and l are the coordinates of reciprocal lattice
</p>
<p>points, the vector d* (hkl) is given by
</p>
<p>d�&eth;hkl&THORN; &frac14; ha� &thorn; kb� &thorn; lc� (2.15)
</p>
<p>Hence, taking the dot product of d*(hkl) with itself, we have
</p>
<p>d�&eth;hkl&THORN; � d�&eth;hkl&THORN; &frac14; d�2&eth;hkl&THORN;
&frac14; h2a�2 &thorn; k2b�2 &thorn; l2c�2
</p>
<p>&thorn; 2klb�c� cos a�
</p>
<p>&thorn; 2lhc�a� cos b�
</p>
<p>&thorn; 2hka�b� cos g� (2.16)
</p>
<p>Now d(hkl) may be obtained from (2.11) and (2.16). Simplifications of (2.16) arise through
</p>
<p>symmetry constraints on the unit-cell vectors in different crystal systems. The reader should check
</p>
<p>the entries in Table 2.4, starting with Table 2.3 and (2.16).
</p>
<p>64 2 Lattices and Space-Group Theory</p>
<p/>
</div>
<div class="page"><p/>
<p>2.5 Unit-Cell Transformations
</p>
<p>Here, we consider the transformations of unit-cell vectors, zone symbols and directions, Miller indices,
</p>
<p>reciprocal unit-cell vectors, and fractional coordinates of sites in the unit cell, all involving no
</p>
<p>change in the origin of the unit cell. Such transformations are necessary when a nonstandard unit cell
</p>
<p>needs to be re-cast in standard form.
</p>
<p>2.5.1 Bravais Unit-Cell Vectors
</p>
<p>Let a, b, and c be transformed to a0, b0, and c0, such that
</p>
<p>a0 &frac14; s11a&thorn; s12b&thorn; s13c
b0 &frac14; s21a&thorn; s22b&thorn; s23c
c0 &frac14; s31a&thorn; s32b&thorn; s33c
</p>
<p>(2.17)
</p>
<p>which may be written in matrix notation as
</p>
<p>a0
</p>
<p>b0
</p>
<p>c0
</p>
<p>2
4
</p>
<p>3
5 &frac14;
</p>
<p>s11 s12 s13
s21 s22 s23
s31 s32 s33
</p>
<p>2
4
</p>
<p>3
5 �
</p>
<p>a
b
c
</p>
<p>2
4
3
5 (2.18)
</p>
<p>or, more concisely, as
</p>
<p>a0 &frac14; S � a (2.19)
</p>
<p>where the dot � here symbolizes matrix multiplication; a and a0 represent the two sets of column
vectors a, b, c, and a0, b0, c0, and S is the 3 	 3 matrix of elements sij. The inverse transformation is
obtained by multiplying both sides of (2.19) by S�1, since S � S�1 &frac14; 1:
</p>
<p>Table 2.4 Expressions for d*2(hkl), and d2(hkl) with k &frac14; 1
System d*2(hkl) d2(hkl)
</p>
<p>Triclinic h2a�2 &thorn; k2b�2 &thorn; l2c�2 &thorn; 2klb�c� cos a�
&thorn; 2lhc�a� cos b� &thorn; 2hka�b� cos g�
</p>
<p>1/d*2(hkl)
</p>
<p>Monoclinic h2a�2 &thorn; k2b�2 &thorn; l2c�2 &thorn; 2hla�c� cos b� 1
sin2b
</p>
<p>h2
</p>
<p>a2
&thorn; l
</p>
<p>2
</p>
<p>c2
� 2hl cos b
</p>
<p>ac
</p>
<p>� �
&thorn; k
</p>
<p>2
</p>
<p>b2
</p>
<p>� ��1
</p>
<p>Orthorhombic h2a�2 &thorn; k2b�2 &thorn; l2c�2 h2
a2
</p>
<p>&thorn; k
2
</p>
<p>b2
&thorn; l
</p>
<p>2
</p>
<p>c2
</p>
<p>� ��1
</p>
<p>Tetragonal &eth;h2 &thorn; k2&THORN;a�2 &thorn; l2c�2 h2 &thorn; k2
a2
</p>
<p>&thorn; l
2
</p>
<p>c2
</p>
<p>� ��1
</p>
<p>Hexagonal and
</p>
<p>trigonal (P)
&eth;h2 &thorn; k2 &thorn; hk&THORN;a�2 &thorn; l2c�2 4&eth;h2 &thorn; k2 &thorn; hk&THORN;
</p>
<p>3a2
&thorn; l
</p>
<p>2
</p>
<p>c2
</p>
<p>� ��1
</p>
<p>Trigonal (R)
</p>
<p>(rhombohedral)
&frac12;h2 &thorn; k2 &thorn; l2 &thorn; 2&eth;hk &thorn; kl&thorn; hl&THORN;&eth;cos a�&THORN;�a�2 a2&eth;TR&THORN;�1, where
</p>
<p>T &frac14; h2 &thorn; k2 &thorn; l2 &thorn; 2&eth;hk &thorn; kl&thorn; hl)
&frac12;&eth;cos2 a� cos a&THORN;=sin2 a� and
</p>
<p>R &frac14; &eth;sin2 a&THORN;=&eth;1� 3 cos2 a&thorn; 2 cos3 a&THORN;
Cubic &eth;h2 &thorn; k2 &thorn; l2&THORN;a�2 h2 &thorn; k2 &thorn; l2
</p>
<p>a2
</p>
<p>� ��1
&frac14; a
</p>
<p>2
</p>
<p>h2 &thorn; k2 &thorn; l2
</p>
<p>2.5 Unit-Cell Transformations 65</p>
<p/>
</div>
<div class="page"><p/>
<p>a &frac14; S�1 � a0 (2.20)
</p>
<p>where S�1 is the matrix
</p>
<p>S�1 &frac14;
t11 t12 t13
t21 t22 t23
t31 t32 t33
</p>
<p>2
4
</p>
<p>3
5 (2.21)
</p>
<p>The elements tij may be obtained by rearranging (2.17), or by the following equations:
</p>
<p>tij &frac14; &eth;�1&THORN;i&thorn;jjMjij=jSj
</p>
<p>jSj &frac14; s11
s22s23
</p>
<p>s32s33
</p>
<p>�����
</p>
<p>�����&thorn; s21
s12s13
</p>
<p>s22s23
</p>
<p>�����
</p>
<p>�����&thorn; s31
s12s13
</p>
<p>s22s23
</p>
<p>�����
</p>
<p>�����
(2.22)
</p>
<p>where jMjij is the minor determinant of S obtained by striking out its jth row and ith column, and jSj is
the determinant value of the matrix S.
</p>
<p>2.5.2 Directions (Zone Symbols)
</p>
<p>From Sect. 2.2, we have
</p>
<p>r &frac14; Ua&thorn; Vb&thorn;Wc (2.23)
</p>
<p>and for the transformed cell
</p>
<p>r &frac14; U0a0 &thorn; V 0b0 &thorn;W0c0 (2.24)
</p>
<p>Thus, from (2.23) and (2.24),
</p>
<p>&frac12;U0V0W0�
a0
</p>
<p>b0
</p>
<p>c0
</p>
<p>0
@
</p>
<p>1
A &frac14; &eth;UVW&THORN;
</p>
<p>a
b
c
</p>
<p>0
@
</p>
<p>1
A &frac14; &eth;UVW&THORN;S�1
</p>
<p>a0
</p>
<p>b0
</p>
<p>c0
</p>
<p>0
@
</p>
<p>1
A (2.25)
</p>
<p>or
</p>
<p>U0V 0W0&eth; &THORN; &frac14; UVW&eth; &THORN;S�1 (2.26)
</p>
<p>Hence, and concisely,
</p>
<p>U0 &frac14; US�1 &frac14; &eth;S�1&THORN;TU (2.27)
</p>
<p>where U and U0 are now column vectors.
</p>
<p>66 2 Lattices and Space-Group Theory</p>
<p/>
</div>
<div class="page"><p/>
<p>Since (S�1)T &frac14; (ST)�1, pre-multiplication of (2.27) by ST leads to
</p>
<p>STU0 &frac14; ST&eth;ST&THORN;�1U &frac14; U (2.28)
or
</p>
<p>U &frac14; ST � U0 (2.29)
</p>
<p>2.5.3 Coordinates of Sites in the Unit Cell
</p>
<p>For any point x, y, z in a unit cell, the vector r from the origin to that point is given by
</p>
<p>r &frac14; xa&thorn; yb&thorn; zc (2.30)
</p>
<p>Comparison of this equation with (2.23), and by a procedure similar to (2.24)&ndash;(2.28), we see that
</p>
<p>coordinates transform as do zone symbols. Thus,
</p>
<p>x0 &frac14; &eth;S�1&THORN;T � x (2.31)
</p>
<p>2.5.4 Miller Indices
</p>
<p>From (2.15) and (2.23), it follows that
</p>
<p>d�&eth;hkl&THORN; � r &frac14; hU &thorn; kV &thorn; lW (2.32)
</p>
<p>Thus, with (2.29),
</p>
<p>d�&eth;hkl&THORN; � r &frac14; &frac12;hkl� �
U
</p>
<p>V
</p>
<p>W
</p>
<p>2
4
</p>
<p>3
5 &frac14; &frac12;hkl� � ST �
</p>
<p>U0
</p>
<p>V0
</p>
<p>W0
</p>
<p>2
4
</p>
<p>3
5 (2.33)
</p>
<p>But also
</p>
<p>d�&eth;h0k0l0&THORN; � r &frac14; &frac12;h0k0l0� �
U0
</p>
<p>V0
</p>
<p>W0
</p>
<p>2
4
</p>
<p>3
5 (2.34)
</p>
<p>because d*(hkl) and d*(h0k0l0) are one and the same vector in the same plane but with different indices.
Hence
</p>
<p>&frac12;h0k0l0� &frac14; &frac12;hkl� � ST (2.35)
</p>
<p>2.5 Unit-Cell Transformations 67</p>
<p/>
</div>
<div class="page"><p/>
<p>Transposing
</p>
<p>h0
</p>
<p>k0
</p>
<p>l0
</p>
<p>2
4
</p>
<p>3
5 &frac14; S �
</p>
<p>h
</p>
<p>k
</p>
<p>l
</p>
<p>2
4
3
5 (2.36)
</p>
<p>or
</p>
<p>h0 &frac14; S � h (2.37)
</p>
<p>where h and h0 are column vectors with components h, k, l and h0, k0, l0, respectively. Thus, Miller
indices transform in the same way as do unit-cell vectors in real space. If we operate on both sides of
</p>
<p>(2.37) by S�1, then
</p>
<p>S�1 � h0 &frac14; S�1 � S � h
</p>
<p>or
</p>
<p>h &frac14; S�1 � h0 (2.38)
</p>
<p>We may note here that if a plane (hkl) lies in the [UVW] zone and the normal to the plane is d�&eth;hkl&THORN;
then d�&eth;hkl&THORN; � r &frac14; 0, then from (2.15) and (2.23), it follows that hU + kV + lW &frac14; 0, which is the Weiss
Zone Law, since products such as a�a* and a�b* are unity and zero, respectively (k &frac14; 1).
</p>
<p>2.5.5 Reciprocal Unit-Cell Vectors
</p>
<p>From (2.15), we develop
</p>
<p>d�&eth;hkl&THORN; &frac14; &frac12;a�b�c�� �
h
</p>
<p>k
</p>
<p>l
</p>
<p>2
64
</p>
<p>3
75
</p>
<p>&frac14; &frac12;a�b�c�� � S�1 �
h0
</p>
<p>k0
</p>
<p>l0
</p>
<p>2
64
</p>
<p>3
75
</p>
<p>(2.39)
</p>
<p>In the transformed reciprocal unit cell
</p>
<p>d�&eth;hkl&THORN; &frac14; &frac12;a0�b0�c0�� �
h0
</p>
<p>k0
</p>
<p>l0
</p>
<p>2
4
</p>
<p>3
5 (2.40)
</p>
<p>so that
</p>
<p>&frac12;a0�b0�c0�� &frac14; &frac12;a�b�c�� � S�1 (2.41)
</p>
<p>68 2 Lattices and Space-Group Theory</p>
<p/>
</div>
<div class="page"><p/>
<p>Transposing
</p>
<p>a0�
</p>
<p>b0�
</p>
<p>c0�
</p>
<p>2
4
</p>
<p>3
5 &frac14; &eth;S�1&THORN;T �
</p>
<p>a�
</p>
<p>b�
</p>
<p>c�
</p>
<p>2
4
</p>
<p>3
5 (2.42)
</p>
<p>or
</p>
<p>a0
� &frac14; &eth;S�1&THORN;T � a� (2.43)
</p>
<p>so that reciprocal unit-cell vectors transform in the same way as do zone symbols.
</p>
<p>As an example of the transformations that we have just derived, let a transformation matrix from
</p>
<p>unit cell 1 to unit cell 2 may be written as
</p>
<p>S &frac14;
1 0 1
</p>
<p>0 1 �2
</p>
<p>1 2 1
</p>
<p>2
4
</p>
<p>3
5
</p>
<p>Given the plane &eth;1�35&THORN; and the site �0.10, 0.15, 0.25 in unit cell 1, determine the corresponding
values for unit cell 2.
</p>
<p>Miller indices: h2 &frac14; h1 &thorn; l1 &frac14; 6
k2 &frac14; k1 � 2l1 &frac14; 13
l2 &frac14; h1 &thorn; 2k1 &thorn; l1 &frac14; 0
</p>
<p>that is, the plane is &eth;613; 0&THORN; in unit cell 2.
For the coordinates we need the matrix (S�1)T. The determinant jSj is 4. Then, applying (2.22),
</p>
<p>S�1 &frac14;
5=4 1=2 �1=4
�1=2 0 1=2
�1=4 �1=2 1=4
</p>
<p>2
4
</p>
<p>3
5
</p>
<p>whereupon the transpose becomes
</p>
<p>&eth;S�1&THORN;T &frac14;
5=4 �1=2 �1=4
1=2 0 �1=2
�1=4 1=2 1=4
</p>
<p>2
4
</p>
<p>3
5
</p>
<p>then the transformed coordinates are
</p>
<p>x2 &frac14; 5x1=4� y1=2� z1=4 &frac14; �0:2625
y2 &frac14; x1=2� z1=2 &frac14; �0:1750
z2 &frac14; �x14&thorn; y1=2&thorn; z1=4 &frac14; 0:1625
</p>
<p>that is, the site �0.2625, �0.1750, 0.1625.
</p>
<p>2.5 Unit-Cell Transformations 69</p>
<p/>
</div>
<div class="page"><p/>
<p>A reciprocal lattice has the same symmetry as the Bravais lattice from which it was deduced. This
</p>
<p>fact may be appreciated from a comparison of the constructions of the reciprocal lattice and the
</p>
<p>stereogram. Both of these constructions are built up from normals to planes, so that the symmetry
</p>
<p>expressed through the poles of a stereogram is the same as that at the reciprocal lattice points, but the
</p>
<p>reciprocal lattice adds dimensions to the representation; see also Sect. 3.4.1.
</p>
<p>The transformations that we have discussed can be summarized by the mnemonic scheme in
</p>
<p>Fig. 2.16, for any matrix M and its inverse M�1. The arrow symbols, such as
</p>
<p>Fig. 2.16 Mnemonic
scheme for operating on a
</p>
<p>matrix or its inverse and its
</p>
<p>inverse; two examples are
</p>
<p>shown
</p>
<p>70 2 Lattices and Space-Group Theory</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec20_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec20_3</a></div>
</div>
<div class="page"><p/>
<p>should be interpreted as a0 in terms of a, and so on. The scheme for the inverse is equivalent to writing,
</p>
<p>x0 &frac14; &eth;M�1&THORN;T � x and then multiplying in the usual manner.
</p>
<p>2.6 Rotational Symmetries of Lattices
</p>
<p>We now discuss analytically the permissible rotational symmetries in the lattices of periodic crystals,
</p>
<p>already stated to be of degrees 1, 2, 3, 4, and 6. In Fig. 2.17, let A and B represent two adjacent lattice
</p>
<p>points, of repeat distance t, in any row. An R-fold rotation axis is imagined to act at each point and to
</p>
<p>lie normal to the plane of the diagram. An anticlockwise rotation of F about A maps B on to B0, and a
clockwise rotation of the same value F about B maps A on to A0. It follows from the geometry of the
figure that AB is parallel to A0B0 and, from the property of lattices, A0B0 &frac14; Jt, where J is an integer.
Lines A0S and B0T are drawn perpendicular to AB, as shown. Hence,
</p>
<p>A0B0 &frac14; TS &frac14; AB� &eth;AT &thorn; BS&THORN; (2.44)
</p>
<p>or
</p>
<p>Jt &frac14; t� 2t cos F (2.45)
</p>
<p>whence
</p>
<p>cos F &frac14; &eth;1� J&THORN;=2 &frac14; M=2 (2.46)
</p>
<p>Fig. 2.17 Rotational symmetry in crystal lattices. Permissible values of F are 360(0), 180, 120, 90, and 60�,
corresponding to one-, two-, three-, four-, and sixfold rotations, respectively
</p>
<p>2.6 Rotational Symmetries of Lattices 71</p>
<p/>
</div>
<div class="page"><p/>
<p>where M is another integer. Since � 1
 cos F
 1, it follows from (2.46) that the only admissible
values forM are 0, �1, �2, and these values give rise to the rotational symmetries already discussed.
This treatment gives a quantitative aspect to the packing considerations mentioned previously,
</p>
<p>Sect. 1.4.2.
</p>
<p>2.7 Space Groups
</p>
<p>In order to extend our study of crystals further into the realm of atomic arrangements, we must
</p>
<p>consider now the symmetry of extended, ideally infinite, patterns in space. We recall that a point
</p>
<p>group describes the symmetry of a finite body, and that a lattice constitutes a mechanism for
</p>
<p>repetition, to an infinite extent, by translations parallel to three noncoplanar directions. We may
</p>
<p>ask, therefore, what is the result of repeating a point-group pattern by the translations of a Bravais
</p>
<p>lattice? It is a space group, and we shall see that it produces an arrangement like atoms in a crystal.
</p>
<p>A space group can be described as an infinite set of symmetry elements, the operation with
</p>
<p>respect to any of which brings the infinite array of points to which they refer into a state that is
</p>
<p>indistinguishable from that before the operation. In practice, we may apply space-group rules to
</p>
<p>crystals because the dimensions of crystals used in experimental investigations are very large in
</p>
<p>comparison with the repeat distances of the pattern. For example, the dimension a of the face-
</p>
<p>centered cubic unit cell of sodium chloride is 0.564 nm. Thus, in a crystal of experimental size
</p>
<p>(ca. 0.2, 0.2, 0.2 mm), there are approximately 4.5 	 1016 unit cells.
A space group may be considered to be made up of two parts, a pattern motif and a repeat
</p>
<p>mechanism. An analogy can be drawn with a wallpaper-type pattern, a simple example of which is
</p>
<p>shown in Fig. 2.18a. We shall analyze this pattern.
</p>
<p>The conventional unit cell for this pattern is indicated by the vectors a and b. If we choose a pattern
motif consisting of two flowers, Fig. 2.18b, and continue it indefinitely by the repeat vectors a and b,
</p>
<p>Fig. 2.18 Wallpaper-type
pattern. (a) Extended pat-
tern. (b) Asymmetric unit,
or pattern motif; the space-
</p>
<p>group symmetry applied to
</p>
<p>the asymmetric unit gener-
</p>
<p>ates the infinite pattern
</p>
<p>72 2 Lattices and Space-Group Theory</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Sec11_1">http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Sec11_1</a></div>
</div>
<div class="page"><p/>
<p>the plane pattern is generated. However, we have ignored the symmetry between the two flowers in
</p>
<p>the pattern motif itself. If one flower (1) is reflected across the dashed line (g) to (10) and then translated
by a/2, it then occupies the position of the second flower (2); thus, the pattern represented in Fig. 2.18a
is brought from one state to another indistinguishable state by this symmetry operation. This operation
</p>
<p>takes place across a glide line, a symmetry element that occurs in some extended two-dimensional
</p>
<p>patterns. The two motions constitute a single symmetry operation.
</p>
<p>The necessary and sufficient pattern motif for a whole, extended figure is a single flower,
</p>
<p>occupying the asymmetric unit&mdash;the unshaded (or shaded) portion of Fig. 2.18b. If the single flower
</p>
<p>is repeated by both the glide-line symmetry and the unit-cell translations, that is, overall by the space-
</p>
<p>group symmetry, then the infinitely extended pattern is generated. Thus, if we know the asymmetric
</p>
<p>unit of a crystal structure, which need not be the whole unit-cell contents, and the space-group symbol
</p>
<p>for the crystal, we can generate the whole structure.
</p>
<p>2.7.1 Two-Dimensional Space Groups (Plane Groups)
</p>
<p>Oblique System
Our discussion leads naturally into two-dimensional space groups, or plane groups. Consider the pattern
</p>
<p>motif showing twofold symmetry, illustrated in Fig. 2.19a; the symmetry symbols that we have used in
</p>
<p>point groups are continued into the realm of space groups. Next, consider a primitive oblique net,
</p>
<p>Fig. 2.19b; it is of infinite extent in the plane, and the framework of lines divides the field, conceptually,
</p>
<p>into a number of identical primitive (p) unit cells. An origin is chosen at a lattice point; it could be
</p>
<p>anywhere in the unit cell, but is desirably, and conventionally, linked to a symmetry element.
</p>
<p>Now, let themotif be repeated around eachpoint in thenet, and in the sameorientation,with the twofold
</p>
<p>rotation points of the motif and the net in coincidence, Fig. 2.19c. It will be seen that additional twofold
</p>
<p>rotation points are introduced at the unique fractional coordinates 0; 1
2
; 1
2
; 0; and 1
</p>
<p>2
; 1
2
in each unit cell,
</p>
<p>Fig. 2.19 Plane group p2.
(a) Twofold symmetry
motif. (b) Oblique net with
p unit cells outlined. (c)
Extended pattern of plane
</p>
<p>group p2 obtained by a
</p>
<p>combination of (a) with (b)
</p>
<p>2.7 Space Groups 73</p>
<p/>
</div>
<div class="page"><p/>
<p>see Sect. 2.2.2. We must always look for such &ldquo;extra&rdquo; symmetry elements after the point-group motif
</p>
<p>has been operated on by the unit-cell translations. Ultimately, this will be found to be quite straightfor-
</p>
<p>ward. Meanwhile, a simple check consists in ensuring that any point on the diagram can be reached
</p>
<p>from any other point bymeans of a single symmetry operation, including translations as necessary. This
</p>
<p>plane group is given the symbol p2.
</p>
<p>In general, we shall not need to draw several unit cells; one cell will suffice provided that the pattern
</p>
<p>motif is completed around all lattice points intercepted by the given unit cell. Figure 2.20 illustrates
</p>
<p>the standard drawing of p2: the origin is taken on a twofold point, the x axis runs from top to bottom,
</p>
<p>and the y axis runs from left to right. Thus, the origin is considered to be in the top left-hand corner of
</p>
<p>the cell as drawn, but each twofold rotation point could be an equivalent origin; we must remember
</p>
<p>always that the drawing is a representative portion of an infinite array, whether in two or three
</p>
<p>dimensions.
</p>
<p>The asymmetric unit (which may be a chemical species) represented here by O, may be placed
</p>
<p>anywhere in the unit cell, but for convenience, near the origin. It is then repeated by the symmetry p2
</p>
<p>to build up the complete picture, taking care to complete the arrangements around each corner of the
</p>
<p>unit cell. The additional twofold points can then be identified. The reader should now carry out this
</p>
<p>construction.
</p>
<p>The list of fractional coordinates in Fig. 2.20 refers to the unique symmetry-related sites in the unit
</p>
<p>cell. The first row of these sites, related by the space-group symmetry, lists the general equivalent
</p>
<p>positions. In p2 they are given the coordinates x, y, and �x, �y. We could use 1 � x, 1 � y instead of �x, �y,
but it is more usual to list the set of coordinates near one and the same origin.
</p>
<p>Each coordinate line in the space-group description lists, in order from left to right, the number of
</p>
<p>positions in each set, the Wyckoff [2] notation, used for reference purposes, the symmetry at each site
</p>
<p>in the set, and the fractional coordinates of all sites in the set.
</p>
<p>In a conceptual two-dimensional crystal, or projected real atomic arrangement, the asymmetric
</p>
<p>unit may contain either a single atom or a group of atoms. If it consists of part, half, in this plane
</p>
<p>group, of one molecule then the whole molecule, as seen in projection at least, must contain twofold
</p>
<p>rotational symmetry, or a symmetry of which 2 is a subgroup.
</p>
<p>There are four unique twofold rotation points in the unit cell; in theWyckoff notation they are the sets
</p>
<p>(a), (b), (c), and (d), and they constitute the sets of special equivalent positions, point symmetry 2 in this
</p>
<p>plane group. Notice that general positions always have symmetry 1, whereas special positions always
</p>
<p>have a higher crystallographic point-group symmetry. Where the unit cell contains fewer (an integral
</p>
<p>submultiple) of a species than the number of general equivalent positions in its space group, then it may
</p>
<p>be assumed that the species are occupying special equivalent positions and have the symmetry consistent
</p>
<p>with that of the special site. Exceptions to this rule may arise in disordered structures, Sect. 8.9.
</p>
<p>Fig. 2.20 Standard drawing and description of plane group p2. The lines which divide the unit cell into four quadrants
are, as usual, drawn for convenience only
</p>
<p>74 2 Lattices and Space-Group Theory</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_8#Sec44_8">http://dx.doi.org/10.1007/978-1-4614-3954-7_8#Sec44_8</a></div>
</div>
<div class="page"><p/>
<p>Rectangular System
We move next to the rectangular system, which includes point groups m and 2mm, and both p and c
</p>
<p>unit cells. We shall consider first plane groups pm and cm.
</p>
<p>The formation of these plane groups may be considered along the lines already described for p2,
</p>
<p>and we refer immediately to Fig. 2.21a. The origin is chosen on m, but its y coordinate is not defined
</p>
<p>by this symmetry element. In a structure of this symmetry, the origin is specified by fixing arbitrarily
</p>
<p>the y coordinate of one of the atoms in the unit cell. In pm, the general equivalent positions are two in
</p>
<p>number, and there are two sets of special equivalent positions on m lines.
</p>
<p>Plane group cm, Fig. 2.21b, introduces several new features. The coordinate list is headed by the
</p>
<p>expression &eth;0; 0; 1
2
; 1
2
&THORN; &thorn; ; this means that the two translations 0, 0 and 1
</p>
<p>2
; 1
2
are added to all the listed
</p>
<p>coordinates. Hence, the full list of general (equivalent) positions would read
</p>
<p>x; y; �x; y; 1
2
&thorn; x; 1
</p>
<p>2
&thorn; y; 1
</p>
<p>2
� x; 1
</p>
<p>2
&thorn; y
</p>
<p>Fig. 2.21 Plane groups
in the rectangular system.
</p>
<p>(a) pm. (b) cm; glide lines
(g) are indicated by the
</p>
<p>dashed lines
</p>
<p>2.7 Space Groups 75</p>
<p/>
</div>
<div class="page"><p/>
<p>F
ig
.
2
.2
2
</p>
<p>76 2 Lattices and Space-Group Theory</p>
<p/>
</div>
<div class="page"><p/>
<p>F
ig
.
2
.2
2
</p>
<p>(c
o
n
ti
n
u
ed
)
U
n
it
ce
ll
s
o
f
th
e
1
7
p
la
n
e
g
ro
u
p
s.
T
h
e
g
en
er
al
eq
u
iv
al
en
t
p
o
si
ti
o
n
s
ar
e
re
p
re
se
n
te
d
h
er
e
b
y
sc
al
en
e
tr
ia
n
g
le
m
o
ti
fs
in
st
ea
d
o
f
th
e
u
su
al
ci
rc
le
.
It
m
ay
</p>
<p>b
e
</p>
<p>n
o
te
d
th
at
</p>
<p>th
e
co
n
v
en
ti
o
n
al
</p>
<p>u
n
it
-c
el
l
d
ra
w
in
g
sh
o
w
s
th
e
im
</p>
<p>m
ed
ia
te
</p>
<p>en
v
ir
o
n
m
en
t
o
f
ea
ch
</p>
<p>ce
ll
</p>
<p>2.7 Space Groups 77</p>
<p/>
</div>
<div class="page"><p/>
<p>Given x, the distance 1
2
� x, for example, is found by first moving 1
</p>
<p>2
along the a axis from the origin
</p>
<p>and then moving back along the same line by the amount x.
</p>
<p>The centering of the unit cell in conjunction with the m lines introduces the glide-line symmetry
</p>
<p>element, symbol g and graphic symbol - - - . The glide lines interleave the mirror lines, and their action is
</p>
<p>a combination of reflection and translation, the two movements comprising again a single symmetry
</p>
<p>operation. The translational component is one half of the repeat distance in the direction of the glide line.
</p>
<p>Thus, the pair of general positions x, y and 1
2
� x; 1
</p>
<p>2
&thorn; y are related by the g line at x &frac14; 1
</p>
<p>4
, y &frac14; 0.We shall
</p>
<p>encounter glide lines in any centered plane group wherem lines are present, and in certain other groups.
</p>
<p>For example, we may ask if there is any meaning to the symbol pg, a glide-symmetry motif repeated by
</p>
<p>the translations of a p unit-cell? The answer is that pg is a possible plane group; in fact, it is the symmetry
</p>
<p>of the pattern in Fig. 2.18. The differing orientations of the glide lines in Figs. 2.18 and 2.22 (standard) are
</p>
<p>expressed by the full symbols p11g (g &perp; y) and p1g1 (g &perp; x), respectively.
</p>
<p>There is only one set of special positions in cm, in contrast to two sets in pm. This situation arises
</p>
<p>because the centering condition in cm requires that both mirror lines in the unit cell be included in
</p>
<p>one and the same set. If we try to postulate two sets, by analogy with pm, we obtain
</p>
<p>0; y; 1
2
; 1
2
&thorn; y (2.47)
</p>
<p>and
</p>
<p>1
2
; y; 0&eth;or 1&THORN;;1
</p>
<p>2
&thorn; y (2.48)
</p>
<p>However, expressions (2.47) and (2.48) involve only a shift in the origin, and therefore do not
</p>
<p>constitute two different sets of special equivalent positions.
</p>
<p>We could refer to plane group cm by the symbol cg. If we begin with the origin on g and mark in the
</p>
<p>general positions as before, we should find now the glide lines interleaved withm lines. Two patterns that
</p>
<p>differ only in the choice of origin or in the numerical values attached to the coordinates of the equivalent
</p>
<p>positions do not constitute different space groups. The reader can illustrate this statement by drawing cg,
</p>
<p>and by drawing pg also, can show that pm and pg are different. The glide line or, indeed, any translational
</p>
<p>symmetry element is not encountered in point groups; it is a property of infinite patterns.
</p>
<p>The 17 Plane Groups
The 17 plane groups are illustrated in Fig. 2.22. The two diagrams for each plane group show the general
</p>
<p>equivalent positions and the symmetry elements. The asymmetric unit is represented therein by a scalene
</p>
<p>triangle instead of by the usual circle. Space groups that are derived by the repetition of a point-group
</p>
<p>motif by the lattice translations are termed symmorphic space groups, as with p2, pm, and c2mm, but
</p>
<p>otherwise as non-symmorphic space groups, as with pg, p2mg, and p2gg.
</p>
<p>Conditions Governing X-Ray Reflection
Our main reason for studying space-group symmetry is that it provides information about the repeat
</p>
<p>patterns of atoms in crystal structures. X-ray diffraction spectra are characterized in position by the
</p>
<p>indices of the families of planes from which, in the Bragg treatment of diffraction which we
</p>
<p>consider in Sect. 3.3.2, the X-rays are considered to be reflected. The pattern of the indices of the
</p>
<p>reflecting planes reveals information about the space group of the crystal. Where a space group
</p>
<p>contains translational symmetry, certain sets of reflections will be systematically absent from the
</p>
<p>experimental diffraction data record. We meet this situation for the first time in cm, Fig. 2.21b; two-
</p>
<p>dimensional reflections hk (l &frac14; 0) are limited to those for which the sum h + k is an even number.
</p>
<p>78 2 Lattices and Space-Group Theory</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec16_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec16_3</a></div>
</div>
<div class="page"><p/>
<p>Figure 2.23 illustrates a rectangular lattice. Two unit cells are depicted on this lattice, a centered
</p>
<p>cell with vectors A and B, and a primitive cell with vectors a and b. The relationship between them
is summarized by the equations
</p>
<p>A &frac14; a� b
B &frac14; a&thorn; b (2.49)
</p>
<p>We have shown in Sect. 2.5.4 that Miller indices of planes transform in the same way as unit-cell
</p>
<p>vectors, so it follows that
</p>
<p>H &frac14; h� k
K &frac14; h&thorn; k
</p>
<p>(2.50)
</p>
<p>where H and K apply to the unit cellA, B and h and k to the unit cell a, b. Adding equations (2.50), we
obtain
</p>
<p>H &thorn; K &frac14; 2h (2.51)
</p>
<p>which is even for all values of h. Thus, in this centered unit cell, reflections can occur only when the
</p>
<p>sum of the indices, H + K, is an even integer. This topic is discussed more fully in Sect. 3.7ff,
</p>
<p>whereupon the significance of the extreme right-hand column of data in figures such as Figs. 2.21
</p>
<p>and 2.24 will become clear.
</p>
<p>2.7.2 Plane Groups Related to 2mm
</p>
<p>Point group 2mm belongs to the rectangular system and, as a final example in two dimensions, we
</p>
<p>shall study plane group p2gg. It is often helpful to recall the &ldquo;parent&rdquo; point group of any space group:
</p>
<p>we ignore the unit-cell symbol, and replace any translational symmetry elements by the
</p>
<p>Fig. 2.23 Centered rectangular unit cell A, B and primitive unit cell a, b within the same lattice
</p>
<p>2.7 Space Groups 79</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec27_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec27_3</a></div>
</div>
<div class="page"><p/>
<p>corresponding nontranslational symmetry elements. Thus, pg is derived from point groupm, and p2gg
</p>
<p>from 2mm.
</p>
<p>In point group 2mm, we know that the two m lines intersect in the twofold rotation point, and this
</p>
<p>remains true for plane group p2mm. In p2gg, however, we may not assume that the twofold rotation
</p>
<p>point lies at the intersection of the g lines. In our study of point groups, we saw that the symmetry
</p>
<p>elements in a given symbol have a definite relative orientation with respect to the crystallographic
</p>
<p>axes; this is preserved in the corresponding space groups. Thus, we know that the g lines are normal to
</p>
<p>the x and y axes, and we can take an origin, initially, at their intersection, Fig. 2.24a. In Fig. 2.24b, the
</p>
<p>general equivalent positions have been inserted; this diagram reveals the positions of the twofold
</p>
<p>points, inserted now in Fig. 2.24c, together with the additional g lines in the unit cell. The standard
</p>
<p>Fig. 2.24 Formation and
description of plane group
</p>
<p>p2gg
</p>
<p>80 2 Lattices and Space-Group Theory</p>
<p/>
</div>
<div class="page"><p/>
<p>orientation of p2gg places the twofold point at the origin; Fig. 2.24d shows this setting and the
</p>
<p>description of this plane group. We see again that two interacting symmetry elements have a
</p>
<p>combined action which is equivalent to that of a third symmetry element, but their positions must
</p>
<p>be chosen correctly. This question did not arise in point groups because, by definition, all symmetry
</p>
<p>elements pass through a point, the origin. What group would arise if we did place the twofold rotation
</p>
<p>point at the intersection of the glide lines?
</p>
<p>There are two sets of special equivalent positions in p2gg, but the pairs of twofold rotation points
</p>
<p>that constitute each set must be selected correctly. One way of ensuring a proper selection is by
</p>
<p>inserting the coordinate values of the point-group symmetry element constituting a special position
</p>
<p>into the coordinates of the general positions. Thus, by taking x &frac14; y &frac14; 0, for one of the twofold points,
we obtain a set of special positions with coordinates 0,0 and 1
</p>
<p>2
; 1
2
. If we had chosen 0,0 and 0; 1
</p>
<p>2
as a set,
</p>
<p>the resulting pattern would not have conformed to p2gg symmetry, but to pm, as Fig. 2.25 shows.
</p>
<p>Special positions always form a subset of the general positions, under the same space-group symmetry.
</p>
<p>The general equivalent positions give rise to two conditions limiting reflections, because
</p>
<p>the structure is &ldquo;halved&rdquo; with respect to a for the reflections h0, and with respect to b for the
</p>
<p>reflections 0k. The special positions take both of these conditions, and the extra conditions shown,
</p>
<p>because occupancy of the special positions in this plane group gives rise to centered arrangements.
</p>
<p>The entities occupying special positions must, themselves, be consistent with the symmetry of the
</p>
<p>crystal structure.
</p>
<p>After the development of the structure factor in Sects. 3.2.3ff and 3.5.1ff, limiting conditions will
</p>
<p>be derived analytically.
</p>
<p>2.7.3 Three-Dimensional Space Groups
</p>
<p>The principles that have emerged from the discussion on plane groups can be extended to
</p>
<p>three dimensions. Whereas the plane groups are limited to 17 in number, there are 230 space groups.
</p>
<p>We shall limit our discussion to a few space groups mainly in the monoclinic and orthorhombic
</p>
<p>systems. We believe this will prove a satisfactory working procedure because many of the important
</p>
<p>principles will evolve and, from a practical point of view, a large percentage of crystals belong to these
</p>
<p>two systems.
</p>
<p>Monoclinic Space Groups
In the monoclinic system, the lattices are characterized by P and C unit-cell descriptors, and the point
</p>
<p>groups are 2, m, and 2/m. We consider first space groups P2 and C2.
</p>
<p>Fig. 2.25 Occupation of the special positions 0, 0 and 0; 1
2
in p2gg leads to pm (p1m1) symmetry, even though the
</p>
<p>occupying entity has itself symmetry 2
</p>
<p>2.7 Space Groups 81</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec11_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec11_3</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec23_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec23_3</a></div>
</div>
<div class="page"><p/>
<p>Aswith the plane groups, wemay begin with amotif, which has twofold symmetry, but now about a
</p>
<p>line or axis, in three-dimensional space. This motif is arranged in a fixed orientation with respect to the
</p>
<p>points of a monoclinic lattice. Figure 2.26 shows a stereoscopic pair of illustrations for a unit cell of
</p>
<p>space group C2, drawn with respect to the conventional right-handed axes.
</p>
<p>In Fig. 2.27, space groups P2 and C2 are shown in projection. The standard drawing of space-
</p>
<p>group diagrams is on the a, b plane of the unit cell, with +x running from top to bottom, &thorn;y from left
to right, both in the plane of the paper, and +z directed upwards from the paper. The positive or
</p>
<p>negative signs attached to the representative points indicate the z coordinates, that is, in the
</p>
<p>symbolism O+ and O�, the signs stand for z and �z, respectively. The relationship with the preferred
stereogram notation Sect. 1.3, will be evident here.
</p>
<p>In both P2 and C2, the origin is chosen on 2, and is, thus, defined with respect to the x and z axes,
</p>
<p>but not with respect to y; compare pm and cm. How is the origin fixed in Pm? The graphic symbol
</p>
<p>for a diad axis in the plane of the diagram is!; if the axis lies at, say, z &frac14; 1
4
, the symbol ! 1
</p>
<p>4
is used.
</p>
<p>In space group P2, the general and special equivalent positions may be derived quite readily. The
</p>
<p>special sets (b) and (d) should be noted carefully; they are sometimes forgotten by the beginner
</p>
<p>because symmetry elements distant c/2 from those drawn in the a, b plane are not indicated on the
</p>
<p>conventional diagrams. The diad axis at x &frac14; 0, 1
2
, for example, relates x, y, z to a point at �x, y, 1 � z; its
</p>
<p>presence, and that of the diad at x &frac14; z &frac14; 1
2
, may be illustrated by drawing the space group in projection
</p>
<p>on the ac plane of the unit cell. The reader should make this drawing and compare it with Fig. 2.27a.
</p>
<p>It is often useful to consider a structure in projection on to one of the principal planes (100), (010),
</p>
<p>or (001). The symmetry of a projected space group corresponds to that of a plane group, and the
</p>
<p>symmetries of the principal projections are included with the space-group description, Fig. 2.27. The
</p>
<p>full plane-group symbols, given in parentheses, indicate the orientations of all symmetry elements,
</p>
<p>including identity, in the space group, Table 1.5. In C2, certain projections produce more than one
</p>
<p>repeat in some directions; the projected cell dimensions, represented by a0 and b0 are then halved with
respect to their original values. The Miller indices transform with the change of unit cell: thus, for
</p>
<p>example, with b halved, 220 becomes 210, and 210 becomes 410 (which is equivalent to halving the k
</p>
<p>index in each case).
</p>
<p>The projection of C2 on to (100) is shown by Fig. 2.28 in three stages, starting from the y and z
</p>
<p>coordinates of the set of general equivalent positions. The symmetry of the projection is determined
</p>
<p>by the arrangements of points, now in two dimensions, and the relation between them is clearly that of
</p>
<p>Fig. 2.26 Stereoscopic pair of illustrations of the environs of one unit cell of space group C2; the general equivalent
positions are shown. The diagram reveals nine axes of symmetry 2, and six axes of symmetry 21. Can you identify
</p>
<p>their positions?
</p>
<p>82 2 Lattices and Space-Group Theory</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Sec8_1">http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Sec8_1</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Tab5_1">http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Tab5_1</a></div>
</div>
<div class="page"><p/>
<p>Fig. 2.27 Monoclinic space groups in the standard setting. (a) P2. (b) C2
</p>
<p>2.7 Space Groups 83</p>
<p/>
</div>
<div class="page"><p/>
<p>m symmetry. A correct and sufficient projected unit cell is determined by a and b/2. It is important to
remember that, in the plane groups, as in the two-dimensional point groups, all symmetry operations
</p>
<p>take place wholly within the plane of the figure.
</p>
<p>The general equivalent positions in C2 may be obtained by adding the translations 1
2
; 1
2
; 0, namely,
</p>
<p>those associated with a C unit cell (Table 2.2), to the equivalent positions of P2. This operation is
</p>
<p>equivalent to repeating the original twofold motif at the lattice points of the C monoclinic unit cell.
</p>
<p>This simple relationship between P and C cells is indicated by the heading &eth;0; 0; 0; 1
2
; 1
2
; 0&THORN; &thorn; of the
</p>
<p>coordinate list in C2; it may be compared with that for cm, Fig. 2.21b.
</p>
<p>There are four sets of special positions in P2, but only two sets in C2; the reason for this has been
</p>
<p>discussed in relation to plane groups pm and cm, Sect. 2.7.1.
</p>
<p>2.7.4 Screw Axes
</p>
<p>Screw axes are symmetry elements that can relate points in an infinite, three-dimensional, regular
</p>
<p>array; they are not a feature of point groups. A screw-axis operation may be thought of as a
</p>
<p>combination of rotation and translation, although it is a single symmetry operation: an infinitely
</p>
<p>long spiral staircase gives an indication of the nature of the symmetry operation.
</p>
<p>Imagine that the bottom step, Fig. 2.29, is rotated, anticlockwise, looking in a direction down the
</p>
<p>stairs, by 60� about the vertical support, or axis, and then translated upward by one sixth of the repeat
</p>
<p>Fig. 2.28 Projection of C2 on to (100). (a) y, z Positions from C2 (z axis left to right). (b) Two-dimensional symmetry
elements, m lines, added. (c) One unit cell: p1m (p11m), b0 &frac14; b/2, c0 &frac14; c. Plane groups p11m and p1m1 are equivalent
because they correspond only to an interchange of the x and y axes; 1 is the trivial symmetry element
</p>
<p>84 2 Lattices and Space-Group Theory</p>
<p/>
</div>
<div class="page"><p/>
<p>distance between steps in similar orientations; it then takes the place of the second step, which itself
</p>
<p>moves upward in a similar manner. Clearly, if this procedure were repeated six times, the bottom step
</p>
<p>would reach the position and orientation of the sixth step up; we symbolize this screw axis as 61.
</p>
<p>Infinite length is, theoretically, a requirement because as the bottom step is rotated and translated
</p>
<p>upward, so another step, below the figure, comes up into its position in order that indistinguishability
</p>
<p>is maintained. The spiral staircases of the Monument in London and of the Statue of Liberty in
</p>
<p>New York seem to be of infinite length, and might be considered as macroscopic near-examples of
</p>
<p>screw axes. Examine them carefully on your next visit and determine their symmetry nature.
</p>
<p>The centering of the unit cell in C2 introduces screw axes which interleave the diad axes,
</p>
<p>Fig. 2.27. A screw axis may be designated Rp (p &lt; R) and a screw-axis operation consists of an R-fold
rotation coupled with a translation parallel to the screw axis of p/R times the repeat in the direction of
</p>
<p>the axis. Forp &frac14; R, the translation parallel to the screwaxis is unity, and result of the operation corresponds
effectively to simple rotation: P22 � P2. In C2, the screw axis is of the type 21 and has a translational
component of 1
</p>
<p>2
parallel to b. The general equivalent positions x, y, z and 1
</p>
<p>2
� x; 1
</p>
<p>2
&thorn; y; �z are related by
</p>
<p>a 21 axis along &frac12;14; y; 0�.
6 Screw axes are present in the positions shown by their graphic symbol (see
</p>
<p>also Table 2.5).
</p>
<p>Limiting Conditions in C2
Wereferredbriefly to limiting conditions inSect. 2.7.1. The limiting conditions forC2are listed inFig. 2.27.
</p>
<p>Two of them are placed in parentheses; this notation is used to indicate that they are dependent upon amore
</p>
<p>general condition. Thus, since we know that the hkl reflections are limited by the condition h + k &frac14; 2n
(even), because thecell isC-centered, it follows that theh0l reflectionsare limitedbyh &frac14; 2n (0 is effectively
an even number). There are several other nonindependent conditions that could have been listed.
</p>
<p>For example, 0kl: k &frac14; 2n and h00: h &frac14; 2n. However, in the monoclinic system, in addition to the
</p>
<p>Fig. 2.29 Spiral staircase: an illustration of 61 screw-axis symmetry
</p>
<p>6We use this notation to describe lines, in this example, the line parallel to the y axis through x &frac14; 1
4
, z &frac14; 0.
</p>
<p>2.7 Space Groups 85</p>
<p/>
</div>
<div class="page"><p/>
<p>hkl reflections, we are concerned particularly onlywith h0l and 0k0, because the symmetry plane is parallel
</p>
<p>to (010) and the symmetry axis is parallel to [010]. This feature is discussed more fully in Sect. 3.7ff.
</p>
<p>Space Group P21
Space groupsC2 andC21 are equivalent andmaybe comparedwith the pair cm and cg. On the other hand,
</p>
<p>P2 contains no translational symmetry, soP21 is a new space group, Fig. 2.30; it occurs with a frequency
</p>
<p>of6%among recorded structures. Thereare no special positions inP21. Special positions cannot exist ona
</p>
<p>translational symmetry element, since it would mean that the entity placed on such an element consisted
</p>
<p>of a pattern of infinite repeat.
</p>
<p>Table 2.5 Notation for symmetry axes in space groups, and limiting
conditions for screw axes
</p>
<p>Notes: (1) The 31 and 32 axes are referred to the hexagonal setting of the
</p>
<p>trigonal system. (2) Compare the 21, 42, and 63 axes, the 41 and 43 axes, and
</p>
<p>31, 32, 62, 64 axes
</p>
<p>86 2 Lattices and Space-Group Theory</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec27_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec27_3</a></div>
</div>
<div class="page"><p/>
<p>2.7.5 Glide Planes
</p>
<p>Consider again Fig. 2.18, but let each dashed line be now the trace of a glide plane normal to b. Whereas
</p>
<p>in two dimensions, the direction of translation, after the reflection part of the operation, is unequivocal, in
</p>
<p>three dimensions there are several possibilities, although each of them will not necessarily give rise to a
</p>
<p>different space group.
</p>
<p>In the case of the glide plane normal to b, the reflection is across the mirror plane normal to b, and
</p>
<p>the direction of translation could be along a of amount a/2, along c of amount c/2, along a diagonal
</p>
<p>direction n of amount (a + c)/2, or, in certain groups, along a diagonal direction d of amount such as
</p>
<p>(a � c)/4. The d-glide plane is not often encountered in practice, and will not be discussed in detail
here [3].
</p>
<p>The graphic symbols for glide planes carry information about the glide planes. Thus, in Fig. 2.36, an a-
</p>
<p>glide plane at c/4 is shown by the arrow with 1
4
adjacent to it. The symbol n may refer to more than one
</p>
<p>orientation (Table 2.6), but the space-group symbol here, which relates back to the corresponding point-
</p>
<p>group symbol in Table 1.5, provides the necessary information. Thus, if the n-glide plane is normal to a,
</p>
<p>the translation component of the n-glide-symmetry operation must be (b + c)/2. This is why it is so
</p>
<p>important to understand fully the Hermann&ndash;Mauguin point-group notation [4], Table 1.5, because that
</p>
<p>for space groups follows in a parallel manner. The translational components for screw axes and for glide
</p>
<p>planes are always integer fractions of the repeat distances.
</p>
<p>If a space group is formed from the combination of a point group with m planes and a lattice of
</p>
<p>centered unit cells, glide planes are always introduced into the space group. The nature and direction
</p>
<p>of the translations in screw-axis and glide-plane symmetries are implicit in their symbolism (see
</p>
<p>Tables 2.5 and 2.6).
</p>
<p>Space Group P21/c
As an example of a space group with glide planes, we shall study P21/c, a space group encountered
</p>
<p>frequently (36%) in practice. This space group is derived from point group 2/m, and must, therefore,
</p>
<p>be centrosymmetric. However, the center of symmetry does not lie at the intersection of 21 and c.
</p>
<p>Fig. 2.30 General equivalent positions and symmetry elements in space group P21
</p>
<p>2.7 Space Groups 87</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Tab5_1">http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Tab5_1</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Tab5_1">http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Tab5_1</a></div>
</div>
<div class="page"><p/>
<p>It is normally desirable to place the origin on a center of symmetry in centrosymmetric space groups
</p>
<p>and, in this example, we must determine the appropriate positions of the symmetry elements in the
</p>
<p>unit cell. We note here that sometimes an origin will have a point symmetry greater than 1, for
</p>
<p>example, 2/m or mmm, but 1 is a subgroup of such symmetries. We shall approach the solution of
</p>
<p>this problem in two ways, the first of which is similar to our treatment of plane group p2gg.
</p>
<p>Since the screw axis must intersect the glide plane normally, according to the space-group symbol,
</p>
<p>the point of intersection will be taken as an origin and the space group drawn, Fig. 2.31. We see now
</p>
<p>that the centers of symmetry lie at points such as 0; 1
4
; 1
4
. This point may be taken as a new origin, and
</p>
<p>the space group redrawn, Fig. 2.32; the fraction 1
4
placed next to the center of symmetry symbol
</p>
<p>indicates its fractional position above (and below) the ab plane.
</p>
<p>It is desirable, however, to be able to draw the standard space-group illustration at the outset. From
</p>
<p>a choice of origin, and using the full meaning of the space-group symbol, we can obtain the positions
</p>
<p>of the symmetry elements by means of a simple scheme:
</p>
<p>Let the symmetry elements be placed as follows:
�1 at 0, 0, 0 (choice of origin)
</p>
<p>21 along [p, y, r], parallel to the y axis
</p>
<p>c the plane (x, q, z), normal to the y axis
</p>
<p>It is important to note that we have employed only the standard choice of origin and the
</p>
<p>information contained in the space-group symbol. Next, we carry out the symmetry operations as
</p>
<p>shown in the scheme of Fig. 2.33.
</p>
<p>Table 2.6 Notation for symmetry planes in space groups, and limiting conditions for glide planes
</p>
<p>Symbol Graphic symbol
</p>
<p>Glide plane
</p>
<p>orientation
</p>
<p>and translation
</p>
<p>Limiting
</p>
<p>condition
</p>
<p>m &perp; paper &ndash; &ndash;
</p>
<p>jj paper &ndash; &ndash;
</p>
<p>a &perp; paper (h0l) a/2 h0l: h &frac14; 2n
jj paper (hk0) a/2 hk0: h &frac14; 2n
</p>
<p>b &perp; paper (0kl) b/2 0kl: k &frac14; 2n
jj paper (hk0) b/2 hk0: k &frac14; 2n
</p>
<p>c ?paper
?paper
</p>
<p>(
(0kl) c/2 0kl: l &frac14; 2n
(h0l) c/2 h0l: l &frac14; 2n
</p>
<p>n ?paper
?paper
</p>
<p>(
(0kl) (b + c)/2 0kl: k + l &frac14; 2n
(h0l) (c + a)/2 h0l: l + h &frac14; 2n
</p>
<p>jj paper (hk0) (a + b)/2 hk0: h + k &frac14; 2n
</p>
<p>d ?paper
?paper
</p>
<p>(
(0kl) (b � c)/4 0kl: k + l &frac14; 4n
(h0l) (c � a)/4 h0l: l + h &frac14; 4n
</p>
<p>jj paper (hk0) (a � b)/4 hk0: h + k &frac14; 4n
</p>
<p>Notes: (1) The trigonal system is here referred to hexagonal axes. (2) An arrow shows the direction of the
</p>
<p>glide translation. A fraction indicates the z height of the plane. (3) The condition (a + b + c)/4 exists for
</p>
<p>d-glide planes parallel to f1�10g in the tetragonal and cubic systems
</p>
<p>88 2 Lattices and Space-Group Theory</p>
<p/>
</div>
<div class="page"><p/>
<p>Fig. 2.31 Space group P21/c with the origin at an intersection of 21 and c
</p>
<p>Fig. 2.32 Space group P21/c with the origin on �1 (standard setting)
</p>
<p>Fig. 2.33 Operation about a 21 axis along the line [p, y, 0]: The x coordinate of point 2 relative to that of point 1 is
2p � x. A similar construction may be used for the y coordinate in the c-glide operation
</p>
<p>2.7 Space Groups 89</p>
<p/>
</div>
<div class="page"><p/>
<p>The symbol �c is used to indicate that the c-glide translation of 1
2
is subtracted, which is
</p>
<p>crystallographically equivalent to being added.7
</p>
<p>We use the fact that the combined effect of two operations is equivalent to a third operation,
</p>
<p>starting from the original point (1). Symbolically, in operator notation, c 21 &frac14; �1, that is, 21 followed
by c is equivalent to �1. Thus, points (3) and (4) are one and the same, whence, by comparing
</p>
<p>coordinates, p &frac14; 0 and q &frac14; r &frac14; 1
4
. Comparison with Fig. 2.32 shows that these conditions lead to
</p>
<p>the desired positions of the symmetry elements in P21/c.
</p>
<p>The change in the x coordinate in the operation (1) ! (2) is illustrated in Fig. 2.33; the argument
can be applied to any similar situation in other space groups, and we consider one coordinate at a time.
</p>
<p>The completion of the details of this space group forms the basis of a problem at the end of this chapter.
</p>
<p>We shall not discuss centered monoclinic space groups, but they do not present difficulty once the
</p>
<p>primitive space groups have been mastered. Figure 2.34 shows a stereoscopic pair of illustrations of the
</p>
<p>zinc and iodine atoms in the structure of diiodo-(N, N, N0, N0-tetramethylethylenediamine)zinc(II) [5]. It
crystallizes in space groupC2/cwith fourmolecules per unit cell; the zinc atoms lie on twofold axes. The
</p>
<p>reader should make a drawing of C2/c, putting in all the symmetry elements and a set of
</p>
<p>general equivalent positions, for comparison with Fig. 2.34.
</p>
<p>2.7.6 Analysis of the Space-Group Symbol
</p>
<p>In this section we consider the general relationship between space-group symbols and point-group
</p>
<p>symbols.On encountering a space-group symbol, the first problem is to determine the parent point group.
</p>
<p>This process has been discussed, Sect. 2.7.2; here are a few more examples. It is not necessary to have
</p>
<p>explored all space groups in order to carry out this exercise:
</p>
<p>P21=c ! &eth;21=c&THORN; ! &eth;2=c&THORN; ! 2=m
Ibca ! mmm
P41212 ! 422
F43c ! 43m
</p>
<p>7�1 may always be added to a coordinate to give a crystallographically equivalent position.
</p>
<p>Fig. 2.34 Stereoview of the unit cell for the structure of diiodo-(N, N, N0, N0-tetramethylethylenediamine)zinc(II),
showing the zinc and iodine (larger circles) atoms
</p>
<p>90 2 Lattices and Space-Group Theory</p>
<p/>
</div>
<div class="page"><p/>
<p>Next we must identify a crystal system for each point group:
</p>
<p>2=m ! monoclinic
mmm ! orthorhormbic
422 ! tetragonal
43m ! cubic
</p>
<p>Now, from Table 1.5, we can associate certain crystallographic directions with each symmetry
</p>
<p>element in the space group symbol:
</p>
<p>P21/c: Primitive, monoclinic unit cell; c-glide plane &perp; b; 21 axis jj b; centrosymmetric.
Ibca: Body-centered, orthorhombic unit cell; b-glide plane&perp; a; c-glide plane&perp; b; a-glide plane&perp; c;
</p>
<p>centrosymmetric.
</p>
<p>P41212: Primitive, tetragonal unit cell; 41 axis jj c; 21 axes jj a and b; twofold axes at 45� to a and b, in
the ab plane; non-centrosymmetric.
</p>
<p>F�43c: Face-centered, cubic unit cell; �4 axes jj a, b, and c; threefold axes jj h111i; c-glide planes &perp;
h110i; non-centrosymmetric.
It should be noted carefully that the symmetry elements, where there are more than two present, in
</p>
<p>a given space-group symbol may not intersect in the third, equivalent symmetry element, and the
</p>
<p>origin must always be selected with care. Appropriate procedures for the monoclinic and orthorhom-
</p>
<p>bic systems have been discussed; in working with higher symmetry space groups, similar rules can be
</p>
<p>drawn up, as we shall see.
</p>
<p>Because of the similarities between space groups and their parent point groups, a reflection
</p>
<p>symmetry, for example, in a given orientation with respect to the crystallographic axes always
</p>
<p>produces similar changes in the signs of the coordinates. Thus, an m plane perpendicular to z in
</p>
<p>point groupmmm changes x, y, z to x, y, �z. The a-glide plane in Pnma, which is at c/4, changes x, y, z to
1
2
&thorn; x; y; 1
</p>
<p>2
� z; the translational components of 1
</p>
<p>2
are a feature of the space group, but the signs of x, y,
</p>
<p>and z are still +, +, and � after the operation, as with mmm.
</p>
<p>2.7.7 Orthorhombic Space Groups
</p>
<p>We shall consider two orthorhombic space groups,P212121 andPnma. The first is illustrated in Fig. 2.35;
</p>
<p>it should be noted that the three mutually perpendicular 21 axes do not intersect one another in this space
</p>
<p>group. Although P212121, which occurs to the extent of ca. 10%, is a non-centrosymmetric space group,
</p>
<p>the three principal projections are centrosymmetric; each corresponds to the two-dimensional space
</p>
<p>group p2gg.
</p>
<p>Change of Origin
Consider the projection of P212121 on to (001). From the general equivalent positions we obtain the
</p>
<p>two-dimensional set of coordinates:
</p>
<p>x; y; 1
2
� x; �y; 1
</p>
<p>2
&thorn; x; 1
</p>
<p>2
� y; �x; 1
</p>
<p>2
&thorn; y
</p>
<p>It is convenient to change the origin to a twofold rotation point, currently at 1
4
; 0. To carry out this
</p>
<p>transformation, the coordinates of the new origin are subtracted from the original coordinates:
</p>
<p>x� 1
4
; y; 1
</p>
<p>4
� x; �y; 1
</p>
<p>4
&thorn; x; 1
</p>
<p>2
� y; �x� 1
</p>
<p>4
; 1
2
&thorn; y
</p>
<p>2.7 Space Groups 91</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Tab5_1">http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Tab5_1</a></div>
</div>
<div class="page"><p/>
<p>Next, new variables x0 and y0 are chosen such that x0 &frac14; x� 14 and y0 &frac14; y. Then, by substitution,
we obtain:
</p>
<p>x0; y0; �x0; �y0;
1
2
&thorn; x0; 12 � y0; 12 � x0; 12 &thorn; y0
</p>
<p>If the subscript is dropped, these coordinates are exactly those given already for p2gg, Fig. 2.24d,
</p>
<p>which is the plane group of the projection of P212121 on (001), and also on (100) and (010). This type
</p>
<p>of change of origin is useful when studying projections.
</p>
<p>The orthorhombic space group Pnma is shown with the origin on �1 in Fig. 2.36. The symbol tells us
</p>
<p>that the unit cell is primitive, with an n-glide plane normal to the x axis (see Table 2.6), anm plane normal
</p>
<p>to y, and an a-glide plane normal to z. Although this space group is derived from point group mmm, we
</p>
<p>have a problem similar to that discussed with P21/c. The solution of this type of problem depends upon
</p>
<p>the fact that, in the standard orientation,m m m &frac14;1, and is illustrated fully in Problem 2.10 at the end of
this chapter. It may be noted that a double application of Euler&rsquo;s theorem is used here:
</p>
<p>m m &frac14; mm2 and m &eth;mm2&THORN; � 1
The coordinates of the general and the special equivalent positions can be derived easily from the
</p>
<p>diagram. The translational symmetry elements n and a give rise to the limiting conditions shown on
</p>
<p>the diagram.Nonindependent conditions are shown in parentheses; in the orthorhombic system, all of the
</p>
<p>classes of reflection listed should be considered, as will be discussed in Sect. 3.7ff.
</p>
<p>Fig. 2.35 Space group P212121. In space-group diagrams, represents a 21 axis normal to the plane of projection
(Lonsdale K, Henry NFM (1965) International tables for X-ray crystallography, vol I. Kynoch Press. Reproduced by
</p>
<p>courtesy of I. U. Cr.)
</p>
<p>92 2 Lattices and Space-Group Theory</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec27_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec27_3</a></div>
</div>
<div class="page"><p/>
<p>It is useful to remember that among the triclinic, monoclinic, and orthorhombic space groups, at
</p>
<p>least, pairs of coordinates which have one sign change of x, y, or z indicate a symmetry plane normal
</p>
<p>to the axis of the coordinate with the changed sign. If two sign changes exist, a symmetry axis lies
</p>
<p>parallel to the axis of the coordinate that has not changed sign. Three sign changes indicate a center of
</p>
<p>symmetry. In these three systems, where any coordinate, say x, is related by symmetry to another at
</p>
<p>t � x, that symmetry element intersects the x axis at t/2, by virtue of Fig. 2.33 mutatis mutandis.
</p>
<p>2.7.8 Relative Orientations of Symmetry Elements in Space Groups
</p>
<p>Earlier in this chapter, we looked briefly at the problem of choosing the relative positions of the
</p>
<p>symmetry elements in space groups while keeping a particular symmetry element at a given site, such
</p>
<p>as a center of symmetry at the origin in space groups of class 2/m. We now discuss some simple rules
</p>
<p>whereby this task can be accomplished readily, with due regard to the relative orientations of the
</p>
<p>symmetry elements given by the space-group symbol itself, Tables 1.5 and 2.5. We shall consider
</p>
<p>here the symmetry planes and symmetry axes in space groups derived from point groups mmm and
</p>
<p>2/m, although the rules can be applied more widely.
</p>
<p>Fig. 2.36 Space group Pnma; the full space-group symbol is P 21
n
</p>
<p>21
m
</p>
<p>21
a
(Lonsdale K, Henry NFM (1965) International
</p>
<p>tables for X-ray crystallography, vol I. Kynoch Press. Reproduced by courtesy of I. U. Cr.)
</p>
<p>2.7 Space Groups 93</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Tab5_1">http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Tab5_1</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_2#Tab5_2">http://dx.doi.org/10.1007/978-1-4614-3954-7_2#Tab5_2</a></div>
</div>
<div class="page"><p/>
<p>Half-Translation Rule
</p>
<p>Location of Symmetry Planes
</p>
<p>Consider space group Pnna: the translations associated with the three symmetry planes are (b + c)/2,
</p>
<p>(c + a)/2 and a/2, respectively. If they are summed, the result T is (a + b/2 + c). We disregard the
</p>
<p>whole translations a and c because they refer to unit-cell repetitions. Thus, the center of symmetry is
</p>
<p>found displaced by T/2, or b/4, from the point of intersection of the three symmetry planes n, n, and a.
</p>
<p>This means that, with �1 at the origin, we have njj&eth;0; y; z&THORN;; njj&eth;x; 1
4
; z&THORN; and ajj(x, y, 0). As a second
</p>
<p>example, consider Pmma. The only translation is a/2; thus, T &frac14; a/2, and the center of symmetry is
displaced by a/4 from the intersection of m, m and a.
</p>
<p>Space group Immamay be formed from Pmma by introducing the body-centering translation 1
2
; 1
2
; 1
2
,
</p>
<p>Fig. 6.18b. Alternatively, the half-translation rule may be applied to the complete space-group
</p>
<p>symbol. In all, Imma contains the translations (a + b + c)/2 and a/2, so that T &frac14; a + (b + c)/2, or
(b + c)/2; hence, the center of symmetry is displaced by (b + c)/4 from the intersection ofm,m and a.
</p>
<p>This center of symmetry lies in one of the four sets, Wyckoff (a)&ndash;(d), that are introduced by the body-
</p>
<p>centering translation at 1
4
; 1
4
; 1
4
, half the I translation, from a Pmma center of symmetry. This alternative
</p>
<p>setting is given in the International Tables for X-Ray Crystallography [3]; it corresponds to that in
</p>
<p>Fig. 6.18b with the origin shifted to the center of symmetry at 1
4
; 1
4
; 1
4
. Space groups in classmmm based
</p>
<p>on A, B, C, and F unit cells similarly introduce additional sets of centers of symmetry. The reader may
</p>
<p>care to apply these rules to space group Pnma and then check the result with Fig. 2.36. Note that there
</p>
<p>are two sets of special equivalent positions on 1, which is why an origin on either center of symmetry
</p>
<p>can be chosen.
</p>
<p>Type and Location of Symmetry Axes
</p>
<p>The quantity T also shows the types of twofold axes parallel to a, b, and c. Thus, if T contains an a/
</p>
<p>2 component, then if a twofold axis parallel to a exists in the space group, it is a 21 axis. Similarly for
</p>
<p>twofold axes parallel to y and z. Thus, in Pnna, T &frac14; b/2, and so 2x � 2, 2y � 21, and 2z � 2. In Pbca,
T &frac14; (b + c + a)/2; hence, all axes are 21 and the full space-group symbol is P
</p>
<p>21
</p>
<p>b
</p>
<p>21
</p>
<p>c
</p>
<p>21
</p>
<p>a
.
</p>
<p>The location of each twofold axis may be obtained from the orientation of the symmetry plane
</p>
<p>perpendicular to it, being displaced by half the corresponding glide translation, where appropriate.
</p>
<p>Thus, in Pnna, we find 2 along &frac12;x1
4
; 1
4
�, 21 along &frac12;14; y; 14� and 2 along &frac12;14; 0; z�. In Pmma, 21 is along [x, 0,
</p>
<p>0], 2 is along [0, y, 0] and 2 is along &frac12;1
4
; 0; z�. The reader may care to continue the study with space
</p>
<p>group Pnma, and then check the results against Fig. 2.36.
</p>
<p>In the monoclinic space groups of class 2/m, a 21 axis with a translational component of b/2 shifts
</p>
<p>the center of symmetry by b/4 with respect to the point of intersection of 21 with m; carry out Problem
</p>
<p>7.3 and check you result in Tutorial Solution 7.3. In P2/c, the center of symmetry is shifted by c/4 with
</p>
<p>respect to 2/c, and in P21/c the corresponding shift is (b + c)/4, see Fig. 2.32.
</p>
<p>General Equivalent Positions
Once we know the positions of the symmetry elements in a space-group pattern, the coordinates of the
</p>
<p>general equivalent positions in the unit cell follow readily.
</p>
<p>Consider Pmma. Following out the above analysis, we may write the orientation of the symmetry
</p>
<p>elements:
�1 at 0,0,0; choice of origin
</p>
<p>mx, the plane &eth;14; y; z&THORN;
my, the plane (x, 0, z)
</p>
<p>a, the plane (x, y, 0)
</p>
<p>94 2 Lattices and Space-Group Theory</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_17#Fig18_6">http://dx.doi.org/10.1007/978-1-4614-3954-7_17#Fig18_6</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_6#Fig18_6">http://dx.doi.org/10.1007/978-1-4614-3954-7_6#Fig18_6</a></div>
</div>
<div class="page"><p/>
<p>Taking a point x, y, z across each of the three symmetry planes, we have, from Fig. 2.33:
</p>
<p>x; y; z�!mx 1
2
� x; y; z
</p>
<p>�!my x; y; z
�!a 1
</p>
<p>2
&thorn; x; y; z
</p>
<p>If these four points are now operated on by �1 the total of eight equivalent positions for Pmma are
</p>
<p>obtained: � fx; y; z; 1
2
� x; y; z; x; y; z; 1
</p>
<p>2
&thorn; x; y; zg.
</p>
<p>A similar analysis may be carried out for the space groups in the mm2 class, with respect to origins
</p>
<p>on 2 or 21, although we have not discussed these space groups in this book. For example, work
</p>
<p>through the space group Pma�, and check your result with Sect. 3.7.2 and Fig. 3.25, or with the
International Tables for X-ray Crystallography [3].
</p>
<p>2.7.9 Tetragonal and Hexagonal Space Groups
</p>
<p>We shall examine one space group from each of the tetragonal and hexagonal systems because new
</p>
<p>features arise on account of the higher rotational symmetry in these two systems.
</p>
<p>Tetragonal Space Group P4nc
It is evident that this space group is based on the point group 4mm. Reference to Table 1.5 shows that
</p>
<p>the symbol has the following interpretation: a fourfold axis along z; n-glide planes normal to x (and to
</p>
<p>y, because of the fourfold symmetry); c-glide planes normal to [110] and its fourfold symmetry-
</p>
<p>related direction &frac12;1�10�. The orientation of the n glides can be handled in the manner already discussed.
In the case of the c glide, it is straightforward to show, from Problem 2.21, that if the glide plane
</p>
<p>intercepts the x and y axes at the value q, then a point x, y, z is reflected across the glide plane and
</p>
<p>translated to the position q� y; q� x; 1
2
&thorn; z. Thus, as in Sect. 2.7.5, we can set up the interpretation of
</p>
<p>the symbol P4nc, again using Euler&rsquo;s theorem, that the combination of any two operations is
</p>
<p>equivalent to a third operation. Thus, n 4 &frac14; c, but, in contradistinction to the point group 4mm, the
three operators do not all pass through the origin point.
</p>
<p>Let the symmetry elements be placed as follows:
</p>
<p>4 along the z axis, that is, the line [0, 0, z]
</p>
<p>n normal to x, being the plane (x, b, z)
</p>
<p>c normal to &frac12;110�, the plane (q, q, z)
A point x, y, z (1) rotated about the 4-axis becomes �y; x; z (2); this point is taken across the n glide to
</p>
<p>1
2
� y; 2b� x; 1
</p>
<p>2
&thorn; z (3). If we now operate on the original point (1) by the c glide, then x, y, z is
</p>
<p>reflected to q� y; q� x; 1
2
&thorn; z (4). Now, points (3) and (4) are one and the same, so that q &frac14; 1
</p>
<p>2
and
</p>
<p>a &frac14; 1
4
. This setting of the symmetry elements gives rise to the standard diagram for P4nc, shown in
</p>
<p>Fig. 2.37. A similar result may be obtained by an initial clockwise rotation and the equivalent n glide
</p>
<p>parallel to (a, y, z). The positions of the additional symmetry elements, not apparent from the symbol,
</p>
<p>should again be noted. The diagram of the unit cell and its environs is complete, because any point
</p>
<p>shown can be reached from any other point on the diagram by a single symmetry operation, plus unit-
</p>
<p>cell translations as necessary.
</p>
<p>Hexagonal Space Group P63/m
In this space group we encounter sixfold and threefold rotation operations. From Web Appendix
</p>
<p>WA4, we show that a point x, y, z on hexagonal axes rotated anticlockwise about a 63 screw axis along
</p>
<p>2.7 Space Groups 95</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec29_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec29_3</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Fig25_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Fig25_3</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Tab5_1">http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Tab5_1</a></div>
</div>
<div class="page"><p/>
<p>z would be moved to the position x� y; x; 1
2
&thorn; z. The translation of 1
</p>
<p>2
accompanying the z coordinate
</p>
<p>arises from the translation associated with the 63 axis, namely, a translation of 3/6, or
1
2
along z. The
</p>
<p>sequence of points obtained by the successive operations of 63 about [0001] are:
</p>
<p>x; y; z; x� y; x; 1
2
&thorn; z; �y; x� y; z; �x; �y; 1
</p>
<p>2
&thorn; z;
</p>
<p>&eth;1&THORN; &eth;2&THORN; &eth;3&THORN; &eth;4&THORN;
y� x; �x; z; y; y� x; 1
</p>
<p>2
&thorn; z
</p>
<p>&eth;5&THORN; &eth;6&THORN;
</p>
<p>Points (1) and (3) are related by a threefold rotation: note that 3 � 623, that is, two successive
operations of 63, whereas points (1) and (4) are related by 21 symmetry. The space group is completed
</p>
<p>by introducing the m plane at z &frac14; 1
4
: this position ensures that the center of symmetry is at the origin;
</p>
<p>actually the symmetry at the origin is �3: �1 is a subgroup of �3. Other important symmetry elements now
</p>
<p>in evidence include �6, 3, and �1.
</p>
<p>Figure 2.38 illustrates space group P63/m. The 12 general equivalent positions comprise the six
</p>
<p>listed above and another six obtained by their inversion across the center of symmetry at the origin; all
</p>
<p>coordinates change sign. Consider point (2) reflected across the m plane to x� y; x; 1
2
� z. How may
</p>
<p>this point be reached from x, y, z in a single operation? Either a clockwise �3 operation, or an
</p>
<p>Fig. 2.37 Diagrams to show the general equivalent positions and symmetry elements for the tetragonal space group
P4nc (Lonsdale K, Henry NFM (1965) International tables for X-ray crystallography, vol I. Kynoch Press. Reproduced
</p>
<p>by courtesy of I. U. Cr.)
</p>
<p>96 2 Lattices and Space-Group Theory</p>
<p/>
</div>
<div class="page"><p/>
<p>anticlockwise �32 operation, which is equivalent to two successive anticlockwise �3 operations, relates
</p>
<p>these two points; we note in passing that both 3 and 32 are symmetry operations in this group, related
</p>
<p>to the single symmetry element �3.
</p>
<p>A scheme for handling hexagonal space groups, similar to those used for the lower-symmetry
</p>
<p>systems, could be devised, but it will be more straightforward to use matrix operations, as we shall
</p>
<p>now demonstrate.
</p>
<p>Fig. 2.38 Diagrams to show the general equivalent positions and symmetry elements for the hexagonal space group
P63/m (Lonsdale K, Henry NFM (1965) International tables for X-ray crystallography, vol I. Kynoch Press. Reproduced
</p>
<p>by courtesy of I. U. Cr.)
</p>
<p>2.7 Space Groups 97</p>
<p/>
</div>
<div class="page"><p/>
<p>2.8 Matrix Representation of Symmetry Operations
</p>
<p>The representation of symmetry operations by matrices has a certain inherent elegance, and is useful
</p>
<p>for displaying the close relationship between point groups and space groups. In this discussion, we
</p>
<p>shall use the triplet x, y, z to represent a point in three-dimensional space. It could lie on the normal to
</p>
<p>the face of a crystal or be an atom in a crystal structure, and we can indicate it concisely by the vector x.
A symmetry operation may be written as
</p>
<p>R x&thorn; t &frac14; x0 (2.52)
</p>
<p>where x and x0 are column vector triplets before and after the operation, R is a matrix representing the
symmetry operation, and t is a translation vector with components parallel to x, y, and z.
</p>
<p>2.8.1 Matrices in Point-Group Symmetry
</p>
<p>From the definition of point group, Sect. 1.4, it follows that t is identically zero in a point group. All
symmetry elements pass through a single point, the origin: if it were not the case, then parallel
</p>
<p>symmetry axes, for example, could be generated. The consequence of this arrangement for a twofold
</p>
<p>axis is shown in Fig. 2.39.
</p>
<p>Thus, for point groups, (2.52) reduces to
</p>
<p>R x&frac14;x0 (2.53)
</p>
<p>Let R1 represent an m plane perpendicular to the x axis, as in the orthorhombic system, for
example. Then, we have
</p>
<p>1 0 0
</p>
<p>0 1 0
</p>
<p>0 0 1
</p>
<p>2
4
</p>
<p>3
5
</p>
<p>R1
</p>
<p>�
x
</p>
<p>y
</p>
<p>z
</p>
<p>2
4
3
5
</p>
<p>x
</p>
<p>&frac14;
�x
</p>
<p>y
</p>
<p>z
</p>
<p>2
4
3
5
</p>
<p>x0
</p>
<p>(2.54)
</p>
<p>Fig. 2.39 Points 1 and 2, related by the diad (twofold axis) a, when rotated about the axis b produce points 4 and 3. But
3 and 4 are now related by another diad, c. The effect of diad c on points 1 and 2 is to produce points 6 and 5. But these
</p>
<p>points are related to 3 and 4 by diad d and to each other by diad e. Now 3 and 4, for example, can be rotated about e, and
</p>
<p>so on. Clearly, this process would lead to an infinite number of parallel, equidistant diad axes, together with the
</p>
<p>symmetry-related points, a situation that is totally incompatible with a point group
</p>
<p>98 2 Lattices and Space-Group Theory</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Sec9_1">http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Sec9_1</a></div>
</div>
<div class="page"><p/>
<p>The multiplication is carried out, as usual, along the row and down the column, with the result at
</p>
<p>the intersection marked; that is,
</p>
<p>(2.55)
</p>
<p>�x &frac14; �1	 x&thorn; 0	 y&thorn; 0	 z (2.56)
</p>
<p>and similarly for y and z.
</p>
<p>Let the triplet x0 now suffer reflection across a mirror plane normal to y, using matrix R2:
</p>
<p>1 0 0
</p>
<p>0 �1 0
</p>
<p>0 0 1
</p>
<p>2
4
</p>
<p>3
5
</p>
<p>R2
</p>
<p>�
�x
</p>
<p>y
</p>
<p>z
</p>
<p>2
4
3
5
</p>
<p>x
</p>
<p>&frac14;
�x
</p>
<p>�y
</p>
<p>z
</p>
<p>2
4
3
5
</p>
<p>x00
</p>
<p>(2.57)
</p>
<p>It should be clear that the relationship between x and x00 is that of a twofold rotation about the z
axis. Thus, for the two m planes,
</p>
<p>m m&frac14;2 (2.58)
</p>
<p>as we have seen already, Sect. 1.4.2.
</p>
<p>Another way of reaching the same final result is first to combine the two matrices R1 and R2,
</p>
<p>1 0 0
</p>
<p>0 1 0
</p>
<p>0 0 1
</p>
<p>2
64
</p>
<p>3
75
</p>
<p>R2
</p>
<p>�
1 0 0
</p>
<p>0 1 0
</p>
<p>0 0 1
</p>
<p>2
64
</p>
<p>3
75
</p>
<p>R1
</p>
<p>&frac14;
1 0 0
</p>
<p>0 1 0
</p>
<p>0 0 1
</p>
<p>2
64
</p>
<p>3
75
</p>
<p>R3
</p>
<p>(2.59)
</p>
<p>and then to use the right-hand side of (2.59) in (2.53):
</p>
<p>�1 0 0
</p>
<p>0 �1 0
</p>
<p>0 0 1
</p>
<p>2
4
</p>
<p>3
5
</p>
<p>R3
</p>
<p>�
x
</p>
<p>y
</p>
<p>z
</p>
<p>2
4
3
5
</p>
<p>x
</p>
<p>&frac14;
�x
</p>
<p>y
</p>
<p>z
</p>
<p>2
4
3
5
</p>
<p>x00
</p>
<p>(2.60)
</p>
<p>Equation (2.59) corresponds to operation R2 R1 (R1 followed by R2), the order of multiplication
following (2.55). If a rotational symmetry axis forming an operator R is less than or equal to degree
2 or to m, the order of multiplication need not be followed, but it is good practice to multiply the
</p>
<p>matrices in the standard manner; we can highlight this feature by considering point group 4mm.
</p>
<p>The matrices for a fourfold rotation along the z axis and anm plane perpendicular to x are, in order,
</p>
<p>1 0 0
</p>
<p>0 1 0
</p>
<p>0 0 1
</p>
<p>2
4
</p>
<p>3
5
</p>
<p>R2
</p>
<p>m?x
</p>
<p>�
0 1 0
</p>
<p>1 0 0
</p>
<p>0 0 1
</p>
<p>2
4
</p>
<p>3
5
</p>
<p>R3
</p>
<p>4 along z
</p>
<p>&frac14;
0 1 0
</p>
<p>1 0 0
</p>
<p>0 0 1
</p>
<p>2
4
</p>
<p>3
5
</p>
<p>R3
</p>
<p>(2.61)
</p>
<p>2.8 Matrix Representation of Symmetry Operations 99</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Sec11_1">http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Sec11_1</a></div>
</div>
<div class="page"><p/>
<p>Hence, R2 R1 &frac14; R3, and R3 x&eth;x; y; z&THORN; &frac14; x0&eth;y; x; z&THORN;; R3 represents an m plane symmetry operator
normal to &frac12;1�10�. Multiplying in the reverse order, that is,
</p>
<p>&eth;R1 R2&THORN;x &frac14; x00 (2.62)
</p>
<p>gives
</p>
<p>R1 R2 &frac14; R4 (2.63)
</p>
<p>where x00 is now �y, �x, z, andR4 is a matrix operator representing anm plane normal to [110]. Write out
this matrix. The m planes represented by R3 and R4 are equivalent under symmetry R1 or R2, but lead
to physically different sites. Thus, if we are expecting x0 from x and obtain x00 instead, it may be
confusing and, in considering some physical properties, could be significantly different. All other
</p>
<p>point groups may be treated in the standard manner just described.
</p>
<p>2.8.2 Matrices in Space-Group Symmetry
</p>
<p>In space-group symmetry, t in (2.52) is not necessarily equal to zero. Such a situation will exist
whenever the space group under consideration contains translational symmetry. We will consider first
</p>
<p>space group P21/c, Sect. 2.7.5. As before, we set the origin on �1 &eth;R3&THORN;, 21 (R1) along [p, y, r], and c
(R2) the plane (x, q, z). The operation R1 followed by R2, from our previous discussion, may be
formulated as
</p>
<p>1 0 0
</p>
<p>0 �1 0
</p>
<p>0 0 1
</p>
<p>2
4
</p>
<p>3
5
</p>
<p>R2
</p>
<p>&thorn;
0
</p>
<p>2q
1
2
</p>
<p>2
4
</p>
<p>3
5
</p>
<p>t2
</p>
<p>�
�1 0 0
</p>
<p>0 1 0
</p>
<p>0 0 �1
</p>
<p>2
4
</p>
<p>3
5
</p>
<p>R1
</p>
<p>&thorn;
2p
1
2
</p>
<p>2r
</p>
<p>2
4
</p>
<p>3
5
</p>
<p>t1
</p>
<p>&frac14;
�1 0 0
</p>
<p>0 �1 0
</p>
<p>0 0 �1
</p>
<p>2
4
</p>
<p>3
5
</p>
<p>R3
</p>
<p>&thorn;
0
</p>
<p>0
</p>
<p>0
</p>
<p>2
4
3
5
</p>
<p>t3
</p>
<p>(2.64)
</p>
<p>Matrix R1 is just that for twofold rotation about a line parallel to the y axis, as represented above,
and R2 is the matrix for an m plane normal to y, as given above. The translation vectors t1 and t2 may
be obtained from the setting, following the argument relating to Fig. 2.33. Matrix R3 is the multipli-
cation R2 R1 and, clearly, is equivalent to a center of symmetry (�1) at the origin. Since, by definition
of the standard origin, t3 must be zero, we have the translation vectors
</p>
<p>t2 &thorn; t1 &frac14; t3 &frac14; 0 (2.65)
</p>
<p>It follows that p &frac14; 0, q &frac14; 1
4
and r &frac14; 1
</p>
<p>4
, as before. These results may be regarded as a matrix
</p>
<p>justification of the scheme used in Sect. 2.7.5, and expressed in the half-translation rule, Sect. 2.7.8.
</p>
<p>As a final example, we shall consider space group Pnma, see Sect. 2.7.7. From the symbol, we can
</p>
<p>write
</p>
<p>R1: n is the plane (p, y, z) with n-translation 0,
1
2
; 1
2
</p>
<p>R2: m is the plane (x, q, z) with no translation
R3: a is the plane (x, y, r) with a-translation
</p>
<p>1
2
; 0; 0
</p>
<p>R4: �1 is the center of symmetry at 0,0,0 (no translation)
We know that, for space groups in the mmm class, we have
</p>
<p>R3 R2 R1 &frac14; R4 (2.66)
</p>
<p>100 2 Lattices and Space-Group Theory</p>
<p/>
</div>
<div class="page"><p/>
<p>Hence,
</p>
<p>1 0 0
</p>
<p>0 1 0
</p>
<p>0 0 �1
</p>
<p>2
64
</p>
<p>3
75
</p>
<p>R3
</p>
<p>&thorn;
1
2
</p>
<p>0
</p>
<p>2r
</p>
<p>2
64
</p>
<p>3
75
</p>
<p>t3
</p>
<p>8
&gt;&gt;&lt;
</p>
<p>&gt;&gt;:
</p>
<p>1 0 0
</p>
<p>0 �1 0
</p>
<p>0 0 1
</p>
<p>2
64
</p>
<p>3
75
</p>
<p>R2
</p>
<p>&thorn;
0
</p>
<p>2q
</p>
<p>0
</p>
<p>2
64
</p>
<p>3
75
</p>
<p>t2
</p>
<p>�1 0 0
</p>
<p>0 1 0
</p>
<p>0 0 1
</p>
<p>2
64
</p>
<p>3
75
</p>
<p>R1
</p>
<p>&thorn;
2p
1
2
1
2
</p>
<p>2
64
</p>
<p>3
75
</p>
<p>t1
</p>
<p>9
&gt;&gt;=
</p>
<p>&gt;&gt;;
</p>
<p>&frac14;
�1 0 0
</p>
<p>0 �1 0
</p>
<p>0 0 �1
</p>
<p>2
64
</p>
<p>3
75
</p>
<p>R4
</p>
<p>&thorn;
0
</p>
<p>0
</p>
<p>0
</p>
<p>2
64
</p>
<p>3
75
</p>
<p>t4
</p>
<p>(2.67)
</p>
<p>And we have
</p>
<p>t3 &thorn; t2 &thorn; t1 &frac14; t4 &frac14; 0 (2.68)
</p>
<p>Multiplying the matrices and adding the translation vectors we obtain p &frac14; 1
4
; q &frac14; 1
</p>
<p>4
, and r &frac14; 1
</p>
<p>4
as
</p>
<p>given in Fig. 2.36. The full symbol of point group mmm is
2
</p>
<p>m
</p>
<p>2
</p>
<p>m
</p>
<p>2
</p>
<p>m
, so that in Pnma there are 2 or 21
</p>
<p>axes normal to the symmetry planes. We can obtain the results readily from (2.67), inserting the
</p>
<p>values of p, q, and r into the translation vectors; if the fraction 1
2
appears in line with the x coordinate in
</p>
<p>a plane normal to x, then the axis is 21, and similarly for the y and z positions. Hence, the full symbol
</p>
<p>for this space group is P
21
</p>
<p>n
</p>
<p>21
</p>
<p>m
</p>
<p>21
</p>
<p>a
. The same result could be achieved with the scheme used for
</p>
<p>solving Problem 2.10, perhaps with less elegance.
</p>
<p>The essential difference between point groups and space groups rests in the translation vectors, and
</p>
<p>the infinite space to which the space groups refer. Symmorphic space groups such as Pm, C2/m, and
</p>
<p>Imm2, some of which contain translational symmetry elements, do not need any special treatment to
</p>
<p>determine the orientation of the symmetry elements with respect to the origin, since the symmorphic
</p>
<p>space groups contain the point-group symbol, the origin is given immediately, for example, on m in
</p>
<p>Pm, at 2/m (�1) in C2/m, and along mm2 in Imm2; all translation vectors in equations such as (2.64) are
</p>
<p>zero in these space groups. The half-translation rule, once understood, is the simplest method of
</p>
<p>locating the origin, certainly for the non-symmorphic space groups in the monoclinic and orthorhom-
</p>
<p>bic systems, which represent the majority of known crystals.
</p>
<p>2.9 Diffraction Symbols
</p>
<p>We look ahead briefly to some results in later chapters, and note that after a crystal has been examined to
</p>
<p>the extent that indices can be assigned to the X-ray diffraction spectra, the totality of the diffraction
</p>
<p>information can be assembled into a diffraction symbol. This parameter includes the Laue group and the
</p>
<p>symmetry determined through the systematic absences.
</p>
<p>2.9 Diffraction Symbols 101</p>
<p/>
</div>
<div class="page"><p/>
<p>In Table 2.7, we list the diffraction symbols for the orthorhombic space groups. A full discussion
</p>
<p>of diffraction symbols for the 230 space groups may be found in the International Tables for X-Ray
</p>
<p>Crystallography (2002, Volume A) or (1965, Volume I).
</p>
<p>Table 2.7 Orthorhombic space group diffraction symbols
</p>
<p>Point group
</p>
<p>Diffraction symbol 222 mm2 mmm
</p>
<p>mmmP . . . P222 Pmm2 Pmmm
</p>
<p>mmmP . . 21 P2221
mmmP 21 21 . P21212
</p>
<p>mmmP 21 21 21 P212121
mmmP c . . Pc2m &frac14;
</p>
<p>Pcm21 &frac14;
Pma2
</p>
<p>Pmc21
</p>
<p> !
Pcmm &frac14; Pmma
</p>
<p>mmmP n . . Pnm21 &frac14; Pmn21 Pnmm &frac14; Pmmn
mmmP c c . Pcc2 Pccm
</p>
<p>mmmP c a . Pca21 Pcam &frac14; Pbcm
mmmP b a . Pba2 Pbam
</p>
<p>mmmP n c . Pnc2 Pncm &frac14; Pmna
mmmP n a . Pna21 Pnam &frac14; Pnma
mmmP n n . Pnn2 Pnnm
</p>
<p>mmmP c c a Pcca
</p>
<p>mmmP b c a Pbca
</p>
<p>mmmP c c n Pccn
</p>
<p>mmmP b a n Pban
</p>
<p>mmmP b c n Pbcn
</p>
<p>mmmP n n a Pnna
</p>
<p>mmmP n n n Pnnn
</p>
<p>mmmC . . . C222 Cmm2 &frac14;
Cm2m &frac14;
</p>
<p>Cmm2
</p>
<p>Amm2
</p>
<p> !
Cmmm
</p>
<p>mmmC . . 21 C2221
mmmC . c . Cmc21 &frac14;
</p>
<p>C2cm &frac14;
Cmc21
</p>
<p>Ama2
</p>
<p> !
Cmcm
</p>
<p>mmmC . . a C2ma &frac14; Abm2 Cmma
mmmC . c a C2ca &frac14; Aba2 Cmca
mmmC c c . Ccc2 Cccm
</p>
<p>mmmC c c a Ccca
</p>
<p>mmmI . . . I222
</p>
<p>I212121
</p>
<p>" #
Imm2 Immm
</p>
<p>mmmI . a . Ima2 Imam &frac14; Imma
mmmI b a . Iba2 Ibam
</p>
<p>mmmI b c a Ibca
</p>
<p>mmmF . . . F222 Fmm2 Fmmm
</p>
<p>mmmF d d . Fdd2
</p>
<p>mmmF d d d Fddd
</p>
<p>Notes: (1) Space groups shown in bold type, e.g. P212121, are uniquely determinable when the Laue group is known. (2)
Space groups shown in italic type, e.g. Pccm, are not uniquely determinable even when the Laue group is known. (3)
</p>
<p>Special pairs of space groups are enclosed in brackets, e.g. [I222, I212121]. (4) Space groups enclosed in parentheses,
</p>
<p>e.g. Pma2, Pc2m, are determinable if the point group and its orientation are known. (5) In rows containing two symbols,
</p>
<p>e.g. Pc2m and Pma2, the symbol on the right is the standard setting, whether or not it is in parentheses
</p>
<p>102 2 Lattices and Space-Group Theory</p>
<p/>
</div>
<div class="page"><p/>
<p>2.10 Some Other Types of Symmetry
</p>
<p>The symmetry concepts dealtwith so far have referred to the classical &ldquo;non-color&rdquo; groups.Considerationof
</p>
<p>other patterns, such as those of wallpapers, tiled walls and floors of the Alhambra, reveal the existence of
</p>
<p>color symmetry, the simplest example of which is black-white symmetry.
</p>
<p>An example of the classical symmetry that we have been studying is shown in Fig. 2.40. At the
</p>
<p>bottom of the illustration there are three fourfold rotation points, assuming a two-dimensional pattern.
</p>
<p>If we choose the center point as an origin, then another three points in identical orientation form the
</p>
<p>corners of a plane unit cell, set at 45� to the borders of the figure. It may be found convenient to make
a copy of the figure for this study. Not surprisingly, twofold rotation points exist at the mid-points of
</p>
<p>the unit-cell edges, but the fourfold point at the center of the unit cell is in a different orientation from
</p>
<p>those at the corners. There are also m lines and g lines in the pattern: the plane group is p4mg: see also
</p>
<p>Fig. 2.22; p4mg � p4gm by interchange of axes.
</p>
<p>2.10.1 Black-White Symmetry
</p>
<p>The simplest nonclassical symmetry is black-white symmetry, of which Fig. 2.41 is an example. The
</p>
<p>elements of this pattern are black beetles and white beetles, and the same symmetry elements as in
</p>
<p>Fig. 2.40 are present in this illustration. The m lines in the figure are classical, but the g lines involve a
</p>
<p>color change fromwhite to black and vice versa as do the fourfold rotation points. The plane groupmay
</p>
<p>be designated p40gm.
</p>
<p>Fig. 2.40 Classical plane
group of symmetry p4mg
</p>
<p>(see also Fig. 2.22) (Mac-
</p>
<p>gillavry CH (1965) Sym-
</p>
<p>metry aspects of
</p>
<p>M. C. Escher&rsquo;s periodic
</p>
<p>drawings. Reproduced
</p>
<p>by courtesy of I. U. Cr.).
</p>
<p>Scheltema and Holkema,
</p>
<p>Bohn (for I. U. Cr., 1976)
</p>
<p>2.10 Some Other Types of Symmetry 103</p>
<p/>
</div>
<div class="page"><p/>
<p>Potassium Chloride
Apractical example of very closely black-white symmetry is found in the structure of potassiumchloride,
</p>
<p>which consists of the isoelectronic K+ and Cl� ions, Fig. 2.42. Because X-rays are scattered by electrons
</p>
<p>in a crystal structure, each of these species appears identical in an X-ray beam. Thus, the structure
</p>
<p>appeared on first examination to be based on a cubic P unit cell,8 since the resolution of the X-ray pattern
</p>
<p>at that time was not high.
</p>
<p>After other alkali halides, notably sodium chloride, had been examined and their structures found
</p>
<p>to be cubic F, a more detailed examination showed that potassium chloride, too, was cubic F, and the
</p>
<p>true repeat distance was revealed. The X-ray reflections that would have been indicated the F cubic
</p>
<p>structure of potassium chloride were too weak to be revealed by the first experiments with the X-ray
</p>
<p>ionization spectrometer. The correct repeat period is found also by neutron scattering, since the
</p>
<p>scattering powers of the K+ and Cl� species differ significantly for neutron radiation, Sect. 12.5.
</p>
<p>2.10.2 Color Symmetry
</p>
<p>As an example of color symmetry, we examine Fig. 2.43. It comprises fish in four different colors and
</p>
<p>orientations, but all fish of any given color have identical orientations. The 90� difference in
orientation between the pairs white-green, green-red, red-blue, and blue-white fish indicate the
</p>
<p>presence of fourfold color-rotation points. The almost square elements of fins, of sequence white,
</p>
<p>Fig. 2.41 Black/white
plane group of symmetry
</p>
<p>p40gm (Macgillavry CH
(1965) Symmetry aspects
</p>
<p>of M. C. Escher&rsquo;s periodic
</p>
<p>drawings. Reproduced by
</p>
<p>courtesy of I. U. Cr.)
</p>
<p>8 See Bibliography (Bragg 1949).
</p>
<p>104 2 Lattices and Space-Group Theory</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_12#Sec1_12">http://dx.doi.org/10.1007/978-1-4614-3954-7_12#Sec1_12</a></div>
</div>
<div class="page"><p/>
<p>Fig. 2.43 An example of a color symmetry plane group (Macgillavry CH (1965) Symmetry aspects of M. C. Escher&rsquo;s
periodic drawings. Reproduced by courtesy of I. U. Cr.)
</p>
<p>Fig. 2.42 The structure of potassium chloride, KCl, as seen in projection on to a cube face. Since K+ and Cl� are
isoelectronic (18 electrons each), their scattering of X-rays (q.v.) is closely similar
</p>
<p>2.10 Some Other Types of Symmetry 105</p>
<p/>
</div>
<div class="page"><p/>
<p>green, red, blue, at the bottom center of the figure and three others in similar orientation form the
</p>
<p>corners of a square unit cell.
</p>
<p>The fourfold color-rotation point at the center of the unit cell, consisting of areas of fish tails,
</p>
<p>shows the same color sequence but in a different orientation. The twofold rotation points are again
</p>
<p>evident at the mid-points of the cell edges. In this pattern, however, the twofold rotations involve a
</p>
<p>change of color, as indicated by the motifs at the fourfold rotation points: they are twofold color-
</p>
<p>rotation points.
</p>
<p>For further discussions on black-white and color symmetry, the reader is referred to the works of
</p>
<p>Macgillavry and Shubnikov listed in the Bibliography at the end of the chapter.
</p>
<p>2.11 Problems
</p>
<p>2.1. Figure P2.1a shows the molecule of cyclosporin H repeated by translations in two dimensions.
</p>
<p>In Fig. P2.1b, the molecules are related also by twofold rotation operations, while still subjected
</p>
<p>to the same translations as in Fig. P2.1a. Four parallelogram-shaped, adjacent repeat units of
</p>
<p>pattern from an ideally infinite array are shown in each diagram. Convince yourself that
</p>
<p>Fig. P2.1 (a) The mole-
cule of cyclosporin H
</p>
<p>repeated by translations in
</p>
<p>two dimensions. (b) The
molecules are related also
</p>
<p>by twofold rotation opera-
</p>
<p>tions
</p>
<p>106 2 Lattices and Space-Group Theory</p>
<p/>
</div>
<div class="page"><p/>
<p>Fig. P2.1a is formed by repeating a single molecule by the unit-cell translations shown, and that
</p>
<p>Fig. P2.1b follows from it by the addition of a single twofold operation acting at any parallelo-
</p>
<p>gram corner. Furthermore, for Fig. P2.1b state in words:
</p>
<p>(a) The locations of all twofold symmetry operators belonging to a single parallelogram unit.
</p>
<p>(b) How many of these twofold operators are unique to a single parallelogram unit?
</p>
<p>2.2. Two nets are described by the unit cells (1) a &frac14; b, g &frac14; 90� and (2) a &frac14; b, g &frac14; 120�. In each
case: (a) What is the symmetry at each net point? (b) To which two-dimensional system does
</p>
<p>the net belong? (c) What are the results of centering the unit cell?
</p>
<p>2.3. A monoclinic F unit cell has the dimensions a &frac14; 6.000 Å, b &frac14; 7.000 Å, c &frac14; 8.000 Å, and
b &frac14; 110.0�. Show that an equivalent monoclinic C unit cell, with an obtuse b angle, can
represent the same lattice, and calculate its dimensions. What is the ratio of the volume of the C
</p>
<p>cell to that of the F cell?
</p>
<p>2.4. Carry out the following exercises with drawings of a tetragonal P unit cell:
</p>
<p>(a) Center the B faces. Comment on the result.
</p>
<p>(b) Center the A and B faces. Comment on the result.
</p>
<p>(c) Center all faces. What conclusions can you draw now?
</p>
<p>2.5. Calculate the length of &frac12;31�2� for both unit cells in Problem 2.3.
2.6. The relationships a 6C b 6C c, a 6C b 6C 90 or 120�, and g &frac14; 90� may be said to define a diclinic
</p>
<p>system. Is this an eighth system? Give reasons for your answer.
</p>
<p>2.7. (a) Draw a diagram to show the symmetry elements and general equivalent positions in c2mm,
</p>
<p>origin on 2mm. Write the coordinates and point symmetry of the general and special positions, in
</p>
<p>their correct sets, and give the conditions limiting X-ray reflections in this plane group. (b) Draw a
</p>
<p>diagram of the symmetry elements in plane group p2mg, origin on 2; take care not to put the
</p>
<p>twofold point at the intersection ofm and g. Why? On the diagram, insert each of the motifs P, V,
</p>
<p>and Z in turn, each letter drawn in its most symmetrical manner, using the minimum number of
</p>
<p>motifs consistent with the space-group symmetry.
</p>
<p>2.8. (a) Continue the study of space group P21/c, Sect. 2.7.5. Write the coordinates of the general and
</p>
<p>special positions, in their correct sets. Give the limiting conditions for all sets of positions, andwrite
</p>
<p>the plane-group symbols for the three principal projections. Draw a diagram of the space group as
</p>
<p>seen along the b axis. (b) Biphenyl, , crystallizes in space group P21/c with two molecules
</p>
<p>per unit cell. What can be deduced about both the positions of the molecules in the unit cell and the
</p>
<p>molecular conformation? The benzene rings in the molecule may be assumed to be planar.
</p>
<p>2.9. Write the coordinates of the vectors between all pairs of general equivalent positions in P21/c
</p>
<p>with respect to the origin, and note that they are of two types. What is the &ldquo;weight,&rdquo; or
</p>
<p>multiplicity, of each vector set? Remember that � 1
2
and &thorn; 1
</p>
<p>2
in a coordinate are crystallo-
</p>
<p>graphically equivalent, because we can always add or subtract 1 from a fractional coordinate
</p>
<p>without altering its crystallographic implication.
</p>
<p>2.10. The orientation of the symmetry elements in the orthorhombic space group Pban may be
</p>
<p>written as follows9:
</p>
<p>�1 at 0; 0; 0 (choice of origin)
</p>
<p>b - glide k &eth;p; y; z&THORN;
a - glide k &eth;x; q; z&THORN;
n - glide k &eth;x; y; r&THORN;
</p>
<p>9
&gt;&gt;=
</p>
<p>&gt;&gt;;
&eth;from the space - group symbol&THORN;
</p>
<p>9 In general, the symbol ǁ in this context indicates the plane (or line) specified; for example, the b-glide plane will be the
</p>
<p>plane (p, y, z).
</p>
<p>2.11 Problems 107</p>
<p/>
</div>
<div class="page"><p/>
<p>Determine p, q, and r from the following scheme, using the fact that n a b � 1:
</p>
<p>2.11. Construct a space-group diagram for Pbam, with the origin at the intersection of the three
</p>
<p>symmetry planes. List the coordinates of both the general equivalent positions and the centers
</p>
<p>of symmetry. Derive the standard coordinates for the general positions by transforming the
</p>
<p>origin to a center of symmetry.
</p>
<p>2.12. Show that space groups Pa, Pc, and Pn represent the same pattern, but that Ca is different from
</p>
<p>Cc. What is the more usual symbol for space group Ca? What would be the space group for Cc
</p>
<p>after an interchange of the x and z axes? Is Cn another monoclinic space group?
</p>
<p>2.13. For each of the space groups P2/c, Pca21, Cmcm, P�421c, P6122, and Pa3:
</p>
<p>(a) Write down the parent point group and crystal system.
</p>
<p>(b) List the full meaning conveyed by the symbol.
</p>
<p>(c) State the independent conditions limiting X-ray reflections.
</p>
<p>(d) List the Buerger diffraction symbols for these space groups.
</p>
<p>2.14. Consider Fig. 2.25. What would be the result of constructing this diagram with Z alone, and not
</p>
<p>using its mirror image?
</p>
<p>2.15. (a) Draw a P unit cell of a cubic lattice in the standard orientation.
</p>
<p>(b) Center the A faces. What system and standard unit-cell type now exist?
</p>
<p>(c) From the position at the end of (b), let c and all other lines parallel to it be angled backward
</p>
<p>a few degrees in the ac plane. What system and standard unit-cell type now exist?
</p>
<p>From the position at the end of (c), let c and all other lines parallel to it be angled sideways a
</p>
<p>few degrees in the bc plane. What system and standard unit-cell type now exist? For (b) to
</p>
<p>(d), write the transformation equations that take the unit cell as drawn into its standard
</p>
<p>orientation.
</p>
<p>2.16. Set up matrices for the following symmetry operations: �4 along the z axis, m normal to the y
</p>
<p>axis. Hence, determine the Miller indices of a plane obtained by operating on (hkl) by �4, and on
</p>
<p>the resulting plane by the operation m. What are the nature and orientation of the symmetry
</p>
<p>element represented by the given combination of �4 followed by m?
</p>
<p>2.17. The matrices for an n-glide plane normal to a and an a-glide plane normal to b in an orthorhom-
</p>
<p>bic space group are as follows:
</p>
<p>1 0 0
</p>
<p>0 �1 0
</p>
<p>0 0 1
</p>
<p>2
64
</p>
<p>3
75&thorn;
</p>
<p>1
2
</p>
<p>0
</p>
<p>0
</p>
<p>2
64
</p>
<p>3
75
</p>
<p>�1 0 0
</p>
<p>0 1 0
</p>
<p>0 0 1
</p>
<p>2
64
</p>
<p>3
75&thorn;
</p>
<p>0
1
2
1
2
</p>
<p>2
64
</p>
<p>3
75
</p>
<p>a t n t
</p>
<p>What are the nature and orientation of the symmetry element arising from the combination of n
</p>
<p>followed by a? What is the space-group symbol and its class?
</p>
<p>2.18. (a) Determine thematrices for both a 63 rotation about [0, 0, z] and a c-glide plane normal to the y
</p>
<p>axis and passing through the origin in space group P63c�. Use the fact that a threefold right-
handed rotation converts the point x, y, z to �y, x � y, z, and that 2 32 &frac14; 6. (b) What is symmetry
represented by the symbol � in the space group symbol and what are the point-group and
</p>
<p>108 2 Lattices and Space-Group Theory</p>
<p/>
</div>
<div class="page"><p/>
<p>space-group symbols? (c)What is thematrix for the symmetry operation found in (b)? (d) Draw a
</p>
<p>diagram for the space group. List the number of general equivalent positions, their Wyckoff
</p>
<p>notation, point symmetry, coordinates, and conditions limiting reflections for the space group. (d)
</p>
<p>Are there any special equivalent positions? If so, list them as under (c).
</p>
<p>2.19. A unit cell is determined as a &frac14; b &frac14; 3 Å, c &frac14; 9 Å, a &frac14; b &frac14; 90�, g &frac14; 120�. Later, it proves to
be a triply primitive hexagonal unit cell. With reference to Fig. 2.11, determine the equations
</p>
<p>for the unit-cell transformation Rhex ! Robv, and calculate the parameters of the rhombohedral
unit cell.
</p>
<p>2.20. In relation to Problem 2.19, given the plane (13*4) and zone symbol &frac12;1�2 � 3� in the hexagonal
unit cell, determine these parameters in the obverse rhombohedral unit cell. The symbol&lowast; here
</p>
<p>indicates that the three integers given relate to the x, y, and z axes, respectively.
</p>
<p>2.21. By means of a diagram, or otherwise, show that a site x, y, z reflected across the plane (qqz) in
</p>
<p>the tetragonal system has the coordinates q � y, q � x, z after reflection.
2.22. Deduce a diffraction symbol table for the monoclinic space groups.
</p>
<p>2.23. Draw the projection of an orthorhombic unit cell on (001), and insert the trace of the (210) plane
</p>
<p>and the parallel plane through the origin.
</p>
<p>(a) Consider the transformation a0 &frac14; a/2, b0 &frac14; b, c0 &frac14; c. Using the appropriate transformation
matrix, write the indices of the (210) plane with respect to the new unit cell. Draw the new
</p>
<p>unit cell and insert the planes at the same perpendicular spacing, starting with the plane
</p>
<p>through the origin. Does the geometry of the diagram confirm the indices obtained from the
</p>
<p>matrix?
</p>
<p>(b) Make a new drawing, like the first, but now consider the transformation a0 &frac14; a, b0 &frac14; b/2,
c0 &frac14; c. What does (210) become under this transformation? Draw the new unit cell and
insert the planes as before. Does the geometry confirm the result from the matrix?
</p>
<p>2.24. Why are space groups Cmm2 and Amm2 distinct, yet Cmmm and Ammm are equivalent?
</p>
<p>2.25. An orthorhombic P unit cell has the dimensions a &frac14; 5.50 Å, b &frac14; 6.75 Å, c &frac14; 12.20 Å, and
their reciprocals (k &frac14; 1) are a* &frac14; 0.1818 Å�1, b* &frac14; 0.1481 Å�1, c* &frac14; 0.08197 Å�1. Use the
matrix M to transform: (a) The unit cell. (b) The Miller indices (312). (c) The zone symbol
[102]. (d) The reciprocal unit cell dimensions. (e) The point x &frac14; 0.3142, y &frac14; 0.4703,
z &frac14; �0.5174.
</p>
<p>M &frac14;
1 �1 1
</p>
<p>2
</p>
<p>1 1 �1
2
</p>
<p>1
4
</p>
<p>3
4
</p>
<p>1
8
</p>
<p>2
4
</p>
<p>3
5
</p>
<p>References
</p>
<p>1. Ladd MFC (1997) J Chem Educ 74:461
</p>
<p>2. Wyckoff RWG (1963&ndash;1971) Crystal structures, vols 1&ndash;6. Wiley, New York
</p>
<p>3. See Bibliography 1: Henry et al. (1965); Ladd (1989)
</p>
<p>4. Hermann C, Mauguin Ch (1931) Z Kristallogr 76:559
</p>
<p>5. Htoon S, Ladd MFC (1974) J Cryst Mol Struct 4:357
</p>
<p>Bibliography: Lattices and Space Groups
</p>
<p>Bragg WL (1949) The Crystalline State, Vol. 1, G. Bell and Sons
</p>
<p>Burns G, Glazer AM (2013) Space groups for solid state scientists. 3rd (ed.), Elsevier
</p>
<p>References 109</p>
<p/>
</div>
<div class="page"><p/>
<p>Hahn T (ed) (2002) International tables for crystallography, vol A, 5th edn. Kluwer Academic, Dordrecht
</p>
<p>Henry NFM, Lonsdale K (eds) (1965) International tables for X-ray crystallography, vol I. Kynoch Press, Birmingham
</p>
<p>Ladd MFC (1989) Symmetry in molecules and crystals. Ellis Horwood, Chichester
</p>
<p>Ladd MFC (1998) Symmetry and group theory in chemistry. Horwood Publishing, Chichester
</p>
<p>Ladd M, Structures C (1999) Lattices and solids in stereoview. Horwood Publishing, Chichester
</p>
<p>Shmueli U (ed) (2001) International tables for X-ray crystallography. Kluwer Academic, Dordrecht
</p>
<p>Black-White and Color Symmetry
</p>
<p>Macgillavry CH (1976) Symmetry aspects of M C Escher&rsquo;s periodic drawings. Scheltema &amp; Holkema, Bohn (for I. U.
</p>
<p>Cr., 1976)
</p>
<p>Shubnikov AV, Belov NV et al (1964) Coloured symmetry. Pergamon Press, Oxford
</p>
<p>110 2 Lattices and Space-Group Theory</p>
<p/>
</div>
<div class="page"><p/>
<p>X-Rays and X-Ray Diffraction 3
</p>
<p>3.1 Generation and Properties of X-Rays
</p>
<p>X-rays are an electromagnetic radiation of short wavelength, and can be produced by the sudden
</p>
<p>deceleration of rapidly moving electrons at a target material. If an electron falls through a potential
</p>
<p>difference of V volt, it acquires an energy eV electron-volt (eV), where e is the charge on an electron.
</p>
<p>This energy may be expressed as quanta of X-rays of wavelength l, where each quantum is given by
</p>
<p>l &frac14; hc=&eth;eV&THORN; (3.1)
</p>
<p>h being the Planck constant and c the speed of light in vacuum. Substitution of numerical values into
</p>
<p>(3.1) leads to
</p>
<p>l &frac14; 12:4=V (3.2)
</p>
<p>where V is measured in kilovolt and l is given in Angstrom units (Å). The wavelength range of X-rays
</p>
<p>is approximately 0.1&ndash;100 Å, but for the purposes of practical X-ray crystallography, the range used is
</p>
<p>restricted to 0.7&ndash;2.5 Å.
</p>
<p>3.1.1 X-Rays and White Radiation
</p>
<p>Except for synchrotron radiation, which is discussed in Sect. 3.1.6, a widely used source of X-rays in
</p>
<p>conventional crystallography laboratories is the sealed hot-cathode tube with a rotating anode,
</p>
<p>illustrated diagrammatically in Fig. 3.1. Electrons are emitted from a heated tungsten filament, the
</p>
<p>cathode, and accelerated by a high voltage, 40 kV or more, towards a water-cooled target anode,
</p>
<p>usually made of copper or molybdenum. A large proportion of the energy reaching the target is
</p>
<p>dissipated as heat on account of multiple collisions within the target material, but about 10% of it is
</p>
<p>converted usefully for X-ray crystallographic purposes. In order to dissipate the heat rapidly and
</p>
<p>efficiently, the water-cooled anode is rotated, as indicated in the diagram of Fig. 3.1.
</p>
<p>M. Ladd and R. Palmer, Structure Determination by X-ray Crystallography:
</p>
<p>Analysis by X-rays and Neutrons, DOI 10.1007/978-1-4614-3954-7_3,
# Springer Science+Business Media New York 2013
</p>
<p>111</p>
<p/>
</div>
<div class="page"><p/>
<p>As a consequence, a higher accelerating voltage can be applied to the tube, which results in a more
</p>
<p>powerful X-ray source. If the energy eV is not too high, there will be a continuous distribution of
</p>
<p>X-ray wavelengths, &ldquo;white&rdquo; radiation, or Bremsstrahlung (Ger. &frac14; braking radiation), as shown in
Fig. 3.2. With an increase in the accelerating voltage V, the intensity of the radiation increases, and
</p>
<p>the maximum of the curve moves to shorter wavelengths.
</p>
<p>Fig. 3.1 Schematic diagram of a sealed crystallographic X-ray tube. The target anode is provided with a means
of rotation, so as to aid the dissipation of heat generated by the electron impact on the target and to prolong the life of
</p>
<p>the target
</p>
<p>112 3 X-Rays and X-Ray Diffraction</p>
<p/>
</div>
<div class="page"><p/>
<p>3.1.2 Characteristic Radiation
</p>
<p>At a particular higher value of V, the impinging electrons excite inner electrons in the target atoms.
</p>
<p>Other electrons from higher energy levels then fall back to the inner levels, and their transitions are
</p>
<p>accompanied by the emission of X-radiation of high intensity, characteristic of the material of
</p>
<p>the target. The X-ray wavelength depends on the energies of the two levels involved, E1 and E2,
</p>
<p>such that
</p>
<p>l &frac14; hc=jE2 � E1j (3.3)
</p>
<p>Figure 3.3 illustrates the curve of radiation intensity against X-ray wavelength, when the accel-
</p>
<p>erating voltage is sufficient to excite the K spectrum of the target metal. The K spectrum consists of
</p>
<p>the Ka and Kb wavelengths, which are always produced together, and correspond to electrons
</p>
<p>falling back to the K level from the L and M levels respectively. Two slightly different L energy
</p>
<p>levels exist, so that the important Ka spectrum consists of two components, Ka1 and Ka2, of
</p>
<p>closely similar wavelength. Similarly, M ! K transitions give rise to Kb characteristic radiation.
The wavelengths of the K radiations for a target material of copper are: Kb1 &frac14; 1.39222 Å,
Kb2 &frac14; 1.38109 Å, Ka1 &frac14; 1.54056 Å, and Ka2 &frac14; 1.54439 Å. The mean value for Ka is obtained
by averaging the Ka1 and Ka2wavelengths in their intensity ratio of 2:1, thus giving the average value
</p>
<p>of 1.54184 Å for Ka. The a1, a2 doublet is resolved when the angle of scatter is large, that is, at high
</p>
<p>values of the Bragg angle y (q.v.).
</p>
<p>Fig. 3.2 Variation of intensity with wavelength for an X-ray tube, for three different operating voltages; as V increases,
the maximum wavelength in the continuous spectrum moves to shorter wavelengths, in accordance with (3.2)
</p>
<p>3.1 Generation and Properties of X-Rays 113</p>
<p/>
</div>
<div class="page"><p/>
<p>3.1.3 Absorption of X-Rays
</p>
<p>All materials absorb X-rays, and the transmitted intensity is attenuated according to an exponential law:
</p>
<p>I &frac14; I0 exp&eth;�mt&THORN; (3.4)
</p>
<p>I0 and I are, respectively, the incident and transmitted intensities, m is the linear absorption coefficient
</p>
<p>of the material, and t is the path length for X-rays through the material. The absorption of X-rays
</p>
<p>increases with the atomic numbers of the elements in the absorbing material.
</p>
<p>The variation of m with wavelength is illustrated in Fig. 3.4, which refers to elemental nickel. The
</p>
<p>absorption coefficient m of any material decreases approximately as l5/2, so that as l falls, the energy
</p>
<p>of the radiation (hc/l) becomes greater and more penetrating. With continuing decrease in wave-
</p>
<p>length, a position is reached where the energy of the radiation is sufficient to eject an electron from the
</p>
<p>L energy level of an atom of the material. At this point, known as the absorption edge, or resonance
</p>
<p>level, the value of m is greatly enhanced. As the wavelength decreases further, the absorption
</p>
<p>coefficient continues to fall off as before. In the case of nickel, this particular L absorption edge
</p>
<p>occurs at a wavelength of 1.4886 Å.
</p>
<p>Absorption edges are important in selecting the correct radiation for a particular application. For
</p>
<p>example, copper X-radiation would be unsuitable for materials containing a high percentage of iron.
</p>
<p>The K absorption edge for iron is 1.7433 Å, so that radiation of this wavelength would be strongly
</p>
<p>absorbed by the iron moiety and subsequently re-emitted as the characteristic K spectrum of iron.
</p>
<p>In such a case, molybdenum radiation, l(Ka) &frac14; 0.71073 Å, would be a satisfactory alternative.
</p>
<p>Fig. 3.3 Characteristic
K spectrum from an X-ray
</p>
<p>tube superimposed upon
</p>
<p>the &ldquo;white&rdquo; radiation, or
</p>
<p>continuous spectrum
</p>
<p>114 3 X-Rays and X-Ray Diffraction</p>
<p/>
</div>
<div class="page"><p/>
<p>If a material consists of a single elemental species, m in (3.4) may be termed the atomic absorption
</p>
<p>coefficient [1] ma, given by
</p>
<p>ma &frac14; Mrm=&eth;DmL&THORN; (3.5)
</p>
<p>whereMr is the relative atomic mass, Dm the density of the material, and L the Avogadro constant. Of
</p>
<p>more general applicability is the mass absorption coefficient mm, given by mm &frac14; m/Dm, so that
ma &frac14; mm(Mr/L). For a compound. we have
</p>
<p>m &frac14;
X
</p>
<p>i
</p>
<p>mm;iDm;i (3.6)
</p>
<p>where mm,i is the mass absorption coefficient for the ith species of partial density Dm,i in the
</p>
<p>compound; Dm,i is calculated for the ith species as Dm;iMm;i=Mr. For example, sodium chloride,
</p>
<p>NaCl, has a density of 2165 kg m�3, and the relative atomic masses and mass absorption coefficients
</p>
<p>for Na and Cl are 22.98 and 3.01 m2 kg�1, and 35.45 m2 kg�1 and 10.6 m2 kg�1 for Na and Cl,
respectively. Hence, the linear absorption coefficient for NaCl is given by
</p>
<p>m &frac14; 2165&frac12;&eth;3:01� 22:98=58:43&THORN; &thorn; &eth;10:6� 35:45=58:43&THORN;� &frac14; 1:65� 104 m�1
</p>
<p>and this parameter is needed in the correction of X-ray intensities, Sect. 4.1.3. The attenuation factor
</p>
<p>I/I0, for a crystal of NaCl of thickness 0.1 mm in the path of the X-ray beam, is then exp(�1.65
� 104 � 0.1 � 10�3), or 0.192.
</p>
<p>Fig. 3.4 Variation with
wavelength of the linear
</p>
<p>absorption coefficient m for
</p>
<p>nickel; the discontinuity at
</p>
<p>approximately 1.4886 Å
</p>
<p>corresponds with the L
</p>
<p>absorption edge of the
</p>
<p>element
</p>
<p>3.1 Generation and Properties of X-Rays 115</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec4_4">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec4_4</a></div>
</div>
<div class="page"><p/>
<p>3.1.4 Monochromatic Radiation
</p>
<p>Usually X-ray structure analysis requires monochromatic radiation, but Figs. 3.2 and 3.3 show that
</p>
<p>X-ray sources contain a range of wavelengths. However, and in particular for radiation from a
</p>
<p>copper target, we note that the absorption edge for nickel (1.4886 Å) lies between the wavelengths
</p>
<p>for Cu Ka and Cu Kb radiations. The effect of passing the X-rays from a copper target through a
</p>
<p>nickel foil of ca. 0.018 mm thickness is shown in Fig. 3.5, a superposition of Figs. 3.3 and 3.4.
</p>
<p>The Kb radiation is almost totally absorbed by the nickel, and the &ldquo;white&rdquo; radiation is decreased
</p>
<p>significantly in intensity. There is also a loss in intensity of the Ka radiation, but the intense part of the
</p>
<p>beam behaves as a closely monochromatic, or filtered, radiation. Evidence for a residual presence of
</p>
<p>Kb and white radiation may be seen in Fig. 9.5 as weak reflections just below the strong reflections,
</p>
<p>that is, at lesser y values; the very strong reflections also show spots from tungsten La radiation at
</p>
<p>still lower y values (lLa &frac14; 1.476 Å). A similar degree of monochromatization can be obtained with
molybdenum radiation if a filter of zirconium foil is used.
</p>
<p>Crystals themselves can act as monochromators through application of the Bragg equation (q.v.);
</p>
<p>we consider this topic in Sect. 5.8.1ff.
</p>
<p>3.1.5 Collimation
</p>
<p>A collimator is used to define the angular limit of the X-ray beam. The simplest collimator is the
</p>
<p>pinhole type, a cylindrical device in which the beam is limited usually by two defining circular
</p>
<p>apertures together with a guard aperture. The collimation provided is satisfactory in many situations,
</p>
<p>and a divergence angle of approximately 0.02 rad is typical. Figure 3.6a illustrates an elegant
</p>
<p>apparatus for taking a back-reflection Laue photograph of a metal casting, showing also the type of
</p>
<p>collimator just described.
</p>
<p>Fig. 3.5 Diagrammatic
superposition of the curves
</p>
<p>of Figs. 3.3 and 3.4, for Cu
</p>
<p>Ka X-radiation; the
</p>
<p>strongly preferential
</p>
<p>absorption of the Kb
</p>
<p>radiation results in an
</p>
<p>almost monochromatic Ka
</p>
<p>radiation
</p>
<p>116 3 X-Rays and X-Ray Diffraction</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Fig5_8">http://dx.doi.org/10.1007/978-1-4614-3954-7#Fig5_8</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec40_5">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec40_5</a></div>
</div>
<div class="page"><p/>
<p>Improved collimators have been described [2] in which monochromatic X-rays, obtained by
</p>
<p>crystal reflection, are guided by multiple reflections within thin-walled borosilicate glass capillary
</p>
<p>tubes (Fig. 3.6b). It is necessary for the X-rays to be reflected from the tube walls at less than the
</p>
<p>critical angle yc:
</p>
<p>yc=mrad &frac14; 30=V
</p>
<p>where the voltage V is measured in keV. The divergence at input could be 60&ndash;120 mrad, or 5&ndash;10�, and
</p>
<p>2&ndash;4 mrad at output. The system allows a useful X-ray flux from Cu Ka radiation to be produced that is
</p>
<p>four to five times stronger than that obtained by simple pinhole collimation, with comparable initial
</p>
<p>divergence, and the output beam produces uniform diffraction peaks while the higher energy
</p>
<p>Bremsstrahlung and background radiations are insignificant.
</p>
<p>Fig. 3.6 Collimation of X-rays. (a) Colli-
mator employed in a back-reflection Laue
</p>
<p>experiment (courtesy of Jackson Electron-
</p>
<p>ics Newark, UK)
</p>
<p>(b) Cross-section of a multifiber
polycapillary fiber collimator, with
</p>
<p>about 400 50 m diameter channels
</p>
<p>(Gibson D, Gibson W (2002) Adv X-ray
</p>
<p>Anal 45; reproduced by permission of
</p>
<p>ICDD)
</p>
<p>3.1 Generation and Properties of X-Rays 117</p>
<p/>
</div>
<div class="page"><p/>
<p>3.1.6 Synchrotron Sources
</p>
<p>A synchrotron is a large-scale particle accelerator designed primarily as a tool for fundamental studies
</p>
<p>in particle physics. However, it has many applications, and in X-ray crystallography it functions as a
</p>
<p>very powerful source of X-rays, with an intensity of several orders of magnitude greater than that of
</p>
<p>the sealed X-ray tube. It can generate brilliant beams of electromagnetic radiation from infra-red to
</p>
<p>X-ray wavelengths.
</p>
<p>The Diamond Synchrotron Source
Figure 3.7 is a schematic diagram of the Diamond synchrotron that was opened at Harwell Science
</p>
<p>and Innovation Campus in 2007. It consists of several elements, as follow:
</p>
<p>&bull; In the injection system (1), an electron gun produces a series of pulses of electrons with an energy
</p>
<p>of approximately 90 keV. These electrons are then accelerated in the linear accelerator LINAC,
</p>
<p>which raises their energy to about 100 MeV.
</p>
<p>&bull; The 100 MeV electrons are further accelerated in the booster synchrotron (2). Thirty-six dipole
</p>
<p>bending magnets of field up to 0.8 T are used to curve the electron beam around the bends in the
</p>
<p>ring, and then they are ejected from the booster with energy of 3 GeV.
</p>
<p>&bull; The storage ring (3) contains 48 bending magnets that curve the electron beam between the
</p>
<p>straight sections. The total deflection is 360� and the length of the orbit is 561.6 m. The ring is
</p>
<p>under vacuum, and the 3 GeV electrons complete each circuit in 1.9 � 10�6 s, thus traveling at
</p>
<p>98.6% of the speed of light.
</p>
<p>&bull; Each beamline (4) comprises three main sections: an optics hutch, an experimental hutch that
</p>
<p>houses experimental equipment, and a control hutch, and is designed for a specific experimental
</p>
<p>application.
</p>
<p>&bull; The front end (5) channels the synchrotron light into a beamline. It monitors the beam in passing
</p>
<p>through to the optics hutch.
</p>
<p>&bull; The optics hutch (6) contains mirrors and diffraction gratings or crystals, the purpose of which is to
</p>
<p>filter the beam so as to obtain the desired radiation wavelength, and also to focus it on to the sample
</p>
<p>under investigation.
</p>
<p>&bull; The experimental hutch (7) houses the technical equipment for carrying out the experiments and
</p>
<p>provides a number of different specialized X-ray detectors for the various kinds of experiments.
</p>
<p>Fig. 3.7 Schematic illustration of the Diamond synchrotron facility at Harwell; the numbered sections are described in
the text (reproduced by courtesy of the Diamond Light Source, Harwell Science and Innovation Campus)
</p>
<p>118 3 X-Rays and X-Ray Diffraction</p>
<p/>
</div>
<div class="page"><p/>
<p>&bull; The control hutch (8) provides the operators with a computer-controlled monitoring system for the
</p>
<p>experiment in progress.
</p>
<p>&bull; A radiofrequency cavity in the region of (9) contains an electromagnetic field that makes up for the
</p>
<p>energy that the electrons lose in the form of synchrotron radiation as they travel around the ring.
</p>
<p>There is also Diamond House, which provides for Diamond staff accommodation, meeting, and
</p>
<p>conference rooms, and also access to the synchrotron facility itself.
</p>
<p>The Diamond IO2 beamline, which is concerned with macromolecular X-ray crystallography, has
</p>
<p>a flux of about 1.5 � 1012 photon s�1 and, typically, an operating wavelength of 0.98 Å, giving a
resolution of 1.09 Å. Beamlines IO3 and IO4 are similar to IO2, and beamlines IO4-1 and I24 are
</p>
<p>available for microfocus macromolecular crystallography. A fascinating video-description of a trip
</p>
<p>around the Diamond ring may be found at a web site [3].
</p>
<p>Polarization of Synchrotron Radiation
An important difference between X-radiation from sealed tube or rotating anode sources and X-rays
</p>
<p>produced by a synchrotron is their physical state of polarization; see also Sect. 3.2.4. X-rays generated
</p>
<p>from conventional laboratory sources are totally non-polarized unless a crystal monochromator is
</p>
<p>employed. In contrast, synchrotron radiation is 100% linearly polarized in the plane of the electron
</p>
<p>beam orbit and elliptically polarized above and below the plane. In addition, the output radiation is
</p>
<p>pulsed, because the electrons do not form a uniform stream.
</p>
<p>Figure 3.8a considers a three-dimensional X-ray wave at the origin, with components of oscillation
</p>
<p>vibrating in the y and z directions. The beam from a synchrotron is plane-polarized, with the
</p>
<p>component in the z direction being an order of magnitude less in intensity. X-ray beams from crystal
</p>
<p>monochromators, because they have undergone Bragg diffraction, are also polarized, but the extent is
</p>
<p>considerably less. The effect of this primary or beam polarization of synchrotron radiation on the
</p>
<p>intensity of the diffraction pattern must be taken into account during data processing. This is achieved
</p>
<p>through application of a polarization factor P that includes a source-dependent property, with
</p>
<p>components normal and parallel to the plane of polarization:
</p>
<p>P &frac14; &eth;Ik � I?&THORN;=&eth;Ik &thorn; I?&THORN;
</p>
<p>where Ik and I&perp; are the intensities of the electrical fields of the X-ray beam along and normal to the
axis of a rotation camera, respectively. The value of P is calibrated for each workstation at a
</p>
<p>synchrotron installation. For the Diamond station, P is approximately 0.8, whereas for a graphite
</p>
<p>monochromator, P is equal to 0.11.
</p>
<p>A typical synchrotron radiation spectrum is shown in Fig. 3.8b and may be compared with that
</p>
<p>from the sealed tube. The photon intensity is given in units of photon per second for a horizontal
</p>
<p>angular aperture of 1 mrad (3.4 min of arc), with a 1 A beam current and a 0.1% spectral bandwidth,
</p>
<p>after performing vertical integration over the full angular divergence of the radiation above and below
</p>
<p>the orbital plane.
</p>
<p>The flux attainable in practice depends upon the multiplying factors set by the values of the
</p>
<p>dependent parameters. A horizontal aperture of an experimental workstation may be less than 1 mrad
</p>
<p>for topography, typically 5&ndash;10 mrad for the majority of spectroscopy experiments, and up to 40 mrad
</p>
<p>for the high-aperture port used for time-resolved measurements. The flux available will change
</p>
<p>proportionally if this resolution is varied. The stored current and, hence the photo flux, gradually
</p>
<p>decline as electrons are lost by scattering from closed electron orbits. The beam lifetime, that is, the
</p>
<p>time of fall to approximately 1/e of the initial intensity, is approximately 8 h.
</p>
<p>3.1 Generation and Properties of X-Rays 119</p>
<p/>
</div>
<div class="page"><p/>
<p>Insertion Devices: Wigglers and Undulators
The output characteristics of the synchrotron can be modified by devices inserted into the straight
</p>
<p>sections between themagnets. An insertion device is an array of magnets which can be inserted into the
</p>
<p>straight sections of the storage ring and cause the electron beam to follow a wiggling or undulating
</p>
<p>Fig. 3.8 (a) A three-
dimensional wave
</p>
<p>traversing the origin, with
</p>
<p>oscillatory components
</p>
<p>vibrating along the x and y
</p>
<p>directions: (i) Conventional
</p>
<p>X-ray tube radiation, (ii)
</p>
<p>Synchrotron radiation (not
</p>
<p>to scale). The radiation
</p>
<p>from a synchrotron is
</p>
<p>plane-polarized, with the z
</p>
<p>component an order of
</p>
<p>magnitude less than that in
</p>
<p>the y direction. The spread
</p>
<p>of radiation in the vertical
</p>
<p>(z) direction is given as
</p>
<p>D &frac14; mec2/E, where E is the
electron beam energy.
</p>
<p>(b) Spectral curves in the
X-ray region from a normal
</p>
<p>bending magnet and a
</p>
<p>wiggler for a 2 GeV 1 A
</p>
<p>beam in the synchrotron
</p>
<p>radiation source, and the
</p>
<p>types of experiment used in
</p>
<p>the wavelength regions
</p>
<p>specified. The peak of the
</p>
<p>curve is approximately
</p>
<p>1.4lc, corresponding to the
</p>
<p>maximum output of energy
</p>
<p>per unit wavelength; lc is
</p>
<p>the critical wavelength for
</p>
<p>the synchrotron
</p>
<p>120 3 X-Rays and X-Ray Diffraction</p>
<p/>
</div>
<div class="page"><p/>
<p>path, so as to produce more intense, tuneable light. Insertion devices come in two main types: wigglers
</p>
<p>and undulators.
</p>
<p>A wiggler is an insertion device that consists of an array of dipolar magnets of alternating
</p>
<p>polarity which cause the electron beam to follow a wiggling path of oscillations perpendicular to its
</p>
<p>general direction. This causes the light to be produced in a wide cone, spanning a broad spectrum of
</p>
<p>X-rays. Wigglers are used in beamlines where the priority is for very high energy X-rays.
</p>
<p>An undulator, which is more common at the Diamond installation, is an insertion device that
</p>
<p>produces a very bright light in a very narrow beam. By varying the separation of the magnet arrays, it
</p>
<p>is possible to tune the undulator and choose the energy that is generated. They can be used to produce
</p>
<p>very high energy X-rays over a continuous frequency range, which is essential for many experiments,
</p>
<p>particularly in protein crystallography.
</p>
<p>The synchrotron source may be said to have revolutionized X-ray crystallography in certain
</p>
<p>applications. It has enabled a rapid collection of data to be achieved, and so is of great value in
</p>
<p>dealing with relatively unstable crystals, such as proteins, with poorly diffracting specimens or
</p>
<p>polymers, in time-resolved studies, or in solid-state reactions and other transformations, including
</p>
<p>enzyme-catalyzed processes, or in X-ray topographical studies of crystal defects.
</p>
<p>Laser-Wakefield Acceleration
Relatively recent work on synchrotron sources has concentrated on a reduction in the physical size of
</p>
<p>the accelerator. If a high-powered, femtosecond (10�15 s) laser pulse is focused into a plasma, an
</p>
<p>electrostatic wake is produced which can be harnessed to accelerate electrons to GeV energies within
</p>
<p>centimeter distances. Electrons become separated from the positive ions, and a ponderomotive force,
</p>
<p>which is a non-linear force that a charged particle experiences in an inhomogeneous oscillating
</p>
<p>electromagnetic field, arising from the laser light pushes aside the plasma electrons so as to create a
</p>
<p>wake from the plasma. The associated electrostatic fields can produce an accelerating field of three to
</p>
<p>four orders of magnitude greater than in the conventional accelerator.
</p>
<p>In one application of the procedure [4], a laser pulse of 37 fs is fired at a cell of hydrogen gas
</p>
<p>atoms, thus energizing the electrons and causing them to break free. The positive attraction of the
</p>
<p>nucleus acts to retain the electrons and the result is an oscillation about the nucleus that produces
</p>
<p>a plasma wave in the cell. This excitation is similar to the water-wave behind a motorboat; hence, the
</p>
<p>term wake-field (or wakefield). Other electrons &ldquo;ride&rdquo; this wave at relativistic speeds and so generate
</p>
<p>X-rays through their oscillatory changes.
</p>
<p>An important feature in the process is the use of a miniature undulator. The combination of a
</p>
<p>1.5 cm accelerator and a 30 cm magnetic undulator produced electron energies of 210 MeV. Further
</p>
<p>research aims at higher electron energies, so that laser-wakefield acceleration can generate X-rays in
</p>
<p>the useful wavelength range for diffraction experiments, and with facilities of centimeter size. This
</p>
<p>development has great potential for X-ray studies on proteins and viruses, where powerful X-ray
</p>
<p>sources are desirable, because of the low stability of some of these compounds.
</p>
<p>3.2 X-Ray Scattering
</p>
<p>Scattering occurs generally when electromagnetic radiation interacts with matter. Two everyday
</p>
<p>examples of scattering are the blue color of the sky and the haloes around distant car lights at night
</p>
<p>that arise from Rayleigh scattering, a highly wavelength-dependent elastic scattering of the light by
</p>
<p>gas molecules or dust particles in the air. The pattern seen when looking at a sodium street-lamp
</p>
<p>through a stretched handkerchief or an umbrella, which are approximately two-dimensional net
</p>
<p>structures, shows an aspect of diffraction similar to that which we shall be discussing with X-rays
</p>
<p>3.2 X-Ray Scattering 121</p>
<p/>
</div>
<div class="page"><p/>
<p>and crystals, namely the interaction of radiation with a periodic distribution of matter. The X-ray
</p>
<p>scattering from a crystal is described in terms of the intensity of the scattering function and the angle
</p>
<p>of scatter.
</p>
<p>3.2.1 Scattering by a Single Electron
</p>
<p>If a plane monochromatic X-ray beam of wavelength l is incident upon an electron at an
</p>
<p>origin O, then the amplitude C2y of the scattered beam at a point P in the forward direction at
</p>
<p>unit distance from the origin is given by [5]
</p>
<p>C2y &frac14; f2yC0 (3.7)
</p>
<p>where C0 is the amplitude of the incident wave, 2y is the scattering angle, and f2y is a constant
</p>
<p>of proportionality that we shall discuss more fully shortly.
</p>
<p>3.2.2 Scattering by Two or More Electrons
</p>
<p>Let a second electron be introduced at a point A, Fig. 3.9. We need now to determine the phase
</p>
<p>difference at any point P in the forward direction, whereOP is much greater thanOA, for the wavelets
</p>
<p>scattered by the two electrons. The path difference d between the two wavelets scattered byO and A is
</p>
<p>OY � AX; thus the phase difference f is (2p/l)d, that is,
</p>
<p>f &frac14; &eth;2p=l&THORN;&eth;OY � AX&THORN; (3.8)
</p>
<p>If the distance OA is jrj, then the incident and diffracted waves may be defined by the unit vectors
s0 and s, respectively, such that AX &frac14; r&middot;s0 and OY &frac14; r&middot;s. Thus,
</p>
<p>f &frac14; 2p&eth;r � s� r � s0&THORN;=l &frac14; 2pr � &eth;s=l� s0=l&THORN; &frac14; 2pr � S (3.9)
</p>
<p>From Fig. 3.10, it is clear that S is a vector normal to a plane through O that may be regarded
conveniently as a reflecting plane; hence,
</p>
<p>S &frac14; js� s0j=l (3.10)
</p>
<p>so that the magnitude jSj, or S, is equal to 2 sin y=l.
</p>
<p>Fig. 3.9 Combined
scattering at two centers
</p>
<p>O and A; s0 and s are unit
vectors in the incident and
</p>
<p>scattered beams,
</p>
<p>respectively; dv is a small
</p>
<p>volume element at A.
</p>
<p>122 3 X-Rays and X-Ray Diffraction</p>
<p/>
</div>
<div class="page"><p/>
<p>The amplitude of the wave scattered at P by the two electrons is thus
</p>
<p>C2y &frac14; f2y &thorn; f2y exp&eth;i2pr � S&THORN; &frac14; f2y&frac12;1&thorn; exp&eth;i2pr � S&THORN;� (3.11)
</p>
<p>The use of the exponential term to represent relative phase is explained below in Sect. 3.2.3 on the
</p>
<p>Argand Diagram. In the case that neither of the electrons of the previous example occupies the origin,
</p>
<p>so there is no zero relative phase involved, then the number 1 on the right-hand side of (3.11) would
</p>
<p>be replaced by another exponential term. In general, the result for n electrons is given by
</p>
<p>C2y;n &frac14; f2y
Xn
</p>
<p>j&frac14;1
</p>
<p>exp&eth;i2prj � S&THORN; (3.12)
</p>
<p>This equation assumes that the n scattering species have equal power, which would be true if they
</p>
<p>were all electrons, or identical atoms. In the event that the scattering species are unequal, then the
</p>
<p>function f2y would be unique to each species and included within the summation as f2y,j. The scattering
</p>
<p>process itself introduces a phase shift with respect to the origin, but this applies equally to all atoms and
</p>
<p>may be ignored for our purposes.
</p>
<p>3.2.3 Waves and Wave Sums
</p>
<p>In (3.12), we considered the summation of waves of equal scattering power. In general, each
</p>
<p>scattering entity will be allocated its own scattering function and phase. Each wave then takes the
</p>
<p>form fj exp(ifj), where fj &frac14; rj � S. For convenience, we may drop the subscript y (or 2y) to f ; unless
we specifically want to emphasize its dependence on y; we know that such dependence is always
</p>
<p>present. Using de Moivre&rsquo;s theorem, the exponential term may be expanded into cosine (real) and
</p>
<p>sine (imaginary) components: exp&eth;�if&THORN; &frac14; cosf� i sinf, and a straightforward way of representing
</p>
<p>a wave is in the complex plane of an Argand diagram.
</p>
<p>Fig. 3.10 Relationship of
the scattering vector S, or
(s � s0)/l, to the reflecting
plane (hkl). The vector S is
normal to the plane (hkl),
</p>
<p>and its magnitude jSj, or S,
is 2 sin y=l
</p>
<p>3.2 X-Ray Scattering 123</p>
<p/>
</div>
<div class="page"><p/>
<p>Argand Diagram
In Fig. 3.11, we show the combination of two waves of scattering factors f1 and f2 on an Argand
</p>
<p>diagram to give a resultant F, which we shall later identify with the structure factor, Sect. 3.5.1:
</p>
<p>F &frac14; f1 exp&eth;if1&THORN; &thorn; f2 exp&eth;if2&THORN; (3.13)
</p>
<p>F is a scalar quantity but may be manipulated like a vector in the complex plane, having both a
</p>
<p>magnitude and direction; exp(if) may be regarded as an operator that rotates f counterclockwise on
</p>
<p>an Argand diagram by the angle f measured from the positive, real axis.
</p>
<p>The foregoing analysis may be extended to n waves. The resultant sum F is, from (3.13),
</p>
<p>F &frac14; f1 exp&eth;if1&THORN; &thorn; f2 exp&eth;if2&THORN; &thorn; � � � &thorn; fj exp&eth;ifj&THORN; &thorn; � � � &frac14;
Xn
</p>
<p>j&frac14;1
</p>
<p>fj exp&eth;ifj&THORN; (3.14)
</p>
<p>The Argand diagram, Fig. 3.12, expresses (3.14) as a polygon of f-&ldquo;vectors&rdquo; for the combination of
</p>
<p>six waves, and the resultant F may be expressed as
</p>
<p>F &frac14; jFj exp&eth;if&THORN; (3.15)
</p>
<p>where the amplitude jFj is obtained from
</p>
<p>jFj &frac14; &eth;FF�&THORN;1=2 (3.16)
</p>
<p>F* is the complex conjugate of F, that is, jFj exp&eth;�if&THORN;, Fig. 3.13. Resolving the resultant F of the six
</p>
<p>waves into its real and imaginary components, we have
</p>
<p>Fig. 3.11 Combination of
the two waves f1 exp(if1)
</p>
<p>and f2 exp(if2), shown as
</p>
<p>vectors on an Argand
</p>
<p>diagram. The resultant is
</p>
<p>F, and its phase is
</p>
<p>expressed by the angle
</p>
<p>f between F and the R
</p>
<p>(real) axis
</p>
<p>124 3 X-Rays and X-Ray Diffraction</p>
<p/>
</div>
<div class="page"><p/>
<p>Fig. 3.12 Combination of six waves on an Argand diagram: F &frac14;P6j&frac14;1 fj exp&eth;ifj&THORN;
</p>
<p>Fig. 3.13 The quantity F(hkl) for a reflection from an (hkl) family of planes and its components A0(hkl) and B0(hkl);
f(hkl) is the phase angle for F(hkl). The conjugate F*(hkl), or F&eth;h k l&THORN;, has the same magnitude for A0 and B0, but the sign
of B0 is reversed; hence f&eth;hkl&THORN; &frac14; �f&eth;h k l&THORN;. In a centrosymmetric structure, F(hkl) lies along the real axis, so that F
(hkl) &frac14; A0(hkl), or F(hkl), and its phase is either 0 or p. Then, since A0(hkl) &frac14; jF(hkl)j cos f, it is common to speak of
the sign of F(hkl), that is, �F(hkl), in a centrosymmetric structure
</p>
<p>3.2 X-Ray Scattering 125</p>
<p/>
</div>
<div class="page"><p/>
<p>jFj &frac14; &eth;A02 &thorn; B02&THORN;1=2 (3.17)
</p>
<p>where
</p>
<p>A0 &frac14;
Xn
</p>
<p>j&frac14;1
fj cosfj (3.18)
</p>
<p>and
</p>
<p>B0 &frac14;
Xn
</p>
<p>j&frac14;1
fj sinfj (3.19)
</p>
<p>The phase f of the resultant F is given by
</p>
<p>f &frac14; tan�1&eth;B0=A0&THORN; (3.20)
</p>
<p>Graphical Representation of Wave Sums
When waves of equal frequency, those in which we are interested, combine, the resultant can be
</p>
<p>determined through (3.17&ndash;3.20). If two waves are in exact register&mdash;path difference, nl&mdash;the
</p>
<p>resultant amplitude is simply the sum of the individual amplitudes. If the waves are exactly out of
</p>
<p>phase&mdash;path difference, (n + 1)l/2&mdash;the resultant is the difference between the amplitudes. All cases
</p>
<p>between these extremes are possible, depending on the values of the amplitudes jFj and phase anglesf.
In Fig. 3.14, we represent the combination of two waves, one of amplitude 100 (f1) and phase 0
</p>
<p>�, and
</p>
<p>the other of amplitude 50 (f2) and phase 240
�, with respect to the origin. From the foregoing, the
</p>
<p>amplitude of the resultant becomes
</p>
<p>jFj &frac14; &frac12;&eth;f1 cosf1 &thorn; f2 cosf2&THORN;
2 &thorn; &eth;f1 sinf1 &thorn; f2 sinf2&THORN;
</p>
<p>2�1=2 (3.21)
</p>
<p>and the phase of the resultant is
</p>
<p>f &frac14; tan�1 &frac12;&eth;f1 sinf1 &thorn; f2 sinf2&THORN;=&eth;f1 cosf1 &thorn; f2 cosf2&THORN;� (3.22)
</p>
<p>Importance of Correct Phases
As shown in the legend to Fig. 3.14, the resultant amplitude jFj is 86.6 and its phase f is 330�. From
</p>
<p>the figure we can see the importance of the correct phase of a wave (reflection). The peaks (atomic
</p>
<p>positions) in this figure occur at 330� � 2p (�360�). If, on one hand, we decrease f2 to 25, the
resultant amplitude is 90.1, because the sine term subtracts a smaller amount, and the phase is 346�, a
</p>
<p>small change in position. If on the other hand, we decrease the phase by half, leaving f2 unchanged,
</p>
<p>then the resultant amplitude is again 86.6, but the phase (atomic position) is now at +30�, a very
</p>
<p>different situation; see Problem 7.11.
</p>
<p>126 3 X-Rays and X-Ray Diffraction</p>
<p/>
</div>
<div class="page"><p/>
<p>3.2.4 Coherent and Incoherent Scattering
</p>
<p>Coherent (Thomson) Scattering: Polarization
In coherent scattering, the incident and scattered waves have the samewavelength, and there is a definite
</p>
<p>phase relationship between the incident and scattered radiations. When X-rays fall upon an electron, the
</p>
<p>alternating electric-field vector imparts an alternating acceleration to the electron. Classical electromag-
</p>
<p>netic wave theory shows that an accelerated charged particle emits radiation, through a process of
</p>
<p>absorption and re-emission, the emitted radiation traveling in all directions for a given angle of scatter. A
</p>
<p>theoretical treatment of Thomson scattering [5] shows that the intensity I2y of the scattered radiation of
</p>
<p>incident intensity I0, defined as power per unit solid angle, is
</p>
<p>I2y &frac14;
1
</p>
<p>2
&frac12;e2=&eth;4pe0c2me&THORN;�2&eth;1&thorn; cos22y&THORN;I0 (3.23)
</p>
<p>Fig. 3.14 Combination of two waves of amplitudes 100 (f1) and 50 (f2) with phases 0
� (f1) and 240
</p>
<p>� (f2), respectively.
</p>
<p>The resultant wave has an amplitude jFj and a phase f. (a) First wave. (b) Second wave. (c) Resultant wave F, given
by &frac14; [(100 + 50 cos 240)2 + (50 sin 240)2]1/2 &frac14; 86.6; f &frac14; tan�1[(50 sin 240)/(100 + 50 cos 240)] &frac14; �30� (330�)
</p>
<p>3.2 X-Ray Scattering 127</p>
<p/>
</div>
<div class="page"><p/>
<p>where e is the charge on an electron, e0 is the permittivity of a vacuum, c is the speed of light in a
</p>
<p>vacuum, and me is the mass of the electron. The factor 1=me shows how electrons are the only
</p>
<p>effective scattering species for X-rays: even the lightest species, hydrogen, although it has the same
</p>
<p>magnitude of charge, is ca. 1,840 times heavier than the electron. This result shows also that neutrons
</p>
<p>are excluded from the category of X-ray scattering species because they are uncharged particles. The
</p>
<p>term 1
2
&eth;1&thorn; cos22y&THORN; is a geometrical factor known as the polarization factor p for X-ray scattering,
</p>
<p>which we shall encounter further in ensuing chapters.
</p>
<p>p &frac14; 1
2
&eth;1&thorn; cos22y&THORN; (3.24)
</p>
<p>Incoherent (Compton) Scattering
With incoherent scattering, the wavelength of the scattered radiation is longer than that of the incident
</p>
<p>radiation, which implies a loss of energy in the scattering process, owing to elastic collisions of
</p>
<p>electrons with photons. An analysis [5] shows that in Compton scattering, the wavelength change dl
</p>
<p>is given by
</p>
<p>dl &frac14; &frac12;h=&eth;mec&THORN;�&eth;1� cos 2y&THORN; (3.25)
</p>
<p>or, by inserting the fundamental constants in appropriate units,
</p>
<p>dl &eth;A) &frac14; 0:0243&eth;1� cos 2y&THORN; (3.26)
</p>
<p>Thomson scattering illustrates the particle property of the electron, whereas Compton scattering
</p>
<p>shows its wave nature.
</p>
<p>3.2.5 Scattering by an Atom
</p>
<p>In an atom, electrons are bound in levels of distinct energies, and in the scattering of X-rays by an
</p>
<p>atom both coherent and incoherent scattering are involved. A full analysis of the scattering process
</p>
<p>requires a wave-mechanical treatment, involving both modes of scattering, from which we obtain the
</p>
<p>electron density function r, where r &frac14; CC* andC* is conjugate toC; if we assume a real nature for
the electron density function then r &frac14; jCj2, and the expression may be interpreted such thatC2 dt, or
</p>
<p>r dt, represents the probability of finding the electron in a volume element dt. We shall use the
</p>
<p>Thomson formula only because incoherent scattering contributes to the background radiation and is
</p>
<p>but a small fraction of the total intensity in the case of crystalline materials.
</p>
<p>Consider a plane of atoms in a crystal. We demonstrate in the ensuing sections that all atoms on this
</p>
<p>plane scatter in phase with one another andwith the atoms in parallel planes, for a given scattering vector
</p>
<p>S. Thus, we need to consider how the electrons in any one atom combine in order to obtain the total
scattering amplitude for the atom.
</p>
<p>Let r&eth;r&THORN; dt be the probability that an electron in the chosen atom lies in a small volume element dt
distant r from the origin, the center of the atom, where r is the magnitude of the vector r, as shown in
Fig. 3.15. If f(S) represents the scattering power of the atom in the direction S, then we have from the
foregoing:
</p>
<p>f &eth;S&THORN; &frac14;
</p>
<p>&eth;
r&eth;r&THORN; exp&eth;i2pr � S&THORN; dt (3.27)
</p>
<p>128 3 X-Rays and X-Ray Diffraction</p>
<p/>
</div>
<div class="page"><p/>
<p>Let S make an angle C with the direction of r. Then
</p>
<p>2pr � S &frac14; &eth;4p=l&THORN; sin y r cosC &frac14; mr cosC
</p>
<p>where m &frac14; 4p(sin y)/l. Since spherical symmetry has been assumed, the volume element dt is a
</p>
<p>spherical annulus of radius r and thickness dr on S as axis, so that dt &frac14; 2pr2 dC dr.
Let mr cos C &frac14; x, so that dx &frac14; �mr sin C dC. Now (3.27) may be expressed as
</p>
<p>f &eth;S&THORN; &frac14; 2p
</p>
<p>&eth;1
</p>
<p>0
</p>
<p>r2r&eth;r&THORN;=mr dr
</p>
<p>&eth;�mr
</p>
<p>mr
</p>
<p>� exp&eth;ix&THORN; dx &frac14; 4p
</p>
<p>&eth;1
</p>
<p>0
</p>
<p>r2r&eth;r&THORN;&eth;sinmr&THORN; mr dr= (3.28)
</p>
<p>Since m &frac14; (4p sin y)/l, (3.28) may be recast as a function of jSj:
</p>
<p>f &eth;S&THORN; &frac14; 4p
</p>
<p>&eth;1
</p>
<p>0
</p>
<p>r2r&eth;r&THORN;&eth;sin 2prS&THORN; &eth;2prS&THORN;= dr (3.29)
</p>
<p>where S is 2 sin y=l.
The atomic scattering factor may be defined as the ratio of the amplitude of coherent scattering
</p>
<p>from an atom to that scattering by a single electron at the center of the atom. It follows from (3.29)
</p>
<p>that, for scattering in the forward direction, when (sin 2prS)/(2prS) &frac14; 1, the expression
&ETH;1
0
</p>
<p>r2r&eth;r&THORN; dr
becomes the total electron density for the atom. Hence, we may write
</p>
<p>f &eth;S&THORN;S&frac14;0 &frac14; Z (3.30)
</p>
<p>Fig. 3.15 Scattering by a
single atom. A section of a
</p>
<p>sphere of radius r, showing
</p>
<p>an annular ring of thickness
</p>
<p>dr at an angle C to the
</p>
<p>scattering vector S. The
volume of the annular ring
</p>
<p>is &frac12;p&eth;r &thorn; dr&THORN;2 � pr2�r sinC
</p>
<p>dC &frac14; 2pr2 sinC dC dr
(neglecting second order
</p>
<p>terms in dr)
</p>
<p>3.2 X-Ray Scattering 129</p>
<p/>
</div>
<div class="page"><p/>
<p>where Z is the total number of electrons in the atom, or atomic number; S &frac14; 0 is equivalent to
sin y &frac14; 0 in terms of the y-angle.
</p>
<p>As an example calculation, we consider the contribution of a 1s electron to f(S). We equate r(r) to
jC1sj2 and use Slater&rsquo;s analytical wavefunctions. The Slater one-electron 1s wavefunction C1s may be
written as &eth;1= ffiffiffipp &THORN;c3=21 exp&eth;�c1=r&THORN;; c1 is (Z � s)/a0, where s is Slater&rsquo;s quantummechanical screening
constant [6] and a0 is the Bohr radius for hydrogen. With lithium, for example, s1s &frac14; 0.30 and,
remembering that r / C2, we have from (3.29),
</p>
<p>f1s&eth;S&THORN; &frac14; 2c31=&eth;pS&THORN;
&eth;1
</p>
<p>0
</p>
<p>r exp&eth;�c1r&THORN; sin&eth;2pSr&THORN; dr
</p>
<p>From a table of standard integrals, or by use of the G function (see Web Appendix WA7),
</p>
<p>&eth;1
</p>
<p>0
</p>
<p>x exp&eth;�ax&THORN; sin&eth;bx&THORN; dx &frac14; 2ab=&eth;a2 &thorn; b2&THORN;2
</p>
<p>so that
</p>
<p>f1s&eth;S&THORN; &frac14; c41=&eth;c21 &thorn; p2S2&THORN;
2
</p>
<p>(3.31)
</p>
<p>In lithium, for example, there are two contributions from (3.31) and one contribution from a
</p>
<p>similar expression for the 2s electron (see also Problem 3.3). They are added to obtain the value of f at
</p>
<p>a given value of S. Atomic scattering factor data are readily available, quoted normally as functions of
sin y=l. Such data refer to systems of electrons at rest: at a finite temperature, the effective scattering
</p>
<p>from an atom is less than the value at rest, and we shall discuss this situation later in this chapter.
</p>
<p>A satisfactory calculation of rest atomic scattering factors is afforded by the equation
</p>
<p>f &eth;s&THORN; &frac14;
X4
</p>
<p>j&frac14;1
aj exp&eth;�bjs2&THORN; &thorn; cj (3.32)
</p>
<p>where s is sin y=l, and the nine constants required by the equation have been recorded for all atomic
</p>
<p>and some ionic species [7].
</p>
<p>3.3 Scattering by Regular Arrays of Atoms
</p>
<p>The interaction of X-rays with a crystal is a complex process, often described as a diffraction
</p>
<p>phenomenon although, strictly speaking, it is a combined scattering and interference effect.
</p>
<p>Two treatments, those of von Laue and Bragg, describe the process, and we shall consider them in
</p>
<p>that order.
</p>
<p>3.3.1 Laue Equations
</p>
<p>Figure 3.16 represents a regular, one-dimensional array of atoms of spacing b, imagined in three-
</p>
<p>dimensional space. Parallel X-rays are incident at an angle f2 and scattered at an angle C2 to
</p>
<p>the direction of b. The path difference for rays scattered by neighboring centers is represented by
</p>
<p>130 3 X-Rays and X-Ray Diffraction</p>
<p/>
</div>
<div class="page"><p/>
<p>AQ � BP, or b(cos C2 � cos f2); this difference must be equal to an integral number of wavelengths
for reinforcement to occur, so that
</p>
<p>b&eth;cosC2 � cosf2&THORN; &frac14; kl &frac14; b � s� b � s0 (3.33)
</p>
<p>where s and s0 have meanings as before. This equation may be written alternatively, using (3.10), as
</p>
<p>b � &eth;s� s0&THORN;=l &frac14; b � S &frac14; k (3.34)
</p>
<p>The generators of a cone of semivertical angleC2, coaxial with the row in Fig. 3.16, satisfy (3.33).
</p>
<p>For a given value of f2, there will be a series of cones corresponding to the orders of k (k &frac14; 0, 1,
2, . . .), as shown in Fig. 3.17.
</p>
<p>Fig. 3.16 Diffraction
from a row of scattering
</p>
<p>centers, of spacing b along
</p>
<p>the y axis. The Laue
</p>
<p>equation b(cos C2 �
cos f2) &frac14; kl, or b � S &frac14; k,
is satisfied by any
</p>
<p>generator of the cone
</p>
<p>Fig. 3.17 Several orders
of diffraction can arise
</p>
<p>from a row of scattering
</p>
<p>centers, for a given value
</p>
<p>of f2, corresponding to
</p>
<p>integral values for k
</p>
<p>3.3 Scattering by Regular Arrays of Atoms 131</p>
<p/>
</div>
<div class="page"><p/>
<p>The discussion is extended readily to a net a, b, so giving rise to a second condition
</p>
<p>a � &eth;s� s0&THORN;=l &frac14; a � S &frac14; h (3.35)
</p>
<p>A second cone now intersects the first cone generally in two lines, but for the special case that both
</p>
<p>(3.34) and (3.35) hold simultaneously, the two lines coincide and the atoms of the net scatter in phase,
</p>
<p>with the incident and diffracted beams lying in the plane of the net.
</p>
<p>Generalizing the argument to three dimensions, we obtain the complete Laue equations:
</p>
<p>a � S &frac14; h
</p>
<p>b � S &frac14; k
</p>
<p>c � S &frac14; l
</p>
<p>(3.36)
</p>
<p>Any of the three possible pairs of equations define scattering from the corresponding net, but for
</p>
<p>the particular case that all three equations apply simultaneously, the three-dimensional array scatters
</p>
<p>in phase and produces the hkl spectrum.
</p>
<p>Referring to Fig. 1.12 mutatis mutandis, we can rewrite (3.36) as
</p>
<p>a � S &frac14; 2a sin y=l cos a &frac14; h
</p>
<p>b � S &frac14; 2b sin y=l cos b &frac14; k
</p>
<p>c � S &frac14; 2c sin y=l cos g &frac14; l
</p>
<p>(3.37)
</p>
<p>where h, k, and l are integers. The direction cosines cos a, cos b, and cos g of the vector S, normal to
the reflecting plane, with respect to the directions of a, b, and c are, therefore, proportional to a/h, b/k,
and c/l, respectively. Successive planes (hkl) in the crystal intersect the x, y, and z axes at a/h, b/k, and
</p>
<p>c/l, respectively, so that they are parallel to the reflecting plane (hkl). Thus, (3.37) show that a
</p>
<p>scattered beam may be considered as derived from the incident beam by &ldquo;reflection&rdquo; from the (hkl)
</p>
<p>family of planes. Furthermore, if d(hkl) is the interplanar spacing, then from Sect. 1.2.2,
</p>
<p>d&eth;hkl&THORN; &frac14; a=h cos a &frac14; b=k cos b &frac14; l=c cos g (3.38)
</p>
<p>and (3.37) shows that
</p>
<p>2d&eth;hkl&THORN; sin y &frac14; l (3.39)
</p>
<p>a relation deduced by Bragg, originally in the form 2d(hkl) sin y &frac14; nl.
</p>
<p>3.3.2 Bragg Equation
</p>
<p>The deduction of the Bragg equation, sometimes thought to be ad hoc, was occasioned by the
</p>
<p>observation that if a crystal in a position that produced a scattered X-ray beam was rotated through
</p>
<p>an angle j to another scattering position, then the scattered beam had been rotated through 2j, as in
</p>
<p>the reflection of light from a plane mirror.
</p>
<p>In Fig. 3.18, two planes from a family of planes (hkl) are shown, together with the incident and
</p>
<p>reflected rays. The part of the incident beam that is not reflected at a given level passes on to be
</p>
<p>132 3 X-Rays and X-Ray Diffraction</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Fig12_1">http://dx.doi.org/10.1007/978-1-4614-3954-7#Fig12_1</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec4_1">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec4_1</a></div>
</div>
<div class="page"><p/>
<p>reflected from a deeper level in the crystal. Furthermore, all rays reflected from a given level remain
</p>
<p>in phase after reflection, because there is no path difference between them.
</p>
<p>The path difference d between the two reflected rays shown is given by
</p>
<p>d &frac14; QA2 &thorn; A2R &frac14; A1A2 cosf2 &frac14; A1A2&eth;cosf1 &thorn; cosf2&THORN;
&frac14; 2A1A2 cos&frac12;&eth;f1 � f2&THORN;=2� cos&frac12;&eth;f1 &thorn; f2&THORN;=2� (3.40)
</p>
<p>which, by simple manipulation, becomes
</p>
<p>d &frac14; 2d sin y (3.41)
</p>
<p>Since d is independent of f1 and f2, (3.41) applies to all rays in the bundle reflected from
</p>
<p>the adjacent planes. By the usual rules that apply to the combination of waves, the reflected rays
</p>
<p>will interfere with one another, the interference being at least partially destructive unless the path
</p>
<p>difference d is equal to an integral number of wavelengths. Hence, we obtain the Bragg equation, as
</p>
<p>originally formulated:
</p>
<p>2d sin y &frac14; nl (3.42)
</p>
<p>where n is an integer. The mirror-reflection analogy breaks down in practice because this equation
</p>
<p>must be satisfied for a reflection to occur, but the treatment is, nevertheless, a very useful geometrical
</p>
<p>way of looking at the X-ray diffraction process.
</p>
<p>In (3.42), n is the order of the Bragg reflection. From Sect. 2.4, we recall that d(hkl)/n &frac14; d(nh, nk,
nl), with h, k, and l taking common factors as necessary. Thus, n is included in the definition of d(hkl),
</p>
<p>and the Bragg equation now written as
</p>
<p>2d&eth;hkl&THORN; sin y&eth;hkl&THORN; &frac14; l (3.43)
</p>
<p>Each reflection from a crystal must now be considered, effectively, as first-order from the (hkl)
</p>
<p>family of planes, specified uniquely by their Miller indices. To illustrate this argument further,
</p>
<p>Table 3.1 lists data for planes parallel to (120) in a cube of side 5 Å.
</p>
<p>The Bragg and von Laue treatments are equivalent, and we shall use them as the
</p>
<p>occasion demands. It has been convenient for this discussion, although not necessary, to consider
</p>
<p>Fig. 3.18 Geometry of
X-ray reflection. The path
</p>
<p>difference between the two
</p>
<p>typical rays reflected from
</p>
<p>successive planes is
</p>
<p>(QA2 + A2R). When this
</p>
<p>difference is equal to an
</p>
<p>integral number of
</p>
<p>wavelengths l, a reflection
</p>
<p>is obtained, according to
</p>
<p>the Bragg equation
</p>
<p>3.3 Scattering by Regular Arrays of Atoms 133</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec7_2">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec7_2</a></div>
</div>
<div class="page"><p/>
<p>that the scattering material is concentrated at lattice points, or on lattice planes. In general, electron
</p>
<p>density is a continuous function, albeit with a lattice-like distribution in the crystal.
</p>
<p>3.3.3 Equivalence of the Laue and Bragg Equations
</p>
<p>A way of demonstrating the equivalence between the Laue and Bragg treatments is illustrated by
</p>
<p>Fig. 3.19. Let p be the spacing between adjacent scattering points A and B in any row of a three-
</p>
<p>dimensional lattice. An X-ray beam makes the incident angle f with the row of points, and C is the
</p>
<p>angle between the diffracted ray and the same row. A &ldquo;reflecting&rdquo; plane must be in such a position as
</p>
<p>to make an angle y with both rays; the dashed line in the figure is the trace of such a plane. Following
</p>
<p>Sect. 3.3.1, we write
</p>
<p>p&eth;cosC� cosf&THORN; &frac14; nl (3.44)
</p>
<p>where m is an integer. Expanding (3.44), we obtain � 2p sin&eth;&eth;C&thorn; j&THORN;=2&THORN; sin&eth;&eth;C� j&THORN;=2&THORN; &frac14; nl.
It follows from the diagram that f� y &frac14; C&thorn; y so that &eth;f=2&THORN; &frac14; &eth;C=2&THORN; &thorn; y, whereupon
</p>
<p>p sin&eth;&eth;C&thorn; f&THORN;=2&THORN; &frac14; p sin&eth;C&thorn; y&THORN; &frac14; d and &eth;&eth;C� f&THORN;=2&THORN; &frac14; &eth;C=2&THORN; � &eth;&eth;C=2&THORN; &thorn; y&THORN; &frac14; �y. Hence,
</p>
<p>Table 3.1 Nomenclature for interplanar spacings
</p>
<p>Original Bragg notation Current usage
</p>
<p>hkl Order d (Å) hkl d (Å)
</p>
<p>120 1 2.236 120 2.236
2 2.236 240 1.118
3 2.236 360 0.745
4 2.236 480 0.559
</p>
<p>Fig. 3.19 Equivalence
of the Laue and Bragg
</p>
<p>treatments of X-ray
</p>
<p>diffraction
</p>
<p>134 3 X-Rays and X-Ray Diffraction</p>
<p/>
</div>
<div class="page"><p/>
<p>2d sin y &frac14; nl (3.45)
</p>
<p>which is the Bragg equation (3.42) derived above.
</p>
<p>3.3.4 Further Analysis of the Path Difference
</p>
<p>In (3.12) we expressed the resultant wave from n scattering centers, there thought of as electrons.
</p>
<p>Now, we can consider them as atoms each specified in scattering power by f(S).
The contribution to a wave scattered by the Ath atom in a unit cell is given by fA&eth;S&THORN; exp&eth;i2prA � S&THORN;,
</p>
<p>where the exponential term is the phase of the contribution from the Ath atom. Now the distance r
</p>
<p>from the origin to atom A is expressed through
</p>
<p>rA &frac14; xAa&thorn; yAb&thorn; zAc (3.46)
</p>
<p>where xA, yA, and zA are the fractional coordinates of atom A, as before. Since jSj &frac14; 2 sin y&eth;hkl&THORN;=l,
which from (3.43) is 1/d(hkl), or d*(hkl) (taking k as 1), S is the reciprocal lattice vector d*(hkl).
Applying (2.15) we have
</p>
<p>rA � S &frac14; &eth;xAa&thorn; yAb&thorn; zAc&THORN; � &eth;ha
� &thorn; kb� &thorn; lc�&THORN; &frac14; hxA &thorn; kyA &thorn; lzA (3.47)
</p>
<p>since a � a� &frac14; 1, and a � b� &frac14; 0, and similarly for b and c.
Thus, from (3.12), the phase angle fA for an atomic species A and its phase contribution are now
</p>
<p>given by 2prA � S, or
</p>
<p>fA &frac14; 2p&eth;hxA &thorn; kyA &thorn; lzA&THORN; (3.48)
</p>
<p>and the phase contribution is, therefore, exp&frac12;i2p&eth;hxA &thorn; kyA &thorn; lzA&THORN;�.
</p>
<p>3.4 Reciprocal Lattice: Analytical Treatment
</p>
<p>We considered a geometrical derivation of the reciprocal lattice in Sect. 2.4, as we believe that
</p>
<p>treatment forms a straightforward introduction to it. Here, we shall discuss the reciprocal lattice in
</p>
<p>greater detail.
</p>
<p>In considering the stereographic projection, we showed that the morphology of a crystal could be
</p>
<p>represented by a bundle of lines, drawn from a point, normal to the faces of the crystal. This description,
</p>
<p>although angle-true, lacks linear definition. The representationmay be extended by giving each normal a
</p>
<p>length that is inversely proportional to the corresponding interplanar spacing in real space, and applying
</p>
<p>it to all possible lattice planes, so forming a reciprocal lattice.
</p>
<p>Let a Bravais (real-space) lattice be represented by the unit cell vectors a, b, c. The reciprocal
lattice unit cell is defined by the vectors a*, b*, c*, such that a* is perpendicular to b and c, and so
on. Then,
</p>
<p>a� � b &frac14; a� � c &frac14; b� � a &frac14; b� � c &frac14; c� � a &frac14; c� � b &frac14; 0 (3.49)
</p>
<p>The magnitudes of the reciprocal unit cell vectors are defined by
</p>
<p>a� � a &frac14; b� � b &frac14; c� � c &frac14; k (3.50)
</p>
<p>3.4 Reciprocal Lattice: Analytical Treatment 135</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_2#Eq15_2">http://dx.doi.org/10.1007/978-1-4614-3954-7_2#Eq15_2</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec7_2">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec7_2</a></div>
</div>
<div class="page"><p/>
<p>where k is a constant, normally equal to unity in theoretical discussions, and to an X-ray wavelength
</p>
<p>in practical applications where the size of the reciprocal lattice is important.
</p>
<p>In Fig. 3.20, the z*(c*) axis is normal to the plane a, b. Since c � c� &frac14; cc� cos ffCOR, that is, taking
k &frac14; 1 in this discussion,
</p>
<p>c� &frac14; jc�j &frac14; 1=&eth;c cos ffCOR&THORN; (3.51)
</p>
<p>the magnitude of c* in reciprocal space is inversely proportional to the c-spacing in real, or Bravais,
space; similar deductions can be made for both a* and b*. Since c* is normal to both b and c, it lies in
the direction of their vector product:
</p>
<p>c� &frac14; �&eth;a� b&THORN; (3.52)
</p>
<p>where � is a constant. Let V be the unit cell volume in real space. Then,
</p>
<p>V &frac14; c � &eth;a� b&THORN; (3.53)
</p>
<p>Now a � b is a vector of magnitude ab sin g, the area ofOADB, directed normal to the plane of a, b
and forming a right-handed set of directions with a and b. Then,
</p>
<p>c � c� &frac14; �c � &eth;a� b&THORN; &frac14; �V &frac14; 1 (3.54)
</p>
<p>Hence,
</p>
<p>c� &frac14; jc�j &frac14; &eth;ab sin g&THORN;=V (3.55)
</p>
<p>with values for a* and b* obtained by cyclic permutation.
</p>
<p>The angle g* between a* and b* can be obtained by the equations of spherical trigonometry (seeWeb
Appendices WA2 and WA3). From the discussion therein, we derive
</p>
<p>Fig. 3.20 Triclinic unit
cell, showing its vectors
</p>
<p>a, b, and c, and the
corresponding reciprocal
</p>
<p>unit-cell vectors, a*, b*,
and c*
</p>
<p>136 3 X-Rays and X-Ray Diffraction</p>
<p/>
</div>
<div class="page"><p/>
<p>cos g� &frac14; &eth;cos a cos b� cos g&THORN;= sin a sinb (3.56)
</p>
<p>with corresponding expressions for a* and b* obtained by cyclic permutation. Simplified expressions
</p>
<p>obtain for (3.51)&ndash;(3.56) when the crystal symmetry is higher than triclinic.
</p>
<p>3.4.1 Reciprocal Lattice Properties
</p>
<p>In the Web Appendix WA6, we derive a number of useful properties of the reciprocal lattice, and we
</p>
<p>summarize the results here.
</p>
<p>Unit-Cell Volumes in Real and Reciprocal Space
The volume of the unit cell may be evaluated as follows. From (3.53), and expressing a, b, and c in
terms of a set of orthogonal unit vectors, we derive the equation for a unit-cell volume Vc in terms of
</p>
<p>its six constants, as shown in Web Appendix WA6; thus
</p>
<p>Vc &frac14; abc&eth;1� cos2 a� cos2 b� cos2 g&thorn; 2 cos a cos b cos g&THORN;1=2 (3.57)
</p>
<p>In the same Appendix, we show also that
</p>
<p>VcV
� &frac14; k (3.58)
</p>
<p>where V* is the volume of the reciprocal unit cell and k has the meaning as before.
</p>
<p>Interplanar Spacings
From (2.16), the general equation for d*(hkl) is obtained in terms of the reciprocal unit cell
</p>
<p>constants from
</p>
<p>jd�&eth;hkl&THORN;j2 &frac14; &eth;ha� &thorn; kb� &thorn; lc�&THORN; � &eth;ha� &thorn; kb� &thorn; lc�&THORN;
&frac14; h2a�2 &thorn; k2b�2 &thorn; l2c�2 &thorn; 2klb�c� cos a� &thorn; 2lhc�a� cos b� &thorn; 2hka�b� cos g� (3.59)
</p>
<p>and so
</p>
<p>sin2y&eth;hkl&THORN; &frac14; l2=&frac12;4d2&eth;hkl&THORN;� &frac14; l2d�2&eth;hkl&THORN;=4 (3.60)
</p>
<p>Simplifications of (3.59) arise in the presence of symmetry higher than triclinic and have been
</p>
<p>detailed adequately in Table 2.4.
</p>
<p>Angle Between Planes
Given any two planes h1k1l1 and h2k2l2, the angle between them can be found as the supplement of the
</p>
<p>angle between the two normals, d*(h1k1l1) and d
*(h2k2l2); this angle is the interfacial angle of
</p>
<p>the stereographic projection, Sect. 1.3.
</p>
<p>In general, the angle f between the forward directions of two vectors p and q is given through
</p>
<p>cosf &frac14; &eth;p � q&THORN;=pq (3.61)
</p>
<p>3.4 Reciprocal Lattice: Analytical Treatment 137</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_2#Eq16_2">http://dx.doi.org/10.1007/978-1-4614-3954-7_2#Eq16_2</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Tab4_2">http://dx.doi.org/10.1007/978-1-4614-3954-7#Tab4_2</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec8_1">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec8_1</a></div>
</div>
<div class="page"><p/>
<p>Applying (2.16) for the two planes h1k1l1 and h2k2l2, we find
</p>
<p>cosf &frac14; &frac12;h1h2a�2 &thorn; k1k2b�2 &thorn; l1l2c�2 &thorn; &eth;k1l2 &thorn; k2l1&THORN;b�c� cos a� &thorn; &eth;l1h2 &thorn; l2h1&THORN;c�a� cos b�
&thorn; &eth;h1k2 &thorn; h2k1&THORN;a�b� cos g��=&frac12;d�&eth;h1k1l1&THORN;d�&eth;h2k2l2&THORN;� (3.62)
</p>
<p>Reciprocity of F and I Unit Cells
We show in the Web Appendix WA6 how an F unit cell reciprocates into an I unit cell (and
</p>
<p>conversely), where the I unit cell is defined by the vectors 2a�F; 2b
�
F, and 2c
</p>
<p>�
F. If, as is customary in
</p>
<p>practice, we define the reciprocal of an I unit cell by vectors a�F; b
�
F, and c
</p>
<p>�
F, then only those reciprocal
</p>
<p>lattice points for which each of h + k, k + l (and l + h) is an even integer belong to the reciprocal of
</p>
<p>the I unit cell. In other words, Bragg reflections from an F unit cell have indices of the same parity;
</p>
<p>see also Sect. 3.7.1 and Table 3.2.
</p>
<p>3.4.2 Reciprocal Lattice and Reflection Condition: Ewald Sphere
</p>
<p>A section of a reciprocal lattice is shown in Fig. 3.21a: the vectors s0/l and s/l lie in the incident and
scattered X-ray beams respectively; a crystal is situated at the pointC. The vectorQP, which is also the
</p>
<p>direction of S, must be normal to an (hkl) plane and have the magnitude 2 sin y=l. When the conditions
for diffraction are satisfied, according to (3.43), S &frac14; jd*(hkl)j. A sphere of radius 1/l is described on
C as center, and passing through Q, the origin of the reciprocal lattice. If another reciprocal lattice
</p>
<p>point, such as P, lies on the sphere, then a diffracted beam arises and the vector CP is the direction of
</p>
<p>this beam; the incident beam vector is along CQ. The sphere is known as the sphere of reflection, or
</p>
<p>Ewald sphere [8], and will be required in subsequent chapters. We note that although we always refer
</p>
<p>to the &ldquo;Ewald&rdquo; sphere, the first suggestion for this device came from Bernal [9].
</p>
<p>Figure 3.21b is an alternative way of looking at the Ewald sphere construction, in this case taking
</p>
<p>the radius of the sphere as unity, so that reciprocal space has the units of length�1. The conclusions
</p>
<p>from the construction are, of course, the same, as we can see: from the figure AQ &frac14; 2 and
</p>
<p>ffAPQ &frac14; 90�, so that QP &frac14; AQ sin y&eth;hkl&THORN;, or 2 sin y&eth;hkl&THORN;. From (3.43), 2 sin y &eth;hkl&THORN; &frac14; l=d&eth;hkl&THORN;,
</p>
<p>and from Sect. 2.4, we may identify the point P with the reciprocal lattice point hkl. Hence, QP &frac14;
d*(hkl). Since k &frac14; l in this construction, d�&eth;hkl&THORN; &frac14; 2 sin y&eth;hkl&THORN;. Thus, the reflection from the (hkl)
</p>
<p>Table 3.2 Limiting conditions for centered unit cells
</p>
<p>Unit-cell-type Limiting conditions Associated translations
</p>
<p>Structure factor
</p>
<p>multiplier G
</p>
<p>P None None 1
</p>
<p>A hkl: k + l &frac14; 2n b/2 + c/2 2
</p>
<p>B hkl: l + h &frac14; 2n c/2 + a/2 2
</p>
<p>C hkl: h + k &frac14; 2n a/2 + b/2 2
</p>
<p>I hkl: h + k + l &frac14; 2n a/2 + b/2 + c/2 2
</p>
<p>F hkl: h + k &frac14; 2n a/2 + b/2 4
hkl: k + l &frac14; 2n b/2 + c/2
hkl: (l + h &frac14; 2n)a c/2 + a/2
</p>
<p>Rhex
b hkl: �h + k + l &frac14; 3nobv a/3 + 2b/3 + 2c/3 3
</p>
<p>2a/3 + b/3 + c/3
or
hkl: h � k + l &frac14; 3nrev a/3 + 2b/3 + c/3 3
</p>
<p>2a/3 + b/3 + 2c/3
</p>
<p>aThis condition is not independent of the other two
bSee Sect. 2.2.3 and Table 2.3
</p>
<p>138 3 X-Rays and X-Ray Diffraction</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_2#Eq16_2">http://dx.doi.org/10.1007/978-1-4614-3954-7_2#Eq16_2</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec7_2">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec7_2</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec5_2">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec5_2</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Tab3_2">http://dx.doi.org/10.1007/978-1-4614-3954-7#Tab3_2</a></div>
</div>
<div class="page"><p/>
<p>plane occurs when the hkl reciprocal lattice point lies on the sphere of reflection, and the direction of
</p>
<p>reflection is, again, that of CP.
</p>
<p>3.5 Scattering by a Crystal Structure
</p>
<p>In Sect. 2.2ff, we discussed the geometrical properties of the lattices on which crystal structures are
</p>
<p>based and showed how the arrays of atoms or molecules may be arranged according to space-group
</p>
<p>symmetry. We need next to consider the diffraction of X-rays by a crystal. We refer the process of
</p>
<p>scattering to a conventional unit cell and determine the resultant effect of all atoms in the unit cell,
</p>
<p>Fig. 3.21 (a) Sphere of
reflection, radius 1/l, with
</p>
<p>the crystal at C, the center.
</p>
<p>The origin of the reciprocal
</p>
<p>lattice is at Q. When a
</p>
<p>reciprocal lattice point,
</p>
<p>such as P, lies on the Ewald
</p>
<p>sphere, a reflection arises
</p>
<p>along the direction CP.
</p>
<p>(b) Alternative, equivalent
picture, but with the sphere
</p>
<p>of radius 1 (dimensionless),
</p>
<p>showing the position of
</p>
<p>crystal planes. By
</p>
<p>geometry, QP is parallel to
</p>
<p>the normal d*(hkl) to the
</p>
<p>crystal planes and is now
</p>
<p>2 sin y&eth;hkl&THORN;
</p>
<p>3.5 Scattering by a Crystal Structure 139</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec2_2">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec2_2</a></div>
</div>
<div class="page"><p/>
<p>which leads directly to the equation for the structure factor, F. The scattering of X-rays by a crystal
</p>
<p>comprises two parts: a geometrical part that depends on the symmetry of the arrangement of its
</p>
<p>components, and a structural part that depends upon both the nature of the atoms or molecules
</p>
<p>comprising those entities and their relative positions in the unit cell.
</p>
<p>3.5.1 Structure Factor Equation
</p>
<p>We need to express the equation that has been deduced for F in a manner that includes the coordinates
</p>
<p>of the atoms in the unit cell. It follows from (3.46)&ndash;(3.48) that the phase for the jth atom, with respect
</p>
<p>to the origin, can be represented by 2p(hxj + kyj + lzj). Hence, we use this expression in (3.14) to give
</p>
<p>the structure factor equation for the hkl reflection:
</p>
<p>F(hkl&THORN; &frac14;
Xn
</p>
<p>j&frac14;1
fj;y exp&frac12;i2p&eth;hxj &thorn; kyj &thorn; lzj&THORN;� (3.63)
</p>
<p>The structure factor F(hkl) is a dimensionless scalar quantity and refers to the combined scattering
</p>
<p>from the n atoms in the unit cell to give the hkl spectrum, or equally to the wave from the (hkl) family
</p>
<p>of planes, relative to the scattering by a single electron at the origin; see also Sect. 3.2.3. The atomic
</p>
<p>scattering factor fj,y for the jth atom indicates its dependence on y, an alternative parameter to S in this
context. Again, frequently the y dependence of f is not expressed in the formula, although it is
</p>
<p>implicitly always present. The atomic coordinates in (3.63) are fractional values, Sect. 2.2.3, and so
</p>
<p>are independent of the size of the unit cell.
</p>
<p>3.6 Using the Structure Factor Equation
</p>
<p>In the next two sections, we explore some of the properties and applications of the structure factor
</p>
<p>equation that are encountered in practical X-ray crystallography. The trigonometrical relations in
</p>
<p>Web Appendix WA5 may be helpful in some of the ensuing arguments.
</p>
<p>3.6.1 Friedel&rsquo;s Law
</p>
<p>Except where anomalous scattering is significant, Sect. 7.6, X-ray diffraction spectra form a centro-
</p>
<p>symmetric array. The diffraction spectra from a crystal may be thought of as an expression
</p>
<p>of its reciprocal lattice, with each spectrum hkl weighted by the corresponding value of the amplitude
</p>
<p>jF(hkl)j, or the intensity I(hkl). Friedel&rsquo;s law expresses the centrosymmetric property as
</p>
<p>I&eth;hkl&THORN; &frac14; I&eth;�h �k �l&THORN; or jF&eth;hkl&THORN;j &frac14; jF&eth;h k l&THORN;j (3.64)
</p>
<p>within the limits of experimental error, and may be derived as follows.
</p>
<p>Since the atomic scattering factor is a function of (sin y)/l, it will have the same value for both the
</p>
<p>hkl and h k l reflections. Thus, fy &frac14; f�y, because reflections from opposite sides of any plane occur at
the same value of the Bragg angle y.
</p>
<p>140 3 X-Rays and X-Ray Diffraction</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec5_2">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec5_2</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec31_7">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec31_7</a></div>
</div>
<div class="page"><p/>
<p>From (3.63)
</p>
<p>F&eth;�h �k �l&THORN; &frac14;
Xn
</p>
<p>j&frac14;1
fj exp&frac12;�i2p&eth;hxj &thorn; kyj &thorn; lzj&THORN;� (3.65)
</p>
<p>and from Fig. 3.13
</p>
<p>F&eth;hkl&THORN; &frac14; A0&eth;hkl&THORN; &thorn; iB0&eth;hkl&THORN;F&eth;�h �k �l&THORN; &frac14; A0&eth;�h �k �l&THORN; &thorn; iB0&eth;�h �k �l&THORN; &frac14; A0&eth;hkl&THORN; � iB0&eth;hkl&THORN; (3.66)
</p>
<p>Hence, the following relations hold:
</p>
<p>f&eth;hkl&THORN; &frac14; �f&eth;�h �k �l&THORN;jF&eth;hkl&THORN;j &frac14; jF&eth;�h �k �l&THORN;j &frac14; &frac12;A0
2
&eth;hkl&THORN; &thorn; B0
</p>
<p>2
&eth;hkl&THORN;�1=2 (3.67)
</p>
<p>and, since I &frac14; jFj2
</p>
<p>I&eth;hkl&THORN; &frac14; I&eth;�h �k �l&THORN; (3.68)
</p>
<p>which is Friedel&rsquo;s law.
</p>
<p>3.6.2 Structure Factor for a Centrosymmetric Crystal
</p>
<p>One of the questions that frequently arises at the outset of a crystal structure determination is whether
</p>
<p>or no the space group is centrosymmetric. In a centrosymmetric structure, with the origin on a center
</p>
<p>of symmetry, the n atoms in the unit cell lie in related pairs, with coordinates �(x, y, z). From (3.18)
</p>
<p>and (3.19), we write for the two parts of the structure factor equation:
</p>
<p>A0&eth;hkl&THORN; &frac14;
Xn=2
</p>
<p>j&frac14;1
</p>
<p>fj&frac12;cos 2p&eth;hxj &thorn; kyj &thorn; lzj&THORN; &thorn; cos 2p&eth;�hxj � kyj � lzj&THORN;�
</p>
<p>&frac14; 2
Xn=2
</p>
<p>j&frac14;1
</p>
<p>fj cos 2p&eth;hxj &thorn; kyj &thorn; lzj&THORN; (3.69)
</p>
<p>where j ranges over the n/2 atoms in the unit cell not related by the center of symmetry.
</p>
<p>For B0(hkl), we write
</p>
<p>B0&eth;hkl&THORN; &frac14;
Xn=2
</p>
<p>j&frac14;1
</p>
<p>fj&frac12;sin 2p&eth;hxs &thorn; kys &thorn; lzs&THORN; &thorn; sin 2p&eth;�hxs � kys � lzs&THORN;� &frac14; 0 (3.70)
</p>
<p>Here, B0 &frac14; 0 because sin(�f) &frac14; �sin(f) for all f, in accord with (3.20). In this case, A0(hkl) &frac14;
</p>
<p>F(hkl), and f(hkl) can take only the values 0 or p, so that the phase angle attaches itself to jF(hkl)j as a
</p>
<p>positive or negative sign. Hence, we often speak of the signs, s, of reflections in centrosymmetric
</p>
<p>crystals, so that F(hkl) &frac14; s(hkl) jF(hkl)j. Clearly, these results apply only when the origin of the unit
</p>
<p>cell is taken on �1; in any other setting of the origin in a centrosymmetric space group, there will
</p>
<p>normally be a non-zero component in B0(hkl). Centrosymmetric crystals usually present fewer
</p>
<p>3.6 Using the Structure Factor Equation 141</p>
<p/>
</div>
<div class="page"><p/>
<p>difficulties to the structure analyst than do non-centrosymmetric crystals because of the above
</p>
<p>restriction on the phase angles.
</p>
<p>3.7 Limiting Conditions and Systematic Absences
</p>
<p>We considered limiting conditions briefly in Sect. 2.7.1; here we investigate them more fully through
</p>
<p>the structure factor equation and show how they are handled in several different example symmetries.
</p>
<p>An X-ray diffraction pattern can be used to determine the type of unit cell that corresponds to the
</p>
<p>chosen system of reference axes. From (3.63), it would be an unexpected coincidence for many
</p>
<p>intensities to be zero. With unit cells having no translational symmetry, the intensity of a reflection is
</p>
<p>not usually zero for any particular combinations of h, k, and l, that is, no limiting conditions apply in
</p>
<p>such a case. Vanishingly weak intensities may arise for certain reflections because of the particular
</p>
<p>structure under investigation; we call these reflections accidental absences, and we shall discuss them
</p>
<p>further, in Sect. 4.2.3. In centered unit cells or in the presence of other translational symmetry, glide
</p>
<p>planes and screw axes, reflections of certain combinations of h, k, and l are totally absent; we call such
</p>
<p>unobservable reflections systematic absences.
</p>
<p>3.7.1 Body-Centered Unit Cell
</p>
<p>As a first example, we know that in a body-centered (I) unit cell, the atoms are related in pairs as x, y, z
</p>
<p>and 1
2
&thorn; x; 1
</p>
<p>2
&thorn; y; 1
</p>
<p>2
&thorn; z. Using (3.63), we have
</p>
<p>F&eth;hkl&THORN; &frac14;
Xn=2
</p>
<p>j&frac14;1
fj exp&frac12;i2p&eth;hxj &thorn; kyj &thorn; lzj&THORN;� &thorn; exp&frac12;i2p&eth;hxj &thorn; kyj &thorn; lzj &thorn; h=2&thorn; k=2&thorn; l=2&THORN;�
� �
</p>
<p>(3.71)
</p>
<p>The term within the braces {. . .} may be expressed as exp&frac12;i2p&eth;hxj &thorn; kyj &thorn; lzj&THORN;�f1&thorn; exp&frac12;i2
</p>
<p>p&eth;h&thorn; k &thorn; l&THORN;=2�g. Since h + k + l is integral, f1&thorn; exp&frac12;i2p&eth;h&thorn; k &thorn; l&THORN;=2�g &frac14; 1&thorn; cos&frac12;2p&eth;h&thorn; k &thorn; l&THORN;
</p>
<p>=2� &frac14; 2 cos2&frac12;2p&eth;h&thorn; k &thorn; l&THORN;=4� &frac14; G, where G is a multiplying factor for the reduced structure factor
</p>
<p>equation in centered unit cells; in the body-centered unit cellG &frac14; 2, so that
</p>
<p>F&eth;hkl&THORN; &frac14; 2cos2&frac12;2p&eth;h&thorn; k &thorn; l&THORN;=4�
X
</p>
<p>n=2
</p>
<p>j&frac14;1
</p>
<p>fj exp&frac12;i2p&eth;hxj &thorn; kyj &thorn; lzj&THORN;� (3.72)
</p>
<p>This equation may be broken down into its two components, A0(hkl) and B0(hkl), in the usual way.
</p>
<p>Further simplification is possible: in this example, G takes the value 2 if h + k + l is even, and 0 if
</p>
<p>h + k + l is odd. Hence, we write the limiting condition that shows which reflections are permitted by
</p>
<p>the geometry of an I unit cell as
</p>
<p>hkl : h&thorn; k &thorn; l &frac14; 2n; n &frac14; 0;�1;�2; . . .
</p>
<p>The same situation expressed as systematic absences, the condition under which reflections are
</p>
<p>forbidden by the space-group geometry, is
</p>
<p>hkl : h&thorn; k &thorn; l &frac14; 2n&thorn; 1; n &frac14; 0;�1;�2; . . .
</p>
<p>142 3 X-Rays and X-Ray Diffraction</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec16_2">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec16_2</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec13_4">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec13_4</a></div>
</div>
<div class="page"><p/>
<p>Both terms are in common use, and the reader should distinguish between them. We could have
</p>
<p>reached the same conclusion from (3.71) more speedily: the expression {1 + exp[i2p(h + k + l)/2]} is
</p>
<p>equal to 1 + exp(ipn), where n is the sum of the integers h, k, and l, and [1 + exp(ipn)] is 2 or 0 for n
</p>
<p>even or odd. But it is preferable to highlight the dependence on h, k, and l.
</p>
<p>Analogous expressions can be derived for any centered unit cell. The G factors for all types of
</p>
<p>centering have been summarized in Table 3.2. It is evident that where a reflection arises in a centered
</p>
<p>unit cell, the structure factor equation has the same form as that for the corresponding primitive unit
</p>
<p>cell, but multiplied by the G factor appropriate to the unit cell type. The summation in the reduced
</p>
<p>structure factor equation is then taken over that fraction of atoms not related by the centering
</p>
<p>symmetry.
</p>
<p>In practice, the diffraction pattern is recorded, indices allocated to the spectra and then scrutinized
</p>
<p>for systematic absences, so as to determine the unit cell type. The reader may care to work through the
</p>
<p>derivations of F(hkl) for, say, a C and an F unit cell, and determine the limiting conditions for each
</p>
<p>unit cell type; see also Sect. 3.8.
</p>
<p>3.7.2 Screw Axes and Glide Planes
</p>
<p>As we are concerned in this discussion with the geometry of the unit cell rather than the chemical
</p>
<p>nature of its contents, it is convenient to introduce the following nomenclature. Let N be the total
</p>
<p>number of atoms in the unit cell, and let n of them be the number in the asymmetric unit, with the
</p>
<p>number of asymmetric units being m, so that N &frac14; nm. Symbolically, we may write
</p>
<p>XN
</p>
<p>j&frac14;1


Xn
</p>
<p>r&frac14;1
</p>
<p>Xm
</p>
<p>s&frac14;1
</p>
<p>where the sum over r refers to the symmetry-independent atoms, and that over s to the symmetry-
</p>
<p>related species. Thus, the structure factor equation contains two parts that may be considered
</p>
<p>separately. The sum over m symmetry-related atoms is expressed through the coordinates of a set
</p>
<p>of general equivalent positions. Thus,
</p>
<p>Ar&eth;hkl&THORN; &frac14;
Xm
</p>
<p>s&frac14;1
cos 2p&eth;hxs &thorn; kys &thorn; lzs&THORN;
</p>
<p>Br&eth;hkl&THORN; &frac14;
Xm
</p>
<p>s&frac14;1
sin 2p&eth;hxs &thorn; kys &thorn; lzs&THORN;
</p>
<p>(3.73)
</p>
<p>Extending to the n atoms in the asymmetric unit, with one such term for each atom,
</p>
<p>A0&eth;hkl&THORN; &frac14;
Xn
</p>
<p>r&frac14;1
frAr&eth;hkl&THORN;
</p>
<p>B0&eth;hkl&THORN; &frac14;
Xn
</p>
<p>r&frac14;1
frBr&eth;hkl&THORN;
</p>
<p>(3.74)
</p>
<p>The terms Ar(hkl) and Br(hkl) are independent of the nature and arrangement of the atoms in the
</p>
<p>asymmetric unit; they are a property of the space-group symmetry and are called geometrical
</p>
<p>3.7 Limiting Conditions and Systematic Absences 143</p>
<p/>
</div>
<div class="page"><p/>
<p>structure factors. We shall consider some examples taken from the monoclinic and orthorhombic
</p>
<p>systems, in order to show how glide-plane and screw-axis symmetries give rise to limiting conditions
</p>
<p>with special classes of reflections. For this discussion, the subscript s in (3.73) need not be retained,
</p>
<p>because allm positions are related to the position x, y, z by symmetry, and in the geometrical structure
</p>
<p>factors we shall, for simplicity, drop the subscript r.
</p>
<p>Space Group P21
General equivalent positions: x; y; z; �x; 1
</p>
<p>2
&thorn; y; �z, see Fig. 2.30.
</p>
<p>Geometrical structure factors:
</p>
<p>A&eth;hkl&THORN; &frac14; cos 2p&eth;hx&thorn; ky&thorn; lz&THORN; &thorn; cos 2p&eth;�hx&thorn; ky� lz&thorn; k=2&THORN;
&frac14; 2 cos 2p&eth;hx&thorn; lz� k=4&THORN; cos 2p&eth;ky&thorn; k=4&THORN; (3.75)
</p>
<p>In a similar way,
</p>
<p>B&eth;hkl&THORN; &frac14; sin 2p&eth;hx&thorn; ky&thorn; lz&THORN; &thorn; sin 2p&eth;�hx&thorn; ky� lz&thorn; k=2&THORN;
&frac14; 2 cos 2p&eth;hx&thorn; lz� k=4&THORN; sin 2p&eth;ky&thorn; k=4&THORN; (3.76)
</p>
<p>Limiting Conditions in P21
Geometrical structure factors enable us to determine limiting conditions, that is, to predict which
</p>
<p>classes of reflections are capable of arising in an X-ray diffraction pattern. If we can show, for given
</p>
<p>values of h, k, and l, that both A(hkl) and B(hkl) are systematically zero, then F(hkl) will be zero,
</p>
<p>regardless of the atomic positions.
</p>
<p>For P21, we can cast (3.75) and (3.76) in the following forms, according to the parity (evenness or
</p>
<p>oddness) of k. Expanding (3.75), we have (see Web Appendix WA5):
</p>
<p>A&eth;hkl&THORN;=2 &frac14; &frac12;cos 2p&eth;hx&thorn; lz&THORN; &thorn; cos 2p&eth;k=4&THORN; &thorn; sin 2p&eth;hx&thorn; lz&THORN; sin 2p&eth;k=4&THORN;�
� &frac12;cos 2p&eth;ky&THORN; cos 2p&eth;k=4&THORN; � sin 2p&eth;ky&THORN; sin 2p&eth;k=4&THORN;� (3.77)
</p>
<p>In expanding the right-hand side of (3.77), terms such as
</p>
<p>cos 2p&eth;hx&thorn; lz&THORN; cos 2p&eth;k=4&THORN; sin 2p&eth;ky&THORN; sin 2p&eth;k=4&THORN;
</p>
<p>occur. This particular term is equivalent to
</p>
<p>1
</p>
<p>2
cos 2p&eth;hx&thorn; lz&THORN; sin 2p&eth;ky&THORN; sin 4p&eth;k=4&THORN;
</p>
<p>which is zero, because k is an integer. Hence, (3.77) becomes
</p>
<p>A&eth;hkl&THORN;=2 &frac14; &frac12;cos 2p&eth;hx&thorn; lz&THORN; cos 2p&eth;ky&THORN;cos22p&eth;k=4&THORN;� � &frac12;sin 2p&eth;hx&thorn; lz&THORN;
</p>
<p>� sin 2p&eth;ky&THORN; sin22p&eth;k=4&THORN;� (3.78)
</p>
<p>In a similar manner, we find from (3.76)
</p>
<p>B&eth;hkl&THORN;=2 &frac14; &frac12;cos 2p&eth;hx&thorn; lz&THORN; sin 2p&eth;ky&THORN; cos22p&eth;k=4&THORN;� &thorn; &frac12;sin 2p&eth;hx&thorn; lz&THORN;
</p>
<p>� cos 2p&eth;ky&THORN; sin22p&eth;k=4&THORN;� (3.79)
</p>
<p>144 3 X-Rays and X-Ray Diffraction</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Fig30_2">http://dx.doi.org/10.1007/978-1-4614-3954-7#Fig30_2</a></div>
</div>
<div class="page"><p/>
<p>Separating for k even and k odd, we obtain
</p>
<p>k &frac14; 2n:A&eth;hkl&THORN; &frac14; 2 cos 2p&eth;hx&thorn; lz&THORN; cos 2p&eth;ky&THORN; (3.80)
</p>
<p>B&eth;hkl&THORN; &frac14; 2 cos 2p&eth;hx&thorn; lz&THORN; sin 2p&eth;ky&THORN; (3.81)
</p>
<p>k &frac14; 2n&thorn; 1:A&eth;hkl&THORN; &frac14; �2 sin 2p&eth;hx&thorn; lz&THORN; sin 2p&eth;ky&THORN; (3.82)
</p>
<p>B&eth;hkl&THORN; &frac14; 2 sin 2p&eth;hx&thorn; lz&THORN; cos 2p&eth;ky&THORN; (3.83)
</p>
<p>Only one systematic condition can be extracted from these equations: if both h and l are zero, then
</p>
<p>from (3.82) and (3.83).
</p>
<p>A&eth;hkl&THORN; &frac14; B&eth;hkl&THORN; &frac14; 0
</p>
<p>In other words, the limiting conditions associated with a 21 axis are
</p>
<p>hkl None (P unit cell)
</p>
<p>0k0 k &frac14; 2n
</p>
<p>The example of the 21 axis has been treated in detail; it shows again how a diffraction record may
</p>
<p>be used to reveal information about the translational symmetry elements of a space group. We can
</p>
<p>show how the limiting conditions for a 21 axis arise from a consideration of the Bragg equation.
</p>
<p>Figure 3.22 is a schematic illustration of a 21 symmetry pattern; the motif represents a structure
</p>
<p>at a height z, and the structure at a height �z after operating on it with the 21 axis. The planes
</p>
<p>MM0 represent the family (0k0) and NN0 the family (02k,0).
Reflections of the type (0k0) fromMM0 planes are canceled by the reflections from the NN0 planes,
</p>
<p>because their phase change relative to MM0 is 180�. Clearly, this result is not obtained with the
02k,0 reflections. Although the figure illustrates the situation for k &frac14; 1, the same argument can be
applied to any pair of values k and 2k, where k is an odd integer. Limiting conditions for other screw
</p>
<p>axes, and in other orientations, can be deduced as above, and the results are summarized in Table 3.3.
</p>
<p>Notice that pure rotation axes, as in space group P2, do not introduce any limiting conditions.
</p>
<p>Fig. 3.22 Pattern of a
structure containing a 21
screw axis: d(NN0) &frac14; d
(MM0)/2, so that the MM0
</p>
<p>planes are halved by the
</p>
<p>NN0 family
</p>
<p>3.7 Limiting Conditions and Systematic Absences 145</p>
<p/>
</div>
<div class="page"><p/>
<p>Centric Zones
Centric zones, sometimes loosely termed centrosymmetric zones, are of particular importance in
</p>
<p>crystal structure determination; see also Sect. 4.2.3. In space group P21 and other space groups of
</p>
<p>crystal class 2, the h0l reflections are of special interest. Among (3.80)&mdash;(3.83), only (3.80) is relevant
</p>
<p>here because zero behaves as an even number, and sin(2pky)k&frac14;0 &frac14; 0. Hence,
</p>
<p>A&eth;h0l&THORN; &frac14; 2 cos 2p&eth;hx&thorn; lz&THORN; B&eth;h0l&THORN; &frac14; 0 (3.84)
</p>
<p>From (3.20), f(h0l) is either 0 or p; in other words, the [010] zone is centric for this space group.
</p>
<p>Centric zones occur in the non-centrosymmetric space groups that have symmetry 2 as a subgroup of
</p>
<p>their point groups; see Sects. 1.4.2 and 2.7.3ff.
</p>
<p>Space Group Pc
General equivalent positions: x, y, z; x; �y; 1
</p>
<p>2
&thorn; z.
</p>
<p>Geometrical structure factors: Proceeding as before, we obtain
</p>
<p>A&eth;hkl&THORN; &frac14; 2 cos 2p&eth;hx&thorn; lz&thorn; l=4&THORN; cos 2p&eth;ky� k=4&THORN;
B&eth;hkl&THORN; &frac14; 2 sin 2p&eth;hx&thorn; lz&thorn; l=4&THORN; cos 2p&eth;ky� l=4&THORN;
</p>
<p>(3.85)
</p>
<p>If we expand these equations, as with the example for P21, and then separate the terms for l even
</p>
<p>and l odd, we shall find systematic absences only for the h0l reflections. Thus, the limiting
</p>
<p>conditions for Pc are
</p>
<p>hkl None
</p>
<p>h0l l &frac14; 2n
</p>
<p>The relationship between the index (l) involved in the condition and the symmetry translation (c/2)
</p>
<p>is clear. A space group diagram for Pc is shown in Fig. 3.23.
</p>
<p>Space Group P21/c
This space group contains the two translational symmetry operations already discussed, namely, a 21
axis parallel to y and a c-glide normal to y, Fig. 2.32 and Problem 2.8a. It is a centrosymmetric space
</p>
<p>group, and the general equivalent positions may be summarized as
</p>
<p>Table 3.3 Limiting conditions for screw axes
</p>
<p>Screw axis Orientation Limiting condition Translational component
</p>
<p>21 k x h00: h &frac14; 2n a/2
21 k y 0k0: k &frac14; 2n b/2
21 k z 00l: l &frac14; 2n c/2
31 or 32 k z 000l: l &frac14; 3n c/3 or 2c/3
41 or 43 k z 00l: l &frac14; 4n c/4 or 3c/4
42 k z 00l: l &frac14; 2n 2c/4 (c/2)
61 or 65 k z 000l: l &frac14; 6n c/6 or 5c/6
62 or 64 k z 000l: l &frac14; 3n 2c/6 (c/3), c/6(2c/3)
63 k z 000l: l &frac14; 2n 3c/6 (c/2)
</p>
<p>In the cubic system, 41, 43, and 42 axes parallel to x and y exhibit limiting conditions similar to those parallel to z by
</p>
<p>cyclic permutation
</p>
<p>146 3 X-Rays and X-Ray Diffraction</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec13_4">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec13_4</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7# Sec11_1">http://dx.doi.org/10.1007/978-1-4614-3954-7# Sec11_1</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec18_2">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec18_2</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Fig32_2">http://dx.doi.org/10.1007/978-1-4614-3954-7#Fig32_2</a></div>
</div>
<div class="page"><p/>
<p>� fx; y; z; x; 1
2
� y; 1
</p>
<p>2
&thorn; zg
</p>
<p>Geometrical structure factors: In the standard setting of this space group, the origin is on �1, so that we
</p>
<p>can immediately apply (3.69) and (3.70) and write
</p>
<p>A&eth;hkl&THORN; &frac14; 2fcos 2p&frac12;hx&thorn; ky&thorn; lz� &thorn; cos 2p&frac12;hx� ky&thorn; lz&thorn; &eth;k &thorn; l&THORN;=2�gB&eth;hkl&THORN; &frac14; 0 (3.86)
</p>
<p>Combining the two cosine terms
</p>
<p>A&eth;hkl&THORN; &frac14; 4 cos 2p&frac12;hx&thorn; lz&thorn; &eth;k &thorn; l&THORN;=4� cos 2p&frac12;ky� &eth;k &thorn; l&THORN;=4&THORN;�
</p>
<p>Separating for k + l even and odd, we obtain
</p>
<p>k &thorn; l &frac14; 2n; A&eth;hkl&THORN; &frac14; 4 cos 2p&eth;hx&thorn; lz&THORN; cos 2p&eth;ky&THORN;
</p>
<p>k &thorn; l &frac14; 2n&thorn; 1; A&eth;hkl&THORN; &frac14; �4 sin 2p&eth;hx&thorn; lz&THORN; sin 2p&eth;ky&THORN;
(3.87)
</p>
<p>We now deduce the limiting conditions as
</p>
<p>hkl None
</p>
<p>h0l l &frac14; 2n (c-glide normal to y)
</p>
<p>0k0 k &frac14; 2n (21 axis k y)
</p>
<p>These three classes of reflections are important in monoclinic reciprocal space, because only with
</p>
<p>them can we determine the characteristic systematic absences in the space groups within this system.
</p>
<p>Despite Friedel&rsquo;s law, the diffraction symmetry reveals the true space group in this example.
</p>
<p>Figure 3.24 illustrates weighted reciprocal space levels for a monoclinic crystal of space group Pc,
</p>
<p>P2/c, or P21/c.
</p>
<p>Fig. 3.23 Space group Pc. (a) Viewed along c. (b) Viewed along b
</p>
<p>3.7 Limiting Conditions and Systematic Absences 147</p>
<p/>
</div>
<div class="page"><p/>
<p>Space Group Pma2
From the data in Fig. 3.25 we can write down expressions for the geometrical structure factors:
</p>
<p>A&eth;hkl&THORN; &frac14; cos 2p&eth;hx&thorn; ky&thorn; lz&THORN; &thorn; cos 2p&eth;�hx� ky&thorn; lz&THORN; &thorn; cos 2p&eth;�hx&thorn; ky&thorn; lz&thorn; h=2&THORN;
&thorn; cos 2p&eth;hx� ky&thorn; lz&thorn; h=2&THORN; (3.88)
</p>
<p>Combining the first and third, and second and fourth terms, we have
</p>
<p>A&eth;hkl&THORN; &frac14; 2 cos 2p&eth;ky&thorn; lz&thorn; h=4&THORN; &thorn; cos 2p&eth;hx� h=4&THORN; &thorn; 2 cos 2p&eth;�ky&thorn; lz&thorn; h=4&THORN;
� cos 2p&eth;hx&thorn; h=4&THORN; (3.89)
</p>
<p>Further simplification of this expression requires the separate parts to contain a common factor.
</p>
<p>We return to (3.88) and make a minor alteration to the term cos 2p(hx � ky + lz + h/2). Since h is an
integer,wemaywrite this termas the crystallographically equivalent term cos 2p(hx � ky + lz � h/2).
Another way of looking at this process is that the fourth general equivalent position has been changed to
</p>
<p>� 1
2
&thorn; x; �y; z, which is equivalent to moving through one repeat a in the negative direction to a
</p>
<p>crystallographically equivalent position, a perfectly valid and generally applicable tactic.
</p>
<p>Returning to Pma2, (3.89) now becomes
</p>
<p>A&eth;hkl&THORN; &frac14; 2 cos 2p&eth;ky&thorn; lz&thorn; h=4&THORN; cos 2p&eth;hx� h=4&THORN;
&thorn; 2 cos 2p&eth;�ky&thorn; lz� h=4&THORN; cos 2p&eth;hx� h=4&THORN; (3.90)
</p>
<p>which simplifies to
</p>
<p>A&eth;hkl&THORN; &frac14; &frac12;2 cos 2p&eth;hx� h=4&THORN;�&frac12;cos 2p&eth;ky&thorn; lz&thorn; h=4&THORN; &thorn; cos 2p&eth;�ky&thorn; lz� h=4&THORN;� (3.91)
</p>
<p>Fig. 3.24 Reciprocal nets in the x*, z* plane appropriate to space groups Pc, P2/c, and P21/c. (a) k &frac14; 0 and (b) k &gt; 0.
The c-glide plane, which is perpendicular to b, causes a halving of the rows parallel to x* when k &frac14; 0, so that only the
rows with l &frac14; 2n are present. Hence, the true c* spacing is not observed on the reciprocal lattice level k &frac14; 0, but can be
determined from higher levels. The symmetry on both levels is 2, in accordance with the diffraction symmetry 2/m:
</p>
<p>jF&eth;hkl&THORN;j &frac14; jF&eth;h k l&THORN;j. The reciprocal lattice points are weighted according to jF(hkl)j, or to jF(hkl)j2, and we speak of
such diffraction patterns as weighted reciprocal lattices
</p>
<p>148 3 X-Rays and X-Ray Diffraction</p>
<p/>
</div>
<div class="page"><p/>
<p>Combining again:
</p>
<p>A&eth;hkl&THORN; &frac14; 4&frac12;cos 2p&eth;hx� h=4&THORN;� cos 2p&eth;ky&thorn; h=4&THORN; cos 2plz (3.92)
</p>
<p>Similarly,
</p>
<p>B&eth;hkl&THORN; &frac14; 4&frac12;cos 2p&eth;hx� h=4&THORN;� cos 2p&eth;ky&thorn; h=4&THORN; sin 2plz (3.93)
</p>
<p>In the orthorhombic system, seven regions of reciprocal space of particular importance are listed
</p>
<p>on the right-hand side of Fig. 3.25. Separating (3.92) and (3.93) for even and odd values of h, we
</p>
<p>obtain
</p>
<p>h &frac14; 2n : A&eth;hkl&THORN; &frac14; 4 cos 2phx cos 2pky cos 2plz
B&eth;hkl&THORN; &frac14; 2 cos 2phx cos 2pky sin 2plz
</p>
<p>(3.94)
</p>
<p>h &frac14; 2n&thorn; 1 : A&eth;hkl&THORN; &frac14; �4 sin 2phx sin 2pky cos 2plz
B&eth;hkl&THORN; &frac14; �4 sin 2phx sin 2pky sin 2plz
</p>
<p>(3.95)
</p>
<p>from which we find the limiting conditions
</p>
<p>hkl None
</p>
<p>h0l h &frac14; 2n
</p>
<p>Fig. 3.25 General equivalent positions and symmetry elements in space group Pma2, with the origin on 2; the origin is
not fixed in the z direction by the symmetry elements. The diagram shows inter alia the coordinates of the special
</p>
<p>equivalent positions and the limiting conditions
</p>
<p>3.7 Limiting Conditions and Systematic Absences 149</p>
<p/>
</div>
<div class="page"><p/>
<p>The listed condition h00: (h &frac14; 2n) should be considered carefully. One might be excused for
thinking at first that it implies the existence of a 21 axis parallel to the x axis, but for the knowledge
</p>
<p>that there are no symmetry axes parallel to the x axis in class mm2. This particular limiting condition
</p>
<p>is dependent upon the h0l condition: h00 is in the h0l zone.
</p>
<p>We emphasize here that confusion can very easily arise if the limiting conditions are interpreted in
</p>
<p>other than the following hierarchal order:
</p>
<p>Order of inspection Downward
</p>
<p>hkl Unit cell type #0kl Glide plane&perp;xh0l Glide plane&perp;yhk0 Glide plane &perp;zh00 21 axis jj x
0k0 21 axis jj y
00l 21 axis jj z
</p>
<p>One should proceed to a lower level in this list only after considering the full implications of the
</p>
<p>conditions at higher levels. Conditions such as that for h00 in Pma2 are called redundant or dependent
</p>
<p>and are placed in parentheses on the diagram. Reflections involved in such conditions are certainly
</p>
<p>absent from a diffraction record, but do not contribute to the determination of space-group symmetry.
</p>
<p>Table 3.4 summarizes the limiting conditions for glide-plane symmetry.
</p>
<p>Space Group Pman
This space group may be derived from Pma2 by the addition of an n-glide plane perpendicular to
</p>
<p>the z axis, with a translational component of (a + b)/2. We have now seen on several occasions that
</p>
<p>it is advantageous to set the origin at �1 wherever possible; Fig. 3.26 shows Pman drawn in this
</p>
<p>orientation. It is left to the reader to show that the geometrical structure factors are
</p>
<p>A&eth;hkl&THORN; &frac14; 8 cos 2phx cos 2p&frac12;ky� &eth;h&thorn; k&THORN;=4� cos 2p&frac12;lz&thorn; &eth;h&thorn; k&THORN;=4�
B&eth;hkl&THORN; &frac14; 0
</p>
<p>(3.96)
</p>
<p>and subsequently to derive the limiting conditions for this space group.
</p>
<p>Table 3.4 Limiting conditions for glide planes
</p>
<p>Glide plane Orientation Limiting condition Translational component
</p>
<p>a &perp;b h0l: h &frac14; 2n a/2
</p>
<p>a &perp;c hk0: h &frac14; 2n a/2
</p>
<p>b &perp;a 0kl: k &frac14; 2n b/2
</p>
<p>b &perp;c hk0: k &frac14; 2n b/2
</p>
<p>c &perp;a 0kl: l &frac14; 2n c/2
</p>
<p>c &perp;b h0l: l &frac14; 2n c/2
</p>
<p>n &perp;a 0kl: k + l &frac14; 2n (b + c)/2
</p>
<p>n &perp;b h0l: l + h &frac14; 2n (c + a)/2
</p>
<p>n &perp;c hk0: h + k &frac14; 2n (a + b)/2
</p>
<p>d &perp;a 0kl: k + l &frac14; 4n (k, l &frac14; 2n) (b � c)/4
</p>
<p>d &perp;b h0l: l + h &frac14; 4n (l, h &frac14; 2n) (c � a)/4
</p>
<p>d &perp;c hk0: h + k &frac14; 4n (h, k &frac14; 2n) (a � b)/4
</p>
<p>150 3 X-Rays and X-Ray Diffraction</p>
<p/>
</div>
<div class="page"><p/>
<p>Space Group P63 &frasl;m
Finally here, and in order to show that these manipulations are fundamentally no more difficult with a
</p>
<p>space group not based on orthogonal axes, we study the hexagonal space group P63/m; we will
</p>
<p>address the tetragonal system through space group P4nc in a problem.
</p>
<p>From Fig. 2.38, we list the coordinates as:
</p>
<p>� fx; y; z; y; x� y; z; x&thorn; y; x; z; x; y; 1
2
&thorn; z; y; x&thorn; y; 1
</p>
<p>2
&thorn; z; x� y; x; 1
</p>
<p>2
&thorn; zg
</p>
<p>Fig. 3.26 Space group Pman. (a) General equivalent positions. (b) Symmetry elements; the origin is on �1. The diagram
shows also the limiting conditions for this space group.
</p>
<p>3.7 Limiting Conditions and Systematic Absences 151</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Fig38_2">http://dx.doi.org/10.1007/978-1-4614-3954-7#Fig38_2</a></div>
</div>
<div class="page"><p/>
<p>Since the structure is centrosymmetric, with the origin on 6/m &eth;1&THORN;, we need consider only the
cosine part of the structure factor equation (B &frac14; 0). Thus, following (3.69) we can write the
geometrical structure factor as:
</p>
<p>A=2 &frac14; cos 2p&eth;hx&thorn; ky&thorn; lz&THORN; &thorn; cos 2p&eth;�hy&thorn; k&frac12;x� y� &thorn; lz&THORN; &thorn; cos 2p&eth;h&frac12; y� x� � kx&thorn; lz&THORN;
&thorn; cos 2p&eth;�hx� ky&thorn; lz&thorn; l=2&THORN; &thorn; cos 2p&eth;hy&thorn; k&frac12; y� x� &thorn; lz&thorn; l=2&THORN;
</p>
<p>&thorn; cos 2p&eth;h&frac12;x� y� &thorn; kx&thorn; lz&thorn; l=2&THORN;
</p>
<p>Combining the cosine terms in pairs, we obtain
</p>
<p>A=4 &frac14; cos 2p&eth;lz� l=4&THORN;fcos 2p&eth;hx&thorn; ky&thorn; l=4&THORN; &thorn; cos 2p&eth;kx&thorn; iy&thorn; l=4&THORN;
</p>
<p>&thorn; cos 2p&eth;ix&thorn; hy&thorn; l=4&THORN;g
</p>
<p>[remember that i &frac14; �(h + k)]. We can now separate into equations for l even and l odd by expanding
the cosine terms (you may need Web Appendix WA5):
</p>
<p>l &frac14; 2n: A &frac14; 4 cos 2plzfcos 2p&eth;hx&thorn; ky&THORN; &thorn; cos 2p&eth;kx&thorn; iy&THORN; &thorn; cos 2p&eth;ix&thorn; hy&THORN;g
</p>
<p>l &frac14; 2n&thorn; 1: A &frac14; �4 sin 2plzfsin 2p&eth;hx&thorn; ky&THORN; &thorn; sin 2p&eth;kx&thorn; iy&THORN; &thorn; sin 2p&eth;ix&thorn; hy&THORN;g
</p>
<p>For l even, there are no reflection conditions; for l odd, A &frac14; 0 if h &frac14; k &frac14; 0. Thus, the only
</p>
<p>condition limiting X-ray reflections is 000l &frac14; l &frac14; 2n. The expression above for A/2 differs in a
trivial manner from that given in the International Tables [10]: the value therein may be achieved
</p>
<p>by using initially the crystallographically equivalent position &eth;�
1
</p>
<p>2
&thorn; z&THORN;, in place of the
</p>
<p>1
</p>
<p>2
&thorn; z
</p>
<p>employed here.
</p>
<p>3.8 Practical Determination of Space Groups from Diffraction Data
</p>
<p>The determination of the space group of a crystal is an important and early feature in the X-ray
</p>
<p>analysis of its structure. We shall assume that we have available the X-ray diffraction record for the
</p>
<p>several examples of monoclinic and orthorhombic crystals to be examined. It is necessary to bear in
</p>
<p>mind that X-ray techniques can reveal the presence of that translational symmetry which can arise
</p>
<p>through symmetry operations:
</p>
<p>1. Translations relating to centering of the unit cell (a/2 and/or b/2 and/or c/2).
</p>
<p>2. Translations relating to glide planes.
</p>
<p>3. Translations relating to screw axes.
</p>
<p>or any combination of these symmetry operations. All categories lead to systematic absences, and the
</p>
<p>totality of the translational symmetry, together with the Laue group, forms the diffraction symbol,
</p>
<p>Sect. 2.9.
</p>
<p>152 3 X-Rays and X-Ray Diffraction</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec28_2">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec28_2</a></div>
</div>
<div class="page"><p/>
<p>3.8.1 Monoclinic Space Groups
</p>
<p>Single crystal X-ray photographs taken with a monoclinic crystal showed typically the reflections
</p>
<p>listed in Table 3.5.
</p>
<p>From the important reflection types, hkl, h0l, and 0k0, we deduce the limiting conditions:
</p>
<p>hkl h + k &frac14; 2n
h0l (h &frac14; 2n)
0k0 (k &frac14; 2n)
</p>
<p>Using Table 3.7, we conclude that for crystal I, the space group is one of C2, Cm, or C2/m.
</p>
<p>The diffraction data alone do not distinguish between these three possible space groups; we show in
</p>
<p>Sect. 4.2.3ff how this ambiguity might be resolved.
</p>
<p>Table 3.6, for monoclinic crystal II, provides the next list of diffraction data for inspection.
</p>
<p>Table 3.5 Some reflection data for monoclinic
crystal I
</p>
<p>hkl 200 401 112 510
201 402 113 020
202 600 114 040
203 110 310 060
400 111 311 080
</p>
<p>Table 3.6 Some reflection data for monoclinic
crystal II
</p>
<p>hkl 100 204 111 322
200 402 122 020
300 502 113 040
400 110 311 060
202 310 123 080
</p>
<p>Table 3.7 Limiting conditions for the mono-
clinic space groups
</p>
<p>Conditions limiting possible
</p>
<p>X-ray reflections Space groups
</p>
<p>hkl: none
</p>
<p>h0l: none
</p>
<p>0k0: none
</p>
<p>)
P2, Pm, P2/m
</p>
<p>hkl: none
</p>
<p>h0l: none
</p>
<p>0k0: k &frac14; 2n
</p>
<p>)
P21, P21/m
</p>
<p>hkl: none
</p>
<p>h0l: l &frac14; 2n
0k0: none
</p>
<p>)
Pc, P2/c
</p>
<p>hkl: none
</p>
<p>h0l: l &frac14; 2n
0k0: k &frac14; 2n
</p>
<p>)
P21/c
</p>
<p>hkl: h&thorn; k &frac14; 2n
h0l: none
</p>
<p>0k0: none
</p>
<p>)
C2, Cm, C2/m
</p>
<p>hkl: h&thorn; k &frac14; 2n
h0l: l &frac14; 2n&eth;h &frac14; 2n&THORN;
0k0: none
</p>
<p>)
Cc, C2/c
</p>
<p>3.8 Practical Determination of Space Groups from Diffraction Data 153</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec13_4">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec13_4</a></div>
</div>
<div class="page"><p/>
<p>There is no condition on hkl, but h0l are restricted by l being even, and 0k0 by k being even: this
</p>
<p>space group is identified uniquely as P21/c.
</p>
<p>The limiting conditions for the 13 monoclinic space groups are listed in Table 3.7, in their standard
</p>
<p>orientations. In practice, it is possible, by an inadvertent choice of axes, to find oneself working with a
</p>
<p>non-standard space-group symbol. Generally, a fairly straightforward transformation of axes will lead
</p>
<p>to the standard setting (see Problems 2.12 and 3.17).
</p>
<p>3.8.2 Orthorhombic Space Groups
</p>
<p>We begin with the sample data in Table 3.8. From these data, we deduce the conditions below:
</p>
<p>Limiting conditions deduced:
</p>
<p>hkl None h00 h &frac14; 2n
0kl None 0k0 k &frac14; 2n
h0l None 00l l &frac14; 2n
hk0 None
</p>
<p>Examining in the prescribed hierarchy, we find only 21 axes parallel to x, y, and z: the space group
</p>
<p>is determined uniquely as P212121, Sect. 2.7.7 and Table 2.7.
</p>
<p>In the final two examples, we consider only the conclusions drawn from an inspection of the
</p>
<p>diffraction records. In the first instance, we have:
</p>
<p>hkl None h00 None
</p>
<p>0kl k &frac14; 2n 0k0 (k &frac14; 2n)
h0l l &frac14; 2n 00l (l &frac14; 2n)
hk0 None
</p>
<p>The diffraction symbol ismmmPbc� so that the space group is either Pbc21 or Pbcm; the distinction
between them depends upon the presence, or otherwise, of a center of symmetry.
</p>
<p>In the second example, we have:
</p>
<p>hkl None h00 (h &frac14; 2n)
0kl k &frac14; 2n 0k0 (k &frac14; 2n)
h0l l &frac14; 2n 00l (l &frac14; 2n)
hk0 h &frac14; 2n
</p>
<p>and space group Pbca is uniquely determined.
</p>
<p>These results seem quite reasonable and straightforward, but nevertheless, one might be tempted to
</p>
<p>question their validity. For example, in the first orthorhombic crystal, is there a space group in class
</p>
<p>mmm that would give the same systematic absences as those in Table 3.8? Experience tells us that
</p>
<p>there is not. Since no glide planes are indicated by the systematic absences, the three symmetry
</p>
<p>planes, if present, would have to be m-planes. Three m-planes could not be involved with three 21
</p>
<p>Table 3.8 Some reflection data for an ortho-
rhombic crystal
</p>
<p>hkl 111 011 110 020
112 021 120 040
212 012 310 060
312 101 200 002
322 203 400 004
332 303 600 006
</p>
<p>154 3 X-Rays and X-Ray Diffraction</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec22_2">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec22_2</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Tab7_2">http://dx.doi.org/10.1007/978-1-4614-3954-7#Tab7_2</a></div>
</div>
<div class="page"><p/>
<p>axes unless the unit cell were centered, for example, as in Immm, which would restrict the hkl
</p>
<p>reflections to h + k + l &frac14; 2n. Hence, our original conclusion is correct.
</p>
<p>3.8.3 Tetragonal Space Groups
</p>
<p>The following reflections conditions were obtained for two tetragonal crystals of Laue groups
4
</p>
<p>m
mm
</p>
<p>and
4
</p>
<p>m
, respectively. It may help to consider again Table 1.5.
</p>
<p>(a) hkl : none hk0 : h&thorn; k &frac14; 2n 0kl : k &frac14; 2n
P n? z b? x
</p>
<p>Consulting the International Tables, Volume 1 (or Volume A) on diffraction symbols shows that
</p>
<p>this space group is P
4
</p>
<p>n
bm.
</p>
<p>(b) hkl : h&thorn; k &thorn; l &frac14; 2n hk0 : h; &eth;k&THORN; &frac14; 2n 00l : l &frac14; 4n
I a? z 41 jj z
</p>
<p>This space group is I
41
</p>
<p>a
.
</p>
<p>3.8.4 Hexagonal Space Groups
</p>
<p>The following reflections conditions were obtained for two hexagonal crystals of point groups (a) 622
</p>
<p>and (b) either
6
</p>
<p>m
mm or 6mm, respectively.
</p>
<p>(a) hkil : none 000l : l &frac14; 6n
P 61 or 65
</p>
<p>The diffraction symbol is: 622P 61&eth;65&THORN; � �; thus, the space group is either P6122 or its enantio-
morph P6522.
</p>
<p>(b) hkl : none h0�hl : l &frac14; 2n
P c?x; &eth;y&THORN;
</p>
<p>If the point group is
6
</p>
<p>m
mm, reference to the International Tables for diffraction symbol
</p>
<p>6
</p>
<p>m
mmP�c�
</p>
<p>leads to the space group is P
63
</p>
<p>m
cm, whereas if the point group is 6mm, similar considerations indicate
</p>
<p>space group P63cm.
</p>
<p>The practicing X-ray crystallographer is assisted by the information on space groups in Volume A
</p>
<p>(and the earlier Volume 1) of the International Tables for Crystallography [10].
</p>
<p>Combined with a working knowledge of symmetry, these tables enable most symmetry situations
</p>
<p>arising in the course of a structure analysis to be treated correctly.
</p>
<p>3.9 Problems
</p>
<p>3.1. What is the change in wavelength of an X-ray photon scattered incoherently by a free electron
</p>
<p>at 45� to the forward direction of the incident beam? If the wavelength of the incident photon is
</p>
<p>1 Å, what is the energy of the scattered photon?
</p>
<p>3.9 Problems 155</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Tab5_1">http://dx.doi.org/10.1007/978-1-4614-3954-7#Tab5_1</a></div>
</div>
<div class="page"><p/>
<p>3.2. Two identical coherent scattering centers are separated by the distance 2l, and X-rays
</p>
<p>fall normally on to the line joining the two centers. For 2y &frac14; 0 to 180� in steps of 30�, calculate
the scattered amplitudes and intensities as fractions of the results with both scatterers at
</p>
<p>one point.
</p>
<p>3.3. Calculate the atomic scattering factor f for beryllium at (sin y)/l &frac14; 0.0, 0.2, and 0.5. The
expression for f(1s) has been given in the text. The Slater wave function for the 2s electron
</p>
<p>may be given as C2s &frac14; &eth;c
5
2= 96p&THORN;
</p>
<p>1=2
r exp&eth;c2r=2&THORN;; you may need the general result&ETH;1
</p>
<p>0
xn exp&eth;�ax&THORN; sin bx dx &frac14; n!&frac12;&eth;a&thorn; ib&THORN;n&thorn;1�&eth;a� ib&THORN;n&thorn;1�=&frac12;2i&eth;a2&thorn; b2&THORN;n&thorn;1�, from which the
</p>
<p>similar expression given in the text for the 1s wavefunction applies for the case n &frac14; 1. The
</p>
<p>screening constants for beryllium are s1s &frac14; 0.3 and s2s &frac14; 2.05. Compare the results that are
obtained for f with those from the expression f &frac14;
</p>
<p>P4
j aj exp&eth;�bjs
</p>
<p>2&THORN; &thorn; c, where s is (sin y)/l
</p>
<p>and the values of a, b, and c for beryllium are listed below:
</p>
<p>a1 b1 a2 b2 a3 b3 a4 b4 c
</p>
<p>1.5919 43.6427 1.1278 1.8623 0.5391 103.483 0.7029 0.5420 0.0385
</p>
<p>3.4. An X-ray tube is operated at 30 kV. What is the energy, in J, associated with each X-ray
</p>
<p>photon produced by the tube?
</p>
<p>3.5. Calculate the transmittance factor (I/I0) for a 1 mm crystal plate of benzene (C6H6), the density
</p>
<p>of which is 1,124 kg m�3. The mass absorption coefficient for Cu Ka X-radiation and the
</p>
<p>relative atomic masses are as follow:
</p>
<p>C H
</p>
<p>m/m2 kg�1 0.46 0.04
</p>
<p>Mr 12.01 1.008
</p>
<p>3.6. There are eight combinations of one to three negative signs among the indices hkl for any
</p>
<p>general reflection. With the aid of the geometrical structure factors given in the text, derive the
</p>
<p>relationships between the eight forms of the phase angle f(hkl) for (a) space group P21 and
</p>
<p>(b) space group Pma2.
</p>
<p>3.7. A triclinic unit cell has the dimensions a &frac14; 7.36 Å, b &frac14; 9.21 Å, c &frac14; 13.47 Å, and
</p>
<p>a &frac14; 101.22�, b &frac14; 110.62�, g &frac14; 123.41�. Calculate (a) the six parameters of the reciprocal
unit cell for Cu Ka radiation (l &frac14; 1.5418 Å), and (b) the volumes of the real and reciprocal
</p>
<p>unit cells.
</p>
<p>3.8. In the direct unit cell of Problem 3.7, two atoms are situated at the fractional coordinates 0.10,
</p>
<p>0.30, 0.20 and 0.10, 0.15, 0.35 for x, y, z, respectively. By means of vector expressions,
</p>
<p>calculate the distance between the two atoms, and the angle subtended at the origin by the
</p>
<p>vectors from the origin to each of the two atoms.
</p>
<p>3.9. Three atoms have the following amplitudes and phases with respect to the real axis of an
</p>
<p>Argand diagram:
</p>
<p>(a) 13.1, 16.23�
</p>
<p>(b) 21.4, 154.87�
</p>
<p>(c) 37.9, �113.26�
</p>
<p>Calculate the amplitude and phase of the resultant sum.
</p>
<p>3.10. Express the structure factor equation in a reduced form for an A-face centered unit cell. Hence,
</p>
<p>deduce the limiting conditions associated with A centering.
</p>
<p>156 3 X-Rays and X-Ray Diffraction</p>
<p/>
</div>
<div class="page"><p/>
<p>3.11. A two-dimensional structure has four atoms per unit cell, two of type P and two of type Q,
</p>
<p>with the following fractional coordinates:
</p>
<p>x y
</p>
<p>P1 0.1 0.2
</p>
<p>P2 0.9 0.8
</p>
<p>Q1 0.2 0.7
</p>
<p>Q2 0.8 0.3
</p>
<p>Calculate jF(hk)j for the reflections 5 0, 0 5, 5 5 and 5 10 in terms of the scattering factors gP
and gQ for the two species. If gP &frac14; 2gQ, what are the phase angles for these reflections?
</p>
<p>3.12. a-Uranium crystallizes in the orthorhombic system with four uranium atoms in special
</p>
<p>positions:
</p>
<p>� f0; y; 1
4
; 1
</p>
<p>2
; 1
2
&thorn; y; 1
</p>
<p>4
g
</p>
<p>Use the data below to decide whether y is better chosen as 0.10 or 0.15.
</p>
<p>hkl jF(hkl)j gU(hkl)
020 88.5 70.0
</p>
<p>110 268.9 80.0
</p>
<p>3.13. The unit cell dimensions of a-uranium are a &frac14; 2.85 Å, b &frac14; 5.87 Å, c &frac14; 5.00 Å. Use the value
of yU from Problem 3.12 to determine the shortest U&ndash;U distance in the structure. It may be
</p>
<p>helpful to plot the uranium atom positions in a few neighboring unit cells.
</p>
<p>3.14. In the examples listed below for monoclinic crystals, the conditions limiting possible X-ray
</p>
<p>reflections are given. In each case, write the possible space groups corresponding to the
</p>
<p>information given.
</p>
<p>(a)
</p>
<p>hkl None
</p>
<p>h0l None
</p>
<p>0k0 k &frac14; 2n
</p>
<p>(b)
</p>
<p>hkl None
</p>
<p>h0l h &frac14; 2n
0k0 None
</p>
<p>(c)
</p>
<p>hkl h + k &frac14; 2n
h0l l &frac14; 2n (h &frac14; 2n)
0k0 (k &frac14; 2n)
</p>
<p>(d)
</p>
<p>hkl None
</p>
<p>h0l None
</p>
<p>0k0 None
</p>
<p>3.9 Problems 157</p>
<p/>
</div>
<div class="page"><p/>
<p>3.15. Repeat Problem 3.14, but for the limiting conditions below relating to orthorhombic crystals.
</p>
<p>(a)
</p>
<p>hkl None h00 h &frac14; 2n
0kl None 0k0 k &frac14; 2n
h0l None 00l None
</p>
<p>hk0 None
</p>
<p>(b)
</p>
<p>hkl None h00 None
</p>
<p>0kl k &frac14; 2n 0k0 k &frac14; 2n
h0l None 00l None
</p>
<p>hk0 None
</p>
<p>(c)
</p>
<p>hkl h + k + l &frac14; 2n h00 h &frac14; 2n
0kl k &frac14; 2n, l &frac14; 2n 0k0 k &frac14; 2n
h0l h + l &frac14; 2n 00l l &frac14; 2n
hk0 h + k &frac14; 2n
</p>
<p>3.16. (a) Write the independent conditions limiting possible X-ray reflections for the following
</p>
<p>space groups: (i) P21/a; (ii) Pc; (iii) C2; (iv) P2122; (v) Pcc2; (vi) Imam. In each case, write
</p>
<p>the symbols of the space groups, if any, in the same crystal system with the same limiting
</p>
<p>conditions. (b) Write the conditions limiting possible X-ray reflections in the monoclinic space
</p>
<p>group P21/n (non-standard setting). (c) Give the conventional symbols for the space groups
</p>
<p>A2/a and B21221.
</p>
<p>3.17. (a) Space group Pcab corresponds to the non-standard setting acb, that is, a along x,�c along y,
and b along z. What is the symbol in the standard (abc) setting? (b) What is the essential
difference between the space groups represented by the standard symbols Pmna and Pnma?
</p>
<p>What are their full symbols?
</p>
<p>3.18. The absorption correction for a crystal ground into a sphere of radius r is dependent on r, m, and
</p>
<p>y. Assume that extinction effects are negligible and determine the ideal intensity for an hkl
</p>
<p>reflection, given that the measured intensity less background is 56.3, and that r &frac14; 0.11 mm and
m &frac14; 18.2 � 103 m�1. For this reflection, y &frac14; 30�, and some tabulated data are listed above,
corresponding to the numerical integration A &frac14; f&eth;1=V&THORN;
</p>
<p>&ETH;
dx
&ETH;
dy
&ETH;
exp&frac12;�m &eth;r0 &thorn; r&THORN;� dzg
</p>
<p>�1
,
</p>
<p>where r0 and r are, respectively, the incident and diffracted paths lengths in the crystal.
</p>
<p>Include the y-dependent Lorentz and polarization corrections.
</p>
<p>Transmission factors A for a sphere
</p>
<p>of radius R and linear absorption
</p>
<p>coefficient �
</p>
<p>y/�
</p>
<p>mR 25 30 35
</p>
<p>1 3.88 3.79 3.70
2 10.9 10.0 9.26
3 22.4 19.5 17.1
4 37.2 31.0 26.3
</p>
<p>158 3 X-Rays and X-Ray Diffraction</p>
<p/>
</div>
<div class="page"><p/>
<p>3.19. For space group P63/m, what are (a) the Sch&euro;onflies point group symbol, (b) the full
</p>
<p>Hermann&ndash;Mauguin point-group and space-group symbols, (c) the crystal system, (d) the crystal
</p>
<p>class, (e) the lattice, and (f) the conventional unit cell?
</p>
<p>3.20. Using the coordinates of the general equivalent positions for space group P4nc, Fig. 2.37,
</p>
<p>derive (a) the geometrical structure factors, (b) the amplitude symmetry, (c) the phase-angle
</p>
<p>symmetry.
</p>
<p>References
</p>
<p>1. See Bibliography, International tables for X-ray crystallography, vol 4
</p>
<p>2. Gilfrich JV et al (1995) Advances in X-ray analysis, vol 39. In: Proceedings of the 45th annual conference on
</p>
<p>applications of X-ray analysis, Colorado Springs
</p>
<p>3. http://blog.the-scientist.com/2011/04/11/multipole-wigglers/
</p>
<p>4. Karsch S et al (2010) Max-Planck Instit&euro;ut f&euro;ur Quantenoptik. In: Osterhoff Jens (ed) Laser-plasma acceleration.
Universit&euro;at Hamburg, Hamburg
</p>
<p>5. See Bibliography: James (1958), Woolfson (1977) or http://www.gwyndafevans.co.uk/thesis-html/node11.html
</p>
<p>6. Ladd M (1998) Introduction to physical chemistry, 3rd edn. Cambridge University Press, Cambridge
</p>
<p>7. (1962) International tables for X-ray crystallography, vol IV. Kynoch Press
</p>
<p>8. Ewald PP (1913) Z Phys 14:465
</p>
<p>9. Holmes KC (2006) Nature 440:149
</p>
<p>10. See Bibliography, Chapter 1, Henry et al (1965)
</p>
<p>Bibliography: Synchrotron Radiation
</p>
<p>Helliwell JR (1992) Macromolecular crystallography with synchrotron radiation. Cambridge University Press,
</p>
<p>Cambridge
</p>
<p>Structure Factor and Intensity
</p>
<p>James RW (1958) Optical principles of the diffraction of X-rays: the crystalline state, vol 2. Bell, London
</p>
<p>Prince E, Wilson AJC (1999) International tables for X-ray crystallography, vol C, 2nd edn. Kluwer Academic,
</p>
<p>Dordrecht
</p>
<p>Woolfson MM (1977) An introduction to X-ray crystallography, 2nd edn. Cambridge University Press, Cambridge
</p>
<p>Atomic Scattering Factors
</p>
<p>Ibers JA, Hamilton WC (eds) (1974) International tables for X-ray crystallography, vol 4. Kynoch Press, Birmingham
</p>
<p>References 159</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Fig37_2">http://dx.doi.org/10.1007/978-1-4614-3954-7#Fig37_2</a></div>
<div class="annotation"><a href="http://blog.the-scientist.com/2011/04/11/multipole-wigglers/">http://blog.the-scientist.com/2011/04/11/multipole-wigglers/</a></div>
<div class="annotation"><a href="http://www.gwyndafevans.co.uk/thesis-html/node11.html">http://www.gwyndafevans.co.uk/thesis-html/node11.html</a></div>
</div>
<div class="page"><p/>
<p>Intensities and Intensity Statistics 4
</p>
<p>4.1 Intensity Expressions and Factors Affecting Intensities
</p>
<p>The measurement of the intensity of a diffracted X-ray beam can be carried out photographically by
</p>
<p>camera methods, but almost always today by quantum counting with diffractometer techniques.
</p>
<p>We can measure either a peak intensity or an integrated intensity, the latter parameter being preferred
</p>
<p>for the expression of the intensity of X-ray reflection.
</p>
<p>Real crystals are not geometrically perfect, so that a given reflection will be observed over a small,
</p>
<p>finite angular range. Hence, we need to be able to determine the area under a curve such as that shown
</p>
<p>in Fig. 4.1 in order to represent a total intensity of a reflection. In the photographic method, the peak
</p>
<p>intensity is recorded over a grid of points and the integrated result imposed onto a photographic film.
</p>
<p>In collecting intensities with a diffractometer, a scintillation counter sweeps through a pre-set angular
</p>
<p>range�dy0, so recording the total number of counts, or integrated intensity. We shall discuss some of
the practical implications of these techniques in the next chapter, but much of the ensuing discussion
</p>
<p>in this chapter will have the collection of intensity data by an X-ray diffractometer and its subsequent
</p>
<p>treatment in mind.
</p>
<p>The total energy of a given diffraction spectrum E&eth;hkl&THORN; at any given angle y0, for a crystal
sufficiently small that absorption may be neglected, and completely bathed in an X-ray beam and
</p>
<p>rotating with a uniform angular velocity o, is given for unpolarized incident radiation of incident
</p>
<p>intensity I0, by
</p>
<p>E&eth;hkl&THORN;o=I0 &frac14; Qdn (4.1)
</p>
<p>where E&eth;hkl&THORN;o=I0 is known as the integrated reflection, dv is the volume of the crystal, and Q is
given by
</p>
<p>Q &frac14; &eth;N2l3= sin 2y0&THORN;jF&eth;hkl&THORN;j2&frac12;e2=&eth;4pe0mec2&THORN;2&eth;1&thorn; cos22y0&THORN;=2� (4.2)
</p>
<p>where N is the number of unit cells per unit volume of the crystal, and the other terms have
</p>
<p>their conventional meanings. The derivation of these expressions has been discussed in detail
</p>
<p>elsewhere [1].
</p>
<p>Since the value of the integrated reflection does not actually depend upon the angular velocity, we
</p>
<p>let R&eth;y&THORN;I0 be the radiation reflected at the angle y0 by the crystal, so that R(y) may be called the
reflecting power. Then,
</p>
<p>M. Ladd and R. Palmer, Structure Determination by X-ray Crystallography:
</p>
<p>Analysis by X-rays and Neutrons, DOI 10.1007/978-1-4614-3954-7_4,
# Springer Science+Business Media New York 2013
</p>
<p>161</p>
<p/>
</div>
<div class="page"><p/>
<p>E &frac14;
&eth;
R&eth;y&THORN;I0=o dy
</p>
<p>so that
</p>
<p>Eo=I0 &frac14;
&eth;
R&eth;y&THORN; dy &frac14; Qdn (4.3)
</p>
<p>The term
&ETH;
R&eth;y&THORN; dy expresses the area under the curve in Fig. 4.1. From (4.2), we can write
</p>
<p>Eo=I0 &frac14; KC&eth;hkl&THORN;&eth;1=sin2y&THORN;
1
</p>
<p>2
&eth;1&thorn; cos22y&THORN;jF&eth;hkl&THORN;j2 (4.4)
</p>
<p>where K is a scaling factor and C(hkl) is a factor that depends upon absorption and extinction, both of
</p>
<p>which we shall discuss shortly. Essentially, the area under the curve of Fig. 4.1 may be expressed as
</p>
<p>&eth;
R&eth;y&THORN; dy &frac14; KC&eth;hkl&THORN;LpjF&eth;hkl&THORN;j2 (4.5)
</p>
<p>where the L and p are the trigonometrical terms in (4.4), to be discussed next.
</p>
<p>4.1.1 Polarization and Lorentz Factors
</p>
<p>In (4.5), L and p represent the Lorentz and polarization factors, respectively, thus linking the quantity
</p>
<p>measured, the reflecting power, to the quantity sought, the corrected jF(hkl)j2 value, which we may
refer to as the ideal intensity.
</p>
<p>The polarization factor p, discussed in Sect. 3.2.4, takes into account the fact that the output of a
</p>
<p>conventionalX-ray tube is unpolarized radiation,whereas the radiation after reflection froma crystal plane
</p>
<p>is polarized, thus decreasing the intensity of the diffracted beam as a function of the scattering angle 2y.
</p>
<p>Fig. 4.1 Variation of reflection power R(y) with y; the intensity at the Bragg angle y0 is recorded over the angular
range �dy0
</p>
<p>162 4 Intensities and Intensity Statistics</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec12_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec12_3</a></div>
</div>
<div class="page"><p/>
<p>Where the incident beam is polarized, for example, after reflection from a crystal monochromator,
</p>
<p>the polarization factor is modified to &eth;1&thorn; cos22ycos22ym&THORN;=&eth;1&thorn; cos22ym&THORN;, where 2ym is the angle
between the incident and scattered beams at the monochromator.
</p>
<p>The Lorentz factor L depends on the diffraction geometry and expresses a time-of-reflection
</p>
<p>opportunity for a crystal plane in the X-ray beam. For a rotating crystal with the X-ray beam
</p>
<p>normal to a reflecting plane the L factor is 1=&eth;sin 2y&THORN;; with a powder specimen it takes the form
2=&eth;sin y sin 2y&THORN;.
</p>
<p>In order to give expression to the Lorentz factor in a particular case, let P be a point on a zero level of
</p>
<p>the reciprocal lattice, normal to a rotation axis that passes through its originO; the crystal is atQ and the
</p>
<p>incident X-ray beam direction is QO, as shown in Fig. 4.2. The constant angular velocity of the crystal
</p>
<p>is o, so that the reciprocal lattice point P has a linear velocity jSjo. The speed with which P moves
through the surface of the Ewald sphere is the component of its velocity along the radius QP. Since
</p>
<p>ffQOP &frac14; ffQPO &frac14; &eth;90� y&THORN;�, the velocity v of the point P as it passes through the sphere is given by
</p>
<p>n &frac14; ojSj cos y (4.6)
</p>
<p>Since jSj &frac14; 2 sin y=l, the velocity v is equal to (o/l) sin 2y. The time t taken for P to pass through the
reflecting position is proportional to 1/v, so that this time-of-reflection is given by
</p>
<p>t &frac14; k=n &frac14; k=&eth;ojSj cos y&THORN; (4.7)
</p>
<p>where k is a constant depending on both the size of the reciprocal lattice, in practice, the wavelength
</p>
<p>of X-radiation, and the limits�dy for finite reflection. The denominator in (4.7) depends on the time-
of-reflection opportunity for the given crystal plane; it is the Lorentz factor when the rotation axis is
</p>
<p>normal to the reflecting plane. From (4.6), o/v &frac14; 1/(jSj cos y), so that
</p>
<p>L &frac14; o=n &frac14; l=&eth;2 sin y cos y&THORN; &frac14; l=&eth;sin 2y&THORN; (4.8)
</p>
<p>Fig. 4.2 Lorentz factor: sphere of reflection with the crystal at its center Q; O is the origin of the reciprocal lattice, P is
a reciprocal lattice point hkl in the position for a reflection from the corresponding plane. The distanceOP is jSj, and the
normal to it represents the velocity vector of P; ffQOP &frac14; ffQPO &frac14; &eth;90� y&THORN;�
</p>
<p>4.1 Intensity Expressions and Factors Affecting Intensities 163</p>
<p/>
</div>
<div class="page"><p/>
<p>Since both o and l remain constant, L is equal to 1= sin 2y for the given experimental arrangement. This
argument assumes k = 1; if k = l, S = 2 sin y, and the same result obtains.
</p>
<p>4.1.2 Extinction
</p>
<p>We consider a crystal bathed in the X-ray beam, under the conditions for normal Bragg reflection,
</p>
<p>with all unit cells stacked together in a regular manner. Figure 4.3 shows a family of planes, all in the
</p>
<p>same orientation y with respect to the X-ray beam. It is clear that the first-reflected ray BC is in the
</p>
<p>correct orientation for a second reflection CD, and so on. Since there is always an inherent phase
</p>
<p>change of p/2 on reflection, the doubly reflected ray CD has a phase difference of pwith respect to the
</p>
<p>incident ray AB. We note in passing that the phase change of p is neglected in crystal-structure
</p>
<p>calculations since it occurs equally for all reflections.
</p>
<p>Primary Extinction
In general, rays that are reflected n and (n � 2) times differ in phase by p, so that the net result is a
reduction in intensity of the incident X-ray beam and, hence, in the diffracted beam in passing through
</p>
<p>the crystal. Energy is effectively conserved in this process, because each beam is depleted in energy
</p>
<p>by scattering into another beam, while being enhanced in energy by that which is scattered into the
</p>
<p>beam itself from other beams [2]. This effect is termed primary extinction, but it is very much reduced
</p>
<p>if the crystal is not perfect in its stacking. In fact, very few crystals are perfect: they are composed of
</p>
<p>an array of slightly misaligned blocks, constituting the so-called mosaic character of the crystal,
</p>
<p>Fig. 4.4. The ranges of geometric perfection are generally very small, less than about 10�3 mm, and
</p>
<p>even crystals that show primary extinction possess some mosaic character. For the ideally perfect
</p>
<p>crystal, I / jFj, whereas for the ideally imperfect crystal, I / jFj2. Since perfection is rare and very
difficult to produce in a specimen, the imperfect state with I / jFj2 is the normal state in X-ray
crystallography.
</p>
<p>Primary extinction is most noticeable with low-order, high-intensity reflections and, if it is suspected,
</p>
<p>its effect may be very substantially reduced by the thermal shock occasioned by dipping the crystal in
</p>
<p>liquid air, thereby increasing the imperfection of the mosaic structure of the crystal.
</p>
<p>Secondary Extinction
In Sect. 3.1.3, we considered the absorption of X-rays by materials, a process that is quite independent
</p>
<p>of the mechanism of diffraction. However, under Bragg reflection another feature may arise with an
</p>
<p>attendant attenuation of the energy of the incident X-ray beam; this effect is known as secondary
</p>
<p>extinction.
</p>
<p>Fig. 4.3 Primary extinction: the phase changes by reflection at B and C are each p/2, so that between the directions BE
and CD, the total phase change is p. Thus, there is an attenuation of the incident X-ray beam reaching planes deeper into
</p>
<p>the crystal
</p>
<p>164 4 Intensities and Intensity Statistics</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec4_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec4_3</a></div>
</div>
<div class="page"><p/>
<p>Consider a situation in which the first of a set of parallel planes encountered by the X-ray beam
</p>
<p>reflects a high proportion of the incident X-ray beam. Then, planes in this set deeper into the crystal
</p>
<p>receive less incident intensity, so that they reflect less than would be expected. The effect is most
</p>
<p>noticeable with large crystals and intense, often low-order, reflections. Crystals that have a high
</p>
<p>degree of imperfection generally show very little secondary extinction, because only a relatively
</p>
<p>small number of planes in the set are in the exact reflecting position at a given time. The ideally
</p>
<p>imperfect crystal shows least secondary extinction, and often only a few very strong reflections are
</p>
<p>affected, and they will not materially affect the structure determination. Nevertheless, it is possible to
</p>
<p>bring secondary extinction into a least-squares refinement in terms of an additional variable, the
</p>
<p>extinction parameter z; see Sect. 8.4. The quantity then minimized in the refinement of the atomic and
</p>
<p>scale parameters is
</p>
<p>X
</p>
<p>hkl
</p>
<p>w&frac12;Fo � &eth;1=Kz&THORN;jFcj�2 (4.9)
</p>
<p>where Fo is the observed and jFcj the calculated structure factor amplitudes.
</p>
<p>4.1.3 Absorption Measurement and Correction
</p>
<p>Here we consider how the intensity may be corrected for absorption in obtaining a value for the ideal
</p>
<p>intensity jF(hkl)j2.
From (3.4), the transmission factor T for an X-ray beam through a crystal is given by
</p>
<p>T &frac14; I=I0 &frac14; exp&frac12;�m&eth;ti � td&THORN;� (4.10)
</p>
<p>where ti and td are path lengths through the crystal for the incident and diffracted beams, respectively.
</p>
<p>If the shape of the crystal is known exactly, then it is possible to correct for absorption:
</p>
<p>T &frac14; &eth;1=V&THORN;
&eth;
</p>
<p>V
</p>
<p>exp&frac12;�m&eth;ti &thorn; td&THORN;� dV (4.11)
</p>
<p>Fig. 4.4 Mosaic character in a crystal: the angular misalignment between blocks may vary from 2 to 30 min of arc
</p>
<p>4.1 Intensity Expressions and Factors Affecting Intensities 165</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_8#Sec29_8">http://dx.doi.org/10.1007/978-1-4614-3954-7_8#Sec29_8</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Eq4_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Eq4_3</a></div>
</div>
<div class="page"><p/>
<p>where dV is an infinitesimal portion of the volume V of the crystal [3]. Frequently, however, the
</p>
<p>crystal faces are not sufficiently well defined for this method, and an empirical procedure may be
</p>
<p>preferred.
</p>
<p>Empirical Absorption Correction with Diffractometer Data
An empirical absorption correction is easily applied to data collected with a diffractometer, Sect.
</p>
<p>5.5ff. Consider Fig. 4.5: the incident and diffracted X-rays for a general reflection with f &frac14; f0 will
intersect the transmission profile at f0 � d and f0 + d, where
</p>
<p>d &frac14; tan�1&eth;tan y cos w&THORN;
</p>
<p>Hence, d &frac14; 0 at w &frac14; �90�. The transmission profile used is that with y nearest to the appropriate
equi-inclination angle n where
</p>
<p>n &frac14; sin�1&eth;sin y sin w&THORN;
</p>
<p>The transmission factor T is given either as the arithmetic mean or as the geometric mean of the
</p>
<p>estimated incident and reflected ray transmissions:
</p>
<p>T &frac14; &frac12;Tn&eth;f� d&THORN; &thorn; Tn&eth;f&thorn; d&THORN;�=2 or T &frac14; &frac12;Tn&eth;f� d&THORN;Tn&eth;f&thorn; d&THORN;�1 2= (4.12)
</p>
<p>Transmission Profiles
The transmission is measured for axial reflections, w &frac14; 90�, as a function of f, Fig. 4.5. The
transmission is given by
</p>
<p>Ty&eth;f&THORN; &frac14; Iy&eth;f&THORN;=Iy&eth;max&THORN; (4.13)
</p>
<p>The variation of T with y is neglected as it has the same effect as a small isotropic temperature factor.
</p>
<p>A set of profiles of T as a function of f is obtained for different values of y, and applied in data
</p>
<p>processing as detailed above.
</p>
<p>Fig. 4.5 Geometry of the empirical absorption correction: the crystal rotates on the f-circle of a diffractometer with
the w-circle at �90�
</p>
<p>166 4 Intensities and Intensity Statistics</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_5#Sec19_5">http://dx.doi.org/10.1007/978-1-4614-3954-7_5#Sec19_5</a></div>
</div>
<div class="page"><p/>
<p>Absorption Correction with Area Detector Data
The empirical method just described requires single Bragg reflections to be isolated and scanned so as
</p>
<p>to produce absorption correction curves. This procedure is not possible with intensity data collected
</p>
<p>with an area detector, Sect. 5.7, and other methods for applying absorption corrections have been
</p>
<p>developed. One commonly used procedure [4] uses a least-squares method to model an empirical
</p>
<p>transmission surface as sampled by multiple symmetry-equivalent and/or azimuth rotation-equivalent
</p>
<p>intensity measurements. The fitting functions are sums of real spherical harmonics of even order:
</p>
<p>Ylm&frac12;�u&eth;0&THORN;� &thorn; Ylm&frac12;�u&eth;1&THORN;� &eth;2 � l &frac14; 2n � 8&THORN;
</p>
<p>The arguments of the functions are the components of unit direction vectors, �u(0) relating to the
reverse incident beam and �u(1) to the scattered beam, with respect to crystal-fixed Cartesian axes.
The procedure had been verified against standard absorption correction data.
</p>
<p>4.1.4 Scaling
</p>
<p>Fluctuations in the incident X-ray beam intensity and possible radiation damage to the crystal may be
</p>
<p>monitored on a diffractometer by measuring four standard reflections of moderate intensity at regular
</p>
<p>intervals, say, hourly. Two of these reflections should have w-values of approximately 0�, and two
with w near 90�, with each pair approximately 90� apart in f. The average of these intensities relative
to the average of their starting values is smoothed and used to rescale the raw intensity data. If S is this
</p>
<p>scale factor, different from the scale factor K applied to Fo, then the measured intensity Io,meas is
</p>
<p>corrected to the intensity Io,corr:
</p>
<p>Io;corr &frac14; Io;meas&eth;Lp&THORN;�1T�1S�1 (4.14)
</p>
<p>with an estimated standard deviation given by
</p>
<p>s&eth;Icorr&THORN; &frac14; sImeas&eth;Lp&THORN;�1T�1S�1 (4.15)
</p>
<p>4.1.5 Merging Equivalent Reflections
</p>
<p>Where more than the symmetry-independent region of weighted reciprocal space is measured for any
</p>
<p>given reflection, a weighted mean intensity is calculated:
</p>
<p>�I &frac14;
X
</p>
<p>j
</p>
<p>wjIj
X
</p>
<p>j
</p>
<p>wj
</p>
<p>,
(4.16)
</p>
<p>where the sum is over all n measured symmetry-equivalent values of the given reflection, and wj is
</p>
<p>given by
</p>
<p>wj &frac14; s�2j (4.17)
</p>
<p>4.1 Intensity Expressions and Factors Affecting Intensities 167</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_5#Sec27_5">http://dx.doi.org/10.1007/978-1-4614-3954-7_5#Sec27_5</a></div>
</div>
<div class="page"><p/>
<p>A chi-squared test may be used to detect equivalents that have a systematic error:
</p>
<p>w2 &frac14;
X
</p>
<p>j
</p>
<p>&frac12;&eth;Ij � �Ij&THORN;=sj�2 (4.18)
</p>
<p>where the sum is again over n symmetry-equivalent reflections and the number of degrees of freedom
</p>
<p>is (n � 1). If w2 exceeds w2n�1 at a probability level of 0.001, then the symmetry-equivalent reflection
with the highest weighted deviation from the mean, wj(Ij � �Ij), is rejected and the test repeated on the
remaining equivalents. If n &frac14; 2, the smaller intensity value is rejected. The merging Rint value1 is
defined by
</p>
<p>Rint &frac14;
X
</p>
<p>hkl
</p>
<p>X
</p>
<p>j
</p>
<p>jIj � �Ijj
 !
</p>
<p>X
</p>
<p>hkl
</p>
<p>X
</p>
<p>j
</p>
<p>Ij
</p>
<p> !,
(4.19)
</p>
<p>4.1.6 Practical Intensity Expression and its Standard Deviation
</p>
<p>We have developed the necessary theory to express the intensity of a reflection and the corrections
</p>
<p>that need to be applied to it in order to obtain the ideal intensity. A measurement of intensity involves
</p>
<p>values for both the intensity of the reflection, over a range �dy0, and the background. These
parameters are measured in diffractometry, by a step-scan moving-window method [5]. The standard
</p>
<p>deviation s(I) in I arising from statistical fluctuations is given by
</p>
<p>s&eth;I&THORN; &frac14; &eth;I &thorn; rB&thorn; r2B&THORN;1=2 (4.20)
</p>
<p>where r is the ratio of the time spent in measuring the intensity I to that spent in measuring the
</p>
<p>background B; typically a value of r is 1.5.
</p>
<p>We now express the ideal intensity in a practical form, assuming the absence of primary extinction, as
</p>
<p>jF&eth;hkl&THORN;j2 &frac14; I&eth;hkl&THORN;T�1S�1L�1p�1 (4.21)
</p>
<p>where I(hkl) represents the intensity of the hkl reflection that has been adjusted for fluctuations in the
</p>
<p>incident X-ray beam, corrected for the background B and merged with symmetry-equivalent reflec-
</p>
<p>tions, then further corrected for absorption (and extinction) T, for scaling S, and for Lorentz L, and
</p>
<p>polarization p factors, to give ideal intensity values on a correct relative scale, with standard
</p>
<p>deviations s(I).
</p>
<p>All the corrections to intensity values that we have considered so far have been concerned with
</p>
<p>adjustments to the experimentally measured expression of the intensity of reflection. There are other
</p>
<p>related correcting factors, one of which is the secondary extinction parameter which has already been
</p>
<p>discussed; the scale factor for Fo, that is actually applied to jFcj during refinement, and the temperature
factors are considered next.
</p>
<p>1Elsewhere, Rint is also called Rm, Rmerge and Req.
</p>
<p>168 4 Intensities and Intensity Statistics</p>
<p/>
</div>
<div class="page"><p/>
<p>4.1.7 Scale Factor for Fo
</p>
<p>In the initial stage of a structure analysis, the scaling factor K for Fo can be calculated by Wilson&rsquo;s
</p>
<p>method, which we describe more fully in the context of intensity statistics, in Sect. 4.2.1. We write the
</p>
<p>scaling factor K in terms of Fo and jFcj as
</p>
<p>jFcj &frac14; KFo (4.22)
</p>
<p>and a simplistic calculation of K during a structure analysis is evidently
</p>
<p>K &frac14;
X
</p>
<p>hkl
</p>
<p>jFcj
X
</p>
<p>hkl
</p>
<p>Fo
</p>
<p>,
(4.23)
</p>
<p>where the sums are taken over all data for which Fo and jFcj are available. Normally, K is adjusted in a
least-squares refinement, where the scale factor is applied inversely to jFcj, as indicated in (4.9).
</p>
<p>4.1.8 Thermal Vibrations and the Temperature Factor
</p>
<p>The picture of a crystal with a total of j atoms in fixed positions with coordinates xj, yj, zj needs to be
</p>
<p>modified to take into account their motion arising from the vibrational thermal energy that the atoms
</p>
<p>possess at any finite temperature. Bonding forces permit small degrees of random, relative movement
</p>
<p>of atoms, dependent upon the temperature, so that a crystal contains atoms that are vibrating about
</p>
<p>their mean positions.
</p>
<p>The effect of thermal vibration is that the electron density is smeared out over a finite volume,
</p>
<p>rather than being concentrated at the atomic sites. Since the frequencies of vibrations are low relative
</p>
<p>to the time taken for an X-ray beam to traverse a crystal under normal experimental conditions, the
</p>
<p>crystal may be pictured as a time average of atoms randomly displaced from their mean positions, and
</p>
<p>this condition is imposed upon the diffraction pattern of the crystal.
</p>
<p>Thermal Vibration in One Dimension
Consider first a one-dimensional periodic arrangement of scattering centers in a row of repeat
</p>
<p>distance a, and let the jth scattering species of mean fractional position xj be displaced by a small,
</p>
<p>absolute distance uj. Since all unit cells in this structure are not identical, the structure factor F(h),
</p>
<p>using (3.63) in the x dimension alone, is given by the time and space average
</p>
<p>F&eth;h&THORN; &frac14;
X
</p>
<p>j
</p>
<p>fjexp&frac12;i2ph&eth;xj &thorn; uj=a&THORN;�
</p>
<p>&frac14;
X
</p>
<p>j
</p>
<p>fjexp&eth;i2phuj=a&THORN; exp&eth;i2phxj&THORN;
(4.24)
</p>
<p>Since the displacements uj are small, the exponential term may be expanded to three terms and,
</p>
<p>remembering that for the symmetrical vibrations of simple harmonic motion uj &frac14; 0, the average
value of exp(i2phuj/a) is approximately &eth;1� 2p2h2u2j =a2&THORN; to the third term, expressed conveniently,
and to the same approximation, as exp&eth;1� 2p2h2u2j =a2&THORN;. In the one-dimensional analysis,
h=a &frac14; 2 sin y=l; hence, from (4.23), we obtain
</p>
<p>4.1 Intensity Expressions and Factors Affecting Intensities 169</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Eq63_3l">http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Eq63_3l</a></div>
</div>
<div class="page"><p/>
<p>F&eth;h&THORN; &frac14;
X
</p>
<p>j
</p>
<p>fj;y exp&eth;�8p2u2j l�2sin2y&THORN; exp&eth;i2phxj&THORN; (4.25)
</p>
<p>The factor exp&eth;�8p2u2j sin2y=l2&THORN;, where u2j is the mean square atomic displacement in the x
direction, modifies fj strictly fj,y, to take account of thermal vibration. Normally, 8p
</p>
<p>2u2j is written as
</p>
<p>the isotropic temperature factor Bj, known as the Debye&ndash;Waller factor; initially in a structure
</p>
<p>determination, an overall value B, Sect. 4.2.1, may be applied to all atoms.
</p>
<p>Thermal Vibration in Three Dimensions
We extend the discussion now to three dimensions, that is, to a lattice of scattering centers, or atoms,
</p>
<p>so as to obtain an expression analogous to (4.25). From (3.12), we derive an expression for the
</p>
<p>observed intensity Io for a lattice of atoms at rest by multiplying this equation by its conjugate, which
</p>
<p>leads to:
</p>
<p>Io &frac14; C22y
X
</p>
<p>n
</p>
<p>X
</p>
<p>m
</p>
<p>expfi2p&frac12;&eth;rn � rm&THORN; � S�g (4.26)
</p>
<p>where each summation extends over the total number of atoms in the unit cell. Small, vector
</p>
<p>displacements uj are now applied to each atom, so that rn is replaced by rn + un, and similarly for
rm, so that (4.26) becomes
</p>
<p>Io &frac14; C22y
X
</p>
<p>n
</p>
<p>X
</p>
<p>m
</p>
<p>expfi2p&frac12;&eth;rn � rm&THORN; � S�g 	 expfi2p&frac12;&eth;un � um&THORN; � S�g (4.27)
</p>
<p>The isotropic vibration of the lattice of atoms is expressed by the mean value of the second
</p>
<p>exponential term in (4.27). Let 2p&frac12;&eth;un � um&THORN; � S� be written as pn,m; then, for any particular value of
pn,m, we can write its mean value as
</p>
<p>exp &eth;ip&THORN; &frac14; 1&thorn; ip� p2=2!� ip3=3!&thorn; p4=4!&thorn; � � � &frac14; 1� p2=2!&thorn; p4=4! (4.28)
</p>
<p>the mean values of the odd powers of p are zero, because positive and negative displacements are
</p>
<p>equally probable. A satisfactory approximation to (4.28) is then
</p>
<p>exp &eth;ip&THORN; &frac14; exp&eth;�p2=2&THORN; (4.29)
</p>
<p>so that the mean value of (4.26) becomes
</p>
<p>�Io &frac14; C22y
X
</p>
<p>n
</p>
<p>X
</p>
<p>m
</p>
<p>expfi2p&frac12;&eth;rn � rm&THORN; � S�gexp&eth;�p2n;m=2&THORN; (4.30)
</p>
<p>Now pn,m &frac14; (4/l) p sin y(un,S � um,S), where un,S is the component of the nth displacement vector
in the direction of the vector S. Hence, we need to evaluate the mean value &eth;un;S � um;S&THORN;2, which is
equivalent to u2n;S &thorn; u2m;S � 2un;Sum;S. We make the approximation that the coupling of the vibrations of
atoms in a lattice is negligible, whereupon un;Sum;S &frac14; 0, and u2n;S &frac14; u2n;S &frac14; u2S.
</p>
<p>170 4 Intensities and Intensity Statistics</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Eq12_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Eq12_3</a></div>
</div>
<div class="page"><p/>
<p>In (4.26), the double summation contains N2 terms. Those with n &frac14; m, a total of N, have an
exponential factor of unity and pn,m is equal to zero. Where n 6&frac14; m; p2n;m=2 is constant, because the
vibration has been taken to be isotropic, and is equal to 2B, where B is now given as
</p>
<p>B &frac14; 8p2u2S (4.31)
</p>
<p>We can now write (4.27) as
</p>
<p>Io &frac14; C22y
</p>
<p>(
X
</p>
<p>n
</p>
<p>X
</p>
<p>m
</p>
<p>n6&frac14;m
</p>
<p>expfi2p&frac12;&eth;rn � rm&THORN; � S�gexp&frac12;�2B&eth;sin2y&THORN;=l2� &thorn; N
)
</p>
<p>(4.32)
</p>
<p>In the expression for the mean isotropic temperature factor B; u2S is the mean square atomic
</p>
<p>displacement in the direction of vector S, that is, normal to the reflecting plane to which sin y
corresponds. Table 4.1 shows the effect of the exponential factor on the atomic scattering factor of
</p>
<p>carbon, for two values of B and from sin y=l &frac14; 0� 0:7.
A better approximation for temperature correction assumes that the motion remains isotropic, but
</p>
<p>allows B to take a particular value Bj for each atom j in a unit cell of a structure. This procedure is used
</p>
<p>in the least-squares routine in the XRAY program, Sect. 13.4.4. In general, however, each atom in a
</p>
<p>structure vibrates anisotropically, and the time-averaged electron density for an atom has the form of
</p>
<p>a triaxial ellipsoid. This ellipsoid is represented by a 3 	 3 tensor, where six Bij components are
needed in the most general case of triclinic symmetry; the tensor is symmetric, that is, Bij &frac14; Bji. The
Bij values can be calculated from the isotropic Bj or B values, but normally are allowed to evolve in a
</p>
<p>least-squares refinement of atomic parameters.
</p>
<p>Thermal vibrations increase the effective volume of the atom, so that interference within the atom
</p>
<p>becomes more noticeable. Consequently, f falls off with increasing sin y=l more rapidly than with
that calculated for an atom at rest, as shown in Fig. 4.6. The thermal vibrations of less rigidly retained
</p>
<p>atoms in a structure often have higher thermal vibrations than atoms that are more constrained by the
</p>
<p>stereochemistry. An example of this effect may be seen on the electron density map in Fig. 1.7: the
</p>
<p>carbon atoms in the eight-membered side chain have a greater freedom of movement than do those in
</p>
<p>the ring system; consequently, their thermal vibrations are larger and their electron density contours
</p>
<p>more diffuse.
</p>
<p>Table 4.1 Debye&ndash;Waller corrections for a carbon atom
</p>
<p>l�1 sin y (l�1 sin y)2 f
</p>
<p>exp&eth;�B2sin2y=l2&THORN;
B &frac14; 2 Å2 B &frac14; 4 Å2
</p>
<p>0 0 6 6 6
</p>
<p>0.10 0.01 5.126 5.024 4.925
</p>
<p>0.20 0.04 3.581 3.306 3.052
</p>
<p>0.30 0.09 2.502 .090 1.746
</p>
<p>0.40 0.16 1.950 1.416 1.028
</p>
<p>0.50 0.25 1.685 1.022 0.620
</p>
<p>0.60 0.36 1.536 0.748 0.364
</p>
<p>0.70 0.49 1.426 0.535 0.201
</p>
<p>4.1 Intensity Expressions and Factors Affecting Intensities 171</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_13#Sec10_13">http://dx.doi.org/10.1007/978-1-4614-3954-7_13#Sec10_13</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Fig7_1">http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Fig7_1</a></div>
</div>
<div class="page"><p/>
<p>Statistical Expectation Value of the Debye&ndash; Waller Factor
If the unit-cell distribution of the mean square displacement parameters of the atoms is assumed to be
</p>
<p>Gaussian, a normal distribution, with a mean �m equal to �B, and a variance s2 equal to &eth;B� �B&THORN;2, then
the expectation value W2 for the Debye&ndash;Waller factor is given as W2 &frac14; exp&frac12;�2&eth;m �s2s2&THORN;2�, where
W2 &frac14; exp&eth;�2Bs2&THORN;2 and s &frac14; sin y=l. This result has been incorporated into procedures for scaling and
normalizing measured intensities to the Wilson expectation values. The procedures can be used to
</p>
<p>determine both isotropic mB and sB, and anisotropic mUij and sUij distribution parameters. Tests with
</p>
<p>experimental data and refined structural models for several protein crystals have yielded reliable
</p>
<p>normalized structure factors, Sect. 4.2.5, with Sh E�j jEcjj=ShE 
 5%.
</p>
<p>4.2 Intensity Statistics
</p>
<p>Statistics form an important adjunct to many aspects of X-ray crystallography. They are used in
</p>
<p>assessing the precision of unit-cell and atomic parameters, for predicting the phase angles of
</p>
<p>reflections by direct methods, as discussed in Chap. 8, for determining scale and temperature factors,
</p>
<p>to name but three. In this section, we shall be concerned with the statistics of intensity distributions,
</p>
<p>and we consider first the Wilson statistics, and show how they may be used to obtain scale and
</p>
<p>temperature factors for a crystal.
</p>
<p>4.2.1 Determining Scale and Temperature Factors
</p>
<p>Wilson Plot
An important and familiar aspect of the statistics of the weighted reciprocal lattice is based on the
</p>
<p>equation developed byWilson [6] for the average ideal intensity. We write (3.63), for convenience, in
</p>
<p>a compact form:
</p>
<p>F&eth;h&THORN; &frac14;
X
</p>
<p>j
</p>
<p>gj;y exp&frac12;i2p&eth;h � rj&THORN;� (4.33)
</p>
<p>Fig. 4.6 Atomic scattering factors. (a) Stationary atom, fj,y. (b) Atom corrected for thermal vibration, fj,y Tj,y, also
called gj,y, where Tj;y &frac14; exp&eth;�B2sin2y=l2&THORN;.
</p>
<p>172 4 Intensities and Intensity Statistics</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_8">http://dx.doi.org/10.1007/978-1-4614-3954-7_8</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Eq63_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Eq63_3</a></div>
</div>
<div class="page"><p/>
<p>where h represents the reciprocal lattice point hkl, rj is the position vector of the jth atom, that is,
rj &frac14; xja + yjb + zjc; gj,y is the atomic scattering factor for the jth atom, fj, modified by a temperature
factor, such as exp&eth;�B2sin2y=l2&THORN;, and the sum is over all atoms in the unit cell.
</p>
<p>If we now multiply (4.33) by its conjugate, we obtain an expression for the ideal intensity jF(h)j2:
</p>
<p>jF&eth;h&THORN;j2 &frac14;
X
</p>
<p>j
</p>
<p>g2j;y&thorn;
X
</p>
<p>j
</p>
<p>X
</p>
<p>k
</p>
<p>j6&frac14;k
</p>
<p>gj;ygk;y exp&eth;i2ph �rj;k&THORN;
(4.34)
</p>
<p>where rj,k is the vector distance rj � rk. If the distribution of atoms is uniform over the unit cell, then
the second term on the right-hand side of (4.34) will tend to a negligible value because the many rj,k
vectors will tend to cancel one another: then, the average ideal intensity is given by
</p>
<p>jF&eth;h&THORN;j2 &frac14;
X
</p>
<p>j
</p>
<p>g2j;y (4.35)
</p>
<p>and is the basis for obtaining a preliminary scale factor for Fo and a temperature factor for f.
</p>
<p>Equation (4.35) has been found to hold satisfactorily over a wide range of structures, provided that
</p>
<p>the values of F2o are averaged over small, local ranges r in reciprocal space, such that f is not varying
</p>
<p>rapidly within any range.
</p>
<p>Applying the scale and temperature factors to (4.35), we have
</p>
<p>K2Fo&eth;h&THORN;2 &frac14; exp&eth;�2Bsin2yr=l2&THORN;
X
</p>
<p>j
</p>
<p>f 2j;yr (4.36)
</p>
<p>where yr is a representative value of y for each range and fj;yr the corresponding rest-atomic scattering
</p>
<p>factor. Taking logarithms of both sides, we write
</p>
<p>ln qr &frac14; 2 ln K &thorn; 2Bsin2yr=l2 (4.37)
</p>
<p>where qr is given by
</p>
<p>qr &frac14;
X
</p>
<p>j
</p>
<p>f 2j;yr
</p>
<p> !
Fo&eth;h&THORN;2y;r
</p>
<p>.
(4.38)
</p>
<p>and the sum is taken over all j atoms in the unit cell. If ln qr is plotted against sin
2yr=l
</p>
<p>2 and the best
</p>
<p>straight line drawn, the slope is equal to 2B and the intercept on the ordinate is equal to 2 ln K. This
</p>
<p>graph is often called a Wilson plot and is best obtained through the following procedure.
</p>
<p>Methodology
</p>
<p>1. Three-dimensional space is divided into a number of spherical shells, Fig. 4.7a, such that there are
</p>
<p>80&ndash;100 reflections in each range. Although the plot of Fig. 4.7b is against sin2y=l2, it is convenient
</p>
<p>to form the range demarcations in terms of sin3y=l3, since this parameter has the dimensions of
reciprocal volume; the demarcations can be converted into the equivalent values of sin2y=l2 later.
</p>
<p>2. Average values of Fo(h)
2 for each range are calculated, including either symmetry-equivalent
</p>
<p>reflections, or according each reflection in the asymmetric unit its correct multiplicity of planes. It
</p>
<p>is necessary also to allocate values to the accidental absences, that is, possible reflections lying
</p>
<p>within the experimental Ewald sphere that are too weak to be recorded.
</p>
<p>4.2 Intensity Statistics 173</p>
<p/>
</div>
<div class="page"><p/>
<p>Wilson [6] has shown that the most probable values for accidental absences are 0.55Fo,min for
</p>
<p>centrosymmetric crystals, 0.66Fo,min for non-centrosymmetric crystals, and 0.59Fo,min where this
</p>
<p>symmetry is undetermined; Fo,min is the minimum value of Fo(h) in the range under consideration.
Systematic absences are always ignored, as are those in a region up to the second order on each
</p>
<p>axis because they are atypical of the general distribution of intensities. In addition, the average
</p>
<p>intensity multiple (e-factor) should be applied as a divisor of each intensity value; this parameter is
</p>
<p>discussed and tabulated in Sect. 4.2.3.
</p>
<p>3. The mean values of sin2yr=l
2 may be obtained as
</p>
<p>sin2yr=l
2 &frac14; 1
</p>
<p>2
&eth;sin2yn=l2 &thorn; sin2yn&thorn;1=l2&THORN;
</p>
<p>where n + 1 is the number of the outer boundary shell defining the rth shell, starting at n &frac14; 0 for
which value sin y &frac14; sin ymin, Fig. 4.7a. Alternatively, sin2yr=l2 values may be obtained as
averages of sin2y=l2 over each range. Atomic scattering factor data are tabulated and readily
available [7].
</p>
<p>Auxiliary Plot
The Wilson plot can sometimes be non-linear, possibly because (4.35) does not hold well for the
</p>
<p>given data. It may be useful to make an auxiliary plot in which the numerator in (4.38) is replaced
</p>
<p>by
P
</p>
<p>j Z
2
j , that is, the sum of the squares of fj at y &frac14; 0. Both curves should have the same intercept,
</p>
<p>and experience indicates that a compromise between the Wilson and auxiliary plots often leads to a
</p>
<p>better result.
</p>
<p>Fig. 4.7 Scale and temperature factors. (a) Division of reciprocal space into spherical shells; the void region contains
data for which h, k, l &lt; 2. (b) Wilson plot: the intercept is 2 ln K, and the slope is 2B
</p>
<p>174 4 Intensities and Intensity Statistics</p>
<p/>
</div>
<div class="page"><p/>
<p>4.2.2 Other Aspects of the Wilson Plot
</p>
<p>In papers dealing with the calculation of jEj values, Hall and Subramanian consider the Wilson plot in
some detail [8]. They stress the need for the shells in the Wilson plot to be equally populated, which
</p>
<p>means that Fo values for the accidentally absent reflections should be estimated with care. The
</p>
<p>program GENEV [9] provides two methods for calculating jEj values: one uses the values of B and K
from the linear Wilson plot, whereas the other provides inter alia the options of linear scale, profile
</p>
<p>scale, and random fragment; the program GENEV uses Bayesian statistics [10] to obtain reliable
</p>
<p>estimates of weak reflections.
</p>
<p>Pathological Cases
Assuming that all possible care has been taken in collecting the intensity data, there may still remain
</p>
<p>some situations in which the Wilson plot is non-linear, for a variety of reasons: there may be
</p>
<p>significant non-zero values of the double summation in (4.34) or the temperature factor is non-
</p>
<p>Gaussian, both conditions leading to a partial breakdown of (4.35), on which equation the Wilson
</p>
<p>plot is based. Alternatively, the number of atoms in the unit cell may be too low to provide a uniform
</p>
<p>distribution; a proportion of atoms may lie on symmetry elements, and so contribute only to certain
</p>
<p>reflections; the proportion of heavy atoms in the unit cell may be unduly large; hypersymmetry may
</p>
<p>be present. Some of these problems have been addressed by several workers, but no totally
</p>
<p>satisfactory procedure has emerged. In most cases, the problem is resolved in the least-squares
</p>
<p>refinement of the structure; hypersymmetry is discussed briefly in Sect. 4.2.5.
</p>
<p>In handling two-dimensional data, the annular regions of equal area may contain too few reflec-
</p>
<p>tions for a true sample of the reciprocal lattice. Then it is possible to employ a batch procedure, that is,
</p>
<p>to combine groups (1, 2), (2, 3), (3, 4),. . . before the averages are taken.
</p>
<p>In the program XRAY, the routine for the Wilson plot uses this batch technique for the two-
</p>
<p>dimensional data sets. Not all of the example data sets provided give equally satisfactory results; this
</p>
<p>situation is considered again in Sect. 4.2.5.
</p>
<p>4.2.3 Statistics of Reciprocal Space
</p>
<p>The weighted reciprocal lattice exhibits four types of regularity and one type that may be described
</p>
<p>as irregular; we shall consider them in turn.
</p>
<p>Accidental Absences
Accidental absences occur in the diffraction pattern of most structures, and they are disposed in an
</p>
<p>irregularmanner within the Ewald sphere. From (3.63), it is not surprising to find that there are some
</p>
<p>instances where the sum of the vectors fj exp[i2p(hxj + kyj + lzj)] tend to cancel to a negligible value.
</p>
<p>The result depends upon the particular atomic arrangement in the structure rather than on its
</p>
<p>symmetry. Such permitted reflections of negligible intensity can be estimated in the manner discussed
</p>
<p>in Sect. 4.2.1: it is not uncommon to omit these reflections from a structure analysis, but without real
</p>
<p>justification. We now consider regular features of the weighted reciprocal lattice.
</p>
<p>Laue Symmetry
The positions of the reciprocal lattice points and the intensities associated with them conform to one
</p>
<p>of the Laue groups, that is, one of the eleven centrosymmetric point groups discussed in Sect. 1.4.2.
</p>
<p>This situation arises because of Friedel&rsquo;s Law, Sect. 3.6.1, and holds in all normal situations, that is, in
</p>
<p>the absence of resonance excitation, Sect. 3.1.3.
</p>
<p>4.2 Intensity Statistics 175</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Eq63_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Eq63_3</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Sec11_1">http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Sec11_1</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec25_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec25_3</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3# Sec4_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3# Sec4_3</a></div>
</div>
<div class="page"><p/>
<p>Systematic Absences
In the presence of translational symmetry, that is, structures with centered unit cells, glide planes,
</p>
<p>or screw axes, certain characteristic groups of reflections are absent from the diffraction records,
</p>
<p>Sect. 3.7ff. The diffracted energy that is so excluded is redistributed over other reflections. For
</p>
<p>example in a C-centered unit cell, hkl reflections are absent for h + k &frac14; 2n + 1. However, the
structure factor equation now takes the form
</p>
<p>F&eth;hkl&THORN; &frac14; 2&frac12;cos2p&eth;h&thorn; k&THORN;=4�
Xn
</p>
<p>j&frac14;1
gj 	 exp&frac12;i2p&eth;hxj &thorn; kyj &thorn; lzj&THORN;� (4.39)
</p>
<p>from which it is evident that for the reflections present, (h + k) even, jF(hkl)j has twice the value that
it would have for a corresponding primitive unit cell.
</p>
<p>Abnormal Averages
We have shown in Sect. 3.2.3 that the components of (4.33) can be represented in phase and
</p>
<p>amplitude on an Argand diagram. All types of symmetry link the rj (rj &frac14; xja + yjb + zjc) in groups
of two or more; (3.69) represents the simplest example of this feature. Thus,
</p>
<p>P
j g
</p>
<p>2
j , which we shall
</p>
<p>write as S, is enhanced and becomes a distribution parameter S (not to be confused with
</p>
<p>S &frac14; 2 sin y=l):
</p>
<p>S &frac14; eS &frac14; e
Xn
</p>
<p>j&frac14;1
g2j (4.40)
</p>
<p>Consider space group Pm, where the mirror plane is normal to y and cuts this axis at y &frac14; 0; then,
atoms are related in pairs x, y, z, and x, �y, z. Simple manipulation shows that the structure factor
</p>
<p>equation for this example becomes
</p>
<p>F&eth;hkl&THORN; &frac14;
Xn=2
</p>
<p>j&frac14;1
gj exp&frac12;i2p&eth;hxj &thorn; kyj&THORN;�&eth;2 cos 2plzj&THORN; &frac14; A0&eth;hkl&THORN; &thorn; iB0&eth;hkl&THORN; (4.41)
</p>
<p>where
</p>
<p>A0&eth;hkl&THORN; &frac14; 2
Xn=2
</p>
<p>j&frac14;1
gj cos 2p&eth;hxj &thorn; lzj&THORN; cos 2pkyj (4.42)
</p>
<p>and
</p>
<p>B0&eth;hkl&THORN; &frac14; 2
Xn=2
</p>
<p>j&frac14;1
gj sin 2p&eth;hxj &thorn; lzj&THORN; cos2pkyj (4.43)
</p>
<p>We need now to invoke the central limit theorem which states that in a sequence of independent
</p>
<p>random variables x1, x2, . . ., xj, . . ., xn, where the mean values are expressed by mj and the variances
</p>
<p>by s2j , the sum x &frac14; Sjxj tends to a normal (Gaussian) distribution, with a mean m equal to Sjmj and a
variance s2 equal to Sjs
</p>
<p>2
j , as the number of terms (n) in the sequence tends, ideally, to infinity.
</p>
<p>176 4 Intensities and Intensity Statistics</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec27_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec27_3</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_#Sec11_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_#Sec11_3</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Eq69_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Eq69_3</a></div>
</div>
<div class="page"><p/>
<p>In our application, the mean values A0&eth;hkl&THORN; and B0&eth;hkl&THORN; both tend to zero, since the positive and
negative values of these terms are equally probable and will tend to cancel one another in a normal
</p>
<p>distribution. The variance for a large sample is given generally by
</p>
<p>s2 &frac14; &eth;1=n&THORN;
X
</p>
<p>j
</p>
<p>&eth;xj � �x&THORN;2 &frac14; &eth;1=n&THORN;
X
</p>
<p>j
</p>
<p>x2j � x2 (4.44)
</p>
<p>since �x &frac14; 0 in our example, s2 &frac14; x2j . Applying this result first to A0(hkl), the jth individual variance is
given by 2g2j cos
</p>
<p>22p&eth;hxj &thorn; lzj&THORN;cos2 2p&eth;kyj&THORN;. Thus, the variance of A0(hkl) will be equal to A0&eth;hkl&THORN;2
which, by the central limit theorem, is given by
</p>
<p>A0&eth;hkl&THORN;2 &frac14;
Xn=2
</p>
<p>j&frac14;1
4g2j cos
</p>
<p>22p&eth;hxj &thorn; lzj&THORN;cos22pkyj
</p>
<p>&frac14;
Xn=2
</p>
<p>j&frac14;1
4g2j cos
</p>
<p>22p&eth;hxj &thorn; lzj&THORN; cos22pkyj
</p>
<p>(4.45)
</p>
<p>It is straightforward to show, since cos2y &frac14; &eth;1=p&THORN;
&ETH; p
0
cos2y dy, that the average value of cos2 y is 1
</p>
<p>2
.
</p>
<p>Hence, from (4.40),
</p>
<p>A0&eth;hkl&THORN;2 &frac14;
Xn=2
</p>
<p>j&frac14;1
g2j &frac14;
</p>
<p>1
</p>
<p>2
S (4.46)
</p>
<p>In a similar manner, we can show that the average B0&eth;hkl&THORN;2 is also equal to 1
2
S,so that
</p>
<p>jF&eth;hkl&THORN;2j &frac14; S (4.47)
</p>
<p>However, if we consider the zone of reflections for which k &frac14; 0, a similar analysis shows that
</p>
<p>jF&eth;h0l&THORN;2j &frac14; 2S (4.48)
</p>
<p>Hence, the e-factor for the intensities in this zone of P2/m is 2. The e-factor is dependent on the crystal
</p>
<p>class, and Table 4.2 lists the e-factors that arise in the 32 point groups.
</p>
<p>Another way of looking at these e-factors is by means of stereograms. Consider Fig. 1.31, point
</p>
<p>group �42m, and imagine the radiating normals that give rise to the poles as vectors. When projected
</p>
<p>onto the z axis there is a fourfold superposition of the gj vectors, but when projected onto the plane
</p>
<p>normal to z there is no such superposition; hence, 4/1 arises for the first direction, along z, that is,
</p>
<p>e(00l) &frac14; 4 and e(hk0) &frac14; 1.
</p>
<p>4.2.4 Acentric and Centric Distributions of Structure Factors
</p>
<p>The measured intensities of the whole reciprocal lattice or of certain two- or even one-dimensional
</p>
<p>regions of it may conform to an acentric,2 a centric, or a hypercentric distribution, and we shall
</p>
<p>consider the properties and uses of their distribution functions.
</p>
<p>2Not &ldquo;non-centrosymmetric&rdquo; and &ldquo;centrosymmetric.&rdquo;
</p>
<p>4.2 Intensity Statistics 177</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Fig31_1">http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Fig31_1</a></div>
</div>
<div class="page"><p/>
<p>Acentric Distribution
In the acentric distribution, typically for space group P1, the components A0(h) and B0(h) of the
structure factor must be considered separately. Following earlier discussions we write A0(h), omitting
the subscript y to g, as
</p>
<p>Table 4.2 Centric reflections and multiples (e-factors) for intensities in the 32 crystal classes
</p>
<p>Crystal class New diffraction symbol Centric sets Multiples
</p>
<p>1 1P None 1/1
</p>
<p>�1 �1P All 1/1
</p>
<p>m 2/mP&ndash;/&ndash; (0k0) 1/2
</p>
<p>2 2/mP&ndash;/&ndash; (h0l) 2/1
</p>
<p>2/m 2/mP&ndash;/&ndash; All 2/2
</p>
<p>mm2 mmmP&ndash; &ndash; &ndash; [(hk0) masks (h00), (0k0)] 2/2; 2/2; 4/1
</p>
<p>222 mmmP&ndash; &ndash; &ndash; 3 principal zones only 2/1; 2/1; 2/1
</p>
<p>mmm mmmP&ndash; &ndash; &ndash; All 4/2; 4/2; 4/2
</p>
<p>4 4/mP&ndash;/&ndash; (hk0) 4/1
</p>
<p>�4 4/mP&ndash;/&ndash; (hk0); (00l) 2/1
</p>
<p>4/m 4/mP&ndash;/&ndash; All 4/2
</p>
<p>�42m 4/mmmP&ndash;/&ndash; &ndash; &ndash; [(hk0), {hh0}]; [{h0l}, (00l)] 4/1; 2/1; 2/2
</p>
<p>4mm 4/mmmP&ndash;/&ndash; &ndash; &ndash; [(hk0), {h00}, {hh0}] 8/1; 2/2; 2/2
</p>
<p>422 4/mmmP&ndash;/&ndash; &ndash; &ndash; (hk0); {h0l}; {hhl} 4/2; 2/1; 2/1
</p>
<p>4/mmm 4/mmmP&ndash;/&ndash; &ndash; &ndash; All 8/2; 4/2; 4/2
</p>
<p>3 �3P&ndash; None 3/1
</p>
<p>�3 �3P&ndash; All 3/1
</p>
<p>3m(1) �3mlP&ndash; &ndash; &ndash; fh0�h0g 6/1; 1/2; 2/1
32(1) �3mlP&ndash; &ndash; &ndash; fh0�hlg 3/1; 2/1; 1/1
�3m(1) �3mlP&ndash; &ndash; &ndash; All 6/1; 2/2; 2/1
</p>
<p>6 6/mP&ndash;/&ndash; (hk0) 6/1
</p>
<p>�6 6/mP&ndash;/&ndash; (00l) 3/2
</p>
<p>6/m 6/mP&ndash;/&ndash; All 6/2
</p>
<p>�6m2 6/mmmP&ndash;/&ndash; &ndash; &ndash; [{hhl}, {hh0}, (00l)] 6/2; 2/2; 4/1
</p>
<p>6mm 6/mmmP&ndash;/&ndash; &ndash; &ndash; [(hk0), {hh0}, {h00}] 12/1; 2/2; 2/2
</p>
<p>622 6/mmmP&ndash;/&ndash; &ndash; &ndash; (hk0); (h0l); (hhl) 6/1; 2/1; 2/1
</p>
<p>6/mmm 6/mmmP&ndash;/&ndash; &ndash; &ndash; All 12/2; 4/2; 4/2
</p>
<p>23 m3P&ndash; &ndash; {hk0} 2/1; 3/1; 1/1
</p>
<p>m�3 m3P&ndash; &ndash; All 4/2; 3/1; 2/1
</p>
<p>�43m m3mP&ndash; &ndash; &ndash; [{hk0}, {hh0}] 4/1; 6/1; 2/2
</p>
<p>432 m3mP&ndash; &ndash; &ndash; {hk0}; {hhl} 4/1; 3/1; 2/1
</p>
<p>m�3m m3mP&ndash; &ndash; &ndash; All 8/2; 6/1; 4/2
</p>
<p>Column 1: Crystal class (also point group symbol)
</p>
<p>Column 2: Buerger diffraction symbols: centric zones are underlined, and the unit cell symbol is underlined where the
</p>
<p>point group is centrosymmetric. Note that the centric distribution occurs (i) for all hkl if the lattice is centrosymmetric;
</p>
<p>(ii) for a zone if the corresponding projection is centrosymmetric; (iii) for a central lattice row if the corresponding
</p>
<p>one-dimensional projection is centrosymmetric
</p>
<p>Column 3: Centric reflections are listed explicitly
</p>
<p>Column 4: Average intensity crystal class-dependent multiples (e-factors). Each p/q symbol gives the multiple
</p>
<p>p (e-factor) for a reciprocal lattice row and q that for the zone normal to the row
</p>
<p>Column 5. It may be helpful to recall the full symbols 1m1, 121 (monoclinic m and 2).
</p>
<p>178 4 Intensities and Intensity Statistics</p>
<p/>
</div>
<div class="page"><p/>
<p>A0&eth;h&THORN; &frac14;
Xn
</p>
<p>j&frac14;1
gj cos 2p&eth;h � rj&THORN;
</p>
<p>and its average value over j atoms is zero, as discussed above; then from (4.46)
</p>
<p>A0&eth;h&THORN;2 &frac14;
Xn
</p>
<p>j&frac14;1
g2j cos
</p>
<p>22p&eth;h � rj&THORN; &frac14;
1
</p>
<p>2
S (4.49)
</p>
<p>Similarly, B0&eth;h&THORN; &frac14; 0, and B0&eth;h&THORN;2 &frac14; 1
2
S. The probabilities that A0 lies between A0 and A0 + dA0, and
</p>
<p>that B0 lies between B0 and B0 + dB0, following a normal distribution of the type P&eth;x&THORN; &frac14; &eth;2ps2&THORN;�1=2
</p>
<p>exp&frac12;�&eth;x� �x&THORN;2=2s2�, are
</p>
<p>Pa&eth;A0&THORN; dA0 &frac14; 1=&eth;pS&THORN;1=2 exp&eth;�A02=S&THORN; dA0
</p>
<p>Pa&eth;B0&THORN; dB0 &frac14; 1=&eth;pS&THORN;1=2 exp&eth;�B02=S&THORN; dB0
(4.50)
</p>
<p>The region of area defined by dA0 dB0 is an infinitesimal portion of an annular ring on an Argand
diagram, Fig. 4.8, distant jFj from the origin. Since A0 and B0 are not correlated, the joint probability
that the structure amplitude jFj lies between jFj and jFj + djFj is
</p>
<p>Pa&eth;jFj&THORN; djFj &frac14; Pa&eth;A0&THORN;Pa&eth;B0&THORN; dA0 dB0 &frac14; &eth;1=pS&THORN; exp&frac12;�&eth;A02 &thorn; B02&THORN;=S� dA0 dB0
</p>
<p>&frac14; &eth;1=pS&THORN; exp&frac12;&eth;�jFj2&THORN;=S� djSj (4.51)
</p>
<p>where djSj represents an area dA0 dB0 on the Argand diagram and has the value 2pjFj djFj. Thus, the
joint probability refers to that area of the annular ring on the Argand diagram with radii jFj and
jFj + djFj, so that the acentric distribution function is
</p>
<p>Pa&eth;jFj&THORN; &frac14; &eth;2jFj=S&THORN; exp&frac12;&eth;�jFj2&THORN;=S� (4.52)
</p>
<p>Centric Distribution
Space group P�1 provides a typical centric distribution of intensity data. The structure factors are real
</p>
<p>and are given by the A0 component of the structure factor equation, that is,
</p>
<p>F&eth;h&THORN; &frac14; A0&eth;h&THORN; &frac14; 2
Xn=2
</p>
<p>j&frac14;1
gj cos 2p&eth;h � rj&THORN; (4.53)
</p>
<p>where h and rj have the meanings as before. From the central limit theorem, if the set of A
0(h) follows
</p>
<p>a normal distribution, the mean A0&eth;h&THORN; is zero, and the variance A0&eth;h&THORN;2 is the sum of n/2 terms of the
form 4g2j cos
</p>
<p>22p&eth;h � rj&THORN;, which evaluates to S, the distribution parameter defined above. Hence, the
probability that a structure factor lies between F and F + dF is given by
</p>
<p>Pc&eth;F&THORN; dF &frac14; 1=&eth;2pS&THORN;1=2 exp&eth;�F2=2S&THORN; dF (4.54)
</p>
<p>and the centric distribution function becomes
</p>
<p>4.2 Intensity Statistics 179</p>
<p/>
</div>
<div class="page"><p/>
<p>Pc&eth;F&THORN; &frac14; &eth;2pS&THORN;�1=2 exp&eth;�F2=2S&THORN; (4.55)
</p>
<p>We note that, if we wish to consider here only the amplitudes of the centric structure factors, a
</p>
<p>phase-restricted set, then it follows that
</p>
<p>Pc&eth;Famp&THORN; &frac14; 2Pc&eth;F&THORN; &frac14; &eth;2=pS&THORN;1=2 exp&eth;�F2=2S&THORN; (4.56)
</p>
<p>since the amplitude of F can be derived from either (+)F or (�)F, Fig. 4.8. We use this function in
deriving Nc&eth;E&THORN; shortly, because we will be concerned only with positive values in the centric
distribution of amplitudes.
</p>
<p>Mean Values
We are now in a position to derive mean values for jFj and jFj2 and other parameters in the two
distributions derived. The mean value x for any distribution f(x) is given generally by
</p>
<p>�x &frac14;
&ETH;
xf&eth;x&THORN; dx&ETH;
f&eth;x&THORN; dx
</p>
<p>but, because we are dealing with a normal distribution,
&ETH;
f&eth;x&THORN; dx &frac14; 1, so that the average value of x is
</p>
<p>given simply by
</p>
<p>�x &frac14;
&eth;
xf&eth;x&THORN; dx (4.57)
</p>
<p>The acentric and centric intensity distributions are plotted in terms of jFj in Fig. 4.9. It is evident
that the centric distribution is characterized by a significant proportion of both strong and weak
</p>
<p>intensities, whereas the acentric distribution has a low dispersion of intensities. These features can
</p>
<p>sometimes be recognized in precession X-ray photographs, Sect. 5.4ff.
</p>
<p>Fig. 4.8 Region on an Argand diagram for structure amplitude lying between jFj and jFj + djFj in an acentric
distribution. In a centric distribution, F can have only two possible values, shown at F1 (+jFj) and F2 (�jFj)
</p>
<p>180 4 Intensities and Intensity Statistics</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_5#Sec14_5">http://dx.doi.org/10.1007/978-1-4614-3954-7_5#Sec14_5</a></div>
</div>
<div class="page"><p/>
<p>Because the centric and acentric distributions differ, it is reasonable to suppose that the distributions
</p>
<p>may be used to distinguish between centrosymmetric and non-centrosymmetric crystals. We have noted
</p>
<p>in Sect. 3.8 that the space group suggested by the diffraction data of a crystal may often be ambiguous.
</p>
<p>The cause of the ambiguity lies frequently in Friedel&rsquo;s law, that is, the inability to determine from the
</p>
<p>positions of the diffraction spectra alone whether or not the crystal itself is centrosymmetric.
</p>
<p>A parameter that may be used as a discriminator is the ratio of the square of the average jFj to the
average of jFj2:
</p>
<p>M &frac14; jFj2=jFj2 (4.58)
</p>
<p>For the acentric distribution, we have
</p>
<p>jFj &frac14; &eth;2=S&THORN;
&eth;1
</p>
<p>0
</p>
<p>jFj2 exp&eth;�jFj2=S&THORN;djFj
</p>
<p>This integral, and many similar to it, can be solved readily by making use of the properties of the
</p>
<p>gamma (G) function (see Web Appendix WA7). Let jFj2/S &frac14; t, so that 2jFj djFj &frac14; S dt. Then,
</p>
<p>jFj &frac14; S1=2
&eth;1
</p>
<p>0
</p>
<p>t1=2 exp&eth;�t&THORN; dt
</p>
<p>The term t1/2 may be written as t(3/2&ndash;1), so that the value of the integral is G(3/2) or 1
2
p1 2= , so that
</p>
<p>jFj &frac14; 1
2
&eth;pS&THORN;1=2 (4.59)
</p>
<p>In a similar manner,
</p>
<p>jFj2 &frac14; &eth;2=S&THORN;
&eth;1
</p>
<p>0
</p>
<p>jFj3 exp&eth;�jFj2=S&THORN; djFj (4.60)
</p>
<p>Fig. 4.9 Distribution function for structure amplitudes: (a) acentric, (b) centric . The acentric distribution has a low
dispersion of jFj values, whereas the centric distribution has significant proportions of both small and large jFj values
</p>
<p>4.2 Intensity Statistics 181</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec30_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec30_3</a></div>
</div>
<div class="page"><p/>
<p>By making substitutions as before, it is straightforward to show that the integral in (4.60) equates
</p>
<p>to S. Thus,
</p>
<p>Ma &frac14;
1
</p>
<p>4
pS=S &frac14; p=4 &frac14; 0:785 (4.61)
</p>
<p>In a centric distribution, the corresponding parameter Mc is readily shown to be
</p>
<p>Mc &frac14; &eth;2S=p&THORN;=S &frac14; 2=p &frac14; 0:637 (4.62)
</p>
<p>A disadvantage inherent in these discriminators, even when the data are divided into ranges in
</p>
<p>which the variation of f (or g) with y is small, is that the variation is imposed on the results. It is
</p>
<p>preferable, therefore, to use a parameter that is not dependent upon f.
</p>
<p>4.2.5 Normalized Structure Factors
</p>
<p>In the previous section, we stressed the importance of placing intensity data on a common statistical
</p>
<p>scale, and we discussed the e-factor for the crystal classes. For improved statistical results, either
</p>
<p>unitary structure factors U(h) or normalized structure factors E(h) are employed: jU&eth;h&THORN;j2 &frac14;
jE&eth;h&THORN;j2Sjg2j =&eth;Sjgj&THORN;
</p>
<p>2
. We shall use the parameter jEj in discussing intensity statistics and direct
</p>
<p>methods.
</p>
<p>The normalized structure factor E is given by the equation
</p>
<p>jEj2 &frac14; jFj2 e
X
</p>
<p>j
</p>
<p>g2j
</p>
<p>,
(4.63)
</p>
<p>where jFj is on an absolute scale. For special classes of reflections, the e-factor must be applied in
accordance with Table 3.10.
</p>
<p>From (4.52), the acentric distribution function for normalized structure factors follows as
</p>
<p>Pa&eth;jEj&THORN; &frac14; 2jEj exp&eth;�jEj2&THORN; (4.64)
</p>
<p>and from (4.55), that for the centric distribution is
</p>
<p>Pc&eth;E&THORN; &frac14; &eth;2p&THORN;�1=2 exp&eth;�E2=2&THORN; (4.65)
</p>
<p>Again, as with the centric distribution of jFj, the distribution of jEj amplitudes in the centric case is
twice that given in (4.65), because jEj here includes both +E and &ndash;E:
</p>
<p>Pc&eth;jEj&THORN; &frac14; &eth;2=p&THORN;1=2 exp&eth;�jEj2=2&THORN; (4.66)
</p>
<p>It will be evident that these distribution equations do not involve the atomic scattering factors, so
</p>
<p>they are independent of the particular structure. As in the previous section, we can calculate mean
</p>
<p>values related to the new variable jEj. For both the acentric and the centric distributions, the average
value of jEj2 is unity. For the average value of jEj in the acentric distribution, we have
</p>
<p>182 4 Intensities and Intensity Statistics</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Table10_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Table10_3</a></div>
</div>
<div class="page"><p/>
<p>jEj &frac14; 2
&eth;1
</p>
<p>0
</p>
<p>jEj2 exp&eth;�jEj2&THORN; djEj
</p>
<p>Making the substitution jEj2 &frac14; t, the integral becomes
</p>
<p>jEj &frac14;
&eth;1
</p>
<p>0
</p>
<p>t1=2 exp&eth;�t&THORN; dt
</p>
<p>Since t1/2 may be written as t3/2&ndash;1, the integral becomes G(3/2), or 1
2
G&eth;1=2&THORN;, which is 1
</p>
<p>2
</p>
<p>p
p. Hence,
</p>
<p>jEj &frac14; 0:886. In the case of the centric distribution, a similar calculation shows that jEj &frac14; 0:798.
The parameter jEj2 � 1j offers another useful discriminant between acentric and centric distribu-
</p>
<p>tions. Here, we evaluate this parameter for the centric distribution; that for the acentric distribution
</p>
<p>forms a problem at the end of the chapter. We now use jE2 � 1j for the centric case&mdash;&eth;&thorn;E&THORN;2 &frac14; &eth;�E)2:
</p>
<p>jE2 � 1j &frac14; &eth;2=p&THORN;1=2
&eth;1
</p>
<p>0
</p>
<p>jEj2 � 1j exp&eth;�E2=2&THORN; dE
</p>
<p>&frac14; &eth;2=p&THORN;1=2
&eth;1
</p>
<p>0
</p>
<p>&eth;1� E2&THORN; exp&eth;�E2=2&THORN; dE&thorn; &eth;2=p&THORN;1=2
&eth;1
</p>
<p>1
</p>
<p>&eth;E2 � 1&THORN; exp&eth;�E2=2&THORN; dE
</p>
<p>Since, generally,
</p>
<p>&eth;
&eth;1� X2&THORN; exp&eth;�X2=2&THORN; dX &frac14;
</p>
<p>&eth;
d&frac12;X exp&eth;�X2=2&THORN;� (4.67)
</p>
<p>it follows that
</p>
<p>jE2 � 1j &frac14; &eth;2=p&THORN;1=2
(&eth;1
</p>
<p>0
</p>
<p>d&frac12;E exp&eth;�E2=2&THORN;� &thorn;
&eth;1
</p>
<p>1
d&frac12;E exp&eth;�E2=2&THORN;�
</p>
<p>)
</p>
<p>&frac14; &eth;2=p&THORN;1=2fE exp&eth;�E2=2&THORN;j10 &thorn; E exp&eth;�E2=2&THORN;j
1
1g
</p>
<p>&frac14; &eth;2=p&THORN;1=22e�1=2 &frac14; 0:968
</p>
<p>A range of parameters can be determined from the probability functions for the two distributions; a
</p>
<p>few of them are listed in Table 4.3.
</p>
<p>The centric distribution is addressed again through Problems 4.3 and 4.5.
</p>
<p>Table 4.3 Parameters in the
acentric and centric distributions
</p>
<p>of jEj values
Parameter Acentric Centric
</p>
<p>jEj 0.886 0.798
</p>
<p>jEj2 1 1
</p>
<p>jEj2 � 1 0.736 0.968
</p>
<p>&eth;jEj2 � 1&THORN;2 1 2
</p>
<p>4.2 Intensity Statistics 183</p>
<p/>
</div>
<div class="page"><p/>
<p>Cumulative Distributions
Rather than considering individual parameters, such asM or jEj, the determination of the centricity or
otherwise of the distribution may be approached by means of cumulative distributions of jEj values. In
the acentric distribution, the fractional number of jEj values less than or equal to a given value of jEj is
the integral of the probability function from the lower limit to that given value. Thus, we write for the
</p>
<p>acentric distribution
</p>
<p>Na&eth;jEj&THORN; &frac14; 2
&eth;jEj
</p>
<p>0
</p>
<p>jEj exp&eth;�jEj2&THORN; djEj &frac14; 1� exp&eth;�E2&THORN; (4.68)
</p>
<p>Similarly, for the centric distribution, we have
</p>
<p>Na&eth;jEj&THORN; &frac14; 2
&eth;jEj
</p>
<p>0
</p>
<p>jEj exp&eth;�jEj2&THORN; djEj
</p>
<p>&frac14; erf &eth;jEj=
ffiffiffi
2
</p>
<p>p
&THORN;
</p>
<p>(4.69)
</p>
<p>where erf(. . .) represents the statistical error function, which is tabulated in most texts on statistics.
</p>
<p>Table 4.4 list the values of N(jEj) for the two distributions up to jEj &frac14; 3.0, and Fig. 4.10 illustrates
these distributions; the region of greatest discrimination is clearly 0&lt;jEj&lt;1.
</p>
<p>Hypersymmetry
Hypersymmetry (hyper-centrosymmetry) can arise when non-crystallographic centers of symmetry
</p>
<p>are present in the asymmetric unit of a structure. Pyrene [11] and benzo[a]pyrene [12] are examples
</p>
<p>of molecules that are, themselves, centrosymmetric; the crystal structure of pyrene has been reported
</p>
<p>in space group P21/a, which is a non-standard setting of P21/c. The degree of hypersymmetry depends
</p>
<p>upon the number of additional centers of symmetry. Figure 4.10 includes the curve for Nh&eth;jEj&THORN; when
one additional center is present in the asymmetric unit. Further discussions on hypersymmetry may be
</p>
<p>found in the literature [13].
</p>
<p>Table 4.4 Acentric and centric
cumulative distributions
</p>
<p>jEj Na(jEj) Nc(jEj)
0 0 0
</p>
<p>0.2 0.039 0.159
</p>
<p>0.4 0.148 0.311
</p>
<p>0.6 0.302 0.451
</p>
<p>0.8 0.473 0.576
</p>
<p>1.0 0.632 0.683
</p>
<p>1.2 0.763 0.770
</p>
<p>1.4 0.859 0.838
</p>
<p>1.6 0.923 0.890
</p>
<p>1.8 0.961 0.928
</p>
<p>2.0 0.982 0.954
</p>
<p>2.2 0.992 0.972
</p>
<p>2.4 0.997 0.984
</p>
<p>2.6 0.999 0.991
</p>
<p>2.8 1.000 0.995
</p>
<p>3.0 1.000 0.997
</p>
<p>184 4 Intensities and Intensity Statistics</p>
<p/>
</div>
<div class="page"><p/>
<p>4.3 Problems
</p>
<p>4.1. An organic crystal has a large overall isotropic temperature factor of 6.8 Å2. What is the
</p>
<p>percentage reduction of the atomic scattering factor of a carbon atom at room temperature for
</p>
<p>a reflection at y &frac14; 27.55�, with Cu Ka radiation (l &frac14; 1.5418 Å) compared to that for a carbon
</p>
<p>atom at rest scattering under the same conditions? What is the root mean square amplitude of
</p>
<p>vibration of the atom in a direction normal to the given reflecting plane? How might the data
</p>
<p>collection process for this crystal be improved?
</p>
<p>4.2. Calculate the ideal intensities, jFj2, for the 111 and 222 reflections for NaCl and KCl. Hence,
discuss these reflections in the light of Fig. 12.10. The necessary data are as follows: a (NaCl)
</p>
<p>&frac14; 5.627 Å, a (KCl) &frac14; 6.278; f values may be calculated from the equationP4
1 ai exp&eth;�bis2&THORN; &thorn; ci, where s &frac14; (sin y)/l, and the constants for the equation are listed below.
</p>
<p>Fig. 4.10 Cumulative distributions N(jEj). The region 0 &lt; jEj &lt; 1 is the most discriminatory, as the curves tend to
converge and actually cross over at higher values of jEj
</p>
<p>(sin y)/l fC
</p>
<p>0 6
</p>
<p>0.1 5.108
</p>
<p>0.2 3.560
</p>
<p>0.3 2.494
</p>
<p>0.4 1.948
</p>
<p>0.5 1.686
</p>
<p>a1 b1 a2 b2 a3 b3 a4 b4 c
</p>
<p>Na+ 3.2565 2.6671 3.9362 6.1153 1.3998 0.2001 1.0032 14.0390 0.4040
</p>
<p>K+ 7.9578 12.6331 7.4917 0.7674 6.3590 �0.0020 1.1915 31.9128 �4.9978
Cl� 18.2915 0.0066 7.2084 1.1717 6.5337 19.5424 2.3386 60.4486 �16.3780
</p>
<p>4.3 Problems 185</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_12#Fig10_12">http://dx.doi.org/10.1007/978-1-4614-3954-7_12#Fig10_12</a></div>
</div>
<div class="page"><p/>
<p>4.3. Show that, in the centric distribution, Mc&eth;&frac14; F2=F2&THORN; &frac14; 0:637.
4.4. The value of jEj3 for the acentric distribution is 1.329. Find the value of jEj3 for the centric
</p>
<p>distribution.
</p>
<p>4.5. The value of jE2 � 1j in the centric distribution has been shown to be 0.968. Find
the corresponding value in the acentric distribution.
</p>
<p>4.6. Compare the statistically distinguishable features of space groups Pm, P2, and P2/m and work
</p>
<p>out a scheme for distinguishing between them.
</p>
<p>4.7. How may one distinguish between the space groups that have the diffraction symbols (a)
</p>
<p>mmm Pc � � and (b) mmm C � � � ?
</p>
<p>References
</p>
<p>1. See Bibliography, James (1958) International tables for X-ray crystallography, vol C. 2nd edn. (1999)
</p>
<p>2. Zachariasen WH (1967) Acta Crystallogr 23:558
</p>
<p>3. Busing WR, Levy HA (1957) Acta Crystallogr 10:180
</p>
<p>4. Blessing RH (1995) Acta Crystallogr A51:33; idem J Appl Crystallogr 30:421 (1997)
</p>
<p>5. Tickle IJ (1975) Acta Crystallogr B31:329
</p>
<p>6. Wilson AJC (1942) Nature 150:152
</p>
<p>7. See Bibliography, Ibers and Hamilton (1974)
</p>
<p>8. Hall SR, Subramanian V (1982) Acta Crystallogr A38:590; idem ibid A38:598 (1982)
</p>
<p>9. http://xtal.source.forge.net/man/genev-desc.html# 2776482
</p>
<p>10. French S, Wilson K (1978) Acta Crystallogr A34:517
</p>
<p>11. Robertson JM, White JG (1947) J Chem Soc 358
</p>
<p>12. Carrell CJ et al (1997) Carcinogenesis 18:415
</p>
<p>13. Rogers D, Wilson AJC (1953) Acta Crystallogr 6:439
</p>
<p>Bibliography: Intensity and Intensity Statistics
</p>
<p>James RW (1958) Optical principles of the diffraction of X-rays: the crystalline state, vol 2. Bell, London
</p>
<p>Prince E, Wilson AJC (1999) International tables for X-ray crystallography, vol C, 2nd edn. Kluwer, London
</p>
<p>Woolfson MM (1977) An introduction to X-ray crystallography, 2nd edn. Cambridge University Press, Cambridge
</p>
<p>186 4 Intensities and Intensity Statistics</p>
<p/>
<div class="annotation"><a href="http://xtal.source.forge.net/man/genev-desc.html#2776482">http://xtal.source.forge.net/man/genev-desc.html#2776482</a></div>
</div>
<div class="page"><p/>
<p>Examination of Single Crystals: Optical
and X-Ray Diffraction Practice 5
</p>
<p>5.1 Introduction
</p>
<p>The preliminary optical examination of crystalline specimens is interesting and useful in its own right
</p>
<p>and is a major tool still employed by mineralogists and geologists. However, in structure determina-
</p>
<p>tions with modern equipment, it is not uncommon nowadays to by-pass this step and proceed
</p>
<p>immediately with X-ray studies. This is because in most cases, the X-ray technique is straightforward
</p>
<p>and data can be quickly scanned with a single-crystal X-ray diffractometer, Sects. 5.5 and 5.6, or with
</p>
<p>an area detector (see Sect. 5.7), and the suitability and quality of the crystal assessed. There are other
</p>
<p>situations, however, where complications may arise, for example, because of an unusual crystal habit,
</p>
<p>Sect. 5.3.5, pseudosymmetry, Sects. 7.2.2, 7.5.4, and Sect. 8.5.3, or twinning, Sect. 5.10. In such
</p>
<p>cases, it might be possible to extract useful information from an optical examination of a crystal
</p>
<p>before the more detailed, costly and time-consuming X-ray methods are tried.
</p>
<p>5.2 Crystal Growing
</p>
<p>It is frequently the case that the crystals supplied for crystallographic work are not suitable for
</p>
<p>immediate use; then, it is necessary to recrystallize the material in order to obtain specimens
</p>
<p>appropriate for both X-ray and optical studies.
</p>
<p>5.2.1 Growing Crystals for X-Ray Diffraction
</p>
<p>In Sect. 10.2ff we give considerable space to the discussion of growing crystals of proteins because
</p>
<p>this specialized and important process can be difficult. It is, however, equally important to obtain
</p>
<p>good crystals of inorganic and organic smaller-molecule materials, as the quality of the crystal will
</p>
<p>influence both the value of an optical investigation and that of the X-ray diffraction data and, thus, of
</p>
<p>the structure determination itself. Crystals used for X-ray diffraction are commonly of size ranging
</p>
<p>between 0.1 and 0.5 mm, and crystals of such dimensions are generally obtained by a relatively slow
</p>
<p>growth process. The slow growth favors the production of crystals that are not twinned (q.v.) or
</p>
<p>disordered (q.v.), and from which occluded substances, such as dust or solvent molecules are absent,
</p>
<p>except where the solvent is germane to the crystal structure. There are several techniques that are
</p>
<p>M. Ladd and R. Palmer, Structure Determination by X-ray Crystallography:
</p>
<p>Analysis by X-rays and Neutrons, DOI 10.1007/978-1-4614-3954-7_5,
# Springer Science+Business Media New York 2013
</p>
<p>187</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_7#Sec4_7">http://dx.doi.org/10.1007/978-1-4614-3954-7_7#Sec4_7</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_7#Sec26_7">http://dx.doi.org/10.1007/978-1-4614-3954-7_7#Sec26_7</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_8#Sec38_8">http://dx.doi.org/10.1007/978-1-4614-3954-7_8#Sec38_8</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_10#Sec3_10">http://dx.doi.org/10.1007/978-1-4614-3954-7_10#Sec3_10</a></div>
</div>
<div class="page"><p/>
<p>employed in general crystal growing. It is not possible to give strict directions and it is often necessary
</p>
<p>to vary the conditions of growth in order to obtain satisfactory crystals.
</p>
<p>5.2.2 Crystallization from Solution
</p>
<p>In this method a suitable solvent must be found, and it is useful to remember that many inorganic
</p>
<p>materials and organic material containing polar functional groups, such as &ndash;OH or &ndash;CO2H, for
</p>
<p>example, will dissolve in water or ethanol. Non-polar molecules frequently dissolve in solvents
</p>
<p>such as benzene or toluene. In some cases, mixed solvents are an asset. For example, steroid-like
</p>
<p>substances can often be recrystallized well from a mixture of methanol and dichloromethane.
</p>
<p>Sufficient material to form a near-saturated solution may be dissolved in warm solvent in a clean
</p>
<p>glass vessel protected from direct exposure to the surroundings and allowed to cool slowly. Slow
</p>
<p>cooling may be achieved by suspending the crystallization vessel in a Dewar flask or by using a
</p>
<p>thermal reservoir. Many variations on this procedure are possible.
</p>
<p>Crystallization from solution is essentially a process of concentration of the solution until crystals
</p>
<p>begin to form either by cooling a solution of the substance or by slow room-temperature evaporation.
</p>
<p>If the vessel in which the crystals are to be grown is very smooth-walled, crystallization may need to
</p>
<p>be induced by rubbing the wall of the vessel with a clean glass rod.
</p>
<p>5.2.3 Crystallization by Diffusion
</p>
<p>Another technique for bringing about crystallization is by diffusion into the solution of a miscible
</p>
<p>solvent in which the crystals have a small solubility. Thus, &ldquo;ibuprofen,&rdquo; a-methyl-4-(2-methylpropyl)
</p>
<p>benzeneaceticacid (C12H18O2), can be crystallized by diffusion from ethanol/diethylether, and
</p>
<p>ammonium nitrate (NH3NO3) can be crystallized by diffusing ethanol into its aqueous solution.
</p>
<p>Crystallization by diffusion may be brought about also by introducing the vapor of the solvent in
</p>
<p>which the substance is least soluble above the surface of the solution of the substance in a suitable
</p>
<p>vessel. This technique and the related hanging-drop method are discussed in Sect. 10.2.9, and are not
</p>
<p>restricted to proteins.
</p>
<p>5.2.4 Crystallization by Sublimation
</p>
<p>Some materials are suitable for crystallization by sublimation; iodine, I2, and menthol (C10H20O) are
</p>
<p>substances that can be treated in this manner. In principle, the material is warmed and the vapor
</p>
<p>allowed to impinge on a cool surface; it can sometimes be advantageous for the material to be under
</p>
<p>reduced pressure. Not surprisingly, the size of the crystals is governed by the rate of sublimation.
</p>
<p>5.2.5 Other Issues
</p>
<p>There are materials, particularly inorganic and refractory substances and polymers that cannot be
</p>
<p>formed into single crystals. They exist as microcrystalline powders of size in the region of 10�3 mm,
and can rarely be recrystallized to larger dimensions. For this reason, we treat structure determination
</p>
<p>from powder materials in Chap. 12, where we show that considerable success has been achieved with
</p>
<p>this X-ray technique.
</p>
<p>188 5 Examination of Single Crystals: Optical and X-Ray Diffraction Practice</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_10#Sec12_10">http://dx.doi.org/10.1007/978-1-4614-3954-7_10#Sec12_10</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_12">http://dx.doi.org/10.1007/978-1-4614-3954-7_12</a></div>
</div>
<div class="page"><p/>
<p>5.3 Optical Techniques
</p>
<p>We discuss first some aspects of the interaction between crystals and two different electromagnetic
</p>
<p>radiations, light and X-rays. Light, with its longer wavelength (5000&ndash;6000 Å) can reveal only limited
</p>
<p>information about crystal structures, whereas X-rays with wavelengths of less than about 2 Å can be
</p>
<p>used to determine the relative positions of atoms in crystals. A preliminary examination of a crystal
</p>
<p>aims to determine its space group and unit-cell dimensions, and is generally carried out by X-ray
</p>
<p>techniques. However, the optical methods described here are simple but often very effective; they
</p>
<p>should be regarded as a desirable prerequisite to an X-ray structure determination, particularly where
</p>
<p>automated intensity measurement is used.
</p>
<p>5.3.1 Polarized Light
</p>
<p>An ordinary light source emits wave trains, or pulses of light, vibrating in all directions perpendicular
</p>
<p>to the direction of propagation, as shown in Fig. 5.1; the light is said to be unpolarized. The vibrations
</p>
<p>of interest to us are those of the electric vector associated with the waves. Any one of these random
</p>
<p>vibrations can be resolved into two mutually perpendicular components, and the resultant vibration
</p>
<p>may, therefore, be considered as the sum of all components in these two perpendicular directions. In
</p>
<p>order to study the optical properties of crystals, we need to restrict the resultant vibration of the light
</p>
<p>source to one direction only by eliminating the component at right angles to it.
</p>
<p>A polarizer (P), consisting of a sheet of Polaroid, transmits light vibrating in the horizontal
</p>
<p>direction LM and absorbs all components vibrating in the direction perpendicular to LM. Thus,
</p>
<p>light passing through the polarizer vibrates in one plane only, and is said to be plane-polarized.
</p>
<p>The plane contains the vibration direction that is perpendicular to the direction of propagation and to
</p>
<p>the direction of propagation itself. A second Polaroid, the analyzer (A), is placed after the polarizer
</p>
<p>and rotated so that its vibration transmission direction (MN) is at 90� to that of the polarizer. It
</p>
<p>receives no component parallel to its transmission direction and, therefore, absorbs all the light
</p>
<p>transmitted by the polarizer. The two Polaroids are then said to be crossed. This effect may be
</p>
<p>demonstrated by cutting a Polaroid sheet marked with a straight line LMN into two sections, P and A.
</p>
<p>When superimposed, the two halves will not transmit light if the reference lines (vibration directions)
</p>
<p>LM and MN are exactly perpendicular, or crossed. In intermediate positions, the intensity of light
</p>
<p>Fig. 5.1 Production of plane-polarized light by passing unpolarized light through a sheet of Polaroid film (the
polarizer, P). A second, identical sheet of Polaroid (the analyzer, A), rotated through 90� with respect to P, completely
</p>
<p>absorbs all light transmitted by P. The lines LM and MN were parallel on the sheet from which P and A were cut
</p>
<p>5.3 Optical Techniques 189</p>
<p/>
</div>
<div class="page"><p/>
<p>transmitted varies from a maximum, where they are parallel, to zero (in the crossed position). The
</p>
<p>production and use of plane-polarized light by this method is used in the polarizing microscope.
</p>
<p>5.3.2 Optical Classification of Crystals
</p>
<p>Crystals may be grouped, optically, under two main headings: isotropic crystals and anisotropic, or
</p>
<p>birefringent, crystals. All crystals belonging to the cubic system are optically isotropic: the refractive
</p>
<p>index of a cubic crystal is independent of the direction of the light incident upon it, and its optical
</p>
<p>characteristics are similar to those of glass. Non-cubic crystals exhibit a dependence on direction in
</p>
<p>their interaction with light.
</p>
<p>Anisotropic crystals are divided into two groups: uniaxial crystals, which have one optically
</p>
<p>isotropic section and include the tetragonal, hexagonal, and trigonal crystal systems, and biaxial
</p>
<p>crystals, which have two optically isotropic sections and belong to the orthorhombic, monoclinic, or
</p>
<p>triclinic crystal systems.
</p>
<p>A preliminary optical examination of a crystal will usually show whether it is isotropic, uniaxial,
</p>
<p>or biaxial. Distinction between the three biaxial crystal systems is often possible in practice and,
</p>
<p>depending on how well developed the crystals are, a similar differentiation may also be effected for
</p>
<p>the uniaxial crystals. Even if an unambiguous determination of the crystal system is not forthcom-
</p>
<p>ing, the examination should, at least, enable the principal symmetry directions to be identified;
</p>
<p>Table 5.1 summarizes this information.
</p>
<p>5.3.3 Uniaxial Crystals
</p>
<p>As an example of the use of the polarizing microscope, we consider a tetragonal crystal, such as
</p>
<p>potassium dihydrogen phosphate (KH2PO4), lying on a microscope slide with its y axis parallel to the
</p>
<p>axis of the optical path through a microscope, illustrated in Fig. 5.2. The microscope is fitted with a
</p>
<p>polarizer (P), and an analyzer (A) which is crossed with respect to P and may be removed from the
</p>
<p>optical path. The crystal can be rotated on the microscope stage between P and A. With the Polaroids
</p>
<p>crossed and no crystal in between, the field of view is uniformly dark. However, with the crystal
</p>
<p>interposed, this situation will not necessarily be obtained.
</p>
<p>The tetragonal crystal is lying with its (010) plane on the microscope slide; both the x and z axes are,
</p>
<p>therefore, perpendicular to the microscope axis. In general, some of the light passing through the crystal
</p>
<p>will be transmitted by the analyzer, even though P and A are crossed. The intensity of the transmitted
</p>
<p>light varies as the crystal is rotated on the microscope stage between the polarizer and the analyzer.
</p>
<p>During a complete revolution of the stage, the intensity of transmitted light passes through four maxima
</p>
<p>Table 5.1 Crystal directions readily derivable from an optical study
</p>
<p>Optical classification Crystal system Information relating to crystal axes likely to be revealed
</p>
<p>Isotropic Cubic Axes may be assigned from the crystal morphology
</p>
<p>Anisotropic, uniaxial Tetragonal Direction of the z axis
Hexagonal Direction of the z axis
Trigonala Direction of the z axis
</p>
<p>Anisotropic, biaxial Orthorhombic Direction of at least the x, y, or z axis, possibly all three axes
Monoclinic Direction parallel to the y axis
Triclinic No special relationship between the crystal axes and the
</p>
<p>vibration directions
</p>
<p>aReferred to hexagonal axes
</p>
<p>190 5 Examination of Single Crystals: Optical and X-Ray Diffraction Practice</p>
<p/>
</div>
<div class="page"><p/>
<p>and four minima. At the minimum positions, the crystal is usually only just visible. These positions are
</p>
<p>called extinction positions, and they occur at exactly 90� intervals of rotation. Maximum intensity is
</p>
<p>observed with the crystal at 45� to these directions.
</p>
<p>These changes would be observed if the crystal itself were replaced by a sheet of Polaroid.
</p>
<p>Extinction would occur when the vibrations of the &ldquo;crystal Polaroid&rdquo; were perpendicular to those
</p>
<p>of P or A. A simple explanation of these effects is that the crystal behaves as a polarizer. Incident
</p>
<p>plane-polarized light from P is resolved by the crystal into two perpendicular components, Fig. 5.3.
</p>
<p>In the tetragonal crystal, the vibration directions associated with this polarizing effect are parallel to
</p>
<p>its x and z axes. Rotating the crystal on the microscope stage will, therefore, produce extinction
</p>
<p>whenever x and z are parallel to the vibration directions of P and A. The x and z axes of a tetragonal
</p>
<p>Fig. 5.2 Schematic
experimental arrangement
</p>
<p>for examining extinction
</p>
<p>directions. A tetragonal
</p>
<p>crystal is shown on the
</p>
<p>microscope stage, and the
</p>
<p>incident light is
</p>
<p>perpendicular to the z axis
</p>
<p>of the crystal
</p>
<p>5.3 Optical Techniques 191</p>
<p/>
</div>
<div class="page"><p/>
<p>crystal correspond to its extinction directions: it should be remembered that the x and y directions are
</p>
<p>equivalent under the fourfold symmetry of the crystal.
</p>
<p>5.3.4 Birefringence
</p>
<p>The vibration components produced by a crystal are associated with its different refractive indices.
</p>
<p>With reference to Fig. 5.3, a tetragonal crystal in light vibrating parallel to the fourfold symmetry axis
</p>
<p>(z) has a refractive index e, whereas light vibrating perpendicular to z has a different refractive index,
</p>
<p>o; the crystal is said to be birefringent, or optically anisotropic. Figure 5.4 represents plane-polarized
</p>
<p>light incident in a general direction with respect to the crystallographic axes. It is resolved into two
</p>
<p>components, one with an associated refractive index o and the other with an associated refractive
</p>
<p>index e0, both vibrating perpendicular to each other and to the direction of incidence. In general, the
value of e0 lies between those of o and e.
</p>
<p>Two special cases arise: one, already discussed, where the incident light is perpendicular to z, for
</p>
<p>which e0 &frac14; e, and the other where the incident light is parallel to z, for which e0 &frac14; o. It follows that
where the direction of incidence is parallel to the z axis, the refractive index is always o for any
</p>
<p>vibration direction in the xy plane. Plane-polarized light incident parallel to the z axis will pass through
</p>
<p>the crystal unmodified. In this particular direction, the crystal is optically isotropic, and if rotated on
</p>
<p>the microscope stage between crossed Polaroids, it remains in extinction. The z direction of a uniaxial
</p>
<p>crystal is called the optic axis, and there is only one such direction in the crystal; it is the fourfold
</p>
<p>symmetry axis in the example that we are using.
</p>
<p>Identification of the z Axis of a Uniaxial Crystal
A polarizing microscope is usually fitted with eyepiece cross-wires arranged parallel and perpendic-
</p>
<p>ular to the vibration directions of the polarizer, and therefore we can relate the crystal vibration
</p>
<p>Fig. 5.3 Resolution of
incident light into
</p>
<p>components vibrating
</p>
<p>parallel to the x and z axes
</p>
<p>of a tetragonal crystal lying
</p>
<p>with its y axis parallel to the
</p>
<p>incident beam; o and e are
</p>
<p>the refractive indices for
</p>
<p>light vibrating,
</p>
<p>respectively, perpendicular
</p>
<p>and parallel to z
</p>
<p>192 5 Examination of Single Crystals: Optical and X-Ray Diffraction Practice</p>
<p/>
</div>
<div class="page"><p/>
<p>directions to its morphology. There are two important optical orientations for a tetragonal crystal,
</p>
<p>namely with the z axis either perpendicular or parallel to the axis of the microscope. These orienta-
</p>
<p>tions are, in fact, important for all uniaxial crystals, and will be described in more detail.
</p>
<p>z Axis Perpendicular to the Microscope Axis
</p>
<p>In this position, shown by Fig. 5.5, a birefringent orientation is always presented to the incident light
</p>
<p>beam. Extinction will occur whenever the z axis is parallel to a cross-wire, no matter how the crystal is
</p>
<p>rotated, or flipped over, while keeping z normal to the microscope axis. The success of this operation
</p>
<p>depends to a large extent on having a crystal with well-developed (hk0) faces. The term straight
</p>
<p>extinction is used to indicate that the field of view is dark when a crystal edge is aligned with a cross-
</p>
<p>wire. A face of a uniaxial crystal for which one edge is parallel to z, an (hk0) face or to its trace on a
</p>
<p>crystal face, such as (h0l), will show straight extinction.
</p>
<p>z Axis Parallel to the Microscope Axis
</p>
<p>The crystal now presents an isotropic section to the incident light beam, and will remain extinguished
</p>
<p>for all rotations of the crystal, while keeping z along the microscope axis. A reasonably thin section of
</p>
<p>the crystal is required in order to observe this effect. Because of the needle-shaped habit of the crystal,
</p>
<p>it is necessary to cut the crystal carefully so as to obtain the desired specimen, Sect. 5.10.1.
</p>
<p>The section of a uniaxial crystal normal to the z axis, if well developed, may provide a clue to the
</p>
<p>crystal system. Tetragonal crystals often have edges at 90� to one another, whereas hexagonal and
</p>
<p>Fig. 5.4 Uniaxial crystal
showing a light ray OP
</p>
<p>resolved into two
</p>
<p>components. One
</p>
<p>component, with refractive
</p>
<p>index o, vibrates in the xy
</p>
<p>plane, the other, with
</p>
<p>refractive index e0, vibrates
</p>
<p>parallel to both o and the
</p>
<p>ray direction
</p>
<p>5.3 Optical Techniques 193</p>
<p/>
</div>
<div class="page"><p/>
<p>trigonal crystals often exhibit edges at 60 or 120� to one another. These angles are external
</p>
<p>manifestations of the internal symmetry; idealized uniaxial crystal sections are shown in Fig. 5.6.
</p>
<p>5.3.5 Biaxial Crystals
</p>
<p>Biaxial crystals have two optic axes and, correspondingly, two isotropic directions. The reason for
</p>
<p>this effect lies in the low symmetry associated with the orthorhombic, monoclinic, and triclinic
</p>
<p>systems, which, in turn, results in less symmetric optical characteristics. Biaxial crystals have three
</p>
<p>principal refractive indices, n1, n2, and n3 (normally, n1 &lt; n2 &lt; n3), associated with light vibrating
parallel to three mutually perpendicular directions in the crystal. The optic axes that derive from this
</p>
<p>property are not necessarily directly related to the crystallographic axes. We shall not concern
</p>
<p>ourselves here with a detailed treatment of the optical properties of biaxial crystals, but will
</p>
<p>concentrate on relating the extinction directions to the crystal symmetry.
</p>
<p>Orthorhombic Crystals
In the orthorhombic system, the vibration directions associated with n1, n2, and n3 are parallel to the
</p>
<p>crystallographic axes, but any combination of x, y, and z with n1, n2, and n3may occur. Consequently,
</p>
<p>recognition of the extinction directions facilitates identification of the directions of the crystallo-
</p>
<p>graphic axes. For a crystal with x, y, or z perpendicular to the microscope axis, the extinction,
</p>
<p>directions will be either parallel or perpendicular to the axis in question, as illustrated in Fig. 5.7.
</p>
<p>Fig. 5.5 Extinction position for a tetragonal crystal lying with its z axis parallel to the microscope slide. Any [UV 0]
direction could be parallel to the microscope axis; extinction will always be straight with respect to the z axis or its trace
</p>
<p>Fig. 5.6 Idealized uniaxial crystals as seen along the z axis. (a) Tetragonal. (b) Hexagonal. (c) Trigonal. The refractive
index for light vibrating perpendicular to the z axis is given the symbol o, and the crystals appear isotropic in this
</p>
<p>orientation
</p>
<p>194 5 Examination of Single Crystals: Optical and X-Ray Diffraction Practice</p>
<p/>
</div>
<div class="page"><p/>
<p>If the crystal is a well-developed orthorhombic prism, the three crystallographic axes may be
</p>
<p>identified by this optical method. A common alternative habit of orthorhombic crystals has one
</p>
<p>axis, x, for example, as a needle axis with the {011} form prominent. The appearance of such a crystal
</p>
<p>viewed along x is shown in Fig. 5.8, and is an example of a symmetric extinction.
</p>
<p>Monoclinic Crystals
The lower symmetry of monoclinic crystals results in a corresponding modification of the optical
</p>
<p>properties in this system. The symmetry axis y is, conventionally, set parallel to one of the vibration
</p>
<p>directions; x and z are related arbitrarily to the other two mutually perpendicular vibration directions.
</p>
<p>Hence, two directions are of importance in monoclinic crystals, namely, perpendicular to and parallel
</p>
<p>to the y axis.
</p>
<p>When viewed between crossed Polaroids, a monoclinic crystal lying with its y axis perpendicular
</p>
<p>to the microscope axis will always show straight extinction, with the cross-wires parallel (and
</p>
<p>perpendicular) to y. Often, the y axis is a well-developed needle axis; rotation of the crystal about
</p>
<p>this axis while keeping it perpendicular to the microscope axis will not cause any change in the
</p>
<p>position of extinction, Fig. 5.9.
</p>
<p>Fig. 5.7 Extinction directions in an orthorhombic crystal viewed along the x, y, or z axis
</p>
<p>Fig. 5.8 Extinction directions as seen along the x-axis of an orthorhombic crystal with {011} development: an
example of symmetric extinction
</p>
<p>5.3 Optical Techniques 195</p>
<p/>
</div>
<div class="page"><p/>
<p>If the monoclinic crystal is arranged so that y is parallel to the microscope axis, the (010) plane will
</p>
<p>lie on the microscope slide. Extinction in this position will, in general, be oblique, as shown in
</p>
<p>Fig. 5.10, thus giving further evidence for the position of the y-axis direction. The appearance of
</p>
<p>extinction in a monoclinic crystal in this orientation may be somewhat similar to that of an
</p>
<p>orthorhombic crystal showing prominent {011} development, compare Figs. 5.8 and 5.10, and
</p>
<p>confusion may sometimes occur in practice.
</p>
<p>Triclinic Crystals
The mutually perpendicular vibration directions associated with n1, n2, and n3 are arbitrarily related to
</p>
<p>the crystallographic axes, which are selected initially from morphological and X-ray studies.
</p>
<p>Reference to Table 5.1 should now enable the reader to consolidate the ideas presented in the
</p>
<p>discussion of extinction directions in the seven crystal systems. Although it gives only limited
</p>
<p>Fig. 5.9 Extinction directions in a monoclinic crystal viewed perpendicular to the y axis: an example of straight
extinction
</p>
<p>Fig. 5.10 Extinction directions in a monoclinic crystal viewed along the y axis&mdash;an example of oblique extinction. If
the forms developed are pinacoids, that is, {100}, {010}, and {001}, then the extinction angle; the angle between a
</p>
<p>crystal edge and a cross-wire will be related in a simple way to the b angle
</p>
<p>196 5 Examination of Single Crystals: Optical and X-Ray Diffraction Practice</p>
<p/>
</div>
<div class="page"><p/>
<p>information on the optical properties of crystals, a practical study of a crystal along these lines can
</p>
<p>often provide useful information about both its system and its axial directions. Further details on the
</p>
<p>optical properties of crystals may be found in standard works on this subject [1].
</p>
<p>5.4 Single-Crystal X-Ray Diffraction Techniques: Intensity Data Collection
</p>
<p>With the development of sophisticated X-ray diffractometers, there has been a distinct move away
</p>
<p>from traditional photographic methods of recording diffraction patterns of crystals, although they still
</p>
<p>have a value for displaying a picture of the reciprocal lattice, and for investigating the symmetry or
</p>
<p>perfection of crystals. The introduction of various types of area detectors swings methodology away
</p>
<p>from the serial, &ldquo;one reflection at a time,&rdquo; diffractometer toward image-plate principles, reminiscent of
</p>
<p>the earlier photographic techniques. An understanding of Laue and X-ray oscillation photography and
</p>
<p>the related geometry is invaluable to those taking the almost inevitable step of progressing from in-
</p>
<p>house facilities to the use of the synchrotron. The precession camera [2, 3] is sadly no longer used per
</p>
<p>se but was historically a brilliant innovation that allowed undistorted records of the weighted
</p>
<p>reciprocal lattice, Sect. 3.4.2, to be produced as a matter of routine. Figure 5.11a, b show zero-level
</p>
<p>and first-level precession photographs of the orthorhombic crystal of trypsin. Such records can
</p>
<p>however now be produced from intensity data files using software such as that of Bruker AXS
</p>
<p>(XPREP/RLATT) [4] thus preserving the art of recording the undistorted reciprocal lattice, Fig. 5.11c.
</p>
<p>Over the past 15 years, the role of powder crystallography has been transformed from one of very
</p>
<p>minor use in structure analysis to one in which it is rapidly becoming a major alternative choice.
</p>
<p>Chap. 12 provides details of the advances that have taken place to revolutionize this method, while
</p>
<p>the present chapter discusses the interpretation of the X-ray diffraction data recording techniques in
</p>
<p>current use for single crystals and methods for the n data.
</p>
<p>5.4.1 Laue Method
</p>
<p>The three variables in the Bragg equation provide a basis for the interpretation of X-ray crystallo-
</p>
<p>graphic experiments. In the Laue method, Fig. 5.12, the Bragg equation is satisfied by effectively
</p>
<p>varying l, utilizing a beam of continuous (white) radiation. Since the crystal is stationary with respect
</p>
<p>to the X-ray beam, it acts as a sort of filter, selecting the correct wavelength for each reflection
</p>
<p>according to the Bragg equation.
</p>
<p>The diffraction spots on a Laue photograph lie on ellipses, all of which have one end of their major
</p>
<p>axis at the center of the photographic film, Fig. 5.13. All spots on one ellipse arise through reflections
</p>
<p>from planes that lie in one and the same zone. In Fig. 5.14, a zone axis for a given Bragg angle y is
</p>
<p>represented by ZZ0. A reflected ray is labeled R, and we can simulate the effect of the zone by
imagining the crystal to be rotated about ZZ0, taking the reflected beam with it. Rays such as R,
generate a cone, coaxial with ZZ0 and with a semivertical angle y. The lower limit, in the diagram, of
R is the direction (XY) of the X-ray beam, and the general intersection of a circle with a plane (the flat
</p>
<p>film) is an ellipse. Hence, we can understand the general appearance of the Laue photograph shown in
</p>
<p>Fig. 5.13. On each ellipse, discrete spots appear instead of continuous bands because only those
</p>
<p>orientations parallel to zone axes, such as ZZ0, that exist for actual crystal planes can give rise to X-ray
reflections.
</p>
<p>5.4 Single-Crystal X-Ray Diffraction Techniques: Intensity Data Collection 197</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec21_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec21_3</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_12">http://dx.doi.org/10.1007/978-1-4614-3954-7_12</a></div>
</div>
<div class="page"><p/>
<p>Fig. 5.11 Precession photographs. (a) A zero-level precession photograph of an orthorhombic protein crystal, trypsin:
a &frac14; 41.5444, b &frac14; 113.4716, c &frac14; 50.7156, space group P21212. (b) The first-level photograph of the same crystal.
(c) A simulated precession photograph of an orthorhombic crystal precessing about a (Cu Ka, l &frac14; 1.5418 Å).
The undistorted, weighted reciprocal lattice zero level is shown; it enables b*, c*, and a* to be measured directly
</p>
<p>from the film. The lengths b*and c* are magnified by the crystal-to-film distance of 60.00 mm, and the diagram has been
</p>
<p>scaled down by a factor of 0.415 (from the original photograph). By direct measurement on the film, 14b*and 14c* are
</p>
<p>67.9 and 68.5 mm, respectively. Hence, b� &frac14; &eth;67:9=0:415&THORN;=&eth;60� 14&THORN; &frac14; 0:1948. Similarly, c* &frac14; 0.1965. Thus,
a &frac14; 7.91 Å, b &frac14; 7.84 Å and g &frac14; 90�
</p>
<p>198 5 Examination of Single Crystals: Optical and X-Ray Diffraction Practice</p>
<p/>
</div>
<div class="page"><p/>
<p>Fig. 5.12 Schematic experimental arrangement for taking a Laue photograph on a flat-plate film
</p>
<p>Fig. 5.13 Sketch of a Laue photograph of a-Al2O3; the threefold symmetry direction is normal to the photograph
(along the X-ray beam)
</p>
<p>Fig. 5.14 Geometry of the Laue method: XY, X-ray beam direction; ZZ0, a zone axis; R, generator of the cone of
diffracted rays of semi-angle y, the Bragg angle; Y, central spot on the film and extremity of the major axis of the ellipse
</p>
<p>formed by the intersection of the cone with the film
</p>
<p>5.4 Single-Crystal X-Ray Diffraction Techniques: Intensity Data Collection 199</p>
<p/>
</div>
<div class="page"><p/>
<p>5.4.2 Symmetry in Laue Photographs
</p>
<p>One of the useful features of Laue photographs is the symmetry observable on them. The crystal
</p>
<p>orientation with respect to the X-ray beam is selected by the experimenter from morphological and
</p>
<p>optical considerations. This orientation, together with the crystal point group, controls the symmetry
</p>
<p>on the Laue photograph.
</p>
<p>In practice, a complication arises by the introduction of a center of symmetry into an X-ray
</p>
<p>diffraction pattern, in normal circumstances, whether or not the crystal is centrosymmetric. This
</p>
<p>situation is embodied in Friedel&rsquo;s law, the theoretical grounds for which have been discussed in
</p>
<p>Sect. 3.6.1. As a result of this law, the diffraction pattern may not reveal the true point-group
</p>
<p>symmetry of a crystal. It may be recalled that Table 1.6 shows the classification of the 32 crystallo-
</p>
<p>graphic point groups according to Laue diffraction symmetry.
</p>
<p>It cannot be over emphasized that the Laue group assigned to a crystal describes the symmetry of
</p>
<p>the complete X-ray diffraction pattern from that crystal. No single X-ray photograph can exhibit the
</p>
<p>complete diffraction symmetry, only that of a selected portion which is a projection on to the film,
</p>
<p>along the direction of the X-ray beam, of the symmetry information that would be encountered in that
</p>
<p>direction in a crystal having the Laue group of the given crystal.
</p>
<p>It follows that in the triclinic system, no symmetry higher than 1 is ever observable on a Laue
</p>
<p>photograph. In other crystal systems, the Laue-projection symmetry depends on the orientation of the
</p>
<p>crystal with respect to the X-ray beam. Rotation axes of any order reveal their true symmetry when
</p>
<p>the X-ray beam is parallel to the symmetry axis. Even-order rotation axes, 2, 4, or 6, give rise to
</p>
<p>mirror diffraction symmetry in the plane normal to the rotation axis when the X-ray beam is normal to
</p>
<p>that axis. A mirror plane itself shows m symmetry parallel to the mirror plane when the X-ray beam is
</p>
<p>contained by the plane. Various combinations of these effects may be observable, depending upon the
</p>
<p>Laue group in question.
</p>
<p>The nature of the X-ray results, supplementary to those obtained in the optical examination, should
</p>
<p>now be evident. Uniaxial crystals can be allocated to their correct systems by a Laue photograph
</p>
<p>taken with the X-ray beam along the z axis. The Laue photograph in Fig. 5.13 exhibits the Laue
</p>
<p>symmetry 3m. Distinction between the monoclinic and orthorhombic systems, which is not always
</p>
<p>possible in an optical examination, is fairly straightforward with Laue photographs, Table 1.6. Cubic
</p>
<p>crystals can exhibit a variety of symmetries, but with the X-ray beam along h100i the differentiation
between Laue groups m�3 and m�3m is clear.
</p>
<p>In practice, the symmetry pattern on a Laue photograph is very sensitive to the precise orientation
</p>
<p>of the crystal with respect to the X-ray beam [5]. Slight deviation from the ideal position will result in
</p>
<p>a distortion of the relative positions and intensities of the spots on the photographs.
</p>
<p>5.4.3 Laue Method and Synchrotron Radiation
</p>
<p>The synchrotron is an extremely powerful source of X-rays and produces a very wide range of
</p>
<p>wavelengths, Sect. 3.1.6, and is ideally suited to the Laue method of recording diffraction patterns.
</p>
<p>Since the crystal is in a fixed orientation, the angle of incidence of the X-ray beam is thus set for each
</p>
<p>(hkl) plane. For a reflection to take place at a preset y angle, the plane must effectively select the
</p>
<p>wavelength required to satisfy the Bragg equation. A reflection on a Laue photograph thus comprises
</p>
<p>four parameters, the usual hkl indices and the wavelength selected by the crystal.
</p>
<p>200 5 Examination of Single Crystals: Optical and X-Ray Diffraction Practice</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec25_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec25_3</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Tab6_1">http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Tab6_1</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Tab6_1">http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Tab6_1</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec7_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec7_3</a></div>
</div>
<div class="page"><p/>
<p>Consider a fixed crystal in a Laue diffraction experiment. For planes (hkl) the preset angle of
</p>
<p>incidence is y, and the wavelength required to be selected for the Bragg equation to be satisfied for the
</p>
<p>reflection hkl is l(hkl)y, given by
</p>
<p>l&eth;hkl&THORN;y &frac14; 2d&eth;hkl&THORN; sin y (5.1)
</p>
<p>For planes (2h, 2k, 2l), the wavelength to be selected is
</p>
<p>l&eth;2h; 2k; 2l&THORN;y &frac14; 2d&eth;2h; 2k; 2l&THORN; sin y (5.2)
</p>
<p>From Sect. 2.3,
</p>
<p>l&eth;2h; 2k; 2l&THORN;y &frac14; 2d&eth;hkl&THORN; sin y=2 &frac14; l&eth;hkl&THORN;y=2 (5.3)
</p>
<p>which is easily generalized to
</p>
<p>l&eth;nh; nk; nl&THORN;y &frac14; l&eth;hkl&THORN;y=n (5.4)
</p>
<p>Since the Bragg angle y is common to the sets of reflections (hkl, 2h, 2k, 2l, . . ., nh, nk, nl,
</p>
<p>n &frac14; 1, 2, 3, . . .) these sets will be superimposed on the Laue photograph. The extent of super-
imposition depends on the range of values of l available from the X-ray source, which may be
</p>
<p>selected to minimize this overlap of reflections.
</p>
<p>The interpretation of a Laue photograph may thus be complicated both by the possible existence of
</p>
<p>multiple reflection orders and by the need to assign the correct value of l to each I(hkl) prior to use in
</p>
<p>structure analysis. The latter requirement is, of course, necessary in view of the dependence of the
</p>
<p>atomic scattering factor f on both y and l. Other factors requiring special attention include sensitivity
</p>
<p>characteristics and absorption by the photographic film, both of which are wavelength-dependent.
</p>
<p>As an example, consider the following situation: an orthorhombic crystal with a &frac14; 10.0 Å,
b &frac14; 15.0 Å, c &frac14; 20.0 Å is mounted with c vertical and perpendicular to the X-ray beam, such that
bmakes an angle f of 30� with the beam direction in the horizontal plane. A diffraction spot P occurs
</p>
<p>on a flat-plate film such that its coordinates are X mm (horizontal) and Y mm (vertical), the plate
</p>
<p>being placed at a distance R mm from the crystal, Fig. 5.15, Y &frac14; 0. Since the wavelength is variable,
</p>
<p>Fig. 5.15 Geometry of Laue diffraction at the level Y &frac14; 0 on a film placed perpendicularly to the incident X-ray beam:
C, crystal; P, position of a Bragg reflection; 2y, scattering angle
</p>
<p>5.4 Single-Crystal X-Ray Diffraction Techniques: Intensity Data Collection 201</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_2#Sec6_2">http://dx.doi.org/10.1007/978-1-4614-3954-7_2#Sec6_2</a></div>
</div>
<div class="page"><p/>
<p>we define d*(hkl) from (2.11) with k &frac14; 1, which results in the Ewald sphere having a wavelength-
dependent radius of 1/l. Thus, if the extremes of l used in the experiment are lmin and lmax, the
</p>
<p>corresponding Ewald spheres will have radii 1/lmin and 1/lmax, Fig. 5.16. The reciprocal lattice in this
</p>
<p>treatment has fixed dimensions, a* &frac14; 1/a, b* &frac14; 1/b, and c* &frac14; 1/c. We can predict possible Laue
reflections at Y &frac14; 0 with the aid of Fig. 5.16. The a*b* reciprocal lattice net is shown rotated at 30� to
the X-ray beam, as described above. Traces of the outer and inner Ewald spheres, corresponding
</p>
<p>to the minimum and maximum wavelengths, contain a reciprocal lattice area which includes all
</p>
<p>reciprocal lattice points, shown intensified in the diagram, that are able to give rise to Laue reflections
</p>
<p>with an appropriate wavelength. The recording geometry relevant to the above experimental arrange-
</p>
<p>ment is shown in Fig. 5.15, from which it can be seen that for Y &frac14; 0 (corresponding to l &frac14; 0 in
</p>
<p>the example)
</p>
<p>tan 2y&eth;hk0&THORN; &frac14; X=R
</p>
<p>where X is the horizontal distance of spot P from the origin O, and R is the crystal&ndash;film constant.
</p>
<p>Reciprocal lattice points lying within the allowed region, Fig. 5.16, include 350, 470, 480, 490,. . .,
</p>
<p>313,0, for the limiting wavelengths lmin &frac14; 1.0 Å, lmax &frac14; 1.5 Å used in the example. We now
determine the coordinates of Laue diffraction spots and the wavelengths used in producing them.
</p>
<p>The wavelength selected for a particular d*(hk0) can be calculated with reference to Fig. 5.17.
</p>
<p>In this diagram
</p>
<p>Fig. 5.16 Reciprocal net a*b* for an orthorhombic crystal with a &frac14; 10 Å and b &frac14; 15 Å, tilted at f &frac14; 30� to the X-ray
beam; c is vertical. The wavelength range chosen is 1.0 Å (lmin) to 1.5 Å (lmax). Reciprocal lattice points that can give
</p>
<p>rise to diffraction spots on the right-hand side of the film (including the 350 and 470 used in the example calculations)
</p>
<p>and on the left-hand side have been emphasized. They lie between the limiting arcs of the Ewald spheres as shown.
</p>
<p>The range of hk0 reflections is limited also by the resolution limit d�max, which is governed by the crystal. For the
</p>
<p>reciprocal net chosen, a* &frac14; 1/a &frac14; 0.1 RU and b* &frac14; 1/b &frac14; 0.0667 RU; the scale is 1 RU &frac14; 3.52 cm
</p>
<p>202 5 Examination of Single Crystals: Optical and X-Ray Diffraction Practice</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_2#Eq11_2">http://dx.doi.org/10.1007/978-1-4614-3954-7_2#Eq11_2</a></div>
</div>
<div class="page"><p/>
<p>tan e &frac14; ha � =kb� &frac14; hb=ka
</p>
<p>and
</p>
<p>y&eth;hk0&THORN; &frac14; &frac12;90� &eth;f&thorn; e&THORN;�
</p>
<p>Thus,
</p>
<p>l&eth;hk0&THORN;y &frac14; 2d&eth;hk0&THORN; sin y&eth;hk0&THORN; &frac14; 2d&eth;hk0&THORN; sin&frac12;90� &eth;f&thorn; e&THORN;� &frac14; 2d&eth;hk0&THORN; cos&eth;f&thorn; e&THORN;
</p>
<p>Taking f &frac14; 30�:
1. Reflection 350
</p>
<p>tan e &frac14; 3� 15=&eth;5� 10&THORN; &frac14; 0:900 e &frac14; 41:99�
</p>
<p>d&eth;350&THORN; &frac14;
9
</p>
<p>102
&thorn;
</p>
<p>25
</p>
<p>152
</p>
<p>� ��1=2
&frac14; 2:2299A
</p>
<p>�
&frac14; 2:2299A; l&eth;350&THORN;y &frac14; 2� 2:2299 cos&eth;30&thorn; 41:99&THORN;
</p>
<p>&frac14; 1:3789A y&eth;350&THORN; &frac14; 90� &eth;30&thorn; 41:99&THORN; &frac14; 18:01�
</p>
<p>Hence for R &frac14; 60 mm
</p>
<p>X&eth;350&THORN; &frac14; 60 tan&eth;2� 18:01&THORN; &frac14; 43:62 mm
</p>
<p>Fig. 5.17 Geometry of
wavelength selection
</p>
<p>through a reciprocal lattice
</p>
<p>vector d* (hk0). The angle
</p>
<p>f defines the crystal
</p>
<p>orientation: C, crystal; O,
</p>
<p>origin of reciprocal lattice;
</p>
<p>CQ, direction of diffracted
</p>
<p>beam
</p>
<p>5.4 Single-Crystal X-Ray Diffraction Techniques: Intensity Data Collection 203</p>
<p/>
</div>
<div class="page"><p/>
<p>2. Reflection 470
</p>
<p>tan e &frac14; 4� 15=&eth;7� 10&THORN; &frac14; 0:8571 e &frac14; 40:60�
</p>
<p>d&eth;470&THORN; &frac14;
16
</p>
<p>102
&thorn;
</p>
<p>49
</p>
<p>152
</p>
<p>� ��1=2
&frac14; 1:6270A
</p>
<p>�
l&eth;470&THORN;y &frac14; 2� 1:6270 cos&eth;30&thorn; 40:60&THORN;
</p>
<p>&frac14; 1:0809A
� y&eth;470&THORN; &frac14; 90� &eth;30&thorn; 40:60&THORN; &frac14; 19:40� X&eth;470&THORN; &frac14; 60 tan&eth;2� 19:40&THORN;
</p>
<p>&frac14; 48:24mm
</p>
<p>Note the dependence on these results of the crystal orientation parameter f. It is easy to show that
</p>
<p>for f &frac14; 0�
</p>
<p>1. l(350)y &frac14; 3.3148 Å
2. l(470)y &frac14; 2.4707 Å
</p>
<p>Thus, neither of these reflections would be recorded for this orientation using the given wavelength
</p>
<p>range.
</p>
<p>Laue photographs can be scanned optically with a densitometer, which records both intensity and
</p>
<p>position of each spot on the film; alternatively, this information can be derived from the diffraction
</p>
<p>pattern recorded on a flat reusable CCD plate. These data are then processed by computer using
</p>
<p>powerful software, which will index and also refine the unit-cell parameters as required, Sect. 11.2.3ff.
</p>
<p>The interpretation of such records, which can contain thousands of spots, is a skilled operation but
</p>
<p>usually requires a minimum of user intervention. Knowledge of the crystal unit cell is a useful
</p>
<p>prerequisite; otherwise the crystal can be, and usually is, randomly oriented with respect to the
</p>
<p>direction of the X-ray beam. With a crystal of high symmetry it is possible to record most of the three-
</p>
<p>dimensional diffraction pattern on a single photograph. Being independent of mechanical constraints,
</p>
<p>Laue data can be recorded very rapidly, in seconds for good quality crystals at a high-intensity
</p>
<p>synchrotron radiation facility using a CCD plate (q.v.) instead of a photographic film. Even the
</p>
<p>problem previously mentioned of multiplicity of orders has proved to be less of a difficulty to an
</p>
<p>application of the method than at first thought, for example, by using a wavelength range of 0.6&ndash;1.6 Å
</p>
<p>in protein crystallography.
</p>
<p>The method thus facilitates novel studies using synchrotron radiation, such as time-dependent
</p>
<p>solid-state reactions and enzyme-driven transformations. The latter can be synchronized by employing
</p>
<p>tailor-made photosensitive substrates to delay the biochemical reactions until the X-ray experiment is
</p>
<p>ready. Such studies provide exciting new dimensions to X-ray crystallography.
</p>
<p>There are several important differences between synchrotron radiation and X-rays generated from
</p>
<p>a conventional laboratory source. The latter are usually emitted as characteristic radiation from a
</p>
<p>copper or molybdenum metal target, and comprise the predominant characteristic a and b wave-
</p>
<p>lengths, and a more general, less intense, polychromatic background. An appropriately selected metal
</p>
<p>filter allows the production an effectively monochromatic beam, Sect. 3.1.4. Synchrotron radiation
</p>
<p>(SR) is of extremely high intensity, a property which can be exploited for the examination of weakly
</p>
<p>diffracting or very small crystals. A particular wavelength can be selected from the continuous SR
</p>
<p>wavelength range by an appropriate filter, for either single-crystal rotation or powder diffraction
</p>
<p>experiments. Alternatively, the continuous polychromatic beam can be used for Laue photography
</p>
<p>with a stationary crystal, to record diffraction data efficiently and rapidly. Finally, SR has a very low
</p>
<p>beam divergence which results in very sharp diffraction spots, Fig. 5.18. This is particularly useful in
</p>
<p>providing good intensity data from poorly diffracting crystals, such as proteins, resulting subse-
</p>
<p>quently in greatly improved resolution of their X-ray Fourier images.
</p>
<p>204 5 Examination of Single Crystals: Optical and X-Ray Diffraction Practice</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_11#Sec6_11">http://dx.doi.org/10.1007/978-1-4614-3954-7_11#Sec6_11</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec5_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec5_3</a></div>
</div>
<div class="page"><p/>
<p>5.4.4 Oscillation Method
</p>
<p>Flat-Plate Oscillation Technique: Basic Ideas
This technique is now used for both preliminary assessment of the suitability of a crystal prior to
</p>
<p>X-ray data collection and data collection itself with monochromatic X-radiation. The diffraction
</p>
<p>pattern was originally recorded on a flat film, but has been replaced in modern applications by one of
</p>
<p>several electronic devices known as area detectors or position-sensitive devices that we discuss in
</p>
<p>Sect. 5.7.
</p>
<p>The intensity data are recorded in a series of exposures in which the crystal is rotated or
</p>
<p>oscillated through a small angular range d&rsquo; of about 0.1&ndash;2�. For crystals with large unit cells, it
is necessary to employ a large crystal&ndash;film distance, at least 60 mm, in order to effect separation of
</p>
<p>the diffraction spots on the film. A large effective surface area is necessary in order to record data to
</p>
<p>high resolution, that is, to high y values, for the production of good quality X-ray structure
</p>
<p>determinations. Previously, cameras were designed specifically for this purpose, and employed a
</p>
<p>flat-plate film perpendicular to the X-ray beam together with a mechanism for automatically
</p>
<p>changing the cassette, a carousel device, thus enabling several exposures to be set concurrently
</p>
<p>[6]. Area detectors, Sect. 5.7, are reusable, thus eliminating this requirement.
</p>
<p>The geometry necessary to define the (X, Y) coordinates of an upper-level spot on the film or plate
</p>
<p>is indicated in Fig. 5.19. In this diagram CO0 is perpendicular to the film plane. Triangle CO0P0 is
</p>
<p>right-angled at O0, and PP0O0 at P0; triangles DD0C and PP0C are similar, so that
</p>
<p>Fig. 5.18 Synchrotron
Laue photograph recorded
</p>
<p>from a crystal of
</p>
<p>Photoactive Yellow Protein
</p>
<p>(PYP) recorded on the
</p>
<p>ESRF Laue beam line
</p>
<p>(ID09B). The exposure was
</p>
<p>performed by accumulating
</p>
<p>16 single pulses each
</p>
<p>100 ps long (courtesy H.
</p>
<p>Ihee and M. Wulff, ESRF,
</p>
<p>Grenoble)
</p>
<p>5.4 Single-Crystal X-Ray Diffraction Techniques: Intensity Data Collection 205</p>
<p/>
</div>
<div class="page"><p/>
<p>Y=CP0 &frac14; z=CD0 &frac14; z=&eth;1� z2&THORN;1=2
</p>
<p>Y &frac14; CP0z=&eth;1� z2&THORN;1=2
</p>
<p>CP0
2 &frac14; R2 &thorn; X2
</p>
<p>Y2 &frac14; &eth;R2 &thorn; X2&THORN;z=&eth;1� z2&THORN;
</p>
<p>(5.5)
</p>
<p>where R is the distance CO0.
Further,
</p>
<p>r=R &frac14; tan 2y and r2 &frac14; X2 &thorn; Y2
</p>
<p>where r is the distance PO0. Therefore,
</p>
<p>R2tan22y &frac14; X2 &thorn; Y2 (5.6)
</p>
<p>Combining these results, it follows that
</p>
<p>X &frac14; R&frac12;tan22y&eth;1� z2&THORN; � z2�1=2 (5.7)
</p>
<p>Y &frac14; Rz&frac12;1&thorn; tan22y�1=2 (5.8)
</p>
<p>If a reciprocal lattice axis is parallel to the rotation axis, z RU corresponds to the appropriate level
</p>
<p>along the axis.
</p>
<p>Fig. 5.19 Coordinate pair X, Y of the diffraction spot P on a flat-plate oscillation photograph for a general upper-level
reflection: C, crystal; CC0, oscillation, or &rsquo; axis; O, reciprocal lattice origin; O0, origin of coordinates on the film;
OD &frac14; d*(hkl); CP, reflected beam, intersecting the Ewald sphere at D; DD0, perpendicular distance above the zero
level. Since the radius of the Ewald sphere is 1 RU, the radius of the upper level at level z is
</p>
<p>ffiffiffiffiffiffiffiffiffiffiffiffiffiffiffiffi
&eth;1� z2&THORN;
</p>
<p>q
; CO0 is the
</p>
<p>crystal-to-film distance
</p>
<p>206 5 Examination of Single Crystals: Optical and X-Ray Diffraction Practice</p>
<p/>
</div>
<div class="page"><p/>
<p>We will illustrate these ideas with the following worked example. A protein crystal with a
</p>
<p>monoclinic unit cell, a &frac14; 30.0 Å, b &frac14; 50.0 Å, c &frac14; 40.0 Å, b &frac14; 100�, is mounted on an oscillation
camera equippedwith a flat-plate cassette placedwith its plane perpendicular to theX-ray beam and at a
</p>
<p>distance R &frac14; 60 mm from the center of oscillation. The b axis of the crystal is vertical and perpendicu-
</p>
<p>lar to the X-ray beam, with �a parallel to the beam at the start of a counterclockwise rotation;
</p>
<p>X-rays with wavelength of 1.2 Å are used in the experiment. Consider the following problems:
</p>
<p>1. At what point in the rotation will the 004 reflection take place?
</p>
<p>2. Calculate the (X, Y) coordinates (in mm) of the 004 spot on the film, X being the horizontal
</p>
<p>coordinate and Y the vertical.
</p>
<p>3. Calculate the (X,Y) coordinates (in mm) of the 014 spot on the film.
</p>
<p>1. Refer to Fig. 5.18 to see that the required rotation is y(004), calculated as follows:
</p>
<p>d�&eth;004&THORN; &frac14; 4c�
</p>
<p>where c�&frac14; l=c sin b&frac14;1:2=&eth;40 sin 100&THORN;&frac14;0:03046, so that
</p>
<p>d�&eth;004&THORN; &frac14; 0:1219 &frac14; 2 sin y&eth;004&THORN;
</p>
<p>Hence, sin y&eth;004&THORN; &frac14; 0:06095 and y(004) &frac14; 3.494� (the rotation angle from starting position).
</p>
<p>2. From Fig. 5.20, we have
</p>
<p>X&eth;004&THORN; &frac14; R tan 2y&eth;004&THORN;
</p>
<p>&frac14; 60 tan 6:988
</p>
<p>&frac14; 7:354mm
</p>
<p>Y&eth;004&THORN; &frac14; 0mm
</p>
<p>3. For the 014 reflection:
</p>
<p>z &frac14; b� &frac14; l=b &frac14; 0:0240
</p>
<p>Using (2.16), it follows that
</p>
<p>d�2&eth;014&THORN; &frac14; b�2 &thorn; 16c�2
</p>
<p>&frac14; 0:02402 &thorn; 16&eth;0:03046&THORN;2
</p>
<p>&frac14; 0:01542
</p>
<p>so that
</p>
<p>d�&eth;014&THORN; &frac14; 0:1242 &frac14; 2 sin y&eth;014&THORN;
</p>
<p>Hence, sin y&eth;014&THORN; &frac14; 0:06210, and 2y &frac14; 7.121�, by (5.7). Thus,
</p>
<p>X&eth;014&THORN; &frac14; 60&frac12;tan27:121&eth;1� 0:02402&THORN; � 0:02402�1=2 &frac14; 7:352mm
</p>
<p>5.4 Single-Crystal X-Ray Diffraction Techniques: Intensity Data Collection 207</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_2#Eq16_2">http://dx.doi.org/10.1007/978-1-4614-3954-7_2#Eq16_2</a></div>
</div>
<div class="page"><p/>
<p>and from (5.8)
</p>
<p>Y&eth;014&THORN; &frac14; 60� 0:0240&frac12;1&thorn; tan27:121�1=2 &frac14; 1:451 mm
</p>
<p>Hence, the 004 reflection has coordinates (7.354, 0.0) mm; the 014, (7.352, 1.451) mm.
</p>
<p>5.5 Measurement of the Intensities of Diffraction Data
</p>
<p>A variety of options is open to the crystallographer for obtaining measurements of X-ray intensities.
</p>
<p>These include the use of a single counter or serial diffractometer, and the latest area detectors. A list
</p>
<p>of some of the currently available equipment is to be found in Sect. 5.7.10. Examples include
</p>
<p>the computer-controlled four-circle serial diffractometer (Nonius CAD4), and the kappa geometry
</p>
<p>CCD diffractometer (Nonius KappaCCD). A cryostream device may be attached for work at low
</p>
<p>temperatures.
</p>
<p>Fig. 5.20 Plan view of flat-plate oscillation geometry, looking down the oscillation axis, CC0 in Fig. 5.19, at C. In the
accompanying example, the y axis of the crystal (C) is perpendicular to the diagram, and coincides with the rotation
</p>
<p>axis. Because the crystal is monoclinic, the a*c* levels superimpose in this orientation. The diagram shows c* at
</p>
<p>the beginning of an anticlockwise rotation perpendicular to the X-ray beam. The crystal rotates through y(004) and, at
</p>
<p>the reciprocal lattice point 004, d* (004) intersects the Ewald sphere and the 004 reflection is produced. For the
</p>
<p>014 reflection, a further small rotation is necessary
</p>
<p>208 5 Examination of Single Crystals: Optical and X-Ray Diffraction Practice</p>
<p/>
</div>
<div class="page"><p/>
<p>5.5.1 Single Counter or Serial Diffractometers
</p>
<p>Traditional diffractometers that employ scintillation counters1 to detect and measure X-ray reflec-
</p>
<p>tions from single crystals, one by one, incorporate a mechanical goniometer to orientate the crystal
</p>
<p>into the correct reflecting position for each hkl reflection, and to rotate the counter to receive the
</p>
<p>scattered radiation from this single reflection. The energy is transformed electronically into a form
</p>
<p>suitable for conversion to I(hkl). Because each reflection is measured individually, with a count time
</p>
<p>typically of around 60 s, the process is very slow, particularly for proteins, which routinely involve
</p>
<p>the measurement of tens of thousands of reflections. The dynamic range of the instrument is the
</p>
<p>quantified range of intensities from very weak to very strong over which the instrument can produce
</p>
<p>measured values of intensity. This can be enhanced to enable measurement of very strong reflections
</p>
<p>through the use of a calibrated attenuator, such as a strip of nickel foil in the case of copper radiation.
</p>
<p>The wider the dynamic range available the better will be the measured set of intensity data. For
</p>
<p>relatively unstable crystals, proteins and other macromolecules, the crystal needs to be replaced once
</p>
<p>it is observed to be dying. Consequently, several crystals may be required in the production of a
</p>
<p>complete data set, which is subsequently achieved by scaling the individual sets together. However,
</p>
<p>the accuracy attainable is possibly better than for most other methods, and this advantage offsets the
</p>
<p>lengthy time scale that is required for the data collection.
</p>
<p>The disadvantages of a slow data-collection rate and the need for several crystals with single
</p>
<p>counter diffractometry have been largely overcome by the use of electronic film area detectors and
</p>
<p>image plates, which methods we discuss in Sect. 5.7.
</p>
<p>5.6 Single-Crystal X-Ray Diffractometry
</p>
<p>It has become commonplace for X-ray diffraction data to be collected by means of a diffractometer.
</p>
<p>We give here a brief description of the Enraf&ndash;Nonius CAD4 kappa-diffractometer. Two other similar
</p>
<p>systems are considered later.
</p>
<p>5.6.1 Instrument Geometry
</p>
<p>The CAD4 kappa-diffractometer is characterized by its k-goniometer, which differs in geometry
</p>
<p>from other four-circle diffractometers. The k-goniometer carries the goniometer head and arcs, and
</p>
<p>keeps the crystal at the center of the diffractometer throughout the data collection. The k-goniometer
</p>
<p>is a combination of three parts, which carry the rotation axes. All axes intersect in the center of the
</p>
<p>diffractometer.
</p>
<p>The arcs are mounted on the f axis, on the k-block, as shown in Fig. 5.21; the angle of rotation
</p>
<p>about this axis is fk. The k-block is rotated about the k axis, being itself carried by the o-block.
</p>
<p>In turn, the o-block is rotated about the axis (ok) and is carried on the diffractometer base. The angle
</p>
<p>w between the o and X axes is ca. 50�, and that between the k and f axes is also ca. 50�. Thus, the
</p>
<p>goniometer can access all directions of w within about 100� of the zero setting of the instrument. This
</p>
<p>suspension gives the CAD4 an enhanced flexibility over the traditional Eulerian cradle. There is also a
</p>
<p>2y axis, coinciding with the o axis, which carries the scintillation counter detector. The coincidence
</p>
<p>of the f and o axes corresponds to k &frac14; 0; ok &frac14; 0 for k in the plane of the diffractometer axes X and
</p>
<p>1 Similar to Geiger counters.
</p>
<p>5.6 Single-Crystal X-Ray Diffractometry 209</p>
<p/>
</div>
<div class="page"><p/>
<p>Z, with the k block opposite +X, and 2y &frac14; 0 when the center of the detector lies in the plane of X and
Z and opposite +X. The definition of fk is arbitrary, and a suitable working procedure is arranged.
</p>
<p>Starting from k &frac14; ok &frac14; 0, positive rotations of y, o, and f move a vector from Y toward X, and a
positive rotation about k moves a vector from Y to a position below the horizontal plane.
</p>
<p>5.6.2 Rotation of the Crystal into a Diffracting Position
</p>
<p>In the zero position of the CAD4, a vector c is assumed to be attached to the crystal, with components
c1, c2, and c3 parallel to X, Y, and Z, respectively. The operation to be applied to c is given in terms of
the angles ok, k, fk by
</p>
<p>c&eth;o; k;f&THORN; &frac14; Z&eth;ok&THORN; � Y&eth;�a&THORN; � Z&eth;k&THORN; � Y&eth;a&THORN; � Z&eth;fk&THORN; � jcj (5.9)
</p>
<p>We can define general clockwise rotations of amounts a, b, and g about X, Y, and Z, respectively,
</p>
<p>by the following matrices:
</p>
<p>X&eth;a&THORN;Z!Y &frac14;
1 0 0
</p>
<p>0 cos a sin a
</p>
<p>0 � sin a cos a
</p>
<p>2
4
</p>
<p>3
5 (5.10)
</p>
<p>Fig. 5.21 Schematic
diagram of the Nonius
</p>
<p>CAD4 single crystal X-ray
</p>
<p>k diffractometer, showing
</p>
<p>the X, Y, Z instrumental
</p>
<p>coordinate system and the
</p>
<p>rotation directions
</p>
<p>(reproduced with the
</p>
<p>permission of
</p>
<p>Enraf&ndash;Nonius, Delft)
</p>
<p>210 5 Examination of Single Crystals: Optical and X-Ray Diffraction Practice</p>
<p/>
</div>
<div class="page"><p/>
<p>Y&eth;b&THORN;X!Z &frac14;
cos b 0 � sin b
0 1 0
</p>
<p>sin b 0 cos b
</p>
<p>2
4
</p>
<p>3
5 (5.11)
</p>
<p>Z&eth;g&THORN;Y!X &frac14;
cos g sin g 0
</p>
<p>� sin g cos g 0
0 0 1
</p>
<p>2
4
</p>
<p>3
5 (5.12)
</p>
<p>In (5.9), a is the angle of rotation of the k-block about Y such that the k and Z axes coincide. The
</p>
<p>term Y(�a) in (5.9) returns the goniometer to its original position.
</p>
<p>5.6.3 Transformation from Miller Indices to Diffractometer Angles
</p>
<p>To set a reflecting plane in the diffracting position, its Miller indices are transformed to a scattering
</p>
<p>vector c by an orientation matrix R:
</p>
<p>R &frac14;
a�X b
</p>
<p>�
X c
</p>
<p>�
X
</p>
<p>a�Y b
�
Y c
</p>
<p>�
Y
</p>
<p>a�Z b
�
Z c
</p>
<p>�
Z
</p>
<p>2
4
</p>
<p>3
5 (5.13)
</p>
<p>where the reciprocal unit-cell vectors a*, b*, and c* are resolved into components along X, Y, and Z
respectively. Then
</p>
<p>c1
</p>
<p>c2
</p>
<p>c3
</p>
<p>2
64
</p>
<p>3
75 &frac14; R �
</p>
<p>h
</p>
<p>k
</p>
<p>l
</p>
<p>2
64
</p>
<p>3
75 (5.14)
</p>
<p>or
</p>
<p>c &frac14; R � h (5.15)
</p>
<p>Other similar transformations are built into the software of the diffractometer so that the scattering
</p>
<p>vector c is brought into the horizontal plane and, via Eulerian forward and inverse transformations,
for mathematical convenience, the values of the angles ok, k, and fk, by which the diffractometer
</p>
<p>circles must be moved, are determined. Then the intensity of the reflection that corresponds to c(ok,
k, fk) can be measured and recorded.
</p>
<p>5.6.4 Data Collection
</p>
<p>The flexible routines that are built into the CAD4 permit selection of the hkl reflection in differing
</p>
<p>ways, including Friedel pairs hkl and �h �k �l, Sects. 3.6.1 and 7.5. The azimuthal angle, the angle
</p>
<p>between the vertical and the normal to the f-circle, can be variously specified, including multiple
</p>
<p>measurements of reflections at different azimuth values so as to derive an empirical absorption
</p>
<p>correction curve as a function of the angle f, see Sect. 4.1.3.
</p>
<p>5.6 Single-Crystal X-Ray Diffractometry 211</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec25_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec25_3</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_7#Sec22_7">http://dx.doi.org/10.1007/978-1-4614-3954-7_7#Sec22_7</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_4#Sec4_4">http://dx.doi.org/10.1007/978-1-4614-3954-7_4#Sec4_4</a></div>
</div>
<div class="page"><p/>
<p>Intensity and background measurements are carried out, and the crystal may be monitored for
</p>
<p>decay and movement. Where the latter occurs, the crystal may be reoriented as necessary during the
</p>
<p>data collection. If the space group is known, only the geometrically permitted reflections need to be
</p>
<p>scanned. This procedure is of great significance with centered unit cells. In addition, more than the
</p>
<p>unique portion of reciprocal space can be explored. By means of the CAD4 Structure Determination
</p>
<p>Package, the symmetry-equivalent reflections can be merged to give the best set of unique data,
</p>
<p>Lorentz and polarization corrections applied, and an absorption correction included if deemed
</p>
<p>desirable. The data set can be truncated so as to exclude weak reflections of lower accuracy, for
</p>
<p>example, those for which F2o is less than 3s&eth;F2o&THORN;, or they may be remeasured over a longer time period
so as to increase their precision. The CAD4 is provided with a means of taking X-ray photographs of
</p>
<p>the crystal on the diffractometer, using Polaroid film. This procedure may be very helpful where there
</p>
<p>is a &ldquo;difficult&rdquo; space group, or where twinning is suspected, but it may not prove a substitute for a
</p>
<p>thorough initial examination of the crystal by conventional X-ray photography where necessary.
</p>
<p>The automatic search and indexing routines for determining the unit cell are not always definitive.
</p>
<p>This situation may arise for various reasons, such as twinning, poor crystal quality, leading to weak
</p>
<p>reflections, or insufficient spread of reflections in reciprocal space. Several options are available for
</p>
<p>operator intervention. One of the most useful is to take a Polaroid rotation photograph about the
</p>
<p>randomly oriented f-axis and use it to determine the x, y coordinates of reflections on the film.
</p>
<p>About ten such pairs of values will often suffice to determine an approximate unit cell. Then the unit
</p>
<p>cell can be refined by locating more reflections, well distributed in reciprocal space, and applying a
</p>
<p>least-squares procedure, Sect. 8.4.2. The f-axis photograph may detect poor crystal quality but not
</p>
<p>twinning, and it will not help with space-group determination; normal rotation X-ray photographs about
</p>
<p>the crystallographic axes may be helpful in these situations, but this procedure is rarely used nowadays.
</p>
<p>The techniques available with the CAD4 have been well tested in laboratories throughout the
</p>
<p>world, and their careful application can lead to a data set of high quality, capable of solving and
</p>
<p>refining a crystal structure to high precision, with R-factor (q.v.) between 5 and 1%.
</p>
<p>5.6.5 Scanning Over a Peak: v/u Versus v Scans
</p>
<p>The most widely used, and arguably the best, scanning option employed with the CAD4 and similar
</p>
<p>types of diffractometer involves a coupled rotation of the crystal (o-axis) and the counter (2y-axis),
</p>
<p>Fig. 5.21, both rotation axes being vertical during the scan. For a given reflection, the center of the
</p>
<p>scan coincides with the calculated position of the maximum intensity, based on the orientation matrix
</p>
<p>and unit-cell parameters already determined. Figure 5.22 shows the variation of intensity recorded for
</p>
<p>such a scan, using a good quality crystal. In order to obtain a measure of the intensity it is necessary to
</p>
<p>evaluate the counts recorded in the area P on the diagram. For most reflections there is general
</p>
<p>background intensity, shown in the diagram as being slightly asymmetrical. The background levels
</p>
<p>B1 and B2 are established by short scans on either side of the peak.
</p>
<p>The scans are not smooth, and consist of, say, n short steps in both B1 and B2 andm similar steps in
</p>
<p>P. If the total counts are b1 for region B1, b2 for region B2, and N for the peak P, it can be shown that
</p>
<p>the number of counts in P is &frac12;N � m&eth;b1&thorn; b2&THORN;=2n�. For most small-molecule crystals, the peak width
would be about 40&ndash;60min of arc and the total scan time about 60 s. In this method the two backgrounds
</p>
<p>are usually fairly similar. The facility for inspecting individual scans is extremely useful for establish-
</p>
<p>ing the correct scan width to use for a given crystal, and to check for crystal splitting, which would
</p>
<p>produce a double, or even a multiple, peak.
</p>
<p>For crystals with very large unit cells, which occur mainly withmacromolecules, neighboring peaks
</p>
<p>are usually too close together, because the reciprocal lattice constants are very small, to allow the use
</p>
<p>212 5 Examination of Single Crystals: Optical and X-Ray Diffraction Practice</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_8#Sec31_8">http://dx.doi.org/10.1007/978-1-4614-3954-7_8#Sec31_8</a></div>
</div>
<div class="page"><p/>
<p>ofo/y scans. Instead, the alternativeo-scan method is used, which involves a stationary counter, fixed
</p>
<p>at the correct value of 2y. This procedure reduces the possibility of the diffraction maxima for
</p>
<p>neighboring reflections overlapping, but does not eliminate it altogether.
</p>
<p>5.7 Area Detectors (Position-Sensitive Detectors)
</p>
<p>The disadvantages of single counter diffractometry, mentioned earlier, that apply mainly to large-
</p>
<p>molecule structures, are a slow data-collection rate and the requirement of several crystals for
</p>
<p>collection of a complete data set, with the attendant errors associated with scaling and crystal
</p>
<p>deterioration: macromolecular crystals may contain 40&ndash;50% of solvent, which renders them suscepti-
</p>
<p>ble to decay in the X-ray beam. These problems have been largely overcome by the use of electronic
</p>
<p>film area detectors and image plates that have enjoyed rapid development in recent years. In many
</p>
<p>laboratories, these methods of detection are used for both large- and small-molecule crystallography,
</p>
<p>and especially in structure determination at low temperature.
</p>
<p>The use of an area detector or image plate allows many nonoverlapping reflections to be recorded in
</p>
<p>a single exposure. Each exposure is produced by oscillating the crystal through a small angle, D&rsquo;. Each
</p>
<p>reflection is received by a detector, which is an effectively flat surface with a uniform sensitivity to
</p>
<p>X-rays. A series of sequential exposures is recorded, during which the crystal may be rotated through a
</p>
<p>total of 180�, for a triclinic crystal, or as little as 30�, for a hexagonal or cubic crystal. TheX,Y position of
</p>
<p>each reflection (Fig. 5.19) enables its hkl index to be computed, and the signals received at the detector
</p>
<p>are converted electronically into intensity data. Area detectors are therefore also position-sensitive
</p>
<p>detectors, because both the intensity of a diffracted beam and the exact position where it strikes the
</p>
<p>detector are determined. Several designs are commercially available, as discussed next.
</p>
<p>5.7.1 Multiwire Proportional Counter
</p>
<p>The multiwire proportional counter (MWPC) is a digital detector, working on the principle of
</p>
<p>detection of X-rays by ionization. This type of detector was used by protein crystallographers in
</p>
<p>Fig. 5.22 Profile of a reflection peak scanned in the o/ymode. The background level to be subtracted on either side of
the peak is indicated. It is typically slightly asymmetrical
</p>
<p>5.7 Area Detectors (Position-Sensitive Detectors) 213</p>
<p/>
</div>
<div class="page"><p/>
<p>the 1980s and 1990s. The design of the MWPC is based on a xenon-filled chamber with an anode
</p>
<p>lying between two cathodes, Fig. 5.23. Each anode and each cathode consists of a plane of parallel
</p>
<p>wires, with the direction of the wires in the first cathode and the anode perpendicular to those in the
</p>
<p>second cathode, Fig. 5.24. During exposure to X-radiation, X-ray photons enter the chamber. For each
</p>
<p>photon that is absorbed, a xenon molecule is ionized into a positive xenon ion and an inner shell
</p>
<p>electron having a kinetic energy that is virtually all of the energy of the absorbed photon. The
</p>
<p>liberated electron ionizes further xenon molecules. In the case of Cu Ka radiation, a single 8 keV
</p>
<p>photon, the energy of a Cu Ka photon, produces about 300 ions and primary electrons. An electric
</p>
<p>Fig. 5.23 Principle of the multiwire proportional counter (MWPC)
</p>
<p>Fig. 5.24 Design of the MWPC
</p>
<p>214 5 Examination of Single Crystals: Optical and X-Ray Diffraction Practice</p>
<p/>
</div>
<div class="page"><p/>
<p>field accelerates the positive ions toward the first cathode and the cloud of electrons toward the anode.
</p>
<p>The electron cloud passes the first cathode to hit the anode plane, producing an avalanche of
</p>
<p>secondary ionization, with an amplification factor of the order of 104 or more for each primary
</p>
<p>electron. While the electrons give a negative pulse on the nearest anode wire, the positive ions move
</p>
<p>toward the second cathode giving a positive pulse on the nearest cathode wire. To prevent ultraviolet
</p>
<p>photons, produced in the secondary ionization, from restarting ionization in another region of the
</p>
<p>chamber, a small amount of quenching gas, such as methane, is added to the xenon and absorbs excess
</p>
<p>photons and dissipates energy. Counting is accomplished through the orthogonal array of wires that
</p>
<p>samples the centroid of the charge distribution. The analog signals from the detector are digitized to
</p>
<p>produce the corresponding reading.
</p>
<p>This type of counter has a low noise level, is extremely sensitive, and covers a wide dynamic range
</p>
<p>of intensities. Unfortunately the MWPC is severely hampered by size limitations, which reduce the
</p>
<p>resolution range achievable with Cu Ka radiation, suffer from errors due to parallax at high angles of
</p>
<p>incidence, and require a helium path in order to limit absorption, because of the large crystal-to-
</p>
<p>detector distances employed. Commercially available MWPC devices include the Bruker AXS HI-
</p>
<p>STAR (11.5 cm diameter) and the larger Xuong&ndash;Hamlin (also called San Diego Multiwire System or
</p>
<p>SDMS, 30 � 30 cm).
</p>
<p>5.7.2 FAST Area Detector (Enraf&ndash;Nonius FAST) [3]
</p>
<p>In thismethod, theX-ray reflections strike a phosphor coated fiber optics screen that converts the signals
</p>
<p>into light photons, which are then intensified, integrated, and digitized. The recording device incorpo-
</p>
<p>rates a television scanning system and the goniometer is essentially that used in the CAD4 diffractome-
</p>
<p>ter. The disadvantages of this method include high electronic noise and consequent low dynamic range
</p>
<p>(1:103) and the need for remeasurement of strong intensities at decreased camera voltage. In view of
</p>
<p>these inherent difficulties the FAST technique is now seldom used.
</p>
<p>5.7.3 Image Plate
</p>
<p>The image-plate detector [7, 8], shown diagrammatically in Fig. 5.25a, consists of a barium halide
</p>
<p>phosphor doped with divalent europium, BaFBrEu(II), and held in an organic binder. On exposure to
</p>
<p>X-rays, an X-ray photon hits the image plate and its energy is stored in a Br� &ldquo;hole&rdquo; (an F-center
</p>
<p>defect) and the Eu(II) species is excited into the metastable Eu(III) state until further stimulated:
</p>
<p>Eu&eth;II&THORN;�!
X-rays
</p>
<p>Eu&eth;III&THORN; &thorn; e
</p>
<p>(see also Fig. 5.25b). After completion of each exposure, during which time the crystal is oscillated in
</p>
<p>the X-ray beam for a few minutes, the image plate is scanned with a narrow {helium&ndash; neon laser beam
</p>
<p>(Fig. 5.25c), which causes the regions converted to Eu(III), where the X-ray spots would be located on
</p>
<p>a photographic film, to emit violet light of wavelength 3900Å with intensity proportional to the
</p>
<p>absorbed X-ray energy.
</p>
<p>Eu&eth;III&THORN; &thorn; e�!
laser
</p>
<p>Eu(II)&thorn; hn &eth;blue - violet&THORN;
</p>
<p>5.7 Area Detectors (Position-Sensitive Detectors) 215</p>
<p/>
</div>
<div class="page"><p/>
<p>This light is then detected with a photomultiplier system, integrated and digitized. After reading
</p>
<p>the stored data, the plate is cleaned by exposure to bright yellow light, Fig. 5.25d, and it is then ready
</p>
<p>to record the next image. Image plates up to 345 mm diameter are now available (Mar Image Plate
</p>
<p>345), which enable data to be collected to a resolution of about 1.4 Å using Cu Ka radiation, and even
</p>
<p>Fig. 5.25 (a) Resting state
of the image plate. (b)
Exposure to X-ray
</p>
<p>diffraction from the
</p>
<p>oscillating crystal. (c)
Scanning with laser light
</p>
<p>and measurement of the
</p>
<p>emitted light from each
</p>
<p>spot with a photomultiplier.
</p>
<p>(d) Cleaning the plate
ready for further use (e)
Front view of MAR 345
</p>
<p>image plate
</p>
<p>216 5 Examination of Single Crystals: Optical and X-Ray Diffraction Practice</p>
<p/>
</div>
<div class="page"><p/>
<p>better resolution with synchrotron radiation at smaller radiation wavelengths. The resolution restric-
</p>
<p>tion normally precludes the choice of this device for routine small-molecule analysis, but it is very
</p>
<p>popular for large-molecule work. An approximately 1:105 range of intensities can be recorded
</p>
<p>compared with ca. 1:200 for X-ray film.
</p>
<p>5.7.4 Charge-Coupled Device Area Detectors
</p>
<p>In this type of system, which is rapidly becoming the method of choice in both small-molecule and
</p>
<p>macromolecular crystallography, the video tube of the now almost obsolete FAST area detector,
</p>
<p>Sect. 5.7.2, or alternatively the image plate, Sect. 5.7.3, is replaced by a charge-coupled device
</p>
<p>(CCD). The CCD has been developed for a multitude of applications where extremely efficient
</p>
<p>detection of photons is required, as in astronomy and other branches of physics as well as X-ray
</p>
<p>crystallography. We could say that the ideal characteristics of a detector are:
</p>
<p>&bull; 100% quantum efficiency (independent of wavelength)
</p>
<p>&bull; Perfectly uniform response and unlimited dynamic range (both energy-wise)
</p>
<p>&bull; Electronically noiseless
</p>
<p>&bull; Completely characterized components
</p>
<p>Since the conception [9] of the CCD in 1970, the current state of the art has now effectively
</p>
<p>achieved all of these criteria and the design of the CCD will probably not be improved further. In
</p>
<p>theory, a CCD performs the following tasks:
</p>
<p>1. Generation of charge. When a photon strikes a CCD, it generates electron&ndash;hole pairs by the
</p>
<p>photoelectric effect [10].
</p>
<p>2. Collection of the resulting charge or charges in the pixels formed by an array of electrodes or gates.
</p>
<p>3. Transference of the charge on each pixel by application of a differential voltage across the gates.
</p>
<p>This information is conveyed pixel by pixel for counting.
</p>
<p>4. Detection of the individual charges and conversion to an output voltage, followed by digital
</p>
<p>encoding prior to computer processing [11, 12]. Figure 5.26 shows a possible design for a
</p>
<p>CCD chip.
</p>
<p>The detector used with the Nonius Kappa CCD diffractometer, Fig. 5.27, achieves high efficiency
</p>
<p>through the use of a special grade phosphor (Gd2O2S) with directly bonded high-quality fiber optics, in
</p>
<p>turn also directly bonded to the CCD chip. The CCD chip needs to be actively cooled to between�20
and �50�C, which is achieved through the incorporation of a stack of four Peltier elements again
bonded directly to the chip. This results in a stable low temperature at the chip, which is a requirement
</p>
<p>for low noise and consequent high sensitivity with low background. Figure 5.28 shows a composite
</p>
<p>Fig. 5.26 Design of a
CCD chip
</p>
<p>5.7 Area Detectors (Position-Sensitive Detectors) 217</p>
<p/>
</div>
<div class="page"><p/>
<p>Fig. 5.27 A view of the Nonius Kappa CCD diffractometer
</p>
<p>Fig. 5.28 Composite of the molecule 1,2-ethene-3,30-di-tert-butyl-diimidazole-2,20-diylidene indium(III) trihydride,
C16H32In2N4, superimposed on part of its diffraction pattern, as recorded on a Nonius Kappa CCD diffractometer
</p>
<p>218 5 Examination of Single Crystals: Optical and X-Ray Diffraction Practice</p>
<p/>
</div>
<div class="page"><p/>
<p>representation of the novel molecule 1,2-ethene-3,30-di-tert-butyl-diimidazole-2,20-diylidene indium
(III) trihydride, C16H32In2N4, superimposed on part of its diffraction pattern, which was recorded on a
</p>
<p>Nonius Kappa CCD diffractometer [13].
</p>
<p>5.7.5 The Tiled CCD
</p>
<p>The MAR (Rayonix USA) company has recently introduced the MX-series of detectors described as
</p>
<p>the tiled continuous CCD which uses either a 3 � 3 or a 4 � 4 array of fiber optic elements so as to
</p>
<p>make a larger active surface. This is achieved by coupling the detector to a fiber optic, and linking
</p>
<p>multiple CCD units together.
</p>
<p>5.7.6 Charge-Coupled Device Including Tiled CCD Versus Image Plate
</p>
<p>Image-plate detectors may be considered to be the heart of the workforce in many of the current
</p>
<p>applications in recording and digitizing X-ray diffraction data. They have a large active surface, good
</p>
<p>spatial resolution, and are relatively easy on the financial budget, which is a key consideration in
</p>
<p>many science departments. Their main drawback has been that the readout time of 60 s is relatively
</p>
<p>slow. However, recent design improvements, such as the MAR 345 IP, have resulted in a significant
</p>
<p>reduction of this time to as low as 26 s and can record high-quality data from both small-molecule
</p>
<p>and macromolecular crystals. The readout time is still slow compared to 1 s or less for a CCD
</p>
<p>which can record atomic resolution data for small proteins at the synchrotron [13, 14]. In a local
</p>
<p>laboratory environment this is a minor consideration, because the limited power of the X-ray
</p>
<p>source means that exposure times of 5&ndash;10 min are the norm. At modern synchrotron stations, where
</p>
<p>beam time is at an absolute premium, IP detectors while being extremely reliable are far too slow.
</p>
<p>The CCD detector on the other hand is very fast (30&ndash;90 s exposures) and the small active surface
</p>
<p>problems which restrict the range of data collection have now been alleviated by the introduction of
</p>
<p>the tiled CCD, described in the previous section.
</p>
<p>With their high sensitivity, high resolution, and low noise, the CCD is now in regular use for high-
</p>
<p>resolution macromolecular structure determinations [14, 15]. Diffractometers employing CCD,
</p>
<p>including software packages, are available from Nonius [16], Bruker [17], and Oxford-diffraction
</p>
<p>[18], a system which includes optional liquid nitrogen or liquid helium cooling of the crystal
</p>
<p>(Fig. 5.29a); other systems are run routinely with liquid nitrogen cooling. Figure 5.29b shows a
</p>
<p>close-up of the Oxford Cryosystems CCD detector.
</p>
<p>5.7.7 Data Collection Strategies
</p>
<p>X-ray intensity data are measured most frequently using the oscillation technique as described in
</p>
<p>Sect. 5.4.4, or with one or two minor variations. The crystal may be flash frozen and immediately held
</p>
<p>in theX-ray beam on a goniometer head for data collection. In-houseX-rays are generated using a rotating
</p>
<p>anode or other high outputX-ray tube, Sect. 3.1. For small-molecule analysis a molybdenumX-ray tube
</p>
<p>may be used and for proteins an X-ray tube with a copper target is chosen; most protein crystals would
</p>
<p>not give a high-quality data set with molybdenum X-radiation. A highly intense synchrotron source
</p>
<p>with a wavelength range of 0.7&ndash;1.0 Å would be ideal in most cases.
</p>
<p>The crystal can be rotated or oscillated about the f axis of the equipment by a preselected amount.
</p>
<p>Most crystals will arrive on the goniometer head in a random orientation unless care has been taken to
</p>
<p>5.7 Area Detectors (Position-Sensitive Detectors) 219</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec1_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec1_3</a></div>
</div>
<div class="page"><p/>
<p>ensure that a particular crystallographic axis is parallel to thef axis.With an IP orCCDdetector, data are
</p>
<p>collected in a series of small angle oscillations ofDfwhereDfwould be selected between 0.1 and 2.0�.
</p>
<p>The overall angular range of f for the whole data could be 180 or 360� in order to ensure collection of
</p>
<p>Friedel equivalent reflections. The final data set would then include sufficient measurements to enable
</p>
<p>the unit cell, space group andRint to be evaluated. It is often possible, and advisable, to derive the unit cell
</p>
<p>and space group by indexing a subset of the data at an earlier stage in the collection process in order to
</p>
<p>enable any problems that may emerge to be dealt with. Most software packages will allow an efficient
</p>
<p>data collection strategy to be designed in order to minimize problems such as radiation damage. Highly
</p>
<p>intense X-ray sources available at the present time often produce data images that are sufficiently intense
</p>
<p>to record after a single pass of the crystal through the angular range Df.
</p>
<p>The method of data collection may be designated as fine-f-slicing if Df is less than or equal to half
</p>
<p>the reflection, or rocking, width, which is a function of the crystal mosaic spread and beam divergence
</p>
<p>usually in the range 0.1&ndash;1�, or thick-f-slicing if Df is greater than half the reflection width. At a
</p>
<p>Fig. 5.29 Oxford-
diffraction: (a) The CCD
detector in position at the
</p>
<p>Oxford-diffraction
</p>
<p>Xcalibur diffractometer.
</p>
<p>(b) A close-up of the
Oxford-diffraction CCD
</p>
<p>detector (reproduced by
</p>
<p>permission of Agilent,
</p>
<p>formerly Oxford
</p>
<p>Diffraction)
</p>
<p>220 5 Examination of Single Crystals: Optical and X-Ray Diffraction Practice</p>
<p/>
</div>
<div class="page"><p/>
<p>synchrotron source, where the beam divergence is extremely small, a data set with images, or frames,
</p>
<p>of 1.5� rotation would be a thick data set if the crystal mosaicity was 0.75� but would be a thin or fine
</p>
<p>data set if the crystal mosaicity was 0.5�.
</p>
<p>A data set of thick images usually has more fully-recorded reflections, fewer partially recorded
</p>
<p>reflections, more spatial overlaps, higher X-ray background, more saturated pixels, and a lower total
</p>
<p>number of images. A fine data set has no fully recorded reflections, fewer spatial overlaps, lower X-ray
</p>
<p>background, fewer saturated pixels, more time consumed during data collection in reading out the
</p>
<p>detector between oscillations or rotations, and a much larger total number of images. Each reflection
</p>
<p>may span two or three adjacent images. Figure 5.30 is a good example of a clean oscillation exposure
</p>
<p>from a cubic insulin protein crystal [19].
</p>
<p>5.7.8 The CMOS Detector, Pilatus 1M Detector System, and Continuous Rotation
</p>
<p>A significant recent development in flat-plate X-ray intensity measurement technology, particularly
</p>
<p>appropriate for macromolecular crystallography, is based on the X-ray complementary metal-oxide
</p>
<p>semiconductor (CMOS) system.2 The Pilatus 1M detector [20] is a CMOS with hybrid-pixel
</p>
<p>technology, a detector with over one million pixels that operates in single-photon counting mode.
</p>
<p>This detector was designed also for applications in macromolecular crystallography, and is one of the
</p>
<p>largest pixel array detectors currently in use at synchrotron sources. It consists of 18 multichip
</p>
<p>Fig. 5.30 A high-quality
oscillation photograph of
</p>
<p>cryo-cooled cubic insulin
</p>
<p>recorded using a Bruker
</p>
<p>Microstar rotating anode
</p>
<p>source with a Proteum X8
</p>
<p>CCD plate; Cu Ka X-
</p>
<p>radiation was used, with
</p>
<p>crystal-to-detector distance
</p>
<p>of 45 mm, and a 0.5�
</p>
<p>oscillation. The image
</p>
<p>display software was
</p>
<p>COSMO/SAINT
</p>
<p>(reproduced by courtesy of
</p>
<p>Dr. S. M. Prince and
</p>
<p>Dr. P. K. Bryant, Faculty of
</p>
<p>Life Sciences, Manchester)
</p>
<p>2Hamamatsu Photonics KK C10158DK.
</p>
<p>5.7 Area Detectors (Position-Sensitive Detectors) 221</p>
<p/>
</div>
<div class="page"><p/>
<p>modules covering an area of about 500 cm2 and was designed and built at the Swiss Light Source
</p>
<p>synchrotron location within the Paul Scherrer Institute, Switzerland. The modules are read out in
</p>
<p>parallel, leading to a full-frame readout time of 6.7 ms. This allows crystallographic data to be
</p>
<p>acquired in the fine-slicing mode, but with continuous rotation of the sample, not broken up into small
</p>
<p>angle oscillations, and the X-ray shutter is kept continuously in the open position during the entire
</p>
<p>data collection period. Using these techniques, the accuracy of the intensity data measurements can
</p>
<p>be improved without increasing the data collection time.
</p>
<p>The Pilatus silicon pixel detector was used in the analysis of the crystal structure of the protein
</p>
<p>thaumatin [21]; the processing of the data produced extremely satisfactory merging R-factors of about
</p>
<p>8.5% and led to a refined electron density map of the protein. The Pilatus system is marketed by
</p>
<p>Dectris [22], who also produce the Mythen detector (q.v.) for powder crystallography. Other CMOS
</p>
<p>detectors are manufactured by Dexela [23].
</p>
<p>With shutterless data collection, images are still produced and are effectively shuttered by the
</p>
<p>instrument readout. The CCD has pixels defined by a taper of fiber optic elements and the chip is read
</p>
<p>out as a whole after the shutter is closed. For the CMOS detector each pixel has its own circuitry and
</p>
<p>is read out directly and more rapidly. Both techniques are suitable for fine-slicing [24] and the data are
</p>
<p>processed similarly.
</p>
<p>5.7.9 Data Processing Software
</p>
<p>A number of software packages are available for processing the data recorded by the oscillation
</p>
<p>technique. These include HKL2000 [25], which has superceded the very popular DENZO, and
</p>
<p>MOSFILM [26], and is supported by CCP4. Both of these programs have excellent indexing routines
</p>
<p>that are based on slightly different versions of the data processing software algorithm [27]. It has been
</p>
<p>recommended that it is sometimes useful to run both of these programs separately for initial
</p>
<p>assessment of a given set of data [28].
</p>
<p>Other software packages are d*TREK [24], sold with MSC X-ray detectors, and PROTUEM,3 a
</p>
<p>commercial program associated with Bruker detectors. The expert system ELVES [29] was devel-
</p>
<p>oped with the intention of taking the user from data collection frames to a protein structure without
</p>
<p>human intervention. Perhaps the lessons learnt in Sect. 11.7 will encourage crystallographers to treat
</p>
<p>this option with caution.
</p>
<p>5.7.10 Detectors and Diffractometers
</p>
<p>The following list indicates some of the devices that are currently available for the measurement of
</p>
<p>X-ray diffraction intensities; it is not exhaustive, and some of them have been mentioned in the above
</p>
<p>discussions.
</p>
<p>PILATUS 6M pixel CMOS detector
</p>
<p>ADSC CCD Q-series detectors
</p>
<p>NOIR-1 lens focused CCD detector (ALS)
</p>
<p>MAR345 IP detector
</p>
<p>MAR555 Flatpanel detector (TFT/Se)
</p>
<p>MAR-MX-series tiled CCD detector
</p>
<p>STOE IP-PSD IP detector
</p>
<p>3 http://www.bruker-axs.com/x8_proteum.html
</p>
<p>222 5 Examination of Single Crystals: Optical and X-Ray Diffraction Practice</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_11#Sec16_11">http://dx.doi.org/10.1007/978-1-4614-3954-7_11#Sec16_11</a></div>
<div class="annotation"><a href="http://www.bruker-axs.com/x8_proteum.html">http://www.bruker-axs.com/x8_proteum.html</a></div>
</div>
<div class="page"><p/>
<p>SIEMENS X100A Multiwire detector
</p>
<p>OXFORD-DIFFRACTION Xcalibur Nova CCD detector
</p>
<p>OXFORD CRYOSYSTEMS CCD detector
</p>
<p>ADSC Quantum 4 CCD detector
</p>
<p>LABORATORY DESIGN GUIDE [30]
</p>
<p>CRYO-CRYSTALLOGRAPHY [31]
</p>
<p>5.7.11 Other Diffractometer Systems
</p>
<p>Bruker D8 Systems
A new system from Bruker, the D8 QUEST, offers a single-crystal diffractometer with a choice of
</p>
<p>monochromatic wavelength source (Ag, Mo, Cu, or Cr). Alternatively, the ImS microfocus source is
</p>
<p>available: it consumes only 30W but is of long life, requires no cooling and produces intensity greater
</p>
<p>than that of the traditional water-cooled rotating anode source; the diffractometer accommodates
</p>
<p>devices for low-temperature diffraction. The goniometer has a &ldquo;sphere of confusion&rdquo; of only 7 mm, so
</p>
<p>that the specimen is always centered in the X-ray beam. Detection is by the PHOTON 100 CMOS
</p>
<p>detector of high gain and sensitivity, and the system incorporates the APEX2 and SHELXTL
</p>
<p>computer packages for structure solution.
</p>
<p>The companion system D8 VENTURE has substantially the same facilities as the QUEST, but
</p>
<p>provides a more spacious enclosure for the equipment. In addition, it offers either a fixed w or a k
</p>
<p>sample stage, giving a high degree of freedom in positioning the sample.
</p>
<p>Rigaku SCXmini System
Among the crystallographic products offered by Rigaku, the SCXmini is the first bench-top diffrac-
</p>
<p>tometer system; it requires minimal training and support. Data collection is carried out by
</p>
<p>the CrystalClear package, and structure solution by Oxford CRYSTALS and SHELXL. Twin crystals
</p>
<p>may be handled by the TwinSolve software.
</p>
<p>In both of these examples of diffractometer systems, full information may be obtained from the
</p>
<p>manufacturers.4,5
</p>
<p>5.8 Monochromators
</p>
<p>In all aspects of diffraction crystallography, except for Laue diffraction, Sects. 5.4.1ff and 11.2.3ff,
</p>
<p>where a continuous spectrum is employed, intensity data are measured with radiations that are as
</p>
<p>close as possible to being monochromatic. We have seen in Sect. 3.1ff that characteristic radiation
</p>
<p>from a sealed tube can be monochromatized effectively by means of an absorption filter, but with
</p>
<p>considerable loss in intensity. In this section, we consider monochromators, which are devices
</p>
<p>designed for the production of effectively single-wavelength X-ray beams. Both the traditional sealed
</p>
<p>X-ray tube and the synchrotron source will be considered.
</p>
<p>The Bragg equation shows that scattered X-radiation occurs at y angles that depend upon the
</p>
<p>wavelength l of the X-radiation. Hence, in selecting y by reflection from a crystal plane, a mono-
</p>
<p>chromatic radiation may be obtained. A small scattering angle is chosen so that the loss of energy due
</p>
<p>to polarization is minimized.
</p>
<p>4 http://www.bruker-axs.com/chemical_crystallography.html.
5 http://www.rigaku.com/smc/scxmini.html.
</p>
<p>5.8 Monochromators 223</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_11#Sec6_11">http://dx.doi.org/10.1007/978-1-4614-3954-7_11#Sec6_11</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec1_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec1_3</a></div>
<div class="annotation"><a href="http://www.bruker-axs.com/chemical_crystallography.html">http://www.bruker-axs.com/chemical_crystallography.html</a></div>
<div class="annotation"><a href="http://www.rigaku.com/smc/scxmini.html">http://www.rigaku.com/smc/scxmini.html</a></div>
</div>
<div class="page"><p/>
<p>5.8.1 Single-Type Crystal Monochromators
</p>
<p>The crystal structure of graphite consists of layers of covalently bonded, planar, hexagonal carbon
</p>
<p>rings stacked perpendicular to the c-axis. The stacking distance c/2 of 3.41 Å approximates to the van
</p>
<p>der Waals distance for carbon. This layered structure maximizes the carbon content of successive
</p>
<p>(0002) planes which thus produces an extremely strong 0002 reflection. The single-type crystal
</p>
<p>graphite monochromator operates by using this reflection as the primary beam. For a copper target,
</p>
<p>there is an approximately 20% greater loss of intensity than that suffered with a nickel filter.
</p>
<p>However, the range of selected wavelengths is very narrow and effectively monochromatic. This
</p>
<p>results in a dramatic improvement in the peak to background ratio, owing to a virtual elimination of
</p>
<p>the general background radiation produced by a sealed-tube source. Reduced heating effects resulting
</p>
<p>in improved crystal life have also been reported for some proteins.
</p>
<p>For a monochromator tuned to wavelength l, the harmonically related wavelengths l/2, l/3,. . .
</p>
<p>will be reflected at the same angle. This could, of course, present problems with the measured
</p>
<p>intensities, which would then relate to a composite of wavelengths. However, in practice, the
</p>
<p>harmonic wavelengths tend to be very weak in intensity and the choice of a structurally suitable
</p>
<p>monochromator material, such as graphite, will further ensure that the corresponding I(h/n, k/n, l/n)
</p>
<p>values are also negligibly small.
</p>
<p>5.8.2 Double-Type Crystal Monochromators
</p>
<p>A disadvantage with the single-type crystal monochromator is that once tuned to select a given
</p>
<p>wavelength, the instrument cannot easily be adapted for use with a different wavelength source,
</p>
<p>because the reflecting angle will be different. This can be overcome by employing a double-type
</p>
<p>crystal monochromator, Fig. 5.31, in which the incident and emergent beam directions are the same.
</p>
<p>Tuning of the instrument is achieved by pivots fitted to both halves of the monochromator. Obviously,
</p>
<p>a further loss of intensity will be suffered with this arrangement, from the second crystal, which
</p>
<p>offsets the above advantage to some extent.
</p>
<p>Fig. 5.31 Double-type crystal monochromator. The incident X-ray beam Xi is reflected as an essentially monochro-
matic component Xr1 from the first crystal M1. The second crystal M2 is set at the same angle and is therefore tuned to
</p>
<p>the same wavelength and reflects the emergent component Xr2. Tuning to a different wavelength is achieved by
</p>
<p>coupled rotation of the two crystals (curved arrows) and maintains the direction of incident and emergent beams, thus
</p>
<p>requiring minimal adjustment of the diffractometer. The second crystal M2 is absent in a single-type crystal
</p>
<p>monochromator
</p>
<p>224 5 Examination of Single Crystals: Optical and X-Ray Diffraction Practice</p>
<p/>
</div>
<div class="page"><p/>
<p>5.8.3 Monochromators for Synchrotron Radiation
</p>
<p>The divergence of the X-ray beam from a synchrotron is small but undesirable. The effect can be
</p>
<p>corrected by using a focusing monochromator, which may be either single- or double-type. A single
</p>
<p>crystal of germanium or silicon is used for this purpose, because it has the additional advantage of a
</p>
<p>much lower bandwidth selection, or wavelength range, compared to graphite. The plane of the crystal
</p>
<p>monochromator is carefully bent in order to produce the focusing effect, which is in one direction
</p>
<p>only, forming a line focus. Further focusing may be achieved by the use of highly polished quartz or
</p>
<p>glass toroidal mirrors. Thin diamond plates, which have a high thermal tolerance, also provide
</p>
<p>excellent mirrors but suffer from the disadvantage of being extremely small [32].
</p>
<p>5.9 Focusing Mirrors
</p>
<p>We have noted that important features of the X-ray beam used in data collection include mono-
</p>
<p>chromatization, small beam divergence, high intensity, and optimal focusing. In macromolecular
</p>
<p>crystallography, a small divergence of the beam enables the use of a larger oscillation range for
</p>
<p>individual exposures without overlap of diffraction spots and consequently fewer exposures in total.
</p>
<p>It will also improve the resolution of individual spots that are close together in a crystal of large unit-
</p>
<p>cell dimensions.
</p>
<p>X-rays can be reflected by mirrors when the angle of incidence is smaller than a critical angle,
</p>
<p>ca. 0.1�, and the use of the Franks&rsquo; double focusing mirror to improve X-ray beam intensity, first
</p>
<p>introduced nearly 50 years ago [33], has been revived recently. This device consists of a pair of
</p>
<p>curved mirrors with perpendicular axes of curvature, which produces a point-focused X-ray beam,
</p>
<p>with consequent high intensity. The first mirror thus focuses the beam in one direction and the
</p>
<p>second mirror focuses the beam in the perpendicular direction to give a small, highly concentrated
</p>
<p>spot size.
</p>
<p>The design and construction of confocal ellipsoidal mirrors for use with microfocus X-ray tubes
</p>
<p>has been described [34]. In this device, two mirrors are glued together in perpendicular arrangement
</p>
<p>at the same distance from the X-ray source, giving a fixed focus beam. Alternatively, G&euro;obel mirrors,
</p>
<p>using parabolic focusing in a sequential double-mirror arrangement, can be used to produce a
</p>
<p>collimated parallel beam.
</p>
<p>The advantages of the use of mirrors include an increased flux by a factor of at least three or
</p>
<p>four compared to a graphite monochromator, a narrow angular divergence and higher brilliance, a
</p>
<p>small spot size and higher brightness, a low background and some degree of monochromatization.
</p>
<p>The disadvantages are a degree of competition between monochromatization and beam flux,
</p>
<p>leading to the use of a nickel filter, for copper radiation, placed in front of the mirror, associated
</p>
<p>problems with alignment of the X-ray generator and filament with a rotating anode tube, and the
</p>
<p>requirement of a helium path to counter air-absorption, because of path lengths of approximately
</p>
<p>200 mm.
</p>
<p>Commercially available mirrors include the following:
</p>
<p>&bull; Franks&rsquo; mirrors: MAR, MAC Science, Nonius XOS, and Charles Supper
</p>
<p>&bull; G&euro;obel mirrors: Bruker AXS
</p>
<p>&bull; Confocal mirrors: Osmic&rsquo;s Max Flux&trade; (Bruker), MSC, MAR, and Nonius (AXS Confocal
</p>
<p>MaxFlux optics)
</p>
<p>5.9 Focusing Mirrors 225</p>
<p/>
</div>
<div class="page"><p/>
<p>5.10 Twinning
</p>
<p>5.10.1 Morphology of Twinning
</p>
<p>Crystals during their formation may undergo some form of growth stress that causes them to continue
</p>
<p>the process of crystallization in other directions. The resultant material is called a twin crystal. A
</p>
<p>twinned crystal exhibits two or more orientations in one and the same crystal fragment. The portions
</p>
<p>occur in a well-defined relationship to one another: composite blocks of unit cells unite in differing
</p>
<p>orientations to form the crystal fragment. Although the portions of the fragment grow simultaneously
</p>
<p>during crystallization, it is convenient to speak of them in terms of an operation about a line, the twin
</p>
<p>axis, or a plane, the twin plane, which would bring the fragments into congruence. In some instances
</p>
<p>crystal fragments may be formed by mechanical deformation after growth, but they behave as
</p>
<p>twinned crystals. A reflection twin may be described as a symmetric twin, and the rotation twin as
</p>
<p>a hemitropic twin. Re-entrant angles are frequently an indication of twinning, whereas the interfacial
</p>
<p>angles on single crystals are salient. The plane across which two portions of a twinned crystal appear
</p>
<p>to be united is termed the composition plane. There is generally no physical discontinuity across this
</p>
<p>plane, merely a change of orientation; often a twinned crystal may be separated across its composition
</p>
<p>plane in order to obtain a single crystal. Another form of twinning is shown by interpenetrant twins,
</p>
<p>on which the crystals making up the fragment appear to have grown through one another. A
</p>
<p>composition plane is no longer possible in such cases, but the orientations of the crystals in the
</p>
<p>fragment are still related by an operation of rotation or reflection.
</p>
<p>In some crystal fragments, the twin plane is also the composition plane; an example is the
</p>
<p>symmetric twin of gypsum, CaSO4�2H2O. The twin may be described in terms of a rotation of
</p>
<p>180� about the X-axis, (100) being the twin plane, Fig. 5.32a. The re-entrant angle is clearly evident
</p>
<p>and the specimen also exhibits an external pseudosymmetry of mm2, the true point-group symmetry
</p>
<p>being 2/m. Another form of gypsum, twined on the same law, is shown in Fig. 5.32b, but this is an
</p>
<p>example of an interpenetrant twin, with mmm pseudosymmetry.
</p>
<p>Fig. 5.32 Twinning in gypsum: (a) Symmetric twin of gypsum, with a (100) twin plane. (b) Interpenetrant twin of
gypsum, also twinned on (100)
</p>
<p>226 5 Examination of Single Crystals: Optical and X-Ray Diffraction Practice</p>
<p/>
</div>
<div class="page"><p/>
<p>The mineral fluorite, CaF2, exhibits interpenetrant twinning, Fig. 5.33a, and also contact twinning,
</p>
<p>about a threefold axis, Fig. 5.33b. Figure 5.33c is an example of naturally occurring fluorite exhibiting
</p>
<p>both interpenetrant and contact twinning.
</p>
<p>The scalenohedral twin habit of calcite is illustrated in Fig. 5.34. It may be regarded as formed by a
</p>
<p>rotation about the vertical, triad axis, and is thus a hemitropic twin. A crystallographic axis of
</p>
<p>symmetry of even degree (2, 4 or 6) cannot function as a twin axis, because no new orientations
</p>
<p>are formed by such rotations. Similarly a crystallographic reflection plane (m) cannot function as the
</p>
<p>twin plane though it may, in some cases, be a composition plane.
</p>
<p>In addition to these types of twinning, multiple twins can occur. The components may take the
</p>
<p>form of lath-like fragments, parallel to the composition plane, the whole crystal being termed a
</p>
<p>lamellar twin.
</p>
<p>It is often possible to identify twinning under the polarizing microscope. Figure 5.35 shows two
</p>
<p>possible situations where adjacent parts of a crystal specimen are not simultaneously in extinction.
</p>
<p>Before proceeding to an X-ray examination, a single crystal may often be extracted from a twin by
</p>
<p>cutting the fragment under the microscope with a sharp, thin (razor) blade and retaining collar,
</p>
<p>Fig. 5.36. Any minute fragments adhering to the crystal can then be removed by judicious use of a
</p>
<p>solvent. The same apparatus can be used to cut crystals generally, as the occasion demands.
</p>
<p>Fig. 5.33 Twinning in fluorite: (a) Interpenetrant twin, about (111). (b) Contact twin of fluorite, also twinned about
(111). (c) Natural specimen of fluorite, showing both interpenetrant and contact twinning
</p>
<p>Fig. 5.34 Scalenohedral habit of calcite, CaCO3, twinned about the vertical, triad axis (compare the scalenohedron
in Fig. 1.3)
</p>
<p>5.10 Twinning 227</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Fig3_1">http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Fig3_1</a></div>
</div>
<div class="page"><p/>
<p>5.10.2 Twinning and X-Ray Diffraction
</p>
<p>The consequence of twinning in an X-ray study is the appearance of more than one set of diffraction
</p>
<p>data for the &ldquo;crystal&rdquo; under examination. The two lattices may coincide, so that the two sets of
</p>
<p>reflections are superimposed in position and intensity. If the twin law is a symmetry operation of the
</p>
<p>crystal system but not of the point group of the crystal,merohedral twinning, such as twinning on {110}
</p>
<p>in point group
4
</p>
<p>m
, occurs, and this twinning will simulate point group
</p>
<p>4
</p>
<p>m
mm, and reflections hkl and khl
</p>
<p>will be superimposed because of the apparent higher symmetry of the holosymmetric point group.
</p>
<p>A special case of merohedral twinning is racemic twinning, in which a non-centrosymmetric
</p>
<p>structure is twinned with its inversion, thus leading to a superposition of reflections hkl and h k l,
</p>
<p>which are not equivalent under Friedel&rsquo;s law, see also Sect. 7.6. The twin operation may not be of the
</p>
<p>crystal system but of a higher symmetry system, pseudomerohedral twinning, to which the structure
</p>
<p>approximates. A common example would be a crystal in the monoclinic system with a b-angle close
</p>
<p>to 90�. In such a case the two sets of reflections hkl; hk l; h kl; h k l and hkl; h kl; hk l; hkl each
</p>
<p>equivalent and separate under Friedel&rsquo;s law, but distinct in symmetry
2
</p>
<p>m
, are all equivalent and
</p>
<p>superposed under symmetry mmm.
</p>
<p>Inmerohedral and pseudomerohedral twinning, the X-ray diffraction pattern can normally be indexed
</p>
<p>on a single unit cell, but indications of twinning arise in space group determination, intensity statistics or
</p>
<p>in the structure determination proper. Where the components do not overlap, non-merohedral twinning,
</p>
<p>problems with the unit-cell determination may arise, leading to a doubling of one axis or more.
</p>
<p>One way of avoiding problems with twinning may be by its early recognition. Interpenetrant twins
</p>
<p>may be recognized under the microscope by their shape, and their use avoided. Contact twins can
</p>
<p>Fig. 5.35 Thin sections of twinned crystals viewed in polarized light. (a) Simple contact twin. (b) Lamellar twin. In
each case, adjacent parts of the crystal are not in extinction simultaneously
</p>
<p>Fig. 5.36 Crystal cutting apparatus: B is a half razor blade, M is a metal or plastic collar and C is the crystal under
examination
</p>
<p>228 5 Examination of Single Crystals: Optical and X-Ray Diffraction Practice</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_7#Sec31_7">http://dx.doi.org/10.1007/978-1-4614-3954-7_7#Sec31_7</a></div>
</div>
<div class="page"><p/>
<p>often be determined with the microscope under polarized light. The two portions of the fragment may
</p>
<p>appear in extinction or show polarization colors at different positions of the microscope stage. If the
</p>
<p>boundary of two crystals comprising the fragment can be recognized, the crystal may be cut using the
</p>
<p>simple apparatus shown in Fig. 5.36: then one is again dealing with a single crystal.
</p>
<p>The SHELX structure solution program (see Appendix D) incorporates facilities (BASF and
</p>
<p>TWIN) that enable tests to be carried out in the penultimate stages of least-squares refinement, and
</p>
<p>interesting accounts of twinning may be found on the web site references below [35, 36].
</p>
<p>5.11 Problems
</p>
<p>5.1. Crystals of KH2PO4 are needle-shaped and show straight extinction parallel to the needle axis.
</p>
<p>A Laue photograph taken with the X-rays parallel to the needle axis shows symmetry 4mm.
</p>
<p>(a) What is the crystal system and Laue group, and how is the optic axis oriented?
</p>
<p>(b) Describe and explain the appearance between crossed Polaroids of a section cut perpendic-
</p>
<p>ular to the needle axis.
</p>
<p>(c) What minimum symmetry, would be observed on both general and symmetric oscillation
</p>
<p>photographs taken with the crystal mounted on the needle axis?
</p>
<p>5.2. Crystals of acetanilide (C8H9NO) are brick-shaped parallelepipeda, showing straight extinction
</p>
<p>for sections cut normal to each of the three edges of the &ldquo;brick&rdquo;
</p>
<p>(a) What system would you assign to the crystals?
</p>
<p>(b) Allocate suitable crystallographic axes.
</p>
<p>(c) What minimum symmetry would be shown by general oscillation photographs taken, in
</p>
<p>turn, about each of the three crystallographic axes?
</p>
<p>(d) What symmetrywould an oscillation photograph exhibit where the crystal is oscillating about
</p>
<p>the a axis such that b is parallel to the X-ray beam at the center of the oscillation range?
</p>
<p>5.3. Crystals of sucrose show the extinction directions indicated on the schematic crystal drawing of
</p>
<p>Fig. P5.1; the arrows indicate the directions of the cross-wires at extinction.
</p>
<p>(a) To what crystal system does sucrose belong?
</p>
<p>(b) How are the morphological directions, p, q, and r related to the crystallographic axes?
</p>
<p>(c) How would you mount the crystal in order to test your conclusions with (i) Laue photo-
</p>
<p>graphs, in a single mounting of the crystal, and (ii) oscillation photographs? In each case,
</p>
<p>indicate the symmetry you would expect the photographs to exhibit in the orientations you
</p>
<p>have chosen.
</p>
<p>5.4. A cubic crystal of side a &frac14; 5.0 Å is mounted on a flat-plate Laue camera so that one axis (b or
c) lies along the rotation axis and a is inclined to the X-ray beam at an angle f &frac14; 70�. If the
crystal is irradiated with an X-ray beam possessing the wavelength range 0.2&ndash;2.5 Å, determine
</p>
<p>Fig. P5.1 Crystal section
of sucrose, showing
</p>
<p>extinction directions
</p>
<p>5.11 Problems 229</p>
<p/>
</div>
<div class="page"><p/>
<p>the highest order h00 reflection that can be recorded. If the cassette is placed at a distance R of
</p>
<p>60 mm from the crystal and the film plate is 125 mm2, determine whether this reflection will be
</p>
<p>recorded and, if so, find its coordinates on the film in mm.
</p>
<p>5.5. A Laue photograph of a protein crystal is recorded on a pack of two films separated by black
</p>
<p>paper. A particular reflection hkl for a wavelength l in the &ldquo;white&rdquo; radiation is overlapped on
</p>
<p>the films by the reflection 2h, 2k, 2l for the wavelength l/2. The black paper transmits 65% of
</p>
<p>the radiation of shorter wavelength and 35% of the longer wavelength radiation. If the relative
</p>
<p>intensity of the composite reflection measured on the film first to receive the reflected beam is
</p>
<p>300 but only 130 on the second film, what are the relative intensities of the two reflections on
</p>
<p>the first film?
</p>
<p>5.6. b-Zinc sulphide, ZnS, crystallizes in space group F�43m, with a &frac14; 5.41 Å. A flat-plate Laue
photograph, taken with a crystal-to-film distance of 30.00 mm, exhibits symmetry 2mm.
</p>
<p>(a) What was the direction of the incident beam in the crystal?
</p>
<p>(b) The horizontal m line on the film displays two pairs of reflections, symmetrically disposed
</p>
<p>about the center of the film. The two reflections of the outer pair are 77.5 mm apart, and the
</p>
<p>inner pair 43.5 mm apart. For each reflection in the two pairs, find (i) the Bragg angle y, (ii)
</p>
<p>the indices hkl and (iii) the wavelength producing each spot.
</p>
<p>5.7. Oscillation photographs of manganese carbonate, MnCO3, taken about three mutually perpen-
</p>
<p>dicular axes, and Laue photographs along the same three axes gave the following results
</p>
<p>(l &frac14; 1.5418 Å):
</p>
<p>Axis 1 2 3
</p>
<p>Layer line 8 4 2
</p>
<p>Distance/mm
</p>
<p>from zero layer
</p>
<p>38.40 33.63 25.40
</p>
<p>Laue symmetry 3m m 2
</p>
<p>Determine the unit-cell dimensions and the Laue group of the substance; the diameter of the
</p>
<p>oscillation camera was 60.00 mm.
</p>
<p>5.8. The 0002 reflection from a single crystal of graphite can be used to obtain monochromatic
</p>
<p>X-radiation from its (0001) plane. If the c dimension of graphite is 6.696 Å, calculate the
</p>
<p>appropriate Bragg angle for reflection of (a) Cu Ka (l &frac14; 1.5418 Å) and (b) Mo Ka (l &frac14; 0.7107
Å) radiation at which the crystal monochromator must be set.
</p>
<p>5.9. An orthorhombic crystal was set in a random orientation on an automatic four-circle single-
</p>
<p>crystal X-ray diffractometer. A peak search led to a primitive reciprocal unit cell of dimensions
</p>
<p>a* &frac14; b* &frac14; 0.239, c &frac14; 0.184, a* &frac14; b* &frac14; 90, g* &frac14; 152.4�, using Mo Ka radiation (l &frac14; 0.7107
Å). Determine the transformation matrix to convert the &ldquo;diffractometer&rdquo; cell to the true cell,
</p>
<p>which was known from photographs to have the approximate dimensions a &frac14; 3.1 Å, b &frac14; 12.5
</p>
<p>Å, and c &frac14; 3.9 Å, (a) in real space and (b) in reciprocal space.
</p>
<p>5.10. (a) The diffraction pattern from a protein crystal is required to be recorded on an image plate
</p>
<p>having a diameter of 345 mm. If the resolution dmin required is 1.0 Å and an X-ray
</p>
<p>wavelength of 1.05 Å, is being used, calculate the furthest distance R in mm that the
</p>
<p>plate can be set from the crystal.
</p>
<p>(b) What would happen to the diffraction pattern in (a) if the plate were set 10 mm nearer to the
</p>
<p>crystal?
</p>
<p>(c) What would happen to the diffraction pattern in (a) if the plate were set 10 mm further from
</p>
<p>the crystal?
</p>
<p>5.11. Refer to Fig. P5.2a&ndash;c. Each figure poses a problem relating to X-ray records.
</p>
<p>230 5 Examination of Single Crystals: Optical and X-Ray Diffraction Practice</p>
<p/>
</div>
<div class="page"><p/>
<p>Fig. P5.2 Diffraction images from a given crystal, with a fixed wavelength, but with varying oscillation angle,
crystal&ndash;detector distance, and exposure time</p>
<p/>
</div>
<div class="page"><p/>
<p>References
</p>
<p>1. Bibliography, Gay (1967); Hartshorne and Stuart (1970)
</p>
<p>2. See Bibliography, Buerger MJ (2009)
</p>
<p>3. Ladd M, Palmer R (2003) Structure determination by X-ray crystallography, 4th edn. Kluwer Academic, New York
</p>
<p>4. http://www.bruker-axs.com/spacegroupdetermination.html
</p>
<p>5. See Bibliography, Jeffrey
</p>
<p>6. Arndt UW, Wonacott AJ (eds) (1977) The rotation method in crystallography. North-Holland, Amsterdam
</p>
<p>7. Mar Research (MAR IP) Norderstedt, Germany
</p>
<p>8. Miyahara J et al (1986) Nucl Instrum Methods A246:572
</p>
<p>9. Boyle W, Smith G (1970) Charge coupled semiconductor devices. Bell Syst Tech J 49:587
</p>
<p>10. Einstein A (1905) Ann Phys 17:132
</p>
<p>11. Gruner SM (1994) Curr Opin Struct Biol 4:765
</p>
<p>12. Westbrook EM, Naday I (1987) Methods Enzymol 276:244
</p>
<p>13. Jones C (2001) Chem Commun 2293
</p>
<p>14. Deacon A et al (1997) J Chem Soc Farad Trans 93:4305
</p>
<p>15. Walter RL et al (1995) Structure 3:835
</p>
<p>16. http://www.nonius.com
</p>
<p>17. http://www.bruker-axs.com
</p>
<p>18. Oxford Cryosystems, http://www.oxcryo.com
</p>
<p>19. Prince SM, Bryant PK (private communication) Manchester University
</p>
<p>20. Br&euro;onnimann Ch et al (2004) Synchrotron Radiat News 17:23
21. Br&euro;onnimann Ch et al (2006) J Synchrotron Radiat 13:120&ndash;130
22. http://www.dectris.ch/sites/products.html
</p>
<p>23. http://www.dexela.com
</p>
<p>24. Plugrath JW (1999) Acta Crystallogr D55:1718
</p>
<p>25. Otwinowski Z, Minor W (2001) &lsquo;Denzo&rsquo; and &lsquo;Scalepack 211&rsquo;. in Arnold, E et al. (2012) (eds) International
</p>
<p>Tables for Crystallography, vol. F, I.U.Cr.
</p>
<p>26. Leslie A (1993) Data collection and processing. In: Sawyer L, Isaacs N, Bailey S (eds) Proceedings of the CCP4
</p>
<p>study weekend. Daresbury Laboratories, Warrington, UK, pp 44&ndash;51
</p>
<p>27. Rossmann MG, van Beek CG (1999) Acta Crystallogr D55:1631
</p>
<p>28. Sanderson MR (2007) Macromolecular crystallography conventional and high-throughput methods, Chapter 5.
</p>
<p>Oxford University Press, Oxford
</p>
<p>29. Yarris Lynn, lcyarris@lbl.gov
</p>
<p>30. http://www.xtl.ox.ac.uk/equip.html
</p>
<p>31. http://people.mbi.ucla.edu/sawaya/m230d/Data/data.html
</p>
<p>32. Freund AK (1996) Structure 4:155
</p>
<p>33. Franks A (1955) Proc Phys Soc London B68:1054
</p>
<p>34. Bloomer AC, Arndt UW (1999) Acta Crystallogr D55:1672
</p>
<p>35. http://www.lcm3b.uhp-nancy.fr/mathcryst/twins.htm
</p>
<p>36. http://xrayweb.chem.ou.edu/notes/twin.html
</p>
<p>37. http://www.bruker-axs.com/d8_advance.html
</p>
<p>38. http://www.rigaku.com/downloads/journal/Vol12.2.19 95/raxisiv.pdf
</p>
<p>Bibliography: Crystal Optics
</p>
<p>Gay P (1967) An introduction to crystal optics. Longmans, London
</p>
<p>Hartshorne NH, Stuart A (1970) Crystals and the polarising microscope. Arnold, London
</p>
<p>X-Ray Scattering and Reciprocal Lattice
</p>
<p>Arndt UW, Wonacott AJ (eds) (1977) The rotation method in crystallography: data collection from macromolecular
</p>
<p>crystals. North-Holland, Amsterdam
</p>
<p>232 5 Examination of Single Crystals: Optical and X-Ray Diffraction Practice</p>
<p/>
<div class="annotation"><a href="http://www.bruker-axs.com/spacegroupdetermination.html">http://www.bruker-axs.com/spacegroupdetermination.html</a></div>
<div class="annotation"><a href="http://www.nonius.com">http://www.nonius.com</a></div>
<div class="annotation"><a href="http://www.bruker-axs.com">http://www.bruker-axs.com</a></div>
<div class="annotation"><a href="http://www.oxford-diffraction.com">http://www.oxford-diffraction.com</a></div>
<div class="annotation"><a href="http://www.dectris.ch/sites/products.html">http://www.dectris.ch/sites/products.html</a></div>
<div class="annotation"><a href="http://www.dexela.com">http://www.dexela.com</a></div>
<div class="annotation"><a href="http://www.xtl.ox.ac.uk/equip.html">http://www.xtl.ox.ac.uk/equip.html</a></div>
<div class="annotation"><a href="http://people.mbi.ucla.edu/sawaya/m230d/Data/data.html">http://people.mbi.ucla.edu/sawaya/m230d/Data/data.html</a></div>
<div class="annotation"><a href="http://www.lcm3b.uhp-nancy.fr/mathcryst/twins.htm">http://www.lcm3b.uhp-nancy.fr/mathcryst/twins.htm</a></div>
<div class="annotation"><a href="http://xrayweb.chem.ou.edu/notes/twin.html">http://xrayweb.chem.ou.edu/notes/twin.html</a></div>
<div class="annotation"><a href="http://www.bruker-axs.com/d8_advance.html">http://www.bruker-axs.com/d8_advance.html</a></div>
<div class="annotation"><a href="http://www.rigaku.com/downloads/journal/Vol12.2.1995/raxisiv.pdf">http://www.rigaku.com/downloads/journal/Vol12.2.1995/raxisiv.pdf</a></div>
</div>
<div class="page"><p/>
<p>Buerger MJ (2009) X-ray crystallography, 11th edn. Wiley, New York
</p>
<p>Jeffrey JW (1971) Methods in X-ray crystallography. Academic, London
</p>
<p>Woolfson MM (1997) An introduction to X-ray crystallography, 2nd edn. Cambridge University Press, Cambridge
</p>
<p>Interpretation of X-Ray Diffraction Photographs
</p>
<p>Henry NFM, Lipson H, Wooster WA (1960) The interpretation of X-ray diffraction photographs. Macmillan, London
</p>
<p>Jeffery JW (1971) Methods in X-ray crystallography. Academic, London
</p>
<p>Precession Method
</p>
<p>Buerger MJ (1964) The precession method. Wiley, New York
</p>
<p>Ladd MFC, Palmer RA (2003) Structure determination by X-ray crystallography, 4th edn. Kluwer Academic,
</p>
<p>New York
</p>
<p>Diffractometry
</p>
<p>Arndt UW, Willis BTM (1966) Single crystal diffractometry. Cambridge University Press, Cambridge
</p>
<p>Cryo-Crystallography
</p>
<p>Ealick S et al. Cryocrystallography: a practical approach http://people.mbi.ucla.edu/Sawaya/m230d/Data/data.htmlhtml
</p>
<p>Macromolecular Crystallography
</p>
<p>Carter CW Jr, Sweet RM (eds) (1997) Macromolecular crystallography. Academic, New York
</p>
<p>Arnold, E et al. (2012) (eds) International Tables for Crystallography, vol. F, I.U.Cr.
</p>
<p>Data Collection
</p>
<p>Evans PR (1999) Acta Crystallogr D55:1771
</p>
<p>Dauter Z (1999) Acta Crysatallogr D55:1703
</p>
<p>Data Collection Services
</p>
<p>The EPSC National Data Collection Service (University of Southampton) will collect, process, and optionally solve
</p>
<p>crystal structures (but there&rsquo;s not much fun in that) for clients who supply suitable crystals. In a recent use of this
</p>
<p>service the following facilities were used
</p>
<p>Diffractometer: Nonius Kappa CCD Area Detector (f scans and o scans to fill the asymmetric unit sphere)
</p>
<p>Cell determination: DirAx (Duisenberg AJM (1992) J Appl Crystallogr 25:92)
</p>
<p>Data collection: Collect (Data collection software, Hooft R, Nonius BV (1998))
</p>
<p>Data reduction and cell refinement: Denzo (Otwinowski Z, Minor W (1997) Methods Enzymol 276); Macromol
</p>
<p>Crystallogr, loc. cit.
</p>
<p>Absorption correction: SORTAV (Blessing RH (1995) Acta Crystallogr A51:33); idem., J Appl Crystallogr 30:421 (1997)
</p>
<p>Structure solution: SHELXS-97 (Sheldrick GM (1990) Acta Crystallogr A46:467)
</p>
<p>Structure refinement: SHELXL-97 (Sheldrick GM (1997) University of G&euro;ottingen)
Graphics: ORTEP3 for windows (Farrugia LJ (1997) J Appl Crystallogr 30:365)
</p>
<p>References 233</p>
<p/>
<div class="annotation"><a href="http://people.mbi.ucla.edu/Sawaya/m230d/Data/data.html">http://people.mbi.ucla.edu/Sawaya/m230d/Data/data.html</a></div>
</div>
<div class="page"><p/>
<p>Fourier Series and Fourier Transforms 6
</p>
<p>6.1 Image Formation and Focusing
</p>
<p>In Sect. 5.3, we touched upon an analogy between the scattering of X-rays and that of visible light.
</p>
<p>Here, we extend that discussion to consider aspects of Fourier series and Fourier transforms that are
</p>
<p>germane to our study of X-ray diffraction.
</p>
<p>The formation of the optical image of an object involves first, a scattering of light from it, and then
</p>
<p>a recombination of the scattered light rays to form the image. We could, for example, prepare a
</p>
<p>transparency of the sodium chloride structure in Fig. 1.6 and project it on to a screen, thus revealing
</p>
<p>an enlarged image of the model of the structure. If we now take the lens out of the projector, there will
</p>
<p>be just a patch of light on the screen, even though the object, the transparency, is still in the same
</p>
<p>position. All the information provided by the transparency is still present in the beam of light, but it is
</p>
<p>not immediately decipherable. The lens has no information about the transparency, but once it is in
</p>
<p>the correct position in the projector the image becomes clear. The lens rearranges the scattered light
</p>
<p>so as to be understandable to us.
</p>
<p>The process of focusing is complex, but we perform it by adjusting the position of the lens until the
</p>
<p>image assumes our expectation of the object. Evidently, some foreknowledge of the probable
</p>
<p>appearance of the object is needed, and we assume that the presence of sharply defined boundaries
</p>
<p>in the image is a condition of being in focus. If we do not have the necessary foreknowledge, we must
</p>
<p>determine the relative positions of transparency, lens and screen by the methods of geometrical
</p>
<p>optics. It is possible to show that the scattered radiation is everywhere within the beam of scattered
</p>
<p>light by moving through it with a hand lens and a piece of white card, although the resulting image
</p>
<p>will not be quite as satisfactory as that obtained with the projector lens system. If we cannot obtain
</p>
<p>recognizable sharp boundaries in an image, we need an aid to focusing. One simple practical method,
</p>
<p>sometime used in microscopy, would be to have a minute speck of dust adhering to the surface of the
</p>
<p>transparency. When the speck is in focus, it can be assumed that the whole image is also in focus.
</p>
<p>Visible light can resolve separations in an object down to approximately 2500 Å, if a high-quality
</p>
<p>microscope is employed. Resolution and radiation wavelength go hand-in-hand. The human eye can
</p>
<p>observe two objects as separate entities provided that they are no less than approximately 0.15 mm
</p>
<p>apart. The Rayleigh formula for the limit of resolution, or resolving power, R of a microscope is
</p>
<p>given as
</p>
<p>R &frac14; 0:61l=&eth;n sin y&THORN; &frac14; 0:61l=N (6.1)
</p>
<p>M. Ladd and R. Palmer, Structure Determination by X-ray Crystallography:
</p>
<p>Analysis by X-rays and Neutrons, DOI 10.1007/978-1-4614-3954-7_6,
# Springer Science+Business Media New York 2013
</p>
<p>235</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_5#Sec8_5">http://dx.doi.org/10.1007/978-1-4614-3954-7_5#Sec8_5</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Fig 6_1">http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Fig 6_1</a></div>
</div>
<div class="page"><p/>
<p>where 2y is the angle of scatter, n is the refractive index of the medium between the lens and the
</p>
<p>specimen, and N is the numerical aperture of the objective. For a microscope working dry, N can
</p>
<p>reach 0.95, but up to 1.46 with oil immersion. Hence, taking a mean l for visible light as 5800 Å, R is
</p>
<p>approximately 3725 Å, or 2425 Å, corresponding to magnifications of 400 and 600, respectively.
</p>
<p>More recently, higher resolution has been demonstrated by scanning near-field optical microscopy
</p>
<p>(SNOM) [1, 2], and resolution as low as 2 nm has been reported.
</p>
<p>The refractive index of materials for X-rays is different from unity by about 10�6, so that from
</p>
<p>(6.1) wavelengths in the range 1&ndash;1.5 Å, are required in order to resolve an atomic separation of 1&ndash;2 Å;
</p>
<p>such wavelengths are obtained with X-rays or neutrons. Although these radiations can be used to
</p>
<p>resolve atomic detail, they can be focused only by special systems of curved mirrors, which lead to
</p>
<p>impracticably low magnification. With X-rays or neutrons, the scattered radiation must be recom-
</p>
<p>bined by calculation.
</p>
<p>The electron microscope can provide resolution of atomic detail, provided that the structure is not
</p>
<p>too complex. The theoretical limit of resolution with an accelerating voltage of 100 kV is about 0.8 Å,
</p>
<p>but spherical aberration of the focusing system may prevent this value from being reached. High-
</p>
<p>resolution transmission electron microscopy (HRTEM) has been developed with very small crystals
</p>
<p>(&lt;0.1 mm) and has attained resolution in the range 0.8&ndash;0.5 Å.
</p>
<p>Certain simplifications exist in the applications of these radiations to crystal structures compared
</p>
<p>with examination in visible light. The regular packing of atoms and molecules in crystals and a
</p>
<p>restriction of the radiation used to a monochromatic source together give rise to a spot diffraction
</p>
<p>pattern rather than a diffuse patch.
</p>
<p>As we have remarked earlier, Sect. 3.2, a close analogy to two-dimensional X-ray diffraction is the
</p>
<p>view of a sodium street-lamp through a fine, stretched gauze, such as a handkerchief or umbrella;
</p>
<p>these materials simulate a two-dimensional lattice, or net. The spot pattern that is obtained is invariant
</p>
<p>under translation of the object, but rotates as the object is rotated; we shall return to these two
</p>
<p>properties later.
</p>
<p>To form an image from the diffraction pattern, the scattered radiation must be recombined in both
</p>
<p>amplitude and phase. The lens system used with visible light enables this process to be carried out
</p>
<p>directly. With X-rays or neutrons, however, not only can the focusing not be done directly, but the
</p>
<p>important phase information required is not obtained explicitly from the experimental procedure: we
</p>
<p>record I(hkl), but from Fig. 3.10, we see that jF&eth;hkl&THORN;j2 &frac14; &frac12;A0&eth;hkl&THORN; &thorn; B0&eth;hkl&THORN;�2 &frac14; jF&eth;hkl&THORN;j2cos2 f&eth;hkl&THORN;
&thorn;jF&eth;hkl&THORN;j2sin2f&eth;hkl&THORN; &frac14; I&eth;hkl&THORN;, so that information about f(hkl) is not given directly by the
measured data.
</p>
<p>In crystal structure analysis, we use Fourier series to carry out the focusing process that is
</p>
<p>automatic with the lens system of a visible-light microscope. In the next section, we shall assume
</p>
<p>that, for the moment, we have the necessary phase information; the acquisition of this phase
</p>
<p>information will be the subject of much of the remainder of this book.
</p>
<p>6.2 Fourier Series
</p>
<p>We wish to consider in detail the function C(X) in Fig. 6.1: it is continuous, single-valued, and
</p>
<p>periodic in the repeat distance a. The mathematician Jean Baptiste Fourier (1768&ndash;1830) solved
</p>
<p>problems on thermal conduction by means of a series consisting of an ideally infinite number of
</p>
<p>cosine and sine terms. Such series had been used earlier by Euler and Bernoulli to investigate
</p>
<p>periodicity in vibrating strings and in astronomy.
</p>
<p>According to the Fourier theorem,C(X) can be represented by a series of cosine and sine terms that
</p>
<p>may be written conveniently as
</p>
<p>236 6 Fourier Series and Fourier Transforms</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3# Sec8_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3# Sec8_3</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Fig 10_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Fig 10_3</a></div>
</div>
<div class="page"><p/>
<p>C&eth;X&THORN; &frac14;
X1
</p>
<p>H&frac14;�1
C&eth;H&THORN; cos&eth;2pHX=a&THORN; &thorn; S&eth;H&THORN; sin&eth;2pHX=a&THORN;f g (6.2)
</p>
<p>which is a trigonometric or Fourier series; C and S are amplitude terms, and the integer index H, that
</p>
<p>defines the term of this number in this series, is a wave number, or reciprocal wavelength, that is, the
</p>
<p>number of times its own wavelength fits into the repeat period.
</p>
<p>Physical phenomena that are periodic, such as tides or heartbeats, can be represented by similar
</p>
<p>Fourier series. For our application, we may make use of a one-dimensional analogue of Friedel&rsquo;s law,
</p>
<p>and express C(X) as
</p>
<p>C&eth;X&THORN; &frac14; C&eth;0&THORN; &thorn; 2
(X1
</p>
<p>H&frac14;1
C&eth;H&THORN; cos&eth;2pHX=a&THORN; &thorn; S&eth;H&THORN; sin&eth;2pHX=a&THORN;
</p>
<p>)
(6.3)
</p>
<p>In order to find the coefficients C(H) and S(H), we integrate both sides of (6.3) with respect to X,
</p>
<p>between the limits 0 and a for one period, andmake use of the fact that the integral of a sum is equal to the
</p>
<p>sum of the integrals of the separate parts:
</p>
<p>&eth;a
</p>
<p>0
</p>
<p>C&eth;X&THORN; dX &frac14;
&eth;a
</p>
<p>0
</p>
<p>C&eth;0&THORN; dX &thorn; 2
X1
</p>
<p>H&frac14;1
</p>
<p>&eth;a
</p>
<p>0
</p>
<p>C&eth;H&THORN;&eth;cos 2pHX=a&THORN; dX &thorn; 2
</p>
<p>�
X1
</p>
<p>H&frac14;1
</p>
<p>&eth;a
</p>
<p>0
</p>
<p>S&eth;H&THORN;&eth;sin 2pHX=a&THORN; dX (6.4)
</p>
<p>We see readily that
&ETH; a
0
cos&eth;2pHX=a&THORN; dX &frac14; &eth;a=2pH&THORN; sin&eth;2pHX=a&THORN;ja0 &frac14; 0, since H is an integer; a
</p>
<p>similar result arises for the sine function. Hence,
&ETH; a
0
C&eth;X&THORN; dX &frac14; C&eth;0&THORN;a, so that
</p>
<p>C&eth;0&THORN; &frac14;
1
</p>
<p>a
</p>
<p>&eth;a
</p>
<p>0
</p>
<p>C&eth;X&THORN; dX (6.5)
</p>
<p>In order to evaluate general expressions for C(H) and S(H), we multiply both sides of (6.2) by
</p>
<p>cos 2pKX=a, where K is another integer, greater than zero:
</p>
<p>&eth;a
</p>
<p>0
</p>
<p>C&eth;X&THORN; cos&eth;2pKX=a&THORN; dX &frac14;
</p>
<p>&eth;a
</p>
<p>0
</p>
<p>(
C&eth;0&THORN; &thorn; 2
</p>
<p>X1
</p>
<p>H&frac14;1
</p>
<p>&frac12;C&eth;H&THORN;&eth;cos 2pHX=a&THORN; &thorn; S&eth;H&THORN; sin&eth;2pHX=a&THORN;�
</p>
<p>)
</p>
<p>� cos&eth;2pKX=a&THORN; dX (6.6)
</p>
<p>Fig. 6.1 One-dimensional periodic function C(X) of repeat a
</p>
<p>6.2 Fourier Series 237</p>
<p/>
</div>
<div class="page"><p/>
<p>Using again the rule of the integral of a sum, we have
</p>
<p>&eth;a
</p>
<p>0
</p>
<p>C&eth;X&THORN; cos&eth;2pKX=a&THORN; dX &frac14; C&eth;0&THORN;
&eth;a
</p>
<p>0
</p>
<p>cos&eth;2pKX=a&THORN; dX &thorn; 2
X1
</p>
<p>H&frac14;1
C&eth;H&THORN;
</p>
<p>&eth;a
</p>
<p>0
</p>
<p>cos&eth;2pHX=a&THORN;
</p>
<p>� cos&eth;2pKX=a&THORN; dX &thorn; 2
X1
</p>
<p>H&frac14;1
</p>
<p>S&eth;H&THORN;
</p>
<p>&eth;a
</p>
<p>0
</p>
<p>sin&eth;2pHX=a&THORN;
</p>
<p>� cos&eth;2pKX=a&THORN; dX (6.7)
</p>
<p>The terms involving the integral of the product of the cosine and sine functions are zero for all
</p>
<p>values of H and K, as discussed in Web Appendix WA8. However, the integral
</p>
<p>&eth;a
</p>
<p>0
</p>
<p>cos&eth;2pHX=a&THORN; cos&eth;2pKX=a&THORN; dX (6.8)
</p>
<p>is zero for K 6&frac14; H, but for K &frac14; H, it becomes
</p>
<p>&eth;a
</p>
<p>0
</p>
<p>cos2&eth;2pHX=a&THORN; dX (6.9)
</p>
<p>which is readily shown to be equal to a/2. Thus, from (6.7)
</p>
<p>&eth;a
</p>
<p>0
</p>
<p>C&eth;X&THORN; cos&eth;2pKX=a&THORN; dX &frac14; 2fC&eth;K&THORN;a=2g &frac14; C&eth;K&THORN;a (6.10)
</p>
<p>Solving for C(K) and then replacing K by H, we obtain from (6.4)
</p>
<p>C&eth;H&THORN; &frac14;
1
</p>
<p>a
</p>
<p>&eth;a
</p>
<p>0
</p>
<p>C&eth;X&THORN; cos&eth;2pHX=a&THORN; dX (6.11)
</p>
<p>Similarly, if we multiply both sides (6.2) by sin 2pKX=a and follow through the same procedure,
</p>
<p>we obtain
</p>
<p>S&eth;H&THORN; &frac14;
1
</p>
<p>a
</p>
<p>&eth;a
</p>
<p>o
</p>
<p>C&eth;X&THORN; sin&eth;2pHX=a&THORN; dX (6.12)
</p>
<p>Thus, if the form of the function C(X) is known, the coefficients C and S can be evaluated. We
</p>
<p>shall carry out this process for the square wave shown in Fig. 6.2, and subsequently extend these
</p>
<p>arguments to two- and three-dimensional functions.
</p>
<p>6.2.1 Analysis of the Square Wave
</p>
<p>Let the square-wave function C(X), shown in Fig. 6.2, be defined in the range �p � X � p, with a
</p>
<p>repeat of 2p. For X &lt; 0,C(X) &frac14; 0, and for 0 &lt; X � p,C(X) &frac14; p; we shall see from the analysis that
</p>
<p>C(X) &frac14; p/2 at X &frac14; 0. Hence, from (6.11) and (6.12),
</p>
<p>238 6 Fourier Series and Fourier Transforms</p>
<p/>
</div>
<div class="page"><p/>
<p>C&eth;h&THORN; &frac14; 1
2p
</p>
<p>&eth;p
</p>
<p>0
</p>
<p>p cos 2ph
X
</p>
<p>2p
</p>
<p>� �
dX (6.13)
</p>
<p>and
</p>
<p>S&eth;h&THORN; &frac14; 1
2p
</p>
<p>&eth;p
</p>
<p>0
</p>
<p>p sin 2ph
X
</p>
<p>2p
</p>
<p>� �
dX (6.14)
</p>
<p>Integration gives the result that for h 6&frac14; 0, C(h) &frac14; 0, whereas for h &frac14; 0, C&eth;h&THORN; &frac14; 1
2
</p>
<p>&ETH; p
0
dx &frac14; p=2. In a
</p>
<p>similar manner, for h &frac14; 0, S(h) &frac14; 0, but for h 6&frac14; 0, S&eth;h&THORN; &frac14; &eth;1=2h&THORN;&frac12;1� cos&eth;hp&THORN;�. Substituting these
results in (6.3), we find
</p>
<p>C&eth;X&THORN; &frac14; p=2&thorn; 2
X1
</p>
<p>h&frac14;1
</p>
<p>&eth;1=2h&THORN;&frac12;1� cos&eth;ph&THORN;� sin&eth;hX&THORN;
</p>
<p>or
</p>
<p>C&eth;X&THORN; &frac14; p=2&thorn; 2
X1
</p>
<p>h&frac14;1
&eth;h&frac14;2n&thorn;1&THORN;
</p>
<p>&eth;1=h&THORN; sin&eth;hX&THORN; (6.15)
</p>
<p>since the term [1 � cos(ph)] is zero for even values of h. Hence, finite values of C(X) arise for
</p>
<p>h &frac14; 2n&thorn; 1 &eth;n &frac14; 0; 1; 2; . . .&THORN;.
</p>
<p>Range of X
In (6.15), the variable X defines a sampling point in any repeat interval between X &frac14; �p.
For convenience, we will choose a zero arbitrarily at X &frac14; 0 and sample the function at intervals
</p>
<p>of 2pn/50, that is, we shall calculate the function from n &frac14; 0 to 50. The results will show that
</p>
<p>we could, and in general would, make use of reflection symmetry in the function at
</p>
<p>X &frac14; &eth;2m&thorn; 1&THORN;p=2; &eth;m &frac14; 0; 1; 2; . . . ;1&THORN; in order to decrease the amount of calculation.
</p>
<p>Range of h
The summations of Fourier series extend, theoretically, from �1 to 1. In practice, however, the
</p>
<p>range becomes hmin � h � hmax, where the limits of h are set by the experimental conditions. In this
</p>
<p>Fig. 6.2 Square wave
C(X), defined in the range
</p>
<p>�p to p, with a repeat of 2p
</p>
<p>6.2 Fourier Series 239</p>
<p/>
</div>
<div class="page"><p/>
<p>example hmin is unity and hmax values of 3 and 7 are used; the results of summing (6.15) are presented in
</p>
<p>Fig. 6.3. It is notable, even with these few numbers of terms, that increasing hmax has a dramatic effect
</p>
<p>on the series. As hmax increases so the series (6.15) approaches more closely the square-wave function
</p>
<p>in Fig. 6.2. In general, the more independent terms that can be included in a Fourier series, the better it
</p>
<p>represents the periodic function under investigation, from which the terms have been derived.
</p>
<p>The process of determining the coefficients of a Fourier series is called Fourier analysis, and the
</p>
<p>process of reconstructing the function by the summation of a series such as (6.15) is Fourier synthesis.
</p>
<p>A microscope, in forming an image of an object, effectively performs a Fourier synthesis of the
</p>
<p>scattered light, a process that we shall have occasion to refer to again later.
</p>
<p>6.2.2 Exponential Forms of Fourier Series
</p>
<p>Let a function G(h) be defined by
</p>
<p>G&eth;h&THORN; &frac14; C&eth;h&THORN; &thorn; iS&eth;h&THORN;
G&eth;�h&THORN; &frac14; C&eth;h&THORN; � iS&eth;h&THORN;
</p>
<p>(6.16)
</p>
<p>so that G is of the nature of the structure factor, F(h). Multiplying G(h) and G(�h) by exp(�2phX/a),
and using de Moivre&rsquo;s theorem &frac12;exp&eth;�iX&THORN; &frac14; cos&eth;X&THORN; � i sin&eth;X&THORN;� it follows that
</p>
<p>G&eth;h&THORN; exp&eth;�i2phX=a&THORN; &thorn; G&eth;�h&THORN; exp&eth;�i2p&eth;�h&THORN;X=a&THORN;
</p>
<p>&frac14; &frac12;C&eth;h&THORN; &thorn; iS&eth;h&THORN;�&frac12;cos&eth;2phX=a&THORN; � i sin&eth;2phX=a&THORN;� &thorn; &frac12;C&eth;h&THORN; � iS&eth;h&THORN;�&frac12;cos&eth;2phX=a&THORN;
</p>
<p>&thorn; i sin&eth;2phX=a&THORN;� &frac14; 2C&eth;h&THORN; cos&eth;2phX=a&THORN; &thorn; 2S&eth;h&THORN; sin&eth;2phX=a&THORN;
</p>
<p>(6.17)
</p>
<p>Fig. 6.3 Square waves
calculated from (6.15).
</p>
<p>(a) hmax &frac14; 3, (b) hmax &frac14; 7.
The positive and negative
</p>
<p>fluctuations of the
</p>
<p>calculated function arise
</p>
<p>because there are
</p>
<p>insufficient terms to
</p>
<p>provide good convergence
</p>
<p>of the Fourier series; they
</p>
<p>are known as series
</p>
<p>termination errors
</p>
<p>240 6 Fourier Series and Fourier Transforms</p>
<p/>
</div>
<div class="page"><p/>
<p>From the foregoing and (6.3) we can write C(X) as
</p>
<p>C&eth;X&THORN; &frac14; G&eth;0&THORN; &thorn; 1
2
</p>
<p>X1
</p>
<p>h&frac14;1
fjG&eth;h&THORN;j exp&eth;�i2phX=a&THORN; &thorn; jG&eth;�h&THORN;j exp&eth;i2phX=a&THORN;g (6.18)
</p>
<p>or
</p>
<p>C&eth;X&THORN; &frac14;
X1
</p>
<p>h&frac14;�1
jG&eth;h&THORN;j exp&eth;�i2phX=a&THORN; (6.19)
</p>
<p>The multiplier of 1
2
and the term G(0) in (6.18) disappear in (6.19) because the summation limits
</p>
<p>are now �1 through zero to 1, whereas in (6.18) the positive and negative values of h appear
explicitly under the summation. Following Sect. 6.2, we can show that
</p>
<p>G&eth;h&THORN; &frac14; 1
a
</p>
<p>&eth;a
</p>
<p>0
</p>
<p>C&eth;X&THORN; exp&eth;i2phX=a&THORN; dX (6.20)
</p>
<p>Equations (6.19) and (6.20) are Fourier transforms of each other, a topic that we shall consider in
</p>
<p>more detail shortly; the signs of the exponents should be noted.
</p>
<p>6.3 Fourier Series in X-Ray Crystallography
</p>
<p>The lattice basis of a crystal structure introduces a three-dimensional periodicity which pervades its
</p>
<p>properties, including the electron density distribution. The square wave that we have just analyzed
</p>
<p>may be likened to a one-dimensional crystal, or the projection of a crystal structure on to a single axis.
</p>
<p>The first applications of Fourier series in crystallography were with one-dimensional series, and for
</p>
<p>good reason. We must consider first the significance of the functions C(X) and G(h).
</p>
<p>We have shown in Sect. 3.2ff how X-rays are scattered by the electrons associated with atoms in a
</p>
<p>crystal. The concentration of electrons and their distribution around an atom is called the electron
</p>
<p>density. At any point X, Y, Z the electron density may be written as r(XYZ), and we may identify
</p>
<p>r(XYZ) as a three-dimensional analogue of the function C(X) in (6.19), and we must now determine
</p>
<p>the meaning of G(h).
</p>
<p>We consider electrons in an atom as though they were concentrated at a point, but specify their
</p>
<p>distribution by a shape factor, the atomic scattering factor f, which is an amplitude term. The
</p>
<p>exponential term in (6.19) represents the phase of the wave of frequency index h in its contribution
</p>
<p>to C(X), with respect to origin of the unit cell.
</p>
<p>6.3.1 One-Dimensional Function
</p>
<p>The one-dimensional electron density function r(X) shown in Fig. 6.4 was calculated from a small
</p>
<p>number of experimental terms. In a small interval dX along the X axis, the electron density may be
</p>
<p>regarded as being constant, so that the associated electron count is r&eth;X&THORN; dX. Its contribution to the hth
structure factor F(h) is given, following Sect. 3.2.3, by r&eth;X&THORN; exp&eth;i2phX&THORN; dX,where exp(i2phX) is the
phase associated with r&eth;X&THORN; dX with respect to the origin. The contribution from the whole repeat
period, the structure factor F(h), is now given by
</p>
<p>6.3 Fourier Series in X-Ray Crystallography 241</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3# Sec8_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3# Sec8_3</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3# Sec11_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3# Sec11_3</a></div>
</div>
<div class="page"><p/>
<p>F&eth;h&THORN; &frac14;
&eth;a
</p>
<p>0
</p>
<p>r&eth;X&THORN; exp&eth;i2phX=a&THORN; dX (6.21)
</p>
<p>This equation is a generalized one-dimensional structure factor; analogous expressions can
</p>
<p>be written in two and three dimensions. Substituting for r(X), equivalent to C(X), from (6.19),
</p>
<p>we have
</p>
<p>F&eth;h&THORN; &frac14;
&eth;a
</p>
<p>0
</p>
<p>X1
</p>
<p>h0&frac14;�1
jG&eth;h0&THORN;j exp&eth;�i2ph0X=a&THORN; exp&eth;i2phX=a&THORN; dX (6.22)
</p>
<p>where h0 indicates the range of values of h under the summation sign. Since the integral of a sum is
equal to the sum of the integrals of the separate terms, we write
</p>
<p>F&eth;h&THORN; &frac14;
X1
</p>
<p>h0&frac14;�1
jG&eth;h0&THORN;j
</p>
<p>&eth;a
</p>
<p>0
</p>
<p>exp&frac12;i2p&eth;h� h0&THORN;X=a� dX (6.23)
</p>
<p>The integral evaluates to a=i2p&eth;h� h0&THORN; expfi2p&eth;h� h0&THORN;&eth;X=a&THORN;g. Since both h and h0 are integers,
</p>
<p>the numerator of this expression is zero, except when h0 &frac14; h. In this special case, we can see from
</p>
<p>(6.23) that the integral becomes
&ETH; a
0
dX, which has the value a. It follows immediately that
</p>
<p>G&eth;h&THORN; &frac14; F&eth;h&THORN;=a, so that from (6.19) we write
</p>
<p>r&eth;X&THORN; &frac14;
1
</p>
<p>a
</p>
<p>X1
</p>
<p>h&frac14;�1
</p>
<p>jF&eth;h&THORN;j exp&eth;�i2phX=a&THORN; (6.24)
</p>
<p>which is the Fourier transform of (6.21).
</p>
<p>From Fig. 3.13 we can write
</p>
<p>F&eth;h&THORN; &frac14; A&eth;h&THORN; &thorn; iB&eth;h&THORN;
</p>
<p>F&eth;�h&THORN; &frac14; A&eth;h&THORN; � iB&eth;h&THORN;
(6.25)
</p>
<p>Fig. 6.4 One-dimensional
electron density projection
</p>
<p>r(X) for pyrite, FeS2
</p>
<p>242 6 Fourier Series and Fourier Transforms</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Fig 13_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Fig 13_3</a></div>
</div>
<div class="page"><p/>
<p>so that (6.24) may be written as
</p>
<p>r&eth;X&THORN; &frac14; 1
a
</p>
<p>(
F&eth;0&THORN; &thorn;
</p>
<p>X1
</p>
<p>h&frac14;1
&frac12;A&eth;h&THORN; &thorn; iB&eth;h&THORN;�&frac12;cos&eth;2phX=a&THORN; � i sin&eth;2phX=a&THORN;� &thorn;
</p>
<p>X1
</p>
<p>h&frac14;1
</p>
<p>&frac12;A&eth;h&THORN; � iB&eth;h&THORN;�
</p>
<p>� &frac12;cos&eth;2phX=a&THORN; &thorn; i sin&eth;�2phX=a&THORN;�
</p>
<p>)
(6.26)
</p>
<p>which reduces to
</p>
<p>r&eth;X&THORN; &frac14;
1
</p>
<p>a
</p>
<p>(
F&eth;0&THORN; &thorn; 2
</p>
<p>X1
</p>
<p>h&frac14;1
</p>
<p>&frac12;A&eth;h&THORN; cos&eth;2phX=a&THORN; &thorn; B&eth;h&THORN; sin&eth;2phX=a&THORN;�
</p>
<p>)
(6.27)
</p>
<p>which may be compared with (6.3).
</p>
<p>6.3.2 Two- and Three-Dimensional Functions
</p>
<p>Analogous summation expressions can be formulated in two and three dimensions. We will consider
</p>
<p>the two-dimensional case in detail, as the program system XRAY described in Sect. 13.4 uses this
</p>
<p>form of the Fourier series. First, we state, by analogy with (6.24), and using fractional coordinates x,
</p>
<p>y, z, the three-dimensional electron density equation as
</p>
<p>r&eth;xyz&THORN; &frac14;
1
</p>
<p>Vc
</p>
<p>X1
</p>
<p>h&frac14;�1
</p>
<p>X1
</p>
<p>k&frac14;�1
</p>
<p>X1
</p>
<p>l&frac14;�1
</p>
<p>jF&eth;hkl&THORN;j exp&frac12;�i2p&eth;hx&thorn; ky&thorn; lz&THORN;� (6.28)
</p>
<p>Since, from Fig. 3.13, A&eth;hkl&THORN; &frac14; jF&eth;hkl&THORN;j cosf&eth;hkl&THORN; and B&eth;hkl&THORN; &frac14; jF&eth;hkl&THORN;j sinf&eth;hkl&THORN;, it follows
</p>
<p>from the form of (6.27) and Web Appendix WA5 equation (WA5.6), that r(xyz) may be written as
</p>
<p>r&eth;xyz&THORN; &frac14;
1
</p>
<p>Vc
</p>
<p>X1
</p>
<p>h&frac14;�1
</p>
<p>X1
</p>
<p>k&frac14;�1
</p>
<p>X1
</p>
<p>l&frac14;�1
</p>
<p>jF&eth;hkl&THORN;j cos&frac12;2p&eth;hx&thorn; ky&thorn; lz&THORN; � f&eth;hkl&THORN;� (6.29)
</p>
<p>which serves to show how the electron density depends upon the phase angles: only jF(hkl)j is
</p>
<p>measured through the experimental procedure; f(hkl) must be determined before (6.29) can be
</p>
<p>summed. This situation constitutes the phase problem in crystallography, of which we shall
</p>
<p>hear more.
</p>
<p>Equation (6.28) can be rewritten in the form of (6.27), which is more convenient for calculation,
</p>
<p>by again making use of Friedel&rsquo;s law, namely that jF&eth;�h�k �l&THORN;j &frac14; jF(hkl&THORN;j, or A&eth;�h�k �l&THORN; &frac14; A&eth;hkl&THORN; and
</p>
<p>B&eth;�h�k�l&THORN; &frac14; B&eth;hkl&THORN;. Thus, we obtain
</p>
<p>r&eth;xyz&THORN; &frac14;
1
</p>
<p>Vc
</p>
<p>(
F&eth;000&THORN; &thorn; 2
</p>
<p>X1
</p>
<p>h&frac14;1
</p>
<p>X1
</p>
<p>k&frac14;�1
</p>
<p>X1
</p>
<p>l&frac14;�1
</p>
<p>A&eth;hkl&THORN; cos&frac12;2p&eth;hx&thorn; ky&thorn; lz&THORN;�
</p>
<p>&thorn; B&eth;hkl&THORN; sin&frac12;2p&eth;hx&thorn; ky&thorn; lz&THORN;�
</p>
<p>) (6.30)
</p>
<p>6.3 Fourier Series in X-Ray Crystallography 243</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_13#Sec6_13">http://dx.doi.org/10.1007/978-1-4614-3954-7_13#Sec6_13</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Fig 13_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Fig 13_3</a></div>
</div>
<div class="page"><p/>
<p>From the structure factor equation and the Bragg equation, it follows that F(0,0,0) &frac14;P
j &eth;fj&THORN;y&frac14;0 &frac14; Z, the total number of electrons in the unit cell. It is normally an integer quantity, but
</p>
<p>need not be for a non-stoichiometric compound.
</p>
<p>Again, this equation could be put in the form of (6.29) if desired, and variations can be developed
</p>
<p>according to space-group symmetry, as discussed in Sect. 3.5.1ff for the structure factor equation.
</p>
<p>We consider the two-dimensional electron density equation. The generalized structure factor
</p>
<p>F(hkl) is given by
</p>
<p>F&eth;hkl&THORN; &frac14; V
&eth;1
</p>
<p>0
</p>
<p>&eth;1
</p>
<p>0
</p>
<p>&eth;1
</p>
<p>0
</p>
<p>r&eth;xyz&THORN; exp&frac12;i2p&eth;hx&thorn; ky&thorn; lz&THORN;� dx dy dz (6.31)
</p>
<p>we are using the integration limits 0&ndash;1 because we are now using the fractional coordinates x, y, and z,
</p>
<p>relating to a single unit cell. For a projection along the z axis, we need the F(hk0) reflections, where
</p>
<p>F&eth;hk0&THORN; &frac14; Vc
</p>
<p>&eth;1
</p>
<p>0
</p>
<p>&eth;1
</p>
<p>0
</p>
<p>(&eth;1
</p>
<p>0
</p>
<p>r&eth;xyz&THORN; dz
</p>
<p>)
exp&frac12;i2p&eth;hx&thorn; ky&thorn; lz&THORN;� dx dy (6.32)
</p>
<p>To interpret the integral over z, consider an element of structure of cross-sectional area dx dy and
</p>
<p>length c along the z axis. In an element of length dz, the electron content is r(xyz) dz, so that the total
</p>
<p>electron content in the element of length c is c dx dx
&ETH; 1
0
r&eth;xyz&THORN; dz. Hence, the projected electron
</p>
<p>density at a point x, y is given by
</p>
<p>r&eth;xy&THORN; &frac14; c
</p>
<p>&eth;1
</p>
<p>0
</p>
<p>r&eth;xyz&THORN; dx dy dz
</p>
<p>from which F(hk0), or F(hk) becomes
</p>
<p>F&eth;hk&THORN; &frac14; A
</p>
<p>&eth;1
</p>
<p>0
</p>
<p>&eth;1
</p>
<p>0
</p>
<p>r&eth;xy&THORN; exp&frac12;�i2p&eth;hx&thorn; ky&THORN;� dx dy (6.33)
</p>
<p>since
&ETH; 1
0
exp&eth;�i2plz&THORN; dz is zero unless l &frac14; 0; A is the area of the a, b face of the unit cell. It follows
</p>
<p>that
</p>
<p>r&eth;xy&THORN; &frac14;
1
</p>
<p>A
</p>
<p>X1
</p>
<p>h&frac14;�1
</p>
<p>X1
</p>
<p>k&frac14;�1
</p>
<p>jF&eth;hk&THORN;j exp&frac12;�i2p&eth;hx&thorn; ky&THORN;� (6.34)
</p>
<p>which may be written more conveniently as
</p>
<p>r&eth;xy&THORN; &frac14;
1
</p>
<p>A
</p>
<p>(
F&eth;00&THORN; &thorn; 2
</p>
<p>X1
</p>
<p>h&frac14;1
</p>
<p>X1
</p>
<p>k&frac14;�1
</p>
<p>fA&eth;hk&THORN; cos 2p&eth;hx&thorn; ky&THORN; &thorn; B&eth;hk&THORN; sin 2p&eth;hx&thorn; ky&THORN;g
</p>
<p>)
(6.35)
</p>
<p>We have defined F(000) above: in projection on to a plane or line, the electron density in the unit
</p>
<p>cell is projected on to that plane or line; hence, F(000) &frac14; F(00) &frac14; F(0). Unless one is concerned with
</p>
<p>absolute values of electron density, these terms can be omitted and the electron density scaled to a
</p>
<p>convenient maximum value. In neutron diffraction, F(000) counts the total scattering power in the
</p>
<p>unit cell, and is a positive or negative quantity.
</p>
<p>244 6 Fourier Series and Fourier Transforms</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec23_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec23_3</a></div>
</div>
<div class="page"><p/>
<p>We conclude this section by a consideration of the one-dimensional electron density function
</p>
<p>which we discussed in Sect. 6.2. We draw Fig. 6.1 again as Fig. 6.5, now with the amplitudes and
</p>
<p>phases of the waves shown. Following the arguments above, and using a one-dimensional analogue of
</p>
<p>(6.29), we can write the electron density function in terms of amplitude and phase as
</p>
<p>r&eth;x&THORN; &frac14; 1
5
f20&thorn; 5 cos&frac12;2p&eth;x&THORN; � 3:023� &thorn; 2:5 cos&frac12;2p&eth;2x&THORN; � 2:400� &thorn; 4 cos&frac12;2p&eth;3x&THORN; � 4:060�g
</p>
<p>This equation would then be evaluated at a suitable interval x, 20ths of awould be appropriate, and
</p>
<p>the function plotted. Practice with Fourier series follows from the Problems section.
</p>
<p>6.3.3 Units of Electron Density
</p>
<p>A consideration of (6.28), (3.29), and (3.63), shows that the units of r(xyz) are reciprocal volume&mdash;a
</p>
<p>number (of electrons) density. The often-quoted eÅ�3 unit for electron density is incorrect: integra-
</p>
<p>tion of electron density figure field of a peak in r(xyz) gives the number of electrons associated with
</p>
<p>the atoms of that peak.
</p>
<p>6.4 Holes and Atoms
</p>
<p>When visible light is incident upon a circular hole in an otherwise opaque card, its diffraction
</p>
<p>pattern is somewhat diffuse. The theory of this scattering is complex, but it can be simplified by
</p>
<p>considering the pattern at an ideally infinite distance, the Fraunhofer diffraction pattern of the hole.
</p>
<p>Fig. 6.5 The periodic function of Fig. 6.1, with amplitudes and phases for waves of indices 0, 1, 2, and 3; the
amplitudes for the indices 1, 2, and 3 have been given twice their weight according to (6.27) for r(X). The portion Amay
</p>
<p>be regarded as the electron density r(X) with a repeat distance a of 5 Å. The portion B corresponds to the structure
</p>
<p>factors, or harmonics, with their phases. X-ray diffraction may be regarded as the path A ! B. Structure determination
is the path B ! A, and the need for correct phases is clear. The values for the phases (which must be in radians) given
are based on their own wavelengths, 0, a, 2a, and 3a in order to comply with the definition of phase, Sect. 3.2.3
</p>
<p>6.4 Holes and Atoms 245</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Eq29_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Eq29_3</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Eq63_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Eq63_3</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec11_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec11_3</a></div>
</div>
<div class="page"><p/>
<p>Figure 6.6 shows a simple schematic experimental arrangement for viewing a Fraunhofer diffrac-
</p>
<p>tion pattern. Parallel light from a laser source S is incident upon the object hole at O, which must be
</p>
<p>smaller than the diameter of the laser beam. The diffraction pattern can be viewed or photographed
</p>
<p>at F, the back focal plane of the lens L. The whole of the radiation is not completely in phase at F,
</p>
<p>and its intensity falls off with the distance from F. The larger the hole, the more rapid is the fall off,
</p>
<p>since the addition of waves at a given distance fromF is then less complete. The diffraction pattern of
</p>
<p>the hole has circular symmetry, and it may be represented by a radial distribution function of the form
</p>
<p>A&eth;y&THORN; &frac14; 2J1&eth;X&THORN;=X (6.36)
</p>
<p>where A(y) is the amplitude of the scattered radiation at an angle y to the plane of the hole, J1 is a first-
</p>
<p>order Bessel function, and X is a function that is proportional to the radius of the hole and to sin y=l.
</p>
<p>Figure 6.7a illustrates the diffraction pattern of a hole; Fig. 6.7b is a plot of (6.36), normalized to
</p>
<p>A(y) &frac14; 6 at sin y=l &frac14; 0, together with the atomic scattering factor curve for carbon, atomic number
6. Whereas A(y) alternates in sign, the atomic scattering factor f does not, but at low values of sin y=l
</p>
<p>the two curves are closely similar. The diffraction pattern of the hole is its Fourier transform, and
</p>
<p>(6.36) is a mathematical representation of the transform. Similarly, f is the transform of an atom, and
</p>
<p>the f-curve shows its variation with sin y=l.
</p>
<p>6.5 Generalized Fourier Transform
</p>
<p>We refer back to Fig. 3.9, and the development thereof. Let the three-dimensional electron density for
</p>
<p>the body at the point A, distant jrj from the origin, be r(r) with respect to a single electron at the origin
O. An element of volume dV around A has an electron content of r(r)dV, and its phase with respect to
O is 2p r�S. Hence, the contribution of the quantity r(r)dV to scattering in the direction y is
</p>
<p>r&eth;r&THORN; exp&frac12;i2p&eth;r � S&THORN;�dV (6.37)
</p>
<p>Then, the total scattering for the body is
</p>
<p>&eth;
</p>
<p>V
</p>
<p>r&eth;r&THORN; exp&frac12;i2p&eth;r � S&THORN;� dV (6.38)
</p>
<p>where the integral extends over the volume V of the body. This expression is the Fourier transform of
</p>
<p>the body, and may be written
</p>
<p>Fig. 6.6 Simple experimental arrangement for producing Fraunhofer diffraction: S helium&ndash;neon laser source; O
object; L lens (focal length ca. 1 m); F back focal plane of lens L, where the diffraction pattern may be recorded
</p>
<p>(reproduced from Taylor and Lipson 1964)
</p>
<p>246 6 Fourier Series and Fourier Transforms</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Fig 9_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Fig 9_3</a></div>
</div>
<div class="page"><p/>
<p>G&eth;S&THORN; &frac14;
&eth;
</p>
<p>V
</p>
<p>r&eth;r&THORN; exp&frac12;i2p&eth;r � S&THORN;� dV (6.39)
</p>
<p>Following Sect. 3.2.3, we have
</p>
<p>I&eth;S&THORN; &frac14; G&eth;S&THORN;G�&eth;S&THORN; (6.40)
</p>
<p>so that I(S) is the intensity of the transform. For an atom, we may usually assume spherical symmetry,
so that the Fourier transform for an atom is just f(S), where f is the atomic scattering factor. Since
centrosymmetry is a subgroup of spherical symmetry
</p>
<p>f &eth;S&THORN; &frac14; f �&eth;S&THORN; (6.41)
</p>
<p>Fig. 6.7 Scattering from a
circular hole. (a) Diffraction
pattern (Airy disc) for a
</p>
<p>hole of 5 mm diameter.
</p>
<p>(b) Scattered amplitude
function A&eth;y&THORN; for a given
hole radius, normalized
</p>
<p>to equal the value of f
</p>
<p>for carbon at sin y=l
(reproduced from Taylor
</p>
<p>and Lipson 1964)
</p>
<p>6.5 Generalized Fourier Transform 247</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec11_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec11_3</a></div>
</div>
<div class="page"><p/>
<p>so that
</p>
<p>f &eth;S&THORN; &frac14; &frac12; f &eth;S&THORN; f �&eth;S&THORN;�1=2 &frac14; &frac12; f 2&eth;S&THORN;�1=2 (6.42)
</p>
<p>In practice, f(S) is defined to be positive for all values of sin y. Strictly, there is a phase change of p
when X-rays are scattered, as we discussed in Sect. 4.1.2, but since it is true for all atoms it is usually
</p>
<p>ignored, and f(S) is given by the Fourier transform of (6.39), where r&eth;r&THORN; &frac14; 4pr2C&eth;r&THORN;, C(r) being an
appropriate radial wave function, or a combinations of wave functions, for the atom. Evidently,
</p>
<p>we evaluated the Fourier transform f(S) surreptitiously in Sect. 3.2.5 (see also Problem 3.3).
</p>
<p>6.5.1 Fourier Transform of a Molecule
</p>
<p>Let the n atoms of a molecule be characterized by the coordinates Xj, Yj, Zj, (j &frac14; 1, 2, 3, . . ., n), each
</p>
<p>with a scattering factor fj(S). The vector rj from the origin to the jth atom is
</p>
<p>rj &frac14; Xj &thorn; Yj &thorn; Zj (6.43)
</p>
<p>From (6.38), the wave scattered by the jth atom at a distance r from it, with respect to the origin, is
given by
</p>
<p>&eth;
</p>
<p>V
</p>
<p>r&eth;r&THORN; exp&frac12;i2p&eth;r&thorn; rj&THORN; � S� dV (6.44)
</p>
<p>or
</p>
<p>&eth;
</p>
<p>V
</p>
<p>r&eth;r&THORN; exp&frac12;i2p&eth;r � S&THORN;� dV exp&frac12;i2p&eth;rj � S&THORN;� (6.45)
</p>
<p>which becomes
</p>
<p>fj&eth;S&THORN; exp&frac12;i2p&eth;rj � S&THORN;� (6.46)
</p>
<p>Thus, the total wave from all n discrete atoms in the molecule is its Fourier transform G(S),
given by
</p>
<p>G&eth;S&THORN; &frac14;
Xn
</p>
<p>j&frac14;1
</p>
<p>fj exp&frac12;i2p&eth;rj � S&THORN;� (6.47)
</p>
<p>For convenience, we write fj for fj(S), because we have already decided that fj is spherically
symmetrical, and its variation with y is understood.
</p>
<p>6.5.2 Fourier Transform of a Unit Cell
</p>
<p>Let xj, yj, zj now be the fractional coordinates of the jth atom in a unit cell, so that
</p>
<p>rj &frac14; xja&thorn; yjb&thorn; zjc (6.48)
</p>
<p>248 6 Fourier Series and Fourier Transforms</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_4#Sec3_4">http://dx.doi.org/10.1007/978-1-4614-3954-7_4#Sec3_4</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec13_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec13_3</a></div>
</div>
<div class="page"><p/>
<p>From the Bragg equation, 2 sin y=l &frac14; 1=d, and the fact that
</p>
<p>S &frac14; d� &frac14; ha� &thorn; kb� &thorn; lc� (6.49)
</p>
<p>we have
</p>
<p>rj � S &frac14; hxj &thorn; kyj &thorn; lzj (6.50)
</p>
<p>Then, the total transform for the unit cell is
</p>
<p>G&eth;S&THORN; &frac14;
Xn
</p>
<p>j&frac14;1
</p>
<p>fj exp&frac12;i2p&eth;hxj &thorn; kyj &thorn; lzj&THORN;� (6.51)
</p>
<p>which is identical to the structure factor equation (3.63) for the reflection hkl, where hkl is related to
</p>
<p>S through (6.49).
The Fourier transform (6.51) is valid for all values of h, k, and l. In a crystal, however, because of
</p>
<p>interference, the transform can be observed only at those points where scattering is reinforced, that is,
</p>
<p>at the reciprocal lattice points hkl, where h, k, and l are necessarily integral. Thus, the structure factor
</p>
<p>equation for a crystal is its Fourier transform sampled at the reciprocal lattice points. We may imagine
</p>
<p>the Fourier transform for a molecule overlaid by the reciprocal lattice in the correct orientation: only
</p>
<p>those points that satisfy the limiting conditions for the space group (and lie within the sphere of
</p>
<p>reflection) could give rise to X-ray reflections.
</p>
<p>6.6 Practice with Transforms
</p>
<p>We can calculate transforms by (6.51), and we can prepare them experimentally by means of
</p>
<p>the optical diffractometer. Both techniques can provide useful results in developing Fourier
</p>
<p>transform theory.
</p>
<p>6.6.1 Optical Diffractometer
</p>
<p>The optical diffractometer permits the preparation of diffraction patterns in a relatively straightfor-
</p>
<p>ward manner. In Fig. 6.8, S is a helium&ndash;neon laser of ca. 50 mW intensity,M1 andM2 are mirrors, E is
</p>
<p>an expander that extends the laser beam without loss of spatial coherence, L1 and L2 are lenses, and O
</p>
<p>is the object. The diffraction pattern is brought to a focus at F, the back focal plane of lens L2, where it
</p>
<p>may be viewed with an eyepiece, photographed or input to a television camera; Figs. 6.7a and 6.9
</p>
<p>were produced with such an instrument.
</p>
<p>6.6.2 Single Hole
</p>
<p>The diffraction pattern for a single hole has been illustrated in Fig. 6.7a, and its transform (6.36), was
</p>
<p>shown in Fig. 6.7b. The diffraction pattern of a circular hole is often called the Airy disc, after its
</p>
<p>discoverer.
</p>
<p>6.6 Practice with Transforms 249</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Eq 63_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Eq 63_3</a></div>
</div>
<div class="page"><p/>
<p>6.6.3 Two or More Holes
</p>
<p>The Fourier transform of a single hole may be represented generally by (6.39), as well as by (6.36).
</p>
<p>A second hole, displaced by a vector distant a from the first hole, has an identical transform function
but with a vector distance r + a from the origin. The total transform GT(S) is the sum of the displaced
and undisturbed transforms:
</p>
<p>GT&eth;S&THORN; &frac14;
&eth;
</p>
<p>V
</p>
<p>r&eth;r&THORN; exp&frac12;i2p&eth;r � S&THORN;� dV &thorn;
&eth;
</p>
<p>V
</p>
<p>r&eth;r&THORN; exp&frac12;i2p&eth;r&thorn; a&THORN; � S� dV
</p>
<p>&frac14;
</p>
<p>&eth;
</p>
<p>V
</p>
<p>r&eth;r&THORN; exp&frac12;i2p&eth;r � S&THORN;� dVf1&thorn; exp&frac12;i2p&eth;a � S&THORN;�g &frac14; G0&eth;S&THORN;f1&thorn; exp&frac12;i2p&eth;a � S&THORN;�g (6.52)
</p>
<p>where G0(S) is the transform of the undisturbed function. The term f1&thorn; exp&frac12;i2p&eth;a � S&THORN;�g is a fringe
functionmodifying G0(S). It has the value 2 when a�S is integral, and is zero halfway between, as can
be shown from de Moivre&rsquo;s theorem. Thus, the total transform is that of a single hole, Figs. 6.7a
</p>
<p>or 6.9a, crossed by a system of planar fringes, Young&rsquo;s fringes, as shown in Fig. 6.9b. The fringe
</p>
<p>Fig. 6.8 Schematic arrangement of an optical diffractometer: S helium&ndash;neon laser source; M1 and M2 mirrors; E
beam expander; O object; L1 and L2 lenses; F back focal plane of lens L2, where the diffraction pattern may be recorded
</p>
<p>(reproduced from Taylor and Lipson 1964)
</p>
<p>Fig. 6.9 Diffraction patterns from circular holes. (a) One hole 1 mm diameter. (b) Two holes each of 1 mm diameter,
set 3 mm apart in the horizontal direction (reproduced from Taylor and Lipson 1964)
</p>
<p>250 6 Fourier Series and Fourier Transforms</p>
<p/>
</div>
<div class="page"><p/>
<p>system lies perpendicular to the direction joining the two holes. As the distance a is increased, the
</p>
<p>distance between the individual fringes becomes smaller, an example of the reciprocal nature of the
</p>
<p>diffraction process. The amplitudes of the fringes vary sinusoidally, alternate fringes having a relative
</p>
<p>phase difference of p.
</p>
<p>The addition of further pairs of holes of different spacings and differing orientations gives rise to
</p>
<p>more fringe systems, all with the same reciprocal property discussed above. The complete diffraction
</p>
<p>pattern of a molecule may be thought of as a superposition of many sets of such fringes. The sequence
</p>
<p>of optical transforms in Fig. 6.10 shows not only an increase in the number of pairs of holes, as with
</p>
<p>Fig. 6.10 Optical
diffraction patterns
</p>
<p>illustrating scattering.
</p>
<p>(a) Two-dimensional
lattice ( portion only).
</p>
<p>(b) Single molecule
(simulated benzene ring).
</p>
<p>(c) Two such molecules.
(d) Four molecules.
(e) A row ( portion only)
of six molecules. (f) A net
(portion only) of
</p>
<p>molecules; this pattern
</p>
<p>resembles closely an X-ray
</p>
<p>diffraction photograph
</p>
<p>(reproduced from Taylor
</p>
<p>and Lipson 1964)
</p>
<p>6.6 Practice with Transforms 251</p>
<p/>
</div>
<div class="page"><p/>
<p>the benzene ring itself, but also the effect of increasing the numbers of benzene ring entities in both
</p>
<p>one and two dimensions.
</p>
<p>6.6.4 Change of Origin
</p>
<p>If the origin to which a scattering species is referred be changed by the addition of a fixed vector p to
all rj vectors, then from (6.47)
</p>
<p>Gp&eth;S&THORN; &frac14;
Xn
</p>
<p>j&frac14;1
fj exp&frac12;i2p&eth;rj &thorn; p&THORN; � S� &frac14;
</p>
<p>Xn
</p>
<p>j&frac14;1
</p>
<p>fj exp&frac12;i2p&eth;rj � S&THORN;� exp&frac12;i2p&eth;p � S&THORN;�
</p>
<p>&frac14; G0&eth;S&THORN; exp&frac12;i2p&eth;p � S&THORN;� (6.53)
</p>
<p>In (6.53), G0(S) is modified by the fringe function exp&frac12;i2p&eth;p � S&THORN;�. Since, in practice, &eth;p � S&THORN; is
integral at points where the transform can be observed, it follows that exp&frac12;i2p&eth;p � S&THORN;� has a magnitude
of unity. Thus, the amplitude (and intensity) of the transform is invariant under translation, which can
</p>
<p>be demonstrated as mentioned in Sect. 6.1. The phase, however, is dependent upon position, from
</p>
<p>which it is apparent that a true position cannot be determined without knowledge of relative phases,
</p>
<p>which is the central problem in crystal structure determination, as we have remarked already.
</p>
<p>6.6.5 Systematic Absences
</p>
<p>At this point, we can show how the Fourier transform of a crystal can predict systematic absences.
</p>
<p>Suppose that the vector displacement a in (6.52) is replaced by &eth;a=2&thorn; b=2&THORN;, consistent with a
C-centered unit cell. Then, we have
</p>
<p>GC&eth;S&THORN; &frac14; G0&eth;S&THORN; 1&thorn; exp i2p 1=2 &eth;a&thorn; b&THORN; � S&frac12; �f g (6.54)
</p>
<p>which, from (6.49), becomes
</p>
<p>GC&eth;S&THORN; &frac14; G0&eth;S&THORN;f1&thorn; exp&frac12;ip&eth;h&thorn; k&THORN;�g (6.55)
</p>
<p>For (h + k) odd, GC(S) is identically zero, but it equals 2G0(S) for (h + k) even, which we
recognize as characteristic for a C-centered unit cell; see also Sect. 3.7. Equations like (6.55) can
</p>
<p>be developed for all translational symmetries. Such results show clearly those reciprocal lattice points
</p>
<p>that cannot be sampled, whatever the nature of the contents of the unit cell.
</p>
<p>6.6.6 Reconstruction of the Image
</p>
<p>Consider again the diffraction pattern of the hypothetical two-dimensional crystal in Fig. 6.10f. The
</p>
<p>lattice in this example is defined by the separations of the simulated benzene molecules. The weighted
</p>
<p>reciprocal lattice is the transform of a single molecule, Fig. 6.10b, crossed by fringes, the principal
</p>
<p>sets of which are governed by the basic translations of the lattice, Fig. 6.10a; the continuous transform
</p>
<p>is decomposed into, or sampled as, a spot pattern.
</p>
<p>252 6 Fourier Series and Fourier Transforms</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec27_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec27_3</a></div>
</div>
<div class="page"><p/>
<p>In reconstructing an image from its diffraction pattern, our object is to attain, say, the structure of
</p>
<p>Fig. 6.10f given only its diffraction pattern, or transform. The fringes themselves are rather like a
</p>
<p>diffraction grating. A set of true sinusoidal fringes itself used as a mask, or grating, would produce a
</p>
<p>diffraction pattern that is the original two-hole object; see Sect. 6.6.3. Thus, in reconstructing the
</p>
<p>object from its diffraction image, we are really seeking the diffraction pattern of the diffraction
</p>
<p>pattern, or the transform of the transform. We may express this result mathematically in the following
</p>
<p>manner.
</p>
<p>The transform of (6.39) for a crystal gives an expression for the distribution of the electron density
</p>
<p>r(r) in the form (note the change of sign of the exponent)
</p>
<p>r&eth;r&THORN; &frac14;
&eth;
</p>
<p>v
</p>
<p>jG&eth;S&THORN;j exp&frac12;�i2p&eth;r � S&THORN;� dv (6.56)
</p>
<p>where the integral extends over a volume n in reciprocal space. Using (6.40) in the context of the unit
</p>
<p>cell of a crystal, we can now write
</p>
<p>r&eth;r&THORN; &frac14;
</p>
<p>&eth;
</p>
<p>v
</p>
<p>jF&eth;hkl&THORN;j exp&frac12;�i2p&eth;r � S&THORN;� dv (6.57)
</p>
<p>Equation (6.56), or the equivalent (6.57), is the Fourier transform of (6.39). Thus, the transform in
</p>
<p>Fig. 6.10f corresponds to the calculation in (6.39), whereas the structure in Fig. 6.10f is the result of
</p>
<p>evaluating (6.56), or (6.57).
</p>
<p>From the (h0l) weighted reciprocal lattice section of the crystalline platinum derivative of
</p>
<p>phthalocyanine, a mask, Fig. 6.11a, was prepared in which the relative amplitudes of the structure
</p>
<p>factors were indicated by the sizes of the holes. When this mask was used in the optical diffrac-
</p>
<p>tometer, the transform obtained was that shown in Fig. 6.11b. Comparison with the electron density
</p>
<p>map from the fully solved structure, Fig. 6.11h, shows that the recombination obtained furnishes a
</p>
<p>good reconstruction of the molecule which, in the crystal, gave the spot pattern from which the
</p>
<p>mask was prepared. One will ask immediately how the phase problem has been overcome in this
</p>
<p>reconstruction, as it appears not to have been even considered.
</p>
<p>We have, in this example, a rather special case. In the crystal of platinum phthalocyanine, the
</p>
<p>space group is P21/c and there are two molecules in the unit cell. Thus, the platinum atom occupies a
</p>
<p>special position at the origin in the asymmetric unit, and makes a positive contribution to all structure
</p>
<p>factors. It is such a heavy atom (high atomic number) that it dominates the contributions from all
</p>
<p>other atoms in the structure so causing all F(h0l) structure factors to have a positive sign (zero phase
</p>
<p>angle). Thus, all h0l structure factors have the same relative phase, and the transform of the
</p>
<p>diffraction pattern gives a true representation of the structure without further reference to phase.
</p>
<p>Figure 6.11c&ndash;g and c0&ndash;g0 shows different portions of the h0l reciprocal lattice section and the
</p>
<p>corresponding transforms. The effect of the cutoff on the resolution is well illustrated.
</p>
<p>Representation of Fourier Transforms
If an object is centrosymmetric and the origin is taken at a center of symmetry, then the Fourier
</p>
<p>transform is real, Sect. 3.6.2, and only one diagram is needed to represent it. This situation exists for
</p>
<p>the h0l section of the transform of platinum phthalocyanine, Fig. 6.11a. If the object is not centro-
</p>
<p>symmetric, then it is necessary to use two diagrams to display the transform, either amplitude and
</p>
<p>phase or real and imaginary parts. The phase, or the real and imaginary components, will vary
</p>
<p>according to the choice of origin, but the amplitudes remain invariant under change of origin;
</p>
<p>see also Sect. 6.6.4.
</p>
<p>6.6 Practice with Transforms 253</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec26_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec26_3</a></div>
</div>
<div class="page"><p/>
<p>Fig. 6.11 Platinum phthalocyanine. (a) Mask of the h0l diffraction pattern. (b) Optical transform of (a), showing a
complete molecule and portions of neighboring molecules in the (projected) crystal structure [3]. (c&ndash;g) Increasingly large
portions of the diffraction pattern. (c0&ndash;g0) Corresponding transforms&mdash;the effect of the cutoff of the pattern on the
resolution in the transform is evident. (h) Electron density contour map. All the diffraction patterns relate to the h0l
data, so that the corresponding transforms are x, z projections in real space. (Reproduced from Taylor and Lipson 1964)</p>
<p/>
</div>
<div class="page"><p/>
<p>6.6.7 Transforms and Inverse Transforms
</p>
<p>Consider a one-dimensional square-wave function defined by
</p>
<p>f &eth;x&THORN; &frac14; 2 for� a=2 � x � a=2
0 for� a=2&gt;x&gt;a=2
</p>
<p>�
</p>
<p>which has the form shown in Fig. 6.12. The Fourier transform in the one dimension x is, from (6.39),
</p>
<p>G&eth;S&THORN; &frac14;
</p>
<p>&eth;a=2
</p>
<p>�a=2
</p>
<p>2 exp&eth;i2pSx&THORN; dx (6.58)
</p>
<p>which is solved readily to give
</p>
<p>G&eth;S&THORN; &frac14; 2 sin&eth;pSa&THORN;=&eth;pS&THORN; &frac14; 2a sin&eth;pSa&THORN;=&eth;pSa&THORN; (6.59)
</p>
<p>This transform, in which the function sin pSa=pSa is typical for the transform of a pulse
</p>
<p>waveform, has the form shown in Fig. 6.13, where G(S) is plotted as a function of S in units of 1/a.
</p>
<p>It may be noted that, characteristically, the length 2/a between the first two nodes on each side of the
</p>
<p>central maximum is the reciprocal of the width of the function f(x). We would have obtained the same
</p>
<p>result if we had used the real part of exp &eth;i2pSx&THORN; in (6.58), that is, 2
&ETH; a=2
0
</p>
<p>2 cos&eth;2pSx&THORN; dx, because
</p>
<p>Fig. 6.12 indicates an even function: f(�x) &frac14; f(x).
</p>
<p>Fig. 6.12 Square wave of amplitude 2.0, defined for the period�a/2 to a/2; f(x) &frac14; 2.0 for � a=2 � x � a=2; f(x) &frac14; 0
for jxj &gt; a/2
</p>
<p>Fig. 6.13 Fourier transform of the square wave in Fig. 6.12. Subsidiary, decreasing maxima arise at intervals
&eth;2n&thorn; 1&THORN;a=2 &eth;n &frac14; 1; 2; . . . &THORN;
</p>
<p>6.6 Practice with Transforms 255</p>
<p/>
</div>
<div class="page"><p/>
<p>Consider next the function
</p>
<p>f &eth;x&THORN; &frac14;
1� 2jxj for� 1
</p>
<p>2
� x � 1
</p>
<p>2
</p>
<p>0 for jxj&gt; 1
2
</p>
<p>(
</p>
<p>which has a saw-tooth wave form, Fig. 6.14. Because of the symmetry of this function about the point
</p>
<p>x &frac14; 0, the Fourier transform of this function has the general form
</p>
<p>G&eth;S&THORN; &frac14; 2
</p>
<p>&eth;0
</p>
<p>�1=2
</p>
<p>&eth;1&thorn; 2x&THORN; cos&eth;2pSx&THORN; dx&thorn; 2
</p>
<p>&eth;1=2
</p>
<p>0
</p>
<p>&eth;1� 2x&THORN; cos&eth;2pSx&THORN; dx (6.60)
</p>
<p>which may be simplified to 2
&ETH; 1=2
0
</p>
<p>cos&eth;2pSx&THORN; dx�4
&ETH; 1=2
0
</p>
<p>x cos&eth;2pSx&THORN; dx. The first of these two integrals
solves to sin pS=pS. Integrating by parts, the second integral gives &eth;1=p2S2&THORN;&frac12;1� cos&eth;pS&THORN;��
</p>
<p>&eth;1=pS&THORN; sin&eth;pS&THORN;, so that the total result is
</p>
<p>G&eth;S&THORN; &frac14; &eth;1=p2S2&THORN;&frac12;1� cos&eth;pS&THORN;� (6.61)
</p>
<p>It is left as an exercise to the reader to plot this transform, as a function of S, (S &frac14; 0, 0.05, 0.10,. . .),
</p>
<p>and to show that (6.61) is obtained also if exp&eth;i2pSx&THORN; is used in place of cos(2pSx) in (6.60). The plot
</p>
<p>of the transform should have a maximum at S &frac14; 0, zero values at S &frac14; 2n (n &frac14; 1,2,3,. . .), and small
decreasing maxima at S &frac14; 2n + 1 (n &frac14; 1,2,3,. . .).
</p>
<p>A function and its Fourier transform are reciprocally related; we noted this feature in studying
</p>
<p>Fourier series in Sect. 6.3. We can illustrate this property by means of two programs, TRANS1 and
</p>
<p>FOUR1D, which are part of the suite of programs that are described in Sects. 13.6.1 and 13.6.3.
</p>
<p>Consider the one-dimensional, periodic function f(X) listed in Table 6.1 at 30ths of the repeat
</p>
<p>distance along the X axis, from n &frac14; 0 to 30/30. It can be represented by a series of sine and cosine
terms with coefficients determined by the program TRANS1, which calculates the Fourier transform
</p>
<p>Fig. 6.14 Saw-toothed wave form for the period �
1
</p>
<p>2
to
</p>
<p>1
</p>
<p>2
; f &eth;x&THORN; &frac14; 1� 2jxj for jxj �
</p>
<p>1
</p>
<p>2
; f &eth;x&THORN; &frac14; 0 for jxj&gt;
</p>
<p>1
</p>
<p>2
</p>
<p>256 6 Fourier Series and Fourier Transforms</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_13# Sec26_13">http://dx.doi.org/10.1007/978-1-4614-3954-7_13# Sec26_13</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_13#Sec28_13">http://dx.doi.org/10.1007/978-1-4614-3954-7_13#Sec28_13</a></div>
</div>
<div class="page"><p/>
<p>of the function. When this program is executed on the data shown in Table 6.1, with hmax &frac14; 10, the
coefficients listed in Table 6.2 are obtained. Note that only the values of the function f(x) are entered
</p>
<p>as data. If these coefficients are used with the program FOUR1D, which calculates a one-dimensional
</p>
<p>Fourier series, the initial function is regained, Table 6.3&mdash;the transform of the transform is the
</p>
<p>original function. Details on the manipulation of these programs are given in Sects. 13.6.1 and 13.6.3.
</p>
<p>However, we mention here that the value of h should be below the critical value of D/2, where D is
</p>
<p>the sampling interval, 30 in this example. The sampling theorem states that if the Fourier transform of
</p>
<p>a function is zero for all frequencies greater than a critical frequency fc, that is, its bandwidth is
</p>
<p>limited to frequencies smaller than fc, then the continuous function can be determined from a
</p>
<p>knowledge of its sampled values. The function in Table 6.1 is not of this character, so that the
</p>
<p>maximum frequency h used should be 1
4
to 1
</p>
<p>3
of the sampling interval, 30. The reader may care to
</p>
<p>investigate this property by using the programs with different values of h. Further reading about
</p>
<p>sampling may be found in the literature [4].
</p>
<p>This problem is rarely manifested in the normal Fourier syntheses of X-ray crystallography. The
</p>
<p>experimental Fo data fall off in magnitude at the higher values of h, k, and l because of the attenuation
</p>
<p>arising from the temperature factor effects on the atomic scattering factors, Sect. 4.1.8. In working
</p>
<p>Table 6.1 Periodic function r(X), listed at 30ths (n) of the repeat distance along the X axis
</p>
<p>n f(x) n f(x) n f(x) n f(x)
</p>
<p>0 50 1 52 2 57 3 63
</p>
<p>4 69 5 76 6 81 7 84
</p>
<p>8 85 9 83 10 79 11 73
</p>
<p>12 66 13 60 14 50 15 42
</p>
<p>16 37 17 32 18 41 19 62
</p>
<p>20 90 21 117 22 126 23 123
</p>
<p>24 113 25 95 26 78 27 65
</p>
<p>28 56 29 51 30 50
</p>
<p>Table 6.2 Coefficients A(h) and B(h), for h &frac14; 0&ndash;10, transformed from r(X) data in Table 6.1 using TRANS1
h A(h) B(h) h A(h) B(h) h A(h) B(h) h A(h) B(h)
</p>
<p>0 71.87 0.00 1 2.22 �4.61 2 �15.08 �2.53 3 0.55 5.84
4 2.14 �1.54 5 �0.99 �0.21 6 0.19 0.36 7 0.15 �0.25
8 �0.26 �0.04 9 �0.07 0.13 10 0.43 0.06
</p>
<p>Table 6.3 Transformation of the coefficients A(h) and B(h) from Table 6.2, using FOUR1D, to give the original
function f(x) of Table 6.1
</p>
<p>n f(x) n f(x) n f(x) n f(x)
</p>
<p>0 50.43 1 51.95 2 56.47 3 63.52
</p>
<p>4 69.15 5 75.50 6 81.72 7 83.67
</p>
<p>8 84.48 9 83.91 10 78.39 11 72.55
</p>
<p>12 67.43 13 59.02 14 50.57 15 43.00
</p>
<p>16 34.66 17 32.62 18 41.49 19 60.90
</p>
<p>20 90.76 21 117.64 22 126.27 23 122.34
</p>
<p>24 112.70 25 95.35 26 77.50 27 65.48
</p>
<p>28 56.99 29 50.54 30 50.43
</p>
<p>6.6 Practice with Transforms 257</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_13# Sec26_13">http://dx.doi.org/10.1007/978-1-4614-3954-7_13# Sec26_13</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_13#Sec28_13">http://dx.doi.org/10.1007/978-1-4614-3954-7_13#Sec28_13</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_4#Sec9_4">http://dx.doi.org/10.1007/978-1-4614-3954-7_4#Sec9_4</a></div>
</div>
<div class="page"><p/>
<p>with the sharpened Patterson function and the E map (q.v.), the sharpening inherent in the coefficients
</p>
<p>in the former, reduces the tendency to zero of the higher order spectra, whereas in the latter the
</p>
<p>coefficients in the Fourier transformation to an E map are both sharpened and significantly decreased
</p>
<p>in number. We consider in later chapters how these cases are treated in practice.
</p>
<p>6.6.8 Delta Function
</p>
<p>Another important function, mentioned in Sects. 4.1.8 and 4.2.2, is the Gaussian distribution
</p>
<p>f &eth;x&THORN; &frac14; &eth;1=k&THORN; exp&eth;�px2=k2&THORN; (6.62)
</p>
<p>where k may be regarded as a width of the function. In finding the Fourier transform of this function,
</p>
<p>we can make use of the fact that it is an even function, so that we use the cosine part of exp(i2pSx) and
</p>
<p>determine twice the sum from zero to infinity, that is,
</p>
<p>1
</p>
<p>k
</p>
<p>&eth;1
</p>
<p>�1
exp&eth;�px2=k2&THORN; exp&eth;i2pSx&THORN; dx &frac14; 2
</p>
<p>k
</p>
<p>&eth;1
</p>
<p>0
</p>
<p>exp&eth;�px2=k2&THORN; cos&eth;2pSx&THORN; dx
</p>
<p>From tables of standard integrals, or otherwise [5],
</p>
<p>&eth;1
</p>
<p>0
</p>
<p>exp&eth;�a2x2 cos&eth;bx&THORN;&THORN; dx &frac14; &eth;p=4a2&THORN;1=2 exp&eth;�b2=4a2&THORN;
</p>
<p>so that the required Fourier transform becomes
</p>
<p>G&eth;S&THORN; &frac14; exp&eth;�pk2S2&THORN; (6.63)
</p>
<p>Thus, the transform of a Gaussian function is another Gaussian, in reciprocal space, of a width 1/k,
</p>
<p>the reciprocal of the width k in real space. The integral
</p>
<p>1
</p>
<p>k
</p>
<p>&eth;1
</p>
<p>�1
exp&eth;�px2=k2&THORN; dx
</p>
<p>evaluates to &eth;1= ffiffiffipp &THORN;G 1
2
</p>
<p>� �
, which is unity (normalized) for all values of k.
</p>
<p>Consider next the function
</p>
<p>d&eth;x&THORN; &frac14; lim
k!0
</p>
<p>&eth;1=k&THORN; exp&eth;�px2=k2&THORN; (6.64)
</p>
<p>This function has the following properties:
</p>
<p>d&eth;x&THORN; &frac14; 0 for x 6&frac14; 01 for x &frac14; 0
</p>
<p>�
</p>
<p>&eth;1
</p>
<p>�1
d&eth;x&THORN; dx &frac14; 1
</p>
<p>258 6 Fourier Series and Fourier Transforms</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_4#Sec9_4">http://dx.doi.org/10.1007/978-1-4614-3954-7_4#Sec9_4</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_4#Sec12_4">http://dx.doi.org/10.1007/978-1-4614-3954-7_4#Sec12_4</a></div>
</div>
<div class="page"><p/>
<p>and is known as the (Dirac) d-function; it corresponds to an infinitely sharp line of unit weight at the
</p>
<p>origin. The shape of the function resembles somewhat that of the atomic scattering factor. As the
</p>
<p>width k in (6.64) tends to zero, so the transform of the d-function tends to unity, and reaches it in the
</p>
<p>limit where k &frac14; 0. We can picture it from Fig. 6.12 but with a height of 1.0: as a ! 0 so f(x) ! d(x);
at a &frac14; 0, f(x) &frac14; d(x) &frac14; 1, that is, the Fourier transform of a d-function, G(S), is unity.
</p>
<p>This result arises also from the general integral
</p>
<p>&eth;1
</p>
<p>�1
f &eth;x&THORN;d&eth;x&THORN; dx &frac14; f &eth;0&THORN; (6.65)
</p>
<p>and if we let f &eth;x&THORN; &frac14; exp&eth;i2pSx&THORN;, then
&eth;1
</p>
<p>�1
exp&eth;i2pSx&THORN;d&eth;x&THORN; dx &frac14; exp&eth;0&THORN; &frac14; 1
</p>
<p>If the d-function is located at x &frac14; x0, we have f &eth;x&THORN; &frac14; d&eth;x� x0&THORN;. The Fourier transform of f(x)
is then
</p>
<p>G&eth;S&THORN; &frac14;
&eth;1
</p>
<p>�1
d&eth;x� x0&THORN; exp&eth;i2pSx&THORN; dx &frac14;
</p>
<p>&eth;1
</p>
<p>�1
d&eth;x&THORN; exp&frac12;i2pS&eth;x&thorn; x0&THORN; dx
</p>
<p>which, from (6.65), leads to
</p>
<p>G&eth;S&THORN; &frac14; exp&eth;i2pSx0&THORN; (6.66)
</p>
<p>Thus, the Fourier transform of a d-function at a point x0 is equal to exp(i2pSx0). When the
</p>
<p>d-function is at the origin, that is, x0 &frac14; 0, its Fourier transform is at unity for all values of S.
Of particular interest is a set of d-functions that define a one-dimensional lattice of spacing a.
</p>
<p>Its Fourier transform will be another set of d-functions of spacing 1/a that define the corresponding
</p>
<p>one-dimensional reciprocal lattice. While our discussions have been confined, for convenience, to
</p>
<p>one-dimensional space, the results are equally true in higher dimensions, and we shall consider such
</p>
<p>applications in later sections.
</p>
<p>6.6.9 Weighted Reciprocal Lattice
</p>
<p>As a final, practical example, we illustrate the power of the Fourier transform with the crystal
</p>
<p>structure of euphenyl iodoacetate, Figs. 1.7 and 1.8. The crystal is monoclinic, with unit-cell
</p>
<p>dimensions a &frac14; 7.260 Å, b &frac14; 11.547 Å, c &frac14; 19.217 Å, b &frac14; 94.10	. There are two molecules in
the unit cell, and systematic absences indicated space group P21 or P21/m. The latter space group is
</p>
<p>not possible, because the two molecules in the unit cell would have to lie on special positions of
</p>
<p>symmetry either m or �1. The chiral nature of the molecule, a tetracyclic triterpene, precludes both of
</p>
<p>these symmetries, so that the two molecules must occupy general positions in space group P21.
</p>
<p>Figure 6.15 is a reconstruction of the X-ray photograph, or weighted reciprocal lattice, of the
</p>
<p>h0l section for this crystal; this section is centric, symmetry p2. Since the iodine atom is very
</p>
<p>much heavier than the other atoms present in the crystal, the unit cell may be imagined, to a first
</p>
<p>approximation, as a pair of iodine atoms distant jdj apart.
</p>
<p>6.6 Practice with Transforms 259</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Fig 7_1">http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Fig 7_1</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Fig 8_1">http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Fig 8_1</a></div>
</div>
<div class="page"><p/>
<p>From Sect. 6.6.3, we would expect to find a set of fringe systems running through the transform,
</p>
<p>arising from the heavy iodine atoms. In Fig. 6.15, the two most obvious fringe systems at an angle of
</p>
<p>85.90	 (b*, or the supplement of the b-angle in the monoclinic system) to each other arise from the
unit-cell translations a and c. In addition, the dashed lines indicate another prominent fringe system,
</p>
<p>that arising from the iodine atoms. Note that the system is more prominent at higher angles, because
</p>
<p>the scattering from heavy atoms falls of relatively much less than that from the lighter atoms present
</p>
<p>in the structure.
</p>
<p>The perpendicular spacing of the fringes is 0.18 reciprocal units which, because Cu KaX-radiation
</p>
<p>(l &frac14; 1.5418 Å) was used in the experiment, corresponds to an I&ndash;I spacing of 8.565 Å in projection,
lying in a direction normal to the fringe system, that is, at approximately 45	 to z*, Fig. 6.16.
Measuring X0 and z along the x* and z axes respectively, we have
</p>
<p>X0 &frac14; &eth;d=2&THORN; sin&eth;45&thorn; b� 90&THORN; &frac14; 3:237A
</p>
<p>Z &frac14; &eth;d=2&THORN; cos&eth;45&thorn; b� 90&THORN; &frac14; 2:804A
</p>
<p>From the figure, X &frac14; X0 cos&eth;b� 90&THORN;, whence X &frac14; 3.229 Å. Hence, the fractional coordinates are
x &frac14; 0.445 and z &frac14; 0.146. After the structure was fully solved [6], the refined values of these
coordinates were x &frac14; 0.4274 and z &frac14; 0.1431.
</p>
<p>Fig. 6.15 The h0l section of the weighted reciprocal lattice of euphenyl iodoacetate. The I&ndash;I fringe spacing (three lines
of zero amplitude are indicated) is 0.18 reciprocal lattice units, and the I&ndash;I vector makes an angle of ca. 45	 with the
z* axis (It may help to inspect the diagram edgewise.)
</p>
<p>260 6 Fourier Series and Fourier Transforms</p>
<p/>
</div>
<div class="page"><p/>
<p>6.7 Some General Properties of Transforms
</p>
<p>We summarize here in Table 6.4 some of the more important properties of Fourier transforms in
</p>
<p>their applications to X-ray crystallography; for each property, real and reciprocal space may be
</p>
<p>interchanged.
</p>
<p>6.8 Convolution
</p>
<p>We consider here the last of the properties of transforms listed in Table 6.4. The convolution integral is
</p>
<p>often called the &ldquo;folding&rdquo; integral, for a reason that will become clear as we continue the discussion.
</p>
<p>6.8.1 Convolution and Diffraction
</p>
<p>The Fourier transform of a slit diffraction grating of given width can be considered in terms of the
</p>
<p>diffraction at a single slit of that width, together with that of an ideal, infinite grating. We have, first,
</p>
<p>the diffraction pattern, or transform, of the single slit, shown in Fig. 6.17a, and then the transform of
</p>
<p>the ideal grating, Fig. 6.17b. These two transforms, in reciprocal space, are multiplied, point by point,
</p>
<p>to give the product in real space (Fig. 6.17c), which is called the convolution of the two functions.
</p>
<p>In a crystal, we have the contents of the unit cell and a point function, the lattice, in Bravais space.
</p>
<p>The convolution of these two functions is the diffraction pattern in reciprocal space. The transform of
</p>
<p>a lattice is another set of points, the reciprocal lattice points, and it has unit value at each point and
</p>
<p>zero value elsewhere. The product of the two transforms is the transform of the contents of the unit
</p>
<p>cell sampled at the reciprocal lattice points.
</p>
<p>Fig. 6.16 Geometry of the
reconstruction of the x and
</p>
<p>z coordinates of the iodine
</p>
<p>atoms in euphenyl
</p>
<p>iodoacetate. The b angle is
</p>
<p>94.1	; X, X0, and Z are
measured along the x, x*,
</p>
<p>and z axes, respectively.
</p>
<p>Note that since the space
</p>
<p>group of euphenyl
</p>
<p>iodoacetate is P21, the y
</p>
<p>coordinates of the iodine
</p>
<p>atoms may be set at �
1
</p>
<p>4
,
</p>
<p>so as to fix the position of
</p>
<p>the origin with respect to
</p>
<p>the symmetry elements
</p>
<p>6.8 Convolution 261</p>
<p/>
</div>
<div class="page"><p/>
<p>The image of a diffraction pattern is influenced by the divergence of the incident X-ray beam, the
</p>
<p>crystal shape and the mosaic structure of the crystal, because all of these factors have a bearing on the
</p>
<p>Fourier transform of the crystal.
</p>
<p>6.8.2 Convolution Integral
</p>
<p>The convolution of two functions is a physical concept of significance in divers scientific fields.
</p>
<p>We consider the process of convolution now in a little more detail. Consider two functions f(r) and
g(r) in Bravais (real) space, and let their transforms in reciprocal space be F(S) and G(S), respec-
tively. The Fourier transform of the product of these two transforms may be written as
</p>
<p>T&eth;r&THORN; &frac14;
&eth;
F&eth;S&THORN;G&eth;S&THORN; exp&eth;�i2pr � S&THORN; dVS (6.67)
</p>
<p>and is a function in Bravais space. From earlier sections we know that F(S) is given by
</p>
<p>F&eth;S&THORN; &frac14;
</p>
<p>&eth;
f &eth;r0&THORN; exp&eth;i2pr0 � S&THORN; dVr0 (6.68)
</p>
<p>where r0 is a vector different from r in general, but ranging over the same Bravais space, and the
change of sign of the exponential term arises because we are considering the reverse transformation.
</p>
<p>Thus, we have
</p>
<p>Table 6.4 Properties of Fourier transforms in real and reciprocal space
</p>
<p>Operation in Bravais space Result in reciprocal space
</p>
<p>Rotation about an axis Rotation about a parallel axis at the same speed
</p>
<p>Change of scale in a given direction Reciprocal change of scale in the same direction
</p>
<p>Translation Modulus unchanged; phase modified by a fringe function
</p>
<p>Addition of n units Vector summation of n transforms referred to a common origin
(a) Two parallel units Transform for one unit crossed by parallel, planar fringes;
</p>
<p>maximum amplitude doubled
(b) Two units related by a center of symmetry Transform for one unit crossed by wavy fringes which may be
</p>
<p>approximately planar in limited regions
</p>
<p>Convolution of two functions Transform is the product of the individual transforms
</p>
<p>Fig. 6.17 Relationship between amplitudes of diffraction. (a) Single slit of finite width (thin line). (b) Ideal slit grating
(dashed line). (c) Grating of finite-width slits (bold line). The transform of the grating of finite-width slits is the convolution
of the transforms of the single slit and the ideal grating (reproduced from Taylor and Lipson 1964)
</p>
<p>262 6 Fourier Series and Fourier Transforms</p>
<p/>
</div>
<div class="page"><p/>
<p>T&eth;r&THORN; &frac14;
&eth;
G&eth;S&THORN;
</p>
<p>&eth;
f &eth;r0&THORN; exp&eth;i2pr0 � S&THORN; dVr0
</p>
<p>� �
</p>
<p>exp&eth;�i2pr � S&THORN; dVS
</p>
<p>&frac14;
</p>
<p>&eth;
</p>
<p>f &eth;r0&THORN;
</p>
<p>&eth;
</p>
<p>G&eth;S&THORN; exp&frac12;�i2p&eth;r� r0&THORN; � S� dVS
</p>
<p>� �
</p>
<p>dVr0 &frac14;
</p>
<p>&eth;
</p>
<p>f &eth;r0&THORN;g&eth;r� r0&THORN; dVr0 (6.69)
</p>
<p>which is the convolution f &eth;r&THORN; � g&eth;r&THORN;, or c(r); g(r � r0) is the function g(r) with its origin moved
from r &frac14; 0 to r &frac14; r0.
</p>
<p>We can give a physical interpretation to this process. In Fig. 6.18a, an element dr0 of the function
f(r0) is defined. According to (6.69) each such element of f(r0) must be multiplied by the sharp
function g(r � r0) in Fig. 6.18b before integration. In Fig. 6.18c, this process is shown for three
elements, with the result in Fig. 6.18d. The completed convolution is shown in Fig. 6.18e; it is evident
</p>
<p>that the function f(r0) has been repeated at each value of g(r). Note that g(r � r0) is g(r0) mirrored, or
folded, across the ordinate axis and shifted by an amount r; the convolution integral is often referred
to as the folding integral.
</p>
<p>The converse of (6.69) is equally true, that is,
</p>
<p>T&frac12;c&eth;r&THORN;� &frac14; F&eth;S&THORN;G&eth;S&THORN; &frac14;
</p>
<p>&eth; &eth;
</p>
<p>f &eth;r0&THORN;g&eth;r� r0&THORN; dVr0
</p>
<p>� �
</p>
<p>exp&eth;�i2pr � S&THORN; dVr
</p>
<p>&frac14;
</p>
<p>&eth;
</p>
<p>f &eth;r0&THORN;
</p>
<p>&eth;
</p>
<p>g&eth;r� r0&THORN; dVr
</p>
<p>� �
</p>
<p>exp&eth;�i2pr � S&THORN; dVr0 (6.70)
</p>
<p>If now we let r � r0 &frac14; r00, where r00 is another independent variable in the same space as r and r00,
then
</p>
<p>Fig. 6.18 Convolution of two functions in the one-dimension space of r. (a) Function f(r0), showing an element
of width dr0. (b) Function g(r), showing two sharp peaks. (c) Function f(r0) with three selected elements dr0.
(d) Convolution of (b) and (c); the three elements are reproduced by g(r), modified by the values of that function.
(e) The completed convolution f &eth;r0&THORN;�g&eth;r&THORN;; because the function g(r) does not have infinitesimal width, f(r0) is
reproduced in a slightly modified form (6.14)
</p>
<p>6.8 Convolution 263</p>
<p/>
</div>
<div class="page"><p/>
<p>T&frac12;c&eth;r&THORN;� &frac14;
&eth;
f &eth;r0&THORN; exp&eth;�i2pr0 � S&THORN; dVr0
</p>
<p>&eth;
g&eth;r00&THORN; exp&eth;�i2pr00 � S&THORN; dVr0 &frac14; F&eth;S&THORN;G&eth;S&THORN; (6.71)
</p>
<p>so that the transform of the convolution is the product of the individual transforms.
</p>
<p>6.8.3 Convolution and Crystal Structure
</p>
<p>We have shown in the previous section, and illustrated in Fig. 6.18, that the convolution, in one-
</p>
<p>dimensional space, of a function f(r) with another function g(r) consisting of two sharp peaks, results
in f(r) being repeated, modified slightly in accordance with the width of g(r), at the two locations in g(r).
A pair of d-functions can be derived from g(r) by allowing the width of the sharp peaks, normalized to
unit area, to be reduced while maintaining the area at unity. In the limit as the width becomes infinitely
</p>
<p>small, the function becomes infinitely high at each location.
</p>
<p>We now have a pair of d-functions that have the value unity at each location but zero otherwise. If
</p>
<p>these d-functions were to be convolved with f(r), then f(r) would be repeated exactly at the locations of
the r vectors. We may write this process in mathematical terms in the following way. Initially, we have
the convolution expressed as
</p>
<p>c&eth;r&THORN; &frac14;
</p>
<p>&eth;
f &eth;r0 � r&THORN;g&eth;r&THORN; dVr
</p>
<p>where g(r) is normalized to unit area. As the width tends to infinitesimal size, it becomes
</p>
<p>c&eth;r&THORN; &frac14;
</p>
<p>&eth;
f &eth;r0 � r&THORN;d&eth;r&THORN; dVr &frac14; f &eth;r
</p>
<p>0&THORN; (6.72)
</p>
<p>from Sect. 6.6.8.
</p>
<p>If we translate the d-function by a vector r00 along the positive axis, so that the d-function is defined
as d(r � r00), then the convolution becomes
</p>
<p>c&eth;r&THORN; &frac14;
</p>
<p>&eth;
f &eth;r0 � r&THORN;d&eth;r� r00&THORN; dVr (6.73)
</p>
<p>so that the function f(r0) is reproduced at a vector distance r00 from the origin, that is, it is the function f
(r � r00). It is straightforward to extend the argument to a one-dimensional lattice. The convolution of
f(r0) with an infinite array of d-functions, g&eth;r&THORN; &frac14;
</p>
<p>P1
j&frac14;�1 d&eth;r� jr
</p>
<p>00&THORN;, is given by
</p>
<p>c&eth;r&THORN; &frac14;
X1
</p>
<p>j&frac14;�1
</p>
<p>f &eth;r� jr00&THORN; (6.74)
</p>
<p>which is periodic in r00, Fig. 6.19.
Thus, a crystal structure may be regarded first as the convolution of a unit cell and its contents,
</p>
<p>delineated by vectors a, b, and c, with a three-dimensional set of d-functions that have the value unity
at each Bravais lattice point but zero elsewhere.
</p>
<p>The transform of the unit-cell contents may involve one or more molecules, and is a continuous
</p>
<p>function in reciprocal space. However, the conditions for interference lead to the fact that this
</p>
<p>264 6 Fourier Series and Fourier Transforms</p>
<p/>
</div>
<div class="page"><p/>
<p>transform may be sampled, experimentally, only at the reciprocal lattice points, so that the convolu-
</p>
<p>tion of the unit cell and the lattice of d-functions is the product of the transform of the unit-cell
</p>
<p>contents and the reciprocal lattice, and is made manifest in the experimental diffraction pattern.
</p>
<p>The essence of this process may be grasped from Fig. 6.20, in which the transform of the contents
</p>
<p>of the centrosymmetrical unit cell of naphthalene, projected on to the ac plane, has been overlaid with
</p>
<p>a drawing of the x*, z* reciprocal lattice in the correct orientation. The weights of the transform at the
</p>
<p>reciprocal lattice points h0l correspond to the values of jF(h0l)j.
</p>
<p>Fig. 6.19 Convolution. (a) Structural entity f(r). (b) The function g(r) is an infinite set of d-functions d(r).
(c) Convolution c&eth;r&THORN; &frac14; f &eth;r&THORN; � g&eth;r&THORN;
</p>
<p>Fig. 6.20 Fourier transform of the h0l projection of the centrosymmetric structure of naphthalene, showing the a*
</p>
<p>and c* directions. The transform is overlaid with a drawing of the reciprocal lattice in the correct orientation. The
</p>
<p>amplitudes of the jF(h0l)j data are the amplitudes of the transform at the reciprocal lattice points. It can be seen readily
that reflections such as 201, 202, and 801 are sensitive to small changes in the orientation of the reciprocal lattice,
</p>
<p>whereas reflections such as 203, 603, and 802 are relatively insensitive under rotation
</p>
<p>6.8 Convolution 265</p>
<p/>
</div>
<div class="page"><p/>
<p>6.9 Structure Solution in Brief
</p>
<p>An X-ray diffraction pattern of a crystal can be recombined to form an image of the object from which
</p>
<p>it was derived, provided that we know the indices h, k, l, the amplitude jFj and the relative phase f
for each reflection. The absence of direct measurements of the phases means that they must be
</p>
<p>obtained indirectly. If it can be done satisfactorily, a Fourier transform (Fourier series) can be used to
</p>
<p>convert the reciprocal-space diffraction pattern into a real-space object, the crystal structure. This
</p>
<p>section is just a brief overview of some of the problems in crystal structure analysis, and how they
</p>
<p>may be overcome.
</p>
<p>6.9.1 Use of Heavy Atoms
</p>
<p>We considered a special case of this method in Sect. 6.6.6, where all F(h0l) data for platinum
</p>
<p>phthalocyanine were positive in sign. In a more general case, the heavy atom which acts like the
</p>
<p>speck of dust on the transparency in Sect. 6.1 will lie at a general position in the unit cell, probably
</p>
<p>related to one or more atoms by symmetry.
</p>
<p>Assuming for the moment that we can determine the coordinates of the heavy atoms in the unit
</p>
<p>cell, see, for example, Sect. 6.6.9, we could calculate approximate structure factor amplitudes
</p>
<p>and phases for the reflections. Then, the Fourier transform (6.57) becomes a summation, or series,
</p>
<p>with coefficients of the form Fo exp&eth;ifc&THORN;, where the Fo data are the experimentally determined
structure amplitudes and fc is the relative phase calculated from the positions of the heavy atoms
</p>
<p>alone.
</p>
<p>We may write these coefficients in the form
</p>
<p>Fo exp&eth;�ifc&THORN; &frac14; &eth;Fo=jFcj&THORN;jFcj exp&eth;�ifc&THORN; (6.75)
</p>
<p>where jFcj is the amplitude produced by the heavy atoms alone. The right-hand side of (6.75) is the
product of the transform of the heavy-atom portion of the structure and a function (Fo/jFcj) which is of
zero phase for every reflection. The result in real space is the convolution of the heavy-atom portion
</p>
<p>of structure with the transform of (Fo/jFcj). This transform, because of its zero phase, will have a large
peak at the origin, and the convolution will tend to make the heavy-atom part of the structure
</p>
<p>dominant. However, because Fo is not equal jFcj, there will be a background effect imposed upon
the heavy-atom part of the structure that will modify it in the direction of Fo. If the heavy-atom
</p>
<p>positions are correct, or nearly so, the convolution will lead to an improved model for the structure. It
</p>
<p>is important to realize, however, that because of the nature of (6.75) such a convolution is always
</p>
<p>biased toward the heavy-atom part of the structure, and success with this method depends strongly on
</p>
<p>the degree of correctness of the heavy-atom positions.
</p>
<p>Series Termination Effect
If a Fourier series is calculated with data only up to a certain value sin ymax, it is equivalent to
</p>
<p>multiplying the transform of the structure by a function that is unity up to sin ymaxand zero above it.
</p>
<p>The result is the convolution of the complete transform with that of the exclusion function. In two
</p>
<p>dimensions, the exclusion function simulates a circular hole, and its transform is a Bessel function,
</p>
<p>Fig. 6.7, which has a central maximum surrounded by maxima of alternating sign and decreasing
</p>
<p>magnitude. The transform of the limited data set is, thus, the complete transform convoluted with the
</p>
<p>Bessel function. The result can be seen in electron density maps as contours around the atomic
</p>
<p>266 6 Fourier Series and Fourier Transforms</p>
<p/>
</div>
<div class="page"><p/>
<p>positions that decrease to zero from the central maximum, and then are surrounded by ripples, or rings
</p>
<p>of decreasing magnitude and alternating sign. The effect is most noticeable in the regions around the
</p>
<p>heavy atoms in a structure; see, for example, Fig. 11.2a.
</p>
<p>6.9.2 General Phase-Free Transform: Patterson Function
</p>
<p>Although a diffraction pattern is invariant under the operation of translation, a representation of
</p>
<p>translation is contained within the pattern, because when the pattern is recombined the translational
</p>
<p>property of the object crystal is revealed. The relative phases at different parts of a transform do
</p>
<p>change under translation, but the changes are not evident because we observe the intensity of the
</p>
<p>transform:
</p>
<p>I &frac14; jFj2 (6.76)
</p>
<p>Equation (6.76) is the counterpart of (6.40) for observations at reciprocal lattice points.
</p>
<p>Recombination based on intensity alone contains no information about the lateral position of the
</p>
<p>object. All possible pairs of scattering points in the object will be reproduced with the correct
</p>
<p>orientation and separation, but symmetrically disposed about the center of the image. Hence, the
</p>
<p>image from the transform of intensities is that of vector positions between pairs of scattering points
</p>
<p>(atoms) in the object, all taken to a common origin. This is the Patterson function, Sect. 7.4.2, and
</p>
<p>may be thought of as the result of the superposition of numerous fringe systems, all of which have a
</p>
<p>positive sign at the origin of the unit cell. Mathematically, the transform of intensity, G(S)G*(S), is
the convolution of the transform of G(S), which is the electron density, with the transform of G*(S),
which is the electron density inverted in the origin.
</p>
<p>In the hypothetical rectangular four-atom structure, Fig. 6.21a, there are 42, or 16, interatomic vectors.
</p>
<p>Four of them are of zero length and coincide at the origin, Fig. 6.21b, and 12 are arranged in centrosym-
</p>
<p>metric pairs. Those formed by them symmetries of the rectangle, Fig. 6.21a, are of doubleweight,whereas
</p>
<p>those formedby symmetry 2, are of singleweight. Figure 6.21b can be considered in terms of Fig. 6.21a by
</p>
<p>transferring all interatomic vectors (arrowed) to a common origin.
</p>
<p>Fig. 6.21 Convolution
with a rectangle of
</p>
<p>scattering centers (atoms).
</p>
<p>(a) Hypothetical four-atom
structure of coordinates
</p>
<p>� (0.1, 0.2); interatomic
vectors are shown by
</p>
<p>arrows. (b) Schematic
convolution of the four-
</p>
<p>atom structure with its
</p>
<p>inversion in the origin.
</p>
<p>Multi-weight peaks arise
</p>
<p>from a superposition of
</p>
<p>identical vectors
</p>
<p>6.9 Structure Solution in Brief 267</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_11#Fig 2_11">http://dx.doi.org/10.1007/978-1-4614-3954-7_11#Fig 2_11</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_7#Sec14_7">http://dx.doi.org/10.1007/978-1-4614-3954-7_7#Sec14_7</a></div>
</div>
<div class="page"><p/>
<p>Alternatively, we can think of Fig. 6.21b as Fig. 6.21a convoluted with its inversion in the origin
</p>
<p>(which is the same as itself in this centrosymmetric arrangement). Thus, the structure in Fig. 6.21a is
</p>
<p>drawn four times, with its center of inversion at each of the atoms of the inverted structure in turn, and
</p>
<p>in the same orientation to give Fig. 6.21b.
</p>
<p>We can arrive at the same results by calculating the appropriate transforms. Using (6.51) with
</p>
<p>fj &frac14; 6 (carbon), and neglecting the change in fj with y, Fig. 6.22 was plotted for h &frac14; 0&ndash;10 and
k &frac14; 0&ndash;10. The contours (disregard the three vectors highlighted for this application) show clearly the
fringe systems that are perpendicular to each of the six pairs of points in Fig. 6.21a. In Fig. 6.23, the
</p>
<p>phase-free transform of the intensities, which are the squares of the values shown in Fig. 6.22 suitably
</p>
<p>scaled, we see the 16 peaks and 4 rectangles to which we referred above, just as in Fig. 6.21b. The
</p>
<p>orientations and separations of the peaks are correct, but there is a fourfold ambiguity with respect to
</p>
<p>the lateral positions. If, by some stratagem, we were able to determine the correct relative phases, that
</p>
<p>those shown in Fig. 6.22 were in fact correct, then the transform of these amplitudes, with those
</p>
<p>phases, would lead to the one correct result, Fig. 6.24, from which we determine the atomic
</p>
<p>positions � (0.1, 0.2). We shall investigate the required cunning in the next two chapters.
</p>
<p>6.9.3 Sign Relationships
</p>
<p>Relationships between signs (or phases, in the most general case) form a basis of the direct method of
</p>
<p>structure determination, discussed in see Chap. 8. At this stage, it is interesting to show how an
</p>
<p>important sign relationship may be deduced from the transform of a centrosymmetric arrangement of
</p>
<p>atoms, Fig. 6.22, that will be considered to exist within a crystal unit cell.
</p>
<p>Three vectors of the type h, k, and h � k, where we use h to represent the triplet hkl and k another
triplet h0k0l0, form a triangle in reciprocal space, with one vertex of the triangle at the origin; these
</p>
<p>vector terminations correspond to X-ray reflections.
</p>
<p>Fig. 6.22 Calculated
transform for the four-atom
</p>
<p>structure (Fig. 6.21a)
</p>
<p>convolved with its
</p>
<p>inversion in the origin;
</p>
<p>contours are shown at �20,
�10, 10, and 20. Fringe
systems are apparent,
</p>
<p>corresponding to the
</p>
<p>normals to the directions
</p>
<p>between all pairs of atoms
</p>
<p>in the four-atom structure
</p>
<p>(The vectors labeled h, k,
and h � k refer to the
discussion in Sect. 6.9.3.)
</p>
<p>268 6 Fourier Series and Fourier Transforms</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_8">http://dx.doi.org/10.1007/978-1-4614-3954-7_8</a></div>
</div>
<div class="page"><p/>
<p>If, as in this chosen example, the vectors terminate in regions of relatively high magnitude in the
</p>
<p>transform (strong reflections), then by counting the number of times that the zero-boundary contour is
</p>
<p>traversed we can arrive at the signs of the transform at the reciprocal lattice points h, k, and h � k,
since the transform must be positive at the origin:
</p>
<p>Vector hkl
</p>
<p>Times zero
</p>
<p>boundary crossed Sign
</p>
<p>h 450 3 �
k 420 2 +
</p>
<p>h � k 030 1 �
</p>
<p>Fig. 6.23 Calculated
transform from Fig. 6.22
</p>
<p>for h and k &frac14; 0&ndash;10, using
the squares of the given
</p>
<p>amplitudes (phase-free)
</p>
<p>suitably scaled; contours
</p>
<p>are shown at 20, 30, 40, and
</p>
<p>50, with the origin peak
</p>
<p>shaded. The rectangles are
</p>
<p>correct in size and
</p>
<p>orientation, but there is a
</p>
<p>fourfold ambiguity in the
</p>
<p>lateral positions in the x, y
</p>
<p>plane. The ambiguity is
</p>
<p>related to the symmetry and
</p>
<p>not to the number of atoms
</p>
<p>in the structure
</p>
<p>Fig. 6.24 Calculated
transform from Fig. 6.22
</p>
<p>for h and k &frac14; 0&ndash;10, using
the given amplitudes and
</p>
<p>signs; contours are shown
</p>
<p>at 6, 14, 22, and 30.
</p>
<p>A single, correct structure
</p>
<p>is now revealed
</p>
<p>6.9 Structure Solution in Brief 269</p>
<p/>
</div>
<div class="page"><p/>
<p>Hence,
</p>
<p>s&eth;h&THORN;s&eth;k&THORN;s&eth;h� k&THORN; &frac14; &thorn;1 (6.77)
</p>
<p>which is an expression of the triple product (&sum;2) sign relationship, or Sayre&rsquo;s equation, Sect. 8.2.3, for
</p>
<p>a centrosymmetric crystal. This &sum;2 relationship was used, apparently unwittingly as such, in solving
</p>
<p>the structure of hexamethylbenzene, Sect. 8.2.3.
</p>
<p>There will always be some vectors that terminate in a region of low magnitude in the transform
</p>
<p>(weak reflections), such as 810 and 840 in Fig. 6.22. Then, no certain conclusion can be drawn about
</p>
<p>their phases, because of the difficulty in locating exactly the zero-boundary contours in practice.
</p>
<p>6.10 Problems
</p>
<p>6.1 Show that
&ETH; c=2
�c=2 sin&eth;2pmx=c&THORN; cos&eth;2pnx=c&THORN; dx is zero for all values of m and n, where m and n
</p>
<p>are both integers. One or more of the identities in Web Appendix WA5 may be useful.
</p>
<p>6.2 Magnesium fluoride is tetragonal, with space P
42
</p>
<p>m
nm, a &frac14; 4.625 Å, c &frac14; 3.052 Å and two species
</p>
<p>MgF2 per unit cell at positions
</p>
<p>2Mg 0; 0; 0; 1=2; 1=2; 1=2
</p>
<p>4F � x; x; 0; 1=2&thorn; x; 1=2� x; 1=2&eth; &THORN;
</p>
<p>The Fo(h00) data are listed below:
</p>
<p>h 2 4 6 8 10 12
</p>
<p>Fo(h00) 2.7 12.0 7.2 0.1 3.2 0.1
</p>
<p>Calculate r(x) using the program FOUR1D with a subdivision of 40, and plot the function.
</p>
<p>Determine x. What length of the repeat is sufficient to define the complete function? Investigate
</p>
<p>the effects of (a) using terms up to h &frac14; 6, and (b) changing the sign attached to Fo(600).
</p>
<p>6.3 Find the Fourier transform of the function
</p>
<p>f &eth;x&THORN; &frac14;
a forjxj&lt;p
a=2 for x &frac14; �p
0 forjxj&gt;p
</p>
<p>8
&lt;
:
</p>
<p>Then, transform the resulting transform, and show that it regenerates the original function.
</p>
<p>6.4 Find the Fourier transform and its inversion for the periodic function
</p>
<p>f &eth;t&THORN; &frac14; A cos&eth;2pf0t&THORN;:
</p>
<p>6.5 Using (6.47), deduce the Fourier transform for two asymmetric scattering units related by a center
</p>
<p>of symmetry; the center may be taken as the origin. What general feature might be expected in the
</p>
<p>resulting transform?
</p>
<p>6.6 Show, from Fourier transform theory, the nature of the systematic absences that would arise
</p>
<p>from a 21 screw axis along the line 0; 1;
1
</p>
<p>4
</p>
<p>� �
in an orthorhombic unit cell.
</p>
<p>6.7 Tetraethyldiphosphine disulfide, &eth;C2H5&THORN;4P2 S2, crystallizes in the triclinic system,with space group
</p>
<p>P�1 and onemolecule in the unit cell. Figure P6.1 is the hk0 section of the weighted reciprocal lattice
</p>
<p>for this crystal. Make a photocopy of the diagram, identify, and draw lines to indicate, the fringe
</p>
<p>270 6 Fourier Series and Fourier Transforms</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_8#Sec5_8">http://dx.doi.org/10.1007/978-1-4614-3954-7_8#Sec5_8</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_8#Sec5_8">http://dx.doi.org/10.1007/978-1-4614-3954-7_8#Sec5_8</a></div>
</div>
<div class="page"><p/>
<p>system for the (double weight) P&ndash;S vector. Hence, allocate signs to the more intense reflections
</p>
<p>(about 25) in the asymmetric portion of the reciprocal-space diagram.
</p>
<p>6.8 A hypothetical, two-dimensional, three-atom structure has the atomic coordinates 0.1, �0.1;
�0.1, 0.1; 0.2, 0.3, with respect to the origin of rectangular axes. Draw a diagram to represent the
convolution of this structure with its inversion in the origin.
</p>
<p>Fig. P6.1 The hk0 section of the weighted reciprocal lattice of tetraethyldiphosphine disulphide (reproduced from
Taylor and Lipson 1964)
</p>
<p>Fig. P6.2 Figure field of the phase-free Fourier transform of the intensities of the hypothetical structure in Problem 6.8
</p>
<p>6.10 Problems 271</p>
<p/>
</div>
<div class="page"><p/>
<p>6.9 The figure field of Fig. P6.2 represents the phase-free Fourier transform of the intensities of the
</p>
<p>diffraction pattern of the three-atom structure in Problem 6.8. How many images of the
</p>
<p>structure are present, and what are the sets of possible atomic coordinates?
</p>
<p>6.10 Refer to Fig. 6.20. What are the signs of jFj for the reflections 00l, l &frac14; 1&ndash;6?
6.11 If f &eth;x&THORN; &frac14; 1=
</p>
<p>ffiffiffiffiffiffi
2p
</p>
<p>p
exp&eth;�x2=2&THORN; and g&eth;x&THORN; &frac14; d&eth;x� 2&THORN; find the convolution c&eth;x&THORN; &frac14; f &eth;x&THORN; � g&eth;x&THORN; by
</p>
<p>multiplying the transforms of f(x) and g(x).
</p>
<p>6.12 Write a computer program in any language with which you are familiar and compute the square-
</p>
<p>wave function, Sect. 6.2.1, with values of h equal to, say, 10, 100, and 1,000.
</p>
<p>(a) How does the form of the function f(x) change with increasing value of h?
</p>
<p>(b) Where in f(x) do the results indicate m-line symmetry?
</p>
<p>(c) Where does f(x) &frac14; p/2, and for what reason?
</p>
<p>References
</p>
<p>1. Sandoghdar V (2001) Proceedings of the International School of Physics, Enrico Fermi, IOS Press, Amsterdam
</p>
<p>2. Hecht B et al (2000) J Chem Phys 18:112
</p>
<p>3. Dunkerley BD, Lipson H (1955) Nature 176:82
</p>
<p>4. See Bibliography, Brigham
</p>
<p>5. Weast RC (ed) (1974&ndash;1975) Handbook of chemistry and physics. Chemical Rubber Company, Cleveland, OH
</p>
<p>6. Carlisle CH, Ladd MFC loc cit (chapter 1)
</p>
<p>Bibliography
</p>
<p>Harburn G, Taylor CA, Welberry TR (1975) An atlas of optical transforms. Bell, London
</p>
<p>James RW (1950) The crystalline state, 2nd edn, The optical principles of the diffraction of X-rays. Bell, London
</p>
<p>Lipson H (1972) Optical transforms. Academic, London
</p>
<p>Oran Brigham E (1988) The fast Fourier transform and its applications. Prentice Hall, Englewood Cliffs
</p>
<p>Taylor CA (1975) Diffraction. Institute of Physics
</p>
<p>Taylor CA, Lipson H (1964) Optical transformations. Bell, London
</p>
<p>Woolfson MM (1997) An introduction to crystallography, 2nd edn. Cambridge University Press, New York
</p>
<p>272 6 Fourier Series and Fourier Transforms</p>
<p/>
</div>
<div class="page"><p/>
<p>Fourier Techniques in X-Ray Structure
Determination 7
</p>
<p>7.1 Introduction
</p>
<p>We have reached the stage where we can consider how to attack the solving of a crystal structure.
</p>
<p>After the earliest trial and error determinations in the 1920s with very simple and highly symmetrical
</p>
<p>structures, it was found that the application of Fourier series, initially in one dimension, led to the
</p>
<p>electron density function, in which peak maxima in the electron density corresponded to atomic
</p>
<p>positions. As we have seen in the previous chapters, it is necessary to have the phases of the structure
</p>
<p>factors for a Fourier synthesis to be carried out meaningfully. One way in which phase information
</p>
<p>may be obtained is through the Patterson function of vector density, a function of interatomic vectors
</p>
<p>in the crystal structure.
</p>
<p>In this chapter, we examine the application of Fourier series to crystal structure analysis, together
</p>
<p>with some of its developments. However, in any structure analysis there are certain preliminary
</p>
<p>investigations that can throw light on the problem in hand. One such investigation, which we have
</p>
<p>already discussed, leads to the unit-cell dimensions and space group of the crystal. We study next
</p>
<p>some example crystal structures in order to show how other, readily available information can be used
</p>
<p>to assist in the structure solution process.
</p>
<p>7.2 Analysis of the Unit-Cell Contents
</p>
<p>The density Dm of the crystals under examination may be measured by suspending them in a liquid or
</p>
<p>liquid mixture. The composition of the liquid is altered until the crystals neither rise nor fall; then the
</p>
<p>density of the liquid, equal to Dm, is measured with a pyknometer. Many organic materials can be
</p>
<p>suspended in aqueous sodium bromide. At flotation equilibrium, the refractive index of the solution may
</p>
<p>be measured, and the density determined by interpolation of a graph of the density of the solution as a
</p>
<p>function of its refractive index. The flotation procedure is best carried out in a thermostat. It may
</p>
<p>still happen, however, that the demarcation between sinking and floating is a little ill defined. Inclusion
</p>
<p>of air or solvent in the crystal will lead to a smaller apparent density, and the flotation position
</p>
<p>corresponding to a maximum value for the density measured should be most appropriate.
</p>
<p>If the crystal unit cell contains a number Z of chemical species, each of relative molar mass Mr,
</p>
<p>then the following relationship holds:
</p>
<p>Dm &frac14; ZMrmu=Vc (7.1)
</p>
<p>M. Ladd and R. Palmer, Structure Determination by X-ray Crystallography:
</p>
<p>Analysis by X-rays and Neutrons, DOI 10.1007/978-1-4614-3954-7_7,
# Springer Science+Business Media New York 2013
</p>
<p>273</p>
<p/>
</div>
<div class="page"><p/>
<p>where mu is the atomic mass unit. If the volume Vc of the unit cell is in Å
3 and the density in g cm�3,
</p>
<p>(7.1) may be written as
</p>
<p>Dm &frac14; 1:6605 ZMr=Vc (7.2)
</p>
<p>If several measurements are made, the standard deviation s(Dm) can be deduced. It is useful to
</p>
<p>calculate the density Dc from the unit-cell volume and the (integral) value of Z. A significant
</p>
<p>discrepancy between Dm and Dc should be examined, as it might point to an error in the unit-cell
</p>
<p>dimensions or to solvent of crystallization not included in Mr at that stage.
</p>
<p>7.2.1 Papaverine Hydrochloride, C20H21NO4&middot;HCl
</p>
<p>Crystal Data
</p>
<p>System: monoclinic
</p>
<p>Unit-cell dimensions: a &frac14; 13.059 Å, b &frac14; 15.620 Å, c &frac14; 9.130 Å, b &frac14; 92.13�
Vc: 1861.1 Å
</p>
<p>3
</p>
<p>Dm: 1.33 g cm
�3
</p>
<p>Mr: 375.85
</p>
<p>Z: 4 to the nearest integer (3.97 from (7.2))
</p>
<p>Unit-cell contents: 80C, 88H, 4N, 16O, 4C1 atoms
</p>
<p>Absent spectra: h0l: l odd; 0k0: k odd
</p>
<p>Space group: P21/c
</p>
<p>All atoms are in general equivalent positions. The molecular conformation, obtained by a complete
</p>
<p>structure analysis [1], is shown in Fig. 7.1.
</p>
<p>Fig. 7.1 Stereoview of the molecular conformation of papaverine hydrochloride; the circles, in order of decreasing
size, represent Cl, O, N, C, and H
</p>
<p>274 7 Fourier Techniques in X-Ray Structure Determination</p>
<p/>
</div>
<div class="page"><p/>
<p>7.2.2 Naphthalene, C10H8
</p>
<p>Crystal Data
</p>
<p>System: monoclinic
</p>
<p>Unit-cell dimensions: a &frac14; 8.658 Å, b &frac14; 6.003 Å, c &frac14; 8.235 Å, b &frac14; 122.92�
Vc: 359.28 Å
</p>
<p>3
</p>
<p>Dm: 1.152 g cm
�3
</p>
<p>Mr: 128.17
</p>
<p>Z: 2 to the nearest integer (1.94 from (7.2))
</p>
<p>Unit-cell contents: 20C, 16H atoms
</p>
<p>Absent spectra: h0l: l &frac14; 2n + 1; 0k0: l &frac14; 2n + 1
</p>
<p>Space group: P21/c
</p>
<p>7.2.3 Molecular Symmetry
</p>
<p>In papaverine hydrochloride, the four molecules in the unit cell occupy a set of general positions; each
</p>
<p>atom at coordinates xj, yj, zj, ( j &frac14; 1, 2, . . ., 48) is repeated by the space-group symmetry so as to build up
</p>
<p>the crystal structure. There are, therefore, 48 atoms, including hydrogen, in the asymmetric unit to be
</p>
<p>located by the structure analysis.
</p>
<p>Naphthalene is not quite so straightforward. With two molecules per unit cell, there are 20 carbon
</p>
<p>atoms and 16 hydrogen atoms that may be distributed in 4 equivalent-position sets of five and four
</p>
<p>atoms, respectively, in the unit cell. This means that in order to solve the structure, we have to locate
</p>
<p>five carbon atoms and four hydrogen atoms. This number is only half that expected: since Z is 2, each
</p>
<p>atom is related by one of the symmetry elements of the space group to a second atom of the same
</p>
<p>type in the same molecule, so as to generate C10H8 from C5H4. There are three different symmetry
</p>
<p>elements to consider: the 21 axis, the c-glide plane, and the center of symmetry. The screw axis
</p>
<p>and glide plane are discounted because they involve translational symmetry, which would generate
</p>
<p>an infinite molecule with translational repeats. We must, therefore, conclude that the atom pairs are
</p>
<p>related by a center of symmetry, which in turn means that the molecule of naphthalene is
</p>
<p>centrosymmetric.
</p>
<p>The symmetry analysis for naphthalene has served two very useful purposes: it has halved thework of
</p>
<p>the subsequent structure analysis, and shown that the molecules in the crystal exhibit a certain minimum
</p>
<p>symmetry (�1). This result is, of course, in agreement with chemical knowledge, which ordinarily we are
</p>
<p>quite entitled to use. The conventional notion that naphthalene should havemmm symmetry, Fig. 7.2, is
</p>
<p>not supported directly, although the crystal structure analysis shows that this symmetry holds within
</p>
<p>experimental error.
</p>
<p>Fig. 7.2 Naphthalene molecular structure; the 9,10 carbon&ndash;carbon bond lies on a center of symmetry
</p>
<p>7.2 Analysis of the Unit-Cell Contents 275</p>
<p/>
</div>
<div class="page"><p/>
<p>7.2.4 Special Positions
</p>
<p>Themolecules ofnaphthalene lie onspecial positions inP21/c, Fig. 7.3. Special position sites correspond in
</p>
<p>symmetry to one of the 32 crystallographic point groups and, in subsequent examples, we shall see that
</p>
<p>both atoms and molecules can occupy special positions.
</p>
<p>Glide planes and screw axes do not usually accommodate atoms or molecules; an atom lying
</p>
<p>exactly on a translational symmetry element would introduce a pseudo-half-axial translation, thus
</p>
<p>creating special reflection conditions which, depending on the atomic number, may be observable
</p>
<p>among the X-ray data (see Problem 7.1).
</p>
<p>Although the molecules of naphthalene are in special positions, they are subject to the space-group
</p>
<p>symmetry inherent in the general positions: if one molecule is located at 0, 0, 0, then the second
</p>
<p>molecule is at 0, 1
2
, 1
2
; see Sect. 2.7.5. This set may be determined by substituting x &frac14; y &frac14; z &frac14; 0 into
</p>
<p>the set of general positions. The structure of naphthalene [2] is shown in Fig. 7.4. The reader may like
</p>
<p>to consider the three other possible sets of special positions that could be used to represent this
</p>
<p>structure, and then show from the structure factor equation that jF(hkl)j is invariant with respect to
each set of special positions.
</p>
<p>7.2.5 Nickel Tungstate, NiWO4
</p>
<p>Crystal Data
</p>
<p>System: monoclinic
</p>
<p>Unit-cell dimensions: a &frac14; 4.60 Å, b &frac14; 5.66 Å, c &frac14; 4.91 Å, b &frac14; 90.1�
Vc: 127.84 Å
</p>
<p>3
</p>
<p>Dm: 7.964 g cm
�3
</p>
<p>Mr: 306.81
</p>
<p>Z: 2 to the nearest integer (2.00 from (7.2))
</p>
<p>Unit-cell contents: 2Ni, 2W, 8O atoms
</p>
<p>Absent spectra: h0l: l &frac14; 2n + 1
Possible space groups: Pc or P2/c
</p>
<p>We shall use space group P2/c, since the structure was determined successfully only with this
</p>
<p>space group [3].
</p>
<p>Fig. 7.3 Grouping of one of the special positions sets in P21/c; the arrangement of molecules (symmetry �1) with their
centers at 0, 0, 0 and 0, 1
</p>
<p>2
, 1
2
is shown
</p>
<p>276 7 Fourier Techniques in X-Ray Structure Determination</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec20_2">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec20_2</a></div>
</div>
<div class="page"><p/>
<p>The general equivalent positions in P2/c are
</p>
<p>� fx; y; z; x; �y; 1
2
&thorn; zg
</p>
<p>but in order to study NiWO4 further, we must consider the possible special positions for this space
</p>
<p>group; they are located on either the twofold axes or the centers of symmetry. The reader should make
</p>
<p>a drawing for space group P2/c, using the coordinates listed above and inserting the symmetry
</p>
<p>elements.
</p>
<p>Special Positions on Twofold Axes
The twofold axes lie along the lines 0; y;
</p>
<p>1
</p>
<p>4
</p>
<p>� �
</p>
<p>;
1
</p>
<p>2
; y;
</p>
<p>1
</p>
<p>4
</p>
<p>� �
</p>
<p>; 0; y;
3
</p>
<p>4
</p>
<p>� �
</p>
<p>; and
1
</p>
<p>2
; y;
</p>
<p>3
</p>
<p>4
</p>
<p>� �
</p>
<p>. The equivalent
</p>
<p>positions generated by the space-group symmetry show that the special position sets are
</p>
<p>� f0; y; 1
4
g or � f1
</p>
<p>2
; y; 1
</p>
<p>4
g
</p>
<p>and each set satisfies P2/c symmetry by accommodating in the unit cell two structural entities with
</p>
<p>symmetry 2.
</p>
<p>Special Positions on Centers of Symmetry
If we repeat the above analysis for the eight centers of symmetry in the space group, we will develop
</p>
<p>four special position sets:
</p>
<p>0; 0; 0 0; 0; 1
2
</p>
<p>1
2
; 0; 0 1
</p>
<p>2
; 0; 1
</p>
<p>2
</p>
<p>0; 1
2
; 0 0; 1
</p>
<p>2
; 1
2
</p>
<p>1
2
; 1
2
; 0 1
</p>
<p>2
; 1
2
; 1
2
</p>
<p>Fig. 7.4 Stereoview of the crystal structure of naphthalene; for clarity, H atoms are not shown
</p>
<p>7.2 Analysis of the Unit-Cell Contents 277</p>
<p/>
</div>
<div class="page"><p/>
<p>The Ni and W atoms must lie on special positions, with either 2 or �1 symmetry. Nothing can be said
</p>
<p>about the position of the oxygen atoms, and without further detailed analysis we cannot define this
</p>
<p>structure further. However, to complete the picture, we list the atomic parameters for this structure, and
</p>
<p>illustrate it in Fig. 7.5:
</p>
<p>2Ni �f1
2
; 0:653; 1
</p>
<p>4
g
</p>
<p>2W �f0; 0:180; 1
4
g
</p>
<p>4O �f0:22; 0:11; 0:96; 0:22; 0:89; 0:46g
4O0 �f0:26; 0:38; 0:39; 0:26; 0:62; 0:89g
</p>
<p>The heavy atoms (W and Ni) were found to occupy the four twofold axes in pairs. This conclusion,
</p>
<p>although not uniquely derivable from the symmetry analysis alone, was at least partially indicated by
</p>
<p>it. Once again, a pencil and paper operation saved considerable effort in the subsequent detailed
</p>
<p>structure analysis by pointing to the proper course of action.
</p>
<p>In these few examples, we have shown the value of a symmetry analysis in the early stages of a
</p>
<p>structure determination. The procedure may be regarded as a routine to be carried out before the more
</p>
<p>detailed calculations required in the elucidation of the atomic parameters.
</p>
<p>7.3 Interpretation of Electron Density Distributions
</p>
<p>We have discussed different forms of the electron density equation in Sect. 6.3ff and we now make
</p>
<p>use of that theory in studying the distribution of electron density in crystal structures.
</p>
<p>Electron density is concentrated in the vicinity of atoms, and rises to peaks at electron density
</p>
<p>maxima, which correspond to atomic positions within the limits of experimental error, and fall to
</p>
<p>relatively low values between the peaks. The wavelengths of X-rays used in crystal structure analysis
</p>
<p>are too long to reveal the intimate electronic structure of atoms themselves, which appear, therefore,
</p>
<p>somewhat blurred in the calculated electron density function. In general, the more complete and
</p>
<p>accurate the experimental Fo data, the better will be the atomic resolution and the more precise the
</p>
<p>final structure model.
</p>
<p>Fig. 7.5 Structure of NiWO4, showing the WO6 and NiO6 octahedra: large open circles O, small open circles W, small
black circles Ni
</p>
<p>278 7 Fourier Techniques in X-Ray Structure Determination</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec5_6">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec5_6</a></div>
</div>
<div class="page"><p/>
<p>7.3.1 Peak Heights and Weights
</p>
<p>To a first approximation, the heights of the peaks in an electron density distribution of a crystal are
</p>
<p>proportional to the corresponding atomic numbers. The hydrogen atom, at the extreme low end of the
</p>
<p>atomic numbers, is rarely resolved in electron density maps; its small electron density merges into the
</p>
<p>background density that arises from errors in both the data and the structure model. However,
</p>
<p>hydrogen atoms can be detected by a difference-Fourier technique, as discussed later, Sect. 8.4.5,
</p>
<p>and by neutron diffraction; see also Sects. 11.1, 11.4.1, and 11.5.
</p>
<p>A better measure of the electron content of a given atom may be obtained from an integrated peak
</p>
<p>weight, in which the absolute values of r(xyz) are summed over the volume occupied by the atom.
</p>
<p>This technique makes some allowance for the variation of individual atomic temperature factors, high
</p>
<p>values of which tend to decrease peak heights for a given electron content.
</p>
<p>7.3.2 Computation and Display of Electron Density Distributions
</p>
<p>Assuming for the moment that phases are available, the electron density function may be calculated
</p>
<p>over a grid of chosen values of x, y, and z. For this purpose, the unit cell is divided into a selected number
</p>
<p>of equal divisions, in a manner similar to that employed in the synthesis of the square-wave function,
</p>
<p>Sect. 6.2.1. Intervals corresponding to about 0.3 Å are satisfactory for most electron density maps. The
</p>
<p>symmetry of r(xyz) is that of the space group of the crystal under investigation. Consequently, a
</p>
<p>summation over a volume either equal to or just greater than that of the asymmetric unit is adequate.
</p>
<p>In order to facilitate the interpretation of r(xyz), it is essential to present the distribution of the
</p>
<p>numerical values in such a way that the geometric relationships between the peaks are easily
</p>
<p>inspected. This feature is afforded by first calculating the electron density in sections, each
</p>
<p>corresponding to a constant value of x, y, or z. Each section consists of a field of figures arranged
</p>
<p>on a grid, which may be true to scale for preference, and can be contoured by lines passing through
</p>
<p>points of equal electron density, interpolating as necessary, Fig. 7.6. The grading of the contour
</p>
<p>intervals is selected to produce a reasonable number of contours around the higher-density areas. The
</p>
<p>contouring should be carried out with care; this exercise leads to fairly precise peak positions and a
</p>
<p>desirable familiarity with the problem. Sophisticated map-plotting and peak-searching facilities are
</p>
<p>available, but they should be treated with caution by the beginner.
</p>
<p>The contoured sections may be transferred to a transparent medium, such as thin perspex or clear
</p>
<p>acetate sheets, which are then stacked at the requisite spatial intervals and viewed over a diffuse light
</p>
<p>source. The diagram in Fig. 1.7 is a photograph of such a display, extending through 17 sections.
</p>
<p>An alternative method of displaying the results of an electron density calculation is by means of a
</p>
<p>ball-and-stick model. An example of this form of representation is shown in Fig. 7.7.
</p>
<p>In the analysis of small molecules, it is not considered necessary always to plot and contour the
</p>
<p>electron density function, although it can be done through some program packages, such as WinGX.
</p>
<p>There are many graphics programs available that recognize the highest peaks and carry out a
</p>
<p>geometrical interpretation in terms of their coordinates: Platon for small molecules, and O, Turbo
</p>
<p>Frodo or Coot for proteins (see Appendix D).
</p>
<p>7.3.3 Projections
</p>
<p>The use of two-dimensional studies in crystallography is fairly restrictive but, nevertheless, worthy of
</p>
<p>mention because of the relative ease of calculation and preparation of Fourier maps. For example, the
</p>
<p>function
</p>
<p>7.3 Interpretation of Electron Density Distributions 279</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec34_8">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec34_8</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec1_11">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec1_11</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec11_11">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec11_11</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec14_11">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec14_11</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec3_6">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec3_6</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Fig7_1">http://dx.doi.org/10.1007/978-1-4614-3954-7#Fig7_1</a></div>
</div>
<div class="page"><p/>
<p>Fig. 7.6 Two-dimensional electron density projection r(xy) for azidopurine monohydrate, C5H3N7&middot;H2O, calculated
from the data of Glusker et al. [3a]. The isolated peak Ow in the lower right-hand region of the map represents
</p>
<p>the oxygen atom of the water molecule. Hydrogen-atom positions are not obtained in this electron density synthesis.
</p>
<p>The field figures are 10 r(xy) Å�2 contoured at intervals of 20 units
</p>
<p>Fig. 7.7 Three-dimensional model of euphenyl iodoacetate; see also Figs. 1.7 and 1.8</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Fig7_1">http://dx.doi.org/10.1007/978-1-4614-3954-7#Fig7_1</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Fig8_1">http://dx.doi.org/10.1007/978-1-4614-3954-7#Fig8_1</a></div>
</div>
<div class="page"><p/>
<p>r&eth;xz&THORN; &frac14; 2
A
</p>
<p>X
</p>
<p>h
</p>
<p>X
</p>
<p>l
</p>
<p>jF&eth;h0l&THORN;j cos&frac12;2p&eth;hx&thorn; lz&THORN; � f&eth;h0l&THORN;� (7.3)
</p>
<p>is calculated with the data from only one level of the reciprocal lattice, the zero level, perpendicular to
</p>
<p>b, and plotted over the area A of the a, c plane, or the asymmetric portion thereof. The simplification
</p>
<p>in the calculations is offset, however, by a corresponding complexity in the interpretation of the maps,
</p>
<p>arising from the superposition of peaks in projection on to the given plane, although this effect is not
</p>
<p>as severe as in one dimension. Equation (7.3) corresponds to the projection of the electron density
</p>
<p>along the b axis: it is essential to appreciate the difference between the meaning of r(xz) and r(x0z);
</p>
<p>the latter represents the section of the three-dimensional electron density function at y &frac14; 0. Equations
for projections along other principal axes may be written by analogy with (7.3).
</p>
<p>Even simple atomic arrangements may appear distorted in projections, with individual molecules
</p>
<p>overlapping to some degree, but we would not wish to discourage their consideration. We shall
</p>
<p>restrict their use to examples illustrating various aspects of structure analysis. Practice in the
</p>
<p>calculation and interpretation of Fourier series is provided by Problems 7.11 and 7.12, and also the
</p>
<p>exercises with the XRAY system; see Sect. 13.4.
</p>
<p>7.4 Methods of Solving the Phase Problem
</p>
<p>At this point, it may be convenient for the reader to revise some of the ideas presented in Sect. 6.9ff
</p>
<p>since a general overview of the present topic was presented there.
</p>
<p>The set of Fo(hkl) data constitutes the starting point of an X-ray structure determination. The
</p>
<p>approximate number of symmetry-independent reflections measurable may be calculated in the
</p>
<p>following manner.
</p>
<p>7.4.1 Number of Reflections in the Data Set
</p>
<p>The radius of the limiting sphere is 2 RU (reciprocal lattice units), and its volume is therefore
</p>
<p>33.510 RU3, taking the reciprocal lattice constant k as l, Sect. 2.4. The number of reciprocal lattice
</p>
<p>points within the limiting sphere is approximately equal to the number of times the reciprocal unit-
</p>
<p>cell volume V* will fit into 33.510; since, for this application, V* &frac14; l/V, this number is 33.510 Vc/l3.
The number of symmetry-independent reflections observable, Nmax, in a given experiment in which
</p>
<p>ymax represents the practical upper limit of y is given by
</p>
<p>Nmax &frac14; 33:510Vcsin3ymax=l3Gm (7.4)
</p>
<p>whereG is the unit-cell translation constant, Table 3.2, for non-zero reflections andm is the number of
</p>
<p>symmetry-equivalent reflections, or the number of general equivalent points in the appropriate Laue
</p>
<p>group. For zones and rows, mmay take different values from that for hkl, and a number of systematic
</p>
<p>absences within the sphere of radius 2 sin ymax may have to be subtracted.
</p>
<p>As an example, consider an orthorhombic crystal of space group Cmm2, with unit-cell dimensions
</p>
<p>a &frac14; 9.00 Å, b &frac14; 10.00 Å, and c &frac14; 11.00 Å. For Cu Ka radiation (l &frac14; 1:5418A) and ymax of 85�
(dmin &frac14; 0.77 Å), Nmax is (33.510 � 9 � 10 � 11� sin3 85)/(1.54183 � 2 � 8) &frac14; 559. If Mo Ka
radiation (with ymax � 27� and �l &frac14; 0:71073 Å) had been used instead of Cu Ka, the number would
have been 5709. Say the structure contains 15 atoms in the asymmetric unit. In the structure analysis, each
</p>
<p>7.4 Methods of Solving the Phase Problem 281</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec6_13">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec6_13</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec28_6">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec28_6</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec7_2">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec7_2</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Tab2_3">http://dx.doi.org/10.1007/978-1-4614-3954-7#Tab2_3</a></div>
</div>
<div class="page"><p/>
<p>atom would be determined by three positional parameters (xj, yj, zj) and, say, one isotropic thermal
</p>
<p>vibration parameter, which, with an overall scale factor, totals 61 variables. Even with Cu Ka radiation,
</p>
<p>there are nine reflections per variable, a situation which, from a mathematical point of view, is consider-
</p>
<p>ably overdetermined. This feature is important, since the experimental intensity measurements contain
</p>
<p>random errors which cannot be eliminated, and a preponderance of data is needed to ensure good
</p>
<p>precision in the structural parameters. We shall meet this situation again in Chap. 8 but such a degree
</p>
<p>of over-determination is not usually possible with macromolecules, as we explain in Chap. 10.
</p>
<p>7.4.2 The Patterson Function
</p>
<p>Although the connection between Fourier theory and X-ray diffraction was recorded first in 1915, it
</p>
<p>was not until about 1930 that very much practical use was made of it. Before the advent of computing
</p>
<p>facilities, the calculation of even a Fourier projection, involved considerable time and effort. Add to
</p>
<p>this the phase problem, which necessitated many such calculations, and it is easy to understand that
</p>
<p>X-ray analysts were not anxious to become involved with extensive Fourier calculations; many early
</p>
<p>structure analyses were based on two projections.
</p>
<p>In 1934, Patterson reported a new Fourier series which could be calculated directly from the
</p>
<p>experimental intensity data. However, because phase information is not required in the Patterson
</p>
<p>series, the result cannot be interpreted as a set of atomic positions, but rather as a collection of
</p>
<p>interatomic vectors all taken to a common origin, Sect. 6.9.2. Patterson was led to the formulation of
</p>
<p>his series from considerations of an earlier theory of Debye on the scattering of X-rays by liquids&mdash;a
</p>
<p>much more difficult problem.
</p>
<p>Patterson functions are of considerable importance in X-ray structure analysis, and their applica-
</p>
<p>tion will be considered in some detail. We will study first a one-dimensional function.
</p>
<p>One-Dimensional Patterson Function
The electron density at any fractional coordinate x is r(x), and that at the point (x + u) is r(x + u). The
</p>
<p>average product of these two electron densities in a repeat of length a, for a given value of u, is
</p>
<p>A&eth;u&THORN; &frac14;
&eth;1
</p>
<p>0
</p>
<p>r&eth;x&THORN;r&eth;x&thorn; u&THORN; dx (7.5)
</p>
<p>where the upper limit of integration corresponds to the use of fractional coordinates. Using (6.21) in a
</p>
<p>form appropriate to a one-dimensional unit repeat, we obtain
</p>
<p>A&eth;u&THORN; &frac14;
&eth;1
</p>
<p>0
</p>
<p>1
</p>
<p>a2
</p>
<p>X
</p>
<p>h
</p>
<p>jFj&eth;h&THORN;e�i2phx
X
</p>
<p>h0
jFj&eth;h0&THORN;e�i2ph0&eth;x&thorn;u&THORN; dx (7.6)
</p>
<p>where jFj is an amplitude (phase-free) term. The index h0 lies within the same range as h, but is used to
effect distinction between the Fourier series for r(x) and r(x + u). Separating the parts dependent
</p>
<p>upon x, and remembering that the integral of a sum is the sum of the integrals of the separate terms,
</p>
<p>we may write
</p>
<p>A&eth;u&THORN; &frac14; 1
a2
</p>
<p>X
</p>
<p>h
</p>
<p>X
</p>
<p>h0
jFj&eth;h&THORN;jFj&eth;h0&THORN;e�i2ph0u
</p>
<p>&eth;1
</p>
<p>0
</p>
<p>e�i2p&eth;h&thorn;h
0&THORN;x dx (7.7)
</p>
<p>282 7 Fourier Techniques in X-Ray Structure Determination</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_8">http://dx.doi.org/10.1007/978-1-4614-3954-7_8</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_10">http://dx.doi.org/10.1007/978-1-4614-3954-7_10</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec30_6">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec30_6</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Equ21_6">http://dx.doi.org/10.1007/978-1-4614-3954-7#Equ21_6</a></div>
</div>
<div class="page"><p/>
<p>Considering the integral
</p>
<p>&eth;1
</p>
<p>0
</p>
<p>e�i2p&eth;h&thorn;h
0&THORN;xdx &frac14; e
</p>
<p>�i2p&eth;h&thorn;h0&THORN;x
</p>
<p>�i2p&eth;h&thorn; h0&THORN;
</p>
<p>�1
</p>
<p>0
</p>
<p>(7.8)
</p>
<p>e�i2p(h+h
0) is unity, since h and h0 are integral, from de Moivre&rsquo;s theorem, Sect. 3.2.3, and the integral
</p>
<p>is, in general, zero. However, for the particular value of h0 equal to �h, it becomes indeterminate and
we must consider making this substitution before integration. Thus,
</p>
<p>&eth;1
</p>
<p>0
</p>
<p>dx &frac14; 1 (7.9)
</p>
<p>Hence, from (7.7), for non-zero values of A(u), where h0 &frac14; �h,
</p>
<p>A&eth;u&THORN; &frac14; 1
a2
</p>
<p>X
</p>
<p>h
</p>
<p>X
</p>
<p>�h
jFj&eth;h&THORN;jFj&eth;�h&THORN;ei2phu (7.10)
</p>
<p>Equation (7.10) is not really a double summation, since h and �h cover the same field of the
function. From Fig. 3.23, we see that the magnitudes jFj(h) and jFj(�h) are equal; hence, (7.10)
becomes
</p>
<p>A&eth;u&THORN; &frac14; 1
a2
</p>
<p>X
</p>
<p>h
</p>
<p>jF&eth;h&THORN;j2ei2phu (7.11)
</p>
<p>where the index h ranges from �1 to 1. Taking h from 0 to 1, (7.11) may be written as
</p>
<p>A&eth;u&THORN; &frac14; 1
a2
</p>
<p>X
</p>
<p>h
</p>
<p>&eth;jF&eth;h&THORN;j2ei2phu &thorn; jF&eth;h&THORN;j2e�i2phu&THORN; (7.12)
</p>
<p>Applying de Moivre&rsquo;s theorem, we obtain
</p>
<p>A&eth;u&THORN; &frac14; 2
a2
</p>
<p>X
</p>
<p>h
</p>
<p>jF&eth;h&THORN;j2 cos 2phu (7.13)
</p>
<p>The corresponding Patterson function P(u) is usually defined as
</p>
<p>P&eth;u&THORN; &frac14; 2
a
</p>
<p>X
</p>
<p>h
</p>
<p>jF&eth;h&THORN;j2 cos 2phu (7.14)
</p>
<p>a small difference from the averaging function A(u).
</p>
<p>The practical evaluation of P(u) proceeds through (7.14), but its physical interpretation is best
</p>
<p>considered in terms of (7.5), neglecting the small difference between P(u) and A(u).
</p>
<p>Figure 7.8a shows one unit cell of a one-dimensional structure containing two different atoms A
</p>
<p>and B situated at fractional coordinates xA and xB, respectively. Equation (7.5) represents the value of
</p>
<p>the electron density product r(x)r(x + u), for any given value of u, averaged over the repeat period of
</p>
<p>the unit cell. The average will be zero if one end of the vector u always lies in a zero region of electron
density, small if both ends of the vector encounter low electron densities, large if the electron density
</p>
<p>7.4 Methods of Solving the Phase Problem 283</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec11_3">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec11_3</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Fig23_3">http://dx.doi.org/10.1007/978-1-4614-3954-7#Fig23_3</a></div>
</div>
<div class="page"><p/>
<p>products are large, and amaximum where u is of such a length that it spans two atomic positions in the
</p>
<p>unit cell.
</p>
<p>For values of u less than umin in Fig. 7.8a, no peak will arise from the pair of atoms. As u is
</p>
<p>increased, however, both ends of the vector will come simultaneously under the electron density
</p>
<p>peaks, and from (7.5) a finite value of A(u), or P(u), will be obtained. The integration can be simulated
</p>
<p>by sliding a vector of a given magnitude u along the x axis, evaluating the average product r(x)
</p>
<p>r(x + u) for all sampling intervals between zero and unit fractional repeat; this process is carried out
</p>
<p>for all fractional values of u between 0 and 1. The graph of P(u) as a function of u is similar in
</p>
<p>appearance to an electron density function, but we must be careful not to interpret it in this way.
</p>
<p>As we proceed through the values of u, we encounter upeak, the interatomic vector A&ndash;B, which gives
</p>
<p>rise to the maximum value of P(u), labeled AB in Fig. 7.8b. As u increases to umax, the electron density
</p>
<p>product falls to zero and P(u) decreases correspondingly. Since we are concerned with interatomic
</p>
<p>vectors, negative values of u are equally important;�AB is marked off on the negative side of the origin,
or at BA within the given unit cell.
</p>
<p>If we consider next very small values of u, both ends of such vectors will lie inside one and the
</p>
<p>same electron density peak, and P(u) will be large. In the limit as u ! 0, the product involves that
of the electron density maximum with itself, which is a local maximum for each atom, and a very
</p>
<p>large peak at the origin (u &frac14; 0) is to be expected. Thus the Patterson function is represented as a
map of interatomic vectors, including null vectors, all taken to the origin, Sect. 6.9.2.
</p>
<p>Fig. 7.8 Development of a
one-dimensional Patterson
</p>
<p>function (b) for a two-atom
structure (a). Note the
centrosymmetry of the
</p>
<p>Patterson function that is
</p>
<p>lacking in the r(x) function
</p>
<p>284 7 Fourier Techniques in X-Ray Structure Determination</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec30_6">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec30_6</a></div>
</div>
<div class="page"><p/>
<p>The reader should confirm from Fig. 7.8, using tracing paper, that the positions of the peaks in
</p>
<p>Patterson space can be plotted graphically by placing each atom of the structure r(x) in turn at the
</p>
<p>origin of the Patterson map, in parallel orientation, and marking the positions of the other atoms on
</p>
<p>to the Patterson unit cell. Because of the centrosymmetry of the Patterson function, implicit in
</p>
<p>(7.14), it is not strictly necessary to plot vectors lying outside one-half of the unit cell.
</p>
<p>Three-Dimensional Patterson Function
If we replace r(x) and r(x + u) in (7.5) by the three-dimensional analogs r(xyz) and r(x + u, y + v,
</p>
<p>z + w) and integrate over a unit fractional volume, we can derive the three-dimensional Patterson
</p>
<p>function:
</p>
<p>P&eth;uvw&THORN; &frac14; 2
Vc
</p>
<p>X
</p>
<p>h
</p>
<p>X
</p>
<p>k
</p>
<p>X
</p>
<p>l
</p>
<p>jF&eth;hkl&THORN;j2 cos 2p&eth;hu&thorn; kv&thorn; lw&THORN; (7.15)
</p>
<p>where the summations range, in the most general case, over one half of experimental reciprocal space.
</p>
<p>This equation should be compared with (7.14) and (6.29) in Chap. 6: it is a Fourier series with zero
</p>
<p>phases and jFj2 as coefficients. Since jFj2 is F � F	, we see from (6.28) that (7.15) represents the
convolution of the electron density r(r) with its inversion in the origin, that is, with r(�r). In
practice, (7.15) may be handled like the corresponding electron density equation, with u, v, w repla-
</p>
<p>cing x, y, z. Both series explore the same field but their interpretation is different, as we shall see. The
</p>
<p>roving vector is now specified by three coordinates, u, v, and w, and P(uvw) is a maximum where the
</p>
<p>corresponding vector spans two atoms in the crystal.
</p>
<p>7.4.3 Positions and Weights of Peaks in the Patterson Function
</p>
<p>The positions of the peaks in P(uvw) may be plotted in three dimensions by placing each atom of the
</p>
<p>unit cell of a structure in turn at the origin of Patterson space, in parallel orientation, and mapping the
</p>
<p>positions of all other atoms on to the Patterson unit cell. Examples of this process are illustrated
</p>
<p>graphically in Fig. 7.9; for simplicity the origin peak is not shown in Fig. 7.9d. In Fig. 7.9a, all atoms
</p>
<p>and their translation equivalents produce vector peaks lying on the points of a lattice that is identical in
</p>
<p>shape and size to the crystal lattice. For example, atom 1 at x, y, z and its translation equivalent, 10, at x,
1 + y, z give rise to a vector ending at 0, 1, 0 on the Patterson map. Peaks of this nature accumulate at
</p>
<p>the corners of the Patterson unit cell in exactly the same way as those of the origin peak, P(000). From
</p>
<p>(7.15), we can derive the height of the origin peak:
</p>
<p>P&eth;000&THORN; &frac14; 2
Vc
</p>
<p>X
</p>
<p>h&frac14;0
</p>
<p>X1
</p>
<p>k
</p>
<p>X
</p>
<p>l&frac14;�1
jFo&eth;hkl&THORN;j2 (7.16)
</p>
<p>In general, (7.16) is equivalent to a superposition at the origin of all N products like r(xyz)r(xyz),
</p>
<p>where N is the number of atoms in the unit cell. Since r(xyz), is proportional to the atomic number Zj
of the jth atom, Sect. 7.3, we have
</p>
<p>P&eth;000&THORN; /
XN
</p>
<p>j&frac14;1
Z2j (7.17)
</p>
<p>A single vector interaction between two atoms j and k, Fig. 7.9b, will have a Patterson peak of
</p>
<p>height proportional to ZjZk. Hence, the height H&eth; j; k&THORN; of this peak will be given by
</p>
<p>7.4 Methods of Solving the Phase Problem 285</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Equ29_6">http://dx.doi.org/10.1007/978-1-4614-3954-7#Equ29_6</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_6">http://dx.doi.org/10.1007/978-1-4614-3954-7_6</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Equ28_6">http://dx.doi.org/10.1007/978-1-4614-3954-7#Equ28_6</a></div>
</div>
<div class="page"><p/>
<p>H&eth; j; k&THORN; � P&eth;000&THORN;ZjZk=
XN
</p>
<p>j&frac14;1
Z2j (7.18)
</p>
<p>where P(000) is calculated from (7.16). This equation can serve as a useful guide, but overlapping
</p>
<p>vectors may give rise to misleading indications. The reservations on peak heights already mentioned
</p>
<p>in Sect. 7.3 apply also to Patterson peaks. It should be remembered that the correct geometrical
</p>
<p>interpretation of Patterson peaks is of far greater significance than is an adherence to (7.18).
</p>
<p>In a structure with N atoms per unit cell, each atom forms a vector with the remaining N � 1
atoms. There are, thus, N(N � 1) non-origin peaks per unit cell. From (7.15), substitution of �u, �v,
�w for u, v, w, respectively, leaves P(uvw) unaltered, which is a statement of the centrosymmetry of
the Patterson function.
</p>
<p>The Patterson unit cell is of the same size and shape as the crystal unit cell, but it has to
</p>
<p>accommodate N2 rather than N peaks and is, therefore, correspondingly overcrowded. Thus, peaks
</p>
<p>in Patterson space tend to overlap when there are many atoms in the unit cell, a feature which
</p>
<p>introduces difficulties into the process of unraveling the function in terms of the correct distribution of
</p>
<p>atoms in the crystal.
</p>
<p>Fig. 7.9 Effects of symmetry-related and symmetry-independent atomson thePatterson function.Theweights of the peaks
are approximately proportional to the diameters of the circles: (a) P1(N &frac14; 1). (b) P1(N &frac14; 2): two atoms per unit cell
produce (22 � 2) non-origin peaks. (c) Pm(N &frac14; 2): two non-origin peaks, but with coordinates � {0, 2y, 0}.
(d) Pm(N &frac14; 4): 12 non-origin peaks per unit cell; for clarity the origin peak has not been drawn. The Patterson space
group is P�1 in (a) and (b) and P2/m in (c) and (d)
</p>
<p>286 7 Fourier Techniques in X-Ray Structure Determination</p>
<p/>
</div>
<div class="page"><p/>
<p>7.4.4 Sharpened Patterson Function
</p>
<p>In a conceptual point atom, the electrons would be concentrated at a point. The atomic scattering
</p>
<p>factor curves, Fig. 4.6, would be parallel to the abscissa and fwould be equal to the atomic number for
</p>
<p>all values of sin y=l and at all temperatures. The electron density for a crystal composed of point
atoms would show a much higher degree of resolution than does that for a real crystal. Put another
</p>
<p>way, the broad peaks representing real atoms, Fig. 6.4, would be replaced by peaks of very narrow
</p>
<p>breadth in the point-atom crystal.
</p>
<p>A plot of the mean value of Fo
2 against sin y=l for a typical set of data is shown in Fig. 7.10. The
</p>
<p>radial decrease in Fo
2 can be reduced by modifying Fo
</p>
<p>2 by a function which increases as sin y=l
increases. The coefficients for a sharpened Patterson synthesis may be calculated by the following
</p>
<p>equation. Sharpening can be effected also through the use of jEj values, Sect. 8.2.1.
</p>
<p>Fo;mod
2 &frac14; Fo
</p>
<p>2
</p>
<p>exp&frac12;�2B&eth;sin2y&THORN;=l2�
PN
</p>
<p>j&frac14;1 fj
n o2
</p>
<p>8
&gt;&lt;
&gt;:
</p>
<p>9
&gt;=
&gt;;
hkl
</p>
<p>(7.19)
</p>
<p>where N is the number of atoms in the unit cell and B is an overall isotropic temperature factor,
</p>
<p>Sect. 4.1.8.
</p>
<p>The effect of sharpening on a Patterson synthesis is illustrated in Fig. 7.17d, the Harker section
</p>
<p>u;
1
</p>
<p>2
;w
</p>
<p>� �
</p>
<p>for papaverine hydrochloride. It should be compared with Fig. 7.17b; the increased
</p>
<p>resolution is very apparent.
</p>
<p>Over-sharpening of Patterson coefficients may lead to spurious peaks because of series termination
</p>
<p>errors, Sect. 6.9.1, particularly where heavy atoms are present, and the technique should not be
</p>
<p>Fig. 7.10 Effect of
sharpening on the radial
</p>
<p>decrease of the local
</p>
<p>average intensity Fo
2
</p>
<p>7.4 Methods of Solving the Phase Problem 287</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Fig6_4">http://dx.doi.org/10.1007/978-1-4614-3954-7#Fig6_4</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Fig4_6">http://dx.doi.org/10.1007/978-1-4614-3954-7#Fig4_6</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec3_8">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec3_8</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec9_4">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec9_4</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec29_6">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec29_6</a></div>
</div>
<div class="page"><p/>
<p>applied without care. Sometimes the coefficients can be further modified to advantage by multipli-
</p>
<p>cation by a function such as exp(�m sin3 y), where m is chosen by trial, but might be about 5. This
function has the effect of decreasing the magnitude of the Fo
</p>
<p>2 curve at high y-values. Many other
</p>
<p>sharpening functions have been proposed, but we shall not dwell on this subject. It is often helpful to
</p>
<p>calculate both the normal and sharpened Patterson functions for comparison. Practice can be gained
</p>
<p>through exercises with the XRAY program system.
</p>
<p>7.4.5 Symmetry of the Patterson Function for a Crystal of Space Group Pm
</p>
<p>An inspection of Fig. 7.9c, d shows that the peaks on the line [0, v, 0] arise from atom pairs related by
</p>
<p>the m planes. The vector interactions for case (d) are listed in Table 7.1, and may be easily verified by
</p>
<p>the reader; the values z1 &frac14; z2 &frac14; 0.0 were chosen for convenience only.
The m planes in Pm are carried over into Patterson space, and relate the following pairs of peaks in
</p>
<p>the vector set:
</p>
<p>1; �1; 2; 5; �2; �5; 3; 4; �3; �4; 6; �6 (7.20)
</p>
<p>Furthermore, the presence of a center of symmetry in the diffraction pattern generates 2/m symmetry
</p>
<p>in the Patterson map, which corresponds to the Laue symmetry of all monoclinic crystals. Evidently, the
</p>
<p>symmetry of the diffraction pattern is impressed on to the Patterson function by the use of Fo
2
</p>
<p>coefficients in the Patterson series. As a consequence, the Patterson synthesis is computed in the
</p>
<p>primitive space group corresponding to the Laue symmetry of a crystal, and this situation is similar for
</p>
<p>all space groups.
</p>
<p>We can detect the presence of the twofold axis parallel to b in Fig. 7.9d through vector peaks such as 5,
�2 and 3, �4. Finally, the symmetry-related pairs of atoms in the crystal, 1, 3 and 2, 4, give rise to vectors
</p>
<p>along the line [0, v, 0] corresponding to the peaks 1, 6, �6, and �1 in Patterson space. The presence of a large
</p>
<p>number of peaks along an axis in a three-dimensional Pattersonmapmay be used as evidence for amirror
</p>
<p>plane perpendicular to that axis in the crystal. This feature is important because anm plane does not give
</p>
<p>rise to systematic absences in the diffraction pattern, Table 3.7. The existence of peaks, arising from
</p>
<p>symmetry-related atoms in certain regions of Patterson space was noted first by Harker in 1936. The line
</p>
<p>Table 7.1 Vectors generated by two independent atoms and their symmetry equivalents in space group Pma
</p>
<p>Subtraction of coordinates Reduced to one unit cell
</p>
<p>Atom pair Analytical form of vector u v u v Point in Fig. 7.9d
</p>
<p>(1), (3) �{0, 2y1, 0} 0 0.10 0 0.10 1
0 �0.10 0 0.90 �1
</p>
<p>(1), (2) �{x1 � x2, y1 � y2, z1 � z2} 0.15 �0.15 0.15 0.85 2
�0.15 0.15 0.85 0.15 �2
</p>
<p>(1), (4) �{x1 � x2, y1 + y2, z1 � z2} 0.15 0.25 0.15 0.25 3
�0.15 �0.25 0.85 0.75 �3
</p>
<p>(2), (3) �{x1 � x2, �y1 � y2, z1 � z2} 0.15 �0.25 0.15 0.75 4
�0.15 0.25 0.85 0.25 �4
</p>
<p>(3), (4) �{x1 � x2, �y1 + y2, z1 + z2} 0.15 0.15 0.15 0.15 5
�0.15 �0.15 0.85 0.85 �5
</p>
<p>(2), (4) �{0, 2y2, 0} 0 0.40 0 0.40 6
0 �0.40 0 0.60 �6
</p>
<p>aThe coordinates of the four atoms in two sets of general positions are x, y, z; x, �y, z with x1 &frac14; 0.20, y1 &frac14; 0.05,
x2 &frac14; 0.05, y2 &frac14; 0.20, and z1 &frac14; z2 &frac14; 0.00
</p>
<p>288 7 Fourier Techniques in X-Ray Structure Determination</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Tab7_3">http://dx.doi.org/10.1007/978-1-4614-3954-7#Tab7_3</a></div>
</div>
<div class="page"><p/>
<p>[0, v, 0] for Pm is called a Harker line; planes containing peaks arising from pairs of symmetry-related
</p>
<p>atoms are called Harker sections. We shall consider some examples below.
</p>
<p>7.4.6 Vector Interactions in Other Space Groups
</p>
<p>We shall consider atoms in general positions in a number of space groups that should be now familiar.
</p>
<p>Space Group P�1
</p>
<p>General positions: x, y, z; �x; �y; �z.
</p>
<p>Vectors: �{2x, 2y, 2z}.
Harker peaks lie in general positions in Patterson space.
</p>
<p>Space Group P2
</p>
<p>General positions: x, y, z; �x; y; �z.
Vectors: �{2x, 0, 2z}.
Harker section: (u, 0, w).
</p>
<p>It may be noted that for complex structures, not all of the peaks on Harker sections are necessarily
</p>
<p>true Harker peaks. If in this structure there are two atoms not related by symmetry, which, by chance,
</p>
<p>have the same or very nearly the same y coordinates, the vector between them will produce a peak on
</p>
<p>the Harker section.
</p>
<p>Space Group P2/m
</p>
<p>Vectors:
</p>
<p>�{2x, 0, 2z} Double weight Type 1
�{0, 2y, 0} Double weight Type 2
�{2x, 2y, 2z} Single weight Type 3
�{2x, 2�y, 2z} Single weight Type 4
</p>
<p>Harker section: (u, 0, w).
</p>
<p>Harker line: [0, u, 0].
</p>
<p>Vector type 1 arises in two ways, once from the pair x, y, z; �x; y; �z, and once from the pair x, �y, z;
�x; �y; �z. These two interactions give rise to identical vectors, which therefore superimpose in Patterson
space and form a double-weight peak. Similar comments apply to type 2, but the centrosymmetrically
</p>
<p>related atoms give rise to single-weight peaks, types 3 and 4. Figure 7.11 illustrates these vectors, as
</p>
<p>seen along the z axis. The reader may now consider how the Patterson function might be used to
</p>
<p>differentiate among space groups P2, Pm, and P2/m. Statistical methods, discussed in Chap. 4, are
</p>
<p>often employed to verify the results obtained from a study of the vector distribution.
</p>
<p>7.4.7 Examples of the Use of the Patterson Function in Solving
the Phase Problem
</p>
<p>In this section, we shall consider how the Patterson function was used in the solution of three quite
</p>
<p>different structures.
</p>
<p>Bisdiphenylmethyldiselenide, &eth;C6H5&THORN;2CHSe2 CH&eth;C6H5&THORN;2
Crystals of this compound form yellow needles, with straight extinction under crossed Polaroids for
</p>
<p>all directions parallel to the needle axis, and oblique extinction on the section normal to the needle
</p>
<p>7.4 Methods of Solving the Phase Problem 289</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_4">http://dx.doi.org/10.1007/978-1-4614-3954-7_4</a></div>
</div>
<div class="page"><p/>
<p>axis. Photographs taken with the crystal oscillating about its needle axis show only a horizontal m
</p>
<p>line, while zero- and upper-layer Weissenberg [4] photographs show only symmetry 2. The crystals
</p>
<p>are therefore monoclinic, with b along the needle direction.
</p>
<p>Crystal Data
</p>
<p>System: monoclinic
</p>
<p>Unit-cell dimensions: a &frac14; 18.72, b &frac14; 5.773, c &frac14; 12.594 Å, b &frac14; 125.47�
Vc: 1,108.5 Å
</p>
<p>3
</p>
<p>Dm: 1.49 g cm
�3
</p>
<p>Mr: 492.38
</p>
<p>Z: 2.02 or 2 to the nearest integer
</p>
<p>Unit-cell contents: 4Se, 52C, and 44H atoms
</p>
<p>Absent spectra: hkl: h + k &frac14; 2n
Possible space groups: C2, Cm, C2/m
</p>
<p>Symmetry Analysis
</p>
<p>Where the space group is not determined uniquely by the X-ray diffraction pattern, it may be possible to
</p>
<p>eliminate certain alternatives at the outset of the structure determination by other means.
</p>
<p>Space groups C2 and Cm each require four general positions:
</p>
<p>C2 : &eth;0; 0; 0; 1
2
; 1
2
; 0&THORN; &thorn; fx; y; z; �x; y; �zg Cm : &eth;0; 0; 0; 1
</p>
<p>2
; 1
2
; 0&THORN; &thorn; fx; y; z; x; �y; zg
</p>
<p>Since Z is 2, the molecular symmetry is either 2, in C2, or m, in Cm. In both C2 and Cm, all atoms
</p>
<p>could satisfy general position requirements, and neither arrangement would be stereochemically
</p>
<p>unreasonable.
</p>
<p>Fig. 7.11 (a) Vectors
between symmetry-related
</p>
<p>atoms in general equivalent
</p>
<p>positions in space group
</p>
<p>P2/m. Coordinates like �x
have been treated as
</p>
<p>(1 � x) in drawing the
vectors. (b) One unit cell of
the Patterson function: the
</p>
<p>twofold axes intersect the
</p>
<p>m planes in centers of
</p>
<p>symmetry. Note the single-
</p>
<p>weight and double-weight
</p>
<p>peaks, and their relation to
</p>
<p>the space-group symmetry
</p>
<p>290 7 Fourier Techniques in X-Ray Structure Determination</p>
<p/>
</div>
<div class="page"><p/>
<p>Space group C2/m requires eight general equivalent positions per unit cell. Only special position
</p>
<p>sets, such as 0, 0, 0 and 1
2
, 1
2
, 0 correspond with Z &frac14; 2. These positions have symmetry 2/m, but it is not
</p>
<p>possible to construct the molecule in this symmetry without contradicting known chemical facts.
</p>
<p>Consequently, we shall regard this space group as highly improbable for the compound under
</p>
<p>investigation.
</p>
<p>Patterson Studies
</p>
<p>Whatever the answer to the questions remaining from this symmetry analysis, we expect, from the
</p>
<p>covalent radius of selenium (1.22 Å) that the two selenium atoms will be covalently bonded at a
</p>
<p>distance of about 2.4 Å. This Se&ndash;Se interaction will produce a strong peak in the Patterson function at
</p>
<p>about this distance from the origin.
</p>
<p>The atomic numbers of Se, C, and H are 34, 6, and 1, respectively. Hence, the important vectors in the
</p>
<p>Patterson function would have approximate single-weight peak heights, from (7.18), as follows:
</p>
<p>(a) Se&ndash;Se: 1156.
</p>
<p>(b) Se&ndash;C: 204.
</p>
<p>(c) C&ndash;C: 36.
</p>
<p>Because of the presence of identical vectors arising from the C unit cell, all vectors will be double
</p>
<p>these values.
</p>
<p>Figure 7.12 is the Patterson section P(u0w), calculated with 1053 Fo(hkl)
2 data, with grid intervals
</p>
<p>of 50th along u, v, and w. The origin peak P(000) was scaled to 100 and, from (7.17),PN
j&frac14;1 Z
</p>
<p>2
j &frac14; 6540. Hence, the vector interactions (a), (b), and (c) should have peak heights in the
</p>
<p>approximate ratio 32:5.7:1.
</p>
<p>The section is dominated by a large peak of height 39 at a distance of about 2.4 Å from the origin.
</p>
<p>Making the reasonable assumption that it represents the Se&ndash;Se vector, and since there are no
</p>
<p>Fig. 7.12 Patterson section, P(u0w); the origin peak (height &frac14; 100) has not been contoured. Contours around the
Se&ndash;Se peaks are at intervals of 4; elsewhere at intervals of 2
</p>
<p>7.4 Methods of Solving the Phase Problem 291</p>
<p/>
</div>
<div class="page"><p/>
<p>significant peaks on the v axis, the Harker line in Cm, it follows that the space group cannot be Cm,
</p>
<p>thus leaving C2 as the most logical choice.
</p>
<p>In space group C2 from the above list, it follows that vectors between atoms in general positions take
</p>
<p>the form: (u, 0, w), where u &frac14; 2x, and w &frac14; 2z, plus the C-centered equivalents. For the sulphur atom
vectors u &frac14; 2xSe, w &frac14; 2zSe. Hence by measurement on the section, the Patterson coordinates are
u &frac14; 6.7/50 &frac14; 0.134 and w &frac14; 2.2/50 &frac14; 0.044, so that xSe &frac14; 0.067 and zSe &frac14; 0.022.
</p>
<p>In space group C2, the unit-cell origin is fixed in the x, z plane by the twofold axis. There is no
</p>
<p>symmetry element that defines the origin in the y direction, which must be fixed by specifying the y
</p>
<p>coordinate for a selected atom. For convenience, we may set ySe &frac14; 0, and our analysis so far may be
given as the positions
</p>
<p>Se: 0.067, 0, 0.022
</p>
<p>Se0: �0.067, 0, �0.022
A space-group ambiguity is not always resolved in this manner. Sometimes it is necessary to proceed
</p>
<p>further with the structure analysis, even to refinement stages, before confirmation is obtained.
</p>
<p>What of the atoms other than selenium? Is it possible to determine the positions of the carbon and
</p>
<p>hydrogen atoms? We shall find that we can locate the carbon atoms in this structure from the Patterson
</p>
<p>synthesis. To explain the procedure, we consider first only part of the structure, including one phenyl ring
</p>
<p>of the asymmetric unit, Fig. 7.13a, and neglect all but the C&ndash;Se vectors. The vector set generated by the
</p>
<p>2 Se atoms and 14 C atoms in this hypothetical arrangement contains two images of the structure
</p>
<p>fragment, one per Se atom, which are displaced from each other by the Se&ndash;Se vector. The idealized
</p>
<p>vector set is shown in Fig. 7.13b. By shifting one of these images by a reverse1 Se&ndash;Se vector
</p>
<p>displacement, it is possible to bring the two images into coincidence. Verify this statement by making
</p>
<p>a transparent copy of Fig. 7.13b and placing its origin over an Se&ndash;Se vector position in the original figure,
</p>
<p>keeping the pairs of u and w axes parallel. Certain peaks overlap, producing a single, displaced image of
</p>
<p>the structure. Shade the peaks that overlap. This image is displaced with respect to the true space-group
</p>
<p>origin, which we know to be midway between the two Se atoms. A correctly placed image of the
</p>
<p>structure can be recovered by inserting the true origin position on to the tracing and neglecting any peaks
</p>
<p>that are not shaded.
</p>
<p>The partial vector set was formed from the image of all atoms of the fragment in each Se atom;
</p>
<p>each image is weighted by Zj, the atomic number of the jth atom, (carbon, in this example), imaged in
</p>
<p>Se. The displacement arises because the Patterson synthesis transfers all vectors to a common origin.
</p>
<p>Fig. 7.13 (a) Hypothetical
structure fragment
</p>
<p>C6H5CHSe2CHC6H5 in
</p>
<p>real space; (b) idealized set
of Se&ndash;Se and Se&ndash;C vectors
</p>
<p>in Patterson space
</p>
<p>1 If the forward direction of this vector is used, the structure obtainedwould, in general, be inverted through the origin. This
</p>
<p>does not happen with the example under study because the molecule possesses only twofold symmetry.
</p>
<p>292 7 Fourier Techniques in X-Ray Structure Determination</p>
<p/>
</div>
<div class="page"><p/>
<p>Patterson Superposition
</p>
<p>The technique just described depends upon the recognition of the vector interaction from a given
</p>
<p>pair of atoms, the two Se atoms in this example. At least a partial unscrambling of the structure
</p>
<p>images in the Patterson function was effected by correctly displacing two copies of the Patterson
</p>
<p>map and noting the positions of overlap.
</p>
<p>To illustrate the method further and to derive a systematic procedure for its implementation, we
</p>
<p>return to the Patterson section in Fig. 7.12. The two Se atoms have the same y coordinate, which
</p>
<p>means that the vector shift takes place in this section. Now, make two copies on tracing paper of the
</p>
<p>half unit-cell outline, x &frac14; 0� 1
2
and z &frac14; 0&ndash;1, and label them copy 1 and copy 2.
</p>
<p>On copy 1 mark in the position S of the point,�(xSe, zSe), which is at�0.067,�0.022, and on copy
2 mark in the position S0 of the point (xSe0 ; zSe0), which is at 0.067, 0.022. Think of these two unit cells
as existing in crystal space, not Patterson space. Place copy 1 over the Patterson (u, 0, w) section,
</p>
<p>maintaining a parallel orientation, with S over the origin, and trace out the Patterson map, Fig. 7.14a,
</p>
<p>excluding the origin peak in each case. Repeat this procedure with copy 2, placing S0 over the
Patterson section origin, Fig. 7.14b.
</p>
<p>Finally, superimpose copy 1 and copy 2. As in the exercise with Fig. 7.13a, b, some peaks overlap
</p>
<p>and some lie over blank regions in one or the other map. The overlaps correspond to regions of high
</p>
<p>electron density in the crystal. They are best mapped out by compiling a new diagram which contains
</p>
<p>theminimum value of the vector density between copy 1 and copy 2 for each point, thus eliminating or
</p>
<p>decreasing in height those regions where one copy has no or only slight overlap. A map prepared in
</p>
<p>this way is shown in Fig. 7.14c.
</p>
<p>Minimum Function
The technique outlined above follows the method of Buerger.2 An analytical expression for the
</p>
<p>minimum functionMn(xyz) is given by (7.21); it may be regarded as an approximation to the electron
</p>
<p>density r(xyz).
</p>
<p>Mn&eth;xyz&THORN; &frac14; Min&frac12;P&eth;u� x1; v� y1;w� z1&THORN;; P&eth;u� x2; v� y2;w� z2&THORN;; . . . ;
P&eth;u� xn; v� yn;w� zn&THORN;�
</p>
<p>(7.21)
</p>
<p>where Min(P1, P2, . . ., Pn) represents the lowest value at the point x, y, z in the set of super positions
</p>
<p>P1, P2, . . ., Pn; n corresponds with the number of known or trial atomic positions. The following
</p>
<p>general comments on the application of the minimum function procedure should be noted:
</p>
<p>1. The n trial atoms should form within themselves a set or sets of points related by the appropriate
</p>
<p>space-group symmetry.
</p>
<p>2. In a non-centrosymmetric space group, n should be three or more in order to remove the Patterson
</p>
<p>center of symmetry.
</p>
<p>3. If the various n trial atoms have different atomic numbers, the corresponding Patterson copies
</p>
<p>should be weighted accordingly in order to even out the different image strengths.
</p>
<p>4. Incorrectly placed atoms in the trial set tend to confuse the structure image. New atom sites
</p>
<p>therefore should be added to the model with caution.
</p>
<p>Figure 7.15 shows a composite electron density map of the atoms in the asymmetric unit that
</p>
<p>were revealed by a three-dimensional minimum function M2. This result is quite satisfactory; only
</p>
<p>C(9), C(10), and C(11) are not yet located. The composite map of the complete structure [5] and the
</p>
<p>2 See Bibliography, Buerger (1959).
</p>
<p>7.4 Methods of Solving the Phase Problem 293</p>
<p/>
</div>
<div class="page"><p/>
<p>Fig. 7.14 Bisdiphenylmethyldiselenide. (a, b) Shifted copies 1 and 2 prepared from the (u0w) section. (c) Minimum-
function M2 section at y &frac14; 0.0; C0(12) and C0(13) are symmetry-related to C(12) and C(13) in Fig. 7.15
</p>
<p>294 7 Fourier Techniques in X-Ray Structure Determination</p>
<p/>
</div>
<div class="page"><p/>
<p>packing of the molecules in the unit cell are shown in Fig. 7.16. In favorable circumstances, the
</p>
<p>Patterson function can be solved for the majority of the heavier atoms in the crystal structure. The
</p>
<p>atoms not located by M2 in this example were obtained from an electron density map phased
</p>
<p>on those atoms that were found, a standard method for attempting to complete a partial structure,
</p>
<p>see Sect. 7.5.
</p>
<p>Fig. 7.15 Composite map of the three-dimensional minimum function M2(xyz) for bisdiphenylmethyldiselenide
</p>
<p>Fig. 7.16 Bisdiphenylmethyldiselenide: (a) Composite electron density map as seen along b; (b) crystal structure as
seen along b; the dashed lines indicate the closest intermolecular contacts
</p>
<p>7.4 Methods of Solving the Phase Problem 295</p>
<p/>
</div>
<div class="page"><p/>
<p>7.4.8 Determination of the Chlorine Atom Positions in Papaverine
Hydrochloride [1]
</p>
<p>The crystal data for this compound have been given in Sect. 7.2.1. The calculated origin peak height is
</p>
<p>approximately 4700, and a single-weight Cl&ndash;Cl vector would have a height of about 6% of that of the
</p>
<p>origin peak. The Cl&ndash;Cl vector may not be located as easily as that of Se&ndash;Se in the previous example.
</p>
<p>The general equivalent positions in P21/c give rise to the vectors shown in Table 7.2. The assignment
</p>
<p>of coordinates to the chlorine atoms follows the recognition of peaks A, B, and C as Cl&ndash;Cl vectors on
</p>
<p>the Patterson maps, Fig. 7.17a&ndash;c. Figure 7.17d is the sharpened section, &eth;u; 1
2
; w&THORN;. The steps in the
</p>
<p>solution of the problem are set out in Table 7.3.
</p>
<p>The results are completely self-consistent, and we may list the Cl coordinates in the unit cell:
</p>
<p>4Cl : 0:025; 0:169; 0:038; 0:025; 0:331; 0:538
</p>
<p>� 0:025; �0:169; �0:038; �0:025; �0:331; �0:538
</p>
<p>For simplicity, peak A was assigned as � &eth;1
2
&thorn; 2y&THORN;, which is crystallographically the same as
</p>
<p>1
</p>
<p>2
� 2y, in order to obtain y
 1
</p>
<p>2
. For a similar reason, B was retained as
</p>
<p>1
</p>
<p>2
&thorn; 2z.
</p>
<p>The specification of the peak parameters in this manner is, to some extent, dependent on the observer.
</p>
<p>A different choice, for example,
1
</p>
<p>2
&thorn; 2y in A, merely results in a set of atomic positions located with
</p>
<p>respect to one of the other centers of symmetry as origin. In space groups where the origin might be
</p>
<p>defined with respect to other symmetry elements, similar arbitrary peak specifications may be possible.
</p>
<p>7.4.9 Determination of the Mercury Atom Positions in KHg2
</p>
<p>This example illustrates the application of the Patterson function to the determination of the
</p>
<p>coordinates of atoms in special positions of space group Imma.
</p>
<p>Crystal Data3
</p>
<p>System: orthorhombic
</p>
<p>Unit-cell dimensions: a &frac14; 8.10, b &frac14; 5.16, c &frac14; 8.77 Å
Vc: 366.55 Å
</p>
<p>3
</p>
<p>Dm: 7.95 g cm
�3
</p>
<p>Mr: 440.28
</p>
<p>Table 7.2 Patterson peaks in space group P21/c
</p>
<p>Label Vector Peak strength Harker region
</p>
<p>A � f0; 1
2
&thorn; 2y; 1
</p>
<p>2
g Double weight Line: &eth;0; v; 1
</p>
<p>2
&THORN;
</p>
<p>B � f2x; 1
2
; 1
2
&thorn; 2zg Double weight Section: &eth;u; 1
</p>
<p>2
;w&THORN;
</p>
<p>C �{2x, 2y, 2z} Single weight General region
D � f2x; 2�y; 2xg Single weight General region
</p>
<p>3 In the original paper, the origin in Imma was chosen on a center of symmetry displaced by 1
4
, 1
4
, 1
4
from this origin.
</p>
<p>296 7 Fourier Techniques in X-Ray Structure Determination</p>
<p/>
</div>
<div class="page"><p/>
<p>Fig. 7.17 Three-dimensional Patterson sections for papaverine hydrochloride; the Cl&ndash;Cl vectors are labeled A, B, and
</p>
<p>C. (a) v &frac14; 8.4/52, (b) v &frac14; 1
2
. (c) v &frac14; 17.6/52. (d) v &frac14; 1
</p>
<p>2
(sharpened section)
</p>
<p>Table 7.3 Heavy-atom coordinates for papaverine hydrochloride
</p>
<p>Patterson map Label Vector coordinatesa Cl coordinates
</p>
<p>Figure 7.17a, level v &frac14; 8.4/52 A 1
2
� 2y &frac14; 8:4=52 y &frac14; 0.169
</p>
<p>Figure 7.17b, level v &frac14; 1
2
</p>
<p>B 2x &frac14; 2.2/44 x &frac14; 0.025
1
2
&thorn; 2z &frac14; 17:3=30 z &frac14; 0.038
</p>
<p>Figure 7.17c, level v &frac14; 17.6/52 C 2x &frac14; 2.2/44 x &frac14; 0.025
2y &frac14; 17.6/52 y &frac14; 0.169
2z &frac14; 2.3/30 z &frac14; 0.038
</p>
<p>aThe Patterson synthesis was computed with the intervals of subdivision 44, 52, and 30 and
</p>
<p>along u, v, and w, respectively</p>
<p/>
</div>
<div class="page"><p/>
<p>Z: 3.99 or 4 to the nearest integer
</p>
<p>Unit-cell contents: 4K and 8Hg atom
</p>
<p>Absent spectra: hkl: h + k + l &frac14; 2n + 1; hk0: h &frac14; 2n + 1 (k &frac14; 2n + 1)
From the diffraction data, possible space groups are Im2a, I2ma, or Imma. In the absence of further
</p>
<p>information on the space group, we shall proceed with the analysis in Imma, Fig. 7.18a, b; the reader
</p>
<p>may like to consider how easily this figure may be derived from Pmma, origin on �1, + I.
</p>
<p>Symmetry and Packing Analyses
Since Z is 4 and there are 16 general equivalent positions in Imma, all atomsmust lie in special positions.
</p>
<p>Table 7.4 lists these positions for this space group, with a center of symmetry (2/m) as origin [6].
</p>
<p>This list presents a quite formidable number of alternatives for examination. The eight Hg
</p>
<p>atoms could lie in (f), (g), (h), or (i). However, further consideration of sets (f), (g), and (i) and sets
</p>
<p>(c) and (d) shows that they would all involve pairs of Hg atoms being separated by distances less
</p>
<p>than b/2 (2.58 Å). This value is much shorter than known Hg&ndash;Hg bond distances in other structures,
</p>
<p>and we shall reject these sets. The positions in these sets may be plotted to scale in order to verify
</p>
<p>the spatial limitations.
</p>
<p>Fig. 7.18 Space group
Imma (rotated by 90�
</p>
<p>from the standard setting):
</p>
<p>(a) general equivalent
positions, (b) symmetry
elements
</p>
<p>298 7 Fourier Techniques in X-Ray Structure Determination</p>
<p/>
</div>
<div class="page"><p/>
<p>Of the remaining sets, (a) and (b) together would again place neighboring Hg atoms too close to one
</p>
<p>another. There are three likely models:
</p>
<p>Model I: four Hg in (a) + four Hg in (e).
</p>
<p>Model II: four Hg in (b) + four Hg in (e).
</p>
<p>Model III: eight Hg in (h).
</p>
<p>The Patterson function enables us to differentiate among these alternative models.
</p>
<p>Fig. 7.19 Patterson projection P(uw) for KHg2; the origin peak has not been contoured and the labeled peaks are
Hg&ndash;Hg vectors
</p>
<p>Table 7.4 Special positions in Imma
</p>
<p>4 (a) 2/m 0; 0; 0;
1
</p>
<p>2
; 0; 0;
</p>
<p>1
</p>
<p>2
;
1
</p>
<p>2
;
1
</p>
<p>2
; 0;
</p>
<p>1
</p>
<p>2
;
1
</p>
<p>2
</p>
<p>4 (b) 2/m 0;
1
</p>
<p>2
; 0;
</p>
<p>1
</p>
<p>2
;
1
</p>
<p>2
; 0;
</p>
<p>1
</p>
<p>2
; 0;
</p>
<p>1
</p>
<p>2
; 0; 0;
</p>
<p>1
</p>
<p>2
</p>
<p>4 (c) 2/m
1
</p>
<p>4
;
1
</p>
<p>4
;
1
</p>
<p>4
;
</p>
<p>1
</p>
<p>4
;
3
</p>
<p>4
;
1
</p>
<p>4
;
</p>
<p>3
</p>
<p>4
;
3
</p>
<p>4
;
3
</p>
<p>4
;
</p>
<p>3
</p>
<p>4
;
1
</p>
<p>4
;
3
</p>
<p>4
</p>
<p>4 (d) 2/m
1
</p>
<p>4
;
3
</p>
<p>4
;
3
</p>
<p>4
;
</p>
<p>1
</p>
<p>4
;
1
</p>
<p>4
;
3
</p>
<p>4
;
</p>
<p>3
</p>
<p>4
;
1
</p>
<p>4
;
1
</p>
<p>4
;
</p>
<p>3
</p>
<p>4
;
3
</p>
<p>4
;
1
</p>
<p>4
</p>
<p>4 (e) mm2
1
</p>
<p>4
; 0; z;
</p>
<p>3
</p>
<p>4
; 0; �z;
</p>
<p>3
</p>
<p>4
;
1
</p>
<p>2
;
1
</p>
<p>2
&thorn; z; 1
</p>
<p>4
;
1
</p>
<p>2
;
1
</p>
<p>2
� z
</p>
<p>8 (f) 2 � 0; y; 0; 1
2
; y; 0;
</p>
<p>1
</p>
<p>2
;
1
</p>
<p>2
&thorn; y; 1
</p>
<p>2
; 0;
</p>
<p>1
</p>
<p>2
&thorn; y; 1
</p>
<p>2
</p>
<p>� �
</p>
<p>8 (g) 2 � x; 1
4
;
1
</p>
<p>4
; x;
</p>
<p>3
</p>
<p>4
;
1
</p>
<p>4
;
</p>
<p>1
</p>
<p>2
&thorn; x; 3
</p>
<p>4
;
3
</p>
<p>4
;
</p>
<p>1
</p>
<p>2
&thorn; x; 1
</p>
<p>4
;
3
</p>
<p>4
</p>
<p>� �
</p>
<p>8 (h) m � x; 0; z; 1
2
� x; 0; z; 1
</p>
<p>2
&thorn; x; 1
</p>
<p>2
;
1
</p>
<p>2
&thorn; z; �x; 1
</p>
<p>2
;
1
</p>
<p>2
&thorn; z
</p>
<p>� �
</p>
<p>8 (i) m � 1
4
; y; z;
</p>
<p>1
</p>
<p>4
; �y; z;
</p>
<p>3
</p>
<p>4
;
1
</p>
<p>2
&thorn; y; 1
</p>
<p>2
&thorn; z; 3
</p>
<p>4
;
1
</p>
<p>2
� y; 1
</p>
<p>2
&thorn; z
</p>
<p>� �
</p>
<p>7.4 Methods of Solving the Phase Problem 299</p>
<p/>
</div>
<div class="page"><p/>
<p>Vector Analysis of the Alternative Hg Positions
Model I would produce, among others, an Hg&ndash;Hg vector at u &frac14; 1
</p>
<p>2
, w &frac14; 0, from the atoms in set (a).
</p>
<p>The b axis Patterson projection, Fig. 7.19, shows no peak at that position, and we eliminate model I.
</p>
<p>For a similar reason, with the atoms of set (b), model II is rejected. It is necessary to show next that
</p>
<p>model III is consistent with the Patterson function. The a-axis projection is shown in Fig. 7.20.
</p>
<p>Interpretation of P(uw)
</p>
<p>In this projection, no reference is made to the y coordinates, and we look for vectors of the type
</p>
<p>� f1
2
&thorn; 2x; 0g and �{ 1
</p>
<p>2
, 2z}, and four vectors related by 2mm symmetry �{2x, 2z} and �f2�x, 2z}.
</p>
<p>The double-weight peak labeled B(1) is on the line w &frac14; 0, and B(2) is on the line u &frac14; 1
2
. Hence,
</p>
<p>xHg &frac14; 0.064 and zHg &frac14; 0.161. These values are corroborated by measurements from the single
weight peak B(3).
</p>
<p>Interpretation of P(vw)
</p>
<p>Vectors likeA, Fig. 7.20, are of the type�{0, 2z}.We deduce zHg &frac14; 0.161, in excellent agreement with the
value obtained from the b-axis projection.
</p>
<p>Superposition techniques applied to the a-axis projection indicate that the K atoms are in special
</p>
<p>positions (b), but this result is not supported by the b-axis projection. Evidently, the Patterson results can
</p>
<p>give only a partial structure, and supplementary methods are needed to carry the analysis to completion.
</p>
<p>In summary, we have determined the positions of the mercury atoms to be in set (h),4 Table 7.4, with
</p>
<p>x &frac14; 0.064, z &frac14; 0.161.
</p>
<p>Fig. 7.20 Patterson projection P(vw) for KHg2; the origin peak has not been contoured
</p>
<p>4 In the work of Duwell and Baenziger [6], the positions listed are 8 (i), with x &frac14; 0.186 and z &frac14; 0.089, each being 1
4
</p>
<p>minus the value given here.
</p>
<p>300 7 Fourier Techniques in X-Ray Structure Determination</p>
<p/>
</div>
<div class="page"><p/>
<p>7.5 Heavy-Atom Method and Partial Fourier Synthesis
</p>
<p>The heavy-atom method was conceived originally as a method for determining the positions of light
</p>
<p>atoms in a structure containing a relatively small number of heavier atoms. However, the technique
</p>
<p>can be applied to most situations where a partial structure analysis has been effected, provided that
</p>
<p>certain conditions are met.
</p>
<p>Imagine a situation where Nk, of the N atoms in a unit cell have been located; Nk may be only one
</p>
<p>atom, if it is a heavy atom. There will be Nu atoms remaining to be located, and we may express the
</p>
<p>structure factor in terms of known (k) and unknown (u) atoms:
</p>
<p>F&eth;hkl&THORN; &frac14;
XNk
</p>
<p>j&frac14;1
gj exp&frac12;i2p&eth;hxj &thorn; kyj &thorn; lzj&THORN;� &thorn;
</p>
<p>XNu
</p>
<p>u&frac14;1
gu exp&frac12;i2p&eth;hxu &thorn; kyu &thorn; lzu&THORN;� (7.22)
</p>
<p>or
</p>
<p>jF&eth;hkl&THORN;j &frac14; jFc&eth;hkl&THORN;j &thorn; jFu&eth;hkl&THORN;j (7.23)
</p>
<p>In practice, Fo data, appropriately scaled, replace jF(hkl)j, and Fc(hkl) refers to the known (Nk)
atomic positions. As more of the structure becomes known, the values of jFc(hkl)j approach Fo(hkl)
and the phase angle fc approaches the unobservable but required value f(hkl). Figure 7.21 illustrates
</p>
<p>this argument for any given reflection. The values of fc may provide sufficiently reasonable
</p>
<p>approximations to f(hkl) for an electron density map to be calculated with some confidence. The
</p>
<p>nearer jFcj is to Fo, the better the values of the phase angles, and this is clearly dependent upon the
percentage of the scattering power which is known. As a guide to the effective phasing power of a
</p>
<p>partial structure, the quantity r may be calculated:
</p>
<p>r &frac14;
XNk
</p>
<p>j&frac14;1
Z2j =
</p>
<p>XNu
</p>
<p>u&frac14;1
Z2u (7.24)
</p>
<p>Fig. 7.21 Partial-structure phasing; F(hkl) is the true structure factor of modulus Fo(hkl) and phase f(hkl)
</p>
<p>7.5 Heavy-Atom Method and Partial Fourier Synthesis 301</p>
<p/>
</div>
<div class="page"><p/>
<p>where Z refers to the atomic number of a species. A value of r near unity is considered to provide a
</p>
<p>useful basis for application of the heavy-atommethod. However, values of r quite different from unity
</p>
<p>have produced successful results, because for a given reflection the important quantity is really r0, the
scattering ratio, given by
</p>
<p>r0 &frac14;
XNk
</p>
<p>j&frac14;1
g2j =
</p>
<p>XNu
</p>
<p>u&frac14;1
g2u (7.25)
</p>
<p>If r is large, however, the heavy-atom contributions tend to swamp those from the lighter atoms,
</p>
<p>which may then not be located very precisely from electron density maps. On the other hand, if r is
</p>
<p>small, the calculated phases may deviate widely from the desired values, and the resulting electron
</p>
<p>density map could be very difficult to interpret. These extreme situations are found in two of the
</p>
<p>structures just studied, bisdiphenylmethyldiselenide (r &frac14; 2.4) and papaverine hydrochloride
(r &frac14; 0.28), based, in each case, on the heavy atoms alone in Nk.
</p>
<p>The underlying philosophy of the heavy-atom method depends on the acceptance of calculated
</p>
<p>phases, even if they contain errors, for the computation of the electron density synthesis. Large phase
</p>
<p>errors give rise to high background features, which mask the image of the correct structure. The
</p>
<p>calculated phases fc contain errors arising from inadequacies in the model, but the Fo data, although
</p>
<p>subject to experimental errors, hold information on the complete structure. Phase errors may be counter-
</p>
<p>acted to some extent by weighting the Fourier coefficients according to the degree of confidence in a
</p>
<p>particular phase. For centrosymmetric structures, the weight w(hkl) by which Fo(hkl) is multiplied is
</p>
<p>given by [7]
</p>
<p>w&eth;hkl&THORN; &frac14; tanh&eth;w=2&THORN; (7.26)
</p>
<p>where w is given by
</p>
<p>w &frac14; 2FojFcj=
X
</p>
<p>g2u (7.27)
</p>
<p>The subscripts c and u refer, respectively, to the known and unknown parts of the structure. In non-
</p>
<p>centrosymmetric structures, w(hkl) can be obtained from the graph [8] in Fig. 7.22. Weighting factors
</p>
<p>should be applied to Fo values that have been placed on an absolute, or approximately absolute, scale.
</p>
<p>We can show, in a simplified manner, how one may reasonably expect the heavy-atom procedure
</p>
<p>to be successful. In a centrosymmetric structure, the two terms on the right-hand side of (7.22) would
</p>
<p>Fig. 7.22 Weighting
factor w(hkl) as a
</p>
<p>function of w in non-
</p>
<p>centrosymmetric crystals
</p>
<p>302 7 Fourier Techniques in X-Ray Structure Determination</p>
<p/>
</div>
<div class="page"><p/>
<p>be cosine expressions. The sum over the Nk atoms would have a magnitudeM1 and be either + or� in
sign. Similarly, the sum over the Nu atoms would have a magnitudeM2 together with a + or a� sign.
Over a number of reflections, we may say that there is a 50% chance that the true signs are those given
</p>
<p>by the heavy atom. For the other 50%, there is a 25% chance that M1 &gt; M2, so that again the sign
</p>
<p>given by the heavy atom is correct. Thus, there is a good chance that a large percentage of the
</p>
<p>reflections will be given the correct sign in a favorable heavy-atom application.
</p>
<p>Bearing all these points in mind, it follows that the best electron density map one can calculate
</p>
<p>with phases determined from a partial structure is given by
</p>
<p>r&eth;xyz&THORN; &frac14; 2
Vc
</p>
<p>X
</p>
<p>h&frac14;0
</p>
<p>X1
</p>
<p>k&frac14;�1
</p>
<p>X
</p>
<p>l&frac14;�1
w&eth;hkl&THORN;Fo&eth;hkl&THORN; cos&frac12;2p&eth;hx&thorn; ky&thorn; lz&THORN; � fc&eth;hkl&THORN;� (7.28)
</p>
<p>where
</p>
<p>fc&eth;hkl&THORN; &frac14; tan�1&frac12;B0c&eth;hkl&THORN;=A0c&eth;hkl&THORN;� (7.29)
</p>
<p>and A0c&eth;hkl&THORN; and B0c&eth;hkl&THORN; are the real and imaginary components, respectively, of the calculated
structure factor, which is included in the right-hand side of (7.23).
</p>
<p>Electron density maps calculated from partial-structure phasing contain features which character-
</p>
<p>ize both the true structure and the partial, or trial, structure. We have considered this situation in Sect.
</p>
<p>6.9.1. Now, we may let each observation in (6.75), be multiplied by the weight w so as to give a better
</p>
<p>statistical significance to each term in the calculation.
</p>
<p>If the model includes atoms in reasonably accurate positions, we can expect two important features
</p>
<p>in the electron density map: (a) atoms of the trial structure should appear, possibly in corrected
</p>
<p>positions, and (b) additional atoms should be revealed by the presence of peaks in stereochemically
</p>
<p>sensible positions.
</p>
<p>If neither of these features is observed in the electron density synthesis, it may be concluded that
</p>
<p>the trial structure contains very serious errors, and we would be on a false trail. Correspondingly,
</p>
<p>there would be poor agreement in the pattern of relationship between Fo and jFcj.
</p>
<p>7.5.1 Reliability Factor
</p>
<p>Introduction
The use of different R factors is a popular way of quantifying various procedures at different stages of
</p>
<p>an analysis. While the calculation of each R factor is not necessarily based on rigorous principles, it is
</p>
<p>a simple and convenient way to record a quantitative assessment of quality as an aid to subsequent
</p>
<p>decision making. In essence each R factor measures an agreement in terms of the ratio R &frac14; D=M,
where D is the mean difference between measurements that are expected to have the same value
</p>
<p>within experimental error and M is the mean value of the set of measured quantities. These averages
</p>
<p>are usually derived from a large number of measurements in a typical crystal structure analysis. As an
</p>
<p>example, we consider intensity data recorded from a crystal having a twofold symmetry axis parallel
</p>
<p>to the y axis. Ideally, I&eth;hkl&THORN; &frac14; I&eth;h k �l&THORN; so that DI &frac14; jI&eth;hkl&THORN; � I&eth;h k �l&THORN;j may be used to calculate an R
factor called Rsym.
</p>
<p>If the intensity measurements do actually correspond according to twofold symmetry, Rsym
calculated in this way will be very small, because each DI term will tend to be zero within
</p>
<p>experimental error. An unacceptably high value of Rsym could indicate that either there is not
</p>
<p>7.5 Heavy-Atom Method and Partial Fourier Synthesis 303</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec29_6">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec29_6</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Equ75_6">http://dx.doi.org/10.1007/978-1-4614-3954-7#Equ75_6</a></div>
</div>
<div class="page"><p/>
<p>actually a twofold axis parallel to y, or the crystal quality is poor. A solution to this problem would be
</p>
<p>to repeat data measurements with a fresh crystal with subsequent reassessment of the situation. In the
</p>
<p>case of a macromolecular crystal, it should be remembered that diffracted intensities tend to
</p>
<p>much more accurate values at low resolution than at high resolution where the actual intensity
</p>
<p>values become much smaller owing to the effect of disorder and high thermal factors. For this reason,
</p>
<p>it is common to calculate Rsym as a function of resolution by partitioning the data into resolution
</p>
<p>&ldquo;shells&rdquo; or &ldquo;bins.&rdquo;
</p>
<p>A quantity very similar to Rsym, known as Rmerge, may be calculated as a by-product of data
</p>
<p>collection: Rsym and Rmerge are based on different sets of data and should not be treated as being
</p>
<p>equivalent. As a practical warning it should be remembered that neither Rsym nor Rmerge, as with other
</p>
<p>R indices, can stand up to rigorous analysis, and should be treated only as guides to data quality and
</p>
<p>aides to decision making. As a rule of thumb for high-resolution small-molecule structures Rsym may
</p>
<p>be as low as 2% and more usually 4&ndash;6%. For low resolution protein data Rsym would rarely fall below
</p>
<p>5%, and may rise to about 20% for high resolution data. A worse result than this would indicate poor
</p>
<p>data quality, assuming that a sufficient sweep of intensity data has been recorded. In such cases it may
</p>
<p>be wise either to exercise caution and discard some of the data by using a resolution cut-off or to
</p>
<p>remeasure the data with a fresh crystal.
</p>
<p>Rsym
If separate intensity measurements of I&eth;hkl&THORN; and its symmetry-related equivalents have been made,
each measurement may be expressed as Ii&eth;hkl&THORN;, (i &frac14; 1, 2, . . ., n) where n is greater than 2 but has
possibly different values for different reflections, with a maximum value equal to the total number of
</p>
<p>symmetry equivalents for the space group in question. The best estimate of a given I&eth;hkl&THORN; is
�I&eth;hkl&THORN; &frac14;
</p>
<p>P
Ii&eth;hkl&THORN;=n. The discrepancy between the ith observation of I&eth;hkl&THORN; and the best estimate
</p>
<p>is Ii&eth;hkl&THORN; � �I&eth;hkl&THORN;; and the mean discrepancy of all the observations of this intensity isP
n &eth;Ii&eth;hkl&THORN; � �I&eth;hkl&THORN;&THORN;=n. In a given group of intensity measurements, let there be N for which at
</p>
<p>least two symmetry equivalents have been measured. To form Rsym the mean discrepancy for each of
</p>
<p>these N reflections is compared to their mean intensity:
</p>
<p>Rsym &frac14;
X
</p>
<p>N
</p>
<p>X
</p>
<p>n
</p>
<p>jIi&eth;hkl&THORN; � �I&eth;hkl&THORN;j=n
" # X
</p>
<p>N
</p>
<p>�I&eth;hkl&THORN;
,
</p>
<p>Weights derived from the data collection statistics may be employed in order to make a better
</p>
<p>estimate of �I&eth;hkl&THORN; as:
</p>
<p>�I&eth;hkl&THORN; &frac14;
X
</p>
<p>i&frac14;1;n
oiIi&eth;hkl&THORN;
</p>
<p>X
</p>
<p>i&frac14;1;n
wi
</p>
<p>,
</p>
<p>Rsym can be thought of as the mean error of an intensity measurement, compared to the mean
</p>
<p>intensity, averaged over the chosen group of reflections.
</p>
<p>We draw attention to the facts (1) that Rsym as defined here is calculated on the basis of intensity
</p>
<p>measurements, (2) that some programs, such as SHELX-97, use the notation Rint instead of Rsym and
</p>
<p>employ jFj2(hkl) instead of I(hkl), and (3) that since I(hkl) is related to jFj2(hkl), the fractional error in
I(hkl) is approximately double the fractional error in jFj(hkl) so that if Rsym is 0.04, the mean
fractional error in jFj would be about 0.02. This sort of value would usually be considered to be
&ldquo;highly acceptable.&rdquo;
</p>
<p>304 7 Fourier Techniques in X-Ray Structure Determination</p>
<p/>
</div>
<div class="page"><p/>
<p>Rmerge
The essential difference between Rmerge and Rsym lies in the data subsets from which they are usually
</p>
<p>calculated: Rmerge is calculated, not from symmetry-equivalent intensities, but from repeats of the
</p>
<p>same I(hkl) measured during the course of a data collection session. Such repeat measurements may
</p>
<p>be the result of employing overlapping angular ranges when the crystal is rotated. The program
</p>
<p>SHELX-97 uses the notation Rsigma instead of Rmerge.
</p>
<p>It is possible to merge the repeat measurements to produce a new basic data set prior to merging
</p>
<p>the symmetry equivalents. A value of Rsym could then be derived from this new basic set and the
</p>
<p>symmetry equivalents combined to produce the final working data set. A possible scheme is set out as
</p>
<p>follows.
</p>
<p>Intensity Data Collection and Processing Scheme
</p>
<p>Data Set 1
</p>
<p>Data are collected to the maximum possible y for which intensities are strong enough to measure
</p>
<p>satisfactorily. This is the basic data set of non-merged I(hkl) data, and may include repeats:
</p>
<p>&bull; For some unique I(hkl) data whose values should be equal within experimental error;
</p>
<p>&bull; For a number of space group symmetry-equivalent I(hkl) data whose values should also be equal
</p>
<p>within experimental error.
</p>
<p>Data Set 2
</p>
<p>Data processing can be performed first by merging the unique repeats in data set 1. Rmerge is thus
</p>
<p>produced and examined for quality; if acceptable, it defines data set 2 which is then further processed.
</p>
<p>Data Set 3
</p>
<p>Data set 2 is next used to produce the final data set 3 in which the space group symmetry-equivalent I
</p>
<p>(hkl) data have now been combined, and Rsym is produced as a result of this process. If Rsym has an
</p>
<p>acceptable value, data set 3 becomes the current standard data set for the analysis.
</p>
<p>Data Collection from Macromolecules
If Rmerge from data set 2 is not acceptable, poor crystal quality may be indicated. The data collection
</p>
<p>should be repeated with a new crystal until a good specimen is found.
</p>
<p>If Rsym from data set 3 is not acceptable, it could indicate an incorrect assignment of the space
</p>
<p>group. Then, other possible space groups should be investigated by further analysis of the data and
</p>
<p>recalculation of Rsym until a satisfactory result is obtained. If it is still unsuccessful, it may be
</p>
<p>necessary to recollect data from fresh crystals.
</p>
<p>Anomalous Scattering
</p>
<p>Small Molecules: Calculation of Rsym for Non-centrosymmetric Structures
</p>
<p>In small-molecule analysis for structures in non-centrosymmetric space groups, it is possible to
</p>
<p>determine the absolute configuration of the structure by virtue of anomalous differences between Friedel
</p>
<p>pairs I&eth;hkl&THORN; and I&eth;hkl&THORN;. The success of this operation depends on the types of atom present and the
radiation used for data collection. Thus, it is usual in non-centrosymmetric small-molecule structure
</p>
<p>analysis not to merge Friedel opposites or any opposites generated by the space group symmetry
</p>
<p>(Bijvoet differences, q.v.) unless it can be shown from calculation of the Flack parameter (Sect. 7.6.1)
</p>
<p>that the absolute configuration cannot be determinedwith certainty. If andwhen this stage is reached, the
</p>
<p>7.5 Heavy-Atom Method and Partial Fourier Synthesis 305</p>
<p/>
</div>
<div class="page"><p/>
<p>refinement should be repeated and completed with a data set for which the Friedel opposites and Bijvoet
</p>
<p>pairs have been merged.
</p>
<p>Ranom in Macromolecular Analysis
</p>
<p>With macromolecular structures that are also non-centrosymmetric, the application of anomalous
</p>
<p>scattering for phasing in structure analysis also requires similar anomalous measurable differences to
</p>
<p>exist. Again, this would preclude the inclusion of such reflections from the calculation of Rsym. In
</p>
<p>order to assess the possible strength of anomalous scattering, an R index is initially calculated as
</p>
<p>Ranom &frac14; D=M where D is now the mean Friedel or Bijvoet difference and M is the mean of all
amplitudes involved.
</p>
<p>We know that Bijvoet amplitude differences are equivalent by space group symmetry to Friedel
</p>
<p>opposite differences. In a given data collection experiment, one or both quantities may be measured.
</p>
<p>In practice, it is possible to derive useful phase indications from anomalous differences for a
</p>
<p>macromolecular crystal which includes one or more strong anomalous scatters in relation to the
</p>
<p>radiation employed provided that 2Ranom � Rmerge. The factor 2 derives from the fact that Rmerge is
based on intensity whereas Ranom is based on amplitude.
</p>
<p>Rderiv
This R factor is used to assess the phasing power of a heavy-atom derivative in macromolecular
</p>
<p>structure analysis by the isomorphous replacement method:
</p>
<p>Rderiv &frac14;
X
</p>
<p>jjFPHj � jFPjj
X
</p>
<p>jFPj
.
</p>
<p>The summations here are over the N-independent reflections for which both jFPj and jFPHj have
been measured. It is assumed that jFPj and jFPHj have both been scaled, for example, by Wilson
statistics (Sect. 4.2.1). For a moderately sized protein with one fully occupied heavy atom site, the
</p>
<p>value of Rderiv calculated for the data between 10 and 3.5 Å resolution would be expected to have a
</p>
<p>value between 0.1 and 0.2. Larger changes than this may indicate several sites of heavy atom
</p>
<p>substitution or may be due to lack of isomorphism. If the latter is the case, the heavy atom derivative
</p>
<p>will not provide any useful phasing and would be discarded; Rderiv is also known as Rdiff and Riso.
</p>
<p>Reliability Factor in Structure Refinement
The conventional R factor that is used in structure refinement assessment reflects the difference
</p>
<p>between the scaled-observed and the calculated structure-factor amplitudes, and provides a measure
</p>
<p>of the quality of the trial structure. Large differences correspond to poor reliability, and vice versa.
</p>
<p>The R is defined as
</p>
<p>R &frac14;
X
</p>
<p>hkl
</p>
<p>jKFo � jFcjj
X
</p>
<p>hkl
</p>
<p>KFo
</p>
<p>,
(7.30)
</p>
<p>R is sometimes also known as R1, where R2 is used for the corresponding weighted value. For a well-
</p>
<p>refined structure model, the value of R approaches a small value, about 1% at the very best,
</p>
<p>corresponding to the residual errors in both the experimental data and the model. In the early stages
</p>
<p>of the analysis, however, Rmay lie between 0.4 and 0.5. It expresses the first criterion of correctness,
</p>
<p>namely, good agreement between Fo and jFcj. It should be noted that trial structures with an R factor
of more than 50% have been known to be capable of refinement; R is only a rough guide at that stage
</p>
<p>of the analysis. A better basis for judgment is a comparison of the pattern of Fo and jFcj, which
requires care and experience.
</p>
<p>306 7 Fourier Techniques in X-Ray Structure Determination</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec11_4">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec11_4</a></div>
</div>
<div class="page"><p/>
<p>Refinement R Indices and Weights with SHELX-97
Structure refinement with the SHELX-97 system is carried out against jFj2 instead of jFj and the
values of the R indices tend to be more than twice as high. For comparison with older refinements,
</p>
<p>SHELX-97 produces a conventional index R1 based on Fo values &gt;4s(Fo) and a weighted R factor
</p>
<p>wR2 based on jFj2 where R1 is governed by (7.30) and the weighted R factor by
</p>
<p>wR2 &frac14;
X
</p>
<p>&frac12;w&eth;Fo2 � jFcj2&THORN;2�
X
</p>
<p>&frac12;w&eth;Fo2&THORN;2�
.n o1=2
</p>
<p>where Fo is a scaled value, as in (7.30), and w &frac14; 1=&frac12;s2&eth;F2o&THORN; &thorn; &eth;aP&THORN;
2 &thorn; bP� where P &frac14; &frac12;2jFcj2&thorn;
</p>
<p>max&eth;Fo2; 0&THORN;�=3. The use of this combination of F2o and jFcj
2
was shown [9] to reduce statistical bias.
</p>
<p>As a rough guide, for high resolution small-molecule structures R1 may be as low as 2% but more
</p>
<p>usually 4&ndash;6%. For low resolution protein data R1would rarely fall below 5%, and may rise to 15&ndash;20%
</p>
<p>at high resolution. A result less than this would indicate poor data quality and require careful
</p>
<p>investigation.
</p>
<p>Rfree and Protein Refinement
The Rfree index [10] is currently used in macromolecular structure analysis, and this is a requirement in
</p>
<p>most journals as a condition of publication. Use of this technique requires that the measured Fo(hkl)
</p>
<p>data available for refinement of the protein structure is partitioned into two separate sets. The smaller
</p>
<p>portion of the data is usually selected at random and comprises 5&ndash;10% of the whole data set. The small
</p>
<p>subset is used in the calculation of Rfree and the larger subset is known as the &ldquo;working&rdquo; data set. The
</p>
<p>Rfree subset of data is used only for calculation of Rfree. These reflections are subsequently excluded
</p>
<p>from the summation in the equations forR (R1) given above, which is calculated only from the working
</p>
<p>data set. Consequently there is a permanent loss of this selected data from the refinement procedure.
</p>
<p>This test subset is never allowed to be included in the refinement process from start to finish. The
</p>
<p>purpose of excluding this subset of data is to enable the calculation of
</p>
<p>Rfree &frac14;
X
</p>
<p>hkl
</p>
<p>jKFo � jFcjj
X
</p>
<p>hkl
</p>
<p>KFo
</p>
<p>,
</p>
<p>where the summations are carried out over the randomly selected 5&ndash;10% subset of test data, and K is
</p>
<p>the scale factor for the data set.
</p>
<p>Since these randomly selected data are not available at any stage in the refinement procedure, any
</p>
<p>improvements such as a decrease in Rfree during the course of the refinement will reflect genuine
</p>
<p>improvements in the model, rather than the fitting of an incorrect model to the Fo data which could,
</p>
<p>within limits, improve the conventional R factor. After each cycle of refinement calculations, it is
</p>
<p>essential to review the refined model for errors by studying in detail the Fo � jFcj difference map. If
this map has any significant positive features, they are a clear indication that a part of the map needs to
</p>
<p>be reinterpreted. In practice, during the course of structure refinement Rfree lags behind R because it is
</p>
<p>calculated from reflections that the refinement procedure does not &ldquo;know about.&rdquo; A lag of about 3% is
</p>
<p>usually considered acceptable. Further guidelines can be found in the documentation for the Phenix
</p>
<p>program [11].
</p>
<p>Rfree as a Simple Quantitative Guide to Correct Coordinate in a Refined Protein Structure
A method [12] that was originally intended to aid evaluation of the accuracy of well-resolved atomic
</p>
<p>structures in small-molecule crystallography has been modified [13, 14] for application to refined
</p>
<p>protein structures. As with the original method, the modification is based largely on the R factor.
</p>
<p>7.5 Heavy-Atom Method and Partial Fourier Synthesis 307</p>
<p/>
</div>
<div class="page"><p/>
<p>It enables the positional error for an atom having an average isotropic temperature factor to be
</p>
<p>estimated. From (7.30), we see that R will be small if the Fo values have been measured accurately
</p>
<p>and the jFcj values are derived from a well behaved and highly reliable structure. A modified
approximation [14] provides a rapid method for estimating the precision s&eth;rx;y;z; �B&THORN; of the positions
of atoms in a refined protein crystal structure:
</p>
<p>s&eth;rx;y;z; �B&THORN; &frac14; 2:2
ffiffiffiffiffiffiffiffiffiffiffiffi
Natoms
</p>
<p>p
V1=3a n
</p>
<p>�5=6
o Rfree
</p>
<p>where Natoms is the number of fully occupied protein sites in the structure, Va is the volume of the
</p>
<p>crystal asymmetric unit in Å3, no is the number of I(hkl) measurements in the refinement data set, and
</p>
<p>Rfree is its final value after refinement is complete. This expression applies only to protein structures,
</p>
<p>because the average atomic weight of atoms present in the structure is assumed to be about 14.1. The
</p>
<p>measure s&eth;rx;y;z; �B&THORN; is actually the standard error of position for an atom that has an average B value; it
does not include any information about s&eth;B&THORN;, and the atom positions involved in N are those that are
fully occupied. Some of the atoms in a protein structure may be either partially occupied or have
</p>
<p>larger than average B values. According to this analysis, s&eth;rx;y;z; �B&THORN; has the following very approxi-
mate expectation values as a function of resolution dmin and Rfree:
</p>
<p>dmin (Å) Rfree s&eth;rx;y;z; �B&THORN; (Å)
1.0 0.25 0.06
</p>
<p>2.0 0.25 0.30
</p>
<p>The values of Rfree at this level would be considered worryingly high.
</p>
<p>dmin (Å) Rfree s&eth;rx;y;z; �B&THORN; (Å)
1.0 0.20 0.045
</p>
<p>2.0 0.20 0.25
</p>
<p>1.0 0.15 0.035
</p>
<p>2.0 0.15 0.18
</p>
<p>The values of Rfree within these ranges would generally be considered to be acceptable.
</p>
<p>7.5.2 Pseudosymmetry in Electron Density Maps
</p>
<p>The electron density map calculated with phases derived from the heavy-atom positions may not
</p>
<p>exhibit the true space-group symmetry. Suppose space group P21, for example, has one heavy atom
</p>
<p>per asymmetric unit. The origin is defined with respect to the x and z axes by the 21 axis along [0, y, 0],
</p>
<p>but the y coordinate of the origin is determined with respect to an arbitrarily assigned y coordinate for
</p>
<p>one of the atoms. Consider the heavy atoms at x, y, z and symmetry-related at �x,
1
</p>
<p>2
&thorn; y, �z. This
</p>
<p>arrangement of heavy atoms has the symmetry of P21/m, with the m planes cutting the y axis at
</p>
<p>whatever y coordinate is chosen for the heavy atom, say yH, and at
1
</p>
<p>2
&thorn; yH. If yH is selected as
</p>
<p>1
</p>
<p>4
, a
</p>
<p>center of symmetry is at the origin, and the calculated phases will be 0 or p. This situation is
</p>
<p>illustrated in Fig. 7.23, which indicates that an unscrambling of the images must be carried out.
</p>
<p>If the heavy atom is given any general value for yH, B
0(hkl) will not be zero and the phase angles will
</p>
<p>not be 0 or p, but the pseudosymmetry will still exist.
</p>
<p>308 7 Fourier Techniques in X-Ray Structure Determination</p>
<p/>
</div>
<div class="page"><p/>
<p>7.5.3 Successive Fourier Refinement
</p>
<p>A single application of the Fourier method described above does not usually produce a complete set
</p>
<p>of atomic coordinates. It should lead to the inclusion of more atoms into subsequent structure factor
</p>
<p>calculations and so to a better electron density map, and so on. This iterative process of Fourier
</p>
<p>refinement should, after several cycles, result in the identification of all non-hydrogen atoms in the
</p>
<p>structure to within about 0.1 Å of their true positions. Further improvement of the structure would
</p>
<p>normally be carried out by the method of least squares, which is described in Sect. 8.4.
</p>
<p>7.5.4 Difference-Fourier Synthesis
</p>
<p>Some errors present in the trial structure may not be revealed by Fourier synthesis. In particular, the
</p>
<p>following situations are important.
</p>
<p>1. Atoms in completely wrong positions tend to be returned by the Fourier process with similar
</p>
<p>fractional coordinates, but sometimes with a comparatively low electron density.
</p>
<p>2. Correctly placed atoms, may have been assigned either the wrong atomic number, for example, C
</p>
<p>for N, or an incorrectly estimated temperature factor.
</p>
<p>3. Small corrections to the fractional coordinates may be difficult to assess from the Fourier map.
</p>
<p>In these circumstances, a difference-Fourier synthesis is valuable. We shall symbolize the Fourier
</p>
<p>series with Fo coefficients as ro(xyz) and the corresponding synthesis with jFcj instead as rc(xyz); the
difference-Fourier synthesis Dr(xyz) may be obtained in a single-stage calculation from the equation,
</p>
<p>using the difference of observed and calculated structure amplitudes:
</p>
<p>Dr&eth;xyz&THORN; &frac14; 2
Vc
</p>
<p>X
</p>
<p>h
</p>
<p>X
</p>
<p>k
</p>
<p>X
</p>
<p>l
</p>
<p>&eth;Fo � jFcj&THORN; cos&frac12;2p&eth;hx&thorn; ky&thorn; lz&THORN; � fc� (7.31)
</p>
<p>Fig. 7.23 Introduction of pseudosymmetry into space group P21 by single heavy-atom phasing. H is the heavy atom
and C is a center of symmetry introduced between H and its P21 equivalent H
</p>
<p>0. The space group (for the heavy atoms
alone) thus appears as P21/m with mirror (and �1) pseudosymmetry. The electron density map, phased on the H and H
</p>
<p>0
</p>
<p>species, will contain two mirror-related images in the asymmetric unit, with a certain degree of confusion between them
</p>
<p>7.5 Heavy-Atom Method and Partial Fourier Synthesis 309</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec29_8">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec29_8</a></div>
</div>
<div class="page"><p/>
<p>Since the phases are substantially correct at this stage, it is in effect, a subtraction, point by point, of
</p>
<p>the &ldquo;calculated,&rdquo; or trial, Fourier synthesis from that of the &ldquo;observed,&rdquo; or experimentally based,
</p>
<p>synthesis. The difference synthesis has the following useful properties.
</p>
<p>1. Incorrectly placed atoms correspond to regions of high electron density in rc(xyz) and low density
</p>
<p>in ro(xyz); Dr(xyz) is therefore negative in these regions.
</p>
<p>2. A correctly placed atom with either too small an atomic number or too high a temperature factor
</p>
<p>shows up as a small positive area in Dr. The converse situations produce negative peaks in Dr.
</p>
<p>3. An atom requiring a small positional correction tends to lie in a negative area at the side of a small
</p>
<p>positive peak. The correction is applied by moving the atom into the positive area.
</p>
<p>4. Very light atoms, such as hydrogen, may be revealed by a Dr synthesis when the phases are
</p>
<p>essentially correct, after least-squares refinement has been carried out.
</p>
<p>5. As one final test of the validity of a refined structure, theDr synthesis should be effectively featureless
</p>
<p>within two to three times the standard deviation of the electron density, Sect. 8.7.
</p>
<p>7.5.5 Limitations of the Heavy-Atom Method
</p>
<p>The Patterson and heavy-atom techniques are effective over a wide range of structures. It is
</p>
<p>sometimes necessary to introduce heavy atoms artificially into structures. This process may not be
</p>
<p>desirable because a possible structural interference may arise, and there will be a loss in the accuracy
</p>
<p>of the light-atom positions. An introduction to direct methods, capable of solving the phase problem
</p>
<p>for such structures, is given in the next chapter.
</p>
<p>7.5.6 Patterson Selection
</p>
<p>It is possible that the Patterson function for the crystal of a heavy-atom compound may not reveal
</p>
<p>the heavy-atom vector unambiguously. Figure 7.24a is the Patterson projection on to the x, z plane
</p>
<p>for euphenyl iodoacetate, C32H53O2I, and Fig. 7.24b is the Harker section for the same material.
</p>
<p>There are two high peaks, A and B, in the asymmetric unit where only one was expected. For the
</p>
<p>Harker section, the coefficients were sharpened and further modified by the multiplicative function
</p>
<p>exp&eth;�9sin3y&THORN; so as to smooth out any undesirable fluctuations caused by the sharpening which
enhances the high-order reflections relative to those of low order.
</p>
<p>In the selection process, the Fo
2 data, averaged in zones of 2 sin y, are plotted as a function of 2 sin y
</p>
<p>in Fig. 7.24c(i). Between the values for 2 sin y of 0.5 and 0.7, the average values of Fo
2 are enhanced,
</p>
<p>owing to the multiplicity of similar distances in the structure, unconnected with the heavy atom,
</p>
<p>Fig. 1.7, compared with the corresponding smoothed curve, Fig. 7.24c(ii). The many equal, or nearly
</p>
<p>equal, vectors between atoms in the molecule are superimposed in the Patterson function and lead to
</p>
<p>additional large peaks.
</p>
<p>The Fo
2 data were &ldquo;selected&rdquo; by excluding from the next Patterson synthesis all those data lying
</p>
<p>within the range 0:5&lt;2 sin y&lt;0:7, and the resulting sharpened Harker section is shown in Fig. 7.25;
clearly, the heavy-atom vector is at B. The selection process has effectively removed the &ldquo;structure&rdquo;
</p>
<p>that was giving rise to the additional peak A, so that the heavy-atom vector was then sought among the
</p>
<p>vectors from a more random array of atoms [15].
</p>
<p>Figure 7.24c(iii) is the average Fo
2 curve for a random arrangement corresponding to C32O2I, with
</p>
<p>the same unit-cell geometry as that of euphenyl iodoacetate, and with no two atoms closer than 1.6 Å.
</p>
<p>The R factor for this hypothetical structure was 0.58; the value for a completely random
</p>
<p>310 7 Fourier Techniques in X-Ray Structure Determination</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7# Sec41_8">http://dx.doi.org/10.1007/978-1-4614-3954-7# Sec41_8</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Fig7_1">http://dx.doi.org/10.1007/978-1-4614-3954-7#Fig7_1</a></div>
</div>
<div class="page"><p/>
<p>Fig. 7.24 Patterson studies on the heavy-atom compound C32H53O2I (P21; Z &frac14; 2). (a) Projection P(uw).
(b) Sharpened Patterson&ndash;Harker section P&eth;u 1
</p>
<p>2
w&THORN;. (c) Plots of average Fo2 against 2 sin y: (i) experimental data,
</p>
<p>(ii) &ldquo;smooth&rdquo; curve through the experimental points, (iii) hypothetical (random) structure C32O2I, with the same
</p>
<p>crystal geometry as C32H53O2I
</p>
<p>Fig. 7.25 Sharpened Patterson&ndash;Harker section P&eth;u 1
2
w&THORN;, with the Fo2 data selected as described in the text. The I&ndash;I
</p>
<p>vector is clearly at B, whereas in Fig. 7.24a, b, there were two peaks of equal height in the asymmetric unit</p>
<p/>
</div>
<div class="page"><p/>
<p>non-centrosymmetric structure is 0.586 [16]. The curve evolves in a manner similar to an f 2 curve
</p>
<p>and, as would be expected, shows no structural effects of the nature of Fig. 7.24c(i).
</p>
<p>7.5.7 Isomorphous Replacement
</p>
<p>A common feature of biologically important substances is their high molecular weight. Proteins and
</p>
<p>enzymes, for example, are polymers built up from various amino acid residues and form very large
</p>
<p>atomic assemblies with molecular weights greater than about 5000. The study of the conformations of
</p>
<p>these giant molecules is necessary for an understanding of their biological functions, and the principal
</p>
<p>method of obtaining structural detail is by X-ray analysis.
</p>
<p>Because of their high molecular weight, protein structures do not yield to analysis by the
</p>
<p>straightforward heavy-atom method. The value of r, from (7.24), is typically 0.03 for a protein
</p>
<p>molecule of molecular weight 5000 containing one incorporated mercury atom. This value of r is too
</p>
<p>small to be useful. Another difficulty is that most proteins and enzymes contain neither very heavy
</p>
<p>atoms nor easily replaceable groups to facilitate the introduction of heavy atoms. In spite of these
</p>
<p>difficulties, if a heavy-atom derivative of a large molecule can be prepared, it may be possible to
</p>
<p>induce it to crystallize in a similar size of unit cell and with the same space group as the native
</p>
<p>compound. Such pairs of crystals are said to be isomorphous.
</p>
<p>The structure factor of the heavy-atom derivative FPH may be handled, as before, in the manner of
</p>
<p>a vector as
</p>
<p>FPH &frac14; FP &thorn; FH (7.32)
</p>
<p>where FP and FH are the structure factors for the parent protein and the heavy atoms alone,
</p>
<p>respectively, for the same reflection. This relationship is shown in Fig. 7.26.
</p>
<p>Fig. 7.26 Graphical interpretation of the isomorphous replacement equation. In practice, the phases fP and fPH are
unknown initially. FH may be known with a fair degree of accuracy if the heavy-atom position in each isomorphous
</p>
<p>derivative is known. This enables a solution, as illustrated in Fig. 7.27, to be obtained
</p>
<p>312 7 Fourier Techniques in X-Ray Structure Determination</p>
<p/>
</div>
<div class="page"><p/>
<p>Assuming that the positions of the NH heavy atoms in the unit cell can be determined, their
</p>
<p>contribution can be calculated:
</p>
<p>FH &frac14;
XNH
</p>
<p>j&frac14;1
g0j exp&frac12;i2p&eth;hxj &thorn; kyj &thorn; lzj&THORN;� (7.33)
</p>
<p>where g0j &frac14; f 0j exp&eth;�Bjsin2y=l2&THORN;, and f 0j &frac14; Kjfj: Kj is a site occupation factor, less than or equal to
unity, provided that jFPj and jFPHj are on an absolute scale, and depending on the degree of
substitution at the heavy-atom site j; some heavy-atom binding sites of the protein molecules in the
</p>
<p>crystal may not be fully substituted.
</p>
<p>In order to obtain an idea of the effect of a heavy atom on the intensities of X-ray reflections from a
</p>
<p>protein, we shall carry out a simple calculation for a crystal containing one protein molecule per unit
</p>
<p>cell in space group P1. Assuming that it has a molecular weight of about 13000, about 1000 non-
</p>
<p>hydrogen atoms would also be present in the molecule; we shall assume that the non-hydrogen atoms
</p>
<p>are all carbon (ZC &frac14; 6). Accepting Wilson&rsquo;s approximation (4.35), and replacing gj by fC, we have
</p>
<p>jFPj2 �
X1000
</p>
<p>j&frac14;1
f 2C (7.34)
</p>
<p>At sin y &frac14; 0, jFPj2 is 36000. If the derivative contains one mercury atom (ZHg &frac14; 80), then
</p>
<p>jFPHj2 �
X1000
</p>
<p>j&frac14;1
f 2C &thorn;
</p>
<p>XNH
</p>
<p>j&frac14;1
f 2j (7.35)
</p>
<p>which has the value 42400 at sin y &frac14; 0&eth; fj &frac14; 80&THORN;. Hence, the maximum change in intensity is about
18%, which is a surprisingly high value.
</p>
<p>Experimentally, two sets of data jFP(hkl)j and jFPH(hkl)j are measured and, because of the
comparative nature of the phase-determining procedure with isomorphous compounds, they must
</p>
<p>be placed on the same relative scale, which can be achieved by Wilson&rsquo;s method. Rewriting (7.32),
</p>
<p>we have
</p>
<p>jFPj exp&eth;ifP&THORN; &frac14; jFPHj exp&eth;ifPH&THORN; � FH (7.36)
</p>
<p>Assuming that FH can be determined, this equation involves two unknown quantities, fP and fPH,
</p>
<p>and cannot yield a unique solution. However, Fig. 7.27 shows that ideally only two solutions for fP are
</p>
<p>real, corresponding to the vectors OP1 and OP2, one of which is the true FP vector. A second
</p>
<p>isomorphous derivative with a different set of heavy-atom positions will also have two solutions for
</p>
<p>FP. The derivatives are denoted 1 and 2, and the solutions for FP are OP11 and OP12 (derivative 1) and
</p>
<p>OP21 adOP22 (derivative 2), as shown in Fig. 7.28. Two of the solutions should agree with eitherOP1 or
</p>
<p>OP2 within experimental error, thus resolving the ambiguity, Fig. 7.28. With a more extensive series of
</p>
<p>isomorphous derivatives, it is possible to obtain phases capable of yielding interpretable electron density
</p>
<p>maps. Many protein structures have been investigated successfully by this technique. Further details are
</p>
<p>given in Sect. 7.4.8.
</p>
<p>Centrosymmetric Projections
Proteins always crystallize in non-centrosymmetric space groups because the amino acid residues in
</p>
<p>the polypeptide have left-handed configurations about the a-carbon atoms. Amino acid residues with
</p>
<p>7.5 Heavy-Atom Method and Partial Fourier Synthesis 313</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Equ35_4">http://dx.doi.org/10.1007/978-1-4614-3954-7#Equ35_4</a></div>
</div>
<div class="page"><p/>
<p>Fig. 7.27 Single
isomorphous replacement
</p>
<p>(SIR) phase&ndash;amplitude
</p>
<p>diagram in the SIR method.
</p>
<p>OH(�FH) is the known
reversed heavy-atom
</p>
<p>vector. The triangles OHP1
and OHP2 both satisfy
</p>
<p>(7.32), giving a twofold
</p>
<p>ambiguity with vectors
</p>
<p>either OP1 or OP2 as the
</p>
<p>solution for FP
</p>
<p>Fig. 7.28 Multiple
isomorphous replacement
</p>
<p>(MIR) resolution of the
</p>
<p>phase ambiguity with a
</p>
<p>second isomorphous
</p>
<p>heavy-atom derivative
</p>
<p>PH2. The determined
</p>
<p>direction of FP is near
</p>
<p>vectors OP11 and OP21. In
</p>
<p>practice, P11 and P21 rarely
</p>
<p>coincide, due to
</p>
<p>inaccuracies in the heavy-
</p>
<p>atom parameters and lack
</p>
<p>of complete isomorphism
</p>
<p>314 7 Fourier Techniques in X-Ray Structure Determination</p>
<p/>
</div>
<div class="page"><p/>
<p>right-handed configurations are very rare in nature. Although non-centrosymmetric structures usually
</p>
<p>present more difficulties than centrosymmetric structures, there is a compensation in the relative ease
</p>
<p>of determination of the space group; ambiguities such as P21 and P21/m do not exist for the protein
</p>
<p>crystallographer. Most non-centrosymmetric space groups have at least one centric zone. In such a
</p>
<p>case, (7.32) becomes
</p>
<p>sPHFPH &frac14; sPFP &thorn; sHFH (7.37)
</p>
<p>where s, the sign of the structure factor is �1.
Unless both FPH and FP are very small compared with FH, it is unlikely that sPHwill differ from sP.
</p>
<p>Generally FP and FPH are pointing in the same direction. Accepting this statement, we may substitute,
</p>
<p>sP for sPH in (7.37):
</p>
<p>sP&eth;FPH � FP&THORN; &frac14; sHFH (7.38)
</p>
<p>or
</p>
<p>sP &frac14; sHFH=DF (7.39)
where DF &frac14; FPH � FP. Since we are interested only in the signs, (7.39) may be rewritten as
</p>
<p>sP &frac14; sHsD (7.40)
</p>
<p>where sD is +1 if FPH &gt; FP and �1 if FPH &lt; FP. In this way, signs can often be determined for
centric reflections in a protein crystal with only a single isomorphous derivative, and we shall
</p>
<p>illustrate the method by the following example.
</p>
<p>Sign Determination for Centric Reflections in Protein Structures
We shall consider data for both the enzyme ribonuclease and a heavy-atom derivative prepared by
</p>
<p>soaking pre-grown crystals of the enzyme in K2[PtCl6] solution.
</p>
<p>Crystal Data for Ribonuclease
</p>
<p>System: monoclinic
</p>
<p>Unit-cell dimensions: a &frac14; 30.31 Å, b &frac14; 38.26 Å, c &frac14; 52.91 Å, b &frac14; 105.9�
Mr: 13500 (ribonuclease)
</p>
<p>ZP: two molecules of ribonuclease plus an unknown number of water molecules
</p>
<p>ZPH: as for ZP + NH [PtCl6]
�2 groups per unit cell (NH is the number of heavy atoms groups)
</p>
<p>Absent spectra: 0k0: k &frac14; 2n + 1
Space group: P21. The h0l zone is centrosymmetric
</p>
<p>Given the heavy-atom positions, Table 7.5 shows how the signs for some h0l reflections have been
</p>
<p>determined.Notice that experimental errors in FP and FPH, togetherwith errors in the calculated FH arising
</p>
<p>from inaccuracies in the heavy-atom model, are reflected in the inequality of DF and FH. The validity of
</p>
<p>(7.40) is upheld by these data.
</p>
<p>Location of Heavy-Atom Positions in Proteins
In a centrosymmetric zone, it follows from (7.39), since sP and sH are �1, that
</p>
<p>FH &frac14; jDFj (7.41)
</p>
<p>where jDFj &frac14; jFPHj � jFPj. A Patterson function calculated with jFHj2 as coefficients would give the
vector set of the substituted heavy atoms in the protein molecule. Since jFHj cannot be observed
</p>
<p>7.5 Heavy-Atom Method and Partial Fourier Synthesis 315</p>
<p/>
</div>
<div class="page"><p/>
<p>directly, one calculates a difference Patterson map with (DF)2 as coefficients. If the experimental
</p>
<p>errors in jFPj and jFPHj are not significant, and not too many sign &ldquo;cross-overs&rdquo; with sP and sPH occur,
then the (DF)2 Patterson projection would be expected to reveal the heavy-atom vectors. In the case of
</p>
<p>general non-centrosymmetric reflections, we note in Fig. 7.29 that sinceOQ &frac14; jFPHj,OP &frac14; jFPj, and
OR &frac14; jFPj cos(fPH � fP), we have
</p>
<p>RQ &frac14; jFHj cos&eth;fH � fPH&THORN; and
OQ &frac14; OR&thorn; RQ
</p>
<p>(7.42)
</p>
<p>Hence,
</p>
<p>jFPHj � jFPj cos&eth;fPH � fP&THORN; &frac14; jFHj cos&eth;fH � fPH&THORN; (7.43)
</p>
<p>Table 7.5 h0l Data for ribonuclease
</p>
<p>Observed data Calculated data
Deduced sign
</p>
<p>hkl FP FPH jDFj sD jFHj sH sp &frac14; sHsD
003 437 326 111 �1 50 +1 �1
006 59 48 11 �1 27 �1 +1
007 182 109 73 �1 90 �1 +1
10; 17 144 196 52 +1 31 �1 �1
10,13 146 82 64 �1 52 +1 �1
10�9 97 165 68 +1 55 �1 �1
106 183 242 59 +1 45 +1 +1
</p>
<p>30�4 746 861 115 +1 72 +1 +1
</p>
<p>405 103 57 46 �1 56 +1 �1
</p>
<p>Fig. 7.29 Location of heavy-atom positions in proteins
</p>
<p>316 7 Fourier Techniques in X-Ray Structure Determination</p>
<p/>
</div>
<div class="page"><p/>
<p>If fPH � fP is small, it follows that cos(fPH � fP) � 1.0 and
</p>
<p>jFPHj � jFPj &frac14; DF � jFHj cos&eth;fH � fPH&THORN; (7.44)
</p>
<p>In practice, since the angle fH � fPH is undeterminable at this stage, one can calculate a Patterson
function with (DF)2 coefficients as for centrosymmetric reflections, but as an added precaution to
</p>
<p>ensure that (fPH � fP) is small, use only those terms for which both jFPj and jFPHj are large.
Although the non-centrosymmetric (DF)2 synthesis is not a true Patterson function, it has been used
</p>
<p>successfully to determine the heavy-atom distribution in proteins.
</p>
<p>The most useful derivatives contain a small number of highly substituted sites. Unlike the structure
</p>
<p>analysis of smaller molecules, it is not known initially how many heavy-atom sites have been
</p>
<p>incorporated into the molecule.
</p>
<p>As an example, we shall consider the (DF)2 Patterson map for the Pt derivative of ribonuclease,
</p>
<p>space group P21. The vectors between symmetry-related atoms occur on the Harker section &eth;u;
1
</p>
<p>2
;w&THORN;.
</p>
<p>Eight peaks occur on the Harker section and four at v &frac14; 0, Fig. 7.30a, b. This result suggests that there
is more than one heavy-atom site per protein molecule. The most obvious choice is two, since four
</p>
<p>heavy atoms per unit cell would give rise to 12 non-origin peaks. If the two sites are labeled 1 and 2,
</p>
<p>their Harker peaks will be of the form � &eth;2x1;
1
</p>
<p>2
; 2z1&THORN; and � &eth;2x2;
</p>
<p>1
</p>
<p>2
; 2z2&THORN;.
</p>
<p>Interpretation of the Patterson function is best undertaken in terms of the Harker section, assuming
</p>
<p>that the peaks represent nonoverlapping vectors and ignoring the possibility that some peaks could be
</p>
<p>Fig. 7.30 (DF)2 Patterson
sections for the Pt
</p>
<p>derivative of ribonuclease:
</p>
<p>(a) P&eth;u1
2
w&THORN;. (b) P(u0w)
</p>
<p>7.5 Heavy-Atom Method and Partial Fourier Synthesis 317</p>
<p/>
</div>
<div class="page"><p/>
<p>non-Harker peaks. Since the true Harker peaks are of the form 2x, 2z, values of x and z can be obtained
</p>
<p>from the fractional coordinates on the Harker section.
</p>
<p>This analysis may be carried out graphically. The peak positions from the Harker section are
</p>
<p>replotted, on tracing paper, on a unit-cell projection in which the a and c dimensions are each reduced
</p>
<p>by a factor of 1
2
. This procedure results in one quadrant of Fig. 7.31a. The diagram is completed by
</p>
<p>operating on the first quadrant with the translation of a/2, and then on both quadrants by c/2, thus
</p>
<p>completing an area the size of the true unit-cell projection.
</p>
<p>All points marked on this map locate potential (x, z) coordinates for the heavy atoms. In fact, it
</p>
<p>contains four equivalent solutions with respect to the four unique 21 axes in the unit cell. Cross-vector
</p>
<p>peaks are found by moving this implication diagram5 to other sections of the Patterson function, using
</p>
<p>pairs of potential sites to generate potential vectors. To see how this mechanism operates, place the
</p>
<p>site marked 2 on the tracing paper over the origin of the section v &frac14; 0 and note the coincidence of site
1 with the peak 2-1. Similarly, the peak 2��1 and others on the section u &frac14; 1
</p>
<p>2
can be generated from
</p>
<p>the sites 1, 2, �1, and �2 on the implication diagram. Peaks S and �S are not explained in this way; they
</p>
<p>may be assumed to be spurious: remember (DF)2 is not a true representation of jFHj2.
Figure 7.31b shows a composite electron density map of the Pt atom sites which were prepared by
</p>
<p>an independent method, and confirms the Patterson analysis. The y coordinates of the two heavy-atom
</p>
<p>sites are almost equal, which accounts for the presence of the non-Harker peaks�(2-1) on the Harker
</p>
<p>Fig. 7.31 Interpretation of
the (DF)2 Patterson
</p>
<p>sections for ribonuclease.
</p>
<p>(a) Implication diagram.
(b) Electron density map
showing the Pt atom sites
</p>
<p>5 See Bibliography, Buerger (1959).
</p>
<p>318 7 Fourier Techniques in X-Ray Structure Determination</p>
<p/>
</div>
<div class="page"><p/>
<p>section. Figure 7.32 is a stereo-pair showing the course of the polypeptide chain in ribonuclease and
</p>
<p>the position of the main site in the Pt derivative.
</p>
<p>7.5.8 Further Details of the Isomorphous Replacement Phasing Procedure
</p>
<p>In single isomorphous replacement (SIR), the ambiguity in fP is best resolved, Fig. 7.33, by taking
</p>
<p>fP &frac14; fH with FP along the median OM between P1 and P2; jFPj should be weighted [17] by
m &frac14; cosC where C is the semi-angle between P1 and P2. By the cosine rule
</p>
<p>m &frac14; cosC &frac14; jjFPHj2 � jFPj2
</p>
<p>� jFHj2j=&eth;2jFPjjFHj&THORN;
(7.45)
</p>
<p>Fig. 7.32 Stereoviews of
the polypeptide chain in
</p>
<p>ribonuclease; the main site
</p>
<p>in the Pt derivative is
</p>
<p>shown as a simulated
</p>
<p>octahedrally coordinated
</p>
<p>group
</p>
<p>Fig. 7.33 Phase-
amplitude diagram in the
</p>
<p>SIR method; OH(�FH) is
the known reversed heavy-
</p>
<p>atom vector. The triangles
</p>
<p>OHP1 and OHP2 both
</p>
<p>satisfy (7.32), giving
</p>
<p>twofold ambiguity with the
</p>
<p>vectors either OP1 or OP2
as the solution for FP; OM
</p>
<p>defines the weighted SIR
</p>
<p>solution for FP
</p>
<p>7.5 Heavy-Atom Method and Partial Fourier Synthesis 319</p>
<p/>
</div>
<div class="page"><p/>
<p>we see that m would have a maximum value of unity in the special case for which jFHj &frac14; jjFPHj �
jFPjj, where the two circles in Fig. 7.33 are tangential. The coefficients in the SIR electron density
map would be composed of mjFPj and fH. The electron density of such a map would be subject to the
pseudo-symmetry effects discussed in Sect. 7.4.4. Taking fP &frac14; fH is thus the SIR equivalent of the
initial stage of the heavy-atom method, in which we take f &frac14; fH.
</p>
<p>Analytical Calculation of Phases in SIR and MIR
The geometrical determination of phases by the isomorphous replacement method using Harker&rsquo;s
</p>
<p>construction is impractical for several reasons:
</p>
<p>1. In MIR (multiple isomorphous replacement), phase-circle intersections, arising from accumulated
</p>
<p>errors, do not usually give absolutely clear indications of fP, as shown in Fig. 7.34 by P11 and P21.
</p>
<p>Actual phase determination in MIR, exemplified by Fig. 7.35, contains a complexity of multiple-
</p>
<p>derivative phase indications, the phase circles intersecting in rather ill-defined regions.
</p>
<p>2. The size of the task of estimating thousands of fP(hkl) values in a typical protein analysis
</p>
<p>necessitated the development of an analytical formula suitable for computer programming, as
</p>
<p>outlined below.
</p>
<p>The basis for a computational algorithm, alternative to the Harker construction for SIR, is shown in
</p>
<p>Fig. 7.36, in which the inner circle represents jFPj and the spokes represent fT, a series of trial values
of fP for fT &frac14; 0 to 360� in steps of 30�. The amplitude jFHj, which would be calculated from (7.33),
using the known heavy-atom parameters, is plotted at the end of each spoke. In order to simplify the
</p>
<p>drawing, the third side of the isomorphous replacement triangle, representing jFPHj, has not been
joined up. The SIR solutions, corresponding to P1 and P2 in Fig. 7.33, would occur when jFHj just
touches the jFPHj circle, which is plotted concentrically with the jFPj circle and is the outer circle in
Fig. 7.36. These two positions are indicated in the diagram, which should be compared with Fig. 7.33.
</p>
<p>Because of the method of selecting fT, neither is generated exactly in this method.
</p>
<p>Fig. 7.34 MIR resolution
of the phase ambiguity with
</p>
<p>a second isomorphous
</p>
<p>derivative PH2. The
</p>
<p>determined direction is
</p>
<p>near the vectors OP11
and OP21. In practice P11
and P21 rarely coincide,
</p>
<p>owing to inaccuracies in
</p>
<p>the heavy-atom parameters
</p>
<p>and a lack of true
</p>
<p>isomorphism
</p>
<p>320 7 Fourier Techniques in X-Ray Structure Determination</p>
<p/>
</div>
<div class="page"><p/>
<p>Now consider Fig. 7.37, which shows a more detailed representation of the case where fT &frac14; 30�,
Fig. 7.36. This is one of the general cases, neither P1 nor P2, where the jFPHj trial value on the Argand
diagram would not close the third side of the phase triangle properly. In the case shown jFPHj is too
short; other situations evident in Fig. 7.36 would correspond to jFPHj being too long. In these
situations there is a lack-of-closure error denoted by e&eth;fT&THORN;, Fig. 7.37. For the SIR solutions,
e&eth;fT&THORN; &frac14; 0. In general, e&eth;fT&THORN; may be calculated as follows:
</p>
<p>D2fT &frac14; jFPj
2 &thorn; jFHj2 &thorn; 2jFPjjFHj cos&eth;fT � fH&THORN; (7.46)
</p>
<p>and
</p>
<p>e2&eth;fT&THORN; &frac14; &frac12;jFPHj � D&eth;fT&THORN;�2 (7.47)
</p>
<p>Fig. 7.35 Examples ofMIRwhere the phase circles do not intersect at a point. Themost probable value of the phasef (fM
at Pmax) and the centroid phase (fB at Pcent) are indicated. (a) P(f) calculated for the two-derivative case in Fig. 7.34, with
fB &frac14; 43�, fM &frac14; 31�, and m &frac14; 0.80. (b) An example of three-derivative phasing (see Problem 7.14) with fB &frac14; 204�,
fM &frac14; 185�, andm &frac14; 0.59. Although based on three derivatives, the probability distribution in (b) is not as sharp as that in
(a), resulting in a lower figure of merit, m, and larger fB ! fM difference
</p>
<p>7.5 Heavy-Atom Method and Partial Fourier Synthesis 321</p>
<p/>
</div>
<div class="page"><p/>
<p>The SIR solutions could be determined to a satisfactory degree of precision by plotting e&eth;fT&THORN;
against fT and locating the two fT values for which e&eth;fT&THORN; &frac14; 0. This is shown for the example in
Figs. 7.33 and 7.36 by the graph of Fig. 7.38. Both solutions P1 and P2 are of course equally probable
</p>
<p>in the SIR method. In the theory of phase analysis by the MIR method, errors may be assumed to
</p>
<p>reside in jFPHj [18], which simplifies the calculations. For a given trial value of fT, the probability
that fT is the correct value is
</p>
<p>Fig. 7.37 Calculation of the lack-of-closure error e&eth;fT&THORN;
</p>
<p>Fig. 7.36 The concept of lack of closure; the inner circle represents jFPj and the outer circle jFPHj. Trial values of fP
are plotted at 30� intervals, each carrying the known FH. At P1 and P2, FH ends exactly on the FPH circle; otherwise it
fails to close, being too long for the smaller region spanning P1&ndash;P2 and too short for the rest; see Fig. 7.27 for the Harker
</p>
<p>construction of this SIR case
</p>
<p>322 7 Fourier Techniques in X-Ray Structure Determination</p>
<p/>
</div>
<div class="page"><p/>
<p>P&eth;fT&THORN; &frac14; exp&eth;�e2&eth;fT&THORN;=2E2&THORN; (7.48)
</p>
<p>where E here is the root-mean-square error in jFPHj arising from data errors.
In MIR there would be one value of e&eth;fT&THORN; per derivative. Let ei(fT) be the value for derivative i,
</p>
<p>where i &frac14; 1, 2, . . . to the total number of derivatives. Then the probability for the ith derivative is
</p>
<p>Pi&eth;fT&THORN; &frac14; exp&frac12;�e2i &eth;fT&THORN;=2E2i � (7.49)
</p>
<p>and the joint probability over all derivatives is
</p>
<p>P&eth;fT&THORN; &frac14; P1&eth;fT&THORN;P2&eth;fT&THORN;P&eth;fT&THORN; . . .
</p>
<p>or
</p>
<p>P&eth;fT&THORN; &frac14; exp �
X
</p>
<p>e2i &eth;fT&THORN;=2E2i
h i
</p>
<p>(7.50)
</p>
<p>Typical examples of probability distributions met in practice are given in Fig. 7.35. Generally
</p>
<p>the distributions are bimodal, indicating a stronger preference for one maximum over the other.
</p>
<p>The most probable electron density map uses coefficients (jFPj, fM), where fM is the phase angle
corresponding to the maximum probability in the range 0&ndash;360�. However, the electron density map
with the least overall root-mean-square error uses coefficients (jFPj, fB), where fB is the &ldquo;best&rdquo; phase
angle, corresponding to the centroid of the probability distribution, and m is a weighting function
</p>
<p>(or figure of merit), given by
</p>
<p>m cosfB &frac14;
X
</p>
<p>fT
</p>
<p>P&eth;fT&THORN; cosfT
X
</p>
<p>P&eth;fT&THORN;
.
</p>
<p>(7.51)
</p>
<p>Fig. 7.38 Values of e&eth;fT&THORN; plotted against fT; P1 and P2 are the two positions for which efT &frac14; 0, corresponding to the
SIR solutions; see Figs. 7.33 and 7.36
</p>
<p>7.5 Heavy-Atom Method and Partial Fourier Synthesis 323</p>
<p/>
</div>
<div class="page"><p/>
<p>m sinfB &frac14;
X
</p>
<p>fT
</p>
<p>P&eth;fT&THORN; sinfT
X
</p>
<p>P&eth;fT&THORN;
.
</p>
<p>(7.52)
</p>
<p>It is convenient in practice to evaluate these expressions by stepping from 0 to 360� in regular
intervals of 5 or 10�. The probability distributions and corresponding phases may be readily evaluated
by suitable programming [19].
</p>
<p>For each derivative, the root-mean-square estimate of error may be taken initially as the average
</p>
<p>E2j &frac14; h&eth;jDFij � jFHi j&THORN;
2ihkl (7.53)
</p>
<p>where the quantity
</p>
<p>DFi &frac14; jFPHi j � jFPj (7.54)
</p>
<p>is evaluated for centric reflections only.
</p>
<p>The error in a phase angle may be defined as Df &frac14; fB � fM and m &frac14; cos Df. A value of
m &frac14; 0.7 corresponds to �45�. The average value of m is a measure of the average of cos Df. In a
typical protein analysis at resolution 2 Å6 (sin ymax &frac14; l/4), an average m of 0.6&ndash;0.7 would be
acceptable. Further practical details of MIR are to be found in Sect. 10.2.12ff.
</p>
<p>Electron Density Maps Used in Large-Molecule Analysis
The correlation of heavy-atom sites between derivatives requires one to establish the coordinates of
</p>
<p>heavy atoms in a derivative i with respect to those of another derivative or combination of derivatives
</p>
<p>for which phases fP 6&frac14;i have been determined. A difference electron density map may be calculated as
</p>
<p>rDi&eth;xyz&THORN; &frac14;
1
</p>
<p>Vc
</p>
<p>X
</p>
<p>h
</p>
<p>X
</p>
<p>k
</p>
<p>X
</p>
<p>l
</p>
<p>&eth;jFPHi j � jFPj&THORN; cos&frac12;2p&eth;hx&thorn; ky&thorn; lz&THORN; � fP 6&frac14;i� (7.55)
</p>
<p>This should reveal the heavy atoms in derivative i with respect to the same origin as in the other
</p>
<p>heavy-atom derivatives. Derivative i can then be added into the MIR procedure.
</p>
<p>For a trial structure in which phases have been calculated, as in small-molecule analysis, a
</p>
<p>difference electron density map may be used in order to effect corrections to the structure:
</p>
<p>Dr&eth;xyz&THORN; &frac14; 1
Vc
</p>
<p>X
</p>
<p>j
</p>
<p>X
</p>
<p>k
</p>
<p>X
</p>
<p>l
</p>
<p>&eth;jFPj � jFcj&THORN; cos&frac12;2p&eth;hx&thorn; ky&thorn; lz&THORN; � fc� (7.56)
</p>
<p>Alternatively a double-difference map
</p>
<p>r0&eth;xyz&THORN; &frac14; 1
Vc
</p>
<p>X
</p>
<p>h
</p>
<p>X
</p>
<p>k
</p>
<p>X
</p>
<p>l
</p>
<p>&eth;2jFPj � jFcj&THORN; cos&frac12;2p&eth;hx&thorn; ky&thorn; lz&THORN; � fc� (7.57)
</p>
<p>6The resolution of a protein X-ray analysis is loosely defined as dmin, where dmin &frac14; l/2 sin ymax, ymax being the
maximum Bragg angle associated with the analysis: initially ymax may be temporarily restricted in order to limit the
</p>
<p>work required, but at the expense of the quality of the electron density image.
</p>
<p>324 7 Fourier Techniques in X-Ray Structure Determination</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec15_10">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec15_10</a></div>
</div>
<div class="page"><p/>
<p>where r0(xyz), equal to rc(xyz) + Dr(xyz), may be used since new features may be more easily
recognized in Dr(xyz) against the background of the known rc(xyz) structure. This map is very useful
</p>
<p>in computer graphics analysis, Sect. 10.4.4.
</p>
<p>In MIR the most error-free electron density is calculated as
</p>
<p>rP&eth;xyz&THORN; &frac14;
1
</p>
<p>Vc
</p>
<p>X
</p>
<p>h
</p>
<p>X
</p>
<p>k
</p>
<p>X
</p>
<p>l
</p>
<p>mjFPj
</p>
<p>� cos&frac12;2p&eth;hx&thorn; ky&thorn; lz&THORN; � fB�
(7.58)
</p>
<p>where fB is the MIR phase corresponding to the centroid of the phase probability distribution (the
</p>
<p>best phase) and m is the figure of merit; see (7.51) and (7.52).
</p>
<p>7.6 Anomalous Scattering
</p>
<p>Friedel&rsquo;s law is not an exact relationship, and becomes less so as the atomic numbers of the
</p>
<p>constituent atoms in a crystal increase. The law breaks down severely if X-rays are used that have
</p>
<p>a wavelength just less than that of an absorption edge of an atom in the crystal, Sect. 3.1.3. However,
</p>
<p>this criterion is not essential for anomalous scattering to be used in two important aspects of crystal
</p>
<p>structure analysis, namely, the determination of absolute stereochemical configurations and the
</p>
<p>phasing of reflections.
</p>
<p>Anomalous scattering introduces a phase change into a given atomic scattering factor, which
</p>
<p>becomes complex:
</p>
<p>f &frac14; fo &thorn; Df 0 &thorn; iDf 00 &frac14; f 0 &thorn; iDf 00 (7.59)
</p>
<p>Df 0 is a real correction, usually negative, and Df00 is an imaginary component which is rotated
anticlockwise through 90� in the complex plane with respect to fo and Df
</p>
<p>0, that is, to f 0.
A possible situation is illustrated in Fig. 7.39. In Fig. 7.39a, atom A is assumed to be scattering in
</p>
<p>accordance with Friedel&rsquo;s law, and it is clear that jF&eth;h&THORN;j &frac14; jF&eth;�h&THORN;j, where h stands for hkl. In Fig. 7.39b,
atom A is represented as an anomalous scattering species, with its three components according to (7.59).
</p>
<p>In this situation, jF&eth;h&THORN;j 6&frac14; jF&eth;�h&THORN;j, and intensity measurements of Friedel pairs of reflections produce
different values.
</p>
<p>We can assume safely that procedures for measuring I(hkl) differentiate correctly between hkl and
�h �k �l. In any non-centrosymmetric space group a structure model can be inverted, as if through a center
</p>
<p>of symmetry, and used to recalculate the structure factors. Because structure factor formulae involve
</p>
<p>(hx + ky + lz), the two models will produce different values for jFc(hkl)j and jFc&eth;�h �k �l&THORN;j. The correct
enantiomorph is expected to produce better agreement between Fo and jFcj and, thus, a lower R-factor.
Some typical results are listed in Table 7.6, from which it may be deduced that the structure giving
</p>
<p>jFcj�x;�y;�z corresponds to the absolute configuration. An equivalent procedure would be to measure the
values of both jF(hkl)j and jF&eth;�h �k �l&THORN;j and compare them with jFcjx,y,z. The technique can be used only
with crystals which are non-centrosymmetric, because jF&eth;h&THORN;j &frac14; jF&eth;�h&THORN;j in centrosymmetric crystals,
but this limitation is not important because molecules which crystallize with a single enantiomorph
</p>
<p>cannot do so in a space group containing any form of inversion symmetry.
</p>
<p>7.6 Anomalous Scattering 325</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec24_10">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec24_10</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec4_3">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec4_3</a></div>
</div>
<div class="page"><p/>
<p>7.6.1 The Flack x Parameter
</p>
<p>It is clear from the above treatment that a more complete, rigorous, and easily applied method for the
</p>
<p>determination of absolute configuration would be highly desirable. To this end several proposals have
</p>
<p>been made: Hamilton&rsquo;s R-ratio test [20] requires the number of observations to be known; Rogers&rsquo; �
parameter [21] attempted to overcome operational difficulties associated with Hamilton&rsquo;s test, but
</p>
<p>assumed that reflections for which jF&eth;hkl&THORN;j &frac14; jF&eth;�h �k �l&THORN;j, and there are always some except in space
group P1, had no influence on the refinement of � and its standard deviation. Flack&rsquo;s x parameter
method [22] showed that the assumption of Rogers was not always valid, and it has proved to be the
</p>
<p>most useful and reliable method; it is currently employed in program packages such as SHELX-97.
</p>
<p>The Flack x parameter is refined by least squares, together with the other structural parameters,
</p>
<p>Sect. 8.4.3, to a final value and corresponding standard deviation. The Flack parameter x is defined in
</p>
<p>terms of jFcj by the equation
</p>
<p>jFc&eth;hkl&THORN;j2 &frac14; &eth;1� x&THORN;jFc&eth;hkl&THORN;j2 &thorn; xjFc&eth;�h �k �l&THORN;j2 (7.60)
</p>
<p>Fig. 7.39 Anomalous
scattering of atom A with
</p>
<p>respect to the rest of the
</p>
<p>structure R. (a) Normal
case: jF&eth;h&THORN;j &frac14; jF&eth;�h&THORN;j.
(b) Anomalous
case: jF&eth;h&THORN;j 6&frac14; jF&eth;�h&THORN;j.
The general non-
</p>
<p>centrosymmetric case
</p>
<p>is illustrated. For
</p>
<p>centrosymmetric crystals,
</p>
<p>jFhj always equals jFhj, but
fh differs to a small degree
</p>
<p>from 0 or p if anomalous
</p>
<p>scattering is significant
</p>
<p>Table 7.6 Example of some Friedel pairs and the
corresponding jFj values
hkl Fo jFcjx, y, z jFcjx;y;z
121 17.0 19.1 18.3
</p>
<p>122 21.2 22.9 21.9
</p>
<p>123 41.4 44.4 42.8
</p>
<p>341 36.7 38.7 35.5
</p>
<p>342 7.8 9.5 8.2
</p>
<p>413 14.2 15.3 13.5
</p>
<p>326 7 Fourier Techniques in X-Ray Structure Determination</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec32_8">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec32_8</a></div>
</div>
<div class="page"><p/>
<p>When the atomic coordinate set and the crystal have the same chirality, x takes the value zero; if
</p>
<p>they are different, x is equal to unity. A result is considered acceptable if x lies within 3 standard
</p>
<p>deviations, that is, �3s(x) as calculated by the program, of either zero or unity. If a value of unity is
returned, the coordinates of the model should be inverted in the origin; the data collection routines
</p>
<p>ensure that the hkl data have the correct polarity. After further refinement a value of 0.0 � 3s(x) for
x should result.
</p>
<p>This method has withstood innumerable tests and can be considered reliable in most circum-
</p>
<p>stances. Some examples are given below. A further advantage of this method is that although, as with
</p>
<p>all analyses, the measured intensity data set should cover as large a volume of reciprocal space as
</p>
<p>possible, it is not absolutely necessary to measure Friedel or Bijvoet pairs, Sect. 7.6.2. In practice, for
</p>
<p>the test to be reliable, s(x) in x should be 0.05 or less. Because the refinement routine in SHELX
</p>
<p>refines against jFj2 as opposed to jFj Sect. 8.4.2, the program determines x within a small number of
iterations. It should also be noted that since the value of xmay be related to twinning, Sect. 5.10.2, it is
</p>
<p>advisable to carry out further tests for this effect as prescribed, for example, in the SHELX system
</p>
<p>manual. Quite rightly the SHELX manual advises that it is important to refine every non-
</p>
<p>centrosymmetric structure as the correct absolute structure in order to avoid introducing systematic
</p>
<p>errors into the molecular geometry. In fact, it cannot be overemphasized that it is often imperative to
</p>
<p>determine the absolute configuration, especially in the case of new drug molecules for which a
</p>
<p>reversal of chirality is almost certain to lead to a dramatic change in potency and other binding
</p>
<p>characteristics. The classic example [23] of this is that of thalidomide in the early 1960s. In some
</p>
<p>cases the absolute structure will be known with certainty, for example, proteins which are predomi-
</p>
<p>nantly L-amino acid polypeptides, but in others it has to be deduced from the X-ray structure.
</p>
<p>If the structure includes a reasonably heavy atom, such as phosphorus, sulphur or chlorine, the
</p>
<p>anomalous scattering effect from these atoms using Cu Ka or even accurate high-resolution low-
</p>
<p>temperatureMoKa radiationmay be sufficient for determining the absolute configuration, particularly
</p>
<p>if Friedel opposites have been measured. Sometimes the presence of as few as two oxygen atoms per
</p>
<p>molecule has been shown to produce a reliable result, Sect. 9.4.4. In SHELX the refinement program
</p>
<p>estimates [22] the parameters x and s(x) for the absolute structure.
</p>
<p>The parameter x is effectively the fractional contribution of the inverted component of a &ldquo;racemic
</p>
<p>twin&rdquo; and, as we have said, should be zero if the absolute structure is correct, or 1 if it needs to be
</p>
<p>inverted in the origin; but it could be somewhere between 0 and 1 if racemic twinning is really
</p>
<p>present. If this is the case, x can be refined along with all the other parameters, and the instructions for
</p>
<p>TWIN refinement in the SHELXmanual must be followed faithfully; fortunately such cases are fairly
</p>
<p>uncommon.
</p>
<p>Examples to Illustrate Results of Flack Parameter Refinement
The following six crystal structures were all refined using SHELX-97 and have different degrees of
</p>
<p>heavy atom content. The Flack parameter produced in the final least-squares cycle was within three
</p>
<p>standard deviations of zero in each of the first five structures. All six structures are, of course, in non-
</p>
<p>centrosymmetric space groups and the first four have single molecular species of one specific hand,
</p>
<p>determined absolutely by the X-ray analysis. The fifth structure, atropine, was crystallized from a
</p>
<p>commercial racemic mixture in the unusual space group Fdd2 and contains both enantiomorphic
</p>
<p>forms of the atropine molecule. The space group itself is enantiomorphous, Tables 2.7, and the crystal
</p>
<p>structure has therefore established, via the Flack parameter, the correct hand of the three-dimensional
</p>
<p>arrangement ofmolecules in the crystal. For the sixth example the value of xwas notwithin 3s of 0 (or 1).
</p>
<p>1. Form I of Cholesteryl Iodide [24], C27H45I
</p>
<p>X-ray data were collected at low temperature (100 K) with Mo Ka radiation (l &frac14; 0.71073 Å). The
crystals are monoclinic, space group P21 with a &frac14; 12.577(1) Å, b &frac14; 9.009(1) Å, c &frac14; 12.862
</p>
<p>7.6 Anomalous Scattering 327</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec31_8">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec31_8</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec21_9">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec21_9</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Tab7_2">http://dx.doi.org/10.1007/978-1-4614-3954-7#Tab7_2</a></div>
</div>
<div class="page"><p/>
<p>(1) Å, b &frac14; 119.00(1)�, and Z &frac14; 2molecules per unit cell. The crystal size was 0.35, 0.56, 0.06mm,
and the y-range for data collection 1.81&ndash;29.15�. The number of reflections collected was 14457,
with 5729 unique reflections (Rint &frac14; 0.0632). Refinement by full-matrix least-squares on jFj2 gave
final R indices (I &gt; 2sI) R1 &frac14; 0.0499, weighted R2 &frac14; 0.1119; R indices on all data were
R1 &frac14; 0.0751 and weighted R2 &frac14; 0.1345. The absolute structure parameter x was �0.04(3)
which is zero within 3s, s being less than 0.05.
</p>
<p>2. Form II of Cholesteryl Iodide [24], C27H45I
</p>
<p>X-ray data were collected at room temperature with Cu Ka radiation (l &frac14; 1.54178 Å). Crystals are
monoclinic, space group P21 with a &frac14; 11.005(1) Å, b &frac14; 10.469(1) Å, c &frac14; 11.6840(10) Å,
b &frac14; 106.827(6)�, and Z &frac14; 2. The crystal size was 0.24, 0.42, 0.24 mm, and the y-range for data
collection was 3.95&ndash;69.91�. The number of reflections collected was 3543, with 2470 unique
reflections (Rint &frac14; 0.0191). Refinement was by full-matrix least-squares on jFj2 which gave final R
indices (I &gt; 2sI): R1 &frac14; 0.0480, weighted R2 &frac14; 0.1299; R indices on all data were R1 &frac14; 0.0537,
weighted R2 &frac14; 0.1447. The absolute structure parameter x &frac14; 0.00(1) which is zero within 3s, s
being less than 0.05.
</p>
<p>The structures (1) and (2) are polymorphs, and have exactly the same chemical composition.
</p>
<p>The iodine atom is a strong anomalous scattering species for either Mo radiation in the case of (1)
</p>
<p>or Cu radiation in the case of (2). Both analyses confirm the absolute configuration of cholesterol,
</p>
<p>an important biological substance, and of steroids in general.
</p>
<p>3. Compound BW202W92: C12H12Cl3FN4O3 (C11H9Cl3FN4&middot;CH3SO3) (R-form) [25]
</p>
<p>X-ray data were collected at room temperature (293 K) with Cu Ka radiation (l &frac14; 1.54178 Å). The
crystals are monoclinic, space group P21 with a &frac14; 8.384(2) Å, b &frac14; 16.984(3) Å, c &frac14; 12.480(3) Å,
b &frac14; 104.14(6)�, and Z &frac14; 2. The crystal size was 0.30, 0.16, 0.16 mm, and the y-range for data
collection 4.49&ndash;74.42�. The number of reflections collected was 3990 reflections, with 3192 unique
reflections (Rint &frac14; 0.0496). Refinement was by full-matrix least-squares on jFj2 which gave final R
indices (I &gt; 2sI) R1 &frac14; 0.0490, weighted R2 &frac14; 0.1254; R indices on all data were R1 &frac14; 0.0644,
weighted R2 &frac14; 0.1324. The absolute structure parameter x &frac14; 0.05(3) which is zero within
3s, s being less than 0.05.
</p>
<p>4. Compound BW203W92: C12H12Cl3FN4O3 (C11H9Cl3FN4&middot;CH3SO3) (S-form) [25]
</p>
<p>X-ray data were collected at low temperature (123 K) with Mo Ka radiation (l &frac14; 0.71073 Å).
The crystals are triclinic, space group P1 with a &frac14; 7.716(2) Å, b &frac14; 8.120(2) Å, c &frac14; 13.719
(3) Å, a &frac14; 74.91(3)�, b &frac14; 87.69(3)�, g &frac14; 89.83(3)�, and Z &frac14; 1. The crystal size was 0.30, 0.20,
0.20 mm, and the y-range for data collection 2.60&ndash;27.47�. The number of reflections collected
was 6456, with 5302 unique reflections (Rint &frac14; 0.0733). Refinement was by full-matrix least-
squares on jFj2, which gave final R indices (I &gt; 2sI) R1 &frac14; 0.0614, weighted R2 &frac14; 0.1204; R
indices on all data were R1 &frac14; 0.1248, weighted R2 &frac14; 0.1468. The absolute structure parameter
x &frac14; �0.10(9) which is zero within 3s, but s here is greater than 0.05.
</p>
<p>Structures (3) and (4) are atropisomers [25] and have exactly the same chemical structure.
</p>
<p>Chemical and spectroscopic studies have identified (3) as the R-isomer and (4) as the S-isomer, but
</p>
<p>with insufficient details to enable their three-dimensional structures to be finalized; this was
</p>
<p>achieved through X-ray structure analyses. The structures are good candidates for anomalous
</p>
<p>X-ray scattering by virtue of their three chlorine, one sulphur, and three oxygen atoms. It is clear,
</p>
<p>however, that the result for structure (3) is more convincing, being based on copper X-radiation
</p>
<p>data and resulting in x &frac14; 0.05(3), which is zero within 3s (s &lt; 0.05), while structure (4), based on
molybdenum X-radiation data, has x &frac14; �0.10(9) which is also zero within 3s, but with s greater
than 0.05, slightly larger than the recommended limit. Nevertheless, it is safe to say that the X-ray
</p>
<p>analyses have established the absolute configurations of these important drugs for which (3) is a
</p>
<p>potent/selective sodium-ion channel blocker and (4) is about 30 times less potent in this capacity.
</p>
<p>328 7 Fourier Techniques in X-Ray Structure Determination</p>
<p/>
</div>
<div class="page"><p/>
<p>The R and S enantiomers derived from the X-ray analyses are shown in Fig. 7.40; see also
</p>
<p>Sect. 8.2.16.
</p>
<p>5. Atropine [28], C17H23NO3
X-ray data were collected at low temperature (100 K) with Mo Ka radiation (l &frac14; 0.71073 Å). The
crystals are orthorhombic, space group Fdd2 with a &frac14; 24.291(5) Å, b &frac14; 39.538(8) Å, c &frac14; 6.473
(1) Å, and Z &frac14; 16. The crystal size was 0.20, 0.15, 0.35 mm, and the y-range for data collection
3.30&ndash;25.02�. The number of independent reflections collected was 14353, with 2710 unique
reflections (Rint &frac14; 0.0820). Refinement was by full-matrix least-squares on jFj2, which gave
final R indices (I &gt; 2sI) R1 &frac14; 0.0452, weighted R2 &frac14; 0.1206; R indices on all data
R1 &frac14; 0.0558, weighted R2 &frac14; 0.1294. The absolute structure parameter x &frac14; �0.1(2) which is
zero within 3s, but with s greater than 0.05. Considering that there are no heavy anomalous
</p>
<p>scattering species in this structure, only three oxygen atoms, this is an excellent result and fully
</p>
<p>acceptable; see also Sect. 8.3.7.
</p>
<p>6. 1-Benzyl-1H-tetrazole, C8H8N4
X-ray data were collected at low temperature (120 K) with Mo Ka radiation (l &frac14; 0.71073Å). The
crystals aremonoclinic, space groupP21with a &frac14; 7.6843(5) Å, b &frac14; 5.5794(7) Å, c &frac14; 9.4459(7) Å,
and Z &frac14; 2. The crystal size was 0.09, 0.03, 0.02mm, and the y-range for data collection 3.14&ndash;27.45�.
The number of independent reflections collected was 5624, with 1718 unique reflections (Rint
&frac14; 0.0588). The refinement was by full-matrix least-squares on jFj2, which gave final R indices
(I &gt; 2sI) R1 &frac14; 0.0447, weighted R2 &frac14; 0.0786; R indices on all data were R1 &frac14; 0.0695, weighted
R2 &frac14; 0.0870. The absolute structure parameter x &frac14; 0(3). Inversion of the structure produced a result
with an absolute structure parameter x &frac14; 1(3).
The molecular structure is shown in Fig. 7.41. What does the Flack parameter mean in this case? In
</p>
<p>fact, the values of x for the structure and its inverse are exactly correct, 0 and 1, but the estimated
</p>
<p>standard deviation of 3 in each case deprives this result of a statistical significance. We can relate the
</p>
<p>Fig. 7.40 Molecular conformations of BW202W92: C11H9Cl3FN4&middot;CH3SO3 (R-form) and BW203W92:
C12H12Cl3FN4O3 (S-form) [25] the R and S enantiomers respectively of 2,4-diamino-6-fluoromethyl-5-(2,3,5-trichlor-
</p>
<p>ophenyl)-pyrimidine (Drawn with Ortep/Raster [26, 27]). Thermal ellipsoids are shown at 50% probability; for clarity
</p>
<p>the CH3SO3 moieties have been omitted. The views are shown perpendicular to the pyrimidine ring (RHS) in order to
</p>
<p>emphasize the difference between the two molecules with respect to the central linking bond
</p>
<p>7.6 Anomalous Scattering 329</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec18_8">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec18_8</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec27_8">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec27_8</a></div>
</div>
<div class="page"><p/>
<p>failure of the Flack refinement to the lack of significant anomalous scattering species in the molecule
</p>
<p>as well as to the choice of molybdenum radiation. Inspection of Fig. 7.41 reveals that there are no
</p>
<p>chiral centers in this molecule but the space group P21 is enantiomorphic, Table 10.1, and therefore
</p>
<p>could be more associated with molecules which are chiral. In this case, there is no need for concern
</p>
<p>over the absolute configuration of the molecule as it does not have one. It should be noted that in fact
</p>
<p>molecules chemically similar to this species usually crystallize in centrosymmetric space groups,
</p>
<p>avoiding space groups such as P21.
</p>
<p>Conclusions
These examples illustrate the successful use of Flack parameter refinement to determine absolute
</p>
<p>configuration. It is easy to apply and has a high chance of success particularly where there are a good
</p>
<p>number of anomalous scattering species in the structure and the diffraction data has been measured
</p>
<p>carefully and completely. The cause of the failed result lay in the given experimental conditions. It
</p>
<p>should be said, however, that it is always sensible to refine the Flack x parameter in all cases and run the
</p>
<p>final refinement stages again if the test fails.
</p>
<p>7.6.2 Effect of Anomalous Scattering on the Symmetry of Diffraction Patterns
</p>
<p>We have seen that when Friedel&rsquo;s law holds, the X-ray diffraction pattern, considered as a three-
</p>
<p>dimensional weighted reciprocal lattice, exhibits a symmetry equivalent to that of the point group of
</p>
<p>the crystal with an additional center of symmetry, if not already present, that is, to one of the eleven
</p>
<p>diffraction symmetry groups (Laue groups) listed in Table 1.6. However, for a structure in which
</p>
<p>some of the atoms scatter anomalously Friedel&rsquo;s law breaks down and the symmetry of the diffraction
</p>
<p>pattern then reverts to that of the point group of the crystal. For a centrosymmetric crystal jF(hkl)j still
equals jF&eth;�h �k �l&THORN;j, although the phase angle is no longer 0 or p. As an example to illustrate the effects of
anomalous scattering let us consider a crystal in space group P21.
</p>
<p>Diffraction Symmetry for a Crystal in Space Group P21 with No Anomalous Scattering
A given crystal is monoclinic, belonging to point group 2, and the diffraction symmetry if Friedel&rsquo;s
</p>
<p>law holds is 2/m. The jF(hkl)j equivalents are thus jF&eth;hkl&THORN;j &frac14; jF&eth;�h k �l&THORN;j &frac14; jF&eth;h �k l&THORN;j &frac14; jF&eth;�h �k �l&THORN;j. On
X-ray photographs, for example, zero-level and upper-level a-axis precession photographs, this
</p>
<p>symmetry will be manifested as indicated in Fig. 7.42. The zero level, Fig. 7.42a(i), clearly
</p>
<p>Fig. 7.41 X-ray molecular structure of 1-benzyl-H-tetrazole [29] (Drawn with Ortep/Raster [26, 27]). Thermal
ellipsoids are shown at 50 % probability
</p>
<p>330 7 Fourier Techniques in X-Ray Structure Determination</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Tab1_10">http://dx.doi.org/10.1007/978-1-4614-3954-7#Tab1_10</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Tab6_1">http://dx.doi.org/10.1007/978-1-4614-3954-7#Tab6_1</a></div>
</div>
<div class="page"><p/>
<p>demonstrates symmetry mm2 with jF&eth;0kl&THORN;j &frac14; jF&eth;0�k l&THORN;j &frac14; jF&eth;0k�l&THORN;j &frac14; jF&eth;0�k �l&THORN;j, while upper-level
photographs exhibit symmetry m (perpendicular to b*) with, for example jF&eth;1kl&THORN;j &frac14; jF&eth;1�kl&THORN;j and
jF&eth;1k�l&THORN;j &frac14; jF&eth;1�k �l&THORN;j respectively, but jF&eth;1kl&THORN;j 6&frac14; jF&eth;1k�l&THORN;j, Fig. 7.42a(ii).
</p>
<p>Diffraction Symmetry for a Crystal in Space Group P21 for a Structure Containing Some
Anomalous Scattering Species
The diffraction symmetry is now that of point group 2, for which jF&eth;hkl&THORN;j &frac14; jF&eth;�hk�l&THORN;j and
jF&eth;�h �k �l&THORN;j &frac14; jF&eth;h�kl&THORN;j. On the a-axis precession photographs shown schematically in Fig. 7.42b(i) we
observe, on the zero level, jF&eth;0kl&THORN;j &frac14; jF&eth;0k�l&THORN;j and jF&eth;0�kl&THORN;j &frac14; jF&eth;0�k �l&THORN;j, but jF&eth;0kl&THORN;j 6&frac14; jF&eth;0�kl&THORN;j and
jF&eth;0k�l&THORN;j 6&frac14; jF&eth;0�k �l&THORN;j because there is now no m plane perpendicular to b*. On the upper-level photo-
graphs, no symmetry is observable and all four jFj values are different, Fig. 7.40b(ii).
</p>
<p>We may define the anomalous difference as
</p>
<p>DFANO &frac14; &eth;jF&eth;hkl&THORN;j � jF&eth;�h �k �l&THORN;j&THORN;
or as
</p>
<p>DFANO &frac14; &eth;jF&eth;&thorn;&THORN;j � jF&eth;�&THORN;j&THORN;
(7.61)
</p>
<p>From the above discussion, for the case of space group P21 with anomalous scattering species
</p>
<p>present, DFANO is also given by
</p>
<p>DFANO &frac14; &eth;jF&eth;hkl&THORN;j � jF&eth;h�kl&THORN;j&THORN; (7.62)
</p>
<p>since jF&eth;�h �k �l&THORN;j &frac14; jF&eth;h�kl&THORN;j. Differences such as (7.62), equivalent by point-group symmetry to the
difference between jFj for a Friedel pair, are known as Bijvoet differences, the two reflections
involved being denoted as a Bijvoet pair [30].
</p>
<p>Fig. 7.42 Schematic representation of symmetry exhibited in b*c* sections of the reciprocal lattice for a crystal with space
groupP21 (or any othermonoclinic space group in crystal class 2). (a) No anomalous scattering; the diffraction symmetry is
2/m. (i) Symmetry of 0kl section ismm2. (ii) Symmetry of 1kl, 2kl, . . . sections ism&perp; b*. Equivalent hkl are represented by
the same symbolX, etc. (b) Anomalous scattering case; the diffraction symmetry is 2. (i) Symmetry of 0kl section ism&perp; b.*
</p>
<p>(ii) Symmetry of 1kl, 2kl, . . . sections is reduced to 1 (no m present)
</p>
<p>7.6 Anomalous Scattering 331</p>
<p/>
</div>
<div class="page"><p/>
<p>Bijvoet differences can be observed on X-ray photographs represented in Fig. 7.42a, b as follows:
</p>
<p>(a) Zero level: h &frac14; 0
</p>
<p>&eth;jF&eth;0kl&THORN;j � jF&eth;0�kl&THORN;j&THORN; Bijvoet pair
&eth;jF&eth;0k�l&THORN;j � jF&eth;0�k �l&THORN;j&THORN; Bijvoet pair
</p>
<p>These differences will be equivalent by symmetry.
</p>
<p>(b) Upper level
</p>
<p>&eth;jF&eth;hkl&THORN;j � jF&eth;h�kl&THORN;j&THORN; Bijvoet pair
&eth;jF&eth;hk�l&THORN;j � jF&eth;h�k �l&THORN;j&THORN; Bijvoet pair
</p>
<p>These differences are not equivalent.
</p>
<p>It is, thus, possible to monitor anomalous differences indirectly on the same photograph through the
</p>
<p>use of Bijvoet pairs, whereas Friedel pairs, in the true sense of the definition, would necessarily always
</p>
<p>occur on different photographs (except for axial reflections, such as 00l and 00�l).
</p>
<p>7.6.3 Form of the Structure Factor for a Structure Composed of Heavy-Atom
Anomalous Scattering Species
</p>
<p>For a structure composed of NH heavy-atom anomalous scattering species, the structure factor
</p>
<p>becomes
</p>
<p>F&eth;hkl&THORN; &frac14;
XNH
</p>
<p>j&frac14;1
&eth; f 0j &thorn; iDf 00j &THORN; exp 2pi&eth;hxj &thorn; kyj &thorn; lzj&THORN; (7.63)
</p>
<p>which can be written as
</p>
<p>FH&eth;&thorn;&THORN; &frac14; F0H&eth;&thorn;&THORN; &thorn; iF00 H&eth;&thorn;&THORN;
</p>
<p>where
</p>
<p>F
0
</p>
<p>H&eth;&thorn;&THORN; &frac14;
XNH
</p>
<p>j&frac14;1
f
0
</p>
<p>j exp 2pi&eth;hxj &thorn; kyj &thorn; lzj&THORN; (7.64)
</p>
<p>and
</p>
<p>F00H&eth;&thorn;&THORN; &frac14;
XNH
</p>
<p>j&frac14;1
Df 00j exp 2pi&eth;hxj &thorn; kyj &thorn; lzj&THORN; (7.65)
</p>
<p>Similarly the structure factor F&eth;�h �k �l&THORN; can be written as
</p>
<p>FH&eth;�&THORN; &frac14; F0H&eth;�&THORN; &thorn; iF00H&eth;�&THORN;
</p>
<p>where
</p>
<p>332 7 Fourier Techniques in X-Ray Structure Determination</p>
<p/>
</div>
<div class="page"><p/>
<p>F0H&eth;�&THORN; &frac14;
XNH
</p>
<p>j&frac14;1
f 0j expf�2pi&eth;hxj &thorn; kyj &thorn; lzj&THORN;g (7.66)
</p>
<p>and
</p>
<p>F00H&eth;�&THORN; &frac14;
XNH
</p>
<p>j&frac14;1
Df 00j expf�2pi&eth;hxj &thorn; kyj &thorn; lzj&THORN;g (7.67)
</p>
<p>The form of the structure factor for a protein heavy-atom derivative crystal composed of protein
</p>
<p>atoms P, light atom, negligible-anomalous scattering species, mainly C, N, O atoms, and heavy atoms
</p>
<p>H, anomalous scattering species, becomes
</p>
<p>FPH&eth;&thorn;&THORN; &frac14; FP&eth;&thorn;&THORN; &thorn; F0H&eth;&thorn;&THORN; &thorn; iF00H&eth;&thorn;&THORN; for hkl (7.68)
</p>
<p>FPH&eth;�&THORN; &frac14; FP&eth;�&THORN; &thorn; F0H&eth;�&THORN; &thorn; iF00H&eth;�&THORN; for �h �k �l (7.69)
</p>
<p>FP is the normal protein structure factor and FP(+) is clearly equal to FP(�), Fig. 7.39a.
The structure factors FPH(+) and FPH(�) are represented in Fig. 7.43, where OC &frac14; FPH(+) and
</p>
<p>OD &frac14; FPH(�). Clearly jFPH(+)j 6&frac14; jFPH(�)j. Figure 7.43 also shows the effect of reflecting FPH(�)
across the real axis OR. This device simplifies the following calculation. Using the cosine rule in
</p>
<p>triangles OBC and OBD0, we obtain
</p>
<p>jFPH&eth;&thorn;&THORN;j2 &frac14; jFPHj2 &thorn; jF00Hj2 � 2jFPHjjF00Hj
� cos&eth;fPH � fH &thorn; 90&THORN;
</p>
<p>(7.70)
</p>
<p>jFPH&eth;�&THORN;j2 &frac14; jFPHj2 &thorn; jF00Hj2 � 2jFPHjjF00Hj
� cos&eth;90� fPH � fH&THORN;
</p>
<p>(7.71)
</p>
<p>Subtracting (7.71) from (7.70) gives
</p>
<p>jFPH&eth;&thorn;&THORN;j2 � jFPH&eth;�&THORN;j2
</p>
<p>&frac14; &frac12;jFPH&eth;&thorn;&THORN;j &thorn; jFPH&eth;�&THORN;j�&frac12;jFPH&eth;&thorn;&THORN;j � jFPH&eth;�&THORN;j�
&frac14; �2jFPHjjF00Hj&frac12;cos&eth;fPH � fH &thorn; 90� � cos&eth;90� fPH � fH&THORN;�
&frac14; 4jFPHjjF00Hj sin&eth;fPH � fH&THORN;
</p>
<p>(7.72)
</p>
<p>But, intuitively,
</p>
<p>jFPH&eth;&thorn;&THORN;j &thorn; jFPH&eth;�&THORN;j &frac14; OC&thorn; OD0 � 2OB&eth;&frac14; 2jFPHj&THORN; (7.73)
</p>
<p>Thus, it follows that the anomalous difference
</p>
<p>DFANO &frac14; jFPH&eth;&thorn;&THORN;j � jFPH&eth;�&THORN;j � 2jF00Hj sin&eth;fPH � fH&THORN; (7.74)
</p>
<p>For a given heavy-atom type, it is known that the ratio jF0Hj=jF00Hj &frac14; f 0=Df 00, and is approximately
constant, k, say. Thus,
</p>
<p>7.6 Anomalous Scattering 333</p>
<p/>
</div>
<div class="page"><p/>
<p>DFANO � &eth;2jF0Hj=k&THORN; sin&eth;fPH � fH&THORN;
</p>
<p>or
</p>
<p>jF0Hj sin&eth;fPH � fH&THORN; � &eth;k=2&THORN;DFANO (7.75)
</p>
<p>From (7.44), we see that DFISO � jFHj cos(fH � fPH), and in the case of anomalous scattering
jFHj becomes jF0Hj. It follows that jF0Hj � &frac12;&eth;DFISO&THORN;2 &thorn; &eth;k=2&THORN;DFANO&THORN;2�1=2.
</p>
<p>This provides a possible method for estimating jF0Hj for calculation of a difference Patterson map
when anomalous scattering measurements are available.
</p>
<p>7.6.4 Phasing by Use of Anomalous Scattering
</p>
<p>Anomalous scattering can be used in phasing reflections. We saw in the previous section that the
</p>
<p>isomorphous replacement technique in non-centrosymmetric crystals leads to an ambiguity in phase
</p>
<p>determination, Fig. 7.27. The ambiguity cannot be resolved unless the replaceable site is changed.
</p>
<p>Merely using a third derivative with the same replaceable site would lead to a situation comparable
</p>
<p>with that in Fig. 7.27. The heavy-atom vector would still be directed along OH, and its different
</p>
<p>length would be just balanced by the change in Fo, so that three circles would intersect at P1 and P2.
</p>
<p>Fig. 7.43 Structure factors FPH(+) and FPH(�) for a protein crystal containing heavy-atom anomalous scattering
species: dROC &frac14; fPH&eth;&thorn;&THORN;; dROD &frac14; dROD; dROB &frac14; fPH; dROA &frac14; fP; dOAB &frac14; fP &thorn; &eth;180� fH&THORN;; OA &frac14; jFP(+)j; AB &frac14;
jFH(+)j; BC &frac14; jF&rsquo;&rsquo;H&eth;&thorn;&THORN;j; OB &frac14; jFPHj; OC &frac14; jFPH(+)j; OD &frac14; jFPH(�)j; OD0 &frac14; mirror reflection of OD; OE &frac14;
jFP(�)j; EF &frac14; jFH(�)j; FD &frac14; jF&rsquo;&rsquo;H&eth;�&THORN;j; dOBC &frac14; w &frac14; fPH � fH &thorn; 90; AC &frac14; AD0 &frac14; ED &frac14; jFHj
</p>
<p>334 7 Fourier Techniques in X-Ray Structure Determination</p>
<p/>
</div>
<div class="page"><p/>
<p>Avector change in the heavy-atom contribution can be brought about through anomalous scattering in
</p>
<p>a given derivative, instead of invoking a different replaceable site. Following Fig. 7.39b we see that two
</p>
<p>differentFo values can arise for h and �h. Consequently,f&eth;h&THORN; 6&frac14; f&eth;�h&THORN;, and the ambiguity can be resolved
by the experimental data. This technique is particularly important with synchrotron radiation, where the
</p>
<p>wavelength can be tuned to the absorption edge of a relatively heavy atom in the structure so as to obtain
</p>
<p>the maximum difference between jFo(h)j and Fo&eth;�h&THORN;.
</p>
<p>7.6.5 Resolution of the Phase Problem for Proteins Using Anomalous Scattering
Measurements (SIRAS Method)7
</p>
<p>AHarker diagram for any hkl reflection FP(+) may be constructed as discussed earlier by first drawing
</p>
<p>a circle of center O and radius jFP(+)j. From C+, the end of the vector �FH(+) as center, a second
circle of radius jFPH(+)j is drawn, as shown in Fig. 7.44. It intersects the FP(+) circle in points P1(+)
and P2(+); OP1(+) and OP2(+) represent the SIR phase ambiguity noted previously, Sect. 7.4.8.
</p>
<p>The Harker diagram for the corresponding �h �k �l reflection FP(�) is constructed by drawing a circle of
radius jFP(�)j, center O, and finally a circle of radius jFPH(�)j, centered at C�, the end of the value of
�FH(�), as shown inFig. 7.45. The two circles this time intersect in pointsP1(�) andP2(�), representing
the ambiguous SIR solution for FP(�).
</p>
<p>Since FP(+) and FP(�) are related by reflection across the real axisOR, Figs. 7.44 and 7.45, the correct
solutions for FP(+) and FP(�) will be mirror-related in this way, while the incorrect pair will not. In the
given case the corresponding correct solutions areP1(+) andP1(�) andP2(+) andP2(�) are the unwanted,
erroneous solutions. In order to rationalize this process a combinedHarker diagram can be conceived as in
</p>
<p>Fig. 7.46. This involves plotting the FPH(�) circle at the end of themirrored FH(�) vector, thus enabling
</p>
<p>Fig. 7.44 Harker diagram
for FP(+)(hkl) for a protein
</p>
<p>crystal with heavy atoms
</p>
<p>scattering anomalously.
</p>
<p>OJ+ is the vector
</p>
<p>� F&rsquo;H&eth;&thorn;&THORN;, and J+C+ is the
imaginary component of
</p>
<p>OC+, the vector for
</p>
<p>�FH(+). The FP(+) and
FPH(+) circles intersect at
</p>
<p>P1(+) and P2(+). OP1(+)
</p>
<p>and OP2(+) are the
</p>
<p>ambiguous solutions for
</p>
<p>FP(+)
</p>
<p>7 Single Isomorphous Replacement with Anomalous Scattering.
</p>
<p>7.6 Anomalous Scattering 335</p>
<p/>
</div>
<div class="page"><p/>
<p>solutions P1(+) and P1(�) mirrored to coalesce. In practice, the phasing process is carried out by
calculating a probability distribution in a similar manner to that used in the MIR technique. Algorithms
</p>
<p>for carrying out these computations have been developed.
</p>
<p>Fig. 7.46 Combined
Harker diagram for FP(+)
</p>
<p>and FP(�), with FP(�)
mirrored across the real
</p>
<p>axis of the Argand
</p>
<p>diagram: C�m is the mirrored
point of C�, where OC� is
the vector for �FH(�), as
in Fig. 7.43. The correct
</p>
<p>solution for FP(+) is
</p>
<p>OP1(+), coinciding with
</p>
<p>OP1(�) mirrored
</p>
<p>Fig. 7.45 Harker diagram
for FP(�)(hkl) for a protein
crystal with heavy atoms
</p>
<p>scattering anomalously.
</p>
<p>OJ� is the vector for
�FH(�), and J�C� is the
imaginary component of
</p>
<p>OC�, the vector for
�FH(�). The FP(�) and
FPH(�) circles intersect at
P1(�) and P2(�). OP1(�)
and OP2(�) are the
ambiguous solutions for
</p>
<p>FP(�)
</p>
<p>336 7 Fourier Techniques in X-Ray Structure Determination</p>
<p/>
</div>
<div class="page"><p/>
<p>7.6.6 Protein Phasing Using the Multiple-Wavelength Anomalous Dispersion
Technique (MAD) with Synchrotron Radiation (SR)
</p>
<p>The above treatment shows that, in principle, the phase ambiguity associated with the SIR
</p>
<p>technique can be resolved by incorporating anomalous dispersion measurements. Conventional
</p>
<p>laboratories are usually equipped with either sealed-tube or rotating-anode X-ray sources
</p>
<p>generating X-radiation from a copper target. Anomalous scattering effects for Cu Ka radiation are
</p>
<p>quite small, Table 7.7, the actual differences between jF(hkl)j and jF&eth;�h �k �l&THORN;j being difficult to detect
without extremely careful measurements. This practical limitation to the method may be overcome by
</p>
<p>the use of synchrotron radiation. Optimization of anomalous scattering information can be achieved by
</p>
<p>selection of a wavelength close to an absorption edge of the heavy atom, where Df 00 is a maximum. It
should be remembered, however, that anomalous differences are still small, even for measurements
</p>
<p>made for wavelengths tuned in this way.
</p>
<p>There is another advantage to be gained by the availability of SR radiation, namely that measure-
</p>
<p>ments can be made at different wavelengths, possibly even on the same crystal specimen. The second
</p>
<p>wavelength should be selected such that f 0 is large and f 00 is small for the anomalous scattering
species. Measurements for this wavelength would be made only for hkl reflections. This technique is
</p>
<p>known as the multiple-wavelength anomalous dispersion (MAD) method for phasing. The measure-
</p>
<p>ments required are jFP(hkl)j, jFPH&eth;&thorn;&THORN;jl1 , jFPH&eth;�&THORN;jl2 , and jFPH&eth;&thorn;&THORN;jl2 ; Fig. 7.47 shows an idealized
phase diagram for a typical hkl reflection using the MAD technique.
</p>
<p>Table 7.7 Values of the real (dispersion) Df 0 and imaginary (absorption) Df 00 components of anomalous scattering for
Cu Ka X-rays
</p>
<p>Atomic number
</p>
<p>Cu Ka radiation, l &frac14; 1.5418 Å
Df 0 Df 00
</p>
<p>(sin y)/l &frac14; 0 &frac14;0.6 (sin y)/l &frac14; 0 &frac14;0.4 ma
</p>
<p>C 6 0 0 0 0 4.6
</p>
<p>N 7 0 0 0 0 7.52
</p>
<p>O 8 0 0 0.1 0.1 11.5
</p>
<p>S 16 0.3 0.3 0.6 0.6 89.1
</p>
<p>Fe 26 �1.1 �1.1 3.4 3.3 308
Zn 30 �1.7 �1.7 0.8 0.7 60.3
Pd 46 �0.5 �0.6 4.3 4.1 206
Ag 47 �0.5 �0.6 4.7 4.5 218
I 53 �1.1 �1.3 7.2 6.9 294
Sm 62 �6.6 �6.7 13.3 12.8 397
Gd 64 �12 �12 12.0 11.6 439
Lu 71 �7 �7 5 5 153
Pt 78 �5 �5 8 7 200
Au 79 �5 �5 8 8 208
Hg 80 �5 �5 9 8 216
Pb 82 �4 �5 10 9 232
U 92 �4 �5 16 16 306
The atoms selected are those found in proteins and those used frequently for heavy-atom derivatives
am is the absorption coefficient (see Sect. 3.1.3)
</p>
<p>7.6 Anomalous Scattering 337</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec4_3">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec4_3</a></div>
</div>
<div class="page"><p/>
<p>7.7 Charge Flipping
</p>
<p>The determination of good phases may sometimes be achieved by a process known as charge flipping,
</p>
<p>which is based on an algorithm by Oszlányi and S&euro;ut&euro;u [31]. The process requires a set of experimental
</p>
<p>jFoj values, but no other prior information about the structure at this stage. It can be applied to both
classical crystals and the aperiodic structures that were discussed briefly in Sect. 1.4.3, and with either
</p>
<p>single crystal or powder specimens.
</p>
<p>It begins by assigning random phases to all values of jFoj in the data set. A Fourier transform of this
phased data produces an electron density map r1(xyz) which will contain both positive and negative
</p>
<p>regions of density, and will most probably have little if any resemblance to a chemical structure.
</p>
<p>Then, a new density function r2(xyz) is derived, by inverting the signs of all density less than a chosen
</p>
<p>positive value, D; this is the charge flipping stage. The new function r2(xyz) is transformed to give jFcj
values, which are discarded, and a set of phases. This set of phases is combined with the experimen-
</p>
<p>tally determined jFoj values so as to create r3(xyz). Again, the signs are reversed, and the iterative
process is continued until a sensible chemical structure is obtained. The process is shown diagram-
</p>
<p>matically in Fig. 7.48.
</p>
<p>Fig. 7.47 Multiple-wavelength anomalous dispersion (MAD) phasing for wavelengths l1 and l2. The circles based on
l1 for jFPH(+)j and jFPH(�)j are drawn as before. Adding the measurement at l2 of jFPH&eth;&thorn;&THORN;jl2 is comparable to the
information from a second heavy-atom derivative in the MIR technique. OK+ is the real component and K+D+ the
</p>
<p>imaginary component of OD+, the direction FH&eth;&thorn;&THORN;l2 . For clarity the FP circle is not shown. FP gives the MAD solution
for the protein structure factor. The measurements are sometimes made only for the derivative crystal, using first l1 and
</p>
<p>then l2, during the same experimental session at the synchrotron radiation station. In such a case, the jFPj data may not
be available, whereupon the phasing would be carried out for jFPHj instead. The ensuing electron density map would
then apply to the derivative crystal
</p>
<p>338 7 Fourier Techniques in X-Ray Structure Determination</p>
<p/>
</div>
<div class="page"><p/>
<p>The convergence of the process may be judged in the usual way, by both the chemical sense of the
</p>
<p>electron density map and the R factor. The procedure has a single free parameter, namely the charge
</p>
<p>flipping threshold, D, and no structural or symmetry information is needed. A number of structures
</p>
<p>has been solved by this technique [32&ndash;36], and there is an interesting applet available [33] which
</p>
<p>allows an interactive appreciation of the process.
</p>
<p>In order to complete the structure determination, it is, of course, necessary to know the unit cell
</p>
<p>data and space group, since the structure will need to be refined by least squares in the usual way. It
</p>
<p>may be required also to have recourse to normal Fourier methods, if the phases produced by charge
</p>
<p>flipping do not lead to all atoms in the molecule. The charge flipping procedure is another useful tool
</p>
<p>in the crystallographer&rsquo;s armoury, and is also useful in studying incommensurate structures, such as
</p>
<p>those based on interpenetrating periodic lattices, and aperiodic structures [37]. The algorithm is
</p>
<p>incorporated into the SHELX and WinGX (Superflip) program systems.
</p>
<p>7.8 Location of Hydrogen Atoms
</p>
<p>In other sections of this book we have commented on the location of hydrogen atoms: by difference
</p>
<p>Fourier synthesis, Sect. 8.4.5, from molecular geometry, Sect. 9.2.4, and by neutron diffraction, Sect.
</p>
<p>11.7.2. Here we consider a method by which hydrogen atoms can be located in hydrated ionic crystal
</p>
<p>structures from calculations of electrostatic energy.
</p>
<p>The location of the positions of hydrogen atoms in crystalline hydrates can sometimes be carried
</p>
<p>out by a calculation based on electrostatic energy [38]. The coordinates at the positive end of the
</p>
<p>dipole in a water molecule may be located by a vector sum of the weighted electrical field strengths
</p>
<p>around the oxygen atom in the water molecule. All atoms less than 4 Å from the oxygen atom are
</p>
<p>included in calculating x0 from (7.76); the O&ndash;H bond distance chosen was 0.99 Å and the H&ndash;O&ndash;H
angle 104.5�.
</p>
<p>x0 &frac14;
X
</p>
<p>i
</p>
<p>&eth;Zixi=d2i &THORN; (7.76)
</p>
<p>where xi is x-coordinate of the ith atom in the structure, distant di from the oxygen atom, and Zi is its
</p>
<p>charge including the sign. For the ions in the structure, their formal charges were used; for the oxygen
</p>
<p>Fig. 7.48 Iterative process
for phasing by charge
</p>
<p>flipping (FFT &frac14; fast
Fourier transform)
</p>
<p>7.8 Location of Hydrogen Atoms 339</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec34_8">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec34_8</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec6_9">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec6_9</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec18_11">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec18_11</a></div>
</div>
<div class="page"><p/>
<p>and hydrogen atoms, theoretical calculations on a water molecule gave charges of�0.310 and +0.155
for ZO and ZH, respectively. Results for y
</p>
<p>0 and z0 follow similarly.
A number of possible positions for each hydrogen atom in the water molecule was generated in
</p>
<p>steps f around the point x0, y0, z0, such that all O&ndash;H distances were 0.99 Å. Madelung constants were
calculated [39] for the complete structure with all possible hydrogen atoms locations and plotted
</p>
<p>against the step angle f. The positions chosen for the hydrogen atoms were those corresponding to
</p>
<p>minimum electrostatic energy; about 90% of the energy is electrostatic in ionic hydrate structures.
</p>
<p>Results were obtained for barium chloride dihydrate, Table 7.8, and sodium bromide dihydrate,
</p>
<p>Table 7.9.
</p>
<p>This procedure can be useful for structures in which heavy atoms are present, because the electrostatic
</p>
<p>contribution of a hydrogen atom in relation to the heavy atoms is significant. Subsequent to the location of
</p>
<p>the hydrogen atoms, their positions can be refined by least squares.
</p>
<p>7.9 Problems
</p>
<p>7.1. A structure with the apparent space group P21/c consists of atoms at 0.2,
1
4
, 0.1 and the
</p>
<p>symmetry-related positions; the center of symmetry is at the origin. Evaluate the geometric
</p>
<p>structure factor for the four general equivalent positions in the unit cell, and, hence, determine
</p>
<p>the systematic absences among the hkl reflections. What are the consequences of these absences
</p>
<p>as far as the true structure is concerned? Sketch the structure in projection along b. What is the
</p>
<p>true space group?
</p>
<p>7.2. Rh2B crystallizes in space group Pnma with a &frac14; 5.42 Å, b &frac14; 3.98 Å, c &frac14; 7.44 Å, and Z &frac14; 4.
Consider Fig. 2.36: show that if no two Rh atoms may approach within 2.5 Å of each other, they
</p>
<p>cannot lie in general positions. Where could the Rh atoms be placed? Illustrate your answer
</p>
<p>with a sketch showing possible positions for these atoms in projection on (010).
</p>
<p>Table 7.8 Hydrogen atom positions in BaCl2&middot;2H2O
</p>
<p>Experimental By (7.76)
</p>
<p>x y z x y z
</p>
<p>H11 0.3534 0.0616 0.8824 0.381 0.064 0.874
</p>
<p>H12 0.4756 0.1255 0.0402 0.440 0.119 0.061
</p>
<p>H21 0.0977 0.1368 0.5967 0.082 0.154 0.586
</p>
<p>H22 0.2466 0.0654 0.4877 0.294 0.091 0.573
</p>
<p>Table 7.9 Hydrogen atom positions in NaBr&middot;2H2O
</p>
<p>Experimental By (7.76)
</p>
<p>x y z x y z
</p>
<p>H11 0.269 0.029 0.635 0.298 0.052 0.657
</p>
<p>H12 0.129 �0.017 0.765 0.124 �0.001 0.744
H21 0.861 0.221 0.725 0.804 0.222 0.753
</p>
<p>H22 0.741 0.098 0.792 0.772 0.101 0.882
</p>
<p>340 7 Fourier Techniques in X-Ray Structure Determination</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Fig36_2">http://dx.doi.org/10.1007/978-1-4614-3954-7#Fig36_2</a></div>
</div>
<div class="page"><p/>
<p>7.3. Trimethylammonium chloride,
</p>
<p>H3C
</p>
<p>n
H3C ��N��H
</p>
<p>=
H3C
</p>
<p>2
66664
</p>
<p>3
77775
</p>
<p>&thorn;
</p>
<p>Cl�
</p>
<p>crystallizes in a monoclinic, centrosymmetric space group, with a &frac14; 6.09 Å, b &frac14; 7.03 Å,
c &frac14; 7.03 Å, b &frac14; 95.73�, and Z &frac14; 2. The only limiting condition is 0k0: k &frac14; 2n. What is the
space group? Comment on the probable positions of (a) Cl, (b) C, (c) N, and (d) H atoms.
</p>
<p>7.4. Potassium hexachloroplatinate(IV), K2 [PtCl6], is cubic, with a &frac14; 9.755 Å. The atomic positions
are as follows (Z &frac14; 4):
</p>
<p>&eth;0; 0; 0; 0; 1
2
; 1
2
; 1
</p>
<p>2
; 0; 1
</p>
<p>2
; 1
</p>
<p>2
; 1
2
; 0&THORN; &thorn;
</p>
<p>4 Pt 0, 0, 0
</p>
<p>8 K 1
4
, 1
4
, 1
4
; 3
4
, 3
4
, 3
4
</p>
<p>24 Cl �{x, 0, 0; 0, x, 0; 0, 0, x}
</p>
<p>Show that jFc(hhh)j &frac14; A0(hhh), where
</p>
<p>A0&eth;hhh&THORN; &frac14; 4gPt &thorn; 8gK cos&eth;3ph=2&THORN; &thorn; 24gCl cos 2phxCl
</p>
<p>Calculate jFc(hhh)j for the values of h tabulated below, with xCl &frac14; 0.23 and 0.24. ObtainR factors
for the scaled Fo data for the two values of xCl, and indicate which value of xCl is the more
</p>
<p>acceptable. Calculate the Pt&ndash;Cl distance, and sketch the [PtCl6]
�2 ion. What is the point group of
</p>
<p>this species?
</p>
<p>hkl 111 222 333
</p>
<p>Fo 491 223 281
</p>
<p>gPt 73.5 66.5 59.5
</p>
<p>gK 17.5 14.5 12.0
</p>
<p>gCl 15.5 13.0 10.5
</p>
<p>Atomic scattering factors gj may be taken to be temperature-corrected values.
</p>
<p>7.5. USi crystallizes in space group Pbnm, with a &frac14; 5.65 Å, b &frac14; 7.65 Å, c &frac14; 3.90 Å, and Z &frac14; 4.
The U atoms lie at the positions
</p>
<p>� fx; y; 1
4
; 1
</p>
<p>2
� x; 1
</p>
<p>2
&thorn; y; 1
</p>
<p>4
g
</p>
<p>Obtain a simplified expression for the geometric structure factor (�1 at 0, 0, 0) for the U atoms.
</p>
<p>From the data below, determine approximate values for xU and yU; the Si contributions may be
</p>
<p>neglected.
</p>
<p>hkl 200 111 210 231 040 101 021 310
</p>
<p>Io(hkl) 0 236 251 200 0 170 177 0
</p>
<p>Proceed by using 200 to find a probable value for xU. Then find yU from 111, 231, and 040.
</p>
<p>7.9 Problems 341</p>
<p/>
</div>
<div class="page"><p/>
<p>7.6. Methylamine forms a complex with boron trifluoride of composition CH3NH2BF3.
</p>
<p>Crystal Data
</p>
<p>System: monoclinic
</p>
<p>Unit-cell dimensions: a &frac14; 5.06 Å, b &frac14; 7.28 Å, c &frac14; 5.81 Å, b &frac14; 101.5�
Vc: 209.7 Å
</p>
<p>3
</p>
<p>Dm: 1.54 g cm
�3
</p>
<p>Mr: 98.86
</p>
<p>Z: 1.97 or 2 to the nearest integer
</p>
<p>Unit-cell contents: 2C, 10H, 2N, 2B, and 6F atoms
</p>
<p>Absent spectra: 0k0 : k &frac14; 2n + 1
Possible space groups: P21 or P21/m (P21/m may be assumed)
</p>
<p>Determine what you can about the crystal structure.
</p>
<p>7.7. Write the symmetry-equivalent amplitudes of jF(hkl)j, jF(0kl)j, and jF(h0l)j in (a) the triclinic,
(b) the monoclinic, and (c) the orthorhombic crystal systems; Friedel&rsquo;s law may be assumed.
</p>
<p>7.8. (a) Determine the orientations of the Harker lines and sections in Pa, P2/a, and P2221.
</p>
<p>(b) A monoclinic, non-centrosymmetric crystal with a primitive space group shows concentra-
</p>
<p>tions of peaks on (u, 0, w) and [0, v, 0]. How might this situation arise?
</p>
<p>7.9. Diphenyl sulfoxide, (C6H5)2SO, is monoclinic, with a &frac14; 8.90 Å, b &frac14; 14.08 Å, c &frac14; 8.32 Å,
b &frac14; 101.12�, and Z &frac14; 4. The conditions limiting possible X-ray reflections are as follows.
</p>
<p>hkl : none; h0l : h&thorn; l &frac14; 2n;
0k0 : k &frac14; 2n
</p>
<p>(a) What is the space group?
</p>
<p>(b) Figure P7.1a&ndash;c are Patterson sections at v &frac14; 1
2
, 0.092, and 0.408, respectively, and
</p>
<p>contain S&ndash;S vector peaks. Write the coordinates of the non-origin S&ndash;S vectors in terms
</p>
<p>of x, y, and z, and from the sections provided determine the best values for the S atoms in
</p>
<p>the unit cell. Plot these atomic positions as seen along the b axis, with an indication of the
</p>
<p>heights of the atoms with respect to the plane of the diagram.
</p>
<p>7.10. Figure P7.2 shows an idealized u, w vector set for a hypothetical structure C6H5S in space group
</p>
<p>P2 with Z &frac14; 2, projected down the b axis. Only the S&ndash;S and S&ndash;C vector interactions are
considered.
</p>
<p>(a) Determine the x and z coordinates for the S atoms and plot them to the scale of this
</p>
<p>projection.
</p>
<p>(b) Use the Patterson superposition method to locate the carbon atom positions on a map of the
</p>
<p>same projection.
</p>
<p>7.11. Hafnium disilicide, HfSi2, is orthorhombic, with a &frac14; 3.677 Å, b &frac14; 14.55 Å, c &frac14; 3.649 Å, and
Z &frac14; 4. The space group is Cmcm, and the Hf and Si atoms occupy three sets of special positions
of the type
</p>
<p>� f0; y; 1
4
; 1
</p>
<p>2
; 1
2
&thorn; y; 1
</p>
<p>4
g
</p>
<p>The contributions from the Hf atoms dominate the structure factors. By combining the terms
</p>
<p>cos2pky and cos 2p(ky + k/2), show that the geometric structure factor A(0k0) is approximately
</p>
<p>proportional to cos2pyHf. The Fo(0k0) data are listed below, from which the values of Fo(0k0)
2,
</p>
<p>divided by 10 and rounded to the nearest integer, have been derived.
</p>
<p>342 7 Fourier Techniques in X-Ray Structure Determination</p>
<p/>
</div>
<div class="page"><p/>
<p>Fig. P7.1 Patterson
</p>
<p>sections at (a) v &frac14; 1
2
, (b)
</p>
<p>v &frac14; 0.092, (c) v &frac14; 0.408
</p>
<p>7.9 Problems 343</p>
<p/>
</div>
<div class="page"><p/>
<p>0k0 020 040 060 080 010,0 012,0 014,0 016,0
</p>
<p>Fo(0k0) 7 14 18 13 12 &lt;1 20 &lt;1
</p>
<p>Fo(0k0)
2 5 20 32 17 14 0 40 0
</p>
<p>(a) Calculate the one-dimensional Patterson function P(v), using the equation
</p>
<p>P&eth;v&THORN; /
X
</p>
<p>k
</p>
<p>Fo&eth;0k0&THORN;2 cos 2pkv
</p>
<p>The multiplying factor 2/b and the F(0) term have been omitted8 to simplify the calculation;
</p>
<p>they can never change the form of the synthesized function, although the neglect of the term
</p>
<p>involving F(0) gives rise to negative values in the calculated P(v). The Fourier summation
</p>
<p>here can be carried out readily by means of the program FOUR1D, which also gives a plot
</p>
<p>of the function at the online printer.
</p>
<p>Fig. P7.2 Idealized Vector map for C6H4S
</p>
<p>8These omissions give rise to the proportionality sign.
</p>
<p>344 7 Fourier Techniques in X-Ray Structure Determination</p>
<p/>
</div>
<div class="page"><p/>
<p>However, for many years following their introduction in 1936, the summation aid known
</p>
<p>as Beevers&ndash;Lipson9 strips was used for Fourier summations in one and two dimensions, and
</p>
<p>even Harker sections in three dimensions. For historical interest, Table P7.1 shows the strips
</p>
<p>that would be used for this summation. Each line contains the value of the positive
</p>
<p>coefficients &eth;F2o=10&THORN; cos 2ph&eth;n=60&THORN; for n &frac14; 0 to 15/60; for negative jFj the sign of all
terms for n &frac14; 0&ndash;15 are changed. For reflection symmetry at 1
</p>
<p>4
, only the values of n from
</p>
<p>0 to 15 are needed. The columns are added vertically to give the sum over h at each value of
</p>
<p>n. The range of n can be increased by making use of the properties of the cosine function.
</p>
<p>Table P7.2 shows clearly how an error is of greater consequence in a phase (sign) than in
</p>
<p>an amplitude. Consider changing the sign of 080 in the calculation of r(y) and then the
</p>
<p>amplitude by, say, 20%.
</p>
<p>Plot the function, extend it to one repeat unit, interpret the four highest non-origin peaks,
</p>
<p>and determine yHf.
</p>
<p>(b) Use the value of yHf and the form of the geometric structure factor A(0k0) to determine the
</p>
<p>signs for the 0k0 reflections. Hence, compute the electron density:
</p>
<p>r&eth;y&THORN; /
X
</p>
<p>k
</p>
<p>�Fo&eth;0k0&THORN;j cos 2pky
</p>
<p>Again the 2/b factor and F(0) have been omitted. Plot the function and determine yHf.
</p>
<p>What can be deduced about the positions of the Si atoms? In the light of your results,
</p>
<p>study P(v) again. Table P7.2 contains the simulated, relevant Beevers&ndash;Lipson strips, with
</p>
<p>positive values of jFj.
</p>
<p>Table P7.1 Simulated Beevers-Lipson strips appropriate to problem 7.11
</p>
<p>Amplitude Index 0
60
</p>
<p>1
60
</p>
<p>2
60
</p>
<p>3
60
</p>
<p>4
60
</p>
<p>5
60
</p>
<p>6
60
</p>
<p>7
60
</p>
<p>8
60
</p>
<p>9
60
</p>
<p>10
60
</p>
<p>11
60
</p>
<p>12
60
</p>
<p>13
60
</p>
<p>14
60
</p>
<p>15
60
</p>
<p>Add columns for
P
</p>
<p>k 5
</p>
<p>20
</p>
<p>32
</p>
<p>17
</p>
<p>14
</p>
<p>40
</p>
<p>2
</p>
<p>4
</p>
<p>6
</p>
<p>8
</p>
<p>10
</p>
<p>14
</p>
<p>5
</p>
<p>20
</p>
<p>32
</p>
<p>17
</p>
<p>14
</p>
<p>40
</p>
<p>128
</p>
<p>5
</p>
<p>18
</p>
<p>26
</p>
<p>11
</p>
<p>7
</p>
<p>4
</p>
<p>71
</p>
<p>5
</p>
<p>13
</p>
<p>10
�2
�7
39
</p>
<p>20:::
</p>
<p>4
</p>
<p>6
</p>
<p>10
</p>
<p>14
</p>
<p>14
</p>
<p>12
</p>
<p>3
�2
26
</p>
<p>17
�7
37
</p>
<p>2
</p>
<p>10
</p>
<p>32
</p>
<p>8
</p>
<p>7
</p>
<p>20
</p>
<p>2
</p>
<p>16
</p>
<p>26
</p>
<p>5
</p>
<p>14
</p>
<p>32
</p>
<p>1
</p>
<p>20
</p>
<p>10
</p>
<p>16
</p>
<p>7
</p>
<p>27
</p>
<p>�1
20
</p>
<p>10
</p>
<p>16
</p>
<p>7
</p>
<p>27
</p>
<p>�2
16
</p>
<p>26
</p>
<p>5
</p>
<p>14
</p>
<p>32
</p>
<p>�2
10
</p>
<p>32
</p>
<p>8
</p>
<p>7
</p>
<p>20
</p>
<p>�3
2
</p>
<p>26
</p>
<p>17
</p>
<p>7
</p>
<p>32
</p>
<p>�4
6
</p>
<p>10
</p>
<p>14
</p>
<p>14
</p>
<p>12
</p>
<p>�5
13
</p>
<p>10
</p>
<p>2
</p>
<p>7
</p>
<p>39
</p>
<p>�5
18
</p>
<p>26
</p>
<p>11
</p>
<p>7
</p>
<p>4
</p>
<p>�5
20
</p>
<p>32
</p>
<p>17
</p>
<p>14
</p>
<p>40
</p>
<p>Table P7.2 Simulated Beevers&ndash;Lipson strips
</p>
<p>Amplitude Index 0
60
</p>
<p>1
60
</p>
<p>2
60
</p>
<p>3
60
</p>
<p>4
60
</p>
<p>5
60
</p>
<p>6
60
</p>
<p>7
60
</p>
<p>8
60
</p>
<p>9
60
</p>
<p>10
60
</p>
<p>11
60
</p>
<p>12
60
</p>
<p>13
60
</p>
<p>14
60
</p>
<p>15
60
</p>
<p>7 2 7 7 6 6 5 3 2 1 �1 �2 �3 �5 �6 �6 �7 �7
</p>
<p>14 4 14 13 9 4 �1 �7 11 14 14 11 �7 �1 4 9 13 14
</p>
<p>18 6 18 15 6 �6 15 18 15 �6 6 15 18 15 6 �6 15 18
</p>
<p>13 8 13 9 �1 11 13 �6 4 12 12 4 �6 13 11 �1 9 13
</p>
<p>12 10 12 6 �6 12 �6 6 12 6 �6 12 �6 6 12 6 �6 12
</p>
<p>20 14 20 2 20 �6 18 10 16 13 13 16 10 18 6 20 �2 20
</p>
<p>9Lipson H, Beevers CA (1936) Proc Phys Soc 48:772.
</p>
<p>7.9 Problems 345</p>
<p/>
</div>
<div class="page"><p/>
<p>7.12. The alums, MAl(NO4)2&middot;12H2O, where M &frac14; NH4, K, Rb, Tl, and N &frac14; S, Se, are isomorphous.
They crystallize in the cubic centrosymmetric space group Pa3, with the unit-cell side a in the
</p>
<p>range 12.2&ndash;12.4 Å and Z &frac14; 4.
A symmetry analysis leads to the following atomic positions:
</p>
<p>4M 0, 0, 0; 0,
1
</p>
<p>2
,
1
</p>
<p>2
;
1
</p>
<p>2
, 0,
</p>
<p>1
</p>
<p>2
;
1
</p>
<p>2
,
1
</p>
<p>2
, 0
</p>
<p>4Al
1
</p>
<p>2
,
1
</p>
<p>2
,
1
</p>
<p>2
;
1
</p>
<p>2
, 0, 0; 0,
</p>
<p>1
</p>
<p>2
, 0; 0, 0,
</p>
<p>1
</p>
<p>2
</p>
<p>8N � x; x; x; 1
2
&thorn; x; 1
</p>
<p>2
� x; x; �x; 1
</p>
<p>2
&thorn; x; 1
</p>
<p>2
� x; 1
</p>
<p>2
� x; �x; 1
</p>
<p>2
&thorn; x
</p>
<p>� �
</p>
<p>The N atoms lie on cube diagonals, and xN may be obtained by a one-dimensional Fourier
</p>
<p>synthesis along the line [111], using F(hhh) data. Table P7.3 lists these data for four alums
</p>
<p>(N &frac14; S). Tl may be assumed to be sufficiently heavy to make all Fo values positive in this
derivative. The same sites in each crystal are occupied by the replaceable atoms.
</p>
<p>(a) Use the isomorphous replacement technique to determine the signs of the reflections in
</p>
<p>Table P7.3.
</p>
<p>(b) Compute r[111] for K alum, using the following equation:
</p>
<p>r&eth;D&THORN; /
X
</p>
<p>h
</p>
<p>�Fo&eth;hhh&THORN; cos 2phD
</p>
<p>where D is the sampling interval along [111]. Plot the function and determine a probable
</p>
<p>value for ds.
</p>
<p>(c) The corresponding hhh data for the isomorphous K/Se alum are listed below. The signs
</p>
<p>have been allocated by a similar isomorphous replacement procedure. Calculate and plot
</p>
<p>r(D) for these data. Compare the two electron density plots and comment upon the results.
</p>
<p>hkl 111 222 333 444 555 666 777 888
</p>
<p>Fo �48 �52 64 0 116 100 �16 0
</p>
<p>7.13. A crystal contains five atoms per unit cell. Four of them contribute together 100eif to F(010).
</p>
<p>The fifth atom has fractional coordinates 0.00, 0.10, 0.00, and its atomic scattering factor
</p>
<p>components fo, Df
0, and Df00 are 52.2, �2.7, and 8.0, respectively. If f &frac14; 60�, determine,
</p>
<p>graphically or otherwise, jF(010)j, jF&eth;0�10&THORN;j, f(010), and f&eth;0�10&THORN;.
7.14. A protein crystal structure is to be solved usingMIR. Three isomorphous derivatives are prepared
</p>
<p>using platinum, uranium, and iodine compounds. For the reflection 060, the following measure-
</p>
<p>ments were recorded:
</p>
<p>Table P7.3 Fo(hhh) for isomorphous alums
</p>
<p>hkl NH&thorn;4 (10 electrons) K
+ (18 electrons) Rb+ (36 electrons) Tl+ (80 electrons)
</p>
<p>111 86 38 19 113
</p>
<p>222 0 19 79 195
</p>
<p>333 111 125 158 236
</p>
<p>444 25 6 55 125
</p>
<p>555 24 49 64 131
</p>
<p>666 86 86 122 164
</p>
<p>777 53 34 0 18
</p>
<p>888 0 16 22 56
</p>
<p>346 7 Fourier Techniques in X-Ray Structure Determination</p>
<p/>
</div>
<div class="page"><p/>
<p>For protein: jFPj &frac14; 858
For Pt derivative: jFPH1 j &frac14; 756, jFH1 j &frac14; 141, fH1 &frac14; 78�
U derivative: jFPH2 j &frac14; 856, jFH2 j &frac14; 154, fH2 &frac14; 63�
I derivative: jFPH3 j &frac14; 940, jFH3 j &frac14; 100, fH3 &frac14; 146�
Use a Harker construction to obtain an estimate for fP for this reflection from the native protein
</p>
<p>crystal.
</p>
<p>7.15. Consider (7.19), reproduced here:
</p>
<p>C&eth;x&THORN; &frac14; p
2
&thorn; 2
</p>
<p>X1
</p>
<p>h&frac14;1
</p>
<p>1
</p>
<p>h
sin hx
</p>
<p>Show that sin hx can be replaced by cos (hx � f), where f &frac14; p/2.
7.16. A protein with molecular weight 18000 Da crystallizes in space group C2 with unit-cell
</p>
<p>dimensions a &frac14; 40 Å, b &frac14; 50 Å, c &frac14; 60 Å, b &frac14; 100�.
(a) Estimate the number of protein molecules per unit cell if there are equal masses of protein
</p>
<p>and solvent in the unit cell (take MH as 1.66 � 10�24 g).
(b) What symmetry would the protein molecule need to adopt in the crystalline state?
</p>
<p>7.17. A non-centrosymmetric structure is composed entirely of NH identical heavy atom anomalous
</p>
<p>scattering species per unit cell; it may be assumed here that no normal scattering species is present.
</p>
<p>Show, graphically or otherwise, that jF&eth;hkl&THORN;j &frac14; jF&eth;�h �k �l&THORN;j but that f&eth;hkl&THORN; 6&frac14; �f&eth;�h �k �l&THORN;.
7.18. A centrosymmetric structure contains a mixture of anomalous and normal scattering atoms.
</p>
<p>Show that jF&eth;hkl&THORN;j2 &frac14; jF&eth;�h �k �l&THORN;j2 and f&eth;hkl&THORN; &frac14; f&eth;�h �k �l&THORN; 6&frac14; 0 or p.
7.19. Given the jF(hkl)j equivalents for each of the following space groups, list corresponding
</p>
<p>Bijvoet pairs:
</p>
<p>(a) C2: jF&eth;hkl&THORN;j &frac14; jF&eth;�hk�l&THORN;j &frac14; jF&eth;h�kl&THORN;j &frac14; jF&eth;�h �k �l&THORN;j
(b) Pm: jF&eth;hkl&THORN;j &frac14; jF&eth;�hk�l&THORN;j &frac14; jF&eth;h�kl&THORN;j &frac14; jF&eth;�h �k �l&THORN;j
(c) P212121:
</p>
<p>jF&eth;hkl&THORN;j &frac14;jF&eth;�hkl&THORN;j&frac14;jF&eth;h�kl&THORN;j&frac14;jF&eth;hk�l&THORN;j &frac14; jF &eth;�hk�l&THORN; j &frac14; jF &eth;h�k �l&THORN;j &frac14; jF&eth;�h �k l&THORN;j &frac14; jF&eth;�h �k �l&THORN;j
(d) P4:
</p>
<p>jF&eth;hkl&THORN;j &frac14; jF&eth;�khl&THORN;j &frac14; jF&eth;�h �k l&THORN;j &frac14; jF&eth;k�hl&THORN;j &frac14; jF&eth;k �h �l&THORN;j &frac14; jF&eth;hk�l&THORN;j &frac14; jF&eth;�kh�l&THORN;j &frac14; jF&eth;�h �k �l&THORN;j
7.20. X-ray intensity data are to be measured for an orthorhombic crystalline protein in three stages:
</p>
<p>(a) For 0 &lt; y &lt; 10�, both I(hkl) and I&eth;�h �k �l&THORN; symmetry-equivalent reflections are measured
(b) For 10 &lt; y &lt; 20�, I(hkl) alone are measured
(c) For 20 &lt; y &lt; 25�, I(hkl) alone are measured
If the unit cell is primitive, with a &frac14; 30 Å, b &frac14; 50 Å and c &frac14; 40 Å, and the X-ray wavelength is
1.5 Å, estimate the number of reflections measured and the corresponding resolution for the three
</p>
<p>data sets.
</p>
<p>References
</p>
<p>1. Reynolds CD et al (1974) J Cryst Mol Struct 4:213
</p>
<p>2. Cruickshank DWJ (1957) Acta Crystallogr 10:504 [This study used the non-standard space group P21/a, equiva-
</p>
<p>lent to P21/c, with a and c interchanged and b changed to �b]
3. Keeling RO (1957) Acta Crystallogr 10:209
</p>
<p>3a. Glusker JP et al (1968) Acta Crystallogr B24:359
</p>
<p>4. Ladd M, Palmer R (2003) Structure determination by X-ray crystallography, 4th edn. Kluwer Academic/Plenum,
</p>
<p>New York
</p>
<p>5. Palmer HT, Palmer RA (1969) Acta Crystallogr B25:1090
</p>
<p>6. Duwell EJ, Baenziger NC (1995) Acta Crystallogr 8:705
</p>
<p>References 347</p>
<p/>
</div>
<div class="page"><p/>
<p>7. Woolfson MM (1956) Acta Crystallogr 9:804
</p>
<p>8. Sim GA (1960) Acta Crystallogr 13:511
</p>
<p>9. Br&euro;unger AT (1992) Nature 355:472
10. Phenix Tutorial8. http://www.phenix-online.org/documentation/refinement-examples.htm
</p>
<p>11. Wilson AJC (1976) Acta Crystallogr A32:994
</p>
<p>12. Cruickshank DWJ (1949) Acta Crystallogr 2:154
</p>
<p>13. idem. ibid. (1999) D55:583
</p>
<p>14. Blow DM (2002) Acta Crystallogr D58:792
</p>
<p>15. Ladd MFC (1967) Z Kristallogr 124:64
</p>
<p>16. Wilson AJC (1950) Acta Crystallogr 3:397&ndash;398
</p>
<p>17. Rossman MG, Blow DM (1961) Acta Crystallogr 14:641
</p>
<p>18. Blow DM, Crick FHC (1959) Acta Crystallogr 12:794
</p>
<p>19. Dickerson RE, Weintzierl JE, Palmer RA (1968) Acta Crystallogr B24:997
</p>
<p>20. Hamilton WC (1965) Acta Crystallogr 18:502
</p>
<p>21. Rogers D (1981) Acta Crystallogr A37:734
</p>
<p>22. Flack HD (1983) Acta Crystallogr A39:876
</p>
<p>23. http://en.wikipedia.org/wiki/Stereochemistry
</p>
<p>24. Palmer RA, Potter BS, Palmer HT, Frampton CS (2007) J Chem Cystallogr 37:1
</p>
<p>25. Palmer RA, Potter BS, Leach MJ, Jenkins TC, Chowdhry BZ (2010) Med Chem Commun 1:45
</p>
<p>26. Barnes CL (1997) J Appl Crystallogr 30:568 [Based on ORTEP-III (v 1.0.3) by CK Johnson and MN Burnett]
</p>
<p>27. Merrit EA, Bacon DJ (1997) Methods Enzymol 277:505 [Implemented in WinGX (qv)]
</p>
<p>28. Tanczos AC, Palmer RA, Potter BS, Saldanha JW, Howlin BJ (2004) Comput Biol Chem 28(5&ndash;6):375
</p>
<p>29. Spencer J et al (2012) CrystEngComm 14(20), 6441
</p>
<p>30. Peerdeman AF, Bijvoet JM (1956) Acta Crystallogr 9:1012
</p>
<p>31. Oszlányi G and S&euro;ut&euro;u A (2004) Acta Cryst A60:134
32. Palatinus L (2004) Acta Cryst A60:604
</p>
<p>33. http://escher.epfl.ch/eCrystallography/ (Sect. 7.4); http://user.web.psi.ch/powder08/downloads/Hinrichsen-Charge
</p>
<p>%20flipping.pdf
</p>
<p>34. Baerlocher Ch et al (2007) Science 315:1113
</p>
<p>35. McCusker LB, Baerlocher Ch (2007) Zeit f&euro;ur Kristallogr 222:47
36. Sisak D et al (2012) J Appl Crystallogr (in press)
</p>
<p>37. Van Smaalen S (2007) Incommensurate Crystallography, Oxford University Press
</p>
<p>38. Ladd MFC (1968) Z Kristallogr 126:147
</p>
<p>39. Ladd MFC (1998) Introduction to physical chemistry, 3rd edn. Cambridge University Press, Cambridge
</p>
<p>Bibliography: Introductory Structure Analysis
</p>
<p>Bragg WL (1949) A general survey, vol I, The crystalline state. Bell, London
</p>
<p>Hahn T (ed) (2002) International tables for crystallography, vol A, 5th edn. Kluwer Academic, Dordrecht
</p>
<p>Henry NFM, Lonsdale K (eds) (1965) International tables for X-ray crystallography, vol I. Kynoch Press, Birmingham
</p>
<p>General Structure Analysis
</p>
<p>Buerger MJ (1959) Vector space. Wiley, New York
</p>
<p>Dunitz JD (1995) X-ray analysis and the structure of organic molecules. Verlag Helvetica Chimica Acta, Basel
</p>
<p>Woolfson MM (1997) An introduction to X-ray crystallography, 2nd edn. Cambridge University Press, Cambridge
</p>
<p>Protein Crystallography
</p>
<p>Blundell TL, Johnson LN (1977) Protein crystallography. Academic, New York
</p>
<p>Eisenberg D (1970) in The Enzymes, vol I, 3rd edition (ed Boyer PD), Academic, New York
</p>
<p>Phillips DC (1966) Adv Struct Res Diffraction Methods 2:75
</p>
<p>348 7 Fourier Techniques in X-Ray Structure Determination</p>
<p/>
<div class="annotation"><a href="http://www.phenix-online.org/documentation/refinement-examples.htm">http://www.phenix-online.org/documentation/refinement-examples.htm</a></div>
<div class="annotation"><a href="http://en.wikipedia.org/wiki/Stereochemistry">http://en.wikipedia.org/wiki/Stereochemistry</a></div>
<div class="annotation"><a href="http://escher.epfl.ch/eCrystallography/">http://escher.epfl.ch/eCrystallography/</a></div>
<div class="annotation"><a href="http://user.web.psi.ch/powder08/downloads/Hinrichsen-Charge%20flipping.pdf">http://user.web.psi.ch/powder08/downloads/Hinrichsen-Charge%20flipping.pdf</a></div>
<div class="annotation"><a href="http://user.web.psi.ch/powder08/downloads/Hinrichsen-Charge%20flipping.pdf">http://user.web.psi.ch/powder08/downloads/Hinrichsen-Charge%20flipping.pdf</a></div>
</div>
<div class="page"><p/>
<p>Crystallographic Computing
</p>
<p>Ahmed FR et al (eds) (1970) Crystallographic computing. Munksgaard, Copenhagen
</p>
<p>SHELX-97 manual. http://shelx.uni-ac.gwdg.de/SHELX/shelx.pdf
</p>
<p>CCP4. http://www.ccp4.ac.uk/download/downloadman.php
</p>
<p>Chemical Data
</p>
<p>Sutton LE (ed) (1958; suppl 1965) Tables of interatomic distances and configuration in molecules and ions. The
</p>
<p>Chemical Society, London
</p>
<p>References 349</p>
<p/>
<div class="annotation"><a href="http://shelx.uni-ac.gwdg.de/SHELX/shelx.pdf">http://shelx.uni-ac.gwdg.de/SHELX/shelx.pdf</a></div>
<div class="annotation"><a href="http://www.ccp4.ac.uk/download/downloadman.php">http://www.ccp4.ac.uk/download/downloadman.php</a></div>
</div>
<div class="page"><p/>
<p>Direct Methods and Refinement 8
</p>
<p>8.1 Introduction
</p>
<p>In this chapter, we consider direct methods, also known as phase probability methods, of solving the
</p>
<p>phase problem, together with Patterson search techniques, least-squares refinement, and other
</p>
<p>important procedures that are involved in the overall investigation of crystal and molecular structure.
</p>
<p>8.2 Direct Methods of Phase Determination
</p>
<p>Direct methods of solving the phase problem are an important technique, particularly in their ability
</p>
<p>to yield good phase information for structures containing no heavy atoms. One feature common to the
</p>
<p>structure-determining methods that we have encountered so far is that values for phases of X-ray
</p>
<p>reflections are derived initially by structure factor calculations, albeit on only part of the structure.
</p>
<p>Since the data from which the best phases are ultimately derived are the Fo values, we may imagine
</p>
<p>that the phases are encoded somehow in these quantities, even though their actual values are not
</p>
<p>recorded experimentally. This philosophy led to the search for analytical methods of phase determi-
</p>
<p>nation, which are independent of trial structures, and initiated the development of direct methods.
</p>
<p>8.2.1 Normalized Structure Factors
</p>
<p>An important stage in direct phase-determining formulae results by first replacing Fo(hkl) by
</p>
<p>the corresponding normalized structure factor jE(hkl)j, which is given by the equation
</p>
<p>jE&eth;hkl&THORN;j2 &frac14; K
2Fo&eth;hkl&THORN;2
</p>
<p>e
PN
</p>
<p>j&frac14;1 g
2
j
</p>
<p>(8.1)
</p>
<p>The summation in the denominator of (8.1), which is y-dependent through gj, may be obtained
</p>
<p>through a K-curve, similar to that in Fig. 7.24c(ii). The data must be on an absolute scale, Sect. 4.2.1,
</p>
<p>and e is incorporated in setting up the K-curve; see also Sects. 13.4.6 and 13.4.10.
</p>
<p>The jEj values have properties similar to those of the sharpened Fo values derived for a point-atom
model, Sect. 7.4.4: they are compensated for the fall-off in f with sin y. High-order reflections with
</p>
<p>M. Ladd and R. Palmer, Structure Determination by X-ray Crystallography:
</p>
<p>Analysis by X-rays and Neutrons, DOI 10.1007/978-1-4614-3954-7_8,
# Springer Science+Business Media New York 2013
</p>
<p>351</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Fig24_7">http://dx.doi.org/10.1007/978-1-4614-3954-7#Fig24_7</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec11_4">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec11_4</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec12_13">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec12_13</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec16_13">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec16_13</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec16_7">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec16_7</a></div>
</div>
<div class="page"><p/>
<p>comparatively small jFj values can have relatively large jEj values, an important fact in the applica-
tion of direct methods. We may note in passing that jEj2 values, or jEj jFj, can be used as coefficients
in a sharpened Patterson function; also, since jEj2 &frac14; 1, Table 8.1 and Sect. 4.2.5, the coefficients
(jEj2 � 1) produce a sharpened Patterson function with the origin peak removed. This technique is
useful because, in addition to the general sharpening effect, vectors of small magnitude that are often
</p>
<p>swamped by the origin peak may be revealed.
</p>
<p>Epsilon (e) Factor
Because of the importance of individual reflections in direct phasing methods, care must be taken to
</p>
<p>obtain the best possible jEj values. The factor e in the denominator of (8.1) takes account of the fact that
reflections in certain reciprocal lattice zones or rows may have average intensity greater than that for
</p>
<p>general reflections. The e-factor depends upon the crystal class, and its values for all crystal classes are
</p>
<p>listed in Table 4.2. Detailed considerations of the e factor and of the statistics of jEj values have been
presented in Sect. 4.2.3.
</p>
<p>Distributions of jEj Values
The distribution of jEj values holds useful information about the space group of a crystal. Theoretical
quantities derived for equal-atom structures in space groups P1 and P�1 are indicated in Table 8.1,
</p>
<p>together with the experimental results for two crystals.
</p>
<p>Crystal 1 is pyridoxal phosphate oxime dihydrate, C8H11N2O6P&middot;2H2O, which is triclinic. The
</p>
<p>values in Table 8.1 favor the centric distribution C, and the structure analysis [1] confirmed the
</p>
<p>assignment of space group P�1. Crystal 2 is a penta-uloside sugar; the results correspond, on the
</p>
<p>whole, to an acentric distribution A, as expected for a crystal of space group P212121 [2]. It should
</p>
<p>be noted that the experimentally derived quantities do not always have a completely one-to-one
</p>
<p>correspondence with the theoretical values, and care should be exercised in using these statistics to
</p>
<p>select a space group.
</p>
<p>8.2.2 Structure Invariants and Origin-Fixing Reflections
</p>
<p>The formulae used in direct phasing require, initially, the use of a few reflections with phases known,
</p>
<p>either uniquely or symbolically; we consider first centrosymmetric primitive space groups.
</p>
<p>Table 8.1 Some theoretical and experimental values related to jEj values
statistics
</p>
<p>Mean values
</p>
<p>Theoretical values
</p>
<p>Experimental values and
</p>
<p>conclusions
</p>
<p>P�1&eth;C&THORN; P1(A) Crystal 1 Crystal 2
jEj2 1.00 1.00 0.99 0.98
jEj 0.80 0.89 0.85 A/C 0.84 A/C
jjEj2 � 1j 0.97 0.74 0.91 C 0.82 A
</p>
<p>Distribution % % % %
</p>
<p>jEj &gt; 3.0 0.30 0.01 0.20 C 0.05 A
jEj &gt; 2.5 1.24 0.19 0.90 C 0.98 C
jEj &gt; 2.0 4.60 1.80 2.70 A/C 2.84 A/C
jEj &gt; 1.75 8.00 4.71 7.14 C 6.21 A/C
jEj &gt; 1.5 13.4 10.5 12.9 C 10.5 A
jEj &gt; 1.0 32.0 36.8 33.7 C 37.1 A
</p>
<p>352 8 Direct Methods and Refinement</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec15_4">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec15_4</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Tab2_4">http://dx.doi.org/10.1007/978-1-4614-3954-7#Tab2_4</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec13_4">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec13_4</a></div>
</div>
<div class="page"><p/>
<p>In this group of crystals, the origin is usually taken on the center of symmetry in the unit cell at
</p>
<p>0, 0, 0 and we speak of the sign s(hkl) of the reflection; s(hkl) is F(hkl)/jF(hkl)j and is either + or�. We
shall show that, in any primitive centrosymmetic space group, arbitrary signs can be allocated to three
</p>
<p>reflections, chosen according to certain rules, in order to specify the origin at one of the eight centers
</p>
<p>of symmetry in the unit cell. These signs form a basic set, or &ldquo;fountainhead,&rdquo; from which more signed
</p>
<p>reflections emerge as the analysis proceeds. From (3.69), remembering that we are dealing with a
</p>
<p>centrosymmetric crystal, we write
</p>
<p>F&eth;hkl&THORN;0;0;0 &frac14;
XN
</p>
<p>j&frac14;1
gj cos 2p&eth;hxj &thorn; kyj &thorn; lzj&THORN; (8.2)
</p>
<p>where F(hkl)0,0,0 indicates an origin of coordinates at the point 0, 0, 0. If this origin is moved to a
</p>
<p>center of symmetry at
1
</p>
<p>2
,
1
</p>
<p>2
, 0, the point that was originally xj, yj, zj, now becomes xj �
</p>
<p>1
</p>
<p>2
, yj �
</p>
<p>1
</p>
<p>2
, zj,
</p>
<p>Fig. 8.1, with p &frac14; q &frac14; 1
2
. The structure factor equation may now be written as
</p>
<p>F&eth;hkl&THORN;1=2;1=2;0 &frac14;
XN
</p>
<p>j&frac14;1
gj cos 2p&frac12;&eth;hxj &thorn; kyj &thorn; lzj&THORN; � &eth;h&thorn; k&THORN;=2� (8.3)
</p>
<p>Expanding the cosine term (see Web Appendix WA5), and remembering that sin[2p(h + k)/2]
</p>
<p>&frac14; sin[np] &frac14; 0, and cos[2p(h + k)/2] &frac14; cos[np] &frac14; (�1)n, we obtain
</p>
<p>F&eth;hkl&THORN;1=2;1=2;0 &frac14; &eth;�1&THORN;
h&thorn;k
</p>
<p>F&eth;hkl&THORN;0;0;0 (8.4)
</p>
<p>Equation (8.4) demonstrates that the amplitude jF(hkl)j is invariant under change of origin, as
</p>
<p>would be expected, but that a change of sign may occur, depending on the parity of the indices hkl.
</p>
<p>The complete results are presented in Table 8.2, the use of which is illustrated by the following
</p>
<p>examples.1
</p>
<p>Fig. 8.1 Transformation of the point P(x, y), with respect to two-dimensional axes, by moving the origin from O to O0
</p>
<p>(p, q); the transformed coordinates of P are (x � p, y � q)
</p>
<p>1 See also Bibliography, Ladd and Palmer (1980).
</p>
<p>8.2 Direct Methods of Phase Determination 353</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Equ69_3">http://dx.doi.org/10.1007/978-1-4614-3954-7#Equ69_3</a></div>
</div>
<div class="page"><p/>
<p>For example, the reflection 312 belongs to the odd-odd-even (ooe) parity group 7. If s(312) is given
</p>
<p>a plus sign, the origin could be regarded as being restricted to one from the following list:
</p>
<p>0; 0; 0; 0; 0; 1
2
; 1
</p>
<p>2
; 1
2
; 0; 1
</p>
<p>2
; 1
2
; 1
2
</p>
<p>Similarly, if s(322), parity group 2(oee), is also given a plus sign, the possible origins are
</p>
<p>0; 0; 0; 0; 1
2
; 0; 0; 0; 1
</p>
<p>2
; 0; 1
</p>
<p>2
; 1
2
</p>
<p>Combining these two sign allocations, the common origins are
</p>
<p>0; 0; 0; 0; 0; 1
2
</p>
<p>In order to fix the origin uniquely at, say, 0, 0, 0, we select another reflection with a plus sign with
</p>
<p>respect to 0, 0, 0. Reference to Table 8.2 shows that parity groups 4, 5, 6, and 8 each meet this
</p>
<p>requirement. For the following reasons, parity groups 1 and 3 are excluded from the choice as the
</p>
<p>third origin-specifying reflection. Group 1 is a special case because signs of eee reflections do not
</p>
<p>change sign among the permitted (�1) origins, and Group 3 (eoe) is related to groups 2 and 7 through an
</p>
<p>addition of indices:
</p>
<p>312&thorn; 322 ! 634 (8.5)
</p>
<p>or, more generally,
</p>
<p>ooe&thorn; oee ! eoe (8.6)
</p>
<p>since o + o &frac14; e + e &frac14; e, and e + o &frac14; o. Parity groups 2, 3, and 7 are said to be linearly related, and
cannot be used together in defining the choice of origin. As stated above, structure factors belonging
</p>
<p>to parity group 1 do not change sign on change of origin, as is evident from both the development of
</p>
<p>(8.4) and Table 8.2. Reflections in this group are called structure seminvariants; their actual signs
</p>
<p>depend on the structure among the permitted origins, and cannot be used to restrict the origin.
</p>
<p>Hence, the origin will be fixed by choosing a third reflection from one of the groups 4, 5, 6, or 8; the
</p>
<p>reflections so chosen, three in this case, form a starting set.
</p>
<p>Table 8.2 Effect of a change of the origin coordinates, among centers of symmetry, on the sign of a structure factor
</p>
<p>Centers of symmetry
</p>
<p>Parity group
</p>
<p>1 2 3 4 5 6 7 8
</p>
<p>h even h odd h even h even h even h odd h odd h odd
</p>
<p>k even k even k odd k even k odd k even k odd k odd
l even l even l even l odd l odd l odd l even l odd
</p>
<p>0, 0, 0 + + + + + + + +
1
2
, 0, 0 + � + + + � � �
0, 1
</p>
<p>2
, 0 + + � + � + � �
</p>
<p>0, 0, 1
2
</p>
<p>+ + + � � � + �
0, 1
</p>
<p>2
, 1
2
</p>
<p>+ + � � + � � +
1
2
, 0, 1
</p>
<p>2
+ � + � � + � +
</p>
<p>1
2
, 1
2
, 0 + � � + � � + +
</p>
<p>1
2
, 1
2
, 1
2
</p>
<p>+ � � � + + + �
</p>
<p>354 8 Direct Methods and Refinement</p>
<p/>
</div>
<div class="page"><p/>
<p>8.2.3 Sign Determination: Centrosymmetric Crystals
</p>
<p>Many equations have been proposed that are capable of providing sign information for centrosym-
</p>
<p>metric crystals. Two of these expressions have proved to be outstandingly useful, and it is to them that
</p>
<p>we first turn our attention.
</p>
<p>Triple-Product Sign Relationship
In 1952, Sayre [3] derived a general formula for hypothetical structures containing identical resolved
</p>
<p>atoms. For centrosymmetric crystals, it was given in the form
</p>
<p>s&eth;hkl&THORN; s&eth;h0k0l0&THORN; s&eth;h� h0; k � k0; l� l0&THORN; � &thorn;1 (8.7)
</p>
<p>where � here means &ldquo;is probably equal to.&rdquo; The vectors associated with these reflections, d*(hkl),
</p>
<p>d*(h0k0l0), and d*(h � h0, k � k0, l � l0) form a closed triangle, or vector triplet, in reciprocal space.
In practice, it may be possible to form several such vector triplets for a given hkl; Fig. 8.2a shows two
</p>
<p>triplets for the vector 300. If any two of the signs in (8.7) are known, the third can be deduced, andwe can
</p>
<p>then extend the sign information beyond that given in the starting set.
</p>
<p>A physical meaning can be given to (8.7) by drawing the traces, in real space, of the three
</p>
<p>planes that form a vector triplet in reciprocal space (Fig. 8.2a, b). For a centrosymmetric crystal,
</p>
<p>we may write
</p>
<p>r&eth;xyz&THORN; &frac14;
2
</p>
<p>Vc
</p>
<p>X
</p>
<p>h
</p>
<p>X
</p>
<p>k
</p>
<p>X
</p>
<p>l
</p>
<p>�jF&eth;hkl&THORN;j cos 2p&eth;hx&thorn; ky&thorn; lz&THORN; (8.8)
</p>
<p>The jF(hkl)j terms in this equation take a positive sign if the traces of the corresponding planes pass
</p>
<p>through the origin, like the full lines in Fig. 8.2b, and a negative sign if they lie midway between these
</p>
<p>positions, like the dashed lines in the figure. The combined contributions from the three planes in
</p>
<p>question will thus have maxima at the points of their mutual intersections, which are therefore
</p>
<p>potential atomic sites, and correspond to regions of high electron density.
</p>
<p>This argument is particularly strong if the three planes have large jEj values, because in jEj the
damping effect of f has been significantly reduced, leaving a term that is governed by the structure
</p>
<p>itself. It may be seen from the diagram that triple intersections occur only at points where either three
</p>
<p>full lines (+ + +) meet, or two dashed lines and one full line meet (some combination of + � �).
This result is in direct agreement with (8.7). It is interesting to recall that the structure of hexam-
</p>
<p>ethylbenzene was solved in 1929 by Lonsdale [4] through drawing the traces of three high-order,
</p>
<p>high-intensity reflection planes, 7�30, 340, and 4�70, and placing atoms at their intersections. These
</p>
<p>planes form a vector triplet, and this structure determination contained, therefore, the first, but
</p>
<p>apparently inadvertent, use of direct methods.
</p>
<p>�2�Formula
Hauptman and Karle [5] have given the more general form of (8.7) as
</p>
<p>s&frac12;E&eth;hkl&THORN;� � s
X
</p>
<p>h0k0l0
</p>
<p>E&eth;h0k0l0&THORN;E&eth;h� h0; k � k0; l� l0&THORN;
</p>
<p>" #
(8.9)
</p>
<p>where E(h) is the value of jE(h)j with its sign, and the summation is taken over all vector pairs with
known signs which form a triplet with hkl. The probability associated with (8.9) is given by [6]
</p>
<p>8.2 Direct Methods of Phase Determination 355</p>
<p/>
</div>
<div class="page"><p/>
<p>P&thorn;&eth;hkl&THORN; &frac14; 12 &thorn; 12 tanh&frac12;&eth;s3=s
3=2
2 &THORN;a0� (8.10)
</p>
<p>where a0 and sn are given by
</p>
<p>a0 &frac14; jE&eth;hkl&THORN;j
X
</p>
<p>h0k0l0
</p>
<p>E&eth;h0k0l0&THORN;E&eth;h� h0; k � k0; l� l0&THORN; (8.11)
</p>
<p>sn &frac14;
X
</p>
<p>j
</p>
<p>Znj (8.12)
</p>
<p>and Zj is the atomic number of the jth atom. For a structure containing N identical atoms, s3=s
3=2
2 is
</p>
<p>equal to N�1/2. From (8.11), the probability is strongly dependent upon the magnitudes of the jEj
</p>
<p>values. Furthermore, unless glide-plane or screw-axis symmetry is present, or there exist some other
</p>
<p>means of generating negative signs, (8.9) will produce only positive signs for all jE(hkl)j used. Such
</p>
<p>a situation would correspond to a structure with a very heavy atom at the origin and would, in
</p>
<p>general, indicate an incorrect solution.
</p>
<p>Fig. 8.2 (a) Vector triplets 300, 204, 10�4 and 300, 30�4, 004. (b) Physical interpretation of (8.7): the lines are traces of
the families (300), (004), and &eth;30�4&THORN;, and the points of triple intersection correspond with (8.7)
</p>
<p>356 8 Direct Methods and Refinement</p>
<p/>
</div>
<div class="page"><p/>
<p>If the combination of signs under the summation in (8.11) produces a large and negative value for
</p>
<p>a0, the corresponding value of P+(hkl) may tend to zero. This result indicates that s(hkl) is negative,
with a probability that tends to unity. Probability curves for different numbers N of atoms in the unit
</p>
<p>cell as a function of a0 are shown in Fig. 8.3. Since the most reliable signs from (8.9) are associated
with large jEj values, we can now add to the origin-specifying criteria the requirements of both large
jEj magnitudes and a large number of &sum;2 interactions for each reflection in the starting set. In this
way, strong and reliable sign propagation is encouraged.
</p>
<p>In order to illustrate the operation of the &sum;2 relationship, we shall consider the two vector triplets in
</p>
<p>Fig. 8.2. The sign to be determined is s(300), the others are assumed to be known. It may be noted that
</p>
<p>sometimes we speak of a sign as + or� and at other times as +1 or�1. The latter formulation is clearly
more appropriate to computational methods. The data are tabulated as follows:
</p>
<p>hkl jE(hkl)j
300 2.40
</p>
<p>h0k0l0 E(h0k0l0)
</p>
<p>h � h0,
k � k0,
l � l0
</p>
<p>E(h � h0,
k � k0,
l � l0) a0 s(hkl) P+(hkl) (%)
</p>
<p>10�4 +2.03 204 �2.22 �19.3 �1 0.8
004 �1.95 30�4 +1.81 �1
</p>
<p>Assuming an N of 64, the indication given is that s(300) is negative with a probability of 99.2%.
</p>
<p>Fig. 8.3 Percentage
probability of a single
</p>
<p>triple-product (&sum;2) sign
</p>
<p>relationship as a function of
</p>
<p>a0 for different numbers N
of atoms in a unit cell,
</p>
<p>according to (8.10)
</p>
<p>8.2 Direct Methods of Phase Determination 357</p>
<p/>
</div>
<div class="page"><p/>
<p>8.2.4 Amplitude Symmetry and Phase Symmetry
</p>
<p>In space group P�1, the only symmetry-related structure factors are F(hkl) and F&eth;�h �k �l&THORN;. According
to Friedel&rsquo;s law the intensities and, hence the amplitudes, of these structure factors are equal, and in
</p>
<p>centrosymmetric space groups s&eth;hkl&THORN; &frac14; s&eth;�h �k �l&THORN;. Thus, the amplitude symmetry and the phase symmetry
follow the same law, but this will not necessarily be true in other space groups.
</p>
<p>From the geometric structure factor for space group P21/c, see (3.87),
</p>
<p>jF&eth;hkl&THORN;j &frac14; jF&eth;�h �k �l&THORN;j &frac14; jF&eth;h�kl&THORN;j &frac14; jF&eth;�hk�l&THORN;j (8.13)
</p>
<p>and for the signs there are two possibilities to consider:
</p>
<p>k &thorn; l &frac14; 2n: s&eth;hkl&THORN; &frac14; s&eth;�h �k �l&THORN; &frac14; s&eth;h�kl&THORN; &frac14; s&eth;�hk�l&THORN; (8.14)
</p>
<p>k &thorn; l &frac14; 2n&thorn; 1: s&eth;hkl&THORN; &frac14; s&eth;�h �k �l&THORN; &frac14; �s&eth;h�kl&THORN; &frac14; �s&eth;�hk�l&THORN; (8.15)
</p>
<p>These relationships provide enhanced opportunities for &sum;2 relationships to be developed, including
</p>
<p>negative values for signs, and in this way space-group symmetry can improve the chances of a successful
</p>
<p>phase determination. When considering phase relationships in a non-centrosymmetric space group, we
</p>
<p>need to take note of the change in sign and magnitude of both the A and the B components of the
</p>
<p>geometrical structure factor, as h, k, or l take a negative sign. Consider, for example, space group Pc (b
</p>
<p>axis unique; origin on c). We can separate (3.85) into two parts, according to whether l is odd or even:
</p>
<p>l &frac14; 2n
</p>
<p>A &frac14; 2 cos 2p&eth;hx&thorn; lz&THORN; cos 2pky
B &frac14; 2 sin 2p&eth;hx&thorn; lz&THORN; cos 2pky
</p>
<p>l &frac14; 2n + 1
</p>
<p>A &frac14; �2 sin 2p&eth;hx&thorn; lz&THORN; sin 2pky
B &frac14; 2 cos 2p&eth;hx&thorn; lz&THORN; sin 2pky
</p>
<p>It is clear that, in all cases, f&eth;hkl&THORN; &frac14; �f&eth;�h �k �l&THORN; 6&frac14; f&eth;�hkl&THORN;. However, for the two parities of l, the
following expressions hold.
</p>
<p>l &frac14; 2n: f&eth;hkl&THORN; &frac14; f&eth;h�kl&THORN; and f&eth;�hkl&THORN; &frac14; �f&eth;hk�l&THORN;
l &frac14; 2n&thorn; 1: f&eth;hkl&THORN; &frac14; p&thorn; f&eth;h�kl&THORN; and f&eth;�hkl&THORN; &frac14; p� f&eth;hk�l&THORN;
</p>
<p>The amplitude symmetry and phase symmetry for all space groups are contained in the Interna-
</p>
<p>tional Tables for X-ray Crystallography, Volume A (or Volume 1) [7].
</p>
<p>8.2.5 &sum;2-Listing
</p>
<p>Because of both the increased probability in relationships developed for reflections with high jEj
values and the existence of many vector triplets in a complete set of data, the initial application of
</p>
<p>direct methods is limited to reflections with large jEj values, say, greater than 1.5.
A&sum;2 listing is prepared by considering each value of jE(hkl)j greater than the preset limit in order of
</p>
<p>decreasingmagnitude, as a basic hkl vector, and searching the data for all possible interactions with h0k0l0
</p>
<p>358 8 Direct Methods and Refinement</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Equ87_3">http://dx.doi.org/10.1007/978-1-4614-3954-7#Equ87_3</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Equ85_3">http://dx.doi.org/10.1007/978-1-4614-3954-7#Equ85_3</a></div>
</div>
<div class="page"><p/>
<p>and h � h0, k � k0, l � l0. Some reflections will enter into many such interactions, while others will
produce only a small number.
</p>
<p>8.2.6 Symbolic-Addition Procedure: Example
</p>
<p>Karle and Karle [8] described a technique for the systematic application of the &sum;2 formula for
</p>
<p>building up a self-consistent sign set. The various steps involved are outlined below, using results
</p>
<p>obtained with pyridoxal phosphate oxime dihydrate [9].
</p>
<p>Crystal Data
</p>
<p>Formula: C8H11N2O6P&middot;2H2O
</p>
<p>System: triclinic
</p>
<p>Unit-cell dimensions: a &frac14; 10.94 Å, b &frac14; 8.06 Å, c &frac14; 9.44 Å; a &frac14; 57.18�, b &frac14; 107.68�, g &frac14; 116.53�
Vc: 623.75 Å
</p>
<p>3
</p>
<p>Dm: 1.57 g cm
�3
</p>
<p>Mr: 298.19
</p>
<p>Z: 1.98 or 2 to the nearest integer
</p>
<p>Absent spectra: none
</p>
<p>Possible space groups: P1 or P�1; P�1 was chosen on the basis of intensity statistics
</p>
<p>All atoms are in general positions.
</p>
<p>Sign Determination
1. A total of 163 reflections for which jEj � 1.5 was arranged in descending order of magnitude, and
</p>
<p>a &sum;2 listing was obtained, Table 8.4.
</p>
<p>2. From a study of the &sum;2 listing, three reflections were allocated + signs, Table 8.3; they are the
</p>
<p>origin-fixing reflections, selected according to the procedures already discussed.
</p>
<p>3. Equation (8.9) was used by searching, initially between members of the origin-fixing set and
</p>
<p>other reflections. In order to maintain a high probability, only the highest jEj values were used.
For example, 9�5 �5 with jEj &frac14; 2.31 is generated by the combination of 8�1 �5 and �140
</p>
<p>s&eth;9 �5 �5&THORN; � s&eth;8�1 �5&THORN; s&eth;�140&THORN; &frac14; &eth;&thorn;1&THORN;&eth;&thorn;1&THORN; &frac14; &thorn;1 (8.16)
</p>
<p>From (8.11), a0 is 16.5, and Fig. 8.3, withN &frac14; 38 and excluding hydrogen, shows that the probability
</p>
<p>of this indication is about 99.7%. The new sign was accepted and used to generate more signs.
</p>
<p>This process was continued until no new signs could be developed with high probability.
</p>
<p>Table 8.3 Starting set for
the symbolic-addition pro-
</p>
<p>cedure
</p>
<p>hkl jEj Sign
</p>
<p>9�1 �4 2.97 +
</p>
<p>8�1 �5 3.00 +
</p>
<p>�140 2.38 +
</p>
<p>020 4.50 A
</p>
<p>253 2.24 B
</p>
<p>822 2.71 C
</p>
<p>303 2.69 D
</p>
<p>023 2.28 E
</p>
<p>8.2 Direct Methods of Phase Determination 359</p>
<p/>
</div>
<div class="page"><p/>
<p>4. At this stage, it is often found that the number of signs developed with confidence is small. This
</p>
<p>situation arose with pyridoxal phosphate oxime dihydrate, and the&sum;2 formula was then applied to
</p>
<p>reflections with symbolic signs. In this technique, a reflection was selected, again by virtue of its
</p>
<p>high jEj value and long &sum;2 listing, and allocated a letter symbol, as shown in Table 8.3; the letter
symbols A&ndash;E represent a + or � sign. Generally, less than five symbolic phases are sufficient, and
there are no necessary restrictions on the parities of these reflections. However, it is desirable that
</p>
<p>there are no redundancies in the complete starting set, that is, no three reflections in the set should
</p>
<p>themselves be related by a triple-product relationship.
</p>
<p>As a symbol became involved in a sign of a reflection, it was written into the &sum;2 listing. The
</p>
<p>example in Table 8.4 shows a &sum;2 entry for reflection 9�8 �6. Reading across the table, sign
</p>
<p>combinations are seen to be generated by multiplying s(h0k0l0) by s(h � h0, k � k0, l � l0), which
are then written as s&eth;9�8 �6&THORN; in the penultimate column. Recurring symbol combinations, such as
ABD, gave rise to consistent indications. If the probability that s&eth;9�8 �6&THORN; &frac14; s&eth;ABD&THORN; is sufficiently
large, this sign value is entered for s&eth;9�8 �6&THORN; wherever these indices occur. In the final column of the
table, the probability of each sign indication is listed. Although they are small individually, the
</p>
<p>combined probability from (8.10) that s&eth;9�8 �6&THORN; was ABD is 100%.
5. When this process had been exhausted, the results were examined for agreement among sign
</p>
<p>relationships. For example, in Table 8.4 there is a weak indication that ABD &frac14; EC. The most
significant relationships found overall were AC &frac14; E, C &frac14; EB, B &frac14; ED, AD &frac14; E, and AB &frac14; CD.
Multiplying the first by the second, and the first by the fourth, and remembering that products such
</p>
<p>as A2 equal +1 reduces this list to A &frac14; B, C &frac14; D, and E &frac14; AC. The five symbols were reduced,
effectively, to two, A and C. The sign determination was rewritten in terms of signs and the
</p>
<p>symbols A and C; reflections with either uncertain or undetermined signs were rejected from the
</p>
<p>first electron density calculation.
</p>
<p>8.2.7 Calculation of E Maps
</p>
<p>The result of the above analysis meant that four possible sign sets could be generated by the
</p>
<p>substitutions A &frac14; �1, C &frac14; �1. The set with A &frac14; C &frac14; +1 was rejected immediately because this
phase assignment implies a very heavy atom at the origin of the unit cell. The three other sign
</p>
<p>Table 8.4 &sum;2 listing for reflection 9�8 �6 of pyridoxal oxime phosphate with symbols added
a
</p>
<p>h0k0l0 s(h0k0l0)
jE2j:
jE(h0k0l0)j
</p>
<p>h � h0,
k � k0,
l � l0
</p>
<p>s(h � h0,
k � k0,
l � l0)
</p>
<p>jE3j: jE(h � h0,
k � k0,
l � l0)j jE1jjE2jjE3j s&eth;9�8 �6&THORN; P&thorn;&eth;9�8 �6&THORN; (%)
</p>
<p>1 �5 0 BD 2.16 8, �3; �6 A 1.63 6.40 ABD 90
</p>
<p>10; �2 �2 AB 2.04 1, 6, 4 D 1.88 6.97 ABD 91
</p>
<p>10; �7 �1 D 1.87 1, 1, 5 AB 1.63 5.54 ABD 87
</p>
<p>4 �8 �3 D 1.83 5, 0, �3 ECD 1.58 5.25 EC 85
</p>
<p>3 �9 �4 1.76 6, 1, �2 1.58 5.03
</p>
<p>3 �5 �6 1.70 6, �3, 0 1.51 4.66
</p>
<p>6 �7 �2 1.68 3, �1, �4 1.63 4.98
</p>
<p>10; �4 �2 B 1.62 1, 4, 4 AD 1.67 4.93 ABD 84
</p>
<p>0 �2 0 �A 4.50 9, 6, �6 1.73 14.08
0 �8 0 + 2.48 9, 0, �6 1.85 8.30
</p>
<p>ajE&eth;98 6&THORN;j &frac14; jE1j &frac14; 1:89. We can use both h � k and h + k in these triple products
</p>
<p>360 8 Direct Methods and Refinement</p>
<p/>
</div>
<div class="page"><p/>
<p>combinations were used to calculate E maps. These maps were obtained by Fourier syntheses, using
</p>
<p>(8.8), but with jEj replacing jFj as the coefficients. The sharp nature of jEj implicit in (8.1) is
advantageous when using a limited number of data to resolve atomic peaks in the electron density
</p>
<p>map, although normally about eight reflections per atom in the asymmetric unit are desirable.
</p>
<p>Spurious peaks can arise, however, like that in the vicinity of O(8), because of the limited number
</p>
<p>of coefficients in the Fourier series for an E map. The sign combination for pyridoxal phosphate
</p>
<p>oxime dihydrate that led to an interpretable E map was A &frac14; C &frac14; �1. The atomic positions from this
map, Fig. 8.4a, were used in a successful refinement of the structure, and Fig. 8.4b shows the chemical
</p>
<p>formula and Fig. 8.4c shows the molecular conformation determined in this study.
</p>
<p>If there are n symbolic signs in the final centrosymmetric phase solution, there will be 2n
</p>
<p>combinations, each of which can give rise to an E map, and it is desirable to set up criteria that
</p>
<p>will seek the most probable set. We shall consider such criteria during our discussion of the non-
</p>
<p>centrosymmetric case, where they are of even greater importance.
</p>
<p>8.2.8 Phase Determination: Non-centrosymmetric Crystals
</p>
<p>The non-centrosymmetric case is more difficult, both in theory and in practice. Much of this difficulty
</p>
<p>stems from the fact that the phase angle can take on any value between 0 and 2p, with a consequent
</p>
<p>imprecision in its determination. Nevertheless, direct methods are used regularly to solve structures,
</p>
<p>the limiting factor being the number of atoms N in the unit cell, which is space-group dependent.
</p>
<p>Structures where N is 400 or more can be expected to be solved without great difficulty using current
</p>
<p>versions of the various programs now available.
</p>
<p>Fig. 8.4 Pyridoxal
phosphate oxime dihydrate.
</p>
<p>(a) Composite three-
dimensional E map as seen
</p>
<p>along b. (b) Chemical
formula. (c) Molecular
conformation derived from
</p>
<p>(a) excluding H atoms
(drawn with RASMOL)
</p>
<p>8.2 Direct Methods of Phase Determination 361</p>
<p/>
</div>
<div class="page"><p/>
<p>Equations for a non-centrosymmetric crystal are, not surprisingly, more generalized expressions of
</p>
<p>those such as (8.9), which relates to centrosymmetric crystals. Using the fact that the electron density
</p>
<p>distribution is a nonnegative function, Karle and Hauptman derived a set of inequality relationships [10].
</p>
<p>The first three inequalities may be written as nonnegative functions.
</p>
<p>F000 � 0 (8.17)
</p>
<p>jFhj � F000 (8.18)
</p>
<p>Fh � dh;k � jrj (8.19)
</p>
<p>where
</p>
<p>dh;k &frac14; Fh�kF
a
k=F000 (8.20)
</p>
<p>and
</p>
<p>jrj &frac14;
</p>
<p>F000 F
	
h�k
</p>
<p>Fh�k F000
</p>
<p>�
</p>
<p>�
</p>
<p>�
</p>
<p>�
</p>
<p>�
</p>
<p>�
</p>
<p>�
</p>
<p>�
</p>
<p>F000 F
	
k
</p>
<p>Fk F000
</p>
<p>�
</p>
<p>�
</p>
<p>�
</p>
<p>�
</p>
<p>�
</p>
<p>�
</p>
<p>�
</p>
<p>�
</p>
<p>F000
(8.21)
</p>
<p>We use here the convenient notation h for hkl, k for h0k0l0, h � k for a third reflection that forms a
vector triplet with h and k, and, in order to avoid excessive parentheses, Fh for F(h). Equations (8.17)
and (8.18) are immediately acceptable in terms of earlier discussions in this book; FkFh�k is a
</p>
<p>multiplication of two structure factors and may be interpreted on an Argand diagram as jFkjjFh�kj
exp{i(&rsquo;k + &rsquo;h�k)}.
</p>
<p>The structure factor Fh, given only jFhj, must lie on a circle of that radius on an Argand diagram,
</p>
<p>Fig. 8.5. Equation (8.19) then indicates that Fh lies within a circle, center dh,k and radius jrj, between
</p>
<p>Fig. 8.5 Illustration
of (8.19) at equality
</p>
<p>362 8 Direct Methods and Refinement</p>
<p/>
</div>
<div class="page"><p/>
<p>the points P and Q. Expansions of the determinants in (8.21), remembering that FF* &frac14; jFj2, shows
that the larger the values of jFkj2 and jFh�kj2, the closer Fh approaches dh,k. For a given h, as k is
varied, Fh is proportional to the average over k:
</p>
<p>Fh / hFkFh�kik (8.22)
</p>
<p>the proportionality constant being F(000). We can see how (8.22) can give rise to (8.9) or (8.7) for a
</p>
<p>single interaction.
</p>
<p>Using the general relation
</p>
<p>Fh &frac14; jFhj exp&eth;ifh&THORN; (8.23)
</p>
<p>we obtain the phase addition formula
</p>
<p>fh � fk &thorn; fh�k (8.24)
</p>
<p>The sign � indicates an approximation which is better the larger the values of the corresponding
</p>
<p>structure factors. Where several triplets are involved with a given h, (8.24) becomes
</p>
<p>fh � fk &thorn; fh�kh ik (8.25)
</p>
<p>where h ik implies an average, taken over a number of triple product relationships (TPRs) common
to h.
</p>
<p>The Fo data derived experimentally are converted to jEj values. Again, we commence phase
</p>
<p>determination with jEj values greater than about 1.5 in order to maintain acceptable probability limits.
Equation (8.25) is illustrated by an Argand diagram in Fig. 8.6 for four values of k; fh is the estimated
</p>
<p>phase angle associated with the resultant Rh. Each direction labeled k depends on a product
</p>
<p>jEkjjEh�kj and may be resolved into components A and B along the real and imaginary axes,
respectively, such that
</p>
<p>A &frac14; jEkjjEh�kj cos&eth;fk &thorn; fh�k&THORN; (8.26)
</p>
<p>and
</p>
<p>B &frac14; jEkjjEh�kj sin&eth;fk &thorn; fh�k&THORN; (8.27)
</p>
<p>It follows from (8.25) and (8.26), including a frequency enhancement weight wh, that
</p>
<p>tanfh �
</p>
<p>P
h whjEkjjEh�kj sin&eth;fk &thorn; fh�k&THORN;P
h whjEkjjEh�kj cos&eth;fk &thorn; fh�k&THORN;
</p>
<p>(8.28)
</p>
<p>Equation (8.28) is a weighted tangent formula. Weights may be used as defined below under
</p>
<p>MULTAN. Current phase-determining procedures are based largely on (8.28), and the reliability of
</p>
<p>(8.28) can be measured by the variance V&eth;fh&THORN;. Figure 8.7 shows V&eth;fh&THORN; as a function of ah, where
</p>
<p>a2h &frac14;
X
</p>
<p>h
</p>
<p>khk cos&eth;fk &thorn; fh�k&THORN;
</p>
<p>" #2
&thorn;
</p>
<p>X
</p>
<p>k
</p>
<p>khk sin&eth;fk &thorn; fh�k&THORN;
</p>
<p>" #2
(8.29)
</p>
<p>8.2 Direct Methods of Phase Determination 363</p>
<p/>
</div>
<div class="page"><p/>
<p>and
</p>
<p>khk &frac14; 2s3s�3=22 jEhjjEkjjEh�kj (8.30)
</p>
<p>with
</p>
<p>sn &frac14;
XN
</p>
<p>j&frac14;1
Znj (8.31)
</p>
<p>as before, and the sum is taken over the N atoms in the unit cell. The parameter ah gives a measure of
</p>
<p>the reliability with which fh is determined by the tangent formula. When (8.29) contains only one
</p>
<p>term, as it may in the initial stages of phase determination, then ah &frac14; khk and is strongly dependent
on the product jEhjjEkjjEh�kj. Figure 8.7 shows clearly that V&eth;fh&THORN; has acceptably small values when
ah is greater than about 4, corresponding to V
</p>
<p>1/2 &lt; 30�, but increases rapidly for ah decreasing below
about 3, corresponding to V1/2 &gt; 40�: ah depends also on s3s
</p>
<p>�3=2
2 , which, in turn, depends on the number
</p>
<p>and types of atoms in the unit cell. This dependence may be illustrated by hypothetical structures
</p>
<p>containing different numbers N of identical atoms; ah (&frac14;khk) is then given by
</p>
<p>ah &frac14;
2
</p>
<p>N1=2
jEhjjEkjjEh�kj (8.32)
</p>
<p>Fig. 8.6 Summation of
four &ldquo;vectors&rdquo; k1&ndash;k4 on an
</p>
<p>Argand diagram; the
</p>
<p>resultant is Rh, with a phase
</p>
<p>angle fh
</p>
<p>364 8 Direct Methods and Refinement</p>
<p/>
</div>
<div class="page"><p/>
<p>Table 8.5 lists the values of jEminj needed to obtain ah &frac14; 3 for selected values of N from 25 to 100.
The table illustrates clearly an important limitation of direct methods: the required jEminj increases
dramatically as a function of N, whereas, as indicated earlier, the distribution of jEj values is largely
independent of structural complexity. Therefore, it becomes more and more difficult to form a good
</p>
<p>starting set as N becomes larger and larger. Calculation of ah from (8.29) is possible only when phases
</p>
<p>are available. In the initial stages of phase determination this is not practicable, and the following
</p>
<p>formula for the expectation value (a2E) of a
2
h, which uses only the values of khk, has been developed:
</p>
<p>a2E &frac14;
X
</p>
<p>k
</p>
<p>k2hk &thorn;
X
</p>
<p>k
</p>
<p>X
</p>
<p>k0
khkkhk0
</p>
<p>k 6&frac14;k0
</p>
<p>I1&eth;khk&THORN;I1&eth;khk0&THORN;
I0&eth;khk&THORN;I0&eth;khk0&THORN; (8.33)
</p>
<p>where I0 and I1 are modified Bessel functions of the zero and first orders, respectively. The function
</p>
<p>I1(k)/I0(k) has the form shown in Fig. 8.8 and may be expressed as the polynomial
</p>
<p>I1&eth;k&THORN;=I0&eth;k&THORN; � 0:5658k� 0:1304k2 &thorn; 0:0106k3
</p>
<p>in the range 0 � k � 6; for larger values of k the function is essentially unity. These principles, used
in conjunction with those discussed earlier for selecting the origin-determining reflections, may help a
</p>
<p>direct methods analysis to be established on a sound basis right from the beginning, and so lead to a
</p>
<p>number of sufficiently accurate phases to give an interpretable E map. Experience shows, however,
</p>
<p>that even with great care, the development of phases may not always be successful. In such an event
</p>
<p>the remedy is often to try again with a different starting set of reflections.
</p>
<p>Fig. 8.7 Variance &eth;Vfh&THORN; as a function of ah
</p>
<p>8.2 Direct Methods of Phase Determination 365</p>
<p/>
</div>
<div class="page"><p/>
<p>&sum;1-Relationships
In (8.24), let �k &frac14; h; then, since � f�h &frac14; fh,
</p>
<p>fh � f�h &thorn; f2h � f2h=2 (8.34)
</p>
<p>If the structure is centrosymmetric then, in (8.7), sh is �1; since sh s�h &frac14; 1, it follows that
</p>
<p>s2h � 1 (8.35)
</p>
<p>These two&sum;1 relationships, like (8.7) and (8.24) themselves, require large values of both jEhj and jE2hj
</p>
<p>for a high probability of their validity.
</p>
<p>Multan System
We refer to equations of the MULTAN program system [11, 12] in order to define the weights
</p>
<p>required in (8.28). Multan employs a modified weighted tangent formula given by
</p>
<p>tanfh &frac14;
X
</p>
<p>k
</p>
<p>Qh; k sin&eth;fk &thorn; fh � k&THORN;
</p>
<p>" # X
</p>
<p>k
</p>
<p>Qh; k cos&eth;fk &thorn; fh � k&THORN;
</p>
<p>" #,
&frac14; Th=Bh
</p>
<p>where
</p>
<p>Qh;k &frac14; okoh�kjEkjjEh�kj=&eth;1� jUhj
2&THORN;
</p>
<p>where the unitary structure factor jUhj is defined by
</p>
<p>jUhj &frac14; jFhj e
1=2
X
</p>
<p>j
gj
</p>
<p>� �.
</p>
<p>Fig. 8.8 Variation of
I1(k)/I0(k) with k
</p>
<p>Table 8.5 Values of jEminj
for ah &frac14; 3.0 in structures
containing N identical
</p>
<p>atoms per unit cell
</p>
<p>N jEminj
</p>
<p>25 1.96
</p>
<p>36 2.08
</p>
<p>49 2.19
</p>
<p>64 2.29
</p>
<p>81 2.38
</p>
<p>100 2.47
</p>
<p>366 8 Direct Methods and Refinement</p>
<p/>
</div>
<div class="page"><p/>
<p>and j varies from 1 to N, the number of atoms in the unit cell; gj is fj corrected for temperature; ok is
</p>
<p>given by
</p>
<p>oh &frac14; tan&frac12;s3s�3=22 jEhj&eth;T2h &thorn; B2h&THORN;�
</p>
<p>and sn is defined by (8.31). Thus, each phase assignment carries a weight designed to ensure that
</p>
<p>poorly determined phases have but little effect on the generation of new phases, while inclusion of all
</p>
<p>phases ensures an efficient propagation of phase information throughout the data set.
</p>
<p>8.2.9 Enantiomorph Selection
</p>
<p>In non-centrosymmetric space groups, such as P21 and P212121, that contain no inversion symmetry,
</p>
<p>but can exist in enantiomorphous forms, it is always possible to specify two enantiomorphic
</p>
<p>arrangements of the atoms in the structure that will lead to the same values of jFj. For example, in
</p>
<p>the structure in Fig. 1.8, which has two molecules per unit cell in space group P21, the two
</p>
<p>enantiomorphic arrangements would be related by inversion through the origin, and will be referred
</p>
<p>to as the structure (S) and its inverse (I). From the structure factor theory discussed earlier, we
</p>
<p>can write
</p>
<p>F&eth;h&THORN;S &frac14; A&eth;h&THORN;S &thorn; iB&eth;h&THORN;S
</p>
<p>for the structure, and
</p>
<p>F&eth;h&THORN;I &frac14; A&eth;h&THORN;I &thorn; iB&eth;h&THORN;I (8.36)
</p>
<p>for its inverse. From the inversion relationship, we know that F(h)S and F(h)I are complex conjugates;
hence,
</p>
<p>A&eth;h&THORN;S &frac14; A&eth;h&THORN;I
</p>
<p>and
</p>
<p>B&eth;h&THORN;S &frac14; �B&eth;h&THORN;I (8.37)
</p>
<p>For either the structure or its inverse, we can choose B(h) to be positive, so that the corresponding
phase angle f&eth;h&THORN;lies in the range 0 � f&eth;h&THORN; � p. This procedure was followed in the structure
analysis of tubercidin, Sect. 8.2.10, where the phase of symbolic reflection a&eth;13�8&THORN; was restricted to
</p>
<p>a value between 0 and p, specifically 3p/4.
</p>
<p>In P212121, a space group of frequent occurrence in practice, the zonal reflections 0kl, h0l, and hk0
</p>
<p>are centric, and may be given phases equal to mp/2, since no 21 axis passes through the origin. The
</p>
<p>value ofm takes the same parity as the Miller index following zero, working in a cyclic manner. Thus,
</p>
<p>an origin and an enantiomorph could be specified in this space group by the selection
</p>
<p>5 2 0 &thorn;p=2
</p>
<p>0 1 1 &thorn;p=2
</p>
<p>11 3 0 &thorn;p=2
</p>
<p>9
&gt;=
&gt;;
</p>
<p>Origin
</p>
<p>11 0 0 &thorn;p=2 Enantiomorph
</p>
<p>8.2 Direct Methods of Phase Determination 367</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Fig8_1">http://dx.doi.org/10.1007/978-1-4614-3954-7#Fig8_1</a></div>
</div>
<div class="page"><p/>
<p>A detailed practical treatment on the origin and enantiomorph for all space groups has been given
</p>
<p>by Rogers [13]. It is important not to confuse the specification of the enantiomorph with the selection
</p>
<p>of the absolute configuration of a structure: in both cases the same type of space group is involved.
</p>
<p>Selection of the enantiomorph is essential to a correct application of direct methods to a structure with
</p>
<p>an enantiomorphous space group. However, the derived solution of the structure may correspond to
</p>
<p>either the absolute configuration or its inverse. This dilemma has to be resolved by further tests,
</p>
<p>usually involving anomalous scattering, Sect. 7.6. We discuss enantiomorph selection and related
</p>
<p>topics in Appendix E.
</p>
<p>8.2.10 Phase Determination in Space Group P21
</p>
<p>The method of symbolic addition can be used for phase determination in non-centrosymmetric
</p>
<p>crystals, but it can be somewhat laborious because of the general nature of f&eth;hkl&THORN;. The structure of
tubercidin was determined by Stroud [14] using this method: Table 8.6 lists the crystal data for this
</p>
<p>compound.
</p>
<p>In space group P21, jE(hkl)j has the following symmetry equivalence:
</p>
<p>jE&eth;hkl&THORN;j &frac14; jE&eth;�hk�l&THORN;j &frac14; jE&eth;h�kl&THORN;j &frac14; jE&eth;�h �k �l&THORN;j (8.38)
</p>
<p>The phases of the symmetry-related reflections in this space group are also linked, but in a different
</p>
<p>way, according to the parity of k:
</p>
<p>k &frac14; 2n : f&eth;hkl&THORN; &frac14; f&eth;�hk�l&THORN; &frac14; �f&eth;h�kl&THORN;
&frac14; �f&eth;�h �k �l&THORN;
</p>
<p>(8.39)
</p>
<p>k &frac14; 2n&thorn; 1 : f&eth;hkl&THORN; &frac14; p&thorn; f&eth;�hk�l&THORN; &frac14; p� f&eth;h�kl&THORN;
&frac14; �f&eth;�h �k �l&THORN; (8.40)
</p>
<p>The h0l zone of this space group is centric, with the origin transferred to the twofold rotation point
</p>
<p>at x &frac14; z &frac14; 0; hence, the phases are restricted to 0 or p, according to the arguments developed in
Sect. 8.2.2, so that permitted origins here are 0, 0, 0; 1
</p>
<p>2
, 0, 0; 0, 0, 1
</p>
<p>2
; 1
2
, 0, 1
</p>
<p>2
.
</p>
<p>Again, the origin was specified by assigning phases to three reflections, following the
</p>
<p>rules discussed above, and shown in Table 8.7. Next, new phases were determined according
</p>
<p>Table 8.6 Crystal data for tubercidin
</p>
<p>Formula, Mr C11H14N4O4, 266.26
</p>
<p>Space group P21
</p>
<p>a (Å) 9.724(9)
</p>
<p>b (Å) 9.346(11)
</p>
<p>c (Å) 6.762(10)
</p>
<p>b (�) 94.64(10)
</p>
<p>Vc (Å
3) 612.52
</p>
<p>Dm (g cm
�3) 1.449
</p>
<p>Dc (g cm
�3) 1.444
</p>
<p>Z 2
</p>
<p>F(000) 280
</p>
<p>368 8 Direct Methods and Refinement</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec31_7">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec31_7</a></div>
</div>
<div class="page"><p/>
<p>to (8.24) or (8.25). In order to maintain an expected variance V&eth;fh&THORN;, Fig. 8.7, of no more than 0.5 rad2,
the product jEhjjEkjjEh�kj must be greater than 8.5 for this structure. Two new phases f&eth;80�2&THORN; and
f&eth;612&THORN; were thus determined from the origin set and added to it; further phases were determined in
terms of the symbols allocated as shown in Table 8.8.
</p>
<p>Eleven phases were generated in terms of the origin set and symbol a, 20 after adding symbol b, and
</p>
<p>47 after adding the third symbolic phase c. Table 8.9 illustrates the initial stages of this process. The criteria
</p>
<p>for accepting a phase were as follows:
</p>
<p>1. That V&eth;fh&THORN;, irrespective of the actual choice for phases c and a should be less than 0.5 rad2, no
matter how many contributors there were to the sum in (8.25); symbol b, in parity group eee, is a
</p>
<p>structure seminvariant with phase 0 or p with respect to the permitted origins.
</p>
<p>Table 8.7 Origin-specifying phases
for tubercidin
</p>
<p>hkl jEhj fh
10�6 1.95 0
</p>
<p>40�1 2.09 0
</p>
<p>71�4 2.45 0
</p>
<p>Table 8.8 Course of the phase determination procedure for
tubercidin
</p>
<p>hkl fh jEhj
Number of numerical
</p>
<p>or symbolic phases
</p>
<p>Origin set;
</p>
<p>Table 8.7
</p>
<p>5
</p>
<p>13�8 a 2.99 11
</p>
<p>206 b 2.20 20
</p>
<p>790 c 2.76 47
</p>
<p>Table 8.9 Initial development of phases for tubercidin
</p>
<p>h jEhj fh jEhjjEkjjEh�kj
Origin set
</p>
<p>40�1 2.09 0
</p>
<p>10�6 1.95 0
</p>
<p>71�4 2.45 0 9.98
</p>
<p>New phases (marked *)
</p>
<p>40�1 2.09 0
</p>
<p>40�1 2.09 0
</p>
<p>80�2* 2.33 0 10.2
</p>
<p>�106a 1.95 0
</p>
<p>71�4 2.45 0
612* 1.83 0 8.74
</p>
<p>First symbol
</p>
<p>13�8 2.99 a
612 1.83 0
</p>
<p>74�6* 2.20 a 12.0
</p>
<p>7�1 �4b 2.45 p
�138b 2.99 p + a
624* 2.19 a 16.0
</p>
<p>aSymmetry-related through (8.39)
bSymmetry-related through (8.40)
</p>
<p>8.2 Direct Methods of Phase Determination 369</p>
<p/>
</div>
<div class="page"><p/>
<p>2. That where there were two or more different indications for a phase, the phase would be accepted
</p>
<p>only when indications of one type predominated strongly. Manual phase determination using a&sum;2
listing indicated relationships such as
</p>
<p>f&eth;63�3&THORN; &frac14; c� 2a� b (8.41)
</p>
<p>and
</p>
<p>b &frac14; 0 (8.42)
</p>
<p>which were strong because they both come from 6 and 3 multiple indications, respectively.
</p>
<p>By reiteration of the phase addition procedure above, the results in Table 8.10 indicate relation-
</p>
<p>ships between a and c. Bearing in mind that the objective is to obtain a self-consistent set of
</p>
<p>phases, it is well to consider how this might now be achieved. Refinement of phases could in
</p>
<p>principle be achieved by application of (8.28). However, this would be possible only if numerical
</p>
<p>values for a and c (taking b as zero) were available. Alternatively, if a working formula relating a
</p>
<p>and c could be found, (8.28) could be implemented by substitution of values for one symbol only.
</p>
<p>Table 8.10 shows that there were 41 indications that
</p>
<p>c &frac14; p&thorn; pa (8.43)
</p>
<p>where the value for p was chosen as the weighted average of the first three indications for c in
</p>
<p>Table 8.10. Thus,
</p>
<p>c &frac14; p&thorn; 3:29a (8.44)
</p>
<p>The symbol a was then limited to the range
</p>
<p>0&lt; a&lt; p (8.45)
</p>
<p>Table 8.10 Relationships between letter symbols
</p>
<p>Form of relationship Number of indications
</p>
<p>c &frac14; p + 2a 7
c &frac14; p + 3a 15
c &frac14; p + 4a 19
c &frac14; 3a 5
c &frac14; 4a 2
c &frac14; �3a 4 or 5
a &frac14; 0 2
a &frac14; p 2
b &frac14; 0 Many
b &frac14; p None
</p>
<p>370 8 Direct Methods and Refinement</p>
<p/>
</div>
<div class="page"><p/>
<p>in order to fix the enantiomorph, Sect. 8.2.9. Values for a were chosen such that
</p>
<p>a &frac14; np=8 &eth;n &frac14; 1; 2; . . . ; 8&THORN; (8.46)
</p>
<p>and converted into phases by the above relationships. Each set was expanded and refined by (8.28),
</p>
<p>taking wh &frac14; 1, for up to 419 reflections with jEminj � 1.0. Some phases were rejected because of
inconsistencies in their phase indications. An interpretable E map was obtained using the refined
</p>
<p>phase set with a &frac14; 6p/8; a composite diagram is given in Fig. 8.9.2
</p>
<p>8.2.11 Advantages and Disadvantages of Symbolic Addition
</p>
<p>Symbolic addition has several advantages and disadvantages, which we summarize as follows:
</p>
<p>Advantages
</p>
<p>1. The user is in control throughout the analysis, and has the responsibility of making sure that all
</p>
<p>formulae, including symmetry relationships, are applied correctly.
</p>
<p>2. The user can make decisions regarding criteria of acceptance of phase indications, the number of
</p>
<p>jEj values to include, the number of symbolic phases, the choice of starting set, and so on.
</p>
<p>Fig. 8.9 Tubercidin, C11H14N4O4. (a) Structural formula in approximately the same orientation as in the E map. (b)
Composite E map, with idealized contours drawn at arbitrary equal intervals. Some peaks are heavier than others
</p>
<p>because of the limited data set used; peak T was the only significant spurious peak. (c) Molecular structure derived from
(b), and drawn with RASMOL
</p>
<p>2 See also Bibliography, Ladd and Palmer (1980).
</p>
<p>8.2 Direct Methods of Phase Determination 371</p>
<p/>
</div>
<div class="page"><p/>
<p>Disadvantages
</p>
<p>1. The analysis can be carried out only by a specialist in crystallography.
</p>
<p>2. The procedure is slow, requiring many hours of preparation before meaningful results emerge.
</p>
<p>3. If a large number of symbols are required, many phase sets will be produced, each of which requires
</p>
<p>refinement by the tangent formula.
</p>
<p>Not surprisingly, alternative rapid and more automatic methods of applying direct methods formulas
</p>
<p>were sought in the late 1960s, leading to development of the multi-solution methods starting with the
</p>
<p>program MULTAN, now superseded by other systems, particularly SHELX which we discuss in a
</p>
<p>subsequent section. Where automatic methods fail, and sometimes they do, it is important to recall how
</p>
<p>structures like tubercidin were solved by an intelligent manual approach.
</p>
<p>8.2.12 Signs of Trouble, and Past Remedies When
the Structure Failed to Solve
</p>
<p>We summarize some of our own experiences, and record factors involved when a structure solution
</p>
<p>failed to emerge from direct phase-determining methods. Although these comments apply primarily
</p>
<p>to earlier program systems, similar considerations apply to other comparable procedures:
</p>
<p>1. Too few &sum;2 interactions are being used. Increase the number of interactions by lowering jEminj.
2. The origin-defining set is poor or incorrect. Try another one, choosing it with the aid of the rules
</p>
<p>given.
</p>
<p>3. The E map contains one very large peak. The phases are probably very inaccurate; the heavy peak
</p>
<p>may be located in the center of a closed ring. Start again; do not waste time trying to interpret the
</p>
<p>E map.
</p>
<p>4. The E map is not interpretable or chemically sensible. The are incorrect. Try again.
</p>
<p>5. If heavy atoms are present in the structure, they alone may show up. Proceed to Fourier methods
</p>
<p>using interpretable heavy-atom sites. A check against the Patterson function might prove useful
</p>
<p>here.
</p>
<p>6. Only a small molecular fragment is discernible on the E map. Try recycling basing phases on the
</p>
<p>fragment found, or try to obtain more phases by increasing the initial data set.
</p>
<p>7. The program selects an incorrect or poor starting set, such as too few&sum;2 interactions. Select your
</p>
<p>own starting set. If you suspect that a program may contain a fault, inform the author; do not
</p>
<p>attempt to correct it.
</p>
<p>8. The solution still fails to emerge. Review the calculation of the jEj values; perhaps omit reflections
that appear to have a bad influence on the phase-determining pathway.
</p>
<p>9. All fails. Go back to fundamentals. Check the space group, data collection, processing, and any
</p>
<p>other factor that might be at fault.
</p>
<p>If you exhaust these possibilities without achieving success, try another method for determining
</p>
<p>the structure. Or give it a rest and try again later&mdash;or study recent Bibliography on direct methods.
</p>
<p>8.2.13 Triplets, Quartets, and the SHELX Program Strategy [15]
</p>
<p>We discuss in Appendix E and Sect. 8.2.2 entities known as structure invariants and structure
</p>
<p>seminvariants. As we have shown, they may comprise one or more reflections, most often three or
</p>
<p>four in practice, and have important properties that are related to the choice of origin, or permitted
</p>
<p>372 8 Direct Methods and Refinement</p>
<p/>
</div>
<div class="page"><p/>
<p>origin when symmetry is present. In this section, we examine how these invariants may be used to
</p>
<p>enhance direct methods of phase determination. The structure factor equation has been expressed as
</p>
<p>F&eth;hkl&THORN; &frac14;
XN
</p>
<p>j&frac14;1
gj exp&frac12;i 2p&eth;hxj &thorn; kyj &thorn; lzj&THORN;� &frac14; jF&eth;hkl&THORN;j exp if&eth;hkl&THORN; (8.47)
</p>
<p>where N is the number of atoms in one unit cell. In the following discussions it will generally be
</p>
<p>assumed, for simplicity, that we are dealing with an equal-atom structure.
</p>
<p>Referring to Sect. 8.2.2 and Fig. 8.1, it can be seen that changing the origin to any point (Dx, Dy,
</p>
<p>Dz), changes each atom coordinate to (xj � Dx, yj � Dy, zj � Dz), and that the structure factor in
</p>
<p>(8.47) will change to a new structure factor F0(hkl), given by
</p>
<p>F0&eth;hkl&THORN;
XN
</p>
<p>j&frac14;1
</p>
<p>fgj exp&frac12;i2p&eth;hxj &thorn; kyj&thorn;lzj&THORN;�gfexp&frac12;�i2p&eth;hDx&thorn; kDy&thorn; lDz&THORN;�g (8.48)
</p>
<p>where {exp[�i2p(hDx + kDy + lDz)]} is a term external to the summation, so that
</p>
<p>F0&eth;hkl&THORN; &frac14; F&eth;hkl&THORN;fexp&frac12;�i2p&eth;hDx&thorn; kDy&thorn; lDz&THORN;�g (8.49)
</p>
<p>Thus, changing the origin to the point (Dx, Dy, Dz) causes a change in &rsquo;(hkl) given by
</p>
<p>Df&eth;hkl&THORN; &frac14; �2p&eth;hDx&thorn; kDy&thorn; lDz&THORN; (8.50)
</p>
<p>jF(hkl)j is, of course, invariant, as is the intensity of reflection, I(hkl). If we now consider the structure
</p>
<p>factors for two reflections where the two reflections have indices h1k1l1 and h2k2l2, we show in
</p>
<p>Appendix E that for a product of two structure factors, in the absence of any symmetry, if
</p>
<p>h1 + h2 &frac14; 0, then it follows that the phase sum (&rsquo;1 + &rsquo;2) is a structure invariant. This result can
be generalized for three and four (or more) reflections, thus:
</p>
<p>&eth;f1 &thorn; f2 &thorn; f3&THORN; is a structure invariant if h1 &thorn; h2 &thorn; h3 &frac14; 0 (8.51)
</p>
<p>&eth;f1 &thorn; f2 &thorn; f3 &thorn; f4&THORN; is a structure invariant if h1 &thorn; h2 &thorn; h3 &thorn; h4 &frac14; 0 (8.52)
</p>
<p>The presence of symmetry normally requires that the origin be chosen on a symmetry element.
</p>
<p>Thus, in P�1, for example, we have:
</p>
<p>&eth;f1 &thorn; f2 &thorn; f3&THORN; is a structure invariant if h1 &thorn; h2 &thorn; h3 &frac14; 0 (8.53)
</p>
<p>but if h1 + h2 + h3 &frac14; 0 modulo (2 2 2) the corresponding phase sum is a structure seminvariant. We
discuss these invariants more fully in Appendix E.
</p>
<p>Notice that the individual phases in these expressions are not themselves invariant, and that
</p>
<p>nothing has been said so far about the actual values of these phase sums. For (8.51), we can show
</p>
<p>that we have assumed tacitly, Sect. 8.2, that, for sufficiently large values of a, that is, of
</p>
<p>2N�1=2jEh1 jjEh2 jjEh3 j,
</p>
<p>f1 &thorn; f2 &thorn; f3 � 0 (8.54)
</p>
<p>8.2 Direct Methods of Phase Determination 373</p>
<p/>
</div>
<div class="page"><p/>
<p>Rearranging (8.54) gives:
</p>
<p>f1 � �f2 � f3 (8.55)
</p>
<p>We make a small change in notation, so as to accord with that used in the SHELX-97 program
</p>
<p>system and to focus on determining &rsquo;h. Let h1 &frac14; h, �h2 &frac14; k, �h3 &frac14; h � k; then if Friedel&rsquo;s law
holds, that is f&eth;hkl&THORN; &frac14; �f&eth;�h �k �l&THORN;, then (8.55) becomes
</p>
<p>fh � fk &thorn; fh�k (8.56)
</p>
<p>which is the phase addition formula (8.24) from which the tangent formula (8.28) was developed as a
</p>
<p>general means of exploiting TPRs for expanding and refining phases by direct methods. Equation
</p>
<p>(8.52) represents a four-phase quartet, and like three-phase triplets, may be a source of phase
</p>
<p>information, since it can be shown [16, 17] that for a sufficiently large value of the expression
</p>
<p>2N�1jEh1 jjEh2 jjEh3 jjEh4 j
</p>
<p>cos&eth;f1 &thorn; f2 &thorn; f3 &thorn; f4&THORN; � &thorn;1 (8.57)
</p>
<p>provided that jEh1&thorn;h2 j, jEh1&thorn;h3 j, and jEh1&thorn;h4 j are all large. Alternatively,
</p>
<p>cos&eth;f1 &thorn; f2 &thorn; f3 &thorn; f4&THORN; � �1 (8.58)
</p>
<p>if jEh1&thorn;h2 j, jEh1&thorn;h3 j, and jEh1&thorn;h4 j are all small. Equations (8.57) and (8.58) are positive quartets and
</p>
<p>negative quartets (NQRs), respectively. Thus, the sum of four phases is dependent not only on the
</p>
<p>intensities of the four corresponding reflections, the primary terms, but also on the intensities of three
</p>
<p>other index-related reflections, the cross-terms. As in the treatment for triplets, we need to cast the
</p>
<p>indices of the phase-quartet in terms of a target reflection h. The quartet phase sum then becomes
</p>
<p>&eth;fh &thorn; f�k &thorn; f�l &thorn; f�h&thorn;k&thorn;l&THORN; (8.59)
</p>
<p>where h &frac14; h1, �k &frac14; h2, �l &frac14; h3, and �h + k + l &frac14; h4. The primary terms are jEhj, jEkj, jElj, and
jE�h+k+lj, which should all be large; the cross terms are jEh�kj, jEh�lj, and jEk+lj, which should all be
</p>
<p>large for a positive quartet and small for an NQR.
</p>
<p>We have seen in Sect. 8.2.11 that symbolic-addition techniques for phase determination employ a
</p>
<p>number of initial phase assignments for a relatively small number of reflections. Each phase set is
</p>
<p>then expanded via the tangent formula, which is based on the use of strong triplets. Only a very small
</p>
<p>number of such assignments can be expected to converge to produce a phase set that is somewhere
</p>
<p>near the correct one. To avoid the use of the time-consuming inspection of a large number of E maps,
</p>
<p>experience has shown that, with large structures, results based only on the consistency of triplet
</p>
<p>relationships may not discriminate the correct phase set. This failure has led to the use of NQRs, as
</p>
<p>well as triple-phase relations, for good phase determination.
</p>
<p>8.2.14 The SHELX Computer Program System
</p>
<p>The SHELX program suite, one of the most popular and widely used set of crystallographic programs,
</p>
<p>was developed originally from SHELX-76 which was specifically aimed at the determination and
</p>
<p>374 8 Direct Methods and Refinement</p>
<p/>
</div>
<div class="page"><p/>
<p>refinement of small-molecule structures [18, 19]. The latest version, SHELX-97 [20&ndash;22], is available
</p>
<p>free of charge to academic users by registering online [23]. It is recommended that the program be
</p>
<p>incorporated by the user into a specialized program system such as WinGX [22] which is download-
</p>
<p>able [24]. SHELX and other programs have been integrated into WinGX to produce a comprehensive
</p>
<p>work-base for small-molecule crystallography. The implementation of the SHELX programs for both
</p>
<p>large and small molecules is fully covered in the usermanual.When usingWinGX, the crystallographer
</p>
<p>has access to the SHELX direct methods structure-solving programs SHELXS-86, SHELXS-97,
</p>
<p>SHELD, and PATSEE; see Sect. 8.3, which employs a Patterson Search method. The SHELXL
</p>
<p>programs are relatively easy to use and require only two input files:
</p>
<p>&bull; Atoms and instructions
</p>
<p>&bull; Reflection data
</p>
<p>The instruction file includes initially only unit-cell dimensions, wavelength, atom types and their
</p>
<p>numbers, space group information, and the type of input data being used. The programs are written so
</p>
<p>as to handle all space groups in all settings, both standard and nonstandard, and are thus perfectly
</p>
<p>general. This has been achieved by requiring the coordinates of the equivalent positions in the unit
</p>
<p>cell to be specified together with a centrosymmetric/non-centrosymmetric flag, and therefore
</p>
<p>demands a good working knowledge of crystallography.
</p>
<p>The reflection data file contains h,k,l, Fo
2 and s(Fo
</p>
<p>2) as output by the data collection system. If
</p>
<p>using SHELX with WinGX, the user has access also to graphics and editing facilities and programs
</p>
<p>for monitoring the quality of the structure analysis prior to deposition and publication of the data.
</p>
<p>Users are advised to read the operation manuals of both SHELX-97 and WinGX very carefully and to
</p>
<p>keep an eye open for changes in either system.
</p>
<p>8.2.15 The WinGX Program System
</p>
<p>WinGX is an MS-Windows system of programs for solving, refining, and analyzing single crystal X-
</p>
<p>ray diffraction data for small molecules. It provides a consistent and user-friendly Graphical User
</p>
<p>Interface for some of the best publically available crystallographic programs, and has interfaces to
</p>
<p>other popular programs such as SHELX-97 and SirWare programs (SIR-97, SIR-2008). Users of
</p>
<p>WinGX must be registered users of the SHELX and SirWare programs. It is the responsibility of the
</p>
<p>user to register with the appropriate sources as per instructions; WinGX can be downloaded free of
</p>
<p>charge [23].
</p>
<p>Installation of the system is fully described in the manual. The installation stage includes setting up
</p>
<p>of environment variables which may need the help of an expert. Once installed and called from the PC
</p>
<p>desktop shortcut, the menu shown in Fig. 8.10a is displayed. In this example, the project name is
</p>
<p>BW202W92(R) and the necessary files BW202W92(R).ins and BW202W92(R).hkl are stored in a
</p>
<p>folder called BW202W92(R) on the C:\ drive. Other menus in WinGX to which we shall refer are
</p>
<p>shown in Fig. 8.10b, c.
</p>
<p>8.2.16 Direct Methods in the Program SHELX-97 for Small Molecules
</p>
<p>We are concerned at this point with methods used to determine small-molecule crystal structures
</p>
<p>using the SHELX-97 system, but hasten to add that this system is in fact now capable of handling both
</p>
<p>small-molecule and protein structure analyses. For these purposes it contains a number of executable
</p>
<p>programs, such as those that solve structures by either Patterson or direct methods, carry out detailed
</p>
<p>least-squares refinement, and locate water molecules of crystallization in proteins. The heavy-atom
</p>
<p>8.2 Direct Methods of Phase Determination 375</p>
<p/>
</div>
<div class="page"><p/>
<p>method can be invoked, and the Patterson search method used is effectively PATSEE, which we
</p>
<p>describe in Sect. 8.3.6. The least-squares procedures encompass inter alia dispersion, absorption, and
</p>
<p>extinction corrections, together with a wide range of constraints and restraints, and routines for fixing
</p>
<p>and refining hydrogen-atom positions. The direct methods routine is based on a random start multi-
</p>
<p>solution strategy, or more accurately a multi-permutation single-solution procedure, since it endea-
</p>
<p>vors to identify the correct solution and then to improve on it by E map partial structure extension, or
</p>
<p>successive refinement. Geometry routines calculate bond lengths, bond angles, and torsion angles and
</p>
<p>identify possible hydrogen bonds, all of the results being tabulated for publication purposes. The
</p>
<p>system is strongly recommended to the serious structure analyst.
</p>
<p>The triple-phase and NQR relationships discussed above are employed in the SHELX-97 to
</p>
<p>generate phases by a modified tangent formula. We write
</p>
<p>a &frac14; 2N�1=2jEhjEkEh�k (8.60)
</p>
<p>h &frac14; gN�1jEhjEkElEh�k�l (8.61)
</p>
<p>where Ej &frac14; jEjj(cos &rsquo;j + i sin &rsquo;j), (j &frac14; k, l), and g is a positive constant set by the program to account
for the cross-term jEj values; (8.60) and (8.61) are subject to the same conditions as (8.59). It has been
found that computer time is optimized for cases where the number of NQRs is restricted to between
</p>
<p>1000 and 8000. Only the most reliable relationships are retained in this process, strictly where all three
</p>
<p>cross-terms have actually been measured and found to be weak. However, all interconnecting triple-
</p>
<p>phase relationships are used, except for &sum;1 terms, Sect. 8.2.8, and those which all involve restricted
</p>
<p>phases that prevent the resultant phase from being zero [25].
</p>
<p>A process of phase annealing, based on a principle similar to that of simulated annealing used in
</p>
<p>the refinement of macromolecular structures, Sect. 10.9.1, is employed in the next stage of phase
</p>
<p>refinement. The results from these two stages are then applied to a full tangent formula refinement for
</p>
<p>the best retained reflections. The total number of different attempts using these procedures can be set
</p>
<p>by the user and may be as many as 5000 for really difficult structures.
</p>
<p>Fig. 8.10 The WinGX
program system. (a)
WinGX Menu showing the
</p>
<p>Basic Functions available.
</p>
<p>(b) Selection of SHELXS-
86 direct methods program
</p>
<p>from the WinGX menu.
</p>
<p>(c) Selection of the
graphics facility
</p>
<p>SXGRAPH in WinGX is
</p>
<p>activated by the option
</p>
<p>&ldquo;Model&rdquo; on the main menu
</p>
<p>(a); see also Fig. 8.11b, c
</p>
<p>376 8 Direct Methods and Refinement</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec46_10">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec46_10</a></div>
</div>
<div class="page"><p/>
<p>Initial Stages
In the initial stages of applying SHELX-97 to a structure determination, a number of cycles of
</p>
<p>weighted tangent formula (8.28) are performed, starting with a selected number of randomly
</p>
<p>generated phases. The best phase sets, as judged by NQR and triplet consistency, are retained
</p>
<p>and the process repeated to give a number of parallel-generated phase sets. After each iteration, the
</p>
<p>total number of phase sets processed in parallel is reduced until only 25% of the original number of
</p>
<p>phase sets is generated. Typically 8 or 16 best phase sets are retained from each cycle for a total run of
</p>
<p>128 parallel permutations. The best reflections and strongest TPRs are retained and passed to the next
</p>
<p>stage of the procedure. The program uses a TPR figure of merit (FOM) based on the indicator NQUAL
</p>
<p>(a negative quantity), where the best phase set has the smallest value:
</p>
<p>NQUAL &frac14; Sja 
 hj=Sjajjhj (8.62)
</p>
<p>Phase Annealing
This method is used to refine the phase sets retained after the initial stages. Phase annealing supplies a
</p>
<p>correction to the phase produced by the tangent formula. As only a limited number of reflections
</p>
<p>are involved, the process uses computer time efficiently, employing only the strongest triplets and
</p>
<p>quartets.
</p>
<p>Final Stages in Phase Determination
The total number of directmethods attempts (the variable np on theTREF instruction in SHELX-97) to be
</p>
<p>employedmay have to be a very high value for difficult structures. The routines in SHELX-97 are written
</p>
<p>with efficiency inmindand the program requires only a fewminutes on a PC for calculations onquite large
</p>
<p>structures, evenwhen TREF is as high as 5000. The programwill select the number nE of reflections to be
</p>
<p>involvedat this stage if not otherwise stipulated, butnEmayhave to be reset to ahighervalue if the program
</p>
<p>fails to produce an interpretable E map. An example of the use of this program in solving a difficult
</p>
<p>structure (cyclosporin H) is given in Sect. 11.10.
</p>
<p>8.2.17 Example of a SHELX-97 Structure Solution: Crystal Code Name
BW202W92(R)
</p>
<p>The SHELXS-97 direct determination described here was undertaken as part of the study of the
</p>
<p>mesylate salts of a novel voltage-gated sodium channel-binding ligand R-(�)-BW202W92 and its
much less active S-(+)-enantiomer (BW203W92) (Fig. 8.11a) in order to determine the absolute
</p>
<p>configuration of each, a factor which is of vital importance to a full understanding of the biological
</p>
<p>activity of the drug. In addition each enantiomer exists as two distinct atropisomeric forms in the solid
</p>
<p>state, i.e., each has two independent molecules per asymmetric unit with slightly different configu-
</p>
<p>rational properties [26]. The solution for the R-form is described below.
</p>
<p>Crystals of BW202W92(R) mesylate salt, C11H9Cl3FN4&middot;CH3SO3 Fig. 8.11a (I), are monoclinic in
</p>
<p>space group P21, with a &frac14; 8.384(2) Å, b &frac14; 16.984(3) Å, c &frac14; 12.480(3) Å, b &frac14; 104.14(2)
�, and with
</p>
<p>two independent molecules of each type per asymmetric unit. The X-ray data collected with Cu Ka
</p>
<p>radiation had 3192 independent hkl data to a resolution of 0.800 Å (ymax &frac14; 74.42�). The files
BW202W92(R).ins and BW202W92(R).hkl were stored in the folder BW202W92(R) which was read
</p>
<p>intoWinGX as described in the previous section, Fig. 8.10a. The program SHELXS-86 was selected by
</p>
<p>using the &ldquo;Solve&rdquo; option on the WinGX menu, Fig. 8.10b. The resulting E map from this calculation,
</p>
<p>obtained using model/SXGRAPH, Fig. 8.10c, from the WinGX menu, is shown in Fig. 8.11b.
</p>
<p>8.2 Direct Methods of Phase Determination 377</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec32_11">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec32_11</a></div>
</div>
<div class="page"><p/>
<p>The program SHELXS-97 was then implemented in a similar manner and resulted in the map
</p>
<p>shown in Fig. 8.11c. The input data for the SHELXS-97 run is shown in Table 8.11a and the output
</p>
<p>data in Table 8.11b, both with explanatory notes.
</p>
<p>In all, 256 phase sets were refined; the best fit corresponds to the value 0.0503, a very promising
</p>
<p>and acceptable result. However, in the final analysis it is the electron density map and subsequent
</p>
<p>refinement which provide the final proof of correctness of the direct phase set. The RE value for this
</p>
<p>best phase set prior to examination of the electron density had a value of 22.4% which is very
</p>
<p>satisfactory at that stage.
</p>
<p>Fig. 8.11 (a) Top:
schematic chemical
</p>
<p>structure for the sterically
</p>
<p>hindered rotamers of I and
</p>
<p>II, respectively. Bottom:
</p>
<p>general numbering scheme
</p>
<p>used for the enantiomeric
</p>
<p>compounds (R-
</p>
<p>form &frac14; BW202W92,
S-form &frac14; BW203W92).
Note that the pyrimidine
</p>
<p>ring is N10-protonated in
each of the mesylate acid
</p>
<p>addition salts. (b) E map of
BW202W92(R) from
</p>
<p>SHELXS-86. The structure
</p>
<p>is essentially correct but the
</p>
<p>heavy atoms Cl and S are
</p>
<p>not identified and one
</p>
<p>molecule is fragmented,
</p>
<p>that is, the required
</p>
<p>symmetry operation to join
</p>
<p>the two halves up has not
</p>
<p>been applied. (c) E map of
BW202W92(R) from
</p>
<p>SHELXS-97. Heavy atoms
</p>
<p>are recognized as such but
</p>
<p>apart from one S atom are
</p>
<p>incorrectly atom-typed.
</p>
<p>The structure is essentially
</p>
<p>correct apart from this fault
</p>
<p>which was easily rectified
</p>
<p>378 8 Direct Methods and Refinement</p>
<p/>
</div>
<div class="page"><p/>
<p>It is of interest to compare the resulting E maps from SHELXS-86 (Fig. 8.11b) and SHELXS-97
</p>
<p>(Fig. 8.11c). The SHELXS-97 program recognizes that the structure contains the heavy atoms Cl
</p>
<p>(chlorine) and S (sulfur) and has attempted to assign the highest peaks in the electron density
</p>
<p>calculated from the best phase set as either Cl or S (Fig. 8.11c). From Fig. 8.11c we can confirm
</p>
<p>that the program has in fact assigned the correct peaks as heavy atoms but has not been able to
</p>
<p>correctly identify all of the atom types: only one S atom has been correctly assigned as such. The
</p>
<p>earlier program SHELXS-86, the results of which are shown in Fig. 8.11b, did not attempt to make
</p>
<p>these heavy-atom assignments at all, and in addition did not apply the necessary symmetry relation-
</p>
<p>ship needed to bring the two halves of one of the molecules together. In both respects SHELX-97 is
</p>
<p>the superior program. Figure 8.11c shows the peaks generated by SHELXS-97 and displayed by the
</p>
<p>WinGX facility model/SXGRAPH, which has the following features: three-dimensional rotation,
</p>
<p>translation,magnification, and demagnification of the set of peaks displayed; calculation of bond lengths,
</p>
<p>bond angles and torsion angles for selected peaks; deletion of unwanted peaks; assignment of individual
</p>
<p>Table 8.11a Input data for crystal BW202W92(R) to SHELXS-97
</p>
<p>Input file BW202W92(R).ins
</p>
<p>TITL BW202W92(R)
CELL 1.54180 8.385 16.984 12.481 90.000 104.15 90.000
ZERR 4.00 0.002 0.003 0.003 0.000 0.02 0.000
LATT �1
SYMM � X, 1/2 + Y, - Z
SFAC C H O N CL S F
UNIT 48 48 12 16 12 4 4
HKLF 3
END
</p>
<p>Explanation of input file BW202W92(R).ins
</p>
<p>TITL Title
CELL Wavelength and unit-cell parameters
ZERR Number of chemical formula units per unit cell, with esds
LATT Non-centrosymmetric, -1 (centrosymmetric, 1)
SYMM Equivalent position(s) related to x,y,z (space group is P21)
SFAC Atom types in the structure
UNIT Number of each atom type in the unit cell (in the order of the previous
</p>
<p>line)
HKLF Data input order: h,k,l, Fo, and s(Fo)
</p>
<p>Note: This HKLF input is not the recommended form. The more usual input data
</p>
<p>would be HKLF 4 corresponding to h,k,l, Fo
2, and s(Fo
</p>
<p>2). These data were
</p>
<p>collected on a CAD4 diffractometer, which produces output data as Fo and
</p>
<p>s(Fo); thus, the input is restricted to HKLF 3. However, refinement is still
</p>
<p>carried out on jFj2
</p>
<p>Input file BW202W92(R).hkl
</p>
<p>�8 0 0 43.79 1.30 1
�8 1 0 8.19 2.75 1
�8 3 1 11.18 2.59 1
�8 1 1 22.52 1.67 1
�8 0 2 48.56 1.28 1
�8 2 2 29.63 1.57 1
�8 3 2 10.44 2.63 1
�8 4 2 13.78 2.03 1
�8 4 3 7.48 4.08 1
�8 3 3 20.62 1.65 1
�8 2 3 13.71 2.33 1
. . .and so on, to the end of the hkl data set. At the end of each line is the batch
number, which is usually 1 for contemporary data measurement schemes. A total
</p>
<p>of 3192 hkl values formed this data set, corresponding to a resolution of 0.800 Å
</p>
<p>8.2 Direct Methods of Phase Determination 379</p>
<p/>
</div>
<div class="page"><p/>
<p>atom numbers, types and thermal parameters; saving of the present model as the new .ins file (for
</p>
<p>refinement); and initiation of structure refinement. After completion of the refinement, programs within
</p>
<p>WinGX assess the quality of refinement, produce files (.cif)3 for checking and publication, and prepare
</p>
<p>high quality diagrams, either using the programs ORTEP/RASTER inWinGX or from the .cif3 file or a .
</p>
<p>pdb file, using freely downloadable programs such as MERCURY or RASMOL (see Appendix D for
</p>
<p>references to these programs).
</p>
<p>8.3 Patterson Search Methods
</p>
<p>In our earlier discussion in this chapter, we showed that a Patterson synthesis must contain a complete
</p>
<p>set of peaks, that is,N2 � N non-origin peaks for a crystal containingN atoms in the unit cell. Since we
can always calculate the vector set for a model structure, or just a part of it, the Patterson function could
</p>
<p>be unscrambled, wholly or partially, in terms of a set of atomic coordinates. This idea has led to a
</p>
<p>technique in structure analysis called Patterson search methods.
</p>
<p>Table 8.11b Selected output from the SHELX-97 direct methods calculations with WinGX
</p>
<p>Summary of parameters for BW202W92(R)
</p>
<p>ESEL jEjmin 1.200 jEjmax 5.000 DelU 0.005 renorm 0.700 axis 0
OMIT s 4.00 2theta(lim) 180.0
INIT nn 15 nf 16 s+ 0.800 s- 0.200 wr 0.200
PHAN steps 10 cool 0.900 Boltz 0.300 ns 282 mtpr 40 mnqr 10
TREF np 256. nE 420 kapscal 0.800 ntan 3 wn -0.750
FMAP code 8
PLAN npeaks -65 del1 0.500 del2 1.500
MORE verbosity 1
TIME t 9999999.
</p>
<p>This first part of the output consists of various parameters which in this case have been set by default. For difficult
</p>
<p>structures some manual intervention may be necessary
282 Reflections 4356 unique TPR for phase annealing
420 Phases refined, using 12351 unique TPR used
443 Reflections 13782 unique TPR for R(alpha)
7416 Unique negative quartets found 3768 used for phase refinement
Highest memory used to derive phase relations 5424 / 71182
This second selected part of the output lists some information about the direct methods results, including how many
</p>
<p>triplet phase relationships (TPRs) have been found. Valid phases have been calculated from a random starting phase set
</p>
<p>using TPRs and NQRs
</p>
<p>Measure of fit results
</p>
<p>FOM range Frequency
</p>
<p>0.000 - 0.020 0
0.020 - 0.040 0
0.040 - 0.060 184
0.060 - 0.080 0
0.080 - 0.100 2
0.100 - 0.120 15
0.120 - 0.140 4
0.140 - 0.160 3
0.160 - 0.180 6
0.180 - 0.200 5
0.200 - 0.220 1
0.220 - 0.240 and so on
</p>
<p>3Crystallographic Information File.
</p>
<p>380 8 Direct Methods and Refinement</p>
<p/>
</div>
<div class="page"><p/>
<p>8.3.1 General Comments for Small Molecules and Macromolecules
</p>
<p>We introduce here a method for structure determination that is useful and applicable either to small
</p>
<p>molecules, for which the method would normally be used in case of failure of direct methods, or to
</p>
<p>macromolecules, for which the method is now frequently used as a first choice where possible.
</p>
<p>The method may be designated Patterson Search for small molecules and Molecular Replacement
</p>
<p>(MR) in the case of macromolecules. In order to apply the method to solve a structure, the initial
</p>
<p>requirements are:
</p>
<p>1. For the crystal under investigation, the target structure, a set of Fo(hkl) data to as high a resolution
</p>
<p>as possible, which would normally be atomic resolution for small molecules, but generally less
</p>
<p>for a macromolecular crystal because such crystals tend to diffract X-rays weakly, Sects. 4.1
</p>
<p>and 10.4.7
</p>
<p>2. The availability of the coordinates of a good quality structure, the search structure, which forms a
</p>
<p>relatively small fragment of the target structure for small molecules, whereas with macromole-
</p>
<p>cules the search structure should ideally be similar in size and structure to the target molecule and/
</p>
<p>or with at least 40% sequence homology
</p>
<p>3. A sound understanding of the principles and practices involved
</p>
<p>4. State-of-the-art software and hardware
</p>
<p>Conceptually the basic principle of Patterson Search or Molecular Replacement (MR) is quite
</p>
<p>straightforward. Remember, we know that the intensity data Fo
2(hkl) contain phase information. The
</p>
<p>MR method as proposed by Rossman and Blow [27] for protein structures involves a critical and
</p>
<p>quantitative comparison of the Patterson functions of the target and search models. Similar com-
</p>
<p>ments apply to a Patterson search method implemented successfully some years ago by Braun et al.
</p>
<p>in the Vector Verification Method [28] for small molecules. Although quite successful and relatively
</p>
<p>easy to use, this method has now been superseded by the program PATSEE, which will be described
</p>
<p>in detail below.
</p>
<p>In contemporary software for carrying out MR applications, the method is strengthened through
</p>
<p>the use of a variety of other lengthy crystallographic techniques, which are now within the capabilities
</p>
<p>of modern computers. For example, with proteins:
</p>
<p>1. In the MR module of CCP4 [29] an automated procedure called MrBump is used
</p>
<p>2. Model generation is used with the program Chainsaw
</p>
<p>3. The program Phaser is used in CCP4
</p>
<p>Similar routines are used with the programs:
</p>
<p>4. MOLREP (developed from AmoRe)
</p>
<p>5. BALBES
</p>
<p>6. X-PLOR
</p>
<p>7. CNS
</p>
<p>In a similar manner, PATSEE is strengthened by the use of direct methods.
</p>
<p>The Patterson function has peaks of high density at locations corresponding to the ends of
</p>
<p>atom&ndash;atom vector pairs, with one atom of each pair occupying the common origin. For complex
</p>
<p>structures like proteins, the interatomic vectors are densely packed in the unit cell, and most of them
</p>
<p>will not be resolved in the Patterson map; lack of atomic resolution in the X-ray data will also cause a
</p>
<p>further blurring of the vector distribution density. The two atoms forming the Patterson vector can be
</p>
<p>in either the same molecule, intramolecular, or between atoms in either symmetry-related molecules
</p>
<p>or nonsymmetry-related molecules, intermolecular.
</p>
<p>8.3 Patterson Search Methods 381</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec27_10">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec27_10</a></div>
</div>
<div class="page"><p/>
<p>8.3.2 Intramolecular Interatomic Vectors and Molecular Orientation
</p>
<p>Intramolecular vectors tend to be shorter than the intermolecular vectors that span different molecules
</p>
<p>in the unit cell. This self-vector set, as it is also known, is consequently situated around the origin of
</p>
<p>the Patterson function, the longest vectors arising between atoms at extreme ends of the molecule.
</p>
<p>Each atom, in theory, images [30] both the structure and its inverse, that is, vector types AB and BA.
</p>
<p>Because of the lack of resolution in most protein structures, this will be in the form of a blurredmolecular
</p>
<p>envelope of density, whereas for small molecules at atomic resolution the individual images will bemuch
</p>
<p>more clearly defined. There will be one such image of the structure, plus its inverse, per atom, forming a
</p>
<p>centrosymmetric distribution. Figure 8.12 shows two similar (homologous) two-dimensional &ldquo;mole-
</p>
<p>cules,&rdquo; in which S represents a suitable search molecule, and T the target molecule whose structure is
</p>
<p>being determined. In the case of a small molecule we can consider the four or five discs to represent
</p>
<p>resolvable atoms. For macromolecules, the surrounding sheaths in these diagrams represent the overall
</p>
<p>molecular shape that MR technique is seeking.
</p>
<p>Figure 8.13 shows the two actual structures in their different unit cells. In small-molecule
</p>
<p>structures, where the search and target molecules include a relatively small part, maybe 50%, of
</p>
<p>the structure in common, we would expect the two unit cells and space groups to be different. For
</p>
<p>macromolecular structures, which are more similar to each other, this need not be so, although
</p>
<p>frequently it is. As can be seen in this example, the known search structure S is based on a non-
</p>
<p>orthogonal unit cell, while the unit cell for the target structure T is orthogonal. Both cells incorporate
</p>
<p>twofold symmetry; Fig. 8.13b shows some of the intermolecular interatomic vectors, as discussed in
</p>
<p>Sect. 8.3.3. For the two molecules S and T, Fig. 8.14 shows the corresponding resolved self-vector set
</p>
<p>peak positions in the Patterson functions for small molecules and the corresponding simulated vector
</p>
<p>sheaths representing the envelope of the Patterson function, for macromolecules; for clarity only
</p>
<p>those peaks not related by the Patterson center of symmetry are shown.
</p>
<p>Rotation Stage of Patterson Search or Molecular Replacement
In the above simulated example the vectors defining the orientation of the unknown structure T in its
</p>
<p>unit cell, Fig. 8.14b, can be seen to occur in the Patterson of the known search structure S, Fig. 8.14a.
</p>
<p>To demonstrate this fact the reader should make a transparent copy of Fig. 8.14a, which shows the
</p>
<p>Patterson peaks of the known molecule S, and place it over Fig. 8.13b, the Patterson function of the
</p>
<p>unknown structure T. Locate the two diagrams such that their origins are in register, and show that
</p>
<p>the maximum correspondence occurs for an anticlockwise rotation of your copy by 67�.
</p>
<p>Fig. 8.12 (a) Known, search model S to be used in Patterson Search or Molecular Replacement. (b) The unknown
target molecule T, that is similar to the search model but has an extra structural feature
</p>
<p>382 8 Direct Methods and Refinement</p>
<p/>
</div>
<div class="page"><p/>
<p>Fig. 8.13 (a) The search
model S, and (b) the target
molecule T packed in their
</p>
<p>respective unit cells. It is
</p>
<p>necessary that the structure
</p>
<p>of S is known and that a set
</p>
<p>of atomic coordinates is
</p>
<p>available. The structure
</p>
<p>(b) is to be determined by
Patterson Search or
</p>
<p>Molecular Replacement.
</p>
<p>Some of the intermolecular
</p>
<p>Patterson vectors are
</p>
<p>shown in (b)
</p>
<p>Fig. 8.14 Interatomic vectors for (a) the search model S, and (b) the target molecule T. Only those vectors
independent of the Patterson center of symmetry are shown. Both diagrams contain multiple images, or part images,
</p>
<p>of structures. There is a rotational relationship between the two Patterson functions that the reader should try to
</p>
<p>determine (see text). Note that the sets of vectors shown here are arranged around the origin of the Patterson function at
</p>
<p>distances corresponding to the distances between atoms within a given molecule. The high vector density which always
</p>
<p>occurs at the Patterson origin has been omitted
</p>
<p>8.3 Patterson Search Methods 383</p>
<p/>
</div>
<div class="page"><p/>
<p>In a real structure determination this orientation angle would be calculated by one of the available
</p>
<p>computer programs, as described below. Note that because the unknown structure Fig. 8.12b contains
</p>
<p>an additional moiety, vectors for this part of the structure are missing in the search Patterson,
</p>
<p>Fig. 8.14a. For small-molecule analysis the missing vectors may form a large fraction of the target
</p>
<p>molecule, whereas for macromolecules only a small fraction would normally be involved. Subsequent
</p>
<p>Fourier and least-squares refinement of the target structure, once the search model has been located,
</p>
<p>will serve both to locate any missing atoms or groups and to eliminate others which may have been
</p>
<p>included, but which do not form part of the target structure.
</p>
<p>In practice the process of matching the orientations of the two Patterson functions, the rotation
</p>
<p>stage, is carried out by computation, testing over a series of orientation angles. For three-dimensional
</p>
<p>structures, it is necessary to perform these rotations about three independent axes, normally called a,
</p>
<p>b, and g&mdash;not to be confused with the unit-cell angles. Axial systems used for the rotation procedure
</p>
<p>differ among different programs, and it is advisable to read the program manual in each case for
</p>
<p>information on this point (see Appendix D). The angular rotation ranges and intervals have to be
</p>
<p>chosen carefully in order to cover a sufficient number of possibilities, thus ensuring that the correct
</p>
<p>angular triplet is not overlooked. Although computationally wasteful, it is better to include too many
</p>
<p>trials rather than too few. In PATSEE it is not unusual to use several thousand random angle triplets in
</p>
<p>the rotation stage of the analysis.
</p>
<p>On account of the complexity of protein structures, the following two conditions apply to the
</p>
<p>rotation function:
</p>
<p>1. In order to limit the Patterson vectors to lengths that include self-vectors (intramolecular vectors)
</p>
<p>but exclude cross vectors (intermolecular vectors), the rotation function R(a, b, g) should be
</p>
<p>calculated over a restricted spherical volume U centered at the origin, and having a radius known
</p>
<p>as the radius of integration.
</p>
<p>2. The large number of values of R(a, b, g), calculated over the required angular range, generally
</p>
<p>contains many peaks in addition to those that belong to the correct solution, and having compara-
</p>
<p>ble magnitudes. These peaks are simply signifying that there is a degree of correspondence
</p>
<p>between the two Patterson functions in this orientation, albeit a wrong one. Because of this
</p>
<p>uncertainty, some programs, such as AmoRe, retain all peaks greater than 50% of the highest
</p>
<p>peak (even more in some programs) for transference to the translation stage of MR. The problem
</p>
<p>is less significant for small molecules but nevertheless must not be overlooked. Figures of merit are
</p>
<p>used to discriminate between the true and false solutions.
</p>
<p>8.3.3 Intermolecular Interatomic Vectors: Translation Stage of MR
</p>
<p>Assuming that the correct orientation of the search fragment or molecule has been determined in the
</p>
<p>rotation stage, the correctly oriented structure must then be located spatially in its true position in the
</p>
<p>unit cell of the target crystal by means of a translation stage. The origin with respect to this translation
</p>
<p>process is usually governed by the space group. Translation is carried out rigorously by placing the
</p>
<p>oriented search fragment in a large number of test positions located on a fine grid. This process must
</p>
<p>cover a sufficient number of finely selected translational increments in three dimensions, designated
</p>
<p>as either Dt1, Dt2, Dt3, or tx, ty, tz, in order to ensure that the correct location of the molecule is
</p>
<p>scanned. In essence, the correct structure is recognized initially as corresponding to the highest degree
</p>
<p>of overlap between the calculated and observed Patterson functions, when superimposed after
</p>
<p>applying the given translation vector.
</p>
<p>During the above process all of the lattice and symmetry operations for the target crystal are fully
</p>
<p>applied. It is therefore absolutely essential that the space group of the target crystal has been correctly
</p>
<p>384 8 Direct Methods and Refinement</p>
<p/>
</div>
<div class="page"><p/>
<p>assigned. If this is not the case there will be no outstanding solution at the translation stage, and further
</p>
<p>refinement of the structure will not be possible. Figure 8.13b, which represents the correct solution for
</p>
<p>the target structure, includes some of the intermolecular interatomic vectors. For comparison the
</p>
<p>corresponding vectors are shown in Fig. 8.15, in which the search molecule S is correctly oriented but
</p>
<p>incorrectly translated into the target unit cell. Very large changes in these vectors are evident and
</p>
<p>would have a profound effect on the quality of this particular translation when tested computationally.
</p>
<p>The changes in how the molecules are packed (see below), while retaining the twofold symmetry and
</p>
<p>unit-cell translations in the structure, should be noted by comparing Figs. 8.13b and 8.15.
</p>
<p>8.3.4 Crystal Packing and Refinement of the Structure
</p>
<p>We consider next the packing of the rotated&ndash;translated (RT)molecules in the unit cell. An incorrect RT
</p>
<p>solution will produce incorrect crystal packing, which can be easily detected from a thorough
</p>
<p>calculation of interatomic distances. What usually happens is that some atoms in different molecules
</p>
<p>will be unacceptably close to each other; see Tables 8.21 and 8.22. Evidence of the correctness of the
</p>
<p>combined RT search results can thus be easily achieved by inspecting the crystal packing generated
</p>
<p>through each promising set of rotation and translation parameters that has been retained, usually
</p>
<p>because the figures of merit are encouraging. Crystal packing can be examined readily with the use of a
</p>
<p>suitable molecular graphics program. For small molecules this can be carried out, for example, in the
</p>
<p>program package WinGX using the routine PLATON99, while for macromolecules the program
</p>
<p>MOLPAK has been found to be useful and reliable (see Appendix D). These programs provide not
</p>
<p>only graphical representations of the packed molecules but also quantitative values for the intermo-
</p>
<p>lecular contacts; in particular, any violent clashes between the symmetry-related molecules that would
</p>
<p>exclude a physically viable solution are flagged.
</p>
<p>Fig. 8.15 A possible solution for Patterson Search or Molecular Replacement, where the correct rotation (or
orientation) has been applied to the search model S but the subsequent translation is incorrect (compare with
</p>
<p>Fig. 8.14b). Three intermolecular vectors are shown for this situation which, because the translation is incorrect, will
</p>
<p>not be present in the Patterson function for the target molecule. Consequently, there will be a lack of correspondence
</p>
<p>between the S and T Patterson functions, resulting in poor figures of merit for this solution; such solutions will therefore
</p>
<p>be rejected
</p>
<p>8.3 Patterson Search Methods 385</p>
<p/>
</div>
<div class="page"><p/>
<p>Expansion and Refinement of the Structure
The final stage of this procedure, as with other methods of structure analysis, attempts to locate atoms
</p>
<p>that are missing from the model and to refine the positions of all atoms to produce a value for the R-
</p>
<p>factor, Sect. 7.6.1, that is compatible with the accuracy of the intensity data. As we have seen, missing
</p>
<p>atoms can be located by calculating the electron density, based on phasing from the partial model.
</p>
<p>Initially the calculated structure factors will be of low accuracy, but with a reasonably sized fragment,
</p>
<p>as will be seen in the following examples, the electron density for the correct solution can be expected
</p>
<p>to reveal at least some of the missing atoms, which are then added to the phasing process and passed
</p>
<p>on to the next cycle.
</p>
<p>Program suites likeWinGX have facilities for checking themolecular geometry at each stage, so as to
</p>
<p>reduce the possibility of adding incorrectly placed atoms to the model. The partial model may be
</p>
<p>subjected to a few cycles of least-squares refinement, as this can accelerate the procedure toward
</p>
<p>convergence. However, it should be noted that only when all atoms are present in the model will the R
</p>
<p>factor be meaningful. This refinement procedure will be designated loosely as &ldquo;Fourier-least-squares,&rdquo;
</p>
<p>see Sect. 8.4.2.
</p>
<p>8.3.5 Patterson Search Methods for Small Molecules
</p>
<p>In the description of the heavy-atom method, Sect. 7.5, we showed that, for a structure containing a
</p>
<p>small number of relatively heavy atoms, the Patterson function can lead to a successful determination
</p>
<p>of the atomic coordinates of the heavy atoms, the remaining atoms being located by subsequent
</p>
<p>calculation of electron density maps prior to refinement. Direct methods can be used to solve light-
</p>
<p>atom structures, for which there are none of the predominant interatomic vectors that would be
</p>
<p>necessary for an application of the heavy-atom method. We have seen that the Patterson function
</p>
<p>necessarily contains the complete set of interatomic vectors for the given crystal structure, in the form
</p>
<p>of peaks located at the ends of such vectors, one end of each vector being located at the origin.
</p>
<p>Reference to Fig. 8.15 shows that in effect each atom forms an image of the structure with itself at the
</p>
<p>origin. When the Patterson center of symmetry is added, there result 2N displaced images of the
</p>
<p>structure present around the origin. A single weight peak, that is, one generated by a given pair of atoms
</p>
<p>1 and 2, will have a value or peak height roughly proportional to the atomic number product Z1Z2.
</p>
<p>The total of N2 � N non-origin Patterson peaks per unit cell containing N atoms increases rapidly
with N: the peaks may become overcrowded and overlapping, and unresolvable for larger structures.
</p>
<p>In this section we describe applications of Patterson methods to approximately equal-atom structures.
</p>
<p>These structures are usually of relatively low molecular weight, containing 30&ndash;40 carbon-like atoms,
</p>
<p>with crystals that diffract to atomic resolution, leading to well-resolved interatomic vectors in the
</p>
<p>Patterson function. Macromolecules, which generally diffract far short of atomic resolution, will be
</p>
<p>discussed in Chap. 10.
</p>
<p>The Patterson function for small molecules necessarily contains the complete set of interatomic
</p>
<p>vectors for the crystal structure. Thinking of the logistics in reverse: for a known light-atom structure
</p>
<p>we could use the atomic coordinates to generate the set of interatomic vector coordinates, and these
</p>
<p>would match the Patterson function within limits of error that are dictated by the quality of the Fo data,
</p>
<p>assuming that peak overlap is not a major problem. Since each atom in the structure forms in the
</p>
<p>Patterson function an image of the structure in itself, Sect. 7.4.2, there areN such images scrambled, or
</p>
<p>convoluted together, in P(uvw). The complexity of the Patterson function generally makes deconvo-
</p>
<p>luting it in terms of the individual atomic coordinates almost impossible unless some additional
</p>
<p>information can be used initially. This information is usually in the form of a reasonably precise
</p>
<p>model of the geometry of a fragment of the molecule whose three-dimensional structure is known.
</p>
<p>386 8 Direct Methods and Refinement</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec32_7">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec32_7</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec22_7">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec22_7</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_10">http://dx.doi.org/10.1007/978-1-4614-3954-7_10</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec14_7">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec14_7</a></div>
</div>
<div class="page"><p/>
<p>The method thus depends on a good knowledge of the chemical identity of the molecule in
</p>
<p>question. Knowing the geometry of such a fragment of the structure, interatomic vectors for the
</p>
<p>fragment may be calculated and searched for in the Patterson function. Coordinates for the molecular
</p>
<p>fragment to be used may be derived from the library of known crystal structures in the Cambridge
</p>
<p>Crystallographic Data Base, or generated from graphics programs, such as the excellent ChemSketch
</p>
<p>[31], which incorporate standard bond lengths and angles; see Sect. 8.7. For example, a suitable
</p>
<p>molecular fragment may simply be a benzene ring, which comprises a flat regular hexagon with sides
</p>
<p>of 1.40 Å and internal angles of 120�. The coordinates for the search model in this case could be
obtained by drawing (see Problem 8.9). If the structure contains a cyclohexane ring, this structure is
</p>
<p>truly three-dimensional, and one of the other two methods mentioned above must be used to derive its
</p>
<p>coordinates; see also Sect. 13.6.6. The examples discussed below both involve somewhat more
</p>
<p>complex search fragments.
</p>
<p>8.3.6 The Program PATSEE
</p>
<p>In the discussion so far, we have established that a Patterson Search in vector space consists of the
</p>
<p>following stages:
</p>
<p>1. Acquisition of a set of accurate atom coordinates for a suitable search model
</p>
<p>2. Calculation and storage of the Patterson function (self-vector set) for the model
</p>
<p>3. Calculation and storage of the Patterson function for the target crystal, based on the Fo(hkl) data
</p>
<p>4. Rotation search, which provides several possible orientations for the search model to occupy in the
</p>
<p>target unit cell, listed in order of likely feasibility, or precedence
</p>
<p>5. Translation search which attempts to place each of the rotation solutions in turn into the correction
</p>
<p>position in the target unit cell
</p>
<p>6. Refinement and expansion of the rotated and translated models, again in order of precedence, to
</p>
<p>finally converge on the correct solution for the target structure
</p>
<p>The program PATSEE follows stages (1)&ndash;(5) in the above list, using some important and powerful
</p>
<p>new algorithms in its implementation. Stage (6) reverts to the standard procedures for structure
</p>
<p>refinement described in the next section. The PATSEE program is highly recommended, is easy to
</p>
<p>use, and readily available to crystallographers, being incorporated into the structure analysis suite
</p>
<p>WinGX. As with other methods that utilize Patterson Search, the program determines the orientation
</p>
<p>of the search fragment from the rotation function. However, instead of then using the conventional
</p>
<p>translational function, direct methods are applied in order to determine the position of the oriented
</p>
<p>fragment in the unit cell. This stage involves the use of several triple-phase invariants, Sect. 8.2.12,
</p>
<p>selected by the program as being sensitive to the location of the fragment within the unit cell. The
</p>
<p>weighted sum of the cosines of these phases is maximized with respect to the position of the search
</p>
<p>fragment in order to determine its most probable location.
</p>
<p>As a very large number of possible solutions for the orientation and position of the search fragment
</p>
<p>in the unit cell are explored, it is necessary to try to pinpoint the correct solution by using a figure of
</p>
<p>merit (FOM) based upon:
</p>
<p>1. The agreement with the Patterson function
</p>
<p>2. The triple-phase consistency
</p>
<p>3. An R-index between the observed jEj, Sect. 8.2.1, and the jEcj values, Sect. 13.4.10, for the partial
structure
</p>
<p>The Patterson function is calculated using jEjFo as coefficients instead of Fo2 in order to sharpen or
improve the resolution of peaks. The complete Patterson function is then stored in the computer, each
</p>
<p>grid value being represented by a digit between 0 and 7 so that it can be stored in three bits of
</p>
<p>8.3 Patterson Search Methods 387</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec31_13">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec31_13</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec16_13">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec16_13</a></div>
</div>
<div class="page"><p/>
<p>computer memory. This is one step better than the method employed in the Vector Verification,
</p>
<p>which employed only a two-bit representation. In order to make efficient use of computer memory,
</p>
<p>the Patterson values are encoded according to seven test levels, with level 2 equal to the median of the
</p>
<p>cumulative Patterson distribution, the difference between two successive test levels being about half
</p>
<p>the expected height of the highest single vector. The user can, in fact, supply different test levels from
</p>
<p>those noted above but it is probably not necessary.
</p>
<p>Rotation Search Strategy Used in PATSEE
The rotation search procedure used in PATSEE is summarized in Table 8.12.
</p>
<p>Translation Search Strategy Used in PATSEE
The translation search procedure used in PATSEE is summarized in Table 8.13.
</p>
<p>8.3.7 Examples of Structure Solution Using PATSEE
</p>
<p>The following examples, 5,7-methoxy-8-(3-methyl-1-buten-3-ol)-coumarin and atropine, indicate vari-
</p>
<p>ous other features of the PATSEE program. In particular, the use of the figures of merit RFOM,
</p>
<p>TPSRSUM, and CFOM is to pinpoint the correct Patterson Search solution.
</p>
<p>Structure of 5,7-Methoxy-8-(3-Methyl-1-Buten-3-ol)-Coumarin
The crystal and molecular structure of the antimalarial compound 5,7-methoxy-8-(3-methyl-1-buten-3-
</p>
<p>ol)-coumarin, C16H18O5,Mr &frac14; 290.3 has been reported [32]. Thismolecule, Fig. 8.16a,was selected for
investigation of the features of the PATSEE program, because the coumarin moiety, Fig. 8.16b, is
</p>
<p>known to be fairly rigid and planar (rings A and B) and as such is an ideal search molecule. The
</p>
<p>presence of potentially more flexible side groups provides an element of challenge to the method.
</p>
<p>The atomic coordinates of the 11 atoms in the coumarin moiety are readily available from either the
</p>
<p>published structure [33] or the Cambridge Crystallographic Data Base [34]. In fact, we chose the
</p>
<p>method of molecular graphics employing the Chem-X package to generate coumarin search model 1.
</p>
<p>The program Chem-X is no longer available but ChemSketch [31] is a good alternative for this
</p>
<p>type of work.
</p>
<p>Table 8.12 Rotation search in PATSEE
</p>
<p>Compilation of the intramolecular self-vector set from the model coordinates with distances between 2 Å, which are too
</p>
<p>short to provide orientation data, and 6 Å, which is the point at which errors in distance affect the accuracy with which
</p>
<p>vectors superimpose
</p>
<p>Generation of random orientations, typically between 10000 and 60000 angle triplets (a, b, g) at about 7� intervals
</p>
<p>For each orientation, calculation of the correlation between the rotated intramolecular vector set and the Patterson
</p>
<p>function. This is computed as a sum function to give a figure of merit (FOM)
</p>
<p>RFOM &frac14; &eth;1=n&THORN;P
n
</p>
<p>i&frac14;1
Pi=wi (8.63)
</p>
<p>where n � 0.3ntotal, wi is the calculated vector weight, Pi is the nearest Patterson grid value, and ntotal is the number of
worst-fitting vectors, that is, those with the lowest Pi/wi value
</p>
<p>An overlap or packing test ensures that when symmetry and lattice translations are applied there are no serious
</p>
<p>interatomic clashes
</p>
<p>An equivalence test which excludes similar solutions from being retained
</p>
<p>Sorting of the solutions in descending order of RFOM
</p>
<p>Refinement of the best solutions, carried out by testing up to 1000 additional random rotations around each retained
</p>
<p>solution, at approximately 2� intervals
</p>
<p>388 8 Direct Methods and Refinement</p>
<p/>
</div>
<div class="page"><p/>
<p>Search Model 1 from Chem-X
The material crystallizes in the monoclinic space group P21/c with four molecules per unit cell of
</p>
<p>dimensions a &frac14; 8.9044(9) Å, b &frac14; 17.623(1) Å, c &frac14; 10.175(1) Å, b &frac14; 113.97(1)�. The X-ray inten-
sity data were collected on a Nonius CAD4 diffractometer, Sect. 5.6ff, using Cu Ka radiation.
</p>
<p>A total of 3193 reflections was collected, to a ymax of 74.22
�, of which 2972 are independent with
</p>
<p>an Rint, Sect. 10.4.7, of 0.0175. Using the coumarin ring system as the search model provides a
</p>
<p>fragment consisting of 11 out of 21 non-hydrogen atoms. The fractional scattering power (p2), defined
</p>
<p>in the previous section, for this model is very high (50.4%). The coordinates generated by Chem-X for
</p>
<p>the first search model are shown in Table 8.14, which is a complete listing of the PATSEE input data,
</p>
<p>whereas Table 8.15 summarizes the search results.
</p>
<p>Table 8.13 Translation search in PATSEE
</p>
<p>Search for the most probable direct methods TPRs
</p>
<p>Calculation of jE(hkl)j for a given orientation of the search model
Selection of suitable phase relationships from a relatively small number of large jE(hkl)j data. This strategy
considerably speeds up the procedure and especially enhances the efficiency for larger structures. Note that in small-
</p>
<p>molecule work, the search fragment is of course very incomplete and possibly inaccurate. Nevertheless, if its scattering
</p>
<p>power �
P
</p>
<p>j Z
2
j
</p>
<p>� �
</p>
<p>is large enough, the TPRs used here should hold, at least approximately, for the correct solution, and
</p>
<p>be nonrandom in character. Hence the importance of having a sufficiently large fragment of the whole structure as the
</p>
<p>search model, as discussed further in the next section. The reader should refer to Sect. 8.2 to aid the appreciation of these
</p>
<p>three steps
</p>
<p>For each oriented search model from the rotation stage, sets of atom coordinates are generated, the oriented search
</p>
<p>model being placed at a position in the unit cell generated in a random manner. Each of these sets of coordinates is used
</p>
<p>for calculation of trial phases and assessment through their agreement with the selected TPRs from stage 1
</p>
<p>A packing test is carried out on rotated-translated fragments from stage 4 and a model is eliminated if short
</p>
<p>intermolecular distances occur
</p>
<p>Initial refinement of an RT fragment: this procedure involves optimizing a figure of merit TPSRSUM by fine tuning the
</p>
<p>position of the fragment:
</p>
<p>TPSRSUM &frac14;
P
</p>
<p>jEhjjEkjjE�h�kj cos&eth;fh &thorn; fk &thorn; f�h�k&THORN;&frac12; �
P
</p>
<p>jEhjjEkjjE�h�kj&frac12; �= (8.64)
</p>
<p>the summations being taken over all selected three-phase structure invariants. TPSRSUM is expected to be large and
</p>
<p>positive for the correct solution, up to a maximum value of 1.0. During this process the step sizes are reduced from
</p>
<p>around 0.2 to 0.05 Å
</p>
<p>A further distance test is then carried out to check the packing of the fragment
</p>
<p>Solutions that have survived all of these rigorous tests are further tested against the Patterson function of the target
</p>
<p>crystal. In earlier Patterson Search programs, such as Vector Verification, this stage was carried out immediately after
</p>
<p>the rotation stage [87], and consequently more time consuming and less likely to succeed.
</p>
<p>In PATSEE the correlation between the Patterson function and the fragment-derived intermolecular vector set is
</p>
<p>examined by comparing the weight of each vector with the nearest grid value. The fit is measured by calculating a
</p>
<p>further FOM:
</p>
<p>TFOM &frac14; &eth;1=n&THORN;
P
</p>
<p>n
</p>
<p>i&frac14;1
</p>
<p>Pi=wi (8.65)
</p>
<p>where n � 0.2ntotal, and the other parameters are defined as before
</p>
<p>Final selection and ordering of the possible solutions. At this stage a small number of the most promising solutions
</p>
<p>according to TPSRSUM and TFOM will have been stored in the computer. An RE index is calculated as
</p>
<p>RE &frac14;
P
</p>
<p>&eth;Eo � jEcj&THORN;=p&frac12; �=
P
</p>
<p>Eo (8.66)
</p>
<p>where p2, equal to
P
</p>
<p>Z2frag=
P
</p>
<p>Z2molecule, is the fractional scattering power of the search model (frag) compared to that of
</p>
<p>the whole model. Only positive terms in (8.66) are considered, as it is assumed that negative terms indicate complete
</p>
<p>agreement. The solutions are then sorted according to a combined FOM:
</p>
<p>CFOM &frac14; 0:1&eth;RFOM&thorn; TFOM&THORN;&eth;TPRSUM1=2&THORN;=RE (8.67)
</p>
<p>8.3 Patterson Search Methods 389</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec21_5">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec21_5</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec27_10">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec27_10</a></div>
</div>
<div class="page"><p/>
<p>Fig. 8.16 Chemical
formulae: (a) the Coumarin
derivative; (b) Coumarin:
rings A and B form a planar
</p>
<p>group for use in Patterson
</p>
<p>Search (Diagrams
</p>
<p>produced by Chemwindow
</p>
<p>(Softshell International
</p>
<p>Limited))
</p>
<p>Table 8.14 Input data for coumarin model 1 PATSEE search
</p>
<p>TITLE Coumarin derivative Chem-X model for PATSEE
</p>
<p>CELL 1.5418 8.9044 17.6236 10.1757 90.0 113.97 90.0
</p>
<p>ZERR 4 0.0009 0.001 0.001 0 0.01 0
</p>
<p>LATT 1
</p>
<p>SYMM X, 0.5 � Y, 0.5 + Z
SFAC C H O
</p>
<p>UNIT 64 72 20
</p>
<p>ROTS 10000 20
</p>
<p>TRAN
</p>
<p>FRAG 1 100.00 100.00 100.00 90.0 90.0 90.0
</p>
<p>O1 3 �0.01015 0.00495 0.000
C2 1 �0.02219 �0.00188 0.000
O2 3 �0.03271 0.00429 0.000
C3 1 �0.02228 �0.01572 0.000
C4 1 �0.01034 �0.02273 0.000
C40 1 0.00169 �0.01589 0.000
C5 1 0.01363 �0.02289 0.000
C6 1 0.02566 �0.01605 0.000
C7 1 0.02576 �0.00221 0.000
C8 1 0.01382 0.00479 0.000
</p>
<p>C80 1 0.00179 �0.00205 0.000
Notes: The above data have the following interpretation: CELL:
</p>
<p>wavelength and target unit-cell parameters; ZERR: number of mole-
</p>
<p>cules in target cell and errors in experimental unit-cell parameters a,
</p>
<p>b, c, a, b, g; LATT 1: primitive unit cell and centrosymmetric space
</p>
<p>group; SYMM: other space-group symmetry operations; SFAC: atom
</p>
<p>types in molecule; UNIT: number of each atom type in unit cell;
</p>
<p>ROTS: 10000 20 (use 10,000 random test orientations and retain the
</p>
<p>best 20); TRAN: initiate translation search using nt random positions
</p>
<p>calculated by the program (the user also has the option of supplying a
</p>
<p>value for this number here); FRAG: 1 &frac14; fragment 1 (a second frag-
ment can be supplied in addition as in the example below). The next
</p>
<p>six values are the unit-cell parameters for the fragment. This is not a
</p>
<p>real crystal cell as this fragment was model built in Chem-X. The final
</p>
<p>11 lines of data here represent the model atom names, types, and
</p>
<p>coordinates in the FRAG unit cell
</p>
<p>390 8 Direct Methods and Refinement</p>
<p/>
</div>
<div class="page"><p/>
<p>Discussion of Results and Expansion and Refinement of the Model
From the extract from the PATSEE output, Table 8.15, we can see:
</p>
<p>1. The top 15 values of RFOM, after the rotation stage, are quite similar
</p>
<p>2. The values of CFOM, after the translation stage, are more widely spread and only solution 8 has a
</p>
<p>value greater than 1.0
</p>
<p>3. Solution 9 has CFOM &frac14; 0.909 and somewhat similar rotation solution angles
Using the coordinates of the coumarin fragment corresponding to solution 8, all of the atoms in the
</p>
<p>structure were located and refined in two iterations of Fourier-least-squares in the program WinGX.
</p>
<p>Figure 8.17 shows how the structure developed at these three stages. At stage (a), Fig. 8.17a, the R
</p>
<p>factor was very high, as expected, with a value of 57.3%, reducing to 42.3% after the stage of
</p>
<p>Fig. 8.17b, and dramatically to 15.5% represented by Fig. 8.17c, which included all non-hydrogen
</p>
<p>atoms. Experience tells us that a structure which refines with isotropic temperature factors to this sort
</p>
<p>of R-value is probably correct. Further refinement of the model led to the published structure for
</p>
<p>which R &frac14; 3.9%, Fig. 8.17d. It has been shown that the use of model 1 with PATSEE has led to a
successful determination of the structure with the RT solution corresponding to the highest CFOM
</p>
<p>value. The strategy of carrying 20 rotation stage solutions through to the 00translation stage was
</p>
<p>necessary in view of the lack of discrimination in the RFOM values.
</p>
<p>Search Model 2 from Chem-X
In order to probe further the working and effectiveness of the PATSEE program, the above
</p>
<p>calculations were repeated with a smaller search model. A smaller, six-atom fragment, model
</p>
<p>comprising ring B, Fig. 8.16b, was generated again using Chem-X. This is essentially a benzene
</p>
<p>ring with six equivalent bonds and angles. The fractional scattering power p2 is now much lower,
</p>
<p>Table 8.15 Summary of results for coumarin model 1
</p>
<p>Solution RFOM
</p>
<p>Rotation stage Translation stage
</p>
<p>a (�) b (�) g (�) CFOM
</p>
<p>1 0.514 5.103 5.844 2.295 0.559
</p>
<p>2 0.514 5.109 5.887 0.527 0.678
</p>
<p>3 0.514 1.261 3.580 0.844 0.540
</p>
<p>4 0.514 1.317 2.667 0.593 0.665
</p>
<p>5 0.514 3.524 4.462 1.948 0.542
</p>
<p>6 0.514 3.185 1.821 0.953 0.854
</p>
<p>7 0.514 1.325 3.594 2.521 0.698
</p>
<p>8 0.514 6.163 5.078 0.884 1.053&larr;
</p>
<p>9 0.514 5.153 5.841 0.462 0.909
</p>
<p>10 0.514 1.832 0.475 0.891 0.881
</p>
<p>11 0.514 2.868 4.993 1.146 0.554
</p>
<p>12 0.514 5.136 0.437 2.633 0.716
</p>
<p>13 0.491 1.939 5.704 2.240 0.470
</p>
<p>14 0.491 1.862 5.706 2.253 0.594
</p>
<p>15 0.490 1.315 2.543 2.291 0.547
</p>
<p>16 0.364 5.305 0.977 1.172 0.103
</p>
<p>17 0.356 0.438 1.273 2.666 0.409
</p>
<p>18 0.329 4.078 4.008 1.198 0.094
</p>
<p>19 0.299 5.091 0.103 2.478 0.089
</p>
<p>20 0.298 1.162 2.826 2.258 0.132
</p>
<p>8.3 Patterson Search Methods 391</p>
<p/>
</div>
<div class="page"><p/>
<p>Fig. 8.17 Stages in determination of the structure of the coumarin derivative. (a) The search model. (b) The partial
structure developed after one cycle of refinement. (c) After two cycles of refinement. (d) After further refinement
(diagrams by POV-Ray&trade; VERSION 3.1 (http://www.povray.org), as implemented in WinGX and generated by Ortep-
</p>
<p>3 for Windows)
</p>
<p>392 8 Direct Methods and Refinement</p>
<p/>
<div class="annotation"><a href="http://www.povray.org">http://www.povray.org</a></div>
</div>
<div class="page"><p/>
<p>24.1% instead of 50.4% for model 1, and is therefore expected to be less effective in the Patterson
</p>
<p>search. In other words even if PATSEE can locate the model correctly in the unit cell, this may not
</p>
<p>be the top solution and may be more difficult to expand to the full molecule. The program was again
</p>
<p>instructed to retain the top 20 rotation solutions, which were then passed on to the translation
</p>
<p>procedure, with the results as summarized in Table 8.16.
</p>
<p>Expansion and Refinement of the Model
The coordinates for the six atoms corresponding to the best RT solution (solution 16) were input to the
</p>
<p>WinGX package and refined with SHELX-97 for two least-squares cycles. Inspection of a subsequent
</p>
<p>electron densitymap, using the programSXGRAPH inWinGX, allowed all of themissing structure to be
</p>
<p>built in and refinedwith two iterations of Fourier and isotropic least squares to anR-factor of 15.5%. The
</p>
<p>above result is quite pleasing, and may be somewhat surprising in view of the relatively few atoms used
</p>
<p>in the search.
</p>
<p>Search Model 3 from Chem-X
The search model was further modified and reduced by removing atoms C40 and C80, Fig. 8.16b, and
the PATSEE procedure was repeated. A correctly rotated and translated search model was found in
</p>
<p>the PATSEE RT listing. It was expanded and refined using WinGX to the same isotropic R-factor as
</p>
<p>in the previous cases, but with a little more difficulty. The reason this model did not produce the top
</p>
<p>RT solution is undoubtedly due to its size. Compared to models 1 and 2, the fractional scattering
</p>
<p>power for model 3 is only 16.1%. It is therefore a tribute to the method that the correct crystal
</p>
<p>structure could be generated with PATSEE without too much difficulty; remember though, that it is
</p>
<p>often much easier to solve a problem when the answer is known.
</p>
<p>Table 8.16 Summary of RT results for coumarin model 2
</p>
<p>Solution RFOM
</p>
<p>Rotation stage Translation stage
</p>
<p>a (�) b (�) g (�) CFOM
</p>
<p>1 0.604 0.515 1.328 2.756 0.960
</p>
<p>2 0.604 5.015 5.907 2.162 0.661
</p>
<p>3 0.604 5.088 5.846 0.637 0.881
</p>
<p>4 0.604 3.746 1.826 1.492 0.437
</p>
<p>5 0.604 5.088 5.284 2.035 0.264
</p>
<p>6 0.604 1.973 5.816 0.615 0.983
</p>
<p>7 0.604 5.956 4.351 2.546 1.335
</p>
<p>8 0.604 1.278 2.682 2.257 1.234
</p>
<p>9 0.604 1.895 5.873 0.623 1.071
</p>
<p>10 0.604 0.353 1.714 1.263 0.783
</p>
<p>11 0.604 3.158 4.603 2.316 0.787
</p>
<p>12 0.604 3.353 4.417 2.025 0.879
</p>
<p>13 0.604 2.724 1.756 0.429 0.830
</p>
<p>14 0.604 4.548 3.606 0.925 0.715
</p>
<p>15 0.604 1.468 3.602 0.781 0.691
</p>
<p>16 0.604 6.289 4.886 0.936 1.478&larr;
</p>
<p>17 0.604 2.741 4.932 1.130 0.697
</p>
<p>18 0.604 4.301 3.585 2.717 1.055
</p>
<p>19 0.604 1.843 5.736 2.330 1.174
</p>
<p>20 0.604 1.217 3.681 2.402 0.650
</p>
<p>8.3 Patterson Search Methods 393</p>
<p/>
</div>
<div class="page"><p/>
<p>The above experiments with a second and third model emphasize the power of the method and it
</p>
<p>should be remembered that some structures do still fail to be determined by direct methods. We know
</p>
<p>that a 21-atom structure can be solved with a 4-atom search model. Similar models may be useful for
</p>
<p>solving small polypeptide structures, for example, using the well-defined planar peptide group in the
</p>
<p>initial search. A dipeptide search model with 1� of rotational freedom may enable a tetrapeptide
structure to be solved. This procedure will become clearer after studying the next section.
</p>
<p>Structure of Atropine: a-[Hydroxymethyl]Benzeneacetic Acid 8-Methyl-8-
Azabicyclo[3.2.1]oct-3-yl Ester
</p>
<p>The chemical formula of the atropine molecule is shown in Fig. 8.18. Until recently [35] there were
</p>
<p>no reports of the structure of this classic molecule in the literature. Atropine is a competitive
</p>
<p>antagonist at central and peripheral synapses and has been used, somewhat unadvisedly, as a
</p>
<p>beautifying agent; hence its alternative, better known name Belladonna.
</p>
<p>Atropine, C17H23NO3, Mr &frac14; 289.4, is a-[hydroxymethyl]benzeneacetic acid 8-methyl-8-azabicy-
clo [3.2.1]oct-3-yl ester, and is known also as tropine tropate.
</p>
<p>Practical Details
Unit-cell determination and refinement, data collection, and data reduction were carried out on a Nonius
</p>
<p>Kappa CCD diffractometer [36] using Mo Ka radiation at �173 �C (liquid nitrogen temperature) with
the aid of an Oxford Cryostreams cooler. At this temperature, the crystal diffracted strongly over the
</p>
<p>90 min period of data collection. Data frames were processed using the Nonius software. The material
</p>
<p>crystallizes in the orthorhombic system, with the unusual space group, Fdd2, Fig. 8.19, with 16
</p>
<p>molecules per unit cell of dimensions a &frac14; 24.291(5) Å, b &frac14; 39.538(8) Å, c &frac14; 6.472(1) Å. A total of
2701 independent reflections were collected, to ymax 25.02
</p>
<p>�.
</p>
<p>Patterson Search Models
The molecule contains a benzene ring, as in the previous example, which is an obvious fragment to use for
</p>
<p>Patterson Search. This can be easily extended by adding atom C7 to form a slightly larger fragment
</p>
<p>comprising 7 of the 21 non-hydrogen atoms, with a fractional scattering power p2 &frac14; 0.30. However with
this model, trials with PATSEE failed to identify anRT solution that could be expanded and refined into the
</p>
<p>complete molecule.
</p>
<p>Fig. 8.18 Chemical formula of atropine (diagram produced by Chemwindow (Softshell International Limited))
</p>
<p>394 8 Direct Methods and Refinement</p>
<p/>
</div>
<div class="page"><p/>
<p>It can be demonstrated, by applying the usual Fourier-least-squares procedure to the known atom
</p>
<p>coordinates for this fragment, that phasing on the 7 atoms of this model leads to a complete and refinable
</p>
<p>structure.We conclude therefore that if the correct solution is in fact produced by PATSEE for this small
</p>
<p>fragment, it does not appear as one of the best solutions, as judged by the figures of merit.
</p>
<p>In order to overcome this problem and further test the features of PATSEE, a model was built
</p>
<p>with Chem-X, which included fragment 1 (atoms 1&ndash;6) and five other atoms, C7, C9, O1, O3, and
</p>
<p>C8, as fragment 2, Fig. 8.20. Fragment 2 is itself structurally rigid, but for the combined search
</p>
<p>model of fragments 1 and 2 there is rotational freedom about the bond C6&ndash;C7. The PATSEE
</p>
<p>program allows the linkage torsion angle, Sect. 8.5.2, between two such fragments to be systemati-
</p>
<p>cally varied during Patterson Search to produce a series of test models. Each new set of coordinates
</p>
<p>produced by a change of this torsion angle is then treated as an independent search model. The
</p>
<p>computer time required to complete the rotation search is thus multiplied by the number of
</p>
<p>individual models explored. The best models are again passed to the translation stage.
</p>
<p>Patterson Search for Atropine
The input data are shown in Table 8.17. The reader should identify the differences between the data
</p>
<p>in this table and those in Table 8.14 used for coumarin. Obvious changes arise from the differences
</p>
<p>in unit-cell dimensions and space group and molecular formula. The reader should study Fig. 8.19
</p>
<p>so as to identify the entries that appear in Table 8.17; LATT �4 is the code for a non-
centrosymmetric space group with an F unit cell. The rotation (ROTS) and translation (TRAN)
</p>
<p>instructions have been retained from the coumarin example without change; see Table 8.14 and
</p>
<p>footnote, especially for information on ROTS and TRAN. The next major change occurs in the
</p>
<p>FRAG listing of atom positions: the instruction TWIS 0 2 360 causes the program to vary the
</p>
<p>Fig. 8.19 Space group
diagrams for Fdd2.
</p>
<p>(a) General equivalent
positions. (b) Symmetry
elements
</p>
<p>8.3 Patterson Search Methods 395</p>
<p/>
</div>
<div class="page"><p/>
<p>Fig. 8.20 Search model for atropine; torsional flexibility between the two rigid groups occurs about the C6&ndash;C7 bond
indicated (diagram produced by POV-Ray&trade; VERSION 3.1 [85], as implemented in WinORTEP [86])
</p>
<p>Table 8.17 Input data for PATSEE search for atropine frag-
ment
</p>
<p>TITL Atropine PATSEE using dual model
</p>
<p>CELL 0.71073 24.2913 39.5380 6.4727
</p>
<p>90.000 90.000 90.000
</p>
<p>ZERR 16.00 0.0049 0.0079 0.0013 0.000 0.000 0.000
</p>
<p>LATT �4
SYMM �X, �Y, Z
SYMM 0.25 + X, 0.25 � Y, 0.25 + Z
SYMM 0.25 � X, 0.25 + Y, 0.25 + Z
SFAC C H N O
</p>
<p>UNIT 272 368 16 48
</p>
<p>ROTS 10000 20
</p>
<p>TRAN
</p>
<p>FRAG 1 100.00 100.00 100.00 90.0 90.0 90.0
</p>
<p>(Fragment via Chem-X)
</p>
<p>C1 1 0.12068 0.04724 0.06165
</p>
<p>C2 1 0.12991 0.05601 0.06751
</p>
<p>C3 1 0.13407 0.05393 0.08046
</p>
<p>C4 1 0.12927 0.04327 0.08761
</p>
<p>C5 1 0.12003 0.03461 0.08187
</p>
<p>C6 1 0.11568 0.03661 0.06881
</p>
<p>TWIS 0 2 360
</p>
<p>C7 1 0.10546 0.02722 0.06245
</p>
<p>C8 1 0.07132 0.04096 0.06734
</p>
<p>C9 1 0.09194 0.02896 0.06908
</p>
<p>O3 3 0.08480 0.03817 0.06248
</p>
<p>O1 3 0.08835 0.02332 0.07902
</p>
<p>396 8 Direct Methods and Refinement</p>
<p/>
</div>
<div class="page"><p/>
<p>torsion angle systematically about C6&ndash;C7 from 0 to 360� in steps of 2�, thus involving 181 different
search models in the rotation procedure. The final application of the torsional change at 360� gives
the same result as for the first at 0 and acts merely as a check.
</p>
<p>Rotation Stage
The results of the rotation analysis are summarized in Table 8.18. In this table and in Table 8.19, the
</p>
<p>torsion angle refers to the relative orientation, about the C6&ndash;C7 bond, of the two rigid fragments of
</p>
<p>the search model (Fig. 8.20).
</p>
<p>Translation Stage
The possible rotation function solutions are then each transferred to the translation algorithm in turn.
</p>
<p>The best 20 solutions, Table 8.19, were subjected to the translation algorithms as before. Values of
</p>
<p>CFOM after translation and optimization range from 0.929 to 2.640, as shown in the table.
</p>
<p>In this analysis the order of precedence has changed when using CFOM as the final discriminator.
</p>
<p>As would be expected the best solutions (arrowed) all have similar rotational and torsional para-
</p>
<p>meters. Although not shown here, the corresponding translation components derived in the second
</p>
<p>stage of PATSEE are also quite similar, as would be expected.
</p>
<p>Expansion and Refinement of the Model
The overall best solution was taken to be number 2 in Table 8.19, and the model coordinates
</p>
<p>corresponding to this solution were subjected to Fourier-least-squares expansion and refinement as
</p>
<p>before. After two iterations of Fourier-least squares all non-carbon atoms were located and isotropically
</p>
<p>refined to an R factor of 0.105. In the published structure anisotropic refinement led to final R factor of
</p>
<p>0.0453. The completed structure of atropine is shown in Fig. 8.21.
</p>
<p>Table 8.18 Atropine fragment rotation search
</p>
<p>Solution RFOM a (�) b (�) g (�) Torsion angle (�)
</p>
<p>1 1.742 2.942 2.912 0.174 182
</p>
<p>2 1.732 5.892 3.527 1.736 2
</p>
<p>3 1.701 6.036 3.359 2.980 180
</p>
<p>4 1.676 6.052 3.352 2.984 182
</p>
<p>5 1.563 2.932 2.916 0.176 184
</p>
<p>6 1.544 5.827 3.474 1.766 358
</p>
<p>7 1.535 5.933 3.540 1.737 360
</p>
<p>8 1.535 5.933 3.540 1.737 0
</p>
<p>9 1.522 6.101 3.295 2.999 184
</p>
<p>10 1.438 5.938 3.567 1.705 4
</p>
<p>11 1.403 2.883 2.927 0.158 176
</p>
<p>12 1.353 2.690 5.981 1.726 35
</p>
<p>13 1.348 2.912 2.876 0.179 180
</p>
<p>14 1.334 6.105 3.408 2.969 186
</p>
<p>15 1.332 2.702 5.949 1.783 354
</p>
<p>16 1.312 5.862 3.549 1.745 6
</p>
<p>17 1.285 2.946 2.883 0.176 178
</p>
<p>18 1.272 6.114 3.360 3.008 188
</p>
<p>19 1.246 2.594 6.027 1.757 352
</p>
<p>20 1.222 2.588 5.984 1.748 356
</p>
<p>Note: Each entry in this table is the result of 10,000 test rotations
</p>
<p>8.3 Patterson Search Methods 397</p>
<p/>
</div>
<div class="page"><p/>
<p>Conclusions
These examples illustrate many of the features of the powerful Patterson search technique.
</p>
<p>However, for small-molecule analysis, because direct methods are so much more easy to apply,
</p>
<p>if not to understand, Patterson search will remain as a reserve technique, albeit a very useful and
</p>
<p>powerful one, in the crystallographer&rsquo;s armory.
</p>
<p>Table 8.19 Atropine fragment translation search results
</p>
<p>Solution RFOM Torsion angle (�) CFOM
</p>
<p>1 1.742 182 1.408
</p>
<p>2 1.732 2 2.640&larr;
</p>
<p>3 1.701 180 1.373
</p>
<p>4 1.676 182 1.250
</p>
<p>5 1.563 184 1.424
</p>
<p>6 1.544 358 2.284&larr;
</p>
<p>7 1.535 360 1.232
</p>
<p>8 1.535 0 1.232
</p>
<p>9 1.522 184 1.029
</p>
<p>10 1.438 4 2.342&larr;
</p>
<p>11 1.403 176 1.081
</p>
<p>12 1.353 356 1.342
</p>
<p>13 1.348 180 1.039
</p>
<p>14 1.334 186 1.114
</p>
<p>15 1.332 354 1.307
</p>
<p>16 1.312 6 1.793
</p>
<p>17 1.285 178 1.077
</p>
<p>18 1.272 188 0.929
</p>
<p>19 1.246 352 1.763
</p>
<p>20 1.222 356 1.694
</p>
<p>Fig. 8.21 The completed structure of atropine (diagram produced by POV-Ray&trade; VERSION 3.1 [85], as implemented
in WinORTEP [86])
</p>
<p>398 8 Direct Methods and Refinement</p>
<p/>
</div>
<div class="page"><p/>
<p>We have discussed two small-molecule analyses where Patterson search with PATSEE has led
</p>
<p>successfully to refineable models of the crystal structure. Provided that a suitable search model is
</p>
<p>available there is no reason to doubt that most small-molecule structures could be solved in a similar
</p>
<p>way. Themost useful discriminator provided by PATSEE is the combined figure of merit (CFOM). For
</p>
<p>all but the smallest searchmodels employed in the examples discussed, the RT solutionwith the largest
</p>
<p>value of CFOM has proved to be correct.
</p>
<p>A search model with a fractional scattering power as low as 16.1% and including only 6 out of a
</p>
<p>total of 21 non-hydrogen atoms can provide the correct answer, but it requires more sifting of
</p>
<p>coordinate sets, using Fourier and least squares, to be selected. We shall see in Sect. 10.6.4 that
</p>
<p>the related method of Molecular Replacement (MR) is routinely used in macromolecular structure
</p>
<p>analysis, and for this reason the examples discussed in the present chapter will be invaluable for a
</p>
<p>sound understanding of the principles of the method when we come to discuss the large molecule
</p>
<p>studies.
</p>
<p>8.3.8 Shake and Bake
</p>
<p>We mention here, very briefly, another procedure that is a further stage in solving really complicated
</p>
<p>structures. Like PATSEE the program is readily available [37] and is quite fun to use, as the name
</p>
<p>Shake and Bake suggests.
</p>
<p>In outline, Shake and Bake is a direct structure solving procedure that carries out phase refinement
</p>
<p>in reciprocal space and electron density space alternately, in order to achieve a true minimum of R(F),
</p>
<p>a quantity known as the minimal function [38].
</p>
<p>R&eth;F&THORN; &frac14;
X
</p>
<p>H;K
</p>
<p>AHK cos THK �
J1&eth;AHK&THORN;
J0&eth;AHK&THORN;
</p>
<p>� �2
</p>
<p>&thorn;
X
</p>
<p>L;M;N
</p>
<p>jBLMNj cosQLMN �
J1&eth;BLMN&THORN;
J0&eth;BLMN&THORN;
</p>
<p>� �2
(
</p>
<p>&thorn;
X
</p>
<p>L;M;N
</p>
<p>jBLMNj cosQLMN �
J1&eth;BLMN&THORN;
J0&eth;BLMN&THORN;
</p>
<p>� �2
X
</p>
<p>H;K
</p>
<p>AHK &thorn;
X
</p>
<p>L;M;N
</p>
<p>jBLMNj
" #�1
</p>
<p>9
</p>
<p>=
</p>
<p>;
</p>
<p>(8.68)
</p>
<p>where AHK and BLMN are given by (8.69).
</p>
<p>AHK &frac14; &eth;2=N1=2&THORN;jEHEKEH&thorn;KjBLMN
&frac14; &eth;2=N&THORN;jELEMENEL&thorn;M&thorn;Nj&eth;jEL&thorn;Mj2 &thorn; jEM&thorn;Nj2 &thorn; jEN&thorn;Lj2 � 2&THORN; (8.69)
</p>
<p>The structure triple THK is defined by
</p>
<p>THK &frac14; fH &thorn; fK &thorn; f�H�K (8.70)
</p>
<p>and an estimate of cos THK follows the equation
</p>
<p>cos THK &frac14; J1&eth;AHK&THORN;=J0&eth;AHK&THORN; (8.71)
</p>
<p>where I0 and I1 are the modified Bessel functions.
</p>
<p>The structure quartet QLMN is given by
</p>
<p>QLMN &frac14; fL &thorn; fM &thorn; fN &thorn; f�L�M�N (8.72)
</p>
<p>8.3 Patterson Search Methods 399</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec36_10">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec36_10</a></div>
</div>
<div class="page"><p/>
<p>and the phases themselves are functions of the atomic positions:
</p>
<p>E&eth;h&THORN; &frac14; jE&eth;h&THORN;j exp i2pf&eth;h&THORN; &frac14; 1ffiffiffiffi
N
</p>
<p>p
X
</p>
<p>j
</p>
<p>exp i2ph 
 rj (8.73)
</p>
<p>The Shake and Bake procedure is initiated by generating triplet and negative quartet structure
</p>
<p>invariants that are based on a trial structure or partial structure model comprised of N atoms in the
</p>
<p>unit cell, chosen such that no two atoms are closer than ca. 1.2 Å and no atom lies within bonding
</p>
<p>distance of four other atoms. Normalized structure factors Eh are determined and with the phases a
</p>
<p>first electron density map calculated. The phases are then subjected to a cyclical shake-and-bake
</p>
<p>phasing procedure consisting of phase refinement, to minimize R(&rsquo;), and constrained electron
density maps to obtain atom positions. The time per cycle depends on the number of atoms and is
</p>
<p>about 10 s for a 30-atom structure and about 90 s for a structure with 400 atoms in the unit cell. The
</p>
<p>technique has solved a number of structures, including one with over 600 atoms in the asymmetric
</p>
<p>unit. For further details the reader is referred to the Appendix D and the literature [39&ndash;41].
</p>
<p>The procedure has also been used successfully on the 1001 non-hydrogen atom structure of
</p>
<p>lysozyme [38] and on a range of other protein structures [42].
</p>
<p>8.4 Least-Squares Refinement
</p>
<p>If we have two pairs of values of X and Y for measurements which are related by the equation
</p>
<p>Y &frac14; mX &thorn; b (8.74)
</p>
<p>we can obtain a unique answer for the constants m and b. Sometimes, as in the Wilson plot, we have
</p>
<p>several pairs of values, which contain random errors, and we need to obtain those values of m and b
</p>
<p>that best fit the complete set of observations. In practical problems, we have often a situation in which
</p>
<p>the errors in the X values are negligible compared with those in Y.
</p>
<p>Let the best estimates of m and b under these conditions be m0 and b0. Then, the error of fit in the
</p>
<p>ith observation is
</p>
<p>ei &frac14; m0Xi &thorn; b0 � Yi (8.75)
</p>
<p>The principle of least squares states that the best-fit parameters are those that minimize the sum of
</p>
<p>the squares of the errors. Thus,
</p>
<p>X
</p>
<p>i
</p>
<p>e2i &frac14;
X
</p>
<p>i
</p>
<p>&eth;m0Xi &thorn; b0 � Yi&THORN;
2 &eth;i &frac14; 2; . . . ;N&THORN; (8.76)
</p>
<p>has to be minimized over the number N observations. This condition corresponds to differentiating
</p>
<p>partially with respect to m0 and b0, in turn, and equating the derivatives to zero. Hence,
</p>
<p>m0
X
</p>
<p>i
</p>
<p>X2i &thorn; b0
X
</p>
<p>i
</p>
<p>Xi &frac14;
X
</p>
<p>i
</p>
<p>XiYi (8.77)
</p>
<p>400 8 Direct Methods and Refinement</p>
<p/>
</div>
<div class="page"><p/>
<p>m0
X
</p>
<p>i
</p>
<p>Xi &thorn; b0N &frac14;
X
</p>
<p>i
</p>
<p>Yi (8.78)
</p>
<p>which constitute a pair of simultaneous equations (normal equations) easily solved for m0 and b0.
</p>
<p>In a crystal structure analysis, we are always manipulating more observations than there are
</p>
<p>unknown quantities; the system is said to be overdetermined. We shall consider some crystallo-
</p>
<p>graphic applications of the method of least squares.
</p>
<p>8.4.1 Unit-Cell Dimensions
</p>
<p>In Chap. 5, we considered methods for obtaining unit-cell dimensions with moderate accuracy from
</p>
<p>photographic and diffractometer measurements. Generally, we need to enhance the precision of these
</p>
<p>measurements, which may be achieved by a least-squares analysis. Consider, for example, a mono-
</p>
<p>clinic crystal for which the y values of a number of reflections of known indices, preferably high-
</p>
<p>order, have been measured to the nearest 0.01�. In the monoclinic system, sin y is given, Table 2.4
with k &frac14; l, and (3.43), by
</p>
<p>4sin2y &frac14; h2a	2 &thorn; k2b	2 &thorn; l2c	2 &thorn; 2hla	c	 cos b	 (8.79)
</p>
<p>In order to obtain the best values of a*, b*, c*, and cos b*, we write, following (8.76),
</p>
<p>X
</p>
<p>i
</p>
<p>&eth;h2i a	2 &thorn; k2i b	2 &thorn; l2i c	2 &thorn; 2hilia	c	 cos b	 � 4sin2yi&THORN;
2
</p>
<p>(8.80)
</p>
<p>and then minimize this expression, with respect to a*, b*, c*, and cos b*, over the number of
</p>
<p>observations i. The procedure is a little more involved numerically; we obtain four simultaneous
</p>
<p>equations to be solved for the four variables, but the principles are the same as those involved with the
</p>
<p>straight line.
</p>
<p>8.4.2 Least-Squares Parameters
</p>
<p>Correct trial structures are refined by the least-squares method. In essence, this process involves
</p>
<p>adjusting a scale factor and the positional and temperature parameters of the atoms in the unit cell so
</p>
<p>as to obtain the best agreement between the experimental Fo values and the jFcj quantities derived
from the structure model. In its most usual application, the technique minimizes the function.
</p>
<p>R0 &frac14;
X
</p>
<p>h
</p>
<p>w&eth;Fo � G&eth;jFcj&THORN;2 (8.81)
</p>
<p>where the sum is taken over the set of crystallographically independent terms h, w is a weight for each
term, and G is the reciprocal of the scale factor K for Fo. Let pj (j &frac14; 1, 2,. . ., n) be the variables in jFcj
whose values are to be refined. Then
</p>
<p>@R0
</p>
<p>@pj
&frac14; 0 (8.82)
</p>
<p>8.4 Least-Squares Refinement 401</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_5">http://dx.doi.org/10.1007/978-1-4614-3954-7_5</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Tab4_2">http://dx.doi.org/10.1007/978-1-4614-3954-7#Tab4_2</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Equ43_3">http://dx.doi.org/10.1007/978-1-4614-3954-7#Equ43_3</a></div>
</div>
<div class="page"><p/>
<p>or
</p>
<p>X
</p>
<p>h
</p>
<p>wD
@jFcj
@pj
</p>
<p>&frac14; 0 (8.83)
</p>
<p>where D is Fo � jFcj. For a trial set of parameters not too different from the correct values, D is
expanded as a Taylor series to the first order:
</p>
<p>D&eth;p; j&THORN; &frac14; D&eth;p&THORN; �
Xn
</p>
<p>i&frac14;1
xi
@jFcj
@pj
</p>
<p>(8.84)
</p>
<p>where the shift xi is the correction to be applied to parameter pi; p and j represent the complete sets of
variables and corrections. Substituting (8.84) in (8.83) leads to the normal equations
</p>
<p>Xn
</p>
<p>i&frac14;1
</p>
<p>X
</p>
<p>h
</p>
<p>w
@jFcj
@pi
</p>
<p>@jFcj
@pj
</p>
<p>" #
xi &frac14;
</p>
<p>X
</p>
<p>h
</p>
<p>wD
@jFcj
@pj
</p>
<p>(8.85)
</p>
<p>The n normal equations may be expressed neatly in matrix form:
</p>
<p>A j &frac14; b or
X
</p>
<p>i
</p>
<p>aijxi &frac14; bj (8.86)
</p>
<p>where
</p>
<p>aij &frac14;
X
</p>
<p>h
</p>
<p>w
@jFcj
@pi
</p>
<p>@jFcj
@pj
</p>
<p>(8.87)
</p>
<p>and
</p>
<p>bj &frac14;
X
</p>
<p>h
</p>
<p>wD
@jFcj
@pj
</p>
<p>(8.88)
</p>
<p>The solution of the normal equations is a well- documented mathematical procedure that we
</p>
<p>shall not dwell upon. Instead, we draw attention to certain features of least-squares refinement. It is
</p>
<p>important to remember that least squares provides the best fit for the parameters that have been put
</p>
<p>into the model. Hence, it is essential to examine a final difference-Fourier map at the completion of
</p>
<p>a least-squares refinement, after several cycles of calculations have led to negligible differences xi.
</p>
<p>The techniques of least squares have been reported fully at Crystallographic Computing Confer-
</p>
<p>ences (see Bibliography).
</p>
<p>Temperature Factors
The Wilson technique, Sect. 4.2.1, leads to an overall isotropic temperature factor for the structure,
</p>
<p>and we have discussed the theory of one-dimensional isotropic thermal vibration in Sect. 4.1.8. As
</p>
<p>indicated there, a better procedure allots a B parameter to each atom and refines the values as least-
</p>
<p>squares parameters. We write the temperature correction factor Ti as
</p>
<p>402 8 Direct Methods and Refinement</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec11_4">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec11_4</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec9_4">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec9_4</a></div>
</div>
<div class="page"><p/>
<p>Ti &frac14; exp&frac12;�&eth;Bil�2sin2y&THORN;� (8.89)
</p>
<p>and the equation
</p>
<p>Bi &frac14; 8p
2U2i (8.90)
</p>
<p>relates the isotropic temperature factor Bi to the mean square amplitude U
2
i of the ith atom;
</p>
<p>the three-dimensional surface of isotropic vibration is a sphere. A more sophisticated treatment, as we
</p>
<p>indicated already in Sect. 4.1.8, describes the vibrations of each atom by a symmetrical tensor U
having six independent components in the general case:
</p>
<p>Ti &frac14; exp&frac12;�2p
2&eth;U11h
</p>
<p>2a	2 &thorn; U22k
2b	2 &thorn; U33l
</p>
<p>2c	2 &thorn; 2U23klb
	c	 &thorn; 2U31lhc
</p>
<p>	a	
</p>
<p>&thorn; 2U12hka
	b	&THORN;� (8.91)
</p>
<p>and the (anisotropic) Uij parameters are refined as part of the model. The surface of vibration is now a
</p>
<p>biaxial (thermal) ellipsoid, and the mean-square amplitude of vibration in the direction of a unit
</p>
<p>vector L(L1, L2, L3) is given by
</p>
<p>U2i &frac14;
X3
</p>
<p>i&frac14;1
</p>
<p>X3
</p>
<p>j&frac14;1
</p>
<p>UijLiLj (8.92)
</p>
<p>Biaxial ellipsoids for molecule A of the R-enantiomer BW202W92(R) are illustrated in Fig. 8.22;
</p>
<p>see also Sects. 8.2.12 and 8.2.16. Since L is defined with respect to the reciprocal lattice, the
component of U with L(1,0,0), parallel to a*, is
</p>
<p>Fig. 8.22 Anisotropic thermal ellipsoids for molecule A of BW202W92(R) [26]. The ellipsoids are plotted at the 50%
probability level, which means that at the maximum radius drawn the exponential expression of (8.91) is 0.5. In terms of
</p>
<p>an isotropic thermal surface, the radius would be almost 0.6 Å corresponding to sin2 y/l2, or &eth;1
2
d&THORN;2, of 0.694 Å�2 (drawn
</p>
<p>in WinGX with ORTEP)
</p>
<p>8.4 Least-Squares Refinement 403</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec9_4">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec9_4</a></div>
</div>
<div class="page"><p/>
<p>U2i &frac14; U11 (8.93)
</p>
<p>In an orthorhombic crystal, for example, a direction 30� from a* in the a*b* plane has
</p>
<p>L &frac14; &eth;
ffiffi
</p>
<p>3
p
</p>
<p>2
; 1
2
; 0&THORN;, and the component of U in that direction is
</p>
<p>U2i &frac14; U11&eth;
ffiffi
</p>
<p>3
p
</p>
<p>2
&THORN;2 &thorn; U22&eth;12&THORN;
</p>
<p>2 &thorn; 2U12&eth;
ffiffi
</p>
<p>3
p
</p>
<p>2
&THORN;&eth;1
</p>
<p>2
&THORN; (8.94)
</p>
<p>The following relationships among the values of B, U2i , and the root mean square (rms) amplitude
</p>
<p>are often useful:
</p>
<p>The smallest rms amplitudes encountered are ca. 0.05 Å. Values ofB between 3 and 10 Å2 are found
</p>
<p>in organic structures at ambient temperatures. The larger rms amplitudes require caution in interpret-
</p>
<p>ing them in terms of bond lengths and their precision. For example, static or dynamic disorder,
</p>
<p>Sect. 8.9, not included in the model, may be manifested as abnormally large temperature factors.
</p>
<p>Scale Factor
In least-squares refinement the Fo data must not be adjusted, and so the parameter G in (8.81) is
</p>
<p>introduced. The inverse of the refined value ofGmay be applied to Fo at the end of a refinement cycle.
</p>
<p>Several cycles of refinement may be needed before the parameters reach a sensibly constant value.
</p>
<p>Generally full-matrix least-squares refinement is to be preferred. However, where the number of
</p>
<p>parameters is very large or where computer availability is limited, an approximation may be used.
</p>
<p>One such method is the block-diagonal refinement, in which certain off-diagonal aij terms in (8.86) are
</p>
<p>neglected. Generally, more cycles are necessary in this procedure.
</p>
<p>Weights
In the initial stages of refinement, weights may be set at unity or chosen so as to accelerate the
</p>
<p>process, such as down-weighting reflections of small Fo or of high order, or both. In the final stages,
</p>
<p>weights should be related to the precision of Fo, which can be achieved in two ways:
</p>
<p>1.
w&eth;hkl&THORN; &frac14; 1=&frac12;s2&eth;Fo&eth;hkl&THORN;� (8.95)
</p>
<p>where the estimated standard deviation, s[Fo(hkl)], is obtained from counting statistics in diffrac-
</p>
<p>tometer data by the relationship
</p>
<p>s &frac14;
ffiffiffiffi
</p>
<p>N
p
</p>
<p>(8.96)
</p>
<p>N being related to the total counts, peak, and background, for the given reflection. Sometimes a
</p>
<p>quantity pFo
2 is added to the right-hand side of (8.95), where p is adjusted so that wD2 is constant
</p>
<p>over ranges of Fo, where D is Fo � jFcj.
</p>
<p>B/Å2 U2/Å2 rms Amplitude (Å)
</p>
<p>0.10 0.0013 0.036
</p>
<p>0.50 0.0063 0.080
</p>
<p>1.0 0.013 0.11
</p>
<p>5.0 0.063 0.25
</p>
<p>10 0.13 0.36
</p>
<p>404 8 Direct Methods and Refinement</p>
<p/>
</div>
<div class="page"><p/>
<p>2. w&eth;hkl&THORN; &frac14; F&eth;Fo&THORN; (8.97)
</p>
<p>where the function F is chosen so that wD2 is again sensibly constant over ranges of Fo. Another
</p>
<p>weighting scheme is given by
</p>
<p>3. w &frac14; &eth;A&thorn; Fo &thorn; BFo2 &thorn; CFo3&THORN;�1 (8.98)
</p>
<p>where the constants A, B, and C may be obtained by a least-squares fit of mean values of D, in
</p>
<p>ranges of Fo, to the inverse of the right-hand side of (8.98).
</p>
<p>Precision
The choice of absolute weights (8.95) should yield parameters of lowest variances:
</p>
<p>s2&eth;pj&THORN; &frac14; &eth;ajj&THORN;�1 (8.99)
</p>
<p>where (ajj)
�1 is an element of the matrix inverse to that of the aij elements (8.87). With weights
</p>
<p>related to Fo,
</p>
<p>s2&eth;pj&THORN; &frac14; &eth;ajj&THORN;�1
P
</p>
<p>h wD
2
</p>
<p>m� n (8.100)
</p>
<p>where m is the number of reflections and n the number of parameters to be refined. Generally,
</p>
<p>m=n � ca 10 leads to a high quality analysis. In a block-diagonal approximation, the standard deviations
are usually underestimated by 15&ndash;20%.
</p>
<p>Atoms in Special Positions
If any symmetry operation of the space group of a structure leaves an atom invariant, the atom is on a
</p>
<p>special position. Consider onemolecule of formula AB2 in the unit cell of space groupP2. TheA atoms
</p>
<p>occupy, say, the special positions 0, y, 0, and the B atoms occupy the general positions x, y, z and �x; y; �z.
Several important points arise:
</p>
<p>1. The x and z coordinates of atoms A remain invariant at zero during refinement.
</p>
<p>2. In this space group, the origin is along the twofold axis y and must be specified by fixing the y
</p>
<p>coordinate of an atom; the heavier the atom the better. It could be yB &frac14; 0, in which case this
</p>
<p>parameter also remains invariant.
</p>
<p>3. With respect to the symmetry operations of the space group, atoms A must be given an atom
</p>
<p>multiplicity factor of 1
2
so that a total of one A atom per unit cell obtains.
</p>
<p>4. In an isotropic refinement, one of the three twofold axes of the biaxial thermal ellipsoid must lie
</p>
<p>along the twofold axis y of the space group. In this case, the U12 and U23 elements of U remain
invariant at zero value.
</p>
<p>8.4.3 Theory of Least-Squares Refinement and Strategies to Use
</p>
<p>The calculations of least squares, although lengthy, have been implemented in now well-tested
</p>
<p>program systems, such as WinGX and SHELX, that are available to workers in crystal structure
</p>
<p>determination. In a number of instances, the application of least-squares calculations is
</p>
<p>8.4 Least-Squares Refinement 405</p>
<p/>
</div>
<div class="page"><p/>
<p>straightforward, and results of sufficient precision are obtained. In other cases, and perhaps more
</p>
<p>generally, a consideration of the strategy to be employed, within the constraints of the program
</p>
<p>system, is necessary and desirable. In the structure determination process, a model is proposed on the
</p>
<p>basis of the diffraction data, and then subjected to adjustments (refinement) of the parameters of that
</p>
<p>model so as to reach a solution with a defined precision.
</p>
<p>Model
For a least-squares refinement to be successful, the model must be sufficiently close to the truth and
</p>
<p>contain all the components of the structure. The multidimensional surface that corresponds to an
</p>
<p>n-dimensional refinement is complex and contains many false minima, into any of which an insuffi-
</p>
<p>ciently correct model may converge. Hence, the correctness of a structure should be based on several
</p>
<p>criteria, Sect. 8.7.
</p>
<p>Data Errors
Experimental data are subject to systematic and random errors. In X-ray diffraction, a systematic error
</p>
<p>may be, for example, the lack of an absorption correction: the values ofFowill often be less than those of
</p>
<p>jFcj for the partially refined structure, noticeably for the low-order reflections of high intensity. The
minimization of the difference betweenFo and jFcjwill lead, in this case, to unwarranted discrepancies in
the temperature factors of the atoms and the scale factor of the data.
</p>
<p>Increasing the temperature factors and decreasing G, the scale factor applied inversely to jFcj
during refinement, will tend to decrease the jFcj values so as to compensate for the effect of
absorption. Random errors are unavoidable in any physical experiment, but they can be minimized
</p>
<p>by careful attention to the experimental procedure. For example, reflections of low intensity and
</p>
<p>consequent high probable error can be improved in reliability by increasing the measurement time.
</p>
<p>This procedure may be costly in time, but is very straightforward with a diffractometer.
</p>
<p>The data set must be sufficient in size for the work in hand. The success of least-squares refinement
</p>
<p>is partially dependent on the fact that the calculations are appreciably overdetermined. This excess of
</p>
<p>data over variables is needed in order to average out discrepancies in individual measurements. As an
</p>
<p>approximate guide, the ratio of data to variables should be at least 8. The ratio can be improved by
</p>
<p>decreasing the number of variables. There are instances, such as with phenyl rings, where refinement
</p>
<p>of the parameters of the hydrogen atoms might be regarded as exaggerated. The hydrogen atoms can,
</p>
<p>and should, be allowed to contribute to the calculated structure factors, but unless there is reason to
</p>
<p>suppose otherwise they would be expected to display sp2 geometry, with C&ndash;H � 1.00 Å. Their
isotropic temperature factors may be refined, or assumed to be, say, 1.3 times that of the carbon atoms
</p>
<p>to which they are attached. In a structure like euphenyl iodoacetate C32H53O2I, Sect. 1.1, the number
</p>
<p>of variables may be very significantly reduced by this type of approach. The strategy should be
</p>
<p>determined by the nature of the problem in hand, and computer programs that handle least-squares
</p>
<p>refinement have been designed to accommodate a range of constraints.
</p>
<p>If the data-to-variables ratio is too low, the final structure may have an inherent lack of resolution
</p>
<p>that may not be immediately apparent from the numerical least-squares results, although all the
</p>
<p>expected atomic positions will be represented, provided that the model itself was complete. Least
</p>
<p>squares, unlike the Fourier process, cannot find anything that is not present in the model. It will obtain
</p>
<p>a best fit, under the given strategy, for the model supplied. A lack of resolution may arise because of
</p>
<p>an unsatisfactory termination of the data set. If the cut-off criterion for acceptable data leads to too
</p>
<p>few reflections, those that have been excised could be subjected to re-measurement for longer times
</p>
<p>so that they can be accepted under the same criterion. If this is not done, then imperfections will exist
</p>
<p>in the refinement, and may be manifested in large estimated standard deviations for some parameters
</p>
<p>or in unsatisfactory temperature factors. Incorrect values for B factors may lead to highly improbable
</p>
<p>406 8 Direct Methods and Refinement</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec1_1">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec1_1</a></div>
</div>
<div class="page"><p/>
<p>root mean square atomic displacements or to Uij values that are non-positive definite, that is, they do
</p>
<p>not define an ellipsoid.
</p>
<p>There is some concern with published papers on structure determination that report 50% or more
</p>
<p>of the unique experimental Fo data rejected by some criterion, such as I � 3s&eth;I&THORN;, apparently to the
</p>
<p>satisfaction of both the worker and the journal editor. The cosmetic effect of this practice serves
</p>
<p>merely to reduce the R factor rather than to improve the estimated standard deviations of the structural
</p>
<p>parameters. Weak intensity data do, in fact, contain structural information. Protein crystallographers,
</p>
<p>who of necessity work with poorly diffracting crystal specimens, are not in a position to discard data
</p>
<p>in this fashion.
</p>
<p>Least-Squares Refinement Procedure
The least-squares refinement of a model leads to those parameters that minimize (Fo � jFcj)
</p>
<p>2, as in
</p>
<p>(8.76), over the whole data set. The first application of the calculations will generally not lead to the
</p>
<p>best-adjusted parameters, and it will be necessary to cycle through the calculations until the results
</p>
<p>converge; that is, until the calculated shifts are less than the estimated standard deviations of the
</p>
<p>corresponding parameters. The refinement is continued until the shifts in the parameters are some
</p>
<p>small fraction (0.1 is often quoted) of the estimated standard deviations.
</p>
<p>A good procedure will begin by ensuring the maximum number of good observations, and by
</p>
<p>minimizing the number of variables consistent with the requirements of the problem. The starting
</p>
<p>model must be sufficiently good so that the minimization does not fall into a local, false position. This
</p>
<p>is unlikely if Fourier synthesis or difference-Fourier synthesis has been used in establishing the model
</p>
<p>and approximate scale and temperature factors have been obtained by statistical methods. Although
</p>
<p>Fourier methods of refinement are less convenient than those of least squares, they do have the power of
</p>
<p>revealing necessary information that may not be contained in an initial model. From the above, it should
</p>
<p>be clear that such information is vital to a good least-squares refinement.
</p>
<p>Another constraint that may be considered, as well as the fixed C&ndash;H geometry, is the rigid-body
</p>
<p>specification of a group of atoms. For example, unless there is any reason to suppose otherwise, a phenyl
</p>
<p>ring can be considered to obey the geometry ofTables 8.21 and 8.22, or other equivalent compilation, and
</p>
<p>thus be refined as a single, rigid entity.
</p>
<p>The least-squares equations outlined in Sect. 8.4.2 include a weight for each observation. They
</p>
<p>may be unity or calculated from (8.97) for the early stages of refinement. Subsequently, weighting
</p>
<p>schemes based on Fo or sin y or on both of these parameters may be used. The validity of any
</p>
<p>weighting scheme should be checked for the given problem, as already indicated. Attention to detail
</p>
<p>in least-squares refinement is generally rewarding, and it is worth remembering that it consumes
</p>
<p>about three-quarters of the total calculation time of a structure analysis.
</p>
<p>8.4.4 Least-Squares Refinement Against Fo
2
</p>
<p>Some crystallographic program packages, notably SHELX-97, employ full-matrix least-squares
</p>
<p>refinement routines based on Fo
2 and jFcj
</p>
<p>2, as opposed to Fo and jFcj as shown in (8.81). Apart from
</p>
<p>this major difference which will influence the form of the normal equations, the method of solution to
</p>
<p>produce parameter shifts and estimated standard deviations, s, will be entirely parallel to the procedure
</p>
<p>described above for refinement against Fo and jFcj. Studies have shown that refinement against jFj
2
</p>
<p>allows more weighted experimental data to be incorporated into the analysis which enables it to
</p>
<p>proceed more smoothly. A further rationale in favor of this choice is that it is Fo
2 and s(Fo
</p>
<p>2) that are,
</p>
<p>after corrections, obtained from the diffraction experiments so why refine against Fo and jFcj?
</p>
<p>Refinement against Fo also involves mathematical problems with very weak reflections or reflections
</p>
<p>8.4 Least-Squares Refinement 407</p>
<p/>
</div>
<div class="page"><p/>
<p>with apparent negative measured intensity values. There are also difficulties in converting and
</p>
<p>estimating s(F) from the measured s(F2) values for weak or zero measured intensities. Refinement
</p>
<p>against jFj2 avoids these difficulties, and also reduces the probability of the refinement iterations
settling into a local minimum. It also simplifies the treatment of twinned crystals and the determination
</p>
<p>of the absolute configuration in the case of non-centrosymmetric structures, Sect. 7.5.1. For these
</p>
<p>reasons, it is probably currently the most frequently used technique, although it does rely heavily on
</p>
<p>the assignment of reasonable weights to individual reflections. Further details are to be found in the
</p>
<p>SHELX references [21, 23].
</p>
<p>8.4.5 Constraints and Restraints
</p>
<p>The program SHELX-97 allows the user to apply a wide range of controls on the structural
</p>
<p>parameters being refined in any particular case. It is necessary in this respect to distinguish between
</p>
<p>a constraint and a restraint. A constraint is an exact mathematical condition that enables one or more
</p>
<p>least-squares variables to be expressed exactly in terms of other variables or constants, and as such is
</p>
<p>eliminated from the actual refinement process while still contributing as part of the structure to jFcj.
An example is the fixing of an atom position exactly on a center of symmetry. A restraint involves
</p>
<p>additional information that is not exact but is subject to a probability distribution; for example
</p>
<p>chemically but not crystallographically equivalent bonds could be restrained to be approximately
</p>
<p>equal within a specified tolerance. A restraint is treated as an extra experimental observation, with an
</p>
<p>appropriate esd that determines its weight relative to the X-ray data.
</p>
<p>Constraints are flagged in SHELX-97 using appropriate instructions prefixed by AFIX. A wide
</p>
<p>variety of options is available as described in the manual. An HFIX instruction generates AFIX
</p>
<p>instructions and dummy hydrogen atoms bonded to specified atoms. All types of hydrogen atom
</p>
<p>groupings are available. For example, AFIX 6 fits a regular hexagon to six specified atoms, of
</p>
<p>default bond length 1.39 Å for aromatic C&ndash;C bonds, and AFIX 13 specifies an ideal &ndash;CH3 group
</p>
<p>with tetrahedral angles. A DFIX instruction allows the distance between two named atoms to be
</p>
<p>restrained to a target value d within a specified standard deviation.
</p>
<p>These facilities have a wide variety of uses. For example, to economize on parameters if the data/
</p>
<p>parameter ratio is low, or to enable hydrogen atoms to be included in the analysis when it has not been
</p>
<p>possible to locate them from an appropriate electron density map.
</p>
<p>8.5 Molecular Geometry
</p>
<p>When the structure analysis is complete, the results must be expressed in terms of molecular geometry
</p>
<p>and crystal packing. Thus, we need to compute bond lengths, bond angles, intermolecular contact
</p>
<p>distances, torsion angles, where necessary, all with measures of their precision.
</p>
<p>8.5.1 Bond Lengths and Angles
</p>
<p>Consider three atoms with fractional coordinates x1, y1, z1; x2, y2, z2; and x3, y3, z3 in a unit cell of sides
</p>
<p>a, b, and c, Fig. 8.23. The vector rj from the origin O to any atom j is given by
</p>
<p>rj &frac14; xja&thorn; yjb&thorn; zjc (8.101)
</p>
<p>408 8 Direct Methods and Refinement</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec23_7">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec23_7</a></div>
</div>
<div class="page"><p/>
<p>The vector r12 between atoms 1 and 2 is given by
</p>
<p>r12 &frac14; r2 � r1 (8.102)
</p>
<p>or, using (8.101),
</p>
<p>r12 &frac14; &eth;x2 � x1&THORN; a&thorn; &eth;y2 � y1&THORN; b&thorn; &eth;z2 � z1&THORN; c (8.103)
</p>
<p>Forming the dot product of each side with itself, remembering that
</p>
<p>p 
 q &frac14; pq coscpq (8.104)
</p>
<p>we obtain
</p>
<p>r212 &frac14; &eth;x2 � x1&THORN;
2
a2 &thorn; &eth;y2 � y1&THORN;
</p>
<p>2
b2 &thorn; &eth;z2 � z1&THORN;
</p>
<p>2
c2 &thorn; 2&eth;y2 � y1&THORN;&eth;z2 � z1&THORN; bc cos a&thorn; 2&eth;z2
</p>
<p>� z1&THORN;&eth;x2 � x1&THORN; ca cos b&thorn; 2&eth;x2 � x1&THORN;&eth;y2 � y1&THORN; ab cos g (8.105)
</p>
<p>This equation may be simplified for crystal systems other than triclinic. Thus, if the atoms exist in a
</p>
<p>tetragonal unit cell, for example,
</p>
<p>r212 &frac14; &frac12;&eth;x2 � x1&THORN;
2 &thorn; &eth;y2 � y1&THORN;
</p>
<p>2�a2 &thorn; &eth;z2 � z1&THORN;
2
c2 (8.106)
</p>
<p>In a similar manner, we can evaluate r32, Fig. 8.23.
</p>
<p>In the case of a bond angleF123 formed from atoms 1, 2, and 3, and using (8.105) for the tetragonal
</p>
<p>system,
</p>
<p>cosF123 &frac14;
&frac12;&eth;x2 � x1&THORN;&eth;x2 � x3&THORN; &thorn; &eth;y2 � y1&THORN;&eth;y2 � y3&THORN;� a
</p>
<p>2 &thorn; &eth;z2 � z1&THORN;&eth;z2 � z3&THORN; c
2
</p>
<p>r12r32
(8.107)
</p>
<p>Fig. 8.23 Geometry of the
calculation of interatomic
</p>
<p>distances and angles; points
</p>
<p>1, 2, and 3 represent atomic
</p>
<p>positions
</p>
<p>8.5 Molecular Geometry 409</p>
<p/>
</div>
<div class="page"><p/>
<p>where r12 and r32 are evaluated following (8.106). Similar equations enable any distance or angle to
</p>
<p>be calculated, in any crystal system, in terms of the atomic coordinates and unit-cell dimensions.
</p>
<p>When the asymmetric unit of a crystal contains more than one copy of a given molecule, or when
</p>
<p>similar molecules occur in different crystals, the question arises as to whether or not the several sets of
</p>
<p>molecular dimensions are significantly different. The statistical test applicable in this situation is the
</p>
<p>w2 test. It involves calculation of
Pn
</p>
<p>i&frac14;1 &frac12;Di=s&eth;Di&THORN;�
2
, which is distributed as w2with n degrees of freedom;
</p>
<p>Di is the difference between one measured property, such as a bond length, in a pair of molecules, and
</p>
<p>s(Di)is the standard deviation inDi, estimated as [s
2(di1) + s
</p>
<p>2 (di2)]
1/2, assuming no correlation between
</p>
<p>di1 and di2, and n is the number of pairs of measurements.
</p>
<p>Significance Tests and x2 Distributions
The significance of a result can be tested bymaking the null hypothesis that all of the differences can be
</p>
<p>accounted for by random errors in the experimental procedures, and then obtaining from statistical
</p>
<p>tables the significance level of the test, that is, the probability P of incorrectly rejecting a good
</p>
<p>hypothesis. Normally, the test is not regarded as significant unless P � 0.05. Bond lengths and angles
and their estimated standard deviations are calculated and listed by SHELX-97 through the BOND
</p>
<p>instruction in the .ins file.
</p>
<p>As an example we will consider two aspects of the bond lengths in the X-ray structure [43] of the
</p>
<p>cyclic diamino acid peptide N,N0-diacetyl-cyclo(Gly-Gly), shown in the diagram below; the eight ring
</p>
<p>bonds, including the C&frac14;O bonds and the six side-chain bonds, will be considered in the table below.
</p>
<p>The crystal structure has two independent copies of the molecule labeled A and B.
</p>
<p>N,N0-Diacetyl-Cyclo(Gly-Gly) showing the atom numbering scheme (Gly &frac14; glycine)
</p>
<p>Using Table (b) below, we can show that:
</p>
<p>Table a Bond lengths for N,N0-diacetyl-cyclo(Gly-Gly), molecules A and B
</p>
<p>Bond lengths (Å) w2 Components
</p>
<p>Molecule A B Di s(Di) [Di/s(Di)] [Di/s(Di)]
2
</p>
<p>C(1)&ndash;N(2) 1.474(2) 1.472(2) 0.002 0.0028 0.714 0.51 Ca&ndash;Na
</p>
<p>N(2)&ndash;C(3) 1.381(2) 1.383(2) 0.002 0.0028 0.714 0.51 N&ndash;C(O)
C(3)&ndash;C(4) 1.512(3) 1.508(3) 0.004 0.0042 0.953 0.91 C(O)&ndash;Ca
C(4)&ndash;N(5) 1.469(2) 1.474(2) 0.005 0.0028 1.786 3.19 Ca&ndash;N
N(5)&ndash;C(6) 1.386(2) 1.389(2) 0.003 0.0028 1.071 1.15 N&ndash;C(O)
C(1)&ndash;C(6) 1.507(3) 1.505(3) 0.002 0.0042 0.476 0.23 Ca&ndash;C(O)
C(6)&ndash;O(6) 1.217(2) 1.215(2) 0.002 0.0028 0.714 0.51 C(O)&frac14;O
C(3)&ndash;O(3) 1.214(2) 1.209(2) 0.005 0.0028 1.786 3.19 C(O)&frac14;O
Pn
i&frac14;1
</p>
<p>&frac12;Di=s&eth;Di&THORN;�
2 &frac14; 10:20 for n &frac14; 8
</p>
<p>C(2)&ndash;N(2) 1.416(2) 1.413(2) 0.003 0.0028 1.071 1.15 Side-chain bonds
C(2)&ndash;O(2) 1.217(2) 1.218(2) 0.001 0.0028 0.357 0.13
C(2)&ndash;C(21) 1.492(3) 1.493(3) 0.001 0.0042 0.238 0.06
C(5)&ndash;N(5) 1.415(2) 1.410(2) 0.005 0.0028 1.786 3.19
C(5)&ndash;O(5) 1.213(2) 1.212(2) 0.001 0.0028 0.357 0.13
C(5)&ndash;C(51) 1.486(3) 1.474(3) 0.012 0.0042 2.857 8.16
Pn
i&frac14;1
</p>
<p>&frac12;Di=s&eth;Di&THORN;�
2 &frac14; 12:82 for n &frac14; 6
</p>
<p>aStandard peptide bond notation is used here (see Chap. 10)
</p>
<p>410 8 Direct Methods and Refinement</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_10">http://dx.doi.org/10.1007/978-1-4614-3954-7_10</a></div>
</div>
<div class="page"><p/>
<p>1. For the eight ring bond lengths the probability Pring, the difference between molecules A and B, is
</p>
<p>between 0.2 and 0.3, that is greater than 0.05 and is not significant
</p>
<p>2. For the six side-chain bond lengths the probability Pchain, the differences between molecules A and
</p>
<p>B, lies between 0.01 and 0.05, less than 0.05, so that they are significant. Exact values of P are
</p>
<p>[44]: Pring &frac14; 0.251 and Pchain &frac14; 0.0463, thus confirming the above results
It is not too difficult to see that the main difference in side-chain bond lengths between molecules
</p>
<p>A and B occurs for C(5)&ndash;C(51); in practice it may be necessary to seek further reasons for this type of
</p>
<p>result.
</p>
<p>8.5.2 Torsion Angles
</p>
<p>Torsion angles are useful conformational parameters with which to compare different, related
</p>
<p>molecules or, indeed, different conformations of one and the same molecule. In a freely rotating
</p>
<p>moiety, a torsion angle may be a function of the environment of the molecule. Consider
</p>
<p>an arrangement of four atoms 1, 2, 3, 4, Fig. 8.24. The torsion angle w(1, 2, 3, 4) is defined by the
</p>
<p>angle between the planes 1, 2, 3 and 2, 3, 4, and lies in the range �180� &lt; w � 180�; the sign is an
important property of the parameter.
</p>
<p>Table b Chi-square distribution [45] w&sup2; vs. Probability, P. This table gives a number of P values matching to w&sup2; for the
first 10� of freedom, df. A P-value of 0.05 or less is usually regarded as statistically significant, that is, the observed
deviation from the null hypothesis is significant.
</p>
<p>Degrees of
</p>
<p>freedom (df) Probability (P)
</p>
<p>0.95 0.90 0.80 0.70 0.50 0.30 0.20 0.10 0.05 0.01 0.001
</p>
<p>1 0.004 0.02 0.06 0.15 0.46 1.07 1.64 2.71 3.84 6.64 10.83
</p>
<p>2 0.10 0.21 0.45 0.71 1.39 2.41 3.22 4.60 5.99 9.21 13.82
</p>
<p>3 0.35 0.58 1.01 1.42 2.37 3.66 4.64 6.25 7.82 11.34 16.27
</p>
<p>4 0.71 1.06 1.65 2.20 3.36 4.88 5.99 7.78 9.49 13.28 18.47
</p>
<p>5 1.14 1.61 2.34 3.00 4.35 6.06 7.29 9.24 11.07 15.09 20.52
</p>
<p>6 1.63 2.20 3.07 3.83 5.35 7.23 8.56 10.64 12.59 16.81 22.46
</p>
<p>7 2.17 2.83 3.82 4.67 6.35 8.38 9.80 12.02 14.07 18.48 24.32
</p>
<p>8 2.73 3.49 4.59 5.53 7.34 9.52 11.03 13.36 15.51 20.09 26.12
</p>
<p>9 3.32 4.17 5.38 6.39 8.34 10.66 12.24 14.68 16.92 21.67 27.88
</p>
<p>10 3.94 4.86 6.18 7.27 9.34 11.78 13.44 15.99 18.31 23.21 29.59
</p>
<p>Nonsignificant Significant
</p>
<p>Fig. 8.24 The torsion
angle w(1, 2, 3, 4) has a
</p>
<p>positive sign for a rotation
</p>
<p>of 1, 2, about 2, 3 as shown,
</p>
<p>that is, it is positive if,
</p>
<p>looking along r23, a
clockwise rotation is
</p>
<p>required to bring atom
</p>
<p>1 into atom 4
</p>
<p>8.5 Molecular Geometry 411</p>
<p/>
</div>
<div class="page"><p/>
<p>In the planar, eclipsed conformation shown in Fig. 8.24, w is zero. The torsion angle is the amount
</p>
<p>of rotation of 1, 2 about 2, 3 and, looking along the direction 2 ! 3; a positive value of w corresponds
to the clockwise rotation that brings atom 1 into atom 4. Let
</p>
<p>p1 &frac14; r23 ��r12 (8.108)
</p>
<p>and
</p>
<p>p2 &frac14; r23 � r34 (8.109)
</p>
<p>Then
</p>
<p>w&eth;1; 2; 3; 4&THORN; &frac14; cos�1 p1 
 p2
p1p2
</p>
<p>� 	
</p>
<p>(8.110)
</p>
<p>If the torsion angle is calculated by the expression atan2 fjr23jr12:&eth;r23�r34&THORN;; &eth;r12�r23&THORN;:&eth;r23�r34&THORN;g
the angle is obtained with the correct sign. [atan2 is a Fortran function for tan�1, with two arguments].
</p>
<p>Torsion angle calculation is provided by crystallographic software, for example, the program MOL-
</p>
<p>GOM in the program suite, Sect. 13.6.5, or by using the CONF instruction in SHELX-97. WinGX has
</p>
<p>a facility for producing standard tables for publication including molecular geometry: the &ldquo;Publish&rdquo;
</p>
<p>button, Fig. 8.10a, invokes this facility.
</p>
<p>8.5.3 Conformational Analysis
</p>
<p>Confusion has arisen in the literature over the use of torsion angles in conformational analysis. It is
</p>
<p>often convenient to quote values of torsion angles as lying within certain ranges. For example,
</p>
<p>w � 0� may be called cis, w &frac14; 180� is trans, and w � �60� is �gauche. However, because of
</p>
<p>changing conventions, it is best to quote the actual value of w, and to state how it is defined,
</p>
<p>Fig. 8.24. This procedure will minimize ambiguities in the future.
</p>
<p>Ring Conformations
Two types of symmetry (or pseudo-symmetry) must be considered in order to define ring
</p>
<p>conformations [46], namely mirror planes perpendicular to the dominant ring plane and twofold
</p>
<p>axes lying in the ring plane. If there is an odd number (usually 5 or 7) of atoms in the ring, all
</p>
<p>symmetry elements pass through one of the ring atoms and bisect the opposite bond, Fig. 8.25. In
</p>
<p>rings containing an even number of atoms (usually 6), symmetry elements may pass through two
</p>
<p>ring atoms located directly across the ring, or else bisect two opposite ring bonds.
</p>
<p>Ten symmetry elements are possible in five-membered rings. The planar five-membered ring
</p>
<p>possesses all ten, five mirror planes and five twofold axes. The ideal envelope conformation has only a
</p>
<p>single m plane, and it passes through the out-of-plane atom. The ideal half-chair has one twofold axis
</p>
<p>bisecting the bond between the two out-of-plane atoms. Six-membered rings possess 12 locations for
</p>
<p>symmetry elements. In determining the ring conformation, we can ignore the two-, three-, and sixfold
</p>
<p>collinear rotation axes perpendicular to the ring plane. Figure 8.26 illustrates the symmetry elements
</p>
<p>that define the ideal forms of commonly observed conformations. The planar ring, such as in benzene,
</p>
<p>has one m plane and one twofold axis at each of six locations (
6
</p>
<p>m
mm). The chair form of cyclohexane
</p>
<p>has three m planes and three twofold axes (�3m). The boat and twist-boat have point group symmetry
</p>
<p>mm2 and 222, respectively, while the sofa has symmetry m and the half-chair symmetry 2.
</p>
<p>412 8 Direct Methods and Refinement</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec30_13">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec30_13</a></div>
</div>
<div class="page"><p/>
<p>Asymmetry Parameters
Once the atom coordinates are available, torsion angles may be calculated. Because of errors in the
</p>
<p>data and for stereochemical reasons, a particular cyclic structure will often depart from its ideal
</p>
<p>symmetry. The degree of this departure, its asymmetry, may be calculated in terms of asymmetry
</p>
<p>parameters. For this purpose, related or nearly related torsion angles are compared in a way that will
</p>
<p>result in a value of zero for a parameter if the corresponding symmetry is realized in the molecule.
</p>
<p>Mirror-related torsion angles have equal magnitude but opposite sign, and such torsion angles are
</p>
<p>compared by addition. The torsion angles related by twofold symmetry are equal in both magnitude
</p>
<p>and sign, and are compared by subtraction. The rms value of each discrepancy yields a measure of the
</p>
<p>deviation from ideal symmetry at the location in question. We calculate
</p>
<p>DCs &frac14;
Pn
</p>
<p>i&frac14;1 &eth;wi &thorn; w0i&THORN;
2
</p>
<p>n
</p>
<p> !1=2
(8.111)
</p>
<p>in respect of m symmetry, and
</p>
<p>DC2 &frac14;
Pn
</p>
<p>i&frac14;1 &eth;wi � w0i&THORN;
2
</p>
<p>n
</p>
<p> !1=2
(8.112)
</p>
<p>in respect of twofold symmetry; n is the number of individual comparisons, and wi and w
0
i are the
</p>
<p>related torsion angles in question.
</p>
<p>Fig. 8.25 Conformations
in five- and six-membered
</p>
<p>rings. (a) Torsion
angles related by mirror
</p>
<p>planes (&mdash;) have opposite
</p>
<p>signs. (b) Torsion angles
related by twofold rotation
</p>
<p>axes have the same sign
</p>
<p>8.5 Molecular Geometry 413</p>
<p/>
</div>
<div class="page"><p/>
<p>8.5.4 Mean Planes
</p>
<p>In discussing the geometry of a molecule, it may be desirable to test the planarity of a group of atoms.
</p>
<p>For a number n (n &gt; 3) of atoms, the best plane may be obtained by the method of least squares. Let
</p>
<p>the plane be given by
</p>
<p>PX &thorn; QY &thorn; RZ &thorn; S &frac14; 0 (8.113)
</p>
<p>The constants P, Q, R, and S are obtained through an extension of the least-squares procedure
</p>
<p>given in Sect. 8.4; it is desirable to work with Cartesian coordinates (see Appendix C).
</p>
<p>Fig. 8.26 Commonly observed conformations of six-membered rings. The mirror and twofold rotational symmetries
are indicated on the right
</p>
<p>414 8 Direct Methods and Refinement</p>
<p/>
</div>
<div class="page"><p/>
<p>8.6 Precision
</p>
<p>Closely related to the calculations of bond lengths and angles is the expression of the precision of
</p>
<p>these quantities. The least-squares refinement procedure establishes values for an estimated standard
</p>
<p>deviation in each of the variables used in these calculations. Thus, a fractional coordinate of 0.3712
</p>
<p>might have an esd of 0.0003, written as 0.3712(3).
</p>
<p>We need to know further how errors are propagated in a quantity which is a function of several
</p>
<p>variables, each of which contains some uncertainty arising from random errors. The answer is provided
</p>
<p>by the statistical principle of superposition of errors. Let q be a function of several variables pi (i &frac14; 1, 2,
3,. . ., N), with known standard deviations s(pi). Then the esd in q is given through
</p>
<p>s2&eth;q&THORN; &frac14;
XN
</p>
<p>i&frac14;1
</p>
<p>@q
</p>
<p>@pi
s&eth;pi&THORN;
</p>
<p>� �2
</p>
<p>(8.114)
</p>
<p>A simple example may be given, through (8.103) and (8.114), for a bond between two atoms lying
</p>
<p>along the c edge of a tetragonal unit cell. Let c be 10.06(1) Å, z1 be 0.3712(3), and z2 be 0.5418(2).
</p>
<p>From (8.103),
</p>
<p>r12 &frac14; &eth;z2 � z1&THORN;c &frac14; &eth;0:5418� 0:3712&THORN;10:06 &frac14; 1:716A (8.115)
</p>
<p>and from (8.114),
</p>
<p>s2&eth;r12&THORN; &frac14; &eth;0:5418� 0:3712&THORN;2&eth;0:01&THORN;2 &thorn; &eth;10:06&THORN;2&eth;0:0002&THORN;2 &thorn; &eth;10:06&THORN;2&eth;0:0003&THORN;2 (8.116)
</p>
<p>Thus, s(r12) is 0.004 Å and we write r12 &frac14; 1.716(4) Å. Similar calculations may be used for all
distance and angle calculations in all crystal systems, but the general equations are quite involved
</p>
<p>numerically and best handled by computer methods.
</p>
<p>The estimated standard deviation of a torsion angle can be calculated along the lines given in this
</p>
<p>section; the significances of differences between torsion angles may be as important as the differences
</p>
<p>themselves. Again, in a discussion of best-plane results it is essential to evaluate the perpendicular
</p>
<p>distances (deviations) of atoms from the plane, and their esds. Let the jth atom have the coordinates
</p>
<p>Xj, Yj, Zj. Then it is a simple exercise in coordinate geometry to show that the deviation Dj of this atom
</p>
<p>from the best plane (8.113) is given by
</p>
<p>Dj &frac14; &eth;PXj &thorn; QYj &thorn; RZj &thorn; S&THORN;=K (8.117)
</p>
<p>where K is given by
</p>
<p>K &frac14; &eth;P2 &thorn; Q2 &thorn; R2&THORN;1=2 (8.118)
</p>
<p>To obtain the esd in Dj the esds in Xj, Yj, and Zj are first obtained, using (8.114) and results from
</p>
<p>Appendix C. Then
</p>
<p>s&eth;Dj&THORN; &frac14; f&frac12;Ps&eth;Xj&THORN;�2 &thorn; &frac12;Qs&eth;Yj&THORN;�2 &thorn; &frac12;Rs&eth;Zj&THORN;�2g1=2K (8.119)
</p>
<p>8.6 Precision 415</p>
<p/>
</div>
<div class="page"><p/>
<p>8.7 Correctness of a Structure Analysis
</p>
<p>At this stage we may summarize four criteria of correctness of a good structure analysis. If we can
</p>
<p>satisfy these conditions in one and the same structure model, we shall have a high degree of
</p>
<p>confidence in the analysis.
</p>
<p>1. There should be good agreement between Fo and jFcj, expressed through the R factor. Ultimately,
R depends upon the quality of the experimental data. At best, it will probably be about 1% greater
</p>
<p>than the average standard deviation in Fo. Assuming the desirable situation that two or more
</p>
<p>asymmetric units of data have been collected, a value R close to that for Rint, Sect. 10.4.7, for the
</p>
<p>data is acceptable.
</p>
<p>2. The electron density map should show neither positive nor negative density regions that are
</p>
<p>unaccountable, other than Fourier series termination errors.
</p>
<p>3. The difference-Fourier map should be relatively flat. This map eliminates series termination errors
</p>
<p>as they are present in both ro and rc. Random errors produce small fluctuations on a difference
</p>
<p>map, but they should be less than 2.5&ndash;3 times the standard deviation of the electron density s(ro):
</p>
<p>s&eth;ro&THORN; &frac14;
1
</p>
<p>Vc
</p>
<p>X
</p>
<p>hkl
</p>
<p>&eth;DF&THORN;2
" #1=2
</p>
<p>(8.120)
</p>
<p>where DF &frac14; Fo � jFcj and the sum extends over all symmetry-independent reflections.
4. The molecular geometry should be chemically sensible, within the limits of current structural
</p>
<p>knowledge. Abnormal bond lengths and angles may be correct but theymust be supported by strong
</p>
<p>evidence of their validity in order to gain acceptance. Normally a deviation of less than three times
</p>
<p>the corresponding standard deviation is not considered to be statistically significant. As a guide to
</p>
<p>the interpretation of acceptable stereochemistry, we include selections of ionic radii, bond lengths,
</p>
<p>and bond angles in Tables 8.20, 8.21, and 8.22, respectively; see also Table 13.4.
</p>
<p>Table 8.20 Selected ionic radii (Å) referred to coordination number 6a
</p>
<p>Ag+ 1.27 Hg2+ 1.02 Th4+ 0.94
</p>
<p>Al3+ 0.54 K+ 1.44 Ti2+ 0.86
</p>
<p>Ba2+ 1.49 La3+ 1.03 Ti4+ 0.61
</p>
<p>Be2+ 0.48 Li+ 0.86 Tl+ 1.54
</p>
<p>Ca2+ 1.18 Mg2+ 0.87 Tl3+ 1.03
</p>
<p>Cd2+ 0.95 Mn2+ 0.83 Zn2+ 0.74
</p>
<p>Ce3+ 1.01 Na+ 1.12 NH&thorn;4 1.66
</p>
<p>Ce4+ 0.87 Ni2+ 0.69 H� 1.39
</p>
<p>Co2+ 0.75 Pb2+ 1.19 F� 1.19
</p>
<p>Co3+ 0.61 Pd2+ 0.86 Cl� 1.70
</p>
<p>Cr3+ 0.62 Pt2+ 0.80 Br� 1.87
</p>
<p>Cs+ 1.84 Pt4+ 0.63 I� 2.20
</p>
<p>Cu+ 0.77 Ra2+ 1.43 O2� 1.40
</p>
<p>Cu2+ 0.73 Rb+ 1.58 S2� 1.70
</p>
<p>Fe2+ 0.78 Sn2+ 0.93 Se2� 1.81
</p>
<p>Fe3+ 0.65 Sr2+ 1.32 Te2� 1.97
aThe changes in ionic radius from coordination number 6 to coordination
</p>
<p>numbers 8, 4, 3, and 2 are approximately +1.5 %, �1.5 %, �3.0 %, and
�3.5 %, respectively
</p>
<p>416 8 Direct Methods and Refinement</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec27_10">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec27_10</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Tab4_13">http://dx.doi.org/10.1007/978-1-4614-3954-7#Tab4_13</a></div>
</div>
<div class="page"><p/>
<p>8.7.1 Databases
</p>
<p>Tables of standard (average) bond lengths and angles (both with esds) are useful aids to structure
</p>
<p>determination. In any research or advanced study, it is necessary to take cognizance of all work in the
</p>
<p>given field that has already been published. The number of crystal structures that has been solved and
</p>
<p>published is now vast, and data files have been constructed that can be interrogated by computer. The
</p>
<p>best known of these is the Cambridge Structural Database (CSD) [47]. It contains the results of both
</p>
<p>X-ray and neutron diffraction studies on organic and organometallic compounds. At the time of
</p>
<p>writing, over 544,000 crystal structures are filed in this database, summarized in Table 8.23, and the
</p>
<p>database is available in about 25 countries (see also Appendix D). The Cambridge Structural Database
</p>
<p>System (CSDS) is a single product that comprises the following components: Cambridge Structural
</p>
<p>Database (CSD); CSDS Software: ConQuest, Mercury, VISTA, and PreQuest. Knowledge bases
</p>
<p>derived from the CSD: Mogul and IsoStar. Life Sciences Products: SuperStar, Hermes, GOLD,
</p>
<p>GoldMine, Relibase+. Powder Diffraction Products: DASH.
</p>
<p>One criticism of the CSD has been that it does not store the experimental Fo data on which each
</p>
<p>structure is based.TheProteinDataBank (PDB) doeskeep these data for eachprotein structure deposited.
</p>
<p>Not every structure in the above classes that has been ever published will be found in the CSD.
</p>
<p>Table 8.21 Selected bond lengths (Å)a
</p>
<p>Formal single bonds Formal double bonds
</p>
<p>C4&ndash;H 1.09 C3&ndash;C2 1.45 C3&ndash;C3 1.34 C2&ndash;O1 1.16
C3&ndash;H 1.08 C3&ndash;N3 1.40 C3&ndash;C2 1.31 N3&ndash;O1 1.24
C2&ndash;H 1.06 C3&ndash;N2 1.40 C3&ndash;N2 1.32 N2&ndash;N2 1.25
N3&ndash;H 1.01 C3&ndash;O2 1.36 C3&ndash;O1 1.22 N2&ndash;O1 1.22
N2&ndash;H 0.99 C2&ndash;C2 1.38 C2&ndash;C2 1.28 O1&ndash;O1 1.21
O2&ndash;H 0.96 C2&ndash;N3 1.33 C2&ndash;N2 1.32
</p>
<p>Formal triple bonds
</p>
<p>C4&ndash;C4 1.54 C2&ndash;N2 1.33 C2&ndash;C2 1.20 N1&ndash;N1 1.10
C4&ndash;C3 1.52 C2&ndash;O2 1.36 C2&ndash;N1 1.16
C4&ndash;C2 1.46 N3&ndash;N3 1.45
C4&ndash;N3 1.47 N3&ndash;N2 1.45
</p>
<p>Aromatic bonds
</p>
<p>C4&ndash;N2 1.47 N3&ndash;O2 1.36 C2&ndash;C3 1.40 N2&ndash;N2 1.35
C4&ndash;O2 1.43 N2&ndash;N2 1.45 C2&ndash;N2 1.34
C3&ndash;C3 1.46 N2&ndash;O2 1.41
</p>
<p>aThe notation in the table indicates the connectivity of the atoms
</p>
<p>Table 8.22 Selected bond angles
</p>
<p>Atom Geometry Angle (�)
</p>
<p>C4 Tetrahedral 109.47
</p>
<p>C3 Planar 120
</p>
<p>C2 Bent 109.47
</p>
<p>C2 Linear 180
</p>
<p>N4 Tetrahedral 109.47
</p>
<p>N3 Pyramidal 109.47
</p>
<p>N3 Planar 120
</p>
<p>N2 Bent 109.47
</p>
<p>N2 Linear 180
</p>
<p>O3 Pyramidal 109.47
</p>
<p>O2 Bent 109.47
</p>
<p>8.7 Correctness of a Structure Analysis 417</p>
<p/>
</div>
<div class="page"><p/>
<p>Each entry has to pass a scrutiny that involves such checks as the consistency between the published
</p>
<p>coordinates and bond lengths. The information in the CSD falls into three categories, namely, biblio-
</p>
<p>graphic, connective, and crystallographic. The bibliographic file contains information such as the
</p>
<p>chemical name, the type of structure analysis, the chemical class, the molecular formula, and relevant
</p>
<p>literature for a given compound. The connective file contains chemical structural formulae encoded as
</p>
<p>atom and bond parameters, and the crystallographic file contains parameters relevant to the crystal
</p>
<p>structure data and its solving.
</p>
<p>Retrieval from the database is flexible, and the software permits many different types of search,
</p>
<p>such as on chemical name, formula, or class, and compounds containing specific chemical fragments
</p>
<p>can be sought. The results of searches can be recorded by printing and plotting techniques. File
</p>
<p>formats which can be retrieved include .cif, .mol2, .pdb, SHELXL, and laztpulverix. The .cif file can
</p>
<p>be used, for example, to display the structure with MERCURY and is required to be deposited by
</p>
<p>authors to enable detailed checking of the structure determination and results.
</p>
<p>Crystallographic programs such as SHELXL produce the .cif file at the completion of the analysis.
</p>
<p>The file can be checked for errors using programs in WinGX or other analysis systems or by using
</p>
<p>ENCIFER available through the CCDC [47]. Any alert produced by running the checking facilities
</p>
<p>should be dealt with by the authors prior to submission of their results for publication.
</p>
<p>Similar databases have been organized: for proteins by the Resource for Studying Biological
</p>
<p>Molecules (RCSB) [48]; for inorganic structures at the Inorganic Crystal Structure Database (ICSD)
</p>
<p>[49], a database of inorganic and related structures produced cooperatively by FIZ Karlsruhe and the
</p>
<p>National Institute of Standards and Technology (NIST) [50] and searchable via two different web
</p>
<p>browser interfaces, CrystalWeb [51] and ICSD-WWW [49]; for metals, alloys, and intermetallic
</p>
<p>compounds: the CrystMet [52] database at Daresbury, UK which contains some 75,000 crystal
</p>
<p>structure data.
</p>
<p>Table 8.23 Statistics of the Cambridge Structural Database (CSD), 2011
</p>
<p>Structures %CSD
</p>
<p>Total number of structures 596,810 100.0
</p>
<p>Number of different compounds 544,565 &ndash;
</p>
<p>Number of literature sources 1,429 &ndash;
</p>
<p>Organic structures 254,475 42.6
</p>
<p>Transition metal present 319,188 53.5
</p>
<p>Li&ndash;Fr or Be&ndash;Ra present 30,134 5.0
</p>
<p>Main group metal present 36,923 6.2
</p>
<p>3D coordinates present 554,760 93.0
</p>
<p>Error-free coordinates 545,085 98.3{
</p>
<p>Neutron studies 1,534 0.3
</p>
<p>Powder diffraction studies 2,354 0.4
</p>
<p>Low/high temperature studies 250,328 41.9
</p>
<p>Absolute configuration determined 11,111 1.9
</p>
<p>Disorder present in structure 132,349 22.2
</p>
<p>Polymorphic structures 18,386 3.1
</p>
<p>R-factor &lt; 0.100 559,093 93.6
</p>
<p>R-factor &lt; 0.075 506,465 84.9
</p>
<p>R-factor &lt; 0.050 325,440 54.5
</p>
<p>R-factor &lt; 0.030 65,897 11.0
</p>
<p>No. of atoms with 3D coordinates 45,048,092 &ndash;
</p>
<p>{Taken as a precentage of structures for which 3D coordinates are present
</p>
<p>in the CSD
</p>
<p>418 8 Direct Methods and Refinement</p>
<p/>
</div>
<div class="page"><p/>
<p>8.8 Limitations of X-Ray Structure Analysis
</p>
<p>There are certain things that X-ray analysis cannot do well, and it is prudent to consider the more
</p>
<p>important of them.
</p>
<p>Liquids and gases lack three-dimensional order, and cannot be used in diffraction experiments in
</p>
<p>the same way as are crystals. Certain information about the radial distribution of electron density can
</p>
<p>be obtained, but it lacks the distinctive detail of crystal analysis.
</p>
<p>It is not easy to locate light atoms in the presence of heavy atoms. Difference-Fourier maps
</p>
<p>alleviate the situation to some extent, but the atomic positions are not necessarily precise. Least-
</p>
<p>squares refinement of light-atom parameters is not always successful, because the contributions to the
</p>
<p>structure factor from these atoms are relatively small.
</p>
<p>Hydrogen atoms are particularly difficult to locate with precision because of their small scattering
</p>
<p>power and the fact that the center of the hydrogen atom does not, in general, coincide with the
</p>
<p>maximum of its electron density. Terminal hydrogen atoms have a more aspherical electron density
</p>
<p>distribution than do hydrogen-bonded hydrogen atoms, and their bond distances, from X-ray studies,
</p>
<p>often appear short when compared with spectroscopic or neutron diffraction values. For similar
</p>
<p>reasons, refinement of hydrogen-atom parameters in a structure analysis may be imprecise, and the
</p>
<p>standard deviations in their coordinate values may be as much as ten times greater than those for a
</p>
<p>carbon atom in the same structure. It is, nevertheless, very desirable to include hydrogen-atom
</p>
<p>positions in the final structure model. They lead to a best fit, and are useful when comparing the
</p>
<p>results of X-ray structure determination with those of other techniques, notably nuclear magnetic
</p>
<p>resonance.
</p>
<p>In general, bond lengths determined by X-ray methods represent distances between the centers of
</p>
<p>gravity of the electron clouds, which may not be the same as the internuclear separations. Internuclear
</p>
<p>distances can be found from neutron diffraction data, because neutrons are scattered by the atomic
</p>
<p>nuclei. If, for a given crystal, the synthesized neutron scattering density is subtracted from that of the
</p>
<p>X-ray scattering density, a much truer picture of the electron density can be obtained. Neutron
</p>
<p>diffraction is discussed in detail in Chap. 11.
</p>
<p>8.9 Disorder in Single Crystals
</p>
<p>A typical small-molecule analysis may involve less than about 100 non-hydrogen atoms in the
</p>
<p>asymmetric unit. With Cu Ka radiation and ymax tending to 70
�, it would be expected to lead to the
</p>
<p>determination of bond lengths with esds of about 0.005 Å and of bond angles with esds of about 0.2�.
Isotropic thermal parameters for non-hydrogen atoms usually range from 0.050 to 0.090 Å2 and may
</p>
<p>have esds from 0.003 to 0.007 Å2. However, it is sometimes found that the refined thermal parameters
</p>
<p>for certain atoms in a structure have atypical values. For example, Uiso may increase progressively
</p>
<p>and significantly toward the end of a chain-like moiety compared to the more rigid areas of the
</p>
<p>structure. The obvious and reasonable physical interpretation is simply that the atoms near the end of
</p>
<p>the chain experience greater thermal motion than do the atoms in the bulk of the molecule. For
</p>
<p>example, the hydroxyethyl side-chain atoms (excluding hydrogen) of an azasteroid derivative [53]
</p>
<p>have the following Uiso parameters:
</p>
<p>&gt;N &ndash;CH3 &ndash;CH2 &ndash;OH
</p>
<p>0.058 0.073 0.190 0.176
</p>
<p>8.9 Disorder in Single Crystals 419</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_11">http://dx.doi.org/10.1007/978-1-4614-3954-7_11</a></div>
</div>
<div class="page"><p/>
<p>In the analysis of this azasteroid all atoms, including those in the side chain, were resolved and
</p>
<p>refined successfully by least squares.
</p>
<p>Atoms in solvent of crystallization molecules may exhibit high thermal parameters, and for similar
</p>
<p>reasons. Exceptions occur from time to time, and in the above example a well-resolved solvent water-
</p>
<p>oxygen atom had a refined Uiso value of 0.088 Å
2 and was so well ordered that its hydrogen atoms
</p>
<p>were clearly located in a DF map, Fig. 8.27. In this particular case, the clarity of definition in the
</p>
<p>electron density is associated with the formation of two strong hydrogen bonds donated by each of the
</p>
<p>water-hydrogen atoms holding it firmly in position. However, in the structure of another steroid
</p>
<p>derivative [54], the carbon atoms of the side chains were so badly disordered that some atoms were
</p>
<p>not resolved in the difference electron density and appeared as diffuse patches, Figs. 8.28 and 8.29.
</p>
<p>Such disorder is probably of a statistical nature, with the atoms taking up slightly different positions
</p>
<p>from one unit cell to another. The effect can be compensated, albeit somewhat artificially, by the
</p>
<p>refinement of the isotropic thermal parameters assigned to the atoms concerned. In the example, the
</p>
<p>Uiso values are three to six times greater than those of the ordered atoms in the structure.
</p>
<p>Disorder may also arise by groups of atoms either in free rotation in the solid state (dynamic
</p>
<p>disorder) or in more than one position of similar energy (static disorder). Methyl groups in large
</p>
<p>organic molecules often show this type of behavior. It may be possible to distinguish between
</p>
<p>dynamic and static disorder by a complete reexamination of the structure at a much reduced
</p>
<p>temperature.
</p>
<p>Protein structures are of particular topical interest, and innovations in this field include the
</p>
<p>development of techniques for refining these large structures [55]. The molecules involved in protein
</p>
<p>analysis are very large, typically with more than 1000 non-hydrogen atoms in the asymmetric unit.
</p>
<p>Consequently, the crystals have large unit cells, and many possible X-ray reflections occur within a
</p>
<p>given y-range compared to small-molecule crystals. It is customary to limit severely the maximum
</p>
<p>Fig. 8.27 Difference electron density map for azasteroid HS626 showing the hydrogen atom on O1 (hydroxyl) and the
two water-hydrogen atoms, none of which was included in the structure factor calculation. The steroid molecule, part of
</p>
<p>which is shown by the dashed line, has been subtracted out in the difference synthesis (This figure and the next five are
</p>
<p>electron density and difference electron density maps photographed from the screen of an Evans and Sutherland Picture
</p>
<p>System 2 cathode ray tube display unit coupled to a computer that holds the electron density data. The interactive
</p>
<p>computer graphics system is programmed such that the user can simulate a three-dimensional effect by rotating the map
</p>
<p>about one or more of three mutually perpendicular axes. The contouring of the maps encloses the electron density in a
</p>
<p>cage of &ldquo;chicken wire&rdquo; hoops running in several directions. Unlike the sectional contour maps used elsewhere in this
</p>
<p>book, only one contour level is used, selected so as to optimize the desired features of electron density)
</p>
<p>420 8 Direct Methods and Refinement</p>
<p/>
</div>
<div class="page"><p/>
<p>Fig. 8.28 Electron density at the end of the cholesteryl side chain of HS650 (molecule B). The density is smeared out,
</p>
<p>and at least two stereochemically sensible positions for the C25
</p>
<p>C26
</p>
<p>C27
</p>
<p>fragment can be fitted to the density, as indicated
</p>
<p>Fig. 8.29 As for Fig. 8.28, but with molecule A of HS650, in which the disorder in the side chain is more extensive and
encompasses C23&ndash;C27. In this structure the side chains are loosely held, having little contact with neighboring
</p>
<p>molecules in the crystal
</p>
<p>8.9 Disorder in Single Crystals 421</p>
<p/>
</div>
<div class="page"><p/>
<p>y-value during the course of a protein structure analysis, depending on the particular stage reached.
</p>
<p>Corresponding to a given maximum y-value there is a minimum d-value, l=2 sin ymax and it is
</p>
<p>customary to speak of the dmin of an analysis as the nominal resolution. The protein analysis may
</p>
<p>proceed through stages of progressively higher resolution, for example, 6, 3, 2.5, 2, and 1.5 Å, the
</p>
<p>electron density image undergoing gradually improved mathematical focusing in the process.
</p>
<p>In addition to the large quantity of data associated with protein analysis, there is a further
</p>
<p>technical problem, which limits the quality of most studies. During the process of crystallization
</p>
<p>from solution, solvent molecules, typically 40&ndash;60% by weight, are trapped in the structure: the
</p>
<p>protein molecules almost float in a solvated crystalline state [56], and consequently many regions of
</p>
<p>electron density in the protein structure may be subject to the type of disorder described above. Even
</p>
<p>in a good high-resolution analysis, protein data rarely extend beyond 1.5 Å, whence individual atoms
</p>
<p>in the protein molecule may not be revealed. A protein structure refinement involves the use of both
</p>
<p>least-squares analysis and geometrically constrained positioning of groups in order to produce a
</p>
<p>plausible model.
</p>
<p>We conclude this section with an example of electron density determined in the high-resolution X-
</p>
<p>ray analysis of the enzyme ribonuclease [57], a small protein of molecular weight 13700. In the first
</p>
<p>example, a tyrosyl residue, Fig. 8.30, is seen at 0.85 Å extremely high resolution as a hollow ring of
</p>
<p>density and, although the individual atoms are not resolved, the shape of the density image is strikingly
</p>
<p>good. As would be expected, the sulfur atom of a methionine residue, Fig. 8.31, is quite outstanding,
</p>
<p>but it does not swamp the rest of this slender aliphatic side chain. At this resolution, the high quality of
</p>
<p>the refined analysis is evident in the appearance of resolved, solvent (water) molecules, as shown in
</p>
<p>Fig. 8.32. Further practical details of protein analysis are discussed in detail in Chap. 10.
</p>
<p>8.10 Computer Prediction of Crystal Structures
</p>
<p>Recent work has involved computer programs that fit structures to a givenmolecular conformation and
</p>
<p>minimize the lattice energy of the chosen structure model so as to obtain an energetic &ldquo;best fit.&rdquo; As an
</p>
<p>example, we consider the known crystal structure of 5-azauracil, Figs. 8.33 and 8.34.
</p>
<p>8.10.1 Crystal Structure of 5-Azauracil [58]
</p>
<p>A successful modeling procedure requires: (1) an accurate model of the molecule; (2) a formula for
</p>
<p>the intermolecular force function, and (3) a method for generating close-packed structures. In this
</p>
<p>example, the molecular structure model was obtained through an energy optimization of an SCF 6-
</p>
<p>31G** wavefunction, using the program CADPAC [59].
</p>
<p>The model for the electrostatic contribution to the lattice energy involved calculating sets of
</p>
<p>atomic multipoles derived by a distributed multipole analysis [60] of the MP2 6-31G** wavefunction
</p>
<p>for 5-azauracil. Other intermolecular forces were represented by an isotropic atom&ndash;atom repulsion
</p>
<p>and attraction (dispersion) potential. Thus the lattice energy U was written as
</p>
<p>U &frac14;
X
</p>
<p>i21;j22
&eth;AppAqq&THORN;1=2 exp&frac12;�&eth;BppBqq&THORN;Rij=2� � &eth;CppCqq&THORN;=R6ij (8.121)
</p>
<p>where atom i in molecule 1 is of type p, atom j in molecule 2 is of type q, and Rij is the i&ndash;j
</p>
<p>intermolecular distance; the best known values [60, 61] of the parameters A, B, and C for the C, H,
</p>
<p>O, and N atoms were used. This form of the potential function has been found to be successful in
</p>
<p>representing other small, rigid C, H, O, N molecular species.
</p>
<p>422 8 Direct Methods and Refinement</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_10">http://dx.doi.org/10.1007/978-1-4614-3954-7_10</a></div>
</div>
<div class="page"><p/>
<p>The program MOLPAK [62] was used to select close-packed molecular structures, using a simple
</p>
<p>hard-sphere repulsion potential function. The lattice energies were calculated with the
</p>
<p>program DMAREL [62], and minimization of the lattice energy was carried out by a modified
</p>
<p>Newton&ndash;Raphson method that optimized the unit-cell dimensions and the rotations and translations of
</p>
<p>each molecule in the unit cell; the symmetry was generally maintained, although the minimization
</p>
<p>procedure did not enforce it. The ten structures with the lower initial lattice energies corresponded to
</p>
<p>space groups P�1, P21, P21/c, and P212121. Structures in these space groups were minimized, and on the
</p>
<p>basis of the results MOLPAK crystal structures were generated with the less common space groups P1,
</p>
<p>C2/c, Pna21, Pca21, and Pbca. Table 8.25 lists the predicted unit-cell dimensions and hydrogen-bond
</p>
<p>distances, which are very structure-sensitive, for the six lower energy-minimized structures, together
</p>
<p>with the corresponding results from the X-ray study.
</p>
<p>Fig. 8.30 Figures 8.30&ndash;8.33 show extracts from the electron density map (COOT, Sect. 10.8.3) of ribonuclease-A at
0.85 Å extremely high resolution. The maps are calculated with the coefficients Fo + (Fo � jFcj) &frac14; 2Fo � jFcj and
thus show features of both the electron density and the difference electron density. This figure shows a tyrosyl residue,
</p>
<p>HOC6H4CH2CH&lt;, with the hole of the phenyl ring and the &ndash;OH group (at the top) clearly indicated. [Data collected at
the Diamond Light Source synchrotron (Sect. 3.1.6) at 100 K and 0.77 Å wavelength.]
</p>
<p>8.10 Computer Prediction of Crystal Structures 423</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec18_10">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec18_10</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec8_3">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec8_3</a></div>
</div>
<div class="page"><p/>
<p>Fig. 8.31 Methionyl
residue,
</p>
<p>CH3SCH2CH2CH&lt;,
showing the outstanding
</p>
<p>electron density region
</p>
<p>around the sulfur atom
</p>
<p>Fig. 8.32 This electron
density portion shows a
</p>
<p>clearly resolved solvent
</p>
<p>molecule (water), not
</p>
<p>included in the structure
</p>
<p>factor calculation
</p>
<p>424 8 Direct Methods and Refinement</p>
<p/>
</div>
<div class="page"><p/>
<p>Conclusions
The X-ray determination [58] and the best modeled structure of 5-azauracil, labeled * in Table 8.25,
</p>
<p>are in good agreement. Both are in space group Pbca with unit-cell parameters all within 0.3 Å, the
</p>
<p>modeled structure requiring a unit-cell transformation a0 &frac14; b, b0 &frac14; c, c0 &frac14; a in order to conform to
the X-ray setting. The small energy differences, Table 8.25, of the order of 1&ndash;2 kJ mol�1, between the
</p>
<p>best solution and other close agreements, mean that at this stage in the development of structure
</p>
<p>prediction it may be necessary to obtain final confirmation by X-ray methods. In fact both the crystal
</p>
<p>structure and the best predicted structure in this case are very similar, as we have seen, with respect to
</p>
<p>unit cell and space group and this also applies to the structures in the details of packing and the rippled
</p>
<p>sheet H-bond formation, Fig. 8.35. Two strong hydrogen bonds, N1H1. . .O4 and N3H3. . .N5, and
</p>
<p>several weaker intermolecular interactions combine to produce this crinkled sheet structure.
</p>
<p>This crystal structure was independently predicted by a search for minima in the lattice energy, as
</p>
<p>calculated using an ab initio optimized molecular structure and a distributed multipole model for the
</p>
<p>electrostatic interactions. The global minimum in the search thus corresponded to the same Pbca
</p>
<p>space group, with rms errors in the cell lengths of about 3.7%. There is a larger energy gap separating
</p>
<p>the observed hydrogen-bonding motif structure from alternative structures, with different hydrogen
</p>
<p>bonds and connectivity.
</p>
<p>8.10.2 Developments in Computer Crystal Structure Prediction
</p>
<p>The state of the art in the computer prediction of organic crystal structures has been the subject of
</p>
<p>several authoritative reviews [63, 64]. Many programs have been written that attempt to predict
</p>
<p>structures, and a list of them is given in Table 8.26. The assumption is made that the true crystal
</p>
<p>structure will correspond to the global minimum lattice energy among the predicted structures. Since
</p>
<p>Fig. 8.33 Chemical formula of 5-azauracil, C3H3N3O2
</p>
<p>Fig. 8.34 View of the molecule of 5-azauracil molecule in the crystal structure, perpendicular to the molecular plane,
showing 50% thermal ellipsoids
</p>
<p>8.10 Computer Prediction of Crystal Structures 425</p>
<p/>
</div>
<div class="page"><p/>
<p>the structure is calculated for atoms at rest, that is, at 0 K, the lattice energy will differ from the lattice-
</p>
<p>free energy at ambient temperature mainly by an entropic component, which could affect the choice
</p>
<p>of structure when several results of similar lattice energy are calculated.
</p>
<p>8.11 Blind Structure Prediction of the Flexible Molecule
1-Benzyl-1H-Tetrazole
</p>
<p>1-Benzyl-1H-tetrazole is a molecule with 2� of conformational freedom and has an unusual tetrazole
functional group, providing a challenge to crystal structure prediction methods. There are too few
</p>
<p>structures with the C&ndash;H tetrazole fragment in the Cambridge Structural Database [65, 66] for any
</p>
<p>prior expectations of the intermolecular interactions of the tetrazole ring to be made.
</p>
<p>Table 8.24 Crystal data for 5-azauracil, C3H3N3O2
</p>
<p>Color/shape Colorless/needles
</p>
<p>Temperature (K) 289(2)
</p>
<p>Crystal system Orthorhombic
</p>
<p>Space group Pbca
</p>
<p>a (Å) 6.5135(3)
</p>
<p>b (Å) 13.5217(4)
</p>
<p>c (Å) 9.5824(4)
</p>
<p>Z 8
</p>
<p>Diffractometer/scan CAD4/o � 2y
Radiation Cu Ka
</p>
<p>Wavelength (Å) (graphite
</p>
<p>monochromator)
</p>
<p>1.54178
</p>
<p>Crystal dimensions (mm) 0.48, 0.33, 0.14
</p>
<p>Independent/observed reflections 779/763
</p>
<p>y (�) range for data collection 6.54&ndash;74.178
</p>
<p>Corrections applied Lorentz and polarization
</p>
<p>Absorption correction None
</p>
<p>Computer programs CAD4-Express 1988
</p>
<p>Structure solution SHELX-86
</p>
<p>Structure refinement SHELXL-93
</p>
<p>Refinement method Full matrix least squares on jFj2
Data/restraints/parameters 763/0/86
</p>
<p>Goodness-of-fit on F2 1.045
</p>
<p>Final R indices [I � 2s(I)] R1 &frac14; 0.0337; wR2 &frac14; 0.0909
</p>
<p>R indices (all data) R1 &frac14; 0.0409; wR2 &frac14; 0.1192
</p>
<p>Table 8.25 Crystal structure parameters for modeled structures and the X-ray structure
</p>
<p>Space group �U (kJ mol�1) a (Å) b (Å) c (Å) b (�) (Vc/Z) (Å
3) NH. . .O (Å) NH. . .N (Å)
</p>
<p>Pbca* 109.4 13.777 9.197 6.814 90 107.9 2.17 1.87
</p>
<p>P21/c 108.2 9.545 7.293 9.710 140.3 107.9 2.51 1.83
</p>
<p>Pbca 108.1 7.277 9.730 12.243 90 108.3 2.48 1.82
</p>
<p>P21 106.8 4.767 9.717 4.767 80.3 108.8 2.49 1.81
</p>
<p>Pbca 106.4 7.302 9.694 12.296 90 108.8 2.51 1.81
</p>
<p>Pca21 105.2 13.517 3.682 9.129 90 113.6 2.09 1.86
</p>
<p>X-ray &ndash; 6.5135 13.5217 9.5824 90 105.5 2.03 1.99
</p>
<p>426 8 Direct Methods and Refinement</p>
<p/>
</div>
<div class="page"><p/>
<p>Fig. 8.35 The crystal structure of 5-azauracil: four unit cells of the hydrogen bonded sheet, viewed along the a axis;
hydrogen bonds are denoted by dashed lines
</p>
<p>Table 8.26 Programs for organic crystal structure prediction
</p>
<p>Program
</p>
<p>Type of molecule used
</p>
<p>in development Search type
</p>
<p>Chin Crystal engineering: diketo-
</p>
<p>piperazines
</p>
<p>Monte Carlo simulated annealing with hydrogen-bonding bias
</p>
<p>CRYSTALG Rigid organics: amides,
</p>
<p>bases
</p>
<p>Self-consistent basin-to-deformed-basin mapping global optimization
</p>
<p>CRYSCA Pigments, organometallics Random search with steepest descent
</p>
<p>ICE9 Rigid hydrocarbons Systematic grid search to generate close-packed structures
</p>
<p>MDCP Small rigid Constant pressure molecular dynamics to find crude structures
</p>
<p>MOLPAK Energetic materials, rigid Systematic search for high density structures in common coordination
</p>
<p>types
</p>
<p>DMAREL Rigid polar and hydrogen
</p>
<p>bonded
</p>
<p>MPA,
</p>
<p>extended to
</p>
<p>Mpg
</p>
<p>Small rigid Lattman systematic, or random generation of expanded trial unit cell
</p>
<p>Perlstein Moderate sized, semi-
</p>
<p>flexible organics
</p>
<p>Aufbau search for low energy one-dimensional and two-dimensional
</p>
<p>aggregates, primarily for monolayer predictions
</p>
<p>PMC Hydrocarbons Symmetry adapted grid systematic
</p>
<p>Polymorph
</p>
<p>Predictor
</p>
<p>Flexible organics, including
</p>
<p>pharmaceuticals
</p>
<p>Monte Carlo simulated annealing with intermediate clustering
</p>
<p>PROMET Rigid hydrocarbons Selecting cohesive dimer, ribbons, and layer substructures of partial
</p>
<p>space group
</p>
<p>SySe and PP Pigments Grid-based systematic
</p>
<p>UPACK Sugars and alcohols Systematic grid or random search, with intermediate clustering
</p>
<p>8.11 Blind Structure Prediction of the Flexible Molecule 1-Benzyl-1H-Tetrazole 427</p>
<p/>
</div>
<div class="page"><p/>
<p>Crystal Structure of 1-Benzyl-1H-Tetrazole
The crystal structure determination and refinement of 1-benzyl-1H-tetrazole, C8H8N4, has been
</p>
<p>carried out using the programs SHELXS-86 and SHELXL-97 in WinGX [22], with Mo Ka X-ray
</p>
<p>data measured at 120(2) K [67]. The final data/restraints/parameter ratios were 977/1/113; R
</p>
<p>indices, I &gt; 2s(I): R1 &frac14; 0.0426, wR2 &frac14; 0.0753; R indices, all data: R1 &frac14; 0.0636, wR2 &frac14; 0.0827.
Figure 8.36 shows the molecular structure and atom numbering, and Table 8.27 list the crystal data.
</p>
<p>In the crystal structure, the benzyl and tetrazole rings are essentially planar. The individual rings
</p>
<p>are each coplanar with the inter-ring link atom C(6), and H(5) on C(5) of the five-membered tetrazole
</p>
<p>ring is also coplanar with its ring. The dihedral angle between the two rings is 68.52(5)�.
</p>
<p>Crystal Packing: Weak Hydrogen Bonding
The mode of packing of the molecules in this crystal structure is unusual and unexpected. Molecules
</p>
<p>such as 1-benzyl-1H-tetrazole, based on linked delocalized rings, might be expected to form crystal
</p>
<p>structures involving pi. . .pi ring stacking bonding. In this case, however, no such interactions occur.
</p>
<p>Instead the structure is held together through a large number of weak intermolecular hydrogen bonds:
</p>
<p>[68, 69] twelve of them are of the type CH. . .N; and three are of the weaker type CH. . .C, involving
</p>
<p>atoms in the phenyl ring, Table 8.28. The result is a structure composed of infinite S-shaped layers, as
</p>
<p>illustrated in Figs. 8.37 and 8.38. Another interesting and unusual by-product of the hydrogen bonding
</p>
<p>Fig. 8.36 1-Benzyl-1H-tetrazole showing thermal ellipsoids at 50 % probability. The torsion angles x1 &frac14; C(7)&ndash; C(6)&ndash;
N(1)&ndash;C(5) &frac14; �89.8(3)� and x2 &frac14; C(12)&ndash; C(7)&ndash; C(6)&ndash;N(1) &frac14; 98.5(3)� are indicated (drawn with ORTEP/RASTER)
</p>
<p>Table 8.27 Crystal data: experimental, computed structure #2, global minimum,GM
</p>
<p>Experimental #2 GM
</p>
<p>a (Å) 7.6843(5) 7.932 9.992
</p>
<p>b (Å) 5.5794(4) 5.550 8.175
</p>
<p>c (Å) 9.4459(7) 9.437 11.038
</p>
<p>b (�) 100.949(4) 101.323 117.504
</p>
<p>Vc (Å
3) 397.61(5) 407.331 799.726
</p>
<p>Z 2 2 4
</p>
<p>Space group P21 P21 P21/c
</p>
<p>x1 (
�) 89.91 87.38 86.07
</p>
<p>x2 (
�) �98.57 �97.88 �92.07
</p>
<p>Density (g cm�3) 1.338 1.306 1.330
</p>
<p>428 8 Direct Methods and Refinement</p>
<p/>
</div>
<div class="page"><p/>
<p>Table 8.28 Hydrogen bonds for 1-benzyl-1H-tetrazole
</p>
<p>D-H. . .A d(D&ndash;H) (Å) d(H. . .A) (Å) d(D. . .A) (Å) ffDHA (�)
C(8)&ndash;H(8). . .C(8)#1 0.95 2.93 3.743(3) 144.3
</p>
<p>C(9)&ndash;H(9). . .C(6)#1 0.95 2.93 3.668(4) 135.4
</p>
<p>C(10)&ndash;H(10). . .N(3)#2 0.95 2.91 3.723(3) 143.9
</p>
<p>C(10)&ndash;H(10). . .N(4)#2 0.95 2.78 3.674(4) 157.2
</p>
<p>C(12)&ndash;H(12). . .N(2)#3 0.95 2.78 3.617(4) 148.1
</p>
<p>C(6)&ndash;H(6B). . .N(3)#3 0.99 2.65 3.598(4) 159.7
</p>
<p>C(6)&ndash;H(6B). . .N(3)#3 0.99 2.93 3.766(4) 142.4
</p>
<p>C(5)&ndash;H(5). . .N(3)#3 0.90(3) 2.70(3) 3.490(4) 146(2)
</p>
<p>C(6)&ndash;H(6A). . .N(3)#4 0.99 2.75 3.245(3) 111.5
</p>
<p>C(5)&ndash;H(5). . .N(4)#5 0.90(3) 2.64(3) 3.292(4) 130(2)
</p>
<p>C(12)&ndash;H(12). . .N(3)#3 0.95 3.14 3.991(3) 150.5
</p>
<p>C(6)&ndash;H(6B). . .N(3)#3 0.99 2.93 3.766(4) 142.4
</p>
<p>C(9)&ndash;H(9). . .N(4)#6 0.95 3.00 3.664(3) 127.8
</p>
<p>C(10)&ndash;H(10). . .N(4)#6 0.95 3.04 3.679(4) 126.4
</p>
<p>C(10)&ndash;H(10). . .C(5)#6 0.95 3.15 3.882(4) 135.0
</p>
<p>Symmetry transformations used to generate equivalent atoms:
</p>
<p>#1: �x + 1, y� 1
2
, �z + 2; #2: �x + 2, y&thorn; 1
</p>
<p>2
, �z + 2; #3: x, y + 1, z
</p>
<p>#4: �x + 1, y&thorn; 1
2
, �z + 1;#5: �x + 2, y&thorn; 1
</p>
<p>2
, �z + 1; #6: x, y, z + 1
</p>
<p>Additionally there is an intermolecular CH. . .pi interaction: C(8)&ndash;H(8). . .benzyl centroid
#1 &frac14; 2.859 Å
</p>
<p>Fig. 8.37 1-Benzyl-1H-
tetrazole : view of the
</p>
<p>infinite S-shaped layers
</p>
<p>held together by weak
</p>
<p>CH...N and CH...C
</p>
<p>hydrogen bonds (drawn
</p>
<p>with Accelrys Discovery
</p>
<p>Studio 3 with coordinates
</p>
<p>generated in MERCURY)
</p>
<p>8.11 Blind Structure Prediction of the Flexible Molecule 1-Benzyl-1H-Tetrazole 429</p>
<p/>
</div>
<div class="page"><p/>
<p>is the formation of tetrazole clusters, Fig. 8.39, each having a central tetrazole ring coordinated by six
</p>
<p>other tetrazoles. There is in addition one intermolecular CH. . .pi interaction, but no pi. . .pi interaction.
</p>
<p>Structure Prediction
In view of this unusual mode of packing it was decided to initiate a blind crystal structure prediction
</p>
<p>of 1-benzyl-1H-tetrazole. Initially the modeling study investigated possible steric hindrance to
</p>
<p>rotation about the two torsion angles x1 and x2, Fig. 8.36, which indicated a high degree of flexibility
</p>
<p>in the molecule. Such a degree of flexibility, where a wide range of very different molecular shapes is
</p>
<p>possible, is required to be taken into account from the beginning of the structure search. This is a
</p>
<p>significant difference to the approach that can be used for more rigid molecules [72&ndash;79].
</p>
<p>Those working on this prediction study were provided only with a sketch of the 1-benzyl-1H-
</p>
<p>tetrazole molecule and were told only that the crystal structure had one molecule per asymmetric unit.
</p>
<p>The methodology used here for the prediction of crystal structures of flexible molecules had previously
</p>
<p>been extended to a pharmaceutical-like molecule with seven torsion angles linking four aromatic rings
</p>
<p>and a peptide group [73], in the Fifth Blind Test of Crystal Structure Prediction [74]. The search using
</p>
<p>the program CrystalPredictor [75] covered the 59 most commonly occurring space groups and
</p>
<p>generated about 170000 structures. The lattice energy, Elatt &frac14; Uinter + DEintra, where Uinter is the
intermolecular packing energy and DEintra is the energy penalty for changing the conformation of the
</p>
<p>molecule, was minimized by varying the cell parameters and torsion angles x1 and x2.
</p>
<p>Fig. 8.38 Partial view of
the layer structure 1-
</p>
<p>benzyl-1H-tetrazole
</p>
<p>showing alternative benzyl
</p>
<p>(B) and tetrazole (T) layers.
</p>
<p>Note the repeating layer
</p>
<p>sequence BTBT j TBTB
(drawn with MERCURY)
</p>
<p>430 8 Direct Methods and Refinement</p>
<p/>
</div>
<div class="page"><p/>
<p>At this stage, the lattice energy was calculated crudely by using a grid of ab initio calculations for
</p>
<p>DEintra and an atomic point charge model and empirical repulsion-dispersion model for Uinter. This
</p>
<p>reduced the search to 44000 unique structures. At this stage, for each of the lowest 10000 crystal
</p>
<p>structures, the energy of the molecule and its charge distribution were calculated at the PBE0/6-31G
</p>
<p>(d,p) level of theory using GAUSSIAN [76], to provide a better estimate of DEintra, and a more
</p>
<p>accurate representation of the charge density in terms of a distributed multipole model [77]. This was
</p>
<p>combined with an atom&ndash;atom exp&ndash;6 repulsion-dispersion potential, using parameters that had been
</p>
<p>fitted to azahydrocarbon [60] and polar crystal structures [78], which were assumed transferable to
</p>
<p>this tetrazole. The lattice energy of the crystal structure was minimized with DMACRYS [79], with
</p>
<p>the molecule held rigidly. One hundred most stable structures were then further refined by allowing
</p>
<p>the molecular conformation to adjust more accurately to the intermolecular forces using the program
</p>
<p>CrystalOptimizer [73, 80] to combine the GAUSSIAN calculations on the conformations of the
</p>
<p>isolated molecule with the crystal structure minimized by DMACRYS. Finally, the effect of the
</p>
<p>crystal environment on the conformational energies DEintra and charge density was estimated by
</p>
<p>calculating the conformational energy and charge distribution in a polarizable continuum, with the
</p>
<p>dielectric constant e equal to 3, a value typical of organic molecules [81], using the Polarizable
</p>
<p>Continuum Model [82] as implemented in GAUSSIAN [76] at the PBE0/6-31+G(d) level of theory.
</p>
<p>Results
The two most stable crystal structures on the crystal energy landscape, Fig. 8.40, are separated by less
</p>
<p>than 0.1 kJ mol�1 but are very different from one another in terms of structural features. One of these
</p>
<p>structures, labeled #2, is in the same space group, P21, as determined experimentally and has very
</p>
<p>similar unit-cell dimensions, Table 8.27. This computed structure gives an excellent overlay with the
</p>
<p>Fig. 8.39 1-Benzyl-1H-
tetrazole packing, showing
</p>
<p>an unusual tetrazole
</p>
<p>cluster. (drawn with
</p>
<p>Accelrys Discovery Studio
</p>
<p>3 from coordinates
</p>
<p>generated in MERCURY)
</p>
<p>8.11 Blind Structure Prediction of the Flexible Molecule 1-Benzyl-1H-Tetrazole 431</p>
<p/>
</div>
<div class="page"><p/>
<p>experimental structure using the COMPACK [83] facility as implemented in MERCURY [84], with a
</p>
<p>root mean square difference in the positions of the non-hydrogen atoms in a 15-molecule cluster of
</p>
<p>only 0.148 Å. This model #2 provided a starting model that was easily refined in SHELXL-97 against
</p>
<p>the experimental Fo(hkl) data, which is further proof that it is a genuine reproduction of the crystal
</p>
<p>structure.
</p>
<p>Lowest Energy Predicted Crystal Structure
It is of interest to note that the structure corresponding to the global minimum in lattice energy,
</p>
<p>Table 8.28, is in space group P21/c and presents a very different spatial arrangement as shown in
</p>
<p>Fig. 8.41. This structure also has dominant weak hydrogen bonds: eight of type CH. . .N; and one
</p>
<p>Fig. 8.40 Crystal energy
landscape of 1-benzyl-1H-
</p>
<p>tetrazole. Each point
</p>
<p>represents a crystal
</p>
<p>structure that is a lattice
</p>
<p>energy minimum, using
</p>
<p>PCM (e &frac14; 3) at PBE0/6-31
+G level of theory
</p>
<p>molecular calculations. The
</p>
<p>structure that matches the
</p>
<p>experimental crystal
</p>
<p>structure, optimized with
</p>
<p>the same lattice energy
</p>
<p>model, is labeled with its
</p>
<p>energy rank #2. The global
</p>
<p>minimum structure (#1) is
</p>
<p>labeled GM
</p>
<p>Fig. 8.41 Global
minimum predicted
</p>
<p>structure of 1-benzyl-1H-
</p>
<p>tetrazole: the packing
</p>
<p>reveals some of the weak
</p>
<p>CH. . .N and CH. . .C
interactions and the weak
</p>
<p>pi. . .pi interaction;
distances are in Å. There is
</p>
<p>a strong possibility that this
</p>
<p>structure may exist as a
</p>
<p>polymorphic form (drawn
</p>
<p>with Accelrys Discovery
</p>
<p>Studio 3 from coordinates
</p>
<p>generated in MERCURY)
</p>
<p>432 8 Direct Methods and Refinement</p>
<p/>
</div>
<div class="page"><p/>
<p>CH. . .C. There is also one weak pi. . .pi interaction (benzyl. . .benzyl) which is indicated in Fig. 8.41.
</p>
<p>Thermodynamically, this structure could be a polymorph of 1-benzyl-1H-tetrazole, but no polymorph
</p>
<p>screening had been undertaken when it was synthesized.
</p>
<p>Conclusions
1-Benzyl-1H-tetrazole shows an unusual crystal packing with segregated layers of phenyl and
</p>
<p>tetrazole interactions. This was predicted blindly from a structure that modeled accurately the
</p>
<p>electrostatic forces arising from molecular charge density, including the anisotropic forces from the
</p>
<p>lone pairs and pi-electrons, but otherwise had not been tailored to tetrazole. . .tetrazole interactions.
</p>
<p>The successful prediction of the experimental structure as one of the two distinct most stable
</p>
<p>structures shows that the unusual layers do present an optimal compromise between the many
</p>
<p>different weak hydrogen bonds and other intermolecular interactions.
</p>
<p>In structures that can exist in polymorphic modifications, the minimum energy conformation should
</p>
<p>correspond to the thermodynamically most stable form. In practice however, various factors, such as
</p>
<p>temperature, rate of crystallization, and nature of solvent, could affect the form that is actually observed.
</p>
<p>It seems probable that improvements in the precision of the prediction method will involve incorporat-
</p>
<p>ing more accurate forms of the intermolecular potential into the program, as well as allowing for the
</p>
<p>possibility of kinetic control of crystallization. While computer prediction clearly has a part to play in
</p>
<p>crystal structure determination, it will probably remain that X-ray diffraction will be needed for
</p>
<p>confirmation of the structure and particularly for obtaining accurate molecular geometry.
</p>
<p>8.12 Problems
</p>
<p>8.1. Choose three of the following reflections to fix an origin in space group P�1, giving reasons for
</p>
<p>your choice.
</p>
<p>Are there any triplets which meet the vector requirements of the &sum;2 formula?
</p>
<p>8.2. The geometric structure factor formulae for space group P21 are
</p>
<p>A &frac14; 2 cos 2p&eth;hx&thorn; lz&thorn; k=4&THORN; cos 2p&eth;ky� k=4&THORN;
B &frac14; 2 cos 2p&eth;hx&thorn; lz&thorn; k=4&THORN; sin 2p&eth;ky� k=4&THORN;
</p>
<p>Deduce the amplitude symmetry and the phase symmetry for this space group according to the
</p>
<p>two conditions k &frac14; 2n and k &frac14; 2n &thorn; 1.
8.3. In space group P21/c, two starting sets of reflections for the application of the &sum;2 formula are
</p>
<p>proposed:
</p>
<p>hkl jEj hkl jEj
705 2.2 6�1 �7 3.2
</p>
<p>42�6 2.7 203 2.3
</p>
<p>4�3 �2 1.1 8�1 �4 2.1
</p>
<p>Origin-fixing Symbols
</p>
<p>(a) 041, 117, �1 23 242, �162
</p>
<p>(b) 223, 012, 13�7 111, 162
</p>
<p>8.12 Problems 433</p>
<p/>
</div>
<div class="page"><p/>
<p>Using just this information, which starting set would be chosen in practice? Give reasons. What
</p>
<p>modification would have to be made to the starting set if the space group is C2/c?
</p>
<p>8.4. The following values of ln
P
</p>
<p>j f
2
j &eth;hkl&THORN;=
</p>
<p>h
jF2o&eth;hkl&THORN;j� and &eth;sin2y&THORN;=l2 were obtained from a set of
</p>
<p>three-dimensional data for a monoclinic crystal. Use the method of least squares (program
</p>
<p>LSLI) to obtain values for the scale K (of Fo) and temperature factor B by Wilson&rsquo;s method.
</p>
<p>ln
P
</p>
<p>j f
2
j &eth;hkl&THORN;=jF
</p>
<p>2
o&eth;hkl&THORN;
</p>
<p>h i
&eth;sin2y&THORN;=l2
</p>
<p>4.0 0.10
</p>
<p>5.6 0.20
</p>
<p>6.5 0.30
</p>
<p>7.9 0.40
</p>
<p>9.4 0.50
</p>
<p>What is the value of the root mean square atomic displacement corresponding to the derived
</p>
<p>value of B?
</p>
<p>8.5. An orthorhombic crystal contains four molecules of a chloro-compound in a unit cell of
</p>
<p>dimensions a &frac14; 7.210(4) Å, b &frac14; 10.43(1) Å, c &frac14; 15.22(2) Å. The coordinates of the Cl
atoms are
</p>
<p>1
4
; y; z; 3
</p>
<p>4
; �y; z; 1
</p>
<p>4
; &eth;1
</p>
<p>2
&thorn; y&THORN;; &eth;1
</p>
<p>2
&thorn; z&THORN;; 3
</p>
<p>4
; &eth;1
</p>
<p>2
� y&THORN;; &eth;1
</p>
<p>2
&thorn; z&THORN;
</p>
<p>with y &frac14; 0.140(2) and z &frac14; 0.000(2). Calculate the shortest Cl. . .Cl contact distance and its
</p>
<p>estimated standard deviation.
</p>
<p>8.6. The following data give phase indications for the reflection 771 (jEhj &frac14; 2.2, &rsquo;calc &frac14; �14
�) in a
</p>
<p>crystal of space group P212121. Determine &rsquo;h by both (8.25) and (8.28). For simplicity, let wh in
</p>
<p>(8.28) be taken as unity.
</p>
<p>k &rsquo;k (
�) h � k &rsquo;h�k (�) jEkjjEh�kj
</p>
<p>12,10 0 �561 �37 4.4
7�14 177 08�3 �180 5.1
12; 0�1 90 �572 �144 4.5
12,01 90 �570 �90 3.3
613 102 16�2 �64 2.7
�145 �79 83�4 92 3.7
</p>
<p>8.7. Figure 8.13a, b shows unit cells for a hypothetical search model (S) and a target structure (T)
</p>
<p>respectively. Assuming these structures to be correct, which of the intermolecular vectors
</p>
<p>indicated in Fig. 8.13b will not actually occur in the Patterson for the search molecule? Explain
</p>
<p>your answer.
</p>
<p>8.8. (a) Why would you not use a molecular graphics package alone to generate coordinates for all
</p>
<p>atoms in the coumarin derivative shown in Fig. 8.16a?
</p>
<p>(b) The z coordinates of all atoms in the coumarin model built with Chem-X in Table 8.16 are all
</p>
<p>0.0000. Why is this so?
</p>
<p>(c) The unit cell for the coumarin model in Table 8.15 has three sides each of 100 Å and all
</p>
<p>angles 90�. Why do you think this is so? (Hint: Apparently this is not a real unit cell.)
8.9. Determine graphically the X, Y, and Z Cartesian coordinates in Å of the six atoms of a (planar)
</p>
<p>benzene ring to be used in Patterson search. The bond lengths are all 1.40 Å and angles 120�.
[Hint: Construct a regular hexagon with one side parallel to the X (horizontal) axis and its center
</p>
<p>at the origin. The X, Y coordinates of atoms 1 and 2 are then obvious. The positions of the other
</p>
<p>434 8 Direct Methods and Refinement</p>
<p/>
</div>
<div class="page"><p/>
<p>atoms may be generated by applying symmetry. All Z coordinates are of course 0.0]. Check
</p>
<p>your results with the program INTXYZ, Sect. 13.6.6. If the unit cell shown has the dimensions
</p>
<p>a &frac14; 2.959 Å, b &frac14; 5.741 Å, c &frac14; 10.0 Å, a &frac14; b &frac14; g &frac14; 90�, determine the x, y, z fractional
coordinates of the six atoms.
</p>
<p>8.10. From the definition of jEj, show how a Patterson function with j(E2 � 1)j values as coefficients
leads to a sharpened Patterson function with the origin peak removed.
</p>
<p>8.11. When employing Patterson Search methods for structure analysis, under what circumstances
</p>
<p>would you expect the search molecule to be (a) very similar in size to, and (b) much smaller
</p>
<p>than, the target molecule? Discuss your answer in some detail.
</p>
<p>8.12. A third order Karle&ndash;Hauptman determinant for a centrosymmetric crystal may bewritten in
</p>
<p>the form
</p>
<p>E&eth;0&THORN; E&eth;h&THORN; E&eth;k&THORN;
E&eth;�h&THORN; E&eth;0&THORN; E&eth;�h&thorn; k&THORN;
E&eth;�k&THORN; E&eth;�k&thorn; h&THORN; E&eth;0&THORN;
</p>
<p>�
</p>
<p>�
</p>
<p>�
</p>
<p>�
</p>
<p>�
</p>
<p>�
</p>
<p>�
</p>
<p>�
</p>
<p>�
</p>
<p>�
</p>
<p>�
</p>
<p>�
</p>
<p>� 0
</p>
<p>If k &frac14; h, form and evaluate the determinant. If E(0) &frac14; 3 and jE(h)j &frac14; jE(2h)j &frac14; 2, determine
the sign of E(2h).
</p>
<p>8.13. Consider the three triplets 101, 21�2, �1 3�3; 101, 21�2, �1 �13; and 101, 21�2, �3 �11. Decide whether
</p>
<p>each triplet is a structure invariant, a structure seminvariant, or neither, and give reasons.
</p>
<p>Can any of these triplets be used to specify an origin in space group P1? Explain your
</p>
<p>conclusions.
</p>
<p>8.14. Consider a hypothetical crystal structure with a single atom at x &frac14; 0.3, y &frac14; 0.2, and z &frac14; 0.1.
</p>
<p>Assume that f(103) &frac14; 1.0, and calculate jF(103)j and &rsquo;(103) (a) in space group P21 (b) in space
group P212121 when the 21 axis parallel to y (i) passes through the point 0, 0, 0, and (ii) when it
</p>
<p>is placed in the standard orientation for this space group. Comment on the results.
</p>
<p>References
</p>
<p>1. Barrett AN, Palmer RA (1969) Acta Crystallogr B25:688
</p>
<p>2. Palmer HT (1973) Private communication
</p>
<p>3. Sayre D (1952) Acta Crystallogr 5:60
</p>
<p>4. Lonsdale K (1929) Proc R Soc A123:494
</p>
<p>5. Hauptman H, Karle J (1953) Solution of the phase problem, 1. The centrosymmetric crystal, American Crystallo-
</p>
<p>graphic Association monograph. Polycrystal Book Service, New York
</p>
<p>6. See Bibliography, Woolfson
</p>
<p>7. See Bibliography, Chapter 1
</p>
<p>8. Karle J, Karle IL (1966) Acta Crystallogr 21:849
</p>
<p>9. Barrett AN, Palmer RA, loc. cit.
</p>
<p>10. Karle J, Hauptman H (1950) Acta Crystallogr 3:181
</p>
<p>11. Germain G, Main P, Woolfson MM (1971) Acta Crystallogr A27:8
</p>
<p>12. Main P et al (1980) MULTAN 80: a system of computer programs for the automatic solution of crystal structures
</p>
<p>from x-ray diffraction data. University of York, England
</p>
<p>13. Rogers D (1980) In: Ladd MFC, Palmer RA (eds) Theory and practice of direct methods in crystallography.
</p>
<p>Plenum, New York
</p>
<p>14. Stroud RM (1973) Acta Crystallogr B29:60
</p>
<p>15. Sheldrick GM (1990) Acta Crystallogr A46:467
</p>
<p>16. Hauptman H (1972) Crystal structure determination: the role of the cosine seminvariants. Plenum, New York
</p>
<p>17. Hauptman H (1972) Acta Crystallogr B28:2337
</p>
<p>18. Sheldrick GM (1982) In: Sayre D (ed) Computational crystallography. Clarendon Press, Oxford
</p>
<p>19. M&euro;uller P (2006) Crystal structure refinement. Oxford University Press, New York
</p>
<p>References 435</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec31_13">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec31_13</a></div>
</div>
<div class="page"><p/>
<p>20. Sheldrick GM (2008) Acta Crystallogr A64:112
</p>
<p>21. The SHELX-97 manual. http://shelx.uni-ac.gwdg.de/SHELX/shelx.pdf
</p>
<p>22. Farrugia LJ (1999) J Appl Crystallogr 32:837
</p>
<p>23. http://shelx.uni-ac.gwdg.de/SHELX/applfrm.htm
</p>
<p>24. http://www.chem.gla.ac.uk/~louis/software/wingx/
</p>
<p>25. Giacovazzo C (1974) Acta Crystallogr A30:631
</p>
<p>26. Palmer RA et al (2010) Med Chem Commun 1:45
</p>
<p>27. Rossman MG, Blow DM (1961) Acta Crystallogr 14:641
</p>
<p>28. Braun PB, Hornstra J, Leenhouts JI (1969) Philips Res Rep 42:85
</p>
<p>29. http://www.ccp4.ac.uk/
</p>
<p>30. See Bibliography, Chapter 7
</p>
<p>31. http://download.cnet.com/ACD-ChemSketch-Freeware/3000-2054_4-10591465.html
</p>
<p>32. Lisgarten JN et al (2003) J Chem Crystallogr 33:149
</p>
<p>33. Gavuzzo E et al (1974) Acta Crystallogr B30:1351
</p>
<p>34. http://www.ccdc.cam.ac.uk/
</p>
<p>35. Tanczos AC et al (2004) Comput Biol Chem 28:375
</p>
<p>36. EPSRC Data Collection Service, University of Southampton, UK
</p>
<p>37. http://www.ccp14.ac.uk/tutorial/snb/
</p>
<p>38. Miller R et al (1994) J Appl Crystallogr 27:613
</p>
<p>39. DeTitta GT et al (1994) Acta Crystallogr A50:203
</p>
<p>40. idem. ibid. (1994) A50:210
</p>
<p>41. Hauptman H (2002) Z Kristallogr 217:406 and references therein
</p>
<p>42. Deacon AM et al (1998) Proc Natl Acad Sci U S A 95:9284
</p>
<p>43. Mendham AP et al (2011) J Chem Crystallogr 41:1323
</p>
<p>44. http://www2.lv.psu.edu/jxm57/irp/chisquar.html
</p>
<p>45. Fisher RA, Yates F (1974) Statistical tables for biological agricultural and medical research, 6th edn. Oliver &amp;
</p>
<p>Boyd, Edinburgh
</p>
<p>46. Duax WL et al (1975, 1984) Atlas of steroid structure, vols 1 and 2. IFI/Plenum, New York
</p>
<p>47. http://www.ccdc.cam.ac.uk/index.php
</p>
<p>48. http://home.rcsb.org
</p>
<p>49. http://cds.dl.ac.uk/cds/datasets/crys/icsd/llicsd.html
</p>
<p>50. http://www.nist.gov/index.html
</p>
<p>51. http://cds.dl.ac.uk/cds/datasets/crys/cweb/cweb.html
</p>
<p>52. http://www.bath.ac.uk/library/subjects/chem/core/daresbury.html
</p>
<p>53. El-Shora AL et al (1984) J Crystallogr Spectrosc Res 14:89
</p>
<p>54. Husain J et al (1982) Acta Crystallogr B38:2845
</p>
<p>55. Moss DS, Morffew A (1981) Comput Chem 6:1&ndash;3
</p>
<p>56. Bibliography, Chapter 6
</p>
<p>57. Palmer RA et al (2013) (to be published)
</p>
<p>58. Potter BS et al (1999) J Mol Struct 485&ndash;486:349
</p>
<p>59. Amos RD et al (1992) The Cambridge analytical derivatives package, 5th edn
</p>
<p>60. Williams DE, Cox SR (1984) Acta Crystallogr B40:404
</p>
<p>61. Cox SR et al (1981) Acta Crystallogr A37:293
</p>
<p>62. See Appendix D
</p>
<p>63. Price Sarah L (2003) In: Atwood J, Steed J (eds) Encyclopedia of supramolecular chemistry. Marcel Dekker,
</p>
<p>New York
</p>
<p>64. Beyer T et al (2001) CrystEngComm 3:178
</p>
<p>65. Allen FH (2002) Acta Crystallogr B58:380
</p>
<p>66. van de Streek J (2006) Acta Crystallogr B62:567
</p>
<p>67. Spencer J, Patel H, Deadman JJ, Palmer R, Male L, Coles S, Uzoh O, Price S (2012) CrystEngComm 14(20), 6441
</p>
<p>68. Desiraju GR (2002) Acc Chem Res 35:565
</p>
<p>69. Herrebout WA, Suhm MA (2011) Phys Chem Chem Phys 13:13858
</p>
<p>70. Potter BS et al (1999) J Mol Struct 486:349
</p>
<p>71. Nowell H et al (2006) Acta Crystallogr B62:642
</p>
<p>72. D&rsquo;Oria E et al (2010) Cryst Growth Des 10:1749
</p>
<p>73. Kazantsev AV et al (2011) Int J Pharm 418:168
</p>
<p>74. Bardwell DA, Adjiman CS, Arnautova YA, Bartashevich E, Boerrigter SX, Braun DE, Cruz-Cabeza AJ, Day GM,
</p>
<p>la Valle RG, Desiraju GR, van Eijck BP, Facelli JC, Ferraro MB, Grillo D, Habgood M, Hofmann DW, Hofmann F,
</p>
<p>Jose K, Karamertzanis VPG, Kazantsev AV, Kendrick J, Kuleshova LN, Leusen FJ, Maleev AV, Misquitta AJ,
</p>
<p>436 8 Direct Methods and Refinement</p>
<p/>
<div class="annotation"><a href="http://shelx.uni-ac.gwdg.de/SHELX/shelx.pdf">http://shelx.uni-ac.gwdg.de/SHELX/shelx.pdf</a></div>
<div class="annotation"><a href="http://shelx.uni-ac.gwdg.de/SHELX/applfrm.htm">http://shelx.uni-ac.gwdg.de/SHELX/applfrm.htm</a></div>
<div class="annotation"><a href="http://www.chem.gla.ac.uk/~louis/software/wingx/">http://www.chem.gla.ac.uk/~louis/software/wingx/</a></div>
<div class="annotation"><a href="http://www.ccp4.ac.uk/">http://www.ccp4.ac.uk/</a></div>
<div class="annotation"><a href="http://download.cnet.com/ACD-ChemSketch-Freeware/3000-2054_4-10591465.html">http://download.cnet.com/ACD-ChemSketch-Freeware/3000-2054_4-10591465.html</a></div>
<div class="annotation"><a href="http://www.ccdc.cam.ac.uk/">http://www.ccdc.cam.ac.uk/</a></div>
<div class="annotation"><a href="http://www.ccp14.ac.uk/tutorial/snb/">http://www.ccp14.ac.uk/tutorial/snb/</a></div>
<div class="annotation"><a href="http://www2.lv.psu.edu/jxm57/irp/chisquar.html">http://www2.lv.psu.edu/jxm57/irp/chisquar.html</a></div>
<div class="annotation"><a href="http://www.ccdc.cam.ac.uk/index.php">http://www.ccdc.cam.ac.uk/index.php</a></div>
<div class="annotation"><a href="http://home.rcsb.org">http://home.rcsb.org</a></div>
<div class="annotation"><a href="http://cds.dl.ac.uk/cds/datasets/crys/icsd/llicsd.html">http://cds.dl.ac.uk/cds/datasets/crys/icsd/llicsd.html</a></div>
<div class="annotation"><a href="http://www.nist.gov/index.html">http://www.nist.gov/index.html</a></div>
<div class="annotation"><a href="http://cds.dl.ac.uk/cds/datasets/crys/cweb/cweb.html">http://cds.dl.ac.uk/cds/datasets/crys/cweb/cweb.html</a></div>
<div class="annotation"><a href="http://www.bath.ac.uk/library/subjects/chem/core/daresbury.html">http://www.bath.ac.uk/library/subjects/chem/core/daresbury.html</a></div>
</div>
<div class="page"><p/>
<p>Mohamed S, Needs RJ, Neumann MA, Nikylov D, Orendt AM, Pal R, Pantelides CC, Pickard CJ, Price LS, Price
</p>
<p>Sarah L, Scheraga HA, van de Streek J, Thakur TS, Tiwari S, Venuti E, Zhitkov IK (2011) Acta Crystallogr B
</p>
<p>67:535
</p>
<p>75. Karamertzanis PG, Pantelides CC (2007) Mol Phys 105:273
</p>
<p>76. Frisch MJ, Trucks GW, Schlegel HB, Scuseria GE, Robb MA, Cheeseman JR, Montgomery J, Vreven T, Kudin
</p>
<p>KN, Burant JC, Millam JM, Iyengar SS, Tomasi J, Barone V, Mennucci B, Cossi M, Scalmani G, Rega N, Petersson
</p>
<p>GA, Nakatsuji H, Hada M, Ehara M, Toyota K, Fukuda R, Hasegawa J, Ishida M, Nakajima T, Honda Y, Kitao O,
</p>
<p>Nakai H, Klene M, Li X, Knox JE, Hratchian HP, Cross JB, Bakken V, Adamo C, Jaramillo J, Gomperts R,
</p>
<p>Stratmann RE, Yazyev O, Austin AJ, Cammi R, Pomelli C, Ochterski J, Ayala Morokuma SPY, Voth K, Salvador
</p>
<p>GA, Dannenberg P, Zakrzewski JJ, Dapprich VG, Daniels AD, Strain MC, Farkas O, Malick DK, Rabuck AD,
</p>
<p>Raghavachari K, Foresman JB, Ortiz JV, Cui Q, Baboul AG, Clifford S, Cioslowski J, Stefanov BB, Liu G,
</p>
<p>Liashenko A, Piskorz P, Komaromi I, Martin RL, Fox DJ, Keith T, Al Laham MA, Peng CY, Nanayakkara A,
</p>
<p>Challacombe M, Gill PMW, Johnson B, Chen W, Wong MW, Gonzalez C, Pople JA (2004) Gaussian Inc.,
</p>
<p>Wallingford
</p>
<p>77. Stone AJ (2005) J Chem Theory Comput 1:1128
</p>
<p>78. Coombes DS et al (1996) J Phys Chem 100:7352
</p>
<p>79. Price SL et al (2010) Phys Chem Chem Phys 12:8478
</p>
<p>80. Kazantsev AV et al (2010) In: Adjiman CS, Galindo A (eds) Molecular system engineering, vol 6. Wiley-VCH
</p>
<p>Verlag GmbH, Weinheim, p 1
</p>
<p>81. Cooper TG et al (2008) J Chem Theory Comput 4:1795
</p>
<p>82. Cossi M et al (2002) J Chem Phys 117:43
</p>
<p>83. Chisholm JA, Motherwell S (2005) J Appl Crystallogr 38:228
</p>
<p>84. Macrae CF et al (2006) J Appl Crystallogr 39:453
</p>
<p>85. http://www.povray.org
</p>
<p>86. http://www.chem.gla.ac.uk/~louis/ortep3/
</p>
<p>87. Ladd MFC, Palmer RA (1993) Structure determination by X-ray crystallography, 3rd edn. Springer, New York
</p>
<p>Bibliography: Crystallographic Computing
</p>
<p>Ahmed FR et al (eds) (1970) Crystallographic computing. Munksgaard, Copenhagen
</p>
<p>Diamond R, Ramaseshan S, Venkatesan K (eds) (1980) Computing in crystallography. Indian Academy of Sciences,
</p>
<p>Bangalore
</p>
<p>Flack HD, Párkányi L, Simon K (eds) (1993) Crystallographic computing, vol 6, A window on modern crystallography.
</p>
<p>Oxford University Press, Oxford
</p>
<p>http://www.iucr.org/resources/commissions/crystallographic- computing/schools
</p>
<p>http://crystallography.org.uk/
</p>
<p>Isaacs MR (1988) Crystallographic computing, vol 4, Techniques and new technologies. Oxford University Press,
</p>
<p>Oxford
</p>
<p>Mitchell DJ, Lippert EL (1965) Acta Crystallogr 18:559
</p>
<p>Mitchell EJ (1965) PhD thesis, Vanderbilt University
</p>
<p>Moras D, Podjarny AD, Thierry JC (eds) (1991) Crystallographic computing, No. 5: from chemistry to biology,
</p>
<p>International union of crystallography symposia. Oxford University Press/International Union of Crystallography,
</p>
<p>Oxford
</p>
<p>M&euro;uller P (ed) (2006) Crystal structure refinement: a crystallographer&rsquo;s guide to SHELXL. International Union of
Crystallography, Oxford
</p>
<p>Pepinsky R, Robertson JM, Speakman JC (eds) (1961) Computing methods and the phase problem in X-ray crystal
</p>
<p>analysis. Pergamon, Oxford
</p>
<p>Rollett JS (ed) (1965) Computing methods in crystallography. Pergamon, Oxford
</p>
<p>Sheldrick GM (2008) A short history of SHELX. Acta Crystallogr A64:112
</p>
<p>Sheldrick GM (1985) Crystallographic computing, vol 3, Data collection, structure determination, proteins and
</p>
<p>databases. Oxford University Press, Oxford
</p>
<p>Watkin D (1994) Acta Crystallogr A39:158
</p>
<p>References 437</p>
<p/>
<div class="annotation"><a href="http://www.povray.org">http://www.povray.org</a></div>
<div class="annotation"><a href="http://www.chem.gla.ac.uk/~louis/ortep3/">http://www.chem.gla.ac.uk/~louis/ortep3/</a></div>
<div class="annotation"><a href="http://www.iucr.org/resources/commissions/crystallographic-computing/schools">http://www.iucr.org/resources/commissions/crystallographic-computing/schools</a></div>
<div class="annotation"><a href="http://crystallography.org.uk/">http://crystallography.org.uk/</a></div>
</div>
<div class="page"><p/>
<p>Direct Methods
</p>
<p>ACORN ab initio phasing. http://www.ccp4.ac.uk/dist/ccp4i/help/modules/exptphase.html#acorn; http://www.ccp4.ac.uk/
</p>
<p>newsletters/newsletter40/07_acorn.pdf
</p>
<p>Giacovazzo C (1980) Direct methods in crystallography. Academic, New York
</p>
<p>Ladd MFC, Palmer RA (eds) (1980) Theory and practice of direct methods in crystallography. Plenum, New York
</p>
<p>Rodriguez DD et al (2009) Nat Methods 6:651
</p>
<p>Woolfson MM, Hai-Fu H (2005) Physical and non-physical methods of solving crystal structures. Cambridge Univer-
</p>
<p>sity Press, Cambridge
</p>
<p>Woolfson MM (1997) An introduction to X-ray crystallography, 2nd edn. Cambridge University Press, Cambridge
</p>
<p>Woolfson MM (1961) Direct methods in crystallography. Clarendon Press, Oxford
</p>
<p>Chemical Data
</p>
<p>Allen FH et al (1987) J Chem Soc Perkin Trans II:S1&ndash;S19
</p>
<p>Chemical Database Service (2011). http://cds.dl.ac.uk/
</p>
<p>Goodman Group, University of Cambridge (2011). http://www-jmg.ch.cam.ac.uk/
</p>
<p>Kitaigorodskii AI (1973) Molecular crystals and molecules. Academic, New York
</p>
<p>Megaw HD (1973) Crystal structures. Saunders, Philadelphia
</p>
<p>NIST Chemical Webbook (2011). http://webbook.nist.gov/
</p>
<p>Sutton LE (ed) (1958, suppl 1965) Tables of interatomic distances and configurations in molecules and ions. The
</p>
<p>Chemical Society, London
</p>
<p>438 8 Direct Methods and Refinement</p>
<p/>
<div class="annotation"><a href="http://www.ccp4.ac.uk/dist/ccp4i/help/modules/exptphase.html#acorn">http://www.ccp4.ac.uk/dist/ccp4i/help/modules/exptphase.html#acorn</a></div>
<div class="annotation"><a href="http://www.ccp4.ac.uk/newsletters/newsletter40/07_acorn.pdf">http://www.ccp4.ac.uk/newsletters/newsletter40/07_acorn.pdf</a></div>
<div class="annotation"><a href="http://www.ccp4.ac.uk/newsletters/newsletter40/07_acorn.pdf">http://www.ccp4.ac.uk/newsletters/newsletter40/07_acorn.pdf</a></div>
<div class="annotation"><a href="http://cds.dl.ac.uk/">http://cds.dl.ac.uk/</a></div>
<div class="annotation"><a href="http://www-jmg.ch.cam.ac.uk/">http://www-jmg.ch.cam.ac.uk/</a></div>
<div class="annotation"><a href="http://webbook.nist.gov/">http://webbook.nist.gov/</a></div>
</div>
<div class="page"><p/>
<p>Examples of Crystal Structure
Determination 9
</p>
<p>9.1 Introduction
</p>
<p>In this chapterwe draw together, bymeans of actual examples, someof thematerial presented earlier in the
</p>
<p>book. It may be desirable for the reader to refer back to the previous chapters for descriptions of the
</p>
<p>techniques used, since we shall present here mainly the results obtained at each stage.
</p>
<p>The first three examples can be solved by either the heavy-atommethod or directmethods.Nowadays,
</p>
<p>it is quite commonplace to attempt the solving by direct methods of those structures which, at one time,
</p>
<p>would have been treated by the heavy-atom method. Where a powerful and sophisticated computer
</p>
<p>package is available, direct methods frequently provide the most expeditious route to the solution of a
</p>
<p>structure.However, in order thatwemay illustrate themethods described,we shall use both techniques in
</p>
<p>this chapter.
</p>
<p>9.2 Crystal Structure of 2-Bromobenzo[b] Indeno[1,2-e] Pyran [1]
</p>
<p>2-Bromobenzo[b]indeno[1,2-e]pyran (BBIP) is an organic compound which was prepared by heating
</p>
<p>a solution in ethanol of equimolar amounts of 3-bromo-6-hydroxybenzaldehyde (I) and 2-oxoindane
</p>
<p>(II) under reflux in the presence of piperidine acetate. The two molecules condense with the
</p>
<p>elimination of two molecules of water. Upon recrystallization of the product from toluene, it has an
</p>
<p>m.p. of 176.5&ndash;177.0�C. Its molecular formula is C16H9BrO, and its classical structural formula is
shown by III.
</p>
<p>9.2.1 Preliminary Physical and X-Ray Measurements
</p>
<p>The compound was recrystallized from toluene by slow, isothermal evaporation of the solvent at
</p>
<p>room temperature. The crystals were red, with an acicular (needle-shaped) habit, with the forms
</p>
<p>M. Ladd and R. Palmer, Structure Determination by X-ray Crystallography:
</p>
<p>Analysis by X-rays and Neutrons, DOI 10.1007/978-1-4614-3954-7_9,
# Springer Science+Business Media New York 2013
</p>
<p>439</p>
<p/>
</div>
<div class="page"><p/>
<p>(subsequently named) {100}, {110}, {001}, and {011} predominant, Fig. 9.1. The red color is
</p>
<p>characteristic of the chromophoric nature of a conjugated double-bond system.
</p>
<p>The density of the crystals was measured by suspending them in aqueous sodium bromide solution
</p>
<p>in a stoppered measuring cylinder in a thermostat bath at 25�C. Water or concentrated sodium
bromide solution, as necessary, was added to the suspension until the crystals neither settled to the
</p>
<p>bottom of the cylinder nor floated to the surface of the solution. Then, the crystals and solution were
</p>
<p>of the same density, and the density of the solution was measured with a pyknometer. A convenient
</p>
<p>variant here is to measure the refractive index of the final solution with an Abbe refractometer, having
</p>
<p>first prepared a graph of refractive index against density from data in the literature [2].
</p>
<p>Under a polarizing microscope, the crystals showed straight extinction on (100) and (001), and
</p>
<p>oblique extinction (about 3� to a crystal edge) on a section cut normal to the needle axis (y). These
observations suggested that the crystals were probably monoclinic. Although the diffractometer is now
</p>
<p>the main instrument of data collection, photographic work is still carried on in some parts of the world,
</p>
<p>and X-ray photographs can be useful in deciding upon the suitability of the crystal specimen and for
</p>
<p>Fig. 9.1 Crystal habit of BBIP with the crystallographic axes drawn in; the forms shown are {100}, {110}, {001},
and {011}
</p>
<p>440 9 Examples of Crystal Structure Determination</p>
<p/>
</div>
<div class="page"><p/>
<p>information on its symmetry. In this example, we consider the use of photographic methods as a
</p>
<p>preliminary to the diffractometer measurements.
</p>
<p>The crystals chosen for X-ray studies had the approximate dimensions 0.2, 0.4, and 0.3 mm
</p>
<p>parallel to a, b, and c, respectively. A crystal was mounted on the end of an annealed quartz fiber with
</p>
<p>&ldquo;Araldite&rdquo; (or &ldquo;Eastman 910&rdquo;) adhesive and the fiber attached to an X-ray goniometer head, or arcs,
</p>
<p>Fig. 9.2, with dental wax. The arcs were affixed to a single-crystal oscillation camera, and the crystal
</p>
<p>was set with the needle axis accurately parallel to the axis of oscillation, first by eye and finally by X-
</p>
<p>ray methods. Copper Ka radiation (l &frac14; 1.5418 Å) was used throughout the work.
A symmetrical oscillation photograph taken about the b direction is shown in Fig. 9.3. The
</p>
<p>horizontal mirror symmetry line indicates that the Laue group of the crystal has an m plane normal
</p>
<p>to the needle axis. Further X-ray photographs, for example, the Laue photograph in Fig. 9.4, showed
</p>
<p>that the only axial symmetry was 2 parallel to b, thus confirming the monoclinic system for BBIP.
</p>
<p>Notwithstanding Weissenberg photography is rarely employed nowadays, these photographs were
</p>
<p>used in this work, and a sample is shown in Figs. 9.5, 9.6, and 9.7. The straightforward indexing of the
</p>
<p>reflections can be understood with reference to Figs. 9.8 and 9.9. There are no systematic absences for
</p>
<p>the hkl reflections, so that the unit cell is primitive, but systematic absences do arise for h0l with l odd
</p>
<p>and for 0k0 with k odd. These observations confirm the monoclinic symmetry, and the systematic
</p>
<p>absences lead unambiguously to space group P21/c.
</p>
<p>Measurements on the X-ray photographs gave the approximate unit-cell dimensions as:
</p>
<p>a &frac14; 7:51 A; b &frac14; 5:96 A; c &frac14; 26:2 A; and b &frac14; 92:5�
</p>
<p>The Bragg y-angles of 20 high-order reflections of known indices, distributed evenly in reciprocal
</p>
<p>space, were measured to the nearest 0.01� on a four-circle CAD-4 diffractometer. From these data, the
unit-cell dimensions were calculated accurately by the method of least squares; see Sect. 8.4ff. The
</p>
<p>complete crystal data are listed in Table 9.1. The calculated density Dc, for Z &frac14; 4, is in good
agreement with the measured value Dm, which indicates a high degree of self-consistency in the
</p>
<p>Fig. 9.2 Standard goniometer head; A and B are two arcs for angular adjustments; C and D are two sledges for
horizontal adjustments (courtesy of Stoe et Cie)
</p>
<p>9.2 Crystal Structure of 2-Bromobenzo[b] Indeno[1,2-e] Pyran 441</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_8#Sec29_8">http://dx.doi.org/10.1007/978-1-4614-3954-7_8#Sec29_8</a></div>
</div>
<div class="page"><p/>
<p>Fig. 9.3 Symmetrical oscillation photograph taken with the X-ray beam normal to b. The horizontalm line indicates an
m plane in the Laue group of the crystal, normal to the axis of oscillation
</p>
<p>Fig. 9.4 Laue photograph taken with the X-ray beam along b, showing clearly the twofold symmetry axis along this
direction
</p>
<p>442 9 Examples of Crystal Structure Determination</p>
<p/>
</div>
<div class="page"><p/>
<p>Fig. 9.5 Weissenberg
photograph of the h0l layer.
</p>
<p>The more intense
</p>
<p>reflections show spots
</p>
<p>arising from both Cu Ka
</p>
<p>(l &frac14; 1.5418 Å) and Cu Kb
(l &frac14; 1.392 Å) radiations.
In some areas, spots from
</p>
<p>W La radiation
</p>
<p>(l &frac14; 1.48 Å) arise due to
sputtering of the copper
</p>
<p>target in the X-ray tube
</p>
<p>with tungsten from the
</p>
<p>filament; see Sect. 3.1.1. In
</p>
<p>each case, the spots for a
</p>
<p>given h0l reflection lie
</p>
<p>along the directions of axial
</p>
<p>rows at decreasing values
</p>
<p>of sin y, in proportion to
</p>
<p>the change in l. The
</p>
<p>continuous streaks (Laue
</p>
<p>streaks) arise from the
</p>
<p>&ldquo;white&rdquo; radiation; filtering
</p>
<p>(Chap. 3, Sect. 3.1.4) is
</p>
<p>never perfect
</p>
<p>Fig. 9.6 Weissenberg
photograph of the h1l layer.
</p>
<p>The 01l reciprocal lattice
</p>
<p>row, indicated by an
</p>
<p>arrowhead, illustrates
</p>
<p>clearly the effect of slight
</p>
<p>mis-setting of the crystal
</p>
<p>9.2 Crystal Structure of 2-Bromobenzo[b] Indeno[1,2-e] Pyran 443</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec2_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec2_3</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec5_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec5_3</a></div>
</div>
<div class="page"><p/>
<p>parameters involved. The estimated standard deviations of the measured and calculated density
</p>
<p>values are important, because a significant difference (&gt;3s) may indicate the presence of unsuspected
</p>
<p>solvent of crystallization.
</p>
<p>9.2.2 Intensity Measurement and Correction
</p>
<p>Intensity data were collected on the diffractometer up to sin y &frac14; 65�. The number N of data to be
expected may be calculated from the formula N &frac14; (4p/3)[2(sin ymax)/l]3Vc/mG, where mG is the
number of general positions in the Laue group of the crystal, and the other symbols have their usual
</p>
<p>meanings. Since mG &frac14; 4 in the present example, N &frac14; 1990. From this number we subtract the
number of systematic absences, 165, to give 1825. In practice, 1724 data were collected, indicating
</p>
<p>101 accidental absences. Thus, a total of 1623 reflections were used for the structure analysis, giving
</p>
<p>ten reflections per parameter, assuming anisotropic temperature factors for the non-hydrogen atoms
</p>
<p>and a single scale factor in the least-squares refinement.
</p>
<p>Fig. 9.7 Upper part of a
Weissenberg photograph of
</p>
<p>the 0kl layer, Cu Kb spots,
</p>
<p>can be seen for the more
</p>
<p>intense reflections. The 00l
</p>
<p>(z*) reciprocal lattice row,
</p>
<p>indicated by an arrowhead,
</p>
<p>is common to this
</p>
<p>photograph and that of the
</p>
<p>h0l layer
</p>
<p>Fig. 9.8 Sample of
indexed reflections on an
</p>
<p>h0l Weissenberg
</p>
<p>photograph diagram
</p>
<p>444 9 Examples of Crystal Structure Determination</p>
<p/>
</div>
<div class="page"><p/>
<p>The data were corrected for Lorentz and polarization effects, but not for absorption. Approximate
</p>
<p>scale K and overall isotropic temperature B factors were obtained by Wilson&rsquo;s method, Sect. 4.2.1.
</p>
<p>The parameters were fitted by least squares, and the line obtained had the equation
</p>
<p>ln
</p>
<p>P
j f
</p>
<p>2
j
</p>
<p>F2o
</p>
<p>( )
&frac14; �1:759&thorn; 3:480 sin2y (9.1)
</p>
<p>Fig. 9.9 Weissenberg chart: camera diameter 57.30 mm, 2� rotation per mm travel (reproduced with the permission
of the Institute of Physics and the Physical Society, London)
</p>
<p>9.2 Crystal Structure of 2-Bromobenzo[b] Indeno[1,2-e] Pyran 445</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_4#Sec11_4">http://dx.doi.org/10.1007/978-1-4614-3954-7_4#Sec11_4</a></div>
</div>
<div class="page"><p/>
<p>From the slope (2B/l2) and intercept (2 ln K), B &frac14; 4.1 Å2 and K &frac14; 0.41, where K is the scale
factor for F2o. The graphical Wilson plot is shown in Fig. 9.10.
</p>
<p>9.2.3 Structure Analysis in the xz Projection
</p>
<p>This projection of the unit cell has the largest area and thus would be expected to show good resolution
</p>
<p>of the molecule. It is uncommon for a normal three-dimensional study to be preceded by an analysis in
</p>
<p>projection. However, from the standpoint of introductory study, carefully chosen two-dimensional
</p>
<p>examples have much to offer.
</p>
<p>Using Fig. 2.32, we can associate the coordinates of the general equivalent positions
</p>
<p>� x; y; z; x; 1=2� y; 1=2&thorn; zf g
</p>
<p>with the four bromine atoms in the unit cell. In the xz projection, these coordinates give rise to two repeats
</p>
<p>within the length c, so we may consider this projection in terms of plane group p2 and compute the
</p>
<p>Patterson function from 0 to a/2 and 0 to c/2, which is equivalent to one-half of the unit cell in p2. This
</p>
<p>portion of the projection P(uw) would be expected to show one Br&ndash;Br vector at (2x, 2z), as in Fig. 9.11.
</p>
<p>The two rows of peaks indicate that, in this projection, the molecules lie closely parallel to the z
</p>
<p>axis; this conclusion is supported by the large magnitude of jF(200)j, equal to 336. This value may be
compared with F(000) in Table 9.1, and, more significantly, with
</p>
<p>P
j fj;y200 , which is 474. The peak
</p>
<p>arising from the Br&ndash;Br vector is marked A, and by direct measurement we obtain the fractional
</p>
<p>coordinates x &frac14; 0.25, z &frac14; 0.015 for the Br atom in the asymmetric unit.
An electron density map, in this projection, was calculated using the signs given by FBr with the
</p>
<p>experimental values of Fo(h0l). If FBr was less than one third of the corresponding value of Fo for any
</p>
<p>reflection, the sign was assumed to be uncertain and the reflection omitted from the electron density
</p>
<p>calculation at this stage of the analysis. Figure 9.12 shows the electron density map with the molecule,
</p>
<p>fitted with the aid of a model, marked in. The resolution is moderately good, and we can see that we are
</p>
<p>working along the right lines. From the shapes of the rings, it is apparent that the molecule is inclined to
</p>
<p>the plane of this projection, and therewill be a limit to the improvement of the resolution attainable in this
</p>
<p>projection. Consequently, we begin three-dimensional studies.
</p>
<p>Table 9.1 Crystal data for BBIP at 20�Ca
</p>
<p>Molecular formula C16H9BrO
</p>
<p>Mr 297.16
</p>
<p>Space group P21/c
</p>
<p>a (Å) 7.508(4)
</p>
<p>b (Å) 5.959(5)
</p>
<p>c (Å) 26.172(6)
</p>
<p>b (�) 92.55(2)
</p>
<p>Vc (Å
3) 1,169(2)
</p>
<p>Radiation (Cu Ka) 1.5418 Å
</p>
<p>Dm (g cm
�3) 1.68(1)
</p>
<p>Dc (g cm
�3) 1.688(3)
</p>
<p>Z 4
</p>
<p>F(000) 592
</p>
<p>aThe numbers in parentheses are estimated
</p>
<p>standard deviations, to be applied to the least
</p>
<p>significant figure
</p>
<p>446 9 Examples of Crystal Structure Determination</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_2#Fig32_2">http://dx.doi.org/10.1007/978-1-4614-3954-7_2#Fig32_2</a></div>
</div>
<div class="page"><p/>
<p>9.2.4 Three-Dimensional Structure Determination
</p>
<p>In order to obtain values for all spatial coordinates, we proceeded first to a three-dimensional
</p>
<p>Patterson map P(uvw), calculated section by section normal to the b axis.
</p>
<p>The coordinates of the general positions show that the Br&ndash;Br vectors in the asymmetric unit will be
</p>
<p>found at 2x, 2y, 2z (single-weight peak), 0; 1
2
� 2y; 1
</p>
<p>2
(double-weight peak), and 2x; 1
</p>
<p>2
; 1
2
� 2z (double-
</p>
<p>weight peak). Hence, wemust study the Pattersonmap carefully, particularly the Harker line &frac12;0; v; 1
2
� and
</p>
<p>section &eth;u; 1
2
;w&THORN;&mdash;why? Figures 9.13 and 9.14 show these two regions of Patterson space, and Fig. 9.15
</p>
<p>illustrates a general section calculated close to the single-weight peakB. From the peaks B,C, andD, the
</p>
<p>coordinates for the bromine atom in the asymmetric unit were found to be 0.248, 0.188, 0.016.
</p>
<p>Repeating the phasing procedure, but now for hkl reflections, and calculating a three-dimensional
</p>
<p>electron density map produced a good resolution of the complete structure, with the exception of the
</p>
<p>Fig. 9.10 Wilson plot for
BBIP; the slope (2B/l2) is
</p>
<p>3.48 and the intercept
</p>
<p>&eth;2 ln K&THORN; is �1.76
</p>
<p>9.2 Crystal Structure of 2-Bromobenzo[b] Indeno[1,2-e] Pyran 447</p>
<p/>
</div>
<div class="page"><p/>
<p>hydrogen atoms. Figure 9.16 illustrates a composite electron density map, which consists of super-
</p>
<p>imposed sections calculated at intervals along a.
</p>
<p>The scattering of X-rays by hydrogen atoms is small in magnitude, and these atoms are not
</p>
<p>normally resolved by the direct summation of the electron density. If the data are of good quality and
</p>
<p>all other atoms in the structure have been found, a difference-Fourier synthesis, Sect. 7.5.4, will
</p>
<p>generally result in the hydrogen atoms being located, provided the other atoms are not themselves
</p>
<p>too large in scattering power. It is possible also to calculate the positions of hydrogen atoms from the
</p>
<p>geometry of the structure, if the positions of sufficient surrounding groups are known. Nowadays,
</p>
<p>most program systems for X-ray structure analysis include routines for calculating the coordinates of
</p>
<p>hydrogen atoms, according to their nature, for example, R1R2R3CH and R1R2CH2; the hydrogen
</p>
<p>atoms in a CH3 group can be calculated if a position with respect to rotation about the R&ndash;CH3 bond
</p>
<p>can be postulated.
</p>
<p>Finally, we arrive at the complete structure for BBIP, as shown in Fig. 9.17 with a convenient
</p>
<p>numbering scheme.
</p>
<p>Fig. 9.11 Asymmetric unit of P(uw). Since we are concerned here mainly with the Br&ndash;Br vector (A), the slight
distortion arising from drawing b as 90� is inconsequential here
</p>
<p>448 9 Examples of Crystal Structure Determination</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_7#Sec26_7">http://dx.doi.org/10.1007/978-1-4614-3954-7_7#Sec26_7</a></div>
</div>
<div class="page"><p/>
<p>9.2.5 Refinement
</p>
<p>During the final stages of refinement of the structure, the hydrogen atoms were included in the
</p>
<p>evaluation of the structure factors jFcj, but no attempt was made to refine the parameters of the
hydrogen atoms because the main interest in the problem lay in determining the molecular
</p>
<p>conformation. We note also that we have a favorable ratio of 10:1 for data to variables. Had we
</p>
<p>included the hydrogen atoms in the refinement, with isotropic temperature factors, the ratio would
</p>
<p>have been decreased to 9.4; see Sect. 8.4.3. The final adjustments of the structural parameters of
</p>
<p>the Br, O, and C atoms (x, y, z, and anisotropic temperature factors) and the scale factor were
</p>
<p>carried out by the method of least squares. The refinement converged with an R factor of 0.070, and
</p>
<p>a final difference-Fourier synthesis showed no fluctuations in density greater than about twice
</p>
<p>s(ro), and then only around the position of the bromine atom. The analysis was considered to be
</p>
<p>satisfactory, and the refinement was terminated at this stage.
</p>
<p>Fig. 9.12 Asymmetric unit of r(xz) phased on the bromine atoms; the probable atomic positions are marked in
</p>
<p>9.2 Crystal Structure of 2-Bromobenzo[b] Indeno[1,2-e] Pyran 449</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_8#Sec32_8">http://dx.doi.org/10.1007/978-1-4614-3954-7_8#Sec32_8</a></div>
</div>
<div class="page"><p/>
<p>Fig. 9.13 Patterson function along the Harker line &frac12;0; v; 1
2
�, showing a double-weight Br&ndash;Br vector at C
</p>
<p>Fig. 9.14 Patterson section &eth;u; 1
2
;w&THORN; showing a double-weight Br&ndash;Br vector at D</p>
<p/>
</div>
<div class="page"><p/>
<p>9.2.6 Molecular Geometry
</p>
<p>It remained to determine the bond lengths, bond angles, and other features of the geometry of the
</p>
<p>molecule and its relationship with other molecules in the unit cell.
</p>
<p>From the coordinates of the atomic positions (Table 9.2) and using (7.119) and (7.121), bond
</p>
<p>lengths and angles were calculated. They are shown on the drawings of the molecule in Figs. 9.18
</p>
<p>and 9.19. Figure 9.20 illustrates the packing of the molecules in the unit cell, as seen along b; the
</p>
<p>average intermolecular contact distance is 3.7 Å, a typical intermolecular contact distance in
</p>
<p>organic compounds, in which van der Waals forces link the molecules in the solid state.
</p>
<p>In a molecule of this nature, the planarity of the ring system is of stereochemical interest. The
</p>
<p>equation of a plane, Ax + By + Cz &frac14; D, can be solved by three triplets x, y, z. Hence, the best
molecular plane is obtained by a least-squares procedure that minimizes the sum of the squares of the
</p>
<p>deviations d of all of the atoms from the plane. The results are listed in Table 9.3. It can be seen that
</p>
<p>the deviations of the atoms from the best plane are not significant, and it is possible to conclude,
</p>
<p>therefore, that the introduction of the heteroatom has but little effect on the planarity of the
</p>
<p>benzofluorene moiety.
</p>
<p>Fig. 9.15 Patterson section (u,0.375,w), showing a single-weight Br&ndash;Br vector B
</p>
<p>9.2 Crystal Structure of 2-Bromobenzo[b] Indeno[1,2-e] Pyran 451</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_7#Eq119_7">http://dx.doi.org/10.1007/978-1-4614-3954-7_7#Eq119_7</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_7#Eq121_7">http://dx.doi.org/10.1007/978-1-4614-3954-7_7#Eq121_7</a></div>
</div>
<div class="page"><p/>
<p>Fig. 9.16 Composite three-dimensional electron density map with the molecule (excluding H atoms) marked in, as
seen along a. The contour of zero electron density is not shown, and the numbers represent 100x for each atom.
</p>
<p>A symmetry-related position to that chosen in Sect. 9.2.4 has been selected here for the Br atom (What is this symmetry
</p>
<p>operation?)
</p>
<p>Fig. 9.17 Structural formula for BBIP
</p>
<p>452 9 Examples of Crystal Structure Determination</p>
<p/>
</div>
<div class="page"><p/>
<p>Table 9.2 Fractional atomic coordinates in BBIP, with esds in parenthesesa
</p>
<p>Atom x y z
</p>
<p>Br 0.7602(2) 0.3152(3) 0.4848(0)
</p>
<p>C(1) 0.7820(16) 0.4187(22) 0.3789(4)
</p>
<p>C(2) 0.7310(16) 0.4951(24) 0.4252(4)
</p>
<p>C(3) 0.6524(16) 0.7075(25) 0.4297(5)
</p>
<p>C(4) 0.6214(16) 0.8413(23) 0.3871(5)
</p>
<p>C(4a) 0.6794(16) 0.7619(22) 0.3406(4)
</p>
<p>O(5) 0.6520(11) 0.9051(14) 0.2995(3)
</p>
<p>C(5a) 0.6973(14) 0.8329(21) 0.2526(4)
</p>
<p>C(6) 0.6714(14) 0.9397(19) 0.2077(5)
</p>
<p>C(6a) 0.7384(15) 0.7990(19) 0.1678(4)
</p>
<p>C(7) 0.7401(17) 0.8230(24) 0.1150(4)
</p>
<p>C(8) 0.8078(18) 0.6526(24) 0.0858(4)
</p>
<p>C(9) 0.8766(17) 0.4574(24) 0.1079(5)
</p>
<p>C(10) 0.8731(16) 0.4268(21) 0.1605(4)
</p>
<p>C(10a) 0.8035(16) 0.5954(20) 0.1908(4)
</p>
<p>C(10b) 0.7767(14) 0.6076(21) 0.2454(5)
</p>
<p>C(11) 0.8064(15) 0.4734(21) 0.2850(4)
</p>
<p>C(11a) 0.7593(14) 0.5475(20) 0.3359(5)
</p>
<p>H(1) 0.809 0.239 0.375
</p>
<p>H(3) 0.622 0.789 0.460
</p>
<p>H(4) 0.565 0.999 0.389
</p>
<p>H(6) 0.630 0.121 0.208
</p>
<p>H(7) 0.674 0.944 0.097
</p>
<p>H(8) 0.804 0.667 0.043
</p>
<p>H(9) 0.886 0.361 0.076
</p>
<p>H(10) 0.809 0.253 0.375
</p>
<p>H(11) 0.870 0.311 0.285
</p>
<p>aThere are no esd&rsquo;s for the hydrogen atom coordinates because these
</p>
<p>parameters were not included in the least-squares refinement
</p>
<p>Fig. 9.18 Bond lengths in BBIP, with their estimated standard deviations in parentheses</p>
<p/>
</div>
<div class="page"><p/>
<p>Fig. 9.19 Bond angles in BBIP, with their estimated standard deviations in parentheses
</p>
<p>Table 9.3 Deviations of atoms from
the least-squares plane through the
</p>
<p>moleculea
</p>
<p>Atom Deviation (Å)
</p>
<p>Br 0.03
</p>
<p>C(1) 0.01
</p>
<p>C(2) �0.04
C(3) �0.05
C(4) �0.06
C(4a) 0.01
</p>
<p>O(5) 0.05
</p>
<p>C(5a) 0.06
</p>
<p>C(6) 0.02
</p>
<p>C(6a) 0.02
</p>
<p>C(7) �0.06
C(8) �0.08
C(9) �0.03
C(10) 0.03
</p>
<p>C(10a) 0.04
</p>
<p>C(10b) 0.04
</p>
<p>C(11) 0.03
</p>
<p>C(11a) 0.03
</p>
<p>aThe mean estimated standard deviation
</p>
<p>of the deviations is 0.02, so that hardly
</p>
<p>any atoms deviate significantly from the
</p>
<p>best plane at a 3s level
</p>
<p>Fig. 9.20 Stereoview of the molecular packing in the structure of BBIP, as seen along a
</p>
<p>454 9 Examples of Crystal Structure Determination</p>
<p/>
</div>
<div class="page"><p/>
<p>9.3 Crystal Structure of Potassium 2-Hydroxy-3,4-Dioxocyclobut-1-ene-1-
Olate Monohydrate (KHSQ) [3]
</p>
<p>1,2-Dihydroxy-3,4-dioxocyclobut-1-ene (IV) may be prepared by the acid-catalyzed hydrolysis of
</p>
<p>1,2-diethoxy-3,3,4,4-tetrafluorocyclobut-1-ene (V). On recrystallization from water, it has a melting
</p>
<p>point of 293�C, at which temperature it decomposes.
It has been called by the trivial name, squaric acid; the hydrogen atoms in the hydroxyl groups are
</p>
<p>acidic, and can be replaced by a metal. Potassium hydrogen squarate monohydrate (VI), which is the
</p>
<p>subject of this example, can be obtained by mixing hot, concentrated, equimolar aqueous solutions of
</p>
<p>potassium hydroxide and squaric acid and then cooling the reaction mixture.
</p>
<p>9.3.1 Preliminary X-Ray and Physical Measurements
</p>
<p>The compound was recrystallized from water as colorless, prismatic crystals with the forms {001},
</p>
<p>{110}, and {100} predominant, Fig. 9.21. Under a polarizing microscope, straight extinction was
</p>
<p>observed on {001} and {100}, and an extinction angle of about 2� was obtained on a section cut
normal to b. These results suggest strongly that the crystals belong to the monoclinic system. The
</p>
<p>density was measured in the manner indicated in Sect. 9.2.1.
</p>
<p>The crystal specimen chosen for X-ray work had the dimensions 0.5, 0.5, and 0.3 mm parallel to a,
</p>
<p>b, and c, respectively. The details of the preliminary measurements are similar to those described for
</p>
<p>the previous example, and we list the crystal data immediately, Table 9.4. Copper Ka radiation
</p>
<p>(l &frac14; 1.5418 Å) was used throughout this work.
</p>
<p>Fig. 9.21 Crystal habit of potassium hydrogen squarate monohydrate (KHSQ) with the crystallographic axes drawn in
</p>
<p>9.3 Crystal Structure of Potassium 2-Hydroxy-3,4-Dioxocyclobut&ndash;1&ndash;ene&ndash;1-Olate. . . 455</p>
<p/>
</div>
<div class="page"><p/>
<p>9.3.2 Intensity Measurement and Correction
</p>
<p>Nine hundred symmetry-independent intensities with sin y=l � 0:57 A�1 were measured with a
CAD-4 diffractometer. Corrections were applied for polarization and Lorentz effects, but not for
</p>
<p>absorption. Scale (K) and isotropic temperature (B) factors were deduced by Wilson&rsquo;s method, and
</p>
<p>the Fo data were converted to jEj values, see (8.1).
The structure analysis began with 142 jEj values �1.5, representing 15.8% of the experimental
</p>
<p>reflection data, and the jEj statistics are shown in Table 9.5. The agreement with the theoretical values
for a centric distribution of jEj values is very close, in accord with the chosen centrosymmetric space
group, Table 9.4.
</p>
<p>9.3.3 �2-Listing
</p>
<p>The next stage was the preparation of a
P
</p>
<p>2 listing, Sect. 8.2.5. Symmetry-related reflections become
</p>
<p>very important in generating triplet relationships: 300 and 304 can lead to both 004 and 604, the latter
</p>
<p>Table 9.4 Crystal data for KHSQ at 20�C
</p>
<p>Molecular formula C4HO
�
4 , K
</p>
<p>+, H2O
</p>
<p>Mr 170.17
</p>
<p>Space group P21/c
</p>
<p>a (Å) 8.641(1)
</p>
<p>b (Å) 10.909(1)
</p>
<p>c (Å) 6.563(2)
</p>
<p>b (�) 99.81(1)
</p>
<p>Vc (Å
3) 609.6(2)
</p>
<p>Dm (g cm
�3) 1.839(7)
</p>
<p>Dc (g cm
�3) 1.854(1)
</p>
<p>Z 4
</p>
<p>F(000) 344
</p>
<p>Table 9.5 Statistics of jEj values in KHSQ
Acentric Centric This structure
</p>
<p>jEj2 1.00 1.00 1.00
</p>
<p>jEj 0.89 0.80 0.81
</p>
<p>jEj2 � 1 0.74 0.97 0.95
% � 1.0 36.8 31.7 33.9
% � 1.5 10.5 13.4 14.6
% � 1.75 4.7 8.0 8.4
% � 2.0 1.8 4.6 4.9
% � 2.5 0.2 1.2 1.1
</p>
<p>456 9 Examples of Crystal Structure Determination</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_8#Eq1_8">http://dx.doi.org/10.1007/978-1-4614-3954-7_8#Eq1_8</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_8#Sec7_8">http://dx.doi.org/10.1007/978-1-4614-3954-7_8#Sec7_8</a></div>
</div>
<div class="page"><p/>
<p>by replacing 304 by �304, taking note of the phase symmetry. The relevant phase symmetry for space
</p>
<p>group P21/c follows from (3.87), and may be summarized as follows:
</p>
<p>s&eth;hkl&THORN; &frac14; s&eth;�h �k �l&THORN; (9.2)
</p>
<p>s&eth;hkl&THORN; &frac14; s&eth;h�kl&THORN;&eth;�1&THORN;k&thorn;l (9.3)
</p>
<p>A portion of the
P
</p>
<p>2 listing is shown in Table 9.6; full use of the symmetry relationships (9.2) and
</p>
<p>(9.3) has been made in setting up the listing. The numbers in parentheses under each h are the total
numbers of
</p>
<p>P
2 triplets for each of these reflections; k and h � k represent those reflections forming a
</p>
<p>vector triplet with h.
</p>
<p>9.3.4 Specifying the Origin
</p>
<p>Following the procedure described in Sect. 8.2.2 and using the reflections in Table 9.6, three
</p>
<p>reflections were chosen and allocated positive signs, in order to fix the origin at 0,0,0. The symmetry
</p>
<p>relationships in the space group of this compound allowed, in all, 12 signs in the origin set, Table 9.7.
</p>
<p>The reader should check the signs, starting from the first one in each group of four, using (9.2)
</p>
<p>and (9.3).
</p>
<p>Table 9.6 Part of the
P
</p>
<p>2 listing for KHSQ
a
</p>
<p>h jE(h)j k jE(k)j h � k jE(h � k)j jE(h)j jE(k)j jE(h � k)j
531 (37) 2.6 010; 4 2.8 573 2.6 18.9
</p>
<p>041 2.2 572 3.3 17.2
0041 2.0 570 2.7 14.0
114 2.3 625 1.7 10.2
032 1.7 563 2.0 8.8
</p>
<p>114 (45) 2.3 572 3.3 482 1.9 14.4
664 1.8 570 2.7 11.2
681 1.5 573 2.6 9.0
563 2.0 451 1.5 6.9
454 1.6 540 1.5 5.5
</p>
<p>032 (54) 1.7 531 2.6 563b 2.0 8.8
572 3.3 540 1.5 8.4
482 1.9 454 1.6 5.2
51 1.5 481b 2.0 5.1
</p>
<p>112 (39) 2.5 572 3.3 664 1.8 14.9
482 1.9 570 2.7 12.8
114 2.3 002 1.9 10.2
571 1.7 681 1.5 6.4
</p>
<p>010,4 (35) 2.8 332 2.2 372 1.9 11.7
625 1.7 681 1.5 7.1
</p>
<p>332 (46) 2.2 332 2.2 664 1.8 8.7
114 2.3 242 1.7 8.6
313 1.8 041 2.0 7.9
625 1.7 313 1.8 6.7
</p>
<p>002(25) 1.9 041 2.0 041 2.0 7.4
114 2.3 116 1.5 6.6
681 1.5 681 1.6 4.6
</p>
<p>aWe use the notation h for hkl, k for h0k0l0, and h � k for h � h0, k � k0, l � l0
bUses 032
</p>
<p>9.3 Crystal Structure of Potassium 2-Hydroxy-3,4-Dioxocyclobut&ndash;1&ndash;ene&ndash;1-Olate. . . 457</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Eq87_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Eq87_3</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_8#Sec4_8">http://dx.doi.org/10.1007/978-1-4614-3954-7_8#Sec4_8</a></div>
</div>
<div class="page"><p/>
<p>9.3.5 Sign Determination
</p>
<p>The
P
</p>
<p>2 listing was examined with a view to generating new signs, using (8.9), which may be
</p>
<p>given by
</p>
<p>s&frac12;E&eth;h&THORN;� � s
X
</p>
<p>k
</p>
<p>E&eth;k&THORN;E&eth;h� k&THORN;
" #
</p>
<p>(9.4)
</p>
<p>where the sum is taken over the several k triplets all involved with the given h. The probability of
(9.4) is given by (8.10). If only a single
</p>
<p>P
2 interaction is considered, (9.4) becomes
</p>
<p>s&frac12;E&eth;h&THORN;�� s&frac12;E&eth;k&THORN;sE&eth;h� k&THORN;�
</p>
<p>and the probability calculation omits the summation given in (8.11).
</p>
<p>Since the values of the probability P+(h) were sufficiently high, the signs were accepted as
correct; very small or zero values of P+(h) indicated strongly a negative sign for h. Some examples
of the application of (9.5) are given in Table 9.8. It does not matter which reflections in a triplet are
</p>
<p>labeled h and k.
</p>
<p>Table 9.7 Origin-fixing reflections and their symmetry
equivalents
</p>
<p>Reflections Sign jEj No. ofP2 triplets
531 + 2.6 37
</p>
<p>531 +
</p>
<p>531 +
</p>
<p>531 +
</p>
<p>114 + 2.3 45
</p>
<p>114 +
</p>
<p>114 �
114 �
032 + 1.7 54
</p>
<p>032 +
</p>
<p>032 �
032 �
</p>
<p>Table 9.8 Sign determination starting from the origin set
</p>
<p>k h � k h Indication for s(h)
531 + 114� 625 �
531 + 03�2� 563 �
56�3� 114&thorn; 451 �
56�3� 032+ 591 �
451� 03�2� 481 +
</p>
<p>458 9 Examples of Crystal Structure Determination</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_8#Eq9_8">http://dx.doi.org/10.1007/978-1-4614-3954-7_8#Eq9_8</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_8#Eq10_8">http://dx.doi.org/10.1007/978-1-4614-3954-7_8#Eq10_8</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_8#Eq11_8">http://dx.doi.org/10.1007/978-1-4614-3954-7_8#Eq11_8</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_9#Eq5_9">http://dx.doi.org/10.1007/978-1-4614-3954-7_9#Eq5_9</a></div>
</div>
<div class="page"><p/>
<p>Use of Sign Symbols
The above process of sign determination was applied to the entire
</p>
<p>P
2 listing which, although it
</p>
<p>contained 1276 triple products, was exhausted after only 24 signs had been found. To enable further
</p>
<p>progress to be made, three reflections were assigned the symbols A, B, and C, where each symbol
</p>
<p>represented either a plus or minus sign. Twelve symbolic signs, Table 9.9, were thus added to the set,
</p>
<p>and the sign determination was continued, now in terms of both signs and symbols. It may be noted
</p>
<p>that although the symbols are given to reflections with large jEj values and large numbers ofP
2 interactions, there are not, necessarily, any restrictions on either parity groups or the use of
</p>
<p>structure seminvariants.
</p>
<p>Some examples of this stage of the process are given in Table 9.10. The values of h and k are taken
from either Tables 9.7 and 9.9, which constitute the &ldquo;starting set,&rdquo; or as determined through (9.4). The
</p>
<p>reader is invited to follow through the stages in Table 9.10, working out the correct symmetry-
</p>
<p>equivalent signs from (9.2) and (9.3) as necessary.
</p>
<p>From Table 9.10, we see that six more reflections have been allocated signs, and another 17 are
</p>
<p>determined in terms of A, B, and C. Multiple indications can now be seen. For example, there are
</p>
<p>two indications that s(573) &frac14; B, two indications that s(570) &frac14; �, and two indications that s
(540) &frac14; A. Three indications for 041 suggest that both s(041) &frac14; � and A &frac14; �.
</p>
<p>Continuing in this manner, it was found possible to allot signs and symbols to all 142 jEj values
greater than 1.5. The symbols A, B, and C were involved in 65, 72, and 55 relationships,
</p>
<p>respectively. Consistent indications, such as those mentioned above for s(041), led finally to the
</p>
<p>sign relationships A &frac14; AC &frac14; B &frac14; �, from which it follows that C &frac14; +. It does not always turn out
that the signs represented by symbols can be allocated from the analysis in that complete and
</p>
<p>satisfactory manner. If there are n undetermined symbols, then there will be, in general, 2n sets of
</p>
<p>signs to be examined. In this case, figures of merit, such as those discussed in Sect. 8.2.13, can be
</p>
<p>used to indicate that set of signs most likely to be correct. It may not follow that the indicated set is
</p>
<p>correct, and some trials with Fourier syntheses may be needed at this stage in order to elicit the
</p>
<p>correct result.
</p>
<p>9.3.6 The E Map
</p>
<p>The signs of the 142 jEj values used in this procedure were obtained with a high probability, and an
electron density map was computed using the signed jEj values as coefficients. The sections of this
</p>
<p>Table 9.9 Symbolic signs
</p>
<p>Reflection Sign jEj No. ofP2 relationships
112 A 2.5 39
</p>
<p>1 12 A
</p>
<p>1 1 2 �A
112 �A
010,4 B 2.8 35
</p>
<p>010; 4 B
</p>
<p>10; 4 B
</p>
<p>010,4 B
</p>
<p>332 C 2.2 46
</p>
<p>332 C
</p>
<p>332 �C
332 �C
</p>
<p>9.3 Crystal Structure of Potassium 2-Hydroxy-3,4-Dioxocyclobut&ndash;1&ndash;ene&ndash;1-Olate. . . 459</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_8#Sec15_8">http://dx.doi.org/10.1007/978-1-4614-3954-7_8#Sec15_8</a></div>
</div>
<div class="page"><p/>
<p>map r(xyz) at z &frac14; 0.15, 0.20, 0.25, and 0.30 are shown in Figs. 9.22, 9.23, 9.24, and 9.25. They reveal
the K+ ion and the C4O
</p>
<p>�
4 ring system clearly; the oxygen atom Ow of the water molecule was not
</p>
<p>indicated convincingly at this stage of the analysis. A tilt of the plane of the molecule with respect to
</p>
<p>(001) can be inferred from Figs. 9.23, 9.24, and 9.25. Some spurious peaks S may be seen. This is a
</p>
<p>common feature of E maps. We must remember that a limited data set (142 out of 900) is being used,
</p>
<p>and that the jEj values are sharpened coefficients corresponding to an approximate point-atom model.
The data set is therefore terminated while the coefficients for the Fourier series are relatively large, a
</p>
<p>procedure that can lead to spurious maxima; see Sect. 6.9.1. However, such peaks are often of smaller
</p>
<p>weight than those that correspond to true atomic positions.
</p>
<p>Table 9.10 Further sign determinationsa
</p>
<p>k s(k) (h � k) s(h � k) h Sign indication, sh
010.4 B 625 � 681 s&eth;68�1&THORN; &frac14; B
010.4 B 332 C 372 s(372) &frac14; �BC
112 A 681 B 571 s(571) &frac14; AB
531 + 010,4 B 573 s(573) &frac14; B
114 + 681 B 573 s(573) &frac14; B
332 C 332 C 664 s(664&THORN; &frac14; CC &frac14; &thorn;
114 + 664 &thorn; 570 s(570) &frac14; �
112 A 570 � 482 s(482) &frac14; A
112 A 664 &thorn; 572 s(572&THORN; &frac14; �A
114 + 572 �A 482 s(482) &frac14; A
032 + 572 �A 540 s(540) &frac14; A
032 + 482 A 454 s(454) &frac14; �A
114 + 454 �A 540 s(540) &frac14; A
332 C 114 &thorn; 242 s(242) &frac14; �C
112 A 114 &thorn; 002 s(002) &frac14; A
112 A 482 A 570 s(570) &frac14; �AA &frac14; �
570 � 531 &thorn; 041 s(041) &frac14; �
625 � 332 C 313 s(313&THORN; &frac14; C
313 C 332 C 041 s(041) &frac14; �CC &frac14; �
531 + 041 � 572 s&eth;57�2&THORN; &frac14; &thorn;
002 A 041 � 041 s(041) &frac14; A
002 A 681 B 681 s(681) &frac14; AB
002 A 114 &thorn; 116 s(116&THORN; &frac14; A
aSymmetry relations should be employed as necessary
</p>
<p>Fig. 9.22 E map for KHSQ at z &frac14; 0.15
</p>
<p>460 9 Examples of Crystal Structure Determination</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_6#Sec29_6">http://dx.doi.org/10.1007/978-1-4614-3954-7_6#Sec29_6</a></div>
</div>
<div class="page"><p/>
<p>Fig. 9.23 E map for
KHSQ at z &frac14; 0.20
</p>
<p>Fig. 9.24 E map for
KHSQ at z &frac14; 0.25
</p>
<p>Fig. 9.25 E map for
KHSQ at z &frac14; 0.30
</p>
<p>9.3 Crystal Structure of Potassium 2-Hydroxy-3,4-Dioxocyclobut&ndash;1&ndash;ene&ndash;1-Olate. . . 461</p>
<p/>
</div>
<div class="page"><p/>
<p>9.3.7 Completion and Refinement of the Structure
</p>
<p>Sometimes, all atomic positions are not contained among the peaks in an E map. Those peaks that do
</p>
<p>correspond to chemically sensible atomic positions may be used to form a trial structure for
</p>
<p>calculation of structure factors and an Fo electron density map. A certain amount of subjective
</p>
<p>judgment may be required to decide upon the best peaks for the trial structure at such a stage.
</p>
<p>This situation was obtained for KHSQ, although it was not difficult to pick out a good trial
</p>
<p>structure. Coordinates were obtained for all non-hydrogen atoms except the oxygen atom of the water
</p>
<p>molecule. The R-factor for this trial structure was 0.30, and the composite three-dimensional electron
</p>
<p>density map obtained is shown in Fig. 9.26, which now reveals Ow clearly. It may be noted in passing
</p>
<p>that the small peak labeled Ow in Fig. 9.22 corresponds to the position of this atom, but this fact could
</p>
<p>not be determined conclusively at that stage of the analysis.
</p>
<p>Fig. 9.26 Composite electron density map for KHSQ (excludingH atoms); the atomic coordinates are listed in Table 9.11
</p>
<p>Fig. 9.27 Composite difference electron density map for KHSQ. Positive contours are solid lines and negative
contours are broken lines. Bonds in the squarate ring and those involving hydrogen atoms are shown as dotted lines.
</p>
<p>Some spurious, small peaks (unlabeled) we shown by this synthesis
</p>
<p>462 9 Examples of Crystal Structure Determination</p>
<p/>
</div>
<div class="page"><p/>
<p>Further refinement was carried out by the method of least squares, and an R-factor of 0.078 was
</p>
<p>obtained. Figure 9.27 shows a composite three-dimensional difference-Fourier map for KHSQ. Peaks
</p>
<p>numerically greater than 0.5, representing about twice s(r0), are significant, and have been contoured.
</p>
<p>Some of these peaks indicate areas of small disagreement between the true structure and the model.
</p>
<p>Three positive peaks, however, are in positions expected for hydrogen atoms. Inclusion of these atoms in
</p>
<p>the structure factor calculations in the final cycles of least-squares refinement had a small effect on theR-
</p>
<p>factor, bringing it to its final value of 0.077. The fractional atomic coordinates for the atoms in the
</p>
<p>asymmetric unit are listed in Table 9.11.
</p>
<p>Interatomic distances and angles are shown in Fig. 9.28, and a molecular packing diagram, as seen
</p>
<p>along c, is given in Fig. 9.29. From the analysis, we find that intermolecular hydrogen bonds exist
</p>
<p>between O(2) and O(1)0 [2.47(1) Å], between O(3)˝ and Ow [2.76(1) Å], and between O(4)
0 and Ow
</p>
<p>[2.95(1) Å]; they are largely responsible for the cohesion between molecules in the solid state.1
</p>
<p>Electron Counts on Atoms
We cannot determine from this analysis how the electronic charge on the individual species is
</p>
<p>distributed. With X-rays, a more precise analysis would be needed, followed by a peak integration
</p>
<p>of an electron density map. However, an ab initio calculation with Gaussian 94 on a free KHSQ
</p>
<p>species resulted in the following electron population parameters p on the individual atomic species.
</p>
<p>p p p p
</p>
<p>K 18.111 C1 5.581 C2 5.833 C3 5.605
</p>
<p>C4 5.599 O1 8.808 O2 8.760 O3 8.626
</p>
<p>O4 8.544 Ow 8.855 H12 0.564 H3w 0.560
</p>
<p>H4w 0.554
</p>
<p>The value of
P
</p>
<p>p is 86, which is also F(000)/4, as expected, the number of electrons in a single
</p>
<p>species. Although the above results would be modified slightly in the crystal environment, they are in
</p>
<p>excellent agreement with the formula weight and indicate the drawing of electron density from the
</p>
<p>less electronegative species, particularly the hydrogen atoms, towards the more electronegative
</p>
<p>Table 9.11 Fractional atomic coordinates for KHSQ
</p>
<p>x y z
</p>
<p>K+ 0.8249(2) 0.1040(2) 0.1295(3)
</p>
<p>C(1) 0.4353(9) 0.1295(7) 0.2572(12)
</p>
<p>C(2) 0.4495(9) 0.2597(7) 0.2714(12)
</p>
<p>C(3) 0.2795(9) 0.2714(8) 0.2462(11)
</p>
<p>C(4) 0.2659(9) 0.1345(7) 0.2305(12)
</p>
<p>O(1) 0.5399(6) 0.0450(5) 0.2649(10)
</p>
<p>O(2) 0.5649(6) 0.3346(5) 0.2920(10)
</p>
<p>O(3) 0.1874(7) 0.3582(6) 0.2386(10)
</p>
<p>O(4) 0.1578(6) 0.0605(5) 0.2022(10)
</p>
<p>Ow 0.8789(7) 0.3429(6) 0.0424(10)
</p>
<p>H(12) 0.522 0.413 0.246
</p>
<p>H(3w) 1.000 0.346 0.075
</p>
<p>H(4w) 0.826 0.400 0.100
</p>
<p>1 Single and double primes indicate different neighboring molecules.
</p>
<p>9.3 Crystal Structure of Potassium 2-Hydroxy-3,4-Dioxocyclobut&ndash;1&ndash;ene&ndash;1-Olate. . . 463</p>
<p/>
</div>
<div class="page"><p/>
<p>Fig. 9.29 Molecular packing diagram of one layer of the KHSQ structure as seen along c. The circles in order of
decreasing size represent K, O, C, and H. The hydrogen-bond network is shown by dashed lines
</p>
<p>Fig. 9.28 Bond lengths and bond angles in the asymmetric unit of KHSQ; the OH. . .O distances refer to the overall
O. . .O separations. Primes on atom symbols indicate neighboring asymmetric units; this diagram should be studied in
conjunction with Fig. 9.29
</p>
<p>464 9 Examples of Crystal Structure Determination</p>
<p/>
</div>
<div class="page"><p/>
<p>oxygen atoms. This effect leads to bond lengths involving hydrogen that measure shorter by X-ray
</p>
<p>diffraction than with neutrons, as we shall see in Chap. 11.
</p>
<p>9.4 Crystal and Molecular Structure and Absolute Configuration of 3b-
Acetoxy-6,7-Epidithio-19-Norlanosta-5,7,9,11-Tetraene [4]
</p>
<p>This interesting and stable dithiete compound (I) was obtained during an attempt to convert a
</p>
<p>derivative of lanostenol (II) into a derivative of euphenol (III), through the series of reactions
</p>
<p>(IV)&ndash;(VII). There are very few authenticated reports of 1,2-dithiones (VIII) or their tautomers, 1,2-
</p>
<p>dithietes (IX) [5]. It may be noted en passant that (III) is closely related to the compound in Fig. 1.7b,
</p>
<p>but a 3-b-hydroxyl group takes the place of the 3-b-iodoacetoxy group. At the time that this structure
</p>
<p>was reported, it was the first representative of its class. The stability of the dithiete arises most
</p>
<p>probably from a steric protection by the host molecule. From a crystallographic point of view, it has
</p>
<p>the interest of two heavy atoms, not related by symmetry, in the asymmetric unit, and of easily
</p>
<p>sufficient &ldquo;weight&rdquo; for a reliable determination of the absolute configuration of the molecule.
</p>
<p>9.4 Crystal and Molecular Structure and Absolute Configuration. . . 465</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_11">http://dx.doi.org/10.1007/978-1-4614-3954-7_11</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Fig7_1">http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Fig7_1</a></div>
</div>
<div class="page"><p/>
<p>9.4.1 Preparation and Preliminary Optical and X-Ray Examinations
</p>
<p>3-b-Acetoxy-7,7-ethlenedithio-5a-lanost-8-en-11-b-ol (IV) in pyridine heated under reflux with
</p>
<p>phosphorus oxychloride gave a colorless compound, the expected 8,11-diene (V). After a prolonged
</p>
<p>reaction time under reflux, colorless crystals corresponding to the formula C33H48O2S2, m.p.
</p>
<p>161&ndash;163�C, [a]D &frac14; 45� separated. Examination by n.m.r and u.v. indicated a 2,3-dihydro-1,4-
benzodithiine (VI). Photolysis of this compound in n-heptane at �20�C under medium-power
mercury light produced a quantitative conversion to C31H44O2S2 (VII, I) and ethylene (C2H4) [6].
</p>
<p>The compound (I) was recrystallized from a 1:1 mixture of dichloromethane and methanol as pale-
</p>
<p>yellow, lath-like parallelepipeds elongated in the direction of an axis, later to be described as z. The
</p>
<p>crystal forms present in decreasing order of development were {100}, {010}, and {001}. An optical
</p>
<p>examination indicated strongly that the crystals were monoclinic, with a b-angle close to 90�.
The density was measured by flotation in aqueous sodium bromide at 25�C. The result, 1.17
</p>
<p>(1) g cm�3, was obtained by measuring the refractive index of the flotation-equilibrium solution and
then obtaining the density from a calibration chart.
</p>
<p>X-ray photographs confirmed that the crystals were monoclinic and provided approximate unit-
</p>
<p>cell dimensions. The only systematic extinctions were 0k0 for k &frac14; 2n + 1; hence, the space group
was either P21 or P21/m. From the unit-cell and density measurements Z evaluated to 1.996, or 2 to
</p>
<p>the nearest integer. Thus, for crystals of the asymmetric molecule (I) under investigation, space
</p>
<p>group P21/m was eliminated, and so we find two sulphur atoms per unit in space group P21.
</p>
<p>9.4.2 X-Ray Measurement of the Unit-Cell Dimensions and Intensities
</p>
<p>Accurate unit-cell dimensions were obtained from a least-squares fit to the y values of 30 reflections
</p>
<p>with y &gt; 49�, measured on a Siemens four-circle diffractometer. Resolution of the Cu Ka1 peaks
</p>
<p>466 9 Examples of Crystal Structure Determination</p>
<p/>
</div>
<div class="page"><p/>
<p>(l &frac14; 1.54056 Å) and those for Cu Ka2 (l &frac14; 1.54439 Å) was obtained for these reflections, which
were chosen so as to be evenly distributed in reciprocal space. The crystal data are summarized in
</p>
<p>Table 9.12. In the usual order of a &lt; b &lt; c, a and c have been reversed in order that the higher
</p>
<p>convention of b &gt; 90� was preserved.
The intensities of 2605 of a theoretically possible 2629 symmetry-independent reflections with
</p>
<p>sin y=l � 0:60 were measured on the diffractometer by a five-measurement technique [7]. Of the
total measured reflections, 243 for which I &lt; 2.58s(I), were very weak; the average standard
</p>
<p>deviations of the Fo data were 1.8% (2362 data) and 2.7% (all reflections). The very weak reflections
</p>
<p>were omitted from the structure analysis. A reference reflection was measured after every 20
</p>
<p>reflections: the data were scaled, first within batches of 20 reflections and then to the average value
</p>
<p>of the reference reflection. The scaling factors varied over a range of less than�2%, and there was no
evidence of crystal decomposition.
</p>
<p>Corrections were applied for Lorentz and polarization effects, but not for absorption (mR &frac14; 0:78).
jEj values were calculated by the K-curve method; their statistics, listed in Table 9.13, show very
good agreement with those for the ideal acentric distribution.
</p>
<p>Table 9.12 Crystal data for the title compound
</p>
<p>Formula C31H44O2S2
</p>
<p>Mr 512.82
</p>
<p>System/space group Monoclinic/P21
</p>
<p>Crystal dimensions/mm 0.2, 0.3, 0.5
</p>
<p>a (Å) 20.1896(9)
</p>
<p>b (Å) 11.0709(5)
</p>
<p>c (Å) 6.4953(3)
</p>
<p>b (�) 90.578(2)
</p>
<p>Vc (Å
3) 1451.7(1)
</p>
<p>Dm (g cm
�3) 1.17(1)
</p>
<p>Dc (g cm
�3) 1.173(1)
</p>
<p>Z 2
</p>
<p>F(000) 556
</p>
<p>m(Cu Ka) (cm�1) 19P
Z2(S)/
</p>
<p>P
Z2 (light atoms) 0.4
</p>
<p>Table 9.13 Statistics of jEj values
This structure Acentric Centric
</p>
<p>jE2j 1.00 1.00 1.00
</p>
<p>jEj 0.88 0.89 0.80
</p>
<p>jEj2 � 1 0.76 0.74 0.97
% � 1:5 11.2 10.5 13.4
% � 1:75 4.7 4.7 8.0
% � 2:0 2.4 1.8 4.6
% � 2:5 0.4 0.2 1.2
</p>
<p>9.4 Crystal and Molecular Structure and Absolute Configuration. . . 467</p>
<p/>
</div>
<div class="page"><p/>
<p>9.4.3 Structure Determination and Refinement
</p>
<p>Patterson and sharpened (jEj2 � 1) syntheses revealed the expected heavy-atom peaks on the &eth;u; 1
2
;w&THORN;
</p>
<p>Harker section, and in addition four peaks of about double the weight of the Harker peaks, with a
</p>
<p>geometry in accord with that of two pairs of heavy (S) atoms, each pair being related by 21 symmetry.
</p>
<p>The y coordinate of S(1) was fixed at 3
4
, and that for S(2) was deduced to be 0.69. Their difference in y of
</p>
<p>0.66 Å was sufficient to break the pseudo-symmetry which occurs in P21 where the asymmetric unit
</p>
<p>contains two heavy atoms. A similar situation has been reported, for example, with epilimonol
</p>
<p>iodoacetate [8].
</p>
<p>Successive Fourier syntheses located all non-hydrogen atoms in the molecule, and the structure
</p>
<p>was refined on jFj by full-matrix anisotropic least squares. A Fourier difference synthesis enabled
most of the hydrogen atoms to be located and they were refined isotropically. The six hydrogen atoms
</p>
<p>on the terminal methyl groups C(26) and C(27) were relatively poorly defined, and their coordinates
</p>
<p>were determined geometrically in terms of their minimum energy conformations. It is highly probable
</p>
<p>that some disorder exists in this region of the structure; see also Fig. 1.7 and accompanying text.
</p>
<p>Corrections for anomalous dispersion were applied (DfS
00 &frac14; 0.60) and the refinement converged at
</p>
<p>R &frac14; 3.4% (Rw &frac14; 4.7%). All computations were performed by both local programs and the XRAY-72
system [9]. A list of structure factors has been deposited with the British Library Lending Division as
</p>
<p>Supplementary Publication No. SUP 31423 (23 pp. 1 microfiche).
</p>
<p>9.4.4 Absolute Configuration
</p>
<p>The absolute configuration was examined by carrying out the least-squares refinement again, but with
</p>
<p>the signs of the imaginary components Df 00 of the structure factors reversed. Convergence was
attained at R &frac14; 3.6% (Rw &frac14; 4.9%). The ratio of the weighted R values is 1.04, and by Hamilton&rsquo;s
ratio test [10], the result is significant at a level better than 0.01. Thus, the absolute configuration
</p>
<p>corresponds, not surprisingly, to the acetoxy group in the b-configuration, with the configurations at C
</p>
<p>(13), C(14), and C(17) unchanged from those of the starting material (IV). Although the configuration
</p>
<p>was quite clear from this result, the Flack parameter, Sect. 7.6.1, would now constitute the preferred
</p>
<p>test.
</p>
<p>9.5 Discussion of the Structure
</p>
<p>Figure 9.30 is a stereoscopic illustration of themolecule (I) as seen along &frac12;021�. The central ring system
shows a high degree of planarity: the average deviation of the atoms forming the plane C(5)&ndash;C(10) is
</p>
<p>0.01 Å, and that for the atoms S(1), S(2), C(1), C(4), C(11), and C(14) from the same plane is 0.07 Å.
</p>
<p>Fig. 9.30 Stereoscopic illustration of the title compound (I), as seen along &frac12;021�
</p>
<p>468 9 Examples of Crystal Structure Determination</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Fig7_1">http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Fig7_1</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_7#Sec32_7">http://dx.doi.org/10.1007/978-1-4614-3954-7_7#Sec32_7</a></div>
</div>
<div class="page"><p/>
<p>The slightly different environments of the two sulphur atoms are shown by their deviations from the
</p>
<p>plane (0.00 and 0.05 Å) and by the differing S&ndash;C&ndash;C angles (135.7(3)� and 100.5(3)�).
The question of the existence of the two sulphur atoms as a dithione (VIII) or a dithiete (IX) has
</p>
<p>been clearly resolved by the geometry of the epidithio ring system, as shown by the data below:
</p>
<p>Bond lengths (Å) Bond angles (�)
</p>
<p>S(1)&ndash;S(2) &frac14; 2.119(1) C(5)&ndash;C(6)&ndash;C(7) &frac14; 123.8(4)
S(1)&ndash;C(7) &frac14; 1.767(4) C(6)&ndash;C(7)&ndash;C(8) &frac14; 122.1(3)
S(2)&ndash;C(6) &frac14; 1.782(4) C(5)&ndash;C(6)&ndash;S(2) &frac14; 135.7(3)
C(6)&ndash;C(7) &frac14; 1.387(5) C(6)&ndash;C(7)&ndash;S(1) &frac14; 103.3(3)
</p>
<p>C(6)&ndash;S(2)&ndash;S(1) &frac14; 80.3(3)
C(7)&ndash;C(6)&ndash;S(2) &frac14; 100.5(3)
C(8)&ndash;C(7)&ndash;S(1) &frac14; 134.5(3)
C(7)&ndash;S(1)&ndash;S(2) &frac14; 79.3(3)
</p>
<p>Additionally, the Raman spectrum showed a band at 486 cm�1, which corresponds to a &ndash;S&ndash;S&ndash;
</p>
<p>stretch [11]; the &gt;C&frac14;S frequency is in the range 1171&ndash;1180 cm�1.
</p>
<p>A theoretical study was carried out in connection with this unusual dithiete compound (I) [12, 13].
</p>
<p>CNDO/2 calculations were applied to a model consisting of ring B + S(1) and S(2), a hypothetical
</p>
<p>C6S2 molecule, for a range of C(7)&ndash;S(1), C(6)&ndash;S(2), C(7)&ndash;S(1)&ndash;S(2), and C(6)&ndash;S(2)&ndash;S(1) para-
</p>
<p>meters, with the following encouraging results for a minimum energy conformation:
</p>
<p>Calculated Experimental
</p>
<p>C(7)&ndash;S(1) 1.74 Å 1.767(4) Å
</p>
<p>C(6)&ndash;S(2) 1.76 Å 1.782(4) Å
</p>
<p>C(7)&ndash;S(1)&ndash;S(2) 100� 103.3(3)�
</p>
<p>C(6)&ndash;S(2)&ndash;S(1) 99� 100.5(3)�
</p>
<p>A compound C6H4S2, corresponding to the [C6S2] fragment, cannot be isolated, as it dimerizes
</p>
<p>through S&ndash;S bonding to the compound dibenzo[c,g]tetrathiocin, C12H8S4, which could exist in a rigid
</p>
<p>trans form (XI), or in a mobile form with a range of possible conformations: CNDO/2 calculations as
</p>
<p>a function of the dihedral angle between the two rings of the dimer showed the trans form to have the
</p>
<p>most stable (minimum energy) conformation.
</p>
<p>9.5 Discussion of the Structure 469</p>
<p/>
</div>
<div class="page"><p/>
<p>The cohesion of the S&ndash;S bond is most probably occasioned by pd-orbital overlap. This mechanism
</p>
<p>is not possible for the smaller oxygen species and o-quinones are well established as &gt;C&frac14;O species.
Figure 9.31 illustrates the molecular packing of the structure. The intermolecular S. . .S distances are
</p>
<p>approximately 10 Å, and there is no tendency to dimerization. Steric factors probably prevent a close
</p>
<p>approach of the two molecules.
</p>
<p>In the reactions of (IV) to (VII, I), stage (IV) to (V) gave the expected elimination of water and
</p>
<p>formation of the 11,12 double bond. Stage (V) to (VI) was accompanied by the loss of a &ndash;CH3 group
</p>
<p>and three other hydrogen atoms. The final stage (VI) to (VII, I) was quantitative, with loss of
</p>
<p>ethylene. The probable mechanism is a disruption at the spiro-C(7) carbon atom, followed by
</p>
<p>migration of a sulphur atom to C(6) and subsequent aromatization of ring B. The aromatization
</p>
<p>provides a conformational driving force for the elimination of the methyl group at C(19), the methyl
</p>
<p>group that is attached to C(10) in II. Under u.v. activation, the structure is further stabilized by the
</p>
<p>loss of ethylene from the dithian ring.
</p>
<p>9.6 Some Remarks on X-Ray Structure Determination
</p>
<p>No description of the process of X-ray crystal structure analysis can be as complete or as satisfying as
</p>
<p>a practical involvement with the subject. In teaching crystallography and structural chemistry,
</p>
<p>projects that include crystal structure determinations have become increasingly important. However,
</p>
<p>in order to attempt to replicate the practical side of structure analysis, insofar as is possible in isolation
</p>
<p>from the laboratory, problems on this topic additional to those at the end of this chapter are given in
</p>
<p>Chap. 13; they involve the program system XRAY and the several sets of data that accompany it.
</p>
<p>They have been designed to give practice, albeit in two dimensions, with the fundamental
</p>
<p>techniques of solving crystal structures, given the crystal and reflection data. The programs can be
</p>
<p>executed on any IBM-type PC. We encourage the reader strongly to tackle these problems and so
</p>
<p>Fig. 9.31 Stereoview of the packing of the title compound (I), as seen along &frac12;021�
</p>
<p>470 9 Examples of Crystal Structure Determination</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_13">http://dx.doi.org/10.1007/978-1-4614-3954-7_13</a></div>
</div>
<div class="page"><p/>
<p>engage in a practical way with the techniques of crystal structure analysis. We refer the reader also to
</p>
<p>Appendix D, wherein are listed the many program systems now freely available to the structural
</p>
<p>crystallographer.
</p>
<p>9.7 Biomolecular Modeling: Bioinformatics
</p>
<p>We have seen that the technique of X-ray crystallography is a powerful means for the determination
</p>
<p>of accurate crystal structures, and that this in turn provides a source of information on molecular
</p>
<p>geometry in the form of bond lengths, bond angles, conformational parameters, planarity, and
</p>
<p>intermolecular interactions. There are many reasons that prompt such studies. For example, we
</p>
<p>may wish to design new industrial materials, a process which requires access to such experimental
</p>
<p>results. We may wish to design a new drug molecule with modified properties, such as increased or
</p>
<p>decreased potency, improved specificity, or lacking unwanted side effects. Again, detailed knowledge
</p>
<p>of the molecular properties of the starting material, including its molecular geometry and flexibility as
</p>
<p>well as those of the biological target are a necessary prerequisite for such research. One approach
</p>
<p>where the crystallographic data can be applied effectively for such studies involves the use of
</p>
<p>intensive computational techniques in which structural data from one or more starting materials are
</p>
<p>employed. Obviously, there are many parameters and variables involved in this type of work and we
</p>
<p>have selected an example which demonstrates many of the techniques employed.
</p>
<p>We introduce in this section, as an important by-product of X-ray structure determination, the
</p>
<p>technique known as biomolecular modeling. The method described here depends on the availability
</p>
<p>of a known macromolecular structure, usually a protein or a nucleic acid, such as DNA or RNA, and
</p>
<p>another usually much smaller molecule or ligand that is known to bind either chemically or more
</p>
<p>usually to interact non-covalently with the macromolecule. It is usually an advantage, when embark-
</p>
<p>ing on studies of this kind, for both structures to be known from previous X-ray or neutron diffraction
</p>
<p>analysis. Frequently, it is possible to determine the structure of the macromolecule-ligand assembly
</p>
<p>as such through a further diffraction study, as discussed later in Chaps. 10 and 11. However, when this
</p>
<p>is not possible or practicable, biomolecular modeling may provide plausible answers about the mode
</p>
<p>and location of the interactions between the two molecules.
</p>
<p>9.8 Docking Oligomycin into ATP Synthase: Ligand and Receptor
</p>
<p>The macrocyclic naturally occurring antibiotic oligomycin (Fig. 9.32) binds to and thereby inhibits
</p>
<p>the biological activity of the multi-subunit protein enzyme ATP synthase (ATPase). The ATP
</p>
<p>synthase complex is constructed of numerous protein subunits that work together (Fig. 9.33). It is
</p>
<p>of interest to locate the site of interaction between the ligand oligomycin and the receptor ATPase.
</p>
<p>The detailed X-ray structures of the three oligomycin structures [14] A, B, and C described below are
</p>
<p>available for modeling studies designed to locate and characterize this site of interaction.
</p>
<p>9.8.1 Why Modeling Studies?
</p>
<p>Ideally the best way to investigate ligand/receptor binding would be to carry out a new X-ray or
</p>
<p>neutron diffraction study on the complex of the two components. However, the protein (ATPase) is so
</p>
<p>large and complex that it has not been possible to determine the complete structure as such. Instead, a
</p>
<p>possible structure has been assembled from several independent studies, Sect. 9.10. Thus, it has been
</p>
<p>9.8 Docking Oligomycin into ATP Synthase: Ligand and Receptor 471</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_10">http://dx.doi.org/10.1007/978-1-4614-3954-7_10</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_11">http://dx.doi.org/10.1007/978-1-4614-3954-7_11</a></div>
</div>
<div class="page"><p/>
<p>necessary to model the binding of oligomycin to ATPase, at least until a better way is devised for
</p>
<p>studying the structure of the whole complex by diffraction methods.
</p>
<p>9.9 X-Ray Structures and Absolute Configurations of the Antibiotics
Oligomycins A, B, and C: Inhibitors of ATP Synthase
</p>
<p>Oligomycin is a naturally occurring antibiotic that occurs in the gram-positive bacterium Streptomy-
</p>
<p>ces diastatochromogenes. Knowledge of the site on the FO (O &frac14; oligomycin) domain (Fig. 9.33)
where oligomycin binds to ATPase may enable the development of new improved ATPase inhibitors
</p>
<p>that could be produced synthetically. As the structure of oligomycin (Fig. 9.32) contains a 26-
</p>
<p>membered ring, and therefore has numerous conformational possibilities, a detailed X-ray structure
</p>
<p>Fig. 9.32 Chemical structures of oligomycins A, B, and C. The crystallographic numbering scheme is shown and
major features of the structures as discussed in the text are labeled
</p>
<p>472 9 Examples of Crystal Structure Determination</p>
<p/>
</div>
<div class="page"><p/>
<p>is an essential prerequisite for the modeling studies that are aimed at locating and characterizing the
</p>
<p>FO&ndash;ATPase binding site. In fact, recently published X-ray structures [14] of the three common forms,
</p>
<p>of oligomycin, A, B, and C, including their absolute configurations, are available for these modeling
</p>
<p>studies. The X-ray studies reveal regions with differences in the three-dimensional structures and
</p>
<p>hydrogen-bonding propensity between the oligomycins, which may be associated with their potential
</p>
<p>to bind to ATPase. The experimentally determined X-ray structures of the three forms of oligomycin
</p>
<p>were used as the starting point for docking studies.
</p>
<p>9.9.1 Summary
</p>
<p>All three oligomycins crystallized in space group P212121 with four molecules per unit cell.
</p>
<p>Oligomycin A crystallized as the methanol solvate C45H72O11	CH3OH with unit-cell parameters
a &frac14; 10.476(3) Å, b &frac14; 17.342(1) Å, c &frac14; 26.825(5) Å; oligomycin B as an acetic acid solvate
C45H71O12	CH3CO2 with unit-cell parameters a &frac14; 10.351(3) Å, b &frac14; 17.305(1) Å, c &frac14; 26.929
(5) Å; and oligomycin C, C45H74O10, with unit-cell parameters a &frac14; 10.385(2) Å, b &frac14; 11.9510
(9) Å, c &frac14; 38.007(4) Å. Oligomycin A refined with final R indices [I &gt; 2s(I)], R1 &frac14; 0.0734,
wR2 &frac14; 0.1940; R (all data): R1 &frac14; 0.1106, wR2 &frac14; 0.2100, and absolute structure parameter &frac14; �0.7
(4); for oligomycin B, final R indices [I &gt; 2s(I)] are R1 &frac14; 0.0479, wR2 &frac14; 0.1388; R (all data):
R1 &frac14; 0.0581, wR2 &frac14; 0.1435, and absolute structure parameter &frac14; �0.2(2); and for oligomycin C,
final R indices [I &gt; 2s(I)] are R1 &frac14; 0.0454, wR2 &frac14; 0.1130; R (all data): R1 &frac14; 0.1061,
wR2 &frac14; 0.1221, and absolute structure parameter &frac14; 0.1(3).
</p>
<p>Fig. 9.33 Subunit
arrangement for
</p>
<p>Escherichia coli ATP
</p>
<p>synthase
</p>
<p>9.9 X-Ray Structures and Absolute Configurations of the Antibiotics Oligomycins. . . 473</p>
<p/>
</div>
<div class="page"><p/>
<p>The present study has provided:
</p>
<p>1. Corrections to the previously published chemical structures of the oligomycins
</p>
<p>2. Full descriptions of the absolute configurations
</p>
<p>3. Information on regions of the structures with minor but important differences in their three-
</p>
<p>dimensional structures that may create differences between the oligomycins in their potential to
</p>
<p>bind to sites on the ATP synthase molecule. These results are all of major importance for future
</p>
<p>studies designed to establish details of the actual binding of oligomycins to ATP synthase.
</p>
<p>9.9.2 Background
</p>
<p>Oligomycin is an antibiotic that occurs in the Gram-positive bacterium S. diastatochromogenes. It
</p>
<p>is known [15] to act as a potent inhibitor of oxidative phosphorylation, and as a specific inhibitor
</p>
<p>acting at the site of coupling between respiration and phosphorylation. Oligomycin is used
</p>
<p>extensively in experiments designed, for example, to investigate the mechanisms of processes in
</p>
<p>mitochondria [16] and studies involving oligomycin-sensitive ATP synthase [17, 18].
</p>
<p>The three distinct forms or types studied here, known as Oligomycin A, B, and C, differ from
</p>
<p>each other chemically only in one or two substituents, as discussed below. Experiments using
</p>
<p>oligomycin as a probe do not usually attempt to distinguish between possible differences between
</p>
<p>the effects of using a particular type of oligomycin, and often use the unrefined mixture of all three.
</p>
<p>Enzymes known as ATP synthases are complex multi-subunited protein structures that can synthe-
</p>
<p>size adenosine triphosphate (ATP) from adenosine diphosphate (ADP) and inorganic phosphate. In
</p>
<p>mitochondria, the ATP synthase molecule can be visualized as having a major domain, F1, outside
</p>
<p>the cell membrane, and a minor domain, FO, embedded within the membrane. FO derives its name
</p>
<p>from being the oligomycin binding domain, and is also known as OSCP (the oligomycin sensitivity
</p>
<p>conferral protein). The F1 domain, diameter 9 nm, can be seen in the transmission electron
</p>
<p>microscope by negative staining, first observed, at least partially, by Fernandez-Moran [19]. The
</p>
<p>antibiotic oligomycin binds to and inhibits ATP synthase. The crystal structure of the F1 catalytic-
</p>
<p>domain of ATP synthase is known [20] and is largely consistent with Boyer&rsquo;s rotary-catalysis model
</p>
<p>[21].
</p>
<p>9.9.3 Experimental
</p>
<p>Samples of Oligomycin A, B, and C were purchased from Reanal Finechemical company. Small
</p>
<p>seed crystals were produced by controlled evaporation of solutions at room temperature over
</p>
<p>several weeks in (a) 50/50 acetone and acetic acid (for A); (b) 50/50 acetone/ethanol (for B); and
</p>
<p>acetone only (for C). Recrystallization at 4�C was induced by seeding and produced X-ray quality
crystals after 20 days.
</p>
<p>The forms A and B are almost colorless and prismatic or platy crystals of typical dimensions 0.25,
</p>
<p>0.15, 0.15 mm. Form C crystals are also colorless but more plate-like with typical dimension 0.3,
</p>
<p>0.25, 0.10 mm. In all cases, crystal quality was tested by means of Mar IP images taken in-house using
</p>
<p>copper X-radiation prior to data collection.
</p>
<p>Crystals were mounted for room temperature data collection on glass fibers, using epoxy resin
</p>
<p>(araldite), and having dimensions 0.20, 0.15, 0.15 mm3 (A), 0.20, 0.20, 0.15 mm (B), and 0.3, 0.25,
</p>
<p>0.10 mm (C) and then mounted for X-ray intensity measurement on an Enraf-Nonius CAD-4
</p>
<p>automated 4-circle diffractometer equipped with a graphite monochromator for room temperature
</p>
<p>474 9 Examples of Crystal Structure Determination</p>
<p/>
</div>
<div class="page"><p/>
<p>recording of the diffraction pattern. Intensity data were recorded using monochromatic Cu Ka
</p>
<p>radiation; CAD-4 Express Software [22] was used for cell determination and refinement and data
</p>
<p>reduction. In all cases, accurate cell parameters were determined from 25 reflections (25 &lt; y &lt; 28�).
For data collection, o � 2y scans were used under computer control. The intensities 4680 (A),
5266 (B), and 5498 (C) of reflections were measured for y &lt; 70�. The crystals showed no significant
variations in the intensities of the three standard reflections during the course of data collection.
</p>
<p>Lorenz and polarization corrections were applied and a semi-empirical correction for absorption
</p>
<p>[23] was made for each set of data. All three oligomycins were found to have space group P212121,
</p>
<p>the A and B unit cells being very similar but bearing no relationship to that of C.
</p>
<p>9.9.4 Structure Determination and Refinement
</p>
<p>The oligomycin structures were solved using SHELXS-86 [24] and refined with SHELXL-97 [12].
</p>
<p>Non-hydrogen atoms were refined anisotropically by full-matrix least-squares techniques. Apart from
</p>
<p>the exceptions discussed below, hydrogen-atom positions were calculated geometrically and refined
</p>
<p>in riding mode with isotropic displacement parameters fixed by the program and common for all
</p>
<p>hydrogen atoms attached to the same carbon atom. In the riding mode, hydrogen atoms are first
</p>
<p>positioned according to standard molecular geometry. Then, in subsequent refinements, these atoms
</p>
<p>are adjusted in such a way as to preserve the chosen geometry, and the isotropic temperature factors
</p>
<p>Uiso of these atoms can be refined in the usual manner. A very high value of Uiso may indicate a
</p>
<p>misplaced hydrogen atom, and the difference-Fourier map should be carefully examined. Calcula-
</p>
<p>tions were made with SHELXL-97 [25] as implemented in WinGX [26] and geometrical calculations
</p>
<p>were made with the programs PARST and PLATON [27] as implemented in WinGX.
</p>
<p>9.9.5 Results
</p>
<p>Figure 9.32 summarizes the chemical constitutions of the three forms as determined in these structure
</p>
<p>analyses and the atom numbering used: ORTEP [28]/Raster3D [29] generated corresponding views of
</p>
<p>the three molecular conformations which are shown in Fig. 9.34a&ndash;c. Figure 9.35a&ndash;c are surface
</p>
<p>representations prepared with the program VMD [30].
</p>
<p>9.9.6 Discussion
</p>
<p>General
Apart from the exceptions discussed below, corresponding bond lengths and bond angles between the
</p>
<p>three forms of oligomycin are in good agreement with one another and the values are within the
</p>
<p>expected ranges for this type of molecule (see Tables 8.21 and 8.22).
</p>
<p>Side Chain R4
According to the literature available [31, 32] when the X-ray analyses presented here were initiated,
</p>
<p>the R4 side chains (Fig. 9.32) in all three forms were designated as being propanone (P3, Fig. 9.32).
</p>
<p>As the analyses progressed it soon became clear that this was not the correct assignment for any of
</p>
<p>the forms.
</p>
<p>9.9 X-Ray Structures and Absolute Configurations of the Antibiotics Oligomycins. . . 475</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_8#Tab21_8">http://dx.doi.org/10.1007/978-1-4614-3954-7_8#Tab21_8</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_8#Tab22_8">http://dx.doi.org/10.1007/978-1-4614-3954-7_8#Tab22_8</a></div>
</div>
<div class="page"><p/>
<p>Oligomycin A
In the refined Oligomycin A structure, the geometry for side chain R4 is as follows:
</p>
<p>C(1)&ndash;C(2) &frac14; 1.536(10) Å, O(2)&ndash;C(2) &frac14; 1.487(9) Å, C(2)&ndash;C(3) &frac14; 1.437(8) Å, C(3)&ndash;C(4) &frac14; 1.516
(7) Å; C(1)&ndash;C(2)&ndash;C(3) &frac14; 111.0(6)�, O(2)&ndash;C(2)&ndash;C(3) &frac14; 104.0(5)�, C(2)&ndash;C(3)&ndash;C(4) &frac14; 123. 1(5)�.
The bond angle C(2)&ndash;C(3)&ndash;C(4) is aromatic rather than tetrahedral, as is the associated short C
</p>
<p>(2)&ndash;C(3) bond length. In difference electron density maps prior to assignment of hydrogen-atom
</p>
<p>positions, there was only one significant peak close to where hydrogen atoms were expected to be
</p>
<p>attached to C(3). Several different models involving the geometrical assignment of hydrogen atoms
</p>
<p>on C(3) were set up. Only one hydrogen atom, H(3A), was left with an acceptably small temperature
</p>
<p>factor, Uiso &frac14; 0.2054 Å2, after refinement of a starting model in which two tetrahedrally positioned
hydrogen atoms on C(3) were both refined with respect to their positions and isotropic thermal
</p>
<p>parameters. In the very final model, for consistency with the overall model, H(3) was geometrically
</p>
<p>fixed using the aromatic option and tied to C(3), H(3) being close to the former refined H(3A)
</p>
<p>position. These results are consistent with the R4 side chain in Oligomycin A as being propenol
</p>
<p>(R4 &frac14; P1, Fig. 9.32), not propanone (P3) as was previously thought [31]. In view of the unusual
geometry revealed in this analysis for side chain R4, several complete intensity data sets were
</p>
<p>measured for Oligomycin A using fresh crystals; the results were always consistent with those
</p>
<p>reported above.
</p>
<p>Fig. 9.34 Molecular
conformations of
</p>
<p>Oligomycins A, B, and C
</p>
<p>respectively. The view
</p>
<p>selected displays the most
</p>
<p>open projection of the
</p>
<p>molecule (ORTEP/Raster)
</p>
<p>476 9 Examples of Crystal Structure Determination</p>
<p/>
</div>
<div class="page"><p/>
<p>Oligomycins B and C
The R4 side chains for both Oligomycins B and C were also quite clearly not propanone (P3) but were
</p>
<p>consistent with P2 (Fig. 9.32), that is, with propanol. Both side chains refined with perfectly normal
</p>
<p>geometry and well-behaved thermal displacement parameters including H(3A) and H(3B) in both
</p>
<p>structures.
</p>
<p>Oligomycins A and B have very similar unit cells and belong to the same space group. Many of the
</p>
<p>atoms are in similar positions in the two structures. Chemical differences can be noted from the
</p>
<p>formulae (Fig. 9.32). Those noted in the R4 side chains in the above discussion were quite unexpected
</p>
<p>andmay be important for future activity studies.As discussed below, theR4 side chain inOligomycinA
</p>
<p>differs in conformation from the corresponding side chains in Oligomycins B and C and forms an
</p>
<p>internal hydrogen bond between O(2) and O(7) in pyranose ring X (Figs. 9.32 and 9.34a). The correct
</p>
<p>assignment of the chemical features of any biologically active molecule is of course an essential
</p>
<p>prerequisite to an understanding of its function and mode of action.
</p>
<p>Fig. 9.35 Surface
representations of
</p>
<p>Oligomycins A, B, and C
</p>
<p>(VMD SURFACE).
</p>
<p>Red &frac14; oxygen,
blue &frac14; carbon,
white &frac14; hydrogen.
(a) Oligomycin A: OH(25)
forms an intramolecular
</p>
<p>hydrogen bond with O(24)
</p>
<p>which could exclude both
</p>
<p>from receptor binding.
</p>
<p>(b) Oligomycin B: there are
five active groups in this
</p>
<p>surface view of oligomycin
</p>
<p>B, compared to two in
</p>
<p>oligomycin A and zero in
</p>
<p>oligomycin C. This
</p>
<p>suggests that oligomycin B
</p>
<p>has a greater potential for
</p>
<p>receptor binding than A or
</p>
<p>C. (c) Oligomycin C: this
shows three potential
</p>
<p>receptor binding sites. OH
</p>
<p>(24) appears to be partially
</p>
<p>buried but does in fact
</p>
<p>participate in an
</p>
<p>intermolecular hydrogen
</p>
<p>bond. In addition OH(2) is
</p>
<p>quite exposed in the
</p>
<p>structure but out of this
</p>
<p>view
</p>
<p>9.9 X-Ray Structures and Absolute Configurations of the Antibiotics Oligomycins. . . 477</p>
<p/>
</div>
<div class="page"><p/>
<p>Pyranose Rings X and Y
Both of the spiro linked pyranose rings X and Y (Figs. 9.32 and 9.34) are distorted chairs in all three
</p>
<p>forms of oligomycin, with corresponding rings having the same type of chair conformation in all three
</p>
<p>structures. The ring torsion angles range from about �45 to �65� in both rings, and corresponding
torsion angles differ by as much as 9� but usually agree within 4�. The eight spiro linkage torsion
angles are either about �170 or �70�, with similar variations between Oligomycins as exhibited by
the rings.
</p>
<p>9.9.7 Conformational Variations in the Macrocyclic Structures
</p>
<p>Figure 9.32 shows the common macrocyclic structure of the oligomycins. The 26 atom chain runs
</p>
<p>from C(11) to the ether oxygen O(37) which is the only non-carbon atom in the chain. Close
</p>
<p>inspection of Fig. 9.34a&ndash;c reveals small differences in the overall macrocycle ring conformations.
</p>
<p>Structures A and B are generally more similar in conformation to each other than to structure C which
</p>
<p>has differences in torsion angle of up to 18� as discussed below.
</p>
<p>Chain 1: C(36) to C(17) is Predominantly Fully Extended
The chain stretching from O(37) to C(17) and including side chain atom C(171) is fully extended in all
</p>
<p>three structures (torsion angles ca. �180�). The ether oxygen O(37) is associated with a pronounced
kink in the chain (torsion angle about bond O(37)&ndash;C(11) is ca. �80�) in all structures.
</p>
<p>Chain 2: C(17) to C(23)
This chain is also in an extended conformation but is slightly convex due to opposite sense twists at
</p>
<p>either end.
</p>
<p>Chain 3: C(21) to C(34)
This long stretch forms an irregular but gradual clockwise coil and is the most conformationally
</p>
<p>variable part of the structure. Correspondence between the chain torsion angles in the three oligo-
</p>
<p>mycins in this region is generally the most variable with differences as great as 17� about C(24)&ndash;C(25)
between Oligomycin A and C, and differences frequently around 10� elsewhere. Conformational
variability in this region can be seen from close examination of Fig. 9.33a&ndash;c.
</p>
<p>Chain 4: C(33) to C(36)
This short stretch of chain is fully extended in all three structures.
</p>
<p>Bends in the Macrocyclic Chains
</p>
<p>1. The bend between chains 1 and 4:
</p>
<p>This bend is created by a cis twist about bond C(35)&ndash;C(36) of about �3�.
2. The bend between chains 1 and 2:
</p>
<p>This bend is associated with changes in the torsion angles about the bonds C(16)&ndash;C(17) and
</p>
<p>C(17)&ndash;C(18) from the extended mode on either side (ca. 180�) to values of approximately�60 and
100� in all three structures.
</p>
<p>3. The bend between chains 2 and 3:
</p>
<p>This bend similarly involves transitions in two torsion angles, namely about C(21)&ndash;C(22)
</p>
<p>and C(22)&ndash;C(23) from the extended mode, their values being approximately �130 and �60�
respectively.
</p>
<p>478 9 Examples of Crystal Structure Determination</p>
<p/>
</div>
<div class="page"><p/>
<p>4. The bend between chains 3 and 4:
</p>
<p>This bend also involves two torsion angles, namely, C(32)&ndash;C(33) and C(33)&ndash;C(34) changing from
</p>
<p>extended mode on either side. C(32)&ndash;C(33) has values around 55� in all three structures whereas
C(33)&ndash;C(34) is highly variable with values of 134.3, 135.7, and 117.7� in Oligomycins A, B, and C
respectively.
</p>
<p>Side Group Orientations
</p>
<p>Close inspection of Fig. 9.33a&ndash;c reveals small differences in some side group orientations resulting
</p>
<p>from variations in the ring conformation discussed above. These differences may be important factors
</p>
<p>in determining the binding characteristics essential for understanding the biological activities of the
</p>
<p>oligomycins. In the following discussion the term &ldquo;external&rdquo; applies to groups facing in a general
</p>
<p>direction outwards, that is, away from the macrocyclic ring and &ldquo;internal&rdquo; facing groups are pointing.
</p>
<p>CH3 Groups
</p>
<p>In Oligomycin A, B, and C, two CH3 groups are internal, that is, those on C(12) and C(27) (Fig. 9.32).
</p>
<p>All other CH3 groups are external.
</p>
<p>OH Groups
</p>
<p>In Oligomycin A, the five OH groups are all external and four of them participate in hydrogen bonds,
</p>
<p>namely, O(2)H. . .O(7) (in ring X, see Fig. 9.32), stabilizing the R4 side chain; O(25)H. . .O(24)H,
</p>
<p>stabilizing bend 3 (see above); and O(32)H. . .O(30), both stabilizing bend 4 (see above). Group O
</p>
<p>(28)H forms an intermolecular hydrogen bond with O(36) (see below).
</p>
<p>In Oligomycins B and C, the situation is quite different, possibly due in part to the lack of the R25OH
</p>
<p>group (Fig. 9.32). The four OH groups are all external but with O(32)H. . .O(30), stabilizing bend 4,
</p>
<p>being the only intramolecular hydrogen bond formed. This provides Oligomycins B and C with more
</p>
<p>opportunities for hydrogen-bonding to protein receptors, and may explain differences which could exist
</p>
<p>in their binding characteristics compared to Oligomycin A. Owing to the presence of three more
</p>
<p>intramolecular hydrogen bonds, Oligomycin A may be deemed to be a much more stable conformation
</p>
<p>than Oligomycin B or C.
</p>
<p>&frac14;O Groups
In Oligomycin A, B, and C, &frac14;O(26) is internal, and &frac14;O(30) and &frac14;O(36) are external. Oligomycin B
has an additional&frac14;O(7) on pyranose ring X and this side group may be described as external and does
in fact participate in intermolecular hydrogen bonds.
</p>
<p>Summary of the Hydrogen Bonds in Oligomycins A, B, and C
Only one intermolecular hydrogen bond, between OH(28) and O(36), is a consistent feature of these
</p>
<p>structures. It may be concluded therefore that both OH(28) and &frac14;O(36) are highly exposed on the
molecular surface: OH(2) is also exposed in Oligomycins B and C forming hydrogen bonds either to
</p>
<p>solvent, in B, or to symmetry-related molecules in C. In Oligomycin A, OH(2) forms an intramolec-
</p>
<p>ular hydrogen bond to O(9) on the pyranose ring X: OH(24) is also exposed forming hydrogen bonds
</p>
<p>to solvate molecules in A and B and to symmetry-related OH(2) groups in structure C. The OH(32)
</p>
<p>group forms consistently a stabilizing intramolecular hydrogen bond with &frac14;O(30), and both oxygen
atoms can be discounted from binding studies. Apart from the intramolecular hydrogen bond
</p>
<p>between O(2) and O(9) in Oligomycin A, the pyranose oxygen atoms O(9) and O(14) do not
</p>
<p>participate in hydrogen-bonding interactions and can be taken to be inaccessible, as can &frac14;O(26)
and &ndash;O(37)&ndash;. It may therefore be concluded from these studies that the key groups which are likely to
</p>
<p>be involved in intermolecular interactions with protein subunits are: OH(2) (probably only in
</p>
<p>Oligomycins B and C), OH(24), OH(28), and O(36). The conclusions about OH(28) and &frac14;O(36)
</p>
<p>9.9 X-Ray Structures and Absolute Configurations of the Antibiotics Oligomycins. . . 479</p>
<p/>
</div>
<div class="page"><p/>
<p>are clearly borne out by the VMD [22] SURFACE plots of the three molecules shown in Fig. 9.35.
</p>
<p>Other features and differences of the molecular structures discussed above can be clearly seen by
</p>
<p>careful inspection of these plots.
</p>
<p>Absolute Configuration of Oligomycins A, B, and C
Oligomycins A, B, and C all include a number of asymmetric centers which, for the purposes of
</p>
<p>modeling studies, need to be defined in the correct absolute configuration. The X-ray studies reported
</p>
<p>in the present article have in all three cases been of sufficiently high quality to enable the correct
</p>
<p>assignment of configuration in all three cases. The refined Flack [33] parameter, Sect. 7.6.1, and its
</p>
<p>standard deviations for the three structures are: �0.7(4), �0.2(2), and 0.1(3), respectively. These
three values are all equal to 0.0 within 3 standard deviations or better and it may be concluded that
</p>
<p>each structure has been assigned the correct absolute configuration, which can be seen by inspection
</p>
<p>to be the same in each case. All figures depicting the oligomycin structures show them in their correct
</p>
<p>absolute configurations.
</p>
<p>Conclusions
</p>
<p>While Oligomycins A, B, and C are chemically very similar, and A and B are crystallographically
</p>
<p>pseudo-isomorphous, the small differences that do exist between them have significant effects on how
</p>
<p>they take part in intermolecular interactions. This information is important for the assessment of the
</p>
<p>receptor binding potential of oligomycin to sites in the ATP synthase system and, in particular, when
</p>
<p>such assessments are being carried out by graphics-aided computer simulations.
</p>
<p>It has been shown that in the case of Oligomycin A two surface-located sites are available for
</p>
<p>intermolecular binding, while there are four in Oligomycin B and three in Oligomycin C. They are
</p>
<p>highlighted in Table 9.1, counting interactions with solvent as possible intermolecular binding sites.
</p>
<p>This result indicates that Oligomycin B, with an extra oxygen atom, O(7), available on the surface,
</p>
<p>may be better favored than the A or C form for taking part in interactions with protein subunits.
</p>
<p>9.10 Structure of ATP Synthase (ATPase): The Receptor
</p>
<p>The three-dimensional structure of ATPase shown schematically in Fig. 9.33 has been constructed
</p>
<p>from the following components:
</p>
<p>1. The X-ray structure of the catalytic-domain F1 [20]
</p>
<p>2. Collected information about domain FO from biochemical studies
</p>
<p>The crystal structure of F1 is largely consistent with Boyer&rsquo;s [22] rotary-catalysis model. The
</p>
<p>absence of an X-ray structure for domain FO provides the motive for carrying out modeling studies of
</p>
<p>the oligomycin binding. ATPase is a large subunited protein complex found in the mitochondria of
</p>
<p>cells, and its function is to extract energy via the conversion of ADP into adenosine triphosphate
</p>
<p>(ATP). The general reaction scheme is written as:
</p>
<p>ADP&thorn; Phosphate &ETH; ATP
</p>
<p>The mechanism by which this conversion takes place is unique as it employs a biological rotor
</p>
<p>driven by the passage of protons through a membrane. Various studies have shown that the ATPase
</p>
<p>complex is constructed of numerous protein subunits that work together. Figure 9.33 shows schema-
</p>
<p>tically the various protein subunits (a, b, c, a, b, g and d) and how they are arranged to form the
</p>
<p>480 9 Examples of Crystal Structure Determination</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_7#Sec32_7">http://dx.doi.org/10.1007/978-1-4614-3954-7_7#Sec32_7</a></div>
</div>
<div class="page"><p/>
<p>ATPase complex. The ATPase complex straddles a cell membrane, which is itself a complex protein
</p>
<p>assembly. Further details of the ATPase structure are given below.
</p>
<p>The F1 Domain
The a and b subunits of ATPase (Fig. 9.33) contain the catalytic sites that carry out the reaction
</p>
<p>between ADP and ATP; they are arranged cyclically in an ababab assembly. In the center of this
</p>
<p>construct is the &ldquo;stalk&rdquo; which consists of the g subunit with, in the case of mitochondrial ATP
</p>
<p>synthase, d and e subunits at the &ldquo;foot,&rdquo; the arrangement being slightly different in the case of
</p>
<p>Escherichia coli ATP synthase. This foot is in contact with the c-ring of protein subunits which is
</p>
<p>made up of 9&ndash;12 identical c-subunits. The c-ring and central stalk rotate within the ababab assembly
</p>
<p>in order to cause the conformational changes required for catalytic activity. This rotation is a result of
</p>
<p>proton transport facilitated by the a-subunit that allows access to a proton (H+) which binds to a
</p>
<p>hydrophilic channel between the c-ring and the a-subunit, the ring moves around exposing an empty
</p>
<p>site for the next proton. The bound proton stays attached to the c-ring until it has traveled nearly full
</p>
<p>circle back to the a-subunit where it enters a second hydrophilic channel that opens inside, whereupon
</p>
<p>it is released leaving an empty site for the next proton. The top of the structure (ab and g complex) is
</p>
<p>known as the F1 domain and is external to the cell membrane. The crystal structure of the F1
</p>
<p>catalytic-domain of ATP synthase is known [20] and is largely consistent with Boyer&rsquo;s [34] rotary-
</p>
<p>catalysis model.
</p>
<p>The FO Domain
The c-ring, the a-unit, and other transmembrane protein units (Fig. 9.33) are known as the FO domain
</p>
<p>and is a cell membrane embedded component of ATPase. The FO domain is thought to be where the
</p>
<p>antibiotic oligomycin binds, thereby inhibiting the production of ATP. There are numerous examples of
</p>
<p>mutational studies that reveal a possible binding site for oligomycin at the interface between the a-
</p>
<p>subunit and the c-ring [35, 36].
</p>
<p>9.11 Docking Oligomycin into ATPase
</p>
<p>The study reported here locates and characterizes the oligomycin binding mode and site within the
</p>
<p>FO subunit of an homology model of the ATP synthase FO subunit from E. coli, employing the three
</p>
<p>oligomycin molecular structures independently.
</p>
<p>9.11.1 ATP Synthase FO Model
</p>
<p>Inspection of the European Bioinformatics Institute (EBI) database revealed a model [37] of the
</p>
<p>FO structure derived from NMR measurements. This is a solution study as opposed to a single crystal
</p>
<p>study. The PDB entry code for this structure is 1c17. However, this structure does not represent the
</p>
<p>complete FO complex as it has omitted the vital section with residues numbered 167&ndash;198 in the a-
</p>
<p>subunit. Since a complete model of the a-subunit is needed for oligomycin binding studies, it was
</p>
<p>necessary to first build this missing section into the structure, and for this purpose the full sequence of
</p>
<p>the a-subunit was submitted to the SWISSMODEL (now INTERPRO) fully automated protein
</p>
<p>structure homology-modeling facility. This server is accessible either via the ExPASy web server,
</p>
<p>or from the program DeepView (Swiss PDB-Viewer) [38, 39]. The resulting model included the
</p>
<p>stretch missing in 1c17.
</p>
<p>9.11 Docking Oligomycin into ATPase 481</p>
<p/>
</div>
<div class="page"><p/>
<p>9.11.2 Homology Modeling
</p>
<p>The homology modeling procedure consists of four sequential steps:
</p>
<p>1. Template selection
</p>
<p>2. Target-template alignment
</p>
<p>3. Model construction
</p>
<p>4. Model assessment
</p>
<p>Template selection and target-template alignment may be carried out together. The target is the
</p>
<p>unknown structure being investigated and the template is a sought-for structure with a highly similar
</p>
<p>amino acid sequence (homology) which it is hoped to find in the Protein Data Base. Usually a number
</p>
<p>of possible templates will be found. The identification of templates relies on the production of
</p>
<p>sequence alignments.
</p>
<p>The simplest method of template identification relies on serial pairwise sequence alignments aided
</p>
<p>by database search techniques. Choosing the best template from among the candidates is a key step,
</p>
<p>and can affect the final accuracy of the structure significantly. Sometimes several homology models
</p>
<p>are produced for a single query sequence, with the most likely candidate chosen only in the final step.
</p>
<p>9.11.3 Refining the Model: Energy Minimization
</p>
<p>The model provided by SWISSMODEL [40] was superimposed onto the coordinates of the
</p>
<p>corresponding subunits in the PDB structure 1c17 using the program MOE [41] and an energy
</p>
<p>minimization of the three chains involved in ligand binding was then carried out.
</p>
<p>Energy Minimization
Molecules are not static structures, and many biological processes are mediated by molecular
</p>
<p>movement. However, a static molecular model, where the atoms are located at an energy minimum,
</p>
<p>is a useful simplification; ideally this energy minimum should be the global minimum for the given
</p>
<p>collection of atoms.
</p>
<p>The interactions between neighboring atoms include spring-like forces (representing chemical
</p>
<p>bonds), and steric and van der Waals forces between nonbonded atoms. The Lennard-Jones potential
</p>
<p>is commonly used to describe nonbonded interactions, and electrostatic interactions are computed
</p>
<p>from Coulomb&rsquo;s law. The complete potential energy function E for a biomolecule also includes terms
</p>
<p>that constrain the deviation of bond lengths, bond angles, and torsion angles from their equilibrium
</p>
<p>values, in addition to the nonbonded interactions. Thus, we write
</p>
<p>E &frac14; Ebonds &thorn; Eangle &thorn; Etorsion &thorn; Enon�bonded &thorn; Eelectrostatic
</p>
<p>The set of parameters characterizing the potential function consists of equilibrium bond lengths,
</p>
<p>angles, partial charges, force constants, and van der Waals parameters, collectively known as a force
</p>
<p>field. Different implementations of molecular mechanics use differing mathematical expressions and
</p>
<p>different parameters for the potential function. The force fields in use today have been developed by a
</p>
<p>combination of high-level quantum-mechanical calculations and fitting to experimental data.
</p>
<p>Given a model for the potential energy of the asymmetric unit, an equilibrium (zero force) set of
</p>
<p>atomic positions can be found by energy minimization. The method of conjugate gradients is suitable
</p>
<p>for finding the local energy minimum. Starting with an initial intelligent guess for the atomic
</p>
<p>positions, the method of conjugate gradients will minimize the residual forces until local equilibrium
</p>
<p>482 9 Examples of Crystal Structure Determination</p>
<p/>
</div>
<div class="page"><p/>
<p>is achieved. However, this is not necessarily the minimum energy structure: as with the refinement
</p>
<p>that we have discussed in Sect. 8.4ff, a true, or global, minimum will not be reached unless we have a
</p>
<p>reasonably accurate set of coordinates with which to begin. Lower energy states are more thermody-
</p>
<p>namically stable and play the most important role in chemical and biological processes.
</p>
<p>After the FO model, provided by SWISSMODEL, had been superimposed onto the coordinates of
</p>
<p>the corresponding subunits in the PDB structure 1c17, using the program MOE [41], an energy
</p>
<p>minimization of the three chains involved in ligand binding was then carried out until all van der
</p>
<p>Waals contacts had been eliminated. Two models which were used for the docking were created from
</p>
<p>the resultant model, one containing the full 12 c-subunits (Model 1) and the other including only the
</p>
<p>three subunits involved in binding with the ligand (Model 2).
</p>
<p>9.11.4 Creation of a Pocket for Docking Oligomycin into the ATP Synthase FO
</p>
<p>Inspection of the completed FO model indicated that in its present form there was insufficient space
</p>
<p>between the subunits, thought to be involved in the binding of oligomycin, in which to fit this ligand.
</p>
<p>It was therefore necessary to create an artificial pocket in FO for this purpose. This is a reasonable
</p>
<p>requirement because, when actually docking in vivo, it is quite feasible that the ligand would have to
</p>
<p>force its way into the receptor.
</p>
<p>Consequently a small space at the desired location was found using the facility Site Finder in
</p>
<p>MOE. Dummy atoms were then placed into this pocket and Oligomycin B was manually overlaid
</p>
<p>onto these dummy atoms and an energy minimization [42] was carried out using the program
</p>
<p>AMBER99 on the three chains involved in binding (two c-subunits and the a-subunit). The backbone
</p>
<p>atoms of these chains were fixed in order to retain the overall integrity of the structure. To develop
</p>
<p>further the binding pocket, Oligomycin A was overlaid on to the coordinates of the energy minimized
</p>
<p>ligand within the pocket, and energy minimization was again performed. This procedure was
</p>
<p>followed during the creation of the pocket in both Model 1 (12 c-subunits and 1 a-subunit) and
</p>
<p>Model 2 (subunits involved in ligand binding) which was then capable of accommodating any of the
</p>
<p>oligomycins.
</p>
<p>Docking Oligomycin into the ATP Synthase FO
The docking studies performed have highlighted contacts to the following residues that are known to be
</p>
<p>essential for proton transport through the membrane: Asp61, Arg210, Asn214, and Gln252 [43].
</p>
<p>Figure 9.36 demonstrates Oligomycin C bound in place, contacting these integral residues. According
</p>
<p>to the docking study, these residues were contacted preferentially by specific ligating groups on the
</p>
<p>antibiotic. The ligating group that appears to be most essential to binding is O(24) with a preference for
</p>
<p>Asn214; this is a consistent theme for all forms of oligomycin and both FO models. According to the
</p>
<p>results the ligand has a tendency to arrange itself in such a way that the outer ligating groups O(24), O
</p>
<p>(28), O(32), and O(14) are positioned such that they will coordinate with Asn214. This allows for a
</p>
<p>certain amount of mobility within the pocket permitting the ligating groups to contact other residues, for
</p>
<p>example O(28)&ndash;Gln252 and O(2)&ndash;Asp61.
</p>
<p>Conclusions
</p>
<p>Docking of all oligomycins to the FO pocket indicated that the B and C forms would bind more tightly
</p>
<p>than the A form. Consideration of the single crystal X-ray data alone indicated the B form to be the
</p>
<p>best inhibitor and that O(24) was the most important ligating group for binding. This was supported
</p>
<p>Conclusions 483</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_8#Sec29_8">http://dx.doi.org/10.1007/978-1-4614-3954-7_8#Sec29_8</a></div>
</div>
<div class="page"><p/>
<p>by the docking data. The latter revealed Asn214 and other key proton translocating residues to be the
</p>
<p>main residues contacted by the inhibitor.
</p>
<p>These data have allowed the binding modes of different forms of oligomycin to be deduced from
</p>
<p>X-ray single crystal data supported by molecular modeling and computational docking studies.
</p>
<p>9.12 Problems
</p>
<p>9.1. The unit cell of euphenyl iodoacetate, C32H53O2I, has the dimensions: a &frac14; 7.26 Å, b &frac14; 11.55 Å,
c &frac14; 19.22 Å, and b &frac14; 94.07�. The space group is P21 and Z &frac14; 2. Figure P9.1 is the sharpened
and &ldquo;selected&rdquo; Harker section &eth;u; 1
</p>
<p>2
;w&THORN;.
</p>
<p>Fig. 9.36 Example of the
docking results:
</p>
<p>oligomycin C docked,
</p>
<p>showing the hydrogen-
</p>
<p>bonding to Asn 214, Arg
</p>
<p>210, and Asp 61
</p>
<p>484 9 Examples of Crystal Structure Determination</p>
<p/>
</div>
<div class="page"><p/>
<p>(a) Determine the x and z coordinates for the iodine atoms in the unit cell.
</p>
<p>(b) Atomic scattering factor data for iodine are tabulated below; temperature factor corrections
</p>
<p>may be ignored.
</p>
<p>Determine probable signs for the reflections 001 (Fo &frac14; 40), 0014 (Fo &frac14; 37), 106 (Fo &frac14; 33),
and 300 (Fo &frac14; 35). Comment upon the likelihood of the correctness of the signs which you
have determined.
</p>
<p>(c) Calculate the length of the shortest iodine-iodine vector in the structure.
</p>
<p>9.2. The following jEj values were determined for the [100] zone of a crystal of space group P21/a.
Prepare a
</p>
<p>P
2 listing, assign an origin, and determine signs for as many reflections as possible,
</p>
<p>and give reasons for each step that you carry out. In this projection, two reflections for which the
</p>
<p>indices are not both even may be used to specify the origin.
</p>
<p>0kl jEj 0kl jEj
0018 2.4 0310 1.9
</p>
<p>011 1.0 0312 0.1
</p>
<p>021 0.1 059 1.9
</p>
<p>024 2.8 081 2.2
</p>
<p>026 0.3 0817 1.8
</p>
<p>035 1.8 011,7 1.3
</p>
<p>038 2.1 011,9 2.2
</p>
<p>9.3. The chart in Fig. P9.2 shows jEj values taken from the hk0 data for potassium hydrogen squarate.
Take an origin at the center of a sheet of centimeter graph paper and copy the jEj values on to it,
using the top left portion of each appropriate square. For each jEj value plotted, add the
corresponding symmetry-related jEj values to the other three portions of the graphical reciprocal
space representation. Remember to change the signs of jEj in accordance with the space group
symmetry. Next, draw an identical chart on transparent paper, but with the jEj values in the
bottom right portion of each appropriate square.
</p>
<p>Fig. P9.1 Sharpened
Harker section &eth;u; 1
</p>
<p>2
;w&THORN; for
</p>
<p>euphenyl iodoacetate
</p>
<p>(sin y)/l 0.00 0.05 0.10 0.15 0.20 0.25 0.30 0.35 0.40
</p>
<p>f1 53.0 51.7 48.6 45.0 41.6 38.7 36.1 33.7 31.5
</p>
<p>9.12 Problems 485</p>
<p/>
</div>
<div class="page"><p/>
<p>(a) Obtain a
P
</p>
<p>2 listing: take each plotted jEj value in turn on the original chart and superimpose
the transparency, with the origin of the transparency over the chosen jEj value and keeping
the two sets of h, k axes in register. Look for any superimposed jEj values. A
</p>
<p>P
2 triplet is
</p>
<p>given by the jEj value on the original chart under the origin of the transparency, together with
the superimposed values, with the hk indices read, one from the original and the other from
</p>
<p>the transparency. Thus, with the origin of the transparency on the original jE(300)j, we read
840 on the original and 540 on the transparency. Set up the
</p>
<p>P
2 listing as follows:
</p>
<p>h jEhj k jEkj h � k jEh � kj jEhjjEkjjEh�kj
300 1.75 840 1.79 �5�40 1.92 6.01
	
	
	
700 2.26 	 	 	
</p>
<p>The rationale for the graphical procedure may be seen from Fig. P9.3.
</p>
<p>(b) Assign an origin in accordance with the rules discussed in the previous chapter and allocate
</p>
<p>signs to as many reflections as possible; use symbols if necessary. It may be assumed that the
</p>
<p>products jEhjjEkjjEh�kj are all sufficiently large for the indications to be accepted.
</p>
<p>Fig. P9.2 Chart of the jE(hk0)j data for the KHSQ structure
</p>
<p>486 9 Examples of Crystal Structure Determination</p>
<p/>
</div>
<div class="page"><p/>
<p>9.4. The dithiete compound discussed in Sect. 9.4 has the sulphur atoms placed as follows:
</p>
<p>x y z
</p>
<p>S(1) 0.2092(2) 3
4
</p>
<p>�0.1522(2)
S(2) 0.1095(1) 0.6933(1) �0.1216(2)
</p>
<p>(a) Calculate the S(1)&ndash;S(2) bond distance and its esd.
</p>
<p>(b) Draw a diagram to show the positions and weights of the Patterson vector disposition around
</p>
<p>the origin of the unit cell, as seen in projection on to the xz plane. For the purpose of the
</p>
<p>drawing, a b-angle of 90� can be assumed.
</p>
<p>References
</p>
<p>1. Ladd MFC, Povey DC (1972) J Cryst Mol Struct 2:243
</p>
<p>2. International critical tables 3:80 (1928), 7:73 (1930); Mellor&rsquo;s treatise on inorganic and theoretical chemistry 2:941
</p>
<p>(supplement, 1961)
</p>
<p>3. Bull RJ et al (1973) Cryst Struct Commun 2:625
</p>
<p>4. Ladd MFC, Povey DC (1976) Acta Crystallogr B32:1311
</p>
<p>5. Kusters W, de Mayo P (1974) J Am Chem Soc 96:3502
</p>
<p>6. Boar R et al (1975) J Chem Soc Chem Commun 756
</p>
<p>7. Hoppe W (1965) Angew Chem 74:484
</p>
<p>8. Arnott S et al (1961) J Chem Soc 4183
</p>
<p>9. Stewart JM et al (1972) The X-ray system&mdash;version. University of Maryland, College Park
</p>
<p>10. Hamilton WC (1965) Acta Crystallogr 18:502
</p>
<p>11. Cymerman J, Willis JB (1951) J Chem Soc 1332
</p>
<p>12. Ladd MFC (1976) Unpublished results
</p>
<p>13. Boar R (1975) loc. cit.
</p>
<p>14. Palmer RA, Potter BS (2008) J Chem Crystallogr 38:243
</p>
<p>15. Lardy HA (1980) Pharmacol Ther 11:649
</p>
<p>16. Fossati G et al (2003) J Immunol 170:1964
</p>
<p>17. Altamura N et al (1996) FEBS Lett 382:111
</p>
<p>18. von Glehn M et al (1972) FEBS Lett 15:267
</p>
<p>19. Fernandez-Moran H (1962) Circulation 26:1039
</p>
<p>20. Abrahams JP et al (1994) Nature 370:621
</p>
<p>21. Boyer PD (1993) Biochim Biophys Acta 1140:215
</p>
<p>22. Enraf-Nonius (1988) CAD-4 software. Enraf-Nonius, Delft, Holland
</p>
<p>Fig. P9.3
P
</p>
<p>2 relationships: subscript o refers to the original chart, and subscript t refers to the transparency. If the
</p>
<p>origin of the transparency is placed over a chosen ho, it may be seen that coincidences of jEj values given by +kt,
ho + ko or by �kt, ho � ko will represent
</p>
<p>P
2 triplets. It should be noted that this technique applies only to
</p>
<p>centrosymmetric projections of space groups
</p>
<p>References 487</p>
<p/>
</div>
<div class="page"><p/>
<p>23. North ACT et al (1968) Acta Crystallogr A24:351
</p>
<p>24. Sheldrick GM (1986) SHELXS-86. University of G&euro;ottingen, Germany
25. Sheldrick GM (1993) SHELXL-97. University of G&euro;ottingen, Germany
26. Farrugia LJ (1998) J Appl Crystallogr 32:837
</p>
<p>27. Spek AL (1990) Acta Crystallogr A46:C34
</p>
<p>28. Farrugia LJ (1997) J Appl Crystallogr 30:565 (based on ORTEP-III (v 1.0.3) by C. K. Johnson and M. N. Burnett)
</p>
<p>29. Merrit EA, Bacon DJ (1997) Methods Enzymol 277:505 (implemented in WinGX and generated by ORTEP-III for
</p>
<p>Windows)
</p>
<p>30. Humphrey W et al (1996) J Mol Graphics 14:33
</p>
<p>31. Berdy J et al (1987) Handbook of antibiotic compounds, vol IV. CRC Press, Boca Raton, FL
</p>
<p>32. Ramirez F et al (1982) Eur J Biochem 121:275
</p>
<p>33. Flack HD (1983) Acta Crystallogr A39:876
</p>
<p>34. Boyer PD (1994) Biochim Biophys Acta 1140:215
</p>
<p>35. Devenish RJ et al (2000) J Bioenerg Biomembr 32:507
</p>
<p>36. Galanis M et al (1989) FEBS Lett 249:333
</p>
<p>37. Rastogi VK, Girvin ME (1999) Nature 402:263
</p>
<p>38. Arnold K et al (2006) Bioinformatics 22:195
</p>
<p>39. http://www.expasy.org
</p>
<p>40. http://www.ebi.ac.uk/interpro/
</p>
<p>41. http://www.chemcomp.com/
</p>
<p>42. Wang J et al (2000) J Comp Chem 21:1049
</p>
<p>43. de Jonge MR et al (2007) Proteins Struct Funct Bioinform 67:971
</p>
<p>Bibliography: Published Structure Analyses
</p>
<p>Acta Crystallographica (the early issues (1948&ndash;1968) are most suitable for the beginner)
</p>
<p>Journal of Crystallographic and Spectroscopic Research
</p>
<p>Journal of Chemical Crystallography (formerly Journal of Crystal and Molecular Structure)
</p>
<p>Zeitschrift f&euro;ur Kristallographie
</p>
<p>General Structural Data
</p>
<p>Kennard O et al. Cambridge Structural Database (CSD)
</p>
<p>Wyckoff RWG (1963&ndash;1966, 1968, 1971) Crystal structures, vols 1&ndash;6. Wiley, New York
</p>
<p>Molecular Mechanics
</p>
<p>Frenkel D, Smit B (1996) Understanding molecular simulation. Academic, New York
</p>
<p>488 9 Examples of Crystal Structure Determination</p>
<p/>
</div>
<div class="page"><p/>
<p>Proteins and Macromolecular
X-Ray Analysis 10
</p>
<p>10.1 Introduction
</p>
<p>In this chapter, we take a more detailed look at methods of X-ray analysis that are particularly
</p>
<p>applicable to large biological molecules. It will involve some useful reiteration of concepts and ideas
</p>
<p>discussed in previous chapters. We would also remind readers that although there are definite
</p>
<p>distinctions between large and small molecules in the crystallographic arena, there is no reason to
</p>
<p>exclude one from the other, and in fact, there are many advantages in being familiar with both. The
</p>
<p>major differences should become clearer as we progress through this chapter. It follows that while we
</p>
<p>deal mainly with macromolecules here, much of the information provided in this chapter is applicable
</p>
<p>to all areas of crystal structure analysis.
</p>
<p>Traditionally the branch of X-ray crystallography that has developed around the study of macro-
</p>
<p>molecular structures is thought of as being derived from the need in the 1940s and 1950s to extend the
</p>
<p>methods of analysis to include protein and enzyme structures, which range in relative molar mass
</p>
<p>from around 6000 upwards. Structural studies of many other macromolecular types, including
</p>
<p>chromatin, DNA fragments, and other polynucleotides, present similar problems for the crystallogra-
</p>
<p>pher, and consequently many of the methods currently in use for proteins are applicable. Specific
</p>
<p>topics to be covered include: methods for growing crystals, collection, measurement and handling of
</p>
<p>the X-ray data, and methods for analyzing and interpreting the structures. Much of the emphasis here
</p>
<p>will be given to the practical aspects of this type of research, most of the background to the underlying
</p>
<p>theory having been covered in previous chapters. We have seen that X-ray crystallography is highly
</p>
<p>computational, and a great variety of software is available for performing the various stages of the
</p>
<p>calculations and graphics display. A guide to the availability of both equipment and software is
</p>
<p>included in Appendix D, together with important Internet addresses. The flow chart in Fig. 10.1
</p>
<p>indicates the first stages in the X-ray analysis of a macromolecule.
</p>
<p>10.1.1 What Is a Protein?
</p>
<p>Although proteins are very large biological molecules, their chemical formulation is quite straight-
</p>
<p>forward. Figure 10.2a illustrates some of the important features of the first three residues of a protein
</p>
<p>polypeptide chain: Ca atoms are chiral carbon atoms (see below); R1, R2, and R3 are side-groups,
</p>
<p>each of which can be one of 20 [1] different chemical moieties that make up the &ldquo;protein alphabet&rdquo;:
</p>
<p>the N-terminus (�NH2 or &ndash;NH3
+) is the beginning of the polypeptide chain, and at some distance
</p>
<p>M. Ladd and R. Palmer, Structure Determination by X-ray Crystallography:
</p>
<p>Analysis by X-rays and Neutrons, DOI 10.1007/978-1-4614-3954-7_10,
# Springer Science+Business Media New York 2013
</p>
<p>489</p>
<p/>
</div>
<div class="page"><p/>
<p>Fig. 10.1 Flow diagram for the preliminary stages in the X-ray analysis of a macromolecule
</p>
<p>Fig. 10.2 (a) Schematic diagram of a polypeptide chain found in proteins, showing the first two amino acid residues
and the beginning of the third. The R-groups comprise one of the 20 commonly occurring moieties. The peptide groups
</p>
<p>linked on either side to the Ca atoms are predominantly planar, the torsion angleo being usually in the range�20�; two
torsion angles C and f for each residue define the polypeptide main-chain conformation, Sect. 10.11.2. (b) The
absolute configuration about a Ca atom in the L-configuration, looking along H&ndash;C; the sequence C ! O ! R ! N is
encountered in a clockwise sense
</p>
<p>490 10 Proteins and Macromolecular X-Ray Analysis</p>
<p/>
</div>
<div class="page"><p/>
<p>away will be the C-terminus group (�CO2H); the amino acid sequence is R1, R2, R3, . . ., Rn, where n
is the total number of residues in the chain, and is known as the primary structure. Elements of
</p>
<p>secondary structure describe ways in which the polypeptide chain can fold; a-helices (right-handed)
</p>
<p>and b-sheets constitute two of the principle elements of secondary structure; tertiary structure
</p>
<p>describes how the elements of secondary structure are arranged to form the whole structure.
</p>
<p>Two proteins are said to be homologous if their amino acid sequences are similar to within 40%,
</p>
<p>although lesser values have been used; corresponding amino acids can be either exactly the same type
</p>
<p>or conserved by replacement of similar side-groups. The sequence N&ndash;Ca&ndash;C(&frac14;O)&ndash;N&ndash;Ca&ndash;C(&frac14;O)&ndash; . . .
</p>
<p>is known as the main chain. The bond labeled o has limited torsional freedom, with the torsion angle
</p>
<p>o being 180 � 20� (maximum) for the usual trans peptide conformation, and 0 � 20� (maximum)
for the less usual cis peptide; the bonds on either side of the Ca atoms labeled with the torsion angles
</p>
<p>f and C have greater but not unlimited rotational freedom. The folding of the protein chain depends
</p>
<p>on the values that f and C take up along the chain to produce a minimum energy conformation, and
</p>
<p>this in turn will be dependent on the amino acid sequence. Some large proteins comprise more than
</p>
<p>one polypeptide chain, each of which acts as a subunit contributing to the tertiary structure of the
</p>
<p>protein; structurally equivalent subunits can be related either by exact crystallographic symmetry (the
</p>
<p>whole protein then occupies the corresponding symmetry element), by approximate or so-called non-
</p>
<p>crystallographic symmetry (NCS), or in no regular manner at all.
</p>
<p>10.2 Crystallization of Proteins and Complexes for X-Ray Analysis
</p>
<p>10.2.1 Introduction
</p>
<p>Bernal and Crowfoot [2] produced the first X-ray diffraction pattern of the protein crystal pepsin in
</p>
<p>1934. Since then hundreds of biological macromolecules have been crystallized, at first using mainly
</p>
<p>unrefined methodology, and with little or no control of the outcome. Recently, however, it has been
</p>
<p>possible to rationalize the procedures, at least to some extent, through the use of hanging and sitting
</p>
<p>drops, which require minimal amounts of protein, and multiple sampling techniques, which enable a
</p>
<p>wide variety of conditions to be tried.
</p>
<p>Once established, crystallization procedures are generally reproducible. However, problems can
</p>
<p>arise as a result of one or more factors: minor differences in protein composition or purity, possibly
</p>
<p>through extraction from different sources; or failure to reproduce exact crystallization conditions,
</p>
<p>perhaps owing to poor reporting or recording protocols. Even the chance presence of a trace of dirt or
</p>
<p>grease can make a difference to the resulting product. Special properties of protein crystals are
</p>
<p>listed here:
</p>
<p>1. Protein crystals tend to be small compared to common crystals, being rarely greater than 1 mm on
</p>
<p>any edge.
</p>
<p>2. In the vast majority of examples, only one stereoisomer of any biological macromolecule exists in
</p>
<p>nature, so that their crystals contain only symmetry elements with rotation and/or translation;
</p>
<p>inversion and reflection are excluded as these symmetry operations involve a change of hand. As a
</p>
<p>consequence, the crystals themselves tend to exhibit fairly simple shapes compared to many
</p>
<p>minerals and other naturally occurring crystals, such as quartz or ice crystals.
</p>
<p>3. Protein crystals are fragile and require extremely gentle handling. This arises from their high
</p>
<p>solvent content, which can be as much as 70%.
</p>
<p>4. Protein crystals are extremely sensitive to pH, ionic strength, and temperature. Stability to low
</p>
<p>temperature can be improved through the use of cryoprotectants, forming a useful technique for
</p>
<p>improving the extent and quality of X-ray intensity data.
</p>
<p>10.2 Crystallization of Proteins and Complexes for X-Ray Analysis 491</p>
<p/>
</div>
<div class="page"><p/>
<p>5. Protein crystals diffract X-rays weakly, usually far short of atomic resolution; this effect is
</p>
<p>associated with the presence of large amounts of disordered or partially disordered solvent in
</p>
<p>the crystal.
</p>
<p>10.2.2 Crystallization Conditions for Macromolecules
</p>
<p>In keeping with aspects of crystal growth in general, the successful production of protein crystals
</p>
<p>is highly dependent on supersaturation. Macromolecular crystals usually form by nucleation at
</p>
<p>extremely high levels of supersaturation (100&ndash;1000%). In comparison, small-molecule crystals
</p>
<p>usually nucleate at only a few percent supersaturation. Work with proteins usually means that the
</p>
<p>starting material is both expensive and sparse (20 mg is a large amount of pure protein), whereas
</p>
<p>most small-molecule compounds are available in much larger quantities and consequently quite
</p>
<p>easy to crystallize.
</p>
<p>Although high levels of supersaturation may be essential for promoting the nucleation of
</p>
<p>macromolecular crystals, there are general problems in terms of the formation of good quality
</p>
<p>crystals, because supersaturated macromolecular solutions tend to produce amorphous-like preci-
</p>
<p>pitates. Consequently, there is competition between crystals and precipitates at both nucleation and
</p>
<p>growth stages, and this competition is particularly acute because it is promoted at high levels of
</p>
<p>supersaturation. Because amorphous precipitates are kinetically favored, even though they are
</p>
<p>associated with higher energy states, they tend to dominate the solid phase and inhibit or even
</p>
<p>preclude crystal formation.
</p>
<p>10.2.3 Properties of Protein Crystals
</p>
<p>Protein crystals may contain one or more of the following, which may influence their size, quality,
</p>
<p>and X-ray diffraction characteristics: air pockets, disordered molecular deposits or clusters, inclu-
</p>
<p>sions, ordered and disordered solvent, precipitant ions, impurities, such as bound carbohydrate,
</p>
<p>inhibitors, non-covalently bound sugars, prosthetic groups or other ligands, or covalent or non-
</p>
<p>covalently bound heavy atoms. The quality of sample homogeneity is another important factor.
</p>
<p>10.2.4 Crystallization of Proteins
</p>
<p>The formation of macromolecular crystals depends mainly on three factors:
</p>
<p>1. Changing the relationship between the macromolecules and the solution components (water
</p>
<p>molecules and ions)
</p>
<p>2. Altering the structure of the solvent so that the molecules are less well accommodated, thus
</p>
<p>promoting phase separation
</p>
<p>3. Enhancing the number and strength of favorable interactions between macromolecules. If the right
</p>
<p>conditions can be established, the molecules will be continuously associating to form clusters and
</p>
<p>aggregates, to which new molecules are added more rapidly than old ones are lost. A crystal
</p>
<p>nucleus will then be born and growth will proceed
</p>
<p>492 10 Proteins and Macromolecular X-Ray Analysis</p>
<p/>
</div>
<div class="page"><p/>
<p>10.2.5 Molecular Purity
</p>
<p>In order to form crystals, the macromolecules have to be ordered in regular three-dimensional arrays.
</p>
<p>All forms of interference with regular packing will hinder crystallization. Lack of purity and
</p>
<p>homogeneity are major factors in causing unsuccessful and irreproducible crystallization experi-
</p>
<p>ments. It is advisable to carry out crystallization assays on fresh samples without mixing different
</p>
<p>batches of macromolecules. However, some micro-heterogeneities can be tolerated provided they do
</p>
<p>not occur in parts of the molecule involved in packing contacts in the crystal structure, a factor that is
</p>
<p>unpredictable prior to the structure being solved.
</p>
<p>10.2.6 Practical Considerations
</p>
<p>The high supersaturation of the molecules required for crystal nucleation can be achieved using a
</p>
<p>variety of chemical precipitants. Widely used precipitants include ammonium sulphate, polyethylene
</p>
<p>glycol (PEG), methylpentanediol (MPD), and sodium chloride.
</p>
<p>When starting from non-saturated solutions, supersaturation can be reached by varying parameters
</p>
<p>such as temperature or pH. It should be noted that the conditions for optimal nucleation are not the
</p>
<p>same as those for optimal growth. Nucleation may be homogeneous or heterogeneous, occurring in
</p>
<p>the latter case on solid particles. This can lead to epitaxial growth (growth of crystals on other
</p>
<p>crystals). Interface or wall effects and the shape and volume of drops when using hanging or sitting
</p>
<p>drop techniques, Sect. 10.2.9, can affect nucleation or growth. Therefore, the geometry of crystalli-
</p>
<p>zation chambers or drops can be quite critical.
</p>
<p>10.2.7 Batch Crystallization
</p>
<p>This method is not normally used in current research, as large amounts of protein are required. As an
</p>
<p>exercise it can, however, be quite informative, and serious students are encouraged to try it before
</p>
<p>moving to the smaller scale methods described below. In this method protein solutions are prepared in
</p>
<p>milliliter quantities and crystallization is carried out in ordinary test tubes, typically of 5&ndash;10ml capacity.
</p>
<p>The crystals grow on the walls of the test tubes and are harvested for storage when of suitable size.
</p>
<p>To grow large crystals [3a] of the enzyme ribonuclease II (Mr � 13000), put 100 mg protein (e.g., ex
SIGMA Pharmaceuticals) into a test tube and dissolve in 2 ml water, cooled to 0 �C; then gradually add
1 ml ice-cold absolute ethanol with stirring. Measure the pH and adjust to 5.0 by addition of a suitable
</p>
<p>buffer. Keep the tube at room temperature for 3 days. Re-cool and add a further 0.4 ml ethanol with
</p>
<p>stirring. The composition is now approximately 40% v/v. Small crystals should eventually appear on the
</p>
<p>walls of the tube within 10 days. The crystals obtained should be monoclinic, space group P21 [3b].
</p>
<p>10.2.8 Microbatch Screening
</p>
<p>As discussed in Sect. 10.2.6, it is necessary to establish the correct crystallization conditions for a new
</p>
<p>protein. Consequently, initial experiments are usually conducted at a microscale level with sample
</p>
<p>volumes in the range of a few microliters, with protein concentrations 5&ndash;10 mg ml&ndash;1. Since it is
</p>
<p>difficult and expensive to prepare large quantities of highly purified protein, this strategy allows many
</p>
<p>different crystallization conditions to be screened. The experiment is conducted using microbatch
</p>
<p>plates such those of Hampton Research [4]. Each plate, Fig. 10.3, contains a number of small wells in
</p>
<p>10.2 Crystallization of Proteins and Complexes for X-Ray Analysis 493</p>
<p/>
</div>
<div class="page"><p/>
<p>the form of a matrix, so that individual wells can be easily labeled for future reference. This is
</p>
<p>important, as each well will correspond to a different set of conditions and crystals will only form in a
</p>
<p>very small number of wells, perhaps only one.
</p>
<p>To set up the plate, drops are pipetted from a fine syringe under a layer of silicone oil (polydimethyl-
</p>
<p>siloxane) or paraffin oil, or their 1:1 mixture. The drops contain protein and precipitant solution. All the
</p>
<p>reagents involved in the crystallization are present at a specific concentration and no significant concen-
</p>
<p>tration of the protein or of the reagents can occur in the drop. Reagent sampling kits are commercially
</p>
<p>available. Diffusion ofwater from the drop takes place through the oil thus changing the concentration in
</p>
<p>the drop, hopefully toward the required crystallization conditions. Once established, the localized well
</p>
<p>conditions can be refined in order to optimize the crystal size and quality for X-ray diffraction, using one
</p>
<p>of the techniquesdescribedbelow.Microbatchdroplets usually dry upcompletelywithin a fewweeks and
</p>
<p>consequently require to be carefully monitored.
</p>
<p>10.2.9 Vapor Diffusion Techniques
</p>
<p>The most popular and successful techniques for establishing crystallization conditions rapidly and
</p>
<p>efficiently with subsequent production of diffraction quality protein crystals are based on vapor
</p>
<p>diffusion. There are several practical variations in use.
</p>
<p>Hanging Drop
This method is illustrated in Fig. 10.4. Each drop is set up by being rapidly inverted over the prepared
</p>
<p>well where it hangs by surface tension. The wells are again in the form of a matrix, typically 4 � 6
</p>
<p>Fig. 10.3 Microbatch plate, (a) The well matrix, (b) The profile of the wells and detail of the drops
</p>
<p>494 10 Proteins and Macromolecular X-Ray Analysis</p>
<p/>
</div>
<div class="page"><p/>
<p>(e.g., ex Hampton [4]). A great variety of plates is available, many of which are also adaptable for the
</p>
<p>sitting drop technique. In the hanging drop method the droplet (5&ndash;20 ml) containing the macromole-
</p>
<p>cule, a buffer and a precipitating agent is equilibrated against a reservoir (1&ndash;25 ml) containing a
</p>
<p>solution of the same precipitant at a higher concentration than the droplet, say, by a factor of 2.
</p>
<p>Equilibration proceeds by evaporation of the volatile component until the vapor pressure of the
</p>
<p>droplet equals that of the reservoir. Crystals form in the droplet.
</p>
<p>Sitting Drop
The principle here is essentially the same as that in the hanging drop technique except that the drop is
</p>
<p>placed on a bridge which sits over the precipitant reservoir. If successful, crystals grow in the droplet.
</p>
<p>Drop size and conditions of precipitants, buffers, and other factors similar to those used for hanging
</p>
<p>drops apply.
</p>
<p>Vapor Diffusion Rate of Control
As in the microbatch technique, a layer of oil can be used with hanging or sitting drop techniques.
</p>
<p>This limits the rate of vapor diffusion; 200 ml of paraffin or silicone oil or a mixture of the two is
</p>
<p>applied over the reservoir solution. Varying the composition of the mixture provides additional
</p>
<p>control over the vapor diffusion rate.
</p>
<p>Screening Crystallization Conditions
The above methods are all used with multiple screening protocols. A plate containing typically 4 � 6
wells provides a matrix of conditions. The method allows a broad range of salts, polymers and organic
</p>
<p>solvents over a wide range of pH to be sampled. When crystals are obtained, a second, finer screening
</p>
<p>around the relevant conditions can be used in order to optimize crystallization conditions to produce
</p>
<p>X-ray diffraction quality crystals. Ready-to-use reagents formulated from highly pure salts, buffers, and
</p>
<p>precipitants at various concentrations are commercially available.
</p>
<p>Gel Crystallization Using Silica Hydrogel
Gels provide very efficient media for growing protein crystals. Silica gels, in particular are stable over
</p>
<p>a wide range of conditions, and compatible with the additives and precipitants commonly used in
</p>
<p>macromolecular crystal growing. The gel forms a porous network in which the crystals can grow; it
</p>
<p>Fig. 10.4 Diagram of the hanging drop method of crystallization
</p>
<p>10.2 Crystallization of Proteins and Complexes for X-Ray Analysis 495</p>
<p/>
</div>
<div class="page"><p/>
<p>minimizes convection, sedimentation, and nucleation, and therefore promotes the growth of large
</p>
<p>crystals without the strain usually imposed by the presence of a container. Silica hydrogel can be used
</p>
<p>for liquid&ndash;gel, liquid&ndash;gel&ndash;liquid, and vapor diffusion as well as dialysis techniques.
</p>
<p>10.2.10 Co-crystallization
</p>
<p>Many proteins are required to be studied in the presence of ligands, which are usually quite small
</p>
<p>molecules and include sugars, inhibitors, coenzymes, nucleotides, nucleic acid fragments, and sometimes
</p>
<p>other macromolecules such as antibodies.Whenmultiple isomorphous replacement (MIR) is being used
</p>
<p>as a phasing technique for structure analysis; see Sects. 7.5.8 and 10.6.2, it is necessary to incorporate
</p>
<p>heavy-atom reagents into the crystals. One commonly used method of preparing these complex co-
</p>
<p>crystals is to add the material into the protein solution prior to crystallization. Crystals grown in this way
</p>
<p>are quite likely to suffer a change of crystal system,which is of course of no practical use inMIR, and can
</p>
<p>lead to unwanted complications in their applications. This is discussed further in Sect. 10.2.12.
</p>
<p>10.2.11 How to Improve the Crystals
</p>
<p>The following brief guide should help to improve the situation where poor (or no) crystals have
</p>
<p>appeared initially.
</p>
<p>1. Vary the screening conditions: buffer, pH (possibly finer intervals), precipitant, protein concen-
</p>
<p>tration, drop size, method used, temperature (crystals will sometimes form under cold conditions).
</p>
<p>2. Use a finer matrix to vary the conditions around the wells with crystals. Try another method, for
</p>
<p>example, the hanging drop instead of the sitting drop. Try seeding: this works by transferring finely
</p>
<p>crushed crystal particles in the wells using a cat&rsquo;s whisker or a hair.
</p>
<p>3. One or more of the following additives may help the crystallization process: Cu2+, Zn2+, Ca2+, Co2+
</p>
<p>ions, ethylenediamine tetraacetic acid, acetone, dioxane, phenol: for membrane proteins, N-
</p>
<p>octyl-b-D-glucopyranoside (up to critical micelle concentrations) and substrates, cofactors, inhibi-
</p>
<p>tors, or binding sugars.
</p>
<p>4. With limited amounts of protein, concentrate on a few parameters known to be important: pH,
</p>
<p>initially in 0.5 intervals of pH unit, decreasing the interval as conditions are established; tempera-
</p>
<p>ture, 4, 22, and 37 �C are most commonly used; precipitants, such as ammonium sulfate, sodium
chloride, PEG, ethanol, or MPD are highly favored.
</p>
<p>5. Crystals are obtained but their X-ray diffraction pattern looks like that of a small molecule, that is,
</p>
<p>spots widely separated on the photograph. Try the click test on one of the crystals: small-molecule
</p>
<p>crystals such as ammonium sulphate are usually physically hard and difficult to crush and will
</p>
<p>audibly &ldquo;click&rdquo; when poked with a needle.
</p>
<p>6. Large crystals are obtained but the diffraction pattern disappears after ca. 30 min exposure to the
</p>
<p>X-ray beam, before a full set of data can be collected. Try freezing the crystals and book a session
</p>
<p>on a synchrotron facility.
</p>
<p>7. Crystals are very soft and disintegrate when mounted in a glass capillary tube. Try being more
</p>
<p>careful when mounting: transfer through a larger pipette; use a larger diameter glass capillary.
</p>
<p>Because of the fundamental importance of acquiring suitable crystals for diffraction work, a great
</p>
<p>deal of effort is input by researchers endeavoring to improve the techniques involved. For example a
</p>
<p>recent paper [5] on protein crystallization describes a simple method to perform matrix microseeding
</p>
<p>in which crystals grown in one set of conditions are seeded into a secondary screen of 96 crystalliza-
</p>
<p>tion solutions. The seed stocks can be stored at 193 K and can survive many cycles of freezing and
</p>
<p>thawing without a decrease in the nucleation effect observed. This technique has great potential for
</p>
<p>496 10 Proteins and Macromolecular X-Ray Analysis</p>
<p/>
</div>
<div class="page"><p/>
<p>improving &ldquo;hit rates&rdquo; in early stages of screening for crystallization conditions. Figure 10.5 illustrates
</p>
<p>results that can be achieved by this method.
</p>
<p>Another recent paper [6] suggests practical variations for growing large protein crystals.
</p>
<p>Figure 10.6 shows examples of protein crystals grown using this method in semi-solid agarose gel.
</p>
<p>(a) Xylanase. (b) Insulin at neutral pH. (c) Insulin at higher pH. (d) Glucose isomerase.
</p>
<p>For further details of the methods described later in this chapter and a list of equipment suppliers see
</p>
<p>Bibliography: Sanderson and Skelly; Sherwood and Cooper. An interesting video on the production of
</p>
<p>protein crystals for X-ray crystallography is also worth viewing [7].
</p>
<p>10.2.12 Heavy-Atom Derivatives for MIR
</p>
<p>A very wide range of heavy-atom reagents has been compiled for this purpose, mainly organic or
</p>
<p>inorganic compounds of mercury, gold, platinum, or uranium. The initial choice of possible
</p>
<p>compounds may be influenced by a knowledge of the amino acid sequence of the protein [8].
</p>
<p>Lists of known useful compounds may also be consulted [9, 10]. If good quality native crystals
</p>
<p>are already available, soaking them in heavy-atom solutions may be the best way to prepare
</p>
<p>derivatives. This can be carried out on single, mounted crystals, as in Fig. 10.7 except that it will
</p>
<p>be necessary to open the capillary tube and wash the crystal at the end of the soaking period so
</p>
<p>as to stop uptake of heavy-atom material. The procedure should be carried out under controlled
</p>
<p>conditions of pH, temperature, time and concentration. Heavy-atom concentrations in the range
</p>
<p>0.5&ndash;20 M may be tried for times between a few minutes and several days.
</p>
<p>The crystals may change color but should not undergo any other significant physical change.
</p>
<p>Detailed screening by X-ray techniques is necessary in order to establish isomorphism and to ensure
</p>
<p>that the heavy atoms have been incorporated into useful locations; see Sect. 7.5.7. Alternatively, co-
</p>
<p>crystallization can be tried. This involves setting up crystallization experiments using one of the
</p>
<p>methods described in the previous section, but with the addition of a heavy-atom reagent to the
</p>
<p>solution. It is essential to maintain the integrity of the solution prior to crystallization attempts. This
</p>
<p>may require careful adjustment of the conditions and a good deal of patience. If crystals form they
</p>
<p>will again require careful monitoring by X-rays so as to establish the desired incorporation of heavy
</p>
<p>atoms. A change of crystal unit cell and space group is more likely to occur in this method.
</p>
<p>Fig. 10.5 An example of protein crystallization results that can be achieved by the method of D&rsquo;Arcy et al. [6]. This
compares (a) USP7 protein crystals used for seeds grown in 30% PEG (polyethylene glycol) 3350 and 100 mMHEPES
[4-(2-hydroxyethyl)-1-piperazineethanesulphonic acid] at pH 7.0 with (b) Crystals grown after seeding in 20% PEG
3350 and 200 mM magnesium hexahydrate. The solid bar represents 200 mm
</p>
<p>10.2 Crystallization of Proteins and Complexes for X-Ray Analysis 497</p>
<p/>
</div>
<div class="page"><p/>
<p>10.2.13 Protein Complex Crystals with Small Molecules
</p>
<p>Many proteins associate with small molecules as part of their biological function, for example:
</p>
<p>enzyme&ndash;inhibitor complexes, protein&ndash;carbohydrate complexes (lectins), protein&ndash;nucleic acid frag-
</p>
<p>ments, protein&ndash;peptide complexes, such as cyclophillin/cyclosporin, and antibody&ndash;antigen
</p>
<p>Fig. 10.6 Examples of protein crystals grown using the method of Matsumara et al. [5]. The crystals were grow in
semi-solid agarose gel. (a) Xylanase. (b) Insulin at neutral pH. (c) Insulin at higher pH. (d) Glucose isomerase
</p>
<p>Fig. 10.7 Capillarymountedprotein crystal, as used for heavy-atomsoakingor forX-raydata collection at room temperature
</p>
<p>498 10 Proteins and Macromolecular X-Ray Analysis</p>
<p/>
</div>
<div class="page"><p/>
<p>complexes. Complex crystals may again be formed either by soaking or co-crystallization under
</p>
<p>controlled conditions. The methodology is similar to that used for the preparation of heavy-atom
</p>
<p>derivatives, and requires similar precautions and screening to establish incorporation of the adduct
</p>
<p>molecules. There is a slight preference for co-crystallization, as it is more likely to lead to full
</p>
<p>occupation of the ligand binding sites. Again, non-isomorphous derivatives may occur, but this
</p>
<p>problem should be easily overcome by using molecular replacement (MR) to determine the structure,
</p>
<p>assuming the native structure is known; see Sect. 8.3ff.
</p>
<p>10.3 Crystal Mounting for X-Ray Data Collection
</p>
<p>The two methods currently in use for X-ray data collection depend principally upon the temperature,
</p>
<p>either room temperature or that of liquid nitrogen.
</p>
<p>10.3.1 Mounting at Room Temperature
</p>
<p>For X-ray data collection at room temperature, protein crystals are usually mounted in sealed thin-
</p>
<p>walled glass capillary tubes as shown in Fig. 10.7. This method is rarely employed now and is
</p>
<p>retained here mainly for historical reasons.
</p>
<p>The presence of a drop of mother liquor inside the tube allows a stable equilibrium to be
</p>
<p>established, which prevents the crystal from drying out and becoming denatured. Most proteins
</p>
<p>rapidly deteriorate once exposed to X-rays, through the formation of free radicals. Crystal lifetime
</p>
<p>can be anything from a few minutes to several days under normal laboratory conditions. Exposure to
</p>
<p>the highly intense beam from a synchrotron source usually decreases crystal lifetime dramatically. It
</p>
<p>is compensated, however, by an increased beam intensity, which permits shorter individual exposures
</p>
<p>and consequent increase in the quantity of data recorded.
</p>
<p>10.3.2 Cryo-Crystallography
</p>
<p>When the temperature of a crystal is lowered, the thermal motion of the constituent atoms becomes
</p>
<p>less marked and the X-ray diffraction pattern can be improved in both intensity and resolution. These
</p>
<p>two factors are highly desirable in protein crystallography, as protein crystals are notoriously poor
</p>
<p>diffractors. Since the innovation of more efficient crystal cryo-systems in the past decade (a popular
</p>
<p>and efficient model is the Oxford Cryosystems Cryostream Cooler [11]), there has been a marked
</p>
<p>increase in the number of protein structures determined at low temperature, and small-molecule data
</p>
<p>is routinely collected on cooled crystals in some laboratories. However, for a new protein crystal, or a
</p>
<p>new derivative crystal, the method requires to be carefully set up empirically in order to establish the
</p>
<p>best conditions. The crystal is mounted and flash-cooled, usually to 100 K, free from ice, and set up
</p>
<p>in the X-ray beam. Damage from freezing is prevented through the use of a cryoprotectant liquid.
</p>
<p>The cryoprotectant may be incorporated in the mother liquor after crystallization or, less commonly,
</p>
<p>as a component of the crystallization reagents.
</p>
<p>The procedure involves supporting the crystal in a film of cryoprotected mother liquor in a small fiber
</p>
<p>loop, Fig. 10.8a, b, which is then cooled in liquid nitrogen. The liquid surrounding the crystal must freeze
</p>
<p>10.3 Crystal Mounting for X-Ray Data Collection 499</p>
<p/>
</div>
<div class="page"><p/>
<p>as an amorphous glass to avoid crystal damage and diffraction fromordered ice crystals. Glycerol ismost
</p>
<p>frequently used as a cryoprotectant in this type of work. The required glycerol concentration must be
</p>
<p>carefully established; unfortunately this may involve the loss of several crystals in the early stages of the
</p>
<p>experiment.
</p>
<p>Fig. 10.8 Cryo-crystallography. (a) The loop mounting of a crystal in a film of cryoprotected mother liquor, which is
then cooled in a gaseous stream of evaporating liquid nitrogen. (b) Arrangement for aligning the crystal in the stream of
cold gas. The goniometer head is usually a standard commercial device, with perpendicular translation slides, but
</p>
<p>normally no angular adjustments, because they are not needed for image-plate data collection
</p>
<p>500 10 Proteins and Macromolecular X-Ray Analysis</p>
<p/>
</div>
<div class="page"><p/>
<p>For the low temperature data collection of mistletoe lectin I (MLI) [12], a complex mixture was
</p>
<p>used, consisting of 0.1 M glycine buffer at pH 3.4, with 0.9 M ammonium sulfate, 0.05 M galactose
</p>
<p>and 30% glycerol, soaking for a few minutes prior to flash freezing. Once frozen, protein crystals are
</p>
<p>usually extremely stable, transportable, and can be stored and kept ready for subsequent X-ray
</p>
<p>diffraction experiments.
</p>
<p>10.4 Macromolecular Crystallography
</p>
<p>10.4.1 Space Groups
</p>
<p>Proteins are polymers of L-amino acids, the Ca atoms being specifically left-handed chiral centers,
</p>
<p>Fig. 10.2b. Consequently, only axes of symmetry occur in protein crystals. This is because centers of
</p>
<p>symmetry, mirror planes, and glide planes would necessarily produce D-amino acids, which do
</p>
<p>not normally occur. Similar considerations, in fact, apply to the majority of biologically active mole-
</p>
<p>cules. As a consequence, protein crystals can occur in only 65 of the possible 230 space groups,
</p>
<p>Table 10.1; space group determination is therefore a relatively simple procedure.
</p>
<p>10.4.2 X-Ray Diffraction from Macromolecular Crystals
</p>
<p>Crystal Selection
We have seen, or implied in previous sections, that for single crystal analysis, each crystal to be used
</p>
<p>for measuring diffraction data should be well ordered and produce a single, clear diffraction pattern.
</p>
<p>Poor quality specimens that are not single crystals may suffer from a variety of faults including
</p>
<p>splitting, twinning, or the presence of slippage-disorder planes. Split or twinned crystals may diffract
</p>
<p>well but will produce multiple reflection spots that are difficult to interpret; such specimens should be
</p>
<p>discarded. Badly disordered crystals are characterized by the spreading out of each diffraction spot,
</p>
<p>often in an irregular manner, and by the poor overall resolution of the diffraction pattern, which
</p>
<p>rapidly fades away as a function of the radial distance from the center of the pattern.
</p>
<p>Faults of this type are particularly associated with protein crystals that are relatively soft (containing
</p>
<p>up to 70% solvent), and easily damaged through handling when being mounted for X-ray analysis. Even
</p>
<p>single crystals diffract poorly if heavily disordered. If a crystal proves to be unsuitable through any of the
</p>
<p>above faults it should be discarded and another one selected. Refinement of crystallization conditions is
</p>
<p>frequently necessary in order to produce better crystals.
</p>
<p>Structure Factors and Temperature Factors
We have shown previously that the total X-ray scattering associated with a given hkl reflection
</p>
<p>is represented by the structure factor F(hkl). This parameter and the atomic scattering factor have
</p>
<p>been discussed in detail in Sects. 3.5 and 3.6ff. The use of structure factor equations for macro-
</p>
<p>molecules is no different from applications with other molecules, except of course that the number of
</p>
<p>atoms per unit cell is greater than 1000 even for quite small proteins.
</p>
<p>The isotropic temperature factor, has the following typical values for proteins: overall average,
</p>
<p>20&ndash;30 Å2; for main chain atoms, 10&ndash;20 Å2; for side-chain atoms 15&ndash;30 Å2; for solvent and small
</p>
<p>ligand atoms, 25&ndash;45 Å2. As a rule of thumb it is common for an upper limit of around 80 Å2 to be
</p>
<p>placed as a credibility indicator for any individual atom. Atoms with a temperature factor greater than
</p>
<p>this value after refinement, Sect. 10.9, should be carefully re-examined. The values for the tempera-
</p>
<p>ture factor of each atom evolve during the course of structure refinement.
</p>
<p>10.4 Macromolecular Crystallography 501</p>
<p/>
</div>
<div class="page"><p/>
<p>Intensities and Phases
Experimental Fo(hkl) values are derived from the measured intensities Io(hkl) by procedures that are
</p>
<p>discussed below. For a successful structure analysis, as many values as possible of Fo(hkl) must be
</p>
<p>measured to as high a resolution as can be achieved. Values of the phase angles f(hkl) required for
</p>
<p>calculation of electron density r(xyz) are usually determined by application of one of the following
</p>
<p>techniques:
</p>
<p>Table 10.1 The 65 enantiomorphic space groups applicable to protein crystals; corresponding Laue groups are listed
in brackets
</p>
<p>Crystal system Point group
</p>
<p>Space-group symbol
</p>
<p>(Laue-group symbol)
</p>
<p>Number Z of asymmetric
</p>
<p>units per unit cell
</p>
<p>Triclinic 1 P1 1
[�1]
</p>
<p>Monoclinic 2 P2, P21 2
C2 4
[2/m]
</p>
<p>Orthorhombic 222 P222, P2221, P21212 4
P212121 4
C222, C2221 8
I222, I212121 8
F222 16
[mmm]
</p>
<p>Tetragonal 4 P4, P41, P42, P43 4
I4, I41 8
[4/m]
</p>
<p>422 P422, P4212, P4122 8
P41212, P4222 8
P42212
P43212, P4322 8
I422, I4122 16
[4/m mm]
</p>
<p>Cubic 23 P23, P213 12
I23, I213 24
F23 48
[m3]
</p>
<p>432 P432, P4132, P4232 24
P4332 24
I432, I4132 48
F432, F4132 96
[m3m]
</p>
<p>Hexagonal 6 P6, P61, P65, P62 6
P64, P63 6
[6/m]
</p>
<p>622 P622, P6122, P6522 12
P6222, P642, P63 22 12
[6/m mm]
</p>
<p>Trigonala 3 P3, P31, P32 3
R3 3(9)
[�3] 6
</p>
<p>321, 312b P321, P3121, P3221 6
R32 6(18)
P312, P3112, P3212 6
[�3m]
</p>
<p>aTrigonal crystals are referred to hexagonal axes; there is no separate trigonal unit cell. The same choice can be made
</p>
<p>for rhombohedral crystals, in which case there are three times the number of molecules per unit cell; but the cell shape is
</p>
<p>easier to handle
bWhile point groups 321 and 312 are identical under a rotation of the symmetry elements by 30� with respect to the
crystallographic axes, space groups P321 and P312 are different. In the infinite array of a space group, the two space
</p>
<p>groups relate to different arrangements of points in space, that is, differing sets of general equivalent positions. Similar
</p>
<p>arguments apply to the pairs of point groups �3m1=�31m
</p>
<p>502 10 Proteins and Macromolecular X-Ray Analysis</p>
<p/>
</div>
<div class="page"><p/>
<p>1. Direct calculation of phases, using a known isomorphous protein structure
</p>
<p>2. MIR, Sect. 7.5.8
</p>
<p>3. MR, Sect. 8.3ff
</p>
<p>4. Multiple wavelength anomalous dispersion (MAD), Sect. 7.6.6
</p>
<p>5. Single isomorphous replacement with anomalous scattering (SIRAS), Sect. 7.6.5
</p>
<p>Calculated Structure Factors and R-Factors
Once a model of the structure has been proposed, values of the coordinates (x, y, z) for most of the
</p>
<p>non-H atoms will be available. It is then possible to calculate values of the structure factor amplitudes
</p>
<p>and phases based on the model. The quality of the model can be tested by calculating a conventional
</p>
<p>R-factor:
</p>
<p>R &frac14;
X
</p>
<p>jFo&eth;hkl&THORN; � jFc&eth;hkl&THORN;jj
h i.X
</p>
<p>Fo&eth;hkl&THORN; (10.1)
</p>
<p>where the summations are carried out over all reflections in the data set.
</p>
<p>For a protein structure R is rarely less than 0.35 for the initial model, improving to 0.2 or better
</p>
<p>after refinement. The quality of any X-ray structure is restricted by the diffracting power of the
</p>
<p>crystal. The poor diffracting power of protein crystals restricts both the quality and the total number
</p>
<p>of data available compared to the large number of parameters involved in the calculation of the
</p>
<p>structure factors, and is reflected in the poor R-factors attainable. By comparison the R-factor for a
</p>
<p>good low molecular weight X-ray structure would be less than 7%.
</p>
<p>Free R-Factor (Rfree)
When MR is used to determine a protein structure, a known homologous structure is fitted into the unit
</p>
<p>cell of the new (target) structure. It is common to use search structures which have as little as 40%
</p>
<p>identity with the sequence of the target structure and this is quite acceptable as long as the main chains of
</p>
<p>the protein fold in a similar way. The new structure is then refined by gradually transforming each amino
</p>
<p>acid in the sequence, in location and conformation, as necessary, in order to minimize the R-factor.
</p>
<p>Sometimes it is difficult to remove the initial bias, which is built into the new structure as a consequence
</p>
<p>of using the coordinates of another structure, a phenomenon known as feedback or &ldquo;memory,&rdquo; which is
</p>
<p>then built into the Fourier method.
</p>
<p>In order to monitor this process 5&ndash;10% of the Fo data are removed from the data set during the
</p>
<p>refinement process and are not allowed to contribute to the course of the analysis. This subset of data
</p>
<p>is used to calculate an R-factor called the Free R or Rfree [13, 14]. As the refinement proceeds, if
</p>
<p>jFc(hkl)j truly approaches Fo(hkl), Rfree will drop together with R. If the refined model fails to break
</p>
<p>away from the initial model, R will drop because its parameters are changing, but Rfree will fail to
</p>
<p>improve because the model is not actually improving. With a correct model, Rfree will usually
</p>
<p>decrease, but will remain a few percent greater than R.
</p>
<p>10.4.3 Recording X-Ray Diffraction from Macromolecular Crystals
</p>
<p>The recording and measurement of X-ray diffraction patterns are topics which have been thoroughly
</p>
<p>covered inChap. 5.Only points specifically of interest formacromolecular crystallographywill be touched
</p>
<p>upon here.
</p>
<p>10.4 Macromolecular Crystallography 503</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_5">http://dx.doi.org/10.1007/978-1-4614-3954-7_5</a></div>
</div>
<div class="page"><p/>
<p>Conventional X-Ray Laboratory Sources
In conventional crystallography laboratories, university and industrial departments, rotating anode
</p>
<p>generators are reliable sources of intense X-radiation in most crystallographic applications. For
</p>
<p>macromolecular studies, copper X-radiation in conjunction with a graphite monochromator and
</p>
<p>focusing mirrors give excellent results. Focussing mirrors as discussed in Sect. 5.9, are the more
</p>
<p>efficient alternative in practice as the graphite monochromator causes a large proportion of the incident
</p>
<p>radiation to be absorbed with a consequent loss of X-ray intensity. It should also be noted that rotating
</p>
<p>anode generators can be expensive to run, requiring constant pumping to high vacuum and frequent
</p>
<p>filament changes.
</p>
<p>Synchrotron X-Ray Sources
As discussed in detail in Sect. 5.4ff, X-rays are generated at a synchrotron source when high-energy
</p>
<p>electrons are accelerated in a storage ring at relativistic speeds. The X-ray beam is narrow and extremely
</p>
<p>intense, and the wavelength can be selected from a very wide range so as to match experimental
</p>
<p>requirements. In view of the high intensity and fine collimation, it is possible to record complete data
</p>
<p>sets even from small or weakly diffracting protein crystals in a matter of minutes. It is also possible to
</p>
<p>devise time-resolved experiments in order to monitor processes such as modified enzyme-substrate
</p>
<p>interactions. Synchrotron installations [15] are highly specialized research facilities. Experiments to be
</p>
<p>undertaken with synchrotron radiation (SR) require careful planning and in depth consultation with the
</p>
<p>on site staff. A list of world wide synchrotron installations is given in Sect. 11.4ff.
</p>
<p>X-Ray Cameras
Full details of the use of X-ray cameras in macromolecular crystallography are given in Sect. 5.4ff.
</p>
<p>Their use can lead to rapid and reliable evaluation of crystal quality, symmetry and unit cell
</p>
<p>parameters, and is an excellent tool for establishing a sound understanding of the reciprocal lattice.
</p>
<p>The availability of X-ray cameras in both university departments and other research laboratories is
</p>
<p>commonplace and is retained for both training and research. Modern applications employ electronic
</p>
<p>film devices to record and analyze the diffraction patterns in order to check for changes in both the
</p>
<p>unit cell and the intensity pattern in applications involving MIR.
</p>
<p>Diffractometers: Single Counter or Serial Diffractometers
Traditional diffractometers incorporate a mechanical goniometer to orientate the crystal into the
</p>
<p>correct position for each reflection and to rotate the counter, usually a scintillation counter to receive
</p>
<p>the scattered X-radiation from this single reflection. The energy is transformed electronically into a
</p>
<p>form suitable for conversion to intensity. Because each reflection is measured individually, with a
</p>
<p>count time typically of around 60 s, the process is very slow, particularly for proteins, which routinely
</p>
<p>involve the measurement of tens of thousands of reflections. While the accuracy attainable is better
</p>
<p>than for most of the other methods used for intensity measurement, the limited lifetime of protein
</p>
<p>crystals in the X-ray beam permits only a fraction of the available data to be recorded from one crystal
</p>
<p>The use of several crystals for data collection introduces errors associated with the scale factors
</p>
<p>required to merge the various collections into a single data set. Further details of this method are to be
</p>
<p>found in Sect. 5.6.4.
</p>
<p>Diffractometers: Area Detectors
The main disadvantages of single counter diffractometry, such as slow data collection rate and the
</p>
<p>requirement of several crystals for collection of a complete data set, with the attendant errors
</p>
<p>associated with scaling and crystal deterioration, have been largely overcome by the use of &ldquo;electronic
</p>
<p>film&rdquo; area detectors and image plates. They have enjoyed rapid development in recent years and have
</p>
<p>504 10 Proteins and Macromolecular X-Ray Analysis</p>
<p/>
</div>
<div class="page"><p/>
<p>been discussed in Sect. 5.7.3. The principal advantages of this methodwith proteins are that data can be
</p>
<p>collected to a resolution of about 1.4 Å using Cu radiation and better with synchrotron source, and that
</p>
<p>a very wide range of intensities (approximately 105) can be recorded compared to an X-ray film
</p>
<p>(approximately 200).
</p>
<p>10.4.4 Measurement of X-Ray Diffraction from Macromolecular Crystals
</p>
<p>Area Detectors in Practice: Autoindexing
As we have seen in Sect. 5.4ff, diffraction data are often collected now by the oscillation method, with
</p>
<p>the oscillation or rotation axis perpendicular to the X-ray beam. A number of software packages is
</p>
<p>available for processing the data recorded using the oscillation technique. These include HKL2000
</p>
<p>(which has superceded the very popular DENZO), andMOSFILMwhich is supported by CCP4. Both of
</p>
<p>these programs have excellent indexing routines, based on slightly different versions of the DPS
</p>
<p>algorithm; see Sect. 5.6.9. It has been recommended that it is sometimes useful to run both of these
</p>
<p>programs separately for initial assessment of a given set of data (see Appendix D). Other data processing
</p>
<p>software packages include d*TREK, and further developments include ELVES which is an expert
</p>
<p>system designed with the intention of taking the user from data collection frames to a protein structure
</p>
<p>without human intervention. Perhaps the lessons learnt in Sect. 11.7 will encourage crystallographers to
</p>
<p>treat this blind sort of approach with caution, and make every effort to understand fully the principles of
</p>
<p>X-ray crystallography and reap the undoubted rewards it will bring.
</p>
<p>A procedure known as autoindexing is designed to optimize data collection and processing. This
</p>
<p>method allows time to be saved by using the most efficient data collection procedure for a given
</p>
<p>situation. The autoindexing programs may be implemented on the basis of a single frame (or
</p>
<p>exposure), or better still, two frames of data collected at starting points 90� apart. This initial
information leads to a derivation of approximate unit-cell parameters, Laue group, crystal orientation
</p>
<p>(the crystal is not necessarily mounted about a major crystallographic axis), total oscillation range,
</p>
<p>and oscillation range per frame and, hence, the total number of frames to be used for the experiment.
</p>
<p>For a given exposure the crystal is oscillated through a small angle Df, set usually to a value between
</p>
<p>0.1 and 1.5�, depending on the unit cell size. The next frame begins where the previous one ended,
keeping Df at the same setting. The procedure is repeated until the required angular range has been
</p>
<p>covered. Once the X-ray intensities have been measured, structure analysis software usually assumes
</p>
<p>that the I(hkl) data set covers a single asymmetric unit of the Laue group symmetry, together with the
</p>
<p>Friedel equivalents where anomalous dispersion is being utilized.
</p>
<p>If time permits, it is a sound strategy to collect at least two asymmetric units of data and utilize the
</p>
<p>Rint index to monitor data quality. For a crystal arbitrarily mounted, 180
� of data frames will be more
</p>
<p>than sufficient for anything but a triclinic crystal. Such an arbitrary strategy can be wasteful of both
</p>
<p>diffractometer time and crystal lifetime and can be avoided using the programs carefully.
</p>
<p>Image Plate Data Processing
While the procedure for acquiring raw data using an image plate is quite rapid, and this itself is a
</p>
<p>tremendous advantage bearing in mind the fact that protein crystals have notoriously short lifetimes in
</p>
<p>the X-ray beam, processing this data to produce the Fo values can be very time consuming. Each
</p>
<p>frame of data contains information about the (X, Y, Z) coordinates of the diffraction maxima relative
</p>
<p>to the experimental set up, and digitized information from which a measure of the intensity can be
</p>
<p>derived. For a given crystal specimen the following procedures are carried out concomitantly:
</p>
<p>10.4 Macromolecular Crystallography 505</p>
<p/>
</div>
<div class="page"><p/>
<p>1. Refinement of the orientation matrix to define the disposition of the crystal axes with respect to the
</p>
<p>X-ray beam direction
</p>
<p>2. Confirmation of the crystal system and possible space groups; measurement and refinement to the
</p>
<p>highest accuracy possible of the unit-cell parameters and their estimated standard deviations
</p>
<p>3. Determination of the indices hkl and intensities I(hkl) for each recorded diffraction spot
</p>
<p>Partially Recorded Spots and Integration of Intensity Data
Because images are recorded from small oscillations of the crystal and the reciprocal lattice points for a
</p>
<p>protein crystal are close together, each image will contain a proportion of partially recorded reflections
</p>
<p>which are completed either on the previous or next record. To obtain the intensity reading, the two
</p>
<p>partials are added together. This effect will inevitably introduce errors in the intensity values. Recording
</p>
<p>of the diffraction image produces a series of digitized pixels, which can be viewed by computer graphics
</p>
<p>and processed to provide intensity estimates.
</p>
<p>Conversion to an Intensity Reading
In summation integration a volume containing an individual spot is defined. Summation of the
</p>
<p>measured counts in all the pixels in the defined volume and subtraction of the background, determined
</p>
<p>by examining surrounding pixels, leads to an integrated intensity. This method is similar to the
</p>
<p>scanning method used in processing the data from serial diffractometer measurements described in
</p>
<p>Sect. 5.5.
</p>
<p>In an alternative method of profile fitting, an empirically derived model reflection shape is scaled
</p>
<p>to the data and then integrated. This assumes that the reflection shape is independent of intensity. The
</p>
<p>observed profiles vary over the detector face, so that several model profiles are usually required,
</p>
<p>depending on the location on the detector face. Profile fitting is computationally expensive but
</p>
<p>produces more reliable results, and is less susceptible to random errors. The method was first
</p>
<p>proposed for one-dimensional profiles [16], extended to two-dimensional profiles for precession
</p>
<p>films [17], and later applied to oscillation photographs [18].
</p>
<p>Profile fitting depends on the assumption that strong and weak reflections share a common
</p>
<p>intensity profile, so that an observed reflection can be related to a &ldquo;standard&rdquo; profile for the area by
</p>
<p>a simple scale factor. In DENZO this is formulated as follows. The observed profile Mi is approxi-
</p>
<p>mated by Pi
</p>
<p>Pi &frac14; Cpi &thorn; Bi (10.2)
</p>
<p>where C is a constant to be determined, Bi is the predicted value of the background and pi is the
</p>
<p>predicted profile. The index i represents all pixels derived by scanning a spot in either a one-, two-, or
</p>
<p>three-dimensional profile. If the predicted profile is normalized,
P
</p>
<p>i pi &frac14; 1, then the constant is the
fitted intensity I, that is, I &frac14; C. Profile fitting minimizes the function
</p>
<p>&eth;Mi � Pi&THORN;2
vi
</p>
<p>(10.3)
</p>
<p>This solution means that each pixel provides an estimate of the spot intensity with variance vi. A
</p>
<p>profile fitted intensity is then the weighted average of all observations, thus:
</p>
<p>I &frac14;
</p>
<p>P
i &eth;p
</p>
<p>2
i =vi&THORN;&eth;Mi � Bi&THORN;=piP
</p>
<p>i p
2
i =vi
</p>
<p>&frac14;
</p>
<p>P
i pi&eth;Mi � Bi&THORN;=viP
</p>
<p>i p
2
i =vi
</p>
<p>(10.4)
</p>
<p>506 10 Proteins and Macromolecular X-Ray Analysis</p>
<p/>
</div>
<div class="page"><p/>
<p>Post-refinement
One of the aims in data collection is to measure the intensity of each indexed reflection as
</p>
<p>accurately as the experimental conditions allow. For partial reflections, accurate partiality and
</p>
<p>intensity fractions are necessary as described in the previous section. The parameters required for
</p>
<p>the measurement of intensities, such as cell dimensions and crystal orientation can be determined
</p>
<p>more accurately once an initial set of integrated intensities is available. This process of post-
</p>
<p>refinement, allows a more reliable classification of reflections as full or partial, and gives suffi-
</p>
<p>ciently accurate estimates of the intensity fraction of partial reflections for them to be scaled and
</p>
<p>used. The parameters involved include:
</p>
<p>1. Parameters determining the position of the reflection, such as cell dimensions, wavelength, and
</p>
<p>crystal orientation;
</p>
<p>2. Reflection width, which is precisely formulated [19];
</p>
<p>3. A function relating the angular fraction to the intensity fraction for which a model function is
</p>
<p>employed.
</p>
<p>Fully recorded reflections or sums of adjacent partials are used for the reference intensity of a
</p>
<p>partial reflection. Since these data and the flag that indicates whether a reflection is full or partial vary
</p>
<p>during refinement of the parameters mentioned above, as well as by scaling between frames, the post-
</p>
<p>refinement needs to be iterated with scaling several times over. There is a wide selection of software
</p>
<p>available for data processing. A possible procedure using mainly CCP4 software (see Appendix D) is
</p>
<p>indicated in the flow diagram in Fig. 10.9.
</p>
<p>Alternative software that may be used in the various stages of this suggested scheme:
</p>
<p>Stages (I), (II) and (III): DENZO
</p>
<p>Stage (V): SCALEPACK
</p>
<p>Stage (VIII): SCALEPACK2MTZ and CAD
</p>
<p>DENZO/XDISPLAY/SCALEPACK are part of the HKL suite. DENZO enables autoindexing,
</p>
<p>conversion to intensities and cell refinement; XDISPLAY displays the observed and calculated
</p>
<p>spot positions for comparison and validation; SCALEPACK carries out scaling, and merging on an
</p>
<p>iterative basis (post-refinement); MOSFILM is similar to DENZO but has the additional facility for
</p>
<p>spot size adjustment; REFIX allows MOSFILM to autoindex a single image; XDS or MARXDS
</p>
<p>include features of the programs mentioned previously that are required for MAR RESEARCH IPs;
</p>
<p>Fig. 10.9 Possible flow diagram for data processing
</p>
<p>10.4 Macromolecular Crystallography 507</p>
<p/>
</div>
<div class="page"><p/>
<p>MARSCALE is an additional scaling program (see Appendix D). Note. A number of software
</p>
<p>packages are available for processing data recorded using the oscillation technique. These include
</p>
<p>HKL2000 [20] (which has now superceded the very popular DENZO), and MOSFILM [21] which is
</p>
<p>supported by CCP4. Both of these programs have excellent indexing routines that are based on
</p>
<p>slightly different versions of the DPS algorithm [22]. It has been recommended that it is sometimes
</p>
<p>useful to run both of these programs separately for initial assessment of a given set of data [23]. The
</p>
<p>paper by Leslie [24] is essential reading for those about to embark on data processing. Other
</p>
<p>software packages are d*TREK [25] (sold with MSC X-ray detectors and evolved fromMADNESS)
</p>
<p>and PROTUEM a commercial program associated with Bruker detectors. ELVES [26] is an expert
</p>
<p>system developed with the intention of taking the user from data collection frames to a protein
</p>
<p>structure without human intervention.
</p>
<p>10.4.5 Problems with Data Collection and Suggested Cures
</p>
<p>There are certain problems that may be encountered with data collection from protein crystals; the
</p>
<p>main points are enumerated below, together with possible solutions.
</p>
<p>1. Diffraction is observed but the spots are wide apart and clear to very high resolution. This may
</p>
<p>indicate that the crystal is not protein but one of the small-molecule salts used in crystallization,
</p>
<p>very often ammonium sulfate. To establish this further, the click test should be tried.
</p>
<p>2. The crystals may appear to be well formed, optically clear, and with good morphology, and the
</p>
<p>diffraction pattern is typically protein with spots close together. Sometimes, however, the
</p>
<p>individual spots may be diffuse or partially diffuse, &ldquo;tailed&rdquo; or spread out in a tadpole shape,
</p>
<p>obviously split (twinned), or spread out in a particular direction, for example, a principal
</p>
<p>reciprocal lattice direction (statistical or systematic disorder). These symptoms are all indicative
</p>
<p>of poor crystal quality and in all such cases, it is not worth collecting data from these specimens.
</p>
<p>The crystallization conditions should be reviewed and possibly slowed down by cooling or
</p>
<p>adjusting the initial concentration. Crystallization at cold-room temperature or in a refrigerator
</p>
<p>may be tried. Crystal-mounting routines should be considered, as handling protein crystals too
</p>
<p>vigorously will lead to damage; physical contact with the crystals should be minimized. Other
</p>
<p>conditions should also be reviewed including storage time, because protein crystals can have a
</p>
<p>very short shelf life in crystallization trays, and also suffer from rough handling in transit.
</p>
<p>3. Sometimes crystals may be very small and refuse to grow, have good morphology, give good
</p>
<p>diffraction spots, but the intensities are weak and ymax is low. In such cases, the high-intensity SR
</p>
<p>should be tried, or cryo-cooling, or both.
</p>
<p>4. If any one of the unit-cell parameters is large (&gt;150 Å) diffraction spots may be weak and very
</p>
<p>close together on low intensity diffraction equipment. Again it would be better to use SR, which
</p>
<p>enables a larger crystal-to-image plate distance to be used, resulting in better spot separation while
</p>
<p>maintaining satisfactory image strength.
</p>
<p>5. If the crystals diffract well but their lifetime in the X-ray beam is too short to allow a full data set to
</p>
<p>be collected, again use of cryo-cooling to �100�C or even �150�C should be tried, together with
synchrotron radiation and a CCD detector, so as to obtain very rapid collection. Cooling not only
</p>
<p>promotes better resolution in many cases but also improves crystal lifetime almost indefinitely.
</p>
<p>Crystals are usually loop mounted, Sect. 10.3.2, and can be stored and transported in a Dewar flask
</p>
<p>contained in an insulated box.
</p>
<p>Cryo-cooling is the method of choice in many current applications. It has several advantages in
</p>
<p>addition to those listed above, including elimination of crystal slippage and drying out, both of which
</p>
<p>can occur with capillary tube mounting.
</p>
<p>508 10 Proteins and Macromolecular X-Ray Analysis</p>
<p/>
</div>
<div class="page"><p/>
<p>10.4.6 Preliminary Structure Determination: Unit Cell and Symmetry
</p>
<p>Before serious consideration can be given to the determination of the structure it is necessary to
</p>
<p>characterize the crystal system, determine the unit-cell dimensions and cell volume as accurately as
</p>
<p>possible, and identify possible space groups. The unit-cell parameters will rarely be determined to
</p>
<p>better than 0.02 Å for most protein crystals, unless extreme care is taken and the diffraction pattern
</p>
<p>extends to better than 1 Å resolution. Some image plates allow direct measurement of cell dimensions
</p>
<p>or other lattice spacings. Subsequent confirmation and least-squares refinement of the cell parameters
</p>
<p>using the data reduction software produces the final values together with an estimate of the errors. The
</p>
<p>space group of the crystal is determined from a consideration of the following.
</p>
<p>Unit-Cell Parameters
The cell parameter restrictions associated with each of the seven crystal systems have been listed in
</p>
<p>Table 2.3. It must be remembered that the symbol C=means &ldquo;not restricted by symmetry to equal,&rdquo; and
</p>
<p>consequently it is quite possible for the unit cell parameters of a crystal to appear to have higher
</p>
<p>symmetry than is actually the case. For example, a crystal which is actually monoclinic might have
</p>
<p>values of a, b, and g which are all 90� within experimental error; the crystal is then apparently
orthorhombic.
</p>
<p>Laue Symmetry
The diffraction pattern of a given crystal exhibits the symmetry of one of the 11 Laue Groups, Table
</p>
<p>1.6. Provided that enough data have been collected, Laue symmetry is usually reliably indicated as a
</p>
<p>further by-product of data reduction. This serves to pinpoint the correct crystal system and possible
</p>
<p>space groups, Table 10.1, that apply to the given crystal. Initially it is usually best to sample as large a
</p>
<p>portion of reciprocal space as possible in order to guarantee unambiguous determination of the Laue
</p>
<p>group.
</p>
<p>Systematically Absent X-ray Reflections
We show in Table 10.1 that there are several possible space groups for a crystal belonging to a given
</p>
<p>crystal system. For example, a monoclinic protein crystal could have one of the three space groups
</p>
<p>P2, P21, or C2; similar considerations apply to the other crystal systems. Further consideration of the
</p>
<p>X-ray diffraction pattern may enable the exact space group to be indicated. This procedure depends
</p>
<p>on the recognition of systematic absences, and has been considered at length in Sect. 3.7; specific
</p>
<p>related problems can be found at the ends of Chaps. 3 and 10. In general, screw-axis absences, Table
</p>
<p>2.5, must be determined from a careful consideration of intensity values, and may be indicated from
</p>
<p>graphical plots of the reciprocal lattice using software such as HKLVIEW (see Appendix D), which
</p>
<p>provides simulated precession photographs of the reciprocal lattice.
</p>
<p>10.4.7 Ricin Agglutinin
</p>
<p>Determination of the Space Group
Ricin Agglutinin [27] (RCA) belongs to the same family of proteins as the toxin Ricin [28], both
</p>
<p>being derived from Castor beans. From biochemical evidence, it is known that Ricin comprises a
</p>
<p>toxic A-chain and a lectin protein B-chain; the two chains are linked by an S&ndash;S bond to form an A&ndash;B
</p>
<p>heterodimer; RCA has two A-chains and two B-chains that are highly homologous with the Ricin
</p>
<p>chains but are linked as B&ndash;A&ndash;A&ndash;B. The molecular weight of RCA is thus about twice that of Ricin.
</p>
<p>10.4 Macromolecular Crystallography 509</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_2#Tab3_2">http://dx.doi.org/10.1007/978-1-4614-3954-7_2#Tab3_2</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Tab6_1">http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Tab6_1</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_10">http://dx.doi.org/10.1007/978-1-4614-3954-7_10</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_2#Tab5_2">http://dx.doi.org/10.1007/978-1-4614-3954-7_2#Tab5_2</a></div>
</div>
<div class="page"><p/>
<p>There is a possibility that the B&ndash;A&ndash;A&ndash;B assembly could sit across a twofold axis in the crystal either
</p>
<p>an exact space group twofold axis or an approximate, non-crystallographic twofold axis.
</p>
<p>During data processing the unit-cell parameters for RCA found were a &frac14; b &frac14; 100.05 Å, c
&frac14; 212.58 Å, a &frac14; b &frac14; 90�, g &frac14; 120�, and Laue symmetry 3. Together these observations show
that the crystals are trigonal, referred to hexagonal axes. In addition, limiting conditions in the
</p>
<p>diffraction pattern for 00l, l &frac14; 3n, indicate the presence of a 31 or 32 screw axis parallel to c, Table
2.5. Thus, the space group is either P31 or P32, Fig. 10.10 and Table 10.1. The X-ray analysis,
</p>
<p>Sect. 10.6.3, resolved this ambiguity, the space group being shown conclusively to be P32, Fig. 10.10.
</p>
<p>Note that 31 and 32 screw operations are left-hand&ndash;right-hand opposites; it follows that only one of the
</p>
<p>enantiomorphic pair of space groups, P31 or P32, can be correct for a given protein crystal. The
</p>
<p>question regarding the symmetry of the molecule was investigated using the self-rotation function and
</p>
<p>the result was confirmed using MR, Sect. 10.6.3.
</p>
<p>Resolution
The concept of resolution, Sect. 8.9, is perhaps more relevant in macromolecular crystallography than
</p>
<p>in the case of well diffracting small-molecule crystals, which usually diffract to atomic resolution.
</p>
<p>We may define dmin as the resolution of the X-ray diffraction pattern where, by the Bragg equation,
</p>
<p>dmin &frac14; l/2 sin ymax, ymax being the maximum value of y for reflections contained within the
measured data set. Atomic resolution corresponds to a dmin of about 0.8 Å, for copper radiation,
</p>
<p>which is usually out of reach for proteins.
</p>
<p>Most protein crystals fail to diffract to much better than 1.8 Å resolution. This is sufficient
</p>
<p>however to define the positions of non-hydrogen atoms and many of the solvent atoms in the
</p>
<p>structure. At 2.5 Å solvent atoms may be unreliable and at 3.5 Å it may be difficult to refine side-
</p>
<p>chain atoms beyond Cb. Obviously the higher the resolution the better defined the structure will be,
</p>
<p>and it is usually worthwhile to expend the required effort in order to achieve it rather than to cut
</p>
<p>corners and end up with an inferior analysis. The resolution, as we shall see below, seriously affects
</p>
<p>the quality of the initial electron density map upon which the ensuing analysis is based.
</p>
<p>Number of Reflections in and Completeness of the Data Set
For a given crystal it is possible to estimate the number of reciprocal lattice points within the range
</p>
<p>0 to ymax. For a crystal with a primitive lattice, ignoring the fact that some reflections will be related
</p>
<p>by symmetry, the number of reflections n is given by, Sect. 7.4.1,
</p>
<p>33:510Vc sin
3 ymax=l
</p>
<p>3 &frac14; 4:19Vc=d3min (10.5)
</p>
<p>Fig. 10.10 Space
group P32
</p>
<p>510 10 Proteins and Macromolecular X-Ray Analysis</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_2#Tab5_2">http://dx.doi.org/10.1007/978-1-4614-3954-7_2#Tab5_2</a></div>
</div>
<div class="page"><p/>
<p>Depending on how the X-ray data set has been measured, the number of reflections present may
</p>
<p>not be consistent with the number expected from the nominal value of ymax, or dmin. If the number
</p>
<p>of reflection data actually measured is significantly lower than the expected value, it can be
</p>
<p>assumed that the resolution of the analysis will be correspondingly lower although it is not easy
</p>
<p>to estimate by how much. The cause may also be due to an anisotropic intensity distribution,
</p>
<p>associated with the variation of diffracting power with direction. Some data processing programs
</p>
<p>put out a warning when such an effect is detected. The percentage completeness at nominal
</p>
<p>resolution will also be indicated. A value less than about 85% complete indicates the need to
</p>
<p>investigate the causes for the loss of data.
</p>
<p>From the results so far for RCA, it follows that the unit-cell volume is 1.843 � 106 Å3. A data set
was recorded on a 180 mm diameter Mar Image Plate, with a crystal-to-detector distance of 300 mm.
</p>
<p>The resolution dmin of the data was 3.6 Å for an X-ray wavelength of 1.5418 Å. From the results
</p>
<p>above, the expected number of reflections is 165,513. This number includes all symmetry-related
</p>
<p>reflections. Since the unit cell is P and Laue symmetry �3, we divide this number of reflections by G,
</p>
<p>which is unity, Table 3.1, and by the multiplicity of planes m, namely 6, to obtain an estimate of the
</p>
<p>number of unique data, that is, not related by symmetry, and which is therefore 27585. Experimen-
</p>
<p>tally, 62310 reflections were recorded. The data processing program reported that the multiplicity
</p>
<p>factor for the recorded data was 3.1, indicating that 62310/3.1, or 20100 reflections were unique. The
</p>
<p>data set is therefore said to be 20100/27585, or 73% complete at 3.6 Å resolution. We can assume that
</p>
<p>the effective resolution of the data is therefore less than the nominal resolution.
</p>
<p>Internal Consistency, Space Group Ambiguities, and Rint
As a test of data quality, data processing programs produce an internal consistency index, Rint,
</p>
<p>(4.19), which is calculated by comparing I(hkl) values that should be equal by virtue of symmetry.
</p>
<p>To be acceptable, Rint for the complete measured data set for a protein should be 9% or better; for a
</p>
<p>well diffracting small molecule Rmerg should be less than 7%, and is frequently 4% or better. Since
</p>
<p>the mean value of I(hkl) decreases as the Bragg angle y(hkl) increases, errors in the data will be
</p>
<p>more significant for the higher resolution data, that is, for smaller d(hkl). For the outer shells of
</p>
<p>protein data, therefore, the local value of Rint might be as high as 20%. Although this value may
</p>
<p>seem to be inordinately high, such data may still contain a great deal of structural information and
</p>
<p>may serve a useful purpose in the analysis if retained, albeit with a low weight.
</p>
<p>In addition to providing an indication of data quality, the merging R-factor can serve as a means of
</p>
<p>resolving space group ambiguities. By assigning a Laue symmetry, Table 1.6, for the crystal that is
</p>
<p>higher than true, such as mmm instead of 222, falsely high values of Rintwill result. Dropping down to
</p>
<p>the correct Laue symmetry should produce better consistency and lower Rint values, but only if a
</p>
<p>sufficient portion of reciprocal space has been covered during the data collection. If this is not the case
</p>
<p>further data are needed.
</p>
<p>Number of Molecules Per Unit Cell
Each of the possible protein space groups is associated with an expectation number of molecules
</p>
<p>per unit cell, given as Z in Table 10.1. If the actual number n is greater than the expectation value,
</p>
<p>again usually by a simple factor, the protein structure is said to be oligomeric and contains more
</p>
<p>than one molecule in the asymmetric unit, not related to the other by crystal symmetry; in the case
</p>
<p>of proteins frequently an approximate NCS relates such molecules. In this case the X-ray analysis
</p>
<p>will involve the determination of the structures of this number of molecules rather than just one.
</p>
<p>The molecules in this type of situation usually exhibit subtle differences in molecular conforma-
</p>
<p>tion, particularly in loop regions.
</p>
<p>10.4 Macromolecular Crystallography 511</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Tab1_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Tab1_3</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_4#Eq19_4">http://dx.doi.org/10.1007/978-1-4614-3954-7_4#Eq19_4</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Tab6_1">http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Tab6_1</a></div>
</div>
<div class="page"><p/>
<p>From Sect. 7.2, we express the crystal density as
</p>
<p>Dc &frac14; nMrmu=Vc (10.6)
</p>
<p>where the parameters have meanings as before. If Vc is expressed in Å
3, Dc will have units of g cm
</p>
<p>�3.
</p>
<p>In this relationship, both n and Mr, which involves the unknown solvent component, are unknown.
</p>
<p>However, Dc can be assumed, from experience, to lie in the range 1.2&ndash;1.4 g cm
�3 for crystalline
</p>
<p>compounds composed predominantly of the elements C, N, O, and H; the actual value will depend
</p>
<p>heavily on the degree of solvation.
</p>
<p>Let Mr &frac14; MP + sMr, where MP is the relative molar mass of the protein, which is 133000 and s is
</p>
<p>the solvent fraction of the cell mass. ThusMr &frac14; MP/(1 &ndash; s), so that Dc &frac14; nMPmu/[Vc(1 &ndash; s)]. IfMS is
the unknown molecular mass incorporated in the solvent, we define the fraction s as MS/(MP + MS),
</p>
<p>the solvent fraction of the unit-cell mass. Finally if the crystal space group is known, as would be
</p>
<p>normal after data reduction, then since we know that n will be a multiple or submultiple, say m, of Z,
</p>
<p>the number of asymmetric units per unit cell, Table 10.1, it follows that n &frac14; mZ. The density
</p>
<p>expression (10.6) then becomes:
</p>
<p>Dc &frac14; &eth;mZ&THORN;MPmu=&frac12;Vc&eth;1� s&THORN;� (10.7)
</p>
<p>It is known [29&ndash;31] that in crystalline proteins, s is usually in the range 0.27&ndash;0.65. Putting all of
</p>
<p>this information together it is usually possible to determine the appropriate value for m, the number of
</p>
<p>protein molecules per asymmetric unit. These ideas are illustrated by the following example.
</p>
<p>Analysis of the Solvent Content in Ricin Agglutinin
We now continue our study of RCA by investigating the possible values for the parameters m and s.
</p>
<p>Since Vc is 1.843 � 106 Å3, the value for mu here is 1.6605, Z, Table 10.1, is 3 for space group P32,
and using the equation for Dc developed above and with MP &frac14; 133000, we have
</p>
<p>Dc &frac14; 1:6605&eth;mZ&THORN;MP=&frac12;Vc&eth;1� s&THORN;� &frac14; 0:36m=&eth;1� s&THORN; (10.8)
</p>
<p>Assuming m &frac14; 1 RCA molecule per asymmetric unit (3� B&ndash;A&ndash;A&ndash;B per unit cell) and s &frac14; 0.65
(a guess, at the top of the range because the crystals are wet and poorly diffracting), we find that Dc is
</p>
<p>1.03 g cm�3. Since this is a very low value for the density, we try s &frac14; 0.7, which gives Dc &frac14; 1.20
g cm�3, a more realistic value. The physical wetness of the crystals and relatively poor diffracting
</p>
<p>power are consistent with the result that they are highly solvated. As we shall see in the next section,
</p>
<p>RCA has a non-crystallographic twofold axis relating the two halves of the B&ndash;A&ndash;A&ndash;B assembly, and
</p>
<p>this information is used in the X-ray analysis. In addition, the following alternative must be borne in
</p>
<p>mind as the X-ray analysis progresses: assuming m is 2, Dc equals 1.20 g cm
�3 for s &frac14; 0.40, which is
</p>
<p>approximately equal to the average solvation level found in protein crystals, and should therefore be
</p>
<p>considered reasonable until proved otherwise.
</p>
<p>10.5 Types of Fourier Synthesis for Protein Analysis
</p>
<p>10.5.1 Reconstruction of the Molecular Structure
</p>
<p>We have seen in Sect. 6.3ff that the calculation of electron density maps is a major objective of any single
</p>
<p>crystal X-ray analysis, and we recall the formula for calculating the electron density r(xyz) at the general
</p>
<p>point with fractional coordinates (x, y, z):
</p>
<p>512 10 Proteins and Macromolecular X-Ray Analysis</p>
<p/>
</div>
<div class="page"><p/>
<p>r&eth;xyz&THORN; &frac14; 1
Vc
</p>
<p>Xhmax
</p>
<p>hmin
</p>
<p>Xkmax
</p>
<p>kmin
</p>
<p>Xlmax
</p>
<p>lmin
</p>
<p>Fo&eth;hkl&THORN; cos 2p&frac12;&eth;hx&thorn; ky&thorn; lz&THORN; � f&eth;hkl&THORN;� (10.9)
</p>
<p>For protein structures the phases can be determined using the experimentally intense method ofMIR,
</p>
<p>or from a model derived byMR. The theory of MIR has been discussed in Sect. 7.5.8 and MR treated in
</p>
<p>detail in Sect. 8.3.1; other important methods are to be found in this chapter. However, it is important to
</p>
<p>remember that the quality of an electron densitymap is strongly influenced by the accuracy of the phases.
</p>
<p>Poor phases will never produce a good map, whatever the resolution.
</p>
<p>Most contemporary software uses the method of fast Fourier transform (FFT) for rapid calculation
</p>
<p>of electron density maps [32&ndash;34]. Related algorithms are used for calculation of structure factors.
</p>
<p>Programs for calculation and display of electron density and other Fourier series are listed in
</p>
<p>Appendix D. Useful hints on the calculation and interpretation of electron density maps have been
</p>
<p>given elsewhere [35].
</p>
<p>Properties of the Electron Density
We summarize here the important properties of the electron density function that we need to keep in
</p>
<p>mind when interpreting it in terms of a crystal structure:
</p>
<p>1. When calculated at atomic resolution, r(xyz) has local maximum values (peaks) at sites
</p>
<p>corresponding to atom centers (xj, yj, zj).
</p>
<p>2. Density values at these locations are approximately proportional to Zj, the atomic number of the
</p>
<p>corresponding jth atom.
</p>
<p>3. The locations of all atom positions together define the crystal structure. Atom positions not
</p>
<p>determined (for whatever reason) lower the quality of the structure model.
</p>
<p>4. For proteins, where the extent of the X-ray data is usually far short of atomic resolution, the
</p>
<p>electron density tends to be blurred, or unfocused. Atom coordinates have to be inferred in this
</p>
<p>case, usually by model building, which makes extensive use of the known molecular geometry of
</p>
<p>bonds and groups involved in the structures.
</p>
<p>5. Since clearly defined peaks are not a property of protein electron density, the magnitude of given
</p>
<p>density regions is assessed in terms of the overall root mean square value of r(xyz). Significant
</p>
<p>density is taken to have r(xyz) &gt; 3sr, although a value of 2sr (or even lower) can have useful
</p>
<p>features but requires very careful consideration.
</p>
<p>10.5.2 Difference Electron Density
</p>
<p>Difference electron density may be defined as
</p>
<p>Dr&eth;xyz&THORN; &frac14; 1
Vc
</p>
<p>Xhmax
</p>
<p>hmin
</p>
<p>Xkmax
</p>
<p>kmin
</p>
<p>Xlmax
</p>
<p>lmin
</p>
<p>DjF&eth;hkl&THORN;j cos2p&frac12;&eth;hx&thorn; ky&thorn; lz&THORN; � fc&eth;hkl&THORN;� (10.10)
</p>
<p>where DjF&eth;hkl&THORN;j &frac14; Fo&eth;hkl&THORN; � jFc&eth;hkl&THORN;j. This function exhibits positive density corresponding to
atoms not included in the model, and negative density where atoms are missing from the model or
</p>
<p>badly placed in it.
</p>
<p>10.5 Types of Fourier Synthesis for Protein Analysis 513</p>
<p/>
</div>
<div class="page"><p/>
<p>10.5.3 The 2Fo(hkl)&ndash;jFc(hkl )j Map
</p>
<p>By adding r&eth;xyz&THORN;&thorn;Dr&eth;xyz&THORN;we obtain a new type ofmap known simply as a 2&eth;Fo&eth;hkl&THORN; � jFc&eth;hkl&THORN;j&THORN;map
represented by the expression
</p>
<p>1
</p>
<p>Vc
</p>
<p>Xhmax
</p>
<p>hmin
</p>
<p>Xkmax
</p>
<p>kmin
</p>
<p>Xlmax
</p>
<p>lmin
</p>
<p>&frac12;2Fo&eth;hkl&THORN; � jFc&eth;hkl&THORN;j� cos 2p&frac12;&eth;hx&thorn; ky&thorn; lz&THORN; � fc&eth;hkl&THORN;� (10.11)
</p>
<p>Maps of this type are very useful for locating parts of the structure which are not part of the current
</p>
<p>model; they show positive density corresponding to missing atoms and negative density
</p>
<p>corresponding to incorrect atoms against the background of the original model. It provides a very
</p>
<p>useful basis for making corrections and additions to the structure model.
</p>
<p>Omit Maps
An electron density function where part of the model has been deliberately left out is called an omit
</p>
<p>map. Reasons for such omissions include disordered or apparently disordered regions, areas of poor
</p>
<p>stereochemistry, ambiguous side-chains (some pairs of amino acids are very similar), or simply parts of
</p>
<p>the model that are difficult to explain. The omitted parts of the structure should be returned in a
</p>
<p>subsequent electron density calculation as peaks of about half the height of comparable regions included
</p>
<p>in the phasing. An example of a contoured difference electron density map is given later in Fig. 10.19.
</p>
<p>10.6 Determination of the Phases for Protein Crystals
</p>
<p>The Patterson function has been discussed in detail including its use in MIR for location of heavy
</p>
<p>atoms and for locating the position of the search model in an unknown structure in the MR method.
</p>
<p>We discuss now some further uses of the Patterson function for these two methods.
</p>
<p>10.6.1 Introduction
</p>
<p>In the early years of protein structure analysis, the period 1950&ndash;1970, MIR was the primary choice for
</p>
<p>protein structure analysis, and a core of good quality protein structures was established. During this
</p>
<p>time MR became increasingly popular, as the number of well defined structures from which to select a
</p>
<p>search model gradually increased. There are currently more than 19000 protein structures deposited
</p>
<p>in the Protein Data Bank (see Appendix D). Thus, MR is the current method of choice for any protein
</p>
<p>having good structural homology, as judged initially by the amino acid sequence and quality of the
</p>
<p>search structure. The more experimentally demanding MIR technique is still used for proteins where
</p>
<p>this condition does not apply. The advent of fast, large capacity computers, and improved software
</p>
<p>has also contributed significantly to the ease of application of MR.
</p>
<p>10.6.2 Isomorphous Replacement (MIR)
</p>
<p>We know that by incorporating one or more heavy atoms into a protein crystal to form a heavy-atom
</p>
<p>derivative, induces changes in the X-ray intensities. For the derivative crystal to be isomorphous three
</p>
<p>conditions must hold:
</p>
<p>514 10 Proteins and Macromolecular X-Ray Analysis</p>
<p/>
</div>
<div class="page"><p/>
<p>1. The heavy atoms must not disturb the protein structure significantly.
</p>
<p>2. The crystallographic space group must remain unaltered.
</p>
<p>3. The unit-cell parameters must not differ significantly (less than 0.5% in any cell length).
</p>
<p>Useful heavy atoms usually have atomic numbers 53 (iodine) or greater. The average fractional
</p>
<p>change in intensity DI(hkl)/I(hkl) at sin y/l &frac14; 0 can be estimated as:
P
</p>
<p>NH
Z2H=
</p>
<p>P
NP
</p>
<p>Z2P where the
</p>
<p>summations are over the number of heavy atoms NH per unit cell and the number of non-heavy atoms
</p>
<p>NP per unit cell, respectively. For a protein crystal containing a protein of relative molar mass 20000
</p>
<p>with 1 molecule per unit cell, NP would be equivalent to about 3200 carbon atoms. If NH represents
</p>
<p>two atoms of mercury (atomic number 80) per unit cell, then DI&eth;hkl&THORN;=I&eth;hkl&THORN; &frac14; &eth;2� 802&THORN;=
&eth;3200� 62&THORN;, or 11 %. In practice some I(hkl) values are increased and some are decreased; intensity
differences of the order illustrated here would be useful for phasing.
</p>
<p>Heavy Atoms and Compounds for Isomorphous Replacement
There is a very large repertoire of compounds [8, 9, 36] known to produce heavy-atom derivatives of
</p>
<p>proteins that are suitable for MIR. Heavy atoms are usually non-covalently linked to the native
</p>
<p>protein molecule, often in surface pockets or other easily accessible regions of the protein structure.
</p>
<p>The most popular heavy atoms are platinum, mercury, gold, silver, and uranium. The selection of
</p>
<p>appropriate compounds to screen for useful heavy-atom derivatives depends to a great extent on
</p>
<p>knowledge of the amino acid sequence.
</p>
<p>10.6.3 Preparation and Screening of Heavy-Atom Derivatives
</p>
<p>Two methods are used for the preparation of heavy-atom derivatives for isomorphous replacement,
</p>
<p>namely, soaking pre-grown native crystals in heavy-atom solutions, and co-crystallization of the protein
</p>
<p>and heavy atoms together from solution.
</p>
<p>Soaking Method
Soaking can be carried out on crystals in hanging or sitting drops or on crystals mounted for X-ray
</p>
<p>diffraction in glass capillaries, Sect. 10.2. This method is highly sensitive and dependent on
</p>
<p>correct pH, concentration, temperature and arid time. Heavy-atom solution concentrations around
</p>
<p>0.5&ndash;10 M are frequently used with soaking times ranging from a few minutes to several days or
</p>
<p>longer. This method can cause, often-visible, deterioration of crystal quality, but is not likely to
</p>
<p>induce changes in unit cell and symmetry.
</p>
<p>Co-crystallization in the Presence of Heavy Atoms
Generally speaking, the introduction of heavy atoms by co-crystallization will initially follow the
</p>
<p>method established for crystallization of the native protein, Sect. 10.2. Co-crystallization can be
</p>
<p>undertaken in situ in hanging or sitting drops, or in test tubes where batch crystallization is used. Co-
</p>
<p>crystallization may produce better quality crystals than soaking, but symmetry and unit-cell changes
</p>
<p>are more likely to occur as a consequence of the different crystallization conditions, including the
</p>
<p>presence of heavy atoms, and is therefore not usually the first method of choice.
</p>
<p>Screening Possible Derivatives
All possible heavy-atom derivatives, produced as described in the previous section, have to be
</p>
<p>tested for isomorphism, for intensity changes that would indicate successful incorporation of heavy
</p>
<p>10.6 Determination of the Phases for Protein Crystals 515</p>
<p/>
</div>
<div class="page"><p/>
<p>atoms and for crystal quality. This screening of possible heavy-atom derivative crystals can be time
</p>
<p>consuming. The easiest method of detecting changes in I(hkl) is by comparing X-ray diffraction
</p>
<p>records covering the same region of reciprocal space, Sect. 5.6.4, previously achieved via preces-
</p>
<p>sion photographs. The same diffraction records can provide a sensitive check on symmetry and
</p>
<p>unit-cell dimensions. Alternatively low resolution, 5&ndash;6 Å, data sets can be measured for all the
</p>
<p>crystals of interest but this will be time consuming, particularly if no heavy atoms have been
</p>
<p>introduced.
</p>
<p>How Many Derivatives Are Required
Wehave discussed in Sect. 7.5.8 the procedures for locating heavy atoms and deriving phase information
</p>
<p>inMIR, and it is advisable to use at least three good heavy-atom derivatives for phasing purposes. Useful
</p>
<p>phase information can be obtained with two derivatives and to a much lesser degree, with a single
</p>
<p>derivative, but the quality of the calculated electron density will suffer.
</p>
<p>The Initial MIR Model
As in direct methods phasing in small-molecule analysis, MIR phases are not calculated from the
</p>
<p>structure factors of a model, but from probability functions. Thus, MIR phases are free from model
</p>
<p>bias, unlike those fromMR. As little or no information may be available about the secondary structure
</p>
<p>of the protein, unless intensity data have been measured to a resolution of 1 Å or better, which is
</p>
<p>unlikely, it is imperative that a detailed knowledge of the amino acid sequence of the protein is
</p>
<p>available. The MIR phases are used to calculate the electron density distribution, which may first be
</p>
<p>enhanced by modification techniques (Sect. 10.8.2) and interpreted using molecular graphics,
</p>
<p>Sect. 10.8.3.
</p>
<p>If the phases are of sufficiently high quality the main-chain density should be strong and continu-
</p>
<p>ous, and on average greater than 3s(r), thus enabling this part of the structure to be modeled. Once
</p>
<p>the ends of the polypeptide chain have been identified, its course through the electron density can be
</p>
<p>retraced and side-groups fitted to the model and added to the coordinate list. This lengthy process will
</p>
<p>then be followed by further refinement, Sects. 10.8 and 10.9. Figure 10.11 is a suggested flow diagram
</p>
<p>for MIR.
</p>
<p>10.6.4 Molecular Replacement (MR)
</p>
<p>Introduction
In contrast to MIR, MR is not such an experimentally intense method of phase determination. The
</p>
<p>initial requirements of this method are as set down under Sect. 8.3.1. In essence MR [37] involves a
</p>
<p>critical and quantitative comparison of the Patterson functions of the target and search models. In
</p>
<p>contemporary software for carrying out MR applications (AmoRe [38, 39], X-PLOR [40], MOL-
</p>
<p>REP [41, 42], CNS [43] and PHASER [44, 45]), the method is sometimes strengthened through
</p>
<p>the availability of other crystallographic techniques, as described below, which are now within the
</p>
<p>capabilities of modern computers. For example, the Fast Rotation Function has been developed by
</p>
<p>Jorge Navaza [46]. The programs MERLOT, ALMN, POLARRFN [47] are part of the CCP4 suite,
</p>
<p>(q.v.), MOLREP [41, 42] and AmoRe [38, 39] use the Fast Rotation Function. CNS [43] and
</p>
<p>X-PLOR [40] use the conventional Rotation Function (RFn). The programs BEAST [48] and
</p>
<p>PHASER [44, 45], a highly speeded up version of BEAST, use novel maximum likelihood
</p>
<p>functions.
</p>
<p>516 10 Proteins and Macromolecular X-Ray Analysis</p>
<p/>
</div>
<div class="page"><p/>
<p>The following comprehensive list summarizes commonly available MR programs:
</p>
<p>Fig. 10.11 Flow diagram
for single or multiple
</p>
<p>isomorphous replacement
</p>
<p>PHASER Has rotation and translation functions using a hybrid maximum likelihood scoring method with
</p>
<p>traditional fast Rotation and Translation Functions. The program has a good operating speed and is
</p>
<p>readily available in the Phenix and CCP4 suites
</p>
<p>MOLREP This program has rotation and translation functions with many of the features of AmoRe, and includes
</p>
<p>special features for dealing with multiple molecules in the asymmetric unit, and is part of the CCP4 suite
</p>
<p>EPMR This program uses a genetic algorithmic 6D search and is very popular but quite slow. The new
</p>
<p>OpenEPMR is readily available to users
</p>
<p>AmoRe This program has rotation and translation functions which include facilities for searching long lists of
</p>
<p>possible solutions. It is very fast and highly recommended. It can search for more than one model of more
</p>
<p>than one type to assemble multidomain solutions and is currently part of CCP4
</p>
<p>(continued)
</p>
<p>10.6 Determination of the Phases for Protein Crystals 517</p>
<p/>
</div>
<div class="page"><p/>
<p>Of these programs AmoRe, which is very fast, and PHASER are both highly recommended. Other
</p>
<p>programs may have to be considered if these two fail: MOLREP is probably just as powerful but less
</p>
<p>user friendly. The following web pages may also be examined:
</p>
<p>&bull; &ldquo;An introduction to molecular replacement,&rdquo; from the CCP4 workshop proceedings (Phil Evans).
</p>
<p>&bull; Molecular Replacement course from Cambridge Institute for Medical Research.
</p>
<p>&bull; Molecular Replacement example, with some theory and practice.
</p>
<p>&bull; Molecular Replacement programs page UCLA.
</p>
<p>&bull; &ldquo;Making the most of your search model&rdquo; (Gerard Kleywegt).
</p>
<p>&bull; PHASER v2.1 home page and documentation.
</p>
<p>&bull; Mr BUMP home page.
</p>
<p>&bull; BALBES home page.
</p>
<p>&bull; CCP4&rsquo;s AmoRe, POLARRFN, ALMN and program list (CHAINSAW, CAD, etc.).
</p>
<p>&bull; BEAST reprint in pdf.
</p>
<p>For complex structures like proteins, interatomic vectors are densely packed in the unit cell and
</p>
<p>most will not be resolved in the Patterson map. Lack of atomic resolution in the X-ray data will also
</p>
<p>cause a further blurring of the vector distribution density. In the rotation stage of MR, the shorter
</p>
<p>intramolecular and interatomic vectors are used to determine possible orientations for the search
</p>
<p>molecule in the unit cell of the target molecule; the best subset of solutions is retained. The
</p>
<p>translation stage seeks to optimize the actual location of each of these selected possible molecular
</p>
<p>orientations in the target unit cell. Again the best solutions are retained for further development.
</p>
<p>Ideally there will be an outstanding solution, which will eventually yield a refineable structure for
</p>
<p>the target protein.
</p>
<p>Self-Rotation Function and Non-crystallographic Symmetry
A complication can arise in practice for target crystal structures containing more than one protein
</p>
<p>molecule in the asymmetric unit. A clue to the presence of NCS may already have been given by the
</p>
<p>preliminary analysis of unit-cell contents, Sect. 10.4.6, but ambiguities can arise, mainly because of
</p>
<p>uncertainties in solvent content. In practice it is necessary to establish the presence of NCS as early as
</p>
<p>possible in the MR analysis, as this can save time by allowing the correct procedures to be followed.
</p>
<p>Calculation of the self-rotation function may help to resolve, or at least to provide clues as to, the
</p>
<p>actual situation with a given protein crystal.
</p>
<p>CNS Uses Patterson-based RFn and TFn implementations which tend to run slowly. Its main advantage is PC
</p>
<p>refinement which allows optimization of internal degrees of freedom in a model prior to the translation
</p>
<p>function
</p>
<p>BEAST This program has rotation and translation functions which employ a novel maximum likelihood scoring
</p>
<p>method. It was essentially the precursor to PHASER
</p>
<p>GLRF The mnemonic stands for Global Locked Rotation Function; it allows inclusion of non-crystallographic
</p>
<p>symmetry constraints in MR searches. The programs REPLACE and COMO are probably derived from
</p>
<p>this program
</p>
<p>Mr BUMP An automated CCP4 Molecular Replacement program using MOLREP or PHASER
</p>
<p>BALBES MR replacement
</p>
<p>MERLOT This was the first MR program suite. It is now completely supplanted by PHASER, AmoRe and
</p>
<p>MOLREP
</p>
<p>ALMN Fast RFn program, using the Crowther algorithm for cross-rotation searches. Results expressed as
</p>
<p>Eulerian angles; it is a part of the CCP4 suite
</p>
<p>POLARRFN Fast RFn program using the Crowther algorithm for self- and cross-rotation searches. Results are
</p>
<p>expressed as Polar angles. Most useful for self-rotation searches where the angle system makes the
</p>
<p>symmetry easier to interpret, and part of CCP4 suite
</p>
<p>518 10 Proteins and Macromolecular X-Ray Analysis</p>
<p/>
</div>
<div class="page"><p/>
<p>Molecules related by NCS frequently differ in both position and spatial orientation with respect to
</p>
<p>each other and are often, but by no means always, related by pseudo twofold rotational symmetry.
</p>
<p>The pseudo-rotation axis will not coincide with any of the space group symmetry elements and, of
</p>
<p>course, will not necessarily involve an exact 180� rotation. Other types of rotational NCS can also
occur. In such cases, the Patterson function will include, around the origin, a differently oriented copy
</p>
<p>of the molecular interatomic vectors for as many protein molecules as are involved in the NCS.
</p>
<p>Consequently if two exact copies of the Patterson function are superimposed origin to origin,
</p>
<p>then by rotating one copy of the replica correctly with respect to the other, the peak distributions
</p>
<p>can be made to coincide, thus determining the relative orientations of the NCS related molecules.
</p>
<p>For this correct relative orientation, the rotation function will have a significantly large local
</p>
<p>maximum value.
</p>
<p>Programs are available for plotting the results of this type of self-rotation and for calculating the
</p>
<p>relative molecular orientations in three dimensions. Initially the crystal system is orthogonalized as
</p>
<p>shown in the example in Fig. 10.12, and spherical polar angles with respect to this axial system are
</p>
<p>used to define orientation, instead of the usual Eulerian angles. In this way, as shown in Fig. 10.13, it
</p>
<p>is possible to specify a single rotation w (also called k) about an axis whose direction is defined by the
</p>
<p>spherical polar anglesC (also called o) and f. The pointC &frac14; 0� on the pole projects on to the center
of the circular plot, and points on the equator with C &frac14; 90� project on to the circumference. Points
with f &frac14; 0� project on to the X1-axis, or meridian; f increases in an anticlockwise direction. For
orthogonal unit cells, some programs place the crystallographic a-axis at f &frac14; 0�, the b-axis at
f &frac14; 90� and the c-axis at C &frac14; 0�. Other conventions are used, however, and each program should
be checked by the prospective user. When working in a monoclinic space group, it is convenient to
</p>
<p>place the unique y-axis along the polar direction. The use of spherical polar angles in this type of
</p>
<p>analysis enables sections of constant w to be plotted.
</p>
<p>The two most common cases of NCS that are found in practice are pseudo twofold axes,
</p>
<p>corresponding to w &frac14; 180�, and pseudo threefold axes, corresponding to w &frac14; 120�. Other sections
</p>
<p>Fig. 10.12 Relationship of the orthogonal axes X1, X2, X3, used in calculating the self-rotation function, and the
generalized crystallographic axes a1, a2, a3: X2 is along a2, X1 is normal to a2 and a3, and X3 is normal to both X1 and X2
</p>
<p>10.6 Determination of the Phases for Protein Crystals 519</p>
<p/>
</div>
<div class="page"><p/>
<p>of constant w can be accommodated as required. In practice several self-rotation plots should be
</p>
<p>calculated by varying the radius of integration, Sect. 8.3.2, and/or the resolution cut-off applied to the
</p>
<p>Fo(hkl) input data. It is important to recognize features of the rotation function which persist under
</p>
<p>these different conditions as they are most likely to contain the information sought. Both of these
</p>
<p>parameters can be specified in the data input to the program. The use of plots with different radii of
</p>
<p>integration enables the effect of eliminating cross vectors, which are the longer interatomic vectors in
</p>
<p>the Patterson function, to be seen. It is of course not possible to eliminate all cross vectors, as the
</p>
<p>molecules in a crystal are packed close together. As a rule of thumb, for an approximately spherical
</p>
<p>molecule a search radius of 75&ndash;80% of the molecular diameter will include about 90% of the self-
</p>
<p>vectors. Non-spherical molecules present more of challenge.
</p>
<p>10.6.5 Example of a Self-Rotation Function: Ricin Agglutinin
</p>
<p>We have seen that ricin agglutin (RCA) probably has a twofold NCS axis across the B&ndash;A&ndash;A&ndash;B
</p>
<p>assembly of protein chains. Figure 10.14 shows the section of the self-rotation function at w &frac14; 180�,
calculated with the program POLARRFN, and includes data to 4.0 Å resolution. For space group P32
the symmetry of this section must contain a threefold axis, corresponding to the symmetry of Laue
</p>
<p>group 3. The presence of non-crystallographic twofold axes is exemplified by the very large peak,
</p>
<p>(arrowed) on the stereogram, with spherical polar coordinates of approximately f &frac14; 20,C &frac14; 90, and
w &frac14; 180�. This peak corresponds to a pseudo twofold axis perpendicular to the c-axis, and at
approximately 20� to b. This example is extended in the next section in which the rotation and
translation functions are described.
</p>
<p>Intermolecular Atomic Vectors and Translational Non-crystallographic Symmetry
If two non-crystallographically related molecules differ only by a non-lattice translation, with no
</p>
<p>rotational component, the Patterson function will contain an outstandingly high peak at a position
</p>
<p>corresponding to the non-crystallographic intermolecular vector, because all vectors between the two
</p>
<p>molecules will be lined up. Again, it is important to determine these situations prior to embarking on
</p>
<p>the full MR analysis, by calculating the Patterson function and inspecting the regions of high density.
</p>
<p>Fig. 10.13 Variables C and f are spherical polar coordinates that specify a direction about which the axes may be
rotated through an angle w
</p>
<p>520 10 Proteins and Macromolecular X-Ray Analysis</p>
<p/>
</div>
<div class="page"><p/>
<p>Remember that the origin peak (0, 0, 0) always has the highest density in the Patterson. Most
</p>
<p>Patterson density will be a very small fraction of P(000). We are therefore considering here Patterson
</p>
<p>density that is, a numerically significant fraction of P(000), possibly as high as 20&ndash;50%.
</p>
<p>10.6.6 Molecular Replacement in Practice
</p>
<p>The Search Model
One of the following situations may provide a suitable opportunity for using MR:
</p>
<p>1. Determination of a protein derivative structure containing a small ligand or other modification,
</p>
<p>crystallizing in a space group different from that of the known native protein structure, for
</p>
<p>example, enzyme&ndash;inhibitor or lectin&ndash;carbohydrate complexes.
</p>
<p>2. Determination of macromolecular complex combinations when one or both components have
</p>
<p>known structures. It may be possible to carry out successive MR searches in order to provide a
</p>
<p>starting model for the complete structural combination. If only one macromolecular component
</p>
<p>can be located by MR, lengthy Fourier development and refinement may follow. Examples
</p>
<p>include antibody&ndash; protein and protein&ndash;nucleic acid complexes.
</p>
<p>3. Functionally similar proteins with high sequence homology, one member of the series having a
</p>
<p>known structure. Sequence homology as low as 20% may be sufficient especially if many of the
</p>
<p>sequence changes are conservative, that is, chemically similar. Sequence homology may also be
</p>
<p>used in less obvious cases of structural similarity, such as proteins not belonging to the same or
</p>
<p>related families, but this carries a greater risk of failure.
</p>
<p>Fig. 10.14 Non-crystallographic symmetry: the section w &frac14; 180� of the self-rotation function for the protein RCA,
which crystallizes in space group P32, calculated with the program POLARRFN. The peak related by non-
</p>
<p>crystallographic twofold symmetry approximately perpendicular to the crystallographic 32 axis, the only true symmetry
</p>
<p>axis of this space group, is arrowed
</p>
<p>10.6 Determination of the Phases for Protein Crystals 521</p>
<p/>
</div>
<div class="page"><p/>
<p>4. Proteins grown by a recombinant technique from a DNA template, or site-directed mutants (proteins
</p>
<p>grown from modified DNA) having known native structures. The modified proteins may sometimes
</p>
<p>crystallize isomorphously or otherwise in different space groups.
</p>
<p>Data Base Searches
Several protein sequence data bases are readily available to accommodate sequence homology
</p>
<p>searches for suitable MR test structures; a list is given in Appendix D. Depending on the type of
</p>
<p>protein involved it is often known in advance what likely target structures are available. For example,
</p>
<p>Ricin is the parent structure for the ribosome inactivating protein (RIP type II) family and would be the
</p>
<p>first choice to use in MR search for determination of another member of the family such as RCA.
</p>
<p>Sequence matching protocols and algorithms are under active development. Many of these routines are
</p>
<p>freely available on the internet and can be accessed in combination with known protein sequences that
</p>
<p>are accessible through data bases such as BLAST.
</p>
<p>Where possible, well-defined structures should be chosen as search models. In simple terms, this
</p>
<p>means that the resolution should be 2 Å or better, and the R-factor should be 20% or less. It is more
</p>
<p>risky to use low resolution X-ray structures or nuclear magnetic resonance derived models, both of
</p>
<p>which will suffer from a lack of refinement, but these may have to be employed if no other models are
</p>
<p>available.
</p>
<p>The Target Patterson Function
Calculation of the target Patterson function for the unknown structure is straightforward in practice as the
</p>
<p>Fo(hkl) data are already available from diffraction measurements. It has been suggested that different
</p>
<p>resolution cut-off values should be tried, say, between 3 and 5 Å; in other words, it is best to employ
</p>
<p>medium to low resolution data in order to blur the differences between the search and target models. An
</p>
<p>inner low-resolution data cut-off should also be used, within 10&ndash;20 Å, as such data are influencedmainly
</p>
<p>by the solvent structure and therefore do not directly contribute to the protein structure at all. Facilities for
</p>
<p>easily setting up these cut-off values are available in most software. Again, different radii of integration
</p>
<p>should also be tried in order to pinpoint persistent solutions.
</p>
<p>In the case of the target structure, it is more difficult to restrict the rotation vectors to the
</p>
<p>intramolecular type alone because the observed Patterson distribution around the origin contains
</p>
<p>both intramolecular and interatomic vectors. This problem can be overcome to some extent, by
</p>
<p>restricting the volume of the target Patterson used in the search. The radius of integration used for this
</p>
<p>purpose can be selected by the user in an attempt to optimize the signal to noise ratio. Several trials
</p>
<p>may be required in practice in order to achieve the desired result. It has been recommended [49] to
</p>
<p>start with a value for this radius of integration approximately equal to 0.5dm, where dm is the
</p>
<p>approximate molecular diameter, increasing to about 0.75&ndash;0.8dm for further trials as necessary.
</p>
<p>The Search Patterson Function
Several approaches are available for generating the Patterson function for the known search model in
</p>
<p>macromolecular studies:
</p>
<p>1. Construct the search Patterson as a vector array from the structure coordinates, as in procedures
</p>
<p>used for small molecules, Sect. 8.3ff. This type of approach generates an extremely large number
</p>
<p>of vectors and is computationally very costly and slow to apply but nevertheless is used effectively
</p>
<p>in the X-PLOR [40] suite of programs.
</p>
<p>2. In the program AmoRe, the search model coordinates are transformed with respect to an artificial
</p>
<p>triclinic unit cell with angles of 90� and cell edges of pdm, where dm is again the molecular
diameter and the factor p is selected by the user, frequently with a value of 2.0 or less. Values of
</p>
<p>jFc(hkl)j calculated from the triclinic coordinates are then used to calculate the search Patterson.
</p>
<p>522 10 Proteins and Macromolecular X-Ray Analysis</p>
<p/>
</div>
<div class="page"><p/>
<p>This approach is computationally efficient and furthermore allows models that do not correspond
</p>
<p>to any crystal structure, as derived from NMR studies or from model building, to be used. One or
</p>
<p>both of these procedures may be used in practice. In either case, the need to consider symmetry is
</p>
<p>eliminated, thereby ensuring that only intramolecular vectors are generated for the search model.
</p>
<p>Recognition of the Correct Rotation Solution
Several powerful programs are currently available for MR analysis. Of these AmoRe and
</p>
<p>X-PLOR are commonly used. X-PLOR is a more general program suite, covering most aspects
</p>
<p>of macromolecular crystal structure analysis and refinement, whereas AmoRe is specific to MR.
</p>
<p>The use of these programs to locate the correct rotation function solution involves searching
</p>
<p>through a large number of trial rotation angle triplets (a, b, g). Incremental values of 5 or 10� for
each of these three angles are typical, resulting in a large number of trials. These programs
</p>
<p>optimize the rotation angles around local maximum regions of Patterson overlap. In AmoRe, each
</p>
<p>promising trial carries a correlation coefficient Cc indicating the level of agreement between the
</p>
<p>rotated Pattersons. The correct, or near correct, solution should have the highest Cc value.
</p>
<p>However, at the rotation stage, Cc values do not always differ significantly between potential
</p>
<p>solutions, and the values tend to be much smaller, typically around 5&ndash;15%, than at the later
</p>
<p>translation and optimization stages. The AmoRe program also calculates the rms value sp for the
</p>
<p>Patterson density and gauges the significance of local density regions with respect to this value.
</p>
<p>A peak greater than 3sp is considered significant.
</p>
<p>If different search models, resolutions or integration radii, of any of the other parameters available,
</p>
<p>have been varied over a series of searches, consistency in the occurrence of solutions for the angle
</p>
<p>triplets a, b, g, with high Cc values, is a good indication that the given solution is worth carrying over
</p>
<p>to the translation stage. Because the Cc indicator is not absolutely reliable, all peaks greater than 50%
</p>
<p>of the maximum value, ranked in order, are normally retained for transfer to the translation stage. The
</p>
<p>density values in terms of sp are also printed. In this way the probability of overlooking the correct
</p>
<p>rotation solution should be greatly decreased. The AmoRe system can accommodate the retained
</p>
<p>rotation solutions simultaneously into the translation routine. It also calculates an R-factor, Rf (not to
</p>
<p>be confused with Rfree) based on observed and calculated jFj values for each possible solution.
Because the model is usually very incomplete at this stage, it is possible only to compare the relative
</p>
<p>values of Rf for the different solutions, but this practice has been found to be quite useful and
</p>
<p>effective.
</p>
<p>The Rotation Function and Non-crystallographic Symmetry
For structures that contain more than one molecule per asymmetric unit related by NCS, the rotation
</p>
<p>function would be expected to produce a corresponding number of solutions, equivalent, or nearly so,
</p>
<p>with respect to both the Cc index and reproducibility under different computational conditions such as
</p>
<p>resolution, radius of integration, search unit-cell size and initial search model orientation. Only
</p>
<p>genuine solutions will produce acceptable translated models at the next stage of MR. There is no
</p>
<p>guarantee that such translated models will occupy their true relative positions in the unit cell, and this
</p>
<p>problem requires further attention, as explained below.
</p>
<p>The Translation Function
It cannot be overemphasized that success with the MR technique depends heavily on the precision
</p>
<p>with which the correct rotation parameters have been established. Programs such as AmoRe and
</p>
<p>X-PLOR include optimization procedures at the rotation stage to improve the chances of success later
</p>
<p>on. For each of the potential rotation solutions carried over into this stage of MR a set of atomic
</p>
<p>coordinates is generated. It is necessary to place this oriented model into the correct location in the
</p>
<p>10.6 Determination of the Phases for Protein Crystals 523</p>
<p/>
</div>
<div class="page"><p/>
<p>unit cell. Full attention to the symmetry elements and their relationship with one another and the
</p>
<p>standard origin must be maintained. In general, only one of the retained rotation solutions will
</p>
<p>provide a successfully translated model.
</p>
<p>When NCS is present the user looks for the corresponding number of successfully translated
</p>
<p>models. Each test model, carried over from the rotation stage, is moved by incremental translations
</p>
<p>called (Dt1, Dt2, Dt3) along three independent axial directions. The small steps used, typically 1 Å or
</p>
<p>more initially, are further optimized, as with the rotation solutions, so as to obtain the most promising
</p>
<p>models. The translation function routines attempt to correlate, at least conceptually, the observed and
</p>
<p>calculated Patterson functions. As we have seen, optimal correspondence is measured in AmoRe by
</p>
<p>the correlation coefficient Cc, which approaches 1.0 for the best solution, but is usually much less than
</p>
<p>unity. In addition, it is possible to compare Fo(hkl) with jFc(hkl)j for each translated model and to
calculate Rf, which is usually found to be lowest in value for the best solution. The AmoRe system
</p>
<p>employs both Cc and Rf in order to discriminate between potential solutions. As with the earlier
</p>
<p>rotation stage, consistency under different trial conditions is an important and extremely useful means
</p>
<p>of generating confidence in the validity of the solution of choice, prior to embarking on what can be
</p>
<p>very lengthy model building and refinement.
</p>
<p>Rigid-Body Refinement
Recent innovations in software are aimed at improving the likelihood of choosing the correct model
</p>
<p>resulting from an MR search. At the same time the accuracy of the three rotational and three
</p>
<p>translational parameters of the best solution are optimized, and will therefore correspond to a model
</p>
<p>which will be easier to refine. To this end, both AmoRe andX-PLOR include refinement routines which
</p>
<p>fine tune the rotation and translation parameters by adjusting the position of the search model without
</p>
<p>making changes to the geometry of themodel. This technique is known as rigid-body refinement. In this
</p>
<p>approach, the validity of the rotated and translated model is given a final check in terms of Cc, which
</p>
<p>should approach 1.0, and of Rf, which should minimize.
</p>
<p>Subunits and Non-crystallographic Symmetry
Individual molecules related by NCS may produce independent acceptable solutions as a result of the
</p>
<p>rotation and translation stages of MR. Because there is no guarantee that the coordinate sets for the
</p>
<p>individual molecules produced at this stage are properly correlated in the unit cell it is not usually
</p>
<p>possible to simply combine them to form the final trial structure. The main reason for this is
</p>
<p>concerned with the existence of more than one equivalent origin in many space groups. For example
</p>
<p>in space group P21 we know that there are four distinct 21 axes per unit cell, at [0, y, 0], &frac12;12; 2; y; 0�,
&frac12;0; y; 1
</p>
<p>2
�, and &frac12;1
</p>
<p>2
; y; 1
</p>
<p>2
�. Both AmoRe and X-PLOR have the facility for allowing a rotated&ndash;translated model
</p>
<p>to be included as a rigid group, which is subtracted from the unknown structure in order to enhance the
</p>
<p>signal from the missing non-crystallographically related molecule or molecules. If successful, this
</p>
<p>process will result in the non-crystallographically related molecules being properly placed relative to
</p>
<p>one another in the unit cell. This is an extremely powerful facility and will produce a full trial structure
</p>
<p>to be developed further by refinement procedures.
</p>
<p>Phases Derived from Molecular Replacement
The phase information derived from Patterson Search methods is initially derived through the
</p>
<p>calculation of structure factors, inspection of electron density maps, and least-squares refinement&mdash;
</p>
<p>the method of Fourier-least-squares. Similar methods apply to macromolecular analysis but with
</p>
<p>some important variations.
</p>
<p>524 10 Proteins and Macromolecular X-Ray Analysis</p>
<p/>
</div>
<div class="page"><p/>
<p>10.6.7 Application of the AmoRe Algorithms to Ricin Agglutinin
</p>
<p>We continue with the study of RCA, which we introduced in Sect. 10.4.7, as an example of the use of
</p>
<p>the program AmoRe to solve a protein structure [27]. The amino acid sequences of the individual A-
</p>
<p>and B-chains in both Ricin itself and in RCA are very similar, the overall homology being about 85%.
</p>
<p>This suggests that, although Ricin is only half the size of RCA, its structure [28, 50, 51] would be
</p>
<p>expected to make an ideal search molecule for RCA, with two final solutions corresponding to the two
</p>
<p>halves of the RCA molecule.
</p>
<p>A number of trials was used to establish the rotation search for RCA:
</p>
<p>1. The search-model Patterson maps were calculated using intensity data restricted to various
</p>
<p>resolution ranges, the most useful being 4.0&ndash;10 Å.
</p>
<p>2. The Ricin search model was placed in an orthogonal cell of P1 symmetry, with unit-cell
</p>
<p>parameters 100, 80, 60 Å, a &ldquo;special&rdquo; triclinic unit cell. Other search model unit cells were used
</p>
<p>with edges from 60 to 100 Å.
</p>
<p>3. The radius of integration was set at values between 25 and 35 Å, and the rotation function was
</p>
<p>calculated at angular steps of 2.5�. An overall temperature factor applied to F2o was set to �20 Å
2,
</p>
<p>which has the effect of sharpening the Patterson peaks, Sect. 7.4. Carrying over the most promising
</p>
<p>peaks from the rotation stage and using space group P32, two different outstanding translation
</p>
<p>function solutions were produced, as expected for the two halves of the RCA molecule. They had
</p>
<p>peak heights of 4.8sp and 4.5sp above the highest noise peak (a value greater than 3sp is normally
</p>
<p>considered to be outstanding) with Rf values of 49.2% and 54.0% and Cc 44.5% and 37.1%,
</p>
<p>respectively, corresponding to solutions 1 and 2, after application of the rigid body refinement
</p>
<p>protocol of AmoRe, Tables 10.2 and 10.3 [52].
</p>
<p>Using both solutions simultaneously, all symmetry-related molecules in the P32 unit cell for
</p>
<p>this solution were generated and examined graphically using the program MOLPAK. This
</p>
<p>program detected no inadmissibly short intermolecular contacts in the packing of this model.
</p>
<p>As anticipated however the symmetry-related version of solution 2 (Ricin 2) was found to be
</p>
<p>situated in close proximity to solution 1 (Ricin 1). Closer examination strongly suggested that
</p>
<p>Table 10.2 AmoRe rotation function for RCA in
space group P32
</p>
<p>a (�) b (�) g (�)
</p>
<p>1 109.7 80.2 80.7
</p>
<p>2 41.3 83.1 169.9
</p>
<p>3 14.3 46.5 11.5
</p>
<p>4 70.0 141.7 103.0
</p>
<p>5 15.6 139.9 177.1
</p>
<p>Data range 10&ndash;4 Å; radius of integration 35 Å; search
</p>
<p>cell 110, 110, 90 Å. The top five R-function peaks are
</p>
<p>shown. The correct solutions (&ldquo;peaks&rdquo;) 1 and 2 are in
</p>
<p>bold type. Solutions 1 and 2 alone persist under other
</p>
<p>conditions, such as: data range 10&ndash;4 Å; radius of inte-
</p>
<p>gration 32 Å; search cell 100, 90, 80 Å; data range
</p>
<p>10&ndash;4 Å; radius of integration 25 Å; search cell 100, 80,
</p>
<p>60 Å; Data range 10&ndash;4 Å; radius of integration 37 Å;
</p>
<p>search Cell 100, 100, 80 Å [52].
</p>
<p>10.6 Determination of the Phases for Protein Crystals 525</p>
<p/>
</div>
<div class="page"><p/>
<p>Ricin 1 and Ricin 2 were covalently linked, and after much further refinement the linkage was
</p>
<p>shown to be a disulfide S&ndash;S bond. The same rotation function results when used with the
</p>
<p>enantiomorphic space group P31 produced no translation function peaks comparable with those
</p>
<p>in space group P32. It was concluded, therefore, that MR had produced a potentially refineable
</p>
<p>model for RCA, and subsequent development and refinement of the P32 structure proved this to be
</p>
<p>the case. Figure 10.15 shows a view of the RCA double heterodimer molecule drawn with the
</p>
<p>program RASMOL.
</p>
<p>The Initial MR Model
Unlike phases from MIR, those produced with MR are calculated from the structure factors of the re-
</p>
<p>positioned search model, and as such are subject to model bias. As a consequence, the calculated
</p>
<p>electron density will tend to reproduce the model even if it is incorrectly placed. However, there will
</p>
<p>usually be significant differences between the search model and the target model, particularly in terms
</p>
<p>of the amino acid sequence; such residues may have been omitted from the search model. The
</p>
<p>electron density calculated with model-based MR phases and the Fo(hkl) data of the target protein
</p>
<p>should show electron density in these regions of the amino acid sequence, thus allowing the new side
</p>
<p>groups to be modeled and added to the atom list. Failure of the electron density to reveal these
</p>
<p>modifications is an indication of further problems.
</p>
<p>Another indication is a lack of correspondence between Fo(hkl) and jFc(hkl)j, resulting in a very
high R-factor. However, as the model will be very incomplete at this stage, the R-factor alone should
</p>
<p>not be used to make decisions on the quality of the model. As in the case of MIR phasing, MR phases
</p>
<p>may first be enhanced by modification techniques, Sect. 10.8.2. Interpretation of the electron density
</p>
<p>is again achieved with molecular graphics, after which further refinement of the model is then carried
</p>
<p>out, Sects. 10.8 and 10.9.
</p>
<p>10.7 SIRAS and MAD Phasing
</p>
<p>In Sects. 7.6.5 and 7.6.6, we discussed two methods that are applicable to solving the phase problem
</p>
<p>for proteins, both of which require the use of tunable SR for making anomalous scattering measure-
</p>
<p>ments. In the SIRAS method, two sets of intensity data are required: one from the native crystal, and a
</p>
<p>Table 10.3 Extension of the rotations solutions 1 and 2 in Table 10.2 to the translation stage using space groups P32
and P31
</p>
<p>a (�) b (�) g (�) tx ty tz Cc (%) Rf (%) Ds
</p>
<p>P32 109.7 80.2 80.7 0.6334 0.1427 0.5674 44.5 49.2 4.8
1
</p>
<p>P32 41.3 38.1 169.9 0.9679 0.0321 0.00 37.1 54.0 4.5
2
</p>
<p>P31 109.7 80.2 80.7 0.5128 0.3910 0.00 9.5 77.9 1.5
1
</p>
<p>P31 41.3 83.1 169.9 0.7372 0.3397 0.00 10.9 77.8 1.4
2
</p>
<p>Compared to the other solutions in Table 10.2, the solutions in Table 10.3 for space group P32 are outstanding in terms
</p>
<p>of Cc (highest) and Rf (lowest), indicating the correctness of the solution. Rotation solutions 1 and 2 fail to provide any
</p>
<p>reasonable translation solutions in space group P31, which is therefore eliminated. As a further test of the method, the
</p>
<p>starting Ricin model was rotated randomly to a new starting position, and the rotation and translation functions
</p>
<p>recalculated. Two outstanding peaks were again found that proved to be consistent with the above solutions 1 and 2.
</p>
<p>The packing of the molecules from the solution in P32 shows no disallowed contacts, and reveals also how the two
</p>
<p>halves of the Ricin molecule associate.
</p>
<p>526 10 Proteins and Macromolecular X-Ray Analysis</p>
<p/>
</div>
<div class="page"><p/>
<p>second from the heavy-atom derivative for which Friedel or Bijvoet pairs are measured, thus enabling
</p>
<p>the phase ambiguity of SIR to be resolved. In the Multiple-Wavelength Anomalous Dispersion
</p>
<p>technique (MAD), the anomalous dispersion measurements are made at two different wavelengths,
</p>
<p>so providing an extra data set. In both methods the wavelengths used are selected in order to optimize
</p>
<p>the anomalous dispersion effects from heavy atoms present in the crystals.
</p>
<p>The MAD technique is often used to solve the phase problem for proteins containing selenium-
</p>
<p>mutated methionine residues, produced by recombinant DNA technology, in which sulfur has been
</p>
<p>replaced with the heavier selenium atom (ZS &frac14; 16, ZSe &frac14; 34) in order to enhance the strength of
the anomalous scattering signals. In such cases [53] the structure determined is often that of the
</p>
<p>modified protein.
</p>
<p>Fig. 10.15 (a) A view of the RCA double heterodimer molecule, drawn by the program RASMOL. The position of the
pseudo-twofold axis relating the two halves of the molecule that are joined by a disulfide bridge is shown by the arrow. This
</p>
<p>figure is a ribbon diagram showing the course of themain chain, and it is possible to pick out regions of helix structure, sheet
</p>
<p>structure, and random coils. (b) Stick-bond diagram of RCA; the central S&ndash;S bond is indicated by the arrow
</p>
<p>10.7 SIRAS and MAD Phasing 527</p>
<p/>
</div>
<div class="page"><p/>
<p>10.8 Use of Phase Information and Density Modification
</p>
<p>Successful application of one of the four methods MIR, MR, SIRAS or MAD, will provide at least
</p>
<p>approximate values of the phases required for calculation of a preliminary electron density map of the
</p>
<p>native protein structure. Any errors associated with these phases will affect the quality of the
</p>
<p>calculated electron density and its interpretation in terms of the protein structure. To enable improve-
</p>
<p>ment in both the phases and electron density to be made at an early stage, density modification
</p>
<p>techniques have been developed to optimize the electron density map, particularly in cases where
</p>
<p>there is reason to believe the initial phasing is of poor quality, and also to enable a rational
</p>
<p>interpretation to be carried out. Some applications in which this technique might be necessary include
</p>
<p>SIR or MIR where the mean figure of merit is too low, say, less than 0.5, and in MR where Cc is too
</p>
<p>low, or Rf too high, or where no outstanding or persistent solution is observed.
</p>
<p>10.8.1 Properties of r(xyz) for Proteins
</p>
<p>The inherently poor resolution of protein X-ray data means that, generally speaking, only heavy
</p>
<p>atoms, those with atomic number greater than 20, are expected to produce distinct peaks in the
</p>
<p>electron density map. The general features of protein electron density maps may be summarized as
</p>
<p>follows:
</p>
<p>1. Main-chain density tends to be continuous and relatively strong. This is because atoms in the
</p>
<p>polypeptide chain are generally held firmly by interactions between neighboring atoms and are
</p>
<p>therefore less susceptible to thermal and statistical disorder, Sect. 8.9. Recognition of strong,
</p>
<p>continuous density in the map can therefore enable the main polypeptide chain to be traced and the
</p>
<p>molecular envelope to be outlined.
</p>
<p>2. Side-chain atoms tend to be less rigidly held in the structure than main-chain atoms, with
</p>
<p>correspondingly weaker electron density.
</p>
<p>3. Protein crystals can contain anything between 35 and 70% in solvent molecules. This part is the
</p>
<p>most disordered of the structure and will consequently be associated with very weak electron
</p>
<p>density, situated in the intermolecular interstices and solvent channels. Consideration of the
</p>
<p>regions of weak density should therefore enhance the recognition of the protein molecular
</p>
<p>boundary. Solvent molecules, such as water, closest to the protein are generally better ordered
</p>
<p>than bulk solvent atoms, and may be identified once some structure refinement has been effected.
</p>
<p>10.8.2 Programs for Density Modification
</p>
<p>We provide here a brief list of programs available for density modification (DM) in protein structure
</p>
<p>analysis.
</p>
<p>DM [54] is a CCP4-supported program. It applies constraints to the observed electron density and
</p>
<p>derives new phases via Fourier transform techniques. Phases can be calculated for reflections not
</p>
<p>involved in the original phasing. However this procedure, known as phase extension, should be
</p>
<p>applied with caution: for example, it is unlikely that a 6 Å MIR map will rapidly expand to 2 Å, but
</p>
<p>the method might more realistically expand the phases from 2.5 to 2 Å.
</p>
<p>SOLV [55] uses a method known as solvent flattening, which establishes uniformity of density in
</p>
<p>solvent regions, on the assumption that regions of disordered solvent are essentially without
</p>
<p>structure.
</p>
<p>528 10 Proteins and Macromolecular X-Ray Analysis</p>
<p/>
</div>
<div class="page"><p/>
<p>HIST [56] applies histogram matching based on the known characteristics of biological structures so
</p>
<p>as to predict the histogram of density values in the protein region. The current density map is then
</p>
<p>systematically modified according to the predicted histogram. This technique is complementary to
</p>
<p>solvent flattening. Between the two, the image of the protein structure should become much clearer
</p>
<p>and facilitate model building prior to further refinement (see below). The combined process of
</p>
<p>SOLV/HIST may require 10&ndash;20 iterations to converge.
</p>
<p>SKEL uses skeletonization [57, 58] to provide a sound basis for molecular graphics model building of
</p>
<p>the structure, by enhancing the connectivity of the electron density in main-chain regions.
</p>
<p>SAYR [59] attempts to achieve phase improvement for data at 2 Å resolution or better by applying
</p>
<p>phase relationships adapted from a classical small-molecule phasing method.
</p>
<p>AVER [60, 61] uses molecular averaging for structures where either NCS or exact crystallographic
</p>
<p>symmetry is present; it provides better phasing for initial model building. This constraint should, if
</p>
<p>possible, be released for the final structure.
</p>
<p>Figure 10.16 is a suggested flow diagram for use with these programs.
</p>
<p>10.8.3 Preparing to Refine the Structure
</p>
<p>At some stage in the determination of any structure using X-ray analysis, it has to be decided whether
</p>
<p>the current model can be refined further. For small molecules, during the initial stages of Fourier
</p>
<p>refinement we are guided by a combination of good molecular geometry and the behavior of the R-
</p>
<p>factor, which should fall to 16&ndash;18% prior to least-squares refinement. Even a small protein of Mr
approximately 15,000 contains about 1,000 non-hydrogen atoms, so that examination of the electron
</p>
<p>density is much more arduous and time consuming.
</p>
<p>This work is carried out at a computer workstation, employing programs such as FRODO or its
</p>
<p>updated version O, TURBO-FRODO, Coot (see below) and SKEL [57, 58], which allow a geometri-
</p>
<p>cally accurate protein model to be built into the electron density. Figure 10.17a, b show examples of
</p>
<p>fitting benzene ring atoms into electron density with Coot at two different resolutions. It should be
</p>
<p>Fig. 10.16 Flow diagram for electron density modification (DM). The final electron density is used for model building,
and for further refinement if of sufficient quality
</p>
<p>10.8 Use of Phase Information and Density Modification 529</p>
<p/>
</div>
<div class="page"><p/>
<p>Fig. 10.17 Coot (2Fo � jFjc) electron density maps. (a) Electron density for mistletoe lectin I (MLI) lactose complex
at 2.5 Å resolution in the vicinity of phenylalanine 88 (PHE 88), contoured at the 1.59s level. (b) Bovine ribonuclease A
at 0.95 Å in the vicinity of phenylalanine 20 (PHE 20), contoured at the 1.6s level: s is the rms deviation of the average
</p>
<p>electron density, a quantitative way of assessing the noise level of the map. The choice of contour levels determines
</p>
<p>both the appearance of the output map and the ease of interpretation. The effect of resolution is clearly apparent
</p>
<p>emphasizing the need for a program like Coot, which assigns coordinates to the model atoms in such a way as to
</p>
<p>preserve the molecular geometry. Assignment of atomic coordinates is obviously much easier for the higher resolution
</p>
<p>map (b) (Coot maps provided by courtesy of Dr. David Lisgarten)
</p>
<p>530 10 Proteins and Macromolecular X-Ray Analysis</p>
<p/>
</div>
<div class="page"><p/>
<p>emphasized that knowledge of the amino acid sequence of the protein is a desirable prerequisite
</p>
<p>which will help residues to be identified in the complex electron density map. After each round of
</p>
<p>model building, structure factors should be calculated and the R-factor inspected. A very satisfactory
</p>
<p>R-factor for a protein structure at this stage would be around 25%; a value up to 30% could be
</p>
<p>tolerated but should be treated with caution. Further computational refinement may then be carried
</p>
<p>out, as described in the following discussion.
</p>
<p>10.9 Macromolecular Structure Refinement and Solvent and Ligand Fitting
</p>
<p>10.9.1 Refinement Techniques
</p>
<p>Many programs are currently available for the refinement of macromolecular crystal structures. As
</p>
<p>we shall see some are mainly concerned with refinement, while others will already have been
</p>
<p>discussed for their particular use, for example in MR. It is a matter of personal choice and experience
</p>
<p>whether to use a specialized refinement program or to continue a particular study using the same
</p>
<p>program from which the structure was initially derived.
</p>
<p>The process of structure refinement involves optimization of the agreement between the observed
</p>
<p>and calculated diffraction patterns, represented by Fo(hkl) and jFc(hkl)j, and validation of the resulting
molecular structure. If isomorphous replacement has been used to derive phase information, the initial
</p>
<p>structure model will be derived from an electron density map using molecular graphics (see below).
</p>
<p>For anMR analysis, it is likely that rigid-body refinement of the model will have been carried out at an
</p>
<p>earlier stage. Whichever method has been used for the initial structure analysis, refinement of the
</p>
<p>model should be undertaken by a combination of the following techniques:
</p>
<p>1. Fourier refinement using successive Fourier synthesis
</p>
<p>2. Simulated annealing
</p>
<p>3. Least-squares analysis
</p>
<p>Further Details of Fourier Refinement and Model Building
Lack of resolution in protein X-ray data results in poor definition in the electron density map whether
</p>
<p>from MIR, MR, or any other phase determining method. It will be necessary to interpret this map by
</p>
<p>fitting the protein structure to the density. If MR has been used for the preliminary structure
</p>
<p>determination, electron density calculations will again play an important role initially, but the
</p>
<p>crystallographer will have prior knowledge of structural features of the search model that will help
</p>
<p>to establish the new structure. There are three aids to assist the initial process of interpreting the
</p>
<p>electron density:
</p>
<p>1. The amino acid sequence of the protein. If this is not known, interpretation of the crystal structure
</p>
<p>will necessarily be difficult, and probably not possible unless very good phasing at a high resolution
</p>
<p>has been achieved. If MR has been used, the target molecule will be subject to amino acid sequence
</p>
<p>changes; in addition, insertions and/or deletions may be required. These features have to be built into
</p>
<p>the density.
</p>
<p>2. Knowledge of the standard geometry of proteins [62, 63] in terms of main-chain and side-chain
</p>
<p>bond lengths and bond angles, and of secondary structural features, particularly the a-helix and b-
</p>
<p>sheet, Sect. 10.11. Software such as PROCHECK [64] and MolProbity [65] have been developed
</p>
<p>as extremely useful aids to protein structure verification, including bond length, bond angle and
</p>
<p>conformational checks, using the Ramachandran plot [66].
</p>
<p>10.9 Macromolecular Structure Refinement and Solvent and Ligand Fitting 531</p>
<p/>
</div>
<div class="page"><p/>
<p>3. Molecular graphics. The effective interpretation of electron density maps has been completely
</p>
<p>revolutionized over the past 25&ndash;30 years because of the development and availability of graphics
</p>
<p>software. These facilities enable structural features to be built into density maps, following
</p>
<p>standard geometry protocols, and provide the user with structure files for use in further analysis.
</p>
<p>Software available for carrying out these procedures include FRODO [67] and O [68] (O is an
</p>
<p>enhanced version of FRODO), XtalView [69], Coot [70] and MolProbity [65] enable the molecu-
</p>
<p>lar model to be fitted optimally into the electron density map. Output from these routines includes
</p>
<p>a pdb file of atomic coordinates for each amino acid and ligand recognized and fitted graphically.
</p>
<p>The pdb file will be used initially in the Fourier refinement stage and eventually as input to a least
</p>
<p>squares refinement program.
</p>
<p>The Fourier refinement process is represented in the flow diagram in Fig. 10.18. It consists of
</p>
<p>successive cycles of electron density calculation, graphical interpretation, updating structure
</p>
<p>files, and calculation of structure factors, and R-factors. Improvement in the model and its fit to the
</p>
<p>electron density should cause significant decrease in the R-factors. This process terminates when no
</p>
<p>further enhancement is evident. The structure file at this stage consists of atom names, individual
</p>
<p>atom fractional coordinates (x, y, z), overall or average temperature factor B, for the whole crystal and
</p>
<p>a scale factor that converts Fo values to an absolute scale. Individual Bj values are not refined at this
</p>
<p>stage of the analysis.
</p>
<p>Fig. 10.18 Basic flow diagram for protein structure refinement. Initial refinement is carried out by repeated cycles of
Fourier synthesis/molecular graphics until no further atoms can be added to the model. Once this process has converged,
</p>
<p>least squares (REFMAC, PHENIX or SHELX-97) or simulated annealing (X-PLOR) can be used in step IV. The cycle
</p>
<p>IV-II-III-IV is repeated until no further changes occur in the parameters of the structure; final convergence is then
</p>
<p>considered to have been reached. Coot is now a popular choice to employ throughout for model building with the
</p>
<p>electron density and molecular graphics analysis
</p>
<p>532 10 Proteins and Macromolecular X-Ray Analysis</p>
<p/>
</div>
<div class="page"><p/>
<p>10.9.2 Simulated Annealing
</p>
<p>After the initial model building of a protein structure, it is to be expected that there will be some
</p>
<p>regions of the model that are some distance from their true positions. Such errors are difficult to
</p>
<p>correct with Fourier methods alone, as the detailed inspection of each map is a lengthy and difficult
</p>
<p>process. We have described the routine refinement of small-molecule structures through the calcula-
</p>
<p>tion of successive Fourier syntheses followed by least-squares analysis. The Fourier method has also
</p>
<p>been shown to be an essential tool for macromolecular structure analysis, and in the following
</p>
<p>sections of this chapter, other techniques for refining large molecule structures will also be described.
</p>
<p>In order to apply least squares satisfactorily, the structure should not contain any gross errors; if it does
</p>
<p>the method will be unable to cope, as it assumes that the structural parameters are close to their
</p>
<p>true values.
</p>
<p>Another difficulty is that the ratio of reflections in the data set to the number of parameters in the
</p>
<p>model (data/parameter ratio) is much smaller for a large molecule (see Problem 10.7). The least
</p>
<p>squares method necessarily requires the problem to be over-determined to the extent of 3&ndash;4
</p>
<p>reflections per parameter or better, which is extremely difficult to achieve for proteins. In order
</p>
<p>to overcome these difficulties with the initial model, the program X-PLOR [40] or its updated
</p>
<p>version CNS [43] can be used to carry out the process of simulated annealing, which employs
</p>
<p>molecular dynamics.
</p>
<p>The main objective of this method is the elimination from the model those regions of structure that
</p>
<p>are far from their true positions. Such regions may have been introduced through the initial search
</p>
<p>model in MR, or by faulty model building at a later stage. It is likely that such regions will be
</p>
<p>associated with bad interatomic contacts and inadmissibly high potential energy. In simulated
</p>
<p>annealing the structure is given a large perturbation (&ldquo;heated to a high temperature&rdquo;), yielding an
</p>
<p>ensemble of energetically allowed structures that is then allowed to recover (&ldquo;cool&rdquo;) while preserving
</p>
<p>or re-establishing the correct minimum-energy molecular geometry. This process allows energy
</p>
<p>barriers that are associated with faults in the X-ray model to be overcome and corrected. The X-
</p>
<p>PLOR protocol involves minimization of an overall total potential energy term Etot, which is made up
</p>
<p>the empirical energy Eemp and the effective energy Eeff, where Eeff &frac14; EX-ray + Ep + Enb. The terms
Ep + Enb are included to take into account experimental information about phases and crystal
</p>
<p>packing, while EX-ray is a &ldquo;pseudo energy&rdquo; that involves differences in the observed and calculated
</p>
<p>structure factors:
</p>
<p>EX�ray &frac14; &eth;Wa=Na&THORN;
X
</p>
<p>hklWhkl&eth;Fo � KjFcj&THORN; (10.12)
</p>
<p>whereWa puts the term on to the same basis as &ldquo;energy&rdquo; (established through a dummy dynamics run
</p>
<p>without EX-ray), Na is a normalization factor which renders Wa independent of resolution, Whkl
provides a weighting scheme and K is a scale factor. X-PLOR is sometimes used without further
</p>
<p>refinement, and this can lead to a satisfactory structure determination. Individual isotropic thermal
</p>
<p>parameters Bj can be included in the refinement if the data collection resolution is sufficiently high, at
</p>
<p>least 2.5 Å. The conventional R-factor should drop to ca. 20% or lower unless there are big problems
</p>
<p>with the data, such as poor crystal quality or very weak diffraction, in which case it will not be
</p>
<p>possible to derive a highly significant structure with any refinement protocol. It may be desirable to
</p>
<p>undertake least-squares refinement of the structural parameters even after simulated annealing has
</p>
<p>been carried out. In any case, routines such as PROCHECK [64] or WHAT_CHECK [71],
</p>
<p>Sect. 10.11, should be used to provide rigorous validation of the structure model.
</p>
<p>10.9 Macromolecular Structure Refinement and Solvent and Ligand Fitting 533</p>
<p/>
</div>
<div class="page"><p/>
<p>10.9.3 Least-Squares Refinement: Constrained, Restrained and Other Protocols
</p>
<p>The method of least squares is used routinely in small-molecule structure analysis, optimizing the fit
</p>
<p>between jFcj andFo, and providing refined parameters and their standard deviations. Becausemost small-
molecule crystals diffract to around 0.8 Å, the data set provides about ten Fo data per parameter even
</p>
<p>when an anisotropic thermal displacement model is used. This degree of over-determination is found
</p>
<p>rarely in protein crystals. It is more usual with poorly diffracting protein crystals for there to be only one
</p>
<p>Fo or less per structural parameter. Methods have therefore been developed, which attempt to economize
</p>
<p>on the number of parameters defining the structure model, and provide data in addition to the X-ray data
</p>
<p>that can contribute toward the refinement of the model. Software for carrying out such refinements is
</p>
<p>usually based on constrained or restrained protocols.
</p>
<p>Constrained or Rigid-Body Refinement
Rigid-body refinement was mentioned in Sect. 10.6.3 as an option in someMR protocols. The method
</p>
<p>involves repositioning of the whole model in the unit cell without any further adjustment of individual
</p>
<p>atom coordinates. The method can be useful for refinement of molecules related by NCS by initially
</p>
<p>averaging the two or more molecules so as to improve the data/parameter ratio; this method is
</p>
<p>available, for example, in the program X-PLOR. Constrained refinement can also be used for fixing
</p>
<p>rigidly predefined bond length and angle types and planarity of groups such as rings or peptide groups.
</p>
<p>While possibly effecting some degree of refinement in the structure model, constrained refine-
</p>
<p>ment will usually require further refinement by restrained least squares, where a degree of tolerance
</p>
<p>is built into the bond lengths and angles, if the data/parameter ratio allows. As an example of the
</p>
<p>savings in parameters that can be achieved by applying constraints, consider a flat six-membered
</p>
<p>ring. In terms of free atom parameters there are 18 positional parameters. By constraining the 6
</p>
<p>bond lengths, 6 bond angles, and 3 cross-ring distances, only 15 parameters are required, a saving of
</p>
<p>3 parameters. When applied repeatedly within a macromolecular structure, small savings like this
</p>
<p>can accumulate to a very worthwhile total, and at the same time ensure that the geometry of the
</p>
<p>refined structure will be acceptable. The input data set does, however, require a great deal of
</p>
<p>preparation in order to set up the required constraints.
</p>
<p>Restrained Refinement
This and other methods of refinement should not be used unless considerable effort has been applied
</p>
<p>to the Fourier refinement and possibly supplemented by simulated annealing as well, so as to reduce
</p>
<p>the errors present in the structure and to include as many atoms as possible in the coordinate set.
</p>
<p>Restrained geometry refinement is more flexible than constrained refinement: each of the standard
</p>
<p>values of bond lengths, bond angles and other distances is tagged with a tolerance which specifies an
</p>
<p>acceptable range for the refined value of the parameter, Sect. 10.11. The following programs are
</p>
<p>important in restrained refinement.
</p>
<p>RESTRAIN
This program [72], which has now been superseded by REFMAC and PHENIX, employed a least-
</p>
<p>squares algorithm, that used terms involving differences (DF)2, or (Fo � jFcj)
2, standard geometry
</p>
<p>values, Sect. 10.11, and the planarity of groups. Corrections to structure parameter values were
</p>
<p>derived by minimizing the function
</p>
<p>M &frac14;
X
</p>
<p>Wf&eth;DF&THORN;
2 &thorn;
</p>
<p>X
Wf&eth;fo � fc&THORN;
</p>
<p>2 &thorn;
X
</p>
<p>Wd&eth;do � dc&THORN;
2 &thorn;
</p>
<p>X
Wa&eth;ao � ac&THORN;
</p>
<p>2
</p>
<p>&thorn;
X
</p>
<p>Wb&eth;bo � bc&THORN;
2 &thorn;
</p>
<p>X
WvV (10.13)
</p>
<p>534 10 Proteins and Macromolecular X-Ray Analysis</p>
<p/>
</div>
<div class="page"><p/>
<p>where fo is the (usually) MIR phase, d is a bond length, a is a cross-bond distance, such as A&ndash;C in the
</p>
<p>sequence A&ndash;B&ndash;C and therefore a measure of bond angle, b is a term for non-bonded distances, V is a
</p>
<p>planarity restraint, and the weights W apply to various types of terms and require very careful
</p>
<p>adjustment during the course of the refinement to ensure proper calculation of standard deviations
</p>
<p>for the refined parameters. Chirality is preserved by applying restraints to the edges of all chiral
</p>
<p>tetrahedra. Anisotropic thermal displacement parameters, nine for each freely refined atom, can be
</p>
<p>included if the data/parameter ratio allows. RESTRAIN was a CCP4 program: it employed non-FFT
</p>
<p>calculations for structure factors and partial derivatives and was consequently slow to perform each
</p>
<p>cycle, but required fewer cycles to converge than FFT-based algorithms.
</p>
<p>REFMAC with Maximum Likelihood
This is a currently supported and widely used CCP4 program [73] with many facilities, including the
</p>
<p>fitting and refinement of ligands such as sugars. The refinement method used is based on the statistical
</p>
<p>method of maximum likelihood. Use of the method of least squares assumes inherently that errors in
</p>
<p>the data and themodel are random, uncorrelated andwith equal and finite variance. Some or all of these
</p>
<p>conditions may break down in practice, for example, if part of the molecular structure is missing
</p>
<p>initially from the model, even though the rest of the model is acceptable. Considerations such as these
</p>
<p>lead to the use of another method for function minimization based on the principle of maximum
</p>
<p>likelihood or probability. This requires the associated probabilities of the observed Fo(hkl) to be a
</p>
<p>known function of the parameters of the model.
</p>
<p>Maximum likelihood refinement adjusts the parameters of the model so as to maximize the
</p>
<p>probability that the data and model are compatible. The overall conditional probability is calculated
</p>
<p>as the product of the probabilities for each reflection, estimated from the individual Fo(hkl), jFc(hkl)j
and s(hkl) values [73]. In practice, use is also made, as we have emphasized in other respects, of the
</p>
<p>known stereochemistry of the macromolecular system. Maximum likelihood refinement can also be
</p>
<p>carried out with the programs CNS [43] and BUSTER-TNT [74].
</p>
<p>SHELXL-97
SHELX programs [75] are well known for their use in small-molecule crystal structure analysis and
</p>
<p>have been discussed in earlier chapters. The SHELXL-97 version has been adapted to accommo-
</p>
<p>date protein refinement by incorporating the SHELXPRO computing interface. The geometry
</p>
<p>refinement options and restraints are similar to those described previously and include additional
</p>
<p>facilities such as: anisotropic scaling, refinement progress display, and thermal displacement
</p>
<p>analysis. For conformational analysis, tabulation, display, and publication purposes the program
</p>
<p>carries out the following:
</p>
<p>1. Checks for validity of main-chain conformations using a Ramachandran plot [76] (see below)
</p>
<p>2. Produces an electron density map file for use in the graphics program O
</p>
<p>3. Generates a pdb file which can be used in most graphics display programs (see Appendix D). The
</p>
<p>pdb file is a standardized record of the structural parameters and is used as an input to programs for
</p>
<p>analyzing and displaying these results as well as for deposition at the Protein Data Bank; see also
</p>
<p>Problem 11.7
</p>
<p>Practical Details
Initially, small ligands and/or solvent can be excluded, but should be added through further use of
</p>
<p>electron density maps as the refinement progresses. Least-squares analysis predicts corrections for the
</p>
<p>structural parameters in the current model (coordinates, temperature factors and scale factor). It may
</p>
<p>be possible to refine individual Bj values if the data/parameter ratio is at least 2&ndash;3; anisotropic thermal
</p>
<p>displacement factors are unlikely to be refineable in all but very high-resolution protein structure
</p>
<p>10.9 Macromolecular Structure Refinement and Solvent and Ligand Fitting 535</p>
<p/>
</div>
<div class="page"><p/>
<p>determinations. Selected groups of atoms can be given a single temperature factor, which provides an
</p>
<p>easy means of saving parameters.
</p>
<p>PHENIX
This relatively new software suite [77, 78] for the automated determination of macromolecular
</p>
<p>structures using X-ray crystallography and other methods and described as a &ldquo;comprehensive
</p>
<p>Python-based system for macromolecular structure solution,&rdquo; is now widely used and highly recom-
</p>
<p>mended. Among its refinement features is a facility for torsion angle parameterization [79] which is
</p>
<p>also available in the program CNS [43]. This requires a much smaller number of parameters and is
</p>
<p>therefore an approach which is expected to improve the refinement process, for example by speeding
</p>
<p>up convergence and reducing the number of cycles required.
</p>
<p>Solvent Molecules and Small Ligands: Fitting to the Electron Density
The assignment of solvent atoms, usually water, in the crystal structure model is one of the last
</p>
<p>procedures to be carried out. These atoms make only small individual contributions to X-ray
</p>
<p>intensities, but several hundred water molecules can influence considerably the values of jFcj. The
total level of solvation in protein crystals is usually around 50%, and most of the solvent is
</p>
<p>unstructured, forming very disordered, fluid regions in the intermolecular channels. However within
</p>
<p>a layer closest to the protein molecule, it is to be expected that many solvent molecules will form
</p>
<p>strong hydrogen-bond interactions with atoms of the protein, and consequently will acquire an
</p>
<p>ordered state approaching that of the protein itself.
</p>
<p>A measure of the ordering may be derived from the refinement if individual atomic thermal
</p>
<p>displacement parameters Bj are used, but this is possible only for analyses employing a resolution
</p>
<p>better than about 1.8 Å. Generally, solvent atoms will have Bj values 10&ndash;20% larger than the protein
</p>
<p>atoms with which they are associated. Hydrogen-bonded interactions are responsible for ordering
</p>
<p>water molecules or other hydrophilic solvent molecules, such as ethanol. The interactions can involve
</p>
<p>any main-chain or side-chain atom which can act as a hydrogen-bond donor or acceptor. In order to
</p>
<p>locate possible water molecules it is necessary to inspect an (Fo � jFcj) or a (2Fo � jFcj) difference
</p>
<p>density map in detail. Possible water sites may be assigned to significant density regions, say, greater
</p>
<p>than 2s(r) where s(r) is the rms electron density, and located within 2.5&ndash;3.3 Å (hydrogen-bonding
</p>
<p>distances) from one or more possible hydrogen-bond donors, usually &ndash;OH or &gt;NH groups, or
</p>
<p>acceptor atoms, usually oxygen or nitrogen; there should be no other close contacts present. Suitable
</p>
<p>electron density regions should ideally be spherical and small in volume.
</p>
<p>Since this procedure can be very time consuming, programs have been developed to expedite the
</p>
<p>process, including routines in O [68] and SHELXPRO [75]. It should be emphasized that water sites
</p>
<p>assigned by such automatic procedures should be critically checked manually using molecular
</p>
<p>graphics such as Coot [70]. Graphical examples of water molecules located in this way are shown
</p>
<p>in Fig. 10.19. If the analysis is at sufficiently high resolution, the Bj values should be refined in the
</p>
<p>subsequent analysis. Solvent sites, as with other included atoms, for which Bj refines to a value greater
</p>
<p>than 80 Å2 should be discarded as being unrealistically weak. The inclusion of significant numbers of
</p>
<p>solvent atoms, which meet all these requirements should cause the R-factor to decrease by 1&ndash;2% or
</p>
<p>more. In many cases solvent atoms are functionally important and their recognition through the X-ray
</p>
<p>structure may be regarded as an added bonus.
</p>
<p>The assignment of small ligands in the structural model follows a route similar to the above
</p>
<p>procedures for water molecules. Atom coordinates should be assigned and added to the data file for
</p>
<p>as many atoms as possible, paying full attention to the validity of the molecular geometry involving the
</p>
<p>new atoms. As much information as possible should be incorporated into the procedure, including
</p>
<p>536 10 Proteins and Macromolecular X-Ray Analysis</p>
<p/>
</div>
<div class="page"><p/>
<p>known or preconceived stereochemistry of the group, known or expected binding region of the protein,
</p>
<p>proximity of protein atoms with ligand atoms, and validation of contact distances. Figure 10.18
</p>
<p>provides a possible flow diagram for the refinement process just described. An example of difference
</p>
<p>electron density showing a bound inhibitor molecule in the protein RNase T1 [80] is shown in
</p>
<p>Fig. 10.20.
</p>
<p>10.10 Structure Validation: Final Checks
</p>
<p>In this final section, we mention some of the checks that can and should be made with respect to the
</p>
<p>protein structure when all of the above techniques have been exhausted. These concern both the
</p>
<p>technicalities of the X-ray analysis itself and the finished product.
</p>
<p>10.10.1 R-Factors
</p>
<p>One of the principal checks during the course of the analysis involves R-factors. While emphasizing
</p>
<p>that this should not be the only means of assessing the final structure, it does nevertheless provide the
</p>
<p>first indication that the analysis has progressed well. From the outset of the analysis, the overall
</p>
<p>quality of the intensity data will have been assessed from the value (or values) of the merging R-
</p>
<p>factor, Sect. 10.4.6. This in turn provides an indication of how we can expect the R-factor of the
</p>
<p>structure to behave.
</p>
<p>Fig. 10.19 Location of water molecules (actually water O atoms) in bovine ribonuclease A at 0.95 Å resolution for
data collected at �50 �C. Two water oxygen atoms are clearly visible in this (2Fo � jFcj) map contoured and
interpreted by Coot. One water molecule forms hydrogen bonds to the carbonyl O atom of the residue Arg 39, and to
</p>
<p>the main chain N atoms of both Asp 38 and Arg 39, as indicated by the distances. This information provides sound
</p>
<p>confirmation that the new peak, which is both strong in density and well resolved, is a water oxygen. The other water
</p>
<p>does not form hydrogen-bonds directly to the protein and is therefore in an outer hydration shell of the structure, which
</p>
<p>has about 40% total solvent. Subsequent least squares refinement of the B values of the proposed water oxygens to
</p>
<p>values less than about 40 Å2 would be taken as final confirmation of the assignment. Several hundred water molecules
</p>
<p>have been located in this structure (Coot map supplied by courtesy of Dr David Lisgarten)
</p>
<p>10.10 Structure Validation: Final Checks 537</p>
<p/>
</div>
<div class="page"><p/>
<p>Over-refinement, that is, an unjustified fitting of too many parameters, can lead to unrealistically
</p>
<p>low R-factors, whereas they should conform to the quality of the X-ray data. Depending on the
</p>
<p>resolution attained, which is typically around 1.8&ndash;2.2 Å for a good protein structure, we can expect
</p>
<p>the R-factor to drop to 18&ndash;20% on completion of the refinement. On the one hand, at higher resolutions
</p>
<p>the R-factor should be less because more of the structure, particularly water molecules, will have been
</p>
<p>fitted, and the structure will be less subjected to regions of disorder that are difficult to model. On the
</p>
<p>other hand, for lower resolutions the R-factor will tend to lie between 20 and 30%. In contemporary
</p>
<p>analyses it is expected that the Rfree index will be determined as a check on model bias, particularly in
</p>
<p>structures based on MR. Because it is unbiased by the refinement process, as a general rule of thumb
</p>
<p>Rfree should decrease as the R-factor falls but exhibit values 2&ndash;3% higher.
</p>
<p>Fig. 10.20 An example of an (Fo � jFcj) electron density plot for a protein structure in the active site region of
RnaseT1 [80]. Two moieties are clearly visible: a 30-guanosine monophosphate (lower density region) and a guanosine
</p>
<p>subsite (upper density). The resolution is 1.7 Å and find R value 14.5%
</p>
<p>538 10 Proteins and Macromolecular X-Ray Analysis</p>
<p/>
</div>
<div class="page"><p/>
<p>10.10.2 Evaluation of Errors
</p>
<p>Programs such as REFMAC, SHELX-97 and PHENIX provide estimated standard deviations for the
</p>
<p>refined structural parameters and these should be carefully reviewed to enable poorly refined parts of
</p>
<p>the structure to be identified. For a typical small-molecule analysis a good estimate of the error in
</p>
<p>each of the refined parameters is calculated as a by-product of the least-squares procedure, the best
</p>
<p>estimates being from a full-matrix refinement. This is possible because of the high degree of over-
</p>
<p>determination in small-molecule analysis, with a data/parameter ratio greater than about 10, whereas
</p>
<p>in protein analysis this ratio is more likely to be 2&ndash;3, and the estimated standard deviations will
</p>
<p>therefore not be so reliable. Several methods have been suggested for estimating the rms valueffiffiffiffiffiffiffiffiffiffi
jDrj2
</p>
<p>q
of the atomic coordinate error of a protein structure. They include the method given by Read
</p>
<p>[81] which derives jDrj2 from the slope of a plot of ln(sA) against (sin y/l)2 in the formula
</p>
<p>ln sA&frac14;
1
</p>
<p>2
ln
</p>
<p>P
PP
N
</p>
<p>� �
</p>
<p>�
8p3
</p>
<p>3
jDrj2
</p>
<p>sin2 y
</p>
<p>l2
</p>
<p>� �
</p>
<p>(10.14)
</p>
<p>where
P
</p>
<p>N is
P
</p>
<p>f 2j summed over all atoms in the structure, and
P
</p>
<p>P is
P
</p>
<p>f 2j summed over all atoms
</p>
<p>in the partial structure; sA is obtained from
</p>
<p>&eth;sA&THORN;
2&frac14;
</p>
<p>P
</p>
<p>&eth;jEobsj
2�jEobsj
</p>
<p>2&THORN;&eth;jEcalcj
2�jEcalcj
</p>
<p>2&THORN;
h i
</p>
<p>P
</p>
<p>&eth;jEobsj
2�jEobsj
</p>
<p>2&THORN;
2
P
</p>
<p>&eth;jEcalcj
2�jEcalcj
</p>
<p>2&THORN;
2
</p>
<p>� �1=2 (10.15)
</p>
<p>where jEobsj is obtained from Fo and jEcalcj is calculated from the structure; both observed jEoj
2
and
</p>
<p>calculated jEcalcj
2
should equal unity, Table 4.3. Values of sA estimated from this method are not
</p>
<p>absolute, and can only be considered to be on a relative basis for comparison purposes. According to
</p>
<p>current opinion, a value of sA around 0.3 Å or less is acceptable.
</p>
<p>10.11 Geometry Validation: Final Checks
</p>
<p>The geometrical features of proteins and related compounds have been subjected to intense study for
</p>
<p>over 50 years [62, 63]. Information on the bond lengths, bond angles and torsion angles of smaller
</p>
<p>peptide structures are available in the Cambridge Crystallographic Data Base, and of proteins in the
</p>
<p>PDB (see Appendix D). It is possible, therefore, to inspect a newly refined protein structure in detail
</p>
<p>and to make assessments of its geometry in terms of standard features. If either unusual features or
</p>
<p>possible errors are suspected, it will be necessary to look at them in detail and to repeat some of the
</p>
<p>refinement protocols so as to make corrections where necessary.
</p>
<p>10.11.1 Bond Lengths, Bond Angles, Planarity, and Chirality
</p>
<p>Refinement programs have built-in checks that assess the derived geometry of the current model in
</p>
<p>terms of standard values. Average values of features such as peptide bonds are also calculated together
</p>
<p>with the spread of values found in the structure. Examination of these values provides a useful check on
</p>
<p>possible errors. Software such as PROCHECK [64], MolProbity [65] and WHAT-CHECK [71] have
</p>
<p>10.11 Geometry Validation: Final Checks 539</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_4#Tab3_4">http://dx.doi.org/10.1007/978-1-4614-3954-7_4#Tab3_4</a></div>
</div>
<div class="page"><p/>
<p>been developed for protein structure verification, including bond length, bond angle, and conforma-
</p>
<p>tional checks, and should be used as a matter of routine in protein structure analysis.
</p>
<p>Known planar groups, such as phenyl rings, will probably have been constrained, and the effect of
</p>
<p>this should be checked to see if it has been successful. Chiral carbon atoms are usually of known hand
</p>
<p>and this can be restrained in some refinement programs. The chiral volume calculated as the scalar triple
</p>
<p>product of the vectors from the central atom to three attached atoms will have the correct sign only for
</p>
<p>the correct enantiomeric form. If a side group has been attached to a chiral center incorrectly, the chiral
</p>
<p>volume will have the wrong sign and this can be corrected by rebuilding the model at this location.
</p>
<p>10.11.2 Conformation
</p>
<p>Main-Chain Conformation
</p>
<p>Fig. 10.21 Ramachandran plot for the protein SNAII (Sambucus nigra agglutinin II), a Gal/GalNAc specific lectin
extracted from the bark of the Elder, calculated with the program PROCHECK. The plot provides an overall picture of
</p>
<p>the (f, C) coordinates (white squares) observed for this protein. There are no unusual or disallowed conformations
</p>
<p>present (Gal, galactose; GalNAc N-acetylgalactosamine)
</p>
<p>540 10 Proteins and Macromolecular X-Ray Analysis</p>
<p/>
</div>
<div class="page"><p/>
<p>The peptide torsion angle o, Sect. 10.1, is rarely outside the limits of 180 � 20� (trans) or
0 � 20� (cis). Deviations from these limits in the model should be examined carefully. The
features of protein secondary structure that are governed by folding of the polypeptide in terms of
</p>
<p>the torsion angles f and C are well documented and can be visualized and evaluated conve-
</p>
<p>niently in terms of a Ramachandran plot [66]. This provides an overall picture of the (f, C)
</p>
<p>values observed for the protein structure, all of which are plotted on to the same diagram. The
</p>
<p>shaded areas of Fig. 10.21 show the regions of allowed conformation more or less as originally
</p>
<p>plotted. These regions were defined from studies of protein models in which the side-chains were
</p>
<p>all alanine (ala), which has a short methyl side group, and are based on those (f, C) combina-
</p>
<p>tions that are free from steric clashes.
</p>
<p>Although protein structures are found to conform mainly to the allowed regions displayed here,
</p>
<p>some extension outside and between these areas is accepted on account of the large data base of known
</p>
<p>proteins now available. In Fig. 10.21 the white squares denote the (f,C) coordinates calculated for the
</p>
<p>refined structure of Sambucus nigra agglutinin II (SNAII) [12, 82], which is a lectin with high
</p>
<p>specificity for binding to the sugar b-D-Galactose (Gal) and its derivative b-D-N-acetylgalactosamine
</p>
<p>(GalNAc), and is extracted from the bark of the Elder, S. nigra. All of the (f,C) values are seen to be
</p>
<p>within allowed regions, indicating that it is an acceptable structure. Any points derived from a new
</p>
<p>protein structure that are significantly outside the regions shown here should be carefully examined.
</p>
<p>Side-Chain Conformations: Molecular Graphics Display Programs and Bound Ligands
In terms of stereochemistry there is a great variety amongst the 20 amino acids that occur in proteins,
</p>
<p>and their individual conformations are well characterized from known structures. As with all
</p>
<p>empirical data, tolerances in these conformations exist and should be borne in mind when evaluating
</p>
<p>a new structure. Programs such as PROCHECK, MolProbity and WHAT-CHECK provide a detailed
</p>
<p>analysis of the side-chain conformations in a given protein structure and flag any unusual or dubious
</p>
<p>ones for further investigation.
</p>
<p>Molecular Graphics Display Programs
One of the rewards of completing a protein structure is to see graphical representations of it. In the
</p>
<p>1950s these were produced by artists, resulting in what Bernal described as &ldquo;extremely beautiful
</p>
<p>pictures. . ..&rdquo; In his innovative 1969 book [83] Dickerson used clear and imaginative drawings
</p>
<p>produced by the artist Irving Geis to great effect. Geis&rsquo;s renowned paintings, sketches and drawings
</p>
<p>helped generations of researchers to visualize complex macromolecules, such as cytochromes and
</p>
<p>viruses [84]. Since then computerized molecular graphics has been developed and researchers are
</p>
<p>fortunate to have at their disposal a wide range of software to produce accurate representations of
</p>
<p>their structures which have a beauty of their own. An important feature of these programs is that
</p>
<p>they allow different parts of a structure to be selected and presented graphically in a variety of
</p>
<p>ways. One of the first programs to be developed, which is still extremely popular is RASMOL [85]
</p>
<p>several examples of which can be found in Chap. 11. The CCP4 suite has a versatile graphics
</p>
<p>facility [24] as can be seen in Fig. 10.22. Graphics produced by Accelrys Discovery [86] are also
</p>
<p>extremely powerful and relatively easy to obtain from the versatile menu provided, Fig. 10.23. This
</p>
<p>program also features a useful energy minimization procedure. Other useful graphics programs
</p>
<p>include: MOLSCRIPT [87], PYMOL [88], and ArgusLab [89], all of which produce excellent
</p>
<p>molecular drawings, such as the drawing of an a-helix in Fig. 10.24, of all sizes and has useful
</p>
<p>energy minimization and other features; MOE [90] also has a wide range of useful features.
</p>
<p>Finally there is another aspect of protein structure analysis which involves the binding of ligands to
</p>
<p>the main protein molecule [91]. For example lectin proteins bind to specific cell surface sugar
</p>
<p>molecules, and lectin structures are often determined with bound sugars. In the case of MLI, the
</p>
<p>10.11 Geometry Validation: Final Checks 541</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_11">http://dx.doi.org/10.1007/978-1-4614-3954-7_11</a></div>
</div>
<div class="page"><p/>
<p>Fig. 10.23 General view of MLI galactose complex [92], drawn with Accelrys Discovery [85]. Regions of a-helix
(red) and b-sheet (blue) are shown as ribbons
</p>
<p>Fig. 10.22 CCP4 molecular graphics example showing a periplasmic ferric enterobactin binding protein Ceu of
Campylobacter jejeuni with a bound dinuclear ferric enterobactin mimic [95]
</p>
<p>542 10 Proteins and Macromolecular X-Ray Analysis</p>
<p/>
</div>
<div class="page"><p/>
<p>structure crystallizes in the presence either of b-D-galactose [27, 92, 93], which shows two galactose
</p>
<p>binding sites in the B-chain, or with lactose [94]. The binding sites are designated: site 1, towards the
</p>
<p>N-terminus (a); and site 2, towards the C-terminus (b). Figure 10.25a, b were drawn from the pdb file
</p>
<p>for MLI-lactose with LIGPLOT [95], which automatically analyses and depicts the interactions
</p>
<p>between the protein and the ligand in each case. Similarities in the binding of galactose in MLI and
</p>
<p>ricin [28] were revealed using this software. The sites are quite complex and it is virtually impossible to
</p>
<p>represent the three-dimensional graphics in a direct two-dimensional diagram. This problem is
</p>
<p>overcome by LIGPLOT which produces an easy to read cartoon type representation.
</p>
<p>Fig. 10.24 Part of the a-helix from MLI [92] drawn with ArgusLab [88]. Hydrogen bonds are shown as white dots
</p>
<p>10.11 Geometry Validation: Final Checks 543</p>
<p/>
</div>
<div class="page"><p/>
<p>Fig. 10.25 The two sugar binding sites of MLI [27, 91, 92]. (a) Site 1, towards the N-terminus. (b) Site 2, towards the
C-terminus. Drawn from the pdb file with LIGPLOT [94], which automatically analyses and depicts the interactions
</p>
<p>between the protein and the ligand. Similarities in the binding of galactose in MLI and ricin [28] are revealed using this
</p>
<p>software. The sites are geometrically quite complex and it is virtually impossible to represent the three-dimensional
</p>
<p>graphics in an undistorted two-dimensional diagram. This problem is overcome by LIGPLOT which produces these
</p>
<p>comprehensible cartoon-type representations
</p>
<p>544 10 Proteins and Macromolecular X-Ray Analysis</p>
<p/>
</div>
<div class="page"><p/>
<p>10.12 Humidity Control and the Use of Cryoprotectants in Protein
Crystallography
</p>
<p>Soaking a protein crystal in a cryoprotectant causes it to suffer an abrupt shock which may disrupt the
</p>
<p>crystalline order. This strongly suggests that most cryoprotectants have a lower relative humidity than
</p>
<p>typical crystal well solutions. In order to overcome this problem experiments are now undertaken
</p>
<p>with a view to controlling the humidity environment at the crystal to promote data enhancement and
</p>
<p>identify the relative humidity of the mother liquor. Exposure of the crystal to X-rays is then carried
</p>
<p>out while using a humidity control device to maintain the crystal at the humidity of the mother liquor,
</p>
<p>thus providing an impression of crystal quality at room temperature. It is then possible to gradually
</p>
<p>lower the humidity with the aim of improving the diffraction quality and possibly removing the need
</p>
<p>to cryoprotect at all (below about 92% relative humidity). The humidity is controlled by passing a
</p>
<p>humid stream of air over the sample. It is now considered to be an advantage to be able to review
</p>
<p>crystals at room temperature so that the effect of the addition of cryo-protectant can be monitored.
</p>
<p>This is not possible with crystals in volatile mother liquors as the mother liquor never stabilizes. Such
</p>
<p>experiments are expedited by the use of crystal mounting robots which can handle 16 samples in one
</p>
<p>setting (http://doc.diamond.ac.uk/MXManual/Sample-Changers.html). [We are indebted to Dr. C.
</p>
<p>Lobley and Professor T. Sorensen, Diamond Light Source, for providing this update.] [97, 98].
</p>
<p>10.13 Problems
</p>
<p>10.1. Typical specimens used in single crystal studies have dimensions between 0.1 and 0.5 mm.
</p>
<p>Crystals of small organic molecules have unit-cell dimensions in the range 5&ndash;100 Å and
</p>
<p>macromolecules 20&ndash;400 Å. Estimate the number of unit cells in a protein crystal of dimen-
</p>
<p>sions 0.2, 0.3, 0.4 mm if the unit-cell dimensions are a &frac14; 30 Å, b &frac14; 40 Å, c &frac14; 50 Å. Repeat
the calculation for a small organic molecule with unit cell dimensions a &frac14; 10 Å, b &frac14; 12 Å,
c &frac14; 15 Å. Assume that all crystal and unit cell angles are 90�. Comment on the relative
diffracting power of the crystals.
</p>
<p>10.2. Diffraction data are to be measured on an image plate using SR. The exact wavelength of the
</p>
<p>radiation can vary from time to time and needs to be calibrated. A smear of amorphous beeswax
</p>
<p>was placedwhere the crystal is normally mounted at a distance of 300mm from the image plate.
</p>
<p>The wax has a dominant powder line of spacing 3.5 Å and produced a ring of diameter 140 mm
</p>
<p>on the image plate. Calculate the wavelength of the X-radiation.
</p>
<p>10.3. Refer to Problem 10.2. It is required to collect data for a protein crystal with a known unit-cell
</p>
<p>length of about 300 Å. If a spot separation of at least 1 mm is required, comment on whether a
</p>
<p>crystal-to-detector distance of 450 mm would be appropriate.
</p>
<p>10.4. During data processing for the crystal in Problems 10.2 and 10.3, the unit-cell parameters were
</p>
<p>found to be a &frac14; b &frac14; 110.79 Å, c &frac14; 308.53 Å, a &frac14; b &frac14; 90�, g &frac14; 120�; the Laue symmetry
was 6
</p>
<p>m
mm. The only limiting conditions in the diffraction pattern were for 00l: l &frac14; 6n. State the
</p>
<p>space group or possible space groups for this crystal.
</p>
<p>10.5. For mistletoe lectin MLI, the space group is P6522 with Z &frac14; 12,MP is 63000 and the unit-cell
parameters are a &frac14; b &frac14; 110.79 Å, c &frac14; 308.53 Å, a &frac14; b &frac14; 90�, g &frac14; 120�. Investigate possible
values for m, the number of molecules per asymmetric unit, and s, the fractional solvent content.
</p>
<p>10.6. For mistletoe lectin MLI (see Problem 10.5), there are 21000 recorded reflections in a data set
</p>
<p>having dmin &frac14; 2.9 Å for wavelength l &frac14; 0.8 Å. Estimate the total number of reflections
</p>
<p>10.13 Problems 545</p>
<p/>
<div class="annotation"><a href="http://doc.diamond.ac.uk/MXManual/Sample-Changers.html">http://doc.diamond.ac.uk/MXManual/Sample-Changers.html</a></div>
</div>
<div class="page"><p/>
<p>expected in the data set. If 21000 unique reflections were recorded what is the percentage
</p>
<p>completeness of the data set?
</p>
<p>10.7. A protein of relative molar mass 27000 to be studied by X-ray crystallography, crystallizes in
</p>
<p>space group P21 with 2 molecules per unit cell. Estimate the number of non-hydrogen atoms
</p>
<p>to be located in the analysis given that the hydrogen content of the protein is 10% and
</p>
<p>assuming that of the 40%, by weight of solvent (water) in the unit cell only 10% is ordered.
</p>
<p>If the unit-cell dimensions are a &frac14; 58.2 Å, b &frac14; 38.3 Å, c &frac14; 54.2 Å and b &frac14; 106.5�, estimate
the data/parameter ratios corresponding to 6, 2.5, and 1.0 Å resolution data sets, and comment
</p>
<p>on these values.
</p>
<p>10.8. Derive the general equivalent positions for space group P32.What conditions limit reflections
</p>
<p>in this space group? (Web Appendix WA5 may be helpful.)
</p>
<p>10.9. The choice of methods for the determination of protein crystal structures includes (a) MR, (b)
</p>
<p>MIR, and (c) MAD. Discuss the circumstances under which each of these methods would be
</p>
<p>the most appropriate to use for a given protein.
</p>
<p>10.10. When employing Patterson Search methods for structure analysis, under what circumstances
</p>
<p>would you expect the search molecule to be (a) very similar in size to, and (b) much smaller
</p>
<p>than, the target molecule? Discuss your answer in some detail.
</p>
<p>References
</p>
<p>1. Branden C and Tooze J (1990) Introduction to Protein Structure, 2nd edition, Garland Publishing, NY
</p>
<p>2. Bernal JD, Crowfoot D (1934) Nature 133:794
</p>
<p>3a. Rosemeyer MA (1961) PhD thesis. University of London
</p>
<p>3b. Bernal JD, Carlisle CH, Rosemeyer MA (1959) Acta Crystallogr 12:227
</p>
<p>4. http://hamptonresearch.com/
</p>
<p>5. D&rsquo;Arcy A, Villard F, Marsh M (2007) Acta Crystallogr D63:550
</p>
<p>6. Matsumura H et al (2011) J Synchrotron Radiat 18:16
</p>
<p>7. http://www.jove.com/video/2285/protein-crystallization-for-x-ray-crystallography
</p>
<p>8. Petsko GA (1985) Methods Enzymol 14:147
</p>
<p>9. Blundell TL, Jenkins JA (1977) Chem Soc Rev 6:139
</p>
<p>10. McRee DE (1993) Practical protein crystallography. Academic, San Diego
</p>
<p>11. http://www.oxford-diffraction.com
</p>
<p>12. Niwa H (2001) PhD thesis, University of London
</p>
<p>13. Br&euro;unger AT (1992) Nature 355:472
14. Tickle IJ et al (2000) Acta Crystallogr D56:442&ndash;450
</p>
<p>15. See Appendix D
</p>
<p>16. Diamond R (1969) Acta Crystallogr A25:43
</p>
<p>17. Ford GC (1974) J Appl Crystallogr 7:555
</p>
<p>18. Rossmann MG (1979) J Appl Crystallogr 12:225
</p>
<p>19. Greenhough TJ, Helliwell J (1982) J Appl Crystallogr 15:493
</p>
<p>20. Otwinowski Z, Minor W. International tables for crystallography. In: Rossman MG, Arnold E (eds) IUCr [DENZO
</p>
<p>and SCALEPACK]. Kluwer, Dordrecht, pp 226&ndash;235
</p>
<p>21. Leslie A (1993) Data processing. In: Sawyer L, Isaacs N, Bailey S (eds) Proceedings of the CCP4 study weekend.
</p>
<p>Daresbury Laboratories, Warrington, pp 44&ndash;51
</p>
<p>22. Rossmann MG, van Beek CG (1999) Acta Crystallogr D55:1631
</p>
<p>23. Sanderson MR (2007) Macromolecular crystallography conventional and high-throughput methods, Chapter 5.
</p>
<p>Oxford University Press, Oxford
</p>
<p>24. Leslie A (2006) Acta Crystallogr D62:48
</p>
<p>25. Plugrath JW (1999) Acta Crystallogr D55:1718
</p>
<p>26. Contact: Lynn Yarris; lcyarris@lbl.gov
</p>
<p>27. Sweeney EC et al (1997) Proteins 28:586
</p>
<p>28. Monfort W et al (1987) J Biol Chem 262:5398
</p>
<p>29. Matthews BW (1974). In: Neurath M, Hill RL (eds) The proteins, 3rd edn. Academic, New York
</p>
<p>546 10 Proteins and Macromolecular X-Ray Analysis</p>
<p/>
<div class="annotation"><a href="http://hamptonresearch.com/">http://hamptonresearch.com/</a></div>
<div class="annotation"><a href="http://www.jove.com/video/2285/protein-crystallization-for-x-ray-crystallography">http://www.jove.com/video/2285/protein-crystallization-for-x-ray-crystallography</a></div>
<div class="annotation"><a href="http://www.oxford-diffraction.com">http://www.oxford-diffraction.com</a></div>
</div>
<div class="page"><p/>
<p>30. Matthews BW (1974) J Mol Biol 82:513
</p>
<p>31. Quillin ML, Matthews BW (2000) Acta Crystallogr D56:791
</p>
<p>32. Cooley JW, Tukey JW (1965) Math Comput 19:297
</p>
<p>33. Barrett AN, Zwick M (1971) Acta Crystallogr A27:6
</p>
<p>34. Ten Eyck LF (1985) Methods Enzymol 115:324
</p>
<p>35. http://www.ysbl.york.ac.uk/~ccp4mg/ccp4mg_help/maps.html
</p>
<p>36. Wood SP (1990) Protein purification applications: a practical approach. In: Harris ELV, Angal S (eds). IRL Press,
</p>
<p>Oxford
</p>
<p>37. Rossman MG, Blow DB (1961) Acta Crystallogr 14:641
</p>
<p>38. Navaza J (1994) Acta Crystallogr D57:1367
</p>
<p>39. Navaza J (2008) Acta Crystallogr D64:70
</p>
<p>40. Br&euro;unger AT (1993) X-PLOR version 3.1 manual. Yale University Press, New Haven, CT
41. Vagin A, Teflyakov A (1997) J Appl Crystallogr 30:1022
</p>
<p>42. Lebedev AA, Vagin A, Mushudov GN (2008) Acta Crystallogr D64:33
</p>
<p>43. Br&euro;unger AT et al (1998) Acta Crystallogr D54:905
44. Storoni LC, McCoy AJ, Read RJ (2004) Acta Crystallogr D60:432
</p>
<p>45. McCoy AJ, Grosse-Kunstleve RW, Adams PD, Winn MD, Storoni LC, Read RJ (2007) J Appl Crystallogr
</p>
<p>40:658&ndash;674 (see also http://www-structmed.cimr.cam.ac.uk/Personal/randy/pubs/pubs.html)
</p>
<p>46. Navaza J (1994) Acta Crystallogr A50:157
</p>
<p>47. http://www.ccp4.ac.uk/html/polarrfn.html
</p>
<p>48. Read RJ (2001) Acta Crystallogr D57:1373
</p>
<p>49. Tickle IJ, Driessen HPC (1996) In: Jones C et al (eds) Methods in molecular biology, vol 56. Humana Press, Totowa
</p>
<p>50. Rutenber E, Robertus J (1991) Proteins 10:260
</p>
<p>51. Katzin BJ et al (1991) Proteins, structure, function and genetics 10:251 (the ricin structure can be accessed at http://
</p>
<p>www.rcsb.org/pdb, Internet Protein Data Bank ID code 2AAI)
</p>
<p>52. Sweeney EC (1996) PhD thesis, University of London (1996)
</p>
<p>53. Erskine PT et al (1997) Nat Struct Biol 4:1025
</p>
<p>54. Cowtan K (1994) Joint CCP4 and ESRF-EACBM newsletter. Protein Crystallogr 31:34
</p>
<p>55. Wang BC (1985) Methods Enzymol 115:90
</p>
<p>56. Zhang KYJ, Main P (1990) Acta Crystallogr A46:377
</p>
<p>57. Baker D et al (1994) Acta Crystallogr D49:429
</p>
<p>58. Swanson S (1994) Acta Crystallogr D50:695
</p>
<p>59. Sayre D (1974) Acta Crystallogr A30:180
</p>
<p>60. Bricogne G (1974) Acta Crystallogr A30:395
</p>
<p>61. Schuller D (1996) Acta Crystallogr D52:425
</p>
<p>62. Pauling L, Corey RB (1951) Proc Natl Acad Sci U S A 37:729
</p>
<p>63. Engh RA, Huber R (1991) Acta Crystallogr A47:392
</p>
<p>64. Laskowski RA, MacArthur MW, Moss DS, Thornton JM (1993) J Appl Crystallogr 26:283
</p>
<p>65. Lovell SC et al (2003) Proteins 50:437
</p>
<p>66. Ramachandran GN, Sasisekheran V (1968) Adv Prot Chem 23:283
</p>
<p>67. Jones TA (1978) J Appl Crystallogr 11:268
</p>
<p>68. Kleywegt GJ, Jones TA (1997) Methods Enzymol 277:525
</p>
<p>69. McCree DE (1993) J Struct Biol 125:156
</p>
<p>70. Emsley P, Cowtan K (2004) Acta Crystallogr D60:2126
</p>
<p>71. http://swift.cmbi.ru.nl/gv/refs/
</p>
<p>72. Haneef I et al (1985) Acta Crystallogr A41:426
</p>
<p>73. Murshudov GN, Vagin AA, Dodson EJ (1997) Acta Crystallogr D53:240
</p>
<p>74. http://www.ccp4.ac.uk/stwk/2004/abstracts/roversi.html
</p>
<p>75. Sheldrick GM (2008) Acta Crystallogr A64:112
</p>
<p>76. Ramachandran GN et al (1963) J Mol Biol 7:95
</p>
<p>77. Adams PD et al (2010) Acta Crystallogr D66:213
</p>
<p>78. http://www.phenix-online.org/
</p>
<p>79. Rice LM, Br&euro;unger AT (1994) Proteins 19:277
80. Zegers I (1994) PhD thesis. Free University of Brussels
</p>
<p>81. Read RJ (1986) Acta Crystallogr A42:140
</p>
<p>82. Maveyraud L, Niwa H, Guillet V, Svergun DI, Konarev P, Palmer RA, Peumans WJ, Rougé P, Van Damme EJM,
</p>
<p>Reynolds CD, Mourey L (2009) Proteins 75:89
</p>
<p>83. Dickerson RE, Geis I (1969) The structure and action of proteins. Harper Row, New York
</p>
<p>84. http://www.hhmi.org/news/geis.html
</p>
<p>References 547</p>
<p/>
<div class="annotation"><a href="http://www.ysbl.york.ac.uk/~ccp4mg/ccp4mg_help/maps.html">http://www.ysbl.york.ac.uk/~ccp4mg/ccp4mg_help/maps.html</a></div>
<div class="annotation"><a href="http://www-structmed.cimr.cam.ac.uk/Personal/randy/pubs/pubs.html">http://www-structmed.cimr.cam.ac.uk/Personal/randy/pubs/pubs.html</a></div>
<div class="annotation"><a href="http://www.ccp4.ac.uk/html/polarrfn.html">http://www.ccp4.ac.uk/html/polarrfn.html</a></div>
<div class="annotation"><a href="http://www.rcsb.org/pdb">http://www.rcsb.org/pdb</a></div>
<div class="annotation"><a href="http://www.rcsb.org/pdb">http://www.rcsb.org/pdb</a></div>
<div class="annotation"><a href="http://swift.cmbi.ru.nl/gv/refs/">http://swift.cmbi.ru.nl/gv/refs/</a></div>
<div class="annotation"><a href="http://www.ccp4.ac.uk/stwk/2004/abstracts/roversi.html">http://www.ccp4.ac.uk/stwk/2004/abstracts/roversi.html</a></div>
<div class="annotation"><a href="http://www.phenix-online.org/">http://www.phenix-online.org/</a></div>
<div class="annotation"><a href="http://www.hhmi.org/news/geis.html">http://www.hhmi.org/news/geis.html</a></div>
</div>
<div class="page"><p/>
<p>85. Sayle R (1994) RASMOL, a molecular visualisation program. Glaxo Research and Development, Bruno, IJ
</p>
<p>86. http://accelrys.com/about/
</p>
<p>87. http://www.avatar.se/molscript/download.html
</p>
<p>88. http://pymol.org/
</p>
<p>89. Thompson MA. Planaria software LLC, Seattle, WA, http://www.arguslab.com
</p>
<p>90. http://www.chemcomp.com/aboutccg-focus.htm
</p>
<p>91. Palmer RA, Niwa H (2003) Biochem Soc Trans 31:973
</p>
<p>92. Sweeney EC, Palmer RA, Pfuller UJ (1993) Mol Biol 234:1279
</p>
<p>93. Niwa H, Tonevitsky AG, Agapov II, Saward S, Pf&euro;uller U, Palmer RA (2003) Eur J Biochem 270:2739
94. Niwa (2003) personal communication
</p>
<p>95. http://www.uclb-elicensing.com/informatics_software/cheminformatics/protein_ligand_2D_representation_ win-
</p>
<p>dows.html
</p>
<p>96. Russi S et al (2011) Inducing phase changes in crystals of macromolecules: status and perspectives for controlled
</p>
<p>crystal dehydration. J Struct Biol 175(2):236&ndash;243
</p>
<p>97. Sanchez-Weatherby J et al (2009) Improving diffraction by humidity control: a novel device compatible with X-ray
</p>
<p>beamlines. Acta Crystallogr D Biol Crystallogr 65(12):1237&ndash;1246
</p>
<p>Bibliography
</p>
<p>Branden C, Tooze J (1999) Introduction to protein structure, 2nd edn. Garland, New York
</p>
<p>Carter W (1990) Protein and nucleic acid crystallization. Academic, New York
</p>
<p>Drenth J (1999) Principles of protein X-ray crystallography, 2nd edn. Springer, New York
</p>
<p>McRee DE (1993) Practical protein crystallography. Academic, San Diego
</p>
<p>McPherson A (1989) Preparation and analysis of protein crystals. Krieger Publishing, Melbourne, FL
</p>
<p>Sanderson MR, Skelly JV (eds) (2007) Macromolecular crystallography. Oxford University Press, Oxford
</p>
<p>Sherwood D, Cooper J (2011) Crystals, X-rays and proteins. Oxford University Press, Oxford
</p>
<p>Chayen NE, Helliwell JR, Snell EH (2010) Macromolecular crystallization and crystal perfection (International Union
</p>
<p>of Crystallography Monographs on Crystallography)
</p>
<p>http://www.ccp4.ac.uk/schools/China-2011/talks/Oasis_Fan.pdf
</p>
<p>548 10 Proteins and Macromolecular X-Ray Analysis</p>
<p/>
<div class="annotation"><a href="http://accelrys.com/about/">http://accelrys.com/about/</a></div>
<div class="annotation"><a href="http://www.avatar.se/molscript/download.html">http://www.avatar.se/molscript/download.html</a></div>
<div class="annotation"><a href="http://pymol.org/">http://pymol.org/</a></div>
<div class="annotation"><a href="http://www.arguslab.com">http://www.arguslab.com</a></div>
<div class="annotation"><a href="http://www.chemcomp.com/aboutccg-focus.htm">http://www.chemcomp.com/aboutccg-focus.htm</a></div>
<div class="annotation"><a href="http://www.uclb-elicensing.com/informatics_software/cheminformatics/protein_ligand_2D_representation_windows.html">http://www.uclb-elicensing.com/informatics_software/cheminformatics/protein_ligand_2D_representation_windows.html</a></div>
<div class="annotation"><a href="http://www.uclb-elicensing.com/informatics_software/cheminformatics/protein_ligand_2D_representation_windows.html">http://www.uclb-elicensing.com/informatics_software/cheminformatics/protein_ligand_2D_representation_windows.html</a></div>
<div class="annotation"><a href="http://www.ccp4.ac.uk/schools/China-2011/talks/Oasis_Fan.pdf">http://www.ccp4.ac.uk/schools/China-2011/talks/Oasis_Fan.pdf</a></div>
</div>
<div class="page"><p/>
<p>Neutron Diffraction from Single
Crystals 11
</p>
<p>11.1 Introduction
</p>
<p>Whenwediscussed the variation of observable symmetry according to the nature of the examining probe,
</p>
<p>Sect. 1.4, we stated that one possible probe is a neutron beam. As an example, when the structure of
</p>
<p>elemental chromium is examined by X-ray diffraction, it shows a body-centered cubic arrangement,
</p>
<p>Fig. 11.1a. Elemental chromium has the electronic configuration (Ar 3d5 4s1); it is antiferromagnetic at
</p>
<p>room temperature, and its electron spins, arising from the unpaired electrons, give rise to the magnetic
</p>
<p>structure shown inFig. 11.1b. Themagneticmoment of the neutron interactswith the permanent dipole of
</p>
<p>chromium to form this structure, and consequently the diffraction record shows a primitive cubic
</p>
<p>structure.
</p>
<p>Neutron diffraction can distinguish between isotopes, aid the study of alloy systems, such as the
</p>
<p>Cu&ndash;Zn phases (similar atomic numbers) and magnetically ordered structures, and reveal the presence of
</p>
<p>light atoms in the presence of heavy atoms, such as hydrogen in sodium hydride&mdash;a difficult matter with
</p>
<p>X-rays.
</p>
<p>As an example of the detection of hydrogen atoms, we cite the structure of potassium dihydrogen
</p>
<p>phosphate, KH2PO4. Hydrogen atoms scatter neutrons strongly, but because the scattering cross section
</p>
<p>(q.v.) is negative, the nuclear density contours are also negative for this atom; it also produces a high
</p>
<p>background owing to incoherent scattering.Deuterium, however, has a large, positive coherent scattering
</p>
<p>with only small background. Thus, it is common practice to replace hydrogen by deuterium in neutron
</p>
<p>diffraction studies of hydrogen atom positions in crystal structures. Notable examples of this application
</p>
<p>are the precise location of the deuterium (hydrogen) atom positions in potassium dihydrogen phosphate,
</p>
<p>KH2PO4, Fig. 11.2, and sucrose, C12H22O11.
</p>
<p>Neutron diffraction is also an important and powerful technique used in combination with X-ray
</p>
<p>diffraction for locating hydrogen atoms or protons in both small molecules and macromolecules. For
</p>
<p>example, neutron scattering has been used in protein crystallography to distinguish between the side
</p>
<p>chain C&frac14;Ooxygen atom and the NH2 nitrogen atom in asparagine and glutamine residues because of the
significant difference between the neutron scattering lengths of oxygen and nitrogen: this distinction is
</p>
<p>difficult to achievewithX-ray diffraction. It is also possible to distinguish betweenN&ndash;H andC&ndash;Hgroups
</p>
<p>on histidine rings (q.v.). In both cases the proper positioning of these groups is an important factor in the
</p>
<p>understanding of the biological action of the protein; see also Sect. 9.4.3. It is also possible to ascertain
</p>
<p>whether a given nitrogen atom is protonated (deuterated) or unprotonated, an ambiguity which again
</p>
<p>may be difficult to resolve by other means [1].
</p>
<p>M. Ladd and R. Palmer, Structure Determination by X-ray Crystallography:
</p>
<p>Analysis by X-rays and Neutrons, DOI 10.1007/978-1-4614-3954-7_11,
# Springer Science+Business Media New York 2013
</p>
<p>549</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Sec1_1">http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Sec1_1</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_9#Sect20_9">http://dx.doi.org/10.1007/978-1-4614-3954-7_9#Sect20_9</a></div>
</div>
<div class="page"><p/>
<p>11.1.1 Refinement of Hydrogen Atom Positions
</p>
<p>The neutron diffraction results for sucrose give the value 1.09 Å for the C&ndash;H bond length, whereas by
</p>
<p>X-rays it is shown to be 0.98 Å; the corresponding O&ndash;H bond length values are 0.98 and 0.80 Å.
</p>
<p>These results are of interest in connection with the refinement of a structure. If the hydrogen atoms
</p>
<p>are included in the calculations but are not themselves refined, a not uncommon procedure, then with
</p>
<p>X-ray data it is reasonable to constrain the C&ndash;H bond length to be about 0.98 Å rather than 1.09 Å.
</p>
<p>The polarity of the C&ndash;H bond leads to a displacement of the electron density of hydrogen toward the
</p>
<p>carbon atom. Thus, the position of the electron density maximum for the hydrogen atom will not
</p>
<p>coincide with that given by neutron diffraction.
</p>
<p>Neutron diffraction investigations can also be carried out on powdered crystalline samples, and we
</p>
<p>address this topic in Chap. 12. We proceed next to discuss neutron diffraction in detail and to give
</p>
<p>examples of structures determined by neutron crystallography.
</p>
<p>Fig. 11.1 Unit cell of the crystal structure of elemental chromium. (a) By X-rays: body-centred cubic. (b) By neutrons:
primitive cubic. The arrows represent the magnetic moment vector directions in the metal atoms
</p>
<p>Fig. 11.2 Projections onto the x,y plane of the neutron scattering density for deuterated potassium dihydrogen
phosphate, KD2PO4, in the room temperature tetragonal structure. (a) Direct synthesis, showing all atoms in the
structure; the most intense peaks arise from the K and P atoms, which are superimposed in this projection. The other
</p>
<p>peaks are O atoms (full lines) and D atoms (dashed lines, indicating the negative scattering amplitude of this
</p>
<p>species). (b) Corresponding difference synthesis, in which only the D atoms appear. It is noteworthy that the
diffraction ripples (series termination errors) around the heavy-atom positions in (a) are subtracted out in (b), so that
the contours of D are free from the distorting effects of the ripples
</p>
<p>550 11 Neutron Diffraction from Single Crystals</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_12">http://dx.doi.org/10.1007/978-1-4614-3954-7_12</a></div>
</div>
<div class="page"><p/>
<p>11.2 Neutrons, Neutron Sources, and Data Collection
</p>
<p>11.2.1 Neutrons
</p>
<p>In 1923, some years after the historical Laue experiment, de Broglie introduced the wave-particle
</p>
<p>duality theory, which proposed that every material particle can behave also as a non-electromagnetic
</p>
<p>wave, opposed to the electromagnetic waves which include X-rays, light, and other types of wave
</p>
<p>phenomena such as heat and sound. According to de Broglie&rsquo;s theory, the particle wavelength l is
</p>
<p>given by:
</p>
<p>l &frac14; h=p &frac14; h=mv (11.1)
</p>
<p>where h is the Planck constant and p, m, and v are, respectively, the momentum, mass, and speed of
</p>
<p>the particle.
</p>
<p>Neutrons are sub-atomic particles which are bound to most atomic nuclei. A neutron has no net
</p>
<p>electric charge and a mass slightly greater than that of a proton. The number of protons in a nucleus is
</p>
<p>numerically equal to the atomic number and defines the type of the element. The number of neutrons
</p>
<p>determines the isotope of the element. For example, the abundant 12C isotope has six protons and six
</p>
<p>neutrons, while the rare radioactive 14C isotope has six protons and eight neutrons. While bound
</p>
<p>neutrons in stable nuclei are stable, free neutrons are unstable and undergo b-decay with a mean
</p>
<p>lifetime of just under 15 min. The nucleus of deuterium (heavy hydrogen) contains one proton and
</p>
<p>one neutron, whereas the hydrogen nucleus contains just one proton. This distinction has an important
</p>
<p>consequence in some neutron diffraction experiments, as we shall see.
</p>
<p>11.2.2 Neutron Sources
</p>
<p>Free neutrons are produced in both nuclear fission and fusion processes. Thus, large dedicated neutron
</p>
<p>sources are either nuclear fission reactors, or high flux spallation sources in which protons have been
</p>
<p>accelerated to high energies and directed on to a target so as to produce free neutrons. Spallation
</p>
<p>describes what happens when a high-energy proton is accelerated into a heavy target: a number of
</p>
<p>spallation particles, including neutrons, are produced. For every proton striking the nucleus, twenty to
</p>
<p>thirty neutrons are expelled. Meson production limits spallation efficiency above 140 MeV. At the
</p>
<p>1 GeV proton energy level, the Spallation Neutron Source, Sect. 11.4.2, requires 30 MeV per neutron
</p>
<p>produced. All of these sources produce neutron beams that are useful in neutron diffraction experi-
</p>
<p>ments. Inside a reactor, thermal neutrons have an average energy corresponding to 308 K (35 �C), but
there is actually a Boltzmann distribution of neutron energies, so that the emerging beam will have a
</p>
<p>range of energies (or temperature) and, more importantly, a range of wavelengths, according to
</p>
<p>(11.1). Hence an essentially &ldquo;white&rdquo; beam of neutron wavelengths is available, which can be limited,
</p>
<p>if required, to a desired wavelength range.
</p>
<p>11.2.3 Neutron Data Collection
</p>
<p>Monochromatic Neutron Data Collection
This method of data collection as the title suggests involves the isolation of a single neutron
</p>
<p>wavelength or limited range of wavelengths using suitable filters, corresponding closely to the use
</p>
<p>11.2 Neutrons, Neutron Sources, and Data Collection 551</p>
<p/>
</div>
<div class="page"><p/>
<p>of monochromatic X-radiation discussed in Sect. 3.1.4. Thus, when a beam of neutrons emanating
</p>
<p>from a reactor is slowed down and selected properly, wavelengths appropriate for crystal diffraction
</p>
<p>experiments can be produced. However, these processes cause the neutron beam to be significantly
</p>
<p>depleted in intensity, which has the over-riding consequence of prolonging data collection times to
</p>
<p>the unmanageable proportions of days or weeks. As a result, alternative methods of data collection
</p>
<p>have been developed.
</p>
<p>Laue Diffraction Neutron Data Collection
In this now commonly used technique, a white neutron beam is used to produce Laue diffraction
</p>
<p>photographs or image records [2], Sect. 5.4.1ff, that can be processed to create a neutron intensity
</p>
<p>data set. This approach allows the full incident intensity of the neutron beam to be used without the
</p>
<p>use of filters, which enables data to be collected far more rapidly.
</p>
<p>Monochromatic versus Laue Diffraction Neutron Data Collection
It is of interest to compare these two methods of data collection. In particular, what are the limits to
</p>
<p>data set completeness imposed by a Laue experiment vs. those of monochromatic data collection?
</p>
<p>These methods of data collection are common to both X-ray and neutron diffraction techniques.
</p>
<p>Rigorous theoretical considerations [3, 4] have laid the foundations of Laue data collection, which,
</p>
<p>with the complications inherent in the method imposed by a lack of direct knowledge of the
</p>
<p>wavelength associated with a given diffraction intensity, initially seemed to be an unlikely method
</p>
<p>of successful data collection. From the Bragg equation, when a crystal diffracts a polychromatic beam
</p>
<p>of X-rays or neutrons, many orders of each Bragg reflection, hkl, 2h,2k,2l,. . .., may occur simulta-
</p>
<p>neously and overlap exactly in scattering angle. The overlap of these multiple orders along a ray, a
</p>
<p>central line in reciprocal space, poses a problem for Laue methods. It has been shown [3] that:
</p>
<p>1. More than 83% of all Bragg reflections occur on single rays when experimental values of lmax and
</p>
<p>lmin are used.
</p>
<p>2. This proportion depends only on the ratio of lmax to lmin and not on the space group, unit-cell
</p>
<p>dimensions, crystal orientation, or the limiting resolution of the crystal, d�max, provided d
�
</p>
<p>max is less
</p>
<p>than 2/lmax.
</p>
<p>3. The total number of single rays, like the total number of all stimulated Bragg reflections, is
</p>
<p>approximately proportional to the wavelength range.
</p>
<p>4. The proportion of reflections at a given resolution d* that lie on single or double rays depends
</p>
<p>markedly on d*, and on the ratio of lmax to lmin&mdash;it is generally lower at low resolution than at
</p>
<p>high.
</p>
<p>5. Restricted angular acceptance of the detector can reduce significantly both the proportion and the
</p>
<p>total number of single rays.
</p>
<p>6. Agreement between the theoretical distributions and those derived from analysis of X-ray Laue
</p>
<p>photographs of macromolecular crystals, and from extensive computer simulations, is good.
</p>
<p>It is evident that, under a wide variety of experimental conditions, the effect of multiple orders is
</p>
<p>not a serious limitation on the use of the Laue method for structure determination. An analysis has
</p>
<p>also been presented [4] of the angular distribution of reflections in Laue diffraction, with particular
</p>
<p>application to the spatial overlap problem in synchrotron macromolecular crystallography. This
</p>
<p>shows the factors that govern the spatial overlap of spots and indicates tactics for experimental
</p>
<p>design. The analysis is also relevant to polychromatic neutron diffraction. These considerations have
</p>
<p>led to the development of both hardware and software [5] for Laue diffraction which has now come to
</p>
<p>be accepted as a standard collection method capable of producing data of high quality, completeness,
</p>
<p>and resolution for both X-ray and neutron diffraction from single crystals.
</p>
<p>552 11 Neutron Diffraction from Single Crystals</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3# Sec5_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3# Sec5_3</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_5#Sec15_5">http://dx.doi.org/10.1007/978-1-4614-3954-7_5#Sec15_5</a></div>
</div>
<div class="page"><p/>
<p>Time-Resolved Laue Neutron Diffraction Data Collection Using a Spallation Source
This is a third method of neutron data collection currently undergoing intense development. Neutrons
</p>
<p>are produced in a spallation source by bombarding a metal target with protons from a particle
</p>
<p>accelerator or synchrotron. Essentially each neutron from a spallation source can be assigned its
</p>
<p>own wavelength. Data collection by this method is analogous to carrying out hundreds of monochro-
</p>
<p>matic experiments simultaneously. Further details on this type of experiment are given in Sect. 11.4.
</p>
<p>11.2.4 Thermal Neutrons
</p>
<p>The term &ldquo;thermal neutron&rdquo; is used to describe any free neutron that has been ejected from an atomic
</p>
<p>nucleus and has kinetic energy corresponding to the average energy of the particles at 308 K. Thermal
</p>
<p>neutrons are produced by slowing down more energetic atomic neutrons by passing them through a
</p>
<p>moderator. The neutron wavelength can be calculated from a consideration of its kinetic energy E:
</p>
<p>E &frac14; 1
2
mnv
</p>
<p>2 &frac14; kT (11.2)
</p>
<p>where mn is the neutron mass and v its speed; k is the Boltzmann constant and T the absolute
</p>
<p>temperature. It follows that
</p>
<p>v &frac14; &eth;2kT=mn&THORN;1=2
</p>
<p>and since mv &frac14; p &frac14; &eth;2kTmn&THORN;1=2,
</p>
<p>l &frac14; h=p &frac14; A=T1=2 (11.3)
</p>
<p>where A &frac14; h=&eth;2kmn&THORN;1=2 and is a constant for a given particle type such as neutrons. Since mn&frac14;1.6749
� 10�27 kg, A &frac14; 3.0810 � 10�9mK1/2; hence, at 308 K
</p>
<p>l &frac14; 1:756� 10�10 m &frac14; 1:756A
�
</p>
<p>(11.4)
</p>
<p>This wavelength is suitable for crystal diffraction measurements. Since the neutron beam may be
</p>
<p>designed to comprise a range of wavelengths, the diffraction data are then processed using a Laue
</p>
<p>procedure, as referred to in Sect. 11.2. As in contemporary X-ray diffraction experiments, the sample
</p>
<p>may be flash frozen in order to prevent decomposition in the neutron beam and thus enable a full set of
</p>
<p>diffraction data to be collected on a single crystal sample.
</p>
<p>11.3 Neutron Scattering
</p>
<p>Whereas the scattering of X-rays increases with increasing atomic number and decreases with sin y
</p>
<p>for a given wavelength, neutron scattering factors, scattering lengths, do not depend on atomic
</p>
<p>number in any regular manner. One important difference from X-ray diffraction is that the hydrogen
</p>
<p>scattering length is comparable to those of other atoms, but of negative sign. This means that for
</p>
<p>hydrogen, 1H, the beam scattered by the nucleus differs in phase by 180� from the scattering from a
nucleus in the same position that has a positive scattering factor. However, deuterium, 2H, has a
</p>
<p>positive neutron scattering length that is comparable in magnitude to that of hydrogen. Consequently,
</p>
<p>11.3 Neutron Scattering 553</p>
<p/>
</div>
<div class="page"><p/>
<p>it is frequently a useful option to replace hydrogen with deuterium, that is, to deuterate crystal
</p>
<p>samples prior to neutron diffraction data collection in order to render positive the subsequent neutron
</p>
<p>density at these sites. In that way, they are easier to handle, for example, when using molecular
</p>
<p>graphics. This is not always possible if the sample is unstable. It should also be noted that in general
</p>
<p>all C&ndash;H hydrogen atoms are not exchangeable in the deuteration process. Because neutron scattering
</p>
<p>for hydrogen or deuterium is comparable in magnitude with that from other atoms, it is not so easy to
</p>
<p>classify atoms as &ldquo;light&rdquo; or &ldquo;heavy&rdquo; as we have done for X-ray diffraction, and this introduces a
</p>
<p>technical difficulty in the actual solving of crystal structures ab initio from neutron diffraction data.
</p>
<p>An advantage, however, is afforded by the use of 2H over 1H, that is a much lower associated
</p>
<p>background scattering level, because the incoherent scattering cross section of 2H is about 40 times
</p>
<p>less than that of 1H. Thus, the signal-noise ratio is greatly enhanced in neutron diffraction data
</p>
<p>measured from deuterated samples [6].
</p>
<p>We have assumed tacitly that the number of hkl data is the same for both X-ray and neutron
</p>
<p>diffraction. Usually, neutron data sets are less extensive than their X-ray counterparts, and this serves
</p>
<p>to exaggerate still further these problems. It is therefore almost always necessary to solve a crystal
</p>
<p>structure initially from X-ray data, and subsequently to use this structure as a basis for a detailed
</p>
<p>neutron analysis, as has been done with the three examples of varying structural complexity given
</p>
<p>later in this chapter.
</p>
<p>11.3.1 Neutron Scattering Lengths
</p>
<p>Neutron scattering lengths for different atoms can be obtained from various sources including the
</p>
<p>website of the National Institute of Standards and Technology [7, 8]. Examples of scattering length
</p>
<p>data incorporated into the SHELX program suite [9] for the atoms H, D, N, O, F, and S are listed in
</p>
<p>Table 11.1.
</p>
<p>11.4 Experimental Neutron Diffraction Data Collection
</p>
<p>The practical use of neutron diffraction requires access to highly specialized neutron sources. These
</p>
<p>installations are purpose built to cater for a variety of neutron physics applications, in the same way as
</p>
<p>synchrotron installations are designed not only for X-ray diffraction, but can accommodate a wide
</p>
<p>range of studies from Raman spectroscopy to circular dichroism studies, for example. A list of
</p>
<p>neutron sources available to scientists on a global scale can be found at the end of this chapter. For
</p>
<p>present purposes, experimental arrangements that are available at two of these installations will be
</p>
<p>described in some detail.
</p>
<p>Table 11.1 Neutron scattering lengths for common
elements
</p>
<p>Atom Atomic number Scattering length (fm)
</p>
<p>H 1 �3.739
D 2 6.671
</p>
<p>N 7 9.360
</p>
<p>O 8 5.803
</p>
<p>F 9 5.650
</p>
<p>S 16 2.847
</p>
<p>554 11 Neutron Diffraction from Single Crystals</p>
<p/>
</div>
<div class="page"><p/>
<p>11.4.1 LADI-III and VIVALDI at ILL, Grenoble
</p>
<p>At the Institut Laue-Langevin (ILL), the Laue diffraction instrument LADI-III produces extremely
</p>
<p>high-quality diffraction data. It employs a quasi-white cold-neutron beam with a large-area cylindri-
</p>
<p>cal detector based on neutron image plates and is used mainly for single crystal studies of relatively
</p>
<p>small proteins at medium or high resolution with the objective of locating individual hydrogen atoms
</p>
<p>in compounds of special interest, water molecules, or small ligands that can be deuterated to improve
</p>
<p>their scattering power. Good data sets from macromolecular structures that tend to have relatively
</p>
<p>weak scattering are measured routinely on this instrument. Protein crystals with unit-cell dimensions
</p>
<p>up to about 150 Å are accommodated and sample sizes can be as small as 0.1&ndash;0.2 mm3.
</p>
<p>The facility VIVALDI, Fig. 11.3, (Very-Intense Vertical-Axis Laue Diffractometer) [10, 11] was
</p>
<p>set up for use in 2001. This instrument allows large volumes of reciprocal space to be surveyed very
</p>
<p>quickly using an image-plate photographic Laue technique with a white neutron beam. Figure 11.4 is
</p>
<p>an example of a neutron Laue frame taken on VIVALDI. As well as single crystal analysis, it can
</p>
<p>accommodate applications for magnetism, nuclear charge-density waves, high-pressure studies, and
</p>
<p>structural phase transitions and allows rapid preliminary investigation of new materials even when
</p>
<p>only small single crystals are available. VIVALDI uses a technique similar to that of LADI-III, but
</p>
<p>employs a thermal neutron beam and is very well suited to fast data collection for crystal structures
</p>
<p>with smaller unit cells. Both VIVALDI and LADI-III have the detector cylinder axis vertical. In the
</p>
<p>case of VIVALDI, this facilitates acceptance of an &ldquo;Orange&rdquo; cryostat, and in both installations, the
</p>
<p>image plates are mounted and read on the inside of the cylinder in order to improve the detection
</p>
<p>efficiency. Consequently with VIVALDI, it is now possible to study larger biological complexes using
</p>
<p>smaller crystals than was previously possible [12]. As on LADI, the neutron-sensitive plates are based
</p>
<p>on the BaFBr storage-phosphor doped with Eu(II) ions, which is also used for X-ray image plates,
</p>
<p>Sects. 5.7.3 and 12.4.2, but with Gd2O3 added. The gadolinium nuclei act as neutron scintillators by
</p>
<p>creating a cascade of g-rays and conversion electrons: an electron is released from the atomic shell by
</p>
<p>transferring the energy of a g-quantum emitted from the same nucleus to its electron. Its kinetic energy
</p>
<p>is equal to the energy of the g-quantum reduced by the binding energy of the electron.
</p>
<p>The full thermal spectrum can be accepted without detrimental overlap of reflections for primitive
</p>
<p>unit cells up to 25 Å edge length. An optional filter based on multilayer super-mirrors can be placed
</p>
<p>upstream to deflect wavelengths longer than 3 Å which would otherwise contribute primarily to the
</p>
<p>background. A time-of-flight spectrum analyzer is located downstream to monitor changes in the
</p>
<p>wavelength. The entire instrument pivots on air cushions around the filter to allow selection of
</p>
<p>Fig. 11.3 Diagrammatic layout of the ultra-high speed VIVALDI neutron diffractometer (reproduced from the
&ldquo;yellow book&rdquo; by courtesy of the Institut Laue-Langevin, Grenoble)
</p>
<p>11.4 Experimental Neutron Diffraction Data Collection 555</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_5#Sec30_5">http://dx.doi.org/10.1007/978-1-4614-3954-7_5#Sec30_5</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_12# Sec12_12">http://dx.doi.org/10.1007/978-1-4614-3954-7_12# Sec12_12</a></div>
</div>
<div class="page"><p/>
<p>wavelengths shorter than 3 Å, wavelengths longer than 3 Å, or the unfiltered beam. The beam size
</p>
<p>used is up to 10 � 10 mm2 with a flux (unfiltered) of 109 neutron cm�2 s&ndash;1.
</p>
<p>The Orange cryostat can be used to select a sample temperature in the range 1.5&ndash;300 K. A typical
</p>
<p>experiment produces a small number of stationary-crystal, stationary-detector Laue diffraction
</p>
<p>patterns, distinguished by means of 20&ndash;30� rotations of the crystal about the vertical axis. The
Laue data-analysis software developed on LADI from the CCP4 suite (see Appendix D) is used to
</p>
<p>produce the final data set for subsequent analysis.
</p>
<p>Although VIVALDI was unparalled in data collection speed when it was introduced in 2001, it is
</p>
<p>now even faster. This has been brought about partly through the introduction of the more efficient Fuji
</p>
<p>&ldquo;Niimura Special&rdquo; white image plates, the replacement of some sections of the neutron guide by super-
</p>
<p>mirror guides, and a guide realignment, resulting in about three times as much neutron flux at the
</p>
<p>sample. Overall there has been a ninefold gain in efficiency. A short video showing the instrument set
</p>
<p>and each stage of the measurement of a neutron diffraction pattern is available on the ILLwebsite [13].
</p>
<p>11.4.2 Oak Ridge National Laboratory (ORNL) [14]
</p>
<p>The High Flux Isotope Reactor (HFIR) and the HB-3A Diffractometer
Operating at 85 MW, HFIR is the highest flux reactor-based source of neutrons for research in the
</p>
<p>United States, and it provides one of the highest steady-state neutron fluxes of any research reactor in
</p>
<p>the world. The thermal and cold neutrons produced by HFIR are used to study physics, chemistry,
</p>
<p>materials science, engineering, and biology. The intense neutron flux, constant power density, and
</p>
<p>constant-length fuel cycles are used by more than 200 researchers each year for neutron scattering
</p>
<p>research into the fundamental properties of condensed matter.
</p>
<p>Fig. 11.4 A neutron Laue diffraction frame
</p>
<p>556 11 Neutron Diffraction from Single Crystals</p>
<p/>
</div>
<div class="page"><p/>
<p>For single crystal diffraction studies, a four-circle diffractometer installation, the HB-3A,
</p>
<p>Fig. 11.5a, is available; see Sect. 5.6ff. The Huber goniometer has a full w-circle with a 10 K
</p>
<p>closed-cycle helium refrigerator. The detector is 3He with a seven-anode array in a honeycomb
</p>
<p>pattern. The upper limit of 2y is 100�. A multilayer [110] silicon wafer monochromator with the
reflection from planes of the &lt;011&gt; zone ensures sharp diffraction peaks in specified ranges of
</p>
<p>detector angles by control of the horizontal radius of curvature.
</p>
<p>Any plane from the &lt;011&gt; zone can be set in the Bragg reflection position, but only the (155),
</p>
<p>(133), (022) with (044), and (111) with (333) reflection planes are of practical interest. For a fixed
</p>
<p>monochromator angle of 48�, these reflections provide principal incident wavelengths of 0.618 Å,
1.01 Å, 1.56 Å, and 2.55 Å, respectively. A PC-based Lab-View system provides user-friendly
</p>
<p>Fig. 11.5 (a) The HB-3A
neutron single crystal
</p>
<p>diffractometer at ORNL,
</p>
<p>Tennessee (reproduced by
</p>
<p>courtesy of Dr Cao Huibo,
</p>
<p>ORNL, Tennessee.)
</p>
<p>(b) Detector array for the
MaNDi instrument. The
</p>
<p>detectors are designed to
</p>
<p>cover a large solid-angle
</p>
<p>to record most of the
</p>
<p>neutrons scattered from
</p>
<p>a single crystal sample
</p>
<p>regardless of the reflection
</p>
<p>angle. The instrument
</p>
<p>design accommodates this
</p>
<p>by situating detectors
</p>
<p>approximately spherically
</p>
<p>around the sample
</p>
<p>11.4 Experimental Neutron Diffraction Data Collection 557</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_5#Sec21_5">http://dx.doi.org/10.1007/978-1-4614-3954-7_5#Sec21_5</a></div>
</div>
<div class="page"><p/>
<p>diffractometer control and data acquisition. The beam size is 5 � 5 mm2, and the minimum crystal
size is 1 mm3. The maximum crystal dimension is about 4 mm. The flux on the sample is estimated to
</p>
<p>be greater than 5 � 106 neutron cm�2 s�1, whereas that of VIVALDI is 109 neutron cm�2 s�1.
This instrument is suitable for a wide range of small unit-cell crystallography studies, from
</p>
<p>structure solution and refinement to charge and nuclear density mapping. Problems have been
</p>
<p>addressed from chemistry, physics, materials science, and mineralogy. Specific areas of study include
</p>
<p>hydrogen bonding and weak interactions, organometallics, supra-molecular chemistry and crystal
</p>
<p>engineering, metal hydrides, charge density, pharmaceuticals, and magnetic structures.
</p>
<p>Design for the Future: The ORNL Spallation Neutron Source (SNS) High-Resolution
Time-of-Flight Single Crystal Macromolecular Neutron Diffractometer (MaNDi)
The ORNL spallation neutron source is an accelerator-based neutron source and forms a unique
</p>
<p>facility providing for the most intense pulsed neutron beams in the world dedicated to scientific
</p>
<p>research and industrial development.
</p>
<p>It is reasonable to assume from our previous considerations that neutron macromolecular crystal-
</p>
<p>lography (NMC) is capable of providing accurate hydrogen atom positions, protonation states and
</p>
<p>hydration states, as well as hydrogen/deuterium exchange information in macromolecular crystals. In
</p>
<p>fact, this is possible even at a moderate resolution of approximately 2 Å. In contrast, in order to
</p>
<p>observe hydrogen atoms via ultra-high-resolution macromolecular X-ray crystallography, diffraction
</p>
<p>data beyond 1.0 Å are required. X-ray diffraction beyond this limit can only be achieved with highly
</p>
<p>ordered crystals, which is rarely achieved. The introduction of the Spallation Neutron Source (SNS)
</p>
<p>at ORNL with over an order of magnitude increase in neutron flux, in combination with advances in
</p>
<p>neutron optics and detectors, structural genomics, and protein deuteration, provides new opportu-
</p>
<p>nities for NMC to become a routine and essential structural tool for enzymology, structural biology,
</p>
<p>and functional genomics.
</p>
<p>In order to satisfy the needs of the structural biology community, a dedicated, extremely
</p>
<p>high-quality high-resolution time-of-flight single crystal macromolecular neutron diffractometer
</p>
<p>(MaNDi) is now under construction at the SNS [15]. The design of this instrument has been
</p>
<p>optimized so that it will enable data collection rates over 50 times faster than existing facilities.
</p>
<p>It depends on the high regularity of the neutron pulses from the spallation source and the ability of
</p>
<p>highly sensitive detectors to distinguish the velocity-dependent wavelength of each particle. This
</p>
<p>process has been likened to measuring diffraction data from innumerable monochromatic experi-
</p>
<p>ments and will enable studies of crystals with lattice constants substantially larger than currently
</p>
<p>possible. It is expected that the unprecedented speed and resolution limits achievable with MaNDi
</p>
<p>for NMC experiments will greatly advance the fields of structural biology, enzymology, and
</p>
<p>computational chemistry.
</p>
<p>The SNS macromolecular diffractometer (MaNDi) will be a state of the art high-resolution
</p>
<p>macromolecular crystal diffractometer. Optimized for rapid data collection from large structures,
</p>
<p>MaNDi will achieve a 1.5 Å resolution from crystal volumes between 0.1 and 1.0 mm3 with unit-
</p>
<p>cell repeats in the order of 150 Å. The instrument will use a decoupled hydrogen moderator for
</p>
<p>optimal resolution and separation of Bragg peaks. The design utilizes a 24 m flight path and a
</p>
<p>variable wavelength bandwidth of 2.7 Å to accommodate different types of experiments. This
</p>
<p>bandwidth variation is achieved by the use of three disc choppers in the incident flight path. With
</p>
<p>crystals larger than 1 mm3, it will be possible to obtain useful data in the resolution range 2.0&ndash;2.5 Å
</p>
<p>for unit-cell repeats of up to 300 Å, a revolution in neutron macromolecular crystallography
</p>
<p>(NMC). Figure 11.5b shows the design of the detector array for the MaNDi instrument.
</p>
<p>558 11 Neutron Diffraction from Single Crystals</p>
<p/>
</div>
<div class="page"><p/>
<p>11.4.3 Other Neutron Sources
</p>
<p>We conclude this section with a global list of neutron facilities:
</p>
<p>North America
</p>
<p>&bull; Spallation Neutron Source, Oak Ridge
</p>
<p>&bull; Los Alamos Neutron Science Center (LANSCE)
</p>
<p>&bull; University of Missouri Research Reactor Center
</p>
<p>&bull; High Flux Isotope Reactor, Oak Ridge
</p>
<p>&bull; Canadian Neutron Beam Centre, Chalk River, Canada
</p>
<p>&bull; Indiana University Cyclotron Facility
</p>
<p>Europe
</p>
<p>&bull; ISIS-Rutherford-Appleton Laboratories, UK
</p>
<p>&bull; Institut Laue-Langevin, Grenoble, France
</p>
<p>&bull; Leon Brillouin Laboratory, Saclay, France
</p>
<p>&bull; Berlin Neutron Scattering Center, Germany
</p>
<p>&bull; GEMS at Helmholtz-Zentrum Geesthacht, Germany
</p>
<p>&bull; Juelich Center for Neutron Science, Germany
</p>
<p>&bull; FRM-II, Munich, Germany
</p>
<p>&bull; Budapest Neutron Centre, Hungary
</p>
<p>&bull; RID, Delft, The Netherlands
</p>
<p>&bull; SINQ, Paul Scherrer Institut (PSI), Switzerland
</p>
<p>&bull; Frank Laboratory of Neutron Physics, Dubna, Russia
</p>
<p>&bull; St. Petersburg Neutron Physics Institute, Gatchina, Russia
</p>
<p>Asia and Australia
</p>
<p>&bull; ISSP Neutron Scattering Laboratory, Tokai, Japan
</p>
<p>&bull; JAEA Research Reactors, Tokai, Japan
</p>
<p>&bull; KENS Neutron Scattering Facility, Tsukuba, Japan
</p>
<p>&bull; Hi-Flux Advanced Neutron Application Reactor, Korea
</p>
<p>&bull; Bragg Institute, ANSTO, Australia
</p>
<p>11.5 Deuteration and Perdeuteration
</p>
<p>Deuteration
Deuteration even of simple compounds is best effected by collaborating with an experienced chemist.
</p>
<p>In simple terms, the obvious way to effect H&ndash;D interchange would be to soak the compound of
</p>
<p>interest, or a crystal of it, in D2O. This may work in some cases, but it is nevertheless a hit-or-miss
</p>
<p>experiment. Generally speaking, whereas C&ndash;H hydrogen atoms would probably not be exchanged in
</p>
<p>this way, O&ndash;H and N&ndash;H hydrogen atoms would.
</p>
<p>Deuteration may be carried out on pre-grown crystals or on the pure sample itself prior to
</p>
<p>crystallization. With some crystal materials that are sensitive to liquids, it may not be possible to
</p>
<p>perform the experiments necessary to effect deuteration, Sect. 11.7; help can be obtained from
</p>
<p>appropriate websites [16&ndash;18].
</p>
<p>Perdeuteration or Biodeuteration
Neutron protein crystallography provides a powerful complement to X-ray crystallography by
</p>
<p>enabling key hydrogen atoms to be located in biological structures where they cannot be seen by
</p>
<p>X-ray analysis alone. The neutron Laue diffractometer LADI, run jointly by EMBL and ILL at the
</p>
<p>11.5 Deuteration and Perdeuteration 559</p>
<p/>
</div>
<div class="page"><p/>
<p>ILL high flux reactor in Grenoble, is a dedicated facility for neutron protein crystallography at high-
</p>
<p>resolution (1.5 Å) and provides 10&ndash;100-fold gains in efficiency compared with conventional neutron
</p>
<p>diffractometers. The availability of a fully deuterated (perdeuterated or biodeuterated) protein
</p>
<p>eliminates the hydrogen incoherent scattering contribution to the background and brings approxi-
</p>
<p>mately tenfold improvements in the signal to noise ratios.
</p>
<p>The production of deuterium-labeled macromolecules is carried out using bacterial expression
</p>
<p>systems incorporating perdeuterated glycerol (CH2OH)2CHOH as the sole carbon source in the Deuter-
</p>
<p>ation Laboratory at Grenoble [19]. Protein crystals are grown using methods described in Chap. 10, but
</p>
<p>with D2O replacing H2O. A recent paper [20] provides an up to date example of this type of research.
</p>
<p>The aim of the facility at the Institut Laue-Langevin, in collaboration with the European Molecular
</p>
<p>Biology Laboratory, is to provide a focus for scientists wishing to make their own deuterated
</p>
<p>materials prior to carrying out neutron scattering experiments. In support of the effort to encourage
</p>
<p>deuteration studies, a major EU grant has been obtained jointly by ILL, EMBL, the University of
</p>
<p>Oxford, and the Institut de Biologie Structurale, Grenoble in order to develop different protocols and
</p>
<p>procedures of deuteration both for neutron scattering and NMR studies.
</p>
<p>11.6 Examples of Structure Determination by Neutron Crystallography
</p>
<p>We now discuss in detail three recent examples of single-crystal analyses that have been enhanced
</p>
<p>through the use of neutron diffraction. The first example is an atomic resolution study of a small
</p>
<p>organic molecule, the second is an intermediate resolution study of a lectin protein molecule, and the
</p>
<p>third is a high-resolution analysis of the 11-membered peptide cyclosporin H. All three examples
</p>
<p>combine, for reasons given previously, the neutron study with a parallel X-ray study, and they
</p>
<p>illustrate different aspects of the current state of neutron diffraction technology, indicating its
</p>
<p>versatility and potential. We show during the determination of the first structure how potential pitfalls
</p>
<p>of &ldquo;black box&rdquo; crystallography can be encountered.
</p>
<p>11.7 X-Ray and Neutron Structure of 1,8-(3,6,9-Trioxaundecane-1,
11-diyldioxy)-9,10-dihydro-10-10 dimethylanthracene-9-ol [21]
</p>
<p>This compound was prepared in high yield by bis-methylation of anthralin at the 10-position,
</p>
<p>followed by bis-O-alkylation to give a ketone (I) that was reduced to the alcohol by sodium amalgam.
</p>
<p>A detailed molecular structure of the formal hydrolysis product, Fig. 11.6, of the parent cationic form,
</p>
<p>Fig. 11.7, was required.
</p>
<p>The product has been shown from combined X-ray and neutron diffraction analyses to be
</p>
<p>1,8-(3,6, 9-trioxaundecane-1,11-diyldioxy)-9,10-dihydro-10,10-dimethylanthracene-9-ol, and is the
</p>
<p>parent compound for an important series of derivatives; unequivocal positions for the hydrogen
</p>
<p>atoms were obtained from this study. Both sets of data were measured on single crystal four-circle
</p>
<p>diffractometers with monochromatic radiation. In terms of the X-ray analysis, there were 96 carbon
</p>
<p>560 11 Neutron Diffraction from Single Crystals</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_10">http://dx.doi.org/10.1007/978-1-4614-3954-7_10</a></div>
</div>
<div class="page"><p/>
<p>atoms and 28 oxygen atoms per unit cell. A solution of this structure by direct methods was therefore
</p>
<p>quite feasible. In the neutron structure diffraction analysis, a further 128 hydrogen atoms of compara-
</p>
<p>ble size (in neutron diffraction terms) to the carbon and oxygen atoms were present per unit cell.
</p>
<p>Solution of the neutron structure ab initio was therefore not attempted; it was solved by a Fourier
</p>
<p>expansion of the X-ray structure. This process was not impaired by the changes in unit cell related to
</p>
<p>the different experimental temperatures used in the two analyses. The crystal and structure analyses
</p>
<p>data for this compound are summarized in Table 11.2.
</p>
<p>11.7.1 Experimental
</p>
<p>Sample Preparation
The slow evaporation of an ethanolic solution yielded crystals of melting point 151&ndash;153�C, which
were used in the crystallographic studies. No attempt was made to exclude water from the purified
</p>
<p>solvent used for crystallization. However, the presence of water in the crystalline material was not
</p>
<p>evident from its spectra.
</p>
<p>Fig. 11.7 The cationic form of the compound illustrated by Fig. 11.6
</p>
<p>Fig. 11.6 Themolecular structure of 1,8-(3,6,9-trioxaundecane-1,11-diyldioxy)-9,10-dihydro-10,10-dimethylanthracene-9-ol
</p>
<p>11.7 X-Ray and Neutron Structure. . . 561</p>
<p/>
</div>
<div class="page"><p/>
<p>X-Ray Diffraction Measurements
A crystal was centered on a Nonius CAD4 diffractometer. The software package CAD4 Express 1988
</p>
<p>was used to determine a unit cell and orientation matrix based on 25 reflections selected by the
</p>
<p>software, employing Cu Ka radiation. The diffractometer system, in automatic mode, favored an
</p>
<p>orthorhombic C-centered cell which refined to the dimensions listed in Table 11.2.
</p>
<p>The assignment of crystal system by the software was based solely on the close proximity of the
</p>
<p>cell angles to 90�. This software was designed to collect one asymmetric unit of intensity data for the
initially assigned point group, there being no automatic Laue check facility at this stage. However, it
</p>
<p>was possible to carry out a check for Laue symmetry after data collection, but this was not an easy
</p>
<p>procedure to carry out with this version of the software. We note that the later version of the software
</p>
<p>Table 11.2 X-ray and neutron crystal data for 1,8-(3,6,9-Trioxaundecane-11-diyldioxy)-9,10-
dihydro-10,10,10-dimethylanthracene-9-ol
</p>
<p>Chemical formula C24H30O6&middot;H2O
</p>
<p>Mr 432.50
</p>
<p>Color/shape Colorless/plates
</p>
<p>X-ray Neutron
</p>
<p>Temperature 293(2) K 15 K
</p>
<p>Crystal system/space group Monoclinic/P21 Monoclinic/P21
</p>
<p>Unit cell dimensions/Å
</p>
<p>(25 reflections with 25 &lt; y &lt; 30�)
a 12.845(5) 12.605(5)
b 14.575(3) 14.458(3)
c 13.779(2) 13.588(2)
b 117.72(2)� 117.12(2)�
</p>
<p>Unit cell volume/Å3 Vc 2283.6(1) 2204.1(1)
</p>
<p>Z (2 molecules/asymmetric unit) 4 4
</p>
<p>Dc (g cm
�3) 1.258 1.302
</p>
<p>mc (cm
�1) 0.754 0.754
</p>
<p>Diffractometer/scan Nonius CAD4/o�2y D19 ILL/o
Radiation Cu Ka Neutron
</p>
<p>Monochromator Graphite Graphite
</p>
<p>Wavelength (Å) 1.54180 1.538, 1.312
</p>
<p>Crystal dimensions (mm) 0.35, 0.25, 0.20 2.5, 1.6, 0.2
</p>
<p>Reflections measured 17753 1931
</p>
<p>Independent/observed 9092/9092 1931/1881
</p>
<p>Rint 0.0459
</p>
<p>ymax� 72.93 46.12
</p>
<p>Range of h, k, l � 15;�18;�17 10� 11; 0� 13; �9� 12
</p>
<p>Data processing CAD4-Express 1992 RETREAT
</p>
<p>Structure solution SHELX-S-86 From X-ray
</p>
<p>Structure refinement SHELXL-97 SHELXL-97
</p>
<p>Refinement method Full-matrix least-squares on F2
</p>
<p>Data/restraints/parameters 9092/217/605 1881/1/507
</p>
<p>Goodness-of-fit on F2 0.908 0.779
</p>
<p>Final R [I &gt; 2s(I)] R1/wR2 0.0619/0.1506 0.0462/0.1184
</p>
<p>R indices (all data) R1/wR2 0.1001/0.1688 0.04093/0.1320
</p>
<p>Absolute structure parameter 0.25(35)
</p>
<p>Largest diff. peak/hole 0.232/-0.260 0.055/�0.058
</p>
<p>562 11 Neutron Diffraction from Single Crystals</p>
<p/>
</div>
<div class="page"><p/>
<p>(CAD4 Express, 1994) does carry out a Laue check in automatic mode prior to data collection proper.
</p>
<p>Intensity data were routinely collected for one asymmetric unit of assumed orthorhombic reciprocal
</p>
<p>space, for 1 &lt; y &lt; 74�.
The space group was assigned by inspection of the systematic absences in the intensity data, after
</p>
<p>processing in C2221. The unit cell was calculated to include eight molecules of C24H30O6 (two
</p>
<p>molecules per asymmetric unit). It was discovered subsequently, during refinement of the structure,
</p>
<p>that there were additionally two water molecules per asymmetric unit. It should be noted that the X-
</p>
<p>ray results were obtained under some pressure, being required urgently in order to coordinate with a
</p>
<p>scheduled neutron data collection time at ILL. Therefore, the X-ray analysis proceeded at this stage
</p>
<p>without any hint that an extraordinary situation would soon arise as a result of a very unusual
</p>
<p>geometrical relationship between the unit-cell parameters. These parameters appeared to be quite
</p>
<p>reasonable at this stage of the analysis. The crystal had moderate diffracting power and good single-
</p>
<p>crystal quality was indicated from the diffractometer readings incorporated into the software. The first
</p>
<p>X-ray data set, after processing and correction for Lorentz and polarization effects, gave 2572 unique
</p>
<p>intensities, 1395 with Fo &gt; 4s(Fo), and with few symmetry repeats Rint was indeterminate.
The direct methods program SHELX-S produced no solution in space group C2221. The mono-
</p>
<p>clinic space group C21 was then tried (non-standard, c-unique setting: determine what this is
</p>
<p>equivalent to and draw a space-group diagram). It produced an outstanding solution giving a
</p>
<p>promising E map which contained the essential features of both molecules in the unit cell, despite
</p>
<p>being based on only part of the monoclinic data set, through an incorrect assumption of the point
</p>
<p>group. This structure failed to refine properly, owing to the incompleteness of the &ldquo;orthorhombic&rdquo;
</p>
<p>intensity data set. The best R-factor achieved with this model using isotropic thermal displacement
</p>
<p>parameters was 0.24, with a data/parameter ratio of 10.8.
</p>
<p>A second set of intensity data was collected, with the unit cell constrained to be monoclinic, an
</p>
<p>option allowed by the software. The data for the monoclinic unit-cell and other data are listed in
</p>
<p>Table 11.2. A total of 9092 unique, corrected (excluding absorption) intensities was recorded, 4804
</p>
<p>with Fo &gt; 4s(Fo), with Rint &frac14; 4.59%. Subsequent refinement of the X-ray structure, Sect. 11.7.2,
confirmed this assignment of unit cell and space group.
</p>
<p>Neutron Diffraction Measurements
Prior to the start of the neutron diffraction data collection, the reflection indices and cell parameters of the
</p>
<p>octant of X-ray diffraction data that had been collected in the monoclinic C-centered setting were
</p>
<p>transformed to a primitive, monoclinic setting with b-axis unique and space group P21. A crystal with
</p>
<p>approximate dimensions 2.5, 1.6, 0.2 mm was mounted on a vanadium support on the D19 four-circle
</p>
<p>diffractometer at ILL, with the longest unit-cell dimension of the plate along the diffractometer f-axis.
</p>
<p>The neutron wavelength was 1.538 Å and an initial scan of reciprocal space located six reflections which
</p>
<p>could be indexed and a preliminary orientation matrix calculated. Additional reflections, which were
</p>
<p>expected to be of significant intensity on the basis of the X-ray data, were then located and an improved
</p>
<p>orientation matrix established.
</p>
<p>The 4 � 64� area detector of D19 enabled the three-dimensional peak shapes of the Bragg
reflections to be monitored, and neither splitting nor streaking was observed. The crystal was cooled
</p>
<p>to 15 K while a strong reflection was continuously monitored and a new orientation matrix estab-
</p>
<p>lished. The peak shapes of a number of reflections were checked. Collection of the � h�k �l data
commenced in shells of 2y &frac14; 92�, using o-scans in equatorial geometry. The area detector allowed
additional reflections to be measured and so checks were made of mirror-related reflections and
</p>
<p>repeated measurements of the same or Friedel-related (q.v.) reflections, as well as the usual three
</p>
<p>reference reflections repeated regularly during data collection.The wavelength was then changed to
</p>
<p>1.312 Å in order to measure higher-angle data, a new orientation matrix calculated and data collection
</p>
<p>11.7 X-Ray and Neutron Structure. . . 563</p>
<p/>
</div>
<div class="page"><p/>
<p>continued using normal beam geometry. Because of a slow helium leak, a significant reduction in
</p>
<p>detector efficiency had occurred. Nevertheless, the efficiency was effectively constant on the time
</p>
<p>scale of the experiment, but this situation, and the rather small crystal volume of 0.8 mm3, contributed
</p>
<p>to the fact that half of the intensity data were weak, although significantly above background. The
</p>
<p>program RETREAT was used to integrate the Bragg peaks in three dimensions. No absorption
</p>
<p>correction was applied, although the crystal shape would have caused certain reflections to have
</p>
<p>suffered significant absorption effects. The data were merged and sorted to provide 1931 unique
</p>
<p>reflections.
</p>
<p>11.7.2 Structure Analysis and Refinement
</p>
<p>X-ray Structure
Using the monoclinic X-ray data set, the direct methods routine in SHELX-S produced an
</p>
<p>outstanding solution to the structure in space group P21. The corresponding E map revealed the
</p>
<p>positions of all non-hydrogen atoms in both molecules (here designated A and B). Difference
</p>
<p>electron density maps indicated the presence of bound water in each molecule. The structure was
</p>
<p>completed and refined by full-matrix least-squares on jFj2, with anisotropic displacement para-
meters for non-hydrogen atoms, and isotropic parameters for the hydrogen atoms, which were fixed
</p>
<p>geometrically in riding mode. Methyl group hydrogen atoms were located using the circular Fourier
</p>
<p>facility in SHELXL-93; only one hydrogen atom was found on each water molecule in the
</p>
<p>difference electron density maps.
</p>
<p>The structure refined to convergence with SHELX-97 with a mean s/err of 0.020 (max. 0.071); soft
</p>
<p>constraints (SADI) were applied in order to equilibrate bond distances between molecules A and B.
</p>
<p>The final measures of agreement for a total of 9092 reflections with I &gt; 2s(I) and 605 parameters are
</p>
<p>listed in Table 11.2 together with those for the neutron analysis which follows.
</p>
<p>Neutron Structure
At the start of the refinement, the coordinates from the X-ray refinement were used together with the
</p>
<p>Fo data from neutron diffraction. Unexpectedly there was no agreement, although the unit-cell
</p>
<p>parameters from the two sets of measurements were in close agreement, and any difference could
</p>
<p>be ascribed to the temperature difference between the two sets of measurement conditions. This
</p>
<p>situation, which gave considerable cause for concern, was traced eventually to the idiosyncrasies in
</p>
<p>the shape of the unit cell. Thus, when the transformation from the original X-ray, apparently
</p>
<p>rectangular, C-centered cell to the correct, primitive monoclinic cell was made, two alternative but
</p>
<p>extremely similar unit cells could be chosen as shown in Fig. 11.8. It transpired that, by chance, one
</p>
<p>unit cell had been chosen for the X-ray data collection and the other for the neutron data. The two
</p>
<p>intensity data sets were therefore not matched. In order to rectify this situation, the indices and cell
</p>
<p>parameters of the neutron data were suitably transformed and the analysis was then carried out as
</p>
<p>described below.
</p>
<p>The b-angle now established differs from that listed in Table 11.2 because the g-angle was
</p>
<p>inadvertently and incorrectly assumed to be exactly 90� instead of the true value of 90.10�.
The atomic coordinates from the X-ray refinement, which were now compatible with the trans-
</p>
<p>formed unit cell and hkl neutron data (at R &frac14; 0.10), were used as the starting set for the neutron
structure. Isotropic refinement led rapidly to an R-factor of 0.06. Although the X-ray structure had
</p>
<p>revealed most of the hydrogen atoms, one hydrogen atom from each water molecule had been
</p>
<p>undetected. A difference neutron synthesis revealed the missing atoms. Refinement was undertaken by
</p>
<p>564 11 Neutron Diffraction from Single Crystals</p>
<p/>
</div>
<div class="page"><p/>
<p>full-matrix least-squares on jFj2, with isotropic displacement parameters for all atoms. The structure was
refined to convergence using SHELX-97. The final measures of agreement are listed in Table 11.2.
</p>
<p>The least-squares refinement converged with a mean s/err of 0.006 (max. �0.12). Neutron scattering
lengths were taken from Volume III of the International Tables for X-ray Crystallography (1952).
</p>
<p>11.7.3 Discussion of the Structure
</p>
<p>The unit cell dimensions for the &ldquo;orthorhombic&rdquo; unit cell, suggested by the version of the software for
</p>
<p>the CAD4 then available, were a &frac14; 12.856(1) Å, b &frac14; 24.396(4) Å, c &frac14; 14.564(2) Å, a &frac14; 89.98
(1)�, b &frac14; 89.987(9)�, g &frac14; 90.10(1)�. The potential for designing automated diffractometer systems
without adequate symmetry checks and the dangers of misuse with some automated diffractometer
</p>
<p>systems became clear. We emphasize strongly the importance of the role of photographic methods,
</p>
<p>which are still in use in many laboratories and teaching schools, and/or an early Laue photograph to
</p>
<p>check the determination of the Laue group and crystal system. We believe that many important
</p>
<p>Fig. 11.8 Relationship between alternative unit-cell transformations. (a) Crystal unit cells. (b) Reciprocal lattice. Note
that I is the ab face corresponding to the original C-centred X-ray unit cell. The angle g of this unit cell is not exactly
</p>
<p>90 but &frac14;90.1(1)� and consequently the derived primitive true monoclinic cells II (the transformed X-ray cell) and
III (the original neutron cell) have slightly different b angles of 117.93(2) and &frac14;117.12(2)�, respectively. Only the
values of unit-cell parameters c and b are changed by this modification, and the indices transform as h0 &frac14; h; k0 &frac14;
�k; l0 &frac14; �&eth;l&thorn; h&THORN;
</p>
<p>11.7 X-Ray and Neutron Structure. . . 565</p>
<p/>
</div>
<div class="page"><p/>
<p>structures may be passed over or discarded through a failure to use, or simply a lack of awareness, of
</p>
<p>these procedures. The other crystallographic situation illustrated by this structure is the possibility of
</p>
<p>different unit-cell choices appearing identical, with the consequential difficulties in the comparison
</p>
<p>of structural data.
</p>
<p>11.7.4 Hydrogen Bonding
</p>
<p>The hydrogen bonding was elucidated from the neutron diffraction results. Molecules A and B exhibit
</p>
<p>similar hydrogen bonding patterns, there being strong interactions between each molecule and its
</p>
<p>solvated water, Figs. 11.9 and 11.10. There are no hydrogen bonds between the molecules A and B, or
</p>
<p>otherwise.
</p>
<p>Each molecule includes a hydrogen-bonded water molecule which stabilizes the macrocyclic ring
</p>
<p>structure. Both molecules A and B exhibit very close pseudo-symmetry across a plane perpendicular to
</p>
<p>the molecular plane and through atoms C(9) and O(18); in addition they show predominantly planar
</p>
<p>structures. The crystal packing includes hydrogen bonding patterns for molecules A and B, with three
</p>
<p>strong interactions between eachmolecule and its solvatedwater. TheX-ray analysis failed to reveal one
</p>
<p>hydrogen atom per water molecule, each being subsequently included after location and refinement in
</p>
<p>the neutron analysis. The initial automatic assignment of a very convincing &ldquo;orthorhombic&rdquo; X-ray unit
</p>
<p>cell and the subsequent monoclinic transformations in the two analyses reaffirms the need for caution in
</p>
<p>applying procedures automatically.
</p>
<p>Fig. 11.9 Molecule A from the neutron diffraction analysis, showing the water molecule and the hydrogen bonding
(built by Accelerys)
</p>
<p>566 11 Neutron Diffraction from Single Crystals</p>
<p/>
</div>
<div class="page"><p/>
<p>11.8 The Pointless Program in CCP4
</p>
<p>We conclude this section on a slightly more optimistic note. It has been drawn to our attention that
</p>
<p>since the above research was undertaken a new CCP4 program known as Pointless [22] has become
</p>
<p>available. Using the observed Fo(hkl) data, this program determines all the possible Laue groups
</p>
<p>consistent with the unit cell, which is based on cell dimension restraints. It does this by matching
</p>
<p>potential symmetry equivalent reflections. For chiral systems, the Laue group uniquely implies the
</p>
<p>point group. It then checks sets of reflections which may be systematically absent to suggest a
</p>
<p>possible space group. There is no check for unit-cell centering, that is, no check for whole classes of
</p>
<p>reflections having zero intensity, although there is a check for potentially centered reflections missing
</p>
<p>from the list. Strictly speaking, the program determines the &ldquo;Patterson group&rdquo; rather than the Laue
</p>
<p>group, since the Laue group is a point group, but not combined with any unit-cell centring type (P, C,
</p>
<p>I, F, H, or R). Unit-cell centring is included in the reported Laue group, and re-indexing from the
</p>
<p>original setting may change the unit-cell type.
</p>
<p>11.9 Determination of the Positions of the Deuterium Atoms of the Bound
Water Molecules in the Lectin Protein Concanavalin A by Neutron Laue
Crystallography [23]
</p>
<p>11.9.1 Introduction
</p>
<p>Concanavalin A is a saccharide binding protein isolated from the Jack Bean and is the most
</p>
<p>thoroughly studied member of a class of proteins known as plant hemagglutinins or lectins [24]. Its
</p>
<p>biological role is unknown, but it is thought to mediate cell&ndash;cell interactions by binding to polysac-
</p>
<p>charide on the cell surface. It may also have an antifungal plant-defense role. The molecule has a
</p>
<p>Fig. 11.10 The structure of molecule B, built from molecule A using Accelerys and drawn with RASMOL
</p>
<p>11.9 Determination of the Positions of the Deuterium Atoms. . . 567</p>
<p/>
</div>
<div class="page"><p/>
<p>molecular weight of 25 kDa, contains 237 amino acids, and binds a calcium and a transition metal ion,
</p>
<p>for example Mn, as in the present structure.
</p>
<p>Structural studies by neutron diffraction permit elucidation of hydrogen/deuterium exchange,
</p>
<p>which can yield vital information for the understanding of catalytic processes at the molecular
</p>
<p>level [25]. This approach exploits the difference in neutron scattering lengths between hydrogen
</p>
<p>and deuterium, which are equal under X-ray diffraction. Hydrogen atoms constitute a large proportion
</p>
<p>of any biological material and to understand the structure or the catalytic mechanism of such
</p>
<p>materials it is desirable to locate those atoms. Exchanging hydrogen for deuterium in a neutron
</p>
<p>diffraction experiment has three advantages:
</p>
<p>1. A possible reduction of the incoherent scattering by hydrogen.
</p>
<p>2. Deuterium has a neutron scattering length of 6.671, which is similar in sign and magnitude to those
</p>
<p>of the other atoms of the protein, whereas hydrogen has a scattering length of �3.739; this allows
contrast-variation studies.
</p>
<p>3. Deuteration also improves the determination of water deuterium atom positions, as the scattering
</p>
<p>lengths of deuterium and oxygen are both positive in sign and the shape of the D2O nuclear density
</p>
<p>indicates the D2O orientation.
</p>
<p>A high level of D2O exchange is important in a detailed study of boundwater by neutron diffraction.
</p>
<p>In addition to the bulk water and water of hydration, which comprise approximately 50% of most
</p>
<p>protein crystals, the exchangeable atoms are those attached to nitrogen or oxygen atoms, namely:
</p>
<p>1. Main chain N&ndash;-H bonds (except in prolines).
</p>
<p>2. The side chains of Thr, Tyr, Asn, Ser, His, Lys, Arg, Trp, and Gln with N&ndash;H or O&ndash;H bonds.
</p>
<p>3. The side chains of Asp and Glu, if originally protonated.
</p>
<p>The unexchangeable hydrogen atoms in the structure are those attached to carbon atoms, as in C&ndash;H
</p>
<p>bonds, for example. In concanavalin A, there are 399 exchangeable hydrogen atoms and 1356 that are
</p>
<p>non-exchangeable.
</p>
<p>11.9.2 Deuteration of the Concanavalin A Crystals
</p>
<p>Large crystals of concanavalin A were selected for the neutron study. They were transferred to a 1 ml
</p>
<p>portion of crystallization buffer in a tightly sealed test tube. The D2O solution was changed four times
</p>
<p>during the four-month soaking period. The crystal form studied here has the space group I222. The
</p>
<p>chosen crystal, of dimensions 3, 2.5, 2 mm, was mounted in a quartz capillary and used for neutron
</p>
<p>diffraction data collection on the LADI diffractometer at ILL, as discussed above.
</p>
<p>11.9.3 Data Collection and Analysis
</p>
<p>In the data collection, the range of neutron energies corresponding to a wavelength range of
</p>
<p>2.49&ndash;3.52 Å was obtained using a system of silicon crystal mirrors each with 748 alternating
</p>
<p>74&ndash;90 Å thick titanium and nickel layers. The shift towards shorter wavelength helps to enhance
</p>
<p>the resolution of the data. A total of 20 neutron Laue images were collected from �60 to +54� in
intervals of 6�. The exposure time per image was 23.5 h; 5 min were needed to scan and download the
data from the image plate and one minute to erase it before re-use.
</p>
<p>The neutron Laue data were processed using the Daresbury Laboratory software developed for
</p>
<p>synchrotron Laue data processing, but modified for neutron Laue data with a cylindrical detector
</p>
<p>geometry of diameter 318.3 mm. The orientation of the crystal was determined by automatic indexing
</p>
<p>568 11 Neutron Diffraction from Single Crystals</p>
<p/>
</div>
<div class="page"><p/>
<p>of the spots using the LAUEGENprogram, aswas the prediction of the spots of each image. The unit cell
</p>
<p>was refined by LAUEGEN and the unit-cell parameters determined were a &frac14; 88.7 Å, b &frac14; 86.5 Å, and
c &frac14; 62.5 Å. The resolution limit was 2.4 Å. The INTLAUE program was used for integration of each
Laue reflection, and the LAUENORM program was used to derive the wavelength-normalization curve
</p>
<p>using the intensities of symmetry equivalent reflections measured at different wavelengths. This
</p>
<p>essentially puts all of the reflections on the same wavelength basis. A unique set of 8605 reflections
</p>
<p>from these measurements was produced giving a final working neutron data at 2.4 Å resolution.
</p>
<p>The neutron Laue geometry employs a much broader band of neutron wavelengths than a
</p>
<p>monochromatic beam and, with a very large image-plate detector, allows data collection in a
</p>
<p>reasonable time, days instead of months, for small crystals and structures with large unit cells.
</p>
<p>11.9.4 X-Ray Model Refinement
</p>
<p>In order to carry out a neutron diffraction refinement of the concanavalin A structure, a good quality
</p>
<p>starting model at an appropriate resolution was first needed, and to this end an X-ray data set was
</p>
<p>measured using a crystal of dimensions 1.6, 0.95, 0.5 mm deuterated under the same conditions as the
</p>
<p>crystal used for the neutron diffractionmeasurements. TheX-ray data set was collected using anR-AXIS
</p>
<p>II area detector with Cu Ka (rotating anode) radiation. The data were processed in DENZO/SCALE-
</p>
<p>PACK. The refined unit-cell parameters were a &frac14; 89.11 Å, b &frac14; 87.58 Å, and c &frac14; 63.26 Å. There were
24110 unique reflections to a resolution of 1.7 Å. The X-ray structure was generated with respect to this
</p>
<p>data set using as a starting model the coordinates from the PDB file [26] and refined using X-PLOR to
</p>
<p>1.8 Å resolution. In this X-ray model, hydrogen (deuterium) atoms were generated for the bound water
</p>
<p>oxygen atoms using X-PLOR.
</p>
<p>11.9.5 Neutron Structure Refinement
</p>
<p>The refinedX-ray structure of the deuterated crystal at a resolution of 1.8 Å, determined as described in
</p>
<p>the previous section, was used as the starting model for the neutron refinement using X-PLOR version
</p>
<p>3.851. The structure, including the calculated hydrogen atoms, comprised 3566 protein atoms and 148
</p>
<p>water molecules. The neutron scattering lengths used throughout all the subsequent refinements were
</p>
<p>those of the Atomic Energy of Canada (1992) as shown in Table 11.1 and 11.3.
</p>
<p>The unit cell used was determined during the course of the Cu Ka X-ray data reduction. Using
</p>
<p>LSQKAB [27], the two models were superimposed and the average displacements for the 444 bound
</p>
<p>water molecules calculated to be 0.04 Å. For water molecule six, featured in Fig. 11.11, the oxygen
</p>
<p>atom shift was 0.03 Å. The interpretation of hydrogen/deuterium exchange in the peptide chain was
</p>
<p>also needed and, in order to undertake this, the output model of the final cycle of X-PLOR neutron
</p>
<p>Table 11.3 Bound neutron coherent scattering
lengths
</p>
<p>Atom type Scattering length (fm)
</p>
<p>H �3.74
D 6.67
</p>
<p>C 6.65
</p>
<p>N 9.36
</p>
<p>Ca 4.70
</p>
<p>Mn �3.73
</p>
<p>11.9 Determination of the Positions of the Deuterium Atoms. . . 569</p>
<p/>
</div>
<div class="page"><p/>
<p>refinement was input to SHELXL to determine the occupancy refinement. To determine the extent of
</p>
<p>hydrogen/deuterium exchange of the N&ndash;DH amides on the protein (228 atoms), the occupancy of each
</p>
<p>exchangeable hydrogen atom was tied to the occupancy of a deuterium atom so that the sum of their
</p>
<p>occupancies was constrained to be equal to 1.0. The occupancies of hydrogen and deuterium varied
</p>
<p>from 0 to 1.0 for most of the peptides.
</p>
<p>11.9.6 The Bound Water Structure
</p>
<p>There are 148 bound water molecules to be investigated. Many water molecules appear as D2O with
</p>
<p>elongated positive density. The water deuterium atoms are thereby important contributors to
</p>
<p>the overall scattering. There are many examples where the initial hydrogen atom positions assigned
</p>
<p>by X-PLOR to the X-ray model were not found in the nuclear density map, but after refinement
</p>
<p>against the neutron data the fit to the density was greatly improved and taken to be correct. These are
</p>
<p>water molecules that are hydrogen-bond donors or acceptors either with the protein or with other
</p>
<p>water molecules, for example, in water clusters. Figure 11.11 is a stereo illustration of a well-ordered
</p>
<p>D2O molecule where the oxygen atom and its two deuterium atoms enhance one another so to give a
</p>
<p>positive neutron density with a shape evident for the nuclear density. Two water molecules, numbered
</p>
<p>22 and 27 in the PDB file 1c57, which may be downloaded from the Protein Data Bank, are
</p>
<p>coordinated to the Mn atom in the transition metal-binding site, and another two water molecules,
</p>
<p>numbers 29 and 30, are coordinated to the Ca atom in the nearby calcium-binding site. They are
</p>
<p>illustrated in Figs. 11.12, 11.13, and 11.14. Of the 148 defined bound waters, 93 are in the primary
</p>
<p>hydration shell and 55 are in the secondary hydration shell.
</p>
<p>11.9.7 The Metal Sites
</p>
<p>The transition metal manganese is coordinated to Glu8, Asp10, Asp19, His24, and two D2O water
</p>
<p>molecules. The neutron scattering length for manganese is �0.373, essentially the same as the
scattering length of hydrogen, and is observable in the negative density. The distances of Mn and
</p>
<p>Ca from their ligands as determined in this study by the X-ray refinement are shown in Tables 11.4
</p>
<p>and 11.5, respectively.
</p>
<p>Fig. 11.11 Stereo diagram showing the orientation of a well-ordered molecule of D2O, number 6 in the PDB, in the
neutron density
</p>
<p>570 11 Neutron Diffraction from Single Crystals</p>
<p/>
</div>
<div class="page"><p/>
<p>11.9.8 The Saccharide Binding Site
</p>
<p>In the saccharide-free crystal form studied here, the saccharide binding site is occupied by the three water
</p>
<p>molecules 31, 34, and 33, which form hydrogen bonds to the protein as shown in Figs. 11.15 and 11.16.
</p>
<p>Water molecule 31 is hydrogen bonded to both water molecules 33 and 34. It is probable that water
</p>
<p>molecule 33 is hydrogen bonded to water molecule 34, and thus the three water molecules are linked
</p>
<p>to form a triangle. These water molecules fill the binding site snugly and water molecules 33 and 34
</p>
<p>form hydrogen bonds to the adjacent protein. Water molecule 33 donates deuterium to form a
</p>
<p>Fig. 11.12 Mn coordination sphere, Table 11.4, drawn with RASMOL; see also Problem 11.5
</p>
<p>Fig. 11.13 The Mn coordination sphere
</p>
<p>11.9 Determination of the Positions of the Deuterium Atoms. . . 571</p>
<p/>
</div>
<div class="page"><p/>
<p>hydrogen bond to Asp208 OD1, while its oxygen atom can form a hydrogen bond to the adjacent
</p>
<p>Asn14 HD22. Water molecule 34 is close enough to form hydrogen bonds to Tyr100 HN and
</p>
<p>Asp208 OD2.
</p>
<p>11.9.9 Conclusion
</p>
<p>This neutron diffraction study has provided details of the manganese ligand environment, that is,
</p>
<p>the coordination of the hydrogen and deuterium atoms of the two associated water molecules that
</p>
<p>are necessary for an in-depth understanding of features such as the EPR spectroscopy of Conca-
</p>
<p>navalin A. The extensive direct soaking of the crystal in D2O has yielded the deuterium atom
</p>
<p>Fig. 11.14 The Mn and Ca coordination spheres; note the common link from OD1 (Asp 10)
</p>
<p>Table 11.4 The Mn coordination-sphere
ligands
</p>
<p>Mn&ndash;Ligand Distance (Å)
</p>
<p>Mn-OE1 (Glu8) 2.19
</p>
<p>Mn-OD1 (Asp10) 2.17
</p>
<p>Mn-OD1 (Asp19) 2.24
</p>
<p>Mn-NE2 (His24) 2.28
</p>
<p>Mn-O (water 22) 2.17
</p>
<p>Mn-O (water 27) 2.28
</p>
<p>Table 11.5 The Ca coordination-sphere
ligands
</p>
<p>Ca&ndash;Ligand Distance (Å)
</p>
<p>Ca-OD1 (Asp10) 2.47
</p>
<p>Ca-OD2 (Asp10) 2.48
</p>
<p>Ca-O (Tyr12) 2.34
</p>
<p>Ca-OD1 (Asn14) 2.36
</p>
<p>Ca-OD2 (Asp19) 2.44
</p>
<p>Ca-O (water 29) 2.42
</p>
<p>Ca-O (water 30) 2.38
</p>
<p>572 11 Neutron Diffraction from Single Crystals</p>
<p/>
</div>
<div class="page"><p/>
<p>positions of the bound water molecules. The reduction of the number of hydrogen atoms via this
</p>
<p>D2O soaking method has also enhanced the neutron diffraction resolution.
</p>
<p>Neutron diffraction of the bound D2Omolecules, even at the relatively low resolution of this study,
</p>
<p>is seen to be more effective than ultra-high-resolution X-ray diffraction, assigning many more
</p>
<p>Fig. 11.15 The saccharide binding site, drawn by RASMOL
</p>
<p>Fig. 11.16 Saccharide binding site details, built by Accelerys
</p>
<p>11.9 Determination of the Positions of the Deuterium Atoms. . . 573</p>
<p/>
</div>
<div class="page"><p/>
<p>coordinates of the deuterium atoms in bound D2O; 62 complete D2O molecules were revealed, plus
</p>
<p>20 with only one deuterium atom, as well as oxygen. In comparison, for the bound water in the ultra-
</p>
<p>high-resolution X-ray structure, there are only twelve complete H2O molecules, plus 35 with one
</p>
<p>hydrogen atom visible, as well as oxygen, and 272 with no hydrogen atom visible even at 110 K.
</p>
<p>Thus, the complementarity of the neutron and X-ray approaches is clearly evident.
</p>
<p>In the saccharide binding site of concanavalin A, the neutron study has revealed the deuterium
</p>
<p>atoms of the bound water molecules at that binding site. As a sugar ligand approaches the receptor
</p>
<p>binding site on the protein, the mutual orientation of these water molecules together is now known.
</p>
<p>This is a new level of detail for molecular recognition and molecular modeling studies in the future.
</p>
<p>Joint X-Ray and Neutron Refinement
It may be noted that it is possible to combine X-ray diffraction data with neutron diffraction data and
</p>
<p>to carry out refinement using both sets of data simultaneously. In this way the data-parameter ratio
</p>
<p>can be substantially increased, thus effecting significant improvements in the refined model. This has
</p>
<p>been undertaken most extensively on protein structures [28]. The best results are obtained if the same
</p>
<p>crystal can be used in the measurement of both sets of data, or if this is not possible to use two crystals
</p>
<p>grown in the same crystallization experiment. This ensures that the two crystals are isomorphous.
</p>
<p>11.10 The Neutron Structure of the Formyl Peptide Receptor Antagonist
Cyclosporin H (CsH) Unambiguously Determines the Solvent and
Hydrogen Bonding Structure for Crystal Form II [16]
</p>
<p>11.10.1 Introduction
</p>
<p>In the final example of these three recent neutron diffraction studies, we describe the analysis carried out
</p>
<p>on Cyclosporin H in order to resolve some minor problems that were encountered in an earlier X-ray
</p>
<p>study: we do not report that study [29] here, but use the X-ray coordinates from it in this neutron structure
</p>
<p>determination. Single-crystal neutron diffraction data have been collected at 20 K to a resolution of
</p>
<p>1.05 Å on a crystal of the inverse formyl peptide receptor agonist Cyclosporin H (CsH-II) on the
</p>
<p>VIVALDI Laue diffractometer at ILL, Grenoble. The solvent structure and hydrogen bonding network
</p>
<p>of CsH-II have been unambiguously determined by single-crystal neutron diffraction; the agreement
</p>
<p>factor on jFj2 is 13.5% for all 2726 reflections. All hydrogen atom positions, including methyl group
orientations, have been determined by crystallographic refinement. The neutron diffraction structure of
</p>
<p>cyclosporin H provides unique and complementary insights into methyl orientation, hydrogen bonding,
</p>
<p>and solvent interactions that are not available from X-ray analysis alone.
</p>
<p>The hydrogen atoms in a biological material play an essential part in its structure and function. For a
</p>
<p>full understanding of hydrogen bonding and solvent structure, it is essential to know the accurate
</p>
<p>locations of the hydrogen atoms in various situations: terminal &ndash;CH3, &ndash;OH groups, and solvated water
</p>
<p>molecules. Neutron diffraction enhances the possibility of determining their locations reliably. Cyclos-
</p>
<p>porins are cyclic undecapeptides of fungal origin, the best known of which, cyclosporin A (CsA), is a
</p>
<p>lead clinical immunosuppressant; cyclosporin H is an inverse formyl peptide receptor agonist, differing
</p>
<p>from CsA only in the chiral inversion of MeVal-11 from L to D configuration. The peptide sequence of
</p>
<p>cyclosporin H is (L)Bmt-(L)Abu-Sar-(L)MeLeu-(L)Val-(L)MeLeu-(L)Ala-(D)Ala-(L)MeLeu-(L)
</p>
<p>MeLeu-(D)MeVal. The X-ray structure of CsA crystallized from chloroform shows a very similar
</p>
<p>conformation to that determined by both 1H and 13C NMR spectroscopy in chloroform solution.
</p>
<p>Several other crystal forms of CsA and derivatives, crystallized from a range of solvents,
</p>
<p>including ethanol, ether, and acetone, have also been examined; in most cases the CsA
</p>
<p>574 11 Neutron Diffraction from Single Crystals</p>
<p/>
</div>
<div class="page"><p/>
<p>conformation is very similar to within about 0.2 Å for a-carbon atoms. These studies include
</p>
<p>crystallographic (X-ray and neutron) and NMR spectroscopic studies of CsA and its complex
</p>
<p>with cyclophilin.
</p>
<p>Two detailed X-ray crystal structure determinations of CsH [30], crystallized in the presence or in
</p>
<p>the absence of Mg2+ (forms I and II, respectively), have been undertaken at 0.80 Å resolution and
</p>
<p>have identified structural and surface features important for biological activity which may aid the
</p>
<p>future design of new CsH derivatives. A major and surprising outcome of this X-ray study was the
</p>
<p>observation that a minor local chemical change between CsA and CsH&mdash; the chiral inversion of one
</p>
<p>peptide&mdash;is associated with a major structural transformation from open b-sheet in CsA to a saddle-
</p>
<p>shaped &ldquo;baseball-stitch&rdquo; conformation in CsH, Fig. 11.17a, b.
</p>
<p>CsH forms I and II both adopt the same saddle-shaped secondary structure, and the peptide bonds
</p>
<p>in this highly convoluted loop conformation are all trans. Unlike CsA, with just one ordered water
</p>
<p>molecule in the crystal structure, the structures of both CsH forms are heavily solvated, with eight and
</p>
<p>seven water molecules, respectively. Of the seven solvation water molecules in CsH form II, O1W is
</p>
<p>extremely well ordered, as judged by the refined anisotropic displacement parameters. Consideration
</p>
<p>of its interactions with neighboring atoms rules out the possibility that this could be a coordinated
</p>
<p>Mg2+ cation, which implies that these cations improve crystal ordering without being incorporated
</p>
<p>into the structure itself.
</p>
<p>Here we describe the analysis of CsH form II by neutron diffraction. The positions of all hydrogen
</p>
<p>atoms in the peptide and solvent molecules were determined by crystallographic refinement.
</p>
<p>Fig. 11.17 A single stereochemical inversion at residue 11 induces the conversion of Cyclosporin A (a) to cyclosporin
H (b); the structures adopt dramatically different loop configurations. The peptide nitrogen atoms are labeled. CsA data
was taken from PDB 2WFJ, and the figures drawn with RASMOL
</p>
<p>11.10 The Neutron Structure of the Formyl Peptide Receptor. . . 575</p>
<p/>
</div>
<div class="page"><p/>
<p>11.10.2 Experimental
</p>
<p>Highly purified cyclosporin H samples were supplied by Sandoz&ndash;Novartis, Basle, and crystals of CsH
</p>
<p>form-II were grown over twenty-one days at �20�C from methanol in the presence of magnesium
perchlorate in partially sealed glass vials. Magnesium perchlorate was observed to extend the
</p>
<p>resolution of the X-ray diffraction pattern and to have a stabilizing effect on the structure, especially
</p>
<p>the long MeBmt-1 side chain, which is considerably disordered in the absence of the perchlorate. The
</p>
<p>crystals disintegrate spontaneously on exposure to air or water. For this reason it was not possible to
</p>
<p>deuterate the sample, unlike concanavalin A discussed above.
</p>
<p>Neutron Data Collection and Processing
A crystal of cyclosporin H with approximate dimensions 0.5, 0.5, 0.3 mm was dipped in Fomblin
</p>
<p>oil, wrapped in thin aluminum foil, mounted on a thin V-pin, and rapidly cooled to 20 K in a cryo-
</p>
<p>refrigerator. Data were collected on VIVALDI. Eight Laue diffraction patterns were collected on a
</p>
<p>cylindrical image-plate detector at 10&ndash;20� intervals in a rotation of the crystal perpendicular to the
incident beam. The crystal was held stationary during each 7-h Laue exposure, and then rotated to
</p>
<p>the next Laue position where it was held stationary again. The reflections were indexed using the
</p>
<p>program LAUEGEN and integrated using the program ARGONNE_BOXES, which is based on a
</p>
<p>two-dimensional implementation of the three-dimensional minimum s(I)/I algorithm [31]. Correc-
</p>
<p>tion for absorption was unnecessary owing to the small nearly isotropic sample volume. The
</p>
<p>integrated reflections were wavelength normalized and scaled using the program LSCALE. A
</p>
<p>total of 9222 reflections was recorded of which 2849 were independent, resulting in 71.8% overall
</p>
<p>completeness for the resolution range 9.7&ndash;1.05 Å. Shell-wise merging and completeness statistics
</p>
<p>were computed with SCALA. Five percent (123) of the reflections were segregated for Rfree
calculations in order to monitor the refinement and help avoid the risk of over-fitting the data.
</p>
<p>Data collection, processing, and refinement statistics are presented in Table 11.6.
</p>
<p>Since only the ratios between unit-cell dimensions could be accurately determined using the white
</p>
<p>beam Laue technique, the cell dimensions obtained by monochromatic X-ray diffraction at 298 K
</p>
<p>were used to index the neutron data. The crystal displayed anisotropic contraction with cooling, and it
</p>
<p>was necessary to refine the axis-length ratios: with the a axis held constant at 17.4 Å, the length of the
</p>
<p>c axis decreased by 3% from 23.2 to 22.4 Å. The unit cell length esd values from monochromatic
</p>
<p>X-ray measurement at room temperature were used for analysis.
</p>
<p>11.10.3 Structure Refinement
</p>
<p>Refinement against jFj2 was performed using the Windows version of SHELXL-97. Initial atomic
positions, including riding hydrogen atoms, were obtained from the results of the earlier X-ray
</p>
<p>structure, to which we have referred already. Owing to the relatively low data/parameter ratio, the
</p>
<p>refinement protocol was based on macromolecular refinement techniques, starting with a rigid-body
</p>
<p>refinement, then successively easing the restraints after each round of refinement and rebuilding,
</p>
<p>computing Rfree on jFj2at each step to monitor over-fitting of the data. In order to compensate for the
change in unit-cell parameters relative to those at room temperature, the AFIX 9 command was used
</p>
<p>in the initial refinement cycle to model a &ldquo;breathing&rdquo; rigid-body fit. This initial refinement cycle with
</p>
<p>only 226 parameters resulted in an agreement factor of R(jFj2) of 28% and an Rfree(jFj2) of 34%.
Inspection of the initial nuclear density map showed well-resolved nuclear density for all atoms in the
</p>
<p>cyclic peptide, including clear nuclear density for a methyl rotor at residue 4, which had been
</p>
<p>modeled in the wrong conformation in the X-ray structure. The density for the hydrogen atoms on
</p>
<p>576 11 Neutron Diffraction from Single Crystals</p>
<p/>
</div>
<div class="page"><p/>
<p>some solvent molecules was less clear, and ambiguous hydrogen atoms were deleted during this
</p>
<p>round of rebuilding. In the second round of least-squares refinement, the cyclic molecule was fitted as
</p>
<p>twelve rigid peptides and seven rigid water molecules.
</p>
<p>Constraints were replaced with soft restraints as refinement progressed; finally, many of the
</p>
<p>restraints were removed entirely. Default SHELXL weights were employed for restraints, except for
</p>
<p>DELU (0.0001) and ISOR (0.005&ndash;0.01). Several common distances and angles were restrained to
</p>
<p>refined FVAR parameters. In the final cycle of refinement, these were: methyl C&ndash;H, amide N&ndash;H, water
</p>
<p>O&ndash;H, carbon C&ndash;H, C&ndash;C, N&ndash;CA and C&frac14;O bond distances; methyl H&ndash;C&ndash;H and water H&ndash;O&ndash;H angles.
Releasing these restraints caused the R1,free index to increase from 0.168 to 0.183 with only modest
</p>
<p>diminishment of R1, so the restraints were retained. Refinement statistics are listed in Table 11.6.
</p>
<p>After completion of the refinement, each water molecule was omitted in turn for the calculation of
</p>
<p>least-squares water-omit maps. Every water hydrogen atom reappeared in its simulated annealing omit
</p>
<p>map with negative density greater in magnitude than 3s in the Fo � jFjc nuclear density map,
</p>
<p>Table 11.6 Crystal and neutron structure refinement data for CsH-II
</p>
<p>Empirical formula C62H123N11O18
</p>
<p>Mr 1309.00
</p>
<p>Temperature (K) 20(2)
</p>
<p>Neutron wavelength (Å) 1.2&ndash;2.6
</p>
<p>Crystal system Orthorhombic
</p>
<p>Space group P212121
</p>
<p>Unit cell dimensions (Å) a &frac14; 17.392(2)
b &frac14; 19.3729(10)
c &frac14; 22.437(3)
</p>
<p>Volume (Å3) 7559.6(14)
</p>
<p>Z 4
</p>
<p>Resolution range (last shell) (Å) 9.69&ndash;1.05(1.11&ndash;1.05)
</p>
<p>Rmerge 0.109(0.112)
</p>
<p>Rmeas (all I+ and I�) 0.124(0.130)
Rpim (all I+ and I�) 0.057(0.064)
Total number of observations 9184(857)
</p>
<p>Total number unique 2849(309)
</p>
<p>Mean((I)/s) 9.7(7.2)
</p>
<p>Completeness 71.8(55.5)
</p>
<p>Multiplicity 3.2(2.8)
</p>
<p>Dc/Mg (m
�3) 0.288
</p>
<p>F(000) 160
</p>
<p>Crystal size (mm) 0.5, 0.5, 0.5
</p>
<p>Index ranges 0 � h � 16; 0 � k � 18; 0 � l � 21
Reflections collected 9222
</p>
<p>Independent reflections 2849
</p>
<p>Reflections used for SHELX 2726
</p>
<p>Rint 0.1140
</p>
<p>Overall completeness (last shell) 71.8%(55.5%)
</p>
<p>Refinement method Full-matrix least-squares on jFj2
Data/restraints/parameters 2726/998/1303
</p>
<p>Goodness-of-fit on jFj2 1.572
Final R indices [I &gt; 2s(I)] R1 &frac14; 0.1222, wR2 &frac14; 0.2906
R indices (all data) R1 &frac14; 0.1382, wR2 &frac14; 0.2988
</p>
<p>11.10 The Neutron Structure of the Formyl Peptide Receptor. . . 577</p>
<p/>
</div>
<div class="page"><p/>
<p>Fig. 11.18a&ndash;g. Figure 11.19 shows a view of the neutron structure. The final maps and ellipsoid plots
</p>
<p>were of high quality, Fig. 11.20.
</p>
<p>Cyclosporin H-II with co-crystallized water has 217 atoms. With full positional and anisotropic
</p>
<p>thermal displacement parameters for all non-hydrogen atoms (the U values for some methyl hydrogen
</p>
<p>Fig. 11.18 (a&ndash;g) Mosiac of the water molecules in CsH in the &ldquo;water-omit-refinement&rdquo; Fo � jFcj nuclear difference
density. Nuclear density is contoured at 3s (blue) and �3s (red)
</p>
<p>Fig. 11.19 View of CsH neutron structure
</p>
<p>578 11 Neutron Diffraction from Single Crystals</p>
<p/>
</div>
<div class="page"><p/>
<p>atoms were constrained to 1.5 times that of their carbon atom), the number of refined parameters in the
</p>
<p>final refinement cycle totaled 1303.With 2849 independent reflections and 217 atoms, the resulting data/
</p>
<p>atom ratio was 13.1 with a data/parameter ratio of 2.2. A total of 998 restraints was used for refinement:
</p>
<p>185 to restrain chemically comparable bonds to common lengths, 73 to restrain angles, and 740 to
</p>
<p>restrain anisotropic thermal displacement ellipsoids. Refined bond lengths and angles generally com-
</p>
<p>pared well with those from the X-ray refinement, except that the C&ndash;H, N&ndash;H, and O&ndash;H bonds refined to
</p>
<p>larger values. Least-squares refinement of all atomic coordinates and anisotropic temperature factors
</p>
<p>resulted in a final agreement factor ofR1(F
2) &frac14; 0.122 for 2191 independent reflections withF &gt; 4s(F).
</p>
<p>Relevant crystallographic data are summarized in Table 11.6. The unconventional nomenclature for
</p>
<p>atom labeling reflects the presence of D-peptides and several non-standard amino acids; atom naming
</p>
<p>follows that established in the X-ray structure analysis of cyclosporin H-II.
</p>
<p>11.10.4 Description of the Neutron Structure and Comparison
with the X-Ray Structure
</p>
<p>The overall structure of CsH-II is a saddle-shaped, cyclic undecapeptide, Fig. 11.17a, b; the peptide
</p>
<p>sequence of CsH is (L)Bmt-(L)Abu-Sar-(L)MeLeu-(L)Val-(L)MeLeu-(L)Ala-(D)Ala-(L)
</p>
<p>MeLeu-(L)MeLeu-(D)MeVal. There are several non-standard amino acids in the polypeptide, includ-
</p>
<p>ing seven residues with methylated amide nitrogen atoms. Position 1 is assigned to the amino acid with
</p>
<p>the longest unbranched side chain, L-MeBMT. Position 2 is L-Abu. The amide nitrogen atoms at
</p>
<p>residues 1, 3, 4, 6, 8, 9, 10, and 11 are methylated. Residues 8 and 11 have the D-configuration at the a-
</p>
<p>C position (chiral inversion relative to the usual L-amino acid configuration). While six of the seven
</p>
<p>water molecules make at least one hydrogen bond with the peptide chain, Fig. 11.18a&ndash;g, water
</p>
<p>molecule seven does not interact directly with cyclosporin H, and may thus be described as occupying
</p>
<p>Fig. 11.20 (a) 2Fo�jFcj nuclear density for residue 4 of CsH-II is shown to display the map quality. Positive density at
the 2s level is shown in teal, at �2s in orange. (b) An ORTEP plot with ellipsoids shown at the 50 % confidence level.
The hydrogen atoms are displayed with the radii proportional to refined U values
</p>
<p>11.10 The Neutron Structure of the Formyl Peptide Receptor. . . 579</p>
<p/>
</div>
<div class="page"><p/>
<p>the second hydration sphere. Not coincidentally, the oxygen atom of water molecule seven exhibits the
</p>
<p>largest thermal motion and weakest nuclear density of all the water oxygen atoms, Fig. 11.18g.
</p>
<p>The room temperature X-ray and cryo-neutron structures were superposed by the program LSQKAB,
</p>
<p>using the a-carbon atoms for alignment, Fig. 11.21; individual atom-by-atom displacements were com-
</p>
<p>puted with Perl Script. The average displacement for the 11 a-carbon atoms is 0.11 Å; for all peptide non-
</p>
<p>hydrogen atoms, the average displacement is 0.15 Å; for all water oxygen atoms, it is 0.16 Å. For all
</p>
<p>peptide hydrogen atoms, the average displacement is 0.27 Å; for water hydrogen atoms, it is 0.26 Å. The
</p>
<p>shifts are not anisotropic, that is, they do not correspond to shifts along the contracted cell axis.As neutrons
</p>
<p>are scattered by nuclei rather than electron clouds, the C&ndash;H, N&ndash;H, and O&ndash;H bond lengths are generally
</p>
<p>0.14, 0.28, and 0.05 Å, respectively, longer than in the X-ray structure, Fig. 11.19, the largest difference
</p>
<p>being 0.56 Å at HW42, water molecule four.
</p>
<p>There are minor differences in hydrogen atom positions at solvent sites 3, 4, 6, and 7.
</p>
<p>The maximum atomic displacements between the X-ray structure and the neutron structure arose for
</p>
<p>the methyl hydrogen atoms of the amide nitrogen atom of residue 4. This methyl rotor had been
</p>
<p>misoriented by 60� in the X-ray structure determination. As would be expected, the hydrogen bonds
found are consistent for both the X-ray and the neutron analyses. That is, all of the hydrogen bonds
</p>
<p>determined in the neutron study were also recognized in the X-ray structure. As in the X-ray structure,
</p>
<p>there is no indication of Mg2+ ions in the neutron structure, and it may be concluded that the presence of
</p>
<p>these ions in the crystallization mediummerely serves to aid the crystallization process. This conclusion
</p>
<p>is supported by the distinct negative nuclear density for hydrogen atoms obtained in water-omit maps for
</p>
<p>every water molecule in the neutron structure analysis.
</p>
<p>11.10.5 Conclusion
</p>
<p>The complete atomic structure of cyclosporin H form II has been determined by single-crystal neutron
</p>
<p>diffraction. The relatively low data/parameter ratio required a mixed refinement strategy, similar to that
</p>
<p>employed in protein crystallography, Sect. 11.8. The unambiguous determination of all hydrogen atom
</p>
<p>positions completes the high-resolution single-crystal structure of cyclosporin H form II. The water
</p>
<p>hydrogen bonding interactions have now been directly determined, confirming that hydrogen bonding
</p>
<p>patterns inferred from X-ray refinement were essentially correct, but with minor differences, especially
</p>
<p>in the positions of hydrogen atoms on the solventmolecules. The refined neutron structure is illustrated in
</p>
<p>Figs. 11.22 and 11.23.
</p>
<p>Fig. 11.21 The refined NH bond lengths in the neutron structure: average 1.135 Å compared with 0.86 Å in the X-ray
structure
</p>
<p>580 11 Neutron Diffraction from Single Crystals</p>
<p/>
</div>
<div class="page"><p/>
<p>Fig. 11.22 Neutron data allowed correction of a CH3 group orientation at residue 4. X-ray structure is shown in teal
and green. Neutron structure in blue (C atoms and 2s nuclear density) and purple (H atoms and &ndash;2s nuclear density).
</p>
<p>C&ndash;H bond lengths refined to larger values in the neutron structure.
</p>
<p>Fig. 11.23 CsH neutron structure showing the seven water molecules and trace of the main chain N1&ndash;N11. All water
hydrogen atoms were determined experimentally. Thermal ellipsoids are plotted at 85% probability. The drawing was
</p>
<p>made with ORTEP-III/RASTER as implemented in the program suite WinGX and generated by ORTEP-3 for
</p>
<p>Windows. The main chain trace was drawn by RASMOL and inserted independently
</p>
<p>11.10 The Neutron Structure of the Formyl Peptide Receptor. . . 581</p>
<p/>
</div>
<div class="page"><p/>
<p>11.11 Problems
</p>
<p>11.1. A neutron beam has a wavelength spread of 1.0&ndash;2.0 Å. The beam is not quite parallel, the
</p>
<p>angular divergence being �0.25� from the ideal path. The beam is monochromatized with a
single crystal of lead (cubic, a &frac14; 4.954 Å) by reflection from the (111) face. (a) At what angle
should the (111) planes be set so as to give a wavelength of 1.25 Å? (b) What would be the
</p>
<p>approximate spread of wavelength in the monochromatized beam?
</p>
<p>11.2. Sodium hydride has the sodium chloride structure type, with a &frac14; 4.88 Å, Na+ at 0, 0, 0, and H�
at 0, 0, 1/2. Using the data below, calculate F(111) and F(220) for NaH and NaD, for both
</p>
<p>X-rays and neutrons; the temperature factor may be neglected for the X-ray case. First,
</p>
<p>formulate a simplified structure factor equation for the calculation.
</p>
<p>fNa&thorn;&eth;111&THORN; 8.1 bNa&thorn;Z 0.35
fH�=D� &eth;111&THORN; 0.38 bH�Z �0.37
fNa&thorn; &eth;220&THORN; 6.7 bD�Z 0.67
fH�=D� &eth;220&THORN; 0.21
</p>
<p>11.3. When using the VIVALDI instrument to collect neutron diffraction data, why might it be
</p>
<p>necessary first to undertake preliminary studies of a crystal using X-ray diffraction?
</p>
<p>11.4. The neutron spallation beam at ORNL is extremely powerful. Why is it recommended that
</p>
<p>users should supply crystals as large as 1 mm3, whereas VIVALDI users can frequently
</p>
<p>collect excellent neutron diffraction data on much smaller crystals?
</p>
<p>11.5. Calculate the wavelength of a neutron beam having an associated temperature T &frac14; 273 K.
11.6. Use RASMOL to reproduce as closely as possible Fig. 11.12 showing the Mn coordination
</p>
<p>sphere in Concanavalin A. Table 8.4 may also be useful.
</p>
<p>Suggested strategy:
</p>
<p>1. Download the program RASMOL.
</p>
<p>2. Go to the PDB website and download PDB file 1scs.
</p>
<p>3. Read the file 1scs using RASMOL.
</p>
<p>4. Maximize the display.
</p>
<p>5. Locate the independent command display for RASMOL which will be at the bottom of your
</p>
<p>screen.
</p>
<p>6. Enter the following commands using this display:
</p>
<p>582 11 Neutron Diffraction from Single Crystals</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_8#Tab8_4">http://dx.doi.org/10.1007/978-1-4614-3954-7_8#Tab8_4</a></div>
</div>
<div class="page"><p/>
<p>References
</p>
<p>1. Myles DAA (2006) Curr Opin Struct Biol 16:630
</p>
<p>2. Helliwell JR et al (1989) J Appl Crystallogr 22:483&ndash;497
</p>
<p>3. Cruickshank DW, Helliwell JR, Moffat K (1987) Acta Crystallogr A43:656
</p>
<p>4. Cruickshank DW, Helliwell JR, Moffat K (1991) Acta Crystallogr A47:352
</p>
<p>5. Ren Z et al (1999) J Synch Rad 6:891
</p>
<p>6. http://www.ncnr.nist.gov/resources/n-lengths/ele ments/h.html
</p>
<p>7. Neutron News, 3:29 (1992)
</p>
<p>8. http://www.ncnr.nist.gov/resources/n-lengths
</p>
<p>9. Sheldrick GM (2008) Acta Crystallogr A64:112
</p>
<p>10. McIntyre G et al (2006) Phys B Condens Matter 385:1055
</p>
<p>11. Wilkinson C et al (2002) Neutron News 13:37
</p>
<p>12. Teixera SC et al (2008) Chem Phys 345:133
</p>
<p>13. http://www.ill.eu/html/about/movies/experiments/vivaldi- laue-diffraction
</p>
<p>14. http://neutrons.ornl.gov/facilities/
</p>
<p>References 583</p>
<p/>
<div class="annotation"><a href="http://www.ncnr.nist.gov/resources/n-lengths/elements/h.html">http://www.ncnr.nist.gov/resources/n-lengths/elements/h.html</a></div>
<div class="annotation"><a href="http://www.ncnr.nist.gov/resources/n-lengths">http://www.ncnr.nist.gov/resources/n-lengths</a></div>
<div class="annotation"><a href="http://www.ill.eu/html/about/movies/experiments/vivaldi-laue-diffraction">http://www.ill.eu/html/about/movies/experiments/vivaldi-laue-diffraction</a></div>
<div class="annotation"><a href="http://neutrons.ornl.gov/facilities/">http://neutrons.ornl.gov/facilities/</a></div>
</div>
<div class="page"><p/>
<p>15. http://neutrons.ornl.gov/instruments/SNS/MaNDi/
</p>
<p>16. http://www.b2bfreezone.com/product-search/deut erated-compounds.htm
</p>
<p>17. http://www.patentstorm.us/patents/5733984/descrip tion.html
</p>
<p>18. http://www.ansto.gov.au/research/bragg_institute/facilities/molecular_deuteration
</p>
<p>19. ILL-EMBL Deuteration Laboratory, Partnership for structural biology, Grenoble, France
</p>
<p>20. Petit-Haertlein I, Blakeley MP, Howard E, Hazemann I, Mitschler A, Haertlein M, Podjarny A (2009) Acta
</p>
<p>Crystallogr F Struct Biol Cryst Commun 1:406
</p>
<p>21. Palmer RA et al (2001) Acta Crystallogr B57:339 and references therein
</p>
<p>22. http://www.ccp4.ac.uk/html/pointless.html
</p>
<p>23. Habash J et al (2000) Acta Crystallogr D56:541 and references therein
</p>
<p>24. Kalb (Gilboa) AJ et al (1999) In: Sigel A, Sigel H (eds) Metal ions in biological systems: manganese and its role in
</p>
<p>biological processes.vol 37. Marcel Dekker, New York, p 279
</p>
<p>25. Wlodawer A (1982) Prog Biophys Mol Biol 40:115
</p>
<p>26. Emmerich C et al (1994) Acta Crystallogr D50:749
</p>
<p>27. Collaborative Computational Project, Number 4 [CCP4] (1994)
</p>
<p>28. Wlodawer A (1982) loc cit
</p>
<p>29. Potter BS (2001) PhD Thesis, University of London
</p>
<p>30. Brian Potter BS, Palmer RA, Withnall R, Jenkins T, Chowdhry BZ (2003) Org Biomol Chem 1:1466 and references
</p>
<p>therein
</p>
<p>31. Wilkinson C et al (1988) J Appl Crystallogr B21:47
</p>
<p>32. Gardberg A, Potter B, Palmer R, Myles D, McDonald G (2011) J Chem Cryst 41:470 and references therein
</p>
<p>Bibliography: Neutron Diffraction
</p>
<p>Bacon GE (1975) Neutron diffraction, 3rd edn. Clarendon Press, Oxford
</p>
<p>Bacon GE (1963) Applications of neutron diffraction in chemistry. Pergamon Press, New York
</p>
<p>Niimura N, Podjarny A (2011) Neutron protein crystallography, I. U. Cr. Monograph (Book 25)
</p>
<p>Sherwood D, Cooper J (2011) Crystals, X-rays and protein crystallography (chapter 16). OUP, Oxford
</p>
<p>Wilson CC (2000) Single crystal neutron diffraction from molecular materials. World Scientific, Singapore. Onlineli-
</p>
<p>brary.wiley.com/doi/10.1107/S01087673010 05980/abstract
</p>
<p>Structure of Proteins
</p>
<p>Branden C, Tooze J (1999) Introduction to protein structure, 2nd edn. Garland Publishing, New York
</p>
<p>Blow DM (2002) Outline of crystallography for biologists. OUP, Oxford
</p>
<p>Palmer RA (2000) X-ray crystallographic studies of protein-ligand interactions in protein-ligand interactions (chapter
</p>
<p>12). In: Chowdhury B, Harding S (eds) Practical approach series. OUP, England
</p>
<p>Proteins
</p>
<p>ChayenNE, Helliwell JR, Snell EH (2010)Macromolecular crystallization and crystal perfection. OxfordUniversity Press,
</p>
<p>Oxford
</p>
<p>584 11 Neutron Diffraction from Single Crystals</p>
<p/>
<div class="annotation"><a href="http://neutrons.ornl.gov/instruments/SNS/MaNDi/">http://neutrons.ornl.gov/instruments/SNS/MaNDi/</a></div>
<div class="annotation"><a href="http://www.b2bfreezone.com/product-search/deuterated-compounds.htm">http://www.b2bfreezone.com/product-search/deuterated-compounds.htm</a></div>
<div class="annotation"><a href="http://www.patentstorm.us/patents/5733984/description.html">http://www.patentstorm.us/patents/5733984/description.html</a></div>
<div class="annotation"><a href="http://www.ansto.gov.au/research/bragg_institute/facilities/molecular_deuteration">http://www.ansto.gov.au/research/bragg_institute/facilities/molecular_deuteration</a></div>
<div class="annotation"><a href="http://www.ccp4.ac.uk/html/pointless.html">http://www.ccp4.ac.uk/html/pointless.html</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1107/S0108767301005980/abstract">http://dx.doi.org/10.1107/S0108767301005980/abstract</a></div>
</div>
<div class="page"><p/>
<p>Powder Diffraction 12
</p>
<p>12.1 Introduction
</p>
<p>The powder method was devised by Hull soon after the discovery of X-ray diffraction, and developed
</p>
<p>in detail by Hull [1] and by Debye and Scherrer [2]. Since that time, X-ray diffraction from powdered
</p>
<p>specimens has been used in divers investigations of materials. The main interest in this book is
</p>
<p>structure determination for which powder methods have, until recent years, been inappropriate,
</p>
<p>mainly because of the problem of overlap of the X-ray reflections which causes three-dimensional
</p>
<p>data information to collapse on to a one-dimensional powder record. The vast improvement in
</p>
<p>instrument technology in recent years has led to powder photographs and diffractograms of sufficient
</p>
<p>precision to be interpretable in terms of the underlying crystal structures, and powder techniques have
</p>
<p>now been developed as a very significant tool in X-ray structure determination. Before launching into
</p>
<p>this topic, however, we summarize here some of the many applications of X-ray powder diffraction
</p>
<p>other than in crystal structure determination.
</p>
<p>12.1.1 Identification
</p>
<p>An X-ray powder diffraction data file was inaugurated by Hanawalt in 1930, and his work has
</p>
<p>developed into the powder index of today. Well over a half a million substances have been indexed
</p>
<p>and recorded, and are represented by the Powder Diffraction File (PDF) of the International Centre
</p>
<p>for Diffraction Data (ICDD) [3], formerly the Joint Committee for Powder Diffraction Studies, which
</p>
<p>maintains a database of powder diffraction patterns, including inter alia the d-spacings and relative
</p>
<p>intensities of observable data.
</p>
<p>12.1.2 Crystallinity: Size and Strain Broadening
</p>
<p>Crystals produce sharp X-ray diffraction peaks, whereas amorphous materials, such as glass, liquids,
</p>
<p>and some polymers, produce a few diffraction maxima and a diffuse background. A decrease in
</p>
<p>crystallite size in a crystal sample causes an increase in the width of diffraction maxima, often
</p>
<p>characterized by the peak full width at half maximum height (FWHM), b, which is related to the mean
</p>
<p>crystallite size l by b &frac14; Kl=&eth;l cos y&THORN;, where the scale factor K is 0.89 for spherical crystallites, 0.94
</p>
<p>M. Ladd and R. Palmer, Structure Determination by X-ray Crystallography:
</p>
<p>Analysis by X-rays and Neutrons, DOI 10.1007/978-1-4614-3954-7_12,
# Springer Science+Business Media New York 2013
</p>
<p>585</p>
<p/>
</div>
<div class="page"><p/>
<p>for cubic grains and otherwise generally close to unity. Powder methods can be used to determine the
</p>
<p>degree of crystallinity by comparing the intensity of the background pattern to that of the sharp peaks.
</p>
<p>Most real crystals contain imperfections which produce distortions in the microstructure. Both
</p>
<p>decrease in crystallite size and increase of strain in a crystal will cause broadening of the recorded
</p>
<p>diffraction peaks. The Williamson-Hall analysis [4] allows both size and strain to be determined
</p>
<p>through the expression bhkl &frac14; 2ed�hkl &thorn;1=l, where b and l are as before and e is the microstrain. A plot
of b against 1/l leads to values for both the crystallite size and the microstrain.
</p>
<p>12.1.3 Unit-Cell Parameters
</p>
<p>Powder diffraction data can be indexed and unit-cell parameters determined readily for cubic,
</p>
<p>hexagonal, and tetragonal crystals. Other crystal systems present more difficulty and we address
</p>
<p>this matter in Sect. 12.5, where we shall see that the most important factor is the precision obtainable
</p>
<p>in the measurement of 2y from the diffraction data.
</p>
<p>12.1.4 Expansion Properties
</p>
<p>The unit-cell parameters depend on temperature and pressure. Changes in the d-spacings with
</p>
<p>changes in temperature or pressure allow the thermal expansivity and bulk modulus to be determined.
</p>
<p>As with the unit-cell parameters, measurements of 2y are required. The bulk modulus K is given by
</p>
<p>the expression K &frac14; �V&eth;@p=@V&THORN;T, and accurate unit-cell parameters are required to determine the
volume change under uniform compression.
</p>
<p>12.1.5 Phase Transitions and Alloy Systems
</p>
<p>At certain critical conditions of temperature and pressure, a crystal may change its structure to a
</p>
<p>polymorphic form, and this is revealed by the appearance of new peaks in the diffraction record and
</p>
<p>the disappearance of some exhibited previously. Allied to this property is the study of alloy systems.
</p>
<p>Figure 12.1 shows an early powder pattern of the binary silver-cadmium alloy system [5], and
</p>
<p>Fig. 12.2 is the phase diagram [7] deduced from that study.
</p>
<p>12.2 Crystal Structure Analysis with Powders
</p>
<p>In recent years, great progress has been made in the determination of crystal structures from powder
</p>
<p>data, and these developments in the powder method have added a new and powerful tool for the
</p>
<p>determination of the structures of the many substances that could be obtained only in microcrystalline
</p>
<p>form, of particle size ca. 10�3 mm. This topic forms the subject matter for the rest of this chapter;
</p>
<p>more detailed discussions of powder methods can be found in the literature [8&ndash;11].
</p>
<p>12.2.1 Crystal Structure Determination Scheme
</p>
<p>A scheme for the structure determination of a crystal from powder diffraction data may be set out in
</p>
<p>five stages:
</p>
<p>586 12 Powder Diffraction</p>
<p/>
</div>
<div class="page"><p/>
<p>Fig. 12.1 Early X-ray powder photographs of silver&ndash;cadmium alloys, taken with Fe Ka X-radiation. Silver, the
a-phase, takes up cadmium into solid solution to the extent of ca. 40 % Cd. The unit-cell side a of the cubic a-phase is
</p>
<p>proportional to the concentration cCd of cadmium (Vegard&rsquo;s law [6]). Since the metallic radius of cadmium (1.50 Å) is
</p>
<p>greater than that of silver (1.44 Å), lines in the a-phase move to lower y as cCd increases. Several films show a1a2
splitting at high y (after Westgren and Phragmén, loc. cit. [4])
</p>
<p>12.2 Crystal Structure Analysis with Powders 587</p>
<p/>
</div>
<div class="page"><p/>
<p>1. Collecting and indexing the powder diffraction pattern, and determining the unit cell
</p>
<p>2. Identifying the space group from the diffraction record, if possible
</p>
<p>3. Decomposing the powder pattern into integrated intensity data, I(hkl)
</p>
<p>4. Solving the phase problem
</p>
<p>5. Refining the model structure
</p>
<p>As with single-crystal X-ray analysis, the first three stages lead to the acquisition of a set of relative
</p>
<p>Fo(hkl) data; the indexed data will enable the space group to be determined as far as possible. The phase
</p>
<p>problem may be tackled by traditional Patterson or direct methods, or with direct-space techniques such
</p>
<p>as Monte Carlo or genetic algorithms, which make use of the power of modern computers. The
</p>
<p>refinement of the structure follows the Rietveld technique, developed in the late 1960s, which has been
</p>
<p>the driving force for the extension of the powder method to the more complex materials now studied.
</p>
<p>The experimental data collection procedures have progressed greatly since the initial film based
</p>
<p>camera techniques, leading to refined camera methods and then to diffractometer methods, and with
</p>
<p>neutron and synchrotron sources as well as X-rays.
</p>
<p>12.3 Basis of the Powder Method
</p>
<p>The geometrical basis of X-ray powder photography is illustrated in Fig. 12.3. In the camera method,
</p>
<p>monochromatic, collimated X-rays enter along a diameter of the cylindrical camera, through a small
</p>
<p>hole in the X-ray film, and exit through a diametrically opposed aperture into a beam trap. In this type
</p>
<p>Fig. 12.2 The phase equilibrium diagram for the silver&ndash;cadmium system [5]; silver and cadmium form a continuous
range of solid solutions, up to 42.5% Cd; the range at the high cadmium end is much smaller (reproduced by courtesy of
</p>
<p>John Wiley and Sons)
</p>
<p>588 12 Powder Diffraction</p>
<p/>
</div>
<div class="page"><p/>
<p>of camera, the specimen is a very finely ground powder contained in a thin-walled borosilicate glass
</p>
<p>capillary tube, so providing a cylindrical-shaped specimen. The mounting for the sample tube is
</p>
<p>provided with mechanical slides, so that it can be adjusted to rotate on its axis within its own volume.
</p>
<p>The rotation serves to increase the effective randomness of the orientations of the crystallites in the
</p>
<p>sample, but is not a necessary feature of the method itself. Other experimental techniques may require
</p>
<p>a specimen in the form of a flat plate.
</p>
<p>Typical reflection positions on a powder film are shown at A1 and A2, at a scattering angle of 2y.
</p>
<p>From the geometry of the camera, it is clear that y is obtained from
</p>
<p>arc A1A2 &frac14; 4Ry (12.1)
</p>
<p>where R is the radius of the film. Since the crystallites lie in all possible orientations, the Bragg
</p>
<p>equation for a given reflection hkl is satisfied by any generator of a cone of semi-vertical angle 2yhkl
and axis along the transmitted X-ray beam. A complete cone of diffraction spots intercepts the
</p>
<p>cylindrical film strip in circular arcs, A1 and A2, of uniform intensity. If all possible orientations do not
</p>
<p>exist, as when the crystallites are too large, the arcs are broken to give a &ldquo;spotty&rdquo; powder photograph.
</p>
<p>When the Bragg equation is satisfied for a reflection to occur, we have
</p>
<p>sin2y &frac14; l2=&eth;4d2&THORN; (12.2)
</p>
<p>so that d values are also obtainable. An important limitation of the powder method is immediately
</p>
<p>evident, namely, that we can measure only one geometrical parameter, the Bragg angle y for each
</p>
<p>reflection.
</p>
<p>Fig. 12.3 The basic Debye&ndash;Scherrer arrangement for X-ray diffraction from polycrystalline specimens: C, camera
housing of radius R, with film strip F fitted tightly against its inner surface; X, incident X-ray beam traveling along a
</p>
<p>diameter of the camera; H1 and H2, two small, diametrically opposed holes in the camera and film; S, the powder
</p>
<p>specimen; L, trap for the transmitted X-ray beam; A1 and A2, positions of two arcs of one and the same powder
</p>
<p>reflection; 2y, the angle of scatter (twice the Bragg angle)
</p>
<p>12.3 Basis of the Powder Method 589</p>
<p/>
</div>
<div class="page"><p/>
<p>For a structure analysis, we know that we must obtain the unit-cell dimensions, the space group,
</p>
<p>and the hkl indices for each reflection together with its integrated intensity. Until the development of
</p>
<p>both high speed computing and improvements in the data collection procedures, the task was
</p>
<p>formidable, although some simplification arises for high-symmetry materials.
</p>
<p>12.4 Data Collection
</p>
<p>In powder diffraction work, many fewer resolved reflections are available than with single-crystal
</p>
<p>X-ray crystallography. In addition, the problem of determining the unit-cell dimensions, space group,
</p>
<p>and indices and intensities of reflections is less straightforward.
</p>
<p>Powdered specimens contain numerous very small crystallites arranged in completely random
</p>
<p>orientations, so that there is no explicit information on the location of a reciprocal lattice point other
</p>
<p>than its distance d* from the origin. A powder pattern is a collapsed, tangential projection of the
</p>
<p>weighted reciprocal lattice, in which each diffraction maximum is characterized by its Bragg angle y
</p>
<p>and its intensity.
</p>
<p>Powder lines have a significant breadth and tend to merge or overlap with both increasing y and
</p>
<p>increasing unit-cell size. A sample containing crystallites that are too large may be ground in an
</p>
<p>appropriate apparatus, made from agate or tungsten carbide. Care must be exercised because over-
</p>
<p>grinding could introduce strain with consequent line broadening
</p>
<p>In order that powder lines should be sufficiently well resolved for their positions to be measured to
</p>
<p>better than the desired 0.01� in y, it is necessary that the X-radiation is monochromatic, very finely
focused on to the film, and produces a low intensity of background radiation.
</p>
<p>12.4.1 Guinier-Type Cameras
</p>
<p>One method of obtaining the desired experimental conditions is with a Guinier-type camera, employ-
</p>
<p>ing a curved crystal monochromator. This X-ray monochromator consists essentially of an asymmet-
</p>
<p>rically ground, curved crystal of quartz, silicon, or germanium. The (10�10) planes of a quartz crystal,
</p>
<p>or the (111) planes of silicon or germanium, are set to diffract at a prearranged angle to the crystal
</p>
<p>surface, Fig. 12.4a. The crystal is ground to a cylinder of radius 2R and then bent over the fitting area.
</p>
<p>X-rays from the line focus of the X-ray tube strike the crystal as a divergent bundle. In the crystal, a
</p>
<p>narrow wave band is separated from the polychromatic source by diffraction according to the Bragg
</p>
<p>equation. The diffracted rays leave the crystal as a convergent bundle, at the glancing angle y. The
</p>
<p>most intense spectral line is the Ka1, and the crystal is set to reflect this wavelength, with the
</p>
<p>elimination of other wavelengths from the diffracted beam, albeit with a slight loss in the a1 intensity.
</p>
<p>Figure 12.4b shows the location of the various determinants involved in the setting of the crystal
</p>
<p>monochromator. The distance a between the line focus F of the X-ray tube and the center of the
</p>
<p>crystal, and the distance b between the center of the crystal and the focal line F0 are maintained
constant for different wavelengths by a judicious choice of t, the angle between the (10�10) or (111)
</p>
<p>planes and the crystal surface. This feature allows a change of X-ray tube (and wavelength) without
</p>
<p>major alterations in the experimental arrangement.
</p>
<p>The principle of Guinier operation is illustrated in Fig. 12.4c. The Guinier-type camera can
</p>
<p>function in transmission geometry, as with low&ndash;medium absorbing specimens, or in reflection
</p>
<p>geometry, in the case of strongly absorbing materials. In the photographic technique, the X-ray
</p>
<p>film is mounted on the Seeman&ndash;Bohlin focusing cylinder. Figure 12.5 shows a complete Guinier-type
</p>
<p>590 12 Powder Diffraction</p>
<p/>
</div>
<div class="page"><p/>
<p>Fig. 12.4 (a) A curved crystal monochromator, showing the incident divergent bundle of rays from the X-ray tube at F
reflected at the glancing angle y, and the bundle convergent at the point F0. (b) The curved crystal, showing the
determinants involved in the setting of the crystal; the constancy of a and b with change of wavelength depends on the
</p>
<p>correct choice of t. (c) The principle of the Guinier method, showing the disposition of X-ray tube, monochromator,
powder specimen, and focusing cylinder. The focusing cylinder carries a photographic film, or can be replaced by a
</p>
<p>scintillation counter (diffractometer) or an imaging plate (imaging camera) (courtesy Huber Diffraktionstechnik GmbH,
</p>
<p>83253 Rimsting, Germany)
</p>
<p>12.4 Data Collection 591</p>
<p/>
</div>
<div class="page"><p/>
<p>film-camera assembly for flat, powdered specimens. If the final film is assessed with a vernier
</p>
<p>measuring instrument, the desired accuracy in y can be achieved.
</p>
<p>12.4.2 Image Plate Camera
</p>
<p>In a modern development of the Guinier technique, the X-ray film is replaced by an image (photo-
</p>
<p>stimulated) plate, leading to greatly improved speed of data collection without loss of the accuracy
</p>
<p>that is so important in powder indexing. The X-ray imaging plate is a flexible strip, like an X-ray film,
</p>
<p>Fig. 12.4 (continued)
</p>
<p>Fig. 12.5 Huber&ndash;Guinier Powder Camera assembly (courtesy Huber Diffraktionstechnik GmbH, 83253 Rimsting,
Germany)
</p>
<p>592 12 Powder Diffraction</p>
<p/>
</div>
<div class="page"><p/>
<p>that contains a metastable phosphor, such as a deposited 150 mm layer of BaFBr.Eu(II) which
</p>
<p>contains F-centers (color centers) in an organic binder, as discussed in Sect. 5.7.3.
</p>
<p>The advantages of the imaging plate are parallel data collection over a large area, with high
</p>
<p>dynamic range, good resolution, and digital readout. The image-recorded intensity tends to decay
</p>
<p>with time, so that the plates are best handled in darkened conditions.
</p>
<p>Figure 12.6 is a schematic diagram of the Huber Imaging-Plate Guinier camera. The incident beam
</p>
<p>is monochromatic Ka1 X-radiation, focused on to powder specimen in a flat holder. The camera
</p>
<p>enclosuremay be evacuated, so as to decrease the background scattering. The imaging plate containing
</p>
<p>the powder pattern is scanned by a laser, and software handles the scanned data to produce a profile
</p>
<p>record and data files. Collection times may be reckoned in minutes, particularly where a synchrotron
</p>
<p>source is available. After use, the imaging plate is restored to its original condition by means of the
</p>
<p>halogen erasure lamp. The data files are produced in a format appropriate for Rietveld refinement
</p>
<p>(q.v.). Fuller details of the performance of the camera are available from the manufacturer [12].
</p>
<p>12.4.3 Powder Diffractometers
</p>
<p>The diffractometer is the primary instrument for recording powder diffraction spectra, whether produced
</p>
<p>by X-ray or neutron radiation sources. Many X-ray powder diffractometers use the Bragg-Brentano
</p>
<p>Fig. 12.6 Schematic diagram of the Huber Imaging Powder Camera. The powder pattern, collected on the imaging
plate, is scanned by a laser and then digitized for computer analysis (courtesy Huber Diffraktionstechnik GmbH, 83253
</p>
<p>Rimsting, Germany)
</p>
<p>12.4 Data Collection 593</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_5#Sec30_5">http://dx.doi.org/10.1007/978-1-4614-3954-7_5#Sec30_5</a></div>
</div>
<div class="page"><p/>
<p>parafocusing principle in reflection geometry, Fig. 12.7. The take-off angle of X-rays from the tube
</p>
<p>anode is about 6�, impinging as a divergent beam at the monochromator. In a high-resolution experi-
mental arrangement, the pre-sample crystal monochromator, of the type in Fig. 12.4b, removes the
</p>
<p>unwanted Ka2 radiation, since the path of this radiation from the monochromator deviates from that of
</p>
<p>theKa1. The reflectedX-rays converge at a knife-edge slit and then diverge on to the specimen. Scattered
</p>
<p>X-rays from the flat sample converge at a receiving slit and are received at the proportional counter, gas
</p>
<p>or solid state device, detector assembly. The detector rotates about the sample axis synchronously
</p>
<p>through twice the angular rotation of the sample in the y/2y scan mode. The effective source, receiving
</p>
<p>slit and sample lie on the focusing circle, which has a radius that depends on y. It is important to avoid
</p>
<p>preferred orientation in the flat plate powder sample when preparing it for diffractometer [13].
</p>
<p>The sample is spun about an axis normal to the plate so as to obtain a good average diffraction
</p>
<p>record. It is also desirable to move the capillary tube containing the powder sample such that a different
</p>
<p>part of it is irradiated as the data collection proceeds, so as to counteract the effects of any decomposition
</p>
<p>of the sample. Figure 12.8 illustrates a typical high-resolution powder diffractometer assembly.
</p>
<p>Synchrotron Radiation
The use of synchrotron radiation instead of X-rays leads to better resolution, but in view of the cost of
</p>
<p>SR facilities, it is necessary to carry out a preliminary X-ray examination, as with single crystals, so as
</p>
<p>to obtain the unit-cell data and to index the pattern. Particular advantages of synchrotron radiation are
</p>
<p>the high intensity and excellent vertical collimation, leading to greatly improved resolution compared
</p>
<p>with laboratory X-ray sources. These features decrease the difficulties caused by overlapping reflec-
</p>
<p>tions, and have enabled structures of considerable complexity to be solved, such as that of the
</p>
<p>compound La3Ti5Al15O37, which has 60 atoms in the asymmetric unit [14].
</p>
<p>12.4.4 Diffractometry at a Neutron Source
</p>
<p>We have discussed features of neutron diffraction in detail in Chap. 11, and the technique is of
</p>
<p>importance in structural studies with powder specimens. The majority of neutron research projects are
</p>
<p>Fig. 12.7 Schematic diagram of a powder diffractometer with Bragg-Brentano parafocusing: S, X-ray source focus;
M, curved-crystal monochromator; F, adjustable receiving slit; O, rotation axis; D detector; R, radius of focusing circle
</p>
<p>(after Huber Diffraktionstechnik GmbH, 83253 Rimsting, Germany)
</p>
<p>594 12 Powder Diffraction</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_11">http://dx.doi.org/10.1007/978-1-4614-3954-7_11</a></div>
</div>
<div class="page"><p/>
<p>Fig. 12.8 (a) The Stoe STADI P powder diffractometer system: the radiation is Ka1, in the range available from Fe to
Mo targets; the focusing circle diameter is 130 mm, the 2y range is 0&ndash;140�, and the minimum step size is 0.0005�.
Intensities are measured by a scintillation counter, and the whole operation is handled by built-in structure-solving
</p>
<p>software. (b) A diffractogram of solecite, a calcium-containing zeolite, measured in the Debye&ndash;Scherrer transmission
mode in a capillary of 0.3 mm diameter: calculated and experimental peaks (with probable errors indicated), positions
</p>
<p>of main peaks (vertical lines) and, below, the difference curve are shown (courtesy Stoe &amp; Cie GmbH, Darmstadt,
</p>
<p>Germany)
</p>
<p>12.4 Data Collection 595</p>
<p/>
</div>
<div class="page"><p/>
<p>carried out at the nuclear reactor at the Institut Laue-Langevin in Grenoble. The fission process used
</p>
<p>to produce a neutron flux generates much heat and has to be water-cooled, by D2O. The deuterated
</p>
<p>water also slows the &ldquo;thermal&rdquo; neutrons to the ambient temperature, leading to a neutron source of
</p>
<p>wavelength of 1&ndash;2 Å.
</p>
<p>Owing to the size of the neutron beam, the single crystal for the monochromator must be large.
</p>
<p>Typically, germanium crystals are cut so that the (hhl) planes, often the (220), are in a position to act
</p>
<p>as a monochromatic source. Filters may be incorporated in order to select a particular wavelength.
</p>
<p>Neutrons are detected mainly by absorption in a gas (often helium) proportional counter. In the
</p>
<p>detection process, the (n, g) reaction is
</p>
<p>3
2He&thorn; 10n ! 42He&thorn; g
</p>
<p>and the energy of the g-radiation is proportional to the intensity of the X-ray signal. The quantum
</p>
<p>efficiency Q of the detector is given by
</p>
<p>Q &frac14; 1� exp&eth;spl&THORN;
</p>
<p>where s is the neutron cross section, p is the pressure of the helium gas, and l is the length of the
</p>
<p>chamber of the proportional counter.
</p>
<p>The resolution is high and the peak shape, though complicated, is well understood and can be
</p>
<p>modeled very satisfactorily in a refinement process. Because the neutron cross section, the neutron
</p>
<p>equivalent of the X-ray form factor, is almost constant over the whole range of sin y=l, atomic
</p>
<p>coordinates and temperature factors can be obtained with high precision. The method can be applied
</p>
<p>both at low temperatures and at high pressures. Figure 12.9 is a diagrammatic illustration of a high-
</p>
<p>resolution powder diffractometer at the Institut Laue-Langevin that uses a neutron source and a
</p>
<p>graphite filter. It was designed to achieve the ultimate resolution, Dd=d &frac14; 5� 10�4, limited only by
</p>
<p>Fig. 12.9 Schematic
diagram of the D2B very
</p>
<p>high-resolution, two-axis
</p>
<p>neutron powder
</p>
<p>diffractometer D2B at the
</p>
<p>Institut Laue-Langevin
</p>
<p>(reproduced from the
</p>
<p>&ldquo;yellow book&rdquo; by courtesy
</p>
<p>of the Institut Laue-
</p>
<p>Langevin, Grenoble)
</p>
<p>596 12 Powder Diffraction</p>
<p/>
</div>
<div class="page"><p/>
<p>particle size. Data can be collected on a 500 mg sample in a few hours and on a gram sample in a few
</p>
<p>minutes, with obvious advantages.
</p>
<p>Crystal Structure of Benzene
The crystal structure of benzene at low temperature has been studied by this technique [15].
</p>
<p>Notwithstanding its highly symmetrical molecular shape, benzene crystallizes in the orthorhombic
</p>
<p>system, with space group Pbca: a &frac14; 7.3551(3) Å, b &frac14; 9.3712(4) Å, c &frac14; 6.6994(3) Å, and four
molecules in the unit cell; the figures in parentheses are the estimated standard deviations of the
</p>
<p>unit-cell parameters.
</p>
<p>The molecules lie in special positions of this space group so that the molecular symmetry that is
</p>
<p>indicated crystallographically is only �1. In practice, deuterated benzene was used because the incoherent
</p>
<p>scattering from hydrogen results in a very high background level; deuteration of hydrogen-containing
</p>
<p>materials is standard practice with neutron experiments. Furthermore, the neutron scattering length for
</p>
<p>deuterium is positive at 6.671 fm1 whereas that for hydrogen is �3.739 fm.
The precision of the structure analysis can be judged from the following bond length parameters,
</p>
<p>quoted in Å:
</p>
<p>C1�C2 &frac14; 1:3969&eth;7&THORN;; C1�D1 &frac14; 1:0879&eth;9&THORN;
C2�C3 &frac14; 1:3970&eth;8&THORN;; C1�D1 &frac14; 1:0869&eth;9&THORN;
C3�C1 &frac14; 1:3976&eth;7&THORN;; C1�D1 &frac14; 1:0843&eth;8&THORN;
</p>
<p>The mean values for C&ndash;C and C&ndash;D are 1.3972(5) Å and 1.0864(7) Å, respectively. The differ-
</p>
<p>ences between the individual C&ndash;C and C&ndash;D bond lengths are not significant at the 3s level, so that
</p>
<p>there is no evidence to show that the geometry of benzene is other than truly hexagonal. For further
</p>
<p>discussions on neutron diffractometry, the reader is referred to the literature [16, 17].
</p>
<p>Time-of-Flight Techniques
Time-of-flight (TOF) techniques use a polychromatic beam and a pulsed neutron source. From the de
</p>
<p>Broglie equation, it follows that the neutron wavelength is inversely proportional to its speed. By
</p>
<p>detecting the time of arrival of each neutron of a particular pulse, its wavelength and the corresponding
</p>
<p>d-spacing can be determined:
</p>
<p>l &frac14; h=mnv &frac14; ht=mnl &frac14; 2d sin y
</p>
<p>where l is the flight path length and t is the TOF. By differentiating the above expression, it follows
</p>
<p>that a long flight path and large scattering angle enable good resolution at small d values to be
</p>
<p>achieved. The TOF technique permits good use of the total neutron energy. Peak shapes are more
</p>
<p>complicated and consequently more difficult to model than those from monochromatic sources.
</p>
<p>Time-Resolved Studies: MYTHEN Detector
The MYTHEN single-photon-counting silicon microstrip detector has been developed at the Swiss
</p>
<p>Light Source of the Paul Scherrer Institute, Switzerland [18] for time-resolved powder diffraction
</p>
<p>experiments. In these experiments, a reaction is triggered rapidly and synchrotron X-rays used to probe
</p>
<p>the structural changes as a function of time. This technique has application, for example, in the study of
</p>
<p>1 fm &frac14; femtometer, 10�15 m.
</p>
<p>12.4 Data Collection 597</p>
<p/>
</div>
<div class="page"><p/>
<p>how protein molecules behave under protein action. The procedure needs sub-second time resolution,
</p>
<p>and the Laue stationary crystal method is used.
</p>
<p>The MYTHEN detector [19] allows the acquisition of diffraction patterns over 120� in 2y in
fractions of seconds. The quality of the data obtainable is comparable with that of normal high-
</p>
<p>resolution detectors in terms of FWHM resolution and in profile shape. It has the additional advantage
</p>
<p>of fast and simultaneous acquisition of the full diffraction pattern. MYTHEN is therefore highly
</p>
<p>desirable for time-resolved studies, such as radiation damage in organic compounds [20].
</p>
<p>12.5 Indexing Powder Patterns
</p>
<p>The simplest powder diffraction pattern is obtained for cubic crystals. In the cubic system, (12.2) may
</p>
<p>be written as
</p>
<p>sin2y &frac14; l
2
</p>
<p>4a2
&eth;h2 &thorn; k2 &thorn; l2&THORN; &frac14; l
</p>
<p>2
</p>
<p>4a2
Nc (12.3)
</p>
<p>where Nc is the integer sum &eth;h2 &thorn; k2 &thorn; l2&THORN;. It follows that the values of sin2 y must exhibit integer
ratios with cubic crystals, within the limits of experimental error, and that the integer Nc follows one
</p>
<p>of the three clearly defined possible patterns: 1,2,3,4,. . . for a P unit cell; 2,4,6,8,. . . for I; and
</p>
<p>3,4,8,11,. . . for F, according to the known limiting conditions for X-ray reflections. Hence, the
</p>
<p>indexing of a cubic substance is normally a simple matter. But even with this high symmetry, more
</p>
<p>than one reflection can have the same value of y. For example, pairs such as 300 and 221 (Nc &frac14; 9),
and 411 and 330 (Nc &frac14; 18) are superimposed on the X-ray film. Certain values of Nc, such as 7, 15,
23, 28, 31,. . ., in general, m2(n � 1) values, where m and n are integers, cannot arise in the cubic
system because they cannot be derived as the sum of the squares of three integers. Although the unit
</p>
<p>cell can be identified readily, we see that there is already a problem in assigning the correct relative
</p>
<p>intensities to these superimposed reflections.
</p>
<p>We referred to an apparent P unit cell for potassium chloride in Sect. 2.10.1. It is, perhaps,
</p>
<p>interesting here to note that Fig. 12.10 shows indexed powder photographs of sodium and potassium
</p>
<p>Fig. 12.10 Indexed X-ray powder photographs for KCl and NaCl. Both structures have the space group Fm�3m, but the
apparent absence of powder lines with h, k, and l all odd in KCl causes it to appear as Pm�3m with a spacing
aapparent &frac14; atrue/2 (after Bragg, The Crystalline State, vol 1, [94])
</p>
<p>598 12 Powder Diffraction</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_2#Sec30_2">http://dx.doi.org/10.1007/978-1-4614-3954-7_2#Sec30_2</a></div>
</div>
<div class="page"><p/>
<p>chlorides. Although both potassium chloride and sodium chloride have the same structure type, the
</p>
<p>absences in the pattern for potassium chloride caused it to be considered as primitive cubic at first.
</p>
<p>The reflections for which h, k, and l are all odd integers are accidental absences, that is, reflections too
</p>
<p>weak to be recorded; see Sect. 4.2.3.
</p>
<p>Powder patterns may be indexed by trial and error methods for simple crystal structures of the
</p>
<p>cubic, tetragonal, and hexagonal systems, by use of equations like (12.3), but as the crystal structure
</p>
<p>becomes complex, with lower symmetry and larger unit-cell dimensions, the problem is enhanced:
</p>
<p>the powder lines become very close, merging into and overlapping one another, so that indexing and
</p>
<p>intensity measurement become increasingly difficult.
</p>
<p>12.5.1 General Indexing
</p>
<p>The indexing of powder diffraction patterns is carried out today by computer programs, and success
</p>
<p>in this process depends to some extent on the complexity of the structure, and to a very large extent on
</p>
<p>the accuracy of the primary y-data. An early general procedure for indexing was set out by Ito [21].
</p>
<p>We recall the general equation for d*(hkl), developed in Sect. 2.4:
</p>
<p>d�2&eth;hkl&THORN; &frac14; h2a�2 &thorn; k2b�2 &thorn; l2c�2 &thorn; 2klb�c� cos a� &thorn; 2lhc�a� cos b� &thorn; 2hka�b� cos g� (12.4)
</p>
<p>Generally, this equation is cast in terms of Q values, for convenience, such that Qhkl &frac14; d�2&eth;hkl&THORN;,
QA &frac14; a�2; . . . ;QD &frac14; 2b�c� cos a�; . . . , and so on, so that
</p>
<p>Qhkl &frac14; h2QA &thorn; k2QB &thorn; l2QC &thorn; klQD &thorn; lhQE &thorn; hkQF (12.5)
</p>
<p>It follows that
</p>
<p>Qhkl &frac14; 4
sin2y&eth;hkl&THORN;
</p>
<p>l2
(12.6)
</p>
<p>Magnesium Tungstate
As an example, Table 12.1 lists a set of 40Q values formagnesium tungstate,MgWO4 collected by a film
</p>
<p>technique. Any three noncoplanar reciprocal lattice vectors define a possible unit cell. A unit cell thus
</p>
<p>chosen may not be the smallest, conventional unit cell, but if it is a satisfactory cell it may subsequently
</p>
<p>be transformed as we discuss shortly. We will consider Ito&rsquo;s method by means of this hand-solved
</p>
<p>example.
</p>
<p>Normally, there will be no a priori knowledge of three suitable noncoplanar vectors. Hence, we
</p>
<p>consider the first three lines in the diffraction pattern, and label them initially as Q100, Q010, and Q001,
</p>
<p>so that here we have a* &frac14; 0.1761 Å, b* &frac14; 0.2131 Å, and c* &frac14; 0.2683 Å. We look through the pattern
for multiples of these values, so as to improve the values of the reciprocal parameters by averaging the
</p>
<p>results. For example, Q200 &frac14; 4Q100 &frac14; 0.1240, and line number 7 at Q &frac14; 0.1239 is evidently Q200,
within experimental error. Proceeding in this way as far as possible, we obtain the average values
</p>
<p>a* &frac14; 0.1761 Å, b* &frac14; 0.2131 Å, and c* &frac14; 0.2684 Å.
Next, we consider a zone in which one index is zero, say, hk0. If the angle g* is not 90�, we would
</p>
<p>expect to find two lines symmetrically disposed about the position that would correspond to Q110 if g
*
</p>
<p>was 90�. For example, assuming correctness so far, Q100 + Q010 &frac14; 0.0764. In the list of Q values, we
</p>
<p>12.5 Indexing Powder Patterns 599</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_4#Sec13_4">http://dx.doi.org/10.1007/978-1-4614-3954-7_4#Sec13_4</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_2#Sec7_2">http://dx.doi.org/10.1007/978-1-4614-3954-7_2#Sec7_2</a></div>
</div>
<div class="page"><p/>
<p>have line 4 at 0.0764, and no pair of lines exists symmetrically disposed about this position, as there
</p>
<p>would be if g* was not 90�. Thus, we conclude that line 4 is 110, and that g* &frac14; 90�.
We expect now to find other Qhk0 values among those listed in Table 12.1. For example,
</p>
<p>Q210 &frac14; 4QA &thorn; QB &frac14; 0:1694, so that the line at 0.1695 corresponds to Q210 in the given data. A few
hk0 lines are listed in Table 12.2, showing that we are on the right lines.
</p>
<p>In a similar manner, Q010 &thorn; Q001 &frac14; 0:1174. There is no line of this Q value, but lines 5 and 6 are
nearly symmetrically disposed about this position, so that these lines may be taken as 0�11 and 011.
</p>
<p>Since Q011 � Q0�11 &frac14; 4b�c� cos a�, it follows that a* &frac14; 89.45�. As before, we can now check for the
presence of Q0kl values, as desired. For example, Q021 &frac14; 4Q020 &thorn; Q001 &thorn; 2QD&eth;4b�c� cos a�&THORN; &frac14;
0:2558; similarly, Q02�1 &frac14; 0:2514. These Q values correspond to lines 17 and 16.
</p>
<p>We seek b* in a similar manner. By trial and error in this way, it is found ultimately that lines
</p>
<p>11 and 32 are nearly symmetrically disposed about the calculated value, 0.3190, of Q102. Hence, b
*
</p>
<p>evaluates to 49.11�. Summarizing the results, we have the reciprocal unit cell
</p>
<p>a� &frac14; 0:1761
�
A; b� &frac14; 0:2131
</p>
<p>�
A; c� &frac14; 0:2684
</p>
<p>�
A
</p>
<p>a� &frac14; 89:45�; b� &frac14; 49:11�; g� &frac14; 90:00�
</p>
<p>Table 12.1 Values of Q from the powder diffraction pattern of
magnesium tungstate
</p>
<p>Line number Q Line number Q
</p>
<p>1 0.0310 21 0.3246
</p>
<p>2 0.0454 22 0.3322
</p>
<p>3 0.0720 23 0.3364
</p>
<p>4 0.0764 24 0.3418
</p>
<p>5 0.1164 25 0.3451
</p>
<p>6 0.1186 26 0.3505
</p>
<p>7 0.1239 27 0.3646
</p>
<p>8 0.1649 28 0.3723
</p>
<p>9 0.1695 29 0.3824
</p>
<p>10 0.1816 30 0.4016
</p>
<p>11 0.1957 31 0.4083
</p>
<p>12 0.2077 32 0.4432
</p>
<p>13 0.2123 33 0.4465
</p>
<p>14 0.2386 34 0.4500
</p>
<p>15 0.2436 35 0.4608
</p>
<p>16 0.2517 36 0.4659
</p>
<p>17 0.2563 37 0.4770
</p>
<p>18 0.2793 38 0.4863
</p>
<p>19 0.2884 39 0.4918
</p>
<p>20 0.3055 40 0.4960
</p>
<p>Table 12.2 Some calculated and observed
Qhk0 values for magnesium tungstate
</p>
<p>hk0 Qcalc Qobs
</p>
<p>110 0.0764 0.0764
</p>
<p>210 0.1694 0.1695
</p>
<p>310 0.3244 0.3246
</p>
<p>120 0.2126 0.2123
</p>
<p>220 0.3056 0.3055
</p>
<p>600 12 Powder Diffraction</p>
<p/>
</div>
<div class="page"><p/>
<p>Finally, we obtain the direct space unit cell; Sects. 3.4 and 13.6.3&mdash;program RECIP, and then
</p>
<p>calculate the Q values, Sect. 13.6.9, so as to identify and index all lines in the diffraction record, as
</p>
<p>shown in Table 12.3. The results show good agreement between Qobs and Qcalc for most of the lines,
</p>
<p>although certain fits suggest that some errors exist among the data. We see already in this fairly simple
</p>
<p>example the importance of the accurate experimental measurement of y values.
</p>
<p>12.5.2 Reduced and Conventional Unit Cells
</p>
<p>From the results so far, the crystal system of magnesium tungstate appears to be triclinic. However,
</p>
<p>we must determine whether or not this unit cell is the conventional unit cell for the lattice, that is, it is
</p>
<p>correctly related to the lattice symmetry according to the conventions that we have discussed in
</p>
<p>earlier chapters. The real space unit cell is derived using the equations developed in Sect. 3.4, or with
</p>
<p>the program RECIP by entering the reciprocal unit-cell parameters, with the constant K equal to 1.
</p>
<p>Thus, we obtain
</p>
<p>a &frac14; 7:512
�
A; b &frac14; 4:693
</p>
<p>�
A; c &frac14; 4:929
</p>
<p>�
A
</p>
<p>a &frac14; 90:72�; b &frac14; 130:89�; g &frac14; 89:52�
</p>
<p>or, in the usual order of increasing unit-cell lengths, as
</p>
<p>a &frac14; 4:693
�
A; b &frac14; 4:929
</p>
<p>�
A; c &frac14; 7:512
</p>
<p>�
A
</p>
<p>a &frac14; 130:89�; b &frac14; 89:52�; g &frac14; 90:72�
</p>
<p>Table 12.3 Indices and Q values for magnesium tungstate
</p>
<p>Line hkl Qcalc Qobs Line hkl Qcalc Qobs
</p>
<p>1 100 0.03101 0.0310 21 310 0.32444 0.3249
</p>
<p>2 010 0.04541 0.0454 22 41�2 0.33250 0.3322
</p>
<p>3 001 0.07203 0.0720 23 012 0.33568 0.3364
</p>
<p>4 110 0.07644 0.0764 24 22�2 0.34197 0.3418
</p>
<p>5 01�1 0.11637 0.1164 25 32�1 0.34491 0.3451
</p>
<p>6 011 0.11851 0.1186 26 3�2�1 0.34937 0.3505
</p>
<p>7 200 0.12403 0.1239 27 211 0.36626 0.3646
</p>
<p>8 101 0.16491 0.1649 28 12�2 0.37272 0.3723
</p>
<p>9 2�10 0.16948 0.1695 29 1�2�2 0.38135 0.3824
</p>
<p>10 020 0.18165 0.1816 30 40�3 0.40195 0.4016
</p>
<p>11 30�2 0.19597 0.1957 31 030 0.40871 0.4083
</p>
<p>12 21�2 0.20791 0.2077 32 21�3 0.44324 0.4432
</p>
<p>13 3�1�1 0.21202 0.2123 33 41�3 0.44409 0.4465
</p>
<p>14 11�2 0.23864 0.2386 34 2�1�3 0.44973 0.4500
</p>
<p>15 3�1�2 0.24357 0.2436 35 320 0.46063 0.4608
</p>
<p>16 02�1 0.25177 0.2517 36 02�2 0.46548 0.4659
</p>
<p>17 021 0.25582 0.2563 37 4�2�2 0.47534 0.4770
</p>
<p>18 300 0.27908 0.2793 38 1�12 0.48616 0.4863
</p>
<p>19 002 0.28812 0.2884 39 112 0.49041 0.4918
</p>
<p>20 220 0.30562 0.3055 40 400 0.49614 0.4960
</p>
<p>12.5 Indexing Powder Patterns 601</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec19_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec19_3</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_13#Sec28_13">http://dx.doi.org/10.1007/978-1-4614-3954-7_13#Sec28_13</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_13#Sec34_13">http://dx.doi.org/10.1007/978-1-4614-3954-7_13#Sec34_13</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec19_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec19_3</a></div>
</div>
<div class="page"><p/>
<p>The order makes no difference to the reduced cell; it alters only the transformation matrix between
</p>
<p>the input and the reduced unit cells. A two-stage process now allows the determination of the reduced
</p>
<p>and conventional unit cells.
</p>
<p>The unit-cell parameters derived from the indexing procedure are transformed to a unique, reduced
</p>
<p>unit cell based on the three shortest noncoplanar translations in the lattice&mdash;a Niggli-reduced cell
</p>
<p>[22, 23]; such a unit cell is always primitive. In the second stage, the distribution of twofold axes is
</p>
<p>determined, because it will define the lattice symmetry [24], Sect. 1.4.2 and Table 1.5. We know, for
</p>
<p>example, that the normal to a plane in a lattice that contains p intersecting twofold axes is itself a
</p>
<p>p-fold axis, and that a mirror plane exists normal to a p-fold axis (p &frac14; 2, 4, 6) in a lattice.
The two-stage procedure, which is lengthy manually, has been brought together in the program
</p>
<p>LEPAGE [25]. We describe this program further in Sect. 13.6.10, but for the moment we use it with
</p>
<p>the unit cell derived from indexing the MgWO4 diffraction pattern. Thus, we obtain
</p>
<p>a (Å) b (Å) c (Å) a (�) b (�) g (�)
</p>
<p>Input cell P 4.693 4.929 5.712 130.89 89.52 90.72
</p>
<p>Reduced cell P 4.693 4.930 5.679 90.12 90.01 90.72
</p>
<p>Conventional cell P 4.693 4.929 5.679 90.12 90.01 90.72
</p>
<p>We see that the reduced and conventional unit cells are identical in this example. Further
</p>
<p>interpretation now depends upon the error permitted in the collinearity of the twofold axes in the
</p>
<p>real and reciprocal unit cells, the &ldquo;2-axis criterion (in the program).&rdquo; Ideally, it should be zero.
</p>
<p>However, there will be experimental errors in the data that are conveyed to the parameters; if we
</p>
<p>dismiss these errors, we may fail to recognize the true symmetry. In the present case, if all angles are
</p>
<p>regarded as 90� within experimental error, the lattice is P orthorhombic, with a 2-axis criterion of 1�.
A more realistic situation could be to set a and b at 90�; then we obtain monoclinic P, under the more
stringent 2-axis criterion of 0.5�:
</p>
<p>a &frac14; 4:693 &Aring;; b &frac14; 5:679 &Aring;; c &frac14; 4:929&Aring;
a &frac14; 90&eth;89:88&THORN;�; b &frac14; 90:72�; g &frac14; 90�
</p>
<p>where the unit cell has been rearranged so that the unique angle is b. Notice that this result would have
</p>
<p>been obtained immediately from the first reduction by imposing monoclinic symmetry. Some of the
</p>
<p>difficulty in this indexing analysis for magnesium tungstate arose from the fact that the data was not
</p>
<p>of the highest quality. The modern camera and, particularly, the diffractometer techniques described
</p>
<p>above ensure data of a sufficiently high quality.
</p>
<p>12.5.3 Computer Indexing of the Diffraction Pattern
</p>
<p>We have shown that determining the unit cell by hand can be a slow process, and it is not surprising to
</p>
<p>find that the literature today abounds with computer programs for indexing a powder pattern. Most of
</p>
<p>these programs that serve to determine the unit cell are stand-alone programs, but a few are part of a
</p>
<p>structure-solving and refinement package that leads to a complete structure determination, just like
</p>
<p>the single-crystal methods that we have described already.
</p>
<p>Indexing a powder pattern, by whatever method, is an inductive process: we must deduce the
</p>
<p>indices of the diffraction lines from the experimental Q values. The first 20&ndash;30 lines in a pattern,
</p>
<p>starting from the low-y region, are most important in indexing, because Qcalc involves the square of
</p>
<p>the indices, and so errors in Qobs become more important as y increases. We examine first the ITO
</p>
<p>(named after T. Ito, who devised the general method) program system.
</p>
<p>602 12 Powder Diffraction</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Sec11_1">http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Sec11_1</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Tab5_1">http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Tab5_1</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_13#Sec35_13">http://dx.doi.org/10.1007/978-1-4614-3954-7_13#Sec35_13</a></div>
</div>
<div class="page"><p/>
<p>ITO Program System
The program ITO12 is the 12th version of this program system [26]; a later version is now available,
</p>
<p>Sect. 13.7, but it is only very slightly different from version 12; see also Appendix D. It is a deductive
</p>
<p>program based on the properties of crystal zones, that is, planes of reciprocal lattice points passing
</p>
<p>through the origin, and is often described as zone-indexing. It performs best when given 30&ndash;40
</p>
<p>accurately measured powder lines.
</p>
<p>Any zone is specified by three parameters; for example, the zone hk0 may be formulated by
</p>
<p>Qhk0 &frac14; h2QA &thorn; k2QB &thorn; hkQF (12.7)
</p>
<p>Two Q values are selected and assigned as Q100 and Q010, similar to the procedure in Sect. 12.3.
</p>
<p>Expanding (12.7), we have
</p>
<p>Qhk0 &frac14; h2Q100 &thorn; k2Q010 &thorn; 2hk&eth;Q100Q010&THORN;1=2 cos g� (12.8)
</p>
<p>Let
</p>
<p>2hk&eth;Q100Q010&THORN;1=2 cos g� &frac14; R
</p>
<p>&frac14; Qhk0 � h
2Q100 � k2Q010
hk
</p>
<p>(12.9)
</p>
<p>Values forQhk0 are obtained from the experimental data, and used in the right-hand term of (12.9) so as
</p>
<p>to obtain a list of jRj values.Agreements in jRj, within a permitted error, are then used to find a value for the
angle g*. Zones that are found are checked and reduced, and the three zone parameters refined by least
</p>
<p>squares. Zone quality is determined by a parameter 1/P, where P is the probability that a zone is found by
</p>
<p>chance.Pairs of zoneswith a common roware sought and the angle between themcalculated.Theunit cells
</p>
<p>found are reduced and transformed to standard form, and thefirst 20 lines indexedwhere possible. Thefit is
</p>
<p>assessed by theM20 parameter:
</p>
<p>M20 &frac14;
Q20
</p>
<p>2 �QN20
(12.10)
</p>
<p>where Q20 is the Q value for the 20th indexed line, �Q is the average error between Qobs and Qcalc for
</p>
<p>the first 20 lines, and N20 is the number of lines, observed and calculated, up to Q20. The program is
</p>
<p>optimized for the lower symmetry systems, orthorhombic, monoclinic, and triclinic. High-symmetry
</p>
<p>lattices may be reported in an orthorhombic setting, with a note that a higher symmetry lattice may
</p>
<p>exist.
</p>
<p>Notes on the practical use of the program ITO appear in Sect. 13.7, and problems on it are given at
</p>
<p>the end of this chapter. The data for magnesium tungstate, Sect. 12.5.1, when used with ITO12, led
</p>
<p>to the unit-cell dimensions a &frac14; 4.929 Å, b &frac14; 5.678 Å, c &frac14; 4.693 Å, a &frac14; 90�, b &frac14; 90.77�, and
g &frac14; 90�, in good agreement with the values derived in the worked example of this substance.
</p>
<p>CRYSFIRE Program System
The program system CRYSFIRE provides a detailed set of procedures for indexing powder patterns.
</p>
<p>In common with all methods, a prerequisite is a set of Q values with errors less than ca. 0.01� in y. As
we have seen, this level is achievable experimentally, but the number of lines and their potential
</p>
<p>overlap increases as d*3. Since powder lines have a finite width, clear resolution exists only at the
</p>
<p>lower y values; at higher values of y, they merge into semi-continuous profiles of mainly unresolved
</p>
<p>12.5 Indexing Powder Patterns 603</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_13#Sec36_13">http://dx.doi.org/10.1007/978-1-4614-3954-7_13#Sec36_13</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_13#Sec36_13">http://dx.doi.org/10.1007/978-1-4614-3954-7_13#Sec36_13</a></div>
</div>
<div class="page"><p/>
<p>maxima, each of which may contain 5&ndash;50 peaks. As long as the average discrepancy between the
</p>
<p>observed and calculatedQ values is less than about 5%, the true unit cell can be extracted from among
</p>
<p>other approximate solutions.
</p>
<p>A wholly exhaustive search is prohibitive. We have shown in Sect. 2.2.2 that a lattice can be
</p>
<p>described by any number of alternative unit cells, recognized by having the same reduced unit cell
</p>
<p>and volume. Less satisfactory solutions will usually occur, but they may be recognized through
</p>
<p>goodness-of-fit parameters, such asM20, defined by (12.10); the higher the values of these parameters,
</p>
<p>the better is the fit.
</p>
<p>The program system CRYSFIRE is actually a master automatic-indexing script, which operates
</p>
<p>through another program CRYS acting as a front-end &ldquo;wizard&rdquo; to a collection of eight indexing
</p>
<p>programs written by other workers of lengthy experience in the field of powder indexing, each having
</p>
<p>its own strategy and its own best applicability. The CRYSFIRE system has succeeded in indexing
</p>
<p>numerous powder patterns of all symmetries [27, 28].
</p>
<p>As an example of indexing with CRYSFIRE, a set of 40 Q values for a particular powder sample
</p>
<p>was input to the program. A series of possible solutions were obtained, listed in Table 12.4 in order of
</p>
<p>number of lines I20 listed out of the first 20, together with the figure of meritM20, the unit-cell volume
</p>
<p>and other parameters, and the particular link of the program that produced the solution. The program
</p>
<p>also recorded other possible solutions with I20 less than 19. All solutions are indicated as either
</p>
<p>orthorhombic or monoclinic.
</p>
<p>The first solution listed in Table 12.4, with the highest figure of merit and with I20 equal to 20,
</p>
<p>reproduced all 40 observed Q values, although the experimental error in some lines was greater than
</p>
<p>the best achievable. This result may be regarded as the most probable, and a good starting point for
</p>
<p>further investigations. From a perusal of the indices found by the program, the following limiting
</p>
<p>conditions were deduced:
</p>
<p>hkl : h&thorn; k &frac14; 2n 0kl : &eth;k &frac14; 2n&THORN; h0l : &eth;h &frac14; 2n&THORN;
</p>
<p>hk0 : &eth;h&thorn; k &frac14; 2n&THORN;
</p>
<p>h00 : &eth;h &frac14; 2n&THORN; 0k0 : &eth;k &frac14; 2n&THORN;
</p>
<p>00l : None
</p>
<p>Hence, we may conclude that the best solution is, a &frac14; 6.321 Å, b &frac14; 7.746 Å, and c &frac14; 11.482 Å,
reordered so that c &gt; b &gt; a, with the possible space group being one of Cmmm, Cmm2, and C222.
</p>
<p>However, at this stage we have not considered the problem of overlapping lines, and it may be
</p>
<p>Table 12.4 Some results of an indexing with CRYSFIRE
</p>
<p>I20 M20 V (Å
3) a (Å) b (Å) c (Å) a (�) b (�) g (�) Link
</p>
<p>20 22.3 562.15 7.746 11.482 6.321 90.00 90.00 90.00 ITO12
</p>
<p>20 21.2 280.53 5.000 11.475 4.991 90.00 101.6 90.00 KOHL
</p>
<p>20 12.6 560.97 9.982 11.475 5.000 90.00 101.6 90.00 KOHL
</p>
<p>20 12.1 562.15 7.746 11.481 6.321 90.00 90.00 90.00 ITO12
</p>
<p>20 9.0 561.64 9.995 11.505 4.986 90.00 104.3 90.00 TREOR90
</p>
<p>20 6.5 1966.5 16.737 11.507 10.210 90.00 90.00 90.00 DICVOL91
</p>
<p>20 6.0 623.25 13.048 4.894 11.090 90.00 118.3 90.00 TREOR90
</p>
<p>19 24.5 561.56 7.742 11.473 5.001 90.00 90.00 90.00 KOHL
</p>
<p>19 8.0 561.90 9.973 11.508 5.001 90.00 101.8 90.00 TREOR90
</p>
<p>604 12 Powder Diffraction</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_2#Sec4_2">http://dx.doi.org/10.1007/978-1-4614-3954-7_2#Sec4_2</a></div>
</div>
<div class="page"><p/>
<p>necessary to review the deduction of the space group after the powder pattern has been decomposed
</p>
<p>into integrated intensities.
</p>
<p>We consider briefly some of the remaining solutions in Table 12.4. The second solution indicates a
</p>
<p>smaller unit-cell volume, with an apparently monoclinic unit cell. It can be explained by the
</p>
<p>transformation (from solution 1) a0 &frac14; a=2&thorn; b=2, b0 &frac14; b, c0 &frac14; �a=2&thorn; c=2, to give a P unit cell,
but it would not be chosen as the conventional cell. The penultimate solution in Table 12.4, with
</p>
<p>M20 &frac14; 24.5, is almost as satisfactory, and would probably make a suitable starting point in the
absence of solution 1. Several other solutions also involve an interaxial angle greater than 90�. The
only other apparently different orthorhombic unit cell listed in Table 12.4 has a volume of 1966.5 Å3,
</p>
<p>where the transformation (from cell 1) a0 &frac14; 2a + c leads to the value 16.732 Å, and ja + cj is
approximately 10.0 Å. Thus, we have a set of unit cells within one and the same lattice, and we
</p>
<p>have chosen the most probable and conventional one.
</p>
<p>We note in passing that the CRYSFIRE suite is a very powerful indexing system. However,
</p>
<p>because of the demise of the author in 2005, this suite is no longer under development, but is available
</p>
<p>from CCP14. We discuss indexing systems with built-in structure solving and refinement in
</p>
<p>Sect. 12.8.2.
</p>
<p>12.6 Extracting Integrated Intensities from a Powder Pattern
</p>
<p>At this stage, it is always prudent to check whether or no the pattern now indexed has already been
</p>
<p>recorded and to what extent the structure has been determined, by reference to the ICDD [3] records,
</p>
<p>to which we referred in Sect. 12.1. If we find a &ldquo;new&rdquo; pattern, we proceed to obtain the intensity data.
</p>
<p>The extraction of individual intensities from a powder pattern is complicated by the overlapping of
</p>
<p>reflections in the pattern. Overlaps may be exact, as imposed by symmetry, or accidental, arising from
</p>
<p>near-equivalence of d values for nonequivalent reflections. Pattern decomposition is usually carried
</p>
<p>out either by the Le Bail procedure, which is based on the Rietveld method for determining Fo, or by
</p>
<p>the Pawley procedure, which is a least-squares approach with constraints.
</p>
<p>12.7 The Rietveld Procedure
</p>
<p>It is clear from Fig. 12.3 that the circle of which A1A2 is an arc is one of a set arising in reciprocal
</p>
<p>space from a very large number of randomly oriented crystallites in the sample. The intensity of each
</p>
<p>such diffraction record depends on the crystal structure, overlaps of reflections, and the physical and
</p>
<p>structural features of the diffraction procedure itself that we have discussed above.
</p>
<p>The Rietveld refinement [29, 30] is a complex curve fitting problem. The model parameters,
</p>
<p>atomic coordinates, and thermal parameters and occupancies are subjected to a least-squares
</p>
<p>procedure so as to obtain the best fit between the whole experimental powder diffraction profile
</p>
<p>and the corresponding pattern calculated from the trial structure. In this way, the explicit decomposi-
</p>
<p>tion of overlapping reflections can be avoided, because only the points along the observed and
</p>
<p>calculated profiles, and not the individual reflections, are compared.
</p>
<p>A powder diffractometer record consists of a number of peaks, some overlapped, over a smooth
</p>
<p>background; the intensities follow a Poisson distribution which, with sufficient counts at each
</p>
<p>measurement step, is very closely Gaussian. The powder pattern is, thus, digitized into i steps to
</p>
<p>give an intensity function yi; the digitizing parameter is either the scattering angle 2y for X-rays or a
</p>
<p>velocity function in the case of TOF neutron studies. At each step i, there is an observed profile
</p>
<p>intensity yo,i and a background yb,i. A corresponding calculated profile intensity yc,i is a mathematical
</p>
<p>12.7 The Rietveld Procedure 605</p>
<p/>
</div>
<div class="page"><p/>
<p>expression for the ith step comprising the adjustable parameters of the model: x, y, z coordinates,
</p>
<p>temperature factors, population parameters, and geometrical factors of the experiment. Hence, yc,i
may be written in the form
</p>
<p>yc;i &frac14; yb &thorn;
X
</p>
<p>yh (12.11)
</p>
<p>where yb is a background intensity and the summation term (h &frac14; hkl ) includes the Bragg reflections
around the powder pattern step. When the number of observations, the profile measuring points,
</p>
<p>exceeds the number of parameters, a least-squares minimization M becomes appropriate for fitting a
</p>
<p>function to the observed data:
</p>
<p>M &frac14;
X
</p>
<p>w&eth;yo � yc&THORN;2 (12.12)
</p>
<p>The weight for any term is calculated from the variance in the corresponding measurement of yo,
</p>
<p>and the calculated profile is given by
</p>
<p>yc &frac14; KjFhj2F&eth;Ph&THORN; (12.13)
</p>
<p>where K is a constant that includes the scale and other experimental factors and F(Ph) is a profile
</p>
<p>function. Equation (12.13) represents a multidimensional surface with many minima. The minimum
</p>
<p>is found from
</p>
<p>X
w&eth;yo � yc&THORN;
</p>
<p>@yc
@pi
</p>
<p>&frac14; 0 (12.14)
</p>
<p>pi being the ith adjustable parameter.
</p>
<p>Several different forms have been used for the profile function F: one of the simplest, which is
</p>
<p>appropriate for constant wavelength neutron data, is the Gaussian
</p>
<p>ffiffiffiffiffiffiffiffiffi
C
</p>
<p>pH2j
</p>
<p>s
exp �C &eth;2yi � 2yj&THORN;
</p>
<p>2
</p>
<p>H2j
</p>
<p>" #
(12.15)
</p>
<p>where C is 4 ln 2 and Hj is the full width of the peak at half-maximum height of the jth Bragg
</p>
<p>reflection. The pseudo-Voigt distribution, which is a combination of a Gaussian and a Lorentzian, is
</p>
<p>used frequently to model the peak shape in X-ray diffraction patterns [31]; the Gaussian is of the form
</p>
<p>noted above, while the Lorentzian may be expressed as
</p>
<p>2
</p>
<p>pHj
1&thorn; 4 &eth;2yi � 2yj&THORN;
</p>
<p>2
</p>
<p>H2j
</p>
<p>" #�1
(12.16)
</p>
<p>The best fit of the calculated pattern to the observed pattern is judged by means of numerical
</p>
<p>criteria. Some of the criteria functions used are
</p>
<p>RF &frac14;
P
</p>
<p>j
</p>
<p>ffiffiffiffiffiffiffiffiffi
Iobs;j
</p>
<p>p
�
</p>
<p>ffiffiffiffiffiffiffiffiffiffi
Icalc;j
</p>
<p>p�
�
</p>
<p>�
</p>
<p>�
</p>
<p>P
</p>
<p>j
</p>
<p>ffiffiffiffiffiffiffiffiffi
</p>
<p>Iobs;j
p (12.17)
</p>
<p>606 12 Powder Diffraction</p>
<p/>
</div>
<div class="page"><p/>
<p>which is the conventional R-factor, but written in terms of
ffiffi
I
</p>
<p>p
instead of jFj;
</p>
<p>RB &frac14;
P
</p>
<p>j jIobs;j � Icalc;jjP
j Iobs;j
</p>
<p>(12.18)
</p>
<p>which is called the Bragg R-factor;
</p>
<p>Rp &frac14;
P
</p>
<p>j jyobs;j � ycalc;jjP
j yobs;j
</p>
<p>(12.19)
</p>
<p>which is the profile R-factor; and
</p>
<p>Rwp &frac14;
</p>
<p>ffiffiffiffiffiffiffiffiffiffiffiffiffiffiffiffiffiffiffiffiffiffiffiffiffiffiffiffiffiffiffiffiffiffiffiffiffiffiffiffiffiffiP
j wj&eth;yobs;j � ycalc;j&THORN;
</p>
<p>2
</p>
<p>P
j wj&eth;yobs;j&THORN;
</p>
<p>2
</p>
<p>vuut (12.20)
</p>
<p>which is the weighted profile R-factor: wj is a weighting factor for the jth point. A goodness-of-fit
</p>
<p>indicator w2 is also used, given by
</p>
<p>w2 &frac14; &eth;Rwp=Re&THORN;2 (12.21)
</p>
<p>where Re is the statistically expected R-factor:
</p>
<p>Re &frac14;
N � P&thorn; C
P
</p>
<p>j wj&eth;yobs;j&THORN;
2
</p>
<p>(12.22)
</p>
<p>Here, N is the number of observations j in the experimental powder profile, P is the number of
</p>
<p>parameters refined, and C is the number of constraints. In general, N is much greater than (P + C) so
</p>
<p>that the numerator tends closely to N. We shall indicate values for these parameters in the next
</p>
<p>section. Probably the most meaningful parameter is Rwp, since the numerator is the residual in (12.12)
</p>
<p>that is minimized. However, it is sensitive to a small number of poor agreements, which could arise
</p>
<p>from impurity.
</p>
<p>The Rietveld method, now used generally for powder studies, was programmed first by Rietveld
</p>
<p>[26], but it is now available in many program packages. The system RIETAN (see Appendix D), which
</p>
<p>is a system similar to GSAS, offers additionally a choice of three minimization algorithms that can be
</p>
<p>introduced under user-control in one and the same minimization process.
</p>
<p>A problem with Rietveld refinement, as with other forms of minimization, is the possibility of
</p>
<p>converging into a false minimum. The risk can be ameliorated by making reasonable variations to the
</p>
<p>starting model and refining, hopefully, to the same minimum. The three algorithms provided in
</p>
<p>RIETAN also provide for a possible way out of false minima. Guidelines for structure refinement
</p>
<p>using the Rietveld method have been published by the International Union of Crystallography (IUCr)
</p>
<p>Commission on Powder Diffraction [32].
</p>
<p>12.7.1 The Le Bail Method
</p>
<p>In this procedure, Le Bail et al. extended a method used by Rietveld [29, 33] to extract intensities for
</p>
<p>the calculation of Fourier maps during the course of a whole-profile refinement to the case where no
</p>
<p>initial structural model is available.
</p>
<p>12.7 The Rietveld Procedure 607</p>
<p/>
</div>
<div class="page"><p/>
<p>In the Rietveld procedure, the profile is defined by a number j of digitized points and, in the case of
</p>
<p>a resolved peak, the background-subtracted points are summed. For overlapping peaks 1 and 2, the
</p>
<p>integrated intensities Jobs are obtained through the equations
</p>
<p>J1obs &frac14;
X
</p>
<p>j
</p>
<p>I1;calcq1;j
</p>
<p>I1;calcq1;j &thorn; I2;calcq2;j
&eth;Pj � Bj&THORN; (12.23)
</p>
<p>J2obs &frac14;
X
</p>
<p>j
</p>
<p>I2;calcq2;j
</p>
<p>I1;calcq1;j &thorn; I2;calcq2;j
&eth;Pj � Bj&THORN; (12.24)
</p>
<p>where J1,obs is equal to m1F
2
o;1; m1 is the multiplicity factor for Fo,1, I1,calc is the calculated value of
</p>
<p>F2o;1 based on an appropriate model, and q1,j contains the Lorentz, polarization, absorption (if
</p>
<p>necessary), and shape factors associated with peak 1, and similarly for peak 2, (Pj � Bj) is the
measured (peak � background) term for the jth point in the pattern. By summing (12.23) and (12.24)
we obtain
</p>
<p>J1;obs &thorn; J2;obs &frac14;
X
</p>
<p>j
</p>
<p>&eth;Pj � Bj&THORN; (12.25)
</p>
<p>so that the sum of the peak areas is equal to the background-subtracted area, as in the case of resolved
</p>
<p>peaks.
</p>
<p>The Le Bail procedure itself [34] is essentially an iterative version of that of Rietveld, Sect. 12.7, to
</p>
<p>evaluate jFj, in which the observed peak area for the nth iteration is used as the calculated peak area
for the (n + 1)th iteration; generally, a rapid convergence is obtained for overlapping reflections. The
</p>
<p>procedure is programmed and available in systems such as FULLPROF and GSAS (Appendix D). It
</p>
<p>is said that the Le Bail method has led to a marked and radical increase in the ability to solve
</p>
<p>structures from powder diffraction data.
</p>
<p>12.7.2 The Pawley Method
</p>
<p>An alternative procedure for determining peak intensities in the absence of a structural model has
</p>
<p>been given by Pawley [35]. Whereas the Rietveld procedure is a least-squares analysis of a diffraction
</p>
<p>pattern where the principal variable parameters are peak position, peak shape, and those parameters
</p>
<p>dependent on peak area (atomic coordinates, thermal parameters, and intensity correction factors), the
</p>
<p>Pawley method is a least-squares analysis of the powder pattern in which the variable parameters are
</p>
<p>again peak position and shape but those concerning the peak area are the peak area itself. No
</p>
<p>structural model is needed, and the method is available in the program system ALLHKL [35, 36].
</p>
<p>In both the Le Bail and Pawley algorithms, the variables that are related to peak positions and shapes
</p>
<p>are the same as in the Rietveld structure refinement method.
</p>
<p>12.8 Examples of Solved Structures
</p>
<p>Once the intensity data are available, attempts can be made to solve the phase problem, so as to obtain a
</p>
<p>structural model that can be subjected to at least a partial refinement. The attack may take place by
</p>
<p>means of the so-called traditional approach, that is, by Patterson or direct methods along the lines
</p>
<p>608 12 Powder Diffraction</p>
<p/>
</div>
<div class="page"><p/>
<p>discussed for single crystals, or newer techniques such as Monte Carlo, simulated annealing, and
</p>
<p>maximum entropy can be brought to bear on the problem. As with single-crystal structure determi-
</p>
<p>nation, Patterson methods tend to be applied where heavy atoms are present in the molecules, and a
</p>
<p>direct method is used mainly for equal-atom structures. Refinement usually involves a combination
</p>
<p>of difference-Fourier syntheses to locate any missing atoms and Rietveld whole-profile structure
</p>
<p>refinement. We shall examine some structures with a view to indicating the stages that have been
</p>
<p>found necessary in obtaining satisfactory solutions.
</p>
<p>12.8.1 Traditional Methods
</p>
<p>Calcium Uranate
The early work by Rietveld was carried out with data obtained by neutron diffraction, but the use of
</p>
<p>X-rays, and particularly with a synchrotron source, is now well developed. Figure 12.11 shows an
</p>
<p>early diffractogram obtained by neutron diffraction on calcium uranate, CaUO4. The fitted and
</p>
<p>experimental profiles show excellent agreement [37].
</p>
<p>Manganese Phosphate Monohydrate
This structure, originally thought to be a 1
</p>
<p>1
</p>
<p>2
H2O hydrate, was solved by Patterson methods [38],
</p>
<p>using X-ray powder diffractometer data collected at a synchrotron source. X-rays of wavelength
</p>
<p>1.3208 Å were selected by means of a Ge (111) monochromator, and the radiation scattered in the
</p>
<p>vertical plane was measured.
</p>
<p>The pattern was indexed by the program ITO, and the unit cell at anM20 of 196 had the dimensions
</p>
<p>a &frac14; 6.916 Å, b &frac14; 7.475 Å, c &frac14; 7.361 Å, and b &frac14; 112.32�; Z was 4, and the space group either Cc or
</p>
<p>Fig. 12.11 Neutron diffraction pattern of a powdered sample of CaUO4, measured at l &frac14; 2.565 Å; the full line shows
the calculated profile and the dots indicate the observed profile
</p>
<p>12.8 Examples of Solved Structures 609</p>
<p/>
</div>
<div class="page"><p/>
<p>C2/c from systematic absences. We may note here that it may be inadvisable to decide the presence or
</p>
<p>absence of a center of symmetry solely by statistical tests, as the distribution of Fo values can be
</p>
<p>affected by the methods of pattern decomposition [39, 40]; space group C2/c was confirmed by the
</p>
<p>structure analysis.
</p>
<p>Pattern decomposition was carried out with the pseudo-Voigt function, leading to 61 intensity
</p>
<p>data; Fig. 12.12 shows a typical modeled peak. The intensity data were used to generate a Patterson
</p>
<p>map. The manganese atom coordinates were determined from this map and a difference-Fourier
</p>
<p>synthesis located the phosphorus and three oxygen atoms.
</p>
<p>Refinement was carried out by minimizing the function
P
</p>
<p>j wj&eth;yobs;j � Cycalc;j&THORN;
2
, where the terms
</p>
<p>have the meanings already discussed, and C is a scaling constant. The weights for each jth point were
</p>
<p>calculated from the expression wj &frac14; &frac12;&eth;yj &thorn; bj&THORN;� &frac14; s2&eth;bj&THORN;��1, where bj is the background at the jth
point and s(bj) is the esd of bj. During refinement it was determined that the substance was, in fact, a
</p>
<p>monohydrate. Satisfactory final refinement was achieved, with the following agreement factors:
</p>
<p>RB &frac14; 4:74%; Rp &frac14; 12:2%; Rwp &frac14; 16:1%; Re &frac14; 15:4%; w2 &frac14; 1:1
</p>
<p>Cimetidine
The structure of cimetidine, C10H16N6S, has been solved from powder diffraction data [41]. In
</p>
<p>this example, a diffractometer and a synchrotron source of wavelength 1.4599(1) Å were used to
</p>
<p>obtain the data. The diffractogram was indexed by the program TREOR (M20 &frac14; 176) and the pattern
decomposed by a modified Pawley method. Several attempts were made with direct methods pro-
</p>
<p>grams, which located only sulfur and three other atoms. Further phase extraction was carried out with
</p>
<p>the direct methods program SIR88 (see below).
</p>
<p>The structure was completed by iterative Fourier and least-squares procedures. A difference-Fourier
</p>
<p>synthesis then located all hydrogen atoms.After finalRietveld refinement, the structure convergedwith the
</p>
<p>following residuals:
</p>
<p>RB &frac14; 1:9%; Rwp &frac14; 8:5%; Re &frac14; 6:9%; w2 &frac14; 1:5
</p>
<p>It is noteworthy that in the absence of the hydrogen atoms RB and Rwp were 11.3% and 16.2%,
</p>
<p>respectively. Figure 12.13 illustrates the profile refinement of cimetidine with and without the
</p>
<p>inclusion of the hydrogen atoms.
</p>
<p>Fig. 12.12 Manganese
phosphate monohydrate,
</p>
<p>MnPO4�H2O, modeled
profile: plot of the intensity
</p>
<p>of the observed (dots) and
</p>
<p>calculated (solid line)
</p>
<p>profiles as a function of 2y
</p>
<p>for the 31�1 peak. The lower
difference curve shows
</p>
<p>the accuracy of the pseudo-
</p>
<p>Voigt modeling function
</p>
<p>610 12 Powder Diffraction</p>
<p/>
</div>
<div class="page"><p/>
<p>12.8.2 SIR Program System
</p>
<p>In the context of direct methods here, SIR, as in SIR2004 [36], refers to phase determination by the
</p>
<p>method of seminvariants representations and should not be confused with SIR as used in Chaps. 7 and
</p>
<p>10 wherein it means single isomorphous replacement.
</p>
<p>A structure seminvariant is a linear combination of phases, the value of which is uniquely determined
</p>
<p>by the crystal structure alone, irrespective of the choice of permissible origin. For example, in space group
</p>
<p>P21 (y axis unique), f(h0l) is a seminvariant if h and l are both even. Again, if h1 &thorn; h2 &thorn; h3 &frac14; 2n,
k1 &thorn; k2 &thorn; k3 &frac14; 0, and l1&thorn; l2 &thorn; l3 &frac14; 2n0,where bothn andn0 are integers, thenf&eth;h1&THORN; &thorn; f&eth;h2&THORN; &thorn; f&eth;h3&THORN; is
</p>
<p>Fig. 12.13 Rietveld
profile refinement of
</p>
<p>cimetidine. (a) With
hydrogen atoms included,
</p>
<p>Rwp &frac14; 8.5 %. (b) Without
hydrogen atoms included,
</p>
<p>Rwp &frac14; 16.2 % (after
Cernik loc. cit.)
</p>
<p>12.8 Examples of Solved Structures 611</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_7">http://dx.doi.org/10.1007/978-1-4614-3954-7_7</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_10">http://dx.doi.org/10.1007/978-1-4614-3954-7_10</a></div>
</div>
<div class="page"><p/>
<p>a seminvariant. Further discussions of the properties of structure seminvariants may be found in the
</p>
<p>literature [42].
</p>
<p>The SIR method is based on the estimation of 1- and 2-phase structure seminvariants and 3- and
</p>
<p>4-phase structure invariants, according to the theory of representations [43, 44]. The program
</p>
<p>functions in all space groups without user intervention, although a knowledge of partial structure
</p>
<p>moieties may be exploited with advantage. A later version of SIR, SIR2008, has been published
</p>
<p>[45, 46] (see also Appendix D).
</p>
<p>Silver&ndash;Pyrazole Complex
The structure of the silver&ndash;pyrazole complex [Ag(pz)]3 (pz &frac14; pyrazole), C9H9N6Ag6, has been
solved by direct methods [47] as part of a program investigating complexes between pyrazole,
</p>
<p>C3H4N2, and copper or silver.
</p>
<p>Powder diffractometer data were collected with graphite-monochromatized X-radiation of wave-
</p>
<p>length 1.5418 Å, at values of 17&ndash;85� in 2y-steps of 0.02�. The program TREOR was used to index the
pattern, and it gave the orthorhombic unit cell a &frac14; 13.13 Å, b &frac14; 10.56 Å, and c &frac14; 8.79 Å with
M20 &frac14; 15. The indexed data suggested the space group Pbcn, which was confirmed by the structure
analysis. The refined unit cell had the dimensions a &frac14; 13.1469(4), b &frac14; 10.5702(10), and
c &frac14; 8.7921(4). Since Z &frac14; 4, there are 12 Ag(pz) moieties in the unit cell.
</p>
<p>Pattern decomposition was achieved by the Pawley method through application of the program
</p>
<p>ALLHKL to 924 reflection data, using 3400 points. The program SIRPOW92 (see below) was used to
</p>
<p>extract a direct methods trial model comprising the three silver atoms in the asymmetric unit, one of
</p>
<p>which lies on a twofold axis. At this stage, Rp &frac14; 0.29 and Rwp &frac14; 0.37. The structure was completed and
Rietveld refinement, excluding hydrogen atoms, converged with the following set of indicating para-
</p>
<p>meters:
</p>
<p>RF &frac14; 6:0%; Rp &frac14; 11:5%; Rwp &frac14; 14:9%;
Re &frac14; 6:6%; w2 &frac14; 5:1
</p>
<p>The value of w2 is larger than normal for a refined structure. It has been noted [48] that the
</p>
<p>parameters Re and w
2 depend on the intensity counting statistics and are, at best, only gross goodness-
</p>
<p>of-fit parameters.
</p>
<p>12.8.3 EXPO Program System
</p>
<p>The program SIRPOW92 that was employed in the structure determination just described has been
</p>
<p>incorporated into the powerful structure-solving program system EXPO that is now much used. The
</p>
<p>main addition to the SIRPOW92 system is the routine EXTRA, the purpose of which is to
</p>
<p>provide for the decomposition of the pattern according to the Le Bail algorithm. Figure 12.14
</p>
<p>illustrates the flow diagram for EXPO. Advances in pattern decomposition have made it possible
</p>
<p>to introduce into EXTRA the positivity of the electron density and Patterson functions, the
</p>
<p>treatment of pseudo-translational symmetry and preferred orientation, and the availability of a
</p>
<p>partial structure. Such information, when detected in SIRPOW, is recycled as shown in
</p>
<p>Fig. 12.14, so as to obtain better extracted values for F2o. Additional user-friendly facilities
</p>
<p>include contouring of Fourier maps, representation of crystal structures by coordination polyhe-
</p>
<p>dra, and automatic preliminary Rietveld refinement, which is triggered when the structure is
</p>
<p>incomplete. A more recent innovation is the labeling of peaks in electron density maps, based on
</p>
<p>chemical information rather than on electron density peak heights [36].
</p>
<p>612 12 Powder Diffraction</p>
<p/>
</div>
<div class="page"><p/>
<p>The EXPO system has been made available to the academic community by the courtesy of its
</p>
<p>authors, and can be obtained from a web site (see Appendix D); several versions have been described
</p>
<p>in the literature [49, 50], the latest version [51] being EXPO2009.
</p>
<p>12.9 Direct-Space Methods
</p>
<p>Direct-space methods, not to be confused with traditional direct methods, have evolved from model
</p>
<p>building techniques in which chemical information is used in order to construct a sensible structural
</p>
<p>model. Such a model can then be used as a basis for calculating a diffraction pattern with which to
</p>
<p>compare the measured pattern. The inclusion of information on bond lengths, bond angles, connec-
</p>
<p>tivities, torsion angles, orientation, and position means that a very large number of chemically
</p>
<p>sensible models could be produced. Thus, computing power must be brought to bear in order for
</p>
<p>such an approach to be practicable. Encoding frequently results in algorithms being specific to a given
</p>
<p>class of compounds.
</p>
<p>In the case of molecular compounds, information on bond lengths and bond angles is well
</p>
<p>documented, so that the number of variables is reduced to that defining orientation and position,
</p>
<p>and possibly torsion. In the case of zeolite structures, for example, the chemical composition and
</p>
<p>connectivities of structural units are of significance in designing useful structural models.
</p>
<p>Fig. 12.14 Flow diagram for the program system EXPO; structural information detected by SIRPOW can be recycled
to EXTRA, so as to obtain more reliable intensity data (after Altomare et al., loc. cit. [49])
</p>
<p>12.9 Direct-Space Methods 613</p>
<p/>
</div>
<div class="page"><p/>
<p>An alternative approach involves generating a model by the random placement of atoms of the
</p>
<p>required number and types in the unit cell, then applying shifts to the atoms in predetermined
</p>
<p>amounts, and calculating and comparing patterns so as to find a best fit. Constraints in terms of
</p>
<p>known chemical information can be introduced into the model, so as to increase the plausibility of the
</p>
<p>model and to reduce the time consumption of computing facilities. We consider examples of these
</p>
<p>and other related techniques in the remainder of this section.
</p>
<p>12.9.1 Zeolites and the FOCUS Algorithm
</p>
<p>Zeolite structures have many important applications as molecular sieves, absorbents, catalysts, and
</p>
<p>ion-exchange materials. These properties are related to both their unusual structures and, in particular,
</p>
<p>to their framework topologies, that is, the way in which the tetrahedral structural units are linked in
</p>
<p>the solid state. As zeolites are microcrystalline, powder diffraction proves to be the only method
</p>
<p>available for their detailed structural examination.
</p>
<p>The structures are complex, often with high symmetry, such as P
63
</p>
<p>m
mc or Fm�3c, and can have unit
</p>
<p>cells, with dimensions up to 40 Å, so that a high degree of overlap of diffraction maxima arises.
</p>
<p>Generally, direct methods have been used for solving the structures, with difference-Fourier synthesis
</p>
<p>and Rietveld profile refinement to complete and refine the structure [16].
</p>
<p>A different approach is used in the program environment FOCUS [52], indicated in Fig. 12.15,
</p>
<p>where Fourier recycling is combined with a specialized topology search and topology classification
</p>
<p>scheme. The method makes use of crystal-chemical information such as chemical composition,
</p>
<p>probable interatomic distances, and the fact that all zeolite structures have three-dimensional four-
</p>
<p>connected frameworks, in order to aid in the interpretation of electron density maps.
</p>
<p>TheQ values of about 20 high-accuracy powder lines serve to determine the unit-cell parameters and
</p>
<p>index the pattern. The space group follows from the indexed lines, and integrated intensities are extracted.
</p>
<p>Random phases consistent with the space group are assigned to the extracted reflection inten-
</p>
<p>sities and an electron density map generated. If enough of the phases are (by chance) correct, some
</p>
<p>of the features of the structure will appear, and an attempt is made to interpret the map using the
</p>
<p>chemical information outlined above. The resulting model is then used to generate new phases and
</p>
<p>a new electron density map. This Fourier interpretation and recycling procedure is continued until
</p>
<p>either phase convergence or a maximum number of cycles is reached. Then the process is started
</p>
<p>again, if necessary, with a new set of random phases. Each time an electron density map is
</p>
<p>generated, a search is also made for a three-dimensional four-connected net.
</p>
<p>If one is found, it is classified and written to a file. The procedure is terminated when a sufficient
</p>
<p>number of such nets have been found. The net that occurs most frequently is usually the correct
</p>
<p>framework structure. The model is then used as a starting point for structure completion and Rietveld
</p>
<p>refinement. Figure 12.16 illustrates the FOCUS algorithm, the detail of that section of Fig. 12.15 that
</p>
<p>is enclosed by dashed lines.
</p>
<p>12.9.2 Zinc&ndash;Silicate Complex VIP-9
</p>
<p>As an example of a large structure solved only when the FOCUS procedure was applied, we cite the
</p>
<p>zinc-silicate molecular sieve complex [54] VPI-95. The synthesized material corresponds in chemical
</p>
<p>composition to Rb38&ndash;43K5&ndash;10[Si96Zn24O240]�48H2O, from chemical analysis and
29Si NMR spectros-
</p>
<p>copy. Room temperature powder patterns were collected on a Scintag XDS 2000 powder
</p>
<p>614 12 Powder Diffraction</p>
<p/>
</div>
<div class="page"><p/>
<p>diffractometer, operating in Bragg-Brentano geometry, with a flat-plate sample and Cu Ka radiation
</p>
<p>(l &frac14; 1.54184 Å). Most of the peaks were indexed on a hexagonal unit cell, with a � 9.9 Å and
c � 37 Å.
</p>
<p>A sample exchanged with (NH4&THORN;&thorn; was used, in addition to the as-synthesized sample, in order to
obtain high-resolution powder patterns at the European Synchrotron Radiation Facility (ESRF) at
</p>
<p>Grenoble. The pattern of the (NH4&THORN;&thorn;-exchanged sample was indexed completely in space group
P
42
n
cm, with a &frac14; 9.8946 Å and c &frac14; 36.8715 Å. The framework topology was determined by applying
</p>
<p>the FOCUS procedure, as described above. While a preliminary Rietveld refinement indicated that
</p>
<p>the topologywas probably correct, thismodelwas not pursued because the exchangewas incomplete and
</p>
<p>the detection of (NH4&THORN;&thorn; in the presence of H2O would be difficult.
The pattern of the as-synthesized material was indexed only in space group P41212, with
</p>
<p>a &frac14; 9.8837(1) Å and c &frac14; 73.6505(6) Å, an approximate doubling along c compared to the
(NH4&THORN;&thorn;-exchanged compound. Rietveld refinement was applied with restraints on the (Si, Zn)&ndash;O
bond length, and the O&ndash;(Si, Zn)&ndash;O and (Si, Zn)&ndash;O&ndash;(Si, Zn) bond angles. A series of difference-Fourier
</p>
<p>Fig. 12.15 The FOCUS
structure-solving
</p>
<p>environment, tailored to the
</p>
<p>solving of zeolite
</p>
<p>structures. The essential
</p>
<p>algorithm for the zeolites is
</p>
<p>enclosed in dashed lines;
</p>
<p>the remaining blocks in the
</p>
<p>diagram refer to standard
</p>
<p>procedures [53]
</p>
<p>12.9 Direct-Space Methods 615</p>
<p/>
</div>
<div class="page"><p/>
<p>maps based on iteratively improved models led to better agreement, but only as far as RF &frac14; 0.167 and
Rwp &frac14; 0.453.
</p>
<p>By applying further chemical reasoning to the model, together with a new series of difference-Fourier
</p>
<p>maps and Rietveld refinement, gradual improvements in the model were obtained. The final refinement
</p>
<p>converged atRF &frac14; 0.069,Rwp &frac14; 0.147,Re &frac14; 0.099, and w2 &frac14; 2.2. In all, 170 structural parameters were
refined, which is one of the largest framework topology structures solved from powder data without
</p>
<p>manual intervention. Figure 12.17 showsprofiles for theRietveld refinement of the as-synthesizedVIP-10.
</p>
<p>The framework topology has seven T-sites, or nodes [51, 55] in the asymmetric unit. The
</p>
<p>framework can be described in terms of two types of layers linked by isolated tetrahedra,
</p>
<p>Fig. 12.16 The FOCUS
algorithm in detail
</p>
<p>(compare Fig. 11.14):
</p>
<p>automatic Fourier
</p>
<p>recycling and topology
</p>
<p>searching is carried out
</p>
<p>here (after Grosse-
</p>
<p>Kuntsleve, loc. cit. [53])
</p>
<p>616 12 Powder Diffraction</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_11#Fig14_11">http://dx.doi.org/10.1007/978-1-4614-3954-7_11#Fig14_11</a></div>
</div>
<div class="page"><p/>
<p>Figs. 12.18 and 12.19. The simpler layer A is a 4.82 net, which is a two-dimensional string of
</p>
<p>undulating 4-membered rings (4-rings) and 8-rings, as shown in Fig. 12.18.
</p>
<p>The building unit of layers B is a polyhedron consisting of a 3-ring with three bent 5-rings attached
</p>
<p>to it, which may be termed a [533] structural unit. The polyhedra share 3-ring faces on one side, and
</p>
<p>5-ring edges on the other side, thus forming infinite chains parallel to h110i. Neighboring, parallel
chains are shifted relative to one another by a half-chain length, and these structural units are new to
</p>
<p>zeolite data.
</p>
<p>12.10 Monte Carlo Method
</p>
<p>In this approach, a series of structural models in direct space is generated by random movements of
</p>
<p>a set of atoms in the unit cell, and each state of the system is evaluated on the basis of the agreement
</p>
<p>between the observed and calculated powder diffraction patterns. The models are postulated
</p>
<p>independently of the diffraction data, and once a satisfactory model has been obtained, it is refined
</p>
<p>by the Rietveld procedure. The atom positions may be chosen independently at random, as a group of
</p>
<p>atoms known to be part of the structure, particularly if the group forms a rigid body, or in terms of the
</p>
<p>connectivity of a molecule. In the latter case, the molecule is usually described in terms of internal
</p>
<p>coordinates, that is, bond lengths, bond angles, and torsion angles, that are converted into Cartesian
</p>
<p>coordinates, and the variables are the orientation and positions of the molecule in the unit cell. In some
</p>
<p>cases, unknown torsion angles may be additional variables.
</p>
<p>The Monte Carlo method itself is based on the well-known Metropolis algorithm [56], and each
</p>
<p>state of the system is tested by calculating a residual Rwp, rather than an energy term as in its original
</p>
<p>applications. An initial configuration of atoms xi is displaced in a random manner but with
</p>
<p>Fig. 12.17 Profile refinement for VIP-9, 0&ndash;30� (top) and 30&ndash;60� (bottom) in 2y. In each diagram, the top, middle and
bottom profiles refer to the observed, calculated, and difference patterns, respectively
</p>
<p>12.10 Monte Carlo Method 617</p>
<p/>
</div>
<div class="page"><p/>
<p>constraints, relating, for example, to the amount of change permitted in the parameters of the set xi,
which may involve translations (in x, y, z), rotations about orthogonal axes (in y, f, C) and torsion
</p>
<p>movements (in t1, t2, t3, . . .).
</p>
<p>After each movement, the powder pattern is calculated, scaled to the observed pattern, and the
</p>
<p>whole-profile Rwp factor calculated. Alternatively, the extracted intensities can be used as long as an
</p>
<p>account is taken of the correlations between neighboring reflections [57].
</p>
<p>Each trial structure is assessed on the basis of the difference Z, such that
</p>
<p>Z &frac14; Rwp&eth;xcurrent&THORN; � Rwp&eth;xprevious&THORN; (12.26)
</p>
<p>(Z 	 0), xcurrent is accepted; if Z &gt; 0, xcurrent is accepted with a probability exp(�Z/S), where S is a
scaling factor that operates like the energy parameter kT in the more conventional applications of the
</p>
<p>Metropolis algorithm [58]. It follows that the probability for rejection is [1 � exp(�Z/S)]: if xcurrent is
rejected, the previous trial structure now becomes &ldquo;current.&rdquo; These stages are repeated, Fig. 12.20, so
</p>
<p>generating aMarkov chain, that is, a sequence of events in which the outcome of each step is independent
</p>
<p>of the previous step; the probability of the change xi ! xj depends only on the states i and j. Eventually,
the event showing the lowestRwp value is subjected to Rietveld refinement. This technique has now been
</p>
<p>applied successfully to a number of structures [11, 59]. An important feature of theMetropolis algorithm
</p>
<p>is that it biases the generation of configurations towards those that are significant for the true solution.
</p>
<p>p-Bromophenylethanoic Acid
An interesting application of the Monte Carlo method elucidated the previously unknown structure of
</p>
<p>p-bromophenylethanoic acid, BrC6H4CH2CO2H [60]. By computer indexing of the first 20 lines
</p>
<p>of the powder pattern, the unit-cell dimension found were a &frac14; 16.020 Å, b &frac14; 4.607 Å, c &frac14; 11.715 Å,
and b &frac14; 109.33�. The systematic absences indicated space group P21/c, with Z &frac14; 4.
</p>
<p>The Monte Carlo technique was applied in two stages. In stage 1, the bromine atom alone was
</p>
<p>used. Of 1000 moves, one was found to be the most probable, at Rwp &frac14; 5.2%; the value of Rwp for a
</p>
<p>Fig. 12.18 The framework topology of VIP-9 viewed along [110]; oxygen atoms have been omitted for clarity.
</p>
<p>Layer B at z &frac14; 1
2
is rotated by 90� with respect to those at z &frac14; 0 and 1. Similarly layer A at z &frac14; 3
</p>
<p>4
is rotated by 90�
</p>
<p>relative to that at z &frac14; 1
4
</p>
<p>618 12 Powder Diffraction</p>
<p/>
</div>
<div class="page"><p/>
<p>totally random placement is ca. 55%. In the second stage, the bromine atom was constrained in its best
</p>
<p>determined position, and the rigid C7 fragment of the molecule rotated at random about an axis
</p>
<p>passing through the bromine atom. From the best fit model from this stage, the remaining atoms were
</p>
<p>obtained by difference-Fourier methods, and Rietveld refinement of the structure converged at
</p>
<p>Rwp &frac14; 6.66%.
Figure 12.21 illustrates the powder diffraction profiles from the Rietveld refinement. The differ-
</p>
<p>ence between the stage 1 and final bromine atom positions was 0.2 Å, and other differences up to
</p>
<p>1.3 Å were recorded for other atoms. This analysis shows well the power of this technique, which may
</p>
<p>now be considered one of the standard methods in the crystallographer&rsquo;s armory.
</p>
<p>Another structure that has been solved successfully by the Monte Carlo method is that of 5-
</p>
<p>bromonicotinic acid [61].
</p>
<p>Fig. 12.19 Layer-like
building units in VIP-9:
</p>
<p>layer A, 4.82 net; layer B,
</p>
<p>chains of [533] polyhedra
</p>
<p>12.10 Monte Carlo Method 619</p>
<p/>
</div>
<div class="page"><p/>
<p>Fig. 12.20 Monte Carlo cycling procedure. If a trial structure xi is accepted, it becomes xi+1 and is cycled back;
otherwise xi is returned for new random movements
</p>
<p>Fig. 12.21 Profile refinement for p-BrC6H4CH2CO2H; the experimental points are marked + and the calculated
profile is the full line. The lower line is the difference-Fourier profile</p>
<p/>
</div>
<div class="page"><p/>
<p>12.10.1 Simulated Annealing
</p>
<p>The simulated annealing procedure [62] can be combined with the Monte Carlo method. In this case,
</p>
<p>the Metropolis algorithm is applied as before but with a systematic decreasing of the parameter S in
</p>
<p>the exponential function above, which is similar to a decrease in temperature of the energy quantity
</p>
<p>kT. The temperature controls the potential energy surface that is scanned, and the Metropolis loop
</p>
<p>accepts the lower energy solution. A higher energy solution can be accepted if the temperature is
</p>
<p>raised. The starting value of S is set such that all trial structures are accepted; then, as S is decreased,
</p>
<p>so the poorer fits (larger Rp values) are excluded. The final structure, the best fit, is then subjected to
</p>
<p>Rietveld refinement. In cases where the model does not contain the whole structure, the remainder can
</p>
<p>usually be located by difference-Fourier synthesis, Sect. 8.4.5.
</p>
<p>Variations in this procedure have been proposed differing in the selection of the S parameter [63]
</p>
<p>and the rate at which it is changed or in the generation of trial configurations. One such variant has
</p>
<p>been incorporated into the program system DASH [65].
</p>
<p>12.11 ESPOIR Program System
</p>
<p>While Monte Carlo methods imply an element of random-sampling, the fitting of a model to
</p>
<p>scattering data in this context is sometimes termed a reverse Monte Carlo procedure. The program
</p>
<p>system ESPOIR [35, 66] employs the reverse Monte Carlo technique coupled with simulated
</p>
<p>annealing for ab initio structure determination. It can use a completely random starting model, or
</p>
<p>else incorporates a structural entity of known geometry, in which case it functions similar to
</p>
<p>molecular replacement.
</p>
<p>The program fits the starting model to either Fo data extracted from a powder pattern, or to single-
</p>
<p>crystal data in the unlikely event that one of the single-crystal techniques fails. The author of the
</p>
<p>ESPOIR program system has kindly consented to its inclusion with the suite of programs supplied
</p>
<p>with this book.
</p>
<p>An innovative and important computer-time-saving feature of this program relates to the problem
</p>
<p>of overlapping peaks. Direct-space methods generally either fit the raw data to a model and derive
</p>
<p>measures of fit for each model, or fit some equation involving the extracted Fo data, taking into
</p>
<p>account overlapping of peaks. ESPOIR follows a method intermediate between these two: instead of
</p>
<p>fitting the raw data, a pseudo powder pattern P(2y) is reconstructed from the extracted Fo data. In this
</p>
<p>way no background, Lorentz, polarization, absorption, asymmetry, profile shape, or reflection
</p>
<p>multiplicity corrections have to be considered. A Gaussian shape function G is used for fast
</p>
<p>calculation and gross approximation to the overlapping, and the best fit between the pseudo pattern
</p>
<p>and the calculated pattern may be judged from the parameter Rwp, where
</p>
<p>Rwp &frac14;
P
</p>
<p>jP&eth;2y&THORN;obs � KP&eth;2y&THORN;calcjP
P&eth;2y&THORN;obs
</p>
<p>(12.27)
</p>
<p>where P(2y)obs &frac14; GFo and K is a scale factor. Simulated annealing is introduced so as to reduce
progressively the magnitude of the atomic displacements. An additional variable parameter permits
</p>
<p>the acceptance of Monte Carlo events that do not necessarily decrease Rwp, so as to avoid false
</p>
<p>minima; about 40% of such events are retained typically. Nevertheless, at least ten independent runs
</p>
<p>are recommended for a chance of success. With a more complex structural problem, more runs may
</p>
<p>be needed.
</p>
<p>12.11 ESPOIR Program System 621</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_8#Sec34_8">http://dx.doi.org/10.1007/978-1-4614-3954-7_8#Sec34_8</a></div>
</div>
<div class="page"><p/>
<p>The ESPOIR procedure is illustrated with examples and problems at the end of Sect. 13.8ff. It
</p>
<p>must be borne in mind that ESPOIR is a program for obtaining a starting model for further
</p>
<p>refinement, either by the Rietveld technique or by traditional procedures. As with all other
</p>
<p>procedures, ESPOIR is not guaranteed to lead to success: (Fr. espoir &frac14; hope).
</p>
<p>a-Lanthanum Tungstate
a-Lanthanum tungstate, La2W2O9, crystallizes in space group P�1, with a &frac14; 7.2489(1) Å, b &frac14; 7.2878
(1) Å, c &frac14; 7.0435(1) Å, a &frac14; 96.367(1)�, b &frac14; 94.715(1)�, g &frac14; 70.286(1)�, and Z &frac14; 2. Diffraction
patterns were obtained with both X-rays (Cu Ka) and neutrons (l &frac14; 1.5939 Å). The pattern was
indexed by TREOR to give a single triclinic solution (M20 &frac14; 24). Using the X-ray data, intensities
were extracted by the program FULLPROF and the structure solved by SHELXS-86 to locate two
</p>
<p>lanthanum and two tungsten sites. No full solution was found at this stage: the scattering from oxygen
</p>
<p>in the presence of lanthanum and tungsten is relatively weak.
</p>
<p>The structure was solved with the neutron data using the program ESPOIR [67]. A [WO4]
</p>
<p>tetrahedron was first used as a rigid-body search fragment, rotating around fixed positions for
</p>
<p>tungsten and lanthanum. However, Rp would reduce no lower than 25%. Finally, the lanthanum
</p>
<p>and tungsten positions were fixed at the coordinates given by the X-ray study, and nine oxygen atoms
</p>
<p>searched with ESPOIR by a random approach, using the neutron data. An Rp of 6.2% was achieved,
</p>
<p>and then the whole neutron pattern subjected to a final Rietveld refinement. Figure 12.22 illustrates
</p>
<p>the final profile refinement, and Table 12.5 summarizes results from the X-ray and neutron studies.
</p>
<p>The differences between the two tabulated sets of results probably relate to preferred orientation,
</p>
<p>which was considered to be the cause of the failure to find the oxygen atoms by difference-Fourier
</p>
<p>synthesis, thus leading to the poorer agreement indices. One aspect of the structure is shown in
</p>
<p>projection in Fig. 12.23. The tungsten atoms are of two types: W1 is coordinated by five oxygen
</p>
<p>atoms forming a trigonal bipyramid, whereas W2 is coordinated octahedrally. This structure could
</p>
<p>explain the failure of the models based on [WO4] tetrahedral units. The sharing of corners builds up
</p>
<p>Fig. 12.22 Final profile refinement for a-La2W2O9: circles, observed pattern; full line, calculated pattern; vertical
lines, reflection positions; bottom profile, difference pattern
</p>
<p>622 12 Powder Diffraction</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_13#Sec37_13">http://dx.doi.org/10.1007/978-1-4614-3954-7_13#Sec37_13</a></div>
</div>
<div class="page"><p/>
<p>4-rings to give [W4O18] structural units in a cyclo-tungstate structure. The two lanthanum atoms are
</p>
<p>coordinated, respectively, by nine and ten oxygen atoms.
</p>
<p>12.12 Powder Diffraction with Proteins
</p>
<p>X-ray diffraction analysis of the structure of a protein is often limited by the availability of suitable
</p>
<p>single crystals. However, it has been shown that the absence of single crystals may not present an
</p>
<p>insurmountable difficulty in this field any more than it does in materials science, as powder diffraction
</p>
<p>techniques have developed to the point where, as we have seen in the previous section, complex
</p>
<p>oxides, zeolites, and small organic molecular structures are often solved from powder data alone.
</p>
<p>12.12.1 T3R3 Zinc&ndash;Insulin Complex
</p>
<p>The applicability of powder diffraction methods to proteins has been demonstrated, for example, with
</p>
<p>the structure solution and refinement of a new variant of the T3R3 human zinc&ndash;insulin complex [68],
</p>
<p>produced by mechanically grinding a polycrystalline sample.
</p>
<p>Table 12.5 Results from the X-ray and neutron data collection for a-
La2W2O9
</p>
<p>X-ray Neutron
</p>
<p>Number of reflections 1317 1366
</p>
<p>Number of parameters refined 58 69
</p>
<p>Peak-shape function Pseudo-Voigt Pseudo-Voigt
</p>
<p>RB 0.114 0.030
</p>
<p>Rp 0.156 0.060
</p>
<p>Rwp 0.186 0.069
</p>
<p>Re 0.064 0.019
</p>
<p>Fig. 12.23 Projection of the structure of a-La2W2O9 along a, showing the two environments of the tungsten atoms
</p>
<p>12.12 Powder Diffraction with Proteins 623</p>
<p/>
</div>
<div class="page"><p/>
<p>High-resolution synchrotron X-ray powder-diffraction data were used to solve this crystal struc-
</p>
<p>ture by molecular replacement adapted for Rietveld refinement. A complete Rietveld refinement of
</p>
<p>the 1630-atom protein structure, achieved by combining 7981 stereochemical restraints with a 4800-
</p>
<p>step (dmin &frac14; 3.24 Å) powder-diffraction pattern, yielded the residuals Rwp &frac14; 3.73%, Rp &frac14; 2.84%,
and RF &frac14; 8.25%.
</p>
<p>It was found that the grinding-induced phase change was accompanied by 9.5 and 17.2� rotations
of the two T3R3 moieties in the crystal structure. The material reverts in 2&ndash;3 days to give the original
</p>
<p>T3R3 crystal structure. A Rietveld refinement of this 815-atom protein structure, by combining 3886
</p>
<p>stereochemical restraints with a 6000-step (dmin &frac14; 3.06 Å) powder-diffraction pattern, yielded the
residuals Rwp &frac14; 3.46%, Rp &frac14; 2.64%, and RF &frac14; 7.10%.
</p>
<p>The ability, demonstrated by this work, to solve and refine a protein crystal structure from powder
</p>
<p>diffraction data indicates that this approach could be employed, for example, to examine structural
</p>
<p>changes in a series of protein derivatives in which the structure of one member is known from a single-
</p>
<p>crystal study. An interesting paper on the application of powder methods to proteins has been given in
</p>
<p>the literature [69].
</p>
<p>12.13 Maximum Entropy in Crystal Structure Analysis
</p>
<p>12.13.1 Most Probable Distribution
</p>
<p>Entropy is a thermodynamic concept that may be used to describe the degree of order in a system.
</p>
<p>Although it was defined originally in terms of the operation of heat engines, it can also be addressed in
</p>
<p>terms of the probability of a system at a molecular level.
</p>
<p>Initially, imagine L gaseous electrons behaving ideally, where L is the Avogadro number,
</p>
<p>constrained to one-half V1 of a total containing volume V2 and then allowed to expand reversibly at
</p>
<p>constant temperature so as to occupy the total volume V2. The probability that any given electron is
</p>
<p>now present in the volume V1 is
1
</p>
<p>2
. The probability that two such electrons are present in V1 and the
</p>
<p>probability that the two electrons are in the volume V1 would be &eth;
1
</p>
<p>2
&THORN;2, since these two events are
</p>
<p>uncorrelated. Thus, the probability that all L electrons occupy the volume V1 is &eth;
1
</p>
<p>2
&THORN;L, and W1/W2 is,
</p>
<p>therefore, &eth;1
2
&THORN;L, where W1 is less than W2. Since any similar ratio of volumes could be chosen,
</p>
<p>W1/W2 &frac14; (V1/V2)L. From the study of heat engines, we know that the heat change qrev in the expansion
of a gas behaving ideally from a volume V1 to a volume V2 is given by qrev &frac14; nRT ln(V2=V1&THORN; per mole,
where n is the amount of substance and R is the universal gas constant. But qrev/T is a measure of the
</p>
<p>molar entropy change DS, so that DS &frac14; R ln&eth;V2=V1&THORN;. In our analysis, n &frac14; 1 and the volume that of the
electron gas;
</p>
<p>DS &frac14; S2 � S1 &frac14; R ln&eth;V2=V1&THORN; &frac14; R&eth;ln W2 � ln W1�1=L &frac14; &eth;R=L&THORN;&frac12;ln W2 � ln W1�
&frac14; kB&frac12;ln W2 � ln W1� (12.28)
</p>
<p>where kB is the Boltzmann constant. Thus the entropy of a system is identified with probability
</p>
<p>through the Boltzmann equation:
</p>
<p>S &frac14; kB ln W (12.29)
</p>
<p>624 12 Powder Diffraction</p>
<p/>
</div>
<div class="page"><p/>
<p>We can look upon the probability W as the number of ways a system can be constructed within
</p>
<p>a fixed framework. Consider distributing five identical but distinguishable electrons among a set
</p>
<p>of boxes B0&ndash;B4 such that the total number of boxes is 10. The possible arrangements are set out
</p>
<p>in Table 12.6.
</p>
<p>Of the 381 possible arrangements, one distribution is four times more probable than the next most
</p>
<p>probable distribution. As the number of electrons increases, one distribution becomes outstandingly
</p>
<p>more probable: this is the maximum entropy distribution and is assumed to be the true distribution.
</p>
<p>12.13.2 Electron Density Map
</p>
<p>Consider the grid points at which an electron density map is normally calculated. The number of
</p>
<p>electrons available for these sites is given and the unit cell defines the boundaries for the electron
</p>
<p>distribution. If the electrons were distributed at random, a map could be produced but it would be
</p>
<p>unlikely to be correct. The number of ways in which the electrons can be arranged to form a map is a
</p>
<p>measure of the probability that we need to determine.
</p>
<p>The problem is similar to that described for the distribution of electrons in boxes, but instead of
</p>
<p>electrons in boxes we have electrons in cells corresponding to the grid points at which an electron
</p>
<p>density map is calculated. We consider a total of N electrons in a unit cell divided intoM grid points.
</p>
<p>The first electron has a choice of N cells in which to go: a mutually exclusive assignment has a
</p>
<p>probability pi &frac14; ni=N, where pi is the probability of the ith electron assignment and ni the number of
electrons assigned in that choice. If the total number of electrons is large, the probability of any
</p>
<p>particular result is the multinomial distribution familiar in Maxwell-Boltzmann-statistics: P(p) &frac14;
W/MN, where W is given by
</p>
<p>W &frac14; N!
n1!n2!n3! . . . nm!
</p>
<p>(12.30)
</p>
<p>The most probable result is that which maximizes the value ofW, the total number of arrangements
</p>
<p>of the N electrons among the M cells. We write
</p>
<p>Table 12.6 Numbers W of arrangements of five electrons among
five boxes B0&ndash;B4
</p>
<p>B4 B3 B2 B1 B0 W
</p>
<p>1 1 1 1 1 5!/1! &frac14; 120
2 1 2)
</p>
<p>2 2 1)
</p>
<p>1 2 2) 5!(2!2!)
</p>
<p>1 2 2 ) 30 each &frac14; 180
2 2 1)
</p>
<p>2 1 2 )
</p>
<p>1 3 1)
</p>
<p>1 1 3 ) 5!/5! &frac14; 80
3 1 1) 20 each
</p>
<p>1 3 1
</p>
<p>5 1
</p>
<p>12.13 Maximum Entropy in Crystal Structure Analysis 625</p>
<p/>
</div>
<div class="page"><p/>
<p>1
</p>
<p>N
ln W &frac14; 1
</p>
<p>N
ln
</p>
<p>N!
</p>
<p>n1!n2! . . . nM
</p>
<p>� �
</p>
<p>&frac14; 1
N
</p>
<p>ln
N!
</p>
<p>Np1!Np2! . . .NpM
</p>
<p>� �
</p>
<p>&frac14; 1
N
</p>
<p>ln N!�
X
</p>
<p>M
</p>
<p>i&frac14;1
ln Npi!
</p>
<p> !
</p>
<p>(12.31)
</p>
<p>Applying Stirling&rsquo;s approximation for factorials:
</p>
<p>1
</p>
<p>N
ln W &frac14; 1
</p>
<p>N
N lnN �
</p>
<p>X
</p>
<p>M
</p>
<p>i&frac14;1
Npi lnNpi
</p>
<p> !
</p>
<p>&frac14; lnN �
X
</p>
<p>M
</p>
<p>i�1
pi lnNpi
</p>
<p>&frac14; lnN � lnN
X
</p>
<p>M
</p>
<p>i&frac14;1
pi �
</p>
<p>X
</p>
<p>M
</p>
<p>i&frac14;1
pi ln pi &frac14; 1�
</p>
<p>X
</p>
<p>M
</p>
<p>i&frac14;1
pi
</p>
<p> !
</p>
<p>lnN �
X
</p>
<p>M
</p>
<p>i&frac14;1
pi ln pi
</p>
<p>&frac14; �
X
</p>
<p>M
</p>
<p>i&frac14;1
pi ln pi (12.32)
</p>
<p>We showed in Sect. 12.13.1 that entropy is proportional to ln W, so that
</p>
<p>S / �
X
</p>
<p>M
</p>
<p>i&frac14;1
pi ln pi (12.33)
</p>
<p>Replacing pi by the more usual symbol for electron density, we write
</p>
<p>S / �
X
</p>
<p>M
</p>
<p>i&frac14;1
ri ln&eth;ri=ri&THORN; (12.34)
</p>
<p>where ri is the electron density in the ith grid cell and ri is a reference density. Themaximum value of S
</p>
<p>gives the most probable electron density distribution. Desirably [70], the reference density can
</p>
<p>incorporate constraints, such as the total number of electrons in the unit cell or reflections with
</p>
<p>known phase, or it may be taken as
P
</p>
<p>i ri. The maximum entropy method has been discussed in the
</p>
<p>literature [71, 72] and used successfully to solve structures [70]. The method has been programmed
</p>
<p>successfully in BayMEM [73, 74], which can be used for crystallographic applications of the
</p>
<p>maximum entropy method. It can derive the electron density in the unit cell from phased X-ray
</p>
<p>diffraction data. Different types of Bayesian prior probable densities are available.
</p>
<p>12.14 Log-Likelihood Gain in the Phase Problem
</p>
<p>In 1984, Bricogne, and Bricogne and co-workers published approaches to the solution of the phase
</p>
<p>problem that involved a combination of multi-solution, maximum entropy, and likelihood ranking
</p>
<p>techniques [75&ndash;79].
</p>
<p>12.14.1 Basis Set and Expansion of Reflections
</p>
<p>We start with the preliminary data and the extraction of intensities, as already discussed. The extracted
</p>
<p>intensities can be divided into an &ldquo;overlapped&rdquo; set and a &ldquo;non-overlapped&rdquo; set. For example, in the
</p>
<p>626 12 Powder Diffraction</p>
<p/>
</div>
<div class="page"><p/>
<p>structure determination of Mg3BN3 [80], only about 3% of the intensity was overlapped, whereas with
</p>
<p>1,3,4,6-tetrathiopentalene-2,5-dione [81], C4O2S4, about 60% of reflections were overlapped.
</p>
<p>An origin is set by assigning appropriate phases to a number of reflections, preferably of large jEj
values, according to the rules already discussed; thesen reflections constitute the starting, or basis, set {B}.
</p>
<p>The determination of jEj values requires the use of aWilson plot or aK-curve procedure, Sects. 4.2.1 and
8.2.1. The basis set is used as constraints in constructing a maximum entropy map m(x) [82], where
maximum entropy has been discussed in Sect. 12.13ff. The Fourier transform of m(x) reproduces the set
{B} and also estimates structure amplitudes and phases for m non-basis set reflections, {N}.
</p>
<p>Reflections from {N} are added to {B} with appropriate phases: 0 or p for centric reflections and
</p>
<p>
p/4, 
 3p/4 for acentric reflections, Sect. 8.2.9. Each phase combination creates a node: nc centric
reflections create 2nc nodes and acentric phases create 4na nodes. The nodes develop a phase tree in
</p>
<p>which the root node is the origin-fixing reflection set from which subsequent levels are built by phase
</p>
<p>permutations.
</p>
<p>12.14.2 Log-Likelihood Gain
</p>
<p>Following Bricogne [83], a log-likelihood gain (LLG) expression is formulated as
</p>
<p>LLG &frac14; LH� LH0 (12.35)
</p>
<p>where LH is the LLG function, namely, the probability that a given overlap of a set of Eh,0 values and
</p>
<p>the corresponding jEh,cj from maximum entropy calculations, and LH0 represents the null hypothesis,
that is, with zero number of calculated intensities. Table 12.7 shows some results from an LLG
</p>
<p>calculation. The inclusion of overlapped reflections generally leads to the best result. With four
</p>
<p>second-level node phases for the structure of lithium zirconium diphosphate [9], Li6Zr2O7, space
</p>
<p>group C2/c, there are 148 non-overlapped and 109 overlapped reflections. Node 135 generated a
</p>
<p>correct electron density map. Here, entropy is a poor figure of merit but LLG, including overlaps,
</p>
<p>indicates the correct solution well.
</p>
<p>12.14.3 Centroid Maps
</p>
<p>In maximum entropy calculations, centroid electron density maps [84], including overlapped reflec-
</p>
<p>tions, are most useful. These maps employ experimentally determined jEj values with phases from the
relevant tree node for the basis set reflections, whereas non-basis set reflections incorporate jEj
amplitudes generated from the jFj data set and phases generated from a maximum entropy process
and with Sim-type weighting.
</p>
<p>Table 12.7 LLG data for Li6Zr2O7
</p>
<p>Node Entropy LLG (non-overlap) LLG (all data)
</p>
<p>135 �2.03 7.10 15.1
109 �1.75 5.93 12.8
99 �1.87 4.89 11.3
97 �2.90 7.51 12.5
</p>
<p>12.14 Log-Likelihood Gain in the Phase Problem 627</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_4#Sec11_4">http://dx.doi.org/10.1007/978-1-4614-3954-7_4#Sec11_4</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_8#Sec3_8">http://dx.doi.org/10.1007/978-1-4614-3954-7_8#Sec3_8</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_8#Sec11_8">http://dx.doi.org/10.1007/978-1-4614-3954-7_8#Sec11_8</a></div>
</div>
<div class="page"><p/>
<p>12.15 Genetic Algorithms
</p>
<p>Another optimization technique, apart fromMonte Carlo and simulated annealing, is based on genetic
</p>
<p>algorithms [85, 86] which make use of the evolutionary strategy of Darwinian theory.
</p>
<p>The procedure follows a series of steps [87]:
</p>
<p>1. An individual random starting population, normally of a given size, is generated, and may be
</p>
<p>likened to a chromosome. The components are a collection of variables of the crystal structure, the
</p>
<p>genes xj, (j &frac14; 1, 2, . . . n).
2. The individual is characterized by its fitness, that is, the degree to which its calculated diffraction
</p>
<p>pattern agrees with the experimentally observed pattern.
</p>
<p>3. Then, by applying the genetic operators of crossover and mutation, new and improved structures
</p>
<p>are produced from a population and subjected to fitness tests.
</p>
<p>In a crossover, or mating, the chromosomes of two individuals, or parents, are cut, either singly
</p>
<p>or at multiple points, and the cut parts interchanged:
</p>
<p>Parent 1 Parent 2
</p>
<p>Initial state ABCDEF GHIJKL
</p>
<p>Crossover state ABCJKL GHIDEF
</p>
<p>Mutation can be carried out by assigning a new random value to one or more of the xi genes, or by
</p>
<p>inverting a variable through the origin.
</p>
<p>4. Natural selection is applied in the form of a fitness test: by the principle of the &ldquo;survival of the
</p>
<p>fittest,&rdquo; the population evolves towards a correct, or substantially correct structure, suitable for
</p>
<p>further refinement.
</p>
<p>The processes 1&ndash;4 represent one generation; further generation are executed by cycling back until a
</p>
<p>global minimum is found. In common with least-squares refinement and other minimization proce-
</p>
<p>dures, a local minimum may be encountered leading to a stagnation in the population. The way out is
</p>
<p>always the fitness test based on a comparison of Fo and jFcj. The genetic algorithm technique has been
programmed and has solved crystal structures successfully [88].
</p>
<p>12.16 Energy Minimization Techniques
</p>
<p>We have described a method of structure analysis by energy minimization in the context of single
</p>
<p>crystal studies in Sect. 9.11.3. It has been shown that a similar process can be successful with data
</p>
<p>obtained by powder methods [89]. The procedure with powder data can be summarized in the
</p>
<p>following stages:
</p>
<p>&bull; The experimental powder pattern is indexed and, where possible, the space group deduced
</p>
<p>&bull; A plausible structure is devised based on standard molecular geometry, with constraints dependent
</p>
<p>upon symmetry and upon packing parameters, such as the unit-cell dimensions, and positional and
</p>
<p>orientational disposition of the molecule within the asymmetric unit
</p>
<p>&bull; Possible crystal structures are calculated by minimization of the lattice energy of the postulated
</p>
<p>structure model, and powder patterns are calculated for the possible structures
</p>
<p>&bull; The better solutions are selected by comparison with the experimental powder pattern and then
</p>
<p>subjected to Rietveld refinement, so as to obtain a minimum lattice energy structure
</p>
<p>628 12 Powder Diffraction</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_9#Sec39_9">http://dx.doi.org/10.1007/978-1-4614-3954-7_9#Sec39_9</a></div>
</div>
<div class="page"><p/>
<p>12.17 Concluding Remarks
</p>
<p>The examples of structure determination discussed in this chapter demonstrate the feasibility of
</p>
<p>solving crystal structures from powder diffraction data. It is to be expected that the methods will
</p>
<p>become even more widely used in future, since it opens the way for the investigation of a wide
</p>
<p>spectrum of materials that have previously resisted detailed structural analysis. At the present time,
</p>
<p>however, it remains that the resolution obtained from powder diffraction is not yet as good as that
</p>
<p>obtained from single crystal studies.
</p>
<p>The totality of techniques has been well reviewed [90&ndash;92] (and Bibliography), and many programs
</p>
<p>and program systems have been devised that address both the individual stages and the complete
</p>
<p>process of solving crystal structures from powder data [32]. The serious powder analyst is strongly
</p>
<p>recommended to consult these literature sources. In addition, there are several important references
</p>
<p>relating to work that have not all been published elsewhere that can be accessed from web sites; they
</p>
<p>are listed in Table 12.8.
</p>
<p>As an example of the results obtainable with state-of-the-art equipment, we illustrate the
</p>
<p>high-resolution powder diffraction pattern of (E)-1-cyclopropyl-6-fluoro-4-oxo-7-(4-(3-oxo-3-phenyl-
</p>
<p>prop-1-enyl)piperazin-1-yl)-1,4-dihydroquinoline-3-carboxylic acid, Fig. 12.24 (R &frac14; C6H5). It was
taken at the Diamond Light Source synchrotron, which was discussed in Sect. 3.1.6, on beamline
</p>
<p>IO3with a wavelength of 0.9763 Å over a period of 1 s, and recorded on a Dectris Pilatus 6M large area
</p>
<p>pixel detector. The photograph is shown in Fig. 12.25.
</p>
<p>Table 12.8 Crystallographic data on web sites
</p>
<p>Topic Web site
</p>
<p>Structure determination from Powder
</p>
<p>Diffractometry Round Robin (SDPDRR)
</p>
<p>http://www.cristal.org/SDPDRR/index.html, http://www.ccp14.ac.
</p>
<p>uk/ccp/web-mirrors/armel/SDPDRR/index.html
</p>
<p>Optimum data collection strategy http://www.ccp14.ac.uk/solution/powder_data_collection.html,
</p>
<p>http://www.ccp.14.ac.uk/solution/gsas/convert_vct_data_to_gsas.
</p>
<p>html
</p>
<p>Discussion on variable count time (VCT)
</p>
<p>data collection
</p>
<p>http://www.ccp14.ac.uk/solution/vct/index.html
</p>
<p>Fig. 12.24 The molecular structure of (E)-1-cyclopropyl-6-fluoro-4-oxo-7-(4-(3-oxo-3-phenylprop-1-enyl)piper-
azin-1-yl)-1,4-dihydroquinoline-3-carboxylic acid (R &frac14; C6H5)
</p>
<p>12.17 Concluding Remarks 629</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec7_3">http://dx.doi.org/10.1007/978-1-4614-3954-7_3#Sec7_3</a></div>
<div class="annotation"><a href="http://www.cristal.org/SDPDRR/index.html">http://www.cristal.org/SDPDRR/index.html</a></div>
<div class="annotation"><a href="http://www.ccp14.ac.uk/ccp/web-mirrors/armel/SDPDRR/index.html">http://www.ccp14.ac.uk/ccp/web-mirrors/armel/SDPDRR/index.html</a></div>
<div class="annotation"><a href="http://www.ccp14.ac.uk/ccp/web-mirrors/armel/SDPDRR/index.html">http://www.ccp14.ac.uk/ccp/web-mirrors/armel/SDPDRR/index.html</a></div>
<div class="annotation"><a href="http://www.ccp14.ac.uk/solution/powder_data_collection.html">http://www.ccp14.ac.uk/solution/powder_data_collection.html</a></div>
<div class="annotation"><a href="http://www.ccp.14.ac.uk/solution/gsas/convert_vct_data_to_gsas.html">http://www.ccp.14.ac.uk/solution/gsas/convert_vct_data_to_gsas.html</a></div>
<div class="annotation"><a href="http://www.ccp.14.ac.uk/solution/gsas/convert_vct_data_to_gsas.html">http://www.ccp.14.ac.uk/solution/gsas/convert_vct_data_to_gsas.html</a></div>
<div class="annotation"><a href="http://www.ccp14.ac.uk/solution/vct/index.html">http://www.ccp14.ac.uk/solution/vct/index.html</a></div>
</div>
<div class="page"><p/>
<p>12.18 Problems
</p>
<p>12.1. A cylindrical powder camera has a radius of 57.30 mm. A given powder specimen is examined
</p>
<p>with Cu Ka radiation. At what value of the Bragg angle y would the a1a2 doublet begin to be
</p>
<p>resolved on the film, if the lines in that region of the film are of approximately 0.5 mm
</p>
<p>thickness? The a1 and a2 components have wavelengths 1.5405 Å and 1.5443 Å, respectively,
</p>
<p>and an intensity ratio a1:a2 &frac14; 2.
12.2. The following sequence of sin2 y values was measured for lines on a powder photograph of a
</p>
<p>cubic substance taken with Cu Ka radiation, l &frac14; 1.5148 Å. Determine (a) the unit-cell type, (b)
the indices of the lines, (c) a best value for the unit-cell dimension a. For (c), plot a, calculated
</p>
<p>for each line, against f &eth;y&THORN; &frac14; 1
2
&eth;cos2y= sin y&thorn; cos2y=y&THORN;, with y in radian, and extrapolate to f
</p>
<p>(y) &frac14; 0. This (Nelson&ndash;Riley) function tends to compensate for errors arising from absorption,
specimen eccentricity, camera radius, and beam-divergence.
</p>
<p>0:0465; 0:0635; 0:1717; 0:2486;
</p>
<p>0:3712; 0:4170; 0:5394; 0:5544;
</p>
<p>0:6609; 0:7368
</p>
<p>Fig. 12.25 The powder diffraction pattern of crystals of the compound in Fig. 12.24. The grid lines are a feature of the
detector system, and do not influence the measurement of intensities (Sample supplied by Professor Ali El Dissouky
</p>
<p>(Alexandria University). Powder record by courtesy of J. Nicholson and E. Shotton, Diamond Light Source, UK.)
</p>
<p>630 12 Powder Diffraction</p>
<p/>
</div>
<div class="page"><p/>
<p>12.3. Confirm the values of the parameters for the reduced unit cell ofmagnesium tungstate discussed in
</p>
<p>Sects. 12.3 and 12.3.1 with the program LEPAGE: use the default value of the collinearity
</p>
<p>parameter C, and also C &frac14; 0.5�.
12.4. The followingunit-cell parameterswerededuced froman indexedpowderphotograph:a &frac14; 8.515Å,
</p>
<p>b &frac14; 8.515 Å, c &frac14; 6.021 Å, a &frac14; 135.0�, b &frac14; 69.3�, g &frac14; 90.00�. Determine the reduced and
conventional unit cell, and list its parameters. What is the ratio of the volume of the conventional
</p>
<p>unit cell to that of the given unit cell?
</p>
<p>12.5. The following values (104Q) were obtained for the first 40 lines on an accurate powder
</p>
<p>diffraction record of the microcrystalline single substance X. Deduce by induction a possible
</p>
<p>unit cell. Remember that the first three lines need not necessarily correspond with a*, b*, and
</p>
<p>c*. Hence, index the lines on the basis of this unit cell. What can be said about the unit-cell
</p>
<p>type and space group for crystal X? Obtain the reduced and conventional unit cells, using
</p>
<p>the program LEPAGE. If the conventional unit cell is different from the unit cell first derived,
</p>
<p>transform the indices. What now are the conventional unit-cell type and the possible
</p>
<p>space group(s)?
</p>
<p>Line number 104Q Line number 104Q
</p>
<p>1 83.1 21 892.4
</p>
<p>2 810.1 22 916.0
</p>
<p>3 172.2 23 962.3
</p>
<p>4 2410.8 24 981.7
</p>
<p>5 332.6 25 9910.1
</p>
<p>6 356.1 26 1016.0
</p>
<p>7 416.0 27 1105.0
</p>
<p>8 421.5 28 11210.0
</p>
<p>9 4310.3 29 1134.0
</p>
<p>10 516.9 30 1248.0
</p>
<p>11 5510.8 31 12410.0
</p>
<p>12 642.9 32 1308.0
</p>
<p>13 648.6 33 1330.0
</p>
<p>14 683.3 34 1361.0
</p>
<p>15 688.8 35 13610.0
</p>
<p>16 732.1 36 1397.0
</p>
<p>17 748.4 37 14110.0
</p>
<p>18 801.5 38 1425.0
</p>
<p>19 837.2 39 1444.0
</p>
<p>20 884.5 40 1461.0
</p>
<p>12.6. In this problem and the next two, sets of data are provided to demonstrate the power and
</p>
<p>applicability of computer indexing, using the program ITO12. The first data set has been
</p>
<p>included with the suite of computer programs, because it is vital to set up the data in the
</p>
<p>prescribed format. For those familiar with FORTRAN, the fields for numerical input are
</p>
<p>F10.5; see also Sect. 13.7 for comments on ITO12. Crystal XL1: data are provided as values
</p>
<p>of 104Q; Q values are sometimes defined as 104=d2hkl. Use the program to determine the
unit-cell parameters and as much information as possible about the space group.
</p>
<p>12.18 Problems 631</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_13#Sec36_13">http://dx.doi.org/10.1007/978-1-4614-3954-7_13#Sec36_13</a></div>
</div>
<div class="page"><p/>
<p>12.7. Crystal XL2: data are provided as values of 104 Q. Use the program to determine the unit-cell
</p>
<p>parameters and as much information as possible about the space group.
</p>
<p>12.8. Crystal XL3: data are provided as values of 2y. Use the program to determine the unit-cell
</p>
<p>parameters and as much information as possible about the space group.
</p>
<p>Problems on structure-solving from power data are presented in Chapter 13, wherein ESPOIR
</p>
<p>is discussed further.
</p>
<p>References
</p>
<p>1. Hull AW (1917) Phys Rev 10:661
</p>
<p>2. Debye P, Scherrer P (1816) Phys Zeit 17:277
</p>
<p>3. http://www.icdd.com/education/xrd.htm
</p>
<p>4. Williamson GK, Hall WH (1953) Acta Metall 1:22
</p>
<p>5. Westgren A, Phragmén G (1928) Metallwirtschaft 7:700
</p>
<p>6. Vegard L (1916) Phil Mag 32:187
</p>
<p>7. Westgren A (1932) Angew Chem 45:33
</p>
<p>8. Bibliography, Dinnebier RE (ed)
</p>
<p>9. Bibliography, David WIF et al
</p>
<p>10. Harris KDM et al (2001) Angew Chem Int Edit 40:1626
</p>
<p>11. Harris KDM, Tremayne M (1996) Angew Chem Mater 8:2554
</p>
<p>12. http://www.xhuber.com
</p>
<p>13. Parrish W, Huang TC (1983) Adv X-ray Anal 26:35
</p>
<p>14. Morris RE et al (1994) J Solid State Chem 111:52
</p>
<p>15. Jeffrey GA et al (1987) Proc R Soc A 414:47
</p>
<p>16. Bibliography, David et al
</p>
<p>17. Bibliography, Young
</p>
<p>18. http://pilatus.web.psi.ch/mythen.htm
</p>
<p>361.0 459.4 475.8 701.7 717.5 968.9 1310.0 1312.0
</p>
<p>1059.0 1088.0 1363.0 1411.0 1428.0 1444.0 1653.0 1724.0
</p>
<p>1785.0 1838.0 1902.0 2039.0 2081.0 2145.0 2300.0 2332.0
</p>
<p>2380.0 2451.0 2485.0 2639.0 2656.0 2675.0 2807.0 2871.0
</p>
<p>996.0 3185.0 3212.0 3250.0 3265.0 3402.0 3507.0 3428.0
</p>
<p>311.5 364.2 442.0 675.4 877.8 1,020.3 1,065.2 1,111.0
</p>
<p>1,150.3 1,246.3 1,384.4 1,456.3 1,500.2 1,534.1 1,605.3 1,610.0
</p>
<p>1,767.6 1,773.4 1,839.5 1,954.9 2,045.3 2,157.1 2,228.7 2,242.5
</p>
<p>2,311.0 2,318.4 2,476.7 2,702.3 2,746.8 2,773.3 2,834.6 2,865.6
</p>
<p>2,925.3 3,003.0 3,019.4 3,062.3 3,145.7 3,198.0 3,236.9 3,276.7
</p>
<p>8.44 15.45 15.61 16.19 17.00 17.22 18.32 18.80
</p>
<p>19.02 20.71 22.20 22.29 23.98 24.92 25.14 25.76
</p>
<p>26.95 27.34 28.01 29.10 29.27 30.70 31.18 31.62
</p>
<p>32.04 32.49 32.83 33.30 33.60 34.67 34.87 35.45
</p>
<p>36.63 36.84 37.11 37.34 38.08 38.57 38.89 38.95
</p>
<p>632 12 Powder Diffraction</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_13">http://dx.doi.org/10.1007/978-1-4614-3954-7_13</a></div>
<div class="annotation"><a href="http://www.icdd.com/education/xrd.htm">http://www.icdd.com/education/xrd.htm</a></div>
<div class="annotation"><a href="http://www.xhuber.com">http://www.xhuber.com</a></div>
<div class="annotation"><a href="http://pilatus.web.psi.ch/mythen.htm">http://pilatus.web.psi.ch/mythen.htm</a></div>
</div>
<div class="page"><p/>
<p>19. http://www.dectris.ch/sites/mythen1k.html
</p>
<p>20. Bergamaschi A (2010) J Synchrotron Radiat 17:653
</p>
<p>21. Ito T (1949) Nature 164:755
</p>
<p>22. Křivý I, Gruber B (1976) Acta Crystallogr A32:297
</p>
<p>23. Niggli P (1928) Handbuch der Experimentalphysik, vol 7. Akademische Verlagsgesellschaft, Leipzig
</p>
<p>24. Le Page Y (1982) J Appl Crystallogr B15:255
</p>
<p>25. Spek AL (1988) J Appl Crystallogr 21:578
</p>
<p>26. http://www.ccp14.ac.uk/tutorial/crys/program/ito12.htm
</p>
<p>27. Shirley R (2002) Crysfire. http://www.ccp14.ac.uk
</p>
<p>28. Shirley R (2002) Powder indexing with Crysfire. International Union of Crystallography Commission on Powder
</p>
<p>Diffraction, Newsletter, July 2002
</p>
<p>29. Rietveld HM (1969) J Appl Crystallogr 2:65
</p>
<p>30. Young RA (ed) (1993) The Rietveld method. Oxford University Press, Oxford
</p>
<p>31. Young RA, Wiles DB (1982) J Appl Crystallogr 15:430
</p>
<p>32. McCusker LB et al (1999) J Appl Crystallogr 32:36
</p>
<p>33. Young RA (1993) The Rietveld method. Oxford University Press, Oxford
</p>
<p>34. Le Bail A (1988) Mat Res Bull 23:447
</p>
<p>35. Pawley GS (1981) J Appl Crystallogr 14:357
</p>
<p>36. http://www.ccp14.ac.uk
</p>
<p>37. Loopstra BO, Rietveld HM (1969) Acta Crystallogr 2:65
</p>
<p>38. Lightfoot P et al (1987) Inorg Chem 26:3544
</p>
<p>39. Cascarano G et al (1992) J Appl Crystallogr 25:310
</p>
<p>40. Estermann MA et al (1992) J Appl Crystallogr 25:539
</p>
<p>41. Cernik RJ et al (1991) J Appl Crystallogr 24:222
</p>
<p>42. Ladd MFC, Palmer RA (eds) (1980), Theory and practice of direct methods. In: Crystallography. Plenum Press,
</p>
<p>New York, and references therein
</p>
<p>43. Giacovazzo C (1977) Acta Crystallogr A33:933
</p>
<p>44. idem. ibid. A36:362 (1980)
</p>
<p>45. Burla MC et al (2003) J Appl Crystallogr 36:1103
</p>
<p>46. http://www.nanomegass.com/files/eldiff.pdf
</p>
<p>47. Masciocchi N et al (1994) J Am Chem Soc 116:7668
</p>
<p>48. Hill RJ, Madsen IC (1987) Powder Diffr 2:146
</p>
<p>49. Altomare A et al (2004) J Appl Crystallogr 37:1025; J Res Natl Stand Technol 109:125
</p>
<p>50. Brunelli M et al (2007) J Appl Crystallogr 40:702
</p>
<p>51. http://wwwba.ic.cnr.it/content/expo2009
</p>
<p>52. Grosse-Kunstleve RW et al (1997) J Appl Crystallogr 30:985
</p>
<p>53. Grosse-Kuntsleve RW (1996) PhD thesis, Dissertation ETH no. 11422, Z&euro;urich. http://cci.lbl.gov/~rwgk/Disserta-
tion/
</p>
<p>54. McCusker LB et al (1996) Micropor Mater 6:295
</p>
<p>55. Grosse-Kuntsleve RW et al (1999) J Appl Crystallogr 32:536
</p>
<p>56. Metropolis N et al (1957) J Chem Phys 21:1087
</p>
<p>57. David WIF et al (2001) DASH user manual. CCD, Cambridge, UK
</p>
<p>58. Allen MP, Tildesley DJ (1987) Computer simulation in liquids. Oxford University Press, Oxford
</p>
<p>59. Tremayne M et al (1996) J Mater Chem 6:1601
</p>
<p>60. Harris KDM et al (1994) J Am Chem Soc 116:3543
</p>
<p>61. Harris KDM, Tremayne M (1996) Chem Med 8:2554
</p>
<p>62. Engel GE et al (2001) Mol Cryst Liq Cryst 356:355
</p>
<p>63. Andreev YG, Bruce PG (1998) J Chem Soc Dalton Trans 4071
</p>
<p>64. David WIF et al (1998) Chem Commun 931
</p>
<p>65. http://www.ccdc.cam.ac.uk/products/powder_diffrac tion/dash/
</p>
<p>66. Le Bail A (2000) Seventh European powder diffraction conference, Barcelona. http://www.cristal.org/sdpd/espoir/
</p>
<p>67. Laligant Y et al (2001) J Solid State Chem 159:223
</p>
<p>68. von Dreele RB et al (2000) Acta Crystallogr D56:1549
</p>
<p>69. Margiolaki I, Wright JP (2008) Acta Crystallogr A64:169
</p>
<p>70. Bagautdinov B et al (1998) Acta Crystallogr B54:626
</p>
<p>71. Gilmore C (1996) Acta Crystallogr A52:561
</p>
<p>72. Sivia D, David W (1994) Acta Crystallogr A50:703
</p>
<p>73. http://www.crystal.uni-bayreuth.de/en/baymem/index.html
</p>
<p>74. van Smaalen S et al (2003) Acta Crystallogr A59:459
</p>
<p>References 633</p>
<p/>
<div class="annotation"><a href="http://www.dectris.ch/sites/mythen1k.html">http://www.dectris.ch/sites/mythen1k.html</a></div>
<div class="annotation"><a href="http://www.ccp14.ac.uk/tutorial/crys/program/ito12.htm">http://www.ccp14.ac.uk/tutorial/crys/program/ito12.htm</a></div>
<div class="annotation"><a href="http://www.ccp14.ac.uk">http://www.ccp14.ac.uk</a></div>
<div class="annotation"><a href="http://www.ccp14.ac.uk">http://www.ccp14.ac.uk</a></div>
<div class="annotation"><a href="http://www.nanomegass.com/files/eldiff.pdf">http://www.nanomegass.com/files/eldiff.pdf</a></div>
<div class="annotation"><a href="http://wwwba.ic.cnr.it/content/expo2009">http://wwwba.ic.cnr.it/content/expo2009</a></div>
<div class="annotation"><a href="http://cci.lbl.gov/~rwgk/Dissertation/">http://cci.lbl.gov/~rwgk/Dissertation/</a></div>
<div class="annotation"><a href="http://cci.lbl.gov/~rwgk/Dissertation/">http://cci.lbl.gov/~rwgk/Dissertation/</a></div>
<div class="annotation"><a href="http://www.ccdc.cam.ac.uk/products/powder_diffraction/dash/">http://www.ccdc.cam.ac.uk/products/powder_diffraction/dash/</a></div>
<div class="annotation"><a href="http://www.cristal.org/sdpd/espoir/">http://www.cristal.org/sdpd/espoir/</a></div>
<div class="annotation"><a href="http://www.crystal.uni-bayreuth.de/en/baymem/index.html">http://www.crystal.uni-bayreuth.de/en/baymem/index.html</a></div>
</div>
<div class="page"><p/>
<p>75. Bricogne G (1984) Acta Crystallogr A40:410
</p>
<p>76. idem. ibid. A44:517 (1988)
</p>
<p>77. Bricogne G et al (1992) J Mater Chem 2:1301
</p>
<p>78. Bannister C et al (1989) Bayesian methods and maximum entropy. In: Skilling J (ed) Kluwer, Dordrecht
</p>
<p>79. Gilmore CJ et al (1990) Acta Crystallogr A46:284
</p>
<p>80. Hiraguchi H et al (1991) J Appl Crystallogr 24:286
</p>
<p>81. Gilmore CJ et al (1993) Proc Math Phys Soc A442:97
</p>
<p>82. Sakata M, Sato M (1990) Acta Crystallogr A46:263
</p>
<p>83. http://erice2005.docking.org/vcourse/19thu/1145-Bri cogne/Bricogne.pdf
</p>
<p>84. Gilmore CJ et al (1991) Acta Crystallogr A47:830
</p>
<p>85. Kariuki BM et al (1999) J Chem Soc Chem Commun 1677
</p>
<p>86. Shankland K et al (1998) Int J Pharmacol 165:117
</p>
<p>87. Polak E (1971) Computational methods in optimization. Math Sci Eng 77
</p>
<p>88. Johnston RJ et al (1997) The algorithm GAPSS. University of Birmingham, UK, and references therein
</p>
<p>89. http://www.usc.es/export/sites/default/gl/investigacion/riaidt/raiosx/descargas/manuais/ and references therein
</p>
<p>90. http://sdpd.univ-lemans.fr/egypt-2004/
</p>
<p>91. http://www.cristal.org/rapport/LeBail_promo2004_ prod.pdf
</p>
<p>92. http://lachlan.bluehaze.com.au/cpdnews98/art18.htm, and references therein
</p>
<p>93. http://www.ccp14.ac.uk
</p>
<p>94. Bragg WL (1949) The Crystalline State, Vol 1, Bell, London
</p>
<p>Bibliography
</p>
<p>David WIF, Shankland K, McCusker LB, Baerlocher C (eds) (2002) Structure determination from powder diffraction
</p>
<p>data. International Union of Crystallography, Oxford University Press, Oxford
</p>
<p>Dinnebier RE, Billinge SJL (eds) (2008) Powder diffraction: theory and practice. RSC Publishing, Cambridge
</p>
<p>Pecharsky VK, Zavalij PY (2005) Fundamentals of powder diffraction. Springer, Berlin
</p>
<p>Peschar R (1990) Molecular structure solution. Procedures 3:59
</p>
<p>Prince E (1985) In: Structure and statistics in crystallography. Wilson AJC (ed) Adenine Press
</p>
<p>Schenk H (1982) In: Computational crystallography. Sayre D (ed) Oxford University Press, Oxford
</p>
<p>Young RA (1993) The Rietveld method. International Union of Crystallography. Oxford University Press, Oxford
</p>
<p>634 12 Powder Diffraction</p>
<p/>
<div class="annotation"><a href="http://erice2005.docking.org/vcourse/19thu/1145-Bricogne/Bricogne.pdf">http://erice2005.docking.org/vcourse/19thu/1145-Bricogne/Bricogne.pdf</a></div>
<div class="annotation"><a href="http://www.usc.es/export/sites/default/gl/investigacion/riaidt/raiosx/descargas/manuais/">http://www.usc.es/export/sites/default/gl/investigacion/riaidt/raiosx/descargas/manuais/</a></div>
<div class="annotation"><a href="http://sdpd.univ-lemans.fr/egypt-2004/">http://sdpd.univ-lemans.fr/egypt-2004/</a></div>
<div class="annotation"><a href="http://www.cristal.org/rapport/LeBail_promo2004_prod.pdf">http://www.cristal.org/rapport/LeBail_promo2004_prod.pdf</a></div>
<div class="annotation"><a href="http://lachlan.bluehaze.com.au/cpdnews98/art18.htm">http://lachlan.bluehaze.com.au/cpdnews98/art18.htm</a></div>
<div class="annotation"><a href="http://www.ccp14.ac.uk">http://www.ccp14.ac.uk</a></div>
</div>
<div class="page"><p/>
<p>Computer-Aided Crystallography 13
</p>
<p>13.1 Introduction
</p>
<p>This chapter has been designed to further the knowledge gained from a study of the earlier chapters of
</p>
<p>this book. The computer programs that are supplied as theWeb Program Packages and described here
</p>
<p>are complementary to that work, and enable the reader to gain practical experience of concepts and
</p>
<p>methods germane to X-ray structure analysis. While these stand-alone programs are provided for
</p>
<p>quick and easy access for problem solving within the context of this book, we emphasize that the
</p>
<p>serious structure analyst must also refer to the other important program systems readily available.
</p>
<p>13.1.1 Collaborative Computational Projects
</p>
<p>The Collaborative Computational Projects Number 4, for Macromolecular Crystallography (CCP4),
</p>
<p>and Number 14 for Powder and Small-Molecule Single-Crystal Diffraction (CCP14), aim to collect
</p>
<p>and support the best and most commonly used programs for crystal structure determination by single-
</p>
<p>crystal and powder techniques. The software is located on web sites [1, 2] and is freely available to
</p>
<p>workers in the academic and research communities, and the wide dissemination of new ideas,
</p>
<p>techniques, and practice is encouraged.
</p>
<p>Computing is an essential feature in any modern X-ray crystallographic investigation. Here, we can
</p>
<p>provide only a flavor of what is available, but enough, we hope, to demonstrate the great importance of
</p>
<p>an intelligent application of computational methods to this subject. The program suite provided here
</p>
<p>has the following functions:
</p>
<p>&bull; To study the derivation of point groups;
</p>
<p>&bull; To carry out systematic point-group recognition using crystal or molecular models;
</p>
<p>&bull; To simulate the procedures and calculations involved in the determination of crystal structures by
</p>
<p>X-ray diffraction data from both single-crystal and powder specimens.
</p>
<p>&bull; To carry out various other related computations.
</p>
<p>On the basis of a familiarity with these programs, it should be only a small step to proceed, when
</p>
<p>needed, with the more comprehensive and detailed crystallographic software to which we have
</p>
<p>referred here and in earlier chapters.
</p>
<p>The Web Program Packages can be accessed under the web site reference http://extras.springer.
com. The programs are very straightforward to use, and it is recommended that the complete suite of
</p>
<p>M. Ladd and R. Palmer, Structure Determination by X-ray Crystallography:
</p>
<p>Analysis by X-rays and Neutrons, DOI 10.1007/978-1-4614-3954-7_13,
# Springer Science+Business Media New York 2013
</p>
<p>635</p>
<p/>
<div class="annotation"><a href="http://extras.springer.com">http://extras.springer.com</a></div>
<div class="annotation"><a href="http://extras.springer.com">http://extras.springer.com</a></div>
</div>
<div class="page"><p/>
<p>programs, together with the data files supplied, should be downloaded to a folder in the PC, according
</p>
<p>to instructions given on the web site by the publisher (the ISBN number of the book will be needed).
</p>
<p>The programs may be amended from time to time, as improvements and additions are applied to
</p>
<p>them. The first set with this fifth edition will be dated 1 January 2012 (Version 5.1), so that any date
</p>
<p>after that implies a revision or addition to those programs, and will be so notified on the web site.
</p>
<p>However, the programs ITO12, ESPOIR, and LEPAGE will not be subject to such changes unless
</p>
<p>they are first revised by their own authors.
</p>
<p>13.1.2 Structure of the Web Program Packages
</p>
<p>The companion program suite comprises four folders: XRSYST contains the programs for the single-
</p>
<p>crystal techniques, PDSYST contains the programs for powder techniques, POWDER contains a single
</p>
<p>program system (ESPOIR) also for powder data, and GNSYST contains all other general programs
</p>
<p>referred to in the text and problem sections of the book; all the folders contain data as necessary. There
</p>
<p>are also six GINO files present, concerned with plotting Fourier maps: they must remain unaltered.
</p>
<p>The structure of the program suite is illustrated in Fig. 13.1. In general, the programs, which are
</p>
<p>provided as IBM-compatible .EXE files, are opened by a double click on the program name or icon. In
</p>
<p>some cases, it can be helpful to open the programs in a Command Prompt window and enlarge the
</p>
<p>screen. We now describe the different programs and their uses in detail, with the aid of examples.
</p>
<p>13.2 Derivation of Point Groups (EULR)
</p>
<p>In Sect. 1.4ff we discussed the symmetry operations R and �R (R &frac14; 1, 2, 3, 4, 6), taken singly or in
combinations, and gave stereogram representations of them in Fig. 1.32 of that chapter. The first
</p>
<p>nontrivial combinations of symmetry operations follow from combining R with �1, and it is easy to
</p>
<p>show, with the aid of stereograms, that R and �1 together lead to R/m for R &frac14; 2, 4, and 6; for R &frac14; 1
</p>
<p>&darr;
&darr; &darr; &darr; &darr;
</p>
<p>&darr; &darr; &darr; &darr;
</p>
<p>PROGRAM
</p>
<p>SUITE
</p>
<p>XRSYST PDSYST POWDER GNSYS
</p>
<p>XRAYa
</p>
<p>(AND GINO
</p>
<p>        FILES)
</p>
<p>MAKDAT
</p>
<p>+
</p>
<p>Data Sets
</p>
<p>ITO12
</p>
<p>LEPAGE
</p>
<p>Q-VALS
</p>
<p>RECIP
</p>
<p>+
</p>
<p>Data Sets
</p>
<p>ESPOIR
</p>
<p>RASMOLb
</p>
<p>+
</p>
<p>Data Sets
</p>
<p>EULR
</p>
<p>FOUR1D
</p>
<p>FOUR2D
</p>
<p>INTXYZ
</p>
<p>LSLI
</p>
<p>MATOPS
</p>
<p>MOLGOM
</p>
<p>SQUARE
</p>
<p>SYMM
</p>
<p>TRANS1
</p>
<p>ZONE
</p>
<p>+
</p>
<p>Data Sets
</p>
<p>Fig. 13.1 Structure of the
program packages on web
</p>
<p>site http://extras.springer.
</p>
<p>com (the IBSN number of
</p>
<p>the book will be needed
</p>
<p>here). aThere are also six
</p>
<p>files (five .DLL and one .
</p>
<p>CON) present concerned
</p>
<p>with plotting that must
</p>
<p>remain unaltered. bThe
</p>
<p>RASMOL file is concerned
</p>
<p>with drawing. There are
</p>
<p>also two other data files in
</p>
<p>POWDER for each
</p>
<p>substance that must remain
</p>
<p>unaltered: one with crystal
</p>
<p>data (.DAT) and one with
</p>
<p>reflection data (.HKL)
</p>
<p>636 13 Computer-Aided Crystallography</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Fig32_1">http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Fig32_1</a></div>
<div class="annotation"><a href="http://extras.springer.com">http://extras.springer.com</a></div>
<div class="annotation"><a href="http://extras.springer.com">http://extras.springer.com</a></div>
</div>
<div class="page"><p/>
<p>and 3, point groups �R already include �1 as an operation in the group. Thus, we have quickly derived 13
</p>
<p>crystallographic point groups: 1, 2, 3, 4, 6, �1, m ( � �2), �3, �4, �6, 2/m, 4/m, and 6/m.
</p>
<p>The point groups that contain more than one symmetry operator display the essence of Euler&rsquo;s
</p>
<p>theorem on the combination of rotations. We have used this theorem implicitly in Sect. 1.4.2,
</p>
<p>explicitly in Sect. 2.8ff, and we may state it formally as
</p>
<p>R2R1 � R3 (13.1)
</p>
<p>which means that, from a given situation, symmetry operation R1 followed by operation R2 is
</p>
<p>equivalent to operation R3 applied to the original situation, and we know from the definition of
</p>
<p>point group that the three symmetry operations have at least one point in common. We saw this
</p>
<p>theorem in operation in the examples of point groups mm2 and 4mm, Figs. 1.28 and 1.29, and it is the
</p>
<p>basis of a procedure for determining the remainder of the 32 crystallographic point groups. In general,
</p>
<p>the order of carrying out the symmetry operations is important, although the result is not affected with
</p>
<p>symmetry operations of degree 2 or less.
</p>
<p>We next combine the operations R with another symmetry, operation, say 2, and we need to know
</p>
<p>immediately the relative orientations of the rotation axes R and 2 that we use symbolically to
</p>
<p>represent these operations. Are they perpendicular to each other or even coincident, and are there
</p>
<p>other possibilities to consider?
</p>
<p>The program EULR, opened by a double click on the file name, has been devised to follow the
</p>
<p>steps of the derivation of point groups described elsewhere, for example, in the references that appear
</p>
<p>on the monitor screen when this program is opened. This program is not interactive, but it shows how
</p>
<p>Euler&rsquo;s theorem can be used with the combinations of operations 2 and the permitted values of R to
</p>
<p>develop six sets of orientations of rotation axes.
</p>
<p>Then, independently of the program, consider replacing two of the rotation axes in each of the six
</p>
<p>sets by inversion axes. Why not just one of the rotation axes, or all three of them?
</p>
<p>As a final step, we must consider if any new point groups are obtained by incorporating a center of
</p>
<p>symmetry into any point group where one is not already present. An extension of the program caters
</p>
<p>for certain non-crystallographic point groups that are encountered in studying the symmetry of simple
</p>
<p>molecules.
</p>
<p>13.3 Point-Group Recognition (SYMM)
</p>
<p>There are several ways in which one can approach systematically the recognition of the point group of
</p>
<p>a crystal or a molecular model. In the method used here [3], molecules and crystals are divided into
</p>
<p>four symmetry types, Table 13.1, dependent upon the presence of a center of symmetry and one
</p>
<p>mirror plane or more, or a center of symmetry alone, or one mirror plane or more but no center of
</p>
<p>Table 13.1 Crystallographic point groups typed by m and/or �1 or neither
</p>
<p>Neither m nor �1 Only m Only �1 Both m and �1
</p>
<p>1, 2, 3, 4, �4, 6 m, mm2, 3m �1, �3 2
</p>
<p>m
, mmm, �3m
</p>
<p>222, 32, 422 4mm, �42m 4
</p>
<p>m
,
4
</p>
<p>m
mm
</p>
<p>622, 23, 432 �6, 6mm 6
</p>
<p>m
,
6
</p>
<p>m
mm
</p>
<p>�6m2, �43m m�3, m�3m
</p>
<p>13.3 Point-Group Recognition (SYMM) 637</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Fig28_1">http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Fig28_1</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Fig29_1">http://dx.doi.org/10.1007/978-1-4614-3954-7_1#Fig29_1</a></div>
</div>
<div class="page"><p/>
<p>symmetry, or neither of these symmetry elements; hence, the first step in the scheme is a search for
</p>
<p>these elements.
</p>
<p>In order to demonstrate the presence of a center of symmetry, place the given model in any
</p>
<p>orientation on a flat surface; then, if either the plane through the uppermost atoms (for a chemical
</p>
<p>species), or the uppermost face (for a crystal), is parallel to the plane surface supporting the model
</p>
<p>and the two planes in question are both equivalent and inverted across the center of the model, then a
</p>
<p>center of symmetry is present.
</p>
<p>If a mirror plane is present, it divides the model into enantiomorphic (right-hand/left-hand) halves.
</p>
<p>A correct identification of these symmetry elements at this stage places the model into one of the four
</p>
<p>types listed in Table 13.1.
</p>
<p>The reader may care to examine a cube or a model of the SF6 molecule, which shows both a center
</p>
<p>of symmetry and mirror planes, and a tetrahedron or a model of the CH4 molecule, which shows
</p>
<p>mirror planes but no center of symmetry. Models of a cube and a tetrahedron may be constructed
</p>
<p>easily:
</p>
<p>Cube
</p>
<p>Draw a square of side, say 40 mm, on a thin card. On each side of this square draw another identical
</p>
<p>square. Lightly score the edges of the first square and fold the other four to form five faces of a cube,
</p>
<p>and fasten with &ldquo;Selotape.&rdquo; There is an advantage in leaving the sixth face of the cube open, as we
</p>
<p>shall see, but we shall imagine its presence when needed.
</p>
<p>Tetrahedron
</p>
<p>On similar card, draw an equilateral triangle of side ca. 39
ffiffiffi
2
</p>
<p>p
mm. On each side of the triangle, draw
</p>
<p>another identical triangle. Lightly score the edges of the first triangle, fold the other three triangles in
</p>
<p>the same sense to meet at an apex, and fasten with &ldquo;Selotape.&rdquo;
</p>
<p>Note that on placing the tetrahedron inside the cube, it will be found that an edge of the tetrahedron
</p>
<p>is a face diagonal of the cube, thus aligning the symmetry elements common to both models.
</p>
<p>If these models are to be used with the point-group recognition program, allocate model numbers
</p>
<p>7 and 19 for the cube (or SF6) and tetrahedron (or CH4), respectively. The identification of the point
</p>
<p>group of a model then proceeds along the lines indicated by the block diagram of Fig. 13.2, on which
</p>
<p>the program SYMM is based [4].
</p>
<p>After assigning the model to one of the four &ldquo;types,&rdquo; the principal rotation axis, the rotation axis of
</p>
<p>highest degree, is identified, together with the number of such axes if more than one, the presence and
</p>
<p>orientations of mirror planes, twofold rotation axes, and so on.
</p>
<p>The program SYMM is interactive and the directions on the monitor screen should be followed.
</p>
<p>If an incorrect response is given during a path through the program, the user will be returned to that
</p>
<p>question in the program where the error occurred, for an alternative response to be made. Two such
</p>
<p>returns are allowed before the program rejects that particular examination for a further preliminary
</p>
<p>appraisal.
</p>
<p>It is necessary that the crystal and molecular models to be used are allocated a model number
</p>
<p>appropriate to their symmetry, and Table 13.2 provides the necessary key, based on Krantz wood
</p>
<p>crystal models, together with molecular examples or possible molecular examples of the point
</p>
<p>groups; a set of solid crystal models, appropriately numbered, is equally satisfactory.
</p>
<p>It will be realized that some of the molecules listed are not rigid bodies, and will show the
</p>
<p>required symmetry only if their functional groups are orientated correctly. The program responds to
</p>
<p>the non-crystallographic point groups 1m and 1=m only. The 1 symbol is replaced by the word
infinity in the program description itself, but zero is used to input 1 to the program when asked for
the point-group symbol.
</p>
<p>638 13 Computer-Aided Crystallography</p>
<p/>
</div>
<div class="page"><p/>
<p>F
ig
.
1
3
.2
</p>
<p>F
lo
w
d
ia
g
ra
m
</p>
<p>fo
r
th
e
p
o
in
t-
g
ro
u
p
re
co
g
n
it
io
n</p>
<p/>
</div>
<div class="page"><p/>
<p>13.4 Structure Determination Simulation (XRAY)
</p>
<p>The purpose of the XRAY program package is to facilitate an understanding of the practical
</p>
<p>applications of techniques of structure determination by single-crystal X-ray diffraction that we
</p>
<p>have discussed in the earlier chapters, albeit here in two dimensions. Example structures have been
</p>
<p>selected that give, in projection, results that are readily interpretable in terms of chemical structures.
</p>
<p>It is all too easy, we believe, to use modern, sophisticated structure-solving packages without really
</p>
<p>understanding the nature of the calculations taking place within them. In particular, for reasons that we
</p>
<p>have discussed in Sect. 11.7.1, very subtle difficulties are sometimes encountered.
</p>
<p>A subsidiary program, MAKDAT, enables sets of primary data to be constructed in the precise
</p>
<p>format required by XRAY. Alternatively, one may choose to prepare a data set independently, in which
</p>
<p>case the layout of data shown by the example sets provided should be followed exactly.
</p>
<p>It is stressed that the program system does not teach the subject of structure determination. Rather,
</p>
<p>it provides the basic concepts with a medium for their exploration, and so relates closely to techniques
</p>
<p>Table 13.2 Point groups and model numbers for the program SYMM with molecular examples
or possible examples
</p>
<p>Point group Model number/s Example or possible example
</p>
<p>Crystallographic point groups
1 91 CHBrClF, monochlorofluoromethane
2 77 H2O2, hydrogen peroxide
3 84, 93 H3PO4, phosphoric acid
4 85, 94 (CH3)4C4, tetramethylcyclobutadiene
6 88, 97 C6&eth;CH3&THORN;�6, hexamethylcyclohexadienyl
�1 78, 79 C6H5CH2CH2C6H5, dibenzyl
�3 48 [Ni(NO2)6]
</p>
<p>4�, hexanitronickel(II) ion
�4 86, 95 [H2+PO4]
</p>
<p>�, dihydrogen phosphate ion
�6 89, 98 C3H3N3(N3)3, 2,4,6-triazidotriazine
2/m 68&ndash;70, 72&ndash;75, 80 CHCl&frac14;CHCl, trans-1,2-dichloroethene
4/m 56 [Ni(CN)4]
</p>
<p>2�, tetracyanonickel(II) ion
6/m 37 C6(CH3)6, hexamethylbenzene
m &eth;�2&THORN; 83, 92, 99 C6H5Cl3, 1,2,4-trichlorobenzene
mm2 16, 64, 71, 76 C6H5Cl, chlorobenzene
3m 42 CHCl3, trichloromethane
4mm 87, 96, 100 [SbF5]
</p>
<p>2�, pentafluoroantimony(III) ion
6mm 81 C6(CH2Cl)6, hexa (chloromethyl) benzene
222 67 C8H12,cycloocta-1,5-diene
32 43, 47 [S2O6]
</p>
<p>2�, dithionate ion
422 55 Co(H2O)4Cl2, tetraaquodichlorocobalt
622 36 C6(NH2)6, hexaminobenzene
mmm 59&ndash;63, 65, 66 C6H4Cl2, 1,4-dichlorobenzene
�6m2 44&ndash;46, 90 [CO3]
</p>
<p>2�, carbonate ion
4/m mm 49&ndash;54 [AuBr4]
</p>
<p>�, tetrabromogold(III) ion
6/m mm 29&ndash;35 C6H6, benzene
�42m 57, 58 ThBr4, thorium tetrabromide
�3m 38&ndash;41 C6H12, chair-cyclohexane
23 27 C(CH3)4, 2,2-dimethylpropane
m�3 22&ndash;25 [Co(NO2)6]
</p>
<p>3�, hexanitrocobalt(III) ion
�43m 17&ndash;21, 28 CH4, methane
432 26 C8(CH3)8, octamethylcubane
m�3m 1&ndash;15 SF6, sulfur hexafluoride
</p>
<p>Non-crystallographic point groups
�82m &ndash; S8, sulfur
5 &ndash; C5(CH3&THORN;
</p>
<p>�
5, pentamethylcyclopentadienyl
</p>
<p>5m &ndash; C5H5NiNO, nitrosylcyclopentadienylnickel
�1�0m2 &ndash; (C5H5)2Re, bis-cyclopentadienylruthenium
�5m &ndash; (C5H5)2Fe, bis-cyclopentadienyliron
1m 101 HCl, hydrogen chloride
1=m 102 CO2, carbon dioxide
</p>
<p>640 13 Computer-Aided Crystallography</p>
<p/>
</div>
<div class="page"><p/>
<p>that are in use today. The program is interactive, and the messages on the monitor screen indicate the
</p>
<p>steps to be carried out by the user. Nevertheless, it will be useful to discuss here some of the features
</p>
<p>and capabilities of the system.
</p>
<p>Each data set contains information about the crystal unit cell, crystal symmetry in the appropriate
</p>
<p>projection, wavelength of radiation used, a set of Fo(hk0) data, and other information required for the
</p>
<p>calculation of electron density maps, structure factors, and jEj values. As the procedures are two-
dimensional, the symbols a, b, g, h, k, x, y, and so on are employed. Where the projection is other than
</p>
<p>that on (001), appropriate adjustments are made in setting up the primary data file; thus, true y and z
</p>
<p>become x and y to the program.
</p>
<p>The following basic procedures may be carried out within the XRAY program package:
</p>
<p>&bull; Patterson function, normal and sharpened
</p>
<p>&bull; Superposition (minimum) function
</p>
<p>&bull; Structure factor calculation
</p>
<p>&bull; Least-squares refinement
</p>
<p>&bull; Electron density function
</p>
<p>&bull; Direct methods: calculation of jEj values
&bull; Direct methods: calculation of E map
</p>
<p>&bull; Distances and angles calculation
</p>
<p>&bull; Scale and temperature factors by Wilson&rsquo;s method
</p>
<p>&bull; jEj values calculated from the structure
The program is executed in the usual manner. Several PAUSE situations occur throughout the
</p>
<p>routines in the system, so that material on the screen may be read. Continuation is effected by just
</p>
<p>depressing the Enter key. All primary data file names must be four-letter words plus the suffix .TXT,
</p>
<p>for example, NIOP.TXT, corresponding to a nickel o-phenanthroline complex, although only the first
</p>
<p>four letters of the name, NIOP, are used to call the data set; the program checks for the suffix .TXT. A
</p>
<p>name for the output results file, say NOUT, is entered at the keyboard. In some routines, such as the
</p>
<p>calculation of jEj values, other output files are organized, with appropriate messages to the screen.
Certain other files pertaining to coordinates are created at an appropriate stage, and their significance
</p>
<p>will become clear later. The particular calculations available are then listed on the monitor screen,
</p>
<p>and we shall give a brief description of each routine in turn.
</p>
<p>13.4.1 Patterson Function
</p>
<p>The Patterson function P(uv) may be calculated using either F2o or a &ldquo;sharpened&rdquo; mode, in which the
</p>
<p>coefficients used are &eth;jEj2 � 1&THORN; values, thus providing a sharpened, origin-removed Patterson map.
For the sharpened Patterson, jEj values must be first calculated, and the program so directs. Then, the
system returns to the Patterson routine. Sharpening generally introduces a small number of spurious
</p>
<p>maxima, and it is always useful to compare sharpened and unsharpened maps.
</p>
<p>The Patterson and electron density maps may be viewed as contoured maps on the screen, which
</p>
<p>enhances the interactive nature of the program system. It is possible, for example, to determine the
</p>
<p>coordinates of the heavy atom directly from the map on the screen, then go to the structure factor
</p>
<p>routine and input the heavy atom x, y coordinates, so as to obtain partial phase information, and
</p>
<p>thence calculate a first electron density map. It should be possible then to recognize other atom
</p>
<p>peaks from this first Fourier map, so beginning the process of structure determination by successive
</p>
<p>Fourier synthesis.
</p>
<p>At some stage it may be helpful to print and contour a Patterson or electron density figure field, so
</p>
<p>as to get a clearer picture of the projection, particularly where contours below the relative level 10 are
</p>
<p>13.4 Structure Determination Simulation (XRAY) 641</p>
<p/>
</div>
<div class="page"><p/>
<p>involved. All Fourier maps are scaled to a maximum of 100, and 10 is the lowest level that is
</p>
<p>contoured on the screen.
</p>
<p>It should be noted that, if the WINDOWS &ldquo;screen saver&rdquo; comes into operation while a plot is on
</p>
<p>the screen, then owing to some delay, the cross-wires might become fixed in position. On reactivating
</p>
<p>the plot, a second, mobile cross-wires may appear. To avoid these events, the screen saver may be
</p>
<p>deactivated.
</p>
<p>13.4.2 Superposition Function
</p>
<p>This routine calculates a minimum function M(x, y) at each grid point of the projection, Sect. 7.4.6:
</p>
<p>M&eth;x; y&THORN; &frac14; Minimum of fP&eth;x&thorn; Du1; y&thorn;Dv1&THORN;;P&eth;x&thorn; Du2; y&thorn; Dv2&THORN;; . . . ;P&eth;x&thorn; Dun; y&thorn; Dvn&THORN;g where
Du1;Dv1;Du2;Dv2; :::Dun;Dvn represent n displacement vectors, which may indicate either atom
positions or a set of vectors, including 0; 0, obtained from a partial interpretation of a Patterson map;
</p>
<p>symmetry-related positions should be entered. The grid points x and y are determined by the values
</p>
<p>set in the primary data. The minimum function, if successful, should indicate atomic positions that
</p>
<p>can then be used as discussed above.
</p>
<p>13.4.3 Structure Factor Calculation
</p>
<p>Each atom contributing to the structure factor calculation requires the following data:
</p>
<p>1. Atom type identity number: a list is given at the start of the routine.
</p>
<p>2. Fractional x and y coordinates of the atoms.
</p>
<p>3. Population parameter: 1, unless the atom is in a special position.
</p>
<p>4. An overall (isotropic) temperature factor is given initially from the primary data; it may be altered
</p>
<p>by routine 10. In either case, the value will be allocated to each atom, in preparation for subsequent
</p>
<p>refinement.
</p>
<p>In the calculation of structure factors, the coordinates may be entered either from a file or at the
</p>
<p>keyboard. If entered at the keyboard, the final line must be END. If entered from a file, the file must be
</p>
<p>named XYS.TXT and the first line must be the number of atoms to follow. The output from this
</p>
<p>routine is self-explanatory, and after an jFcj calculation and least-squares refinement, the current
coordinates are retained in the file COORDS.TXT. This file may be invoked in a subsequent
</p>
<p>calculation, or edited as desired, before being used as input data. The file XYS.TXT remains
</p>
<p>unaltered by the program.
</p>
<p>13.4.4 Least-Squares Refinement
</p>
<p>The routine uses the atoms already located. It may be just the heavy atoms, or a number of other atoms
</p>
<p>may be included as well. The routine uses the diagonal least-squares approximation to the ideal full-
</p>
<p>matrix procedure: it is fast, and satisfactory for emphasizing the principles involved. The x and y
</p>
<p>coordinates and the isotropic temperature factor B (initially the same for each atom, from the primary
</p>
<p>data) are refined.
</p>
<p>The changes dx, dy, and dB are determined and applied to each atom, and the R factor listed
</p>
<p>together with other parameters. The cycle can be repeated until no further improvement is obtained,
</p>
<p>as judged by near-constancy in the R factor, or by very small changes in the dx and dy shifts. The scale
</p>
<p>642 13 Computer-Aided Crystallography</p>
<p/>
</div>
<div class="page"><p/>
<p>factor is determined as SjFcj/SFo and may be applied to the data at the end of any cycle of refinement.
In this diagonal approximation, 60% of the calculated shifts are applied by the program. The value
</p>
<p>may be altered at any cycle, as desired. At each cycle, the new coordinates are stored in the file
</p>
<p>COORDS.TXT (see Solution 13.1 for the results).
</p>
<p>13.4.5 Electron Density Maps
</p>
<p>When some phase information becomes available, an electron density summation with the phases
</p>
<p>allocated to the corresponding Fo data should reveal a portion of the structure, if the information from
</p>
<p>the partial structure is correct. Then, more atom positions can be interpreted from the map and built
</p>
<p>into the next structure factor calculation, and so on.
</p>
<p>The same program routine will also calculate a difference electron density map, using Fo � jFcj as
coefficients, provided that the R factor is less than 0.3. In the difference synthesis, atoms placed
</p>
<p>incorrectly appear as low or negative density regions whereas unallocated atomic positions will show
</p>
<p>positive density, both relative to the general level of the figure field. It follows that when all atom
</p>
<p>positions have been determined correctly, the difference map figures should show a nearly level,
</p>
<p>ideally zero, figure field. Thus, a difference map may be used to make appropriate adjustments to
</p>
<p>atomic positions.
</p>
<p>Sometimes a fairly well-refined structure may show a significant positive region on the difference
</p>
<p>map, which may indicate the presence of solvent of crystallization. We note also that incorrect
</p>
<p>temperature factors can lead to variations in the level of the figure field, although this effect would be
</p>
<p>expected to be small when individual temperature factors are applied, as from a least-squares
</p>
<p>refinement.
</p>
<p>13.4.6 Direct Methods: Calculation of |E| Values
</p>
<p>We have shown that jEj values may be calculated from the equation
</p>
<p>jEj2 &frac14;
K2F2o
</p>
<p>e
P
j
</p>
<p>f 2j exp&eth;�Bl
�2 sin2 y&THORN;
</p>
<p>( ) (13.2)
</p>
<p>where the symbols have the meanings as described elsewhere, and the values of B and K may be
</p>
<p>obtained from a Wilson plot. However, a single isotropic B factor may not be representative of the
</p>
<p>structure; consequently, a Wilson plot may deviate from linearity. In an alternative procedure [5], we
</p>
<p>write
</p>
<p>jEj2 &frac14;
K&eth;s&THORN;F2o;corr
</p>
<p>e
P
j
</p>
<p>f 2j
</p>
<p>( ) (13.3)
</p>
<p>where K(s) is a factor that includes adjustments for the scaling of Fo and the temperature effect on fj.
</p>
<p>In implementing this method, a number n of ranges is set up in equal increments of s2, where
</p>
<p>s &frac14; (sin y)/l. For each range, K(s) is computed as
P
</p>
<p>es2
P
</p>
<p>F2o
�
</p>
<p>, where each value of Fo is given
</p>
<p>13.4 Structure Determination Simulation (XRAY) 643</p>
<p/>
</div>
<div class="page"><p/>
<p>its appropriate reflection multiplicity, according to the space group. The K(s) values, then as a
</p>
<p>function of s, are interpolated [6] so as to derive F2o;corr for each reflection. Then, jEj2 is given by
</p>
<p>jEj2 &frac14; jFo;corrj
2
</p>
<p>es2
(13.4)
</p>
<p>It is desirable to enter only the s and K(s) values corresponding to the extreme ends of the data
</p>
<p>range, because extrapolation, particularly at the low y end of a K-curve, can be uncertain. Data are
</p>
<p>output to the monitor that enable these values to be estimated. A facility is provided for printing in the
</p>
<p>main output file any reflections excluded by the procedure. There should be none, but if there are any,
</p>
<p>it means that the values entered for the extreme ends of the K-curve have not been chosen
</p>
<p>satisfactorily.
</p>
<p>The jEj values are written to a file EVALS, together with some statistics of the jEj distribution; jEj
values greater than or equal to a chosen limit, ELIM, are written to the main output file, and a
</p>
<p>P
2
</p>
<p>listing is set up in the file SIG2, in descending order of jEj magnitude. The file EDATA contains the
jEj values greater than or equal to ELIM arranged in parity groups.
</p>
<p>The program comes to a halt at this stage, so that the
P
</p>
<p>2 listing can be printed and signs for the
</p>
<p>jEj values developed along the lines already discussed in the text; see Sect. 8.2.2. Data sets are
provided only for centrosymmetric structures, but similar procedures are used at the beginning of a
</p>
<p>determination of a non-centrosymmetric crystal structure. In practice, it may be necessary to re-run
</p>
<p>the jEj values link with a slightly lower value of the limit ELIM, if insufficient data are produced for a
successful sign determination process.
</p>
<p>After a set of signs has been determined, the next program, Routine, 7, is used to prepare an E map.
</p>
<p>Alternatively, where the plane group is p2gg, the program FOUR2D, Sect. 13.6.2, can be used.
</p>
<p>13.4.7 Calculation of E Maps
</p>
<p>An jEj map is an electron density map calculated, in the case of a centrosymmetric structure,
with �jEj values. This routine in the program provides for a straightforward transfer of the jEj
values, with their signs as determined through the
</p>
<p>P
2 routine, to a Fourier calculation. As jEj values
</p>
<p>are sharpened coefficients, a few spurious peaks may be anticipated. Thus, chemical knowledge has to
</p>
<p>be brought to bear on the extraction of a sensible chemical structure or fragment. Once this has been
</p>
<p>done, and shown to be satisfactory, electron density calculations may be carried out with normal Fo
coefficients.
</p>
<p>This link of the program, however, permits modifications of signs, for further E maps, without
</p>
<p>restarting the system from scratch. A second call to routine 7 lists the current set of data as h, k, and s.
</p>
<p>A value of zero for the sign s indicates an unsigned reflection which does not contribute to the E map.
</p>
<p>To the question &ldquo;Do you want to retain some of the current values of cos(phi)?&rdquo; the answer no implies a
</p>
<p>re-input of new signs. The answer yes implies that some changes are desired, and the opportunity to do
</p>
<p>so follows. The changes are then made, and a further E map calculated.
</p>
<p>From the E map, atomic positions should be found that can be entered into the structure factor
</p>
<p>calculation, with or without a least-squares refinement, and an Fo Fourier map then calculated as
</p>
<p>described above.
</p>
<p>644 13 Computer-Aided Crystallography</p>
<p/>
</div>
<div class="page"><p/>
<p>13.4.8 Bond Lengths and Bond Angles
</p>
<p>This routine calculates bond lengths and bond angles, and distances between nonbonded atoms in the
</p>
<p>structures. The amount of information extracted depends on the distance limits input to the routine.
</p>
<p>Generally, there is no need to set a limit greater than the van der Waals contact distances, typical
</p>
<p>values of which are listed in Table 13.3. In interpreting the results from this routine, it must be
</p>
<p>remembered that, in working in two dimensions, some variations from standard numerical values are
</p>
<p>to be expected, because of the distortion of the molecule in projection. In order to minimize this
</p>
<p>effect, however, structures have been selected in which the plane of the molecule lies nearly in the
</p>
<p>plane of the projection.
</p>
<p>13.4.9 Scale and Temperature Factors by Wilson&rsquo;s Method
</p>
<p>The general procedure of Wilson&rsquo;s method has been discussed elsewhere, and is implemented in this
</p>
<p>routine. The data output contains a breakdown of the individual parts of the calculation. In particular, it
</p>
<p>lists the data for plotting the Wilson line, and thus checking on the linearity of the plot.
</p>
<p>For reliable statistics, the portion of reciprocal space under consideration should include all
</p>
<p>reflections other than systematic absences. Accidental absences, that is, reflections with intensities
</p>
<p>too low to be recorded, should be included at values 0.55 of the localized minimum Fo, that is, the
</p>
<p>minimum Fo in a given range, for a centric distribution, and 0.66 of the localized minimum Fo for an
</p>
<p>acentric distribution. Some of the data sets do not have the accidental absences included. The NO2G
</p>
<p>data is, however, complete in this respect. A check on this aspect of a data set is given by inspecting
</p>
<p>the number of reflections in each of the ranges of the Wilson plot routine; they should be approxi-
</p>
<p>mately equal.
</p>
<p>13.4.10 |E| Values Calculated from the Structure
</p>
<p>This link has been incorporated to show how jEj values may be calculated from a crystal structure. It
follows that all atoms, preferably including hydrogen atoms, must be present in their correct
</p>
<p>locations. Then, jE(hk)j is calculated from
</p>
<p>jE&eth;hk&THORN;j &frac14; 1ffiffiffiffiffiffiffiffiffiffi
ehks2
</p>
<p>p &eth;A2Z &thorn; B2Z&THORN;
1=2
</p>
<p>(13.5)
</p>
<p>Table 13.3 Van der Waals radii for some common species
</p>
<p>Atom Radius (Å) Atom Radius (Å)
</p>
<p>H 1.20 C 1.85
</p>
<p>N 1.50 O 1.40
</p>
<p>F 1.35 Si 2.10
</p>
<p>P 1.90 S 1.83
</p>
<p>Cl 1.80 As 2.00
</p>
<p>Se 2.00 Br 1.95
</p>
<p>Sb 2.20 Te 2.20
</p>
<p>I 2.15 &ndash;CH3 2.00
</p>
<p>&gt;CH2 2.00 &ndash;C6H5 1.85
a
</p>
<p>aHalf-thickness of the phenyl ring
</p>
<p>13.4 Structure Determination Simulation (XRAY) 645</p>
<p/>
</div>
<div class="page"><p/>
<p>where ehk is the epsilon factor for the hk reflection, s2 is given by
</p>
<p>s2 &frac14;
X
</p>
<p>j
</p>
<p>Z2j (13.6)
</p>
<p>and AZ and BZ are given by
</p>
<p>AZ &frac14;
X
</p>
<p>j
</p>
<p>Zj cos 2p&eth;hxj &thorn; kyj&THORN;; BZ &frac14;
X
</p>
<p>j
</p>
<p>Zj sin 2p&eth;hxj &thorn; kyj&THORN; (13.7)
</p>
<p>and the sums are taken over the N atoms in a complete unit cell. It may be noted that the temperature
</p>
<p>factor is not involved in the definition of jEj, because for a &ldquo;point atom&rdquo; fj &frac14; Zj for all h, k. The phase
associated with jEj is given generally by tan&ndash;1(BZ/AZ), having due regard to the sign of both AZ and BZ,
but will be 0 or p for centrosymmetric structures. It follows that the term E(00)1 is given by
</p>
<p>&eth;
P
</p>
<p>j Zj&THORN;=&eth;
P
</p>
<p>j Z
2
j
&THORN;1=2 which, for identical atoms, is
</p>
<p>ffiffiffiffi
</p>
<p>N
p
</p>
<p>; the value of E(00) is listed, with jEj statistics
in the results files EVALS and ECALC.
</p>
<p>13.5 Crystal Structure Analysis Problems
</p>
<p>These problems have been devised in conjunction with the XRAY program system. The different data
</p>
<p>sets may not all operate equally well with all methods of structure solving provided by the system.
</p>
<p>Hence, although the operations available are indicated on the monitor screen during execution of the
</p>
<p>program, we suggest here those procedures by which satisfactory results may be obtained for each
</p>
<p>data set provided. Organic species mostly have been chosen because it is not difficult to find examples
</p>
<p>that show well resolved and interpretable projections.
</p>
<p>There are certain features associated with working in two dimensions that we should remember:
</p>
<p>&bull; Because of the relatively small amount of data and a certain degree of inclination of the molecule
</p>
<p>to the plane of projection, some bond lengths and angles will not calculate to typical values.
</p>
<p>&bull; Fourier maps will not necessarily be true to scale, and will not present the b angle in oblique
</p>
<p>projections, but they will be satisfactory insofar as they give good practice with the structure
</p>
<p>determining methods, and enable atoms to be located. When the axis of projection is not
</p>
<p>perpendicular to the plane of projection, the true axes of the projection should be modified by
</p>
<p>an angular term. For example, for a monoclinic unit cell projected on to (100), the axes are b and
</p>
<p>c sin b. The sinb term may be important where the coordinates are measured from a map and the
</p>
<p>b-angle is very different from 90�. For the projection of a triclinic cell on to (001), the axes are
</p>
<p>a sin b and b sin a. However, since coordinates are almost always refined by least squares, the
</p>
<p>correction may often be ignored.
</p>
<p>&bull; It is rare to be able to locate hydrogen atoms in structures determined from projections, but they
</p>
<p>may be positioned by geometrical considerations.
</p>
<p>The XRAY program is entered as already described. Generally, it will help to enlarge the screen.
</p>
<p>Solutions are provided for the structure determinations in Sects. 13.5.1 and 13.5.6 but, with other
</p>
<p>1E(00) &frac14; E(000).
</p>
<p>646 13 Computer-Aided Crystallography</p>
<p/>
</div>
<div class="page"><p/>
<p>examples, the correctness of the results should be judged according to the criteria of correctness
</p>
<p>already discussed.
</p>
<p>13.5.1 Ni o-Phenanthroline Complex (NIOP)
</p>
<p>Enter the data file name NIOP and then a name for the general output file, say NOUT. The compound
</p>
<p>crystallizes in space group P212121, the plane group of the projection is p2gg and Z &frac14; 4. Open the
menu, select the Patterson link, carry out Patterson and sharpened Patterson syntheses, and plot the
</p>
<p>maps on the screen. Print at least one of the maps, so as to make for easy comparison with the other. In
</p>
<p>p2gg, the general equivalent positions are
</p>
<p>� x; y; 1
2
&thorn; x; 1
</p>
<p>2
� y
</p>
<p>� �
</p>
<p>so that interatomic vectors will have the Patterson coordinates
</p>
<p>� 2x; 2y; 1
2
; 1
2
� 2y; 1
</p>
<p>2
� 2x; 1
</p>
<p>2
&THORN;
</p>
<p>�
</p>
<p>On the screen, along the lines x &frac14;
1
</p>
<p>2
and y &frac14;
</p>
<p>1
</p>
<p>2
there are two large peaks that may be taken as
</p>
<p>Ni&ndash;Ni vectors; they are double-weight (why?). From them we obtain the atomic coordinates as ca.
</p>
<p>0.24, 0.18; the peak corresponding to the 2x, 2y vector is not well resolved in this projection. Other
</p>
<p>peaks indicate possible Ni&ndash;S vectors, but the results may not be completely satisfactory. It may be
</p>
<p>useful to keep copies of the Patterson maps for later reference.
</p>
<p>Use the coordinates of the nickel atom in the asymmetric unit to calculate structure factors and
</p>
<p>then an electron density map. This map shows the Ni atom positions and two other strong peaks to
</p>
<p>which the S atoms can be allocated. Repeat the structure factor and electron density calculations with
</p>
<p>these atoms (three per asymmetric unit), or first apply a least-squares refinement. The electron density
</p>
<p>map may not be obvious to interpret in terms of all carbon atoms. If necessary, print the asymmetric
</p>
<p>unit of the Fourier map and contour it carefully; the lowest contour on these plots is 10 and the
</p>
<p>maximum is 100.
</p>
<p>It may help in this example to contour the figure field at level 5. Search for peaks that would make
</p>
<p>up the picture of the phenanthroline complex, Fig. 13.3. It may not be possible to find all the
</p>
<p>remaining atomic positions at this stage, but enough will be located to enable a better electron density
</p>
<p>map to be calculated.
</p>
<p>When all 21 atoms, excluding hydrogen, have been found, several cycles of least-squares refine-
</p>
<p>ment, with scale factor adjustment, should converge with an R-factor of about 9.8%, which is
</p>
<p>probably the best result that can be obtained with this data set. The difference electron density map
</p>
<p>at this stage will be almost featureless. A small, negative peak near the location of the nickel atom
</p>
<p>may indicate that the isotropic temperature factor is not a completely satisfactory approximation for
</p>
<p>this species, or that there are insufficient terms for true convergence of the Fourier series, Sects. 6.2.1
</p>
<p>and 6.9.1.
</p>
<p>Bond lengths and angles may be calculated. Because the c dimension is only 4.77 Å, the molecule
</p>
<p>is quite well resolved in this projection, and the lengths and angles should have fairly sensible
</p>
<p>chemical values. The results for the nickel and sulfur atoms are Ni&ndash;S(1) &frac14; 1.953 Å, Ni&ndash;S
</p>
<p>(2) &frac14; 1.917 Å, and S(1)&ndash;Ni&ndash;S(2) &frac14; 95.08�; small variations from these values may reflect the
state of the refinement.
</p>
<p>13.5 Crystal Structure Analysis Problems 647</p>
<p/>
</div>
<div class="page"><p/>
<p>13.5.2 2-Amino-4,6-dichloropyrimidine (CL1P)
</p>
<p>The title compound, C4H3N3Cl2, crystallizes in space group P21/awith Z &frac14; 4. The plane group of the
projection on (010) is p2, doubled along the x-axis because of the translation of the a-glide plane. The
</p>
<p>data for this structure and for CL2P produce satisfactory Wilson plots.
</p>
<p>There are two chlorine atoms in the asymmetric unit, not related by symmetry. Hence, these two
</p>
<p>chlorine atoms together with the two related by the twofold symmetry will give rise to eight non-
</p>
<p>origin Cl&ndash;Cl vectors:
</p>
<p>Type 1, single weight for each: � &eth;2x1; 2y1&THORN;; �&eth;2x2; 2y2&THORN;
Type 2, double weight for each: � &eth;x2 � x1; y2 � y1&THORN;; �&eth;x2 &thorn; x1; y2 &thorn; y1&THORN; where the single-weight
</p>
<p>vectors terminate at the corners of a parallelogram, and the double-weight vectors terminate at the
</p>
<p>mid-points of its sides.
</p>
<p>Solve the Patterson projection for positions of the chlorine atoms, and then complete the structure
</p>
<p>determination for the non-hydrogen atoms. It will be helpful to print more than one copy of the
</p>
<p>Patterson map, and then to join them such that the origin is at the center of the composite. (Hint: the
</p>
<p>coordinates of one of the chlorine atoms are ca. 0.16, 0.16.)
</p>
<p>The data for this projection will refine to ca. 12.1%. The bond lengths and angles from this
</p>
<p>projection indicate a tilt of the molecule out of the plane of projection.
</p>
<p>13.5.3 2-Amino-4-methyl-6-chloropyrimidine (CL2P)
</p>
<p>Consider the unit cell data for this compound, C4H6N2Cl, and that for the dichloropyrimidine
</p>
<p>just studied:
</p>
<p>2-Amino-4-methyl-6-chloropyrimidine 2-Amino-4,6-dichloropyrimidine
</p>
<p>a (Å) 16.426 16.447
</p>
<p>b (Å) 4.000 3.845
</p>
<p>c (Å) 10.313 10.283
</p>
<p>b (�) 109.13 107.97
</p>
<p>Z 4 4
</p>
<p>Space group P21/a P21/a
</p>
<p>Fig. 13.3 The molecular structure of C12H14N6S2Ni
</p>
<p>648 13 Computer-Aided Crystallography</p>
<p/>
</div>
<div class="page"><p/>
<p>The two sets of crystal data are sufficiently similar for the two pyrimidine derivatives to be treated
</p>
<p>as isomorphous. Hence, it should be possible to allocate trial atomic coordinates from the structure of
</p>
<p>the dichloropyrimidine; one of the chlorine atoms in the dichloro-compound has been replaced here
</p>
<p>by a methyl group.
</p>
<p>Calculate Patterson maps for this compound and, by comparison with the previous structure,
</p>
<p>obtain atomic coordinates for a trial structure of this compound. Then refine the trial structure by
</p>
<p>successive Fourier syntheses and least squares. Alternatively, use the coordinates from CL1P mutatis
</p>
<p>mutandis and calculate a first electron density map.
</p>
<p>This data set refines to an R value of approximately 12.3%, but some of the bond lengths differ
</p>
<p>from the accepted values for this compound because of the inclination of the molecule to the plane of
</p>
<p>projection.
</p>
<p>13.5.4 m-Tolidine Dihydrochloride (MTOL)
</p>
<p>m-Tolidine dihydrochloride, crystallizes in space group I2, a non standard setting of C2, with Z &frac14; 2.
The molecules occupy special positions on twofold axes. The plane group is p2 in the projection on to
</p>
<p>(010). Thus, the chlorine atoms are related by twofold symmetry to give a Patterson vector at 2x, 2z.
</p>
<p>The projection can be solved by the heavy-atom method, and refined to ca. 22% with the given
</p>
<p>data. The atoms are well resolved, albeit with some distortion and the bond lengths are at variance
</p>
<p>with standard values, because no account can be taken of the third dimension in their calculation from
</p>
<p>this projection.
</p>
<p>13.5.5 Nitroguanidine (NO2G)
</p>
<p>Nitroguanidine, C(NH2)2NNO2, crystallizes in space group Fdd2, with the unit cell dimensions
</p>
<p>a &frac14; 17.639 Å, b &frac14; 24.873 Å, c &frac14; 3.5903 Å, and Z &frac14; 16. The small c dimension, approximately
equal to the van der Waals nonbonded distance between carbon atoms, means that good resolution
</p>
<p>will arise in the projection on (001). The plane group of this projection is p2gg, with the a and b
</p>
<p>dimensions halved and four molecules in the transformed unit cell.
</p>
<p>This structure is suitable for the direct methods procedure. In two dimensions, two reflections
</p>
<p>suffice to fix the origin provided they are chosen one from any two of the parity groups h even/k
</p>
<p>odd, h odd/k even, h odd/k odd; h even k even is a structure seminvariant and cannot be used to restrict
</p>
<p>the origin.
</p>
<p>In p2gg, the sign relationships in reciprocal space may be summarized as s&eth;hk&THORN; &frac14; s&eth;h k&THORN; &frac14;
&eth;�1&THORN;h&thorn;ks&eth;�hk&THORN;, so that both positive and negative signs will be generated by the
</p>
<p>P
2 equation.
</p>
<p>One or more letter symbols may be used as necessary, in order to aid the sign allocation process.
</p>
<p>In some cases, the signs attaching to such letters evolve during the procedure; otherwise, trial
</p>
<p>E maps must be constructed with permuted values for the letter signs. This structure will refine to an
</p>
<p>R value of approximately 5%.
</p>
<p>As an alternative procedure at the E map stage, the data can be assembled as lines of h, k, jEj, s(E),
</p>
<p>and used in conjunction with the program FOUR2D, which has been written for plane group p2gg.
</p>
<p>13.5 Crystal Structure Analysis Problems 649</p>
<p/>
</div>
<div class="page"><p/>
<p>13.5.6 Bis(6-sulfanyloxy-1,3,5-triazin-2(1H)-one) (BSTO)
</p>
<p>This compound, (C3H2N3O2S)2, crystallizes in space group P21/m, with Z &frac14; 2, so that the molecules
occupy special positions on m-planes. In projection on (010), the plane group is p2 and the molecules
</p>
<p>occupy general positions in the plane group.
</p>
<p>The Patterson maps indicate more than one peak of similar height in the asymmetric unit, so that it
</p>
<p>may be necessary to investigate both of them in order to find a good trial structure. This structure
</p>
<p>responds well to the superposition technique, so that the solution is not as difficult as might have been
</p>
<p>expected. Refinement to ca. 15.5% can be achieved.
</p>
<p>13.5.7 2-S-methylthiouracil (SMTX and SMTY)
</p>
<p>2-S-methylthiouracil, C4H6N2OS, is triclinic, with space group P�1 and Z &frac14; 2. Data for the (100) and
(010) projections are supplied, for which the plane group is p2 in each case. This structure may
</p>
<p>present more difficulty than that in the previous example, because several peaks of similar height
</p>
<p>occur in the Patterson maps.
</p>
<p>The correct choice refines here to ca. 13.0% (R &frac14; 7.3% with three-dimensional data has been
reported in the literature). The molecule is mostly well resolved, but not all atoms, particularly the
</p>
<p>carbon attached to sulfur, are clearly resolved in this projection. Note that, for the (100) projection, the
</p>
<p>axes marked x and y on the plot are, strictly, y sin g and z sin b, respectively.
</p>
<p>The second data set for this compound relates to the (010) projection. By solving it, a three-
</p>
<p>dimensional model for the compound can be built up.
</p>
<p>The given selection of problems provides good practice in current, basic structure-solving meth-
</p>
<p>ods. Other problems can be built up as desired; data for suitable structures can be found in the early
</p>
<p>volumes of Acta Crystallographica. If you do this, remember to adopt the correct format.
</p>
<p>13.6 General Crystal Structure and Other Programs
</p>
<p>13.6.1 One-Dimensional Fourier Summation (FOUR1D)
</p>
<p>This program calculates a one-dimensional Fourier summation, r(x). The data comprise lines of index
</p>
<p>h and coefficients A(h) and B(h),which must be available in a file named ABDAT.TXT (example
</p>
<p>given). If r(x) is centrosymmetric, B(h) should be entered as zero for each data line. The figure field
</p>
<p>for r(x) is established in the file RHOX.TXT from whence it can be plotted; it is normalized to a
</p>
<p>maximum value of 50. The file FUNCTN.TXT contains the true values of the data in a form suitable
</p>
<p>for the Fourier transform program TRANS1. The desired interval of subdivision N is entered at the
</p>
<p>keyboard; its maximum value in the program is 100.
</p>
<p>13.6.2 Two-Dimensional Fourier Summation (FOUR2D)
</p>
<p>This program computes a two-dimensional Fourier summation for plane group p2gg. The datamust be in
</p>
<p>a file named TWODAT.TXT (example given) as lines of h, k, Fo, and s; s is the sign (�1) that multiplies
Fo. The interval of subdivision is 40 along both the x and y axes, and the output from the file RHOXY.
</p>
<p>650 13 Computer-Aided Crystallography</p>
<p/>
</div>
<div class="page"><p/>
<p>TXT may be joined along the duplicated lines y &frac14; 20. Four molecules are contained within the unit
cell of the plot, which is normalized to a maximum of 100. Contour the plot in steps of 10 units.
</p>
<p>The source code FOUR2D.F90 for this program, written in FORTRAN 90, is also supplied.
</p>
<p>Those conversant with FORTRAN 90 may wish to modify this program for other plane groups.
</p>
<p>The procedure is straightforward [7], making use of the electron density equations given in the
</p>
<p>International Tables for X-ray Crystallography, vol I [8]. Consider, for example, plane groups p2:
</p>
<p>we can write r(xy) as
</p>
<p>r&eth;xy&THORN; &frac14; K&eth;CT2FC1 � ST2GS1&THORN; (13.8)
</p>
<p>where K is a constant involving the normalization of the output results, CT2 and S
T
2 are the transposes
</p>
<p>of matrices of cos(2phX) and sin(2phnY), of order hmax � nX (nX &frac14; nY &frac14; 0/40, 1/40, 2/40,. . ., as set
currently in the program), and C1 and S1 are matrices of cos(2pknY/b) and sin(2pknY/b), of order
kmax � nY. In plane group p2, F and G are matrices of order hmax � kmax, with elements
</p>
<p>&frac12;Fo&eth;hk&THORN; &thorn; Fo&eth;hk&THORN;� and &frac12;Fo&eth;hk&THORN;� Fo&eth;hk&THORN;�, respectively. A step to form these elements could be
</p>
<p>inserted into the program.
</p>
<p>13.6.3 One-Dimensional Fourier Transform (TRANS1)
</p>
<p>This program calculates the Fourier transform of a one-dimensional function f(x). The function is
</p>
<p>divided into an even number of intervals, up to a maximum of 100, and contained, one datum to a line,
</p>
<p>in the file FUNCTN.TXT; only the values of the function are used as data. The number of data (the
</p>
<p>interval N of subdivision of the function) is entered at the keyboard, followed by the maximum
</p>
<p>frequency hmax for the output coefficients.
</p>
<p>Because of sampling conditions, Sect. 6.6.7, if N is chosen as 30, hmax could be conveniently
</p>
<p>10&ndash;15. The output in the file ABDAT.TXT can then be used with FOUR1D to recreate the original
</p>
<p>function, f(x). Note: Because FOUR1D writes a file named FUNCTN.TXT, the original values of this
</p>
<p>function will be lost unless saved in another file.
</p>
<p>13.6.4 Reciprocal Unit Cell (RECIP)
</p>
<p>This program determines the parameters of the reciprocal unit cell from those of the corresponding
</p>
<p>direct space unit cell (or vice versa) and the volumes of both cells. The input consists of the reciprocal
</p>
<p>constant K (chosen as unity here) and the parameters a, b, c, a, b, and y, say, 5.0 Å, 6.0 Å, 7.0 Å, 90.0�,
</p>
<p>105.0�, and 90.0�. The output is self-explanatory.
</p>
<p>13.6.5 Molecular Geometry (MOLGOM)
</p>
<p>This program calculates bond lengths, bond angles, and torsion angles. It requires the following data
</p>
<p>input from a file named MOLDAT.TXT (example given):
</p>
<p>Unit-cell parameters: 5.0, 6.0, 7.0, 90.0, 105.0, 90.0
</p>
<p>Number of atoms (e.g.,)
</p>
<p>Atom number and x, y, z coordinates
</p>
<p>13.6 General Crystal Structure and Other Programs 651</p>
<p/>
</div>
<div class="page"><p/>
<p>As prompted, the atoms forming a torsion angle are entered at the keyboard. The convention
</p>
<p>relating to the sign of the torsion angle has been discussed, Sect. 8.5.2 and Appendix C. The results
</p>
<p>are listed in the file GEOM.TXT. Note that a program error at the torsion angle stage means that either
</p>
<p>the data is incorrect or that a torsion angle cannot be defined by the order of the atoms given. If the
</p>
<p>coordinates are in absolute measure, as with the test data, then a, b and c are each entered as unity.
</p>
<p>13.6.6 Internal and Cartesian Coordinates (INTXYZ)
</p>
<p>This program converts the geometry of a molecule in terms of its internal coordinates, that is, bond
</p>
<p>lengths, bond angles, and torsion angles, to a set of Cartesian coordinates for the molecule. The data
</p>
<p>must be supplied from a file named CART.TXT (example given), and take the form of lines of atom
</p>
<p>code number, bond angle, torsion angle, bond length, as indicated on the monitor screen (with results)
</p>
<p>after the program is opened; the convention for torsion angles given in Appendix C applies. The first
</p>
<p>entry is always
</p>
<p>0 0:0 0:0 0:0
</p>
<p>In subsequent lines, the code number of the current atom is the atom number of a previous atom to
</p>
<p>which the current atom is linked. As an example of input data, consider the molecular fragment shown
</p>
<p>here in Fig. 13.4, with its internal coordinates, or geometry, as given. Then the input data has the
</p>
<p>following format:
</p>
<p>Atom code number
</p>
<p>(not entered) Input data for file CART.TXT
</p>
<p>1 0 0.0 0.0 0.0
</p>
<p>2 1 0.0 0.0 1.49
</p>
<p>3 2 109.0 0.0 1.50
</p>
<p>4 3 110.0 0.0 1.54
</p>
<p>5 4 107.0 180.0 1.51
</p>
<p>6 4 105.0 �30.0 1.52
</p>
<p>Atom number 1 is at the origin, and the fragment 1&ndash;2&ndash;3&ndash;4&ndash;5 is planar. Does the 4&ndash;6 bond lie
</p>
<p>above or below this plane? The Cartesian coordinates given by the program are in the file
</p>
<p>METRIC.TXT as:
</p>
<p>Atom X Y Z
</p>
<p>1 0.0000 0.0000 0.0000
</p>
<p>2 �1.4900 0.0000 0.0000
</p>
<p>3 �1.9784 1.4183 0.0000
</p>
<p>4 �0.7815 2.3874 0.0000
</p>
<p>5 �1.3472 3.7875 0.0000
</p>
<p>6 0.32437 1.64686 0.7341
</p>
<p>652 13 Computer-Aided Crystallography</p>
<p/>
</div>
<div class="page"><p/>
<p>13.6.7 Linear Least Squares (LSLI)
</p>
<p>This program determines the best-fit straight line to a series of data points that must number at least 3
</p>
<p>and, in this program, cannot exceed 100. Data must be entered from the keyboard or from a file named
</p>
<p>LSSQ.TXT (example given). The first two lines of data, title, and number of data pairs, must always
</p>
<p>be entered at the keyboard. Then the remainder of the data follow in lines of xi, yi; unit weights for
</p>
<p>each observation are assumed in this program.
</p>
<p>It is implicit that errors in x are significantly smaller than those in y. The goodness-of-fit is reflected
</p>
<p>by the values of s(a), s(b), and Pearsons&rsquo;s r coefficient. If the errors in the parameters a and b are to
</p>
<p>be propagated to another quantity z, then they follow the law given in Sect. 8.6.
</p>
<p>13.6.8 Matrix Operations (MATOPS)
</p>
<p>This program accepts an input of two 3 � 3 matricesA and B, and forms A + B,A � B,A � B,AT,
BT, Trace(A), Trace(B), Det(A), Det(B), Cofactor(A), Cofactor(B), A�1, and B�1. If results are
required on only one matrix, A, then B is set by the program as the unit matrix
</p>
<p>1 0 0
</p>
<p>0 1 0
</p>
<p>0 0 1
</p>
<p>0
@
</p>
<p>1
A
</p>
<p>13.6.9 Q Values (QVALS)
</p>
<p>This program is useful in conjunction with work on indexing powder diffraction patterns. Given the
</p>
<p>unit cell parameters a, b, and c in Å, and a, b, and g in degrees, the program produces a set of values of
</p>
<p>104Q; maximum values for h, k, and l are also entered at the keyboard. Provision is made for the
</p>
<p>indices k and l to take negative values. Thus, in any symmetry higher than triclinic, duplicate values
</p>
<p>of Q will be generated and may be discarded as required. The results are in the file INDEX.
</p>
<p>Fig. 13.4 Hypothetical fragment C6, for input to program INTXYZ
</p>
<p>13.6 General Crystal Structure and Other Programs 653</p>
<p/>
</div>
<div class="page"><p/>
<p>13.6.10 Le Page Unit-Cell Reduction (LEPAGE)
</p>
<p>This program was written by A.L. Spek of the University of Utrecht and kindly made available by him
</p>
<p>to the academic community. For reduction, choose the D-option when prompted, and enter the unit
</p>
<p>cell parameters as indicated, one to a line. It may be desirable to vary the &ldquo;2-axis criterion&rdquo; by means
</p>
<p>of the C-option.
</p>
<p>The program reports inter alia the input unit cell, the reduced unit cell, the conventional
</p>
<p>crystallographic unit cell (which may be the same as the reduced cell), and the transformation
</p>
<p>matrices for a, b, c and x, y, z. Other options are provided by the program, but they need not concern
</p>
<p>us here. Note that this program refers to the triclinic system by the symbol a: the alternative name for
</p>
<p>the triclinic system is anorthic.
</p>
<p>A useful mnemonic for using any transformation matrix M and its inverse has been given in Sect.
2.5.5 and the scheme of Fig. 2.16 in the same chapter.
</p>
<p>13.6.11 Zone symbols/Miller indices (ZONE)
</p>
<p>This program calculates the Miller indices of a plane from the input of two zone symbols, or the
</p>
<p>symbol of a zone from the input of the Miller indices for two planes.
</p>
<p>13.7 Automatic Powder Indexing: ITO12
</p>
<p>This program has been made available to the academic community by courtesy of Dr. J. Vissser,
</p>
<p>Technisch Physische Dienst, Delft [9]. The format of the input data is very specific and must be
</p>
<p>followed. The data file must be named ITOINP.DAT, and set out as follows; the parentheses indicate
</p>
<p>FORTRAN formats:
</p>
<p>Line 1: Title, up to 80 alphanumeric characters (A80).
</p>
<p>Line 2: Leave blank; it is related to a number of parameters that take default values in the program,
</p>
<p>and which we need not discuss here.
</p>
<p>Line 3: Four parameters: a zeroshift, 0.0 is recommended; a print controlminimum value ofM20 for a
</p>
<p>lattice to be printed, 4.0 is recommended; a print control minimum value of lines indexed for a
</p>
<p>lattice to be printed, 14.0 is recommended; the number of data, between 20 and 40. These four
</p>
<p>parameters are entered as real numbers, and terminate at character numbers 10, 20, 30, and 40,
</p>
<p>respectively, in the line (4F10.5).
</p>
<p>Lines 4: The data, 104Q or sin2 y or 2y values (in ascending order), or d (in descending order), in n
</p>
<p>lines each containing eight such values, the data ending at character numbers 10, 20,. . .,80 in each
</p>
<p>line (8F10.5).
</p>
<p>Line 5: Leave blank.
</p>
<p>Line 6: The word END as its first three characters (A3).
</p>
<p>An example input file ITOINP.DAT, which relates to Problem 9.6a, is provided with the Program
</p>
<p>Suite POWDER folder. One output file is generated by the program, and provides an echo of the input
</p>
<p>data file. A second output file contains the results of the indexing; it is mostly self-explanatory.
</p>
<p>However, the following column heads have meanings as follow:
</p>
<p>654 13 Computer-Aided Crystallography</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7_2#Fig16_2">http://dx.doi.org/10.1007/978-1-4614-3954-7_2#Fig16_2</a></div>
</div>
<div class="page"><p/>
<p>The best results follow based on the first 20 indexed lines, and then those based on all lines in the
</p>
<p>data set, up to 40; finally the best, refined unit cells are printed. From a study of the observed
</p>
<p>reflections, deductions may be made about the space group of the crystal. The program and example
</p>
<p>data set are together in a folder named ITO12.
</p>
<p>13.8 Automatic Powder Structure Solving: ESPOIR
</p>
<p>This program has been made available to the academic community by courtesy of Professor A Le
</p>
<p>Bail, Laboratoire Fluorures, Université du Main, Le Mans, and we discussed it in Sect. 12.11. Only
</p>
<p>the essentials needed to run a data set are included here, and the reader is referred to an original
</p>
<p>reference [10] for a fuller exposition of the features of this technique.
</p>
<p>We follow through a sequence of instructions for using ESPOIR. Because a large number of files
</p>
<p>can be generated in using this program, we have placed the program and data in a separate directory,
</p>
<p>named POWDER.
</p>
<p>13.8.1 Aragonite
</p>
<p>As a first example, we consider the aragonite form of calcium carbonate, CaCO3. The space group is
</p>
<p>Pmcn, a nonstandard setting of Pnma. Since there are four formula entities per unit cell, the Ca and C
</p>
<p>atoms lie on special positions, but the oxygen atoms could occupy one set of general positions plus
</p>
<p>one set of special positions or three sets of special positions. In practice, both arrangements may need
</p>
<p>to be tried. We report here the successful choice, that is, with occupancies of
1
</p>
<p>2
for each of Ca, C, and
</p>
<p>O1, and unit occupancy for O2. The following procedure is typical:
</p>
<p>Open Espoir.exe.
</p>
<p>From the Espoir window: File ! Open File and Open Arag.dat
From the Espoir window: Run ! Espoir
</p>
<p>A number (10) of tests follow; the end is signified by the Run Terminate box
</p>
<p>From the Espoir window: View ! Open .spf for the chosen, best run. Open .imp for the run data and
the 10 test results.
</p>
<p>The chosen set of x, y, z coordinates is shown below, together with general equivalent position
</p>
<p>appropriate to space group Pmcn; the parameters can be refined by least squares.
</p>
<p>NEWNR New sequence number for the zone after evaluation (cp. OLDNR)
</p>
<p>A Provisional value of QA for the zone
</p>
<p>B Provisional value of QB
</p>
<p>FMAAS Provisional value of QF
</p>
<p>QUALITY Measure of fit for the zone based on probability theory
</p>
<p>OLDNR Old sequence number
</p>
<p>CNTR Flag, equal to 0 for primitive zone or 1 for a centered zone
</p>
<p>NOBS &amp; NCALC Number of observed and calculated lines used in evaluating QUALITY
</p>
<p>ZERSHFT Estimate of 2y zero error for the zone
</p>
<p>13.8 Automatic Powder Structure Solving: ESPOIR 655</p>
<p/>
</div>
<div class="page"><p/>
<p>The known parameters for the Aragonite structure are given below, showing that there is good
</p>
<p>agreement with the structure determined, taking into account symmetry-related atoms:
</p>
<p>x y z
</p>
<p>Ca 1
4
</p>
<p>0.4151 0.2405
</p>
<p>C 1
4
</p>
<p>0.7621 0.0852
</p>
<p>O1 14 0.9222 0.0956
</p>
<p>O2 0.4735 0.6807 0.0873
</p>
<p>13.8.2 a-Alumina (Corundum)
</p>
<p>In the POWDER folder there are three pairs of files for a-alumina, labeled Al2O3A, Al2O3B, and
</p>
<p>Al2O3C. For each of A, B, and C, one file type (.dat) contains crystal data and program settings, and
</p>
<p>the other (.hkl) contains the Fo(hkl) data. We consider just Al2O3A, in which the structure is treated
</p>
<p>from scratch, applying distance constraints for Al&ndash;Al of 3.0, Al&ndash;O of 1.6, and O&ndash;O of 1.6 Å,
</p>
<p>respectively. We know also the space group, R�3c, and that Z &frac14; 6. This data contains more output
options, as we shall see.
</p>
<p>Open Espoir.exe.
</p>
<p>From the Espoir window: File ! Open File and Open Al2O3A.dat.
From Espoir window: Run ! Espoir.
</p>
<p>Ten test trials are now performed, followed by Run Terminate.
</p>
<p>From the Espoir window: Open ! View. The .spf and .imp files are present, as with previous run. In
addition, several other flies are presented; of particular interest are Profile and Structure.
</p>
<p>From the Espoir window: Open ! Profile. The red plot shows the pattern of agreement between
calculated and observed jFj values, whereas the blue plot is the difference pattern, which highlights
their discrepancies. Close Profile.
</p>
<p>From the Espoir window: View ! Structure. The RASMOL program is now invoked.
From the RasWin window: Open ! Display.
</p>
<p>Under Display, Sticks, Spacefill and Ball &amp; Sticks are probably the most useful. Select Ball &amp;
</p>
<p>Spokes.
</p>
<p>From the RasWin window: Open ! Colours. Select ! CPK (normal).
From the RasWin window: Open! Options. Select Specular. This mode enhances the appearance of
</p>
<p>the model. Under Option, the links Labels is useful, but can be best seen withMonochrome (under
</p>
<p>Colours). A Stereo link also exists under Options. A stereo viewer will be needed, and it may be
</p>
<p>necessary to decrease the horizontal distance between the stereo pairs.
</p>
<p>From the RasWin window: Open! Export, and write the diagram in the .BMP mode for subsequent
printing.
</p>
<p>On returning to the original folder (POWDER), a number of new files relating to Al2O3A will
</p>
<p>be found. One of the files, Al2O3Astru (a .dat file), contains structural data about the crystal and
</p>
<p>x y z x y z
</p>
<p>Ca 0.7552 0.9150 0.2602 �x; 1
2
&thorn; y; 1
</p>
<p>2
� z ! 0.2448 0.4150 0.2498
</p>
<p>C 0.2498 0.7624 0.0859 x; y; z ! 0.2498 0.7624 0.0959
</p>
<p>O1 0.7553 0.0781 0.9074 �x; �y; �z ! 0.2447 0.9219 0.0926
</p>
<p>O2 0.5289 0.3188 0.9123 �x; �y; �z ! 0.4711 0.6812 0.0877
</p>
<p>656 13 Computer-Aided Crystallography</p>
<p/>
</div>
<div class="page"><p/>
<p>the view, together with a list of x, y, z coordinates found from that run; the results for two such
</p>
<p>runs were:
</p>
<p>x y z
</p>
<p>A1 Al 0.6667 0.3333 0.9812
</p>
<p>O 0.0000 0.6667 0.7500
</p>
<p>A2 Al 0.3274 0.6725 0.8146
</p>
<p>O 0.6887 0.9995 0.2498
</p>
<p>The two runs do not show apparently the same coordinates because of the random nature of the
</p>
<p>process. We shall discuss these results shortly, but first we consider the other two data sets.
</p>
<p>Next, carry out the procedure with Al2O3B. This data set is arranged to fit to the Fo data rather than a
</p>
<p>regenerated pattern, and makes use of chosen values for the occupation numbers; the execution time
</p>
<p>is much shorter. One set of coordinates, the best fit, will be produced.
</p>
<p>From the View window: Select Al2O3B.spf.
</p>
<p>Alternatively, return to the POWDER folder: Select A12O3B.spf. The results of two runs are
</p>
<p>listed below.
</p>
<p>x y z
</p>
<p>B1 Al 0.3422 0.6702 0.8146
</p>
<p>O 0.6417 0.6716 0.9165
</p>
<p>B2 Al 0.3414 0.6668 0.8146
</p>
<p>O 0.3424 0.9772 0.9159
</p>
<p>Finally, repeat the second procedure now with Al2O3C. In this example, the constraints of the
</p>
<p>special positions of the type 0, 0, z for Al and x, 0,
1
</p>
<p>4
for O have been added to the data set.We obtain
</p>
<p>the coordinates and a plot of the best-fit structure. Two such runs are shown below:
</p>
<p>x y z
</p>
<p>C1 A1 0.0000 0.0000 0.6470
</p>
<p>O 0.6940 0.0000 0.2500
</p>
<p>C2 A1 0.0000 0.0000 0.6479
</p>
<p>O 0.3060 0.0000 0.2500
</p>
<p>In order to interpret the totality of these results, we list the special positions for space group R�3c
</p>
<p>with Z &frac14; 6, and the centers of symmetry in the unit cell:
</p>
<p>&eth;0; 0; 0; 1
3
; 2
3
; 2
3
; 2
3
; 1
3
; 1
3
&THORN;&thorn;
</p>
<p>12 Al at� &eth;0; 0; z; 0; 0; 1
2
&thorn; z&THORN; 18O at � &eth;x; 0; 1
</p>
<p>4
; 0; x; 1
</p>
<p>4
; �x; �x; 1
</p>
<p>4
&THORN;
</p>
<p>�1 at &eth;0; 0; 0; 0; 0; 1
2
; 0; 1
</p>
<p>2
; 0; 1
</p>
<p>2
; 0; 0; 0; 1
</p>
<p>2
; 1
2
; 1
</p>
<p>2
; 0; 1
</p>
<p>2
; 1
</p>
<p>2
; 1
2
; 0; 1
</p>
<p>2
; 1
2
; 1
2
&THORN;
</p>
<p>The program does not necessarily select all atoms from one and the same asymmetric unit, so that
</p>
<p>we have to consider the full implication of the space group symmetry and choice of origin. For
</p>
<p>example, we take result C2 as a norm. Then, if O in set Cl is moved across the center of symmetry at
</p>
<p>1,0,0, that set then agrees with C2. In sets B, we add the translations 2
3
; 1
3
; 1
3
in each case, which leads to
</p>
<p>	0, 	0, 0.1479 (�0.6479); 	0, 	0, 0.1479 (�0.6479) for B1: and 	0, 	0, 0.1479 and 	0, 	0,
</p>
<p>13.8 Automatic Powder Structure Solving: ESPOIR 657</p>
<p/>
</div>
<div class="page"><p/>
<p>0.2492 for B2, both of which agree reasonably with set C. In a similar way, set A can be transformed
</p>
<p>to 0, 0, 0.6479 and 0, 0.3333, 0.2500, and 	0, 	0, 0.1479 (�0.6479) and 0.3113, 	0,0.2498. Except
</p>
<p>for Al2O3C, where the constraints are strong, we would not always expect to get the same numerical
</p>
<p>values exactly in subsequent trials, because of the random nature of the movement of the atoms.
</p>
<p>Recently reported parameters for the a-alumina structure are:
</p>
<p>13.9 Problems
</p>
<p>The problems for this chapter arise throughout the text itself, Sect. 13.5.1ff. Solutions are provided
</p>
<p>for two of them.
</p>
<p>References
</p>
<p>1. http://www.ccp4.ac.uk/
</p>
<p>2. http://www.ccp14.ac.uk/
</p>
<p>3. Ladd MFC, Int J (1976) Math Educ Sci Technol 7:395
</p>
<p>4. Bibliography, Ladd (1989)
</p>
<p>5. Karle J et al (1958) Acta Crystallogr 11:757
</p>
<p>6. Ladd MFC (1978) Z Kristallogr 147:279
</p>
<p>7. Ladd MFC, Davies M (1968) Z Kristallogr 126:210
</p>
<p>8. Loc. cit., Chapter 1
</p>
<p>9. http://www.ccp14.ac.uk/tutorial/crys/program/ito12.htm
</p>
<p>10. Mileur M, Le Bail A. http://www.cristal.org/sdpd/espoir
</p>
<p>x y z
</p>
<p>Al 0 0 0.6477
</p>
<p>O 0.3064 0 1
4
</p>
<p>658 13 Computer-Aided Crystallography</p>
<p/>
<div class="annotation"><a href="http://www.ccp4.ac.uk/">http://www.ccp4.ac.uk/</a></div>
<div class="annotation"><a href="http://www.ccp14.ac.uk/">http://www.ccp14.ac.uk/</a></div>
<div class="annotation"><a href="http://www.ccp14.ac.uk/tutorial/crys/program/ito12.htm">http://www.ccp14.ac.uk/tutorial/crys/program/ito12.htm</a></div>
<div class="annotation"><a href="http://www.cristal.org/sdpd/espoir">http://www.cristal.org/sdpd/espoir</a></div>
</div>
<div class="page"><p/>
<p>Appendix A: Stereoviews
and Crystal Models
</p>
<p>A.1 Stereoviews
</p>
<p>Stereoviews of crystal structures began to be used to illustrate three-dimensional structures in 1926.
</p>
<p>Nowadays, this technique is quite commonplace, and computer programs exist (see Appendices D4
</p>
<p>and D8.7) that prepare the two views needed for producing a three-dimensional image of a crystal or
</p>
<p>molecular structure.
</p>
<p>Two diagrams of a given object are necessary in order to form a three-dimensional visual image.
</p>
<p>They should be approximately 63 mm apart and correspond to the views seen by the eyes in normal
</p>
<p>vision. Correct viewing of a stereoscopic diagram requires that each eye sees only the appropriate half
</p>
<p>of the complete illustration, and there are two ways in which it may be accomplished.
</p>
<p>The simplest procedure is with a stereoviewer. A supplier of a stereoviewer that is relatively
</p>
<p>inexpensive is 3Dstereo.com. Inc., 1930 Village Center Circle, #3-333, Las Vegas, NV 89134, USA.
</p>
<p>The pair of drawings is viewed directly with the stereoviewer, whereupon the three-dimensional
</p>
<p>image appears centrally between the two given diagrams.
</p>
<p>Another procedure involves training the unaided eyes to defocus, so that each eye sees only the
</p>
<p>appropriate diagram. The eyes must be relaxed and look straight ahead. The viewing process may be
</p>
<p>aided by holding a white card edgeways between the two drawings. It may be helpful to close the eyes
</p>
<p>for a moment, then to open them wide and allow them to relax without consciously focusing on the
</p>
<p>diagram.
</p>
<p>Finally, we give instructions whereby a simple stereoviewer can be constructed with ease. A pair
</p>
<p>of plano-convex or bi-convex lenses, each of focal length approximately 100 mm and diameter
</p>
<p>approximately 30 mm, is mounted between two opaque cards such that the centers of the lenses
</p>
<p>are approximately 63 mm apart. The card frame must be so shaped that the lenses may be brought
</p>
<p>close to the eyes. Figure A.1 illustrates the construction of the stereoviewer.
</p>
<p>A.2 Model of a Tetragonal Crystal
</p>
<p>A model similar to that illustrated in Fig. 1.23 can be constructed easily. This particular model has
</p>
<p>been chosen because it exhibits a four-fold inversion axis, which is one of the more difficult symmetry
</p>
<p>elements to appreciate from drawings.
</p>
<p>M. Ladd and R. Palmer, Structure Determination by X-ray Crystallography:
</p>
<p>Analysis by X-rays and Neutrons, DOI 10.1007/978-1-4614-3954-7,
# Springer Science+Business Media New York 2013
</p>
<p>659</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Fig23_1">http://dx.doi.org/10.1007/978-1-4614-3954-7#Fig23_1</a></div>
</div>
<div class="page"><p/>
<p>A good quality paper or thin card should be used for the model. The card should be marked out in
</p>
<p>accordance with Fig. A.2 and then cut out along the solid lines, discarding the shaded portions. Folds
</p>
<p>are made in the same sense along all dotted lines, the flaps ADNP and CFLM are glued internally, and
</p>
<p>the flap EFHJ is glued externally. What is the point group of the resulting model?
</p>
<p>Fig. A.1 Construction of a simple stereoviewer. Cut out two pieces of card as shown and discard the shaded portions.
Make cuts along the double lines. Glue the two cards together with lenses EL and ER in position, fold the portions A and
</p>
<p>B backward, and engage the projection P into the cut atQ. Strengthen the fold with a strip of &ldquo;Sellotape.&rdquo; View from the
</p>
<p>side marked B. It may be helpful to obscure a segment on each lens of maximum depth ca. 30 % of the lens diameter,
</p>
<p>closest to the nose region
</p>
<p>660 Appendix A</p>
<p/>
</div>
<div class="page"><p/>
<p>Fig. A.2 Construction of a tetragonal crystal with a �4 axis: NQ &frac14; AD &frac14; BD &frac14; BC &frac14; DE &frac14; CE &frac14; CF &frac14;
KM &frac14; 100 mm; AB &frac14; CD &frac14; EF &frac14; GJ &frac14; 50 mm; AP &frac14; PQ &frac14; FL &frac14; KL &frac14; 20 mm; AQ &frac14; DN &frac14; CM &frac14; FK &frac14;
FG &frac14; FH &frac14; EJ &frac14; 10 mm
</p>
<p>Appendix A 661</p>
<p/>
</div>
<div class="page"><p/>
<p>Appendix B: Schönflies&rsquo;
Symmetry Notation
</p>
<p>Theoretical chemists and spectroscopists generally use the Schönflies notation for describing
</p>
<p>point-group symmetry but, although both the crystallographic (Hermann&ndash;Mauguin) and Schönflies
</p>
<p>notations are adequate for point groups, only the Hermann&ndash;Mauguin system is satisfactory also for
</p>
<p>space groups.
</p>
<p>The Schönflies notation uses the rotation axis and mirror plane symmetry elements that we have
</p>
<p>discussed in Sect. 1.4.2, albeit with differing notation, but introduces the alternating axis of symmetry
</p>
<p>in place of the roto-inversion axis.
</p>
<p>B.1 Alternating Axis of Symmetry
</p>
<p>A crystal is said to have an alternating axis of symmetry Sn of degree n, if it can be brought from one state
</p>
<p>to another indistinguishable state by the operation of rotation through (360/n)� about the axis and
reflection across a plane normal to that axis, overall a single symmetry operation. It should be stressed
</p>
<p>that this plane is not necessarily a mirror plane in the point group.
</p>
<p>Operations Sn are non-performable physically with models (see Sects. 1.4.1 and 1.4.2). Figure B.1
</p>
<p>shows stereograms for S2 and S4; crystallographically, we recognize them as �1 and �4, respectively.
</p>
<p>The reader should consider what point groups are obtained if the plane of the diagram were a mirror
</p>
<p>plane in point groups S2 and S4.
</p>
<p>B.2 Symmetry Notations
</p>
<p>Rotation axes are symbolized by Cn in the Schönflies notation (cyclic group of degree n); n takes the
</p>
<p>meaning ofR in theHermann&ndash;Mauguin system.Mirror planes are indicated by subscripts v, d, andh; v and
</p>
<p>d refer to mirror planes containing the principal axis, and h indicates a mirror plane normal to that axis. In
</p>
<p>addition, d refers to those vertical planes that are set diagonally, between the crystallographic axes normal
</p>
<p>to the principal axis.
</p>
<p>663</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec11_1">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec11_1</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec10_1">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec10_1</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec11_1">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec11_1</a></div>
</div>
<div class="page"><p/>
<p>The mirror plane symmetry element is denoted by s in the Schönflies system. The symbol Dn
(dihedral group of degree n) is introduced for point groups in which there are n two-fold axes in a
</p>
<p>plane normal to the principal axis of degree n. The cubic point groups are represented through the
</p>
<p>special symbols T (tetrahedral) and O (octahedral). In point group symbols, subscripts h and d are
</p>
<p>used to indicate the presence of horizontal and vertical (dihedral) mirror planes, respectively
</p>
<p>Table B.1 compares the Schönflies and Hermann&ndash;Mauguin symmetry notations.
</p>
<p>Fig. B.1 Stereograms of point groups: (a) S2, (b) S4
</p>
<p>Table B.1 Schönflies and Hermann&ndash;Mauguin pointgroup symbols
</p>
<p>Schönflies Hermann&ndash;Mauguina Schönflies Hermann&ndash;Mauguina
</p>
<p>C1 1 D4 422
</p>
<p>C2 2 D6 622
</p>
<p>C3 3 D2h mmm
</p>
<p>C4 4 D3h �6m2
C6 6
</p>
<p>Ci, S2 �1 D4h 4
</p>
<p>m
mm
</p>
<p>Cs, S1 m; 2
</p>
<p>S6 �3
S4 �4 D6h 6
</p>
<p>m
mm
</p>
<p>C3h, S3 �6
a
</p>
<p>C2h 2/m
b D2d �42m
</p>
<p>C4h 4/m
b D3d �3m
</p>
<p>C6h 6/m
b T 23
</p>
<p>C2v mm2 Th m�3m
C3v 3m O 432
</p>
<p>C4v 4mm Td �43m
C6v 6mm Oh m3m
</p>
<p>D2 222 C1v 1
D3 32 D1h 1=m&eth; �1&THORN;
aThe usual Schönflies symbol for �6 is C3h (3/m). The reason that 3/m
is not used in the Hermann&ndash;Mauguin system is that point groups
</p>
<p>containing the element �6 describe crystals that belong to the hexago-
nal system rather than to the trigonal system; �6 cannot operate on a
rhombohedral lattice.
</p>
<p>bR/m is an acceptable way of writing
R
</p>
<p>m
; but R/mmm is not as
</p>
<p>satisfactory as
R
</p>
<p>m
mm; R/mmm is a marginally acceptable alternative.
</p>
<p>664 Appendix B</p>
<p/>
</div>
<div class="page"><p/>
<p>Appendix C: Cartesian Coordinates
</p>
<p>In calculations that lead to results in absolute measure, such as bond distance and angle calculations
</p>
<p>and location of hydrogen-atom positions, it may be desirable to convert the crystallographic
</p>
<p>fractional coordinates x, y, z, which are dimensionless, to Cartesian (orthogonal) coordinates X, Y,
</p>
<p>and Z, in Å or nm.
</p>
<p>C.1 Cartesian to Crystallographic Transformation and Its Inverse
</p>
<p>Instead of considering immediately the transformation A &frac14; Ma, it is simpler to consider first the
inverse transformation a &frac14; M�1 a, where M is the transformation matrix for the triplet A&eth;A;B;C&THORN; to
the triplet a&eth;a; b; c&THORN;, because the components of a along the Cartesian axes are direction cosines (see
Web Appendix WA1).
</p>
<p>Figure C.1 illustrates the two sets of axes. Let A be a unit vector along a, B a unit vector normal to
a, and in the a, b plane, and C a unit vector normal to both A and B.
</p>
<p>Then, we can write
</p>
<p>a=a
</p>
<p>b=b
</p>
<p>c=c
</p>
<p>2
64
</p>
<p>3
75 &frac14;
</p>
<p>l1 m1 n1
</p>
<p>l2 m2 n2
</p>
<p>l3 m3 n3
</p>
<p>2
64
</p>
<p>3
75
</p>
<p>A
</p>
<p>B
</p>
<p>C
</p>
<p>2
64
</p>
<p>3
75 (C.1)
</p>
<p>From the figure, we can write down some of the elements of M�1:
</p>
<p>M�1 &frac14;
1 0 0
</p>
<p>cos g sin g 0
</p>
<p>cos b m3 n3
</p>
<p>2
64
</p>
<p>3
75 (C.2)
</p>
<p>From the properties of direction cosines, we have
</p>
<p>cos a &frac14; l2l3 &thorn; m2m3 &thorn; n2n3 &frac14; cos b cos g&thorn; m3 sin g
</p>
<p>so that
</p>
<p>m3 &frac14; &eth;cos a� cos b cos g&THORN;= sin g &frac14; � cos a� sin b (C.3)
</p>
<p>665</p>
<p/>
</div>
<div class="page"><p/>
<p>Since the sums of the squares of the direction cosines is unity,
</p>
<p>n23 &frac14; 1� cos2b� sin2bcos2a� &frac14; sin2bsin2a�
</p>
<p>so that
</p>
<p>n3 &frac14; sin b sin a
� &frac14; v= sin g (C.4)
</p>
<p>since1 V &frac14; abc sin a* sin b sin g, and v here refers to the volume of the unit parallelepiped a/a, b/b,
c/c, that is, v &frac14; (1 � cos2 a � cos2 b � cos2 g + 2 cos a cos b cos g). Hence, we can write the
transformation in terms of the direct unit-cell parameters, multiplying the lines of the matrix by a, b,
</p>
<p>or c, as appropriate:
</p>
<p>a
</p>
<p>b
</p>
<p>c
</p>
<p>2
664
</p>
<p>3
775 &frac14;
</p>
<p>a 0 0
</p>
<p>b cos g b sin g 0
</p>
<p>c cos b c&eth;cos a� cos b cos g&THORN;= sin g cv= sin g
</p>
<p>2
664
</p>
<p>3
775
</p>
<p>A
</p>
<p>B
</p>
<p>C
</p>
<p>2
664
</p>
<p>3
775 (C.5)
</p>
<p>which, in matrix notation, is a &frac14; M�1 A. From the transformations discussed in Sect. 2.5.5, we have
X &frac14; &eth;M�1&THORN;Tx, or
</p>
<p>X
</p>
<p>Y
</p>
<p>Z
</p>
<p>2
64
</p>
<p>3
75 &frac14;
</p>
<p>a b cos g c cosb
</p>
<p>0 b sin g c&eth;cos a� cos b cos g&THORN;= sin g
0 0 cv= sin g
</p>
<p>2
64
</p>
<p>3
75
</p>
<p>x
</p>
<p>y
</p>
<p>z
</p>
<p>2
64
</p>
<p>3
75 (C.6)
</p>
<p>The deduction of M, the inverse of M�1, is straightforward for a 3 � 3 matrix, albeit somewhat
laborious, and can be found in most elementary treatments of vectors. Thus, we have A &frac14; M a and
x &frac14; MTX, where
</p>
<p>Fig. C.1 A, B and C are unit vectors on Cartesian (orthogonal) axes X, Y, Z, and a/a, b/b, and c/c are unit vectors on the
conventional crystallographic axes x, y, z
</p>
<p>1Buerger MJ (1942) X-ray crystallography. Wiley, New York.
</p>
<p>666 Appendix C</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec13_2">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec13_2</a></div>
</div>
<div class="page"><p/>
<p>M &frac14;
1=a 0 0
</p>
<p>� cos g=&eth;a sin g&THORN; 1=&eth;b sin g&THORN; 0
&eth;cos g cos a� cos b&THORN;=&eth;av sin b&THORN; &eth;cos g cos b� cos a&THORN;=&eth;bv sin g&THORN; sin g=&eth;cv&THORN;
</p>
<p>0
B@
</p>
<p>1
CA (C.7)
</p>
<p>The transformation (C.6) is employed in the program INTXYZ (see Sect. 13.6.6) for the calcula-
</p>
<p>tion of bond lengths, bond angles, and torsion angles from crystallographic parameters. The sign of a
</p>
<p>torsion angle is governed by the convention discussed in Section 8.5.2. For the sequence of atoms,
</p>
<p>P, Q, R, S in Fig. C.2, the torsion angle wPQRS is positive if a clockwise rotation of PQ about QR,
</p>
<p>as seen along QR, brings PQ over RS.
</p>
<p>Fig. C.2 Convention for torsion angles: wPQRS is reckoned positive as shown, when the atom succession P � Q � R
� S is viewed along QR
</p>
<p>Appendix C 667</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec31_13">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec31_13</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec37_8">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec37_8</a></div>
</div>
<div class="page"><p/>
<p>Appendix D: Crystallographic Software
</p>
<p>This Appendix lists software for X-ray and neutron crystallographic applications that are available to the
</p>
<p>academic community. The list is not exhaustive, and many of the packages are listed under one or other
</p>
<p>section of the Collaborative Computational Projects.2,3 Often, the program systems are mirrored by the
</p>
<p>Engineering and Physical Sciences Research Council (EPSRC) funded CCP projects,4 which have
</p>
<p>mirror sites in the U. S. A. and in Canada. In addition to the programs referenced here, a complete set
</p>
<p>of crystallographic programs has been promulgated elsewhere.5
</p>
<p>The program systems are divided into a number of sections, and an appropriate reference has been
</p>
<p>provided for each entry, including author, e-mail address, and web site reference as appropriate.
</p>
<p>&bull; Single Crystal Suites
</p>
<p>&bull; Single Crystal Structure Solving Programs
</p>
<p>&bull; Single Crystal Twinning Software
</p>
<p>&bull; Freestanding Structure Visualization Software
</p>
<p>&bull; Powder Diffraction Data: Powder Indexing Suites
</p>
<p>&bull; Structure Solution from Powder Diffraction Data
</p>
<p>&bull; Software for Macromolecular Crystallography
</p>
<p>Data Processing; Fourier and Structure Factor Calculations; Molecular Replacement; Single
</p>
<p>and Double Isomorphous Replacement; Software for Packing and Molecular; Geometry; Software
</p>
<p>for Graphics and Model Building; Software for Molecular Graphics and Display; Software for
</p>
<p>Refinement; Software for Molecular Dynamics and Energy Minimization; Data Bases
</p>
<p>&bull; Bioinformatics
</p>
<p>Molecular Modelling Software; External Links; Useful Homepages
</p>
<p>D.1 Single Crystal Suites
</p>
<p>Most single crystal program suites have a large variety of functionality; WinGX is an example of a
</p>
<p>suite linking to several other programs in a seamless manner via graphical user interfaces. In most
</p>
<p>cases, programs link to multiple versions of a structure solution program, such as SHELXS-97 or
</p>
<p>SIR2008.
</p>
<p>2 http://www.ccp4.ac.uk.
3 http://www.ccp14.ac.uk.
4 http://www.epsrc.ac.uk/Pages/default.aspx.
5 http://ww1.iucr.org/sincris-top/logiciel/lmno.html#O.
</p>
<p>669</p>
<p/>
<div class="annotation"><a href="http://www.ccp4.ac.uk">http://www.ccp4.ac.uk</a></div>
<div class="annotation"><a href="http://www.ccp14.ac.uk">http://www.ccp14.ac.uk</a></div>
<div class="annotation"><a href="http://www.epsrc.ac.uk/Pages/default.aspx">http://www.epsrc.ac.uk/Pages/default.aspx</a></div>
<div class="annotation"><a href="http://ww1.iucr.org/sincris-top/logiciel/lmno.html#O">http://ww1.iucr.org/sincris-top/logiciel/lmno.html#O</a></div>
</div>
<div class="page"><p/>
<p>D.2 Single Crystal Structure Solution Programs
</p>
<p>CAOS
</p>
<p>Automated Patterson method. Spagna R et al. http://www.ic.cnr.it/caos/what.html
</p>
<p>CRYSTALS 14.23
</p>
<p>Watkin D. http://www.xtl.ox.ac.uk
</p>
<p>DIRDIF 2008
</p>
<p>Automated Patterson methods and fragment searching: Windows version ported by L. Farrugia and
</p>
<p>available via the WinGX website. http://www.chem.gla.ac.uk/~louis/software/dirdif/
</p>
<p>OLEX2
</p>
<p>User-friendly structure solution and refinement suite with inter alia archiving and report generation.
</p>
<p>Dolomanov OV et al (2008) J Appl Crystallogr 42:339. http://olex2.org
</p>
<p>PATSEE
</p>
<p>Fragment searching methods. http://www.ccp14.ac.uk/ccp/web-mirrors/patsee/egert/html/patsee.
</p>
<p>html. Windows version by Farrugia, L. and available via the WinGX website.
</p>
<p>System S
</p>
<p>SHELXS, DIRDIF, SIR, and CRUNCH for solution; EXOR, DIRDIF, SIR, and CRUNCH for autobuild-
</p>
<p>ing; SHELXL for refinement. Spek AL. http://www.ccp14.ac.uk/tutorial/platon/index.html
</p>
<p>SIR 2008
</p>
<p>http://www.ba.ic.cnr.it/content/il-milione-and-sir2008
</p>
<p>SNB (SHAKE AND BAKE)
</p>
<p>Direct methods. Weeks CM et al. http://www.hwi.buffalo.edu/SnB
</p>
<p>WinGX
</p>
<p>SHELXS, DIRDIF, SIR, and PATSEE for solution; DIRDIF phases for autobuilding; SHELX for
</p>
<p>refinement. Farrugia L. http://www.chem.gla.ac.uk/~louis/software/wingx
</p>
<p>D.3 Single Crystal Twinning Software
</p>
<p>TWIN 3.0
</p>
<p>Kahlenberg V et al. http://www.ccp14.ac.uk/solution/twinning/index.html
</p>
<p>TwinRotMac
</p>
<p>Spek AL. http://www.ccp14.ac.uk/solution/ twinning/index.html
</p>
<p>Windows version by Farrugia L. http://www.chem.gla.ac.uk/~louis/software/platon
</p>
<p>D.4 Freestanding Structure Visualization Software
</p>
<p>ORTEP-III
</p>
<p>Burnett MN et al. http://www.chem.gla.ac.uk/~louis/software/ortep3/
</p>
<p>670 Appendix D</p>
<p/>
<div class="annotation"><a href="http://www.ic.cnr.it/caos/what.html">http://www.ic.cnr.it/caos/what.html</a></div>
<div class="annotation"><a href="http://www.xtl.ox.ac.uk">http://www.xtl.ox.ac.uk</a></div>
<div class="annotation"><a href="http://www.chem.gla.ac.uk/~louis/software/dirdif/">http://www.chem.gla.ac.uk/~louis/software/dirdif/</a></div>
<div class="annotation"><a href="http://olex2.org">http://olex2.org</a></div>
<div class="annotation"><a href="http://www.ccp14.ac.uk/ccp/web-mirrors/patsee/egert/html/patsee.html">http://www.ccp14.ac.uk/ccp/web-mirrors/patsee/egert/html/patsee.html</a></div>
<div class="annotation"><a href="http://www.ccp14.ac.uk/ccp/web-mirrors/patsee/egert/html/patsee.html">http://www.ccp14.ac.uk/ccp/web-mirrors/patsee/egert/html/patsee.html</a></div>
<div class="annotation"><a href="http://www.ccp14.ac.uk/tutorial/platon/index.html">http://www.ccp14.ac.uk/tutorial/platon/index.html</a></div>
<div class="annotation"><a href="http://www.ba.ic.cnr.it/content/il-milione-and-sir2008">http://www.ba.ic.cnr.it/content/il-milione-and-sir2008</a></div>
<div class="annotation"><a href="http://www.hwi.buffalo.edu/SnB">http://www.hwi.buffalo.edu/SnB</a></div>
<div class="annotation"><a href="http://www.chem.gla.ac.uk/~louis/software/wingx">http://www.chem.gla.ac.uk/~louis/software/wingx</a></div>
<div class="annotation"><a href="http://www.ccp14.ac.uk/solution/twinning/index.html">http://www.ccp14.ac.uk/solution/twinning/index.html</a></div>
<div class="annotation"><a href="http://www.ccp14.ac.uk/solution/twinning/index.html">http://www.ccp14.ac.uk/solution/twinning/index.html</a></div>
<div class="annotation"><a href="http://www.chem.gla.ac.uk/~louis/software/platon">http://www.chem.gla.ac.uk/~louis/software/platon</a></div>
<div class="annotation"><a href="http://www.chem.gla.ac.uk/~louis/software/ortep3/">http://www.chem.gla.ac.uk/~louis/software/ortep3/</a></div>
</div>
<div class="page"><p/>
<p>D.5 Powder Diffraction Data: Powder Indexing Suites (Dedicated and Other)
</p>
<p>Checkcell
</p>
<p>Laugier J. http://www.ccp14.ac.uk/tutorial/lmgp/achekcelld.htm
</p>
<p>CRYSFIRE
</p>
<p>Shirley R. http://www.ccp14.ac.uk/tutorial/crys/ (includes the programs ITO, DICVOL, TREOR,
</p>
<p>TAUP, KOHL, LZON, LOSH, and FJZN, but is no longer under development)
</p>
<p>DICVOL91
</p>
<p>Louër D. http://www.ccp14.ac.uk/tutorial/crys/program/dicvol91.htm
</p>
<p>ITO12/13
</p>
<p>Visser JW. http://www.iucr.org/resources/commissions/crystallographic-computing/software-museum
</p>
<p>ITO15 (Included in FULLPROF)
</p>
<p>Visser J et al. http://www.ill.eu/sites/fullprof/php/programs.html
</p>
<p>LOSH/LZON
</p>
<p>Bergmann J et al. http://www.ccp14.ac.uk/tutorial/tutorial.htm
</p>
<p>TAUP/Powder
</p>
<p>Taupin D. http://www.ccp14.ac.uk/tutorial/crys/taup.htm
</p>
<p>TREOR90 (Included in FULLPROF)
</p>
<p>http://www.ill.eu/sites/fullprof/php/programsdc cc.html?pagina&frac14;Treor90
</p>
<p>D.6 Powder Pattern Decomposition
</p>
<p>ALLHKL
</p>
<p>Pawley GS. http://www.ccp14.ac.uk/solution/pawley/index.html
</p>
<p>WPPF
</p>
<p>Hatashi S, Toraya H. http://www.icdd.com/resources/axa/vol41/V41_66.pdf
</p>
<p>D.7 Structure Solution from Powder Diffraction Data
</p>
<p>ESPOIR
</p>
<p>Mileur M, Le Bail A. http://www.cristal.org/sdpd/espoir/
</p>
<p>EXTRA (Included in EXPO)
</p>
<p>http://www.ccp14.ac.uk/tutorial/expo/index.html
</p>
<p>FULLPROF
</p>
<p>http://www.ill.eu/sites/fullprof/
</p>
<p>Appendix D 671</p>
<p/>
<div class="annotation"><a href="http://www.ccp14.ac.uk/tutorial/lmgp/achekcelld.htm">http://www.ccp14.ac.uk/tutorial/lmgp/achekcelld.htm</a></div>
<div class="annotation"><a href="http://www.ccp14.ac.uk/tutorial/crys/">http://www.ccp14.ac.uk/tutorial/crys/</a></div>
<div class="annotation"><a href="http://www.ccp14.ac.uk/tutorial/crys/program/dicvol91.htm">http://www.ccp14.ac.uk/tutorial/crys/program/dicvol91.htm</a></div>
<div class="annotation"><a href="http://www.iucr.org/resources/commissions/crystallographic-computing/software-museum">http://www.iucr.org/resources/commissions/crystallographic-computing/software-museum</a></div>
<div class="annotation"><a href="http://www.ill.eu/sites/fullprof/php/programs.html">http://www.ill.eu/sites/fullprof/php/programs.html</a></div>
<div class="annotation"><a href="http://www.ccp14.ac.uk/tutorial/tutorial.htm">http://www.ccp14.ac.uk/tutorial/tutorial.htm</a></div>
<div class="annotation"><a href="http://www.ccp14.ac.uk/tutorial/crys/taup.htm">http://www.ccp14.ac.uk/tutorial/crys/taup.htm</a></div>
<div class="annotation"><a href="http://www.ill.eu/sites/fullprof/php/programsdccc.html?pagina=Treor90">http://www.ill.eu/sites/fullprof/php/programsdccc.html?pagina=Treor90</a></div>
<div class="annotation"><a href="http://www.ill.eu/sites/fullprof/php/programsdccc.html?pagina=Treor90">http://www.ill.eu/sites/fullprof/php/programsdccc.html?pagina=Treor90</a></div>
<div class="annotation"><a href="http://www.ccp14.ac.uk/solution/pawley/index.html">http://www.ccp14.ac.uk/solution/pawley/index.html</a></div>
<div class="annotation"><a href="http://www.icdd.com/resources/axa/vol41/V41_66.pdf">http://www.icdd.com/resources/axa/vol41/V41_66.pdf</a></div>
<div class="annotation"><a href="http://www.cristal.org/sdpd/espoir/">http://www.cristal.org/sdpd/espoir/</a></div>
<div class="annotation"><a href="http://www.ccp14.ac.uk/tutorial/expo/index.html">http://www.ccp14.ac.uk/tutorial/expo/index.html</a></div>
<div class="annotation"><a href="http://www.ill.eu/sites/fullprof/">http://www.ill.eu/sites/fullprof/</a></div>
</div>
<div class="page"><p/>
<p>GSAS
</p>
<p>http://www.ccp14.ac.uk/solution/gsas
</p>
<p>RIETAN
</p>
<p>Izumi F. http://homepage.mac.com/fujioizumi/download/download_Eng.html
</p>
<p>SIRPOW (Included in EXPO)
</p>
<p>http://www.ccp14.ac.uk/tutorial/expo/index.html
</p>
<p>POWDER SOLVE
</p>
<p>http://accelrys.com/resource-center/case-studies/powder-solve.html
</p>
<p>D.8 Software for Macromolecular Crystallography
</p>
<p>Much of the software listed in this section is fast moving, and the CCP sites1,2 should be consulted for
</p>
<p>the latest developments.
</p>
<p>D.8.1 Data Processing
</p>
<p>HKL 4 (Includes DENZO, XDISPLAY, and SCALEPACK)
</p>
<p>Gerwith D (2003) The HKL manual, 6th edn. http://www.hkl-xray.com/hkl_web1/hkl/manual_
</p>
<p>online.pdf
</p>
<p>STRATEGY
</p>
<p>Ravelli RBG et al. http://www.crystal.chem.uu.nl/distr/strategy.html
</p>
<p>PREDICT
</p>
<p>Noble M. http://biop.ox.ac.uk/www/distrib/predict.html
</p>
<p>D.8.2 Fourier and Structure Factor Calculations
</p>
<p>SFALL (Structure Factors). http://www.ccp4.ac.uk/html/sfall.html
</p>
<p>FFT (Fast Fourier Transform). http://www.ccp4.ac.uk/html/fft.html
</p>
<p>D.8.3 Molecular Replacement
</p>
<p>AmoRe
</p>
<p>Navaza J (Autostruct 2001). http://www.ccp4.ac.uk/autostruct/amore/
</p>
<p>CNS Solve 1.1
</p>
<p>Brünger AT et al. http://cns.csb.yale.edu/v1.1/
</p>
<p>MOLREP
</p>
<p>Vagin AA. alexei@ysbl.york.ac.uk
</p>
<p>672 Appendix D</p>
<p/>
<div class="annotation"><a href="http://www.ccp14.ac.uk/solution/gsas">http://www.ccp14.ac.uk/solution/gsas</a></div>
<div class="annotation"><a href="http://homepage.mac.com/fujioizumi/download/download_Eng.html">http://homepage.mac.com/fujioizumi/download/download_Eng.html</a></div>
<div class="annotation"><a href="http://www.ccp14.ac.uk/tutorial/expo/index.html">http://www.ccp14.ac.uk/tutorial/expo/index.html</a></div>
<div class="annotation"><a href="http://accelrys.com/resource-center/case-studies/powder-solve.html">http://accelrys.com/resource-center/case-studies/powder-solve.html</a></div>
<div class="annotation"><a href="http://www.hkl-xray.com/hkl_web1/hkl/manual_online.pdf">http://www.hkl-xray.com/hkl_web1/hkl/manual_online.pdf</a></div>
<div class="annotation"><a href="http://www.hkl-xray.com/hkl_web1/hkl/manual_online.pdf">http://www.hkl-xray.com/hkl_web1/hkl/manual_online.pdf</a></div>
<div class="annotation"><a href="http://www.crystal.chem.uu.nl/distr/strategy.html">http://www.crystal.chem.uu.nl/distr/strategy.html</a></div>
<div class="annotation"><a href="http://biop.ox.ac.uk/www/distrib/predict.html">http://biop.ox.ac.uk/www/distrib/predict.html</a></div>
<div class="annotation"><a href="http://www.ccp4.ac.uk/html/sfall.html">http://www.ccp4.ac.uk/html/sfall.html</a></div>
<div class="annotation"><a href="http://www.ccp4.ac.uk/html/fft.html">http://www.ccp4.ac.uk/html/fft.html</a></div>
<div class="annotation"><a href="http://www.ccp4.ac.uk/autostruct/amore/">http://www.ccp4.ac.uk/autostruct/amore/</a></div>
<div class="annotation"><a href="http://cns.csb.yale.edu/v1.1/">http://cns.csb.yale.edu/v1.1/</a></div>
<div class="annotation"><a href="http://alexei@ysbl.york.ac.uk">http://alexei@ysbl.york.ac.uk</a></div>
</div>
<div class="page"><p/>
<p>MOLPACK
</p>
<p>Wang D et al. http://www.ccp4.ac.uk/html/molrep.html
</p>
<p>REPLACE
</p>
<p>http://como.bio.columbia.edu/tong/Public/Replace/replace.html
</p>
<p>D.8.4 Schematic Structure Plots
</p>
<p>LIGPLOT
</p>
<p>Laskowski RA. http://www.ebi.ac.uk/thornton-srv/software/LIGPLOT/
</p>
<p>SHELXS-86
</p>
<p>Location of heavy-atom positions. Sheldrick GM (1994) Crystallographic computing, 3rd edn.
</p>
<p>Oxford University Press, Oxford
</p>
<p>D.8.5 Software for Packing, Molecular Geometry, Validation and Deposition
</p>
<p>COOT
</p>
<p>Emsley P et al (2010) Acta Cryst D66:486. http://lmb.bioch.ox.ac.uk/coot/
</p>
<p>PROCHECK
</p>
<p>Laskowski RS et al. http://www.ebi.ac.uk/thornton-srv/software/PROCHECK
</p>
<p>WHATCHECK
</p>
<p>Hooft RWW et al. http://www.ccp4.ac.uk/dist/ccp4i/help/modules/valdep.html
</p>
<p>D.8.6 Software for Graphics and Model Building
</p>
<p>FRODO
</p>
<p>Jones TA. http://www.mendeley.com/research/tek-frodo-new-version-frodo-tektronix-graphics-sta-
</p>
<p>tions/
</p>
<p>O
</p>
<p>Jones TA et al. http://xray0.princeton.edu/~phil/Facility/ono.html
</p>
<p>TURBO-FRODO
</p>
<p>Jones TA et al. Bio-graphics. http://www.afmb.univ-mrs.fr/-TURBO-
</p>
<p>D.8.7 Software for Molecular Graphics and Display
</p>
<p>MERCURY
</p>
<p>http://www.ccdc.cam.ac.uk/products/csd_system/ mercury_csd/index.php
</p>
<p>ORTEP
</p>
<p>Barnes CL (1997) ORTEP-3 for Windows, J Appl Cryst 30:568 [based on ORTEP-III by Johnson CK
</p>
<p>and Burnett MN.]
</p>
<p>Appendix D 673</p>
<p/>
<div class="annotation"><a href="http://www.ccp4.ac.uk/html/molrep.html">http://www.ccp4.ac.uk/html/molrep.html</a></div>
<div class="annotation"><a href="http://como.bio.columbia.edu/tong/Public/Replace/replace.html">http://como.bio.columbia.edu/tong/Public/Replace/replace.html</a></div>
<div class="annotation"><a href="http://www.ebi.ac.uk/thornton-srv/software/LIGPLOT/">http://www.ebi.ac.uk/thornton-srv/software/LIGPLOT/</a></div>
<div class="annotation"><a href="http://lmb.bioch.ox.ac.uk/coot/l">http://lmb.bioch.ox.ac.uk/coot/l</a></div>
<div class="annotation"><a href="http://www.ebi.ac.uk/thornton-srv/software/PROCHECK">http://www.ebi.ac.uk/thornton-srv/software/PROCHECK</a></div>
<div class="annotation"><a href="http://www.ccp4.ac.uk/dist/ccp4i/help/modules/valdep.html">http://www.ccp4.ac.uk/dist/ccp4i/help/modules/valdep.html</a></div>
<div class="annotation"><a href="http://www.mendeley.com/research/tek-frodo-new-version-frodo-tektronix-graphics-stations/">http://www.mendeley.com/research/tek-frodo-new-version-frodo-tektronix-graphics-stations/</a></div>
<div class="annotation"><a href="http://www.mendeley.com/research/tek-frodo-new-version-frodo-tektronix-graphics-stations/">http://www.mendeley.com/research/tek-frodo-new-version-frodo-tektronix-graphics-stations/</a></div>
<div class="annotation"><a href="http://xray0.princeton.edu/~phil/Facility/ono.html">http://xray0.princeton.edu/~phil/Facility/ono.html</a></div>
<div class="annotation"><a href="http://www.afmb.univ-mrs.fr/-TURBO-">http://www.afmb.univ-mrs.fr/-TURBO-</a></div>
<div class="annotation"><a href="http://www.ccdc.cam.ac.uk/products/csd_system/mercury_csd/index.php">http://www.ccdc.cam.ac.uk/products/csd_system/mercury_csd/index.php</a></div>
</div>
<div class="page"><p/>
<p>RASMOL
</p>
<p>Sayle R. http://www.umass.edu/microbio/rasmol/
</p>
<p>RASTER 3.0
</p>
<p>Bacon DJ et al. http://skuld.bmsc.washington.edu/raster3d
</p>
<p>SETOR
</p>
<p>Evans SV. http://www.ncbi.nlm.nih.gov/pubmed/8347566
</p>
<p>MOLSCRIPT 1.4
</p>
<p>Kraulis PJ. http://www.avatar.se/molscript
</p>
<p>BOBSCRIPT 2.4 (Extension to MOLSCRIPT 1.4)
</p>
<p>Esnouf R. http://www.csb.yale.edu/userguides/graphics/bobscript/bobscript.html
</p>
<p>D.8.8 Software for Refinement
</p>
<p>X-PLOR 3.1
</p>
<p>Brünger AT. http://yalepress.yale.edu/book.asp?isbn&frac14;9780300054026
CNS Solve 1.1
</p>
<p>Brünger AT et al. http://cns.csb.yale.edu/v1.1/
</p>
<p>RESTRAIN
</p>
<p>Driessen HPC et al. http://scripts.iucr.org/cgi-bin/paper?gl0109
</p>
<p>SHELXS-86
</p>
<p>Sheldrick GM (1994) Crystallographic computing, 3rd edn. Oxford University Press, Oxford
</p>
<p>SHELX-97 and SHELXL-97
</p>
<p>Sheldrick GM. http://shelx.uni-ac.gwdg.de/SHELX/
</p>
<p>REFMAC 5
</p>
<p>http://www.ccp4.ac.uk/html/refmac5.html
</p>
<p>D.8.9 Software for Molecular Dynamics and Energy Minimization
</p>
<p>SYBYL-X
</p>
<p>http://tripos.com/index.php?family&frac14;modules,SimplePage,,,&amp;page&frac14;SYBYL-X
</p>
<p>D.8.10 Data Bases
</p>
<p>Protein Data Bank (PDB)
</p>
<p>http://www.pdb.org/pdb/static.do?p&frac14;search/index.html
Basic Local Alignment Search Tool (BLAST)
</p>
<p>http://blast.ncbi.nlm.nih.gov/Blast.cgi?PAGE&frac14; Proteins
Cambridge Crystallographic Data Centre (CCDC)
</p>
<p>http://www.ccdc.cam.ac.uk
</p>
<p>674 Appendix D</p>
<p/>
<div class="annotation"><a href="http://www.umass.edu/microbio/rasmol/">http://www.umass.edu/microbio/rasmol/</a></div>
<div class="annotation"><a href="http://skuld.bmsc.washington.edu/raster3d">http://skuld.bmsc.washington.edu/raster3d</a></div>
<div class="annotation"><a href="http://www.ncbi.nlm.nih.gov/pubmed/8347566">http://www.ncbi.nlm.nih.gov/pubmed/8347566</a></div>
<div class="annotation"><a href="http://www.avatar.se/molscript">http://www.avatar.se/molscript</a></div>
<div class="annotation"><a href="http://www.csb.yale.edu/userguides/graphics/bobscript/bobscript.html">http://www.csb.yale.edu/userguides/graphics/bobscript/bobscript.html</a></div>
<div class="annotation"><a href="http://yalepress.yale.edu/book.asp?isbn=9780300054026">http://yalepress.yale.edu/book.asp?isbn=9780300054026</a></div>
<div class="annotation"><a href="http://yalepress.yale.edu/book.asp?isbn=9780300054026">http://yalepress.yale.edu/book.asp?isbn=9780300054026</a></div>
<div class="annotation"><a href="http://cns.csb.yale.edu/v1.1/">http://cns.csb.yale.edu/v1.1/</a></div>
<div class="annotation"><a href="http://scripts.iucr.org/cgi-bin/paper?gl0109">http://scripts.iucr.org/cgi-bin/paper?gl0109</a></div>
<div class="annotation"><a href="http://shelx.uni-ac.gwdg.de/SHELX/">http://shelx.uni-ac.gwdg.de/SHELX/</a></div>
<div class="annotation"><a href="http://www.ccp4.ac.uk/html/refmac5.html">http://www.ccp4.ac.uk/html/refmac5.html</a></div>
<div class="annotation"><a href="http://tripos.com/index.php?family=modules,SimplePage,,,&amp;page=SYBYL-X">http://tripos.com/index.php?family=modules,SimplePage,,,&amp;page=SYBYL-X</a></div>
<div class="annotation"><a href="http://tripos.com/index.php?family=modules,SimplePage,,,&amp;page=SYBYL-X">http://tripos.com/index.php?family=modules,SimplePage,,,&amp;page=SYBYL-X</a></div>
<div class="annotation"><a href="http://tripos.com/index.php?family=modules,SimplePage,,,&amp;page=SYBYL-X">http://tripos.com/index.php?family=modules,SimplePage,,,&amp;page=SYBYL-X</a></div>
<div class="annotation"><a href="http://www.pdb.org/pdb/static.do?p=search/index.html">http://www.pdb.org/pdb/static.do?p=search/index.html</a></div>
<div class="annotation"><a href="http://www.pdb.org/pdb/static.do?p=search/index.html">http://www.pdb.org/pdb/static.do?p=search/index.html</a></div>
<div class="annotation"><a href="http://blast.ncbi.nlm.nih.gov/Blast.cgi?PAGE=Proteins">http://blast.ncbi.nlm.nih.gov/Blast.cgi?PAGE=Proteins</a></div>
<div class="annotation"><a href="http://blast.ncbi.nlm.nih.gov/Blast.cgi?PAGE=Proteins">http://blast.ncbi.nlm.nih.gov/Blast.cgi?PAGE=Proteins</a></div>
<div class="annotation"><a href="http://www.ccdc.cam.ac.uk">http://www.ccdc.cam.ac.uk</a></div>
</div>
<div class="page"><p/>
<p>ReLiBase (Finds Ligands for Protein Families)
</p>
<p>http://www.ccdc.cam.ac.uk/free_services/reliba se_free/
</p>
<p>ChemSpider (Contains Much Chemical Information on ca. 25 Million Compounds)
</p>
<p>http://cs.m.chemspider.com
</p>
<p>D.8.11 Synchrotron Web Page
</p>
<p>http://www.esrf.eu/computing/scientific/people/srio/publications/SPIE04_XOP.pdf
</p>
<p>D.9 Bioinformatics
</p>
<p>D.9.1 Molecular Modelling Software
</p>
<p>The sources listed below provide software for molecular modelling. Some of them, for example,
</p>
<p>COSMOS and Sybyl are listed above. Others are readily obtained from the web sites that are given by
</p>
<p>the names, for example, Abaloneclassical: http://www.sciencedirect.com/science/article/pii/
</p>
<p>S0928493110002894.
</p>
<p>The following names may be interrogated in a similar manner:
</p>
<p>&bull; Abaloneclassical
&bull; ADFquantum
&bull; AMBERclassical
&bull; Ascalaph Designerclassical and quantum [http://en.wikipedia.org/wiki/Main_Page#cite_note-0]
</p>
<p>&bull; AutoDock
</p>
<p>&bull; AutoDock Vina
</p>
<p>&bull; BALLView
</p>
<p>&bull; Biskit
</p>
<p>&bull; BOSSclassical
&bull; Cerius2
</p>
<p>&bull; CHARMMclassical
&bull; Chimera
</p>
<p>&bull; Coot [http://en.wikipedia.org/wiki/Main_Pa ge#cite_note-1]
</p>
<p>&bull; COSMOS (software) [http://en.wikipedia.org/wiki/Main_Page#cite_note-2]
</p>
<p>&bull; CP2Kquantum
&bull; CPMDquantum
&bull; Culgi
</p>
<p>&bull; Discovery Studioclassical and quantum [http://en.wikipedia.org/wiki/Main_Page#cite_note-3]
</p>
<p>&bull; DOCKclassical
&bull; Fireflyquantum
&bull; FoldX
</p>
<p>&bull; GAMESS (UK)quantum
&bull; GAMESS (US)quantum
&bull; GAUSSIANquantum
&bull; Ghemical
</p>
<p>&bull; Gorgon [http://en.wikipedia.org/wiki/Main_ Page#cite_note-4]
</p>
<p>&bull; GROMACSclassical
&bull; GROMOSclassical
</p>
<p>Appendix D 675</p>
<p/>
<div class="annotation"><a href="http://www.ccdc.cam.ac.uk/free_services/relibase_free/">http://www.ccdc.cam.ac.uk/free_services/relibase_free/</a></div>
<div class="annotation"><a href="http://cs.m.chemspider.com">http://cs.m.chemspider.com</a></div>
<div class="annotation"><a href="http://www.esrf.eu/computing/scientific/people/srio/publications/SPIE04_XOP.pdf">http://www.esrf.eu/computing/scientific/people/srio/publications/SPIE04_XOP.pdf</a></div>
<div class="annotation"><a href="http://www.sciencedirect.com/science/article/pii/S0928493110002894">http://www.sciencedirect.com/science/article/pii/S0928493110002894</a></div>
<div class="annotation"><a href="http://www.sciencedirect.com/science/article/pii/S0928493110002894">http://www.sciencedirect.com/science/article/pii/S0928493110002894</a></div>
<div class="annotation"><a href="http://en.wikipedia.org/wiki/Main_Page#cite_note-0">http://en.wikipedia.org/wiki/Main_Page#cite_note-0</a></div>
<div class="annotation"><a href="http://en.wikipedia.org/wiki/Main_Page#cite_note-1">http://en.wikipedia.org/wiki/Main_Page#cite_note-1</a></div>
<div class="annotation"><a href="http://en.wikipedia.org/wiki/Main_Page#cite_note-2">http://en.wikipedia.org/wiki/Main_Page#cite_note-2</a></div>
<div class="annotation"><a href="http://en.wikipedia.org/wiki/Main_Page#cite_note-3">http://en.wikipedia.org/wiki/Main_Page#cite_note-3</a></div>
<div class="annotation"><a href="http://en.wikipedia.org/wiki/Main_Page#cite_note-4">http://en.wikipedia.org/wiki/Main_Page#cite_note-4</a></div>
</div>
<div class="page"><p/>
<p>&bull; InsightIIclassical and quantum
&bull; LAMMPSclassical
&bull; Lead Finderclassical [http://en.wikipedia.org/wiki/Main_Page#cite_note-5]
</p>
<p>&bull; LigandScout
</p>
<p>&bull; MacroModelclassical
&bull; MADAMM [http://en.wikipedia.org/wiki/Main_Page#cite_note-6; http://en.wikipedia.org/wiki/
</p>
<p>Main_Page#cite_note-Cerqueira-7]
</p>
<p>&bull; MarvinSpace [http://en.wikipedia.org/wiki/Main_Page#cite_note-8]
</p>
<p>&bull; Materials and Processes Simulations [http://en.wikipedia.org/wiki/Main_Page#cite_note-9]
</p>
<p>&bull; Materials Studioclassical and quantum [http://en.wikipedia.org/wiki/Main_Page#cite_note-10]
</p>
<p>&bull; MDynaMixclassical
&bull; MMTK
</p>
<p>&bull; Molecular Docking Server
</p>
<p>&bull; Molecular Operating Environment
</p>
<p>(MOE)classical and quantum
&bull; MolIDEhomology modelling [http://en.wikipedia.org/wiki/Main_Page#cite_note-11]
</p>
<p>&bull; Molsoft ICM [http://en.wikipedia.org/wiki/Main_Page#cite_note-12]
</p>
<p>&bull; MOPACquantum
&bull; NAMDclassical
&bull; NOCH
</p>
<p>&bull; Oscail X
</p>
<p>&bull; PyMOLvisualization
&bull; Q-Chemquantum
&bull; ReaxFF
</p>
<p>&bull; ROSETTA
</p>
<p>&bull; SCWRLside-chain prediction [http://en.wikipedia.org/wiki/Main_Page#cite_note-13]
</p>
<p>&bull; Sirius
</p>
<p>&bull; Spartan (software)quantum [http://en.wikipedia.org/wiki/Main_Page#cite_note-14]
</p>
<p>&bull; StruMM3D (STR3DI32) [http://en.wikipedia.org/wiki/Main_Page#cite_note-15]
</p>
<p>&bull; Sybyl (software)classical [http://en.wikipedia.org/wiki/Main_Page#cite_note-16]
</p>
<p>&bull; MCCCS Towhee [http://en.wikipedia.org/wiki/Main_Page#cite_note-17]
</p>
<p>&bull; TURBOMOLEquantum
&bull; VMDvisualization
&bull; VLifeMDSIntegrated molecular modelling and simulation
&bull; WHAT IF [http://en.wikipedia.org/wiki/Main_Page#cite_note-18]
</p>
<p>&bull; xeo [http://en.wikipedia.org/wiki/Main_Page #cite_note-19]
</p>
<p>&bull; YASARA [http://en.wikipedia.org/wiki/Main_Page#cite_note-20]
</p>
<p>&bull; Zodiac (software) [http://en.wikipedia.org/wiki/Main_Page#cite_note-21]
</p>
<p>D.9.2 External Links
</p>
<p>These links relate to important sites on molecular modelling and molecular simulation, but are by no
</p>
<p>means exhaustive.
</p>
<p>Center for Molecular Modelling at the National Institutes of Health (NIH) (U.S. Government
</p>
<p>Agency): http://www.bing.com/search?q&frac14;Center+for+Molecular+Modelling+at++the+National
+Institutes+of+Health+%28NIH%29+%28U.S.+Government+Agency%29&amp;src&frac14;ie9tr
</p>
<p>676 Appendix D</p>
<p/>
<div class="annotation"><a href="http://en.wikipedia.org/wiki/Main_Page#cite_note-5">http://en.wikipedia.org/wiki/Main_Page#cite_note-5</a></div>
<div class="annotation"><a href="http://en.wikipedia.org/wiki/Main_Page#cite_note-6">http://en.wikipedia.org/wiki/Main_Page#cite_note-6</a></div>
<div class="annotation"><a href="http://en.wikipedia.org/wiki/Main_Page#cite_note-Cerqueira-7">http://en.wikipedia.org/wiki/Main_Page#cite_note-Cerqueira-7</a></div>
<div class="annotation"><a href="http://en.wikipedia.org/wiki/Main_Page#cite_note-Cerqueira-7">http://en.wikipedia.org/wiki/Main_Page#cite_note-Cerqueira-7</a></div>
<div class="annotation"><a href="http://en.wikipedia.org/wiki/Main_Page#cite_note-8">http://en.wikipedia.org/wiki/Main_Page#cite_note-8</a></div>
<div class="annotation"><a href="http://en.wikipedia.org/wiki/Main_Page#cite_note-9">http://en.wikipedia.org/wiki/Main_Page#cite_note-9</a></div>
<div class="annotation"><a href="http://en.wikipedia.org/wiki/Main_Page#cite_note-10">http://en.wikipedia.org/wiki/Main_Page#cite_note-10</a></div>
<div class="annotation"><a href="http://en.wikipedia.org/wiki/Main_Page#cite_note-11">http://en.wikipedia.org/wiki/Main_Page#cite_note-11</a></div>
<div class="annotation"><a href="http://en.wikipedia.org/wiki/Main_Page#cite_note-12">http://en.wikipedia.org/wiki/Main_Page#cite_note-12</a></div>
<div class="annotation"><a href="http://en.wikipedia.org/wiki/Main_Page#cite_note-13">http://en.wikipedia.org/wiki/Main_Page#cite_note-13</a></div>
<div class="annotation"><a href="http://en.wikipedia.org/wiki/Main_Page#cite_note-14">http://en.wikipedia.org/wiki/Main_Page#cite_note-14</a></div>
<div class="annotation"><a href="http://en.wikipedia.org/wiki/Main_Page#cite_note-15">http://en.wikipedia.org/wiki/Main_Page#cite_note-15</a></div>
<div class="annotation"><a href="http://en.wikipedia.org/wiki/Main_Page#cite_note-16">http://en.wikipedia.org/wiki/Main_Page#cite_note-16</a></div>
<div class="annotation"><a href="http://en.wikipedia.org/wiki/Main_Page#cite_note-17">http://en.wikipedia.org/wiki/Main_Page#cite_note-17</a></div>
<div class="annotation"><a href="http://en.wikipedia.org/wiki/Main_Page#cite_note-18">http://en.wikipedia.org/wiki/Main_Page#cite_note-18</a></div>
<div class="annotation"><a href="http://en.wikipedia.org/wiki/Main_Page#cite_note-19">http://en.wikipedia.org/wiki/Main_Page#cite_note-19</a></div>
<div class="annotation"><a href="http://en.wikipedia.org/wiki/Main_Page#cite_note-20">http://en.wikipedia.org/wiki/Main_Page#cite_note-20</a></div>
<div class="annotation"><a href="http://en.wikipedia.org/wiki/Main_Page#cite_note-21">http://en.wikipedia.org/wiki/Main_Page#cite_note-21</a></div>
<div class="annotation"><a href="http://www.bing.com/search?q=Center+for+Molecular+Modelling+at++the+National+Institutes+of+Health+%28NIH%29+%28U.S.+Government+Agency%29&amp;src=ie9tr">http://www.bing.com/search?q=Center+for+Molecular+Modelling+at++the+National+Institutes+of+Health+%28NIH%29+%28U.S.+Government+Agency%29&amp;src=ie9tr</a></div>
<div class="annotation"><a href="http://www.bing.com/search?q=Center+for+Molecular+Modelling+at++the+National+Institutes+of+Health+%28NIH%29+%28U.S.+Government+Agency%29&amp;src=ie9tr">http://www.bing.com/search?q=Center+for+Molecular+Modelling+at++the+National+Institutes+of+Health+%28NIH%29+%28U.S.+Government+Agency%29&amp;src=ie9tr</a></div>
<div class="annotation"><a href="http://www.bing.com/search?q=Center+for+Molecular+Modelling+at++the+National+Institutes+of+Health+%28NIH%29+%28U.S.+Government+Agency%29&amp;src=ie9tr">http://www.bing.com/search?q=Center+for+Molecular+Modelling+at++the+National+Institutes+of+Health+%28NIH%29+%28U.S.+Government+Agency%29&amp;src=ie9tr</a></div>
<div class="annotation"><a href="http://www.bing.com/search?q=Center+for+Molecular+Modelling+at++the+National+Institutes+of+Health+%28NIH%29+%28U.S.+Government+Agency%29&amp;src=ie9tr">http://www.bing.com/search?q=Center+for+Molecular+Modelling+at++the+National+Institutes+of+Health+%28NIH%29+%28U.S.+Government+Agency%29&amp;src=ie9tr</a></div>
</div>
<div class="page"><p/>
<p>Molecular Simulation, details for theMolecular Simulation journal, ISSN: 0892-7022 (print), 1029-
</p>
<p>0435 (online): http://www.bing.com/sear ch?q&frac14;Center+for+MolecularModelling+at+the+National
+Institutes+of+Health+%28NIH%29+%28U.S.+Government+Agency%29&amp;src&frac14;ie9tr
</p>
<p>The Cheminfo Network and Community of Practice in Informatics and Modelling: http://www.
</p>
<p>bing.com/search?q&frac14;The+Cheminfo+Network+and+Community+of+Practice+in+Informatics
+and+Modelling.&amp;src&frac14;ie9tr
</p>
<p>D.9.3 Useful Homepages
</p>
<p>These sites relate to situations wherein extensive work on protein crystallography is being persued.
</p>
<p>Again, it is not an exhaustive list.
</p>
<p>York Structural Biology Laboratory
</p>
<p>http://www.york.ac.uk/chemistry/research/groups/ ysbl/
</p>
<p>COSMOS&mdash;Computer Simulation of Molecular Structures
</p>
<p>http://www.mybiosoftware.com/3d-molecular-model/1968
</p>
<p>Accelrys Inc.
</p>
<p>http://accelrys.com/
</p>
<p>http://www.ccp4.ac.uk
</p>
<p>http://www.ccp14.ac.uk
</p>
<p>http://epsrc.ac.uk/Pages/default.aspx
</p>
<p>http://ww1.iusr.org/sincris-top/logicel/Imno.html#O
</p>
<p>Appendix D 677</p>
<p/>
<div class="annotation"><a href="http://www.bing.com/search?q=Center+for+MolecularModelling+at+the+National+Institutes+of+Health+%28NIH%29+%28U.S.+Government+Agency%29&amp;src=ie9tr">http://www.bing.com/search?q=Center+for+MolecularModelling+at+the+National+Institutes+of+Health+%28NIH%29+%28U.S.+Government+Agency%29&amp;src=ie9tr</a></div>
<div class="annotation"><a href="http://www.bing.com/search?q=Center+for+MolecularModelling+at+the+National+Institutes+of+Health+%28NIH%29+%28U.S.+Government+Agency%29&amp;src=ie9tr">http://www.bing.com/search?q=Center+for+MolecularModelling+at+the+National+Institutes+of+Health+%28NIH%29+%28U.S.+Government+Agency%29&amp;src=ie9tr</a></div>
<div class="annotation"><a href="http://www.bing.com/search?q=Center+for+MolecularModelling+at+the+National+Institutes+of+Health+%28NIH%29+%28U.S.+Government+Agency%29&amp;src=ie9tr">http://www.bing.com/search?q=Center+for+MolecularModelling+at+the+National+Institutes+of+Health+%28NIH%29+%28U.S.+Government+Agency%29&amp;src=ie9tr</a></div>
<div class="annotation"><a href="http://www.bing.com/search?q=Center+for+MolecularModelling+at+the+National+Institutes+of+Health+%28NIH%29+%28U.S.+Government+Agency%29&amp;src=ie9tr">http://www.bing.com/search?q=Center+for+MolecularModelling+at+the+National+Institutes+of+Health+%28NIH%29+%28U.S.+Government+Agency%29&amp;src=ie9tr</a></div>
<div class="annotation"><a href="http://www.bing.com/search?q=The+Cheminfo+Network+and+Community+of+Practice+in+Informatics+and+Modelling.&amp;src=ie9tr">http://www.bing.com/search?q=The+Cheminfo+Network+and+Community+of+Practice+in+Informatics+and+Modelling.&amp;src=ie9tr</a></div>
<div class="annotation"><a href="http://www.bing.com/search?q=The+Cheminfo+Network+and+Community+of+Practice+in+Informatics+and+Modelling.&amp;src=ie9tr">http://www.bing.com/search?q=The+Cheminfo+Network+and+Community+of+Practice+in+Informatics+and+Modelling.&amp;src=ie9tr</a></div>
<div class="annotation"><a href="http://www.bing.com/search?q=The+Cheminfo+Network+and+Community+of+Practice+in+Informatics+and+Modelling.&amp;src=ie9tr">http://www.bing.com/search?q=The+Cheminfo+Network+and+Community+of+Practice+in+Informatics+and+Modelling.&amp;src=ie9tr</a></div>
<div class="annotation"><a href="http://www.bing.com/search?q=The+Cheminfo+Network+and+Community+of+Practice+in+Informatics+and+Modelling.&amp;src=ie9tr">http://www.bing.com/search?q=The+Cheminfo+Network+and+Community+of+Practice+in+Informatics+and+Modelling.&amp;src=ie9tr</a></div>
<div class="annotation"><a href="http://www.bing.com/search?q=The+Cheminfo+Network+and+Community+of+Practice+in+Informatics+and+Modelling.&amp;src=ie9tr">http://www.bing.com/search?q=The+Cheminfo+Network+and+Community+of+Practice+in+Informatics+and+Modelling.&amp;src=ie9tr</a></div>
<div class="annotation"><a href="http://www.york.ac.uk/chemistry/research/groups/ysbl/">http://www.york.ac.uk/chemistry/research/groups/ysbl/</a></div>
<div class="annotation"><a href="http://www.mybiosoftware.com/3d-molecular-model/1968">http://www.mybiosoftware.com/3d-molecular-model/1968</a></div>
<div class="annotation"><a href="http://accelrys.com/">http://accelrys.com/</a></div>
<div class="annotation"><a href="http://www.ccp4.ac.uk">http://www.ccp4.ac.uk</a></div>
<div class="annotation"><a href="http://www.ccp14.ac.uk">http://www.ccp14.ac.uk</a></div>
<div class="annotation"><a href="http://epsrc.ac.uk/Pages/default.aspx">http://epsrc.ac.uk/Pages/default.aspx</a></div>
<div class="annotation"><a href="http://ww1.iusr.org/sincris-top/logicel/Imno.html#O">http://ww1.iusr.org/sincris-top/logicel/Imno.html#O</a></div>
</div>
<div class="page"><p/>
<p>Appendix E: Structure Invariants, Structure
Seminvariants, Origin and Enantiomorph
Specifications
</p>
<p>E.1 Structure Invariants
</p>
<p>As we have seen in Sect. 2.2.2, there is an infinite number of ways in which a crystal unit cell may be
</p>
<p>chosen. Conventionally, however, any crystal lattice is represented by one of the 14 Bravais lattices
</p>
<p>described in Sect. 2.2.3. For a given unit cell, the origin of the x, y, and z coordinates can be relocated
</p>
<p>for convenience, as we have seen in Sect. 2.7.7 for space group P212121. The possible effects of such
</p>
<p>origin transformations were mentioned in Sect. 6.6.4, when discussing of Fourier transforms. As a
</p>
<p>general rule, the origin of a given space group is chosen with respect to its symmetry elements; for
</p>
<p>example, in centrosymmetric space groups the origin is specified on a center of symmetry. Conven-
</p>
<p>tions associated with the specification of the origin are fully described for all space groups in the
</p>
<p>literature. With no symmetry elements apart from the lattice translations, space group P1 is the
</p>
<p>exception and can accommodate an origin of coordinates in any arbitrary position. We discuss here
</p>
<p>relationships between structure factors that arise from changes in the location of the coordinate origin.
</p>
<p>Following (3.63) we write the structure factor in the form
</p>
<p>F&eth;h&THORN; &frac14;
X
</p>
<p>j
</p>
<p>fj exp&eth;i2ph � rj&THORN; (E.1)
</p>
<p>where h represents a reciprocal lattice vector corresponding to reflection hkl and rj is the real space
vector corresponding to the point x, y, z, so that h � rj &frac14; hxj &thorn; kyj &thorn; lzj. If the origin is changed to the
point r0, then (E.1) becomes
</p>
<p>F&eth;h&THORN;r0 &frac14;
X
</p>
<p>j
</p>
<p>fj exp&frac12;i2ph � &eth;rj � r0&THORN;�
</p>
<p>&frac14;
X
</p>
<p>j
</p>
<p>fj exp&eth;i2ph � rj&THORN; exp&eth;�i2ph � r0&THORN;
(E.2)
</p>
<p>679</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec4_2">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec4_2</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec5_2">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec5_2</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec22_2">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec22_2</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec17_6">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec17_6</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Equ63_3">http://dx.doi.org/10.1007/978-1-4614-3954-7#Equ63_3</a></div>
</div>
<div class="page"><p/>
<p>Thus, we can write
</p>
<p>F&eth;h&THORN;r0 &frac14; F&eth;h&THORN; exp&eth;�i2ph � r0&THORN; (E.3)
</p>
<p>so that
</p>
<p>jF&eth;h&THORN;r0 j &frac14; jF&eth;h&THORN;j (E.4)
</p>
<p>and
</p>
<p>f&eth;h&THORN;r0 &frac14; f&eth;h&THORN;r � 2ph � r0 (E.5)
</p>
<p>Thus, a change of origin leaves the amplitude of the structure factor unaltered, but changes the
</p>
<p>phase by � 2ph � r0whatever the value of r0. The relationships (E.3)&ndash;(E.5) apply equally to the
normalized structure factors E(hkl) that are used in direct methods of phase determination. We can
</p>
<p>illustrate (E.3)&ndash;(E.5) by a simple example.
</p>
<p>Consider an atom at 0.3, 0.2, 0.7 in space group P1. For a reflection, say 213, and taking f as 1.0,
</p>
<p>we find A01 &frac14; 0.8090, B
0
1 &frac14; �0.5878, so that jF1j &frac14; 1 and f1 &frac14; 324�. We change the origin to the
</p>
<p>point 0.1, 0.1, 0.1, whereupon A02 &frac14; �0.3090, B02 &frac14; 0.9511, so that jF2j &frac14; 1 and f2 &frac14; 108�.
Finally, using the third term in (E.5), we find Df &frac14; 2ph � r0
&frac14; 360 &frac12;2 � &eth;0:1&THORN; &thorn; 1� &eth;0:1&THORN; &thorn; 3�&eth;0:1&THORN;� &frac14; 216, which is equal to f1 � f2. (Remember to set
tan�1(B0/A0) in the correct quadrant according to the signs of A0 and B0, and to evaluate f in the
positive range 0&ndash;2p.)
</p>
<p>The values of jEj are determined by the structure, whatever the origin, whereas the values of f are
determined by both the structure and the choice of origin. Thus, the values of jEj alone cannot determine
unique values for the phases.We need a process to obtain phases from the values of jEj that incorporates a
specification of the origin. Consider the product of three normalized structure factors in the absence of
</p>
<p>symmetry, that is, for space group P1. From (3.15), we can write
</p>
<p>E1E2E3 &frac14; jE1jjE2jjE3j exp&frac12;i&eth;f&eth;h1&THORN;
&thorn; &eth;f&eth;h2&THORN; &thorn; &eth;f&eth;h3&THORN;� (E.6)
</p>
<p>If the origin is moved from 0,0,0 to a point r0, it follows from the foregoing that (E.6) becomes
</p>
<p>E1E2E3 &frac14; jE1jjE2jjE3j exp&frac12;�i2p&eth;h1 &thorn; h2 &thorn; h3&THORN; � r0� (E.7)
</p>
<p>Thus, the condition that the product of three structure factors be a structure invariant N3, that is, a
</p>
<p>change of origin has no effect on its value in the non-centrosymmetric space group P1, is that
</p>
<p>h1 &thorn; h2 &thorn; h3 &frac14; 0 (E.8)
</p>
<p>Equation (E.8) is a triplet structure invariant; it may be extended to a quartet such that the product
</p>
<p>of four structure factors is a structure invariant N4 if
</p>
<p>h1 &thorn; h2 &thorn; h3 &thorn; h4 &frac14; 0 (E.9)
</p>
<p>680 Appendix E</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Equ15_3">http://dx.doi.org/10.1007/978-1-4614-3954-7#Equ15_3</a></div>
</div>
<div class="page"><p/>
<p>and more:
</p>
<p>N&THORN;n &frac14;
Yn
</p>
<p>j&frac14;1
E&eth;hj&THORN; (E.10)
</p>
<p>provided that the condition
</p>
<p>Xn
</p>
<p>j&frac14;1
hj &frac14; 0 (E.11)
</p>
<p>For n &frac14; 1, the structure factor, which is a structure invariant, is E(0) and has a phase of zero for any
origin. For n &frac14; 2, h1 + h2 &frac14; 0, or h2 &frac14; �h1 so that E1E2 &frac14; E(h1)E(�h1) &frac14; jE(h1)j2, which is phase
independent. For n &gt; 2, we have (E.10) and (E.11) as already discussed. For n &frac14; 3 or more, we have
equations such as (E.8) and (E.9).
</p>
<p>E.2 Structure Seminvariants
</p>
<p>Equations such as (E.8) apply also to P�1, because the sums of the indices, as in (E.13) below, are each
</p>
<p>zero. However, consider next a structure with symmetry P�1, wherein the origin is chosen, normally,
</p>
<p>on one of the eight centers of symmetry unique to the unit cell. In the presence of symmetry elements,
</p>
<p>it is always desirable to choose the origin on one of these elements, albeit such a choice may not
</p>
<p>define the origin point uniquely, such as on the twofold axis parallel to the line [0,y,0] in space group
</p>
<p>P2.
</p>
<p>The normally permitted origins in P�1 are listed in Table 8.2. In general, the sign of E(hkl) depends
</p>
<p>on the choice of origin except for reflection in the group eee, for which reflections6
</p>
<p>&eth;hkl&THORN; modulo2 &eth;222&THORN; &frac14; &eth;000&THORN; (E.12)
</p>
<p>Such reflections are structure seminvariants (semi-invariants) since their signs (phases) do not
</p>
<p>change for variation among the permitted origins. If three structure factors are chosen from different
</p>
<p>parity groups, other than eee, such that
</p>
<p>h1 &thorn; h2 &thorn; h3; k1 &thorn; k2 &thorn; k3; l1 &thorn; l2 &thorn; l3 modulo &eth;222&THORN; (E.13)
</p>
<p>then the product of the three structure factors is not a structure seminvariant (semi-invariant), and can
</p>
<p>be either positive or negative. An arbitrary sign can be chosen for each such structure factor in the
</p>
<p>product, and for one of the eight possible origins the choice will be true, and the origin is fixed
</p>
<p>according to that choice. Thus, for example, the reflections 10�6, 40�1, and 71�4 may be chosen to
</p>
<p>specify an origin, and if we allocate a + sign arbitrarily to each, the origin is defined as 0, 0, 0. If we
</p>
<p>choose instead the reflections 10�6, 40�1, and �507, then the origin is not specified uniquely because the
</p>
<p>determinant is less than or equal to zero. The triplet is not linearly independent (see E.14 and text):
</p>
<p>6 a � b modulo n if a � b &frac14; kn, where k is an integer.
</p>
<p>Appendix E 681</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Tab2_8">http://dx.doi.org/10.1007/978-1-4614-3954-7#Tab2_8</a></div>
</div>
<div class="page"><p/>
<p>1 &thorn; 4 � 5 &frac14; 0, 0 &thorn; 0 &thorn; 0 &frac14; 0 and �6 � 1 &thorn; 7 &frac14; 0, or oee + eeo + oeo &frac14; oee, which does not
constitute linear independence. The relation h1 + h2 + h3 &frac14; 0 modulo (222) has no special signifi-
cance in space group P1.
</p>
<p>The structure invariants and structure seminvariants have been well described in the literature for
</p>
<p>all space groups.7&ndash;11
</p>
<p>E.2.1 Difference Between Structure Invariant and Structure Seminvariant
</p>
<p>Consider two triplets E&eth;3�32&THORN;E&eth;012&THORN;E&eth;�32�4&THORN; and E&eth;3�32&THORN;E&eth;012&THORN;E&eth;344&THORN;. The first product is a structure
invariant because the sums h1 + h2 + h3, k1 + k2 + k3 and l1 + l2 + l3 are each equal to zero. It is a
</p>
<p>structure invariant inP1 andP�1wherever the origin point is placed in the unit cell. The second product is
</p>
<p>a structure seminvariant because the sums h1 + h2 + h3, k1 + k2 + k3 and l1 + l2 + l3 are each equal to
</p>
<p>zero modulo (2), and its sign (phase) is not changed by moving to another permitted origin in P�1, but it
</p>
<p>would change if the origin were moved to a general point in the unit cell. Note that in both examples,
</p>
<p>these reflections would not serve to specify an origin because the parities sum to eee in each case.
</p>
<p>E.3 Origin Specification
</p>
<p>From the foregoing, we see that for space group P1, which contains no symmetry other than that of
</p>
<p>the basic translations, three reflections that form a linearly independent combination will specify the
</p>
<p>origin. The three reflections E(h1k1l1), E(h2k2l2), and E(h3k3l3) will specify an origin provided that the
</p>
<p>determinant D satisfies the condition
</p>
<p>D &frac14;
h1 k1 l1
h2 k2 l2
h3 k3 l3
</p>
<p>�
</p>
<p>�
</p>
<p>�
</p>
<p>�
</p>
<p>�
</p>
<p>�
</p>
<p>�
</p>
<p>�
</p>
<p>�
</p>
<p>�
</p>
<p>�
</p>
<p>�
</p>
<p>&gt; 0 (E.14)
</p>
<p>or D modulo (222) &frac14; 	1; the determinant is evaluated in the normal manner.
</p>
<p>Normally, the position 0, 0, 0 is chosen for the origin in P1; there is no purpose in choosing any
</p>
<p>other site. The three independent phases can be given values between 0 and 2p; generally they are
</p>
<p>chosen as zero.
</p>
<p>In any space group of symmetry greater than 1, the origin is normally chosen on that symmetry
</p>
<p>element. We have discussed the case for P�1 sufficiently for our purposes in Sect. 8.2.2.
</p>
<p>E.4 Choice of Enantiomorph
</p>
<p>In any of the 65 enantiomorphous space groups listed in Table 10.1, there exists the need to specify a
</p>
<p>molecular enantiomorph. From (E.1) we can write
</p>
<p>7Hauptman H, Karle J (1953) The solution of the phase problem I, ACA monograph 3.
8 idem. (1956) Acta Crystallogr 9:45.
9 idem. ibid. (1959) 12:93.
10Karle J, Hauptman H (1961) ibid. 14:217.
11Lessinger L, Wondratschek H (1975) ibid. A31:382.
</p>
<p>682 Appendix E</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec4_8">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec4_8</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Tab1_10">http://dx.doi.org/10.1007/978-1-4614-3954-7#Tab1_10</a></div>
</div>
<div class="page"><p/>
<p>jE&eth;h&THORN;j &frac14;
X
</p>
<p>j
</p>
<p>Zj exp&frac12;i2p&eth;h � rj&THORN;�
�
</p>
<p>�
</p>
<p>�
</p>
<p>�
</p>
<p>�
</p>
<p>�
</p>
<p>�
</p>
<p>�
</p>
<p>�
</p>
<p>�
</p>
<p>(E.15)
</p>
<p>If each rj is replaced by its inverse, the right-hand side of (E.15) and, hence, jE(h)j remain unchanged.
The jEj values relate to both a structure and its inverse, or roto-reflection, through a point. If this point is
the origin 0, 0, 0, then the structure factors are E(h) and its conjugate E*(h) and its phases are f(h) and
�f(h). Thus, the two values for a structure invariant differ only in sign.
</p>
<p>If a structure invariant phase is 0 or p, then it has the same value for both enantiomorphs. If a
</p>
<p>structure invariant is enantiomorph-sensitive, then its value differs significantly from 0 or p, and its
</p>
<p>value may be specified arbitrarily within this range, generally a value of p/2, p/4, or 3p/4. Of course,
</p>
<p>the structure determined may not correspond to the true chemical configuration and that problem must
</p>
<p>be addressed (see Sect. 7.6.1). The selection of an enantiomorph has been discussed in a practical
</p>
<p>manner through the structure analysis in Sect. 8.2.10.
</p>
<p>Appendix E 683</p>
<p/>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec31_7">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec31_7</a></div>
<div class="annotation"><a href="http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec11_8">http://dx.doi.org/10.1007/978-1-4614-3954-7#Sec11_8</a></div>
</div>
<div class="page"><p/>
<p>Tutorial Solutions
</p>
<p>Solutions 1
</p>
<p>1.1. Extend CA to cut the x0 axis in H. All angles in the figure are easily calculated (OA = OC = x).
EvaluateOP, or a0 (1.623x), andOH (2.732x). ExpressOH, the required intercept on the x0 axis,
as a fraction of a0 (1.683). The intercept along b0 (and b) remains unaltered, so that the fractional
intercepts of the line CA are 1.683 and 1/2 along x0 and y respectively. Hence, CA has the Miller
indices (0.5941, 2), or (1, 3.366), referred to the oblique axes.
</p>
<p>1.2. (a) h = a/(a/2) = 2, k &frac14; b=&eth;�b=2&THORN; &frac14; �2, l = c/1 = 0; hence &eth;2 �2 0&THORN;. Similarly,
(b) (164) (c) &eth;0 0 �1&THORN; (d) &eth;3 �3 4&THORN; (e) &eth;0 �4 3&THORN; (f) &eth;�4 2 �3&THORN;
</p>
<p>1.3. Use (1.6), (1.7), and (1.8). More simply, set down the planes twice in each of the two rows,
</p>
<p>ignore the first and final indices in each row, and then cross-multiply, similarly to the evaluation
</p>
<p>of a determinant.
</p>
<p>1 2 3 1 2 3
</p>
<p>� � �
0 1 1 0 1 0
</p>
<p>Hence, U = 2 � (�3) = 5, V = 0 � 1 = �1, W = �1 � 0 = �1 so that the zone symbol is
&frac12;5 �1 �1�. If we had written the planes down in the reverse order, we would have obtained &frac12;�5 1 1�.
(What is the interpretation of this result?) Similarly:
</p>
<p>(b) &frac12;3 �5 2� (c) &frac12;�1 �1 �1� (d) [110]
1.4. Use (1.9) or, more simply, set down the procedure as in Solution 1.3, but with zone symbols,
</p>
<p>which leads to &eth;5 2 3&THORN;. This plane and &eth;5 2 3&THORN; are parallel; [UVW] and &frac12;UVW� are coincident.
1.5. Formally, one could write 422, 4 2 �2, 4 �2 2, 4 �2 �2, �4 2 2, �4 2 �2, �4 �2 2, �4 �2 �2. However, the interac-
</p>
<p>tion of two inversion axes leads to an intersecting pure rotation axis, so that all symbols with
</p>
<p>one or three inversion axes are invalid. Now �4 2 �2 and �4 �2 2 are equivalent under rotation of the x
</p>
<p>and y axes in the x, y plane by 45 deg, so that there remain 422, 4 �2 �2, and �4 2 �2 as unique point
</p>
<p>groups. Their standard symbols are 422, 4mm, and �4 2m, respectively. Note that if we do
</p>
<p>postulate a group with the symbol 4 2 �2, for example, it is straightforward to show, with the aid
</p>
<p>of a stereogram, that it is equivalent to, and a non-standard description of
4
</p>
<p>m
mm.
</p>
<p>1.6. (a) mmm (b) 2/m (c) 1
</p>
<p>1.7. Refer to Fig. S1.1 (a) mmm;mmm � �1 (b) 2=m; 2m � �1
</p>
<p>685</p>
<p/>
</div>
<div class="page"><p/>
<p>1.8.
</p>
<p>{010} f1 1 0g f1 1 3g
2/m 2 4 4
</p>
<p>4 2m 4 4 8
</p>
<p>m3 6 12 24
</p>
<p>1.9. (a) 1; (b)m; (c) 2; (d)m; (e) 1; (f) 2; (g) 6; (h) 6mm; (i) 3; (j) 2mm. (Did you remember to use the
</p>
<p>Laue group for each example?)
</p>
<p>1.10. (a) From a thin card, cut out four but identical quadrilaterals; when fitted together, they make a
</p>
<p>(plane) figure of symmetry 2. (b) m. (A beer &ldquo;jug&rdquo; has the same symmetry.) (c) a, 1=m; b, 3;
4
</p>
<p>m
mm; d, 102m; e,
</p>
<p>6
</p>
<p>m
mm; f, m.
</p>
<p>1.11.
</p>
<p>(a) �6m 2 D3h
</p>
<p>(b) 4
</p>
<p>m
mm
</p>
<p>D4h
</p>
<p>(c) m�3m Oh
</p>
<p>(d) �4 3m Td
</p>
<p>(e) 3m C3v
</p>
<p>(f) 1 C1
</p>
<p>(g) 6
</p>
<p>m
mm
</p>
<p>D6h
</p>
<p>(h) mm2 C2v
</p>
<p>(i) mmm D2h
</p>
<p>(j) mm2 C2v
</p>
<p>(k) 2 C2
</p>
<p>(l) 3 C3
</p>
<p>(m) 1 Ci
</p>
<p>(n) 3 S6
</p>
<p>(o) 4 S4
</p>
<p>(p) m Cs
</p>
<p>(q) 6 C3h
</p>
<p>(r) 2/m C2h
</p>
<p>(s) 222 D2
</p>
<p>(t) 422 D4
</p>
<p>(u) 4mm C4v
</p>
<p>(v) 4 2m D2d
</p>
<p>Fig. S2.1
</p>
<p>686 Tutorial Solutions</p>
<p/>
</div>
<div class="page"><p/>
<p>1.12. Remember first to project the general form of the point group on to a plane of the given form,
</p>
<p>and then relate the projected symmetry to one of the two-dimensional point groups. In some
</p>
<p>cases, you will have more than one set of representative points in two dimensions.
</p>
<p>(a) 2 (b) m (c) 1 (d) m (e) 1 (f) 1 (g) 3 (h)3m (i) 3 (j) 2mm
</p>
<p>1.13. (10), (01), &eth;1 0&THORN;, &eth;0 1&THORN;. They are the same for the parallelogram, provided that the axes are
chosen parallel to the sides of the figure.
</p>
<p>1.14. Refer to Fig. S1.2, and from the definition of Miller indices: OA = a/h; OB = b/k. Let the plane
</p>
<p>(hkil) intercept the u axis at p; drawDE parallel to AO. SinceOD bisects&lt;AOB, AOD = 60 deg,
</p>
<p>so that DODE is equilateral; hence OD &frac14; DE &frac14; OE &frac14; p. Triangles EBD and OBA are similar;
hence EB/DE = OB/OA = (b/k)/(a/h). Now EB = b/k � p, and from the above, it follows that
p &frac14; ab=&eth;ak&thorn; bh&THORN;. Since a = b = u, from the symmetry, u/p = h &thorn; k. We write u/p as�i, since
p lies on the negative side of the u axis (OD = �u/p), so that
</p>
<p>i &frac14; �&eth;h&thorn; k&THORN;
</p>
<p>1.15. Refer to Chap. 1, Fig. P1.6; the points ACGEmark out one of the diagonalm planes of the cube.
</p>
<p>From the symmetry of the cube, the currents through the resistors have the values as shown.
</p>
<p>Hence, any path through the cube from A to G has a resistance of 5/6 O.
</p>
<p>Fig. S1.2
</p>
<p>Tutorial Solutions 687</p>
<p/>
</div>
<div class="page"><p/>
<p>Solutions 2
</p>
<p>2.1. The translations, equal to the lengths of the two sides of any parallelogram unit, repeat the
</p>
<p>molecule ad infinitum in the two dimensions shown. A two-fold rotation point placed at any
</p>
<p>corner of a parallelogram is, itself, repeated by the same translations (see Fig. P2.1).
</p>
<p>(i) The two-fold rotation points lie at each corner, half-way along each edge and at the
</p>
<p>geometrical center of each parallelogram unit.
</p>
<p>(ii) There are fourunique two-fold points per parallelogramunit: one at a corner, one at the center of
</p>
<p>each of two non-collinear edges, and one at the geometrical center.
</p>
<p>2.2.
</p>
<p>(i) (ii)
</p>
<p>(a) 4mm 6mm
</p>
<p>(b) Square Hexagonal
</p>
<p>(c) (i) If unit cell is centered, then another square can be drawn to form a conventional unit cell
</p>
<p>of half the area of the centered unit cell.
</p>
<p>(ii) If unit cell is centered it is no longer hexagonal; each point is degraded to the 2mm
</p>
<p>symmetry of the rectangular system, and may be described by a conventional p unit cell.
</p>
<p>The transformation equations in each example are:
</p>
<p>a0 &frac14; a=2&thorn; b=2; b0 &frac14; �a=2&thorn; b=2
</p>
<p>Note. A regular hexagon of &ldquo;lattice&rdquo; points with another point placed at its center is not a
</p>
<p>centered hexagonal unit cell: it represents three adjacent p hexagonal unit cells in different
</p>
<p>relative orientations. (Without the point at the center, the hexagon of points is not even a
</p>
<p>lattice.)
</p>
<p>2.3. A C unit cell may be obtained by the transformations:
</p>
<p>aC &frac14; aF; bC &frac14; bF; cC &frac14; �aF=2&thorn; cF=2:
</p>
<p>The new c dimension is obtained from evaluating the dot product:
</p>
<p>&eth;�a=2&thorn; c=2&THORN; � &eth;�a=2&thorn; c=2&THORN;
</p>
<p>to give c0 5.7627 Å; a and b are unchanged. The angle b0 in the transformed unit cell is obtained
by evaluating
</p>
<p>cos b 0 &frac14; a &middot; &eth;�a=2&thorn; c=2&THORN;=a 0c 0 &frac14; &eth;�a&thorn; c cos b&THORN;=&eth;2c 0&THORN;
</p>
<p>so that b0 = 139.29�.
VC&eth;C cell&THORN;=VF&eth;F cell&THORN; &frac14; 12. (Count the number of unique lattice points in each cell: each lattice
point is associated with a unique portion of the volume.)
</p>
<p>2.4. (a) The symmetry is no longer tetragonal, although the lattice is true: it is now orthorhombic.
</p>
<p>(b) The tetragonal symmetry is apparently restored, but the lattice is no longer true: the lattice
</p>
<p>points are not all in the same environment in the same orientation.
</p>
<p>(c) A tetragonal F unit cell is formed and represents a true tetragonal lattice.
</p>
<p>688 Tutorial Solutions</p>
<p/>
</div>
<div class="page"><p/>
<p>However, tetragonal F is equivalent to tetragonal I (of smaller volume) under the transfor-
</p>
<p>mation
</p>
<p>aI &frac14; aF=2&thorn; bF=2; bI &frac14; �aF=2&thorn; bF=2; c0I &frac14; cF
</p>
<p>2.5. F unit cell: r2&frac12;312�=Å
2 &frac14; r&frac12;312� � r&frac12;31�2� &frac14; 32a2&thorn;12b2&thorn;22c2&thorn;2 � 3 � &eth;�2&THORN; � 6� 8� cos 110, so
</p>
<p>that r = 28.64 Å. To obtain the value in the C unit cell, we could repeat this calculation with
</p>
<p>the dimensions of the C unit cell, leading to 28.64 Å. Alternatively, we could use the
</p>
<p>transformation matrix to obtain the F equivalent of &frac12;31�2�c, and then use the original F cell
dimensions on it. The matrix for this F cell in terms of the C is:
</p>
<p>S &frac14;
1 0 0
</p>
<p>0 1 0
</p>
<p>1 0 2
</p>
<p>2
4
</p>
<p>3
5 so that &eth;S�1&THORN;T &frac14;
</p>
<p>1 0 � 1
2
</p>
<p>0 1 0
</p>
<p>0 0 1
2
</p>
<p>2
4
</p>
<p>3
5
</p>
<p>Then, &frac12;UVW�F&frac14;&eth;S�1&THORN;
T
�&frac12;UVW�C&frac14;&frac12;41�1�F, so that r&frac12;41�1�F &frac14; 28:64 Å .
</p>
<p>2.6. It is not an eighth crystal system because the symmetry at each lattice point is �1. It is a special
</p>
<p>case of the triclinic system in which the g angle is 90�.
2.7. (a) Plane group c2mm is shown in Fig. S2.1, with the coordinates listed below it.
</p>
<p>(b) Plane group p2mg is shown in Fig. S2.2; this diagram also shows the minimum number of
</p>
<p>motifs p, V, and Z.
</p>
<p>Note that if the symmetry elements are arranged with 2 at the intersection ofm and g, they
</p>
<p>do not form a group. Attempts to draw such an arrangement lead to continued halving of the
</p>
<p>repeat distance parallel to the g line.
</p>
<p>Fig. S2.1
</p>
<p>Tutorial Solutions 689</p>
<p/>
</div>
<div class="page"><p/>
<p>2.8. (a)
</p>
<p>Space group P21/c is shown in Fig. S2.3, on the (010) plane.
</p>
<p>(b) Figure S2.4 shows the molecular formula of biphenyl, excluding the hydrogen atoms.
</p>
<p>The two molecules in the unit cell lie on any set of special positions, Wyckoff (a)&ndash;(d),
</p>
<p>with the center of the C(1)&ndash;C(1)0 bond on �1. Hence, the molecule is centrosymmetric and
planar. The planarity imposes a conjugation on the molecule, including the C(1)&ndash;C(1)0
</p>
<p>bond. (This result is supported by the bond lengths C(1)&ndash;C(1)0 
 1.49 Å and Carom&ndash;-
Carom 
 1.40 Å. In the free-molecule state, the rings rotate about the C(1)&ndash;C(1)
</p>
<p>0 bond to
</p>
<p>the energetically favorable conformation with the ring planes at approximately 45� to
each other).
</p>
<p>Fig. S2.2
</p>
<p>Fig. S2.3
</p>
<p>690 Tutorial Solutions</p>
<p/>
</div>
<div class="page"><p/>
<p>2.9. Each pair of positions forms two vectors, between the origin and the points: 	{(x2 � x1),
(y2 � y1), (z2 � z1)}. Thus, there is a single vector at each of the positions:
</p>
<p>2x; 2y; 2z; 2�x; 2�y; 2�z; 2x; 2�y; 2z; 2�x; 2y; 2�z
</p>
<p>and two superimposed vectors at each of the positions:
</p>
<p>2x; 1=2; 1=2 &thorn; 2z; 0; 1=2 &thorn; 2y; 1=2; 2�x; 1=2; 1=2 � 2z; 0; 1=2 � 2y; 1=2
</p>
<p>Note: � &eth;2x; 1=2; 1=2 &thorn; 2z&THORN; � 2�x; 1=2; 1=2 �2z
2.10.
</p>
<p>Since �x; �y; �z and 2p � x, 2q � y, 2r � z are one and the same point, p = q = r = 0, so that the
three symmetry planes intersect in a center of symmetry at the origin.
</p>
<p>Otherwise, by applying the half-translation rule, T &frac14; a=2&thorn; b=2&thorn; a=2 &thorn;b=2 &frac14; 0. Hence,
the center of symmetry lies at the intersection of the three symmetry planes.
</p>
<p>2.11. Figure S2.5 shows space group Pbam.
</p>
<p>Fig. S2.4
</p>
<p>Fig. S2.5
</p>
<p>Tutorial Solutions 691</p>
<p/>
</div>
<div class="page"><p/>
<p>Coordinates of general equivalent positions
</p>
<p>x; y; z; 1=2 � x; 1=2 � y; z; 1=2 &thorn; x; �y; z; �x; 1=2 &thorn; y; z;
x; y; �z; 1=2 � x; 1=2 � y; �z; 1=2 &thorn; x; �y; �z; �x; 1=2 &thorn; y; �z
Coordinates of centers of symmetry
</p>
<p>1=4; 1=4; 0; 1=4; 3=4; 0; 3=4; 1=4; 0; 3=4; 3=4; 0;
</p>
<p>1=4; 1=4; 1=2; 1=4; 3=4; 1=2; 3=4; 1=4; 1=2; 3=4; 3=4; 1=2
</p>
<p>Change of origin to 1
4
; 1
4
; 0:
</p>
<p>(i) Subtract 1
4
; 1
4
; 0 from the above set of coordinates of general equivalent positions.
</p>
<p>(ii) Let x0 &frac14; x� &frac14;, y0 = y � &frac14;, and z0 = z.
(iii) After making all substitutions, drop the subscript, and rearrange to give:
</p>
<p>	 fx; y; z; x; y; z; 1=2 &thorn; x; 1=2 � y; z; 1=2 � x; 1=2 &thorn; y; zg
</p>
<p>This result may be confirmed by redrawing the space group with the origin on �1.
</p>
<p>2.12. Figure S2.6 shows two adjacent unit cells of space group Pn on the (010) plane. In the
</p>
<p>transformation to Pc, only the c spacing is changed:
</p>
<p>cPc &frac14; �aPn &thorn; cPn
</p>
<p>Hence, Pn � Pc. By interchanging the labels of the x and z axes, which are not constrained by
the two-fold symmetry, we see that Pc � Pa. Note that it is necessary to invert the sign on b, so
as to preserve a right-handed set of axes. The translation a/2 in the C unit cell in Cmmeans that
</p>
<p>Ca � Cm. Since there is no half-translation along c in Cm, Cm is not equivalent to Cc, although
Cc is equivalent to Cn. If the x and z axes in Cc are interchanged, with due attention to b, the
symbol becomes Aa. (The standard symbols among these groups are Pc, Cm, and Cc.)
</p>
<p>2.13. P2/c
</p>
<p>(a) 2/m; monoclinic.
</p>
<p>(b) Primitive unit cell; c-glide plane &perp; b; two-fold axis jj b.
(c) h0l: l = 2n.
</p>
<p>(d) 12/m1 P . c.
</p>
<p>Fig. S2.6
</p>
<p>692 Tutorial Solutions</p>
<p/>
</div>
<div class="page"><p/>
<p>Pca21
(a) mm2; orthorhombic.
</p>
<p>(b) Primitive unit cell; c-glide plane &perp; a; a-glide plane &perp; b; 21 axis jj ic.
(c) 0kl: l = 2n; h0l: h = 2n.
</p>
<p>(d) mmm P c a .
</p>
<p>Cmcm
</p>
<p>(a) mmm; orthorhombic.
</p>
<p>(b) C-face centered unit cell; m plane &perp; a; c-glide plane &perp; b; m plane &perp; c.
</p>
<p>(c) hkl: h + k = 2n; h0l :l = 2n.
</p>
<p>(d) mmm C . c .
</p>
<p>P�421c
</p>
<p>(a) �42m; tetragonal.
</p>
<p>(b) Primitive unit cell; �4 axis jj c; 21 axes jj a and b; c-glide planes &perp; [110] and &frac12;1�10�.
(c) hhl: l = 2n; h00: h = 2n.
</p>
<p>(d)
4
</p>
<p>m
mm P . 21 c
</p>
<p>P6122
</p>
<p>(a) 622; hexagonal.
</p>
<p>(b) Primitive unit cell; 61 axis jj c; two-fold axes jj a, b, and u; two-fold axes 30� to a, b, and u,
and in the (0001) plane.
</p>
<p>(c) 000 l: l = 6n (Similarly for P6522).
</p>
<p>(d)
6
</p>
<p>m
mm P61 . . .
</p>
<p>Pa�3
</p>
<p>(a) m�3; cubic.
</p>
<p>(b) Primitive unit cell; a-glide plane &perp; b (equivalent statements are b-glide plane &perp; c, c-glide
</p>
<p>plane &perp; a); three-fold axes jj [111], &frac12;1�11�, &frac12;�111�, and &frac12;�111�.
(c) 0kl: k = 2n; (equivalent statements are h0l: l = 2n; hk0: h = 2n.)
</p>
<p>(d) m�3 Pa.
</p>
<p>2.14. Plane group p2; the unit cell repeat along b is halved, and g has the particular value of 90�.
Note that, because of the contents of the unit cell, it cannot belong to the rectangular
</p>
<p>two-dimensional system.
</p>
<p>2.15. (a) Refer to Fig. 2.24, number 10, for a cubic P unit cell (vectors a, b, and c).
(b) Tetragonal P
</p>
<p>aP = b/2 &thorn; c/2
bP = �b/2 &thorn; c/2
cP = a
</p>
<p>(c) Monoclinic C
</p>
<p>aC = c
bC = �b
cC = a
</p>
<p>(d) Triclinic P
</p>
<p>aT = a
bT = b/2 &thorn; c/2
cT = �b/2 &thorn; c/2
</p>
<p>Tutorial Solutions 693</p>
<p/>
</div>
<div class="page"><p/>
<p>2.16.
�4 along z m?b
0 1 0
</p>
<p>�1 0 0
</p>
<p>0 0 �1
</p>
<p>2
64
</p>
<p>3
75
</p>
<p>1 0 0
</p>
<p>0 �1 0
</p>
<p>0 0 1
</p>
<p>2
64
</p>
<p>3
75
</p>
<p>R1 R2
</p>
<p>R2R1h = h
0. Forming first R3 = R2R1, remembering the order of multiplication, we then
</p>
<p>evaluate
</p>
<p>0 1 0
</p>
<p>1 0 0
</p>
<p>0 0 �1
</p>
<p>2
64
</p>
<p>3
75
</p>
<p>h
</p>
<p>k
</p>
<p>l
</p>
<p>2
64
</p>
<p>3
75 &frac14;
</p>
<p>k
</p>
<p>h
</p>
<p>�l
</p>
<p>2
64
</p>
<p>3
75
</p>
<p>R3 h h
0
</p>
<p>that is, R3h = h
0, so that h0 &frac14; kh�l; R3 represents a two-fold rotation axis along [110].
</p>
<p>2.17. The matrices are multiplied in the usual way, and the components of the translation vectors are
</p>
<p>added, resulting in
</p>
<p>�1 0 0
</p>
<p>0 �1 0
</p>
<p>0 0 1
</p>
<p>2
4
</p>
<p>3
5&thorn;
</p>
<p>1=2
1=2
1=2
</p>
<p>2
4
</p>
<p>3
5
</p>
<p>which corresponds to a 21 axis along 1=4;&frac12; 1=4; z�. The space group symbol is Pna21.
2.18. (a) We can see from the hexagonal stereograms (Fig. 1.32) that 2 32 � 6. Hence the matrix for
</p>
<p>63 about [0, 0, z] is
</p>
<p>1 1 0
</p>
<p>1 0 1
</p>
<p>0 0 1
</p>
<p>2
4
</p>
<p>3
5&thorn;
</p>
<p>0
</p>
<p>0
1=2
</p>
<p>2
4
</p>
<p>3
5
</p>
<p>and that for the c- glide is
</p>
<p>0 1 0
</p>
<p>1 0 0
</p>
<p>0 0 1
</p>
<p>2
4
</p>
<p>3
5&thorn;
</p>
<p>0
</p>
<p>0
1=2
</p>
<p>2
4
</p>
<p>3
5
</p>
<p>(b) Since the sum of the translation vectors of 63 and c is zero, the symbol � represents an m
</p>
<p>plane; the point-group symbol is 6mm and the space-group symbol is P63cm.
</p>
<p>(c) The matrix for the m plane in this space group is given by (remember to multiply the matrices
</p>
<p>and add the translation vectors)
</p>
<p>694 Tutorial Solutions</p>
<p/>
</div>
<div class="page"><p/>
<p>0 1 0
</p>
<p>�1 0 0
</p>
<p>0 0 1
</p>
<p>2
664
</p>
<p>3
775&thorn;
</p>
<p>0
</p>
<p>0
</p>
<p>1=2
</p>
<p>2
664
</p>
<p>3
775
</p>
<p>1 1 0
</p>
<p>1 0 0
</p>
<p>0 0 1
</p>
<p>2
664
</p>
<p>3
775&thorn;
</p>
<p>0
</p>
<p>0
</p>
<p>1=2
</p>
<p>2
664
</p>
<p>3
775
</p>
<p>63 c
</p>
<p>&frac14;
</p>
<p>1 0 0
</p>
<p>1 1 0
</p>
<p>0 0 1
</p>
<p>2
664
</p>
<p>3
775&thorn;
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>2
664
</p>
<p>3
775
</p>
<p>m
</p>
<p>(d) Refer to Fig. S2.7; not all symmetry symbols are entirely standard (red = c glide; m =
</p>
<p>mirror plane). The general equivalent positions are:
</p>
<p>12 d 1 x; y; z; x� y; x; 1=2 &thorn; z; y; x� y; z; x; y; 1=2 &thorn; z; y� x; x; z; y; y� x; 1=2 &thorn; z;
y; x; 1=2 &thorn; z; x; y� x; z; y� x; y; 1=2 &thorn; z; y; x; z; x; x� y; 1=2 &thorn; z; x� y; y; z:
</p>
<p>There are three sets of special equivalent positions:
</p>
<p>6 c m x; 0; z; x; x; 1=2 &thorn; z; 0; x; z; x; 0; 1=2 &thorn; z; x; x; z; 0; x; 1=2 &thorn; z
4 b 3 1=3; 2=3; z; 2=3; 1=3; z; 1=3; 2=3; 1=2 &thorn; z; 2=3; 1=3; 1=2 &thorn; z
2 a 3m 0; 0; z; 0; 0; 1=2 &thorn; z
</p>
<p>Wyckoff site Limiting conditions
</p>
<p>d hkil none
</p>
<p>hh2hl none
</p>
<p>hh0l none
</p>
<p>c as above
</p>
<p>b as above + hkil: l = 2n
</p>
<p>a as for site b
</p>
<p>[Courtesy Professor Steven Dutch, University of Wisconsin-Green Bay]
</p>
<p>Fig. S2.7
</p>
<p>Tutorial Solutions 695</p>
<p/>
</div>
<div class="page"><p/>
<p>2.19. From Chap. 2, Fig. 2.11, it follows that
</p>
<p>aR &frac14; 2aH=3&thorn; bH=3&thorn; cH=3
bR &frac14; �aH=3&thorn; bH=3&thorn; cH=3
cR &frac14; �aH=3� 2bH=3&thorn; cH=3
</p>
<p>Following Sect. 2.2.3, we have aR � aR &frac14; &eth;2aH=3&thorn; bH=3&thorn; cH=3&THORN;� &eth;2aH=3&thorn; bH=3&thorn; cH=3&THORN; &frac14;
</p>
<p>3a2=9&thorn; c2=9 &frac14; 12 Å2, so that aR &frac14; 3.464 Å. Similarly, cos aR &frac14; &eth;2aH=3&thorn; bH=3&thorn; cH=3&THORN;�
</p>
<p>&eth;�aH=3&thorn; bH=3&thorn; cH=3&THORN;=a2R, so that aR = 51.32�. (Remember that a = b = c and a = b = g in
a rhombohedral unit cell.)
</p>
<p>2.20. The transformation matrix S for Rhex ! Robv is given, from the solution to Problem 2.19, by
</p>
<p>S &frac14;
2=3 1=3 1=3
</p>
<p>�1=3 1=3 1=3
�1=3 �2=3 1=3
</p>
<p>2
64
</p>
<p>3
75
</p>
<p>and its inverse is
</p>
<p>S�1 &frac14;
1 �1 0
</p>
<p>0 1 �1
</p>
<p>1 1 1
</p>
<p>2
4
</p>
<p>3
5
</p>
<p>so that the transpose becomes
</p>
<p>&eth;S�1&THORN;T &frac14;
1 0 1
�1 1 1
</p>
<p>0 �1 1
</p>
<p>2
4
</p>
<p>3
5
</p>
<p>Hence (13*4)hex is transformed to &eth;32�1&THORN;obv, and &frac12;1�2*3�hex to [405]obv.
</p>
<p>Fig. S2.8
</p>
<p>696 Tutorial Solutions</p>
<p/>
</div>
<div class="page"><p/>
<p>2.21. Figure S2.8 illustrates the reflection of x, y, z across the plane (qqz), where OC = OW = q, so
</p>
<p>that OCW = 45� Q is the point q � x, q � y, z, and the remainder of the diagram is self-
explanatory.
</p>
<p>As an alternative procedure, we know that in point group 4mm, 4my = mdiag. Hence, x, y is
transformed to �y, �x by the operation mdiag. If we now move the origin to the point �q, �q, it
follows that �y, �x then becomes q � y, q � x.
</p>
<p>2.22.
</p>
<p>Diffraction symbol Point group
</p>
<p>2 m 2/m
</p>
<p>12/m1 P &middot; &middot; &middot; P2 Pm P2/m
</p>
<p>12/m1 P &middot; c &middot; Pc P2/c
</p>
<p>12/m1 P &middot; 21 &middot; P21 P21/m
</p>
<p>12/m1 P &middot; 21/c &middot; P21/c
</p>
<p>12/m1 C &middot; &middot; &middot; C2 Cm C2/m
</p>
<p>12/m1 C &middot; c &middot; Cc C2/c
</p>
<p>2.23. (a) From the matrix
</p>
<p>1=2 0 0
0 1 0
</p>
<p>0 0 1
</p>
<p>2
4
</p>
<p>3
5; (210) becomes (110) and may be confirmed by drawing
</p>
<p>to scale.
</p>
<p>(b) From the matrix
</p>
<p>1 0 0
</p>
<p>0 1=2 0
0 0 1
</p>
<p>2
4
</p>
<p>3
5; (210) becomes (410), after clearing the fraction.
</p>
<p>By drawing to scale, we see that the original (210) plane is now the second plane from the
</p>
<p>origin in the (410) family of planes; d(410)new = d(210)old/2 under the given transforma-
</p>
<p>tion. In each case, the Miller index corresponding to the unit cell halving is also halved.
</p>
<p>2.24. In Cmm2, the polar (two-fold) axis is normal to the centered plane, but parallel to it in Amm2.
</p>
<p>Cmmm and Ammm are equivalent by interchange of axes, so that they are not two distinct
</p>
<p>arrangements of points.
</p>
<p>2.25. (a) a0 = 4.850, b0 = 6.150, c0 = 7.963 Å
(b) (12,12,7)
</p>
<p>The following matrix may be helpful
</p>
<p>&eth;M�1&THORN;T &frac14;
1=2 1=4 1=2
1=2 0 1
0 1 2
</p>
<p>2
4
</p>
<p>3
5
</p>
<p>(c) &frac12;338�
(d) 0.09486, 0.008930, 0.3120 Å-1
</p>
<p>(e) x0 = �0.2192, y0 = 0.6745, z0 = �0.5645
</p>
<p>Solutions 3
</p>
<p>3.1. dl/Å = 0.0243 (1 � cos45) = 0.00712. Energy/J &frac14; hc/(1 + 0.00712) &frac14; 1.97� 10�15.
3.2. Set an origin at the center of a line joining the two scattering centers; then the coordinates are
</p>
<p>	l. The amplitude of the separated points Al &frac14; 2 cos&eth;2p� l� 2l
�1 sin y� cos y&THORN; &frac14;
</p>
<p>Tutorial Solutions 697</p>
<p/>
</div>
<div class="page"><p/>
<p>2 cos&eth;2p sin 2y&THORN;, the angle between r and S being y. For the two centers at one point (r = 0) the
amplitude Ap = 2. Hence:
</p>
<p>2y Ratio Al/Ap Intensity
</p>
<p>0 1 1
</p>
<p>30, 150 �1 1
60, 120 0.666 0.444
</p>
<p>90 1 1
</p>
<p>180 1 1
</p>
<p>3.3. We proved in the text that f1s &frac14; c41=&eth;c41 &thorn; p2S2&THORN;
2
; where c1=(4�0.3)/0.529 = 6.994 Å�1 and
</p>
<p>S = 2(sin y)/l. For the 2s contribution, the integral
&ETH;1
0
</p>
<p>x3 exp&eth;�ax&THORN; sin bx dx evaluates to
4(a3b � ab3)/(a2 + b2)4, so that f2s, becomes &frac12;2pc52=&eth;96pS&THORN;�� 3�16pSc2&frac12;c22 � &eth;2pS&THORN;
</p>
<p>2�=&frac12;c2 &thorn;
4p2S2�4 &frac14; c62&frac12;c22 � 4p2S2&THORN;=&eth;c22 &thorn; 4p2S2&THORN;
</p>
<p>4; where c2 = (4 � 2.05)/0.529 = 3.685 Å�1. Hence:
</p>
<p>Scattering formula Exponential formula
</p>
<p>sin y=lZ f1s f2s (2f1s &thorn; 2f2s) f
0.0 1.000 1.000 4.000 4.002
</p>
<p>0.2 0.938 0.116 2.108 2.060
</p>
<p>0.5 0.692 �0.0082 1.368 1.360
</p>
<p>3.4. Photon energy = hv= hc/l = hc/(hc/eV) = 1.6021� 10�19� 30000 = 4.806� 10�15 J.
</p>
<p>3.5. Mr(C6H6) = 78.11. Mr(C)/Mr(C6H6) = 0.154; Mr(H)/Mr(C6H6) = 0.0129. Hence, m = 1124
</p>
<p>[0.154 � 0.46 � 6) + (0.0129 � 0.04 � 6)] = 481.2 m�1, so that the transmittance (I/I0) is
exp(�481.2 � 1 � 10�3) = 0.618, or 61.8 %.
</p>
<p>3.6. It is necessary to note carefully the changes in sign of both A(hkl) and B(hkl). Thus, the
</p>
<p>following diagram is helpful, together with the changes in sign of the argument of the
</p>
<p>trigonometric functions. For example, if both A and B change sign, f is not unaltered by
</p>
<p>canceling the signs, but becomes p + f
</p>
<p>P21: Use (3.80)&ndash;(3.83) for k even and k odd
</p>
<p>k &frac14; 2n : f&eth;hkl&THORN; &frac14; �f&eth;�h �k �l&THORN; &frac14; �f&eth;h �k l&THORN; &frac14; f&eth;�h k �l&THORN; 6&frac14; f&eth;�hkl&THORN;
f&eth;�hkl&THORN; &frac14; �f&eth;h �k �l&THORN; &frac14; f&eth;hk�l&THORN; &frac14; �f&eth;�h �k l&THORN;
</p>
<p>k &frac14; 2n&thorn; 1 : f&eth;hkl&THORN; &frac14; �f&eth;�h �k �l&THORN; &frac14; p� f&eth;h�kl&THORN; &frac14; p&thorn; f&eth;�h k �l&THORN; 6&frac14; f&eth;�h k l&THORN;
f&eth;�hkl&THORN; &frac14; �f&eth;h �k �l&THORN; &frac14; p&thorn; f&eth;hk �l&THORN; &frac14; p� f&eth;�h �k l&THORN;
</p>
<p>698 Tutorial Solutions</p>
<p/>
</div>
<div class="page"><p/>
<p>Pma2: Use (3.94) and (3.95) for h even and odd
</p>
<p>h even : f&eth;hkl&THORN; &frac14; �f&eth;�h �k �l&THORN; &frac14; f&eth;�h k l&THORN; &frac14; f&eth;h �k l&THORN; &frac14; �f&eth;h k �l&THORN; &frac14; �f&eth;h �k �l&THORN;
&frac14; �f&eth;�h k �l&THORN; &frac14; f&eth;�h �k l&THORN;
</p>
<p>h odd : f&eth;hkl&THORN; &frac14; �f&eth;�h �k �l&THORN; &frac14; p&thorn; f&eth;�h k l&THORN; &frac14; p&thorn; f&eth;h �k l&THORN; &frac14; �f&eth;h k �l&THORN;
&frac14; p� f&eth;h �k �l&THORN; &frac14; p� f&eth;�h k �l&THORN; &frac14; f&eth;�h �k l&THORN;
</p>
<p>3.7. From the equations developed in Sects. 3.4 and 3.4.1, but taking the reciprocal space constant k as
</p>
<p>the X-ray wavelength of 1.5418 Å, we find:
</p>
<p>a* = 0.30314, b* = 0.23115, c* = 0.14096, a* = 60.182, b* = 55.878, g* = 47.591�.
V = 618.916 Å3; V* = 5.9218 � 10�3. The reciprocal unit-cell lengths are dimensionless here,
</p>
<p>and V* may be calculated as l3/V.
</p>
<p>3.8. If r1 and r2 are the distances of the two atoms from the origin, then we use r1 = x1a + y1b +
z1c and r2 = x2a + y2b + z2c. Then r1 = (r1&middot;r1)
</p>
<p>1/2 (not forgetting the cross-products), and
</p>
<p>similarly for r2. The angle y at the origin is given by cos y = r1&middot;r2/(r1 r2). Thus, the two
distances are 2.986 and 4.310 Å, and y = 45.58�.
</p>
<p>3.9. The resultant R is obtained in terms of the amplitude jRj and phase f from
Rj j&frac14; &frac12;&eth;SjA cosfj&THORN;2 &thorn; &eth;SjB sinfj&THORN;2�
</p>
<p>1=2 &frac14; &frac12;&eth;�21:763&THORN;2 &thorn;&eth;�22:070&THORN;2�1=2 &frac14; 31:00, and f =
tan�1[(�22.070)/(�21.763)] = 45.40�, but because both the numerator and denominator are
negative the phase angle lies in the third quadrant, and 180� must be added to give f = 225.40�.
</p>
<p>3.10. A-centering implies pairs of positions x, y, z and x; 1
2
&thorn; y; 1
</p>
<p>2
&thorn; z. Hence, we write
</p>
<p>F&eth;hkl&THORN; &frac14;
Xn=2
</p>
<p>j&frac14;1
fjfexp&frac12;i2p&eth;hxj &thorn; kyj &thorn; lzj&THORN;� &thorn; exp&frac12;i2p&eth;hxj &thorn; kyj &thorn; lzj &thorn; k 2&thorn; l 2== &THORN;�g
</p>
<p>The terms within the braces {} may be expressed as exp[i2p(hxj + kyj + lzj)]{1 + exp[i2p(k/
</p>
<p>2 + l/2)]} which is 2 for (k + l) even, and zero for (k + l) odd (einp = 1/0 for n even/odd).
</p>
<p>Hence, the limiting condition is hkl: k + l = 2n.
</p>
<p>3.11. The coordinates show that the structure is centrosymmetric. Hence, F&eth;hk0&THORN; &frac14;
A&eth;hk0&THORN; &frac14; 2&frac12;gP cos 2p&eth;hxP &thorn; kyP&THORN; &thorn;gQ cos 2p&eth;hxQ &thorn; kyQ&THORN;�
</p>
<p>hk A(hk) hk A(hk) hk A(hk) hk A(hk)
</p>
<p>5 0 2(�gP + gQ) 0 5 2(gP�gQ) 5 5 2(�gP � gQ) 5 10 2(�gP + gQ)
</p>
<p>For gP = 2gQ, f (0 5) = 0, f (5 0) = f (5 5) = f (5 10) = p.
</p>
<p>3.12. F(hk0) = 4gU cos 2p[kyU + (h + k)/4] cos 2p(h + k)/4 which, because (h + k) is even in the
</p>
<p>data, reduces to F(hk0) = 4gU cos 2p kyU.
</p>
<p>hk0 jF(hk0)y=0.10j jF(hk0)y=0.15j
020 86.5 86.5
</p>
<p>110 258.9 188.1
</p>
<p>Hence, 0.10 is the better value for yU in terms of the two reflections given.
</p>
<p>3.13. The shortest U&ndash;U distance dU&ndash;U is from 0; y;
1
4
to 0; �y; 3
</p>
<p>4
, so that dU&ndash;U = [(0.20b)
</p>
<p>2 + (0.5
</p>
<p>c)2]1/2 = 2.76 Å.
</p>
<p>3.14. (a) P21, P21/m; (b) Pa, P2/a; (c) Cc, C2/c; (d) P2, Pm, P2/m.
</p>
<p>Tutorial Solutions 699</p>
<p/>
</div>
<div class="page"><p/>
<p>3.15. (a) P21212; (b) Pbm2, Pbmm; (c) Ibm2, Ibmm. Note that Ibm2, for example, might have been
</p>
<p>named Icm21: normally, where more than one symmetry element lies in a given orientation, the
</p>
<p>rules of precedence in naming is m &gt; a &gt; b &gt; c &gt; n &gt; d and 2 &gt; 21. In a few cases the rules
</p>
<p>may be ignored. For example, I4cm could be named I4bm, but with the origin on 4, the c-glides
</p>
<p>pass through the origin, and the former symbol is preferred.
</p>
<p>Writing example (c) with the redundancies indicated, we have
</p>
<p>hkl: h + k + l = 2n
</p>
<p>0kl: k = 2n, (l = 2n) or l = 2n, (k = 2n)
</p>
<p>h0l: (h + l = 2n)
</p>
<p>hk0: (h + k = 2n)
</p>
<p>h00: (h = 2n)
</p>
<p>0k0: (k = 2n)
</p>
<p>00l: (l = 2n)
</p>
<p>3.16. (a)
</p>
<p>(i) h0l: h = 2n; 0k0: k = 2n.
</p>
<p>(ii) h0l: l = 2n
</p>
<p>(iii) hkl: h + k = 2n
</p>
<p>(iv) h00: h = 2n
</p>
<p>(v) 0kl: l = 2n; h0l : l = 2n
</p>
<p>(vi) hkl : h + k + l = 2n; h0l: h = 2n
</p>
<p>Other space groups with the same conditions: (i) None; (ii) P2/c; (iii) C2, C2/m; (iv) None;
</p>
<p>(v) Pccm; (vi) Ima2 (I2am)
</p>
<p>(b) hkl: None
</p>
<p>h0l: h + l = 2n
</p>
<p>0k0: k = 2n
</p>
<p>(c) C2/c; C222
</p>
<p>3.17. (a) In the given setting x0 and a are normal to a c-glide, y0 and �c are normal to an a-glide, and z0
</p>
<p>and b are normal to a b-glide. In the standard setting, x is along x0 and the plane normal to
has its glide in the new y direction, so that it is a b-glide; y is along z0 and the plane normal
to it is a glide now in the direction of z, a c glide; z is along �y0 and the plane normal to it is
now an a-glide. Thus, the symbol in the standard setting is Pbca.
</p>
<p>(b) In Pmna the symmetry leads to translations of (c + a)/2 and a/2, overall c/2, and in Pnma
</p>
<p>the translations arising are a/2, b/2, and c/2. Hence, the full symbol for Pmna is P
2
</p>
<p>m
</p>
<p>2
</p>
<p>n
</p>
<p>21
</p>
<p>a
,
</p>
<p>whereas that for Pnma is P
21
</p>
<p>n
</p>
<p>21
</p>
<p>m
</p>
<p>21
</p>
<p>a
.
</p>
<p>3.18. mR = 2.00, so that A = 10.0. Hence, jF(hkl)j2 = I � Lp�1 � A = 56.3 � 0.625 � 1.1547
� 10.0 = 406.3.
</p>
<p>3.19. &eth;a&THORN; C6h &eth;b&THORN;
6
</p>
<p>m
11; P
</p>
<p>63
</p>
<p>m
11 (c) Hexagonal/Trigonal (d) Hexagonal (e) Hexagonal (f) P.
</p>
<p>3.20. In this example, we need the A and B terms of the geometrical structure factor. From the
</p>
<p>coordinates of the general equivalent position, we have
</p>
<p>A &frac14; cos 2p&eth;hx&thorn; ky&thorn; lz&THORN; &thorn; cos 2p&eth;�hx� ky&thorn; lz&THORN; &thorn; cos 2p&eth;�hy&thorn; kx&thorn; lz&THORN; &thorn; cos 2p&eth;hy� kx&thorn; lz&THORN;
</p>
<p>&thorn; cos 2p hx� ky&thorn; lz&thorn; h&thorn; k &thorn; l
2
</p>
<p>� �
</p>
<p>&thorn; cos 2p �hx&thorn; ky&thorn; lz&thorn; h&thorn; k &thorn; l
2
</p>
<p>� �
</p>
<p>&thorn; cos 2p hy&thorn; kx&thorn; lz&thorn; h&thorn; k &thorn; l
2
</p>
<p>� �
</p>
<p>&thorn; cos 2p �hy� kx&thorn; lz&thorn; h&thorn; k &thorn; l
2
</p>
<p>� �
</p>
<p>700 Tutorial Solutions</p>
<p/>
</div>
<div class="page"><p/>
<p>Combining the terms appropriately:
</p>
<p>A=2 &frac14; cos 2plzfcos 2p&eth;hx&thorn; ky&THORN; &thorn; cos 2p&eth;�hy&thorn; kx&THORN;g &thorn; cos 2p lz&thorn; h&thorn; k &thorn; l
2
</p>
<p>� �
</p>
<p>cos 2p&eth;hx� ky&THORN;
</p>
<p>&thorn; cos 2p lz&thorn; h&thorn; k &thorn; l
2
</p>
<p>� �
</p>
<p>cos 2p&eth;hy&thorn; kx&THORN;
</p>
<p>The expansion of cos 2p lz&thorn; h&thorn; k &thorn; l
2
</p>
<p>� �
</p>
<p>shows that we need to consider the cases of
</p>
<p>h + k + l even and odd, and recalling that cosP	 cosQ &frac14; cosP cosQ� sinP sinQ, we find
</p>
<p>the following:
</p>
<p>h +k + l = 2n
</p>
<p>A &frac14; 4 cos 2plz&eth;cos 2phx cos 2pky� sin 2phy sin 2pkx&THORN;
</p>
<p>Similarly
</p>
<p>B &frac14; 4 sin 2plz&eth;cos 2phx cos 2pky� sin 2phy sin 2pkx&THORN;
</p>
<p>From the equations for jF(hkl)j and f(hkl), we find h + k + l = 2n + 1.
Proceeding in a similar manner, we now find
</p>
<p>A &frac14; 4 cos 2plz&eth;� sin 2phx sin 2pky&thorn; sin 2phy sin 2pkx&THORN;
</p>
<p>and
</p>
<p>B &frac14; � 4 sin 2plz&eth;� sin 2phx sin 2pky&thorn; sin 2phy sin 2pkx&THORN;
</p>
<p>It is clear now that for h + k + l odd, A = B = 0 if h = 0, or k = 0, or h = 	k. Hence, the limiting
</p>
<p>conditions: 0kl: k + l = 2n12 (h0l: h + l = 2n), and hhl: l = 2n. The first of these conditions
</p>
<p>corresponds to an n-glide &perp;a, (b) while the second indicates a c-glide &perp;&lt;110&gt;, consistent with
</p>
<p>space group P4nc.
</p>
<p>Solutions 4
</p>
<p>4.1. For thegiven reflection, (sin y)/l = 0.30, forwhich fC = 2.494.Hence, exp[�B(sin2y)/l2] = 0.5423,
so that fC,27.55� = 1.352, which is 54.2 % of what its value would be at rest. The root mean square
</p>
<p>displacement is [6.8/(8p2)]1/2 = 0.29 Å. Since vibrational energy is proportional to kT, where k is the
</p>
<p>Boltzmann constant, a reduced temperature factor with concomitant enhanced scattering would be
</p>
<p>achieved by conducting the experiment at a low temperature.
</p>
<p>4.2. For NaCl, d111 &frac14; a=
p
3 &frac14; 2:2487 Å, so that (sin y111)/l = 0.1539 Å�1 and (sin y222)/
</p>
<p>l = 0.3078 Å�1. Similarly, for KCl, (sin y111)/l = 0.1379 Å
�1 and (sin y222)/l = 0.1379 Å
</p>
<p>�1.
</p>
<p>12 [h + k + l &frac14; 2n + 1 &rarr; k + l &frac14; 2n + 1 for h &frac14; 0].
</p>
<p>Tutorial Solutions 701</p>
<p/>
</div>
<div class="page"><p/>
<p>Using the structure factor equation for the NaCl structure type, we have
</p>
<p>F&eth;111&THORN; &frac14; 4&frac12;fNa&thorn;=K&thorn; &thorn; fCl� cos&eth;3p&THORN;� &frac14; 4&frac12;fNa&thorn;=K&thorn;� fCl� �, whereas F(222) = 4[fNa+/K+ + fCl�].
Thus, we obtain the following results:
</p>
<p>111 222
</p>
<p>NaCl KCl NaCl KCl
</p>
<p>(sin y)/l 0.1539 0.1379 0.3078 0.2759
</p>
<p>f+ 8.979 15.652 6.777 11.576
</p>
<p>f� 13.593 14.207 9.387 9.997
</p>
<p>F �18.46 1.445 64.66 86.29
</p>
<p>Remembering that wemeasure jFj2, it is clear that jF(111)j for KCl is relatively vanishingly small.
4.3. F &frac14; &eth;2=pS&THORN;1=2
</p>
<p>&ETH;1
0
</p>
<p>F exp&eth;�F2=2S&THORN;dF. Let F2=2S &frac14; t, so that dF &frac14; &eth;S=2t&THORN;1=2dt.
Then, F &frac14; &eth;2S=p&THORN;1=2
</p>
<p>&ETH;1
0
</p>
<p>t0 exp&eth;�t&THORN;dt. Since t0 = t(1�1), the integral (see Web Appendix WA7)
is G(1) = 1, Hence, F &frac14; &eth;2S=p&THORN;1=2.
</p>
<p>Making the above substitution again, we have F2 &frac14; &eth;2S=p&THORN;1=2
&ETH;1
0
</p>
<p>t1=2 exp&eth;�t&THORN;dt &frac14; &eth;2S=p1=2&THORN;
1=2 G&eth;1=2&THORN; &frac14; S.
Thus, Mc &frac14; &eth;2S=p&THORN;=S &frac14; 2=p &frac14; 0:637.
</p>
<p>4.4. E3 &frac14; &eth;2=p&THORN;1=2
&ETH;1
0
</p>
<p>E3 exp&eth;�E2=2&THORN;dE. Let E2=2 &frac14; t, so that dE = (2t)�1/2dt. Then,
E3 &frac14; &eth;8=p&THORN;1=2
</p>
<p>&ETH;1
0
</p>
<p>t exp&eth;�t&THORN; dt &frac14; &eth;8=p&THORN; 1
2
G&eth;2&THORN; &frac14; 1:596.
</p>
<p>4.5.
</p>
<p>jE2 � 1j &frac14; 2
&eth;1
</p>
<p>0
</p>
<p>jE2 � 1jE exp&eth;�E2&THORN; dE
</p>
<p>By making the substitution E2 = t, we have
</p>
<p>jE2 � 1j &frac14;
&eth;1
</p>
<p>0
</p>
<p>&eth;1� t&THORN; exp&eth;�t&THORN; dt&thorn;
&eth;1
</p>
<p>1
</p>
<p>&eth;t� 1&THORN; exp&eth;�t&THORN;dt
</p>
<p>&frac14; &eth;�e�tj10 &thorn; &eth;te�tj
1
0 &thorn; &eth;e�tj
</p>
<p>1
0 � &eth;te�tj
</p>
<p>1
1 � &eth;e�tj
</p>
<p>1
1 &thorn; &eth;e�tj
</p>
<p>1
1
</p>
<p>&frac14; 2=e &frac14; 0:736
</p>
<p>4.6. The statistically distinguishable features of classes 2, m and 2/m are summarized as follows:
</p>
<p>P2 Pm P2/m
</p>
<p>hkl 1A 1A 1C
</p>
<p>h0l 1C 2A 2C
</p>
<p>0k0 2A 1C 2C
</p>
<p>When finding the average intensities, do not mix the h0l and 0k0 reflections either with
</p>
<p>themselves or with the hkl reflections until the space-group ambiguity has been resolved. Instead
</p>
<p>get them from some other zone, excluding any terms it contains that lie in [h0l] or [0k0] zones,
</p>
<p>and check the distribution of this chosen zone. If it is centric, the space group is P2/m. To
</p>
<p>distinguish between the other two space group, examine the distribution in the [h0l] zone.
</p>
<p>Generally there will be insufficient 0k0 reflections alone to give reliable results.
</p>
<p>4.7. (a) mmmPc - - leaves the following space groups unresolved:
</p>
<p>Pcm21 2/2; 2/2; 4(1)
</p>
<p>Pc2m 2/2; 4/(1); 2/2
</p>
<p>Pcmm (4/2; 4/2; 4/2)
</p>
<p>702 Tutorial Solutions</p>
<p/>
</div>
<div class="page"><p/>
<p>The numbers are the multiples for the principal rows and zones (see Table 4.2). Parentheses
</p>
<p>indicate centric zones or the complete weighted reciprocal lattice. One would examine the
</p>
<p>distribution in both the [h0l] and [0k0] zones. Alternatively, an examination of the [0k0]
</p>
<p>zone, excluding the h0l and 0k0 data, could be considered. A centric distribution would
</p>
<p>identify Pccm. The other two could be separated by reference to zones, but distinction may
</p>
<p>be difficult at this stage.
</p>
<p>(b) mmmC - - - leaves the following space groups unresolved:
</p>
<p>Cmm2 2/2; 2/2; 4(1)
</p>
<p>Cm2m 2/2; 4/(1); 2/2
</p>
<p>C222 2/(1); 2/(1) 2/(1)
</p>
<p>Cmmm (4/2; 4/2; 4/2)
</p>
<p>Again, parentheses indicate centric zones or the complete weighted reciprocal lattice. All
</p>
<p>principal zones must be examined in order to resolve the ambiguities here.
</p>
<p>Solutions 5
</p>
<p>5.1. (a) The crystal system is tetragonal, and the Laue group is
4
</p>
<p>m
mm; the optic axis lies along the
</p>
<p>needle axis (z) of the crystal.
</p>
<p>(b) The section is in extinction for any rotation in the x,y plane, normal to the needle axis; the
</p>
<p>section is optically isotropic.
</p>
<p>(c) For a general oscillation photograph with the X-ray beam normal to z, the symmetry is m.
</p>
<p>For a symmetrical oscillation photograph with the beam along a, b or any direction in the
</p>
<p>form h110i at the mid-point of the oscillation, the symmetry is 2mm.
5.2. (a) The crystal system is orthorhombic.
</p>
<p>(b) Suitable axes may be taken parallel to three non-coplanar edges of the brick.
</p>
<p>(c) Symmetry m.
</p>
<p>(d) Symmetry 2mm, with the m lines horizontal and vertical.
</p>
<p>5.3. (a) Monoclinic, or possibly orthorhombic.
</p>
<p>(b) If monoclinic, p is parallel to the y axis. If orthorhombic, p is parallel to one of x, y, or z.
</p>
<p>(c) (i) Mount the crystal perpendicular to p, about either q or r, and take a Laue photograph with
</p>
<p>the X-ray beam parallel to p. If the crystal is monoclinic, symmetry 2 would be observed. If
</p>
<p>orthorhombic, the symmetry would be 2mm, with the m lines in positions on the film that
</p>
<p>define the directions of the crystallographic axes normal to p. If the crystal is rotated such
</p>
<p>that the X-rays travel through the crystal perpendicular to p, a vertical m line would appear
</p>
<p>on the Laue photograph of either a monoclinic or an orthorhombic crystal. (ii) Use the same
</p>
<p>crystal mounting as in (i), but take a symmetrical oscillation photograph with the X-ray
</p>
<p>beam parallel or perpendicular to p at the mid-point of the oscillation. The rest of the answer
</p>
<p>is as in (i).
</p>
<p>5.4. Refer to Fig. S5.1. Let hmax represent the maximum value sought. Since we are concerned with
</p>
<p>a large d* value, we take l 
 0.2 Å, the minimum value in the white radiation. Now
</p>
<p>d* = ha* = (2/l) sin y and since, from the diagram, y is the angle subtended at the circumfer-
</p>
<p>ence by d*, y = 20�, so that hmax is the integral part of 2=&eth;0:2l&THORN; sin 20, which is 17.
The X-coordinate on the film is 60 tan 40 = 50.35 mm. The half-width of the film is
</p>
<p>62.5 mm, so the 17,00 reflection will be recorded on the film.
</p>
<p>Tutorial Solutions 703</p>
<p/>
</div>
<div class="page"><p/>
<p>5.5. For the first film, we can write I&eth;hkl&THORN; &thorn; I&eth;2h; 2k; 2l&THORN; &frac14; 300, and for the second film,
after absorption, we have 0:35I&eth;hkl&THORN; &thorn; 0:65I&eth;2h; 2k; 2l&THORN; &frac14; 130. Solving these equations gives
I(hkl) = 216.7 and I(2h, 2k, 2l) &frac14; 83.3.
</p>
<p>5.6. (a) For symmetry 2mm in Laue group m�3m, the X-ray beam must be traveling along a &lt;110&gt;
</p>
<p>direction (Table 1.6); we will choose [110], so that a and b lie in the horizontal plane; c is then
</p>
<p>the vertical direction.
</p>
<p>(b) We can use Fig. 5.17, changing the sign of�a*, and with f = 45� because XO is [110] for the
present problem. For an inner spot, it follows readily that 2y = tan�1(43.5/60.0), so that 2y =
</p>
<p>35.94�, and e = 27.03� (Chap. 5, Fig. 5.17).
(c) Now, tan 27:03 &frac14; 0:5102 &frac14; h=k, since a = b. In the given orientation, the reflections on the
</p>
<p>horizontal line are hk0 and, since the unit cell is F, h and k must be both even, with k = 2h,
</p>
<p>from above. Possible reflections are, therefore, 240, 480, 612,0, . . . It is straightforward to
</p>
<p>show that l &frac14; 2a sin y=
ffiffiffiffi
N
</p>
<p>p
, where N = h2 + k2.
</p>
<p>For 240, l = 0.746 Å, for 480, l = 0.373 Å, which is unreasonably small in crystallographic
</p>
<p>work.We note from the orientation of the a and b axes (a* and b*) that one of h and kmust be
</p>
<p>negative; we can choose k. For an outer spot, we find in a similar manner that tan e = 0.3418,
</p>
<p>so that k = 3h. Reasonable indices correspond to h = 2 and k = 6, again with one index
</p>
<p>negative; here, l = 0.753 Å. To summarize:
</p>
<p>The X-ray beam is along [110]. For the inner spots: y = 17.97�; 2 �4 0 and 4 �2 0;
l = 0.746 Å. For the outer spots: y = 26.13�; 2�60 and 6�20; l = 0.753 Å.
</p>
<p>5.7. Since the crystal is uniaxial, it must be hexagonal, tetragonal, or trigonal. The Laue symmetry
</p>
<p>along axis 1 indicates that the crystal is trigonal, referred to hexagonal axes, and that axis 1 is
</p>
<p>therefore c. Following Chap. 5, Sect. 5.4.3, we find for the repeat distances along the three axes:
</p>
<p>Fig. S5.1
</p>
<p>704 Tutorial Solutions</p>
<p/>
</div>
<div class="page"><p/>
<p>Axis 1 2 3
</p>
<p>Repeat=Å 15:65 8:264 4:772
</p>
<p>The smallest repeat distance corresponds to the unit-cell dimension a, direction &frac12;10�10�, Laue
symmetry 2 (Chap. 1, Fig. 1.36 and Table 1.6). Axis 2 must be a direction in the x, y plane, and it
</p>
<p>is straightforward to show that it is the repeat distance along &frac12;12�30�, Laue symmetrym. Thus, we
have: a = b = 4.772, c = 15.65 Å; a = b = 90�, g = 120�; the Laue group is �3m.
</p>
<p>5.8. Applying the Bragg equation, l &frac14; 2d sin y, where d = 6.696/2 Å. Thus, (a) y0002 (Cu) = 13.31�,
and (b) y0002 (Mo) = 6.093
</p>
<p>�.
5.9. (a) The data indicate a pseudo-monoclinic unit cell with g unique. Following Chap. 2, Sect. 2.5,
</p>
<p>we find a = b = 6.418, c = 3.863 Å. It would appear that the c dimension is true, and that
</p>
<p>the ab plane is centered. It is straightforward to show that a and b are the half-diagonals of a
</p>
<p>rectangle with sides a0 = a � b and b0 = a + b. Thus, the orthorhombic unit cell has the
dimensions a = 3.062, b = 12.465, and c = 3.863 Å. The transformation can be written as
</p>
<p>atrue = Madiff, where
</p>
<p>M &frac14;
1 �1 0
1 1 0
</p>
<p>0 0 1
</p>
<p>2
4
</p>
<p>3
5
</p>
<p>(b) The reciprocal cell is transformed according to a�true &frac14; &eth;M
�1&THORN;Ta�diff . The transpose of the
</p>
<p>inverse matrix is
</p>
<p>1=2 �1=2 0
1=2 1=2 0
</p>
<p>0 0 1
</p>
<p>2
64
</p>
<p>3
75
</p>
<p>Hence, a* = 0.2321, b* = 0.05702, c* = 0.1840. These values may be confirmed by dividing
</p>
<p>the &ldquo;true&rdquo; values, for the orthorhombic cell, into the wavelength.
</p>
<p>5.10. (a) Refer to Sect. 5.2.4: tan 2ymax &frac14; r=R, where r is the radius of the plate, 172.5 mm.
2dmin sin ymax &frac14; 1:05= &eth;2� 1:0&THORN; &frac14; 0:525; and ymax &frac14; 63:336�. Hence, R &frac14; 172:5=
tan 63:336; or 86:6mm.
</p>
<p>(b) The whole image would shrink but would still contain the same amount of data, and the
</p>
<p>spots would become closer together.
</p>
<p>(c) Some of the pattern would be lost because the angle subtended at the edge of the plate would
</p>
<p>become less: the spots would then be further apart.
</p>
<p>5.11. (a) A 5, B 1, C 0.1 mm
</p>
<p>(b) A 450, B 250, C 80 mm
</p>
<p>(c) A 12, B 60, C 300 s
</p>
<p>Solutions 6
</p>
<p>6.1.
&ETH; c=2
�c=2 sin&eth;2pmx=c&THORN; cos&eth;2p n x=c&THORN;dx &frac14;
</p>
<p>&ETH; c=2
�c=2 f
</p>
<p>1
</p>
<p>2
sin&frac12;2p&eth;m&thorn; n&THORN;x=c� &thorn; 1
</p>
<p>2
sin&frac12;2p&eth;m� n&THORN;x=c�gdx
</p>
<p>using identities from Web Appendix WA5. Integration leads to � &frac12;c=2p&eth;m&thorn; n&THORN;� cos&frac12;2p
&eth;m&thorn; n&THORN;x=c� jc=2�c=2 � &frac12;c=2p&eth;m� n&THORN;� cos&frac12;2p&eth;m� n&THORN;x=c�j
</p>
<p>c=2
�c=2. Since m and n are integers the
</p>
<p>integral is zero for m 6&frac14; n. For m = n, the original integral becomes
&ETH; c=2
�c=2
</p>
<p>1
2
sin&eth;4pmx=c&THORN; dx,
</p>
<p>which is also zero.
</p>
<p>Tutorial Solutions 705</p>
<p/>
</div>
<div class="page"><p/>
<p>6.2. A plot of r(x) as a function of x (in 40ths) shows peaks at 0, 20, and 40 for Mg (as expected),
</p>
<p>and at ca 8.25, 11.75, 28.25, and 31.75, for the 4F atoms per repeat unit a; thus, xF is 	(0.206;
0.706). Only the function to a/4 need be calculated, since there is m symmetry across the points
1
4
&eth;10=40&THORN;; 1
</p>
<p>2
&eth;20=40&THORN; and 3
</p>
<p>4
&eth;30=40&THORN;.
</p>
<p>(a) The first three terms alone are insufficient to resolve clearly the pairs of fluorine peaks that
</p>
<p>are closest in projection.
</p>
<p>(b) Changing the sign of the 600 reflection results in single peaks for fluorine at 10/40 and
</p>
<p>30/40. The error in sign (phase) is clearly the more serious fault.
</p>
<p>6.3. G&eth;S&THORN; &frac14;
&ETH; p
�p a exp &eth;i2pSx&THORN; dx &frac14; a
</p>
<p>&ETH; p
�p cos &eth;2pSx&THORN; &thorn; ia
</p>
<p>&ETH; p
�p sin&eth;2pSx&THORN; dx. The second integral is
</p>
<p>zero, because the integrand is an odd function. Hence,
</p>
<p>G&eth;S&THORN; &frac14; a &eth;2p S&THORN; sin&eth;2pSx&THORN;jp�p &frac14; 2ap sin&eth;2pSp&THORN;=2pSp&THORN;
</p>
<p>and we retain the parameters which would obviously cancel, so as to preserve the characteristic
</p>
<p>sin (ax)/(ax) form. To obtain the original function, we evaluate
</p>
<p>f &eth;x&THORN; &frac14; &eth;a=p&THORN;
&eth;1
</p>
<p>�1
&eth;1=S&THORN; sin&eth;2pSp&THORN; exp&eth;�i2pxS&THORN; dS &frac14; &eth;a=p&THORN;
</p>
<p>&eth;1
</p>
<p>�1
&eth;1=S&THORN; sin&eth;2pSp&THORN; cos&eth;2pxS&THORN; dS
</p>
<p>where the sine term from the expanded integrand is zero as before. Using results from Web
</p>
<p>Appendix WA5, the integral becomes
</p>
<p>&eth;a=2p&THORN;
&eth;1
</p>
<p>�1
&eth;1=S&THORN; sin&eth;2pS&eth;p&thorn; x&THORN;� dS&thorn;
</p>
<p>&eth;1
</p>
<p>�1
&eth;1=S&THORN; sin&eth;2pS&eth;p� x&THORN;�
</p>
<p>� �
</p>
<p>dS
</p>
<p>a&eth;p� x&THORN;
&eth;1
</p>
<p>�1
sin&frac12;2pS&eth;p� x&THORN;�=&frac12;2p&eth;p� x&THORN;� dS:
</p>
<p>From Web Appendix WA9,
&ETH;1
�1 &eth;sin y=y&THORN; dy &frac14; p; hence, we derive
</p>
<p>f &eth;x&THORN; &frac14; &eth;a=2&THORN;&eth;p&thorn; x&THORN;=jp&thorn; xj &thorn; &eth;a=2&THORN;&eth;p� x&THORN;=jp� xj:
</p>
<p>It is clear from this result that f(x) = a for j x j &lt; p, f(x) = a/2 for x = 	p, and f(x) = 0 for
j x j = 0, which correspond to the starting conditions.
</p>
<p>6.4.
</p>
<p>G&eth;f &THORN; &frac14; A
</p>
<p>&eth;1
</p>
<p>�1
cos&eth;2pf0t&THORN; exp&eth;�i2pft&THORN; dt
</p>
<p>&frac14; &eth;A=2&THORN;
&eth;1
</p>
<p>�1
f&frac12;exp&eth;i2pf0t&THORN; &thorn; exp&eth;�i2pft&THORN;� exp&eth;�i2pf t&THORN;g dt
</p>
<p>&frac14; &eth;A=2&THORN;
&eth;1
</p>
<p>�1
fexp&frac12;�i2p&eth;f � f0&THORN;t� &thorn; exp&frac12;�i2p&eth;f &thorn; f0&THORN;t�g dt
</p>
<p>&frac14; &eth;A=2&THORN;d&eth;f &thorn; f0&THORN; &thorn; &eth;A=2&THORN;d&eth;f � f0&THORN;:
</p>
<p>In the inversion, the d-function repeats the function at f = f0. Thus,
</p>
<p>f &eth;t&THORN; &frac14; &eth;A=2&THORN;
&eth;1
</p>
<p>�1
&frac12;d&eth;f &thorn; f0&THORN; &thorn; d&eth;f � f0&THORN;� exp&eth;i2pft&THORN; df&frac14; &eth;A=2&THORN;&frac12;exp&eth;i2pf0t&THORN; &thorn; exp&eth;�i2pf0t&THORN;�
</p>
<p>&frac14; A cos&eth;2pf0t&THORN;:
</p>
<p>706 Tutorial Solutions</p>
<p/>
</div>
<div class="page"><p/>
<p>6.5. The molecules have the displacements p and �p from the origin. Hence, the total transform
GT(S) is given by
</p>
<p>GT&eth;S&THORN; &frac14; G0&eth;S&THORN; exp&eth;i2pp � S&THORN; &thorn; G�0&eth;S&THORN; exp&eth;�i2p p � S&THORN;
</p>
<p>Using results from Sect. 3.2.3, we can write G0(S) = jG0j exp(if), and G
�
0&eth;S&THORN; &frac14; jG0j exp &eth;�f&THORN;,
</p>
<p>where f is a phase angle. Hence,
</p>
<p>GT&eth;S&THORN; &frac14; jG0j exp&eth;i2pp � S&thorn; f&THORN; &thorn; exp&eth;�i2pp � S� f&THORN;f &frac14; 2jG0j cos&eth;2pp � S&thorn; f&THORN;
</p>
<p>As discussed in Sect. 6.6.3, the maximum value of the transform is 2jG0j, at those points where
</p>
<p>cos(2pp&middot;S + f) is equal to unity. In this example, however, such points do not lie in planes and,
consequently, the fringe systems are curved rather than planar.
</p>
<p>6.6. The atoms related by the screw axis would have the fractional coordinates x, y, z and
</p>
<p>�x; 1
2
&thorn; y; 1
</p>
<p>2
� z. From (6.50), we have
</p>
<p>G&eth;S&THORN; &frac14;
Xn=2
</p>
<p>j&frac14;1
fj exp&frac12;i2p&eth;hxj &thorn; kyj &thorn; lzj&THORN;� &thorn; exp&frac12;i2p&eth;�hxj &thorn; kyj � lzj &thorn; k=2&thorn; l=2&THORN;�
� 	
</p>
<p>where the summation is over n/2 atoms in the unit cell not related by the 21 symmetry. Hence,
</p>
<p>G&eth;S&THORN; &frac14;
X
</p>
<p>n=2
</p>
<p>j&frac14;1
fjfexp&frac12;i2pkyjfexp&frac12;i2p&eth;hxj &thorn; lzj&THORN;&thorn; exp&frac12;i2p&eth;�hxj � lzj &thorn; k=2&thorn; l=2&THORN;�g
</p>
<p>In a general transform, h, k, and l could take any values. However, in a crystal they are integers,
</p>
<p>but in order to obtain a special condition, we must also consider the case that h = l = 0:
</p>
<p>G&eth;S&THORN;h&frac14;l&frac14;0 &frac14; 2
X
</p>
<p>n=2
</p>
<p>j&frac14;1
fj exp&eth;i2pkyj&frac12;exp&eth;ipk&THORN;�&THORN;
</p>
<p>Then, we have G(S)h=l=0 = 0 for k = 2n + 1, that is, the 0k0 reflections are systematically
</p>
<p>absent when k is odd.
</p>
<p>6.7. Figure S6.1 indicates the nodal lines for the P&ndash;S fringe system. Since the transform is chosen to
</p>
<p>be positive at the origin, 	 regions can be allocated to the transform, as shown. Hence, the
intense reflections can be allocated signs, as follow:
</p>
<p>240� 250� 410 � 520 �
650+ 710+ 720 + 820 +
</p>
<p>�130&thorn; �140&thorn; �230&thorn; �240&thorn;
�370&thorn; �440� �470&thorn; �530�
�540� �670� �710� �760�
�910&thorn; �920&thorn; 10; 00&thorn; 10; 10&thorn;
10; 20&thorn;
</p>
<p>6.8. In Fig. S6.2, the three points are plotted in (a).A transparency ismadeof the structure in (a), inverted
</p>
<p>in the origin. The structure (a) is then drawn three times on the transparency, with each of the atoms
</p>
<p>of the inversion, in turn, over the origin of (a), and in the same orientation. The completed diagram
</p>
<p>(b) is the required convolution: the six triangles outlined in (b) all produce the same set of nine
</p>
<p>vectors (three superimposed at the origin).
</p>
<p>Tutorial Solutions 707</p>
<p/>
</div>
<div class="page"><p/>
<p>Fig. S6.1
</p>
<p>Fig. S6.2
</p>
<p>708 Tutorial Solutions</p>
<p/>
</div>
<div class="page"><p/>
<p>6.9. Figure S6.3 shows the contoured figure field of Fig. S6.2. The same triangles are revealed,
</p>
<p>giving six sets of atom coordinates, as follow:
</p>
<p>1 0.15, 0.10 �0.15,�0.10 �0.05, 0.30
2 0.05, 0.20 �0.05,�0.20 �0.20,�0.20
3 0.10,�0.10 �0.10, 0.10 �0.20,�0.30
4 0.05,�0.30 0.15, 0.10 �0.15,�0.10
5 0.25, 0.00 0.05, 0.20 �0.05,�0.20
6 0.10,�0.10 �0.10, 0.10 0.20, 0.30
</p>
<p>6.10. The transform is positive in sign at the origin. Hence, by noting the succession of contours
</p>
<p>along the 00l row, we arrive at the following result:
</p>
<p>001 002 003 004 005 006
</p>
<p>+ � + + � �
</p>
<p>6.11. The transform of f(x) is given by
</p>
<p>fT&eth;x&THORN; &frac14;
1ffiffiffiffiffiffi
2p
</p>
<p>p
&eth;1
</p>
<p>�1
&frac12;exp&eth;�x2=2&THORN; exp&eth;i2pSx&THORN;�dx &frac14; 2ffiffiffiffiffiffi
</p>
<p>2p
p
</p>
<p>&eth;1
</p>
<p>0
</p>
<p>&frac12;exp&eth;�x2=2&THORN; cos&eth;2pSx&THORN;�dx;
</p>
<p>because fT(x) is an even function. Hence, using standard tables of integrals,
</p>
<p>Fig. S6.3
</p>
<p>Tutorial Solutions 709</p>
<p/>
</div>
<div class="page"><p/>
<p>fT&eth;x&THORN; &frac14;
2ffiffiffiffiffiffi
2p
</p>
<p>p
ffiffiffi
p
</p>
<p>p
</p>
<p>2
ffiffiffiffiffiffiffiffi
1=2
</p>
<p>p exp&eth;�4p2S2=2&THORN; &frac14; exp&eth;�2p2S2&THORN;:
</p>
<p>The transform of g(x), gT(x), is a d-function with the origin at the point x = 2, so that gT(x) =
</p>
<p>exp(i4pS), from Sect. 6.6.8. Hence, c(x) = fT(x)*gT(x) = exp(i4pS � 2p2S2).
6.12. (a) As h is increased, the form of f(x) approaches a square more and more closely.
</p>
<p>(b) m-lines occur at 	&frac14;, that is, at 15/60 and 45/60 in x
</p>
<p>(c) At x = 0 and 2p the sine term in (6.15) is zero, so that f(x) = p/2. At x = p, the sine term is
</p>
<p>sin(2ph)30/60, or sin(ph). Since h is an integer, then again f(x) = p /2.
</p>
<p>Solutions 7
</p>
<p>7.1. In P21/c, the general positions are 	 &eth;x; y; z; x;
1
2
� y; 1
</p>
<p>2
&thorn; z&THORN;, so that A&eth;hkl&THORN; &frac14;
</p>
<p>2fcos 2p&eth;hx&thorn; ky&thorn; lz&THORN; &thorn; cos 2p&eth;hx� ky&thorn; lz&thorn; k=2&thorn; l=2&THORN;g&frac14; 4 cos 2p&eth;hx&thorn; lz &thorn; k=4&thorn;l=4&THORN;
cos 2p&eth;ky� k=4� l=4&THORN;. Introducing the y-coordinate of &frac14;, A&eth;hkl&THORN; &frac14; 4 cos 2p&eth;hx&thorn;
lz&thorn; k=4&thorn; l=4&THORN; cos 2p&eth;l=4&THORN;, so that the hkl reflections will be systematically absent for
l = 2n + 1. The indication is that the c spacing should be halved, so that the true unit cell
</p>
<p>contains two species in space group P21 (see Fig. S7.1). This problem illustrates the conse-
</p>
<p>quences of sitting an atom on a glide plane: although we have considered here a hypothetical
</p>
<p>structure containing one atom in the asymmetric unit, in a multi-atom structure, an atom may,
</p>
<p>by chance, be situated on a translational symmetry element.
</p>
<p>7.2. Refer to Chap. 2, Fig. 2.37, and Figs. S7.2 and S7.3.13 There are eight rhodium atoms in the unit
</p>
<p>cell. If the atoms are in general positions, the minimum separation of atoms across any m plane
</p>
<p>is 1=2 � 2y. For any value of y, the distance would be too small to accommodate two rhodium
atoms. Hence, they must occupy two sets of special positions. Positions on centers of symmetry
</p>
<p>may be excluded on the same grounds as above. Thus, the atoms are located on two sets of m
</p>
<p>planes as follow:
</p>
<p>4 Rh	 &eth;x1; 1=4; z1; 1=2 � x1; 3=4; 1=2 &thorn; z1&THORN;
4Rh	 &eth;x2; 1=4; z2; 1=2 � x2; 3=4; 1=2 &thorn; z2&THORN;
</p>
<p>Fig. S7.1
</p>
<p>13Mooney R, Welch AJE (1954) Acta Crystallogr 7:49.
</p>
<p>710 Tutorial Solutions</p>
<p/>
</div>
<div class="page"><p/>
<p>7.3. The space group is P21/m. The molecular symmetry cannot be �1, but it can be m.
</p>
<p>Hence, we can make the following assignments:
</p>
<p>(a) Cl on m; (b) N on m; (c) two C on m, with four other C probably in general positions; (d)
</p>
<p>sixteen H in four sets of general positions, two H (in N&ndash;H groups) on m, and two H from CH3
groups onm&mdash;those that have their C atoms onm. This arrangement is shown in Fig. S7.4a. The
</p>
<p>species CH3, H1 and H2 lie above and below the m plane. The alternative space group P21 was
</p>
<p>considered, but the full structure analysis14 confirmed P21/m. Figure S7.4b illustrates P21/m,
</p>
<p>and is reproduced from the International Tables for X-ray Crystallography, Vol. I, by kind
</p>
<p>permission of the International Union of Crystallography.
</p>
<p>7.4. A&eth;hhh&THORN; &frac14; 4fgPt &thorn; gK&frac12;cos 2p&eth;3h=4&THORN; &thorn; cos 2p&eth;9h=4&THORN;� &thorn; 6gCl&frac12;3 cos 2p&eth;hx&THORN;�g, where the factor
4 relates to an F unit cell (see Sect. 3.7.1). B(hhh) = 0, so that F(hhh) = A(hhh), and A(hhh)
</p>
<p>Fig. S7.2
</p>
<p>Fig. S7.3
</p>
<p>14Lindgren J, Olovsson I (1968) Acta Crystallogr B24:554.
</p>
<p>Tutorial Solutions 711</p>
<p/>
</div>
<div class="page"><p/>
<p>simplifies to 4fgPt&thorn; 2gK cos&eth;3ph=2&THORN; &thorn; 6gCl cos&eth;2phx&THORN;g. We can now calculate F(hhh) for the
two values of x given:
</p>
<p>x = 0.23 x = 0.24
</p>
<p>hhh Fo jFcj K1Fo jFcj K2Fo
111 491 340.6 314.7 317.4 329.5
</p>
<p>222 223 152.2 142.9 159.5 149.6
</p>
<p>333 281 145.2 180.1 190.8 188.6
</p>
<p>K1 = 0.641 R1 = 0.11 K2 = 0.671 R2 = 0.036
</p>
<p>Clearly x = 0.24 is the preferred value. Pt&ndash;Cl = 2.34 Å. For a sketch and the point group, see
</p>
<p>Problem 1.11(c) and its solution.
</p>
<p>7.5. AU&eth;hkl&THORN; &frac14; 2 cos 2p&eth;hx&thorn; ky&thorn; l=4&THORN;&thorn;cos 2p&eth;�hx&thorn; ky&thorn; l=4&thorn; h=2&thorn; k=2&THORN;f g &frac14; 4fcos 2p&frac12;ky&thorn;
&eth;h&thorn; k &thorn; l&THORN;=4� cos 2p&frac12;hx� &eth;h&thorn; k&THORN; =4�g. For (200), AU / j cos 2p &eth;2x� 12&THORN;j and, for this
reflection to have zero intensity, 2p&eth;2x� 1
</p>
<p>2
&THORN; 
 &eth;2n&thorn; 1&THORN;p 2= . For n = 1, x 
 �1/8 (by sym-
</p>
<p>metry, the values 1/8, 5/8, and 7/8 are included). Conveniently, we choose the smallest of the
</p>
<p>Fig. S7.4
</p>
<p>712 Tutorial Solutions</p>
<p/>
</div>
<div class="page"><p/>
<p>symmetry-related values, that is, 1/8. For (111), and using this value for
</p>
<p>x; AU / cos 2p&eth;y&thorn; 3=4&THORN; cos 2p&eth;1=8 � 1=2&THORN;. For high intensity, j cos 2p&eth;y&thorn; 3=4&THORN;j 
 0; np.
For n = 0, y = &frac34; (and &frac14; by symmetry). For n = 1, y is again 1
</p>
<p>4
and &frac34;. Proceeding in this
</p>
<p>manner with (231) leads to y = 1/6 (by symmetry, the values 1/3, 2/3, and 5/6 are included),
</p>
<p>and with (040) we find y = 3/16 (by symmetry, 5/16, 11/16, and 13/16 are included). The
</p>
<p>mean for the three value of y is (1/4 + 1/6 + 3/16)/3, or approximately 0.20.
</p>
<p>7.6. Since there are two molecules per unit cell in P21/m in this structure, and the molecules cannot
</p>
<p>have �1 symmetry, the special positions sets 	 &eth;x; 1
4
; z&THORN; are selected. TheB, C, andN atoms lie onm.
</p>
<p>Since the shortest distance between m planes is 3.64 Å, the F1, B, N, C, and H1 atoms must lie on
</p>
<p>one and the samem plane (see Fig. S7.5a). Hence, the remaining two F and four H atoms must be
</p>
<p>placed symmetrically across the samem plane. These conclusions were borne out by the structure
</p>
<p>analysis.15 Figure S7.5b is a stereoview of the packing diagram for CH3NH2BF3, showing the H1,
</p>
<p>C, N, B and three F atoms. The m plane is normal to the vertical direction in the diagram and the
</p>
<p>remaining two pairs of H atoms are disposed across the m plane as described above.
</p>
<p>7.7. (a) (i) jF&eth;hkl&THORN;j &frac14; jF&eth;�h �k �l&THORN;j; (ii) jF&eth;0kl&THORN;j &frac14; jF&eth;0 �k �l&THORN;j; (iii) jF&eth;h0l&THORN;j &frac14; jF&eth;�h 0 �l&THORN;j
(b) (i) jF&eth;hkl&THORN;j &frac14; jF&eth;�h �k �l&THORN;j &frac14; jF&eth;h �k l&THORN;;
</p>
<p>(ii) jF&eth;0kl&THORN;j &frac14; F&eth;0 �k �l&THORN;j &frac14; jF&eth;0 �k l&THORN;
</p>
<p>(iii) jF&eth;h0l&THORN;j &frac14; jF&eth;�h 0 �l&THORN;j
</p>
<p>Fig. S7.5
</p>
<p>15Geller S, Hoard JL (1950) Acta Crystallogr 3:121.
</p>
<p>Tutorial Solutions 713</p>
<p/>
</div>
<div class="page"><p/>
<p>(c) (i) jF&eth;h k l&THORN;j &frac14; jF&eth;�h �k �l&THORN;j &frac14; jF&eth;�h k l&THORN;j &frac14; jF&eth;h �k l&THORN;j &frac14; jF&eth;h k �l&THORN;j
(ii) jF&eth;0 k l&THORN;j &frac14; jF&eth;0 �k �l&THORN;j &frac14; jF&eth;0 �k l&THORN;j;
(iii) jF&eth;h 0 l&THORN;j &frac14; jF&eth;�h 0 �l&THORN;j &frac14; jF&eth;�h 0 l&THORN;j
</p>
<p>Any combination of 	hkl not listed follows the pattern of (a) (i). In (b), for example,
</p>
<p>jF&eth;�h 0 l&THORN;j &frac14; jF&eth;h 0 �l&THORN;j
</p>
<p>7.8. (a) In Pa, the symmetry element relates the sites x, y, z and 1
2
&thorn; x; y; z, so that the Harker line is
</p>
<p>&frac12;1=2 v 0�. In P2/a, the Harker section is (u 0 w) and the line &frac12;12 v 0�.
In P2221, there are three Harker sections, (0 v w), (u 0 w), and &eth;u v 12&THORN;.
</p>
<p>(b) The Harker section (u 0 w) must arise through the symmetry-related sites x, y, z and �x; y; �z,
</p>
<p>which correspond to a two-fold axis along y. Similarly, the line [0v0] arises from a mirror
</p>
<p>plane in the Patterson normal to y. Since the crystal is non-centrosymmetric, the space
</p>
<p>group must be P2 or Pm. If it is P2, there must be, by chance, closely similar y coordinates
</p>
<p>for many of the atoms in the structure. If it is Pm, chance coincidences occur between the x
</p>
<p>and z coordinates. [These conditions are somewhat unlikely, especially when many atoms
</p>
<p>are present, so that Harker sections and lines can sometimes be used to distinguish between
</p>
<p>space groups that are not determined by diffraction symmetry alone.]
</p>
<p>7.9. (a) P21/n, a non-standard setting of P21/c (see also Chap. 2, Problem 2.12).
</p>
<p>(b) The S&ndash;S vectors have the following Patterson coordinates:
</p>
<p>(1) 	 &eth;1
2
; 1
2
&thorn; 2y; 1
</p>
<p>2
&THORN; Double weight
</p>
<p>(2) 	 &eth;1
2
&thorn; 2x; 1
</p>
<p>2
; 1
2
&thorn; 2z&THORN; Double weight
</p>
<p>(3) 	(2x, 2y, 2z) Single weight
</p>
<p>(4) 	 &eth;2x; 2�y; 2z&THORN; Single weight
</p>
<p>Section v &frac14; 1
2
</p>
<p>Type 2 vector x = 0.182, z = 0.235
</p>
<p>Section v = 0.092 Type 1 vector y = 0.204
</p>
<p>Section v = 0.408 Type 3 or 4 vector x = 0.183, y = 0.204, z = 0.234
</p>
<p>Thus we have four S&ndash;S vectors at: 	(0.183, 0.204, 0.235; 0.683, 0.296, 0.735). Any one of the
</p>
<p>other seven centers of symmetry, unique to the unit cell, may be chosen as the origin,
</p>
<p>whereupon the coordinates would be transformed accordingly. The sulfur atom positions are
</p>
<p>plotted in Fig. S7.6 [Small differences in the third decimal places of the coordinates determined
</p>
<p>from the maps in Problems 7.9 and 7.10 are not significant.]
</p>
<p>Fig. S7.6
</p>
<p>714 Tutorial Solutions</p>
<p/>
</div>
<div class="page"><p/>
<p>7.10. (a) By direct measurement, the sulfur atom coordinates are S (0.266, 0.141) and S0 (�0.266,
�0.141)
</p>
<p>(b) Draw an outline of the unit cell on tracing paper, and plot the position of &ndash;S on it. Place the
</p>
<p>tracing over the idealized Patterson map (Fig. P7.2), in the same orientation, with
</p>
<p>the position of &ndash;S over the origin of the Patterson map, and copy the Patterson map on
</p>
<p>to the tracing (Fig. S7.7a). On another tracing, carry out the same procedure with respect to
</p>
<p>the position of �S0 (Fig. S7.7b). Superimpose the two tracings (Fig. S7.7c). Atomic
positions correspond to positive regions of the two superimposed maps.
</p>
<p>Fig. S7.7
</p>
<p>Tutorial Solutions 715</p>
<p/>
</div>
<div class="page"><p/>
<p>7.11. (a) The summation to form P(v) can be carried out with program FOUR1D. In using the program,
</p>
<p>each data line should contain k,Fo(0k0)
2, and 0.0, the zero datum representing theB coefficient
</p>
<p>of the Fourier series. P(v) shows three non-origin peaks. If the highest of them is assumed to
</p>
<p>arise from theHf&ndash;Hf vector, then yHf = 0.105; the smaller peaks are Hf&ndash;Si vectors, fromwhich
</p>
<p>we could obtain approximate y parameters for the silicon atoms. Their difference in height
</p>
<p>arisesmainly from the fact that one of them is, in projection, close to the origin peak. However,
</p>
<p>the simplified structure factor equation for Fo(0k0), based on the hafnium atoms alone, is
</p>
<p>Fo&eth;0k0&THORN; / cos&eth;2pkyHf&THORN;
</p>
<p>so that the signs of the reflections, ignoring the weak reflections 012,0 and 016,0, are, in order,
</p>
<p>+ � � + + �. (b) We can now calculate r(y) with these signs attached to the Fo(0k0) values.
From the result, we obtain yHf = 0.107, ySi1 = 0.033, and ySi2 = 0.25. These values for ySi lead
</p>
<p>to vectors which appear on P(v). We conclude that the small peak on r(y) at y = 0.17 is
</p>
<p>spurious, arising most probably from both the small number of data and experimental errors
</p>
<p>in them.
</p>
<p>7.12. Since the sites of the replaceable atoms are the same in each derivative, and the space group is
</p>
<p>centrosymmetric, we can write F&eth;M1&THORN; &frac14; F&eth;M2&THORN; &thorn; 4&eth;fM1 � fM2&THORN;, where f may be approximated
by the corresponding atomic number, Z. Hence, we can draw up the following table:
</p>
<p>(a)
</p>
<p>M
</p>
<p>h NH4 K Rb T1
</p>
<p>1 � � + +
2 a + + +
</p>
<p>3 + + + +
</p>
<p>4 � a + +
5 + + + +
</p>
<p>6 � � a +
7 a + + +
</p>
<p>8 a + + +
</p>
<p>a = Indeterminate, because F is small or zero.
</p>
<p>(b) The peak at 0 represents K and Al, superimposed in projection. The peak at 0.35 would then
</p>
<p>be presumed to be due to the S atom.
</p>
<p>(c) The effect of the isomorphous replacement of S by Se can be seen at once in the increases in
</p>
<p>Fo(555) and Fo(666) and decrease in Fo(333). These changes are not in accord with the
</p>
<p>findings in (b). Comparison of the two electron density plots shows that dS/Se must be
</p>
<p>0.19 (the x coordinate is d=
ffiffiffi
3
</p>
<p>p
). The peak at 0.35 arises from a superposition of oxygen
</p>
<p>atoms in projection, and is not appreciably altered by the isomorphous replacement.
</p>
<p>7.13. A &frac14; 100 cos 60&thorn; &eth;fo &thorn; Df 0&THORN; cos 36&thorn; 8 cos126 &frac14; 50&thorn; 40:046� 4:702 &frac14; 85:344: B &frac14; 100
sin 60&thorn;&eth;fo &thorn; Df 0&THORN; sin 36&thorn; 8 sin 126 &frac14; 86:603&thorn; 29:095&thorn; 6:472 &frac14; 122:17. Hence, jF(010)j =
149.0, and f (010) = 55.06�. For the 0�10 reflection, we have A &frac14; 100 cos 60&thorn;
&eth; fo &thorn; Df 0&THORN; cos 36&thorn; 8 cos 54&frac14; 50&thorn;40:046&thorn; 4:702 &frac14; 94:748: B &frac14; 100 sin&eth;�60&THORN; &thorn; &eth;fo &thorn; Df 0&THORN;
sin&eth;�36&THORN; &thorn; 8 sin 54 &frac14; �86:603� 29:095&thorn; 6:472 &frac14; �109:226. Hence, jF&eth;0�10&THORN;j &frac14; 144:6,
and f&eth;0�10&THORN; &frac14; � 49.06�.
</p>
<p>7.14. Draw a circle, at a suitable scale, to represent an amplitude jFPj of 858. From the center of this
circle, set up a &ldquo;vector&rdquo; to represent jFH1jexp(if1), where jFH1j = 141 and f1 = (78 + 180) deg.
</p>
<p>716 Tutorial Solutions</p>
<p/>
</div>
<div class="page"><p/>
<p>At the termination of this vector, draw a circle of radius 756 to represent jFPH1j. Repeat this
procedure for the other two derivatives (Fig. S7.8). The six intersections 1&ndash;10, 2&ndash;20, and 3&ndash;30 are
strongest in the region indicated by &bull;- - -&bull;. The required phase angle fM , calculated from (7.50),
</p>
<p>lies in this region. The centroid phase angle fB is biased slightly towards point 1.
</p>
<p>7.15. Cos(hx � f) expands to cos hx cosf&thorn; sin hx sinf which, for f = p/2, reduces to sinhx.
Hence, c&eth;x&THORN; &frac14; p=2&thorn; 2
</p>
<p>P1
h&frac14;1 &eth;1=h&THORN; sin hx &frac14; p=2&thorn; 2
</p>
<p>P1
h&frac14;1 &eth;1=h&THORN; cos&eth;hx� f&THORN;: This equation
</p>
<p>resembles closely a Fourier series (see Sect. 6.2).
</p>
<p>7.16. (a) The total mass of protein per unit cell is 18000Z � 1.6605 � 10�24 g, where Z is the number
</p>
<p>of protein molecules per unit cell. Since there is an equal mass of solvent water in the unit cell,
</p>
<p>D/2Z = (18000 � 1.6605 � 10�24)/(40 � 50 � 60 � 10�24 sin 100�) = 0.2529 g cm�3, so
thatD = 0.5058Z g cm-3. Sensible values for Z in C2 are 4 and 2. The former leads to a density
</p>
<p>that is much too large for a protein; Z = 2 gives D = 1.012 g cm&ndash;3, which is an acceptable
</p>
<p>answer.
</p>
<p>(b) In space group C2 there are four general equivalent positions (see Sect. 2.7.3). Since Z = 2,
</p>
<p>the protein molecule must occupy special positions on twofold axes, so that the molecule
</p>
<p>has symmetry 2.
</p>
<p>7.17. In the notation of the text, we have for F(hkl)
</p>
<p>FH&eth;&thorn;&THORN; &frac14; F0H&eth;&thorn;&THORN; &thorn; iF00H&eth;&thorn;&THORN;
and for jF&eth;�h �k �l&THORN;j
</p>
<p>FH&eth;�&THORN; &frac14; F0H&eth;�&THORN; &thorn; iF00H&eth;�&THORN;
</p>
<p>where F0H&eth;&thorn;&THORN; and F0H&eth;�&THORN; are the structure factor components derived from the real part of (7.64)
and (7.66), and F00H&eth;&thorn;&THORN; and F00H&eth;�&THORN; are its anomalous components. It is clear from Fig. S7.9 that
</p>
<p>Fig. S7.8
</p>
<p>Tutorial Solutions 717</p>
<p/>
</div>
<div class="page"><p/>
<p>the moduli jFH(+)j and jFH(�)j are equal, but that fH(+) 6&frac14; fH(�). In terms of the structure
factor equations, we can write a single atom vector for h and �h
</p>
<p>F&eth;h&THORN; &frac14; &eth;f 0 &thorn; iDf 00&THORN; exp&frac12;i&eth;2ph � r&thorn; p=2&THORN;�
F&eth;�h&THORN; &frac14; &eth;f &thorn; iDf 00&THORN; exp&frac12;�i&eth;2ph � r&thorn; p=2&THORN;�
</p>
<p>from which we have jF&eth;h&THORN;j &frac14; F&eth;�h&THORN;j, but f&eth;h&THORN; 6&frac14; f&eth;�h&THORN;; p/2 acts in the same sense (positive) in
each case.
</p>
<p>7.18. In the notation of the text, and for a centrosymmetric structure, we have FPH&eth;&thorn;&THORN; &frac14;
AP&eth;&thorn;&THORN; &thorn; A0H&eth;&thorn;&THORN; &thorn; iA00H&eth;&thorn;&THORN; where
</p>
<p>AP&eth;&thorn;&THORN; &frac14;
XNP
</p>
<p>j&frac14;1
fj cos 2p&eth;hxj &thorn; kyj &thorn; lzj&THORN;
</p>
<p>A0H&eth;&thorn;&THORN; &frac14;
XNH
</p>
<p>j&frac14;1
f 0j cos 2p&eth;hxj &thorn; kyj &thorn; lzj&THORN;
</p>
<p>A00H&eth;&thorn;&THORN; &frac14;
XNH
</p>
<p>j&frac14;1
Df 00j cos 2p&eth;hxj &thorn; kyj &thorn; lzj&THORN;
</p>
<p>Clearly, jF&eth;h k l&THORN;j &frac14; jF&eth;�h �k �l&THORN;j &frac14; &eth;A2&thorn; B2&THORN;1=2, where A &frac14; AP&eth;&thorn;&THORN; &thorn; A0H&eth;&thorn;&THORN; and B &frac14; A00H
&eth;&thorn;&THORN;; f&eth;hkl&THORN; &frac14; f&eth;�h �k �l&THORN; &frac14; tan�1 &eth;B=A&THORN;, and cannot equal 0 or p because of the finite value of
A00H&eth;&thorn;&THORN;.
</p>
<p>7.19. If, for a crystal of a given space group, Friedel&rsquo;s Law breaks down, then the diffraction
</p>
<p>symmetry reverts to that of the corresponding point group. Thus, we have
</p>
<p>jF(hkl)j equivalents Bijvoet pairs
(a) C2 (2) hkl=�h k �l hkl=�h k �lwith h �k l=�h �k �l
</p>
<p>(continued)
</p>
<p>Fig. S7.9
</p>
<p>718 Tutorial Solutions</p>
<p/>
</div>
<div class="page"><p/>
<p>(b) Pm (m) hkl=h �k l hkl=h �k lwith �h k �l=�h �k �l
</p>
<p>(c) P212121 (222) hkl=h �k �l=�h k �l=�h �k l hkl=h �k �l=�h k �l=�h �k l with h k l=h �k l=h k �l=�h �k �l
</p>
<p>(d) P4(4) hkl=�k h l=�h �k l=k �h l hkl=�k h l=�h �k l=k �h l with k �h �l=h k �l=�k h �l=�h �k �l
</p>
<p>Strictly, pairs related as hkl and �h �k �l should be discounted, as they are, of course, Friedel pairs.
</p>
<p>7.20. The number N of symmetry-independent reciprocal lattice points with a range 0 &lt; y &lt; ymax is
</p>
<p>33.510 Vc sin
3 y/(l3Gm), from Chap. 7. The volume Vc of the unit cell is 6 � 10
</p>
<p>4 Å3,G = 1 for
</p>
<p>a P unit cell, and m, the number of symmetry-equivalent general reflections, is 8 for the Laue
</p>
<p>group mmm. Hence, N = 74466.7 sin3 ymax.
</p>
<p>(a) 0 &lt; y &lt; 10� : sin3 ymax = 5.236 � 10
�3, so that N = 389 (779 if we consider the hkl and
</p>
<p>�h �k l reflections).
</p>
<p>(b) 10 &lt; y &lt; 20� : sin3 ymax = 4.001 � 10
�2, so that N = 2979 � 389, or 2590.
</p>
<p>(c) 20 &lt; y &lt; 25�: sin3 ymax = 7.548 � 10
�3, so that N = 5620 � 2979, or 2641.
</p>
<p>The resolution, defined in terms of dmin, is dmin = l/(2 sin ymax)
</p>
<p>(a) For ymax = 10
�: dmin = 4.32 Å
</p>
<p>(b) For ymax = 20
�: dmin = 2.19 Å
</p>
<p>(c) For ymax = 25
�: dmin = 1.77 Å
</p>
<p>Solutions 8
</p>
<p>8.1. A possible set, with the larger jEj values, is 705, 6 1 7, and 8�1 4. Reflection 4 2 �6 is a structure
seminvariant, and 203 is linearly related to the pair 8�1�4 and 6�1�7. Reflection 43 2 has a low jEj
value, so that triple relationships involving it would not have a high probability. Alternative sets
</p>
<p>are 705, 203, 8�1�4 and 705, 203, 6 �1 �7. A vector triplet exists between 81 4, 42�6, and 4�32.
</p>
<p>8.2. The equations for A and B lead to the following relationships:
</p>
<p>jF&eth;hkl&THORN;j &frac14; jF&eth;�h�k �l &THORN;j &frac14; jF&eth;h�kl&THORN;j &frac14; jF&eth;�hk�l&THORN;j 6&frac14; jF&eth;�hkl&THORN;j; jF&eth;�hkl&THORN;j &frac14; jF&eth;hk�l&THORN;j
</p>
<p>Because of the existence of the k/4 term, the phase relationships depend on the parity of k:
</p>
<p>k &frac14; 2n : f&eth;hkl&THORN; &frac14; �f&eth;h k l&THORN; &frac14; �f&eth;h�kl&THORN; &frac14; f&eth;�hk�l 6&frac14; f&eth;�hkl&THORN;;
f&eth;�hkl&THORN; &frac14; f&eth;hk�l&THORN;
</p>
<p>k &frac14; 2n&thorn; 1 : f&eth;hkl&THORN; &frac14; �f&eth;�h �k �l&THORN; &frac14; p� f&eth;h�kl&THORN; &frac14; p&thorn; f&eth;�hk�l&THORN; 6&frac14; f&eth;�hkl;
f&eth;�hkl&THORN; &frac14; p&thorn; f&eth;hk�l&THORN;
</p>
<p>8.3. Set (b) would be chosen: there is a redundancy in set (a) among 041, �162, and �123, because
</p>
<p>jF&eth;041&THORN;j&frac14; jF&eth;0 �4 1&THORN;j in this space group. In space group C2/c, h + k is even for reflections to
occur, so that reflections 012, �123, 162, and �162 would not occur. The origin could be fixed by
</p>
<p>223 and 13�7: there are only four parity groups for a C-centered unit cell.
</p>
<p>8.4. Following the procedure given in Chap. 4, Sect. 4.2, it will be found that K = 4.0 	 0.4,
and B = 6.6 	 0.3 Å2. Since B &frac14; 8p2U2, the root mean square atomic displacement is
</p>
<p>[6.6/(8p2)]1/2, or 0.29 Å. (You were not expected to derive the standard errors in K and
</p>
<p>B; they are quoted in order to give an idea of the precision obtainable from a Wilson plot.)
</p>
<p>8.5. A plot of the atomic positions in the unit cell and its environs shows that the shortest Cl. . .Cl
</p>
<p>contact distance is between atoms at &frac14;, y, z, and 3=4; �y; z. Hence, d
2(Cl. . .Cl) = a2/4 + 4y2b2,
</p>
<p>Tutorial Solutions 719</p>
<p/>
</div>
<div class="page"><p/>
<p>so that d(Cl. . .Cl) = 4.639 Å. The superposition of errors (see Sect. 8.6) shows that the variance
</p>
<p>of d(Cl. . .Cl) is obtained from
</p>
<p>&frac12;2ds&eth;d&THORN;�2 &frac14; 2as&eth;a&THORN;=4&frac12; �2 &thorn; &frac12;8y2bs&eth;b&THORN;�2 &thorn; &frac12;8b2ys&eth;y&THORN;�2
</p>
<p>so that s(d) = 0.026 Å. (It may be noted that this answer calculates as 0.02637 Å to four
</p>
<p>significant figures. Note. If we use only the third term, that in s(y), then the result is 0.02626 Å.
</p>
<p>Thus, the error in a distance between atoms arises mostly from the errors in the corresponding
</p>
<p>atomic coordinates.)
</p>
<p>8.6. In the first instance we average the sum of fk and fh&ndash;k, namely, (�37 � 3 � 54 + 38 + 13)/6,
or �7.17�. Applying the tangent expression leads to the better value of �11.32�.
</p>
<p>8.7. Vectors of the type labeled P1- - - -P2 will not occur in the search Patterson as they involve
</p>
<p>atoms, in the region of P1, within the additional loop of the target molecule that are absent in
</p>
<p>the search molecule. Only the search molecule will be positioned by rotation and translation,
</p>
<p>and the missing parts of the structure, particularly in the loop, need to be located initially using
</p>
<p>Fourier and possibly least-squares methods, as in small-molecule analysis.
</p>
<p>8.8. (a) It is not clear how the side chain comprising atoms 8&ndash;13 is oriented with respect to the rest of
</p>
<p>the molecule, which is predominantly flat. The facility in the PATSEE program for varying the
</p>
<p>linkage torsion angle could be used but was not necessary in practice because a sufficiently large
</p>
<p>independent search fragment was available.
</p>
<p>(b) By chance the molecular graphics program oriented the search model, which is perfectly
</p>
<p>flat, to be in the XY plane. Hence all Z coordinates are zero in this plane.
</p>
<p>(c) The CHEM-X (or ChemSketch) program allows a chemical model of the molecule to be
</p>
<p>constructed and provides coordinates for the atoms. These coordinates are given not as
</p>
<p>fractional coordinates but as Å values with respect to the internal orthogonal axis system of
</p>
<p>the program. To convert to fractional coordinates for the purpose of this problem, the X, Y, and
</p>
<p>Z values were each divided by 100 for all atoms. This set then belongs to an artificial unit cell
</p>
<p>with dimensions given in the question.
</p>
<p>8.9. In Fig. S8.1, OP = 1.400 Å, OQ = 1.400 sin 60, and Q1 = 1.400 cos 60. Thus:
</p>
<p>Coordinates in the unit cell are:
</p>
<p>Atom 1: X = 0.700 Y = 1.212 Z = 0.000
</p>
<p>Atom 2: X = 1.400 Y = 0.000 Z = 0.000
</p>
<p>Atom 3: X = 0.700 Y = �1.212 Z = 0.000
Atom 4: X = �0.700 Y = �1.212 Z = 0.000
Atom 5: X = �1.400 Y = 0.000 Z = 0.000
Atom 6: X = �0.700 Y = 1.212 Z = 0.000
</p>
<p>Fractional coordinates in the given unit cell:
</p>
<p>Atom 1: X = 0.237 Y = 0.211 Z = 0.000
</p>
<p>Atom 2: X = 0.473 Y = 0.000 Z = 0.000
</p>
<p>Atom 3: X = 0.237 Y = �0.211 Z = 0.000
Atom 4: X = �0.237 Y = �0.211 Z = 0.000
Atom 5: X = �0.473 Y = 0.000 Z = 0.000
Atom 6: X = �0.237 Y = 0.211 Z = 0.000
</p>
<p>720 Tutorial Solutions</p>
<p/>
</div>
<div class="page"><p/>
<p>8.10. From Chap. 4, (4.34) and Chap. 8, (8.1), with N atoms per unit cell, assuming scaled Fo values,
</p>
<p>dividing throughout by
PN
</p>
<p>j&frac14;1 g
2
j;y (the e factor is assumed to be unity), gives
</p>
<p>Ej j2 &frac14; 1&thorn;


1
XN
</p>
<p>j&frac14;1 g
2
j;y
</p>
<p>. �X
</p>
<p>j6&frac14;k
gj;ygk;y exp i2ph � rj;k
</p>
<p>� &#13;
 !
</p>
<p>The second term on the right-hand side represents sharpened jFj2 coefficients [see also Chap. 7,
</p>
<p>(7.19)]. The term in the Patterson function that creates the origin peak,
PN
</p>
<p>j&frac14;1 g
2
j;y, is now unity,
</p>
<p>so that a Patterson function with coefficients (jEj2 � 1) produces a sharpened Patterson
function with the origin peak removed.
</p>
<p>8.11. (a) When using Molecular Replacement in macromolecular crystallography the search and
</p>
<p>target molecules should be compatible in size as well as in their three-dimensional
</p>
<p>structures. If this is not the case problems may be encountered in obtaining a dominant
</p>
<p>solution to MR. The more possible solutions which have to be inspected, using Fourier
</p>
<p>methods, the more laborious the process becomes, maybe to the point where the analysis
</p>
<p>becomes untenable.
</p>
<p>(b) For small-molecule analysis it is more usual for the search &ldquo;molecule&rdquo; to be a fairly small
</p>
<p>fragment of the target molecule. In this case the search molecule must be as accurate as
</p>
<p>possible in bond lengths and angles because the data are at atomic resolution and the
</p>
<p>Patterson peaks similarly resolved. Programs such as PATSEE allow for more complex
</p>
<p>search molecules to be used which have one degree of torsional freedom, thus increasing the
</p>
<p>size of the whole search fragment.
</p>
<p>8.12. The required determinant is
</p>
<p>E&eth;0&THORN; E&eth;h&THORN; E&eth;2h&THORN;
E&eth;�h&THORN; E&eth;0&THORN; E&eth;h&THORN;
E&eth;�2h&THORN; E&eth;�h&THORN; E&eth;0&THORN;
</p>
<p>�
</p>
<p>�
</p>
<p>�
</p>
<p>�
</p>
<p>�
</p>
<p>�
</p>
<p>�
</p>
<p>�
</p>
<p>�
</p>
<p>�
</p>
<p>�
</p>
<p>�
</p>
<p>� 0, which evaluates as
</p>
<p>E&eth;0&THORN;3&thorn;E&eth;h&THORN;2E&eth;�2h&THORN;&thorn;E&eth;2h&THORN;&frac12;E&eth;�h&THORN;�2� E&eth;0&THORN;jE&eth;2h&THORN;j2�E&eth;0&THORN;E&eth;h&THORN;j2�E&eth;0&THORN;jE&eth;h&THORN;j2� 0 or E&eth;0&THORN;
fE&eth;0&THORN;2 � jE&eth;2h&THORN;j2� 2jE&eth;h&THORN;j2g &thorn; 2 jE&eth;h&THORN;j2 E&eth;2h&THORN;� 0. Inserting the given values for E(0),
jE(h)j, and jE(2h)j, we obtain 3f9� 4� 8g&thorn; 8&eth;	2&THORN;� 0, from which it is clear E(2h) is
positive in sign in order to satisfy the determinant expression.
</p>
<p>8.13. For the first triplet, h + k + l = 0 0 0 modulo (2 2 2), where h = h1 + h2 + h3, and similarly for
k and l. Hence, the triplet is a structure seminvariant. The second triplet is also a structure
seminvariant, for the same reason. In the third triplet, h + k + l = 0 0 0 and is a structure
invariant. None of these triplets is suitable for specifying an origin because their determinants
</p>
<p>are either zero or zero modulo 2 (see Appendix E).
</p>
<p>8.14. Apply the structure factor (3.63). (a) In space group P21, the [010] zone is centric, plane group
</p>
<p>p2, so that B0 &frac14; 0 and A0 &frac14; 2 cos 2p&frac12;&eth;1� 0:3&THORN; &thorn; &eth;3� 0:1&THORN;�, so that jFj = 1.62 and f = 180�.
</p>
<p>Fig. S8.1
</p>
<p>Tutorial Solutions 721</p>
<p/>
</div>
<div class="page"><p/>
<p>(b) (i) Space group P212121 in projection on to (h0l) becomes plane group p2gg, with
</p>
<p>coordinates 	(x, z; &frac12; � x, &frac12; + z). Proceeding as in (a), jFj = 0.38 and f = 180�. (ii) In the
standard orientation, the coordinates are (see Fig. 2.35): x, z; &frac12; � x, &frac12; + z; &frac12; + x, �z; �x,
&frac12; � z. Proceeding as before, A0 = 0 and B0 = �1.18, so that jFj = 1.18 and f = &ndash;90�. Alter-
natively, we recall from Appendix E that the phase change for an origin shift r is &ndash;2ph.r, which
is &ndash;2p(&frac34;), so that f = 180 &ndash; 270 = �90�.
</p>
<p>Solutions 9
</p>
<p>9.1. (a) In space groupP21, symmetry-related vectors have the coordinates 	 &eth;2x; 1=2; 2z&THORN;; the I&ndash;I vector
</p>
<p>in the half unit cell is easily discerned. By measurement on the map, x1 = 0.422 and zI = 0.144,
</p>
<p>with respect to the origin O.
</p>
<p>(b) The contribution of the iodine atoms, FI, to the structure factors is given by 2fI cos 2p
</p>
<p>(0.422xI + 0.144zI). Hence, the following table:
</p>
<p>hkl (sin y)/l 2fI fI Fo
</p>
<p>001 0.026 105 65 40
</p>
<p>0014 0.364 67 67 37
</p>
<p>106 0.175 88 �20 33
300 0.207 84 �8 35
</p>
<p>The signs of 001, 0014, and 106 are probably +, +, and �, respectively. The magnitude
jFI (300)j is a small fraction of Fo, and could easily be outweighed by the contribution from
the rest of the structure. Thus, its sign remains uncertain from the data given. Small
</p>
<p>variations in the values determined for fI are acceptable; they derive, most probably, from
</p>
<p>small differences in the graphical interpolation of the fI values.
</p>
<p>(c) The shortest I&ndash;I vector is that between the positions listed above. Hence, dI&ndash;I = {[2 � 0.422
� 7.26]2 + [0.5 � 11.55]2 + [2 � 0.144 � 19.22]2 + [2 � 0.422 � 0.144 � 7.26 � 19.22
</p>
<p>cos (94.07)]}1/2 = 10.05 Å.
</p>
<p>9.2. A S2 listing is prepared as follows:
</p>
<p>h k h � k jE(h)j jE(k)j jE(h &ndash; k)j
0018 081 0817 9.5
</p>
<p>011 024 035 5.0
</p>
<p>026 035 0.5
</p>
<p>021 038 059 0.4
</p>
<p>0310 059 0.4
</p>
<p>024 035 059 9.6
</p>
<p>038 059 0817 7.2
</p>
<p>081 011,7 6.0
</p>
<p>081 011,9 10.2
</p>
<p>0310 059 081 7.9
</p>
<p>081 011,9 9.2
</p>
<p>(Note the convention, that a two-figure Miller index takes a comma after it unless it is the
</p>
<p>third index.)
</p>
<p>In space group P21/a, s&eth;hkl&THORN; &frac14; s&eth;�h �k �l&THORN; &frac14; &eth;�1&THORN;h&thorn;ks&eth;h �k l&THORN;, and s&eth;h k �l&THORN; &frac14; &eth;�1&THORN;h&thorn;k s&eth;�h k l&THORN;.
In using only two-dimensional reflections from the data set, we need just two reflections to
</p>
<p>722 Tutorial Solutions</p>
<p/>
</div>
<div class="page"><p/>
<p>specify an origin, say, 0, 0. We take s(081) = s(011,9) = + and proceed to the determination of
</p>
<p>signs, as follows:
</p>
<p>The two indications for s(021) and the single indication for s(026) will have low probabilities,
</p>
<p>because of low jEj values, and must be regarded as unreliable at this stage. Within the data set, no
conclusion can be reached about s(a); both + and� signs are equally likely. Reflection 0312 does
not interact within the data set.
</p>
<p>9.3. The space group is P21/c, from Chap. 9, Table 9.4. Thus, sjE&eth;hkl&THORN;j &frac14; sjE&eth;�h �k �l&THORN;j
&frac14; &eth;�1&THORN;k&thorn;1sjE&eth;h �k l&THORN;j; for the hk reflections, set l = 0 in these relationships. Figure S9.1 shows
the completed chart. A S2 listing follows; an N indicates that no new relationships were
</p>
<p>derivable with the reflection so marked; negative signs are represented by bars over the jEj values.
</p>
<p>S2 Listing
</p>
<p>Number h k h � k jEhj jEkj jEh�kj
1 300 040 3�40 3.5
</p>
<p>2 840 5 40 6.0
</p>
<p>3 570 2 70 10.0
</p>
<p>4 700 570 2�70 13.0
</p>
<p>5 800 670 2�70 10.1
</p>
<p>6 340 5�40 7.7
</p>
<p>7 411,0 4 11; 0 4.9
</p>
<p>8 040 8�40 4.2
</p>
<p>9 730 0 �4 0 770 3.1
</p>
<p>10 5 �4 0 270 6.9
</p>
<p>11 040 N
</p>
<p>12 340 7�70 �4 1 1; 0 4.1
</p>
<p>(continued)
</p>
<p>Tutorial Solutions 723</p>
<p/>
</div>
<div class="page"><p/>
<p>An origin at 0, 0 may be chosen by specifying 270 (eoe, and occurring four times) and 540 (oee,
</p>
<p>and occurring three times), both as +. From the S2 listing, we now have:
</p>
<p>Number Conclusion Comments
</p>
<p>10 s(730) = +
</p>
<p>7 s(800) = � s &eth;411; 0&THORN; &frac14; �s&eth;4 11; 0&THORN;
5 s(670) = +
</p>
<p>6 s(340) = � Sign propagation has ended.
Now let s(040) = a
</p>
<p>1 s(300) = �a
2 s(840) = �a
3 s(570) = �a
4 s(700) = a
</p>
<p>8 s(840) = �a
9 s(770) = a
</p>
<p>11 s(411,0) = �a
</p>
<p>Fig. S9.1
</p>
<p>13 540 N
</p>
<p>14 840 N
</p>
<p>15 270 N
</p>
<p>16 570 N
</p>
<p>17 670 N
</p>
<p>18 770 N
</p>
<p>19 411,0 N
</p>
<p>724 Tutorial Solutions</p>
<p/>
</div>
<div class="page"><p/>
<p>The symbol a would be determined by calculating electron density maps with both + and �
values, and assessing the results in terms of sensible chemical entities. In a more extended,
</p>
<p>experimental data set, the sign of a may evolve. No S2 relationship is noticeably weak, and the
</p>
<p>above solution to the problem may be regarded as acceptable.
</p>
<p>9.4. (a) Use Chap. 8, Sect. 8.5.1, (8.105); since a = g = 90 deg, the fourth and sixth terms on the
</p>
<p>right-hand side are zero. Thus, the bond length is 2.119 Å. From Sect. 8.6, (8.114), the esd
</p>
<p>evaluates to 0.0001 Å. Thus, we write S(1)&ndash;S(2) = 2.119(1) Å.
</p>
<p>(b) Writing down all Patterson vectors on the x,z projection of space group P21, we obtain:
</p>
<p>A
2x1, 2z1; 2x2, 2z2
&ndash;2x1, &ndash;2z1; &ndash;2x2, &ndash;2z2
B
x1 &ndash; x2, z1 &ndash; z2; &ndash;x1 &ndash; x2, &ndash;z1 &ndash; z2; &ndash;x1 + x2, &ndash;z1 + z2; x1 + x2, z1 + z2;
</p>
<p>&ndash;x1 &ndash; x2, &ndash;z1 + z2; x1 + x2, z1 + z2; x1 &ndash; x2, z1 &ndash; z2; &ndash;x1 &ndash; x2, &ndash;z1 &ndash; z2
Group A vectors are of single weight whereas group B vectors are of double weight. Hence the
</p>
<p>vectors around the origin would have the geometry shown in Fig. S9.2, and we expect the
</p>
<p>following arrangement, excluding the origin peak:
</p>
<p>where S1D1 = D1S2 = S3D3 = D3S4 and S2D2 = D2S3 = S4D4 = D4S1.
</p>
<p>Solutions 10
</p>
<p>10.1. The number N of unit cells in a crystal is V(crystal)/V(unit cell). Both crystals have the volume
</p>
<p>V(crystal) = 2.4 � 10�2 mm3. The protein unit-cell volume V(protein) = 60000 Å3, or
60000 � 10�21 mm3. The total number of protein unit cells NP is therefore = 4 � 10
</p>
<p>14.
</p>
<p>For the organic crystal unit cell, V(organic) = 1800 Å3, or 1800 � 10�21 mm3, so that the
</p>
<p>total number of organic unit cells N0 is 1.333 � 10
16.
</p>
<p>From Chap. 4, (4.1) and (4.2), we write
</p>
<p>E&eth;hkl&THORN; &frac14; &eth;I0=o&THORN;&eth;N
2l3&THORN;&frac12;e4=m2ec
</p>
<p>4�LpAjF&eth;hkl&THORN;j2V&eth;crystal&THORN; (S10.1)
</p>
<p>where N is the number of unit cells per unit volume of the crystal, L, p, and A are the Lorentz,
</p>
<p>polarization, and absorption correction factors, and the other symbols have their usual meanings.
</p>
<p>Historically, this equation was derived in 191416 and confirmed by careful measurements on
</p>
<p>a crystal of sodium chloride in 192117. In (S10.1), E (hkl) is the experimentally derived quantity
and jF(hkl)j is the term required in X-ray analysis. For our purposes, we write
</p>
<p>Fig. S9.2
</p>
<p>16Darwin CG (1914) Philosophical Magazine 27, 315.
17Bragg WL, James RW, Bosanquet CM (1921) Philosophical Magazine 42:1.
</p>
<p>Tutorial Solutions 725</p>
<p/>
</div>
<div class="page"><p/>
<p>E&eth;hkl&THORN; / N2 &frac14; &frac12;Ncells=V&eth;crystal&THORN;�2 (S10.2)
</p>
<p>where Ncells is the total number of unit cells in the crystal volume V(crystal). Since diffraction
</p>
<p>power D is proportional to energy, we have for the two cases under discussion:
</p>
<p>D &eth;organic&THORN;=D &eth;protein&THORN; &frac14; &frac12;Ncells&eth;organic&THORN; Ncells&eth;protein&THORN;= �2 &frac14; &frac12;&eth;1:333� 1016&THORN;=&eth;4� 1014&THORN;�
2
</p>
<p>&frac14; 1110:6
</p>
<p>Based on these considerations alone, the organic crystal will diffract over 1000 times more
</p>
<p>powerfully than the protein crystal. However, most protein data sets are now collected with
</p>
<p>synchrotron radiation, the intensity of which more than makes up for the deficiency in
</p>
<p>diffracting power calculated above. Other factors affect the intensity: in particular, it follows
</p>
<p>from Chap. 4, Sect. 4.2.1 that a local average value of jF(hkl)j2 is proportional to Ncf 2 if, for
simplicity, we assume an equal-atom structure, where Nc is here the number atoms per unit cell,
</p>
<p>which, to a first approximation, is proportional to the V (unit cell). Hence, the diffracting power
</p>
<p>of the crystal is directly proportional to V(unit cell), so that the above &ldquo;squared effect&rdquo; is
</p>
<p>somewhat diminished by the second factor.
</p>
<p>10.2. The experimental arrangement and coordinate systems are shown in the diagram, Fig. S10.2.
</p>
<p>For the powder ring, the two coordinates Yd and Zd will be the same, that is, 70 mm, and the
</p>
<p>distance D is 300 mm. The angle subtended from O by the diffracted beam is 2y so that
</p>
<p>tan2y = 70/300, or y = 6.654�. From the Bragg equation, l = 0.811 Å.
10.3. (Following on from 10.2.) Let the separation of spots for the 300 Å spacing be DZd; then
</p>
<p>DZd=D &frac14; tan 2y for a single diffraction order. Using the Bragg equation, we have 2 � 300 �
sin y = 0.811 and y = 0.0774 deg. If DZd &frac14; 1mm then D = 1/tan 2y = 370 mm. Using a value
of D of 450 mm will be more than adequate. Note that the intensity falls off as the square of the
</p>
<p>distance, so that, in practice, moving the detector too far away will be costly in terms of lost
</p>
<p>data for a weakly diffracting protein crystal.
</p>
<p>10.4. The information on limiting conditions indicates that there is either a 61 or a 65 screw axis in the
</p>
<p>crystal (Chap. 2, Table 10.2). As the Laue symmetry is
6
</p>
<p>m
mm, it follows from Chap. 10, Table
</p>
<p>Fig. S10.2
</p>
<p>726 Tutorial Solutions</p>
<p/>
</div>
<div class="page"><p/>
<p>10.1 that space group is either P6122 or P6522. Only the X-ray analysis can resolve this
</p>
<p>remaining ambiguity. Note that 61 and 65 screw operations are left-hand&ndash;right-hand opposites;
</p>
<p>only one can be correct for a given protein crystal.
</p>
<p>10.5. The volume Vc is 3.280 � 10
6 Å3. Substituting known values into the equation Dc =mZMPmu/
</p>
<p>Vc(1 � s) gives 0.383 m/(1 � s) for Dc, where m is the number of molecules per asymmetric
unit, and s is the fractional solvent content to be found by trial and error. Assuming that m is 1
</p>
<p>molecule per asymmetric unit and s is 0.68, that is, the crystal contains 68% solvent by weight
</p>
<p>(the top value of the known range), then it follows that Dc = 1.20 g cm
�3, a reasonable result.
</p>
<p>Note that we could make s = 0.70, slightly higher than normal, and this would give Dc = 1.28
</p>
<p>g cm�3, which is again quite acceptable. The important result for the structure analysis is that
m = 1 so that Z = 12.
</p>
<p>As s from the above analysis is on the high side, we increase the number of molecules m to 2.
</p>
<p>Then Dc = 2 � 0.383 m/(1 � s), or 0.766/(1 � s) which, for Dc = 1.4 g cm3 (see Chap. 10,
Sect. 10.4.7), gives s = 0.45. This result is again reasonable, so that there is some ambiguity for
</p>
<p>this protein. All that can be done is to bear these results in mind during the X-ray analysis, and
</p>
<p>make use of any other facts which are known about the crystal. In the case of the protein MLI, it
</p>
<p>was known that the crystals diffracted X-rays only poorly, which is often a sign of high solvent
</p>
<p>content, and this fact is more consistent with m = 1.
</p>
<p>10.6. The expected number of reflections &frac14; 4:19Vc=d3min, or 563450; this number includes all
symmetry-related reflections. Since the Laue symmetry (Chap. 1, Table 1.6) is
</p>
<p>6
</p>
<p>m
mm, the
</p>
<p>number of unique data is 1/24 times the number in the complete sphere, namely, 23479. If only
</p>
<p>21000 reflections are recorded, the data set would be approximately 89 % complete at the
</p>
<p>nominal resolution of 2.9 Å. This result corresponds more appropriately to 3.0 Å resolution
</p>
<p>(working backwards). Note that the above discussion is based on the number of reciprocal
</p>
<p>lattice points scanned in data collection and processing. Because protein crystals diffract
</p>
<p>poorly the number of reflections with significant intensities may well be as low as 50 %.
</p>
<p>These weak data do actually contain structural information and will usually be retained in the
</p>
<p>working data set.
</p>
<p>10.7. The asymmetric unit is one protein molecule. About 10% of the 27000 Da is hydrogen leaving
</p>
<p>27000 � 2700 = 24300 Da, which is equivalent to 2025 carbon atoms (C = 12). For the atoms
in the water molecules to be located (neglecting hydrogen atoms) we add a further 30 % of this
</p>
<p>number. The total number of non-hydrogen atoms to be located is then 2025 + 608, or 2633
</p>
<p>(O = 16). The number of parameters required for isotropic refinement (3 positional and 1
</p>
<p>temperature factor per non-hydrogen atom) is (2633 � 4) + 1 (scale factor), or 10533. The
unit-cell volume is 58.2 � 38.3 � 54.2 sin (106.5), which equates to 115840 Å3. Using the
</p>
<p>equation for the number N of reciprocal lattice points in the whole sphere at a given resolution
</p>
<p>limit, 4:19Vc d
3
min
</p>
<p>�
, and dividing by a factor 4 for (Laue group 2/m) we have the following
</p>
<p>results for the different resolutions:
</p>
<p>Reflections in 1 asymmetric unit Data/parameter ratio
</p>
<p>6 Å N = 2246/4 = 562 0.053
</p>
<p>2.5 Å N = 31066/4 = 7766 0.74
</p>
<p>1 Å N = 485400/4 = 121342 11.7
</p>
<p>Comments. The 6 Å structure is completely unrefinable. The 2.5 Å structure is refinable, but
</p>
<p>only if heavily restrained. The 1 Å isotropic model structure should refine provided the data
</p>
<p>quality is adequate.
</p>
<p>Tutorial Solutions 727</p>
<p/>
</div>
<div class="page"><p/>
<p>10.8. From the general expression Web Appendix A4, (WA4.6), with g = 120� and f = 120�, we
derive the matrix
</p>
<p>0 �1 0
</p>
<p>1 �1 0
</p>
<p>0 0 1
</p>
<p>2
64
</p>
<p>3
75 which; together with the translation vector
</p>
<p>0
</p>
<p>0
2
3
</p>
<p>2
64
</p>
<p>3
75
</p>
<p>for the 32 screw axis, leads to the general equivalent position set: x, y, z; �y, x� y; 23 &thorn; z;
y � x; �x; 1
</p>
<p>3
&thorn; z. The only condition limiting reflections is 000l: = 3n.
</p>
<p>10.9. If the protein belongs to a family, or group, of proteins having similar functions or biological
</p>
<p>or other properties in common, and the structure of one member of the family is known, either
</p>
<p>from an ab initio or other structure determination, molecular replacement can be attempted.
</p>
<p>The method usually requires the two proteins involved in MR to have amino acid sequences
</p>
<p>which correspond either identically or are of very similar types, that is, conserved for at least
</p>
<p>30% of their total lengths (30% homology). Note that if the two proteins crystallize in the same
</p>
<p>space group and have very similar unit cells, they are very likely to be isomorphous, and the
</p>
<p>new structure should be determinable initially by Fourier methods alone. If the protein belongs
</p>
<p>to a new family for which no known structures exist, an ab inito method, MIR or MAD, has to
</p>
<p>be used for structure analysis. In the case of MAD, a tuneable source of synchrotron radiation
</p>
<p>is required.
</p>
<p>Solutions 11
</p>
<p>11.1. (a) From the Bragg equation, 1.25 = 2d(111) sin y(111). Since d&eth;111&THORN; &frac14; a=p3,
y(111) = 12.62�. (b) Differentiating the Bragg equation with respect to y, we obtain
dl &frac14; 2d&eth;111&THORN; cos y&eth;111&THORN; dy. Remembering that dy here is measured in radian, dl = 0.0243 Å.
</p>
<p>11.2. For the NaCl structure type, we can write F(hkl) = 4[fNa
+ + (�1)lfH�/D�]. Hence, the following
</p>
<p>results are obtained:
</p>
<p>(111) (220)
</p>
<p>NaH NaD NaH NaD
</p>
<p>X-rays 30.9 30.9 27.6 27.6
</p>
<p>Neutrons 2.88 &ndash;1.28 &ndash;0.08 4.08
</p>
<p>11.3. VIVALDI uses a white-beam Laue technique to record the diffraction data. Consequently each
</p>
<p>diffraction record (spot) will have a different wavelength associated with it which will have first
</p>
<p>to be determined in order for the spot to be assigned its hkl indices. This would usually require
</p>
<p>the unit cell of the crystal to be known, which would be easier to carry out first with
</p>
<p>monochromatic X-rays in the user&rsquo;s laboratory.
</p>
<p>11.4. The spallation neutron beam at ORNL has been filtered in order to cover as small a wavelength
</p>
<p>range as is required in order to measure the data on a four-circle diffractometer. This would
</p>
<p>result in some loss of beam flux (power) which, in turn, would require the use of larger crystals
</p>
<p>in order to produce good quality diffraction data. VIVALDI uses a Laue white radiation
</p>
<p>technique which does not require the use of flux reducing filters.
</p>
<p>11.5. Applying Chap. 11, (11.3) gives l = 1.865 Å. [Work out the units of A in (11.3).]
</p>
<p>11.6. The final picture should look something like Chap. 11, Fig. 11.12. It can be exported and saved
</p>
<p>in a variety of ways from the RASMOL menu.
</p>
<p>728 Tutorial Solutions</p>
<p/>
</div>
<div class="page"><p/>
<p>Solutions 12
</p>
<p>12.1. Since R = 57.30 mm, 1 mm on the film is equal to 1� in y. Thus, 0.5 mm = 0.5� = 0.00873 rad.
The mean Cu Ka wavelength is 1.5418 Å. Differentiating the Bragg equation with respect to y:
</p>
<p>dl &frac14; 2d cos y dy &frac14; l cot y dy
</p>
<p>Since dl = 0.0038, we have cot y = 0.0038/(1.5418 � 0.00873) = 0.2823, so that y 
 74�,
the angle at which the a1a2 doublet would be resolved under the given conditions.
</p>
<p>12.2. Perusal of the cubic unit-cell types leads us to expect that (sin2y)/n for the first line (the low y
</p>
<p>region) where n = 1, 2, 3, . . . , would result in a factor that would divide into all other experimental
</p>
<p>values of sin2y to give integer or near-integer results.By trial,wefind that, for thefirst line, (sin2y)/3
</p>
<p>leads to a sequence of values that correspond closely to those for a cubicF unit cell. Thus, dividing
</p>
<p>all other values of sin2y by 0.0155 we obtain:
</p>
<p>Line no. sin2y/0.0155 N a/Å hkl y/o
</p>
<p>1 3.00 3 6.192 111 12.45
</p>
<p>2 4.10 4 6.118 200 14.60
</p>
<p>3 11.08 11 6.170 311 24.48
</p>
<p>4 16.04 16 6.185 400 29.91
</p>
<p>5 23.95 24 6.199 422 37.54
</p>
<p>6 26.90 27 6.203 333, 511 40.22
</p>
<p>7 34.80 35 6.210 531 47.26
</p>
<p>8 35.77 36 6.212 600, 442 48.12
</p>
<p>9 42.64 43 6.218 533 54.39
</p>
<p>10 47.54 48 6.222 444 59.13
</p>
<p>Some accidental absences appear in this sequence of lines. Extrapolation of a v. f(y) by the
</p>
<p>method of least squares (program LSLI) gives a = 6.217 Å. However, lines 1 and 2 produce
</p>
<p>significantly greater errors of fit than do the remaining eight lines. Since low-angle measure-
</p>
<p>ments tend to be less reliable, we can justifiably exclude lines 1 and 2. Least squares on lines 3
</p>
<p>to 10 gives a probably better value, a = 6.223 Å.
</p>
<p>12.3. The LEPAGE program gives the following results with the C-factor set at 1�:
Reduced Cell: P 4.693, 4.929, 5.679 Å; 90.12, 90.01, 90.72�
</p>
<p>Conventional cell: Orthorhombic P 4.693, 4.929, 5.679 Å; 90.12, 90.01, 90.72�,
where all three angles are assumed to be 90� within experimental error. If we select the more
stringent LEPAGE parameter C = 0.5�, we obtain
Reduced Cell: P 4.693, 4.929, 5.679 Å; 90.12, 90.01, 90.7�
</p>
<p>Conventional cell: Monoclinic P 4.693, 5.679, 4.929 Å; 90.72, 89.99�
</p>
<p>where a and gmay taken now to be 90� within experimental error. The parameters are reordered
so that b is the unique angle.
</p>
<p>12.4. From the program LEPAGE, we find:
</p>
<p>Reduced Cell: P 6.021, 6.021, 8.515 Å; 110.70, 110.70, 90.01�
</p>
<p>Conventional cell: Tetragonal I 6.021, 6.021, 14.75 Å; 90.00, 90.00, 90.00�
</p>
<p>V(conventional unit cell)/V(given unit cell) = 2.
</p>
<p>Tutorial Solutions 729</p>
<p/>
</div>
<div class="page"><p/>
<p>12.5. Let lines 1, 2, and 3 be 100, 010, and 001, respectively. Then, from multiples of their Q values,
</p>
<p>we have a* = 0.09118 (average of 100, 200, 300, and 400; lines 1, 5, 17, and 33, respectively),
</p>
<p>b* = 0.09437 (average of 010, 020, 030, and 040; lines 2, 6, 18, and 38, respectively), and
</p>
<p>c* = 0.1312 (average of 002 and 003; lines 3 and 15, respectively). Consider next the possible
</p>
<p>hk0 lines. Q110 = Q100 + Q010 = 172.2; this line has been allocated to 001, which is probably
</p>
<p>erroneous. Continue with the [001] zone:
</p>
<p>hk0 Qhk0 Line number
</p>
<p>110 172.2 3
</p>
<p>120 439.5 9
</p>
<p>130 885.0 20
</p>
<p>210 421.5 8
</p>
<p>220 688.8 15
</p>
<p>230 1134.3 29
</p>
<p>310 837.0 19
</p>
<p>320 1104.3 27
</p>
<p>This zone is well represented, and it follows that g* is 90�. If line 4 is now taken as 001, then
line 25 could be 002. We check this assignment by forming expected Q0kl values:Q011 = 338.9,
</p>
<p>but there is no line at this Q value, nor a pair of lines equidistant above and below this value,
</p>
<p>as there would be if the assignment is correct and a* 6&frac14; 90�. Q021 = 606.2, but this line fails the
above test. It seems probable that line 4 is not 001. However, it must involve the l index and
</p>
<p>one of the indices h or k. If it is 101, then, for b* = 90�, Q001 = 249.8 � 83.1 = 166.7, and if
it is 011, then, for a* = 90�, Q001 = 249.8 � 89.1 = 160.7. For the second of these assign-
ments, although a line at 160.7 is not present, there are the multiples 002 and 003 at lines
</p>
<p>12 (642.8) and 39 (1446.3), respectively. With this assumption, line 4 is 011, line 12 is 002,
</p>
<p>and line 39 is 003.
</p>
<p>Confirmation arises from 012, 021, and 022 at lines 16 (731.9), line 10 (517.1), and line 25
</p>
<p>(999.2). Thus, an average c* = 0.1267 and a* = 90�. We now search for h0l lines. For b* = 90�,
Q101 = 243.8; this line cannot be fitted into the pattern. Q102 = 725.9: there is no line at this
</p>
<p>value, but lines 11 and 21 are very nearly equidistant (166.1 and 166.5) from 725.9. Hence, the
</p>
<p>difference, 332.6 is 104 (8c* a* cos b*), so that b* = 68.91�. We have now a set of reciprocal
unit-cell parameters from which, since two angles are 90�, the direct unit cell is calculated as
b = 111.09�, a = 1/(a* sin b) = 11.755, b = 1/b* = 10.597, c = 1/(c* sin b) = 8.459 Å. We
make the conventional interchange of a and c, so that b is the unique angle, and c &gt; b &gt; a, and
</p>
<p>now apply the further check of calculating the Q values for this unit cell, using the program
</p>
<p>QVALS, with the results listed in Table S12.1.
</p>
<p>In using this program, we remember that the unit cell appears to be monoclinic, so that we
</p>
<p>need to consider hkl and h k �l reflections. From Table S12.1, it is evident that several reflections
</p>
<p>overlap, within the given experimental error. The unit cell type is P. The h0l reflections are
</p>
<p>present only when h is an even integer. The reflections 30�3 and 300, at the Q values 1444.8 and
</p>
<p>1444.9, respectively are probably not present and overlapped by the 23�2 and 230. Hence, the
</p>
<p>space group is probably Pa (non-standard form of Pc) or P2/a (non-standard form of P2/c).
</p>
<p>To consider if the symmetry is actually higher than monoclinic, the unit cell is reduced, using
</p>
<p>730 Tutorial Solutions</p>
<p/>
</div>
<div class="page"><p/>
<p>the program LEPAGE. We find that the first unit cell is reduced, but the conventional unit cell is
</p>
<p>orthorhombic B (� C or A), with a high degree of precision:
</p>
<p>a &frac14; 8:459; b &frac14; 10:597; c &frac14; 21:935 A; a &frac14; b &frac14; g &frac14; 90:00o
</p>
<p>Since Miller indices transform as unit-cell vectors, we find from the transformation matrix
</p>
<p>given by the program LEPAGE that hB = �h, kB = �k, and lB = h + 2l; the transformed
indices are listed in Table S12.2. We note that the indices are listed as directly transformed.
</p>
<p>If we were dealing with structure factors, we could negate all the negative indices, because
</p>
<p>jF&eth;hkl&THORN;j&frac14;jF&eth;h k l&THORN; &frac14; jF&eth;h k l&THORN;j &frac14; jF&eth;h �k l&THORN;j &frac14; jF&eth;h k �l&THORN;j in the orthorhombic system.
</p>
<p>Table S12.1 Observed and calculated Q values
for substance X and the hkl indices of the lines
</p>
<p>referred to the first unit cell
</p>
<p>Q(obs) hkl Q(calc)
</p>
<p>83.1 0 0 1 83.1
</p>
<p>89.1 0 1 0 89.1
</p>
<p>172.2 0 1 1 172.2
</p>
<p>249.8 1 1 0 249.6
</p>
<p>1 1 �1 249.6
</p>
<p>332.6 0 0 2 332.5
</p>
<p>356.1 0 2 0 356.2
</p>
<p>416.0 1 1 �2 415.8
</p>
<p>1 1 1 415.9
</p>
<p>421.5 0 1 2 421.6
</p>
<p>439.3 0 2 1 439.3
</p>
<p>516.9 1 2 �1 516.7
</p>
<p>1 2 0 516.7
</p>
<p>559.8 2 0 �1 559.0
</p>
<p>642.9 2 0 �2 642.1
</p>
<p>2 0 0 642.2
</p>
<p>648.6 2 1 �1 648.1
</p>
<p>683.3 1 2 �2 683.0
</p>
<p>1 2 1 683.0
</p>
<p>688.8 0 2 2 688.7
</p>
<p>732.1 2 1 �2 731.2
</p>
<p>2 1 0 731.2
</p>
<p>748.4 0 0 3 748.2
</p>
<p>1 1 �3 748.4
</p>
<p>1 1 2 748.4
</p>
<p>801.5 0 3 0 801.5
</p>
<p>837.2 0 1 3 837.3
</p>
<p>884.5 0 3 1 884.6
</p>
<p>892.4 2 0 �3 891.5
</p>
<p>2 0 1 891.6
</p>
<p>916.0 2 2 �1 915.2
</p>
<p>962.3 1 3 �1 962.0
</p>
<p>1 3 0 962.0
</p>
<p>(continued)
</p>
<p>Tutorial Solutions 731</p>
<p/>
</div>
<div class="page"><p/>
<p>From an inspection of the hkl indices in Table S12.2 for the transformed unit cell, we find
</p>
<p>hkl: h + l = 2n
</p>
<p>0kl: None
</p>
<p>h0l: h = 2n; (l = 2n)
</p>
<p>hk0: (h = 2n)
</p>
<p>Table S12.1 (continued)
</p>
<p>981.7 2 1 �3 980.6
</p>
<p>2 1 1 980.6
</p>
<p>999.1 2 2 �2 998.3
</p>
<p>2 2 0 998.4
</p>
<p>1016. 1 2 �3 1015.5
</p>
<p>1 2 2 1015.6
</p>
<p>1015. 0 2 3 1104.4
</p>
<p>1129. 1 3 �2 1128.2
</p>
<p>1 3 1 1128.3
</p>
<p>1134. 0 3 2 1134.0
</p>
<p>1248. 1 1 �4 1247.2
</p>
<p>1 1 3 1247.2
</p>
<p>1249. 2 2 �3 1247.7
</p>
<p>2 2 1 1247.8
</p>
<p>1308. 2 0 �4 1307.2
</p>
<p>2 0 2 1307.3
</p>
<p>1330. 0 0 4 1330.1
</p>
<p>1361. 2 3 �1 1360.5
</p>
<p>1369. 3 1 �2 1367.6
</p>
<p>31�1 1367.6
</p>
<p>1397. 2 1 �4 1396.2
</p>
<p>2 1 2 1396.3
</p>
<p>1419. 0 1 4 1419.2
</p>
<p>1425. 0 4 0 1424.8
</p>
<p>1444. 2 3 �2 1443.6
</p>
<p>2 3 0 1443.6
</p>
<p>1461. 1 3 �3 1460.8
</p>
<p>1 3 2 1460.8
</p>
<p>Table S12.2 Transformation of the hkl indices from the monoclinic (first) unit cell to the orthorhombic B unit
cell
</p>
<p>h k l hB kB lB h k l hB kB lB
</p>
<p>0 0 1 ! 0 0 2 1 3 �1 ! �1 �3 �1
0 1 0 ! 0 �1 0 1 3 0 ! �1 �3 1
0 1 1 ! 0 �1 2 2 1 �3 ! �2 �1 �4
1 1 0 ! �1 �1 1 2 1 1 ! �2 �1 4
1 1 �1 ! �1 �1 �1 2 2 �2 ! �2 �2 �2
</p>
<p>(continued)
</p>
<p>732 Tutorial Solutions</p>
<p/>
</div>
<div class="page"><p/>
<p>giving the diffraction symbol as B . a . which, under the transformation a0 = a, b0 = c, c0 = �c,
becomes C � � a. Hence, the space group is either Cmma or C2ma (� Abm2).
</p>
<p>12.6. Crystal XL1: a = 6.425, b = 9.171, c = 5.418 Å, a = 90, b = 90, g = 90�. The unit cell is
orthorhombic. The systematic absences indicate the diffraction symbol as mmm P n a � � ,
</p>
<p>which corresponds to either Pna21 or Pnam. The latter is the ac b setting of Pnma. [Reported:
KNO3; 9.1079, 6.4255, 5.4175 Å; Pbnm, which is the cab setting of Pnma.] The LEPAGE
reduction confirms the above cell as reduced and conventional, under reordering, such that
</p>
<p>a &lt; b &lt; c. What is the space group now?
</p>
<p>12.7. Crystal XL2: a = 10.482, b = 11.332, c = 3.757 Å, a = 90, b = 90, g = 90�. The unit cell is
orthorhombic, with space group Pbca. The LEPAGE reduction confirms the above cell as
</p>
<p>reduced and conventional, under reordering such that a &lt; b &lt; c.
</p>
<p>12.8. Crystal XL3: a = 6.114, b = 10.722, c = 5.960 Å, a = 97.59, b = 107.25, g = 77.42�. The unit
cell is triclinic, space group P1 or P�1. The LEPAGE reduction gives a = 5.960, b = 6.114,
</p>
<p>c = 10.722 Å, a = 77.42, b = 82.41, g = 72.75�. [Reported: CuSO4.5H2O: 6.1130, 10.7121,
5.9576 Å, 82.30, 107.29, 102.57�; P�1.]
</p>
<p>Table S12.2 (continued)
</p>
<p>h k l hB kB lB h k l hB kB lB
</p>
<p>0 0 2 ! 0 0 4 2 2 0 ! �2 �2 2
0 2 0 ! 0 �2 0 1 2 �3 ! �1 �2 �5
1 1 �2 ! �1 �1 �3 1 2 2 ! �1 �2 5
1 1 1 ! �1 �1 3 0 2 3 ! 0 �2 6
0 1 2 ! 0 �1 4 1 3 �2 ! �1 �3 �3
0 2 1 ! 0 �2 2 1 3 1 ! �1 �3 3
1 2 �1 ! �1 �2 �1 0 3 2 ! 0 �3 4
1 2 0 ! �1 �2 1 1 1 �4 ! �1 �1 �7
2 0 �1 ! �2 0 0 1 1 3 ! �1 �1 7
2 0 �2 ! �2 0 �2 2 2 �3 ! �2 �2 �4
2 0 0 ! �2 0 2 2 2 1 ! �2 �2 4
2 1 �1 ! �2 �1 0 2 0 �4 ! �2 0 �6
1 2 �2 ! �1 �2 �3 2 0 2 ! �2 0 6
1 2 1 ! �1 �2 3 0 0 4 ! 0 0 8
0 2 2 ! 0 �2 4 2 3 �1 ! �2 �3 0
2 1 �2 ! �2 �1 �2 3 1 �2 ! �3 �1 �1
2 1 0 ! �2 �1 2 3 1 �1 ! �3 �1 1
0 0 3 ! 0 0 6 2 1 �4 ! �2 �1 �6
1 1 �3 ! �1 �1 �5 2 1 2 ! �2 �1 6
1 1 2 ! �1 �1 5 0 1 4 ! 0 �1 8
0 3 0 ! 0 �3 0 0 4 0 ! 0 �4 0
0 1 3 ! 0 �1 6 2 3 �2 ! �2 �3 �2
0 3 1 ! 0 �3 2 2 3 0 ! �2 �3 2
2 0 �3 ! �2 0 �4 1 3 �3 ! �1 �3 �5
2 0 1 ! �2 0 4 1 3 2 ! �1 �3 5
2 2 �1 ! �2 �2 0
</p>
<p>Tutorial Solutions 733</p>
<p/>
</div>
<div class="page"><p/>
<p>Solutions 13
</p>
<p>The solutions given here apply to the structure determinations of (1) the nickel o-phenanthroline
</p>
<p>complex (NIOP) and (2) 2-S-methylthiouracil (SMTX&amp; SMTY). The correctness of the other XRAY
</p>
<p>structure examples should be judged by both the state of the refinement achieved and the chemical
</p>
<p>plausibility of the structure, as discussed in Sect. 8.7.
</p>
<p>13.1. NIOP
Table S13.1 lists the refined x, y and B parameters for the atoms in the Ni o-phenanthroline
</p>
<p>complex; two-dimensional refinement by XRAY to R 
 9.8 %.
</p>
<p>Table S13.1
</p>
<p>Atom x y Pop. B/Å2
</p>
<p>Ni 0.23511 0.17804 1.000 2.02
</p>
<p>S1 0.31780 0.10370 1.000 2.10
</p>
<p>S2 0.15409 0.10022 1.000 2.21
</p>
<p>C1 0.46164 0.15015 1.000 2.05
</p>
<p>C2 0.39174 0.22786 1.000 2.22
</p>
<p>C3 0.32252 0.24182 1.000 3.55
</p>
<p>C4 0.14559 0.23252 1.000 1.14
</p>
<p>C5 0.08009 0.21567 1.000 2.83
</p>
<p>C6 0.00165 0.13681 1.000 2.08
</p>
<p>C7 0.00321 0.95462 1.000 5.63
</p>
<p>C8 0.27653 0.44349 1.000 2.91
</p>
<p>C9 0.47796 0.08704 1.000 0.85
</p>
<p>C10 0.39398 0.16063 1.000 2.09
</p>
<p>C11 0.33069 0.30042 1.000 3.47
</p>
<p>C12 0.27678 0.33480 1.000 2.85
</p>
<p>C13 0.31090 0.39206 1.000 2.40
</p>
<p>C14 0.18671 0.44236 1.000 4.71
</p>
<p>C15 0.15426 0.37717 1.000 3.69
</p>
<p>C16 0.19049 0.33477 1.000 1.65
</p>
<p>C17 0.14168 0.29263 1.000 1.35
</p>
<p>C18 0.07108 0.16047 1.000 3.90
</p>
<p>Note: The total number of non-hydrogen atoms in the molecule is 21. The ID
</p>
<p>numbers refer to atoms as follow: 1 Ni, 2 S, 3 N, 4 C.
</p>
<p>734 Tutorial Solutions</p>
<p/>
</div>
<div class="page"><p/>
<p>13.2. SMTX and SMTY
Figure S13.1 shows the molecular structure of 2-S-methylthiouracil. Not all angles have been
</p>
<p>listed; the values for S&ndash;CH3 and C(l)&ndash;S&ndash;CH3 will evolve from your result for the position of
</p>
<p>the &ndash;CH3 group.
</p>
<p>Fig. S13.1
</p>
<p>Tutorial Solutions 735</p>
<p/>
</div>
<div class="page"><p/>
<p>Index
</p>
<p>A
Ab initio methods, 425, 438, 463
</p>
<p>Absences in x-ray diffraction spectra
</p>
<p>accidental, 142
</p>
<p>local average intensity for, 287
</p>
<p>systematic
</p>
<p>for centered unit cell, 142, 252
</p>
<p>and geometric structure factor, 144ff
</p>
<p>for glide planes, 142&ndash;152, 176
</p>
<p>and limiting conditions, 142&ndash;152
</p>
<p>and m plane, 154
</p>
<p>for screw axes, 142ff, 176
</p>
<p>and translational symmetry, 142,
</p>
<p>152, 176
</p>
<p>Absolute configuration of chiral entities, 328
</p>
<p>Absorption
</p>
<p>coefficients, 114, 156, 337
</p>
<p>correction, 166, 211, 233, 406,
</p>
<p>426, 725
</p>
<p>edge, 114, 325, 335
</p>
<p>effects
</p>
<p>with neutrons, 564
</p>
<p>with x-rays, 114ff, 164, 406
</p>
<p>measurement of, 165ff
</p>
<p>Accuracy. See Precision
</p>
<p>3-β-Acetoxy-6,7-epidithio-19-norlanosta-5,7,9,11-
</p>
<p>tetraene
</p>
<p>absolute configuration of, 465
</p>
<p>chemistry of, 465
</p>
<p>crystal structure of, 465
</p>
<p>Airy disk, 247, 249
</p>
<p>Alkali-metal halides, 4
</p>
<p>Alternating axis of symmetry, 663
</p>
<p>Alums, crystal structure of, 346
</p>
<p>Amorphous substance, 7, 585
</p>
<p>Amplitude symmetry, 159, 358, 433. See also
</p>
<p>Phase, symmetry
</p>
<p>Angle. See also Bond lengths and angles
</p>
<p>Bragg, 201
</p>
<p>dihedral, 428, 469
</p>
<p>Eulerian, 518
</p>
<p>of incidence, 200, 225
</p>
<p>interaxial, 7, 9, 26, 55, 605
</p>
<p>interfacial, 1, 15, 137, 226
</p>
<p>between lines, 134
</p>
<p>between planes, 137, 138, 411, 590
</p>
<p>torsion, 411, 652
</p>
<p>Ångstr&oslash;m unit, 111
</p>
<p>Angular frequency
</p>
<p>Anisotropic thermal vibration. See Thermal vibrations
</p>
<p>Anisotropy, optical, 192. See also Biaxial crystals;
</p>
<p>Uniaxial crystals
</p>
<p>Anomalous dispersion, 30, 468, 505, 527
</p>
<p>Anomalous scattering
</p>
<p>and diffraction symmetry, 330ff
</p>
<p>and heavy atoms, 333ff
</p>
<p>and phasing reflections, 325, 334
</p>
<p>and protein phasing, 337
</p>
<p>and structure factor, 325, 332ff, 338
</p>
<p>and symmetry, 330&ndash;332
</p>
<p>Aperiodic crystals, 37, 51
</p>
<p>Aperiodic structure, 38, 338
</p>
<p>Area detector, 167, 187, 197, 205ff, 233,
</p>
<p>504, 563ff. See also Intensity measurement
</p>
<p>Argand diagram, 123ff, 156, 176ff, 321,
</p>
<p>336, 362
</p>
<p>Assemblage, 17
</p>
<p>Asymmetric unit, 21, 73
</p>
<p>Asymmetry parameter, 413
</p>
<p>Atom
</p>
<p>mass of, 156
</p>
<p>scattering by, 335, 347
</p>
<p>Atomic number, 130, 286
</p>
<p>Atomic scattering factor
</p>
<p>and anomalous scattering, 325, 332ff
</p>
<p>corrections to, 485
</p>
<p>and electron density, 129
</p>
<p>exponential formula for, 670
</p>
<p>factors affecting, 161ff
</p>
<p>and spherical symmetry of atoms, 247
</p>
<p>temperature correction of, 171
</p>
<p>variation with, 248
</p>
<p>ATPsynthase
</p>
<p>docking with oligomycin, 471, 481
</p>
<p>domains in, 472ff
</p>
<p>structure of, 480
</p>
<p>Attenuation, 116, 164, 257
</p>
<p>Average intensity multiple. See Epsilon
</p>
<p>(ε) factor
</p>
<p>Averaging function. See Patterson, function
</p>
<p>M. Ladd and R. Palmer, Structure Determination by X-ray Crystallography:
</p>
<p>Analysis by X-rays and Neutrons, DOI 10.1007/978-1-4614-3954-7,
# Springer Science+Business Media New York 2013
</p>
<p>737</p>
<p/>
</div>
<div class="page"><p/>
<p>Axes
</p>
<p>Cartesian right-handed, 167
</p>
<p>conventional, 10, 60, 81
</p>
<p>crystallographic, 8ff
</p>
<p>for hexagonal system, 22, 59
</p>
<p>Axial ratios, 12
</p>
<p>B
Background scattering
</p>
<p>with neutrons, 554
</p>
<p>with x-rays, 554, 593
</p>
<p>Balanced filter, 223, 225. See also Monochromators
</p>
<p>Barlow, W., 3
</p>
<p>Bayesian statistics, 175
</p>
<p>Beam stop, 442
</p>
<p>Beevers&ndash;Lipson strips, 345
</p>
<p>Bernal, J.D., 138, 491, 541
</p>
<p>Bessel function, 246, 266, 365, 399
</p>
<p>B factor, 169ff, 173
</p>
<p>Biaxial crystals
</p>
<p>optical behaviour of, 190
</p>
<p>refractive indices of, 190
</p>
<p>Bijvoet difference, 305, 331
</p>
<p>Bijvoet pairs, 306, 327ff, 347, 527, 718
</p>
<p>Biodeuteration. See Perdeuteration
</p>
<p>Bioinformatics, 471, 481. See also ATP synthase;
</p>
<p>Oligomycins A, B, C
</p>
<p>Biological molecules, 489
</p>
<p>Biomolecular modelling, 471
</p>
<p>Birefringence, 192ff
</p>
<p>Bond lengths and angles
</p>
<p>neutron, 417
</p>
<p>tables of, 410
</p>
<p>x-ray, 410
</p>
<p>Bragg equation, 132, 134
</p>
<p>Bragg reflection (diffraction)
</p>
<p>and Laue diffraction, 200, 223
</p>
<p>equivalence of with Bragg reflection, 134
</p>
<p>order of, 133
</p>
<p>Bragg, W.H., 4
</p>
<p>Bragg, W.L., 4
</p>
<p>Bravais, A., 3
</p>
<p>Bravais lattices
</p>
<p>and crystal systems, 54, 60, 64
</p>
<p>direct, 63
</p>
<p>notation and terminology of, 61
</p>
<p>plane, 55, 57, 62
</p>
<p>reciprocal, 63ff
</p>
<p>representative portion of, 52, 60, 74
</p>
<p>rotational symmetry of, 52, 71, 74, 94, 98
</p>
<p>and space groups, 72ff
</p>
<p>symmetry of, 70, 97&ndash;99
</p>
<p>tables of, 59
</p>
<p>three-dimensional, 54
</p>
<p>translations, 51 ff, 72
</p>
<p>two-dimensional, 52ff, 73ff
</p>
<p>unit cells of, 52ff, 72ff, 86, 93
</p>
<p>vector, 65ff
</p>
<p>Buckyballs, 32&ndash;39
</p>
<p>C
Camera methods, importance of, 588, 590
</p>
<p>Capillary crystal mount, 496, 508, 568
</p>
<p>Carangeot, A., 1, 15
</p>
<p>Carbon monoxide, molecular symmetry of, 31
</p>
<p>CCD. See Charge-coupled device (CCD);
</p>
<p>Charge-coupled type area detector (CCD)
</p>
<p>CCP4. See Collaborative computational projects
</p>
<p>CCP14. See Collaborative computational projects
</p>
<p>Central limit theorem, 176
</p>
<p>Centred unit cells
</p>
<p>in three dimensions, 56
</p>
<p>Centre of symmetry (inversion)
</p>
<p>alternative origins on, 95
</p>
<p>and diffraction pattern, 330
</p>
<p>Centric reflection. See Signs of reflections in
</p>
<p>centrosymmetric crystals
</p>
<p>Centric zones, 146, 178, 703
</p>
<p>Centrosymmetric crystals
</p>
<p>point groups of, 18&ndash;31
</p>
<p>projection of, 17, 23ff, 36
</p>
<p>structure, 32, 36
</p>
<p>structure factor for, 141
</p>
<p>and x-ray patterns, 103
</p>
<p>zones (see Centric zones)
</p>
<p>Change of hand, 22, 23, 491
</p>
<p>Change of origin, 90, 252, 353, 692. See also
</p>
<p>Origin, change of
</p>
<p>Characteristic symmetry, 24, 26
</p>
<p>Characteristic x-radiation, 113
</p>
<p>Charge-coupled type area detector (CCD), 217ff
</p>
<p>Chirality, 327, 534, 539&ndash;540
</p>
<p>Chi-square (χ2) distribution, 411
</p>
<p>Closure error, 321, 322
</p>
<p>CMOS. See Complementary metal-oxide
</p>
<p>semiconductor (CMOS)
</p>
<p>Coherent scattering. See Scattering, coherent
</p>
<p>Collaborative computational projects, 635
</p>
<p>Collimation, 117, 504, 594
</p>
<p>Collimator
</p>
<p>multiple beam, 116
</p>
<p>traditional, 116
</p>
<p>Complementary metal-oxide semiconductor
</p>
<p>(CMOS), 221
</p>
<p>Complex numbers
</p>
<p>conjugate of, 124, 367
</p>
<p>plane, 123, 325
</p>
<p>Compton scattering. See Scattering, incoherent
</p>
<p>Computer graphics, 325, 420, 506
</p>
<p>Computer prediction of crystal structure
</p>
<p>developments in, 425
</p>
<p>lattice energy and, 423
</p>
<p>programs for, 422
</p>
<p>Conformational parameters, 411, 471
</p>
<p>Constant interfacial angles, law of, 1, 15
</p>
<p>Convolution
</p>
<p>and crystal structure, 264
</p>
<p>and diffraction, 261
</p>
<p>folding integral, 262ff
</p>
<p>738 Index</p>
<p/>
</div>
<div class="page"><p/>
<p>and structure solution, 266ff
</p>
<p>transform of, 261ff
</p>
<p>Coordinates
</p>
<p>Cartesian, 414, 434, 617, 652
</p>
<p>fractional, 54, 65, 135
</p>
<p>orthogonal, 151, 720
</p>
<p>symmetry related, 74, 290, 308, 722
</p>
<p>transformation of axes of, 353
</p>
<p>Copper
</p>
<p>pyrites crystal, 30
</p>
<p>sulphate, 4, 5
</p>
<p>Corrections to measured intensities
</p>
<p>absorption, 165ff
</p>
<p>extinction, 164, 168
</p>
<p>Lorentz, 162, 168, 456
</p>
<p>polarization, 162, 168, 456
</p>
<p>scale, 169, 174, 456
</p>
<p>temperature factor, 169ff, 456
</p>
<p>Correct phases, importance of, 126
</p>
<p>Correlation coefficient, 523
</p>
<p>Cross vector, 318, 384, 520
</p>
<p>Crown ether derivative, 574
</p>
<p>Cryoprotectants for proteins, 491, 543
</p>
<p>Crystal. See also Crystalline substance
</p>
<p>class, 3, 24, 31, 146, 159, 177, 182, 331
</p>
<p>and ε factor, 177, 182, 352
</p>
<p>classification of
</p>
<p>by optics, 90
</p>
<p>by symmetry, 24
</p>
<p>definition of, 37
</p>
<p>density of
</p>
<p>calculated, 444
</p>
<p>measured, 444
</p>
<p>external symmetry of, 17ff
</p>
<p>faces of, 9ff, 28, 62, 166, 193
</p>
<p>geometry of, 26, 311
</p>
<p>growth of, 492ff
</p>
<p>habit of, 3, 187, 440, 455
</p>
<p>ideally imperfect, 164
</p>
<p>ideally perfect, 164
</p>
<p>imperfections in, 164, 586
</p>
<p>internal symmetry of (see Space groups)
</p>
<p>lattice (see Lattice)
</p>
<p>models of, 638
</p>
<p>monoclinic, 147, 152ff, 195, 288, 401, 434
</p>
<p>mosaic character in, 164, 165
</p>
<p>mounting of, 499ff, 508, 545, 703
</p>
<p>optical classification of, 190
</p>
<p>perfection of, 164, 197
</p>
<p>periodicity in, 35
</p>
<p>permitted symmetry of, 212
</p>
<p>point group (see Point groups)
</p>
<p>size of, 328, 494, 558, 577
</p>
<p>as stack of unit cells, 1, 52, 164
</p>
<p>symmetry and physical properties of, 17
</p>
<p>unit cell of, 53, 54
</p>
<p>Crystal growing
</p>
<p>by diffusion, 188
</p>
<p>from solution, 155, 188
</p>
<p>by sublimation, 188
</p>
<p>Crystalline state, 4, 6ff
</p>
<p>Crystalline substance, 7
</p>
<p>Crystallographic computing. See Software for
</p>
<p>crystallography
</p>
<p>Crystallographic point groups. See also
</p>
<p>Point groups
</p>
<p>and crystal systems, 3, 24ff, 200
</p>
<p>derivation of, 636
</p>
<p>and general form, 3, 23ff, 40, 687
</p>
<p>and Laue groups, 29ff
</p>
<p>notation for, 15, 29
</p>
<p>recognition of, scheme for, 24ff, 636
</p>
<p>restrictions on, 22
</p>
<p>and space groups, 26, 38
</p>
<p>and special form, 24, 28, 35
</p>
<p>stereograms for, 15, 19ff, 29, 31, 177
</p>
<p>tables of, 26, 30
</p>
<p>Crystallographic software. See Software for
</p>
<p>crystallography
</p>
<p>Crystal models, 638
</p>
<p>Crystal morphology, 1ff
</p>
<p>Crystal systems
</p>
<p>and characteristic symmetry, 24, 26
</p>
<p>and crystallographic axes, 8, 28
</p>
<p>cubic, 12, 15, 28, 190
</p>
<p>hexagonal, 22, 59, 61, 94, 155, 599
</p>
<p>and lattices, 51ff
</p>
<p>and Laue groups, 29ff, 502
</p>
<p>monoclinic, 147, 152ff, 195, 288, 401, 434
</p>
<p>and optical behaviour, 190
</p>
<p>orthorhombic, 24, 60, 100, 154, 157,
</p>
<p>194ff, 201, 230, 281, 342,
</p>
<p>404, 434, 703
</p>
<p>and point group scheme, 25ff
</p>
<p>recognition of, 28
</p>
<p>and symmetry in Laue photographs, 552
</p>
<p>table of, 26
</p>
<p>tetragonal, 26, 190, 502
</p>
<p>triclinic, 26, 190, 502
</p>
<p>trigonal, 26, 190, 502
</p>
<p>CSD. See Data bases
</p>
<p>Cube, model of, 638
</p>
<p>Cubic crystal system, 28
</p>
<p>D
Data bases
</p>
<p>Cambridge crystallographic data base (CSD), 387,
</p>
<p>417, 539
</p>
<p>International Centre for Diffraction Data
</p>
<p>(ICSD), 585
</p>
<p>Protein data base, 482, 674
</p>
<p>Research Collabotory for Structural Bioinfomatics
</p>
<p>(RCSB), 418
</p>
<p>searching of, 359
</p>
<p>Data collection
</p>
<p>strategy
</p>
<p>with neutrons, 551ff, 563, 568, 576, 623
</p>
<p>with x-rays, 197ff, 219, 225, 564, 623
</p>
<p>Index 739</p>
<p/>
</div>
<div class="page"><p/>
<p>Data processing
</p>
<p>neutron, 562
</p>
<p>x-ray, 562
</p>
<p>Data/variables ratio, 406
</p>
<p>Debye&ndash;Waller expectation factor, 170
</p>
<p>Delta (δ) function, 258&ndash;259
</p>
<p>de Moivre&rsquo;s theorem, 123, 240, 250, 283
</p>
<p>Density
</p>
<p>calculated, 115, 273
</p>
<p>and contents of unit cell, 273
</p>
<p>importance of, 444, 513
</p>
<p>measurement of, 466
</p>
<p>optical, 204
</p>
<p>Detector
</p>
<p>neutron, 555, 558
</p>
<p>x-ray, 118, 222, 508
</p>
<p>Determination of absolute configuration, 326
</p>
<p>Deuteration, 554ff, 597
</p>
<p>Diad, 23, 54, 81, 97
</p>
<p>Diagonal (n) glide plane, 86, 94, 107, 150
</p>
<p>Diamond (d) glide plane, 86
</p>
<p>Difference-Fourier series, 615. See also Fourier series
</p>
<p>and correctness of structure analysis
</p>
<p>and least-squares refinement, 309, 384, 391, 393ff,
</p>
<p>419, 524, 532, 610, 649
</p>
<p>Difference-Patterson, 316
</p>
<p>anomalous, 334
</p>
<p>Diffraction. See also X-ray scattering (diffraction)
</p>
<p>by atoms, 130, 549, 561, 573
</p>
<p>by crystals, 552ff
</p>
<p>grating, 118, 253, 261
</p>
<p>by holes, 245, 249
</p>
<p>pattern, 4, 18, 33, 51, 103, 200ff (see also Atomic
</p>
<p>scattering factor)
</p>
<p>of assemblage, 17
</p>
<p>of atoms, 245
</p>
<p>of holes, 245, 249, 250
</p>
<p>by regular arrays of scattering centres, 130ff
</p>
<p>symbol, 100ff, 152ff, 178, 186, 697, 732, 733
</p>
<p>of visible light, 235, 245
</p>
<p>Diffractometer
</p>
<p>CAD4 (Nonius)
</p>
<p>data collection with, 211, 389
</p>
<p>structure determination with, 212
</p>
<p>kappa CCD (Nonius)
</p>
<p>crystal positioning, 210
</p>
<p>data collection and strategies, 394
</p>
<p>instrument geometry, 209
</p>
<p>scans, ω/θ and θ, 212
</p>
<p>optical, 249ff
</p>
<p>powder, 204, 417, 418, 585&ndash;632, 654
</p>
<p>serial, 208, 504 (see also Intensity measurement
</p>
<p>single-counter, Intensity measurement
</p>
<p>single-crystal)
</p>
<p>transformations with, 211
</p>
<p>Dihedral group, 428, 469
</p>
<p>Directions
</p>
<p>angle between, 14
</p>
<p>cosines, 7, 132 (see also Web Appendix WA1)
</p>
<p>Direct lattices, 63, 508. See also
</p>
<p>Bravais lattices
</p>
<p>Direct methods of phase determination
</p>
<p>Σ1 equation, 366, 376
</p>
<p>Σ2 equation, 355ff, 370
</p>
<p>enantiomorph selection, 367
</p>
<p>example of the use of, 377, 525
</p>
<p>experience with, 366, 372, 391
</p>
<p>figures of merit in, 384ff, 395
</p>
<p>origin specification, 354, 369
</p>
<p>Sayre&rsquo;s equation, 270
</p>
<p>starting set in, 354ff, 372
</p>
<p>structure invariant, 352ff, 372
</p>
<p>structure seminvariant, 354
</p>
<p>success with, 372
</p>
<p>symbolic addition, 359ff
</p>
<p>use of SHELX in, 372ff
</p>
<p>Direct methods of phasing, 352, 516
</p>
<p>Direct-space methods, 588, 613ff, 621
</p>
<p>Discrepancy index. See R factors
</p>
<p>Disorder
</p>
<p>dynamic, 404, 420
</p>
<p>in single crystals, 419ff
</p>
<p>static, 420
</p>
<p>Disphenoid, 4, 24
</p>
<p>Distribution
</p>
<p>acentric, 177ff, 352, 467, 645
</p>
<p>centric, 177ff, 352, 456, 645, 703
</p>
<p>cumulative, 184, 388
</p>
<p>Gaussian, 172, 176
</p>
<p>transform of, 258
</p>
<p>mean values of, 180
</p>
<p>parameter, 172, 176
</p>
<p>radial, 246, 419
</p>
<p>Duality, 551
</p>
<p>E
Electromagnetic radiation, interaction of crystal with,
</p>
<p>121, 189
</p>
<p>Electron
</p>
<p>particle properties of, 128
</p>
<p>scattering, 121ff, 140
</p>
<p>wave nature of, 128
</p>
<p>Electron density, 5, 128, 134ff, 169ff, 241ff, 278ff,
</p>
<p>355ff, 446ff, 502ff, 564ff, 612ff, 641ff. See also
</p>
<p>Difference-Fourier series
</p>
<p>ball-and-stick model for, 279
</p>
<p>computation and display of, 279
</p>
<p>contour map of, 5, 254, 420
</p>
<p>and criteria for correctness of structure
</p>
<p>analysis, 416
</p>
<p>determined from partial structures, 303
</p>
<p>equations for, 243, 278, 285, 651
</p>
<p>fitting to, 536
</p>
<p>and Fourier series, 241, 273, 416, 513
</p>
<p>and Fourier transform of, 528
</p>
<p>and hydrogen atom positions, 280, 476
</p>
<p>interpretation of, 278ff, 513, 527, 531, 614
</p>
<p>in large molecule analysis, 324
</p>
<p>740 Index</p>
<p/>
</div>
<div class="page"><p/>
<p>map, 5, 171, 266, 279, 293, 301ff, 339, 416, 459ff,
</p>
<p>513ff, 563, 607, 625, 642
</p>
<p>non-negativity of, 362
</p>
<p>and Patterson function, 282, 612, 641
</p>
<p>peak heights and peak weights in, 279, 612
</p>
<p>periodicity of, 241
</p>
<p>and phase problem, 243
</p>
<p>projections of, 242, 279ff, 446, 641
</p>
<p>superposition of peaks in, 281
</p>
<p>pseudosymmetry in, 308, 320
</p>
<p>real nature of, 128
</p>
<p>resolution of, 278, 423, 447, 513, 530
</p>
<p>standard deviation of, 416
</p>
<p>structure factors and, 244, 513, 641, 648
</p>
<p>successive Fourier refinement and, 309
</p>
<p>units of, 245
</p>
<p>E maps
</p>
<p>calculation of, 360
</p>
<p>&lsquo;sharp&rsquo; nature of, 361
</p>
<p>Epsilon (ε) factor, 177, 352, 646
</p>
<p>Equivalent positions, 74, 78, 82, 94. See also General
</p>
<p>equivalent positions; Special equivalent positions
</p>
<p>Errors, 415
</p>
<p>superposition of, 415, 720
</p>
<p>Esd. See Estimated standard deviation (esd)
</p>
<p>Estimated standard deviation (esd), 415
</p>
<p>jEj values, 175, 183, 287, 643. See also Direct methods
of phase determination calculation of
</p>
<p>distribution of, 182ff, 352, 365, 456
</p>
<p>statistics of, 644
</p>
<p>structure factors and, 182, 641
</p>
<p>Evans-Sutherland picture system, 420
</p>
<p>Even function, 255, 258, 709
</p>
<p>Ewald, P.P.
</p>
<p>construction, 138
</p>
<p>Ewald sphere, 138, 163, 173, 202ff
</p>
<p>Extinction
</p>
<p>optical
</p>
<p>for biaxial crystals, 190, 194ff
</p>
<p>for uniaxial crystals, 190
</p>
<p>x-ray
</p>
<p>parameter, 165, 168
</p>
<p>primary, 164
</p>
<p>secondary, 164ff
</p>
<p>F
FAST detector, 215
</p>
<p>Fast freezing, 496ff
</p>
<p>Fast Fourier transform (FFT), 272, 339, 513, 672
</p>
<p>Federov, E.S. See Fyodorov, Y.
</p>
<p>Figure of merit (FOM), 321ff, 377ff, 528, 604, 627
</p>
<p>File
</p>
<p>cif, 416
</p>
<p>pdb, 532, 535, 569
</p>
<p>Filtered x-radiation. See Monochromators
</p>
<p>Flack parameter, 305, 326ff, 468. See also Determination
</p>
<p>of absolute configuration; Hamilton ratio test
</p>
<p>Flash freezing (shock cooling), 501
</p>
<p>Focusing mirrors
</p>
<p>folding integral (see Convolution; Integrals)
</p>
<p>Franks, A., 225
</p>
<p>Göbel, 235
</p>
<p>FOM. See Figure of merit (FOM)
</p>
<p>Form
</p>
<p>of directions, 51
</p>
<p>of planes, 15
</p>
<p>Fourier analysis. See Fourier series
</p>
<p>Fourier, J.B., 236
</p>
<p>Fourier map
</p>
<p>difference electron density, 324
</p>
<p>electron density, 641
</p>
<p>Fo-jFcj map, 266
2Fo-jFcj map, 423, 530, 536, 537, 579
</p>
<p>Fourier series, 235ff, 281, 285, 309, 361, 416, 460, 513,
</p>
<p>648, 716, 717. See also Difference-Fourier series
</p>
<p>coefficients of, 240, 258, 266, 285, 302, 309, 361,
</p>
<p>460, 716
</p>
<p>exponential form of, 240
</p>
<p>frequency variable in, 241, 257
</p>
<p>one-dimensional, 241, 257, 346, 651
</p>
<p>partial, 301ff
</p>
<p>refinement with successive, 309, 468, 531
</p>
<p>series termination errors in, 240, 416
</p>
<p>in structure analysis, 236, 266, 273, 281
</p>
<p>summation of, 239, 344, 651
</p>
<p>three-dimensional, 241ff, 463
</p>
<p>two-dimensional, 36, 243ff, 651
</p>
<p>wavenumber variable in, 237
</p>
<p>weighting of coefficients for, 302
</p>
<p>Fourier summation tables. See BeeversûLipson strips
</p>
<p>Fourier transform
</p>
<p>of atom, 245ff
</p>
<p>and change of origin, 252
</p>
<p>conjugate of, 125, 367 (see also Friedel&rsquo;s law)
</p>
<p>of electron density, 528
</p>
<p>of euphenyl iodoacetate, 259
</p>
<p>fast, 339, 513
</p>
<p>generalized, 246ff
</p>
<p>general properties of, 261
</p>
<p>and heavy-atom technique, 266
</p>
<p>of hole, 245ff
</p>
<p>inverse of, 255ff
</p>
<p>of molecule, 248
</p>
<p>and optical diffractometer, 249
</p>
<p>and Patterson function, 273, 282ff, 372
</p>
<p>phase free (see Fourier transform; Patterson, function)
</p>
<p>of platinum phthalocyanine, 253, 266
</p>
<p>practice with, 249ff
</p>
<p>reconstruction of image by, 252ff
</p>
<p>representation of, 246, 253ff
</p>
<p>sampling, 239, 257
</p>
<p>and sign relationships, 268
</p>
<p>structure factors as, 240, 266, 309
</p>
<p>and systematic absences, 252, 270
</p>
<p>transform of, 36, 235ff, 338, 528, 627, 651
</p>
<p>of two or more holes, 250ff
</p>
<p>of unit cell, 248
</p>
<p>and weighted reciprocal lattice, 259ff
</p>
<p>Index 741</p>
<p/>
</div>
<div class="page"><p/>
<p>Four-phase structure invariants. See Quartets
</p>
<p>Fractional coordinates, 54ff, 65, 106, 248
</p>
<p>Frankenheim, M.L., 3
</p>
<p>Fraunhofer diffraction, 245
</p>
<p>Free rotation, 420
</p>
<p>Friedel pairs, 211, 305, 325, 331, 719.
</p>
<p>See also Bijvoet pairs
</p>
<p>Friedel&rsquo;s law, 140, 147, 175, 181, 200, 228, 237, 243, 330,
</p>
<p>342, 358, 374, 718
</p>
<p>and absorption edges, 325
</p>
<p>Friedrich, W., 5
</p>
<p>Fringe function, 250ff, 262
</p>
<p>Fullerenes, 38
</p>
<p>Fyodorov, Y., 3
</p>
<p>G
Gamma (Γ) function, 181. See alsoWeb Appendix WA7
</p>
<p>Gaussian function, 258
</p>
<p>General equivalent positions, 74ff, 89ff, 143ff
</p>
<p>molecules in, 106
</p>
<p>General form of planes, 30. See also Form
</p>
<p>for crystallographic point groups, 31
</p>
<p>Generator, x-ray. See X-rays, generators
</p>
<p>Geodesic dome, 38
</p>
<p>Geometric structure factor, 340ff, 358, 433
</p>
<p>Gessner, C., 1
</p>
<p>Glass (silica) structure, 7
</p>
<p>Glide line, 73, 75, 76, 80
</p>
<p>Glide plane, 86, 94, 142ff, 176, 275
</p>
<p>Goniometer
</p>
<p>contact, 1
</p>
<p>optical, 14
</p>
<p>x-ray, 441
</p>
<p>Graphic symmetry symbols, 25, 73, 87, 695. See also
</p>
<p>Stereograms
</p>
<p>change of hand, 21, 491
</p>
<p>for diad axis, 81
</p>
<p>for glide line, 73, 76
</p>
<p>for glide plane, 86
</p>
<p>for inverse monad (centre of symmetry), 25
</p>
<p>for inversion axis, 22, 685 (see also
</p>
<p>Roto-inversion axes)
</p>
<p>for mirror line, 76
</p>
<p>for mirror plane, 22ff, 200, 637, 695
</p>
<p>for pole, 21, 28
</p>
<p>for representative point, 21
</p>
<p>for rotation axes, 22, 26, 200
</p>
<p>for rotation points, 74, 102
</p>
<p>for screw axes, 83ff, 142, 176, 276
</p>
<p>table of, 25, 85
</p>
<p>Great circle, 15, 38
</p>
<p>Guglielmini, G., 1
</p>
<p>H
Half-translation rule, 93ff, 691
</p>
<p>Hamilton ratio test, 468. See also Flack parameter
</p>
<p>determination of absolute configuration, 468
</p>
<p>Harker (and non-Harker) sections of the Patterson
</p>
<p>function, 287, 310ff, 450, 468
</p>
<p>Hauptman, H., 355, 362, 435
</p>
<p>Haüy, R.J. (Abbé), 1
</p>
<p>Heavy-atom method, 301ff, 386, 439, 649. See also
</p>
<p>Anomalous dispersion; Isomorphous replacement;
</p>
<p>Patterson, search; Patterson, selection; Structure
</p>
<p>analysis
</p>
<p>examples of, 303, 308
</p>
<p>and Fourier transform, 266
</p>
<p>limitations of, 310
</p>
<p>and Patterson function, 310, 315
</p>
<p>Hermann-Mauguin notation, 27, 29, 41, 86, 159
</p>
<p>and Schönflies notation, 29, 41, 159
</p>
<p>Hessel, J.F.C., 3
</p>
<p>Hexad, 25
</p>
<p>Hexagonal crystal system, 1, 22, 26, 34, 155
</p>
<p>Hexagonal two-dimensional system, 22
</p>
<p>Hexamethylbenzene, 270, 355, 640
</p>
<p>Hex(akis)octahedron, 4, 24
</p>
<p>Hierarchy for considering limiting conditions, 154
</p>
<p>High Flux Isotope Reactor (HFIR) (ORNL,
</p>
<p>Tennessee), 556
</p>
<p>High-resolution transmission electron microscopy
</p>
<p>(HRTEM), 236
</p>
<p>Homology modelling, 481
</p>
<p>HRTEM. See High-resolution transmission electron
</p>
<p>microscopy (HRTEM)
</p>
<p>Hydrogen atom positions, 280, 340, 376, 419, 475, 549,
</p>
<p>558, 570ff
</p>
<p>Hydrogen atoms, 5, 279, 339ff, 406ff, 419, 468, 549.
</p>
<p>See also Light atoms
</p>
<p>bonds with, 408
</p>
<p>location of
</p>
<p>by calculation, 340, 406
</p>
<p>by difference-Fourier, 279
</p>
<p>Hypersymmetry, 175, 184
</p>
<p>I
I(hkl), 140, 161ff, 305
</p>
<p>ICDD. See International Centre for Diffraction
</p>
<p>Data (ICDD)
</p>
<p>Icosahedral group, 39
</p>
<p>ICSD. See Data bases; Inorganic Crystal Structure
</p>
<p>Database (ICSD)
</p>
<p>Ideal intensity, 162
</p>
<p>Identity symmetry element, 38, 81
</p>
<p>Image formation, 235&ndash;236
</p>
<p>Image plate, 209, 215ff, 504ff, 592. See also Intensity
</p>
<p>measurement
</p>
<p>Implication diagram, 318
</p>
<p>Incoherent scattering, 127ff, 554ff, 597. See also
</p>
<p>Scattering
</p>
<p>Independent reflections, 329, 394, 577
</p>
<p>Indexing reflections, 441
</p>
<p>Indistinguishability in symmetry, 25
</p>
<p>In-house data collection, 219
</p>
<p>Inorganic Crystal Structure Database
</p>
<p>(ICSD), 418
</p>
<p>Integral sin x / x. See Web Appendix WA7
</p>
<p>Integrated reflection, 161, 576
</p>
<p>742 Index</p>
<p/>
</div>
<div class="page"><p/>
<p>Intensity
</p>
<p>averages,
</p>
<p>abnormal, 176
</p>
<p>enhanced, 176, 310
</p>
<p>data, 197
</p>
<p>data collection, 197ff, 305
</p>
<p>data quality, 304
</p>
<p>distributions (see Distribution)
</p>
<p>expressions for, 161ff
</p>
<p>factors affecting, 161
</p>
<p>ideal (see Ideal Intensity)
</p>
<p>measurement of, 167, 189, 221, 282, 303, 325, 444,
</p>
<p>456, 474, 504, 599
</p>
<p>relative, 230
</p>
<p>of scattered x-rays (see X-ray scattering (diffraction)
</p>
<p>by crystals)
</p>
<p>statistics of, 161ff, 228, 359
</p>
<p>variance of, 506
</p>
<p>weighted mean, 167
</p>
<p>Intensity data, 197
</p>
<p>chi-squared test for equivalent reflections in, 168
</p>
<p>completeness of, 510
</p>
<p>merging of equivalent reflections in, 167
</p>
<p>scaling of, 167
</p>
<p>by Wilson&rsquo;s method, 169, 313, 445ff, 641, 646
</p>
<p>Intensity measurement
</p>
<p>by area detector
</p>
<p>charge-coupled type, 217
</p>
<p>FAST type, 215
</p>
<p>background radiation in, 128, 590
</p>
<p>by diffractometer
</p>
<p>powder, 593ff, 605ff
</p>
<p>serial, 208, 504
</p>
<p>single-counter, 208, 504
</p>
<p>by image plate, 215ff
</p>
<p>Interatomic distances, 385, 409, 463, 614
</p>
<p>Interference of x-rays
</p>
<p>constructive, 225
</p>
<p>destructive, 133
</p>
<p>and finite atom size, 171
</p>
<p>Intermolecular contact distances, 408, 451
</p>
<p>Intermolecular potential, 433
</p>
<p>International Centre for Diffraction Data (ICDD), 117,
</p>
<p>585, 605
</p>
<p>International Union of Crystallography (IUCr), 37,
</p>
<p>607, 711
</p>
<p>Internuclear distance, 419
</p>
<p>Interplanar spacings, 62, 132ff
</p>
<p>Inversion axes, 22, 637, 659. See also Roto-inversion axes
</p>
<p>Ionic radii, 416
</p>
<p>table of, 416
</p>
<p>Ionization spectrometer, x-ray, 5, 103
</p>
<p>Isomorphous pairs, 312
</p>
<p>Isomorphous replacement, 306, 312ff, 334, 346, 514&ndash;515,
</p>
<p>531, 716. See also Heavy-atom method
</p>
<p>for alums, 346
</p>
<p>multiple (MIR), 314, 320ff, 337, 346, 496ff, 503,
</p>
<p>513ff, 527, 531, 534, 545, 728
</p>
<p>for proteins, 514
</p>
<p>single (SIR), 314, 319ff, 335, 503, 527, 611 (see also
</p>
<p>Powder method; Seminvariant representations
</p>
<p>(SIR))
</p>
<p>single, with anomalous scattering (SIRAS and
</p>
<p>MIRAS), 335ff, 503, 527
</p>
<p>Isomorphous replacement, single, with anomalous
</p>
<p>scattering, 335ff, 503, 527
</p>
<p>Isotropic crystals. See Optically isotropic crystals
</p>
<p>Isotropic thermal vibrations. See Thermal vibrations
</p>
<p>Isotropy, 166ff, 171ff, 190ff, 287
</p>
<p>IUCr. See International Union of Crystallography (IUCr)
</p>
<p>J
Joint Committee on Powder Diffraction Standards
</p>
<p>(JCPDS). See International Centre for
</p>
<p>Diffraction Data
</p>
<p>K
Karle&ndash;Hauptman inequalities, 435
</p>
<p>Karle, J., 355, 359
</p>
<p>Kepler, J., 1
</p>
<p>Knipping, P., 4
</p>
<p>K-spectrum, 113ff
</p>
<p>L
Lack of closure error. See Closure error
</p>
<p>LADI-III (ILL, Grenoble), 555
</p>
<p>Laser wakefield acceleration, 121
</p>
<p>Lattice energy, 422ff, 430, 628. See also Computer
</p>
<p>prediction of crystal
</p>
<p>structure
</p>
<p>minimization of, 423, 628
</p>
<p>and thermodynamic stability, 433, 483
</p>
<p>Lattices. See Bravais lattices
</p>
<p>Lattice, two-dimensional, 52, 60
</p>
<p>Laue class. See Laue group
</p>
<p>Laue equations, 130
</p>
<p>Laue group
</p>
<p>and point group, 30, 101, 175, 200, 330, 502
</p>
<p>projection symmetry of, 29, 34, 200
</p>
<p>Laue method. See also Laue x-ray image/photograph
</p>
<p>experimental arrangement for, 199, 202
</p>
<p>and synchrotron radiation, 200ff
</p>
<p>Laue projection symmetry, 30, 34, 200
</p>
<p>Laue symmetry, 175, 180, 225, 288, 509, 545,
</p>
<p>562, 704, 726
</p>
<p>Laue treatment of x-ray diffraction, 76, 130ff
</p>
<p>equivalence with Bragg treatment, 134
</p>
<p>Laue x-ray image/photograph. See also Laue method
</p>
<p>symmetry of, 200
</p>
<p>and uniaxial crystals, 190, 200
</p>
<p>Layer lines, 230. See also Oscillation method
</p>
<p>Least squares refinement
</p>
<p>data errors, 406
</p>
<p>data/variables ratio, 406
</p>
<p>and estimated standard deviation, 406
</p>
<p>and light atoms, 419
</p>
<p>and parameter refinement, 401ff
</p>
<p>precision in, 405
</p>
<p>Index 743</p>
<p/>
</div>
<div class="page"><p/>
<p>Least squares refinement (cont.)
</p>
<p>refinement against jFoj and jFoj2, 410
refinement strategy, 405ff
</p>
<p>rigid-body constraint in, 534
</p>
<p>scale factor in, 406, 444
</p>
<p>and secondary extinction in, 137
</p>
<p>special positions in, 405
</p>
<p>strategy in, 405
</p>
<p>temperature factors in, 402ff
</p>
<p>unit-cell dimensions, 401
</p>
<p>and weights, 406
</p>
<p>Light atoms, 301, 310, 386, 419, 467, 549. See also
</p>
<p>Hydrogen atom positions
</p>
<p>Limiting conditions. See also Geometric structure factor;
</p>
<p>Space groups unit cell, centred; Translational
</p>
<p>symmetry
</p>
<p>hierarchal order of considering, 150
</p>
<p>non-independent (redundant), 84, 91
</p>
<p>redundant (see Non-independent limiting
</p>
<p>conditions)
</p>
<p>in space group P21, 144
</p>
<p>in space group Pc, 146
</p>
<p>in space group P21/c, 146
</p>
<p>in space group Pma2, 148
</p>
<p>and systematic absences, 142
</p>
<p>for translational symmetry, 142
</p>
<p>for unit-cell types, 138
</p>
<p>Limiting sphere, 281. See also Ewald sphere;
</p>
<p>Sphere of reflection
</p>
<p>Line, equation of, 13, 135
</p>
<p>intercept form of, 8
</p>
<p>Liquid nitrogen shock cooling, 219, 394. See also
</p>
<p>Fast freezing
</p>
<p>Liquids, x-ray diffraction from, 273, 419
</p>
<p>Long-range order, 7
</p>
<p>Lonsdale, K., 91ff, 355
</p>
<p>Lorentz factor, 163
</p>
<p>Low energy neutrons, 573
</p>
<p>Low temperature measurements, 208
</p>
<p>M
Macromolecular structure analysis
</p>
<p>flow diagram for, 490
</p>
<p>free-R factor (see also Macromolecular structure
</p>
<p>analysis; Protein structure analysis; Rfree)
</p>
<p>heavy-atoms derivatives for, 306
</p>
<p>multiple isomorphous replace (MIR) in structure
</p>
<p>factors in, 501
</p>
<p>temperature factors in, 501
</p>
<p>protein, crystallization of
</p>
<p>&lsquo;click&rsquo; test in, 494
</p>
<p>improvement of crystals, 496
</p>
<p>precipitants, 493
</p>
<p>quality screening, 495
</p>
<p>protein, make-up
</p>
<p>L-amino acids, 499
</p>
<p>amino acid sequence, 497
</p>
<p>α-carbon atom, 489, 499
</p>
<p>polypeptide chain, 489
</p>
<p>primary structure, 489
</p>
<p>secondary structure, 490
</p>
<p>tertiary structure, 490
</p>
<p>protein, properties, 491
</p>
<p>protein, purity, 492
</p>
<p>protein, structure analysis
</p>
<p>cryo-crystallography, 499
</p>
<p>crystal mounting for, 498
</p>
<p>crystal selection, 501
</p>
<p>data collection
</p>
<p>with area detector, 197
</p>
<p>by camera, 197
</p>
<p>by diffractometer, 197, 209
</p>
<p>with image plate, 209, 500
</p>
<p>problems in and possible cures, 508
</p>
<p>radiation sources in, 593
</p>
<p>example structure determination, Ricin
</p>
<p>Agglutinin (RCA)
</p>
<p>AmoRe algorithm, 525
</p>
<p>density, 510
</p>
<p>difference electron density, 512
</p>
<p>electron density in, 512
</p>
<p>errors in, 510
</p>
<p>2Fo-jFcj map for, 514
geometry of molecule, 387
</p>
<p>initial MR model, 527
</p>
<p>number of reflections in data set, 281
</p>
<p>omit map for, 514
</p>
<p>Patterson function for, 282ff
</p>
<p>phase information and density modification
</p>
<p>in programs for, 527ff
</p>
<p>phasing by MAD in, 527
</p>
<p>phasing by single isomorphous
</p>
<p>techniques in, 527
</p>
<p>radius of integration, 525
</p>
<p>resolution, of data, 304
</p>
<p>R factor, 511
</p>
<p>Rfree, 503
</p>
<p>self-rotation function for, 509, 520
</p>
<p>solvent analysis, 536
</p>
<p>space group for, 509
</p>
<p>structure validation, 537
</p>
<p>unit cell of, 509
</p>
<p>heavy-atoms derivatives for, 497,
</p>
<p>514, 527
</p>
<p>intensity data for, 506
</p>
<p>Laue symmetry, 509
</p>
<p>multiple isomorphous replace (MIR) in, 496,
</p>
<p>514, 527
</p>
<p>multiple isomorphous replace (MIR) in:
</p>
<p>structure factors in, 527
</p>
<p>NCS (see Non-crystallographic symmetry)
</p>
<p>non-crystallographic symmetry, 518ff, 523
</p>
<p>post-refinement, 506ff
</p>
<p>profile fitting, 506
</p>
<p>temperature factors in, 501
</p>
<p>unit cell determination, 509
</p>
<p>MAD. See Multiple wavelength anomalous
</p>
<p>dispersion (MAD)
</p>
<p>744 Index</p>
<p/>
</div>
<div class="page"><p/>
<p>Magnetic
</p>
<p>field effect, 119, 121
</p>
<p>materials, 128
</p>
<p>moment, 549
</p>
<p>scattering
</p>
<p>amplitude, 128
</p>
<p>form factor, 128
</p>
<p>of neutrons, 553
</p>
<p>Matrix
</p>
<p>inverse of, 405, 705
</p>
<p>multiplication of, 65
</p>
<p>notation, 65
</p>
<p>representation of symmetry operations, 97ff
</p>
<p>transformation, 69, 108, 230, 602, 654, 689, 696, 731
</p>
<p>transpose of, 651, 705
</p>
<p>Maximum entropy, 609, 624
</p>
<p>Maximum likelihood, 516, 535
</p>
<p>Mean planes, 414
</p>
<p>Miller&ndash;Bravais indices, 11, 22
</p>
<p>Miller indices
</p>
<p>common factor in, 602
</p>
<p>in stereograms, 70
</p>
<p>transformations of, 69
</p>
<p>Miller, W., 3, 9
</p>
<p>Minimum function, 293, 641. See also Patterson,
</p>
<p>superposition
</p>
<p>MIR. See Multiple isomorphous replacement
</p>
<p>MIRAS. See Multiple isomorphous replacement with
</p>
<p>anomalous scattering
</p>
<p>Mirror plane, 18, 22ff, 86, 132, 176, 200, 288, 412, 501,
</p>
<p>602, 637, 714
</p>
<p>Mirror symmetry. See Reflection symmetry
</p>
<p>Model
</p>
<p>bias, 516, 527, 538
</p>
<p>building, 513, 523ff, 613
</p>
<p>structure, 4, 278, 306, 325, 380, 400, 416ff, 513, 531,
</p>
<p>533ff, 588, 628, 727
</p>
<p>Modulus, 262, 301, 586
</p>
<p>Mohs, F., 3
</p>
<p>Molecular
</p>
<p>geometry, 327, 336, 408ff, 433, 451ff, 471, 513, 529ff,
</p>
<p>628, 652
</p>
<p>graphics, 385, 388, 434, 516, 527, 531, 541, 554, 720
</p>
<p>programs for, 385
</p>
<p>Molecular replacement
</p>
<p>correct solution, recognition of, 523
</p>
<p>phases from, 524
</p>
<p>programs for, 518
</p>
<p>rigid-body refinement in, 524
</p>
<p>rotation function in, 523
</p>
<p>search model in, 522
</p>
<p>data bases, use of, 522
</p>
<p>search Patterson function for, 522
</p>
<p>subunits in, 524
</p>
<p>target Patterson function for, 522
</p>
<p>translation function, 522
</p>
<p>Monad, 25
</p>
<p>Monochromators
</p>
<p>double-type, 224
</p>
<p>filter type, 224
</p>
<p>single-type, 224
</p>
<p>for synchrotron radiation, 225
</p>
<p>Monoclinic crystal system, 195
</p>
<p>Morphology, 1&ndash;49ff, 135, 190, 226, 508
</p>
<p>Mosaic spread, 220
</p>
<p>Motif, 18, 72ff, 83, 105, 145, 425, 689
</p>
<p>MR. See Molecular replacement (MR)
</p>
<p>Multiple isomorphous replacement (MIR), 314, 320,
</p>
<p>496, 517
</p>
<p>Multiple isomorphous replacement with anomalous
</p>
<p>scattering (MIRAS), 527
</p>
<p>Multiple wavelength anomalous dispersion (MAD), 337,
</p>
<p>503, 527. See also Protein (macromolecule),
</p>
<p>structure determination of
</p>
<p>Multiplicity of reflection data, 511
</p>
<p>Multisolution procedure, 372
</p>
<p>Multiwire proportional counter (MWPC), 213
</p>
<p>N
Naphthalene, 265, 275
</p>
<p>symmetry analysis of, 275
</p>
<p>Net, 52, 73, 106, 121, 132, 164, 202, 236, 251, 551, 614.
</p>
<p>See also Lattice
</p>
<p>Neutron
</p>
<p>crystallography, 550, 560
</p>
<p>density map, 570, 576
</p>
<p>detectors, 471, 557
</p>
<p>diffraction
</p>
<p>complementary to x-ray diffraction, 549, 553
</p>
<p>data collection by, 552
</p>
<p>Laue method in, 552
</p>
<p>location of light atoms by, 549
</p>
<p>with monochromatic radiation, 560
</p>
<p>refinement of light atoms by, 550, 564, 574
</p>
<p>structure determination examples by, 560, 574
</p>
<p>time-resolved Laue method in, 560
</p>
<p>scattering, 103, 419, 550ff, 560, 565
</p>
<p>scattering length, 549, 553ff, 568, 597
</p>
<p>sources of, 551, 559
</p>
<p>spallation source, 551, 558
</p>
<p>spectrum, 555
</p>
<p>thermal, 553
</p>
<p>Nickel tungstate, 276
</p>
<p>Niggli cell, 602
</p>
<p>Non-crystallographic point groups, 31, 637, 640
</p>
<p>Non-crystallographic structure. See Aperiodic structure
</p>
<p>Non-crystallographic symmetry, 491, 518ff
</p>
<p>Non-independent limiting conditions, 84, 91
</p>
<p>Normal distribution, 172, 177, 179, 180
</p>
<p>Normalized structure factor, 172, 182ff, 351ff
</p>
<p>Notation, xxxiii, 15, 21, 23ff, 54ff, 74, 85ff,
</p>
<p>134, 362, 663
</p>
<p>Nucleic acids, 471, 496, 521
</p>
<p>Numerical data, 64, 132, 339, 534, 555
</p>
<p>O
Oblique axes, 40
</p>
<p>Oblique extinction, 196, 289, 440
</p>
<p>Index 745</p>
<p/>
</div>
<div class="page"><p/>
<p>Oblique two-dimensional system, 21, 52, 73. See also
</p>
<p>Plane groups
</p>
<p>Occupancy, 80, 570, 656
</p>
<p>Oligomycins A, B, C
</p>
<p>absolute configuration of, 472ff
</p>
<p>crystal structure of, 474
</p>
<p>docking of, 471, 481
</p>
<p>hydrogen bonds in, 473, 477, 484
</p>
<p>Omit map, 514, 577, 580
</p>
<p>Omitted coordinates, 375, 577
</p>
<p>Optical classification of crystals, 190
</p>
<p>Optical diffraction pattern, 251
</p>
<p>Optical diffractometer, 249
</p>
<p>Optically anisotropic crystals, 192
</p>
<p>Optically isotropic crystals, 192
</p>
<p>Optical methods of crystal examination, 190, 466
</p>
<p>Optic axis
</p>
<p>and biaxial crystals, 194
</p>
<p>and crystallographic axes, 192
</p>
<p>and uniaxial crystals, 193
</p>
<p>Order of diffraction, 131
</p>
<p>Orientation of crystals, 226, 384, 574
</p>
<p>Origin
</p>
<p>change of, 90, 252, 353, 692
</p>
<p>choice of, 76, 87, 93, 106, 253, 354, 372, 657
</p>
<p>Origin-fixing reflections, 352ff, 359, 458, 627
</p>
<p>Orthogonal axes, 7, 151, 519, 618
</p>
<p>Orthogonal function, 9, 137, 151, 215, 382, 525. See also
</p>
<p>Web Appendix WA8
</p>
<p>Orthorhombic crystal system, 342, 426, 577. See also
</p>
<p>Crystal systems
</p>
<p>Oscillation method, 205ff
</p>
<p>Oscillation record (image/photograph), 197, 206, 221,
</p>
<p>229, 441, 506. See also Layer lines
</p>
<p>experimental arrangement for, 199, 202
</p>
<p>flat-plate technique in, 199, 205
</p>
<p>and protein crystal, 207
</p>
<p>symmetry indications from, 305, 441
</p>
<p>Over-determination, 282, 533, 538
</p>
<p>P
Packing, 3, 295, 298ff, 385ff, 408, 428ff, 493,
</p>
<p>628, 673
</p>
<p>Parametral line, 39
</p>
<p>Parametral plane, 8, 26
</p>
<p>Parity, 138, 144, 353, 354, 367ff, 459, 644, 649, 719
</p>
<p>Parity group, 354, 369, 459, 644, 649, 719
</p>
<p>Partial-structure phasing. See also Phase, determination
</p>
<p>effective power of, 301
</p>
<p>for proteins, 400, 539
</p>
<p>for small molecules, 538
</p>
<p>Path difference
</p>
<p>analysis of, 135
</p>
<p>in Bragg reflection, 164, 201, 552
</p>
<p>PATSEE Patterson search program
</p>
<p>crystal packing in, 385
</p>
<p>detailed examples of use of, 28, 418
</p>
<p>example structure analysis, 388
</p>
<p>expansion and refinement in, 386, 391, 397
</p>
<p>interatomic vectors in, 267, 381ff
</p>
<p>molecular orientation in, 382
</p>
<p>Patterson function, storage of in, 387
</p>
<p>refinement with, 384
</p>
<p>rotation search, 387, 395
</p>
<p>strategy for, 388
</p>
<p>search model construction for, 384ff
</p>
<p>scattering power of, 389ff
</p>
<p>translation search, 387
</p>
<p>vector verification, comparison with, 381, 388
</p>
<p>Pattern motif, 51, 72
</p>
<p>and asymmetric unit, 72
</p>
<p>Patterson
</p>
<p>function (see also Harker sections; Peaks of
</p>
<p>Patterson function)
</p>
<p>centrosymmetry of, 284
</p>
<p>convolution and, 266, 271
</p>
<p>electron density product in, 283
</p>
<p>Fourier series of, 235ff
</p>
<p>Fourier transform of, 235ff
</p>
<p>heavy atom in, 310
</p>
<p>Laue symmetry of, 288
</p>
<p>as map of interatomic vectors, 284
</p>
<p>non-origin peaks in, 286
</p>
<p>one-dimensional, 241
</p>
<p>origin peak in, 269, 285, 291, 296, 299, 352, 435,
</p>
<p>521, 716, 725
</p>
<p>over-sharpening of, 287
</p>
<p>packing analysis in, 298
</p>
<p>partial electron density from, 463
</p>
<p>peaks in, 285, 463
</p>
<p>practical evaluation of, 283
</p>
<p>in projection, 279, 446
</p>
<p>resolution of, 287, 381, 387
</p>
<p>search methods (see Patterson, search)
</p>
<p>sharpened, 287, 352
</p>
<p>solution of phase problem and, 289ff
</p>
<p>successive Fourier refinement and, 257
</p>
<p>symmetry analysis in, 288
</p>
<p>space group Pm, 288
</p>
<p>and symmetry-related and symmetry-
</p>
<p>independent atoms, 286
</p>
<p>three-dimensional, 286
</p>
<p>and vector interactions, 289
</p>
<p>search
</p>
<p>crystal packing in, 285
</p>
<p>deconvolution by, 286
</p>
<p>expansion of structure from, 286
</p>
<p>figures of merit in, 385, 395
</p>
<p>general procedure with, 381
</p>
<p>overlap in, 523
</p>
<p>random angle triplets in, 384
</p>
<p>refinement of structure from, 531
</p>
<p>rotation stage in, 382, 397, 523
</p>
<p>search structure, 381
</p>
<p>for small molecules, 375, 382ff, 522
</p>
<p>target structure for, 382ff
</p>
<p>translation stage in, 384, 395, 523
</p>
<p>vectors in, 381ff
</p>
<p>746 Index</p>
<p/>
</div>
<div class="page"><p/>
<p>sections
</p>
<p>for papaverine hydrochloride, 297
</p>
<p>selection, 310ff
</p>
<p>space, 285, 286, 288, 292, 447
</p>
<p>space group, 286
</p>
<p>superposition, 293 (see also Minimum function)
</p>
<p>Peaks. See also Patterson, function
</p>
<p>arbitrariness in location of, 296
</p>
<p>cross-vector, 384
</p>
<p>electron density, in, 284
</p>
<p>in Harker lines and sections, 289, 448
</p>
<p>heights and weights
</p>
<p>in electron density, 278, 612
</p>
<p>in Patterson function, 285
</p>
<p>implication diagram for, 318
</p>
<p>non-origin, 286, 317, 380, 386
</p>
<p>Patterson, 285, 296, 382, 524
</p>
<p>positions of, 285
</p>
<p>spurious, 287, 361, 371, 460
</p>
<p>and symmetry-related atoms, 288,
</p>
<p>290, 302
</p>
<p>weights of, 285
</p>
<p>Peaks of Patterson function, 285, 296
</p>
<p>Penrose tiling, 36
</p>
<p>Perdeuteration, 559
</p>
<p>Periodicity, 35, 236, 241
</p>
<p>Periodic table, 256
</p>
<p>Phase. See also Partial-structure phasing; Structure
</p>
<p>analysis; Structure factor
</p>
<p>angle, 125, 135, 141, 156, 159, 172, 216, 253, 301,
</p>
<p>308, 323, 330, 361, 363, 367, 502, 703, 717
</p>
<p>annealing, 376ff
</p>
<p>by anomalous scattering, 140, 305, 325ff, 527
</p>
<p>best, 323, 351, 377
</p>
<p>in centrosymmetric crystals, 141, 270, 302, 353, 361
</p>
<p>change, 145, 164, 248, 325, 624, 722
</p>
<p>combined, 130, 336, 627
</p>
<p>determination, 351ff, 516, 611
</p>
<p>difference, 122, 164, 251
</p>
<p>direct methods of determining, 351ff
</p>
<p>error in, 302, 321
</p>
<p>extension, 376, 528, 588
</p>
<p>heavy-atom method of determining, 301
</p>
<p>importance of correct, 126
</p>
<p>in non-centrosymmetric crystals, 141, 302, 334, 361ff
</p>
<p>from Patterson function, 282ff, 641
</p>
<p>power, 123&ndash;108, 135, 244, 301, 306
</p>
<p>probability methods, 351 (see also Direct methods)
</p>
<p>problem, 243, 253, 281ff, 310, 335, 351, 527, 588,
</p>
<p>608, 626
</p>
<p>in space group P1, 179, 270, 289, 352, 358, 433,
</p>
<p>622, 650
</p>
<p>in space group P21, 86, 106, 146, 184, 259, 296, 330,
</p>
<p>367, 377, 432, 466, 524, 564, 611, 647, 710, 721
</p>
<p>of structure factor, 351 (see also Structure factor)
</p>
<p>symmetry, 358, 433, 457
</p>
<p>variance of, 179, 363, 535
</p>
<p>of wave, 124, 363
</p>
<p>of resultant wave, 127, 135
</p>
<p>Phase symmetry, 358, 433, 457
</p>
<p>Photograph. See X-rays, photograph
</p>
<p>Photon, 119, 128, 155, 214, 221, 597, 698
</p>
<p>PILATUS detector. See Complementary metal-oxide
</p>
<p>semiconductor (CMOS)
</p>
<p>Plane groups, 60, 73ff, 102ff, 394, 446, 644ff, 689, 693.
</p>
<p>See also Projections; Two-dimensional system
</p>
<p>the 17 patterns of, 76
</p>
<p>by symbol
</p>
<p>cm, 75, 78, 81, 83
</p>
<p>c2mm, 76
</p>
<p>p2, 73, 446, 651, 693, 721
</p>
<p>pg, 76, 79
</p>
<p>p2gg, 76, 78, 87, 90, 644, 649, 721
</p>
<p>pm, 75, 80, 100, 153, 176, 186, 286, 288
</p>
<p>Planes
</p>
<p>equation of, 3, 7, 451
</p>
<p>intercept form of, 8
</p>
<p>family of, 62, 125, 132, 140, 164, 697
</p>
<p>form of, 15
</p>
<p>indices (Miller) for, 8ff, 37ff, 56, 62ff, 78, 81, 107,
</p>
<p>133, 211, 359, 654, 687, 731
</p>
<p>mirror, 18, 22, 25, 28, 87, 99, 132, 176, 200, 288, 412,
</p>
<p>413, 501, 602, 637, 638, 714
</p>
<p>multiplicity of, 173, 511 (see also Epsilon
</p>
<p>(å) factor)
</p>
<p>spacing of, 12
</p>
<p>Platinum derivative of ribonuclease, 317
</p>
<p>Point atom, 287, 351, 460, 646
</p>
<p>Point groups. See also Crystallographic point groups;
</p>
<p>Non-crystallographic point groups; Stereograms
</p>
<p>centrosymmetric, 25, 175
</p>
<p>and crystal systems, 24
</p>
<p>derivation of, 636
</p>
<p>examples of (crystals and molecules), 637ff
</p>
<p>incompatibility with translation, 97
</p>
<p>key to study of, 26
</p>
<p>and Laue groups, 29, 229, 444
</p>
<p>matrix representation of, 97ff
</p>
<p>non-crystallographic, 31, 35, 184, 640
</p>
<p>notation for, 29
</p>
<p>one-dimensional, 19
</p>
<p>point group mm2, 27
</p>
<p>point group 4mm, 27, 94, 98, 695
</p>
<p>projection symmetry of, 30, 42
</p>
<p>recognition scheme for, 637ff
</p>
<p>and space groups, 97, 100
</p>
<p>three-dimensional
</p>
<p>Hermann-Mauguin symbols for, 27, 29, 159
</p>
<p>Schönflies symbols for, 29, 31
</p>
<p>two-dimensional, 19, 30, 83
</p>
<p>Pointless program, 567
</p>
<p>Polarization correction, 158, 212, 475
</p>
<p>Polarized light, 189, 228
</p>
<p>and structure analysis, 189
</p>
<p>Polarizing microscope
</p>
<p>analyser, 190
</p>
<p>examples of use of, 190
</p>
<p>polarizer, 189
</p>
<p>Index 747</p>
<p/>
</div>
<div class="page"><p/>
<p>Polarizing microscope (cont.)
</p>
<p>Polaroid, 189
</p>
<p>Pole. See Stereogram
</p>
<p>Polypeptide, 313, 319, 327, 394, 489ff, 516, 528, 540, 579
</p>
<p>Position-sensitive detector. See Area detector
</p>
<p>by computer, 618
</p>
<p>CRYSFIRE program system in, 603
</p>
<p>figures of merit in, 395, 459
</p>
<p>general indexing, 599
</p>
<p>in high-symmetry systems, 268
</p>
<p>ITO12 program system in, 603
</p>
<p>by Ito&rsquo;s method, 599
</p>
<p>of magnesium tungstate, 599
</p>
<p>Powder method
</p>
<p>basis of, 588
</p>
<p>Bragg-Brentano geometry in, 593, 614
</p>
<p>centroid maps in, 627
</p>
<p>conventional unit cells in, 601
</p>
<p>crystallite size in, 585
</p>
<p>data collection in, 590
</p>
<p>overlap of lines in, 590
</p>
<p>data indexing (see Powder indexing)
</p>
<p>difference-Fourier method in, 619
</p>
<p>model building with, 613
</p>
<p>zeolites as examples of
</p>
<p>FOCUS algorithm for, 614
</p>
<p>with Fourier recycling, 614
</p>
<p>topology studies in, 614
</p>
<p>zinc-silicate complex VIP-9, structure of, 614
</p>
<p>energy minimization in, 628
</p>
<p>expansivity by, 586
</p>
<p>genetic algorithms in, 588, 627
</p>
<p>geometry of, 590, 606
</p>
<p>and Guinier-type cameras, 590
</p>
<p>image plate in, 592
</p>
<p>identification by, 585
</p>
<p>image plate camera, 592
</p>
<p>log-likelihood gain in, 626
</p>
<p>maximum entropy in, 624
</p>
<p>and Monte Carlo technique of structure solving
</p>
<p>acceptance criteria in, 606
</p>
<p>Markov chain in, 618
</p>
<p>Metropolis algorithm in, 618
</p>
<p>and Niggli cell, 602
</p>
<p>with simulated annealing, 621
</p>
<p>starting model for, 622
</p>
<p>MYTHEN detector in
</p>
<p>neutron source for, 597
</p>
<p>diffractometry with, 597
</p>
<p>peak modelling in, 596
</p>
<p>phase transitions by, 586
</p>
<p>and protein structures, 624
</p>
<p>reduced unit cells, 602, 629, 654 (see also Niggli cell)
</p>
<p>refinement, 588, 593ff, 605
</p>
<p>R factors in, 493
</p>
<p>Rietveld refinement in, 593, 605ff, 614, 618ff
</p>
<p>by Patterson method, 609
</p>
<p>seminvariant (see Structure seminvariant)
</p>
<p>simulated annealing in, 531ff, 609, 621
</p>
<p>SIR program system for, 611
</p>
<p>specimen preparation for
</p>
<p>mounting of, 589
</p>
<p>preferred orientation in, 594, 612, 622
</p>
<p>strain broadening by, 585
</p>
<p>time-of-flight studies in, 597
</p>
<p>time-resolved studies in, 597
</p>
<p>unit-cell parameters by, 509, 510, 515, 586
</p>
<p>unit-cell reduction (LEPAGE), 654
</p>
<p>use of synchrotron radiation in, 594
</p>
<p>and zinc-insulin T3R3 complex, 623
</p>
<p>Powder pattern
</p>
<p>integrated intensities, extraction of
</p>
<p>by Le Bail method, 605
</p>
<p>overlap problem in, 552
</p>
<p>by Pawley method, 608ff
</p>
<p>Rietveld whole-profile refinement, 609
</p>
<p>x-ray structure determination, scheme of
</p>
<p>operation, 585
</p>
<p>Precession method, 36, 197, 233, 330, 506
</p>
<p>unit-cell dimensions from, 401, 516
</p>
<p>Precision, 308, 405, 415. See also Errors
</p>
<p>Precision of calculations, 415
</p>
<p>Primary extinction. See Extinction
</p>
<p>Primitive, 15, 52, 54
</p>
<p>circle, 15, 89
</p>
<p>plane, 15
</p>
<p>unit cell, 52, 54
</p>
<p>Principal symmetry axis, 26, 98
</p>
<p>Probability of triple product sign relationship, 355ff
</p>
<p>Programs. See Software for crystallography, Web
</p>
<p>program suite
</p>
<p>Projections. See also Space groups; Stereograms
</p>
<p>centrosymmetric, 313ff, 487
</p>
<p>electron density, 242, 280
</p>
<p>Patterson, 299, 300, 310, 316, 649
</p>
<p>spherical, 15
</p>
<p>stereographic, 15ff, 50, 135
</p>
<p>Proportional counter, 213, 594
</p>
<p>Protein alphabet, 489
</p>
<p>Protein Data Bank (PDB), 380, 417, 481&ndash;483, 514, 531ff,
</p>
<p>569ff, 582
</p>
<p>Protein structure analysis, 422, 514, 528, 539ff
</p>
<p>Protein (macromolecule), structure determination of
</p>
<p>by anomalous scattering, 503
</p>
<p>by co-crystallization, 496, 515
</p>
<p>electron density, properties of, 513
</p>
<p>2Fo-jFcj map, 423, 514, 530ff, 579
heavy-atom location in, 523
</p>
<p>by heavy-atom method, 386, 649
</p>
<p>by molecular replacement (MR), 381ff, 516ff
</p>
<p>by multiple isomorphous replacement (MIR), 314,
</p>
<p>320, 496, 517
</p>
<p>MIR model, 516
</p>
<p>by multiple wavelength anomalous dispersion
</p>
<p>(MAD), 503, 527, 545, 728
</p>
<p>and non-crystallographic symmetry, 518ff, 523
</p>
<p>phases, determination of, 514ff
</p>
<p>post refinement of, 506
</p>
<p>748 Index</p>
<p/>
</div>
<div class="page"><p/>
<p>and powder methods
</p>
<p>example structure determination, zinc-insulin
</p>
<p>T3R3 complex, 623
</p>
<p>structure analysis of, 586
</p>
<p>use of synchrotron radiation in, 594, 726
</p>
<p>preparation of derivatives for, 515
</p>
<p>sign determination for centric reflections of, 315
</p>
<p>structure analysis of (see Macromolecular structure
</p>
<p>analysis)
</p>
<p>Pseudosymmetry, 186, 226, 308, 320, 412, 468, 566
</p>
<p>in trial structure, 309
</p>
<p>Q
Quantum theory, 111, 130, 161, 482, 555, 596
</p>
<p>Quartets, 372ff, 399
</p>
<p>Quartz, crystal structure of, 2, 7, 225, 441, 491, 568, 590
</p>
<p>Quasicrystals, 32ff
</p>
<p>R
Radial distribution function, 246
</p>
<p>Radiation damage, 167, 220, 598
</p>
<p>Radi
</p>
<p>ionic, 416
</p>
<p>van der Waals, 645
</p>
<p>Ramachandran plot, 531, 535, 540
</p>
<p>Rational intercepts (indices), law of, 3, 8ff
</p>
<p>Rayleigh formula, 235
</p>
<p>Real space, 63, 68, 135ff, 230, 254ff, 292, 355, 601
</p>
<p>Reciprocal lattice. See also Lattice; Unit cell.
</p>
<p>analytical treatment of, 135ff. See also Web
</p>
<p>Appendix WA6
</p>
<p>diffraction pattern as weighted, 148
</p>
<p>geometrical treatment of, 63
</p>
<p>points of, in limiting sphere, 281
</p>
<p>properties of, 137
</p>
<p>and reflection conditions, 138
</p>
<p>rows in, 64, 178, 443
</p>
<p>and sphere of reflection, 138 (see also Ewald sphere)
</p>
<p>statistics of, 175
</p>
<p>symmetry of, 70, 197, 719
</p>
<p>unit cell in, 64, 135, 261, 265
</p>
<p>volume of, 281
</p>
<p>unit cell-size in, 312
</p>
<p>units of, 64, 260, 281
</p>
<p>vector treatment of, 135
</p>
<p>weighted, 148, 172, 197, 252, 259ff, 270, 330,
</p>
<p>590, 703
</p>
<p>Recombinant protein expression, 522. See also Selenium-
</p>
<p>mutated methionine
</p>
<p>Rectangular axes. See Axes
</p>
<p>Rectangular two-dimensional system, 693
</p>
<p>two-dimensional (plane) space groups, 73, 90
</p>
<p>Reduced structure factor equation. See Geometric
</p>
<p>structure factor
</p>
<p>Reduced unit cell. See Unit cell, reduction of
</p>
<p>Redundant (non-independent) limiting conditions, 150
</p>
<p>Reference axes. See Axes
</p>
<p>Refinement. See Least squares refinement
</p>
<p>Reflecting power, 161
</p>
<p>Reflection (mirror) line, 22
</p>
<p>Reflection (mirror) plane, 22
</p>
<p>Reflections. See also Limiting conditions; Signs of
</p>
<p>reflections; Structure analysis; X-ray scattering
</p>
<p>number of in data set, 281
</p>
<p>origin-fixing, 352ff, 359, 433, 458, 627
</p>
<p>unobserved, 142
</p>
<p>local average intensity for, 287
</p>
<p>Reflection, sphere of. See Ewald sphere
</p>
<p>Reflection symmetry. See also Symmetry
</p>
<p>of square-wave function, 238
</p>
<p>in three dimensions, 24
</p>
<p>in two dimensions, 21
</p>
<p>Reflection, x-ray
</p>
<p>integrated, 161, 576
</p>
<p>intensity, theory of, 219, 221, 347, 389, 474,
</p>
<p>491, 504
</p>
<p>phase change on, 145, 164, 248
</p>
<p>Refractive index, 190ff, 236, 273, 440, 466
</p>
<p>Reliability (R) factors, 303ff. See also R factors
</p>
<p>and correctness of structure analysis, 416&ndash;418
</p>
<p>and parameter refinement, 642
</p>
<p>Repeat period of a function, 103, 237, 241, 283
</p>
<p>Repeat vector, 72
</p>
<p>Repetition, 51, 72, 76, 93
</p>
<p>Replaceable site, 334
</p>
<p>Resolution, 119, 192, 202, 235ff, 304, 324, 335ff. See also
</p>
<p>Reflections, number of in data set
</p>
<p>optical, 236
</p>
<p>Rayleigh formula for, 235
</p>
<p>by x-rays, 522
</p>
<p>Resonance level, 114ff
</p>
<p>Resultant phase, 376
</p>
<p>Resultant wave, 127, 135
</p>
<p>Reticular density, 12
</p>
<p>R factors, 303ff, 339, 416, 503, 537, 607. See also
</p>
<p>Reliability factors
</p>
<p>Ranom, 306
</p>
<p>Rderiv, 306
</p>
<p>Rdiff, 306
</p>
<p>Rfree, 307, 503, 523, 538, 576
</p>
<p>Rint, 168, 220, 304, 328, 389, 416, 505, 511, 562, 577
</p>
<p>Riso, 306
</p>
<p>Rλ,
</p>
<p>Rmeas, 577
</p>
<p>Rmerge, 168, 304, 577
</p>
<p>Rfree, 307, 503, 523, 538, 576
</p>
<p>Rhombic dodecahedron, 3, 15
</p>
<p>Rhombohedral symmetry,
</p>
<p>Rhombohedral unit cell, 27, 61, 108, 696
</p>
<p>Ribonuclease, 315ff, 422, 493, 530, 537
</p>
<p>platinum derivative of, 317
</p>
<p>Richard, B.F., 38
</p>
<p>Rietveld refinement, 593, 605ff. See also Powder method
</p>
<p>chi-squared test in, 168
</p>
<p>false minima in, 607
</p>
<p>profile functions in
</p>
<p>Gaussian, 606
</p>
<p>pseudo-Voigt, 610
</p>
<p>Index 749</p>
<p/>
</div>
<div class="page"><p/>
<p>Rietveld refinement (cont.)
</p>
<p>R factors in
</p>
<p>Bragg, 607
</p>
<p>conventional, 306, 503, 533, 607
</p>
<p>profile, 607
</p>
<p>statistical expectation, 172
</p>
<p>weighted-profile, 607
</p>
<p>Right-handed coordinate system, 210
</p>
<p>Rigid-body
</p>
<p>motion,
</p>
<p>refinement, 524ff, 576
</p>
<p>Ring conformation, 412, 478
</p>
<p>Romé de l&rsquo;Isle, J.-B., 1
</p>
<p>Root-mean square amplitude of vibration, 185
</p>
<p>Rotating crystal measurement, 163
</p>
<p>Rotational symmetry, 21ff, 32ff, 52, 71ff, 94, 98,
</p>
<p>414, 519
</p>
<p>Rotation axis, 18, 26, 71, 163, 200ff, 229, 505, 594, 638,
</p>
<p>659, 694
</p>
<p>Rotation function, 384, 387, 397, 516, 518ff
</p>
<p>Rotation function search, 523
</p>
<p>Rotation matrices, 98. See also Web Appendix WA4
</p>
<p>Rotation point, 73ff, 90, 102, 368, 688
</p>
<p>Rotation x-ray photograph, 212
</p>
<p>Roto-inversion axes, 40
</p>
<p>Row, 51ff, 131ff
</p>
<p>R ratio, 326
</p>
<p>S
Sampling interval, 257, 284, 346
</p>
<p>Sayre&rsquo;s equation, 270
</p>
<p>Scale factor, 167, 173, 282, 307, 401ff, 444ff, 504, 532,
</p>
<p>585, 621, 647, 727
</p>
<p>Scaling of intensity data, 167. See also Intensity data,
</p>
<p>scaling of
</p>
<p>Scan procedure, 212
</p>
<p>Scattering. See also Anomalous scattering; X-ray
</p>
<p>scattering (diffraction) by crystals
</p>
<p>anomalous (see Anomalous scattering)
</p>
<p>coherent, 127, 155, 549, 569
</p>
<p>everyday examples of, 121
</p>
<p>factor
</p>
<p>for neutrons, 553
</p>
<p>for x-rays, 124, 129, 142, 155
</p>
<p>incoherent, 127, 549, 554, 560, 597
</p>
<p>of light, 128
</p>
<p>ratio, r, 302
</p>
<p>vector, 124, 129, 211
</p>
<p>Schönflies, A., 3
</p>
<p>Schönflies notation
</p>
<p>and Hermann&ndash;Mauguin notation, compared, 41
</p>
<p>for point groups, 86, 159
</p>
<p>Scintillation counter, 161, 209, 504, 591, 595
</p>
<p>Screw axis
</p>
<p>limiting conditions for, 84
</p>
<p>notation for, 84
</p>
<p>Search structure, 381, 430, 503, 514
</p>
<p>Secondary extinction. See Extinction
</p>
<p>Seeding, 474, 496
</p>
<p>Selenium-mutated methionine, 527
</p>
<p>Self-rotation function, 510, 518ff
</p>
<p>Self-translation function, 441, 517
</p>
<p>Self-vector set, 382, 387
</p>
<p>Seminvariant. See Structure seminvariant
</p>
<p>Seminvariant representations (SIR), 611
</p>
<p>Series termination error, 240, 287, 416, 550
</p>
<p>Shake and Bake, 399
</p>
<p>SHELX-99, 304ff, 326, 349, 374ff, 393, 407ff, 532,
</p>
<p>538, 564
</p>
<p>example analysis with, 377ff
</p>
<p>SHELX program system, 374
</p>
<p>Shock cooling (flash freezing), 501
</p>
<p>Sigma one (Σ1), formula for, 376
</p>
<p>Sigma two (Σ2). See also Signs of reflections in
</p>
<p>centrosymmetric crystals
</p>
<p>examples of, 360
</p>
<p>formula for, 359
</p>
<p>listing, 358, 370
</p>
<p>and symbolic addition, 359
</p>
<p>Sigma weighting, 365
</p>
<p>Sign determination. See also Centrosymmetric crystals;
</p>
<p>Direct methods of phasing; Sigma two;
</p>
<p>Triple product relationship
</p>
<p>of reflections in centrosymmetric crystals, 355
</p>
<p>by symbolic addition, 359
</p>
<p>Significance test, 410
</p>
<p>Sign relationships and Fourier transform, 268
</p>
<p>Signs of reflections in centrosymmetric crystals, 141
</p>
<p>Silica glass, 7
</p>
<p>Simulated annealing, 376, 427, 531ff, 577, 609, 621, 627.
</p>
<p>See also Powder method
</p>
<p>Single-crystal x-ray diffraction techniques, 197ff
</p>
<p>data collection in, 197
</p>
<p>Single isomorphous replacement (SIR), 314, 335,
</p>
<p>503, 611
</p>
<p>Single isomorphous replacement with anomalous
</p>
<p>scattering, 335, 503
</p>
<p>Sinusoidal wave, 251
</p>
<p>SIR. See Single isomorphous replacement (SIR)
</p>
<p>SIRAS. See Isomorphous replacement, single, with
</p>
<p>anomalous scattering
</p>
<p>Site directed mutagenesis and Siras, 522
</p>
<p>Slater wavefunction, 155
</p>
<p>Small-angle scattering, 223
</p>
<p>Small circle, 28
</p>
<p>Small-molecule structures, 304ff, 375, 382, 399, 532
</p>
<p>Sodium chloride, 4, 72, 103, 115, 235, 493,
</p>
<p>582, 599, 725
</p>
<p>Software for crystallography, 344
</p>
<p>Solid state detector, 594
</p>
<p>Solvent flattening, 528
</p>
<p>Solvent of crystallization, 274, 420, 643
</p>
<p>Space-filling patterns, 25
</p>
<p>Space groups
</p>
<p>&lsquo;additional&rsquo; symmetry elements of, 94
</p>
<p>ambiguity in determination of, 269, 319
</p>
<p>center of symmetry in, 22, 30, 38, 88, 94, 141, 200
</p>
<p>enantiomorphous pairs of, 327, 367
</p>
<p>750 Index</p>
<p/>
</div>
<div class="page"><p/>
<p>equivalent positions in
</p>
<p>general, 74, 83, 93, 274, 288, 296
</p>
<p>special, 74
</p>
<p>fractional coordinates in, 54, 74, 244, 260, 282, 309
</p>
<p>and geometric structure factor, 358
</p>
<p>hexagonal, 94, 151, 155
</p>
<p>limiting conditions for x-ray reflection in, 598
</p>
<p>matrix representation of, 97ff
</p>
<p>monoclinic, 81ff, 107, 152, 158, 327ff, 389, 493,
</p>
<p>519, 563
</p>
<p>origin shift for, 93 (see also Origin, change of)
</p>
<p>orthorhombic, 59, 90, 101, 106, 154, 329, 426, 577
</p>
<p>pattern for, 33, 72, 509
</p>
<p>and point groups, 86ff
</p>
<p>practical determination of, 152ff
</p>
<p>projections of, 36, 487
</p>
<p>as repetition of point group pattern by Bravais
</p>
<p>lattice, 72
</p>
<p>&lsquo;special&rsquo; pairs of, 101
</p>
<p>standard diagrams for, 94
</p>
<p>symbol, analysis of, 89
</p>
<p>table of, 92, 101, 352, 431, 456
</p>
<p>tetragonal, 94, 154
</p>
<p>theory, 51ff
</p>
<p>three-dimensional, 7, 23, 80
</p>
<p>three-dimensional by symbol
</p>
<p>C2, 39, 81, 240
</p>
<p>Imma, 247
</p>
<p>P1,
</p>
<p>P1: amplitude and phase symmetry for, 261,
</p>
<p>273, 569ff
</p>
<p>P1: diagram for, 286, 293
</p>
<p>P1: jjEj statistics for, 644
P1: general equivalent positions in, 149
</p>
<p>P1: origin-fixing reflections in, 352ff, 458
</p>
<p>P2, 81, 289
</p>
<p>P21
P21: amplitude and phase symmetry for, 282,
</p>
<p>314, 320
</p>
<p>P21: diagram for, 276, 309
</p>
<p>P21: general equivalent positions in, 148, 656
</p>
<p>P21: geometric structure factor for, 358
</p>
<p>P21: limiting conditions in, 85
</p>
<p>P21: origin-fixing reflections for, 458
</p>
<p>P212121, 90, 101, 154, 347, 352, 367, 423,
</p>
<p>434, 473, 502, 577, 647, 719
</p>
<p>Pc
</p>
<p>Pc: equivalence with Pa and Pn, 142
</p>
<p>Pc: general equivalent positions in, 149
</p>
<p>Pc: geometric structure factor for, 358
</p>
<p>Pc: limiting conditions in, 146
</p>
<p>Pc: reciprocal net for, 148
</p>
<p>P21/c
</p>
<p>P21/c: amplitude and phase symmetry, 126
</p>
<p>P21/c: analysis of symbol for, 152
</p>
<p>P21/c: diagram for, 148, 276
</p>
<p>P21/c: general equivalent positions in, 149
</p>
<p>P21/c: geometric structure factor for, 358
</p>
<p>P21/c: limiting conditions in, 153
</p>
<p>P21/c: origin-fixing reflections in, 352ff, 458
</p>
<p>P21/c: special equivalent positions in, 149, 151
</p>
<p>P63/m, 95, 96, 151
</p>
<p>Pma2, 101, 148
</p>
<p>Pman, 150
</p>
<p>Pmma, 93, 101, 298
</p>
<p>P4nc, 94, 151
</p>
<p>Pnma, 90, 99&ndash;101, 655
</p>
<p>two-dimensional (see Plane groups)
</p>
<p>Space groups unit cell, centred, 565
</p>
<p>Special equivalent positions, 74, 149, 151. See also
</p>
<p>Equivalent positions
</p>
<p>molecules in, 74
</p>
<p>Special form, 21ff, 35
</p>
<p>Special intensity distributions. See Distribution
</p>
<p>Sphere of reflection, 138, 163, 249. See also Ewald sphere
</p>
<p>Spherical harmonics, 167
</p>
<p>Spherical projection, 15
</p>
<p>Spherical symmetry, 129, 247
</p>
<p>Spherical triangle, 38
</p>
<p>Spherical trigonometry, 136. See also Web
</p>
<p>Appendix WA3
</p>
<p>Square two-dimensional system, 52
</p>
<p>Square wave
</p>
<p>as Fourier series, 240
</p>
<p>Fourier transform of, 255
</p>
<p>termination errors for, 240
</p>
<p>Standard deviation of electron density, 416
</p>
<p>Statistics. See Intensity, statistics
</p>
<p>Stensen, N., 1
</p>
<p>Step-scan moving window measurement, 168
</p>
<p>Stereogram. See also Point groups; Projections;
</p>
<p>Stereographic projection
</p>
<p>fundamental properties of, 556
</p>
<p>indexing of, 28
</p>
<p>notation for, 23, 34, 81
</p>
<p>for point groups, 28
</p>
<p>for three-dimensional point groups, 22
</p>
<p>for two-dimensional point groups, 20
</p>
<p>uses of, 17, 27
</p>
<p>Stereograms, 15ff, 70, 81, 177, 520, 636, 659, 694
</p>
<p>Stereographic projection, 15ff, 50, 135. See also
</p>
<p>Stereograms, Web Appendix WA2
</p>
<p>Stereoscopic images. See Stereoviewer; Stereoviews
</p>
<p>Stereoviewer, 5, 659ff
</p>
<p>Stereoviews, 4, 22, 34, 57, 89, 274, 277, 319, 454,
</p>
<p>470, 713
</p>
<p>Straight extinction, 193ff, 229, 289, 440, 455
</p>
<p>Structural data. See Bond lengths and angles; Data bases;
</p>
<p>Ionic radii; van der Waals contact distances
</p>
<p>Structure analysis. See also Direct methods of phase
</p>
<p>determination; Heavy-atom method; Reflection,
</p>
<p>x-ray; X-ray scattering (diffraction) by crystals
</p>
<p>accuracy of (see Precision of calculations)
</p>
<p>of atropine, 327, 329, 394
</p>
<p>of 5-azauracil, 422
</p>
<p>of azidopurine monohydrate, 280
</p>
<p>of benzene, 597
</p>
<p>of 1-benzyl-1H-tetrazole, 329, 426
</p>
<p>Index 751</p>
<p/>
</div>
<div class="page"><p/>
<p>Structure analysis. (cont.)
</p>
<p>of bisdiphenylmethyldiselenide, 289, 295
</p>
<p>of bromobenzo[b]indeno[1,2-e]pyran, 439
</p>
<p>of calcium uranate, 609
</p>
<p>of cholesteryl iodide, 327
</p>
<p>of cimetidine, 610
</p>
<p>computer use in, 223, 325, 384, 422
</p>
<p>of concanavalin A, 568, 576
</p>
<p>of coumarin derivative, 390, 392
</p>
<p>criteria for correctness, 406, 416
</p>
<p>of crown ether derivative, 574
</p>
<p>of cyclosporin H, 560, 574ff
</p>
<p>diiodo-(N, N, N&rsquo;, N&rsquo;)-tetramethylethylenediamine)
</p>
<p>zinc(II), 89
</p>
<p>errors in trial structure during, 303
</p>
<p>euphenyl iodoacetate, 5, 260, 280
</p>
<p>of α-lanthanum tungstate, 622
</p>
<p>limitations of, 419
</p>
<p>of manganese phosphate monohydrate, 609
</p>
<p>and neutron diffraction, 568 (see also Crown ether
</p>
<p>derivative)
</p>
<p>as over-determined problem, 533
</p>
<p>of papaverine hydrochloride, 274
</p>
<p>of p-bromophenylethanoic acid, 618
</p>
<p>phase problem in, 253, 282ff, 351, 527, 588, 608
</p>
<p>of potassium dihydrogen phosphate, 190, 549
</p>
<p>of potassium dimercury, 296
</p>
<p>of potassium hexachloroplatinate (IV), 285
</p>
<p>of potassium 2-hydroxy-3,4-dioxocyclobut-1-en-1-
</p>
<p>olate monohydrate, 455
</p>
<p>precision of, 596
</p>
<p>preliminary stages of, 509, 518, 531
</p>
<p>of proteins, 6, 514
</p>
<p>pyridoxal phosphate oxime dehydrate, 352ff
</p>
<p>symbolic addition in, 359, 368
</p>
<p>refinement in, 385, 533, 550, 588
</p>
<p>of ricin agglutinin, 509, 512, 520ff
</p>
<p>of silver-pyrazole complex, 612
</p>
<p>and symmetry analysis, 290
</p>
<p>1,8-(3,6,9-trioxaundecane-1,11-diyldioxy)-9,10-
</p>
<p>dihydro-10,10-dimethylanthracene-9-ol, 560
</p>
<p>tubercidin, 367, 372
</p>
<p>of zeolites, 614, 623
</p>
<p>of zinc&ndash;insulin complex (T3R3), 623
</p>
<p>of zinc&ndash;silicate complex (VIP-9), 614
</p>
<p>Structure factor. See also Phase, determination; Phase,
</p>
<p>of structure factor
</p>
<p>agreement of, 338, 357, 401
</p>
<p>amplitude of
</p>
<p>absolute scale of, 215, 351, 563
</p>
<p>invariance under change of origin, 253, 353
</p>
<p>amplitude symmetry of, 159, 358, 389, 433
</p>
<p>with anomalous scattering, 325ff
</p>
<p>applications of equation for, 174ff, 249, 276, 351, 532,
</p>
<p>582, 702, 714
</p>
<p>for body-centred (l) unit cell, 142
</p>
<p>calculated, 165, 303, 386, 406, 503, 533
</p>
<p>for centrosymmetric crystal, 141
</p>
<p>change of origin and, 252, 353
</p>
<p>conjugate, 367
</p>
<p>defined, 179, 240, 306, 363, 400
</p>
<p>equation for, 140
</p>
<p>as Fourier transform of electron density, 235ff
</p>
<p>generalized form of, 242, 244
</p>
<p>geometric, 340ff, 358, 433
</p>
<p>invariance of under change of origin, 253, 353
</p>
<p>local average value of, 287
</p>
<p>normalized, 172, 182, 351, 400
</p>
<p>observed, 165, 306, 533
</p>
<p>and parity group, 354
</p>
<p>phase of, 140, 180, 301
</p>
<p>phase symmetry of, 358, 457
</p>
<p>plotted on Argand diagram, 124ff
</p>
<p>reduced equation for, 142
</p>
<p>representation as vector, 97, 135, 318
</p>
<p>sign-determining formula for, 315, 355ff, 458
</p>
<p>and special equivalent positions sets
</p>
<p>for space group P21, 85, 144, 156
</p>
<p>for space group Pc, 146
</p>
<p>for space group P21/c, 146, 358
</p>
<p>for space group Pma2, 148, 156
</p>
<p>for space group Pman, 150
</p>
<p>and symmetry elements, 373
</p>
<p>and translational symmetry, 176
</p>
<p>unitary, 182, 366
</p>
<p>Structure invariant, 352, 372, 389, 400, 435, 612, 692
</p>
<p>Structure refinement
</p>
<p>from neutron data, 569
</p>
<p>from x-ray data, 569
</p>
<p>Structure seminvariant, 354, 369, 372, 435, 459, 611, 649,
</p>
<p>719, 721
</p>
<p>Subgroup, 28, 34, 74, 87, 95,
</p>
<p>146, 247
</p>
<p>Subunit, 471, 473, 479ff, 491, 524
</p>
<p>Superposition technique, 300, 650
</p>
<p>Symbolic addition procedure, 359
</p>
<p>advantages and disadvantages of, 371
</p>
<p>Symbolic phases, 360, 369, 372
</p>
<p>Symbolic signs for reflections, 360, 459
</p>
<p>Symmetric extinction, 195
</p>
<p>Symmetry
</p>
<p>black-white, 102ff
</p>
<p>potassium chloride, 103
</p>
<p>characteristic, 24, 26
</p>
<p>colour, 102ff
</p>
<p>cylindrical, 31
</p>
<p>definition of, 36
</p>
<p>of diffraction pattern, 330
</p>
<p>examples of molecular, 637
</p>
<p>external, 17&ndash;39
</p>
<p>glide plane, 86, 142, 150, 275, 501
</p>
<p>and indistinguishability, 25
</p>
<p>internal, 194
</p>
<p>inversion axis, 22 (see also Roto-inversion axes)
</p>
<p>of Laue photograph, 199
</p>
<p>mirror (see Reflection symmetry)
</p>
<p>notations for, 85, 87
</p>
<p>onefold, 21
</p>
<p>752 Index</p>
<p/>
</div>
<div class="page"><p/>
<p>operation, 17, 22, 40, 52, 73ff, 94ff, 146, 152, 228,
</p>
<p>378, 384, 390, 405, 452, 491, 636
</p>
<p>operator, 99, 106, 637
</p>
<p>permitted, 175, 372
</p>
<p>and physical properties, 99
</p>
<p>rotational, 21, 25, 32, 36, 52, 71, 74, 94, 98, 414, 519
</p>
<p>screw axis, 84, 144, 356
</p>
<p>and structure analysis, 5
</p>
<p>translational, 38, 76ff, 85ff, 120, 142ff, 176, 252,
</p>
<p>275, 710
</p>
<p>of x-ray diffraction pattern, 4, 30, 142, 200, 290,
</p>
<p>330, 509
</p>
<p>Symmetry analysis, 275ff, 290, 346
</p>
<p>Symmetry axis
</p>
<p>alternating, See Schönflies notation
</p>
<p>principal, 26
</p>
<p>Symmetry elements. See also Symmetry operations,
</p>
<p>combinations of
</p>
<p>combinations of, 21
</p>
<p>geometrical extension of, 19
</p>
<p>matrix representation of, 97ff
</p>
<p>notation for, 85ff
</p>
<p>sign changes in, 92
</p>
<p>in three dimensions, 22ff
</p>
<p>in two dimensions, 19ff
</p>
<p>Symmetry-equivalent points. See Equivalent positions
</p>
<p>Symmetry-independent species, 143
</p>
<p>Symmetry operations, combinations of, 24, 636
</p>
<p>Symmetry operator, 99, 106, 637
</p>
<p>Symmetry plane. See Reflection plane
</p>
<p>Symmetry point, 21, 36, 74, 87, 106
</p>
<p>Symmetry-related atoms and molecules, 317
</p>
<p>Synchrotron radiation (SR)
</p>
<p>diamond installation for, 121
</p>
<p>filtered, 116 (see also Monochromators)
</p>
<p>insertion devices for
</p>
<p>undulators, 120
</p>
<p>wigglers, 120
</p>
<p>oscillation method with, 205ff
</p>
<p>worked example of, 207
</p>
<p>photon intensity from, 119
</p>
<p>polarization of radiation from, 119
</p>
<p>radiation flux from, 117
</p>
<p>sources, 118, 223, 504
</p>
<p>Systematic absences. See Absences; Limiting conditions
</p>
<p>Systems
</p>
<p>three-dimensional, 21, 27 (see also Crystal systems)
</p>
<p>two-dimensional, 21, 52, 73
</p>
<p>T
Tangent formula, 362ff, 372ff
</p>
<p>Target structure, 381ff, 434, 503, 522
</p>
<p>Temperature factor. See also Thermal vibrations
</p>
<p>anisotropic, 444, 449, 579
</p>
<p>correction, 402
</p>
<p>isotropic, 166ff, 171ff, 190ff, 287, 445
</p>
<p>and scale factor, 168, 173, 282, 307, 401ff, 504, 585,
</p>
<p>621, 647, 727
</p>
<p>Termination errors in Fourier series, 240, 416
</p>
<p>Tetrad, 25, 60
</p>
<p>Tetragonal crystal system. See also Crystal systems
</p>
<p>model of a crystal in,
</p>
<p>optical behaviour of crystals in, 190ff
</p>
<p>space groups in, 154
</p>
<p>symmetry of, 9, 26ff, 192
</p>
<p>unit cells in, 60, 90, 409
</p>
<p>Tetrahedron, model of, 638
</p>
<p>Thermal vibrations. See also Temperature factor
</p>
<p>anisotropic, 171
</p>
<p>Debye&ndash;Waller factor, 170
</p>
<p>statistical expectation value of, 172
</p>
<p>isotropic, 282, 433
</p>
<p>and mean square atom displacement, 170
</p>
<p>one-dimensional analysis of, 169
</p>
<p>and smearing of electron density, 6
</p>
<p>three-dimensional analysis, 403
</p>
<p>Thomson scattering. See Coherent; Scattering
</p>
<p>Three-phase structure invariants. See Triplets
</p>
<p>Tiled CCD, 219, 222. See also Charge-coupled type
</p>
<p>area detector (CCD)
</p>
<p>Time-of-flight neutron diffractometer, 558
</p>
<p>Time-of-reflection opportunity, 163
</p>
<p>Transformations
</p>
<p>of coordinate axes, 69, 353
</p>
<p>of coordinates in unit cell, 60, 647
</p>
<p>of directions, 58, 65
</p>
<p>inverse, 65, 211, 255
</p>
<p>of Miller indices, 68, 81, 731
</p>
<p>mnemonic for, 654
</p>
<p>of reciprocal unit cell vectors, 65, 68, 211
</p>
<p>of unit cell vectors, 54ff, 78, 731
</p>
<p>of zone symbols, 65ff
</p>
<p>Translation
</p>
<p>function, 517ff, 523ff
</p>
<p>search, 387ff
</p>
<p>vector, 51, 97, 99, 100, 384, 692, 669
</p>
<p>symmetry, 38, 76, 78, 79, 85, 91, 99, 100, 142,
</p>
<p>145, 146, 152, 176, 275, 276, 707
</p>
<p>Transmission
</p>
<p>factor, 159, 165
</p>
<p>profile, 166
</p>
<p>Triad, 25, 60, 227
</p>
<p>Trial-and-error method, 273, 599, 600, 727
</p>
<p>Triclinic crystal system, 190
</p>
<p>Trigonal crystal system, 190
</p>
<p>Trigonal lattice, 61
</p>
<p>Trigonometric formulae, 140, 162, 237. See also Web
</p>
<p>Appendix WA5
</p>
<p>Triple phase relationship (TPR), 374, 376
</p>
<p>Triple product relationship, 360, 363
</p>
<p>Triple product sign relationship. See also Signs of
</p>
<p>reflections in centrosymmetric crystals
</p>
<p>physical interpretation of, 356
</p>
<p>Σ2 formula for, 355ff
</p>
<p>Triplets, 97, 268, 355, 363, 372ff, 400, 433ff, 451ff, 486,
</p>
<p>523, 721
</p>
<p>Triply primitive hexagonal unit cell, 61, 108
</p>
<p>Tungsten Lα-radiation, 116
</p>
<p>Index 753</p>
<p/>
</div>
<div class="page"><p/>
<p>Twin
</p>
<p>axis of, 226
</p>
<p>contact, 227
</p>
<p>hemitropic, 226
</p>
<p>interpenetrant, 226
</p>
<p>lamellar, 227, 228
</p>
<p>mechanical separation of, 226
</p>
<p>symmetric, 226
</p>
<p>Twinning
</p>
<p>in calcite, 227
</p>
<p>composition plane in, 226
</p>
<p>in fluorite, 227
</p>
<p>in gypsum, 226
</p>
<p>merohedral, 228
</p>
<p>morphology of, 226
</p>
<p>non-merohedral, 228
</p>
<p>pseudo-merohedral, 228
</p>
<p>x-ray diffraction and, 228
</p>
<p>Two-dimensional system, 21, 52, 106, 693
</p>
<p>U
Uniaxial crystals. See also Anisotropy, optical; Crystal;
</p>
<p>Optically anisotropic crystals
</p>
<p>defined, 192
</p>
<p>idealized cross-section of, 194
</p>
<p>idealized interference figure for, 171
</p>
<p>and Laue photograph, 200
</p>
<p>optic axis of, 192
</p>
<p>Unit cell. See also Bravais lattices; Reciprocal lattice
</p>
<p>centred, 550, 565
</p>
<p>contents of, 73, 264, 273ff, 290, 298, 342, 518
</p>
<p>conventional choice of, 10
</p>
<p>dimensions of, 401, 423, 466, 555
</p>
<p>limiting conditions for type of, 80ff, 106, 138, 142ff,
</p>
<p>249, 510, 598, 604
</p>
<p>notation for, 54
</p>
<p>number of lattice points in, 55, 57, 61
</p>
<p>one-dimensional, 52, 169, 178, 241, 282, 284, 427
</p>
<p>parameters
</p>
<p>errors in, 390, 599
</p>
<p>measurement of, 506
</p>
<p>primitive, 52ff, 73ff
</p>
<p>reciprocity of F and I, 138
</p>
<p>reduction of, 569, 654
</p>
<p>scattering of x-rays by, 103, 282
</p>
<p>symbols for, 52ff, 73, 178
</p>
<p>three-dimensional, 54ff, 83, 244
</p>
<p>transformations of, 65ff, 565
</p>
<p>translations associated with type of, 138
</p>
<p>triply primitive hexagonal, 60
</p>
<p>volume of, 54, 137
</p>
<p>Units, prefixes to, 408
</p>
<p>&lsquo;Unobserved&rsquo; reflections. See Reflections, unobserved
</p>
<p>Uranium heptafluoride, molecular symmetry of, 34
</p>
<p>V
van der Waals contact distances, 224, 645, 649
</p>
<p>Vector interactions, 288ff, 342. See also Vectors
</p>
<p>Vectors. See also Vector interactions
</p>
<p>algebra and reciprocal lattice, 173
</p>
<p>complex, 124, 381
</p>
<p>cross, 318, 384
</p>
<p>interatomic, in Patterson map, 284, 381ff,
</p>
<p>518ff, 647
</p>
<p>map, 285
</p>
<p>overlap, 293
</p>
<p>repeat, 72
</p>
<p>scalar product of, 58, 64
</p>
<p>superposition map, 267
</p>
<p>translation, 51, 97ff, 384, 694, 728
</p>
<p>triplet, 97, 355ff, 457, 719
</p>
<p>vector product of, 136
</p>
<p>verification, 381, 388
</p>
<p>Velocity, wave, 558
</p>
<p>Vibration directions, 189ff
</p>
<p>Volume
</p>
<p>of real (direct) unit cell, 63
</p>
<p>of reciprocal unit cell, 137, 281
</p>
<p>von Groth, P., 3
</p>
<p>von Laue, M., 4, 130, 133
</p>
<p>W
Wave
</p>
<p>amplitude of, 123
</p>
<p>and Argand diagram, 124ff
</p>
<p>combinations, 124ff, 133, 248
</p>
<p>energy associated with, 120, 522
</p>
<p>phase of, 123, 241
</p>
<p>resultant of combination of, 127, 135
</p>
<p>Wavelength
</p>
<p>of neutrons, 596
</p>
<p>of x-rays, 111ff, 164, 189, 202ff
</p>
<p>Wave sums, graphical representation of, 123ff
</p>
<p>Web appendix materials.
</p>
<p>See also http://extras.springer.com
</p>
<p>angle between planes, 137
</p>
<p>angle between two lines, 132
</p>
<p>direct and reciprocal unit-cell volumes, 136
</p>
<p>direction cosines, 7, 353
</p>
<p>interplanar spacing, 136
</p>
<p>plane trigonometry formulae, 136
</p>
<p>reciprocal lattice, 137
</p>
<p>reciprocity of I and F unit cells, 138
</p>
<p>rotation matrices, 94
</p>
<p>spherical trigonometry, 136
</p>
<p>Web program packages.
</p>
<p>See also http://extras.springer.com
</p>
<p>general
</p>
<p>coordinate transformation (INTXYZ), 652
</p>
<p>linear least squares (LSLI), 653
</p>
<p>matrix operations (MATOPS), 653
</p>
<p>molecular geometry (MOLGOM),
</p>
<p>327, 649
</p>
<p>one-dimensional Forier transform (TRANS1),
</p>
<p>256, 651
</p>
<p>one-dimensional Fourier series (FOUR1D), 256,
</p>
<p>257, 270, 650
</p>
<p>two-dimensional Fourier series (FOUR2D),
</p>
<p>644, 650
</p>
<p>source code for, 651
</p>
<p>754 Index</p>
<p/>
</div>
<div class="page"><p/>
<p>point groups
</p>
<p>derivation (EULR), 24, 636
</p>
<p>recognition (SYMM), 28, 379, 390, 396, 637ff
</p>
<p>powder
</p>
<p>direct-reciprocal unit-cell parameters (RECIP),
</p>
<p>601, 652
</p>
<p>indexing, automatic, 568, 604
</p>
<p>CRYSFIRE, 604
</p>
<p>ITO12, 654
</p>
<p>Q values (QVALS), 653
</p>
<p>strucure determination (ESPOIR), 621ff
</p>
<p>unit-cell reduction (LEPAGE), 602, 654
</p>
<p>structure determination, simulated (XRAY)
</p>
<p>data preparation (MAKDAT), 640
</p>
<p>direct methods, 268, 355
</p>
<p>example data sets for, 175, 504, 516, 554, 641
</p>
<p>fourier series(electron density), 241
</p>
<p>geometry (distance-angle), 415
</p>
<p>least-squares refinement, 375, 384, 407, 419
</p>
<p>Patterson function, 282ff, 641
</p>
<p>Patterson superposition, 341
</p>
<p>structure factors, 501ff
</p>
<p>Wilson method, 169 Web program suite,
</p>
<p>http://extras.springer.com
</p>
<p>Weight
</p>
<p>function for, 285
</p>
<p>measurement of, 606
</p>
<p>Weighted reciprocal lattice, 148, 172ff, 197, 259ff,
</p>
<p>330, 590,703
</p>
<p>Weighted tangent formula, 363ff. See also Tangent
</p>
<p>formula
</p>
<p>Weiss, C.S., 3
</p>
<p>Weissenberg photograph, 441
</p>
<p>unit cell dimensions from, 441
</p>
<p>Weissenbrg chart, 445
</p>
<p>Weiss zone law, 13, 68
</p>
<p>Whewell, W., 9
</p>
<p>&lsquo;White&rsquo; radiation, 111ff, 197, 230, 443, 703, 728
</p>
<p>Wilson, A.J.C., 173
</p>
<p>Wilson plot, 172, 400, 446, 627, 644ff, 690
</p>
<p>auxillary plot, 174
</p>
<p>Wyckoff space group notation, 74
</p>
<p>X
X-radiation
</p>
<p>copper, 114ff
</p>
<p>dependence on wavelength, 123, 140
</p>
<p>filtered (see Monochromators)
</p>
<p>molybdenum, 220, 328
</p>
<p>monochromatic, 205, 230, 552, 593
</p>
<p>tungsten, 116
</p>
<p>&lsquo;white,&rsquo; 111ff, 197, 230, 443, 703, 728
</p>
<p>X-rays. See also X-radiation
</p>
<p>absorption of, 114ff
</p>
<p>characteristic, 113
</p>
<p>detectors, 118, 213, 222, 508
</p>
<p>diffraction and reciprocal lattice, 136, 197, 202, 232,
</p>
<p>504, 509 (see also Reciprocal lattice)
</p>
<p>diffraction by liquids (see Liquids, x-ray diffraction
</p>
<p>from)
</p>
<p>diffraction pattern (see also X-ray scattering by
</p>
<p>crystals; X-rays, diffraction photograph/image)
</p>
<p>centrosymmetric nature of, 29, 140
</p>
<p>and Friedel&rsquo;s law, 140, 147, 332
</p>
<p>and geometric structure of crystal, 358
</p>
<p>intensity in, 474, 504
</p>
<p>position in, 143, 200, 289, 510, 575
</p>
<p>symmetry of, 36, 146, 200
</p>
<p>as weighted reciprocal lattice, 148, 172ff, 197,
</p>
<p>259ff, 330
</p>
<p>diffraction photograph/image
</p>
<p>important features of, 121, 225
</p>
<p>indexing of, 23, 592
</p>
<p>by Laue method, 197ff
</p>
<p>measurement of intensity of reflection on, 304
</p>
<p>by oscillation method, 205
</p>
<p>for powder sample, 594
</p>
<p>by precession method, 36, 180, 197, 233, 330, 506
</p>
<p>for single crystal, 152, 187ff, 375,
</p>
<p>441, 501
</p>
<p>extinction of, 195
</p>
<p>generation of, 217
</p>
<p>generators
</p>
<p>Bruker AXS, 197, 215, 225
</p>
<p>Rigaku MSC, 223
</p>
<p>rotating anode, 111ff, 219ff, 337, 504, 569
</p>
<p>non-focussing property of, 503
</p>
<p>photograph, 30, 152, 180, 200, 212, 259, 330, 332,
</p>
<p>440, 441, 466
</p>
<p>properties of, 111ff
</p>
<p>reflections (see Reflection, x-ray)
</p>
<p>tube
</p>
<p>rotating anode, 111ff, 219ff, 337, 504, 569
</p>
<p>sealed tube, 119, 223, 337
</p>
<p>wavelengths of, 111ff, 136, 230, 347, 511, 699
</p>
<p>wave-like properties of, 4
</p>
<p>X-ray scattering (diffraction) by crystals. See also
</p>
<p>Diffraction; Reflection; X-rays,
</p>
<p>diffraction pattern; X-rays, diffraction
</p>
<p>photograph/image
</p>
<p>anomalous, 140, 306, 325, 330ff
</p>
<p>by atom, 128, 331
</p>
<p>Bragg treatment of, 76, 134
</p>
<p>coherent, 127, 549, 569
</p>
<p>Compton (see Scattering, incoherent)
</p>
<p>by crystal structure, 5, 553
</p>
<p>as a Fourier analysis, 240
</p>
<p>generalized treatment of, 201
</p>
<p>incoherent, 127, 155, 549, 554, 560ff, 597
</p>
<p>and indices of planes, 8ff, 78
</p>
<p>intensity
</p>
<p>ideal, 159ff, 172ff
</p>
<p>Laue treatment of, 133
</p>
<p>equivalence with Bragg treatment, 76, 134
</p>
<p>from monoclinic crystals, 152
</p>
<p>order of, 576
</p>
<p>for orthorhombic crystals, 152
</p>
<p>phase difference in, 164
</p>
<p>by regular array of atoms, 130ff
</p>
<p>for single crystal, 501, 556
</p>
<p>Index 755</p>
<p/>
</div>
<div class="page"><p/>
<p>X-ray scattering (diffraction) by crystals. (cont.)
</p>
<p>by single electron, 122, 139
</p>
<p>and space group determination, 228, 501
</p>
<p>and symmetry of Patterson function, 273, 383
</p>
<p>theory of, 127
</p>
<p>Thomson (see Coherent; Scattering)
</p>
<p>total energy of, 161
</p>
<p>by two or more electrons, 122, 624
</p>
<p>from unit cell, 286, 626, 650
</p>
<p>vector, 136, 386, 450
</p>
<p>X-ray scattering (diffraction) by lattice array of
</p>
<p>scattering points
</p>
<p>one-dimensional, 19, 52, 131, 169, 264
</p>
<p>three-dimensional, 241ff, 285, 328, 446
</p>
<p>two-dimensional, 121, 175, 236, 251, 279
</p>
<p>Y
Young&rsquo;s fringes, 250
</p>
<p>Z
Zinc sulphide, 4
</p>
<p>Zone
</p>
<p>axis, 13ff, 197ff
</p>
<p>circle, 16
</p>
<p>indexing (see Powder indexing)
</p>
<p>symbol, 13, 40, 51, 65ff, 108, 654
</p>
<p>756 Index</p>
<p/>
</div>
<ul>	<li>Foreword</li>
	<li>Preface to the Fifth Edition</li>
	<li>Disclaimer</li>
	<li>Table of Contents</li>
	<li>Physical Constants and OtherNumerical Data</li>
	<li>1: Crystal Morphology and Crystal Symmetry&#13;</li>
<ul>	<li>1.1 Brief Historical Introduction</li>
	<li>1.2 The Crystalline State</li>
<ul>	<li>1.2.1 Crystallographic Reference Axes</li>
	<li>1.2.2 Equation of a Plane</li>
	<li>1.2.3 Indices of Planes and the Law of Rational Intercepts</li>
	<li>1.2.4 Axial Ratios</li>
	<li>1.2.5 Zones</li>
</ul>
	<li>1.3 Stereographic Projection: Brief Survey</li>
	<li>1.4 External Symmetry of Crystals</li>
<ul>	<li>1.4.1 Two-Dimensional Point Groups</li>
	<li>1.4.2 Three-Dimensional Point Groups</li>
	<li>1.4.3 Quasicrystals, Buckyballs, and Icosahedral Symmetry</li>
</ul>
	<li>1.5 Problems</li>
	<li>References and Bibliography</li>
<ul>	<li>Bibliography: General and Historical Study of Crystallography</li>
	<li>Crystal Morphology and Stereographic Projection</li>
	<li>Crystal Symmetry and Point Groups</li>
</ul>
</ul>
	<li>2:&#13;Lattices and Space-Group Theory</li>
<ul>	<li>2.1 Introduction</li>
	<li>2.2 Lattices</li>
<ul>	<li>2.2.1 Two-Dimensional Lattices</li>
	<li>2.2.2 Choice of Unit Cell</li>
	<li>2.2.3 Three-Dimensional Lattices</li>
</ul>
	<li>2.3 Families of Planes and Interplanar Spacings</li>
	<li>2.4 Reciprocal Lattice: Geometrical Treatment</li>
	<li>2.5 Unit-Cell Transformations</li>
<ul>	<li>2.5.1 Bravais Unit-Cell Vectors</li>
	<li>2.5.2 Directions (Zone Symbols)</li>
	<li>2.5.3 Coordinates of Sites in the Unit Cell</li>
	<li>2.5.4 Miller Indices</li>
	<li>2.5.5 Reciprocal Unit-Cell Vectors</li>
</ul>
	<li>2.6 Rotational Symmetries of Lattices</li>
	<li>2.7 Space Groups</li>
<ul>	<li>2.7.1 Two-Dimensional Space Groups (Plane Groups)</li>
	<li>2.7.2 Plane Groups Related to 2mm</li>
	<li>2.7.3 Three-Dimensional Space Groups</li>
	<li>2.7.4 Screw Axes</li>
	<li>2.7.5 Glide Planes</li>
	<li>2.7.6 Analysis of the Space-Group Symbol</li>
	<li>2.7.7 Orthorhombic Space Groups</li>
	<li>2.7.8 Relative Orientations of Symmetry Elements in Space Groups</li>
	<li>2.7.9 Tetragonal and Hexagonal Space Groups</li>
</ul>
	<li>2.8 Matrix Representation of Symmetry Operations</li>
<ul>	<li>2.8.1 Matrices in Point-Group Symmetry</li>
	<li>2.8.2 Matrices in Space-Group Symmetry</li>
</ul>
	<li>2.9 Diffraction Symbols</li>
	<li>2.10 Some Other Types of Symmetry</li>
<ul>	<li>2.10.1 Black-White Symmetry</li>
	<li>2.10.2 Color Symmetry</li>
</ul>
	<li>2.11 Problems</li>
	<li>References</li>
<ul>	<li>Bibliography: Lattices and Space Groups</li>
	<li>Black-White and Color Symmetry</li>
</ul>
</ul>
	<li>3:&#13;X-Rays and X-Ray Diffraction</li>
<ul>	<li>3.1 Generation and Properties of X-Rays</li>
<ul>	<li>3.1.1 X-Rays and White Radiation</li>
	<li>3.1.2 Characteristic Radiation</li>
	<li>3.1.3 Absorption of X-Rays</li>
	<li>3.1.4 Monochromatic Radiation</li>
	<li>3.1.5 Collimation</li>
	<li>3.1.6 Synchrotron Sources</li>
</ul>
	<li>3.2 X-Ray Scattering</li>
<ul>	<li>3.2.1 Scattering by a Single Electron</li>
	<li>3.2.2 Scattering by Two or More Electrons</li>
	<li>3.2.3 Waves and Wave Sums</li>
	<li>3.2.4 Coherent and Incoherent Scattering</li>
	<li>3.2.5 Scattering by an Atom</li>
</ul>
	<li>3.3 Scattering by Regular Arrays of Atoms</li>
<ul>	<li>3.3.1 Laue Equations</li>
	<li>3.3.2 Bragg Equation</li>
	<li>3.3.3 Equivalence of the Laue and Bragg Equations</li>
	<li>3.3.4 Further Analysis of the Path Difference</li>
</ul>
	<li>3.4 Reciprocal Lattice: Analytical Treatment</li>
<ul>	<li>3.4.1 Reciprocal Lattice Properties</li>
	<li>3.4.2 Reciprocal Lattice and Reflection Condition: Ewald Sphere</li>
</ul>
	<li>3.5 Scattering by a Crystal Structure</li>
<ul>	<li>3.5.1 Structure Factor Equation</li>
</ul>
	<li>3.6 Using the Structure Factor Equation</li>
<ul>	<li>3.6.1 Friedel&acute;s Law</li>
	<li>3.6.2 Structure Factor for a Centrosymmetric Crystal</li>
</ul>
	<li>3.7 Limiting Conditions and Systematic Absences</li>
<ul>	<li>3.7.1 Body-Centered Unit Cell</li>
	<li>3.7.2 Screw Axes and Glide Planes</li>
</ul>
	<li>3.8 Practical Determination of Space Groups from Diffraction Data</li>
<ul>	<li>3.8.1 Monoclinic Space Groups</li>
	<li>3.8.2 Orthorhombic Space Groups</li>
	<li>3.8.3 Tetragonal Space Groups</li>
	<li>3.8.4 Hexagonal Space Groups</li>
</ul>
	<li>3.9 Problems</li>
	<li>References</li>
<ul>	<li>Bibliography: Synchrotron Radiation</li>
	<li>Structure Factor and Intensity</li>
	<li>Atomic Scattering Factors</li>
</ul>
</ul>
	<li>4: Intensities and Intensity Statistics&#13;</li>
<ul>	<li>4.1 Intensity Expressions and Factors Affecting Intensities</li>
<ul>	<li>4.1.1 Polarization and Lorentz Factors</li>
	<li>4.1.2 Extinction</li>
	<li>4.1.3 Absorption Measurement and Correction</li>
	<li>4.1.4 Scaling</li>
	<li>4.1.5 Merging Equivalent Reflections</li>
	<li>4.1.6 Practical Intensity Expression and its Standard Deviation</li>
	<li>4.1.7 Scale Factor for Fo</li>
	<li>4.1.8 Thermal Vibrations and the Temperature Factor</li>
</ul>
	<li>4.2 Intensity Statistics</li>
<ul>	<li>4.2.1 Determining Scale and Temperature Factors</li>
	<li>4.2.2 Other Aspects of the Wilson Plot</li>
	<li>4.2.3 Statistics of Reciprocal Space</li>
	<li>4.2.4 Acentric and Centric Distributions of Structure Factors</li>
	<li>4.2.5 Normalized Structure Factors</li>
</ul>
	<li>4.3 Problems</li>
	<li>References</li>
<ul>	<li>Bibliography: Intensity and Intensity Statistics</li>
</ul>
</ul>
	<li>5:&#13;Examination of Single Crystals: Optical and X-Ray Diffraction Practice</li>
<ul>	<li>5.1 Introduction</li>
	<li>5.2 Crystal Growing</li>
<ul>	<li>5.2.1 Growing Crystals for X-Ray Diffraction</li>
	<li>5.2.2 Crystallization from Solution</li>
	<li>5.2.3 Crystallization by Diffusion</li>
	<li>5.2.4 Crystallization by Sublimation</li>
	<li>5.2.5 Other Issues</li>
</ul>
	<li>5.3 Optical Techniques</li>
<ul>	<li>5.3.1 Polarized Light</li>
	<li>5.3.2 Optical Classification of Crystals</li>
	<li>5.3.3 Uniaxial Crystals</li>
	<li>5.3.4 Birefringence</li>
	<li>5.3.5 Biaxial Crystals</li>
</ul>
	<li>5.4 Single-Crystal X-Ray Diffraction Techniques: Intensity Data Collection</li>
<ul>	<li>5.4.1 Laue Method</li>
	<li>5.4.2 Symmetry in Laue Photographs</li>
	<li>5.4.3 Laue Method and Synchrotron Radiation</li>
	<li>5.4.4 Oscillation Method</li>
</ul>
	<li>5.5 Measurement of the Intensities of Diffraction Data</li>
<ul>	<li>5.5.1 Single Counter or Serial Diffractometers</li>
</ul>
	<li>5.6 Single-Crystal X-Ray Diffractometry</li>
<ul>	<li>5.6.1 Instrument Geometry</li>
	<li>5.6.2 Rotation of the Crystal into a Diffracting Position</li>
	<li>5.6.3 Transformation from Miller Indices to Diffractometer Angles</li>
	<li>5.6.4 Data Collection</li>
	<li>5.6.5 Scanning Over a Peak: omega/theta Versus omega Scans</li>
</ul>
	<li>5.7 Area Detectors (Position-Sensitive Detectors)</li>
<ul>	<li>5.7.1 Multiwire Proportional Counter</li>
	<li>5.7.2 FAST Area Detector (Enraf-Nonius FAST) [3]</li>
	<li>5.7.3 Image Plate</li>
	<li>5.7.4 Charge-Coupled Device Area Detectors</li>
	<li>5.7.5 The Tiled CCD</li>
	<li>5.7.6 Charge-Coupled Device Including Tiled CCD Versus Image Plate</li>
	<li>5.7.7 Data Collection Strategies</li>
	<li>5.7.8 The CMOS Detector, Pilatus 1M Detector System, and Continuous Rotation</li>
	<li>5.7.9 Data Processing Software</li>
	<li>5.7.10 Detectors and Diffractometers</li>
	<li>5.7.11 Other Diffractometer Systems</li>
</ul>
	<li>5.8 Monochromators</li>
<ul>	<li>5.8.1 Single-Type Crystal Monochromators</li>
	<li>5.8.2 Double-Type Crystal Monochromators</li>
	<li>5.8.3 Monochromators for Synchrotron Radiation</li>
</ul>
	<li>5.9 Focusing Mirrors</li>
	<li>5.10 Twinning</li>
<ul>	<li>5.10.1 Morphology of Twinning</li>
	<li>5.10.2 Twinning and X-Ray Diffraction</li>
</ul>
	<li>5.11 Problems</li>
	<li>References</li>
<ul>	<li>Bibliography: Crystal Optics</li>
	<li>X-Ray Scattering and Reciprocal Lattice</li>
	<li>Interpretation of X-Ray Diffraction Photographs</li>
	<li>Precession Method</li>
	<li>Diffractometry</li>
	<li>Cryo-Crystallography</li>
	<li>Macromolecular Crystallography</li>
	<li>Data Collection</li>
	<li>Data Collection Services</li>
</ul>
</ul>
	<li>6: Fourier Series and Fourier Transforms&#13;</li>
<ul>	<li>6.1 Image Formation and Focusing</li>
	<li>6.2 Fourier Series</li>
<ul>	<li>6.2.1 Analysis of the Square Wave</li>
	<li>6.2.2 Exponential Forms of Fourier Series</li>
</ul>
	<li>6.3 Fourier Series in X-Ray Crystallography</li>
<ul>	<li>6.3.1 One-Dimensional Function</li>
	<li>6.3.2 Two- and Three-Dimensional Functions</li>
	<li>6.3.3 Units of Electron Density</li>
</ul>
	<li>6.4 Holes and Atoms</li>
	<li>6.5 Generalized Fourier Transform</li>
<ul>	<li>6.5.1 Fourier Transform of a Molecule</li>
	<li>6.5.2 Fourier Transform of a Unit Cell</li>
</ul>
	<li>6.6 Practice with Transforms</li>
<ul>	<li>6.6.1 Optical Diffractometer</li>
	<li>6.6.2 Single Hole</li>
	<li>6.6.3 Two or More Holes</li>
	<li>6.6.4 Change of Origin</li>
	<li>6.6.5 Systematic Absences</li>
	<li>6.6.6 Reconstruction of the Image</li>
	<li>6.6.7 Transforms and Inverse Transforms</li>
	<li>6.6.8 Delta Function</li>
	<li>6.6.9 Weighted Reciprocal Lattice</li>
</ul>
	<li>6.7 Some General Properties of Transforms</li>
	<li>6.8 Convolution</li>
<ul>	<li>6.8.1 Convolution and Diffraction</li>
	<li>6.8.2 Convolution Integral</li>
	<li>6.8.3 Convolution and Crystal Structure</li>
</ul>
	<li>6.9 Structure Solution in Brief</li>
<ul>	<li>6.9.1 Use of Heavy Atoms</li>
	<li>6.9.2 General Phase-Free Transform: Patterson Function</li>
	<li>6.9.3 Sign Relationships</li>
</ul>
	<li>6.10 Problems</li>
	<li>References</li>
<ul>	<li>Bibliography</li>
</ul>
</ul>
	<li>7:&#13;Fourier Techniques in X-Ray Structure Determination</li>
<ul>	<li>7.1 Introduction</li>
	<li>7.2 Analysis of the Unit-Cell Contents</li>
<ul>	<li>7.2.1 Papaverine Hydrochloride, C20H21NO4HCl</li>
	<li>7.2.2 Naphthalene, C10H8</li>
	<li>7.2.3 Molecular Symmetry</li>
	<li>7.2.4 Special Positions</li>
	<li>7.2.5 Nickel Tungstate, NiWO4</li>
</ul>
	<li>7.3 Interpretation of Electron Density Distributions</li>
<ul>	<li>7.3.1 Peak Heights and Weights</li>
	<li>7.3.2 Computation and Display of Electron Density Distributions</li>
	<li>7.3.3 Projections</li>
</ul>
	<li>7.4 Methods of Solving the Phase Problem</li>
<ul>	<li>7.4.1 Number of Reflections in the Data Set</li>
	<li>7.4.2 The Patterson Function</li>
	<li>7.4.3 Positions and Weights of Peaks in the Patterson Function</li>
	<li>7.4.4 Sharpened Patterson Function</li>
	<li>7.4.5 Symmetry of the Patterson Function for a Crystal of Space Group Pm</li>
	<li>7.4.6 Vector Interactions in Other Space Groups</li>
	<li>7.4.7 Examples of the Use of the Patterson Function in Solving the Phase Problem</li>
	<li>7.4.8 Determination of the Chlorine Atom Positions in Papaverine Hydrochloride [1]</li>
	<li>7.4.9 Determination of the Mercury Atom Positions in KHg2</li>
</ul>
	<li>7.5 Heavy-Atom Method and Partial Fourier Synthesis</li>
<ul>	<li>7.5.1 Reliability Factor</li>
	<li>7.5.2 Pseudosymmetry in Electron Density Maps</li>
	<li>7.5.3 Successive Fourier Refinement</li>
	<li>7.5.4 Difference-Fourier Synthesis</li>
	<li>7.5.5 Limitations of the Heavy-Atom Method</li>
	<li>7.5.6 Patterson Selection</li>
	<li>7.5.7 Isomorphous Replacement</li>
	<li>7.5.8 Further Details of the Isomorphous Replacement Phasing Procedure</li>
</ul>
	<li>7.6 Anomalous Scattering</li>
<ul>	<li>7.6.1 The Flack x Parameter</li>
	<li>7.6.2 Effect of Anomalous Scattering on the Symmetry of Diffraction Patterns</li>
	<li>7.6.3 Form of the Structure Factor for a Structure Composed of Heavy-Atom Anomalous Scattering Species</li>
	<li>7.6.4 Phasing by Use of Anomalous Scattering</li>
	<li>7.6.5 Resolution of the Phase Problem for Proteins Using Anomalous Scattering Measurements (SIRAS Method)</li>
	<li>7.6.6 Protein Phasing Using the Multiple-Wavelength Anomalous Dispersion Technique (MAD) with Synchrotron Radiation (SR)</li>
</ul>
	<li>7.7 Charge Flipping</li>
	<li>7.8 Location of Hydrogen Atoms</li>
	<li>7.9 Problems</li>
	<li>References</li>
<ul>	<li>Bibliography: Introductory Structure Analysis</li>
	<li>General Structure Analysis</li>
	<li>Protein Crystallography</li>
	<li>Crystallographic Computing</li>
	<li>Chemical Data</li>
</ul>
</ul>
	<li>8: Direct Methods and Refinement&#13;</li>
<ul>	<li>8.1 Introduction</li>
	<li>8.2 Direct Methods of Phase Determination</li>
<ul>	<li>8.2.1 Normalized Structure Factors</li>
	<li>8.2.2 Structure Invariants and Origin-Fixing Reflections</li>
	<li>8.2.3 Sign Determination: Centrosymmetric Crystals</li>
	<li>8.2.4 Amplitude Symmetry and Phase Symmetry</li>
	<li>8.2.5 2-Listing</li>
	<li>8.2.6 Symbolic-Addition Procedure: Example</li>
	<li>8.2.7 Calculation of E Maps</li>
	<li>8.2.8 Phase Determination: Non-centrosymmetric Crystals</li>
	<li>8.2.9 Enantiomorph Selection</li>
	<li>8.2.10 Phase Determination in Space Group P21</li>
	<li>8.2.11 Advantages and Disadvantages of Symbolic Addition</li>
	<li>8.2.12 Signs of Trouble, and Past Remedies When the Structure Failed to Solve</li>
	<li>8.2.13 Triplets, Quartets, and the SHELX Program Strategy [15]</li>
	<li>8.2.14 The SHELX Computer Program System</li>
	<li>8.2.15 The WinGX Program System</li>
	<li>8.2.16 Direct Methods in the Program SHELX-97 for Small Molecules</li>
	<li>8.2.17 Example of a SHELX-97 Structure Solution: Crystal Code Name BW202W92(R)</li>
</ul>
	<li>8.3 Patterson Search Methods</li>
<ul>	<li>8.3.1 General Comments for Small Molecules and Macromolecules</li>
	<li>8.3.2 Intramolecular Interatomic Vectors and Molecular Orientation</li>
	<li>8.3.3 Intermolecular Interatomic Vectors: Translation Stage of MR</li>
	<li>8.3.4 Crystal Packing and Refinement of the Structure</li>
	<li>8.3.5 Patterson Search Methods for Small Molecules</li>
	<li>8.3.6 The Program PATSEE</li>
	<li>8.3.7 Examples of Structure Solution Using PATSEE</li>
	<li>8.3.8 Shake and Bake</li>
</ul>
	<li>8.4 Least-Squares Refinement</li>
<ul>	<li>8.4.1 Unit-Cell Dimensions</li>
	<li>8.4.2 Least-Squares Parameters</li>
	<li>8.4.3 Theory of Least-Squares Refinement and Strategies to Use</li>
	<li>8.4.4 Least-Squares Refinement Against Fo2</li>
	<li>8.4.5 Constraints and Restraints</li>
</ul>
	<li>8.5 Molecular Geometry</li>
<ul>	<li>8.5.1 Bond Lengths and Angles</li>
	<li>8.5.2 Torsion Angles</li>
	<li>8.5.3 Conformational Analysis</li>
	<li>8.5.4 Mean Planes</li>
</ul>
	<li>8.6 Precision</li>
	<li>8.7 Correctness of a Structure Analysis</li>
<ul>	<li>8.7.1 Databases</li>
</ul>
	<li>8.8 Limitations of X-Ray Structure Analysis</li>
	<li>8.9 Disorder in Single Crystals</li>
	<li>8.10 Computer Prediction of Crystal Structures</li>
<ul>	<li>8.10.1 Crystal Structure of 5-Azauracil [58]</li>
	<li>8.10.2 Developments in Computer Crystal Structure Prediction</li>
</ul>
	<li>8.11 Blind Structure Prediction of the Flexible Molecule 1-Benzyl-1H-Tetrazole</li>
	<li>8.12 Problems</li>
	<li>References</li>
<ul>	<li>Bibliography: Crystallographic Computing</li>
	<li>Direct Methods</li>
	<li>Chemical Data</li>
</ul>
</ul>
	<li>9:Examples of Crystal Structure Determination&#13;</li>
<ul>	<li>9.1 Introduction</li>
	<li>9.2 Crystal Structure of 2-Bromobenzo[b] Indeno[1,2-e] Pyran [1]</li>
<ul>	<li>9.2.1 Preliminary Physical and X-Ray Measurements</li>
	<li>9.2.2 Intensity Measurement and Correction</li>
	<li>9.2.3 Structure Analysis in the xz Projection</li>
	<li>9.2.4 Three-Dimensional Structure Determination</li>
	<li>9.2.5 Refinement</li>
	<li>9.2.6 Molecular Geometry</li>
</ul>
	<li>9.3 Crystal Structure of Potassium 2-Hydroxy-3,4-Dioxocyclobut-1-ene-1-Olate Monohydrate (KHSQ) [3]</li>
<ul>	<li>9.3.1 Preliminary X-Ray and Physical Measurements</li>
	<li>9.3.2 Intensity Measurement and Correction</li>
	<li>9.3.3 2-Listing</li>
	<li>9.3.4 Specifying the Origin</li>
	<li>9.3.5 Sign Determination</li>
	<li>9.3.6 The E Map</li>
	<li>9.3.7 Completion and Refinement of the Structure</li>
</ul>
	<li>9.4 Crystal and Molecular Structure and Absolute Configuration of 3beta-Acetoxy-6,7-Epidithio-19-Norlanosta-5,7,9,11-Tetraene ...</li>
<ul>	<li>9.4.1 Preparation and Preliminary Optical and X-Ray Examinations</li>
	<li>9.4.2 X-Ray Measurement of the Unit-Cell Dimensions and Intensities</li>
	<li>9.4.3 Structure Determination and Refinement</li>
	<li>9.4.4 Absolute Configuration</li>
</ul>
	<li>9.5 Discussion of the Structure</li>
	<li>9.6 Some Remarks on X-Ray Structure Determination</li>
	<li>9.7 Biomolecular Modeling: Bioinformatics</li>
	<li>9.8 Docking Oligomycin into ATP Synthase: Ligand and Receptor</li>
<ul>	<li>9.8.1 Why Modeling Studies?</li>
</ul>
	<li>9.9 X-Ray Structures and Absolute Configurations of the Antibiotics Oligomycins A, B, and C: Inhibitors of ATP Synthase</li>
<ul>	<li>9.9.1 Summary</li>
	<li>9.9.2 Background</li>
	<li>9.9.3 Experimental</li>
	<li>9.9.4 Structure Determination and Refinement</li>
	<li>9.9.5 Results</li>
	<li>9.9.6 Discussion</li>
	<li>9.9.7 Conformational Variations in the Macrocyclic Structures</li>
</ul>
	<li>Conclusions</li>
	<li>9.10 Structure of ATP Synthase (ATPase): The Receptor</li>
	<li>9.11 Docking Oligomycin into ATPase</li>
<ul>	<li>9.11.1 ATP Synthase FO Model</li>
	<li>9.11.2 Homology Modeling</li>
	<li>9.11.3 Refining the Model: Energy Minimization</li>
	<li>9.11.4 Creation of a Pocket for Docking Oligomycin into the ATP Synthase FO</li>
</ul>
	<li>Conclusions</li>
	<li>9.12 Problems</li>
	<li>References</li>
<ul>	<li>Bibliography: Published Structure Analyses</li>
	<li>General Structural Data</li>
	<li>Molecular Mechanics</li>
</ul>
</ul>
	<li>10:&#13;Proteins and Macromolecular X-Ray Analysis</li>
<ul>	<li>10.1 Introduction</li>
<ul>	<li>10.1.1 What Is a Protein?</li>
</ul>
	<li>10.2 Crystallization of Proteins and Complexes for X-Ray Analysis</li>
<ul>	<li>10.2.1 Introduction</li>
	<li>10.2.2 Crystallization Conditions for Macromolecules</li>
	<li>10.2.3 Properties of Protein Crystals</li>
	<li>10.2.4 Crystallization of Proteins</li>
	<li>10.2.5 Molecular Purity</li>
	<li>10.2.6 Practical Considerations</li>
	<li>10.2.7 Batch Crystallization</li>
	<li>10.2.8 Microbatch Screening</li>
	<li>10.2.9 Vapor Diffusion Techniques</li>
	<li>10.2.10  Co-crystallization</li>
	<li>10.2.11  How to Improve the Crystals</li>
	<li>10.2.12  Heavy-Atom Derivatives for MIR</li>
	<li>10.2.13  Protein Complex Crystals with Small Molecules</li>
</ul>
	<li>10.3 Crystal Mounting for X-Ray Data Collection</li>
<ul>	<li>10.3.1 Mounting at Room Temperature</li>
	<li>10.3.2 Cryo-Crystallography</li>
</ul>
	<li>10.4 Macromolecular Crystallography</li>
<ul>	<li>10.4.1 Space Groups</li>
	<li>10.4.2 X-Ray Diffraction from Macromolecular Crystals</li>
	<li>10.4.3 Recording X-Ray Diffraction from Macromolecular Crystals</li>
	<li>10.4.4 Measurement of X-Ray Diffraction from Macromolecular Crystals</li>
	<li>10.4.5 Problems with Data Collection and Suggested Cures</li>
	<li>10.4.6 Preliminary Structure Determination: Unit Cell and Symmetry</li>
	<li>10.4.7 Ricin Agglutinin</li>
</ul>
	<li>10.5 Types of Fourier Synthesis for Protein Analysis</li>
<ul>	<li>10.5.1 Reconstruction of the Molecular Structure</li>
	<li>10.5.2 Difference Electron Density</li>
	<li>10.5.3 The 2Fo(hkl)--Fc(hkl) Map</li>
</ul>
	<li>10.6 Determination of the Phases for Protein Crystals</li>
<ul>	<li>10.6.1 Introduction</li>
	<li>10.6.2 Isomorphous Replacement (MIR)</li>
	<li>10.6.3 Preparation and Screening of Heavy-Atom Derivatives</li>
	<li>10.6.4 Molecular Replacement (MR)</li>
	<li>10.6.5 Example of a Self-Rotation Function: Ricin Agglutinin</li>
	<li>10.6.6 Molecular Replacement in Practice</li>
	<li>10.6.7 Application of the AmoRe Algorithms to Ricin Agglutinin</li>
</ul>
	<li>10.7 SIRAS and MAD Phasing</li>
	<li>10.8 Use of Phase Information and Density Modification</li>
<ul>	<li>10.8.1 Properties of rho(xyz) for Proteins</li>
	<li>10.8.2 Programs for Density Modification</li>
	<li>10.8.3 Preparing to Refine the Structure</li>
</ul>
	<li>10.9 Macromolecular Structure Refinement and Solvent and Ligand Fitting</li>
<ul>	<li>10.9.1 Refinement Techniques</li>
	<li>10.9.2 Simulated Annealing</li>
	<li>10.9.3 Least-Squares Refinement: Constrained, Restrained and Other Protocols</li>
</ul>
	<li>10.10 Structure Validation: Final Checks</li>
<ul>	<li>10.10.1 R-Factors</li>
	<li>10.10.2 Evaluation of Errors</li>
</ul>
	<li>10.11 Geometry Validation: Final Checks</li>
<ul>	<li>10.11.1 Bond Lengths, Bond Angles, Planarity, and Chirality</li>
	<li>10.11.2 Conformation</li>
</ul>
	<li>10.12 Humidity Control and the Use of Cryoprotectants in Protein Crystallography</li>
	<li>10.13 Problems</li>
	<li>References</li>
<ul>	<li>Bibliography</li>
</ul>
</ul>
	<li>11:&#13;Neutron Diffraction from Single Crystals</li>
<ul>	<li>11.1 Introduction</li>
<ul>	<li>11.1.1 Refinement of Hydrogen Atom Positions</li>
</ul>
	<li>11.2 Neutrons, Neutron Sources, and Data Collection</li>
<ul>	<li>11.2.1 Neutrons</li>
	<li>11.2.2 Neutron Sources</li>
	<li>11.2.3 Neutron Data Collection</li>
	<li>11.2.4 Thermal Neutrons</li>
</ul>
	<li>11.3 Neutron Scattering</li>
<ul>	<li>11.3.1 Neutron Scattering Lengths</li>
</ul>
	<li>11.4 Experimental Neutron Diffraction Data Collection</li>
<ul>	<li>11.4.1 LADI-III and VIVALDI at ILL, Grenoble</li>
	<li>11.4.2 Oak Ridge National Laboratory (ORNL) [14]</li>
	<li>11.4.3 Other Neutron Sources</li>
</ul>
	<li>11.5 Deuteration and Perdeuteration</li>
	<li>11.6 Examples of Structure Determination by Neutron Crystallography</li>
	<li>11.7 X-Ray and Neutron Structure of 1,8-(3,6,9-Trioxaundecane-1,11-diyldioxy)-9,10-dihydro-10-10 dimethylanthracene-9-ol [21]</li>
<ul>	<li>11.7.1 Experimental</li>
	<li>11.7.2 Structure Analysis and Refinement</li>
	<li>11.7.3 Discussion of the Structure</li>
	<li>11.7.4 Hydrogen Bonding</li>
</ul>
	<li>11.8 The Pointless Program in CCP4</li>
	<li>11.9 Determination of the Positions of the Deuterium Atoms of the Bound Water Molecules in the Lectin Protein Concanavalin A b...</li>
<ul>	<li>11.9.1 Introduction</li>
	<li>11.9.2 Deuteration of the Concanavalin A Crystals</li>
	<li>11.9.3 Data Collection and Analysis</li>
	<li>11.9.4 X-Ray Model Refinement</li>
	<li>11.9.5 Neutron Structure Refinement</li>
	<li>11.9.6 The Bound Water Structure</li>
	<li>11.9.7 The Metal Sites</li>
	<li>11.9.8 The Saccharide Binding Site</li>
	<li>11.9.9 Conclusion</li>
</ul>
	<li>11.10 The Neutron Structure of the Formyl Peptide Receptor Antagonist Cyclosporin H (CsH) Unambiguously Determines the Solvent...</li>
<ul>	<li>11.10.1 Introduction</li>
	<li>11.10.2 Experimental</li>
	<li>11.10.3 Structure Refinement</li>
	<li>11.10.4 Description of the Neutron Structure and Comparison with the X-Ray Structure</li>
	<li>11.10.5 Conclusion</li>
</ul>
	<li>11.11 Problems</li>
	<li>References</li>
<ul>	<li>Bibliography: Neutron Diffraction</li>
	<li>Structure of Proteins</li>
	<li>Proteins</li>
</ul>
</ul>
	<li>12:&#13;Powder Diffraction</li>
<ul>	<li>12.1 Introduction</li>
<ul>	<li>12.1.1 Identification</li>
	<li>12.1.2 Crystallinity: Size and Strain Broadening</li>
	<li>12.1.3 Unit-Cell Parameters</li>
	<li>12.1.4 Expansion Properties</li>
	<li>12.1.5 Phase Transitions and Alloy Systems</li>
</ul>
	<li>12.2 Crystal Structure Analysis with Powders</li>
<ul>	<li>12.2.1 Crystal Structure Determination Scheme</li>
</ul>
	<li>12.3 Basis of the Powder Method</li>
	<li>12.4 Data Collection</li>
<ul>	<li>12.4.1 Guinier-Type Cameras</li>
	<li>12.4.2 Image Plate Camera</li>
	<li>12.4.3 Powder Diffractometers</li>
	<li>12.4.4 Diffractometry at a Neutron Source</li>
</ul>
	<li>12.5 Indexing Powder Patterns</li>
<ul>	<li>12.5.1 General Indexing</li>
	<li>12.5.2 Reduced and Conventional Unit Cells</li>
	<li>12.5.3 Computer Indexing of the Diffraction Pattern</li>
</ul>
	<li>12.6 Extracting Integrated Intensities from a Powder Pattern</li>
	<li>12.7 The Rietveld Procedure</li>
<ul>	<li>12.7.1 The Le Bail Method</li>
	<li>12.7.2 The Pawley Method</li>
</ul>
	<li>12.8 Examples of Solved Structures</li>
<ul>	<li>12.8.1 Traditional Methods</li>
	<li>12.8.2 SIR Program System</li>
	<li>12.8.3 EXPO Program System</li>
</ul>
	<li>12.9 Direct-Space Methods</li>
<ul>	<li>12.9.1 Zeolites and the FOCUS Algorithm</li>
	<li>12.9.2 Zinc-Silicate Complex VIP-9</li>
</ul>
	<li>12.10 Monte Carlo Method</li>
<ul>	<li>12.10.1 Simulated Annealing</li>
</ul>
	<li>12.11 ESPOIR Program System</li>
	<li>12.12 Powder Diffraction with Proteins</li>
<ul>	<li>12.12.1 T3R3 Zinc-Insulin Complex</li>
</ul>
	<li>12.13 Maximum Entropy in Crystal Structure Analysis</li>
<ul>	<li>12.13.1 Most Probable Distribution</li>
	<li>12.13.2 Electron Density Map</li>
</ul>
	<li>12.14 Log-Likelihood Gain in the Phase Problem</li>
<ul>	<li>12.14.1 Basis Set and Expansion of Reflections</li>
	<li>12.14.2 Log-Likelihood Gain</li>
	<li>12.14.3 Centroid Maps</li>
</ul>
	<li>12.15 Genetic Algorithms</li>
	<li>12.16 Energy Minimization Techniques</li>
	<li>12.17 Concluding Remarks</li>
	<li>12.18 Problems</li>
	<li>References</li>
<ul>	<li>Bibliography</li>
</ul>
</ul>
	<li>13:&#13;Computer-Aided Crystallography</li>
<ul>	<li>13.1 Introduction</li>
<ul>	<li>13.1.1 Collaborative Computational Projects</li>
	<li>13.1.2 Structure of the Web Program Packages</li>
</ul>
	<li>13.2 Derivation of Point Groups (EULR)</li>
	<li>13.3 Point-Group Recognition (SYMM)</li>
	<li>13.4 Structure Determination Simulation (XRAY)</li>
<ul>	<li>13.4.1 Patterson Function</li>
	<li>13.4.2 Superposition Function</li>
	<li>13.4.3 Structure Factor Calculation</li>
	<li>13.4.4 Least-Squares Refinement</li>
	<li>13.4.5 Electron Density Maps</li>
	<li>13.4.6 Direct Methods: Calculation of |E| Values</li>
	<li>13.4.7 Calculation of E Maps</li>
	<li>13.4.8 Bond Lengths and Bond Angles</li>
	<li>13.4.9 Scale and Temperature Factors by Wilson&acute;s Method</li>
	<li>13.4.10  |E| Values Calculated from the Structure</li>
</ul>
	<li>13.5 Crystal Structure Analysis Problems</li>
<ul>	<li>13.5.1 Ni o-Phenanthroline Complex (NIOP)</li>
	<li>13.5.2 2-Amino-4,6-dichloropyrimidine (CL1P)</li>
	<li>13.5.3 2-Amino-4-methyl-6-chloropyrimidine (CL2P)</li>
	<li>13.5.4 m-Tolidine Dihydrochloride (MTOL)</li>
	<li>13.5.5 Nitroguanidine (NO2G)</li>
	<li>13.5.6 Bis(6-sulfanyloxy-1,3,5-triazin-2(1H)-one) (BSTO)</li>
	<li>13.5.7 2-S-methylthiouracil (SMTX and SMTY)</li>
</ul>
	<li>13.6 General Crystal Structure and Other Programs</li>
<ul>	<li>13.6.1 One-Dimensional Fourier Summation (FOUR1D)</li>
	<li>13.6.2 Two-Dimensional Fourier Summation (FOUR2D)</li>
	<li>13.6.3 One-Dimensional Fourier Transform (TRANS1)</li>
	<li>13.6.4 Reciprocal Unit Cell (RECIP)</li>
	<li>13.6.5 Molecular Geometry (MOLGOM)</li>
	<li>13.6.6 Internal and Cartesian Coordinates (INTXYZ)</li>
	<li>13.6.7 Linear Least Squares (LSLI)</li>
	<li>13.6.8 Matrix Operations (MATOPS)</li>
	<li>13.6.9 Q Values (QVALS)</li>
	<li>13.6.10 Le Page Unit-Cell Reduction (LEPAGE)</li>
	<li>13.6.11 Zone symbols/Miller indices (ZONE)</li>
</ul>
	<li>13.7 Automatic Powder Indexing: ITO12</li>
	<li>13.8 Automatic Powder Structure Solving: ESPOIR</li>
<ul>	<li>13.8.1 Aragonite</li>
	<li>13.8.2 α-Alumina (Corundum)</li>
</ul>
	<li>13.9 Problems&#13;</li>
	<li>References</li>
</ul>
	<li>Appendix A: Stereoviews&#13;and Crystal Models</li>
<ul>	<li>A.1 Stereoviews</li>
	<li>A.2 Model of a Tetragonal Crystal</li>
	<li>B.1 Alternating Axis of Symmetry</li>
	<li>B.2 Symmetry Notations</li>
	<li>C.1 Cartesian to Crystallographic Transformation and Its Inverse</li>
	<li>D.1 Single Crystal Suites</li>
	<li>D.2 Single Crystal Structure Solution Programs</li>
	<li>D.3 Single Crystal Twinning Software</li>
	<li>D.4 Freestanding Structure Visualization Software</li>
	<li>D.5 Powder Diffraction Data: Powder Indexing Suites (Dedicated and Other)</li>
	<li>D.6 Powder Pattern Decomposition</li>
	<li>D.7 Structure Solution from Powder Diffraction Data</li>
	<li>D.8 Software for Macromolecular Crystallography</li>
<ul>	<li>D.8.1 Data Processing</li>
	<li>D.8.2 Fourier and Structure Factor Calculations</li>
	<li>D.8.3 Molecular Replacement</li>
	<li>D.8.4 Schematic Structure Plots</li>
	<li>D.8.5 Software for Packing, Molecular Geometry, Validation and Deposition</li>
	<li>D.8.6 Software for Graphics and Model Building</li>
	<li>D.8.7 Software for Molecular Graphics and Display</li>
	<li>D.8.8 Software for Refinement</li>
	<li>D.8.9 Software for Molecular Dynamics and Energy Minimization</li>
	<li>D.8.10 Data Bases</li>
	<li>D.8.11 Synchrotron Web Page</li>
</ul>
	<li>D.9 Bioinformatics</li>
<ul>	<li>D.9.1 Molecular Modelling Software</li>
	<li>D.9.2 External Links</li>
	<li>D.9.3 Useful Homepages</li>
</ul>
	<li>E.1 Structure Invariants</li>
	<li>E.2 Structure Seminvariants</li>
<ul>	<li>E.2.1 Difference Between Structure Invariant and Structure Seminvariant</li>
</ul>
	<li>E.3 Origin Specification</li>
	<li>E.4 Choice of Enantiomorph</li>
</ul>
	<li>Tutorial Solutions</li>
<ul>	<li>Solutions 1</li>
	<li>Solutions 2</li>
	<li>Solutions 3</li>
	<li>Solutions 4</li>
	<li>Solutions 5</li>
	<li>Solutions 6</li>
	<li>Solutions 7</li>
	<li>Solutions 8</li>
	<li>Solutions 9</li>
	<li>Solutions 10</li>
	<li>Solutions 11</li>
	<li>Solutions 12</li>
	<li>Solutions 13</li>
</ul>
	<li>Index</li>
</ul>
</body></html>