<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<title>Untitled</title>
</head>
<body><div class="page"><p/>
</div>
<div class="page"><p/>
<p>Springer Series in Statistics
</p>
<p>Advisors:
</p>
<p>P. Bickel, P. Diggle, S. Fienberg, U. Gather,
</p>
<p>I. Olkin, S. Zeger</p>
<p/>
</div>
<div class="page"><p/>
<p>Springer Series in Statistics
</p>
<p>For other titles published in this series, go to
</p>
<p>http://www.springer.com/series/692</p>
<p/>
</div>
<div class="page"><p/>
<p>Trevor Hastie
</p>
<p>Robert Tibshirani
</p>
<p>Jerome Friedman
</p>
<p>Data Mining, Inference, and Prediction
</p>
<p>The Elements of Statistical
</p>
<p>Second Edition
</p>
<p>Learning</p>
<p/>
</div>
<div class="page"><p/>
<p>c
</p>
<p>All rights reserved. This work may not be translated or copied in whole or in part without the written
permission of the publisher (Springer Science+Business Media, LLC, 233 Spring Street, New York, NY
10013, USA), except for brief excerpts in connection with reviews or scholarly analysis. Use in connection
with any form of information storage and retrieval, electronic adaptation, computer software, or by similar
or dissimilar methodology now known or hereafter developed is forbidden.
The use in this publication of trade names, trademarks, service marks, and similar terms, even if they are
not identified as such, is not to be taken as an expression of opinion as to whether or not they are subject
to proprietary rights.
</p>
<p>springer.com
</p>
<p>Trevor Hastie
Stanford University
Dept. of Statistics
Stanford CA 94305
USA
</p>
<p>Robert Tibshirani
Stanford University
Dept. of Statistics
Stanford CA 94305
</p>
<p>Jerome Friedman
Stanford University
Dept. of Statistics
Stanford CA 94305
</p>
<p>USA
</p>
<p>Library of Congress Control Number: 2008941148
</p>
<p>USA
hastie@stanford.edu tibs@stanford.edu
</p>
<p>jhf@stanford.edu
</p>
<p>ISSN: 0172-7397
e-ISBN: 978-0-387-84858-7ISBN: 978-0-387-84857-0
</p>
<p>&copy; Springer Science+Business Media, LLC 2009, Corrected at 12th printing 2017
</p>
<p>DOI: 10.1007/b94608
</p>
<p>Printed on acid-free paper</p>
<p/>
</div>
<div class="page"><p/>
<p>To our parents:
</p>
<p>Valerie and Patrick Hastie
</p>
<p>Vera and Sami Tibshirani
</p>
<p>Florence and Harry Friedman
</p>
<p>and to our families:
</p>
<p>Samantha, Timothy, and Lynda
</p>
<p>Charlie, Ryan, Julie, and Cheryl
</p>
<p>Melanie, Dora, Monika, and Ildiko</p>
<p/>
</div>
<div class="page"><p/>
<p>Preface to the Second Edition
</p>
<p>In God we trust, all others bring data.
</p>
<p>&ndash;William Edwards Deming (1900-1993)1
</p>
<p>We have been gratified by the popularity of the first edition of The
Elements of Statistical Learning. This, along with the fast pace of research
in the statistical learning field, motivated us to update our book with a
second edition.
We have added four new chapters and updated some of the existing
</p>
<p>chapters. Because many readers are familiar with the layout of the first
edition, we have tried to change it as little as possible. Here is a summary
of the main changes:
</p>
<p>1On the Web, this quote has been widely attributed to both Deming and Robert W.
Hayden; however Professor Hayden told us that he can claim no credit for this quote,
and ironically we could find no &ldquo;data&rdquo; confirming that Deming actually said this.</p>
<p/>
</div>
<div class="page"><p/>
<p>viii Preface to the Second Edition
</p>
<p>Chapter What&rsquo;s new
</p>
<p>1. Introduction
2. Overview of Supervised Learning
3. Linear Methods for Regression LAR algorithm and generalizations
</p>
<p>of the lasso
4. Linear Methods for Classification Lasso path for logistic regression
5. Basis Expansions and Regulariza-
tion
</p>
<p>Additional illustrations of RKHS
</p>
<p>6. Kernel Smoothing Methods
7. Model Assessment and Selection Strengths and pitfalls of cross-
</p>
<p>validation
8. Model Inference and Averaging
9. Additive Models, Trees, and
Related Methods
10. Boosting and Additive Trees New example from ecology; some
</p>
<p>material split off to Chapter 16.
11. Neural Networks Bayesian neural nets and the NIPS
</p>
<p>2003 challenge
12. Support Vector Machines and
Flexible Discriminants
</p>
<p>Path algorithm for SVM classifier
</p>
<p>13. Prototype Methods and
Nearest-Neighbors
14. Unsupervised Learning Spectral clustering, kernel PCA,
</p>
<p>sparse PCA, non-negative matrix
factorization archetypal analysis,
nonlinear dimension reduction,
Google page rank algorithm, a
direct approach to ICA
</p>
<p>15. Random Forests New
16. Ensemble Learning New
17. Undirected Graphical Models New
18. High-Dimensional Problems New
</p>
<p>Some further notes:
</p>
<p>&bull; Our first edition was unfriendly to colorblind readers; in particular,
we tended to favor red/green contrasts which are particularly trou-
blesome. We have changed the color palette in this edition to a large
extent, replacing the above with an orange/blue contrast.
</p>
<p>&bull; We have changed the name of Chapter 6 from &ldquo;Kernel Methods&rdquo; to
&ldquo;Kernel Smoothing Methods&rdquo;, to avoid confusion with the machine-
learning kernel method that is discussed in the context of support vec-
tor machines (Chapter 12) and more generally in Chapters 5 and 14.
</p>
<p>&bull; In the first edition, the discussion of error-rate estimation in Chap-
ter 7 was sloppy, as we did not clearly differentiate the notions of
conditional error rates (conditional on the training set) and uncondi-
tional rates. We have fixed this in the new edition.</p>
<p/>
</div>
<div class="page"><p/>
<p>Preface to the Second Edition ix
</p>
<p>&bull; Chapters 15 and 16 follow naturally from Chapter 10, and the chap-
ters are probably best read in that order.
</p>
<p>&bull; In Chapter 17, we have not attempted a comprehensive treatment
of graphical models, and discuss only undirected models and some
new methods for their estimation. Due to a lack of space, we have
specifically omitted coverage of directed graphical models.
</p>
<p>&bull; Chapter 18 explores the &ldquo;p ≫ N&rdquo; problem, which is learning in high-
dimensional feature spaces. These problems arise in many areas, in-
cluding genomic and proteomic studies, and document classification.
</p>
<p>We thank the many readers who have found the (too numerous) errors in
the first edition. We apologize for those and have done our best to avoid er-
rors in this new edition. We thank Mark Segal, Bala Rajaratnam, and Larry
Wasserman for comments on some of the new chapters, and many Stanford
graduate and post-doctoral students who offered comments, in particular
Mohammed AlQuraishi, John Boik, Holger Hoefling, Arian Maleki, Donal
McMahon, Saharon Rosset, Babak Shababa, Daniela Witten, Ji Zhu and
Hui Zou. We thank John Kimmel for his patience in guiding us through this
new edition. RT dedicates this edition to the memory of Anna McPhee.
</p>
<p>Trevor Hastie
Robert Tibshirani
Jerome Friedman
</p>
<p>Stanford, California
August 2008</p>
<p/>
</div>
<div class="page"><p/>
<p>Preface to the First Edition
</p>
<p>We are drowning in information and starving for knowledge.
</p>
<p>&ndash;Rutherford D. Roger
</p>
<p>The field of Statistics is constantly challenged by the problems that science
and industry brings to its door. In the early days, these problems often came
from agricultural and industrial experiments and were relatively small in
scope. With the advent of computers and the information age, statistical
problems have exploded both in size and complexity. Challenges in the
areas of data storage, organization and searching have led to the new field
of &ldquo;data mining&rdquo;; statistical and computational problems in biology and
medicine have created &ldquo;bioinformatics.&rdquo; Vast amounts of data are being
generated in many fields, and the statistician&rsquo;s job is to make sense of it
all: to extract important patterns and trends, and understand &ldquo;what the
data says.&rdquo; We call this learning from data.
The challenges in learning from data have led to a revolution in the sta-
</p>
<p>tistical sciences. Since computation plays such a key role, it is not surprising
that much of this new development has been done by researchers in other
fields such as computer science and engineering.
The learning problems that we consider can be roughly categorized as
</p>
<p>either supervised or unsupervised. In supervised learning, the goal is to pre-
dict the value of an outcome measure based on a number of input measures;
in unsupervised learning, there is no outcome measure, and the goal is to
describe the associations and patterns among a set of input measures.</p>
<p/>
</div>
<div class="page"><p/>
<p>xii Preface to the First Edition
</p>
<p>This book is our attempt to bring together many of the important new
ideas in learning, and explain them in a statistical framework. While some
mathematical details are needed, we emphasize the methods and their con-
ceptual underpinnings rather than their theoretical properties. As a result,
we hope that this book will appeal not just to statisticians but also to
researchers and practitioners in a wide variety of fields.
Just as we have learned a great deal from researchers outside of the field
</p>
<p>of statistics, our statistical viewpoint may help others to better understand
different aspects of learning:
</p>
<p>There is no true interpretation of anything; interpretation is a
vehicle in the service of human comprehension. The value of
interpretation is in enabling others to fruitfully think about an
idea.
</p>
<p>&ndash;Andreas Buja
</p>
<p>We would like to acknowledge the contribution of many people to the
conception and completion of this book. David Andrews, Leo Breiman,
Andreas Buja, John Chambers, Bradley Efron, Geoffrey Hinton, Werner
Stuetzle, and John Tukey have greatly influenced our careers. Balasub-
ramanian Narasimhan gave us advice and help on many computational
problems, and maintained an excellent computing environment. Shin-Ho
Bang helped in the production of a number of the figures. Lee Wilkinson
gave valuable tips on color production. Ilana Belitskaya, Eva Cantoni, Maya
Gupta, Michael Jordan, Shanti Gopatam, Radford Neal, Jorge Picazo, Bog-
dan Popescu, Olivier Renaud, Saharon Rosset, John Storey, Ji Zhu, Mu
Zhu, two reviewers and many students read parts of the manuscript and
offered helpful suggestions. John Kimmel was supportive, patient and help-
ful at every phase; MaryAnn Brickner and Frank Ganz headed a superb
production team at Springer. Trevor Hastie would like to thank the statis-
tics department at the University of Cape Town for their hospitality during
the final stages of this book. We gratefully acknowledge NSF and NIH for
their support of this work. Finally, we would like to thank our families and
our parents for their love and support.
</p>
<p>Trevor Hastie
Robert Tibshirani
Jerome Friedman
</p>
<p>Stanford, California
May 2001
</p>
<p>The quiet statisticians have changed our world; not by discov-
ering new facts or technical developments, but by changing the
ways that we reason, experiment and form our opinions ....
</p>
<p>&ndash;Ian Hacking</p>
<p/>
</div>
<div class="page"><p/>
<p>Contents
</p>
<p>Preface to the Second Edition vii
</p>
<p>Preface to the First Edition xi
</p>
<p>1 Introduction 1
</p>
<p>2 Overview of Supervised Learning 9
</p>
<p>2.1 Introduction . . . . . . . . . . . . . . . . . . . . . . . . . 9
2.2 Variable Types and Terminology . . . . . . . . . . . . . . 9
2.3 Two Simple Approaches to Prediction:
</p>
<p>Least Squares and Nearest Neighbors . . . . . . . . . . . 11
2.3.1 Linear Models and Least Squares . . . . . . . . 11
2.3.2 Nearest-Neighbor Methods . . . . . . . . . . . . 14
2.3.3 From Least Squares to Nearest Neighbors . . . . 16
</p>
<p>2.4 Statistical Decision Theory . . . . . . . . . . . . . . . . . 18
2.5 Local Methods in High Dimensions . . . . . . . . . . . . . 22
2.6 Statistical Models, Supervised Learning
</p>
<p>and Function Approximation . . . . . . . . . . . . . . . . 28
2.6.1 A Statistical Model
</p>
<p>for the Joint Distribution Pr(X,Y ) . . . . . . . 28
2.6.2 Supervised Learning . . . . . . . . . . . . . . . . 29
2.6.3 Function Approximation . . . . . . . . . . . . . 29
</p>
<p>2.7 Structured Regression Models . . . . . . . . . . . . . . . 32
2.7.1 Difficulty of the Problem . . . . . . . . . . . . . 32</p>
<p/>
</div>
<div class="page"><p/>
<p>xiv Contents
</p>
<p>2.8 Classes of Restricted Estimators . . . . . . . . . . . . . . 33
2.8.1 Roughness Penalty and Bayesian Methods . . . 34
2.8.2 Kernel Methods and Local Regression . . . . . . 34
2.8.3 Basis Functions and Dictionary Methods . . . . 35
</p>
<p>2.9 Model Selection and the Bias&ndash;Variance Tradeoff . . . . . 37
Bibliographic Notes . . . . . . . . . . . . . . . . . . . . . . . . . 39
Exercises . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 39
</p>
<p>3 Linear Methods for Regression 43
</p>
<p>3.1 Introduction . . . . . . . . . . . . . . . . . . . . . . . . . 43
3.2 Linear Regression Models and Least Squares . . . . . . . 44
</p>
<p>3.2.1 Example: Prostate Cancer . . . . . . . . . . . . 49
3.2.2 The Gauss&ndash;Markov Theorem . . . . . . . . . . . 51
3.2.3 Multiple Regression
</p>
<p>from Simple Univariate Regression . . . . . . . . 52
3.2.4 Multiple Outputs . . . . . . . . . . . . . . . . . 56
</p>
<p>3.3 Subset Selection . . . . . . . . . . . . . . . . . . . . . . . 57
3.3.1 Best-Subset Selection . . . . . . . . . . . . . . . 57
3.3.2 Forward- and Backward-Stepwise Selection . . . 58
3.3.3 Forward-Stagewise Regression . . . . . . . . . . 60
3.3.4 Prostate Cancer Data Example (Continued) . . 61
</p>
<p>3.4 Shrinkage Methods . . . . . . . . . . . . . . . . . . . . . . 61
3.4.1 Ridge Regression . . . . . . . . . . . . . . . . . 61
3.4.2 The Lasso . . . . . . . . . . . . . . . . . . . . . 68
3.4.3 Discussion: Subset Selection, Ridge Regression
</p>
<p>and the Lasso . . . . . . . . . . . . . . . . . . . 69
3.4.4 Least Angle Regression . . . . . . . . . . . . . . 73
</p>
<p>3.5 Methods Using Derived Input Directions . . . . . . . . . 79
3.5.1 Principal Components Regression . . . . . . . . 79
3.5.2 Partial Least Squares . . . . . . . . . . . . . . . 80
</p>
<p>3.6 Discussion: A Comparison of the Selection
and Shrinkage Methods . . . . . . . . . . . . . . . . . . . 82
</p>
<p>3.7 Multiple Outcome Shrinkage and Selection . . . . . . . . 84
3.8 More on the Lasso and Related Path Algorithms . . . . . 86
</p>
<p>3.8.1 Incremental Forward Stagewise Regression . . . 86
3.8.2 Piecewise-Linear Path Algorithms . . . . . . . . 89
3.8.3 The Dantzig Selector . . . . . . . . . . . . . . . 89
3.8.4 The Grouped Lasso . . . . . . . . . . . . . . . . 90
3.8.5 Further Properties of the Lasso . . . . . . . . . . 91
3.8.6 Pathwise Coordinate Optimization . . . . . . . . 92
</p>
<p>3.9 Computational Considerations . . . . . . . . . . . . . . . 93
Bibliographic Notes . . . . . . . . . . . . . . . . . . . . . . . . . 94
Exercises . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 94</p>
<p/>
</div>
<div class="page"><p/>
<p>Contents xv
</p>
<p>4 Linear Methods for Classification 101
</p>
<p>4.1 Introduction . . . . . . . . . . . . . . . . . . . . . . . . . 101
4.2 Linear Regression of an Indicator Matrix . . . . . . . . . 103
4.3 Linear Discriminant Analysis . . . . . . . . . . . . . . . . 106
</p>
<p>4.3.1 Regularized Discriminant Analysis . . . . . . . . 112
4.3.2 Computations for LDA . . . . . . . . . . . . . . 113
4.3.3 Reduced-Rank Linear Discriminant Analysis . . 113
</p>
<p>4.4 Logistic Regression . . . . . . . . . . . . . . . . . . . . . . 119
4.4.1 Fitting Logistic Regression Models . . . . . . . . 120
4.4.2 Example: South African Heart Disease . . . . . 122
4.4.3 Quadratic Approximations and Inference . . . . 124
4.4.4 L1 Regularized Logistic Regression . . . . . . . . 125
4.4.5 Logistic Regression or LDA? . . . . . . . . . . . 127
</p>
<p>4.5 Separating Hyperplanes . . . . . . . . . . . . . . . . . . . 129
4.5.1 Rosenblatt&rsquo;s Perceptron Learning Algorithm . . 130
4.5.2 Optimal Separating Hyperplanes . . . . . . . . . 132
</p>
<p>Bibliographic Notes . . . . . . . . . . . . . . . . . . . . . . . . . 135
Exercises . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 135
</p>
<p>5 Basis Expansions and Regularization 139
</p>
<p>5.1 Introduction . . . . . . . . . . . . . . . . . . . . . . . . . 139
5.2 Piecewise Polynomials and Splines . . . . . . . . . . . . . 141
</p>
<p>5.2.1 Natural Cubic Splines . . . . . . . . . . . . . . . 144
5.2.2 Example: South African Heart Disease (Continued)146
5.2.3 Example: Phoneme Recognition . . . . . . . . . 148
</p>
<p>5.3 Filtering and Feature Extraction . . . . . . . . . . . . . . 150
5.4 Smoothing Splines . . . . . . . . . . . . . . . . . . . . . . 151
</p>
<p>5.4.1 Degrees of Freedom and Smoother Matrices . . . 153
5.5 Automatic Selection of the Smoothing Parameters . . . . 156
</p>
<p>5.5.1 Fixing the Degrees of Freedom . . . . . . . . . . 158
5.5.2 The Bias&ndash;Variance Tradeoff . . . . . . . . . . . . 158
</p>
<p>5.6 Nonparametric Logistic Regression . . . . . . . . . . . . . 161
5.7 Multidimensional Splines . . . . . . . . . . . . . . . . . . 162
5.8 Regularization and Reproducing Kernel Hilbert Spaces . 167
</p>
<p>5.8.1 Spaces of Functions Generated by Kernels . . . 168
5.8.2 Examples of RKHS . . . . . . . . . . . . . . . . 170
</p>
<p>5.9 Wavelet Smoothing . . . . . . . . . . . . . . . . . . . . . 174
5.9.1 Wavelet Bases and the Wavelet Transform . . . 176
5.9.2 Adaptive Wavelet Filtering . . . . . . . . . . . . 179
</p>
<p>Bibliographic Notes . . . . . . . . . . . . . . . . . . . . . . . . . 181
Exercises . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 181
Appendix: Computational Considerations for Splines . . . . . . 186
</p>
<p>Appendix: B-splines . . . . . . . . . . . . . . . . . . . . . 186
Appendix: Computations for Smoothing Splines . . . . . 189</p>
<p/>
</div>
<div class="page"><p/>
<p>xvi Contents
</p>
<p>6 Kernel Smoothing Methods 191
</p>
<p>6.1 One-Dimensional Kernel Smoothers . . . . . . . . . . . . 192
6.1.1 Local Linear Regression . . . . . . . . . . . . . . 194
6.1.2 Local Polynomial Regression . . . . . . . . . . . 197
</p>
<p>6.2 Selecting the Width of the Kernel . . . . . . . . . . . . . 198
6.3 Local Regression in IRp . . . . . . . . . . . . . . . . . . . 200
6.4 Structured Local Regression Models in IRp . . . . . . . . 201
</p>
<p>6.4.1 Structured Kernels . . . . . . . . . . . . . . . . . 203
6.4.2 Structured Regression Functions . . . . . . . . . 203
</p>
<p>6.5 Local Likelihood and Other Models . . . . . . . . . . . . 205
6.6 Kernel Density Estimation and Classification . . . . . . . 208
</p>
<p>6.6.1 Kernel Density Estimation . . . . . . . . . . . . 208
6.6.2 Kernel Density Classification . . . . . . . . . . . 210
6.6.3 The Naive Bayes Classifier . . . . . . . . . . . . 210
</p>
<p>6.7 Radial Basis Functions and Kernels . . . . . . . . . . . . 212
6.8 Mixture Models for Density Estimation and Classification 214
6.9 Computational Considerations . . . . . . . . . . . . . . . 216
Bibliographic Notes . . . . . . . . . . . . . . . . . . . . . . . . . 216
Exercises . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 216
</p>
<p>7 Model Assessment and Selection 219
</p>
<p>7.1 Introduction . . . . . . . . . . . . . . . . . . . . . . . . . 219
7.2 Bias, Variance and Model Complexity . . . . . . . . . . . 219
7.3 The Bias&ndash;Variance Decomposition . . . . . . . . . . . . . 223
</p>
<p>7.3.1 Example: Bias&ndash;Variance Tradeoff . . . . . . . . 226
7.4 Optimism of the Training Error Rate . . . . . . . . . . . 228
7.5 Estimates of In-Sample Prediction Error . . . . . . . . . . 230
7.6 The Effective Number of Parameters . . . . . . . . . . . . 232
7.7 The Bayesian Approach and BIC . . . . . . . . . . . . . . 233
7.8 Minimum Description Length . . . . . . . . . . . . . . . . 235
7.9 Vapnik&ndash;Chervonenkis Dimension . . . . . . . . . . . . . . 237
</p>
<p>7.9.1 Example (Continued) . . . . . . . . . . . . . . . 239
7.10 Cross-Validation . . . . . . . . . . . . . . . . . . . . . . . 241
</p>
<p>7.10.1 K-Fold Cross-Validation . . . . . . . . . . . . . 241
7.10.2 The Wrong and Right Way
</p>
<p>to Do Cross-validation . . . . . . . . . . . . . . . 245
7.10.3 Does Cross-Validation Really Work? . . . . . . . 247
</p>
<p>7.11 Bootstrap Methods . . . . . . . . . . . . . . . . . . . . . 249
7.11.1 Example (Continued) . . . . . . . . . . . . . . . 252
</p>
<p>7.12 Conditional or Expected Test Error? . . . . . . . . . . . . 254
Bibliographic Notes . . . . . . . . . . . . . . . . . . . . . . . . . 257
Exercises . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 257
</p>
<p>8 Model Inference and Averaging 261
</p>
<p>8.1 Introduction . . . . . . . . . . . . . . . . . . . . . . . . . 261</p>
<p/>
</div>
<div class="page"><p/>
<p>Contents xvii
</p>
<p>8.2 The Bootstrap and Maximum Likelihood Methods . . . . 261
8.2.1 A Smoothing Example . . . . . . . . . . . . . . 261
8.2.2 Maximum Likelihood Inference . . . . . . . . . . 265
8.2.3 Bootstrap versus Maximum Likelihood . . . . . 267
</p>
<p>8.3 Bayesian Methods . . . . . . . . . . . . . . . . . . . . . . 267
8.4 Relationship Between the Bootstrap
</p>
<p>and Bayesian Inference . . . . . . . . . . . . . . . . . . . 271
8.5 The EM Algorithm . . . . . . . . . . . . . . . . . . . . . 272
</p>
<p>8.5.1 Two-Component Mixture Model . . . . . . . . . 272
8.5.2 The EM Algorithm in General . . . . . . . . . . 276
8.5.3 EM as a Maximization&ndash;Maximization Procedure 277
</p>
<p>8.6 MCMC for Sampling from the Posterior . . . . . . . . . . 279
8.7 Bagging . . . . . . . . . . . . . . . . . . . . . . . . . . . . 282
</p>
<p>8.7.1 Example: Trees with Simulated Data . . . . . . 283
8.8 Model Averaging and Stacking . . . . . . . . . . . . . . . 288
8.9 Stochastic Search: Bumping . . . . . . . . . . . . . . . . . 290
Bibliographic Notes . . . . . . . . . . . . . . . . . . . . . . . . . 292
Exercises . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 293
</p>
<p>9 Additive Models, Trees, and Related Methods 295
</p>
<p>9.1 Generalized Additive Models . . . . . . . . . . . . . . . . 295
9.1.1 Fitting Additive Models . . . . . . . . . . . . . . 297
9.1.2 Example: Additive Logistic Regression . . . . . 299
9.1.3 Summary . . . . . . . . . . . . . . . . . . . . . . 304
</p>
<p>9.2 Tree-Based Methods . . . . . . . . . . . . . . . . . . . . . 305
9.2.1 Background . . . . . . . . . . . . . . . . . . . . 305
9.2.2 Regression Trees . . . . . . . . . . . . . . . . . . 307
9.2.3 Classification Trees . . . . . . . . . . . . . . . . 308
9.2.4 Other Issues . . . . . . . . . . . . . . . . . . . . 310
9.2.5 Spam Example (Continued) . . . . . . . . . . . 313
</p>
<p>9.3 PRIM: Bump Hunting . . . . . . . . . . . . . . . . . . . . 317
9.3.1 Spam Example (Continued) . . . . . . . . . . . 320
</p>
<p>9.4 MARS: Multivariate Adaptive Regression Splines . . . . . 321
9.4.1 Spam Example (Continued) . . . . . . . . . . . 326
9.4.2 Example (Simulated Data) . . . . . . . . . . . . 327
9.4.3 Other Issues . . . . . . . . . . . . . . . . . . . . 328
</p>
<p>9.5 Hierarchical Mixtures of Experts . . . . . . . . . . . . . . 329
9.6 Missing Data . . . . . . . . . . . . . . . . . . . . . . . . . 332
9.7 Computational Considerations . . . . . . . . . . . . . . . 334
Bibliographic Notes . . . . . . . . . . . . . . . . . . . . . . . . . 334
Exercises . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 335
</p>
<p>10 Boosting and Additive Trees 337
</p>
<p>10.1 Boosting Methods . . . . . . . . . . . . . . . . . . . . . . 337
10.1.1 Outline of This Chapter . . . . . . . . . . . . . . 340</p>
<p/>
</div>
<div class="page"><p/>
<p>xviii Contents
</p>
<p>10.2 Boosting Fits an Additive Model . . . . . . . . . . . . . . 341
10.3 Forward Stagewise Additive Modeling . . . . . . . . . . . 342
10.4 Exponential Loss and AdaBoost . . . . . . . . . . . . . . 343
10.5 Why Exponential Loss? . . . . . . . . . . . . . . . . . . . 345
10.6 Loss Functions and Robustness . . . . . . . . . . . . . . . 346
10.7 &ldquo;Off-the-Shelf&rdquo; Procedures for Data Mining . . . . . . . . 350
10.8 Example: Spam Data . . . . . . . . . . . . . . . . . . . . 352
10.9 Boosting Trees . . . . . . . . . . . . . . . . . . . . . . . . 353
10.10 Numerical Optimization via Gradient Boosting . . . . . . 358
</p>
<p>10.10.1 Steepest Descent . . . . . . . . . . . . . . . . . . 358
10.10.2 Gradient Boosting . . . . . . . . . . . . . . . . . 359
10.10.3 Implementations of Gradient Boosting . . . . . . 360
</p>
<p>10.11 Right-Sized Trees for Boosting . . . . . . . . . . . . . . . 361
10.12 Regularization . . . . . . . . . . . . . . . . . . . . . . . . 364
</p>
<p>10.12.1 Shrinkage . . . . . . . . . . . . . . . . . . . . . . 364
10.12.2 Subsampling . . . . . . . . . . . . . . . . . . . . 365
</p>
<p>10.13 Interpretation . . . . . . . . . . . . . . . . . . . . . . . . 367
10.13.1 Relative Importance of Predictor Variables . . . 367
10.13.2 Partial Dependence Plots . . . . . . . . . . . . . 369
</p>
<p>10.14 Illustrations . . . . . . . . . . . . . . . . . . . . . . . . . . 371
10.14.1 California Housing . . . . . . . . . . . . . . . . . 371
10.14.2 New Zealand Fish . . . . . . . . . . . . . . . . . 375
10.14.3 Demographics Data . . . . . . . . . . . . . . . . 379
</p>
<p>Bibliographic Notes . . . . . . . . . . . . . . . . . . . . . . . . . 380
Exercises . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 384
</p>
<p>11 Neural Networks 389
</p>
<p>11.1 Introduction . . . . . . . . . . . . . . . . . . . . . . . . . 389
11.2 Projection Pursuit Regression . . . . . . . . . . . . . . . 389
11.3 Neural Networks . . . . . . . . . . . . . . . . . . . . . . . 392
11.4 Fitting Neural Networks . . . . . . . . . . . . . . . . . . . 395
11.5 Some Issues in Training Neural Networks . . . . . . . . . 397
</p>
<p>11.5.1 Starting Values . . . . . . . . . . . . . . . . . . . 397
11.5.2 Overfitting . . . . . . . . . . . . . . . . . . . . . 398
11.5.3 Scaling of the Inputs . . . . . . . . . . . . . . . 398
11.5.4 Number of Hidden Units and Layers . . . . . . . 400
11.5.5 Multiple Minima . . . . . . . . . . . . . . . . . . 400
</p>
<p>11.6 Example: Simulated Data . . . . . . . . . . . . . . . . . . 401
11.7 Example: ZIP Code Data . . . . . . . . . . . . . . . . . . 404
11.8 Discussion . . . . . . . . . . . . . . . . . . . . . . . . . . 408
11.9 Bayesian Neural Nets and the NIPS 2003 Challenge . . . 409
</p>
<p>11.9.1 Bayes, Boosting and Bagging . . . . . . . . . . . 410
11.9.2 Performance Comparisons . . . . . . . . . . . . 412
</p>
<p>11.10 Computational Considerations . . . . . . . . . . . . . . . 414
Bibliographic Notes . . . . . . . . . . . . . . . . . . . . . . . . . 415</p>
<p/>
</div>
<div class="page"><p/>
<p>Contents xix
</p>
<p>Exercises . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 415
</p>
<p>12 Support Vector Machines and
</p>
<p>Flexible Discriminants 417
</p>
<p>12.1 Introduction . . . . . . . . . . . . . . . . . . . . . . . . . 417
12.2 The Support Vector Classifier . . . . . . . . . . . . . . . . 417
</p>
<p>12.2.1 Computing the Support Vector Classifier . . . . 420
12.2.2 Mixture Example (Continued) . . . . . . . . . . 421
</p>
<p>12.3 Support Vector Machines and Kernels . . . . . . . . . . . 423
12.3.1 Computing the SVM for Classification . . . . . . 423
12.3.2 The SVM as a Penalization Method . . . . . . . 426
12.3.3 Function Estimation and Reproducing Kernels . 428
12.3.4 SVMs and the Curse of Dimensionality . . . . . 431
12.3.5 A Path Algorithm for the SVM Classifier . . . . 432
12.3.6 Support Vector Machines for Regression . . . . . 434
12.3.7 Regression and Kernels . . . . . . . . . . . . . . 436
12.3.8 Discussion . . . . . . . . . . . . . . . . . . . . . 438
</p>
<p>12.4 Generalizing Linear Discriminant Analysis . . . . . . . . 438
12.5 Flexible Discriminant Analysis . . . . . . . . . . . . . . . 440
</p>
<p>12.5.1 Computing the FDA Estimates . . . . . . . . . . 444
12.6 Penalized Discriminant Analysis . . . . . . . . . . . . . . 446
12.7 Mixture Discriminant Analysis . . . . . . . . . . . . . . . 449
</p>
<p>12.7.1 Example: Waveform Data . . . . . . . . . . . . . 451
Bibliographic Notes . . . . . . . . . . . . . . . . . . . . . . . . . 455
Exercises . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 455
</p>
<p>13 Prototype Methods and Nearest-Neighbors 459
</p>
<p>13.1 Introduction . . . . . . . . . . . . . . . . . . . . . . . . . 459
13.2 Prototype Methods . . . . . . . . . . . . . . . . . . . . . 459
</p>
<p>13.2.1 K-means Clustering . . . . . . . . . . . . . . . . 460
13.2.2 Learning Vector Quantization . . . . . . . . . . 462
13.2.3 Gaussian Mixtures . . . . . . . . . . . . . . . . . 463
</p>
<p>13.3 k-Nearest-Neighbor Classifiers . . . . . . . . . . . . . . . 463
13.3.1 Example: A Comparative Study . . . . . . . . . 468
13.3.2 Example: k-Nearest-Neighbors
</p>
<p>and Image Scene Classification . . . . . . . . . . 470
13.3.3 Invariant Metrics and Tangent Distance . . . . . 471
</p>
<p>13.4 Adaptive Nearest-Neighbor Methods . . . . . . . . . . . . 475
13.4.1 Example . . . . . . . . . . . . . . . . . . . . . . 478
13.4.2 Global Dimension Reduction
</p>
<p>for Nearest-Neighbors . . . . . . . . . . . . . . . 479
13.5 Computational Considerations . . . . . . . . . . . . . . . 480
Bibliographic Notes . . . . . . . . . . . . . . . . . . . . . . . . . 481
Exercises . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 481</p>
<p/>
</div>
<div class="page"><p/>
<p>xx Contents
</p>
<p>14 Unsupervised Learning 485
</p>
<p>14.1 Introduction . . . . . . . . . . . . . . . . . . . . . . . . . 485
14.2 Association Rules . . . . . . . . . . . . . . . . . . . . . . 487
</p>
<p>14.2.1 Market Basket Analysis . . . . . . . . . . . . . . 488
14.2.2 The Apriori Algorithm . . . . . . . . . . . . . . 489
14.2.3 Example: Market Basket Analysis . . . . . . . . 492
14.2.4 Unsupervised as Supervised Learning . . . . . . 495
14.2.5 Generalized Association Rules . . . . . . . . . . 497
14.2.6 Choice of Supervised Learning Method . . . . . 499
14.2.7 Example: Market Basket Analysis (Continued) . 499
</p>
<p>14.3 Cluster Analysis . . . . . . . . . . . . . . . . . . . . . . . 501
14.3.1 Proximity Matrices . . . . . . . . . . . . . . . . 503
14.3.2 Dissimilarities Based on Attributes . . . . . . . 503
14.3.3 Object Dissimilarity . . . . . . . . . . . . . . . . 505
14.3.4 Clustering Algorithms . . . . . . . . . . . . . . . 507
14.3.5 Combinatorial Algorithms . . . . . . . . . . . . 507
14.3.6 K-means . . . . . . . . . . . . . . . . . . . . . . 509
14.3.7 Gaussian Mixtures as Soft K-means Clustering . 510
14.3.8 Example: Human Tumor Microarray Data . . . 512
14.3.9 Vector Quantization . . . . . . . . . . . . . . . . 514
14.3.10 K-medoids . . . . . . . . . . . . . . . . . . . . . 515
14.3.11 Practical Issues . . . . . . . . . . . . . . . . . . 518
14.3.12 Hierarchical Clustering . . . . . . . . . . . . . . 520
</p>
<p>14.4 Self-Organizing Maps . . . . . . . . . . . . . . . . . . . . 528
14.5 Principal Components, Curves and Surfaces . . . . . . . . 534
</p>
<p>14.5.1 Principal Components . . . . . . . . . . . . . . . 534
14.5.2 Principal Curves and Surfaces . . . . . . . . . . 541
14.5.3 Spectral Clustering . . . . . . . . . . . . . . . . 544
14.5.4 Kernel Principal Components . . . . . . . . . . . 547
14.5.5 Sparse Principal Components . . . . . . . . . . . 550
</p>
<p>14.6 Non-negative Matrix Factorization . . . . . . . . . . . . . 553
14.6.1 Archetypal Analysis . . . . . . . . . . . . . . . . 554
</p>
<p>14.7 Independent Component Analysis
and Exploratory Projection Pursuit . . . . . . . . . . . . 557
14.7.1 Latent Variables and Factor Analysis . . . . . . 558
14.7.2 Independent Component Analysis . . . . . . . . 560
14.7.3 Exploratory Projection Pursuit . . . . . . . . . . 565
14.7.4 A Direct Approach to ICA . . . . . . . . . . . . 565
</p>
<p>14.8 Multidimensional Scaling . . . . . . . . . . . . . . . . . . 570
14.9 Nonlinear Dimension Reduction
</p>
<p>and Local Multidimensional Scaling . . . . . . . . . . . . 572
14.10 The Google PageRank Algorithm . . . . . . . . . . . . . 576
Bibliographic Notes . . . . . . . . . . . . . . . . . . . . . . . . . 578
Exercises . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 579</p>
<p/>
</div>
<div class="page"><p/>
<p>Contents xxi
</p>
<p>15 Random Forests 587
</p>
<p>15.1 Introduction . . . . . . . . . . . . . . . . . . . . . . . . . 587
15.2 Definition of Random Forests . . . . . . . . . . . . . . . . 587
15.3 Details of Random Forests . . . . . . . . . . . . . . . . . 592
</p>
<p>15.3.1 Out of Bag Samples . . . . . . . . . . . . . . . . 592
15.3.2 Variable Importance . . . . . . . . . . . . . . . . 593
15.3.3 Proximity Plots . . . . . . . . . . . . . . . . . . 595
15.3.4 Random Forests and Overfitting . . . . . . . . . 596
</p>
<p>15.4 Analysis of Random Forests . . . . . . . . . . . . . . . . . 597
15.4.1 Variance and the De-Correlation Effect . . . . . 597
15.4.2 Bias . . . . . . . . . . . . . . . . . . . . . . . . . 600
15.4.3 Adaptive Nearest Neighbors . . . . . . . . . . . 601
</p>
<p>Bibliographic Notes . . . . . . . . . . . . . . . . . . . . . . . . . 602
Exercises . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 603
</p>
<p>16 Ensemble Learning 605
</p>
<p>16.1 Introduction . . . . . . . . . . . . . . . . . . . . . . . . . 605
16.2 Boosting and Regularization Paths . . . . . . . . . . . . . 607
</p>
<p>16.2.1 Penalized Regression . . . . . . . . . . . . . . . 607
16.2.2 The &ldquo;Bet on Sparsity&rdquo; Principle . . . . . . . . . 610
16.2.3 Regularization Paths, Over-fitting and Margins . 613
</p>
<p>16.3 Learning Ensembles . . . . . . . . . . . . . . . . . . . . . 616
16.3.1 Learning a Good Ensemble . . . . . . . . . . . . 617
16.3.2 Rule Ensembles . . . . . . . . . . . . . . . . . . 622
</p>
<p>Bibliographic Notes . . . . . . . . . . . . . . . . . . . . . . . . . 623
Exercises . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 624
</p>
<p>17 Undirected Graphical Models 625
</p>
<p>17.1 Introduction . . . . . . . . . . . . . . . . . . . . . . . . . 625
17.2 Markov Graphs and Their Properties . . . . . . . . . . . 627
17.3 Undirected Graphical Models for Continuous Variables . 630
</p>
<p>17.3.1 Estimation of the Parameters
when the Graph Structure is Known . . . . . . . 631
</p>
<p>17.3.2 Estimation of the Graph Structure . . . . . . . . 635
17.4 Undirected Graphical Models for Discrete Variables . . . 638
</p>
<p>17.4.1 Estimation of the Parameters
when the Graph Structure is Known . . . . . . . 639
</p>
<p>17.4.2 Hidden Nodes . . . . . . . . . . . . . . . . . . . 641
17.4.3 Estimation of the Graph Structure . . . . . . . . 642
17.4.4 Restricted Boltzmann Machines . . . . . . . . . 643
</p>
<p>Exercises . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 645
</p>
<p>18 High-Dimensional Problems: p ≫ N 649
18.1 When p is Much Bigger than N . . . . . . . . . . . . . . 649</p>
<p/>
</div>
<div class="page"><p/>
<p>xxii Contents
</p>
<p>18.2 Diagonal Linear Discriminant Analysis
and Nearest Shrunken Centroids . . . . . . . . . . . . . . 651
</p>
<p>18.3 Linear Classifiers with Quadratic Regularization . . . . . 654
18.3.1 Regularized Discriminant Analysis . . . . . . . . 656
18.3.2 Logistic Regression
</p>
<p>with Quadratic Regularization . . . . . . . . . . 657
18.3.3 The Support Vector Classifier . . . . . . . . . . 657
18.3.4 Feature Selection . . . . . . . . . . . . . . . . . . 658
18.3.5 Computational Shortcuts When p ≫ N . . . . . 659
</p>
<p>18.4 Linear Classifiers with L1 Regularization . . . . . . . . . 661
18.4.1 Application of Lasso
</p>
<p>to Protein Mass Spectroscopy . . . . . . . . . . 664
18.4.2 The Fused Lasso for Functional Data . . . . . . 666
</p>
<p>18.5 Classification When Features are Unavailable . . . . . . . 668
18.5.1 Example: String Kernels
</p>
<p>and Protein Classification . . . . . . . . . . . . . 668
18.5.2 Classification and Other Models Using
</p>
<p>Inner-Product Kernels and Pairwise Distances . 670
18.5.3 Example: Abstracts Classification . . . . . . . . 672
</p>
<p>18.6 High-Dimensional Regression:
Supervised Principal Components . . . . . . . . . . . . . 674
18.6.1 Connection to Latent-Variable Modeling . . . . 678
18.6.2 Relationship with Partial Least Squares . . . . . 680
18.6.3 Pre-Conditioning for Feature Selection . . . . . 681
</p>
<p>18.7 Feature Assessment and the Multiple-Testing Problem . . 683
18.7.1 The False Discovery Rate . . . . . . . . . . . . . 687
18.7.2 Asymmetric Cutpoints and the SAM Procedure 690
18.7.3 A Bayesian Interpretation of the FDR . . . . . . 692
</p>
<p>18.8 Bibliographic Notes . . . . . . . . . . . . . . . . . . . . . 693
Exercises . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . 694
</p>
<p>References 699
</p>
<p>Author Index 729
</p>
<p>Index 737</p>
<p/>
</div>
<div class="page"><p/>
<p>1
Introduction
</p>
<p>Statistical learning plays a key role in many areas of science, finance and
industry. Here are some examples of learning problems:
</p>
<p>&bull; Predict whether a patient, hospitalized due to a heart attack, will
have a second heart attack. The prediction is to be based on demo-
graphic, diet and clinical measurements for that patient.
</p>
<p>&bull; Predict the price of a stock in 6 months from now, on the basis of
company performance measures and economic data.
</p>
<p>&bull; Identify the numbers in a handwritten ZIP code, from a digitized
image.
</p>
<p>&bull; Estimate the amount of glucose in the blood of a diabetic person,
from the infrared absorption spectrum of that person&rsquo;s blood.
</p>
<p>&bull; Identify the risk factors for prostate cancer, based on clinical and
demographic variables.
</p>
<p>The science of learning plays a key role in the fields of statistics, data
mining and artificial intelligence, intersecting with areas of engineering and
other disciplines.
This book is about learning from data. In a typical scenario, we have
</p>
<p>an outcome measurement, usually quantitative (such as a stock price) or
categorical (such as heart attack/no heart attack), that we wish to predict
based on a set of features (such as diet and clinical measurements). We
have a training set of data, in which we observe the outcome and feature
</p>
<p> 
</p>
<p>1
</p>
<p>&copy; Springer Science+Business Media, LLC 2009 
</p>
<p>T. Hastie et al., The Elements of Statistical Learning, Second Edition, 
</p>
<p>DOI: 10.1007/b94608_1,</p>
<p/>
</div>
<div class="page"><p/>
<p>2 1. Introduction
</p>
<p>TABLE 1.1. Average percentage of words or characters in an email message
equal to the indicated word or character. We have chosen the words and characters
showing the largest difference between spam and email.
</p>
<p>george you your hp free hpl ! our re edu remove
</p>
<p>spam 0.00 2.26 1.38 0.02 0.52 0.01 0.51 0.51 0.13 0.01 0.28
email 1.27 1.27 0.44 0.90 0.07 0.43 0.11 0.18 0.42 0.29 0.01
</p>
<p>measurements for a set of objects (such as people). Using this data we build
a prediction model, or learner, which will enable us to predict the outcome
for new unseen objects. A good learner is one that accurately predicts such
an outcome.
The examples above describe what is called the supervised learning prob-
</p>
<p>lem. It is called &ldquo;supervised&rdquo; because of the presence of the outcome vari-
able to guide the learning process. In the unsupervised learning problem,
we observe only the features and have no measurements of the outcome.
Our task is rather to describe how the data are organized or clustered. We
devote most of this book to supervised learning; the unsupervised problem
is less developed in the literature, and is the focus of Chapter 14.
Here are some examples of real learning problems that are discussed in
</p>
<p>this book.
</p>
<p>Example 1: Email Spam
</p>
<p>The data for this example consists of information from 4601 email mes-
sages, in a study to try to predict whether the email was junk email, or
&ldquo;spam.&rdquo; The objective was to design an automatic spam detector that
could filter out spam before clogging the users&rsquo; mailboxes. For all 4601
email messages, the true outcome (email type) email or spam is available,
along with the relative frequencies of 57 of the most commonly occurring
words and punctuation marks in the email message. This is a supervised
learning problem, with the outcome the class variable email/spam. It is also
called a classification problem.
Table 1.1 lists the words and characters showing the largest average
</p>
<p>difference between spam and email.
Our learning method has to decide which features to use and how: for
</p>
<p>example, we might use a rule such as
</p>
<p>if (%george &lt; 0.6) &amp; (%you &gt; 1.5) then spam
else email.
</p>
<p>Another form of a rule might be:
</p>
<p>if (0.2 &middot; %you &minus; 0.3 &middot; %george) &gt; 0 then spam
else email.</p>
<p/>
</div>
<div class="page"><p/>
<p>1. Introduction 3
</p>
<p>lpsa
</p>
<p>&minus;1 1 2 3 4
</p>
<p>oo oo
o
</p>
<p>o ooo o
oo ooooo oo o
</p>
<p>o oo oo ooo o oo
o ooo ooo ooo oo
</p>
<p>ooo oo ooo oo o oo
ooo oo o oooo ooo o ooo o
</p>
<p>ooo oooo oooo o
oo oo o
</p>
<p>oo
o
</p>
<p>oo
o
</p>
<p>o oo o
o
</p>
<p>o oooo
oooo oo ooo o
</p>
<p>oooo oo ooo o o
oo oo ooo ooo oo
ooo oooo o oo oo o
</p>
<p>o o oo ooo oo ooo o ooo oo
oo ooo oo o oooo
</p>
<p>oo oo o
o o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>40 50 60 70 80
</p>
<p>o o oo
o
</p>
<p>o ooo o
ooo oo o ooo o
</p>
<p>ooo o ooooooo
o oo oo oo oo o oo
</p>
<p>o oo ooo oooo oo o
oo oo oo oooo o oooo ooo
</p>
<p>oo o ooooo ooo o
o oo oo
</p>
<p>o o
o
</p>
<p>o o
o
</p>
<p>oooo
o
o o ooo
o ooooo ooo o
o oo o ooo ooo o
</p>
<p>oooo ooo oooo o
o oo ooo oo oooo o
o oo o ooo oo o o oo oo oo o
o o oooo ooo o oo
</p>
<p>ooo oo
oo
</p>
<p>o
o o
</p>
<p>o
</p>
<p>0.0 0.4 0.8
</p>
<p>oooo
o
ooooo
oooooooooo
ooooooooooo
ooooooo ooooo
ooo oooooooooo
ooooo oo ooooooo oo oo
</p>
<p>oooo oooo ooo o
o oooo
</p>
<p>oo
o
oo
o
</p>
<p>oooo
o
ooooo
oo oo oo o ooo
o oo ooo oo o oo
ooo oo oo oooo o
</p>
<p>ooo ooo ooo o ooo
ooooo oo oo o ooo o oo o o
</p>
<p>o ooo ooo oo oo o
o o ooo
o o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>6.0 7.0 8.0 9.0
</p>
<p>oo oo
o
ooooo
oo oooo oooo
o oo ooo ooooo
oooo o oooo ooo
</p>
<p>ooo oooo oo oooo
oo ooo o ooo oooo oooo o
ooooooooo ooo
</p>
<p>o oooo
oo
o
oo
o
</p>
<p>0
1
</p>
<p>2
3
</p>
<p>4
5
</p>
<p>oo oo
o
ooooo
oo oooo oooo
o oo ooo oo ooo
ooooo oo oo ooo
o oo oooo oo ooo o
oo o oo o ooo o oooo ooo o
</p>
<p>o o oo ooo oo oo o
o o o oo
</p>
<p>o o
o
</p>
<p>o o
o
</p>
<p>&minus;
1
</p>
<p>1
2
</p>
<p>3
4
</p>
<p>o
o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
o
oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
oo
o
o
</p>
<p>o
</p>
<p>o
oo
o
oo
</p>
<p>o
</p>
<p>o
o
oo
o
</p>
<p>oo
</p>
<p>o
ooo
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>oo
o
</p>
<p>lcavol
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>oo o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
o
</p>
<p>oo
</p>
<p>o
oo o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>oo
</p>
<p>o
ooo
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o o
</p>
<p>o
ooo
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
o
oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
oo
o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
o
</p>
<p>oo
</p>
<p>o
o oo
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
oo
o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
o
</p>
<p>o o
</p>
<p>o
o oo
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
o
o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo
o
oo
</p>
<p>o
</p>
<p>o
o
oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
ooo
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
o
o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o o
</p>
<p>o
o oo
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
ooo
o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oooo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
oo
</p>
<p>oo
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
ooo
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o o
</p>
<p>oo
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>lweight
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
ooo
o
</p>
<p>oo
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>ooo o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
oo
</p>
<p>oo
</p>
<p>oo
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o oo
o
</p>
<p>o o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o oo o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
oo
</p>
<p>oo
</p>
<p>oo
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
ooo
o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oooo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
oo
</p>
<p>oo
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
ooo
o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
ooo
o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o oo o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
o o
</p>
<p>oo
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>2
.5
</p>
<p>3
.5
</p>
<p>4
.5
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
ooo
o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o oo o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
oo
</p>
<p>oo
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>4
0
</p>
<p>5
0
</p>
<p>6
0
</p>
<p>7
0
</p>
<p>8
0
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oooo
o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>oo
oooooo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o o
oo o ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o ooo
o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>oo
ooo o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>age
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>oo
o ooo oo
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oooo
o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>oo
oooooo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>oo
oo o ooo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>oo
oooooo
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>oo
oo oooo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oooo oo
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>oooo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>oo oo oo
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>oooo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>oo
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o oo ooo
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>oo oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o o oo oo
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o oo o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o lbph
</p>
<p>oooooo
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>oooo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>oooooo
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>oo oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo oooo
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>oooo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>&minus;
1
</p>
<p>0
1
</p>
<p>2
</p>
<p>oo oooo
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>oooo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>0
.0
</p>
<p>0
.4
</p>
<p>0
.8
</p>
<p>oooo oooooooooooooooooooooooooooooooooo
</p>
<p>o
</p>
<p>ooooooo
</p>
<p>o
</p>
<p>oooooooooooooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oooooo
</p>
<p>o
</p>
<p>o
</p>
<p>oooo
</p>
<p>oo
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>oooooo
</p>
<p>oo oo oo ooo ooo ooooo oo o o oo oo ooo o ooo ooo ooo
</p>
<p>o
</p>
<p>oo oo ooo
</p>
<p>o
</p>
<p>o ooo oo o ooooo oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo ooo o
</p>
<p>o
</p>
<p>o
</p>
<p>o o oo
</p>
<p>o o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>oo ooo o
</p>
<p>o oo ooo oooo oooo oo ooo ooooo oo ooo o ooo oo ooo
</p>
<p>o
</p>
<p>oo ooooo
</p>
<p>o
</p>
<p>ooo o oo oo oo o oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o ooo o o
</p>
<p>o
</p>
<p>o
</p>
<p>oo oo
</p>
<p>oo
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o ooo o o
</p>
<p>o o oo oo ooo oooo oo o ooo oooo o oooooooo oo oo oo
</p>
<p>o
</p>
<p>o o ooo oo
</p>
<p>o
</p>
<p>oo oooo oo ooo oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o ooo
</p>
<p>o
</p>
<p>o
</p>
<p>oo oo
</p>
<p>o o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o oo o oo
</p>
<p>oooooo o oooo ooooo ooo oo oo o ooo ooo ooooo ooo
</p>
<p>o
</p>
<p>ooo oo oo
</p>
<p>o
</p>
<p>oo oo oooo oo oo o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o o oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo o
</p>
<p>oo
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>oooo oo
</p>
<p>svi
</p>
<p>oooooooooooo oo oo o oooo oo ooo oo o ooooo oo oo
</p>
<p>o
</p>
<p>ooo o ooo
</p>
<p>o
</p>
<p>oo ooo o ooo ooooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o ooo o
</p>
<p>o
</p>
<p>o
</p>
<p>o o o o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o o ooo o
</p>
<p>oo oooooooooo oooo ooooo oo ooo ooooooooo o oo
</p>
<p>o
</p>
<p>o ooo ooo
</p>
<p>o
</p>
<p>ooo oo oooooo ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oooo o
</p>
<p>o
</p>
<p>o
</p>
<p>o ooo
</p>
<p>oo
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>oooooo
</p>
<p>oo oooooooooo oooo ooooo oo ooo oo oooooooo oo
</p>
<p>o
</p>
<p>o oooo oo
</p>
<p>o
</p>
<p>ooo oo ooo ooo o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o oooo
</p>
<p>o
</p>
<p>o
</p>
<p>o oo o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o o o
</p>
<p>o
</p>
<p>o ooo oo
</p>
<p>oooo oooooooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
ooo
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oooo
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo oo oo ooo ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo ooo oooo oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o oo o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o oo oo ooo ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o oo o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oooooo o oooo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>oo o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oooooooooooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
ooo
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oooo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>lcp
</p>
<p>oo oooooooooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
ooo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o ooo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>&minus;
1
</p>
<p>0
1
</p>
<p>2
3
</p>
<p>oo oooooooooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
ooo
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o o oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>6
.0
</p>
<p>7
.0
</p>
<p>8
.0
</p>
<p>9
.0
</p>
<p>oo
</p>
<p>o
</p>
<p>o oooooooo
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>oooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>ooo
</p>
<p>oooooo
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>ooooo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>oooo
</p>
<p>o
</p>
<p>ooooooooo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>oooooo
</p>
<p>oo
</p>
<p>o
</p>
<p>o oo ooo ooo
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>oo o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>oo o
</p>
<p>ooo ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o o ooo
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o ooo
</p>
<p>o
</p>
<p>ooo oooo oo
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>oo ooo o
</p>
<p>o o
</p>
<p>o
</p>
<p>ooo oooo oo
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>oo oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>ooo
</p>
<p>o ooo oo
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o oo oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>ooo o
</p>
<p>o
</p>
<p>oo ooo oo o o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o ooo o o
</p>
<p>o o
</p>
<p>o
</p>
<p>o oo ooo ooo
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>ooo
</p>
<p>ooo oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o o
</p>
<p>o
</p>
<p>oo oo
</p>
<p>o
</p>
<p>oo o ooooo o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o oo o oo
</p>
<p>oo
</p>
<p>o
</p>
<p>ooo o oooo o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>oo oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o oo
</p>
<p>o ooooo
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo oo
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o o
</p>
<p>o
</p>
<p>oo oo
</p>
<p>o
</p>
<p>o o oooo ooo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>oooo oo
</p>
<p>oo
</p>
<p>o
</p>
<p>ooooooooo
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>oooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>ooo
</p>
<p>oooooo
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>ooooo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o oo o
</p>
<p>o
</p>
<p>oooo oooo o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>oooooo
</p>
<p>oo
</p>
<p>o
</p>
<p>ooooooooo
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>oooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo o
</p>
<p>ooooo o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o ooo o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o oo o
</p>
<p>o
</p>
<p>o ooo ooo oo
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o o ooo o
gleason
</p>
<p>oo
</p>
<p>o
</p>
<p>ooooooooo
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>oooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo o
</p>
<p>oooooo
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>ooo oo
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o ooo
</p>
<p>o
</p>
<p>o o oo ooo oo
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o o o
</p>
<p>o
</p>
<p>o ooo oo
</p>
<p>0 1 2 3 4 5
</p>
<p>oo
</p>
<p>o
</p>
<p>o oooooooo
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>oooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooooooo
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o oo ooo ooo
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>oo o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo ooo o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>2.5 3.5 4.5
</p>
<p>o o
</p>
<p>o
</p>
<p>ooo oooo oo
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>oo oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o ooo oo o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o oo ooo ooo
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>oo oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo oo oo
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>&minus;1 0 1 2
</p>
<p>oo
</p>
<p>o
</p>
<p>ooo o oooo o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>oo oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o ooooo o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>ooooooooo
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>oooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooooooo
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>&minus;1 0 1 2 3
</p>
<p>oo
</p>
<p>o
</p>
<p>ooooooooo
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>oooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooooo oo
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>ooooooooo
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>oooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oooooo o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>0 20 60 100
0
</p>
<p>2
0
</p>
<p>6
0
</p>
<p>1
0
</p>
<p>0
</p>
<p>pgg45
</p>
<p>FIGURE 1.1. Scatterplot matrix of the prostate cancer data. The first row shows
the response against each of the predictors in turn. Two of the predictors, svi and
gleason, are categorical.
</p>
<p>For this problem not all errors are equal; we want to avoid filtering out
good email, while letting spam get through is not desirable but less serious
in its consequences. We discuss a number of different methods for tackling
this learning problem in the book.
</p>
<p>Example 2: Prostate Cancer
</p>
<p>The data for this example, displayed in Figure 1.11, come from a study
by Stamey et al. (1989) that examined the correlation between the level of
</p>
<p>1There was an error in these data in the first edition of this book. Subject 32 had
a value of 6.1 for lweight, which translates to a 449 gm prostate! The correct value is
</p>
<p>44.9 gm. We are grateful to Prof. Stephen W. Link for alerting us to this error.</p>
<p/>
</div>
<div class="page"><p/>
<p>4 1. Introduction
</p>
<p>FIGURE 1.2. Examples of handwritten digits from U.S. postal envelopes.
</p>
<p>prostate specific antigen (PSA) and a number of clinical measures, in 97
men who were about to receive a radical prostatectomy.
The goal is to predict the log of PSA (lpsa) from a number of measure-
</p>
<p>ments including log cancer volume (lcavol), log prostate weight lweight,
age, log of benign prostatic hyperplasia amount lbph, seminal vesicle in-
vasion svi, log of capsular penetration lcp, Gleason score gleason, and
percent of Gleason scores 4 or 5 pgg45. Figure 1.1 is a scatterplot matrix
of the variables. Some correlations with lpsa are evident, but a good pre-
dictive model is difficult to construct by eye.
This is a supervised learning problem, known as a regression problem,
</p>
<p>because the outcome measurement is quantitative.
</p>
<p>Example 3: Handwritten Digit Recognition
</p>
<p>The data from this example come from the handwritten ZIP codes on
envelopes from U.S. postal mail. Each image is a segment from a five digit
ZIP code, isolating a single digit. The images are 16&times;16 eight-bit grayscale
maps, with each pixel ranging in intensity from 0 to 255. Some sample
images are shown in Figure 1.2.
The images have been normalized to have approximately the same size
</p>
<p>and orientation. The task is to predict, from the 16 &times; 16 matrix of pixel
intensities, the identity of each image (0, 1, . . . , 9) quickly and accurately. If
it is accurate enough, the resulting algorithm would be used as part of an
automatic sorting procedure for envelopes. This is a classification problem
for which the error rate needs to be kept very low to avoid misdirection of</p>
<p/>
</div>
<div class="page"><p/>
<p>1. Introduction 5
</p>
<p>mail. In order to achieve this low error rate, some objects can be assigned
to a &ldquo;don&rsquo;t know&rdquo; category, and sorted instead by hand.
</p>
<p>Example 4: DNA Expression Microarrays
</p>
<p>DNA stands for deoxyribonucleic acid, and is the basic material that makes
up human chromosomes. DNA microarrays measure the expression of a
gene in a cell by measuring the amount of mRNA (messenger ribonucleic
acid) present for that gene. Microarrays are considered a breakthrough
technology in biology, facilitating the quantitative study of thousands of
genes simultaneously from a single sample of cells.
Here is how a DNA microarray works. The nucleotide sequences for a few
</p>
<p>thousand genes are printed on a glass slide. A target sample and a reference
sample are labeled with red and green dyes, and each are hybridized with
the DNA on the slide. Through fluoroscopy, the log (red/green) intensities
of RNA hybridizing at each site is measured. The result is a few thousand
numbers, typically ranging from say &minus;6 to 6, measuring the expression level
of each gene in the target relative to the reference sample. Positive values
indicate higher expression in the target versus the reference, and vice versa
for negative values.
A gene expression dataset collects together the expression values from a
</p>
<p>series of DNA microarray experiments, with each column representing an
experiment. There are therefore several thousand rows representing individ-
ual genes, and tens of columns representing samples: in the particular ex-
ample of Figure 1.3 there are 6830 genes (rows) and 64 samples (columns),
although for clarity only a random sample of 100 rows are shown. The fig-
ure displays the data set as a heat map, ranging from green (negative) to
red (positive). The samples are 64 cancer tumors from different patients.
The challenge here is to understand how the genes and samples are or-
</p>
<p>ganized. Typical questions include the following:
</p>
<p>(a) which samples are most similar to each other, in terms of their expres-
sion profiles across genes?
</p>
<p>(b) which genes are most similar to each other, in terms of their expression
profiles across samples?
</p>
<p>(c) do certain genes show very high (or low) expression for certain cancer
samples?
</p>
<p>We could view this task as a regression problem, with two categorical
predictor variables&mdash;genes and samples&mdash;with the response variable being
the level of expression. However, it is probably more useful to view it as
unsupervised learning problem. For example, for question (a) above, we
think of the samples as points in 6830&ndash;dimensional space, which we want
to cluster together in some way.</p>
<p/>
</div>
<div class="page"><p/>
<p>6 1. Introduction
</p>
<p>SID42354
SID31984
SID301902
SIDW128368
SID375990
SID360097
SIDW325120
ESTsChr.10
SIDW365099
SID377133
SID381508
SIDW308182
SID380265
SIDW321925
ESTsChr.15
SIDW362471
SIDW417270
SIDW298052
SID381079
SIDW428642
TUPLE1TUP1
ERLUMEN
SIDW416621
SID43609
ESTs
SID52979
SIDW357197
SIDW366311
ESTs
SMALLNUC
SIDW486740
ESTs
SID297905
SID485148
SID284853
ESTsChr.15
SID200394
SIDW322806
ESTsChr.2
SIDW257915
SID46536
SIDW488221
ESTsChr.5
SID280066
SIDW376394
ESTsChr.15
SIDW321854
WASWiskott
HYPOTHETIC
SIDW376776
SIDW205716
SID239012
SIDW203464
HLACLASSI
SIDW510534
SIDW279664
SIDW201620
SID297117
SID377419
SID114241
ESTsCh31
SIDW376928
SIDW310141
SIDW298203
PTPRC
SID289414
SID127504
ESTsChr.3
SID305167
SID488017
SIDW296310
ESTsChr.6
SID47116
MITOCHOND
Chr
SIDW376586
Homosapiens
SIDW487261
SIDW470459
SID167117
SIDW31489
SID375812
DNAPOLYME
SID377451
ESTsChr.1
MYBPROTO
SID471915
ESTs
SIDW469884
HumanmRNA
SIDW377402
ESTs
SID207172
RASGTPASE
SID325394
H.sapiensmR
GNAL
SID73161
SIDW380102
SIDW299104
</p>
<p>B
R
</p>
<p>E
A
</p>
<p>S
T
</p>
<p>R
E
</p>
<p>N
A
</p>
<p>L
M
</p>
<p>E
L
</p>
<p>A
N
</p>
<p>O
M
</p>
<p>A
M
</p>
<p>E
L
</p>
<p>A
N
</p>
<p>O
M
</p>
<p>A
M
</p>
<p>C
F
</p>
<p>7
D
</p>
<p>-r
e
</p>
<p>p
ro
</p>
<p>C
O
</p>
<p>L
O
</p>
<p>N
C
</p>
<p>O
L
</p>
<p>O
N
</p>
<p>K
5
</p>
<p>6
2
</p>
<p>B
-r
</p>
<p>e
p
</p>
<p>ro
C
</p>
<p>O
L
</p>
<p>O
N
</p>
<p>N
S
</p>
<p>C
L
</p>
<p>C
L
</p>
<p>E
U
</p>
<p>K
E
</p>
<p>M
IA
</p>
<p>R
E
</p>
<p>N
A
</p>
<p>L
M
</p>
<p>E
L
</p>
<p>A
N
</p>
<p>O
M
</p>
<p>A
B
</p>
<p>R
E
</p>
<p>A
S
</p>
<p>T
C
</p>
<p>N
S
</p>
<p>C
N
</p>
<p>S
R
</p>
<p>E
N
</p>
<p>A
L
</p>
<p>M
C
</p>
<p>F
7
</p>
<p>A
-r
</p>
<p>e
p
</p>
<p>ro
N
</p>
<p>S
C
</p>
<p>L
C
</p>
<p>K
5
</p>
<p>6
2
</p>
<p>A
-r
</p>
<p>e
p
</p>
<p>ro
C
</p>
<p>O
L
</p>
<p>O
N
</p>
<p>C
N
</p>
<p>S
N
</p>
<p>S
C
</p>
<p>L
C
</p>
<p>N
S
</p>
<p>C
L
</p>
<p>C
L
</p>
<p>E
U
</p>
<p>K
E
</p>
<p>M
IA
</p>
<p>C
N
</p>
<p>S
O
</p>
<p>V
A
</p>
<p>R
IA
</p>
<p>N
B
</p>
<p>R
E
</p>
<p>A
S
</p>
<p>T
L
</p>
<p>E
U
</p>
<p>K
E
</p>
<p>M
IA
</p>
<p>M
E
</p>
<p>L
A
</p>
<p>N
O
</p>
<p>M
A
</p>
<p>M
E
</p>
<p>L
A
</p>
<p>N
O
</p>
<p>M
A
</p>
<p>O
V
</p>
<p>A
R
</p>
<p>IA
N
</p>
<p>O
V
</p>
<p>A
R
</p>
<p>IA
N
</p>
<p>N
S
</p>
<p>C
L
</p>
<p>C
R
</p>
<p>E
N
</p>
<p>A
L
</p>
<p>B
R
</p>
<p>E
A
</p>
<p>S
T
</p>
<p>M
E
</p>
<p>L
A
</p>
<p>N
O
</p>
<p>M
A
</p>
<p>O
V
</p>
<p>A
R
</p>
<p>IA
N
</p>
<p>O
V
</p>
<p>A
R
</p>
<p>IA
N
</p>
<p>N
S
</p>
<p>C
L
</p>
<p>C
R
</p>
<p>E
N
</p>
<p>A
L
</p>
<p>B
R
</p>
<p>E
A
</p>
<p>S
T
</p>
<p>M
E
</p>
<p>L
A
</p>
<p>N
O
</p>
<p>M
A
</p>
<p>L
E
</p>
<p>U
K
</p>
<p>E
M
</p>
<p>IA
C
</p>
<p>O
L
</p>
<p>O
N
</p>
<p>B
R
</p>
<p>E
A
</p>
<p>S
T
</p>
<p>L
E
</p>
<p>U
K
</p>
<p>E
M
</p>
<p>IA
C
</p>
<p>O
L
</p>
<p>O
N
</p>
<p>C
N
</p>
<p>S
M
</p>
<p>E
L
</p>
<p>A
N
</p>
<p>O
M
</p>
<p>A
N
</p>
<p>S
C
</p>
<p>L
C
</p>
<p>P
R
</p>
<p>O
S
</p>
<p>T
A
</p>
<p>T
E
</p>
<p>N
S
</p>
<p>C
L
</p>
<p>C
R
</p>
<p>E
N
</p>
<p>A
L
</p>
<p>R
E
</p>
<p>N
A
</p>
<p>L
N
</p>
<p>S
C
</p>
<p>L
C
</p>
<p>R
E
</p>
<p>N
A
</p>
<p>L
L
</p>
<p>E
U
</p>
<p>K
E
</p>
<p>M
IA
</p>
<p>O
V
</p>
<p>A
R
</p>
<p>IA
N
</p>
<p>P
R
</p>
<p>O
S
</p>
<p>T
A
</p>
<p>T
E
</p>
<p>C
O
</p>
<p>L
O
</p>
<p>N
B
</p>
<p>R
E
</p>
<p>A
S
</p>
<p>T
R
</p>
<p>E
N
</p>
<p>A
L
</p>
<p>U
N
</p>
<p>K
N
</p>
<p>O
W
</p>
<p>N
</p>
<p>FIGURE 1.3. DNA microarray data: expression matrix of 6830 genes (rows)
and 64 samples (columns), for the human tumor data. Only a random sample
of 100 rows are shown. The display is a heat map, ranging from bright green
(negative, under expressed) to bright red (positive, over expressed). Missing values
are gray. The rows and columns are displayed in a randomly chosen order.</p>
<p/>
</div>
<div class="page"><p/>
<p>1. Introduction 7
</p>
<p>Who Should Read this Book
</p>
<p>This book is designed for researchers and students in a broad variety of
fields: statistics, artificial intelligence, engineering, finance and others. We
expect that the reader will have had at least one elementary course in
statistics, covering basic topics including linear regression.
We have not attempted to write a comprehensive catalog of learning
</p>
<p>methods, but rather to describe some of the most important techniques.
Equally notable, we describe the underlying concepts and considerations
by which a researcher can judge a learning method. We have tried to write
this book in an intuitive fashion, emphasizing concepts rather than math-
ematical details.
As statisticians, our exposition will naturally reflect our backgrounds and
</p>
<p>areas of expertise. However in the past eight years we have been attending
conferences in neural networks, data mining and machine learning, and our
thinking has been heavily influenced by these exciting fields. This influence
is evident in our current research, and in this book.
</p>
<p>How This Book is Organized
</p>
<p>Our view is that one must understand simple methods before trying to
grasp more complex ones. Hence, after giving an overview of the supervis-
ing learning problem in Chapter 2, we discuss linear methods for regression
and classification in Chapters 3 and 4. In Chapter 5 we describe splines,
wavelets and regularization/penalization methods for a single predictor,
while Chapter 6 covers kernel methods and local regression. Both of these
sets of methods are important building blocks for high-dimensional learn-
ing techniques. Model assessment and selection is the topic of Chapter 7,
covering the concepts of bias and variance, overfitting and methods such as
cross-validation for choosing models. Chapter 8 discusses model inference
and averaging, including an overview of maximum likelihood, Bayesian in-
ference and the bootstrap, the EM algorithm, Gibbs sampling and bagging,
A related procedure called boosting is the focus of Chapter 10.
In Chapters 9&ndash;13 we describe a series of structured methods for su-
</p>
<p>pervised learning, with Chapters 9 and 11 covering regression and Chap-
ters 12 and 13 focusing on classification. Chapter 14 describes methods for
unsupervised learning. Two recently proposed techniques, random forests
and ensemble learning, are discussed in Chapters 15 and 16. We describe
undirected graphical models in Chapter 17 and finally we study high-
dimensional problems in Chapter 18.
</p>
<p>At the end of each chapter we discuss computational considerations im-
portant for data mining applications, including how the computations scale
with the number of observations and predictors. Each chapter ends with
Bibliographic Notes giving background references for the material.</p>
<p/>
</div>
<div class="page"><p/>
<p>8 1. Introduction
</p>
<p>We recommend that Chapters 1&ndash;4 be first read in sequence. Chapter 7
should also be considered mandatory, as it covers central concepts that
pertain to all learning methods. With this in mind, the rest of the book
can be read sequentially, or sampled, depending on the reader&rsquo;s interest.
</p>
<p>The symbol indicates a technically difficult section, one that can
be skipped without interrupting the flow of the discussion.
</p>
<p>Book Website
</p>
<p>The website for this book is located at
</p>
<p>http://www-stat.stanford.edu/ElemStatLearn
</p>
<p>It contains a number of resources, including many of the datasets used in
this book.
</p>
<p>Note for Instructors
</p>
<p>We have successively used the first edition of this book as the basis for a
two-quarter course, and with the additional materials in this second edition,
it could even be used for a three-quarter sequence. Exercises are provided at
the end of each chapter. It is important for students to have access to good
software tools for these topics. We used the R and S-PLUS programming
languages in our courses.</p>
<p/>
</div>
<div class="page"><p/>
<p>2
Overview of Supervised Learning
</p>
<p>2.1 Introduction
</p>
<p>The first three examples described in Chapter 1 have several components
in common. For each there is a set of variables that might be denoted as
inputs, which are measured or preset. These have some influence on one or
more outputs. For each example the goal is to use the inputs to predict the
values of the outputs. This exercise is called supervised learning.
We have used the more modern language of machine learning. In the
</p>
<p>statistical literature the inputs are often called the predictors, a term we
will use interchangeably with inputs, and more classically the independent
variables. In the pattern recognition literature the term features is preferred,
which we use as well. The outputs are called the responses, or classically
the dependent variables.
</p>
<p>2.2 Variable Types and Terminology
</p>
<p>The outputs vary in nature among the examples. In the glucose prediction
example, the output is a quantitative measurement, where some measure-
ments are bigger than others, and measurements close in value are close
in nature. In the famous Iris discrimination example due to R. A. Fisher,
the output is qualitative (species of Iris) and assumes values in a finite set
G = {Virginica, Setosa and Versicolor}. In the handwritten digit example
the output is one of 10 different digit classes: G = {0, 1, . . . , 9}. In both of
</p>
<p> &copy; Springer Science+Business Media, LLC 2009 
</p>
<p>T. Hastie et al., The Elements of Statistical Learning, Second Edition, 9
</p>
<p>DOI: 10.1007/b94608_2,</p>
<p/>
</div>
<div class="page"><p/>
<p>10 2. Overview of Supervised Learning
</p>
<p>these there is no explicit ordering in the classes, and in fact often descrip-
tive labels rather than numbers are used to denote the classes. Qualitative
variables are also referred to as categorical or discrete variables as well as
factors.
</p>
<p>For both types of outputs it makes sense to think of using the inputs to
predict the output. Given some specific atmospheric measurements today
and yesterday, we want to predict the ozone level tomorrow. Given the
grayscale values for the pixels of the digitized image of the handwritten
digit, we want to predict its class label.
This distinction in output type has led to a naming convention for the
</p>
<p>prediction tasks: regression when we predict quantitative outputs, and clas-
sification when we predict qualitative outputs. We will see that these two
tasks have a lot in common, and in particular both can be viewed as a task
in function approximation.
Inputs also vary in measurement type; we can have some of each of qual-
</p>
<p>itative and quantitative input variables. These have also led to distinctions
in the types of methods that are used for prediction: some methods are
defined most naturally for quantitative inputs, some most naturally for
qualitative and some for both.
A third variable type is ordered categorical, such as small, medium and
</p>
<p>large, where there is an ordering between the values, but no metric notion
is appropriate (the difference between medium and small need not be the
same as that between large and medium). These are discussed further in
Chapter 4.
Qualitative variables are typically represented numerically by codes. The
</p>
<p>easiest case is when there are only two classes or categories, such as &ldquo;suc-
cess&rdquo; or &ldquo;failure,&rdquo; &ldquo;survived&rdquo; or &ldquo;died.&rdquo; These are often represented by a
single binary digit or bit as 0 or 1, or else by &minus;1 and 1. For reasons that will
become apparent, such numeric codes are sometimes referred to as targets.
When there are more than two categories, several alternatives are available.
The most useful and commonly used coding is via dummy variables. Here a
K-level qualitative variable is represented by a vector of K binary variables
or bits, only one of which is &ldquo;on&rdquo; at a time. Although more compact coding
schemes are possible, dummy variables are symmetric in the levels of the
factor.
We will typically denote an input variable by the symbol X. If X is
</p>
<p>a vector, its components can be accessed by subscripts Xj . Quantitative
outputs will be denoted by Y , and qualitative outputs by G (for group).
We use uppercase letters such as X, Y or G when referring to the generic
aspects of a variable. Observed values are written in lowercase; hence the
ith observed value of X is written as xi (where xi is again a scalar or
vector). Matrices are represented by bold uppercase letters; for example, a
set of N input p-vectors xi, i = 1, . . . , N would be represented by the N&times;p
matrix X. In general, vectors will not be bold, except when they have N
components; this convention distinguishes a p-vector of inputs xi for the</p>
<p/>
</div>
<div class="page"><p/>
<p>2.3 Least Squares and Nearest Neighbors 11
</p>
<p>ith observation from the N -vector xj consisting of all the observations on
variable Xj . Since all vectors are assumed to be column vectors, the ith
row of X is xTi , the vector transpose of xi.
</p>
<p>For the moment we can loosely state the learning task as follows: given
the value of an input vector X, make a good prediction of the output Y,
denoted by Ŷ (pronounced &ldquo;y-hat&rdquo;). If Y takes values in IR then so should
Ŷ ; likewise for categorical outputs, Ĝ should take values in the same set G
associated with G.
For a two-class G, one approach is to denote the binary coded target
</p>
<p>as Y , and then treat it as a quantitative output. The predictions Ŷ will
typically lie in [0, 1], and we can assign to Ĝ the class label according to
whether ŷ &gt; 0.5. This approach generalizes to K-level qualitative outputs
as well.
We need data to construct prediction rules, often a lot of it. We thus
</p>
<p>suppose we have available a set of measurements (xi, yi) or (xi, gi), i =
1, . . . , N , known as the training data, with which to construct our prediction
rule.
</p>
<p>2.3 Two Simple Approaches to Prediction: Least
Squares and Nearest Neighbors
</p>
<p>In this section we develop two simple but powerful prediction methods: the
linear model fit by least squares and the k-nearest-neighbor prediction rule.
The linear model makes huge assumptions about structure and yields stable
but possibly inaccurate predictions. The method of k-nearest neighbors
makes very mild structural assumptions: its predictions are often accurate
but can be unstable.
</p>
<p>2.3.1 Linear Models and Least Squares
</p>
<p>The linear model has been a mainstay of statistics for the past 30 years
and remains one of our most important tools. Given a vector of inputs
XT = (X1, X2, . . . , Xp), we predict the output Y via the model
</p>
<p>Ŷ = β̂0 +
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>Xj β̂j . (2.1)
</p>
<p>The term β̂0 is the intercept, also known as the bias in machine learning.
Often it is convenient to include the constant variable 1 in X, include β̂0 in
the vector of coefficients β̂, and then write the linear model in vector form
as an inner product
</p>
<p>Ŷ = XT β̂, (2.2)</p>
<p/>
</div>
<div class="page"><p/>
<p>12 2. Overview of Supervised Learning
</p>
<p>where XT denotes vector or matrix transpose (X being a column vector).
Here we are modeling a single output, so Ŷ is a scalar; in general Ŷ can be
a K&ndash;vector, in which case β would be a p&times;K matrix of coefficients. In the
(p + 1)-dimensional input&ndash;output space, (X, Ŷ ) represents a hyperplane.
If the constant is included in X, then the hyperplane includes the origin
and is a subspace; if not, it is an affine set cutting the Y -axis at the point
(0, β̂0). From now on we assume that the intercept is included in β̂.
Viewed as a function over the p-dimensional input space, f(X) = XTβ
</p>
<p>is linear, and the gradient f &prime;(X) = β is a vector in input space that points
in the steepest uphill direction.
How do we fit the linear model to a set of training data? There are
</p>
<p>many different methods, but by far the most popular is the method of
least squares. In this approach, we pick the coefficients β to minimize the
residual sum of squares
</p>
<p>RSS(β) =
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>(yi &minus; xTi β)2. (2.3)
</p>
<p>RSS(β) is a quadratic function of the parameters, and hence its minimum
always exists, but may not be unique. The solution is easiest to characterize
in matrix notation. We can write
</p>
<p>RSS(β) = (y &minus;Xβ)T (y &minus;Xβ), (2.4)
where X is an N &times; p matrix with each row an input vector, and y is an
N -vector of the outputs in the training set. Differentiating w.r.t. β we get
the normal equations
</p>
<p>XT (y &minus;Xβ) = 0. (2.5)
If XTX is nonsingular, then the unique solution is given by
</p>
<p>β̂ = (XTX)&minus;1XTy, (2.6)
</p>
<p>and the fitted value at the ith input xi is ŷi = ŷ(xi) = x
T
i β̂. At an arbi-
</p>
<p>trary input x0 the prediction is ŷ(x0) = x
T
0 β̂. The entire fitted surface is
</p>
<p>characterized by the p parameters β̂. Intuitively, it seems that we do not
need a very large data set to fit such a model.
Let&rsquo;s look at an example of the linear model in a classification context.
</p>
<p>Figure 2.1 shows a scatterplot of training data on a pair of inputs X1 and
X2. The data are simulated, and for the present the simulation model is
not important. The output class variable G has the values BLUE or ORANGE,
and is represented as such in the scatterplot. There are 100 points in each
of the two classes. The linear regression model was fit to these data, with
the response Y coded as 0 for BLUE and 1 for ORANGE. The fitted values Ŷ
are converted to a fitted class variable Ĝ according to the rule
</p>
<p>Ĝ =
</p>
<p>{
ORANGE if Ŷ &gt; 0.5,
</p>
<p>BLUE if Ŷ &le; 0.5. (2.7)</p>
<p/>
</div>
<div class="page"><p/>
<p>2.3 Least Squares and Nearest Neighbors 13
</p>
<p>Linear Regression of 0/1 Response
</p>
<p>.. . . . . .
</p>
<p>. . . . . . . . . . . .. . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . .
</p>
<p>. . . . . . . .. .
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>FIGURE 2.1. A classification example in two dimensions. The classes are coded
as a binary variable (BLUE = 0, ORANGE = 1), and then fit by linear regression.
The line is the decision boundary defined by xT β̂ = 0.5. The orange shaded region
denotes that part of input space classified as ORANGE, while the blue region is
classified as BLUE.
</p>
<p>The set of points in IR2 classified as ORANGE corresponds to {x : xT β̂ &gt; 0.5},
indicated in Figure 2.1, and the two predicted classes are separated by the
decision boundary {x : xT β̂ = 0.5}, which is linear in this case. We see
that for these data there are several misclassifications on both sides of the
decision boundary. Perhaps our linear model is too rigid&mdash; or are such errors
unavoidable? Remember that these are errors on the training data itself,
and we have not said where the constructed data came from. Consider the
two possible scenarios:
</p>
<p>Scenario 1: The training data in each class were generated from bivariate
Gaussian distributions with uncorrelated components and different
means.
</p>
<p>Scenario 2: The training data in each class came from a mixture of 10 low-
variance Gaussian distributions, with individual means themselves
distributed as Gaussian.
</p>
<p>A mixture of Gaussians is best described in terms of the generative
model. One first generates a discrete variable that determines which of</p>
<p/>
</div>
<div class="page"><p/>
<p>14 2. Overview of Supervised Learning
</p>
<p>the component Gaussians to use, and then generates an observation from
the chosen density. In the case of one Gaussian per class, we will see in
Chapter 4 that a linear decision boundary is the best one can do, and that
our estimate is almost optimal. The region of overlap is inevitable, and
future data to be predicted will be plagued by this overlap as well.
In the case of mixtures of tightly clustered Gaussians the story is dif-
</p>
<p>ferent. A linear decision boundary is unlikely to be optimal, and in fact is
not. The optimal decision boundary is nonlinear and disjoint, and as such
will be much more difficult to obtain.
We now look at another classification and regression procedure that is
</p>
<p>in some sense at the opposite end of the spectrum to the linear model, and
far better suited to the second scenario.
</p>
<p>2.3.2 Nearest-Neighbor Methods
</p>
<p>Nearest-neighbor methods use those observations in the training set T clos-
est in input space to x to form Ŷ . Specifically, the k-nearest neighbor fit
for Ŷ is defined as follows:
</p>
<p>Ŷ (x) =
1
</p>
<p>k
</p>
<p>&sum;
</p>
<p>xi&isin;Nk(x)
</p>
<p>yi, (2.8)
</p>
<p>where Nk(x) is the neighborhood of x defined by the k closest points xi in
the training sample. Closeness implies a metric, which for the moment we
assume is Euclidean distance. So, in words, we find the k observations with
xi closest to x in input space, and average their responses.
In Figure 2.2 we use the same training data as in Figure 2.1, and use
</p>
<p>15-nearest-neighbor averaging of the binary coded response as the method
of fitting. Thus Ŷ is the proportion of ORANGE&rsquo;s in the neighborhood, and
so assigning class ORANGE to Ĝ if Ŷ &gt; 0.5 amounts to a majority vote in
the neighborhood. The colored regions indicate all those points in input
space classified as BLUE or ORANGE by such a rule, in this case found by
evaluating the procedure on a fine grid in input space. We see that the
decision boundaries that separate the BLUE from the ORANGE regions are far
more irregular, and respond to local clusters where one class dominates.
Figure 2.3 shows the results for 1-nearest-neighbor classification: Ŷ is
</p>
<p>assigned the value yℓ of the closest point xℓ to x in the training data. In
this case the regions of classification can be computed relatively easily, and
correspond to a Voronoi tessellation of the training data. Each point xi
has an associated tile bounding the region for which it is the closest input
point. For all points x in the tile, Ĝ(x) = gi. The decision boundary is even
more irregular than before.
The method of k-nearest-neighbor averaging is defined in exactly the
</p>
<p>same way for regression of a quantitative output Y , although k = 1 would
be an unlikely choice.</p>
<p/>
</div>
<div class="page"><p/>
<p>2.3 Least Squares and Nearest Neighbors 15
</p>
<p>15-Nearest Neighbor Classifier
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . .
</p>
<p>..
</p>
<p>. .. .
</p>
<p>. .. .
</p>
<p>. . .. . .
</p>
<p>. . .. . . . .
</p>
<p>. . . . . .. . . . . . .
</p>
<p>. . . . . . . .. . . . . . . . .
</p>
<p>. . . . . . . . .. . . . . . . . .
</p>
<p>. . . . . . . . . .. . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>FIGURE 2.2. The same classification example in two dimensions as in Fig-
ure 2.1. The classes are coded as a binary variable (BLUE = 0, ORANGE = 1) and
then fit by 15-nearest-neighbor averaging as in (2.8). The predicted class is hence
chosen by majority vote amongst the 15-nearest neighbors.
</p>
<p>In Figure 2.2 we see that far fewer training observations are misclassified
than in Figure 2.1. This should not give us too much comfort, though, since
in Figure 2.3 none of the training data are misclassified. A little thought
suggests that for k-nearest-neighbor fits, the error on the training data
should be approximately an increasing function of k, and will always be 0
for k = 1. An independent test set would give us a more satisfactory means
for comparing the different methods.
It appears that k-nearest-neighbor fits have a single parameter, the num-
</p>
<p>ber of neighbors k, compared to the p parameters in least-squares fits. Al-
though this is the case, we will see that the effective number of parameters
of k-nearest neighbors is N/k and is generally bigger than p, and decreases
with increasing k. To get an idea of why, note that if the neighborhoods
were nonoverlapping, there would be N/k neighborhoods and we would fit
one parameter (a mean) in each neighborhood.
It is also clear that we cannot use sum-of-squared errors on the training
</p>
<p>set as a criterion for picking k, since we would always pick k = 1! It would
seem that k-nearest-neighbor methods would be more appropriate for the
mixture Scenario 2 described above, while for Gaussian data the decision
boundaries of k-nearest neighbors would be unnecessarily noisy.</p>
<p/>
</div>
<div class="page"><p/>
<p>16 2. Overview of Supervised Learning
</p>
<p>1&minus;Nearest Neighbor Classifier
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>FIGURE 2.3. The same classification example in two dimensions as in Fig-
ure 2.1. The classes are coded as a binary variable (BLUE = 0, ORANGE = 1), and
then predicted by 1-nearest-neighbor classification.
</p>
<p>2.3.3 From Least Squares to Nearest Neighbors
</p>
<p>The linear decision boundary from least squares is very smooth, and ap-
parently stable to fit. It does appear to rely heavily on the assumption
that a linear decision boundary is appropriate. In language we will develop
shortly, it has low variance and potentially high bias.
On the other hand, the k-nearest-neighbor procedures do not appear to
</p>
<p>rely on any stringent assumptions about the underlying data, and can adapt
to any situation. However, any particular subregion of the decision bound-
ary depends on a handful of input points and their particular positions,
and is thus wiggly and unstable&mdash;high variance and low bias.
Each method has its own situations for which it works best; in particular
</p>
<p>linear regression is more appropriate for Scenario 1 above, while nearest
neighbors are more suitable for Scenario 2. The time has come to expose
the oracle! The data in fact were simulated from a model somewhere be-
tween the two, but closer to Scenario 2. First we generated 10 means mk
from a bivariate Gaussian distribution N((1, 0)T , I) and labeled this class
BLUE. Similarly, 10 more were drawn from N((0, 1)T , I) and labeled class
ORANGE. Then for each class we generated 100 observations as follows: for
each observation, we picked an mk at random with probability 1/10, and</p>
<p/>
</div>
<div class="page"><p/>
<p>2.3 Least Squares and Nearest Neighbors 17
</p>
<p>Degrees of Freedom &minus; N/k
</p>
<p>T
e
</p>
<p>s
t 
</p>
<p>E
rr
</p>
<p>o
r
</p>
<p>0
.1
</p>
<p>0
0
</p>
<p>.1
5
</p>
<p>0
.2
</p>
<p>0
0
</p>
<p>.2
5
</p>
<p>0
.3
</p>
<p>0
</p>
<p>  2   3   5   8  12  18  29  67 200
</p>
<p>151 101  69  45  31  21  11   7   5   3   1
</p>
<p>Train
Test
Bayes
</p>
<p>k &minus;  Number of Nearest Neighbors
</p>
<p>Linear
</p>
<p>FIGURE 2.4. Misclassification curves for the simulation example used in Fig-
ures 2.1, 2.2 and 2.3. A single training sample of size 200 was used, and a test
sample of size 10, 000. The orange curves are test and the blue are training er-
ror for k-nearest-neighbor classification. The results for linear regression are the
bigger orange and blue squares at three degrees of freedom. The purple line is the
optimal Bayes error rate.
</p>
<p>then generated a N(mk, I/5), thus leading to a mixture of Gaussian clus-
ters for each class. Figure 2.4 shows the results of classifying 10,000 new
observations generated from the model. We compare the results for least
squares and those for k-nearest neighbors for a range of values of k.
</p>
<p>A large subset of the most popular techniques in use today are variants of
these two simple procedures. In fact 1-nearest-neighbor, the simplest of all,
captures a large percentage of the market for low-dimensional problems.
The following list describes some ways in which these simple procedures
have been enhanced:
</p>
<p>&bull; Kernel methods use weights that decrease smoothly to zero with dis-
tance from the target point, rather than the effective 0/1 weights used
by k-nearest neighbors.
</p>
<p>&bull; In high-dimensional spaces the distance kernels are modified to em-
phasize some variable more than others.</p>
<p/>
</div>
<div class="page"><p/>
<p>18 2. Overview of Supervised Learning
</p>
<p>&bull; Local regression fits linear models by locally weighted least squares,
rather than fitting constants locally.
</p>
<p>&bull; Linear models fit to a basis expansion of the original inputs allow
arbitrarily complex models.
</p>
<p>&bull; Projection pursuit and neural network models consist of sums of non-
linearly transformed linear models.
</p>
<p>2.4 Statistical Decision Theory
</p>
<p>In this section we develop a small amount of theory that provides a frame-
work for developing models such as those discussed informally so far. We
first consider the case of a quantitative output, and place ourselves in the
world of random variables and probability spaces. Let X &isin; IRp denote a
real valued random input vector, and Y &isin; IR a real valued random out-
put variable, with joint distribution Pr(X,Y ). We seek a function f(X)
for predicting Y given values of the input X. This theory requires a loss
function L(Y, f(X)) for penalizing errors in prediction, and by far the most
common and convenient is squared error loss: L(Y, f(X)) = (Y &minus; f(X))2.
This leads us to a criterion for choosing f ,
</p>
<p>EPE(f) = E(Y &minus; f(X))2 (2.9)
</p>
<p>=
</p>
<p>&int;
[y &minus; f(x)]2 Pr(dx, dy), (2.10)
</p>
<p>the expected (squared) prediction error . By conditioning1 on X, we can
write EPE as
</p>
<p>EPE(f) = EXEY |X
(
[Y &minus; f(X)]2|X
</p>
<p>)
(2.11)
</p>
<p>and we see that it suffices to minimize EPE pointwise:
</p>
<p>f(x) = argmincEY |X
(
[Y &minus; c]2|X = x
</p>
<p>)
. (2.12)
</p>
<p>The solution is
</p>
<p>f(x) = E(Y |X = x), (2.13)
the conditional expectation, also known as the regression function. Thus
the best prediction of Y at any point X = x is the conditional mean, when
best is measured by average squared error.
The nearest-neighbor methods attempt to directly implement this recipe
</p>
<p>using the training data. At each point x, we might ask for the average of all
</p>
<p>1Conditioning here amounts to factoring the joint density Pr(X,Y ) = Pr(Y |X)Pr(X)
</p>
<p>where Pr(Y |X) = Pr(Y,X)/Pr(X), and splitting up the bivariate integral accordingly.</p>
<p/>
</div>
<div class="page"><p/>
<p>2.4 Statistical Decision Theory 19
</p>
<p>those yis with input xi = x. Since there is typically at most one observation
at any point x, we settle for
</p>
<p>f̂(x) = Ave(yi|xi &isin; Nk(x)), (2.14)
</p>
<p>where &ldquo;Ave&rdquo; denotes average, and Nk(x) is the neighborhood containing
the k points in T closest to x. Two approximations are happening here:
</p>
<p>&bull; expectation is approximated by averaging over sample data;
</p>
<p>&bull; conditioning at a point is relaxed to conditioning on some region
&ldquo;close&rdquo; to the target point.
</p>
<p>For large training sample size N , the points in the neighborhood are likely
to be close to x, and as k gets large the average will get more stable.
In fact, under mild regularity conditions on the joint probability distri-
bution Pr(X,Y ), one can show that as N, k &rarr; &infin; such that k/N &rarr; 0,
f̂(x) &rarr; E(Y |X = x). In light of this, why look further, since it seems
we have a universal approximator? We often do not have very large sam-
ples. If the linear or some more structured model is appropriate, then we
can usually get a more stable estimate than k-nearest neighbors, although
such knowledge has to be learned from the data as well. There are other
problems though, sometimes disastrous. In Section 2.5 we see that as the
dimension p gets large, so does the metric size of the k-nearest neighbor-
hood. So settling for nearest neighborhood as a surrogate for conditioning
will fail us miserably. The convergence above still holds, but the rate of
convergence decreases as the dimension increases.
How does linear regression fit into this framework? The simplest explana-
</p>
<p>tion is that one assumes that the regression function f(x) is approximately
linear in its arguments:
</p>
<p>f(x) &asymp; xTβ. (2.15)
This is a model-based approach&mdash;we specify a model for the regression func-
tion. Plugging this linear model for f(x) into EPE (2.9) and differentiating
we can solve for β theoretically:
</p>
<p>β = [E(XXT )]&minus;1E(XY ). (2.16)
</p>
<p>Note we have not conditioned on X; rather we have used our knowledge
of the functional relationship to pool over values of X. The least squares
solution (2.6) amounts to replacing the expectation in (2.16) by averages
over the training data.
So both k-nearest neighbors and least squares end up approximating
</p>
<p>conditional expectations by averages. But they differ dramatically in terms
of model assumptions:
</p>
<p>&bull; Least squares assumes f(x) is well approximated by a globally linear
function.</p>
<p/>
</div>
<div class="page"><p/>
<p>20 2. Overview of Supervised Learning
</p>
<p>&bull; k-nearest neighbors assumes f(x) is well approximated by a locally
constant function.
</p>
<p>Although the latter seems more palatable, we have already seen that we
may pay a price for this flexibility.
Many of the more modern techniques described in this book are model
</p>
<p>based, although far more flexible than the rigid linear model. For example,
additive models assume that
</p>
<p>f(X) =
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>fj(Xj). (2.17)
</p>
<p>This retains the additivity of the linear model, but each coordinate function
fj is arbitrary. It turns out that the optimal estimate for the additive model
uses techniques such as k-nearest neighbors to approximate univariate con-
ditional expectations simultaneously for each of the coordinate functions.
Thus the problems of estimating a conditional expectation in high dimen-
sions are swept away in this case by imposing some (often unrealistic) model
assumptions, in this case additivity.
Are we happy with the criterion (2.11)? What happens if we replace the
</p>
<p>L2 loss function with the L1: E|Y &minus; f(X)|? The solution in this case is the
conditional median,
</p>
<p>f̂(x) = median(Y |X = x), (2.18)
</p>
<p>which is a different measure of location, and its estimates are more robust
than those for the conditional mean. L1 criteria have discontinuities in
their derivatives, which have hindered their widespread use. Other more
resistant loss functions will be mentioned in later chapters, but squared
error is analytically convenient and the most popular.
What do we do when the output is a categorical variable G? The same
</p>
<p>paradigm works here, except we need a different loss function for penalizing
prediction errors. An estimate Ĝ will assume values in G, the set of possible
classes. Our loss function can be represented by a K &times;K matrix L, where
K = card(G). L will be zero on the diagonal and nonnegative elsewhere,
where L(k, ℓ) is the price paid for classifying an observation belonging to
class Gk as Gℓ. Most often we use the zero&ndash;one loss function, where all
misclassifications are charged a single unit. The expected prediction error
is
</p>
<p>EPE = E[L(G, Ĝ(X))], (2.19)
</p>
<p>where again the expectation is taken with respect to the joint distribution
Pr(G,X). Again we condition, and can write EPE as
</p>
<p>EPE = EX
</p>
<p>K&sum;
</p>
<p>k=1
</p>
<p>L[Gk, Ĝ(X)]Pr(Gk|X) (2.20)</p>
<p/>
</div>
<div class="page"><p/>
<p>2.4 Statistical Decision Theory 21
</p>
<p>Bayes Optimal Classifier
</p>
<p>..
. .. . . .
</p>
<p>. . . . .. . . . . .
</p>
<p>. . . . . . .. . . . . . . . .
</p>
<p>. . . . . . . . . .. . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . .. . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . .. . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . .. . . . . . . . . . .
</p>
<p>. . . . . . . . . . .. . . . . . . . . . .
. . . . . . . . . . .. . . . . . . . . . .
. . . . . . . . . . .. . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . .. . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . .. . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . .. . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . .. . . . . . . . . . . . .
. . . . . . . . . . . . .. . . . . . . . . . . . .
. . . . . . . . . . . . .. . . . . . . . . . . . .
. . . . . . . . . . . . .. . . . . . . . . . . . .
. . . . . . . . . . . . .. . . . . . . . . . . . .
. . . . . . . . . . . . .. . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . .. . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . .. . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . .. . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . .. . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . .. . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . .. . . . . . . . . . . .
. . . . . . . . . . . .
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>FIGURE 2.5. The optimal Bayes decision boundary for the simulation example
of Figures 2.1, 2.2 and 2.3. Since the generating density is known for each class,
this boundary can be calculated exactly (Exercise 2.2).
</p>
<p>and again it suffices to minimize EPE pointwise:
</p>
<p>Ĝ(x) = argming&isin;G
</p>
<p>K&sum;
</p>
<p>k=1
</p>
<p>L(Gk, g)Pr(Gk|X = x). (2.21)
</p>
<p>With the 0&ndash;1 loss function this simplifies to
</p>
<p>Ĝ(x) = argming&isin;G [1&minus; Pr(g|X = x)] (2.22)
</p>
<p>or simply
</p>
<p>Ĝ(x) = Gk if Pr(Gk|X = x) = max
g&isin;G
</p>
<p>Pr(g|X = x). (2.23)
</p>
<p>This reasonable solution is known as the Bayes classifier, and says that
we classify to the most probable class, using the conditional (discrete) dis-
tribution Pr(G|X). Figure 2.5 shows the Bayes-optimal decision boundary
for our simulation example. The error rate of the Bayes classifier is called
the Bayes rate.</p>
<p/>
</div>
<div class="page"><p/>
<p>22 2. Overview of Supervised Learning
</p>
<p>Again we see that the k-nearest neighbor classifier directly approximates
this solution&mdash;a majority vote in a nearest neighborhood amounts to ex-
actly this, except that conditional probability at a point is relaxed to con-
ditional probability within a neighborhood of a point, and probabilities are
estimated by training-sample proportions.
Suppose for a two-class problem we had taken the dummy-variable ap-
</p>
<p>proach and coded G via a binary Y , followed by squared error loss estima-
tion. Then f̂(X) = E(Y |X) = Pr(G = G1|X) if G1 corresponded to Y = 1.
Likewise for a K-class problem, E(Yk|X) = Pr(G = Gk|X). This shows
that our dummy-variable regression procedure, followed by classification to
the largest fitted value, is another way of representing the Bayes classifier.
Although this theory is exact, in practice problems can occur, depending
on the regression model used. For example, when linear regression is used,
f̂(X) need not be positive, and we might be suspicious about using it as
an estimate of a probability. We will discuss a variety of approaches to
modeling Pr(G|X) in Chapter 4.
</p>
<p>2.5 Local Methods in High Dimensions
</p>
<p>We have examined two learning techniques for prediction so far: the stable
but biased linear model and the less stable but apparently less biased class
of k-nearest-neighbor estimates. It would seem that with a reasonably large
set of training data, we could always approximate the theoretically optimal
conditional expectation by k-nearest-neighbor averaging, since we should
be able to find a fairly large neighborhood of observations close to any x
and average them. This approach and our intuition breaks down in high
dimensions, and the phenomenon is commonly referred to as the curse
of dimensionality (Bellman, 1961). There are many manifestations of this
problem, and we will examine a few here.
Consider the nearest-neighbor procedure for inputs uniformly distributed
</p>
<p>in a p-dimensional unit hypercube, as in Figure 2.6. Suppose we send out a
hypercubical neighborhood about a target point to capture a fraction r of
the observations. Since this corresponds to a fraction r of the unit volume,
the expected edge length will be ep(r) = r
</p>
<p>1/p. In ten dimensions e10(0.01) =
0.63 and e10(0.1) = 0.80, while the entire range for each input is only 1.0.
So to capture 1% or 10% of the data to form a local average, we must cover
63% or 80% of the range of each input variable. Such neighborhoods are no
longer &ldquo;local.&rdquo; Reducing r dramatically does not help much either, since
the fewer observations we average, the higher is the variance of our fit.
Another consequence of the sparse sampling in high dimensions is that
</p>
<p>all sample points are close to an edge of the sample. Consider N data points
uniformly distributed in a p-dimensional unit ball centered at the origin.
Suppose we consider a nearest-neighbor estimate at the origin. The median</p>
<p/>
</div>
<div class="page"><p/>
<p>2.5 Local Methods in High Dimensions 23
</p>
<p>1
</p>
<p>1
</p>
<p>0
</p>
<p>Unit Cube
</p>
<p>Fraction of Volume
</p>
<p>D
is
</p>
<p>ta
n
c
e
</p>
<p>0.0 0.2 0.4 0.6
</p>
<p>0
.0
</p>
<p>0
.2
</p>
<p>0
.4
</p>
<p>0
.6
</p>
<p>0
.8
</p>
<p>1
.0
</p>
<p>p=1
</p>
<p>p=2
p=3
</p>
<p>p=10
</p>
<p>Neighborhood
</p>
<p>FIGURE 2.6. The curse of dimensionality is well illustrated by a subcubical
neighborhood for uniform data in a unit cube. The figure on the right shows the
side-length of the subcube needed to capture a fraction r of the volume of the data,
for different dimensions p. In ten dimensions we need to cover 80% of the range
of each coordinate to capture 10% of the data.
</p>
<p>distance from the origin to the closest data point is given by the expression
</p>
<p>d(p,N) =
(
1&minus; 1
</p>
<p>2
</p>
<p>1/N)1/p
(2.24)
</p>
<p>(Exercise 2.3). A more complicated expression exists for the mean distance
to the closest point. For N = 500, p = 10 , d(p,N) &asymp; 0.52, more than
halfway to the boundary. Hence most data points are closer to the boundary
of the sample space than to any other data point. The reason that this
presents a problem is that prediction is much more difficult near the edges
of the training sample. One must extrapolate from neighboring sample
points rather than interpolate between them.
Another manifestation of the curse is that the sampling density is pro-
</p>
<p>portional to N1/p, where p is the dimension of the input space and N is the
sample size. Thus, if N1 = 100 represents a dense sample for a single input
problem, then N10 = 100
</p>
<p>10 is the sample size required for the same sam-
pling density with 10 inputs. Thus in high dimensions all feasible training
samples sparsely populate the input space.
Let us construct another uniform example. Suppose we have 1000 train-
</p>
<p>ing examples xi generated uniformly on [&minus;1, 1]p. Assume that the true
relationship between X and Y is
</p>
<p>Y = f(X) = e&minus;8||X||
2
</p>
<p>,
</p>
<p>without any measurement error. We use the 1-nearest-neighbor rule to
predict y0 at the test-point x0 = 0. Denote the training set by T . We can</p>
<p/>
</div>
<div class="page"><p/>
<p>24 2. Overview of Supervised Learning
</p>
<p>compute the expected prediction error at x0 for our procedure, averaging
over all such samples of size 1000. Since the problem is deterministic, this
is the mean squared error (MSE) for estimating f(0):
</p>
<p>MSE(x0) = ET [f(x0)&minus; ŷ0]2
= ET [ŷ0 &minus; ET (ŷ0)]2 + [ET (ŷ0)&minus; f(x0)]2
= VarT (ŷ0) + Bias
</p>
<p>2(ŷ0). (2.25)
</p>
<p>Figure 2.7 illustrates the setup. We have broken down the MSE into two
components that will become familiar as we proceed: variance and squared
bias. Such a decomposition is always possible and often useful, and is known
as the bias&ndash;variance decomposition. Unless the nearest neighbor is at 0,
ŷ0 will be smaller than f(0) in this example, and so the average estimate
will be biased downward. The variance is due to the sampling variance of
the 1-nearest neighbor. In low dimensions and with N = 1000, the nearest
neighbor is very close to 0, and so both the bias and variance are small. As
the dimension increases, the nearest neighbor tends to stray further from
the target point, and both bias and variance are incurred. By p = 10, for
more than 99% of the samples the nearest neighbor is a distance greater
than 0.5 from the origin. Thus as p increases, the estimate tends to be 0
more often than not, and hence the MSE levels off at 1.0, as does the bias,
and the variance starts dropping (an artifact of this example).
Although this is a highly contrived example, similar phenomena occur
</p>
<p>more generally. The complexity of functions of many variables can grow
exponentially with the dimension, and if we wish to be able to estimate
such functions with the same accuracy as function in low dimensions, then
we need the size of our training set to grow exponentially as well. In this
example, the function is a complex interaction of all p variables involved.
The dependence of the bias term on distance depends on the truth, and
</p>
<p>it need not always dominate with 1-nearest neighbor. For example, if the
function always involves only a few dimensions as in Figure 2.8, then the
variance can dominate instead.
Suppose, on the other hand, that we know that the relationship between
</p>
<p>Y and X is linear,
</p>
<p>Y = XTβ + ε, (2.26)
</p>
<p>where ε &sim; N(0, σ2) and we fit the model by least squares to the train-
ing data. For an arbitrary test point x0, we have ŷ0 = x
</p>
<p>T
0 β̂, which can
</p>
<p>be written as ŷ0 = x
T
0 β +
</p>
<p>&sum;N
i=1 ℓi(x0)εi, where ℓi(x0) is the ith element
</p>
<p>of X(XTX)&minus;1x0. Since under this model the least squares estimates are</p>
<p/>
</div>
<div class="page"><p/>
<p>2.5 Local Methods in High Dimensions 25
</p>
<p>X
</p>
<p>f(
X
</p>
<p>)
</p>
<p>-1.0 -0.5 0.0 0.5 1.0
</p>
<p>0
.0
</p>
<p>0
.2
</p>
<p>0
.4
</p>
<p>0
.6
</p>
<p>0
.8
</p>
<p>1
.0
</p>
<p>&bull;
</p>
<p>1-NN in One Dimension
</p>
<p>X1
</p>
<p>X
2
</p>
<p>-1.0 -0.5 0.0 0.5 1.0
</p>
<p>-1
.0
</p>
<p>-0
.5
</p>
<p>0
.0
</p>
<p>0
.5
</p>
<p>1
.0
</p>
<p>&bull; &bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>1-NN in One vs. Two Dimensions
</p>
<p>Dimension
</p>
<p>A
v
e
ra
</p>
<p>g
e
 D
</p>
<p>is
ta
</p>
<p>n
c
e
 t
o
 N
</p>
<p>e
a
re
</p>
<p>s
t 
N
</p>
<p>e
ig
</p>
<p>h
b
o
r
</p>
<p>2 4 6 8 10
</p>
<p>0
.0
</p>
<p>0
.2
</p>
<p>0
.4
</p>
<p>0
.6
</p>
<p>0
.8
</p>
<p>&bull; &bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>Distance to 1-NN vs. Dimension
</p>
<p>Dimension
</p>
<p>M
s
e
</p>
<p>2 4 6 8 10
</p>
<p>0
.0
</p>
<p>0
.2
</p>
<p>0
.4
</p>
<p>0
.6
</p>
<p>0
.8
</p>
<p>1
.0
</p>
<p>&bull; &bull; &bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull;&bull; &bull; &bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull; &bull;
</p>
<p>MSE vs. Dimension
</p>
<p>&bull; MSE
</p>
<p>&bull; Variance
</p>
<p>&bull; Sq. Bias
</p>
<p>FIGURE 2.7. A simulation example, demonstrating the curse of dimensional-
ity and its effect on MSE, bias and variance. The input features are uniformly
distributed in [&minus;1, 1]p for p = 1, . . . , 10 The top left panel shows the target func-
tion (no noise) in IR: f(X) = e&minus;8||X||
</p>
<p>2
</p>
<p>, and demonstrates the error that 1-nearest
neighbor makes in estimating f(0). The training point is indicated by the blue tick
mark. The top right panel illustrates why the radius of the 1-nearest neighborhood
increases with dimension p. The lower left panel shows the average radius of the
1-nearest neighborhoods. The lower-right panel shows the MSE, squared bias and
variance curves as a function of dimension p.</p>
<p/>
</div>
<div class="page"><p/>
<p>26 2. Overview of Supervised Learning
</p>
<p>X
</p>
<p>f(
X
</p>
<p>)
</p>
<p>-1.0 -0.5 0.0 0.5 1.0
</p>
<p>0
1
</p>
<p>2
3
</p>
<p>4
</p>
<p>&bull;
</p>
<p>1-NN in One Dimension
</p>
<p>Dimension
M
</p>
<p>S
E
</p>
<p>2 4 6 8 10
</p>
<p>0
.0
</p>
<p>0
.0
</p>
<p>5
0
</p>
<p>.1
0
</p>
<p>0
.1
</p>
<p>5
0
</p>
<p>.2
0
</p>
<p>0
.2
</p>
<p>5
</p>
<p>&bull; &bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull; &bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull;
&bull; &bull;
</p>
<p>MSE  vs. Dimension
</p>
<p>&bull; MSE
</p>
<p>&bull; Variance
</p>
<p>&bull; Sq. Bias
</p>
<p>FIGURE 2.8. A simulation example with the same setup as in Figure 2.7. Here
the function is constant in all but one dimension: f(X) = 1
</p>
<p>2
(X1 + 1)
</p>
<p>3. The
variance dominates.
</p>
<p>unbiased, we find that
</p>
<p>EPE(x0) = Ey0|x0ET (y0 &minus; ŷ0)2
</p>
<p>= Var(y0|x0) + ET [ŷ0 &minus; ET ŷ0]2 + [ET ŷ0 &minus; xT0 β]2
= Var(y0|x0) + VarT (ŷ0) + Bias2(ŷ0)
= σ2 + ET x
</p>
<p>T
0 (X
</p>
<p>TX)&minus;1x0σ
2 + 02. (2.27)
</p>
<p>Here we have incurred an additional variance σ2 in the prediction error,
since our target is not deterministic. There is no bias, and the variance
depends on x0. If N is large and T were selected at random, and assuming
E(X) = 0, then XTX &rarr; NCov(X) and
</p>
<p>Ex0EPE(x0) &sim; Ex0xT0 Cov(X)&minus;1x0σ2/N + σ2
= trace[Cov(X)&minus;1Cov(x0)]σ
</p>
<p>2/N + σ2
</p>
<p>= σ2(p/N) + σ2. (2.28)
</p>
<p>Here we see that the expected EPE increases linearly as a function of p,
with slope σ2/N . If N is large and/or σ2 is small, this growth in vari-
ance is negligible (0 in the deterministic case). By imposing some heavy
restrictions on the class of models being fitted, we have avoided the curse
of dimensionality. Some of the technical details in (2.27) and (2.28) are
derived in Exercise 2.5.
Figure 2.9 compares 1-nearest neighbor vs. least squares in two situa-
</p>
<p>tions, both of which have the form Y = f(X) + ε, X uniform as before,
and ε &sim; N(0, 1). The sample size is N = 500. For the orange curve, f(x)</p>
<p/>
</div>
<div class="page"><p/>
<p>2.5 Local Methods in High Dimensions 27
</p>
<p>Dimension
</p>
<p>E
P
</p>
<p>E
 R
</p>
<p>a
ti
o
</p>
<p>2 4 6 8 10
</p>
<p>1
.6
</p>
<p>1
.7
</p>
<p>1
.8
</p>
<p>1
.9
</p>
<p>2
.0
</p>
<p>2
.1
</p>
<p>&bull; &bull; &bull; &bull;
&bull;
</p>
<p>&bull; &bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull; &bull; &bull; &bull;
&bull; &bull; &bull;
</p>
<p>&bull; &bull; &bull;
</p>
<p>Expected Prediction Error of 1NN vs. OLS
</p>
<p>&bull; Linear
</p>
<p>&bull; Cubic
</p>
<p>FIGURE 2.9. The curves show the expected prediction error (at x0 = 0) for
1-nearest neighbor relative to least squares for the model Y = f(X) + ε. For the
orange curve, f(x) = x1, while for the blue curve f(x) =
</p>
<p>1
2
(x1 + 1)
</p>
<p>3.
</p>
<p>is linear in the first coordinate, for the blue curve, cubic as in Figure 2.8.
Shown is the relative EPE of 1-nearest neighbor to least squares, which
appears to start at around 2 for the linear case. Least squares is unbiased
in this case, and as discussed above the EPE is slightly above σ2 = 1.
The EPE for 1-nearest neighbor is always above 2, since the variance of
f̂(x0) in this case is at least σ
</p>
<p>2, and the ratio increases with dimension as
the nearest neighbor strays from the target point. For the cubic case, least
squares is biased, which moderates the ratio. Clearly we could manufacture
examples where the bias of least squares would dominate the variance, and
the 1-nearest neighbor would come out the winner.
By relying on rigid assumptions, the linear model has no bias at all and
</p>
<p>negligible variance, while the error in 1-nearest neighbor is substantially
larger. However, if the assumptions are wrong, all bets are off and the
1-nearest neighbor may dominate. We will see that there is a whole spec-
trum of models between the rigid linear models and the extremely flexible
1-nearest-neighbor models, each with their own assumptions and biases,
which have been proposed specifically to avoid the exponential growth in
complexity of functions in high dimensions by drawing heavily on these
assumptions.
Before we delve more deeply, let us elaborate a bit on the concept of
</p>
<p>statistical models and see how they fit into the prediction framework.</p>
<p/>
</div>
<div class="page"><p/>
<p>28 2. Overview of Supervised Learning
</p>
<p>2.6 Statistical Models, Supervised Learning and
Function Approximation
</p>
<p>Our goal is to find a useful approximation f̂(x) to the function f(x) that
underlies the predictive relationship between the inputs and outputs. In the
theoretical setting of Section 2.4, we saw that squared error loss lead us
to the regression function f(x) = E(Y |X = x) for a quantitative response.
The class of nearest-neighbor methods can be viewed as direct estimates
of this conditional expectation, but we have seen that they can fail in at
least two ways:
</p>
<p>&bull; if the dimension of the input space is high, the nearest neighbors need
not be close to the target point, and can result in large errors;
</p>
<p>&bull; if special structure is known to exist, this can be used to reduce both
the bias and the variance of the estimates.
</p>
<p>We anticipate using other classes of models for f(x), in many cases specif-
ically designed to overcome the dimensionality problems, and here we dis-
cuss a framework for incorporating them into the prediction problem.
</p>
<p>2.6.1 A Statistical Model for the Joint Distribution Pr(X, Y )
</p>
<p>Suppose in fact that our data arose from a statistical model
</p>
<p>Y = f(X) + ε, (2.29)
</p>
<p>where the random error ε has E(ε) = 0 and is independent of X. Note that
for this model, f(x) = E(Y |X = x), and in fact the conditional distribution
Pr(Y |X) depends on X only through the conditional mean f(x).
The additive error model is a useful approximation to the truth. For
</p>
<p>most systems the input&ndash;output pairs (X,Y ) will not have a deterministic
relationship Y = f(X). Generally there will be other unmeasured variables
that also contribute to Y , including measurement error. The additive model
assumes that we can capture all these departures from a deterministic re-
lationship via the error ε.
For some problems a deterministic relationship does hold. Many of the
</p>
<p>classification problems studied in machine learning are of this form, where
the response surface can be thought of as a colored map defined in IRp.
The training data consist of colored examples from the map {xi, gi}, and
the goal is to be able to color any point. Here the function is deterministic,
and the randomness enters through the x location of the training points.
For the moment we will not pursue such problems, but will see that they
can be handled by techniques appropriate for the error-based models.
The assumption in (2.29) that the errors are independent and identically
</p>
<p>distributed is not strictly necessary, but seems to be at the back of our mind</p>
<p/>
</div>
<div class="page"><p/>
<p>2.6 Statistical Models, Supervised Learning and Function Approximation 29
</p>
<p>when we average squared errors uniformly in our EPE criterion. With such
a model it becomes natural to use least squares as a data criterion for
model estimation as in (2.1). Simple modifications can be made to avoid
the independence assumption; for example, we can have Var(Y |X = x) =
σ(x), and now both the mean and variance depend on X. In general the
conditional distribution Pr(Y |X) can depend on X in complicated ways,
but the additive error model precludes these.
So far we have concentrated on the quantitative response. Additive error
</p>
<p>models are typically not used for qualitative outputs G; in this case the tar-
get function p(X) is the conditional density Pr(G|X), and this is modeled
directly. For example, for two-class data, it is often reasonable to assume
that the data arise from independent binary trials, with the probability of
one particular outcome being p(X), and the other 1 &minus; p(X). Thus if Y is
the 0&ndash;1 coded version of G, then E(Y |X = x) = p(x), but the variance
depends on x as well: Var(Y |X = x) = p(x)[1&minus; p(x)].
</p>
<p>2.6.2 Supervised Learning
</p>
<p>Before we launch into more statistically oriented jargon, we present the
function-fitting paradigm from a machine learning point of view. Suppose
for simplicity that the errors are additive and that the model Y = f(X)+ε
is a reasonable assumption. Supervised learning attempts to learn f by
example through a teacher. One observes the system under study, both
the inputs and outputs, and assembles a training set of observations T =
(xi, yi), i = 1, . . . , N . The observed input values to the system xi are also
fed into an artificial system, known as a learning algorithm (usually a com-
</p>
<p>puter program), which also produces outputs f̂(xi) in response to the in-
puts. The learning algorithm has the property that it can modify its in-
put/output relationship f̂ in response to differences yi&minus; f̂(xi) between the
original and generated outputs. This process is known as learning by exam-
ple. Upon completion of the learning process the hope is that the artificial
and real outputs will be close enough to be useful for all sets of inputs likely
to be encountered in practice.
</p>
<p>2.6.3 Function Approximation
</p>
<p>The learning paradigm of the previous section has been the motivation
for research into the supervised learning problem in the fields of machine
learning (with analogies to human reasoning) and neural networks (with
biological analogies to the brain). The approach taken in applied mathe-
matics and statistics has been from the perspective of function approxima-
tion and estimation. Here the data pairs {xi, yi} are viewed as points in a
(p+ 1)-dimensional Euclidean space. The function f(x) has domain equal
to the p-dimensional input subspace, and is related to the data via a model</p>
<p/>
</div>
<div class="page"><p/>
<p>30 2. Overview of Supervised Learning
</p>
<p>such as yi = f(xi) + εi. For convenience in this chapter we will assume the
domain is IRp, a p-dimensional Euclidean space, although in general the
inputs can be of mixed type. The goal is to obtain a useful approximation
to f(x) for all x in some region of IRp, given the representations in T .
Although somewhat less glamorous than the learning paradigm, treating
supervised learning as a problem in function approximation encourages the
geometrical concepts of Euclidean spaces and mathematical concepts of
probabilistic inference to be applied to the problem. This is the approach
taken in this book.
Many of the approximations we will encounter have associated a set of
</p>
<p>parameters θ that can be modified to suit the data at hand. For example,
the linear model f(x) = xTβ has θ = β. Another class of useful approxi-
mators can be expressed as linear basis expansions
</p>
<p>fθ(x) =
</p>
<p>K&sum;
</p>
<p>k=1
</p>
<p>hk(x)θk, (2.30)
</p>
<p>where the hk are a suitable set of functions or transformations of the input
vector x. Traditional examples are polynomial and trigonometric expan-
sions, where for example hk might be x
</p>
<p>2
1, x1x
</p>
<p>2
2, cos(x1) and so on. We
</p>
<p>also encounter nonlinear expansions, such as the sigmoid transformation
common to neural network models,
</p>
<p>hk(x) =
1
</p>
<p>1 + exp(&minus;xTβk)
. (2.31)
</p>
<p>We can use least squares to estimate the parameters θ in fθ as we did
for the linear model, by minimizing the residual sum-of-squares
</p>
<p>RSS(θ) =
N&sum;
</p>
<p>i=1
</p>
<p>(yi &minus; fθ(xi))2 (2.32)
</p>
<p>as a function of θ. This seems a reasonable criterion for an additive error
model. In terms of function approximation, we imagine our parameterized
function as a surface in p + 1 space, and what we observe are noisy re-
alizations from it. This is easy to visualize when p = 2 and the vertical
coordinate is the output y, as in Figure 2.10. The noise is in the output
coordinate, so we find the set of parameters such that the fitted surface
gets as close to the observed points as possible, where close is measured by
the sum of squared vertical errors in RSS(θ).
</p>
<p>For the linear model we get a simple closed form solution to the mini-
mization problem. This is also true for the basis function methods, if the
basis functions themselves do not have any hidden parameters. Otherwise
the solution requires either iterative methods or numerical optimization.
While least squares is generally very convenient, it is not the only crite-
</p>
<p>rion used and in some cases would not make much sense. A more general</p>
<p/>
</div>
<div class="page"><p/>
<p>2.6 Statistical Models, Supervised Learning and Function Approximation 31
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
&bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull; &bull; &bull;&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>FIGURE 2.10. Least squares fitting of a function of two inputs. The parameters
of fθ(x) are chosen so as to minimize the sum-of-squared vertical errors.
</p>
<p>principle for estimation is maximum likelihood estimation. Suppose we have
a random sample yi, i = 1, . . . , N from a density Prθ(y) indexed by some
parameters θ. The log-probability of the observed sample is
</p>
<p>L(θ) =
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>log Prθ(yi). (2.33)
</p>
<p>The principle of maximum likelihood assumes that the most reasonable
values for θ are those for which the probability of the observed sample is
largest. Least squares for the additive error model Y = fθ(X) + ε, with
ε &sim; N(0, σ2), is equivalent to maximum likelihood using the conditional
likelihood
</p>
<p>Pr(Y |X, θ) = N(fθ(X), σ2). (2.34)
So although the additional assumption of normality seems more restrictive,
the results are the same. The log-likelihood of the data is
</p>
<p>L(θ) = &minus;N
2
log(2π)&minus;N log σ &minus; 1
</p>
<p>2σ2
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>(yi &minus; fθ(xi))2, (2.35)
</p>
<p>and the only term involving θ is the last, which is RSS(θ) up to a scalar
negative multiplier.
A more interesting example is the multinomial likelihood for the regres-
</p>
<p>sion function Pr(G|X) for a qualitative output G. Suppose we have a model
Pr(G = Gk|X = x) = pk,θ(x), k = 1, . . . ,K for the conditional probabil-
ity of each class given X, indexed by the parameter vector θ. Then the</p>
<p/>
</div>
<div class="page"><p/>
<p>32 2. Overview of Supervised Learning
</p>
<p>log-likelihood (also referred to as the cross-entropy) is
</p>
<p>L(θ) =
N&sum;
</p>
<p>i=1
</p>
<p>log pgi,θ(xi), (2.36)
</p>
<p>and when maximized it delivers values of θ that best conform with the data
in this likelihood sense.
</p>
<p>2.7 Structured Regression Models
</p>
<p>We have seen that although nearest-neighbor and other local methods focus
directly on estimating the function at a point, they face problems in high
dimensions. They may also be inappropriate even in low dimensions in
cases where more structured approaches can make more efficient use of the
data. This section introduces classes of such structured approaches. Before
we proceed, though, we discuss further the need for such classes.
</p>
<p>2.7.1 Difficulty of the Problem
</p>
<p>Consider the RSS criterion for an arbitrary function f ,
</p>
<p>RSS(f) =
N&sum;
</p>
<p>i=1
</p>
<p>(yi &minus; f(xi))2. (2.37)
</p>
<p>Minimizing (2.37) leads to infinitely many solutions: any function f̂ passing
through the training points (xi, yi) is a solution. Any particular solution
chosen might be a poor predictor at test points different from the training
points. If there are multiple observation pairs xi, yiℓ, ℓ = 1, . . . , Ni at each
value of xi, the risk is limited. In this case, the solutions pass through
the average values of the yiℓ at each xi; see Exercise 2.6. The situation is
similar to the one we have already visited in Section 2.4; indeed, (2.37) is
the finite sample version of (2.11) on page 18. If the sample size N were
sufficiently large such that repeats were guaranteed and densely arranged,
it would seem that these solutions might all tend to the limiting conditional
expectation.
In order to obtain useful results for finite N , we must restrict the eligible
</p>
<p>solutions to (2.37) to a smaller set of functions. How to decide on the
nature of the restrictions is based on considerations outside of the data.
These restrictions are sometimes encoded via the parametric representation
of fθ, or may be built into the learning method itself, either implicitly or
explicitly. These restricted classes of solutions are the major topic of this
book. One thing should be clear, though. Any restrictions imposed on f
that lead to a unique solution to (2.37) do not really remove the ambiguity</p>
<p/>
</div>
<div class="page"><p/>
<p>2.8 Classes of Restricted Estimators 33
</p>
<p>caused by the multiplicity of solutions. There are infinitely many possible
restrictions, each leading to a unique solution, so the ambiguity has simply
been transferred to the choice of constraint.
In general the constraints imposed by most learning methods can be
</p>
<p>described as complexity restrictions of one kind or another. This usually
means some kind of regular behavior in small neighborhoods of the input
space. That is, for all input points x sufficiently close to each other in
some metric, f̂ exhibits some special structure such as nearly constant,
linear or low-order polynomial behavior. The estimator is then obtained by
averaging or polynomial fitting in that neighborhood.
The strength of the constraint is dictated by the neighborhood size. The
</p>
<p>larger the size of the neighborhood, the stronger the constraint, and the
more sensitive the solution is to the particular choice of constraint. For
example, local constant fits in infinitesimally small neighborhoods is no
constraint at all; local linear fits in very large neighborhoods is almost a
globally linear model, and is very restrictive.
The nature of the constraint depends on the metric used. Some methods,
</p>
<p>such as kernel and local regression and tree-based methods, directly specify
the metric and size of the neighborhood. The nearest-neighbor methods
discussed so far are based on the assumption that locally the function is
constant; close to a target input x0, the function does not change much, and
so close outputs can be averaged to produce f̂(x0). Other methods such
as splines, neural networks and basis-function methods implicitly define
neighborhoods of local behavior. In Section 5.4.1 we discuss the concept
of an equivalent kernel (see Figure 5.8 on page 157), which describes this
local dependence for any method linear in the outputs. These equivalent
kernels in many cases look just like the explicitly defined weighting kernels
discussed above&mdash;peaked at the target point and falling away smoothly
away from it.
One fact should be clear by now. Any method that attempts to pro-
</p>
<p>duce locally varying functions in small isotropic neighborhoods will run
into problems in high dimensions&mdash;again the curse of dimensionality. And
conversely, all methods that overcome the dimensionality problems have an
associated&mdash;and often implicit or adaptive&mdash;metric for measuring neighbor-
hoods, which basically does not allow the neighborhood to be simultane-
ously small in all directions.
</p>
<p>2.8 Classes of Restricted Estimators
</p>
<p>The variety of nonparametric regression techniques or learning methods fall
into a number of different classes depending on the nature of the restrictions
imposed. These classes are not distinct, and indeed some methods fall in
several classes. Here we give a brief summary, since detailed descriptions</p>
<p/>
</div>
<div class="page"><p/>
<p>34 2. Overview of Supervised Learning
</p>
<p>are given in later chapters. Each of the classes has associated with it one
or more parameters, sometimes appropriately called smoothing parameters,
that control the effective size of the local neighborhood. Here we describe
three broad classes.
</p>
<p>2.8.1 Roughness Penalty and Bayesian Methods
</p>
<p>Here the class of functions is controlled by explicitly penalizing RSS(f)
with a roughness penalty
</p>
<p>PRSS(f ;λ) = RSS(f) + λJ(f). (2.38)
</p>
<p>The user-selected functional J(f) will be large for functions f that vary too
rapidly over small regions of input space. For example, the popular cubic
smoothing spline for one-dimensional inputs is the solution to the penalized
least-squares criterion
</p>
<p>PRSS(f ;λ) =
N&sum;
</p>
<p>i=1
</p>
<p>(yi &minus; f(xi))2 + λ
&int;
[f &prime;&prime;(x)]2dx. (2.39)
</p>
<p>The roughness penalty here controls large values of the second derivative
of f , and the amount of penalty is dictated by λ &ge; 0. For λ = 0 no penalty
is imposed, and any interpolating function will do, while for λ = &infin; only
functions linear in x are permitted.
Penalty functionals J can be constructed for functions in any dimension,
</p>
<p>and special versions can be created to impose special structure. For ex-
ample, additive penalties J(f) =
</p>
<p>&sum;p
j=1 J(fj) are used in conjunction with
</p>
<p>additive functions f(X) =
&sum;p
</p>
<p>j=1 fj(Xj) to create additive models with
smooth coordinate functions. Similarly, projection pursuit regression mod-
els have f(X) =
</p>
<p>&sum;M
m=1 gm(α
</p>
<p>T
mX) for adaptively chosen directions αm, and
</p>
<p>the functions gm can each have an associated roughness penalty.
Penalty function, or regularization methods, express our prior belief that
</p>
<p>the type of functions we seek exhibit a certain type of smooth behavior, and
indeed can usually be cast in a Bayesian framework. The penalty J corre-
sponds to a log-prior, and PRSS(f ;λ) the log-posterior distribution, and
minimizing PRSS(f ;λ) amounts to finding the posterior mode. We discuss
roughness-penalty approaches in Chapter 5 and the Bayesian paradigm in
Chapter 8.
</p>
<p>2.8.2 Kernel Methods and Local Regression
</p>
<p>These methods can be thought of as explicitly providing estimates of the re-
gression function or conditional expectation by specifying the nature of the
local neighborhood, and of the class of regular functions fitted locally. The
local neighborhood is specified by a kernel function Kλ(x0, x) which assigns</p>
<p/>
</div>
<div class="page"><p/>
<p>2.8 Classes of Restricted Estimators 35
</p>
<p>weights to points x in a region around x0 (see Figure 6.1 on page 192). For
example, the Gaussian kernel has a weight function based on the Gaussian
density function
</p>
<p>Kλ(x0, x) =
1
</p>
<p>λ
exp
</p>
<p>[
&minus;||x&minus; x0||
</p>
<p>2
</p>
<p>2λ
</p>
<p>]
(2.40)
</p>
<p>and assigns weights to points that die exponentially with their squared
Euclidean distance from x0. The parameter λ corresponds to the variance
of the Gaussian density, and controls the width of the neighborhood. The
simplest form of kernel estimate is the Nadaraya&ndash;Watson weighted average
</p>
<p>f̂(x0) =
</p>
<p>&sum;N
i=1 Kλ(x0, xi)yi&sum;N
i=1 Kλ(x0, xi)
</p>
<p>. (2.41)
</p>
<p>In general we can define a local regression estimate of f(x0) as fθ̂(x0),
</p>
<p>where θ̂ minimizes
</p>
<p>RSS(fθ, x0) =
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>Kλ(x0, xi)(yi &minus; fθ(xi))2, (2.42)
</p>
<p>and fθ is some parameterized function, such as a low-order polynomial.
Some examples are:
</p>
<p>&bull; fθ(x) = θ0, the constant function; this results in the Nadaraya&ndash;
Watson estimate in (2.41) above.
</p>
<p>&bull; fθ(x) = θ0 + θ1x gives the popular local linear regression model.
Nearest-neighbor methods can be thought of as kernel methods having a
</p>
<p>more data-dependent metric. Indeed, the metric for k-nearest neighbors is
</p>
<p>Kk(x, x0) = I(||x&minus; x0|| &le; ||x(k) &minus; x0||),
</p>
<p>where x(k) is the training observation ranked kth in distance from x0, and
I(S) is the indicator of the set S.
</p>
<p>These methods of course need to be modified in high dimensions, to avoid
the curse of dimensionality. Various adaptations are discussed in Chapter 6.
</p>
<p>2.8.3 Basis Functions and Dictionary Methods
</p>
<p>This class of methods includes the familiar linear and polynomial expan-
sions, but more importantly a wide variety of more flexible models. The
model for f is a linear expansion of basis functions
</p>
<p>fθ(x) =
M&sum;
</p>
<p>m=1
</p>
<p>θmhm(x), (2.43)</p>
<p/>
</div>
<div class="page"><p/>
<p>36 2. Overview of Supervised Learning
</p>
<p>where each of the hm is a function of the input x, and the term linear here
refers to the action of the parameters θ. This class covers a wide variety of
methods. In some cases the sequence of basis functions is prescribed, such
as a basis for polynomials in x of total degree M .
</p>
<p>For one-dimensional x, polynomial splines of degreeK can be represented
by an appropriate sequence of M spline basis functions, determined in turn
byM&minus;K&minus;1 knots. These produce functions that are piecewise polynomials
of degree K between the knots, and joined up with continuity of degree
K &minus; 1 at the knots. As an example consider linear splines, or piecewise
linear functions. One intuitively satisfying basis consists of the functions
b1(x) = 1, b2(x) = x, and bm+2(x) = (x &minus; tm)+, m = 1, . . . ,M &minus; 2,
where tm is the mth knot, and z+ denotes positive part. Tensor products
of spline bases can be used for inputs with dimensions larger than one
(see Section 5.2, and the CART and MARS models in Chapter 9.) The
parameter M controls the degree of the polynomial or the number of knots
in the case of splines.
Radial basis functions are symmetric p-dimensional kernels located at
</p>
<p>particular centroids,
</p>
<p>fθ(x) =
M
&sum;
</p>
<p>m=1
</p>
<p>Kλm(&micro;m, x)θm; (2.44)
</p>
<p>for example, the Gaussian kernel Kλ(&micro;, x) = e
&minus;||x&minus;&micro;||2/2λ is popular.
</p>
<p>Radial basis functions have centroids &micro;m and scales λm that have to
be determined. The spline basis functions have knots. In general we would
like the data to dictate them as well. Including these as parameters changes
the regression problem from a straightforward linear problem to a combi-
natorially hard nonlinear problem. In practice, shortcuts such as greedy
algorithms or two stage processes are used. Section 6.7 describes some such
approaches.
A single-layer feed-forward neural network model with linear output
</p>
<p>weights can be thought of as an adaptive basis function method. The model
has the form
</p>
<p>fθ(x) =
M
&sum;
</p>
<p>m=1
</p>
<p>βmσ(α
T
mx+ bm), (2.45)
</p>
<p>where σ(x) = 1/(1 + e&minus;x) is known as the activation function. Here, as
in the projection pursuit model, the directions αm and the bias terms bm
have to be determined, and their estimation is the meat of the computation.
Details are given in Chapter 11.
These adaptively chosen basis function methods are also known as dictio-
</p>
<p>nary methods, where one has available a possibly infinite set or dictionary
D of candidate basis functions from which to choose, and models are built
up by employing some kind of search mechanism.</p>
<p/>
</div>
<div class="page"><p/>
<p>2.9 Model Selection and the Bias&ndash;Variance Tradeoff 37
</p>
<p>2.9 Model Selection and the Bias&ndash;Variance
Tradeoff
</p>
<p>All the models described above and many others discussed in later chapters
have a smoothing or complexity parameter that has to be determined:
</p>
<p>&bull; the multiplier of the penalty term;
</p>
<p>&bull; the width of the kernel;
</p>
<p>&bull; or the number of basis functions.
</p>
<p>In the case of the smoothing spline, the parameter λ indexes models ranging
from a straight line fit to the interpolating model. Similarly a local degree-
m polynomial model ranges between a degree-m global polynomial when
the window size is infinitely large, to an interpolating fit when the window
size shrinks to zero. This means that we cannot use residual sum-of-squares
on the training data to determine these parameters as well, since we would
always pick those that gave interpolating fits and hence zero residuals. Such
a model is unlikely to predict future data well at all.
The k-nearest-neighbor regression fit f̂k(x0) usefully illustrates the com-
</p>
<p>peting forces that affect the predictive ability of such approximations. Sup-
pose the data arise from a model Y = f(X) + ε, with E(ε) = 0 and
Var(ε) = σ2. For simplicity here we assume that the values of xi in the
sample are fixed in advance (nonrandom). The expected prediction error
at x0, also known as test or generalization error, can be decomposed:
</p>
<p>EPEk(x0) = E[(Y &minus; f̂k(x0))2|X = x0]
= σ2 + [Bias2(f̂k(x0)) + VarT (f̂k(x0))] (2.46)
</p>
<p>= σ2 +
[
f(x0)&minus;
</p>
<p>1
</p>
<p>k
</p>
<p>k&sum;
</p>
<p>ℓ=1
</p>
<p>f(x(ℓ))
]2
</p>
<p>+
σ2
</p>
<p>k
. (2.47)
</p>
<p>The subscripts in parentheses (ℓ) indicate the sequence of nearest neighbors
to x0.
</p>
<p>There are three terms in this expression. The first term σ2 is the ir-
reducible error&mdash;the variance of the new test target&mdash;and is beyond our
control, even if we know the true f(x0).
The second and third terms are under our control, and make up the
</p>
<p>mean squared error of f̂k(x0) in estimating f(x0), which is broken down
into a bias component and a variance component. The bias term is the
squared difference between the true mean f(x0) and the expected value of
</p>
<p>the estimate&mdash;[ET (f̂k(x0))&minus; f(x0)]2&mdash;where the expectation averages the
randomness in the training data. This term will most likely increase with
k, if the true function is reasonably smooth. For small k the few closest
neighbors will have values f(x(ℓ)) close to f(x0), so their average should</p>
<p/>
</div>
<div class="page"><p/>
<p>38 2. Overview of Supervised Learning
</p>
<p>High Bias
</p>
<p>Low Variance
</p>
<p>Low Bias
</p>
<p>High Variance
</p>
<p>P
re
d
ic
ti
o
n
E
rr
o
r
</p>
<p>Model Complexity
</p>
<p>Training Sample
</p>
<p>Test Sample
</p>
<p>Low High
</p>
<p>FIGURE 2.11. Test and training error as a function of model complexity.
</p>
<p>be close to f(x0). As k grows, the neighbors are further away, and then
anything can happen.
The variance term is simply the variance of an average here, and de-
</p>
<p>creases as the inverse of k. So as k varies, there is a bias&ndash;variance tradeoff.
More generally, as the model complexity of our procedure is increased, the
</p>
<p>variance tends to increase and the squared bias tends to decrease. The op-
posite behavior occurs as the model complexity is decreased. For k-nearest
neighbors, the model complexity is controlled by k.
</p>
<p>Typically we would like to choose our model complexity to trade bias
off with variance in such a way as to minimize the test error. An obvious
estimate of test error is the training error 1N
</p>
<p>&sum;
i(yi &minus; ŷi)2. Unfortunately
</p>
<p>training error is not a good estimate of test error, as it does not properly
account for model complexity.
Figure 2.11 shows the typical behavior of the test and training error, as
</p>
<p>model complexity is varied. The training error tends to decrease whenever
we increase the model complexity, that is, whenever we fit the data harder.
However with too much fitting, the model adapts itself too closely to the
training data, and will not generalize well (i.e., have large test error). In
</p>
<p>that case the predictions f̂(x0) will have large variance, as reflected in the
last term of expression (2.46). In contrast, if the model is not complex
enough, it will underfit and may have large bias, again resulting in poor
generalization. In Chapter 7 we discuss methods for estimating the test
error of a prediction method, and hence estimating the optimal amount of
model complexity for a given prediction method and training set.</p>
<p/>
</div>
<div class="page"><p/>
<p>Exercises 39
</p>
<p>Bibliographic Notes
</p>
<p>Some good general books on the learning problem are Duda et al. (2000),
Bishop (1995),(Bishop, 2006), Ripley (1996), Cherkassky and Mulier (2007)
and Vapnik (1996). Parts of this chapter are based on Friedman (1994b).
</p>
<p>Exercises
</p>
<p>Ex. 2.1 Suppose each of K-classes has an associated target tk, which is a
vector of all zeros, except a one in the kth position. Show that classifying to
the largest element of ŷ amounts to choosing the closest target, mink ||tk &minus;
ŷ||, if the elements of ŷ sum to one.
</p>
<p>Ex. 2.2 Show how to compute the Bayes decision boundary for the simula-
tion example in Figure 2.5.
</p>
<p>Ex. 2.3 Derive equation (2.24).
</p>
<p>Ex. 2.4 The edge effect problem discussed on page 23 is not peculiar to
uniform sampling from bounded domains. Consider inputs drawn from a
spherical multinormal distribution X &sim; N(0, Ip). The squared distance
from any sample point to the origin has a χ2p distribution with mean p.
Consider a prediction point x0 drawn from this distribution, and let a =
x0/||x0|| be an associated unit vector. Let zi = aTxi be the projection of
each of the training points on this direction.
Show that the zi are distributed N(0, 1) with expected squared distance
</p>
<p>from the origin 1, while the target point has expected squared distance p
from the origin.
Hence for p = 10, a randomly drawn test point is about 3.1 standard
</p>
<p>deviations from the origin, while all the training points are on average
one standard deviation along direction a. So most prediction points see
themselves as lying on the edge of the training set.
</p>
<p>Ex. 2.5
</p>
<p>(a) Derive equation (2.27). The last line makes use of (3.8) through a
conditioning argument.
</p>
<p>(b) Derive equation (2.28), making use of the cyclic property of the trace
operator [trace(AB) = trace(BA)], and its linearity (which allows us
to interchange the order of trace and expectation).
</p>
<p>Ex. 2.6 Consider a regression problem with inputs xi and outputs yi, and a
parameterized model fθ(x) to be fit by least squares. Show that if there are
observations with tied or identical values of x, then the fit can be obtained
from a reduced weighted least squares problem.</p>
<p/>
</div>
<div class="page"><p/>
<p>40 2. Overview of Supervised Learning
</p>
<p>Ex. 2.7 Suppose we have a sample of N pairs xi, yi drawn i.i.d. from the
distribution characterized as follows:
</p>
<p>xi &sim; h(x), the design density
yi = f(xi) + εi, f is the regression function
</p>
<p>εi &sim; (0, σ2) (mean zero, variance σ2)
</p>
<p>We construct an estimator for f linear in the yi,
</p>
<p>f̂(x0) =
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>ℓi(x0;X )yi,
</p>
<p>where the weights ℓi(x0;X ) do not depend on the yi, but do depend on the
entire training sequence of xi, denoted here by X .
(a) Show that linear regression and k-nearest-neighbor regression are mem-
</p>
<p>bers of this class of estimators. Describe explicitly the weights ℓi(x0;X )
in each of these cases.
</p>
<p>(b) Decompose the conditional mean-squared error
</p>
<p>EY|X (f(x0)&minus; f̂(x0))2
</p>
<p>into a conditional squared bias and a conditional variance component.
Like X , Y represents the entire training sequence of yi.
</p>
<p>(c) Decompose the (unconditional) mean-squared error
</p>
<p>EY,X (f(x0)&minus; f̂(x0))2
</p>
<p>into a squared bias and a variance component.
</p>
<p>(d) Establish a relationship between the squared biases and variances in
the above two cases.
</p>
<p>Ex. 2.8 Compare the classification performance of linear regression and k&ndash;
nearest neighbor classification on the zipcode data. In particular, consider
only the 2&rsquo;s and 3&rsquo;s, and k = 1, 3, 5, 7 and 15. Show both the training and
test error for each choice. The zipcode data are available from the book
website www-stat.stanford.edu/ElemStatLearn.
</p>
<p>Ex. 2.9 Consider a linear regression model with p parameters, fit by least
squares to a set of training data (x1, y1), . . . , (xN , yN ) drawn at random
</p>
<p>from a population. Let β̂ be the least squares estimate. Suppose we have
some test data (x̃1, ỹ1), . . . , (x̃M , ỹM ) drawn at random from the same pop-
</p>
<p>ulation as the training data. If Rtr(β) =
1
N
</p>
<p>&sum;N
1 (yi &minus; βTxi)2 and Rte(β) =
</p>
<p>1
M
</p>
<p>&sum;M
1 (ỹi &minus; βT x̃i)2, prove that
</p>
<p>E[Rtr(β̂)] &le; E[Rte(β̂)],</p>
<p/>
</div>
<div class="page"><p/>
<p>Exercises 41
</p>
<p>where the expectations are over all that is random in each expression. [This
exercise was brought to our attention by Ryan Tibshirani, from a homework
assignment given by Andrew Ng.]</p>
<p/>
</div>
<div class="page"><p/>
<p>3
Linear Methods for Regression
</p>
<p>3.1 Introduction
</p>
<p>A linear regression model assumes that the regression function E(Y |X) is
linear in the inputs X1, . . . , Xp. Linear models were largely developed in
the precomputer age of statistics, but even in today&rsquo;s computer era there
are still good reasons to study and use them. They are simple and often
provide an adequate and interpretable description of how the inputs affect
the output. For prediction purposes they can sometimes outperform fancier
nonlinear models, especially in situations with small numbers of training
cases, low signal-to-noise ratio or sparse data. Finally, linear methods can be
applied to transformations of the inputs and this considerably expands their
scope. These generalizations are sometimes called basis-function methods,
and are discussed in Chapter 5.
In this chapter we describe linear methods for regression, while in the
</p>
<p>next chapter we discuss linear methods for classification. On some topics we
go into considerable detail, as it is our firm belief that an understanding
of linear methods is essential for understanding nonlinear ones. In fact,
many nonlinear techniques are direct generalizations of the linear methods
discussed here.
</p>
<p> &copy; Springer Science+Business Media, LLC 2009 
</p>
<p>T. Hastie et al., The Elements of Statistical Learning, Second Edition, 43
</p>
<p>DOI: 10.1007/b94608_3,</p>
<p/>
</div>
<div class="page"><p/>
<p>44 3. Linear Methods for Regression
</p>
<p>3.2 Linear Regression Models and Least Squares
</p>
<p>As introduced in Chapter 2, we have an input vectorXT=(X1, X2, . . . , Xp),
and want to predict a real-valued output Y . The linear regression model
has the form
</p>
<p>f(X) = β0 +
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>Xjβj . (3.1)
</p>
<p>The linear model either assumes that the regression function E(Y |X) is
linear, or that the linear model is a reasonable approximation. Here the
βj &rsquo;s are unknown parameters or coefficients, and the variables Xj can come
from different sources:
</p>
<p>&bull; quantitative inputs;
</p>
<p>&bull; transformations of quantitative inputs, such as log, square-root or
square;
</p>
<p>&bull; basis expansions, such asX2 = X21 ,X3 = X31 , leading to a polynomial
representation;
</p>
<p>&bull; numeric or &ldquo;dummy&rdquo; coding of the levels of qualitative inputs. For
example, if G is a five-level factor input, we might create Xj , j =
1, . . . , 5, such that Xj = I(G = j). Together this group of Xj repre-
sents the effect of G by a set of level-dependent constants, since in&sum;5
</p>
<p>j=1 Xjβj , one of the Xjs is one, and the others are zero.
</p>
<p>&bull; interactions between variables, for example, X3 = X1 &middot;X2.
</p>
<p>No matter the source of the Xj , the model is linear in the parameters.
Typically we have a set of training data (x1, y1) . . . (xN , yN ) from which
</p>
<p>to estimate the parameters β. Each xi = (xi1, xi2, . . . , xip)
T is a vector
</p>
<p>of feature measurements for the ith case. The most popular estimation
method is least squares, in which we pick the coefficients β=(β0, β1, . . . , βp)
</p>
<p>T
</p>
<p>to minimize the residual sum of squares
</p>
<p>RSS(β) =
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>(yi &minus; f(xi))2
</p>
<p>=
N&sum;
</p>
<p>i=1
</p>
<p>(
yi &minus; β0 &minus;
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>xijβj
</p>
<p>)2
. (3.2)
</p>
<p>From a statistical point of view, this criterion is reasonable if the training
observations (xi, yi) represent independent random draws from their popu-
lation. Even if the xi&rsquo;s were not drawn randomly, the criterion is still valid
if the yi&rsquo;s are conditionally independent given the inputs xi. Figure 3.1
illustrates the geometry of least-squares fitting in the IRp+1-dimensional</p>
<p/>
</div>
<div class="page"><p/>
<p>3.2 Linear Regression Models and Least Squares 45
</p>
<p>&bull;&bull; &bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull; &bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull; &bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull; &bull;
&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>X1
</p>
<p>X2
</p>
<p>Y
</p>
<p>FIGURE 3.1. Linear least squares fitting with X &isin; IR2. We seek the linear
function of X that minimizes the sum of squared residuals from Y .
</p>
<p>space occupied by the pairs (X,Y ). Note that (3.2) makes no assumptions
about the validity of model (3.1); it simply finds the best linear fit to the
data. Least squares fitting is intuitively satisfying no matter how the data
arise; the criterion measures the average lack of fit.
How do we minimize (3.2)? Denote by X the N &times; (p + 1) matrix with
</p>
<p>each row an input vector (with a 1 in the first position), and similarly let
y be the N -vector of outputs in the training set. Then we can write the
residual sum-of-squares as
</p>
<p>RSS(β) = (y &minus;Xβ)T (y &minus;Xβ). (3.3)
This is a quadratic function in the p + 1 parameters. Differentiating with
respect to β we obtain
</p>
<p>&part;RSS
</p>
<p>&part;β
= &minus;2XT (y &minus;Xβ)
</p>
<p>&part;2RSS
</p>
<p>&part;β&part;βT
= 2XTX.
</p>
<p>(3.4)
</p>
<p>Assuming (for the moment) that X has full column rank, and hence XTX
is positive definite, we set the first derivative to zero
</p>
<p>XT (y &minus;Xβ) = 0 (3.5)
to obtain the unique solution
</p>
<p>β̂ = (XTX)&minus;1XTy. (3.6)</p>
<p/>
</div>
<div class="page"><p/>
<p>46 3. Linear Methods for Regression
</p>
<p>x1
</p>
<p>x2
</p>
<p>y
</p>
<p>ŷ
</p>
<p>FIGURE 3.2. The N-dimensional geometry of least squares regression with two
predictors. The outcome vector y is orthogonally projected onto the hyperplane
spanned by the input vectors x1 and x2. The projection ŷ represents the vector
of the least squares predictions
</p>
<p>The predicted values at an input vector x0 are given by f̂(x0) = (1 : x0)
T β̂;
</p>
<p>the fitted values at the training inputs are
</p>
<p>ŷ = Xβ̂ = X(XTX)&minus;1XTy, (3.7)
</p>
<p>where ŷi = f̂(xi). The matrix H = X(X
TX)&minus;1XT appearing in equation
</p>
<p>(3.7) is sometimes called the &ldquo;hat&rdquo; matrix because it puts the hat on y.
Figure 3.2 shows a different geometrical representation of the least squares
</p>
<p>estimate, this time in IRN .We denote the column vectors ofX by x0,x1, . . . ,xp,
with x0 &equiv; 1. For much of what follows, this first column is treated like any
other. These vectors span a subspace of IRN , also referred to as the column
space of X. We minimize RSS(β) = ‖y &minus;Xβ‖2 by choosing β̂ so that the
residual vector y &minus; ŷ is orthogonal to this subspace. This orthogonality is
expressed in (3.5), and the resulting estimate ŷ is hence the orthogonal pro-
jection of y onto this subspace. The hat matrix H computes the orthogonal
projection, and hence it is also known as a projection matrix.
It might happen that the columns of X are not linearly independent, so
</p>
<p>that X is not of full rank. This would occur, for example, if two of the
inputs were perfectly correlated, (e.g., x2 = 3x1). Then X
</p>
<p>TX is singular
</p>
<p>and the least squares coefficients β̂ are not uniquely defined. However,
the fitted values ŷ = Xβ̂ are still the projection of y onto the column
space of X; there is just more than one way to express that projection
in terms of the column vectors of X. The non-full-rank case occurs most
often when one or more qualitative inputs are coded in a redundant fashion.
There is usually a natural way to resolve the non-unique representation,
by recoding and/or dropping redundant columns in X. Most regression
software packages detect these redundancies and automatically implement</p>
<p/>
</div>
<div class="page"><p/>
<p>3.2 Linear Regression Models and Least Squares 47
</p>
<p>some strategy for removing them. Rank deficiencies can also occur in signal
and image analysis, where the number of inputs p can exceed the number
of training cases N . In this case, the features are typically reduced by
filtering or else the fitting is controlled by regularization (Section 5.2.3 and
Chapter 18).
Up to now we have made minimal assumptions about the true distribu-
</p>
<p>tion of the data. In order to pin down the sampling properties of β̂, we now
assume that the observations yi are uncorrelated and have constant vari-
ance σ2, and that the xi are fixed (non random). The variance&ndash;covariance
matrix of the least squares parameter estimates is easily derived from (3.6)
and is given by
</p>
<p>Var(β̂) = (XTX)&minus;1σ2. (3.8)
</p>
<p>Typically one estimates the variance σ2 by
</p>
<p>σ̂2 =
1
</p>
<p>N &minus; p&minus; 1
N&sum;
</p>
<p>i=1
</p>
<p>(yi &minus; ŷi)2.
</p>
<p>The N &minus; p &minus; 1 rather than N in the denominator makes σ̂2 an unbiased
estimate of σ2: E(σ̂2) = σ2.
To draw inferences about the parameters and the model, additional as-
</p>
<p>sumptions are needed. We now assume that (3.1) is the correct model for
the mean; that is, the conditional expectation of Y is linear in X1, . . . , Xp.
We also assume that the deviations of Y around its expectation are additive
and Gaussian. Hence
</p>
<p>Y = E(Y |X1, . . . , Xp) + ε
</p>
<p>= β0 +
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>Xjβj + ε, (3.9)
</p>
<p>where the error ε is a Gaussian random variable with expectation zero and
variance σ2, written ε &sim; N(0, σ2).
Under (3.9), it is easy to show that
</p>
<p>β̂ &sim; N(β, (XTX)&minus;1σ2). (3.10)
</p>
<p>This is a multivariate normal distribution with mean vector and variance&ndash;
covariance matrix as shown. Also
</p>
<p>(N &minus; p&minus; 1)σ̂2 &sim; σ2χ2N&minus;p&minus;1, (3.11)
</p>
<p>a chi-squared distribution with N &minus;p&minus;1 degrees of freedom. In addition β̂
and σ̂2 are statistically independent. We use these distributional properties
to form tests of hypothesis and confidence intervals for the parameters βj .</p>
<p/>
</div>
<div class="page"><p/>
<p>48 3. Linear Methods for Regression
</p>
<p>Z
</p>
<p>T
a
</p>
<p>il 
P
</p>
<p>ro
b
</p>
<p>a
b
</p>
<p>ili
ti
e
</p>
<p>s
</p>
<p>2.0 2.2 2.4 2.6 2.8 3.0
</p>
<p>0
.0
</p>
<p>1
0
</p>
<p>.0
2
</p>
<p>0
.0
</p>
<p>3
0
</p>
<p>.0
4
</p>
<p>0
.0
</p>
<p>5
0
</p>
<p>.0
6
</p>
<p>t30
t100
normal
</p>
<p>FIGURE 3.3. The tail probabilities Pr(|Z| &gt; z) for three distributions, t30, t100
and standard normal. Shown are the appropriate quantiles for testing significance
at the p = 0.05 and 0.01 levels. The difference between t and the standard normal
becomes negligible for N bigger than about 100.
</p>
<p>To test the hypothesis that a particular coefficient βj = 0, we form the
standardized coefficient or Z-score
</p>
<p>zj =
β̂j
</p>
<p>σ̂
&radic;
vj
</p>
<p>, (3.12)
</p>
<p>where vj is the jth diagonal element of (X
TX)&minus;1. Under the null hypothesis
</p>
<p>that βj = 0, zj is distributed as tN&minus;p&minus;1 (a t distribution with N &minus; p &minus; 1
degrees of freedom), and hence a large (absolute) value of zj will lead to
rejection of this null hypothesis. If σ̂ is replaced by a known value σ, then
zj would have a standard normal distribution. The difference between the
tail quantiles of a t-distribution and a standard normal become negligible
as the sample size increases, and so we typically use the normal quantiles
(see Figure 3.3).
Often we need to test for the significance of groups of coefficients simul-
</p>
<p>taneously. For example, to test if a categorical variable with k levels can
be excluded from a model, we need to test whether the coefficients of the
dummy variables used to represent the levels can all be set to zero. Here
we use the F statistic,
</p>
<p>F =
(RSS0 &minus; RSS1)/(p1 &minus; p0)
</p>
<p>RSS1/(N &minus; p1 &minus; 1)
, (3.13)
</p>
<p>where RSS1 is the residual sum-of-squares for the least squares fit of the big-
ger model with p1+1 parameters, and RSS0 the same for the nested smaller
model with p0+1 parameters, having p1&minus;p0 parameters constrained to be</p>
<p/>
</div>
<div class="page"><p/>
<p>3.2 Linear Regression Models and Least Squares 49
</p>
<p>zero. The F statistic measures the change in residual sum-of-squares per
additional parameter in the bigger model, and it is normalized by an esti-
mate of σ2. Under the Gaussian assumptions, and the null hypothesis that
the smaller model is correct, the F statistic will have a Fp1&minus;p0,N&minus;p1&minus;1 dis-
tribution. It can be shown (Exercise 3.1) that the zj in (3.12) are equivalent
to the F statistic for dropping the single coefficient βj from the model. For
large N , the quantiles of Fp1&minus;p0,N&minus;p1&minus;1 approach those of χ
</p>
<p>2
p1&minus;p0/(p1&minus;p0).
</p>
<p>Similarly, we can isolate βj in (3.10) to obtain a 1&minus;2α confidence interval
for βj :
</p>
<p>(β̂j &minus; z(1&minus;α)v
1
2
j σ̂, β̂j + z
</p>
<p>(1&minus;α)v
1
2
j σ̂). (3.14)
</p>
<p>Here z(1&minus;α) is the 1&minus; α percentile of the normal distribution:
</p>
<p>z(1&minus;0.025) = 1.96,
z(1&minus;.05) = 1.645, etc.
</p>
<p>Hence the standard practice of reporting β̂ &plusmn; 2 &middot; se(β̂) amounts to an ap-
proximate 95% confidence interval. Even if the Gaussian error assumption
does not hold, this interval will be approximately correct, with its coverage
approaching 1&minus; 2α as the sample size N &rarr; &infin;.
</p>
<p>In a similar fashion we can obtain an approximate confidence set for the
entire parameter vector β, namely
</p>
<p>Cβ = {β|(β̂ &minus; β)TXTX(β̂ &minus; β) &le; σ̂2χ2p+1
(1&minus;α)}, (3.15)
</p>
<p>where χ2ℓ
(1&minus;α)
</p>
<p>is the 1 &minus; α percentile of the chi-squared distribution on ℓ
degrees of freedom: for example, χ25
</p>
<p>(1&minus;0.05)
= 11.1, χ25
</p>
<p>(1&minus;0.1)
= 9.2. This
</p>
<p>confidence set for β generates a corresponding confidence set for the true
function f(x) = xTβ, namely {xTβ|β &isin; Cβ} (Exercise 3.2; see also Fig-
ure 5.4 in Section 5.2.2 for examples of confidence bands for functions).
</p>
<p>3.2.1 Example: Prostate Cancer
</p>
<p>The data for this example come from a study by Stamey et al. (1989). They
examined the correlation between the level of prostate-specific antigen and
a number of clinical measures in men who were about to receive a radical
prostatectomy. The variables are log cancer volume (lcavol), log prostate
weight (lweight), age, log of the amount of benign prostatic hyperplasia
(lbph), seminal vesicle invasion (svi), log of capsular penetration (lcp),
Gleason score (gleason), and percent of Gleason scores 4 or 5 (pgg45).
The correlation matrix of the predictors given in Table 3.1 shows many
strong correlations. Figure 1.1 (page 3) of Chapter 1 is a scatterplot matrix
showing every pairwise plot between the variables. We see that svi is a
binary variable, and gleason is an ordered categorical variable. We see, for</p>
<p/>
</div>
<div class="page"><p/>
<p>50 3. Linear Methods for Regression
</p>
<p>TABLE 3.1. Correlations of predictors in the prostate cancer data.
</p>
<p>lcavol lweight age lbph svi lcp gleason
</p>
<p>lweight 0.300
age 0.286 0.317
lbph 0.063 0.437 0.287
svi 0.593 0.181 0.129 &minus;0.139
lcp 0.692 0.157 0.173 &minus;0.089 0.671
</p>
<p>gleason 0.426 0.024 0.366 0.033 0.307 0.476
pgg45 0.483 0.074 0.276 &minus;0.030 0.481 0.663 0.757
</p>
<p>TABLE 3.2. Linear model fit to the prostate cancer data. The Z score is the
coefficient divided by its standard error (3.12). Roughly a Z score larger than two
in absolute value is significantly nonzero at the p = 0.05 level.
</p>
<p>Term Coefficient Std. Error Z Score
</p>
<p>Intercept 2.46 0.09 27.60
lcavol 0.68 0.13 5.37
lweight 0.26 0.10 2.75
</p>
<p>age &minus;0.14 0.10 &minus;1.40
lbph 0.21 0.10 2.06
svi 0.31 0.12 2.47
lcp &minus;0.29 0.15 &minus;1.87
</p>
<p>gleason &minus;0.02 0.15 &minus;0.15
pgg45 0.27 0.15 1.74
</p>
<p>example, that both lcavol and lcp show a strong relationship with the
response lpsa, and with each other. We need to fit the effects jointly to
untangle the relationships between the predictors and the response.
We fit a linear model to the log of prostate-specific antigen, lpsa, after
</p>
<p>first standardizing the predictors to have unit variance. We randomly split
the dataset into a training set of size 67 and a test set of size 30. We ap-
plied least squares estimation to the training set, producing the estimates,
standard errors and Z-scores shown in Table 3.2. The Z-scores are defined
in (3.12), and measure the effect of dropping that variable from the model.
A Z-score greater than 2 in absolute value is approximately significant at
the 5% level. (For our example, we have nine parameters, and the 0.025 tail
quantiles of the t67&minus;9 distribution are &plusmn;2.002!) The predictor lcavol shows
the strongest effect, with lweight and svi also strong. Notice that lcp is
not significant, once lcavol is in the model (when used in a model without
lcavol, lcp is strongly significant). We can also test for the exclusion of
a number of terms at once, using the F -statistic (3.13). For example, we
consider dropping all the non-significant terms in Table 3.2, namely age,</p>
<p/>
</div>
<div class="page"><p/>
<p>3.2 Linear Regression Models and Least Squares 51
</p>
<p>lcp, gleason, and pgg45. We get
</p>
<p>F =
(32.81&minus; 29.43)/(9&minus; 5)
</p>
<p>29.43/(67&minus; 9) = 1.67, (3.16)
</p>
<p>which has a p-value of 0.17 (Pr(F4,58 &gt; 1.67) = 0.17), and hence is not
significant.
The mean prediction error on the test data is 0.521. In contrast, predic-
</p>
<p>tion using the mean training value of lpsa has a test error of 1.057, which
is called the &ldquo;base error rate.&rdquo; Hence the linear model reduces the base
error rate by about 50%. We will return to this example later to compare
various selection and shrinkage methods.
</p>
<p>3.2.2 The Gauss&ndash;Markov Theorem
</p>
<p>One of the most famous results in statistics asserts that the least squares
estimates of the parameters β have the smallest variance among all linear
unbiased estimates. We will make this precise here, and also make clear
that the restriction to unbiased estimates is not necessarily a wise one. This
observation will lead us to consider biased estimates such as ridge regression
later in the chapter. We focus on estimation of any linear combination of
the parameters θ = aTβ; for example, predictions f(x0) = x
</p>
<p>T
0 β are of this
</p>
<p>form. The least squares estimate of aTβ is
</p>
<p>θ̂ = aT β̂ = aT (XTX)&minus;1XTy. (3.17)
</p>
<p>Considering X to be fixed, this is a linear function cT0 y of the response
</p>
<p>vector y. If we assume that the linear model is correct, aT β̂ is unbiased
since
</p>
<p>E(aT β̂) = E(aT (XTX)&minus;1XTy)
</p>
<p>= aT (XTX)&minus;1XTXβ
</p>
<p>= aTβ. (3.18)
</p>
<p>The Gauss&ndash;Markov theorem states that if we have any other linear estima-
tor θ̃ = cTy that is unbiased for aTβ, that is, E(cTy) = aTβ, then
</p>
<p>Var(aT β̂) &le; Var(cTy). (3.19)
</p>
<p>The proof (Exercise 3.3) uses the triangle inequality. For simplicity we have
stated the result in terms of estimation of a single parameter aTβ, but with
a few more definitions one can state it in terms of the entire parameter
vector β (Exercise 3.3).
</p>
<p>Consider the mean squared error of an estimator θ̃ in estimating θ:
</p>
<p>MSE(θ̃) = E(θ̃ &minus; θ)2
= Var(θ̃) + [E(θ̃)&minus; θ]2. (3.20)</p>
<p/>
</div>
<div class="page"><p/>
<p>52 3. Linear Methods for Regression
</p>
<p>The first term is the variance, while the second term is the squared bias.
The Gauss-Markov theorem implies that the least squares estimator has the
smallest mean squared error of all linear estimators with no bias. However,
there may well exist a biased estimator with smaller mean squared error.
Such an estimator would trade a little bias for a larger reduction in variance.
Biased estimates are commonly used. Any method that shrinks or sets to
zero some of the least squares coefficients may result in a biased estimate.
We discuss many examples, including variable subset selection and ridge
regression, later in this chapter. From a more pragmatic point of view, most
models are distortions of the truth, and hence are biased; picking the right
model amounts to creating the right balance between bias and variance.
We go into these issues in more detail in Chapter 7.
Mean squared error is intimately related to prediction accuracy, as dis-
</p>
<p>cussed in Chapter 2. Consider the prediction of the new response at input
x0,
</p>
<p>Y0 = f(x0) + ε0. (3.21)
</p>
<p>Then the expected prediction error of an estimate f̃(x0) = x
T
0 β̃ is
</p>
<p>E(Y0 &minus; f̃(x0))2 = σ2 + E(xT0 β̃ &minus; f(x0))2
= σ2 +MSE(f̃(x0)). (3.22)
</p>
<p>Therefore, expected prediction error and mean squared error differ only by
the constant σ2, representing the variance of the new observation y0.
</p>
<p>3.2.3 Multiple Regression from Simple Univariate Regression
</p>
<p>The linear model (3.1) with p &gt; 1 inputs is called the multiple linear
regression model. The least squares estimates (3.6) for this model are best
understood in terms of the estimates for the univariate (p = 1) linear
model, as we indicate in this section.
Suppose first that we have a univariate model with no intercept, that is,
</p>
<p>Y = Xβ + ε. (3.23)
</p>
<p>The least squares estimate and residuals are
</p>
<p>β̂ =
</p>
<p>&sum;N
1 xiyi&sum;N
1 x
</p>
<p>2
i
</p>
<p>,
</p>
<p>ri = yi &minus; xiβ̂.
(3.24)
</p>
<p>In convenient vector notation, we let y = (y1, . . . , yN )
T , x = (x1, . . . , xN )
</p>
<p>T
</p>
<p>and define
</p>
<p>〈x,y〉 =
N&sum;
</p>
<p>i=1
</p>
<p>xiyi,
</p>
<p>= xTy, (3.25)</p>
<p/>
</div>
<div class="page"><p/>
<p>3.2 Linear Regression Models and Least Squares 53
</p>
<p>the inner product between x and y1. Then we can write
</p>
<p>β̂ =
〈x,y〉
〈x,x〉 ,
</p>
<p>r = y &minus; xβ̂.
(3.26)
</p>
<p>As we will see, this simple univariate regression provides the building block
for multiple linear regression. Suppose next that the inputs x1,x2, . . . ,xp
(the columns of the data matrix X) are orthogonal; that is 〈xj ,xk〉 = 0
for all j �= k. Then it is easy to check that the multiple least squares esti-
mates β̂j are equal to 〈xj ,y〉/〈xj ,xj〉&mdash;the univariate estimates. In other
words, when the inputs are orthogonal, they have no effect on each other&rsquo;s
parameter estimates in the model.
Orthogonal inputs occur most often with balanced, designed experiments
</p>
<p>(where orthogonality is enforced), but almost never with observational
data. Hence we will have to orthogonalize them in order to carry this idea
further. Suppose next that we have an intercept and a single input x. Then
the least squares coefficient of x has the form
</p>
<p>β̂1 =
〈x&minus; x̄1,y〉
</p>
<p>〈x&minus; x̄1,x&minus; x̄1〉 , (3.27)
</p>
<p>where x̄ =
&sum;
</p>
<p>i xi/N , and 1 = x0, the vector of N ones. We can view the
estimate (3.27) as the result of two applications of the simple regression
(3.26). The steps are:
</p>
<p>1. regress x on 1 to produce the residual z = x&minus; x̄1;
</p>
<p>2. regress y on the residual z to give the coefficient β̂1.
</p>
<p>In this procedure, &ldquo;regress b on a&rdquo; means a simple univariate regression of b
on a with no intercept, producing coefficient γ̂ = 〈a,b〉/〈a,a〉 and residual
vector b&minus; γ̂a. We say that b is adjusted for a, or is &ldquo;orthogonalized&rdquo; with
respect to a.
Step 1 orthogonalizes x with respect to x0 = 1. Step 2 is just a simple
</p>
<p>univariate regression, using the orthogonal predictors 1 and z. Figure 3.4
shows this process for two general inputs x1 and x2. The orthogonalization
does not change the subspace spanned by x1 and x2, it simply produces an
orthogonal basis for representing it.
This recipe generalizes to the case of p inputs, as shown in Algorithm 3.1.
</p>
<p>Note that the inputs z0, . . . , zj&minus;1 in step 2 are orthogonal, hence the simple
regression coefficients computed there are in fact also the multiple regres-
sion coefficients.
</p>
<p>1The inner-product notation is suggestive of generalizations of linear regression to
</p>
<p>different metric spaces, as well as to probability spaces.</p>
<p/>
</div>
<div class="page"><p/>
<p>54 3. Linear Methods for Regression
</p>
<p>x1
</p>
<p>x2
</p>
<p>y
</p>
<p>ŷ
</p>
<p>zzzzz
</p>
<p>FIGURE 3.4. Least squares regression by orthogonalization of the inputs. The
vector x2 is regressed on the vector x1, leaving the residual vector z. The regres-
sion of y on z gives the multiple regression coefficient of x2. Adding together the
projections of y on each of x1 and z gives the least squares fit ŷ.
</p>
<p>Algorithm 3.1 Regression by Successive Orthogonalization.
</p>
<p>1. Initialize z0 = x0 = 1.
</p>
<p>2. For j = 1, 2, . . . , p
</p>
<p>Regress xj on z0, z1, . . . , , zj&minus;1 to produce coefficients γ̂ℓj =
〈zℓ,xj〉/〈zℓ, zℓ〉, ℓ = 0, . . . , j &minus; 1 and residual vector zj =
xj &minus;
</p>
<p>&sum;j&minus;1
k=0 γ̂kjzk.
</p>
<p>3. Regress y on the residual zp to give the estimate β̂p.
</p>
<p>The result of this algorithm is
</p>
<p>β̂p =
〈zp,y〉
〈zp, zp〉
</p>
<p>. (3.28)
</p>
<p>Re-arranging the residual in step 2, we can see that each of the xj is a linear
combination of the zk, k &le; j. Since the zj are all orthogonal, they form
a basis for the column space of X, and hence the least squares projection
onto this subspace is ŷ. Since zp alone involves xp (with coefficient 1), we
see that the coefficient (3.28) is indeed the multiple regression coefficient of
y on xp. This key result exposes the effect of correlated inputs in multiple
regression. Note also that by rearranging the xj , any one of them could
be in the last position, and a similar results holds. Hence stated more
generally, we have shown that the jth multiple regression coefficient is the
univariate regression coefficient of y on xj&middot;012...(j&minus;1)(j+1)...,p, the residual
after regressing xj on x0,x1, . . . ,xj&minus;1,xj+1, . . . ,xp:</p>
<p/>
</div>
<div class="page"><p/>
<p>3.2 Linear Regression Models and Least Squares 55
</p>
<p>The multiple regression coefficient β̂j represents the additional
contribution of xj on y, after xj has been adjusted for x0,x1, . . . ,xj&minus;1,
xj+1, . . . ,xp.
</p>
<p>If xp is highly correlated with some of the other xk&rsquo;s, the residual vector
</p>
<p>zp will be close to zero, and from (3.28) the coefficient β̂p will be very
unstable. This will be true for all the variables in the correlated set. In
such situations, we might have all the Z-scores (as in Table 3.2) be small&mdash;
any one of the set can be deleted&mdash;yet we cannot delete them all. From
(3.28) we also obtain an alternate formula for the variance estimates (3.8),
</p>
<p>Var(β̂p) =
σ2
</p>
<p>〈zp, zp〉
=
</p>
<p>σ2
</p>
<p>‖zp‖2
. (3.29)
</p>
<p>In other words, the precision with which we can estimate β̂p depends on
the length of the residual vector zp; this represents how much of xp is
unexplained by the other xk&rsquo;s.
Algorithm 3.1 is known as the Gram&ndash;Schmidt procedure for multiple
</p>
<p>regression, and is also a useful numerical strategy for computing the esti-
mates. We can obtain from it not just β̂p, but also the entire multiple least
squares fit, as shown in Exercise 3.4.
We can represent step 2 of Algorithm 3.1 in matrix form:
</p>
<p>X = ZΓ, (3.30)
</p>
<p>where Z has as columns the zj (in order), and Γ is the upper triangular ma-
trix with entries γ̂kj . Introducing the diagonal matrix D with jth diagonal
entry Djj = ‖zj‖, we get
</p>
<p>X = ZD&minus;1DΓ
</p>
<p>= QR, (3.31)
</p>
<p>the so-called QR decomposition of X. Here Q is an N &times; (p+1) orthogonal
matrix, QTQ = I, and R is a (p+ 1)&times; (p+ 1) upper triangular matrix.
The QR decomposition represents a convenient orthogonal basis for the
</p>
<p>column space of X. It is easy to see, for example, that the least squares
solution is given by
</p>
<p>β̂ = R&minus;1QTy, (3.32)
</p>
<p>ŷ = QQTy. (3.33)
</p>
<p>Equation (3.32) is easy to solve because R is upper triangular
(Exercise 3.4).</p>
<p/>
</div>
<div class="page"><p/>
<p>56 3. Linear Methods for Regression
</p>
<p>3.2.4 Multiple Outputs
</p>
<p>Suppose we have multiple outputs Y1, Y2, . . . , YK that we wish to predict
from our inputs X0, X1, X2, . . . , Xp. We assume a linear model for each
output
</p>
<p>Yk = β0k +
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>Xjβjk + εk (3.34)
</p>
<p>= fk(X) + εk. (3.35)
</p>
<p>With N training cases we can write the model in matrix notation
</p>
<p>Y = XB+E. (3.36)
</p>
<p>Here Y is the N&times;K response matrix, with ik entry yik, X is the N&times;(p+1)
input matrix, B is the (p + 1) &times; K matrix of parameters and E is the
N &times;K matrix of errors. A straightforward generalization of the univariate
loss function (3.2) is
</p>
<p>RSS(B) =
K&sum;
</p>
<p>k=1
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>(yik &minus; fk(xi))2 (3.37)
</p>
<p>= tr[(Y &minus;XB)T (Y &minus;XB)]. (3.38)
</p>
<p>The least squares estimates have exactly the same form as before
</p>
<p>B̂ = (XTX)&minus;1XTY. (3.39)
</p>
<p>Hence the coefficients for the kth outcome are just the least squares es-
timates in the regression of yk on x0,x1, . . . ,xp. Multiple outputs do not
affect one another&rsquo;s least squares estimates.
If the errors ε = (ε1, . . . , εK) in (3.34) are correlated, then it might seem
</p>
<p>appropriate to modify (3.37) in favor of a multivariate version. Specifically,
suppose Cov(ε) = Σ, then the multivariate weighted criterion
</p>
<p>RSS(B;Σ) =
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>(yi &minus; f(xi))TΣ&minus;1(yi &minus; f(xi)) (3.40)
</p>
<p>arises naturally from multivariate Gaussian theory. Here f(x) is the vector
function (f1(x), . . . , fK(x))
</p>
<p>T , and yi the vector of K responses for obser-
vation i. However, it can be shown that again the solution is given by
(3.39); K separate regressions that ignore the correlations (Exercise 3.11).
If the Σi vary among observations, then this is no longer the case, and the
solution for B no longer decouples.
In Section 3.7 we pursue the multiple outcome problem, and consider
</p>
<p>situations where it does pay to combine the regressions.</p>
<p/>
</div>
<div class="page"><p/>
<p>3.3 Subset Selection 57
</p>
<p>3.3 Subset Selection
</p>
<p>There are two reasons why we are often not satisfied with the least squares
estimates (3.6).
</p>
<p>&bull; The first is prediction accuracy: the least squares estimates often have
low bias but large variance. Prediction accuracy can sometimes be
improved by shrinking or setting some coefficients to zero. By doing
so we sacrifice a little bit of bias to reduce the variance of the predicted
values, and hence may improve the overall prediction accuracy.
</p>
<p>&bull; The second reason is interpretation. With a large number of predic-
tors, we often would like to determine a smaller subset that exhibit
the strongest effects. In order to get the &ldquo;big picture,&rdquo; we are willing
to sacrifice some of the small details.
</p>
<p>In this section we describe a number of approaches to variable subset selec-
tion with linear regression. In later sections we discuss shrinkage and hybrid
approaches for controlling variance, as well as other dimension-reduction
strategies. These all fall under the general heading model selection. Model
selection is not restricted to linear models; Chapter 7 covers this topic in
some detail.
With subset selection we retain only a subset of the variables, and elim-
</p>
<p>inate the rest from the model. Least squares regression is used to estimate
the coefficients of the inputs that are retained. There are a number of dif-
ferent strategies for choosing the subset.
</p>
<p>3.3.1 Best-Subset Selection
</p>
<p>Best subset regression finds for each k &isin; {0, 1, 2, . . . , p} the subset of size k
that gives smallest residual sum of squares (3.2). An efficient algorithm&mdash;
the leaps and bounds procedure (Furnival and Wilson, 1974)&mdash;makes this
feasible for p as large as 30 or 40. Figure 3.5 shows all the subset models
for the prostate cancer example. The lower boundary represents the models
that are eligible for selection by the best-subsets approach. Note that the
best subset of size 2, for example, need not include the variable that was
in the best subset of size 1 (for this example all the subsets are nested).
The best-subset curve (red lower boundary in Figure 3.5) is necessarily
decreasing, so cannot be used to select the subset size k. The question of
how to choose k involves the tradeoff between bias and variance, along with
the more subjective desire for parsimony. There are a number of criteria
that one may use; typically we choose the smallest model that minimizes
an estimate of the expected prediction error.
Many of the other approaches that we discuss in this chapter are similar,
</p>
<p>in that they use the training data to produce a sequence of models varying
in complexity and indexed by a single parameter. In the next section we use</p>
<p/>
</div>
<div class="page"><p/>
<p>58 3. Linear Methods for Regression
</p>
<p>Subset Size k
</p>
<p>R
e
</p>
<p>s
id
</p>
<p>u
a
</p>
<p>l 
S
</p>
<p>u
m
</p>
<p>&minus;
o
</p>
<p>f&minus;
S
</p>
<p>q
u
</p>
<p>a
re
</p>
<p>s
</p>
<p>0
2
</p>
<p>0
4
</p>
<p>0
6
</p>
<p>0
8
</p>
<p>0
1
</p>
<p>0
0
</p>
<p>0 1 2 3 4 5 6 7 8
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;&bull;
&bull;
</p>
<p>&bull;
&bull;&bull;
</p>
<p>&bull;&bull;
&bull;&bull;&bull;&bull;
&bull;
&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;
&bull;&bull;&bull;
&bull;&bull;
&bull;
&bull;
</p>
<p>&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;
&bull;&bull;
&bull;
</p>
<p>&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;
</p>
<p>&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;
&bull;&bull;
</p>
<p>&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;
&bull;&bull;
&bull;&bull;&bull;&bull;&bull;
&bull;&bull;
</p>
<p>&bull;&bull;&bull;&bull;&bull;&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull; &bull; &bull; &bull; &bull; &bull; &bull;
</p>
<p>FIGURE 3.5. All possible subset models for the prostate cancer example. At
each subset size is shown the residual sum-of-squares for each model of that size.
</p>
<p>cross-validation to estimate prediction error and select k; the AIC criterion
is a popular alternative. We defer more detailed discussion of these and
other approaches to Chapter 7.
</p>
<p>3.3.2 Forward- and Backward-Stepwise Selection
</p>
<p>Rather than search through all possible subsets (which becomes infeasible
for pmuch larger than 40), we can seek a good path through them. Forward-
stepwise selection starts with the intercept, and then sequentially adds into
the model the predictor that most improves the fit. With many candidate
predictors, this might seem like a lot of computation; however, clever up-
dating algorithms can exploit the QR decomposition for the current fit to
rapidly establish the next candidate (Exercise 3.9). Like best-subset re-
gression, forward stepwise produces a sequence of models indexed by k, the
subset size, which must be determined.
Forward-stepwise selection is a greedy algorithm, producing a nested se-
</p>
<p>quence of models. In this sense it might seem sub-optimal compared to
best-subset selection. However, there are several reasons why it might be
preferred:</p>
<p/>
</div>
<div class="page"><p/>
<p>3.3 Subset Selection 59
</p>
<p>&bull; Computational; for large p we cannot compute the best subset se-
quence, but we can always compute the forward stepwise sequence
(even when p ≫ N).
</p>
<p>&bull; Statistical; a price is paid in variance for selecting the best subset
of each size; forward stepwise is a more constrained search, and will
have lower variance, but perhaps more bias.
</p>
<p>0 5 10 15 20 25 30
</p>
<p>0
.6
</p>
<p>5
0
</p>
<p>.7
0
</p>
<p>0
.7
</p>
<p>5
0
</p>
<p>.8
0
</p>
<p>0
.8
</p>
<p>5
0
</p>
<p>.9
0
</p>
<p>0
.9
</p>
<p>5
</p>
<p>Best Subset
Forward Stepwise
Backward Stepwise
Forward Stagewise
</p>
<p>E
||β̂
</p>
<p>(k
)
&minus;
</p>
<p>β
||2
</p>
<p>Subset Size k
</p>
<p>FIGURE 3.6. Comparison of four subset-selection techniques on a simulated lin-
ear regression problem Y = XTβ+ ε. There are N = 300 observations on p = 31
standard Gaussian variables, with pairwise correlations all equal to 0.85. For 10 of
the variables, the coefficients are drawn at random from a N(0, 0.4) distribution;
the rest are zero. The noise ε &sim; N(0, 6.25), resulting in a signal-to-noise ratio of
0.64. Results are averaged over 50 simulations. Shown is the mean-squared error
of the estimated coefficient β̂(k) at each step from the true β.
</p>
<p>Backward-stepwise selection starts with the full model, and sequentially
deletes the predictor that has the least impact on the fit. The candidate for
dropping is the variable with the smallest Z-score (Exercise 3.10). Backward
selection can only be used when N &gt; p, while forward stepwise can always
be used.
Figure 3.6 shows the results of a small simulation study to compare
</p>
<p>best-subset regression with the simpler alternatives forward and backward
selection. Their performance is very similar, as is often the case. Included in
the figure is forward stagewise regression (next section), which takes longer
to reach minimum error.</p>
<p/>
</div>
<div class="page"><p/>
<p>60 3. Linear Methods for Regression
</p>
<p>On the prostate cancer example, best-subset, forward and backward se-
lection all gave exactly the same sequence of terms.
Some software packages implement hybrid stepwise-selection strategies
</p>
<p>that consider both forward and backward moves at each step, and select
the &ldquo;best&rdquo; of the two. For example in the R package the step function uses
the AIC criterion for weighing the choices, which takes proper account of
the number of parameters fit; at each step an add or drop will be performed
that minimizes the AIC score.
Other more traditional packages base the selection on F -statistics, adding
</p>
<p>&ldquo;significant&rdquo; terms, and dropping &ldquo;non-significant&rdquo; terms. These are out
of fashion, since they do not take proper account of the multiple testing
issues. It is also tempting after a model search to print out a summary of
the chosen model, such as in Table 3.2; however, the standard errors are
not valid, since they do not account for the search process. The bootstrap
(Section 8.2) can be useful in such settings.
Finally, we note that often variables come in groups (such as the dummy
</p>
<p>variables that code a multi-level categorical predictor). Smart stepwise pro-
cedures (such as step in R) will add or drop whole groups at a time, taking
proper account of their degrees-of-freedom.
</p>
<p>3.3.3 Forward-Stagewise Regression
</p>
<p>Forward-stagewise regression (FS) is even more constrained than forward-
stepwise regression. It starts like forward-stepwise regression, with an in-
tercept equal to ȳ, and centered predictors with coefficients initially all 0.
At each step the algorithm identifies the variable most correlated with the
current residual. It then computes the simple linear regression coefficient
of the residual on this chosen variable, and then adds it to the current co-
efficient for that variable. This is continued till none of the variables have
correlation with the residuals&mdash;i.e. the least-squares fit when N &gt; p.
Unlike forward-stepwise regression, none of the other variables are ad-
</p>
<p>justed when a term is added to the model. As a consequence, forward
stagewise can take many more than p steps to reach the least squares fit,
and historically has been dismissed as being inefficient. It turns out that
this &ldquo;slow fitting&rdquo; can pay dividends in high-dimensional problems. We
see in Section 3.8.1 that both forward stagewise and a variant which is
slowed down even further are quite competitive, especially in very high-
dimensional problems.
Forward-stagewise regression is included in Figure 3.6. In this example it
</p>
<p>takes over 1000 steps to get all the correlations below 10&minus;4. For subset size
k, we plotted the error for the last step for which there where k nonzero
coefficients. Although it catches up with the best fit, it takes longer to
do so.</p>
<p/>
</div>
<div class="page"><p/>
<p>3.4 Shrinkage Methods 61
</p>
<p>3.3.4 Prostate Cancer Data Example (Continued)
</p>
<p>Table 3.3 shows the coefficients from a number of different selection and
shrinkage methods. They are best-subset selection using an all-subsets search,
ridge regression, the lasso, principal components regression and partial least
squares. Each method has a complexity parameter, and this was chosen to
minimize an estimate of prediction error based on tenfold cross-validation;
full details are given in Section 7.10. Briefly, cross-validation works by divid-
ing the training data randomly into ten equal parts. The learning method
is fit&mdash;for a range of values of the complexity parameter&mdash;to nine-tenths of
the data, and the prediction error is computed on the remaining one-tenth.
This is done in turn for each one-tenth of the data, and the ten prediction
error estimates are averaged. From this we obtain an estimated prediction
error curve as a function of the complexity parameter.
Note that we have already divided these data into a training set of size
</p>
<p>67 and a test set of size 30. Cross-validation is applied to the training set,
since selecting the shrinkage parameter is part of the training process. The
test set is there to judge the performance of the selected model.
The estimated prediction error curves are shown in Figure 3.7. Many of
</p>
<p>the curves are very flat over large ranges near their minimum. Included
are estimated standard error bands for each estimated error rate, based on
the ten error estimates computed by cross-validation. We have used the
&ldquo;one-standard-error&rdquo; rule&mdash;we pick the most parsimonious model within
one standard error of the minimum (Section 7.10, page 244). Such a rule
acknowledges the fact that the tradeoff curve is estimated with error, and
hence takes a conservative approach.
Best-subset selection chose to use the two predictors lcvol and lweight.
</p>
<p>The last two lines of the table give the average prediction error (and its
estimated standard error) over the test set.
</p>
<p>3.4 Shrinkage Methods
</p>
<p>By retaining a subset of the predictors and discarding the rest, subset selec-
tion produces a model that is interpretable and has possibly lower predic-
tion error than the full model. However, because it is a discrete process&mdash;
variables are either retained or discarded&mdash;it often exhibits high variance,
and so doesn&rsquo;t reduce the prediction error of the full model. Shrinkage
methods are more continuous, and don&rsquo;t suffer as much from high
variability.
</p>
<p>3.4.1 Ridge Regression
</p>
<p>Ridge regression shrinks the regression coefficients by imposing a penalty
on their size. The ridge coefficients minimize a penalized residual sum of</p>
<p/>
</div>
<div class="page"><p/>
<p>62 3. Linear Methods for Regression
</p>
<p>Subset Size
</p>
<p>C
V
</p>
<p> E
rr
</p>
<p>o
r
</p>
<p>0 2 4 6 8
</p>
<p>0
.6
</p>
<p>0
.8
</p>
<p>1
.0
</p>
<p>1
.2
</p>
<p>1
.4
</p>
<p>1
.6
</p>
<p>1
.8
</p>
<p>&bull;
</p>
<p>&bull;
&bull; &bull; &bull; &bull; &bull; &bull; &bull;
</p>
<p>All Subsets
</p>
<p>Degrees of Freedom
</p>
<p>C
V
</p>
<p> E
rr
</p>
<p>o
r
</p>
<p>0 2 4 6 8
</p>
<p>0
.6
</p>
<p>0
.8
</p>
<p>1
.0
</p>
<p>1
.2
</p>
<p>1
.4
</p>
<p>1
.6
</p>
<p>1
.8
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull; &bull; &bull; &bull; &bull;
</p>
<p>Ridge Regression
</p>
<p>Shrinkage Factor s
</p>
<p>C
V
</p>
<p> E
rr
</p>
<p>o
r
</p>
<p>0.0 0.2 0.4 0.6 0.8 1.0
</p>
<p>0
.6
</p>
<p>0
.8
</p>
<p>1
.0
</p>
<p>1
.2
</p>
<p>1
.4
</p>
<p>1
.6
</p>
<p>1
.8
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull; &bull; &bull; &bull; &bull; &bull;
</p>
<p>Lasso
</p>
<p>Number of Directions
</p>
<p>C
V
</p>
<p> E
rr
</p>
<p>o
r
</p>
<p>0 2 4 6 8
</p>
<p>0
.6
</p>
<p>0
.8
</p>
<p>1
.0
</p>
<p>1
.2
</p>
<p>1
.4
</p>
<p>1
.6
</p>
<p>1
.8
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull; &bull; &bull; &bull; &bull; &bull;
</p>
<p>Principal Components Regression
</p>
<p>Number of  Directions
</p>
<p>C
V
</p>
<p> E
rr
</p>
<p>o
r
</p>
<p>0 2 4 6 8
</p>
<p>0
.6
</p>
<p>0
.8
</p>
<p>1
.0
</p>
<p>1
.2
</p>
<p>1
.4
</p>
<p>1
.6
</p>
<p>1
.8
</p>
<p>&bull;
</p>
<p>&bull;
&bull; &bull; &bull; &bull; &bull; &bull; &bull;
</p>
<p>Partial Least Squares
</p>
<p>FIGURE 3.7. Estimated prediction error curves and their standard errors for
the various selection and shrinkage methods. Each curve is plotted as a function
of the corresponding complexity parameter for that method. The horizontal axis
has been chosen so that the model complexity increases as we move from left to
right. The estimates of prediction error and their standard errors were obtained by
tenfold cross-validation; full details are given in Section 7.10. The least complex
model within one standard error of the best is chosen, indicated by the purple
vertical broken lines.</p>
<p/>
</div>
<div class="page"><p/>
<p>3.4 Shrinkage Methods 63
</p>
<p>TABLE 3.3. Estimated coefficients and test error results, for different subset
and shrinkage methods applied to the prostate data. The blank entries correspond
to variables omitted.
</p>
<p>Term LS Best Subset Ridge Lasso PCR PLS
</p>
<p>Intercept 2.465 2.477 2.452 2.468 2.497 2.452
lcavol 0.680 0.740 0.420 0.533 0.543 0.419
lweight 0.263 0.316 0.238 0.169 0.289 0.344
</p>
<p>age &minus;0.141 &minus;0.046 &minus;0.152 &minus;0.026
lbph 0.210 0.162 0.002 0.214 0.220
svi 0.305 0.227 0.094 0.315 0.243
lcp &minus;0.288 0.000 &minus;0.051 0.079
</p>
<p>gleason &minus;0.021 0.040 0.232 0.011
pgg45 0.267 0.133 &minus;0.056 0.084
</p>
<p>Test Error 0.521 0.492 0.492 0.479 0.449 0.528
Std Error 0.179 0.143 0.165 0.164 0.105 0.152
</p>
<p>squares,
</p>
<p>β̂ridge = argmin
β
</p>
<p>{ N&sum;
</p>
<p>i=1
</p>
<p>(
yi &minus; β0 &minus;
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>xijβj
)2
</p>
<p>+ λ
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>β2j
</p>
<p>}
. (3.41)
</p>
<p>Here λ &ge; 0 is a complexity parameter that controls the amount of shrink-
age: the larger the value of λ, the greater the amount of shrinkage. The
coefficients are shrunk toward zero (and each other). The idea of penaliz-
ing by the sum-of-squares of the parameters is also used in neural networks,
where it is known as weight decay (Chapter 11).
</p>
<p>An equivalent way to write the ridge problem is
</p>
<p>β̂ridge = argmin
β
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>(
yi &minus; β0 &minus;
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>xijβj
</p>
<p>)2
,
</p>
<p>subject to
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>β2j &le; t,
(3.42)
</p>
<p>which makes explicit the size constraint on the parameters. There is a one-
to-one correspondence between the parameters λ in (3.41) and t in (3.42).
When there are many correlated variables in a linear regression model,
their coefficients can become poorly determined and exhibit high variance.
A wildly large positive coefficient on one variable can be canceled by a
similarly large negative coefficient on its correlated cousin. By imposing a
size constraint on the coefficients, as in (3.42), this problem is alleviated.
The ridge solutions are not equivariant under scaling of the inputs, and
</p>
<p>so one normally standardizes the inputs before solving (3.41). In addition,</p>
<p/>
</div>
<div class="page"><p/>
<p>64 3. Linear Methods for Regression
</p>
<p>notice that the intercept β0 has been left out of the penalty term. Penal-
ization of the intercept would make the procedure depend on the origin
chosen for Y ; that is, adding a constant c to each of the targets yi would
not simply result in a shift of the predictions by the same amount c. It
can be shown (Exercise 3.5) that the solution to (3.41) can be separated
into two parts, after reparametrization using centered inputs: each xij gets
</p>
<p>replaced by xij &minus; x̄j . We estimate β0 by ȳ = 1N
&sum;N
</p>
<p>1 yi. The remaining co-
efficients get estimated by a ridge regression without intercept, using the
centered xij . Henceforth we assume that this centering has been done, so
that the input matrix X has p (rather than p+ 1) columns.
</p>
<p>Writing the criterion in (3.41) in matrix form,
</p>
<p>RSS(λ) = (y &minus;Xβ)T (y &minus;Xβ) + λβTβ, (3.43)
</p>
<p>the ridge regression solutions are easily seen to be
</p>
<p>β̂ridge = (XTX+ λI)&minus;1XTy, (3.44)
</p>
<p>where I is the p&times;p identity matrix. Notice that with the choice of quadratic
penalty βTβ, the ridge regression solution is again a linear function of
y. The solution adds a positive constant to the diagonal of XTX before
inversion. This makes the problem nonsingular, even if XTX is not of full
rank, and was the main motivation for ridge regression when it was first
introduced in statistics (Hoerl and Kennard, 1970). Traditional descriptions
of ridge regression start with definition (3.44). We choose to motivate it via
(3.41) and (3.42), as these provide insight into how it works.
Figure 3.8 shows the ridge coefficient estimates for the prostate can-
</p>
<p>cer example, plotted as functions of df(λ), the effective degrees of freedom
implied by the penalty λ (defined in (3.50) on page 68). In the case of or-
thonormal inputs, the ridge estimates are just a scaled version of the least
squares estimates, that is, β̂ridge = β̂/(1 + λ).
Ridge regression can also be derived as the mean or mode of a poste-
</p>
<p>rior distribution, with a suitably chosen prior distribution. In detail, sup-
pose yi &sim; N(β0 + xTi β, σ2), and the parameters βj are each distributed as
N(0, τ2), independently of one another. Then the (negative) log-posterior
density of β, with τ2 and σ2 assumed known, is equal to the expression
in curly braces in (3.41), with λ = σ2/τ2 (Exercise 3.6). Thus the ridge
estimate is the mode of the posterior distribution; since the distribution is
Gaussian, it is also the posterior mean.
The singular value decomposition (SVD) of the centered input matrix X
</p>
<p>gives us some additional insight into the nature of ridge regression. This de-
composition is extremely useful in the analysis of many statistical methods.
The SVD of the N &times; p matrix X has the form
</p>
<p>X = UDVT . (3.45)</p>
<p/>
</div>
<div class="page"><p/>
<p>3.4 Shrinkage Methods 65
</p>
<p>C
o
</p>
<p>e
ff
</p>
<p>ic
ie
</p>
<p>n
ts
</p>
<p>0 2 4 6 8
</p>
<p>&minus;
0
</p>
<p>.2
0
</p>
<p>.0
0
</p>
<p>.2
0
</p>
<p>.4
0
</p>
<p>.6
</p>
<p>&bull;
</p>
<p>&bull;&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>lcavol
</p>
<p>&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;&bull;&bull;&bull;
</p>
<p>&bull;
</p>
<p>lweight
</p>
<p>&bull;
&bull;&bull;&bull;
</p>
<p>&bull;&bull;
&bull;&bull;
</p>
<p>&bull;&bull;
&bull;&bull;
</p>
<p>&bull;&bull;
&bull;&bull;&bull;
</p>
<p>&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;
</p>
<p>age
</p>
<p>&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;
</p>
<p>lbph
</p>
<p>&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
</p>
<p>&bull;
</p>
<p>svi
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;&bull;
</p>
<p>&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
</p>
<p>&bull;
</p>
<p>lcp
</p>
<p>&bull;
&bull;&bull;&bull;&bull;
</p>
<p>&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;
</p>
<p>&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;
</p>
<p>gleason
</p>
<p>&bull;
</p>
<p>&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
</p>
<p>&bull;
</p>
<p>pgg45
</p>
<p>df(λ)
</p>
<p>FIGURE 3.8. Profiles of ridge coefficients for the prostate cancer example, as
the tuning parameter λ is varied. Coefficients are plotted versus df(λ), the effective
degrees of freedom. A vertical line is drawn at df = 5.0, the value chosen by
cross-validation.</p>
<p/>
</div>
<div class="page"><p/>
<p>66 3. Linear Methods for Regression
</p>
<p>Here U and V are N &times; p and p&times; p orthogonal matrices, with the columns
of U spanning the column space of X, and the columns of V spanning the
row space. D is a p &times; p diagonal matrix, with diagonal entries d1 &ge; d2 &ge;
&middot; &middot; &middot; &ge; dp &ge; 0 called the singular values of X. If one or more values dj = 0,
X is singular.
Using the singular value decomposition we can write the least squares
</p>
<p>fitted vector as
</p>
<p>Xβ̂ls = X(XTX)&minus;1XTy
</p>
<p>= UUTy, (3.46)
</p>
<p>after some simplification. Note that UTy are the coordinates of y with
respect to the orthonormal basis U. Note also the similarity with (3.33);
Q and U are generally different orthogonal bases for the column space of
X (Exercise 3.8).
Now the ridge solutions are
</p>
<p>Xβ̂ridge = X(XTX+ λI)&minus;1XTy
</p>
<p>= U D(D2 + λI)&minus;1D UTy
</p>
<p>=
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>uj
d2j
</p>
<p>d2j + λ
uTj y, (3.47)
</p>
<p>where the uj are the columns of U. Note that since λ &ge; 0, we have d2j/(d2j+
λ) &le; 1. Like linear regression, ridge regression computes the coordinates of
y with respect to the orthonormal basisU. It then shrinks these coordinates
by the factors d2j/(d
</p>
<p>2
j + λ). This means that a greater amount of shrinkage
</p>
<p>is applied to the coordinates of basis vectors with smaller d2j .
</p>
<p>What does a small value of d2j mean? The SVD of the centered matrix
X is another way of expressing the principal components of the variables
in X. The sample covariance matrix is given by S = XTX/N , and from
(3.45) we have
</p>
<p>XTX = VD2VT , (3.48)
</p>
<p>which is the eigen decomposition of XTX (and of S, up to a factor N).
The eigenvectors vj (columns of V) are also called the principal compo-
nents (or Karhunen&ndash;Loeve) directions of X. The first principal component
direction v1 has the property that z1 = Xv1 has the largest sample vari-
ance amongst all normalized linear combinations of the columns of X. This
sample variance is easily seen to be
</p>
<p>Var(z1) = Var(Xv1) =
d21
N
</p>
<p>, (3.49)
</p>
<p>and in fact z1 = Xv1 = u1d1. The derived variable z1 is called the first
principal component of X, and hence u1 is the normalized first principal</p>
<p/>
</div>
<div class="page"><p/>
<p>3.4 Shrinkage Methods 67
</p>
<p>-4 -2 0 2 4
</p>
<p>-4
-2
</p>
<p>0
2
</p>
<p>4
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>Largest Principal
Component
</p>
<p>Smallest Principal
Component
</p>
<p>X1
</p>
<p>X
2
</p>
<p>FIGURE 3.9. Principal components of some input data points. The largest prin-
cipal component is the direction that maximizes the variance of the projected data,
and the smallest principal component minimizes that variance. Ridge regression
projects y onto these components, and then shrinks the coefficients of the low&ndash;
variance components more than the high-variance components.
</p>
<p>component. Subsequent principal components zj have maximum variance
d2j/N , subject to being orthogonal to the earlier ones. Conversely the last
principal component has minimum variance. Hence the small singular val-
ues dj correspond to directions in the column space of X having small
variance, and ridge regression shrinks these directions the most.
Figure 3.9 illustrates the principal components of some data points in
</p>
<p>two dimensions. If we consider fitting a linear surface over this domain
(the Y -axis is sticking out of the page), the configuration of the data allow
us to determine its gradient more accurately in the long direction than
the short. Ridge regression protects against the potentially high variance
of gradients estimated in the short directions. The implicit assumption is
that the response will tend to vary most in the directions of high variance
of the inputs. This is often a reasonable assumption, since predictors are
often chosen for study because they vary with the response variable, but
need not hold in general.</p>
<p/>
</div>
<div class="page"><p/>
<p>68 3. Linear Methods for Regression
</p>
<p>In Figure 3.7 we have plotted the estimated prediction error versus the
quantity
</p>
<p>df(λ) = tr[X(XTX+ λI)&minus;1XT ],
</p>
<p>= tr(Hλ)
</p>
<p>=
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>d2j
d2j + λ
</p>
<p>. (3.50)
</p>
<p>This monotone decreasing function of λ is the effective degrees of freedom
of the ridge regression fit. Usually in a linear-regression fit with p variables,
the degrees-of-freedom of the fit is p, the number of free parameters. The
idea is that although all p coefficients in a ridge fit will be non-zero, they
are fit in a restricted fashion controlled by λ. Note that df(λ) = p when
λ = 0 (no regularization) and df(λ) &rarr; 0 as λ &rarr; &infin;. Of course there
is always an additional one degree of freedom for the intercept, which was
removed apriori. This definition is motivated in more detail in Section 3.4.4
and Sections 7.4&ndash;7.6. In Figure 3.7 the minimum occurs at df(λ) = 5.0.
Table 3.3 shows that ridge regression reduces the test error of the full least
squares estimates by a small amount.
</p>
<p>3.4.2 The Lasso
</p>
<p>The lasso is a shrinkage method like ridge, with subtle but important dif-
ferences. The lasso estimate is defined by
</p>
<p>β̂lasso = argmin
β
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>(
yi &minus; β0 &minus;
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>xijβj
</p>
<p>)2
</p>
<p>subject to
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>|βj | &le; t. (3.51)
</p>
<p>Just as in ridge regression, we can re-parametrize the constant β0 by stan-
dardizing the predictors; the solution for β̂0 is ȳ, and thereafter we fit a
model without an intercept (Exercise 3.5). In the signal processing litera-
ture, the lasso is also known as basis pursuit (Chen et al., 1998).
</p>
<p>We can also write the lasso problem in the equivalent Lagrangian form
</p>
<p>β̂lasso = argmin
β
</p>
<p>{
1
</p>
<p>2
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>(
yi &minus; β0 &minus;
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>xijβj
)2
</p>
<p>+ λ
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>|βj |
}
. (3.52)
</p>
<p>Notice the similarity to the ridge regression problem (3.42) or (3.41): the
L2 ridge penalty
</p>
<p>&sum;p
1 β
</p>
<p>2
j is replaced by the L1 lasso penalty
</p>
<p>&sum;p
1 |βj |. This
</p>
<p>latter constraint makes the solutions nonlinear in the yi, and there is no
closed form expression as in ridge regression. Computing the lasso solution</p>
<p/>
</div>
<div class="page"><p/>
<p>3.4 Shrinkage Methods 69
</p>
<p>is a quadratic programming problem, although we see in Section 3.4.4 that
efficient algorithms are available for computing the entire path of solutions
as λ is varied, with the same computational cost as for ridge regression.
Because of the nature of the constraint, making t sufficiently small will
cause some of the coefficients to be exactly zero. Thus the lasso does a kind
of continuous subset selection. If t is chosen larger than t0 =
</p>
<p>&sum;p
1 |β̂j | (where
</p>
<p>β̂j = β̂
ls
j , the least squares estimates), then the lasso estimates are the β̂j &rsquo;s.
</p>
<p>On the other hand, for t = t0/2 say, then the least squares coefficients are
shrunk by about 50% on average. However, the nature of the shrinkage
is not obvious, and we investigate it further in Section 3.4.4 below. Like
the subset size in variable subset selection, or the penalty parameter in
ridge regression, t should be adaptively chosen to minimize an estimate of
expected prediction error.
In Figure 3.7, for ease of interpretation, we have plotted the lasso pre-
</p>
<p>diction error estimates versus the standardized parameter s = t/
&sum;p
</p>
<p>1 |β̂j |.
A value ŝ &asymp; 0.36 was chosen by 10-fold cross-validation; this caused four
coefficients to be set to zero (fifth column of Table 3.3). The resulting
model has the second lowest test error, slightly lower than the full least
squares model, but the standard errors of the test error estimates (last line
of Table 3.3) are fairly large.
Figure 3.10 shows the lasso coefficients as the standardized tuning pa-
</p>
<p>rameter s = t/
&sum;p
</p>
<p>1 |β̂j | is varied. At s = 1.0 these are the least squares
estimates; they decrease to 0 as s &rarr; 0. This decrease is not always strictly
monotonic, although it is in this example. A vertical line is drawn at
s = 0.36, the value chosen by cross-validation.
</p>
<p>3.4.3 Discussion: Subset Selection, Ridge Regression and the
Lasso
</p>
<p>In this section we discuss and compare the three approaches discussed so far
for restricting the linear regression model: subset selection, ridge regression
and the lasso.
In the case of an orthonormal input matrix X the three procedures have
</p>
<p>explicit solutions. Each method applies a simple transformation to the least
squares estimate β̂j , as detailed in Table 3.4.
</p>
<p>Ridge regression does a proportional shrinkage. Lasso translates each
coefficient by a constant factor λ, truncating at zero. This is called &ldquo;soft
thresholding,&rdquo; and is used in the context of wavelet-based smoothing in Sec-
tion 5.9. Best-subset selection drops all variables with coefficients smaller
than the Mth largest; this is a form of &ldquo;hard-thresholding.&rdquo;
Back to the nonorthogonal case; some pictures help understand their re-
</p>
<p>lationship. Figure 3.11 depicts the lasso (left) and ridge regression (right)
when there are only two parameters. The residual sum of squares has ellip-
tical contours, centered at the full least squares estimate. The constraint</p>
<p/>
</div>
<div class="page"><p/>
<p>70 3. Linear Methods for Regression
</p>
<p>0.0 0.2 0.4 0.6 0.8 1.0
</p>
<p>&minus;
0
</p>
<p>.2
0
</p>
<p>.0
0
</p>
<p>.2
0
</p>
<p>.4
0
</p>
<p>.6
</p>
<p>Shrinkage Factor s
</p>
<p>C
o
</p>
<p>e
ff
</p>
<p>ic
ie
</p>
<p>n
ts
</p>
<p>lcavol
</p>
<p>lweight
</p>
<p>age
</p>
<p>lbph
</p>
<p>svi
</p>
<p>lcp
</p>
<p>gleason
</p>
<p>pgg45
</p>
<p>FIGURE 3.10. Profiles of lasso coefficients, as the tuning parameter t is varied.
Coefficients are plotted versus s = t/
</p>
<p>&sum;p
</p>
<p>1 |β̂j |. A vertical line is drawn at s = 0.36,
the value chosen by cross-validation. Compare Figure 3.8 on page 65; the lasso
profiles hit zero, while those for ridge do not. The profiles are piece-wise linear,
and so are computed only at the points displayed; see Section 3.4.4 for details.</p>
<p/>
</div>
<div class="page"><p/>
<p>3.4 Shrinkage Methods 71
</p>
<p>TABLE 3.4. Estimators of βj in the case of orthonormal columns of X. M and λ
are constants chosen by the corresponding techniques; sign denotes the sign of its
argument (&plusmn;1), and x+ denotes &ldquo;positive part&rdquo; of x. Below the table, estimators
are shown by broken red lines. The 45◦ line in gray shows the unrestricted estimate
for reference.
</p>
<p>Estimator Formula
</p>
<p>Best subset (size M) β̂j &middot; I(|β̂j | &ge; |β̂(M)|)
Ridge β̂j/(1 + λ)
</p>
<p>Lasso sign(β̂j)(|β̂j | &minus; λ)+
</p>
<p>(0,0) (0,0) (0,0)
</p>
<p>|β̂(M)|
</p>
<p>λ
</p>
<p>Best Subset Ridge Lasso
</p>
<p>β
^
</p>
<p>β
^2
</p>
<p>. .β
</p>
<p>1
</p>
<p>β 2
</p>
<p>β
1
</p>
<p>β
</p>
<p>FIGURE 3.11. Estimation picture for the lasso (left) and ridge regression
(right). Shown are contours of the error and constraint functions. The solid blue
areas are the constraint regions |β1| + |β2| &le; t and β21 + β22 &le; t2, respectively,
while the red ellipses are the contours of the least squares error function.</p>
<p/>
</div>
<div class="page"><p/>
<p>72 3. Linear Methods for Regression
</p>
<p>region for ridge regression is the disk β21 + β
2
2 &le; t, while that for lasso is
</p>
<p>the diamond |β1| + |β2| &le; t. Both methods find the first point where the
elliptical contours hit the constraint region. Unlike the disk, the diamond
has corners; if the solution occurs at a corner, then it has one parameter
βj equal to zero. When p &gt; 2, the diamond becomes a rhomboid, and has
many corners, flat edges and faces; there are many more opportunities for
the estimated parameters to be zero.
We can generalize ridge regression and the lasso, and view them as Bayes
</p>
<p>estimates. Consider the criterion
</p>
<p>β̃ = argmin
β
</p>
<p>{
N&sum;
</p>
<p>i=1
</p>
<p>(
yi &minus; β0 &minus;
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>xijβj
)2
</p>
<p>+ λ
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>|βj |q
}
</p>
<p>(3.53)
</p>
<p>for q &ge; 0. The contours of constant value of &sum;j |βj |q are shown in Fig-
ure 3.12, for the case of two inputs.
Thinking of |βj |q as the log-prior density for βj , these are also the equi-
</p>
<p>contours of the prior distribution of the parameters. The value q = 0 corre-
sponds to variable subset selection, as the penalty simply counts the number
of nonzero parameters; q = 1 corresponds to the lasso, while q = 2 to ridge
regression. Notice that for q &le; 1, the prior is not uniform in direction, but
concentrates more mass in the coordinate directions. The prior correspond-
ing to the q = 1 case is an independent double exponential (or Laplace)
distribution for each input, with density (1/2τ) exp(&minus;|β|/τ) and τ = 1/λ.
The case q = 1 (lasso) is the smallest q such that the constraint region
is convex; non-convex constraint regions make the optimization problem
more difficult.
In this view, the lasso, ridge regression and best subset selection are
</p>
<p>Bayes estimates with different priors. Note, however, that they are derived
as posterior modes, that is, maximizers of the posterior. It is more common
to use the mean of the posterior as the Bayes estimate. Ridge regression is
also the posterior mean, but the lasso and best subset selection are not.
Looking again at the criterion (3.53), we might try using other values
</p>
<p>of q besides 0, 1, or 2. Although one might consider estimating q from
the data, our experience is that it is not worth the effort for the extra
variance incurred. Values of q &isin; (1, 2) suggest a compromise between the
lasso and ridge regression. Although this is the case, with q &gt; 1, |βj |q is
differentiable at 0, and so does not share the ability of lasso (q = 1) for
</p>
<p>q = 4 q = 2 q = 1 q = 0.5 q = 0.1
</p>
<p>FIGURE 3.12. Contours of constant value of
&sum;
</p>
<p>j |βj |q for given values of q.</p>
<p/>
</div>
<div class="page"><p/>
<p>3.4 Shrinkage Methods 73
</p>
<p>q = 1.2 α = 0.2
</p>
<p>Lq Elastic Net
</p>
<p>FIGURE 3.13. Contours of constant value of
&sum;
</p>
<p>j |βj |q for q = 1.2 (left plot),
and the elastic-net penalty
</p>
<p>&sum;
</p>
<p>j(αβ
2
j +(1&minus;α)|βj |) for α = 0.2 (right plot). Although
</p>
<p>visually very similar, the elastic-net has sharp (non-differentiable) corners, while
the q = 1.2 penalty does not.
</p>
<p>setting coefficients exactly to zero. Partly for this reason as well as for
computational tractability, Zou and Hastie (2005) introduced the elastic-
net penalty
</p>
<p>λ
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>(
αβ2j + (1&minus; α)|βj |
</p>
<p>)
, (3.54)
</p>
<p>a different compromise between ridge and lasso. Figure 3.13 compares the
Lq penalty with q = 1.2 and the elastic-net penalty with α = 0.2; it is
hard to detect the difference by eye. The elastic-net selects variables like
the lasso, and shrinks together the coefficients of correlated predictors like
ridge. It also has considerable computational advantages over the Lq penal-
ties. We discuss the elastic-net further in Section 18.4.
</p>
<p>3.4.4 Least Angle Regression
</p>
<p>Least angle regression (LAR) is a relative newcomer (Efron et al., 2004),
and can be viewed as a kind of &ldquo;democratic&rdquo; version of forward stepwise
regression (Section 3.3.2). As we will see, LAR is intimately connected
with the lasso, and in fact provides an extremely efficient algorithm for
computing the entire lasso path as in Figure 3.10.
Forward stepwise regression builds a model sequentially, adding one vari-
</p>
<p>able at a time. At each step, it identifies the best variable to include in the
active set, and then updates the least squares fit to include all the active
variables.
Least angle regression uses a similar strategy, but only enters &ldquo;as much&rdquo;
</p>
<p>of a predictor as it deserves. At the first step it identifies the variable
most correlated with the response. Rather than fit this variable completely,
LAR moves the coefficient of this variable continuously toward its least-
squares value (causing its correlation with the evolving residual to decrease
in absolute value). As soon as another variable &ldquo;catches up&rdquo; in terms of
correlation with the residual, the process is paused. The second variable
then joins the active set, and their coefficients are moved together in a way
that keeps their correlations tied and decreasing. This process is continued</p>
<p/>
</div>
<div class="page"><p/>
<p>74 3. Linear Methods for Regression
</p>
<p>until all the variables are in the model, and ends at the full least-squares
fit. Algorithm 3.2 provides the details. The termination condition in step 5
requires some explanation. If p &gt; N &minus; 1, the LAR algorithm reaches a zero
residual solution after N &minus; 1 steps (the &minus;1 is because we have centered the
data).
</p>
<p>Algorithm 3.2 Least Angle Regression.
</p>
<p>1. Standardize the predictors to have mean zero and unit norm. Start
with the residual r = y &minus; ȳ, β1, β2, . . . , βp = 0.
</p>
<p>2. Find the predictor xj most correlated with r.
</p>
<p>3. Move βj from 0 towards its least-squares coefficient 〈xj , r〉, until some
other competitor xk has as much correlation with the current residual
as does xj .
</p>
<p>4. Move βj and βk in the direction defined by their joint least squares
coefficient of the current residual on (xj ,xk), until some other com-
petitor xl has as much correlation with the current residual.
</p>
<p>5. Continue in this way until all p predictors have been entered. After
min(N &minus; 1, p) steps, we arrive at the full least-squares solution.
</p>
<p>Suppose Ak is the active set of variables at the beginning of the kth
step, and let βAk be the coefficient vector for these variables at this step;
there will be k&minus; 1 nonzero values, and the one just entered will be zero. If
rk = y&minus;XAkβAk is the current residual, then the direction for this step is
</p>
<p>δk = (X
T
AkXAk)
</p>
<p>&minus;1XTAkrk. (3.55)
</p>
<p>The coefficient profile then evolves as βAk(α) = βAk +α &middot; δk. Exercise 3.23
verifies that the directions chosen in this fashion do what is claimed: keep
the correlations tied and decreasing. If the fit vector at the beginning of
this step is f̂k, then it evolves as f̂k(α) = f̂k + α &middot; uk, where uk = XAkδk
is the new fit direction. The name &ldquo;least angle&rdquo; arises from a geometrical
interpretation of this process; uk makes the smallest (and equal) angle
with each of the predictors in Ak (Exercise 3.24). Figure 3.14 shows the
absolute correlations decreasing and joining ranks with each step of the
LAR algorithm, using simulated data.
By construction the coefficients in LAR change in a piecewise linear fash-
</p>
<p>ion. Figure 3.15 [left panel] shows the LAR coefficient profile evolving as a
function of their L1 arc length
</p>
<p>2. Note that we do not need to take small
</p>
<p>2The L1 arc-length of a differentiable curve β(s) for s &isin; [0, S] is given by TV(β, S) =&int; S
0 ||β̇(s)||1ds, where β̇(s) = &part;β(s)/&part;s. For the piecewise-linear LAR coefficient profile,
</p>
<p>this amounts to summing the L1 norms of the changes in coefficients from step to step.</p>
<p/>
</div>
<div class="page"><p/>
<p>3.4 Shrinkage Methods 75
</p>
<p>0 5 10 15
</p>
<p>0
.0
</p>
<p>0
.1
</p>
<p>0
.2
</p>
<p>0
.3
</p>
<p>0
.4
</p>
<p>v2 v6 v4 v5 v3 v1
</p>
<p>L1 Arc Length
</p>
<p>A
b
so
lu
te
</p>
<p>C
o
rr
e
la
ti
o
n
s
</p>
<p>FIGURE 3.14. Progression of the absolute correlations during each step of the
LAR procedure, using a simulated data set with six predictors. The labels at the
top of the plot indicate which variables enter the active set at each step. The step
length are measured in units of L1 arc length.
</p>
<p>0 5 10 15
</p>
<p>&minus;
1
.5
</p>
<p>&minus;
1
.0
</p>
<p>&minus;
0
.5
</p>
<p>0
.0
</p>
<p>0
.5
</p>
<p>Least Angle Regression
</p>
<p>0 5 10 15
</p>
<p>&minus;
1
.5
</p>
<p>&minus;
1
.0
</p>
<p>&minus;
0
.5
</p>
<p>0
.0
</p>
<p>0
.5
</p>
<p>Lasso
</p>
<p>L1 Arc LengthL1 Arc Length
</p>
<p>C
o
e
ffi
c
ie
n
ts
</p>
<p>C
o
e
ffi
c
ie
n
ts
</p>
<p>FIGURE 3.15. Left panel shows the LAR coefficient profiles on the simulated
data, as a function of the L1 arc length. The right panel shows the Lasso profile.
They are identical until the dark-blue coefficient crosses zero at an arc length of
about 18.</p>
<p/>
</div>
<div class="page"><p/>
<p>76 3. Linear Methods for Regression
</p>
<p>steps and recheck the correlations in step 3; using knowledge of the covari-
ance of the predictors and the piecewise linearity of the algorithm, we can
work out the exact step length at the beginning of each step (Exercise 3.25).
The right panel of Figure 3.15 shows the lasso coefficient profiles on the
</p>
<p>same data. They are almost identical to those in the left panel, and differ
for the first time when the blue coefficient passes back through zero. For the
prostate data, the LAR coefficient profile turns out to be identical to the
lasso profile in Figure 3.10, which never crosses zero. These observations
lead to a simple modification of the LAR algorithm that gives the entire
lasso path, which is also piecewise-linear.
</p>
<p>Algorithm 3.2a Least Angle Regression: Lasso Modification.
</p>
<p>4a. If a non-zero coefficient hits zero, drop its variable from the active set
of variables and recompute the current joint least squares direction.
</p>
<p>The LAR(lasso) algorithm is extremely efficient, requiring the same order
of computation as that of a single least squares fit using the p predictors.
Least angle regression always takes p steps to get to the full least squares
estimates. The lasso path can have more than p steps, although the two
are often quite similar. Algorithm 3.2 with the lasso modification 3.2a is
an efficient way of computing the solution to any lasso problem, especially
when p ≫ N . Osborne et al. (2000a) also discovered a piecewise-linear path
for computing the lasso, which they called a homotopy algorithm.
We now give a heuristic argument for why these procedures are so similar.
</p>
<p>Although the LAR algorithm is stated in terms of correlations, if the input
features are standardized, it is equivalent and easier to work with inner-
products. Suppose A is the active set of variables at some stage in the
algorithm, tied in their absolute inner-product with the current residuals
y &minus;Xβ. We can express this as
</p>
<p>xTj (y &minus;Xβ) = γ &middot; sj , &forall;j &isin; A (3.56)
</p>
<p>where sj &isin; {&minus;1, 1} indicates the sign of the inner-product, and γ is the
common value. Also |xTk (y &minus; Xβ)| &le; γ &forall;k �&isin; A. Now consider the lasso
criterion (3.52), which we write in vector form
</p>
<p>R(β) = 12 ||y &minus;Xβ||22 + λ||β||1. (3.57)
</p>
<p>Let B be the active set of variables in the solution for a given value of λ.
For these variables R(β) is differentiable, and the stationarity conditions
give
</p>
<p>xTj (y &minus;Xβ) = λ &middot; sign(βj), &forall;j &isin; B (3.58)
Comparing (3.58) with (3.56), we see that they are identical only if the
sign of βj matches the sign of the inner product. That is why the LAR</p>
<p/>
</div>
<div class="page"><p/>
<p>3.4 Shrinkage Methods 77
</p>
<p>algorithm and lasso start to differ when an active coefficient passes through
zero; condition (3.58) is violated for that variable, and it is kicked out of the
active set B. Exercise 3.23 shows that these equations imply a piecewise-
linear coefficient profile as λ decreases. The stationarity conditions for the
non-active variables require that
</p>
<p>|xTk (y &minus;Xβ)| &le; λ, &forall;k �&isin; B, (3.59)
</p>
<p>which again agrees with the LAR algorithm.
Figure 3.16 compares LAR and lasso to forward stepwise and stagewise
</p>
<p>regression. The setup is the same as in Figure 3.6 on page 59, except here
N = 100 here rather than 300, so the problem is more difficult. We see
that the more aggressive forward stepwise starts to overfit quite early (well
before the 10 true variables can enter the model), and ultimately performs
worse than the slower forward stagewise regression. The behavior of LAR
and lasso is similar to that of forward stagewise regression. Incremental
forward stagewise is similar to LAR and lasso, and is described in Sec-
tion 3.8.1.
</p>
<p>Degrees-of-Freedom Formula for LAR and Lasso
</p>
<p>Suppose that we fit a linear model via the least angle regression procedure,
stopping at some number of steps k &lt; p, or equivalently using a lasso bound
t that produces a constrained version of the full least squares fit. How many
parameters, or &ldquo;degrees of freedom&rdquo; have we used?
Consider first a linear regression using a subset of k features. If this subset
</p>
<p>is prespecified in advance without reference to the training data, then the
degrees of freedom used in the fitted model is defined to be k. Indeed, in
classical statistics, the number of linearly independent parameters is what
is meant by &ldquo;degrees of freedom.&rdquo; Alternatively, suppose that we carry out
a best subset selection to determine the &ldquo;optimal&rdquo; set of k predictors. Then
the resulting model has k parameters, but in some sense we have used up
more than k degrees of freedom.
We need a more general definition for the effective degrees of freedom of
</p>
<p>an adaptively fitted model. We define the degrees of freedom of the fitted
vector ŷ = (ŷ1, ŷ2, . . . , ŷN ) as
</p>
<p>df(ŷ) =
1
</p>
<p>σ2
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>Cov(ŷi, yi). (3.60)
</p>
<p>Here Cov(ŷi, yi) refers to the sampling covariance between the predicted
value ŷi and its corresponding outcome value yi. This makes intuitive sense:
the harder that we fit to the data, the larger this covariance and hence
df(ŷ). Expression (3.60) is a useful notion of degrees of freedom, one that
can be applied to any model prediction ŷ. This includes models that are</p>
<p/>
</div>
<div class="page"><p/>
<p>78 3. Linear Methods for Regression
</p>
<p>0.0 0.2 0.4 0.6 0.8 1.0
</p>
<p>0
.5
</p>
<p>5
0
</p>
<p>.6
0
</p>
<p>0
.6
</p>
<p>5
</p>
<p>Forward Stepwise
LAR
Lasso
Forward Stagewise
Incremental Forward Stagewise
</p>
<p>E
||β̂
</p>
<p>(k
)
&minus;
</p>
<p>β
||2
</p>
<p>Fraction of L1 arc-length
</p>
<p>FIGURE 3.16. Comparison of LAR and lasso with forward stepwise, forward
stagewise (FS) and incremental forward stagewise (FS0) regression. The setup
is the same as in Figure 3.6, except N = 100 here rather than 300. Here the
slower FS regression ultimately outperforms forward stepwise. LAR and lasso
show similar behavior to FS and FS0. Since the procedures take different numbers
of steps (across simulation replicates and methods), we plot the MSE as a function
of the fraction of total L1 arc-length toward the least-squares fit.
</p>
<p>adaptively fitted to the training data. This definition is motivated and
discussed further in Sections 7.4&ndash;7.6.
Now for a linear regression with k fixed predictors, it is easy to show
</p>
<p>that df(ŷ) = k. Likewise for ridge regression, this definition leads to the
closed-form expression (3.50) on page 68: df(ŷ) = tr(Sλ). In both these
cases, (3.60) is simple to evaluate because the fit ŷ = Hλy is linear in y.
If we think about definition (3.60) in the context of a best subset selection
of size k, it seems clear that df(ŷ) will be larger than k, and this can be
verified by estimating Cov(ŷi, yi)/σ
</p>
<p>2 directly by simulation. However there
is no closed form method for estimating df(ŷ) for best subset selection.
For LAR and lasso, something magical happens. These techniques are
</p>
<p>adaptive in a smoother way than best subset selection, and hence estimation
of degrees of freedom is more tractable. Specifically it can be shown that
after the kth step of the LAR procedure, the effective degrees of freedom of
the fit vector is exactly k. Now for the lasso, the (modified) LAR procedure</p>
<p/>
</div>
<div class="page"><p/>
<p>3.5 Methods Using Derived Input Directions 79
</p>
<p>often takes more than p steps, since predictors can drop out. Hence the
definition is a little different; for the lasso, at any stage df(ŷ) approximately
equals the number of predictors in the model. While this approximation
works reasonably well anywhere in the lasso path, for each k it works best
at the last model in the sequence that contains k predictors. A detailed
study of the degrees of freedom for the lasso may be found in Zou et al.
(2007).
</p>
<p>3.5 Methods Using Derived Input Directions
</p>
<p>In many situations we have a large number of inputs, often very correlated.
The methods in this section produce a small number of linear combinations
Zm, m = 1, . . . ,M of the original inputs Xj , and the Zm are then used in
place of the Xj as inputs in the regression. The methods differ in how the
linear combinations are constructed.
</p>
<p>3.5.1 Principal Components Regression
</p>
<p>In this approach the linear combinations Zm used are the principal com-
ponents as defined in Section 3.4.1 above.
Principal component regression forms the derived input columns zm =
</p>
<p>Xvm, and then regresses y on z1, z2, . . . , zM for some M &le; p. Since the zm
are orthogonal, this regression is just a sum of univariate regressions:
</p>
<p>ŷ
pcr
(M) = ȳ1+
</p>
<p>M&sum;
</p>
<p>m=1
</p>
<p>θ̂mzm, (3.61)
</p>
<p>where θ̂m = 〈zm,y〉/〈zm, zm〉. Since the zm are each linear combinations
of the original xj , we can express the solution (3.61) in terms of coefficients
of the xj (Exercise 3.13):
</p>
<p>β̂pcr(M) =
M&sum;
</p>
<p>m=1
</p>
<p>θ̂mvm. (3.62)
</p>
<p>As with ridge regression, principal components depend on the scaling of
the inputs, so typically we first standardize them. Note that if M = p, we
would just get back the usual least squares estimates, since the columns of
Z = UD span the column space of X. For M &lt; p we get a reduced regres-
sion. We see that principal components regression is very similar to ridge
regression: both operate via the principal components of the input ma-
trix. Ridge regression shrinks the coefficients of the principal components
(Figure 3.17), shrinking more depending on the size of the corresponding
eigenvalue; principal components regression discards the p &minus; M smallest
eigenvalue components. Figure 3.17 illustrates this.</p>
<p/>
</div>
<div class="page"><p/>
<p>80 3. Linear Methods for Regression
</p>
<p>Index
</p>
<p>S
h
ri
n
k
a
g
e
 F
</p>
<p>a
c
to
</p>
<p>r
</p>
<p>2 4 6 8
</p>
<p>0
.0
</p>
<p>0
.2
</p>
<p>0
.4
</p>
<p>0
.6
</p>
<p>0
.8
</p>
<p>1
.0
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull; &bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull; &bull; &bull; &bull; &bull; &bull;
</p>
<p>&bull; &bull;
</p>
<p>ridge
</p>
<p>pcr
</p>
<p>FIGURE 3.17. Ridge regression shrinks the regression coefficients of the prin-
cipal components, using shrinkage factors d2j/(d
</p>
<p>2
j + λ) as in (3.47). Principal
</p>
<p>component regression truncates them. Shown are the shrinkage and truncation
patterns corresponding to Figure 3.7, as a function of the principal component
index.
</p>
<p>In Figure 3.7 we see that cross-validation suggests seven terms; the re-
sulting model has the lowest test error in Table 3.3.
</p>
<p>3.5.2 Partial Least Squares
</p>
<p>This technique also constructs a set of linear combinations of the inputs
for regression, but unlike principal components regression it uses y (in ad-
dition to X) for this construction. Like principal component regression,
partial least squares (PLS) is not scale invariant, so we assume that each
xj is standardized to have mean 0 and variance 1. PLS begins by com-
puting ϕ̂1j = 〈xj ,y〉 for each j. From this we construct the derived input
z1 =
</p>
<p>&sum;
j ϕ̂1jxj , which is the first partial least squares direction. Hence
</p>
<p>in the construction of each zm, the inputs are weighted by the strength
of their univariate effect on y3. The outcome y is regressed on z1 giving
coefficient θ̂1, and then we orthogonalize x1, . . . ,xp with respect to z1. We
continue this process, until M &le; p directions have been obtained. In this
manner, partial least squares produces a sequence of derived, orthogonal
inputs or directions z1, z2, . . . , zM . As with principal-component regres-
sion, if we were to construct all M = p directions, we would get back a
solution equivalent to the usual least squares estimates; using M &lt; p di-
rections produces a reduced regression. The procedure is described fully in
Algorithm 3.3.
</p>
<p>3Since the xj are standardized, the first directions ϕ̂1j are the univariate regression
</p>
<p>coefficients (up to an irrelevant constant); this is not the case for subsequent directions.</p>
<p/>
</div>
<div class="page"><p/>
<p>3.5 Methods Using Derived Input Directions 81
</p>
<p>Algorithm 3.3 Partial Least Squares.
</p>
<p>1. Standardize each xj to have mean zero and variance one. Set ŷ
(0) =
</p>
<p>ȳ1, and x
(0)
j = xj , j = 1, . . . , p.
</p>
<p>2. For m = 1, 2, . . . , p
</p>
<p>(a) zm =
&sum;p
</p>
<p>j=1 ϕ̂mjx
(m&minus;1)
j , where ϕ̂mj = 〈x
</p>
<p>(m&minus;1)
j ,y〉.
</p>
<p>(b) θ̂m = 〈zm,y〉/〈zm, zm〉.
(c) ŷ(m) = ŷ(m&minus;1) + θ̂mzm.
</p>
<p>(d) Orthogonalize each x
(m&minus;1)
j with respect to zm: x
</p>
<p>(m)
j = x
</p>
<p>(m&minus;1)
j &minus;
</p>
<p>[〈zm,x(m&minus;1)j 〉/〈zm, zm〉]zm, j = 1, 2, . . . , p.
</p>
<p>3. Output the sequence of fitted vectors {ŷ(m)}p1. Since the {zℓ}m1 are
linear in the original xj , so is ŷ
</p>
<p>(m) = Xβ̂pls(m). These linear coeffi-
cients can be recovered from the sequence of PLS transformations.
</p>
<p>In the prostate cancer example, cross-validation chose M = 2 PLS direc-
tions in Figure 3.7. This produced the model given in the rightmost column
of Table 3.3.
What optimization problem is partial least squares solving? Since it uses
</p>
<p>the response y to construct its directions, its solution path is a nonlinear
function of y. It can be shown (Exercise 3.15) that partial least squares
seeks directions that have high variance and have high correlation with the
response, in contrast to principal components regression which keys only
on high variance (Stone and Brooks, 1990; Frank and Friedman, 1993). In
particular, the mth principal component direction vm solves:
</p>
<p>maxα Var(Xα) (3.63)
</p>
<p>subject to ||α|| = 1, αTSvℓ = 0, ℓ = 1, . . . ,m&minus; 1,
</p>
<p>where S is the sample covariance matrix of the xj . The conditions α
TSvℓ =
</p>
<p>0 ensures that zm = Xα is uncorrelated with all the previous linear com-
binations zℓ = Xvℓ. The mth PLS direction ϕ̂m solves:
</p>
<p>maxα Corr
2(y,Xα)Var(Xα) (3.64)
</p>
<p>subject to ||α|| = 1, αTSϕ̂ℓ = 0, ℓ = 1, . . . ,m&minus; 1.
</p>
<p>Further analysis reveals that the variance aspect tends to dominate, and
so partial least squares behaves much like ridge regression and principal
components regression. We discuss this further in the next section.
If the input matrix X is orthogonal, then partial least squares finds the
</p>
<p>least squares estimates after m = 1 steps. Subsequent steps have no effect</p>
<p/>
</div>
<div class="page"><p/>
<p>82 3. Linear Methods for Regression
</p>
<p>since the ϕ̂mj are zero for m &gt; 1 (Exercise 3.14). It can also be shown that
the sequence of PLS coefficients for m = 1, 2, . . . , p represents the conjugate
gradient sequence for computing the least squares solutions (Exercise 3.18).
</p>
<p>3.6 Discussion: A Comparison of the Selection and
Shrinkage Methods
</p>
<p>There are some simple settings where we can understand better the rela-
tionship between the different methods described above. Consider an exam-
ple with two correlated inputs X1 and X2, with correlation ρ. We assume
that the true regression coefficients are β1 = 4 and β2 = 2. Figure 3.18
shows the coefficient profiles for the different methods, as their tuning pa-
rameters are varied. The top panel has ρ = 0.5, the bottom panel ρ = &minus;0.5.
The tuning parameters for ridge and lasso vary over a continuous range,
while best subset, PLS and PCR take just two discrete steps to the least
squares solution. In the top panel, starting at the origin, ridge regression
shrinks the coefficients together until it finally converges to least squares.
PLS and PCR show similar behavior to ridge, although are discrete and
more extreme. Best subset overshoots the solution and then backtracks.
The behavior of the lasso is intermediate to the other methods. When the
correlation is negative (lower panel), again PLS and PCR roughly track
the ridge path, while all of the methods are more similar to one another.
It is interesting to compare the shrinkage behavior of these different
</p>
<p>methods. Recall that ridge regression shrinks all directions, but shrinks
low-variance directions more. Principal components regression leaves M
high-variance directions alone, and discards the rest. Interestingly, it can
be shown that partial least squares also tends to shrink the low-variance
directions, but can actually inflate some of the higher variance directions.
This can make PLS a little unstable, and cause it to have slightly higher
prediction error compared to ridge regression. A full study is given in Frank
and Friedman (1993). These authors conclude that for minimizing predic-
tion error, ridge regression is generally preferable to variable subset selec-
tion, principal components regression and partial least squares. However
the improvement over the latter two methods was only slight.
To summarize, PLS, PCR and ridge regression tend to behave similarly.
</p>
<p>Ridge regression may be preferred because it shrinks smoothly, rather than
in discrete steps. Lasso falls somewhere between ridge regression and best
subset regression, and enjoys some of the properties of each.</p>
<p/>
</div>
<div class="page"><p/>
<p>3.6 Discussion: A Comparison of the Selection and Shrinkage Methods 83
</p>
<p>0 1 2 3 4 5 6
</p>
<p>-1
0
</p>
<p>1
2
</p>
<p>3
</p>
<p>Least Squares
</p>
<p>0
</p>
<p>Ridge
</p>
<p>Lasso
</p>
<p>Best Subset
</p>
<p>PLSPCR
</p>
<p>&bull;
</p>
<p>0 1 2 3 4 5 6
</p>
<p>-1
0
</p>
<p>1
2
</p>
<p>3
</p>
<p>Least Squares
</p>
<p>Ridge
</p>
<p>Best Subset
</p>
<p>PLS
</p>
<p>PCR
</p>
<p>Lasso
</p>
<p>&bull;
</p>
<p>0
</p>
<p>ρ = 0.5
</p>
<p>ρ = &minus;0.5
</p>
<p>β1
</p>
<p>β1
</p>
<p>β
2
</p>
<p>β
2
</p>
<p>FIGURE 3.18. Coefficient profiles from different methods for a simple problem:
two inputs with correlation &plusmn;0.5, and the true regression coefficients β = (4, 2).</p>
<p/>
</div>
<div class="page"><p/>
<p>84 3. Linear Methods for Regression
</p>
<p>3.7 Multiple Outcome Shrinkage and Selection
</p>
<p>As noted in Section 3.2.4, the least squares estimates in a multiple-output
linear model are simply the individual least squares estimates for each of
the outputs.
To apply selection and shrinkage methods in the multiple output case,
</p>
<p>one could apply a univariate technique individually to each outcome or si-
multaneously to all outcomes. With ridge regression, for example, we could
apply formula (3.44) to each of the K columns of the outcome matrix Y ,
using possibly different parameters λ, or apply it to all columns using the
same value of λ. The former strategy would allow different amounts of
regularization to be applied to different outcomes but require estimation
of k separate regularization parameters λ1, . . . , λk, while the latter would
permit all k outputs to be used in estimating the sole regularization pa-
rameter λ.
</p>
<p>Other more sophisticated shrinkage and selection strategies that exploit
correlations in the different responses can be helpful in the multiple output
case. Suppose for example that among the outputs we have
</p>
<p>Yk = f(X) + εk (3.65)
</p>
<p>Yℓ = f(X) + εℓ; (3.66)
</p>
<p>i.e., (3.65) and (3.66) share the same structural part f(X) in their models.
It is clear in this case that we should pool our observations on Yk and Yl
to estimate the common f .
Combining responses is at the heart of canonical correlation analysis
</p>
<p>(CCA), a data reduction technique developed for the multiple output case.
Similar to PCA, CCA finds a sequence of uncorrelated linear combina-
tions Xvm, m = 1, . . . ,M of the xj , and a corresponding sequence of
uncorrelated linear combinations Yum of the responses yk, such that the
correlations
</p>
<p>Corr2(Yum,Xvm) (3.67)
</p>
<p>are successively maximized. Note that at most M = min(K, p) directions
can be found. The leading canonical response variates are those linear com-
binations (derived responses) best predicted by the xj ; in contrast, the
trailing canonical variates can be poorly predicted by the xj , and are can-
didates for being dropped. The CCA solution is computed using a general-
ized SVD of the sample cross-covariance matrix YTX/N (assuming Y and
X are centered; Exercise 3.20).
Reduced-rank regression (Izenman, 1975; van der Merwe and Zidek, 1980)
</p>
<p>formalizes this approach in terms of a regression model that explicitly pools
information. Given an error covariance Cov(ε) = Σ, we solve the following</p>
<p/>
</div>
<div class="page"><p/>
<p>3.7 Multiple Outcome Shrinkage and Selection 85
</p>
<p>restricted multivariate regression problem:
</p>
<p>B̂rr(m) = argmin
rank(B)=m
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>(yi &minus;BTxi)TΣ&minus;1(yi &minus;BTxi). (3.68)
</p>
<p>With Σ replaced by the estimate YTY/N , one can show (Exercise 3.21)
that the solution is given by a CCA of Y and X:
</p>
<p>B̂rr(m) = B̂UmU
&minus;
m, (3.69)
</p>
<p>where Um is the K&times;m sub-matrix of U consisting of the first m columns,
and U is the K &times;M matrix of left canonical vectors u1, u2, . . . , uM . U&minus;m
is its generalized inverse. Writing the solution as
</p>
<p>B̂rr(M) = (XTX)&minus;1XT (YUm)U
&minus;
m, (3.70)
</p>
<p>we see that reduced-rank regression performs a linear regression on the
pooled response matrix YUm, and then maps the coefficients (and hence
the fits as well) back to the original response space. The reduced-rank fits
are given by
</p>
<p>Ŷrr(m) = X(XTX)&minus;1XTYUmU
&minus;
m
</p>
<p>= HYPm,
(3.71)
</p>
<p>where H is the usual linear regression projection operator, and Pm is the
rank-m CCA response projection operator. Although a better estimate of
Σ would be (Y&minus;XB̂)T (Y&minus;XB̂)/(N&minus;pK), one can show that the solution
remains the same (Exercise 3.22).
Reduced-rank regression borrows strength among responses by truncat-
</p>
<p>ing the CCA. Breiman and Friedman (1997) explored with some success
shrinkage of the canonical variates between X and Y, a smooth version of
reduced rank regression. Their proposal has the form (compare (3.69))
</p>
<p>B̂c+w = B̂UΛU&minus;1, (3.72)
</p>
<p>where Λ is a diagonal shrinkage matrix (the &ldquo;c+w&rdquo; stands for &ldquo;Curds
and Whey,&rdquo; the name they gave to their procedure). Based on optimal
prediction in the population setting, they show that Λ has diagonal entries
</p>
<p>λm =
c2m
</p>
<p>c2m +
p
N (1&minus; c2m)
</p>
<p>, m = 1, . . . ,M, (3.73)
</p>
<p>where cm is the mth canonical correlation coefficient. Note that as the ratio
of the number of input variables to sample size p/N gets small, the shrink-
age factors approach 1. Breiman and Friedman (1997) proposed modified
versions of Λ based on training data and cross-validation, but the general
form is the same. Here the fitted response has the form
</p>
<p>Ŷc+w = HYSc+w, (3.74)</p>
<p/>
</div>
<div class="page"><p/>
<p>86 3. Linear Methods for Regression
</p>
<p>where Sc+w = UΛU&minus;1 is the response shrinkage operator.
Breiman and Friedman (1997) also suggested shrinking in both the Y
</p>
<p>space and X space. This leads to hybrid shrinkage models of the form
</p>
<p>Ŷridge,c+w = AλYS
c+w, (3.75)
</p>
<p>where Aλ = X(X
TX+λI)&minus;1XT is the ridge regression shrinkage operator,
</p>
<p>as in (3.46) on page 66. Their paper and the discussions thereof contain
many more details.
</p>
<p>3.8 More on the Lasso and Related Path
Algorithms
</p>
<p>Since the publication of the LAR algorithm (Efron et al., 2004) there has
been a lot of activity in developing algorithms for fitting regularization
paths for a variety of different problems. In addition, L1 regularization has
taken on a life of its own, leading to the development of the field compressed
sensing in the signal-processing literature. (Donoho, 2006a; Candes, 2006).
In this section we discuss some related proposals and other path algorithms,
starting off with a precursor to the LAR algorithm.
</p>
<p>3.8.1 Incremental Forward Stagewise Regression
</p>
<p>Here we present another LAR-like algorithm, this time focused on forward
stagewise regression. Interestingly, efforts to understand a flexible nonlinear
regression procedure (boosting) led to a new algorithm for linear models
(LAR). In reading the first edition of this book and the forward stagewise
</p>
<p>Algorithm 3.4 Incremental Forward Stagewise Regression&mdash;FSǫ.
</p>
<p>1. Start with the residual r equal to y and β1, β2, . . . , βp = 0. All the
predictors are standardized to have mean zero and unit norm.
</p>
<p>2. Find the predictor xj most correlated with r
</p>
<p>3. Update βj &larr; βj + δj , where δj = ǫ &middot; sign[〈xj , r〉] and ǫ &gt; 0 is a small
step size, and set r &larr; r&minus; δjxj .
</p>
<p>4. Repeat steps 2 and 3 many times, until the residuals are uncorrelated
with all the predictors.
</p>
<p>Algorithm 16.1 of Chapter 164, our colleague Brad Efron realized that with
</p>
<p>4In the first edition, this was Algorithm 10.4 in Chapter 10.</p>
<p/>
</div>
<div class="page"><p/>
<p>3.8 More on the Lasso and Related Path Algorithms 87
</p>
<p>&minus;
0
.2
</p>
<p>0
.0
</p>
<p>0
.2
</p>
<p>0
.4
</p>
<p>0
.6
</p>
<p>lcavol
</p>
<p>lweight
</p>
<p>age
</p>
<p>lbph
</p>
<p>svi
</p>
<p>lcp
</p>
<p>gleason
</p>
<p>pgg45
</p>
<p>0 50 100 150 200
&minus;
</p>
<p>0
.2
</p>
<p>0
.0
</p>
<p>0
.2
</p>
<p>0
.4
</p>
<p>0
.6
</p>
<p>lcavol
</p>
<p>lweight
</p>
<p>age
</p>
<p>lbph
</p>
<p>svi
</p>
<p>lcp
</p>
<p>gleason
</p>
<p>pgg45
</p>
<p>0.0 0.5 1.0 1.5 2.0
</p>
<p>FSǫ FS0
</p>
<p>Iteration
</p>
<p>C
o
effi
</p>
<p>ci
en
ts
</p>
<p>C
o
effi
</p>
<p>ci
en
ts
</p>
<p>L1 Arc-length of Coefficients
</p>
<p>FIGURE 3.19. Coefficient profiles for the prostate data. The left panel shows
incremental forward stagewise regression with step size ǫ = 0.01. The right panel
shows the infinitesimal version FS0 obtained letting ǫ &rarr; 0. This profile was fit by
the modification 3.2b to the LAR Algorithm 3.2. In this example the FS0 profiles
are monotone, and hence identical to those of lasso and LAR.
</p>
<p>linear models, one could explicitly construct the piecewise-linear lasso paths
of Figure 3.10. This led him to propose the LAR procedure of Section 3.4.4,
as well as the incremental version of forward-stagewise regression presented
here.
Consider the linear-regression version of the forward-stagewise boosting
</p>
<p>algorithm 16.1 proposed in Section 16.1 (page 608). It generates a coefficient
profile by repeatedly updating (by a small amount ǫ) the coefficient of the
variable most correlated with the current residuals. Algorithm 3.4 gives
the details. Figure 3.19 (left panel) shows the progress of the algorithm on
the prostate data with step size ǫ = 0.01. If δj = 〈xj , r〉 (the least-squares
coefficient of the residual on jth predictor), then this is exactly the usual
forward stagewise procedure (FS) outlined in Section 3.3.3.
Here we are mainly interested in small values of ǫ. Letting ǫ &rarr; 0 gives
</p>
<p>the right panel of Figure 3.19, which in this case is identical to the lasso
path in Figure 3.10. We call this limiting procedure infinitesimal forward
stagewise regression or FS0. This procedure plays an important role in
non-linear, adaptive methods like boosting (Chapters 10 and 16) and is the
version of incremental forward stagewise regression that is most amenable
to theoretical analysis. Bühlmann and Hothorn (2007) refer to the same
procedure as &ldquo;L2boost&rdquo;, because of its connections to boosting.</p>
<p/>
</div>
<div class="page"><p/>
<p>88 3. Linear Methods for Regression
</p>
<p>Efron originally thought that the LAR Algorithm 3.2 was an implemen-
tation of FS0, allowing each tied predictor a chance to update their coeffi-
cients in a balanced way, while remaining tied in correlation. However, he
then realized that the LAR least-squares fit amongst the tied predictors
can result in coefficients moving in the opposite direction to their correla-
tion, which cannot happen in Algorithm 3.4. The following modification of
the LAR algorithm implements FS0:
</p>
<p>Algorithm 3.2b Least Angle Regression: FS0 Modification.
</p>
<p>4. Find the new direction by solving the constrained least squares prob-
lem
</p>
<p>min
b
</p>
<p>||r&minus;XAb||22 subject to bjsj &ge; 0, j &isin; A,
</p>
<p>where sj is the sign of 〈xj , r〉.
</p>
<p>The modification amounts to a non-negative least squares fit, keeping the
signs of the coefficients the same as those of the correlations. One can show
that this achieves the optimal balancing of infinitesimal &ldquo;update turns&rdquo;
for the variables tied for maximal correlation (Hastie et al., 2007). Like
lasso, the entire FS0 path can be computed very efficiently via the LAR
algorithm.
As a consequence of these results, if the LAR profiles are monotone non-
</p>
<p>increasing or non-decreasing, as they are in Figure 3.19, then all three
methods&mdash;LAR, lasso, and FS0&mdash;give identical profiles. If the profiles are
not monotone but do not cross the zero axis, then LAR and lasso are
identical.
Since FS0 is different from the lasso, it is natural to ask if it optimizes
</p>
<p>a criterion. The answer is more complex than for lasso; the FS0 coefficient
profile is the solution to a differential equation. While the lasso makes op-
timal progress in terms of reducing the residual sum-of-squares per unit
increase in L1-norm of the coefficient vector β, FS0 is optimal per unit
increase in L1 arc-length traveled along the coefficient path. Hence its co-
efficient path is discouraged from changing directions too often.
FS0 is more constrained than lasso, and in fact can be viewed as a mono-
</p>
<p>tone version of the lasso; see Figure 16.3 on page 614 for a dramatic exam-
ple. FS0 may be useful in p ≫ N situations, where its coefficient profiles
are much smoother and hence have less variance than those of lasso. More
details on FS0 are given in Section 16.2.3 and Hastie et al. (2007). Fig-
ure 3.16 includes FS0 where its performance is very similar to that of the
lasso.</p>
<p/>
</div>
<div class="page"><p/>
<p>3.8 More on the Lasso and Related Path Algorithms 89
</p>
<p>3.8.2 Piecewise-Linear Path Algorithms
</p>
<p>The least angle regression procedure exploits the piecewise linear nature of
the lasso solution paths. It has led to similar &ldquo;path algorithms&rdquo; for other
regularized problems. Suppose we solve
</p>
<p>β̂(λ) = argminβ [R(β) + λJ(β)] , (3.76)
</p>
<p>with
</p>
<p>R(β) =
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>L(yi, β0 +
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>xijβj), (3.77)
</p>
<p>where both the loss function L and the penalty function J are convex.
Then the following are sufficient conditions for the solution path β̂(λ) to
be piecewise linear (Rosset and Zhu, 2007):
</p>
<p>1. R is quadratic or piecewise-quadratic as a function of β, and
</p>
<p>2. J is piecewise linear in β.
</p>
<p>This also implies (in principle) that the solution path can be efficiently
computed. Examples include squared- and absolute-error loss, &ldquo;Huberized&rdquo;
losses, and the L1, L&infin; penalties on β. Another example is the &ldquo;hinge loss&rdquo;
function used in the support vector machine. There the loss is piecewise
linear, and the penalty is quadratic. Interestingly, this leads to a piecewise-
linear path algorithm in the dual space; more details are given in Sec-
tion 12.3.5.
</p>
<p>3.8.3 The Dantzig Selector
</p>
<p>Candes and Tao (2007) proposed the following criterion:
</p>
<p>minβ ||β||1 subject to ||XT (y &minus;Xβ)||&infin; &le; s. (3.78)
</p>
<p>They call the solution the Dantzig selector (DS). It can be written equiva-
lently as
</p>
<p>minβ ||XT (y &minus;Xβ)||&infin; subject to ||β||1 &le; t. (3.79)
</p>
<p>Here || &middot; ||&infin; denotes the L&infin; norm, the maximum absolute value of the
components of the vector. In this form it resembles the lasso, replacing
squared error loss by the maximum absolute value of its gradient. Note
that as t gets large, both procedures yield the least squares solution if
N &lt; p. If p &ge; N , they both yield the least squares solution with minimum
L1 norm. However for smaller values of t, the DS procedure produces a
different path of solutions than the lasso.
Candes and Tao (2007) show that the solution to DS is a linear pro-
</p>
<p>gramming problem; hence the name Dantzig selector, in honor of the late</p>
<p/>
</div>
<div class="page"><p/>
<p>90 3. Linear Methods for Regression
</p>
<p>George Dantzig, the inventor of the simplex method for linear program-
ming. They also prove a number of interesting mathematical properties for
the method, related to its ability to recover an underlying sparse coeffi-
cient vector. These same properties also hold for the lasso, as shown later
by Bickel et al. (2008).
Unfortunately the operating properties of the DS method are somewhat
</p>
<p>unsatisfactory. The method seems similar in spirit to the lasso, especially
when we look at the lasso&rsquo;s stationary conditions (3.58). Like the LAR al-
gorithm, the lasso maintains the same inner product (and correlation) with
the current residual for all variables in the active set, and moves their co-
efficients to optimally decrease the residual sum of squares. In the process,
this common correlation is decreased monotonically (Exercise 3.23), and at
all times this correlation is larger than that for non-active variables. The
Dantzig selector instead tries to minimize the maximum inner product of
the current residual with all the predictors. Hence it can achieve a smaller
maximum than the lasso, but in the process a curious phenomenon can
occur. If the size of the active set is m, there will be m variables tied with
maximum correlation. However, these need not coincide with the active set!
Hence it can include a variable in the model that has smaller correlation
with the current residual than some of the excluded variables (Efron et
al., 2007). This seems unreasonable and may be responsible for its some-
times inferior prediction accuracy. Efron et al. (2007) also show that DS
can yield extremely erratic coefficient paths as the regularization parameter
s is varied.
</p>
<p>3.8.4 The Grouped Lasso
</p>
<p>In some problems, the predictors belong to pre-defined groups; for example
genes that belong to the same biological pathway, or collections of indicator
(dummy) variables for representing the levels of a categorical predictor. In
this situation it may be desirable to shrink and select the members of a
group together. The grouped lasso is one way to achieve this. Suppose that
the p predictors are divided into L groups, with pℓ the number in group
ℓ. For ease of notation, we use a matrix Xℓ to represent the predictors
corresponding to the ℓth group, with corresponding coefficient vector βℓ.
The grouped-lasso minimizes the convex criterion
</p>
<p>min
β&isin;IRp
</p>
<p>(
||y &minus; β01&minus;
</p>
<p>L&sum;
</p>
<p>ℓ=1
</p>
<p>Xℓβℓ||22 + λ
L&sum;
</p>
<p>ℓ=1
</p>
<p>&radic;
pℓ||βℓ||2
</p>
<p>)
, (3.80)
</p>
<p>where the
&radic;
pℓ terms accounts for the varying group sizes, and || &middot; ||2 is
</p>
<p>the Euclidean norm (not squared). Since the Euclidean norm of a vector
βℓ is zero only if all of its components are zero, this procedure encourages
sparsity at both the group and individual levels. That is, for some values of
λ, an entire group of predictors may drop out of the model. This procedure</p>
<p/>
</div>
<div class="page"><p/>
<p>3.8 More on the Lasso and Related Path Algorithms 91
</p>
<p>was proposed by Bakin (1999) and Lin and Zhang (2006), and studied and
generalized by Yuan and Lin (2007). Generalizations include more general
L2 norms ||η||K = (ηTKη)1/2, as well as allowing overlapping groups of
predictors (Zhao et al., 2008). There are also connections to methods for
fitting sparse additive models (Lin and Zhang, 2006; Ravikumar et al.,
2008).
</p>
<p>3.8.5 Further Properties of the Lasso
</p>
<p>A number of authors have studied the ability of the lasso and related pro-
cedures to recover the correct model, as N and p grow. Examples of this
work include Knight and Fu (2000), Greenshtein and Ritov (2004), Tropp
(2004), Donoho (2006b), Meinshausen (2007), Meinshausen and Bühlmann
(2006), Tropp (2006), Zhao and Yu (2006), Wainwright (2006), and Bunea
et al. (2007). For example Donoho (2006b) focuses on the p &gt; N case and
considers the lasso solution as the bound t gets large. In the limit this gives
the solution with minimum L1 norm among all models with zero training
error. He shows that under certain assumptions on the model matrix X, if
the true model is sparse, this solution identifies the correct predictors with
high probability.
Many of the results in this area assume a condition on the model matrix
</p>
<p>of the form
</p>
<p>max
j&isin;Sc
</p>
<p>||xTj XS(XSTXS)&minus;1||1 &le; (1&minus; ǫ) for some ǫ &isin; (0, 1]. (3.81)
</p>
<p>Here S indexes the subset of features with non-zero coefficients in the true
underlying model, and XS are the columns of X corresponding to those
features. Similarly Sc are the features with true coefficients equal to zero,
and XSc the corresponding columns. This says that the least squares coef-
ficients for the columns of XSc on XS are not too large, that is, the &ldquo;good&rdquo;
variables S are not too highly correlated with the nuisance variables Sc.
Regarding the coefficients themselves, the lasso shrinkage causes the esti-
</p>
<p>mates of the non-zero coefficients to be biased towards zero, and in general
they are not consistent5. One approach for reducing this bias is to run
the lasso to identify the set of non-zero coefficients, and then fit an un-
restricted linear model to the selected set of features. This is not always
feasible, if the selected set is large. Alternatively, one can use the lasso to
select the set of non-zero predictors, and then apply the lasso again, but
using only the selected predictors from the first step. This is known as the
relaxed lasso (Meinshausen, 2007). The idea is to use cross-validation to
estimate the initial penalty parameter for the lasso, and then again for a
second penalty parameter applied to the selected set of predictors. Since
</p>
<p>5Statistical consistency means as the sample size grows, the estimates converge to
</p>
<p>the true values.</p>
<p/>
</div>
<div class="page"><p/>
<p>92 3. Linear Methods for Regression
</p>
<p>the variables in the second step have less &ldquo;competition&rdquo; from noise vari-
ables, cross-validation will tend to pick a smaller value for λ, and hence
their coefficients will be shrunken less than those in the initial estimate.
Alternatively, one can modify the lasso penalty function so that larger co-
</p>
<p>efficients are shrunken less severely; the smoothly clipped absolute deviation
(SCAD) penalty of Fan and Li (2005) replaces λ|β| by Ja(β, λ), where
</p>
<p>dJa(β, λ)
</p>
<p>dβ
= λ &middot; sign(β)
</p>
<p>[
I(|β| &le; λ) + (aλ&minus; |β|)+
</p>
<p>(a&minus; 1)λ I(|β| &gt; λ)
]
</p>
<p>(3.82)
</p>
<p>for some a &ge; 2. The second term in square-braces reduces the amount of
shrinkage in the lasso for larger values of β, with ultimately no shrinkage
as a &rarr; &infin;. Figure 3.20 shows the SCAD penalty, along with the lasso and
</p>
<p>&minus;4 &minus;2 0 2 4
</p>
<p>0
1
</p>
<p>2
3
</p>
<p>4
5
</p>
<p>&minus;4 &minus;2 0 2 4
</p>
<p>0
.0
</p>
<p>0
.5
</p>
<p>1
.0
</p>
<p>1
.5
</p>
<p>2
.0
</p>
<p>2
.5
</p>
<p>&minus;4 &minus;2 0 2 4
</p>
<p>0
.5
</p>
<p>1
.0
</p>
<p>1
.5
</p>
<p>2
.0
</p>
<p>|β| SCAD |β|1&minus;ν
</p>
<p>βββ
</p>
<p>FIGURE 3.20. The lasso and two alternative non-convex penalties designed to
penalize large coefficients less. For SCAD we use λ = 1 and a = 4, and ν = 1
</p>
<p>2
in
</p>
<p>the last panel.
</p>
<p>|β|1&minus;ν . However this criterion is non-convex, which is a drawback since it
makes the computation much more difficult. The adaptive lasso (Zou, 2006)
</p>
<p>uses a weighted penalty of the form
&sum;p
</p>
<p>j=1 wj |βj | where wj = 1/|β̂j |ν , β̂j is
the ordinary least squares estimate and ν &gt; 0. This is a practical approxi-
mation to the |β|q penalties (q = 1&minus;ν here) discussed in Section 3.4.3. The
adaptive lasso yields consistent estimates of the parameters while retaining
the attractive convexity property of the lasso.
</p>
<p>3.8.6 Pathwise Coordinate Optimization
</p>
<p>An alternate approach to the LARS algorithm for computing the lasso
solution is simple coordinate descent. This idea was proposed by Fu (1998)
and Daubechies et al. (2004), and later studied and generalized by Friedman
et al. (2007), Wu and Lange (2008) and others. The idea is to fix the penalty
parameter λ in the Lagrangian form (3.52) and optimize successively over
each parameter, holding the other parameters fixed at their current values.
Suppose the predictors are all standardized to have mean zero and unit
</p>
<p>norm. Denote by β̃k(λ) the current estimate for βk at penalty parameter</p>
<p/>
</div>
<div class="page"><p/>
<p>3.9 Computational Considerations 93
</p>
<p>λ. We can rearrange (3.52) to isolate βj ,
</p>
<p>R(β̃(λ), βj) =
1
</p>
<p>2
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>(
yi &minus;
</p>
<p>&sum;
</p>
<p>k �=j
</p>
<p>xikβ̃k(λ)&minus; xijβj
)2
</p>
<p>+ λ
&sum;
</p>
<p>k �=j
</p>
<p>|β̃k(λ)|+ λ|βj |,
</p>
<p>(3.83)
where we have suppressed the intercept and introduced a factor 12 for con-
venience. This can be viewed as a univariate lasso problem with response
</p>
<p>variable the partial residual yi &minus; ỹ(j)i = yi &minus;
&sum;
</p>
<p>k �=j xikβ̃k(λ). This has an
explicit solution, resulting in the update
</p>
<p>β̃j(λ) &larr; S
(
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>xij(yi &minus; ỹ(j)i ), λ
)
. (3.84)
</p>
<p>Here S(t, λ) = sign(t)(|t|&minus;λ)+ is the soft-thresholding operator in Table 3.4
on page 71. The first argument to S(&middot;) is the simple least-squares coefficient
of the partial residual on the standardized variable xij . Repeated iteration
of (3.84)&mdash;cycling through each variable in turn until convergence&mdash;yields
</p>
<p>the lasso estimate β̂(λ).
We can also use this simple algorithm to efficiently compute the lasso
</p>
<p>solutions at a grid of values of λ. We start with the smallest value λmax
for which β̂(λmax) = 0, decrease it a little and cycle through the variables
until convergence. Then λ is decreased again and the process is repeated,
using the previous solution as a &ldquo;warm start&rdquo; for the new value of λ. This
can be faster than the LARS algorithm, especially in large problems. A
key to its speed is the fact that the quantities in (3.84) can be updated
quickly as j varies, and often the update is to leave β̃j = 0. On the other
hand, it delivers solutions over a grid of λ values, rather than the entire
solution path. The same kind of algorithm can be applied to the elastic
net, the grouped lasso and many other models in which the penalty is a
sum of functions of the individual parameters (Friedman et al., 2010). It
can also be applied, with some substantial modifications, to the fused lasso
(Section 18.4.2); details are in Friedman et al. (2007).
</p>
<p>3.9 Computational Considerations
</p>
<p>Least squares fitting is usually done via the Cholesky decomposition of
the matrix XTX or a QR decomposition of X. With N observations and p
features, the Cholesky decomposition requires p3+Np2/2 operations, while
the QR decomposition requires Np2 operations. Depending on the relative
size of N and p, the Cholesky can sometimes be faster; on the other hand,
it can be less numerically stable (Lawson and Hansen, 1974). Computation
of the lasso via the LAR algorithm has the same order of computation as
a least squares fit.</p>
<p/>
</div>
<div class="page"><p/>
<p>94 3. Linear Methods for Regression
</p>
<p>Bibliographic Notes
</p>
<p>Linear regression is discussed in many statistics books, for example, Seber
(1984), Weisberg (1980) and Mardia et al. (1979). Ridge regression was
introduced by Hoerl and Kennard (1970), while the lasso was proposed by
Tibshirani (1996). Around the same time, lasso-type penalties were pro-
posed in the basis pursuit method for signal processing (Chen et al., 1998).
The least angle regression procedure was proposed in Efron et al. (2004);
related to this is the earlier homotopy procedure of Osborne et al. (2000a)
and Osborne et al. (2000b). Their algorithm also exploits the piecewise
linearity used in the LAR/lasso algorithm, but lacks its transparency. The
criterion for the forward stagewise criterion is discussed in Hastie et al.
(2007). Park and Hastie (2007) develop a path algorithm similar to least
angle regression for generalized regression models. Partial least squares
was introduced by Wold (1975). Comparisons of shrinkage methods may
be found in Copas (1983) and Frank and Friedman (1993).
</p>
<p>Exercises
</p>
<p>Ex. 3.1 Show that the F statistic (3.13) for dropping a single coefficient
from a model is equal to the square of the corresponding z-score (3.12).
</p>
<p>Ex. 3.2 Given data on two variables X and Y , consider fitting a cubic
polynomial regression model f(X) =
</p>
<p>&sum;3
j=0 βjX
</p>
<p>j . In addition to plotting
the fitted curve, you would like a 95% confidence band about the curve.
Consider the following two approaches:
</p>
<p>1. At each point x0, form a 95% confidence interval for the linear func-
tion aTβ =
</p>
<p>&sum;3
j=0 βjx
</p>
<p>j
0.
</p>
<p>2. Form a 95% confidence set for β as in (3.15), which in turn generates
confidence intervals for f(x0).
</p>
<p>How do these approaches differ? Which band is likely to be wider? Conduct
a small simulation experiment to compare the two methods.
</p>
<p>Ex. 3.3 Gauss&ndash;Markov theorem:
</p>
<p>(a) Prove the Gauss&ndash;Markov theorem: the least squares estimate of a
parameter aTβ has variance no bigger than that of any other linear
unbiased estimate of aTβ (Section 3.2.2).
</p>
<p>(b) The matrix inequality B � A holds if A&minus;B is positive semidefinite.
Show that if V̂ is the variance-covariance matrix of the least squares
estimate of β and Ṽ is the variance-covariance matrix of any other
linear unbiased estimate, then V̂ � Ṽ.</p>
<p/>
</div>
<div class="page"><p/>
<p>Exercises 95
</p>
<p>Ex. 3.4 Show how the vector of least squares coefficients can be obtained
from a single pass of the Gram&ndash;Schmidt procedure (Algorithm 3.1). Rep-
resent your solution in terms of the QR decomposition of X.
</p>
<p>Ex. 3.5 Consider the ridge regression problem (3.41). Show that this prob-
lem is equivalent to the problem
</p>
<p>β̂c = argmin
βc
</p>
<p>{
N&sum;
</p>
<p>i=1
</p>
<p>[
yi &minus; βc0 &minus;
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>(xij &minus; x̄j)βcj
]2
</p>
<p>+ λ
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>βcj
2
</p>
<p>}
. (3.85)
</p>
<p>Give the correspondence between βc and the original β in (3.41). Char-
acterize the solution to this modified criterion. Show that a similar result
holds for the lasso.
</p>
<p>Ex. 3.6 Show that the ridge regression estimate is the mean (and mode)
of the posterior distribution, under a Gaussian prior β &sim; N(0, τI), and
Gaussian sampling model y &sim; N(Xβ, σ2I). Find the relationship between
the regularization parameter λ in the ridge formula, and the variances τ
and σ2.
</p>
<p>Ex. 3.7 Assume yi &sim; N(β0 + xTi β, σ2), i = 1, 2, . . . , N , and the parameters
βj , j = 1, . . . , p are each distributed as N(0, τ
</p>
<p>2), independently of one
another. Assuming σ2 and τ2 are known, and β0 is not governed by a
prior (or has a flat improper prior), show that the (minus) log-posterior
</p>
<p>density of β is proportional to
&sum;N
</p>
<p>i=1(yi &minus; β0 &minus;
&sum;
</p>
<p>j xijβj)
2 + λ
</p>
<p>&sum;p
j=1 β
</p>
<p>2
j
</p>
<p>where λ = σ2/τ2.
</p>
<p>Ex. 3.8 Consider the QR decomposition of the uncentered N &times; (p + 1)
matrix X (whose first column is all ones), and the SVD of the N &times; p
centered matrix X̃. Show that Q2 and U span the same subspace, where
Q2 is the sub-matrix of Q with the first column removed. Under what
circumstances will they be the same, up to sign flips?
</p>
<p>Ex. 3.9 Forward stepwise regression. Suppose we have the QR decomposi-
tion for the N&times;q matrix X1 in a multiple regression problem with response
y, and we have an additional p&minus;q predictors in the matrix X2. Denote the
current residual by r. We wish to establish which one of these additional
variables will reduce the residual-sum-of squares the most when included
with those in X1. Describe an efficient procedure for doing this.
</p>
<p>Ex. 3.10 Backward stepwise regression. Suppose we have the multiple re-
gression fit of y on X, along with the standard errors and Z-scores as in
Table 3.2. We wish to establish which variable, when dropped, will increase
the residual sum-of-squares the least. How would you do this?
</p>
<p>Ex. 3.11 Show that the solution to the multivariate linear regression prob-
lem (3.40) is given by (3.39). What happens if the covariance matrices Σi
are different for each observation?</p>
<p/>
</div>
<div class="page"><p/>
<p>96 3. Linear Methods for Regression
</p>
<p>Ex. 3.12 Show that the ridge regression estimates can be obtained by
ordinary least squares regression on an augmented data set. We augment
the centered matrix X with p additional rows
</p>
<p>&radic;
λI, and augment y with p
</p>
<p>zeros. By introducing artificial data having response value zero, the fitting
procedure is forced to shrink the coefficients toward zero. This is related to
the idea of hints due to Abu-Mostafa (1995), where model constraints are
implemented by adding artificial data examples that satisfy them.
</p>
<p>Ex. 3.13 Derive the expression (3.62), and show that β̂pcr(p) = β̂ls.
</p>
<p>Ex. 3.14 Show that in the orthogonal case, PLS stops after m = 1 steps,
because subsequent ϕ̂mj in step 2 in Algorithm 3.3 are zero.
</p>
<p>Ex. 3.15 Verify expression (3.64), and hence show that the partial least
squares directions are a compromise between the ordinary regression coef-
ficient and the principal component directions.
</p>
<p>Ex. 3.16 Derive the entries in Table 3.4, the explicit forms for estimators
in the orthogonal case.
</p>
<p>Ex. 3.17 Repeat the analysis of Table 3.3 on the spam data discussed in
Chapter 1.
</p>
<p>Ex. 3.18 Read about conjugate gradient algorithms (Murray et al., 1981, for
example), and establish a connection between these algorithms and partial
least squares.
</p>
<p>Ex. 3.19 Show that ‖β̂ridge‖ increases as its tuning parameter λ &rarr; 0. Does
the same property hold for the lasso and partial least squares estimates?
For the latter, consider the &ldquo;tuning parameter&rdquo; to be the successive steps
in the algorithm.
</p>
<p>Ex. 3.20 Consider the canonical-correlation problem (3.67). Show that the
leading pair of canonical variates u1 and v1 solve the problem
</p>
<p>max
uT (YT Y)u=1
</p>
<p>vT (X
T X)v=1
</p>
<p>uT (YTX)v, (3.86)
</p>
<p>a generalized SVD problem. Show that the solution is given by u1 =
(YTY)&minus;
</p>
<p>1
2u&lowast;1, and v1 = (X
</p>
<p>TX)&minus;
1
2 v&lowast;1 , where u
</p>
<p>&lowast;
1 and v
</p>
<p>&lowast;
1 are the leading left
</p>
<p>and right singular vectors in
</p>
<p>(YTY)&minus;
1
2 (YTX)(XTX)&minus;
</p>
<p>1
2 = U&lowast;D&lowast;V&lowast;T . (3.87)
</p>
<p>Show that the entire sequence um, vm, m = 1, . . . ,min(K, p) is also given
by (3.87).
</p>
<p>Ex. 3.21 Show that the solution to the reduced-rank regression problem
(3.68), with Σ estimated by YTY/N , is given by (3.69). Hint: Transform</p>
<p/>
</div>
<div class="page"><p/>
<p>Exercises 97
</p>
<p>Y to Y&lowast; = YΣ&minus;
1
2 , and solved in terms of the canonical vectors u&lowast;m. Show
</p>
<p>that Um = Σ
&minus; 12U&lowast;m, and a generalized inverse is U
</p>
<p>&minus;
m = U
</p>
<p>&lowast;
m
</p>
<p>TΣ
1
2 .
</p>
<p>Ex. 3.22 Show that the solution in Exercise 3.21 does not change if Σ is
estimated by the more natural quantity (Y&minus;XB̂)T (Y&minus;XB̂)/(N &minus; pK).
</p>
<p>Ex. 3.23 Consider a regression problem with all variables and response hav-
ing mean zero and standard deviation one. Suppose also that each variable
has identical absolute correlation with the response:
</p>
<p>1
</p>
<p>N
|〈xj ,y〉| = λ, j = 1, . . . , p.
</p>
<p>Let β̂ be the least-squares coefficient of y on X, and let u(α) = αXβ̂ for
α &isin; [0, 1] be the vector that moves a fraction α toward the least squares fit
u. Let RSS be the residual sum-of-squares from the full least squares fit.
</p>
<p>(a) Show that
</p>
<p>1
</p>
<p>N
|〈xj ,y &minus; u(α)〉| = (1&minus; α)λ, j = 1, . . . , p,
</p>
<p>and hence the correlations of each xj with the residuals remain equal
in magnitude as we progress toward u.
</p>
<p>(b) Show that these correlations are all equal to
</p>
<p>λ(α) =
(1&minus; α)&radic;
</p>
<p>(1&minus; α)2 + α(2&minus;α)N &middot;RSS
&middot; λ,
</p>
<p>and hence they decrease monotonically to zero.
</p>
<p>(c) Use these results to show that the LAR algorithm in Section 3.4.4
keeps the correlations tied and monotonically decreasing, as claimed
in (3.55).
</p>
<p>Ex. 3.24 LAR directions. Using the notation around equation (3.55) on
page 74, show that the LAR direction makes an equal angle with each of
the predictors in Ak.
</p>
<p>Ex. 3.25 LAR look-ahead (Efron et al., 2004, Sec. 2). Starting at the be-
ginning of the kth step of the LAR algorithm, derive expressions to identify
the next variable to enter the active set at step k+1, and the value of α at
which this occurs (using the notation around equation (3.55) on page 74).
</p>
<p>Ex. 3.26 Forward stepwise regression enters the variable at each step that
most reduces the residual sum-of-squares. LAR adjusts variables that have
the most (absolute) correlation with the current residuals. Show that these
two entry criteria are not necessarily the same. [Hint: let xj.A be the jth</p>
<p/>
</div>
<div class="page"><p/>
<p>98 3. Linear Methods for Regression
</p>
<p>variable, linearly adjusted for all the variables currently in the model. Show
that the first criterion amounts to identifying the j for which Cor(xj.A, r)
is largest in magnitude.
</p>
<p>Ex. 3.27 Lasso and LAR: Consider the lasso problem in Lagrange multiplier
form: with L(β) = 12
</p>
<p>&sum;
i(yi &minus;
</p>
<p>&sum;
j xijβj)
</p>
<p>2, we minimize
</p>
<p>L(β) + λ
&sum;
</p>
<p>j
</p>
<p>|βj | (3.88)
</p>
<p>for fixed λ &gt; 0.
</p>
<p>(a) Setting βj = β
+
j &minus; β&minus;j with β+j , β&minus;j &ge; 0, expression (3.88) becomes
</p>
<p>L(β) + λ
&sum;
</p>
<p>j(β
+
j + β
</p>
<p>&minus;
j ). Show that the Lagrange dual function is
</p>
<p>L(β) + λ
&sum;
</p>
<p>j
</p>
<p>(β+j + β
&minus;
j )&minus;
</p>
<p>&sum;
</p>
<p>j
</p>
<p>λ+j β
+
j &minus;
</p>
<p>&sum;
</p>
<p>j
</p>
<p>λ&minus;j β
&minus;
j (3.89)
</p>
<p>and the Karush&ndash;Kuhn&ndash;Tucker optimality conditions are
</p>
<p>&nabla;L(β)j + λ&minus; λ+j = 0
&minus;&nabla;L(β)j + λ&minus; λ&minus;j = 0
</p>
<p>λ+j β
+
j = 0
</p>
<p>λ&minus;j β
&minus;
j = 0,
</p>
<p>along with the non-negativity constraints on the parameters and all
the Lagrange multipliers.
</p>
<p>(b) Show that |&nabla;L(β)j | &le; λ &forall;j, and that the KKT conditions imply one
of the following three scenarios:
</p>
<p>λ = 0 &rArr; &nabla;L(β)j = 0 &forall;j
β+j &gt; 0, λ &gt; 0 &rArr; λ+j = 0, &nabla;L(β)j = &minus;λ &lt; 0, β&minus;j = 0
β&minus;j &gt; 0, λ &gt; 0 &rArr; λ&minus;j = 0, &nabla;L(β)j = λ &gt; 0, β+j = 0.
</p>
<p>Hence show that for any &ldquo;active&rdquo; predictor having βj �= 0, we must
have &nabla;L(β)j = &minus;λ if βj &gt; 0, and &nabla;L(β)j = λ if βj &lt; 0. Assuming
the predictors are standardized, relate λ to the correlation between
the jth predictor and the current residuals.
</p>
<p>(c) Suppose that the set of active predictors is unchanged for λ0 &ge; λ &ge; λ1.
Show that there is a vector γ0 such that
</p>
<p>β̂(λ) = β̂(λ0)&minus; (λ&minus; λ0)γ0 (3.90)
</p>
<p>Thus the lasso solution path is linear as λ ranges from λ0 to λ1(Efron
et al., 2004; Rosset and Zhu, 2007).</p>
<p/>
</div>
<div class="page"><p/>
<p>Exercises 99
</p>
<p>Ex. 3.28 Suppose for a given t in (3.51), the fitted lasso coefficient for
</p>
<p>variable Xj is β̂j = a. Suppose we augment our set of variables with an
identical copy X&lowast;j = Xj . Characterize the effect of this exact collinearity
</p>
<p>by describing the set of solutions for β̂j and β̂
&lowast;
j , using the same value of t.
</p>
<p>Ex. 3.29 Suppose we run a ridge regression with parameter λ on a single
variable X, and get coefficient a. We now include an exact copy X&lowast; = X,
and refit our ridge regression. Show that both coefficients are identical, and
derive their value. Show in general that if m copies of a variable Xj are
included in a ridge regression, their coefficients are all the same.
</p>
<p>Ex. 3.30 Consider the elastic-net optimization problem:
</p>
<p>min
β
</p>
<p>||y &minus;Xβ||2 + λ
[
α||β||22 + (1&minus; α)||β||1
</p>
<p>]
. (3.91)
</p>
<p>Show how one can turn this into a lasso problem, using an augmented
version of X and y.</p>
<p/>
</div>
<div class="page"><p/>
<p>4
Linear Methods for Classification
</p>
<p>4.1 Introduction
</p>
<p>In this chapter we revisit the classification problem and focus on linear
methods for classification. Since our predictor G(x) takes values in a dis-
crete set G, we can always divide the input space into a collection of regions
labeled according to the classification. We saw in Chapter 2 that the bound-
aries of these regions can be rough or smooth, depending on the prediction
function. For an important class of procedures, these decision boundaries
are linear; this is what we will mean by linear methods for classification.
There are several different ways in which linear decision boundaries can
</p>
<p>be found. In Chapter 2 we fit linear regression models to the class indicator
variables, and classify to the largest fit. Suppose there are K classes, for
convenience labeled 1, 2, . . . ,K, and the fitted linear model for the kth
indicator response variable is f̂k(x) = β̂k0 + β̂
</p>
<p>T
k x. The decision boundary
</p>
<p>between class k and ℓ is that set of points for which f̂k(x) = f̂ℓ(x), that is,
</p>
<p>the set {x : (β̂k0 &minus; β̂ℓ0) + (β̂k &minus; β̂ℓ)Tx = 0}, an affine set or hyperplane.1
Since the same is true for any pair of classes, the input space is divided
into regions of constant classification, with piecewise hyperplanar decision
boundaries. This regression approach is a member of a class of methods
that model discriminant functions δk(x) for each class, and then classify x
to the class with the largest value for its discriminant function. Methods
</p>
<p>1Strictly speaking, a hyperplane passes through the origin, while an affine set need
not. We sometimes ignore the distinction and refer in general to hyperplanes.
</p>
<p> &copy; Springer Science+Business Media, LLC 2009 
</p>
<p>T. Hastie et al., The Elements of Statistical Learning, Second Edition, 101
</p>
<p>DOI: 10.1007/b94608_4,</p>
<p/>
</div>
<div class="page"><p/>
<p>102 4. Linear Methods for Classification
</p>
<p>that model the posterior probabilities Pr(G = k|X = x) are also in this
class. Clearly, if either the δk(x) or Pr(G = k|X = x) are linear in x, then
the decision boundaries will be linear.
Actually, all we require is that some monotone transformation of δk or
</p>
<p>Pr(G = k|X = x) be linear for the decision boundaries to be linear. For
example, if there are two classes, a popular model for the posterior proba-
bilities is
</p>
<p>Pr(G = 1|X = x) = exp(β0 + β
Tx)
</p>
<p>1 + exp(β0 + βTx)
,
</p>
<p>Pr(G = 2|X = x) = 1
1 + exp(β0 + βTx)
</p>
<p>.
</p>
<p>(4.1)
</p>
<p>Here the monotone transformation is the logit transformation: log[p/(1&minus;p)],
and in fact we see that
</p>
<p>log
Pr(G = 1|X = x)
Pr(G = 2|X = x) = β0 + β
</p>
<p>Tx. (4.2)
</p>
<p>The decision boundary is the set of points for which the log-odds are zero,
and this is a hyperplane defined by
</p>
<p>{
x|β0 + βTx = 0
</p>
<p>}
. We discuss two very
</p>
<p>popular but different methods that result in linear log-odds or logits: linear
discriminant analysis and linear logistic regression. Although they differ in
their derivation, the essential difference between them is in the way the
linear function is fit to the training data.
A more direct approach is to explicitly model the boundaries between
</p>
<p>the classes as linear. For a two-class problem in a p-dimensional input
space, this amounts to modeling the decision boundary as a hyperplane&mdash;in
other words, a normal vector and a cut-point. We will look at two methods
that explicitly look for &ldquo;separating hyperplanes.&rdquo; The first is the well-
known perceptron model of Rosenblatt (1958), with an algorithm that finds
a separating hyperplane in the training data, if one exists. The second
method, due to Vapnik (1996), finds an optimally separating hyperplane if
one exists, else finds a hyperplane that minimizes some measure of overlap
in the training data. We treat the separable case here, and defer treatment
of the nonseparable case to Chapter 12.
While this entire chapter is devoted to linear decision boundaries, there is
</p>
<p>considerable scope for generalization. For example, we can expand our vari-
able setX1, . . . , Xp by including their squares and cross-productsX
</p>
<p>2
1 , X
</p>
<p>2
2 , . . . ,
</p>
<p>X1X2, . . ., thereby adding p(p+1)/2 additional variables. Linear functions
in the augmented space map down to quadratic functions in the original
space&mdash;hence linear decision boundaries to quadratic decision boundaries.
Figure 4.1 illustrates the idea. The data are the same: the left plot uses
linear decision boundaries in the two-dimensional space shown, while the
right plot uses linear decision boundaries in the augmented five-dimensional
space described above. This approach can be used with any basis transfor-</p>
<p/>
</div>
<div class="page"><p/>
<p>4.2 Linear Regression of an Indicator Matrix 103
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>11
</p>
<p>1
1
</p>
<p>1
1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>11
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
11
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
11
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
2
</p>
<p>2
2
</p>
<p>22
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2 2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2 2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>22
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2 2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
22
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2 2
</p>
<p>2
2 2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>22
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2 2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>22
</p>
<p>2 2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3 3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3 3
</p>
<p>3 3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3 3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3 3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3 3
</p>
<p>3
</p>
<p>3
</p>
<p>3 3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3 3
</p>
<p>3
3
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>11
</p>
<p>1
1
</p>
<p>1
1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>11
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
11
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
11
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
2
</p>
<p>2
2
</p>
<p>22
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2 2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2 2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>22
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2 2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
22
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2 2
</p>
<p>2
2 2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>22
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2 2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>22
</p>
<p>2 2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3 3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3 3
</p>
<p>3 3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3 3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3 3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3 3
</p>
<p>3
</p>
<p>3
</p>
<p>3 3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3 3
</p>
<p>3
3
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>FIGURE 4.1. The left plot shows some data from three classes, with linear
decision boundaries found by linear discriminant analysis. The right plot shows
quadratic decision boundaries. These were obtained by finding linear boundaries
in the five-dimensional space X1, X2, X1X2, X
</p>
<p>2
1 , X
</p>
<p>2
2 . Linear inequalities in this
</p>
<p>space are quadratic inequalities in the original space.
</p>
<p>mation h(X) where h : IRp �&rarr; IRq with q &gt; p, and will be explored in later
chapters.
</p>
<p>4.2 Linear Regression of an Indicator Matrix
</p>
<p>Here each of the response categories are coded via an indicator variable.
Thus if G has K classes, there will be K such indicators Yk, k = 1, . . . ,K,
with Yk = 1 if G = k else 0. These are collected together in a vector
Y = (Y1, . . . , YK), and the N training instances of these form an N &times; K
indicator response matrix Y. Y is a matrix of 0&rsquo;s and 1&rsquo;s, with each row
having a single 1. We fit a linear regression model to each of the columns
of Y simultaneously, and the fit is given by
</p>
<p>Ŷ = X(XTX)&minus;1XTY. (4.3)
</p>
<p>Chapter 3 has more details on linear regression. Note that we have a coeffi-
cient vector for each response column yk, and hence a (p+1)&times;K coefficient
matrix B̂ = (XTX)&minus;1XTY. Here X is the model matrix with p+1 columns
corresponding to the p inputs, and a leading column of 1&rsquo;s for the intercept.
A new observation with input x is classified as follows:
</p>
<p>&bull; compute the fitted output f̂(x)T = (1, xT )B̂, a K vector;
</p>
<p>&bull; identify the largest component and classify accordingly:
</p>
<p>Ĝ(x) = argmaxk&isin;G f̂k(x). (4.4)</p>
<p/>
</div>
<div class="page"><p/>
<p>104 4. Linear Methods for Classification
</p>
<p>What is the rationale for this approach? One rather formal justification
is to view the regression as an estimate of conditional expectation. For the
random variable Yk, E(Yk|X = x) = Pr(G = k|X = x), so conditional
expectation of each of the Yk seems a sensible goal. The real issue is: how
good an approximation to conditional expectation is the rather rigid linear
regression model? Alternatively, are the f̂k(x) reasonable estimates of the
posterior probabilities Pr(G = k|X = x), and more importantly, does this
matter?
It is quite straightforward to verify that
</p>
<p>&sum;
k&isin;G f̂k(x) = 1 for any x, as
</p>
<p>long as there is an intercept in the model (column of 1&rsquo;s in X). However,
</p>
<p>the f̂k(x) can be negative or greater than 1, and typically some are. This
is a consequence of the rigid nature of linear regression, especially if we
make predictions outside the hull of the training data. These violations in
themselves do not guarantee that this approach will not work, and in fact
on many problems it gives similar results to more standard linear meth-
ods for classification. If we allow linear regression onto basis expansions
h(X) of the inputs, this approach can lead to consistent estimates of the
probabilities. As the size of the training set N grows bigger, we adaptively
include more basis elements so that linear regression onto these basis func-
tions approaches conditional expectation. We discuss such approaches in
Chapter 5.
A more simplistic viewpoint is to construct targets tk for each class,
</p>
<p>where tk is the kth column of the K &times;K identity matrix. Our prediction
problem is to try and reproduce the appropriate target for an observation.
With the same coding as before, the response vector yi (ith row of Y) for
observation i has the value yi = tk if gi = k. We might then fit the linear
model by least squares:
</p>
<p>min
B
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>||yi &minus; [(1, xTi )B]T ||2. (4.5)
</p>
<p>The criterion is a sum-of-squared Euclidean distances of the fitted vectors
from their targets. A new observation is classified by computing its fitted
vector f̂(x) and classifying to the closest target:
</p>
<p>Ĝ(x) = argmin
k
</p>
<p>||f̂(x)&minus; tk||2. (4.6)
</p>
<p>This is exactly the same as the previous approach:
</p>
<p>&bull; The sum-of-squared-norm criterion is exactly the criterion for multi-
ple response linear regression, just viewed slightly differently. Since
a squared norm is itself a sum of squares, the components decouple
and can be rearranged as a separate linear model for each element.
Note that this is only possible because there is nothing in the model
that binds the different responses together.</p>
<p/>
</div>
<div class="page"><p/>
<p>4.2 Linear Regression of an Indicator Matrix 105
</p>
<p>Linear Regression
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>11
11
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 11
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>11
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1 1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>11
1 1
</p>
<p>1
</p>
<p>1
1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>11
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>11
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>11
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>11
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>11
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>11
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>11
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
11
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>11
1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>11
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>11
</p>
<p>1
</p>
<p>1 1
</p>
<p>11
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>11
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
11
</p>
<p>1
1
</p>
<p>1
1
</p>
<p>1
</p>
<p>11
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>11
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>11
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
1
</p>
<p>11
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>11
</p>
<p>1 1
</p>
<p>1
1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
1
</p>
<p>1 1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>11
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
2
</p>
<p>22
</p>
<p>2
</p>
<p>2 2
</p>
<p>2
</p>
<p>2 2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2 2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2 22
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>22
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
22
</p>
<p>2
</p>
<p>22
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2
</p>
<p>2
</p>
<p>2 2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2 2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2 2
</p>
<p>2
</p>
<p>2 2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>22
2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2 2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2 2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2 2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2 2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2
</p>
<p>2 22
</p>
<p>2
</p>
<p>2
</p>
<p>2 2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2 2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2 2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>22
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>22 2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2 2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2
</p>
<p>2 2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3 3 3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>33
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>33
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>33
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3 3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3 3
3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>33
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3 3
</p>
<p>3
3
</p>
<p>3 3
33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>33
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>33
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3 33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
33
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3 3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3 3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>33
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
3
</p>
<p>3
3
</p>
<p>3
3 3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
3
</p>
<p>3
3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3 3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3 3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>33
</p>
<p>33
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>Linear Discriminant Analysis
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>11
11
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 11
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>11
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1 1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>11
1 1
</p>
<p>1
</p>
<p>1
1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>11
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>11
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>11
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>11
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>11
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>11
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>11
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
11
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>11
1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>11
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>11
</p>
<p>1
</p>
<p>1 1
</p>
<p>11
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>11
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
11
</p>
<p>1
1
</p>
<p>1
1
</p>
<p>1
</p>
<p>11
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>11
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>11
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
1
</p>
<p>11
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>11
</p>
<p>1 1
</p>
<p>1
1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
1
</p>
<p>1 1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>11
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
2
</p>
<p>22
</p>
<p>2
</p>
<p>2 2
</p>
<p>2
</p>
<p>2 2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2 2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2 22
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>22
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
22
</p>
<p>2
</p>
<p>22
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2
</p>
<p>2
</p>
<p>2 2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2 2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2 2
</p>
<p>2
</p>
<p>2 2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>22
2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2 2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2 2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2 2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2 2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2
</p>
<p>2 22
</p>
<p>2
</p>
<p>2
</p>
<p>2 2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2 2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2 2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>22
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>22 2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2 2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2
</p>
<p>2 2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3 3 3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>33
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>33
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>33
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3 3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3 3
3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>33
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3 3
</p>
<p>3
3
</p>
<p>3 3
33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>33
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>33
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3 33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
33
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3 3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3 3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>33
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
3
</p>
<p>3
3
</p>
<p>3
3 3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
3
</p>
<p>3
3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3 3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3 3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>33
</p>
<p>33
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>X1X1
</p>
<p>X
2
</p>
<p>X
2
</p>
<p>FIGURE 4.2. The data come from three classes in IR2 and are easily separated
by linear decision boundaries. The right plot shows the boundaries found by linear
discriminant analysis. The left plot shows the boundaries found by linear regres-
sion of the indicator response variables. The middle class is completely masked
(never dominates).
</p>
<p>&bull; The closest target classification rule (4.6) is easily seen to be exactly
the same as the maximum fitted component criterion (4.4).
</p>
<p>There is a serious problem with the regression approach when the number
of classes K &ge; 3, especially prevalent when K is large. Because of the rigid
nature of the regression model, classes can be masked by others. Figure 4.2
illustrates an extreme situation whenK = 3. The three classes are perfectly
separated by linear decision boundaries, yet linear regression misses the
middle class completely.
In Figure 4.3 we have projected the data onto the line joining the three
</p>
<p>centroids (there is no information in the orthogonal direction in this case),
and we have included and coded the three response variables Y1, Y2 and
Y3. The three regression lines (left panel) are included, and we see that
the line corresponding to the middle class is horizontal and its fitted values
are never dominant! Thus, observations from class 2 are classified either
as class 1 or class 3. The right panel uses quadratic regression rather than
linear regression. For this simple example a quadratic rather than linear
fit (for the middle class at least) would solve the problem. However, it
can be seen that if there were four rather than three classes lined up like
this, a quadratic would not come down fast enough, and a cubic would
be needed as well. A loose but general rule is that if K &ge; 3 classes are
lined up, polynomial terms up to degree K &minus; 1 might be needed to resolve
them. Note also that these are polynomials along the derived direction
passing through the centroids, which can have arbitrary orientation. So in</p>
<p/>
</div>
<div class="page"><p/>
<p>106 4. Linear Methods for Classification
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>11
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
11
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>11
11
</p>
<p>1
</p>
<p>11
</p>
<p>11
1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>11
</p>
<p>1
11
</p>
<p>1
1
</p>
<p>1
</p>
<p>11
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>111
1
</p>
<p>1
1
</p>
<p>11
</p>
<p>1
</p>
<p>1
</p>
<p>11
</p>
<p>11
</p>
<p>1
</p>
<p>111
</p>
<p>1
</p>
<p>11
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>11
</p>
<p>11
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
11
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
11
</p>
<p>1
</p>
<p>222 222 2 2 2222 22 22 2 22 22222 2 22 222 22 2 22 22 22 22222 22 2 222 2
2222 222 222 222 2222 22 22 2 22 222 2222 22 222 222 222 222 222 2 22 22
</p>
<p>222 222 22 22 2222 22 2 222 2 22 2 22 2 22 22 222 2 222 222 222 2
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>33
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>33
33
</p>
<p>3
</p>
<p>33
</p>
<p>33
3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>33
</p>
<p>3
33
</p>
<p>3
3
</p>
<p>3
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>333
3
</p>
<p>3
3
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>33
</p>
<p>33
</p>
<p>3
</p>
<p>33
3
</p>
<p>3
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>33
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
33
</p>
<p>3
3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
33
</p>
<p>3
</p>
<p>0.0
</p>
<p>0.5
</p>
<p>1.0
</p>
<p>0.0 0.2 0.4 0.6 0.8 1.0
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
11
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>11
</p>
<p>1
1
</p>
<p>1
</p>
<p>11
</p>
<p>11
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>11
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>11
</p>
<p>1
11
</p>
<p>1 1
</p>
<p>1
</p>
<p>11
1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
1
</p>
<p>111
1
</p>
<p>1
1
</p>
<p>11
</p>
<p>1
</p>
<p>1
</p>
<p>11
</p>
<p>11
</p>
<p>1
</p>
<p>111
</p>
<p>1
</p>
<p>11
1
</p>
<p>11 11 11
1 1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1 11 111
1
</p>
<p>11 1
1
</p>
<p>11 1 11 1
11 1
1
</p>
<p>1 11
1
</p>
<p>11 1 111 111 111
1
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>22
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2
2
</p>
<p>2
</p>
<p>22
</p>
<p>22
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2 2
22
</p>
<p>2
2
</p>
<p>22
2
</p>
<p>22 222 2222 22
22 2
</p>
<p>2
</p>
<p>2
2
</p>
<p>22 2222 22 222
2
</p>
<p>22 222 222 2
22 2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>333 3
3
</p>
<p>3
3 3 333
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
3
</p>
<p>3
3
</p>
<p>3 333
3
</p>
<p>3
</p>
<p>3 33 33
3
</p>
<p>3
</p>
<p>3
</p>
<p>3 33 3
3
</p>
<p>33 333
33 33 3 33
</p>
<p>3 3
</p>
<p>33
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>33
</p>
<p>3
33
</p>
<p>3 3
</p>
<p>3
</p>
<p>33
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>333
3
</p>
<p>3
3
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>33
</p>
<p>33
</p>
<p>3
</p>
<p>333
</p>
<p>3
</p>
<p>33
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>0.0
</p>
<p>0.5
</p>
<p>1.0
</p>
<p>0.0 0.2 0.4 0.6 0.8 1.0
</p>
<p>Degree = 1; Error = 0.33 Degree = 2; Error = 0.04
</p>
<p>FIGURE 4.3. The effects of masking on linear regression in IR for a three-class
problem. The rug plot at the base indicates the positions and class membership of
each observation. The three curves in each panel are the fitted regressions to the
three-class indicator variables; for example, for the blue class, yblue is 1 for the
blue observations, and 0 for the green and orange. The fits are linear and quadratic
polynomials. Above each plot is the training error rate. The Bayes error rate is
0.025 for this problem, as is the LDA error rate.
</p>
<p>p-dimensional input space, one would need general polynomial terms and
cross-products of total degree K &minus; 1, O(pK&minus;1) terms in all, to resolve such
worst-case scenarios.
The example is extreme, but for large K and small p such maskings
</p>
<p>naturally occur. As a more realistic illustration, Figure 4.4 is a projection
of the training data for a vowel recognition problem onto an informative
two-dimensional subspace. There are K = 11 classes in p = 10 dimensions.
This is a difficult classification problem, and the best methods achieve
around 40% errors on the test data. The main point here is summarized in
Table 4.1; linear regression has an error rate of 67%, while a close relative,
linear discriminant analysis, has an error rate of 56%. It seems that masking
has hurt in this case. While all the other methods in this chapter are based
on linear functions of x as well, they use them in such a way that avoids
this masking problem.
</p>
<p>4.3 Linear Discriminant Analysis
</p>
<p>Decision theory for classification (Section 2.4) tells us that we need to know
the class posteriors Pr(G|X) for optimal classification. Suppose fk(x) is
the class-conditional density of X in class G = k, and let πk be the prior
probability of class k, with
</p>
<p>&sum;K
k=1 πk = 1. A simple application of Bayes</p>
<p/>
</div>
<div class="page"><p/>
<p>4.3 Linear Discriminant Analysis 107
</p>
<p>Coordinate 1 for Training Data
</p>
<p>C
o
</p>
<p>o
rd
</p>
<p>in
a
</p>
<p>te
 2
</p>
<p> f
o
</p>
<p>r 
T
</p>
<p>ra
in
</p>
<p>in
g
</p>
<p> D
a
</p>
<p>ta
</p>
<p>-4 -2 0 2 4
</p>
<p>-6
-4
</p>
<p>-2
0
</p>
<p>2
4
</p>
<p>o
ooooo
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
o
</p>
<p>o o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o oo
</p>
<p>o
oo o oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>oo oo
oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
ooo
</p>
<p>o o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oooo
</p>
<p>o
oo o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oooo
o
</p>
<p>oo
o
oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>o o
</p>
<p>ooo o o
</p>
<p>o
</p>
<p>o
oo ooo
</p>
<p>oo o o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>ooooo
o
</p>
<p>o
ooooo
</p>
<p>o
o
</p>
<p>ooo
o
</p>
<p>oooooo
</p>
<p>oo
ooo o
</p>
<p>ooooo
o
</p>
<p>oooooo
</p>
<p>o
o
</p>
<p>oooo
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oooo
o
</p>
<p>o
oooo
</p>
<p>o
</p>
<p>ooo
ooo
</p>
<p>o
o
o
oo
o
</p>
<p>oo
o
ooo
</p>
<p>oo
ooo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>ooooo o
</p>
<p>o o
o
o
</p>
<p>o o
o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oooo
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>ooo
</p>
<p>o
o ooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo o
o oo
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o o o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
ooo
</p>
<p>o
o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>oo
o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
o
o o
</p>
<p>o oo
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
oo
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
o
ooo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
ooo
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
o
</p>
<p>o oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oooo
o
o
</p>
<p>o
oo o
</p>
<p>o
o
</p>
<p>ooo
oo o
</p>
<p>o
o
o
o o
</p>
<p>o
</p>
<p>ooo
ooo
</p>
<p>o
o
</p>
<p>o
oo o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o o
</p>
<p>oo
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
ooo
</p>
<p>oo
o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
o ooo
</p>
<p>o
o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>oo
o
</p>
<p>oo
o
</p>
<p>ooo
</p>
<p>oo
ooo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>&bull;&bull;
&bull;&bull;
</p>
<p>&bull;&bull;
&bull;&bull; &bull;&bull;
</p>
<p>&bull;&bull; &bull;&bull;
&bull;&bull;
</p>
<p>&bull;&bull;
&bull;&bull;
</p>
<p>&bull;&bull;
</p>
<p>Linear Discriminant Analysis
</p>
<p>FIGURE 4.4. A two-dimensional plot of the vowel training data. There are
eleven classes with X &isin; IR10, and this is the best view in terms of a LDA model
(Section 4.3.3). The heavy circles are the projected mean vectors for each class.
The class overlap is considerable.
</p>
<p>TABLE 4.1. Training and test error rates using a variety of linear techniques
on the vowel data. There are eleven classes in ten dimensions, of which three
account for 90% of the variance (via a principal components analysis). We see
that linear regression is hurt by masking, increasing the test and training error
by over 10%.
</p>
<p>Technique Error Rates
Training Test
</p>
<p>Linear regression 0.48 0.67
Linear discriminant analysis 0.32 0.56
</p>
<p>Quadratic discriminant analysis 0.01 0.53
Logistic regression 0.22 0.51</p>
<p/>
</div>
<div class="page"><p/>
<p>108 4. Linear Methods for Classification
</p>
<p>theorem gives us
</p>
<p>Pr(G = k|X = x) = fk(x)πk&sum;K
ℓ=1 fℓ(x)πℓ
</p>
<p>. (4.7)
</p>
<p>We see that in terms of ability to classify, having the fk(x) is almost equiv-
alent to having the quantity Pr(G = k|X = x).
Many techniques are based on models for the class densities:
</p>
<p>&bull; linear and quadratic discriminant analysis use Gaussian densities;
</p>
<p>&bull; more flexible mixtures of Gaussians allow for nonlinear decision bound-
aries (Section 6.8);
</p>
<p>&bull; general nonparametric density estimates for each class density allow
the most flexibility (Section 6.6.2);
</p>
<p>&bull; Naive Bayes models are a variant of the previous case, and assume
that each of the class densities are products of marginal densities;
that is, they assume that the inputs are conditionally independent in
each class (Section 6.6.3).
</p>
<p>Suppose that we model each class density as multivariate Gaussian
</p>
<p>fk(x) =
1
</p>
<p>(2π)p/2|Σk|1/2
e&minus;
</p>
<p>1
2 (x&minus;μk)
</p>
<p>TΣ
&minus;1
k
</p>
<p>(x&minus;μk). (4.8)
</p>
<p>Linear discriminant analysis (LDA) arises in the special case when we
assume that the classes have a common covariance matrix Σk = Σ &forall;k. In
comparing two classes k and ℓ, it is sufficient to look at the log-ratio, and
we see that
</p>
<p>log
Pr(G = k|X = x)
Pr(G = ℓ|X = x) = log
</p>
<p>fk(x)
</p>
<p>fℓ(x)
+ log
</p>
<p>πk
πℓ
</p>
<p>= log
πk
πℓ
</p>
<p>&minus; 1
2
(μk + μℓ)
</p>
<p>TΣ&minus;1(μk &minus; μℓ)
</p>
<p>+ xTΣ&minus;1(μk &minus; μℓ),
</p>
<p>(4.9)
</p>
<p>an equation linear in x. The equal covariance matrices cause the normal-
ization factors to cancel, as well as the quadratic part in the exponents.
This linear log-odds function implies that the decision boundary between
classes k and ℓ&mdash;the set where Pr(G = k|X = x) = Pr(G = ℓ|X = x)&mdash;is
linear in x; in p dimensions a hyperplane. This is of course true for any pair
of classes, so all the decision boundaries are linear. If we divide IRp into
regions that are classified as class 1, class 2, etc., these regions will be sep-
arated by hyperplanes. Figure 4.5 (left panel) shows an idealized example
with three classes and p = 2. Here the data do arise from three Gaus-
sian distributions with a common covariance matrix. We have included in</p>
<p/>
</div>
<div class="page"><p/>
<p>4.3 Linear Discriminant Analysis 109
</p>
<p>+ +
</p>
<p>+
3
</p>
<p>21
</p>
<p>1
</p>
<p>1
</p>
<p>2
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>1
</p>
<p>2
</p>
<p>3
</p>
<p>3
</p>
<p>2
</p>
<p>1 1 21
</p>
<p>1
</p>
<p>3
</p>
<p>3
</p>
<p>1 2
1
</p>
<p>2
</p>
<p>3
</p>
<p>2
</p>
<p>3
</p>
<p>3
</p>
<p>1
</p>
<p>2
</p>
<p>2
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>3
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>1 3
</p>
<p>2 2
</p>
<p>3
</p>
<p>1
</p>
<p>3
</p>
<p>1
</p>
<p>3
</p>
<p>3 2
</p>
<p>1
</p>
<p>3
</p>
<p>3
</p>
<p>2
</p>
<p>3
</p>
<p>1
</p>
<p>3
</p>
<p>3
</p>
<p>2
</p>
<p>1
</p>
<p>3
3
</p>
<p>2
</p>
<p>2
</p>
<p>3
2
</p>
<p>2
</p>
<p>21 1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>2
</p>
<p>1
</p>
<p>3
</p>
<p>3
</p>
<p>1
1
</p>
<p>3
</p>
<p>3
2
</p>
<p>2
</p>
<p>2
</p>
<p>23
</p>
<p>1
</p>
<p>2
</p>
<p>FIGURE 4.5. The left panel shows three Gaussian distributions, with the same
covariance and different means. Included are the contours of constant density
enclosing 95% of the probability in each case. The Bayes decision boundaries
between each pair of classes are shown (broken straight lines), and the Bayes
decision boundaries separating all three classes are the thicker solid lines (a subset
of the former). On the right we see a sample of 30 drawn from each Gaussian
distribution, and the fitted LDA decision boundaries.
</p>
<p>the figure the contours corresponding to 95% highest probability density,
as well as the class centroids. Notice that the decision boundaries are not
the perpendicular bisectors of the line segments joining the centroids. This
would be the case if the covariance Σ were spherical σ2I, and the class
priors were equal. From (4.9) we see that the linear discriminant functions
</p>
<p>δk(x) = x
TΣ&minus;1μk &minus;
</p>
<p>1
</p>
<p>2
μTkΣ
</p>
<p>&minus;1μk + log πk (4.10)
</p>
<p>are an equivalent description of the decision rule, withG(x) = argmaxkδk(x).
In practice we do not know the parameters of the Gaussian distributions,
</p>
<p>and will need to estimate them using our training data:
</p>
<p>&bull; π̂k = Nk/N , where Nk is the number of class-k observations;
</p>
<p>&bull; μ̂k =
&sum;
</p>
<p>gi=k
xi/Nk;
</p>
<p>&bull; Σ̂ = &sum;Kk=1
&sum;
</p>
<p>gi=k
(xi &minus; μ̂k)(xi &minus; μ̂k)T /(N &minus;K).
</p>
<p>Figure 4.5 (right panel) shows the estimated decision boundaries based on
a sample of size 30 each from three Gaussian distributions. Figure 4.1 on
page 103 is another example, but here the classes are not Gaussian.
With two classes there is a simple correspondence between linear dis-
</p>
<p>criminant analysis and classification by linear regression, as in (4.5). The
LDA rule classifies to class 2 if
</p>
<p>xT Σ̂
&minus;1
</p>
<p>(μ̂2 &minus; μ̂1) &gt;
1
</p>
<p>2
(μ̂2 + μ̂1)
</p>
<p>T Σ̂
&minus;1
</p>
<p>(μ̂2 &minus; μ̂1)&minus; log(N2/N1), (4.11)</p>
<p/>
</div>
<div class="page"><p/>
<p>110 4. Linear Methods for Classification
</p>
<p>and class 1 otherwise. Suppose we code the targets in the two classes as +1
and &minus;1, respectively. It is easy to show that the coefficient vector from least
squares is proportional to the LDA direction given in (4.11) (Exercise 4.2).
[In fact, this correspondence occurs for any (distinct) coding of the targets;
see Exercise 4.2]. However unless N1 = N2 the intercepts are different and
hence the resulting decision rules are different.
Since this derivation of the LDA direction via least squares does not use a
</p>
<p>Gaussian assumption for the features, its applicability extends beyond the
realm of Gaussian data. However the derivation of the particular intercept
or cut-point given in (4.11) does require Gaussian data. Thus it makes
sense to instead choose the cut-point that empirically minimizes training
error for a given dataset. This is something we have found to work well in
practice, but have not seen it mentioned in the literature.
With more than two classes, LDA is not the same as linear regression of
</p>
<p>the class indicator matrix, and it avoids the masking problems associated
with that approach (Hastie et al., 1994). A correspondence between regres-
sion and LDA can be established through the notion of optimal scoring,
discussed in Section 12.5.
Getting back to the general discriminant problem (4.8), if the Σk are
</p>
<p>not assumed to be equal, then the convenient cancellations in (4.9) do not
occur; in particular the pieces quadratic in x remain. We then get quadratic
discriminant functions (QDA),
</p>
<p>δk(x) = &minus;
1
</p>
<p>2
log |Σk| &minus;
</p>
<p>1
</p>
<p>2
(x&minus; μk)TΣ&minus;1k (x&minus; μk) + log πk. (4.12)
</p>
<p>The decision boundary between each pair of classes k and ℓ is described by
a quadratic equation {x : δk(x) = δℓ(x)}.
</p>
<p>Figure 4.6 shows an example (from Figure 4.1 on page 103) where the
three classes are Gaussian mixtures (Section 6.8) and the decision bound-
aries are approximated by quadratic equations in x. Here we illustrate
two popular ways of fitting these quadratic boundaries. The right plot
uses QDA as described here, while the left plot uses LDA in the enlarged
five-dimensional quadratic polynomial space. The differences are generally
small; QDA is the preferred approach, with the LDA method a convenient
substitute 2.
The estimates for QDA are similar to those for LDA, except that separate
</p>
<p>covariance matrices must be estimated for each class. When p is large this
can mean a dramatic increase in parameters. Since the decision boundaries
are functions of the parameters of the densities, counting the number of
parameters must be done with care. For LDA, it seems there are (K &minus;
1) &times; (p + 1) parameters, since we only need the differences δk(x) &minus; δK(x)
</p>
<p>2For this figure and many similar figures in the book we compute the decision bound-
aries by an exhaustive contouring method. We compute the decision rule on a fine lattice
of points, and then use contouring algorithms to compute the boundaries.</p>
<p/>
</div>
<div class="page"><p/>
<p>4.3 Linear Discriminant Analysis 111
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>11
</p>
<p>1
1
</p>
<p>1
1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>11
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
11
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
11
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
2
</p>
<p>2
2
</p>
<p>22
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2 2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2 2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>22
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2 2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
22
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2 2
</p>
<p>2
2 2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>22
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2 2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>22
</p>
<p>2 2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3 3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3 3
</p>
<p>3 3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3 3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3 3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3 3
</p>
<p>3
</p>
<p>3
</p>
<p>3 3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3 3
</p>
<p>3
3
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>11
</p>
<p>1
1
</p>
<p>1
1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>11
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
11
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
11
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
2
</p>
<p>2
2
</p>
<p>22
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2 2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2 2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>22
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2 2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
22
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2 2
</p>
<p>2
2 2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>22
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2 2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>22
</p>
<p>2 2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3 3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3 3
</p>
<p>3 3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3 3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3 3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3 3
</p>
<p>3
</p>
<p>3
</p>
<p>3 3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3 3
</p>
<p>3
3
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>FIGURE 4.6. Two methods for fitting quadratic boundaries. The left plot shows
the quadratic decision boundaries for the data in Figure 4.1 (obtained using LDA
in the five-dimensional space X1, X2, X1X2, X
</p>
<p>2
1 , X
</p>
<p>2
2 ). The right plot shows the
</p>
<p>quadratic decision boundaries found by QDA. The differences are small, as is
usually the case.
</p>
<p>between the discriminant functions where K is some pre-chosen class (here
we have chosen the last), and each difference requires p + 1 parameters3.
Likewise for QDA there will be (K &minus; 1) &times; {p(p + 3)/2 + 1} parameters.
Both LDA and QDA perform well on an amazingly large and diverse set
of classification tasks. For example, in the STATLOG project (Michie et
al., 1994) LDA was among the top three classifiers for 7 of the 22 datasets,
QDA among the top three for four datasets, and one of the pair were in the
top three for 10 datasets. Both techniques are widely used, and entire books
are devoted to LDA. It seems that whatever exotic tools are the rage of the
day, we should always have available these two simple tools. The question
arises why LDA and QDA have such a good track record. The reason is not
likely to be that the data are approximately Gaussian, and in addition for
LDA that the covariances are approximately equal. More likely a reason is
that the data can only support simple decision boundaries such as linear or
quadratic, and the estimates provided via the Gaussian models are stable.
This is a bias variance tradeoff&mdash;we can put up with the bias of a linear
decision boundary because it can be estimated with much lower variance
than more exotic alternatives. This argument is less believable for QDA,
since it can have many parameters itself, although perhaps fewer than the
non-parametric alternatives.
</p>
<p>3Although we fit the covariance matrix Σ̂ to compute the LDA discriminant functions,
a much reduced function of it is all that is required to estimate the O(p) parameters
</p>
<p>needed to compute the decision boundaries.</p>
<p/>
</div>
<div class="page"><p/>
<p>112 4. Linear Methods for Classification
</p>
<p>M
is
</p>
<p>c
la
</p>
<p>s
s
if
ic
</p>
<p>a
ti
o
</p>
<p>n
 R
</p>
<p>a
te
</p>
<p>0.0 0.2 0.4 0.6 0.8 1.0
</p>
<p>0
.0
</p>
<p>0
.1
</p>
<p>0
.2
</p>
<p>0
.3
</p>
<p>0
.4
</p>
<p>0
.5
</p>
<p>&bull;
&bull; &bull; &bull; &bull; &bull; &bull;
</p>
<p>&bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull;
</p>
<p>&bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull;
&bull;
</p>
<p>&bull;
</p>
<p>Regularized Discriminant Analysis on the Vowel Data
</p>
<p>Test Data
</p>
<p>Train Data
</p>
<p>α
</p>
<p>FIGURE 4.7. Test and training errors for the vowel data, using regularized
discriminant analysis with a series of values of α &isin; [0, 1]. The optimum for the
test data occurs around α = 0.9, close to quadratic discriminant analysis.
</p>
<p>4.3.1 Regularized Discriminant Analysis
</p>
<p>Friedman (1989) proposed a compromise between LDA and QDA, which
allows one to shrink the separate covariances of QDA toward a common
covariance as in LDA. These methods are very similar in flavor to ridge
regression. The regularized covariance matrices have the form
</p>
<p>Σ̂k(α) = αΣ̂k + (1&minus; α)Σ̂, (4.13)
</p>
<p>where Σ̂ is the pooled covariance matrix as used in LDA. Here α &isin; [0, 1]
allows a continuum of models between LDA and QDA, and needs to be
specified. In practice α can be chosen based on the performance of the
model on validation data, or by cross-validation.
Figure 4.7 shows the results of RDA applied to the vowel data. Both
</p>
<p>the training and test error improve with increasing α, although the test
error increases sharply after α = 0.9. The large discrepancy between the
training and test error is partly due to the fact that there are many repeat
measurements on a small number of individuals, different in the training
and test set.
Similar modifications allow Σ̂ itself to be shrunk toward the scalar
</p>
<p>covariance,
</p>
<p>Σ̂(γ) = γΣ̂+ (1&minus; γ)σ̂2I (4.14)
</p>
<p>for γ &isin; [0, 1]. Replacing Σ̂ in (4.13) by Σ̂(γ) leads to a more general family
of covariances Σ̂(α, γ) indexed by a pair of parameters.
</p>
<p>In Chapter 12, we discuss other regularized versions of LDA, which are
more suitable when the data arise from digitized analog signals and images.</p>
<p/>
</div>
<div class="page"><p/>
<p>4.3 Linear Discriminant Analysis 113
</p>
<p>In these situations the features are high-dimensional and correlated, and the
LDA coefficients can be regularized to be smooth or sparse in the original
domain of the signal. This leads to better generalization and allows for
easier interpretation of the coefficients. In Chapter 18 we also deal with
very high-dimensional problems, where for example the features are gene-
expression measurements in microarray studies. There the methods focus
on the case γ = 0 in (4.14), and other severely regularized versions of LDA.
</p>
<p>4.3.2 Computations for LDA
</p>
<p>As a lead-in to the next topic, we briefly digress on the computations
required for LDA and especially QDA. Their computations are simplified
by diagonalizing Σ̂ or Σ̂k. For the latter, suppose we compute the eigen-
decomposition for each Σ̂k = UkDkU
</p>
<p>T
k , where Uk is p &times; p orthonormal,
</p>
<p>and Dk a diagonal matrix of positive eigenvalues dkℓ. Then the ingredients
for δk(x) (4.12) are
</p>
<p>&bull; (x&minus; μ̂k)T Σ̂
&minus;1
</p>
<p>k (x&minus; μ̂k) = [UTk (x&minus; μ̂k)]TD&minus;1k [UTk (x&minus; μ̂k)];
</p>
<p>&bull; log |Σ̂k| =
&sum;
</p>
<p>ℓ log dkℓ.
</p>
<p>In light of the computational steps outlined above, the LDA classifier
can be implemented by the following pair of steps:
</p>
<p>&bull; Sphere the data with respect to the common covariance estimate Σ̂:
X&lowast; &larr; D&minus; 12UTX, where Σ̂ = UDUT . The common covariance esti-
mate of X&lowast; will now be the identity.
</p>
<p>&bull; Classify to the closest class centroid in the transformed space, modulo
the effect of the class prior probabilities πk.
</p>
<p>4.3.3 Reduced-Rank Linear Discriminant Analysis
</p>
<p>So far we have discussed LDA as a restricted Gaussian classifier. Part of
its popularity is due to an additional restriction that allows us to view
informative low-dimensional projections of the data.
The K centroids in p-dimensional input space lie in an affine subspace
</p>
<p>of dimension &le; K &minus; 1, and if p is much larger than K, this will be a con-
siderable drop in dimension. Moreover, in locating the closest centroid, we
can ignore distances orthogonal to this subspace, since they will contribute
equally to each class. Thus we might just as well project the X&lowast; onto this
centroid-spanning subspace HK&minus;1, and make distance comparisons there.
Thus there is a fundamental dimension reduction in LDA, namely, that we
need only consider the data in a subspace of dimension at most K &minus; 1.</p>
<p/>
</div>
<div class="page"><p/>
<p>114 4. Linear Methods for Classification
</p>
<p>If K = 3, for instance, this could allow us to view the data in a two-
dimensional plot, color-coding the classes. In doing so we would not have
relinquished any of the information needed for LDA classification.
What if K &gt; 3? We might then ask for a L &lt; K&minus;1 dimensional subspace
</p>
<p>HL &sube; HK&minus;1 optimal for LDA in some sense. Fisher defined optimal to
mean that the projected centroids were spread out as much as possible in
terms of variance. This amounts to finding principal component subspaces
of the centroids themselves (principal components are described briefly in
Section 3.5.1, and in more detail in Section 14.5.1). Figure 4.4 shows such an
optimal two-dimensional subspace for the vowel data. Here there are eleven
classes, each a different vowel sound, in a ten-dimensional input space. The
centroids require the full space in this case, since K &minus; 1 = p, but we have
shown an optimal two-dimensional subspace. The dimensions are ordered,
so we can compute additional dimensions in sequence. Figure 4.8 shows four
additional pairs of coordinates, also known as canonical or discriminant
variables. In summary then, finding the sequences of optimal subspaces
for LDA involves the following steps:
</p>
<p>&bull; compute the K &times; p matrix of class centroids M and the common
covariance matrix W (for within-class covariance);
</p>
<p>&bull; compute M&lowast; = MW&minus; 12 using the eigen-decomposition of W;
&bull; compute B&lowast;, the covariance matrix ofM&lowast; (B for between-class covari-
</p>
<p>ance), and its eigen-decomposition B&lowast; = V&lowast;DBV
&lowast;T . The columns
</p>
<p>v&lowast;ℓ of V
&lowast; in sequence from first to last define the coordinates of the
</p>
<p>optimal subspaces.
</p>
<p>Combining all these operations the ℓth discriminant variable is given by
Zℓ = v
</p>
<p>T
ℓ X with vℓ = W
</p>
<p>&minus; 12 v&lowast;ℓ .
Fisher arrived at this decomposition via a different route, without refer-
</p>
<p>ring to Gaussian distributions at all. He posed the problem:
</p>
<p>Find the linear combination Z = aTX such that the between-
class variance is maximized relative to the within-class variance.
</p>
<p>Again, the between class variance is the variance of the class means of
Z, and the within class variance is the pooled variance about the means.
Figure 4.9 shows why this criterion makes sense. Although the direction
joining the centroids separates the means as much as possible (i.e., max-
imizes the between-class variance), there is considerable overlap between
the projected classes due to the nature of the covariances. By taking the
covariance into account as well, a direction with minimum overlap can be
found.
The between-class variance of Z is aTBa and the within-class variance
</p>
<p>aTWa, where W is defined earlier, and B is the covariance matrix of the
class centroid matrix M. Note that B + W = T, where T is the total
covariance matrix of X, ignoring class information.</p>
<p/>
</div>
<div class="page"><p/>
<p>4.3 Linear Discriminant Analysis 115
</p>
<p>Coordinate 1 
</p>
<p>C
o
</p>
<p>o
rd
</p>
<p>in
a
</p>
<p>te
 3
</p>
<p> 
</p>
<p>-4 -2 0 2 4
</p>
<p>-2
0
</p>
<p>2
</p>
<p>o
o
oooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
oo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o o o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
o
o
</p>
<p>o
ooo
o o
</p>
<p>oo
oo
</p>
<p>o
o
</p>
<p>o
o
oo
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
o
o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
o
oooo
</p>
<p>oo
</p>
<p>o
o
o
</p>
<p>o
o
</p>
<p>o
o
o oo
</p>
<p>o
</p>
<p>ooo
oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o ooo
</p>
<p>oo
o
o
</p>
<p>o
</p>
<p>oo o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
o
o
oo
</p>
<p>oo
</p>
<p>o
o
oo
</p>
<p>o
o
</p>
<p>oooo
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
o
</p>
<p>o
o
</p>
<p>o o
o
o
o
o o
</p>
<p>o
oo
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
oo
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>ooo
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>ooo
o o
</p>
<p>oo
o
</p>
<p>oo
</p>
<p>oo
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>o
</p>
<p>o
</p>
<p>oo oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>ooo
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o oooo
o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o o o
</p>
<p>o
</p>
<p>oooo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
o
</p>
<p>o
oo
oo
</p>
<p>o
</p>
<p>o o o
o
</p>
<p>o ooo
o o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo oo
</p>
<p>ooo
oo
</p>
<p>o o
oo
</p>
<p>ooo
</p>
<p>o
o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>oo
</p>
<p>o
</p>
<p>&bull;&bull;
&bull;&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;&bull;&bull;
&bull; &bull;&bull; &bull;
</p>
<p>&bull;&bull;&bull; &bull;&bull;&bull;&bull;
</p>
<p>Coordinate 2 
</p>
<p>C
o
</p>
<p>o
rd
</p>
<p>in
a
</p>
<p>te
 3
</p>
<p> 
</p>
<p>-6 -4 -2 0 2 4
</p>
<p>-2
0
</p>
<p>2
</p>
<p>o
o
oooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
o
</p>
<p>o
ooo
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
o
o
</p>
<p>o
ooo
</p>
<p>oo
</p>
<p>oo
oo
</p>
<p>o
o
</p>
<p>o
o
oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
o
o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>oooo
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
ooo
o
</p>
<p>ooo
oo
</p>
<p>oo
</p>
<p>o
</p>
<p>oooo
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>ooo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
o
o
oo
</p>
<p>oo
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>oooo
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
oo
o
oo
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
o
</p>
<p>o
o
o
</p>
<p>oo
oo
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
oo
o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>ooo
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
oo
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>ooo
o o
</p>
<p>oo
o
</p>
<p>oo
</p>
<p>oo
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>o
</p>
<p>o
</p>
<p>ooo o
</p>
<p>o
o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>ooo
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>ooooo
o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
oo o
</p>
<p>o
</p>
<p>oooo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>oo
o
</p>
<p>ooo
o
</p>
<p>oo oo
oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oooo
</p>
<p>ooo
oo
</p>
<p>oo
oo
</p>
<p>ooo
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>oo
</p>
<p>o
</p>
<p>&bull;&bull;
&bull;&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;&bull;&bull;
&bull;&bull;&bull;&bull;
</p>
<p>&bull;&bull;&bull;&bull;&bull; &bull;&bull;
</p>
<p>Coordinate 1 
</p>
<p>C
o
</p>
<p>o
rd
</p>
<p>in
a
</p>
<p>te
 7
</p>
<p> 
</p>
<p>-4 -2 0 2 4
</p>
<p>-3
-2
</p>
<p>-1
0
</p>
<p>1
2
</p>
<p>3
</p>
<p>oooooo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o oo
</p>
<p>oooooo
</p>
<p>o o oo
oo
</p>
<p>o
oo
</p>
<p>o
oo
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>ooo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
o
</p>
<p>o
</p>
<p>ooooo
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o ooo
o
</p>
<p>oo
o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
o
o
</p>
<p>oo
</p>
<p>o
o
o
</p>
<p>ooo
oo
</p>
<p>o o
</p>
<p>oo
</p>
<p>o
oo
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>oooo
o
o
</p>
<p>oooo
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>ooo
</p>
<p>oo
oo
</p>
<p>o
</p>
<p>o
o
o
o
</p>
<p>ooo
</p>
<p>oo
ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>o
o
</p>
<p>o oo
o
</p>
<p>oo
</p>
<p>oo
oooo
</p>
<p>o
o
</p>
<p>o
o
oo
</p>
<p>oo
o
</p>
<p>oo
</p>
<p>o
ooo
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>oooo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o o o oo
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o o
</p>
<p>o
</p>
<p>o
o
ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o o
</p>
<p>o
o
o
</p>
<p>o
oo
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>oooo
ooo
</p>
<p>oo oo
o
</p>
<p>oo
o
</p>
<p>o o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>ooo
o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o oo
o
</p>
<p>o
oo
</p>
<p>oo
</p>
<p>o o
</p>
<p>oo
o o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o o
</p>
<p>o
o
o
</p>
<p>ooo
o
</p>
<p>oo
</p>
<p>oooo
oo
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
o
o o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>ooo
o
</p>
<p>o
o
o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>ooo
ooo
</p>
<p>o
oo
</p>
<p>oo
</p>
<p>o
</p>
<p>ooo o
oo
</p>
<p>o o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oooo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>&bull;&bull;&bull;&bull;&bull;
&bull; &bull;&bull; &bull;&bull;&bull;&bull; &bull;&bull; &bull;
</p>
<p>&bull;&bull;&bull; &bull;&bull;&bull;&bull;
</p>
<p>Coordinate 9 
</p>
<p>C
o
</p>
<p>o
rd
</p>
<p>in
a
</p>
<p>te
 1
</p>
<p>0
 
</p>
<p>-2 -1 0 1 2 3
</p>
<p>-2
-1
</p>
<p>0
1
</p>
<p>2
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>oo o
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>oo
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>oooo
</p>
<p>o
o
oooooo
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
o
</p>
<p>o
o
o
</p>
<p>ooo
</p>
<p>oo
o
o ooo
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
o
o o oooo
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
ooo
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
o
</p>
<p>ooooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
oooo
</p>
<p>o
</p>
<p>oooo
</p>
<p>oo
</p>
<p>oo
oo
</p>
<p>o o ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>oo
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
oo
</p>
<p>o o
</p>
<p>oooo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o o
oo
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>ooo oo o
</p>
<p>o o
o
ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oooo
o o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
o o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o o
</p>
<p>oo
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>oooo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>ooo
o
oo
</p>
<p>o
o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
oo
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
</p>
<p>Linear Discriminant Analysis
</p>
<p>FIGURE 4.8. Four projections onto pairs of canonical variates. Notice that as
the rank of the canonical variates increases, the centroids become less spread out.
In the lower right panel they appear to be superimposed, and the classes most
confused.</p>
<p/>
</div>
<div class="page"><p/>
<p>116 4. Linear Methods for Classification
</p>
<p>+
</p>
<p>+
+
</p>
<p>+
</p>
<p>FIGURE 4.9. Although the line joining the centroids defines the direction of
greatest centroid spread, the projected data overlap because of the covariance
(left panel). The discriminant direction minimizes this overlap for Gaussian data
(right panel).
</p>
<p>Fisher&rsquo;s problem therefore amounts to maximizing the Rayleigh quotient,
</p>
<p>max
a
</p>
<p>aTBa
</p>
<p>aTWa
, (4.15)
</p>
<p>or equivalently
max
</p>
<p>a
aTBa subject to aTWa = 1. (4.16)
</p>
<p>This is a generalized eigenvalue problem, with a given by the largest
eigenvalue of W&minus;1B. It is not hard to show (Exercise 4.1) that the optimal
a1 is identical to v1 defined above. Similarly one can find the next direction
a2, orthogonal in W to a1, such that a
</p>
<p>T
2 Ba2/a
</p>
<p>T
2 Wa2 is maximized; the
</p>
<p>solution is a2 = v2, and so on. The aℓ are referred to as discriminant
coordinates, not to be confused with discriminant functions. They are also
referred to as canonical variates, since an alternative derivation of these
results is through a canonical correlation analysis of the indicator response
matrix Y on the predictor matrix X. This line is pursued in Section 12.5.
</p>
<p>To summarize the developments so far:
</p>
<p>&bull; Gaussian classification with common covariances leads to linear deci-
sion boundaries. Classification can be achieved by sphering the data
with respect to W, and classifying to the closest centroid (modulo
log πk) in the sphered space.
</p>
<p>&bull; Since only the relative distances to the centroids count, one can con-
fine the data to the subspace spanned by the centroids in the sphered
space.
</p>
<p>&bull; This subspace can be further decomposed into successively optimal
subspaces in term of centroid separation. This decomposition is iden-
tical to the decomposition due to Fisher.</p>
<p/>
</div>
<div class="page"><p/>
<p>4.3 Linear Discriminant Analysis 117
</p>
<p>Dimension
</p>
<p>M
is
</p>
<p>c
la
</p>
<p>s
s
if
ic
</p>
<p>a
ti
o
</p>
<p>n
 R
</p>
<p>a
te
</p>
<p>2 4 6 8 10
</p>
<p>0
.3
</p>
<p>0
.4
</p>
<p>0
.5
</p>
<p>0
.6
</p>
<p>0
.7
</p>
<p>LDA and Dimension Reduction on the Vowel Data
</p>
<p>&bull;
</p>
<p>&bull; &bull;
&bull; &bull;
</p>
<p>&bull; &bull; &bull; &bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
&bull;
</p>
<p>&bull;
&bull; &bull; &bull; &bull;
</p>
<p>Test Data
</p>
<p>Train Data
</p>
<p>FIGURE 4.10. Training and test error rates for the vowel data, as a function
of the dimension of the discriminant subspace. In this case the best error rate is
for dimension 2. Figure 4.11 shows the decision boundaries in this space.
</p>
<p>The reduced subspaces have been motivated as a data reduction (for
viewing) tool. Can they also be used for classification, and what is the
rationale? Clearly they can, as in our original derivation; we simply limit
the distance-to-centroid calculations to the chosen subspace. One can show
that this is a Gaussian classification rule with the additional restriction
that the centroids of the Gaussians lie in a L-dimensional subspace of IRp.
Fitting such a model by maximum likelihood, and then constructing the
posterior probabilities using Bayes&rsquo; theorem amounts to the classification
rule described above (Exercise 4.8).
Gaussian classification dictates the log πk correction factor in the dis-
</p>
<p>tance calculation. The reason for this correction can be seen in Figure 4.9.
The misclassification rate is based on the area of overlap between the two
densities. If the πk are equal (implicit in that figure), then the optimal
cut-point is midway between the projected means. If the πk are not equal,
moving the cut-point toward the smaller class will improve the error rate.
As mentioned earlier for two classes, one can derive the linear rule using
LDA (or any other method), and then choose the cut-point to minimize
misclassification error over the training data.
As an example of the benefit of the reduced-rank restriction, we return
</p>
<p>to the vowel data. There are 11 classes and 10 variables, and hence 10
possible dimensions for the classifier. We can compute the training and
test error in each of these hierarchical subspaces; Figure 4.10 shows the
results. Figure 4.11 shows the decision boundaries for the classifier based
on the two-dimensional LDA solution.
There is a close connection between Fisher&rsquo;s reduced rank discriminant
</p>
<p>analysis and regression of an indicator response matrix. It turns out that</p>
<p/>
</div>
<div class="page"><p/>
<p>118 4. Linear Methods for Classification
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>Canonical Coordinate 1
</p>
<p>C
a
n
o
n
ic
</p>
<p>a
l 
C
</p>
<p>o
o
rd
</p>
<p>in
a
te
</p>
<p> 2
</p>
<p>Classification in Reduced Subspace
</p>
<p>&bull;&bull;
&bull;&bull;
</p>
<p>&bull;&bull;
&bull;&bull; &bull;&bull;
</p>
<p>&bull;&bull; &bull;&bull;
&bull;&bull;
</p>
<p>&bull;&bull;
&bull;&bull;
</p>
<p>&bull;&bull;
</p>
<p>FIGURE 4.11. Decision boundaries for the vowel training data, in the two-di-
mensional subspace spanned by the first two canonical variates. Note that in
any higher-dimensional subspace, the decision boundaries are higher-dimensional
affine planes, and could not be represented as lines.</p>
<p/>
</div>
<div class="page"><p/>
<p>4.4 Logistic Regression 119
</p>
<p>LDA amounts to the regression followed by an eigen-decomposition of
ŶTY. In the case of two classes, there is a single discriminant variable
that is identical up to a scalar multiplication to either of the columns of Ŷ.
These connections are developed in Chapter 12. A related fact is that if one
transforms the original predictors X to Ŷ, then LDA using Ŷ is identical
to LDA in the original space (Exercise 4.3).
</p>
<p>4.4 Logistic Regression
</p>
<p>The logistic regression model arises from the desire to model the posterior
probabilities of the K classes via linear functions in x, while at the same
time ensuring that they sum to one and remain in [0, 1]. The model has
the form
</p>
<p>log
Pr(G = 1|X = x)
Pr(G = K|X = x) = β10 + β
</p>
<p>T
1 x
</p>
<p>log
Pr(G = 2|X = x)
Pr(G = K|X = x) = β20 + β
</p>
<p>T
2 x
</p>
<p>...
</p>
<p>log
Pr(G = K &minus; 1|X = x)
Pr(G = K|X = x) = β(K&minus;1)0 + β
</p>
<p>T
K&minus;1x.
</p>
<p>(4.17)
</p>
<p>The model is specified in terms of K &minus; 1 log-odds or logit transformations
(reflecting the constraint that the probabilities sum to one). Although the
model uses the last class as the denominator in the odds-ratios, the choice
of denominator is arbitrary in that the estimates are equivariant under this
choice. A simple calculation shows that
</p>
<p>Pr(G = k|X = x) = exp(βk0 + β
T
k x)
</p>
<p>1 +
&sum;K&minus;1
</p>
<p>ℓ=1 exp(βℓ0 + β
T
ℓ x)
</p>
<p>, k = 1, . . . ,K &minus; 1,
</p>
<p>Pr(G = K|X = x) = 1
1 +
</p>
<p>&sum;K&minus;1
ℓ=1 exp(βℓ0 + β
</p>
<p>T
ℓ x)
</p>
<p>, (4.18)
</p>
<p>and they clearly sum to one. To emphasize the dependence on the entire pa-
rameter set θ = {β10, βT1 , . . . , β(K&minus;1)0, βTK&minus;1}, we denote the probabilities
Pr(G = k|X = x) = pk(x; θ).
When K = 2, this model is especially simple, since there is only a single
</p>
<p>linear function. It is widely used in biostatistical applications where binary
responses (two classes) occur quite frequently. For example, patients survive
or die, have heart disease or not, or a condition is present or absent.</p>
<p/>
</div>
<div class="page"><p/>
<p>120 4. Linear Methods for Classification
</p>
<p>4.4.1 Fitting Logistic Regression Models
</p>
<p>Logistic regression models are usually fit by maximum likelihood, using the
conditional likelihood of G given X. Since Pr(G|X) completely specifies the
conditional distribution, the multinomial distribution is appropriate. The
log-likelihood for N observations is
</p>
<p>ℓ(θ) =
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>log pgi(xi; θ), (4.19)
</p>
<p>where pk(xi; θ) = Pr(G = k|X = xi; θ).
We discuss in detail the two-class case, since the algorithms simplify
</p>
<p>considerably. It is convenient to code the two-class gi via a 0/1 response yi,
where yi = 1 when gi = 1, and yi = 0 when gi = 2. Let p1(x; θ) = p(x; θ),
and p2(x; θ) = 1&minus; p(x; θ). The log-likelihood can be written
</p>
<p>ℓ(β) =
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>{
yi log p(xi;β) + (1&minus; yi) log(1&minus; p(xi;β))
</p>
<p>}
</p>
<p>=
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>{
yiβ
</p>
<p>Txi &minus; log(1 + eβ
T xi)
</p>
<p>}
. (4.20)
</p>
<p>Here β = {β10, β1}, and we assume that the vector of inputs xi includes
the constant term 1 to accommodate the intercept.
To maximize the log-likelihood, we set its derivatives to zero. These score
</p>
<p>equations are
</p>
<p>&part;ℓ(β)
</p>
<p>&part;β
=
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>xi(yi &minus; p(xi;β)) = 0, (4.21)
</p>
<p>which are p+1 equations nonlinear in β. Notice that since the first compo-
nent of xi is 1, the first score equation specifies that
</p>
<p>&sum;N
i=1 yi =
</p>
<p>&sum;N
i=1 p(xi;β);
</p>
<p>the expected number of class ones matches the observed number (and hence
also class twos.)
To solve the score equations (4.21), we use the Newton&ndash;Raphson algo-
</p>
<p>rithm, which requires the second-derivative or Hessian matrix
</p>
<p>&part;2ℓ(β)
</p>
<p>&part;β&part;βT
= &minus;
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>xixi
T p(xi;β)(1&minus; p(xi;β)). (4.22)
</p>
<p>Starting with βold, a single Newton update is
</p>
<p>βnew = βold &minus;
(
&part;2ℓ(β)
</p>
<p>&part;β&part;βT
</p>
<p>)&minus;1
&part;ℓ(β)
</p>
<p>&part;β
, (4.23)
</p>
<p>where the derivatives are evaluated at βold.</p>
<p/>
</div>
<div class="page"><p/>
<p>4.4 Logistic Regression 121
</p>
<p>It is convenient to write the score and Hessian in matrix notation. Let
y denote the vector of yi values, X the N &times; (p + 1) matrix of xi values,
p the vector of fitted probabilities with ith element p(xi;β
</p>
<p>old) and W a
N&times;N diagonal matrix of weights with ith diagonal element p(xi;βold)(1&minus;
p(xi;β
</p>
<p>old)). Then we have
</p>
<p>&part;ℓ(β)
</p>
<p>&part;β
= XT (y &minus; p) (4.24)
</p>
<p>&part;2ℓ(β)
</p>
<p>&part;β&part;βT
= &minus;XTWX (4.25)
</p>
<p>The Newton step is thus
</p>
<p>βnew = βold + (XTWX)&minus;1XT (y &minus; p)
= (XTWX)&minus;1XTW
</p>
<p>(
Xβold +W&minus;1(y &minus; p)
</p>
<p>)
</p>
<p>= (XTWX)&minus;1XTWz. (4.26)
</p>
<p>In the second and third line we have re-expressed the Newton step as a
weighted least squares step, with the response
</p>
<p>z = Xβold +W&minus;1(y &minus; p), (4.27)
</p>
<p>sometimes known as the adjusted response. These equations get solved re-
peatedly, since at each iteration p changes, and hence so does W and z.
This algorithm is referred to as iteratively reweighted least squares or IRLS,
since each iteration solves the weighted least squares problem:
</p>
<p>βnew &larr; argmin
β
</p>
<p>(z&minus;Xβ)TW(z&minus;Xβ). (4.28)
</p>
<p>It seems that β = 0 is a good starting value for the iterative procedure,
although convergence is never guaranteed. Typically the algorithm does
converge, since the log-likelihood is concave, but overshooting can occur.
In the rare cases that the log-likelihood decreases, step size halving will
guarantee convergence.
For the multiclass case (K &ge; 3) the Newton algorithm can also be ex-
</p>
<p>pressed as an iteratively reweighted least squares algorithm, but with a
vector of K&minus;1 responses and a nondiagonal weight matrix per observation.
The latter precludes any simplified algorithms, and in this case it is numer-
ically more convenient to work with the expanded vector θ directly (Ex-
ercise 4.4). Alternatively coordinate-descent methods (Section 3.8.6) can
be used to maximize the log-likelihood efficiently. The R package glmnet
(Friedman et al., 2010) can fit very large logistic regression problems ef-
ficiently, both in N and p. Although designed to fit regularized models,
options allow for unregularized fits.
Logistic regression models are used mostly as a data analysis and infer-
</p>
<p>ence tool, where the goal is to understand the role of the input variables</p>
<p/>
</div>
<div class="page"><p/>
<p>122 4. Linear Methods for Classification
</p>
<p>TABLE 4.2. Results from a logistic regression fit to the South African heart
disease data.
</p>
<p>Coefficient Std. Error Z Score
</p>
<p>(Intercept) &minus;4.130 0.964 &minus;4.285
sbp 0.006 0.006 1.023
</p>
<p>tobacco 0.080 0.026 3.034
ldl 0.185 0.057 3.219
</p>
<p>famhist 0.939 0.225 4.178
obesity -0.035 0.029 &minus;1.187
alcohol 0.001 0.004 0.136
</p>
<p>age 0.043 0.010 4.184
</p>
<p>in explaining the outcome. Typically many models are fit in a search for a
parsimonious model involving a subset of the variables, possibly with some
interactions terms. The following example illustrates some of the issues
involved.
</p>
<p>4.4.2 Example: South African Heart Disease
</p>
<p>Here we present an analysis of binary data to illustrate the traditional
statistical use of the logistic regression model. The data in Figure 4.12 are a
subset of the Coronary Risk-Factor Study (CORIS) baseline survey, carried
out in three rural areas of the Western Cape, South Africa (Rousseauw et
al., 1983). The aim of the study was to establish the intensity of ischemic
heart disease risk factors in that high-incidence region. The data represent
white males between 15 and 64, and the response variable is the presence or
absence of myocardial infarction (MI) at the time of the survey (the overall
prevalence of MI was 5.1% in this region). There are 160 cases in our data
set, and a sample of 302 controls. These data are described in more detail
in Hastie and Tibshirani (1987).
We fit a logistic-regression model by maximum likelihood, giving the
</p>
<p>results shown in Table 4.2. This summary includes Z scores for each of the
coefficients in the model (coefficients divided by their standard errors); a
nonsignificant Z score suggests a coefficient can be dropped from the model.
Each of these correspond formally to a test of the null hypothesis that the
coefficient in question is zero, while all the others are not (also known as
the Wald test). A Z score greater than approximately 2 in absolute value
is significant at the 5% level.
There are some surprises in this table of coefficients, which must be in-
</p>
<p>terpreted with caution. Systolic blood pressure (sbp) is not significant! Nor
is obesity, and its sign is negative. This confusion is a result of the corre-
lation between the set of predictors. On their own, both sbp and obesity
are significant, and with positive sign. However, in the presence of many</p>
<p/>
</div>
<div class="page"><p/>
<p>4.4 Logistic Regression 123
</p>
<p>sbp
</p>
<p>0 10 20 30
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
o o
oo
</p>
<p>ooo
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
oo
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
o oo
o
</p>
<p>ooo
o ooo
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo o
oo o
o
oooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>oo
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>oo
</p>
<p>oo oo
oo
</p>
<p>ooo
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o o
o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
ooo
</p>
<p>o
o
</p>
<p>o
</p>
<p>oooo
</p>
<p>oo
o
</p>
<p>o
oooo
</p>
<p>o
o
</p>
<p>o
ooo
</p>
<p>oo
oo
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
oo
</p>
<p>o
</p>
<p>oo
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o o
</p>
<p>oo o
o
</p>
<p>ooo
o o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>oo oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oooo
ooo
</p>
<p>o
o
</p>
<p>oooo o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>oo
o
</p>
<p>oo
o
</p>
<p>oo o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
oo oo
o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
o
oo
</p>
<p>o o
oo o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o o
</p>
<p>o
o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
oo o
</p>
<p>oo
o
</p>
<p>o
o
oo
oo
oo
o
o
</p>
<p>o
</p>
<p>ooo
o
oo
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>oo o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>oo
</p>
<p>oooo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
o
o
</p>
<p>o
</p>
<p>oo
o o
o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o o
o
</p>
<p>ooo
o
</p>
<p>o
ooo
</p>
<p>oo oo
o oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o oo
ooo
</p>
<p>o
oooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o ooo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>oo
</p>
<p>oo
oo
</p>
<p>oo
o
oo
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o ooo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oooo
</p>
<p>o o
o
</p>
<p>o
o o
oo
</p>
<p>o
o
</p>
<p>o
oo o
oo
</p>
<p>o o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
ooo
</p>
<p>o
</p>
<p>oo
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
oo o
o
</p>
<p>ooo
oo
</p>
<p>o o
o
</p>
<p>o
</p>
<p>oooo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>ooo o
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>oooo o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o o
o o
o
</p>
<p>o o
o
</p>
<p>o oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o o
ooo
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
oo
</p>
<p>o
o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo o
oo
</p>
<p>o
</p>
<p>o
o
ooo ooo
</p>
<p>o
o
</p>
<p>o
</p>
<p>ooo oo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>0.0 0.4 0.8
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>ooo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>oo
</p>
<p>o oo
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
o
</p>
<p>o
</p>
<p>o o
o
o
</p>
<p>o
</p>
<p>oo
o o
o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
o
o oo
o o
oo
oo oo
oo o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>ooo
o
oooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o o
</p>
<p>oooo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo
o
</p>
<p>o
</p>
<p>oo
o
o
</p>
<p>oo
oo o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
o
o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
o
</p>
<p>oo
o
</p>
<p>o
o
o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo oo
</p>
<p>o o
o
</p>
<p>o
ooo
o
</p>
<p>o
o
</p>
<p>o
o oo
</p>
<p>o
o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
ooo
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
oo
ooo
o
</p>
<p>o oo
oo
</p>
<p>oo
o
</p>
<p>o
</p>
<p>ooo o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
o
oo
o
</p>
<p>o
o
</p>
<p>ooooo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>oo
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>oo o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
ooo o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o o
oo o
o
</p>
<p>o
o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
ooo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo o
oo
o
</p>
<p>o
o
</p>
<p>o
o oooo
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo oo oo
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
ooo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o ooo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o o o
o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
o
</p>
<p>oo
o
o
</p>
<p>o o o
o o
oo
</p>
<p>o oo oo
oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
oo
</p>
<p>o oo
o
</p>
<p>o oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>oooo
o
o
</p>
<p>o
</p>
<p>o
oo
oo
</p>
<p>oo
oo
</p>
<p>oo
o
o o
</p>
<p>o
</p>
<p>o o o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o ooo o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o oo
o
</p>
<p>o
o
</p>
<p>o
o oo
</p>
<p>oo
oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
oo
</p>
<p>o
</p>
<p>oo
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>ooo
o
</p>
<p>o oo
oo
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
o
</p>
<p>o
ooo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
o o
</p>
<p>ooo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o o oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>oo
o oo
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o o
</p>
<p>o
o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o o o
o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>0 50 100
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
o
o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo
oo
</p>
<p>oooo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o
</p>
<p>o ooo
o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
o
oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>oo
o ooo
ooo
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
o
</p>
<p>o oo
o
oooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o o
</p>
<p>oooo
o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o o
</p>
<p>oo
o
</p>
<p>o
oo
ooo
</p>
<p>o
</p>
<p>oo o
o
</p>
<p>o
</p>
<p>o
oo
o
</p>
<p>o
o
o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
o
oo
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o oo
o
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
ooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o oo
oo
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>ooo
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
oo o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>oo
oo
o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oooo
ooo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>oo
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oooo
o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
o
oo
</p>
<p>oo
ooo
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o o
</p>
<p>o
oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>ooo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
oo
</p>
<p>o
oo
o
</p>
<p>o
o
oooooo
o
o
</p>
<p>o
</p>
<p>o oooo
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>1
0
0
</p>
<p>1
6
0
</p>
<p>2
2
0
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>ooo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
o o
</p>
<p>oo oo
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o o
o
</p>
<p>o
o o
</p>
<p>o
o o
</p>
<p>oo
o oo o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>ooo
o oo
</p>
<p>o
o oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o o
</p>
<p>o o
o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>oo
</p>
<p>oo
o
</p>
<p>o
oo
</p>
<p>oo o
</p>
<p>o
</p>
<p>o o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oooo
</p>
<p>o o
o
</p>
<p>o
oo oo
</p>
<p>o
o
</p>
<p>o
ooo
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
o o o
</p>
<p>o
</p>
<p>oo
o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
o o o
</p>
<p>o
</p>
<p>oo o
o o
</p>
<p>oo
o
</p>
<p>o
o o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o ooo o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
oo oo
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o o
oo o
</p>
<p>o
o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o o
</p>
<p>o
ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
o
</p>
<p>o
</p>
<p>oooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o o
o
</p>
<p>oo
o
</p>
<p>o
o
oooo o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo ooo
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>0
1
0
</p>
<p>2
0
</p>
<p>3
0
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
o o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
o
o
o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>oo oo
o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
o
o
o o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
o o oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
o
ooo
o
</p>
<p>ooooo
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oooo
o
</p>
<p>o
o oooo
o
</p>
<p>o
o
</p>
<p>oo o
o oo
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oooo
o
</p>
<p>oo
oo
</p>
<p>o
oo
</p>
<p>oo
</p>
<p>oo
oo
ooo
</p>
<p>o
</p>
<p>oooo o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>ooo oo
oo
oo
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o ooo o ooo
oo
</p>
<p>oo
oo
</p>
<p>o
o o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>oo
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o o
o
</p>
<p>oo
oo
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o o
oo
</p>
<p>oo o
o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>oooo
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>ooo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>ooooooo o
o
</p>
<p>oooo
</p>
<p>o
</p>
<p>o
o
o oooo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
ooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
o
</p>
<p>o
ooo
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
o
o
</p>
<p>o
o
</p>
<p>oo o
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>ooo oo
o
</p>
<p>o
</p>
<p>o
o
o
o
</p>
<p>oo
o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
o
o
</p>
<p>o
o o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo oo
</p>
<p>o
</p>
<p>ooo oo ooooo
o
</p>
<p>oo
o
</p>
<p>o
oo
ooo
o
</p>
<p>oo
o
</p>
<p>o o
o o o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>tobacco
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
o o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo o
</p>
<p>o
o
oo oo
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
o
o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
o
o
</p>
<p>ooo oo
o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oooo
o
</p>
<p>o
o
</p>
<p>ooo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oooo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o ooo
</p>
<p>o
o o
oo
</p>
<p>o
oo
</p>
<p>oo
</p>
<p>oo ooooo
</p>
<p>o
</p>
<p>oo
o
ooo
oo
</p>
<p>o
</p>
<p>oo
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
o ooo
</p>
<p>ooo o
</p>
<p>o
</p>
<p>o o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oooo oooo
o ooo
o o
o
</p>
<p>oo
o
</p>
<p>o
oo
</p>
<p>oo
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o oo
oo
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>ooo
o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>oo
</p>
<p>o
</p>
<p>oooo
o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
ooo
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo ooo oo
ooooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o ooo
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
oo o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
o oo
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>oo o
oo
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>ooo
o
o
</p>
<p>oo
o
</p>
<p>o
o o
</p>
<p>o
oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo oo
</p>
<p>o
</p>
<p>ooo ooooo o
oo oo
</p>
<p>o
oo
</p>
<p>o
oo o
o
</p>
<p>oo
o
</p>
<p>oooo o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
oo
</p>
<p>o
o
oooo
</p>
<p>o
</p>
<p>o
</p>
<p>ooo oo
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
oo
</p>
<p>o
o
o
</p>
<p>o
o
o oo o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>ooo
o
</p>
<p>oooo o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oooo
o
o
ooo oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o ooo
o
o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
o
ooo
o
o o
o o
</p>
<p>o
oo
</p>
<p>oo
oo
</p>
<p>oo oo o
</p>
<p>o
</p>
<p>ooooo o
oo
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
o
</p>
<p>oo
oo
</p>
<p>ooo o
oooo
</p>
<p>o
</p>
<p>o o
oo
o
o
</p>
<p>o
</p>
<p>o
o
oo oooo oo
</p>
<p>o o
oo
</p>
<p>ooo
oo
o
oo
o
</p>
<p>o o
o
oo
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o oo
ooo o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o o
</p>
<p>oo
</p>
<p>o oo
o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo oo
o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
o
o
</p>
<p>o
</p>
<p>o
o
ooo
o
o
o
</p>
<p>o
</p>
<p>oooooooo
o
</p>
<p>oooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o oo
</p>
<p>o
</p>
<p>o
o
oo
o
</p>
<p>oo oo
o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
oo o
</p>
<p>o
oo o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
o
oooo o
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
oo
</p>
<p>o oo
o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
o
o oo
</p>
<p>o
</p>
<p>ooooo oo ooo
o
o o
</p>
<p>o
</p>
<p>o
o o
o oo
</p>
<p>o
oo
</p>
<p>o
oo
</p>
<p>o
oo
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
o o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>oooo o o
o
</p>
<p>o
</p>
<p>ooo
o o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oooo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>ooo
o
</p>
<p>o oo oo
o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>oo oo
o
</p>
<p>o
o oo oo
o
</p>
<p>o
o
</p>
<p>o ooo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo oo
o
o o
</p>
<p>o o
o
</p>
<p>oo
</p>
<p>oo
</p>
<p>oo
oo
</p>
<p>oo o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>ooo
oo
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
ooo oo
</p>
<p>ooo o
</p>
<p>o
</p>
<p>o o
oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oooo oo ooo o
oo
</p>
<p>oo
o ooo
</p>
<p>o
o
oo
</p>
<p>o
</p>
<p>o o
o
oo
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o o
o
</p>
<p>oo
oo
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
ooo
</p>
<p>o
o
</p>
<p>oo
o
o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o oo
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o oo
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o ooo ooo o
o
</p>
<p>oooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o ooo o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
o
o
o o
</p>
<p>o
ooo
</p>
<p>o
</p>
<p>oo
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>oo o
oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
ooo oo
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>oo
oo o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo oo
</p>
<p>o
</p>
<p>ooooooo oo
o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
oo
ooo
</p>
<p>oo
o
</p>
<p>o
ooo oo
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>ooo ooo
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>o
o
o
o
</p>
<p>o
o
o oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oooo
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
o
</p>
<p>o
ooo
</p>
<p>o
</p>
<p>ooooo
o
</p>
<p>o
o
o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>oooo
o
</p>
<p>o
o
</p>
<p>oo oo
o
</p>
<p>o
o
</p>
<p>ooo
o oo
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oooo
</p>
<p>o
oo
</p>
<p>o o
o
</p>
<p>o o
</p>
<p>oo
</p>
<p>o
oooooo
</p>
<p>o
</p>
<p>oo
o
</p>
<p>ooo
oo
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
ooooo
o
</p>
<p>ooo
</p>
<p>o
</p>
<p>oo
o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oooooooo
ooo o
</p>
<p>oo
o
</p>
<p>oo
o
</p>
<p>oo
o
</p>
<p>oo
o
oo
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>ooo
</p>
<p>oo
oo
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
oo o
</p>
<p>o
o
oo
</p>
<p>o
o
o o
</p>
<p>o
o
o o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oooo
</p>
<p>o
</p>
<p>oooo o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oooo oooo
o
</p>
<p>o ooo
</p>
<p>o
</p>
<p>o
ooooo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
o
</p>
<p>o
ooo
o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
o
</p>
<p>o
o o
</p>
<p>oo o
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
ooo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o o
o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
o
</p>
<p>ooo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o oo
</p>
<p>o
</p>
<p>oooooooooo
o
oo
</p>
<p>o
o
</p>
<p>oo
oo
o oo
</p>
<p>o
</p>
<p>o
oo ooo
</p>
<p>ooo
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
o
</p>
<p>oo
oo oo o o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o oo o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o o o oo
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>oo oo
o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o ooo
o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o oo o
o
</p>
<p>o o
</p>
<p>o o
o
oo
</p>
<p>oo
oo
oo
</p>
<p>oo o
</p>
<p>o
</p>
<p>o o oooo
oo
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>oo o oo
oo
</p>
<p>o o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo oooooo
o o
</p>
<p>oo
oo
</p>
<p>o
oo
</p>
<p>o
oo
o
</p>
<p>oo
o
</p>
<p>oo
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o o o
</p>
<p>o o
oo
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
ooo
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o o
</p>
<p>o
</p>
<p>ooo o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oooo ooo o
o
</p>
<p>o ooo
</p>
<p>o
</p>
<p>o
oo o oo o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
ooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
o
</p>
<p>oo oo
</p>
<p>o
oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o oo
o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>ooooooooo o
o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>oo oo
o ooo
</p>
<p>o
ooo oo
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o ooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
oo
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>oo o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o o
o o
</p>
<p>o
</p>
<p>o oo
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo oo
o
o
</p>
<p>o
oo
</p>
<p>oo
o
</p>
<p>o
o
o
</p>
<p>o
o
o o
o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
o
</p>
<p>oo
oo
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oooo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o ooo
o
</p>
<p>ooo
o
</p>
<p>oo
</p>
<p>oo
o
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>oo
o
</p>
<p>o
oo
</p>
<p>oo
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
oo
</p>
<p>oo
o o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>ooo
oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
o o oo
</p>
<p>oooo
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o o o
</p>
<p>o
</p>
<p>oo
o oo
</p>
<p>oo
ooo o
</p>
<p>o
</p>
<p>oo
o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>oo
o
</p>
<p>o
o o o
</p>
<p>oo oo
o
</p>
<p>o
o
</p>
<p>ooo
o
</p>
<p>o
ooo o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o o
ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>ooo o
o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>o
</p>
<p>o
o ooo
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o ooo o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o ooo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
o
</p>
<p>oo o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o o
o
o
</p>
<p>o
</p>
<p>o oo
o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>ooo o
o
o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
o
o
</p>
<p>oo
o
o
o
o
</p>
<p>o
o
</p>
<p>o
ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oooo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oooo
o
</p>
<p>ooo
o
</p>
<p>oo
</p>
<p>oo o
o
</p>
<p>ooo
o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>oo
</p>
<p>oooo o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
o
o
o
o
o
o
o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oooo
oo
o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
o
o
</p>
<p>o
oo
o
</p>
<p>o
o
</p>
<p>oo oo
oo
</p>
<p>o o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>ooo
o o
</p>
<p>o
o oo
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
ooo o
</p>
<p>o
oo
o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
oo
</p>
<p>ooo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>ooo
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>oo
o ooo oo
</p>
<p>ooo
</p>
<p>o
</p>
<p>o o
o
o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
o oo
</p>
<p>ooo
o
</p>
<p>o
o
</p>
<p>o
oo o
</p>
<p>o
</p>
<p>o ooo o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>oo
o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo oo
o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>o
o
o
o
oo
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o
o
o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ldl
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
o
</p>
<p>o
o
</p>
<p>oo oo
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
o
o oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>ooo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oooo
oo
</p>
<p>o
o oo o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o o
o
</p>
<p>o
o
o
o
o
o
o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
o
o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
ooo
o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo o
o ooo
o
</p>
<p>o o
</p>
<p>oo
o
o
</p>
<p>ooo
o
oo
</p>
<p>o
o
oo
</p>
<p>ooooo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
o o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
o o
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
o oo
</p>
<p>o
oo
o
</p>
<p>o
o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo
o
</p>
<p>o
o
oooo
</p>
<p>oo
oo o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o oo
oo
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>oo
oooo
</p>
<p>o
oo
o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
oo
o
o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o o
o
o
</p>
<p>o o
o
</p>
<p>oo oo
o
</p>
<p>ooo
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
oo
</p>
<p>o o
o
</p>
<p>o
o oo
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
o
oo o
</p>
<p>o
</p>
<p>o
o o
oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
o
</p>
<p>oo
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
oo
</p>
<p>oo
oooo
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>oo o
</p>
<p>o
o
o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>oooo
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
o o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
o oo
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
o
o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo o
o o
</p>
<p>o
oo
</p>
<p>oo
o
</p>
<p>o
o
o
</p>
<p>o o
o oo o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o
o o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo o
o
</p>
<p>o o
oo
</p>
<p>oo
</p>
<p>ooo
o
</p>
<p>oo
o
</p>
<p>o
oo
</p>
<p>o
o
oo
</p>
<p>oo
o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oooo
oo
</p>
<p>o o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o oo
oo
</p>
<p>o
o oo
</p>
<p>o
</p>
<p>o o
o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o o
oo oo
</p>
<p>o
o o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>oooo
o oo o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
oo
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>o
o
o
</p>
<p>o
ooo
</p>
<p>o
o oo
</p>
<p>o
o
</p>
<p>o
o oo
</p>
<p>o
</p>
<p>o
oo oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>oo
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o oo
o
</p>
<p>o
</p>
<p>o
oooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
oo
oo
</p>
<p>o
ooo
</p>
<p>o
</p>
<p>o
o oo
</p>
<p>o
o
o
</p>
<p>o
o o
ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
ooo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
o
</p>
<p>o
o
</p>
<p>ooo o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
oo
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
ooo
</p>
<p>o
o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
</p>
<p>o oo
</p>
<p>o o
o
o
</p>
<p>o
</p>
<p>ooo o
oo
o
</p>
<p>oo
o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o o
oo
o
o
o
</p>
<p>o
o
</p>
<p>o
o
o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
ooo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oooo
o
</p>
<p>o
o oo
</p>
<p>o o
</p>
<p>oo
o
o
</p>
<p>ooo
o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oooo o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
o
o
</p>
<p>o
oo
o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
o o
</p>
<p>oo
oo
o
</p>
<p>o
o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
o
o ooo
</p>
<p>oo
</p>
<p>o o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
o
o
</p>
<p>o
o
o
</p>
<p>o
o
oo
</p>
<p>o
o
o
</p>
<p>ooo
oo
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>ooo
ooo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>ooo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o oo
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>oo
ooo oo oooo
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o o
o
</p>
<p>o
o oo
</p>
<p>o oo
o
o
</p>
<p>o
o
</p>
<p>ooo
o
</p>
<p>o
oo oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>oo
ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
o o oo
</p>
<p>o
oo o
</p>
<p>o
</p>
<p>oo
oo
o
o
oo
oo
ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
oo o
</p>
<p>o
o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>2
6
</p>
<p>1
0
</p>
<p>1
4
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
oo oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
oo o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o o
o o
</p>
<p>o
</p>
<p>o oo
o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo o o
o
</p>
<p>o
o
</p>
<p>oo
oo o
</p>
<p>o
o
o
</p>
<p>o o
oo
</p>
<p>o
oo oo
</p>
<p>o
o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo o
oo o oo
</p>
<p>oo
</p>
<p>oo o
o
</p>
<p>oo
o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>oo
</p>
<p>oo
oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>ooo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
oo
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>oo
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o o
</p>
<p>oo
</p>
<p>o o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo o
o o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o oo
oooo
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>oo
o oo oo
</p>
<p>o
ooo
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
o
</p>
<p>o
oo
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
oo
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o o o
</p>
<p>ooo
o
</p>
<p>o
o
</p>
<p>o
oo o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
ooo o
</p>
<p>o ooo
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>o
</p>
<p>o
oooo
</p>
<p>o o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
o ooo o
</p>
<p>o
o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>0
.0
</p>
<p>0
.4
</p>
<p>0
.8
</p>
<p>o
</p>
<p>o
</p>
<p>o ooo
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>oo
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o o
</p>
<p>oo
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o oo oo
</p>
<p>ooo
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>ooooo
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo oo
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>ooo oo oo
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo oo
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>oooo o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo o
</p>
<p>o o
</p>
<p>oo oooo
</p>
<p>o o
</p>
<p>o oo oo oo oo oo
</p>
<p>oo o o
</p>
<p>oo o
</p>
<p>ooo
</p>
<p>ooo o
</p>
<p>o oooo
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo o ooo oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>ooo ooo
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o oooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oooo
</p>
<p>oo o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o o
</p>
<p>oooo o oo
</p>
<p>o
</p>
<p>ooooooo o
</p>
<p>o
</p>
<p>ooooo
</p>
<p>oo
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>oo
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>oo oo
</p>
<p>oo
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o oooo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o o
</p>
<p>oo
</p>
<p>o
</p>
<p>o oo
</p>
<p>oo
</p>
<p>oo
</p>
<p>oo
</p>
<p>oo
</p>
<p>oo o
</p>
<p>oooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o o
</p>
<p>o
</p>
<p>o o o
</p>
<p>oo ooo
</p>
<p>ooo
</p>
<p>o
</p>
<p>oo
</p>
<p>oo o
</p>
<p>o o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oooo
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
</p>
<p>oo
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>oooo
</p>
<p>o o
</p>
<p>o
</p>
<p>o o oo
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>oo
</p>
<p>o o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>ooooo
</p>
<p>o oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o o
</p>
<p>oo
</p>
<p>o o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o oo oo
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>ooo o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>oo
</p>
<p>oo oo ooo
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oooo
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oooo o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oooo
</p>
<p>oo
</p>
<p>o ooooo
</p>
<p>o o
</p>
<p>o oo oo oooo o o
</p>
<p>oo oo
</p>
<p>ooo
</p>
<p>ooo
</p>
<p>oo oo
</p>
<p>o oo oo
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo oooo oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>oooo oo
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
</p>
<p>oo ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o o
</p>
<p>oo o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>oooo ooo
</p>
<p>o
</p>
<p>oooooooo
</p>
<p>o
</p>
<p>oooo o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>oo
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o ooo
</p>
<p>o o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>oo
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo ooo
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o o
</p>
<p>o
</p>
<p>o oo
</p>
<p>oo
</p>
<p>o o
</p>
<p>oo
</p>
<p>oo
</p>
<p>ooo
</p>
<p>oo oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o oo
</p>
<p>oo oo o
</p>
<p>o oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o oo
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo oo
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>oo
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o ooo
</p>
<p>o o
</p>
<p>o
</p>
<p>o oo o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>ooo oo
</p>
<p>o oo
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o ooo o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>oooo
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>oo
</p>
<p>oo oo o oo
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o ooooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oooo
</p>
<p>o o
</p>
<p>oo oooo
</p>
<p>oo
</p>
<p>o oo ooooooo o
</p>
<p>oo oo
</p>
<p>o oo
</p>
<p>o oo
</p>
<p>o ooo
</p>
<p>ooo oo
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo oo oo oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>oooooo
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>ooooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oooo
</p>
<p>o oo
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o o
</p>
<p>ooooooo
</p>
<p>o
</p>
<p>ooo ooo oo
</p>
<p>o
</p>
<p>oooo o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>oo
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo oo
</p>
<p>o o
</p>
<p>o o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>oo
</p>
<p>o o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo oooo
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
</p>
<p>oo
</p>
<p>o o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>ooo
</p>
<p>o o
</p>
<p>oo
</p>
<p>oo
</p>
<p>oo
</p>
<p>oo o
</p>
<p>oooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o o
</p>
<p>o
</p>
<p>ooo
</p>
<p>oo oo o
</p>
<p>o oo
</p>
<p>o
</p>
<p>oo
</p>
<p>ooo
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o o o
</p>
<p>oo
</p>
<p>o o
</p>
<p>oo
</p>
<p>o
</p>
<p>ooo o
</p>
<p>o
</p>
<p>famhist
</p>
<p>o
</p>
<p>o
</p>
<p>o oo o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>ooo o o
</p>
<p>oo o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o o
</p>
<p>o o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o oooo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>oo ooo oo
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oooo
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oooooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o ooo
</p>
<p>o o
</p>
<p>oooooo
</p>
<p>oo
</p>
<p>o ooooooooo o
</p>
<p>oo oo
</p>
<p>oo o
</p>
<p>o oo
</p>
<p>o ooo
</p>
<p>ooo oo
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo oooo oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>oooooo
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>ooo oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo oo
</p>
<p>oo o
</p>
<p>o o o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>oo ooo oo
</p>
<p>o
</p>
<p>o ooo ooo o
</p>
<p>o
</p>
<p>oooo o
</p>
<p>oo
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>oo oo
</p>
<p>oo
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo oooo
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o o
</p>
<p>oo
</p>
<p>o
</p>
<p>ooo
</p>
<p>oo
</p>
<p>o o
</p>
<p>oo
</p>
<p>oo
</p>
<p>oo o
</p>
<p>oo oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o o
</p>
<p>o
</p>
<p>o oo
</p>
<p>ooo oo
</p>
<p>o oo
</p>
<p>o
</p>
<p>oo
</p>
<p>ooo
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o o
</p>
<p>o o
</p>
<p>oo
</p>
<p>o
</p>
<p>ooo o
</p>
<p>o o
</p>
<p>o
</p>
<p>o o oo
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>oo ooo
</p>
<p>o oo
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o oooo
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>ooo o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>ooooooo
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo oo
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oooooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o ooo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o ooo
</p>
<p>oo
</p>
<p>oooooo
</p>
<p>oo
</p>
<p>o oo oo oooo oo
</p>
<p>oooo
</p>
<p>ooo
</p>
<p>ooo
</p>
<p>oooo
</p>
<p>ooo oo
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooooooooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
</p>
<p>ooo ooo
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o oooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo oo
</p>
<p>ooo
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>ooo o ooo
</p>
<p>o
</p>
<p>oooo oooo
</p>
<p>o
</p>
<p>o ooo o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oooo
</p>
<p>oo
</p>
<p>oo
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo oo o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o o
</p>
<p>o o
</p>
<p>o
</p>
<p>ooo
</p>
<p>oo
</p>
<p>o o
</p>
<p>oo
</p>
<p>o o
</p>
<p>ooo
</p>
<p>oooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o o
</p>
<p>o
</p>
<p>oo o
</p>
<p>ooo oo
</p>
<p>oo o
</p>
<p>o
</p>
<p>oo
</p>
<p>o oo
</p>
<p>oo
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oooo
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o o
</p>
<p>o
</p>
<p>o ooo
</p>
<p>o o
</p>
<p>o
</p>
<p>o ooo
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o o
</p>
<p>oo
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o ooo o
</p>
<p>o oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o o
</p>
<p>oo
</p>
<p>o o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>ooo oo
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo o o
</p>
<p>o
</p>
<p>o o o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>oo
</p>
<p>oo oo o o o
</p>
<p>o o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo oo
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o oo o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o ooo
</p>
<p>o o
</p>
<p>o ooooo
</p>
<p>o o
</p>
<p>o oooo oooo o o
</p>
<p>oo oo
</p>
<p>o o o
</p>
<p>ooo
</p>
<p>o o oo
</p>
<p>o oo oo
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo oooo oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
</p>
<p>oooo o o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
</p>
<p>oo oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo o
</p>
<p>oo o
</p>
<p>o o o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>oo
</p>
<p>ooo o o oo
</p>
<p>o
</p>
<p>oooo ooo o
</p>
<p>o
</p>
<p>o ooo o
</p>
<p>oo
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o o oo
</p>
<p>o o
</p>
<p>oo
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>oo
</p>
<p>o o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo oooo
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>oo
</p>
<p>o o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o oo
</p>
<p>oo
</p>
<p>o o
</p>
<p>oo
</p>
<p>oo
</p>
<p>oo o
</p>
<p>oooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>oo
</p>
<p>o
</p>
<p>o oo
</p>
<p>oo ooo
</p>
<p>ooo
</p>
<p>o
</p>
<p>oo
</p>
<p>oo o
</p>
<p>o o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o oo
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
</p>
<p>oo
</p>
<p>o o
</p>
<p>oo
</p>
<p>o
</p>
<p>oooo
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>ooo
ooo
</p>
<p>o
oo
</p>
<p>o
o o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o o
</p>
<p>o o
o
</p>
<p>ooo o o
o oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>ooo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
oo
</p>
<p>o
oo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
o
</p>
<p>oo o o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo oo
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>oo
</p>
<p>o oooo o oo
</p>
<p>o
o oo o
</p>
<p>ooo
o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
ooo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>oo
oo o
</p>
<p>o
</p>
<p>oo
o
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>oo oo
oo
</p>
<p>o o
o
o o
oo
</p>
<p>oo o
</p>
<p>o
o
</p>
<p>oo
oo
</p>
<p>o o
</p>
<p>o o
o oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o
o
</p>
<p>o
oo o
</p>
<p>oo
o
</p>
<p>o
o
o
</p>
<p>o
o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
oo oo o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
o oo
</p>
<p>o
oo
</p>
<p>o
o o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo o
</p>
<p>o
</p>
<p>oo o
o
</p>
<p>o
</p>
<p>o o
oo
o ooo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
oo o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
oo
</p>
<p>o
o
o
ooo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>ooo o
o o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>oo
o
</p>
<p>o oooo
o oo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
ooo
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
oo
</p>
<p>o
oo
oo
o
</p>
<p>o
</p>
<p>oo
o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
oo oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
ooooo
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>oo
</p>
<p>ooo
o
o o oo
</p>
<p>o
o oo
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
oooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>ooo
o
</p>
<p>oo
o
</p>
<p>o oo
o
</p>
<p>o
o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
o oo
</p>
<p>o
o
</p>
<p>oo
</p>
<p>oo
ooo
</p>
<p>oo
ooo
</p>
<p>o
o
</p>
<p>oo
oo
oo
</p>
<p>o
o o
</p>
<p>o o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
o
</p>
<p>o
o
oo
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
o
oo
o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>oo oo
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o oo
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o oo
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>oo
o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>oo
oo
o
o
o
ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>oooo
o o
</p>
<p>o
oo
</p>
<p>o
o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o oo o
</p>
<p>oo
oooooo
</p>
<p>ooo
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
o o
</p>
<p>o o
o
oo
oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
oooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
ooo
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>oo
</p>
<p>ooooo ooo
</p>
<p>o
o oo
o
ooo
o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
o o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>oo o
o
</p>
<p>oo
oo oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
oooo
</p>
<p>oo
oo
ooo
</p>
<p>oo
ooo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o o
o o
oo o
</p>
<p>o o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o o o
</p>
<p>o
ooo
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
ooooo
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>oo
</p>
<p>o
o
o o
</p>
<p>o
ooo o
</p>
<p>oo
oo
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>oo oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o ooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
oo
</p>
<p>o
o
o
o
</p>
<p>oo
oo
o
</p>
<p>o
o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
o
o
</p>
<p>o o
o
</p>
<p>o
o
</p>
<p>oo
o oo
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>oo
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o ooo
</p>
<p>oo
o
ooo oo
</p>
<p>ooo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o oo
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oooo o
</p>
<p>o
oo
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>oo
o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
oooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o o
o oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>oo
ooooo o
</p>
<p>oo
</p>
<p>o
ooo
o
o
oo
o
o oo
</p>
<p>o
</p>
<p>o
oo
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
o
o
</p>
<p>o
o ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
o
</p>
<p>oo o
o
</p>
<p>o o
o
</p>
<p>ooo
o
</p>
<p>o
o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>oo ooo
oo
oo
oo o
oo
o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o o
oo
oo
</p>
<p>oo o
oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
ooo
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
o
o
o
o
</p>
<p>o
o
</p>
<p>o
o
o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
oo ooo
o
</p>
<p>o
</p>
<p>o o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>oo
oo
</p>
<p>oo
o
</p>
<p>oo
o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo oo
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
oo
oooo
</p>
<p>o
o
o
</p>
<p>o
oo
</p>
<p>o
o
o
</p>
<p>o
o o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o ooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
oo
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
oo
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
oo
</p>
<p>o
o
</p>
<p>o
ooo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>obesity
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>oooo
oo
</p>
<p>o
oo
</p>
<p>o
</p>
<p>oo
oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>ooo
o oo
o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>ooo
oo
</p>
<p>o
oo
o o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
oo
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
o
</p>
<p>o
oooo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>ooo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>oo
</p>
<p>ooo
o
o ooo
</p>
<p>o
o oo
</p>
<p>o
o
o
o o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o oo
o
</p>
<p>o o
o
o oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o ooo o
oo
</p>
<p>o o
ooo
</p>
<p>oo
oo o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
oo
o o
</p>
<p>ooo
oo
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o
oo
o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
o
</p>
<p>o
o
</p>
<p>o
o o o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o
o
o
</p>
<p>o
o
</p>
<p>o
o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
o
o oo o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
o o oo
</p>
<p>o o
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>ooo o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>ooo o
o
o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>oooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>oo
o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>ooooo
o
o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>o 1
5
</p>
<p>2
5
</p>
<p>3
5
</p>
<p>4
5
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
o
</p>
<p>oo
o ooo
</p>
<p>o
oo
</p>
<p>o
o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>oo
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o oo
o o
</p>
<p>o
o oo oo
</p>
<p>o oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
oo o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o o o
</p>
<p>o o
</p>
<p>o
oo
</p>
<p>o o
o
</p>
<p>o
</p>
<p>oo
o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
oo
</p>
<p>o oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
ooo
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>oo
</p>
<p>ooo
o
</p>
<p>o o oo
</p>
<p>o
o oo
</p>
<p>o
o oo
</p>
<p>o
ooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>oo o
o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o o oo o
oo
</p>
<p>oo
o oo
</p>
<p>oo
ooo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
o
</p>
<p>o oo
o o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>o oo o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o o
oo
</p>
<p>o oo
oo
</p>
<p>o o
o
</p>
<p>o o
o o
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>ooo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
ooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>ooo o
o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
oo
o
o
o
o o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>0
5
0
</p>
<p>1
0
0
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
ooo o
</p>
<p>o
</p>
<p>oo oo
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>ooooo
o
ooo
</p>
<p>o
oo o
</p>
<p>o
o
</p>
<p>o
o
o
</p>
<p>o oo
o
</p>
<p>oo oo
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>ooo o oo
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
oooo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o ooooo
oo oo
o
</p>
<p>oo
o
</p>
<p>o oo
oo
</p>
<p>o
</p>
<p>o
oooo oo o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>o o
</p>
<p>o o
o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>ooo
oo o
o
o
</p>
<p>o
o o
o
</p>
<p>o
o
</p>
<p>o oooo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>ooo
oo o
</p>
<p>o
o ooo ooooo
</p>
<p>oo o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o ooo
o
o
oo
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
o
</p>
<p>o
o
</p>
<p>o oo o
oo
</p>
<p>o
o
</p>
<p>o o
o
</p>
<p>oo
o
</p>
<p>o
o
o o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oooo oo
o
</p>
<p>o
</p>
<p>o oo
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>ooo
oo
o
</p>
<p>ooo oo o
o
</p>
<p>ooo
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oooo
o
oo o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
oo ooo
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo oooo o
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>ooo o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o o
</p>
<p>o
o
</p>
<p>ooo oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>ooo
</p>
<p>o o
</p>
<p>o o
oo
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
ooo
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
oo
</p>
<p>oo
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oooo oo oo
oooooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
oo o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
oooo
</p>
<p>o
</p>
<p>ooo o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo ooo
</p>
<p>o
ooo
</p>
<p>o
o
oo
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>ooo
o
</p>
<p>o ooo
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oooo
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>oooo
</p>
<p>o
</p>
<p>ooooo
o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oooooo
o
</p>
<p>ooo
o
</p>
<p>o o
o
</p>
<p>ooooo
</p>
<p>o
</p>
<p>o
oooo ooo
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
o
</p>
<p>ooo
o o
</p>
<p>oo
o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>ooo
o
o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
o
oo ooooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
o o
</p>
<p>oo oooooo o
ooo
</p>
<p>oo
o o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oooo
o
ooo
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
ooo o
</p>
<p>ooo ooo
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o ooo oo o
</p>
<p>o
</p>
<p>o o o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
o
o
</p>
<p>o
ooo o
</p>
<p>o o
oo
ooo o
o
ooo
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>oooo
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
o
oo
</p>
<p>o
</p>
<p>oo oooo
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooooo oo
o
</p>
<p>o
o oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>ooo o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>oo
</p>
<p>o
o
</p>
<p>oo oo o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo o
</p>
<p>o o
</p>
<p>oo
oo
</p>
<p>oo o
</p>
<p>o
o
</p>
<p>o o
o
</p>
<p>o
ooo
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o o
o o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooooooooooo ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
ooo o
</p>
<p>o
</p>
<p>oooo
o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>ooo oo
o
o oo
oo
oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
ooo
</p>
<p>o
oo oo
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>ooooo o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>ooo oo
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oooooo
o o o
</p>
<p>o
o
</p>
<p>o o
o
</p>
<p>ooo
oo
</p>
<p>o
</p>
<p>o
ooo ooo o
</p>
<p>o
</p>
<p>oo
o
o
</p>
<p>o
ooo
</p>
<p>oo
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>ooo
o
o o
o
o
</p>
<p>o
oo
</p>
<p>o
o
o
</p>
<p>o oooo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o o o
o ooo
o
</p>
<p>oo o
o
o
</p>
<p>ooo
oo
</p>
<p>o
o
</p>
<p>o
oo
o
</p>
<p>oooo
o
</p>
<p>ooo
o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo o o
o o
o
</p>
<p>o
o o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>ooooo o
o
</p>
<p>o
</p>
<p>oo o
</p>
<p>oo
o
</p>
<p>o
o o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>oo oo
o o
ooo oo o o ooo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>ooo o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo oo oo
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo oooo
o
o
</p>
<p>oo o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>oo oo
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>oo
</p>
<p>o
o
</p>
<p>oooo o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo o
</p>
<p>oo
</p>
<p>o o
oo
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
o
</p>
<p>o oo
o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oooo ooooo ooo oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o o
oo o
o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
o ooo
</p>
<p>o
</p>
<p>ooo o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>ooo oo
o
</p>
<p>ooo
</p>
<p>o
ooo
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
ooo
o
</p>
<p>oo oo
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo oo o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
o o
oo
o
</p>
<p>o
</p>
<p>oooo o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o ooooo
ooo
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>oo o
oo
</p>
<p>o
</p>
<p>o
oooo oo o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
oooo
oo
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
ooooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
oo o
</p>
<p>oooooo oo
o
</p>
<p>o
ooo
</p>
<p>oo
o
o
</p>
<p>ooo
o
oo oo
</p>
<p>o
o
</p>
<p>o
o
o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oooo
oo
o
o
o o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo ooo
o
</p>
<p>o
</p>
<p>oo o
</p>
<p>oo
o
o
</p>
<p>o o
</p>
<p>o oo
</p>
<p>o
</p>
<p>oooo
oo
oo ooo o
o
</p>
<p>ooo
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oooo
</p>
<p>o
ooo
</p>
<p>o
</p>
<p>o
o
oo
</p>
<p>o
</p>
<p>oooo oo
o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o oooo
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>oo oo
o
o
o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o ooo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
o
</p>
<p>oo
oo
oo o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
oo o
o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oooooo oo
ooooo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
ooo o
</p>
<p>o
</p>
<p>oooo
</p>
<p>oo
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o ooo o
o
</p>
<p>ooo
o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
ooo
</p>
<p>o
ooo o
</p>
<p>o
</p>
<p>oo
o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>oooooo
o
oo
</p>
<p>o
</p>
<p>oo o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o oo oo
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo oo o
oo
o
</p>
<p>o
o
</p>
<p>oo
o
o o o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
ooo ooo o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
oo oo
</p>
<p>oo
oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
ooo o
</p>
<p>o
ooooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
o o
</p>
<p>o
o oooo oo o
o
</p>
<p>o
oo o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o oo
o
oo oo
</p>
<p>o
oo o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>ooo o
ooo oo o
</p>
<p>o
oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>oooooo o
</p>
<p>o
</p>
<p>ooo
</p>
<p>oo o
</p>
<p>o
</p>
<p>o o
</p>
<p>o oo
o
</p>
<p>ooo
oo
o
</p>
<p>o oooo oooo o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o ooo
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo o ooo
o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
oooo o
</p>
<p>o
o
o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
o oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>oo
oo
</p>
<p>o
o
</p>
<p>ooo oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
o
</p>
<p>o o
o o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o oo
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o ooo oooo
oooo oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o ooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
oo o
</p>
<p>o
</p>
<p>alcohol
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o oo o
</p>
<p>o
</p>
<p>ooo o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o ooo o
o
</p>
<p>o oo
o
</p>
<p>o
o o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>o o oo
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o oo oo o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo oo o
o o
</p>
<p>o o
o
</p>
<p>o o
o
</p>
<p>oo o
oo
</p>
<p>o
</p>
<p>o
oooo oo o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o oo o
</p>
<p>oo
oo
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>ooo
o oo
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
ooooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o oo
o oo
</p>
<p>o ooo
o
</p>
<p>ooo
oo
</p>
<p>o oo
o o
</p>
<p>o
o
</p>
<p>o oo
o
oo oo
</p>
<p>o
o
oo
</p>
<p>o oo
</p>
<p>o
o
</p>
<p>o o
o
</p>
<p>oo
</p>
<p>o
o
o
</p>
<p>o
oo oo o
</p>
<p>oo
o
</p>
<p>o
o o
</p>
<p>oo o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o o
oo
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o oooooo
</p>
<p>o
</p>
<p>o oo
</p>
<p>o o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o o
</p>
<p>o
</p>
<p>ooo o
o o
</p>
<p>ooo o o o
o
</p>
<p>ooo
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oooo
o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>ooo ooo
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>ooo oo o o
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o o
o oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
o oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o oo
</p>
<p>o o
</p>
<p>o o
oo
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
ooo
o
</p>
<p>o
</p>
<p>oo o
oo
</p>
<p>oo
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oooooooooo o ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o ooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
ooo
</p>
<p>o
</p>
<p>100 160 220
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o ooo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
ooo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oooo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
ooo
</p>
<p>o oo
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>oo oooooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oooo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oooo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o oo
</p>
<p>ooo
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>oooooooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
o
</p>
<p>2 6 10 14
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>ooo
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
ooo
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o oo
</p>
<p>o o o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo ooooo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o oo
</p>
<p>o
o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
oo oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oooo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>ooo
</p>
<p>ooo
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>oo
</p>
<p>oooo oo oo
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
o
</p>
<p>o
o
</p>
<p>15 25 35 45
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>ooo
</p>
<p>o
o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
ooo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o ooo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo o
</p>
<p>o
o o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oooo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o o
</p>
<p>oo oooo oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo o
</p>
<p>o
o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
oo
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oooo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>ooo
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>oo
</p>
<p>oooooooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
o
</p>
<p>20 40 60
</p>
<p>2
0
</p>
<p>4
0
</p>
<p>6
0
</p>
<p>age
</p>
<p>FIGURE 4.12. A scatterplot matrix of the South African heart disease data.
Each plot shows a pair of risk factors, and the cases and controls are color coded
(red is a case). The variable family history of heart disease (famhist) is binary
(yes or no).</p>
<p/>
</div>
<div class="page"><p/>
<p>124 4. Linear Methods for Classification
</p>
<p>TABLE 4.3. Results from stepwise logistic regression fit to South African heart
disease data.
</p>
<p>Coefficient Std. Error Z score
</p>
<p>(Intercept) &minus;4.204 0.498 &minus;8.45
tobacco 0.081 0.026 3.16
</p>
<p>ldl 0.168 0.054 3.09
famhist 0.924 0.223 4.14
</p>
<p>age 0.044 0.010 4.52
</p>
<p>other correlated variables, they are no longer needed (and can even get a
negative sign).
At this stage the analyst might do some model selection; find a subset
</p>
<p>of the variables that are sufficient for explaining their joint effect on the
prevalence of chd. One way to proceed by is to drop the least significant co-
efficient, and refit the model. This is done repeatedly until no further terms
can be dropped from the model. This gave the model shown in Table 4.3.
A better but more time-consuming strategy is to refit each of the models
</p>
<p>with one variable removed, and then perform an analysis of deviance to
decide which variable to exclude. The residual deviance of a fitted model
is minus twice its log-likelihood, and the deviance between two models is
the difference of their individual residual deviances (in analogy to sums-of-
squares). This strategy gave the same final model as above.
How does one interpret a coefficient of 0.081 (Std. Error = 0.026) for
</p>
<p>tobacco, for example? Tobacco is measured in total lifetime usage in kilo-
grams, with a median of 1.0kg for the controls and 4.1kg for the cases. Thus
an increase of 1kg in lifetime tobacco usage accounts for an increase in the
odds of coronary heart disease of exp(0.081) = 1.084 or 8.4%. Incorporat-
ing the standard error we get an approximate 95% confidence interval of
exp(0.081&plusmn; 2&times; 0.026) = (1.03, 1.14).
We return to these data in Chapter 5, where we see that some of the
</p>
<p>variables have nonlinear effects, and when modeled appropriately, are not
excluded from the model.
</p>
<p>4.4.3 Quadratic Approximations and Inference
</p>
<p>The maximum-likelihood parameter estimates β̂ satisfy a self-consistency
relationship: they are the coefficients of a weighted least squares fit, where
the responses are
</p>
<p>zi = x
T
i β̂ +
</p>
<p>(yi &minus; p̂i)
p̂i(1&minus; p̂i)
</p>
<p>, (4.29)</p>
<p/>
</div>
<div class="page"><p/>
<p>4.4 Logistic Regression 125
</p>
<p>and the weights are wi = p̂i(1&minus; p̂i), both depending on β̂ itself. Apart from
providing a convenient algorithm, this connection with least squares has
more to offer:
</p>
<p>&bull; The weighted residual sum-of-squares is the familiar Pearson chi-
square statistic
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>(yi &minus; p̂i)2
p̂i(1&minus; p̂i)
</p>
<p>, (4.30)
</p>
<p>a quadratic approximation to the deviance.
</p>
<p>&bull; Asymptotic likelihood theory says that if the model is correct, then
β̂ is consistent (i.e., converges to the true β).
</p>
<p>&bull; A central limit theorem then shows that the distribution of β̂ con-
verges to N(β, (XTWX)&minus;1). This and other asymptotics can be de-
rived directly from the weighted least squares fit by mimicking normal
theory inference.
</p>
<p>&bull; Model building can be costly for logistic regression models, because
each model fitted requires iteration. Popular shortcuts are the Rao
score test which tests for inclusion of a term, and the Wald test which
can be used to test for exclusion of a term. Neither of these require
iterative fitting, and are based on the maximum-likelihood fit of the
current model. It turns out that both of these amount to adding
or dropping a term from the weighted least squares fit, using the
same weights. Such computations can be done efficiently, without
recomputing the entire weighted least squares fit.
</p>
<p>Software implementations can take advantage of these connections. For
example, the generalized linear modeling software in R (which includes lo-
gistic regression as part of the binomial family of models) exploits them
fully. GLM (generalized linear model) objects can be treated as linear model
objects, and all the tools available for linear models can be applied auto-
matically.
</p>
<p>4.4.4 L1 Regularized Logistic Regression
</p>
<p>The L1 penalty used in the lasso (Section 3.4.2) can be used for variable
selection and shrinkage with any linear regression model. For logistic re-
gression, we would maximize a penalized version of (4.20):
</p>
<p>max
β0,β
</p>
<p>⎧
⎨
⎩
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>[
yi(β0 + β
</p>
<p>Txi)&minus; log(1 + eβ0+β
T xi)
</p>
<p>]
&minus; λ
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>|βj |
</p>
<p>⎫
⎬
⎭ . (4.31)
</p>
<p>As with the lasso, we typically do not penalize the intercept term, and stan-
dardize the predictors for the penalty to be meaningful. Criterion (4.31) is</p>
<p/>
</div>
<div class="page"><p/>
<p>126 4. Linear Methods for Classification
</p>
<p>concave, and a solution can be found using nonlinear programming meth-
ods (Koh et al., 2007, for example). Alternatively, using the same quadratic
approximations that were used in the Newton algorithm in Section 4.4.1,
we can solve (4.31) by repeated application of a weighted lasso algorithm.
Interestingly, the score equations [see (4.24)] for the variables with non-zero
coefficients have the form
</p>
<p>xTj (y &minus; p) = λ &middot; sign(βj), (4.32)
which generalizes (3.58) in Section 3.4.4; the active variables are tied in
their generalized correlation with the residuals.
Path algorithms such as LAR for lasso are more difficult, because the
</p>
<p>coefficient profiles are piecewise smooth rather than linear. Nevertheless,
progress can be made using quadratic approximations.
</p>
<p>********************************************************************************************************************************************************
**********
</p>
<p>**********
************
</p>
<p>************
************
</p>
<p>**************
***************
</p>
<p>0.0 0.5 1.0 1.5 2.0
</p>
<p>0
.0
</p>
<p>0
.2
</p>
<p>0
.4
</p>
<p>0
.6
</p>
<p>************************************
*****
</p>
<p>******
******
</p>
<p>*******
*******
</p>
<p>*******
*******
</p>
<p>*******
*******
</p>
<p>*******
*******
</p>
<p>*******
*******
</p>
<p>********
********
</p>
<p>********
*********
</p>
<p>*********
*********
</p>
<p>**************
***************
</p>
<p>***************
*******************
</p>
<p>***************************************************
******
</p>
<p>******
******
</p>
<p>******
*******
</p>
<p>*******
*******
</p>
<p>*******
*******
</p>
<p>*******
*******
</p>
<p>*******
*******
</p>
<p>********
********
</p>
<p>*********
*********
</p>
<p>**********
*********
</p>
<p>*********
*********
</p>
<p>*********
**********
</p>
<p>*********
</p>
<p>***********************************
****
</p>
<p>****
******
</p>
<p>******
******
</p>
<p>******
******
</p>
<p>******
******
</p>
<p>******
******
</p>
<p>******
******
</p>
<p>******
******
</p>
<p>******
******
</p>
<p>******
*******
</p>
<p>*******
*******
</p>
<p>*******
*******
</p>
<p>***********
***********
</p>
<p>***********
***********
</p>
<p>**************
*****
</p>
<p>*********************************************************************************************************************************************************************************************************************************************
</p>
<p>********************************************************************************************************************************************************************************************************************************************
*
</p>
<p>**
**
**
**
**
**
**
**
**
**
**
**
**
**
**
******
</p>
<p>******
********
</p>
<p>*********
*********
</p>
<p>*********
********
</p>
<p>********
********
</p>
<p>********
********
</p>
<p>********
*******
</p>
<p>*******
*********
</p>
<p>***********
***********
</p>
<p>************
*************
</p>
<p>*************
**************
</p>
<p>************
***
</p>
<p>obesity
</p>
<p>alcohol
</p>
<p>sbp
</p>
<p>tobacco
ldl
</p>
<p>famhist
</p>
<p>age
</p>
<p>1 2 4 5 6 7
</p>
<p>C
o
effi
</p>
<p>ci
en
ts
</p>
<p>β
j
(λ
)
</p>
<p>||β(λ)||1
</p>
<p>FIGURE 4.13. L1 regularized logistic regression coefficients for the South
African heart disease data, plotted as a function of the L1 norm. The variables
were all standardized to have unit variance. The profiles are computed exactly at
each of the plotted points.
</p>
<p>Figure 4.13 shows the L1 regularization path for the South African
heart disease data of Section 4.4.2. This was produced using the R package
glmpath (Park and Hastie, 2007), which uses predictor&ndash;corrector methods
of convex optimization to identify the exact values of λ at which the active
set of non-zero coefficients changes (vertical lines in the figure). Here the
profiles look almost linear; in other examples the curvature will be more
visible.
Coordinate descent methods (Section 3.8.6) are very efficient for comput-
</p>
<p>ing the coefficient profiles on a grid of values for λ. The R package glmnet</p>
<p/>
</div>
<div class="page"><p/>
<p>4.4 Logistic Regression 127
</p>
<p>(Friedman et al., 2010) can fit coefficient paths for very large logistic re-
gression problems efficiently (large in N or p). Their algorithms can exploit
sparsity in the predictor matrix X, which allows for even larger problems.
See Section 18.4 for more details, and a discussion of L1-regularized multi-
nomial models.
</p>
<p>4.4.5 Logistic Regression or LDA?
</p>
<p>In Section 4.3 we find that the log-posterior odds between class k and K
are linear functions of x (4.9):
</p>
<p>log
Pr(G = k|X = x)
Pr(G = K|X = x) = log
</p>
<p>πk
πK
</p>
<p>&minus; 1
2
(μk + μK)
</p>
<p>TΣ&minus;1(μk &minus; μK)
</p>
<p>+xTΣ&minus;1(μk &minus; μK)
= αk0 + α
</p>
<p>T
k x. (4.33)
</p>
<p>This linearity is a consequence of the Gaussian assumption for the class
densities, as well as the assumption of a common covariance matrix. The
linear logistic model (4.17) by construction has linear logits:
</p>
<p>log
Pr(G = k|X = x)
Pr(G = K|X = x) = βk0 + β
</p>
<p>T
k x. (4.34)
</p>
<p>It seems that the models are the same. Although they have exactly the same
form, the difference lies in the way the linear coefficients are estimated. The
logistic regression model is more general, in that it makes less assumptions.
We can write the joint density of X and G as
</p>
<p>Pr(X,G = k) = Pr(X)Pr(G = k|X), (4.35)
where Pr(X) denotes the marginal density of the inputs X. For both LDA
and logistic regression, the second term on the right has the logit-linear
form
</p>
<p>Pr(G = k|X = x) = e
βk0+β
</p>
<p>T
k x
</p>
<p>1 +
&sum;K&minus;1
</p>
<p>ℓ=1 e
βℓ0+βTℓ x
</p>
<p>, (4.36)
</p>
<p>where we have again arbitrarily chosen the last class as the reference.
The logistic regression model leaves the marginal density of X as an arbi-
</p>
<p>trary density function Pr(X), and fits the parameters of Pr(G|X) by max-
imizing the conditional likelihood&mdash;the multinomial likelihood with proba-
bilities the Pr(G = k|X). Although Pr(X) is totally ignored, we can think
of this marginal density as being estimated in a fully nonparametric and
unrestricted fashion, using the empirical distribution function which places
mass 1/N at each observation.
With LDA we fit the parameters by maximizing the full log-likelihood,
</p>
<p>based on the joint density
</p>
<p>Pr(X,G = k) = φ(X;μk,Σ)πk, (4.37)</p>
<p/>
</div>
<div class="page"><p/>
<p>128 4. Linear Methods for Classification
</p>
<p>where φ is the Gaussian density function. Standard normal theory leads
easily to the estimates μ̂k,Σ̂, and π̂k given in Section 4.3. Since the linear
parameters of the logistic form (4.33) are functions of the Gaussian param-
eters, we get their maximum-likelihood estimates by plugging in the corre-
sponding estimates. However, unlike in the conditional case, the marginal
density Pr(X) does play a role here. It is a mixture density
</p>
<p>Pr(X) =
</p>
<p>K&sum;
</p>
<p>k=1
</p>
<p>πkφ(X;μk,Σ), (4.38)
</p>
<p>which also involves the parameters.
What role can this additional component/restriction play? By relying
</p>
<p>on the additional model assumptions, we have more information about the
parameters, and hence can estimate them more efficiently (lower variance).
If in fact the true fk(x) are Gaussian, then in the worst case ignoring this
marginal part of the likelihood constitutes a loss of efficiency of about 30%
asymptotically in the error rate (Efron, 1975). Paraphrasing: with 30%
more data, the conditional likelihood will do as well.
For example, observations far from the decision boundary (which are
</p>
<p>down-weighted by logistic regression) play a role in estimating the common
covariance matrix. This is not all good news, because it also means that
LDA is not robust to gross outliers.
From the mixture formulation, it is clear that even observations without
</p>
<p>class labels have information about the parameters. Often it is expensive
to generate class labels, but unclassified observations come cheaply. By
relying on strong model assumptions, such as here, we can use both types
of information.
The marginal likelihood can be thought of as a regularizer, requiring
</p>
<p>in some sense that class densities be visible from this marginal view. For
example, if the data in a two-class logistic regression model can be per-
fectly separated by a hyperplane, the maximum likelihood estimates of the
parameters are undefined (i.e., infinite; see Exercise 4.5). The LDA coeffi-
cients for the same data will be well defined, since the marginal likelihood
will not permit these degeneracies.
In practice these assumptions are never correct, and often some of the
</p>
<p>components of X are qualitative variables. It is generally felt that logistic
regression is a safer, more robust bet than the LDA model, relying on fewer
assumptions. It is our experience that the models give very similar results,
even when LDA is used inappropriately, such as with qualitative predictors.</p>
<p/>
</div>
<div class="page"><p/>
<p>4.5 Separating Hyperplanes 129
</p>
<p>FIGURE 4.14. A toy example with two classes separable by a hyperplane. The
orange line is the least squares solution, which misclassifies one of the training
points. Also shown are two blue separating hyperplanes found by the perceptron
learning algorithm with different random starts.
</p>
<p>4.5 Separating Hyperplanes
</p>
<p>We have seen that linear discriminant analysis and logistic regression both
estimate linear decision boundaries in similar but slightly different ways.
For the rest of this chapter we describe separating hyperplane classifiers.
These procedures construct linear decision boundaries that explicitly try
to separate the data into different classes as well as possible. They provide
the basis for support vector classifiers, discussed in Chapter 12. The math-
ematical level of this section is somewhat higher than that of the previous
sections.
Figure 4.14 shows 20 data points in two classes in IR2. These data can be
</p>
<p>separated by a linear boundary. Included in the figure (blue lines) are two
of the infinitely many possible separating hyperplanes. The orange line is
the least squares solution to the problem, obtained by regressing the &minus;1/1
response Y on X (with intercept); the line is given by
</p>
<p>{x : β̂0 + β̂1x1 + β̂2x2 = 0}. (4.39)
</p>
<p>This least squares solution does not do a perfect job in separating the
points, and makes one error. This is the same boundary found by LDA,
in light of its equivalence with linear regression in the two-class case (Sec-
tion 4.3 and Exercise 4.2).
Classifiers such as (4.39), that compute a linear combination of the input
</p>
<p>features and return the sign, were called perceptrons in the engineering liter-</p>
<p/>
</div>
<div class="page"><p/>
<p>130 4. Linear Methods for Classification
</p>
<p>x0 x
</p>
<p>β&lowast;
β0 + β
</p>
<p>Tx = 0
</p>
<p>FIGURE 4.15. The linear algebra of a hyperplane (affine set).
</p>
<p>ature in the late 1950s (Rosenblatt, 1958). Perceptrons set the foundations
for the neural network models of the 1980s and 1990s.
Before we continue, let us digress slightly and review some vector algebra.
</p>
<p>Figure 4.15 depicts a hyperplane or affine set L defined by the equation
f(x) = β0 + β
</p>
<p>Tx = 0; since we are in IR2 this is a line.
Here we list some properties:
</p>
<p>1. For any two points x1 and x2 lying in L, β
T (x1 &minus; x2) = 0, and hence
</p>
<p>β&lowast; = β/||β|| is the vector normal to the surface of L.
</p>
<p>2. For any point x0 in L, β
Tx0 = &minus;β0.
</p>
<p>3. The signed distance of any point x to L is given by
</p>
<p>β&lowast;T (x&minus; x0) =
1
</p>
<p>‖β‖ (β
Tx+ β0)
</p>
<p>=
1
</p>
<p>||f &prime;(x)||f(x). (4.40)
</p>
<p>Hence f(x) is proportional to the signed distance from x to the hyperplane
defined by f(x) = 0.
</p>
<p>4.5.1 Rosenblatt&rsquo;s Perceptron Learning Algorithm
</p>
<p>The perceptron learning algorithm tries to find a separating hyperplane by
minimizing the distance of misclassified points to the decision boundary. If</p>
<p/>
</div>
<div class="page"><p/>
<p>4.5 Separating Hyperplanes 131
</p>
<p>a response yi = 1 is misclassified, then x
T
i β + β0 &lt; 0, and the opposite for
</p>
<p>a misclassified response with yi = &minus;1. The goal is to minimize
</p>
<p>D(β, β0) = &minus;
&sum;
</p>
<p>i&isin;M
</p>
<p>yi(x
T
i β + β0), (4.41)
</p>
<p>where M indexes the set of misclassified points. The quantity is non-
negative and proportional to the distance of the misclassified points to
the decision boundary defined by βTx + β0 = 0. The gradient (assuming
M is fixed) is given by
</p>
<p>&part;
D(β, β0)
</p>
<p>&part;β
= &minus;
</p>
<p>&sum;
</p>
<p>i&isin;M
</p>
<p>yixi, (4.42)
</p>
<p>&part;
D(β, β0)
</p>
<p>&part;β0
= &minus;
</p>
<p>&sum;
</p>
<p>i&isin;M
</p>
<p>yi. (4.43)
</p>
<p>The algorithm in fact uses stochastic gradient descent to minimize this
piecewise linear criterion. This means that rather than computing the sum
of the gradient contributions of each observation followed by a step in the
negative gradient direction, a step is taken after each observation is visited.
Hence the misclassified observations are visited in some sequence, and the
parameters β are updated via
</p>
<p>(
β
β0
</p>
<p>)
&larr;
</p>
<p>(
β
β0
</p>
<p>)
+ ρ
</p>
<p>(
yixi
yi
</p>
<p>)
. (4.44)
</p>
<p>Here ρ is the learning rate, which in this case can be taken to be 1 without
loss in generality. If the classes are linearly separable, it can be shown that
the algorithm converges to a separating hyperplane in a finite number of
steps (Exercise 4.6). Figure 4.14 shows two solutions to a toy problem, each
started at a different random guess.
There are a number of problems with this algorithm, summarized in
</p>
<p>Ripley (1996):
</p>
<p>&bull; When the data are separable, there are many solutions, and which
one is found depends on the starting values.
</p>
<p>&bull; The &ldquo;finite&rdquo; number of steps can be very large. The smaller the gap,
the longer the time to find it.
</p>
<p>&bull; When the data are not separable, the algorithm will not converge,
and cycles develop. The cycles can be long and therefore hard to
detect.
</p>
<p>The second problem can often be eliminated by seeking a hyperplane not
in the original space, but in a much enlarged space obtained by creating</p>
<p/>
</div>
<div class="page"><p/>
<p>132 4. Linear Methods for Classification
</p>
<p>many basis-function transformations of the original variables. This is anal-
ogous to driving the residuals in a polynomial regression problem down
to zero by making the degree sufficiently large. Perfect separation cannot
always be achieved: for example, if observations from two different classes
share the same input. It may not be desirable either, since the resulting
model is likely to be overfit and will not generalize well. We return to this
point at the end of the next section.
A rather elegant solution to the first problem is to add additional con-
</p>
<p>straints to the separating hyperplane.
</p>
<p>4.5.2 Optimal Separating Hyperplanes
</p>
<p>The optimal separating hyperplane separates the two classes and maximizes
the distance to the closest point from either class (Vapnik, 1996). Not only
does this provide a unique solution to the separating hyperplane problem,
but by maximizing the margin between the two classes on the training data,
this leads to better classification performance on test data.
We need to generalize criterion (4.41). Consider the optimization problem
</p>
<p>max
β,β0,||β||=1
</p>
<p>M
</p>
<p>subject to yi(x
T
i β + β0) &ge; M, i = 1, . . . , N.
</p>
<p>(4.45)
</p>
<p>The set of conditions ensure that all the points are at least a signed
distance M from the decision boundary defined by β and β0, and we seek
the largest such M and associated parameters. We can get rid of the ||β|| =
1 constraint by replacing the conditions with
</p>
<p>1
</p>
<p>||β||yi(x
T
i β + β0) &ge; M, (4.46)
</p>
<p>(which redefines β0) or equivalently
</p>
<p>yi(x
T
i β + β0) &ge; M ||β||. (4.47)
</p>
<p>Since for any β and β0 satisfying these inequalities, any positively scaled
multiple satisfies them too, we can arbitrarily set ||β|| = 1/M . Thus (4.45)
is equivalent to
</p>
<p>min
β,β0
</p>
<p>1
</p>
<p>2
||β||2
</p>
<p>subject to yi(x
T
i β + β0) &ge; 1, i = 1, . . . , N.
</p>
<p>(4.48)
</p>
<p>In light of (4.40), the constraints define an empty slab or margin around the
linear decision boundary of thickness 1/||β||. Hence we choose β and β0 to
maximize its thickness. This is a convex optimization problem (quadratic</p>
<p/>
</div>
<div class="page"><p/>
<p>4.5 Separating Hyperplanes 133
</p>
<p>criterion with linear inequality constraints). The Lagrange (primal) func-
tion, to be minimized w.r.t. β and β0, is
</p>
<p>LP =
1
</p>
<p>2
||β||2 &minus;
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>αi[yi(x
T
i β + β0)&minus; 1]. (4.49)
</p>
<p>Setting the derivatives to zero, we obtain:
</p>
<p>β =
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>αiyixi, (4.50)
</p>
<p>0 =
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>αiyi, (4.51)
</p>
<p>and substituting these in (4.49) we obtain the so-called Wolfe dual
</p>
<p>LD =
N&sum;
</p>
<p>i=1
</p>
<p>αi &minus;
1
</p>
<p>2
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>N&sum;
</p>
<p>k=1
</p>
<p>αiαkyiykx
T
i xk
</p>
<p>subject to αi &ge; 0 and
N&sum;
</p>
<p>i=1
</p>
<p>αiyi = 0. (4.52)
</p>
<p>The solution is obtained by maximizing LD in the positive orthant, a sim-
pler convex optimization problem, for which standard software can be used.
In addition the solution must satisfy the Karush&ndash;Kuhn&ndash;Tucker conditions,
which include (4.50), (4.51), (4.52) and
</p>
<p>αi[yi(x
T
i β + β0)&minus; 1] = 0 &forall;i. (4.53)
</p>
<p>From these we can see that
</p>
<p>&bull; if αi &gt; 0, then yi(xTi β + β0) = 1, or in other words, xi is on the
boundary of the slab;
</p>
<p>&bull; if yi(xTi β+β0) &gt; 1, xi is not on the boundary of the slab, and αi = 0.
From (4.50) we see that the solution vector β is defined in terms of a linear
combination of the support points xi&mdash;those points defined to be on the
boundary of the slab via αi &gt; 0. Figure 4.16 shows the optimal separating
hyperplane for our toy example; there are three support points. Likewise,
β0 is obtained by solving (4.53) for any of the support points.
</p>
<p>The optimal separating hyperplane produces a function f̂(x) = xT β̂+ β̂0
for classifying new observations:
</p>
<p>Ĝ(x) = signf̂(x). (4.54)
</p>
<p>Although none of the training observations fall in the margin (by con-
struction), this will not necessarily be the case for test observations. The</p>
<p/>
</div>
<div class="page"><p/>
<p>134 4. Linear Methods for Classification
</p>
<p>FIGURE 4.16. The same data as in Figure 4.14. The shaded region delineates
the maximum margin separating the two classes. There are three support points
indicated, which lie on the boundary of the margin, and the optimal separating
hyperplane (blue line) bisects the slab. Included in the figure is the boundary found
using logistic regression (red line), which is very close to the optimal separating
hyperplane (see Section 12.3.3).
</p>
<p>intuition is that a large margin on the training data will lead to good
separation on the test data.
The description of the solution in terms of support points seems to sug-
</p>
<p>gest that the optimal hyperplane focuses more on the points that count,
and is more robust to model misspecification. The LDA solution, on the
other hand, depends on all of the data, even points far away from the de-
cision boundary. Note, however, that the identification of these support
points required the use of all the data. Of course, if the classes are really
Gaussian, then LDA is optimal, and separating hyperplanes will pay a price
for focusing on the (noisier) data at the boundaries of the classes.
Included in Figure 4.16 is the logistic regression solution to this prob-
</p>
<p>lem, fit by maximum likelihood. Both solutions are similar in this case.
When a separating hyperplane exists, logistic regression will always find
it, since the log-likelihood can be driven to 0 in this case (Exercise 4.5).
The logistic regression solution shares some other qualitative features with
the separating hyperplane solution. The coefficient vector is defined by a
weighted least squares fit of a zero-mean linearized response on the input
features, and the weights are larger for points near the decision boundary
than for those further away.
When the data are not separable, there will be no feasible solution to
</p>
<p>this problem, and an alternative formulation is needed. Again one can en-
large the space using basis transformations, but this can lead to artificial</p>
<p/>
</div>
<div class="page"><p/>
<p>Exercises 135
</p>
<p>separation through over-fitting. In Chapter 12 we discuss a more attractive
alternative known as the support vector machine, which allows for overlap,
but minimizes a measure of the extent of this overlap.
</p>
<p>Bibliographic Notes
</p>
<p>Good general texts on classification include Duda et al. (2000), Hand
(1981), McLachlan (1992) and Ripley (1996). Mardia et al. (1979) have
a concise discussion of linear discriminant analysis. Michie et al. (1994)
compare a large number of popular classifiers on benchmark datasets. Lin-
ear separating hyperplanes are discussed in Vapnik (1996). Our account of
the perceptron learning algorithm follows Ripley (1996).
</p>
<p>Exercises
</p>
<p>Ex. 4.1 Show how to solve the generalized eigenvalue problem max aTBa
subject to aTWa = 1 by transforming to a standard eigenvalue problem.
</p>
<p>Ex. 4.2 Suppose we have features x &isin; IRp, a two-class response, with class
sizes N1, N2, and the target coded as &minus;N/N1, N/N2.
(a) Show that the LDA rule classifies to class 2 if
</p>
<p>xT Σ̂
&minus;1
</p>
<p>(μ̂2 &minus; μ̂1) &gt;
1
</p>
<p>2
(μ̂2 + μ̂1)
</p>
<p>T Σ̂
&minus;1
</p>
<p>(μ̂2 &minus; μ̂1)&minus; log(N2/N1),
</p>
<p>and class 1 otherwise.
</p>
<p>(b) Consider minimization of the least squares criterion
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>(yi &minus; β0 &minus; xTi β)2. (4.55)
</p>
<p>Show that the solution β̂ satisfies
[
(N &minus; 2)Σ̂+NΣ̂B
</p>
<p>]
β = N(μ̂2 &minus; μ̂1) (4.56)
</p>
<p>(after simplification), where Σ̂B =
N1N2
N2 (μ̂2 &minus; μ̂1)(μ̂2 &minus; μ̂1)T .
</p>
<p>(c) Hence show that Σ̂Bβ is in the direction (μ̂2 &minus; μ̂1) and thus
</p>
<p>β̂ &prop; Σ̂&minus;1(μ̂2 &minus; μ̂1). (4.57)
</p>
<p>Therefore the least-squares regression coefficient is identical to the
LDA coefficient, up to a scalar multiple.</p>
<p/>
</div>
<div class="page"><p/>
<p>136 4. Linear Methods for Classification
</p>
<p>(d) Show that this result holds for any (distinct) coding of the two classes.
</p>
<p>(e) Find the solution β̂0 (up to the same scalar multiple as in (c), and
</p>
<p>hence the predicted value f̂(x) = β̂0 + x
T β̂. Consider the following
</p>
<p>rule: classify to class 2 if f̂(x) &gt; 0 and class 1 otherwise. Show this is
not the same as the LDA rule unless the classes have equal numbers
of observations.
</p>
<p>(Fisher, 1936; Ripley, 1996)
</p>
<p>Ex. 4.3 Suppose we transform the original predictors X to Ŷ via linear
regression. In detail, let Ŷ = X(XTX)&minus;1XTY = XB̂, where Y is the
indicator response matrix. Similarly for any input x &isin; IRp, we get a trans-
formed vector ŷ = B̂Tx &isin; IRK . Show that LDA using Ŷ is identical to
LDA in the original space.
</p>
<p>Ex. 4.4 Consider the multilogit model with K classes (4.17). Let β be the
(p + 1)(K &minus; 1)-vector consisting of all the coefficients. Define a suitably
enlarged version of the input vector x to accommodate this vectorized co-
efficient matrix. Derive the Newton-Raphson algorithm for maximizing the
multinomial log-likelihood, and describe how you would implement this
algorithm.
</p>
<p>Ex. 4.5 Consider a two-class logistic regression problem with x &isin; IR. Char-
acterize the maximum-likelihood estimates of the slope and intercept pa-
rameter if the sample xi for the two classes are separated by a point x0 &isin; IR.
Generalize this result to (a) x &isin; IRp (see Figure 4.16), and (b) more than
two classes.
</p>
<p>Ex. 4.6 Suppose we have N points xi in IR
p in general position, with class
</p>
<p>labels yi &isin; {&minus;1, 1}. Prove that the perceptron learning algorithm converges
to a separating hyperplane in a finite number of steps:
</p>
<p>(a) Denote a hyperplane by f(x) = βT1 x + β0 = 0, or in more compact
notation βTx&lowast; = 0, where x&lowast; = (x, 1) and β = (β1, β0). Let zi =
x&lowast;i /||x&lowast;i ||. Show that separability implies the existence of a βsep such
that yiβ
</p>
<p>T
sepzi &ge; 1 &forall;i
</p>
<p>(b) Given a current βold, the perceptron algorithm identifies a point zi that
is misclassified, and produces the update βnew &larr; βold + yizi. Show
that ||βnew&minus;βsep||2 &le; ||βold&minus;βsep||2&minus;1, and hence that the algorithm
converges to a separating hyperplane in no more than ||βstart&minus;βsep||2
steps (Ripley, 1996).
</p>
<p>Ex. 4.7 Consider the criterion
</p>
<p>D&lowast;(β, β0) = &minus;
N&sum;
</p>
<p>i=1
</p>
<p>yi(x
T
i β + β0), (4.58)</p>
<p/>
</div>
<div class="page"><p/>
<p>Exercises 137
</p>
<p>a generalization of (4.41) where we sum over all the observations. Consider
minimizing D&lowast; subject to ||β|| = 1. Describe this criterion in words. Does
it solve the optimal separating hyperplane problem?
</p>
<p>Ex. 4.8 Consider the multivariate Gaussian model X|G = k &sim; N(μk,Σ),
with the additional restriction that rank{μk}K1 = L &lt; max(K &minus; 1, p).
Derive the constrained MLEs for the μk and Σ. Show that the Bayes clas-
sification rule is equivalent to classifying in the reduced subspace computed
by LDA (Hastie and Tibshirani, 1996b).
</p>
<p>Ex. 4.9 Write a computer program to perform a quadratic discriminant
analysis by fitting a separate Gaussian model per class. Try it out on the
vowel data, and compute the misclassification error for the test data. The
data can be found in the book website www-stat.stanford.edu/ElemStatLearn.</p>
<p/>
</div>
<div class="page"><p/>
<p>5
Basis Expansions and Regularization
</p>
<p>5.1 Introduction
</p>
<p>We have already made use of models linear in the input features, both for
regression and classification. Linear regression, linear discriminant analysis,
logistic regression and separating hyperplanes all rely on a linear model.
It is extremely unlikely that the true function f(X) is actually linear in
X. In regression problems, f(X) = E(Y |X) will typically be nonlinear and
nonadditive in X, and representing f(X) by a linear model is usually a con-
venient, and sometimes a necessary, approximation. Convenient because a
linear model is easy to interpret, and is the first-order Taylor approxima-
tion to f(X). Sometimes necessary, because with N small and/or p large,
a linear model might be all we are able to fit to the data without overfit-
ting. Likewise in classification, a linear, Bayes-optimal decision boundary
implies that some monotone transformation of Pr(Y = 1|X) is linear in X.
This is inevitably an approximation.
In this chapter and the next we discuss popular methods for moving
</p>
<p>beyond linearity. The core idea in this chapter is to augment/replace the
vector of inputs X with additional variables, which are transformations of
X, and then use linear models in this new space of derived input features.
Denote by hm(X) : IR
</p>
<p>p �&rarr; IR the mth transformation of X, m =
1, . . . ,M . We then model
</p>
<p>f(X) =
M&sum;
</p>
<p>m=1
</p>
<p>βmhm(X), (5.1)
</p>
<p> &copy; Springer Science+Business Media, LLC 2009 
</p>
<p>T. Hastie et al., The Elements of Statistical Learning, Second Edition, 139
</p>
<p>DOI: 10.1007/b94608_5,</p>
<p/>
</div>
<div class="page"><p/>
<p>140 5. Basis Expansions and Regularization
</p>
<p>a linear basis expansion in X. The beauty of this approach is that once the
basis functions hm have been determined, the models are linear in these
new variables, and the fitting proceeds as before.
Some simple and widely used examples of the hm are the following:
</p>
<p>&bull; hm(X) = Xm, m = 1, . . . , p recovers the original linear model.
</p>
<p>&bull; hm(X) = X2j or hm(X) = XjXk allows us to augment the inputs with
polynomial terms to achieve higher-order Taylor expansions. Note,
however, that the number of variables grows exponentially in the de-
gree of the polynomial. A full quadratic model in p variables requires
O(p2) square and cross-product terms, or more generally O(pd) for a
degree-d polynomial.
</p>
<p>&bull; hm(X) = log(Xj),
&radic;
Xj , . . . permits other nonlinear transformations
</p>
<p>of single inputs. More generally one can use similar functions involv-
ing several inputs, such as hm(X) = ||X||.
</p>
<p>&bull; hm(X) = I(Lm &le; Xk &lt; Um), an indicator for a region of Xk. By
breaking the range of Xk up into Mk such nonoverlapping regions
results in a model with a piecewise constant contribution for Xk.
</p>
<p>Sometimes the problem at hand will call for particular basis functions hm,
such as logarithms or power functions. More often, however, we use the basis
expansions as a device to achieve more flexible representations for f(X).
Polynomials are an example of the latter, although they are limited by
their global nature&mdash;tweaking the coefficients to achieve a functional form
in one region can cause the function to flap about madly in remote regions.
In this chapter we consider more useful families of piecewise-polynomials
and splines that allow for local polynomial representations. We also discuss
the wavelet bases, especially useful for modeling signals and images. These
methods produce a dictionary D consisting of typically a very large number
|D| of basis functions, far more than we can afford to fit to our data. Along
with the dictionary we require a method for controlling the complexity
of our model, using basis functions from the dictionary. There are three
common approaches:
</p>
<p>&bull; Restriction methods, where we decide before-hand to limit the class
of functions. Additivity is an example, where we assume that our
model has the form
</p>
<p>f(X) =
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>fj(Xj)
</p>
<p>=
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>Mj&sum;
</p>
<p>m=1
</p>
<p>βjmhjm(Xj). (5.2)</p>
<p/>
</div>
<div class="page"><p/>
<p>5.2 Piecewise Polynomials and Splines 141
</p>
<p>The size of the model is limited by the number of basis functions Mj
used for each component function fj .
</p>
<p>&bull; Selection methods, which adaptively scan the dictionary and include
only those basis functions hm that contribute significantly to the fit of
the model. Here the variable selection techniques discussed in Chap-
ter 3 are useful. The stagewise greedy approaches such as CART,
MARS and boosting fall into this category as well.
</p>
<p>&bull; Regularization methods where we use the entire dictionary but re-
strict the coefficients. Ridge regression is a simple example of a regu-
larization approach, while the lasso is both a regularization and selec-
tion method. Here we discuss these and more sophisticated methods
for regularization.
</p>
<p>5.2 Piecewise Polynomials and Splines
</p>
<p>We assume until Section 5.7 that X is one-dimensional. A piecewise poly-
nomial function f(X) is obtained by dividing the domain of X into contigu-
ous intervals, and representing f by a separate polynomial in each interval.
Figure 5.1 shows two simple piecewise polynomials. The first is piecewise
constant, with three basis functions:
</p>
<p>h1(X) = I(X &lt; ξ1), h2(X) = I(ξ1 &le; X &lt; ξ2), h3(X) = I(ξ2 &le; X).
</p>
<p>Since these are positive over disjoint regions, the least squares estimate of
the model f(X) =
</p>
<p>&sum;3
m=1 βmhm(X) amounts to β̂m = Ȳm, the mean of Y
</p>
<p>in the mth region.
The top right panel shows a piecewise linear fit. Three additional basis
</p>
<p>functions are needed: hm+3 = hm(X)X, m = 1, . . . , 3. Except in special
cases, we would typically prefer the third panel, which is also piecewise
linear, but restricted to be continuous at the two knots. These continu-
ity restrictions lead to linear constraints on the parameters; for example,
f(ξ&minus;1 ) = f(ξ
</p>
<p>+
1 ) implies that β1 + ξ1β4 = β2 + ξ1β5. In this case, since there
</p>
<p>are two restrictions, we expect to get back two parameters, leaving four free
parameters.
A more direct way to proceed in this case is to use a basis that incorpo-
</p>
<p>rates the constraints:
</p>
<p>h1(X) = 1, h2(X) = X, h3(X) = (X &minus; ξ1)+, h4(X) = (X &minus; ξ2)+,
</p>
<p>where t+ denotes the positive part. The function h3 is shown in the lower
right panel of Figure 5.1. We often prefer smoother functions, and these
can be achieved by increasing the order of the local polynomial. Figure 5.2
shows a series of piecewise-cubic polynomials fit to the same data, with</p>
<p/>
</div>
<div class="page"><p/>
<p>142 5. Basis Expansions and Regularization
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O O
</p>
<p>Piecewise Constant
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O O
</p>
<p>Piecewise Linear
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O O
</p>
<p>Continuous Piecewise Linear Piecewise-linear Basis Function
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;&bull; &bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>ξ1ξ1
</p>
<p>ξ1ξ1
</p>
<p>ξ2ξ2
</p>
<p>ξ2ξ2
</p>
<p>(X &minus; ξ1)+
</p>
<p>FIGURE 5.1. The top left panel shows a piecewise constant function fit to some
artificial data. The broken vertical lines indicate the positions of the two knots
ξ1 and ξ2. The blue curve represents the true function, from which the data were
generated with Gaussian noise. The remaining two panels show piecewise lin-
ear functions fit to the same data&mdash;the top right unrestricted, and the lower left
restricted to be continuous at the knots. The lower right panel shows a piecewise&ndash;
linear basis function, h3(X) = (X &minus; ξ1)+, continuous at ξ1. The black points
indicate the sample evaluations h3(xi), i = 1, . . . , N .</p>
<p/>
</div>
<div class="page"><p/>
<p>5.2 Piecewise Polynomials and Splines 143
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O O
</p>
<p>Discontinuous
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O O
</p>
<p>Continuous
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O O
</p>
<p>Continuous First Derivative
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O O
</p>
<p>Continuous Second Derivative
</p>
<p>Piecewise Cubic Polynomials
</p>
<p>ξ1ξ1
</p>
<p>ξ1ξ1
</p>
<p>ξ2ξ2
</p>
<p>ξ2ξ2
</p>
<p>FIGURE 5.2. A series of piecewise-cubic polynomials, with increasing orders of
continuity.
</p>
<p>increasing orders of continuity at the knots. The function in the lower
right panel is continuous, and has continuous first and second derivatives
at the knots. It is known as a cubic spline. Enforcing one more order of
continuity would lead to a global cubic polynomial. It is not hard to show
(Exercise 5.1) that the following basis represents a cubic spline with knots
at ξ1 and ξ2:
</p>
<p>h1(X) = 1, h3(X) = X
2, h5(X) = (X &minus; ξ1)3+,
</p>
<p>h2(X) = X, h4(X) = X
3, h6(X) = (X &minus; ξ2)3+.
</p>
<p>(5.3)
</p>
<p>There are six basis functions corresponding to a six-dimensional linear space
of functions. A quick check confirms the parameter count: (3 regions)&times;(4
parameters per region) &minus;(2 knots)&times;(3 constraints per knot)= 6.</p>
<p/>
</div>
<div class="page"><p/>
<p>144 5. Basis Expansions and Regularization
</p>
<p>More generally, an order-M spline with knots ξj , j = 1, . . . ,K is a
piecewise-polynomial of order M , and has continuous derivatives up to
order M &minus; 2. A cubic spline has M = 4. In fact the piecewise-constant
function in Figure 5.1 is an order-1 spline, while the continuous piece-
wise linear function is an order-2 spline. Likewise the general form for the
truncated-power basis set would be
</p>
<p>hj(X) = X
j&minus;1, j = 1, . . . ,M,
</p>
<p>hM+ℓ(X) = (X &minus; ξℓ)M&minus;1+ , ℓ = 1, . . . ,K.
</p>
<p>It is claimed that cubic splines are the lowest-order spline for which the
knot-discontinuity is not visible to the human eye. There is seldom any
good reason to go beyond cubic-splines, unless one is interested in smooth
derivatives. In practice the most widely used orders are M = 1, 2 and 4.
</p>
<p>These fixed-knot splines are also known as regression splines. One needs
to select the order of the spline, the number of knots and their placement.
One simple approach is to parameterize a family of splines by the number
of basis functions or degrees of freedom, and have the observations xi de-
termine the positions of the knots. For example, the expression bs(x,df=7)
in R generates a basis matrix of cubic-spline functions evaluated at the N
observations in x, with the 7&minus;3 = 41 interior knots at the appropriate per-
centiles of x (20, 40, 60 and 80th.) One can be more explicit, however; bs(x,
degree=1, knots = c(0.2, 0.4, 0.6)) generates a basis for linear splines,
with three interior knots, and returns an N &times; 4 matrix.
</p>
<p>Since the space of spline functions of a particular order and knot sequence
is a vector space, there are many equivalent bases for representing them
(just as there are for ordinary polynomials.) While the truncated power
basis is conceptually simple, it is not too attractive numerically: powers of
large numbers can lead to severe rounding problems. The B-spline basis,
described in the Appendix to this chapter, allows for efficient computations
even when the number of knots K is large.
</p>
<p>5.2.1 Natural Cubic Splines
</p>
<p>We know that the behavior of polynomials fit to data tends to be erratic
near the boundaries, and extrapolation can be dangerous. These problems
are exacerbated with splines. The polynomials fit beyond the boundary
knots behave even more wildly than the corresponding global polynomials
in that region. This can be conveniently summarized in terms of the point-
wise variance of spline functions fit by least squares (see the example in the
next section for details on these variance calculations). Figure 5.3 compares
</p>
<p>1A cubic spline with four knots is eight-dimensional. The bs() function omits by
default the constant term in the basis, since terms like this are typically included with
other terms in the model.</p>
<p/>
</div>
<div class="page"><p/>
<p>5.2 Piecewise Polynomials and Splines 145
</p>
<p>X
</p>
<p>P
o
in
</p>
<p>tw
is
</p>
<p>e
 V
</p>
<p>a
ri
a
n
c
e
s
</p>
<p>0.0 0.2 0.4 0.6 0.8 1.0
</p>
<p>0
.0
</p>
<p>0
.1
</p>
<p>0
.2
</p>
<p>0
.3
</p>
<p>0
.4
</p>
<p>0
.5
</p>
<p>0
.6
</p>
<p>&bull;
</p>
<p>&bull;&bull;
&bull;
&bull;&bull;
</p>
<p>&bull;&bull;&bull;&bull;&bull; &bull;&bull; &bull;&bull;&bull;&bull;&bull;&bull; &bull;&bull; &bull;&bull; &bull; &bull;&bull; &bull;&bull;&bull;&bull; &bull;
&bull;&bull; &bull;&bull; &bull; &bull; &bull;&bull;&bull; &bull;&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
&bull; &bull;&bull;
</p>
<p>&bull;&bull;&bull;&bull;&bull;
&bull;&bull; &bull;&bull;&bull;&bull;&bull;&bull; &bull;&bull; &bull;&bull; &bull; &bull;&bull;
</p>
<p>&bull;&bull;&bull;&bull; &bull; &bull;&bull; &bull;&bull; &bull; &bull;
&bull;&bull;
</p>
<p>&bull; &bull;&bull; &bull;&bull;
&bull;&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
&bull;
&bull;&bull;
</p>
<p>&bull;&bull;&bull;&bull;&bull;
&bull;&bull;
</p>
<p>&bull;&bull;&bull;&bull;
&bull;&bull;
</p>
<p>&bull;&bull; &bull;&bull; &bull; &bull;&bull;
</p>
<p>&bull;&bull;&bull;&bull;
&bull;
</p>
<p>&bull;&bull; &bull;
&bull; &bull; &bull; &bull;&bull;&bull;
</p>
<p>&bull;&bull; &bull;&bull;
</p>
<p>&bull;&bull;
&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;&bull; &bull; &bull;&bull;
&bull;&bull;&bull;&bull;&bull; &bull;&bull; &bull;&bull;&bull;&bull;&bull;&bull; &bull;&bull; &bull;&bull; &bull; &bull;&bull; &bull;&bull;&bull;&bull; &bull; &bull;&bull; &bull;&bull; &bull; &bull;
</p>
<p>&bull;&bull;&bull; &bull;&bull;
&bull;&bull; &bull;&bull;&bull;&bull;
</p>
<p>&bull;
</p>
<p>Global Linear
</p>
<p>Global Cubic Polynomial
</p>
<p>Cubic Spline - 2 knots
</p>
<p>Natural Cubic Spline - 6 knots
</p>
<p>FIGURE 5.3. Pointwise variance curves for four different models, with X con-
sisting of 50 points drawn at random from U [0, 1], and an assumed error model
with constant variance. The linear and cubic polynomial fits have two and four
degrees of freedom, respectively, while the cubic spline and natural cubic spline
each have six degrees of freedom. The cubic spline has two knots at 0.33 and 0.66,
while the natural spline has boundary knots at 0.1 and 0.9, and four interior knots
uniformly spaced between them.
</p>
<p>the pointwise variances for a variety of different models. The explosion of
the variance near the boundaries is clear, and inevitably is worst for cubic
splines.
A natural cubic spline adds additional constraints, namely that the func-
</p>
<p>tion is linear beyond the boundary knots. This frees up four degrees of
freedom (two constraints each in both boundary regions), which can be
spent more profitably by sprinkling more knots in the interior region. This
tradeoff is illustrated in terms of variance in Figure 5.3. There will be a
price paid in bias near the boundaries, but assuming the function is lin-
ear near the boundaries (where we have less information anyway) is often
considered reasonable.
A natural cubic spline with K knots is represented by K basis functions.
</p>
<p>One can start from a basis for cubic splines, and derive the reduced ba-
sis by imposing the boundary constraints. For example, starting from the
truncated power series basis described in Section 5.2, we arrive at (Exer-
cise 5.4):
</p>
<p>N1(X) = 1, N2(X) = X, Nk+2(X) = dk(X)&minus; dK&minus;1(X), (5.4)</p>
<p/>
</div>
<div class="page"><p/>
<p>146 5. Basis Expansions and Regularization
</p>
<p>where
</p>
<p>dk(X) =
(X &minus; ξk)3+ &minus; (X &minus; ξK)3+
</p>
<p>ξK &minus; ξk
. (5.5)
</p>
<p>Each of these basis functions can be seen to have zero second and third
derivative for X &ge; ξK .
</p>
<p>5.2.2 Example: South African Heart Disease (Continued)
</p>
<p>In Section 4.4.2 we fit linear logistic regression models to the South African
heart disease data. Here we explore nonlinearities in the functions using
natural splines. The functional form of the model is
</p>
<p>logit[Pr(chd|X)] = θ0 + h1(X1)T θ1 + h2(X2)T θ2 + &middot; &middot; &middot;+ hp(Xp)T θp, (5.6)
</p>
<p>where each of the θj are vectors of coefficients multiplying their associated
vector of natural spline basis functions hj .
</p>
<p>We use four natural spline bases for each term in the model. For example,
with X1 representing sbp, h1(X1) is a basis consisting of four basis func-
tions. This actually implies three rather than two interior knots (chosen at
uniform quantiles of sbp), plus two boundary knots at the extremes of the
data, since we exclude the constant term from each of the hj .
</p>
<p>Since famhist is a two-level factor, it is coded by a simple binary or
dummy variable, and is associated with a single coefficient in the fit of the
model.
More compactly we can combine all p vectors of basis functions (and
</p>
<p>the constant term) into one big vector h(X), and then the model is simply
h(X)T θ, with total number of parameters df = 1 +
</p>
<p>&sum;p
j=1 dfj , the sum of
</p>
<p>the parameters in each component term. Each basis function is evaluated
at each of the N samples, resulting in a N &times; df basis matrix H. At this
point the model is like any other linear logistic model, and the algorithms
described in Section 4.4.1 apply.
We carried out a backward stepwise deletion process, dropping terms
</p>
<p>from this model while preserving the group structure of each term, rather
than dropping one coefficient at a time. The AIC statistic (Section 7.5) was
used to drop terms, and all the terms remaining in the final model would
cause AIC to increase if deleted from the model (see Table 5.1). Figure 5.4
shows a plot of the final model selected by the stepwise regression. The
functions displayed are f̂j(Xj) = hj(Xj)
</p>
<p>T θ̂j for each variable Xj . The
</p>
<p>covariance matrix Cov(θ̂) = Σ is estimated by Σ̂ = (HTWH)&minus;1, where W
is the diagonal weight matrix from the logistic regression. Hence vj(Xj) =
</p>
<p>Var[f̂j(Xj)] = hj(Xj)
T Σ̂jjhj(Xj) is the pointwise variance function of f̂j ,
</p>
<p>where Cov(θ̂j) = Σ̂jj is the appropriate sub-matrix of Σ̂. The shaded region
</p>
<p>in each panel is defined by f̂j(Xj)&plusmn; 2
&radic;
</p>
<p>vj(Xj).
The AIC statistic is slightly more generous than the likelihood-ratio test
</p>
<p>(deviance test). Both sbp and obesity are included in this model, while</p>
<p/>
</div>
<div class="page"><p/>
<p>5.2 Piecewise Polynomials and Splines 147
</p>
<p>100 120 140 160 180 200 220
</p>
<p>-2
0
</p>
<p>2
4
</p>
<p>0 5 10 15 20 25 30
</p>
<p>0
2
</p>
<p>4
6
</p>
<p>8
</p>
<p>2 4 6 8 10 12 14
</p>
<p>-4
-2
</p>
<p>0
2
</p>
<p>4
</p>
<p>-4
-2
</p>
<p>0
2
</p>
<p>4
</p>
<p>Absent Present
</p>
<p>15 20 25 30 35 40 45
</p>
<p>-2
0
</p>
<p>2
4
</p>
<p>6
</p>
<p>20 30 40 50 60
</p>
<p>-6
-4
</p>
<p>-2
0
</p>
<p>2
</p>
<p>f̂
(s
b
p
)
</p>
<p>sbp
f̂
(t
o
b
a
c
c
o
)
</p>
<p>tobacco
</p>
<p>f̂
(l
d
l
)
</p>
<p>ldl
</p>
<p>f̂
(o
b
e
s
i
t
y
)
</p>
<p>obesity
</p>
<p>f̂
(a
g
e
)
</p>
<p>age
</p>
<p>f̂
(f
a
m
h
i
s
t
)
</p>
<p>famhist
</p>
<p>FIGURE 5.4. Fitted natural-spline functions for each of the terms in the final
model selected by the stepwise procedure. Included are pointwise standard-error
bands. The rug plot at the base of each figure indicates the location of each of the
sample values for that variable (jittered to break ties).</p>
<p/>
</div>
<div class="page"><p/>
<p>148 5. Basis Expansions and Regularization
</p>
<p>TABLE 5.1. Final logistic regression model, after stepwise deletion of natural
splines terms. The column labeled &ldquo;LRT&rdquo; is the likelihood-ratio test statistic when
that term is deleted from the model, and is the change in deviance from the full
model (labeled &ldquo;none&rdquo;).
</p>
<p>Terms Df Deviance AIC LRT P-value
none 458.09 502.09
sbp 4 467.16 503.16 9.076 0.059
</p>
<p>tobacco 4 470.48 506.48 12.387 0.015
ldl 4 472.39 508.39 14.307 0.006
</p>
<p>famhist 1 479.44 521.44 21.356 0.000
obesity 4 466.24 502.24 8.147 0.086
</p>
<p>age 4 481.86 517.86 23.768 0.000
</p>
<p>they were not in the linear model. The figure explains why, since their
contributions are inherently nonlinear. These effects at first may come as
a surprise, but an explanation lies in the nature of the retrospective data.
These measurements were made sometime after the patients suffered a
heart attack, and in many cases they had already benefited from a healthier
diet and lifestyle, hence the apparent increase in risk at low values for
obesity and sbp. Table 5.1 shows a summary of the selected model.
</p>
<p>5.2.3 Example: Phoneme Recognition
</p>
<p>In this example we use splines to reduce flexibility rather than increase it;
the application comes under the general heading of functional modeling. In
the top panel of Figure 5.5 are displayed a sample of 15 log-periodograms
for each of the two phonemes &ldquo;aa&rdquo; and &ldquo;ao&rdquo; measured at 256 frequencies.
The goal is to use such data to classify a spoken phoneme. These two
phonemes were chosen because they are difficult to separate.
The input feature is a vector x of length 256, which we can think of as
</p>
<p>a vector of evaluations of a function X(f) over a grid of frequencies f . In
reality there is a continuous analog signal which is a function of frequency,
and we have a sampled version of it.
The gray lines in the lower panel of Figure 5.5 show the coefficients of
</p>
<p>a linear logistic regression model fit by maximum likelihood to a training
sample of 1000 drawn from the total of 695 &ldquo;aa&rdquo;s and 1022 &ldquo;ao&rdquo;s. The
coefficients are also plotted as a function of frequency, and in fact we can
think of the model in terms of its continuous counterpart
</p>
<p>log
Pr(aa|X)
Pr(ao|X) =
</p>
<p>&int;
X(f)β(f)df, (5.7)</p>
<p/>
</div>
<div class="page"><p/>
<p>5.2 Piecewise Polynomials and Splines 149
</p>
<p>Frequency
</p>
<p>L
o
g
-p
</p>
<p>e
ri
o
d
o
g
ra
</p>
<p>m
</p>
<p>0 50 100 150 200 250
</p>
<p>0
5
</p>
<p>1
0
</p>
<p>1
5
</p>
<p>2
0
</p>
<p>2
5
</p>
<p>Phoneme Examples
</p>
<p>aa
ao
</p>
<p>Frequency
</p>
<p>L
o
g
is
</p>
<p>ti
c
 R
</p>
<p>e
g
re
</p>
<p>s
s
io
</p>
<p>n
 C
</p>
<p>o
e
ff
ic
</p>
<p>ie
n
ts
</p>
<p>0 50 100 150 200 250
</p>
<p>-0
.4
</p>
<p>-0
.2
</p>
<p>0
.0
</p>
<p>0
.2
</p>
<p>0
.4
</p>
<p>Phoneme Classification: Raw and Restricted Logistic Regression
</p>
<p>FIGURE 5.5. The top panel displays the log-periodogram as a function of fre-
quency for 15 examples each of the phonemes &ldquo;aa&rdquo; and &ldquo;ao&rdquo; sampled from a total
of 695 &ldquo;aa&rdquo;s and 1022 &ldquo;ao&rdquo;s. Each log-periodogram is measured at 256 uniformly
spaced frequencies. The lower panel shows the coefficients (as a function of fre-
quency) of a logistic regression fit to the data by maximum likelihood, using the
256 log-periodogram values as inputs. The coefficients are restricted to be smooth
in the red curve, and are unrestricted in the jagged gray curve.</p>
<p/>
</div>
<div class="page"><p/>
<p>150 5. Basis Expansions and Regularization
</p>
<p>which we approximate by
</p>
<p>256&sum;
</p>
<p>j=1
</p>
<p>X(fj)β(fj) =
</p>
<p>256&sum;
</p>
<p>j=1
</p>
<p>xjβj . (5.8)
</p>
<p>The coefficients compute a contrast functional, and will have appreciable
values in regions of frequency where the log-periodograms differ between
the two classes.
The gray curves are very rough. Since the input signals have fairly strong
</p>
<p>positive autocorrelation, this results in negative autocorrelation in the co-
efficients. In addition the sample size effectively provides only four obser-
vations per coefficient.
Applications such as this permit a natural regularization. We force the
</p>
<p>coefficients to vary smoothly as a function of frequency. The red curve in the
lower panel of Figure 5.5 shows such a smooth coefficient curve fit to these
data. We see that the lower frequencies offer the most discriminatory power.
Not only does the smoothing allow easier interpretation of the contrast, it
also produces a more accurate classifier:
</p>
<p>Raw Regularized
</p>
<p>Training error 0.080 0.185
Test error 0.255 0.158
</p>
<p>The smooth red curve was obtained through a very simple use of natural
cubic splines. We can represent the coefficient function as an expansion of
splines β(f) =
</p>
<p>&sum;M
m=1 hm(f)θm. In practice this means that β = Hθ where,
</p>
<p>H is a p &times; M basis matrix of natural cubic splines, defined on the set of
frequencies. Here we used M = 12 basis functions, with knots uniformly
placed over the integers 1, 2, . . . , 256 representing the frequencies. Since
xTβ = xTHθ, we can simply replace the input features x by their filtered
versions x&lowast; = HTx, and fit θ by linear logistic regression on the x&lowast;. The
red curve is thus β̂(f) = h(f)T θ̂.
</p>
<p>5.3 Filtering and Feature Extraction
</p>
<p>In the previous example, we constructed a p&times;M basis matrix H, and then
transformed our features x into new features x&lowast; = HTx. These filtered
versions of the features were then used as inputs into a learning procedure:
in the previous example, this was linear logistic regression.
Preprocessing of high-dimensional features is a very general and pow-
</p>
<p>erful method for improving the performance of a learning algorithm. The
preprocessing need not be linear as it was above, but can be a general</p>
<p/>
</div>
<div class="page"><p/>
<p>5.4 Smoothing Splines 151
</p>
<p>(nonlinear) function of the form x&lowast; = g(x). The derived features x&lowast; can
then be used as inputs into any (linear or nonlinear) learning procedure.
For example, for signal or image recognition a popular approach is to first
</p>
<p>transform the raw features via a wavelet transform x&lowast; = HTx (Section 5.9)
and then use the features x&lowast; as inputs into a neural network (Chapter 11).
Wavelets are effective in capturing discrete jumps or edges, and the neural
network is a powerful tool for constructing nonlinear functions of these
features for predicting the target variable. By using domain knowledge
to construct appropriate features, one can often improve upon a learning
method that has only the raw features x at its disposal.
</p>
<p>5.4 Smoothing Splines
</p>
<p>Here we discuss a spline basis method that avoids the knot selection prob-
lem completely by using a maximal set of knots. The complexity of the fit
is controlled by regularization. Consider the following problem: among all
functions f(x) with two continuous derivatives, find one that minimizes the
penalized residual sum of squares
</p>
<p>RSS(f, λ) =
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>{yi &minus; f(xi)}2 + λ
&int;
{f &prime;&prime;(t)}2dt, (5.9)
</p>
<p>where λ is a fixed smoothing parameter. The first term measures closeness
to the data, while the second term penalizes curvature in the function, and
λ establishes a tradeoff between the two. Two special cases are:
</p>
<p>λ = 0 : f can be any function that interpolates the data.
</p>
<p>λ = &infin; : the simple least squares line fit, since no second derivative can
be tolerated.
</p>
<p>These vary from very rough to very smooth, and the hope is that λ &isin; (0,&infin;)
indexes an interesting class of functions in between.
The criterion (5.9) is defined on an infinite-dimensional function space&mdash;
</p>
<p>in fact, a Sobolev space of functions for which the second term is defined.
Remarkably, it can be shown that (5.9) has an explicit, finite-dimensional,
unique minimizer which is a natural cubic spline with knots at the unique
values of the xi, i = 1, . . . , N (Exercise 5.7). At face value it seems that
the family is still over-parametrized, since there are as many as N knots,
which implies N degrees of freedom. However, the penalty term translates
to a penalty on the spline coefficients, which are shrunk some of the way
toward the linear fit.
Since the solution is a natural spline, we can write it as
</p>
<p>f(x) =
N&sum;
</p>
<p>j=1
</p>
<p>Nj(x)θj , (5.10)</p>
<p/>
</div>
<div class="page"><p/>
<p>152 5. Basis Expansions and Regularization
</p>
<p>Age
</p>
<p>R
e
</p>
<p>la
ti
v
e
</p>
<p> C
h
</p>
<p>a
n
</p>
<p>g
e
</p>
<p> i
n
</p>
<p> S
p
</p>
<p>in
a
</p>
<p>l 
B
</p>
<p>M
D
</p>
<p>10 15 20 25
</p>
<p>-0
.0
</p>
<p>5
0
</p>
<p>.0
0
</p>
<p>.0
5
</p>
<p>0
.1
</p>
<p>0
0
</p>
<p>.1
5
</p>
<p>0
.2
</p>
<p>0
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
&bull; &bull; &bull; &bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull; &bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>Male
</p>
<p>Female
</p>
<p>FIGURE 5.6. The response is the relative change in bone mineral density mea-
sured at the spine in adolescents, as a function of age. A separate smoothing spline
was fit to the males and females, with λ &asymp; 0.00022. This choice corresponds to
about 12 degrees of freedom.
</p>
<p>where the Nj(x) are an N -dimensional set of basis functions for repre-
senting this family of natural splines (Section 5.2.1 and Exercise 5.4). The
criterion thus reduces to
</p>
<p>RSS(θ, λ) = (y &minus;Nθ)T (y &minus;Nθ) + λθTΩNθ, (5.11)
</p>
<p>where {N}ij = Nj(xi) and {ΩN}jk =
&int;
N &prime;&prime;j (t)N
</p>
<p>&prime;&prime;
k (t)dt. The solution is
</p>
<p>easily seen to be
θ̂ = (NTN+ λΩN )
</p>
<p>&minus;1NTy, (5.12)
</p>
<p>a generalized ridge regression. The fitted smoothing spline is given by
</p>
<p>f̂(x) =
N&sum;
</p>
<p>j=1
</p>
<p>Nj(x)θ̂j . (5.13)
</p>
<p>Efficient computational techniques for smoothing splines are discussed in
the Appendix to this chapter.
Figure 5.6 shows a smoothing spline fit to some data on bone mineral
</p>
<p>density (BMD) in adolescents. The response is relative change in spinal
BMD over two consecutive visits, typically about one year apart. The data
are color coded by gender, and two separate curves were fit. This simple</p>
<p/>
</div>
<div class="page"><p/>
<p>5.4 Smoothing Splines 153
</p>
<p>summary reinforces the evidence in the data that the growth spurt for
females precedes that for males by about two years. In both cases the
smoothing parameter λ was approximately 0.00022; this choice is discussed
in the next section.
</p>
<p>5.4.1 Degrees of Freedom and Smoother Matrices
</p>
<p>We have not yet indicated how λ is chosen for the smoothing spline. Later
in this chapter we describe automatic methods using techniques such as
cross-validation. In this section we discuss intuitive ways of prespecifying
the amount of smoothing.
A smoothing spline with prechosen λ is an example of a linear smoother
</p>
<p>(as in linear operator). This is because the estimated parameters in (5.12)
</p>
<p>are a linear combination of the yi. Denote by f̂ the N -vector of fitted values
f̂(xi) at the training predictors xi. Then
</p>
<p>f̂ = N(NTN+ λΩN )
&minus;1NTy
</p>
<p>= Sλy. (5.14)
</p>
<p>Again the fit is linear in y, and the finite linear operator Sλ is known as
the smoother matrix. One consequence of this linearity is that the recipe
for producing f̂ from y does not depend on y itself; Sλ depends only on
the xi and λ.
Linear operators are familiar in more traditional least squares fitting as
</p>
<p>well. Suppose Bξ is a N &times; M matrix of M cubic-spline basis functions
evaluated at the N training points xi, with knot sequence ξ, and M ≪ N .
Then the vector of fitted spline values is given by
</p>
<p>f̂ = Bξ(B
T
ξ Bξ)
</p>
<p>&minus;1BTξ y
</p>
<p>= Hξy. (5.15)
</p>
<p>Here the linear operator Hξ is a projection operator, also known as the hat
matrix in statistics. There are some important similarities and differences
between Hξ and Sλ:
</p>
<p>&bull; Both are symmetric, positive semidefinite matrices.
</p>
<p>&bull; HξHξ = Hξ (idempotent), while SλSλ � Sλ, meaning that the right-
hand side exceeds the left-hand side by a positive semidefinite matrix.
This is a consequence of the shrinking nature of Sλ, which we discuss
further below.
</p>
<p>&bull; Hξ has rank M , while Sλ has rank N .
The expression M = trace(Hξ) gives the dimension of the projection space,
which is also the number of basis functions, and hence the number of pa-
rameters involved in the fit. By analogy we define the effective degrees of</p>
<p/>
</div>
<div class="page"><p/>
<p>154 5. Basis Expansions and Regularization
</p>
<p>freedom of a smoothing spline to be
</p>
<p>dfλ = trace(Sλ), (5.16)
</p>
<p>the sum of the diagonal elements of Sλ. This very useful definition allows
us a more intuitive way to parameterize the smoothing spline, and indeed
many other smoothers as well, in a consistent fashion. For example, in Fig-
ure 5.6 we specified dfλ = 12 for each of the curves, and the corresponding
λ &asymp; 0.00022 was derived numerically by solving trace(Sλ) = 12. There are
many arguments supporting this definition of degrees of freedom, and we
cover some of them here.
Since Sλ is symmetric (and positive semidefinite), it has a real eigen-
</p>
<p>decomposition. Before we proceed, it is convenient to rewrite Sλ in the
Reinsch form
</p>
<p>Sλ = (I+ λK)
&minus;1, (5.17)
</p>
<p>where K does not depend on λ (Exercise 5.9). Since f̂ = Sλy solves
</p>
<p>min
f
</p>
<p>(y &minus; f)T (y &minus; f) + λfTKf , (5.18)
</p>
<p>K is known as the penalty matrix, and indeed a quadratic form in K has
a representation in terms of a weighted sum of squared (divided) second
differences. The eigen-decomposition of Sλ is
</p>
<p>Sλ =
N&sum;
</p>
<p>k=1
</p>
<p>ρk(λ)uku
T
k (5.19)
</p>
<p>with
</p>
<p>ρk(λ) =
1
</p>
<p>1 + λdk
, (5.20)
</p>
<p>and dk the corresponding eigenvalue of K. Figure 5.7 (top) shows the re-
sults of applying a cubic smoothing spline to some air pollution data (128
observations). Two fits are given: a smoother fit corresponding to a larger
penalty λ and a rougher fit for a smaller penalty. The lower panels repre-
sent the eigenvalues (lower left) and some eigenvectors (lower right) of the
corresponding smoother matrices. Some of the highlights of the eigenrep-
resentation are the following:
</p>
<p>&bull; The eigenvectors are not affected by changes in λ, and hence the whole
family of smoothing splines (for a particular sequence x) indexed by
λ have the same eigenvectors.
</p>
<p>&bull; Sλy =
&sum;N
</p>
<p>k=1 ukρk(λ)〈uk,y〉, and hence the smoothing spline oper-
ates by decomposing y w.r.t. the (complete) basis {uk}, and differ-
entially shrinking the contributions using ρk(λ). This is to be con-
trasted with a basis-regression method, where the components are</p>
<p/>
</div>
<div class="page"><p/>
<p>5.4 Smoothing Splines 155
</p>
<p>Daggot Pressure Gradient
</p>
<p>O
z
o
</p>
<p>n
e
</p>
<p> C
o
</p>
<p>n
c
e
</p>
<p>n
tr
</p>
<p>a
ti
o
</p>
<p>n
</p>
<p>-50 0 50 100
</p>
<p>0
1
</p>
<p>0
2
</p>
<p>0
3
</p>
<p>0
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
&bull;
</p>
<p>&bull;
&bull;&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>Order
</p>
<p>E
ig
</p>
<p>e
n
v
a
lu
</p>
<p>e
s
</p>
<p>5 10 15 20 25
</p>
<p>-0
.2
</p>
<p>0
.0
</p>
<p>0
.2
</p>
<p>0
.4
</p>
<p>0
.6
</p>
<p>0
.8
</p>
<p>1
.0
</p>
<p>1
.2
</p>
<p>&bull; &bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull;
</p>
<p>&bull; &bull; &bull; &bull; &bull; &bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull;
</p>
<p>df=5
</p>
<p>df=11
</p>
<p>-50 0 50 100 -50 0 50 100
</p>
<p>FIGURE 5.7. (Top:) Smoothing spline fit of ozone concentration versus Daggot
pressure gradient. The two fits correspond to different values of the smoothing
parameter, chosen to achieve five and eleven effective degrees of freedom, defined
by dfλ = trace(Sλ). (Lower left:) First 25 eigenvalues for the two smoothing-spline
matrices. The first two are exactly 1, and all are &ge; 0. (Lower right:) Third to
sixth eigenvectors of the spline smoother matrices. In each case, uk is plotted
against x, and as such is viewed as a function of x. The rug at the base of the
plots indicate the occurrence of data points. The damped functions represent the
smoothed versions of these functions (using the 5 df smoother).</p>
<p/>
</div>
<div class="page"><p/>
<p>156 5. Basis Expansions and Regularization
</p>
<p>either left alone, or shrunk to zero&mdash;that is, a projection matrix such
as Hξ above has M eigenvalues equal to 1, and the rest are 0. For
this reason smoothing splines are referred to as shrinking smoothers,
while regression splines are projection smoothers (see Figure 3.17 on
page 80).
</p>
<p>&bull; The sequence of uk, ordered by decreasing ρk(λ), appear to increase
in complexity. Indeed, they have the zero-crossing behavior of polyno-
mials of increasing degree. Since Sλuk = ρk(λ)uk, we see how each of
the eigenvectors themselves are shrunk by the smoothing spline: the
higher the complexity, the more they are shrunk. If the domain of X
is periodic, then the uk are sines and cosines at different frequencies.
</p>
<p>&bull; The first two eigenvalues are always one, and they correspond to the
two-dimensional eigenspace of functions linear in x (Exercise 5.11),
which are never shrunk.
</p>
<p>&bull; The eigenvalues ρk(λ) = 1/(1 + λdk) are an inverse function of the
eigenvalues dk of the penalty matrix K, moderated by λ; λ controls
the rate at which the ρk(λ) decrease to zero. d1 = d2 = 0 and again
linear functions are not penalized.
</p>
<p>&bull; One can reparametrize the smoothing spline using the basis vectors
uk (the Demmler&ndash;Reinsch basis). In this case the smoothing spline
solves
</p>
<p>min
θ
</p>
<p>‖y &minus;Uθ‖2 + λθTDθ, (5.21)
where U has columns uk and D is a diagonal matrix with elements
dk.
</p>
<p>&bull; dfλ = trace(Sλ) =
&sum;N
</p>
<p>k=1 ρk(λ). For projection smoothers, all the
eigenvalues are 1, each one corresponding to a dimension of the pro-
jection subspace.
</p>
<p>Figure 5.8 depicts a smoothing spline matrix, with the rows ordered with
x. The banded nature of this representation suggests that a smoothing
spline is a local fitting method, much like the locally weighted regression
procedures in Chapter 6. The right panel shows in detail selected rows of
S, which we call the equivalent kernels. As λ &rarr; 0, dfλ &rarr; N , and Sλ &rarr; I,
the N -dimensional identity matrix. As λ &rarr; &infin;, dfλ &rarr; 2, and Sλ &rarr; H, the
hat matrix for linear regression on x.
</p>
<p>5.5 Automatic Selection of the Smoothing
Parameters
</p>
<p>The smoothing parameters for regression splines encompass the degree of
the splines, and the number and placement of the knots. For smoothing</p>
<p/>
</div>
<div class="page"><p/>
<p>5.5 Automatic Selection of the Smoothing Parameters 157
</p>
<p>115
</p>
<p>100
</p>
<p>75
</p>
<p>50
</p>
<p>25
</p>
<p>12
</p>
<p>Smoother Matrix
</p>
<p>&bull;&bull;&bull;&bull; &bull; &bull; &bull;&bull;&bull;&bull;&bull;&bull; &bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;
</p>
<p>&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;
</p>
<p>&bull;&bull;&bull;
&bull;&bull;&bull;
</p>
<p>&bull;&bull;&bull;
&bull;&bull;&bull;
</p>
<p>&bull;&bull;&bull;&bull;
&bull;&bull;&bull;
</p>
<p>&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull; &bull;&bull; &bull;
&bull;
</p>
<p>Row 115
</p>
<p>&bull;&bull;&bull;&bull; &bull; &bull; &bull;&bull;&bull;&bull;&bull;&bull; &bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;
</p>
<p>&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;
</p>
<p>&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;
</p>
<p>&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;
</p>
<p>&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull; &bull;&bull;
&bull;
</p>
<p>&bull;
</p>
<p>Row 100
</p>
<p>&bull;&bull;&bull;&bull; &bull; &bull; &bull;&bull;&bull;&bull;&bull;&bull; &bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
</p>
<p>&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;
</p>
<p>&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;
</p>
<p>&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull; &bull;&bull; &bull;
</p>
<p>&bull;
</p>
<p>Row 75
</p>
<p>&bull;&bull;&bull;&bull; &bull; &bull;
&bull;&bull;&bull;&bull;&bull;&bull; &bull;
</p>
<p>&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;
</p>
<p>&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;
</p>
<p>&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;
</p>
<p>&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull; &bull;&bull; &bull; &bull;
</p>
<p>Row 50
</p>
<p>&bull;&bull;&bull;&bull;
&bull;
</p>
<p>&bull;
&bull;&bull;&bull;
</p>
<p>&bull;&bull;&bull;
&bull;&bull;&bull;&bull;
</p>
<p>&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull; &bull;&bull; &bull; &bull;
</p>
<p>Row 25
</p>
<p>&bull;&bull;&bull;&bull; &bull; &bull; &bull;&bull;&bull;&bull;&bull;&bull; &bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull; &bull;&bull; &bull; &bull;
</p>
<p>Row 12
</p>
<p>Equivalent Kernels
</p>
<p>FIGURE 5.8. The smoother matrix for a smoothing spline is nearly banded,
indicating an equivalent kernel with local support. The left panel represents the
elements of S as an image. The right panel shows the equivalent kernel or weight-
ing function in detail for the indicated rows.</p>
<p/>
</div>
<div class="page"><p/>
<p>158 5. Basis Expansions and Regularization
</p>
<p>splines, we have only the penalty parameter λ to select, since the knots are
at all the unique training X&rsquo;s, and cubic degree is almost always used in
practice.
Selecting the placement and number of knots for regression splines can be
</p>
<p>a combinatorially complex task, unless some simplifications are enforced.
The MARS procedure in Chapter 9 uses a greedy algorithm with some
additional approximations to achieve a practical compromise. We will not
discuss this further here.
</p>
<p>5.5.1 Fixing the Degrees of Freedom
</p>
<p>Since dfλ = trace(Sλ) is monotone in λ for smoothing splines, we can in-
vert the relationship and specify λ by fixing df. In practice this can be
achieved by simple numerical methods. So, for example, in R one can use
smooth.spline(x,y,df=6) to specify the amount of smoothing. This encour-
ages a more traditional mode of model selection, where we might try a cou-
ple of different values of df, and select one based on approximate F -tests,
residual plots and other more subjective criteria. Using df in this way pro-
vides a uniform approach to compare many different smoothing methods.
It is particularly useful in generalized additive models (Chapter 9), where
several smoothing methods can be simultaneously used in one model.
</p>
<p>5.5.2 The Bias&ndash;Variance Tradeoff
</p>
<p>Figure 5.9 shows the effect of the choice of dfλ when using a smoothing
spline on a simple example:
</p>
<p>Y = f(X) + ε,
</p>
<p>f(X) =
sin(12(X + 0.2))
</p>
<p>X + 0.2
,
</p>
<p>(5.22)
</p>
<p>with X &sim; U [0, 1] and ε &sim; N(0, 1). Our training sample consists of N = 100
pairs xi, yi drawn independently from this model.
The fitted splines for three different values of dfλ are shown. The yellow
</p>
<p>shaded region in the figure represents the pointwise standard error of f̂λ,
that is, we have shaded the region between f̂λ(x) &plusmn; 2 &middot; se(f̂λ(x)). Since
f̂ = Sλy,
</p>
<p>Cov(f̂) = SλCov(y)S
T
λ
</p>
<p>= SλS
T
λ . (5.23)
</p>
<p>The diagonal contains the pointwise variances at the training xi. The bias
is given by
</p>
<p>Bias(f̂) = f &minus; E(f̂)
= f &minus; Sλf , (5.24)</p>
<p/>
</div>
<div class="page"><p/>
<p>5.5 Automatic Selection of the Smoothing Parameters 159
</p>
<p>6 8 10 12 14
</p>
<p>1
.0
</p>
<p>1
.1
</p>
<p>1
.2
</p>
<p>1
.3
</p>
<p>1
.4
</p>
<p>1
.5
</p>
<p>0.0 0.2 0.4 0.6 0.8 1.0
</p>
<p>&minus;
4
</p>
<p>&minus;
2
</p>
<p>0
2
</p>
<p>y
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
OO
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>OO
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>OO
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>OO
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>OOO
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
OO
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>OO
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>OO
</p>
<p>O
</p>
<p>O
</p>
<p>0.0 0.2 0.4 0.6 0.8 1.0
</p>
<p>&minus;
4
</p>
<p>&minus;
2
</p>
<p>0
2
</p>
<p>y
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
OO
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>OO
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>OO
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>OO
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>OOO
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
OO
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>OO
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>OO
</p>
<p>O
</p>
<p>O
</p>
<p>0.0 0.2 0.4 0.6 0.8 1.0
</p>
<p>&minus;
4
</p>
<p>&minus;
2
</p>
<p>0
2
</p>
<p>y
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
OO
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>OO
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>OO
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>OO
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>OOO
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
OO
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>OO
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>OO
</p>
<p>O
</p>
<p>O
</p>
<p>EPE
CV
</p>
<p>XX
</p>
<p>X
</p>
<p>dfλ = 5
</p>
<p>dfλ = 9 dfλ = 15
</p>
<p>dfλ
</p>
<p>Cross-Validation
</p>
<p>E
P
E
(λ
</p>
<p>)
a
n
d
</p>
<p>C
V
(λ
</p>
<p>)
</p>
<p>FIGURE 5.9. The top left panel shows the EPE(λ) and CV(λ) curves for a
realization from a nonlinear additive error model (5.22). The remaining panels
show the data, the true functions (in purple), and the fitted curves (in green) with
yellow shaded &plusmn;2&times; standard error bands, for three different values of dfλ.</p>
<p/>
</div>
<div class="page"><p/>
<p>160 5. Basis Expansions and Regularization
</p>
<p>where f is the (unknown) vector of evaluations of the true f at the training
X&rsquo;s. The expectations and variances are with respect to repeated draws
of samples of size N = 100 from the model (5.22). In a similar fashion
</p>
<p>Var(f̂λ(x0)) and Bias(f̂λ(x0)) can be computed at any point x0 (Exer-
cise 5.10). The three fits displayed in the figure give a visual demonstration
of the bias-variance tradeoff associated with selecting the smoothing
parameter.
</p>
<p>dfλ = 5: The spline under fits, and clearly trims down the hills and fills in
the valleys. This leads to a bias that is most dramatic in regions of
high curvature. The standard error band is very narrow, so we esti-
mate a badly biased version of the true function with great reliability!
</p>
<p>dfλ = 9: Here the fitted function is close to the true function, although a
slight amount of bias seems evident. The variance has not increased
appreciably.
</p>
<p>dfλ = 15: The fitted function is somewhat wiggly, but close to the true
function. The wiggliness also accounts for the increased width of the
standard error bands&mdash;the curve is starting to follow some individual
points too closely.
</p>
<p>Note that in these figures we are seeing a single realization of data and
hence fitted spline f̂ in each case, while the bias involves an expectation
E(f̂). We leave it as an exercise (5.10) to compute similar figures where the
bias is shown as well. The middle curve seems &ldquo;just right,&rdquo; in that it has
achieved a good compromise between bias and variance.
The integrated squared prediction error (EPE) combines both bias and
</p>
<p>variance in a single summary:
</p>
<p>EPE(f̂λ) = E(Y &minus; f̂λ(X))2
</p>
<p>= Var(Y ) + E
[
Bias2(f̂λ(X)) + Var(f̂λ(X))
</p>
<p>]
</p>
<p>= σ2 +MSE(f̂λ). (5.25)
</p>
<p>Note that this is averaged both over the training sample (giving rise to f̂λ),
and the values of the (independently chosen) prediction points (X,Y ). EPE
is a natural quantity of interest, and does create a tradeoff between bias
and variance. The blue points in the top left panel of Figure 5.9 suggest
that dfλ = 9 is spot on!
Since we don&rsquo;t know the true function, we do not have access to EPE, and
</p>
<p>need an estimate. This topic is discussed in some detail in Chapter 7, and
techniques such as K-fold cross-validation, GCV and Cp are all in common
use. In Figure 5.9 we include the N -fold (leave-one-out) cross-validation
curve:</p>
<p/>
</div>
<div class="page"><p/>
<p>5.6 Nonparametric Logistic Regression 161
</p>
<p>CV(f̂λ) =
1
</p>
<p>N
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>(yi &minus; f̂ (&minus;i)λ (xi))2 (5.26)
</p>
<p>=
1
</p>
<p>N
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>(
yi &minus; f̂λ(xi)
1&minus; Sλ(i, i)
</p>
<p>)2
, (5.27)
</p>
<p>which can (remarkably) be computed for each value of λ from the original
fitted values and the diagonal elements Sλ(i, i) of Sλ (Exercise 5.13).
</p>
<p>The EPE and CV curves have a similar shape, but the entire CV curve
is above the EPE curve. For some realizations this is reversed, and overall
the CV curve is approximately unbiased as an estimate of the EPE curve.
</p>
<p>5.6 Nonparametric Logistic Regression
</p>
<p>The smoothing spline problem (5.9) in Section 5.4 is posed in a regression
setting. It is typically straightforward to transfer this technology to other
domains. Here we consider logistic regression with a single quantitative
input X. The model is
</p>
<p>log
Pr(Y = 1|X = x)
Pr(Y = 0|X = x) = f(x), (5.28)
</p>
<p>which implies
</p>
<p>Pr(Y = 1|X = x) = e
f(x)
</p>
<p>1 + ef(x)
. (5.29)
</p>
<p>Fitting f(x) in a smooth fashion leads to a smooth estimate of the condi-
tional probability Pr(Y = 1|x), which can be used for classification or risk
scoring.
We construct the penalized log-likelihood criterion
</p>
<p>ℓ(f ;λ) =
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>[yi log p(xi) + (1&minus; yi) log(1&minus; p(xi))]&minus;
1
</p>
<p>2
λ
</p>
<p>&int;
{f &prime;&prime;(t)}2dt
</p>
<p>=
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>[
yif(xi)&minus; log(1 + ef(xi))
</p>
<p>]
&minus; 1
</p>
<p>2
λ
</p>
<p>&int;
{f &prime;&prime;(t)}2dt, (5.30)
</p>
<p>where we have abbreviated p(x) = Pr(Y = 1|x). The first term in this ex-
pression is the log-likelihood based on the binomial distribution (c.f. Chap-
ter 4, page 120). Arguments similar to those used in Section 5.4 show that
the optimal f is a finite-dimensional natural spline with knots at the unique</p>
<p/>
</div>
<div class="page"><p/>
<p>162 5. Basis Expansions and Regularization
</p>
<p>values of x. This means that we can represent f(x) =
&sum;N
</p>
<p>j=1 Nj(x)θj . We
compute the first and second derivatives
</p>
<p>&part;ℓ(θ)
</p>
<p>&part;θ
= NT (y &minus; p)&minus; λΩθ, (5.31)
</p>
<p>&part;2ℓ(θ)
</p>
<p>&part;θ&part;θT
= &minus;NTWN&minus; λΩ, (5.32)
</p>
<p>where p is the N -vector with elements p(xi), and W is a diagonal matrix
of weights p(xi)(1&minus; p(xi)). The first derivative (5.31) is nonlinear in θ, so
we need to use an iterative algorithm as in Section 4.4.1. Using Newton&ndash;
Raphson as in (4.23) and (4.26) for linear logistic regression, the update
equation can be written
</p>
<p>θnew = (NTWN+ λΩ)&minus;1NTW
(
Nθold +W&minus;1(y &minus; p)
</p>
<p>)
</p>
<p>= (NTWN+ λΩ)&minus;1NTWz. (5.33)
</p>
<p>We can also express this update in terms of the fitted values
</p>
<p>fnew = N(NTWN+ λΩ)&minus;1NTW
(
fold +W&minus;1(y &minus; p)
</p>
<p>)
</p>
<p>= Sλ,wz. (5.34)
</p>
<p>Referring back to (5.12) and (5.14), we see that the update fits a weighted
smoothing spline to the working response z (Exercise 5.12).
The form of (5.34) is suggestive. It is tempting to replace Sλ,w by any
</p>
<p>nonparametric (weighted) regression operator, and obtain general fami-
lies of nonparametric logistic regression models. Although here x is one-
dimensional, this procedure generalizes naturally to higher-dimensional x.
These extensions are at the heart of generalized additive models, which we
pursue in Chapter 9.
</p>
<p>5.7 Multidimensional Splines
</p>
<p>So far we have focused on one-dimensional spline models. Each of the ap-
proaches have multidimensional analogs. Suppose X &isin; IR2, and we have
a basis of functions h1k(X1), k = 1, . . . ,M1 for representing functions of
coordinate X1, and likewise a set of M2 functions h2k(X2) for coordinate
X2. Then the M1 &times;M2 dimensional tensor product basis defined by
</p>
<p>gjk(X) = h1j(X1)h2k(X2), j = 1, . . . ,M1, k = 1, . . . ,M2 (5.35)
</p>
<p>can be used for representing a two-dimensional function:
</p>
<p>g(X) =
</p>
<p>M1&sum;
</p>
<p>j=1
</p>
<p>M2&sum;
</p>
<p>k=1
</p>
<p>θjkgjk(X). (5.36)</p>
<p/>
</div>
<div class="page"><p/>
<p>5.7 Multidimensional Splines 163
</p>
<p>FIGURE 5.10. A tensor product basis of B-splines, showing some selected pairs.
Each two-dimensional function is the tensor product of the corresponding one
dimensional marginals.
</p>
<p>Figure 5.10 illustrates a tensor product basis using B-splines. The coeffi-
cients can be fit by least squares, as before. This can be generalized to d
dimensions, but note that the dimension of the basis grows exponentially
fast&mdash;yet another manifestation of the curse of dimensionality. The MARS
procedure discussed in Chapter 9 is a greedy forward algorithm for includ-
ing only those tensor products that are deemed necessary by least squares.
</p>
<p>Figure 5.11 illustrates the difference between additive and tensor product
(natural) splines on the simulated classification example from Chapter 2.
A logistic regression model logit[Pr(T |x)] = h(x)T θ is fit to the binary re-
sponse, and the estimated decision boundary is the contour h(x)T θ̂ = 0.
The tensor product basis can achieve more flexibility at the decision bound-
ary, but introduces some spurious structure along the way.</p>
<p/>
</div>
<div class="page"><p/>
<p>164 5. Basis Expansions and Regularization
</p>
<p>Additive Natural Cubic Splines - 4 df each
</p>
<p>.. .
</p>
<p>. . .. . . .
</p>
<p>. . . . .. . . . . .
</p>
<p>. . . . . . .. . . . . . .
</p>
<p>. . . . . . . .. . . . . . . . .
</p>
<p>. . . . . . . . . .. . . . . . . . . . .
</p>
<p>. . . . . . . . . . . .. . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . .. . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . .. . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . .. . . . . . . . . . . . . .
. . . . . . . . . . . . .. . . . . . . . . . . . .
. . . . . . . . . . . . .. . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . .. . . . . . . . . . . .
. . . . . . . . . . .. . . . . . . . . . .
. . . . . . . . . . .. . . . . . . . . .
</p>
<p>. . . . . . . . . .. . . . . . . . . .
. . . . . . . . .. . . . . . . . .
. . . . . . . . .. . . . . . . .
</p>
<p>. . . . . . . .. . . . . . . .
. . . . . . .. . . . . . .
. . . . . . .. . . . . .
</p>
<p>. . . . . .. . . . .
. . . . .. . . . .
</p>
<p>. . . .. . . .
</p>
<p>. . . .. . .
. . .. . .
</p>
<p>. .. .
</p>
<p>. ..
.
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>Training Error: 0.23
</p>
<p>Test Error:       0.28
</p>
<p>Bayes Error:    0.21
</p>
<p>Natural Cubic Splines - Tensor Product - 4 df each
</p>
<p>. . . . . . . . . . . . .. . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . .. . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . .. . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . .. . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . .. . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . .. . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . .. . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . .. . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . .. . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . .. . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . .. . . . . . . . . . . . .
. . . . . . . . . . . .. . . . . . . . . . . .
. . . . . . . . . . . .. . . . . . . . . . . .
. . . . . . . . . . . .. . . . . . . . . . .
</p>
<p>. . . . . . . . . . .. . . . . . . . . . .
. . . . . . . . . .. . . . . . . . .
</p>
<p>. . . . . . . .. . . . .
.
</p>
<p>. . . . . .. . . . . . . . . .
. . . . . . . . . . . .. . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . .. . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . .. . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . .
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>Training Error: 0.230
</p>
<p>Test Error:       0.282
</p>
<p>Bayes Error:    0.210
</p>
<p>FIGURE 5.11. The simulation example of Figure 2.1. The upper panel shows the
decision boundary of an additive logistic regression model, using natural splines
in each of the two coordinates (total df = 1 + (4 &minus; 1) + (4 &minus; 1) = 7). The lower
panel shows the results of using a tensor product of natural spline bases in each
coordinate (total df = 4 &times; 4 = 16). The broken purple boundary is the Bayes
decision boundary for this problem.</p>
<p/>
</div>
<div class="page"><p/>
<p>5.7 Multidimensional Splines 165
</p>
<p>One-dimensional smoothing splines (via regularization) generalize to high-
er dimensions as well. Suppose we have pairs yi, xi with xi &isin; IRd, and we
seek a d-dimensional regression function f(x). The idea is to set up the
problem
</p>
<p>min
f
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>{yi &minus; f(xi)}2 + λJ [f ], (5.37)
</p>
<p>where J is an appropriate penalty functional for stabilizing a function f in
IRd. For example, a natural generalization of the one-dimensional roughness
penalty (5.9) for functions on IR2 is
</p>
<p>J [f ] =
</p>
<p>&int; &int;
</p>
<p>IR2
</p>
<p>[(&part;2f(x)
&part;x21
</p>
<p>)2
+2
</p>
<p>(
&part;2f(x)
</p>
<p>&part;x1&part;x2
</p>
<p>)2
+
</p>
<p>(
&part;2f(x)
</p>
<p>&part;x22
</p>
<p>)2]
dx1dx2. (5.38)
</p>
<p>Optimizing (5.37) with this penalty leads to a smooth two-dimensional
surface, known as a thin-plate spline. It shares many properties with the
one-dimensional cubic smoothing spline:
</p>
<p>&bull; as λ &rarr; 0, the solution approaches an interpolating function [the one
with smallest penalty (5.38)];
</p>
<p>&bull; as λ &rarr; &infin;, the solution approaches the least squares plane;
</p>
<p>&bull; for intermediate values of λ, the solution can be represented as a
linear expansion of basis functions, whose coefficients are obtained
by a form of generalized ridge regression.
</p>
<p>The solution has the form
</p>
<p>f(x) = β0 + β
Tx+
</p>
<p>N&sum;
</p>
<p>j=1
</p>
<p>αjhj(x), (5.39)
</p>
<p>where hj(x) = ||x &minus; xj ||2 log ||x &minus; xj ||. These hj are examples of radial
basis functions, which are discussed in more detail in the next section. The
coefficients are found by plugging (5.39) into (5.37), which reduces to a
finite-dimensional penalized least squares problem. For the penalty to be
finite, the coefficients αj have to satisfy a set of linear constraints; see
Exercise 5.14.
Thin-plate splines are defined more generally for arbitrary dimension d,
</p>
<p>for which an appropriately more general J is used.
There are a number of hybrid approaches that are popular in practice,
</p>
<p>both for computational and conceptual simplicity. Unlike one-dimensional
smoothing splines, the computational complexity for thin-plate splines is
O(N3), since there is not in general any sparse structure that can be ex-
ploited. However, as with univariate smoothing splines, we can get away
with substantially less than the N knots prescribed by the solution (5.39).</p>
<p/>
</div>
<div class="page"><p/>
<p>166 5. Basis Expansions and Regularization
</p>
<p>125
</p>
<p>130
</p>
<p>135
</p>
<p>140
</p>
<p>145
</p>
<p>150
</p>
<p>155
</p>
<p>15
</p>
<p>20
</p>
<p>25
</p>
<p>30
</p>
<p>35
</p>
<p>40
</p>
<p>45
</p>
<p>20 30 40 50 60
</p>
<p>Age
</p>
<p>O
b
</p>
<p>e
s
it
y
</p>
<p>Systolic Blood Pressure
</p>
<p>120
</p>
<p>125
</p>
<p>130
</p>
<p>135
</p>
<p>140
</p>
<p>145
</p>
<p>150
</p>
<p>155
</p>
<p>160
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
&bull;
</p>
<p>&bull; &bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull; &bull; &bull; &bull; &bull;
</p>
<p>&bull; &bull; &bull; &bull; &bull; &bull; &bull;
</p>
<p>&bull; &bull; &bull; &bull; &bull; &bull; &bull;
</p>
<p>&bull; &bull; &bull; &bull; &bull; &bull;
</p>
<p>&bull; &bull; &bull; &bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull; &bull; &bull; &bull;
</p>
<p>&bull; &bull; &bull; &bull; &bull; &bull;
</p>
<p>&bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull;
</p>
<p>FIGURE 5.12. A thin-plate spline fit to the heart disease data, displayed as a
contour plot. The response is systolic blood pressure, modeled as a function
of age and obesity. The data points are indicated, as well as the lattice of points
used as knots. Care should be taken to use knots from the lattice inside the convex
hull of the data (red), and ignore those outside (green).
</p>
<p>In practice, it is usually sufficient to work with a lattice of knots covering
the domain. The penalty is computed for the reduced expansion just as
before. Using K knots reduces the computations to O(NK2 + K3). Fig-
ure 5.12 shows the result of fitting a thin-plate spline to some heart disease
risk factors, representing the surface as a contour plot. Indicated are the
location of the input features, as well as the knots used in the fit. Note that
λ was specified via dfλ = trace(Sλ) = 15.
More generally one can represent f &isin; IRd as an expansion in any arbi-
</p>
<p>trarily large collection of basis functions, and control the complexity by ap-
plying a regularizer such as (5.38). For example, we could construct a basis
by forming the tensor products of all pairs of univariate smoothing-spline
basis functions as in (5.35), using, for example, the univariate B-splines
recommended in Section 5.9.2 as ingredients. This leads to an exponential</p>
<p/>
</div>
<div class="page"><p/>
<p>5.8 Regularization and Reproducing Kernel Hilbert Spaces 167
</p>
<p>growth in basis functions as the dimension increases, and typically we have
to reduce the number of functions per coordinate accordingly.
The additive spline models discussed in Chapter 9 are a restricted class
</p>
<p>of multidimensional splines. They can be represented in this general formu-
lation as well; that is, there exists a penalty J [f ] that guarantees that the
solution has the form f(X) = α+ f1(X1) + &middot; &middot; &middot;+ fd(Xd) and that each of
the functions fj are univariate splines. In this case the penalty is somewhat
degenerate, and it is more natural to assume that f is additive, and then
simply impose an additional penalty on each of the component functions:
</p>
<p>J [f ] = J(f1 + f2 + &middot; &middot; &middot;+ fd)
</p>
<p>=
</p>
<p>d&sum;
</p>
<p>j=1
</p>
<p>&int;
f &prime;&prime;j (tj)
</p>
<p>2dtj . (5.40)
</p>
<p>These are naturally extended to ANOVA spline decompositions,
</p>
<p>f(X) = α+
&sum;
</p>
<p>j
</p>
<p>fj(Xj) +
&sum;
</p>
<p>j&lt;k
</p>
<p>fjk(Xj , Xk) + &middot; &middot; &middot; , (5.41)
</p>
<p>where each of the components are splines of the required dimension. There
are many choices to be made:
</p>
<p>&bull; The maximum order of interaction&mdash;we have shown up to order 2
above.
</p>
<p>&bull; Which terms to include&mdash;not all main effects and interactions are
necessarily needed.
</p>
<p>&bull; What representation to use&mdash;some choices are:
</p>
<p>&ndash; regression splines with a relatively small number of basis func-
tions per coordinate, and their tensor products for interactions;
</p>
<p>&ndash; a complete basis as in smoothing splines, and include appropri-
ate regularizers for each term in the expansion.
</p>
<p>In many cases when the number of potential dimensions (features) is large,
automatic methods are more desirable. The MARS and MART procedures
(Chapters 9 and 10, respectively), both fall into this category.
</p>
<p>5.8 Regularization and Reproducing Kernel
Hilbert Spaces
</p>
<p>In this section we cast splines into the larger context of regularization meth-
ods and reproducing kernel Hilbert spaces. This section is quite technical
and can be skipped by the disinterested or intimidated reader.</p>
<p/>
</div>
<div class="page"><p/>
<p>168 5. Basis Expansions and Regularization
</p>
<p>A general class of regularization problems has the form
</p>
<p>min
f&isin;H
</p>
<p>[
N&sum;
</p>
<p>i=1
</p>
<p>L(yi, f(xi)) + λJ(f)
</p>
<p>]
(5.42)
</p>
<p>where L(y, f(x)) is a loss function, J(f) is a penalty functional, and H is
a space of functions on which J(f) is defined. Girosi et al. (1995) describe
quite general penalty functionals of the form
</p>
<p>J(f) =
</p>
<p>&int;
</p>
<p>IRd
</p>
<p>|f̃(s)|2
G̃(s)
</p>
<p>ds, (5.43)
</p>
<p>where f̃ denotes the Fourier transform of f , and G̃ is some positive function
that falls off to zero as ||s|| &rarr; &infin;. The idea is that 1/G̃ increases the penalty
for high-frequency components of f . Under some additional assumptions
they show that the solutions have the form
</p>
<p>f(X) =
K&sum;
</p>
<p>k=1
</p>
<p>αkφk(X) +
N&sum;
</p>
<p>i=1
</p>
<p>θiG(X &minus; xi), (5.44)
</p>
<p>where the φk span the null space of the penalty functional J , and G is the
inverse Fourier transform of G̃. Smoothing splines and thin-plate splines
fall into this framework. The remarkable feature of this solution is that
while the criterion (5.42) is defined over an infinite-dimensional space, the
solution is finite-dimensional. In the next sections we look at some specific
examples.
</p>
<p>5.8.1 Spaces of Functions Generated by Kernels
</p>
<p>An important subclass of problems of the form (5.42) are generated by
a positive definite kernel K(x, y), and the corresponding space of func-
tions HK is called a reproducing kernel Hilbert space (RKHS). The penalty
functional J is defined in terms of the kernel as well. We give a brief and
simplified introduction to this class of models, adapted from Wahba (1990)
and Girosi et al. (1995), and nicely summarized in Evgeniou et al. (2000).
Let x, y &isin; IRp. We consider the space of functions generated by the linear
</p>
<p>span of {K(&middot;, y), y &isin; IRp)}; i.e arbitrary linear combinations of the form
f(x) =
</p>
<p>&sum;
m αmK(x, ym), where each kernel term is viewed as a function
</p>
<p>of the first argument, and indexed by the second. Suppose that K has an
eigen-expansion
</p>
<p>K(x, y) =
</p>
<p>&infin;&sum;
</p>
<p>i=1
</p>
<p>γiφi(x)φi(y) (5.45)
</p>
<p>with γi &ge; 0,
&sum;&infin;
</p>
<p>i=1 γ
2
i &lt; &infin;. Elements of HK have an expansion in terms of
</p>
<p>these eigen-functions,
</p>
<p>f(x) =
</p>
<p>&infin;&sum;
</p>
<p>i=1
</p>
<p>ciφi(x), (5.46)</p>
<p/>
</div>
<div class="page"><p/>
<p>5.8 Regularization and Reproducing Kernel Hilbert Spaces 169
</p>
<p>with the constraint that
</p>
<p>||f ||2HK
def
=
</p>
<p>&infin;&sum;
</p>
<p>i=1
</p>
<p>c2i /γi &lt; &infin;, (5.47)
</p>
<p>where ||f ||HK is the norm induced by K. The penalty functional in (5.42)
for the space HK is defined to be the squared norm J(f) = ||f ||2HK . The
quantity J(f) can be interpreted as a generalized ridge penalty, where
functions with large eigenvalues in the expansion (5.45) get penalized less,
and vice versa.
Rewriting (5.42) we have
</p>
<p>min
f&isin;HK
</p>
<p>[
N&sum;
</p>
<p>i=1
</p>
<p>L(yi, f(xi)) + λ||f ||2HK
</p>
<p>]
(5.48)
</p>
<p>or equivalently
</p>
<p>min
{cj}&infin;1
</p>
<p>⎡
⎣
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>L(yi,
</p>
<p>&infin;&sum;
</p>
<p>j=1
</p>
<p>cjφj(xi)) + λ
</p>
<p>&infin;&sum;
</p>
<p>j=1
</p>
<p>c2j/γj
</p>
<p>⎤
⎦ . (5.49)
</p>
<p>It can be shown (Wahba, 1990, see also Exercise 5.15) that the solution
to (5.48) is finite-dimensional, and has the form
</p>
<p>f(x) =
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>αiK(x, xi). (5.50)
</p>
<p>The basis function hi(x) = K(x, xi) (as a function of the first argument) is
known as the representer of evaluation at xi in HK , since for f &isin; HK , it is
easily seen that 〈K(&middot;, xi), f〉HK = f(xi). Similarly 〈K(&middot;, xi),K(&middot;, xj)〉HK =
K(xi, xj) (the reproducing property of HK), and hence
</p>
<p>J(f) =
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>N&sum;
</p>
<p>j=1
</p>
<p>K(xi, xj)αiαj (5.51)
</p>
<p>for f(x) =
&sum;N
</p>
<p>i=1 αiK(x, xi).
In light of (5.50) and (5.51), (5.48) reduces to a finite-dimensional crite-
</p>
<p>rion
min
α
</p>
<p>L(y,Kα) + λαTKα. (5.52)
</p>
<p>We are using a vector notation, in which K is the N &times;N matrix with ijth
entry K(xi, xj) and so on. Simple numerical algorithms can be used to
optimize (5.52). This phenomenon, whereby the infinite-dimensional prob-
lem (5.48) or (5.49) reduces to a finite dimensional optimization problem,
has been dubbed the kernel property in the literature on support-vector
machines (see Chapter 12).</p>
<p/>
</div>
<div class="page"><p/>
<p>170 5. Basis Expansions and Regularization
</p>
<p>There is a Bayesian interpretation of this class of models, in which f
is interpreted as a realization of a zero-mean stationary Gaussian process,
with prior covariance function K. The eigen-decomposition produces a se-
ries of orthogonal eigen-functions φj(x) with associated variances γj . The
typical scenario is that &ldquo;smooth&rdquo; functions φj have large prior variance,
while &ldquo;rough&rdquo; φj have small prior variances. The penalty in (5.48) is the
contribution of the prior to the joint likelihood, and penalizes more those
components with smaller prior variance (compare with (5.43)).
For simplicity we have dealt with the case here where all members of H
</p>
<p>are penalized, as in (5.48). More generally, there may be some components
in H that we wish to leave alone, such as the linear functions for cubic
smoothing splines in Section 5.4. The multidimensional thin-plate splines
of Section 5.7 and tensor product splines fall into this category as well.
In these cases there is a more convenient representation H = H0 &oplus; H1,
with the null space H0 consisting of, for example, low degree polynomi-
als in x that do not get penalized. The penalty becomes J(f) = ‖P1f‖,
where P1 is the orthogonal projection of f onto H1. The solution has the
form f(x) =
</p>
<p>&sum;M
j=1 βjhj(x) +
</p>
<p>&sum;N
i=1 αiK(x, xi), where the first term repre-
</p>
<p>sents an expansion in H0. From a Bayesian perspective, the coefficients of
components in H0 have improper priors, with infinite variance.
</p>
<p>5.8.2 Examples of RKHS
</p>
<p>The machinery above is driven by the choice of the kernel K and the loss
function L. We consider first regression using squared-error loss. In this
case (5.48) specializes to penalized least squares, and the solution can be
characterized in two equivalent ways corresponding to (5.49) or (5.52):
</p>
<p>min
{cj}&infin;1
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>⎛
⎝yi &minus;
</p>
<p>&infin;&sum;
</p>
<p>j=1
</p>
<p>cjφj(xi)
</p>
<p>⎞
⎠
</p>
<p>2
</p>
<p>+ λ
</p>
<p>&infin;&sum;
</p>
<p>j=1
</p>
<p>c2j
γj
</p>
<p>(5.53)
</p>
<p>an infinite-dimensional, generalized ridge regression problem, or
</p>
<p>min
α
</p>
<p>(y &minus;Kα)T (y &minus;Kα) + λαTKα. (5.54)
</p>
<p>The solution for α is obtained simply as
</p>
<p>α̂ = (K+ λI)&minus;1y, (5.55)
</p>
<p>and
</p>
<p>f̂(x) =
N&sum;
</p>
<p>j=1
</p>
<p>α̂jK(x, xj). (5.56)</p>
<p/>
</div>
<div class="page"><p/>
<p>5.8 Regularization and Reproducing Kernel Hilbert Spaces 171
</p>
<p>The vector of N fitted values is given by
</p>
<p>f̂ = Kα̂
</p>
<p>= K(K+ λI)&minus;1y (5.57)
</p>
<p>= (I+ λK&minus;1)&minus;1y. (5.58)
</p>
<p>The estimate (5.57) also arises as the kriging estimate of a Gaussian ran-
dom field in spatial statistics (Cressie, 1993). Compare also (5.58) with the
smoothing spline fit (5.17) on page 154.
</p>
<p>Penalized Polynomial Regression
</p>
<p>The kernel K(x, y) = (〈x, y〉 + 1)d (Vapnik, 1996), for x, y &isin; IRp, has
M =
</p>
<p>(
p+d
d
</p>
<p>)
eigen-functions that span the space of polynomials in IRp of
</p>
<p>total degree d. For example, with p = 2 and d = 2, M = 6 and
</p>
<p>K(x, y) = 1 + 2x1y1 + 2x2y2 + x
2
1y
</p>
<p>2
1 + x
</p>
<p>2
2y
</p>
<p>2
2 + 2x1x2y1y2 (5.59)
</p>
<p>=
</p>
<p>M&sum;
</p>
<p>m=1
</p>
<p>hm(x)hm(y) (5.60)
</p>
<p>with
</p>
<p>h(x)T = (1,
&radic;
2x1,
</p>
<p>&radic;
2x2, x
</p>
<p>2
1, x
</p>
<p>2
2,
&radic;
2x1x2). (5.61)
</p>
<p>One can represent h in terms of the M orthogonal eigen-functions and
eigenvalues of K,
</p>
<p>h(x) = VD
1
2
γ φ(x), (5.62)
</p>
<p>where Dγ = diag(γ1, γ2, . . . , γM ), and V is M &times;M and orthogonal.
Suppose we wish to solve the penalized polynomial regression problem
</p>
<p>min
{βm}M1
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>(
yi &minus;
</p>
<p>M&sum;
</p>
<p>m=1
</p>
<p>βmhm(xi)
</p>
<p>)2
+ λ
</p>
<p>M&sum;
</p>
<p>m=1
</p>
<p>β2m. (5.63)
</p>
<p>Substituting (5.62) into (5.63), we get an expression of the form (5.53) to
optimize (Exercise 5.16).
The number of basis functions M =
</p>
<p>(
p+d
d
</p>
<p>)
can be very large, often much
</p>
<p>larger than N . Equation (5.55) tells us that if we use the kernel represen-
tation for the solution function, we have only to evaluate the kernel N2
</p>
<p>times, and can compute the solution in O(N3) operations.
This simplicity is not without implications. Each of the polynomials hm
</p>
<p>in (5.61) inherits a scaling factor from the particular form of K, which has
a bearing on the impact of the penalty in (5.63). We elaborate on this in
the next section.</p>
<p/>
</div>
<div class="page"><p/>
<p>172 5. Basis Expansions and Regularization
</p>
<p>&minus;2 &minus;1 0 1 2 3 4
</p>
<p>0
.0
</p>
<p>0
.4
</p>
<p>0
.8
</p>
<p>X
</p>
<p>Radial Kernel in IR1
K
(&middot;,
</p>
<p>x
m
)
</p>
<p>FIGURE 5.13. Radial kernels kk(x) for the mixture data, with scale parameter
ν = 1. The kernels are centered at five points xm chosen at random from the 200.
</p>
<p>Gaussian Radial Basis Functions
</p>
<p>In the preceding example, the kernel is chosen because it represents an
expansion of polynomials and can conveniently compute high-dimensional
inner products. In this example the kernel is chosen because of its functional
form in the representation (5.50).
</p>
<p>The Gaussian kernel K(x, y) = e&minus;ν||x&minus;y||
2
</p>
<p>along with squared-error loss,
for example, leads to a regression model that is an expansion in Gaussian
radial basis functions,
</p>
<p>km(x) = e
&minus;ν||x&minus;xm||
</p>
<p>2
</p>
<p>, m = 1, . . . , N, (5.64)
</p>
<p>each one centered at one of the training feature vectors xm. The coefficients
are estimated using (5.54).
Figure 5.13 illustrates radial kernels in IR1 using the first coordinate of
</p>
<p>the mixture example from Chapter 2. We show five of the 200 kernel basis
functions km(x) = K(x, xm).
</p>
<p>Figure 5.14 illustrates the implicit feature space for the radial kernel
with x &isin; IR1. We computed the 200&times; 200 kernel matrix K, and its eigen-
decomposition ΦDγΦ
</p>
<p>T . We can think of the columns of Φ and the corre-
sponding eigenvalues in Dγ as empirical estimates of the eigen expansion
(5.45)2. Although the eigenvectors are discrete, we can represent them as
functions on IR1 (Exercise 5.17). Figure 5.15 shows the largest 50 eigenval-
ues of K. The leading eigenfunctions are smooth, and they are successively
more wiggly as the order increases. This brings to life the penalty in (5.49),
where we see the coefficients of higher-order functions get penalized more
than lower-order ones. The right panel in Figure 5.14 shows the correspond-
</p>
<p>2The ℓth column of Φ is an estimate of φℓ, evaluated at each of the N observations.
Alternatively, the ith row of Φ is the estimated vector of basis functions φ(xi), evaluated
at the point xi. Although in principle, there can be infinitely many elements in φ, our
</p>
<p>estimate has at most N elements.</p>
<p/>
</div>
<div class="page"><p/>
<p>5.8 Regularization and Reproducing Kernel Hilbert Spaces 173
</p>
<p>*
* ***
</p>
<p>**
* ****
</p>
<p>*
* * *
</p>
<p>*
*
</p>
<p>**
**
</p>
<p>*
*
</p>
<p>*** **
*
</p>
<p>*
</p>
<p>*** **
*
</p>
<p>* ** ** *
*
</p>
<p>* *
*
</p>
<p>** *
*
</p>
<p>*
** *
</p>
<p>*
**
</p>
<p>* *
</p>
<p>** **
*
</p>
<p>*
*
</p>
<p>**
*
</p>
<p>* ******
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>**
*
</p>
<p>*
</p>
<p>** *
* *
</p>
<p>* *
*
</p>
<p>*
*
</p>
<p>*
**
</p>
<p>* *
*
</p>
<p>**
* *
</p>
<p>** *** *** *
</p>
<p>**
</p>
<p>** ***
*
</p>
<p>**
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
**
</p>
<p>*
*
</p>
<p>* ** * ****
* *
</p>
<p>*
</p>
<p>*
*
</p>
<p>***
</p>
<p>*
</p>
<p>**
*
</p>
<p>* *
*
</p>
<p>* *
*
</p>
<p>* ***
* **
</p>
<p>*
** ** ** *
</p>
<p>*
*
</p>
<p>** **
*
</p>
<p>** **
** **
</p>
<p>****
*
</p>
<p>*
</p>
<p>**
*
</p>
<p>***
</p>
<p>*
</p>
<p>*
**
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>**
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>* *
</p>
<p>***
</p>
<p>*
</p>
<p>*
* * ***
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* ** **
</p>
<p>*
*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>**
*****
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>**
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* **
</p>
<p>*
*
</p>
<p>**
*
</p>
<p>* *
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*** **
*
</p>
<p>*
**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
**
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>**
</p>
<p>***
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>**
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
**
</p>
<p>**
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
*** *
</p>
<p>*
**
</p>
<p>**
</p>
<p>**
*
</p>
<p>*
**
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
***
</p>
<p>*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
*
</p>
<p>**
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>* *
</p>
<p>*
</p>
<p>* *
</p>
<p>**
* **
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
*
</p>
<p>**
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
**
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*****
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
*
</p>
<p>* ***
</p>
<p>*
</p>
<p>**
*
</p>
<p>**
*
</p>
<p>** *
</p>
<p>**
**
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>**
</p>
<p>*
**
</p>
<p>*
</p>
<p>*
* ** *
</p>
<p>*
</p>
<p>*
* **
</p>
<p>***
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*** *
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>**
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>* ** *
*
</p>
<p>*
**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>**
</p>
<p>*
</p>
<p>** *
*
</p>
<p>**
</p>
<p>**
</p>
<p>****
</p>
<p>**
**
</p>
<p>*
</p>
<p>*
**
</p>
<p>**
*** **
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
* *
</p>
<p>**
</p>
<p>*
** *
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>**
*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>* **
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>**
*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>**
</p>
<p>*
*
</p>
<p>*
* ***
</p>
<p>*
</p>
<p>*
***
</p>
<p>**
*
</p>
<p>*
* **
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>**
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>* **
*
</p>
<p>* *
</p>
<p>**
</p>
<p>* *
** *** *
</p>
<p>*
*
</p>
<p>*
</p>
<p>**
*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>* **
*
</p>
<p>**
**
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
**
</p>
<p>* *
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>* *
*
</p>
<p>*
</p>
<p>** *
</p>
<p>* **
* ***
</p>
<p>* **
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>* *
</p>
<p>***
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>**
</p>
<p>*
*
</p>
<p>** **
</p>
<p>* **
* *
</p>
<p>**
*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
** *
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*** **
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>* *
*
</p>
<p>**
*
</p>
<p>*
</p>
<p>*
</p>
<p>**
** **
</p>
<p>* *
**
</p>
<p>**
</p>
<p>*
*
</p>
<p>*
</p>
<p>** **
</p>
<p>*
**
</p>
<p>***
</p>
<p>** *
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
**
</p>
<p>*
**
</p>
<p>*
*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>** *
**
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>**
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*****
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
**
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>* **
</p>
<p>*
</p>
<p>** *
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>**
**
</p>
<p>*
</p>
<p>**
*
</p>
<p>*
</p>
<p>** *
*
</p>
<p>*
*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*** **
</p>
<p>**
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>**
</p>
<p>*
*
</p>
<p>**
*
</p>
<p>**
*
</p>
<p>** *
</p>
<p>*
</p>
<p>*
***
</p>
<p>* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
**
</p>
<p>*
</p>
<p>*
</p>
<p>* *
* ** ** *
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
** *
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>**
*
</p>
<p>*
</p>
<p>** **
*
</p>
<p>*
*
</p>
<p>** *
</p>
<p>*
</p>
<p>*
</p>
<p>****
*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
* * *
</p>
<p>*
*
</p>
<p>*
**
</p>
<p>* *
</p>
<p>*
</p>
<p>**
</p>
<p>*
***
</p>
<p>*
** *
</p>
<p>*
*
</p>
<p>*
</p>
<p>**
</p>
<p>*
* *
**
</p>
<p>* **
* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>** *
*
</p>
<p>* *
</p>
<p>****
*
</p>
<p>**
</p>
<p>*
*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>**
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>** *
* **
</p>
<p>*
</p>
<p>*
</p>
<p>**
* *** *
</p>
<p>*
***
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>***
</p>
<p>*
</p>
<p>*
</p>
<p>** *
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>**
</p>
<p>* *
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
**
</p>
<p>*
</p>
<p>* *
*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
* * ***
</p>
<p>* **
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
**
</p>
<p>*
** *
</p>
<p>* *
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>**
**
</p>
<p>*
*
</p>
<p>*** *
*
</p>
<p>*
**
</p>
<p>***
*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
**
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
</p>
<p>* * ***
* *
</p>
<p>**
</p>
<p>*
</p>
<p>*
**
</p>
<p>*
**
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
**
</p>
<p>***
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
**
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
**
</p>
<p>*
*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>* *
*
</p>
<p>*
</p>
<p>** **
</p>
<p>*
</p>
<p>*
*
</p>
<p>***
*
</p>
<p>**
</p>
<p>*
</p>
<p>**
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>***
</p>
<p>*
</p>
<p>*
</p>
<p>*
* **
</p>
<p>**
</p>
<p>*
</p>
<p>*
***
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>** *
* *
</p>
<p>**
*
</p>
<p>**
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
** *
</p>
<p>*
* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
** *
</p>
<p>*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>** *
*
</p>
<p>*
</p>
<p>* ***
</p>
<p>**
</p>
<p>*
</p>
<p>****
</p>
<p>* ** **
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>* *
</p>
<p>* *
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
**
</p>
<p>* *
</p>
<p>*
*
*
</p>
<p>* ***
</p>
<p>*
</p>
<p>**
*
</p>
<p>*
*
</p>
<p>*** *
</p>
<p>* *
</p>
<p>**
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
*
</p>
<p>*
</p>
<p>*
***
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>* *
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>** **
*
</p>
<p>*
</p>
<p>***
*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
**
</p>
<p>*
*
</p>
<p>**
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>**
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>**
**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
**
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>**
*
</p>
<p>**
*
</p>
<p>*
**
</p>
<p>*
</p>
<p>*
</p>
<p>**
* **
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>** *
</p>
<p>*
</p>
<p>** *
* *
</p>
<p>*
</p>
<p>**
* **
</p>
<p>***
**
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>** *
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
**
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>***
*** *
</p>
<p>*
*
</p>
<p>*
*
* *
</p>
<p>*
*
</p>
<p>*
</p>
<p>** **
</p>
<p>*
</p>
<p>*
* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
****
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
**
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>* *
</p>
<p>* *
</p>
<p>**
</p>
<p>*
* **
</p>
<p>*
*
</p>
<p>*
**
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
*
* *
</p>
<p>** ** *
</p>
<p>*
*
</p>
<p>*
</p>
<p>**
*
</p>
<p>*
*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>***
*
</p>
<p>*
**
</p>
<p>* **
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
**
</p>
<p>***
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>**
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>** *
*
</p>
<p>*
</p>
<p>**
</p>
<p>**
</p>
<p>** *
</p>
<p>**
*
</p>
<p>*
</p>
<p>**
*
</p>
<p>* *
** *
</p>
<p>* *
*
</p>
<p>*
**
</p>
<p>**
</p>
<p>*
</p>
<p>****
</p>
<p>*
**
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>* *
</p>
<p>**
**
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>* *
</p>
<p>*** *
</p>
<p>*
*
</p>
<p>*
</p>
<p>* ***
</p>
<p>*
</p>
<p>**
*
</p>
<p>*
*
</p>
<p>*
**
</p>
<p>*
* *
*
</p>
<p>*
</p>
<p>*
**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
**
</p>
<p>*
*
</p>
<p>* **
*
</p>
<p>*
*
</p>
<p>*** *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
**
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
* *
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*** *
*
</p>
<p>*
</p>
<p>*
* *
</p>
<p>*
*
</p>
<p>* ** ****
*
</p>
<p>*
</p>
<p>** *
**
</p>
<p>* **
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
* *
</p>
<p>***
**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
** *
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
**
</p>
<p>*
</p>
<p>*
**
</p>
<p>* ** *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
** *
</p>
<p>*
**
</p>
<p>* **
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
** **
</p>
<p>*
*
</p>
<p>**
*
</p>
<p>*
* *
*
</p>
<p>*
**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>**
</p>
<p>**
*
</p>
<p>**
*
</p>
<p>*
***
</p>
<p>*
</p>
<p>*
*
*
</p>
<p>* *
**
</p>
<p>*** **
*
</p>
<p>*
</p>
<p>**
</p>
<p>*
* *
*
</p>
<p>*
</p>
<p>*
*** *
</p>
<p>*
*
</p>
<p>*
</p>
<p>* ** *
*
</p>
<p>*
* *
</p>
<p>*
**
</p>
<p>* *
*
</p>
<p>** *
**
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
**
</p>
<p>*
**
</p>
<p>** **
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>**
*
</p>
<p>** *
</p>
<p>***
</p>
<p>**
</p>
<p>*
</p>
<p>***
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>**
* *
</p>
<p>* **
</p>
<p>* **
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>**
</p>
<p>*
**
</p>
<p>*
</p>
<p>* * **
</p>
<p>**
</p>
<p>*
</p>
<p>* *
</p>
<p>*
</p>
<p>**
</p>
<p>*
** *
</p>
<p>*
** *
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>**
</p>
<p>** *
</p>
<p>*
</p>
<p>*
** *
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>**
</p>
<p>**
</p>
<p>*
</p>
<p>**
**
</p>
<p>*
</p>
<p>**
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>**
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>**
* *
</p>
<p>*
</p>
<p>**
* *
</p>
<p>**
*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>* *
*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>**
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>***
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
* *
</p>
<p>*
*
</p>
<p>*
* * *
</p>
<p>*
*
</p>
<p>* *
*
</p>
<p>*
</p>
<p>** *
</p>
<p>*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
*
</p>
<p>*
* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
* ***
</p>
<p>*
**
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
* *
</p>
<p>***
</p>
<p>*
*
</p>
<p>**
* *
*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>* *
</p>
<p>*
*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
**
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>**
* ** *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>** *
* *
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>* **
</p>
<p>*
*
</p>
<p>* *
*
</p>
<p>* *
*
</p>
<p>**
*
</p>
<p>*
*
</p>
<p>*
* *
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>**
*
</p>
<p>** *
*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>****
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>* * **
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
**
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* * *
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>* **
</p>
<p>*
</p>
<p>**
*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>**
</p>
<p>*
</p>
<p>**
*
</p>
<p>**
</p>
<p>* ** *
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>**
</p>
<p>* **
*
</p>
<p>*
</p>
<p>*
</p>
<p>**
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>* *** **
</p>
<p>*
</p>
<p>**
</p>
<p>**
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>**
* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
** *
</p>
<p>*
</p>
<p>*
** *
</p>
<p>* **
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
**
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>** *
</p>
<p>*
*
</p>
<p>*
***
</p>
<p>*
</p>
<p>*
* *
</p>
<p>*
**
</p>
<p>* ***
</p>
<p>** *
</p>
<p>*
</p>
<p>*
*
</p>
<p>** *
*
**
</p>
<p>*
*
</p>
<p>*
</p>
<p>**
*
</p>
<p>** *
**
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>**
**
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* **
</p>
<p>*
*
*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>** ***
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
****
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>** *
</p>
<p>*
</p>
<p>*
</p>
<p>* *
*
</p>
<p>*
</p>
<p>*
* *
</p>
<p>*
</p>
<p>*
**
*
</p>
<p>*
** *
</p>
<p>*
*
</p>
<p>***
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>**
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
**
</p>
<p>*
</p>
<p>*
</p>
<p>**
*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
* **
</p>
<p>*
</p>
<p>*
</p>
<p>***
</p>
<p>*
</p>
<p>*
* *
</p>
<p>**
* **
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>** *
</p>
<p>*
</p>
<p>* *
</p>
<p>*
</p>
<p>**
</p>
<p>* ** *
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>***
</p>
<p>*
*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>**
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
**
</p>
<p>*
** *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
*
</p>
<p>*
</p>
<p>*
**
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
**
</p>
<p>*
</p>
<p>**
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
***
</p>
<p>*
</p>
<p>**
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>* *
* *
</p>
<p>*
*
</p>
<p>***
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>****
**
</p>
<p>**
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
** *
</p>
<p>* *
</p>
<p>** *
*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
*
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>**
</p>
<p>*
**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>** *
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
* **
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>** *
* *
</p>
<p>*
</p>
<p>**
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>** *
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>**
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
* *
</p>
<p>*
*
</p>
<p>**
* **
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
* *
</p>
<p>* *
*
</p>
<p>***
</p>
<p>*
*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
* *
</p>
<p>**
</p>
<p>*
*
</p>
<p>**
* *
</p>
<p>*
</p>
<p>*
</p>
<p>* **
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>**
</p>
<p>*
** *
</p>
<p>*
*
</p>
<p>*
*
*
</p>
<p>**
</p>
<p>*
**
</p>
<p>*
*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* **
</p>
<p>* ***
*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
*
</p>
<p>***
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
*
</p>
<p>***
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* * *
</p>
<p>*
** **
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>Orthonormal Basis Φ
</p>
<p>*
</p>
<p>* **
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
</p>
<p>**
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*** *
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
</p>
<p>**
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>**
**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*****
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>**
*
</p>
<p>**
*
</p>
<p>** *
</p>
<p>**
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
* *
</p>
<p>*
**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
**
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>* *
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>**
</p>
<p>*
</p>
<p>**
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
***
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
** *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
**
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>***
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>** *
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>**
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
***
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>* **
*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>** *
</p>
<p>**
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>* *
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>***
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* ** *
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>**
</p>
<p>****
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>**
*** **
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>**
</p>
<p>*
</p>
<p>** *
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
**
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>**
* *
</p>
<p>*
**
</p>
<p>*
**
</p>
<p>*
</p>
<p>*
* *
</p>
<p>* *
*
</p>
<p>*
</p>
<p>*
</p>
<p>**
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* **
*
</p>
<p>*
*
</p>
<p>**
</p>
<p>* *
** *** *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
*
</p>
<p>*
</p>
<p>**
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>** *
</p>
<p>* **
</p>
<p>* ***
*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>***
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>**
**
</p>
<p>* **
</p>
<p>* *
</p>
<p>**
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>** *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>***
**
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
* **
</p>
<p>* *
*
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>**
**
</p>
<p>*
</p>
<p>*
*
</p>
<p>***
</p>
<p>*
* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>** *
**
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
****
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>* **
</p>
<p>*
</p>
<p>** *
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
* *
</p>
<p>*
</p>
<p>*
*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>**
*
</p>
<p>**
**
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>**
*
</p>
<p>**
*
</p>
<p>** *
</p>
<p>*
</p>
<p>*
</p>
<p>**
*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>* *
</p>
<p>* ** *
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
** *
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
*
</p>
<p>*
</p>
<p>**
**
</p>
<p>*
</p>
<p>*
*
</p>
<p>** *
</p>
<p>*
</p>
<p>*
</p>
<p>****
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
* * *
</p>
<p>*
</p>
<p>*
</p>
<p>*
**
</p>
<p>* *
</p>
<p>*
</p>
<p>**
</p>
<p>*
***
</p>
<p>*
**
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>**
</p>
<p>*
* *
**
</p>
<p>* **
</p>
<p>* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>** *
*
</p>
<p>* *
</p>
<p>**
**
</p>
<p>*
</p>
<p>**
</p>
<p>*
*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
* **
</p>
<p>*
</p>
<p>*
</p>
<p>**
*
</p>
<p>*** *
</p>
<p>*
***
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>***
</p>
<p>*
</p>
<p>*
</p>
<p>** *
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>**
</p>
<p>* *
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
**
</p>
<p>*
</p>
<p>* *
*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
* * ***
</p>
<p>* **
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
**
</p>
<p>*
** *
</p>
<p>* *
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>**
**
</p>
<p>*
*
</p>
<p>*** *
*
</p>
<p>*
**
</p>
<p>***
*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
**
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
</p>
<p>* * ***
* *
</p>
<p>**
</p>
<p>*
</p>
<p>*
**
</p>
<p>*
**
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
**
</p>
<p>***
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
**
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
**
</p>
<p>*
*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>* *
*
</p>
<p>*
</p>
<p>** **
</p>
<p>*
</p>
<p>*
*
</p>
<p>***
*
</p>
<p>**
</p>
<p>*
</p>
<p>**
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>***
</p>
<p>*
</p>
<p>*
</p>
<p>*
* **
</p>
<p>**
</p>
<p>*
</p>
<p>*
***
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>** *
* * **
</p>
<p>*
</p>
<p>**
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
**
</p>
<p>*
*
</p>
<p>*
**
</p>
<p>** **
* *
</p>
<p>*
</p>
<p>*
* ** *
</p>
<p>*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>** *
*
</p>
<p>*
</p>
<p>* ***
**
</p>
<p>*
</p>
<p>****
* ** **
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
**
</p>
<p>*
</p>
<p>* *
* *
</p>
<p>*
**
</p>
<p>*
**
</p>
<p>* *
</p>
<p>*
*
*
</p>
<p>* ***
</p>
<p>*
**
*
</p>
<p>*
*
</p>
<p>*** *
* *
**
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
*
</p>
<p>*
*****
</p>
<p>**
</p>
<p>** *
**
</p>
<p>*
</p>
<p>*
**
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>**
</p>
<p>* *
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
** **
</p>
<p>*
</p>
<p>*
</p>
<p>*** *
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>****
*
</p>
<p>*
**
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>** *
</p>
<p>*
*
</p>
<p>*
</p>
<p>** **
*
</p>
<p>*
*
</p>
<p>* **
*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>**
*
</p>
<p>** **
* ***
</p>
<p>* **
*
</p>
<p>*
</p>
<p>*** ** **
*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
* *
</p>
<p>* **
*
</p>
<p>*
*
</p>
<p>*
* *
</p>
<p>*
** *
</p>
<p>*
</p>
<p>** **
*
</p>
<p>*
*** *
</p>
<p>* ****** *
*
</p>
<p>** *
**
</p>
<p>*
</p>
<p>*
*
</p>
<p>**
** * **
</p>
<p>*
* ***
</p>
<p>*
* *
</p>
<p>** ***
*** *** *
</p>
<p>** *
* **
</p>
<p>** **
*
</p>
<p>*
* **
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
* *
</p>
<p>****
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
**
</p>
<p>*
</p>
<p>*
**
</p>
<p>* **
* *
</p>
<p>* *
**
</p>
<p>** **
***
</p>
<p>**
*
</p>
<p>* * *
*
</p>
<p>**
*** ** *
</p>
<p>*
*
</p>
<p>*
** ***
</p>
<p>* *
</p>
<p>*
</p>
<p>** *
*** *
</p>
<p>* *** *
*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
** * ** ***
** * **
</p>
<p>** *
* *
</p>
<p>*
**
</p>
<p>* **
** ** ** *
</p>
<p>**
**
</p>
<p>** *
** *
</p>
<p>* ** *
* *
</p>
<p>*
</p>
<p>** *
* **
</p>
<p>***
**
</p>
<p>*
***** **
</p>
<p>*
* *
</p>
<p>*
* *
</p>
<p>** **
** *
*** *
</p>
<p>*** * * **
* ***
</p>
<p>**
* *** *
</p>
<p>** ** **
*
</p>
<p>* **
*
</p>
<p>** ***
** * ** * **
</p>
<p>*** *
*
</p>
<p>*
*
</p>
<p>***
</p>
<p>*
</p>
<p>***
*
</p>
<p>**
*
</p>
<p>*
* * ***
*
</p>
<p>** **
* ** **
</p>
<p>*
**
</p>
<p>** *** ** **
*
</p>
<p>* ** **** *
</p>
<p>*
</p>
<p>** ***
* ** *** **
</p>
<p>* **** ** * **
*** ** *
**** ** *
</p>
<p>**** ** ** ** ** * ** ** ** *
**
</p>
<p>**
** ** *
</p>
<p>*
</p>
<p>** ** ** *** ** ****** *
* ** *
</p>
<p>** *** **
** * *** * *** * * **
</p>
<p>* *** *** *** ***
** ***
</p>
<p>* *** ** *
** ** * ** * ***** ** *
</p>
<p>* ***
*
</p>
<p>****
**
</p>
<p>* ** *
**** ** *
** ** ** * **** *
** ** **** ** *
</p>
<p>*** *
*
</p>
<p>** **** ** *** *** ***
* ** * ** *** ** * **** ** *
</p>
<p>*
*** ** ** ** ** * ** ** ** *
</p>
<p>** ** ** ** *
*
</p>
<p>** ** ** *** ** ****** ** ** **
* *** ** ** * *** * *** * * *** ***
</p>
<p>*** *** *
** ** **** ***
</p>
<p>** *** ** * **
* ***** ** ** ****
</p>
<p>***
* *** ** * **** ** *** ** ** * **** *** ** ***
</p>
<p>* ** **** **
** ****
</p>
<p>** *** *** **** ** * ** *** ** * **** ** * **** ** ** ** ** * ** ** ** * ** ** ** ** * *** ** ** *** ** ****** ** ** *** *** ** ** * *** * *** * * *** *** *** *** *** ** **** **
* ** *** *
* * ** * ***** ** ** **** **** ***
</p>
<p>** * **** ** *** ** ** * **** *** ** **** ** **** ** ** **** ** *** *** **** ** * ** *** ** * **** ** * **** ** ** ** ** * ** ** ** * ** ** ** ** * *** ** ** *** ** ****** ** ** *** *** ** ** * *** * *** * * *** *** *** *** *** ** **** *** ** *** ** * ** * ***** ** ** **** **** *** ** * **** ** *** ** ** * **** *** ** **** ** **** ** ** **** ** *** *** **** ** * ** *** ** * **** ** * **** ** ** ** ** * ** ** ** * ** ** ** ** * *** ** ** *** ** ****** ** ** *** *** ** ** * *** * *** * * *** *** *** *** *** ** **** *** ** *** ** * ** * ***** ** ** **** **** *** ** * **** ** *** ** ** * **** *** ** **** ** **** ** ** **** ** *** *** **** ** * ** *** ** * **** ** * **** ** ** ** ** * ** ** ** * ** ** ** ** * *** ** ** *** ** ****** ** ** *** *** ** ** * *** * *** * * *** *** *** *** *** ** **** *** ** *** ** * ** * ***** ** ** **** **** *** ** * **** ** *** ** ** * **** *** ** **** ** **** ** ** ****
</p>
<p>Feature Space H
</p>
<p>FIGURE 5.14. (Left panel) The first 16 normalized eigenvectors of K, the
200 &times; 200 kernel matrix for the first coordinate of the mixture data. These are
viewed as estimates φ̂ℓ of the eigenfunctions in (5.45), and are represented as
functions in IR1 with the observed values superimposed in color. They are arranged
in rows, starting at the top left. (Right panel) Rescaled versions hℓ =
</p>
<p>&radic;
γ̂ℓφ̂ℓ of
</p>
<p>the functions in the left panel, for which the kernel computes the &ldquo;inner product.&rdquo;
</p>
<p>0 10 20 30 40 50
</p>
<p>1
e
&minus;
</p>
<p>1
5
</p>
<p>1
e
&minus;
</p>
<p>1
1
</p>
<p>1
e
&minus;
</p>
<p>0
7
</p>
<p>1
e
&minus;
</p>
<p>0
3
</p>
<p>1
e
+
</p>
<p>0
1
</p>
<p>E
ig
</p>
<p>e
n
v
a
lu
</p>
<p>e
</p>
<p>FIGURE 5.15. The largest 50 eigenvalues of K; all those beyond the 30th are
effectively zero.</p>
<p/>
</div>
<div class="page"><p/>
<p>174 5. Basis Expansions and Regularization
</p>
<p>ing feature space representation of the eigenfunctions
</p>
<p>hℓ(x) =
&radic;
</p>
<p>γ̂ℓφ̂ℓ(x), ℓ = 1, . . . , N. (5.65)
</p>
<p>Note that 〈h(xi), h(xi&prime;)〉 = K(xi, xi&prime;). The scaling by the eigenvalues quickly
shrinks most of the functions down to zero, leaving an effective dimension
of about 12 in this case. The corresponding optimization problem is a stan-
dard ridge regression, as in (5.63). So although in principle the implicit
feature space is infinite dimensional, the effective dimension is dramat-
ically lower because of the relative amounts of shrinkage applied to each
basis function. The kernel scale parameter ν plays a role here as well; larger
ν implies more local km functions, and increases the effective dimension of
the feature space. See Hastie and Zhu (2006) for more details.
It is also known (Girosi et al., 1995) that a thin-plate spline (Section 5.7)
</p>
<p>is an expansion in radial basis functions, generated by the kernel
</p>
<p>K(x, y) = ‖x&minus; y‖2 log(‖x&minus; y‖). (5.66)
</p>
<p>Radial basis functions are discussed in more detail in Section 6.7.
</p>
<p>Support Vector Classifiers
</p>
<p>The support vector machines of Chapter 12 for a two-class classification
problem have the form f(x) = α0+
</p>
<p>&sum;N
i=1 αiK(x, xi), where the parameters
</p>
<p>are chosen to minimize
</p>
<p>min
α0,α
</p>
<p>{
N&sum;
</p>
<p>i=1
</p>
<p>[1&minus; yif(xi)]+ +
λ
</p>
<p>2
αTKα
</p>
<p>}
, (5.67)
</p>
<p>where yi &isin; {&minus;1, 1}, and [z]+ denotes the positive part of z. This can be
viewed as a quadratic optimization problem with linear constraints, and
requires a quadratic programming algorithm for its solution. The name
support vector arises from the fact that typically many of the α̂i = 0 [due
</p>
<p>to the piecewise-zero nature of the loss function in (5.67)], and so f̂ is an
expansion in a subset of the K(&middot;, xi). See Section 12.3.3 for more details.
</p>
<p>5.9 Wavelet Smoothing
</p>
<p>We have seen two different modes of operation with dictionaries of basis
functions. With regression splines, we select a subset of the bases, using
either subject-matter knowledge, or else automatically. The more adaptive
procedures such as MARS (Chapter 9) can capture both smooth and non-
smooth behavior. With smoothing splines, we use a complete basis, but
then shrink the coefficients toward smoothness.</p>
<p/>
</div>
<div class="page"><p/>
<p>5.9 Wavelet Smoothing 175
</p>
<p>Time
</p>
<p>0.0 0.2 0.4 0.6 0.8 1.0
</p>
<p>Haar Wavelets
</p>
<p>Time
</p>
<p>0.0 0.2 0.4 0.6 0.8 1.0
</p>
<p>Symmlet-8 Wavelets
</p>
<p>ψ1,0
</p>
<p>ψ2,1
</p>
<p>ψ2,3
</p>
<p>ψ3,2
</p>
<p>ψ3,5
</p>
<p>ψ4,4
</p>
<p>ψ4,9
</p>
<p>ψ5,1
</p>
<p>ψ5,15
</p>
<p>ψ6,15
</p>
<p>ψ6,35
</p>
<p>FIGURE 5.16. Some selected wavelets at different translations and dilations
for the Haar and symmlet families. The functions have been scaled to suit the
display.
</p>
<p>Wavelets typically use a complete orthonormal basis to represent func-
tions, but then shrink and select the coefficients toward a sparse represen-
tation. Just as a smooth function can be represented by a few spline basis
functions, a mostly flat function with a few isolated bumps can be repre-
sented with a few (bumpy) basis functions. Wavelets bases are very popular
in signal processing and compression, since they are able to represent both
smooth and/or locally bumpy functions in an efficient way&mdash;a phenomenon
dubbed time and frequency localization. In contrast, the traditional Fourier
basis allows only frequency localization.
Before we give details, let&rsquo;s look at the Haar wavelets in the left panel
</p>
<p>of Figure 5.16 to get an intuitive idea of how wavelet smoothing works.
The vertical axis indicates the scale (frequency) of the wavelets, from low
scale at the bottom to high scale at the top. At each scale the wavelets are
&ldquo;packed in&rdquo; side-by-side to completely fill the time axis: we have only shown</p>
<p/>
</div>
<div class="page"><p/>
<p>176 5. Basis Expansions and Regularization
</p>
<p>a selected subset. Wavelet smoothing fits the coefficients for this basis by
least squares, and then thresholds (discards, filters) the smaller coefficients.
Since there are many basis functions at each scale, it can use bases where
it needs them and discard the ones it does not need, to achieve time and
frequency localization. The Haar wavelets are simple to understand, but not
smooth enough for most purposes. The symmlet wavelets in the right panel
of Figure 5.16 have the same orthonormal properties, but are smoother.
Figure 5.17 displays an NMR (nuclear magnetic resonance) signal, which
</p>
<p>appears to be composed of smooth components and isolated spikes, plus
some noise. The wavelet transform, using a symmlet basis, is shown in the
lower left panel. The wavelet coefficients are arranged in rows, from lowest
scale at the bottom, to highest scale at the top. The length of each line
segment indicates the size of the coefficient. The bottom right panel shows
the wavelet coefficients after they have been thresholded. The threshold
procedure, given below in equation (5.69), is the same soft-thresholding
rule that arises in the lasso procedure for linear regression (Section 3.4.2).
Notice that many of the smaller coefficients have been set to zero. The
green curve in the top panel shows the back-transform of the thresholded
coefficients: this is the smoothed version of the original signal. In the next
section we give the details of this process, including the construction of
wavelets and the thresholding rule.
</p>
<p>5.9.1 Wavelet Bases and the Wavelet Transform
</p>
<p>In this section we give details on the construction and filtering of wavelets.
Wavelet bases are generated by translations and dilations of a single scal-
ing function φ(x) (also known as the father). The red curves in Figure 5.18
are the Haar and symmlet-8 scaling functions. The Haar basis is particu-
larly easy to understand, especially for anyone with experience in analysis
of variance or trees, since it produces a piecewise-constant representation.
Thus if φ(x) = I(x &isin; [0, 1]), then φ0,k(x) = φ(x&minus;k), k an integer, generates
an orthonormal basis for functions with jumps at the integers. Call this ref-
erence space V0. The dilations φ1,k(x) =
</p>
<p>&radic;
2φ(2x&minus;k) form an orthonormal
</p>
<p>basis for a space V1 &sup; V0 of functions piecewise constant on intervals of
length 12 . In fact, more generally we have &middot; &middot; &middot; &sup; V1 &sup; V0 &sup; V&minus;1 &sup; &middot; &middot; &middot; where
each Vj is spanned by φj,k = 2
</p>
<p>j/2φ(2jx&minus; k).
Now to the definition of wavelets. In analysis of variance, we often rep-
</p>
<p>resent a pair of means μ1 and μ2 by their grand mean μ =
1
2 (μ1 +μ2), and
</p>
<p>then a contrast α = 12 (μ1 &minus;μ2). A simplification occurs if the contrast α is
very small, because then we can set it to zero. In a similar manner we might
represent a function in Vj+1 by a component in Vj plus the component in
the orthogonal complement Wj of Vj to Vj+1, written as Vj+1 = Vj &oplus;Wj .
The component in Wj represents detail, and we might wish to set some ele-
ments of this component to zero. It is easy to see that the functions ψ(x&minus;k)</p>
<p/>
</div>
<div class="page"><p/>
<p>5.9 Wavelet Smoothing 177
</p>
<p>NMR Signal
</p>
<p>0 200 400 600 800 1000
</p>
<p>0
2
0
</p>
<p>4
0
</p>
<p>6
0
</p>
<p>0 200 400 600 800 1000
</p>
<p>Wavelet Transform - Original Signal
</p>
<p>0 200 400 600 800 1000
</p>
<p>Wavelet Transform - WaveShrunk Signal
</p>
<p>SignalSignal
</p>
<p>W9W9
</p>
<p>W8W8
</p>
<p>W7W7
</p>
<p>W6W6
</p>
<p>W5W5
</p>
<p>W4W4
</p>
<p>V4V4
</p>
<p>FIGURE 5.17. The top panel shows an NMR signal, with the wavelet-shrunk
version superimposed in green. The lower left panel represents the wavelet trans-
form of the original signal, down to V4, using the symmlet-8 basis. Each coeffi-
cient is represented by the height (positive or negative) of the vertical bar. The
lower right panel represents the wavelet coefficients after being shrunken using
the waveshrink function in S-PLUS, which implements the SureShrink method
of wavelet adaptation of Donoho and Johnstone.</p>
<p/>
</div>
<div class="page"><p/>
<p>178 5. Basis Expansions and Regularization
</p>
<p>Haar Basis Symmlet Basis
</p>
<p>φ(x)φ(x)
</p>
<p>ψ(x)ψ(x)
</p>
<p>FIGURE 5.18. The Haar and symmlet father (scaling) wavelet φ(x) and mother
wavelet ψ(x).
</p>
<p>generated by themother wavelet ψ(x) = φ(2x)&minus;φ(2x&minus;1) form an orthonor-
mal basis for W0 for the Haar family. Likewise ψj,k = 2
</p>
<p>j/2ψ(2jx&minus; k) form
a basis for Wj .
Now Vj+1 = Vj &oplus; Wj = Vj&minus;1 &oplus; Wj&minus;1 &oplus; Wj , so besides representing a
</p>
<p>function by its level-j detail and level-j rough components, the latter can
be broken down to level-(j&minus; 1) detail and rough, and so on. Finally we get
a representation of the form VJ = V0 &oplus;W0 &oplus;W1 &middot; &middot; &middot; &oplus;WJ&minus;1. Figure 5.16
on page 175 shows particular wavelets ψj,k(x).
Notice that since these spaces are orthogonal, all the basis functions are
</p>
<p>orthonormal. In fact, if the domain is discrete with N = 2J (time) points,
this is as far as we can go. There are 2j basis elements at level j, and
adding up, we have a total of 2J &minus; 1 elements in the Wj , and one in V0.
This structured orthonormal basis allows for a multiresolution analysis,
which we illustrate in the next section.
While helpful for understanding the construction above, the Haar basis
</p>
<p>is often too coarse for practical purposes. Fortunately, many clever wavelet
bases have been invented. Figures 5.16 and 5.18 include the Daubechies
symmlet-8 basis. This basis has smoother elements than the corresponding
Haar basis, but there is a tradeoff:
</p>
<p>&bull; Each wavelet has a support covering 15 consecutive time intervals,
rather than one for the Haar basis. More generally, the symmlet-p
family has a support of 2p &minus; 1 consecutive intervals. The wider the
support, the more time the wavelet has to die to zero, and so it can</p>
<p/>
</div>
<div class="page"><p/>
<p>5.9 Wavelet Smoothing 179
</p>
<p>achieve this more smoothly. Note that the effective support seems to
be much narrower.
</p>
<p>&bull; The symmlet-p wavelet ψ(x) has p vanishing moments; that is,
&int;
</p>
<p>ψ(x)xjdx = 0, j = 0, . . . , p&minus; 1.
</p>
<p>One implication is that any order-p polynomial over theN = 2J times
points is reproduced exactly in V0 (Exercise 5.18). In this sense V0
is equivalent to the null space of the smoothing-spline penalty. The
Haar wavelets have one vanishing moment, and V0 can reproduce any
constant function.
</p>
<p>The symmlet-p scaling functions are one of many families of wavelet
generators. The operations are similar to those for the Haar basis:
</p>
<p>&bull; If V0 is spanned by φ(x &minus; k), then V1 &sup; V0 is spanned by φ1,k(x) =&radic;
2φ(2x&minus;k) and φ(x) = &sum;k&isin;Z h(k)φ1,k(x), for some filter coefficients
</p>
<p>h(k).
</p>
<p>&bull; W0 is spanned by ψ(x) =
&sum;
</p>
<p>k&isin;Z g(k)φ1,k(x), with filter coefficients
</p>
<p>g(k) = (&minus;1)1&minus;kh(1&minus; k).
</p>
<p>5.9.2 Adaptive Wavelet Filtering
</p>
<p>Wavelets are particularly useful when the data are measured on a uniform
lattice, such as a discretized signal, image, or a time series. We will focus on
the one-dimensional case, and having N = 2J lattice-points is convenient.
Suppose y is the response vector, and W is the N&times;N orthonormal wavelet
basis matrix evaluated at the N uniformly spaced observations. Then y&lowast; =
WTy is called the wavelet transform of y (and is the full least squares
regression coefficient). A popular method for adaptive wavelet fitting is
known as SURE shrinkage (Stein Unbiased Risk Estimation, Donoho and
Johnstone (1994)). We start with the criterion
</p>
<p>min
θ
</p>
<p>||y &minus;Wθ||22 + 2λ||θ||1, (5.68)
</p>
<p>which is the same as the lasso criterion in Chapter 3. Because W is or-
thonormal, this leads to the simple solution:
</p>
<p>θ̂j = sign(y
&lowast;
j )(|y&lowast;j | &minus; λ)+. (5.69)
</p>
<p>The least squares coefficients are translated toward zero, and truncated
at zero. The fitted function (vector) is then given by the inverse wavelet
</p>
<p>transform f̂ = Wθ̂.</p>
<p/>
</div>
<div class="page"><p/>
<p>180 5. Basis Expansions and Regularization
</p>
<p>A simple choice for λ is λ = σ
&radic;
2 logN , where σ is an estimate of the
</p>
<p>standard deviation of the noise. We can give some motivation for this choice.
Since W is an orthonormal transformation, if the elements of y are white
noise (independent Gaussian variates with mean 0 and variance σ2), then
so are y&lowast;. Furthermore if random variables Z1, Z2, . . . , ZN are white noise,
the expected maximum of |Zj |, j = 1, . . . , N is approximately σ
</p>
<p>&radic;
2 logN .
</p>
<p>Hence all coefficients below σ
&radic;
2 logN are likely to be noise and are set to
</p>
<p>zero.
The space W could be any basis of orthonormal functions: polynomials,
</p>
<p>natural splines or cosinusoids. What makes wavelets special is the particular
form of basis functions used, which allows for a representation localized in
time and in frequency.
</p>
<p>Let&rsquo;s look again at the NMR signal of Figure 5.17. The wavelet transform
was computed using a symmlet&minus;8 basis. Notice that the coefficients do not
descend all the way to V0, but stop at V4 which has 16 basis functions.
As we ascend to each level of detail, the coefficients get smaller, except in
locations where spiky behavior is present. The wavelet coefficients represent
characteristics of the signal localized in time (the basis functions at each
level are translations of each other) and localized in frequency. Each dilation
increases the detail by a factor of two, and in this sense corresponds to
doubling the frequency in a traditional Fourier representation. In fact, a
more mathematical understanding of wavelets reveals that the wavelets at
a particular scale have a Fourier transform that is restricted to a limited
range or octave of frequencies.
The shrinking/truncation in the right panel was achieved using the SURE
</p>
<p>approach described in the introduction to this section. The orthonormal
N &times;N basis matrix W has columns which are the wavelet basis functions
evaluated at the N time points. In particular, in this case there will be 16
columns corresponding to the φ4,k(x), and the remainder devoted to the
ψj,k(x), j = 4, . . . , 11. In practice λ depends on the noise variance, and has
to be estimated from the data (such as the variance of the coefficients at
the highest level).
Notice the similarity between the SURE criterion (5.68) on page 179,
</p>
<p>and the smoothing spline criterion (5.21) on page 156:
</p>
<p>&bull; Both are hierarchically structured from coarse to fine detail, although
wavelets are also localized in time within each resolution level.
</p>
<p>&bull; The splines build in a bias toward smooth functions by imposing
differential shrinking constants dk. Early versions of SURE shrinkage
treated all scales equally. The S+wavelets function waveshrink() has
many options, some of which allow for differential shrinkage.
</p>
<p>&bull; The spline L2 penalty cause pure shrinkage, while the SURE L1
penalty does shrinkage and selection.</p>
<p/>
</div>
<div class="page"><p/>
<p>Exercises 181
</p>
<p>More generally smoothing splines achieve compression of the original signal
by imposing smoothness, while wavelets impose sparsity. Figure 5.19 com-
pares a wavelet fit (using SURE shrinkage) to a smoothing spline fit (using
cross-validation) on two examples different in nature. For the NMR data in
the upper panel, the smoothing spline introduces detail everywhere in order
to capture the detail in the isolated spikes; the wavelet fit nicely localizes
the spikes. In the lower panel, the true function is smooth, and the noise is
relatively high. The wavelet fit has let in some additional and unnecessary
wiggles&mdash;a price it pays in variance for the additional adaptivity.
The wavelet transform is not performed by matrix multiplication as in
</p>
<p>y&lowast; = WTy. In fact, using clever pyramidal schemes y&lowast; can be obtained
in O(N) computations, which is even faster than the N log(N) of the fast
Fourier transform (FFT). While the general construction is beyond the
scope of this book, it is easy to see for the Haar basis (Exercise 5.19).
</p>
<p>Likewise, the inverse wavelet transform Wθ̂ is also O(N).
This has been a very brief glimpse of this vast and growing field. There is
</p>
<p>a very large mathematical and computational base built on wavelets. Mod-
ern image compression is often performed using two-dimensional wavelet
representations.
</p>
<p>Bibliographic Notes
</p>
<p>Splines and B-splines are discussed in detail in de Boor (1978). Green
and Silverman (1994) and Wahba (1990) give a thorough treatment of
smoothing splines and thin-plate splines; the latter also covers reproducing
kernel Hilbert spaces. See also Girosi et al. (1995) and Evgeniou et al.
(2000) for connections between many nonparametric regression techniques
using RKHS approaches. Modeling functional data, as in Section 5.2.3, is
covered in detail in Ramsay and Silverman (1997).
Daubechies (1992) is a classic and mathematical treatment of wavelets.
</p>
<p>Other useful sources are Chui (1992) and Wickerhauser (1994). Donoho and
Johnstone (1994) developed the SURE shrinkage and selection technology
from a statistical estimation framework; see also Vidakovic (1999). Bruce
and Gao (1996) is a useful applied introduction, which also describes the
wavelet software in S-PLUS.
</p>
<p>Exercises
</p>
<p>Ex. 5.1 Show that the truncated power basis functions in (5.3) represent a
basis for a cubic spline with the two knots as indicated.</p>
<p/>
</div>
<div class="page"><p/>
<p>182 5. Basis Expansions and Regularization
</p>
<p>NMR Signal
</p>
<p>0 200 400 600 800 1000
</p>
<p>0
2
</p>
<p>0
4
</p>
<p>0
6
</p>
<p>0
</p>
<p>spline
wavelet
</p>
<p>Smooth Function (Simulated)
</p>
<p>n
</p>
<p>0.0 0.2 0.4 0.6 0.8 1.0
</p>
<p>-4
-2
</p>
<p>0
2
</p>
<p>4
</p>
<p>spline
wavelet
true
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
&bull;
&bull;&bull;&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
&bull;
&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
&bull;
&bull;
</p>
<p>&bull;
&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;&bull;&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
&bull;
</p>
<p>&bull;&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;&bull;
</p>
<p>&bull;
&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;&bull;
&bull;
&bull;
</p>
<p>&bull;
&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>FIGURE 5.19. Wavelet smoothing compared with smoothing splines on two
examples. Each panel compares the SURE-shrunk wavelet fit to the cross-validated
smoothing spline fit.</p>
<p/>
</div>
<div class="page"><p/>
<p>Exercises 183
</p>
<p>Ex. 5.2 Suppose that Bi,M (x) is an order-M B-spline defined in the Ap-
pendix on page 186 through the sequence (5.77)&ndash;(5.78).
</p>
<p>(a) Show by induction that Bi,M (x) = 0 for x �&isin; [τi, τi+M ]. This shows, for
example, that the support of cubic B-splines is at most 5 knots.
</p>
<p>(b) Show by induction that Bi,M (x) &gt; 0 for x &isin; (τi, τi+M ). The B-splines
are positive in the interior of their support.
</p>
<p>(c) Show by induction that
&sum;K+M
</p>
<p>i=1 Bi,M (x) = 1&forall;x &isin; [ξ0, ξK+1].
</p>
<p>(d) Show that Bi,M is a piecewise polynomial of order M (degree M &minus; 1)
on [ξ0, ξK+1], with breaks only at the knots ξ1, . . . , ξK .
</p>
<p>(e) Show that an order-M B-spline basis function is the density function
of a convolution of M uniform random variables.
</p>
<p>Ex. 5.3 Write a program to reproduce Figure 5.3 on page 145.
</p>
<p>Ex. 5.4 Consider the truncated power series representation for cubic splines
with K interior knots. Let
</p>
<p>f(X) =
3&sum;
</p>
<p>j=0
</p>
<p>βjX
j +
</p>
<p>K&sum;
</p>
<p>k=1
</p>
<p>θk(X &minus; ξk)3+. (5.70)
</p>
<p>Prove that the natural boundary conditions for natural cubic splines (Sec-
tion 5.2.1) imply the following linear constraints on the coefficients:
</p>
<p>β2 = 0,
&sum;K
</p>
<p>k=1 θk = 0,
</p>
<p>β3 = 0,
&sum;K
</p>
<p>k=1 ξkθk = 0.
(5.71)
</p>
<p>Hence derive the basis (5.4) and (5.5).
</p>
<p>Ex. 5.5 Write a program to classify the phoneme data using a quadratic dis-
criminant analysis (Section 4.3). Since there are many correlated features,
you should filter them using a smooth basis of natural cubic splines (Sec-
tion 5.2.3). Decide beforehand on a series of five different choices for the
number and position of the knots, and use tenfold cross-validation to make
the final selection. The phoneme data are available from the book website
www-stat.stanford.edu/ElemStatLearn.
</p>
<p>Ex. 5.6 Suppose you wish to fit a periodic function, with a known period T .
Describe how you could modify the truncated power series basis to achieve
this goal.
</p>
<p>Ex. 5.7 Derivation of smoothing splines (Green and Silverman, 1994). Sup-
pose that N &ge; 2, and that g is the natural cubic spline interpolant to the
pairs {xi, zi}N1 , with a &lt; x1 &lt; &middot; &middot; &middot; &lt; xN &lt; b. This is a natural spline</p>
<p/>
</div>
<div class="page"><p/>
<p>184 5. Basis Expansions and Regularization
</p>
<p>with a knot at every xi; being an N -dimensional space of functions, we can
determine the coefficients such that it interpolates the sequence zi exactly.
Let g̃ be any other differentiable function on [a, b] that interpolates the N
pairs.
</p>
<p>(a) Let h(x) = g̃(x)&minus; g(x). Use integration by parts and the fact that g is
a natural cubic spline to show that
</p>
<p>&int; b
</p>
<p>a
</p>
<p>g&prime;&prime;(x)h&prime;&prime;(x)dx = &minus;
N&minus;1&sum;
</p>
<p>j=1
</p>
<p>g&prime;&prime;&prime;(x+j ){h(xj+1)&minus; h(xj)} (5.72)
</p>
<p>= 0.
</p>
<p>(b) Hence show that &int; b
</p>
<p>a
</p>
<p>g̃&prime;&prime;(t)
2
dt &ge;
</p>
<p>&int; b
</p>
<p>a
</p>
<p>g&prime;&prime;(t)
2
dt,
</p>
<p>and that equality can only hold if h is identically zero in [a, b].
</p>
<p>(c) Consider the penalized least squares problem
</p>
<p>min
f
</p>
<p>[
N&sum;
</p>
<p>i=1
</p>
<p>(yi &minus; f(xi))2 + λ
&int; b
</p>
<p>a
</p>
<p>f &prime;&prime;(t)2dt
</p>
<p>]
.
</p>
<p>Use (b) to argue that the minimizer must be a cubic spline with knots
at each of the xi.
</p>
<p>Ex. 5.8 In the appendix to this chapter we show how the smoothing spline
computations could be more efficiently carried out using a (N + 4) dimen-
sional basis of B-splines. Describe a slightly simpler scheme using a (N+2)
dimensional B-spline basis defined on the N &minus; 2 interior knots.
Ex. 5.9 Derive the Reinsch form Sλ = (I+λK)
</p>
<p>&minus;1 for the smoothing spline.
</p>
<p>Ex. 5.10 Derive an expression for Var(f̂λ(x0)) and bias(f̂λ(x0)). Using the
example (5.22), create a version of Figure 5.9 where the mean and several
</p>
<p>(pointwise) quantiles of f̂λ(x) are shown.
</p>
<p>Ex. 5.11 Prove that for a smoothing spline the null space of K is spanned
by functions linear in X.
</p>
<p>Ex. 5.12 Characterize the solution to the following problem,
</p>
<p>min
f
</p>
<p>RSS(f, λ) =
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>wi{yi &minus; f(xi)}2 + λ
&int;
{f &prime;&prime;(t)}2dt, (5.73)
</p>
<p>where the wi &ge; 0 are observation weights.
Characterize the solution to the smoothing spline problem (5.9) when
</p>
<p>the training data have ties in X.</p>
<p/>
</div>
<div class="page"><p/>
<p>Exercises 185
</p>
<p>Ex. 5.13 You have fitted a smoothing spline f̂λ to a sample of N pairs
(xi, yi). Suppose you augment your original sample with the pair x0, f̂λ(x0),
and refit; describe the result. Use this to derive the N -fold cross-validation
formula (5.26).
</p>
<p>Ex. 5.14 Derive the constraints on the αj in the thin-plate spline expan-
sion (5.39) to guarantee that the penalty J(f) is finite. How else could one
ensure that the penalty was finite?
</p>
<p>Ex. 5.15 This exercise derives some of the results quoted in Section 5.8.1.
Suppose K(x, y) satisfying the conditions (5.45) and let f(x) &isin; HK . Show
that
</p>
<p>(a) 〈K(&middot;, xi), f〉HK = f(xi).
</p>
<p>(b) 〈K(&middot;, xi),K(&middot;, xj)〉HK = K(xi, xj).
</p>
<p>(c) If g(x) =
&sum;N
</p>
<p>i=1 αiK(x, xi), then
</p>
<p>J(g) =
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>N&sum;
</p>
<p>j=1
</p>
<p>K(xi, xj)αiαj .
</p>
<p>Suppose that g̃(x) = g(x) + ρ(x), with ρ(x) &isin; HK , and orthogonal in HK
to each of K(x, xi), i = 1, . . . , N . Show that
</p>
<p>(d)
N&sum;
</p>
<p>i=1
</p>
<p>L(yi, g̃(xi)) + λJ(g̃) &ge;
N&sum;
</p>
<p>i=1
</p>
<p>L(yi, g(xi)) + λJ(g) (5.74)
</p>
<p>with equality iff ρ(x) = 0.
</p>
<p>Ex. 5.16 Consider the ridge regression problem (5.53), and assume M &ge; N .
Assume you have a kernel K that computes the inner product K(x, y) =&sum;M
</p>
<p>m=1 hm(x)hm(y).
</p>
<p>(a) Derive (5.62) on page 171 in the text. How would you compute the
matrices V and Dγ , given K? Hence show that (5.63) is equivalent
to (5.53).
</p>
<p>(b) Show that
</p>
<p>f̂ = Hβ̂
</p>
<p>= K(K+ λI)&minus;1y, (5.75)
</p>
<p>where H is the N &times;M matrix of evaluations hm(xi), and K = HHT
the N &times;N matrix of inner-products h(xi)Th(xj).</p>
<p/>
</div>
<div class="page"><p/>
<p>186 5. Basis Expansions and Regularization
</p>
<p>(c) Show that
</p>
<p>f̂(x) = h(x)T β̂
</p>
<p>=
N&sum;
</p>
<p>i=1
</p>
<p>K(x, xi)α̂i (5.76)
</p>
<p>and α̂ = (K+ λI)&minus;1y.
</p>
<p>(d) How would you modify your solution if M &lt; N?
</p>
<p>Ex. 5.17 Show how to convert the discrete eigen-decomposition of K in
Section 5.8.2 to estimates of the eigenfunctions of K.
</p>
<p>Ex. 5.18 The wavelet function ψ(x) of the symmlet-p wavelet basis has
vanishing moments up to order p. Show that this implies that polynomials
of order p are represented exactly in V0, defined on page 176.
</p>
<p>Ex. 5.19 Show that the Haar wavelet transform of a signal of length N = 2J
</p>
<p>can be computed in O(N) computations.
</p>
<p>Appendix: Computations for Splines
</p>
<p>In this Appendix, we describe the B-spline basis for representing polyno-
mial splines. We also discuss their use in the computations of smoothing
splines.
</p>
<p>B-splines
</p>
<p>Before we can get started, we need to augment the knot sequence defined
in Section 5.2. Let ξ0 &lt; ξ1 and ξK &lt; ξK+1 be two boundary knots, which
typically define the domain over which we wish to evaluate our spline. We
now define the augmented knot sequence τ such that
</p>
<p>&bull; τ1 &le; τ2 &le; &middot; &middot; &middot; &le; τM &le; ξ0;
</p>
<p>&bull; τj+M = ξj , j = 1, &middot; &middot; &middot; ,K;
</p>
<p>&bull; ξK+1 &le; τK+M+1 &le; τK+M+2 &le; &middot; &middot; &middot; &le; τK+2M .
</p>
<p>The actual values of these additional knots beyond the boundary are arbi-
trary, and it is customary to make them all the same and equal to ξ0 and
ξK+1, respectively.
Denote by Bi,m(x) the ith B-spline basis function of order m for the
</p>
<p>knot-sequence τ , m &le; M . They are defined recursively in terms of divided</p>
<p/>
</div>
<div class="page"><p/>
<p>Appendix: Computations for Splines 187
</p>
<p>differences as follows:
</p>
<p>Bi,1(x) =
</p>
<p>{
1 if τi &le; x &lt; τi+1
0 otherwise
</p>
<p>(5.77)
</p>
<p>for i = 1, . . . ,K + 2M &minus; 1. These are also known as Haar basis functions.
</p>
<p>Bi,m(x) =
x&minus; τi
</p>
<p>τi+m&minus;1 &minus; τi
Bi,m&minus;1(x) +
</p>
<p>τi+m &minus; x
τi+m &minus; τi+1
</p>
<p>Bi+1,m&minus;1(x)
</p>
<p>for i = 1, . . . ,K + 2M &minus;m.
(5.78)
</p>
<p>Thus with M = 4, Bi,4, i = 1, &middot; &middot; &middot; ,K + 4 are the K + 4 cubic B-spline
basis functions for the knot sequence ξ. This recursion can be contin-
ued and will generate the B-spline basis for any order spline. Figure 5.20
shows the sequence of B-splines up to order four with knots at the points
0.0, 0.1, . . . , 1.0. Since we have created some duplicate knots, some care
has to be taken to avoid division by zero. If we adopt the convention
that Bi,1 = 0 if τi = τi+1, then by induction Bi,m = 0 if τi = τi+1 =
. . . = τi+m. Note also that in the construction above, only the subset
Bi,m, i = M &minus; m + 1, . . . ,M + K are required for the B-spline basis
of order m &lt; M with knots ξ.
</p>
<p>To fully understand the properties of these functions, and to show that
they do indeed span the space of cubic splines for the knot sequence, re-
quires additional mathematical machinery, including the properties of di-
vided differences. Exercise 5.2 explores these issues.
The scope of B-splines is in fact bigger than advertised here, and has to
</p>
<p>do with knot duplication. If we duplicate an interior knot in the construc-
tion of the τ sequence above, and then generate the B-spline sequence as
before, the resulting basis spans the space of piecewise polynomials with
one less continuous derivative at the duplicated knot. In general, if in ad-
dition to the repeated boundary knots, we include the interior knot ξj
1 &le; rj &le; M times, then the lowest-order derivative to be discontinuous
at x = ξj will be order M &minus; rj . Thus for cubic splines with no repeats,
rj = 1, j = 1, . . . ,K, and at each interior knot the third derivatives (4&minus;1)
are discontinuous. Repeating the jth knot three times leads to a discontin-
uous 1st derivative; repeating it four times leads to a discontinuous zeroth
derivative, i.e., the function is discontinuous at x = ξj . This is exactly what
happens at the boundary knots; we repeat the knots M times, so the spline
becomes discontinuous at the boundary knots (i.e., undefined beyond the
boundary).
The local support of B-splines has important computational implica-
</p>
<p>tions, especially when the number of knots K is large. Least squares com-
putations with N observations and K +M variables (basis functions) take
O(N(K +M)2 + (K +M)3) flops (floating point operations.) If K is some
appreciable fraction of N , this leads to O(N3) algorithms which becomes</p>
<p/>
</div>
<div class="page"><p/>
<p>188 5. Basis Expansions and Regularization
</p>
<p>B-splines of Order 1
</p>
<p>0.0 0.2 0.4 0.6 0.8 1.0
</p>
<p>0
.0
</p>
<p>0
.4
</p>
<p>0
.8
</p>
<p>1
.2
</p>
<p>B-splines of Order 2
</p>
<p>0.0 0.2 0.4 0.6 0.8 1.0
</p>
<p>0
.0
</p>
<p>0
.4
</p>
<p>0
.8
</p>
<p>1
.2
</p>
<p>B-splines of Order 3
</p>
<p>0.0 0.2 0.4 0.6 0.8 1.0
</p>
<p>0
.0
</p>
<p>0
.4
</p>
<p>0
.8
</p>
<p>1
.2
</p>
<p>B-splines of Order 4
</p>
<p>0.0 0.2 0.4 0.6 0.8 1.0
</p>
<p>0
.0
</p>
<p>0
.4
</p>
<p>0
.8
</p>
<p>1
.2
</p>
<p>FIGURE 5.20. The sequence of B-splines up to order four with ten knots evenly
spaced from 0 to 1. The B-splines have local support; they are nonzero on an
interval spanned by M + 1 knots.</p>
<p/>
</div>
<div class="page"><p/>
<p>Appendix: Computations for Splines 189
</p>
<p>unacceptable for large N . If the N observations are sorted, the N&times;(K+M)
regression matrix consisting of the K +M B-spline basis functions evalu-
ated at the N points has many zeros, which can be exploited to reduce the
computational complexity back to O(N). We take this up further in the
next section.
</p>
<p>Computations for Smoothing Splines
</p>
<p>Although natural splines (Section 5.2.1) provide a basis for smoothing
splines, it is computationally more convenient to operate in the larger space
of unconstrained B-splines. We write f(x) =
</p>
<p>&sum;N+4
1 γjBj(x), where γj are
</p>
<p>coefficients and the Bj are the cubic B-spline basis functions. The solution
looks the same as before,
</p>
<p>γ̂ = (BTB+ λΩB)
&minus;1BTy, (5.79)
</p>
<p>except now the N &times; N matrix N is replaced by the N &times; (N + 4) matrix
B, and similarly the (N + 4) &times; (N + 4) penalty matrix ΩB replaces the
N &times; N dimensional ΩN . Although at face value it seems that there are
no boundary derivative constraints, it turns out that the penalty term
automatically imposes them by giving effectively infinite weight to any non
zero derivative beyond the boundary. In practice, γ̂ is restricted to a linear
subspace for which the penalty is always finite.
Since the columns of B are the evaluated B-splines, in order from left
</p>
<p>to right and evaluated at the sorted values of X, and the cubic B-splines
have local support, B is lower 4-banded. Consequently the matrix M =
(BTB+λΩ) is 4-banded and hence its Cholesky decomposition M = LLT
</p>
<p>can be computed easily. One then solves LLTγ = BTy by back-substitution
to give γ and hence the solution f̂ in O(N) operations.
</p>
<p>In practice, when N is large, it is unnecessary to use all N interior knots,
and any reasonable thinning strategy will save in computations and have
negligible effect on the fit. For example, the smooth.spline function in S-
PLUS uses an approximately logarithmic strategy: if N &lt; 50 all knots are
included, but even at N = 5, 000 only 204 knots are used.</p>
<p/>
</div>
<div class="page"><p/>
<p>6
Kernel Smoothing Methods
</p>
<p>In this chapter we describe a class of regression techniques that achieve
flexibility in estimating the regression function f(X) over the domain IRp
</p>
<p>by fitting a different but simple model separately at each query point x0.
This is done by using only those observations close to the target point x0 to
fit the simple model, and in such a way that the resulting estimated function
f̂(X) is smooth in IRp. This localization is achieved via a weighting function
or kernel Kλ(x0, xi), which assigns a weight to xi based on its distance from
x0. The kernels Kλ are typically indexed by a parameter λ that dictates
the width of the neighborhood. These memory-based methods require in
principle little or no training; all the work gets done at evaluation time.
The only parameter that needs to be determined from the training data is
λ. The model, however, is the entire training data set.
We also discuss more general classes of kernel-based techniques , which
</p>
<p>tie in with structured methods in other chapters, and are useful for density
estimation and classification.
The techniques in this chapter should not be confused with those asso-
</p>
<p>ciated with the more recent usage of the phrase &ldquo;kernel methods&rdquo;. In this
chapter kernels are mostly used as a device for localization. We discuss ker-
nel methods in Sections 5.8, 14.5.4, 18.5 and Chapter 12; in those contexts
the kernel computes an inner product in a high-dimensional (implicit) fea-
ture space, and is used for regularized nonlinear modeling. We make some
connections to the methodology in this chapter at the end of Section 6.7.
</p>
<p> &copy; Springer Science+Business Media, LLC 2009 
</p>
<p>T. Hastie et al., The Elements of Statistical Learning, Second Edition, 191
</p>
<p>DOI: 10.1007/b94608_6,</p>
<p/>
</div>
<div class="page"><p/>
<p>192 6. Kernel Smoothing Methods
</p>
<p>Nearest-Neighbor Kernel
</p>
<p>0.0 0.2 0.4 0.6 0.8 1.0
</p>
<p>-1
.0
</p>
<p>-0
.5
</p>
<p>0
.0
</p>
<p>0
.5
</p>
<p>1
.0
</p>
<p>1
.5
</p>
<p>O
</p>
<p>O
</p>
<p>OO
</p>
<p>OO
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>OOO
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
O
</p>
<p>O
</p>
<p>OO
</p>
<p>OO
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O O
</p>
<p>O
</p>
<p>OO
</p>
<p>OO
</p>
<p>O
</p>
<p>OO
</p>
<p>O
O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
O
</p>
<p>O
</p>
<p>OO
</p>
<p>OO
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>&bull;
</p>
<p>x0
</p>
<p>f̂(x0)
</p>
<p>Epanechnikov Kernel
</p>
<p>0.0 0.2 0.4 0.6 0.8 1.0
</p>
<p>-1
.0
</p>
<p>-0
.5
</p>
<p>0
.0
</p>
<p>0
.5
</p>
<p>1
.0
</p>
<p>1
.5
</p>
<p>O
</p>
<p>O
</p>
<p>OO
</p>
<p>OO
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>OOO
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
O
</p>
<p>O
</p>
<p>OO
</p>
<p>OO
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O O
</p>
<p>O
</p>
<p>OO
</p>
<p>OO
</p>
<p>O
</p>
<p>OO
</p>
<p>O
O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
O
</p>
<p>O
</p>
<p>OO
</p>
<p>OO
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O O
</p>
<p>O
</p>
<p>OO
</p>
<p>&bull;
</p>
<p>x0
</p>
<p>f̂(x0)
</p>
<p>FIGURE 6.1. In each panel 100 pairs xi, yi are generated at random from the
blue curve with Gaussian errors: Y = sin(4X)+ε, X &sim; U [0, 1], ε &sim; N(0, 1/3). In
the left panel the green curve is the result of a 30-nearest-neighbor running-mean
smoother. The red point is the fitted constant f̂(x0), and the red circles indicate
those observations contributing to the fit at x0. The solid yellow region indicates
the weights assigned to observations. In the right panel, the green curve is the
kernel-weighted average, using an Epanechnikov kernel with (half) window width
λ = 0.2.
</p>
<p>6.1 One-Dimensional Kernel Smoothers
</p>
<p>In Chapter 2, we motivated the k&ndash;nearest-neighbor average
</p>
<p>f̂(x) = Ave(yi|xi &isin; Nk(x)) (6.1)
</p>
<p>as an estimate of the regression function E(Y |X = x). Here Nk(x) is the set
of k points nearest to x in squared distance, and Ave denotes the average
(mean). The idea is to relax the definition of conditional expectation, as
illustrated in the left panel of Figure 6.1, and compute an average in a
neighborhood of the target point. In this case we have used the 30-nearest
neighborhood&mdash;the fit at x0 is the average of the 30 pairs whose xi values
are closest to x0. The green curve is traced out as we apply this definition
at different values x0. The green curve is bumpy, since f̂(x) is discontinuous
in x. As we move x0 from left to right, the k-nearest neighborhood remains
constant, until a point xi to the right of x0 becomes closer than the furthest
point xi&prime; in the neighborhood to the left of x0, at which time xi replaces xi&prime; .
The average in (6.1) changes in a discrete way, leading to a discontinuous
</p>
<p>f̂(x).
This discontinuity is ugly and unnecessary. Rather than give all the
</p>
<p>points in the neighborhood equal weight, we can assign weights that die
off smoothly with distance from the target point. The right panel shows
an example of this, using the so-called Nadaraya&ndash;Watson kernel-weighted</p>
<p/>
</div>
<div class="page"><p/>
<p>6.1 One-Dimensional Kernel Smoothers 193
</p>
<p>average
</p>
<p>f̂(x0) =
</p>
<p>&sum;N
i=1 Kλ(x0, xi)yi&sum;N
i=1 Kλ(x0, xi)
</p>
<p>, (6.2)
</p>
<p>with the Epanechnikov quadratic kernel
</p>
<p>Kλ(x0, x) = D
</p>
<p>( |x&minus; x0|
λ
</p>
<p>)
, (6.3)
</p>
<p>with
</p>
<p>D(t) =
</p>
<p>{
3
4 (1&minus; t2) if |t| &le; 1;
0 otherwise.
</p>
<p>(6.4)
</p>
<p>The fitted function is now continuous, and quite smooth in the right panel
of Figure 6.1. As we move the target from left to right, points enter the
neighborhood initially with weight zero, and then their contribution slowly
increases (see Exercise 6.1).
In the right panel we used a metric window size λ = 0.2 for the kernel
</p>
<p>fit, which does not change as we move the target point x0, while the size
of the 30-nearest-neighbor smoothing window adapts to the local density
of the xi. One can, however, also use such adaptive neighborhoods with
kernels, but we need to use a more general notation. Let hλ(x0) be a width
function (indexed by λ) that determines the width of the neighborhood at
x0. Then more generally we have
</p>
<p>Kλ(x0, x) = D
</p>
<p>( |x&minus; x0|
hλ(x0)
</p>
<p>)
. (6.5)
</p>
<p>In (6.3), hλ(x0) = λ is constant. For k-nearest neighborhoods, the neigh-
borhood size k replaces λ, and we have hk(x0) = |x0 &minus; x[k]| where x[k] is
the kth closest xi to x0.
There are a number of details that one has to attend to in practice:
</p>
<p>&bull; The smoothing parameter λ, which determines the width of the local
neighborhood, has to be determined. Large λ implies lower variance
(averages over more observations) but higher bias (we essentially as-
sume the true function is constant within the window).
</p>
<p>&bull; Metric window widths (constant hλ(x)) tend to keep the bias of the
estimate constant, but the variance is inversely proportional to the
local density. Nearest-neighbor window widths exhibit the opposite
behavior; the variance stays constant and the absolute bias varies
inversely with local density.
</p>
<p>&bull; Issues arise with nearest-neighbors when there are ties in the xi. With
most smoothing techniques one can simply reduce the data set by
averaging the yi at tied values of X, and supplementing these new
observations at the unique values of xi with an additional weight wi
(which multiples the kernel weight).</p>
<p/>
</div>
<div class="page"><p/>
<p>194 6. Kernel Smoothing Methods
</p>
<p>-3 -2 -1 0 1 2 3
</p>
<p>0
.0
</p>
<p>0
.4
</p>
<p>0
.8
</p>
<p>Epanechnikov
</p>
<p>Tri-cube
</p>
<p>Gaussian
K
</p>
<p>λ
(x
</p>
<p>0
,
x
)
</p>
<p>FIGURE 6.2. A comparison of three popular kernels for local smoothing. Each
has been calibrated to integrate to 1. The tri-cube kernel is compact and has two
continuous derivatives at the boundary of its support, while the Epanechnikov ker-
nel has none. The Gaussian kernel is continuously differentiable, but has infinite
support.
</p>
<p>&bull; This leaves a more general problem to deal with: observation weights
wi. Operationally we simply multiply them by the kernel weights be-
fore computing the weighted average. With nearest neighborhoods, it
is now natural to insist on neighborhoods with a total weight content
k (relative to
</p>
<p>&sum;
wi). In the event of overflow (the last observation
</p>
<p>needed in a neighborhood has a weight wj which causes the sum of
weights to exceed the budget k), then fractional parts can be used.
</p>
<p>&bull; Boundary issues arise. The metric neighborhoods tend to contain less
points on the boundaries, while the nearest-neighborhoods get wider.
</p>
<p>&bull; The Epanechnikov kernel has compact support (needed when used
with nearest-neighbor window size). Another popular compact kernel
is based on the tri-cube function
</p>
<p>D(t) =
</p>
<p>{
(1&minus; |t|3)3 if |t| &le; 1;
0 otherwise
</p>
<p>(6.6)
</p>
<p>This is flatter on the top (like the nearest-neighbor box) and is differ-
entiable at the boundary of its support. The Gaussian density func-
tion D(t) = φ(t) is a popular noncompact kernel, with the standard-
deviation playing the role of the window size. Figure 6.2 compares
the three.
</p>
<p>6.1.1 Local Linear Regression
</p>
<p>We have progressed from the raw moving average to a smoothly varying
locally weighted average by using kernel weighting. The smooth kernel fit
still has problems, however, as exhibited in Figure 6.3 (left panel). Locally-
weighted averages can be badly biased on the boundaries of the domain,</p>
<p/>
</div>
<div class="page"><p/>
<p>6.1 One-Dimensional Kernel Smoothers 195
</p>
<p>N-W Kernel at Boundary
</p>
<p>0.0 0.2 0.4 0.6 0.8 1.0
</p>
<p>-1
.0
</p>
<p>-0
.5
</p>
<p>0
.0
</p>
<p>0
.5
</p>
<p>1
.0
</p>
<p>1
.5
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>OO
O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>OO
O
</p>
<p>O
</p>
<p>OO
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>OO
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O O
</p>
<p>O
O
O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>OO
O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>&bull;
</p>
<p>x0
</p>
<p>f̂(x0)
</p>
<p>Local Linear Regression at Boundary
</p>
<p>0.0 0.2 0.4 0.6 0.8 1.0
</p>
<p>-1
.0
</p>
<p>-0
.5
</p>
<p>0
.0
</p>
<p>0
.5
</p>
<p>1
.0
</p>
<p>1
.5
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>OO
O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>OO
O
</p>
<p>O
</p>
<p>OO
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>OO
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O O
</p>
<p>O
O
O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>OO
O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>&bull;
</p>
<p>x0
</p>
<p>f̂(x0)
</p>
<p>FIGURE 6.3. The locally weighted average has bias problems at or near the
boundaries of the domain. The true function is approximately linear here, but
most of the observations in the neighborhood have a higher mean than the target
point, so despite weighting, their mean will be biased upwards. By fitting a locally
weighted linear regression (right panel), this bias is removed to first order.
</p>
<p>because of the asymmetry of the kernel in that region. By fitting straight
lines rather than constants locally, we can remove this bias exactly to first
order; see Figure 6.3 (right panel). Actually, this bias can be present in the
interior of the domain as well, if the X values are not equally spaced (for
the same reasons, but usually less severe). Again locally weighted linear
regression will make a first-order correction.
Locally weighted regression solves a separate weighted least squares prob-
</p>
<p>lem at each target point x0:
</p>
<p>min
α(x0),β(x0)
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>Kλ(x0, xi) [yi &minus; α(x0)&minus; β(x0)xi]2 . (6.7)
</p>
<p>The estimate is then f̂(x0) = α̂(x0) + β̂(x0)x0. Notice that although we fit
an entire linear model to the data in the region, we only use it to evaluate
the fit at the single point x0.
Define the vector-valued function b(x)T = (1, x). Let B be the N &times; 2
</p>
<p>regression matrix with ith row b(xi)
T , and W(x0) the N &times; N diagonal
</p>
<p>matrix with ith diagonal element Kλ(x0, xi). Then
</p>
<p>f̂(x0) = b(x0)
T (BTW(x0)B)
</p>
<p>&minus;1BTW(x0)y (6.8)
</p>
<p>=
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>li(x0)yi. (6.9)
</p>
<p>Equation (6.8) gives an explicit expression for the local linear regression
estimate, and (6.9) highlights the fact that the estimate is linear in the</p>
<p/>
</div>
<div class="page"><p/>
<p>196 6. Kernel Smoothing Methods
</p>
<p>Local Linear Equivalent Kernel at Boundary
</p>
<p>0.0 0.2 0.4 0.6 0.8 1.0
</p>
<p>-1
.0
</p>
<p>-0
.5
</p>
<p>0
.0
</p>
<p>0
.5
</p>
<p>1
.0
</p>
<p>1
.5
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>OO
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>OO
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>OO
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>OO
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>OO
</p>
<p>OO
O O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>OO
</p>
<p>OO
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>OO
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>OO
O
</p>
<p>&bull;
</p>
<p>&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;
&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull; &bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull; &bull; &bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull; &bull;&bull;&bull;&bull;&bull;&bull; &bull; &bull;&bull; &bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
</p>
<p>x0
</p>
<p>f̂(x0)
</p>
<p>Local Linear Equivalent Kernel in Interior
</p>
<p>0.0 0.2 0.4 0.6 0.8 1.0
</p>
<p>-1
.0
</p>
<p>-0
.5
</p>
<p>0
.0
</p>
<p>0
.5
</p>
<p>1
.0
</p>
<p>1
.5
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>OO
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>OO
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>OO
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>OO
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>OO
</p>
<p>OO
O O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>OO
</p>
<p>OO
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>OO
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>OO
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>OO
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>OO
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>&bull;
</p>
<p>&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull; &bull;&bull;
&bull;
&bull;&bull;&bull;
&bull;&bull;
&bull;&bull;&bull;&bull;
&bull;&bull;
&bull;&bull;&bull;&bull;
</p>
<p>&bull;&bull;&bull;
&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;
</p>
<p>&bull; &bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;
</p>
<p>&bull;
&bull;&bull;
</p>
<p>&bull;
&bull;
&bull;&bull;&bull;&bull;&bull;&bull; &bull;&bull;&bull;&bull;&bull;&bull; &bull; &bull;&bull; &bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
</p>
<p>x0
</p>
<p>f̂(x0)
</p>
<p>FIGURE 6.4. The green points show the equivalent kernel li(x0) for local re-
gression. These are the weights in f̂(x0) =
</p>
<p>&sum;N
</p>
<p>i=1 li(x0)yi, plotted against their
corresponding xi. For display purposes, these have been rescaled, since in fact
they sum to 1. Since the yellow shaded region is the (rescaled) equivalent kernel
for the Nadaraya&ndash;Watson local average, we see how local regression automati-
cally modifies the weighting kernel to correct for biases due to asymmetry in the
smoothing window.
</p>
<p>yi (the li(x0) do not involve y). These weights li(x0) combine the weight-
ing kernel Kλ(x0, &middot;) and the least squares operations, and are sometimes
referred to as the equivalent kernel. Figure 6.4 illustrates the effect of lo-
cal linear regression on the equivalent kernel. Historically, the bias in the
Nadaraya&ndash;Watson and other local average kernel methods were corrected
by modifying the kernel. These modifications were based on theoretical
asymptotic mean-square-error considerations, and besides being tedious to
implement, are only approximate for finite sample sizes. Local linear re-
gression automatically modifies the kernel to correct the bias exactly to
first order, a phenomenon dubbed as automatic kernel carpentry. Consider
the following expansion for Ef̂(x0), using the linearity of local regression
and a series expansion of the true function f around x0,
</p>
<p>Ef̂(x0) =
N&sum;
</p>
<p>i=1
</p>
<p>li(x0)f(xi)
</p>
<p>= f(x0)
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>li(x0) + f
&prime;(x0)
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>(xi &minus; x0)li(x0)
</p>
<p>+
f &prime;&prime;(x0)
</p>
<p>2
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>(xi &minus; x0)2li(x0) +R, (6.10)
</p>
<p>where the remainder term R involves third- and higher-order derivatives of
f , and is typically small under suitable smoothness assumptions. It can be</p>
<p/>
</div>
<div class="page"><p/>
<p>6.1 One-Dimensional Kernel Smoothers 197
</p>
<p>Local Linear in Interior
</p>
<p>0.0 0.2 0.4 0.6 0.8 1.0
</p>
<p>-1
.0
</p>
<p>-0
.5
</p>
<p>0
.0
</p>
<p>0
.5
</p>
<p>1
.0
</p>
<p>1
.5
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>OO
OO
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>OO
</p>
<p>O
</p>
<p>O
</p>
<p>OO
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
O
</p>
<p>OO
</p>
<p>OO
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>OO
</p>
<p>O
</p>
<p>O
</p>
<p>OO
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
O
O
</p>
<p>O
</p>
<p>&bull;
f̂(x0)
</p>
<p>Local Quadratic in Interior
</p>
<p>0.0 0.2 0.4 0.6 0.8 1.0
</p>
<p>-1
.0
</p>
<p>-0
.5
</p>
<p>0
.0
</p>
<p>0
.5
</p>
<p>1
.0
</p>
<p>1
.5
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>OO
OO
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>OO
</p>
<p>O
</p>
<p>O
</p>
<p>OO
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
O
</p>
<p>OO
</p>
<p>OO
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>OO
</p>
<p>O
</p>
<p>O
</p>
<p>OO
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
O
O
</p>
<p>O
</p>
<p>&bull;
f̂(x0)
</p>
<p>FIGURE 6.5. Local linear fits exhibit bias in regions of curvature of the true
function. Local quadratic fits tend to eliminate this bias.
</p>
<p>shown (Exercise 6.2) that for local linear regression,
&sum;N
</p>
<p>i=1 li(x0) = 1 and&sum;N
i=1(xi &minus; x0)li(x0) = 0. Hence the middle term equals f(x0), and since
</p>
<p>the bias is Ef̂(x0) &minus; f(x0), we see that it depends only on quadratic and
higher&ndash;order terms in the expansion of f .
</p>
<p>6.1.2 Local Polynomial Regression
</p>
<p>Why stop at local linear fits? We can fit local polynomial fits of any de-
gree d,
</p>
<p>min
α(x0),βj(x0), j=1,...,d
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>Kλ(x0, xi)
</p>
<p>⎡
⎣yi &minus; α(x0)&minus;
</p>
<p>d&sum;
</p>
<p>j=1
</p>
<p>βj(x0)x
j
i
</p>
<p>⎤
⎦
2
</p>
<p>(6.11)
</p>
<p>with solution f̂(x0) = α̂(x0)+
&sum;d
</p>
<p>j=1 β̂j(x0)x
j
0. In fact, an expansion such as
</p>
<p>(6.10) will tell us that the bias will only have components of degree d+1 and
higher (Exercise 6.2). Figure 6.5 illustrates local quadratic regression. Local
linear fits tend to be biased in regions of curvature of the true function, a
phenomenon referred to as trimming the hills and filling the valleys. Local
quadratic regression is generally able to correct this bias.
There is of course a price to be paid for this bias reduction, and that is
</p>
<p>increased variance. The fit in the right panel of Figure 6.5 is slightly more
wiggly, especially in the tails. Assuming the model yi = f(xi) + εi, with
εi independent and identically distributed with mean zero and variance
σ2, Var(f̂(x0)) = σ
</p>
<p>2||l(x0)||2, where l(x0) is the vector of equivalent kernel
weights at x0. It can be shown (Exercise 6.3) that ||l(x0)|| increases with d,
and so there is a bias&ndash;variance tradeoff in selecting the polynomial degree.
Figure 6.6 illustrates these variance curves for degree zero, one and two</p>
<p/>
</div>
<div class="page"><p/>
<p>198 6. Kernel Smoothing Methods
</p>
<p>V
a
ri
a
n
c
e
</p>
<p>0.0 0.2 0.4 0.6 0.8 1.0
</p>
<p>0
.0
</p>
<p>0
.1
</p>
<p>0
.2
</p>
<p>0
.3
</p>
<p>0
.4
</p>
<p>0
.5
</p>
<p>Constant
</p>
<p>Linear
</p>
<p>Quadratic
</p>
<p>FIGURE 6.6. The variances functions ||l(x)||2 for local constant, linear and
quadratic regression, for a metric bandwidth (λ = 0.2) tri-cube kernel.
</p>
<p>local polynomials. To summarize some collected wisdom on this issue:
</p>
<p>&bull; Local linear fits can help bias dramatically at the boundaries at a
modest cost in variance. Local quadratic fits do little at the bound-
aries for bias, but increase the variance a lot.
</p>
<p>&bull; Local quadratic fits tend to be most helpful in reducing bias due to
curvature in the interior of the domain.
</p>
<p>&bull; Asymptotic analysis suggest that local polynomials of odd degree
dominate those of even degree. This is largely due to the fact that
asymptotically the MSE is dominated by boundary effects.
</p>
<p>While it may be helpful to tinker, and move from local linear fits at the
boundary to local quadratic fits in the interior, we do not recommend such
strategies. Usually the application will dictate the degree of the fit. For
example, if we are interested in extrapolation, then the boundary is of
more interest, and local linear fits are probably more reliable.
</p>
<p>6.2 Selecting the Width of the Kernel
</p>
<p>In each of the kernels Kλ, λ is a parameter that controls its width:
</p>
<p>&bull; For the Epanechnikov or tri-cube kernel with metric width, λ is the
radius of the support region.
</p>
<p>&bull; For the Gaussian kernel, λ is the standard deviation.
</p>
<p>&bull; λ is the number k of nearest neighbors in k-nearest neighborhoods,
often expressed as a fraction or span k/N of the total training sample.</p>
<p/>
</div>
<div class="page"><p/>
<p>6.2 Selecting the Width of the Kernel 199
</p>
<p>&bull;
&bull;
&bull;
&bull;&bull;
&bull;&bull;
&bull;&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;&bull;&bull;&bull;&bull;
&bull;
&bull;&bull;
</p>
<p>&bull;&bull;&bull;&bull;&bull;&bull; &bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull; &bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull; &bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull; &bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
</p>
<p>&bull;
&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
&bull;&bull;
&bull;&bull;&bull;&bull;&bull;
&bull;
&bull;&bull;
</p>
<p>&bull;&bull;&bull;&bull;&bull;&bull; &bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull; &bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull; &bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull; &bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull; &bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull; &bull;&bull;&bull;&bull;&bull;
&bull;
&bull;&bull;
&bull;&bull;
&bull;&bull;
&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;
</p>
<p>&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;
</p>
<p>&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull; &bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull; &bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull; &bull;
</p>
<p>&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;
&bull;&bull;
&bull;
&bull;&bull;&bull;
&bull;&bull;
&bull;&bull;&bull;
&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
</p>
<p>&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull; &bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull; &bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
</p>
<p>FIGURE 6.7. Equivalent kernels for a local linear regression smoother (tri-cube
kernel; orange) and a smoothing spline (blue), with matching degrees of freedom.
The vertical spikes indicates the target points.
</p>
<p>There is a natural bias&ndash;variance tradeoff as we change the width of the
averaging window, which is most explicit for local averages:
</p>
<p>&bull; If the window is narrow, f̂(x0) is an average of a small number of yi
close to x0, and its variance will be relatively large&mdash;close to that of
an individual yi. The bias will tend to be small, again because each
of the E(yi) = f(xi) should be close to f(x0).
</p>
<p>&bull; If the window is wide, the variance of f̂(x0) will be small relative to
the variance of any yi, because of the effects of averaging. The bias
will be higher, because we are now using observations xi further from
x0, and there is no guarantee that f(xi) will be close to f(x0).
</p>
<p>Similar arguments apply to local regression estimates, say local linear: as
the width goes to zero, the estimates approach a piecewise-linear function
that interpolates the training data1; as the width gets infinitely large, the
fit approaches the global linear least-squares fit to the data.
The discussion in Chapter 5 on selecting the regularization parameter for
</p>
<p>smoothing splines applies here, and will not be repeated. Local regression
smoothers are linear estimators; the smoother matrix in f̂ = Sλy is built up
from the equivalent kernels (6.8), and has ijth entry {Sλ}ij = li(xj). Leave-
one-out cross-validation is particularly simple (Exercise 6.7), as is general-
ized cross-validation, Cp (Exercise 6.10), and k-fold cross-validation. The
effective degrees of freedom is again defined as trace(Sλ), and can be used
to calibrate the amount of smoothing. Figure 6.7 compares the equivalent
kernels for a smoothing spline and local linear regression. The local regres-
sion smoother has a span of 40%, which results in df = trace(Sλ) = 5.86.
The smoothing spline was calibrated to have the same df, and their equiv-
alent kernels are qualitatively quite similar.
</p>
<p>1With uniformly spaced xi; with irregularly spaced xi, the behavior can deteriorate.</p>
<p/>
</div>
<div class="page"><p/>
<p>200 6. Kernel Smoothing Methods
</p>
<p>6.3 Local Regression in IRp
</p>
<p>Kernel smoothing and local regression generalize very naturally to two or
more dimensions. The Nadaraya&ndash;Watson kernel smoother fits a constant
locally with weights supplied by a p-dimensional kernel. Local linear re-
gression will fit a hyperplane locally in X, by weighted least squares, with
weights supplied by a p-dimensional kernel. It is simple to implement and
is generally preferred to the local constant fit for its superior performance
on the boundaries.
Let b(X) be a vector of polynomial terms in X of maximum degree d.
</p>
<p>For example, with d = 1 and p = 2 we get b(X) = (1, X1, X2); with d = 2
we get b(X) = (1, X1, X2, X
</p>
<p>2
1 , X
</p>
<p>2
2 , X1X2); and trivially with d = 0 we get
</p>
<p>b(X) = 1. At each x0 &isin; IRp solve
</p>
<p>min
β(x0)
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>Kλ(x0, xi)(yi &minus; b(xi)Tβ(x0))2 (6.12)
</p>
<p>to produce the fit f̂(x0) = b(x0)
T β̂(x0). Typically the kernel will be a radial
</p>
<p>function, such as the radial Epanechnikov or tri-cube kernel
</p>
<p>Kλ(x0, x) = D
</p>
<p>( ||x&minus; x0||
λ
</p>
<p>)
, (6.13)
</p>
<p>where ||&middot;|| is the Euclidean norm. Since the Euclidean norm depends on the
units in each coordinate, it makes most sense to standardize each predictor,
for example, to unit standard deviation, prior to smoothing.
While boundary effects are a problem in one-dimensional smoothing,
</p>
<p>they are a much bigger problem in two or higher dimensions, since the
fraction of points on the boundary is larger. In fact, one of the manifesta-
tions of the curse of dimensionality is that the fraction of points close to the
boundary increases to one as the dimension grows. Directly modifying the
kernel to accommodate two-dimensional boundaries becomes very messy,
especially for irregular boundaries. Local polynomial regression seamlessly
performs boundary correction to the desired order in any dimensions. Fig-
ure 6.8 illustrates local linear regression on some measurements from an
astronomical study with an unusual predictor design (star-shaped). Here
the boundary is extremely irregular, and the fitted surface must also inter-
polate over regions of increasing data sparsity as we approach the boundary.
Local regression becomes less useful in dimensions much higher than two
</p>
<p>or three. We have discussed in some detail the problems of dimensional-
ity, for example, in Chapter 2. It is impossible to simultaneously main-
tain localness (&rArr; low bias) and a sizable sample in the neighborhood (&rArr;
low variance) as the dimension increases, without the total sample size in-
</p>
<p>creasing exponentially in p. Visualization of f̂(X) also becomes difficult in
higher dimensions, and this is often one of the primary goals of smoothing.</p>
<p/>
</div>
<div class="page"><p/>
<p>6.4 Structured Local Regression Models in IRp 201
</p>
<p>East-West
</p>
<p>South-North
</p>
<p>Velocity
</p>
<p>East-West
</p>
<p>South-North
</p>
<p>Velocity
</p>
<p>FIGURE 6.8. The left panel shows three-dimensional data, where the response
is the velocity measurements on a galaxy, and the two predictors record positions
on the celestial sphere. The unusual &ldquo;star&rdquo;-shaped design indicates the way the
measurements were made, and results in an extremely irregular boundary. The
right panel shows the results of local linear regression smoothing in IR2, using a
nearest-neighbor window with 15% of the data.
</p>
<p>Although the scatter-cloud and wire-frame pictures in Figure 6.8 look at-
tractive, it is quite difficult to interpret the results except at a gross level.
From a data analysis perspective, conditional plots are far more useful.
Figure 6.9 shows an analysis of some environmental data with three pre-
</p>
<p>dictors. The trellis display here shows ozone as a function of radiation,
conditioned on the other two variables, temperature and wind speed. How-
ever, conditioning on the value of a variable really implies local to that
value (as in local regression). Above each of the panels in Figure 6.9 is an
indication of the range of values present in that panel for each of the condi-
tioning values. In the panel itself the data subsets are displayed (response
versus remaining variable), and a one-dimensional local linear regression is
fit to the data. Although this is not quite the same as looking at slices of
a fitted three-dimensional surface, it is probably more useful in terms of
understanding the joint behavior of the data.
</p>
<p>6.4 Structured Local Regression Models in IRp
</p>
<p>When the dimension to sample-size ratio is unfavorable, local regression
does not help us much, unless we are willing to make some structural as-
sumptions about the model. Much of this book is about structured regres-
sion and classification models. Here we focus on some approaches directly
related to kernel methods.</p>
<p/>
</div>
<div class="page"><p/>
<p>202 6. Kernel Smoothing Methods
</p>
<p>1
</p>
<p>2
</p>
<p>3
</p>
<p>4
</p>
<p>5
</p>
<p>Temp
Wind
</p>
<p>0 50 150 250
</p>
<p>Temp
Wind
</p>
<p>Temp
Wind
</p>
<p>0 50 150 250
</p>
<p>Temp
Wind
</p>
<p>Temp
Wind
</p>
<p>Temp
Wind
</p>
<p>Temp
Wind
</p>
<p>1
</p>
<p>2
</p>
<p>3
</p>
<p>4
</p>
<p>5
</p>
<p>Temp
Wind
</p>
<p>1
</p>
<p>2
</p>
<p>3
</p>
<p>4
</p>
<p>5
</p>
<p>Temp
Wind
</p>
<p>Temp
Wind
</p>
<p>Temp
Wind
</p>
<p>Temp
Wind
</p>
<p>Temp
Wind
</p>
<p>Temp
Wind
</p>
<p>0 50 150 250
</p>
<p>Temp
Wind
</p>
<p>1
</p>
<p>2
</p>
<p>3
</p>
<p>4
</p>
<p>5
</p>
<p>Temp
Wind
</p>
<p>0 50 150 250
</p>
<p>Solar Radiation (langleys)
</p>
<p>C
u
b
e
 R
</p>
<p>o
o
t 
O
</p>
<p>z
o
n
e
 (
</p>
<p>c
u
b
e
 r
</p>
<p>o
o
t 
p
p
b
)
</p>
<p>FIGURE 6.9. Three-dimensional smoothing example. The response is (cube-root
of) ozone concentration, and the three predictors are temperature, wind speed and
radiation. The trellis display shows ozone as a function of radiation, conditioned
on intervals of temperature and wind speed (indicated by darker green or orange
shaded bars). Each panel contains about 40% of the range of each of the condi-
tioned variables. The curve in each panel is a univariate local linear regression,
fit to the data in the panel.</p>
<p/>
</div>
<div class="page"><p/>
<p>6.4 Structured Local Regression Models in IRp 203
</p>
<p>6.4.1 Structured Kernels
</p>
<p>One line of approach is to modify the kernel. The default spherical ker-
nel (6.13) gives equal weight to each coordinate, and so a natural default
strategy is to standardize each variable to unit standard deviation. A more
general approach is to use a positive semidefinite matrix A to weigh the
different coordinates:
</p>
<p>Kλ,A(x0, x) = D
</p>
<p>(
(x&minus; x0)TA(x&minus; x0)
</p>
<p>λ
</p>
<p>)
. (6.14)
</p>
<p>Entire coordinates or directions can be downgraded or omitted by imposing
appropriate restrictions on A. For example, if A is diagonal, then we can
increase or decrease the influence of individual predictors Xj by increasing
or decreasing Ajj . Often the predictors are many and highly correlated,
such as those arising from digitized analog signals or images. The covariance
function of the predictors can be used to tailor a metric A that focuses less,
say, on high-frequency contrasts (Exercise 6.4). Proposals have been made
for learning the parameters for multidimensional kernels. For example, the
projection-pursuit regression model discussed in Chapter 11 is of this flavor,
where low-rank versions of A imply ridge functions for f̂(X). More general
models for A are cumbersome, and we favor instead the structured forms
for the regression function discussed next.
</p>
<p>6.4.2 Structured Regression Functions
</p>
<p>We are trying to fit a regression function E(Y |X) = f(X1, X2, . . . , Xp) in
IRp, in which every level of interaction is potentially present. It is natural
to consider analysis-of-variance (ANOVA) decompositions of the form
</p>
<p>f(X1, X2, . . . , Xp) = α+
&sum;
</p>
<p>j
</p>
<p>gj(Xj) +
&sum;
</p>
<p>k&lt;ℓ
</p>
<p>gkℓ(Xk, Xℓ) + &middot; &middot; &middot; (6.15)
</p>
<p>and then introduce structure by eliminating some of the higher-order terms.
Additive models assume only main effect terms: f(X) = α+
</p>
<p>&sum;p
j=1 gj(Xj);
</p>
<p>second-order models will have terms with interactions of order at most
two, and so on. In Chapter 9, we describe iterative backfitting algorithms
for fitting such low-order interaction models. In the additive model, for
example, if all but the kth term is assumed known, then we can estimate gk
by local regression of Y &minus;&sum;j �=k gj(Xj) onXk. This is done for each function
in turn, repeatedly, until convergence. The important detail is that at any
stage, one-dimensional local regression is all that is needed. The same ideas
can be used to fit low-dimensional ANOVA decompositions.
An important special case of these structured models are the class of
</p>
<p>varying coefficient models. Suppose, for example, that we divide the p pre-
dictors in X into a set (X1, X2, . . . , Xq) with q &lt; p, and the remainder of</p>
<p/>
</div>
<div class="page"><p/>
<p>204 6. Kernel Smoothing Methods
</p>
<p>10
</p>
<p>12
</p>
<p>14
</p>
<p>16
</p>
<p>18
</p>
<p>20
</p>
<p>22
</p>
<p>24
</p>
<p>Depth
Female
</p>
<p>20 30 40 50 60
</p>
<p>Depth
Female
</p>
<p>Depth
Female
</p>
<p>20 30 40 50 60
</p>
<p>Depth
Female
</p>
<p>Depth
Female
</p>
<p>20 30 40 50 60
</p>
<p>Depth
Female
</p>
<p>Depth
Male
</p>
<p>Depth
Male
</p>
<p>20 30 40 50 60
</p>
<p>Depth
Male
</p>
<p>Depth
Male
</p>
<p>20 30 40 50 60
</p>
<p>Depth
Male
</p>
<p>10
</p>
<p>12
</p>
<p>14
</p>
<p>16
</p>
<p>18
</p>
<p>20
</p>
<p>22
</p>
<p>24
</p>
<p>Depth
Male
</p>
<p>20 30 40 50 60
</p>
<p>Age
</p>
<p>D
ia
</p>
<p>m
e
</p>
<p>te
r
</p>
<p>Aortic Diameter vs Age
</p>
<p>FIGURE 6.10. In each panel the aorta diameter is modeled as a linear func-
tion of age. The coefficients of this model vary with gender and depth down
the aorta (left is near the top, right is low down). There is a clear trend in the
coefficients of the linear model.
</p>
<p>the variables we collect in the vector Z. We then assume the conditionally
linear model
</p>
<p>f(X) = α(Z) + β1(Z)X1 + &middot; &middot; &middot;+ βq(Z)Xq. (6.16)
</p>
<p>For given Z, this is a linear model, but each of the coefficients can vary
with Z. It is natural to fit such a model by locally weighted least squares:
</p>
<p>min
α(z0),β(z0)
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>Kλ(z0, zi) (yi &minus; α(z0)&minus; x1iβ1(z0)&minus; &middot; &middot; &middot; &minus; xqiβq(z0))2 .
</p>
<p>(6.17)
Figure 6.10 illustrates the idea on measurements of the human aorta.
</p>
<p>A longstanding claim has been that the aorta thickens with age. Here we
model the diameter of the aorta as a linear function of age, but allow the
coefficients to vary with gender and depth down the aorta. We used a local
regression model separately for males and females. While the aorta clearly
does thicken with age at the higher regions of the aorta, the relationship
fades with distance down the aorta. Figure 6.11 shows the intercept and
slope as a function of depth.</p>
<p/>
</div>
<div class="page"><p/>
<p>6.5 Local Likelihood and Other Models 205
</p>
<p>Male
A
</p>
<p>g
e
</p>
<p> I
n
</p>
<p>te
rc
</p>
<p>e
p
</p>
<p>t
</p>
<p>Distance Down Aorta
</p>
<p>A
g
</p>
<p>e
 S
</p>
<p>lo
p
</p>
<p>e
</p>
<p>0.0 0.2 0.4 0.6 0.8 1.0
</p>
<p>Female
</p>
<p>1
4
</p>
<p>1
6
</p>
<p>1
8
</p>
<p>2
0
</p>
<p>Distance Down Aorta
</p>
<p>0
.0
</p>
<p>0
.4
</p>
<p>0
.8
</p>
<p>1
.2
</p>
<p>0.0 0.2 0.4 0.6 0.8 1.0
</p>
<p>FIGURE 6.11. The intercept and slope of age as a function of distance down
the aorta, separately for males and females. The yellow bands indicate one stan-
dard error.
</p>
<p>6.5 Local Likelihood and Other Models
</p>
<p>The concept of local regression and varying coefficient models is extremely
broad: any parametric model can be made local if the fitting method ac-
commodates observation weights. Here are some examples:
</p>
<p>&bull; Associated with each observation yi is a parameter θi = θ(xi) = xTi β
linear in the covariate(s) xi, and inference for β is based on the log-
</p>
<p>likelihood l(β) =
&sum;N
</p>
<p>i=1 l(yi, x
T
i β). We can model θ(X) more flexibly
</p>
<p>by using the likelihood local to x0 for inference of θ(x0) = x
T
0 β(x0):
</p>
<p>l(β(x0)) =
N&sum;
</p>
<p>i=1
</p>
<p>Kλ(x0, xi)l(yi, x
T
i β(x0)).
</p>
<p>Many likelihood models, in particular the family of generalized linear
models including logistic and log-linear models, involve the covariates
in a linear fashion. Local likelihood allows a relaxation from a globally
linear model to one that is locally linear.</p>
<p/>
</div>
<div class="page"><p/>
<p>206 6. Kernel Smoothing Methods
</p>
<p>&bull; As above, except different variables are associated with θ from those
used for defining the local likelihood:
</p>
<p>l(θ(z0)) =
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>Kλ(z0, zi)l(yi, η(xi, θ(z0))).
</p>
<p>For example, η(x, θ) = xT θ could be a linear model in x. This will fit
a varying coefficient model θ(z) by maximizing the local likelihood.
</p>
<p>&bull; Autoregressive time series models of order k have the form yt =
β0 + β1yt&minus;1 + β2yt&minus;2 + &middot; &middot; &middot; + βkyt&minus;k + εt. Denoting the lag set by
zt = (yt&minus;1, yt&minus;2, . . . , yt&minus;k), the model looks like a standard linear
model yt = z
</p>
<p>T
t β + εt, and is typically fit by least squares. Fitting
</p>
<p>by local least squares with a kernel K(z0, zt) allows the model to
vary according to the short-term history of the series. This is to be
distinguished from the more traditional dynamic linear models that
vary by windowing time.
</p>
<p>As an illustration of local likelihood, we consider the local version of the
multiclass linear logistic regression model (4.36) of Chapter 4. The data
consist of features xi and an associated categorical response gi&isin;{1, 2, . . . , J},
and the linear model has the form
</p>
<p>Pr(G = j|X = x) = e
βj0+β
</p>
<p>T
j x
</p>
<p>1 +
&sum;J&minus;1
</p>
<p>k=1 e
βk0+βTk x
</p>
<p>. (6.18)
</p>
<p>The local log-likelihood for this J class model can be written
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>Kλ(x0, xi)
</p>
<p>{
βgi0(x0) + βgi(x0)
</p>
<p>T (xi &minus; x0)
</p>
<p>&minus; log
[
1 +
</p>
<p>J&minus;1&sum;
</p>
<p>k=1
</p>
<p>exp
(
βk0(x0) + βk(x0)
</p>
<p>T (xi &minus; x0)
)
]}
</p>
<p>.
</p>
<p>(6.19)
</p>
<p>Notice that
</p>
<p>&bull; we have used gi as a subscript in the first line to pick out the appro-
priate numerator;
</p>
<p>&bull; βJ0 = 0 and βJ = 0 by the definition of the model;
</p>
<p>&bull; we have centered the local regressions at x0, so that the fitted poste-
rior probabilities at x0 are simply
</p>
<p>P̂r(G = j|X = x0) =
eβ̂j0(x0)
</p>
<p>1 +
&sum;J&minus;1
</p>
<p>k=1 e
β̂k0(x0)
</p>
<p>. (6.20)</p>
<p/>
</div>
<div class="page"><p/>
<p>6.5 Local Likelihood and Other Models 207
</p>
<p>Systolic Blood Pressure
</p>
<p>P
re
</p>
<p>v
a
le
</p>
<p>n
c
e
 C
</p>
<p>H
D
</p>
<p>100 140 180 220
</p>
<p>0
.0
</p>
<p>0
.2
</p>
<p>0
.4
</p>
<p>0
.6
</p>
<p>0
.8
</p>
<p>1
.0
</p>
<p>Obesity
</p>
<p>P
re
</p>
<p>v
a
le
</p>
<p>n
c
e
 C
</p>
<p>H
D
</p>
<p>15 25 35 45
</p>
<p>0
.0
</p>
<p>0
.2
</p>
<p>0
.4
</p>
<p>0
.6
</p>
<p>0
.8
</p>
<p>1
.0
</p>
<p>FIGURE 6.12. Each plot shows the binary response CHD (coronary heart dis-
ease) as a function of a risk factor for the South African heart disease data.
For each plot we have computed the fitted prevalence of CHD using a local linear
logistic regression model. The unexpected increase in the prevalence of CHD at
the lower ends of the ranges is because these are retrospective data, and some of
the subjects had already undergone treatment to reduce their blood pressure and
weight. The shaded region in the plot indicates an estimated pointwise standard
error band.
</p>
<p>This model can be used for flexible multiclass classification in moderately
low dimensions, although successes have been reported with the high-
dimensional ZIP-code classification problem. Generalized additive models
(Chapter 9) using kernel smoothing methods are closely related, and avoid
dimensionality problems by assuming an additive structure for the regres-
sion function.
As a simple illustration we fit a two-class local linear logistic model to
</p>
<p>the heart disease data of Chapter 4. Figure 6.12 shows the univariate local
logistic models fit to two of the risk factors (separately). This is a useful
screening device for detecting nonlinearities, when the data themselves have
little visual information to offer. In this case an unexpected anomaly is
uncovered in the data, which may have gone unnoticed with traditional
methods.
Since CHD is a binary indicator, we could estimate the conditional preva-
</p>
<p>lence Pr(G = j|x0) by simply smoothing this binary response directly with-
out resorting to a likelihood formulation. This amounts to fitting a locally
constant logistic regression model (Exercise 6.5). In order to enjoy the bias-
correction of local-linear smoothing, it is more natural to operate on the
unrestricted logit scale.
Typically with logistic regression, we compute parameter estimates as
</p>
<p>well as their standard errors. This can be done locally as well, and so</p>
<p/>
</div>
<div class="page"><p/>
<p>208 6. Kernel Smoothing Methods
</p>
<p>Systolic Blood Pressure (for CHD group)
</p>
<p>D
e
n
s
it
y
 E
</p>
<p>s
ti
m
</p>
<p>a
te
</p>
<p>100 120 140 160 180 200 220
</p>
<p>0
.0
</p>
<p>0
.0
</p>
<p>0
5
</p>
<p>0
.0
</p>
<p>1
0
</p>
<p>0
.0
</p>
<p>1
5
</p>
<p>0
.0
</p>
<p>2
0
</p>
<p>FIGURE 6.13. A kernel density estimate for systolic blood pressure (for the
CHD group). The density estimate at each point is the average contribution from
each of the kernels at that point. We have scaled the kernels down by a factor of
10 to make the graph readable.
</p>
<p>we can produce, as shown in the plot, estimated pointwise standard-error
bands about our fitted prevalence.
</p>
<p>6.6 Kernel Density Estimation and Classification
</p>
<p>Kernel density estimation is an unsupervised learning procedure, which
historically precedes kernel regression. It also leads naturally to a simple
family of procedures for nonparametric classification.
</p>
<p>6.6.1 Kernel Density Estimation
</p>
<p>Suppose we have a random sample x1, . . . , xN drawn from a probability
density fX(x), and we wish to estimate fX at a point x0. For simplicity we
assume for now that X &isin; IR. Arguing as before, a natural local estimate
has the form
</p>
<p>f̂X(x0) =
#xi &isin; N (x0)
</p>
<p>Nλ
, (6.21)
</p>
<p>where N (x0) is a small metric neighborhood around x0 of width λ. This
estimate is bumpy, and the smooth Parzen estimate is preferred
</p>
<p>f̂X(x0) =
1
</p>
<p>Nλ
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>Kλ(x0, xi), (6.22)</p>
<p/>
</div>
<div class="page"><p/>
<p>6.6 Kernel Density Estimation and Classification 209
</p>
<p>Systolic Blood Pressure
</p>
<p>D
e
</p>
<p>n
s
it
y
 E
</p>
<p>s
ti
m
</p>
<p>a
te
</p>
<p>s
</p>
<p>100 140 180 220
</p>
<p>0
.0
</p>
<p>0
.0
</p>
<p>1
0
</p>
<p>0
.0
</p>
<p>2
0
</p>
<p>CHD
no CHD
</p>
<p>Systolic Blood Pressure
</p>
<p>P
o
</p>
<p>s
te
</p>
<p>ri
o
</p>
<p>r 
E
</p>
<p>s
ti
m
</p>
<p>a
te
</p>
<p>100 140 180 220
</p>
<p>0
.0
</p>
<p>0
.2
</p>
<p>0
.4
</p>
<p>0
.6
</p>
<p>0
.8
</p>
<p>1
.0
</p>
<p>FIGURE 6.14. The left panel shows the two separate density estimates for
systolic blood pressure in the CHD versus no-CHD groups, using a Gaussian
kernel density estimate in each. The right panel shows the estimated posterior
probabilities for CHD, using (6.25).
</p>
<p>because it counts observations close to x0 with weights that decrease with
distance from x0. In this case a popular choice forKλ is the Gaussian kernel
Kλ(x0, x) = φ(|x&minus; x0|/λ). Figure 6.13 shows a Gaussian kernel density fit
to the sample values for systolic blood pressure for the CHD group. Letting
φλ denote the Gaussian density with mean zero and standard-deviation λ,
then (6.22) has the form
</p>
<p>f̂X(x) =
1
</p>
<p>N
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>φλ(x&minus; xi)
</p>
<p>= (F̂ ⋆ φλ)(x), (6.23)
</p>
<p>the convolution of the sample empirical distribution F̂ with φλ. The dis-
tribution F̂ (x) puts mass 1/N at each of the observed xi, and is jumpy; in
</p>
<p>f̂X(x) we have smoothed F̂ by adding independent Gaussian noise to each
observation xi.
The Parzen density estimate is the equivalent of the local average, and
</p>
<p>improvements have been proposed along the lines of local regression [on the
log scale for densities; see Loader (1999)]. We will not pursue these here.
In IRp the natural generalization of the Gaussian density estimate amounts
to using the Gaussian product kernel in (6.23),
</p>
<p>f̂X(x0) =
1
</p>
<p>N(2λ2π)
p
2
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>e&minus;
1
2 (||xi&minus;x0||/λ)
</p>
<p>2
</p>
<p>. (6.24)</p>
<p/>
</div>
<div class="page"><p/>
<p>210 6. Kernel Smoothing Methods
</p>
<p>0.0
</p>
<p>0.5
</p>
<p>1.0
</p>
<p>FIGURE 6.15. The population class densities may have interesting structure
(left) that disappears when the posterior probabilities are formed (right).
</p>
<p>6.6.2 Kernel Density Classification
</p>
<p>One can use nonparametric density estimates for classification in a straight-
forward fashion using Bayes&rsquo; theorem. Suppose for a J class problem we fit
nonparametric density estimates f̂j(X), j = 1, . . . , J separately in each of
the classes, and we also have estimates of the class priors π̂j (usually the
sample proportions). Then
</p>
<p>P̂r(G = j|X = x0) =
π̂j f̂j(x0)&sum;J
</p>
<p>k=1 π̂kf̂k(x0)
. (6.25)
</p>
<p>Figure 6.14 uses this method to estimate the prevalence of CHD for the
heart risk factor study, and should be compared with the left panel of Fig-
ure 6.12. The main difference occurs in the region of high SBP in the right
panel of Figure 6.14. In this region the data are sparse for both classes, and
since the Gaussian kernel density estimates use metric kernels, the density
estimates are low and of poor quality (high variance) in these regions. The
local logistic regression method (6.20) uses the tri-cube kernel with k-NN
bandwidth; this effectively widens the kernel in this region, and makes use
of the local linear assumption to smooth out the estimate (on the logit
scale).
If classification is the ultimate goal, then learning the separate class den-
</p>
<p>sities well may be unnecessary, and can in fact be misleading. Figure 6.15
shows an example where the densities are both multimodal, but the pos-
terior ratio is quite smooth. In learning the separate densities from data,
one might decide to settle for a rougher, high-variance fit to capture these
features, which are irrelevant for the purposes of estimating the posterior
probabilities. In fact, if classification is the ultimate goal, then we need only
to estimate the posterior well near the decision boundary (for two classes,
this is the set {x|Pr(G = 1|X = x) = 12}).
</p>
<p>6.6.3 The Naive Bayes Classifier
</p>
<p>This is a technique that has remained popular over the years, despite its
name (also known as &ldquo;Idiot&rsquo;s Bayes&rdquo;!) It is especially appropriate when</p>
<p/>
</div>
<div class="page"><p/>
<p>6.6 Kernel Density Estimation and Classification 211
</p>
<p>the dimension p of the feature space is high, making density estimation
unattractive. The naive Bayes model assumes that given a class G = j, the
features Xk are independent:
</p>
<p>fj(X) =
</p>
<p>p&prod;
</p>
<p>k=1
</p>
<p>fjk(Xk). (6.26)
</p>
<p>While this assumption is generally not true, it does simplify the estimation
dramatically:
</p>
<p>&bull; The individual class-conditional marginal densities fjk can each be
estimated separately using one-dimensional kernel density estimates.
This is in fact a generalization of the original naive Bayes procedures,
which used univariate Gaussians to represent these marginals.
</p>
<p>&bull; If a component Xj of X is discrete, then an appropriate histogram
estimate can be used. This provides a seamless way of mixing variable
types in a feature vector.
</p>
<p>Despite these rather optimistic assumptions, naive Bayes classifiers often
outperform far more sophisticated alternatives. The reasons are related to
Figure 6.15: although the individual class density estimates may be biased,
this bias might not hurt the posterior probabilities as much, especially
near the decision regions. In fact, the problem may be able to withstand
considerable bias for the savings in variance such a &ldquo;naive&rdquo; assumption
earns.
Starting from (6.26) we can derive the logit-transform (using class J as
</p>
<p>the base):
</p>
<p>log
Pr(G = ℓ|X)
Pr(G = J |X) = log
</p>
<p>πℓfℓ(X)
</p>
<p>πJfJ(X)
</p>
<p>= log
πℓ
</p>
<p>&prod;p
k=1 fℓk(Xk)
</p>
<p>πJ
&prod;p
</p>
<p>k=1 fJk(Xk)
</p>
<p>= log
πℓ
πJ
</p>
<p>+
</p>
<p>p&sum;
</p>
<p>k=1
</p>
<p>log
fℓk(Xk)
</p>
<p>fJk(Xk)
</p>
<p>= αℓ +
</p>
<p>p&sum;
</p>
<p>k=1
</p>
<p>gℓk(Xk).
</p>
<p>(6.27)
</p>
<p>This has the form of a generalized additive model, which is described in more
detail in Chapter 9. The models are fit in quite different ways though; their
differences are explored in Exercise 6.9. The relationship between naive
Bayes and generalized additive models is analogous to that between linear
discriminant analysis and logistic regression (Section 4.4.5).</p>
<p/>
</div>
<div class="page"><p/>
<p>212 6. Kernel Smoothing Methods
</p>
<p>6.7 Radial Basis Functions and Kernels
</p>
<p>In Chapter 5, functions are represented as expansions in basis functions:
f(x) =
</p>
<p>&sum;M
j=1 βjhj(x). The art of flexible modeling using basis expansions
</p>
<p>consists of picking an appropriate family of basis functions, and then con-
trolling the complexity of the representation by selection, regularization, or
both. Some of the families of basis functions have elements that are defined
locally; for example, B-splines are defined locally in IR. If more flexibility
is desired in a particular region, then that region needs to be represented
by more basis functions (which in the case of B-splines translates to more
knots). Tensor products of IR-local basis functions deliver basis functions
local in IRp. Not all basis functions are local&mdash;for example, the truncated
power bases for splines, or the sigmoidal basis functions σ(α0 + αx) used
in neural-networks (see Chapter 11). The composed function f(x) can nev-
ertheless show local behavior, because of the particular signs and values
of the coefficients causing cancellations of global effects. For example, the
truncated power basis has an equivalent B-spline basis for the same space
of functions; the cancellation is exact in this case.
Kernel methods achieve flexibility by fitting simple models in a region
</p>
<p>local to the target point x0. Localization is achieved via a weighting kernel
Kλ, and individual observations receive weights Kλ(x0, xi).
Radial basis functions combine these ideas, by treating the kernel func-
</p>
<p>tions Kλ(ξ, x) as basis functions. This leads to the model
</p>
<p>f(x) =
</p>
<p>M&sum;
</p>
<p>j=1
</p>
<p>Kλj (ξj , x)βj
</p>
<p>=
</p>
<p>M&sum;
</p>
<p>j=1
</p>
<p>D
</p>
<p>( ||x&minus; ξj ||
λj
</p>
<p>)
βj , (6.28)
</p>
<p>where each basis element is indexed by a location or prototype parameter ξj
and a scale parameter λj . A popular choice for D is the standard Gaussian
density function. There are several approaches to learning the parameters
{λj , ξj , βj}, j = 1, . . . ,M . For simplicity we will focus on least squares
methods for regression, and use the Gaussian kernel.
</p>
<p>&bull; Optimize the sum-of-squares with respect to all the parameters:
</p>
<p>min
{λj ,ξj ,βj}M1
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>⎛
⎝yi &minus; β0 &minus;
</p>
<p>M&sum;
</p>
<p>j=1
</p>
<p>βj exp
</p>
<p>{
&minus; (xi &minus; ξj)
</p>
<p>T (xi &minus; ξj)
λ2j
</p>
<p>}⎞
⎠
</p>
<p>2
</p>
<p>.
</p>
<p>(6.29)
This model is commonly referred to as an RBF network, an alterna-
tive to the sigmoidal neural network discussed in Chapter 11; the ξj
and λj playing the role of the weights. This criterion is nonconvex</p>
<p/>
</div>
<div class="page"><p/>
<p>6.7 Radial Basis Functions and Kernels 213
</p>
<p>0
2
</p>
<p>4
6
</p>
<p>8
0
</p>
<p>.0
0
</p>
<p>.4
0
</p>
<p>.8
1
</p>
<p>.2
</p>
<p>FIGURE 6.16. Gaussian radial basis functions in IR with fixed width can leave
holes (top panel). Renormalized Gaussian radial basis functions avoid this prob-
lem, and produce basis functions similar in some respects to B-splines.
</p>
<p>with multiple local minima, and the algorithms for optimization are
similar to those used for neural networks.
</p>
<p>&bull; Estimate the {λj , ξj} separately from the βj . Given the former, the
estimation of the latter is a simple least squares problem. Often the
kernel parameters λj and ξj are chosen in an unsupervised way using
the X distribution alone. One of the methods is to fit a Gaussian
mixture density model to the training xi, which provides both the
centers ξj and the scales λj . Other even more adhoc approaches use
clustering methods to locate the prototypes ξj , and treat λj = λ
as a hyper-parameter. The obvious drawback of these approaches is
that the conditional distribution Pr(Y |X) and in particular E(Y |X)
is having no say in where the action is concentrated. On the positive
side, they are much simpler to implement.
</p>
<p>While it would seem attractive to reduce the parameter set and assume
a constant value for λj = λ, this can have an undesirable side effect of
creating holes&mdash;regions of IRp where none of the kernels has appreciable
support, as illustrated in Figure 6.16 (upper panel). Renormalized radial
basis functions,
</p>
<p>hj(x) =
D(||x&minus; ξj ||/λ)&sum;M
k=1 D(||x&minus; ξk||/λ)
</p>
<p>, (6.30)
</p>
<p>avoid this problem (lower panel).
The Nadaraya&ndash;Watson kernel regression estimator (6.2) in IRp can be
</p>
<p>viewed as an expansion in renormalized radial basis functions,
</p>
<p>f̂(x0) =
&sum;N
</p>
<p>i=1 yi
Kλ(x0,xi)&sum;
N
i=1 Kλ(x0,xi)
</p>
<p>=
&sum;N
</p>
<p>i=1 yihi(x0) (6.31)</p>
<p/>
</div>
<div class="page"><p/>
<p>214 6. Kernel Smoothing Methods
</p>
<p>with a basis function hi located at every observation and coefficients yi;
that is, ξi = xi, β̂i = yi, i = 1, . . . , N .
Note the similarity between the expansion (6.31) and the solution (5.50)
</p>
<p>on page 169 to the regularization problem induced by the kernel K. Radial
basis functions form the bridge between the modern &ldquo;kernel methods&rdquo; and
local fitting technology.
</p>
<p>6.8 Mixture Models for Density Estimation and
Classification
</p>
<p>The mixture model is a useful tool for density estimation, and can be viewed
as a kind of kernel method. The Gaussian mixture model has the form
</p>
<p>f(x) =
</p>
<p>M&sum;
</p>
<p>m=1
</p>
<p>αmφ(x;μm,Σm) (6.32)
</p>
<p>with mixing proportions αm,
&sum;
</p>
<p>m αm = 1, and each Gaussian density has
a mean μm and covariance matrix Σm. In general, mixture models can use
any component densities in place of the Gaussian in (6.32): the Gaussian
mixture model is by far the most popular.
The parameters are usually fit by maximum likelihood, using the EM
</p>
<p>algorithm as described in Chapter 8. Some special cases arise:
</p>
<p>&bull; If the covariance matrices are constrained to be scalar: Σm = σmI,
then (6.32) has the form of a radial basis expansion.
</p>
<p>&bull; If in addition σm = σ &gt; 0 is fixed, and M &uarr; N , then the max-
imum likelihood estimate for (6.32) approaches the kernel density
estimate (6.22) where α̂m = 1/N and μ̂m = xm.
</p>
<p>Using Bayes&rsquo; theorem, separate mixture densities in each class lead to flex-
ible models for Pr(G|X); this is taken up in some detail in Chapter 12.
Figure 6.17 shows an application of mixtures to the heart disease risk-
</p>
<p>factor study. In the top row are histograms of Age for the no CHD and CHD
groups separately, and then combined on the right. Using the combined
data, we fit a two-component mixture of the form (6.32) with the (scalars)
Σ1 and Σ2 not constrained to be equal. Fitting was done via the EM
algorithm (Chapter 8): note that the procedure does not use knowledge of
the CHD labels. The resulting estimates were
</p>
<p>μ̂1 = 36.4, Σ̂1 = 157.7, α̂1 = 0.7,
</p>
<p>μ̂2 = 58.0, Σ̂2 = 15.6, α̂2 = 0.3.
</p>
<p>The component densities φ(μ̂1, Σ̂1) and φ(μ̂2, Σ̂2) are shown in the lower-
left and middle panels. The lower-right panel shows these component den-
sities (orange and blue) along with the estimated mixture density (green).</p>
<p/>
</div>
<div class="page"><p/>
<p>6.8 Mixture Models for Density Estimation and Classification 215
</p>
<p>No CHD
</p>
<p>Age
</p>
<p>C
o
u
n
t
</p>
<p>20 30 40 50 60
</p>
<p>0
5
</p>
<p>1
0
</p>
<p>1
5
</p>
<p>2
0
</p>
<p>CHD
</p>
<p>Age
</p>
<p>C
o
u
n
t
</p>
<p>20 30 40 50 60
</p>
<p>0
5
</p>
<p>1
0
</p>
<p>1
5
</p>
<p>Combined
</p>
<p>Age
</p>
<p>C
o
u
n
t
</p>
<p>20 30 40 50 60
</p>
<p>0
5
</p>
<p>1
0
</p>
<p>1
5
</p>
<p>2
0
</p>
<p>2
5
</p>
<p>3
0
</p>
<p>20 30 40 50 60
</p>
<p>0
.0
</p>
<p>0
0
.0
</p>
<p>2
0
.0
</p>
<p>4
0
.0
</p>
<p>6
0
.0
</p>
<p>8
0
.1
</p>
<p>0
</p>
<p>Age
</p>
<p>M
ix
</p>
<p>tu
re
</p>
<p> E
s
ti
m
</p>
<p>a
te
</p>
<p>20 30 40 50 60
</p>
<p>0
.0
</p>
<p>0
0
.0
</p>
<p>2
0
.0
</p>
<p>4
0
.0
</p>
<p>6
0
.0
</p>
<p>8
0
.1
</p>
<p>0
</p>
<p>Age
</p>
<p>M
ix
</p>
<p>tu
re
</p>
<p> E
s
ti
m
</p>
<p>a
te
</p>
<p>20 30 40 50 60
</p>
<p>0
.0
</p>
<p>0
0
.0
</p>
<p>2
0
.0
</p>
<p>4
0
.0
</p>
<p>6
0
.0
</p>
<p>8
0
.1
</p>
<p>0
</p>
<p>Age
M
</p>
<p>ix
tu
</p>
<p>re
 E
</p>
<p>s
ti
m
</p>
<p>a
te
</p>
<p>FIGURE 6.17. Application of mixtures to the heart disease risk-factor study.
(Top row:) Histograms of Age for the no CHD and CHD groups separately, and
combined. (Bottom row:) estimated component densities from a Gaussian mix-
ture model, (bottom left, bottom middle); (bottom right:) Estimated component
densities (blue and orange) along with the estimated mixture density (green). The
orange density has a very large standard deviation, and approximates a uniform
density.
</p>
<p>The mixture model also provides an estimate of the probability that
observation i belongs to component m,
</p>
<p>r̂im =
α̂mφ(xi; μ̂m, Σ̂m)&sum;M
k=1 α̂kφ(xi; μ̂k, Σ̂k)
</p>
<p>, (6.33)
</p>
<p>where xi is Age in our example. Suppose we threshold each value r̂i2 and
hence define δ̂i = I(r̂i2 &gt; 0.5). Then we can compare the classification of
each observation by CHD and the mixture model:
</p>
<p>Mixture model
</p>
<p>δ̂ = 0 δ̂ = 1
</p>
<p>CHD No 232 70
Yes 76 84
</p>
<p>Although the mixture model did not use the CHD labels, it has done a fair
job in discovering the two CHD subpopulations. Linear logistic regression,
using the CHD as a response, achieves the same error rate (32%) when fit to
these data using maximum-likelihood (Section 4.4).</p>
<p/>
</div>
<div class="page"><p/>
<p>216 6. Kernel Smoothing Methods
</p>
<p>6.9 Computational Considerations
</p>
<p>Kernel and local regression and density estimation are memory-based meth-
ods: the model is the entire training data set, and the fitting is done at
evaluation or prediction time. For many real-time applications, this can
make this class of methods infeasible.
The computational cost to fit at a single observation x0 is O(N) flops,
</p>
<p>except in oversimplified cases (such as square kernels). By comparison,
an expansion in M basis functions costs O(M) for one evaluation, and
typically M &sim; O(logN). Basis function methods have an initial cost of at
least O(NM2 +M3).
</p>
<p>The smoothing parameter(s) λ for kernel methods are typically deter-
mined off-line, for example using cross-validation, at a cost of O(N2) flops.
</p>
<p>Popular implementations of local regression, such as the loess function in
S-PLUS and R and the locfit procedure (Loader, 1999), use triangulation
schemes to reduce the computations. They compute the fit exactly at M
carefully chosen locations (O(NM)), and then use blending techniques to
interpolate the fit elsewhere (O(M) per evaluation).
</p>
<p>Bibliographic Notes
</p>
<p>There is a vast literature on kernel methods which we will not attempt to
summarize. Rather we will point to a few good references that themselves
have extensive bibliographies. Loader (1999) gives excellent coverage of lo-
cal regression and likelihood, and also describes state-of-the-art software
for fitting these models. Fan and Gijbels (1996) cover these models from
a more theoretical aspect. Hastie and Tibshirani (1990) discuss local re-
gression in the context of additive modeling. Silverman (1986) gives a good
overview of density estimation, as does Scott (1992).
</p>
<p>Exercises
</p>
<p>Ex. 6.1 Show that the Nadaraya&ndash;Watson kernel smooth with fixed metric
bandwidth λ and a Gaussian kernel is differentiable. What can be said for
the Epanechnikov kernel? What can be said for the Epanechnikov kernel
with adaptive nearest-neighbor bandwidth λ(x0)?
</p>
<p>Ex. 6.2 Show that
&sum;N
</p>
<p>i=1(xi&minus;x0)li(x0) = 0 for local linear regression. Define
bj(x0) =
</p>
<p>&sum;N
i=1(xi &minus; x0)j li(x0). Show that b0(x0) = 1 for local polynomial
</p>
<p>regression of any degree (including local constants). Show that bj(x0) = 0
for all j &isin; {1, 2, . . . , k} for local polynomial regression of degree k. What
are the implications of this on the bias?</p>
<p/>
</div>
<div class="page"><p/>
<p>Exercises 217
</p>
<p>Ex. 6.3 Show that ||l(x)|| (Section 6.1.2) increases with the degree of the
local polynomial.
</p>
<p>Ex. 6.4 Suppose that the p predictors X arise from sampling relatively
smooth analog curves at p uniformly spaced abscissa values. Denote by
Cov(X|Y ) = Σ the conditional covariance matrix of the predictors, and
assume this does not change much with Y . Discuss the nature of Maha-
lanobis choice A = Σ&minus;1 for the metric in (6.14). How does this compare
with A = I? How might you construct a kernel A that (a) downweights
high-frequency components in the distance metric; (b) ignores them
completely?
</p>
<p>Ex. 6.5 Show that fitting a locally constant multinomial logit model of
the form (6.19) amounts to smoothing the binary response indicators for
each class separately using a Nadaraya&ndash;Watson kernel smoother with kernel
weights Kλ(x0, xi).
</p>
<p>Ex. 6.6 Suppose that all you have is software for fitting local regression,
but you can specify exactly which monomials are included in the fit. How
could you use this software to fit a varying-coefficient model in some of the
variables?
</p>
<p>Ex. 6.7 Derive an expression for the leave-one-out cross-validated residual
sum-of-squares for local polynomial regression.
</p>
<p>Ex. 6.8 Suppose that for continuous response Y and predictor X, we model
the joint density of X,Y using a multivariate Gaussian kernel estimator.
Note that the kernel in this case would be the product kernel φλ(X)φλ(Y ).
Show that the conditional mean E(Y |X) derived from this estimate is a
Nadaraya&ndash;Watson estimator. Extend this result to classification by pro-
viding a suitable kernel for the estimation of the joint distribution of a
continuous X and discrete Y .
</p>
<p>Ex. 6.9 Explore the differences between the naive Bayes model (6.27) and
a generalized additive logistic regression model, in terms of (a) model as-
sumptions and (b) estimation. If all the variables Xk are discrete, what can
you say about the corresponding GAM?
</p>
<p>Ex. 6.10 Suppose we haveN samples generated from the model yi = f(xi)+
εi, with εi independent and identically distributed with mean zero and
variance σ2, the xi assumed fixed (non random). We estimate f using a
linear smoother (local regression, smoothing spline, etc.) with smoothing
</p>
<p>parameter λ. Thus the vector of fitted values is given by f̂ = Sλy. Consider
the in-sample prediction error
</p>
<p>PE(λ) = E
1
</p>
<p>N
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>(y&lowast;i &minus; f̂λ(xi))2 (6.34)</p>
<p/>
</div>
<div class="page"><p/>
<p>218 6. Kernel Smoothing Methods
</p>
<p>for predicting new responses at the N input values. Show that the aver-
age squared residual on the training data, ASR(λ), is a biased estimate
(optimistic) for PE(λ), while
</p>
<p>Cλ = ASR(λ) +
2σ2
</p>
<p>N
trace(Sλ) (6.35)
</p>
<p>is unbiased.
</p>
<p>Ex. 6.11 Show that for the Gaussian mixture model (6.32) the likelihood
is maximized at +&infin;, and describe how.
</p>
<p>Ex. 6.12 Write a computer program to perform a local linear discrimi-
nant analysis. At each query point x0, the training data receive weights
Kλ(x0, xi) from a weighting kernel, and the ingredients for the linear deci-
sion boundaries (see Section 4.3) are computed by weighted averages. Try
out your program on the zipcode data, and show the training and test er-
rors for a series of five pre-chosen values of λ. The zipcode data are available
from the book website www-stat.stanford.edu/ElemStatLearn.</p>
<p/>
</div>
<div class="page"><p/>
<p>7
Model Assessment and Selection
</p>
<p>7.1 Introduction
</p>
<p>The generalization performance of a learning method relates to its predic-
tion capability on independent test data. Assessment of this performance
is extremely important in practice, since it guides the choice of learning
method or model, and gives us a measure of the quality of the ultimately
chosen model.
In this chapter we describe and illustrate the key methods for perfor-
</p>
<p>mance assessment, and show how they are used to select models. We begin
the chapter with a discussion of the interplay between bias, variance and
model complexity.
</p>
<p>7.2 Bias, Variance and Model Complexity
</p>
<p>Figure 7.1 illustrates the important issue in assessing the ability of a learn-
ing method to generalize. Consider first the case of a quantitative or interval
scale response. We have a target variable Y , a vector of inputs X, and a
prediction model f̂(X) that has been estimated from a training set T .
The loss function for measuring errors between Y and f̂(X) is denoted by
</p>
<p>L(Y, f̂(X)). Typical choices are
</p>
<p>L(Y, f̂(X)) =
</p>
<p>{
(Y &minus; f̂(X))2 squared error
|Y &minus; f̂(X)| absolute error. (7.1)
</p>
<p> &copy; Springer Science+Business Media, LLC 2009 
</p>
<p>T. Hastie et al., The Elements of Statistical Learning, Second Edition, 219
</p>
<p>DOI: 10.1007/b94608_7,</p>
<p/>
</div>
<div class="page"><p/>
<p>220 7. Model Assessment and Selection
</p>
<p>0 5 10 15 20 25 30 35
</p>
<p>0
.0
</p>
<p>0
.2
</p>
<p>0
.4
</p>
<p>0
.6
</p>
<p>0
.8
</p>
<p>1
.0
</p>
<p>1
.2
</p>
<p>Model Complexity (df)
</p>
<p>P
re
</p>
<p>d
ic
</p>
<p>ti
o
</p>
<p>n
 E
</p>
<p>rr
o
</p>
<p>r
</p>
<p>High Bias Low Bias
</p>
<p>High VarianceLow Variance
</p>
<p>FIGURE 7.1. Behavior of test sample and training sample error as the model
complexity is varied. The light blue curves show the training error err, while the
light red curves show the conditional test error ErrT for 100 training sets of size
50 each, as the model complexity is increased. The solid curves show the expected
test error Err and the expected training error E[err].
</p>
<p>Test error, also referred to as generalization error, is the prediction error
over an independent test sample
</p>
<p>ErrT = E[L(Y, f̂(X))|T ] (7.2)
</p>
<p>where both X and Y are drawn randomly from their joint distribution
(population). Here the training set T is fixed, and test error refers to the
error for this specific training set. A related quantity is the expected pre-
diction error (or expected test error)
</p>
<p>Err = E[L(Y, f̂(X))] = E[ErrT ]. (7.3)
</p>
<p>Note that this expectation averages over everything that is random, includ-
ing the randomness in the training set that produced f̂ .
</p>
<p>Figure 7.1 shows the prediction error (light red curves) ErrT for 100
simulated training sets each of size 50. The lasso (Section 3.4.2) was used
to produce the sequence of fits. The solid red curve is the average, and
hence an estimate of Err.
Estimation of ErrT will be our goal, although we will see that Err is
</p>
<p>more amenable to statistical analysis, and most methods effectively esti-
mate the expected error. It does not seem possible to estimate conditional</p>
<p/>
</div>
<div class="page"><p/>
<p>7.2 Bias, Variance and Model Complexity 221
</p>
<p>error effectively, given only the information in the same training set. Some
discussion of this point is given in Section 7.12.
Training error is the average loss over the training sample
</p>
<p>err =
1
</p>
<p>N
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>L(yi, f̂(xi)). (7.4)
</p>
<p>We would like to know the expected test error of our estimated model
f̂ . As the model becomes more and more complex, it uses the training
data more and is able to adapt to more complicated underlying structures.
Hence there is a decrease in bias but an increase in variance. There is some
intermediate model complexity that gives minimum expected test error.
Unfortunately training error is not a good estimate of the test error,
</p>
<p>as seen in Figure 7.1. Training error consistently decreases with model
complexity, typically dropping to zero if we increase the model complexity
enough. However, a model with zero training error is overfit to the training
data and will typically generalize poorly.
The story is similar for a qualitative or categorical response G taking
</p>
<p>one of K values in a set G, labeled for convenience as 1, 2, . . . ,K. Typically
we model the probabilities pk(X) = Pr(G = k|X) (or some monotone
transformations fk(X)), and then Ĝ(X) = argmaxk p̂k(X). In some cases,
such as 1-nearest neighbor classification (Chapters 2 and 13) we produce
Ĝ(X) directly. Typical loss functions are
</p>
<p>L(G, Ĝ(X)) = I(G �= Ĝ(X)) (0&ndash;1 loss), (7.5)
</p>
<p>L(G, p̂(X)) = &minus;2
K&sum;
</p>
<p>k=1
</p>
<p>I(G = k) log p̂k(X)
</p>
<p>= &minus;2 log p̂G(X) (&minus;2 &times; log-likelihood). (7.6)
The quantity &minus;2 &times; the log-likelihood is sometimes referred to as the
deviance.
Again, test error here is ErrT = E[L(G, Ĝ(X))|T ], the population mis-
</p>
<p>classification error of the classifier trained on T , and Err is the expected
misclassification error.
Training error is the sample analogue, for example,
</p>
<p>err = &minus; 2
N
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>log p̂gi(xi), (7.7)
</p>
<p>the sample log-likelihood for the model.
The log-likelihood can be used as a loss-function for general response
</p>
<p>densities, such as the Poisson, gamma, exponential, log-normal and others.
If Prθ(X)(Y ) is the density of Y , indexed by a parameter θ(X) that depends
on the predictor X, then
</p>
<p>L(Y, θ(X)) = &minus;2 &middot; log Prθ(X)(Y ). (7.8)</p>
<p/>
</div>
<div class="page"><p/>
<p>222 7. Model Assessment and Selection
</p>
<p>The &ldquo;&minus;2&rdquo; in the definition makes the log-likelihood loss for the Gaussian
distribution match squared-error loss.
For ease of exposition, for the remainder of this chapter we will use Y and
</p>
<p>f(X) to represent all of the above situations, since we focus mainly on the
quantitative response (squared-error loss) setting. For the other situations,
the appropriate translations are obvious.
In this chapter we describe a number of methods for estimating the
</p>
<p>expected test error for a model. Typically our model will have a tuning
parameter or parameters α and so we can write our predictions as f̂α(x).
The tuning parameter varies the complexity of our model, and we wish to
find the value of α that minimizes error, that is, produces the minimum of
the average test error curve in Figure 7.1. Having said this, for brevity we
will often suppress the dependence of f̂(x) on α.
It is important to note that there are in fact two separate goals that we
</p>
<p>might have in mind:
</p>
<p>Model selection: estimating the performance of different models in order
to choose the best one.
</p>
<p>Model assessment: having chosen a final model, estimating its predic-
tion error (generalization error) on new data.
</p>
<p>If we are in a data-rich situation, the best approach for both problems is
to randomly divide the dataset into three parts: a training set, a validation
set, and a test set. The training set is used to fit the models; the validation
set is used to estimate prediction error for model selection; the test set is
used for assessment of the generalization error of the final chosen model.
Ideally, the test set should be kept in a &ldquo;vault,&rdquo; and be brought out only
at the end of the data analysis. Suppose instead that we use the test-set
repeatedly, choosing the model with smallest test-set error. Then the test
set error of the final chosen model will underestimate the true test error,
sometimes substantially.
It is difficult to give a general rule on how to choose the number of
</p>
<p>observations in each of the three parts, as this depends on the signal-to-
noise ratio in the data and the training sample size. A typical split might
be 50% for training, and 25% each for validation and testing:
</p>
<p>TestTrain Validation TestTrain Validation TestValidationTrain Validation TestTrain
</p>
<p>The methods in this chapter are designed for situations where there is
insufficient data to split it into three parts. Again it is too difficult to give
a general rule on how much training data is enough; among other things,
this depends on the signal-to-noise ratio of the underlying function, and
the complexity of the models being fit to the data.</p>
<p/>
</div>
<div class="page"><p/>
<p>7.3 The Bias&ndash;Variance Decomposition 223
</p>
<p>The methods of this chapter approximate the validation step either an-
alytically (AIC, BIC, MDL, SRM) or by efficient sample re-use (cross-
validation and the bootstrap). Besides their use in model selection, we also
examine to what extent each method provides a reliable estimate of test
error of the final chosen model.
Before jumping into these topics, we first explore in more detail the
</p>
<p>nature of test error and the bias&ndash;variance tradeoff.
</p>
<p>7.3 The Bias&ndash;Variance Decomposition
</p>
<p>As in Chapter 2, if we assume that Y = f(X) + ε where E(ε) = 0 and
Var(ε) = σ2ε , we can derive an expression for the expected prediction error
</p>
<p>of a regression fit f̂(X) at an input point X = x0, using squared-error loss:
</p>
<p>Err(x0) = E[(Y &minus; f̂(x0))2|X = x0]
= σ2ε + [Ef̂(x0)&minus; f(x0)]2 + E[f̂(x0)&minus; Ef̂(x0)]2
</p>
<p>= σ2ε + Bias
2(f̂(x0)) + Var(f̂(x0))
</p>
<p>= Irreducible Error + Bias2 +Variance. (7.9)
</p>
<p>The first term is the variance of the target around its true mean f(x0), and
cannot be avoided no matter how well we estimate f(x0), unless σ
</p>
<p>2
ε = 0.
</p>
<p>The second term is the squared bias, the amount by which the average of
our estimate differs from the true mean; the last term is the variance; the
expected squared deviation of f̂(x0) around its mean. Typically the more
</p>
<p>complex we make the model f̂ , the lower the (squared) bias but the higher
the variance.
For the k-nearest-neighbor regression fit, these expressions have the sim-
</p>
<p>ple form
</p>
<p>Err(x0) = E[(Y &minus; f̂k(x0))2|X = x0]
</p>
<p>= σ2ε +
</p>
<p>[
f(x0)&minus;
</p>
<p>1
</p>
<p>k
</p>
<p>k&sum;
</p>
<p>ℓ=1
</p>
<p>f(x(ℓ))
</p>
<p>]2
+
</p>
<p>σ2ε
k
. (7.10)
</p>
<p>Here we assume for simplicity that training inputs xi are fixed, and the ran-
domness arises from the yi. The number of neighbors k is inversely related
to the model complexity. For small k, the estimate f̂k(x) can potentially
adapt itself better to the underlying f(x). As we increase k, the bias&mdash;the
squared difference between f(x0) and the average of f(x) at the k-nearest
neighbors&mdash;will typically increase, while the variance decreases.
For a linear model fit f̂p(x) = x
</p>
<p>T β̂, where the parameter vector β with
p components is fit by least squares, we have
</p>
<p>Err(x0) = E[(Y &minus; f̂p(x0))2|X = x0]</p>
<p/>
</div>
<div class="page"><p/>
<p>224 7. Model Assessment and Selection
</p>
<p>= σ2ε + [f(x0)&minus; Ef̂p(x0)]2 + ||h(x0)||2σ2ε . (7.11)
</p>
<p>Here h(x0) = X(X
TX)&minus;1x0, the N -vector of linear weights that produce
</p>
<p>the fit f̂p(x0) = x0
T (XTX)&minus;1XTy, and hence Var[f̂p(x0)] = ||h(x0)||2σ2ε .
</p>
<p>While this variance changes with x0, its average (with x0 taken to be each
of the sample values xi) is (p/N)σ
</p>
<p>2
ε , and hence
</p>
<p>1
</p>
<p>N
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>Err(xi) = σ
2
ε +
</p>
<p>1
</p>
<p>N
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>[f(xi)&minus; Ef̂(xi)]2 +
p
</p>
<p>N
σ2ε , (7.12)
</p>
<p>the in-sample error. Here model complexity is directly related to the num-
ber of parameters p.
The test error Err(x0) for a ridge regression fit f̂α(x0) is identical in
</p>
<p>form to (7.11), except the linear weights in the variance term are different:
h(x0) = X(X
</p>
<p>TX+ αI)&minus;1x0. The bias term will also be different.
For a linear model family such as ridge regression, we can break down
</p>
<p>the bias more finely. Let β&lowast; denote the parameters of the best-fitting linear
approximation to f :
</p>
<p>β&lowast; = argmin
β
</p>
<p>E
(
f(X)&minus;XTβ
</p>
<p>)2
. (7.13)
</p>
<p>Here the expectation is taken with respect to the distribution of the input
variables X. Then we can write the average squared bias as
</p>
<p>Ex0
</p>
<p>[
f(x0)&minus; Ef̂α(x0)
</p>
<p>]2
= Ex0
</p>
<p>[
f(x0)&minus; xT0 β&lowast;
</p>
<p>]2
+ Ex0
</p>
<p>[
xT0 β&lowast; &minus; ExT0 β̂α
</p>
<p>]2
</p>
<p>= Ave[Model Bias]2 +Ave[Estimation Bias]2
</p>
<p>(7.14)
</p>
<p>The first term on the right-hand side is the average squared model bias, the
error between the best-fitting linear approximation and the true function.
The second term is the average squared estimation bias, the error between
the average estimate E(xT0 β̂) and the best-fitting linear approximation.
For linear models fit by ordinary least squares, the estimation bias is zero.
</p>
<p>For restricted fits, such as ridge regression, it is positive, and we trade it off
with the benefits of a reduced variance. The model bias can only be reduced
by enlarging the class of linear models to a richer collection of models, by
including interactions and transformations of the variables in the model.
Figure 7.2 shows the bias&ndash;variance tradeoff schematically. In the case
</p>
<p>of linear models, the model space is the set of all linear predictions from
p inputs and the black dot labeled &ldquo;closest fit&rdquo; is xTβ&lowast;. The blue-shaded
region indicates the error σε with which we see the truth in the training
sample.
Also shown is the variance of the least squares fit, indicated by the large
</p>
<p>yellow circle centered at the black dot labeled &ldquo;closest fit in population,&rsquo;</p>
<p/>
</div>
<div class="page"><p/>
<p>7.3 The Bias&ndash;Variance Decomposition 225
</p>
<p>Realization
</p>
<p>Closest fit in population
</p>
<p>Estimation Bias
</p>
<p>SPACE
</p>
<p>Variance
</p>
<p>Estimation
</p>
<p>Closest fit
</p>
<p>Truth
</p>
<p>Model bias
</p>
<p>RESTRICTED
</p>
<p>Shrunken fit
</p>
<p>MODEL SPACE
</p>
<p>MODEL
</p>
<p>FIGURE 7.2. Schematic of the behavior of bias and variance. The model space
is the set of all possible predictions from the model, with the &ldquo;closest fit&rdquo; labeled
with a black dot. The model bias from the truth is shown, along with the variance,
indicated by the large yellow circle centered at the black dot labeled &ldquo;closest fit
in population.&rdquo; A shrunken or regularized fit is also shown, having additional
estimation bias, but smaller prediction error due to its decreased variance.</p>
<p/>
</div>
<div class="page"><p/>
<p>226 7. Model Assessment and Selection
</p>
<p>Now if we were to fit a model with fewer predictors, or regularize the coef-
ficients by shrinking them toward zero (say), we would get the &ldquo;shrunken
fit&rdquo; shown in the figure. This fit has an additional estimation bias, due to
the fact that it is not the closest fit in the model space. On the other hand,
it has smaller variance. If the decrease in variance exceeds the increase in
(squared) bias, then this is worthwhile.
</p>
<p>7.3.1 Example: Bias&ndash;Variance Tradeoff
</p>
<p>Figure 7.3 shows the bias&ndash;variance tradeoff for two simulated examples.
There are 80 observations and 20 predictors, uniformly distributed in the
hypercube [0, 1]20. The situations are as follows:
</p>
<p>Left panels: Y is 0 if X1 &le; 1/2 and 1 if X1 &gt; 1/2, and we apply k-nearest
neighbors.
</p>
<p>Right panels: Y is 1 if
&sum;10
</p>
<p>j=1 Xj is greater than 5 and 0 otherwise, and we
use best subset linear regression of size p.
</p>
<p>The top row is regression with squared error loss; the bottom row is classi-
fication with 0&ndash;1 loss. The figures show the prediction error (red), squared
bias (green) and variance (blue), all computed for a large test sample.
In the regression problems, bias and variance add to produce the predic-
</p>
<p>tion error curves, with minima at about k = 5 for k-nearest neighbors, and
p &ge; 10 for the linear model. For classification loss (bottom figures), some
interesting phenomena can be seen. The bias and variance curves are the
same as in the top figures, and prediction error now refers to misclassifi-
cation rate. We see that prediction error is no longer the sum of squared
bias and variance. For the k-nearest neighbor classifier, prediction error
decreases or stays the same as the number of neighbors is increased to 20,
despite the fact that the squared bias is rising. For the linear model classi-
fier the minimum occurs for p &ge; 10 as in regression, but the improvement
over the p = 1 model is more dramatic. We see that bias and variance seem
to interact in determining prediction error.
Why does this happen? There is a simple explanation for the first phe-
</p>
<p>nomenon. Suppose at a given input point, the true probability of class 1 is
0.9 while the expected value of our estimate is 0.6. Then the squared bias&mdash;
(0.6&minus;0.9)2&mdash;is considerable, but the prediction error is zero since we make
the correct decision. In other words, estimation errors that leave us on the
right side of the decision boundary don&rsquo;t hurt. Exercise 7.2 demonstrates
this phenomenon analytically, and also shows the interaction effect between
bias and variance.
The overall point is that the bias&ndash;variance tradeoff behaves differently
</p>
<p>for 0&ndash;1 loss than it does for squared error loss. This in turn means that
the best choices of tuning parameters may differ substantially in the two</p>
<p/>
</div>
<div class="page"><p/>
<p>7.3 The Bias&ndash;Variance Decomposition 227
</p>
<p>0
.0
</p>
<p>0
.1
</p>
<p>0
.2
</p>
<p>0
.3
</p>
<p>0
.4
</p>
<p>Number of Neighbors k
</p>
<p>50 40 30 20 10 0
</p>
<p>k&minus;NN &minus; Regression
</p>
<p>5 10 15 20
</p>
<p>0
.0
</p>
<p>0
.1
</p>
<p>0
.2
</p>
<p>0
.3
</p>
<p>0
.4
</p>
<p>Subset Size p
</p>
<p>Linear Model &minus; Regression
</p>
<p>0
.0
</p>
<p>0
.1
</p>
<p>0
.2
</p>
<p>0
.3
</p>
<p>0
.4
</p>
<p>Number of Neighbors k
</p>
<p>50 40 30 20 10 0
</p>
<p>k&minus;NN &minus; Classification
</p>
<p>5 10 15 20
</p>
<p>0
.0
</p>
<p>0
.1
</p>
<p>0
.2
</p>
<p>0
.3
</p>
<p>0
.4
</p>
<p>Subset Size p
</p>
<p>Linear Model &minus; Classification
</p>
<p>FIGURE 7.3. Expected prediction error (orange), squared bias (green) and vari-
ance (blue) for a simulated example. The top row is regression with squared error
loss; the bottom row is classification with 0&ndash;1 loss. The models are k-nearest
neighbors (left) and best subset regression of size p (right). The variance and bias
curves are the same in regression and classification, but the prediction error curve
is different.</p>
<p/>
</div>
<div class="page"><p/>
<p>228 7. Model Assessment and Selection
</p>
<p>settings. One should base the choice of tuning parameter on an estimate of
prediction error, as described in the following sections.
</p>
<p>7.4 Optimism of the Training Error Rate
</p>
<p>Discussions of error rate estimation can be confusing, because we have
to make clear which quantities are fixed and which are random1. Before
we continue, we need a few definitions, elaborating on the material of Sec-
tion 7.2. Given a training set T = {(x1, y1), (x2, y2), . . . (xN , yN )} the gen-
eralization error of a model f̂ is
</p>
<p>ErrT = EX0,Y 0 [L(Y
0, f̂(X0))|T ]; (7.15)
</p>
<p>Note that the training set T is fixed in expression (7.15). The point (X0, Y 0)
is a new test data point, drawn from F , the joint distribution of the data.
Averaging over training sets T yields the expected error
</p>
<p>Err = ET EX0,Y 0 [L(Y
0, f̂(X0))|T ], (7.16)
</p>
<p>which is more amenable to statistical analysis. As mentioned earlier, it
turns out that most methods effectively estimate the expected error rather
than ET ; see Section 7.12 for more on this point.
Now typically, the training error
</p>
<p>err =
1
</p>
<p>N
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>L(yi, f̂(xi)) (7.17)
</p>
<p>will be less than the true error ErrT , because the same data is being used
to fit the method and assess its error (see Exercise 2.9). A fitting method
typically adapts to the training data, and hence the apparent or training
error err will be an overly optimistic estimate of the generalization error
ErrT .
Part of the discrepancy is due to where the evaluation points occur. The
</p>
<p>quantity ErrT can be thought of as extra-sample error, since the test input
vectors don&rsquo;t need to coincide with the training input vectors. The nature
of the optimism in err is easiest to understand when we focus instead on
the in-sample error
</p>
<p>Errin =
1
</p>
<p>N
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>EY 0 [L(Y
0
i , f̂(xi))|T ] (7.18)
</p>
<p>The Y 0 notation indicates that we observe N new response values at
each of the training points xi, i = 1, 2, . . . , N . We define the optimism as
</p>
<p>1Indeed, in the first edition of our book, this section wasn&rsquo;t sufficiently clear.</p>
<p/>
</div>
<div class="page"><p/>
<p>7.4 Optimism of the Training Error Rate 229
</p>
<p>the difference between Errin and the training error err:
</p>
<p>op &equiv; Errin &minus; err. (7.19)
</p>
<p>This is typically positive since err is usually biased downward as an estimate
of prediction error. Finally, the average optimism is the expectation of the
optimism over training sets
</p>
<p>ω &equiv; Ey(op). (7.20)
</p>
<p>Here the predictors in the training set are fixed, and the expectation is
over the training set outcome values; hence we have used the notation Ey
instead of ET . We can usually estimate only the expected error ω rather
than op, in the same way that we can estimate the expected error Err
rather than the conditional error ErrT .
For squared error, 0&ndash;1, and other loss functions, one can show quite
</p>
<p>generally that
</p>
<p>ω =
2
</p>
<p>N
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>Cov(ŷi, yi), (7.21)
</p>
<p>where Cov indicates covariance. Thus the amount by which err underesti-
mates the true error depends on how strongly yi affects its own prediction.
The harder we fit the data, the greater Cov(ŷi, yi) will be, thereby increas-
ing the optimism. Exercise 7.4 proves this result for squared error loss where
ŷi is the fitted value from the regression. For 0&ndash;1 loss, ŷi &isin; {0, 1} is the
classification at xi, and for entropy loss, ŷi &isin; [0, 1] is the fitted probability
of class 1 at xi.
In summary, we have the important relation
</p>
<p>Ey(Errin) = Ey(err) +
2
</p>
<p>N
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>Cov(ŷi, yi). (7.22)
</p>
<p>This expression simplifies if ŷi is obtained by a linear fit with d inputs
or basis functions. For example,
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>Cov(ŷi, yi) = dσ
2
ε (7.23)
</p>
<p>for the additive error model Y = f(X) + ε, and so
</p>
<p>Ey(Errin) = Ey(err) + 2 &middot;
d
</p>
<p>N
σ2ε . (7.24)
</p>
<p>Expression (7.23) is the basis for the definition of the effective number of
parameters discussed in Section 7.6 The optimism increases linearly with</p>
<p/>
</div>
<div class="page"><p/>
<p>230 7. Model Assessment and Selection
</p>
<p>the number d of inputs or basis functions we use, but decreases as the
training sample size increases. Versions of (7.24) hold approximately for
other error models, such as binary data and entropy loss.
An obvious way to estimate prediction error is to estimate the optimism
</p>
<p>and then add it to the training error err. The methods described in the
next section&mdash;Cp , AIC, BIC and others&mdash;work in this way, for a special
class of estimates that are linear in their parameters.
In contrast, cross-validation and bootstrap methods, described later in
</p>
<p>the chapter, are direct estimates of the extra-sample error Err. These gen-
eral tools can be used with any loss function, and with nonlinear, adaptive
fitting techniques.
In-sample error is not usually of direct interest since future values of the
</p>
<p>features are not likely to coincide with their training set values. But for
comparison between models, in-sample error is convenient and often leads
to effective model selection. The reason is that the relative (rather than
absolute) size of the error is what matters.
</p>
<p>7.5 Estimates of In-Sample Prediction Error
</p>
<p>The general form of the in-sample estimates is
</p>
<p>Êrrin = err + ω̂, (7.25)
</p>
<p>where ω̂ is an estimate of the average optimism.
Using expression (7.24), applicable when d parameters are fit under
</p>
<p>squared error loss, leads to a version of the so-called Cp statistic,
</p>
<p>Cp = err + 2 &middot;
d
</p>
<p>N
σ̂ε
</p>
<p>2. (7.26)
</p>
<p>Here σ̂ε
2 is an estimate of the noise variance, obtained from the mean-
</p>
<p>squared error of a low-bias model. Using this criterion we adjust the training
error by a factor proportional to the number of basis functions used.
The Akaike information criterion is a similar but more generally appli-
</p>
<p>cable estimate of Errin when a log-likelihood loss function is used. It relies
on a relationship similar to (7.24) that holds asymptotically as N &rarr; &infin;:
</p>
<p>&minus;2 &middot; E[log Prθ̂(Y )] &asymp; &minus;
2
</p>
<p>N
&middot; E[loglik] + 2 &middot; d
</p>
<p>N
. (7.27)
</p>
<p>Here Prθ(Y ) is a family of densities for Y (containing the &ldquo;true&rdquo; density),
</p>
<p>θ̂ is the maximum-likelihood estimate of θ, and &ldquo;loglik&rdquo; is the maximized
log-likelihood:
</p>
<p>loglik =
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>log Prθ̂(yi). (7.28)</p>
<p/>
</div>
<div class="page"><p/>
<p>7.5 Estimates of In-Sample Prediction Error 231
</p>
<p>For example, for the logistic regression model, using the binomial log-
likelihood, we have
</p>
<p>AIC = &minus; 2
N
</p>
<p>&middot; loglik + 2 &middot; d
N
</p>
<p>. (7.29)
</p>
<p>For the Gaussian model (with variance σ2ε = σ̂ε
2 assumed known), the AIC
</p>
<p>statistic is equivalent to Cp , and so we refer to them collectively as AIC.
To use AIC for model selection, we simply choose the model giving small-
</p>
<p>est AIC over the set of models considered. For nonlinear and other complex
models, we need to replace d by some measure of model complexity. We
discuss this in Section 7.6.
Given a set of models fα(x) indexed by a tuning parameter α, denote
</p>
<p>by err(α) and d(α) the training error and number of parameters for each
model. Then for this set of models we define
</p>
<p>AIC(α) = err(α) + 2 &middot; d(α)
N
</p>
<p>σ̂ε
2. (7.30)
</p>
<p>The function AIC(α) provides an estimate of the test error curve, and we
find the tuning parameter α̂ that minimizes it. Our final chosen model
is fα̂(x). Note that if the basis functions are chosen adaptively, (7.23) no
longer holds. For example, if we have a total of p inputs, and we choose
the best-fitting linear model with d &lt; p inputs, the optimism will exceed
(2d/N)σ2ε . Put another way, by choosing the best-fitting model with d
inputs, the effective number of parameters fit is more than d.
</p>
<p>Figure 7.4 shows AIC in action for the phoneme recognition example
of Section 5.2.3 on page 148. The input vector is the log-periodogram of
the spoken vowel, quantized to 256 uniformly spaced frequencies. A lin-
ear logistic regression model is used to predict the phoneme class, with
coefficient function β(f) =
</p>
<p>&sum;M
m=1 hm(f)θm, an expansion in M spline ba-
</p>
<p>sis functions. For any given M , a basis of natural cubic splines is used
for the hm, with knots chosen uniformly over the range of frequencies (so
d(α) = d(M) = M). Using AIC to select the number of basis functions will
approximately minimize Err(M) for both entropy and 0&ndash;1 loss.
The simple formula
</p>
<p>(2/N)
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>Cov(ŷi, yi) = (2d/N)σ
2
ε
</p>
<p>holds exactly for linear models with additive errors and squared error loss,
and approximately for linear models and log-likelihoods. In particular, the
formula does not hold in general for 0&ndash;1 loss (Efron, 1986), although many
authors nevertheless use it in that context (right panel of Figure 7.4).</p>
<p/>
</div>
<div class="page"><p/>
<p>232 7. Model Assessment and Selection
</p>
<p>Number of Basis Functions
</p>
<p>L
o
</p>
<p>g
-l
ik
</p>
<p>e
lih
</p>
<p>o
o
</p>
<p>d
</p>
<p>0
.5
</p>
<p>1
.0
</p>
<p>1
.5
</p>
<p>2
.0
</p>
<p>2
.5
</p>
<p>Log-likelihood Loss
</p>
<p>2 4 8 16 32 64 128
</p>
<p>O
</p>
<p>O
O
</p>
<p>O O O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O O O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O O O
O O
</p>
<p>Train
Test
AIC
</p>
<p>Number of Basis Functions
</p>
<p>M
is
</p>
<p>c
la
</p>
<p>s
s
if
ic
</p>
<p>a
ti
o
</p>
<p>n
 E
</p>
<p>rr
o
</p>
<p>r
</p>
<p>0
.1
</p>
<p>0
0
</p>
<p>.1
5
</p>
<p>0
.2
</p>
<p>0
0
</p>
<p>.2
5
</p>
<p>0
.3
</p>
<p>0
0
</p>
<p>.3
5
</p>
<p>0-1 Loss
</p>
<p>2 4 8 16 32 64 128
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O O
</p>
<p>O
O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>O O
</p>
<p>O
</p>
<p>O
</p>
<p>O
</p>
<p>FIGURE 7.4. AIC used for model selection for the phoneme recogni-
tion example of Section 5.2.3. The logistic regression coefficient function
β(f) =
</p>
<p>&sum;M
</p>
<p>m=1 hm(f)θm is modeled as an expansion in M spline basis functions.
In the left panel we see the AIC statistic used to estimate Errin using log-likeli-
hood loss. Included is an estimate of Err based on an independent test sample. It
does well except for the extremely over-parametrized case (M = 256 parameters
for N = 1000 observations). In the right panel the same is done for 0&ndash;1 loss.
Although the AIC formula does not strictly apply here, it does a reasonable job in
this case.
</p>
<p>7.6 The Effective Number of Parameters
</p>
<p>The concept of &ldquo;number of parameters&rdquo; can be generalized, especially to
models where regularization is used in the fitting. Suppose we stack the
outcomes y1, y2, . . . , yN into a vector y, and similarly for the predictions
ŷ. Then a linear fitting method is one for which we can write
</p>
<p>ŷ = Sy, (7.31)
</p>
<p>where S is an N &times;N matrix depending on the input vectors xi but not on
the yi. Linear fitting methods include linear regression on the original fea-
tures or on a derived basis set, and smoothing methods that use quadratic
shrinkage, such as ridge regression and cubic smoothing splines. Then the
effective number of parameters is defined as
</p>
<p>df(S) = trace(S), (7.32)
</p>
<p>the sum of the diagonal elements of S (also known as the effective degrees-
of-freedom). Note that if S is an orthogonal-projection matrix onto a basis</p>
<p/>
</div>
<div class="page"><p/>
<p>7.7 The Bayesian Approach and BIC 233
</p>
<p>set spanned byM features, then trace(S) = M . It turns out that trace(S) is
exactly the correct quantity to replace d as the number of parameters in the
Cp statistic (7.26). If y arises from an additive-error model Y = f(X) + ε
</p>
<p>with Var(ε) = σ2ε , then one can show that
&sum;N
</p>
<p>i=1 Cov(ŷi, yi) = trace(S)σ
2
ε ,
</p>
<p>which motivates the more general definition
</p>
<p>df(ŷ) =
</p>
<p>&sum;N
i=1 Cov(ŷi, yi)
</p>
<p>σ2ε
(7.33)
</p>
<p>(Exercises 7.4 and 7.5). Section 5.4.1 on page 153 gives some more intuition
for the definition df = trace(S) in the context of smoothing splines.
For models like neural networks, in which we minimize an error function
</p>
<p>R(w) with weight decay penalty (regularization) α
&sum;
</p>
<p>m w
2
m, the effective
</p>
<p>number of parameters has the form
</p>
<p>df(α) =
M&sum;
</p>
<p>m=1
</p>
<p>θm
θm + α
</p>
<p>, (7.34)
</p>
<p>where the θm are the eigenvalues of the Hessian matrix &part;
2R(w)/&part;w&part;wT .
</p>
<p>Expression (7.34) follows from (7.32) if we make a quadratic approximation
to the error function at the solution (Bishop, 1995).
</p>
<p>7.7 The Bayesian Approach and BIC
</p>
<p>The Bayesian information criterion (BIC), like AIC, is applicable in settings
where the fitting is carried out by maximization of a log-likelihood. The
generic form of BIC is
</p>
<p>BIC = &minus;2 &middot; loglik + (logN) &middot; d. (7.35)
</p>
<p>The BIC statistic (times 1/2) is also known as the Schwarz criterion (Schwarz,
1978).
Under the Gaussian model, assuming the variance σ2ε is known, &minus;2&middot;loglik
</p>
<p>equals (up to a constant)
&sum;
</p>
<p>i(yi&minus;f̂(xi))2/σ2ε , which is N &middot;err/σ2ε for squared
error loss. Hence we can write
</p>
<p>BIC =
N
</p>
<p>σ2ε
</p>
<p>[
err + (logN) &middot; d
</p>
<p>N
σ2ε
</p>
<p>]
. (7.36)
</p>
<p>Therefore BIC is proportional to AIC (Cp), with the factor 2 replaced
by logN . Assuming N &gt; e2 &asymp; 7.4, BIC tends to penalize complex models
more heavily, giving preference to simpler models in selection. As with AIC,
σ2ε is typically estimated by the mean squared error of a low-bias model.
For classification problems, use of the multinomial log-likelihood leads to a
similar relationship with the AIC, using cross-entropy as the error measure.</p>
<p/>
</div>
<div class="page"><p/>
<p>234 7. Model Assessment and Selection
</p>
<p>Note however that the misclassification error measure does not arise in the
BIC context, since it does not correspond to the log-likelihood of the data
under any probability model.
Despite its similarity with AIC, BIC is motivated in quite a different
</p>
<p>way. It arises in the Bayesian approach to model selection, which we now
describe.
Suppose we have a set of candidate models Mm,m = 1, . . . ,M and
</p>
<p>corresponding model parameters θm, and we wish to choose a best model
from among them. Assuming we have a prior distribution Pr(θm|Mm) for
the parameters of each model Mm, the posterior probability of a given
model is
</p>
<p>Pr(Mm|Z) &prop; Pr(Mm) &middot; Pr(Z|Mm) (7.37)
</p>
<p>&prop; Pr(Mm) &middot;
&int;
</p>
<p>Pr(Z|θm,Mm)Pr(θm|Mm)dθm,
</p>
<p>where Z represents the training data {xi, yi}N1 . To compare two models
Mm and Mℓ, we form the posterior odds
</p>
<p>Pr(Mm|Z)
Pr(Mℓ|Z)
</p>
<p>=
Pr(Mm)
Pr(Mℓ)
</p>
<p>&middot; Pr(Z|Mm)
Pr(Z|Mℓ)
</p>
<p>. (7.38)
</p>
<p>If the odds are greater than one we choose model m, otherwise we choose
model ℓ. The rightmost quantity
</p>
<p>BF(Z) =
Pr(Z|Mm)
Pr(Z|Mℓ)
</p>
<p>(7.39)
</p>
<p>is called the Bayes factor, the contribution of the data toward the posterior
odds.
Typically we assume that the prior over models is uniform, so that
</p>
<p>Pr(Mm) is constant. We need some way of approximating Pr(Z|Mm).
A so-called Laplace approximation to the integral followed by some other
simplifications (Ripley, 1996, page 64) to (7.37) gives
</p>
<p>log Pr(Z|Mm) = log Pr(Z|θ̂m,Mm)&minus;
dm
2
</p>
<p>&middot; logN +O(1). (7.40)
</p>
<p>Here θ̂m is a maximum likelihood estimate and dm is the number of free
parameters in model Mm. If we define our loss function to be
</p>
<p>&minus;2 log Pr(Z|θ̂m,Mm),
</p>
<p>this is equivalent to the BIC criterion of equation (7.35).
Therefore, choosing the model with minimum BIC is equivalent to choos-
</p>
<p>ing the model with largest (approximate) posterior probability. But this
framework gives us more. If we compute the BIC criterion for a set of M ,</p>
<p/>
</div>
<div class="page"><p/>
<p>7.8 Minimum Description Length 235
</p>
<p>models, giving BICm, m = 1, 2, . . . ,M , then we can estimate the posterior
probability of each model Mm as
</p>
<p>e&minus;
1
2 &middot;BICm
</p>
<p>&sum;M
ℓ=1 e
</p>
<p>&minus; 12 &middot;BICℓ
. (7.41)
</p>
<p>Thus we can estimate not only the best model, but also assess the relative
merits of the models considered.
For model selection purposes, there is no clear choice between AIC and
</p>
<p>BIC. BIC is asymptotically consistent as a selection criterion. What this
means is that given a family of models, including the true model, the prob-
ability that BIC will select the correct model approaches one as the sample
size N &rarr; &infin;. This is not the case for AIC, which tends to choose models
which are too complex as N &rarr; &infin;. On the other hand, for finite samples,
BIC often chooses models that are too simple, because of its heavy penalty
on complexity.
</p>
<p>7.8 Minimum Description Length
</p>
<p>The minimum description length (MDL) approach gives a selection cri-
terion formally identical to the BIC approach, but is motivated from an
optimal coding viewpoint. We first review the theory of coding for data
compression, and then apply it to model selection.
We think of our datum z as a message that we want to encode and
</p>
<p>send to someone else (the &ldquo;receiver&rdquo;). We think of our model as a way of
encoding the datum, and will choose the most parsimonious model, that is
the shortest code, for the transmission.
Suppose first that the possible messages we might want to transmit are
</p>
<p>z1, z2, . . . , zm. Our code uses a finite alphabet of length A: for example, we
might use a binary code {0, 1} of length A = 2. Here is an example with
four possible messages and a binary coding:
</p>
<p>Message z1 z2 z3 z4
Code 0 10 110 111
</p>
<p>(7.42)
</p>
<p>This code is known as an instantaneous prefix code: no code is the pre-
fix of any other, and the receiver (who knows all of the possible codes),
knows exactly when the message has been completely sent. We restrict our
discussion to such instantaneous prefix codes.
One could use the coding in (7.42) or we could permute the codes, for
</p>
<p>example use codes 110, 10, 111, 0 for z1, z2, z3, z4. How do we decide which
to use? It depends on how often we will be sending each of the messages.
If, for example, we will be sending z1 most often, it makes sense to use the
shortest code 0 for z1. Using this kind of strategy&mdash;shorter codes for more
frequent messages&mdash;the average message length will be shorter.</p>
<p/>
</div>
<div class="page"><p/>
<p>236 7. Model Assessment and Selection
</p>
<p>In general, if messages are sent with probabilities Pr(zi), i = 1, 2, . . . , 4,
a famous theorem due to Shannon says we should use code lengths li =
&minus; log2 Pr(zi) and the average message length satisfies
</p>
<p>E(length) &ge; &minus;
&sum;
</p>
<p>Pr(zi) log2 (Pr(zi)). (7.43)
</p>
<p>The right-hand side above is also called the entropy of the distribution
Pr(zi). The inequality is an equality when the probabilities satisfy pi =
A&minus;li . In our example, if Pr(zi) = 1/2, 1/4, 1/8, 1/8, respectively, then the
coding shown in (7.42) is optimal and achieves the entropy lower bound.
In general the lower bound cannot be achieved, but procedures like the
</p>
<p>Huffman coding scheme can get close to the bound. Note that with an
infinite set of messages, the entropy is replaced by &minus;
</p>
<p>&int;
Pr(z) log2 Pr(z)dz.
</p>
<p>From this result we glean the following:
</p>
<p>To transmit a random variable z having probability density func-
tion Pr(z), we require about &minus; log2 Pr(z) bits of information.
</p>
<p>We henceforth change notation from log2 Pr(z) to log Pr(z) = loge Pr(z);
this is for convenience, and just introduces an unimportant multiplicative
constant.
Now we apply this result to the problem of model selection. We have
</p>
<p>a model M with parameters θ, and data Z = (X,y) consisting of both
inputs and outputs. Let the (conditional) probability of the outputs under
the model be Pr(y|θ,M,X), assume the receiver knows all of the inputs,
and we wish to transmit the outputs. Then the message length required to
transmit the outputs is
</p>
<p>length = &minus; log Pr(y|θ,M,X)&minus; log Pr(θ|M), (7.44)
</p>
<p>the log-probability of the target values given the inputs. The second term
is the average code length for transmitting the model parameters θ, while
the first term is the average code length for transmitting the discrepancy
between the model and actual target values. For example suppose we have
a single target y with y &sim; N(θ, σ2), parameter θ &sim; N(0, 1) and no input
(for simplicity). Then the message length is
</p>
<p>length = constant + log σ +
(y &minus; θ)2
2σ2
</p>
<p>+
θ2
</p>
<p>2
. (7.45)
</p>
<p>Note that the smaller σ is, the shorter on average is the message length,
since y is more concentrated around θ.
The MDL principle says that we should choose the model that mini-
</p>
<p>mizes (7.44). We recognize (7.44) as the (negative) log-posterior distribu-
tion, and hence minimizing description length is equivalent to maximizing
posterior probability. Hence the BIC criterion, derived as approximation to
log-posterior probability, can also be viewed as a device for (approximate)
model choice by minimum description length.</p>
<p/>
</div>
<div class="page"><p/>
<p>7.9 Vapnik&ndash;Chervonenkis Dimension 237
</p>
<p>0.0 0.2 0.4 0.6 0.8 1.0
</p>
<p>-1
.0
</p>
<p>0
.0
</p>
<p>1
.0
</p>
<p>x
</p>
<p>si
n
(5
0
&middot;x
</p>
<p>)
</p>
<p>FIGURE 7.5. The solid curve is the function sin(50x) for x &isin; [0, 1]. The green
(solid) and blue (hollow) points illustrate how the associated indicator function
I(sin(αx) &gt; 0) can shatter (separate) an arbitrarily large number of points by
choosing an appropriately high frequency α.
</p>
<p>Note that we have ignored the precision with which a random variable
z is coded. With a finite code length we cannot code a continuous variable
exactly. However, if we code z within a tolerance δz, the message length
needed is the log of the probability in the interval [z, z+δz] which is well ap-
proximated by δzPr(z) if δz is small. Since log δzPr(z) = log δz+log Pr(z),
this means we can just ignore the constant log δz and use log Pr(z) as our
measure of message length, as we did above.
The preceding view of MDL for model selection says that we should
</p>
<p>choose the model with highest posterior probability. However, many Bayes-
ians would instead do inference by sampling from the posterior distribution.
</p>
<p>7.9 Vapnik&ndash;Chervonenkis Dimension
</p>
<p>A difficulty in using estimates of in-sample error is the need to specify the
number of parameters (or the complexity) d used in the fit. Although the
effective number of parameters introduced in Section 7.6 is useful for some
nonlinear models, it is not fully general. The Vapnik&ndash;Chervonenkis (VC)
theory provides such a general measure of complexity, and gives associated
bounds on the optimism. Here we give a brief review of this theory.
Suppose we have a class of functions {f(x, α)} indexed by a parameter
</p>
<p>vector α, with x &isin; IRp. Assume for now that f is an indicator function,
that is, takes the values 0 or 1. If α = (α0, α1) and f is the linear indi-
cator function I(α0 + α
</p>
<p>T
1 x &gt; 0), then it seems reasonable to say that the
</p>
<p>complexity of the class f is the number of parameters p + 1. But what
about f(x, α) = I(sinα &middot; x) where α is any real number and x &isin; IR? The
function sin(50 &middot; x) is shown in Figure 7.5. This is a very wiggly function
that gets even rougher as the frequency α increases, but it has only one
parameter: despite this, it doesn&rsquo;t seem reasonable to conclude that it has
less complexity than the linear indicator function I(α0 + α1x) in p = 1
dimension.</p>
<p/>
</div>
<div class="page"><p/>
<p>238 7. Model Assessment and Selection
</p>
<p>FIGURE 7.6. The first three panels show that the class of lines in the plane
can shatter three points. The last panel shows that this class cannot shatter four
points, as no line will put the hollow points on one side and the solid points on
the other. Hence the VC dimension of the class of straight lines in the plane is
three. Note that a class of nonlinear curves could shatter four points, and hence
has VC dimension greater than three.
</p>
<p>The Vapnik&ndash;Chervonenkis dimension is a way of measuring the com-
plexity of a class of functions by assessing how wiggly its members can
be.
</p>
<p>The VC dimension of the class {f(x, α)} is defined to be the
largest number of points (in some configuration) that can be
shattered by members of {f(x, α)}.
</p>
<p>A set of points is said to be shattered by a class of functions if, no matter
how we assign a binary label to each point, a member of the class can
perfectly separate them.
Figure 7.6 shows that the VC dimension of linear indicator functions
</p>
<p>in the plane is 3 but not 4, since no four points can be shattered by a
set of lines. In general, a linear indicator function in p dimensions has VC
dimension p+1, which is also the number of free parameters. On the other
hand, it can be shown that the family sin(αx) has infinite VC dimension,
as Figure 7.5 suggests. By appropriate choice of α, any set of points can be
shattered by this class (Exercise 7.8).
So far we have discussed the VC dimension only of indicator functions,
</p>
<p>but this can be extended to real-valued functions. The VC dimension of a
class of real-valued functions {g(x, α)} is defined to be the VC dimension
of the indicator class {I(g(x, α) &minus; β &gt; 0)}, where β takes values over the
range of g.
One can use the VC dimension in constructing an estimate of (extra-
</p>
<p>sample) prediction error; different types of results are available. Using the
concept of VC dimension, one can prove results about the optimism of the
training error when using a class of functions. An example of such a result is
the following. If we fit N training points using a class of functions {f(x, α)}
having VC dimension h, then with probability at least 1&minus; η over training</p>
<p/>
</div>
<div class="page"><p/>
<p>7.9 Vapnik&ndash;Chervonenkis Dimension 239
</p>
<p>sets:
</p>
<p>ErrT &le; err +
ǫ
</p>
<p>2
</p>
<p>(
1 +
</p>
<p>&radic;
1 +
</p>
<p>4 &middot; err
ǫ
</p>
<p>)
(binary classification)
</p>
<p>ErrT &le;
err
</p>
<p>(1&minus; c&radic;ǫ)+
(regression) (7.46)
</p>
<p>where ǫ = a1
h[log (a2N/h) + 1]&minus; log (η/4)
</p>
<p>N
,
</p>
<p>and 0 &lt; a1 &le; 4, 0 &lt; a2 &le; 2
</p>
<p>These bounds hold simultaneously for all members f(x, α), and are taken
from Cherkassky and Mulier (2007, pages 116&ndash;118). They recommend the
value c = 1. For regression they suggest a1 = a2 = 1, and for classification
they make no recommendation, with a1 = 4 and a2 = 2 corresponding
to worst-case scenarios. They also give an alternative practical bound for
regression
</p>
<p>ErrT &le; err
(
1&minus;
</p>
<p>&radic;
ρ&minus; ρ log ρ+ logN
</p>
<p>2N
</p>
<p>)&minus;1
</p>
<p>+
</p>
<p>(7.47)
</p>
<p>with ρ = hN , which is free of tuning constants. The bounds suggest that the
optimism increases with h and decreases with N in qualitative agreement
with the AIC correction d/N given in (7.24). However, the results in (7.46)
are stronger: rather than giving the expected optimism for each fixed func-
tion f(x, α), they give probabilistic upper bounds for all functions f(x, α),
and hence allow for searching over the class.
Vapnik&rsquo;s structural risk minimization (SRM) approach fits a nested se-
</p>
<p>quence of models of increasing VC dimensions h1 &lt; h2 &lt; &middot; &middot; &middot; , and then
chooses the model with the smallest value of the upper bound.
We note that upper bounds like the ones in (7.46) are often very loose,
</p>
<p>but that doesn&rsquo;t rule them out as good criteria for model selection, where
the relative (not absolute) size of the test error is important. The main
drawback of this approach is the difficulty in calculating the VC dimension
of a class of functions. Often only a crude upper bound for VC dimension
is obtainable, and this may not be adequate. An example in which the
structural risk minimization program can be successfully carried out is the
support vector classifier, discussed in Section 12.2.
</p>
<p>7.9.1 Example (Continued)
</p>
<p>Figure 7.7 shows the results when AIC, BIC and SRM are used to select
the model size for the examples of Figure 7.3. For the examples labeled KNN,
the model index α refers to neighborhood size, while for those labeled REG α
refers to subset size. Using each selection method (e.g., AIC) we estimated
the best model α̂ and found its true prediction error ErrT (α̂) on a test
set. For the same training set we computed the prediction error of the best</p>
<p/>
</div>
<div class="page"><p/>
<p>240 7. Model Assessment and Selection
</p>
<p>reg/KNN reg/linear class/KNN class/linear
</p>
<p>0
2
0
</p>
<p>4
0
</p>
<p>6
0
</p>
<p>8
0
</p>
<p>1
0
0
</p>
<p>%
 I
n
c
re
</p>
<p>a
s
e
 O
</p>
<p>v
e
r 
</p>
<p>B
e
s
t
</p>
<p>AIC
</p>
<p>reg/KNN reg/linear class/KNN class/linear
</p>
<p>0
2
0
</p>
<p>4
0
</p>
<p>6
0
</p>
<p>8
0
</p>
<p>1
0
0
</p>
<p>%
 I
n
c
re
</p>
<p>a
s
e
 O
</p>
<p>v
e
r 
</p>
<p>B
e
s
t
</p>
<p>BIC
</p>
<p>reg/KNN reg/linear class/KNN class/linear
</p>
<p>0
2
0
</p>
<p>4
0
</p>
<p>6
0
</p>
<p>8
0
</p>
<p>1
0
0
</p>
<p>%
 I
n
c
re
</p>
<p>a
s
e
 O
</p>
<p>v
e
r 
</p>
<p>B
e
s
t
</p>
<p>SRM
</p>
<p>FIGURE 7.7. Boxplots show the distribution of the relative error
100 &times; [ErrT (α̂) &minus; minα ErrT (α)]/[maxα ErrT (α) &minus; minα ErrT (α)] over the four
scenarios of Figure 7.3. This is the error in using the chosen model relative to
the best model. There are 100 training sets each of size 80 represented in each
boxplot, with the errors computed on test sets of size 10, 000.</p>
<p/>
</div>
<div class="page"><p/>
<p>7.10 Cross-Validation 241
</p>
<p>and worst possible model choices: minα ErrT (α) and maxα ErrT (α). The
boxplots show the distribution of the quantity
</p>
<p>100&times; ErrT (α̂)&minus;minα ErrT (α)
maxα ErrT (α)&minus;minα ErrT (α)
</p>
<p>,
</p>
<p>which represents the error in using the chosen model relative to the best
model. For linear regression the model complexity was measured by the
number of features; as mentioned in Section 7.5, this underestimates the
df, since it does not charge for the search for the best model of that size.
This was also used for the VC dimension of the linear classifier. For k-
nearest neighbors, we used the quantity N/k. Under an additive-error re-
gression model, this can be justified as the exact effective degrees of free-
dom (Exercise 7.6); we do not know if it corresponds to the VC dimen-
sion. We used a1 = a2 = 1 for the constants in (7.46); the results for SRM
changed with different constants, and this choice gave the most favorable re-
sults. We repeated the SRM selection using the alternative practical bound
(7.47), and got almost identical results. For misclassification error we used
σ̂ε
</p>
<p>2 = [N/(N &minus; d)] &middot; err(α) for the least restrictive model (k = 5 for KNN,
since k = 1 results in zero training error). The AIC criterion seems to work
well in all four scenarios, despite the lack of theoretical support with 0&ndash;1
loss. BIC does nearly as well, while the performance of SRM is mixed.
</p>
<p>7.10 Cross-Validation
</p>
<p>Probably the simplest and most widely used method for estimating predic-
tion error is cross-validation. This method directly estimates the expected
extra-sample error Err = E[L(Y, f̂(X))], the average generalization error
</p>
<p>when the method f̂(X) is applied to an independent test sample from the
joint distribution of X and Y . As mentioned earlier, we might hope that
cross-validation estimates the conditional error, with the training set T
held fixed. But as we will see in Section 7.12, cross-validation typically
estimates well only the expected prediction error.
</p>
<p>7.10.1 K-Fold Cross-Validation
</p>
<p>Ideally, if we had enough data, we would set aside a validation set and use
it to assess the performance of our prediction model. Since data are often
scarce, this is usually not possible. To finesse the problem, K-fold cross-
validation uses part of the available data to fit the model, and a different
part to test it. We split the data into K roughly equal-sized parts; for
example, when K = 5, the scenario looks like this:</p>
<p/>
</div>
<div class="page"><p/>
<p>242 7. Model Assessment and Selection
</p>
<p>ValidationTrain
</p>
<p>1 2 3 4 5
</p>
<p>Train Train Train
</p>
<p>For the kth part (third above), we fit the model to the other K&minus;1 parts
of the data, and calculate the prediction error of the fitted model when
predicting the kth part of the data. We do this for k = 1, 2, . . . ,K and
combine the K estimates of prediction error.
Here are more details. Let κ : {1, . . . , N} �&rarr; {1, . . . ,K} be an indexing
</p>
<p>function that indicates the partition to which observation i is allocated by
the randomization. Denote by f̂&minus;k(x) the fitted function, computed with
the kth part of the data removed. Then the cross-validation estimate of
prediction error is
</p>
<p>CV(f̂) =
1
</p>
<p>N
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>L(yi, f̂
&minus;κ(i)(xi)). (7.48)
</p>
<p>Typical choices of K are 5 or 10 (see below). The case K = N is known
as leave-one-out cross-validation. In this case κ(i) = i, and for the ith
observation the fit is computed using all the data except the ith.
Given a set of models f(x, α) indexed by a tuning parameter α, denote
</p>
<p>by f̂&minus;k(x, α) the αth model fit with the kth part of the data removed. Then
for this set of models we define
</p>
<p>CV(f̂ , α) =
1
</p>
<p>N
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>L(yi, f̂
&minus;κ(i)(xi, α)). (7.49)
</p>
<p>The function CV(f̂ , α) provides an estimate of the test error curve, and we
find the tuning parameter α̂ that minimizes it. Our final chosen model is
f(x, α̂), which we then fit to all the data.
It is interesting to wonder about what quantity K-fold cross-validation
</p>
<p>estimates. With K = 5 or 10, we might guess that it estimates the ex-
pected error Err, since the training sets in each fold are quite different
from the original training set. On the other hand, if K = N we might
guess that cross-validation estimates the conditional error ErrT . It turns
out that cross-validation only estimates effectively the average error Err,
as discussed in Section 7.12.
What value should we choose for K? With K = N , the cross-validation
</p>
<p>estimator is approximately unbiased for the true (expected) prediction er-
ror, but can have high variance because the N &ldquo;training sets&rdquo; are so similar
to one another. The computational burden is also considerable, requiring
N applications of the learning method. In certain special problems, this
computation can be done quickly&mdash;see Exercises 7.3 and 5.13.</p>
<p/>
</div>
<div class="page"><p/>
<p>7.10 Cross-Validation 243
</p>
<p>Size of Training Set
</p>
<p>1
-E
</p>
<p>rr
</p>
<p>0 50 100 150 200
</p>
<p>0
.0
</p>
<p>0
.2
</p>
<p>0
.4
</p>
<p>0
.6
</p>
<p>0
.8
</p>
<p>FIGURE 7.8. Hypothetical learning curve for a classifier on a given task: a
plot of 1 &minus; Err versus the size of the training set N . With a dataset of 200
observations, 5-fold cross-validation would use training sets of size 160, which
would behave much like the full set. However, with a dataset of 50 observations
fivefold cross-validation would use training sets of size 40, and this would result
in a considerable overestimate of prediction error.
</p>
<p>On the other hand, with K = 5 say, cross-validation has lower variance.
But bias could be a problem, depending on how the performance of the
learning method varies with the size of the training set. Figure 7.8 shows
a hypothetical &ldquo;learning curve&rdquo; for a classifier on a given task, a plot of
1 &minus; Err versus the size of the training set N . The performance of the
classifier improves as the training set size increases to 100 observations;
increasing the number further to 200 brings only a small benefit. If our
training set had 200 observations, fivefold cross-validation would estimate
the performance of our classifier over training sets of size 160, which from
Figure 7.8 is virtually the same as the performance for training set size
200. Thus cross-validation would not suffer from much bias. However if the
training set had 50 observations, fivefold cross-validation would estimate
the performance of our classifier over training sets of size 40, and from the
figure that would be an underestimate of 1&minus; Err. Hence as an estimate of
Err, cross-validation would be biased upward.
To summarize, if the learning curve has a considerable slope at the given
</p>
<p>training set size, five- or tenfold cross-validation will overestimate the true
prediction error. Whether this bias is a drawback in practice depends on
the objective. On the other hand, leave-one-out cross-validation has low
bias but can have high variance. Overall, five- or tenfold cross-validation
are recommended as a good compromise: see Breiman and Spector (1992)
and Kohavi (1995).
Figure 7.9 shows the prediction error and tenfold cross-validation curve
</p>
<p>estimated from a single training set, from the scenario in the bottom right
panel of Figure 7.3. This is a two-class classification problem, using a lin-</p>
<p/>
</div>
<div class="page"><p/>
<p>244 7. Model Assessment and Selection
</p>
<p>Subset Size p
</p>
<p>M
is
</p>
<p>c
la
</p>
<p>s
s
if
ic
</p>
<p>a
ti
o
n
 E
</p>
<p>rr
o
r
</p>
<p>5 10 15 20
</p>
<p>0
.0
</p>
<p>0
.1
</p>
<p>0
.2
</p>
<p>0
.3
</p>
<p>0
.4
</p>
<p>0
.5
</p>
<p>0
.6
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull; &bull;
</p>
<p>&bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull; &bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull; &bull; &bull; &bull; &bull; &bull; &bull;
&bull;
</p>
<p>&bull; &bull; &bull; &bull;
</p>
<p>FIGURE 7.9. Prediction error (orange) and tenfold cross-validation curve
(blue) estimated from a single training set, from the scenario in the bottom right
panel of Figure 7.3.
</p>
<p>ear model with best subsets regression of subset size p. Standard error bars
are shown, which are the standard errors of the individual misclassification
error rates for each of the ten parts. Both curves have minima at p = 10,
although the CV curve is rather flat beyond 10. Often a &ldquo;one-standard
error&rdquo; rule is used with cross-validation, in which we choose the most par-
simonious model whose error is no more than one standard error above
the error of the best model. Here it looks like a model with about p = 9
predictors would be chosen, while the true model uses p = 10.
Generalized cross-validation provides a convenient approximation to leave-
</p>
<p>one out cross-validation, for linear fitting under squared-error loss. As de-
fined in Section 7.6, a linear fitting method is one for which we can write
</p>
<p>ŷ = Sy. (7.50)
</p>
<p>Now for many linear fitting methods,
</p>
<p>1
</p>
<p>N
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>[yi &minus; f̂&minus;i(xi)]2 =
1
</p>
<p>N
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>[yi &minus; f̂(xi)
1&minus; Sii
</p>
<p>]2
, (7.51)
</p>
<p>where Sii is the ith diagonal element of S (see Exercise 7.3). The GCV
approximation is
</p>
<p>GCV(f̂) =
1
</p>
<p>N
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>[
yi &minus; f̂(xi)
</p>
<p>1&minus; trace(S)/N
</p>
<p>]2
. (7.52)</p>
<p/>
</div>
<div class="page"><p/>
<p>7.10 Cross-Validation 245
</p>
<p>The quantity trace(S) is the effective number of parameters, as defined in
Section 7.6.
GCV can have a computational advantage in some settings, where the
</p>
<p>trace of S can be computed more easily than the individual elements Sii.
In smoothing problems, GCV can also alleviate the tendency of cross-
validation to undersmooth. The similarity between GCV and AIC can be
seen from the approximation 1/(1&minus; x)2 &asymp; 1 + 2x (Exercise 7.7).
</p>
<p>7.10.2 The Wrong and Right Way to Do Cross-validation
</p>
<p>Consider a classification problem with a large number of predictors, as may
arise, for example, in genomic or proteomic applications. A typical strategy
for analysis might be as follows:
</p>
<p>1. Screen the predictors: find a subset of &ldquo;good&rdquo; predictors that show
fairly strong (univariate) correlation with the class labels
</p>
<p>2. Using just this subset of predictors, build a multivariate classifier.
</p>
<p>3. Use cross-validation to estimate the unknown tuning parameters and
to estimate the prediction error of the final model.
</p>
<p>Is this a correct application of cross-validation? Consider a scenario with
N = 50 samples in two equal-sized classes, and p = 5000 quantitative
predictors (standard Gaussian) that are independent of the class labels.
The true (test) error rate of any classifier is 50%. We carried out the above
recipe, choosing in step (1) the 100 predictors having highest correlation
with the class labels, and then using a 1-nearest neighbor classifier, based
on just these 100 predictors, in step (2). Over 50 simulations from this
setting, the average CV error rate was 3%. This is far lower than the true
error rate of 50%.
What has happened? The problem is that the predictors have an unfair
</p>
<p>advantage, as they were chosen in step (1) on the basis of all of the samples.
Leaving samples out after the variables have been selected does not cor-
rectly mimic the application of the classifier to a completely independent
test set, since these predictors &ldquo;have already seen&rdquo; the left out samples.
Figure 7.10 (top panel) illustrates the problem. We selected the 100 pre-
</p>
<p>dictors having largest correlation with the class labels over all 50 samples.
Then we chose a random set of 10 samples, as we would do in five-fold cross-
validation, and computed the correlations of the pre-selected 100 predictors
with the class labels over just these 10 samples (top panel). We see that
the correlations average about 0.28, rather than 0, as one might expect.
Here is the correct way to carry out cross-validation in this example:
</p>
<p>1. Divide the samples into K cross-validation folds (groups) at random.
</p>
<p>2. For each fold k = 1, 2, . . . ,K</p>
<p/>
</div>
<div class="page"><p/>
<p>246 7. Model Assessment and Selection
</p>
<p>Correlations of Selected Predictors with Outcome
</p>
<p>F
re
</p>
<p>q
u
e
n
c
y
</p>
<p>&minus;1.0 &minus;0.5 0.0 0.5 1.0
</p>
<p>0
1
0
</p>
<p>2
0
</p>
<p>3
0
</p>
<p>Wrong way
</p>
<p>Correlations of Selected Predictors with Outcome
</p>
<p>F
re
</p>
<p>q
u
e
n
c
y
</p>
<p>&minus;1.0 &minus;0.5 0.0 0.5 1.0
</p>
<p>0
1
0
</p>
<p>2
0
</p>
<p>3
0
</p>
<p>Right way
</p>
<p>FIGURE 7.10. Cross-validation the wrong and right way: histograms shows the
correlation of class labels, in 10 randomly chosen samples, with the 100 predic-
tors chosen using the incorrect (upper red) and correct (lower green) versions of
cross-validation.
</p>
<p>(a) Find a subset of &ldquo;good&rdquo; predictors that show fairly strong (uni-
variate) correlation with the class labels, using all of the samples
except those in fold k.
</p>
<p>(b) Using just this subset of predictors, build a multivariate classi-
fier, using all of the samples except those in fold k.
</p>
<p>(c) Use the classifier to predict the class labels for the samples in
fold k.
</p>
<p>The error estimates from step 2(c) are then accumulated over allK folds, to
produce the cross-validation estimate of prediction error. The lower panel
of Figure 7.10 shows the correlations of class labels with the 100 predictors
chosen in step 2(a) of the correct procedure, over the samples in a typical
fold k. We see that they average about zero, as they should.
</p>
<p>In general, with a multistep modeling procedure, cross-validation must
be applied to the entire sequence of modeling steps. In particular, samples
must be &ldquo;left out&rdquo; before any selection or filtering steps are applied. There
is one qualification: initial unsupervised screening steps can be done be-
fore samples are left out. For example, we could select the 1000 predictors</p>
<p/>
</div>
<div class="page"><p/>
<p>7.10 Cross-Validation 247
</p>
<p>with highest variance across all 50 samples, before starting cross-validation.
Since this filtering does not involve the class labels, it does not give the
predictors an unfair advantage.
While this point may seem obvious to the reader, we have seen this
</p>
<p>blunder committed many times in published papers in top rank journals.
With the large numbers of predictors that are so common in genomic and
other areas, the potential consequences of this error have also increased
dramatically; see Ambroise and McLachlan (2002) for a detailed discussion
of this issue.
</p>
<p>7.10.3 Does Cross-Validation Really Work?
</p>
<p>We once again examine the behavior of cross-validation in a high-dimensional
classification problem. Consider a scenario with N = 20 samples in two
equal-sized classes, and p = 500 quantitative predictors that are indepen-
dent of the class labels. Once again, the true error rate of any classifier is
50%. Consider a simple univariate classifier: a single split that minimizes
the misclassification error (a &ldquo;stump&rdquo;). Stumps are trees with a single split,
and are used in boosting methods (Chapter 10). A simple argument sug-
gests that cross-validation will not work properly in this setting2:
</p>
<p>Fitting to the entire training set, we will find a predictor that
splits the data very well. If we do 5-fold cross-validation, this
same predictor should split any 4/5ths and 1/5th of the data
well too, and hence its cross-validation error will be small (much
less than 50%.) Thus CV does not give an accurate estimate of
error.
</p>
<p>To investigate whether this argument is correct, Figure 7.11 shows the
result of a simulation from this setting. There are 500 predictors and 20
samples, in each of two equal-sized classes, with all predictors having a
standard Gaussian distribution. The panel in the top left shows the number
of training errors for each of the 500 stumps fit to the training data. We
have marked in color the six predictors yielding the fewest errors. In the top
right panel, the training errors are shown for stumps fit to a random 4/5ths
partition of the data (16 samples), and tested on the remaining 1/5th (four
samples). The colored points indicate the same predictors marked in the
top left panel. We see that the stump for the blue predictor (whose stump
was the best in the top left panel), makes two out of four test errors (50%),
and is no better than random.
What has happened? The preceding argument has ignored the fact that
</p>
<p>in cross-validation, the model must be completely retrained for each fold
</p>
<p>2This argument was made to us by a scientist at a proteomics lab meeting, and led
</p>
<p>to material in this section.</p>
<p/>
</div>
<div class="page"><p/>
<p>248 7. Model Assessment and Selection
</p>
<p>0 100 200 300 400 500
</p>
<p>2
3
</p>
<p>4
5
</p>
<p>6
7
</p>
<p>8
9
</p>
<p>Predictor
</p>
<p>E
rr
</p>
<p>o
r 
</p>
<p>o
n
 F
</p>
<p>u
ll 
</p>
<p>T
ra
</p>
<p>in
in
</p>
<p>g
 S
</p>
<p>e
t
</p>
<p>1 2 3 4 5 6 7 8
</p>
<p>0
1
</p>
<p>2
3
</p>
<p>4
Error on 4/5
</p>
<p>E
rr
</p>
<p>o
r 
</p>
<p>o
n
 1
</p>
<p>/5
</p>
<p>&minus;1 0 1 2
</p>
<p>Predictor 436 (blue)
</p>
<p>C
la
</p>
<p>s
s
 L
</p>
<p>a
b
e
l
</p>
<p>0
1
</p>
<p>full
4/5
</p>
<p>0
.0
</p>
<p>0
.2
</p>
<p>0
.4
</p>
<p>0
.6
</p>
<p>0
.8
</p>
<p>1
.0
</p>
<p>CV Errors
</p>
<p>FIGURE 7.11. Simulation study to investigate the performance of cross vali-
dation in a high-dimensional problem where the predictors are independent of the
class labels. The top-left panel shows the number of errors made by individual
stump classifiers on the full training set (20 observations). The top right panel
shows the errors made by individual stumps trained on a random split of the
dataset into 4/5ths (16 observations) and tested on the remaining 1/5th (4 ob-
servations). The best performers are depicted by colored dots in each panel. The
bottom left panel shows the effect of re-estimating the split point in each fold: the
colored points correspond to the four samples in the 1/5th validation set. The split
point derived from the full dataset classifies all four samples correctly, but when
the split point is re-estimated on the 4/5ths data (as it should be), it commits
two errors on the four validation samples. In the bottom right we see the overall
result of five-fold cross-validation applied to 50 simulated datasets. The average
error rate is about 50%, as it should be.</p>
<p/>
</div>
<div class="page"><p/>
<p>7.11 Bootstrap Methods 249
</p>
<p>of the process. In the present example, this means that the best predictor
and corresponding split point are found from 4/5ths of the data. The effect
of predictor choice is seen in the top right panel. Since the class labels are
independent of the predictors, the performance of a stump on the 4/5ths
training data contains no information about its performance in the remain-
ing 1/5th. The effect of the choice of split point is shown in the bottom left
panel. Here we see the data for predictor 436, corresponding to the blue
dot in the top left plot. The colored points indicate the 1/5th data, while
the remaining points belong to the 4/5ths. The optimal split points for this
predictor based on both the full training set and 4/5ths data are indicated.
The split based on the full data makes no errors on the 1/5ths data. But
cross-validation must base its split on the 4/5ths data, and this incurs two
errors out of four samples.
The results of applying five-fold cross-validation to each of 50 simulated
</p>
<p>datasets is shown in the bottom right panel. As we would hope, the average
cross-validation error is around 50%, which is the true expected prediction
error for this classifier. Hence cross-validation has behaved as it should.
On the other hand, there is considerable variability in the error, underscor-
ing the importance of reporting the estimated standard error of the CV
estimate. See Exercise 7.10 for another variation of this problem.
</p>
<p>7.11 Bootstrap Methods
</p>
<p>The bootstrap is a general tool for assessing statistical accuracy. First we
describe the bootstrap in general, and then show how it can be used to
estimate extra-sample prediction error. As with cross-validation, the boot-
strap seeks to estimate the conditional error ErrT , but typically estimates
well only the expected prediction error Err.
Suppose we have a model fit to a set of training data. We denote the
</p>
<p>training set by Z = (z1, z2, . . . , zN ) where zi = (xi, yi). The basic idea is
to randomly draw datasets with replacement from the training data, each
sample the same size as the original training set. This is done B times
(B = 100 say), producing B bootstrap datasets, as shown in Figure 7.12.
Then we refit the model to each of the bootstrap datasets, and examine
the behavior of the fits over the B replications.
In the figure, S(Z) is any quantity computed from the data Z, for ex-
</p>
<p>ample, the prediction at some input point. From the bootstrap sampling
we can estimate any aspect of the distribution of S(Z), for example, its
variance,
</p>
<p>V̂ar[S(Z)] =
1
</p>
<p>B &minus; 1
B&sum;
</p>
<p>b=1
</p>
<p>(S(Z&lowast;b)&minus; S̄&lowast;)2, (7.53)</p>
<p/>
</div>
<div class="page"><p/>
<p>250 7. Model Assessment and Selection
</p>
<p>  
</p>
<p>Bootstrap
</p>
<p>Bootstrap
</p>
<p>replications
</p>
<p>samples
</p>
<p>sample
TrainingZ = (z1, z2, . . . , zN )
</p>
<p>Z&lowast;1 Z&lowast;2 Z&lowast;B
</p>
<p>S(Z&lowast;1) S(Z&lowast;2) S(Z&lowast;B)
</p>
<p>FIGURE 7.12. Schematic of the bootstrap process. We wish to assess the sta-
tistical accuracy of a quantity S(Z) computed from our dataset. B training sets
Z&lowast;b, b = 1, . . . , B each of size N are drawn with replacement from the original
dataset. The quantity of interest S(Z) is computed from each bootstrap training
set, and the values S(Z&lowast;1), . . . , S(Z&lowast;B) are used to assess the statistical accuracy
of S(Z).
</p>
<p>where S̄&lowast; =
&sum;
</p>
<p>b S(Z
&lowast;b)/B. Note that V̂ar[S(Z)] can be thought of as a
</p>
<p>Monte-Carlo estimate of the variance of S(Z) under sampling from the
empirical distribution function F̂ for the data (z1, z2, . . . , zN ).
How can we apply the bootstrap to estimate prediction error? One ap-
</p>
<p>proach would be to fit the model in question on a set of bootstrap samples,
and then keep track of how well it predicts the original training set. If
f̂&lowast;b(xi) is the predicted value at xi, from the model fitted to the bth boot-
strap dataset, our estimate is
</p>
<p>Êrrboot =
1
</p>
<p>B
</p>
<p>1
</p>
<p>N
</p>
<p>B&sum;
</p>
<p>b=1
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>L(yi, f̂
&lowast;b(xi)). (7.54)
</p>
<p>However, it is easy to see that Êrrboot does not provide a good estimate in
general. The reason is that the bootstrap datasets are acting as the training
samples, while the original training set is acting as the test sample, and
these two samples have observations in common. This overlap can make
overfit predictions look unrealistically good, and is the reason that cross-
validation explicitly uses non-overlapping data for the training and test
samples. Consider for example a 1-nearest neighbor classifier applied to a
two-class classification problem with the same number of observations in</p>
<p/>
</div>
<div class="page"><p/>
<p>7.11 Bootstrap Methods 251
</p>
<p>each class, in which the predictors and class labels are in fact independent.
Then the true error rate is 0.5. But the contributions to the bootstrap
estimate Êrrboot will be zero unless the observation i does not appear in the
bootstrap sample b. In this latter case it will have the correct expectation
0.5. Now
</p>
<p>Pr{observation i &isin; bootstrap sample b} = 1&minus;
(
1&minus; 1
</p>
<p>N
</p>
<p>)N
</p>
<p>&asymp; 1&minus; e&minus;1
= 0.632. (7.55)
</p>
<p>Hence the expectation of Êrrboot is about 0.5 &times; 0.368 = 0.184, far below
the correct error rate 0.5.
By mimicking cross-validation, a better bootstrap estimate can be ob-
</p>
<p>tained. For each observation, we only keep track of predictions from boot-
strap samples not containing that observation. The leave-one-out bootstrap
estimate of prediction error is defined by
</p>
<p>Êrr
(1)
</p>
<p>=
1
</p>
<p>N
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>1
</p>
<p>|C&minus;i|
&sum;
</p>
<p>b&isin;C&minus;i
</p>
<p>L(yi, f̂
&lowast;b(xi)). (7.56)
</p>
<p>Here C&minus;i is the set of indices of the bootstrap samples b that do not contain
</p>
<p>observation i, and |C&minus;i| is the number of such samples. In computing Êrr(1),
we either have to choose B large enough to ensure that all of the |C&minus;i| are
greater than zero, or we can just leave out the terms in (7.56) corresponding
to |C&minus;i|&rsquo;s that are zero.
The leave-one out bootstrap solves the overfitting problem suffered by
</p>
<p>Êrrboot, but has the training-set-size bias mentioned in the discussion of
cross-validation. The average number of distinct observations in each boot-
strap sample is about 0.632 &middot;N , so its bias will roughly behave like that of
twofold cross-validation. Thus if the learning curve has considerable slope
at sample size N/2, the leave-one out bootstrap will be biased upward as
an estimate of the true error.
The &ldquo;.632 estimator&rdquo; is designed to alleviate this bias. It is defined by
</p>
<p>Êrr
(.632)
</p>
<p>= .368 &middot; err + .632 &middot; Êrr(1). (7.57)
</p>
<p>The derivation of the .632 estimator is complex; intuitively it pulls the
leave-one out bootstrap estimate down toward the training error rate, and
hence reduces its upward bias. The use of the constant .632 relates to (7.55).
The .632 estimator works well in &ldquo;light fitting&rdquo; situations, but can break
</p>
<p>down in overfit ones. Here is an example due to Breiman et al. (1984).
Suppose we have two equal-size classes, with the targets independent of
the class labels, and we apply a one-nearest neighbor rule. Then err = 0,</p>
<p/>
</div>
<div class="page"><p/>
<p>252 7. Model Assessment and Selection
</p>
<p>Êrr
(1)
</p>
<p>= 0.5 and so Êrr
(.632)
</p>
<p>= .632 &times; 0.5 = .316. However, the true error
rate is 0.5.
One can improve the .632 estimator by taking into account the amount
</p>
<p>of overfitting. First we define γ to be the no-information error rate: this
is the error rate of our prediction rule if the inputs and class labels were
independent. An estimate of γ is obtained by evaluating the prediction rule
on all possible combinations of targets yi and predictors xi&prime;
</p>
<p>γ̂ =
1
</p>
<p>N2
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>N&sum;
</p>
<p>i&prime;=1
</p>
<p>L(yi, f̂(xi&prime;)). (7.58)
</p>
<p>For example, consider the dichotomous classification problem: let p̂1 be
the observed proportion of responses yi equaling 1, and let q̂1 be the ob-
served proportion of predictions f̂(xi&prime;) equaling 1. Then
</p>
<p>γ̂ = p̂1(1&minus; q̂1) + (1&minus; p̂1)q̂1. (7.59)
With a rule like 1-nearest neighbors for which q̂1 = p̂1 the value of γ̂ is
2p̂1(1&minus;p̂1). The multi-category generalization of (7.59) is γ̂ =
</p>
<p>&sum;
ℓ p̂ℓ(1&minus;q̂ℓ).
</p>
<p>Using this, the relative overfitting rate is defined to be
</p>
<p>R̂ =
Êrr
</p>
<p>(1) &minus; err
γ̂ &minus; err , (7.60)
</p>
<p>a quantity that ranges from 0 if there is no overfitting (Êrr
(1)
</p>
<p>= err) to 1
if the overfitting equals the no-information value γ̂&minus; err. Finally, we define
the &ldquo;.632+&rdquo; estimator by
</p>
<p>Êrr
(.632+)
</p>
<p>= (1&minus; ŵ) &middot; err + ŵ &middot; Êrr(1) (7.61)
</p>
<p>with ŵ =
.632
</p>
<p>1&minus; .368R̂
.
</p>
<p>The weight w ranges from .632 if R̂ = 0 to 1 if R̂ = 1, so Êrr
(.632+)
</p>
<p>ranges from Êrr
(.632)
</p>
<p>to Êrr
(1)
</p>
<p>. Again, the derivation of (7.61) is compli-
cated: roughly speaking, it produces a compromise between the leave-one-
out bootstrap and the training error rate that depends on the amount of
overfitting. For the 1-nearest-neighbor problem with class labels indepen-
</p>
<p>dent of the inputs, ŵ = R̂ = 1, so Êrr
(.632+)
</p>
<p>= Êrr
(1)
</p>
<p>, which has the correct
</p>
<p>expectation of 0.5. In other problems with less overfitting, Êrr
(.632+)
</p>
<p>will
</p>
<p>lie somewhere between err and Êrr
(1)
</p>
<p>.
</p>
<p>7.11.1 Example (Continued)
</p>
<p>Figure 7.13 shows the results of tenfold cross-validation and the .632+ boot-
strap estimate in the same four problems of Figures 7.7. As in that figure,</p>
<p/>
</div>
<div class="page"><p/>
<p>7.11 Bootstrap Methods 253
</p>
<p>reg/KNN reg/linear class/KNN class/linear
</p>
<p>0
2
0
</p>
<p>4
0
</p>
<p>6
0
</p>
<p>8
0
</p>
<p>1
0
0
</p>
<p>%
 I
n
c
re
</p>
<p>a
s
e
 O
</p>
<p>v
e
r 
</p>
<p>B
e
s
t
</p>
<p>Cross&minus;validation
</p>
<p>reg/KNN reg/linear class/KNN class/linear
</p>
<p>0
2
0
</p>
<p>4
0
</p>
<p>6
0
</p>
<p>8
0
</p>
<p>1
0
0
</p>
<p>%
 I
n
c
re
</p>
<p>a
s
e
 O
</p>
<p>v
e
r 
</p>
<p>B
e
s
t
</p>
<p>Bootstrap
</p>
<p>FIGURE 7.13. Boxplots show the distribution of the relative error
100 &middot; [Errα̂ &minus; minα Err(α)]/[maxα Err(α) &minus; minα Err(α)] over the four scenar-
ios of Figure 7.3. This is the error in using the chosen model relative to the best
model. There are 100 training sets represented in each boxplot.
</p>
<p>Figure 7.13 shows boxplots of 100 &middot; [Errα̂ &minus; minα Err(α)]/[maxα Err(α) &minus;
minα Err(α)], the error in using the chosen model relative to the best model.
There are 100 different training sets represented in each boxplot. Both mea-
sures perform well overall, perhaps the same or slightly worse than the AIC
in Figure 7.7.
Our conclusion is that for these particular problems and fitting methods,
</p>
<p>minimization of either AIC, cross-validation or bootstrap yields a model
fairly close to the best available. Note that for the purpose of model selec-
tion, any of the measures could be biased and it wouldn&rsquo;t affect things, as
long as the bias did not change the relative performance of the methods.
For example, the addition of a constant to any of the measures would not
change the resulting chosen model. However, for many adaptive, nonlinear
techniques (like trees), estimation of the effective number of parameters is
very difficult. This makes methods like AIC impractical and leaves us with
cross-validation or bootstrap as the methods of choice.
A different question is: how well does each method estimate test error?
</p>
<p>On the average the AIC criterion overestimated prediction error of its cho-</p>
<p/>
</div>
<div class="page"><p/>
<p>254 7. Model Assessment and Selection
</p>
<p>sen model by 38%, 37%, 51%, and 30%, respectively, over the four scenarios,
with BIC performing similarly. In contrast, cross-validation overestimated
the error by 1%, 4%, 0%, and 4%, with the bootstrap doing about the
same. Hence the extra work involved in computing a cross-validation or
bootstrap measure is worthwhile, if an accurate estimate of test error is
required. With other fitting methods like trees, cross-validation and boot-
strap can underestimate the true error by 10%, because the search for best
tree is strongly affected by the validation set. In these situations only a
separate test set will provide an unbiased estimate of test error.
</p>
<p>7.12 Conditional or Expected Test Error?
</p>
<p>Figures 7.14 and 7.15 examine the question of whether cross-validation does
a good job in estimating ErrT , the error conditional on a given training set
T (expression (7.15) on page 228), as opposed to the expected test error.
For each of 100 training sets generated from the &ldquo;reg/linear&rdquo; setting in
the top-right panel of Figure 7.3, Figure 7.14 shows the conditional error
curves ErrT as a function of subset size (top left). The next two panels show
10-fold and N -fold cross-validation, the latter also known as leave-one-out
(LOO). The thick red curve in each plot is the expected error Err, while
the thick black curves are the expected cross-validation curves. The lower
right panel shows how well cross-validation approximates the conditional
and expected error.
One might have expected N -fold CV to approximate ErrT well, since it
</p>
<p>almost uses the full training sample to fit a new test point. 10-fold CV, on
the other hand, might be expected to estimate Err well, since it averages
over somewhat different training sets. From the figure it appears 10-fold
does a better job than N -fold in estimating ErrT , and estimates Err even
better. Indeed, the similarity of the two black curves with the red curve
suggests both CV curves are approximately unbiased for Err, with 10-fold
having less variance. Similar trends were reported by Efron (1983).
Figure 7.15 shows scatterplots of both 10-fold andN -fold cross-validation
</p>
<p>error estimates versus the true conditional error for the 100 simulations.
Although the scatterplots do not indicate much correlation, the lower right
panel shows that for the most part the correlations are negative, a curi-
ous phenomenon that has been observed before. This negative correlation
explains why neither form of CV estimates ErrT well. The broken lines in
each plot are drawn at Err(p), the expected error for the best subset of
size p. We see again that both forms of CV are approximately unbiased for
expected error, but the variation in test error for different training sets is
quite substantial.
Among the four experimental conditions in 7.3, this &ldquo;reg/linear&rdquo; scenario
</p>
<p>showed the highest correlation between actual and predicted test error. This</p>
<p/>
</div>
<div class="page"><p/>
<p>7.12 Conditional or Expected Test Error? 255
</p>
<p>5 10 15 20
</p>
<p>0
.1
</p>
<p>0
.2
</p>
<p>0
.3
</p>
<p>0
.4
</p>
<p>Prediction Error
</p>
<p>Subset Size p
</p>
<p>E
rr
</p>
<p>o
r
</p>
<p>5 10 15 20
</p>
<p>0
.1
</p>
<p>0
.2
</p>
<p>0
.3
</p>
<p>0
.4
</p>
<p>10&minus;Fold CV Error
</p>
<p>Subset Size p
</p>
<p>E
rr
</p>
<p>o
r
</p>
<p>5 10 15 20
</p>
<p>0
.1
</p>
<p>0
.2
</p>
<p>0
.3
</p>
<p>0
.4
</p>
<p>Leave&minus;One&minus;Out CV Error
</p>
<p>Subset Size p
</p>
<p>E
rr
</p>
<p>o
r
</p>
<p>5 10 15 20
</p>
<p>0
.0
</p>
<p>1
5
</p>
<p>0
.0
</p>
<p>2
5
</p>
<p>0
.0
</p>
<p>3
5
</p>
<p>0
.0
</p>
<p>4
5
</p>
<p>Approximation Error
</p>
<p>Subset Size p
</p>
<p>M
e
a
n
 A
</p>
<p>b
s
o
lu
</p>
<p>te
 D
</p>
<p>e
v
ia
</p>
<p>ti
o
n
</p>
<p>ET |CV10&minus;Err|
ET |CV10&minus;ErrT |
ET |CVN&minus;ErrT |
</p>
<p>FIGURE 7.14. Conditional prediction-error ErrT , 10-fold cross-validation, and
leave-one-out cross-validation curves for a 100 simulations from the top-right
panel in Figure 7.3. The thick red curve is the expected prediction error Err,
while the thick black curves are the expected CV curves ET CV10 and ET CVN .
The lower-right panel shows the mean absolute deviation of the CV curves from
the conditional error, ET |CVK &minus; ErrT | for K = 10 (blue) and K = N (green),
as well as from the expected error ET |CV10 &minus; Err| (orange).</p>
<p/>
</div>
<div class="page"><p/>
<p>256 7. Model Assessment and Selection
</p>
<p>0.10 0.15 0.20 0.25 0.30 0.35 0.40
</p>
<p>0
.1
</p>
<p>0
0
</p>
<p>.2
0
</p>
<p>0
.3
</p>
<p>0
0
</p>
<p>.4
0
</p>
<p>Subset Size 1
</p>
<p>Prediction Error
</p>
<p>C
V
</p>
<p> E
rr
</p>
<p>o
r
</p>
<p>0.10 0.15 0.20 0.25 0.30 0.35 0.40
</p>
<p>0
.1
</p>
<p>0
0
</p>
<p>.2
0
</p>
<p>0
.3
</p>
<p>0
0
</p>
<p>.4
0
</p>
<p>Subset Size 5
</p>
<p>Prediction Error
</p>
<p>C
V
</p>
<p> E
rr
</p>
<p>o
r
</p>
<p>0.10 0.15 0.20 0.25 0.30 0.35 0.40
</p>
<p>0
.1
</p>
<p>0
0
</p>
<p>.2
0
</p>
<p>0
.3
</p>
<p>0
0
</p>
<p>.4
0
</p>
<p>Subset Size 10
</p>
<p>Prediction Error
</p>
<p>C
V
</p>
<p> E
rr
</p>
<p>o
r
</p>
<p>5 10 15 20
</p>
<p>&minus;
0
</p>
<p>.6
&minus;
</p>
<p>0
.4
</p>
<p>&minus;
0
</p>
<p>.2
0
</p>
<p>.0
0
</p>
<p>.2
</p>
<p>Subset Size
</p>
<p>C
o
</p>
<p>rr
e
</p>
<p>la
ti
o
</p>
<p>n
</p>
<p>Leave&minus;one&minus;out
</p>
<p>10&minus;Fold
</p>
<p>FIGURE 7.15. Plots of the CV estimates of error versus the true conditional
error for each of the 100 training sets, for the simulation setup in the top right
panel Figure 7.3. Both 10-fold and leave-one-out CV are depicted in different
colors. The first three panels correspond to different subset sizes p, and vertical
and horizontal lines are drawn at Err(p). Although there appears to be little cor-
relation in these plots, we see in the lower right panel that for the most part the
correlation is negative.</p>
<p/>
</div>
<div class="page"><p/>
<p>Exercises 257
</p>
<p>phenomenon also occurs for bootstrap estimates of error, and we would
guess, for any other estimate of conditional prediction error.
We conclude that estimation of test error for a particular training set is
</p>
<p>not easy in general, given just the data from that same training set. Instead,
cross-validation and related methods may provide reasonable estimates of
the expected error Err.
</p>
<p>Bibliographic Notes
</p>
<p>Key references for cross-validation are Stone (1974), Stone (1977) and
Allen (1974). The AIC was proposed by Akaike (1973), while the BIC
was introduced by Schwarz (1978). Madigan and Raftery (1994) give an
overview of Bayesian model selection. The MDL criterion is due to Rissa-
nen (1983). Cover and Thomas (1991) contains a good description of coding
theory and complexity. VC dimension is described in Vapnik (1996). Stone
(1977) showed that the AIC and leave-one out cross-validation are asymp-
totically equivalent. Generalized cross-validation is described by Golub et
al. (1979) and Wahba (1980); a further discussion of the topic may be found
in the monograph by Wahba (1990). See also Hastie and Tibshirani (1990),
Chapter 3. The bootstrap is due to Efron (1979); see Efron and Tibshi-
rani (1993) for an overview. Efron (1983) proposes a number of bootstrap
estimates of prediction error, including the optimism and .632 estimates.
Efron (1986) compares CV, GCV and bootstrap estimates of error rates.
The use of cross-validation and the bootstrap for model selection is stud-
ied by Breiman and Spector (1992), Breiman (1992), Shao (1996), Zhang
(1993) and Kohavi (1995). The .632+ estimator was proposed by Efron
and Tibshirani (1997).
Cherkassky and Ma (2003) published a study on the performance of
</p>
<p>SRM for model selection in regression, in response to our study of section
7.9.1. They complained that we had been unfair to SRM because had not
applied it properly. Our response can be found in the same issue of the
journal (Hastie et al. (2003)).
</p>
<p>Exercises
</p>
<p>Ex. 7.1 Derive the estimate of in-sample error (7.24).
</p>
<p>Ex. 7.2 For 0&ndash;1 loss with Y &isin; {0, 1} and Pr(Y = 1|x0) = f(x0), show that
</p>
<p>Err(x0) = Pr(Y �= Ĝ(x0)|X = x0)
= ErrB(x0) + |2f(x0)&minus; 1|Pr(Ĝ(x0) �= G(x0)|X = x0),
</p>
<p>(7.62)</p>
<p/>
</div>
<div class="page"><p/>
<p>258 7. Model Assessment and Selection
</p>
<p>where Ĝ(x) = I(f̂(x) &gt; 12 ), G(x) = I(f(x) &gt;
1
2 ) is the Bayes classifier,
</p>
<p>and ErrB(x0) = Pr(Y �= G(x0)|X = x0), the irreducible Bayes error at x0.
Using the approximation f̂(x0) &sim; N(Ef̂(x0),Var(f̂(x0)), show that
</p>
<p>Pr(Ĝ(x0) �= G(x0)|X = x0) &asymp; Φ
(
sign( 12 &minus; f(x0))(Ef̂(x0)&minus; 12 )&radic;
</p>
<p>Var(f̂(x0))
</p>
<p>)
. (7.63)
</p>
<p>In the above,
</p>
<p>Φ(t) =
1&radic;
2π
</p>
<p>&int; t
</p>
<p>&minus;&infin;
</p>
<p>exp(&minus;t2/2)dt,
</p>
<p>the cumulative Gaussian distribution function. This is an increasing func-
tion, with value 0 at t = &minus;&infin; and value 1 at t = +&infin;.
</p>
<p>We can think of sign( 12 &minus; f(x0))(Ef̂(x0) &minus; 12 ) as a kind of boundary-
bias term, as it depends on the true f(x0) only through which side of the
boundary ( 12 ) that it lies. Notice also that the bias and variance combine
</p>
<p>in a multiplicative rather than additive fashion. If Ef̂(x0) is on the same
side of 12 as f(x0), then the bias is negative, and decreasing the variance
</p>
<p>will decrease the misclassification error. On the other hand, if Ef̂(x0) is
on the opposite side of 12 to f(x0), then the bias is positive and it pays to
</p>
<p>increase the variance! Such an increase will improve the chance that f̂(x0)
falls on the correct side of 12 (Friedman, 1997).
</p>
<p>Ex. 7.3 Let f̂ = Sy be a linear smoothing of y.
</p>
<p>(a) If Sii is the ith diagonal element of S, show that for S arising from least
squares projections and cubic smoothing splines, the cross-validated
residual can be written as
</p>
<p>yi &minus; f̂&minus;i(xi) =
yi &minus; f̂(xi)
1&minus; Sii
</p>
<p>. (7.64)
</p>
<p>(b) Use this result to show that |yi &minus; f̂&minus;i(xi)| &ge; |yi &minus; f̂(xi)|.
</p>
<p>(c) Find general conditions on any smoother S to make result (7.64) hold.
</p>
<p>Ex. 7.4 Consider the in-sample prediction error (7.18) and the training
error err in the case of squared-error loss:
</p>
<p>Errin =
1
</p>
<p>N
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>EY 0(Y
0
i &minus; f̂(xi))2
</p>
<p>err =
1
</p>
<p>N
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>(yi &minus; f̂(xi))2.</p>
<p/>
</div>
<div class="page"><p/>
<p>Exercises 259
</p>
<p>Add and subtract f(xi) and Ef̂(xi) in each expression and expand. Hence
establish that the average optimism in the training error is
</p>
<p>2
</p>
<p>N
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>Cov(ŷi, yi),
</p>
<p>as given in (7.21).
</p>
<p>Ex. 7.5 For a linear smoother ŷ = Sy, show that
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>Cov(ŷi, yi) = trace(S)σ
2
ε , (7.65)
</p>
<p>which justifies its use as the effective number of parameters.
</p>
<p>Ex. 7.6 Show that for an additive-error model, the effective degrees-of-
freedom for the k-nearest-neighbors regression fit is N/k.
</p>
<p>Ex. 7.7 Use the approximation 1/(1&minus;x)2 &asymp; 1+2x to expose the relationship
between Cp/AIC (7.26) and GCV (7.52), the main difference being the
model used to estimate the noise variance σ2ε .
</p>
<p>Ex. 7.8 Show that the set of functions {I(sin(αx) &gt; 0)} can shatter the
following points on the line:
</p>
<p>z1 = 10&minus;1, . . . , zℓ = 10&minus;ℓ, (7.66)
</p>
<p>for any ℓ. Hence the VC dimension of the class {I(sin(αx) &gt; 0)} is infinite.
</p>
<p>Ex. 7.9 For the prostate data of Chapter 3, carry out a best-subset linear
regression analysis, as in Table 3.3 (third column from left). Compute the
AIC, BIC, five- and tenfold cross-validation, and bootstrap .632 estimates
of prediction error. Discuss the results.
</p>
<p>Ex. 7.10 Referring to the example in Section 7.10.3, suppose instead that
all of the p predictors are binary, and hence there is no need to estimate
split points. The predictors are independent of the class labels as before.
Then if p is very large, we can probably find a predictor that splits the
entire training data perfectly, and hence would split the validation data
(one-fifth of data) perfectly as well. This predictor would therefore have
zero cross-validation error. Does this mean that cross-validation does not
provide a good estimate of test error in this situation? [This question was
suggested by Li Ma.]</p>
<p/>
</div>
<div class="page"><p/>
<p>8
Model Inference and Averaging
</p>
<p>8.1 Introduction
</p>
<p>For most of this book, the fitting (learning) of models has been achieved by
minimizing a sum of squares for regression, or by minimizing cross-entropy
for classification. In fact, both of these minimizations are instances of the
maximum likelihood approach to fitting.
In this chapter we provide a general exposition of the maximum likeli-
</p>
<p>hood approach, as well as the Bayesian method for inference. The boot-
strap, introduced in Chapter 7, is discussed in this context, and its relation
to maximum likelihood and Bayes is described. Finally, we present some
related techniques for model averaging and improvement, including com-
mittee methods, bagging, stacking and bumping.
</p>
<p>8.2 The Bootstrap and Maximum Likelihood
Methods
</p>
<p>8.2.1 A Smoothing Example
</p>
<p>The bootstrap method provides a direct computational way of assessing
uncertainty, by sampling from the training data. Here we illustrate the
bootstrap in a simple one-dimensional smoothing problem, and show its
connection to maximum likelihood.
</p>
<p> &copy; Springer Science+Business Media, LLC 2009 
</p>
<p>T. Hastie et al., The Elements of Statistical Learning, Second Edition, 261
</p>
<p>DOI: 10.1007/b94608_8,</p>
<p/>
</div>
<div class="page"><p/>
<p>262 8. Model Inference and Averaging
</p>
<p>0.0 0.5 1.0 1.5 2.0 2.5 3.0
</p>
<p>-1
0
</p>
<p>1
2
</p>
<p>3
4
</p>
<p>5
</p>
<p>x
</p>
<p>y
</p>
<p>&bull;
&bull;&bull;&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;&bull;
&bull;&bull;&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
&bull;&bull;
</p>
<p>&bull;
&bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;&bull;
&bull;
&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>0.0 0.5 1.0 1.5 2.0 2.5 3.0
</p>
<p>0
.0
</p>
<p>0
.2
</p>
<p>0
.4
</p>
<p>0
.6
</p>
<p>0
.8
</p>
<p>1
.0
</p>
<p>x
</p>
<p>B
-s
</p>
<p>p
lin
</p>
<p>e
 B
</p>
<p>a
s
is
</p>
<p>FIGURE 8.1. (Left panel): Data for smoothing example. (Right panel:) Set of
seven B-spline basis functions. The broken vertical lines indicate the placement
of the three knots.
</p>
<p>Denote the training data by Z = {z1, z2, . . . , zN}, with zi = (xi, yi),
i = 1, 2, . . . , N . Here xi is a one-dimensional input, and yi the outcome,
either continuous or categorical. As an example, consider the N = 50 data
points shown in the left panel of Figure 8.1.
Suppose we decide to fit a cubic spline to the data, with three knots
</p>
<p>placed at the quartiles of the X values. This is a seven-dimensional lin-
ear space of functions, and can be represented, for example, by a linear
expansion of B-spline basis functions (see Section 5.9.2):
</p>
<p>μ(x) =
</p>
<p>7&sum;
</p>
<p>j=1
</p>
<p>βjhj(x). (8.1)
</p>
<p>Here the hj(x), j = 1, 2, . . . , 7 are the seven functions shown in the right
panel of Figure 8.1. We can think of μ(x) as representing the conditional
mean E(Y |X = x).
Let H be the N &times;7 matrix with ijth element hj(xi). The usual estimate
</p>
<p>of β, obtained by minimizing the squared error over the training set, is
given by
</p>
<p>β̂ = (HTH)&minus;1HTy. (8.2)
</p>
<p>The corresponding fit μ̂(x) =
&sum;7
</p>
<p>j=1 β̂jhj(x) is shown in the top left panel
of Figure 8.2.
The estimated covariance matrix of β̂ is
</p>
<p>V̂ar(β̂) = (HTH)&minus;1σ̂2, (8.3)
</p>
<p>where we have estimated the noise variance by σ̂2 =
&sum;N
</p>
<p>i=1(yi &minus; μ̂(xi))2/N .
Letting h(x)T = (h1(x), h2(x), . . . , h7(x)), the standard error of a predic-</p>
<p/>
</div>
<div class="page"><p/>
<p>8.2 The Bootstrap and Maximum Likelihood Methods 263
</p>
<p>0.0 0.5 1.0 1.5 2.0 2.5 3.0
</p>
<p>-1
0
</p>
<p>1
2
</p>
<p>3
4
</p>
<p>5
</p>
<p>&bull;
&bull;&bull;&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;&bull;
&bull;&bull;&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
&bull;&bull;
</p>
<p>&bull;
&bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;&bull;
&bull;
&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>x
</p>
<p>y
</p>
<p>0.0 0.5 1.0 1.5 2.0 2.5 3.0
</p>
<p>-1
0
</p>
<p>1
2
</p>
<p>3
4
</p>
<p>5
</p>
<p>x
</p>
<p>y
&bull;
&bull;&bull;&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;&bull;
&bull;&bull;&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
&bull;&bull;
</p>
<p>&bull;
&bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;&bull;
&bull;
&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>0.0 0.5 1.0 1.5 2.0 2.5 3.0
</p>
<p>-1
0
</p>
<p>1
2
</p>
<p>3
4
</p>
<p>5
</p>
<p>x
</p>
<p>y
</p>
<p>&bull;
&bull;&bull;&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;&bull;
&bull;&bull;&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
&bull;&bull;
</p>
<p>&bull;
&bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;&bull;
&bull;
&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>0.0 0.5 1.0 1.5 2.0 2.5 3.0
</p>
<p>-1
0
</p>
<p>1
2
</p>
<p>3
4
</p>
<p>5
</p>
<p>x
</p>
<p>y
</p>
<p>&bull;
&bull;&bull;&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;&bull;
&bull;&bull;&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
&bull;&bull;
</p>
<p>&bull;
&bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;&bull;
&bull;
&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>FIGURE 8.2. (Top left:) B-spline smooth of data. (Top right:) B-spline smooth
plus and minus 1.96&times; standard error bands. (Bottom left:) Ten bootstrap repli-
cates of the B-spline smooth. (Bottom right:) B-spline smooth with 95% standard
error bands computed from the bootstrap distribution.</p>
<p/>
</div>
<div class="page"><p/>
<p>264 8. Model Inference and Averaging
</p>
<p>tion μ̂(x) = h(x)T β̂ is
</p>
<p>ŝe[μ̂(x)] = [h(x)T (HTH)&minus;1h(x)]
1
2 σ̂. (8.4)
</p>
<p>In the top right panel of Figure 8.2 we have plotted μ̂(x)&plusmn;1.96 &middot; ŝe[μ̂(x)].
Since 1.96 is the 97.5% point of the standard normal distribution, these
represent approximate 100 &minus; 2 &times; 2.5% = 95% pointwise confidence bands
for μ(x).
Here is how we could apply the bootstrap in this example. We draw B
</p>
<p>datasets each of size N = 50 with replacement from our training data, the
sampling unit being the pair zi = (xi, yi). To each bootstrap dataset Z
</p>
<p>&lowast;
</p>
<p>we fit a cubic spline μ̂&lowast;(x); the fits from ten such samples are shown in the
bottom left panel of Figure 8.2. Using B = 200 bootstrap samples, we can
form a 95% pointwise confidence band from the percentiles at each x: we
find the 2.5%&times; 200 = fifth largest and smallest values at each x. These are
plotted in the bottom right panel of Figure 8.2. The bands look similar to
those in the top right, being a little wider at the endpoints.
There is actually a close connection between the least squares estimates
</p>
<p>(8.2) and (8.3), the bootstrap, and maximum likelihood. Suppose we further
assume that the model errors are Gaussian,
</p>
<p>Y = μ(X) + ε; ε &sim; N(0, σ2),
</p>
<p>μ(x) =
</p>
<p>7&sum;
</p>
<p>j=1
</p>
<p>βjhj(x). (8.5)
</p>
<p>The bootstrap method described above, in which we sample with re-
placement from the training data, is called the nonparametric bootstrap.
This really means that the method is &ldquo;model-free,&rdquo; since it uses the raw
data, not a specific parametric model, to generate new datasets. Consider
a variation of the bootstrap, called the parametric bootstrap, in which we
simulate new responses by adding Gaussian noise to the predicted values:
</p>
<p>y&lowast;i = μ̂(xi) + ε
&lowast;
i ; ε
</p>
<p>&lowast;
i &sim; N(0, σ̂2); i = 1, 2, . . . , N. (8.6)
</p>
<p>This process is repeated B times, where B = 200 say. The resulting boot-
strap datasets have the form (x1, y
</p>
<p>&lowast;
1), . . . , (xN , y
</p>
<p>&lowast;
N ) and we recompute the
</p>
<p>B-spline smooth on each. The confidence bands from this method will ex-
actly equal the least squares bands in the top right panel, as the number of
bootstrap samples goes to infinity. A function estimated from a bootstrap
sample y&lowast; is given by μ̂&lowast;(x) = h(x)T (HTH)&minus;1HTy&lowast;, and has distribution
</p>
<p>μ̂&lowast;(x) &sim; N(μ̂(x), h(x)T (HTH)&minus;1h(x)σ̂2). (8.7)
</p>
<p>Notice that the mean of this distribution is the least squares estimate, and
the standard deviation is the same as the approximate formula (8.4).</p>
<p/>
</div>
<div class="page"><p/>
<p>8.2 The Bootstrap and Maximum Likelihood Methods 265
</p>
<p>8.2.2 Maximum Likelihood Inference
</p>
<p>It turns out that the parametric bootstrap agrees with least squares in the
previous example because the model (8.5) has additive Gaussian errors. In
general, the parametric bootstrap agrees not with least squares but with
maximum likelihood, which we now review.
We begin by specifying a probability density or probability mass function
</p>
<p>for our observations
</p>
<p>zi &sim; gθ(z). (8.8)
</p>
<p>In this expression θ represents one or more unknown parameters that gov-
ern the distribution of Z. This is called a parametric model for Z. As an
example, if Z has a normal distribution with mean μ and variance σ2, then
</p>
<p>θ = (μ, σ2), (8.9)
</p>
<p>and
</p>
<p>gθ(z) =
1&radic;
2πσ
</p>
<p>e&minus;
1
2 (z&minus;μ)
</p>
<p>2/σ2 . (8.10)
</p>
<p>Maximum likelihood is based on the likelihood function, given by
</p>
<p>L(θ;Z) =
</p>
<p>N&prod;
</p>
<p>i=1
</p>
<p>gθ(zi), (8.11)
</p>
<p>the probability of the observed data under the model gθ. The likelihood is
defined only up to a positive multiplier, which we have taken to be one.
We think of L(θ;Z) as a function of θ, with our data Z fixed.
</p>
<p>Denote the logarithm of L(θ;Z) by
</p>
<p>ℓ(θ;Z) =
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>ℓ(θ; zi)
</p>
<p>=
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>log gθ(zi), (8.12)
</p>
<p>which we will sometimes abbreviate as ℓ(θ). This expression is called the
log-likelihood, and each value ℓ(θ; zi) = log gθ(zi) is called a log-likelihood
</p>
<p>component. The method of maximum likelihood chooses the value θ = θ̂
to maximize ℓ(θ;Z).
</p>
<p>The likelihood function can be used to assess the precision of θ̂. We need
a few more definitions. The score function is defined by
</p>
<p>ℓ̇(θ;Z) =
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>ℓ̇(θ; zi), (8.13)</p>
<p/>
</div>
<div class="page"><p/>
<p>266 8. Model Inference and Averaging
</p>
<p>where ℓ̇(θ; zi) = &part;ℓ(θ; zi)/&part;θ. Assuming that the likelihood takes its maxi-
</p>
<p>mum in the interior of the parameter space, ℓ̇(θ̂;Z) = 0. The information
matrix is
</p>
<p>I(θ) = &minus;
N&sum;
</p>
<p>i=1
</p>
<p>&part;2ℓ(θ; zi)
</p>
<p>&part;θ&part;θT
. (8.14)
</p>
<p>When I(θ) is evaluated at θ = θ̂, it is often called the observed information.
The Fisher information (or expected information) is
</p>
<p>i(θ) = Eθ[I(θ)]. (8.15)
</p>
<p>Finally, let θ0 denote the true value of θ.
A standard result says that the sampling distribution of the maximum
</p>
<p>likelihood estimator has a limiting normal distribution
</p>
<p>θ̂ &rarr; N(θ0, i(θ0)&minus;1), (8.16)
</p>
<p>as N &rarr; &infin;. Here we are independently sampling from gθ0(z). This suggests
that the sampling distribution of θ̂ may be approximated by
</p>
<p>N(θ̂, i(θ̂)&minus;1) or N(θ̂, I(θ̂)&minus;1), (8.17)
</p>
<p>where θ̂ represents the maximum likelihood estimate from the observed
data.
The corresponding estimates for the standard errors of θ̂j are obtained
</p>
<p>from
&radic;
</p>
<p>i(θ̂)&minus;1jj and
&radic;
</p>
<p>I(θ̂)&minus;1jj . (8.18)
</p>
<p>Confidence points for θj can be constructed from either approximation
in (8.17). Such a confidence point has the form
</p>
<p>θ̂j &minus; z(1&minus;α) &middot;
&radic;
i(θ̂)&minus;1jj or θ̂j &minus; z(1&minus;α) &middot;
</p>
<p>&radic;
I(θ̂)&minus;1jj ,
</p>
<p>respectively, where z(1&minus;α) is the 1 &minus; α percentile of the standard normal
distribution. More accurate confidence intervals can be derived from the
likelihood function, by using the chi-squared approximation
</p>
<p>2[ℓ(θ̂)&minus; ℓ(θ0)] &sim; χ2p, (8.19)
</p>
<p>where p is the number of components in θ. The resulting 1 &minus; 2α confi-
dence interval is the set of all θ0 such that 2[ℓ(θ̂) &minus; ℓ(θ0)] &le; χ2p
</p>
<p>(1&minus;2α)
,
</p>
<p>where χ2p
(1&minus;2α)
</p>
<p>is the 1&minus;2α percentile of the chi-squared distribution with
p degrees of freedom.</p>
<p/>
</div>
<div class="page"><p/>
<p>8.3 Bayesian Methods 267
</p>
<p>Let&rsquo;s return to our smoothing example to see what maximum likelihood
yields. The parameters are θ = (β, σ2). The log-likelihood is
</p>
<p>ℓ(θ) = &minus;N
2
log σ22π &minus; 1
</p>
<p>2σ2
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>(yi &minus; h(xi)Tβ)2. (8.20)
</p>
<p>The maximum likelihood estimate is obtained by setting &part;ℓ/&part;β = 0 and
&part;ℓ/&part;σ2 = 0, giving
</p>
<p>β̂ = (HTH)&minus;1HTy,
</p>
<p>σ̂2 =
1
</p>
<p>N
</p>
<p>&sum;
(yi &minus; μ̂(xi))2,
</p>
<p>(8.21)
</p>
<p>which are the same as the usual estimates given in (8.2) and below (8.3).
The information matrix for θ = (β, σ2) is block-diagonal, and the block
</p>
<p>corresponding to β is
</p>
<p>I(β) = (HTH)/σ2, (8.22)
</p>
<p>so that the estimated variance (HTH)&minus;1σ̂2 agrees with the least squares
estimate (8.3).
</p>
<p>8.2.3 Bootstrap versus Maximum Likelihood
</p>
<p>In essence the bootstrap is a computer implementation of nonparametric or
parametric maximum likelihood. The advantage of the bootstrap over the
maximum likelihood formula is that it allows us to compute maximum like-
lihood estimates of standard errors and other quantities in settings where
no formulas are available.
In our example, suppose that we adaptively choose by cross-validation
</p>
<p>the number and position of the knots that define the B-splines, rather
than fix them in advance. Denote by λ the collection of knots and their
positions. Then the standard errors and confidence bands should account
for the adaptive choice of λ, but there is no way to do this analytically.
With the bootstrap, we compute the B-spline smooth with an adaptive
choice of knots for each bootstrap sample. The percentiles of the resulting
curves capture the variability from both the noise in the targets as well as
that from λ̂. In this particular example the confidence bands (not shown)
don&rsquo;t look much different than the fixed λ bands. But in other problems,
where more adaptation is used, this can be an important effect to capture.
</p>
<p>8.3 Bayesian Methods
</p>
<p>In the Bayesian approach to inference, we specify a sampling model Pr(Z|θ)
(density or probability mass function) for our data given the parameters,</p>
<p/>
</div>
<div class="page"><p/>
<p>268 8. Model Inference and Averaging
</p>
<p>and a prior distribution for the parameters Pr(θ) reflecting our knowledge
about θ before we see the data. We then compute the posterior distribution
</p>
<p>Pr(θ|Z) = Pr(Z|θ) &middot; Pr(θ)&int;
Pr(Z|θ) &middot; Pr(θ)dθ , (8.23)
</p>
<p>which represents our updated knowledge about θ after we see the data. To
understand this posterior distribution, one might draw samples from it or
summarize by computing its mean or mode. The Bayesian approach differs
from the standard (&ldquo;frequentist&rdquo;) method for inference in its use of a prior
distribution to express the uncertainty present before seeing the data, and
to allow the uncertainty remaining after seeing the data to be expressed in
the form of a posterior distribution.
The posterior distribution also provides the basis for predicting the values
</p>
<p>of a future observation znew, via the predictive distribution:
</p>
<p>Pr(znew|Z) =
&int;
</p>
<p>Pr(znew|θ) &middot; Pr(θ|Z)dθ. (8.24)
</p>
<p>In contrast, the maximum likelihood approach would use Pr(znew|θ̂),
the data density evaluated at the maximum likelihood estimate, to predict
future data. Unlike the predictive distribution (8.24), this does not account
for the uncertainty in estimating θ.
</p>
<p>Let&rsquo;s walk through the Bayesian approach in our smoothing example.
We start with the parametric model given by equation (8.5), and assume
for the moment that σ2 is known. We assume that the observed feature
values x1, x2, . . . , xN are fixed, so that the randomness in the data comes
solely from y varying around its mean μ(x).
</p>
<p>The second ingredient we need is a prior distribution. Distributions on
functions are fairly complex entities: one approach is to use a Gaussian
process prior in which we specify the prior covariance between any two
function values μ(x) and μ(x&prime;) (Wahba, 1990; Neal, 1996).
Here we take a simpler route: by considering a finite B-spline basis for
</p>
<p>μ(x), we can instead provide a prior for the coefficients β, and this implicitly
defines a prior for μ(x). We choose a Gaussian prior centered at zero
</p>
<p>β &sim; N(0, τΣ) (8.25)
</p>
<p>with the choices of the prior correlation matrix Σ and variance τ to be
discussed below. The implicit process prior for μ(x) is hence Gaussian,
with covariance kernel
</p>
<p>K(x, x&prime;) = cov[μ(x), μ(x&prime;)]
</p>
<p>= τ &middot; h(x)TΣh(x&prime;). (8.26)</p>
<p/>
</div>
<div class="page"><p/>
<p>8.3 Bayesian Methods 269
</p>
<p>0.0 0.5 1.0 1.5 2.0 2.5 3.0
</p>
<p>-3
-2
</p>
<p>-1
0
</p>
<p>1
2
</p>
<p>3
</p>
<p>μ
(x
)
</p>
<p>x
</p>
<p>FIGURE 8.3. Smoothing example: Ten draws from the Gaussian prior distri-
bution for the function μ(x).
</p>
<p>The posterior distribution for β is also Gaussian, with mean and covariance
</p>
<p>E(β|Z) =
(
HTH+
</p>
<p>σ2
</p>
<p>τ
Σ&minus;1
</p>
<p>)&minus;1
HTy,
</p>
<p>cov(β|Z) =
(
HTH+
</p>
<p>σ2
</p>
<p>τ
Σ&minus;1
</p>
<p>)&minus;1
σ2,
</p>
<p>(8.27)
</p>
<p>with the corresponding posterior values for μ(x),
</p>
<p>E(μ(x)|Z) = h(x)T
(
HTH+
</p>
<p>σ2
</p>
<p>τ
Σ&minus;1
</p>
<p>)&minus;1
HTy,
</p>
<p>cov[μ(x), μ(x&prime;)|Z] = h(x)T
(
HTH+
</p>
<p>σ2
</p>
<p>τ
Σ&minus;1
</p>
<p>)&minus;1
h(x&prime;)σ2.
</p>
<p>(8.28)
</p>
<p>How do we choose the prior correlation matrix Σ? In some settings the
prior can be chosen from subject matter knowledge about the parameters.
Here we are willing to say the function μ(x) should be smooth, and have
guaranteed this by expressing μ in a smooth low-dimensional basis of B-
splines. Hence we can take the prior correlation matrix to be the identity
Σ = I. When the number of basis functions is large, this might not be suf-
ficient, and additional smoothness can be enforced by imposing restrictions
on Σ; this is exactly the case with smoothing splines (Section 5.8.1).
Figure 8.3 shows ten draws from the corresponding prior for μ(x). To
</p>
<p>generate posterior values of the function μ(x), we generate values β&prime; from its
</p>
<p>posterior (8.27), giving corresponding posterior value μ&prime;(x) =
&sum;7
</p>
<p>1 β
&prime;
jhj(x).
</p>
<p>Ten such posterior curves are shown in Figure 8.4. Two different values
were used for the prior variance τ , 1 and 1000. Notice how similar the
right panel looks to the bootstrap distribution in the bottom left panel</p>
<p/>
</div>
<div class="page"><p/>
<p>270 8. Model Inference and Averaging
</p>
<p>0.0 0.5 1.0 1.5 2.0 2.5 3.0
</p>
<p>-1
0
</p>
<p>1
2
</p>
<p>3
4
</p>
<p>5
</p>
<p>&bull;
&bull;&bull;&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;&bull;
&bull;&bull;&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
&bull;&bull;
</p>
<p>&bull;
&bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;&bull;
&bull;
&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>0.0 0.5 1.0 1.5 2.0 2.5 3.0
</p>
<p>-1
0
</p>
<p>1
2
</p>
<p>3
4
</p>
<p>5
</p>
<p>&bull;
&bull;&bull;&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;&bull;
&bull;&bull;&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
&bull;&bull;
</p>
<p>&bull;
&bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;&bull;
&bull;
&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>μ
(x
)
</p>
<p>μ
(x
)
</p>
<p>xx
</p>
<p>τ = 1 τ = 1000
</p>
<p>FIGURE 8.4. Smoothing example: Ten draws from the posterior distribution
for the function μ(x), for two different values of the prior variance τ . The purple
curves are the posterior means.
</p>
<p>of Figure 8.2 on page 263. This similarity is no accident. As τ &rarr; &infin;, the
posterior distribution (8.27) and the bootstrap distribution (8.7) coincide.
On the other hand, for τ = 1, the posterior curves μ(x) in the left panel
of Figure 8.4 are smoother than the bootstrap curves, because we have
imposed more prior weight on smoothness.
The distribution (8.25) with τ &rarr; &infin; is called a noninformative prior for
</p>
<p>θ. In Gaussian models, maximum likelihood and parametric bootstrap anal-
yses tend to agree with Bayesian analyses that use a noninformative prior
for the free parameters. These tend to agree, because with a constant prior,
the posterior distribution is proportional to the likelihood. This correspon-
dence also extends to the nonparametric case, where the nonparametric
bootstrap approximates a noninformative Bayes analysis; Section 8.4 has
the details.
We have, however, done some things that are not proper from a Bayesian
</p>
<p>point of view. We have used a noninformative (constant) prior for σ2 and
replaced it with the maximum likelihood estimate σ̂2 in the posterior. A
more standard Bayesian analysis would also put a prior on σ (typically
g(σ) &prop; 1/σ), calculate a joint posterior for μ(x) and σ, and then integrate
out σ, rather than just extract the maximum of the posterior distribution
(&ldquo;MAP&rdquo; estimate).</p>
<p/>
</div>
<div class="page"><p/>
<p>8.4 Relationship Between the Bootstrap and Bayesian Inference 271
</p>
<p>8.4 Relationship Between the Bootstrap and
Bayesian Inference
</p>
<p>Consider first a very simple example, in which we observe a single obser-
vation z from a normal distribution
</p>
<p>z &sim; N(θ, 1). (8.29)
</p>
<p>To carry out a Bayesian analysis for θ, we need to specify a prior. The
most convenient and common choice would be θ &sim; N(0, τ) giving posterior
distribution
</p>
<p>θ|z &sim; N
(
</p>
<p>z
</p>
<p>1 + 1/τ
,
</p>
<p>1
</p>
<p>1 + 1/τ
</p>
<p>)
. (8.30)
</p>
<p>Now the larger we take τ , the more concentrated the posterior becomes
around the maximum likelihood estimate θ̂ = z. In the limit as τ &rarr; &infin; we
obtain a noninformative (constant) prior, and the posterior distribution is
</p>
<p>θ|z &sim; N(z, 1). (8.31)
</p>
<p>This is the same as a parametric bootstrap distribution in which we gen-
erate bootstrap values z&lowast; from the maximum likelihood estimate of the
sampling density N(z, 1).
</p>
<p>There are three ingredients that make this correspondence work:
</p>
<p>1. The choice of noninformative prior for θ.
</p>
<p>2. The dependence of the log-likelihood ℓ(θ;Z) on the data Z only
</p>
<p>through the maximum likelihood estimate θ̂. Hence we can write the
log-likelihood as ℓ(θ; θ̂).
</p>
<p>3. The symmetry of the log-likelihood in θ and θ̂, that is, ℓ(θ; θ̂) =
</p>
<p>ℓ(θ̂; θ) + constant.
</p>
<p>Properties (2) and (3) essentially only hold for the Gaussian distribu-
tion. However, they also hold approximately for the multinomial distribu-
tion, leading to a correspondence between the nonparametric bootstrap
and Bayes inference, which we outline next.
Assume that we have a discrete sample space with L categories. Let wj be
</p>
<p>the probability that a sample point falls in category j, and ŵj the observed
proportion in category j. Let w = (w1, w2, . . . , wL), ŵ = (ŵ1, ŵ2, . . . , ŵL).
Denote our estimator by S(ŵ); take as a prior distribution for w a sym-
metric Dirichlet distribution with parameter a:
</p>
<p>w &sim; DiL(a1), (8.32)</p>
<p/>
</div>
<div class="page"><p/>
<p>272 8. Model Inference and Averaging
</p>
<p>that is, the prior probability mass function is proportional to
&prod;L
</p>
<p>ℓ=1 w
a&minus;1
ℓ .
</p>
<p>Then the posterior density of w is
</p>
<p>w &sim; DiL(a1 +Nŵ), (8.33)
</p>
<p>where N is the sample size. Letting a &rarr; 0 to obtain a noninformative prior
gives
</p>
<p>w &sim; DiL(Nŵ). (8.34)
</p>
<p>Now the bootstrap distribution, obtained by sampling with replacement
from the data, can be expressed as sampling the category proportions from
a multinomial distribution. Specifically,
</p>
<p>Nŵ&lowast; &sim; Mult(N, ŵ), (8.35)
</p>
<p>where Mult(N, ŵ) denotes a multinomial distribution, having probability
</p>
<p>mass function
(
</p>
<p>N
Nŵ&lowast;1 ,...,Nŵ
</p>
<p>&lowast;
L
</p>
<p>)&prod;
ŵ
</p>
<p>Nŵ&lowast;ℓ
ℓ . This distribution is similar to the pos-
</p>
<p>terior distribution above, having the same support, same mean, and nearly
the same covariance matrix. Hence the bootstrap distribution of S(ŵ&lowast;) will
closely approximate the posterior distribution of S(w).
</p>
<p>In this sense, the bootstrap distribution represents an (approximate)
nonparametric, noninformative posterior distribution for our parameter.
But this bootstrap distribution is obtained painlessly&mdash;without having to
formally specify a prior and without having to sample from the posterior
distribution. Hence we might think of the bootstrap distribution as a &ldquo;poor
man&rsquo;s&rdquo; Bayes posterior. By perturbing the data, the bootstrap approxi-
mates the Bayesian effect of perturbing the parameters, and is typically
much simpler to carry out.
</p>
<p>8.5 The EM Algorithm
</p>
<p>The EM algorithm is a popular tool for simplifying difficult maximum
likelihood problems. We first describe it in the context of a simple mixture
model.
</p>
<p>8.5.1 Two-Component Mixture Model
</p>
<p>In this section we describe a simple mixture model for density estimation,
and the associated EM algorithm for carrying out maximum likelihood
estimation. This has a natural connection to Gibbs sampling methods for
Bayesian inference. Mixture models are discussed and demonstrated in sev-
eral other parts of the book, in particular Sections 6.8, 12.7 and 13.2.3.
The left panel of Figure 8.5 shows a histogram of the 20 fictitious data
</p>
<p>points in Table 8.1.</p>
<p/>
</div>
<div class="page"><p/>
<p>8.5 The EM Algorithm 273
</p>
<p>0 2 4 6
</p>
<p>0
.0
</p>
<p>0
.2
</p>
<p>0
.4
</p>
<p>0
.6
</p>
<p>0
.8
</p>
<p>1
.0
</p>
<p>y y
</p>
<p>d
e
n
s
it
y
</p>
<p>0 2 4 6
</p>
<p>0
.0
</p>
<p>0
.2
</p>
<p>0
.4
</p>
<p>0
.6
</p>
<p>0
.8
</p>
<p>1
.0 &bull; &bull;&bull; &bull; &bull;&bull; &bull;&bull;&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull; &bull;&bull; &bull; &bull; &bull;&bull; &bull;
</p>
<p>FIGURE 8.5. Mixture example. (Left panel:) Histogram of data. (Right panel:)
Maximum likelihood fit of Gaussian densities (solid red) and responsibility (dotted
green) of the left component density for observation y, as a function of y.
</p>
<p>TABLE 8.1. Twenty fictitious data points used in the two-component mixture
example in Figure 8.5.
</p>
<p>-0.39 0.12 0.94 1.67 1.76 2.44 3.72 4.28 4.92 5.53
0.06 0.48 1.01 1.68 1.80 3.25 4.12 4.60 5.28 6.22
</p>
<p>We would like to model the density of the data points, and due to the
apparent bi-modality, a Gaussian distribution would not be appropriate.
There seems to be two separate underlying regimes, so instead we model
Y as a mixture of two normal distributions:
</p>
<p>Y1 &sim; N(μ1, σ21),
Y2 &sim; N(μ2, σ22), (8.36)
Y = (1&minus;∆) &middot; Y1 +∆ &middot; Y2,
</p>
<p>where ∆ &isin; {0, 1} with Pr(∆ = 1) = π. This generative representation is
explicit: generate a ∆ &isin; {0, 1} with probability π, and then depending on
the outcome, deliver either Y1 or Y2. Let φθ(x) denote the normal density
with parameters θ = (μ, σ2). Then the density of Y is
</p>
<p>gY (y) = (1&minus; π)φθ1(y) + πφθ2(y). (8.37)
Now suppose we wish to fit this model to the data in Figure 8.5 by maxi-
mum likelihood. The parameters are
</p>
<p>θ = (π, θ1, θ2) = (π, μ1, σ
2
1 , μ2, σ
</p>
<p>2
2). (8.38)
</p>
<p>The log-likelihood based on the N training cases is
</p>
<p>ℓ(θ;Z) =
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>log[(1&minus; π)φθ1(yi) + πφθ2(yi)]. (8.39)</p>
<p/>
</div>
<div class="page"><p/>
<p>274 8. Model Inference and Averaging
</p>
<p>Direct maximization of ℓ(θ;Z) is quite difficult numerically, because of
the sum of terms inside the logarithm. There is, however, a simpler ap-
proach. We consider unobserved latent variables ∆i taking values 0 or 1 as
in (8.36): if ∆i = 1 then Yi comes from model 2, otherwise it comes from
model 1. Suppose we knew the values of the ∆i&rsquo;s. Then the log-likelihood
would be
</p>
<p>ℓ0(θ;Z,∆) =
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>[(1&minus;∆i) log φθ1(yi) + ∆i log φθ2(yi)]
</p>
<p>+
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>[(1&minus;∆i) log(1&minus; π) + ∆i log π] , (8.40)
</p>
<p>and the maximum likelihood estimates of μ1 and σ
2
1 would be the sample
</p>
<p>mean and variance for those data with ∆i = 0, and similarly those for μ2
and σ22 would be the sample mean and variance of the data with ∆i = 1.
The estimate of π would be the proportion of ∆i = 1.
Since the values of the ∆i&rsquo;s are actually unknown, we proceed in an
</p>
<p>iterative fashion, substituting for each ∆i in (8.40) its expected value
</p>
<p>γi(θ) = E(∆i|θ,Z) = Pr(∆i = 1|θ,Z), (8.41)
</p>
<p>also called the responsibility of model 2 for observation i. We use a proce-
dure called the EM algorithm, given in Algorithm 8.1 for the special case of
Gaussian mixtures. In the expectation step, we do a soft assignment of each
observation to each model: the current estimates of the parameters are used
to assign responsibilities according to the relative density of the training
points under each model. In the maximization step, these responsibilities
are used in weighted maximum-likelihood fits to update the estimates of
the parameters.
A good way to construct initial guesses for μ̂1 and μ̂2 is simply to choose
</p>
<p>two of the yi at random. Both σ̂
2
1 and σ̂
</p>
<p>2
2 can be set equal to the overall
</p>
<p>sample variance
&sum;N
</p>
<p>i=1(yi&minus; ȳ)2/N . The mixing proportion π̂ can be started
at the value 0.5.
Note that the actual maximizer of the likelihood occurs when we put a
</p>
<p>spike of infinite height at any one data point, that is, μ̂1 = yi for some
i and σ̂21 = 0. This gives infinite likelihood, but is not a useful solution.
Hence we are actually looking for a good local maximum of the likelihood,
one for which σ̂21 , σ̂
</p>
<p>2
2 &gt; 0. To further complicate matters, there can be
</p>
<p>more than one local maximum having σ̂21 , σ̂
2
2 &gt; 0. In our example, we
</p>
<p>ran the EM algorithm with a number of different initial guesses for the
parameters, all having σ̂2k &gt; 0.5, and chose the run that gave us the highest
maximized likelihood. Figure 8.6 shows the progress of the EM algorithm in
maximizing the log-likelihood. Table 8.2 shows π̂ =
</p>
<p>&sum;
i γ̂i/N , the maximum
</p>
<p>likelihood estimate of the proportion of observations in class 2, at selected
iterations of the EM procedure.</p>
<p/>
</div>
<div class="page"><p/>
<p>8.5 The EM Algorithm 275
</p>
<p>Algorithm 8.1 EM Algorithm for Two-component Gaussian Mixture.
</p>
<p>1. Take initial guesses for the parameters μ̂1, σ̂
2
1 , μ̂2, σ̂
</p>
<p>2
2 , π̂ (see text).
</p>
<p>2. Expectation Step: compute the responsibilities
</p>
<p>γ̂i =
π̂φθ̂2(yi)
</p>
<p>(1&minus; π̂)φθ̂1(yi) + π̂φθ̂2(yi)
, i = 1, 2, . . . , N. (8.42)
</p>
<p>3. Maximization Step: compute the weighted means and variances:
</p>
<p>μ̂1 =
</p>
<p>&sum;N
i=1(1&minus; γ̂i)yi&sum;N
i=1(1&minus; γ̂i)
</p>
<p>, σ̂21 =
</p>
<p>&sum;N
i=1(1&minus; γ̂i)(yi &minus; μ̂1)2&sum;N
</p>
<p>i=1(1&minus; γ̂i)
,
</p>
<p>μ̂2 =
</p>
<p>&sum;N
i=1 γ̂iyi&sum;N
i=1 γ̂i
</p>
<p>, σ̂22 =
</p>
<p>&sum;N
i=1 γ̂i(yi &minus; μ̂2)2&sum;N
</p>
<p>i=1 γ̂i
,
</p>
<p>and the mixing probability π̂ =
&sum;N
</p>
<p>i=1 γ̂i/N .
</p>
<p>4. Iterate steps 2 and 3 until convergence.
</p>
<p>TABLE 8.2. Selected iterations of the EM algorithm for mixture example.
</p>
<p>Iteration π̂
</p>
<p>1 0.485
5 0.493
10 0.523
15 0.544
20 0.546
</p>
<p>The final maximum likelihood estimates are
</p>
<p>μ̂1 = 4.62, σ̂
2
1 = 0.87,
</p>
<p>μ̂2 = 1.06, σ̂
2
2 = 0.77,
</p>
<p>π̂ = 0.546.
</p>
<p>The right panel of Figure 8.5 shows the estimated Gaussian mixture density
from this procedure (solid red curve), along with the responsibilities (dotted
green curve). Note that mixtures are also useful for supervised learning; in
Section 6.7 we show how the Gaussian mixture model leads to a version of
radial basis functions.</p>
<p/>
</div>
<div class="page"><p/>
<p>276 8. Model Inference and Averaging
</p>
<p>Iteration
</p>
<p>O
b
s
e
rv
</p>
<p>e
d
 D
</p>
<p>a
ta
</p>
<p> L
o
g
-l
ik
</p>
<p>e
lih
</p>
<p>o
o
d
</p>
<p>5 10 15 20
</p>
<p>-4
4
</p>
<p>-4
3
</p>
<p>-4
2
</p>
<p>-4
1
</p>
<p>-4
0
</p>
<p>-3
9
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o o o o o o o o o o
</p>
<p>FIGURE 8.6. EM algorithm: observed data log-likelihood as a function of the
iteration number.
</p>
<p>8.5.2 The EM Algorithm in General
</p>
<p>The above procedure is an example of the EM (or Baum&ndash;Welch) algorithm
for maximizing likelihoods in certain classes of problems. These problems
are ones for which maximization of the likelihood is difficult, but made
easier by enlarging the sample with latent (unobserved) data. This is called
data augmentation. Here the latent data are the model memberships ∆i.
In other problems, the latent data are actual data that should have been
observed but are missing.
Algorithm 8.2 gives the general formulation of the EM algorithm. Our
</p>
<p>observed data is Z, having log-likelihood ℓ(θ;Z) depending on parameters
θ. The latent or missing data is Zm, so that the complete data is T =
(Z,Zm) with log-likelihood ℓ0(θ;T), ℓ0 based on the complete density. In
the mixture problem (Z,Zm) = (y,∆), and ℓ0(θ;T) is given in (8.40).
</p>
<p>In our mixture example, E(ℓ0(θ
&prime;;T)|Z, θ̂(j)) is simply (8.40) with the ∆i
</p>
<p>replaced by the responsibilities γ̂i(θ̂), and the maximizers in step 3 are just
weighted means and variances.
We now give an explanation of why the EM algorithm works in general.
Since
</p>
<p>Pr(Zm|Z, θ&prime;) = Pr(Z
m,Z|θ&prime;)
</p>
<p>Pr(Z|θ&prime;) , (8.44)
</p>
<p>we can write
</p>
<p>Pr(Z|θ&prime;) = Pr(T|θ
&prime;)
</p>
<p>Pr(Zm|Z, θ&prime;) . (8.45)
</p>
<p>In terms of log-likelihoods, we have ℓ(θ&prime;;Z) = ℓ0(θ
&prime;;T)&minus;ℓ1(θ&prime;;Zm|Z), where
</p>
<p>ℓ1 is based on the conditional density Pr(Z
m|Z, θ&prime;). Taking conditional
</p>
<p>expectations with respect to the distribution of T|Z governed by parameter
θ gives
</p>
<p>ℓ(θ&prime;;Z) = E[ℓ0(θ
&prime;;T)|Z, θ]&minus; E[ℓ1(θ&prime;;Zm|Z)|Z, θ]</p>
<p/>
</div>
<div class="page"><p/>
<p>8.5 The EM Algorithm 277
</p>
<p>Algorithm 8.2 The EM Algorithm.
</p>
<p>1. Start with initial guesses for the parameters θ̂(0).
</p>
<p>2. Expectation Step: at the jth step, compute
</p>
<p>Q(θ&prime;, θ̂(j)) = E(ℓ0(θ
&prime;;T)|Z, θ̂(j)) (8.43)
</p>
<p>as a function of the dummy argument θ&prime;.
</p>
<p>3. Maximization Step: determine the new estimate θ̂(j+1) as the maxi-
mizer of Q(θ&prime;, θ̂(j)) over θ&prime;.
</p>
<p>4. Iterate steps 2 and 3 until convergence.
</p>
<p>&equiv; Q(θ&prime;, θ)&minus;R(θ&prime;, θ). (8.46)
</p>
<p>In the M step, the EM algorithm maximizes Q(θ&prime;, θ) over θ&prime;, rather than
the actual objective function ℓ(θ&prime;;Z). Why does it succeed in maximizing
ℓ(θ&prime;;Z)? Note that R(θ&lowast;, θ) is the expectation of a log-likelihood of a density
(indexed by θ&lowast;), with respect to the same density indexed by θ, and hence
(by Jensen&rsquo;s inequality) is maximized as a function of θ&lowast;, when θ&lowast; = θ (see
Exercise 8.1). So if θ&prime; maximizes Q(θ&prime;, θ), we see that
</p>
<p>ℓ(θ&prime;;Z)&minus; ℓ(θ;Z) = [Q(θ&prime;, θ)&minus;Q(θ, θ)]&minus; [R(θ&prime;, θ)&minus;R(θ, θ)]
&ge; 0. (8.47)
</p>
<p>Hence the EM iteration never decreases the log-likelihood.
This argument also makes it clear that a full maximization in the M
</p>
<p>step is not necessary: we need only to find a value θ̂(j+1) so that Q(θ&prime;, θ̂(j))
</p>
<p>increases as a function of the first argument, that is, Q(θ̂(j+1), θ̂(j)) &gt;
</p>
<p>Q(θ̂(j), θ̂(j)). Such procedures are called GEM (generalized EM) algorithms.
The EM algorithm can also be viewed as a minorization procedure: see
Exercise 8.7.
</p>
<p>8.5.3 EM as a Maximization&ndash;Maximization Procedure
</p>
<p>Here is a different view of the EM procedure, as a joint maximization
algorithm. Consider the function
</p>
<p>F (θ&prime;, P̃ ) = EP̃ [ℓ0(θ
&prime;;T)]&minus; EP̃ [log P̃ (Zm)]. (8.48)
</p>
<p>Here P̃ (Zm) is any distribution over the latent data Zm. In the mixture
example, P̃ (Zm) comprises the set of probabilities γi = Pr(∆i = 1|θ,Z).
Note that F evaluated at P̃ (Zm) = Pr(Zm|Z, θ&prime;), is the log-likelihood of</p>
<p/>
</div>
<div class="page"><p/>
<p>278 8. Model Inference and Averaging
</p>
<p>1 2 3 4 5
</p>
<p>0
1
</p>
<p>2
3
</p>
<p>4
</p>
<p>0.10.3
</p>
<p>0.50.7
0.9
</p>
<p>M
o
d
e
l 
P
</p>
<p>a
ra
</p>
<p>m
e
te
</p>
<p>rs
</p>
<p>Latent Data Parameters
</p>
<p>E
</p>
<p>M
</p>
<p>E M
</p>
<p>FIGURE 8.7. Maximization&ndash;maximization view of the EM algorithm. Shown
are the contours of the (augmented) observed data log-likelihood F (θ&prime;, P̃ ). The
E step is equivalent to maximizing the log-likelihood over the parameters of the
latent data distribution. The M step maximizes it over the parameters of the
log-likelihood. The red curve corresponds to the observed data log-likelihood, a
profile obtained by maximizing F (θ&prime;, P̃ ) for each value of θ&prime;.
</p>
<p>the observed data, from (8.46)1. The function F expands the domain of
the log-likelihood, to facilitate its maximization.
The EM algorithm can be viewed as a joint maximization method for F
</p>
<p>over θ&prime; and P̃ (Zm), by fixing one argument and maximizing over the other.
The maximizer over P̃ (Zm) for fixed θ&prime; can be shown to be
</p>
<p>P̃ (Zm) = Pr(Zm|Z, θ&prime;) (8.49)
</p>
<p>(Exercise 8.2). This is the distribution computed by the E step, for example,
(8.42) in the mixture example. In the M step, we maximize F (θ&prime;, P̃ ) over θ&prime;
</p>
<p>with P̃ fixed: this is the same as maximizing the first term EP̃ [ℓ0(θ
&prime;;T)|Z, θ]
</p>
<p>since the second term does not involve θ&prime;.
Finally, since F (θ&prime;, P̃ ) and the observed data log-likelihood agree when
</p>
<p>P̃ (Zm) = Pr(Zm|Z, θ&prime;), maximization of the former accomplishes maxi-
mization of the latter. Figure 8.7 shows a schematic view of this process.
This view of the EM algorithm leads to alternative maximization proce-
</p>
<p>1 (8.46) holds for all θ, including θ = θ&prime;.</p>
<p/>
</div>
<div class="page"><p/>
<p>8.6 MCMC for Sampling from the Posterior 279
</p>
<p>Algorithm 8.3 Gibbs Sampler.
</p>
<p>1. Take some initial values U
(0)
k , k = 1, 2, . . . ,K.
</p>
<p>2. Repeat for t = 1, 2, . . . , . :
</p>
<p>For k = 1, 2, . . . ,K generate U
(t)
k from
</p>
<p>Pr(U
(t)
k |U
</p>
<p>(t)
1 , . . . , U
</p>
<p>(t)
k&minus;1, U
</p>
<p>(t&minus;1)
k+1 , . . . , U
</p>
<p>(t&minus;1)
K ).
</p>
<p>3. Continue step 2 until the joint distribution of (U
(t)
1 , U
</p>
<p>(t)
2 , . . . , U
</p>
<p>(t)
K )
</p>
<p>does not change.
</p>
<p>dures. For example, one does not need to maximize with respect to all of
the latent data parameters at once, but could instead maximize over one
of them at a time, alternating with the M step.
</p>
<p>8.6 MCMC for Sampling from the Posterior
</p>
<p>Having defined a Bayesian model, one would like to draw samples from
the resulting posterior distribution, in order to make inferences about the
parameters. Except for simple models, this is often a difficult computa-
tional problem. In this section we discuss the Markov chain Monte Carlo
(MCMC) approach to posterior sampling. We will see that Gibbs sampling,
an MCMC procedure, is closely related to the EM algorithm: the main dif-
ference is that it samples from the conditional distributions rather than
maximizing over them.
Consider first the following abstract problem. We have random variables
</p>
<p>U1, U2, . . . , UK and we wish to draw a sample from their joint distribution.
Suppose this is difficult to do, but it is easy to simulate from the conditional
distributions Pr(Uj |U1, U2, . . . , Uj&minus;1, Uj+1, . . . , UK), j = 1, 2, . . . ,K. The
Gibbs sampling procedure alternatively simulates from each of these distri-
butions and when the process stabilizes, provides a sample from the desired
joint distribution. The procedure is defined in Algorithm 8.3.
Under regularity conditions it can be shown that this procedure even-
</p>
<p>tually stabilizes, and the resulting random variables are indeed a sample
from the joint distribution of U1, U2, . . . , UK . This occurs despite the fact
</p>
<p>that the samples (U
(t)
1 , U
</p>
<p>(t)
2 , . . . , U
</p>
<p>(t)
K ) are clearly not independent for dif-
</p>
<p>ferent t. More formally, Gibbs sampling produces a Markov chain whose
stationary distribution is the true joint distribution, and hence the term
&ldquo;Markov chain Monte Carlo.&rdquo; It is not surprising that the true joint dis-
tribution is stationary under this process, as the successive steps leave the
marginal distributions of the Uk&rsquo;s unchanged.</p>
<p/>
</div>
<div class="page"><p/>
<p>280 8. Model Inference and Averaging
</p>
<p>Note that we don&rsquo;t need to know the explicit form of the conditional
densities, but just need to be able to sample from them. After the procedure
reaches stationarity, the marginal density of any subset of the variables
can be approximated by a density estimate applied to the sample values.
However if the explicit form of the conditional density Pr(Uk, |Uℓ, ℓ �= k)
is available, a better estimate of say the marginal density of Uk can be
obtained from (Exercise 8.3):
</p>
<p>P̂rUk(u) =
1
</p>
<p>(M &minus;m+ 1)
M&sum;
</p>
<p>t=m
</p>
<p>Pr(u|U (t)ℓ , ℓ �= k). (8.50)
</p>
<p>Here we have averaged over the last M &minus;m+ 1 members of the sequence,
to allow for an initial &ldquo;burn-in&rdquo; period before stationarity is reached.
Now getting back to Bayesian inference, our goal is to draw a sample from
</p>
<p>the joint posterior of the parameters given the data Z. Gibbs sampling will
be helpful if it is easy to sample from the conditional distribution of each
parameter given the other parameters and Z. An example&mdash;the Gaussian
mixture problem&mdash;is detailed next.
There is a close connection between Gibbs sampling from a posterior and
</p>
<p>the EM algorithm in exponential family models. The key is to consider the
latent data Zm from the EM procedure to be another parameter for the
Gibbs sampler. To make this explicit for the Gaussian mixture problem,
we take our parameters to be (θ,Zm). For simplicity we fix the variances
σ21 , σ
</p>
<p>2
2 and mixing proportion π at their maximum likelihood values so that
</p>
<p>the only unknown parameters in θ are the means μ1 and μ2. The Gibbs
sampler for the mixture problem is given in Algorithm 8.4. We see that
steps 2(a) and 2(b) are the same as the E and M steps of the EM pro-
cedure, except that we sample rather than maximize. In step 2(a), rather
than compute the maximum likelihood responsibilities γi = E(∆i|θ,Z),
the Gibbs sampling procedure simulates the latent data ∆i from the distri-
butions Pr(∆i|θ,Z). In step 2(b), rather than compute the maximizers of
the posterior Pr(μ1, μ2,∆|Z) we simulate from the conditional distribution
Pr(μ1, μ2|∆,Z).
Figure 8.8 shows 200 iterations of Gibbs sampling, with the mean param-
</p>
<p>eters μ1 (lower) and μ2 (upper) shown in the left panel, and the proportion
of class 2 observations
</p>
<p>&sum;
i ∆i/N on the right. Horizontal broken lines have
</p>
<p>been drawn at the maximum likelihood estimate values μ̂1, μ̂2 and
&sum;
</p>
<p>i γ̂i/N
in each case. The values seem to stabilize quite quickly, and are distributed
evenly around the maximum likelihood values.
The above mixture model was simplified, in order to make the clear
</p>
<p>connection between Gibbs sampling and the EM algorithm. More realisti-
cally, one would put a prior distribution on the variances σ21 , σ
</p>
<p>2
2 and mixing
</p>
<p>proportion π, and include separate Gibbs sampling steps in which we sam-
ple from their posterior distributions, conditional on the other parameters.
One can also incorporate proper (informative) priors for the mean param-</p>
<p/>
</div>
<div class="page"><p/>
<p>8.6 MCMC for Sampling from the Posterior 281
</p>
<p>Algorithm 8.4 Gibbs sampling for mixtures.
</p>
<p>1. Take some initial values θ(0) = (μ
(0)
1 , μ
</p>
<p>(0)
2 ).
</p>
<p>2. Repeat for t = 1, 2, . . . , .
</p>
<p>(a) For i = 1, 2, . . . , N generate ∆
(t)
i &isin; {0, 1} with Pr(∆
</p>
<p>(t)
i = 1) =
</p>
<p>γ̂i(θ
(t)), from equation (8.42).
</p>
<p>(b) Set
</p>
<p>μ̂1 =
</p>
<p>&sum;N
i=1(1&minus;∆
</p>
<p>(t)
i ) &middot; yi&sum;N
</p>
<p>i=1(1&minus;∆
(t)
i )
</p>
<p>,
</p>
<p>μ̂2 =
</p>
<p>&sum;N
i=1 ∆
</p>
<p>(t)
i &middot; yi&sum;N
</p>
<p>i=1 ∆
(t)
i
</p>
<p>,
</p>
<p>and generate μ
(t)
1 &sim; N(μ̂1, σ̂21) and μ
</p>
<p>(t)
2 &sim; N(μ̂2, σ̂22).
</p>
<p>3. Continue step 2 until the joint distribution of (∆(t), μ
(t)
1 , μ
</p>
<p>(t)
2 ) doesn&rsquo;t
</p>
<p>change
</p>
<p>Gibbs Iteration
</p>
<p>M
e
</p>
<p>a
n
</p>
<p> P
a
</p>
<p>ra
m
</p>
<p>e
te
</p>
<p>rs
</p>
<p>0 50 100 150 200
</p>
<p>0
2
</p>
<p>4
6
</p>
<p>8
</p>
<p>Gibbs Iteration
</p>
<p>M
ix
</p>
<p>in
g
</p>
<p> P
ro
</p>
<p>p
o
</p>
<p>rt
io
</p>
<p>n
</p>
<p>0 50 100 150 200
</p>
<p>0
.3
</p>
<p>0
.4
</p>
<p>0
.5
</p>
<p>0
.6
</p>
<p>0
.7
</p>
<p>FIGURE 8.8. Mixture example. (Left panel:) 200 values of the two mean param-
eters from Gibbs sampling; horizontal lines are drawn at the maximum likelihood
estimates μ̂1, μ̂2. (Right panel:) Proportion of values with ∆i = 1, for each of the
200 Gibbs sampling iterations; a horizontal line is drawn at
</p>
<p>&sum;
</p>
<p>i γ̂i/N .</p>
<p/>
</div>
<div class="page"><p/>
<p>282 8. Model Inference and Averaging
</p>
<p>eters. These priors must not be improper as this will lead to a degenerate
posterior, with all the mixing weight on one component.
Gibbs sampling is just one of a number of recently developed procedures
</p>
<p>for sampling from posterior distributions. It uses conditional sampling of
each parameter given the rest, and is useful when the structure of the prob-
lem makes this sampling easy to carry out. Other methods do not require
such structure, for example the Metropolis&ndash;Hastings algorithm. These and
other computational Bayesian methods have been applied to sophisticated
learning algorithms such as Gaussian process models and neural networks.
Details may be found in the references given in the Bibliographic Notes at
the end of this chapter.
</p>
<p>8.7 Bagging
</p>
<p>Earlier we introduced the bootstrap as a way of assessing the accuracy of a
parameter estimate or a prediction. Here we show how to use the bootstrap
to improve the estimate or prediction itself. In Section 8.4 we investigated
the relationship between the bootstrap and Bayes approaches, and found
that the bootstrap mean is approximately a posterior average. Bagging
further exploits this connection.
Consider first the regression problem. Suppose we fit a model to our
</p>
<p>training data Z = {(x1, y1), (x2, y2), . . . , (xN , yN )}, obtaining the predic-
tion f̂(x) at input x. Bootstrap aggregation or bagging averages this predic-
tion over a collection of bootstrap samples, thereby reducing its variance.
For each bootstrap sample Z&lowast;b, b = 1, 2, . . . , B, we fit our model, giving
prediction f̂&lowast;b(x). The bagging estimate is defined by
</p>
<p>f̂bag(x) =
1
</p>
<p>B
</p>
<p>B&sum;
</p>
<p>b=1
</p>
<p>f̂&lowast;b(x). (8.51)
</p>
<p>Denote by P̂ the empirical distribution putting equal probability 1/N on
each of the data points (xi, yi). In fact the &ldquo;true&rdquo; bagging estimate is
</p>
<p>defined by EP̂ f̂
&lowast;(x), where Z&lowast; = {(x&lowast;1, y&lowast;1), (x&lowast;2, y&lowast;2), . . . , (x&lowast;N , y&lowast;N )} and each
</p>
<p>(x&lowast;i , y
&lowast;
i ) &sim; P̂. Expression (8.51) is a Monte Carlo estimate of the true
</p>
<p>bagging estimate, approaching it as B &rarr; &infin;.
The bagged estimate (8.51) will differ from the original estimate f̂(x)
</p>
<p>only when the latter is a nonlinear or adaptive function of the data. For
example, to bag the B-spline smooth of Section 8.2.1, we average the curves
in the bottom left panel of Figure 8.2 at each value of x. The B-spline
smoother is linear in the data if we fix the inputs; hence if we sample using
the parametric bootstrap in equation (8.6), then f̂bag(x) &rarr; f̂(x) as B &rarr; &infin;
(Exercise 8.4). Hence bagging just reproduces the original smooth in the</p>
<p/>
</div>
<div class="page"><p/>
<p>8.7 Bagging 283
</p>
<p>top left panel of Figure 8.2. The same is approximately true if we were to
bag using the nonparametric bootstrap.
A more interesting example is a regression tree, where f̂(x) denotes the
</p>
<p>tree&rsquo;s prediction at input vector x (regression trees are described in Chap-
ter 9). Each bootstrap tree will typically involve different features than the
original, and might have a different number of terminal nodes. The bagged
estimate is the average prediction at x from these B trees.
</p>
<p>Now suppose our tree produces a classifier Ĝ(x) for a K-class response.
</p>
<p>Here it is useful to consider an underlying indicator-vector function f̂(x),
</p>
<p>with value a single one and K &minus; 1 zeroes, such that Ĝ(x) = argmaxk f̂(x).
Then the bagged estimate f̂bag(x) (8.51) is a K-vector [p1(x), p2(x), . . . ,
pK(x)], with pk(x) equal to the proportion of trees predicting class k at x.
The bagged classifier selects the class with the most &ldquo;votes&rdquo; from the B
trees, Ĝbag(x) = argmaxk f̂bag(x).
</p>
<p>Often we require the class-probability estimates at x, rather than the
classifications themselves. It is tempting to treat the voting proportions
pk(x) as estimates of these probabilities. A simple two-class example shows
that they fail in this regard. Suppose the true probability of class 1 at x is
0.75, and each of the bagged classifiers accurately predict a 1. Then p1(x) =
1, which is incorrect. For many classifiers Ĝ(x), however, there is already
</p>
<p>an underlying function f̂(x) that estimates the class probabilities at x (for
trees, the class proportions in the terminal node). An alternative bagging
strategy is to average these instead, rather than the vote indicator vectors.
Not only does this produce improved estimates of the class probabilities,
but it also tends to produce bagged classifiers with lower variance, especially
for small B (see Figure 8.10 in the next example).
</p>
<p>8.7.1 Example: Trees with Simulated Data
</p>
<p>We generated a sample of size N = 30, with two classes and p = 5 features,
each having a standard Gaussian distribution with pairwise correlation
0.95. The response Y was generated according to Pr(Y = 1|x1 &le; 0.5) = 0.2,
Pr(Y = 1|x1 &gt; 0.5) = 0.8. The Bayes error is 0.2. A test sample of size 2000
was also generated from the same population. We fit classification trees to
the training sample and to each of 200 bootstrap samples (classification
trees are described in Chapter 9). No pruning was used. Figure 8.9 shows
the original tree and eleven bootstrap trees. Notice how the trees are all
different, with different splitting features and cutpoints. The test error for
the original tree and the bagged tree is shown in Figure 8.10. In this ex-
ample the trees have high variance due to the correlation in the predictors.
Bagging succeeds in smoothing out this variance and hence reducing the
test error.
Bagging can dramatically reduce the variance of unstable procedures
</p>
<p>like trees, leading to improved prediction. A simple argument shows why</p>
<p/>
</div>
<div class="page"><p/>
<p>284 8. Model Inference and Averaging
</p>
<p>|
</p>
<p>x.1 &lt; 0.395
</p>
<p>0 1
</p>
<p>0
</p>
<p>1 0
</p>
<p>1
</p>
<p>1 0
</p>
<p>Original Tree
</p>
<p>|
</p>
<p>x.1 &lt; 0.555
</p>
<p>0
</p>
<p>1 0
</p>
<p>0
</p>
<p>1
</p>
<p>b = 1
</p>
<p>|
</p>
<p>x.2 &lt; 0.205
</p>
<p>0 1
</p>
<p>0 1
</p>
<p>0 1
</p>
<p>b = 2
</p>
<p>|
</p>
<p>x.2 &lt; 0.285
</p>
<p>1 1
</p>
<p>0
1 0
</p>
<p>b = 3
</p>
<p>|
</p>
<p>x.3 &lt; 0.985
</p>
<p>0
</p>
<p>1
</p>
<p>0 1
</p>
<p>1 1
</p>
<p>b = 4
</p>
<p>|
</p>
<p>x.4 &lt; &minus;1.36
</p>
<p>0
</p>
<p>1
1 0
</p>
<p>1
</p>
<p>0
</p>
<p>1 0
</p>
<p>b = 5
</p>
<p>|
</p>
<p>x.1 &lt; 0.395
</p>
<p>1 1 0 0
</p>
<p>1
</p>
<p>b = 6
</p>
<p>|
</p>
<p>x.1 &lt; 0.395
</p>
<p>0 1
</p>
<p>0 1
</p>
<p>1
</p>
<p>b = 7
</p>
<p>|
</p>
<p>x.3 &lt; 0.985
</p>
<p>0 1
</p>
<p>0 0
</p>
<p>1 0
</p>
<p>b = 8
</p>
<p>|
</p>
<p>x.1 &lt; 0.395
</p>
<p>0
</p>
<p>1
</p>
<p>0 1
</p>
<p>1 0
</p>
<p>b = 9
</p>
<p>|
</p>
<p>x.1 &lt; 0.555
</p>
<p>1 0
</p>
<p>1
</p>
<p>0 1
</p>
<p>b = 10
</p>
<p>|
</p>
<p>x.1 &lt; 0.555
</p>
<p>0 1
</p>
<p>0
</p>
<p>1
</p>
<p>b = 11
</p>
<p>FIGURE 8.9. Bagging trees on simulated dataset. The top left panel shows the
original tree. Eleven trees grown on bootstrap samples are shown. For each tree,
the top split is annotated.</p>
<p/>
</div>
<div class="page"><p/>
<p>8.7 Bagging 285
</p>
<p>0 50 100 150 200
</p>
<p>0
.2
</p>
<p>0
0
</p>
<p>.2
5
</p>
<p>0
.3
</p>
<p>0
0
</p>
<p>.3
5
</p>
<p>0
.4
</p>
<p>0
0
</p>
<p>.4
5
</p>
<p>0
.5
</p>
<p>0
</p>
<p>Number of Bootstrap Samples
</p>
<p>T
e
</p>
<p>s
t 
</p>
<p>E
rr
</p>
<p>o
r
</p>
<p>Bagged Trees
</p>
<p>Original Tree
</p>
<p>Bayes
</p>
<p>Consensus
Probability
</p>
<p>FIGURE 8.10. Error curves for the bagging example of Figure 8.9. Shown is
the test error of the original tree and bagged trees as a function of the number of
bootstrap samples. The orange points correspond to the consensus vote, while the
green points average the probabilities.
</p>
<p>bagging helps under squared-error loss, in short because averaging reduces
variance and leaves bias unchanged.
Assume our training observations (xi, yi), i = 1, . . . , N are indepen-
</p>
<p>dently drawn from a distribution P, and consider the ideal aggregate es-
timator fag(x) = EP f̂
</p>
<p>&lowast;(x). Here x is fixed and the bootstrap dataset Z&lowast;
</p>
<p>consists of observations x&lowast;i , y
&lowast;
i , i = 1, 2, . . . , N sampled from P. Note that
</p>
<p>fag(x) is a bagging estimate, drawing bootstrap samples from the actual
population P rather than the data. It is not an estimate that we can use
in practice, but is convenient for analysis. We can write
</p>
<p>EP [Y &minus; f̂&lowast;(x)]2 = EP [Y &minus; fag(x) + fag(x)&minus; f̂&lowast;(x)]2
</p>
<p>= EP [Y &minus; fag(x)]2 + EP [f̂&lowast;(x)&minus; fag(x)]2
&ge; EP [Y &minus; fag(x)]2. (8.52)
</p>
<p>The extra error on the right-hand side comes from the variance of f̂&lowast;(x)
around its mean fag(x). Therefore true population aggregation never in-
creases mean squared error. This suggests that bagging&mdash;drawing samples
from the training data&mdash; will often decrease mean-squared error.
The above argument does not hold for classification under 0-1 loss, be-
</p>
<p>cause of the nonadditivity of bias and variance. In that setting, bagging a</p>
<p/>
</div>
<div class="page"><p/>
<p>286 8. Model Inference and Averaging
</p>
<p>good classifier can make it better, but bagging a bad classifier can make it
worse. Here is a simple example, using a randomized rule. Suppose Y = 1
for all x, and the classifier Ĝ(x) predicts Y = 1 (for all x) with proba-
bility 0.4 and predicts Y = 0 (for all x) with probability 0.6. Then the
misclassification error of Ĝ(x) is 0.6 but that of the bagged classifier is 1.0.
For classification we can understand the bagging effect in terms of a
</p>
<p>consensus of independent weak learners (Dietterich, 2000a). Let the Bayes
optimal decision at x be G(x) = 1 in a two-class example. Suppose each
of the weak learners G&lowast;b have an error-rate eb = e &lt; 0.5, and let S1(x) =&sum;B
</p>
<p>b=1 I(G
&lowast;
b(x) = 1) be the consensus vote for class 1. Since the weak learn-
</p>
<p>ers are assumed to be independent, S1(x) &sim; Bin(B, 1 &minus; e), and Pr(S1 &gt;
B/2) &rarr; 1 as B gets large. This concept has been popularized outside of
statistics as the &ldquo;Wisdom of Crowds&rdquo; (Surowiecki, 2004) &mdash; the collective
knowledge of a diverse and independent body of people typically exceeds
the knowledge of any single individual, and can be harnessed by voting.
Of course, the main caveat here is &ldquo;independent,&rdquo; and bagged trees are
not. Figure 8.11 illustrates the power of a consensus vote in a simulated
example, where only 30% of the voters have some knowledge.
In Chapter 15 we see how random forests improve on bagging by reducing
</p>
<p>the correlation between the sampled trees.
Note that when we bag a model, any simple structure in the model is
</p>
<p>lost. As an example, a bagged tree is no longer a tree. For interpretation
of the model this is clearly a drawback. More stable procedures like near-
est neighbors are typically not affected much by bagging. Unfortunately,
the unstable models most helped by bagging are unstable because of the
emphasis on interpretability, and this is lost in the bagging process.
Figure 8.12 shows an example where bagging doesn&rsquo;t help. The 100 data
</p>
<p>points shown have two features and two classes, separated by the gray
linear boundary x1 + x2 = 1. We choose as our classifier Ĝ(x) a single
axis-oriented split, choosing the split along either x1 or x2 that produces
the largest decrease in training misclassification error.
The decision boundary obtained from bagging the 0-1 decision rule over
</p>
<p>B = 50 bootstrap samples is shown by the blue curve in the left panel.
It does a poor job of capturing the true boundary. The single split rule,
derived from the training data, splits near 0 (the middle of the range of x1
or x2), and hence has little contribution away from the center. Averaging
the probabilities rather than the classifications does not help here. Bagging
estimates the expected class probabilities from the single split rule, that is,
averaged over many replications. Note that the expected class probabilities
computed by bagging cannot be realized on any single replication, in the
same way that a woman cannot have 2.4 children. In this sense, bagging
increases somewhat the space of models of the individual base classifier.
However, it doesn&rsquo;t help in this and many other examples where a greater
enlargement of the model class is needed. &ldquo;Boosting&rdquo; is a way of doing this</p>
<p/>
</div>
<div class="page"><p/>
<p>8.7 Bagging 287
</p>
<p>0
2
</p>
<p>4
6
</p>
<p>8
1
0
</p>
<p>P &minus;  Probability of Informed Person Being Correct
</p>
<p>E
x
p
e
c
te
</p>
<p>d
 C
</p>
<p>o
rr
</p>
<p>e
c
t 
o
u
t 
o
f 
1
0
</p>
<p>Wisdom of Crowds
</p>
<p>Consensus
Individual
</p>
<p>0.25 0.50 0.75 1.00
</p>
<p>FIGURE 8.11. Simulated academy awards voting. 50 members vote in 10 cat-
egories, each with 4 nominations. For any category, only 15 voters have some
knowledge, represented by their probability of selecting the &ldquo;correct&rdquo; candidate in
that category (so P = 0.25 means they have no knowledge). For each category, the
15 experts are chosen at random from the 50. Results show the expected correct
(based on 50 simulations) for the consensus, as well as for the individuals. The
error bars indicate one standard deviation. We see, for example, that if the 15
informed for a category have a 50% chance of selecting the correct candidate, the
consensus doubles the expected performance of an individual.</p>
<p/>
</div>
<div class="page"><p/>
<p>288 8. Model Inference and Averaging
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>Bagged Decision Rule
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>Boosted Decision Rule
</p>
<p>FIGURE 8.12. Data with two features and two classes, separated by a linear
boundary. (Left panel:) Decision boundary estimated from bagging the decision
rule from a single split, axis-oriented classifier. (Right panel:) Decision boundary
from boosting the decision rule of the same classifier. The test error rates are
0.166, and 0.065, respectively. Boosting is described in Chapter 10.
</p>
<p>and is described in Chapter 10. The decision boundary in the right panel is
the result of the boosting procedure, and it roughly captures the diagonal
boundary.
</p>
<p>8.8 Model Averaging and Stacking
</p>
<p>In Section 8.4 we viewed bootstrap values of an estimator as approximate
posterior values of a corresponding parameter, from a kind of nonparamet-
ric Bayesian analysis. Viewed in this way, the bagged estimate (8.51) is
an approximate posterior Bayesian mean. In contrast, the training sample
estimate f̂(x) corresponds to the mode of the posterior. Since the posterior
mean (not mode) minimizes squared-error loss, it is not surprising that
bagging can often reduce mean squared-error.
Here we discuss Bayesian model averaging more generally. We have a
</p>
<p>set of candidate models Mm, m = 1, . . . ,M for our training set Z. These
models may be of the same type with different parameter values (e.g.,
subsets in linear regression), or different models for the same task (e.g.,
neural networks and regression trees).
Suppose ζ is some quantity of interest, for example, a prediction f(x) at
</p>
<p>some fixed feature value x. The posterior distribution of ζ is
</p>
<p>Pr(ζ|Z) =
M&sum;
</p>
<p>m=1
</p>
<p>Pr(ζ|Mm,Z)Pr(Mm|Z), (8.53)</p>
<p/>
</div>
<div class="page"><p/>
<p>8.8 Model Averaging and Stacking 289
</p>
<p>with posterior mean
</p>
<p>E(ζ|Z) =
M&sum;
</p>
<p>m=1
</p>
<p>E(ζ|Mm,Z)Pr(Mm|Z). (8.54)
</p>
<p>This Bayesian prediction is a weighted average of the individual predictions,
with weights proportional to the posterior probability of each model.
This formulation leads to a number of different model-averaging strate-
</p>
<p>gies. Committee methods take a simple unweighted average of the predic-
tions from each model, essentially giving equal probability to each model.
More ambitiously, the development in Section 7.7 shows the BIC criterion
can be used to estimate posterior model probabilities. This is applicable
in cases where the different models arise from the same parametric model,
with different parameter values. The BIC gives weight to each model de-
pending on how well it fits and how many parameters it uses. One can also
carry out the Bayesian recipe in full. If each model Mm has parameters
θm, we write
</p>
<p>Pr(Mm|Z) &prop; Pr(Mm) &middot; Pr(Z|Mm)
&prop; Pr(Mm) &middot;
</p>
<p>&int;
Pr(Z|θm,Mm)Pr(θm|Mm)dθm.
</p>
<p>(8.55)
</p>
<p>In principle one can specify priors Pr(θm|Mm) and numerically com-
pute the posterior probabilities from (8.55), to be used as model-averaging
weights. However, we have seen no real evidence that this is worth all of
the effort, relative to the much simpler BIC approximation.
How can we approach model averaging from a frequentist viewpoint?
</p>
<p>Given predictions f̂1(x), f̂2(x), . . . , f̂M (x), under squared-error loss, we can
seek the weights w = (w1, w2, . . . , wM ) such that
</p>
<p>ŵ = argmin
w
</p>
<p>EP
</p>
<p>[
Y &minus;
</p>
<p>M&sum;
</p>
<p>m=1
</p>
<p>wmf̂m(x)
]2
. (8.56)
</p>
<p>Here the input value x is fixed and the N observations in the dataset Z (and
the target Y ) are distributed according to P. The solution is the population
linear regression of Y on F̂ (x)T &equiv; [f̂1(x), f̂2(x), . . . , f̂M (x)]:
</p>
<p>ŵ = EP [F̂ (x)F̂ (x)
T ]&minus;1EP [F̂ (x)Y ]. (8.57)
</p>
<p>Now the full regression has smaller error than any single model
</p>
<p>EP
</p>
<p>[
Y &minus;
</p>
<p>M&sum;
</p>
<p>m=1
</p>
<p>ŵmf̂m(x)
</p>
<p>]2
&le; EP
</p>
<p>[
Y &minus; f̂m(x)
</p>
<p>]2
&forall;m (8.58)
</p>
<p>so combining models never makes things worse, at the population level.</p>
<p/>
</div>
<div class="page"><p/>
<p>290 8. Model Inference and Averaging
</p>
<p>Of course the population linear regression (8.57) is not available, and it
is natural to replace it with the linear regression over the training set. But
there are simple examples where this does not work well. For example, if
f̂m(x), m = 1, 2, . . . ,M represent the prediction from the best subset of
inputs of size m among M total inputs, then linear regression would put all
of the weight on the largest model, that is, ŵM = 1, ŵm = 0, m &lt; M . The
problem is that we have not put each of the models on the same footing
by taking into account their complexity (the number of inputs m in this
example).
</p>
<p>Stacked generalization, or stacking, is a way of doing this. Let f̂&minus;im (x)
be the prediction at x, using model m, applied to the dataset with the
ith training observation removed. The stacking estimate of the weights is
obtained from the least squares linear regression of yi on f̂
</p>
<p>&minus;i
m (xi), m =
</p>
<p>1, 2, . . . ,M . In detail the stacking weights are given by
</p>
<p>ŵst = argmin
w
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>[
yi &minus;
</p>
<p>M&sum;
</p>
<p>m=1
</p>
<p>wmf̂
&minus;i
m (xi)
</p>
<p>]2
. (8.59)
</p>
<p>The final prediction is
&sum;
</p>
<p>m ŵ
st
mf̂m(x). By using the cross-validated pre-
</p>
<p>dictions f̂&minus;im (x), stacking avoids giving unfairly high weight to models with
higher complexity. Better results can be obtained by restricting the weights
to be nonnegative, and to sum to 1. This seems like a reasonable restriction
if we interpret the weights as posterior model probabilities as in equation
(8.54), and it leads to a tractable quadratic programming problem.
There is a close connection between stacking and model selection via
</p>
<p>leave-one-out cross-validation (Section 7.10). If we restrict the minimization
in (8.59) to weight vectors w that have one unit weight and the rest zero,
this leads to a model choice m̂ with smallest leave-one-out cross-validation
error. Rather than choose a single model, stacking combines them with
estimated optimal weights. This will often lead to better prediction, but
less interpretability than the choice of only one of the M models.
</p>
<p>The stacking idea is actually more general than described above. One
can use any learning method, not just linear regression, to combine the
models as in (8.59); the weights could also depend on the input location
x. In this way, learning methods are &ldquo;stacked&rdquo; on top of one another, to
improve prediction performance.
</p>
<p>8.9 Stochastic Search: Bumping
</p>
<p>The final method described in this chapter does not involve averaging or
combining models, but rather is a technique for finding a better single
model. Bumping uses bootstrap sampling to move randomly through model
space. For problems where fitting method finds many local minima, bump-
ing can help the method to avoid getting stuck in poor solutions.</p>
<p/>
</div>
<div class="page"><p/>
<p>8.9 Stochastic Search: Bumping 291
</p>
<p>Regular 4-Node Tree
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>Bumped 4-Node Tree
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>FIGURE 8.13. Data with two features and two classes (blue and orange), dis-
playing a pure interaction. The left panel shows the partition found by three splits
of a standard, greedy, tree-growing algorithm. The vertical grey line near the left
edge is the first split, and the broken lines are the two subsequent splits. The al-
gorithm has no idea where to make a good initial split, and makes a poor choice.
The right panel shows the near-optimal splits found by bumping the tree-growing
algorithm 20 times.
</p>
<p>As in bagging, we draw bootstrap samples and fit a model to each. But
rather than average the predictions, we choose the model estimated from a
bootstrap sample that best fits the training data. In detail, we draw boot-
strap samples Z&lowast;1, . . . ,Z&lowast;B and fit our model to each, giving predictions
f̂&lowast;b(x), b = 1, 2, . . . , B at input point x. We then choose the model that
produces the smallest prediction error, averaged over the original training
set. For squared error, for example, we choose the model obtained from
bootstrap sample b̂, where
</p>
<p>b̂ = argmin
b
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>[yi &minus; f̂&lowast;b(xi)]2. (8.60)
</p>
<p>The corresponding model predictions are f̂&lowast;b̂(x). By convention we also
include the original training sample in the set of bootstrap samples, so that
the method is free to pick the original model if it has the lowest training
error.
By perturbing the data, bumping tries to move the fitting procedure
</p>
<p>around to good areas of model space. For example, if a few data points are
causing the procedure to find a poor solution, any bootstrap sample that
omits those data points should procedure a better solution.
For another example, consider the classification data in Figure 8.13, the
</p>
<p>notorious exclusive or (XOR) problem. There are two classes (blue and
orange) and two input features, with the features exhibiting a pure inter-</p>
<p/>
</div>
<div class="page"><p/>
<p>292 8. Model Inference and Averaging
</p>
<p>action. By splitting the data at x1 = 0 and then splitting each resulting
strata at x2 = 0, (or vice versa) a tree-based classifier could achieve per-
fect discrimination. However, the greedy, short-sighted CART algorithm
(Section 9.2) tries to find the best split on either feature, and then splits
the resulting strata. Because of the balanced nature of the data, all initial
splits on x1 or x2 appear to be useless, and the procedure essentially gener-
ates a random split at the top level. The actual split found for these data is
shown in the left panel of Figure 8.13. By bootstrap sampling from the data,
bumping breaks the balance in the classes, and with a reasonable number
of bootstrap samples (here 20), it will by chance produce at least one tree
with initial split near either x1 = 0 or x2 = 0. Using just 20 bootstrap
samples, bumping found the near optimal splits shown in the right panel
of Figure 8.13. This shortcoming of the greedy tree-growing algorithm is
exacerbated if we add a number of noise features that are independent of
the class label. Then the tree-growing algorithm cannot distinguish x1 or
x2 from the others, and gets seriously lost.
Since bumping compares different models on the training data, one must
</p>
<p>ensure that the models have roughly the same complexity. In the case of
trees, this would mean growing trees with the same number of terminal
nodes on each bootstrap sample. Bumping can also help in problems where
it is difficult to optimize the fitting criterion, perhaps because of a lack of
smoothness. The trick is to optimize a different, more convenient criterion
over the bootstrap samples, and then choose the model producing the best
results for the desired criterion on the training sample.
</p>
<p>Bibliographic Notes
</p>
<p>There are many books on classical statistical inference: Cox and Hink-
ley (1974) and Silvey (1975) give nontechnical accounts. The bootstrap
is due to Efron (1979) and is described more fully in Efron and Tibshi-
rani (1993) and Hall (1992). A good modern book on Bayesian inference
is Gelman et al. (1995). A lucid account of the application of Bayesian
methods to neural networks is given in Neal (1996). The statistical appli-
cation of Gibbs sampling is due to Geman and Geman (1984), and Gelfand
and Smith (1990), with related work by Tanner and Wong (1987). Markov
chain Monte Carlo methods, including Gibbs sampling and the Metropolis&ndash;
Hastings algorithm, are discussed in Spiegelhalter et al. (1996). The EM
algorithm is due to Dempster et al. (1977); as the discussants in that pa-
per make clear, there was much related, earlier work. The view of EM as
a joint maximization scheme for a penalized complete-data log-likelihood
was elucidated by Neal and Hinton (1998); they credit Csiszar and Tusnády
(1984) and Hathaway (1986) as having noticed this connection earlier. Bag-
ging was proposed by Breiman (1996a). Stacking is due to Wolpert (1992);</p>
<p/>
</div>
<div class="page"><p/>
<p>Exercises 293
</p>
<p>Breiman (1996b) contains an accessible discussion for statisticians. Leblanc
and Tibshirani (1996) describe variations on stacking based on the boot-
strap. Model averaging in the Bayesian framework has been recently advo-
cated by Madigan and Raftery (1994). Bumping was proposed by Tibshi-
rani and Knight (1999).
</p>
<p>Exercises
</p>
<p>Ex. 8.1 Let r(y) and q(y) be probability density functions. Jensen&rsquo;s in-
equality states that for a random variable X and a convex function φ(x),
E[φ(X)] &ge; φ[E(X)]. Use Jensen&rsquo;s inequality to show that
</p>
<p>Eq log[r(Y )/q(Y )] (8.61)
</p>
<p>is maximized as a function of r(y) when r(y) = q(y). Hence show that
R(θ, θ) &ge; R(θ&prime;, θ) as stated below equation (8.46).
</p>
<p>Ex. 8.2 Consider the maximization of the log-likelihood (8.48), over dis-
tributions P̃ (Zm) such that P̃ (Zm) &ge; 0 and &sum;Zm P̃ (Zm) = 1. Use La-
grange multipliers to show that the solution is the conditional distribution
P̃ (Zm) = Pr(Zm|Z, θ&prime;), as in (8.49).
</p>
<p>Ex. 8.3 Justify the estimate (8.50), using the relationship
</p>
<p>Pr(A) =
</p>
<p>&int;
Pr(A|B)d(Pr(B)).
</p>
<p>Ex. 8.4 Consider the bagging method of Section 8.7. Let our estimate f̂(x)
be the B-spline smoother μ̂(x) of Section 8.2.1. Consider the parametric
bootstrap of equation (8.6), applied to this estimator. Show that if we bag
</p>
<p>f̂(x), using the parametric bootstrap to generate the bootstrap samples,
</p>
<p>the bagging estimate f̂bag(x) converges to the original estimate f̂(x) as
B &rarr; &infin;.
</p>
<p>Ex. 8.5 Suggest generalizations of each of the loss functions in Figure 10.4
to more than two classes, and design an appropriate plot to compare them.
</p>
<p>Ex. 8.6 Consider the bone mineral density data of Figure 5.6.
</p>
<p>(a) Fit a cubic smooth spline to the relative change in spinal BMD, as a
function of age. Use cross-validation to estimate the optimal amount
of smoothing. Construct pointwise 90% confidence bands for the un-
derlying function.
</p>
<p>(b) Compute the posterior mean and covariance for the true function via
(8.28), and compare the posterior bands to those obtained in (a).</p>
<p/>
</div>
<div class="page"><p/>
<p>294 8. Model Inference and Averaging
</p>
<p>(c) Compute 100 bootstrap replicates of the fitted curves, as in the bottom
left panel of Figure 8.2. Compare the results to those obtained in (a)
and (b).
</p>
<p>Ex. 8.7 EM as a minorization algorithm(Hunter and Lange, 2004; Wu and
Lange, 2007). A function g(x, y) to said to minorize a function f(x) if
</p>
<p>g(x, y) &le; f(x), g(x, x) = f(x) (8.62)
</p>
<p>for all x, y in the domain. This is useful for maximizing f(x) since it is easy
to show that f(x) is non-decreasing under the update
</p>
<p>xs+1 = argmaxxg(x, x
s) (8.63)
</p>
<p>There are analogous definitions for majorization, for minimizing a function
f(x). The resulting algorithms are known asMM algorithms, for &ldquo;Minorize-
Maximize&rdquo; or &ldquo;Majorize-Minimize.&rdquo;
Show that the EM algorithm (Section 8.5.2) is an example of an MM al-
</p>
<p>gorithm, using Q(θ&prime;, θ)+log Pr(Z|θ)&minus;Q(θ, θ) to minorize the observed data
log-likelihood ℓ(θ&prime;;Z). (Note that only the first term involves the relevant
parameter θ&prime;).</p>
<p/>
</div>
<div class="page"><p/>
<p>9
Additive Models, Trees, and Related
Methods
</p>
<p>In this chapter we begin our discussion of some specific methods for super-
vised learning. These techniques each assume a (different) structured form
for the unknown regression function, and by doing so they finesse the curse
of dimensionality. Of course, they pay the possible price of misspecifying
the model, and so in each case there is a tradeoff that has to be made. They
take off where Chapters 3&ndash;6 left off. We describe five related techniques:
generalized additive models, trees, multivariate adaptive regression splines,
the patient rule induction method, and hierarchical mixtures of experts.
</p>
<p>9.1 Generalized Additive Models
</p>
<p>Regression models play an important role in many data analyses, providing
prediction and classification rules, and data analytic tools for understand-
ing the importance of different inputs.
Although attractively simple, the traditional linear model often fails in
</p>
<p>these situations: in real life, effects are often not linear. In earlier chapters
we described techniques that used predefined basis functions to achieve
nonlinearities. This section describes more automatic flexible statistical
methods that may be used to identify and characterize nonlinear regression
effects. These methods are called &ldquo;generalized additive models.&rdquo;
In the regression setting, a generalized additive model has the form
</p>
<p>E(Y |X1, X2, . . . , Xp) = α+ f1(X1) + f2(X2) + &middot; &middot; &middot;+ fp(Xp). (9.1)
</p>
<p> &copy; Springer Science+Business Media, LLC 2009 
</p>
<p>T. Hastie et al., The Elements of Statistical Learning, Second Edition, 295
</p>
<p>DOI: 10.1007/b94608_9,</p>
<p/>
</div>
<div class="page"><p/>
<p>296 9. Additive Models, Trees, and Related Methods
</p>
<p>As usual X1, X2, . . . , Xp represent predictors and Y is the outcome; the fj &rsquo;s
are unspecified smooth (&ldquo;nonparametric&rdquo;) functions. If we were to model
each function using an expansion of basis functions (as in Chapter 5), the
resulting model could then be fit by simple least squares. Our approach
here is different: we fit each function using a scatterplot smoother (e.g., a
cubic smoothing spline or kernel smoother), and provide an algorithm for
simultaneously estimating all p functions (Section 9.1.1).
For two-class classification, recall the logistic regression model for binary
</p>
<p>data discussed in Section 4.4. We relate the mean of the binary response
μ(X) = Pr(Y = 1|X) to the predictors via a linear regression model and
the logit link function:
</p>
<p>log
</p>
<p>(
μ(X)
</p>
<p>1&minus; μ(X)
</p>
<p>)
= α+ β1X1 + &middot; &middot; &middot;+ βpXp. (9.2)
</p>
<p>The additive logistic regression model replaces each linear term by a more
general functional form
</p>
<p>log
</p>
<p>(
μ(X)
</p>
<p>1&minus; μ(X)
</p>
<p>)
= α+ f1(X1) + &middot; &middot; &middot;+ fp(Xp), (9.3)
</p>
<p>where again each fj is an unspecified smooth function. While the non-
parametric form for the functions fj makes the model more flexible, the
additivity is retained and allows us to interpret the model in much the
same way as before. The additive logistic regression model is an example
of a generalized additive model. In general, the conditional mean μ(X) of
a response Y is related to an additive function of the predictors via a link
function g:
</p>
<p>g[μ(X)] = α+ f1(X1) + &middot; &middot; &middot;+ fp(Xp). (9.4)
Examples of classical link functions are the following:
</p>
<p>&bull; g(μ) = μ is the identity link, used for linear and additive models for
Gaussian response data.
</p>
<p>&bull; g(μ) = logit(μ) as above, or g(μ) = probit(μ), the probit link function,
for modeling binomial probabilities. The probit function is the inverse
Gaussian cumulative distribution function: probit(μ) = Φ&minus;1(μ).
</p>
<p>&bull; g(μ) = log(μ) for log-linear or log-additive models for Poisson count
data.
</p>
<p>All three of these arise from exponential family sampling models, which
in addition include the gamma and negative-binomial distributions. These
families generate the well-known class of generalized linear models, which
are all extended in the same way to generalized additive models.
The functions fj are estimated in a flexible manner, using an algorithm
</p>
<p>whose basic building block is a scatterplot smoother. The estimated func-
tion f̂j can then reveal possible nonlinearities in the effect of Xj . Not all</p>
<p/>
</div>
<div class="page"><p/>
<p>9.1 Generalized Additive Models 297
</p>
<p>of the functions fj need to be nonlinear. We can easily mix in linear and
other parametric forms with the nonlinear terms, a necessity when some of
the inputs are qualitative variables (factors). The nonlinear terms are not
restricted to main effects either; we can have nonlinear components in two
or more variables, or separate curves in Xj for each level of the factor Xk.
Thus each of the following would qualify:
</p>
<p>&bull; g(μ) = XTβ + αk + f(Z)&mdash;a semiparametric model, where X is a
vector of predictors to be modeled linearly, αk the effect for the kth
level of a qualitative input V , and the effect of predictor Z is modeled
nonparametrically.
</p>
<p>&bull; g(μ) = f(X) + gk(Z)&mdash;again k indexes the levels of a qualitative
input V , and thus creates an interaction term g(V,Z) = gk(Z) for
the effect of V and Z.
</p>
<p>&bull; g(μ) = f(X) + g(Z,W ) where g is a nonparametric function in two
features.
</p>
<p>Additive models can replace linear models in a wide variety of settings,
for example an additive decomposition of time series,
</p>
<p>Yt = St + Tt + εt, (9.5)
</p>
<p>where St is a seasonal component, Tt is a trend and ε is an error term.
</p>
<p>9.1.1 Fitting Additive Models
</p>
<p>In this section we describe a modular algorithm for fitting additive models
and their generalizations. The building block is the scatterplot smoother
for fitting nonlinear effects in a flexible way. For concreteness we use as our
scatterplot smoother the cubic smoothing spline described in Chapter 5.
The additive model has the form
</p>
<p>Y = α+
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>fj(Xj) + ε, (9.6)
</p>
<p>where the error term ε has mean zero. Given observations xi, yi, a criterion
like the penalized sum of squares (5.9) of Section 5.4 can be specified for
this problem,
</p>
<p>PRSS(α, f1, f2, . . . , fp) =
N&sum;
</p>
<p>i=1
</p>
<p>(
yi&minus;α&minus;
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>fj(xij)
</p>
<p>)2
+
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>λj
</p>
<p>&int;
f
</p>
<p>&prime;&prime;
</p>
<p>j (tj)
2dtj ,
</p>
<p>(9.7)
where the λj &ge; 0 are tuning parameters. It can be shown that the minimizer
of (9.7) is an additive cubic spline model; each of the functions fj is a</p>
<p/>
</div>
<div class="page"><p/>
<p>298 9. Additive Models, Trees, and Related Methods
</p>
<p>Algorithm 9.1 The Backfitting Algorithm for Additive Models.
</p>
<p>1. Initialize: α̂ = 1N
&sum;N
</p>
<p>1 yi, f̂j &equiv; 0, &forall;i, j.
</p>
<p>2. Cycle: j = 1, 2, . . . , p, . . . , 1, 2, . . . , p, . . . ,
</p>
<p>f̂j &larr; Sj
[
{yi &minus; α̂&minus;
</p>
<p>&sum;
</p>
<p>k �=j
</p>
<p>f̂k(xik)}N1
</p>
<p>]
,
</p>
<p>f̂j &larr; f̂j &minus;
1
</p>
<p>N
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>f̂j(xij).
</p>
<p>until the functions f̂j change less than a prespecified threshold.
</p>
<p>cubic spline in the component Xj , with knots at each of the unique values
of xij , i = 1, . . . , N . However, without further restrictions on the model,
the solution is not unique. The constant α is not identifiable, since we
can add or subtract any constants to each of the functions fj , and adjust
</p>
<p>α accordingly. The standard convention is to assume that
&sum;N
</p>
<p>1 fj(xij) =
0 &forall;j&mdash;the functions average zero over the data. It is easily seen that α̂ =
ave(yi) in this case. If in addition to this restriction, the matrix of input
values (having ijth entry xij) has full column rank, then (9.7) is a strictly
convex criterion and the minimizer is unique. If the matrix is singular, then
the linear part of the components fj cannot be uniquely determined (while
the nonlinear parts can!)(Buja et al., 1989).
Furthermore, a simple iterative procedure exists for finding the solution.
</p>
<p>We set α̂ = ave(yi), and it never changes. We apply a cubic smoothing
</p>
<p>spline Sj to the targets {yi &minus; α̂ &minus;
&sum;
</p>
<p>k �=j f̂k(xik)}N1 , as a function of xij ,
to obtain a new estimate f̂j . This is done for each predictor in turn, using
</p>
<p>the current estimates of the other functions f̂k when computing yi &minus; α̂ &minus;&sum;
k �=j f̂k(xik). The process is continued until the estimates f̂j stabilize. This
</p>
<p>procedure, given in detail in Algorithm 9.1, is known as &ldquo;backfitting&rdquo; and
the resulting fit is analogous to a multiple regression for linear models.
In principle, the second step in (2) of Algorithm 9.1 is not needed, since
</p>
<p>the smoothing spline fit to a mean-zero response has mean zero (Exer-
cise 9.1). In practice, machine rounding can cause slippage, and the ad-
justment is advised.
This same algorithm can accommodate other fitting methods in exactly
</p>
<p>the same way, by specifying appropriate smoothing operators Sj :
</p>
<p>&bull; other univariate regression smoothers such as local polynomial re-
gression and kernel methods;</p>
<p/>
</div>
<div class="page"><p/>
<p>9.1 Generalized Additive Models 299
</p>
<p>&bull; linear regression operators yielding polynomial fits, piecewise con-
stant fits, parametric spline fits, series and Fourier fits;
</p>
<p>&bull; more complicated operators such as surface smoothers for second or
higher-order interactions or periodic smoothers for seasonal effects.
</p>
<p>If we consider the operation of smoother Sj only at the training points, it
can be represented by an N &times; N operator matrix Sj (see Section 5.4.1).
Then the degrees of freedom for the jth term are (approximately) computed
as dfj = trace[Sj ] &minus; 1, by analogy with degrees of freedom for smoothers
discussed in Chapters 5 and 6.
For a large class of linear smoothers Sj , backfitting is equivalent to a
</p>
<p>Gauss&ndash;Seidel algorithm for solving a certain linear system of equations.
Details are given in Exercise 9.2.
For the logistic regression model and other generalized additive models,
</p>
<p>the appropriate criterion is a penalized log-likelihood. To maximize it, the
backfitting procedure is used in conjunction with a likelihood maximizer.
The usual Newton&ndash;Raphson routine for maximizing log-likelihoods in gen-
eralized linear models can be recast as an IRLS (iteratively reweighted
least squares) algorithm. This involves repeatedly fitting a weighted linear
regression of a working response variable on the covariates; each regression
yields a new value of the parameter estimates, which in turn give new work-
ing responses and weights, and the process is iterated (see Section 4.4.1).
In the generalized additive model, the weighted linear regression is simply
replaced by a weighted backfitting algorithm. We describe the algorithm in
more detail for logistic regression below, and more generally in Chapter 6
of Hastie and Tibshirani (1990).
</p>
<p>9.1.2 Example: Additive Logistic Regression
</p>
<p>Probably the most widely used model in medical research is the logistic
model for binary data. In this model the outcome Y can be coded as 0
or 1, with 1 indicating an event (like death or relapse of a disease) and
0 indicating no event. We wish to model Pr(Y = 1|X), the probability of
an event given values of the prognostic factors XT = (X1, . . . , Xp). The
goal is usually to understand the roles of the prognostic factors, rather
than to classify new individuals. Logistic models are also used in applica-
tions where one is interested in estimating the class probabilities, for use
in risk screening. Apart from medical applications, credit risk screening is
a popular application.
The generalized additive logistic model has the form
</p>
<p>log
Pr(Y = 1|X)
Pr(Y = 0|X) = α+ f1(X1) + &middot; &middot; &middot;+ fp(Xp). (9.8)
</p>
<p>The functions f1, f2, . . . , fp are estimated by a backfitting algorithm
within a Newton&ndash;Raphson procedure, shown in Algorithm 9.2.</p>
<p/>
</div>
<div class="page"><p/>
<p>300 9. Additive Models, Trees, and Related Methods
</p>
<p>Algorithm 9.2 Local Scoring Algorithm for the Additive Logistic Regres-
sion Model.
</p>
<p>1. Compute starting values: α̂ = log[ȳ/(1 &minus; ȳ)], where ȳ = ave(yi), the
sample proportion of ones, and set f̂j &equiv; 0 &forall;j.
</p>
<p>2. Define η̂i = α̂+
&sum;
</p>
<p>j f̂j(xij) and p̂i = 1/[1 + exp(&minus;η̂i)].
Iterate:
</p>
<p>(a) Construct the working target variable
</p>
<p>zi = η̂i +
(yi &minus; p̂i)
p̂i(1&minus; p̂i)
</p>
<p>.
</p>
<p>(b) Construct weights wi = p̂i(1&minus; p̂i)
(c) Fit an additive model to the targets zi with weights wi, us-
</p>
<p>ing a weighted backfitting algorithm. This gives new estimates
α̂, f̂j , &forall;j
</p>
<p>3. Continue step 2. until the change in the functions falls below a pre-
specified threshold.
</p>
<p>The additive model fitting in step (2) of Algorithm 9.2 requires a weighted
scatterplot smoother. Most smoothing procedures can accept observation
weights (Exercise 5.12); see Chapter 3 of Hastie and Tibshirani (1990) for
further details.
The additive logistic regression model can be generalized further to han-
</p>
<p>dle more than two classes, using the multilogit formulation as outlined in
Section 4.4. While the formulation is a straightforward extension of (9.8),
the algorithms for fitting such models are more complex. See Yee and Wild
(1996) for details, and the VGAM software currently available from:
</p>
<p>http://www.stat.auckland.ac.nz/&sim;yee.
</p>
<p>Example: Predicting Email Spam
</p>
<p>We apply a generalized additive model to the spam data introduced in
Chapter 1. The data consists of information from 4601 email messages, in
a study to screen email for &ldquo;spam&rdquo; (i.e., junk email). The data is publicly
available at ftp.ics.uci.edu, and was donated by George Forman from
Hewlett-Packard laboratories, Palo Alto, California.
The response variable is binary, with values email or spam, and there are
</p>
<p>57 predictors as described below:
</p>
<p>&bull; 48 quantitative predictors&mdash;the percentage of words in the email that
match a given word. Examples include business, address, internet,</p>
<p/>
</div>
<div class="page"><p/>
<p>9.1 Generalized Additive Models 301
</p>
<p>TABLE 9.1. Test data confusion matrix for the additive logistic regression model
fit to the spam training data. The overall test error rate is 5.5%.
</p>
<p>Predicted Class
True Class email (0) spam (1)
</p>
<p>email (0) 58.3% 2.5%
spam (1) 3.0% 36.3%
</p>
<p>free, and george. The idea was that these could be customized for
individual users.
</p>
<p>&bull; 6 quantitative predictors&mdash;the percentage of characters in the email
that match a given character. The characters are ch;, ch(, ch[, ch!,
ch$, and ch#.
</p>
<p>&bull; The average length of uninterrupted sequences of capital letters:
CAPAVE.
</p>
<p>&bull; The length of the longest uninterrupted sequence of capital letters:
CAPMAX.
</p>
<p>&bull; The sum of the length of uninterrupted sequences of capital letters:
CAPTOT.
</p>
<p>We coded spam as 1 and email as zero. A test set of size 1536 was randomly
chosen, leaving 3065 observations in the training set. A generalized additive
model was fit, using a cubic smoothing spline with a nominal four degrees of
freedom for each predictor. What this means is that for each predictor Xj ,
the smoothing-spline parameter λj was chosen so that trace[Sj(λj)]&minus;1 = 4,
where Sj(λ) is the smoothing spline operator matrix constructed using the
observed values xij , i = 1, . . . , N . This is a convenient way of specifying
the amount of smoothing in such a complex model.
Most of the spam predictors have a very long-tailed distribution. Before
</p>
<p>fitting the GAM model, we log-transformed each variable (actually log(x+
0.1)), but the plots in Figure 9.1 are shown as a function of the original
variables.
The test error rates are shown in Table 9.1; the overall error rate is 5.3%.
</p>
<p>By comparison, a linear logistic regression has a test error rate of 7.6%.
Table 9.2 shows the predictors that are highly significant in the additive
model.
For ease of interpretation, in Table 9.2 the contribution for each variable
</p>
<p>is decomposed into a linear component and the remaining nonlinear com-
ponent. The top block of predictors are positively correlated with spam,
while the bottom block is negatively correlated. The linear component is a
weighted least squares linear fit of the fitted curve on the predictor, while
the nonlinear part is the residual. The linear component of an estimated</p>
<p/>
</div>
<div class="page"><p/>
<p>302 9. Additive Models, Trees, and Related Methods
</p>
<p>TABLE 9.2. Significant predictors from the additive model fit to the spam train-
ing data. The coefficients represent the linear part of f̂j , along with their standard
errors and Z-score. The nonlinear P-value is for a test of nonlinearity of f̂j.
</p>
<p>Name Num. df Coefficient Std. Error Z Score Nonlinear
P -value
</p>
<p>Positive effects
</p>
<p>our 5 3.9 0.566 0.114 4.970 0.052
over 6 3.9 0.244 0.195 1.249 0.004
remove 7 4.0 0.949 0.183 5.201 0.093
internet 8 4.0 0.524 0.176 2.974 0.028
free 16 3.9 0.507 0.127 4.010 0.065
business 17 3.8 0.779 0.186 4.179 0.194
hpl 26 3.8 0.045 0.250 0.181 0.002
ch! 52 4.0 0.674 0.128 5.283 0.164
ch$ 53 3.9 1.419 0.280 5.062 0.354
CAPMAX 56 3.8 0.247 0.228 1.080 0.000
CAPTOT 57 4.0 0.755 0.165 4.566 0.063
</p>
<p>Negative effects
</p>
<p>hp 25 3.9 &minus;1.404 0.224 &minus;6.262 0.140
george 27 3.7 &minus;5.003 0.744 &minus;6.722 0.045
1999 37 3.8 &minus;0.672 0.191 &minus;3.512 0.011
re 45 3.9 &minus;0.620 0.133 &minus;4.649 0.597
edu 46 4.0 &minus;1.183 0.209 &minus;5.647 0.000
</p>
<p>function is summarized by the coefficient, standard error and Z-score; the
latter is the coefficient divided by its standard error, and is considered
significant if it exceeds the appropriate quantile of a standard normal dis-
tribution. The column labeled nonlinear P -value is a test of nonlinearity
of the estimated function. Note, however, that the effect of each predictor
is fully adjusted for the entire effects of the other predictors, not just for
their linear parts. The predictors shown in the table were judged signifi-
cant by at least one of the tests (linear or nonlinear) at the p = 0.01 level
(two-sided).
Figure 9.1 shows the estimated functions for the significant predictors
</p>
<p>appearing in Table 9.2. Many of the nonlinear effects appear to account for
a strong discontinuity at zero. For example, the probability of spam drops
significantly as the frequency of george increases from zero, but then does
not change much after that. This suggests that one might replace each of
the frequency predictors by an indicator variable for a zero count, and resort
to a linear logistic model. This gave a test error rate of 7.4%; including the
linear effects of the frequencies as well dropped the test error to 6.6%. It
appears that the nonlinearities in the additive model have an additional
predictive power.</p>
<p/>
</div>
<div class="page"><p/>
<p>9.1 Generalized Additive Models 303
</p>
<p>0 2 4 6 8
</p>
<p>-5
0
</p>
<p>5
</p>
<p>0 1 2 3
</p>
<p>-5
0
</p>
<p>5
</p>
<p>0 2 4 6
</p>
<p>-5
0
</p>
<p>5
1
0
</p>
<p>0 2 4 6 8 10
</p>
<p>-5
0
</p>
<p>5
1
0
</p>
<p>0 2 4 6 8 10
</p>
<p>-5
0
</p>
<p>5
1
0
</p>
<p>0 2 4 6
</p>
<p>-5
0
</p>
<p>5
1
0
</p>
<p>0 5 10 15 20
</p>
<p>-1
0
</p>
<p>-5
0
</p>
<p>0 5 10
</p>
<p>-1
0
</p>
<p>-5
0
</p>
<p>0 10 20 30
</p>
<p>-1
0
</p>
<p>-5
0
</p>
<p>5
</p>
<p>0 2 4 6
</p>
<p>-5
0
</p>
<p>5
</p>
<p>0 5 10 15 20
</p>
<p>-1
0
</p>
<p>-5
0
</p>
<p>5
</p>
<p>0 5 10 15
</p>
<p>-1
0
</p>
<p>-5
0
</p>
<p>0 10 20 30
</p>
<p>-5
0
</p>
<p>5
1
0
</p>
<p>0 1 2 3 4 5 6
</p>
<p>-5
0
</p>
<p>5
1
0
</p>
<p>0 2000 6000 10000
</p>
<p>-5
0
</p>
<p>5
</p>
<p>0 5000 10000 15000
</p>
<p>-5
0
</p>
<p>5
</p>
<p>our over remove internet
</p>
<p>free business hp hpl
</p>
<p>george 1999 re edu
</p>
<p>ch! ch$ CAPMAX CAPTOT
</p>
<p>f̂
(o
u
r
)
</p>
<p>f̂
(o
v
e
r
)
</p>
<p>f̂
(r
e
m
o
v
e
)
</p>
<p>f̂
(i
n
t
e
r
n
e
t
)
</p>
<p>f̂
(f
r
e
e
)
</p>
<p>f̂
(b
u
s
i
n
e
s
s
)
</p>
<p>f̂
(h
p
)
</p>
<p>f̂
(h
p
l
)
</p>
<p>f̂
(g
e
o
r
g
e
)
</p>
<p>f̂
(1
9
9
9
)
</p>
<p>f̂
(r
e
)
</p>
<p>f̂
(e
d
u
)
</p>
<p>f̂
(c
h
!)
</p>
<p>f̂
(c
h
$
)
</p>
<p>f̂
(C
A
P
M
A
X
)
</p>
<p>f̂
(C
A
P
T
O
T
)
</p>
<p>FIGURE 9.1. Spam analysis: estimated functions for significant predictors. The
rug plot along the bottom of each frame indicates the observed values of the cor-
responding predictor. For many of the predictors the nonlinearity picks up the
discontinuity at zero.</p>
<p/>
</div>
<div class="page"><p/>
<p>304 9. Additive Models, Trees, and Related Methods
</p>
<p>It is more serious to classify a genuine email message as spam, since then
a good email would be filtered out and would not reach the user. We can
alter the balance between the class error rates by changing the losses (see
Section 2.4). If we assign a loss L01 for predicting a true class 0 as class 1,
and L10 for predicting a true class 1 as class 0, then the estimated Bayes
rule predicts class 1 if its probability is greater than L01/(L01 + L10). For
example, if we take L01 = 10, L10 = 1 then the (true) class 0 and class 1
error rates change to 0.8% and 8.7%.
More ambitiously, we can encourage the model to fit better data in the
</p>
<p>class 0 by using weights L01 for the class 0 observations and L10 for the
class 1 observations. As above, we then use the estimated Bayes rule to
predict. This gave error rates of 1.2% and 8.0% in (true) class 0 and class 1,
respectively. We discuss below the issue of unequal losses further, in the
context of tree-based models.
After fitting an additive model, one should check whether the inclusion
</p>
<p>of some interactions can significantly improve the fit. This can be done
&ldquo;manually,&rdquo; by inserting products of some or all of the significant inputs,
or automatically via the MARS procedure (Section 9.4).
This example uses the additive model in an automatic fashion. As a data
</p>
<p>analysis tool, additive models are often used in a more interactive fashion,
adding and dropping terms to determine their effect. By calibrating the
amount of smoothing in terms of dfj , one can move seamlessly between
linear models (dfj = 1) and partially linear models, where some terms are
modeled more flexibly. See Hastie and Tibshirani (1990) for more details.
</p>
<p>9.1.3 Summary
</p>
<p>Additive models provide a useful extension of linear models, making them
more flexible while still retaining much of their interpretability. The familiar
tools for modeling and inference in linear models are also available for
additive models, seen for example in Table 9.2. The backfitting procedure
for fitting these models is simple and modular, allowing one to choose a
fitting method appropriate for each input variable. As a result they have
become widely used in the statistical community.
However additive models can have limitations for large data-mining ap-
</p>
<p>plications. The backfitting algorithm fits all predictors, which is not feasi-
ble or desirable when a large number are available. The BRUTO procedure
(Hastie and Tibshirani, 1990, Chapter 9) combines backfitting with selec-
tion of inputs, but is not designed for large data-mining problems. There
has also been recent work using lasso-type penalties to estimate sparse ad-
ditive models, for example the COSSO procedure of Lin and Zhang (2006)
and the SpAM proposal of Ravikumar et al. (2008). For large problems a
forward stagewise approach such as boosting (Chapter 10) is more effective,
and also allows for interactions to be included in the model.</p>
<p/>
</div>
<div class="page"><p/>
<p>9.2 Tree-Based Methods 305
</p>
<p>9.2 Tree-Based Methods
</p>
<p>9.2.1 Background
</p>
<p>Tree-based methods partition the feature space into a set of rectangles, and
then fit a simple model (like a constant) in each one. They are conceptually
simple yet powerful. We first describe a popular method for tree-based
regression and classification called CART, and later contrast it with C4.5,
a major competitor.
Let&rsquo;s consider a regression problem with continuous response Y and in-
</p>
<p>puts X1 and X2, each taking values in the unit interval. The top left panel
of Figure 9.2 shows a partition of the feature space by lines that are parallel
to the coordinate axes. In each partition element we can model Y with a
different constant. However, there is a problem: although each partitioning
line has a simple description like X1 = c, some of the resulting regions are
complicated to describe.
To simplify matters, we restrict attention to recursive binary partitions
</p>
<p>like that in the top right panel of Figure 9.2. We first split the space into
two regions, and model the response by the mean of Y in each region.
We choose the variable and split-point to achieve the best fit. Then one
or both of these regions are split into two more regions, and this process
is continued, until some stopping rule is applied. For example, in the top
right panel of Figure 9.2, we first split at X1 = t1. Then the region X1 &le; t1
is split at X2 = t2 and the region X1 &gt; t1 is split at X1 = t3. Finally, the
region X1 &gt; t3 is split at X2 = t4. The result of this process is a partition
into the five regions R1, R2, . . . , R5 shown in the figure. The corresponding
regression model predicts Y with a constant cm in region Rm, that is,
</p>
<p>f̂(X) =
</p>
<p>5&sum;
</p>
<p>m=1
</p>
<p>cmI{(X1, X2) &isin; Rm}. (9.9)
</p>
<p>This same model can be represented by the binary tree in the bottom left
panel of Figure 9.2. The full dataset sits at the top of the tree. Observations
satisfying the condition at each junction are assigned to the left branch,
and the others to the right branch. The terminal nodes or leaves of the
tree correspond to the regions R1, R2, . . . , R5. The bottom right panel of
Figure 9.2 is a perspective plot of the regression surface from this model.
For illustration, we chose the node means c1 = &minus;5, c2 = &minus;7, c3 = 0, c4 =
2, c5 = 4 to make this plot.
A key advantage of the recursive binary tree is its interpretability. The
</p>
<p>feature space partition is fully described by a single tree. With more than
two inputs, partitions like that in the top right panel of Figure 9.2 are
difficult to draw, but the binary tree representation works in the same
way. This representation is also popular among medical scientists, perhaps
because it mimics the way that a doctor thinks. The tree stratifies the</p>
<p/>
</div>
<div class="page"><p/>
<p>306 9. Additive Models, Trees, and Related Methods
</p>
<p>|
</p>
<p>t1
</p>
<p>t2
</p>
<p>t3
</p>
<p>t4
</p>
<p>R1
</p>
<p>R1
</p>
<p>R2
</p>
<p>R2
</p>
<p>R3
</p>
<p>R3
</p>
<p>R4
</p>
<p>R4
</p>
<p>R5
</p>
<p>R5
</p>
<p>X1
</p>
<p>X1X1
</p>
<p>X2
</p>
<p>X
2
</p>
<p>X
2
</p>
<p>X1 &le; t1
</p>
<p>X2 &le; t2 X1 &le; t3
</p>
<p>X2 &le; t4
</p>
<p>FIGURE 9.2. Partitions and CART. Top right panel shows a partition of a
two-dimensional feature space by recursive binary splitting, as used in CART,
applied to some fake data. Top left panel shows a general partition that cannot
be obtained from recursive binary splitting. Bottom left panel shows the tree cor-
responding to the partition in the top right panel, and a perspective plot of the
prediction surface appears in the bottom right panel.</p>
<p/>
</div>
<div class="page"><p/>
<p>9.2 Tree-Based Methods 307
</p>
<p>population into strata of high and low outcome, on the basis of patient
characteristics.
</p>
<p>9.2.2 Regression Trees
</p>
<p>We now turn to the question of how to grow a regression tree. Our data
consists of p inputs and a response, for each of N observations: that is,
(xi, yi) for i = 1, 2, . . . , N , with xi = (xi1, xi2, . . . , xip). The algorithm
needs to automatically decide on the splitting variables and split points,
and also what topology (shape) the tree should have. Suppose first that we
have a partition into M regions R1, R2, . . . , RM , and we model the response
as a constant cm in each region:
</p>
<p>f(x) =
</p>
<p>M&sum;
</p>
<p>m=1
</p>
<p>cmI(x &isin; Rm). (9.10)
</p>
<p>If we adopt as our criterion minimization of the sum of squares
&sum;
</p>
<p>(yi &minus;
f(xi))
</p>
<p>2, it is easy to see that the best ĉm is just the average of yi in region
Rm:
</p>
<p>ĉm = ave(yi|xi &isin; Rm). (9.11)
</p>
<p>Now finding the best binary partition in terms of minimum sum of squares
is generally computationally infeasible. Hence we proceed with a greedy
algorithm. Starting with all of the data, consider a splitting variable j and
split point s, and define the pair of half-planes
</p>
<p>R1(j, s) = {X|Xj &le; s} and R2(j, s) = {X|Xj &gt; s}. (9.12)
</p>
<p>Then we seek the splitting variable j and split point s that solve
</p>
<p>min
j, s
</p>
<p>[
min
c1
</p>
<p>&sum;
</p>
<p>xi&isin;R1(j,s)
</p>
<p>(yi &minus; c1)2 +min
c2
</p>
<p>&sum;
</p>
<p>xi&isin;R2(j,s)
</p>
<p>(yi &minus; c2)2
]
. (9.13)
</p>
<p>For any choice j and s, the inner minimization is solved by
</p>
<p>ĉ1 = ave(yi|xi &isin; R1(j, s)) and ĉ2 = ave(yi|xi &isin; R2(j, s)). (9.14)
</p>
<p>For each splitting variable, the determination of the split point s can
be done very quickly and hence by scanning through all of the inputs,
determination of the best pair (j, s) is feasible.
</p>
<p>Having found the best split, we partition the data into the two resulting
regions and repeat the splitting process on each of the two regions. Then
this process is repeated on all of the resulting regions.
How large should we grow the tree? Clearly a very large tree might overfit
</p>
<p>the data, while a small tree might not capture the important structure.</p>
<p/>
</div>
<div class="page"><p/>
<p>308 9. Additive Models, Trees, and Related Methods
</p>
<p>Tree size is a tuning parameter governing the model&rsquo;s complexity, and the
optimal tree size should be adaptively chosen from the data. One approach
would be to split tree nodes only if the decrease in sum-of-squares due to the
split exceeds some threshold. This strategy is too short-sighted, however,
since a seemingly worthless split might lead to a very good split below it.
The preferred strategy is to grow a large tree T0, stopping the splitting
</p>
<p>process only when some minimum node size (say 5) is reached. Then this
large tree is pruned using cost-complexity pruning, which we now describe.
We define a subtree T &sub; T0 to be any tree that can be obtained by
</p>
<p>pruning T0, that is, collapsing any number of its internal (non-terminal)
nodes. We index terminal nodes by m, with node m representing region
Rm. Let |T | denote the number of terminal nodes in T . Letting
</p>
<p>Nm = #{xi &isin; Rm},
</p>
<p>ĉm =
1
</p>
<p>Nm
</p>
<p>&sum;
</p>
<p>xi&isin;Rm
</p>
<p>yi,
</p>
<p>Qm(T ) =
1
</p>
<p>Nm
</p>
<p>&sum;
</p>
<p>xi&isin;Rm
</p>
<p>(yi &minus; ĉm)2,
</p>
<p>(9.15)
</p>
<p>we define the cost complexity criterion
</p>
<p>Cα(T ) =
</p>
<p>|T |&sum;
</p>
<p>m=1
</p>
<p>NmQm(T ) + α|T |. (9.16)
</p>
<p>The idea is to find, for each α, the subtree Tα &sube; T0 to minimize Cα(T ).
The tuning parameter α &ge; 0 governs the tradeoff between tree size and its
goodness of fit to the data. Large values of α result in smaller trees Tα, and
conversely for smaller values of α. As the notation suggests, with α = 0 the
solution is the full tree T0. We discuss how to adaptively choose α below.
For each α one can show that there is a unique smallest subtree Tα that
</p>
<p>minimizes Cα(T ). To find Tα we use weakest link pruning: we successively
collapse the internal node that produces the smallest per-node increase in&sum;
</p>
<p>m NmQm(T ), and continue until we produce the single-node (root) tree.
This gives a (finite) sequence of subtrees, and one can show this sequence
must contain Tα. See Breiman et al. (1984) or Ripley (1996) for details.
Estimation of α is achieved by five- or tenfold cross-validation: we choose
the value α̂ to minimize the cross-validated sum of squares. Our final tree
is Tα̂.
</p>
<p>9.2.3 Classification Trees
</p>
<p>If the target is a classification outcome taking values 1, 2, . . . ,K, the only
changes needed in the tree algorithm pertain to the criteria for splitting
nodes and pruning the tree. For regression we used the squared-error node</p>
<p/>
</div>
<div class="page"><p/>
<p>9.2 Tree-Based Methods 309
</p>
<p>0.0 0.2 0.4 0.6 0.8 1.0
</p>
<p>0
.0
</p>
<p>0
.1
</p>
<p>0
.2
</p>
<p>0
.3
</p>
<p>0
.4
</p>
<p>0
.5
</p>
<p>p
</p>
<p>Entropy
</p>
<p>G
in
i i
nd
</p>
<p>ex
</p>
<p>M
is
cl
as
</p>
<p>si
fic
</p>
<p>at
io
n 
</p>
<p>er
ro
</p>
<p>r
</p>
<p>FIGURE 9.3. Node impurity measures for two-class classification, as a function
of the proportion p in class 2. Cross-entropy has been scaled to pass through
(0.5, 0.5).
</p>
<p>impurity measure Qm(T ) defined in (9.15), but this is not suitable for
classification. In a node m, representing a region Rm with Nm observations,
let
</p>
<p>p̂mk =
1
</p>
<p>Nm
</p>
<p>&sum;
</p>
<p>xi&isin;Rm
</p>
<p>I(yi = k),
</p>
<p>the proportion of class k observations in node m. We classify the obser-
vations in node m to class k(m) = argmaxk p̂mk, the majority class in
node m. Different measures Qm(T ) of node impurity include the following:
</p>
<p>Misclassification error: 1Nm
&sum;
</p>
<p>i&isin;Rm
I(yi �= k(m)) = 1&minus; p̂mk(m).
</p>
<p>Gini index:
&sum;
</p>
<p>k �=k&prime; p̂mkp̂mk&prime; =
&sum;K
</p>
<p>k=1 p̂mk(1&minus; p̂mk).
Cross-entropy or deviance: &minus;&sum;Kk=1 p̂mk log p̂mk.
</p>
<p>(9.17)
For two classes, if p is the proportion in the second class, these three mea-
sures are 1 &minus; max(p, 1 &minus; p), 2p(1 &minus; p) and &minus;p log p &minus; (1 &minus; p) log (1&minus; p),
respectively. They are shown in Figure 9.3. All three are similar, but cross-
entropy and the Gini index are differentiable, and hence more amenable to
numerical optimization. Comparing (9.13) and (9.15), we see that we need
to weight the node impurity measures by the number NmL and NmR of
observations in the two child nodes created by splitting node m.
</p>
<p>In addition, cross-entropy and the Gini index are more sensitive to changes
in the node probabilities than the misclassification rate. For example, in
a two-class problem with 400 observations in each class (denote this by
(400, 400)), suppose one split created nodes (300, 100) and (100, 300), while</p>
<p/>
</div>
<div class="page"><p/>
<p>310 9. Additive Models, Trees, and Related Methods
</p>
<p>the other created nodes (200, 400) and (200, 0). Both splits produce a mis-
classification rate of 0.25, but the second split produces a pure node and is
probably preferable. Both the Gini index and cross-entropy are lower for the
second split. For this reason, either the Gini index or cross-entropy should
be used when growing the tree. To guide cost-complexity pruning, any of
the three measures can be used, but typically it is the misclassification rate.
The Gini index can be interpreted in two interesting ways. Rather than
</p>
<p>classify observations to the majority class in the node, we could classify
them to class k with probability p̂mk. Then the expected training error
rate of this rule in the node is
</p>
<p>&sum;
</p>
<p>k 6=k&prime; p̂mkp̂mk&prime;&mdash;the Gini index. Similarly,
if we code each observation as 1 for class k and zero otherwise, the variance
over the node of this 0-1 response is p̂mk(1 &minus; p̂mk). Summing over classes
k again gives the Gini index.
</p>
<p>9.2.4 Other Issues
</p>
<p>Categorical Predictors
</p>
<p>When splitting a predictor having q possible unordered values, there are
2q&minus;1 &minus; 1 possible partitions of the q values into two groups, and the com-
putations become prohibitive for large q. However, with a 0 &minus; 1 outcome,
this computation simplifies. We order the predictor classes according to the
proportion falling in outcome class 1. Then we split this predictor as if it
were an ordered predictor. One can show this gives the optimal split, in
terms of cross-entropy or Gini index, among all possible 2q&minus;1&minus;1 splits. This
result also holds for a quantitative outcome and square error loss&mdash;the cat-
egories are ordered by increasing mean of the outcome. Although intuitive,
the proofs of these assertions are not trivial. The proof for binary outcomes
is given in Breiman et al. (1984) and Ripley (1996); the proof for quantita-
tive outcomes can be found in Fisher (1958). For multicategory outcomes,
no such simplifications are possible, although various approximations have
been proposed (Loh and Vanichsetakul, 1988).
The partitioning algorithm tends to favor categorical predictors with
</p>
<p>many levels q; the number of partitions grows exponentially in q, and the
more choices we have, the more likely we can find a good one for the data
at hand. This can lead to severe overfitting if q is large, and such variables
should be avoided.
</p>
<p>The Loss Matrix
</p>
<p>In classification problems, the consequences of misclassifying observations
are more serious in some classes than others. For example, it is probably
worse to predict that a person will not have a heart attack when he/she
actually will, than vice versa. To account for this, we define a K &times;K loss
matrix L, with Lkk&prime; being the loss incurred for classifying a class k obser-
vation as class k&prime;. Typically no loss is incurred for correct classifications,</p>
<p/>
</div>
<div class="page"><p/>
<p>9.2 Tree-Based Methods 311
</p>
<p>that is, Lkk = 0 &forall;k. To incorporate the losses into the modeling process,
we could modify the Gini index to
</p>
<p>&sum;
k �=k&prime; Lkk&prime; p̂mkp̂mk&prime; ; this would be the
</p>
<p>expected loss incurred by the randomized rule. This works for the multi-
class case, but in the two-class case has no effect, since the coefficient of
p̂mkp̂mk&prime; is Lkk&prime; + Lk&prime;k. For two classes a better approach is to weight the
observations in class k by Lkk&prime; . This can be used in the multiclass case only
if, as a function of k, Lkk&prime; doesn&rsquo;t depend on k
</p>
<p>&prime;. Observation weighting can
be used with the deviance as well. The effect of observation weighting is to
alter the prior probability on the classes. In a terminal node, the empirical
Bayes rule implies that we classify to class k(m) = argmink
</p>
<p>&sum;
ℓ Lℓkp̂mℓ.
</p>
<p>Missing Predictor Values
</p>
<p>Suppose our data has some missing predictor values in some or all of the
variables. We might discard any observation with some missing values, but
this could lead to serious depletion of the training set. Alternatively we
might try to fill in (impute) the missing values, with say the mean of that
predictor over the nonmissing observations. For tree-based models, there
are two better approaches. The first is applicable to categorical predictors:
we simply make a new category for &ldquo;missing.&rdquo; From this we might dis-
cover that observations with missing values for some measurement behave
differently than those with nonmissing values. The second more general
approach is the construction of surrogate variables. When considering a
predictor for a split, we use only the observations for which that predictor
is not missing. Having chosen the best (primary) predictor and split point,
we form a list of surrogate predictors and split points. The first surrogate
is the predictor and corresponding split point that best mimics the split of
the training data achieved by the primary split. The second surrogate is
the predictor and corresponding split point that does second best, and so
on. When sending observations down the tree either in the training phase
or during prediction, we use the surrogate splits in order, if the primary
splitting predictor is missing. Surrogate splits exploit correlations between
predictors to try and alleviate the effect of missing data. The higher the cor-
relation between the missing predictor and the other predictors, the smaller
the loss of information due to the missing value. The general problem of
missing data is discussed in Section 9.6.
</p>
<p>Why Binary Splits?
</p>
<p>Rather than splitting each node into just two groups at each stage (as
above), we might consider multiway splits into more than two groups. While
this can sometimes be useful, it is not a good general strategy. The problem
is that multiway splits fragment the data too quickly, leaving insufficient
data at the next level down. Hence we would want to use such splits only
when needed. Since multiway splits can be achieved by a series of binary
splits, the latter are preferred.</p>
<p/>
</div>
<div class="page"><p/>
<p>312 9. Additive Models, Trees, and Related Methods
</p>
<p>Other Tree-Building Procedures
</p>
<p>The discussion above focuses on the CART (classification and regression
tree) implementation of trees. The other popular methodology is ID3 and
its later versions, C4.5 and C5.0 (Quinlan, 1993). Early versions of the
program were limited to categorical predictors, and used a top-down rule
with no pruning. With more recent developments, C5.0 has become quite
similar to CART. The most significant feature unique to C5.0 is a scheme
for deriving rule sets. After a tree is grown, the splitting rules that define the
terminal nodes can sometimes be simplified: that is, one or more condition
can be dropped without changing the subset of observations that fall in
the node. We end up with a simplified set of rules defining each terminal
node; these no longer follow a tree structure, but their simplicity might
make them more attractive to the user.
</p>
<p>Linear Combination Splits
</p>
<p>Rather than restricting splits to be of the form Xj &le; s, one can allow splits
along linear combinations of the form
</p>
<p>&sum;
ajXj &le; s. The weights aj and
</p>
<p>split point s are optimized to minimize the relevant criterion (such as the
Gini index). While this can improve the predictive power of the tree, it can
hurt interpretability. Computationally, the discreteness of the split point
search precludes the use of a smooth optimization for the weights. A better
way to incorporate linear combination splits is in the hierarchical mixtures
of experts (HME) model, the topic of Section 9.5.
</p>
<p>Instability of Trees
</p>
<p>One major problem with trees is their high variance. Often a small change
in the data can result in a very different series of splits, making interpre-
tation somewhat precarious. The major reason for this instability is the
hierarchical nature of the process: the effect of an error in the top split
is propagated down to all of the splits below it. One can alleviate this to
some degree by trying to use a more stable split criterion, but the inherent
instability is not removed. It is the price to be paid for estimating a simple,
tree-based structure from the data. Bagging (Section 8.7) averages many
trees to reduce this variance.
</p>
<p>Lack of Smoothness
</p>
<p>Another limitation of trees is the lack of smoothness of the prediction sur-
face, as can be seen in the bottom right panel of Figure 9.2. In classification
with 0/1 loss, this doesn&rsquo;t hurt much, since bias in estimation of the class
probabilities has a limited effect. However, this can degrade performance
in the regression setting, where we would normally expect the underlying
function to be smooth. The MARS procedure, described in Section 9.4,</p>
<p/>
</div>
<div class="page"><p/>
<p>9.2 Tree-Based Methods 313
</p>
<p>TABLE 9.3. Spam data: confusion rates for the 17-node tree (chosen by cross&ndash;
validation) on the test data. Overall error rate is 9.3%.
</p>
<p>Predicted
</p>
<p>True email spam
email 57.3% 4.0%
spam 5.3% 33.4%
</p>
<p>can be viewed as a modification of CART designed to alleviate this lack of
smoothness.
</p>
<p>Difficulty in Capturing Additive Structure
</p>
<p>Another problem with trees is their difficulty in modeling additive struc-
ture. In regression, suppose, for example, that Y = c1I(X1 &lt; t1)+c2I(X2 &lt;
t2) + ε where ε is zero-mean noise. Then a binary tree might make its first
split on X1 near t1. At the next level down it would have to split both nodes
on X2 at t2 in order to capture the additive structure. This might happen
with sufficient data, but the model is given no special encouragement to find
such structure. If there were ten rather than two additive effects, it would
take many fortuitous splits to recreate the structure, and the data analyst
would be hard pressed to recognize it in the estimated tree. The &ldquo;blame&rdquo;
here can again be attributed to the binary tree structure, which has both
advantages and drawbacks. Again the MARS method (Section 9.4) gives
up this tree structure in order to capture additive structure.
</p>
<p>9.2.5 Spam Example (Continued)
</p>
<p>We applied the classification tree methodology to the spam example intro-
duced earlier. We used the deviance measure to grow the tree and mis-
classification rate to prune it. Figure 9.4 shows the 10-fold cross-validation
error rate as a function of the size of the pruned tree, along with &plusmn;2 stan-
dard errors of the mean, from the ten replications. The test error curve is
shown in orange. Note that the cross-validation error rates are indexed by
a sequence of values of α and not tree size; for trees grown in different folds,
a value of α might imply different sizes. The sizes shown at the base of the
plot refer to |Tα|, the sizes of the pruned original tree.
The error flattens out at around 17 terminal nodes, giving the pruned tree
</p>
<p>in Figure 9.5. Of the 13 distinct features chosen by the tree, 11 overlap with
the 16 significant features in the additive model (Table 9.2). The overall
error rate shown in Table 9.3 is about 50% higher than for the additive
model in Table 9.1.
Consider the rightmost branches of the tree. We branch to the right
</p>
<p>with a spam warning if more than 5.5% of the characters are the $ sign.</p>
<p/>
</div>
<div class="page"><p/>
<p>314 9. Additive Models, Trees, and Related Methods
</p>
<p>0 10 20 30 40
</p>
<p>0
.0
</p>
<p>0
.1
</p>
<p>0
.2
</p>
<p>0
.3
</p>
<p>0
.4
</p>
<p>Tree Size
</p>
<p>M
is
</p>
<p>c
la
</p>
<p>s
s
if
ic
</p>
<p>a
ti
o
n
 R
</p>
<p>a
te
</p>
<p>176 21 7 5 3 2 0
</p>
<p>α
</p>
<p>FIGURE 9.4. Results for spam example. The blue curve is the 10-fold cross-val-
idation estimate of misclassification rate as a function of tree size, with standard
error bars. The minimum occurs at a tree size with about 17 terminal nodes (using
the &ldquo;one-standard-error&rdquo; rule). The orange curve is the test error, which tracks
the CV error quite closely. The cross-validation is indexed by values of α, shown
above. The tree sizes shown below refer to |Tα|, the size of the original tree indexed
by α.
</p>
<p>However, if in addition the phrase hp occurs frequently, then this is likely
to be company business and we classify as email. All of the 22 cases in
the test set satisfying these criteria were correctly classified. If the second
condition is not met, and in addition the average length of repeated capital
letters CAPAVE is larger than 2.9, then we classify as spam. Of the 227 test
cases, only seven were misclassified.
In medical classification problems, the terms sensitivity and specificity
</p>
<p>are used to characterize a rule. They are defined as follows:
</p>
<p>Sensitivity: probability of predicting disease given true state is disease.
</p>
<p>Specificity: probability of predicting non-disease given true state is non-
disease.</p>
<p/>
</div>
<div class="page"><p/>
<p>9.2 Tree-Based Methods 315
</p>
<p>600/1536
</p>
<p>280/1177
</p>
<p>180/1065
</p>
<p> 80/861
</p>
<p> 80/652
</p>
<p> 77/423
</p>
<p> 20/238
</p>
<p> 19/236   1/2
</p>
<p> 57/185
</p>
<p> 48/113
</p>
<p> 37/101   1/12
</p>
<p>  9/72
</p>
<p>  3/229
</p>
<p>  0/209
</p>
<p>100/204
</p>
<p> 36/123
</p>
<p> 16/94
</p>
<p> 14/89   3/5
</p>
<p>  9/29
</p>
<p> 16/81
</p>
<p>  9/112
</p>
<p>  6/109   0/3
</p>
<p> 48/359
</p>
<p> 26/337
</p>
<p> 19/110
</p>
<p> 18/109   0/1
</p>
<p>  7/227
</p>
<p>  0/22
</p>
<p>spam
</p>
<p>spam
</p>
<p>spam
</p>
<p>spam
</p>
<p>spam
</p>
<p>spam
</p>
<p>spam
</p>
<p>spam
</p>
<p>spam
</p>
<p>spam
</p>
<p>spam
</p>
<p>spam
</p>
<p>email
</p>
<p>email
</p>
<p>email
</p>
<p>email
</p>
<p>email
</p>
<p>email
</p>
<p>email
</p>
<p>email
</p>
<p>email
</p>
<p>email
</p>
<p>email
</p>
<p>email
</p>
<p>email
</p>
<p>email
</p>
<p>email
</p>
<p>email
</p>
<p>email
</p>
<p>email
</p>
<p>email
</p>
<p>email
</p>
<p>email
</p>
<p>ch$&lt;0.0555
</p>
<p>remove&lt;0.06
</p>
<p>ch!&lt;0.191
</p>
<p>george&lt;0.005
</p>
<p>hp&lt;0.03
</p>
<p>CAPMAX&lt;10.5
</p>
<p>receive&lt;0.125 edu&lt;0.045
</p>
<p>our&lt;1.2
</p>
<p>CAPAVE&lt;2.7505
</p>
<p>free&lt;0.065
</p>
<p>business&lt;0.145
</p>
<p>george&lt;0.15
</p>
<p>hp&lt;0.405
</p>
<p>CAPAVE&lt;2.907
</p>
<p>1999&lt;0.58
</p>
<p>ch$&gt;0.0555
</p>
<p>remove&gt;0.06
</p>
<p>ch!&gt;0.191
</p>
<p>george&gt;0.005
</p>
<p>hp&gt;0.03
</p>
<p>CAPMAX&gt;10.5
</p>
<p>receive&gt;0.125 edu&gt;0.045
</p>
<p>our&gt;1.2
</p>
<p>CAPAVE&gt;2.7505
</p>
<p>free&gt;0.065
</p>
<p>business&gt;0.145
</p>
<p>george&gt;0.15
</p>
<p>hp&gt;0.405
</p>
<p>CAPAVE&gt;2.907
</p>
<p>1999&gt;0.58
</p>
<p>FIGURE 9.5. The pruned tree for the spam example. The split variables are
shown in blue on the branches, and the classification is shown in every node.The
numbers under the terminal nodes indicate misclassification rates on the test data.</p>
<p/>
</div>
<div class="page"><p/>
<p>316 9. Additive Models, Trees, and Related Methods
</p>
<p>Specificity
</p>
<p>S
e
n
s
it
iv
</p>
<p>it
y
</p>
<p>0.0 0.2 0.4 0.6 0.8 1.0
</p>
<p>0
.0
</p>
<p>0
.2
</p>
<p>0
.4
</p>
<p>0
.6
</p>
<p>0
.8
</p>
<p>1
.0 &bull; &bull;
</p>
<p>&bull;
&bull; &bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull; &bull; &bull; &bull; &bull; &bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
&bull;&bull;&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>Tree (0.95)
GAM (0.98)
Weighted Tree (0.90)
</p>
<p>FIGURE 9.6. ROC curves for the classification rules fit to the spam data. Curves
that are closer to the northeast corner represent better classifiers. In this case the
GAM classifier dominates the trees. The weighted tree achieves better sensitivity
for higher specificity than the unweighted tree. The numbers in the legend repre-
sent the area under the curve.
</p>
<p>If we think of spam and email as the presence and absence of disease, re-
spectively, then from Table 9.3 we have
</p>
<p>Sensitivity = 100&times; 33.4
33.4 + 5.3
</p>
<p>= 86.3%,
</p>
<p>Specificity = 100&times; 57.3
57.3 + 4.0
</p>
<p>= 93.4%.
</p>
<p>In this analysis we have used equal losses. As before let Lkk&prime; be the
loss associated with predicting a class k object as class k&prime;. By varying the
relative sizes of the losses L01 and L10, we increase the sensitivity and
decrease the specificity of the rule, or vice versa. In this example, we want
to avoid marking good email as spam, and thus we want the specificity to
be very high. We can achieve this by setting L01 &gt; 1 say, with L10 = 1.
The Bayes&rsquo; rule in each terminal node classifies to class 1 (spam) if the
proportion of spam is &ge; L01/(L10 + L01), and class zero otherwise. The</p>
<p/>
</div>
<div class="page"><p/>
<p>9.3 PRIM: Bump Hunting 317
</p>
<p>receiver operating characteristic curve (ROC) is a commonly used summary
for assessing the tradeoff between sensitivity and specificity. It is a plot of
the sensitivity versus specificity as we vary the parameters of a classification
rule. Varying the loss L01 between 0.1 and 10, and applying Bayes&rsquo; rule to
the 17-node tree selected in Figure 9.4, produced the ROC curve shown
in Figure 9.6. The standard error of each curve near 0.9 is approximately&radic;
0.9(1&minus; 0.9)/1536 = 0.008, and hence the standard error of the difference
</p>
<p>is about 0.01. We see that in order to achieve a specificity of close to 100%,
the sensitivity has to drop to about 50%. The area under the curve is a
commonly used quantitative summary; extending the curve linearly in each
direction so that it is defined over [0, 100], the area is approximately 0.95.
For comparison, we have included the ROC curve for the GAM model fit
to these data in Section 9.2; it gives a better classification rule for any loss,
with an area of 0.98.
Rather than just modifying the Bayes rule in the nodes, it is better to
</p>
<p>take full account of the unequal losses in growing the tree, as was done
in Section 9.2. With just two classes 0 and 1, losses may be incorporated
into the tree-growing process by using weight Lk,1&minus;k for an observation in
class k. Here we chose L01 = 5, L10 = 1 and fit the same size tree as before
(|Tα| = 17). This tree has higher sensitivity at high values of the specificity
than the original tree, but does more poorly at the other extreme. Its top
few splits are the same as the original tree, and then it departs from it.
For this application the tree grown using L01 = 5 is clearly better than the
original tree.
The area under the ROC curve, used above, is sometimes called the c-
</p>
<p>statistic. Interestingly, it can be shown that the area under the ROC curve
is equivalent to the Mann-Whitney U statistic (or Wilcoxon rank-sum test),
for the median difference between the prediction scores in the two groups
(Hanley and McNeil, 1982). For evaluating the contribution of an additional
predictor when added to a standard model, the c-statistic may not be an
informative measure. The new predictor can be very significant in terms
of the change in model deviance, but show only a small increase in the c-
statistic. For example, removal of the highly significant term george from
the model of Table 9.2 results in a decrease in the c-statistic of less than
0.01. Instead, it is useful to examine how the additional predictor changes
the classification on an individual sample basis. A good discussion of this
point appears in Cook (2007).
</p>
<p>9.3 PRIM: Bump Hunting
</p>
<p>Tree-based methods (for regression) partition the feature space into box-
shaped regions, to try to make the response averages in each box as differ-</p>
<p/>
</div>
<div class="page"><p/>
<p>318 9. Additive Models, Trees, and Related Methods
</p>
<p>ent as possible. The splitting rules defining the boxes are related to each
through a binary tree, facilitating their interpretation.
The patient rule induction method (PRIM) also finds boxes in the feature
</p>
<p>space, but seeks boxes in which the response average is high. Hence it looks
for maxima in the target function, an exercise known as bump hunting. (If
minima rather than maxima are desired, one simply works with the negative
response values.)
PRIM also differs from tree-based partitioning methods in that the box
</p>
<p>definitions are not described by a binary tree. This makes interpretation of
the collection of rules more difficult; however, by removing the binary tree
constraint, the individual rules are often simpler.
The main box construction method in PRIM works from the top down,
</p>
<p>starting with a box containing all of the data. The box is compressed along
one face by a small amount, and the observations then falling outside the
box are peeled off. The face chosen for compression is the one resulting in
the largest box mean, after the compression is performed. Then the process
is repeated, stopping when the current box contains some minimum number
of data points.
This process is illustrated in Figure 9.7. There are 200 data points uni-
</p>
<p>formly distributed over the unit square. The color-coded plot indicates the
response Y taking the value 1 (red) when 0.5 &lt; X1 &lt; 0.8 and 0.4 &lt; X2 &lt;
0.6. and zero (blue) otherwise. The panels shows the successive boxes found
by the top-down peeling procedure, peeling off a proportion α = 0.1 of the
remaining data points at each stage.
Figure 9.8 shows the mean of the response values in the box, as the box
</p>
<p>is compressed.
After the top-down sequence is computed, PRIM reverses the process,
</p>
<p>expanding along any edge, if such an expansion increases the box mean.
This is called pasting. Since the top-down procedure is greedy at each step,
such an expansion is often possible.
The result of these steps is a sequence of boxes, with different numbers
</p>
<p>of observation in each box. Cross-validation, combined with the judgment
of the data analyst, is used to choose the optimal box size.
Denote by B1 the indices of the observations in the box found in step 1.
</p>
<p>The PRIM procedure then removes the observations in B1 from the training
set, and the two-step process&mdash;top down peeling, followed by bottom-up
pasting&mdash;is repeated on the remaining dataset. This entire process is re-
peated several times, producing a sequence of boxes B1, B2, . . . , Bk. Each
box is defined by a set of rules involving a subset of predictors like
</p>
<p>(a1 &le; X1 &le; b1) and (b1 &le; X3 &le; b2).
A summary of the PRIM procedure is given Algorithm 9.3.
PRIM can handle a categorical predictor by considering all partitions of
</p>
<p>the predictor, as in CART. Missing values are also handled in a manner
similar to CART. PRIM is designed for regression (quantitative response</p>
<p/>
</div>
<div class="page"><p/>
<p>9.3 PRIM: Bump Hunting 319
</p>
<p>1
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>2
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>3
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>4
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>5
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>6
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>7
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>8
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>12
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>17
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>22
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>27
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>FIGURE 9.7. Illustration of PRIM algorithm. There are two classes, indicated
by the blue (class 0) and red (class 1) points. The procedure starts with a rectangle
(broken black lines) surrounding all of the data, and then peels away points along
one edge by a prespecified amount in order to maximize the mean of the points
remaining in the box. Starting at the top left panel, the sequence of peelings is
shown, until a pure red region is isolated in the bottom right panel. The iteration
number is indicated at the top of each panel.
</p>
<p>Number of Observations in Box
</p>
<p>B
o
</p>
<p>x
 M
</p>
<p>e
a
</p>
<p>n
</p>
<p>50 100 150
</p>
<p>0
.2
</p>
<p>0
.4
</p>
<p>0
.6
</p>
<p>0
.8
</p>
<p>1
.0
</p>
<p>&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;
</p>
<p>&bull;&bull;&bull;
&bull;&bull;
</p>
<p>&bull;&bull;
&bull;&bull;
</p>
<p>&bull;
&bull;
&bull;
&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>FIGURE 9.8. Box mean as a function of number of observations in the box.</p>
<p/>
</div>
<div class="page"><p/>
<p>320 9. Additive Models, Trees, and Related Methods
</p>
<p>Algorithm 9.3 Patient Rule Induction Method.
</p>
<p>1. Start with all of the training data, and a maximal box containing all
of the data.
</p>
<p>2. Consider shrinking the box by compressing one face, so as to peel off
the proportion α of observations having either the highest values of
a predictor Xj , or the lowest. Choose the peeling that produces the
highest response mean in the remaining box. (Typically α = 0.05 or
0.10.)
</p>
<p>3. Repeat step 2 until some minimal number of observations (say 10)
remain in the box.
</p>
<p>4. Expand the box along any face, as long as the resulting box mean
increases.
</p>
<p>5. Steps 1&ndash;4 give a sequence of boxes, with different numbers of obser-
vations in each box. Use cross-validation to choose a member of the
sequence. Call the box B1.
</p>
<p>6. Remove the data in box B1 from the dataset and repeat steps 2&ndash;5 to
obtain a second box, and continue to get as many boxes as desired.
</p>
<p>variable); a two-class outcome can be handled simply by coding it as 0 and
1. There is no simple way to deal with k &gt; 2 classes simultaneously: one
approach is to run PRIM separately for each class versus a baseline class.
An advantage of PRIM over CART is its patience. Because of its bi-
</p>
<p>nary splits, CART fragments the data quite quickly. Assuming splits of
equal size, with N observations it can only make log2(N)&minus; 1 splits before
running out of data. If PRIM peels off a proportion α of training points
at each stage, it can perform approximately &minus; log(N)/ log(1 &minus; α) peeling
steps before running out of data. For example, if N = 128 and α = 0.10,
then log2(N)&minus; 1 = 6 while &minus; log(N)/ log(1&minus;α) &asymp; 46. Taking into account
that there must be an integer number of observations at each stage, PRIM
in fact can peel only 29 times. In any case, the ability of PRIM to be more
patient should help the top-down greedy algorithm find a better solution.
</p>
<p>9.3.1 Spam Example (Continued)
</p>
<p>We applied PRIM to the spam data, with the response coded as 1 for spam
and 0 for email.
The first two boxes found by PRIM are summarized below:</p>
<p/>
</div>
<div class="page"><p/>
<p>9.4 MARS: Multivariate Adaptive Regression Splines 321
</p>
<p>Rule 1 Global Mean Box Mean Box Support
Training 0.3931 0.9607 0.1413
Test 0.3958 1.0000 0.1536
</p>
<p>Rule 1
</p>
<p>⎧
⎪⎪⎪⎪⎪⎪⎪⎪⎪⎪⎨
⎪⎪⎪⎪⎪⎪⎪⎪⎪⎪⎩
</p>
<p>ch! &gt; 0.029
CAPAVE &gt; 2.331
your &gt; 0.705
1999 &lt; 0.040
</p>
<p>CAPTOT &gt; 79.50
edu &lt; 0.070
re &lt; 0.535
ch; &lt; 0.030
</p>
<p>Rule 2 Remain Mean Box Mean Box Support
</p>
<p>Training 0.2998 0.9560 0.1043
Test 0.2862 0.9264 0.1061
</p>
<p>Rule 2
</p>
<p>{
remove &gt; 0.010
george &lt; 0.110
</p>
<p>The box support is the proportion of observations falling in the box.
The first box is purely spam, and contains about 15% of the test data.
The second box contains 10.6% of the test observations, 92.6% of which
are spam. Together the two boxes contain 26% of the data and are about
97% spam. The next few boxes (not shown) are quite small, containing only
about 3% of the data.
The predictors are listed in order of importance. Interestingly the top
</p>
<p>splitting variables in the CART tree (Figure 9.5) do not appear in PRIM&rsquo;s
first box.
</p>
<p>9.4 MARS: Multivariate Adaptive Regression
Splines
</p>
<p>MARS is an adaptive procedure for regression, and is well suited for high-
dimensional problems (i.e., a large number of inputs). It can be viewed as a
generalization of stepwise linear regression or a modification of the CART
method to improve the latter&rsquo;s performance in the regression setting. We
introduce MARS from the first point of view, and later make the connection
to CART.
MARS uses expansions in piecewise linear basis functions of the form
</p>
<p>(x&minus; t)+ and (t&minus; x)+. The &ldquo;+&rdquo; means positive part, so
</p>
<p>(x&minus;t)+ =
{
</p>
<p>x&minus; t, if x &gt; t,
0, otherwise,
</p>
<p>and (t&minus;x)+ =
{
</p>
<p>t&minus; x, if x &lt; t,
0, otherwise.</p>
<p/>
</div>
<div class="page"><p/>
<p>322 9. Additive Models, Trees, and Related Methods
</p>
<p>0.0 0.2 0.4 0.6 0.8 1.0
</p>
<p>0
.0
</p>
<p>0
.1
</p>
<p>0
.2
</p>
<p>0
.3
</p>
<p>0
.4
</p>
<p>0
.5
</p>
<p>(x&minus; t)+(t&minus; x)+
</p>
<p>x
</p>
<p>t
</p>
<p>B
a
si
s
F
u
n
ct
io
n
</p>
<p>FIGURE 9.9. The basis functions (x&minus; t)+ (solid orange) and (t&minus; x)+ (broken
blue) used by MARS.
</p>
<p>As an example, the functions (x&minus; 0.5)+ and (0.5&minus; x)+ are shown in Fig-
ure 9.9.
Each function is piecewise linear, with a knot at the value t. In the
</p>
<p>terminology of Chapter 5, these are linear splines. We call the two functions
a reflected pair in the discussion below. The idea is to form reflected pairs
for each input Xj with knots at each observed value xij of that input.
Therefore, the collection of basis functions is
</p>
<p>C = {(Xj &minus; t)+, (t&minus;Xj)+} t &isin; {x1j , x2j , . . . , xNj}
j = 1, 2, . . . , p.
</p>
<p>(9.18)
</p>
<p>If all of the input values are distinct, there are 2Np basis functions alto-
gether. Note that although each basis function depends only on a single
Xj , for example, h(X) = (Xj &minus; t)+, it is considered as a function over the
entire input space IRp.
The model-building strategy is like a forward stepwise linear regression,
</p>
<p>but instead of using the original inputs, we are allowed to use functions
from the set C and their products. Thus the model has the form
</p>
<p>f(X) = β0 +
M&sum;
</p>
<p>m=1
</p>
<p>βmhm(X), (9.19)
</p>
<p>where each hm(X) is a function in C, or a product of two or more such
functions.
Given a choice for the hm, the coefficients βm are estimated by minimiz-
</p>
<p>ing the residual sum-of-squares, that is, by standard linear regression. The
real art, however, is in the construction of the functions hm(x). We start
with only the constant function h0(X) = 1 in our model, and all functions
in the set C are candidate functions. This is depicted in Figure 9.10.
At each stage we consider as a new basis function pair all products of a
</p>
<p>function hm in the model set M with one of the reflected pairs in C. We
add to the model M the term of the form
</p>
<p>β̂M+1hℓ(X) &middot; (Xj &minus; t)+ + β̂M+2hℓ(X) &middot; (t&minus;Xj)+, hℓ &isin; M,</p>
<p/>
</div>
<div class="page"><p/>
<p>9.4 MARS: Multivariate Adaptive Regression Splines 323
</p>
<p>X1
</p>
<p>X1
</p>
<p>X1
</p>
<p>X1
</p>
<p>X2
</p>
<p>X2
</p>
<p>X2
</p>
<p>X2
</p>
<p>X2
</p>
<p>Xp
</p>
<p>Xp
</p>
<p>Xp
</p>
<p>Constant
</p>
<p>FIGURE 9.10. Schematic of the MARS forward model-building procedure. On
the left are the basis functions currently in the model: initially, this is the constant
function h(X) = 1. On the right are all candidate basis functions to be considered
in building the model. These are pairs of piecewise linear basis functions as in
Figure 9.9, with knots t at all unique observed values xij of each predictor Xj.
At each stage we consider all products of a candidate pair with a basis function
in the model. The product that decreases the residual error the most is added into
the current model. Above we illustrate the first three steps of the procedure, with
the selected functions shown in red.</p>
<p/>
</div>
<div class="page"><p/>
<p>324 9. Additive Models, Trees, and Related Methods
</p>
<p>X1
X2
</p>
<p>h(X1, X2)
</p>
<p>FIGURE 9.11. The function h(X1, X2) = (X1 &minus; x51)+ &middot; (x72 &minus;X2)+, resulting
from multiplication of two piecewise linear MARS basis functions.
</p>
<p>that produces the largest decrease in training error. Here β̂M+1 and β̂M+2
are coefficients estimated by least squares, along with all the other M + 1
coefficients in the model. Then the winning products are added to the
model and the process is continued until the model set M contains some
preset maximum number of terms.
For example, at the first stage we consider adding to the model a function
</p>
<p>of the form β1(Xj &minus; t)+ + β2(t &minus;Xj)+; t &isin; {xij}, since multiplication by
the constant function just produces the function itself. Suppose the best
choice is β̂1(X2 &minus; x72)+ + β̂2(x72 &minus;X2)+. Then this pair of basis functions
is added to the set M, and at the next stage we consider including a pair
of products the form
</p>
<p>hm(X) &middot; (Xj &minus; t)+ and hm(X) &middot; (t&minus;Xj)+, t &isin; {xij},
</p>
<p>where for hm we have the choices
</p>
<p>h0(X) = 1,
</p>
<p>h1(X) = (X2 &minus; x72)+, or
h2(X) = (x72 &minus;X2)+.
</p>
<p>The third choice produces functions such as (X1 &minus; x51)+ &middot; (x72 &minus; X2)+,
depicted in Figure 9.11.
At the end of this process we have a large model of the form (9.19). This
</p>
<p>model typically overfits the data, and so a backward deletion procedure
is applied. The term whose removal causes the smallest increase in resid-
ual squared error is deleted from the model at each stage, producing an
estimated best model f̂λ of each size (number of terms) λ. One could use
cross-validation to estimate the optimal value of λ, but for computational</p>
<p/>
</div>
<div class="page"><p/>
<p>9.4 MARS: Multivariate Adaptive Regression Splines 325
</p>
<p>savings the MARS procedure instead uses generalized cross-validation. This
criterion is defined as
</p>
<p>GCV(λ) =
</p>
<p>&sum;N
i=1(yi &minus; f̂λ(xi))2
(1&minus;M(λ)/N)2 . (9.20)
</p>
<p>The value M(λ) is the effective number of parameters in the model: this
accounts both for the number of terms in the models, plus the number
of parameters used in selecting the optimal positions of the knots. Some
mathematical and simulation results suggest that one should pay a price
of three parameters for selecting a knot in a piecewise linear regression.
Thus if there are r linearly independent basis functions in the model, and
</p>
<p>K knots were selected in the forward process, the formula isM(λ) = r+cK,
where c = 3. (When the model is restricted to be additive&mdash;details below&mdash;
a penalty of c = 2 is used). Using this, we choose the model along the
backward sequence that minimizes GCV(λ).
</p>
<p>Why these piecewise linear basis functions, and why this particular model
strategy? A key property of the functions of Figure 9.9 is their ability to
operate locally; they are zero over part of their range. When they are mul-
tiplied together, as in Figure 9.11, the result is nonzero only over the small
part of the feature space where both component functions are nonzero. As
a result, the regression surface is built up parsimoniously, using nonzero
components locally&mdash;only where they are needed. This is important, since
one should &ldquo;spend&rdquo; parameters carefully in high dimensions, as they can
run out quickly. The use of other basis functions such as polynomials, would
produce a nonzero product everywhere, and would not work as well.
The second important advantage of the piecewise linear basis function
</p>
<p>concerns computation. Consider the product of a function in M with each
of the N reflected pairs for an input Xj . This appears to require the fitting
of N single-input linear regression models, each of which uses O(N) oper-
ations, making a total of O(N2) operations. However, we can exploit the
simple form of the piecewise linear function. We first fit the reflected pair
with rightmost knot. As the knot is moved successively one position at a
time to the left, the basis functions differ by zero over the left part of the
domain, and by a constant over the right part. Hence after each such move
we can update the fit in O(1) operations. This allows us to try every knot
in only O(N) operations.
</p>
<p>The forward modeling strategy in MARS is hierarchical, in the sense that
multiway products are built up from products involving terms already in
the model. For example, a four-way product can only be added to the model
if one of its three-way components is already in the model. The philosophy
here is that a high-order interaction will likely only exist if some of its lower-
order &ldquo;footprints&rdquo; exist as well. This need not be true, but is a reasonable
working assumption and avoids the search over an exponentially growing
space of alternatives.</p>
<p/>
</div>
<div class="page"><p/>
<p>326 9. Additive Models, Trees, and Related Methods
</p>
<p>Rank of Model
</p>
<p>T
e
</p>
<p>s
t 
</p>
<p>M
is
</p>
<p>c
la
</p>
<p>s
s
if
ic
</p>
<p>a
ti
o
</p>
<p>n
 E
</p>
<p>rr
o
</p>
<p>r
</p>
<p>0 20 40 60 80 100
</p>
<p>0
.1
</p>
<p>0
.2
</p>
<p>0
.3
</p>
<p>0
.4
</p>
<p>&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
</p>
<p>&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;
</p>
<p>&bull;&bull;&bull;
</p>
<p>&bull;&bull;
&bull;
</p>
<p>&bull;
</p>
<p>0.055
</p>
<p>GCV choice
</p>
<p>FIGURE 9.12. Spam data: test error misclassification rate for the MARS pro-
cedure, as a function of the rank (number of independent basis functions) in the
model.
</p>
<p>There is one restriction put on the formation of model terms: each input
can appear at most once in a product. This prevents the formation of
higher-order powers of an input, which increase or decrease too sharply
near the boundaries of the feature space. Such powers can be approximated
in a more stable way with piecewise linear functions.
A useful option in the MARS procedure is to set an upper limit on
</p>
<p>the order of interaction. For example, one can set a limit of two, allowing
pairwise products of piecewise linear functions, but not three- or higher-
way products. This can aid in the interpretation of the final model. An
upper limit of one results in an additive model.
</p>
<p>9.4.1 Spam Example (Continued)
</p>
<p>We applied MARS to the &ldquo;spam&rdquo; data analyzed earlier in this chapter. To
enhance interpretability, we restricted MARS to second-degree interactions.
Although the target is a two-class variable, we used the squared-error loss
function nonetheless (see Section 9.4.3). Figure 9.12 shows the test error
misclassification rate as a function of the rank (number of independent ba-
sis functions) in the model. The error rate levels off at about 5.5%, which is
slightly higher than that of the generalized additive model (5.3%) discussed
earlier. GCV chose a model size of 60, which is roughly the smallest model
giving optimal performance. The leading interactions found by MARS in-
volved inputs (ch$, remove), (ch$, free) and (hp, CAPTOT). However, these
interactions give no improvement in performance over the generalized ad-
ditive model.</p>
<p/>
</div>
<div class="page"><p/>
<p>9.4 MARS: Multivariate Adaptive Regression Splines 327
</p>
<p>9.4.2 Example (Simulated Data)
</p>
<p>Here we examine the performance of MARS in three contrasting scenarios.
There are N = 100 observations, and the predictors X1, X2, . . . , Xp and
errors ε have independent standard normal distributions.
</p>
<p>Scenario 1: The data generation model is
</p>
<p>Y = (X1 &minus; 1)+ + (X1 &minus; 1)+ &middot; (X2 &minus; .8)+ + 0.12 &middot; ε. (9.21)
</p>
<p>The noise standard deviation 0.12 was chosen so that the signal-to-
noise ratio was about 5. We call this the tensor-product scenario; the
product term gives a surface that looks like that of Figure 9.11.
</p>
<p>Scenario 2: This is the same as scenario 1, but with p = 20 total predictors;
that is, there are 18 inputs that are independent of the response.
</p>
<p>Scenario 3: This has the structure of a neural network:
</p>
<p>ℓ1 = X1 +X2 +X3 +X4 +X5,
ℓ2 = X6 &minus;X7 +X8 &minus;X9 +X10,
</p>
<p>σ(t) = 1/(1 + e&minus;t),
Y = σ(ℓ1) + σ(ℓ2) + 0.12 &middot; ε.
</p>
<p>(9.22)
</p>
<p>Scenarios 1 and 2 are ideally suited for MARS, while scenario 3 contains
high-order interactions and may be difficult for MARS to approximate. We
ran five simulations from each model, and recorded the results.
In scenario 1, MARS typically uncovered the correct model almost per-
</p>
<p>fectly. In scenario 2, it found the correct structure but also found a few
extraneous terms involving other predictors.
Let μ(x) be the true mean of Y , and let
</p>
<p>MSE0 = avex&isin;Test(ȳ &minus; μ(x))2,
MSE = avex&isin;Test(f̂(x)&minus; μ(x))2.
</p>
<p>(9.23)
</p>
<p>These represent the mean-square error of the constant model and the fitted
MARS model, estimated by averaging at the 1000 test values of x. Table 9.4
shows the proportional decrease in model error or R2 for each scenario:
</p>
<p>R2 =
MSE0 &minus;MSE
</p>
<p>MSE0
. (9.24)
</p>
<p>The values shown are means and standard error over the five simulations.
The performance of MARS is degraded only slightly by the inclusion of the
useless inputs in scenario 2; it performs substantially worse in scenario 3.</p>
<p/>
</div>
<div class="page"><p/>
<p>328 9. Additive Models, Trees, and Related Methods
</p>
<p>TABLE 9.4. Proportional decrease in model error (R2) when MARS is applied
to three different scenarios.
</p>
<p>Scenario Mean (S.E.)
</p>
<p>1: Tensor product p = 2 0.97 (0.01)
2: Tensor product p = 20 0.96 (0.01)
3: Neural network 0.79 (0.01)
</p>
<p>9.4.3 Other Issues
</p>
<p>MARS for Classification
</p>
<p>The MARS method and algorithm can be extended to handle classification
problems. Several strategies have been suggested.
For two classes, one can code the output as 0/1 and treat the problem as
</p>
<p>a regression; we did this for the spam example. For more than two classes,
one can use the indicator response approach described in Section 4.2. One
codes the K response classes via 0/1 indicator variables, and then per-
forms a multi-response MARS regression. For the latter we use a common
set of basis functions for all response variables. Classification is made to
the class with the largest predicted response value. There are, however, po-
tential masking problems with this approach, as described in Section 4.2.
A generally superior approach is the &ldquo;optimal scoring&rdquo; method discussed
in Section 12.5.
Stone et al. (1997) developed a hybrid of MARS called PolyMARS specif-
</p>
<p>ically designed to handle classification problems. It uses the multiple logistic
framework described in Section 4.4. It grows the model in a forward stage-
wise fashion like MARS, but at each stage uses a quadratic approximation
to the multinomial log-likelihood to search for the next basis-function pair.
Once found, the enlarged model is fit by maximum likelihood, and the
process is repeated.
</p>
<p>Relationship of MARS to CART
</p>
<p>Although they might seem quite different, the MARS and CART strategies
actually have strong similarities. Suppose we take the MARS procedure and
make the following changes:
</p>
<p>&bull; Replace the piecewise linear basis functions by step functions I(x&minus;t &gt;
0) and I(x&minus; t &le; 0).
</p>
<p>&bull; When a model term is involved in a multiplication by a candidate
term, it gets replaced by the interaction, and hence is not available
for further interactions.
</p>
<p>With these changes, the MARS forward procedure is the same as the CART
tree-growing algorithm. Multiplying a step function by a pair of reflected</p>
<p/>
</div>
<div class="page"><p/>
<p>9.5 Hierarchical Mixtures of Experts 329
</p>
<p>step functions is equivalent to splitting a node at the step. The second
restriction implies that a node may not be split more than once, and leads
to the attractive binary-tree representation of the CART model. On the
other hand, it is this restriction that makes it difficult for CART to model
additive structures. MARS forgoes the tree structure and gains the ability
to capture additive effects.
</p>
<p>Mixed Inputs
</p>
<p>Mars can handle &ldquo;mixed&rdquo; predictors&mdash;quantitative and qualitative&mdash;in a
natural way, much like CART does. MARS considers all possible binary
partitions of the categories for a qualitative predictor into two groups.
Each such partition generates a pair of piecewise constant basis functions&mdash;
indicator functions for the two sets of categories. This basis pair is now
treated as any other, and is used in forming tensor products with other
basis functions already in the model.
</p>
<p>9.5 Hierarchical Mixtures of Experts
</p>
<p>The hierarchical mixtures of experts (HME) procedure can be viewed as a
variant of tree-based methods. The main difference is that the tree splits
are not hard decisions but rather soft probabilistic ones. At each node an
observation goes left or right with probabilities depending on its input val-
ues. This has some computational advantages since the resulting parameter
optimization problem is smooth, unlike the discrete split point search in the
tree-based approach. The soft splits might also help in prediction accuracy
and provide a useful alternative description of the data.
There are other differences between HMEs and the CART implementa-
</p>
<p>tion of trees. In an HME, a linear (or logistic regression) model is fit in
each terminal node, instead of a constant as in CART. The splits can be
multiway, not just binary, and the splits are probabilistic functions of a
linear combination of inputs, rather than a single input as in the standard
use of CART. However, the relative merits of these choices are not clear,
and most were discussed at the end of Section 9.2.
A simple two-level HME model in shown in Figure 9.13. It can be thought
</p>
<p>of as a tree with soft splits at each non-terminal node. However, the inven-
tors of this methodology use a different terminology. The terminal nodes
are called experts, and the non-terminal nodes are called gating networks.
The idea is that each expert provides an opinion (prediction) about the
response, and these are combined together by the gating networks. As we
will see, the model is formally a mixture model, and the two-level model
in the figure can be extend to multiple levels, hence the name hierarchical
mixtures of experts.</p>
<p/>
</div>
<div class="page"><p/>
<p>330 9. Additive Models, Trees, and Related Methods
</p>
<p>g1 g2
</p>
<p>g1|1 g2|1 g1|2 g2|2
</p>
<p>Gating Gating
</p>
<p>GatingGating
</p>
<p>Gating GatingGating Gating
</p>
<p>Gating
</p>
<p>NetworkNetwork Network
</p>
<p>Network
</p>
<p>Network
</p>
<p>Network
</p>
<p>Network
</p>
<p>NetworkNetwork Network
</p>
<p>Network
</p>
<p>Network
</p>
<p>Network NetworkNetwork Network
</p>
<p>NetworkNetwork Network
</p>
<p>Network
</p>
<p>ExpertExpertExpert ExpertExpertExpert ExpertExpert ExpertExpertExpert
</p>
<p>Pr(y|x, θ11) Pr(y|x, θ21) Pr(y|x, θ12) Pr(y|x, θ22)
</p>
<p>FIGURE 9.13. A two-level hierarchical mixture of experts (HME) model.
</p>
<p>Consider the regression or classification problem, as described earlier in
the chapter. The data is (xi, yi), i = 1, 2, . . . , N , with yi either a continuous
or binary-valued response, and xi a vector-valued input. For ease of nota-
tion we assume that the first element of xi is one, to account for intercepts.
Here is how an HME is defined. The top gating network has the output
</p>
<p>gj(x, γj) =
eγ
</p>
<p>T
j x
</p>
<p>&sum;K
k=1 e
</p>
<p>γT
k
x
, j = 1, 2, . . . ,K, (9.25)
</p>
<p>where each γj is a vector of unknown parameters. This represents a soft
K-way split (K = 2 in Figure 9.13.) Each gj(x, γj) is the probability of
assigning an observation with feature vector x to the jth branch. Notice
that with K = 2 groups, if we take the coefficient of one of the elements of
x to be +&infin;, then we get a logistic curve with infinite slope. In this case,
the gating probabilities are either 0 or 1, corresponding to a hard split on
that input.
At the second level, the gating networks have a similar form:
</p>
<p>gℓ|j(x, γjℓ) =
eγ
</p>
<p>T
jℓx
</p>
<p>&sum;K
k=1 e
</p>
<p>γT
jk
</p>
<p>x
, ℓ = 1, 2, . . . ,K. (9.26)</p>
<p/>
</div>
<div class="page"><p/>
<p>9.5 Hierarchical Mixtures of Experts 331
</p>
<p>This is the probability of assignment to the ℓth branch, given assignment
to the jth branch at the level above.
At each expert (terminal node), we have a model for the response variable
</p>
<p>of the form
</p>
<p>Y &sim; Pr(y|x, θjℓ). (9.27)
</p>
<p>This differs according to the problem.
</p>
<p>Regression: The Gaussian linear regression model is used, with θjℓ =
(βjℓ, σ
</p>
<p>2
jℓ):
</p>
<p>Y = βTjℓx+ ε and ε &sim; N(0, σ2jℓ). (9.28)
</p>
<p>Classification: The linear logistic regression model is used:
</p>
<p>Pr(Y = 1|x, θjℓ) =
1
</p>
<p>1 + e&minus;θ
T
jℓ
x
. (9.29)
</p>
<p>Denoting the collection of all parameters by Ψ = {γj , γjℓ, θjℓ}, the total
probability that Y = y is
</p>
<p>Pr(y|x,Ψ) =
K&sum;
</p>
<p>j=1
</p>
<p>gj(x, γj)
</p>
<p>K&sum;
</p>
<p>ℓ=1
</p>
<p>gℓ|j(x, γjℓ)Pr(y|x, θjℓ). (9.30)
</p>
<p>This is a mixture model, with the mixture probabilities determined by the
gating network models.
To estimate the parameters, we maximize the log-likelihood of the data,&sum;
i log Pr(yi|xi,Ψ), over the parameters in Ψ. The most convenient method
</p>
<p>for doing this is the EM algorithm, which we describe for mixtures in
Section 8.5. We define latent variables ∆j , all of which are zero except for
a single one. We interpret these as the branching decisions made by the top
level gating network. Similarly we define latent variables ∆ℓ|j to describe
the gating decisions at the second level.
In the E-step, the EM algorithm computes the expectations of the ∆j
</p>
<p>and ∆ℓ|j given the current values of the parameters. These expectations
are then used as observation weights in the M-step of the procedure, to
estimate the parameters in the expert networks. The parameters in the
internal nodes are estimated by a version of multiple logistic regression.
The expectations of the ∆j or ∆ℓ|j are probability profiles, and these are
used as the response vectors for these logistic regressions.
The hierarchical mixtures of experts approach is a promising competitor
</p>
<p>to CART trees. By using soft splits rather than hard decision rules it can
capture situations where the transition from low to high response is gradual.
The log-likelihood is a smooth function of the unknown weights and hence
is amenable to numerical optimization. The model is similar to CART with
linear combination splits, but the latter is more difficult to optimize. On</p>
<p/>
</div>
<div class="page"><p/>
<p>332 9. Additive Models, Trees, and Related Methods
</p>
<p>the other hand, to our knowledge there are no methods for finding a good
tree topology for the HME model, as there are in CART. Typically one uses
a fixed tree of some depth, possibly the output of the CART procedure.
The emphasis in the research on HMEs has been on prediction rather than
interpretation of the final model. A close cousin of the HME is the latent
class model (Lin et al., 2000), which typically has only one layer; here
the nodes or latent classes are interpreted as groups of subjects that show
similar response behavior.
</p>
<p>9.6 Missing Data
</p>
<p>It is quite common to have observations with missing values for one or more
input features. The usual approach is to impute (fill-in) the missing values
in some way.
However, the first issue in dealing with the problem is determining wheth-
</p>
<p>er the missing data mechanism has distorted the observed data. Roughly
speaking, data are missing at random if the mechanism resulting in its
omission is independent of its (unobserved) value. A more precise definition
is given in Little and Rubin (2002). Suppose y is the response vector and X
is the N &times; p matrix of inputs (some of which are missing). Denote by Xobs
the observed entries in X and let Z = (y,X), Zobs = (y,Xobs). Finally, if R
is an indicator matrix with ijth entry 1 if xij is missing and zero otherwise,
then the data is said to be missing at random (MAR) if the distribution of
R depends on the data Z only through Zobs:
</p>
<p>Pr(R|Z, θ) = Pr(R|Zobs, θ). (9.31)
</p>
<p>Here θ are any parameters in the distribution of R. Data are said to be
missing completely at random (MCAR) if the distribution of R doesn&rsquo;t
depend on the observed or missing data:
</p>
<p>Pr(R|Z, θ) = Pr(R|θ). (9.32)
</p>
<p>MCAR is a stronger assumption than MAR: most imputation methods rely
on MCAR for their validity.
For example, if a patient&rsquo;s measurement was not taken because the doctor
</p>
<p>felt he was too sick, that observation would not be MAR or MCAR. In this
case the missing data mechanism causes our observed training data to give a
distorted picture of the true population, and data imputation is dangerous
in this instance. Often the determination of whether features are MCAR
must be made from information about the data collection process. For
categorical features, one way to diagnose this problem is to code &ldquo;missing&rdquo;
as an additional class. Then we fit our model to the training data and see
if class &ldquo;missing&rdquo; is predictive of the response.</p>
<p/>
</div>
<div class="page"><p/>
<p>9.6 Missing Data 333
</p>
<p>Assuming the features are missing completely at random, there are a
number of ways of proceeding:
</p>
<p>1. Discard observations with any missing values.
</p>
<p>2. Rely on the learning algorithm to deal with missing values in its
training phase.
</p>
<p>3. Impute all missing values before training.
</p>
<p>Approach (1) can be used if the relative amount of missing data is small,
but otherwise should be avoided. Regarding (2), CART is one learning
algorithm that deals effectively with missing values, through surrogate splits
(Section 9.2.4). MARS and PRIM use similar approaches. In generalized
additive modeling, all observations missing for a given input feature are
omitted when the partial residuals are smoothed against that feature in
the backfitting algorithm, and their fitted values are set to zero. Since the
fitted curves have mean zero (when the model includes an intercept), this
amounts to assigning the average fitted value to the missing observations.
For most learning methods, the imputation approach (3) is necessary.
</p>
<p>The simplest tactic is to impute the missing value with the mean or median
of the nonmissing values for that feature. (Note that the above procedure
for generalized additive models is analogous to this.)
If the features have at least some moderate degree of dependence, one
</p>
<p>can do better by estimating a predictive model for each feature given the
other features and then imputing each missing value by its prediction from
the model. In choosing the learning method for imputation of the features,
one must remember that this choice is distinct from the method used for
predicting y from X. Thus a flexible, adaptive method will often be pre-
ferred, even for the eventual purpose of carrying out a linear regression of y
on X. In addition, if there are many missing feature values in the training
set, the learning method must itself be able to deal with missing feature
values. CART therefore is an ideal choice for this imputation &ldquo;engine.&rdquo;
After imputation, missing values are typically treated as if they were ac-
</p>
<p>tually observed. This ignores the uncertainty due to the imputation, which
will itself introduce additional uncertainty into estimates and predictions
from the response model. One can measure this additional uncertainty by
doing multiple imputations and hence creating many different training sets.
The predictive model for y can be fit to each training set, and the variation
across training sets can be assessed. If CART was used for the imputation
engine, the multiple imputations could be done by sampling from the values
in the corresponding terminal nodes.</p>
<p/>
</div>
<div class="page"><p/>
<p>334 9. Additive Models, Trees, and Related Methods
</p>
<p>9.7 Computational Considerations
</p>
<p>With N observations and p predictors, additive model fitting requires some
number mp of applications of a one-dimensional smoother or regression
method. The required number of cycles m of the backfitting algorithm is
usually less than 20 and often less than 10, and depends on the amount
of correlation in the inputs. With cubic smoothing splines, for example,
N logN operations are needed for an initial sort and N operations for the
spline fit. Hence the total operations for an additive model fit is pN logN+
mpN .
Trees require pN logN operations for an initial sort for each predictor,
</p>
<p>and typically another pN logN operations for the split computations. If the
splits occurred near the edges of the predictor ranges, this number could
increase to N2p.
</p>
<p>MARS requires Nm2 + pmN operations to add a basis function to a
model with m terms already present, from a pool of p predictors. Hence to
build an M -term model requires NM3 + pM2N computations, which can
be quite prohibitive if M is a reasonable fraction of N .
Each of the components of an HME are typically inexpensive to fit at
</p>
<p>each M-step: Np2 for the regressions, and Np2K2 for a K-class logistic
regression. The EM algorithm, however, can take a long time to converge,
and so sizable HME models are considered costly to fit.
</p>
<p>Bibliographic Notes
</p>
<p>The most comprehensive source for generalized additive models is the text
of that name by Hastie and Tibshirani (1990). Different applications of
this work in medical problems are discussed in Hastie et al. (1989) and
Hastie and Herman (1990), and the software implementation in Splus is
described in Chambers and Hastie (1991). Green and Silverman (1994)
discuss penalization and spline models in a variety of settings. Efron and
Tibshirani (1991) give an exposition of modern developments in statistics
(including generalized additive models), for a nonmathematical audience.
Classification and regression trees date back at least as far as Morgan and
Sonquist (1963). We have followed the modern approaches of Breiman et
al. (1984) and Quinlan (1993). The PRIM method is due to Friedman
and Fisher (1999), while MARS is introduced in Friedman (1991), with an
additive precursor in Friedman and Silverman (1989). Hierarchical mixtures
of experts were proposed in Jordan and Jacobs (1994); see also Jacobs et
al. (1991).</p>
<p/>
</div>
<div class="page"><p/>
<p>Exercises 335
</p>
<p>Exercises
</p>
<p>Ex. 9.1 Show that a smoothing spline fit of yi to xi preserves the linear
part of the fit. In other words, if yi = ŷi + ri, where ŷi represents the
linear regression fits, and S is the smoothing matrix, then Sy = ŷ + Sr.
Show that the same is true for local linear regression (Section 6.1.1). Hence
argue that the adjustment step in the second line of (2) in Algorithm 9.1
is unnecessary.
</p>
<p>Ex. 9.2 Let A be a known k &times; k matrix, b be a known k-vector, and z
be an unknown k-vector. A Gauss&ndash;Seidel algorithm for solving the linear
system of equations Az = b works by successively solving for element zj in
the jth equation, fixing all other zj &rsquo;s at their current guesses. This process
is repeated for j = 1, 2, . . . , k, 1, 2, . . . , k, . . . , until convergence (Golub and
Van Loan, 1983).
</p>
<p>(a) Consider an additive model with N observations and p terms, with
the jth term to be fit by a linear smoother Sj . Consider the following
system of equations:
</p>
<p>⎛
⎜⎜⎜⎝
</p>
<p>I S1 S1 &middot; &middot; &middot; S1
S2 I S2 &middot; &middot; &middot; S2
...
</p>
<p>...
...
</p>
<p>. . .
...
</p>
<p>Sp Sp Sp &middot; &middot; &middot; I
</p>
<p>⎞
⎟⎟⎟⎠
</p>
<p>⎛
⎜⎜⎜⎝
</p>
<p>f1
f2
...
fp
</p>
<p>⎞
⎟⎟⎟⎠ =
</p>
<p>⎛
⎜⎜⎜⎝
</p>
<p>S1y
</p>
<p>S2y
...
</p>
<p>Spy
</p>
<p>⎞
⎟⎟⎟⎠ . (9.33)
</p>
<p>Here each fj is an N -vector of evaluations of the jth function at
the data points, and y is an N -vector of the response values. Show
that backfitting is a blockwise Gauss&ndash;Seidel algorithm for solving this
system of equations.
</p>
<p>(b) Let S1 and S2 be symmetric smoothing operators (matrices) with
eigenvalues in [0, 1). Consider a backfitting algorithm with response
vector y and smoothers S1,S2. Show that with any starting values,
the algorithm converges and give a formula for the final iterates.
</p>
<p>Ex. 9.3 Backfitting equations. Consider a backfitting procedure with orthog-
onal projections, and let D be the overall regression matrix whose columns
span V = Lcol(S1) &oplus; Lcol(S2) &oplus; &middot; &middot; &middot; &oplus; Lcol(Sp), where Lcol(S) denotes the
column space of a matrix S. Show that the estimating equations
</p>
<p>⎛
⎜⎜⎜⎝
</p>
<p>I S1 S1 &middot; &middot; &middot; S1
S2 I S2 &middot; &middot; &middot; S2
...
</p>
<p>...
...
</p>
<p>. . .
...
</p>
<p>Sp Sp Sp &middot; &middot; &middot; I
</p>
<p>⎞
⎟⎟⎟⎠
</p>
<p>⎛
⎜⎜⎜⎝
</p>
<p>f1
f2
...
fp
</p>
<p>⎞
⎟⎟⎟⎠ =
</p>
<p>⎛
⎜⎜⎜⎝
</p>
<p>S1y
</p>
<p>S2y
...
</p>
<p>Spy
</p>
<p>⎞
⎟⎟⎟⎠
</p>
<p>are equivalent to the least squares normal equations DTDβ = DTy where
β is the vector of coefficients.</p>
<p/>
</div>
<div class="page"><p/>
<p>336 9. Additive Models, Trees, and Related Methods
</p>
<p>Ex. 9.4 Suppose the same smoother S is used to estimate both terms in a
two-term additive model (i.e., both variables are identical). Assume that S
is symmetric with eigenvalues in [0, 1). Show that the backfitting residual
converges to (I+ S)&minus;1(I&minus; S)y, and that the residual sum of squares con-
verges upward. Can the residual sum of squares converge upward in less
structured situations? How does this fit compare to the fit with a single
term fit by S? [Hint: Use the eigen-decomposition of S to help with this
comparison.]
</p>
<p>Ex. 9.5 Degrees of freedom of a tree. Given data yi with mean f(xi) and
variance σ2, and a fitting operation y &rarr; ŷ, let&rsquo;s define the degrees of
freedom of a fit by
</p>
<p>&sum;
i cov(yi, ŷi)/σ
</p>
<p>2.
Consider a fit ŷ estimated by a regression tree, fit to a set of predictors
</p>
<p>X1, X2, . . . , Xp.
</p>
<p>(a) In terms of the number of terminal nodes m, give a rough formula for
the degrees of freedom of the fit.
</p>
<p>(b) Generate 100 observations with predictors X1, X2, . . . , X10 as inde-
pendent standard Gaussian variates and fix these values.
</p>
<p>(c) Generate response values also as standard Gaussian (σ2 = 1), indepen-
dent of the predictors. Fit regression trees to the data of fixed size 1,5
and 10 terminal nodes and hence estimate the degrees of freedom of
each fit. [Do ten simulations of the response and average the results,
to get a good estimate of degrees of freedom.]
</p>
<p>(d) Compare your estimates of degrees of freedom in (a) and (c) and
discuss.
</p>
<p>(e) If the regression tree fit were a linear operation, we could write ŷ = Sy
for some matrix S. Then the degrees of freedom would be tr(S).
Suggest a way to compute an approximate S matrix for a regression
tree, compute it and compare the resulting degrees of freedom to
those in (a) and (c).
</p>
<p>Ex. 9.6 Consider the ozone data of Figure 6.9.
</p>
<p>(a) Fit an additive model to the cube root of ozone concentration. as a
function of temperature, wind speed, and radiation. Compare your
results to those obtained via the trellis display in Figure 6.9.
</p>
<p>(b) Fit trees, MARS, and PRIM to the same data, and compare the results
to those found in (a) and in Figure 6.9.</p>
<p/>
</div>
<div class="page"><p/>
<p>10
Boosting and Additive Trees
</p>
<p>10.1 Boosting Methods
</p>
<p>Boosting is one of the most powerful learning ideas introduced in the last
twenty years. It was originally designed for classification problems, but as
will be seen in this chapter, it can profitably be extended to regression
as well. The motivation for boosting was a procedure that combines the
outputs of many &ldquo;weak&rdquo; classifiers to produce a powerful &ldquo;committee.&rdquo;
From this perspective boosting bears a resemblance to bagging and other
committee-based approaches (Section 8.8). However we shall see that the
connection is at best superficial and that boosting is fundamentally differ-
ent.
We begin by describing the most popular boosting algorithm due to
</p>
<p>Freund and Schapire (1997) called &ldquo;AdaBoost.M1.&rdquo; Consider a two-class
problem, with the output variable coded as Y &isin; {&minus;1, 1}. Given a vector of
predictor variables X, a classifier G(X) produces a prediction taking one
of the two values {&minus;1, 1}. The error rate on the training sample is
</p>
<p>err =
1
</p>
<p>N
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>I(yi �= G(xi)),
</p>
<p>and the expected error rate on future predictions is EXY I(Y �= G(X)).
A weak classifier is one whose error rate is only slightly better than
</p>
<p>random guessing. The purpose of boosting is to sequentially apply the
weak classification algorithm to repeatedly modified versions of the data,
thereby producing a sequence of weak classifiers Gm(x),m = 1, 2, . . . ,M .
</p>
<p> &copy; Springer Science+Business Media, LLC 2009 
</p>
<p>T. Hastie et al., The Elements of Statistical Learning, Second Edition, 337
</p>
<p>DOI: 10.1007/b94608_10,</p>
<p/>
</div>
<div class="page"><p/>
<p>338 10. Boosting and Additive Trees
</p>
<p>Training  Sample
</p>
<p>Weighted  Sample
</p>
<p>Weighted  Sample
</p>
<p>Weighted  Sample
</p>
<p>Training  Sample
</p>
<p>Weighted  Sample
</p>
<p>Weighted  Sample
</p>
<p>Weighted  SampleWeighted  Sample
</p>
<p>Training  Sample
</p>
<p>Weighted  Sample
</p>
<p>Training  Sample
</p>
<p>Weighted  Sample
</p>
<p>Weighted  SampleWeighted  Sample
</p>
<p>Weighted  Sample
</p>
<p>Weighted  Sample
</p>
<p>Weighted  Sample
</p>
<p>Training  Sample
</p>
<p>Weighted  Sample
</p>
<p>G(x) = sign
[&sum;M
</p>
<p>m=1 αmGm(x)
]
</p>
<p>GM (x)
</p>
<p>G3(x)
</p>
<p>G2(x)
</p>
<p>G1(x)
</p>
<p>Final Classifier
</p>
<p>FIGURE 10.1. Schematic of AdaBoost. Classifiers are trained on weighted ver-
sions of the dataset, and then combined to produce a final prediction.
</p>
<p>The predictions from all of them are then combined through a weighted
majority vote to produce the final prediction:
</p>
<p>G(x) = sign
</p>
<p>(
M&sum;
</p>
<p>m=1
</p>
<p>αmGm(x)
</p>
<p>)
. (10.1)
</p>
<p>Here α1, α2, . . . , αM are computed by the boosting algorithm, and weight
the contribution of each respective Gm(x). Their effect is to give higher
influence to the more accurate classifiers in the sequence. Figure 10.1 shows
a schematic of the AdaBoost procedure.
The data modifications at each boosting step consist of applying weights
</p>
<p>w1, w2, . . . , wN to each of the training observations (xi, yi), i = 1, 2, . . . , N .
Initially all of the weights are set to wi = 1/N , so that the first step simply
trains the classifier on the data in the usual manner. For each successive
iteration m = 2, 3, . . . ,M the observation weights are individually modi-
fied and the classification algorithm is reapplied to the weighted observa-
tions. At step m, those observations that were misclassified by the classifier
Gm&minus;1(x) induced at the previous step have their weights increased, whereas
the weights are decreased for those that were classified correctly. Thus as
iterations proceed, observations that are difficult to classify correctly re-
ceive ever-increasing influence. Each successive classifier is thereby forced</p>
<p/>
</div>
<div class="page"><p/>
<p>10.1 Boosting Methods 339
</p>
<p>Algorithm 10.1 AdaBoost.M1.
</p>
<p>1. Initialize the observation weights wi = 1/N, i = 1, 2, . . . , N .
</p>
<p>2. For m = 1 to M :
</p>
<p>(a) Fit a classifier Gm(x) to the training data using weights wi.
</p>
<p>(b) Compute
</p>
<p>errm =
</p>
<p>&sum;N
i=1 wiI(yi �= Gm(xi))&sum;N
</p>
<p>i=1 wi
.
</p>
<p>(c) Compute αm = log((1&minus; errm)/errm).
(d) Set wi &larr; wi &middot; exp[αm &middot; I(yi �= Gm(xi))], i = 1, 2, . . . , N .
</p>
<p>3. Output G(x) = sign
[&sum;M
</p>
<p>m=1 αmGm(x)
]
.
</p>
<p>to concentrate on those training observations that are missed by previous
ones in the sequence.
Algorithm 10.1 shows the details of the AdaBoost.M1 algorithm. The
</p>
<p>current classifier Gm(x) is induced on the weighted observations at line 2a.
The resulting weighted error rate is computed at line 2b. Line 2c calculates
the weight αm given to Gm(x) in producing the final classifier G(x) (line
3). The individual weights of each of the observations are updated for the
next iteration at line 2d. Observations misclassified by Gm(x) have their
weights scaled by a factor exp(αm), increasing their relative influence for
inducing the next classifier Gm+1(x) in the sequence.
</p>
<p>The AdaBoost.M1 algorithm is known as &ldquo;Discrete AdaBoost&rdquo; in Fried-
man et al. (2000), because the base classifier Gm(x) returns a discrete class
label. If the base classifier instead returns a real-valued prediction (e.g.,
a probability mapped to the interval [&minus;1, 1]), AdaBoost can be modified
appropriately (see &ldquo;Real AdaBoost&rdquo; in Friedman et al. (2000)).
The power of AdaBoost to dramatically increase the performance of even
</p>
<p>a very weak classifier is illustrated in Figure 10.2. The features X1, . . . , X10
are standard independent Gaussian, and the deterministic target Y is de-
fined by
</p>
<p>Y =
</p>
<p>{
1 if
</p>
<p>&sum;10
j=1 X
</p>
<p>2
j &gt; χ
</p>
<p>2
10(0.5),
</p>
<p>&minus;1 otherwise. (10.2)
</p>
<p>Here χ210(0.5) = 9.34 is the median of a chi-squared random variable with
10 degrees of freedom (sum of squares of 10 standard Gaussians). There are
2000 training cases, with approximately 1000 cases in each class, and 10,000
test observations. Here the weak classifier is just a &ldquo;stump&rdquo;: a two terminal-
node classification tree. Applying this classifier alone to the training data
set yields a very poor test set error rate of 45.8%, compared to 50% for</p>
<p/>
</div>
<div class="page"><p/>
<p>340 10. Boosting and Additive Trees
</p>
<p>0 100 200 300 400
</p>
<p>0
.0
</p>
<p>0
.1
</p>
<p>0
.2
</p>
<p>0
.3
</p>
<p>0
.4
</p>
<p>0
.5
</p>
<p>Boosting Iterations
</p>
<p>T
e
s
t 
E
</p>
<p>rr
o
r
</p>
<p>Single Stump
</p>
<p>244 Node Tree
</p>
<p>FIGURE 10.2. Simulated data (10.2): test error rate for boosting with stumps,
as a function of the number of iterations. Also shown are the test error rate for
a single stump, and a 244-node classification tree.
</p>
<p>random guessing. However, as boosting iterations proceed the error rate
steadily decreases, reaching 5.8% after 400 iterations. Thus, boosting this
simple very weak classifier reduces its prediction error rate by almost a
factor of four. It also outperforms a single large classification tree (error
rate 24.7%). Since its introduction, much has been written to explain the
success of AdaBoost in producing accurate classifiers. Most of this work
has centered on using classification trees as the &ldquo;base learner&rdquo; G(x), where
improvements are often most dramatic. In fact, Breiman (NIPS Workshop,
1996) referred to AdaBoost with trees as the &ldquo;best off-the-shelf classifier in
the world&rdquo; (see also Breiman (1998)). This is especially the case for data-
mining applications, as discussed more fully in Section 10.7 later in this
chapter.
</p>
<p>10.1.1 Outline of This Chapter
</p>
<p>Here is an outline of the developments in this chapter:
</p>
<p>&bull; We show that AdaBoost fits an additive model in a base learner,
optimizing a novel exponential loss function. This loss function is</p>
<p/>
</div>
<div class="page"><p/>
<p>10.2 Boosting Fits an Additive Model 341
</p>
<p>very similar to the (negative) binomial log-likelihood (Sections 10.2&ndash;
10.4).
</p>
<p>&bull; The population minimizer of the exponential loss function is shown
to be the log-odds of the class probabilities (Section 10.5).
</p>
<p>&bull; We describe loss functions for regression and classification that are
more robust than squared error or exponential loss (Section 10.6).
</p>
<p>&bull; It is argued that decision trees are an ideal base learner for data
mining applications of boosting (Sections 10.7 and 10.9).
</p>
<p>&bull; We develop a class of gradient boosted models (GBMs), for boosting
trees with any loss function (Section 10.10).
</p>
<p>&bull; The importance of &ldquo;slow learning&rdquo; is emphasized, and implemented
by shrinkage of each new term that enters the model (Section 10.12),
as well as randomization (Section 10.12.2).
</p>
<p>&bull; Tools for interpretation of the fitted model are described (Section 10.13).
</p>
<p>10.2 Boosting Fits an Additive Model
</p>
<p>The success of boosting is really not very mysterious. The key lies in ex-
pression (10.1). Boosting is a way of fitting an additive expansion in a set
of elementary &ldquo;basis&rdquo; functions. Here the basis functions are the individual
classifiers Gm(x) &isin; {&minus;1, 1}. More generally, basis function expansions take
the form
</p>
<p>f(x) =
</p>
<p>M&sum;
</p>
<p>m=1
</p>
<p>βmb(x; γm), (10.3)
</p>
<p>where βm,m = 1, 2, . . . ,M are the expansion coefficients, and b(x; γ) &isin; IR
are usually simple functions of the multivariate argument x, characterized
by a set of parameters γ. We discuss basis expansions in some detail in
Chapter 5.
Additive expansions like this are at the heart of many of the learning
</p>
<p>techniques covered in this book:
</p>
<p>&bull; In single-hidden-layer neural networks (Chapter 11), b(x; γ) = σ(γ0+
γT1 x), where σ(t) = 1/(1+e
</p>
<p>&minus;t) is the sigmoid function, and γ param-
eterizes a linear combination of the input variables.
</p>
<p>&bull; In signal processing, wavelets (Section 5.9.1) are a popular choice with
γ parameterizing the location and scale shifts of a &ldquo;mother&rdquo; wavelet.
</p>
<p>&bull; Multivariate adaptive regression splines (Section 9.4) uses truncated-
power spline basis functions where γ parameterizes the variables and
values for the knots.</p>
<p/>
</div>
<div class="page"><p/>
<p>342 10. Boosting and Additive Trees
</p>
<p>Algorithm 10.2 Forward Stagewise Additive Modeling.
</p>
<p>1. Initialize f0(x) = 0.
</p>
<p>2. For m = 1 to M :
</p>
<p>(a) Compute
</p>
<p>(βm, γm) = argmin
β,γ
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>L(yi, fm&minus;1(xi) + βb(xi; γ)).
</p>
<p>(b) Set fm(x) = fm&minus;1(x) + βmb(x; γm).
</p>
<p>&bull; For trees, γ parameterizes the split variables and split points at the
internal nodes, and the predictions at the terminal nodes.
</p>
<p>Typically these models are fit by minimizing a loss function averaged
over the training data, such as the squared-error or a likelihood-based loss
function,
</p>
<p>min
{βm,γm}M1
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>L
</p>
<p>(
yi,
</p>
<p>M&sum;
</p>
<p>m=1
</p>
<p>βmb(xi; γm)
</p>
<p>)
. (10.4)
</p>
<p>For many loss functions L(y, f(x)) and/or basis functions b(x; γ), this re-
quires computationally intensive numerical optimization techniques. How-
ever, a simple alternative often can be found when it is feasible to rapidly
solve the subproblem of fitting just a single basis function,
</p>
<p>min
β,γ
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>L (yi, βb(xi; γ)) . (10.5)
</p>
<p>10.3 Forward Stagewise Additive Modeling
</p>
<p>Forward stagewise modeling approximates the solution to (10.4) by sequen-
tially adding new basis functions to the expansion without adjusting the
parameters and coefficients of those that have already been added. This is
outlined in Algorithm 10.2. At each iteration m, one solves for the optimal
basis function b(x; γm) and corresponding coefficient βm to add to the cur-
rent expansion fm&minus;1(x). This produces fm(x), and the process is repeated.
Previously added terms are not modified.
For squared-error loss
</p>
<p>L(y, f(x)) = (y &minus; f(x))2, (10.6)</p>
<p/>
</div>
<div class="page"><p/>
<p>10.4 Exponential Loss and AdaBoost 343
</p>
<p>one has
</p>
<p>L(yi, fm&minus;1(xi) + βb(xi; γ)) = (yi &minus; fm&minus;1(xi)&minus; βb(xi; γ))2
= (rim &minus; βb(xi; γ))2, (10.7)
</p>
<p>where rim = yi &minus; fm&minus;1(xi) is simply the residual of the current model
on the ith observation. Thus, for squared-error loss, the term βmb(x; γm)
that best fits the current residuals is added to the expansion at each step.
This idea is the basis for &ldquo;least squares&rdquo; regression boosting discussed in
Section 10.10.2. However, as we show near the end of the next section,
squared-error loss is generally not a good choice for classification; hence
the need to consider other loss criteria.
</p>
<p>10.4 Exponential Loss and AdaBoost
</p>
<p>We now show that AdaBoost.M1 (Algorithm 10.1) is equivalent to forward
stagewise additive modeling (Algorithm 10.2) using the loss function
</p>
<p>L(y, f(x)) = exp(&minus;y f(x)). (10.8)
</p>
<p>The appropriateness of this criterion is addressed in the next section.
For AdaBoost the basis functions are the individual classifiers Gm(x) &isin;
</p>
<p>{&minus;1, 1}. Using the exponential loss function, one must solve
</p>
<p>(βm, Gm) = argmin
β,G
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>exp[&minus;yi(fm&minus;1(xi) + β G(xi))]
</p>
<p>for the classifier Gm and corresponding coefficient βm to be added at each
step. This can be expressed as
</p>
<p>(βm, Gm) = argmin
β,G
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>w
(m)
i exp(&minus;β yi G(xi)) (10.9)
</p>
<p>with w
(m)
i = exp(&minus;yi fm&minus;1(xi)). Since each w
</p>
<p>(m)
i depends neither on β
</p>
<p>nor G(x), it can be regarded as a weight that is applied to each observa-
tion. This weight depends on fm&minus;1(xi), and so the individual weight values
change with each iteration m.
</p>
<p>The solution to (10.9) can be obtained in two steps. First, for any value
of β &gt; 0, the solution to (10.9) for Gm(x) is
</p>
<p>Gm = argmin
G
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>w
(m)
i I(yi �= G(xi)), (10.10)</p>
<p/>
</div>
<div class="page"><p/>
<p>344 10. Boosting and Additive Trees
</p>
<p>which is the classifier that minimizes the weighted error rate in predicting
y. This can be easily seen by expressing the criterion in (10.9) as
</p>
<p>e&minus;β &middot;
&sum;
</p>
<p>yi=G(xi)
</p>
<p>w
(m)
i + e
</p>
<p>β &middot;
&sum;
</p>
<p>yi �=G(xi)
</p>
<p>w
(m)
i ,
</p>
<p>which in turn can be written as
</p>
<p>(
eβ &minus; e&minus;β
</p>
<p>)
&middot;
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>w
(m)
i I(yi �= G(xi)) + e&minus;β &middot;
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>w
(m)
i . (10.11)
</p>
<p>Plugging this Gm into (10.9) and solving for β one obtains
</p>
<p>βm =
1
</p>
<p>2
log
</p>
<p>1&minus; errm
errm
</p>
<p>, (10.12)
</p>
<p>where errm is the minimized weighted error rate
</p>
<p>errm =
</p>
<p>&sum;N
i=1 w
</p>
<p>(m)
i I(yi �= Gm(xi))&sum;N
</p>
<p>i=1 w
(m)
i
</p>
<p>. (10.13)
</p>
<p>The approximation is then updated
</p>
<p>fm(x) = fm&minus;1(x) + βmGm(x),
</p>
<p>which causes the weights for the next iteration to be
</p>
<p>w
(m+1)
i = w
</p>
<p>(m)
i &middot; e&minus;βmyiGm(xi). (10.14)
</p>
<p>Using the fact that &minus;yiGm(xi) = 2 &middot; I(yi �= Gm(xi))&minus; 1, (10.14) becomes
</p>
<p>w
(m+1)
i = w
</p>
<p>(m)
i &middot; eαmI(yi �=Gm(xi)) &middot; e&minus;βm , (10.15)
</p>
<p>where αm = 2βm is the quantity defined at line 2(c) of AdaBoost.M1
(Algorithm 10.1). The factor e&minus;βm in (10.15) multiplies all weights by the
same value, so it has no effect. Thus (10.15) is equivalent to line 2(d) of
Algorithm 10.1.
One can view line 2(a) of the Adaboost.M1 algorithm as a method for
</p>
<p>approximately solving the minimization in (10.11) and hence (10.10). Hence
we conclude that AdaBoost.M1 minimizes the exponential loss criterion
(10.8) via a forward-stagewise additive modeling approach.
Figure 10.3 shows the training-set misclassification error rate and aver-
</p>
<p>age exponential loss for the simulated data problem (10.2) of Figure 10.2.
The training-set misclassification error decreases to zero at around 250 it-
erations (and remains there), but the exponential loss keeps decreasing.
Notice also in Figure 10.2 that the test-set misclassification error continues
to improve after iteration 250. Clearly Adaboost is not optimizing training-
set misclassification error; the exponential loss is more sensitive to changes
in the estimated class probabilities.</p>
<p/>
</div>
<div class="page"><p/>
<p>10.5 Why Exponential Loss? 345
</p>
<p>0 100 200 300 400
</p>
<p>0
.0
</p>
<p>0
.2
</p>
<p>0
.4
</p>
<p>0
.6
</p>
<p>0
.8
</p>
<p>1
.0
</p>
<p>Boosting Iterations
</p>
<p>T
ra
</p>
<p>in
in
</p>
<p>g
 E
</p>
<p>rr
o
r
</p>
<p>Misclassification Rate
</p>
<p>Exponential Loss
</p>
<p>FIGURE 10.3. Simulated data, boosting with stumps: misclassification error
rate on the training set, and average exponential loss: (1/N)
</p>
<p>&sum;N
</p>
<p>i=1 exp(&minus;yif(xi)).
After about 250 iterations, the misclassification error is zero, while the exponential
loss continues to decrease.
</p>
<p>10.5 Why Exponential Loss?
</p>
<p>The AdaBoost.M1 algorithm was originally motivated from a very differ-
ent perspective than presented in the previous section. Its equivalence to
forward stagewise additive modeling based on exponential loss was only
discovered five years after its inception. By studying the properties of the
exponential loss criterion, one can gain insight into the procedure and dis-
cover ways it might be improved.
The principal attraction of exponential loss in the context of additive
</p>
<p>modeling is computational; it leads to the simple modular reweighting Ad-
aBoost algorithm. However, it is of interest to inquire about its statistical
properties. What does it estimate and how well is it being estimated? The
first question is answered by seeking its population minimizer.
It is easy to show (Friedman et al., 2000) that
</p>
<p>f&lowast;(x) = argmin
f(x)
</p>
<p>EY |x(e
&minus;Y f(x)) =
</p>
<p>1
</p>
<p>2
log
</p>
<p>Pr(Y = 1|x)
Pr(Y = &minus;1|x) , (10.16)</p>
<p/>
</div>
<div class="page"><p/>
<p>346 10. Boosting and Additive Trees
</p>
<p>or equivalently
</p>
<p>Pr(Y = 1|x) = 1
1 + e&minus;2f&lowast;(x)
</p>
<p>.
</p>
<p>Thus, the additive expansion produced by AdaBoost is estimating one-
half the log-odds of P (Y = 1|x). This justifies using its sign as the classifi-
cation rule in (10.1).
Another loss criterion with the same population minimizer is the bi-
</p>
<p>nomial negative log-likelihood or deviance (also known as cross-entropy),
interpreting f as the logit transform. Let
</p>
<p>p(x) = Pr(Y = 1 |x) = e
f(x)
</p>
<p>e&minus;f(x) + ef(x)
=
</p>
<p>1
</p>
<p>1 + e&minus;2f(x)
(10.17)
</p>
<p>and define Y &prime; = (Y + 1)/2 &isin; {0, 1}. Then the binomial log-likelihood loss
function is
</p>
<p>l(Y, p(x)) = Y &prime; log p(x) + (1&minus; Y &prime;) log(1&minus; p(x)),
</p>
<p>or equivalently the deviance is
</p>
<p>&minus;l(Y, f(x)) = log
(
1 + e&minus;2Y f(x)
</p>
<p>)
. (10.18)
</p>
<p>Since the population maximizer of log-likelihood is at the true probabilities
p(x) = Pr(Y = 1 |x), we see from (10.17) that the population minimizers of
the deviance EY |x[&minus;l(Y, f(x))] and EY |x[e&minus;Y f(x)] are the same. Thus, using
either criterion leads to the same solution at the population level. Note that
e&minus;Y f itself is not a proper log-likelihood, since it is not the logarithm of
any probability mass function for a binary random variable Y &isin; {&minus;1, 1}.
</p>
<p>10.6 Loss Functions and Robustness
</p>
<p>In this section we examine the different loss functions for classification and
regression more closely, and characterize them in terms of their robustness
to extreme data.
</p>
<p>Robust Loss Functions for Classification
</p>
<p>Although both the exponential (10.8) and binomial deviance (10.18) yield
the same solution when applied to the population joint distribution, the
same is not true for finite data sets. Both criteria are monotone decreasing
functions of the &ldquo;margin&rdquo; yf(x). In classification (with a &minus;1/1 response)
the margin plays a role analogous to the residuals y&minus;f(x) in regression. The
classification rule G(x) = sign[f(x)] implies that observations with positive
margin yif(xi) &gt; 0 are classified correctly whereas those with negative
margin yif(xi) &lt; 0 are misclassified. The decision boundary is defined by</p>
<p/>
</div>
<div class="page"><p/>
<p>10.6 Loss Functions and Robustness 347
</p>
<p>&minus;2 &minus;1 0 1 2
</p>
<p>0
.0
</p>
<p>0
.5
</p>
<p>1
.0
</p>
<p>1
.5
</p>
<p>2
.0
</p>
<p>2
.5
</p>
<p>3
.0 Misclassification
</p>
<p>Exponential
Binomial Deviance
Squared Error
Support Vector
</p>
<p>L
o
ss
</p>
<p>yf
</p>
<p>FIGURE 10.4. Loss functions for two-class classification. The response is
y = &plusmn;1; the prediction is f , with class prediction sign(f). The losses are
misclassification: I(sign(f) 	= y); exponential: exp(&minus;yf); binomial deviance:
log(1 + exp(&minus;2yf)); squared error: (y &minus; f)2; and support vector: (1 &minus; yf)+ (see
Section 12.3). Each function has been scaled so that it passes through the point
(0, 1).
</p>
<p>f(x) = 0. The goal of the classification algorithm is to produce positive
margins as frequently as possible. Any loss criterion used for classification
should penalize negative margins more heavily than positive ones since
positive margin observations are already correctly classified.
Figure 10.4 shows both the exponential (10.8) and binomial deviance
</p>
<p>criteria as a function of the margin yf(x). Also shown is misclassification
loss L(y, f(x)) = I(yf(x) &lt; 0), which gives unit penalty for negative mar-
gin values, and no penalty at all for positive ones. Both the exponential
and deviance loss can be viewed as monotone continuous approximations
to misclassification loss. They continuously penalize increasingly negative
margin values more heavily than they reward increasingly positive ones.
The difference between them is in degree. The penalty associated with bi-
nomial deviance increases linearly for large increasingly negative margin,
whereas the exponential criterion increases the influence of such observa-
tions exponentially.
At any point in the training process the exponential criterion concen-
</p>
<p>trates much more influence on observations with large negative margins.
Binomial deviance concentrates relatively less influence on such observa-</p>
<p/>
</div>
<div class="page"><p/>
<p>348 10. Boosting and Additive Trees
</p>
<p>tions, more evenly spreading the influence among all of the data. It is
therefore far more robust in noisy settings where the Bayes error rate is
not close to zero, and especially in situations where there is misspecification
of the class labels in the training data. The performance of AdaBoost has
been empirically observed to dramatically degrade in such situations.
Also shown in the figure is squared-error loss. The minimizer of the cor-
</p>
<p>responding risk on the population is
</p>
<p>f&lowast;(x) = argmin
f(x)
</p>
<p>EY |x(Y &minus;f(x))2 = E(Y |x) = 2&middot;Pr(Y = 1 |x)&minus;1. (10.19)
</p>
<p>As before the classification rule is G(x) = sign[f(x)]. Squared-error loss
is not a good surrogate for misclassification error. As seen in Figure 10.4, it
is not a monotone decreasing function of increasing margin yf(x). For mar-
gin values yif(xi) &gt; 1 it increases quadratically, thereby placing increasing
influence (error) on observations that are correctly classified with increas-
ing certainty, thereby reducing the relative influence of those incorrectly
classified yif(xi) &lt; 0. Thus, if class assignment is the goal, a monotone de-
creasing criterion serves as a better surrogate loss function. Figure 12.4 on
page 426 in Chapter 12 includes a modification of quadratic loss, the &ldquo;Hu-
berized&rdquo; square hinge loss (Rosset et al., 2004b), which enjoys the favorable
properties of the binomial deviance, quadratic loss and the SVM hinge loss.
It has the same population minimizer as the quadratic (10.19), is zero for
yf(x) &gt; 1, and becomes linear for yf(x) &lt; &minus;1. Since quadratic functions
are easier to compute with than exponentials, our experience suggests this
to be a useful alternative to the binomial deviance.
With K-class classification, the response Y takes values in the unordered
</p>
<p>set G = {G1, . . . ,Gk} (see Sections 2.4 and 4.4). We now seek a classifier
G(x) taking values in G. It is sufficient to know the class conditional proba-
bilities pk(x) = Pr(Y = Gk|x), k = 1, 2, . . . ,K, for then the Bayes classifier
is
</p>
<p>G(x) = Gk where k = argmax
ℓ
</p>
<p>pℓ(x). (10.20)
</p>
<p>In principal, though, we need not learn the pk(x), but simply which one is
largest. However, in data mining applications the interest is often more in
the class probabilities pℓ(x), ℓ = 1, . . . ,K themselves, rather than in per-
forming a class assignment. As in Section 4.4, the logistic model generalizes
naturally to K classes,
</p>
<p>pk(x) =
efk(x)
</p>
<p>&sum;K
l=1 e
</p>
<p>fl(x)
, (10.21)
</p>
<p>which ensures that 0 &le; pk(x) &le; 1 and that they sum to one. Note that
here we have K different functions, one per class. There is a redundancy
in the functions fk(x), since adding an arbitrary h(x) to each leaves the
model unchanged. Traditionally one of them is set to zero: for example,</p>
<p/>
</div>
<div class="page"><p/>
<p>10.6 Loss Functions and Robustness 349
</p>
<p>fK(x) = 0, as in (4.17). Here we prefer to retain the symmetry, and impose
</p>
<p>the constraint
&sum;K
</p>
<p>k=1 fk(x) = 0. The binomial deviance extends naturally
to the K-class multinomial deviance loss function:
</p>
<p>L(y, p(x)) = &minus;
K&sum;
</p>
<p>k=1
</p>
<p>I(y = Gk) log pk(x)
</p>
<p>= &minus;
K&sum;
</p>
<p>k=1
</p>
<p>I(y = Gk)fk(x) + log
(
</p>
<p>K&sum;
</p>
<p>ℓ=1
</p>
<p>efℓ(x)
</p>
<p>)
. (10.22)
</p>
<p>As in the two-class case, the criterion (10.22) penalizes incorrect predictions
only linearly in their degree of incorrectness.
Zhu et al. (2005) generalize the exponential loss for K-class problems.
</p>
<p>See Exercise 10.5 for details.
</p>
<p>Robust Loss Functions for Regression
</p>
<p>In the regression setting, analogous to the relationship between exponential
loss and binomial log-likelihood is the relationship between squared-error
loss L(y, f(x)) = (y&minus;f(x))2 and absolute loss L(y, f(x)) = | y&minus;f(x) |. The
population solutions are f(x) = E(Y |x) for squared-error loss, and f(x) =
median(Y |x) for absolute loss; for symmetric error distributions these are
the same. However, on finite samples squared-error loss places much more
emphasis on observations with large absolute residuals | yi &minus; f(xi) | during
the fitting process. It is thus far less robust, and its performance severely
degrades for long-tailed error distributions and especially for grossly mis-
measured y-values (&ldquo;outliers&rdquo;). Other more robust criteria, such as abso-
lute loss, perform much better in these situations. In the statistical ro-
bustness literature, a variety of regression loss criteria have been proposed
that provide strong resistance (if not absolute immunity) to gross outliers
while being nearly as efficient as least squares for Gaussian errors. They
are often better than either for error distributions with moderately heavy
tails. One such criterion is the Huber loss criterion used for M-regression
(Huber, 1964)
</p>
<p>L(y, f(x)) =
</p>
<p>{
[y &minus; f(x)]2 for | y &minus; f(x) | &le; δ,
</p>
<p>2δ| y &minus; f(x) | &minus; δ2 otherwise. (10.23)
</p>
<p>Figure 10.5 compares these three loss functions.
These considerations suggest that when robustness is a concern, as is
</p>
<p>especially the case in data mining applications (see Section 10.7), squared-
error loss for regression and exponential loss for classification are not the
best criteria from a statistical perspective. However, they both lead to the
elegant modular boosting algorithms in the context of forward stagewise
additive modeling. For squared-error loss one simply fits the base learner
to the residuals from the current model yi &minus; fm&minus;1(xi) at each step. For</p>
<p/>
</div>
<div class="page"><p/>
<p>350 10. Boosting and Additive Trees
</p>
<p>&minus;3 &minus;2 &minus;1 0 1 2 3
</p>
<p>0
2
</p>
<p>4
6
</p>
<p>8
</p>
<p>Squared Error
Absolute Error
Huber
</p>
<p>L
o
ss
</p>
<p>y &minus; f
</p>
<p>FIGURE 10.5. A comparison of three loss functions for regression, plotted as a
function of the margin y&minus;f . The Huber loss function combines the good properties
of squared-error loss near zero and absolute error loss when |y &minus; f | is large.
</p>
<p>exponential loss one performs a weighted fit of the base learner to the
output values yi, with weights wi = exp(&minus;yifm&minus;1(xi)). Using other more
robust criteria directly in their place does not give rise to such simple
feasible boosting algorithms. However, in Section 10.10.2 we show how one
can derive simple elegant boosting algorithms based on any differentiable
loss criterion, thereby producing highly robust boosting procedures for data
mining.
</p>
<p>10.7 &ldquo;Off-the-Shelf&rdquo; Procedures for Data Mining
</p>
<p>Predictive learning is an important aspect of data mining. As can be seen
from this book, a wide variety of methods have been developed for predic-
tive learning from data. For each particular method there are situations
for which it is particularly well suited, and others where it performs badly
compared to the best that can be done with that data. We have attempted
to characterize appropriate situations in our discussions of each of the re-
spective methods. However, it is seldom known in advance which procedure
will perform best or even well for any given problem. Table 10.1 summarizes
some of the characteristics of a number of learning methods.
Industrial and commercial data mining applications tend to be especially
</p>
<p>challenging in terms of the requirements placed on learning procedures.
Data sets are often very large in terms of number of observations and
number of variables measured on each of them. Thus, computational con-</p>
<p/>
</div>
<div class="page"><p/>
<p>10.7 &ldquo;Off-the-Shelf&rdquo; Procedures for Data Mining 351
</p>
<p>TABLE 10.1. Some characteristics of different learning methods. Key: ▲= good,
◆=fair, and ▼=poor.
</p>
<p>Characteristic Neural SVM Trees MARS k-NN,
</p>
<p>Nets Kernels
</p>
<p>Natural handling of data
of &ldquo;mixed&rdquo; type
</p>
<p>▼ ▼ ▲ ▲ ▼
</p>
<p>Handling of missing values ▼ ▼ ▲ ▲ ▲
</p>
<p>Robustness to outliers in
input space
</p>
<p>▼ ▼ ▲ ▼ ▲
</p>
<p>Insensitive to monotone
transformations of inputs
</p>
<p>▼ ▼ ▲ ▼ ▼
</p>
<p>Computational scalability
(large N)
</p>
<p>▼ ▼ ▲ ▲ ▼
</p>
<p>Ability to deal with irrel-
evant inputs
</p>
<p>▼ ▼ ▲ ▲ ▼
</p>
<p>Ability to extract linear
combinations of features
</p>
<p>▲ ▲ ▼ ▼ ◆
</p>
<p>Interpretability ▼ ▼ ◆ ▲ ▼
</p>
<p>Predictive power ▲ ▲ ▼ ◆ ▲
</p>
<p>siderations play an important role. Also, the data are usually messy: the
inputs tend to be mixtures of quantitative, binary, and categorical vari-
ables, the latter often with many levels. There are generally many missing
values, complete observations being rare. Distributions of numeric predic-
tor and response variables are often long-tailed and highly skewed. This
is the case for the spam data (Section 9.1.2); when fitting a generalized
additive model, we first log-transformed each of the predictors in order to
get a reasonable fit. In addition they usually contain a substantial fraction
of gross mis-measurements (outliers). The predictor variables are generally
measured on very different scales.
In data mining applications, usually only a small fraction of the large
</p>
<p>number of predictor variables that have been included in the analysis are
actually relevant to prediction. Also, unlike many applications such as pat-
tern recognition, there is seldom reliable domain knowledge to help create
especially relevant features and/or filter out the irrelevant ones, the inclu-
sion of which dramatically degrades the performance of many methods.
In addition, data mining applications generally require interpretable mod-
</p>
<p>els. It is not enough to simply produce predictions. It is also desirable to
have information providing qualitative understanding of the relationship</p>
<p/>
</div>
<div class="page"><p/>
<p>352 10. Boosting and Additive Trees
</p>
<p>between joint values of the input variables and the resulting predicted re-
sponse value. Thus, black box methods such as neural networks, which can
be quite useful in purely predictive settings such as pattern recognition,
are far less useful for data mining.
These requirements of speed, interpretability and the messy nature of
</p>
<p>the data sharply limit the usefulness of most learning procedures as off-
the-shelf methods for data mining. An &ldquo;off-the-shelf&rdquo; method is one that
can be directly applied to the data without requiring a great deal of time-
consuming data preprocessing or careful tuning of the learning procedure.
Of all the well-known learning methods, decision trees come closest to
</p>
<p>meeting the requirements for serving as an off-the-shelf procedure for data
mining. They are relatively fast to construct and they produce interpretable
models (if the trees are small). As discussed in Section 9.2, they naturally
incorporate mixtures of numeric and categorical predictor variables and
missing values. They are invariant under (strictly monotone) transforma-
tions of the individual predictors. As a result, scaling and/or more general
transformations are not an issue, and they are immune to the effects of pre-
dictor outliers. They perform internal feature selection as an integral part
of the procedure. They are thereby resistant, if not completely immune,
to the inclusion of many irrelevant predictor variables. These properties of
decision trees are largely the reason that they have emerged as the most
popular learning method for data mining.
Trees have one aspect that prevents them from being the ideal tool for
</p>
<p>predictive learning, namely inaccuracy. They seldom provide predictive ac-
curacy comparable to the best that can be achieved with the data at hand.
As seen in Section 10.1, boosting decision trees improves their accuracy,
often dramatically. At the same time it maintains most of their desirable
properties for data mining. Some advantages of trees that are sacrificed by
boosting are speed, interpretability, and, for AdaBoost, robustness against
overlapping class distributions and especially mislabeling of the training
data. A gradient boosted model (GBM) is a generalization of tree boosting
that attempts to mitigate these problems, so as to produce an accurate and
effective off-the-shelf procedure for data mining.
</p>
<p>10.8 Example: Spam Data
</p>
<p>Before we go into the details of gradient boosting, we demonstrate its abili-
ties on a two-class classification problem. The spam data are introduced in
Chapter 1, and used as an example for many of the procedures in Chapter 9
(Sections 9.1.2, 9.2.5, 9.3.1 and 9.4.1).
Applying gradient boosting to these data resulted in a test error rate of
</p>
<p>4.5%, using the same test set as was used in Section 9.1.2. By comparison,
an additive logistic regression achieved 5.5%, a CART tree fully grown and</p>
<p/>
</div>
<div class="page"><p/>
<p>10.9 Boosting Trees 353
</p>
<p>pruned by cross-validation 8.7%, and MARS 5.5%. The standard error of
these estimates is around 0.6%, although gradient boosting is significantly
better than all of them using the McNemar test (Exercise 10.6).
In Section 10.13 below we develop a relative importance measure for
</p>
<p>each predictor, as well as a partial dependence plot describing a predictor&rsquo;s
contribution to the fitted model. We now illustrate these for the spam data.
Figure 10.6 displays the relative importance spectrum for all 57 predictor
</p>
<p>variables. Clearly some predictors are more important than others in sep-
arating spam from email. The frequencies of the character strings !, $, hp,
and remove are estimated to be the four most relevant predictor variables.
At the other end of the spectrum, the character strings 857, 415, table, and
3d have virtually no relevance.
The quantity being modeled here is the log-odds of spam versus email
</p>
<p>f(x) = log
Pr(spam|x)
Pr(email|x) (10.24)
</p>
<p>(see Section 10.13 below). Figure 10.7 shows the partial dependence of the
log-odds on selected important predictors, two positively associated with
spam (! and remove), and two negatively associated (edu and hp). These
particular dependencies are seen to be essentially monotonic. There is a
general agreement with the corresponding functions found by the additive
logistic regression model; see Figure 9.1 on page 303.
Running a gradient boosted model on these data with J = 2 terminal-
</p>
<p>node trees produces a purely additive (main effects) model for the log-
odds, with a corresponding error rate of 4.7%, as compared to 4.5% for the
full gradient boosted model (with J = 5 terminal-node trees). Although
not significant, this slightly higher error rate suggests that there may be
interactions among some of the important predictor variables. This can
be diagnosed through two-variable partial dependence plots. Figure 10.8
shows one of the several such plots displaying strong interaction effects.
One sees that for very low frequencies of hp, the log-odds of spam are
</p>
<p>greatly increased. For high frequencies of hp, the log-odds of spam tend to
be much lower and roughly constant as a function of !. As the frequency
of hp decreases, the functional relationship with ! strengthens.
</p>
<p>10.9 Boosting Trees
</p>
<p>Regression and classification trees are discussed in detail in Section 9.2.
They partition the space of all joint predictor variable values into disjoint
regions Rj , j = 1, 2, . . . , J , as represented by the terminal nodes of the tree.
A constant γj is assigned to each such region and the predictive rule is
</p>
<p>x &isin; Rj &rArr; f(x) = γj .</p>
<p/>
</div>
<div class="page"><p/>
<p>354 10. Boosting and Additive Trees
</p>
<p>!
$
</p>
<p>hp
remove
</p>
<p>free
CAPAVE
</p>
<p>your
CAPMAX
</p>
<p>george
CAPTOT
</p>
<p>edu
you
our
</p>
<p>money
will
</p>
<p>1999
business
</p>
<p>re
(
</p>
<p>receive
internet
</p>
<p>000
email
</p>
<p>meeting
;
</p>
<p>650
over
mail
pm
</p>
<p>people
technology
</p>
<p>hpl
all
</p>
<p>order
address
</p>
<p>make
font
</p>
<p>project
data
</p>
<p>original
report
</p>
<p>conference
lab
</p>
<p>[
credit
parts
</p>
<p>#
85
</p>
<p>table
cs
</p>
<p>direct
415
857
</p>
<p>telnet
labs
</p>
<p>addresses
3d
</p>
<p>0 20 40 60 80 100
</p>
<p>Relative Importance
</p>
<p>FIGURE 10.6. Predictor variable importance spectrum for the spam data. The
variable names are written on the vertical axis.</p>
<p/>
</div>
<div class="page"><p/>
<p>10.9 Boosting Trees 355
</p>
<p>!
</p>
<p>P
a
</p>
<p>rt
ia
</p>
<p>l 
D
</p>
<p>e
p
</p>
<p>e
n
</p>
<p>d
e
</p>
<p>n
c
e
</p>
<p>0.0 0.2 0.4 0.6 0.8 1.0
</p>
<p>-0
.2
</p>
<p>0
.0
</p>
<p>0
.2
</p>
<p>0
.4
</p>
<p>0
.6
</p>
<p>0
.8
</p>
<p>1
.0
</p>
<p>remove
</p>
<p>P
a
</p>
<p>rt
ia
</p>
<p>l 
D
</p>
<p>e
p
</p>
<p>e
n
</p>
<p>d
e
</p>
<p>n
c
e
</p>
<p>0.0 0.2 0.4 0.6
</p>
<p>-0
.2
</p>
<p>0
.0
</p>
<p>0
.2
</p>
<p>0
.4
</p>
<p>0
.6
</p>
<p>0
.8
</p>
<p>1
.0
</p>
<p>edu
</p>
<p>P
a
</p>
<p>rt
ia
</p>
<p>l 
D
</p>
<p>e
p
</p>
<p>e
n
</p>
<p>d
e
</p>
<p>n
c
e
</p>
<p>0.0 0.2 0.4 0.6 0.8 1.0
</p>
<p>-1
.0
</p>
<p>-0
.6
</p>
<p>-0
.2
</p>
<p>0
.0
</p>
<p>0
.2
</p>
<p>hp
</p>
<p>P
a
</p>
<p>rt
ia
</p>
<p>l 
D
</p>
<p>e
p
</p>
<p>e
n
</p>
<p>d
e
</p>
<p>n
c
e
</p>
<p>0.0 0.5 1.0 1.5 2.0 2.5 3.0
</p>
<p>-1
.0
</p>
<p>-0
.6
</p>
<p>-0
.2
</p>
<p>0
.0
</p>
<p>0
.2
</p>
<p>FIGURE 10.7. Partial dependence of log-odds of spam on four important pre-
dictors. The red ticks at the base of the plots are deciles of the input variable.
</p>
<p>0.51.01.52.02.53.0
</p>
<p>0.2
</p>
<p>0.4
</p>
<p>0.6
</p>
<p>0.8
</p>
<p>1.0
</p>
<p>-1.0
</p>
<p>-0.5
</p>
<p> 0.0
</p>
<p> 0.5
</p>
<p> 1.0
</p>
<p>hp
</p>
<p>!
</p>
<p>FIGURE 10.8. Partial dependence of the log-odds of spam vs. email as a func-
tion of joint frequencies of hp and the character !.</p>
<p/>
</div>
<div class="page"><p/>
<p>356 10. Boosting and Additive Trees
</p>
<p>Thus a tree can be formally expressed as
</p>
<p>T (x; Θ) =
</p>
<p>J&sum;
</p>
<p>j=1
</p>
<p>γjI(x &isin; Rj), (10.25)
</p>
<p>with parameters Θ = {Rj , γj}J1 . J is usually treated as a meta-parameter.
The parameters are found by minimizing the empirical risk
</p>
<p>Θ̂ = argmin
Θ
</p>
<p>J&sum;
</p>
<p>j=1
</p>
<p>&sum;
</p>
<p>xi&isin;Rj
</p>
<p>L(yi, γj). (10.26)
</p>
<p>This is a formidable combinatorial optimization problem, and we usually
settle for approximate suboptimal solutions. It is useful to divide the opti-
mization problem into two parts:
</p>
<p>Finding γj given Rj : Given the Rj , estimating the γj is typically trivial,
and often γ̂j = ȳj , the mean of the yi falling in region Rj . For mis-
classification loss, γ̂j is the modal class of the observations falling in
region Rj .
</p>
<p>Finding Rj : This is the difficult part, for which approximate solutions are
found. Note also that finding the Rj entails estimating the γj as well.
A typical strategy is to use a greedy, top-down recursive partitioning
algorithm to find the Rj . In addition, it is sometimes necessary to
approximate (10.26) by a smoother and more convenient criterion for
optimizing the Rj :
</p>
<p>Θ̃ = argmin
Θ
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>L̃(yi, T (xi,Θ)). (10.27)
</p>
<p>Then given the R̂j = R̃j , the γj can be estimated more precisely
using the original criterion.
</p>
<p>In Section 9.2 we described such a strategy for classification trees. The Gini
index replaced misclassification loss in the growing of the tree (identifying
the Rj).
The boosted tree model is a sum of such trees,
</p>
<p>fM (x) =
</p>
<p>M&sum;
</p>
<p>m=1
</p>
<p>T (x; Θm), (10.28)
</p>
<p>induced in a forward stagewise manner (Algorithm 10.2). At each step in
the forward stagewise procedure one must solve
</p>
<p>Θ̂m = argmin
Θm
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>L (yi, fm&minus;1(xi) + T (xi; Θm)) (10.29)</p>
<p/>
</div>
<div class="page"><p/>
<p>10.9 Boosting Trees 357
</p>
<p>for the region set and constants Θm = {Rjm, γjm}Jm1 of the next tree, given
the current model fm&minus;1(x).
Given the regions Rjm, finding the optimal constants γjm in each region
</p>
<p>is typically straightforward:
</p>
<p>γ̂jm = argmin
γjm
</p>
<p>&sum;
</p>
<p>xi&isin;Rjm
</p>
<p>L (yi, fm&minus;1(xi) + γjm) . (10.30)
</p>
<p>Finding the regions is difficult, and even more difficult than for a single
tree. For a few special cases, the problem simplifies.
For squared-error loss, the solution to (10.29) is no harder than for a
</p>
<p>single tree. It is simply the regression tree that best predicts the current
residuals yi &minus; fm&minus;1(xi), and γ̂jm is the mean of these residuals in each
corresponding region.
For two-class classification and exponential loss, this stagewise approach
</p>
<p>gives rise to the AdaBoost method for boosting classification trees (Algo-
rithm 10.1). In particular, if the trees T (x; Θm) are restricted to be scaled
classification trees, then we showed in Section 10.4 that the solution to
</p>
<p>(10.29) is the tree that minimizes the weighted error rate
&sum;N
</p>
<p>i=1 w
(m)
i I(yi �=
</p>
<p>T (xi; Θm)) with weights w
(m)
i = e
</p>
<p>&minus;yifm&minus;1(xi). By a scaled classification
tree, we mean βmT (x; Θm), with the restriction that γjm &isin; {&minus;1, 1}).
</p>
<p>Without this restriction, (10.29) still simplifies for exponential loss to a
weighted exponential criterion for the new tree:
</p>
<p>Θ̂m = argmin
Θm
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>w
(m)
i exp[&minus;yiT (xi; Θm)]. (10.31)
</p>
<p>It is straightforward to implement a greedy recursive-partitioning algorithm
using this weighted exponential loss as a splitting criterion. Given the Rjm,
one can show (Exercise 10.7) that the solution to (10.30) is the weighted
log-odds in each corresponding region
</p>
<p>γ̂jm =
1
</p>
<p>2
log
</p>
<p>&sum;
xi&isin;Rjm
</p>
<p>w
(m)
i I(yi = 1)
</p>
<p>&sum;
xi&isin;Rjm
</p>
<p>w
(m)
i I(yi = &minus;1)
</p>
<p>. (10.32)
</p>
<p>This requires a specialized tree-growing algorithm; in practice, we prefer
the approximation presented below that uses a weighted least squares re-
gression tree.
Using loss criteria such as the absolute error or the Huber loss (10.23) in
</p>
<p>place of squared-error loss for regression, and the deviance (10.22) in place
of exponential loss for classification, will serve to robustify boosting trees.
Unfortunately, unlike their nonrobust counterparts, these robust criteria
do not give rise to simple fast boosting algorithms.
For more general loss criteria the solution to (10.30), given the Rjm,
</p>
<p>is typically straightforward since it is a simple &ldquo;location&rdquo; estimate. For</p>
<p/>
</div>
<div class="page"><p/>
<p>358 10. Boosting and Additive Trees
</p>
<p>absolute loss it is just the median of the residuals in each respective region.
For the other criteria fast iterative algorithms exist for solving (10.30),
and usually their faster &ldquo;single-step&rdquo; approximations are adequate. The
problem is tree induction. Simple fast algorithms do not exist for solving
(10.29) for these more general loss criteria, and approximations like (10.27)
become essential.
</p>
<p>10.10 Numerical Optimization via Gradient
Boosting
</p>
<p>Fast approximate algorithms for solving (10.29) with any differentiable loss
criterion can be derived by analogy to numerical optimization. The loss in
using f(x) to predict y on the training data is
</p>
<p>L(f) =
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>L(yi, f(xi)). (10.33)
</p>
<p>The goal is to minimize L(f) with respect to f , where here f(x) is con-
strained to be a sum of trees (10.28). Ignoring this constraint, minimizing
(10.33) can be viewed as a numerical optimization
</p>
<p>f̂ = argmin
f
</p>
<p>L(f), (10.34)
</p>
<p>where the &ldquo;parameters&rdquo; f &isin; IRN are the values of the approximating func-
tion f(xi) at each of the N data points xi:
</p>
<p>f = {f(x1), f(x2), . . . , f(xN )}T .
Numerical optimization procedures solve (10.34) as a sum of component
</p>
<p>vectors
</p>
<p>fM =
</p>
<p>M&sum;
</p>
<p>m=0
</p>
<p>hm , hm &isin; IRN ,
</p>
<p>where f0 = h0 is an initial guess, and each successive fm is induced based
on the current parameter vector fm&minus;1, which is the sum of the previously
induced updates. Numerical optimization methods differ in their prescrip-
tions for computing each increment vector hm (&ldquo;step&rdquo;).
</p>
<p>10.10.1 Steepest Descent
</p>
<p>Steepest descent chooses hm = &minus;ρmgm where ρm is a scalar and gm &isin; IRN
is the gradient of L(f) evaluated at f = fm&minus;1. The components of the
gradient gm are
</p>
<p>gim =
</p>
<p>[
&part;L(yi, f(xi))
</p>
<p>&part;f(xi)
</p>
<p>]
</p>
<p>f(xi)=fm&minus;1(xi)
</p>
<p>(10.35)</p>
<p/>
</div>
<div class="page"><p/>
<p>10.10 Numerical Optimization via Gradient Boosting 359
</p>
<p>The step length ρm is the solution to
</p>
<p>ρm = argmin
ρ
</p>
<p>L(fm&minus;1 &minus; ρgm). (10.36)
</p>
<p>The current solution is then updated
</p>
<p>fm = fm&minus;1 &minus; ρmgm
</p>
<p>and the process repeated at the next iteration. Steepest descent can be
viewed as a very greedy strategy, since &minus;gm is the local direction in IRN
for which L(f) is most rapidly decreasing at f = fm&minus;1.
</p>
<p>10.10.2 Gradient Boosting
</p>
<p>Forward stagewise boosting (Algorithm 10.2) is also a very greedy strategy.
At each step the solution tree is the one that maximally reduces (10.29),
given the current model fm&minus;1 and its fits fm&minus;1(xi). Thus, the tree predic-
tions T (xi; Θm) are analogous to the components of the negative gradient
(10.35). The principal difference between them is that the tree components
tm = {T (x1; Θm), . . . , T (xN ; Θm)}T are not independent. They are con-
strained to be the predictions of a Jm-terminal node decision tree, whereas
the negative gradient is the unconstrained maximal descent direction.
The solution to (10.30) in the stagewise approach is analogous to the line
</p>
<p>search (10.36) in steepest descent. The difference is that (10.30) performs
a separate line search for those components of tm that correspond to each
separate terminal region {T (xi; Θm)}xi&isin;Rjm .
If minimizing loss on the training data (10.33) were the only goal, steep-
</p>
<p>est descent would be the preferred strategy. The gradient (10.35) is trivial
to calculate for any differentiable loss function L(y, f(x)), whereas solving
(10.29) is difficult for the robust criteria discussed in Section 10.6. Unfor-
tunately the gradient (10.35) is defined only at the training data points xi,
whereas the ultimate goal is to generalize fM (x) to new data not repre-
sented in the training set.
A possible resolution to this dilemma is to induce a tree T (x; Θm) at the
</p>
<p>mth iteration whose predictions tm are as close as possible to the negative
gradient. Using squared error to measure closeness, this leads us to
</p>
<p>Θ̃m = argmin
Θ
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>(&minus;gim &minus; T (xi; Θ))2 . (10.37)
</p>
<p>That is, one fits the tree T to the negative gradient values (10.35) by least
squares. As noted in Section 10.9 fast algorithms exist for least squares
decision tree induction. Although the solution regions R̃jm to (10.37) will
not be identical to the regions Rjm that solve (10.29), it is generally sim-
ilar enough to serve the same purpose. In any case, the forward stagewise</p>
<p/>
</div>
<div class="page"><p/>
<p>360 10. Boosting and Additive Trees
</p>
<p>TABLE 10.2. Gradients for commonly used loss functions.
</p>
<p>Setting Loss Function &minus;&part;L(yi, f(xi))/&part;f(xi)
</p>
<p>Regression 12 [yi &minus; f(xi)]2 yi &minus; f(xi)
</p>
<p>Regression |yi &minus; f(xi)| sign[yi &minus; f(xi)]
</p>
<p>Regression Huber yi &minus; f(xi) for |yi &minus; f(xi)| &le; δm
δmsign[yi &minus; f(xi)] for |yi &minus; f(xi)| &gt; δm
where δm = αth-quantile{|yi &minus; f(xi)|}
</p>
<p>Classification Deviance kth component: I(yi = Gk)&minus; pk(xi)
</p>
<p>boosting procedure, and top-down decision tree induction, are themselves
approximation procedures. After constructing the tree (10.37), the corre-
sponding constants in each region are given by (10.30).
Table 10.2 summarizes the gradients for commonly used loss functions.
</p>
<p>For squared error loss, the negative gradient is just the ordinary residual
&minus;gim = yi &minus; fm&minus;1(xi), so that (10.37) on its own is equivalent to standard
least-squares boosting. With absolute error loss, the negative gradient is
the sign of the residual, so at each iteration (10.37) fits the tree to the
sign of the current residuals by least squares. For Huber M-regression, the
negative gradient is a compromise between these two (see the table).
For classification the loss function is the multinomial deviance (10.22),
</p>
<p>and K least squares trees are constructed at each iteration. Each tree Tkm
is fit to its respective negative gradient vector gkm,
</p>
<p>&minus;gikm =
[
&part;L (yi, f1(xi), . . . , fK(xi))
</p>
<p>&part;fk(xi)
</p>
<p>]
</p>
<p>f(xi)=fm&minus;1(xi)
</p>
<p>= I(yi = Gk)&minus; pk(xi), (10.38)
with pk(x) given by (10.21). Although K separate trees are built at each
iteration, they are related through (10.21). For binary classification (K =
2), only one tree is needed (exercise 10.10).
</p>
<p>10.10.3 Implementations of Gradient Boosting
</p>
<p>Algorithm 10.3 presents the generic gradient tree-boosting algorithm for
regression. Specific algorithms are obtained by inserting different loss cri-
teria L(y, f(x)). The first line of the algorithm initializes to the optimal
constant model, which is just a single terminal node tree. The components
of the negative gradient computed at line 2(a) are referred to as general-
ized or pseudo residuals, r. Gradients for commonly used loss functions are
summarized in Table 10.2.</p>
<p/>
</div>
<div class="page"><p/>
<p>10.11 Right-Sized Trees for Boosting 361
</p>
<p>Algorithm 10.3 Gradient Tree Boosting Algorithm.
</p>
<p>1. Initialize f0(x) = argminγ
&sum;N
</p>
<p>i=1 L(yi, γ).
</p>
<p>2. For m = 1 to M :
</p>
<p>(a) For i = 1, 2, . . . , N compute
</p>
<p>rim = &minus;
[
&part;L(yi, f(xi))
</p>
<p>&part;f(xi)
</p>
<p>]
</p>
<p>f=fm&minus;1
</p>
<p>.
</p>
<p>(b) Fit a regression tree to the targets rim giving terminal regions
Rjm, j = 1, 2, . . . , Jm.
</p>
<p>(c) For j = 1, 2, . . . , Jm compute
</p>
<p>γjm = argmin
γ
</p>
<p>&sum;
</p>
<p>xi&isin;Rjm
</p>
<p>L (yi, fm&minus;1(xi) + γ) .
</p>
<p>(d) Update fm(x) = fm&minus;1(x) +
&sum;Jm
</p>
<p>j=1 γjmI(x &isin; Rjm).
</p>
<p>3. Output f̂(x) = fM (x).
</p>
<p>The algorithm for classification is similar. Lines 2(a)&ndash;(d) are repeated
K times at each iteration m, once for each class using (10.38). The result
at line 3 is K different (coupled) tree expansions fkM (x), k = 1, 2, . . . ,K.
These produce probabilities via (10.21) or do classification as in (10.20).
Details are given in Exercise 10.9. Two basic tuning parameters are the
number of iterations M and the sizes of each of the constituent trees
Jm, m = 1, 2, . . . ,M .
The original implementation of this algorithm was called MART for
</p>
<p>&ldquo;multiple additive regression trees,&rdquo; and was referred to in the first edi-
tion of this book. Many of the figures in this chapter were produced by
MART. Gradient boosting as described here is implemented in the R gbm
package (Ridgeway, 1999, &ldquo;Gradient Boosted Models&rdquo;), and is freely avail-
able. The gbm package is used in Section 10.14.2, and extensively in Chap-
ters 16 and 15. Another R implementation of boosting is mboost (Hothorn
and Bühlmann, 2006). A commercial implementation of gradient boost-
ing/MART called TreeNet is available from Salford Systems, Inc.
</p>
<p>10.11 Right-Sized Trees for Boosting
</p>
<p>Historically, boosting was considered to be a technique for combining mod-
els, here trees. As such, the tree building algorithm was regarded as a</p>
<p/>
</div>
<div class="page"><p/>
<p>362 10. Boosting and Additive Trees
</p>
<p>primitive that produced models to be combined by the boosting proce-
dure. In this scenario, the optimal size of each tree is estimated separately
in the usual manner when it is built (Section 9.2). A very large (oversized)
tree is first induced, and then a bottom-up procedure is employed to prune
it to the estimated optimal number of terminal nodes. This approach as-
sumes implicitly that each tree is the last one in the expansion (10.28).
Except perhaps for the very last tree, this is clearly a very poor assump-
tion. The result is that trees tend to be much too large, especially during
the early iterations. This substantially degrades performance and increases
computation.
The simplest strategy for avoiding this problem is to restrict all trees
</p>
<p>to be the same size, Jm = J &forall;m. At each iteration a J-terminal node
regression tree is induced. Thus J becomes a meta-parameter of the entire
boosting procedure, to be adjusted to maximize estimated performance for
the data at hand.
One can get an idea of useful values for J by considering the properties
</p>
<p>of the &ldquo;target&rdquo; function
</p>
<p>η = argmin
f
</p>
<p>EXY L(Y, f(X)). (10.39)
</p>
<p>Here the expected value is over the population joint distribution of (X,Y ).
The target function η(x) is the one with minimum prediction risk on future
data. This is the function we are trying to approximate.
One relevant property of η(X) is the degree to which the coordinate vari-
</p>
<p>ables XT = (X1, X2, . . . , Xp) interact with one another. This is captured
by its ANOVA (analysis of variance) expansion
</p>
<p>η(X) =
&sum;
</p>
<p>j
</p>
<p>ηj(Xj)+
&sum;
</p>
<p>jk
</p>
<p>ηjk(Xj , Xk)+
&sum;
</p>
<p>jkl
</p>
<p>ηjkl(Xj , Xk, Xl)+ &middot; &middot; &middot; . (10.40)
</p>
<p>The first sum in (10.40) is over functions of only a single predictor variable
Xj . The particular functions ηj(Xj) are those that jointly best approximate
η(X) under the loss criterion being used. Each such ηj(Xj) is called the
&ldquo;main effect&rdquo; of Xj . The second sum is over those two-variable functions
that when added to the main effects best fit η(X). These are called the
second-order interactions of each respective variable pair (Xj , Xk). The
third sum represents third-order interactions, and so on. For many problems
encountered in practice, low-order interaction effects tend to dominate.
When this is the case, models that produce strong higher-order interaction
effects, such as large decision trees, suffer in accuracy.
The interaction level of tree-based approximations is limited by the tree
</p>
<p>size J . Namely, no interaction effects of level greater than J &minus; 1 are pos-
sible. Since boosted models are additive in the trees (10.28), this limit
extends to them as well. Setting J = 2 (single split &ldquo;decision stump&rdquo;)
produces boosted models with only main effects; no interactions are per-
mitted. With J = 3, two-variable interaction effects are also allowed, and</p>
<p/>
</div>
<div class="page"><p/>
<p>10.11 Right-Sized Trees for Boosting 363
</p>
<p>Number of Terms
</p>
<p>T
e
s
t 
E
</p>
<p>rr
o
r
</p>
<p>0 100 200 300 400
</p>
<p>0
.0
</p>
<p>0
.1
</p>
<p>0
.2
</p>
<p>0
.3
</p>
<p>0
.4
</p>
<p>Stumps
10 Node
100 Node
Adaboost
</p>
<p>FIGURE 10.9. Boosting with different sized trees, applied to the example (10.2)
used in Figure 10.2. Since the generative model is additive, stumps perform the
best. The boosting algorithm used the binomial deviance loss in Algorithm 10.3;
shown for comparison is the AdaBoost Algorithm 10.1.
</p>
<p>so on. This suggests that the value chosen for J should reflect the level
of dominant interactions of η(x). This is of course generally unknown, but
in most situations it will tend to be low. Figure 10.9 illustrates the effect
of interaction order (choice of J) on the simulation example (10.2). The
generative function is additive (sum of quadratic monomials), so boosting
models with J &gt; 2 incurs unnecessary variance and hence the higher test
error. Figure 10.10 compares the coordinate functions found by boosted
stumps with the true functions.
Although in many applications J = 2 will be insufficient, it is unlikely
</p>
<p>that J &gt; 10 will be required. Experience so far indicates that 4 &le; J &le; 8
works well in the context of boosting, with results being fairly insensitive
to particular choices in this range. One can fine-tune the value for J by
trying several different values and choosing the one that produces the low-
est risk on a validation sample. However, this seldom provides significant
improvement over using J ≃ 6.</p>
<p/>
</div>
<div class="page"><p/>
<p>364 10. Boosting and Additive Trees
</p>
<p>Coordinate Functions for Additive Logistic Trees
</p>
<p>f1(x1) f2(x2) f3(x3) f4(x4) f5(x5)
</p>
<p>f6(x6) f7(x7) f8(x8) f9(x9) f10(x10)
</p>
<p>FIGURE 10.10. Coordinate functions estimated by boosting stumps for the sim-
ulated example used in Figure 10.9. The true quadratic functions are shown for
comparison.
</p>
<p>10.12 Regularization
</p>
<p>Besides the size of the constituent trees, J , the other meta-parameter of
gradient boosting is the number of boosting iterations M . Each iteration
usually reduces the training risk L(fM ), so that forM large enough this risk
can be made arbitrarily small. However, fitting the training data too well
can lead to overfitting, which degrades the risk on future predictions. Thus,
there is an optimal number M&lowast; minimizing future risk that is application
dependent. A convenient way to estimate M&lowast; is to monitor prediction risk
as a function of M on a validation sample. The value of M that minimizes
this risk is taken to be an estimate of M&lowast;. This is analogous to the early
stopping strategy often used with neural networks (Section 11.4).
</p>
<p>10.12.1 Shrinkage
</p>
<p>Controlling the value of M is not the only possible regularization strategy.
As with ridge regression and neural networks, shrinkage techniques can be
employed as well (see Sections 3.4.1 and 11.5). The simplest implementation
of shrinkage in the context of boosting is to scale the contribution of each
tree by a factor 0 &lt; ν &lt; 1 when it is added to the current approximation.
That is, line 2(d) of Algorithm 10.3 is replaced by
</p>
<p>fm(x) = fm&minus;1(x) + ν &middot;
J&sum;
</p>
<p>j=1
</p>
<p>γjmI(x &isin; Rjm). (10.41)
</p>
<p>The parameter ν can be regarded as controlling the learning rate of the
boosting procedure. Smaller values of ν (more shrinkage) result in larger
training risk for the same number of iterations M . Thus, both ν and M
control prediction risk on the training data. However, these parameters do</p>
<p/>
</div>
<div class="page"><p/>
<p>10.12 Regularization 365
</p>
<p>not operate independently. Smaller values of ν lead to larger values of M
for the same training risk, so that there is a tradeoff between them.
Empirically it has been found (Friedman, 2001) that smaller values of ν
</p>
<p>favor better test error, and require correspondingly larger values of M . In
fact, the best strategy appears to be to set ν to be very small (ν &lt; 0.1)
and then choose M by early stopping. This yields dramatic improvements
(over no shrinkage ν = 1) for regression and for probability estimation. The
corresponding improvements in misclassification risk via (10.20) are less,
but still substantial. The price paid for these improvements is computa-
tional: smaller values of ν give rise to larger values of M , and computation
is proportional to the latter. However, as seen below, many iterations are
generally computationally feasible even on very large data sets. This is
partly due to the fact that small trees are induced at each step with no
pruning.
Figure 10.11 shows test error curves for the simulated example (10.2) of
</p>
<p>Figure 10.2. A gradient boosted model (MART) was trained using binomial
deviance, using either stumps or six terminal-node trees, and with or with-
out shrinkage. The benefits of shrinkage are evident, especially when the
binomial deviance is tracked. With shrinkage, each test error curve reaches
a lower value, and stays there for many iterations.
Section 16.2.1 draws a connection between forward stagewise shrinkage
</p>
<p>in boosting and the use of an L1 penalty for regularizing model parame-
ters (the &ldquo;lasso&rdquo;). We argue that L1 penalties may be superior to the L2
penalties used by methods such as the support vector machine.
</p>
<p>10.12.2 Subsampling
</p>
<p>We saw in Section 8.7 that bootstrap averaging (bagging) improves the
performance of a noisy classifier through averaging. Chapter 15 discusses
in some detail the variance-reduction mechanism of this sampling followed
by averaging. We can exploit the same device in gradient boosting, both
to improve performance and computational efficiency.
With stochastic gradient boosting (Friedman, 1999), at each iteration we
</p>
<p>sample a fraction η of the training observations (without replacement),
and grow the next tree using that subsample. The rest of the algorithm is
identical. A typical value for η can be 12 , although for large N , η can be
substantially smaller than 12 .
</p>
<p>Not only does the sampling reduce the computing time by the same
fraction η, but in many cases it actually produces a more accurate model.
</p>
<p>Figure 10.12 illustrates the effect of subsampling using the simulated
example (10.2), both as a classification and as a regression example. We
see in both cases that sampling along with shrinkage slightly outperformed
the rest. It appears here that subsampling without shrinkage does poorly.</p>
<p/>
</div>
<div class="page"><p/>
<p>366 10. Boosting and Additive Trees
</p>
<p>Boosting Iterations
</p>
<p>T
e
</p>
<p>s
t 
</p>
<p>S
e
</p>
<p>t 
D
</p>
<p>e
v
ia
</p>
<p>n
c
e
</p>
<p>0 500 1000 1500 2000
</p>
<p>0
.0
</p>
<p>0
.5
</p>
<p>1
.0
</p>
<p>1
.5
</p>
<p>2
.0
</p>
<p>No shrinkage
Shrinkage=0.2
</p>
<p>Stumps
Deviance
</p>
<p>Boosting Iterations
</p>
<p>T
e
</p>
<p>s
t 
</p>
<p>S
e
</p>
<p>t 
M
</p>
<p>is
c
la
</p>
<p>s
s
if
ic
</p>
<p>a
ti
o
</p>
<p>n
 E
</p>
<p>rr
o
</p>
<p>r
</p>
<p>0 500 1000 1500 2000
0
</p>
<p>.0
0
</p>
<p>.1
0
</p>
<p>.2
0
</p>
<p>.3
0
</p>
<p>.4
0
</p>
<p>.5
</p>
<p>No shrinkage
Shrinkage=0.2
</p>
<p>Stumps
Misclassification Error
</p>
<p>Boosting Iterations
</p>
<p>T
e
</p>
<p>s
t 
</p>
<p>S
e
</p>
<p>t 
D
</p>
<p>e
v
ia
</p>
<p>n
c
e
</p>
<p>0 500 1000 1500 2000
</p>
<p>0
.0
</p>
<p>0
.5
</p>
<p>1
.0
</p>
<p>1
.5
</p>
<p>2
.0
</p>
<p>No shrinkage
Shrinkage=0.6
</p>
<p>6-Node Trees
Deviance
</p>
<p>Boosting Iterations
</p>
<p>T
e
</p>
<p>s
t 
</p>
<p>S
e
</p>
<p>t 
M
</p>
<p>is
c
la
</p>
<p>s
s
if
ic
</p>
<p>a
ti
o
</p>
<p>n
 E
</p>
<p>rr
o
</p>
<p>r
</p>
<p>0 500 1000 1500 2000
</p>
<p>0
.0
</p>
<p>0
.1
</p>
<p>0
.2
</p>
<p>0
.3
</p>
<p>0
.4
</p>
<p>0
.5
</p>
<p>No shrinkage
Shrinkage=0.6
</p>
<p>6-Node Trees
Misclassification Error
</p>
<p>FIGURE 10.11. Test error curves for simulated example (10.2) of Figure 10.9,
using gradient boosting (MART). The models were trained using binomial de-
viance, either stumps or six terminal-node trees, and with or without shrinkage.
The left panels report test deviance, while the right panels show misclassification
error. The beneficial effect of shrinkage can be seen in all cases, especially for
deviance in the left panels.</p>
<p/>
</div>
<div class="page"><p/>
<p>10.13 Interpretation 367
</p>
<p>0 200 400 600 800 1000
</p>
<p>0
.4
</p>
<p>0
.6
</p>
<p>0
.8
</p>
<p>1
.0
</p>
<p>1
.2
</p>
<p>1
.4
</p>
<p>Boosting Iterations
</p>
<p>T
e
s
t 
S
</p>
<p>e
t 
D
</p>
<p>e
v
ia
</p>
<p>n
c
e
</p>
<p>Deviance
</p>
<p>4&minus;Node Trees
</p>
<p>0 200 400 600 800 1000
</p>
<p>0
.3
</p>
<p>0
0
.3
</p>
<p>5
0
.4
</p>
<p>0
0
.4
</p>
<p>5
0
.5
</p>
<p>0
Boosting Iterations
</p>
<p>T
e
s
t 
S
</p>
<p>e
t 
A
</p>
<p>b
s
o
lu
</p>
<p>te
 E
</p>
<p>rr
o
r
</p>
<p>No shrinkage
Shrink=0.1
Sample=0.5
Shrink=0.1 Sample=0.5
</p>
<p>Absolute Error
</p>
<p>FIGURE 10.12. Test-error curves for the simulated example (10.2), showing
the effect of stochasticity. For the curves labeled &ldquo;Sample= 0.5&rdquo;, a different 50%
subsample of the training data was used each time a tree was grown. In the left
panel the models were fit by gbm using a binomial deviance loss function; in the
right-hand panel using square-error loss.
</p>
<p>The downside is that we now have four parameters to set: J , M , ν and
η. Typically some early explorations determine suitable values for J , ν and
η, leaving M as the primary parameter.
</p>
<p>10.13 Interpretation
</p>
<p>Single decision trees are highly interpretable. The entire model can be com-
pletely represented by a simple two-dimensional graphic (binary tree) that
is easily visualized. Linear combinations of trees (10.28) lose this important
feature, and must therefore be interpreted in a different way.
</p>
<p>10.13.1 Relative Importance of Predictor Variables
</p>
<p>In data mining applications the input predictor variables are seldom equally
relevant. Often only a few of them have substantial influence on the re-
sponse; the vast majority are irrelevant and could just as well have not
been included. It is often useful to learn the relative importance or contri-
bution of each input variable in predicting the response.</p>
<p/>
</div>
<div class="page"><p/>
<p>368 10. Boosting and Additive Trees
</p>
<p>For a single decision tree T , Breiman et al. (1984) proposed
</p>
<p>I2ℓ (T ) =
J&minus;1&sum;
</p>
<p>t=1
</p>
<p>ı̂2t I(v(t) = ℓ) (10.42)
</p>
<p>as a measure of relevance for each predictor variable Xℓ. The sum is over
the J &minus; 1 internal nodes of the tree. At each such node t, one of the input
variables Xv(t) is used to partition the region associated with that node into
two subregions; within each a separate constant is fit to the response values.
The particular variable chosen is the one that gives maximal estimated
improvement ı̂2t in squared error risk over that for a constant fit over the
entire region. The squared relative importance of variable Xℓ is the sum of
such squared improvements over all internal nodes for which it was chosen
as the splitting variable.
This importance measure is easily generalized to additive tree expansions
</p>
<p>(10.28); it is simply averaged over the trees
</p>
<p>I2ℓ =
1
</p>
<p>M
</p>
<p>M&sum;
</p>
<p>m=1
</p>
<p>I2ℓ (Tm). (10.43)
</p>
<p>Due to the stabilizing effect of averaging, this measure turns out to be more
reliable than is its counterpart (10.42) for a single tree. Also, because of
shrinkage (Section 10.12.1) the masking of important variables by others
with which they are highly correlated is much less of a problem. Note
that (10.42) and (10.43) refer to squared relevance; the actual relevances
are their respective square roots. Since these measures are relative, it is
customary to assign the largest a value of 100 and then scale the others
accordingly. Figure 10.6 shows the relevant importance of the 57 inputs in
predicting spam versus email.
For K-class classification, K separate models fk(x), k = 1, 2, . . . ,K are
</p>
<p>induced, each consisting of a sum of trees
</p>
<p>fk(x) =
</p>
<p>M&sum;
</p>
<p>m=1
</p>
<p>Tkm(x). (10.44)
</p>
<p>In this case (10.43) generalizes to
</p>
<p>I2ℓk =
1
</p>
<p>M
</p>
<p>M&sum;
</p>
<p>m=1
</p>
<p>I2ℓ (Tkm). (10.45)
</p>
<p>Here Iℓk is the relevance of Xℓ in separating the class k observations from
the other classes. The overall relevance of Xℓ is obtained by averaging over
all of the classes
</p>
<p>I2ℓ =
1
</p>
<p>K
</p>
<p>K&sum;
</p>
<p>k=1
</p>
<p>I2ℓk. (10.46)</p>
<p/>
</div>
<div class="page"><p/>
<p>10.13 Interpretation 369
</p>
<p>Figures 10.23 and 10.24 illustrate the use of these averaged and separate
relative importances.
</p>
<p>10.13.2 Partial Dependence Plots
</p>
<p>After the most relevant variables have been identified, the next step is to
attempt to understand the nature of the dependence of the approximation
f(X) on their joint values. Graphical renderings of the f(X) as a function
of its arguments provides a comprehensive summary of its dependence on
the joint values of the input variables.
Unfortunately, such visualization is limited to low-dimensional views.
</p>
<p>We can easily display functions of one or two arguments, either continuous
or discrete (or mixed), in a variety of different ways; this book is filled
with such displays. Functions of slightly higher dimensions can be plotted
by conditioning on particular sets of values of all but one or two of the
arguments, producing a trellis of plots (Becker et al., 1996).1
</p>
<p>For more than two or three variables, viewing functions of the corre-
sponding higher-dimensional arguments is more difficult. A useful alterna-
tive can sometimes be to view a collection of plots, each one of which shows
the partial dependence of the approximation f(X) on a selected small sub-
set of the input variables. Although such a collection can seldom provide a
comprehensive depiction of the approximation, it can often produce helpful
clues, especially when f(x) is dominated by low-order interactions (10.40).
Consider the subvectorXS of ℓ &lt; p of the input predictor variablesX
</p>
<p>T =
(X1, X2, . . . , Xp), indexed by S &sub; {1, 2, . . . , p}. Let C be the complement
set, with S &cup; C = {1, 2, . . . , p}. A general function f(X) will in principle
depend on all of the input variables: f(X) = f(XS , XC). One way to define
the average or partial dependence of f(X) on XS is
</p>
<p>fS(XS) = EXCf(XS , XC). (10.47)
</p>
<p>This is a marginal average of f , and can serve as a useful description of the
effect of the chosen subset on f(X) when, for example, the variables in XS
do not have strong interactions with those in XC .
</p>
<p>Partial dependence functions can be used to interpret the results of any
&ldquo;black box&rdquo; learning method. They can be estimated by
</p>
<p>f̄S(XS) =
1
</p>
<p>N
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>f(XS , xiC), (10.48)
</p>
<p>where {x1C , x2C , . . . , xNC} are the values of XC occurring in the training
data. This requires a pass over the data for each set of joint values of XS for
which f̄S(XS) is to be evaluated. This can be computationally intensive,
</p>
<p>1lattice in R.</p>
<p/>
</div>
<div class="page"><p/>
<p>370 10. Boosting and Additive Trees
</p>
<p>even for moderately sized data sets. Fortunately with decision trees, f̄S(XS)
(10.48) can be rapidly computed from the tree itself without reference to
the data (Exercise 10.11).
It is important to note that partial dependence functions defined in
</p>
<p>(10.47) represent the effect of XS on f(X) after accounting for the (av-
erage) effects of the other variables XC on f(X). They are not the effect
of XS on f(X) ignoring the effects of XC . The latter is given by the con-
ditional expectation
</p>
<p>f̃S(XS) = E(f(XS , XC)|XS), (10.49)
</p>
<p>and is the best least squares approximation to f(X) by a function of XS
alone. The quantities f̃S(XS) and f̄S(XS) will be the same only in the
unlikely event that XS and XC are independent. For example, if the effect
of the chosen variable subset happens to be purely additive,
</p>
<p>f(X) = h1(XS) + h2(XC). (10.50)
</p>
<p>Then (10.47) produces the h1(XS) up to an additive constant. If the effect
is purely multiplicative,
</p>
<p>f(X) = h1(XS) &middot; h2(XC), (10.51)
</p>
<p>then (10.47) produces h1(XS) up to a multiplicative constant factor. On
the other hand, (10.49) will not produce h1(XS) in either case. In fact,
(10.49) can produce strong effects on variable subsets for which f(X) has
no dependence at all.
Viewing plots of the partial dependence of the boosted-tree approxima-
</p>
<p>tion (10.28) on selected variables subsets can help to provide a qualitative
description of its properties. Illustrations are shown in Sections 10.8 and
10.14. Owing to the limitations of computer graphics, and human percep-
tion, the size of the subsets XS must be small (l &asymp; 1, 2, 3). There are of
course a large number of such subsets, but only those chosen from among
the usually much smaller set of highly relevant predictors are likely to be
informative. Also, those subsets whose effect on f(X) is approximately
additive (10.50) or multiplicative (10.51) will be most revealing.
For K-class classification, there are K separate models (10.44), one for
</p>
<p>each class. Each one is related to the respective probabilities (10.21) through
</p>
<p>fk(X) = log pk(X)&minus;
1
</p>
<p>K
</p>
<p>K&sum;
</p>
<p>l=1
</p>
<p>log pl(X). (10.52)
</p>
<p>Thus each fk(X) is a monotone increasing function of its respective prob-
ability on a logarithmic scale. Partial dependence plots of each respective
fk(X) (10.44) on its most relevant predictors (10.45) can help reveal how
the log-odds of realizing that class depend on the respective input variables.</p>
<p/>
</div>
<div class="page"><p/>
<p>10.14 Illustrations 371
</p>
<p>10.14 Illustrations
</p>
<p>In this section we illustrate gradient boosting on a number of larger datasets,
using different loss functions as appropriate.
</p>
<p>10.14.1 California Housing
</p>
<p>This data set (Pace and Barry, 1997) is available from the Carnegie-Mellon
StatLib repository2. It consists of aggregated data from each of 20,460
neighborhoods (1990 census block groups) in California. The response vari-
able Y is the median house value in each neighborhood measured in units of
$100,000. The predictor variables are demographics such as median income
MedInc, housing density as reflected by the number of houses House, and the
average occupancy in each house AveOccup. Also included as predictors are
the location of each neighborhood (longitude and latitude), and several
quantities reflecting the properties of the houses in the neighborhood: av-
erage number of rooms AveRooms and bedrooms AveBedrms. There are thus
a total of eight predictors, all numeric.
We fit a gradient boosting model using the MART procedure, with J = 6
</p>
<p>terminal nodes, a learning rate (10.41) of ν = 0.1, and the Huber loss
criterion for predicting the numeric response. We randomly divided the
dataset into a training set (80%) and a test set (20%).
Figure 10.13 shows the average absolute error
</p>
<p>AAE = E |y &minus; f̂M (x)| (10.53)
</p>
<p>as a function for number of iterations M on both the training data and test
data. The test error is seen to decrease monotonically with increasing M ,
more rapidly during the early stages and then leveling off to being nearly
constant as iterations increase. Thus, the choice of a particular value of M
is not critical, as long as it is not too small. This tends to be the case in
many applications. The shrinkage strategy (10.41) tends to eliminate the
problem of overfitting, especially for larger data sets.
The value of AAE after 800 iterations is 0.31. This can be compared to
</p>
<p>that of the optimal constant predictor median{yi} which is 0.89. In terms of
more familiar quantities, the squared multiple correlation coefficient of this
model is R2 = 0.84. Pace and Barry (1997) use a sophisticated spatial auto-
regression procedure, where prediction for each neighborhood is based on
median house values in nearby neighborhoods, using the other predictors as
covariates. Experimenting with transformations they achieved R2 = 0.85,
predicting log Y . Using log Y as the response the corresponding value for
gradient boosting was R2 = 0.86.
</p>
<p>2http://lib.stat.cmu.edu.</p>
<p/>
</div>
<div class="page"><p/>
<p>372 10. Boosting and Additive Trees
</p>
<p>0 200 400 600 800
</p>
<p>0
.0
</p>
<p>0
.2
</p>
<p>0
.4
</p>
<p>0
.6
</p>
<p>0
.8
</p>
<p>Iterations M
</p>
<p>A
b
s
o
lu
</p>
<p>te
 E
</p>
<p>rr
o
r
</p>
<p>Training and Test Absolute Error
</p>
<p>Train Error
Test Error
</p>
<p>FIGURE 10.13. Average-absolute error as a function of number of iterations
for the California housing data.
</p>
<p>Figure 10.14 displays the relative variable importances for each of the
eight predictor variables. Not surprisingly, median income in the neigh-
borhood is the most relevant predictor. Longitude, latitude, and average
occupancy all have roughly half the relevance of income, whereas the others
are somewhat less influential.
Figure 10.15 shows single-variable partial dependence plots on the most
</p>
<p>relevant nonlocation predictors. Note that the plots are not strictly smooth.
This is a consequence of using tree-based models. Decision trees produce
discontinuous piecewise constant models (10.25). This carries over to sums
of trees (10.28), with of course many more pieces. Unlike most of the meth-
ods discussed in this book, there is no smoothness constraint imposed on
the result. Arbitrarily sharp discontinuities can be modeled. The fact that
these curves generally exhibit a smooth trend is because that is what is
estimated to best predict the response for this problem. This is often the
case.
The hash marks at the base of each plot delineate the deciles of the
</p>
<p>data distribution of the corresponding variables. Note that here the data
density is lower near the edges, especially for larger values. This causes the
curves to be somewhat less well determined in those regions. The vertical
scales of the plots are the same, and give a visual comparison of the relative
importance of the different variables.
The partial dependence of median house value on median income is
</p>
<p>monotonic increasing, being nearly linear over the main body of data. House
value is generally monotonic decreasing with increasing average occupancy,
except perhaps for average occupancy rates less than one. Median house</p>
<p/>
</div>
<div class="page"><p/>
<p>10.14 Illustrations 373
</p>
<p>MedInc
</p>
<p>Longitude
</p>
<p>AveOccup
</p>
<p>Latitude
</p>
<p>HouseAge
</p>
<p>AveRooms
</p>
<p>AveBedrms
</p>
<p>Population
</p>
<p>0 20 40 60 80 100
</p>
<p>Relative importance
</p>
<p>FIGURE 10.14. Relative importance of the predictors for the California housing
data.
</p>
<p>value has a nonmonotonic partial dependence on average number of rooms.
It has a minimum at approximately three rooms and is increasing both for
smaller and larger values.
Median house value is seen to have a very weak partial dependence on
</p>
<p>house age that is inconsistent with its importance ranking (Figure 10.14).
This suggests that this weak main effect may be masking stronger interac-
tion effects with other variables. Figure 10.16 shows the two-variable partial
dependence of housing value on joint values of median age and average oc-
cupancy. An interaction between these two variables is apparent. For values
of average occupancy greater than two, house value is nearly independent
of median age, whereas for values less than two there is a strong dependence
on age.
Figure 10.17 shows the two-variable partial dependence of the fitted
</p>
<p>model on joint values of longitude and latitude, displayed as a shaded
contour plot. There is clearly a very strong dependence of median house
value on the neighborhood location in California. Note that Figure 10.17 is
not a plot of house value versus location ignoring the effects of the other
predictors (10.49). Like all partial dependence plots, it represents the effect
of location after accounting for the effects of the other neighborhood and
house attributes (10.47). It can be viewed as representing an extra premium
one pays for location. This premium is seen to be relatively large near the
Pacific coast especially in the Bay Area and Los Angeles&ndash;San Diego re-</p>
<p/>
</div>
<div class="page"><p/>
<p>374 10. Boosting and Additive Trees
</p>
<p>MedInc
</p>
<p>P
a
</p>
<p>rt
ia
</p>
<p>l 
D
</p>
<p>e
p
</p>
<p>e
n
</p>
<p>d
e
</p>
<p>n
c
e
</p>
<p>2 4 6 8 10
</p>
<p>-0
.5
</p>
<p>0
.0
</p>
<p>0
.5
</p>
<p>1
.0
</p>
<p>1
.5
</p>
<p>2
.0
</p>
<p>AveOccup
</p>
<p>P
a
</p>
<p>rt
ia
</p>
<p>l 
D
</p>
<p>e
p
</p>
<p>e
n
</p>
<p>d
e
</p>
<p>n
c
e
</p>
<p>2 3 4 5
</p>
<p>-1
.0
</p>
<p>-0
.5
</p>
<p>0
.0
</p>
<p>0
.5
</p>
<p>1
.0
</p>
<p>1
.5
</p>
<p>HouseAge
</p>
<p>P
a
</p>
<p>rt
ia
</p>
<p>l 
D
</p>
<p>e
p
</p>
<p>e
n
</p>
<p>d
e
</p>
<p>n
c
e
</p>
<p>10 20 30 40 50
</p>
<p>-1
.0
</p>
<p>-0
.5
</p>
<p>0
.0
</p>
<p>0
.5
</p>
<p>1
.0
</p>
<p>AveRooms
</p>
<p>P
a
</p>
<p>rt
ia
</p>
<p>l 
D
</p>
<p>e
p
</p>
<p>e
n
</p>
<p>d
e
</p>
<p>n
c
e
</p>
<p>4 6 8 10
</p>
<p>-1
.0
</p>
<p>-0
.5
</p>
<p>0
.0
</p>
<p>0
.5
</p>
<p>1
.0
</p>
<p>1
.5
</p>
<p>FIGURE 10.15. Partial dependence of housing value on the nonlocation vari-
ables for the California housing data. The red ticks at the base of the plot are
deciles of the input variables.
</p>
<p>2
</p>
<p>3
</p>
<p>4
</p>
<p>5
10
</p>
<p>20
</p>
<p>30
</p>
<p>40
</p>
<p>50
</p>
<p>0.0
</p>
<p>0.5
</p>
<p>1.0
</p>
<p>AveOccup
</p>
<p>HouseAge
</p>
<p>FIGURE 10.16. Partial dependence of house value on median age and aver-
age occupancy. There appears to be a strong interaction effect between these two
variables.</p>
<p/>
</div>
<div class="page"><p/>
<p>10.14 Illustrations 375
</p>
<p>&minus;124 &minus;122 &minus;120 &minus;118 &minus;116 &minus;114
</p>
<p>3
4
</p>
<p>3
6
</p>
<p>3
8
</p>
<p>4
0
</p>
<p>4
2
</p>
<p>Longitude
</p>
<p>L
a
</p>
<p>ti
tu
</p>
<p>d
e
</p>
<p>&minus;1.0
</p>
<p>&minus;0.5
</p>
<p> 0.0
</p>
<p> 0.5
</p>
<p> 1.0
</p>
<p>FIGURE 10.17. Partial dependence of median house value on location in Cal-
ifornia. One unit is $100, 000, at 1990 prices, and the values plotted are relative
to the overall median of $180, 000.
</p>
<p>gions. In the northern, central valley, and southeastern desert regions of
California, location costs considerably less.
</p>
<p>10.14.2 New Zealand Fish
</p>
<p>Plant and animal ecologists use regression models to predict species pres-
ence, abundance and richness as a function of environmental variables.
Although for many years simple linear and parametric models were popu-
lar, recent literature shows increasing interest in more sophisticated mod-
els such as generalized additive models (Section 9.1, GAM), multivariate
adaptive regression splines (Section 9.4, MARS) and boosted regression
trees (Leathwick et al., 2005; Leathwick et al., 2006). Here we model the</p>
<p/>
</div>
<div class="page"><p/>
<p>376 10. Boosting and Additive Trees
</p>
<p>presence and abundance of the Black Oreo Dory, a marine fish found in the
oceanic waters around New Zealand.3
</p>
<p>Figure 10.18 shows the locations of 17,000 trawls (deep-water net fishing,
with a maximum depth of 2km), and the red points indicate those 2353
trawls for which the Black Oreo was present, one of over a hundred species
regularly recorded. The catch size in kg for each species was recorded for
each trawl. Along with the species catch, a number of environmental mea-
surements are available for each trawl. These include the average depth of
the trawl (AvgDepth), and the temperature and salinity of the water. Since
the latter two are strongly correlated with depth, Leathwick et al. (2006)
derived instead TempResid and SalResid, the residuals obtained when these
two measures are adjusted for depth (via separate non-parametric regres-
sions). SSTGrad is a measure of the gradient of the sea surface temperature,
and Chla is a broad indicator of ecosytem productivity via satellite-image
measurements. SusPartMatter provides a measure of suspended particulate
matter, particularly in coastal waters, and is also satellite derived.
The goal of this analysis is to estimate the probability of finding Black
</p>
<p>Oreo in a trawl, as well as the expected catch size, standardized to take
into account the effects of variation in trawl speed and distance, as well
as the mesh size of the trawl net. The authors used logistic regression
for estimating the probability. For the catch size, it might seem natural
to assume a Poisson distribution and model the log of the mean count,
but this is often not appropriate because of the excessive number of zeros.
Although specialized approaches have been developed, such as the zero-
inflated Poisson (Lambert, 1992), they chose a simpler approach. If Y is
the (non-negative) catch size,
</p>
<p>E(Y |X) = E(Y |Y &gt; 0, X) &middot; Pr(Y &gt; 0|X). (10.54)
</p>
<p>The second term is estimated by the logistic regression, and the first term
can be estimated using only the 2353 trawls with a positive catch.
For the logistic regression the authors used a gradient boosted model
</p>
<p>(GBM)4 with binomial deviance loss function, depth-10 trees, and a shrink-
age factor ν = 0.025. For the positive-catch regression, they modeled
log(Y ) using a GBM with squared-error loss (also depth-10 trees, but
ν = 0.01), and un-logged the predictions. In both cases they used 10-fold
cross-validation for selecting the number of terms, as well as the shrinkage
factor.
</p>
<p>3The models, data, and maps shown here were kindly provided by Dr John Leathwick
of the National Institute of Water and Atmospheric Research in New Zealand, and Dr
Jane Elith, School of Botany, University of Melbourne. The collection of the research
trawl data took place from 1979&ndash;2005, and was funded by the New Zealand Ministry of
Fisheries.
</p>
<p>4Version 1.5-7 of package gbm in R, ver. 2.2.0.</p>
<p/>
</div>
<div class="page"><p/>
<p>10.14 Illustrations 377
</p>
<p>FIGURE 10.18. Map of New Zealand and its surrounding exclusive economic
zone, showing the locations of 17,000 trawls (small blue dots) taken between 1979
and 2005. The red points indicate trawls for which the species Black Oreo Dory
were present.</p>
<p/>
</div>
<div class="page"><p/>
<p>378 10. Boosting and Additive Trees
</p>
<p>0 500 1000 1500
</p>
<p>0
.2
</p>
<p>4
0
.2
</p>
<p>6
0
.2
</p>
<p>8
0
.3
</p>
<p>0
0
.3
</p>
<p>2
0
.3
</p>
<p>4
</p>
<p>Number of Trees
</p>
<p>M
e
a
n
 D
</p>
<p>e
v
ia
</p>
<p>n
c
e
</p>
<p>GBM Test
GBM CV
GAM Test
</p>
<p>0.0 0.2 0.4 0.6 0.8 1.0
</p>
<p>0
.0
</p>
<p>0
.2
</p>
<p>0
.4
</p>
<p>0
.6
</p>
<p>0
.8
</p>
<p>1
.0
</p>
<p>Specificity
S
</p>
<p>e
n
s
it
iv
</p>
<p>it
y
</p>
<p>AUC
</p>
<p>GAM 0.97
GBM 0.98
</p>
<p>FIGURE 10.19. The left panel shows the mean deviance as a function of the
number of trees for the GBM logistic regression model fit to the presence/absence
data. Shown are 10-fold cross-validation on the training data (and 1&times; s.e. bars),
and test deviance on the test data. Also shown for comparison is the test deviance
using a GAM model with 8 df for each term. The right panel shows ROC curves
on the test data for the chosen GBM model (vertical line in left plot) and the
GAM model.
</p>
<p>Figure 10.19 (left panel) shows the mean binomial deviance for the se-
quence of GBM models, both for 10-fold CV and test data. There is a mod-
est improvement over the performance of a GAMmodel, fit using smoothing
splines with 8 degrees-of-freedom (df) per term. The right panel shows the
ROC curves (see Section 9.2.5) for both models, which measures predictive
performance. From this point of view, the performance looks very simi-
lar, with GBM perhaps having a slight edge as summarized by the AUC
(area under the curve). At the point of equal sensitivity/specificity, GBM
achieves 91%, and GAM 90%.
Figure 10.20 summarizes the contributions of the variables in the logistic
</p>
<p>GBM fit. We see that there is a well-defined depth range over which Black
Oreo are caught, with much more frequent capture in colder waters. We do
not give details of the quantitative catch model; the important variables
were much the same.
All the predictors used in these models are available on a fine geographi-
</p>
<p>cal grid; in fact they were derived from environmental atlases, satellite im-
ages and the like&mdash;see Leathwick et al. (2006) for details. This also means
that predictions can be made on this grid, and imported into GIS mapping
systems. Figure 10.21 shows prediction maps for both presence and catch
size, with both standardized to a common set of trawl conditions; since the
predictors vary in a continuous fashion with geographical location, so do
the predictions.</p>
<p/>
</div>
<div class="page"><p/>
<p>10.14 Illustrations 379
</p>
<p>OrbVel
Speed
</p>
<p>Distance
DisOrgMatter
CodendSize
</p>
<p>Pentade
TidalCurr
</p>
<p>Slope
ChlaCase2
</p>
<p>SSTGrad
SalResid
</p>
<p>SusPartMatter
AvgDepth
</p>
<p>TempResid
</p>
<p>Relative influence
</p>
<p>0 10 25 &minus;4 0 2 4 6
</p>
<p>&minus;
7
</p>
<p>&minus;
5
</p>
<p>&minus;
3
</p>
<p>&minus;
1
</p>
<p>TempResid
f(
</p>
<p>T
e
</p>
<p>m
p
</p>
<p>R
e
</p>
<p>s
id
</p>
<p>)
</p>
<p>0 500 1000 2000
</p>
<p>&minus;
6
</p>
<p>&minus;
4
</p>
<p>&minus;
2
</p>
<p>AvgDepth
</p>
<p>f(
A
</p>
<p>v
g
</p>
<p>D
e
</p>
<p>p
th
</p>
<p>)
0 5 10 15
</p>
<p>&minus;
7
</p>
<p>&minus;
5
</p>
<p>&minus;
3
</p>
<p>SusPartMatter
</p>
<p>f(
S
</p>
<p>u
s
P
</p>
<p>a
rt
</p>
<p>M
a
</p>
<p>tt
e
</p>
<p>r)
</p>
<p>&minus;0.8 &minus;0.4 0.0 0.4
</p>
<p>&minus;
7
</p>
<p>&minus;
5
</p>
<p>&minus;
3
</p>
<p>&minus;
1
</p>
<p>SalResid
</p>
<p>f(
S
</p>
<p>a
lR
</p>
<p>e
s
id
</p>
<p>)
</p>
<p>0.00 0.05 0.10 0.15
</p>
<p>&minus;
7
</p>
<p>&minus;
5
</p>
<p>&minus;
3
</p>
<p>&minus;
1
</p>
<p>SSTGrad
</p>
<p>f(
S
</p>
<p>S
T
</p>
<p>G
ra
</p>
<p>d
)
</p>
<p>FIGURE 10.20. The top-left panel shows the relative influence computed from
the GBM logistic regression model. The remaining panels show the partial de-
pendence plots for the leading five variables, all plotted on the same scale for
comparison.
</p>
<p>Because of their ability to model interactions and automatically select
variables, as well as robustness to outliers and missing data, GBM models
are rapidly gaining popularity in this data-rich and enthusiastic community.
</p>
<p>10.14.3 Demographics Data
</p>
<p>In this section we illustrate gradient boosting on a multiclass classifica-
tion problem, using MART. The data come from 9243 questionnaires filled
out by shopping mall customers in the San Francisco Bay Area (Impact
Resources, Inc., Columbus, OH). Among the questions are 14 concerning
demographics. For this illustration the goal is to predict occupation us-
ing the other 13 variables as predictors, and hence identify demographic
variables that discriminate between different occupational categories. We
randomly divided the data into a training set (80%) and test set (20%),
and used J = 6 node trees with a learning rate ν = 0.1.
Figure 10.22 shows the K = 9 occupation class values along with their
</p>
<p>corresponding error rates. The overall error rate is 42.5%, which can be
compared to the null rate of 69% obtained by predicting the most numerous</p>
<p/>
</div>
<div class="page"><p/>
<p>380 10. Boosting and Additive Trees
</p>
<p>FIGURE 10.21. Geological prediction maps of the presence probability (left
map) and catch size (right map) obtained from the gradient boosted models.
</p>
<p>class Prof/Man (Professional/Managerial). The four best predicted classes
are seen to be Retired, Student, Prof/Man, and Homemaker.
</p>
<p>Figure 10.23 shows the relative predictor variable importances as aver-
aged over all classes (10.46). Figure 10.24 displays the individual relative
importance distributions (10.45) for each of the four best predicted classes.
One sees that the most relevant predictors are generally different for each
respective class. An exception is age which is among the three most relevant
for predicting Retired, Student, and Prof/Man.
</p>
<p>Figure 10.25 shows the partial dependence of the log-odds (10.52) on age
for these three classes. The abscissa values are ordered codes for respective
equally spaced age intervals. One sees that after accounting for the contri-
butions of the other variables, the odds of being retired are higher for older
people, whereas the opposite is the case for being a student. The odds of
being professional/managerial are highest for middle-aged people. These
results are of course not surprising. They illustrate that inspecting partial
dependences separately for each class can lead to sensible results.
</p>
<p>Bibliographic Notes
</p>
<p>Schapire (1990) developed the first simple boosting procedure in the PAC
learning framework (Valiant, 1984; Kearns and Vazirani, 1994). Schapire</p>
<p/>
</div>
<div class="page"><p/>
<p>10.14 Illustrations 381
</p>
<p>Sales
</p>
<p>Unemployed
</p>
<p>Military
</p>
<p>Clerical
</p>
<p>Labor
</p>
<p>Homemaker
</p>
<p>Prof/Man
</p>
<p>Retired
</p>
<p>Student
</p>
<p>0.0 0.2 0.4 0.6 0.8 1.0
</p>
<p>Error Rate
</p>
<p>Overall Error Rate = 0.425
</p>
<p>FIGURE 10.22. Error rate for each occupation in the demographics data.
</p>
<p>age
</p>
<p>income
</p>
<p>edu
</p>
<p>hsld-stat
</p>
<p>mar-dlinc
</p>
<p>sex
</p>
<p>ethnic
</p>
<p>mar-stat
</p>
<p>typ-home
</p>
<p>lang
</p>
<p>num-hsld
</p>
<p>children
</p>
<p>yrs-BA
</p>
<p>0 20 40 60 80 100
</p>
<p>Relative Importance
</p>
<p>FIGURE 10.23. Relative importance of the predictors as averaged over all
classes for the demographics data.</p>
<p/>
</div>
<div class="page"><p/>
<p>382 10. Boosting and Additive Trees
</p>
<p>age
mar-dlinc
</p>
<p>sex
ethnic
</p>
<p>income
hsld-stat
mar-stat
</p>
<p>lang
typ-home
</p>
<p>children
edu
</p>
<p>num-hsld
yrs-BA
</p>
<p>0 20 40 60 80 100
</p>
<p>Relative Importance
</p>
<p>Class =  Retired
</p>
<p>hsld-stat
age
</p>
<p>income
mar-stat
</p>
<p>edu
ethnic
</p>
<p>num-hsld
typ-home
</p>
<p>sex
mar-dlinc
</p>
<p>lang
yrs-BA
</p>
<p>children
</p>
<p>0 20 40 60 80 100
</p>
<p>Relative Importance
</p>
<p>Class =  Student
</p>
<p>edu
income
</p>
<p>age
mar-dlinc
</p>
<p>ethnic
hsld-stat
</p>
<p>typ-home
sex
</p>
<p>num-hsld
lang
</p>
<p>mar-stat
yrs-BA
</p>
<p>children
</p>
<p>0 20 40 60 80 100
</p>
<p>Relative Importance
</p>
<p>Class =  Prof/Man
</p>
<p>sex
mar-dlinc
</p>
<p>children
ethnic
</p>
<p>num-hsld
edu
</p>
<p>mar-stat
lang
</p>
<p>typ-home
income
</p>
<p>age
hsld-stat
</p>
<p>yrs-BA
</p>
<p>0 20 40 60 80 100
</p>
<p>Relative Importance
</p>
<p>Class =  Homemaker
</p>
<p>FIGURE 10.24. Predictor variable importances separately for each of the four
classes with lowest error rate for the demographics data.</p>
<p/>
</div>
<div class="page"><p/>
<p>10.14 Illustrations 383
</p>
<p>age
</p>
<p>P
a
</p>
<p>rt
ia
</p>
<p>l 
D
</p>
<p>e
p
</p>
<p>e
n
</p>
<p>d
e
</p>
<p>n
c
e
</p>
<p>1 2 3 4 5 6 7
</p>
<p>0
1
</p>
<p>2
3
</p>
<p>4
</p>
<p>Retired
</p>
<p>age
</p>
<p>P
a
</p>
<p>rt
ia
</p>
<p>l 
D
</p>
<p>e
p
</p>
<p>e
n
</p>
<p>d
e
</p>
<p>n
c
e
</p>
<p>1 2 3 4 5 6 7
</p>
<p>-2
-1
</p>
<p>0
1
</p>
<p>2
</p>
<p>Student
</p>
<p>age
</p>
<p>P
a
</p>
<p>rt
ia
</p>
<p>l 
D
</p>
<p>e
p
</p>
<p>e
n
</p>
<p>d
e
</p>
<p>n
c
e
</p>
<p>1 2 3 4 5 6 7
</p>
<p>-2
-1
</p>
<p>0
1
</p>
<p>2
</p>
<p>Prof/Man
</p>
<p>FIGURE 10.25. Partial dependence of the odds of three different occupations
on age, for the demographics data.
</p>
<p>showed that a weak learner could always improve its performance by train-
ing two additional classifiers on filtered versions of the input data stream.
A weak learner is an algorithm for producing a two-class classifier with
performance guaranteed (with high probability) to be significantly better
than a coin-flip. After learning an initial classifier G1 on the first N training
points,
</p>
<p>&bull; G2 is learned on a new sample of N points, half of which are misclas-
sified by G1;
</p>
<p>&bull; G3 is learned on N points for which G1 and G2 disagree;
</p>
<p>&bull; the boosted classifier is GB = majority vote(G1, G2, G3).
</p>
<p>Schapire&rsquo;s &ldquo;Strength of Weak Learnability&rdquo; theorem proves that GB has
improved performance over G1.
Freund (1995) proposed a &ldquo;boost by majority&rdquo; variation which combined
</p>
<p>many weak learners simultaneously and improved the performance of the
simple boosting algorithm of Schapire. The theory supporting both of these</p>
<p/>
</div>
<div class="page"><p/>
<p>384 10. Boosting and Additive Trees
</p>
<p>algorithms requires the weak learner to produce a classifier with a fixed
error rate. This led to the more adaptive and realistic AdaBoost (Freund
and Schapire, 1996a) and its offspring, where this assumption was dropped.
Freund and Schapire (1996a) and Schapire and Singer (1999) provide
</p>
<p>some theory to support their algorithms, in the form of upper bounds on
generalization error. This theory has evolved in the computational learning
community, initially based on the concepts of PAC learning. Other theo-
ries attempting to explain boosting come from game theory (Freund and
Schapire, 1996b; Breiman, 1999; Breiman, 1998), and VC theory (Schapire
et al., 1998). The bounds and the theory associated with the AdaBoost
algorithms are interesting, but tend to be too loose to be of practical im-
portance. In practice, boosting achieves results far more impressive than
the bounds would imply. Schapire (2002) and Meir and Rätsch (2003) give
useful overviews more recent than the first edition of this book.
Friedman et al. (2000) and Friedman (2001) form the basis for our expo-
</p>
<p>sition in this chapter. Friedman et al. (2000) analyze AdaBoost statistically,
derive the exponential criterion, and show that it estimates the log-odds
of the class probability. They propose additive tree models, the right-sized
trees and ANOVA representation of Section 10.11, and the multiclass logit
formulation. Friedman (2001) developed gradient boosting and shrinkage
for classification and regression, while Friedman (1999) explored stochastic
variants of boosting. Mason et al. (2000) also embraced a gradient approach
to boosting. As the published discussions of Friedman et al. (2000) shows,
there is some controversy about how and why boosting works.
Since the publication of the first edition of this book, these debates have
</p>
<p>continued, and spread into the statistical community with a series of papers
on consistency of boosting (Jiang, 2004; Lugosi and Vayatis, 2004; Zhang
and Yu, 2005; Bartlett and Traskin, 2007). Mease and Wyner (2008),
through a series of simulation examples, challenge some of our interpre-
tations of boosting; our response (Friedman et al., 2008a) puts most of
these objections to rest. A recent survey by Bühlmann and Hothorn (2007)
supports our approach to boosting.
</p>
<p>Exercises
</p>
<p>Ex. 10.1 Derive expression (10.12) for the update parameter in AdaBoost.
</p>
<p>Ex. 10.2 Prove result (10.16), that is, the minimizer of the population
version of the AdaBoost criterion, is one-half of the log odds.
</p>
<p>Ex. 10.3 Show that the marginal average (10.47) recovers additive and
multiplicative functions (10.50) and (10.51), while the conditional expec-
tation (10.49) does not.</p>
<p/>
</div>
<div class="page"><p/>
<p>Exercises 385
</p>
<p>Ex. 10.4
</p>
<p>(a) Write a program implementing AdaBoost with trees.
</p>
<p>(b) Redo the computations for the example of Figure 10.2. Plot the train-
ing error as well as test error, and discuss its behavior.
</p>
<p>(c) Investigate the number of iterations needed to make the test error
finally start to rise.
</p>
<p>(d) Change the setup of this example as follows: define two classes, with
the features in Class 1 being X1, X2, . . . , X10, standard indepen-
dent Gaussian variates. In Class 2, the features X1, X2, . . . , X10 are
also standard independent Gaussian, but conditioned on the event&sum;
</p>
<p>j X
2
j &gt; 12. Now the classes have significant overlap in feature space.
</p>
<p>Repeat the AdaBoost experiments as in Figure 10.2 and discuss the
results.
</p>
<p>Ex. 10.5 Multiclass exponential loss (Zhu et al., 2005). For a K-class clas-
sification problem, consider the coding Y = (Y1, . . . , YK)
</p>
<p>T with
</p>
<p>Yk =
</p>
<p>{
1, if G = Gk
&minus; 1K&minus;1 , otherwise.
</p>
<p>(10.55)
</p>
<p>Let f = (f1, . . . , fK)
T with
</p>
<p>&sum;K
k=1 fk = 0, and define
</p>
<p>L(Y, f) = exp
</p>
<p>(
&minus; 1
K
</p>
<p>Y T f
</p>
<p>)
. (10.56)
</p>
<p>(a) Using Lagrange multipliers, derive the population minimizer f&lowast; of
L(Y, f), subject to the zero-sum constraint, and relate these to the
class probabilities.
</p>
<p>(b) Show that a multiclass boosting using this loss function leads to a
reweighting algorithm similar to Adaboost, as in Section 10.4.
</p>
<p>Ex. 10.6 McNemar test (Agresti, 1996). We report the test error rates on
the spam data to be 5.5% for a generalized additive model (GAM), and
4.5% for gradient boosting (GBM), with a test sample of size 1536.
</p>
<p>(a) Show that the standard error of these estimates is about 0.6%.
</p>
<p>Since the same test data are used for both methods, the error rates are
correlated, and we cannot perform a two-sample t-test. We can compare
the methods directly on each test observation, leading to the summary
</p>
<p>GBM
GAM Correct Error
</p>
<p>Correct 1434 18
Error 33 51</p>
<p/>
</div>
<div class="page"><p/>
<p>386 10. Boosting and Additive Trees
</p>
<p>The McNemar test focuses on the discordant errors, 33 vs. 18.
</p>
<p>(b) Conduct a test to show that GAM makes significantly more errors
than gradient boosting, with a two-sided p-value of 0.036.
</p>
<p>Ex. 10.7 Derive expression (10.32).
</p>
<p>Ex. 10.8 Consider a K-class problem where the targets yik are coded as
1 if observation i is in class k and zero otherwise. Suppose we have a
current model fk(x), k = 1, . . . ,K, with
</p>
<p>&sum;K
k=1 fk(x) = 0 (see (10.21) in
</p>
<p>Section 10.6). We wish to update the model for observations in a region R
in predictor space, by adding constants fk(x) + γk, with γK = 0.
</p>
<p>(a) Write down the multinomial log-likelihood for this problem, and its
first and second derivatives.
</p>
<p>(b) Using only the diagonal of the Hessian matrix in (1), and starting
from γk = 0 &forall;k, show that a one-step approximate Newton update
for γk is
</p>
<p>γ1k =
</p>
<p>&sum;
xi&isin;R
</p>
<p>(yik &minus; pik)&sum;
xi&isin;R
</p>
<p>pik(1&minus; pik)
, k = 1, . . . ,K &minus; 1, (10.57)
</p>
<p>where pik = exp(fk(xi))/ exp(
&sum;K
</p>
<p>ℓ=1 fℓ(xi)).
</p>
<p>(c) We prefer our update to sum to zero, as the current model does. Using
symmetry arguments, show that
</p>
<p>γ̂k =
K &minus; 1
K
</p>
<p>(γ1k &minus;
1
</p>
<p>K
</p>
<p>K&sum;
</p>
<p>ℓ=1
</p>
<p>γ1ℓ ), k = 1, . . . ,K (10.58)
</p>
<p>is an appropriate update, where γ1k is defined as in (10.57) for all
k = 1, . . . ,K.
</p>
<p>Ex. 10.9 Consider a K-class problem where the targets yik are coded as
1 if observation i is in class k and zero otherwise. Using the multinomial
deviance loss function (10.22) and the symmetric logistic transform, use
the arguments leading to the gradient boosting Algorithm 10.3 to derive
Algorithm 10.4. Hint: See exercise 10.8 for step 2(b)iii.
</p>
<p>Ex. 10.10 Show that for K = 2 class classification, only one tree needs to
be grown at each gradient-boosting iteration.
</p>
<p>Ex. 10.11 Show how to compute the partial dependence function fS(XS)
in (10.47) efficiently.
</p>
<p>Ex. 10.12 Referring to (10.49), let S = {1} and C = {2}, with f(X1, X2) =
X1. Assume X1 and X2 are bivariate Gaussian, each with mean zero, vari-
ance one, and E(X1X2) = ρ. Show that E(f(X1, X2)|X2) = ρX2, even
though f is not a function of X2.</p>
<p/>
</div>
<div class="page"><p/>
<p>Exercises 387
</p>
<p>Algorithm 10.4 Gradient Boosting for K-class Classification.
</p>
<p>1. Initialize fk0(x) = 0, k = 1, 2, . . . ,K.
</p>
<p>2. For m=1 to M :
</p>
<p>(a) Set
</p>
<p>pk(x) =
efk(x)
</p>
<p>&sum;K
ℓ=1 e
</p>
<p>fℓ(x)
, k = 1, 2, . . . ,K.
</p>
<p>(b) For k = 1 to K:
</p>
<p>i. Compute rikm = yik &minus; pk(xi), i = 1, 2, . . . , N .
ii. Fit a regression tree to the targets rikm, i = 1, 2, . . . , N ,
</p>
<p>giving terminal regions Rjkm, j = 1, 2, . . . , Jm.
</p>
<p>iii. Compute
</p>
<p>γjkm =
K &minus; 1
K
</p>
<p>&sum;
xi&isin;Rjkm
</p>
<p>rikm&sum;
xi&isin;Rjkm
</p>
<p>|rikm|(1&minus; |rikm|)
, j = 1, 2, . . . , Jm.
</p>
<p>iv. Update fkm(x) = fk,m&minus;1(x) +
&sum;Jm
</p>
<p>j=1 γjkmI(x &isin; Rjkm).
</p>
<p>3. Output f̂k(x) = fkM (x), k = 1, 2, . . . ,K.</p>
<p/>
</div>
<div class="page"><p/>
<p>11
Neural Networks
</p>
<p>11.1 Introduction
</p>
<p>In this chapter we describe a class of learning methods that was developed
separately in different fields&mdash;statistics and artificial intelligence&mdash;based
on essentially identical models. The central idea is to extract linear com-
binations of the inputs as derived features, and then model the target as
a nonlinear function of these features. The result is a powerful learning
method, with widespread applications in many fields. We first discuss the
projection pursuit model, which evolved in the domain of semiparamet-
ric statistics and smoothing. The rest of the chapter is devoted to neural
network models.
</p>
<p>11.2 Projection Pursuit Regression
</p>
<p>As in our generic supervised learning problem, assume we have an input
vector X with p components, and a target Y . Let ωm, m = 1, 2, . . . ,M, be
unit p-vectors of unknown parameters. The projection pursuit regression
(PPR) model has the form
</p>
<p>f(X) =
M&sum;
</p>
<p>m=1
</p>
<p>gm(ω
T
mX). (11.1)
</p>
<p>This is an additive model, but in the derived features Vm = ω
T
mX rather
</p>
<p>than the inputs themselves. The functions gm are unspecified and are esti-
</p>
<p> &copy; Springer Science+Business Media, LLC 2009 
</p>
<p>T. Hastie et al., The Elements of Statistical Learning, Second Edition, 389
</p>
<p>DOI: 10.1007/b94608_11,</p>
<p/>
</div>
<div class="page"><p/>
<p>390 Neural Networks
</p>
<p>g(V )
</p>
<p>X1
X2
</p>
<p>g(V )
</p>
<p>X1
X2
</p>
<p>FIGURE 11.1. Perspective plots of two ridge functions.
(Left:) g(V ) = 1/[1 + exp(&minus;5(V &minus; 0.5))], where V = (X1 +X2)/
</p>
<p>&radic;
2.
</p>
<p>(Right:) g(V ) = (V + 0.1) sin(1/(V/3 + 0.1)), where V = X1.
</p>
<p>mated along with the directions ωm using some flexible smoothing method
(see below).
The function gm(ω
</p>
<p>T
mX) is called a ridge function in IR
</p>
<p>p. It varies only
in the direction defined by the vector ωm. The scalar variable Vm = ω
</p>
<p>T
mX
</p>
<p>is the projection of X onto the unit vector ωm, and we seek ωm so that
the model fits well, hence the name &ldquo;projection pursuit.&rdquo; Figure 11.1 shows
some examples of ridge functions. In the example on the left ω=(1/
</p>
<p>&radic;
2)(1, 1)T ,
</p>
<p>so that the function only varies in the direction X1 +X2. In the example
on the right, ω = (1, 0).
</p>
<p>The PPR model (11.1) is very general, since the operation of forming
nonlinear functions of linear combinations generates a surprisingly large
class of models. For example, the product X1 &middot;X2 can be written as [(X1+
X2)
</p>
<p>2 &minus; (X1 &minus;X2)2]/4, and higher-order products can be represented simi-
larly.
In fact, if M is taken arbitrarily large, for appropriate choice of gm the
</p>
<p>PPR model can approximate any continuous function in IRp arbitrarily
well. Such a class of models is called a universal approximator. However
this generality comes at a price. Interpretation of the fitted model is usually
difficult, because each input enters into the model in a complex and multi-
faceted way. As a result, the PPR model is most useful for prediction, and
not very useful for producing an understandable model for the data. The
M = 1 model, known as the single index model in econometrics, is an
exception. It is slightly more general than the linear regression model, and
offers a similar interpretation.
How do we fit a PPR model, given training data (xi, yi), i = 1, 2, . . . , N?
</p>
<p>We seek the approximate minimizers of the error function
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>[
yi &minus;
</p>
<p>M&sum;
</p>
<p>m=1
</p>
<p>gm(ω
T
mxi)
</p>
<p>]2
(11.2)</p>
<p/>
</div>
<div class="page"><p/>
<p>11.2 Projection Pursuit Regression 391
</p>
<p>over functions gm and direction vectors ωm, m = 1, 2, . . . ,M . As in other
smoothing problems, we need either explicitly or implicitly to impose com-
plexity constraints on the gm, to avoid overfit solutions.
Consider just one term (M = 1, and drop the subscript). Given the
</p>
<p>direction vector ω, we form the derived variables vi = ω
Txi. Then we have
</p>
<p>a one-dimensional smoothing problem, and we can apply any scatterplot
smoother, such as a smoothing spline, to obtain an estimate of g.
On the other hand, given g, we want to minimize (11.2) over ω. A Gauss&ndash;
</p>
<p>Newton search is convenient for this task. This is a quasi-Newton method,
in which the part of the Hessian involving the second derivative of g is
discarded. It can be simply derived as follows. Let ωold be the current
estimate for ω. We write
</p>
<p>g(ωTxi) &asymp; g(ωToldxi) + g&prime;(ωToldxi)(ω &minus; ωold)Txi (11.3)
</p>
<p>to give
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>[
yi &minus; g(ωTxi)
</p>
<p>]2 &asymp;
N&sum;
</p>
<p>i=1
</p>
<p>g&prime;(ωToldxi)
2
</p>
<p>[(
ωToldxi+
</p>
<p>yi &minus; g(ωToldxi)
g&prime;(ωToldxi)
</p>
<p>)
&minus;ωTxi
</p>
<p>]2
.
</p>
<p>(11.4)
To minimize the right-hand side, we carry out a least squares regression
</p>
<p>with target ωToldxi+(yi&minus;g(ωToldxi))/g&prime;(ωToldxi) on the input xi, with weights
g&prime;(ωToldxi)
</p>
<p>2 and no intercept (bias) term. This produces the updated coef-
ficient vector ωnew.
These two steps, estimation of g and ω, are iterated until convergence.
</p>
<p>With more than one term in the PPR model, the model is built in a forward
stage-wise manner, adding a pair (ωm, gm) at each stage.
</p>
<p>There are a number of implementation details.
</p>
<p>&bull; Although any smoothing method can in principle be used, it is conve-
nient if the method provides derivatives. Local regression and smooth-
ing splines are convenient.
</p>
<p>&bull; After each step the gm&rsquo;s from previous steps can be readjusted using
the backfitting procedure described in Chapter 9. While this may
lead ultimately to fewer terms, it is not clear whether it improves
prediction performance.
</p>
<p>&bull; Usually the ωm are not readjusted (partly to avoid excessive compu-
tation), although in principle they could be as well.
</p>
<p>&bull; The number of terms M is usually estimated as part of the forward
stage-wise strategy. The model building stops when the next term
does not appreciably improve the fit of the model. Cross-validation
can also be used to determine M .</p>
<p/>
</div>
<div class="page"><p/>
<p>392 Neural Networks
</p>
<p>There are many other applications, such as density estimation (Friedman
et al., 1984; Friedman, 1987), where the projection pursuit idea can be used.
In particular, see the discussion of ICA in Section 14.7 and its relationship
with exploratory projection pursuit. However the projection pursuit re-
gression model has not been widely used in the field of statistics, perhaps
because at the time of its introduction (1981), its computational demands
exceeded the capabilities of most readily available computers. But it does
represent an important intellectual advance, one that has blossomed in its
reincarnation in the field of neural networks, the topic of the rest of this
chapter.
</p>
<p>11.3 Neural Networks
</p>
<p>The term neural network has evolved to encompass a large class of models
and learning methods. Here we describe the most widely used &ldquo;vanilla&rdquo; neu-
ral net, sometimes called the single hidden layer back-propagation network,
or single layer perceptron. There has been a great deal of hype surrounding
neural networks, making them seem magical and mysterious. As we make
clear in this section, they are just nonlinear statistical models, much like
the projection pursuit regression model discussed above.
A neural network is a two-stage regression or classification model, typ-
</p>
<p>ically represented by a network diagram as in Figure 11.2. This network
applies both to regression or classification. For regression, typically K = 1
and there is only one output unit Y1 at the top. However, these networks
can handle multiple quantitative responses in a seamless fashion, so we will
deal with the general case.
For K-class classification, there are K units at the top, with the kth
</p>
<p>unit modeling the probability of class k. There are K target measurements
Yk, k = 1, . . . ,K, each being coded as a 0&minus; 1 variable for the kth class.
Derived features Zm are created from linear combinations of the inputs,
</p>
<p>and then the target Yk is modeled as a function of linear combinations of
the Zm,
</p>
<p>Zm = σ(α0m + α
T
mX), m = 1, . . . ,M,
</p>
<p>Tk = β0k + β
T
k Z, k = 1, . . . ,K,
</p>
<p>fk(X) = gk(T ), k = 1, . . . ,K,
</p>
<p>(11.5)
</p>
<p>where Z = (Z1, Z2, . . . , ZM ), and T = (T1, T2, . . . , TK).
The activation function σ(v) is usually chosen to be the sigmoid σ(v) =
</p>
<p>1/(1+ e&minus;v); see Figure 11.3 for a plot of 1/(1 + e&minus;v). Sometimes Gaussian
radial basis functions (Chapter 6) are used for the σ(v), producing what is
known as a radial basis function network.
Neural network diagrams like Figure 11.2 are sometimes drawn with an
</p>
<p>additional bias unit feeding into every unit in the hidden and output layers.</p>
<p/>
</div>
<div class="page"><p/>
<p>11.3 Neural Networks 393
</p>
<p> Y Y Y 21 K
</p>
<p> Z Z Z1  Z2 3 m
</p>
<p> X X
</p>
<p> Z Z1  Z2 3
</p>
<p>1  Xp X p-1 X2  X3
</p>
<p>M
</p>
<p> X p-13 X2 X1 p
</p>
<p> Z
</p>
<p> Y Y Y
</p>
<p> X
</p>
<p>K1 2
</p>
<p>                                                                                                                                                
</p>
<p>�
�
�
�
�
�
�
�
�
�
�
�
�
�
�
�
�
�
�
�
�
�
�
�
</p>
<p>FIGURE 11.2. Schematic of a single hidden layer, feed-forward neural network.
</p>
<p>Thinking of the constant &ldquo;1&rdquo; as an additional input feature, this bias unit
captures the intercepts α0m and β0k in model (11.5).
</p>
<p>The output function gk(T ) allows a final transformation of the vector of
outputs T . For regression we typically choose the identity function gk(T ) =
Tk. Early work in K-class classification also used the identity function, but
this was later abandoned in favor of the softmax function
</p>
<p>gk(T ) =
eTk
</p>
<p>&sum;K
ℓ=1 e
</p>
<p>Tℓ
. (11.6)
</p>
<p>This is of course exactly the transformation used in the multilogit model
(Section 4.4), and produces positive estimates that sum to one. In Sec-
tion 4.2 we discuss other problems with linear activation functions, in par-
ticular potentially severe masking effects.
The units in the middle of the network, computing the derived features
</p>
<p>Zm, are called hidden units because the values Zm are not directly ob-
served. In general there can be more than one hidden layer, as illustrated
in the example at the end of this chapter. We can think of the Zm as a
basis expansion of the original inputs X; the neural network is then a stan-
dard linear model, or linear multilogit model, using these transformations
as inputs. There is, however, an important enhancement over the basis-
expansion techniques discussed in Chapter 5; here the parameters of the
basis functions are learned from the data.</p>
<p/>
</div>
<div class="page"><p/>
<p>394 Neural Networks
</p>
<p>-10 -5 0 5 10
</p>
<p>0
.0
</p>
<p>0
.5
</p>
<p>1
.0
</p>
<p>1
/
(1
</p>
<p>+
e&minus;
</p>
<p>v
)
</p>
<p>v
</p>
<p>FIGURE 11.3. Plot of the sigmoid function σ(v) = 1/(1+exp(&minus;v)) (red curve),
commonly used in the hidden layer of a neural network. Included are σ(sv) for
s = 1
</p>
<p>2
(blue curve) and s = 10 (purple curve). The scale parameter s controls
</p>
<p>the activation rate, and we can see that large s amounts to a hard activation at
v = 0. Note that σ(s(v &minus; v0)) shifts the activation threshold from 0 to v0.
</p>
<p>Notice that if σ is the identity function, then the entire model collapses
to a linear model in the inputs. Hence a neural network can be thought of
as a nonlinear generalization of the linear model, both for regression and
classification. By introducing the nonlinear transformation σ, it greatly
enlarges the class of linear models. In Figure 11.3 we see that the rate of
activation of the sigmoid depends on the norm of αm, and if ‖αm‖ is very
small, the unit will indeed be operating in the linear part of its activation
function.
Notice also that the neural network model with one hidden layer has
</p>
<p>exactly the same form as the projection pursuit model described above.
The difference is that the PPR model uses nonparametric functions gm(v),
while the neural network uses a far simpler function based on σ(v), with
three free parameters in its argument. In detail, viewing the neural network
model as a PPR model, we identify
</p>
<p>gm(ω
T
mX) = βmσ(α0m + α
</p>
<p>T
mX)
</p>
<p>= βmσ(α0m + ‖αm‖(ωTmX)), (11.7)
</p>
<p>where ωm = αm/‖αm‖ is the mth unit-vector. Since σβ,α0,s(v) = βσ(α0 +
sv) has lower complexity than a more general nonparametric g(v), it is not
surprising that a neural network might use 20 or 100 such functions, while
the PPR model typically uses fewer terms (M = 5 or 10, for example).
</p>
<p>Finally, we note that the name &ldquo;neural networks&rdquo; derives from the fact
that they were first developed as models for the human brain. Each unit
represents a neuron, and the connections (links in Figure 11.2) represent
synapses. In early models, the neurons fired when the total signal passed to
that unit exceeded a certain threshold. In the model above, this corresponds</p>
<p/>
</div>
<div class="page"><p/>
<p>11.4 Fitting Neural Networks 395
</p>
<p>to use of a step function for σ(Z) and gm(T ). Later the neural network was
recognized as a useful tool for nonlinear statistical modeling, and for this
purpose the step function is not smooth enough for optimization. Hence the
step function was replaced by a smoother threshold function, the sigmoid
in Figure 11.3.
</p>
<p>11.4 Fitting Neural Networks
</p>
<p>The neural network model has unknown parameters, often called weights,
and we seek values for them that make the model fit the training data well.
We denote the complete set of weights by θ, which consists of
</p>
<p>{α0m, αm; m = 1, 2, . . . ,M} M(p+ 1) weights,
{β0k, βk; k = 1, 2, . . . ,K} K(M + 1) weights.
</p>
<p>(11.8)
</p>
<p>For regression, we use sum-of-squared errors as our measure of fit (error
function)
</p>
<p>R(θ) =
K&sum;
</p>
<p>k=1
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>(yik &minus; fk(xi))2. (11.9)
</p>
<p>For classification we use either squared error or cross-entropy (deviance):
</p>
<p>R(θ) = &minus;
N&sum;
</p>
<p>i=1
</p>
<p>K&sum;
</p>
<p>k=1
</p>
<p>yik log fk(xi), (11.10)
</p>
<p>and the corresponding classifier is G(x) = argmaxkfk(x). With the softmax
activation function and the cross-entropy error function, the neural network
model is exactly a linear logistic regression model in the hidden units, and
all the parameters are estimated by maximum likelihood.
Typically we don&rsquo;t want the global minimizer of R(θ), as this is likely
</p>
<p>to be an overfit solution. Instead some regularization is needed: this is
achieved directly through a penalty term, or indirectly by early stopping.
Details are given in the next section.
The generic approach to minimizing R(θ) is by gradient descent, called
</p>
<p>back-propagation in this setting. Because of the compositional form of the
model, the gradient can be easily derived using the chain rule for differen-
tiation. This can be computed by a forward and backward sweep over the
network, keeping track only of quantities local to each unit.</p>
<p/>
</div>
<div class="page"><p/>
<p>396 Neural Networks
</p>
<p>Here is back-propagation in detail for squared error loss. Let zmi =
σ(α0m + α
</p>
<p>T
mxi), from (11.5) and let zi = (z1i, z2i, . . . , zMi). Then we have
</p>
<p>R(θ) &equiv;
N&sum;
</p>
<p>i=1
</p>
<p>Ri
</p>
<p>=
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>K&sum;
</p>
<p>k=1
</p>
<p>(yik &minus; fk(xi))2, (11.11)
</p>
<p>with derivatives
</p>
<p>&part;Ri
&part;βkm
</p>
<p>= &minus;2(yik &minus; fk(xi))g&prime;k(βTk zi)zmi,
</p>
<p>&part;Ri
&part;αmℓ
</p>
<p>= &minus;
K&sum;
</p>
<p>k=1
</p>
<p>2(yik &minus; fk(xi))g&prime;k(βTk zi)βkmσ&prime;(αTmxi)xiℓ.
(11.12)
</p>
<p>Given these derivatives, a gradient descent update at the (r + 1)st iter-
ation has the form
</p>
<p>β
(r+1)
km = β
</p>
<p>(r)
km &minus; γr
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>&part;Ri
</p>
<p>&part;β
(r)
km
</p>
<p>,
</p>
<p>α
(r+1)
mℓ = α
</p>
<p>(r)
mℓ &minus; γr
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>&part;Ri
</p>
<p>&part;α
(r)
mℓ
</p>
<p>,
</p>
<p>(11.13)
</p>
<p>where γr is the learning rate, discussed below.
Now write (11.12) as
</p>
<p>&part;Ri
&part;βkm
</p>
<p>= δkizmi,
</p>
<p>&part;Ri
&part;αmℓ
</p>
<p>= smixiℓ.
</p>
<p>(11.14)
</p>
<p>The quantities δki and smi are &ldquo;errors&rdquo; from the current model at the
output and hidden layer units, respectively. From their definitions, these
errors satisfy
</p>
<p>smi = σ
&prime;(αTmxi)
</p>
<p>K&sum;
</p>
<p>k=1
</p>
<p>βkmδki, (11.15)
</p>
<p>known as the back-propagation equations. Using this, the updates in (11.13)
can be implemented with a two-pass algorithm. In the forward pass, the
current weights are fixed and the predicted values f̂k(xi) are computed
from formula (11.5). In the backward pass, the errors δki are computed,
and then back-propagated via (11.15) to give the errors smi. Both sets of
errors are then used to compute the gradients for the updates in (11.13),
via (11.14).</p>
<p/>
</div>
<div class="page"><p/>
<p>11.5 Some Issues in Training Neural Networks 397
</p>
<p>This two-pass procedure is what is known as back-propagation. It has
also been called the delta rule (Widrow and Hoff, 1960). The computational
components for cross-entropy have the same form as those for the sum of
squares error function, and are derived in Exercise 11.3.
The advantages of back-propagation are its simple, local nature. In the
</p>
<p>back propagation algorithm, each hidden unit passes and receives infor-
mation only to and from units that share a connection. Hence it can be
implemented efficiently on a parallel architecture computer.
The updates in (11.13) are a kind of batch learning, with the parame-
</p>
<p>ter updates being a sum over all of the training cases. Learning can also
be carried out online&mdash;processing each observation one at a time, updat-
ing the gradient after each training case, and cycling through the training
cases many times. In this case, the sums in equations (11.13) are replaced
by a single summand. A training epoch refers to one sweep through the
entire training set. Online training allows the network to handle very large
training sets, and also to update the weights as new observations come in.
The learning rate γr for batch learning is usually taken to be a con-
</p>
<p>stant, and can also be optimized by a line search that minimizes the error
function at each update. With online learning γr should decrease to zero
as the iteration r &rarr; &infin;. This learning is a form of stochastic approxima-
tion (Robbins and Munro, 1951); results in this field ensure convergence if
γr &rarr; 0,
</p>
<p>&sum;
r γr = &infin;, and
</p>
<p>&sum;
r γ
</p>
<p>2
r &lt; &infin; (satisfied, for example, by γr = 1/r).
</p>
<p>Back-propagation can be very slow, and for that reason is usually not
the method of choice. Second-order techniques such as Newton&rsquo;s method
are not attractive here, because the second derivative matrix of R (the
Hessian) can be very large. Better approaches to fitting include conjugate
gradients and variable metric methods. These avoid explicit computation
of the second derivative matrix while still providing faster convergence.
</p>
<p>11.5 Some Issues in Training Neural Networks
</p>
<p>There is quite an art in training neural networks. The model is generally
overparametrized, and the optimization problem is nonconvex and unstable
unless certain guidelines are followed. In this section we summarize some
of the important issues.
</p>
<p>11.5.1 Starting Values
</p>
<p>Note that if the weights are near zero, then the operative part of the sigmoid
(Figure 11.3) is roughly linear, and hence the neural network collapses into
an approximately linear model (Exercise 11.2). Usually starting values for
weights are chosen to be random values near zero. Hence the model starts
out nearly linear, and becomes nonlinear as the weights increase. Individual</p>
<p/>
</div>
<div class="page"><p/>
<p>398 Neural Networks
</p>
<p>units localize to directions and introduce nonlinearities where needed. Use
of exact zero weights leads to zero derivatives and perfect symmetry, and
the algorithm never moves. Starting instead with large weights often leads
to poor solutions.
</p>
<p>11.5.2 Overfitting
</p>
<p>Often neural networks have too many weights and will overfit the data at
the global minimum of R. In early developments of neural networks, either
by design or by accident, an early stopping rule was used to avoid over-
fitting. Here we train the model only for a while, and stop well before we
approach the global minimum. Since the weights start at a highly regular-
ized (linear) solution, this has the effect of shrinking the final model toward
a linear model. A validation dataset is useful for determining when to stop,
since we expect the validation error to start increasing.
A more explicit method for regularization is weight decay, which is anal-
</p>
<p>ogous to ridge regression used for linear models (Section 3.4.1). We add a
penalty to the error function R(θ) + λJ(θ), where
</p>
<p>J(θ) =
&sum;
</p>
<p>k,m
</p>
<p>β2km +
&sum;
</p>
<p>m,ℓ
</p>
<p>α2mℓ (11.16)
</p>
<p>and λ &ge; 0 is a tuning parameter. Larger values of λ will tend to shrink
the weights toward zero: typically cross-validation is used to estimate λ.
The effect of the penalty is to simply add terms 2βkm and 2αmℓ to the
respective gradient expressions (11.13). Other forms for the penalty have
been proposed, for example,
</p>
<p>J(θ) =
&sum;
</p>
<p>k,m
</p>
<p>β2km
1 + β2km
</p>
<p>+
&sum;
</p>
<p>m,ℓ
</p>
<p>α2mℓ
1 + α2mℓ
</p>
<p>, (11.17)
</p>
<p>known as the weight elimination penalty. This has the effect of shrinking
smaller weights more than (11.16) does.
Figure 11.4 shows the result of training a neural network with ten hidden
</p>
<p>units, without weight decay (upper panel) and with weight decay (lower
panel), to the mixture example of Chapter 2. Weight decay has clearly
improved the prediction. Figure 11.5 shows heat maps of the estimated
weights from the training (grayscale versions of these are called Hinton
diagrams.) We see that weight decay has dampened the weights in both
layers: the resulting weights are spread fairly evenly over the ten hidden
units.
</p>
<p>11.5.3 Scaling of the Inputs
</p>
<p>Since the scaling of the inputs determines the effective scaling of the weights
in the bottom layer, it can have a large effect on the quality of the final</p>
<p/>
</div>
<div class="page"><p/>
<p>11.5 Some Issues in Training Neural Networks 399
</p>
<p>Neural Network - 10 Units, No Weight Decay
</p>
<p>. . . . . . .. . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . .. . . . . . . . . . . . .
. . . . . . . .. . .
. . .. . .
. . .. . .
. . . .. . . .
. . . .. . . .
. . . .. . . .
. . . .. . . . .
. . . . .. . . . .
. . . . .. . . . .
. . . . .. . . .
</p>
<p>. . .
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>Training Error: 0.100
</p>
<p>Test Error:       0.259
</p>
<p>Bayes Error:    0.210
</p>
<p>Neural Network - 10 Units, Weight Decay=0.02 
</p>
<p>. .. .
</p>
<p>. .. .
</p>
<p>. . .. . .
</p>
<p>. . .. . .
</p>
<p>. . . .. . . .
</p>
<p>. . . .. . . .
</p>
<p>. . . . .. . . . .
</p>
<p>. . . . .. . . . . .
</p>
<p>. . . . . .. . . . . .
</p>
<p>. . . . . . .. . . . . . .
</p>
<p>. . . . . . .. . . . . . . .
</p>
<p>. . . . . . . .. . . . . . . .
</p>
<p>. . . . . . . . .. . . . . . . . .
</p>
<p>. . . . . . . . .. . . . . . . . . .
</p>
<p>. . . . . . . . . .. . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . .. . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . .. . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . .. . . . . . . . . . . . . .
. . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>Training Error: 0.160
</p>
<p>Test Error:       0.223
</p>
<p>Bayes Error:    0.210
</p>
<p>FIGURE 11.4. A neural network on the mixture example of Chapter 2. The
upper panel uses no weight decay, and overfits the training data. The lower panel
uses weight decay, and achieves close to the Bayes error rate (broken purple
boundary). Both use the softmax activation function and cross-entropy error.</p>
<p/>
</div>
<div class="page"><p/>
<p>400 Neural Networks
</p>
<p>11
</p>
<p>11
</p>
<p>x1x1
</p>
<p>x2x2
</p>
<p>y1y1
</p>
<p>y2y2
</p>
<p>z
1
</p>
<p>z
1
</p>
<p>z
1
</p>
<p>z
1
</p>
<p>z
2
</p>
<p>z
2
</p>
<p>z
2
</p>
<p>z
2
</p>
<p>z
3
</p>
<p>z
3
</p>
<p>z
3
</p>
<p>z
3
</p>
<p>z
1
</p>
<p>z
1
</p>
<p>z
1
</p>
<p>z
1
</p>
<p>z
5
</p>
<p>z
5
</p>
<p>z
5
</p>
<p>z
5
</p>
<p>z
6
</p>
<p>z
6
</p>
<p>z
6
</p>
<p>z
6
</p>
<p>z
7
</p>
<p>z
7
</p>
<p>z
7
</p>
<p>z
7
</p>
<p>z
8
</p>
<p>z
8
</p>
<p>z
8
</p>
<p>z
8
</p>
<p>z
9
</p>
<p>z
9
</p>
<p>z
9
</p>
<p>z
9
</p>
<p>z
1
0
</p>
<p>z
1
0
</p>
<p>z
1
0
</p>
<p>z
1
0
</p>
<p>No weight decay Weight decay
</p>
<p>FIGURE 11.5. Heat maps of the estimated weights from the training of neural
networks from Figure 11.4. The display ranges from bright green (negative) to
bright red (positive).
</p>
<p>solution. At the outset it is best to standardize all inputs to have mean zero
and standard deviation one. This ensures all inputs are treated equally in
the regularization process, and allows one to choose a meaningful range for
the random starting weights. With standardized inputs, it is typical to take
random uniform weights over the range [&minus;0.7,+0.7].
</p>
<p>11.5.4 Number of Hidden Units and Layers
</p>
<p>Generally speaking it is better to have too many hidden units than too few.
With too few hidden units, the model might not have enough flexibility to
capture the nonlinearities in the data; with too many hidden units, the
extra weights can be shrunk toward zero if appropriate regularization is
used. Typically the number of hidden units is somewhere in the range of
5 to 100, with the number increasing with the number of inputs and num-
ber of training cases. It is most common to put down a reasonably large
number of units and train them with regularization. Some researchers use
cross-validation to estimate the optimal number, but this seems unneces-
sary if cross-validation is used to estimate the regularization parameter.
Choice of the number of hidden layers is guided by background knowledge
and experimentation. Each layer extracts features of the input for regres-
sion or classification. Use of multiple hidden layers allows construction of
hierarchical features at different levels of resolution. An example of the
effective use of multiple layers is given in Section 11.6.
</p>
<p>11.5.5 Multiple Minima
</p>
<p>The error function R(θ) is nonconvex, possessing many local minima. As a
result, the final solution obtained is quite dependent on the choice of start-</p>
<p/>
</div>
<div class="page"><p/>
<p>11.6 Example: Simulated Data 401
</p>
<p>ing weights. One must at least try a number of random starting configura-
tions, and choose the solution giving lowest (penalized) error. Probably a
better approach is to use the average predictions over the collection of net-
works as the final prediction (Ripley, 1996). This is preferable to averaging
the weights, since the nonlinearity of the model implies that this averaged
solution could be quite poor. Another approach is via bagging, which aver-
ages the predictions of networks training from randomly perturbed versions
of the training data. This is described in Section 8.7.
</p>
<p>11.6 Example: Simulated Data
</p>
<p>We generated data from two additive error models Y = f(X) + ε:
</p>
<p>Sum of sigmoids: Y = σ(aT1 X) + σ(a
T
2 X) + ε1;
</p>
<p>Radial: Y =
10&prod;
</p>
<p>m=1
</p>
<p>φ(Xm) + ε2.
</p>
<p>Here XT = (X1, X2, . . . , Xp), each Xj being a standard Gaussian variate,
with p = 2 in the first model, and p = 10 in the second.
For the sigmoid model, a1 = (3, 3), a2 = (3,&minus;3); for the radial model,
</p>
<p>φ(t) = (1/2π)1/2 exp(&minus;t2/2). Both ε1 and ε2 are Gaussian errors, with
variance chosen so that the signal-to-noise ratio
</p>
<p>Var(E(Y |X))
Var(Y &minus; E(Y |X)) =
</p>
<p>Var(f(X))
</p>
<p>Var(ε)
(11.18)
</p>
<p>is 4 in both models. We took a training sample of size 100 and a test sample
of size 10, 000. We fit neural networks with weight decay and various num-
bers of hidden units, and recorded the average test error ETest(Y &minus; f̂(X))2
for each of 10 random starting weights. Only one training set was gen-
erated, but the results are typical for an &ldquo;average&rdquo; training set. The test
errors are shown in Figure 11.6. Note that the zero hidden unit model refers
to linear least squares regression. The neural network is perfectly suited to
the sum of sigmoids model, and the two-unit model does perform the best,
achieving an error close to the Bayes rate. (Recall that the Bayes rate for
regression with squared error is the error variance; in the figures, we report
test error relative to the Bayes error). Notice, however, that with more hid-
den units, overfitting quickly creeps in, and with some starting weights the
model does worse than the linear model (zero hidden unit) model. Even
with two hidden units, two of the ten starting weight configurations pro-
duced results no better than the linear model, confirming the importance
of multiple starting values.
A radial function is in a sense the most difficult for the neural net, as it is
</p>
<p>spherically symmetric and with no preferred directions. We see in the right</p>
<p/>
</div>
<div class="page"><p/>
<p>402 Neural Networks
</p>
<p>1
.0
</p>
<p>1
.5
</p>
<p>2
.0
</p>
<p>2
.5
</p>
<p>3
.0
</p>
<p>0 1 2 3 4 5 6 7 8 9 10
</p>
<p>Number of Hidden Units
</p>
<p>T
e
</p>
<p>s
t 
</p>
<p>E
rr
</p>
<p>o
r
</p>
<p>Sum of Sigmoids
</p>
<p>0
5
</p>
<p>1
0
</p>
<p>1
5
</p>
<p>2
0
</p>
<p>2
5
</p>
<p>3
0
</p>
<p>0 1 2 3 4 5 6 7 8 9 10
</p>
<p>Number of Hidden Units
</p>
<p>T
e
</p>
<p>s
t 
</p>
<p>E
rr
</p>
<p>o
r
</p>
<p>Radial
</p>
<p>FIGURE 11.6. Boxplots of test error, for simulated data example, relative to
the Bayes error (broken horizontal line). True function is a sum of two sigmoids
on the left, and a radial function is on the right. The test error is displayed for
10 different starting weights, for a single hidden layer neural network with the
number of units as indicated.
</p>
<p>panel of Figure 11.6 that it does poorly in this case, with the test error
staying well above the Bayes error (note the different vertical scale from
the left panel). In fact, since a constant fit (such as the sample average)
achieves a relative error of 5 (when the SNR is 4), we see that the neural
networks perform increasingly worse than the mean.
In this example we used a fixed weight decay parameter of 0.0005, rep-
</p>
<p>resenting a mild amount of regularization. The results in the left panel of
Figure 11.6 suggest that more regularization is needed with greater num-
bers of hidden units.
In Figure 11.7 we repeated the experiment for the sum of sigmoids model,
</p>
<p>with no weight decay in the left panel, and stronger weight decay (λ = 0.1)
in the right panel. With no weight decay, overfitting becomes even more
severe for larger numbers of hidden units. The weight decay value λ = 0.1
produces good results for all numbers of hidden units, and there does not
appear to be overfitting as the number of units increase. Finally, Figure 11.8
shows the test error for a ten hidden unit network, varying the weight decay
parameter over a wide range. The value 0.1 is approximately optimal.
In summary, there are two free parameters to select: the weight decay λ
</p>
<p>and number of hidden units M . As a learning strategy, one could fix either
parameter at the value corresponding to the least constrained model, to
ensure that the model is rich enough, and use cross-validation to choose
the other parameter. Here the least constrained values are zero weight decay
and ten hidden units. Comparing the left panel of Figure 11.7 to Figure
11.8, we see that the test error is less sensitive to the value of the weight</p>
<p/>
</div>
<div class="page"><p/>
<p>11.6 Example: Simulated Data 403
</p>
<p>1
.0
</p>
<p>1
.5
</p>
<p>2
.0
</p>
<p>2
.5
</p>
<p>3
.0
</p>
<p>0 1 2 3 4 5 6 7 8 9 10
</p>
<p>Number of Hidden Units
</p>
<p>T
e
</p>
<p>s
t 
</p>
<p>E
rr
</p>
<p>o
r
</p>
<p>No Weight Decay
</p>
<p>1
.0
</p>
<p>1
.5
</p>
<p>2
.0
</p>
<p>2
.5
</p>
<p>3
.0
</p>
<p>0 1 2 3 4 5 6 7 8 9 10
</p>
<p>Number of Hidden Units
T
</p>
<p>e
s
t 
</p>
<p>E
rr
</p>
<p>o
r
</p>
<p>Weight Decay=0.1
</p>
<p>FIGURE 11.7. Boxplots of test error, for simulated data example, relative to the
Bayes error. True function is a sum of two sigmoids. The test error is displayed
for ten different starting weights, for a single hidden layer neural network with
the number units as indicated. The two panels represent no weight decay (left)
and strong weight decay λ = 0.1 (right).
</p>
<p>1
.0
</p>
<p>1
.2
</p>
<p>1
.4
</p>
<p>1
.6
</p>
<p>1
.8
</p>
<p>2
.0
</p>
<p>2
.2
</p>
<p>0.00 0.02 0.04 0.06 0.08 0.10 0.12 0.14
</p>
<p>Weight Decay Parameter
</p>
<p>T
e
</p>
<p>s
t 
</p>
<p>E
rr
</p>
<p>o
r
</p>
<p>Sum of Sigmoids, 10 Hidden Unit Model
</p>
<p>FIGURE 11.8. Boxplots of test error, for simulated data example. True function
is a sum of two sigmoids. The test error is displayed for ten different starting
weights, for a single hidden layer neural network with ten hidden units and weight
decay parameter value as indicated.</p>
<p/>
</div>
<div class="page"><p/>
<p>404 Neural Networks
</p>
<p>FIGURE 11.9. Examples of training cases from ZIP code data. Each image is
a 16&times; 16 8-bit grayscale representation of a handwritten digit.
</p>
<p>decay parameter, and hence cross-validation of this parameter would be
preferred.
</p>
<p>11.7 Example: ZIP Code Data
</p>
<p>This example is a character recognition task: classification of handwritten
numerals. This problem captured the attention of the machine learning and
neural network community for many years, and has remained a benchmark
problem in the field. Figure 11.9 shows some examples of normalized hand-
written digits, automatically scanned from envelopes by the U.S. Postal
Service. The original scanned digits are binary and of different sizes and
orientations; the images shown here have been deslanted and size normal-
ized, resulting in 16&times; 16 grayscale images (Le Cun et al., 1990). These 256
pixel values are used as inputs to the neural network classifier.
A black box neural network is not ideally suited to this pattern recogni-
</p>
<p>tion task, partly because the pixel representation of the images lack certain
invariances (such as small rotations of the image). Consequently early at-
tempts with neural networks yielded misclassification rates around 4.5%
on various examples of the problem. In this section we show some of the
pioneering efforts to handcraft the neural network to overcome some these
deficiencies (Le Cun, 1989), which ultimately led to the state of the art in
neural network performance(Le Cun et al., 1998)1.
</p>
<p>Although current digit datasets have tens of thousands of training and
test examples, the sample size here is deliberately modest in order to em-
</p>
<p>1The figures and tables in this example were recreated from Le Cun (1989).</p>
<p/>
</div>
<div class="page"><p/>
<p>11.7 Example: ZIP Code Data 405
</p>
<p>16x16
</p>
<p>8x8x2
</p>
<p>16x16
</p>
<p>10
</p>
<p>4x4
</p>
<p>4x4
</p>
<p>8x8x2
</p>
<p>10
</p>
<p>Shared Weights
Net-5Net-4
</p>
<p>Net-1
</p>
<p>4x4x4
</p>
<p>Local Connectivity
</p>
<p>10
10
</p>
<p>10
</p>
<p>Net-3Net-2
</p>
<p>8x8
12
</p>
<p>16x1616x16
16x16
</p>
<p>FIGURE 11.10. Architecture of the five networks used in the ZIP code example.
</p>
<p>phasize the effects. The examples were obtained by scanning some actual
hand-drawn digits, and then generating additional images by random hor-
izontal shifts. Details may be found in Le Cun (1989). There are 320 digits
in the training set, and 160 in the test set.
Five different networks were fit to the data:
</p>
<p>Net-1: No hidden layer, equivalent to multinomial logistic regression.
</p>
<p>Net-2: One hidden layer, 12 hidden units fully connected.
</p>
<p>Net-3: Two hidden layers locally connected.
</p>
<p>Net-4: Two hidden layers, locally connected with weight sharing.
</p>
<p>Net-5: Two hidden layers, locally connected, two levels of weight sharing.
</p>
<p>These are depicted in Figure 11.10. Net-1 for example has 256 inputs, one
each for the 16&times;16 input pixels, and ten output units for each of the digits
0&ndash;9. The predicted value f̂k(x) represents the estimated probability that
an image x has digit class k, for k = 0, 1, 2, . . . , 9.</p>
<p/>
</div>
<div class="page"><p/>
<p>406 Neural Networks
</p>
<p>Training Epochs
</p>
<p>%
 C
</p>
<p>o
rr
</p>
<p>e
c
t 
</p>
<p>o
n
</p>
<p> T
e
</p>
<p>s
t 
</p>
<p>D
a
</p>
<p>ta
</p>
<p>0 5 10 15 20 25 30
</p>
<p>60
</p>
<p>70
</p>
<p>80
</p>
<p>90
</p>
<p>100
</p>
<p>Net-1
</p>
<p>Net-2
</p>
<p>Net-3
</p>
<p>Net-4
Net-5
</p>
<p>FIGURE 11.11. Test performance curves, as a function of the number of train-
ing epochs, for the five networks of Table 11.1 applied to the ZIP code data.
(Le Cun, 1989)
</p>
<p>The networks all have sigmoidal output units, and were all fit with the
sum-of-squares error function. The first network has no hidden layer, and
hence is nearly equivalent to a linear multinomial regression model (Exer-
cise 11.4). Net-2 is a single hidden layer network with 12 hidden units, of
the kind described above.
The training set error for all of the networks was 0%, since in all cases
</p>
<p>there are more parameters than training observations. The evolution of the
test error during the training epochs is shown in Figure 11.11. The linear
network (Net-1) starts to overfit fairly quickly, while test performance of
the others level off at successively superior values.
The other three networks have additional features which demonstrate
</p>
<p>the power and flexibility of the neural network paradigm. They introduce
constraints on the network, natural for the problem at hand, which allow
for more complex connectivity but fewer parameters.
Net-3 uses local connectivity: this means that each hidden unit is con-
</p>
<p>nected to only a small patch of units in the layer below. In the first hidden
layer (an 8&times;8 array), each unit takes inputs from a 3&times;3 patch of the input
layer; for units in the first hidden layer that are one unit apart, their recep-
tive fields overlap by one row or column, and hence are two pixels apart.
In the second hidden layer, inputs are from a 5&times; 5 patch, and again units
that are one unit apart have receptive fields that are two units apart. The
weights for all other connections are set to zero. Local connectivity makes
each unit responsible for extracting local features from the layer below, and</p>
<p/>
</div>
<div class="page"><p/>
<p>11.7 Example: ZIP Code Data 407
</p>
<p>TABLE 11.1. Test set performance of five different neural networks on a hand-
written digit classification example (Le Cun, 1989).
</p>
<p>Network Architecture Links Weights % Correct
</p>
<p>Net-1: Single layer network 2570 2570 80.0%
Net-2: Two layer network 3214 3214 87.0%
</p>
<p>Net-3: Locally connected 1226 1226 88.5%
</p>
<p>Net-4: Constrained network 1 2266 1132 94.0%
</p>
<p>Net-5: Constrained network 2 5194 1060 98.4%
</p>
<p>reduces considerably the total number of weights. With many more hidden
units than Net-2, Net-3 has fewer links and hence weights (1226 vs. 3214),
and achieves similar performance.
Net-4 and Net-5 have local connectivity with shared weights. All units
</p>
<p>in a local feature map perform the same operation on different parts of the
image, achieved by sharing the same weights. The first hidden layer of Net-
4 has two 8&times;8 arrays, and each unit takes input from a 3&times;3 patch just like
in Net-3. However, each of the units in a single 8&times; 8 feature map share the
same set of nine weights (but have their own bias parameter). This forces
the extracted features in different parts of the image to be computed by
the same linear functional, and consequently these networks are sometimes
known as convolutional networks. The second hidden layer of Net-4 has
no weight sharing, and is the same as in Net-3. The gradient of the error
function R with respect to a shared weight is the sum of the gradients of
R with respect to each connection controlled by the weights in question.
Table 11.1 gives the number of links, the number of weights and the
</p>
<p>optimal test performance for each of the networks. We see that Net-4 has
more links but fewer weights than Net-3, and superior test performance.
Net-5 has four 4 &times; 4 feature maps in the second hidden layer, each unit
connected to a 5 &times; 5 local patch in the layer below. Weights are shared
in each of these feature maps. We see that Net-5 does the best, having
errors of only 1.6%, compared to 13% for the &ldquo;vanilla&rdquo; network Net-2.
The clever design of network Net-5, motivated by the fact that features of
handwriting style should appear in more than one part of a digit, was the
result of many person years of experimentation. This and similar networks
gave better performance on ZIP code problems than any other learning
method at that time (early 1990s). This example also shows that neural
networks are not a fully automatic tool, as they are sometimes advertised.
As with all statistical models, subject matter knowledge can and should be
used to improve their performance.
This network was later outperformed by the tangent distance approach
</p>
<p>(Simard et al., 1993) described in Section 13.3.3, which explicitly incorpo-
rates natural affine invariances. At this point the digit recognition datasets
become test beds for every new learning procedure, and researchers worked</p>
<p/>
</div>
<div class="page"><p/>
<p>408 Neural Networks
</p>
<p>hard to drive down the error rates. As of this writing, the best error rates on
a large database (60, 000 training, 10, 000 test observations), derived from
standard NIST2 databases, were reported to be the following: (Le Cun et
al., 1998):
</p>
<p>&bull; 1.1% for tangent distance with a 1-nearest neighbor classifier (Sec-
tion 13.3.3);
</p>
<p>&bull; 0.8% for a degree-9 polynomial SVM (Section 12.3);
</p>
<p>&bull; 0.8% for LeNet-5, a more complex version of the convolutional net-
work described here;
</p>
<p>&bull; 0.7% for boosted LeNet-4. Boosting is described in Chapter 8. LeNet-
4 is a predecessor of LeNet-5.
</p>
<p>Le Cun et al. (1998) report a much larger table of performance results, and
it is evident that many groups have been working very hard to bring these
test error rates down. They report a standard error of 0.1% on the error
estimates, which is based on a binomial average with N = 10, 000 and
p &asymp; 0.01. This implies that error rates within 0.1&mdash;0.2% of one another
are statistically equivalent. Realistically the standard error is even higher,
since the test data has been implicitly used in the tuning of the various
procedures.
</p>
<p>11.8 Discussion
</p>
<p>Both projection pursuit regression and neural networks take nonlinear func-
tions of linear combinations (&ldquo;derived features&rdquo;) of the inputs. This is a
powerful and very general approach for regression and classification, and
has been shown to compete well with the best learning methods on many
problems.
These tools are especially effective in problems with a high signal-to-noise
</p>
<p>ratio and settings where prediction without interpretation is the goal. They
are less effective for problems where the goal is to describe the physical pro-
cess that generated the data and the roles of individual inputs. Each input
enters into the model in many places, in a nonlinear fashion. Some authors
(Hinton, 1989) plot a diagram of the estimated weights into each hidden
unit, to try to understand the feature that each unit is extracting. This
is limited however by the lack of identifiability of the parameter vectors
αm, m = 1, . . . ,M . Often there are solutions with αm spanning the same
linear space as the ones found during training, giving predicted values that
</p>
<p>2The National Institute of Standards and Technology maintain large databases, in-
</p>
<p>cluding handwritten character databases; http://www.nist.gov/srd/.</p>
<p/>
</div>
<div class="page"><p/>
<p>11.9 Bayesian Neural Nets and the NIPS 2003 Challenge 409
</p>
<p>are roughly the same. Some authors suggest carrying out a principal com-
ponent analysis of these weights, to try to find an interpretable solution. In
general, the difficulty of interpreting these models has limited their use in
fields like medicine, where interpretation of the model is very important.
There has been a great deal of research on the training of neural net-
</p>
<p>works. Unlike methods like CART and MARS, neural networks are smooth
functions of real-valued parameters. This facilitates the development of
Bayesian inference for these models. The next sections discusses a success-
ful Bayesian implementation of neural networks.
</p>
<p>11.9 Bayesian Neural Nets and the NIPS 2003
Challenge
</p>
<p>A classification competition was held in 2003, in which five labeled train-
ing datasets were provided to participants. It was organized for a Neural
Information Processing Systems (NIPS) workshop. Each of the data sets
constituted a two-class classification problems, with different sizes and from
a variety of domains (see Table 11.2). Feature measurements for a valida-
tion dataset were also available.
Participants developed and applied statistical learning procedures to
</p>
<p>make predictions on the datasets, and could submit predictions to a web-
site on the validation set for a period of 12 weeks. With this feedback,
participants were then asked to submit predictions for a separate test set
and they received their results. Finally, the class labels for the validation
set were released and participants had one week to train their algorithms
on the combined training and validation sets, and submit their final pre-
dictions to the competition website. A total of 75 groups participated, with
20 and 16 eventually making submissions on the validation and test sets,
respectively.
There was an emphasis on feature extraction in the competition. Arti-
</p>
<p>ficial &ldquo;probes&rdquo; were added to the data: these are noise features with dis-
tributions resembling the real features but independent of the class labels.
The percentage of probes that were added to each dataset, relative to the
total set of features, is shown on Table 11.2. Thus each learning algorithm
had to figure out a way of identifying the probes and downweighting or
eliminating them.
A number of metrics were used to evaluate the entries, including the
</p>
<p>percentage correct on the test set, the area under the ROC curve, and a
combined score that compared each pair of classifiers head-to-head. The
results of the competition are very interesting and are detailed in Guyon et
al. (2006). The most notable result: the entries of Neal and Zhang (2006)
were the clear overall winners. In the final competition they finished first</p>
<p/>
</div>
<div class="page"><p/>
<p>410 Neural Networks
</p>
<p>TABLE 11.2. NIPS 2003 challenge data sets. The column labeled p is the number
of features. For the Dorothea dataset the features are binary. Ntr, Nval and Nte
are the number of training, validation and test cases, respectively
</p>
<p>Dataset Domain Feature p Percent Ntr Nval Nte
Type Probes
</p>
<p>Arcene Mass spectrometry Dense 10,000 30 100 100 700
Dexter Text classification Sparse 20,000 50 300 300 2000
Dorothea Drug discovery Sparse 100,000 50 800 350 800
Gisette Digit recognition Dense 5000 30 6000 1000 6500
Madelon Artificial Dense 500 96 2000 600 1800
</p>
<p>in three of the five datasets, and were 5th and 7th on the remaining two
datasets.
In their winning entries, Neal and Zhang (2006) used a series of pre-
</p>
<p>processing feature-selection steps, followed by Bayesian neural networks,
Dirichlet diffusion trees, and combinations of these methods. Here we focus
only on the Bayesian neural network approach, and try to discern which
aspects of their approach were important for its success. We rerun their
programs and compare the results to boosted neural networks and boosted
trees, and other related methods.
</p>
<p>11.9.1 Bayes, Boosting and Bagging
</p>
<p>Let us first review briefly the Bayesian approach to inference and its appli-
cation to neural networks. Given training data Xtr,ytr, we assume a sam-
pling model with parameters θ; Neal and Zhang (2006) use a two-hidden-
layer neural network, with output nodes the class probabilities Pr(Y |X, θ)
for the binary outcomes. Given a prior distribution Pr(θ), the posterior
distribution for the parameters is
</p>
<p>Pr(θ|Xtr,ytr) =
Pr(θ)Pr(ytr|Xtr, θ)&int;
Pr(θ)Pr(ytr|Xtr, θ)dθ
</p>
<p>(11.19)
</p>
<p>For a test case with features Xnew, the predictive distribution for the
label Ynew is
</p>
<p>Pr(Ynew|Xnew,Xtr,ytr) =
&int;
</p>
<p>Pr(Ynew|Xnew, θ)Pr(θ|Xtr,ytr)dθ (11.20)
</p>
<p>(c.f. equation 8.24). Since the integral in (11.20) is intractable, sophisticated
Markov Chain Monte Carlo (MCMC) methods are used to sample from the
posterior distribution Pr(Ynew|Xnew,Xtr,ytr). A few hundred values θ are
generated and then a simple average of these values estimates the integral.
Neal and Zhang (2006) use diffuse Gaussian priors for all of the parame-
ters. The particular MCMC approach that was used is called hybrid Monte
Carlo, and may be important for the success of the method. It includes
an auxiliary momentum vector and implements Hamiltonian dynamics in
which the potential function is the target density. This is done to avoid</p>
<p/>
</div>
<div class="page"><p/>
<p>11.9 Bayesian Neural Nets and the NIPS 2003 Challenge 411
</p>
<p>random walk behavior; the successive candidates move across the sample
space in larger steps. They tend to be less correlated and hence converge
to the target distribution more rapidly.
Neal and Zhang (2006) also tried different forms of pre-processing of the
</p>
<p>features:
</p>
<p>1. univariate screening using t-tests, and
</p>
<p>2. automatic relevance determination.
</p>
<p>In the latter method (ARD), the weights (coefficients) for the jth feature
to each of the first hidden layer units all share a common prior variance
σ2j , and prior mean zero. The posterior distributions for each variance σ
</p>
<p>2
j
</p>
<p>are computed, and the features whose posterior variance concentrates on
small values are discarded.
There are thus three main features of this approach that could be im-
</p>
<p>portant for its success:
</p>
<p>(a) the feature selection and pre-processing,
</p>
<p>(b) the neural network model, and
</p>
<p>(c) the Bayesian inference for the model using MCMC.
</p>
<p>According to Neal and Zhang (2006), feature screening in (a) is carried
out purely for computational efficiency; the MCMC procedure is slow with
a large number of features. There is no need to use feature selection to avoid
overfitting. The posterior average (11.20) takes care of this automatically.
We would like to understand the reasons for the success of the Bayesian
</p>
<p>method. In our view, power of modern Bayesian methods does not lie in
their use as a formal inference procedure; most people would not believe
that the priors in a high-dimensional, complex neural network model are
actually correct. Rather the Bayesian/MCMC approach gives an efficient
way of sampling the relevant parts of model space, and then averaging the
predictions for the high-probability models.
Bagging and boosting are non-Bayesian procedures that have some simi-
</p>
<p>larity to MCMC in a Bayesian model. The Bayesian approach fixes the data
and perturbs the parameters, according to current estimate of the poste-
rior distribution. Bagging perturbs the data in an i.i.d fashion and then
re-estimates the model to give a new set of model parameters. At the end,
a simple average of the model predictions from different bagged samples is
computed. Boosting is similar to bagging, but fits a model that is additive
in the models of each individual base learner, which are learned using non
i.i.d. samples. We can write all of these models in the form
</p>
<p>f̂(xnew) =
</p>
<p>L&sum;
</p>
<p>ℓ=1
</p>
<p>wℓE(Ynew|xnew, θ̂ℓ) (11.21)</p>
<p/>
</div>
<div class="page"><p/>
<p>412 Neural Networks
</p>
<p>In all cases the θ̂ℓ are a large collection of model parameters. For the
Bayesian model the wℓ = 1/L, and the average estimates the posterior
mean (11.21) by sampling θℓ from the posterior distribution. For bagging,
</p>
<p>wℓ = 1/L as well, and the θ̂ℓ are the parameters refit to bootstrap re-
samples of the training data. For boosting, the weights are all equal to
1, but the θ̂ℓ are typically chosen in a nonrandom sequential fashion to
constantly improve the fit.
</p>
<p>11.9.2 Performance Comparisons
</p>
<p>Based on the similarities above, we decided to compare Bayesian neural
networks to boosted trees, boosted neural networks, random forests and
bagged neural networks on the five datasets in Table 11.2. Bagging and
boosting of neural networks are not methods that we have previously used
in our work. We decided to try them here, because of the success of Bayesian
neural networks in this competition, and the good performance of bagging
and boosting with trees. We also felt that by bagging and boosting neural
nets, we could assess both the choice of model as well as the model search
strategy.
Here are the details of the learning methods that were compared:
</p>
<p>Bayesian neural nets. The results here are taken from Neal and Zhang
(2006), using their Bayesian approach to fitting neural networks. The
models had two hidden layers of 20 and 8 units. We re-ran some
networks for timing purposes only.
</p>
<p>Boosted trees. We used the gbm package (version 1.5-7) in the R language.
Tree depth and shrinkage factors varied from dataset to dataset. We
consistently bagged 80% of the data at each boosting iteration (the
default is 50%). Shrinkage was between 0.001 and 0.1. Tree depth was
between 2 and 9.
</p>
<p>Boosted neural networks. Since boosting is typically most effective with
&ldquo;weak&rdquo; learners, we boosted a single hidden layer neural network with
two or four units, fit with the nnet package (version 7.2-36) in R.
</p>
<p>Random forests. We used the R package randomForest (version 4.5-16)
with default settings for the parameters.
</p>
<p>Bagged neural networks. We used the same architecture as in the Bayesian
neural network above (two hidden layers of 20 and 8 units), fit using
both Neal&rsquo;s C language package &ldquo;Flexible Bayesian Modeling&rdquo; (2004-
11-10 release), and Matlab neural-net toolbox (version 5.1).</p>
<p/>
</div>
<div class="page"><p/>
<p>11.9 Bayesian Neural Nets and the NIPS 2003 Challenge 413
</p>
<p>T
e
s
t 
E
</p>
<p>rr
o
r 
</p>
<p>(%
)
</p>
<p>Arcene Dexter Dorothea Gisette Madelon
</p>
<p>5
1
5
</p>
<p>2
5
</p>
<p>Univariate Screened Features
</p>
<p>Bayesian neural nets
boosted trees 
boosted neural nets
random forests
bagged neural networks 
</p>
<p>T
e
s
t 
E
</p>
<p>rr
o
r 
</p>
<p>(%
)
</p>
<p>Arcene Dexter Dorothea Gisette Madelon
</p>
<p>5
1
5
</p>
<p>2
5
</p>
<p>ARD Reduced Features
</p>
<p>FIGURE 11.12. Performance of different learning methods on five problems,
using both univariate screening of features (top panel) and a reduced feature set
from automatic relevance determination. The error bars at the top of each plot
have width equal to one standard error of the difference between two error rates.
On most of the problems several competitors are within this error bound.
</p>
<p>This analysis was carried out by Nicholas Johnson, and full details may
be found in Johnson (2008)3. The results are shown in Figure 11.12 and
Table 11.3.
The figure and table show Bayesian, boosted and bagged neural networks,
</p>
<p>boosted trees, and random forests, using both the screened and reduced
features sets. The error bars at the top of each plot indicate one standard
error of the difference between two error rates. Bayesian neural networks
again emerge as the winner, although for some datasets the differences
between the test error rates is not statistically significant. Random forests
performs the best among the competitors using the selected feature set,
while the boosted neural networks perform best with the reduced feature
set, and nearly match the Bayesian neural net.
The superiority of boosted neural networks over boosted trees suggest
</p>
<p>that the neural network model is better suited to these particular prob-
lems. Specifically, individual features might not be good predictors here
</p>
<p>3We also thank Isabelle Guyon for help in preparing the results of this section.</p>
<p/>
</div>
<div class="page"><p/>
<p>414 Neural Networks
</p>
<p>TABLE 11.3. Performance of different methods. Values are average rank of test
error across the five problems (low is good), and mean computation time and
standard error of the mean, in minutes.
</p>
<p>Screened Features ARD Reduced Features
Method Average Average Average Average
</p>
<p>Rank Time Rank Time
</p>
<p>Bayesian neural networks 1.5 384(138) 1.6 600(186)
Boosted trees 3.4 3.03(2.5) 4.0 34.1(32.4)
Boosted neural networks 3.8 9.4(8.6) 2.2 35.6(33.5)
Random forests 2.7 1.9(1.7) 3.2 11.2(9.3)
Bagged neural networks 3.6 3.5(1.1) 4.0 6.4(4.4)
</p>
<p>and linear combinations of features work better. However the impressive
performance of random forests is at odds with this explanation, and came
as a surprise to us.
Since the reduced feature sets come from the Bayesian neural network
</p>
<p>approach, only the methods that use the screened features are legitimate,
self-contained procedures. However, this does suggest that better methods
for internal feature selection might help the overall performance of boosted
neural networks.
The table also shows the approximate training time required for each
</p>
<p>method. Here the non-Bayesian methods show a clear advantage.
Overall, the superior performance of Bayesian neural networks here may
</p>
<p>be due to the fact that
</p>
<p>(a) the neural network model is well suited to these five problems, and
</p>
<p>(b) the MCMC approach provides an efficient way of exploring the im-
portant part of the parameter space, and then averaging the resulting
models according to their quality.
</p>
<p>The Bayesian approach works well for smoothly parametrized models like
neural nets; it is not yet clear that it works as well for non-smooth models
like trees.
</p>
<p>11.10 Computational Considerations
</p>
<p>WithN observations, p predictors,M hidden units and L training epochs, a
neural network fit typically requires O(NpML) operations. There are many
packages available for fitting neural networks, probably many more than
exist for mainstream statistical methods. Because the available software
varies widely in quality, and the learning problem for neural networks is
sensitive to issues such as input scaling, such software should be carefully
chosen and tested.</p>
<p/>
</div>
<div class="page"><p/>
<p>Exercises 415
</p>
<p>Bibliographic Notes
</p>
<p>Projection pursuit was proposed by Friedman and Tukey (1974), and spe-
cialized to regression by Friedman and Stuetzle (1981). Huber (1985) gives
a scholarly overview, and Roosen and Hastie (1994) present a formulation
using smoothing splines. The motivation for neural networks dates back
to McCulloch and Pitts (1943), Widrow and Hoff (1960) (reprinted in An-
derson and Rosenfeld (1988)) and Rosenblatt (1962). Hebb (1949) heavily
influenced the development of learning algorithms. The resurgence of neural
networks in the mid 1980s was due to Werbos (1974), Parker (1985) and
Rumelhart et al. (1986), who proposed the back-propagation algorithm.
Today there are many books written on the topic, for a broad range of
audiences. For readers of this book, Hertz et al. (1991), Bishop (1995) and
Ripley (1996) may be the most informative. Bayesian learning for neural
networks is described in Neal (1996). The ZIP code example was taken from
Le Cun (1989); see also Le Cun et al. (1990) and Le Cun et al. (1998).
We do not discuss theoretical topics such as approximation properties of
</p>
<p>neural networks, such as the work of Barron (1993), Girosi et al. (1995)
and Jones (1992). Some of these results are summarized by Ripley (1996).
</p>
<p>Exercises
</p>
<p>Ex. 11.1 Establish the exact correspondence between the projection pur-
suit regression model (11.1) and the neural network (11.5). In particular,
show that the single-layer regression network is equivalent to a PPR model
with gm(ω
</p>
<p>T
mx) = βmσ(α0m + sm(ω
</p>
<p>T
mx)), where ωm is the mth unit vector.
</p>
<p>Establish a similar equivalence for a classification network.
</p>
<p>Ex. 11.2 Consider a neural network for a quantitative outcome as in (11.5),
using squared-error loss and identity output function gk(t) = t. Suppose
that the weights αm from the input to hidden layer are nearly zero. Show
that the resulting model is nearly linear in the inputs.
</p>
<p>Ex. 11.3 Derive the forward and backward propagation equations for the
cross-entropy loss function.
</p>
<p>Ex. 11.4 Consider a neural network for a K class outcome that uses cross-
entropy loss. If the network has no hidden layer, show that the model is
equivalent to the multinomial logistic model described in Chapter 4.
</p>
<p>Ex. 11.5
</p>
<p>(a) Write a program to fit a single hidden layer neural network (ten hidden
units) via back-propagation and weight decay.</p>
<p/>
</div>
<div class="page"><p/>
<p>416 Neural Networks
</p>
<p>(b) Apply it to 100 observations from the model
</p>
<p>Y = σ(aT1 X) + (a
T
2 X)
</p>
<p>2 + 0.30 &middot; Z,
</p>
<p>where σ is the sigmoid function, Z is standard normal,XT = (X1, X2),
each Xj being independent standard normal, and a1 = (3, 3), a2 =
(3,&minus;3). Generate a test sample of size 1000, and plot the training and
test error curves as a function of the number of training epochs, for
different values of the weight decay parameter. Discuss the overfitting
behavior in each case.
</p>
<p>(c) Vary the number of hidden units in the network, from 1 up to 10, and
determine the minimum number needed to perform well for this task.
</p>
<p>Ex. 11.6 Write a program to carry out projection pursuit regression, using
cubic smoothing splines with fixed degrees of freedom. Fit it to the data
from the previous exercise, for various values of the smoothing parameter
and number of model terms. Find the minimum number of model terms
necessary for the model to perform well and compare this to the number
of hidden units from the previous exercise.
</p>
<p>Ex. 11.7 Fit a neural network to the spam data of Section 9.1.2, and compare
the results to those for the additive model given in that chapter. Compare
both the classification performance and interpretability of the final model.</p>
<p/>
</div>
<div class="page"><p/>
<p>12
Support Vector Machines and
Flexible Discriminants
</p>
<p>12.1 Introduction
</p>
<p>In this chapter we describe generalizations of linear decision boundaries
for classification. Optimal separating hyperplanes are introduced in Chap-
ter 4 for the case when two classes are linearly separable. Here we cover
extensions to the nonseparable case, where the classes overlap. These tech-
niques are then generalized to what is known as the support vector machine,
which produces nonlinear boundaries by constructing a linear boundary in
a large, transformed version of the feature space. The second set of methods
generalize Fisher&rsquo;s linear discriminant analysis (LDA). The generalizations
include flexible discriminant analysis which facilitates construction of non-
linear boundaries in a manner very similar to the support vector machines,
penalized discriminant analysis for problems such as signal and image clas-
sification where the large number of features are highly correlated, and
mixture discriminant analysis for irregularly shaped classes.
</p>
<p>12.2 The Support Vector Classifier
</p>
<p>In Chapter 4 we discussed a technique for constructing an optimal separat-
ing hyperplane between two perfectly separated classes. We review this and
generalize to the nonseparable case, where the classes may not be separable
by a linear boundary.
</p>
<p>&copy; Springer Science+Business Media, LLC 2009 
</p>
<p>T. Hastie et al., The Elements of Statistical Learning, Second Edition, 417
</p>
<p>DOI: 10.1007/b94608_12,</p>
<p/>
</div>
<div class="page"><p/>
<p>418 12. Flexible Discriminants
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>margin
</p>
<p>M = 1
‖β‖
</p>
<p>M = 1
‖β‖
</p>
<p>xTβ + β0 = 0
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>margin
</p>
<p>ξ&lowast;1ξ
&lowast;
1ξ
&lowast;
1
</p>
<p>ξ&lowast;2ξ
&lowast;
2ξ
&lowast;
2
</p>
<p>ξ&lowast;3ξ
&lowast;
3
</p>
<p>ξ&lowast;4ξ
&lowast;
4ξ
&lowast;
4 ξ&lowast;5
</p>
<p>M = 1
‖β‖
</p>
<p>M = 1
‖β‖
</p>
<p>xTβ + β0 = 0
</p>
<p>FIGURE 12.1. Support vector classifiers. The left panel shows the separable
case. The decision boundary is the solid line, while broken lines bound the shaded
maximal margin of width 2M = 2/‖β‖. The right panel shows the nonseparable
(overlap) case. The points labeled ξ&lowast;j are on the wrong side of their margin by
an amount ξ&lowast;j = Mξj; points on the correct side have ξ
</p>
<p>&lowast;
j = 0. The margin is
</p>
<p>maximized subject to a total budget
&sum;
</p>
<p>ξi &le; constant. Hence
&sum;
</p>
<p>ξ&lowast;j is the total
distance of points on the wrong side of their margin.
</p>
<p>Our training data consists of N pairs (x1, y1), (x2, y2), . . . , (xN , yN ), with
xi &isin; IRp and yi &isin; {&minus;1, 1}. Define a hyperplane by
</p>
<p>{x : f(x) = xTβ + β0 = 0}, (12.1)
</p>
<p>where β is a unit vector: ‖β‖ = 1. A classification rule induced by f(x) is
</p>
<p>G(x) = sign[xTβ + β0]. (12.2)
</p>
<p>The geometry of hyperplanes is reviewed in Section 4.5, where we show that
f(x) in (12.1) gives the signed distance from a point x to the hyperplane
f(x) = xTβ+β0 = 0. Since the classes are separable, we can find a function
f(x) = xTβ + β0 with yif(xi) &gt; 0 &forall;i. Hence we are able to find the
hyperplane that creates the biggest margin between the training points for
class 1 and &minus;1 (see Figure 12.1). The optimization problem
</p>
<p>max
β,β0,‖β‖=1
</p>
<p>M
</p>
<p>subject to yi(x
T
i β + β0) &ge; M, i = 1, . . . , N,
</p>
<p>(12.3)
</p>
<p>captures this concept. The band in the figure is M units away from the
hyperplane on either side, and hence 2M units wide. It is called the margin.
</p>
<p>We showed that this problem can be more conveniently rephrased as
</p>
<p>min
β,β0
</p>
<p>‖β‖
</p>
<p>subject to yi(x
T
i β + β0) &ge; 1, i = 1, . . . , N,
</p>
<p>(12.4)</p>
<p/>
</div>
<div class="page"><p/>
<p>12.2 The Support Vector Classifier 419
</p>
<p>where we have dropped the norm constraint on β. Note that M = 1/‖β‖.
Expression (12.4) is the usual way of writing the support vector criterion
for separated data. This is a convex optimization problem (quadratic cri-
terion, linear inequality constraints), and the solution is characterized in
Section 4.5.2.
Suppose now that the classes overlap in feature space. One way to deal
</p>
<p>with the overlap is to still maximize M , but allow for some points to be on
the wrong side of the margin. Define the slack variables ξ = (ξ1, ξ2, . . . , ξN ).
There are two natural ways to modify the constraint in (12.3):
</p>
<p>yi(x
T
i β + β0) &ge; M &minus; ξi, (12.5)
</p>
<p>or
</p>
<p>yi(x
T
i β + β0) &ge; M(1&minus; ξi), (12.6)
</p>
<p>&forall;i, ξi &ge; 0,
&sum;N
</p>
<p>i=1 ξi &le; constant. The two choices lead to different solutions.
The first choice seems more natural, since it measures overlap in actual
distance from the margin; the second choice measures the overlap in relative
distance, which changes with the width of the margin M . However, the first
choice results in a nonconvex optimization problem, while the second is
convex; thus (12.6) leads to the &ldquo;standard&rdquo; support vector classifier, which
we use from here on.
Here is the idea of the formulation. The value ξi in the constraint yi(x
</p>
<p>T
i β+
</p>
<p>β0) &ge; M(1 &minus; ξi) is the proportional amount by which the prediction
f(xi) = x
</p>
<p>T
i β+β0 is on the wrong side of its margin. Hence by bounding the
</p>
<p>sum
&sum;
</p>
<p>ξi, we bound the total proportional amount by which predictions
fall on the wrong side of their margin. Misclassifications occur when ξi &gt; 1,
so bounding
</p>
<p>&sum;
ξi at a value K say, bounds the total number of training
</p>
<p>misclassifications at K.
As in (4.48) in Section 4.5.2, we can drop the norm constraint on β,
</p>
<p>define M = 1/‖β‖, and write (12.4) in the equivalent form
</p>
<p>min ‖β‖ subject to
{
</p>
<p>yi(x
T
i β + β0) &ge; 1&minus; ξi &forall;i,
</p>
<p>ξi &ge; 0,
&sum;
</p>
<p>ξi &le; constant.
(12.7)
</p>
<p>This is the usual way the support vector classifier is defined for the non-
separable case. However we find confusing the presence of the fixed scale
&ldquo;1&rdquo; in the constraint yi(x
</p>
<p>T
i β+β0) &ge; 1&minus; ξi, and prefer to start with (12.6).
</p>
<p>The right panel of Figure 12.1 illustrates this overlapping case.
By the nature of the criterion (12.7), we see that points well inside their
</p>
<p>class boundary do not play a big role in shaping the boundary. This seems
like an attractive property, and one that differentiates it from linear dis-
criminant analysis (Section 4.3). In LDA, the decision boundary is deter-
mined by the covariance of the class distributions and the positions of the
class centroids. We will see in Section 12.3.3 that logistic regression is more
similar to the support vector classifier in this regard.</p>
<p/>
</div>
<div class="page"><p/>
<p>420 12. Flexible Discriminants
</p>
<p>12.2.1 Computing the Support Vector Classifier
</p>
<p>The problem (12.7) is quadratic with linear inequality constraints, hence it
is a convex optimization problem. We describe a quadratic programming
solution using Lagrange multipliers. Computationally it is convenient to
re-express (12.7) in the equivalent form
</p>
<p>min
β,β0
</p>
<p>1
</p>
<p>2
‖β‖2 + C
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>ξi
</p>
<p>subject to ξi &ge; 0, yi(xTi β + β0) &ge; 1&minus; ξi &forall;i,
(12.8)
</p>
<p>where the &ldquo;cost&rdquo; parameter C replaces the constant in (12.7); the separable
case corresponds to C = &infin;.
</p>
<p>The Lagrange (primal) function is
</p>
<p>LP =
1
</p>
<p>2
‖β‖2 +C
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>ξi &minus;
N&sum;
</p>
<p>i=1
</p>
<p>αi[yi(x
T
i β+ β0)&minus; (1&minus; ξi)]&minus;
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>μiξi, (12.9)
</p>
<p>which we minimize w.r.t β, β0 and ξi. Setting the respective derivatives to
zero, we get
</p>
<p>β =
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>αiyixi, (12.10)
</p>
<p>0 =
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>αiyi, (12.11)
</p>
<p>αi = C &minus; μi, &forall;i, (12.12)
</p>
<p>as well as the positivity constraints αi, μi, ξi &ge; 0 &forall;i. By substituting
(12.10)&ndash;(12.12) into (12.9), we obtain the Lagrangian (Wolfe) dual objec-
tive function
</p>
<p>LD =
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>αi &minus;
1
</p>
<p>2
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>N&sum;
</p>
<p>i&prime;=1
</p>
<p>αiαi&prime;yiyi&prime;x
T
i xi&prime; , (12.13)
</p>
<p>which gives a lower bound on the objective function (12.8) for any feasible
</p>
<p>point. We maximize LD subject to 0 &le; αi &le; C and
&sum;N
</p>
<p>i=1 αiyi = 0. In
addition to (12.10)&ndash;(12.12), the Karush&ndash;Kuhn&ndash;Tucker conditions include
the constraints
</p>
<p>αi[yi(x
T
i β + β0)&minus; (1&minus; ξi)] = 0, (12.14)
</p>
<p>μiξi = 0, (12.15)
</p>
<p>yi(x
T
i β + β0)&minus; (1&minus; ξi) &ge; 0, (12.16)
</p>
<p>for i = 1, . . . , N . Together these equations (12.10)&ndash;(12.16) uniquely char-
acterize the solution to the primal and dual problem.</p>
<p/>
</div>
<div class="page"><p/>
<p>12.2 The Support Vector Classifier 421
</p>
<p>From (12.10) we see that the solution for β has the form
</p>
<p>β̂ =
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>α̂iyixi, (12.17)
</p>
<p>with nonzero coefficients α̂i only for those observations i for which the
constraints in (12.16) are exactly met (due to (12.14)). These observations
</p>
<p>are called the support vectors, since β̂ is represented in terms of them
alone. Among these support points, some will lie on the edge of the margin
(ξ̂i = 0), and hence from (12.15) and (12.12) will be characterized by
</p>
<p>0 &lt; α̂i &lt; C; the remainder (ξ̂i &gt; 0) have α̂i = C. From (12.14) we can
</p>
<p>see that any of these margin points (0 &lt; α̂i, ξ̂i = 0) can be used to solve
for β0, and we typically use an average of all the solutions for numerical
stability.
Maximizing the dual (12.13) is a simpler convex quadratic programming
</p>
<p>problem than the primal (12.9), and can be solved with standard techniques
(Murray et al., 1981, for example).
</p>
<p>Given the solutions β̂0 and β̂, the decision function can be written as
</p>
<p>Ĝ(x) = sign[f̂(x)]
</p>
<p>= sign[xT β̂ + β̂0]. (12.18)
</p>
<p>The tuning parameter of this procedure is the cost parameter C.
</p>
<p>12.2.2 Mixture Example (Continued)
</p>
<p>Figure 12.2 shows the support vector boundary for the mixture example
of Figure 2.5 on page 21, with two overlapping classes, for two different
values of the cost parameter C. The classifiers are rather similar in their
performance. Points on the wrong side of the boundary are support vectors.
In addition, points on the correct side of the boundary but close to it (in
the margin), are also support vectors. The margin is larger for C = 0.01
than it is for C = 10, 000. Hence larger values of C focus attention more
on (correctly classified) points near the decision boundary, while smaller
values involve data further away. Either way, misclassified points are given
weight, no matter how far away. In this example the procedure is not very
sensitive to choices of C, because of the rigidity of a linear boundary.
The optimal value for C can be estimated by cross-validation, as dis-
</p>
<p>cussed in Chapter 7. Interestingly, the leave-one-out cross-validation error
can be bounded above by the proportion of support points in the data. The
reason is that leaving out an observation that is not a support vector will
not change the solution. Hence these observations, being classified correctly
by the original boundary, will be classified correctly in the cross-validation
process. However this bound tends to be too high, and not generally useful
for choosing C (62% and 85%, respectively, in our examples).</p>
<p/>
</div>
<div class="page"><p/>
<p>422 12. Flexible Discriminants
</p>
<p>.. . . .
</p>
<p>. . . . . . .. . . . . . . . . .
</p>
<p>. . . . . . . . . . . . .. . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . .. . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . .. . . . . . . .
. . . . ..
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>Training Error: 0.270
</p>
<p>Test Error:       0.288
</p>
<p>Bayes Error:    0.210
</p>
<p>C = 10000
</p>
<p>.. .
</p>
<p>. . . .. . . . .
</p>
<p>. . . . . . .. . . . . . . .
</p>
<p>. . . . . . . . . .. . . . . . . . . . .
</p>
<p>. . . . . . . . . . . .. . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . .
. . . . . . . . . . . . . .. . . . . . . . . . . .
</p>
<p>. . . . . . . . . . .. . . . . . . . .
. . . . . . . .. . . . . . .
</p>
<p>. . . . .. . . .
. ..
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>&bull;
</p>
<p>Training Error: 0.26
</p>
<p>Test Error:       0.30
</p>
<p>Bayes Error:    0.21
</p>
<p>C = 0.01
</p>
<p>FIGURE 12.2. The linear support vector boundary for the mixture data exam-
ple with two overlapping classes, for two different values of C. The broken lines
indicate the margins, where f(x) = &plusmn;1. The support points (αi &gt; 0) are all the
points on the wrong side of their margin. The black solid dots are those support
points falling exactly on the margin (ξi = 0, αi &gt; 0). In the upper panel 62% of
the observations are support points, while in the lower panel 85% are. The broken
purple curve in the background is the Bayes decision boundary.</p>
<p/>
</div>
<div class="page"><p/>
<p>12.3 Support Vector Machines and Kernels 423
</p>
<p>12.3 Support Vector Machines and Kernels
</p>
<p>The support vector classifier described so far finds linear boundaries in the
input feature space. As with other linear methods, we can make the pro-
cedure more flexible by enlarging the feature space using basis expansions
such as polynomials or splines (Chapter 5). Generally linear boundaries
in the enlarged space achieve better training-class separation, and trans-
late to nonlinear boundaries in the original space. Once the basis functions
hm(x), m = 1, . . . ,M are selected, the procedure is the same as before. We
fit the SV classifier using input features h(xi) = (h1(xi), h2(xi), . . . , hM (xi)),
</p>
<p>i = 1, . . . , N , and produce the (nonlinear) function f̂(x) = h(x)T β̂ + β̂0.
</p>
<p>The classifier is Ĝ(x) = sign(f̂(x)) as before.
The support vector machine classifier is an extension of this idea, where
</p>
<p>the dimension of the enlarged space is allowed to get very large, infinite
in some cases. It might seem that the computations would become pro-
hibitive. It would also seem that with sufficient basis functions, the data
would be separable, and overfitting would occur. We first show how the
SVM technology deals with these issues. We then see that in fact the SVM
classifier is solving a function-fitting problem using a particular criterion
and form of regularization, and is part of a much bigger class of problems
that includes the smoothing splines of Chapter 5. The reader may wish
to consult Section 5.8, which provides background material and overlaps
somewhat with the next two sections.
</p>
<p>12.3.1 Computing the SVM for Classification
</p>
<p>We can represent the optimization problem (12.9) and its solution in a
special way that only involves the input features via inner products. We do
this directly for the transformed feature vectors h(xi). We then see that for
particular choices of h, these inner products can be computed very cheaply.
The Lagrange dual function (12.13) has the form
</p>
<p>LD =
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>αi &minus;
1
</p>
<p>2
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>N&sum;
</p>
<p>i&prime;=1
</p>
<p>αiαi&prime;yiyi&prime;〈h(xi), h(xi&prime;)〉. (12.19)
</p>
<p>From (12.10) we see that the solution function f(x) can be written
</p>
<p>f(x) = h(x)Tβ + β0
</p>
<p>=
N&sum;
</p>
<p>i=1
</p>
<p>αiyi〈h(x), h(xi)〉+ β0. (12.20)
</p>
<p>As before, given αi, β0 can be determined by solving yif(xi) = 1 in (12.20)
for any (or all) xi for which 0 &lt; αi &lt; C.</p>
<p/>
</div>
<div class="page"><p/>
<p>424 12. Flexible Discriminants
</p>
<p>So both (12.19) and (12.20) involve h(x) only through inner products. In
fact, we need not specify the transformation h(x) at all, but require only
knowledge of the kernel function
</p>
<p>K(x, x&prime;) = 〈h(x), h(x&prime;)〉 (12.21)
</p>
<p>that computes inner products in the transformed space. K should be a
symmetric positive (semi-) definite function; see Section 5.8.1.
Three popular choices for K in the SVM literature are
</p>
<p>dth-Degree polynomial: K(x, x&prime;) = (1 + 〈x, x&prime;〉)d,
Radial basis: K(x, x&prime;) = exp(&minus;γ‖x&minus; x&prime;‖2),
</p>
<p>Neural network: K(x, x&prime;) = tanh(κ1〈x, x&prime;〉+ κ2).
(12.22)
</p>
<p>Consider for example a feature space with two inputs X1 and X2, and a
polynomial kernel of degree 2. Then
</p>
<p>K(X,X &prime;) = (1 + 〈X,X &prime;〉)2
</p>
<p>= (1 +X1X
&prime;
1 +X2X
</p>
<p>&prime;
2)
</p>
<p>2
</p>
<p>= 1 + 2X1X
&prime;
1 + 2X2X
</p>
<p>&prime;
2 + (X1X
</p>
<p>&prime;
1)
</p>
<p>2 + (X2X
&prime;
2)
</p>
<p>2 + 2X1X
&prime;
1X2X
</p>
<p>&prime;
2.
</p>
<p>(12.23)
</p>
<p>Then M = 6, and if we choose h1(X) = 1, h2(X) =
&radic;
2X1, h3(X) =&radic;
</p>
<p>2X2, h4(X) = X
2
1 , h5(X) = X
</p>
<p>2
2 , and h6(X) =
</p>
<p>&radic;
2X1X2, thenK(X,X
</p>
<p>&prime;) =
〈h(X), h(X &prime;)〉. From (12.20) we see that the solution can be written
</p>
<p>f̂(x) =
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>α̂iyiK(x, xi) + β̂0. (12.24)
</p>
<p>The role of the parameter C is clearer in an enlarged feature space,
since perfect separation is often achievable there. A large value of C will
discourage any positive ξi, and lead to an overfit wiggly boundary in the
original feature space; a small value of C will encourage a small value of
‖β‖, which in turn causes f(x) and hence the boundary to be smoother.
Figure 12.3 show two nonlinear support vector machines applied to the
mixture example of Chapter 2. The regularization parameter was chosen
in both cases to achieve good test error. The radial basis kernel produces
a boundary quite similar to the Bayes optimal boundary for this example;
compare Figure 2.5.
In the early literature on support vectors, there were claims that the
</p>
<p>kernel property of the support vector machine is unique to it and allows
one to finesse the curse of dimensionality. Neither of these claims is true,
and we go into both of these issues in the next three subsections.</p>
<p/>
</div>
<div class="page"><p/>
<p>12.3 Support Vector Machines and Kernels 425
</p>
<p>SVM - Degree-4 Polynomial in Feature Space
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . .. . . . . . . . . . . . . . .
. . . . . . . . . . . . . . .. . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . .. . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . .. . . . . . . . . . . . . .
. . . . . . . . . . . . . . .. . . . . . . . . . . . . . .
. . . . . . . . . . . . . .. . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . .. . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . .. . . . . . . . . . . . . .
. . . . . . . . . . . . . .. . . . . . . . . . . . .
. . . . . . . . . . . . .. . . . . . . . . . . .
. . . . . . . . . . .. . . . . . . . . . .
. . . . . . . . . .. . . . . . . . .
</p>
<p>. . . . . . .. . . . .
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>&bull;&bull;
</p>
<p>&bull; &bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull; &bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>Training Error: 0.180
</p>
<p>Test Error:       0.245
</p>
<p>Bayes Error:    0.210
</p>
<p>SVM - Radial Kernel in Feature Space
</p>
<p>. . . . . . .. . . . . . .
</p>
<p>. . . . . .. . . . . .
</p>
<p>. . . . .. . . . .
</p>
<p>. . . .. . . .
</p>
<p>. . . .. . . .
</p>
<p>. . .. . .
</p>
<p>. . .. . .
</p>
<p>. . .. . .
</p>
<p>. . .. . .
</p>
<p>. . .. . .
</p>
<p>. . .. . . .
</p>
<p>. . . .. . . .
</p>
<p>. . . .. . . . .
</p>
<p>. . . . .. . . . . .
</p>
<p>. . . . . .. . . . . . .
</p>
<p>. . . . . . .. . . . . . . .
</p>
<p>. . . . . . . .. . . . . . . . .
</p>
<p>. . . . . . . . .. . . . . . . . . .
</p>
<p>. . . . . . . . . .. . . . . . . . . . .
</p>
<p>. . . . . . . . . . . .. . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . .
. . . . . . . . . . . .. . . . . . . . . . . .
. . . . . . . . . . . .. . . . . . . . . . . .
. . . . . . . . . . . .. . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . .. . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . .. . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . .. . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . .. . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . .. . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . .. . . . . . . . . . . . .
. . . . . . . . . . . . .. . . . . . . . . . . . . .
. . . . . . . . . . . . . .. . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . .. . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . .. . . . . . . . . . . . .
. . . . . . . . . . . . .. . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . .. . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . .. . . . . . . . . . . . .
. . . . . . . . . . . . .. . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . .. . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . .. . . . . . . . . . . . .
. . . . . . . . . . . . . .
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>Training Error: 0.160
</p>
<p>Test Error:       0.218
</p>
<p>Bayes Error:    0.210
</p>
<p>FIGURE 12.3. Two nonlinear SVMs for the mixture data. The upper plot uses
a 4th degree polynomial kernel, the lower a radial basis kernel (with γ = 1). In
each case C was tuned to approximately achieve the best test error performance,
and C = 1 worked well in both cases. The radial basis kernel performs the best
(close to Bayes optimal), as might be expected given the data arise from mixtures
of Gaussians. The broken purple curve in the background is the Bayes decision
boundary.</p>
<p/>
</div>
<div class="page"><p/>
<p>426 12. Flexible Discriminants
</p>
<p>&minus;3 &minus;2 &minus;1 0 1 2 3
</p>
<p>0
.0
</p>
<p>0
.5
</p>
<p>1
.0
</p>
<p>1
.5
</p>
<p>2
.0
</p>
<p>2
.5
</p>
<p>3
.0
</p>
<p>Hinge Loss
Binomial Deviance
Squared Error
Class Huber
</p>
<p>L
o
ss
</p>
<p>yf
</p>
<p>FIGURE 12.4. The support vector loss function (hinge loss), compared to the
negative log-likelihood loss (binomial deviance) for logistic regression, squared-er-
ror loss, and a &ldquo;Huberized&rdquo; version of the squared hinge loss. All are shown as a
function of yf rather than f , because of the symmetry between the y = +1 and
y = &minus;1 case. The deviance and Huber have the same asymptotes as the SVM
loss, but are rounded in the interior. All are scaled to have the limiting left-tail
slope of &minus;1.
</p>
<p>12.3.2 The SVM as a Penalization Method
</p>
<p>With f(x) = h(x)Tβ + β0, consider the optimization problem
</p>
<p>min
β0, β
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>[1&minus; yif(xi)]+ +
λ
</p>
<p>2
‖β‖2 (12.25)
</p>
<p>where the subscript &ldquo;+&rdquo; indicates positive part. This has the form loss +
penalty, which is a familiar paradigm in function estimation. It is easy to
show (Exercise 12.1) that the solution to (12.25), with λ = 1/C, is the
same as that for (12.8).
Examination of the &ldquo;hinge&rdquo; loss function L(y, f) = [1&minus; yf ]+ shows that
</p>
<p>it is reasonable for two-class classification, when compared to other more
traditional loss functions. Figure 12.4 compares it to the log-likelihood loss
for logistic regression, as well as squared-error loss and a variant thereof.
The (negative) log-likelihood or binomial deviance has similar tails as the
SVM loss, giving zero penalty to points well inside their margin, and a</p>
<p/>
</div>
<div class="page"><p/>
<p>12.3 Support Vector Machines and Kernels 427
</p>
<p>TABLE 12.1. The population minimizers for the different loss functions in Fig-
ure 12.4. Logistic regression uses the binomial log-likelihood or deviance. Linear
discriminant analysis (Exercise 4.2) uses squared-error loss. The SVM hinge loss
estimates the mode of the posterior class probabilities, whereas the others estimate
a linear transformation of these probabilities.
</p>
<p>Loss Function L[y, f(x)] Minimizing Function
</p>
<p>Binomial
Deviance log[1 + e&minus;yf(x)] f(x) = log
</p>
<p>Pr(Y = +1|x)
Pr(Y = -1|x)
</p>
<p>SVM Hinge
Loss
</p>
<p>[1&minus; yf(x)]+ f(x) = sign[Pr(Y = +1|x)&minus; 12 ]
</p>
<p>Squared
Error
</p>
<p>[y &minus; f(x)]2 = [1&minus; yf(x)]2 f(x) = 2Pr(Y = +1|x)&minus; 1
</p>
<p>&ldquo;Huberised&rdquo;
Square
Hinge Loss
</p>
<p>&minus;4yf(x), yf(x) &lt; -1
[1&minus; yf(x)]2+ otherwise
</p>
<p>f(x) = 2Pr(Y = +1|x)&minus; 1
</p>
<p>linear penalty to points on the wrong side and far away. Squared-error, on
the other hand gives a quadratic penalty, and points well inside their own
margin have a strong influence on the model as well. The squared hinge
loss L(y, f) = [1 &minus; yf ]2+ is like the quadratic, except it is zero for points
inside their margin. It still rises quadratically in the left tail, and will be
less robust than hinge or deviance to misclassified observations. Recently
Rosset and Zhu (2007) proposed a &ldquo;Huberized&rdquo; version of the squared hinge
loss, which converts smoothly to a linear loss at yf = &minus;1.
We can characterize these loss functions in terms of what they are es-
</p>
<p>timating at the population level. We consider minimizing EL(Y, f(X)).
Table 12.1 summarizes the results. Whereas the hinge loss estimates the
classifier G(x) itself, all the others estimate a transformation of the class
posterior probabilities. The &ldquo;Huberized&rdquo; square hinge loss shares attractive
properties of logistic regression (smooth loss function, estimates probabili-
ties), as well as the SVM hinge loss (support points).
Formulation (12.25) casts the SVM as a regularized function estimation
</p>
<p>problem, where the coefficients of the linear expansion f(x) = β0+h(x)
Tβ
</p>
<p>are shrunk toward zero (excluding the constant). If h(x) represents a hierar-
chical basis having some ordered structure (such as ordered in roughness),</p>
<p/>
</div>
<div class="page"><p/>
<p>428 12. Flexible Discriminants
</p>
<p>then the uniform shrinkage makes more sense if the rougher elements hj in
the vector h have smaller norm.
All the loss-functions in Table 12.1 except squared-error are so called
</p>
<p>&ldquo;margin maximizing loss-functions&rdquo; (Rosset et al., 2004b). This means that
</p>
<p>if the data are separable, then the limit of β̂λ in (12.25) as λ &rarr; 0 defines
the optimal separating hyperplane1.
</p>
<p>12.3.3 Function Estimation and Reproducing Kernels
</p>
<p>Here we describe SVMs in terms of function estimation in reproducing
kernel Hilbert spaces, where the kernel property abounds. This material is
discussed in some detail in Section 5.8. This provides another view of the
support vector classifier, and helps to clarify how it works.
Suppose the basis h arises from the (possibly finite) eigen-expansion of
</p>
<p>a positive definite kernel K,
</p>
<p>K(x, x&prime;) =
</p>
<p>&infin;&sum;
</p>
<p>m=1
</p>
<p>φm(x)φm(x
&prime;)δm (12.26)
</p>
<p>and hm(x) =
&radic;
δmφm(x). Then with θm =
</p>
<p>&radic;
δmβm, we can write (12.25)
</p>
<p>as
</p>
<p>min
β0, θ
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>[
1&minus; yi(β0 +
</p>
<p>&infin;&sum;
</p>
<p>m=1
</p>
<p>θmφm(xi))
</p>
<p>]
</p>
<p>+
</p>
<p>+
λ
</p>
<p>2
</p>
<p>&infin;&sum;
</p>
<p>m=1
</p>
<p>θ2m
δm
</p>
<p>. (12.27)
</p>
<p>Now (12.27) is identical in form to (5.49) on page 169 in Section 5.8, and
the theory of reproducing kernel Hilbert spaces described there guarantees
a finite-dimensional solution of the form
</p>
<p>f(x) = β0 +
N&sum;
</p>
<p>i=1
</p>
<p>αiK(x, xi). (12.28)
</p>
<p>In particular we see there an equivalent version of the optimization crite-
rion (12.19) [Equation (5.67) in Section 5.8.2; see also Wahba et al. (2000)],
</p>
<p>min
β0,α
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>(1&minus; yif(xi))+ +
λ
</p>
<p>2
αTKα, (12.29)
</p>
<p>where K is the N &times;N matrix of kernel evaluations for all pairs of training
features (Exercise 12.2).
These models are quite general, and include, for example, the entire fam-
</p>
<p>ily of smoothing splines, additive and interaction spline models discussed
</p>
<p>1For logistic regression with separable data, β̂λ diverges, but β̂λ/‖β̂λ‖ converges to
</p>
<p>the optimal separating direction.</p>
<p/>
</div>
<div class="page"><p/>
<p>12.3 Support Vector Machines and Kernels 429
</p>
<p>in Chapters 5 and 9, and in more detail in Wahba (1990) and Hastie and
Tibshirani (1990). They can be expressed more generally as
</p>
<p>min
f&isin;H
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>[1&minus; yif(xi)]+ + λJ(f), (12.30)
</p>
<p>where H is the structured space of functions, and J(f) an appropriate reg-
ularizer on that space. For example, suppose H is the space of additive
functions f(x) =
</p>
<p>&sum;p
j=1 fj(xj), and J(f) =
</p>
<p>&sum;
j
</p>
<p>&int;
{f &prime;&prime;j(xj)}2dxj . Then the
</p>
<p>solution to (12.30) is an additive cubic spline, and has a kernel representa-
tion (12.28) with K(x, x&prime;) =
</p>
<p>&sum;p
j=1 Kj(xj , x
</p>
<p>&prime;
j). Each of the Kj is the kernel
</p>
<p>appropriate for the univariate smoothing spline in xj (Wahba, 1990).
Conversely this discussion also shows that, for example, any of the kernels
</p>
<p>described in (12.22) above can be used with any convex loss function, and
will also lead to a finite-dimensional representation of the form (12.28).
Figure 12.5 uses the same kernel functions as in Figure 12.3, except using
the binomial log-likelihood as a loss function2. The fitted function is hence
an estimate of the log-odds,
</p>
<p>f̂(x) = log
P̂r(Y = +1|x)
P̂r(Y = &minus;1|x)
</p>
<p>= β̂0 +
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>α̂iK(x, xi), (12.31)
</p>
<p>or conversely we get an estimate of the class probabilities
</p>
<p>P̂r(Y = +1|x) = 1
1 + e&minus;β̂0&minus;
</p>
<p>&sum;
N
i=1 α̂iK(x,xi)
</p>
<p>. (12.32)
</p>
<p>The fitted models are quite similar in shape and performance. Examples
and more details are given in Section 5.8.
It does happen that for SVMs, a sizable fraction of the N values of αi
</p>
<p>can be zero (the nonsupport points). In the two examples in Figure 12.3,
these fractions are 42% and 45%, respectively. This is a consequence of the
piecewise linear nature of the first part of the criterion (12.25). The lower
the class overlap (on the training data), the greater this fraction will be.
Reducing λ will generally reduce the overlap (allowing a more flexible f).
</p>
<p>A small number of support points means that f̂(x) can be evaluated more
quickly, which is important at lookup time. Of course, reducing the overlap
too much can lead to poor generalization.
</p>
<p>2Ji Zhu assisted in the preparation of these examples.</p>
<p/>
</div>
<div class="page"><p/>
<p>430 12. Flexible Discriminants
</p>
<p>LR - Degree-4 Polynomial in Feature Space
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . .. . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . .. . . . . . . . . . . . . . .
. . . . . . . . . . . . . . .. . . . . . . . . . . . . .
. . . . . . . . . . . . . .. . . . . . . . . . . . .
. . . . . . . . . . . .. . . . . . . . . . .
. . . . . . . . . .. . . . . . . .
</p>
<p>. . . . . . .. . .
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>Training Error: 0.190
</p>
<p>Test Error:       0.263
</p>
<p>Bayes Error:    0.210
</p>
<p>LR - Radial Kernel in Feature Space
</p>
<p>.. . . .
. . . . .. . . . . .
. . . . . . .. . . . . . . .
. . . . . . . . .. . . . . . . . . .
. . . . . . . . . .. . . . . . . . . . .
. . . . . . . . . . . .. . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . .. . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . .. . . . . . . . . . . . .
. . . . . . . . . . . . . .. . . . . . . . . . . . . .
. . . . . . . . . . . . . .. . . . . . . . . . . . . .
. . . . . . . . . . . . . .. . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . .. . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . .. . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . .. . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . .. . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . .. . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . .. . . . . . . . . . . . . . .
. . . . . . . . . . . . . .. . . . . . . . . . . . . .
. . . . . . . . . . . . . .. . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . .. . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . .. . . . . . . . . . . . .
. . . . . . . . . . . . .. . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . .. . . . . . . . . . . . .
. . . . . . . . . . . .. . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . .. . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . .
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>Training Error: 0.150
</p>
<p>Test Error:       0.221
</p>
<p>Bayes Error:    0.210
</p>
<p>FIGURE 12.5. The logistic regression versions of the SVM models in Fig-
ure 12.3, using the identical kernels and hence penalties, but the log-likelihood
loss instead of the SVM loss function. The two broken contours correspond to
posterior probabilities of 0.75 and 0.25 for the +1 class (or vice versa). The bro-
ken purple curve in the background is the Bayes decision boundary.</p>
<p/>
</div>
<div class="page"><p/>
<p>12.3 Support Vector Machines and Kernels 431
</p>
<p>TABLE 12.2. Skin of the orange: Shown are mean (standard error of the mean)
of the test error over 50 simulations. BRUTO fits an additive spline model adap-
tively, while MARS fits a low-order interaction model adaptively.
</p>
<p>Test Error (SE)
Method No Noise Features Six Noise Features
</p>
<p>1 SV Classifier 0.450 (0.003) 0.472 (0.003)
2 SVM/poly 2 0.078 (0.003) 0.152 (0.004)
3 SVM/poly 5 0.180 (0.004) 0.370 (0.004)
4 SVM/poly 10 0.230 (0.003) 0.434 (0.002)
5 BRUTO 0.084 (0.003) 0.090 (0.003)
6 MARS 0.156 (0.004) 0.173 (0.005)
</p>
<p>Bayes 0.029 0.029
</p>
<p>12.3.4 SVMs and the Curse of Dimensionality
</p>
<p>In this section, we address the question of whether SVMs have some edge
on the curse of dimensionality. Notice that in expression (12.23) we are not
allowed a fully general inner product in the space of powers and products.
For example, all terms of the form 2XjX
</p>
<p>&prime;
j are given equal weight, and the
</p>
<p>kernel cannot adapt itself to concentrate on subspaces. If the number of
features p were large, but the class separation occurred only in the linear
subspace spanned by say X1 and X2, this kernel would not easily find the
structure and would suffer from having many dimensions to search over.
One would have to build knowledge about the subspace into the kernel;
that is, tell it to ignore all but the first two inputs. If such knowledge were
available a priori, much of statistical learning would be made much easier.
A major goal of adaptive methods is to discover such structure.
We support these statements with an illustrative example. We generated
</p>
<p>100 observations in each of two classes. The first class has four standard
normal independent features X1, X2, X3, X4. The second class also has four
standard normal independent features, but conditioned on 9 &le; &sum;X2j &le; 16.
This is a relatively easy problem. As a second harder problem, we aug-
mented the features with an additional six standard Gaussian noise fea-
tures. Hence the second class almost completely surrounds the first, like the
skin surrounding the orange, in a four-dimensional subspace. The Bayes er-
ror rate for this problem is 0.029 (irrespective of dimension). We generated
1000 test observations to compare different procedures. The average test
errors over 50 simulations, with and without noise features, are shown in
Table 12.2.
Line 1 uses the support vector classifier in the original feature space.
</p>
<p>Lines 2&ndash;4 refer to the support vector machine with a 2-, 5- and 10-dimension-
al polynomial kernel. For all support vector procedures, we chose the cost
parameter C to minimize the test error, to be as fair as possible to the</p>
<p/>
</div>
<div class="page"><p/>
<p>432 12. Flexible Discriminants
</p>
<p>1e&minus;01 1e+01 1e+03
</p>
<p>0
.2
</p>
<p>0
0
.2
</p>
<p>5
0
.3
</p>
<p>0
0
.3
</p>
<p>5
</p>
<p>1e&minus;01 1e+01 1e+03 1e&minus;01 1e+01 1e+03 1e&minus;01 1e+01 1e+03
</p>
<p>T
e
s
t 
E
</p>
<p>rr
o
r
</p>
<p>C
</p>
<p>Test Error Curves &minus; SVM with Radial Kernel
</p>
<p>γ = 5 γ = 1 γ = 0.5 γ = 0.1
</p>
<p>FIGURE 12.6. Test-error curves as a function of the cost parameter C for
the radial-kernel SVM classifier on the mixture data. At the top of each plot is
the scale parameter γ for the radial kernel: Kγ(x, y) = exp (&minus;γ||x&minus; y||2). The
optimal value for C depends quite strongly on the scale of the kernel. The Bayes
error rate is indicated by the broken horizontal lines.
</p>
<p>method. Line 5 fits an additive spline model to the (&minus;1,+1) response by
least squares, using the BRUTO algorithm for additive models, described
in Hastie and Tibshirani (1990). Line 6 uses MARS (multivariate adaptive
regression splines) allowing interaction of all orders, as described in Chap-
ter 9; as such it is comparable with the SVM/poly 10. Both BRUTO and
MARS have the ability to ignore redundant variables. Test error was not
used to choose the smoothing parameters in either of lines 5 or 6.
In the original feature space, a hyperplane cannot separate the classes,
</p>
<p>and the support vector classifier (line 1) does poorly. The polynomial sup-
port vector machine makes a substantial improvement in test error rate,
but is adversely affected by the six noise features. It is also very sensitive to
the choice of kernel: the second degree polynomial kernel (line 2) does best,
since the true decision boundary is a second-degree polynomial. However,
higher-degree polynomial kernels (lines 3 and 4) do much worse. BRUTO
performs well, since the boundary is additive. BRUTO and MARS adapt
well: their performance does not deteriorate much in the presence of noise.
</p>
<p>12.3.5 A Path Algorithm for the SVM Classifier
</p>
<p>The regularization parameter for the SVM classifier is the cost parameter
C, or its inverse λ in (12.25). Common usage is to set C high, leading often
to somewhat overfit classifiers.
Figure 12.6 shows the test error on the mixture data as a function of
</p>
<p>C, using different radial-kernel parameters γ. When γ = 5 (narrow peaked
kernels), the heaviest regularization (small C) is called for. With γ = 1</p>
<p/>
</div>
<div class="page"><p/>
<p>12.3 Support Vector Machines and Kernels 433
</p>
<p>&minus;0.5 0.0 0.5 1.0 1.5 2.0
</p>
<p>&minus;
1
</p>
<p>.0
&minus;
</p>
<p>0
.5
</p>
<p>0
.0
</p>
<p>0
.5
</p>
<p>1
.0
</p>
<p>1
.5
</p>
<p>7
</p>
<p>8
</p>
<p>9
</p>
<p>10
</p>
<p>11
</p>
<p>12
</p>
<p>1
</p>
<p>2
</p>
<p>3
</p>
<p>4
</p>
<p>5
</p>
<p>6
</p>
<p>1/||β||f(x) = 0
</p>
<p>f(x) = +1
</p>
<p>f(x) = &minus;1
</p>
<p>0.0 0.2 0.4 0.6 0.8 1.0
</p>
<p>0
2
</p>
<p>4
6
</p>
<p>8
1
</p>
<p>0
</p>
<p>1
</p>
<p>2
</p>
<p>3
</p>
<p>4
</p>
<p>5
</p>
<p>6
</p>
<p>7
</p>
<p>8
</p>
<p>9
</p>
<p>10
</p>
<p>11
</p>
<p>12
</p>
<p>αi(λ)
λ
</p>
<p>FIGURE 12.7. A simple example illustrates the SVM path algorithm. (left
panel:) This plot illustrates the state of the model at λ = 1/2. The &lsquo;&lsquo; + 1&rdquo; points
are orange, the &ldquo;&minus;1&rdquo; blue. The width of the soft margin is 2/||β|| = 2 &times; 0.587.
Two blue points {3, 5} are misclassified, while the two orange points {10, 12} are
correctly classified, but on the wrong side of their margin f(x) = +1; each of
these has yif(xi) &lt; 1. The three square shaped points {2, 6, 7} are exactly on
their margins. (right panel:) This plot shows the piecewise linear profiles αi(λ).
The horizontal broken line at λ = 1/2 indicates the state of the αi for the model
in the left plot.
</p>
<p>(the value used in Figure 12.3), an intermediate value of C is required.
Clearly in situations such as these, we need to determine a good choice
for C, perhaps by cross-validation. Here we describe a path algorithm (in
the spirit of Section 3.8) for efficiently fitting the entire sequence of SVM
models obtained by varying C.
It is convenient to use the loss+penalty formulation (12.25), along with
</p>
<p>Figure 12.4. This leads to a solution for β at a given value of λ:
</p>
<p>βλ =
1
</p>
<p>λ
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>αiyixi. (12.33)
</p>
<p>The αi are again Lagrange multipliers, but in this case they all lie in [0, 1].
Figure 12.7 illustrates the setup. It can be shown that the KKT optimal-
</p>
<p>ity conditions imply that the labeled points (xi, yi) fall into three distinct
groups:</p>
<p/>
</div>
<div class="page"><p/>
<p>434 12. Flexible Discriminants
</p>
<p>&bull; Observations correctly classified and outside their margins. They have
yif(xi) &gt; 1, and Lagrange multipliers αi = 0. Examples are the
orange points 8, 9 and 11, and the blue points 1 and 4.
</p>
<p>&bull; Observations sitting on their margins with yif(xi) = 1, with Lagrange
multipliers αi &isin; [0, 1]. Examples are the orange 7 and the blue 2 and
6.
</p>
<p>&bull; Observations inside their margins have yif(xi) &lt; 1, with αi = 1.
Examples are the blue 3 and 5, and the orange 10 and 12.
</p>
<p>The idea for the path algorithm is as follows. Initially λ is large, the
margin 1/||βλ|| is wide, and all points are inside their margin and have
αi = 1. As λ decreases, 1/||βλ|| decreases, and the margin gets narrower.
Some points will move from inside their margins to outside their margins,
and their αi will change from 1 to 0. By continuity of the αi(λ), these points
will linger on the margin during this transition. From (12.33) we see that
the points with αi = 1 make fixed contributions to β(λ), and those with
αi = 0 make no contribution. So all that changes as λ decreases are the
αi &isin; [0, 1] of those (small number) of points on the margin. Since all these
points have yif(xi) = 1, this results in a small set of linear equations that
prescribe how αi(λ) and hence βλ changes during these transitions. This
results in piecewise linear paths for each of the αi(λ). The breaks occur
when points cross the margin. Figure 12.7 (right panel) shows the αi(λ)
profiles for the small example in the left panel.
Although we have described this for linear SVMs, exactly the same idea
</p>
<p>works for nonlinear models, in which (12.33) is replaced by
</p>
<p>fλ(x) =
1
</p>
<p>λ
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>αiyiK(x, xi). (12.34)
</p>
<p>Details can be found in Hastie et al. (2004). An R package svmpath is
available on CRAN for fitting these models.
</p>
<p>12.3.6 Support Vector Machines for Regression
</p>
<p>In this section we show how SVMs can be adapted for regression with a
quantitative response, in ways that inherit some of the properties of the
SVM classifier. We first discuss the linear regression model
</p>
<p>f(x) = xTβ + β0, (12.35)
</p>
<p>and then handle nonlinear generalizations. To estimate β, we consider min-
imization of
</p>
<p>H(β, β0) =
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>V (yi &minus; f(xi)) +
λ
</p>
<p>2
‖β‖2, (12.36)</p>
<p/>
</div>
<div class="page"><p/>
<p>12.3 Support Vector Machines and Kernels 435
</p>
<p>-4 -2 0 2 4
</p>
<p>-1
0
</p>
<p>1
2
</p>
<p>3
4
</p>
<p>-4 -2 0 2 4
</p>
<p>0
2
</p>
<p>4
6
</p>
<p>8
1
0
</p>
<p>1
2
</p>
<p>ǫ&minus;ǫ c&minus;c
</p>
<p>V
H
(r
)
</p>
<p>V
ǫ
(r
)
</p>
<p>rr
</p>
<p>FIGURE 12.8. The left panel shows the ǫ-insensitive error function used by the
support vector regression machine. The right panel shows the error function used
in Huber&rsquo;s robust regression (blue curve). Beyond |c|, the function changes from
quadratic to linear.
</p>
<p>where
</p>
<p>Vǫ(r) =
</p>
<p>{
0 if |r| &lt; ǫ,
|r| &minus; ǫ, otherwise. (12.37)
</p>
<p>This is an &ldquo;ǫ-insensitive&rdquo; error measure, ignoring errors of size less than
ǫ (left panel of Figure 12.8). There is a rough analogy with the support
vector classification setup, where points on the correct side of the deci-
sion boundary and far away from it, are ignored in the optimization. In
regression, these &ldquo;low error&rdquo; points are the ones with small residuals.
It is interesting to contrast this with error measures used in robust re-
</p>
<p>gression in statistics. The most popular, due to Huber (1964), has the form
</p>
<p>VH(r) =
</p>
<p>{
r2/2 if |r| &le; c,
c|r| &minus; c2/2, |r| &gt; c, (12.38)
</p>
<p>shown in the right panel of Figure 12.8. This function reduces from quadratic
to linear the contributions of observations with absolute residual greater
than a prechosen constant c. This makes the fitting less sensitive to out-
liers. The support vector error measure (12.37) also has linear tails (beyond
ǫ), but in addition it flattens the contributions of those cases with small
residuals.
If β̂, β̂0 are the minimizers of H, the solution function can be shown to
</p>
<p>have the form
</p>
<p>β̂ =
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>(α̂&lowast;i &minus; α̂i)xi, (12.39)
</p>
<p>f̂(x) =
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>(α̂&lowast;i &minus; α̂i)〈x, xi〉+ β0, (12.40)</p>
<p/>
</div>
<div class="page"><p/>
<p>436 12. Flexible Discriminants
</p>
<p>where α̂i, α̂
&lowast;
i are positive and solve the quadratic programming problem
</p>
<p>min
αi,α&lowast;i
</p>
<p>ǫ
N&sum;
</p>
<p>i=1
</p>
<p>(α&lowast;i + αi)&minus;
N&sum;
</p>
<p>i=1
</p>
<p>yi(α
&lowast;
i &minus; αi) +
</p>
<p>1
</p>
<p>2
</p>
<p>N&sum;
</p>
<p>i,i&prime;=1
</p>
<p>(α&lowast;i &minus; αi)(α&lowast;i&prime; &minus; αi&prime;)〈xi, xi&prime;〉
</p>
<p>subject to the constraints
</p>
<p>0 &le; αi, α&lowast;i &le; 1/λ,
N&sum;
</p>
<p>i=1
</p>
<p>(α&lowast;i &minus; αi) = 0, (12.41)
</p>
<p>αiα
&lowast;
i = 0.
</p>
<p>Due to the nature of these constraints, typically only a subset of the solution
values (α̂&lowast;i &minus; α̂i) are nonzero, and the associated data values are called the
support vectors. As was the case in the classification setting, the solution
depends on the input values only through the inner products 〈xi, xi&prime;〉. Thus
we can generalize the methods to richer spaces by defining an appropriate
inner product, for example, one of those defined in (12.22).
Note that there are parameters, ǫ and λ, associated with the criterion
</p>
<p>(12.36). These seem to play different roles. ǫ is a parameter of the loss
function Vǫ, just like c is for VH . Note that both Vǫ and VH depend on the
scale of y and hence r. If we scale our response (and hence use VH(r/σ) and
Vǫ(r/σ) instead), then we might consider using preset values for c and ǫ (the
value c = 1.345 achieves 95% efficiency for the Gaussian). The quantity λ
is a more traditional regularization parameter, and can be estimated for
example by cross-validation.
</p>
<p>12.3.7 Regression and Kernels
</p>
<p>As discussed in Section 12.3.3, this kernel property is not unique to sup-
port vector machines. Suppose we consider approximation of the regression
function in terms of a set of basis functions {hm(x)},m = 1, 2, . . . ,M :
</p>
<p>f(x) =
M&sum;
</p>
<p>m=1
</p>
<p>βmhm(x) + β0. (12.42)
</p>
<p>To estimate β and β0 we minimize
</p>
<p>H(β, β0) =
N&sum;
</p>
<p>i=1
</p>
<p>V (yi &minus; f(xi)) +
λ
</p>
<p>2
</p>
<p>&sum;
β2m (12.43)
</p>
<p>for some general error measure V (r). For any choice of V (r), the solution
</p>
<p>f̂(x) =
&sum;
</p>
<p>β̂mhm(x) + β̂0 has the form
</p>
<p>f̂(x) =
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>âiK(x, xi) (12.44)</p>
<p/>
</div>
<div class="page"><p/>
<p>12.3 Support Vector Machines and Kernels 437
</p>
<p>with K(x, y) =
&sum;M
</p>
<p>m=1 hm(x)hm(y). Notice that this has the same form
as both the radial basis function expansion and a regularization estimate,
discussed in Chapters 5 and 6.
For concreteness, let&rsquo;s work out the case V (r) = r2. Let H be the N&times;M
</p>
<p>basis matrix with imth element hm(xi), and suppose that M &gt; N is large.
For simplicity we assume that β0 = 0, or that the constant is absorbed in
h; see Exercise 12.3 for an alternative.
</p>
<p>We estimate β by minimizing the penalized least squares criterion
</p>
<p>H(β) = (y &minus;Hβ)T (y &minus;Hβ) + λ‖β‖2. (12.45)
</p>
<p>The solution is
</p>
<p>ŷ = Hβ̂ (12.46)
</p>
<p>with β̂ determined by
</p>
<p>&minus;HT (y &minus;Hβ̂) + λβ̂ = 0. (12.47)
</p>
<p>From this it appears that we need to evaluate the M &times;M matrix of inner
products in the transformed space. However, we can premultiply by H to
give
</p>
<p>Hβ̂ = (HHT + λI)&minus;1HHTy. (12.48)
</p>
<p>The N &times;N matrix HHT consists of inner products between pairs of obser-
vations i, i&prime;; that is, the evaluation of an inner product kernel {HHT }i,i&prime; =
K(xi, xi&prime;). It is easy to show (12.44) directly in this case, that the predicted
values at an arbitrary x satisfy
</p>
<p>f̂(x) = h(x)T β̂
</p>
<p>=
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>α̂iK(x, xi), (12.49)
</p>
<p>where α̂ = (HHT +λI)&minus;1y. As in the support vector machine, we need not
specify or evaluate the large set of functions h1(x), h2(x), . . . , hM (x). Only
the inner product kernel K(xi, xi&prime;) need be evaluated, at the N training
points for each i, i&prime; and at points x for predictions there. Careful choice
of hm (such as the eigenfunctions of particular, easy-to-evaluate kernels
K) means, for example, that HHT can be computed at a cost of N2/2
evaluations of K, rather than the direct cost N2M .
Note, however, that this property depends on the choice of squared norm
</p>
<p>‖β‖2 in the penalty. It does not hold, for example, for the L1 norm |β|,
which may lead to a superior model.</p>
<p/>
</div>
<div class="page"><p/>
<p>438 12. Flexible Discriminants
</p>
<p>12.3.8 Discussion
</p>
<p>The support vector machine can be extended to multiclass problems, es-
sentially by solving many two-class problems. A classifier is built for each
pair of classes, and the final classifier is the one that dominates the most
(Kressel, 1999; Friedman, 1996; Hastie and Tibshirani, 1998). Alternatively,
one could use the multinomial loss function along with a suitable kernel,
as in Section 12.3.3. SVMs have applications in many other supervised
and unsupervised learning problems. At the time of this writing, empirical
evidence suggests that it performs well in many real learning problems.
Finally, we mention the connection of the support vector machine and
</p>
<p>structural risk minimization (7.9). Suppose the training points (or their
basis expansion) are contained in a sphere of radius R, and let G(x) =
sign[f(x)] = sign[βTx+ β0] as in (12.2). Then one can show that the class
of functions {G(x), ‖β‖ &le; A} has VC-dimension h satisfying
</p>
<p>h &le; R2A2. (12.50)
If f(x) separates the training data, optimally for ‖β‖ &le; A, then with
probability at least 1&minus; η over training sets (Vapnik, 1996, page 139):
</p>
<p>Error Test &le; 4
h[log (2N/h) + 1]&minus; log (η/4)
</p>
<p>N
. (12.51)
</p>
<p>The support vector classifier was one of the first practical learning pro-
cedures for which useful bounds on the VC dimension could be obtained,
and hence the SRM program could be carried out. However in the deriva-
tion, balls are put around the data points&mdash;a process that depends on the
observed values of the features. Hence in a strict sense, the VC complexity
of the class is not fixed a priori, before seeing the features.
The regularization parameter C controls an upper bound on the VC
</p>
<p>dimension of the classifier. Following the SRM paradigm, we could choose C
by minimizing the upper bound on the test error, given in (12.51). However,
it is not clear that this has any advantage over the use of cross-validation
for choice of C.
</p>
<p>12.4 Generalizing Linear Discriminant Analysis
</p>
<p>In Section 4.3 we discussed linear discriminant analysis (LDA), a funda-
mental tool for classification. For the remainder of this chapter we discuss
a class of techniques that produce better classifiers than LDA by directly
generalizing LDA.
Some of the virtues of LDA are as follows:
</p>
<p>&bull; It is a simple prototype classifier. A new observation is classified to the
class with closest centroid. A slight twist is that distance is measured
in the Mahalanobis metric, using a pooled covariance estimate.</p>
<p/>
</div>
<div class="page"><p/>
<p>12.4 Generalizing Linear Discriminant Analysis 439
</p>
<p>&bull; LDA is the estimated Bayes classifier if the observations are multi-
variate Gaussian in each class, with a common covariance matrix.
Since this assumption is unlikely to be true, this might not seem to
be much of a virtue.
</p>
<p>&bull; The decision boundaries created by LDA are linear, leading to deci-
sion rules that are simple to describe and implement.
</p>
<p>&bull; LDA provides natural low-dimensional views of the data. For exam-
ple, Figure 12.12 is an informative two-dimensional view of data in
256 dimensions with ten classes.
</p>
<p>&bull; Often LDA produces the best classification results, because of its
simplicity and low variance. LDA was among the top three classifiers
for 7 of the 22 datasets studied in the STATLOG project (Michie et
al., 1994)3.
</p>
<p>Unfortunately the simplicity of LDA causes it to fail in a number of situa-
tions as well:
</p>
<p>&bull; Often linear decision boundaries do not adequately separate the classes.
When N is large, it is possible to estimate more complex decision
boundaries. Quadratic discriminant analysis (QDA) is often useful
here, and allows for quadratic decision boundaries. More generally
we would like to be able to model irregular decision boundaries.
</p>
<p>&bull; The aforementioned shortcoming of LDA can often be paraphrased
by saying that a single prototype per class is insufficient. LDA uses
a single prototype (class centroid) plus a common covariance matrix
to describe the spread of the data in each class. In many situations,
several prototypes are more appropriate.
</p>
<p>&bull; At the other end of the spectrum, we may have way too many (corre-
lated) predictors, for example, in the case of digitized analogue signals
and images. In this case LDA uses too many parameters, which are
estimated with high variance, and its performance suffers. In cases
such as this we need to restrict or regularize LDA even further.
</p>
<p>In the remainder of this chapter we describe a class of techniques that
attend to all these issues by generalizing the LDA model. This is achieved
largely by three different ideas.
The first idea is to recast the LDA problem as a linear regression problem.
</p>
<p>Many techniques exist for generalizing linear regression to more flexible,
nonparametric forms of regression. This in turn leads to more flexible forms
of discriminant analysis, which we call FDA. In most cases of interest, the
</p>
<p>3This study predated the emergence of SVMs.</p>
<p/>
</div>
<div class="page"><p/>
<p>440 12. Flexible Discriminants
</p>
<p>regression procedures can be seen to identify an enlarged set of predictors
via basis expansions. FDA amounts to LDA in this enlarged space, the
same paradigm used in SVMs.
In the case of too many predictors, such as the pixels of a digitized image,
</p>
<p>we do not want to expand the set: it is already too large. The second idea is
to fit an LDA model, but penalize its coefficients to be smooth or otherwise
coherent in the spatial domain, that is, as an image. We call this procedure
penalized discriminant analysis or PDA. With FDA itself, the expanded
basis set is often so large that regularization is also required (again as in
SVMs). Both of these can be achieved via a suitably regularized regression
in the context of the FDA model.
The third idea is to model each class by a mixture of two or more Gaus-
</p>
<p>sians with different centroids, but with every component Gaussian, both
within and between classes, sharing the same covariance matrix. This allows
for more complex decision boundaries, and allows for subspace reduction
as in LDA. We call this extension mixture discriminant analysis or MDA.
All three of these generalizations use a common framework by exploiting
</p>
<p>their connection with LDA.
</p>
<p>12.5 Flexible Discriminant Analysis
</p>
<p>In this section we describe a method for performing LDA using linear re-
gression on derived responses. This in turn leads to nonparametric and flex-
ible alternatives to LDA. As in Chapter 4, we assume we have observations
with a quantitative response G falling into one of K classes G = {1, . . . ,K},
each having measured features X. Suppose θ : G �&rarr; IR1 is a function that
assigns scores to the classes, such that the transformed class labels are op-
timally predicted by linear regression on X: If our training sample has the
form (gi, xi), i = 1, 2, . . . , N , then we solve
</p>
<p>min
β,θ
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>(
θ(gi)&minus; xTi β
</p>
<p>)2
, (12.52)
</p>
<p>with restrictions on θ to avoid a trivial solution (mean zero and unit vari-
ance over the training data). This produces a one-dimensional separation
between the classes.
More generally, we can find up to L &le; K&minus;1 sets of independent scorings
</p>
<p>for the class labels, θ1, θ2, . . . , θL, and L corresponding linear maps ηℓ(X) =
XTβℓ, ℓ = 1, . . . , L, chosen to be optimal for multiple regression in IR
</p>
<p>p. The
scores θℓ(g) and the maps βℓ are chosen to minimize the average squared
residual,
</p>
<p>ASR =
1
</p>
<p>N
</p>
<p>L&sum;
</p>
<p>ℓ=1
</p>
<p>[
N&sum;
</p>
<p>i=1
</p>
<p>(
θℓ(gi)&minus; xTi βℓ
</p>
<p>)2
]
. (12.53)</p>
<p/>
</div>
<div class="page"><p/>
<p>12.5 Flexible Discriminant Analysis 441
</p>
<p>The set of scores are assumed to be mutually orthogonal and normalized
with respect to an appropriate inner product to prevent trivial zero
solutions.
Why are we going down this road? It can be shown that the sequence
</p>
<p>of discriminant (canonical) vectors νℓ derived in Section 4.3.3 are identical
to the sequence βℓ up to a constant (Mardia et al., 1979; Hastie et al.,
1995). Moreover, the Mahalanobis distance of a test point x to the kth
class centroid μ̂k is given by
</p>
<p>δJ(x, μ̂k) =
K&minus;1&sum;
</p>
<p>ℓ=1
</p>
<p>wℓ(η̂ℓ(x)&minus; η̄kℓ )2 +D(x), (12.54)
</p>
<p>where η̄kℓ is the mean of the η̂ℓ(xi) in the kth class, and D(x) does not
depend on k. Here wℓ are coordinate weights that are defined in terms of
the mean squared residual r2ℓ of the ℓth optimally scored fit
</p>
<p>wℓ =
1
</p>
<p>r2ℓ (1&minus; r2ℓ )
. (12.55)
</p>
<p>In Section 4.3.2 we saw that these canonical distances are all that is needed
for classification in the Gaussian setup, with equal covariances in each class.
To summarize:
</p>
<p>LDA can be performed by a sequence of linear regressions, fol-
lowed by classification to the closest class centroid in the space
of fits. The analogy applies both to the reduced rank version,
or the full rank case when L = K &minus; 1.
</p>
<p>The real power of this result is in the generalizations that it invites. We
can replace the linear regression fits ηℓ(x) = x
</p>
<p>Tβℓ by far more flexible,
nonparametric fits, and by analogy achieve a more flexible classifier than
LDA. We have in mind generalized additive fits, spline functions, MARS
models and the like. In this more general form the regression problems are
defined via the criterion
</p>
<p>ASR({θℓ, ηℓ}Lℓ=1) =
1
</p>
<p>N
</p>
<p>L&sum;
</p>
<p>ℓ=1
</p>
<p>[
N&sum;
</p>
<p>i=1
</p>
<p>(θℓ(gi)&minus; ηℓ(xi))2 + λJ(ηℓ)
]
, (12.56)
</p>
<p>where J is a regularizer appropriate for some forms of nonparametric regres-
sion, such as smoothing splines, additive splines and lower-order ANOVA
spline models. Also included are the classes of functions and associated
penalties generated by kernels, as in Section 12.3.3.
Before we describe the computations involved in this generalization, let
</p>
<p>us consider a very simple example. Suppose we use degree-2 polynomial
regression for each ηℓ. The decision boundaries implied by the (12.54) will
be quadratic surfaces, since each of the fitted functions is quadratic, and as</p>
<p/>
</div>
<div class="page"><p/>
<p>442 12. Flexible Discriminants
</p>
<p>-2 0 2
</p>
<p>-2
0
</p>
<p>2
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>FIGURE 12.9. The data consist of 50 points generated from each of N(0, I) and
N(0, 9
</p>
<p>4
I). The solid black ellipse is the decision boundary found by FDA using
</p>
<p>degree-two polynomial regression. The dashed purple circle is the Bayes decision
boundary.
</p>
<p>in LDA their squares cancel out when comparing distances. We could have
achieved identical quadratic boundaries in a more conventional way, by
augmenting our original predictors with their squares and cross-products.
In the enlarged space one performs an LDA, and the linear boundaries in
the enlarged space map down to quadratic boundaries in the original space.
A classic example is a pair of multivariate Gaussians centered at the origin,
one having covariance matrix I, and the other cI for c &gt; 1; Figure 12.9
illustrates. The Bayes decision boundary is the sphere ‖x‖ = pc log c2(c&minus;1) , which
is a linear boundary in the enlarged space.
Many nonparametric regression procedures operate by generating a basis
</p>
<p>expansion of derived variables, and then performing a linear regression in
the enlarged space. The MARS procedure (Chapter 9) is exactly of this
form. Smoothing splines and additive spline models generate an extremely
large basis set (N&times;p basis functions for additive splines), but then perform
a penalized regression fit in the enlarged space. SVMs do as well; see also
the kernel-based regression example in Section 12.3.7. FDA in this case can
be shown to perform a penalized linear discriminant analysis in the enlarged
space. We elaborate in Section 12.6. Linear boundaries in the enlarged space
map down to nonlinear boundaries in the reduced space. This is exactly the
same paradigm that is used with support vector machines (Section 12.3).
We illustrate FDA on the speech recognition example used in Chapter 4,
</p>
<p>with K = 11 classes and p = 10 predictors. The classes correspond to</p>
<p/>
</div>
<div class="page"><p/>
<p>12.5 Flexible Discriminant Analysis 443
</p>
<p>o
ooooo
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
o
</p>
<p>o o
o
</p>
<p>oo
o
</p>
<p>o
o
o
</p>
<p>o o oo o
o
</p>
<p>oo
o
</p>
<p>ooo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o o
o
</p>
<p>ooo
</p>
<p>oooo
oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
ooooo
</p>
<p>o o
oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oooo
</p>
<p>oooo
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oooo
o
ooooo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>oooo
oo
</p>
<p>ooo o o
o
</p>
<p>o
ooooo
</p>
<p>ooo oo
o
</p>
<p>o o
o
o
</p>
<p>o
o
</p>
<p>ooooo
o
</p>
<p>oooooo
</p>
<p>o
o
</p>
<p>ooo
o
</p>
<p>oooooo
</p>
<p>oo
oooo
</p>
<p>ooooo
o
</p>
<p>oooooo
</p>
<p>o
o
</p>
<p>oooo
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o oooo
o
</p>
<p>oo
oo
</p>
<p>o
o
</p>
<p>oooo
oo
o
o
o
oo
o
</p>
<p>oooooo
</p>
<p>ooo
ooo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>oooooo
o o
oo o o
</p>
<p>o o
oooo
o
oooo
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>ooo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oooo
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>oo
oooo
</p>
<p>o
ooooo
</p>
<p>o
o
</p>
<p>oo o
o oo
</p>
<p>o
oo o
</p>
<p>o o o
</p>
<p>o
o
</p>
<p>o
oo
o
ooo
</p>
<p>o
oooo
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>oo
o
</p>
<p>oo
o
</p>
<p>o
oo
</p>
<p>o oo
o
o o
</p>
<p>ooo
oo
</p>
<p>o
</p>
<p>oo
ooo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
o oo
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
o
</p>
<p>o
o
ooo
</p>
<p>o
</p>
<p>oo
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
o
</p>
<p>oo
ooo
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
o
</p>
<p>o oo
o
</p>
<p>o
o
</p>
<p>oooo
o
o
</p>
<p>o
oo o
</p>
<p>o
o
</p>
<p>oooo
oo
</p>
<p>o o
o
o oo
</p>
<p>ooo
ooo
</p>
<p>o
o
</p>
<p>o
oo o
</p>
<p>o
oo o
</p>
<p>o o
</p>
<p>oo
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
o
</p>
<p>o o
o
</p>
<p>ooo
ooo
</p>
<p>o
oo
</p>
<p>oooooo
</p>
<p>o
o
</p>
<p>o
oo
o
</p>
<p>o oo
o
</p>
<p>o
oo
</p>
<p>oo
oo
</p>
<p>o
</p>
<p>oo
o
</p>
<p>ooo
</p>
<p>oo
oooo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o o
</p>
<p>Coordinate 1 for Training Data
</p>
<p>C
o
</p>
<p>o
rd
</p>
<p>in
a
</p>
<p>te
 2
</p>
<p> f
o
</p>
<p>r 
T
</p>
<p>ra
in
</p>
<p>in
g
</p>
<p> D
a
</p>
<p>ta
Linear Discriminant Analysis
</p>
<p>oooooo
</p>
<p>oo
</p>
<p>o
oo
</p>
<p>o
</p>
<p>ooo
o
o
o
</p>
<p>ooooo
o
</p>
<p>oo
oooo
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
o o
</p>
<p>o
o
</p>
<p>oo oo
o
</p>
<p>o
ooo
</p>
<p>o
</p>
<p>oooo
oo
</p>
<p>o
ooo
</p>
<p>oo o
ooo
</p>
<p>o
</p>
<p>o
oo
</p>
<p>oo
oo
</p>
<p>o
o
o o oo
</p>
<p>o
o
</p>
<p>oo
oo
</p>
<p>o
ooo
o
</p>
<p>o
</p>
<p>oooo o
oo oo
o
</p>
<p>oo
</p>
<p>ooooo
o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>oo
</p>
<p>o
o
oooo
</p>
<p>oooooo
</p>
<p>oooooo
ooooo
o
</p>
<p>oooooo
</p>
<p>oo
o
oo
</p>
<p>o
o
</p>
<p>o
ooo
</p>
<p>o
</p>
<p>ooo
ooo
</p>
<p>ooo
o
oo
</p>
<p>oooo
ooooooo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>ooo
oooo
</p>
<p>oo
oooo
</p>
<p>oo oo
oooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
ooooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
oo
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>oo
ooo
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
o
</p>
<p>ooooo
o
</p>
<p>o
o oo
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>oo o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
oo
</p>
<p>o
</p>
<p>o
oooo
</p>
<p>o
</p>
<p>oooo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
ooo
</p>
<p>ooo
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
oo
o
o o
oo
</p>
<p>o
o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>ooo o
</p>
<p>o
o
</p>
<p>o
o o o
</p>
<p>oooo
o
</p>
<p>o
</p>
<p>o
o
o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo o
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>oooo
oo
</p>
<p>o o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>o
o
</p>
<p>ooooo
o
</p>
<p>o
oo o
oo
</p>
<p>o
oooo o
</p>
<p>oo
ooo
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oooo
</p>
<p>o
ooooo
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o oo
o
</p>
<p>ooo
</p>
<p>oo
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>ooo o
</p>
<p>ooo
oo
o
oooo
o
</p>
<p>o
</p>
<p>oooooo
</p>
<p>o
o
oo
</p>
<p>oo
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>Coordinate 1 for Training Data
C
</p>
<p>o
o
</p>
<p>rd
in
</p>
<p>a
te
</p>
<p> 2
 f
</p>
<p>o
r 
</p>
<p>T
ra
</p>
<p>in
in
</p>
<p>g
 D
</p>
<p>a
ta
</p>
<p>Flexible Discriminant Analysis -- Bruto
</p>
<p>FIGURE 12.10. The left plot shows the first two LDA canonical variates for
the vowel training data. The right plot shows the corresponding projection when
FDA/BRUTO is used to fit the model; plotted are the fitted regression functions
η̂1(xi) and η̂2(xi). Notice the improved separation. The colors represent the eleven
different vowel sounds.
</p>
<p>11 vowel sounds, each contained in 11 different words. Here are the words,
preceded by the symbols that represent them:
</p>
<p>Vowel Word Vowel Word Vowel Word Vowel Word
</p>
<p>i: heed O hod I hid C: hoard
E head U hood A had u: who&rsquo;d
a: hard 3: heard Y hud
</p>
<p>Each of eight speakers spoke each word six times in the training set, and
likewise seven speakers in the test set. The ten predictors are derived from
the digitized speech in a rather complicated way, but standard in the speech
recognition world. There are thus 528 training observations, and 462 test
observations. Figure 12.10 shows two-dimensional projections produced by
LDA and FDA. The FDA model used adaptive additive-spline regression
functions to model the ηℓ(x), and the points plotted in the right plot have
coordinates η̂1(xi) and η̂2(xi). The routine used in S-PLUS is called bruto,
hence the heading on the plot and in Table 12.3. We see that flexible model-
ing has helped to separate the classes in this case. Table 12.3 shows training
and test error rates for a number of classification techniques. FDA/MARS
refers to Friedman&rsquo;s multivariate adaptive regression splines; degree = 2
means pairwise products are permitted. Notice that for FDA/MARS, the
best classification results are obtained in a reduced-rank subspace.</p>
<p/>
</div>
<div class="page"><p/>
<p>444 12. Flexible Discriminants
</p>
<p>TABLE 12.3. Vowel recognition data performance results. The results for neural
networks are the best among a much larger set, taken from a neural network
archive. The notation FDA/BRUTO refers to the regression method used with
FDA.
</p>
<p>Technique Error Rates
Training Test
</p>
<p>(1) LDA 0.32 0.56
Softmax 0.48 0.67
</p>
<p>(2) QDA 0.01 0.53
(3) CART 0.05 0.56
(4) CART (linear combination splits) 0.05 0.54
(5) Single-layer perceptron 0.67
(6) Multi-layer perceptron (88 hidden units) 0.49
(7) Gaussian node network (528 hidden units) 0.45
(8) Nearest neighbor 0.44
(9) FDA/BRUTO 0.06 0.44
</p>
<p>Softmax 0.11 0.50
(10) FDA/MARS (degree = 1) 0.09 0.45
</p>
<p>Best reduced dimension (=2) 0.18 0.42
Softmax 0.14 0.48
</p>
<p>(11) FDA/MARS (degree = 2) 0.02 0.42
Best reduced dimension (=6) 0.13 0.39
Softmax 0.10 0.50
</p>
<p>12.5.1 Computing the FDA Estimates
</p>
<p>The computations for the FDA coordinates can be simplified in many im-
portant cases, in particular when the nonparametric regression procedure
can be represented as a linear operator. We will denote this operator by
Sλ; that is, ŷ = Sλy, where y is the vector of responses and ŷ the vector
of fits. Additive splines have this property, if the smoothing parameters are
fixed, as does MARS once the basis functions are selected. The subscript λ
denotes the entire set of smoothing parameters. In this case optimal scoring
is equivalent to a canonical correlation problem, and the solution can be
computed by a single eigen-decomposition. This is pursued in Exercise 12.6,
and the resulting algorithm is presented here.
We create an N &times;K indicator response matrix Y from the responses gi,
</p>
<p>such that yik = 1 if gi = k, otherwise yik = 0. For a five-class problem Y
might look like the following:</p>
<p/>
</div>
<div class="page"><p/>
<p>12.5 Flexible Discriminant Analysis 445
</p>
<p>⎛
</p>
<p>⎜
</p>
<p>⎜
</p>
<p>⎜
</p>
<p>⎜
</p>
<p>⎜
</p>
<p>⎜
</p>
<p>⎜
</p>
<p>⎜
</p>
<p>⎜
</p>
<p>⎝
</p>
<p>C1 C2 C3 C4 C5
</p>
<p>g1 = 2 0 1 0 0 0
g2 = 1 1 0 0 0 0
g3 = 1 1 0 0 0 0
g4 = 5 0 0 0 0 1
g5 = 4 0 0 0 1 0
...
</p>
<p>...
gN = 3 0 0 1 0 0
</p>
<p>⎞
</p>
<p>⎟
</p>
<p>⎟
</p>
<p>⎟
</p>
<p>⎟
</p>
<p>⎟
</p>
<p>⎟
</p>
<p>⎟
</p>
<p>⎟
</p>
<p>⎟
</p>
<p>⎠
</p>
<p>Here are the computational steps:
</p>
<p>1. Multivariate nonparametric regression. Fit a multiresponse, adaptive
nonparametric regression of Y on X, giving fitted values Ŷ. Let Sλ
be the linear operator that fits the final chosen model, and η&lowast;(x) be
the vector of fitted regression functions.
</p>
<p>2. Optimal scores. Compute the eigen-decomposition of YT Ŷ = YTSλY,
where the eigenvectors Θ are normalized: ΘTDπΘ = I. Here Dπ =
YTY/N is a diagonal matrix of the estimated class prior
probabilities.
</p>
<p>3. Update the model from step 1 using the optimal scores: η(x) = ΘT η&lowast;(x).
</p>
<p>The first of the K functions in η(x) is the constant function&mdash; a trivial
solution; the remaining K&minus;1 functions are the discriminant functions. The
constant function, along with the normalization, causes all the remaining
functions to be centered.
Again Sλ can correspond to any regression method. When Sλ = HX , the
</p>
<p>linear regression projection operator, then FDA is linear discriminant anal-
ysis. The software that we reference in the Computational Considerations
section on page 455 makes good use of this modularity; the fda function
has a method= argument that allows one to supply any regression function,
as long as it follows some natural conventions. The regression functions
we provide allow for polynomial regression, adaptive additive models and
MARS. They all efficiently handle multiple responses, so step (1) is a single
call to a regression routine. The eigen-decomposition in step (2) simulta-
neously computes all the optimal scoring functions.
In Section 4.2 we discussed the pitfalls of using linear regression on an
</p>
<p>indicator response matrix as a method for classification. In particular, se-
vere masking can occur with three or more classes. FDA uses the fits from
such a regression in step (1), but then transforms them further to produce
useful discriminant functions that are devoid of these pitfalls. Exercise 12.9
takes another view of this phenomenon.</p>
<p/>
</div>
<div class="page"><p/>
<p>446 12. Flexible Discriminants
</p>
<p>12.6 Penalized Discriminant Analysis
</p>
<p>Although FDA is motivated by generalizing optimal scoring, it can also be
viewed directly as a form of regularized discriminant analysis. Suppose the
regression procedure used in FDA amounts to a linear regression onto a
basis expansion h(X), with a quadratic penalty on the coefficients:
</p>
<p>ASR({θℓ, βℓ}Lℓ=1) =
1
</p>
<p>N
</p>
<p>L&sum;
</p>
<p>ℓ=1
</p>
<p>[
N&sum;
</p>
<p>i=1
</p>
<p>(θℓ(gi)&minus; hT (xi)βℓ)2 + λβTℓ Ωβℓ
]
. (12.57)
</p>
<p>The choice of Ω depends on the problem. If ηℓ(x) = h(x)βℓ is an expansion
on spline basis functions, Ω might constrain ηℓ to be smooth over IR
</p>
<p>p. In
the case of additive splines, there are N spline basis functions for each
coordinate, resulting in a total of Np basis functions in h(x); Ω in this case
is Np&times;Np and block diagonal.
The steps in FDA can then be viewed as a generalized form of LDA,
</p>
<p>which we call penalized discriminant analysis, or PDA:
</p>
<p>&bull; Enlarge the set of predictors X via a basis expansion h(X).
</p>
<p>&bull; Use (penalized) LDA in the enlarged space, where the penalized
Mahalanobis distance is given by
</p>
<p>D(x, μ) = (h(x)&minus; h(μ))T (ΣW + λΩ)&minus;1(h(x)&minus; h(μ)), (12.58)
</p>
<p>where ΣW is the within-class covariance matrix of the derived vari-
ables h(xi).
</p>
<p>&bull; Decompose the classification subspace using a penalized metric:
</p>
<p>maxuTΣBetu subject to u
T (ΣW + λΩ)u = 1.
</p>
<p>Loosely speaking, the penalized Mahalanobis distance tends to give less
weight to &ldquo;rough&rdquo; coordinates, and more weight to &ldquo;smooth&rdquo; ones; since
the penalty is not diagonal, the same applies to linear combinations that
are rough or smooth.
For some classes of problems, the first step, involving the basis expansion,
</p>
<p>is not needed; we already have far too many (correlated) predictors. A
leading example is when the objects to be classified are digitized analog
signals:
</p>
<p>&bull; the log-periodogram of a fragment of spoken speech, sampled at a set
of 256 frequencies; see Figure 5.5 on page 149.
</p>
<p>&bull; the grayscale pixel values in a digitized image of a handwritten digit.</p>
<p/>
</div>
<div class="page"><p/>
<p>12.6 Penalized Discriminant Analysis 447
</p>
<p>LDA: Coefficient 1 PDA: Coefficient 1 LDA: Coefficient 2 PDA: Coefficient 2 LDA: Coefficient 3 PDA: Coefficient 3
</p>
<p>LDA: Coefficient 4 PDA: Coefficient 4 LDA: Coefficient 5 PDA: Coefficient 5 LDA: Coefficient 6 PDA: Coefficient 6
</p>
<p>LDA: Coefficient 7 PDA: Coefficient 7 LDA: Coefficient 8 PDA: Coefficient 8 LDA: Coefficient 9 PDA: Coefficient 9
</p>
<p>FIGURE 12.11. The images appear in pairs, and represent the nine discrim-
inant coefficient functions for the digit recognition problem. The left member of
each pair is the LDA coefficient, while the right member is the PDA coefficient,
regularized to enforce spatial smoothness.
</p>
<p>It is also intuitively clear in these cases why regularization is needed.
Take the digitized image as an example. Neighboring pixel values will tend
to be correlated, being often almost the same. This implies that the pair
of corresponding LDA coefficients for these pixels can be wildly different
and opposite in sign, and thus cancel when applied to similar pixel values.
Positively correlated predictors lead to noisy, negatively correlated coeffi-
cient estimates, and this noise results in unwanted sampling variance. A
reasonable strategy is to regularize the coefficients to be smooth over the
spatial domain, as with images. This is what PDA does. The computations
proceed just as for FDA, except that an appropriate penalized regression
method is used. Here hT (X)βℓ = Xβℓ, and Ω is chosen so that β
</p>
<p>T
ℓ Ωβℓ
</p>
<p>penalizes roughness in βℓ when viewed as an image. Figure 1.2 on page 4
shows some examples of handwritten digits. Figure 12.11 shows the dis-
criminant variates using LDA and PDA. Those produced by LDA appear
as salt-and-pepper images, while those produced by PDA are smooth im-
ages. The first smooth image can be seen as the coefficients of a linear
contrast functional for separating images with a dark central vertical strip
(ones, possibly sevens) from images that are hollow in the middle (zeros,
some fours). Figure 12.12 supports this interpretation, and with more dif-
ficulty allows an interpretation of the second coordinate. This and other</p>
<p/>
</div>
<div class="page"><p/>
<p>448 12. Flexible Discriminants
</p>
<p>-5 0 5
</p>
<p>-6
-4
</p>
<p>-2
0
</p>
<p>2
4
</p>
<p>6
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0 0
</p>
<p>00
</p>
<p>0
</p>
<p>00
0
0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0 0
</p>
<p>0
0
</p>
<p>0
</p>
<p>0
</p>
<p>0
0
</p>
<p>000
0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0 0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>00
0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
00
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>00
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
00 0
</p>
<p>0
</p>
<p>0
</p>
<p>0 0
</p>
<p>0
0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>00
</p>
<p>0
</p>
<p>0
0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>00
</p>
<p>0
</p>
<p>0
</p>
<p>0
0
</p>
<p>0
</p>
<p>0
</p>
<p>00
</p>
<p>0
</p>
<p>0
0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>00
0
</p>
<p>0
</p>
<p>00
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
0
</p>
<p>00
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
0
0
</p>
<p>0
0
</p>
<p>0
</p>
<p>0
</p>
<p>0 0
</p>
<p>0
</p>
<p>0
</p>
<p>00 0
</p>
<p>0
0
</p>
<p>0
</p>
<p>00
</p>
<p>0
</p>
<p>0
</p>
<p>0
0 0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0 00
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
0
</p>
<p>00 0
</p>
<p>0
</p>
<p>0
</p>
<p>0
0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0 0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0 0
</p>
<p>0
</p>
<p>0
</p>
<p>0
0 0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0 0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>00
0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
0
</p>
<p>0
</p>
<p>0
</p>
<p>0 0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>00
</p>
<p>0
</p>
<p>0
</p>
<p>00
</p>
<p>0
</p>
<p>00
</p>
<p>0
</p>
<p>0
0
</p>
<p>0 0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>00
</p>
<p>0
</p>
<p>00
</p>
<p>0
</p>
<p>0
</p>
<p>0
0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
0
</p>
<p>0
0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
0
</p>
<p>0
0
</p>
<p>0
</p>
<p>0
</p>
<p>0
0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
0
</p>
<p>0
</p>
<p>00
0
</p>
<p>00
</p>
<p>0
</p>
<p>0
0
</p>
<p>0
</p>
<p>0
</p>
<p>0
0
</p>
<p>0
</p>
<p>0
</p>
<p>0
0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
00
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>0
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
111
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>11
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
11
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>11
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>11
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>11
11
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>11
</p>
<p>11
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>11
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>11
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 11
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
11
</p>
<p>11
</p>
<p>1
1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1 1
1 1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
11 11
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>11
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>11 1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>11
</p>
<p>1
1 1
</p>
<p>11
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>22
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>22
2
</p>
<p>2
</p>
<p>2 2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
2 2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2 2 2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>22
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2 2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
2
</p>
<p>2
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3 3
</p>
<p>33
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3 3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>33
</p>
<p>3 3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>333
</p>
<p>3 3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>33
3
</p>
<p>3
3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3 3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
33 3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>4
</p>
<p>4
4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>44
</p>
<p>4
4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
4
</p>
<p>4
</p>
<p>4
</p>
<p>4
4
</p>
<p>4
</p>
<p>4
</p>
<p>4
4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4 4
</p>
<p>4
</p>
<p>4
</p>
<p>4 4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
4
</p>
<p>4
4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4 4
</p>
<p>4
</p>
<p>4
4 4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4 4
</p>
<p>4
</p>
<p>4
</p>
<p>4
4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
4
</p>
<p>4
</p>
<p>4
4
</p>
<p>4
</p>
<p>4
4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
4 4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
4
</p>
<p>4
4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>44 4
4
</p>
<p>4
</p>
<p>4
</p>
<p>4 4
</p>
<p>4
</p>
<p>44 4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
4
</p>
<p>4
</p>
<p>4
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5 5
</p>
<p>55
</p>
<p>5
55
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>55
5
</p>
<p>5
</p>
<p>55
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5 5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
5
</p>
<p>5
5
</p>
<p>55
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5 5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>55
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
5
</p>
<p>5
</p>
<p>5
5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
55
</p>
<p>5
5
5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
5
</p>
<p>5
</p>
<p>5
</p>
<p>5 5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5 5
</p>
<p>5
</p>
<p>5
5
</p>
<p>5
</p>
<p>5
</p>
<p>55
5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
6
</p>
<p>6
</p>
<p>6
</p>
<p>66
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>666
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
6
</p>
<p>6
</p>
<p>6
6
</p>
<p>6
</p>
<p>6
6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>66
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>66 6
</p>
<p>66
</p>
<p>6
</p>
<p>6
6
</p>
<p>6
</p>
<p>6
6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
6
</p>
<p>6
</p>
<p>6
</p>
<p>6
6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6 6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
6
</p>
<p>6 6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>66
</p>
<p>6
</p>
<p>6
</p>
<p>6
6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>6
6
</p>
<p>6
</p>
<p>6
</p>
<p>6
</p>
<p>7
</p>
<p>7
7 7
</p>
<p>7
</p>
<p>77
</p>
<p>7
</p>
<p>7
</p>
<p>7
</p>
<p>7
</p>
<p>7
</p>
<p>7
</p>
<p>7
7
</p>
<p>7 7
</p>
<p>7
</p>
<p>7
</p>
<p>77
</p>
<p>7 7
</p>
<p>7
</p>
<p>7
</p>
<p>7
</p>
<p>7
7
</p>
<p>77
</p>
<p>7
</p>
<p>7
</p>
<p>7
</p>
<p>7
</p>
<p>7
7
</p>
<p>7
</p>
<p>7
</p>
<p>7
</p>
<p>7
</p>
<p>77
7
</p>
<p>7
</p>
<p>7
</p>
<p>7
7
</p>
<p>7
</p>
<p>7
</p>
<p>7
</p>
<p>7 7
</p>
<p>7
</p>
<p>7
</p>
<p>7
</p>
<p>7
</p>
<p>7
</p>
<p>7
</p>
<p>7
7
</p>
<p>7 7
</p>
<p>7
</p>
<p>7
</p>
<p>7
</p>
<p>7
</p>
<p>7
</p>
<p>7 7
</p>
<p>7
</p>
<p>7
</p>
<p>7
</p>
<p>7
</p>
<p>7
</p>
<p>7
</p>
<p>7
</p>
<p>7
</p>
<p>7
</p>
<p>7
7
</p>
<p>7
</p>
<p>7
</p>
<p>7
7
</p>
<p>7
</p>
<p>7
</p>
<p>7
</p>
<p>7
</p>
<p>7
</p>
<p>7
</p>
<p>7
7
</p>
<p>7 7
</p>
<p>7
</p>
<p>77
</p>
<p>7
</p>
<p>7
7
</p>
<p>7
</p>
<p>7
</p>
<p>7
</p>
<p>7
</p>
<p>7
</p>
<p>7
</p>
<p>7
</p>
<p>7
</p>
<p>7
</p>
<p>7
</p>
<p>7
</p>
<p>7
</p>
<p>7
</p>
<p>7
</p>
<p>7
</p>
<p>7 7
7
</p>
<p>7
</p>
<p>7
</p>
<p>7
</p>
<p>7
</p>
<p>7
7
</p>
<p>7
</p>
<p>7
</p>
<p>7
</p>
<p>7
</p>
<p>77
</p>
<p>7
</p>
<p>77
</p>
<p>7
</p>
<p>7
</p>
<p>7
</p>
<p>7
</p>
<p>7 7
</p>
<p>77
7
</p>
<p>7
</p>
<p>7
</p>
<p>7
</p>
<p>7
</p>
<p>7
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8 8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
8
</p>
<p>8
88
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>88
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>88
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8 8
</p>
<p>8
</p>
<p>8
8
</p>
<p>88 8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
8 8
</p>
<p>8
</p>
<p>8 8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
8
</p>
<p>8
8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8 8
</p>
<p>8
</p>
<p>8 8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
8
</p>
<p>8
8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
8
</p>
<p>8
</p>
<p>8
8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
8
</p>
<p>8
</p>
<p>8
8
</p>
<p>8
</p>
<p>8
8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
888
</p>
<p>8
</p>
<p>8 8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>88
88
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>8
</p>
<p>888
</p>
<p>8
</p>
<p>9
</p>
<p>99
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9 9
</p>
<p>9 9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>99
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
9
</p>
<p>9
9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>99
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
9
</p>
<p>9
9
</p>
<p>9
9
</p>
<p>9
</p>
<p>9
</p>
<p>99
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>99
</p>
<p>9
</p>
<p>9
</p>
<p>9
9
</p>
<p>99
</p>
<p>99
</p>
<p>9
9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
9
</p>
<p>9
</p>
<p>9
</p>
<p>9
9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
9
</p>
<p>9
</p>
<p>9
9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
9 9
</p>
<p>9
9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>99
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
9
</p>
<p>99
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9 9
9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
</p>
<p>9
9
</p>
<p>0
</p>
<p>1
</p>
<p>2
</p>
<p>3
</p>
<p>4
</p>
<p>5
</p>
<p>6
</p>
<p>7
</p>
<p>8
</p>
<p>9
</p>
<p>PDA: Discriminant Coordinate 1
</p>
<p>P
D
</p>
<p>A
: 
</p>
<p>D
is
</p>
<p>c
ri
m
</p>
<p>in
a
</p>
<p>n
t 
</p>
<p>C
o
</p>
<p>o
rd
</p>
<p>in
a
</p>
<p>te
 2
</p>
<p>FIGURE 12.12. The first two penalized canonical variates, evaluated for the
test data. The circles indicate the class centroids. The first coordinate contrasts
mainly 0&rsquo;s and 1&rsquo;s, while the second contrasts 6&rsquo;s and 7/9&rsquo;s.</p>
<p/>
</div>
<div class="page"><p/>
<p>12.7 Mixture Discriminant Analysis 449
</p>
<p>examples are discussed in more detail in Hastie et al. (1995), who also show
that the regularization improves the classification performance of LDA on
independent test data by a factor of around 25% in the cases they tried.
</p>
<p>12.7 Mixture Discriminant Analysis
</p>
<p>Linear discriminant analysis can be viewed as a prototype classifier. Each
class is represented by its centroid, and we classify to the closest using an
appropriate metric. In many situations a single prototype is not sufficient
to represent inhomogeneous classes, and mixture models are more appro-
priate. In this section we review Gaussian mixture models and show how
they can be generalized via the FDA and PDA methods discussed earlier.
A Gaussian mixture model for the kth class has density
</p>
<p>P (X|G = k) =
Rk&sum;
</p>
<p>r=1
</p>
<p>πkrφ(X;μkr,Σ), (12.59)
</p>
<p>where the mixing proportions πkr sum to one. This has Rk prototypes for
the kth class, and in our specification, the same covariance matrix Σ is
used as the metric throughout. Given such a model for each class, the class
posterior probabilities are given by
</p>
<p>P (G = k|X = x) =
&sum;Rk
</p>
<p>r=1 πkrφ(X;μkr,Σ)Πk&sum;K
ℓ=1
</p>
<p>&sum;Rℓ
r=1 πℓrφ(X;μℓr,Σ)Πℓ
</p>
<p>, (12.60)
</p>
<p>where Πk represent the class prior probabilities.
We saw these calculations for the special case of two components in
</p>
<p>Chapter 8. As in LDA, we estimate the parameters by maximum likelihood,
using the joint log-likelihood based on P (G,X):
</p>
<p>K&sum;
</p>
<p>k=1
</p>
<p>&sum;
</p>
<p>gi=k
</p>
<p>log
</p>
<p>[
Rk&sum;
</p>
<p>r=1
</p>
<p>πkrφ(xi;μkr,Σ)Πk
</p>
<p>]
. (12.61)
</p>
<p>The sum within the log makes this a rather messy optimization problem
if tackled directly. The classical and natural method for computing the
maximum-likelihood estimates (MLEs) for mixture distributions is the EM
algorithm (Dempster et al., 1977), which is known to possess good conver-
gence properties. EM alternates between the two steps:</p>
<p/>
</div>
<div class="page"><p/>
<p>450 12. Flexible Discriminants
</p>
<p>E-step: Given the current parameters, compute the responsibility of sub-
class ckr within class k for each of the class-k observations (gi = k):
</p>
<p>W (ckr|xi, gi) =
πkrφ(xi;μkr,Σ)&sum;Rk
ℓ=1 πkℓφ(xi;μkℓ,Σ)
</p>
<p>. (12.62)
</p>
<p>M-step: Compute the weighted MLEs for the parameters of each of the
component Gaussians within each of the classes, using the weights
from the E-step.
</p>
<p>In the E-step, the algorithm apportions the unit weight of an observation
in class k to the various subclasses assigned to that class. If it is close to the
centroid of a particular subclass, and far from the others, it will receive a
mass close to one for that subclass. On the other hand, observations halfway
between two subclasses will get approximately equal weight for both.
In the M-step, an observation in class k is used Rk times, to estimate the
</p>
<p>parameters in each of the Rk component densities, with a different weight
for each. The EM algorithm is studied in detail in Chapter 8. The algorithm
requires initialization, which can have an impact, since mixture likelihoods
are generally multimodal. Our software (referenced in the Computational
Considerations on page 455) allows several strategies; here we describe the
default. The user supplies the number Rk of subclasses per class. Within
class k, a k-means clustering model, with multiple random starts, is fitted
to the data. This partitions the observations into Rk disjoint groups, from
which an initial weight matrix, consisting of zeros and ones, is created.
Our assumption of an equal component covariance matrix Σ throughout
</p>
<p>buys an additional simplicity; we can incorporate rank restrictions in the
mixture formulation just like in LDA. To understand this, we review a little-
known fact about LDA. The rank-L LDA fit (Section 4.3.3) is equivalent to
the maximum-likelihood fit of a Gaussian model,where the different mean
vectors in each class are confined to a rank-L subspace of IRp (Exercise 4.8).
We can inherit this property for the mixture model, and maximize the log-
likelihood (12.61) subject to rank constraints on all the
</p>
<p>&sum;
k Rk centroids:
</p>
<p>rank{μkℓ} = L.
Again the EM algorithm is available, and the M-step turns out to be
</p>
<p>a weighted version of LDA, with R =
&sum;K
</p>
<p>k=1 Rk &ldquo;classes.&rdquo; Furthermore,
we can use optimal scoring as before to solve the weighted LDA problem,
which allows us to use a weighted version of FDA or PDA at this stage.
One would expect, in addition to an increase in the number of &ldquo;classes,&rdquo; a
similar increase in the number of &ldquo;observations&rdquo; in the kth class by a factor
of Rk. It turns out that this is not the case if linear operators are used for
the optimal scoring regression. The enlarged indicator Y matrix collapses
in this case to a blurred response matrix Z, which is intuitively pleasing.
For example, suppose there are K = 3 classes, and Rk = 3 subclasses per
class. Then Z might be</p>
<p/>
</div>
<div class="page"><p/>
<p>12.7 Mixture Discriminant Analysis 451
</p>
<p>⎛
⎜⎜⎜⎜⎜⎜⎜⎜⎜⎝
</p>
<p>c11 c12 c13 c21 c22 c23 c31 c32 c33
g1 = 2 0 0 0 0.3 0.5 0.2 0 0 0
g2 = 1 0.9 0.1 0.0 0 0 0 0 0 0
g3 = 1 0.1 0.8 0.1 0 0 0 0 0 0
g4 = 3 0 0 0 0 0 0 0.5 0.4 0.1
g5 = 2 0 0 0 0.7 0.1 0.2 0 0 0
...
</p>
<p>...
gN = 3 0 0 0 0 0 0 0.1 0.1 0.8
</p>
<p>⎞
⎟⎟⎟⎟⎟⎟⎟⎟⎟⎠
</p>
<p>, (12.63)
</p>
<p>where the entries in a class-k row correspond to W (ckr|x, gi).
The remaining steps are the same:
</p>
<p>Ẑ = SZ
</p>
<p>ZT Ẑ = ΘDΘT
</p>
<p>Update πs and Πs
</p>
<p>⎫
⎬
⎭ M-step of MDA.
</p>
<p>These simple modifications add considerable flexibility to the mixture
model:
</p>
<p>&bull; The dimension reduction step in LDA, FDA or PDA is limited by
the number of classes; in particular, for K = 2 classes no reduction is
possible. MDA substitutes subclasses for classes, and then allows us
to look at low-dimensional views of the subspace spanned by these
subclass centroids. This subspace will often be an important one for
discrimination.
</p>
<p>&bull; By using FDA or PDA in the M-step, we can adapt even more to par-
ticular situations. For example, we can fit MDA models to digitized
analog signals and images, with smoothness constraints built in.
</p>
<p>Figure 12.13 compares FDA and MDA on the mixture example.
</p>
<p>12.7.1 Example: Waveform Data
</p>
<p>We now illustrate some of these ideas on a popular simulated example,
taken from Breiman et al. (1984, pages 49&ndash;55), and used in Hastie and
Tibshirani (1996b) and elsewhere. It is a three-class problem with 21 vari-
ables, and is considered to be a difficult pattern recognition problem. The
predictors are defined by
</p>
<p>Xj = Uh1(j) + (1&minus; U)h2(j) + ǫj Class 1,
Xj = Uh1(j) + (1&minus; U)h3(j) + ǫj Class 2, (12.64)
Xj = Uh2(j) + (1&minus; U)h3(j) + ǫj Class 3,
</p>
<p>where j = 1, 2, . . . , 21, U is uniform on (0, 1), ǫj are standard normal vari-
ates, and the hℓ are the shifted triangular waveforms: h1(j) = max(6 &minus;</p>
<p/>
</div>
<div class="page"><p/>
<p>452 12. Flexible Discriminants
</p>
<p>FDA / MARS - Degree 2
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . .
. . . . . . . . . . . . . .. . . . . . . . . . . . .
. . . . . . . . . . . . .. . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . .. . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . .. . . . . . . . . . . . .
. . . . . . . . . . . . . .. . . . . . . . . . . . . .
. . . . . . . . . . . . . .. . . . . . . . . . . . . .
. . . . . . . . . . . . . .. . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . .. . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . .. . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . .. . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . .
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>Training Error: 0.185
</p>
<p>Test Error:       0.235
</p>
<p>Bayes Error:    0.210
</p>
<p>MDA - 5 Subclasses per Class
</p>
<p>..
</p>
<p>.. .
</p>
<p>. .. . .
</p>
<p>. . .. . .
</p>
<p>. . . . .. . . . .
</p>
<p>. . . . . .. . . . . . . .
</p>
<p>. . . . . . . .. . . . . . . . . .
</p>
<p>. . . . . . . . . .. . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . .. . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . .. . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . .
. . . . . . . . . . . . . .. . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . .. . . . . . . . . . . .
. . . . . . . . . . . . .. . . . . . . . . . . . .
. . . . . . . . . . . . .. . . . . . . . . . . . .
. . . . . . . . . . . . .. . . . . . . . . . . . .
. . . . . . . . . . . . .. . . . . . . . . . . . .
. . . . . . . . . . . . . .. . . . . . . . . . . . . .
. . . . . . . . . . . . . .. . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . .. . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . .. . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . .. . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . .. . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . .. . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . .. . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . .. . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . .. . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . .. . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . .. . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . .. . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . .
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>Training Error: 0.17
</p>
<p>Test Error:       0.22
</p>
<p>Bayes Error:    0.21
</p>
<p>FIGURE 12.13. FDA and MDA on the mixture data. The upper plot uses
FDA with MARS as the regression procedure. The lower plot uses MDA with
five mixture centers per class (indicated). The MDA solution is close to Bayes
optimal, as might be expected given the data arise from mixtures of Gaussians.
The broken purple curve in the background is the Bayes decision boundary.</p>
<p/>
</div>
<div class="page"><p/>
<p>12.7 Mixture Discriminant Analysis 453
</p>
<p>1 1 1 1 1 1 1 1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>12 2 2 2 2 2
2 2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>23 3 3 3 3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
3
</p>
<p>34 4 4 4 4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
4
</p>
<p>4
4
</p>
<p>45 5 5 5 5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5 5 5 5 5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>Class 1
</p>
<p>1
1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1 1 1 12 2
2 2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2 2 2 2 23 3 3 3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3 3 3 3 34
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
4
</p>
<p>4
4
</p>
<p>4 4 4 4 45
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
5
</p>
<p>5
5
</p>
<p>5
</p>
<p>5
</p>
<p>5
5
</p>
<p>5
5
</p>
<p>5 5 5 5 5
</p>
<p>Class 2
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
1
</p>
<p>1
1
</p>
<p>1
1
</p>
<p>12
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
2
</p>
<p>2
2
</p>
<p>2
2
</p>
<p>23
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3 3 3 3 3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>34
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
</p>
<p>4
4
</p>
<p>4
4
</p>
<p>4
4
</p>
<p>4
4
</p>
<p>45
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5
</p>
<p>5 5
5 5 5 5 5 5 5
</p>
<p>Class 3
</p>
<p>FIGURE 12.14. Some examples of the waveforms generated from model (12.64)
before the Gaussian noise is added.
</p>
<p>|j &minus; 11|, 0), h2(j) = h1(j &minus; 4) and h3(j) = h1(j + 4). Figure 12.14 shows
some example waveforms from each class.
Table 12.4 shows the results of MDA applied to the waveform data, as
</p>
<p>well as several other methods from this and other chapters. Each train-
ing sample has 300 observations, and equal priors were used, so there are
roughly 100 observations in each class. We used test samples of size 500.
The two MDA models are described in the caption.
Figure 12.15 shows the leading canonical variates for the penalized MDA
</p>
<p>model, evaluated at the test data. As we might have guessed, the classes
appear to lie on the edges of a triangle. This is because the hj(i) are repre-
sented by three points in 21-space, thereby forming vertices of a triangle,
and each class is represented as a convex combination of a pair of vertices,
and hence lie on an edge. Also it is clear visually that all the information
lies in the first two dimensions; the percentage of variance explained by the
first two coordinates is 99.8%, and we would lose nothing by truncating the
solution there. The Bayes risk for this problem has been estimated to be
about 0.14 (Breiman et al., 1984). MDA comes close to the optimal rate,
which is not surprising since the structure of the MDA model is similar to
the generating model.</p>
<p/>
</div>
<div class="page"><p/>
<p>454 12. Flexible Discriminants
</p>
<p>TABLE 12.4. Results for waveform data. The values are averages over ten sim-
ulations, with the standard error of the average in parentheses. The five entries
above the line are taken from Hastie et al. (1994). The first model below the line
is MDA with three subclasses per class. The next line is the same, except that the
discriminant coefficients are penalized via a roughness penalty to effectively 4df.
The third is the corresponding penalized LDA or PDA model.
</p>
<p>Technique Error Rates
Training Test
</p>
<p>LDA 0.121(0.006) 0.191(0.006)
QDA 0.039(0.004) 0.205(0.006)
CART 0.072(0.003) 0.289(0.004)
FDA/MARS (degree = 1) 0.100(0.006) 0.191(0.006)
FDA/MARS (degree = 2) 0.068(0.004) 0.215(0.002)
</p>
<p>MDA (3 subclasses) 0.087(0.005) 0.169(0.006)
MDA (3 subclasses, penalized 4 df) 0.137(0.006) 0.157(0.005)
PDA (penalized 4 df) 0.150(0.005) 0.171(0.005)
Bayes 0.140
</p>
<p>Discriminant Var 1
</p>
<p>D
is
</p>
<p>c
ri
m
</p>
<p>in
a
</p>
<p>n
t 
</p>
<p>V
a
</p>
<p>r 
2
</p>
<p>-6 -4 -2 0 2 4 6
</p>
<p>-6
-4
</p>
<p>-2
0
</p>
<p>2
4
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>11
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>11 1
</p>
<p>1
</p>
<p>11
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>11
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>11
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2 2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
2
2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
22
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
3
</p>
<p>3
</p>
<p>3 3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>33
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
3
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>33
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3 3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3 3
</p>
<p>3 3
3 3
</p>
<p>3
</p>
<p>33
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>33 3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3 3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3 Subclasses, Penalized 4 df
</p>
<p>Discriminant Var 3
</p>
<p>D
is
</p>
<p>c
ri
m
</p>
<p>in
a
</p>
<p>n
t 
</p>
<p>V
a
</p>
<p>r 
4
</p>
<p>-2 -1 0 1 2
</p>
<p>-1
.0
</p>
<p>0
.0
</p>
<p>0
.5
</p>
<p>1
.0
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>11
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>11
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>11
</p>
<p>1
</p>
<p>1
1 1
</p>
<p>1 1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>11
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1 1
</p>
<p>1
</p>
<p>11
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>1
</p>
<p>111
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2 2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>22
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>22
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>22
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2 2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2 2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
</p>
<p>2
22
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3 3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>33
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>33
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>33
3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
</p>
<p>3
3
</p>
<p>3
3
</p>
<p>3
</p>
<p>3
</p>
<p>3
33
</p>
<p>3 Subclasses, Penalized 4 df
</p>
<p>FIGURE 12.15. Some two-dimensional views of the MDA model fitted to a
sample of the waveform model. The points are independent test data, projected
on to the leading two canonical coordinates (left panel), and the third and fourth
(right panel). The subclass centers are indicated.</p>
<p/>
</div>
<div class="page"><p/>
<p>Exercises 455
</p>
<p>Computational Considerations
</p>
<p>With N training cases, p predictors, and m support vectors, the support
vector machine requires m3 + mN + mpN operations, assuming m &asymp; N .
They do not scale well with N , although computational shortcuts are avail-
able (Platt, 1999). Since these are evolving rapidly, the reader is urged to
search the web for the latest technology.
LDA requires Np2 + p3 operations, as does PDA. The complexity of
</p>
<p>FDA depends on the regression method used. Many techniques are linear
in N , such as additive models and MARS. General splines and kernel-based
regression methods will typically require N3 operations.
</p>
<p>Software is available for fitting FDA, PDA and MDA models in the R
package mda, which is also available in S-PLUS.
</p>
<p>Bibliographic Notes
</p>
<p>The theory behind support vector machines is due to Vapnik and is de-
scribed in Vapnik (1996). There is a burgeoning literature on SVMs; an
online bibliography, created and maintained by Alex Smola and Bernhard
Schölkopf, can be found at:
</p>
<p>http://www.kernel-machines.org.
Our treatment is based on Wahba et al. (2000) and Evgeniou et al. (2000),
and the tutorial by Burges (Burges, 1998).
Linear discriminant analysis is due to Fisher (1936) and Rao (1973). The
</p>
<p>connection with optimal scoring dates back at least to Breiman and Ihaka
(1984), and in a simple form to Fisher (1936). There are strong connections
with correspondence analysis (Greenacre, 1984). The description of flexible,
penalized and mixture discriminant analysis is taken from Hastie et al.
(1994), Hastie et al. (1995) and Hastie and Tibshirani (1996b), and all
three are summarized in Hastie et al. (2000); see also Ripley (1996).
</p>
<p>Exercises
</p>
<p>Ex. 12.1 Show that the criteria (12.25) and (12.8) are equivalent.
</p>
<p>Ex. 12.2 Show that the solution to (12.29) is the same as the solution to
(12.25) for a particular kernel.
</p>
<p>Ex. 12.3 Consider a modification to (12.43) where you do not penalize the
constant. Formulate the problem, and characterize its solution.
</p>
<p>Ex. 12.4 Suppose you perform a reduced-subspace linear discriminant anal-
ysis for a K-group problem. You compute the canonical variables of di-</p>
<p/>
</div>
<div class="page"><p/>
<p>456 12. Flexible Discriminants
</p>
<p>mension L &le; K &minus; 1 given by z = UTx, where U is the p &times; L matrix of
discriminant coefficients, and p &gt; K is the dimension of x.
</p>
<p>(a) If L = K &minus; 1 show that
</p>
<p>‖z &minus; z̄k‖2 &minus; ‖z &minus; z̄k&prime;‖2 = ‖x&minus; x̄k‖2W &minus; ‖x&minus; x̄k&prime;‖
2
W ,
</p>
<p>where ‖&middot;‖W denotes Mahalanobis distance with respect to the covari-
ance W.
</p>
<p>(b) If L &lt; K &minus; 1, show that the same expression on the left measures
the difference in Mahalanobis squared distances for the distributions
projected onto the subspace spanned by U.
</p>
<p>Ex. 12.5 The data in phoneme.subset, available from this book&rsquo;s website
</p>
<p>http://www-stat.stanford.edu/ElemStatLearn
</p>
<p>consists of digitized log-periodograms for phonemes uttered by 60 speakers,
each speaker having produced phonemes from each of five classes. It is
appropriate to plot each vector of 256 &ldquo;features&rdquo; against the frequencies
0&ndash;255.
</p>
<p>(a) Produce a separate plot of all the phoneme curves against frequency
for each class.
</p>
<p>(b) You plan to use a nearest prototype classification scheme to classify
the curves into phoneme classes. In particular, you will use aK-means
clustering algorithm in each class (kmeans() in R), and then classify
observations to the class of the closest cluster center. The curves are
high-dimensional and you have a rather small sample-size-to-variables
ratio. You decide to restrict all the prototypes to be smooth functions
of frequency. In particular, you decide to represent each prototype m
as m = Bθ where B is a 256 &times; J matrix of natural spline basis
functions with J knots uniformly chosen in (0, 255) and boundary
knots at 0 and 255. Describe how to proceed analytically, and in
particular, how to avoid costly high-dimensional fitting procedures.
(Hint: It may help to restrict B to be orthogonal.)
</p>
<p>(c) Implement your procedure on the phoneme data, and try it out. Divide
the data into a training set and a test set (50-50), making sure that
speakers are not split across sets (why?). Use K = 1, 3, 5, 7 centers
per class, and for each use J = 5, 10, 15 knots (taking care to start
the K-means procedure at the same starting values for each value of
J), and compare the results.
</p>
<p>Ex. 12.6 Suppose that the regression procedure used in FDA (Section 12.5.1)
is a linear expansion of basis functions hm(x), m = 1, . . . ,M . Let Dπ =
YTY/N be the diagonal matrix of class proportions.</p>
<p/>
</div>
<div class="page"><p/>
<p>Exercises 457
</p>
<p>(a) Show that the optimal scoring problem (12.52) can be written in vector
notation as
</p>
<p>min
θ,β
</p>
<p>‖Yθ &minus;Hβ‖2 , (12.65)
</p>
<p>where θ is a vector of K real numbers, and H is the N &times;M matrix
of evaluations hj(xi).
</p>
<p>(b) Suppose that the normalization on θ is θTDπ1 = 0 and θ
TDπθ = 1.
</p>
<p>Interpret these normalizations in terms of the original scored θ(gi).
</p>
<p>(c) Show that, with this normalization, (12.65) can be partially optimized
w.r.t. β, and leads to
</p>
<p>max
θ
</p>
<p>θTYTSYθ, (12.66)
</p>
<p>subject to the normalization constraints, where S is the projection
operator corresponding to the basis matrix H.
</p>
<p>(d) Suppose that the hj include the constant function. Show that the
largest eigenvalue of S is 1.
</p>
<p>(e) Let Θ be a K &times; K matrix of scores (in columns), and suppose the
normalization is ΘTDπΘ = I. Show that the solution to (12.53) is
given by the complete set of eigenvectors of S; the first eigenvector is
trivial, and takes care of the centering of the scores. The remainder
characterize the optimal scoring solution.
</p>
<p>Ex. 12.7 Derive the solution to the penalized optimal scoring problem
(12.57).
</p>
<p>Ex. 12.8 Show that coefficients βℓ found by optimal scoring are proportional
to the discriminant directions νℓ found by linear discriminant analysis.
</p>
<p>Ex. 12.9 Let Ŷ = XB̂ be the fitted N &times;K indicator response matrix after
linear regression on the N&times;pmatrixX, where p &gt; K. Consider the reduced
features x&lowast;i = B̂
</p>
<p>Txi. Show that LDA using x
&lowast;
i is equivalent to LDA in the
</p>
<p>original space.
</p>
<p>Ex. 12.10 Kernels and linear discriminant analysis. Suppose you wish to
carry out a linear discriminant analysis (two classes) using a vector of
transformations of the input variables h(x). Since h(x) is high-dimensional,
you will use a regularized within-class covariance matrix Wh + γI. Show
that the model can be estimated using only the inner products K(xi, xi&prime;) =
〈h(xi), h(xi&prime;)〉. Hence the kernel property of support vector machines is also
shared by regularized linear discriminant analysis.
</p>
<p>Ex. 12.11 The MDA procedure models each class as a mixture of Gaussians.
Hence each mixture center belongs to one and only one class. A more
general model allows each mixture center to be shared by all classes. We
take the joint density of labels and features to be</p>
<p/>
</div>
<div class="page"><p/>
<p>458 12. Flexible Discriminants
</p>
<p>P (G,X) =
</p>
<p>R&sum;
</p>
<p>r=1
</p>
<p>πrPr(G,X), (12.67)
</p>
<p>a mixture of joint densities. Furthermore we assume
</p>
<p>Pr(G,X) = Pr(G)φ(X;μr,Σ). (12.68)
</p>
<p>This model consists of regions centered at μr, and for each there is a class
profile Pr(G). The posterior class distribution is given by
</p>
<p>P (G = k|X = x) =
&sum;R
</p>
<p>r=1 πrPr(G = k)φ(x;μr,Σ)&sum;R
r=1 πrφ(x;μr,Σ)
</p>
<p>, (12.69)
</p>
<p>where the denominator is the marginal distribution P (X).
</p>
<p>(a) Show that this model (called MDA2) can be viewed as a generalization
of MDA since
</p>
<p>P (X|G = k) =
&sum;R
</p>
<p>r=1 πrPr(G = k)φ(x;μr,Σ)&sum;R
r=1 πrPr(G = k)
</p>
<p>, (12.70)
</p>
<p>where πrk = πrPr(G = k)/
&sum;R
</p>
<p>r=1 πrPr(G = k) corresponds to the
mixing proportions for the kth class.
</p>
<p>(b) Derive the EM algorithm for MDA2.
</p>
<p>(c) Show that if the initial weight matrix is constructed as in MDA, in-
volving separate k-means clustering in each class, then the algorithm
for MDA2 is identical to the original MDA procedure.</p>
<p/>
</div>
<div class="page"><p/>
<p>13
Prototype Methods and
Nearest-Neighbors
</p>
<p>13.1 Introduction
</p>
<p>In this chapter we discuss some simple and essentially model-free methods
for classification and pattern recognition. Because they are highly unstruc-
tured, they typically are not useful for understanding the nature of the
relationship between the features and class outcome. However, as black box
prediction engines, they can be very effective, and are often among the best
performers in real data problems. The nearest-neighbor technique can also
be used in regression; this was touched on in Chapter 2 and works reason-
ably well for low-dimensional problems. However, with high-dimensional
features, the bias&ndash;variance tradeoff does not work as favorably for nearest-
neighbor regression as it does for classification.
</p>
<p>13.2 Prototype Methods
</p>
<p>Throughout this chapter, our training data consists of the N pairs (x1, g1),
. . . , (xn, gN ) where gi is a class label taking values in {1, 2, . . . ,K}. Pro-
totype methods represent the training data by a set of points in feature
space. These prototypes are typically not examples from the training sam-
ple, except in the case of 1-nearest-neighbor classification discussed later.
Each prototype has an associated class label, and classification of a query
</p>
<p>point x is made to the class of the closest prototype. &ldquo;Closest&rdquo; is usually
defined by Euclidean distance in the feature space, after each feature has
</p>
<p> &copy; Springer Science+Business Media, LLC 2009 
</p>
<p>T. Hastie et al., The Elements of Statistical Learning, Second Edition, 459
</p>
<p>DOI: 10.1007/b94608_13,</p>
<p/>
</div>
<div class="page"><p/>
<p>460 13. Prototypes and Nearest-Neighbors
</p>
<p>been standardized to have overall mean 0 and variance 1 in the training
sample. Euclidean distance is appropriate for quantitative features. We
discuss distance measures between qualitative and other kinds of feature
values in Chapter 14.
These methods can be very effective if the prototypes are well positioned
</p>
<p>to capture the distribution of each class. Irregular class boundaries can be
represented, with enough prototypes in the right places in feature space.
The main challenge is to figure out how many prototypes to use and where
to put them. Methods differ according to the number and way in which
prototypes are selected.
</p>
<p>13.2.1 K-means Clustering
</p>
<p>K-means clustering is a method for finding clusters and cluster centers in a
set of unlabeled data. One chooses the desired number of cluster centers, say
R, and the K-means procedure iteratively moves the centers to minimize
the total within cluster variance.1 Given an initial set of centers, the K-
means algorithm alternates the two steps:
</p>
<p>&bull; for each center we identify the subset of training points (its cluster)
that is closer to it than any other center;
</p>
<p>&bull; the means of each feature for the data points in each cluster are
computed, and this mean vector becomes the new center for that
cluster.
</p>
<p>These two steps are iterated until convergence. Typically the initial centers
are R randomly chosen observations from the training data. Details of the
K-means procedure, as well as generalizations allowing for different variable
types and more general distance measures, are given in Chapter 14.
To use K-means clustering for classification of labeled data, the steps
</p>
<p>are:
</p>
<p>&bull; apply K-means clustering to the training data in each class sepa-
rately, using R prototypes per class;
</p>
<p>&bull; assign a class label to each of the K &times;R prototypes;
</p>
<p>&bull; classify a new feature x to the class of the closest prototype.
</p>
<p>Figure 13.1 (upper panel) shows a simulated example with three classes
and two features. We used R = 5 prototypes per class, and show the clas-
sification regions and the decision boundary. Notice that a number of the
</p>
<p>1The &ldquo;K&rdquo; in K-means refers to the number of cluster centers. Since we have already
</p>
<p>reserved K to denote the number of classes, we denote the number of clusters by R.</p>
<p/>
</div>
<div class="page"><p/>
<p>13.2 Prototype Methods 461
</p>
<p>K-means - 5 Prototypes  per Class
</p>
<p>.........
.............................
</p>
<p>...........................................
.......................................................
</p>
<p>........................................................
</p>
<p>........................................................
................................................
</p>
<p>........................................
................................
</p>
<p>........................
......................
</p>
<p>... ........................ .......................
</p>
<p>......... ..................................... .........................
.............................................................................
....................................................................... .
................................. ....................................... ..........
</p>
<p>............................... ............................................ ...................
........................... .................................................. ............................
........................... ........................................................... .....................................
</p>
<p>......................... ................................................................ ..............................................
................. ......................................................... ............................................
........ ............................................... ..........................................
</p>
<p>...............................................................................
...........................................................................
</p>
<p>.....................................................................
.................................................................
</p>
<p>...........................................................
.......................................................
</p>
<p>.................................................
.............................................
</p>
<p>.......................................
...................................
</p>
<p>...............
</p>
<p>....................................................................................................................................................................................................................................
</p>
<p>....................................................................................................................................................................................................................................
</p>
<p>....................................................................................................................................................................................................................................
</p>
<p>....................................................................................................................................................................................................................................
</p>
<p>....................................................................................................................................................................................................................................
</p>
<p>................................................................................................................................................................................ ..................................................
</p>
<p>........................................................... ........................................................................................................ ..............................................
</p>
<p>..................................................... .............................................................................................. ............................................
</p>
<p>............................................... ........................................................................................ ............................................
</p>
<p>......................................... ...................................................................................... ............................................
</p>
<p>.......................................... ...................................................................................... ............................................
</p>
<p>.......................................... ........................................................................................ .............................................
</p>
<p>.............................................. ............................................................................................. .............................................
</p>
<p>.................................................. ................................................................................................. .............................................
</p>
<p>...................................................... ..................................................................................................... .............................................
</p>
<p>.......................................................... ............................................................................ ........................ .............................................
</p>
<p>.......................... ..................... ............................................................... ................. .......................................
</p>
<p>................ ............. .............................................. ........
</p>
<p>...... ....
</p>
<p>.................
.....................................
</p>
<p>.......................... .................................................................. .....................................
................................... ........................................................................... ........................................
</p>
<p>.................................... .............................................................................. ..........................................
</p>
<p>.................................... .......................................................................... ....................................
</p>
<p>................................... .................................................................... ...............................
</p>
<p>.................................... ................................................................ .......................
</p>
<p>..................................... ....................................................... .............
</p>
<p>....................................... ............................................... .......
</p>
<p>........................................ ..................................................... .................
</p>
<p>......................................... ................................................................ ..........................
</p>
<p>.....................................................................................................................................................
</p>
<p>.........................................................................................................................................................
</p>
<p>...............................................................................................................................................................
</p>
<p>...................................................................................................................................................................
</p>
<p>.........................................................................................................................................................................
</p>
<p>.............................................................................................................................................................................
</p>
<p>...................................................................................................................................................................................
</p>
<p>.......................................................................................................................................................................................
</p>
<p>.............................................................................................................................................................................................
</p>
<p>.................................................................................................................................................................................................
</p>
<p>...................................................................................................
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>&bull;
&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull; &bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>LVQ - 5 Prototypes per Class
</p>
<p>.......
</p>
<p>............
...................
</p>
<p>..........................
.................................
.......................................
</p>
<p>..............................................
....................................................
</p>
<p>.........................................................
...............................................................................
...........................................................................
.........................................................................
...................................................................
................................ ................................... .........
</p>
<p>............................... .............................................. ...................
</p>
<p>............................... ...................................................... .............................
.............................. .............................................................. .......................................
</p>
<p>..................... .......................................................... ...........................................
........ ............................................ .........................................
</p>
<p>...............................................................................
...........................................................................
</p>
<p>.......................................................................
...................................................................
</p>
<p>...............................................................
...........................................................
</p>
<p>........................................................
.....................................................
</p>
<p>.................................................
.............................................
</p>
<p>.....................
</p>
<p>....................................................................................................................................................................................................................................
</p>
<p>....................................................................................................................................................................................................................................
</p>
<p>....................................................................................................................................................................................................................................
</p>
<p>....................................................................................................................................................................................................................................
</p>
<p>....................................................................................................................................................................................................................................
</p>
<p>....................................................................................................................................................................................................................................
</p>
<p>....................................................................................................................................................................................................................................
</p>
<p>....................................................................................................................................................................................................................................
</p>
<p>......................................................................................................................................................................... ........................................................
</p>
<p>....................................................... .............................................................................................................. ......................................................
</p>
<p>...................................................... ........................................................................................................... ...................................................
</p>
<p>...................................................... ....................................................................................................... .................................................
</p>
<p>..................................................... ................................................................................................... ..............................................
</p>
<p>.................................................... ................................................................................................. ............................................
</p>
<p>................................................... ............................................................................................. .........................................
</p>
<p>.................................................. .......................................................................................... .......................................
</p>
<p>...................... .................... .................................................... ................
</p>
<p>........ ..............
</p>
<p>.......................... ....................................
............................ ........................................................................ ......................................
</p>
<p>.................................... ........................................................................... ........................................
</p>
<p>.................................... ............................................................................. .........................................
</p>
<p>.................................... ............................................................................... ...........................................
</p>
<p>..................................... ................................................................................. .........................................
</p>
<p>..................................... ......................................................................... .................................
</p>
<p>................................... ................................................................ .......................
</p>
<p>..................................... ....................................................... .............
</p>
<p>...................................... ................................................ ..........
</p>
<p>........................................ ......................................................... ........................
</p>
<p>......................................... ........................................................................................................
</p>
<p>.......................................................................................................................................................
</p>
<p>...........................................................................................................................................................
</p>
<p>...............................................................................................................................................................
</p>
<p>...................................................................................................................................................................
</p>
<p>.......................................................................................................................................................................
</p>
<p>...........................................................................................................................................................................
</p>
<p>.............................................................................................................................................................................
</p>
<p>.................................................................................................................................................................................
</p>
<p>.....................................................................................................................................................................................
</p>
<p>.........................................................................................................................................................................................
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull; &bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>FIGURE 13.1. Simulated example with three classes and five prototypes per
class. The data in each class are generated from a mixture of Gaussians. In the
upper panel, the prototypes were found by applying the K-means clustering algo-
rithm separately in each class. In the lower panel, the LVQ algorithm (starting
from the K-means solution) moves the prototypes away from the decision bound-
ary. The broken purple curve in the background is the Bayes decision boundary.</p>
<p/>
</div>
<div class="page"><p/>
<p>462 13. Prototypes and Nearest-Neighbors
</p>
<p>Algorithm 13.1 Learning Vector Quantization&mdash;LVQ.
</p>
<p>1. Choose R initial prototypes for each class: m1(k),m2(k), . . . ,mR(k),
k = 1, 2, . . . ,K, for example, by samplingR training points at random
from each class.
</p>
<p>2. Sample a training point xi randomly (with replacement), and let (j, k)
index the closest prototype mj(k) to xi.
</p>
<p>(a) If gi = k (i.e., they are in the same class), move the prototype
towards the training point:
</p>
<p>mj(k) &larr; mj(k) + ǫ(xi &minus;mj(k)),
</p>
<p>where ǫ is the learning rate.
</p>
<p>(b) If gi �= k (i.e., they are in different classes), move the prototype
away from the training point:
</p>
<p>mj(k) &larr; mj(k)&minus; ǫ(xi &minus;mj(k)).
</p>
<p>3. Repeat step 2, decreasing the learning rate ǫ with each iteration to-
wards zero.
</p>
<p>prototypes are near the class boundaries, leading to potential misclassifica-
tion errors for points near these boundaries. This results from an obvious
shortcoming with this method: for each class, the other classes do not have
a say in the positioning of the prototypes for that class. A better approach,
discussed next, uses all of the data to position all prototypes.
</p>
<p>13.2.2 Learning Vector Quantization
</p>
<p>In this technique due to Kohonen (1989), prototypes are placed strategically
with respect to the decision boundaries in an ad-hoc way. LVQ is an online
algorithm&mdash;observations are processed one at a time.
The idea is that the training points attract prototypes of the correct class,
</p>
<p>and repel other prototypes. When the iterations settle down, prototypes
should be close to the training points in their class. The learning rate ǫ is
decreased to zero with each iteration, following the guidelines for stochastic
approximation learning rates (Section 11.4.)
Figure 13.1 (lower panel) shows the result of LVQ, using the K-means
</p>
<p>solution as starting values. The prototypes have tended to move away from
the decision boundaries, and away from prototypes of competing classes.
The procedure just described is actually called LVQ1. Modifications
</p>
<p>(LVQ2, LVQ3, etc.) have been proposed, that can sometimes improve per-
formance. A drawback of learning vector quantization methods is the fact</p>
<p/>
</div>
<div class="page"><p/>
<p>13.3 k-Nearest-Neighbor Classifiers 463
</p>
<p>that they are defined by algorithms, rather than optimization of some fixed
criteria; this makes it difficult to understand their properties.
</p>
<p>13.2.3 Gaussian Mixtures
</p>
<p>The Gaussian mixture model can also be thought of as a prototype method,
similar in spirit to K-means and LVQ. We discuss Gaussian mixtures in
some detail in Sections 6.8, 8.5 and 12.7. Each cluster is described in terms
of a Gaussian density, which has a centroid (as in K-means), and a covari-
ance matrix. The comparison becomes crisper if we restrict the component
Gaussians to have a scalar covariance matrix (Exercise 13.1). The two steps
of the alternating EM algorithm are very similar to the two steps in K-
means:
</p>
<p>&bull; In the E-step, each observation is assigned a responsibility or weight
for each cluster, based on the likelihood of each of the correspond-
ing Gaussians. Observations close to the center of a cluster will most
likely get weight 1 for that cluster, and weight 0 for every other clus-
ter. Observations half-way between two clusters divide their weight
accordingly.
</p>
<p>&bull; In the M-step, each observation contributes to the weighted means
(and covariances) for every cluster.
</p>
<p>As a consequence, the Gaussian mixture model is often referred to as a soft
clustering method, while K-means is hard.
</p>
<p>Similarly, when Gaussian mixture models are used to represent the fea-
ture density in each class, it produces smooth posterior probabilities p̂(x) =
{p̂1(x), . . . , p̂K(x)} for classifying x (see (12.60) on page 449.) Often this
is interpreted as a soft classification, while in fact the classification rule is
Ĝ(x) = argmaxk p̂k(x). Figure 13.2 compares the results of K-means and
Gaussian mixtures on the simulated mixture problem of Chapter 2. We
see that although the decision boundaries are roughly similar, those for the
mixture model are smoother (although the prototypes are in approximately
the same positions.) We also see that while both procedures devote a blue
prototype (incorrectly) to a region in the northwest, the Gaussian mixture
classifier can ultimately ignore this region, while K-means cannot. LVQ
gave very similar results to K-means on this example, and is not shown.
</p>
<p>13.3 k-Nearest-Neighbor Classifiers
</p>
<p>These classifiers are memory-based, and require no model to be fit. Given
a query point x0, we find the k training points x(r), r = 1, . . . , k closest in
distance to x0, and then classify using majority vote among the k neighbors.</p>
<p/>
</div>
<div class="page"><p/>
<p>464 13. Prototypes and Nearest-Neighbors
</p>
<p>K-means - 5 Prototypes per Class
</p>
<p>..
. .. . . .
</p>
<p>. . . .. . . . .
</p>
<p>. . . . . . .. . . . . . .
</p>
<p>. . . . . . . . .. . . . . . . . . .
</p>
<p>. . . . . . . . . . . .. . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . .. . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . .. . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . .. . . . . . . . . . . .
. . . . . . . . . . . .. . . . . . . . . . .
</p>
<p>. . . . . . . . . . .. . . . . . . . . . .
</p>
<p>. . . . . . . . . . .. . . . . . . . . . .
</p>
<p>. . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . .. . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . .. . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . .
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>Training Error: 0.170
</p>
<p>Test Error:       0.243
</p>
<p>Bayes Error:    0.210
</p>
<p>Gaussian Mixtures - 5 Subclasses per Class
</p>
<p>..
</p>
<p>.. .
</p>
<p>. .. . .
</p>
<p>. . .. . .
</p>
<p>. . . . .. . . . .
</p>
<p>. . . . . .. . . . . . . .
</p>
<p>. . . . . . . .. . . . . . . . . .
</p>
<p>. . . . . . . . . .. . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . .. . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . .. . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . .
. . . . . . . . . . . . . .. . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . .. . . . . . . . . . . .
. . . . . . . . . . . . .. . . . . . . . . . . . .
. . . . . . . . . . . . .. . . . . . . . . . . . .
. . . . . . . . . . . . .. . . . . . . . . . . . .
. . . . . . . . . . . . .. . . . . . . . . . . . .
. . . . . . . . . . . . . .. . . . . . . . . . . . . .
. . . . . . . . . . . . . .. . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . .. . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . .. . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . .. . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . .. . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . .. . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . .. . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . .. . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . .. . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . .. . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . .. . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . .. . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . .
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>Training Error: 0.17
</p>
<p>Test Error:       0.22
</p>
<p>Bayes Error:    0.21
</p>
<p>FIGURE 13.2. The upper panel shows the K-means classifier applied to the
mixture data example. The decision boundary is piecewise linear. The lower panel
shows a Gaussian mixture model with a common covariance for all component
Gaussians. The EM algorithm for the mixture model was started at the K-means
solution. The broken purple curve in the background is the Bayes decision
boundary.</p>
<p/>
</div>
<div class="page"><p/>
<p>13.3 k-Nearest-Neighbor Classifiers 465
</p>
<p>Ties are broken at random. For simplicity we will assume that the features
are real-valued, and we use Euclidean distance in feature space:
</p>
<p>d(i) = ||x(i) &minus; x0||. (13.1)
Typically we first standardize each of the features to have mean zero and
variance 1, since it is possible that they are measured in different units. In
Chapter 14 we discuss distance measures appropriate for qualitative and
ordinal features, and how to combine them for mixed data. Adaptively
chosen distance metrics are discussed later in this chapter.
Despite its simplicity, k-nearest-neighbors has been successful in a large
</p>
<p>number of classification problems, including handwritten digits, satellite
image scenes and EKG patterns. It is often successful where each class
has many possible prototypes, and the decision boundary is very irregular.
Figure 13.3 (upper panel) shows the decision boundary of a 15-nearest-
neighbor classifier applied to the three-class simulated data. The decision
boundary is fairly smooth compared to the lower panel, where a 1-nearest-
neighbor classifier was used. There is a close relationship between nearest-
neighbor and prototype methods: in 1-nearest-neighbor classification, each
training point is a prototype.
Figure 13.4 shows the training, test and tenfold cross-validation errors
</p>
<p>as a function of the neighborhood size, for the two-class mixture problem.
Since the tenfold CV errors are averages of ten numbers, we can estimate
a standard error.
Because it uses only the training point closest to the query point, the bias
</p>
<p>of the 1-nearest-neighbor estimate is often low, but the variance is high.
A famous result of Cover and Hart (1967) shows that asymptotically the
error rate of the 1-nearest-neighbor classifier is never more than twice the
Bayes rate. The rough idea of the proof is as follows (using squared-error
loss). We assume that the query point coincides with one of the training
points, so that the bias is zero. This is true asymptotically if the dimension
of the feature space is fixed and the training data fills up the space in a
dense fashion. Then the error of the Bayes rule is just the variance of a
Bernoulli random variate (the target at the query point), while the error of
1-nearest-neighbor rule is twice the variance of a Bernoulli random variate,
one contribution each for the training and query targets.
We now give more detail for misclassification loss. At x let k&lowast; be the
</p>
<p>dominant class, and pk(x) the true conditional probability for class k. Then
</p>
<p>Bayes error = 1&minus; pk&lowast;(x), (13.2)
</p>
<p>1-nearest-neighbor error =
</p>
<p>K&sum;
</p>
<p>k=1
</p>
<p>pk(x)(1&minus; pk(x)), (13.3)
</p>
<p>&ge; 1&minus; pk&lowast;(x). (13.4)
The asymptotic 1-nearest-neighbor error rate is that of a random rule; we
pick both the classification and the test point at random with probabili-</p>
<p/>
</div>
<div class="page"><p/>
<p>466 13. Prototypes and Nearest-Neighbors
</p>
<p>15-Nearest Neighbors
</p>
<p>. ....
.... ......
</p>
<p>... ..... ...... .....
.... .............
</p>
<p>..................
.............................
</p>
<p>............................. ...........................
..................................................................
................................................................
</p>
<p>............................................................................
................................... ...................................
</p>
<p>........................................................................... .. .. . .
..................................... .. . .... .... ......................................... ........................
</p>
<p>.......................................... ..................................................................... .............................
.......................................................................................................................................................
</p>
<p>. ...................................................................................................................................
............. .............................................. .. ..........................................
</p>
<p>.......................................... . ........................................
.............................................................................
</p>
<p>.......................................................................
...................................................................
</p>
<p>...............................................................
.............................................................
</p>
<p>.........................................................
......................................................
...................................................
</p>
<p>...............................................
.........................................
</p>
<p>....................................................................................................................................................................................................................................
</p>
<p>....................................................................................................................................................................................................................................
</p>
<p>....................................................................................................................................................................................................................................
</p>
<p>....................................................................................................................................................................................................................................
</p>
<p>....................................................................................................................................................................................................................................
</p>
<p>....................................................................................................................................................................................................................................
</p>
<p>....................................................................................................................................................................................................................................
</p>
<p>....................................................................................................................................................................................................................................
</p>
<p>....................................................................................................................................................................................................................................
</p>
<p>................................................................................................................................................................... . ..............................................................
</p>
<p>.................................................... .............................................................................................................................................................................
</p>
<p>..................................................... ..................................................................................................... ...... ..........................................................
</p>
<p>........................................... ..... ..... ............................................................................................... .... .........................................................
</p>
<p>............................................. ...... ............................................................................................................ ....................................................
</p>
<p>....................................................... ....................................................................... .................................. ................................................
</p>
<p>.............. .... . . ........................ ............................................................. ........................ ................................................
.... ................. ............................................. ....... ... ..........................................
</p>
<p>.............................................................. ....................
</p>
<p>.......... . ...................... ......
</p>
<p>.... .. . ..
................... ....................
</p>
<p>.......... ...........................................................
</p>
<p>....................................... .......................................... .. ................
</p>
<p>........................................ ................................................................ ......................................
</p>
<p>..................................... ................................................................................ ........................................
</p>
<p>..................................... . ........................................................................... .......................................
</p>
<p>................................... .. . . ................................ .................................... . . ................. ......... .. .
</p>
<p>.................................. .................................................... ............
</p>
<p>................................... ........................................
</p>
<p>........................................................................................ .
</p>
<p>................................................................................................... ......
</p>
<p>.................................................. ......... ...................................................................................
</p>
<p>................................................................... ... .............................................................................
</p>
<p>..........................................................................................................................................................
</p>
<p>...............................................................................................................................................................
</p>
<p>...................................................................................................................................................................
</p>
<p>......................................................................................................................................................................
</p>
<p>.........................................................................................................................................................................
</p>
<p>.............................................................................................................................................................................
</p>
<p>...............................................................................................................................................................................
</p>
<p>...................................................................................................................................................................................
</p>
<p>........................................................................................................................................................................................
</p>
<p>..............................................................................................
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>1-Nearest Neighbor
</p>
<p>............
..... ................. ..
</p>
<p>.......... .............. .......
........... ....................... ... ...
</p>
<p>............. ..... ............... ..... ....
......... .... ................ ....
</p>
<p>..... ..... ..... ...... ..................
. ..... ...... ...... .... ....... ... .......... .
</p>
<p>.. .... .. .... . ........ ............ ........
.... ...... ............ .......... ........ .............. ...
..... ........ ............. .......... ...... ............... ....... ...
</p>
<p>.......... ............. ......... .......................... ............. .......................
.. . ..... .. .. ............................. ....... .. . ...... ........................
</p>
<p>.. ....... ............ .............. ............................... .............................. .....................
... ............ ....................... ........................ ....................... ...................
</p>
<p>....................... ...... ..................... ..................... ..... .....................
.... .......... ............... ........ ....... ... ........................... .... ... ................ ....................... .....................
... . ......... ..... .... .............. ............................. ...... ... ...................................
</p>
<p>........ .......... ... ......................................... ...... .......... ...................................
... .......... ......................................... ..............................................
</p>
<p>..... ...................................... ...................................
.. ......................................................................
</p>
<p>........................................................................
..................................................................
</p>
<p>...............................................................
...........................................................
.........................................................
</p>
<p>.....................................................
...................................................
</p>
<p>...............................................
.............................................
</p>
<p>....................................................................................................................................................................................................................................
</p>
<p>....................................................................................................................................................................................................................................
</p>
<p>....................................................................................................................................................................................................................................
</p>
<p>....................................................................................................................................................................................................................................
</p>
<p>....................................................................................................................................................................................................................................
</p>
<p>....................................................................................................................................................................................................................................
</p>
<p>....................................................................................................................................................................................................................................
</p>
<p>................................................................... ........................................................................................................ .............................................
</p>
<p>............................................. .......... ......................................................................................... ........... .................................................
</p>
<p>........................................... ....................... .............................................................................. .................. ...................................
</p>
<p>............................................ ............... ......................................................... .............. ......... ... .......................................
</p>
<p>..................... ............ ....... ... .............................................................. .......... ...... .... ........................................
</p>
<p>.................... ...... .. ....... .................................................................... . .. .............. .................................................
</p>
<p>.................. ... ............ .... ... ........................................................ .. ............. ... .............................................
</p>
<p>............ .............. ..... ... ..... ..... ................................................. ............. ...... . ....... ... ...................................
</p>
<p>............. ............... .. .......... ...... ..... ... .............................................. .................. .......... ........ ...............................
</p>
<p>.............. ..... .......... ......... ............................................... ... ..... ......... ....................................
</p>
<p>............ . ... ... ........ ............................................... ..... ... ....... ............ ...................
</p>
<p>.......... .. .... ..... ............. ... .... .............
</p>
<p>....... .... ..... . ........... ..... ...... .......
....
</p>
<p>.................
........................
</p>
<p>........ ............ .......
.......... ...................
</p>
<p>........................
...................... ...
</p>
<p>............... ..................... ..
.................... ............................ ...........
</p>
<p>.............................. ................................................ .. ... ............
........................ ...... . ...... ...................................... ...... . ..... .............
</p>
<p>................................... .. .................................................. .. . .......................
</p>
<p>......................................... ... ......................................................... ...... ........................
</p>
<p>............................. ..... ........ .................................................... ... . .. .. ... .......
</p>
<p>............................ .. .. .. .. .................................. ........... .... . ....... .... .
</p>
<p>.............................................. ..... ..... ................................................... ...... . ......
</p>
<p>............................................ .... .. ............................................................... ..... .
</p>
<p>......................................................... .............................................................................. .....
</p>
<p>...................................................................... ............................................................................ .......
</p>
<p>.............................................................................................................................................................
</p>
<p>..............................................................................................................................................................
</p>
<p>....................................................................................................................................................................
</p>
<p>.......................................................................................................................................................................
</p>
<p>..........................................................................................................................................................................
</p>
<p>.............................................................................................................................................................................
</p>
<p>................................................................................................................................................................................
</p>
<p>...................................................................................................................................................................................
</p>
<p>......................................................................................................................................................................................
</p>
<p>............................................................................................
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>FIGURE 13.3. k-nearest-neighbor classifiers applied to the simulation data of
Figure 13.1. The broken purple curve in the background is the Bayes decision
boundary.</p>
<p/>
</div>
<div class="page"><p/>
<p>13.3 k-Nearest-Neighbor Classifiers 467
</p>
<p>Number of Neighbors
</p>
<p>M
is
</p>
<p>c
la
</p>
<p>s
s
if
ic
</p>
<p>a
ti
o
</p>
<p>n
 E
</p>
<p>rr
o
</p>
<p>rs
</p>
<p>0 5 10 15 20 25 30
</p>
<p>0
.0
</p>
<p>0
.0
</p>
<p>5
0
</p>
<p>.1
0
</p>
<p>0
.1
</p>
<p>5
0
</p>
<p>.2
0
</p>
<p>0
.2
</p>
<p>5
0
</p>
<p>.3
0
</p>
<p>&bull;
</p>
<p>&bull; &bull; &bull; &bull;
&bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
&bull; &bull;
</p>
<p>&bull;
&bull; &bull;
</p>
<p>&bull; &bull; &bull; &bull; &bull; &bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull; &bull; &bull; &bull; &bull; &bull;
&bull;
</p>
<p>&bull; &bull; &bull;
&bull; &bull;
</p>
<p>Test Error
</p>
<p>10-fold CV
</p>
<p>Training Error
</p>
<p>Bayes Error
</p>
<p>7-Nearest Neighbors
</p>
<p>.. .
</p>
<p>. . .. . . . .
</p>
<p>. . . . . . .. . . . . . .
</p>
<p>. . . . . . . . .. . . . . . . . . . .
</p>
<p>. . . . . . . . . . . .. . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . . .
. . . . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . . . . . . . .. . . . . . . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . . .. . . . . . . . . . . . . . .
. . . . . . . . . . . .. . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . .. . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . .. . . . . . . . . . . . .
. . . . . . . . . . . .. . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . .. . . . . . . . . . . .
. . . . . . . . . . . . .. . . . . . . . . . .
</p>
<p>. . . . . . . . . . . .. . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . .. . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . .. . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . .. . . . . . . . . . . .
. . . . . . . . . . . .. . . . . . . . . . . .
. . . . . . . . . . . . .. . . . . . . . . . . . .
. . . . . . . . . . . . .. . . . . . . . . . . . .
. . . . . . . . . . . . .. . . . . . . . . . . . . .
. . . . . . . . . . . . . .. . . . . . . . . . . . . . .
. . . . . . . . . . . . . . .. . . . . . . . . . . . . . .
. . . . . . . . . . . . . . .. . . . . . . . . . . . . . .
. . . . . . . . . . . . . . .. . . . . . . . . . . . . . .
</p>
<p>. . . . . . . . . . . . . . .
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>Training Error: 0.145
</p>
<p>Test Error:       0.225
</p>
<p>Bayes Error:    0.210
</p>
<p>FIGURE 13.4. k-nearest-neighbors on the two-class mixture data. The upper
panel shows the misclassification errors as a function of neighborhood size. Stan-
dard error bars are included for 10-fold cross validation. The lower panel shows
the decision boundary for 7-nearest-neighbors, which appears to be optimal for
minimizing test error. The broken purple curve in the background is the Bayes
decision boundary.</p>
<p/>
</div>
<div class="page"><p/>
<p>468 13. Prototypes and Nearest-Neighbors
</p>
<p>ties pk(x), k = 1, . . . ,K. For K = 2 the 1-nearest-neighbor error rate is
2pk&lowast;(x)(1 &minus; pk&lowast;(x)) &le; 2(1 &minus; pk&lowast;(x)) (twice the Bayes error rate). More
generally, one can show (Exercise 13.3)
</p>
<p>K&sum;
</p>
<p>k=1
</p>
<p>pk(x)(1&minus; pk(x)) &le; 2(1&minus; pk&lowast;(x))&minus;
K
</p>
<p>K &minus; 1(1&minus; pk&lowast;(x))
2. (13.5)
</p>
<p>Many additional results of this kind have been derived; Ripley (1996) sum-
marizes a number of them.
This result can provide a rough idea about the best performance that
</p>
<p>is possible in a given problem. For example, if the 1-nearest-neighbor rule
has a 10% error rate, then asymptotically the Bayes error rate is at least
5%. The kicker here is the asymptotic part, which assumes the bias of the
nearest-neighbor rule is zero. In real problems the bias can be substantial.
The adaptive nearest-neighbor rules, described later in this chapter, are an
attempt to alleviate this bias. For simple nearest-neighbors, the bias and
variance characteristics can dictate the optimal number of near neighbors
for a given problem. This is illustrated in the next example.
</p>
<p>13.3.1 Example: A Comparative Study
</p>
<p>We tested the nearest-neighbors, K-means and LVQ classifiers on two sim-
ulated problems. There are 10 independent features Xj , each uniformly
distributed on [0, 1]. The two-class 0-1 target variable is defined as follows:
</p>
<p>Y = I
</p>
<p>(
X1 &gt;
</p>
<p>1
</p>
<p>2
</p>
<p>)
; problem 1: &ldquo;easy&rdquo;,
</p>
<p>Y = I
</p>
<p>⎛
⎝sign
</p>
<p>⎧
⎨
⎩
</p>
<p>3&prod;
</p>
<p>j=1
</p>
<p>(
Xj &minus;
</p>
<p>1
</p>
<p>2
</p>
<p>)⎫⎬
⎭ &gt; 0
</p>
<p>⎞
⎠ ; problem 2: &ldquo;difficult.&rdquo;
</p>
<p>(13.6)
</p>
<p>Hence in the first problem the two classes are separated by the hyperplane
X1 = 1/2; in the second problem, the two classes form a checkerboard
pattern in the hypercube defined by the first three features. The Bayes
error rate is zero in both problems. There were 100 training and 1000 test
observations.
Figure 13.5 shows the mean and standard error of the misclassification
</p>
<p>error for nearest-neighbors, K-means and LVQ over ten realizations, as
the tuning parameters are varied. We see that K-means and LVQ give
nearly identical results. For the best choices of their tuning parameters,
K-means and LVQ outperform nearest-neighbors for the first problem, and
they perform similarly for the second problem. Notice that the best value
of each tuning parameter is clearly situation dependent. For example 25-
nearest-neighbors outperforms 1-nearest-neighbor by a factor of 70% in the</p>
<p/>
</div>
<div class="page"><p/>
<p>13.3 k-Nearest-Neighbor Classifiers 469
</p>
<p>Number of Neighbors
</p>
<p>M
is
</p>
<p>c
la
</p>
<p>s
s
if
ic
</p>
<p>a
ti
o
</p>
<p>n
 E
</p>
<p>rr
o
</p>
<p>r
</p>
<p>0 20 40 60
</p>
<p>0
.1
</p>
<p>0
.2
</p>
<p>0
.3
</p>
<p>0
.4
</p>
<p>0
.5
</p>
<p>Nearest Neighbors / Easy
</p>
<p>Number of Prototypes per Class
</p>
<p>M
is
</p>
<p>c
la
</p>
<p>s
s
if
ic
</p>
<p>a
ti
o
</p>
<p>n
 E
</p>
<p>rr
o
</p>
<p>r
</p>
<p>0 5 10 15 20 25 30
</p>
<p>0
.1
</p>
<p>0
.2
</p>
<p>0
.3
</p>
<p>0
.4
</p>
<p>0
.5
</p>
<p>K-means &amp; LVQ / Easy
</p>
<p>Number of Neighbors
</p>
<p>M
is
</p>
<p>c
la
</p>
<p>s
s
if
ic
</p>
<p>a
ti
o
</p>
<p>n
 E
</p>
<p>rr
o
</p>
<p>r
</p>
<p>0 20 40 60
</p>
<p>0
.4
</p>
<p>0
0
</p>
<p>.4
5
</p>
<p>0
.5
</p>
<p>0
0
</p>
<p>.5
5
</p>
<p>0
.6
</p>
<p>0
</p>
<p>Nearest Neighbors / Difficult
</p>
<p>Number of Prototypes per Class
</p>
<p>M
is
</p>
<p>c
la
</p>
<p>s
s
if
ic
</p>
<p>a
ti
o
</p>
<p>n
 E
</p>
<p>rr
o
</p>
<p>r
</p>
<p>0 5 10 15 20 25 30
</p>
<p>0
.4
</p>
<p>0
0
</p>
<p>.4
5
</p>
<p>0
.5
</p>
<p>0
0
</p>
<p>.5
5
</p>
<p>0
.6
</p>
<p>0
</p>
<p>K-means &amp; LVQ / Difficult
</p>
<p>FIGURE 13.5. Mean &plusmn; one standard error of misclassification error for near-
est-neighbors, K-means (blue) and LVQ (red) over ten realizations for two sim-
ulated problems: &ldquo;easy&rdquo; and &ldquo;difficult,&rdquo; described in the text.</p>
<p/>
</div>
<div class="page"><p/>
<p>470 13. Prototypes and Nearest-Neighbors
</p>
<p>Spectral Band 1 Spectral Band 2 Spectral Band 3
</p>
<p>Spectral Band 4 Land Usage Predicted Land Usage
</p>
<p>FIGURE 13.6. The first four panels are LANDSAT images for an agricultural
area in four spectral bands, depicted by heatmap shading. The remaining two
panels give the actual land usage (color coded) and the predicted land usage using
a five-nearest-neighbor rule described in the text.
</p>
<p>first problem, while 1-nearest-neighbor is best in the second problem by a
factor of 18%. These results underline the importance of using an objective,
data-based method like cross-validation to estimate the best value of a
tuning parameter (see Figure 13.4 and Chapter 7).
</p>
<p>13.3.2 Example: k-Nearest-Neighbors and Image Scene
Classification
</p>
<p>The STATLOG project (Michie et al., 1994) used part of a LANDSAT
image as a benchmark for classification (82&times;100 pixels). Figure 13.6 shows
four heat-map images, two in the visible spectrum and two in the infrared,
for an area of agricultural land in Australia. Each pixel has a class label
from the 7-element set G = {red soil, cotton, vegetation stubble, mixture,
gray soil, damp gray soil, very damp gray soil}, determined manually by
research assistants surveying the area. The lower middle panel shows the
actual land usage, shaded by different colors to indicate the classes. The
objective is to classify the land usage at a pixel, based on the information
in the four spectral bands.
Five-nearest-neighbors produced the predicted map shown in the bot-
</p>
<p>tom right panel, and was computed as follows. For each pixel we extracted
an 8-neighbor feature map&mdash;the pixel itself and its 8 immediate neighbors</p>
<p/>
</div>
<div class="page"><p/>
<p>13.3 k-Nearest-Neighbor Classifiers 471
</p>
<p>N
</p>
<p>N
</p>
<p>N N
</p>
<p>X
</p>
<p>N
</p>
<p>N
</p>
<p>N
</p>
<p>N
</p>
<p>FIGURE 13.7. A pixel and its 8-neighbor feature map.
</p>
<p>(see Figure 13.7). This is done separately in the four spectral bands, giving
(1+8)&times;4 = 36 input features per pixel. Then five-nearest-neighbors classi-
fication was carried out in this 36-dimensional feature space. The resulting
test error rate was about 9.5% (see Figure 13.8). Of all the methods used
in the STATLOG project, including LVQ, CART, neural networks, linear
discriminant analysis and many others, k-nearest-neighbors performed best
on this task. Hence it is likely that the decision boundaries in IR36 are quite
irregular.
</p>
<p>13.3.3 Invariant Metrics and Tangent Distance
</p>
<p>In some problems, the training features are invariant under certain natural
transformations. The nearest-neighbor classifier can exploit such invari-
ances by incorporating them into the metric used to measure the distances
between objects. Here we give an example where this idea was used with
great success, and the resulting classifier outperformed all others at the
time of its development (Simard et al., 1993).
The problem is handwritten digit recognition, as discussed is Chapter 1
</p>
<p>and Section 11.7. The inputs are grayscale images with 16 &times; 16 = 256
pixels; some examples are shown in Figure 13.9. At the top of Figure 13.10,
a &ldquo;3&rdquo; is shown, in its actual orientation (middle) and rotated 7.5◦ and 15◦
</p>
<p>in either direction. Such rotations can often occur in real handwriting, and
it is obvious to our eye that this &ldquo;3&rdquo; is still a &ldquo;3&rdquo; after small rotations.
Hence we want our nearest-neighbor classifier to consider these two &ldquo;3&rdquo;s
to be close together (similar). However the 256 grayscale pixel values for a
rotated &ldquo;3&rdquo; will look quite different from those in the original image, and
hence the two objects can be far apart in Euclidean distance in IR256.
</p>
<p>We wish to remove the effect of rotation in measuring distances between
two digits of the same class. Consider the set of pixel values consisting of
the original &ldquo;3&rdquo; and its rotated versions. This is a one-dimensional curve in
IR256, depicted by the green curve passing through the &ldquo;3&rdquo; in Figure 13.10.
Figure 13.11 shows a stylized version of IR256, with two images indicated by
xi and xi&prime; . These might be two different &ldquo;3&rdquo;s, for example. Through each
image we have drawn the curve of rotated versions of that image, called</p>
<p/>
</div>
<div class="page"><p/>
<p>472 13. Prototypes and Nearest-Neighbors
</p>
<p>STATLOG results
</p>
<p>Method
</p>
<p>T
e
</p>
<p>s
t 
</p>
<p>M
is
</p>
<p>c
la
</p>
<p>s
s
if
ic
</p>
<p>a
ti
o
</p>
<p>n
 E
</p>
<p>rr
o
</p>
<p>r
</p>
<p>2 4 6 8 10 12 14
</p>
<p>0
.0
</p>
<p>0
.0
</p>
<p>5
0
</p>
<p>.1
0
</p>
<p>0
.1
</p>
<p>5
</p>
<p>LVQ
</p>
<p>RBF
</p>
<p>ALLOC80
CART Neural
</p>
<p>NewID C4.5
QDA
</p>
<p>SMART
Logistic
</p>
<p>LDA
</p>
<p>DANN
</p>
<p>K-NN
</p>
<p>FIGURE 13.8. Test-error performance for a number of classifiers, as reported
by the STATLOG project. The entry DANN is a variant of k-nearest neighbors,
using an adaptive metric (Section 13.4.2).
</p>
<p>FIGURE 13.9. Examples of grayscale images of handwritten digits.</p>
<p/>
</div>
<div class="page"><p/>
<p>13.3 k-Nearest-Neighbor Classifiers 473
</p>
<p>Tangent
</p>
<p>+ α .
</p>
<p>Transformations of 3
</p>
<p>0 7.5&minus;15 &minus;7.5
</p>
<p>3
</p>
<p>α=0 α=0.1α=&minus; 0.2 α=&minus; 0.1 α=0.2
</p>
<p>Linear equation for 
images above
</p>
<p>15
</p>
<p>Pixel space
</p>
<p>FIGURE 13.10. The top row shows a &ldquo; 3&rdquo; in its original orientation (middle)
and rotated versions of it. The green curve in the middle of the figure depicts
this set of rotated &ldquo; 3&rdquo; in 256-dimensional space. The red line is the tangent line
to the curve at the original image, with some &ldquo; 3&rdquo;s on this tangent line, and its
equation shown at the bottom of the figure.
</p>
<p>invariance manifolds in this context. Now, rather than using the usual
Euclidean distance between the two images, we use the shortest distance
between the two curves. In other words, the distance between the two
images is taken to be the shortest Euclidean distance between any rotated
version of first image, and any rotated version of the second image. This
distance is called an invariant metric.
In principle one could carry out 1-nearest-neighbor classification using
</p>
<p>this invariant metric. However there are two problems with it. First, it is
very difficult to calculate for real images. Second, it allows large trans-
formations that can lead to poor performance. For example a &ldquo;6&rdquo; would
be considered close to a &ldquo;9&rdquo; after a rotation of 180◦. We need to restrict
attention to small rotations.
The use of tangent distance solves both of these problems. As shown in
</p>
<p>Figure 13.10, we can approximate the invariance manifold of the image
&ldquo;3&rdquo; by its tangent at the original image. This tangent can be computed
by estimating the direction vector from small rotations of the image, or by
more sophisticated spatial smoothing methods (Exercise 13.4.) For large
rotations, the tangent image no longer looks like a &ldquo;3,&rdquo; so the problem
with large transformations is alleviated.</p>
<p/>
</div>
<div class="page"><p/>
<p>474 13. Prototypes and Nearest-Neighbors
</p>
<p>Transformations
</p>
<p>Transformations
</p>
<p>xi xi&prime;
</p>
<p>of xi
</p>
<p>of xi&prime;
</p>
<p>Tangent distance
</p>
<p>Euclidean distance
between xi and xi&prime;
</p>
<p>Distance between
transformed
xi and xi&prime;
</p>
<p>FIGURE 13.11. Tangent distance computation for two images xi and xi&prime; .
Rather than using the Euclidean distance between xi and xi&prime; , or the shortest
distance between the two curves, we use the shortest distance between the two
tangent lines.
</p>
<p>The idea then is to compute the invariant tangent line for each training
image. For a query image to be classified, we compute its invariant tangent
line, and find the closest line to it among the lines in the training set. The
class (digit) corresponding to this closest line is our predicted class for the
query image. In Figure 13.11 the two tangent lines intersect, but this is only
because we have been forced to draw a two-dimensional representation of
the actual 256-dimensional situation. In IR256 the probability of two such
lines intersecting is effectively zero.
Now a simpler way to achieve this invariance would be to add into the
</p>
<p>training set a number of rotated versions of each training image, and then
just use a standard nearest-neighbor classifier. This idea is called &ldquo;hints&rdquo; in
Abu-Mostafa (1995), and works well when the space of invariances is small.
So far we have presented a simplified version of the problem. In addition to
rotation, there are six other types of transformations under which we would
like our classifier to be invariant. There are translation (two directions),
scaling (two directions), sheer, and character thickness. Hence the curves
and tangent lines in Figures 13.10 and 13.11 are actually 7-dimensional
manifolds and hyperplanes. It is infeasible to add transformed versions
of each training image to capture all of these possibilities. The tangent
manifolds provide an elegant way of capturing the invariances.
Table 13.1 shows the test misclassification error for a problem with 7291
</p>
<p>training images and 2007 test digits (the U.S. Postal Services database), for
a carefully constructed neural network, and simple 1-nearest-neighbor and</p>
<p/>
</div>
<div class="page"><p/>
<p>13.4 Adaptive Nearest-Neighbor Methods 475
</p>
<p>TABLE 13.1. Test error rates for the handwritten ZIP code problem.
</p>
<p>Method Error rate
</p>
<p>Neural-net 0.049
1-nearest-neighbor/Euclidean distance 0.055
1-nearest-neighbor/tangent distance 0.026
</p>
<p>tangent distance 1-nearest-neighbor rules. The tangent distance nearest-
neighbor classifier works remarkably well, with test error rates near those
for the human eye (this is a notoriously difficult test set). In practice,
it turned out that nearest-neighbors are too slow for online classification
in this application (see Section 13.5), and neural network classifiers were
subsequently developed to mimic it.
</p>
<p>13.4 Adaptive Nearest-Neighbor Methods
</p>
<p>When nearest-neighbor classification is carried out in a high-dimensional
feature space, the nearest neighbors of a point can be very far away, causing
bias and degrading the performance of the rule.
To quantify this, consider N data points uniformly distributed in the unit
</p>
<p>cube [&minus; 12 , 12 ]p. Let R be the radius of a 1-nearest-neighborhood centered at
the origin. Then
</p>
<p>median(R) = v&minus;1/pp
</p>
<p>(
1&minus; 1
</p>
<p>2
</p>
<p>1/N)1/p
, (13.7)
</p>
<p>where vpr
p is the volume of the sphere of radius r in p dimensions. Fig-
</p>
<p>ure 13.12 shows the median radius for various training sample sizes and
dimensions. We see that median radius quickly approaches 0.5, the dis-
tance to the edge of the cube.
What can be done about this problem? Consider the two-class situation
</p>
<p>in Figure 13.13. There are two features, and a nearest-neighborhood at
a query point is depicted by the circular region. Implicit in near-neighbor
classification is the assumption that the class probabilities are roughly con-
stant in the neighborhood, and hence simple averages give good estimates.
However, in this example the class probabilities vary only in the horizontal
direction. If we knew this, we would stretch the neighborhood in the verti-
cal direction, as shown by the tall rectangular region. This will reduce the
bias of our estimate and leave the variance the same.
In general, this calls for adapting the metric used in nearest-neighbor
</p>
<p>classification, so that the resulting neighborhoods stretch out in directions
for which the class probabilities don&rsquo;t change much. In high-dimensional
feature space, the class probabilities might change only a low-dimensional
subspace and hence there can be considerable advantage to adapting the
metric.</p>
<p/>
</div>
<div class="page"><p/>
<p>476 13. Prototypes and Nearest-Neighbors
</p>
<p>Dimension
</p>
<p>M
e
</p>
<p>d
ia
</p>
<p>n
 R
</p>
<p>a
d
</p>
<p>iu
s
</p>
<p>0 5 10 15
</p>
<p>0
.0
</p>
<p>0
.1
</p>
<p>0
.2
</p>
<p>0
.3
</p>
<p>0
.4
</p>
<p>0
.5
</p>
<p>0
.6
</p>
<p>N=100
</p>
<p>N=1,000
</p>
<p>N=10,000
</p>
<p>FIGURE 13.12. Median radius of a 1-nearest-neighborhood, for uniform data
with N observations in p dimensions.
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>&bull;
</p>
<p>5-Nearest Neighborhoods
</p>
<p>FIGURE 13.13. The points are uniform in the cube, with the vertical line sepa-
rating class red and green. The vertical strip denotes the 5-nearest-neighbor region
using only the horizontal coordinate to find the nearest-neighbors for the target
point (solid dot). The sphere shows the 5-nearest-neighbor region using both co-
ordinates, and we see in this case it has extended into the class-red region (and
is dominated by the wrong class in this instance).</p>
<p/>
</div>
<div class="page"><p/>
<p>13.4 Adaptive Nearest-Neighbor Methods 477
</p>
<p>Friedman (1994a) proposed a method in which rectangular neighbor-
hoods are found adaptively by successively carving away edges of a box
containing the training data. Here we describe the discriminant adaptive
nearest-neighbor (DANN) rule of Hastie and Tibshirani (1996a). Earlier,
related proposals appear in Short and Fukunaga (1981) and Myles and
Hand (1990).
At each query point a neighborhood of say 50 points is formed, and the
</p>
<p>class distribution among the points is used to decide how to deform the
neighborhood&mdash;that is, to adapt the metric. The adapted metric is then
used in a nearest-neighbor rule at the query point. Thus at each query
point a potentially different metric is used.
In Figure 13.13 it is clear that the neighborhood should be stretched in
</p>
<p>the direction orthogonal to line joining the class centroids. This direction
also coincides with the linear discriminant boundary, and is the direction
in which the class probabilities change the least. In general this direction
of maximum change will not be orthogonal to the line joining the class cen-
troids (see Figure 4.9 on page 116.) Assuming a local discriminant model,
the information contained in the local within- and between-class covari-
ance matrices is all that is needed to determine the optimal shape of the
neighborhood.
The discriminant adaptive nearest-neighbor (DANN) metric at a query
</p>
<p>point x0 is defined by
</p>
<p>D(x, x0) = (x&minus; x0)TΣ(x&minus; x0), (13.8)
</p>
<p>where
</p>
<p>Σ = W&minus;1/2[W&minus;1/2BW&minus;1/2 + ǫI]W&minus;1/2
</p>
<p>= W&minus;1/2[B&lowast; + ǫI]W&minus;1/2. (13.9)
</p>
<p>Here W is the pooled within-class covariance matrix
&sum;K
</p>
<p>k=1 πkWk and B
</p>
<p>is the between class covariance matrix
&sum;K
</p>
<p>k=1 πk(x̄k &minus; x̄)(x̄k &minus; x̄)T , with
W and B computed using only the 50 nearest neighbors around x0. After
computation of the metric, it is used in a nearest-neighbor rule at x0.
</p>
<p>This complicated formula is actually quite simple in its operation. It first
spheres the data with respect to W, and then stretches the neighborhood
in the zero-eigenvalue directions of B&lowast; (the between-matrix for the sphered
data ). This makes sense, since locally the observed class means do not dif-
fer in these directions. The ǫ parameter rounds the neighborhood, from an
infinite strip to an ellipsoid, to avoid using points far away from the query
point. The value of ǫ = 1 seems to work well in general. Figure 13.14 shows
the resulting neighborhoods for a problem where the classes form two con-
centric circles. Notice how the neighborhoods stretch out orthogonally to
the decision boundaries when both classes are present in the neighborhood.
In the pure regions with only one class, the neighborhoods remain circular;</p>
<p/>
</div>
<div class="page"><p/>
<p>478 13. Prototypes and Nearest-Neighbors
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o oo
o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
o
</p>
<p>o o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>FIGURE 13.14. Neighborhoods found by the DANN procedure, at various query
points (centers of the crosses). There are two classes in the data, with one class
surrounding the other. 50 nearest-neighbors were used to estimate the local met-
rics. Shown are the resulting metrics used to form 15-nearest-neighborhoods.
</p>
<p>in these cases the between matrix B = 0, and the Σ in (13.8) is the identity
matrix.
</p>
<p>13.4.1 Example
</p>
<p>Here we generate two-class data in ten dimensions, analogous to the two-
dimensional example of Figure 13.14. All ten predictors in class 1 are inde-
pendent standard normal, conditioned on the squared radius being greater
than 22.4 and less than 40, while the predictors in class 2 are independent
standard normal without the restriction. There are 250 observations in each
class. Hence the first class almost completely surrounds the second class in
the full ten-dimensional space.
In this example there are no pure noise variables, the kind that a nearest-
</p>
<p>neighbor subset selection rule might be able to weed out. At any given
point in the feature space, the class discrimination occurs along only one
direction. However, this direction changes as we move across the feature
space and all variables are important somewhere in the space.
Figure 13.15 shows boxplots of the test error rates over ten realiza-
</p>
<p>tions, for standard 5-nearest-neighbors, LVQ, and discriminant adaptive
5-nearest-neighbors. We used 50 prototypes per class for LVQ, to make
it comparable to 5 nearest-neighbors (since 250/5 = 50). The adaptive
metric significantly reduces the error rate, compared to LVQ or standard
nearest-neighbors.</p>
<p/>
</div>
<div class="page"><p/>
<p>13.4 Adaptive Nearest-Neighbor Methods 479
</p>
<p>0
.0
</p>
<p>0
.1
</p>
<p>0
.2
</p>
<p>0
.3
</p>
<p>0
.4
</p>
<p>5N
N
</p>
<p>LV
Q
</p>
<p>D
AN
</p>
<p>N
</p>
<p>T
e
s
t 
E
</p>
<p>rr
o
r
</p>
<p>FIGURE 13.15. Ten-dimensional simulated example: boxplots of the test error
rates over ten realizations, for standard 5-nearest-neighbors, LVQ with 50 centers,
and discriminant-adaptive 5-nearest-neighbors
</p>
<p>13.4.2 Global Dimension Reduction for Nearest-Neighbors
</p>
<p>The discriminant-adaptive nearest-neighbor method carries out local di-
mension reduction&mdash;that is, dimension reduction separately at each query
point. In many problems we can also benefit from global dimension re-
duction, that is, apply a nearest-neighbor rule in some optimally chosen
subspace of the original feature space. For example, suppose that the two
classes form two nested spheres in four dimensions of feature space, and
there are an additional six noise features whose distribution is independent
of class. Then we would like to discover the important four-dimensional
subspace, and carry out nearest-neighbor classification in that reduced sub-
space. Hastie and Tibshirani (1996a) discuss a variation of the discriminant-
adaptive nearest-neighbor method for this purpose. At each training point
xi, the between-centroids sum of squares matrix Bi is computed, and then
these matrices are averaged over all training points:
</p>
<p>B̄ =
1
</p>
<p>N
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>Bi. (13.10)
</p>
<p>Let e1, e2, . . . , ep be the eigenvectors of the matrix B̄, ordered from largest
to smallest eigenvalue θk. Then these eigenvectors span the optimal sub-
spaces for global subspace reduction. The derivation is based on the fact
that the best rank-L approximation to B̄, B̄[L] =
</p>
<p>&sum;L
ℓ=1 θℓeℓe
</p>
<p>T
ℓ , solves the
</p>
<p>least squares problem
</p>
<p>min
rank(M)=L
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>trace[(Bi &minus;M)2]. (13.11)
</p>
<p>Since each Bi contains information on (a) the local discriminant subspace,
and (b) the strength of discrimination in that subspace, (13.11) can be seen</p>
<p/>
</div>
<div class="page"><p/>
<p>480 13. Prototypes and Nearest-Neighbors
</p>
<p>as a way of finding the best approximating subspace of dimension L to a
series of N subspaces by weighted least squares (Exercise 13.5.)
</p>
<p>In the four-dimensional sphere example mentioned above and examined
in Hastie and Tibshirani (1996a), four of the eigenvalues θℓ turn out to be
large (having eigenvectors nearly spanning the interesting subspace), and
the remaining six are near zero. Operationally, we project the data into
the leading four-dimensional subspace, and then carry out nearest neighbor
classification. In the satellite image classification example in Section 13.3.2,
the technique labeled DANN in Figure 13.8 used 5-nearest-neighbors in a
globally reduced subspace. There are also connections of this technique
with the sliced inverse regression proposal of Duan and Li (1991). These
authors use similar ideas in the regression setting, but do global rather
than local computations. They assume and exploit spherical symmetry of
the feature distribution to estimate interesting subspaces.
</p>
<p>13.5 Computational Considerations
</p>
<p>One drawback of nearest-neighbor rules in general is the computational
load, both in finding the neighbors and storing the entire training set. With
N observations and p predictors, nearest-neighbor classification requiresNp
operations to find the neighbors per query point. There are fast algorithms
for finding nearest-neighbors (Friedman et al., 1975; Friedman et al., 1977)
which can reduce this load somewhat. Hastie and Simard (1998) reduce
the computations for tangent distance by developing analogs of K-means
clustering in the context of this invariant metric.
Reducing the storage requirements is more difficult, and various editing
</p>
<p>and condensing procedures have been proposed. The idea is to isolate a
subset of the training set that suffices for nearest-neighbor predictions, and
throw away the remaining training data. Intuitively, it seems important to
keep the training points that are near the decision boundaries and on the
correct side of those boundaries, while some points far from the boundaries
could be discarded.
The multi-edit algorithm of Devijver and Kittler (1982) divides the data
</p>
<p>cyclically into training and test sets, computing a nearest neighbor rule on
the training set and deleting test points that are misclassified. The idea is
to keep homogeneous clusters of training observations.
The condensing procedure of Hart (1968) goes further, trying to keep
</p>
<p>only important exterior points of these clusters. Starting with a single ran-
domly chosen observation as the training set, each additional data item is
processed one at a time, adding it to the training set only if it is misclas-
sified by a nearest-neighbor rule computed on the current training set.
These procedures are surveyed in Dasarathy (1991) and Ripley (1996).
</p>
<p>They can also be applied to other learning procedures besides nearest-</p>
<p/>
</div>
<div class="page"><p/>
<p>Exercises 481
</p>
<p>neighbors. While such methods are sometimes useful, we have not had
much practical experience with them, nor have we found any systematic
comparison of their performance in the literature.
</p>
<p>Bibliographic Notes
</p>
<p>The nearest-neighbor method goes back at least to Fix and Hodges (1951).
The extensive literature on the topic is reviewed by Dasarathy (1991);
Chapter 6 of Ripley (1996) contains a good summary. K-means cluster-
ing is due to Lloyd (1957) and MacQueen (1967). Kohonen (1989) intro-
duced learning vector quantization. The tangent distance method is due to
Simard et al. (1993). Hastie and Tibshirani (1996a) proposed the discrim-
inant adaptive nearest-neighbor technique.
</p>
<p>Exercises
</p>
<p>Ex. 13.1 Consider a Gaussian mixture model where the covariance matrices
are assumed to be scalar: Σr = σI &forall;r = 1, . . . , R, and σ is a fixed param-
eter. Discuss the analogy between the K-means clustering algorithm and
the EM algorithm for fitting this mixture model in detail. Show that in the
limit σ &rarr; 0 the two methods coincide.
</p>
<p>Ex. 13.2 Derive formula (13.7) for the median radius of the 1-nearest-
neighborhood.
</p>
<p>Ex. 13.3 Let E&lowast; be the error rate of the Bayes rule in a K-class problem,
where the true class probabilities are given by pk(x), k = 1, . . . ,K. As-
suming the test point and training point have identical features x, prove
(13.5)
</p>
<p>K&sum;
</p>
<p>k=1
</p>
<p>pk(x)(1&minus; pk(x)) &le; 2(1&minus; pk&lowast;(x))&minus;
K
</p>
<p>K &minus; 1(1&minus; pk&lowast;(x))
2.
</p>
<p>where k&lowast; = argmaxk pk(x). Hence argue that the error rate of the 1-
nearest-neighbor rule converges in L1, as the size of the training set in-
creases, to a value E1, bounded above by
</p>
<p>E&lowast;
(
2&minus; E&lowast; K
</p>
<p>K &minus; 1
)
. (13.12)
</p>
<p>[This statement of the theorem of Cover and Hart (1967) is taken from
Chapter 6 of Ripley (1996), where a short proof is also given].</p>
<p/>
</div>
<div class="page"><p/>
<p>482 13. Prototypes and Nearest-Neighbors
</p>
<p>Ex. 13.4 Consider an image to be a function F (x) : IR2 �&rarr; IR1 over the two-
dimensional spatial domain (paper coordinates). Then F (c+x0+A(x&minus;x0))
represents an affine transformation of the image F , where A is a 2 &times; 2
matrix.
</p>
<p>1. Decompose A (via Q-R) in such a way that parameters identifying
the four affine transformations (two scale, shear and rotation) are
clearly identified.
</p>
<p>2. Using the chain rule, show that the derivative of F (c+x0+A(x&minus;x0))
w.r.t. each of these parameters can be represented in terms of the two
spatial derivatives of F .
</p>
<p>3. Using a two-dimensional kernel smoother (Chapter 6), describe how
to implement this procedure when the images are quantized to 16&times;16
pixels.
</p>
<p>Ex. 13.5 Let Bi, i = 1, 2, . . . , N be square p&times; p positive semi-definite ma-
trices and let B̄ = (1/N)
</p>
<p>&sum;
Bi. Write the eigen-decomposition of B̄ as&sum;p
</p>
<p>ℓ=1 θℓeℓe
T
ℓ with θℓ &ge; θℓ&minus;1 &ge; &middot; &middot; &middot; &ge; θ1. Show that the best rank-L approx-
</p>
<p>imation for the Bi,
</p>
<p>min
rank(M)=L
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>trace[(Bi &minus;M)2],
</p>
<p>is given by B̄[L] =
&sum;L
</p>
<p>ℓ=1 θℓeℓe
T
ℓ . (Hint: Write
</p>
<p>&sum;N
i=1 trace[(Bi &minus;M)2] as
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>trace[(Bi &minus; B̄)2] +
N&sum;
</p>
<p>i=1
</p>
<p>trace[(M&minus; B̄)2]).
</p>
<p>Ex. 13.6 Here we consider the problem of shape averaging. In particular,
Li, i = 1, . . . ,M are each N &times; 2 matrices of points in IR2, each sampled
from corresponding positions of handwritten (cursive) letters. We seek an
affine invariant average V, also N &times; 2, VTV = I, of the M letters Li with
the following property: V minimizes
</p>
<p>M&sum;
</p>
<p>j=1
</p>
<p>min
Aj
</p>
<p>‖Lj &minus;VAj‖2 .
</p>
<p>Characterize the solution.
This solution can suffer if some of the letters are big and dominate the
</p>
<p>average. An alternative approach is to minimize instead:
</p>
<p>M&sum;
</p>
<p>j=1
</p>
<p>min
Aj
</p>
<p>∥∥LjA&lowast;j &minus;V
∥∥2 .
</p>
<p>Derive the solution to this problem. How do the criteria differ? Use the
SVD of the Lj to simplify the comparison of the two approaches.</p>
<p/>
</div>
<div class="page"><p/>
<p>Exercises 483
</p>
<p>Ex. 13.7 Consider the application of nearest-neighbors to the &ldquo;easy&rdquo; and
&ldquo;hard&rdquo; problems in the left panel of Figure 13.5.
</p>
<p>1. Replicate the results in the left panel of Figure 13.5.
</p>
<p>2. Estimate the misclassification errors using fivefold cross-validation,
and compare the error rate curves to those in 1.
</p>
<p>3. Consider an &ldquo;AIC-like&rdquo; penalization of the training set misclassifica-
tion error. Specifically, add 2t/N to the training set misclassification
error, where t is the approximate number of parameters N/r, r be-
ing the number of nearest-neighbors. Compare plots of the resulting
penalized misclassification error to those in 1 and 2. Which method
gives a better estimate of the optimal number of nearest-neighbors:
cross-validation or AIC?
</p>
<p>Ex. 13.8 Generate data in two classes, with two features. These features
are all independent Gaussian variates with standard deviation 1. Their
mean vectors are (&minus;1,&minus;1) in class 1 and (1, 1) in class 2. To each feature
vector apply a random rotation of angle θ, θ chosen uniformly from 0 to
2π. Generate 50 observations from each class to form the training set, and
500 in each class as the test set. Apply four different classifiers:
</p>
<p>1. Nearest-neighbors.
</p>
<p>2. Nearest-neighbors with hints: ten randomly rotated versions of each
data point are added to the training set before applying nearest-
neighbors.
</p>
<p>3. Invariant metric nearest-neighbors, using Euclidean distance invari-
ant to rotations about the origin.
</p>
<p>4. Tangent distance nearest-neighbors.
</p>
<p>In each case choose the number of neighbors by tenfold cross-validation.
Compare the results.</p>
<p/>
</div>
<div class="page"><p/>
<p>14
Unsupervised Learning
</p>
<p>14.1 Introduction
</p>
<p>The previous chapters have been concerned with predicting the values
of one or more outputs or response variables Y = (Y1, . . . , Ym) for a
given set of input or predictor variables XT = (X1, . . . , Xp). Denote by
xTi = (xi1, . . . , xip) the inputs for the ith training case, and let yi be a
response measurement. The predictions are based on the training sample
(x1, y1), . . . , (xN , yN ) of previously solved cases, where the joint values of
all of the variables are known. This is called supervised learning or &ldquo;learn-
ing with a teacher.&rdquo; Under this metaphor the &ldquo;student&rdquo; presents an an-
swer ŷi for each xi in the training sample, and the supervisor or &ldquo;teacher&rdquo;
provides either the correct answer and/or an error associated with the stu-
dent&rsquo;s answer. This is usually characterized by some loss function L(y, ŷ),
for example, L(y, ŷ) = (y &minus; ŷ)2.
If one supposes that (X,Y ) are random variables represented by some
</p>
<p>joint probability density Pr(X,Y ), then supervised learning can be formally
characterized as a density estimation problem where one is concerned with
determining properties of the conditional density Pr(Y |X). Usually the
properties of interest are the &ldquo;location&rdquo; parameters μ that minimize the
expected error at each x,
</p>
<p>μ(x) = argmin
θ
</p>
<p>EY |XL(Y, θ). (14.1)
</p>
<p> &copy; Springer Science+Business Media, LLC 2009 
</p>
<p>T. Hastie et al., The Elements of Statistical Learning, Second Edition, 485
</p>
<p>DOI: 10.1007/b94608_14,</p>
<p/>
</div>
<div class="page"><p/>
<p>486 14. Unsupervised Learning
</p>
<p>Conditioning one has
</p>
<p>Pr(X,Y ) = Pr(Y |X) &middot; Pr(X),
</p>
<p>where Pr(X) is the joint marginal density of the X values alone. In su-
pervised learning Pr(X) is typically of no direct concern. One is interested
mainly in the properties of the conditional density Pr(Y |X). Since Y is of-
ten of low dimension (usually one), and only its location μ(x) is of interest,
the problem is greatly simplified. As discussed in the previous chapters,
there are many approaches for successfully addressing supervised learning
in a variety of contexts.
In this chapter we address unsupervised learning or &ldquo;learning without a
</p>
<p>teacher.&rdquo; In this case one has a set of N observations (x1, x2, . . . , xN ) of a
random p-vector X having joint density Pr(X). The goal is to directly infer
the properties of this probability density without the help of a supervisor or
teacher providing correct answers or degree-of-error for each observation.
The dimension of X is sometimes much higher than in supervised learn-
ing, and the properties of interest are often more complicated than simple
location estimates. These factors are somewhat mitigated by the fact that
X represents all of the variables under consideration; one is not required
to infer how the properties of Pr(X) change, conditioned on the changing
values of another set of variables.
In low-dimensional problems (say p &le; 3), there are a variety of effective
</p>
<p>nonparametric methods for directly estimating the density Pr(X) itself at
all X-values, and representing it graphically (Silverman, 1986, e.g.). Owing
to the curse of dimensionality, these methods fail in high dimensions. One
must settle for estimating rather crude global models, such as Gaussian
mixtures or various simple descriptive statistics that characterize Pr(X).
Generally, these descriptive statistics attempt to characterize X-values,
</p>
<p>or collections of such values, where Pr(X) is relatively large. Principal
components, multidimensional scaling, self-organizing maps, and principal
curves, for example, attempt to identify low-dimensional manifolds within
the X-space that represent high data density. This provides information
about the associations among the variables and whether or not they can be
considered as functions of a smaller set of &ldquo;latent&rdquo; variables. Cluster anal-
ysis attempts to find multiple convex regions of the X-space that contain
modes of Pr(X). This can tell whether or not Pr(X) can be represented by
a mixture of simpler densities representing distinct types or classes of ob-
servations. Mixture modeling has a similar goal. Association rules attempt
to construct simple descriptions (conjunctive rules) that describe regions
of high density in the special case of very high dimensional binary-valued
data.
With supervised learning there is a clear measure of success, or lack
</p>
<p>thereof, that can be used to judge adequacy in particular situations and
to compare the effectiveness of different methods over various situations.</p>
<p/>
</div>
<div class="page"><p/>
<p>14.2 Association Rules 487
</p>
<p>Lack of success is directly measured by expected loss over the joint dis-
tribution Pr(X,Y ). This can be estimated in a variety of ways including
cross-validation. In the context of unsupervised learning, there is no such
direct measure of success. It is difficult to ascertain the validity of inferences
drawn from the output of most unsupervised learning algorithms. One must
resort to heuristic arguments not only for motivating the algorithms, as is
often the case in supervised learning as well, but also for judgments as to
the quality of the results. This uncomfortable situation has led to heavy
proliferation of proposed methods, since effectiveness is a matter of opinion
and cannot be verified directly.
In this chapter we present those unsupervised learning techniques that
</p>
<p>are among the most commonly used in practice, and additionally, a few
others that are favored by the authors.
</p>
<p>14.2 Association Rules
</p>
<p>Association rule analysis has emerged as a popular tool for mining com-
mercial data bases. The goal is to find joint values of the variables X =
(X1, X2, . . . , Xp) that appear most frequently in the data base. It is most
often applied to binary-valued data Xj &isin; {0, 1}, where it is referred to as
&ldquo;market basket&rdquo; analysis. In this context the observations are sales trans-
actions, such as those occurring at the checkout counter of a store. The
variables represent all of the items sold in the store. For observation i, each
variable Xj is assigned one of two values; xij = 1 if the jth item is pur-
chased as part of the transaction, whereas xij = 0 if it was not purchased.
Those variables that frequently have joint values of one represent items that
are frequently purchased together. This information can be quite useful for
stocking shelves, cross-marketing in sales promotions, catalog design, and
consumer segmentation based on buying patterns.
More generally, the basic goal of association rule analysis is to find a
</p>
<p>collection of prototype X-values v1, . . . , vL for the feature vector X, such
that the probability density Pr(vl) evaluated at each of those values is rela-
tively large. In this general framework, the problem can be viewed as &ldquo;mode
finding&rdquo; or &ldquo;bump hunting.&rdquo; As formulated, this problem is impossibly dif-
ficult. A natural estimator for each Pr(vl) is the fraction of observations
for which X = vl. For problems that involve more than a small number
of variables, each of which can assume more than a small number of val-
ues, the number of observations for which X = vl will nearly always be too
small for reliable estimation. In order to have a tractable problem, both the
goals of the analysis and the generality of the data to which it is applied
must be greatly simplified.
The first simplification modifies the goal. Instead of seeking values x
</p>
<p>where Pr(x) is large, one seeks regions of the X-space with high probability</p>
<p/>
</div>
<div class="page"><p/>
<p>488 14. Unsupervised Learning
</p>
<p>content relative to their size or support. Let Sj represent the set of all
possible values of the jth variable (its support), and let sj &sube; Sj be a subset
of these values. The modified goal can be stated as attempting to find
subsets of variable values s1, . . . , sp such that the probability of each of the
variables simultaneously assuming a value within its respective subset,
</p>
<p>Pr
</p>
<p>⎡
⎣
</p>
<p>p⋂
</p>
<p>j=1
</p>
<p>(Xj &isin; sj)
</p>
<p>⎤
⎦ , (14.2)
</p>
<p>is relatively large. The intersection of subsets &cap;pj=1(Xj &isin; sj) is called a
conjunctive rule. For quantitative variables the subsets sj are contiguous
intervals; for categorical variables the subsets are delineated explicitly. Note
that if the subset sj is in fact the entire set of values sj = Sj , as is often
the case, the variable Xj is said not to appear in the rule (14.2).
</p>
<p>14.2.1 Market Basket Analysis
</p>
<p>General approaches to solving (14.2) are discussed in Section 14.2.5. These
can be quite useful in many applications. However, they are not feasible
for the very large (p &asymp; 104, N &asymp; 108) commercial data bases to which
market basket analysis is often applied. Several further simplifications of
(14.2) are required. First, only two types of subsets are considered; either
sj consists of a single value of Xj , sj = v0j , or it consists of the entire set
of values that Xj can assume, sj = Sj . This simplifies the problem (14.2)
to finding subsets of the integers J &sub; {1, . . . , p}, and corresponding values
v0j , j &isin; J , such that
</p>
<p>Pr
</p>
<p>⎡
⎣⋂
</p>
<p>j&isin;J
</p>
<p>(Xj = v0j)
</p>
<p>⎤
⎦ (14.3)
</p>
<p>is large. Figure 14.1 illustrates this assumption.
One can apply the technique of dummy variables to turn (14.3) into
</p>
<p>a problem involving only binary-valued variables. Here we assume that
the support Sj is finite for each variable Xj . Specifically, a new set of
variables Z1, . . . , ZK is created, one such variable for each of the values
vlj attainable by each of the original variables X1, . . . , Xp. The number of
dummy variables K is
</p>
<p>K =
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>|Sj |,
</p>
<p>where |Sj | is the number of distinct values attainable by Xj . Each dummy
variable is assigned the value Zk = 1 if the variable with which it is as-
sociated takes on the corresponding value to which Zk is assigned, and
Zk = 0 otherwise. This transforms (14.3) to finding a subset of the integers
K &sub; {1, . . . ,K} such that</p>
<p/>
</div>
<div class="page"><p/>
<p>14.2 Association Rules 489
</p>
<p>X1X1X1
</p>
<p>X
2
</p>
<p>X
2
</p>
<p>X
2
</p>
<p>FIGURE 14.1. Simplifications for association rules. Here there are two inputs
X1 and X2, taking four and six distinct values, respectively. The red squares
indicate areas of high density. To simplify the computations, we assume that the
derived subset corresponds to either a single value of an input or all values. With
this assumption we could find either the middle or right pattern, but not the left
one.
</p>
<p>Pr
</p>
<p>[⋂
</p>
<p>k&isin;K
</p>
<p>(Zk = 1)
</p>
<p>]
= Pr
</p>
<p>[&prod;
</p>
<p>k&isin;K
</p>
<p>Zk = 1
</p>
<p>]
(14.4)
</p>
<p>is large. This is the standard formulation of the market basket problem.
The set K is called an &ldquo;item set.&rdquo; The number of variables Zk in the item
set is called its &ldquo;size&rdquo; (note that the size is no bigger than p). The estimated
value of (14.4) is taken to be the fraction of observations in the data base
for which the conjunction in (14.4) is true:
</p>
<p>P̂r
</p>
<p>[&prod;
</p>
<p>k&isin;K
</p>
<p>(Zk = 1)
</p>
<p>]
=
</p>
<p>1
</p>
<p>N
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>&prod;
</p>
<p>k&isin;K
</p>
<p>zik. (14.5)
</p>
<p>Here zik is the value of Zk for this ith case. This is called the &ldquo;support&rdquo; or
&ldquo;prevalence&rdquo; T (K) of the item set K. An observation i for which&prod;k&isin;K zik =
1 is said to &ldquo;contain&rdquo; the item set K.
In association rule mining a lower support bound t is specified, and one
</p>
<p>seeks all item sets Kl that can be formed from the variables Z1, . . . , ZK
with support in the data base greater than this lower bound t
</p>
<p>{Kl|T (Kl) &gt; t}. (14.6)
</p>
<p>14.2.2 The Apriori Algorithm
</p>
<p>The solution to this problem (14.6) can be obtained with feasible compu-
tation for very large data bases provided the threshold t is adjusted so that
(14.6) consists of only a small fraction of all 2K possible item sets. The
&ldquo;Apriori&rdquo; algorithm (Agrawal et al., 1995) exploits several aspects of the</p>
<p/>
</div>
<div class="page"><p/>
<p>490 14. Unsupervised Learning
</p>
<p>curse of dimensionality to solve (14.6) with a small number of passes over
the data. Specifically, for a given support threshold t:
</p>
<p>&bull; The cardinality |{K|T (K) &gt; t}| is relatively small.
&bull; Any item set L consisting of a subset of the items in K must have
support greater than or equal to that of K, L &sube; K &rArr; T (L) &ge; T (K).
</p>
<p>The first pass over the data computes the support of all single-item sets.
Those whose support is less than the threshold are discarded. The second
pass computes the support of all item sets of size two that can be formed
from pairs of the single items surviving the first pass. In other words, to
generate all frequent itemsets with |K| = m, we need to consider only
candidates such that all of their m ancestral item sets of size m &minus; 1 are
frequent. Those size-two item sets with support less than the threshold are
discarded. Each successive pass over the data considers only those item
sets that can be formed by combining those that survived the previous
pass with those retained from the first pass. Passes over the data continue
until all candidate rules from the previous pass have support less than the
specified threshold. The Apriori algorithm requires only one pass over the
data for each value of |K|, which is crucial since we assume the data cannot
be fitted into a computer&rsquo;s main memory. If the data are sufficiently sparse
(or if the threshold t is high enough), then the process will terminate in
reasonable time even for huge data sets.
There are many additional tricks that can be used as part of this strat-
</p>
<p>egy to increase speed and convergence (Agrawal et al., 1995). The Apriori
algorithm represents one of the major advances in data mining technology.
Each high support item set K (14.6) returned by the Apriori algorithm is
</p>
<p>cast into a set of &ldquo;association rules.&rdquo; The items Zk, k &isin; K, are partitioned
into two disjoint subsets, A &cup;B = K, and written
</p>
<p>A &rArr; B. (14.7)
The first item subset A is called the &ldquo;antecedent&rdquo; and the second B the
&ldquo;consequent.&rdquo; Association rules are defined to have several properties based
on the prevalence of the antecedent and consequent item sets in the data
base. The &ldquo;support&rdquo; of the rule T (A &rArr; B) is the fraction of observations
in the union of the antecedent and consequent, which is just the support
of the item set K from which they were derived. It can be viewed as an
estimate (14.5) of the probability of simultaneously observing both item
sets Pr(A and B) in a randomly selected market basket. The &ldquo;confidence&rdquo;
or &ldquo;predictability&rdquo; C(A &rArr; B) of the rule is its support divided by the
support of the antecedent
</p>
<p>C(A &rArr; B) = T (A &rArr; B)
T (A)
</p>
<p>, (14.8)
</p>
<p>which can be viewed as an estimate of Pr(B |A). The notation Pr(A), the
probability of an item set A occurring in a basket, is an abbreviation for</p>
<p/>
</div>
<div class="page"><p/>
<p>14.2 Association Rules 491
</p>
<p>Pr(
&prod;
</p>
<p>k&isin;A Zk = 1). The &ldquo;expected confidence&rdquo; is defined as the support of
the consequent T (B), which is an estimate of the unconditional probability
Pr(B). Finally, the &ldquo;lift&rdquo; of the rule is defined as the confidence divided by
the expected confidence
</p>
<p>L(A &rArr; B) = C(A &rArr; B)
T (B)
</p>
<p>.
</p>
<p>This is an estimate of the association measure Pr(A and B)/Pr(A)Pr(B).
As an example, suppose the item setK = {peanut butter, jelly, bread}
</p>
<p>and consider the rule {peanut butter, jelly} &rArr; {bread}. A support value
of 0.03 for this rule means that peanut butter, jelly, and bread appeared
together in 3% of the market baskets. A confidence of 0.82 for this rule im-
plies that when peanut butter and jelly were purchased, 82% of the time
bread was also purchased. If bread appeared in 43% of all market baskets
then the rule {peanut butter, jelly} &rArr; {bread} would have a lift of 1.95.
The goal of this analysis is to produce association rules (14.7) with both
</p>
<p>high values of support and confidence (14.8). The Apriori algorithm returns
all item sets with high support as defined by the support threshold t (14.6).
A confidence threshold c is set, and all rules that can be formed from those
item sets (14.6) with confidence greater than this value
</p>
<p>{A &rArr; B |C(A &rArr; B) &gt; c} (14.9)
</p>
<p>are reported. For each item set K of size |K| there are 2|K|&minus;1 &minus; 1 rules of
the form A &rArr; (K &minus; A), A &sub; K. Agrawal et al. (1995) present a variant of
the Apriori algorithm that can rapidly determine which rules survive the
confidence threshold (14.9) from all possible rules that can be formed from
the solution item sets (14.6).
The output of the entire analysis is a collection of association rules (14.7)
</p>
<p>that satisfy the constraints
</p>
<p>T (A &rArr; B) &gt; t and C(A &rArr; B) &gt; c.
</p>
<p>These are generally stored in a data base that can be queried by the user.
Typical requests might be to display the rules in sorted order of confidence,
lift or support. More specifically, one might request such a list conditioned
on particular items in the antecedent or especially the consequent. For
example, a request might be the following:
</p>
<p>Display all transactions in which ice skates are the consequent
that have confidence over 80% and support of more than 2%.
</p>
<p>This could provide information on those items (antecedent) that predicate
sales of ice skates. Focusing on a particular consequent casts the problem
into the framework of supervised learning.
Association rules have become a popular tool for analyzing very large
</p>
<p>commercial data bases in settings where market basket is relevant. That is</p>
<p/>
</div>
<div class="page"><p/>
<p>492 14. Unsupervised Learning
</p>
<p>when the data can be cast in the form of a multidimensional contingency
table. The output is in the form of conjunctive rules (14.4) that are easily
understood and interpreted. The Apriori algorithm allows this analysis to
be applied to huge data bases, much larger that are amenable to other types
of analyses. Association rules are among data mining&rsquo;s biggest successes.
Besides the restrictive form of the data to which they can be applied, as-
</p>
<p>sociation rules have other limitations. Critical to computational feasibility
is the support threshold (14.6). The number of solution item sets, their size,
and the number of passes required over the data can grow exponentially
with decreasing size of this lower bound. Thus, rules with high confidence
or lift, but low support, will not be discovered. For example, a high confi-
dence rule such as vodka &rArr; caviar will not be uncovered owing to the low
sales volume of the consequent caviar.
</p>
<p>14.2.3 Example: Market Basket Analysis
</p>
<p>We illustrate the use of Apriori on a moderately sized demographics data
base. This data set consists of N = 9409 questionnaires filled out by shop-
ping mall customers in the San Francisco Bay Area (Impact Resources, Inc.,
Columbus OH, 1987). Here we use answers to the first 14 questions, relat-
ing to demographics, for illustration. These questions are listed in Table
14.1. The data are seen to consist of a mixture of ordinal and (unordered)
categorical variables, many of the latter having more than a few values.
There are many missing values.
We used a freeware implementation of the Apriori algorithm due to Chris-
</p>
<p>tian Borgelt1. After removing observations with missing values, each ordinal
predictor was cut at its median and coded by two dummy variables; each
categorical predictor with k categories was coded by k dummy variables.
This resulted in a 6876 &times; 50 matrix of 6876 observations on 50 dummy
variables.
The algorithm found a total of 6288 association rules, involving &le; 5
</p>
<p>predictors, with support of at least 10%. Understanding this large set of
rules is itself a challenging data analysis task. We will not attempt this here,
but only illustrate in Figure 14.2 the relative frequency of each dummy
variable in the data (top) and the association rules (bottom). Prevalent
categories tend to appear more often in the rules, for example, the first
category in language (English). However, others such as occupation are
under-represented, with the exception of the first and fifth level.
Here are three examples of association rules found by the Apriori algo-
</p>
<p>rithm:
</p>
<p>Association rule 1: Support 25%, confidence 99.7% and lift 1.03.
</p>
<p>1See http://fuzzy.cs.uni-magdeburg.de/&sim;borgelt.</p>
<p/>
</div>
<div class="page"><p/>
<p>14.2 Association Rules 493
</p>
<p>0 10 20 30 40 50
</p>
<p>0
.0
</p>
<p>0
.0
</p>
<p>2
0
</p>
<p>.0
4
</p>
<p>0
.0
</p>
<p>6
</p>
<p>Attribute
</p>
<p>R
e
</p>
<p>la
ti
v
e
</p>
<p> F
re
</p>
<p>q
u
</p>
<p>e
n
</p>
<p>c
y
 i
n
</p>
<p> D
a
</p>
<p>ta
</p>
<p>in
c
o
</p>
<p>m
e
</p>
<p>s
e
</p>
<p>x
</p>
<p>m
a
</p>
<p>rs
ta
</p>
<p>t
</p>
<p>a
g
</p>
<p>e
</p>
<p>e
d
</p>
<p>u
c
</p>
<p>o
c
c
u
</p>
<p>p
</p>
<p>y
rs
</p>
<p>&minus;
b
</p>
<p>a
y
</p>
<p>d
u
</p>
<p>a
lin
</p>
<p>c
</p>
<p>p
e
</p>
<p>rh
o
</p>
<p>u
s
</p>
<p>p
e
</p>
<p>ry
o
</p>
<p>u
n
</p>
<p>g
</p>
<p>h
o
</p>
<p>u
s
e
</p>
<p>ty
p
</p>
<p>e
h
</p>
<p>o
m
</p>
<p>e
</p>
<p>e
th
</p>
<p>n
ic
</p>
<p>la
n
</p>
<p>g
u
</p>
<p>a
g
</p>
<p>e
</p>
<p>0 10 20 30 40 50
</p>
<p>0
.0
</p>
<p>0
.0
</p>
<p>4
0
</p>
<p>.0
8
</p>
<p>0
.1
</p>
<p>2
</p>
<p>Attribute
</p>
<p>R
e
</p>
<p>la
ti
v
e
</p>
<p> F
re
</p>
<p>q
u
</p>
<p>e
n
</p>
<p>c
y
 i
n
</p>
<p> A
s
s
o
</p>
<p>c
ia
</p>
<p>ti
o
</p>
<p>n
 R
</p>
<p>u
le
</p>
<p>s
</p>
<p>in
c
o
</p>
<p>m
e
</p>
<p>s
e
</p>
<p>x
</p>
<p>m
a
</p>
<p>rs
ta
</p>
<p>t
</p>
<p>a
g
</p>
<p>e
</p>
<p>e
d
</p>
<p>u
c
</p>
<p>o
c
c
u
</p>
<p>p
</p>
<p>y
rs
</p>
<p>&minus;
b
</p>
<p>a
y
</p>
<p>d
u
</p>
<p>a
lin
</p>
<p>c
</p>
<p>p
e
</p>
<p>rh
o
</p>
<p>u
s
</p>
<p>p
e
</p>
<p>ry
o
</p>
<p>u
n
</p>
<p>g
</p>
<p>h
o
</p>
<p>u
s
e
</p>
<p>ty
p
</p>
<p>e
h
</p>
<p>o
m
</p>
<p>e
</p>
<p>e
th
</p>
<p>n
ic
</p>
<p>la
n
</p>
<p>g
u
</p>
<p>a
g
</p>
<p>e
</p>
<p>FIGURE 14.2. Market basket analysis: relative frequency of each dummy vari-
able (coding an input category) in the data (top), and the association rules found
by the Apriori algorithm (bottom).</p>
<p/>
</div>
<div class="page"><p/>
<p>494 14. Unsupervised Learning
</p>
<p>TABLE 14.1. Inputs for the demographic data.
</p>
<p>Feature Demographic # Values Type
</p>
<p>1 Sex 2 Categorical
2 Marital status 5 Categorical
3 Age 7 Ordinal
4 Education 6 Ordinal
5 Occupation 9 Categorical
6 Income 9 Ordinal
7 Years in Bay Area 5 Ordinal
8 Dual incomes 3 Categorical
9 Number in household 9 Ordinal
10 Number of children 9 Ordinal
11 Householder status 3 Categorical
12 Type of home 5 Categorical
13 Ethnic classification 8 Categorical
14 Language in home 3 Categorical
</p>
<p>[
number in household = 1
number of children = 0
</p>
<p>]
</p>
<p>&dArr;
language in home = English
</p>
<p>Association rule 2: Support 13.4%, confidence 80.8%, and lift 2.13.
</p>
<p>⎡
⎣
</p>
<p>language in home = English
householder status = own
</p>
<p>occupation = {professional/managerial}
</p>
<p>⎤
⎦
</p>
<p>&dArr;
income &ge; $40,000
</p>
<p>Association rule 3: Support 26.5%, confidence 82.8% and lift 2.15.
</p>
<p>⎡
⎢⎢⎣
</p>
<p>language in home = English
income &lt; $40,000
</p>
<p>marital status = not married
number of children = 0
</p>
<p>⎤
⎥⎥⎦
</p>
<p>&dArr;
education /&isin; {college graduate, graduate study}</p>
<p/>
</div>
<div class="page"><p/>
<p>14.2 Association Rules 495
</p>
<p>We chose the first and third rules based on their high support. The second
rule is an association rule with a high-income consequent, and could be
used to try to target high-income individuals.
As stated above, we created dummy variables for each category of the
</p>
<p>input predictors, for example, Z1 = I(income &lt; $40, 000) and Z2 =
I(income &ge; $40, 000) for below and above the median income. If we were
interested only in finding associations with the high-income category, we
would include Z2 but not Z1. This is often the case in actual market basket
problems, where we are interested in finding associations with the presence
of a relatively rare item, but not associations with its absence.
</p>
<p>14.2.4 Unsupervised as Supervised Learning
</p>
<p>Here we discuss a technique for transforming the density estimation prob-
lem into one of supervised function approximation. This forms the basis
for the generalized association rules described in the next section.
Let g(x) be the unknown data probability density to be estimated, and
</p>
<p>g0(x) be a specified probability density function used for reference. For ex-
ample, g0(x) might be the uniform density over the range of the variables.
Other possibilities are discussed below. The data set x1, x2, . . . , xN is pre-
sumed to be an i.i.d. random sample drawn from g(x). A sample of size N0
can be drawn from g0(x) using Monte Carlo methods. Pooling these two
data sets, and assigning mass w = N0/(N +N0) to those drawn from g(x),
and w0 = N/(N + N0) to those drawn from g0(x), results in a random
sample drawn from the mixture density (g(x) + g0(x)) /2. If one assigns
the value Y = 1 to each sample point drawn from g(x) and Y = 0 those
drawn from g0(x), then
</p>
<p>μ(x) = E(Y |x) = g(x)
g(x) + g0(x)
</p>
<p>=
g(x)/g0(x)
</p>
<p>1 + g(x)/g0(x)
(14.10)
</p>
<p>can be estimated by supervised learning using the combined sample
</p>
<p>(y1, x1), (y2, x2), . . . , (yN+N0 , xN+N0) (14.11)
</p>
<p>as training data. The resulting estimate μ̂(x) can be inverted to provide an
estimate for g(x)
</p>
<p>ĝ(x) = g0(x)
μ̂(x)
</p>
<p>1&minus; μ̂(x) . (14.12)
</p>
<p>Generalized versions of logistic regression (Section 4.4) are especially well
suited for this application since the log-odds,
</p>
<p>f(x) = log
g(x)
</p>
<p>g0(x)
, (14.13)
</p>
<p>are estimated directly. In this case one has</p>
<p/>
</div>
<div class="page"><p/>
<p>496 14. Unsupervised Learning
</p>
<p>-1 0 1 2
</p>
<p>-2
0
</p>
<p>2
4
</p>
<p>6
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
&bull;
</p>
<p>&bull;&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull; &bull; &bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull; &bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull; &bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
&bull;
</p>
<p>&bull; &bull;&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;&bull; &bull; &bull;
&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;&bull;
&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>-1 0 1 2
</p>
<p>-2
0
</p>
<p>2
4
</p>
<p>6
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
&bull;
</p>
<p>&bull;&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull; &bull; &bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull; &bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull; &bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
&bull;
</p>
<p>&bull; &bull;&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;&bull; &bull; &bull;
&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;&bull;
&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>X1X1
X
</p>
<p>2
</p>
<p>X
2
</p>
<p>FIGURE 14.3. Density estimation via classification. (Left panel:) Training set
of 200 data points. (Right panel:) Training set plus 200 reference data points,
generated uniformly over the rectangle containing the training data. The training
sample was labeled as class 1, and the reference sample class 0, and a semipara-
metric logistic regression model was fit to the data. Some contours for ĝ(x) are
shown.
</p>
<p>ĝ(x) = g0(x) e
f̂(x). (14.14)
</p>
<p>An example is shown in Figure 14.3. We generated a training set of size
200 shown in the left panel. The right panel shows the reference data (blue)
generated uniformly over the rectangle containing the training data. The
training sample was labeled as class 1, and the reference sample class 0,
and a logistic regression model, using a tensor product of natural splines
(Section 5.2.1), was fit to the data. Some probability contours of μ̂(x) are
shown in the right panel; these are also the contours of the density estimate
ĝ(x), since ĝ(x) = μ̂(x)/(1 &minus; μ̂(x)), is a monotone function. The contours
roughly capture the data density.
In principle any reference density can be used for g0(x) in (14.14). In
</p>
<p>practice the accuracy of the estimate ĝ(x) can depend greatly on partic-
ular choices. Good choices will depend on the data density g(x) and the
procedure used to estimate (14.10) or (14.13). If accuracy is the goal, g0(x)
should be chosen so that the resulting functions μ(x) or f(x) are approx-
imated easily by the method being used. However, accuracy is not always
the primary goal. Both μ(x) and f(x) are monotonic functions of the den-
sity ratio g(x)/g0(x). They can thus be viewed as &ldquo;contrast&rdquo; statistics that
provide information concerning departures of the data density g(x) from
the chosen reference density g0(x). Therefore, in data analytic settings, a
choice for g0(x) is dictated by types of departures that are deemed most
interesting in the context of the specific problem at hand. For example, if
departures from uniformity are of interest, g0(x) might be the a uniform
density over the range of the variables. If departures from joint normality</p>
<p/>
</div>
<div class="page"><p/>
<p>14.2 Association Rules 497
</p>
<p>are of interest, a good choice for g0(x) would be a Gaussian distribution
with the same mean vector and covariance matrix as the data. Departures
from independence could be investigated by using
</p>
<p>g0(x) =
</p>
<p>p&prod;
</p>
<p>j=1
</p>
<p>gj(xj), (14.15)
</p>
<p>where gj(xj) is the marginal data density of Xj , the jth coordinate of X.
A sample from this independent density (14.15) is easily generated from the
data itself by applying a different random permutation to the data values
of each of the variables.
As discussed above, unsupervised learning is concerned with revealing
</p>
<p>properties of the data density g(x). Each technique focuses on a particu-
lar property or set of properties. Although this approach of transforming
the problem to one of supervised learning (14.10)&ndash;(14.14) seems to have
been part of the statistics folklore for some time, it does not appear to
have had much impact despite its potential to bring well-developed su-
pervised learning methodology to bear on unsupervised learning problems.
One reason may be that the problem must be enlarged with a simulated
data set generated by Monte Carlo techniques. Since the size of this data
set should be at least as large as the data sample N0 &ge; N , the compu-
tation and memory requirements of the estimation procedure are at least
doubled. Also, substantial computation may be required to generate the
Monte Carlo sample itself. Although perhaps a deterrent in the past, these
increased computational requirements are becoming much less of a burden
as increased resources become routinely available. We illustrate the use of
supervised learning methods for unsupervised learning in the next section.
</p>
<p>14.2.5 Generalized Association Rules
</p>
<p>The more general problem (14.2) of finding high-density regions in the data
space can be addressed using the supervised learning approach described
above. Although not applicable to the huge data bases for which market
basket analysis is feasible, useful information can be obtained from mod-
erately sized data sets. The problem (14.2) can be formulated as finding
subsets of the integers J &sub; {1, 2, . . . , p} and corresponding value subsets
sj , j &isin; J for the corresponding variables Xj , such that
</p>
<p>P̂r
</p>
<p>⎛
⎝⋂
</p>
<p>j&isin;J
</p>
<p>(Xj &isin; sj)
</p>
<p>⎞
⎠ = 1
</p>
<p>N
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>I
</p>
<p>⎛
⎝⋂
</p>
<p>j&isin;J
</p>
<p>(xij &isin; sj)
</p>
<p>⎞
⎠ (14.16)
</p>
<p>is large. Following the nomenclature of association rule analysis, {(Xj &isin;
sj)}j&isin;J will be called a &ldquo;generalized&rdquo; item set. The subsets sj correspond-
ing to quantitative variables are taken to be contiguous intervals within</p>
<p/>
</div>
<div class="page"><p/>
<p>498 14. Unsupervised Learning
</p>
<p>their range of values, and subsets for categorical variables can involve more
than a single value. The ambitious nature of this formulation precludes a
thorough search for all generalized item sets with support (14.16) greater
than a specified minimum threshold, as was possible in the more restric-
tive setting of market basket analysis. Heuristic search methods must be
employed, and the most one can hope for is to find a useful collection of
such generalized item sets.
Both market basket analysis (14.5) and the generalized formulation (14.16)
</p>
<p>implicitly reference the uniform probability distribution. One seeks item
sets that are more frequent than would be expected if all joint data values
(x1, x2, . . . , xN ) were uniformly distributed. This favors the discovery of
item sets whose marginal constituents (Xj &isin; sj) are individually frequent,
that is, the quantity
</p>
<p>1
</p>
<p>N
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>I(xij &isin; sj) (14.17)
</p>
<p>is large. Conjunctions of frequent subsets (14.17) will tend to appear more
often among item sets of high support (14.16) than conjunctions of margin-
ally less frequent subsets. This is why the rule vodka &rArr; caviar is not likely
to be discovered in spite of a high association (lift); neither item has high
marginal support, so that their joint support is especially small. Reference
to the uniform distribution can cause highly frequent item sets with low
associations among their constituents to dominate the collection of highest
support item sets.
Highly frequent subsets sj are formed as disjunctions of the most fre-
</p>
<p>quent Xj-values. Using the product of the variable marginal data densities
(14.15) as a reference distribution removes the preference for highly fre-
quent values of the individual variables in the discovered item sets. This is
because the density ratio g(x)/g0(x) is uniform if there are no associations
among the variables (complete independence), regardless of the frequency
distribution of the individual variable values. Rules like vodka &rArr; caviar
would have a chance to emerge. It is not clear however, how to incorporate
reference distributions other than the uniform into the Apriori algorithm.
As explained in Section 14.2.4, it is straightforward to generate a sample
from the product density (14.15), given the original data set.
After choosing a reference distribution, and drawing a sample from it
</p>
<p>as in (14.11), one has a supervised learning problem with a binary-valued
output variable Y &isin; {0, 1}. The goal is to use this training data to find
regions
</p>
<p>R =
⋂
</p>
<p>j&isin;J
</p>
<p>(Xj &isin; sj) (14.18)
</p>
<p>for which the target function μ(x) = E(Y |x) is relatively large. In addition,
one might wish to require that the data support of these regions</p>
<p/>
</div>
<div class="page"><p/>
<p>14.2 Association Rules 499
</p>
<p>T (R) =
</p>
<p>&int;
</p>
<p>x&isin;R
</p>
<p>g(x) dx (14.19)
</p>
<p>not be too small.
</p>
<p>14.2.6 Choice of Supervised Learning Method
</p>
<p>The regions (14.18) are defined by conjunctive rules. Hence supervised
methods that learn such rules would be most appropriate in this context.
The terminal nodes of a CART decision tree are defined by rules precisely
of the form (14.18). Applying CART to the pooled data (14.11) will pro-
duce a decision tree that attempts to model the target (14.10) over the
entire data space by a disjoint set of regions (terminal nodes). Each region
is defined by a rule of the form (14.18). Those terminal nodes t with high
average y-values
</p>
<p>ȳt = ave(yi |xi &isin; t)
are candidates for high-support generalized item sets (14.16). The actual
(data) support is given by
</p>
<p>T (R) = ȳt &middot;
Nt
</p>
<p>N +N0
,
</p>
<p>where Nt is the number of (pooled) observations within the region repre-
sented by the terminal node. By examining the resulting decision tree, one
might discover interesting generalized item sets of relatively high-support.
These can then be partitioned into antecedents and consequents in a search
for generalized association rules of high confidence and/or lift.
Another natural learning method for this purpose is the patient rule
</p>
<p>induction method PRIM described in Section 9.3. PRIM also produces
rules precisely of the form (14.18), but it is especially designed for finding
high-support regions that maximize the average target (14.10) value within
them, rather than trying to model the target function over the entire data
space. It also provides more control over the support/average-target-value
tradeoff.
Exercise 14.3 addresses an issue that arises with either of these methods
</p>
<p>when we generate random data from the product of the marginal distribu-
tions.
</p>
<p>14.2.7 Example: Market Basket Analysis (Continued)
</p>
<p>We illustrate the use of PRIM on the demographics data of Table 14.1.
Three of the high-support generalized item sets emerging from the PRIM
</p>
<p>analysis were the following:
</p>
<p>Item set 1: Support= 24%.</p>
<p/>
</div>
<div class="page"><p/>
<p>500 14. Unsupervised Learning
</p>
<p>⎡
⎣
</p>
<p>marital status = married
householder status = own
</p>
<p>type of home �= apartment
</p>
<p>⎤
⎦
</p>
<p>Item set 2: Support= 24%.
</p>
<p>⎡
⎢⎢⎣
</p>
<p>age &le; 24
marital status &isin; {living together-not married, single}
</p>
<p>occupation /&isin; {professional, homemaker, retired}
householder status &isin; {rent, live with family}
</p>
<p>⎤
⎥⎥⎦
</p>
<p>Item set 3: Support= 15%.
</p>
<p>⎡
⎢⎢⎢⎢⎢⎢⎣
</p>
<p>householder status = rent
type of home �= house
</p>
<p>number in household &le; 2
number of children = 0
</p>
<p>occupation /&isin; {homemaker, student, unemployed}
income &isin; [$20,000, $150,000]
</p>
<p>⎤
⎥⎥⎥⎥⎥⎥⎦
</p>
<p>Generalized association rules derived from these item sets with confidence
(14.8) greater than 95% are the following:
</p>
<p>Association rule 1: Support 25%, confidence 99.7% and lift 1.35.
</p>
<p>[
marital status = married
</p>
<p>householder status = own
</p>
<p>]
</p>
<p>&dArr;
type of home �= apartment
</p>
<p>Association rule 2: Support 25%, confidence 98.7% and lift 1.97.
</p>
<p>⎡
⎣
</p>
<p>age &le; 24
occupation /&isin; {professional, homemaker, retired}
</p>
<p>householder status &isin; {rent, live with family}
</p>
<p>⎤
⎦
</p>
<p>&dArr;
marital status &isin; {single, living together-not married}
</p>
<p>Association rule 3: Support 25%, confidence 95.9% and lift 2.61.
</p>
<p>[
householder status = own
</p>
<p>type of home �= apartment
</p>
<p>]
</p>
<p>&dArr;
marital status = married</p>
<p/>
</div>
<div class="page"><p/>
<p>14.3 Cluster Analysis 501
</p>
<p>Association rule 4: Support 15%, confidence 95.4% and lift 1.50.
</p>
<p>⎡
⎢⎢⎢⎢⎣
</p>
<p>householder status = rent
type of home �= house
</p>
<p>number in household &le; 2
occupation /&isin; {homemaker, student, unemployed}
</p>
<p>income &isin; [$20,000, $150,000]
</p>
<p>⎤
⎥⎥⎥⎥⎦
</p>
<p>&dArr;
number of children = 0
</p>
<p>There are no great surprises among these particular rules. For the most
part they verify intuition. In other contexts where there is less prior in-
formation available, unexpected results have a greater chance to emerge.
These results do illustrate the type of information generalized association
rules can provide, and that the supervised learning approach, coupled with
a ruled induction method such as CART or PRIM, can uncover item sets
exhibiting high associations among their constituents.
How do these generalized association rules compare to those found earlier
</p>
<p>by the Apriori algorithm? Since the Apriori procedure gives thousands of
rules, it is difficult to compare them. However some general points can be
made. The Apriori algorithm is exhaustive&mdash;it finds all rules with support
greater than a specified amount. In contrast, PRIM is a greedy algorithm
and is not guaranteed to give an &ldquo;optimal&rdquo; set of rules. On the other hand,
the Apriori algorithm can deal only with dummy variables and hence could
not find some of the above rules. For example, since type of home is a
categorical input, with a dummy variable for each level, Apriori could not
find a rule involving the set
</p>
<p>type of home �= apartment.
</p>
<p>To find this set, we would have to code a dummy variable for apartment
versus the other categories of type of home. It will not generally be feasible
to precode all such potentially interesting comparisons.
</p>
<p>14.3 Cluster Analysis
</p>
<p>Cluster analysis, also called data segmentation, has a variety of goals. All
relate to grouping or segmenting a collection of objects into subsets or
&ldquo;clusters,&rdquo; such that those within each cluster are more closely related to
one another than objects assigned to different clusters. An object can be
described by a set of measurements, or by its relation to other objects.
In addition, the goal is sometimes to arrange the clusters into a natural
hierarchy. This involves successively grouping the clusters themselves so</p>
<p/>
</div>
<div class="page"><p/>
<p>502 14. Unsupervised Learning
</p>
<p>&bull; &bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;&bull;
</p>
<p>&bull;&bull; &bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;&bull;
&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
&bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
&bull;
</p>
<p>&bull; &bull;&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull; &bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
&bull; &bull;
&bull;&bull; &bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;&bull;&bull; &bull;&bull;
</p>
<p>X1
</p>
<p>X
2
</p>
<p>FIGURE 14.4. Simulated data in the plane, clustered into three classes (repre-
sented by orange, blue and green) by the K-means clustering algorithm
</p>
<p>that at each level of the hierarchy, clusters within the same group are more
similar to each other than those in different groups.
Cluster analysis is also used to form descriptive statistics to ascertain
</p>
<p>whether or not the data consists of a set distinct subgroups, each group
representing objects with substantially different properties. This latter goal
requires an assessment of the degree of difference between the objects as-
signed to the respective clusters.
Central to all of the goals of cluster analysis is the notion of the degree of
</p>
<p>similarity (or dissimilarity) between the individual objects being clustered.
A clustering method attempts to group the objects based on the definition
of similarity supplied to it. This can only come from subject matter consid-
erations. The situation is somewhat similar to the specification of a loss or
cost function in prediction problems (supervised learning). There the cost
associated with an inaccurate prediction depends on considerations outside
the data.
Figure 14.4 shows some simulated data clustered into three groups via
</p>
<p>the popular K-means algorithm. In this case two of the clusters are not
well separated, so that &ldquo;segmentation&rdquo; more accurately describes the part
of this process than &ldquo;clustering.&rdquo; K-means clustering starts with guesses
for the three cluster centers. Then it alternates the following steps until
convergence:
</p>
<p>&bull; for each data point, the closest cluster center (in Euclidean distance)
is identified;</p>
<p/>
</div>
<div class="page"><p/>
<p>14.3 Cluster Analysis 503
</p>
<p>&bull; each cluster center is replaced by the coordinate-wise average of all
data points that are closest to it.
</p>
<p>We describe K-means clustering in more detail later, including the prob-
lem of how to choose the number of clusters (three in this example). K-
means clustering is a top-down procedure, while other cluster approaches
that we discuss are bottom-up. Fundamental to all clustering techniques is
the choice of distance or dissimilarity measure between two objects. We
first discuss distance measures before describing a variety of algorithms for
clustering.
</p>
<p>14.3.1 Proximity Matrices
</p>
<p>Sometimes the data is represented directly in terms of the proximity (alike-
ness or affinity) between pairs of objects. These can be either similarities or
dissimilarities (difference or lack of affinity). For example, in social science
experiments, participants are asked to judge by how much certain objects
differ from one another. Dissimilarities can then be computed by averaging
over the collection of such judgments. This type of data can be represented
by an N&times;N matrix D, where N is the number of objects, and each element
dii&prime; records the proximity between the ith and i
</p>
<p>&prime;th objects. This matrix is
then provided as input to the clustering algorithm.
Most algorithms presume a matrix of dissimilarities with nonnegative
</p>
<p>entries and zero diagonal elements: dii = 0, i = 1, 2, . . . , N. If the original
data were collected as similarities, a suitable monotone-decreasing function
can be used to convert them to dissimilarities. Also, most algorithms as-
sume symmetric dissimilarity matrices, so if the original matrix D is not
symmetric it must be replaced by (D+DT )/2. Subjectively judged dissimi-
larities are seldom distances in the strict sense, since the triangle inequality
dii&prime; &le; dik+di&prime;k, for all k &isin; {1, . . . , N} does not hold. Thus, some algorithms
that assume distances cannot be used with such data.
</p>
<p>14.3.2 Dissimilarities Based on Attributes
</p>
<p>Most often we have measurements xij for i = 1, 2, . . . , N , on variables
j = 1, 2, . . . , p (also called attributes). Since most of the popular clustering
algorithms take a dissimilarity matrix as their input, we must first construct
pairwise dissimilarities between the observations. In the most common case,
we define a dissimilarity dj(xij , xi&prime;j) between values of the jth attribute,
and then define
</p>
<p>D(xi, xi&prime;) =
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>dj(xij , xi&prime;j) (14.20)
</p>
<p>as the dissimilarity between objects i and i&prime;. By far the most common
choice is squared distance</p>
<p/>
</div>
<div class="page"><p/>
<p>504 14. Unsupervised Learning
</p>
<p>dj(xij , xi&prime;j) = (xij &minus; xi&prime;j)2. (14.21)
However, other choices are possible, and can lead to potentially different
results. For nonquantitative attributes (e.g., categorical data), squared dis-
tance may not be appropriate. In addition, it is sometimes desirable to
weigh attributes differently rather than giving them equal weight as in
(14.20).
We first discuss alternatives in terms of the attribute type:
</p>
<p>Quantitative variables. Measurements of this type of variable or attribute
are represented by continuous real-valued numbers. It is natural to
define the &ldquo;error&rdquo; between them as a monotone-increasing function
of their absolute difference
</p>
<p>d(xi, xi&prime;) = l(|xi &minus; xi&prime; |).
Besides squared-error loss (xi&minus;xi&prime;)2, a common choice is the identity
(absolute error). The former places more emphasis on larger differ-
ences than smaller ones. Alternatively, clustering can be based on the
correlation
</p>
<p>ρ(xi, xi&prime;) =
</p>
<p>&sum;
j(xij &minus; x̄i)(xi&prime;j &minus; x̄i&prime;)&radic;&sum;
</p>
<p>j(xij &minus; x̄i)2
&sum;
</p>
<p>j(xi&prime;j &minus; x̄i&prime;)2
, (14.22)
</p>
<p>with x̄i =
&sum;
</p>
<p>j xij/p. Note that this is averaged over variables, not ob-
servations. If the observations are first standardized, then
</p>
<p>&sum;
j(xij &minus;
</p>
<p>xi&prime;j)
2 &prop; 2(1&minus;ρ(xi, xi&prime;)). Hence clustering based on correlation (simi-
</p>
<p>larity) is equivalent to that based on squared distance (dissimilarity).
</p>
<p>Ordinal variables. The values of this type of variable are often represented
as contiguous integers, and the realizable values are considered to be
an ordered set. Examples are academic grades (A, B, C, D, F), degree
of preference (can&rsquo;t stand, dislike, OK, like, terrific). Rank data are a
special kind of ordinal data. Error measures for ordinal variables are
generally defined by replacing their M original values with
</p>
<p>i&minus; 1/2
M
</p>
<p>, i = 1, . . . ,M (14.23)
</p>
<p>in the prescribed order of their original values. They are then treated
as quantitative variables on this scale.
</p>
<p>Categorical variables. With unordered categorical (also called nominal)
variables, the degree-of-difference between pairs of values must be
delineated explicitly. If the variable assumes M distinct values, these
can be arranged in a symmetric M &times;M matrix with elements Lrr&prime; =
Lr&prime;r, Lrr = 0, Lrr&prime; &ge; 0. The most common choice is Lrr&prime; = 1 for all
r �= r&prime;, while unequal losses can be used to emphasize some errors
more than others.</p>
<p/>
</div>
<div class="page"><p/>
<p>14.3 Cluster Analysis 505
</p>
<p>14.3.3 Object Dissimilarity
</p>
<p>Next we define a procedure for combining the p-individual attribute dissim-
ilarities dj(xij , xi&prime;j), j = 1, 2, . . . , p into a single overall measure of dissim-
ilarity D(xi, xi&prime;) between two objects or observations (xi, xi&prime;) possessing
the respective attribute values. This is nearly always done by means of a
weighted average (convex combination)
</p>
<p>D(xi, xi&prime;) =
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>wj &middot; dj(xij , xi&prime;j);
p&sum;
</p>
<p>j=1
</p>
<p>wj = 1. (14.24)
</p>
<p>Here wj is a weight assigned to the jth attribute regulating the relative
influence of that variable in determining the overall dissimilarity between
objects. This choice should be based on subject matter considerations.
It is important to realize that setting the weight wj to the same value
</p>
<p>for each variable (say, wj = 1 &forall; j) does not necessarily give all attributes
equal influence. The influence of the jth attributeXj on object dissimilarity
D(xi, xi&prime;) (14.24) depends upon its relative contribution to the average
object dissimilarity measure over all pairs of observations in the data set
</p>
<p>D̄ =
1
</p>
<p>N2
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>N&sum;
</p>
<p>i&prime;=1
</p>
<p>D(xi, xi&prime;) =
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>wj &middot; d̄j ,
</p>
<p>with
</p>
<p>d̄j =
1
</p>
<p>N2
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>N&sum;
</p>
<p>i&prime;=1
</p>
<p>dj(xij , xi&prime;j) (14.25)
</p>
<p>being the average dissimilarity on the jth attribute. Thus, the relative in-
fluence of the jth variable is wj &middot; d̄j , and setting wj &sim; 1/d̄j would give all
attributes equal influence in characterizing overall dissimilarity between ob-
jects. For example, with p quantitative variables and squared-error distance
used for each coordinate, then (14.24) becomes the (weighted) squared Eu-
clidean distance
</p>
<p>DI(xi, xi&prime;) =
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>wj &middot; (xij &minus; xi&prime;j)2 (14.26)
</p>
<p>between pairs of points in an IRp, with the quantitative variables as axes.
In this case (14.25) becomes
</p>
<p>d̄j =
1
</p>
<p>N2
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>N&sum;
</p>
<p>i&prime;=1
</p>
<p>(xij &minus; xi&prime;j)2 = 2 &middot; varj , (14.27)
</p>
<p>where varj is the sample estimate of Var(Xj). Thus, the relative impor-
tance of each such variable is proportional to its variance over the data</p>
<p/>
</div>
<div class="page"><p/>
<p>506 14. Unsupervised Learning
</p>
<p>-6 -4 -2 0 2 4
</p>
<p>-6
-4
</p>
<p>-2
0
</p>
<p>2
4
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;&bull;&bull;
</p>
<p>&bull;
&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;&bull;
</p>
<p>&bull;
&bull;
&bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull; &bull; &bull;
&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;&bull;
&bull;&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
&bull;&bull;
</p>
<p>&bull; &bull;&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull; &bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>-2 -1 0 1 2
</p>
<p>-2
-1
</p>
<p>0
1
</p>
<p>2
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;&bull;&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
&bull;
</p>
<p>&bull; &bull;
&bull;
</p>
<p>&bull; &bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;&bull; &bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>X1X1
X
</p>
<p>2
</p>
<p>X
2
</p>
<p>FIGURE 14.5. Simulated data: on the left, K-means clustering (with K=2) has
been applied to the raw data. The two colors indicate the cluster memberships. On
the right, the features were first standardized before clustering. This is equivalent
to using feature weights 1/[2 &middot;var(Xj)]. The standardization has obscured the two
well-separated groups. Note that each plot uses the same units in the horizontal
and vertical axes.
</p>
<p>set. In general, setting wj = 1/d̄j for all attributes, irrespective of type,
will cause each one of them to equally influence the overall dissimilarity
between pairs of objects (xi, xi&prime;). Although this may seem reasonable, and
is often recommended, it can be highly counterproductive. If the goal is to
segment the data into groups of similar objects, all attributes may not con-
tribute equally to the (problem-dependent) notion of dissimilarity between
objects. Some attribute value differences may reflect greater actual object
dissimilarity in the context of the problem domain.
If the goal is to discover natural groupings in the data, some attributes
</p>
<p>may exhibit more of a grouping tendency than others. Variables that are
more relevant in separating the groups should be assigned a higher influ-
ence in defining object dissimilarity. Giving all attributes equal influence
in this case will tend to obscure the groups to the point where a clustering
algorithm cannot uncover them. Figure 14.5 shows an example.
Although simple generic prescriptions for choosing the individual at-
</p>
<p>tribute dissimilarities dj(xij , xi&prime;j) and their weights wj can be comforting,
there is no substitute for careful thought in the context of each individ-
ual problem. Specifying an appropriate dissimilarity measure is far more
important in obtaining success with clustering than choice of clustering
algorithm. This aspect of the problem is emphasized less in the cluster-
ing literature than the algorithms themselves, since it depends on domain
knowledge specifics and is less amenable to general research.</p>
<p/>
</div>
<div class="page"><p/>
<p>14.3 Cluster Analysis 507
</p>
<p>Finally, often observations have missing values in one or more of the
attributes. The most common method of incorporating missing values in
dissimilarity calculations (14.24) is to omit each observation pair xij , xi&prime;j
having at least one value missing, when computing the dissimilarity be-
tween observations xi and x
</p>
<p>&prime;
i. This method can fail in the circumstance
</p>
<p>when both observations have no measured values in common. In this case
both observations could be deleted from the analysis. Alternatively, the
missing values could be imputed using the mean or median of each attribute
over the nonmissing data. For categorical variables, one could consider the
value &ldquo;missing&rdquo; as just another categorical value, if it were reasonable to
consider two objects as being similar if they both have missing values on
the same variables.
</p>
<p>14.3.4 Clustering Algorithms
</p>
<p>The goal of cluster analysis is to partition the observations into groups
(&ldquo;clusters&rdquo;) so that the pairwise dissimilarities between those assigned to
the same cluster tend to be smaller than those in different clusters. Clus-
tering algorithms fall into three distinct types: combinatorial algorithms,
mixture modeling, and mode seeking.
Combinatorial algorithms work directly on the observed data with no
</p>
<p>direct reference to an underlying probability model. Mixture modeling sup-
poses that the data is an i.i.d sample from some population described by a
probability density function. This density function is characterized by a pa-
rameterized model taken to be a mixture of component density functions;
each component density describes one of the clusters. This model is then fit
to the data by maximum likelihood or corresponding Bayesian approaches.
Mode seekers (&ldquo;bump hunters&rdquo;) take a nonparametric perspective, attempt-
ing to directly estimate distinct modes of the probability density function.
Observations &ldquo;closest&rdquo; to each respective mode then define the individual
clusters.
Mixture modeling is described in Section 6.8. The PRIM algorithm, dis-
</p>
<p>cussed in Sections 9.3 and 14.2.5, is an example of mode seeking or &ldquo;bump
hunting.&rdquo; We discuss combinatorial algorithms next.
</p>
<p>14.3.5 Combinatorial Algorithms
</p>
<p>The most popular clustering algorithms directly assign each observation
to a group or cluster without regard to a probability model describing the
data. Each observation is uniquely labeled by an integer i &isin; {1, &middot; &middot; &middot;, N}.
A prespecified number of clusters K &lt; N is postulated, and each one is
labeled by an integer k &isin; {1, . . . ,K}. Each observation is assigned to one
and only one cluster. These assignments can be characterized by a many-
to-one mapping, or encoder k = C(i), that assigns the ith observation to
the kth cluster. One seeks the particular encoder C&lowast;(i) that achieves the</p>
<p/>
</div>
<div class="page"><p/>
<p>508 14. Unsupervised Learning
</p>
<p>required goal (details below), based on the dissimilarities d(xi, xi&prime;) between
every pair of observations. These are specified by the user as described
above. Generally, the encoder C(i) is explicitly delineated by giving its
value (cluster assignment) for each observation i. Thus, the &ldquo;parameters&rdquo;
of the procedure are the individual cluster assignments for each of the N
observations. These are adjusted so as to minimize a &ldquo;loss&rdquo; function that
characterizes the degree to which the clustering goal is not met.
</p>
<p>One approach is to directly specify a mathematical loss function and
attempt to minimize it through some combinatorial optimization algorithm.
Since the goal is to assign close points to the same cluster, a natural loss
(or &ldquo;energy&rdquo;) function would be
</p>
<p>W (C) =
1
</p>
<p>2
</p>
<p>K&sum;
</p>
<p>k=1
</p>
<p>&sum;
</p>
<p>C(i)=k
</p>
<p>&sum;
</p>
<p>C(i&prime;)=k
</p>
<p>d(xi, xi&prime;). (14.28)
</p>
<p>This criterion characterizes the extent to which observations assigned to
the same cluster tend to be close to one another. It is sometimes referred
to as the &ldquo;within cluster&rdquo; point scatter since
</p>
<p>T =
1
</p>
<p>2
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>N&sum;
</p>
<p>i&prime;=1
</p>
<p>dii&prime; =
1
</p>
<p>2
</p>
<p>K&sum;
</p>
<p>k=1
</p>
<p>&sum;
</p>
<p>C(i)=k
</p>
<p>⎛
⎝ &sum;
</p>
<p>C(i&prime;)=k
</p>
<p>dii&prime; +
&sum;
</p>
<p>C(i&prime;) �=k
</p>
<p>dii&prime;
</p>
<p>⎞
⎠ ,
</p>
<p>or
T = W (C) +B(C),
</p>
<p>where dii&prime; = d(xi, xi&prime;). Here T is the total point scatter, which is a constant
given the data, independent of cluster assignment. The quantity
</p>
<p>B(C) =
1
</p>
<p>2
</p>
<p>K&sum;
</p>
<p>k=1
</p>
<p>&sum;
</p>
<p>C(i)=k
</p>
<p>&sum;
</p>
<p>C(i&prime;) �=k
</p>
<p>dii&prime; (14.29)
</p>
<p>is the between-cluster point scatter. This will tend to be large when obser-
vations assigned to different clusters are far apart. Thus one has
</p>
<p>W (C) = T &minus;B(C)
and minimizing W (C) is equivalent to maximizing B(C).
</p>
<p>Cluster analysis by combinatorial optimization is straightforward in prin-
ciple. One simply minimizes W or equivalently maximizes B over all pos-
sible assignments of the N data points to K clusters. Unfortunately, such
optimization by complete enumeration is feasible only for very small data
sets. The number of distinct assignments is (Jain and Dubes, 1988)
</p>
<p>S(N,K) =
1
</p>
<p>K!
</p>
<p>K&sum;
</p>
<p>k=1
</p>
<p>(&minus;1)K&minus;k
(
K
</p>
<p>k
</p>
<p>)
kN . (14.30)
</p>
<p>For example, S(10, 4) = 34, 105 which is quite feasible. But, S(N,K) grows
very rapidly with increasing values of its arguments. Already S(19, 4) ≃</p>
<p/>
</div>
<div class="page"><p/>
<p>14.3 Cluster Analysis 509
</p>
<p>1010, and most clustering problems involve much larger data sets than
N = 19. For this reason, practical clustering algorithms are able to examine
only a very small fraction of all possible encoders k = C(i). The goal is to
identify a small subset that is likely to contain the optimal one, or at least
a good suboptimal partition.
Such feasible strategies are based on iterative greedy descent. An initial
</p>
<p>partition is specified. At each iterative step, the cluster assignments are
changed in such a way that the value of the criterion is improved from
its previous value. Clustering algorithms of this type differ in their pre-
scriptions for modifying the cluster assignments at each iteration. When
the prescription is unable to provide an improvement, the algorithm ter-
minates with the current assignments as its solution. Since the assignment
of observations to clusters at any iteration is a perturbation of that for the
previous iteration, only a very small fraction of all possible assignments
(14.30) are examined. However, these algorithms converge to local optima
which may be highly suboptimal when compared to the global optimum.
</p>
<p>14.3.6 K-means
</p>
<p>The K-means algorithm is one of the most popular iterative descent clus-
tering methods. It is intended for situations in which all variables are of
the quantitative type, and squared Euclidean distance
</p>
<p>d(xi, xi&prime;) =
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>(xij &minus; xi&prime;j)2 = ||xi &minus; xi&prime; ||2
</p>
<p>is chosen as the dissimilarity measure. Note that weighted Euclidean dis-
tance can be used by redefining the xij values (Exercise 14.1).
The within-point scatter (14.28) can be written as
</p>
<p>W (C) =
1
</p>
<p>2
</p>
<p>K&sum;
</p>
<p>k=1
</p>
<p>&sum;
</p>
<p>C(i)=k
</p>
<p>&sum;
</p>
<p>C(i&prime;)=k
</p>
<p>||xi &minus; xi&prime; ||2
</p>
<p>=
</p>
<p>K&sum;
</p>
<p>k=1
</p>
<p>Nk
&sum;
</p>
<p>C(i)=k
</p>
<p>||xi &minus; x̄k||2, (14.31)
</p>
<p>where x̄k = (x̄1k, . . . , x̄pk) is the mean vector associated with the kth clus-
</p>
<p>ter, and Nk =
&sum;N
</p>
<p>i=1 I(C(i) = k). Thus, the criterion is minimized by
assigning the N observations to the K clusters in such a way that within
each cluster the average dissimilarity of the observations from the cluster
mean, as defined by the points in that cluster, is minimized.
An iterative descent algorithm for solving</p>
<p/>
</div>
<div class="page"><p/>
<p>510 14. Unsupervised Learning
</p>
<p>Algorithm 14.1 K-means Clustering.
</p>
<p>1. For a given cluster assignment C, the total cluster variance (14.33) is
minimized with respect to {m1, . . . ,mK} yielding the means of the
currently assigned clusters (14.32).
</p>
<p>2. Given a current set of means {m1, . . . ,mK}, (14.33) is minimized by
assigning each observation to the closest (current) cluster mean. That
is,
</p>
<p>C(i) = argmin
1&le;k&le;K
</p>
<p>||xi &minus;mk||2. (14.34)
</p>
<p>3. Steps 1 and 2 are iterated until the assignments do not change.
</p>
<p>C&lowast; = min
C
</p>
<p>K&sum;
</p>
<p>k=1
</p>
<p>Nk
&sum;
</p>
<p>C(i)=k
</p>
<p>||xi &minus; x̄k||2
</p>
<p>can be obtained by noting that for any set of observations S
</p>
<p>x̄S = argmin
m
</p>
<p>&sum;
</p>
<p>i&isin;S
</p>
<p>||xi &minus;m||2. (14.32)
</p>
<p>Hence we can obtain C&lowast; by solving the enlarged optimization problem
</p>
<p>min
C,{mk}K1
</p>
<p>K&sum;
</p>
<p>k=1
</p>
<p>Nk
&sum;
</p>
<p>C(i)=k
</p>
<p>||xi &minus;mk||2. (14.33)
</p>
<p>This can be minimized by an alternating optimization procedure given in
Algorithm 14.1.
Each of steps 1 and 2 reduces the value of the criterion (14.33), so that
</p>
<p>convergence is assured. However, the result may represent a suboptimal
local minimum. The algorithm of Hartigan and Wong (1979) goes further,
and ensures that there is no single switch of an observation from one group
to another group that will decrease the objective. In addition, one should
start the algorithm with many different random choices for the starting
means, and choose the solution having smallest value of the objective func-
tion.
Figure 14.6 shows some of the K-means iterations for the simulated data
</p>
<p>of Figure 14.4. The centroids are depicted by &ldquo;O&rdquo;s. The straight lines show
the partitioning of points, each sector being the set of points closest to
each centroid. This partitioning is called the Voronoi tessellation. After 20
iterations the procedure has converged.
</p>
<p>14.3.7 Gaussian Mixtures as Soft K-means Clustering
</p>
<p>The K-means clustering procedure is closely related to the EM algorithm
for estimating a certain Gaussian mixture model. (Sections 6.8 and 8.5.1).</p>
<p/>
</div>
<div class="page"><p/>
<p>14.3 Cluster Analysis 511
</p>
<p>-4 -2 0 2 4 6
</p>
<p>-2
0
</p>
<p>2
4
</p>
<p>6
</p>
<p>Initial Centroids
</p>
<p>&bull; &bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;&bull;
</p>
<p>&bull;&bull; &bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;&bull;
&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
&bull; &bull;
&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
&bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
&bull;
</p>
<p>&bull; &bull;&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
&bull;&bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull; &bull; &bull;
</p>
<p>&bull;
&bull; &bull;
</p>
<p>&bull; &bull;
&bull; &bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull; &bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;&bull;&bull; &bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull; &bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;&bull;
</p>
<p>&bull;&bull; &bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;&bull;
&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
&bull; &bull;
&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
&bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
&bull;
</p>
<p>&bull; &bull;&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
&bull;&bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull; &bull; &bull;
</p>
<p>&bull;
&bull; &bull;
</p>
<p>&bull; &bull;
&bull; &bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull; &bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;&bull;&bull; &bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>Initial Partition
</p>
<p>&bull; &bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;&bull;
</p>
<p>&bull;&bull; &bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;&bull;
&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
&bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
&bull;
</p>
<p>&bull; &bull;&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
&bull; &bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull; &bull;&bull; &bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull; &bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;&bull;&bull; &bull;&bull;
</p>
<p>Iteration Number  2
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull; &bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;&bull;
</p>
<p>&bull;&bull; &bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;&bull;
&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
&bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
&bull;
</p>
<p>&bull; &bull;&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull; &bull; &bull;
</p>
<p>&bull;
&bull; &bull;
</p>
<p>&bull; &bull;
&bull;&bull; &bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;&bull;
&bull;
&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;&bull;&bull; &bull;&bull;
</p>
<p>Iteration Number  20
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>FIGURE 14.6. Successive iterations of the K-means clustering algorithm for
the simulated data of Figure 14.4.</p>
<p/>
</div>
<div class="page"><p/>
<p>512 14. Unsupervised Learning
</p>
<p>&bull; &bull;
</p>
<p>R
e
s
p
o
n
s
ib
</p>
<p>ili
ti
e
s
</p>
<p>0
.0
</p>
<p>0
.2
</p>
<p>0
.4
</p>
<p>0
.6
</p>
<p>0
.8
</p>
<p>1
.0
</p>
<p>&bull; &bull;
</p>
<p>R
e
s
p
o
n
s
ib
</p>
<p>ili
ti
e
s
</p>
<p>0
.0
</p>
<p>0
.2
</p>
<p>0
.4
</p>
<p>0
.6
</p>
<p>0
.8
</p>
<p>1
.0
</p>
<p>σ = 1.0σ = 1.0
</p>
<p>σ = 0.2σ = 0.2
</p>
<p>FIGURE 14.7. (Left panels:) two Gaussian densities g0(x) and g1(x) (blue and
orange) on the real line, and a single data point (green dot) at x = 0.5. The colored
squares are plotted at x = &minus;1.0 and x = 1.0, the means of each density. (Right
panels:) the relative densities g0(x)/(g0(x) + g1(x)) and g1(x)/(g0(x) + g1(x)),
called the &ldquo;responsibilities&rdquo; of each cluster, for this data point. In the top panels,
the Gaussian standard deviation σ = 1.0; in the bottom panels σ = 0.2. The
EM algorithm uses these responsibilities to make a &ldquo;soft&rdquo; assignment of each
data point to each of the two clusters. When σ is fairly large, the responsibilities
can be near 0.5 (they are 0.36 and 0.64 in the top right panel). As σ &rarr; 0, the
responsibilities &rarr; 1, for the cluster center closest to the target point, and 0 for
all other clusters. This &ldquo;hard&rdquo; assignment is seen in the bottom right panel.
</p>
<p>The E-step of the EM algorithm assigns &ldquo;responsibilities&rdquo; for each data
point based in its relative density under each mixture component, while
the M-step recomputes the component density parameters based on the
current responsibilities. Suppose we specify K mixture components, each
with a Gaussian density having scalar covariance matrix σ2I. Then the
relative density under each mixture component is a monotone function of
the Euclidean distance between the data point and the mixture center.
Hence in this setup EM is a &ldquo;soft&rdquo; version of K-means clustering, making
probabilistic (rather than deterministic) assignments of points to cluster
centers. As the variance σ2 &rarr; 0, these probabilities become 0 and 1, and
the two methods coincide. Details are given in Exercise 14.2. Figure 14.7
illustrates this result for two clusters on the real line.
</p>
<p>14.3.8 Example: Human Tumor Microarray Data
</p>
<p>We apply K-means clustering to the human tumor microarray data de-
scribed in Chapter 1. This is an example of high-dimensional clustering.</p>
<p/>
</div>
<div class="page"><p/>
<p>14.3 Cluster Analysis 513
</p>
<p>Number of Clusters K
</p>
<p>S
u
</p>
<p>m
 o
</p>
<p>f 
S
</p>
<p>q
u
</p>
<p>a
re
</p>
<p>s
</p>
<p>2 4 6 8 10
</p>
<p>1
6
</p>
<p>0
0
</p>
<p>0
0
</p>
<p>2
0
</p>
<p>0
0
</p>
<p>0
0
</p>
<p>2
4
</p>
<p>0
0
</p>
<p>0
0
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull; &bull;
&bull; &bull;
</p>
<p>FIGURE 14.8. Total within-cluster sum of squares for K-means clustering ap-
plied to the human tumor microarray data.
</p>
<p>TABLE 14.2. Human tumor data: number of cancer cases of each type, in each
of the three clusters from K-means clustering.
</p>
<p>Cluster Breast CNS Colon K562 Leukemia MCF7
</p>
<p>1 3 5 0 0 0 0
2 2 0 0 2 6 2
3 2 0 7 0 0 0
</p>
<p>Cluster Melanoma NSCLC Ovarian Prostate Renal Unknown
1 1 7 6 2 9 1
2 7 2 0 0 0 0
3 0 0 0 0 0 0
</p>
<p>The data are a 6830 &times; 64 matrix of real numbers, each representing an
expression measurement for a gene (row) and sample (column). Here we
cluster the samples, each of which is a vector of length 6830, correspond-
ing to expression values for the 6830 genes. Each sample has a label such
as breast (for breast cancer), melanoma, and so on; we don&rsquo;t use these la-
bels in the clustering, but will examine posthoc which labels fall into which
clusters.
We applied K-means clustering with K running from 1 to 10, and com-
</p>
<p>puted the total within-sum of squares for each clustering, shown in Fig-
ure 14.8. Typically one looks for a kink in the sum of squares curve (or its
logarithm) to locate the optimal number of clusters (see Section 14.3.11).
Here there is no clear indication: for illustration we chose K = 3 giving the
three clusters shown in Table 14.2.</p>
<p/>
</div>
<div class="page"><p/>
<p>514 14. Unsupervised Learning
</p>
<p>FIGURE 14.9. Sir Ronald A. Fisher (1890 &minus; 1962) was one of the founders
of modern day statistics, to whom we owe maximum-likelihood, sufficiency, and
many other fundamental concepts. The image on the left is a 1024&times;1024 grayscale
image at 8 bits per pixel. The center image is the result of 2&times; 2 block VQ, using
200 code vectors, with a compression rate of 1.9 bits/pixel. The right image uses
only four code vectors, with a compression rate of 0.50 bits/pixel
</p>
<p>We see that the procedure is successful at grouping together samples of
the same cancer. In fact, the two breast cancers in the second cluster were
later found to be misdiagnosed and were melanomas that had metastasized.
However,K-means clustering has shortcomings in this application. For one,
it does not give a linear ordering of objects within a cluster: we have simply
listed them in alphabetic order above. Secondly, as the number of clusters
K is changed, the cluster memberships can change in arbitrary ways. That
is, with say four clusters, the clusters need not be nested within the three
clusters above. For these reasons, hierarchical clustering (described later),
is probably preferable for this application.
</p>
<p>14.3.9 Vector Quantization
</p>
<p>The K-means clustering algorithm represents a key tool in the apparently
unrelated area of image and signal compression, particularly in vector quan-
tization or VQ (Gersho and Gray, 1992). The left image in Figure 14.92 is a
digitized photograph of a famous statistician, Sir Ronald Fisher. It consists
of 1024&times; 1024 pixels, where each pixel is a grayscale value ranging from 0
to 255, and hence requires 8 bits of storage per pixel. The entire image oc-
cupies 1 megabyte of storage. The center image is a VQ-compressed version
of the left panel, and requires 0.239 of the storage (at some loss in quality).
The right image is compressed even more, and requires only 0.0625 of the
storage (at a considerable loss in quality).
The version of VQ implemented here first breaks the image into small
</p>
<p>blocks, in this case 2&times;2 blocks of pixels. Each of the 512&times;512 blocks of four
</p>
<p>2This example was prepared by Maya Gupta.</p>
<p/>
</div>
<div class="page"><p/>
<p>14.3 Cluster Analysis 515
</p>
<p>numbers is regarded as a vector in IR4. A K-means clustering algorithm
(also known as Lloyd&rsquo;s algorithm in this context) is run in this space.
The center image uses K = 200, while the right image K = 4. Each of
the 512&times; 512 pixel blocks (or points) is approximated by its closest cluster
centroid, known as a codeword. The clustering process is called the encoding
step, and the collection of centroids is called the codebook.
To represent the approximated image, we need to supply for each block
</p>
<p>the identity of the codebook entry that approximates it. This will require
log2(K) bits per block. We also need to supply the codebook itself, which
is K &times; 4 real numbers (typically negligible). Overall, the storage for the
compressed image amounts to log2(K)/(4 &middot; 8) of the original (0.239 for
K = 200, 0.063 for K = 4). This is typically expressed as a rate in bits
per pixel: log2(K)/4, which are 1.91 and 0.50, respectively. The process
of constructing the approximate image from the centroids is called the
decoding step.
</p>
<p>Why do we expect VQ to work at all? The reason is that for typical
everyday images like photographs, many of the blocks look the same. In
this case there are many almost pure white blocks, and similarly pure gray
blocks of various shades. These require only one block each to represent
them, and then multiple pointers to that block.
What we have described is known as lossy compression, since our im-
</p>
<p>ages are degraded versions of the original. The degradation or distortion is
usually measured in terms of mean squared error. In this case D = 0.89
for K = 200 and D = 16.95 for K = 4. More generally a rate/distortion
curve would be used to assess the tradeoff. One can also perform lossless
compression using block clustering, and still capitalize on the repeated pat-
terns. If you took the original image and losslessly compressed it, the best
you would do is 4.48 bits per pixel.
We claimed above that log2(K) bits were needed to identify each of theK
</p>
<p>codewords in the codebook. This uses a fixed-length code, and is inefficient
if some codewords occur many more times than others in the image. Using
Shannon coding theory, we know that in general a variable length code
will do better, and the rate then becomes &minus;&sum;Kℓ=1 pℓ log2(pℓ)/4. The term
in the numerator is the entropy of the distribution pℓ of the codewords
in the image. Using variable length coding our rates come down to 1.42
and 0.39, respectively. Finally, there are many generalizations of VQ that
have been developed: for example, tree-structured VQ finds the centroids
with a top-down, 2-means style algorithm, as alluded to in Section 14.3.12.
This allows successive refinement of the compression. Further details may
be found in Gersho and Gray (1992).
</p>
<p>14.3.10 K-medoids
</p>
<p>As discussed above, the K-means algorithm is appropriate when the dis-
similarity measure is taken to be squared Euclidean distance D(xi, xi&prime;)</p>
<p/>
</div>
<div class="page"><p/>
<p>516 14. Unsupervised Learning
</p>
<p>Algorithm 14.2 K-medoids Clustering.
</p>
<p>1. For a given cluster assignment C find the observation in the cluster
minimizing total distance to other points in that cluster:
</p>
<p>i&lowast;k = argmin
{i:C(i)=k}
</p>
<p>&sum;
</p>
<p>C(i&prime;)=k
</p>
<p>D(xi, xi&prime;). (14.35)
</p>
<p>Then mk = xi&lowast;
k
, k = 1, 2, . . . ,K are the current estimates of the
</p>
<p>cluster centers.
</p>
<p>2. Given a current set of cluster centers {m1, . . . ,mK}, minimize the to-
tal error by assigning each observation to the closest (current) cluster
center:
</p>
<p>C(i) = argmin
1&le;k&le;K
</p>
<p>D(xi,mk). (14.36)
</p>
<p>3. Iterate steps 1 and 2 until the assignments do not change.
</p>
<p>(14.112). This requires all of the variables to be of the quantitative type. In
addition, using squared Euclidean distance places the highest influence on
the largest distances. This causes the procedure to lack robustness against
outliers that produce very large distances. These restrictions can be re-
moved at the expense of computation.
The only part of the K-means algorithm that assumes squared Eu-
</p>
<p>clidean distance is the minimization step (14.32); the cluster representatives
{m1, . . . ,mK} in (14.33) are taken to be the means of the currently assigned
clusters. The algorithm can be generalized for use with arbitrarily defined
dissimilarities D(xi, xi&prime;) by replacing this step by an explicit optimization
with respect to {m1, . . . ,mK} in (14.33). In the most common form, cen-
ters for each cluster are restricted to be one of the observations assigned
to the cluster, as summarized in Algorithm 14.2. This algorithm assumes
attribute data, but the approach can also be applied to data described
only by proximity matrices (Section 14.3.1). There is no need to explicitly
compute cluster centers; rather we just keep track of the indices i&lowast;k.
</p>
<p>Solving (14.32) for each provisional cluster k requires an amount of com-
putation proportional to the number of observations assigned to it, whereas
for solving (14.35) the computation increases to O(N2k ). Given a set of clus-
ter &ldquo;centers,&rdquo; {i1, . . . , iK}, obtaining the new assignments
</p>
<p>C(i) = argmin
1&le;k&le;K
</p>
<p>dii&lowast;
k
</p>
<p>(14.37)
</p>
<p>requires computation proportional to K &middot;N as before. Thus, K-medoids is
far more computationally intensive than K-means.
Alternating between (14.35) and (14.37) represents a particular heuristic
</p>
<p>search strategy for trying to solve</p>
<p/>
</div>
<div class="page"><p/>
<p>14.3 Cluster Analysis 517
</p>
<p>TABLE 14.3. Data from a political science survey: values are average pairwise
dissimilarities of countries from a questionnaire given to political science students.
</p>
<p>BEL BRA CHI CUB EGY FRA IND ISR USA USS YUG
</p>
<p>BRA 5.58
CHI 7.00 6.50
CUB 7.08 7.00 3.83
EGY 4.83 5.08 8.17 5.83
FRA 2.17 5.75 6.67 6.92 4.92
IND 6.42 5.00 5.58 6.00 4.67 6.42
ISR 3.42 5.50 6.42 6.42 5.00 3.92 6.17
USA 2.50 4.92 6.25 7.33 4.50 2.25 6.33 2.75
USS 6.08 6.67 4.25 2.67 6.00 6.17 6.17 6.92 6.17
YUG 5.25 6.83 4.50 3.75 5.75 5.42 6.08 5.83 6.67 3.67
ZAI 4.75 3.00 6.08 6.67 5.00 5.58 4.83 6.17 5.67 6.50 6.92
</p>
<p>min
C, {ik}K1
</p>
<p>K&sum;
</p>
<p>k=1
</p>
<p>&sum;
</p>
<p>C(i)=k
</p>
<p>diik . (14.38)
</p>
<p>Kaufman and Rousseeuw (1990) propose an alternative strategy for directly
solving (14.38) that provisionally exchanges each center ik with an obser-
vation that is not currently a center, selecting the exchange that produces
the greatest reduction in the value of the criterion (14.38). This is repeated
until no advantageous exchanges can be found. Massart et al. (1983) derive
a branch-and-bound combinatorial method that finds the global minimum
of (14.38) that is practical only for very small data sets.
</p>
<p>Example: Country Dissimilarities
</p>
<p>This example, taken from Kaufman and Rousseeuw (1990), comes from a
study in which political science students were asked to provide pairwise dis-
similarity measures for 12 countries: Belgium, Brazil, Chile, Cuba, Egypt,
France, India, Israel, United States, Union of Soviet Socialist Republics,
Yugoslavia and Zaire. The average dissimilarity scores are given in Ta-
ble 14.3. We applied 3-medoid clustering to these dissimilarities. Note that
K-means clustering could not be applied because we have only distances
rather than raw observations. The left panel of Figure 14.10 shows the
dissimilarities reordered and blocked according to the 3-medoid clustering.
The right panel is a two-dimensional multidimensional scaling plot, with
the 3-medoid clusters assignments indicated by colors (multidimensional
scaling is discussed in Section 14.8.) Both plots show three well-separated
clusters, but the MDS display indicates that &ldquo;Egypt&rdquo; falls about halfway
between two clusters.</p>
<p/>
</div>
<div class="page"><p/>
<p>518 14. Unsupervised Learning
</p>
<p>C
H
</p>
<p>I
</p>
<p>C
U
</p>
<p>B
</p>
<p>U
S
</p>
<p>S
</p>
<p>Y
U
</p>
<p>G
</p>
<p>B
R
</p>
<p>A
</p>
<p>IN
D
</p>
<p>Z
A
</p>
<p>I
</p>
<p>B
E
</p>
<p>L
</p>
<p>E
G
</p>
<p>Y
</p>
<p>F
R
</p>
<p>A
</p>
<p>IS
R
</p>
<p>CUB
</p>
<p>USS
</p>
<p>YUG
</p>
<p>BRA
</p>
<p>IND
</p>
<p>ZAI
</p>
<p>BEL
</p>
<p>EGY
</p>
<p>FRA
</p>
<p>ISR
</p>
<p>USA
</p>
<p>Reordered Dissimilarity Matrix First MDS Coordinate
</p>
<p>S
e
</p>
<p>c
o
</p>
<p>n
d
</p>
<p> M
D
</p>
<p>S
 C
</p>
<p>o
o
</p>
<p>rd
in
</p>
<p>a
te
</p>
<p>-2 0 2 4
</p>
<p>-2
-1
</p>
<p>0
1
</p>
<p>2
3
</p>
<p>CHI
</p>
<p>CUB
</p>
<p>USS
</p>
<p>YUG
</p>
<p>BRA IND
ZAI
</p>
<p>BEL
</p>
<p>EGY
</p>
<p>FRA
</p>
<p>ISR
USA
</p>
<p>FIGURE 14.10. Survey of country dissimilarities. (Left panel:) dissimilarities
reordered and blocked according to 3-medoid clustering. Heat map is coded from
most similar (dark red) to least similar (bright red). (Right panel:) two-dimen-
sional multidimensional scaling plot, with 3-medoid clusters indicated by different
colors.
</p>
<p>14.3.11 Practical Issues
</p>
<p>In order to apply K-means or K-medoids one must select the number of
clusters K&lowast; and an initialization. The latter can be defined by specifying
an initial set of centers {m1, . . . ,mK} or {i1, . . . , iK} or an initial encoder
C(i). Usually specifying the centers is more convenient. Suggestions range
from simple random selection to a deliberate strategy based on forward
stepwise assignment. At each step a new center ik is chosen to minimize
the criterion (14.33) or (14.38), given the centers i1, . . . , ik&minus;1 chosen at the
previous steps. This continues for K steps, thereby producing K initial
centers with which to begin the optimization algorithm.
A choice for the number of clusters K depends on the goal. For data
</p>
<p>segmentation K is usually defined as part of the problem. For example,
a company may employ K sales people, and the goal is to partition a
customer database intoK segments, one for each sales person, such that the
customers assigned to each one are as similar as possible. Often, however,
cluster analysis is used to provide a descriptive statistic for ascertaining the
extent to which the observations comprising the data base fall into natural
distinct groupings. Here the number of such groups K&lowast; is unknown and
one requires that it, as well as the groupings themselves, be estimated from
the data.
Data-based methods for estimating K&lowast; typically examine the within-
</p>
<p>cluster dissimilarityWK as a function of the number of clustersK. Separate
solutions are obtained for K &isin; {1, 2, . . . ,Kmax}. The corresponding values</p>
<p/>
</div>
<div class="page"><p/>
<p>14.3 Cluster Analysis 519
</p>
<p>{W1,W2, . . . ,WKmax} generally decrease with increasing K. This will be
the case even when the criterion is evaluated on an independent test set,
since a large number of cluster centers will tend to fill the feature space
densely and thus will be close to all data points. Thus cross-validation
techniques, so useful for model selection in supervised learning, cannot be
utilized in this context.
The intuition underlying the approach is that if there are actually K&lowast;
</p>
<p>distinct groupings of the observations (as defined by the dissimilarity mea-
sure), then for K &lt; K&lowast; the clusters returned by the algorithm will each
contain a subset of the true underlying groups. That is, the solution will
not assign observations in the same naturally occurring group to different
estimated clusters. To the extent that this is the case, the solution criterion
value will tend to decrease substantially with each successive increase in the
number of specified clusters, WK+1 ≪ WK , as the natural groups are suc-
cessively assigned to separate clusters. For K &gt; K&lowast;, one of the estimated
clusters must partition at least one of the natural groups into two sub-
groups. This will tend to provide a smaller decrease in the criterion as K is
further increased. Splitting a natural group, within which the observations
are all quite close to each other, reduces the criterion less than partitioning
the union of two well-separated groups into their proper constituents.
To the extent this scenario is realized, there will be a sharp decrease in
</p>
<p>successive differences in criterion value, WK &minus; WK+1, at K = K&lowast;. That
is, {WK &minus; WK+1 |K &lt; K&lowast;} ≫ {WK &minus; WK+1 |K &ge; K&lowast;}. An estimate
K̂&lowast; for K&lowast; is then obtained by identifying a &ldquo;kink&rdquo; in the plot of WK as a
function ofK. As with other aspects of clustering procedures, this approach
is somewhat heuristic.
The recently proposed Gap statistic (Tibshirani et al., 2001b) compares
</p>
<p>the curve logWK to the curve obtained from data uniformly distributed
over a rectangle containing the data. It estimates the optimal number of
clusters to be the place where the gap between the two curves is largest.
Essentially this is an automatic way of locating the aforementioned &ldquo;kink.&rdquo;
It also works reasonably well when the data fall into a single cluster, and
in that case will tend to estimate the optimal number of clusters to be one.
This is the scenario where most other competing methods fail.
Figure 14.11 shows the result of the Gap statistic applied to simulated
</p>
<p>data of Figure 14.4. The left panel shows logWK forK = 1, 2, . . . , 8 clusters
(green curve) and the expected value of logWK over 20 simulations from
uniform data (blue curve). The right panel shows the gap curve, which is the
expected curve minus the observed curve. Shown also are error bars of half-
width s&prime;K = sK
</p>
<p>&radic;
1 + 1/20, where sK is the standard deviation of logWK
</p>
<p>over the 20 simulations. The Gap curve is maximized at K = 2 clusters. If
G(K) is the Gap curve at K clusters, the formal rule for estimating K&lowast; is
</p>
<p>K&lowast; = argmin
K
</p>
<p>{K|G(K) &ge; G(K + 1)&minus; s&prime;K+1}. (14.39)</p>
<p/>
</div>
<div class="page"><p/>
<p>520 14. Unsupervised Learning
</p>
<p>Number of Clusters
</p>
<p>2 4 6 8
</p>
<p>-3
.0
</p>
<p>-2
.5
</p>
<p>-2
.0
</p>
<p>-1
.5
</p>
<p>-1
.0
</p>
<p>-0
.5
</p>
<p>0
.0 &bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>Number of Clusters
</p>
<p>G
a
</p>
<p>p
2 4 6 8
</p>
<p>-0
.5
</p>
<p>0
.0
</p>
<p>0
.5
</p>
<p>1
.0
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull; &bull; &bull;
</p>
<p>&bull;l
o
g
W
</p>
<p>K
&minus;
</p>
<p>lo
g
W
</p>
<p>1
</p>
<p>FIGURE 14.11. (Left panel): observed (green) and expected (blue) values of
logWK for the simulated data of Figure 14.4. Both curves have been translated
to equal zero at one cluster. (Right panel): Gap curve, equal to the difference
between the observed and expected values of logWK . The Gap estimate K
</p>
<p>&lowast; is the
smallest K producing a gap within one standard deviation of the gap at K + 1;
here K&lowast; = 2.
</p>
<p>This gives K&lowast; = 2, which looks reasonable from Figure 14.4.
</p>
<p>14.3.12 Hierarchical Clustering
</p>
<p>The results of applying K-means or K-medoids clustering algorithms de-
pend on the choice for the number of clusters to be searched and a starting
configuration assignment. In contrast, hierarchical clustering methods do
not require such specifications. Instead, they require the user to specify a
measure of dissimilarity between (disjoint) groups of observations, based
on the pairwise dissimilarities among the observations in the two groups.
As the name suggests, they produce hierarchical representations in which
the clusters at each level of the hierarchy are created by merging clusters
at the next lower level. At the lowest level, each cluster contains a single
observation. At the highest level there is only one cluster containing all of
the data.
Strategies for hierarchical clustering divide into two basic paradigms: ag-
</p>
<p>glomerative (bottom-up) and divisive (top-down). Agglomerative strategies
start at the bottom and at each level recursively merge a selected pair of
clusters into a single cluster. This produces a grouping at the next higher
level with one less cluster. The pair chosen for merging consist of the two
groups with the smallest intergroup dissimilarity. Divisive methods start
at the top and at each level recursively split one of the existing clusters at</p>
<p/>
</div>
<div class="page"><p/>
<p>14.3 Cluster Analysis 521
</p>
<p>that level into two new clusters. The split is chosen to produce two new
groups with the largest between-group dissimilarity. With both paradigms
there are N &minus; 1 levels in the hierarchy.
Each level of the hierarchy represents a particular grouping of the data
</p>
<p>into disjoint clusters of observations. The entire hierarchy represents an
ordered sequence of such groupings. It is up to the user to decide which
level (if any) actually represents a &ldquo;natural&rdquo; clustering in the sense that
observations within each of its groups are sufficiently more similar to each
other than to observations assigned to different groups at that level. The
Gap statistic described earlier can be used for this purpose.
Recursive binary splitting/agglomeration can be represented by a rooted
</p>
<p>binary tree. The nodes of the trees represent groups. The root node repre-
sents the entire data set. The N terminal nodes each represent one of the
individual observations (singleton clusters). Each nonterminal node (&ldquo;par-
ent&rdquo;) has two daughter nodes. For divisive clustering the two daughters
represent the two groups resulting from the split of the parent; for agglom-
erative clustering the daughters represent the two groups that were merged
to form the parent.
Most agglomerative and some divisive methods (when viewed bottom-
</p>
<p>up) possess a monotonicity property. That is, the dissimilarity between
merged clusters is monotone increasing with the level of the merger. Thus
the binary tree can be plotted so that the height of each node is proportional
to the value of the intergroup dissimilarity between its two daughters. The
terminal nodes representing individual observations are all plotted at zero
height. This type of graphical display is called a dendrogram.
A dendrogram provides a highly interpretable complete description of
</p>
<p>the hierarchical clustering in a graphical format. This is one of the main
reasons for the popularity of hierarchical clustering methods.
For the microarray data, Figure 14.12 shows the dendrogram resulting
</p>
<p>from agglomerative clustering with average linkage; agglomerative cluster-
ing and this example are discussed in more detail later in this chapter.
Cutting the dendrogram horizontally at a particular height partitions the
data into disjoint clusters represented by the vertical lines that intersect
it. These are the clusters that would be produced by terminating the pro-
cedure when the optimal intergroup dissimilarity exceeds that threshold
cut value. Groups that merge at high values, relative to the merger values
of the subgroups contained within them lower in the tree, are candidates
for natural clusters. Note that this may occur at several different levels,
indicating a clustering hierarchy: that is, clusters nested within clusters.
Such a dendrogram is often viewed as a graphical summary of the data
</p>
<p>itself, rather than a description of the results of the algorithm. However,
such interpretations should be treated with caution. First, different hierar-
chical methods (see below), as well as small changes in the data, can lead
to quite different dendrograms. Also, such a summary will be valid only to
the extent that the pairwise observation dissimilarities possess the hierar-</p>
<p/>
</div>
<div class="page"><p/>
<p>522 14. Unsupervised Learning
</p>
<p>C
N
</p>
<p>S
C
</p>
<p>N
S
</p>
<p>C
N
</p>
<p>S
R
</p>
<p>E
N
</p>
<p>A
L
</p>
<p>B
R
</p>
<p>E
A
</p>
<p>S
T
</p>
<p>C
N
</p>
<p>SC
N
</p>
<p>S
</p>
<p>B
R
</p>
<p>E
A
</p>
<p>S
T
</p>
<p>N
S
</p>
<p>C
L
</p>
<p>C
</p>
<p>N
S
</p>
<p>C
L
</p>
<p>C
</p>
<p>R
E
</p>
<p>N
A
</p>
<p>L
R
</p>
<p>E
N
</p>
<p>A
L
</p>
<p>R
E
</p>
<p>N
A
</p>
<p>L
</p>
<p>R
E
</p>
<p>N
A
</p>
<p>L
R
</p>
<p>E
N
</p>
<p>A
L
</p>
<p>R
E
</p>
<p>N
A
</p>
<p>L
</p>
<p>R
E
</p>
<p>N
A
</p>
<p>L
</p>
<p>B
R
</p>
<p>E
A
</p>
<p>S
T
</p>
<p>N
S
</p>
<p>C
L
</p>
<p>C
</p>
<p>R
E
</p>
<p>N
A
</p>
<p>L
</p>
<p>U
N
</p>
<p>K
N
</p>
<p>O
W
</p>
<p>N
O
</p>
<p>V
A
</p>
<p>R
IA
</p>
<p>N
</p>
<p>M
E
</p>
<p>L
A
</p>
<p>N
O
</p>
<p>M
A
</p>
<p>P
R
</p>
<p>O
S
</p>
<p>T
A
</p>
<p>T
E
</p>
<p>O
V
</p>
<p>A
R
</p>
<p>IA
N
</p>
<p>O
V
</p>
<p>A
R
</p>
<p>IA
N
</p>
<p>O
V
</p>
<p>A
R
</p>
<p>IA
N
</p>
<p>O
V
</p>
<p>A
R
</p>
<p>IA
N
</p>
<p>O
V
</p>
<p>A
R
</p>
<p>IA
N
</p>
<p>P
R
</p>
<p>O
S
</p>
<p>T
A
</p>
<p>T
E
</p>
<p>N
S
</p>
<p>C
L
</p>
<p>C
N
</p>
<p>S
C
</p>
<p>L
C
</p>
<p>N
S
</p>
<p>C
L
</p>
<p>C
</p>
<p>L
E
</p>
<p>U
K
</p>
<p>E
M
</p>
<p>IA
</p>
<p>K
5
</p>
<p>6
2
</p>
<p>B
-r
</p>
<p>e
p
</p>
<p>ro
K
</p>
<p>5
6
</p>
<p>2
A
</p>
<p>-r
e
</p>
<p>p
ro
</p>
<p>L
E
</p>
<p>U
K
</p>
<p>E
M
</p>
<p>IA
</p>
<p>L
E
</p>
<p>U
K
</p>
<p>E
M
</p>
<p>IA
</p>
<p>L
E
</p>
<p>U
K
</p>
<p>E
M
</p>
<p>IA
L
</p>
<p>E
U
</p>
<p>K
E
</p>
<p>M
IA
</p>
<p>L
E
</p>
<p>U
K
</p>
<p>E
M
</p>
<p>IA
</p>
<p>C
O
</p>
<p>L
O
</p>
<p>N
C
</p>
<p>O
L
</p>
<p>O
N
</p>
<p>C
O
</p>
<p>L
O
</p>
<p>N
C
</p>
<p>O
L
</p>
<p>O
N
</p>
<p>C
O
</p>
<p>L
O
</p>
<p>N
</p>
<p>C
O
</p>
<p>L
O
</p>
<p>N
C
</p>
<p>O
L
</p>
<p>O
N
</p>
<p>M
C
</p>
<p>F
7
</p>
<p>A
-r
</p>
<p>e
p
</p>
<p>ro
B
</p>
<p>R
E
</p>
<p>A
S
</p>
<p>T
M
</p>
<p>C
F
</p>
<p>7
D
</p>
<p>-r
e
</p>
<p>p
ro
</p>
<p>B
R
</p>
<p>E
A
</p>
<p>S
T
</p>
<p>N
S
</p>
<p>C
L
</p>
<p>C
</p>
<p>N
S
</p>
<p>C
L
</p>
<p>C
N
</p>
<p>S
C
</p>
<p>L
C
</p>
<p>M
E
</p>
<p>L
A
</p>
<p>N
O
</p>
<p>M
A
</p>
<p>B
R
</p>
<p>E
A
</p>
<p>S
T
</p>
<p>B
R
</p>
<p>E
A
</p>
<p>S
T
</p>
<p>M
E
</p>
<p>L
A
</p>
<p>N
O
</p>
<p>M
A
</p>
<p>M
E
</p>
<p>L
A
</p>
<p>N
O
</p>
<p>M
A
</p>
<p>M
E
</p>
<p>L
A
</p>
<p>N
O
</p>
<p>M
A
</p>
<p>M
E
</p>
<p>L
A
</p>
<p>N
O
</p>
<p>M
A
</p>
<p>M
E
</p>
<p>L
A
</p>
<p>N
O
</p>
<p>M
A
</p>
<p>M
E
</p>
<p>L
A
</p>
<p>N
O
</p>
<p>M
A
</p>
<p>FIGURE 14.12. Dendrogram from agglomerative hierarchical clustering with
average linkage to the human tumor microarray data.
</p>
<p>chical structure produced by the algorithm. Hierarchical methods impose
hierarchical structure whether or not such structure actually exists in the
data.
The extent to which the hierarchical structure produced by a dendro-
</p>
<p>gram actually represents the data itself can be judged by the cophenetic
correlation coefficient. This is the correlation between the N(N&minus;1)/2 pair-
wise observation dissimilarities dii&prime; input to the algorithm and their corre-
sponding cophenetic dissimilarities Cii&prime; derived from the dendrogram. The
cophenetic dissimilarity Cii&prime; between two observations (i, i
</p>
<p>&prime;) is the inter-
group dissimilarity at which observations i and i&prime; are first joined together
in the same cluster.
The cophenetic dissimilarity is a very restrictive dissimilarity measure.
</p>
<p>First, the Cii&prime; over the observations must contain many ties, since onlyN&minus;1
of the total N(N &minus; 1)/2 values can be distinct. Also these dissimilarities
obey the ultrametric inequality
</p>
<p>Cii&prime; &le; max{Cik, Ci&prime;k} (14.40)</p>
<p/>
</div>
<div class="page"><p/>
<p>14.3 Cluster Analysis 523
</p>
<p>for any three observations (i, i&prime;, k). As a geometric example, suppose the
data were represented as points in a Euclidean coordinate system. In order
for the set of interpoint distances over the data to conform to (14.40), the
triangles formed by all triples of points must be isosceles triangles with the
unequal length no longer than the length of the two equal sides (Jain and
Dubes, 1988). Therefore it is unrealistic to expect general dissimilarities
over arbitrary data sets to closely resemble their corresponding cophenetic
dissimilarities as calculated from a dendrogram, especially if there are not
many tied values. Thus the dendrogram should be viewed mainly as a de-
scription of the clustering structure of the data as imposed by the particular
algorithm employed.
</p>
<p>Agglomerative Clustering
</p>
<p>Agglomerative clustering algorithms begin with every observation repre-
senting a singleton cluster. At each of the N &minus;1 steps the closest two (least
dissimilar) clusters are merged into a single cluster, producing one less clus-
ter at the next higher level. Therefore, a measure of dissimilarity between
two clusters (groups of observations) must be defined.
Let G and H represent two such groups. The dissimilarity d(G,H) be-
</p>
<p>tween G and H is computed from the set of pairwise observation dissim-
ilarities dii&prime; where one member of the pair i is in G and the other i
</p>
<p>&prime; is
in H. Single linkage (SL) agglomerative clustering takes the intergroup
dissimilarity to be that of the closest (least dissimilar) pair
</p>
<p>dSL(G,H) = min
i&isin;G
i&prime;&isin;H
</p>
<p>dii&prime; . (14.41)
</p>
<p>This is also often called the nearest-neighbor technique. Complete linkage
(CL) agglomerative clustering (furthest-neighbor technique) takes the in-
tergroup dissimilarity to be that of the furthest (most dissimilar) pair
</p>
<p>dCL(G,H) = max
i&isin;G
i&prime;&isin;H
</p>
<p>dii&prime; . (14.42)
</p>
<p>Group average (GA) clustering uses the average dissimilarity between the
groups
</p>
<p>dGA(G,H) =
1
</p>
<p>NGNH
</p>
<p>&sum;
</p>
<p>i&isin;G
</p>
<p>&sum;
</p>
<p>i&prime;&isin;H
</p>
<p>dii&prime; (14.43)
</p>
<p>where NG and NH are the respective number of observations in each group.
Although there have been many other proposals for defining intergroup
dissimilarity in the context of agglomerative clustering, the above three are
the ones most commonly used. Figure 14.13 shows examples of all three.
If the data dissimilarities {dii&prime;} exhibit a strong clustering tendency, with
</p>
<p>each of the clusters being compact and well separated from others, then all
three methods produce similar results. Clusters are compact if all of the</p>
<p/>
</div>
<div class="page"><p/>
<p>524 14. Unsupervised Learning
</p>
<p>Average Linkage Complete Linkage Single Linkage
</p>
<p>FIGURE 14.13. Dendrograms from agglomerative hierarchical clustering of hu-
man tumor microarray data.
</p>
<p>observations within them are relatively close together (small dissimilarities)
as compared with observations in different clusters. To the extent this is
not the case, results will differ.
Single linkage (14.41) only requires that a single dissimilarity dii&prime; , i &isin; G
</p>
<p>and i&prime; &isin; H, be small for two groups G and H to be considered close
together, irrespective of the other observation dissimilarities between the
groups. It will therefore have a tendency to combine, at relatively low
thresholds, observations linked by a series of close intermediate observa-
tions. This phenomenon, referred to as chaining, is often considered a de-
fect of the method. The clusters produced by single linkage can violate the
&ldquo;compactness&rdquo; property that all observations within each cluster tend to
be similar to one another, based on the supplied observation dissimilari-
ties {dii&prime;}. If we define the diameter DG of a group of observations as the
largest dissimilarity among its members
</p>
<p>DG = max
i&isin;G
i&prime;&isin;G
</p>
<p>dii&prime; , (14.44)
</p>
<p>then single linkage can produce clusters with very large diameters.
Complete linkage (14.42) represents the opposite extreme. Two groups
</p>
<p>G and H are considered close only if all of the observations in their union
are relatively similar. It will tend to produce compact clusters with small
diameters (14.44). However, it can produce clusters that violate the &ldquo;close-
ness&rdquo; property. That is, observations assigned to a cluster can be much</p>
<p/>
</div>
<div class="page"><p/>
<p>14.3 Cluster Analysis 525
</p>
<p>closer to members of other clusters than they are to some members of their
own cluster.
Group average clustering (14.43) represents a compromise between the
</p>
<p>two extremes of single and complete linkage. It attempts to produce rel-
atively compact clusters that are relatively far apart. However, its results
depend on the numerical scale on which the observation dissimilarities dii&prime;
</p>
<p>are measured. Applying a monotone strictly increasing transformation h(&middot;)
to the dii&prime; , hii&prime; = h(dii&prime;), can change the result produced by (14.43). In
contrast, (14.41) and (14.42) depend only on the ordering of the dii&prime; and
are thus invariant to such monotone transformations. This invariance is
often used as an argument in favor of single or complete linkage over group
average methods.
One can argue that group average clustering has a statistical consis-
</p>
<p>tency property violated by single and complete linkage. Assume we have
attribute-value data XT = (X1, . . . , Xp) and that each cluster k is a ran-
dom sample from some population joint density pk(x). The complete data
set is a random sample from a mixture of K such densities. The group
average dissimilarity dGA(G,H) (14.43) is an estimate of
</p>
<p>&int; &int;
d(x, x&prime;) pG(x) pH(x
</p>
<p>&prime;) dx dx&prime;, (14.45)
</p>
<p>where d(x, x&prime;) is the dissimilarity between points x and x&prime; in the space
of attribute values. As the sample size N approaches infinity dGA(G,H)
(14.43) approaches (14.45), which is a characteristic of the relationship
between the two densities pG(x) and pH(x) . For single linkage, dSL(G,H)
(14.41) approaches zero as N &rarr; &infin; independent of pG(x) and pH(x) . For
complete linkage, dCL(G,H) (14.42) becomes infinite as N &rarr; &infin;, again
independent of the two densities. Thus, it is not clear what aspects of the
population distribution are being estimated by dSL(G,H) and dCL(G,H).
</p>
<p>Example: Human Cancer Microarray Data (Continued)
</p>
<p>The left panel of Figure 14.13 shows the dendrogram resulting from average
linkage agglomerative clustering of the samples (columns) of the microarray
data. The middle and right panels show the result using complete and single
linkage. Average and complete linkage gave similar results, while single
linkage produced unbalanced groups with long thin clusters. We focus on
the average linkage clustering.
LikeK-means clustering, hierarchical clustering is successful at clustering
</p>
<p>simple cancers together. However it has other nice features. By cutting off
the dendrogram at various heights, different numbers of clusters emerge,
and the sets of clusters are nested within one another. Secondly, it gives
some partial ordering information about the samples. In Figure 14.14, we
have arranged the genes (rows) and samples (columns) of the expression
matrix in orderings derived from hierarchical clustering.</p>
<p/>
</div>
<div class="page"><p/>
<p>526 14. Unsupervised Learning
</p>
<p>Note that if we flip the orientation of the branches of a dendrogram at any
merge, the resulting dendrogram is still consistent with the series of hierar-
chical clustering operations. Hence to determine an ordering of the leaves,
we must add a constraint. To produce the row ordering of Figure 14.14,
we have used the default rule in S-PLUS: at each merge, the subtree with
the tighter cluster is placed to the left (toward the bottom in the rotated
dendrogram in the figure.) Individual genes are the tightest clusters possi-
ble, and merges involving two individual genes place them in order by their
observation number. The same rule was used for the columns. Many other
rules are possible&mdash;for example, ordering by a multidimensional scaling of
the genes; see Section 14.8.
The two-way rearrangement of Figure 14.14 produces an informative pic-
</p>
<p>ture of the genes and samples. This picture is more informative than the
randomly ordered rows and columns of Figure 1.3 of Chapter 1. Further-
more, the dendrograms themselves are useful, as biologists can, for example,
interpret the gene clusters in terms of biological processes.
</p>
<p>Divisive Clustering
</p>
<p>Divisive clustering algorithms begin with the entire data set as a single
cluster, and recursively divide one of the existing clusters into two daugh-
ter clusters at each iteration in a top-down fashion. This approach has not
been studied nearly as extensively as agglomerative methods in the cluster-
ing literature. It has been explored somewhat in the engineering literature
(Gersho and Gray, 1992) in the context of compression. In the clustering
setting, a potential advantage of divisive over agglomerative methods can
occur when interest is focused on partitioning the data into a relatively
small number of clusters.
</p>
<p>The divisive paradigm can be employed by recursively applying any of
the combinatorial methods such as K-means (Section 14.3.6) or K-medoids
(Section 14.3.10), with K = 2, to perform the splits at each iteration. How-
ever, such an approach would depend on the starting configuration specified
at each step. In addition, it would not necessarily produce a splitting se-
quence that possesses the monotonicity property required for dendrogram
representation.
A divisive algorithm that avoids these problems was proposed by Mac-
</p>
<p>naughton Smith et al. (1965). It begins by placing all observations in a
single cluster G. It then chooses that observation whose average dissimi-
larity from all the other observations is largest. This observation forms the
first member of a second cluster H. At each successive step that observation
in G whose average distance from those in H, minus that for the remaining
observations in G is largest, is transferred to H. This continues until the
corresponding difference in averages becomes negative. That is, there are
no longer any observations in G that are, on average, closer to those in
H. The result is a split of the original cluster into two daughter clusters,</p>
<p/>
</div>
<div class="page"><p/>
<p>14.3 Cluster Analysis 527
</p>
<p>FIGURE 14.14. DNA microarray data: average linkage hierarchical clustering
has been applied independently to the rows (genes) and columns (samples), de-
termining the ordering of the rows and columns (see text). The colors range from
bright green (negative, under-expressed) to bright red (positive, over-expressed).</p>
<p/>
</div>
<div class="page"><p/>
<p>528 14. Unsupervised Learning
</p>
<p>the observations transferred to H, and those remaining in G. These two
clusters represent the second level of the hierarchy. Each successive level
is produced by applying this splitting procedure to one of the clusters at
the previous level. Kaufman and Rousseeuw (1990) suggest choosing the
cluster at each level with the largest diameter (14.44) for splitting. An al-
ternative would be to choose the one with the largest average dissimilarity
among its members
</p>
<p>d̄G =
1
</p>
<p>NG
2
</p>
<p>&sum;
</p>
<p>i&isin;G
</p>
<p>&sum;
</p>
<p>i&prime;&isin;G
</p>
<p>dii&prime; .
</p>
<p>The recursive splitting continues until all clusters either become singletons
or all members of each one have zero dissimilarity from one another.
</p>
<p>14.4 Self-Organizing Maps
</p>
<p>This method can be viewed as a constrained version of K-means clustering,
in which the prototypes are encouraged to lie in a one- or two-dimensional
manifold in the feature space. The resulting manifold is also referred to
as a constrained topological map, since the original high-dimensional obser-
vations can be mapped down onto the two-dimensional coordinate system.
The original SOM algorithm was online&mdash;observations are processed one at
a time&mdash;and later a batch version was proposed. The technique also bears
a close relationship to principal curves and surfaces, which are discussed in
the next section.
We consider a SOM with a two-dimensional rectangular grid of K proto-
</p>
<p>types mj &isin; IRp (other choices, such as hexagonal grids, can also be used).
Each of the K prototypes are parametrized with respect to an integer
coordinate pair ℓj &isin; Q1 &times;Q2. Here Q1 = {1, 2, . . . , q1}, similarly Q2, and
K = q1&middot;q2. Themj are initialized, for example, to lie in the two-dimensional
principal component plane of the data (next section). We can think of the
prototypes as &ldquo;buttons,&rdquo; &ldquo;sewn&rdquo; on the principal component plane in a
regular pattern. The SOM procedure tries to bend the plane so that the
buttons approximate the data points as well as possible. Once the model is
fit, the observations can be mapped down onto the two-dimensional grid.
The observations xi are processed one at a time. We find the closest
</p>
<p>prototype mj to xi in Euclidean distance in IR
p, and then for all neighbors
</p>
<p>mk of mj , move mk toward xi via the update
</p>
<p>mk &larr; mk + α(xi &minus;mk). (14.46)
</p>
<p>The &ldquo;neighbors&rdquo; of mj are defined to be all mk such that the distance
between ℓj and ℓk is small. The simplest approach uses Euclidean distance,
and &ldquo;small&rdquo; is determined by a threshold r. This neighborhood always
includes the closest prototype mj itself.</p>
<p/>
</div>
<div class="page"><p/>
<p>14.4 Self-Organizing Maps 529
</p>
<p>Notice that distance is defined in the space Q1&times;Q2 of integer topological
coordinates of the prototypes, rather than in the feature space IRp. The
effect of the update (14.46) is to move the prototypes closer to the data,
but also to maintain a smooth two-dimensional spatial relationship between
the prototypes.
The performance of the SOM algorithm depends on the learning rate
</p>
<p>α and the distance threshold r. Typically α is decreased from say 1.0 to
0.0 over a few thousand iterations (one per observation). Similarly r is
decreased linearly from starting value R to 1 over a few thousand iterations.
We illustrate a method for choosing R in the example below.
We have described the simplest version of the SOM. More sophisticated
</p>
<p>versions modify the update step according to distance:
</p>
<p>mk &larr; mk + αh(‖ℓj &minus; ℓk‖)(xi &minus;mk), (14.47)
</p>
<p>where the neighborhood function h gives more weight to prototypesmk with
indices ℓk closer to ℓj than to those further away.
If we take the distance r small enough so that each neighborhood contains
</p>
<p>only one point, then the spatial connection between prototypes is lost. In
that case one can show that the SOM algorithm is an online version of
K-means clustering, and eventually stabilizes at one of the local minima
found by K-means. Since the SOM is a constrained version of K-means
clustering, it is important to check whether the constraint is reasonable
in any given problem. One can do this by computing the reconstruction
error ‖x &minus; mj‖2, summed over observations, for both methods. This will
necessarily be smaller for K-means, but should not be much smaller if the
SOM is a reasonable approximation.
As an illustrative example, we generated 90 data points in three dimen-
</p>
<p>sions, near the surface of a half sphere of radius 1. The points were in each
of three clusters&mdash;red, green, and blue&mdash;located near (0, 1, 0), (0, 0, 1) and
(1, 0, 0). The data are shown in Figure 14.15
</p>
<p>By design, the red cluster was much tighter than the green or blue ones.
(Full details of the data generation are given in Exercise 14.5.) A 5&times;5 grid
of prototypes was used, with initial grid size R = 2; this meant that about
a third of the prototypes were initially in each neighborhood. We did a
total of 40 passes through the dataset of 90 observations, and let r and α
decrease linearly over the 3600 iterations.
In Figure 14.16 the prototypes are indicated by circles, and the points
</p>
<p>that project to each prototype are plotted randomly within the correspond-
ing circle. The left panel shows the initial configuration, while the right
panel shows the final one. The algorithm has succeeded in separating the
clusters; however, the separation of the red cluster indicates that the man-
ifold has folded back on itself (see Figure 14.17). Since the distances in the
two-dimensional display are not used, there is little indication in the SOM
projection that the red cluster is tighter than the others.</p>
<p/>
</div>
<div class="page"><p/>
<p>530 14. Unsupervised Learning
</p>
<p>&minus;1
</p>
<p>&minus;0.5
</p>
<p>0
</p>
<p>0.5
</p>
<p>1
</p>
<p>1.5
</p>
<p>&minus;1
</p>
<p>&minus;0.5
</p>
<p>0
</p>
<p>0.5
</p>
<p>1
</p>
<p>1.5
&minus;1
</p>
<p>&minus;0.5
</p>
<p>0
</p>
<p>0.5
</p>
<p>1
</p>
<p>1.5
</p>
<p>FIGURE 14.15. Simulated data in three classes, near the surface of a half&ndash;
sphere.
</p>
<p>&bull;&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;&bull;&bull; &bull;
</p>
<p>&bull; &bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull; &bull;&bull;&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull; &bull;
&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>1 2 3 4 5
</p>
<p>1
</p>
<p>2
</p>
<p>3
</p>
<p>4
</p>
<p>5
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull; &bull;
&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull; &bull;&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
&bull; &bull;
</p>
<p>&bull;&bull; &bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>1 2 3 4 5
</p>
<p>1
</p>
<p>2
</p>
<p>3
</p>
<p>4
</p>
<p>5
</p>
<p>FIGURE 14.16. Self-organizing map applied to half-sphere data example. Left
panel is the initial configuration, right panel the final one. The 5 &times; 5 grid of
prototypes are indicated by circles, and the points that project to each prototype
are plotted randomly within the corresponding circle.</p>
<p/>
</div>
<div class="page"><p/>
<p>14.4 Self-Organizing Maps 531
</p>
<p>FIGURE 14.17. Wiremesh representation of the fitted SOM model in IR3. The
lines represent the horizontal and vertical edges of the topological lattice. The
double lines indicate that the surface was folded diagonally back on itself in order
to model the red points. The cluster members have been jittered to indicate their
color, and the purple points are the node centers.
</p>
<p>Figure 14.18 shows the reconstruction error, equal to the total sum of
squares of each data point around its prototype. For comparison we carried
out a K-means clustering with 25 centroids, and indicate its reconstruction
error by the horizontal line on the graph. We see that the SOM significantly
decreases the error, nearly to the level of the K-means solution. This pro-
vides evidence that the two-dimensional constraint used by the SOM is
reasonable for this particular dataset.
In the batch version of the SOM, we update each mj via
</p>
<p>mj =
</p>
<p>&sum;
wkxk&sum;
wk
</p>
<p>. (14.48)
</p>
<p>The sum is over points xk that mapped (i.e., were closest to) neighbors mk
of mj . The weight function may be rectangular, that is, equal to 1 for the
neighbors ofmk, or may decrease smoothly with distance ‖ℓk&minus;ℓj‖ as before.
If the neighborhood size is chosen small enough so that it consists only
of mk, with rectangular weights, this reduces to the K-means clustering
procedure described earlier. It can also be thought of as a discrete version
of principal curves and surfaces, described in Section 14.5.</p>
<p/>
</div>
<div class="page"><p/>
<p>532 14. Unsupervised Learning
</p>
<p>Iteration
</p>
<p>R
e
c
o
n
s
tr
</p>
<p>u
c
ti
o
n
 E
</p>
<p>rr
o
r
</p>
<p>0 500 1000 1500 2000 2500
</p>
<p>0
1
0
</p>
<p>2
0
</p>
<p>3
0
</p>
<p>4
0
</p>
<p>5
0
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
&bull;
&bull;
&bull;
&bull;
&bull;
&bull;
&bull;
&bull;
&bull;
&bull;
&bull;&bull;&bull;
</p>
<p>&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
</p>
<p>FIGURE 14.18. Half-sphere data: reconstruction error for the SOM as a func-
tion of iteration. Error for k-means clustering is indicated by the horizontal line.
</p>
<p>Example: Document Organization and Retrieval
</p>
<p>Document retrieval has gained importance with the rapid development of
the Internet and the Web, and SOMs have proved to be useful for organiz-
ing and indexing large corpora. This example is taken from the WEBSOM
homepage http://websom.hut.fi/ (Kohonen et al., 2000). Figure 14.19 rep-
resents a SOM fit to 12,088 newsgroup comp.ai.neural-nets articles. The
labels are generated automatically by the WEBSOM software and provide
a guide as to the typical content of a node.
In applications such as this, the documents have to be preprocessed in
</p>
<p>order to create a feature vector. A term-document matrix is created, where
each row represents a single document. The entries in each row are the
relative frequency of each of a predefined set of terms. These terms could
be a large set of dictionary entries (50,000 words), or an even larger set
of bigrams (word pairs), or subsets of these. These matrices are typically
very sparse, and so often some preprocessing is done to reduce the number
of features (columns). Sometimes the SVD (next section) is used to reduce
the matrix; Kohonen et al. (2000) use a randomized variant thereof. These
reduced vectors are then the input to the SOM.</p>
<p/>
</div>
<div class="page"><p/>
<p>14.4 Self-Organizing Maps 533
</p>
<p>FIGURE 14.19. Heatmap representation of the SOM model fit to a corpus
of 12,088 newsgroup comp.ai.neural-nets contributions (courtesy WEBSOM
homepage). The lighter areas indicate higher-density areas. Populated nodes are
automatically labeled according to typical content.</p>
<p/>
</div>
<div class="page"><p/>
<p>534 14. Unsupervised Learning
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>v1v1v1v1v1v1v1v1
</p>
<p>ui1d1ui1d1ui1d1ui1d1ui1d1ui1d1ui1d1ui1d1
</p>
<p>xixixixixixixixi
</p>
<p>FIGURE 14.20. The first linear principal component of a set of data. The line
minimizes the total squared distance from each point to its orthogonal projection
onto the line.
</p>
<p>In this application the authors have developed a &ldquo;zoom&rdquo; feature, which
allows one to interact with the map in order to get more detail. The final
level of zooming retrieves the actual news articles, which can then be read.
</p>
<p>14.5 Principal Components, Curves and Surfaces
</p>
<p>Principal components are discussed in Sections 3.4.1, where they shed light
on the shrinkage mechanism of ridge regression. Principal components are
a sequence of projections of the data, mutually uncorrelated and ordered
in variance. In the next section we present principal components as linear
manifolds approximating a set of N points xi &isin; IRp. We then present
some nonlinear generalizations in Section 14.5.2. Other recent proposals
for nonlinear approximating manifolds are discussed in Section 14.9.
</p>
<p>14.5.1 Principal Components
</p>
<p>The principal components of a set of data in IRp provide a sequence of best
linear approximations to that data, of all ranks q &le; p.
</p>
<p>Denote the observations by x1, x2, . . . , xN , and consider the rank-q linear
model for representing them</p>
<p/>
</div>
<div class="page"><p/>
<p>14.5 Principal Components, Curves and Surfaces 535
</p>
<p>f(λ) = μ+Vqλ, (14.49)
</p>
<p>where μ is a location vector in IRp, Vq is a p&times; q matrix with q orthogonal
unit vectors as columns, and λ is a q vector of parameters. This is the
parametric representation of an affine hyperplane of rank q. Figures 14.20
and 14.21 illustrate for q = 1 and q = 2, respectively. Fitting such a model
to the data by least squares amounts to minimizing the reconstruction error
</p>
<p>min
μ,{λi}, Vq
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>‖xi &minus; μ&minus;Vqλi‖2. (14.50)
</p>
<p>We can partially optimize for μ and the λi (Exercise 14.7) to obtain
</p>
<p>μ̂ = x̄, (14.51)
</p>
<p>λ̂i = V
T
q (xi &minus; x̄). (14.52)
</p>
<p>This leaves us to find the orthogonal matrix Vq:
</p>
<p>min
Vq
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>||(xi &minus; x̄)&minus;VqVTq (xi &minus; x̄)||2. (14.53)
</p>
<p>For convenience we assume that x̄ = 0 (otherwise we simply replace the
observations by their centered versions x̃i = xi &minus; x̄). The p &times; p matrix
Hq = VqV
</p>
<p>T
q is a projection matrix, and maps each point xi onto its rank-
</p>
<p>q reconstruction Hqxi, the orthogonal projection of xi onto the subspace
spanned by the columns of Vq. The solution can be expressed as follows.
Stack the (centered) observations into the rows of an N &times; p matrix X. We
construct the singular value decomposition of X:
</p>
<p>X = UDVT . (14.54)
</p>
<p>This is a standard decomposition in numerical analysis, and many algo-
rithms exist for its computation (Golub and Van Loan, 1983, for example).
Here U is an N &times; p orthogonal matrix (UTU = Ip) whose columns uj are
called the left singular vectors; V is a p&times;p orthogonal matrix (VTV = Ip)
with columns vj called the right singular vectors, and D is a p&times;p diagonal
matrix, with diagonal elements d1 &ge; d2 &ge; &middot; &middot; &middot; &ge; dp &ge; 0 known as the sin-
gular values. For each rank q, the solution Vq to (14.53) consists of the first
q columns of V. The columns of UD are called the principal components
of X (see Section 3.5.1). The N optimal λ̂i in (14.52) are given by the first
q principal components (the N rows of the N &times; q matrix UqDq).
The one-dimensional principal component line in IR2 is illustrated in Fig-
</p>
<p>ure 14.20. For each data point xi, there is a closest point on the line, given
by ui1d1v1. Here v1 is the direction of the line and λ̂i = ui1d1 measures
distance along the line from the origin. Similarly Figure 14.21 shows the</p>
<p/>
</div>
<div class="page"><p/>
<p>536 14. Unsupervised Learning
</p>
<p>First principal component
S
</p>
<p>e
c
o
n
d
 p
</p>
<p>ri
n
c
ip
</p>
<p>a
l 
c
o
m
</p>
<p>p
o
n
e
n
t
</p>
<p>&minus;1.0 &minus;0.5 0.0 0.5 1.0
</p>
<p>&minus;1
.0
</p>
<p>&minus;0
.5
</p>
<p>0
.0
</p>
<p>0
.5
</p>
<p>1
.0
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull; &bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull; &bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>FIGURE 14.21. The best rank-two linear approximation to the half-sphere data.
</p>
<p>The right panel shows the projected points with coordinates given by U2D2, the
</p>
<p>first two principal components of the data.
</p>
<p>two-dimensional principal component surface fit to the half-sphere data
(left panel). The right panel shows the projection of the data onto the
first two principal components. This projection was the basis for the initial
configuration for the SOM method shown earlier. The procedure is quite
successful at separating the clusters. Since the half-sphere is nonlinear, a
nonlinear projection will do a better job, and this is the topic of the next
section.
Principal components have many other nice properties, for example, the
</p>
<p>linear combination Xv1 has the highest variance among all linear com-
binations of the features; Xv2 has the highest variance among all linear
combinations satisfying v2 orthogonal to v1, and so on.
</p>
<p>Example: Handwritten Digits
</p>
<p>Principal components are a useful tool for dimension reduction and com-
pression. We illustrate this feature on the handwritten digits data described
in Chapter 1. Figure 14.22 shows a sample of 130 handwritten 3&rsquo;s, each a
digitized 16 &times; 16 grayscale image, from a total of 658 such 3&rsquo;s. We see
considerable variation in writing styles, character thickness and orienta-
tion. We consider these images as points xi in IR
</p>
<p>256, and compute their
principal components via the SVD (14.54).
Figure 14.23 shows the first two principal components of these data. For
</p>
<p>each of these first two principal components ui1d1 and ui2d2, we computed
the 5%, 25%, 50%, 75% and 95% quantile points, and used them to define
the rectangular grid superimposed on the plot. The circled points indicate</p>
<p/>
</div>
<div class="page"><p/>
<p>14.5 Principal Components, Curves and Surfaces 537
</p>
<p>FIGURE 14.22. A sample of 130 handwritten 3&rsquo;s shows a variety of writing
styles.
</p>
<p>those images close to the vertices of the grid, where the distance measure
focuses mainly on these projected coordinates, but gives some weight to the
components in the orthogonal subspace. The right plot shows the images
corresponding to these circled points. This allows us to visualize the nature
of the first two principal components. We see that the v1 (horizontal move-
ment) mainly accounts for the lengthening of the lower tail of the three,
while v2 (vertical movement) accounts for character thickness. In terms of
the parametrized model (14.49), this two-component model has the form
</p>
<p>f̂(λ) = x̄+ λ1v1 + λ2v2
</p>
<p>= + λ1 &middot; + λ2 &middot; . (14.55)
</p>
<p>Here we have displayed the first two principal component directions, v1
and v2, as images. Although there are a possible 256 principal components,
approximately 50 account for 90% of the variation in the threes, 12 ac-
count for 63%. Figure 14.24 compares the singular values to those obtained
for equivalent uncorrelated data, obtained by randomly scrambling each
column of X. The pixels in a digitized image are inherently correlated,
and since these are all the same digit the correlations are even stronger.</p>
<p/>
</div>
<div class="page"><p/>
<p>538 14. Unsupervised Learning
</p>
<p>First Principal Component
</p>
<p>S
e
</p>
<p>c
o
</p>
<p>n
d
</p>
<p> P
ri
n
</p>
<p>c
ip
</p>
<p>a
l 
C
</p>
<p>o
m
</p>
<p>p
o
</p>
<p>n
e
</p>
<p>n
t
</p>
<p>-6 -4 -2 0 2 4 6 8
</p>
<p>-5
0
</p>
<p>5
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull; &bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>O O O O
</p>
<p>O
</p>
<p>O O O
O
</p>
<p>O
</p>
<p>O
O
</p>
<p>O O O
</p>
<p>O O O O
O
</p>
<p>O O O O
O
</p>
<p>FIGURE 14.23. (Left panel:) the first two principal components of the hand-
written threes. The circled points are the closest projected images to the vertices
of a grid, defined by the marginal quantiles of the principal components. (Right
panel:) The images corresponding to the circled points. These show the nature of
the first two principal components.
</p>
<p>Dimension
</p>
<p>S
in
</p>
<p>g
u
la
</p>
<p>r 
V
</p>
<p>a
lu
</p>
<p>e
s
</p>
<p>0 50 100 150 200 250
</p>
<p>0
2
0
</p>
<p>4
0
</p>
<p>6
0
</p>
<p>8
0
</p>
<p>&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
</p>
<p>&bull; Real Trace
&bull; Randomized Trace
</p>
<p>FIGURE 14.24. The 256 singular values for the digitized threes, compared to
those for a randomized version of the data (each column of X was scrambled).</p>
<p/>
</div>
<div class="page"><p/>
<p>14.5 Principal Components, Curves and Surfaces 539
</p>
<p>A relatively small subset of the principal components serve as excellent
lower-dimensional features for representing the high-dimensional data.
</p>
<p>Example: Procrustes Transformations and Shape Averaging
</p>
<p>FIGURE 14.25. (Left panel:) Two different digitized handwritten Ss, each rep-
resented by 96 corresponding points in IR2. The green S has been deliberately
rotated and translated for visual effect. (Right panel:) A Procrustes transforma-
tion applies a translation and rotation to best match up the two set of points.
</p>
<p>Figure 14.25 represents two sets of points, the orange and green, in the
same plot. In this instance these points represent two digitized versions
of a handwritten S, extracted from the signature of a subject &ldquo;Suresh.&rdquo;
Figure 14.26 shows the entire signatures from which these were extracted
(third and fourth panels). The signatures are recorded dynamically using
touch-screen devices, familiar sights in modern supermarkets. There are
N = 96 points representing each S, which we denote by the N &times;2 matrices
X1 and X2. There is a correspondence between the points&mdash;the ith rows
of X1 and X2 are meant to represent the same positions along the two S&rsquo;s.
In the language of morphometrics, these points represent landmarks on
the two objects. How one finds such corresponding landmarks is in general
difficult and subject specific. In this particular case we used dynamic time
warping of the speed signal along each signature (Hastie et al., 1992), but
will not go into details here.
In the right panel we have applied a translation and rotation to the green
</p>
<p>points so as best to match the orange&mdash;a so-called Procrustes3 transforma-
tion (Mardia et al., 1979, for example).
Consider the problem
</p>
<p>min
μ,R
</p>
<p>||X2 &minus; (X1R+ 1μT )||F , (14.56)
</p>
<p>3Procrustes was an African bandit in Greek mythology, who stretched or squashed
</p>
<p>his visitors to fit his iron bed (eventually killing them).</p>
<p/>
</div>
<div class="page"><p/>
<p>540 14. Unsupervised Learning
</p>
<p>with X1 and X2 both N &times; p matrices of corresponding points, R an or-
thonormal p &times; p matrix4, and μ a p-vector of location coordinates. Here
||X||2F = trace(XTX) is the squared Frobenius matrix norm.
</p>
<p>Let x̄1 and x̄2 be the column mean vectors of the matrices, and X̃1 and
X̃2 be the versions of these matrices with the means removed. Consider
the SVD X̃T1 X̃2 = UDV
</p>
<p>T . Then the solution to (14.56) is given by (Exer-
cise 14.8)
</p>
<p>R̂ = UVT
</p>
<p>μ̂ = x̄2 &minus; R̂x̄1,
(14.57)
</p>
<p>and the minimal distances is referred to as the Procrustes distance. From
the form of the solution, we can center each matrix at its column centroid,
and then ignore location completely. Hereafter we assume this is the case.
The Procrustes distance with scaling solves a slightly more general
</p>
<p>problem,
min
β,R
</p>
<p>||X2 &minus; βX1R||F , (14.58)
</p>
<p>where β &gt; 0 is a positive scalar. The solution for R is as before, with
β̂ = trace(D)/||X1||2F .
Related to Procrustes distance is the Procrustes average of a collection
</p>
<p>of L shapes, which solves the problem
</p>
<p>min
{Rℓ}L1 ,M
</p>
<p>L&sum;
</p>
<p>ℓ=1
</p>
<p>||XℓRℓ &minus;M||2F ; (14.59)
</p>
<p>that is, find the shape M closest in average squared Procrustes distance to
all the shapes. This is solved by a simple alternating algorithm:
</p>
<p>0. Initialize M = X1 (for example).
</p>
<p>1. Solve the L Procrustes rotation problems with M fixed, yielding
X&prime;ℓ &larr; XR̂ℓ.
</p>
<p>2. Let M &larr; 1L
&sum;L
</p>
<p>ℓ=1 X
&prime;
ℓ.
</p>
<p>Steps 1. and 2. are repeated until the criterion (14.59) converges.
</p>
<p>Figure 14.26 shows a simple example with three shapes. Note that we can
only expect a solution up to a rotation; alternatively, we can impose a
constraint, such as that M be upper-triangular, to force uniqueness. One
can easily incorporate scaling in the definition (14.59); see Exercise 14.9.
Most generally we can define the affine-invariant average of a set of
</p>
<p>shapes via
</p>
<p>4To simplify matters, we consider only orthogonal matrices which include reflections
as well as rotations [theO(p) group]; although reflections are unlikely here, these methods
</p>
<p>can be restricted further to allow only rotations [SO(p) group].</p>
<p/>
</div>
<div class="page"><p/>
<p>14.5 Principal Components, Curves and Surfaces 541
</p>
<p>FIGURE 14.26. The Procrustes average of three versions of the leading S in
Suresh&rsquo;s signatures. The left panel shows the preshape average, with each of the
shapes X&prime;ℓ in preshape space superimposed. The right three panels map the pre-
shape M separately to match each of the original S&rsquo;s.
</p>
<p>min
{Aℓ}L1 ,M
</p>
<p>L&sum;
</p>
<p>ℓ=1
</p>
<p>||XℓAℓ &minus;M||2F , (14.60)
</p>
<p>where the Aℓ are any p&times; p nonsingular matrices. Here we require a stan-
dardization, such as MTM = I, to avoid a trivial solution. The solution is
attractive, and can be computed without iteration (Exercise 14.10):
</p>
<p>1. Let Hℓ = Xℓ(X
T
ℓ Xℓ)
</p>
<p>&minus;1XTℓ be the rank-p projection matrix defined
by Xℓ.
</p>
<p>2. M is the N&times;p matrix formed from the p largest eigenvectors of H̄ =
1
L
</p>
<p>&sum;L
ℓ=1 Hℓ.
</p>
<p>14.5.2 Principal Curves and Surfaces
</p>
<p>Principal curves generalize the principal component line, providing a smooth
one-dimensional curved approximation to a set of data points in IRp. A prin-
cipal surface is more general, providing a curved manifold approximation
of dimension 2 or more.
We will first define principal curves for random variables X &isin; IRp, and
</p>
<p>then move to the finite data case. Let f(λ) be a parameterized smooth
curve in IRp. Hence f(λ) is a vector function with p coordinates, each a
smooth function of the single parameter λ. The parameter λ can be chosen,
for example, to be arc-length along the curve from some fixed origin. For
each data value x, let λf (x) define the closest point on the curve to x. Then
f(λ) is called a principal curve for the distribution of the random vector
X if
</p>
<p>f(λ) = E(X|λf (X) = λ). (14.61)
</p>
<p>This says f(λ) is the average of all data points that project to it, that is, the
points for which it is &ldquo;responsible.&rdquo; This is also known as a self-consistency
property. Although in practice, continuous multivariate distributes have
infinitely many principal curves (Duchamp and Stuetzle, 1996), we are</p>
<p/>
</div>
<div class="page"><p/>
<p>542 14. Unsupervised Learning
</p>
<p>....
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull; &bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>.....
</p>
<p>f(λ) = [f1(λ), f2(λ)]f(λ) = [f1(λ), f2(λ)]f(λ) = [f1(λ), f2(λ)]f(λ) = [f1(λ), f2(λ)]f(λ) = [f1(λ), f2(λ)]f(λ) = [f1(λ), f2(λ)]f(λ) = [f1(λ), f2(λ)]f(λ) = [f1(λ), f2(λ)]f(λ) = [f1(λ), f2(λ)]
</p>
<p>FIGURE 14.27. The principal curve of a set of data. Each point on the curve
is the average of all data points that project there.
</p>
<p>interested mainly in the smooth ones. A principal curve is illustrated in
Figure 14.27.
Principal points are an interesting related concept. Consider a set of k
</p>
<p>prototypes and for each point x in the support of a distribution, identify
the closest prototype, that is, the prototype that is responsible for it. This
induces a partition of the feature space into so-called Voronoi regions. The
set of k points that minimize the expected distance from X to its prototype
are called the principal points of the distribution. Each principal point is
self-consistent, in that it equals the mean of X in its Voronoi region. For
example, with k = 1, the principal point of a circular normal distribution is
the mean vector; with k = 2 they are a pair of points symmetrically placed
on a ray through the mean vector. Principal points are the distributional
analogs of centroids found by K-means clustering. Principal curves can be
viewed as k = &infin; principal points, but constrained to lie on a smooth curve,
in a similar way that a SOM constrains K-means cluster centers to fall on
a smooth manifold.
To find a principal curve f(λ) of a distribution, we consider its coordinate
</p>
<p>functions f(λ) = [f1(λ), f2(λ), . . . , fp(λ)] and let X
T = (X1, X2, . . . , Xp).
</p>
<p>Consider the following alternating steps:
</p>
<p>(a) f̂j(λ) &larr; E(Xj |λ(X) = λ); j = 1, 2, . . . , p,
(b) λ̂f (x) &larr; argminλ&prime; ||x&minus; f̂(λ&prime;)||2.
</p>
<p>(14.62)
</p>
<p>The first equation fixes λ and enforces the self-consistency requirement
(14.61). The second equation fixes the curve and finds the closest point on</p>
<p/>
</div>
<div class="page"><p/>
<p>14.5 Principal Components, Curves and Surfaces 543
</p>
<p>-0.1 0.0 0.1 0.2
</p>
<p>-0
.2
</p>
<p>-0
.1
</p>
<p>0
.0
</p>
<p>0
.1
</p>
<p>0
.2
</p>
<p>&bull;
&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;&bull;
</p>
<p>&bull;&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;&bull;
</p>
<p>λ1
λ
2
</p>
<p>FIGURE 14.28. Principal surface fit to half-sphere data. (Left panel:) fitted
two-dimensional surface. (Right panel:) projections of data points onto the sur-
face, resulting in coordinates λ̂1, λ̂2.
</p>
<p>the curve to each data point. With finite data, the principal curve algorithm
starts with the linear principal component, and iterates the two steps in
(14.62) until convergence. A scatterplot smoother is used to estimate the
conditional expectations in step (a) by smoothing each Xj as a function of
</p>
<p>the arc-length λ̂(X), and the projection in (b) is done for each of the ob-
served data points. Proving convergence in general is difficult, but one can
show that if a linear least squares fit is used for the scatterplot smoothing,
then the procedure converges to the first linear principal component, and
is equivalent to the power method for finding the largest eigenvector of a
matrix.
Principal surfaces have exactly the same form as principal curves, but
</p>
<p>are of higher dimension. The mostly commonly used is the two-dimensional
principal surface, with coordinate functions
</p>
<p>f(λ1, λ2) = [f1(λ1, λ2), . . . , fp(λ1, λ2)].
</p>
<p>The estimates in step (a) above are obtained from two-dimensional surface
smoothers. Principal surfaces of dimension greater than two are rarely used,
since the visualization aspect is less attractive, as is smoothing in high
dimensions.
Figure 14.28 shows the result of a principal surface fit to the half-sphere
</p>
<p>data. Plotted are the data points as a function of the estimated nonlinear
coordinates λ̂1(xi), λ̂2(xi). The class separation is evident.
Principal surfaces are very similar to self-organizing maps. If we use a
</p>
<p>kernel surface smoother to estimate each coordinate function fj(λ1, λ2),
this has the same form as the batch version of SOMs (14.48). The SOM
weights wk are just the weights in the kernel. There is a difference, however:</p>
<p/>
</div>
<div class="page"><p/>
<p>544 14. Unsupervised Learning
</p>
<p>the principal surface estimates a separate prototype f(λ1(xi), λ2(xi)) for
each data point xi, while the SOM shares a smaller number of prototypes
for all data points. As a result, the SOM and principal surface will agree
only as the number of SOM prototypes grows very large.
There also is a conceptual difference between the two. Principal sur-
</p>
<p>faces provide a smooth parameterization of the entire manifold in terms
of its coordinate functions, while SOMs are discrete and produce only the
estimated prototypes for approximating the data. The smooth parameter-
ization in principal surfaces preserves distance locally: in Figure 14.28 it
reveals that the red cluster is tighter than the green or blue clusters. In
simple examples the estimates coordinate functions themselves can be in-
formative: see Exercise 14.13.
</p>
<p>14.5.3 Spectral Clustering
</p>
<p>Traditional clustering methods like K-means use a spherical or elliptical
metric to group data points. Hence they will not work well when the clus-
ters are non-convex, such as the concentric circles in the top left panel of
Figure 14.29. Spectral clustering is a generalization of standard clustering
methods, and is designed for these situations. It has close connections with
the local multidimensional-scaling techniques (Section 14.9) that generalize
MDS.
The starting point is a N &times;N matrix of pairwise similarities sii&prime; &ge; 0 be-
</p>
<p>tween all observation pairs. We represent the observations in an undirected
similarity graph G = 〈V, E〉. The N vertices vi represent the observations,
and pairs of vertices are connected by an edge if their similarity is positive
(or exceeds some threshold). The edges are weighted by the sii&prime; . Clustering
is now rephrased as a graph-partition problem, where we identify connected
components with clusters. We wish to partition the graph, such that edges
between different groups have low weight, and within a group have high
weight. The idea in spectral clustering is to construct similarity graphs that
represent the local neighborhood relationships between observations.
To make things more concrete, consider a set ofN points xi &isin; IRp, and let
</p>
<p>dii&prime; be the Euclidean distance between xi and xi&prime; . We will use as similarity
matrix the radial-kernel gram matrix; that is, sii&prime; = exp(&minus;d2ii&prime;/c), where
c &gt; 0 is a scale parameter.
There are many ways to define a similarity matrix and its associated
</p>
<p>similarity graph that reflect local behavior. The most popular is the mutual
K-nearest-neighbor graph. Define NK to be the symmetric set of nearby
pairs of points; specifically a pair (i, i&prime;) is in NK if point i is among the
K-nearest neighbors of i&prime;, or vice-versa. Then we connect all symmetric
nearest neighbors, and give them edge weight wii&prime; = sii&prime; ; otherwise the
edge weight is zero. Equivalently we set to zero all the pairwise similarities
not in NK , and draw the graph for this modified similarity matrix.</p>
<p/>
</div>
<div class="page"><p/>
<p>14.5 Principal Components, Curves and Surfaces 545
</p>
<p>Alternatively, a fully connected graph includes all pairwise edges with
weights wii&prime; = sii&prime; , and the local behavior is controlled by the scale param-
eter c.
The matrix of edge weights W = {wii&prime;} from a similarity graph is called
</p>
<p>the adjacency matrix. The degree of vertex i is gi =
&sum;
</p>
<p>i&prime; wii&prime; , the sum of
the weights of the edges connected to it. Let G be a diagonal matrix with
diagonal elements gi.
Finally, the graph Laplacian is defined by
</p>
<p>L = G&minus;W (14.63)
</p>
<p>This is called the unnormalized graph Laplacian; a number of normalized
versions have been proposed&mdash;these standardize the Laplacian with respect
to the node degrees gi, for example, L̃ = I&minus;G&minus;1W.
Spectral clustering finds the m eigenvectors ZN&times;m corresponding to the
</p>
<p>m smallest eigenvalues of L (ignoring the trivial constant eigenvector).
Using a standard method like K-means, we then cluster the rows of Z to
yield a clustering of the original data points.
An example is presented in Figure 14.29. The top left panel shows 450
</p>
<p>simulated data points in three circular clusters indicated by the colors. K-
means clustering would clearly have difficulty identifying the outer clusters.
We applied spectral clustering using a 10-nearest neighbor similarity graph,
and display the eigenvector corresponding to the second and third smallest
eigenvalue of the graph Laplacian in the lower left. The 15 smallest eigen-
values are shown in the top right panel. The two eigenvectors shown have
identified the three clusters, and a scatterplot of the rows of the eigenvector
matrix Y in the bottom right clearly separates the clusters. A procedure
such as K-means clustering applied to these transformed points would eas-
ily identify the three groups.
Why does spectral clustering work? For any vector f we have
</p>
<p>fTLf =
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>gif
2
i &minus;
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>N&sum;
</p>
<p>i&prime;=1
</p>
<p>fifi&prime;wii&prime;
</p>
<p>=
1
</p>
<p>2
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>N&sum;
</p>
<p>i&prime;=1
</p>
<p>wii&prime;(fi &minus; fi&prime;)2. (14.64)
</p>
<p>Formula 14.64 suggests that a small value of fTLf will be achieved if pairs
of points with large adjacencies have coordinates fi and fi&prime; close together.
Since 1TL1 = 0 for any graph, the constant vector is a trivial eigenvector
</p>
<p>with eigenvalue zero. Not so obvious is the fact that if the graph is con-
nected5, it is the only zero eigenvector (Exercise 14.21). Generalizing this
argument, it is easy to show that for a graph withm connected components,
</p>
<p>5A graph is connected if any two nodes can be reached via a path of connected nodes.</p>
<p/>
</div>
<div class="page"><p/>
<p>546 14. Unsupervised Learning
</p>
<p>&minus;4 &minus;2 0 2 4
</p>
<p>&minus;
4
</p>
<p>&minus;
2
</p>
<p>0
2
</p>
<p>4
</p>
<p>x1
</p>
<p>x
2
</p>
<p>0
.0
</p>
<p>0
.1
</p>
<p>0
.2
</p>
<p>0
.3
</p>
<p>0
.4
</p>
<p>0
.5
</p>
<p>Number
</p>
<p>E
ig
</p>
<p>e
n
v
a
lu
</p>
<p>e
</p>
<p>1 3 5 10 15
</p>
<p>0 100 200 300 400
</p>
<p>Eigenvectors
</p>
<p>Index
</p>
<p>2
n
</p>
<p>d
 S
</p>
<p>m
a
</p>
<p>lle
s
t
</p>
<p>3
rd
</p>
<p> S
m
</p>
<p>a
lle
</p>
<p>s
t
</p>
<p>&minus;
0
.0
</p>
<p>5
 0
</p>
<p>.0
5
</p>
<p>&minus;
0
.0
</p>
<p>5
 0
</p>
<p>.0
5
</p>
<p>&minus;0.04 &minus;0.02 0.00 0.02
</p>
<p>&minus;
0
.0
</p>
<p>6
&minus;
</p>
<p>0
.0
</p>
<p>2
0
.0
</p>
<p>2
0
.0
</p>
<p>6
</p>
<p>Second Smallest Eigenvector
</p>
<p>T
h
ir
d
 S
</p>
<p>m
a
lle
</p>
<p>s
t 
E
</p>
<p>ig
e
n
v
e
c
to
</p>
<p>r
</p>
<p>Spectral Clustering
</p>
<p>FIGURE 14.29. Toy example illustrating spectral clustering. Data in top left are
450 points falling in three concentric clusters of 150 points each. The points are
uniformly distributed in angle, with radius 1, 2.8 and 5 in the three groups, and
Gaussian noise with standard deviation 0.25 added to each point. Using a k = 10
nearest-neighbor similarity graph, the eigenvector corresponding to the second and
third smallest eigenvalues of L are shown in the bottom left; the smallest eigen-
vector is constant. The data points are colored in the same way as in the top left.
The 15 smallest eigenvalues are shown in the top right panel. The coordinates of
the 2nd and 3rd eigenvectors (the 450 rows of Z) are plotted in the bottom right
panel. Spectral clustering does standard (e.g., K-means) clustering of these points
and will easily recover the three original clusters.</p>
<p/>
</div>
<div class="page"><p/>
<p>14.5 Principal Components, Curves and Surfaces 547
</p>
<p>the nodes can be reordered so that L is block diagonal with a block for each
connected component. Then L has m eigenvectors of eigenvalue zero, and
the eigenspace of eigenvalue zero is spanned by the indicator vectors of the
connected components. In practice one has strong and weak connections,
so zero eigenvalues are approximated by small eigenvalues.
Spectral clustering is an interesting approach for finding non-convex clus-
</p>
<p>ters. When a normalized graph Laplacian is used, there is another way to
view this method. Defining P = G&minus;1W, we consider a random walk on
the graph with transition probability matrix P. Then spectral clustering
yields groups of nodes such that the random walk seldom transitions from
one group to another.
There are a number of issues that one must deal with in applying spec-
</p>
<p>tral clustering in practice. We must choose the type of similarity graph&mdash;eg.
fully connected or nearest neighbors, and associated parameters such as the
number of nearest of neighbors k or the scale parameter of the kernel c. We
must also choose the number of eigenvectors to extract from L and finally,
as with all clustering methods, the number of clusters. In the toy example
of Figure 14.29 we obtained good results for k &isin; [5, 200], the value 200 cor-
responding to a fully connected graph. With k &lt; 5 the results deteriorated.
Looking at the top-right panel of Figure 14.29, we see no strong separation
between the smallest three eigenvalues and the rest. Hence it is not clear
how many eigenvectors to select.
</p>
<p>14.5.4 Kernel Principal Components
</p>
<p>Spectral clustering is related to kernel principal components, a non-linear
version of linear principal components. Standard linear principal compo-
nents (PCA) are obtained from the eigenvectors of the covariance matrix,
and give directions in which the data have maximal variance. Kernel PCA
(Schölkopf et al., 1999) expand the scope of PCA, mimicking what we would
obtain if we were to expand the features by non-linear transformations, and
then apply PCA in this transformed feature space.
We show in Section 18.5.2 that the principal components variables Z of
</p>
<p>a data matrix X can be computed from the inner-product (gram) matrix
K = XXT . In detail, we compute the eigen-decomposition of the double-
centered version of the gram matrix
</p>
<p>K̃ = (I&minus;M)K(I&minus;M) = UD2UT , (14.65)
with M = 11T /N , and then Z = UD. Exercise 18.15 shows how to com-
pute the projections of new observations in this space.
Kernel PCA simply mimics this procedure, interpreting the kernel ma-
</p>
<p>trix K = {K(xi, xi&prime;)} as an inner-product matrix of the implicit fea-
tures 〈φ(xi), φ(xi&prime;)〉 and finding its eigenvectors. The elements of the mth
component zm (mth column of Z) can be written (up to centering) as
</p>
<p>zim =
&sum;N
</p>
<p>j=1 αjmK(xi, xj), where αjm = ujm/dm (Exercise 14.16).</p>
<p/>
</div>
<div class="page"><p/>
<p>548 14. Unsupervised Learning
</p>
<p>We can gain more insight into kernel PCA by viewing the zm as sam-
ple evaluations of principal component functions gm &isin; HK , with HK the
reproducing kernel Hilbert space generated by K (see Section 5.8.1). The
first principal component function g1 solves
</p>
<p>max
g1&isin;HK
</p>
<p>VarT g1(X) subject to ||g1||HK = 1 (14.66)
</p>
<p>Here VarT refers to the sample variance over training data T . The norm
constraint ||g1||HK = 1 controls the size and roughness of the function g1,
as dictated by the kernel K. As in the regression case it can be shown that
the solution to (14.66) is finite dimensional with representation g1(x) =&sum;N
</p>
<p>j=1 cjK(x, xj). Exercise 14.17 shows that the solution is defined by ĉj =
αj1, j = 1, . . . , N above. The second principal component function is de-
fined in a similar way, with the additional constraint that 〈g1, g2〉HK = 0,
and so on.6
</p>
<p>Schölkopf et al. (1999) demonstrate the use of kernel principal compo-
nents as features for handwritten-digit classification, and show that they
can improve the performance of a classifier when these are used instead of
linear principal components.
Note that if we use the radial kernel
</p>
<p>K(x, x&prime;) = exp(&minus;‖x&minus; x&prime;‖2/c), (14.67)
</p>
<p>then the kernel matrix K has the same form as the similarity matrix S in
spectral clustering. The matrix of edge weights W is a localized version of
K, setting to zero all similarities for pairs of points that are not nearest
neighbors.
Kernel PCA finds the eigenvectors corresponding to the largest eigenval-
</p>
<p>ues of K̃; this is equivalent to finding the eigenvectors corresponding to the
smallest eigenvalues of
</p>
<p>I&minus; K̃. (14.68)
</p>
<p>This is almost the same as the Laplacian (14.63), the differences being the
</p>
<p>centering of K̃ and the fact that G has the degrees of the nodes along the
diagonal.
Figure 14.30 examines the performance of kernel principal components
</p>
<p>in the toy example of Figure 14.29. In the upper left panel we used the ra-
dial kernel with c = 2, the same value that was used in spectral clustering.
This does not separate the groups, but with c = 10 (upper right panel), the
first component separates the groups well. In the lower-left panel we ap-
plied kernel PCA using the nearest-neighbor radial kernel W from spectral
clustering. In the lower right panel we use the kernel matrix itself as the
</p>
<p>6This section benefited from helpful discussions with Jonathan Taylor.</p>
<p/>
</div>
<div class="page"><p/>
<p>14.5 Principal Components, Curves and Surfaces 549
</p>
<p>&minus;0.10 &minus;0.06 &minus;0.02 0.02
</p>
<p>&minus;
0
.1
</p>
<p>0
&minus;
</p>
<p>0
.0
</p>
<p>5
0
.0
</p>
<p>0
0
.0
</p>
<p>5
0
.1
</p>
<p>0
</p>
<p>First Largest Eigenvector
</p>
<p>S
e
c
o
n
d
 L
</p>
<p>a
rg
</p>
<p>e
s
t 
E
</p>
<p>ig
e
n
v
e
c
to
</p>
<p>r
</p>
<p>Radial Kernel (c=2)
</p>
<p>&minus;0.06 &minus;0.02 0.02 0.06
</p>
<p>&minus;
0
.0
</p>
<p>5
0
.0
</p>
<p>0
0
.0
</p>
<p>5
</p>
<p>First Largest Eigenvector
</p>
<p>S
e
c
o
n
d
 L
</p>
<p>a
rg
</p>
<p>e
s
t 
E
</p>
<p>ig
e
n
v
e
c
to
</p>
<p>r
</p>
<p>Radial Kernel (c=10)
</p>
<p>0.00 0.05 0.10 0.15
</p>
<p>&minus;
0
.2
</p>
<p>&minus;
0
.1
</p>
<p>0
.0
</p>
<p>0
.1
</p>
<p>0
.2
</p>
<p>First Largest Eigenvector
</p>
<p>S
e
c
o
n
d
 L
</p>
<p>a
rg
</p>
<p>e
s
t 
E
</p>
<p>ig
e
n
v
e
c
to
</p>
<p>r
</p>
<p>NN Radial Kernel (c=2)
</p>
<p>&minus;0.05 0.00 0.05 0.10 0.15
</p>
<p>&minus;
0
.1
</p>
<p>0
0
.0
</p>
<p>0
0
.0
</p>
<p>5
0
.1
</p>
<p>0
0
.1
</p>
<p>5
</p>
<p>Second Smallest Eigenvector
</p>
<p>T
h
ir
d
 S
</p>
<p>m
a
lle
</p>
<p>s
t 
E
</p>
<p>ig
e
n
v
e
c
to
</p>
<p>r
</p>
<p>Radial Kernel Laplacian (c=2)
</p>
<p>FIGURE 14.30. Kernel principal components applied to the toy example of Fig-
ure 14.29, using different kernels. (Top left:) Radial kernel (14.67) with c = 2.
(Top right:) Radial kernel with c = 10. (Bottom left): Nearest neighbor radial ker-
nel W from spectral clustering. (Bottom right:) Spectral clustering with Laplacian
constructed from the radial kernel.</p>
<p/>
</div>
<div class="page"><p/>
<p>550 14. Unsupervised Learning
</p>
<p>similarity matrix for constructing the Laplacian (14.63) in spectral cluster-
ing. In neither case do the projections separate the two groups. Adjusting
c did not help either.
</p>
<p>In this toy example, we see that kernel PCA is quite sensitive to the scale
and nature of the kernel. We also see that the nearest-neighbor truncation
of the kernel is important for the success of spectral clustering.
</p>
<p>14.5.5 Sparse Principal Components
</p>
<p>We often interpret principal components by examining the direction vectors
vj , also known as loadings, to see which variables play a role. We did this
with the image loadings in (14.55). Often this interpretation is made easier
if the loadings are sparse. In this section we briefly discuss some methods
for deriving principal components with sparse loadings. They are all based
on lasso (L1) penalties.
</p>
<p>We start with an N &times; p data matrix X, with centered columns. The
proposed methods focus on either the maximum-variance property of prin-
cipal components, or the minimum reconstruction error. The SCoTLASS
procedure of Joliffe et al. (2003) takes the first approach, by solving
</p>
<p>max vT (XTX)v, subject to
&sum;p
</p>
<p>j=1 |vj | &le; t, vT v = 1. (14.69)
</p>
<p>The absolute-value constraint encourages some of the loadings to be zero
and hence v to be sparse. Further sparse principal components are found
in the same way, by forcing the kth component to be orthogonal to the
first k &minus; 1 components. Unfortunately this problem is not convex and the
computations are difficult.
Zou et al. (2006) start instead with the regression/reconstruction prop-
</p>
<p>erty of PCA, similar to the approach in Section 14.5.1. Let xi be the ith row
of X. For a single component, their sparse principal component technique
solves
</p>
<p>min
θ,v
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>||xi &minus; θvTxi||22 + λ||v||22 + λ1||v||1 (14.70)
</p>
<p>subject to ||θ||2 = 1.
</p>
<p>Let&rsquo;s examine this formulation in more detail.
</p>
<p>&bull; If both λ and λ1 are zero and N &gt; p, it is easy to show that v = θ
and is the largest principal component direction.
</p>
<p>&bull; When p ≫ N the solution is not necessarily unique unless λ &gt; 0. For
any λ &gt; 0 and λ1 = 0 the solution for v is proportional to the largest
principal component direction.
</p>
<p>&bull; The second penalty on v encourages sparseness of the loadings.</p>
<p/>
</div>
<div class="page"><p/>
<p>14.5 Principal Components, Curves and Surfaces 551
</p>
<p>Walking Speed
</p>
<p>Verbal Fluency
</p>
<p>Principal Components Sparse Principal Components
</p>
<p>FIGURE 14.31. Standard and sparse principal components from a study of
the corpus callosum variation. The shape variations corresponding to significant
principal components (red curves) are overlaid on the mean CC shape (black
curves).
</p>
<p>For multiple components, the sparse principal components procedures
minimizes
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>||xi &minus;ΘVTxi||2 + λ
K&sum;
</p>
<p>k=1
</p>
<p>||vk||22 +
K&sum;
</p>
<p>k=1
</p>
<p>λ1k||vk||1, (14.71)
</p>
<p>subject to ΘTΘ = IK . Here V is a p&times;K matrix with columns vk and Θ
is also p&times;K.
Criterion (14.71) is not jointly convex in V and Θ, but it is convex in
</p>
<p>each parameter with the other parameter fixed7. Minimization over V with
Θ fixed is equivalent to K elastic net problems (Section 18.4) and can be
done efficiently. On the other hand, minimization over Θ with V fixed is a
version of the Procrustes problem (14.56), and is solved by a simple SVD
calculation (Exercise 14.12). These steps are alternated until convergence.
Figure 14.31 shows an example of sparse principal components analysis
</p>
<p>using (14.71), taken from Sjöstrand et al. (2007). Here the shape of the
mid-sagittal cross-section of the corpus callosum (CC) is related to various
clinical parameters in a study involving 569 elderly persons8. In this exam-
</p>
<p>7Note that the usual principal component criterion, for example (14.50), is not jointly
convex in the parameters either. Nevertheless, the solution is well defined and an efficient
algorithm is available.
</p>
<p>8We thank Rasmus Larsen and Karl Sjöstrand for suggesting this application, and
</p>
<p>supplying us with the postscript figures reproduced here.</p>
<p/>
</div>
<div class="page"><p/>
<p>552 14. Unsupervised Learning
</p>
<p>FIGURE 14.32. An example of a mid-saggital brain slice, with the corpus col-
losum annotated with landmarks.
</p>
<p>ple PCA is applied to shape data, and is a popular tool in morphometrics.
For such applications, a number of landmarks are identified along the cir-
cumference of the shape; an example is given in Figure 14.32. These are
aligned by Procrustes analysis to allow for rotations, and in this case scal-
ing as well (see Section 14.5.1). The features used for PCA are the sequence
of coordinate pairs for each landmark, unpacked into a single vector.
In this analysis, both standard and sparse principal components were
</p>
<p>computed, and components that were significantly associated with various
clinical parameters were identified. In the figure, the shape variations cor-
responding to significant principal components (red curves) are overlaid on
the mean CC shape (black curves). Low walking speed relates to CCs that
are thinner (displaying atrophy) in regions connecting the motor control
and cognitive centers of the brain. Low verbal fluency relates to CCs that
are thinner in regions connecting auditory/visual/cognitive centers. The
sparse principal components procedure gives a more parsimonious, and po-
tentially more informative picture of the important differences.</p>
<p/>
</div>
<div class="page"><p/>
<p>14.6 Non-negative Matrix Factorization 553
</p>
<p>14.6 Non-negative Matrix Factorization
</p>
<p>Non-negative matrix factorization (Lee and Seung, 1999) is a recent al-
ternative approach to principal components analysis, in which the data
and components are assumed to be non-negative. It is useful for modeling
non-negative data such as images.
The N &times; p data matrix X is approximated by
</p>
<p>X &asymp; WH (14.72)
where W is N &times; r and H is r &times; p, r &le; max(N, p). We assume that
xij , wik, hkj &ge; 0.
</p>
<p>The matrices W and H are found by maximizing
</p>
<p>L(W,H) =
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>[xij log(WH)ij &minus; (WH)ij ]. (14.73)
</p>
<p>This is the log-likelihood from a model in which xij has a Poisson dis-
tribution with mean (WH)ij&mdash;quite reasonable for positive data.
The following alternating algorithm (Lee and Seung, 2001) converges to
</p>
<p>a local maximum of L(W,H):
</p>
<p>wik &larr; wik
&sum;p
</p>
<p>j=1 hkjxij/(WH)ij&sum;p
j=1 hkj
</p>
<p>hkj &larr; hkj
&sum;N
</p>
<p>i=1 wikxij/(WH)ij&sum;N
i=1 wik
</p>
<p>(14.74)
</p>
<p>This algorithm can be derived as a minorization procedure for maximizing
L(W,H) (Exercise 14.23) and is also related to the iterative-proportional-
scaling algorithm for log-linear models (Exercise 14.24).
Figure 14.33 shows an example taken from Lee and Seung (1999)9, com-
</p>
<p>paring non-negative matrix factorization (NMF), vector quantization (VQ,
equivalent to k-means clustering) and principal components analysis (PCA).
The three learning methods were applied to a database of N = 2, 429 fa-
cial images, each consisting of 19 &times; 19 pixels, resulting in a 2, 429 &times; 381
matrix X. As shown in the 7&times; 7 array of montages (each a 19&times; 19 image),
each method has learned a set of r = 49 basis images. Positive values are
illustrated with black pixels and negative values with red pixels. A par-
ticular instance of a face, shown at top right, is approximated by a linear
superposition of basis images. The coefficients of the linear superposition
are shown next to each montage, in a 7 &times; 7 array10, and the resulting su-
perpositions are shown to the right of the equality sign. The authors point
</p>
<p>9We thank Sebastian Seung for providing this image.
10These 7&times; 7 arrangements allow for a compact display, and have no structural
</p>
<p>significance.</p>
<p/>
</div>
<div class="page"><p/>
<p>554 14. Unsupervised Learning
</p>
<p>out that unlike VQ and PCA, NMF learns to represent faces with a set of
basis images resembling parts of faces.
Donoho and Stodden (2004) point out a potentially serious problem with
</p>
<p>non-negative matrix factorization. Even in situations whereX = WH holds
exactly, the decomposition may not be unique. Figure 14.34 illustrates the
problem. The data points lie in p = 2 dimensions, and there is &ldquo;open space&rdquo;
between the data and the coordinate axes. We can choose the basis vectors
h1 and h2 anywhere in this open space, and represent each data point
exactly with a nonnegative linear combination of these vectors. This non-
uniqueness means that the solution found by the above algorithm depends
on the starting values, and it would seem to hamper the interpretability of
the factorization. Despite this interpretational drawback, the non-negative
matrix factorization and its applications has attracted a lot of interest.
</p>
<p>14.6.1 Archetypal Analysis
</p>
<p>This method, due to Cutler and Breiman (1994), approximates data points
by prototypes that are themselves linear combinations of data points. In
this sense it has a similar flavor to K-means clustering. However, rather
than approximating each data point by a single nearby prototype, archety-
pal analysis approximates each data point by a convex combination of a
collection of prototypes. The use of a convex combination forces the proto-
types to lie on the convex hull of the data cloud. In this sense, the prototypes
are &ldquo;pure,&rdquo;, or &ldquo;archetypal.&rdquo;
As in (14.72), the N &times; p data matrix X is modeled as
</p>
<p>X &asymp; WH (14.75)
</p>
<p>where W is N&times;r and H is r&times;p. We assume that wik &ge; 0 and
&sum;r
</p>
<p>k=1 wik =
1 &forall;i. Hence the N data points (rows of X) in p-dimensional space are
represented by convex combinations of the r archetypes (rows of H). We
also assume that
</p>
<p>H = BX (14.76)
</p>
<p>where B is r &times;N with bki &ge; 0 and
&sum;N
</p>
<p>i=1 bki = 1 &forall;k. Thus the archetypes
themselves are convex combinations of the data points. Using both (14.75)
and (14.76) we minimize
</p>
<p>J(W,B) = ||X&minus;WH||2
= ||X&minus;WBX||2 (14.77)
</p>
<p>over the weights W and B. This function is minimized in an alternating
fashion, with each separate minimization involving a convex optimization.
The overall problem is not convex however, and so the algorithm converges
to a local minimum of the criterion.</p>
<p/>
</div>
<div class="page"><p/>
<p>14.6 Non-negative Matrix Factorization 555
</p>
<p>VQ
</p>
<p>&times; =
</p>
<p>NMF
</p>
<p>=&times;
</p>
<p>PCA
</p>
<p>=&times;
</p>
<p>Original
</p>
<p>FIGURE 14.33. Non-negative matrix factorization (NMF), vector quantization
(VQ, equivalent to k-means clustering) and principal components analysis (PCA)
applied to a database of facial images. Details are given in the text. Unlike VQ
and PCA, NMF learns to represent faces with a set of basis images resembling
parts of faces.</p>
<p/>
</div>
<div class="page"><p/>
<p>556 14. Unsupervised Learning
</p>
<p>h1
</p>
<p>h2
</p>
<p>FIGURE 14.34. Non-uniqueness of the non-negative matrix factorization.
There are 11 data points in two dimensions. Any choice of the basis vectors h1
and h2 in the open space between the coordinate axes and data, gives an exact
reconstruction of the data.
</p>
<p>Figure 14.35 shows an example with simulated data in two dimensions.
The top panel displays the results of archetypal analysis, while the bottom
panel shows the results from K-means clustering. In order to best recon-
struct the data from convex combinations of the prototypes, it pays to
locate the prototypes on the convex hull of the data. This is seen in the top
panels of Figure 14.35 and is the case in general, as proven by Cutler and
Breiman (1994). K-means clustering, shown in the bottom panels, chooses
prototypes in the middle of the data cloud.
We can think of K-means clustering as a special case of the archetypal
</p>
<p>model, in which each row of W has a single one and the rest of the entries
are zero.
Notice also that the archetypal model (14.75) has the same general form
</p>
<p>as the non-negative matrix factorization model (14.72). However, the two
models are applied in different settings, and have somewhat different goals.
Non-negative matrix factorization aims to approximate the columns of the
data matrix X, and the main output of interest are the columns of W
representing the primary non-negative components in the data. Archetypal
analysis focuses instead on the approximation of the rows of X using the
rows ofH, which represent the archetypal data points. Non-negative matrix
factorization also assumes that r &le; p. With r = p, we can get an exact
reconstruction simply choosing W to be the data X with columns scaled
so that they sum to 1. In contrast, archetypal analysis requires r &le; N ,
but allows r &gt; p. In Figure 14.35, for example, p = 2, N = 50 while
r = 2, 4 or 8. The additional constraint (14.76) implies that the archetypal
approximation will not be perfect, even if r &gt; p.
</p>
<p>Figure 14.36 shows the results of archetypal analysis applied to the
database of 3&rsquo;s displayed in Figure 14.22. The three rows in Figure 14.36
are the resulting archetypes from three runs, specifying two, three and four</p>
<p/>
</div>
<div class="page"><p/>
<p>14.7 Independent Component Analysisand Exploratory Projection Pursuit 557
</p>
<p>2 Prototypes 4 Prototypes 8 Prototypes
</p>
<p>FIGURE 14.35. Archetypal analysis (top panels) and K-means clustering (bot-
tom panels) applied to 50 data points drawn from a bivariate Gaussian distribu-
tion. The colored points show the positions of the prototypes in each case.
</p>
<p>archetypes, respectively. As expected, the algorithm has produced extreme
3&rsquo;s both in size and shape.
</p>
<p>14.7 Independent Component Analysis and
Exploratory Projection Pursuit
</p>
<p>Multivariate data are often viewed as multiple indirect measurements aris-
ing from an underlying source, which typically cannot be directly measured.
Examples include the following:
</p>
<p>&bull; Educational and psychological tests use the answers to questionnaires
to measure the underlying intelligence and other mental abilities of
subjects.
</p>
<p>&bull; EEG brain scans measure the neuronal activity in various parts of
the brain indirectly via electromagnetic signals recorded at sensors
placed at various positions on the head.
</p>
<p>&bull; The trading prices of stocks change constantly over time, and reflect
various unmeasured factors such as market confidence, external in-</p>
<p/>
</div>
<div class="page"><p/>
<p>558 14. Unsupervised Learning
</p>
<p>FIGURE 14.36. Archetypal analysis applied to the database of digitized 3&rsquo;s. The
rows in the figure show the resulting archetypes from three runs, specifying two,
three and four archetypes, respectively.
</p>
<p>fluences, and other driving forces that may be hard to identify or
measure.
</p>
<p>Factor analysis is a classical technique developed in the statistical liter-
ature that aims to identify these latent sources. Factor analysis models
are typically wed to Gaussian distributions, which has to some extent hin-
dered their usefulness. More recently, independent component analysis has
emerged as a strong competitor to factor analysis, and as we will see, relies
on the non-Gaussian nature of the underlying sources for its success.
</p>
<p>14.7.1 Latent Variables and Factor Analysis
</p>
<p>The singular-value decomposition X = UDVT (14.54) has a latent variable
representation. Writing S =
</p>
<p>&radic;
NU and AT = DVT /
</p>
<p>&radic;
N , we have X =
</p>
<p>SAT , and hence each of the columns of X is a linear combination of the
columns of S. Now since U is orthogonal, and assuming as before that the
columns of X (and hence U) each have mean zero, this implies that the
columns of S have zero mean, are uncorrelated and have unit variance. In
terms of random variables, we can interpret the SVD, or the corresponding
principal component analysis (PCA) as an estimate of a latent variable
model</p>
<p/>
</div>
<div class="page"><p/>
<p>14.7 Independent Component Analysis and Exploratory Projection Pursuit 559
</p>
<p>X1 = a11S1 + a12S2 + &middot; &middot; &middot;+ a1pSp
X2 = a21S1 + a22S2 + &middot; &middot; &middot;+ a2pSp
...
</p>
<p>...
Xp = ap1S1 + ap2S2 + &middot; &middot; &middot;+ appSp,
</p>
<p>(14.78)
</p>
<p>or simply X = AS. The correlated Xj are each represented as a linear
expansion in the uncorrelated, unit variance variables Sℓ. This is not too
satisfactory, though, because given any orthogonal p&times; p matrix R, we can
write
</p>
<p>X = AS
</p>
<p>= ARTRS
</p>
<p>= A&lowast;S&lowast;, (14.79)
</p>
<p>and Cov(S&lowast;) = RCov(S)RT = I. Hence there are many such decom-
positions, and it is therefore impossible to identify any particular latent
variables as unique underlying sources. The SVD decomposition does have
the property that any rank q &lt; p truncated decomposition approximates
X in an optimal way.
The classical factor analysismodel, developed primarily by researchers in
</p>
<p>psychometrics, alleviates these problems to some extent; see, for example,
Mardia et al. (1979). With q &lt; p, a factor analysis model has the form
</p>
<p>X1 = a11S1 + &middot; &middot; &middot;+ a1qSq + ε1
X2 = a21S1 + &middot; &middot; &middot;+ a2qSq + ε2
...
</p>
<p>...
Xp = ap1S1 + &middot; &middot; &middot;+ apqSq + εp,
</p>
<p>(14.80)
</p>
<p>or X = AS + ε. Here S is a vector of q &lt; p underlying latent variables or
factors, A is a p&times; q matrix of factor loadings, and the εj are uncorrelated
zero-mean disturbances. The idea is that the latent variables Sℓ are com-
mon sources of variation amongst the Xj , and account for their correlation
structure, while the uncorrelated εj are unique to each Xj and pick up the
remaining unaccounted variation. Typically the Sℓ and the εj are modeled
as Gaussian random variables, and the model is fit by maximum likelihood.
The parameters all reside in the covariance matrix
</p>
<p>Σ = AAT +Dε, (14.81)
</p>
<p>where Dε = diag[Var(ε1), . . . ,Var(εp)]. The Sℓ being Gaussian and un-
correlated makes them statistically independent random variables. Thus a
battery of educational test scores would be thought to be driven by the
independent underlying factors such as intelligence, drive and so on. The
columns of A are referred to as the factor loadings, and are used to name
and interpret the factors.</p>
<p/>
</div>
<div class="page"><p/>
<p>560 14. Unsupervised Learning
</p>
<p>Unfortunately the identifiability issue (14.79) remains, since A and ART
</p>
<p>are equivalent in (14.81) for any q &times; q orthogonal R. This leaves a certain
subjectivity in the use of factor analysis, since the user can search for ro-
tated versions of the factors that are more easily interpretable. This aspect
has left many analysts skeptical of factor analysis, and may account for its
lack of popularity in contemporary statistics. Although we will not go into
details here, the SVD plays a key role in the estimation of (14.81). For ex-
ample, if the Var(εj) are all assumed to be equal, the leading q components
of the SVD identify the subspace determined by A.
</p>
<p>Because of the separate disturbances εj for each Xj , factor analysis can
be seen to be modeling the correlation structure of the Xj rather than the
covariance structure. This can be easily seen by standardizing the covari-
ance structure in (14.81) (Exercise 14.14). This is an important distinction
between factor analysis and PCA, although not central to the discussion
here. Exercise 14.15 discusses a simple example where the solutions from
factor analysis and PCA differ dramatically because of this distinction.
</p>
<p>14.7.2 Independent Component Analysis
</p>
<p>The independent component analysis (ICA) model has exactly the same
form as (14.78), except the Sℓ are assumed to be statistically indepen-
dent rather than uncorrelated. Intuitively, lack of correlation determines
the second-degree cross-moments (covariances) of a multivariate distribu-
tion, while in general statistical independence determines all of the cross-
moments. These extra moment conditions allow us to identify the elements
of A uniquely. Since the multivariate Gaussian distribution is determined
by its second moments alone, it is the exception, and any Gaussian inde-
pendent components can be determined only up to a rotation, as before.
Hence identifiability problems in (14.78) and (14.80) can be avoided if we
assume that the Sℓ are independent and non-Gaussian.
Here we will discuss the full p-component model as in (14.78), where the
</p>
<p>Sℓ are independent with unit variance; ICA versions of the factor analysis
model (14.80) exist as well. Our treatment is based on the survey article
by Hyvärinen and Oja (2000).
</p>
<p>We wish to recover the mixing matrix A in X = AS. Without loss
of generality, we can assume that X has already been whitened to have
Cov(X) = I; this is typically achieved via the SVD described above. This
in turn implies that A is orthogonal, since S also has covariance I. So
solving the ICA problem amounts to finding an orthogonal A such that
the components of the vector random variable S = ATX are independent
(and non-Gaussian).
Figure 14.37 shows the power of ICA in separating two mixed signals.
</p>
<p>This is an example of the classical cocktail party problem, where differ-
ent microphones Xj pick up mixtures of different independent sources Sℓ
(music, speech from different speakers, etc.). ICA is able to perform blind</p>
<p/>
</div>
<div class="page"><p/>
<p>14.7 Independent Component Analysis and Exploratory Projection Pursuit 561
</p>
<p>Source Signals Measured Signals
</p>
<p>PCA Solution ICA Solution
</p>
<p>FIGURE 14.37. Illustration of ICA vs. PCA on artificial time-series data. The
upper left panel shows the two source signals, measured at 1000 uniformly spaced
time points. The upper right panel shows the observed mixed signals. The lower
two panels show the principal components and independent component solutions.
</p>
<p>source separation, by exploiting the independence and non-Gaussianity of
the original sources.
Many of the popular approaches to ICA are based on entropy. The dif-
</p>
<p>ferential entropy H of a random variable Y with density g(y) is given by
</p>
<p>H(Y ) = &minus;
&int;
</p>
<p>g(y) log g(y)dy. (14.82)
</p>
<p>A well-known result in information theory says that among all random
variables with equal variance, Gaussian variables have the maximum en-
tropy. Finally, the mutual information I(Y ) between the components of the
random vector Y is a natural measure of dependence:
</p>
<p>I(Y ) =
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>H(Yj)&minus;H(Y ). (14.83)
</p>
<p>The quantity I(Y ) is called the Kullback&ndash;Leibler distance between the
density g(y) of Y and its independence version
</p>
<p>&prod;p
j=1 gj(yj), where gj(yj)
</p>
<p>is the marginal density of Yj . Now if X has covariance I, and Y = A
TX
</p>
<p>with A orthogonal, then it is easy to show that
</p>
<p>I(Y ) =
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>H(Yj)&minus;H(X)&minus; log | detA| (14.84)
</p>
<p>=
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>H(Yj)&minus;H(X). (14.85)
</p>
<p>Finding an A to minimize I(Y ) = I(ATX) looks for the orthogonal trans-
formation that leads to the most independence between its components. In</p>
<p/>
</div>
<div class="page"><p/>
<p>562 14. Unsupervised Learning
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>***
*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>* *
*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>**
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* **
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
</p>
<p>* *
</p>
<p>*
</p>
<p>**
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>* *
*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* * *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>Source S
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>* **
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>**
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>**
</p>
<p>*
*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>* *
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>** *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
</p>
<p>* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>**
*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>**
</p>
<p>**
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>Data X
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>**
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>**
*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
* *
</p>
<p>**
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
* *
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
**
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>**
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
***
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>PCA Solution
</p>
<p>*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>* * **
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>***
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
**
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>* **
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>** *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>**
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>* *
*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
</p>
<p>***
</p>
<p>**
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>* *
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
*
</p>
<p>*
</p>
<p>*
***
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>*
</p>
<p>ICA Solution
</p>
<p>FIGURE 14.38. Mixtures of independent uniform random variables. The upper
left panel shows 500 realizations from the two independent uniform sources, the
upper right panel their mixed versions. The lower two panels show the PCA and
ICA solutions, respectively.
</p>
<p>light of (14.84) this is equivalent to minimizing the sum of the entropies of
the separate components of Y , which in turn amounts to maximizing their
departures from Gaussianity.
For convenience, rather than using the entropy H(Yj), Hyvärinen and
</p>
<p>Oja (2000) use the negentropy measure J(Yj) defined by
</p>
<p>J(Yj) = H(Zj)&minus;H(Yj), (14.86)
where Zj is a Gaussian random variable with the same variance as Yj . Ne-
gentropy is non-negative, and measures the departure of Yj from Gaussian-
ity. They propose simple approximations to negentropy which can be com-
puted and optimized on data. The ICA solutions shown in Figures 14.37&ndash;
14.39 use the approximation
</p>
<p>J(Yj) &asymp; [EG(Yj)&minus; EG(Zj)]2, (14.87)
where G(u) = 1a log cosh(au) for 1 &le; a &le; 2. When applied to a sample
of xi, the expectations are replaced by data averages. This is one of the
options in the FastICA software provided by these authors. More classical
(and less robust) measures are based on fourth moments, and hence look for
departures from the Gaussian via kurtosis. See Hyvärinen and Oja (2000)
for more details. In Section 14.7.4 we describe their approximate Newton
algorithm for finding the optimal directions.
In summary then, ICA applied to multivariate data looks for a sequence
</p>
<p>of orthogonal projections such that the projected data look as far from</p>
<p/>
</div>
<div class="page"><p/>
<p>14.7 Independent Component Analysis and Exploratory Projection Pursuit 563
</p>
<p>Component
 1
</p>
<p>oooo
o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
o oo
o
</p>
<p>oo
</p>
<p>oo
o
</p>
<p>oo o
o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>oo
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
oo
</p>
<p>o
</p>
<p>o
o oooooo
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
oo o
o
</p>
<p>oo
o
oo
</p>
<p>o
ooo
</p>
<p>o o
</p>
<p>o
</p>
<p>oo
o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oooooo oo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
o oo
oo
o
</p>
<p>oo
ooo o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>oo
o oo
</p>
<p>o
o
</p>
<p>o
o
o oo
</p>
<p>o o
ooo
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>ooo oooo
o
o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo oo
o
</p>
<p>o ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo ooo
</p>
<p>o
</p>
<p>o
o o
</p>
<p>oo o
o
</p>
<p>o
o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o o
o
o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
o
o
</p>
<p>o o
o o o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o o
oo o
o
</p>
<p>oo
oo
</p>
<p>oooooooo
o
</p>
<p>o
o
</p>
<p>ooo o
ooo
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
oo o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o oo
</p>
<p>oo
o oo
o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>oo oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
o
o o
</p>
<p>o
oo
oo oo oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>o
o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
oooo
</p>
<p>oo
</p>
<p>o
</p>
<p>oo ooo
</p>
<p>o
</p>
<p>oo
o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
oo o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>o
o
</p>
<p>o o
oo oo
</p>
<p>ooo
o
</p>
<p>ooo
</p>
<p>o
o
oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o o
oo
</p>
<p>ooo
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
oo o
</p>
<p>o o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>oo oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>oooo ooooooo
o
o
o
</p>
<p>o
</p>
<p>o
o ooo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>oo
o
oo
</p>
<p>ooo
o
o
</p>
<p>oo
o o o
</p>
<p>oo
</p>
<p>o o
o
</p>
<p>oo
o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
oo
o
oo
</p>
<p>oo
o
</p>
<p>o
oo o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o o
o
</p>
<p>o
</p>
<p>o
oo o
</p>
<p>oo
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
o o
</p>
<p>o o
ooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o o
o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
oooo
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
ooo
o
</p>
<p>oo
</p>
<p>oo
o
</p>
<p>ooo
o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
oo
</p>
<p>o
</p>
<p>o
o oooooo
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
o oo
</p>
<p>o
o o
</p>
<p>o
oo
</p>
<p>o
o oo
oo
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooooo ooo
o
o
o
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
ooo
</p>
<p>oo
o
</p>
<p>oo
ooo o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
o oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>ooo
oo
</p>
<p>oo o
o
o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>oo ooooo
o
o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o ooo
o
ooo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o oo oo
</p>
<p>o
</p>
<p>o
oo
ooo
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo o
o
</p>
<p>o
o
</p>
<p>oo
o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o o
o oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
ooo
o
</p>
<p>oo
o o
ooooo ooo
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo oo
o oo
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
oo o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
o
o
</p>
<p>o
o
</p>
<p>o oo
</p>
<p>o
</p>
<p>ooo
</p>
<p>oo
o oo
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>oo oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>oo
</p>
<p>o
oo
</p>
<p>oooooo
o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
oo
o
</p>
<p>o
ooo
</p>
<p>o
</p>
<p>o
o oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o o
</p>
<p>ooo o
o o
</p>
<p>o
</p>
<p>ooo oo
</p>
<p>o
</p>
<p>oo
oo o
o
</p>
<p>o
</p>
<p>o
o
ooo
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o
o
</p>
<p>oo
oooo
ooo
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o o
oo
</p>
<p>ooo
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
ooo
oo
o o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>oooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o oooo ooo ooo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
ooo oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o o
o
</p>
<p>oo
ooo
</p>
<p>o
o
</p>
<p>oo
ooo
</p>
<p>oo
</p>
<p>o o
o
o o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
o
o
</p>
<p>o
o
</p>
<p>o
oo
o
</p>
<p>o o
o o
</p>
<p>o
o
ooo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>o o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
o o
</p>
<p>oo
ooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
o
</p>
<p>o
oo oo
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
ooo
o
</p>
<p>o o
</p>
<p>o o
o
oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
oo oooo o
</p>
<p>o
</p>
<p>oo
o
o
</p>
<p>o
ooo
</p>
<p>o
oo
</p>
<p>o
oo
</p>
<p>o
oo o
oo
</p>
<p>o
</p>
<p>oo
o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooooo ooo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
o
</p>
<p>oo o
oo
o
</p>
<p>oo
oo o o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o o
oo o
</p>
<p>o
o
</p>
<p>o
o
ooo
</p>
<p>o o
o oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o oo oo oo
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o ooo
o
</p>
<p>ooo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o ooo
</p>
<p>o
</p>
<p>o
oo
</p>
<p>ooo
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
o o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
ooo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
ooo
</p>
<p>o
</p>
<p>oo
o o
</p>
<p>oooooooo
o
</p>
<p>o
o
</p>
<p>oooo
ooo
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
oo o
</p>
<p>oo
</p>
<p>o
o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>o
o
o
</p>
<p>oo
o
o
</p>
<p>o
o
</p>
<p>oo o
</p>
<p>o
</p>
<p>ooo
</p>
<p>oo
o o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>o ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>oo
</p>
<p>o
oo
</p>
<p>ooo oo o
o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
oo
</p>
<p>oo oo
oo
</p>
<p>o
</p>
<p>ooooo
</p>
<p>o
</p>
<p>oo
ooo
o
</p>
<p>o
</p>
<p>o
o
oo o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>o
o
</p>
<p>oo
ooo o
</p>
<p>ooo
o
</p>
<p>ooo
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o o
o o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o oo
o
oo
</p>
<p>o
</p>
<p>o
ooo
oo
</p>
<p>oo
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>oooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo oo oo ooooo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
ooooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>oo
o
oo
ooo
</p>
<p>o
o
o o
o o o
</p>
<p>oo
</p>
<p>oo
o
</p>
<p>oo
o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
o
o
</p>
<p>o
o
oo
o
oo
o o
o
</p>
<p>o
oo o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>oo
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>oo
ooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
oo oo
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
o oo
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
o
</p>
<p>ooo
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
ooo oooo
</p>
<p>o
</p>
<p>o o
o
o
</p>
<p>o
ooo
</p>
<p>o
oo
</p>
<p>o
oo
</p>
<p>o
o oo
oo
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooooo ooo
o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
oo o
oo
o
</p>
<p>oo
oo o o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
o oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>ooo
o o
</p>
<p>oo o
o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>oo oo ooo
o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo oo
o
</p>
<p>o o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
ooo oo
</p>
<p>o
</p>
<p>o
oo
</p>
<p>ooo
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>oo
oo o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
ooo
</p>
<p>o
</p>
<p>oo
o o
</p>
<p>oooooooo
o
</p>
<p>o
o
</p>
<p>o o oo
ooo
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
ooo
</p>
<p>oo
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>ooo
</p>
<p>o
</p>
<p>ooo
</p>
<p>o o
ooo
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
o
</p>
<p>ooo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
o
oo
</p>
<p>o
o o
</p>
<p>oooo o o
o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
oo o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o ooo
</p>
<p>o o
</p>
<p>o
</p>
<p>o oo oo
</p>
<p>o
</p>
<p>oo
oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o oo
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>o
o
o o
</p>
<p>ooo o
o oo
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
o
o o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o o
oo
</p>
<p>oo o
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o oo
</p>
<p>oo
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>ooo o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo oo ooooooo
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
o
</p>
<p>o o
o
</p>
<p>oo
o
</p>
<p>oo
ooo
o
</p>
<p>o
oo
</p>
<p>ooo
oo
</p>
<p>oo
o
</p>
<p>oo
o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
o
o
</p>
<p>o
o
o o
</p>
<p>o
oo
oo
</p>
<p>o
o
</p>
<p>ooo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
oo
o
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>oo
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>oo
o oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo o oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o o
</p>
<p>o
ooo
</p>
<p>o
</p>
<p>o ooo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o oo
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o o
o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
ooo
o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o o
ooo
</p>
<p>o
</p>
<p>o o
oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
oo o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
oo o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>oo
o
</p>
<p>oo
o
</p>
<p>ooo
</p>
<p>o o o
oo
</p>
<p>oo
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
oo
</p>
<p>oo
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>oo o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>Component
 2
</p>
<p>o
ooo
</p>
<p>ooo
oo
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
o
o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>oo
o
o
</p>
<p>o
o
</p>
<p>oo
o o
</p>
<p>o
o
</p>
<p>o
oo ooooo oo
o
</p>
<p>o
</p>
<p>ooooo
o o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>o ooo
o
</p>
<p>o
o
o
</p>
<p>oo
oo o
</p>
<p>ooo
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
ooooo oo o
</p>
<p>o
o
</p>
<p>o
o
o o
</p>
<p>o
oo
</p>
<p>o
ooo
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
oo o
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
ooooo
ooo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
oo
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
o oo
</p>
<p>o
ooo
</p>
<p>o
</p>
<p>ooo oo
o
</p>
<p>o
oo o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo o
</p>
<p>o
o
</p>
<p>ooo
</p>
<p>o
</p>
<p>oooo
o
</p>
<p>oo
oo
</p>
<p>o o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
o oo
o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o oo o
</p>
<p>oo
o
o
</p>
<p>o o
oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
</p>
<p>ooooo
o
</p>
<p>o
ooo
</p>
<p>o
ooo o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>ooo o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>oo oo o
</p>
<p>o
</p>
<p>oo
o oo
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
o
oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>ooo
o
</p>
<p>o
</p>
<p>ooo
oo
</p>
<p>o
o ooo oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
oo o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o oo
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o oo
</p>
<p>o
o
oo
oo ooo
</p>
<p>o
</p>
<p>o
oo o
</p>
<p>o
o
ooo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
o oo
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>oo
o o
</p>
<p>oo
oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
oooo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o o
o
</p>
<p>ooo o
</p>
<p>o
o
oo
</p>
<p>o
</p>
<p>o o
o
o
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo ooo o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o oo
</p>
<p>o oo
oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
oo
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
o o
</p>
<p>o
o
o
o oooo oo oo
o
</p>
<p>o
</p>
<p>oooo o
ooo
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>oooo
o
</p>
<p>o
o
o
</p>
<p>oo
oo o
</p>
<p>ooo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
o ooo ooo o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o o
</p>
<p>o
o oo
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
o oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
oo ooo
</p>
<p>ooo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
oooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
oo
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>o
o oo
</p>
<p>o
</p>
<p>ooooo
o
</p>
<p>o
oo o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o oo
</p>
<p>o
o
</p>
<p>ooo
</p>
<p>o
</p>
<p>oooo
o
</p>
<p>oo
oo
</p>
<p>oo
oo
o
</p>
<p>o
o
</p>
<p>o
oo
o
</p>
<p>ooo
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
o oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o ooo
</p>
<p>oo
o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o o
o
</p>
<p>ooo oo
o
</p>
<p>o
o oo
o
oooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>ooo o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o oooo
</p>
<p>o
</p>
<p>oo
ooo
</p>
<p>oo
o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
o
ooo
</p>
<p>o
o
</p>
<p>o oo
oo
</p>
<p>o
o ooooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o oo
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o oo
</p>
<p>o
o
</p>
<p>oo
o oo oo
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>o
o
</p>
<p>oo o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>ooo
oo
</p>
<p>o
</p>
<p>o o
oo
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>oo
oo
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
oooo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>ooo o
</p>
<p>o
o
oo
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo ooo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o oo
</p>
<p>o oo
o o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>oo
o
o
</p>
<p>o
o
</p>
<p>oo
o o
</p>
<p>o
o
</p>
<p>o
oo ooooooo
o
</p>
<p>o
</p>
<p>o oooo
oo o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>oooo
o
</p>
<p>o
o
o
oo
oo o
ooo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
o oooo oo o
o
</p>
<p>o
o
</p>
<p>o
o o
</p>
<p>o
oo
</p>
<p>o
ooo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
oo o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
ooo oo
ooo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
oo
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>o
ooo
</p>
<p>o
</p>
<p>ooo oo
o
</p>
<p>o
o oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
ooo
</p>
<p>o
o
ooo
</p>
<p>o
</p>
<p>o ooo
o
</p>
<p>oo
oo
</p>
<p>oo
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
ooo
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
o
o oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o ooo
</p>
<p>oo
o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
</p>
<p>oo o oo
o
o
</p>
<p>o oo
o
oo oo
o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
ooo o o
</p>
<p>o
</p>
<p>o o
o oo
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>oo
o
o
</p>
<p>o
o
</p>
<p>ooo
o
</p>
<p>o
</p>
<p>oo o
o o
</p>
<p>o
ooo o oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
o oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>ooo
o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
ooo
</p>
<p>o
o
oo
ooo oo
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>o
o
</p>
<p>ooo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo o
oo
</p>
<p>o
</p>
<p>oo
o o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o o
oo
</p>
<p>oo
oo
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
ooo
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
ooo o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>ooo o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo o oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o ooo
</p>
<p>o o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
oo
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo o o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>oo
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oooo
</p>
<p>o
oo
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o ooo
oo oo oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o oo
</p>
<p>o
</p>
<p>o
oo o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
oo oo
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oooo o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>oo
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo oo
</p>
<p>o
o o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>oo oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o oo o
o o
</p>
<p>o
oo o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
o
</p>
<p>o
o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
o
o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o oo
</p>
<p>o
</p>
<p>o
o oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>Component
 3
</p>
<p>oo
ooo o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o oo
oo
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
oo o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oooo o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>o
o oo
</p>
<p>oo
o
o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
ooo
o
</p>
<p>oo
o
o
oooo o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o ooo o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o o
o
</p>
<p>o
o
o ooooooo
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
o oooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo oooo ooo oo
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>o o
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo oo
o
</p>
<p>ooo oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo oo o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o o o
oo
</p>
<p>o
oooo
</p>
<p>o
ooo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o o
oo
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>oo
</p>
<p>ooo
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
oooo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
ooo
</p>
<p>ooo
oo
</p>
<p>oo
oo
oo
o
</p>
<p>ooo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
o
</p>
<p>oo
</p>
<p>ooo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
oo
</p>
<p>o oo
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
o o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo ooo o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
ooo
</p>
<p>oo
o
</p>
<p>o
o
oo o
</p>
<p>ooo
</p>
<p>o
</p>
<p>ooo o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>oo oo o
oo
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
oo oo o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
ooo o
</p>
<p>o
ooooooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o oo
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
ooo
</p>
<p>o
o o
</p>
<p>o
o ooo
</p>
<p>o
o
oo
</p>
<p>o
oo
o
oo
</p>
<p>o o
o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
ooo oo
</p>
<p>o
</p>
<p>o
</p>
<p>o o
oo
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o oo
o
</p>
<p>oo
oo o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
o
ooo
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o o
</p>
<p>o
</p>
<p>oooo
o
</p>
<p>o
</p>
<p>oo o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o o
ooo
oo
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>oo
o
</p>
<p>o
</p>
<p>oo
o o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo o
o
</p>
<p>o
o oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
oo
</p>
<p>ooo o
o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
ooo
oo
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>o
o
</p>
<p>o
o
o
</p>
<p>o oooo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o oo
</p>
<p>o
ooo
</p>
<p>oo
o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>o
oo
</p>
<p>o
o
oo ooo
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
o oooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
o
o
</p>
<p>o
oooo oooo
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
ooooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o ooo ooo oooo
</p>
<p>o
</p>
<p>o
o oo
</p>
<p>oo
o o
</p>
<p>o
</p>
<p>o
o
o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo oo
o
</p>
<p>o oo oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo ooo
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>ooo
oo
o
</p>
<p>o ooo
</p>
<p>o
ooo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>oo
oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o oo
</p>
<p>oo
</p>
<p>ooo
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
oooo
</p>
<p>o
</p>
<p>o o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
oo
</p>
<p>oo o
ooo
</p>
<p>o o
oo
</p>
<p>oo
o o
</p>
<p>o
oo o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o o
</p>
<p>o oo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
oo
</p>
<p>o o o
o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo oo oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
ooo
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>oo o
o oo
</p>
<p>o
</p>
<p>oo oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>ooo oo
oo
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
oo oo o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
o ooo
o
</p>
<p>o oo oooo
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
ooo
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
ooo
o
oo
</p>
<p>o
ooo o
o
o
</p>
<p>oo
</p>
<p>o
oo
</p>
<p>o
o o
o o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o oooo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>oo o
o
</p>
<p>oo
oo o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
o
o oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o o
</p>
<p>oo
</p>
<p>o
</p>
<p>ooo o
o
</p>
<p>o
</p>
<p>ooo
</p>
<p>oo
o
</p>
<p>o
</p>
<p>oo
ooo
</p>
<p>oo
o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>oo
o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo o
o
</p>
<p>o
o oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
o
o
</p>
<p>o
oo ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o oo o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
o
o
</p>
<p>ooo
oo
</p>
<p>oo o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>oo
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo oo o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
oo
</p>
<p>o
o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>oo
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
oo
o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o oo oo
oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>ooo
</p>
<p>o
oo
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>ooo o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>ooo o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
oo o
</p>
<p>oo
ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o o
</p>
<p>o
o o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o ooo o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
o o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>oo
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
o o
o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
o
</p>
<p>oo
o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
oo
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o ooo oo oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o o
</p>
<p>o
oo
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o ooo
o
o
</p>
<p>o
o
</p>
<p>oo oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo o o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
o
o
</p>
<p>oo o
o o
o o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>oo
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
o
</p>
<p>o o
o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo oo o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>oo
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o o
</p>
<p>o
oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
o o
o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
o
</p>
<p>o o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>ooooo ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
oo o
</p>
<p>o
o o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o Component
 4
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o o
o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
o oo
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
o
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
o
</p>
<p>o
o
</p>
<p>ooo
ooo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>ooooo
</p>
<p>oo
oo
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>ooo
o
</p>
<p>o
o
o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>oo
oo
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o ooo o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
o
o
</p>
<p>oo ooo oo
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
o
</p>
<p>oo
oo
oo
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
oo
</p>
<p>o
o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o o oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o o
o
</p>
<p>oo
oo
</p>
<p>o
o
</p>
<p>oo o
o o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
ooo
oo
oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
ooo oooo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
oooo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>o
o
</p>
<p>oo
oo
</p>
<p>oo
oo
</p>
<p>o
</p>
<p>o
oo o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>oo ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o o
o
o
</p>
<p>oo oo
</p>
<p>o
</p>
<p>o
ooo o o
o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>ooo
oo
</p>
<p>oo
o
</p>
<p>oooo
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>ooo o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oooo
oo
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
oo
o
</p>
<p>o
o
o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
o
</p>
<p>oo
oo ooo o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>ooo
ooo
</p>
<p>o
o o
oo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o oo oo o
o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oooo
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>ooo
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>ooo
o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
o
o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
ooo
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>oo o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o o o
ooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>ooo
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o o
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>oo
o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo oooo
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o oo
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o ooo
o
</p>
<p>o
</p>
<p>oo o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
ooo
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
o oo
</p>
<p>o
oo
</p>
<p>o
oo o o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>oo
o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o oo
o
o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>oo o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>oo o
oo o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o o
o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>ooo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o o
o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
ooo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>oo
o o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>ooo ooo
</p>
<p>o
</p>
<p>oo
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o oooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo oo o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
ooo
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o
oo
</p>
<p>o
oo o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o o o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>ooo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>oo o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o o
o
</p>
<p>oo o
o oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o o
o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
ooooo o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o oo
o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo ooo
o
</p>
<p>o
</p>
<p>oo o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
ooo
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
oo o
</p>
<p>o
o o
</p>
<p>o
o o oo
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o oo
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>ooo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>oo
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o o
o
</p>
<p>o
o
</p>
<p>ooo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o o o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o o
o
</p>
<p>ooo
ooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
o
</p>
<p>o o
o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o o
o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
o
</p>
<p>o
</p>
<p>oo
o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o oo
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>oo
oo
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o ooooo
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>ooo
o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>ooo oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
ooo
</p>
<p>ooo
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>ooo
</p>
<p>o
</p>
<p>ooo o
</p>
<p>o
oo
</p>
<p>o
o o oo
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>ooo
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>ooo
o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
o Component
</p>
<p> 5
</p>
<p>P
C
</p>
<p>A
 C
</p>
<p>o
m
</p>
<p>p
o
</p>
<p>n
e
</p>
<p>n
ts
</p>
<p>ICA Components
</p>
<p>FIGURE 14.39. A comparison of the first five ICA components computed using
FastICA (above diagonal) with the first five PCA components(below diagonal).
Each component is standardized to have unit variance.
</p>
<p>Gaussian as possible. With pre-whitened data, this amounts to looking for
components that are as independent as possible.
ICA starts from essentially a factor analysis solution, and looks for rota-
</p>
<p>tions that lead to independent components. From this point of view, ICA is
just another factor rotation method, along with the traditional &ldquo;varimax&rdquo;
and &ldquo;quartimax&rdquo; methods used in psychometrics.
</p>
<p>Example: Handwritten Digits
</p>
<p>We revisit the handwritten threes analyzed by PCA in Section 14.5.1. Fig-
ure 14.39 compares the first five (standardized) principal components with
the first five ICA components, all shown in the same standardized units.
Note that each plot is a two-dimensional projection from a 256-dimensional</p>
<p/>
</div>
<div class="page"><p/>
<p>564 14. Unsupervised Learning
</p>
<p>Mean ICA 1 ICA 2 ICA 3 ICA 4 ICA 5
</p>
<p>FIGURE 14.40. The highlighted digits from Figure 14.39. By comparing with
the mean digits, we see the nature of the ICA component.
</p>
<p>space. While the PCA components all appear to have joint Gaussian distri-
butions, the ICA components have long-tailed distributions. This is not too
surprising, since PCA focuses on variance, while ICA specifically looks for
non-Gaussian distributions. All the components have been standardized,
so we do not see the decreasing variances of the principal components.
For each ICA component we have highlighted two of the extreme digits,
</p>
<p>as well as a pair of central digits and displayed them in Figure 14.40.
This illustrates the nature of each of the components. For example, ICA
component five picks up the long sweeping tailed threes.
</p>
<p>Example: EEG Time Courses
</p>
<p>ICA has become an important tool in the study of brain dynamics&mdash;the
example we present here uses ICA to untangle the components of signals
in multi-channel electroencephalographic (EEG) data (Onton and Makeig,
2006).
Subjects wear a cap embedded with a lattice of 100 EEG electrodes,
</p>
<p>which record brain activity at different locations on the scalp. Figure 14.4111
</p>
<p>(top panel) shows 15 seconds of output from a subset of nine of these elec-
trodes from a subject performing a standard &ldquo;two-back&rdquo; learning task over
a 30 minute period. The subject is presented with a letter (B, H, J, C, F, or
K) at roughly 1500-ms intervals, and responds by pressing one of two but-
tons to indicate whether the letter presented is the same or different from
that presented two steps back. Depending on the answer, the subject earns
or loses points, and occasionally earns bonus or loses penalty points. The
time-course data show spatial correlation in the EEG signals&mdash;the signals
of nearby sensors look very similar.
The key assumption here is that signals recorded at each scalp electrode
</p>
<p>are a mixture of independent potentials arising from different cortical ac-
</p>
<p>11Reprinted from Progress in Brain Research, Vol. 159, Julie Onton and Scott Makeig,
&ldquo;Information based modeling of event-related brain dynamics,&rdquo; Page 106 , Copyright
(2006), with permission from Elsevier. We thank Julie Onton and Scott Makeig for
supplying an electronic version of the image.</p>
<p/>
</div>
<div class="page"><p/>
<p>14.7 Independent Component Analysis and Exploratory Projection Pursuit 565
</p>
<p>tivities, as well as non-cortical artifact domains; see the reference for a
detailed overview of ICA in this domain.
The lower part of Figure 14.41 shows a selection of ICA components.
</p>
<p>The colored images represent the estimated unmixing coefficient vectors âj
as heatmap images superimposed on the scalp, indicating the location of
activity. The corresponding time-courses show the activity of the learned
ICA components.
For example, the subject blinked after each performance feedback signal
</p>
<p>(colored vertical lines), which accounts for the location and artifact signal
in IC1 and IC3. IC12 is an artifact associated with the cardiac pulse. IC4
and IC7 account for frontal theta-band activities, and appear after a stretch
of correct performance. See Onton and Makeig (2006) for a more detailed
discussion of this example, and the use of ICA in EEG modeling.
</p>
<p>14.7.3 Exploratory Projection Pursuit
</p>
<p>Friedman and Tukey (1974) proposed exploratory projection pursuit, a
graphical exploration technique for visualizing high-dimensional data. Their
view was that most low (one- or two-dimensional) projections of high-
dimensional data look Gaussian. Interesting structure, such as clusters or
long tails, would be revealed by non-Gaussian projections. They proposed
a number of projection indices for optimization, each focusing on a differ-
ent departure from Gaussianity. Since their initial proposal, a variety of
improvements have been suggested (Huber, 1985; Friedman, 1987), and a
variety of indices, including entropy, are implemented in the interactive
graphics package Xgobi (Swayne et al., 1991, now called GGobi). These
projection indices are exactly of the same form as J(Yj) above, where
Yj = a
</p>
<p>T
j X, a normalized linear combination of the components of X. In
</p>
<p>fact, some of the approximations and substitutions for cross-entropy coin-
cide with indices proposed for projection pursuit. Typically with projection
pursuit, the directions aj are not constrained to be orthogonal. Friedman
(1987) transforms the data to look Gaussian in the chosen projection, and
then searches for subsequent directions. Despite their different origins, ICA
and exploratory projection pursuit are quite similar, at least in the repre-
sentation described here.
</p>
<p>14.7.4 A Direct Approach to ICA
</p>
<p>Independent components have by definition a joint product density
</p>
<p>fS(s) =
</p>
<p>p&prod;
</p>
<p>j=1
</p>
<p>fj(sj), (14.88)
</p>
<p>so here we present an approach that estimates this density directly us-
ing generalized additive models (Section 9.1). Full details can be found in</p>
<p/>
</div>
<div class="page"><p/>
<p>566 14. Unsupervised Learning
</p>
<p>FIGURE 14.41. Fifteen seconds of EEG data (of 1917 seconds) at nine (of
100) scalp channels (top panel), as well as nine ICA components (lower panel).
While nearby electrodes record nearly identical mixtures of brain and non-brain
activity, ICA components are temporally distinct. The colored scalps represent the
ICA unmixing coefficients âj as a heatmap, showing brain or scalp location of the
source.</p>
<p/>
</div>
<div class="page"><p/>
<p>14.7 Independent Component Analysis and Exploratory Projection Pursuit 567
</p>
<p>Hastie and Tibshirani (2003), and the method is implemented in the R
package ProDenICA, available from CRAN.
</p>
<p>In the spirit of representing departures from Gaussianity, we represent
each fj as
</p>
<p>fj(sj) = φ(sj)e
gj(sj), (14.89)
</p>
<p>a tilted Gaussian density. Here φ is the standard Gaussian density, and
gj satisfies the normalization conditions required of a density. Assuming
as before that X is pre-whitened, the log-likelihood for the observed data
X = AS is
</p>
<p>ℓ(A, {gj}p1;X) =
N&sum;
</p>
<p>i=1
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>[
log φj(a
</p>
<p>T
j xi) + gj(a
</p>
<p>T
j xi
</p>
<p>)
], (14.90)
</p>
<p>which we wish to maximize subject to the constraints that A is orthogonal
and that the gj result in densities in (14.89). Without imposing any further
restrictions on gj , the model (14.90) is over-parametrized, so we instead
maximize a regularized version
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>[
1
</p>
<p>N
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>[
log φ(aTj xi)+ gj(a
</p>
<p>T
j xi)
</p>
<p>]
&minus;
&int;
</p>
<p>φ(t)egj(t)dt&minus;λj
&int;
{g&prime;&prime;&prime;j (t)}2(t)dt
</p>
<p>]
.
</p>
<p>(14.91)
We have subtracted two penalty terms (for each j) in (14.91), inspired by
Silverman (1986, Section 5.4.4):
</p>
<p>&bull; The first enforces the density constraint
&int;
φ(t)eĝj(t)dt = 1 on any
</p>
<p>solution ĝj .
</p>
<p>&bull; The second is a roughness penalty, which guarantees that the solution
ĝj is a quartic-spline with knots at the observed values of sij = a
</p>
<p>T
j xi.
</p>
<p>It can further be shown that the solution densities f̂j = φe
ĝj each have
</p>
<p>mean zero and variance one (Exercise 14.18). As we increase λj , these
solutions approach the standard Gaussian φ.
</p>
<p>Algorithm 14.3 Product Density ICA Algorithm: ProDenICA
</p>
<p>1. Initialize A (random Gaussian matrix followed by orthogonalization).
</p>
<p>2. Alternate until convergence of A:
</p>
<p>(a) Given A, optimize (14.91) w.r.t. gj (separately for each j).
</p>
<p>(b) Given gj , j = 1, . . . , p, perform one step of a fixed point algo-
rithm towards finding the optimal A.
</p>
<p>We fit the functions gj and directions aj by optimizing (14.91) in an
alternating fashion, as described in Algorithm 14.3.</p>
<p/>
</div>
<div class="page"><p/>
<p>568 14. Unsupervised Learning
</p>
<p>Step 2(a) amounts to a semi-parametric density estimation, which can
be solved using a novel application of generalized additive models. For
convenience we extract one of the p separate problems,
</p>
<p>1
</p>
<p>N
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>[log φ(si) + g(si)]&minus;
&int;
</p>
<p>φ(t)eg(t)dt&minus; λ
&int;
{g&prime;&prime;&prime;(t)}2(t)dt. (14.92)
</p>
<p>Although the second integral in (14.92) leads to a smoothing spline, the
first integral is problematic, and requires an approximation. We construct
a fine grid of L values s&lowast;ℓ in increments ∆ covering the observed values si,
and count the number of si in the resulting bins:
</p>
<p>y&lowast;ℓ =
#si &isin; (s&lowast;ℓ &minus;∆/2, s&lowast;ℓ +∆/2)
</p>
<p>N
. (14.93)
</p>
<p>Typically we pick L to be 1000, which is more than adequate. We can then
approximate (14.92) by
</p>
<p>L&sum;
</p>
<p>ℓ=1
</p>
<p>{
y&lowast;i [log(φ(s
</p>
<p>&lowast;
ℓ )) + g(s
</p>
<p>&lowast;
ℓ )]&minus;∆φ(s&lowast;ℓ )eg(s
</p>
<p>&lowast;
</p>
<p>ℓ )
}
&minus; λ
</p>
<p>&int;
g&prime;&prime;&prime;
</p>
<p>2
(s)ds. (14.94)
</p>
<p>This last expression can be seen to be proportional to a penalized Poisson
log-likelihood with response y&lowast;ℓ /∆ and penalty parameter λ/∆, and mean
μ(s) = φ(s)eg(s). This is a generalized additive spline model (Hastie and
Tibshirani, 1990; Efron and Tibshirani, 1996), with an offset term log φ(s),
and can be fit using a Newton algorithm in O(L) operations. Although
a quartic spline is called for, we find in practice that a cubic spline is
adequate. We have p tuning parameters λj to set; in practice we make
them all the same, and specify the amount of smoothing via the effective
degrees-of-freedom df(λ). Our software uses 5df as a default value.
</p>
<p>Step 2(b) in Algorithm 14.3 requires optimizing (14.91) with respect to
A, holding the ĝj fixed. Only the first terms in the sum involve A, and
since A is orthogonal, the collection of terms involving φ do not depend on
A (Exercise 14.19). Hence we need to maximize
</p>
<p>C(A) =
1
</p>
<p>N
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>ĝj(a
T
j xi) (14.95)
</p>
<p>=
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>Cj(aj)
</p>
<p>C(A) is a log-likelihood ratio between the fitted density and a Gaussian,
and can be seen as an estimate of negentropy (14.86), with each ĝj a con-
trast function as in (14.87). The fixed point update in step 2(b) is a modified
Newton step (Exercise 14.20)</p>
<p/>
</div>
<div class="page"><p/>
<p>14.7 Independent Component Analysis and Exploratory Projection Pursuit 569
</p>
<p>1. For each j update
</p>
<p>aj &larr; E
{
Xĝ&prime;j(a
</p>
<p>T
j X)&minus; E[ĝ&prime;&prime;j (aTj X)]aj
</p>
<p>}
, (14.96)
</p>
<p>where E represents expectation w.r.t the sample xi. Since ĝj is a fitted
quartic (or cubic) spline, the first and second derivatives are readily
available.
</p>
<p>2. Orthogonalize A using the symmetric square-root transformation
(AAT )&minus;
</p>
<p>1
2A. If A = UDVT is the SVD of A, it is easy to show that
</p>
<p>this leads to the update A &larr; UVT .
</p>
<p>Our ProDenICA algorithm works as well as FastICA on the artificial time
series data of Figure 14.37, the mixture of uniforms data of Figure 14.38,
and the digit data in Figure 14.39.
</p>
<p>Example: Simulations
</p>
<p>a b c
</p>
<p>d e f
</p>
<p>g h i
</p>
<p>j k l
</p>
<p>m n o
</p>
<p>p q r
</p>
<p>Distribution
</p>
<p>A
m
</p>
<p>a
ri
 D
</p>
<p>is
ta
</p>
<p>n
c
e
</p>
<p> f
ro
</p>
<p>m
 T
</p>
<p>ru
e
</p>
<p> A
</p>
<p>a b c d e f g h i j k l m n o p q r
</p>
<p>0
.0
</p>
<p>1
0
</p>
<p>.0
2
</p>
<p>0
.0
</p>
<p>5
0
</p>
<p>.1
0
</p>
<p>0
.2
</p>
<p>0
0
</p>
<p>.5
0
</p>
<p>FastICA
KernelICA
ProdDenICA
</p>
<p>FIGURE 14.42. The left panel shows 18 distributions used for comparisons.
These include the &ldquo;t&rdquo;, uniform, exponential, mixtures of exponentials, symmetric
and asymmetric Gaussian mixtures. The right panel shows (on the log scale)
the average Amari metric for each method and each distribution, based on 30
simulations in IR2 for each distribution.
</p>
<p>Figure 14.42 shows the results of a simulation comparing ProDenICA to
FastICA, and another semi-parametric competitor KernelICA (Bach and
Jordan, 2002). The left panel shows the 18 distributions used as a basis
of comparison. For each distribution, we generated a pair of independent
components (N = 1024), and a random mixing matrix in IR2 with condition
number between 1 and 2. We used our R implementations of FastICA, using
the negentropy criterion (14.87), and ProDenICA. For KernelICA we used</p>
<p/>
</div>
<div class="page"><p/>
<p>570 14. Unsupervised Learning
</p>
<p>the authors MATLAB code.12 Since the search criteria are nonconvex, we
used five random starts for each method. Each of the algorithms delivers
an orthogonal mixing matrix A (the data were pre-whitened), which is
available for comparison with the generating orthogonalized mixing matrix
A0. We used the Amari metric (Bach and Jordan, 2002) as a measure of
the closeness of the two frames:
</p>
<p>d(A0,A) =
1
</p>
<p>2p
</p>
<p>p&sum;
</p>
<p>i=1
</p>
<p>(&sum;p
j=1 |rij |
</p>
<p>maxj |rij |
&minus; 1
</p>
<p>)
+
</p>
<p>1
</p>
<p>2p
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>(&sum;p
i=1 |rij |
</p>
<p>maxi |rij |
&minus; 1
</p>
<p>)
, (14.97)
</p>
<p>where rij = (AoA
&minus;1)ij . The right panel in Figure 14.42 compares the
</p>
<p>averages (on the log scale) of the Amari metric between the truth and the
estimated mixing matrices. ProDenICA is competitive with FastICA and
KernelICA in all situations, and dominates most of the mixture simulations.
</p>
<p>14.8 Multidimensional Scaling
</p>
<p>Both self-organizing maps and principal curves and surfaces map data
points in IRp to a lower-dimensional manifold. Multidimensional scaling
(MDS) has a similar goal, but approaches the problem in a somewhat dif-
ferent way.
We start with observations x1, x2, . . . , xN &isin; IRp, and let dij be the dis-
</p>
<p>tance between observations i and j. Often we choose Euclidean distance
dij = ||xi &minus; xj ||, but other distances may be used. Further, in some ap-
plications we may not even have available the data points xi, but only
have some dissimilarity measure dij (see Section 14.3.10). For example, in
a wine tasting experiment, dij might be a measure of how different a sub-
ject judged wines i and j, and the subject provides such a measure for all
pairs of wines i, j. MDS requires only the dissimilarities dij , in contrast to
the SOM and principal curves and surfaces which need the data points xi.
Multidimensional scaling seeks values z1, z2, . . . , zN &isin; IRk to minimize
</p>
<p>the so-called stress function13
</p>
<p>SM (z1, z2, . . . , zN ) =
&sum;
</p>
<p>i�=i&prime;
</p>
<p>(dii&prime; &minus; ||zi &minus; zi&prime; ||)2. (14.98)
</p>
<p>This is known as least squares or Kruskal&ndash;Shephard scaling. The idea is
to find a lower-dimensional representation of the data that preserves the
pairwise distances as well as possible. Notice that the approximation is
</p>
<p>12Francis Bach kindly supplied this code, and helped us set up the simulations.
13Some authors define stress as the square-root of SM ; since it does not affect the
</p>
<p>optimization, we leave it squared to make comparisons with other criteria simpler.</p>
<p/>
</div>
<div class="page"><p/>
<p>14.8 Multidimensional Scaling 571
</p>
<p>in terms of the distances rather than squared distances (which results in
slightly messier algebra). A gradient descent algorithm is used to minimize
SM .
A variation on least squares scaling is the so-called Sammon mapping
</p>
<p>which minimizes
</p>
<p>SSm(z1, z2, . . . , zN ) =
&sum;
</p>
<p>i�=i&prime;
</p>
<p>(dii&prime; &minus; ||zi &minus; zi&prime; ||)2
dii&prime;
</p>
<p>. (14.99)
</p>
<p>Here more emphasis is put on preserving smaller pairwise distances.
In classical scaling, we instead start with similarities sii&prime; : often we use
</p>
<p>the centered inner product sii&prime; = 〈xi &minus; x̄, xi&prime; &minus; x̄〉. The problem then is to
minimize
</p>
<p>SC(z1, z2, . . . , zN ) =
&sum;
</p>
<p>i,i&prime;
</p>
<p>(sii&prime; &minus; 〈zi &minus; z̄, zi&prime; &minus; z̄〉)2 (14.100)
</p>
<p>over z1, z2, . . . , zN &isin; IRk. This is attractive because there is an explicit
solution in terms of eigenvectors: see Exercise 14.11. If we have distances
rather than inner-products, we can convert them to centered inner-products
if the distances are Euclidean;14 see (18.31) on page 671 in Chapter 18.
If the similarities are in fact centered inner-products, classical scaling is
exactly equivalent to principal components, an inherently linear dimension-
reduction technique. Classical scaling is not equivalent to least squares
scaling; the loss functions are different, and the mapping can be nonlinear.
Least squares and classical scaling are referred to as metric scaling meth-
</p>
<p>ods, in the sense that the actual dissimilarities or similarities are approx-
imated. Shephard&ndash;Kruskal nonmetric scaling effectively uses only ranks.
Nonmetric scaling seeks to minimize the stress function
</p>
<p>SNM(z1, z2, . . . , zN ) =
</p>
<p>&sum;
i�=i&prime;
</p>
<p>[
||zi &minus; zi&prime; || &minus; θ(dii&prime;)
</p>
<p>]2
&sum;
</p>
<p>i�=i&prime; ||zi &minus; zi&prime; ||2
(14.101)
</p>
<p>over the zi and an arbitrary increasing function θ. With θ fixed, we min-
imize over zi by gradient descent. With the zi fixed, the method of iso-
tonic regression is used to find the best monotonic approximation θ(dii&prime;)
to ||zi &minus; zi&prime; ||. These steps are iterated until the solutions stabilize.
Like the self-organizing map and principal surfaces, multidimensional
</p>
<p>scaling represents high-dimensional data in a low-dimensional coordinate
system. Principal surfaces and SOMs go a step further, and approximate
the original data by a low-dimensional manifold, parametrized in the low
dimensional coordinate system. In a principal surface and SOM, points
</p>
<p>14An N &times; N distance matrix is Euclidean if the entries represent pairwise Euclidean
</p>
<p>distances between N points in some dimensional space.</p>
<p/>
</div>
<div class="page"><p/>
<p>572 14. Unsupervised Learning
</p>
<p>First MDS Coordinate
</p>
<p>S
e
</p>
<p>c
o
</p>
<p>n
d
</p>
<p> M
D
</p>
<p>S
 C
</p>
<p>o
o
</p>
<p>rd
in
</p>
<p>a
te
</p>
<p>-1.0 -0.5 0.0 0.5 1.0
</p>
<p>-1
.0
</p>
<p>-0
.5
</p>
<p>0
.0
</p>
<p>0
.5
</p>
<p>1
.0
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull; &bull;
</p>
<p>&bull;&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull; &bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull; &bull;
</p>
<p>&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull; &bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;
</p>
<p>FIGURE 14.43. First two coordinates for half-sphere data, from classical multi-
dimensional scaling.
</p>
<p>close together in the original feature space should map close together on
the manifold, but points far apart in feature space might also map close
together. This is less likely in multidimensional scaling since it explicitly
tries to preserve all pairwise distances.
Figure 14.43 shows the first two MDS coordinates from classical scaling
</p>
<p>for the half-sphere example. There is clear separation of the clusters, and
the tighter nature of the red cluster is apparent.
</p>
<p>14.9 Nonlinear Dimension Reduction and Local
Multidimensional Scaling
</p>
<p>Several methods have been recently proposed for nonlinear dimension re-
duction, similar in spirit to principal surfaces. The idea is that the data lie
close to an intrinsically low-dimensional nonlinear manifold embedded in a
high-dimensional space. These methods can be thought of as &ldquo;flattening&rdquo;
the manifold, and hence reducing the data to a set of low-dimensional co-
ordinates that represent their relative positions in the manifold. They are
useful for problems where signal-to-noise ratio is very high (e.g., physical
systems), and are probably not as useful for observational data with lower
signal-to-noise ratios.
The basic goal is illustrated in the left panel of Figure 14.44. The data
</p>
<p>lie near a parabola with substantial curvature. Classical MDS does not pre-</p>
<p/>
</div>
<div class="page"><p/>
<p>14.9 Nonlinear Dimension Reduction and Local Multidimensional Scaling 573
</p>
<p>&minus;5 0 5
</p>
<p>&minus;
1
5
</p>
<p>&minus;
1
0
</p>
<p>&minus;
5
</p>
<p>0
Classical MDS
</p>
<p>&minus;5 0 5
</p>
<p>&minus;
1
5
</p>
<p>&minus;
1
0
</p>
<p>&minus;
5
</p>
<p>0
</p>
<p>Local MDS
</p>
<p>x1x1
x
2
</p>
<p>x
2
</p>
<p>FIGURE 14.44. The orange points show data lying on a parabola, while the blue
points shows multidimensional scaling representations in one dimension. Classical
multidimensional scaling (left panel) does not preserve the ordering of the points
along the curve, because it judges points on opposite ends of the curve to be close
together. In contrast, local multidimensional scaling (right panel) does a good job
of preserving the ordering of the points along the curve.
</p>
<p>serve the ordering of the points along the curve, because it judges points
on opposite ends of the curve to be close together. The right panel shows
the results of local multi-dimensional scaling, one of the three methods for
non-linear multi-dimensional scaling that we discuss below. These meth-
ods use only the coordinates of the points in p dimensions, and have no
other information about the manifold. Local MDS has done a good job of
preserving the ordering of the points along the curve.
We now briefly describe three new approaches to nonlinear dimension
</p>
<p>reduction and manifold mapping.
Isometric feature mapping (ISOMAP) (Tenenbaum et al., 2000) con-
</p>
<p>structs a graph to approximate the geodesic distance between points along
the manifold. Specifically, for each data point we find its neighbors&mdash;points
within some small Euclidean distance of that point. We construct a graph
with an edge between any two neighboring points. The geodesic distance
between any two points is then approximated by the shortest path be-
tween points on the graph. Finally, classical scaling is applied to the graph
distances, to produce a low-dimensional mapping.
Local linear embedding (Roweis and Saul, 2000) takes a very different ap-
</p>
<p>proach, trying to preserve the local affine structure of the high-dimensional
data. Each data point is approximated by a linear combination of neigh-
boring points. Then a lower dimensional representation is constructed that</p>
<p/>
</div>
<div class="page"><p/>
<p>574 14. Unsupervised Learning
</p>
<p>best preserves these local approximations. The details are interesting, so
we give them here.
</p>
<p>1. For each data point xi in p dimensions, we find its K-nearest neigh-
bors N (i) in Euclidean distance.
</p>
<p>2. We approximate each point by an affine mixture of the points in its
neighborhood:
</p>
<p>min
Wik
</p>
<p>||xi &minus;
&sum;
</p>
<p>k&isin;N (i)
</p>
<p>wikxk||2 (14.102)
</p>
<p>over weights wik satisfying wik = 0, k /&isin; N (i),
&sum;N
</p>
<p>k=1 wik = 1. wik
is the contribution of point k to the reconstruction of point i. Note
that for a hope of a unique solution, we must have K &lt; p.
</p>
<p>3. Finally, we find points yi in a space of dimension d &lt; p to minimize
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>||yi &minus;
N&sum;
</p>
<p>k=1
</p>
<p>wikyk||2 (14.103)
</p>
<p>with wik fixed.
</p>
<p>In step 3, we minimize
</p>
<p>tr[(Y &minus;WY)T (Y &minus;WY)] = tr[YT (I&minus;W)T (I&minus;W)Y] (14.104)
</p>
<p>where W is N &times; N ; Y is N &times; d, for some small d &lt; p. The solutions Ŷ
are the trailing eigenvectors of M = (I&minus;W)T (I&minus;W). Since 1 is a trivial
eigenvector with eigenvalue 0, we discard it and keep the next d. This has
the side effect that 1TY = 0, and hence the embedding coordinates are
mean centered.
Local MDS (Chen and Buja, 2008) takes the simplest and arguably the
</p>
<p>most direct approach. We define N to be the symmetric set of nearby pairs
of points; specifically a pair (i, i&prime;) is in N if point i is among the K-nearest
neighbors of i&prime;, or vice-versa. Then we construct the stress function
</p>
<p>SL(z1, z2, . . . , zN ) =
&sum;
</p>
<p>(i,i&prime;)&isin;N
</p>
<p>(dii&prime; &minus; ||zi &minus; zi&prime; ||)2
</p>
<p>+
&sum;
</p>
<p>(i,i&prime;)/&isin;N
</p>
<p>w &middot; (D &minus; ||zi &minus; zi&prime; ||)2. (14.105)
</p>
<p>Here D is some large constant and w is a weight. The idea is that points
that are not neighbors are considered to be very far apart; such pairs are
given a small weight w so that they don&rsquo;t dominate the overall stress func-
tion. To simplify the expression, we take w &sim; 1/D, and let D &rarr; &infin;.
Expanding (14.105), this gives</p>
<p/>
</div>
<div class="page"><p/>
<p>14.9 Nonlinear Dimension Reduction and Local Multidimensional Scaling 575
</p>
<p>FIGURE 14.45. Images of faces mapped into the embedding space described by
the first two coordinates of LLE. Next to the circled points, representative faces
are shown in different parts of the space. The images at the bottom of the plot
correspond to points along the top right path (linked by solid line), and illustrate
one particular mode of variability in pose and expression.</p>
<p/>
</div>
<div class="page"><p/>
<p>576 14. Unsupervised Learning
</p>
<p>SL(z1, z2, . . . , zN ) =
&sum;
</p>
<p>(i,i&prime;)&isin;N
</p>
<p>(dii&prime; &minus; ||zi &minus; zi&prime; ||)2 &minus; τ
&sum;
</p>
<p>(i,i&prime;)/&isin;N
</p>
<p>||zi &minus; zi&prime; ||,
</p>
<p>(14.106)
where τ = 2wD. The first term in (14.106) tries to preserve local structure
in the data, while the second term encourages the representations zi, zi&prime;
</p>
<p>for pairs (i, i&prime;) that are non-neighbors to be farther apart. Local MDS
minimizes the stress function (14.106) over zi, for fixed values of the number
of neighbors K and the tuning parameter τ .
The right panel of Figure 14.44 shows the result of local MDS, using k = 2
</p>
<p>neighbors and τ = 0.01. We used coordinate descent with multiple starting
values to find a good minimum of the (nonconvex) stress function (14.106).
The ordering of the points along the curve has been largely preserved,
Figure 14.45 shows a more interesting application of one of these meth-
</p>
<p>ods (LLE)15. The data consist of 1965 photographs, digitized as 20 &times; 28
grayscale images. The result of the first two-coordinates of LLE are shown
and reveal some variability in pose and expression. Similar pictures were
produced by local MDS.
In experiments reported in Chen and Buja (2008), local MDS shows su-
</p>
<p>perior performance, as compared to ISOMAP and LLE. They also demon-
strate the usefulness of local MDS for graph layout. There are also close
connections between the methods discussed here, spectral clustering (Sec-
tion 14.5.3) and kernel PCA (Section 14.5.4).
</p>
<p>14.10 The Google PageRank Algorithm
</p>
<p>In this section we give a brief description of the original PageRank algo-
rithm used by the Google search engine, an interesting recent application
of unsupervised learning methods.
We suppose that we have N web pages and wish to rank them in terms
</p>
<p>of importance. For example, the N pages might all contain a string match
to &ldquo;statistical learning&rdquo; and we might wish to rank the pages in terms of
their likely relevance to a websurfer.
The PageRank algorithm considers a webpage to be important if many
</p>
<p>other webpages point to it. However the linking webpages that point to a
given page are not treated equally: the algorithm also takes into account
both the importance (PageRank) of the linking pages and the number of
outgoing links that they have. Linking pages with higher PageRank are
given more weight, while pages with more outgoing links are given less
weight. These ideas lead to a recursive definition for PageRank, detailed
next.
</p>
<p>15Sam Roweis and Lawrence Saul kindly provided this figure.</p>
<p/>
</div>
<div class="page"><p/>
<p>14.10 The Google PageRank Algorithm 577
</p>
<p>Let Lij = 1 if page j points to page i, and zero otherwise. Let cj =
&sum;N
</p>
<p>i=1 Lij equal the number of pages pointed to by page j (number of out-
links). Then the Google PageRanks pi are defined by the recursive rela-
tionship
</p>
<p>pi = (1&minus; d) + d
</p>
<p>N
&sum;
</p>
<p>j=1
</p>
<p>(Lij
cj
</p>
<p>)
</p>
<p>pj (14.107)
</p>
<p>where d is a positive constant (apparently set to 0.85).
The idea is that the importance of page i is the sum of the importances of
</p>
<p>pages that point to that page. The sums are weighted by 1/cj , that is, each
page distributes a total vote of 1 to other pages. The constant d ensures
that each page gets a PageRank of at least 1&minus; d. In matrix notation
</p>
<p>p = (1&minus; d)e+ d &middot; LD&minus;1c p (14.108)
</p>
<p>where e is a vector of N ones and Dc = diag(c) is a diagonal matrix with
diagonal elements cj . Introducing the normalization e
</p>
<p>Tp = N (i.e., the
average PageRank is 1), we can write (14.108) as
</p>
<p>p =
[
</p>
<p>(1&minus; d)eeT /N + dLD&minus;1c
]
</p>
<p>p
</p>
<p>= Ap (14.109)
</p>
<p>where the matrix A is the expression in square braces.
Exploiting a connection with Markov chains (see below), it can be shown
</p>
<p>that the matrix A has a real eigenvalue equal to one, and one is its largest
eigenvalue. This means that we can find p̂ by the power method: starting
with some p = p0 we iterate
</p>
<p>pk &larr; Apk&minus;1; pk &larr; N
pk
</p>
<p>eTpk
. (14.110)
</p>
<p>The fixed points p̂ are the desired PageRanks.
In the original paper of Page et al. (1998), the authors considered PageR-
</p>
<p>ank as a model of user behavior, where a random web surfer clicks on links
at random, without regard to content. The surfer does a random walk on
the web, choosing among available outgoing links at random. The factor
1&minus; d is the probability that he does not click on a link, but jumps instead
to a random webpage.
Some descriptions of PageRank have (1&minus; d)/N as the first term in def-
</p>
<p>inition (14.107), which would better coincide with the random surfer in-
terpretation. Then the page rank solution (divided by N) is the stationary
distribution of an irreducible, aperiodic Markov chain over the N webpages.
Definition (14.107) also corresponds to an irreducible, aperiodic Markov
</p>
<p>chain, with different transition probabilities than those from the (1&minus;d)/N
version. Viewing PageRank as a Markov chain makes clear why the matrix
A has a maximal real eigenvalue of 1. Since A has positive entries with</p>
<p/>
</div>
<div class="page"><p/>
<p>578 14. Unsupervised Learning
</p>
<p>Page 2
</p>
<p>Page 3
</p>
<p>Page 4
</p>
<p>Page 1
</p>
<p>FIGURE 14.46. PageRank algorithm: example of a small network
</p>
<p>each column summing to one, Markov chain theory tells us that it has a
unique eigenvector with eigenvalue one, corresponding to the stationary
distribution of the chain (Bremaud, 1999).
A small network is shown for illustration in Figure 14.46. The link matrix
</p>
<p>is
</p>
<p>L =
</p>
<p>⎛
⎜⎜⎝
</p>
<p>0 0 1 0
1 0 0 0
1 1 0 1
0 0 0 0
</p>
<p>⎞
⎟⎟⎠ (14.111)
</p>
<p>and the number of outlinks is c = (2, 1, 1, 1).
The PageRank solution is p̂ = (1.49, 0.78, 1.58, 0.15). Notice that page 4
</p>
<p>has no incoming links, and hence gets the minimum PageRank of 0.15.
</p>
<p>Bibliographic Notes
</p>
<p>There are many books on clustering, including Hartigan (1975), Gordon
(1999) and Kaufman and Rousseeuw (1990). K-means clustering goes back
at least to Lloyd (1957), Forgy (1965), Jancey (1966) and MacQueen (1967).
Applications in engineering, especially in image compression via vector
quantization, can be found in Gersho and Gray (1992). The k-medoid pro-
cedure is described in Kaufman and Rousseeuw (1990). Association rules
are outlined in Agrawal et al. (1995). The self-organizing map was proposed
by Kohonen (1989) and Kohonen (1990); Kohonen et al. (2000) give a more
recent account. Principal components analysis and multidimensional scal-
ing are described in standard books on multivariate analysis, for example,
Mardia et al. (1979). Buja et al. (2008) have implemented a powerful en-
vironment called Ggvis for multidimensional scaling, and the user manual</p>
<p/>
</div>
<div class="page"><p/>
<p>Exercises 579
</p>
<p>contains a lucid overview of the subject. Figures 14.17, 14.21 (left panel)
and 14.28 (left panel) were produced in Xgobi, a multidimensional data
visualization package by the same authors. GGobi is a more recent im-
plementation (Cook and Swayne, 2007). Goodall (1991) gives a technical
overview of Procrustes methods in statistics, and Ramsay and Silverman
(1997) discuss the shape registration problem. Principal curves and surfaces
were proposed in Hastie (1984) and Hastie and Stuetzle (1989). The idea of
principal points was formulated in Flury (1990), Tarpey and Flury (1996)
give an exposition of the general concept of self-consistency. An excellent
tutorial on spectral clustering can be found in von Luxburg (2007); this was
the main source for Section 14.5.3. Luxborg credits Donath and Hoffman
(1973) and Fiedler (1973) with the earliest work on the subject. A history
of spectral clustering my be found in Spielman and Teng (1996). Indepen-
dent component analysis was proposed by Comon (1994), with subsequent
developments by Bell and Sejnowski (1995); our treatment in Section 14.7
is based on Hyvärinen and Oja (2000). Projection pursuit was proposed by
Friedman and Tukey (1974), and is discussed in detail in Huber (1985). A
dynamic projection pursuit algorithm is implemented in GGobi.
</p>
<p>Exercises
</p>
<p>Ex. 14.1 Weights for clustering. Show that weighted Euclidean distance
</p>
<p>d(w)e (xi, xi&prime;) =
</p>
<p>&sum;p
l=1 wl(xil &minus; xi&prime;l)2&sum;p
</p>
<p>l=1 wl
</p>
<p>satisfies
</p>
<p>d(w)e (xi, xi&prime;) = de(zi, zi&prime;) =
</p>
<p>p&sum;
</p>
<p>l=1
</p>
<p>(zil &minus; zi&prime;l)2, (14.112)
</p>
<p>where
</p>
<p>zil = xil &middot;
(
</p>
<p>wl&sum;p
l=1 wl
</p>
<p>)1/2
. (14.113)
</p>
<p>Thus weighted Euclidean distance based on x is equivalent to unweighted
Euclidean distance based on z.
</p>
<p>Ex. 14.2 Consider a mixture model density in p-dimensional feature space,
</p>
<p>g(x) =
</p>
<p>K&sum;
</p>
<p>k=1
</p>
<p>πkgk(x), (14.114)
</p>
<p>where gk = N(μk,L&middot;σ2) and πk &ge; 0 &forall;k with
&sum;
</p>
<p>k πk = 1. Here {μk, πk}, k =
1, . . . ,K and σ2 are unknown parameters.</p>
<p/>
</div>
<div class="page"><p/>
<p>580 14. Unsupervised Learning
</p>
<p>Suppose we have data x1, x2, . . . , xN &sim; g(x) and we wish to fit the mix-
ture model.
</p>
<p>1. Write down the log-likelihood of the data
</p>
<p>2. Derive an EM algorithm for computing the maximum likelihood es-
timates (see Section 8.1).
</p>
<p>3. Show that if σ has a known value in the mixture model and we take
σ &rarr; 0, then in a sense this EM algorithm coincides with K-means
clustering.
</p>
<p>Ex. 14.3 In Section 14.2.6 we discuss the use of CART or PRIM for con-
structing generalized association rules. Show that a problem occurs with ei-
ther of these methods when we generate the random data from the product-
marginal distribution; i.e., by randomly permuting the values for each of
the variables. Propose ways to overcome this problem.
</p>
<p>Ex. 14.4 Cluster the demographic data of Table 14.1 using a classification
tree. Specifically, generate a reference sample of the same size of the train-
ing set, by randomly permuting the values within each feature. Build a
classification tree to the training sample (class 1) and the reference sample
(class 0) and describe the terminal nodes having highest estimated class 1
probability. Compare the results to the PRIM results near Table 14.1 and
also to the results of K-means clustering applied to the same data.
</p>
<p>Ex. 14.5 Generate data with three features, with 30 data points in each of
three classes as follows:
</p>
<p>θ1 = U(&minus;π/8, π/8)
φ1 = U(0, 2π)
x1 = sin(θ1) cos(φ1) +W11
y1 = sin(θ1) sin(φ1) +W12
z1 = cos(θ1) +W13
</p>
<p>θ2 = U(π/2&minus; π/4, π/2 + π/4)
φ2 = U(&minus;π/4, π/4)
x2 = sin(θ2) cos(φ2) +W21
y2 = sin(θ2) sin(φ2) +W22
z2 = cos(θ2) +W23
</p>
<p>θ3 = U(π/2&minus; π/4, π/2 + π/4)
φ3 = U(π/2&minus; π/4, π/2 + π/4)
x3 = sin(θ3) cos(φ3) +W31
y3 = sin(θ3) sin(φ3) +W32
z3 = cos(θ3) +W33
</p>
<p>Here U(a, b) indicates a uniform variate on the range [a, b] and Wjk are
independent normal variates with standard deviation 0.6. Hence the data</p>
<p/>
</div>
<div class="page"><p/>
<p>Exercises 581
</p>
<p>lie near the surface of a sphere in three clusters centered at (1, 0, 0), (0, 1, 0)
and (0, 0, 1).
</p>
<p>Write a program to fit a SOM to these data, using the learning rates
given in the text. Carry out a K-means clustering of the same data, and
compare the results to those in the text.
</p>
<p>Ex. 14.6 Write programs to implement K-means clustering and a self-
organizing map (SOM), with the prototype lying on a two-dimensional
grid. Apply them to the columns of the human tumor microarray data, us-
ing K = 2, 5, 10, 20 centroids for both. Demonstrate that as the size of the
SOM neighborhood is taken to be smaller and smaller, the SOM solution
becomes more similar to the K-means solution.
</p>
<p>Ex. 14.7 Derive (14.51) and (14.52) in Section 14.5.1. Show that μ̂ is not
unique, and characterize the family of equivalent solutions.
</p>
<p>Ex. 14.8 Derive the solution (14.57) to the Procrustes problem (14.56).
Derive also the solution to the Procrustes problem with scaling (14.58).
</p>
<p>Ex. 14.9 Write an algorithm to solve
</p>
<p>min
{βℓ,Rℓ}L1 ,M
</p>
<p>L&sum;
</p>
<p>ℓ=1
</p>
<p>||XℓRℓ &minus;M||2F . (14.115)
</p>
<p>Apply it to the three S&rsquo;s, and compare the results to those shown in Fig-
ure 14.26.
</p>
<p>Ex. 14.10 Derive the solution to the affine-invariant average problem (14.60).
Apply it to the three S&rsquo;s, and compare the results to those computed in
Exercise 14.9.
</p>
<p>Ex. 14.11 Classical multidimensional scaling. Let S be the centered in-
ner product matrix with elements 〈xi &minus; x̄, xj &minus; x̄〉. Let λ1 &gt; λ2 &gt; &middot; &middot; &middot; &gt;
λk be the k largest eigenvalues of S, with associated eigenvectors Ek =
(e1, e2, . . . , ek). Let Dk be a diagonal matrix with diagonal entries
</p>
<p>&radic;
λ1,&radic;
</p>
<p>λ2, . . . ,
&radic;
λk. Show that the solutions zi to the classical scaling problem
</p>
<p>(14.100) are the rows of EkDk.
</p>
<p>Ex. 14.12 Consider the sparse PCA criterion (14.71).
</p>
<p>1. Show that with Θ fixed, solving for V amounts to K separate elastic-
net regression problems, with responses the K elements of ΘTxi.
</p>
<p>2. Show that with V fixed, solving for Θ amounts to a reduced-rank
version of the Procrustes problem, which reduces to
</p>
<p>max
Θ
</p>
<p>trace(ΘTM) subject to ΘTΘ = IK , (14.116)
</p>
<p>where M and Θ are both p &times;K with K &le; p. If M = UDQT is the
SVD of M, show that the optimal Θ = UQT .</p>
<p/>
</div>
<div class="page"><p/>
<p>582 14. Unsupervised Learning
</p>
<p>Ex. 14.13 Generate 200 data points with three features, lying close to a
helix. In detail, define X1 = cos(s) + 0.1 &middot; Z1, X2 = sin(s) + 0.1 &middot; Z2, X3 =
s+ 0.1 &middot; Z3 where s takes on 200 equally spaced values between 0 and 2π,
and Z1, Z2, Z3 are independent and have standard Gaussian distributions.
</p>
<p>(a) Fit a principal curve to the data and plot the estimated coordinate
functions. Compare them to the underlying functions cos(s), sin(s)
and s.
</p>
<p>(b) Fit a self-organizing map to the same data, and see if you can discover
the helical shape of the original point cloud.
</p>
<p>Ex. 14.14 Pre- and post-multiply equation (14.81) by a diagonal matrix
containing the inverse variances of the Xj . Hence obtain an equivalent
decomposition for the correlation matrix, in the sense that a simple scaling
is applied to the matrix A.
</p>
<p>Ex. 14.15 Generate 200 observations of three variates X1, X2, X3 according
to
</p>
<p>X1 &sim; Z1
X2 = X1 + 0.001 &middot; Z2
X3 = 10 &middot; Z3 (14.117)
</p>
<p>where Z1, Z2, Z3 are independent standard normal variates. Compute the
leading principal component and factor analysis directions. Hence show
that the leading principal component aligns itself in the maximal variance
direction X3, while the leading factor essentially ignores the uncorrelated
component X3, and picks up the correlated component X2 +X1 (Geoffrey
Hinton, personal communication).
</p>
<p>Ex. 14.16 Consider the kernel principal component procedure outlined in
Section 14.5.4. Argue that the number M of principal components is equal
to the rank of K, which is the number of non-zero elements in D. Show
that the mth component zm (mth column of Z) can be written (up to
</p>
<p>centering) as zim =
&sum;N
</p>
<p>j=1 αjmK(xi, xj), where αjm = ujm/dm. Show that
the mapping of a new observation x0 to the mth component is given by
z0m =
</p>
<p>&sum;N
j=1 αjmK(x0, xj).
</p>
<p>Ex. 14.17 Show that with g1(x) =
&sum;N
</p>
<p>j=1 cjK(x, xj), the solution to (14.66)
is given by ĉj = uj1/d1, where u1 is the first column of U in (14.65), and
d1 the first diagonal element of D. Show that the second and subsequent
principal component functions are defined in a similar manner (hint: see
Section 5.8.1.)
</p>
<p>Ex. 14.18 Consider the regularized log-likelihood for the density estimation
problem arising in ICA,</p>
<p/>
</div>
<div class="page"><p/>
<p>Exercises 583
</p>
<p>1
</p>
<p>N
</p>
<p>N
&sum;
</p>
<p>i=1
</p>
<p>[log φ(si) + g(si)]&minus;
</p>
<p>&int;
</p>
<p>φ(t)eg(t)dt&minus; λ
</p>
<p>&int;
</p>
<p>{g&prime;&prime;&prime;(t)}2(t)dt. (14.118)
</p>
<p>The solution ĝ is a quartic smoothing spline, and can be written as ĝ(s) =
q̂(s) + q̂&perp;(s), where q is a quadratic function (in the null space of the
penalty). Let q(s) = θ0 + θ1s+ θ2s
</p>
<p>2. By examining the stationarity condi-
</p>
<p>tions for θ̂k, k = 1, 2, 3, show that the solution f̂ = φe
ĝ is a density, and
</p>
<p>has mean zero and variance one. If we used a second-derivative penalty
&int;
</p>
<p>{g&prime;&prime;(t)}2(t)dt instead, what simple modification could we make to the
problem to maintain the three moment conditions?
</p>
<p>Ex. 14.19 If A is p &times; p orthogonal, show that the first term in (14.91) on
page 567
</p>
<p>p
&sum;
</p>
<p>j=1
</p>
<p>N
&sum;
</p>
<p>i=1
</p>
<p>log φ(aTj xi),
</p>
<p>with aj the jth column of A, does not depend on A.
</p>
<p>Ex. 14.20 Fixed point algorithm for ICA (Hyvärinen et al., 2001). Consider
maximizing C(a) = E{g(aTX)} with respect to a, with ||a|| = 1 and
Cov(X) = I. Use a Lagrange multiplier to enforce the norm constraint,
and write down the first two derivatives of the modified criterion. Use the
approximation
</p>
<p>E{XXT g&prime;&prime;(aTX)} &asymp; E{XXT }E{g&prime;&prime;(aTX)}
</p>
<p>to show that the Newton update can be written as the fixed-point update
(14.96).
</p>
<p>Ex. 14.21 Consider an undirected graph with non-negative edge weights
wii&prime; and graph Laplacian L. Suppose there are m connected components
A1, A2, . . . , Am in the graph. Show that there arem eigenvectors of L corre-
sponding to eigenvalue zero, and the indicator vectors of these components
IA1 , IA2 , . . . , IAm span the zero eigenspace.
</p>
<p>Ex. 14.22
</p>
<p>(a) Show that definition (14.108) implies that the sum of the PageRanks
pi is N , the number of web pages.
</p>
<p>(b) Write a program to compute the PageRank solutions by the power
method using formulation (14.107). Apply it to the network of Fig-
ure 14.47.
</p>
<p>Ex. 14.23 Algorithm for non-negative matrix factorization (Wu and Lange,
2007). A function g(x, y) to said to minorize a function f(x) if</p>
<p/>
</div>
<div class="page"><p/>
<p>584 14. Unsupervised Learning
</p>
<p>Page 2
</p>
<p>Page 1
</p>
<p>Page 3
</p>
<p>Page 4
</p>
<p>Page 5
</p>
<p>Page 6
</p>
<p>FIGURE 14.47. Example of a small network.
</p>
<p>g(x, y) &le; f(x), g(x, x) = f(x) (14.119)
for all x, y in the domain. This is useful for maximizing f(x) since it is easy
to show that f(x) is nondecreasing under the update
</p>
<p>xs+1 = argmaxxg(x, x
s) (14.120)
</p>
<p>There are analogous definitions for majorization, for minimizing a function
f(x). The resulting algorithms are known as MM algorithms, for &ldquo;minorize-
maximize&rdquo; or &ldquo;majorize-minimize&rdquo; (Lange, 2004). It also can be shown
that the EM algorithm (8.5) is an example of an MM algorithm: see Sec-
tion 8.5.3 and Exercise 8.2 for details.
</p>
<p>(a) Consider maximization of the function L(W,H) in (14.73), written
here without the matrix notation
</p>
<p>L(W,H) =
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>[
xij log
</p>
<p>(
r&sum;
</p>
<p>k=1
</p>
<p>wikhkj
</p>
<p>)
&minus;
</p>
<p>r&sum;
</p>
<p>k=1
</p>
<p>wikhkj
</p>
<p>]
.
</p>
<p>Using the concavity of log(x), show that for any set of r values yk &ge; 0
and 0 &le; ck &le; 1 with
</p>
<p>&sum;r
k=1 ck = 1,
</p>
<p>log
</p>
<p>(
r&sum;
</p>
<p>k=1
</p>
<p>yk
</p>
<p>)
&ge;
</p>
<p>r&sum;
</p>
<p>k=1
</p>
<p>ck log(yk/ck)
</p>
<p>Hence
</p>
<p>log
</p>
<p>(
r&sum;
</p>
<p>k=1
</p>
<p>wikhkj
</p>
<p>)
&ge;
</p>
<p>r&sum;
</p>
<p>k=1
</p>
<p>asikj
bsij
</p>
<p>log
</p>
<p>(
bsij
asikj
</p>
<p>wikhkj
</p>
<p>)
,
</p>
<p>where
</p>
<p>asikj = w
s
ikh
</p>
<p>s
kj and b
</p>
<p>s
ij =
</p>
<p>r&sum;
</p>
<p>k=1
</p>
<p>wsikh
s
kj ,
</p>
<p>and s indicates the current iteration.</p>
<p/>
</div>
<div class="page"><p/>
<p>Exercises 585
</p>
<p>(b) Hence show that, ignoring constants, the function
</p>
<p>g(W,H |Ws,Hs) =
</p>
<p>N
&sum;
</p>
<p>i=1
</p>
<p>p
&sum;
</p>
<p>j=1
</p>
<p>r
&sum;
</p>
<p>k=1
</p>
<p>xij
asikj
bsij
</p>
<p>(
</p>
<p>logwik + log hkj
</p>
<p>)
</p>
<p>&minus;
</p>
<p>N
&sum;
</p>
<p>i=1
</p>
<p>p
&sum;
</p>
<p>j=1
</p>
<p>r
&sum;
</p>
<p>k=1
</p>
<p>wikhkj
</p>
<p>minorizes L(W,H).
</p>
<p>(c) Set the partial derivatives of g(W,H | Ws,Hs) to zero and hence
derive the updating steps (14.74).
</p>
<p>Ex. 14.24 Consider the non-negative matrix factorization (14.72) in the
rank one case (r = 1).
</p>
<p>(a) Show that the updates (14.74) reduce to
</p>
<p>wi &larr; wi
</p>
<p>&sum;p
j=1 xij
</p>
<p>&sum;p
j=1 wihj
</p>
<p>hj &larr; hj
</p>
<p>&sum;N
i=1 xij
</p>
<p>&sum;N
i=1 wihj
</p>
<p>(14.121)
</p>
<p>where wi = wi1, hj = h1j . This is an example of the iterative pro-
portional scaling procedure, applied to the independence model for a
two-way contingency table (Fienberg, 1977, for example).
</p>
<p>(b) Show that the final iterates have the explicit form
</p>
<p>wi = c &middot;
</p>
<p>&sum;p
j=1 xij
</p>
<p>&sum;N
i=1
</p>
<p>&sum;p
j=1 xij
</p>
<p>, hk =
1
</p>
<p>c
&middot;
</p>
<p>&sum;N
i=1 xik
</p>
<p>&sum;N
i=1
</p>
<p>&sum;p
j=1 xij
</p>
<p>(14.122)
</p>
<p>for any constant c &gt; 0. These are equivalent to the usual row and
column estimates for a two-way independence model.
</p>
<p>Ex. 14.25 Fit a non-negative matrix factorization model to the collection
of two&rsquo;s in the digits database. Use 25 basis elements, and compare with a
24- component (plus mean) PCA model. In both cases display the W and
H matrices as in Figure 14.33.</p>
<p/>
</div>
<div class="page"><p/>
<p>15
Random Forests
</p>
<p>15.1 Introduction
</p>
<p>Bagging or bootstrap aggregation (section 8.7) is a technique for reducing
the variance of an estimated prediction function. Bagging seems to work
especially well for high-variance, low-bias procedures, such as trees. For
regression, we simply fit the same regression tree many times to bootstrap-
sampled versions of the training data, and average the result. For classifi-
cation, a committee of trees each cast a vote for the predicted class.
Boosting in Chapter 10 was initially proposed as a committee method as
</p>
<p>well, although unlike bagging, the committee of weak learners evolves over
time, and the members cast a weighted vote. Boosting appears to dominate
bagging on most problems, and became the preferred choice.
Random forests (Breiman, 2001) is a substantial modification of bagging
</p>
<p>that builds a large collection of de-correlated trees, and then averages them.
On many problems the performance of random forests is very similar to
boosting, and they are simpler to train and tune. As a consequence, random
forests are popular, and are implemented in a variety of packages.
</p>
<p>15.2 Definition of Random Forests
</p>
<p>The essential idea in bagging (Section 8.7) is to average many noisy but
approximately unbiased models, and hence reduce the variance. Trees are
ideal candidates for bagging, since they can capture complex interaction
</p>
<p> &copy; Springer Science+Business Media, LLC 2009 
</p>
<p>T. Hastie et al., The Elements of Statistical Learning, Second Edition, 587
</p>
<p>DOI: 10.1007/b94608_15,</p>
<p/>
</div>
<div class="page"><p/>
<p>588 15. Random Forests
</p>
<p>Algorithm 15.1 Random Forest for Regression or Classification.
</p>
<p>1. For b = 1 to B:
</p>
<p>(a) Draw a bootstrap sample Z&lowast; of size N from the training data.
</p>
<p>(b) Grow a random-forest tree Tb to the bootstrapped data, by re-
cursively repeating the following steps for each terminal node of
the tree, until the minimum node size nmin is reached.
</p>
<p>i. Select m variables at random from the p variables.
</p>
<p>ii. Pick the best variable/split-point among the m.
</p>
<p>iii. Split the node into two daughter nodes.
</p>
<p>2. Output the ensemble of trees {Tb}
B
1 .
</p>
<p>To make a prediction at a new point x:
</p>
<p>Regression: f̂Brf (x) =
1
B
</p>
<p>&sum;B
b=1 Tb(x).
</p>
<p>Classification: Let Ĉb(x) be the class prediction of the bth random-forest
tree. Then ĈBrf (x) = majority vote {Ĉb(x)}
</p>
<p>B
1 .
</p>
<p>structures in the data, and if grown sufficiently deep, have relatively low
bias. Since trees are notoriously noisy, they benefit greatly from the averag-
ing. Moreover, since each tree generated in bagging is identically distributed
(i.d.), the expectation of an average of B such trees is the same as the ex-
pectation of any one of them. This means the bias of bagged trees is the
same as that of the individual (bootstrap) trees, and the only hope of im-
provement is through variance reduction. This is in contrast to boosting,
where the trees are grown in an adaptive way to remove bias, and hence
are not i.d.
An average of B i.i.d. random variables, each with variance σ2, has vari-
</p>
<p>ance 1Bσ
2. If the variables are simply i.d. (identically distributed, but not
</p>
<p>necessarily independent) with positive pairwise correlation ρ, the variance
of the average is (Exercise 15.1)
</p>
<p>ρσ2 +
1&minus; ρ
</p>
<p>B
σ2. (15.1)
</p>
<p>As B increases, the second term disappears, but the first remains, and
hence the size of the correlation of pairs of bagged trees limits the benefits
of averaging. The idea in random forests (Algorithm 15.1) is to improve
the variance reduction of bagging by reducing the correlation between the
trees, without increasing the variance too much. This is achieved in the
tree-growing process through random selection of the input variables.
Specifically, when growing a tree on a bootstrapped dataset:</p>
<p/>
</div>
<div class="page"><p/>
<p>15.2 Definition of Random Forests 589
</p>
<p>Before each split, select m &le; p of the input variables at random
as candidates for splitting.
</p>
<p>Typically values for m are
&radic;
p or even as low as 1.
</p>
<p>After B such trees {T (x; Θb)}B1 are grown, the random forest (regression)
predictor is
</p>
<p>f̂B
rf
(x) =
</p>
<p>1
</p>
<p>B
</p>
<p>B&sum;
</p>
<p>b=1
</p>
<p>T (x; Θb). (15.2)
</p>
<p>As in Section 10.9 (page 356), Θb characterizes the bth random forest tree in
terms of split variables, cutpoints at each node, and terminal-node values.
Intuitively, reducing m will reduce the correlation between any pair of trees
in the ensemble, and hence by (15.1) reduce the variance of the average.
</p>
<p>0 500 1000 1500 2000 2500
</p>
<p>0
.0
</p>
<p>4
0
</p>
<p>0
.0
</p>
<p>4
5
</p>
<p>0
.0
</p>
<p>5
0
</p>
<p>0
.0
</p>
<p>5
5
</p>
<p>0
.0
</p>
<p>6
0
</p>
<p>0
.0
</p>
<p>6
5
</p>
<p>0
.0
</p>
<p>7
0
</p>
<p>Spam Data
</p>
<p>Number of Trees
</p>
<p>T
e
</p>
<p>s
t 
</p>
<p>E
rr
</p>
<p>o
r
</p>
<p>Bagging
Random Forest
Gradient Boosting (5 Node)
</p>
<p>FIGURE 15.1. Bagging, random forest, and gradient boosting, applied to the
spam data. For boosting, 5-node trees were used, and the number of trees were
chosen by 10-fold cross-validation (2500 trees). Each &ldquo;step&rdquo; in the figure corre-
sponds to a change in a single misclassification (in a test set of 1536).
</p>
<p>Not all estimators can be improved by shaking up the data like this.
It seems that highly nonlinear estimators, such as trees, benefit the most.
For bootstrapped trees, ρ is typically small (0.05 or lower is typical; see
Figure 15.9), while σ2 is not much larger than the variance for the original
tree. On the other hand, bagging does not change linear estimates, such
as the sample mean (hence its variance either); the pairwise correlation
between bootstrapped means is about 50% (Exercise 15.4).</p>
<p/>
</div>
<div class="page"><p/>
<p>590 15. Random Forests
</p>
<p>Random forests are popular. Leo Breiman&rsquo;s1 collaborator Adele Cutler
maintains a random forest website2 where the software is freely available,
with more than 3000 downloads reported by 2002. There is a randomForest
package in R, maintained by Andy Liaw, available from the CRAN website.
</p>
<p>The authors make grand claims about the success of random forests:
&ldquo;most accurate,&rdquo; &ldquo;most interpretable,&rdquo; and the like. In our experience ran-
dom forests do remarkably well, with very little tuning required. A ran-
dom forest classifier achieves 4.88% misclassification error on the spam test
data, which compares well with all other methods, and is not significantly
worse than gradient boosting at 4.5%. Bagging achieves 5.4% which is
significantly worse than either (using the McNemar test outlined in Ex-
ercise 10.6), so it appears on this example the additional randomization
helps.
</p>
<p>RF&minus;1 RF&minus;3 Bagging GBM&minus;1 GBM&minus;6
</p>
<p>0
.0
</p>
<p>0
0
</p>
<p>.0
5
</p>
<p>0
.1
</p>
<p>0
0
</p>
<p>.1
5
</p>
<p>Nested Spheres
</p>
<p>T
e
</p>
<p>s
t 
</p>
<p>M
is
</p>
<p>c
la
</p>
<p>s
s
if
ic
</p>
<p>a
ti
o
</p>
<p>n
 E
</p>
<p>rr
o
</p>
<p>r
</p>
<p>Bayes Error
</p>
<p>FIGURE 15.2. The results of 50 simulations from the &ldquo;nested spheres&rdquo; model in
IR10. The Bayes decision boundary is the surface of a sphere (additive). &ldquo;RF-3&rdquo;
refers to a random forest with m = 3, and &ldquo;GBM-6&rdquo; a gradient boosted model
with interaction order six; similarly for &ldquo;RF-1&rdquo; and &ldquo;GBM-1.&rdquo; The training sets
were of size 2000, and the test sets 10, 000.
</p>
<p>Figure 15.1 shows the test-error progression on 2500 trees for the three
methods. In this case there is some evidence that gradient boosting has
started to overfit, although 10-fold cross-validation chose all 2500 trees.
</p>
<p>1Sadly, Leo Breiman died in July, 2005.
2http://www.math.usu.edu/&sim;adele/forests/</p>
<p/>
</div>
<div class="page"><p/>
<p>15.2 Definition of Random Forests 591
</p>
<p>0 200 400 600 800 1000
</p>
<p>0
.3
</p>
<p>2
0
</p>
<p>.3
4
</p>
<p>0
.3
</p>
<p>6
0
</p>
<p>.3
8
</p>
<p>0
.4
</p>
<p>0
0
</p>
<p>.4
2
</p>
<p>0
.4
</p>
<p>4
</p>
<p>California Housing Data
</p>
<p>Number of Trees
</p>
<p>T
e
</p>
<p>s
t 
</p>
<p>A
v
e
</p>
<p>ra
g
</p>
<p>e
 A
</p>
<p>b
s
o
</p>
<p>lu
te
</p>
<p> E
rr
</p>
<p>o
r
</p>
<p>RF m=2
RF m=6
GBM depth=4
GBM depth=6
</p>
<p>FIGURE 15.3. Random forests compared to gradient boosting on the California
housing data. The curves represent mean absolute error on the test data as a
function of the number of trees in the models. Two random forests are shown, with
m = 2 and m = 6. The two gradient boosted models use a shrinkage parameter
ν = 0.05 in (10.41), and have interaction depths of 4 and 6. The boosted models
outperform random forests.
</p>
<p>Figure 15.2 shows the results of a simulation3 comparing random forests
to gradient boosting on the nested spheres problem [Equation (10.2) in
Chapter 10]. Boosting easily outperforms random forests here. Notice that
smaller m is better here, although part of the reason could be that the true
decision boundary is additive.
Figure 15.3 compares random forests to boosting (with shrinkage) in a
</p>
<p>regression problem, using the California housing data (Section 10.14.1).
Two strong features that emerge are
</p>
<p>&bull; Random forests stabilize at about 200 trees, while at 1000 trees boost-
ing continues to improve. Boosting is slowed down by the shrinkage,
as well as the fact that the trees are much smaller.
</p>
<p>&bull; Boosting outperforms random forests here. At 1000 terms, the weaker
boosting model (GBM depth 4) has a smaller error than the stronger
</p>
<p>3Details: The random forests were fit using the R package randomForest 4.5-11,
with 500 trees. The gradient boosting models were fit using R package gbm 1.5, with
</p>
<p>shrinkage parameter set to 0.05, and 2000 trees.</p>
<p/>
</div>
<div class="page"><p/>
<p>592 15. Random Forests
</p>
<p>0 500 1000 1500 2000 2500
</p>
<p>0
.0
</p>
<p>4
5
</p>
<p>0
.0
</p>
<p>5
5
</p>
<p>0
.0
</p>
<p>6
5
</p>
<p>0
.0
</p>
<p>7
5
</p>
<p>Number of Trees
</p>
<p>M
is
</p>
<p>c
la
</p>
<p>s
s
if
ic
</p>
<p>a
ti
o
n
 E
</p>
<p>rr
o
r
</p>
<p>OOB Error
Test Error
</p>
<p>FIGURE 15.4. oob error computed on the spam training data, compared to the
test error computed on the test set.
</p>
<p>random forest (RF m = 6); a Wilcoxon test on the mean differences
in absolute errors has a p-value of 0.007. For larger m the random
forests performed no better.
</p>
<p>15.3 Details of Random Forests
</p>
<p>We have glossed over the distinction between random forests for classifica-
tion versus regression. When used for classification, a random forest obtains
a class vote from each tree, and then classifies using majority vote (see Sec-
tion 8.7 on bagging for a similar discussion). When used for regression, the
predictions from each tree at a target point x are simply averaged, as in
(15.2). In addition, the inventors make the following recommendations:
</p>
<p>&bull; For classification, the default value for m is &lfloor;&radic;p&rfloor; and the minimum
node size is one.
</p>
<p>&bull; For regression, the default value for m is &lfloor;p/3&rfloor; and the minimum
node size is five.
</p>
<p>In practice the best values for these parameters will depend on the problem,
and they should be treated as tuning parameters. In Figure 15.3 m = 6
performs much better than the default value &lfloor;8/3&rfloor; = 2.
</p>
<p>15.3.1 Out of Bag Samples
</p>
<p>An important feature of random forests is its use of out-of-bag (oob) sam-
ples:</p>
<p/>
</div>
<div class="page"><p/>
<p>15.3 Details of Random Forests 593
</p>
<p>For each observation zi = (xi, yi), construct its random forest
predictor by averaging only those trees corresponding to boot-
strap samples in which zi did not appear.
</p>
<p>An oob error estimate is almost identical to that obtained by N -fold cross-
validation; see Exercise 15.2. Hence unlike many other nonlinear estimators,
random forests can be fit in one sequence, with cross-validation being per-
formed along the way. Once the oob error stabilizes, the training can be
terminated.
Figure 15.4 shows the oob misclassification error for the spam data, com-
</p>
<p>pared to the test error. Although 2500 trees are averaged here, it appears
from the plot that about 200 would be sufficient.
</p>
<p>15.3.2 Variable Importance
</p>
<p>Variable importance plots can be constructed for random forests in exactly
the same way as they were for gradient-boosted models (Section 10.13).
At each split in each tree, the improvement in the split-criterion is the
importance measure attributed to the splitting variable, and is accumulated
over all the trees in the forest separately for each variable. The left plot
of Figure 15.5 shows the variable importances computed in this way for
the spam data; compare with the corresponding Figure 10.6 on page 354 for
gradient boosting. Boosting ignores some variables completely, while the
random forest does not. The candidate split-variable selection increases
the chance that any single variable gets included in a random forest, while
no such selection occurs with boosting.
Random forests also use the oob samples to construct a different variable-
</p>
<p>importance measure, apparently to measure the prediction strength of each
variable. When the bth tree is grown, the oob samples are passed down
the tree, and the prediction accuracy is recorded. Then the values for the
jth variable are randomly permuted in the oob samples, and the accuracy
is again computed. The decrease in accuracy as a result of this permuting
is averaged over all trees, and is used as a measure of the importance of
variable j in the random forest. These are expressed as a percent of the
maximum in the right plot in Figure 15.5. Although the rankings of the
two methods are similar, the importances in the right plot are more uni-
form over the variables. The randomization effectively voids the effect of
a variable, much like setting a coefficient to zero in a linear model (Exer-
cise 15.7). This does not measure the effect on prediction were this variable
not available, because if the model was refitted without the variable, other
variables could be used as surrogates.</p>
<p/>
</div>
<div class="page"><p/>
<p>594 15. Random Forests
</p>
<p>!
$
</p>
<p>remove
free
</p>
<p>CAPAVE
your
</p>
<p>CAPMAX
hp
</p>
<p>CAPTOT
money
</p>
<p>our
you
</p>
<p>george
000
edu
hpl
</p>
<p>business
1999
</p>
<p>internet
(
</p>
<p>will
all
</p>
<p>email
re
</p>
<p>receive
over
mail
</p>
<p>;
650
</p>
<p>meeting
labs
</p>
<p>order
address
</p>
<p>pm
people
</p>
<p>make
#
</p>
<p>credit
font
data
</p>
<p>technology
85
</p>
<p>[
lab
</p>
<p>telnet
report
</p>
<p>original
project
</p>
<p>conference
direct
</p>
<p>415
857
</p>
<p>addresses
3d
cs
</p>
<p>parts
table
</p>
<p>Gini
</p>
<p>0 20 40 60 80 100
</p>
<p>Variable Importance
</p>
<p>!
remove
</p>
<p>$
CAPAVE
</p>
<p>hp
free
</p>
<p>CAPMAX
edu
</p>
<p>george
CAPTOT
</p>
<p>your
our
</p>
<p>1999
re
</p>
<p>you
hpl
</p>
<p>business
000
</p>
<p>meeting
money
</p>
<p>(
will
</p>
<p>internet
650
pm
</p>
<p>receive
over
</p>
<p>email
;
</p>
<p>font
mail
</p>
<p>technology
order
</p>
<p>all
labs
</p>
<p>[
85
</p>
<p>address
original
</p>
<p>lab
telnet
</p>
<p>people
project
</p>
<p>data
credit
</p>
<p>conference
857
</p>
<p>#
415
</p>
<p>make
cs
</p>
<p>report
direct
</p>
<p>addresses
3d
</p>
<p>parts
table
</p>
<p>Randomization
</p>
<p>0 20 40 60 80 100
</p>
<p>Variable Importance
</p>
<p>FIGURE 15.5. Variable importance plots for a classification random forest
grown on the spam data. The left plot bases the importance on the Gini split-
ting index, as in gradient boosting. The rankings compare well with the rankings
produced by gradient boosting (Figure 10.6 on page 354). The right plot uses oob
randomization to compute variable importances, and tends to spread the impor-
tances more uniformly.</p>
<p/>
</div>
<div class="page"><p/>
<p>15.3 Details of Random Forests 595
</p>
<p>Proximity Plot
</p>
<p>1
</p>
<p>2
3
</p>
<p>4
</p>
<p>5
</p>
<p>6
</p>
<p>Random Forest Classifier
</p>
<p>1
</p>
<p>2
</p>
<p>34
</p>
<p>5
6
</p>
<p>Dimension 1
</p>
<p>D
im
</p>
<p>e
n
si
o
n
2
</p>
<p>X1
</p>
<p>X
2
</p>
<p>FIGURE 15.6. (Left): Proximity plot for a random forest classifier grown to
the mixture data. (Right): Decision boundary and training data for random forest
on mixture data. Six points have been identified in each plot.
</p>
<p>15.3.3 Proximity Plots
</p>
<p>One of the advertised outputs of a random forest is a proximity plot. Fig-
ure 15.6 shows a proximity plot for the mixture data defined in Section 2.3.3
in Chapter 2. In growing a random forest, an N &times; N proximity matrix is
accumulated for the training data. For every tree, any pair of oob obser-
vations sharing a terminal node has their proximity increased by one. This
proximity matrix is then represented in two dimensions using multidimen-
sional scaling (Section 14.8). The idea is that even though the data may be
high-dimensional, involving mixed variables, etc., the proximity plot gives
an indication of which observations are effectively close together in the eyes
of the random forest classifier.
Proximity plots for random forests often look very similar, irrespective of
</p>
<p>the data, which casts doubt on their utility. They tend to have a star shape,
one arm per class, which is more pronounced the better the classification
performance.
Since the mixture data are two-dimensional, we can map points from the
</p>
<p>proximity plot to the original coordinates, and get a better understanding of
what they represent. It seems that points in pure regions class-wise map to
the extremities of the star, while points nearer the decision boundaries map
nearer the center. This is not surprising when we consider the construction
of the proximity matrices. Neighboring points in pure regions will often
end up sharing a bucket, since when a terminal node is pure, it is no longer</p>
<p/>
</div>
<div class="page"><p/>
<p>596 15. Random Forests
</p>
<p>split by a random forest tree-growing algorithm. On the other hand, pairs
of points that are close but belong to different classes will sometimes share
a terminal node, but not always.
</p>
<p>15.3.4 Random Forests and Overfitting
</p>
<p>When the number of variables is large, but the fraction of relevant variables
small, random forests are likely to perform poorly with small m. At each
split the chance can be small that the relevant variables will be selected.
Figure 15.7 shows the results of a simulation that supports this claim. De-
tails are given in the figure caption and Exercise 15.3. At the top of each
pair we see the hyper-geometric probability that a relevant variable will be
selected at any split by a random forest tree (in this simulation, the relevant
variables are all equal in stature). As this probability gets small, the gap
between boosting and random forests increases. When the number of rele-
vant variables increases, the performance of random forests is surprisingly
robust to an increase in the number of noise variables. For example, with 6
relevant and 100 noise variables, the probability of a relevant variable being
selected at any split is 0.46, assuming m =
</p>
<p>&radic;
(6 + 100) &asymp; 10. According to
</p>
<p>Figure 15.7, this does not hurt the performance of random forests compared
with boosting. This robustness is largely due to the relative insensitivity of
misclassification cost to the bias and variance of the probability estimates
in each tree. We consider random forests for regression in the next section.
</p>
<p>Another claim is that random forests &ldquo;cannot overfit&rdquo; the data. It is
certainly true that increasing B does not cause the random forest sequence
to overfit; like bagging, the random forest estimate (15.2) approximates the
expectation
</p>
<p>f̂rf(x) = EΘT (x; Θ) = lim
B&rarr;&infin;
</p>
<p>f̂(x)Brf (15.3)
</p>
<p>with an average over B realizations of Θ. The distribution of Θ here is con-
ditional on the training data. However, this limit can overfit the data; the
average of fully grown trees can result in too rich a model, and incur unnec-
essary variance. Segal (2004) demonstrates small gains in performance by
controlling the depths of the individual trees grown in random forests. Our
experience is that using full-grown trees seldom costs much, and results in
one less tuning parameter.
Figure 15.8 shows the modest effect of depth control in a simple regression
</p>
<p>example. Classifiers are less sensitive to variance, and this effect of over-
fitting is seldom seen with random-forest classification.</p>
<p/>
</div>
<div class="page"><p/>
<p>15.4 Analysis of Random Forests 597
</p>
<p>T
e
</p>
<p>s
t 
</p>
<p>M
is
</p>
<p>c
la
</p>
<p>s
s
if
ic
</p>
<p>a
ti
o
</p>
<p>n
 E
</p>
<p>rr
o
</p>
<p>r
</p>
<p>0
.1
</p>
<p>0
0
.1
</p>
<p>5
0
.2
</p>
<p>0
0
.2
</p>
<p>5
0
.3
</p>
<p>0
</p>
<p>Bayes Error
</p>
<p>(2, 5) (2, 25) (2, 50) (2, 100) (2, 150)
</p>
<p>Number of (Relevant, Noise) Variables
</p>
<p>0.52 0.34 0.25 0.19 0.15
</p>
<p>Random Forest
</p>
<p>Gradient Boosting
</p>
<p>FIGURE 15.7. A comparison of random forests and gradient boosting on prob-
</p>
<p>lems with increasing numbers of noise variables. In each case the true decision
</p>
<p>boundary depends on two variables, and an increasing number of noise variables
</p>
<p>are included. Random forests uses its default value m =
&radic;
</p>
<p>p. At the top of each
</p>
<p>pair is the probability that one of the relevant variables is chosen at any split.
</p>
<p>The results are based on 50 simulations for each pair, with a training sample of
</p>
<p>300, and a test sample of 500. See Exercise 15.3.
</p>
<p>15.4 Analysis of Random Forests
</p>
<p>In this section we analyze the mechanisms at play with the additional
randomization employed by random forests. For this discussion we focus
on regression and squared error loss, since this gets at the main points,
and bias and variance are more complex with 0&ndash;1 loss (see Section 7.3.1).
Furthermore, even in the case of a classification problem, we can consider
the random-forest average as an estimate of the class posterior probabilities,
for which bias and variance are appropriate descriptors.
</p>
<p>15.4.1 Variance and the De-Correlation Effect
</p>
<p>The limiting form (B &rarr; &infin;) of the random forest regression estimator is
</p>
<p>f̂rf(x) = EΘ|ZT (x; Θ(Z)), (15.4)
</p>
<p>where we have made explicit the dependence on the training data Z. Here
we consider estimation at a single target point x. From (15.1) we see that</p>
<p/>
</div>
<div class="page"><p/>
<p>598 15. Random Forests
</p>
<p>50 30 20 10 5
</p>
<p>1
.0
</p>
<p>0
1
</p>
<p>.0
5
</p>
<p>1
.1
</p>
<p>0
</p>
<p>Minimum Node Size
</p>
<p>M
e
</p>
<p>a
n
</p>
<p> S
q
</p>
<p>u
a
</p>
<p>re
d
</p>
<p> T
e
</p>
<p>s
t 
</p>
<p>E
rr
</p>
<p>o
r
</p>
<p>Shallow Deep
</p>
<p>FIGURE 15.8. The effect of tree size on the error in random forest regres-
sion. In this example, the true surface was additive in two of the 12 variables,
plus additive unit-variance Gaussian noise. Tree depth is controlled here by the
minimum node size; the smaller the minimum node size, the deeper the trees.
</p>
<p>Varf̂rf(x) = ρ(x)σ
2(x). (15.5)
</p>
<p>Here
</p>
<p>&bull; ρ(x) is the sampling correlation between any pair of trees used in the
averaging:
</p>
<p>ρ(x) = corr[T (x; Θ1(Z)), T (x; Θ2(Z))], (15.6)
</p>
<p>where Θ1(Z) and Θ2(Z) are a randomly drawn pair of random forest
trees grown to the randomly sampled Z;
</p>
<p>&bull; σ2(x) is the sampling variance of any single randomly drawn tree,
</p>
<p>σ2(x) = VarT (x; Θ(Z)). (15.7)
</p>
<p>It is easy to confuse ρ(x) with the average correlation between fitted trees
in a given random-forest ensemble; that is, think of the fitted trees as N -
vectors, and compute the average pairwise correlation between these vec-
tors, conditioned on the data. This is not the case; this conditional corre-
lation is not directly relevant in the averaging process, and the dependence
on x in ρ(x) warns us of the distinction. Rather, ρ(x) is the theoretical
correlation between a pair of random-forest trees evaluated at x, induced
by repeatedly making training sample draws Z from the population, and
then drawing a pair of random forest trees. In statistical jargon, this is the
correlation induced by the sampling distribution of Z and Θ.
</p>
<p>More precisely, the variability averaged over in the calculations in (15.6)
and (15.7) is both</p>
<p/>
</div>
<div class="page"><p/>
<p>15.4 Analysis of Random Forests 599
</p>
<p>&bull; conditional on Z: due to the bootstrap sampling and feature sampling
at each split, and
</p>
<p>&bull; a result of the sampling variability of Z itself.
</p>
<p>In fact, the conditional covariance of a pair of tree fits at x is zero, because
the bootstrap and feature sampling is i.i.d; see Exercise 15.5.
</p>
<p>1 4 7 13 19 25 31 37 43 49
</p>
<p>0
.0
</p>
<p>0
0
</p>
<p>.0
2
</p>
<p>0
.0
</p>
<p>4
0
</p>
<p>.0
6
</p>
<p>0
.0
</p>
<p>8
</p>
<p>Number of Randomly Selected Splitting Variables m
</p>
<p>C
o
</p>
<p>rr
e
</p>
<p>la
ti
o
</p>
<p>n
 b
</p>
<p>e
tw
</p>
<p>e
e
</p>
<p>n
 T
</p>
<p>re
e
</p>
<p>s
</p>
<p>FIGURE 15.9. Correlations between pairs of trees drawn by a random-forest
regression algorithm, as a function of m. The boxplots represent the correlations
at 600 randomly chosen prediction points x.
</p>
<p>The following demonstrations are based on a simulation model
</p>
<p>Y =
1&radic;
50
</p>
<p>50&sum;
</p>
<p>j=1
</p>
<p>Xj + ε, (15.8)
</p>
<p>with all the Xj and ε iid Gaussian. We use 500 training sets of size 100, and
a single set of test locations of size 600. Since regression trees are nonlinear
in Z, the patterns we see below will differ somewhat depending on the
structure of the model.
Figure 15.9 shows how the correlation (15.6) between pairs of trees de-
</p>
<p>creases as m decreases: pairs of tree predictions at x for different training
sets Z are likely to be less similar if they do not use the same splitting
variables.
In the left panel of Figure 15.10 we consider the variances of single tree
</p>
<p>predictors, VarT (x; Θ(Z)) (averaged over 600 prediction points x drawn
randomly from our simulation model). This is the total variance, and can be</p>
<p/>
</div>
<div class="page"><p/>
<p>600 15. Random Forests
</p>
<p>decomposed into two parts using standard conditional variance arguments
(see Exercise 15.5):
</p>
<p>VarΘ,ZT (x; Θ(Z)) = VarZEΘ|ZT (x; Θ(Z)) + EZVarΘ|ZT (x; Θ(Z))
</p>
<p>Total Variance = VarZf̂rf(x) + within-Z Variance
(15.9)
</p>
<p>The second term is the within-Z variance&mdash;a result of the randomization,
which increases as m decreases. The first term is in fact the sampling vari-
ance of the random forest ensemble (shown in the right panel), which de-
creases as m decreases. The variance of the individual trees does not change
appreciably over much of the range of m, hence in light of (15.5), the vari-
ance of the ensemble is dramatically lower than this tree variance.
</p>
<p>0 10 20 30 40 50
</p>
<p>1
.8
</p>
<p>0
1
.8
</p>
<p>5
1
.9
</p>
<p>0
1
.9
</p>
<p>5
</p>
<p>Single Tree
</p>
<p>m
</p>
<p>V
a
ri
a
n
c
e
</p>
<p>Within Z
Total
</p>
<p>0 10 20 30 40 50
</p>
<p>0
.6
</p>
<p>5
0
.7
</p>
<p>0
0
.7
</p>
<p>5
0
.8
</p>
<p>0
0
.8
</p>
<p>5
Random Forest Ensemble
</p>
<p>m
</p>
<p>M
e
a
n
 S
</p>
<p>q
u
a
re
</p>
<p>d
 E
</p>
<p>rr
o
r 
</p>
<p>a
n
d
 S
</p>
<p>q
u
a
re
</p>
<p>d
 B
</p>
<p>ia
s
</p>
<p>V
a
ri
a
n
c
e
</p>
<p>Mean Squared Error
Squared Bias
Variance
</p>
<p>0
.0
</p>
<p>0
.0
</p>
<p>5
0
.1
</p>
<p>0
0
.1
</p>
<p>5
0
.2
</p>
<p>0
</p>
<p>FIGURE 15.10. Simulation results. The left panel shows the average variance of
a single random forest tree, as a function of m. &ldquo;Within Z&rdquo; refers to the average
within-sample contribution to the variance, resulting from the bootstrap sampling
and split-variable sampling (15.9). &ldquo;Total&rdquo; includes the sampling variability of
Z. The horizontal line is the average variance of a single fully grown tree (with-
out bootstrap sampling). The right panel shows the average mean-squared error,
squared bias and variance of the ensemble, as a function of m. Note that the
variance axis is on the right (same scale, different level). The horizontal line is
the average squared-bias of a fully grown tree.
</p>
<p>15.4.2 Bias
</p>
<p>As in bagging, the bias of a random forest is the same as the bias of any
of the individual sampled trees T (x; Θ(Z)):</p>
<p/>
</div>
<div class="page"><p/>
<p>15.4 Analysis of Random Forests 601
</p>
<p>Bias(x) = μ(x)&minus; EZf̂rf(x)
= μ(x)&minus; EZEΘ|ZT (x; Θ(Z)). (15.10)
</p>
<p>This is also typically greater (in absolute terms) than the bias of an un-
pruned tree grown to Z, since the randomization and reduced sample space
impose restrictions. Hence the improvements in prediction obtained by bag-
ging or random forests are solely a result of variance reduction.
Any discussion of bias depends on the unknown true function. Fig-
</p>
<p>ure 15.10 (right panel) shows the squared bias for our additive model simu-
lation (estimated from the 500 realizations). Although for different models
the shape and rate of the bias curves may differ, the general trend is that
as m decreases, the bias increases. Shown in the figure is the mean-squared
error, and we see a classical bias-variance trade-off in the choice of m. For
all m the squared bias of the random forest is greater than that for a single
tree (horizontal line).
These patterns suggest a similarity with ridge regression (Section 3.4.1).
</p>
<p>Ridge regression is useful (in linear models) when one has a large number
of variables with similarly sized coefficients; ridge shrinks their coefficients
toward zero, and those of strongly correlated variables toward each other.
Although the size of the training sample might not permit all the variables
to be in the model, this regularization via ridge stabilizes the model and al-
lows all the variables to have their say (albeit diminished). Random forests
with small m perform a similar averaging. Each of the relevant variables
get their turn to be the primary split, and the ensemble averaging reduces
the contribution of any individual variable. Since this simulation exam-
ple (15.8) is based on a linear model in all the variables, ridge regression
achieves a lower mean-squared error (about 0.45 with df(λopt) &asymp; 29).
</p>
<p>15.4.3 Adaptive Nearest Neighbors
</p>
<p>The random forest classifier has much in common with the k-nearest neigh-
bor classifier (Section 13.3); in fact a weighted version thereof. Since each
tree is grown to maximal size, for a particular Θ&lowast;, T (x; Θ&lowast;(Z)) is the re-
sponse value for one of the training samples4. The tree-growing algorithm
finds an &ldquo;optimal&rdquo; path to that observation, choosing the most informative
predictors from those at its disposal. The averaging process assigns weights
to these training responses, which ultimately vote for the prediction. Hence
via the random-forest voting mechanism, those observations close to the
target point get assigned weights&mdash;an equivalent kernel&mdash;which combine to
form the classification decision.
Figure 15.11 demonstrates the similarity between the decision boundary
</p>
<p>of 3-nearest neighbors and random forests on the mixture data.
</p>
<p>4We gloss over the fact that pure nodes are not split further, and hence there can be
</p>
<p>more than one observation in a terminal node</p>
<p/>
</div>
<div class="page"><p/>
<p>602 15. Random Forests
</p>
<p>Random Forest Classifier
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>Training Error: 0.000
Test Error:       0.238
Bayes Error:    0.210
</p>
<p>3&minus;Nearest Neighbors
</p>
<p>o
o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
oo o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
o
</p>
<p>o
oo
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
</p>
<p>o
o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>o
</p>
<p>Training Error: 0.130
Test Error:       0.242
Bayes Error:    0.210
</p>
<p>FIGURE 15.11. Random forests versus 3-NN on the mixture data. The axis-ori-
ented nature of the individual trees in a random forest lead to decision regions
with an axis-oriented flavor.
</p>
<p>Bibliographic Notes
</p>
<p>Random forests as described here were introduced by Breiman (2001), al-
though many of the ideas had cropped up earlier in the literature in dif-
ferent forms. Notably Ho (1995) introduced the term &ldquo;random forest,&rdquo; and
used a consensus of trees grown in random subspaces of the features. The
idea of using stochastic perturbation and averaging to avoid overfitting was
introduced by Kleinberg (1990), and later in Kleinberg (1996). Amit and
Geman (1997) used randomized trees grown on image features for image
classification problems. Breiman (1996a) introduced bagging, a precursor
to his version of random forests. Dietterich (2000b) also proposed an im-
provement on bagging using additional randomization. His approach was
to rank the top 20 candidate splits at each node, and then select from the
list at random. He showed through simulations and real examples that this
additional randomization improved over the performance of bagging. Fried-
man and Hall (2007) showed that sub-sampling (without replacement) is
an effective alternative to bagging. They showed that growing and aver-
aging trees on samples of size N/2 is approximately equivalent (in terms
bias/variance considerations) to bagging, while using smaller fractions of
N reduces the variance even further (through decorrelation).
</p>
<p>There are several free software implementations of random forests. In
this chapter we used the randomForest package in R, maintained by Andy
Liaw, available from the CRAN website. This allows both split-variable se-
lection, as well as sub-sampling. Adele Cutler maintains a random forest
website http://www.math.usu.edu/&sim;adele/forests/ where (as of Au-
gust 2008) the software written by Leo Breiman and Adele Cutler is freely</p>
<p/>
</div>
<div class="page"><p/>
<p>Exercises 603
</p>
<p>available. Their code, and the name &ldquo;random forests&rdquo;, is exclusively li-
censed to Salford Systems for commercial release. The Weka machine learn-
ing archive http://www.cs.waikato.ac.nz/ml/weka/ at Waikato Univer-
sity, New Zealand, offers a free java implementation of random forests.
</p>
<p>Exercises
</p>
<p>Ex. 15.1 Derive the variance formula (15.1). This appears to fail if ρ is
negative; diagnose the problem in this case.
</p>
<p>Ex. 15.2 Show that as the number of bootstrap samples B gets large, the
oob error estimate for a random forest approaches its N -fold CV error
estimate, and that in the limit, the identity is exact.
</p>
<p>Ex. 15.3 Consider the simulation model used in Figure 15.7 (Mease and
Wyner, 2008). Binary observations are generated with probabilities
</p>
<p>Pr(Y = 1|X) = q + (1&minus; 2q) &middot; I
</p>
<p>
</p>
<p>
</p>
<p>J
&sum;
</p>
<p>j=1
</p>
<p>Xj &gt; J/2
</p>
<p>
</p>
<p>, (15.11)
</p>
<p>where X &sim; U [0, 1]p, 0 &le; q &le; 1
2
, and J &le; p is some predefined (even)
</p>
<p>number. Describe this probability surface, and give the Bayes error rate.
</p>
<p>Ex. 15.4 Suppose xi, i = 1, . . . , N are iid (&micro;, σ
2). Let x̄&lowast;1 and x̄
</p>
<p>&lowast;
2 be two
</p>
<p>bootstrap realizations of the sample mean. Show that the sampling cor-
relation corr(x̄&lowast;1, x̄
</p>
<p>&lowast;
2) =
</p>
<p>n
2n&minus;1 &asymp; 50%. Along the way, derive var(x̄
</p>
<p>&lowast;
1) and
</p>
<p>the variance of the bagged mean x̄bag. Here x̄ is a linear statistic; bagging
produces no reduction in variance for linear statistics.
</p>
<p>Ex. 15.5 Show that the sampling correlation between a pair of random-
forest trees at a point x is given by
</p>
<p>ρ(x) =
VarZ[EΘ|ZT (x; Θ(Z))]
</p>
<p>VarZ[EΘ|ZT (x; Θ(Z))] + EZVarΘ|Z[T (x; Θ(Z)]
. (15.12)
</p>
<p>The term in the numerator is VarZ[f̂rf(x)], and the second term in the
denominator is the expected conditional variance due to the randomization
in random forests.
</p>
<p>Ex. 15.6 Fit a series of random-forest classifiers to the spam data, to explore
the sensitivity to the parameter m. Plot both the oob error as well as the
test error against a suitably chosen range of values for m.</p>
<p/>
</div>
<div class="page"><p/>
<p>604 15. Random Forests
</p>
<p>Ex. 15.7 Suppose we fit a linear regression model to N observations with
response yi and predictors xi1, . . . , xip. Assume that all variables are stan-
dardized to have mean zero and standard deviation one. Let RSS be the
mean-squared residual on the training data, and β̂ the estimated coefficient.
Denote by RSS&lowast;j the mean-squared residual on the training data using the
</p>
<p>same β̂, but with the N values for the jth variable randomly permuted
before the predictions are calculated. Show that
</p>
<p>EP [RSS
&lowast;
j &minus;RSS] = 2β̂2j , (15.13)
</p>
<p>where EP denotes expectation with respect to the permutation distribution.
Argue that this is approximately true when the evaluations are done using
an independent test set.</p>
<p/>
</div>
<div class="page"><p/>
<p>16
Ensemble Learning
</p>
<p>16.1 Introduction
</p>
<p>The idea of ensemble learning is to build a prediction model by combining
the strengths of a collection of simpler base models. We have already seen
a number of examples that fall into this category.
Bagging in Section 8.7 and random forests in Chapter 15 are ensemble
</p>
<p>methods for classification, where a committee of trees each cast a vote for
the predicted class. Boosting in Chapter 10 was initially proposed as a
committee method as well, although unlike random forests, the committee
of weak learners evolves over time, and the members cast a weighted vote.
Stacking (Section 8.8) is a novel approach to combining the strengths of
a number of fitted models. In fact one could characterize any dictionary
method, such as regression splines, as an ensemble method, with the basis
functions serving the role of weak learners.
Bayesian methods for nonparametric regression can also be viewed as
</p>
<p>ensemble methods: a large number of candidate models are averaged with
respect to the posterior distribution of their parameter settings (e.g. (Neal
and Zhang, 2006)).
Ensemble learning can be broken down into two tasks: developing a pop-
</p>
<p>ulation of base learners from the training data, and then combining them
to form the composite predictor. In this chapter we discuss boosting tech-
nology that goes a step further; it builds an ensemble model by conducting
a regularized and supervised search in a high-dimensional space of weak
learners.
</p>
<p> &copy; Springer Science+Business Media, LLC 2009 
</p>
<p>T. Hastie et al., The Elements of Statistical Learning, Second Edition, 605
</p>
<p>DOI: 10.1007/b94608_16,</p>
<p/>
</div>
<div class="page"><p/>
<p>606 16. Ensemble Learning
</p>
<p>An early example of a learning ensemble is a method designed for multi-
class classification using error-correcting output codes (Dietterich and Bakiri,
1995, ECOC). Consider the 10-class digit classification problem, and the
coding matrix C given in Table 16.1.
</p>
<p>TABLE 16.1. Part of a 15-bit error-correcting coding matrix C for the 10-class
digit classification problem. Each column defines a two-class classification prob-
lem.
</p>
<p>Digit C1 C2 C3 C4 C5 C6 &middot; &middot; &middot; C15
0 1 1 0 0 0 0 &middot; &middot; &middot; 1
1 0 0 1 1 1 1 &middot; &middot; &middot; 0
2 1 0 0 1 0 0 &middot; &middot; &middot; 1
..
.
</p>
<p>..
</p>
<p>.
..
.
</p>
<p>..
</p>
<p>.
..
.
</p>
<p>..
</p>
<p>.
..
. &middot; &middot; &middot;
</p>
<p>..
</p>
<p>.
8 1 1 0 1 0 1 &middot; &middot; &middot; 1
9 0 1 1 1 0 0 &middot; &middot; &middot; 0
</p>
<p>Note that the ℓth column of the coding matrix Cℓ defines a two-class
variable that merges all the original classes into two groups. The method
works as follows:
</p>
<p>1. Learn a separate classifier for each of the L = 15 two class problems
defined by the columns of the coding matrix.
</p>
<p>2. At a test point x, let p̂ℓ(x) be the predicted probability of a one for
the ℓth response.
</p>
<p>3. Define δk(x) =
&sum;L
</p>
<p>ℓ=1 |Ckℓ &minus; p̂ℓ(x)|, the discriminant function for the
kth class, where Ckℓ is the entry for row k and column ℓ in Table 16.1.
</p>
<p>Each row of C is a binary code for representing that class. The rows have
more bits than is necessary, and the idea is that the redundant &ldquo;error-
correcting&rdquo; bits allow for some inaccuracies, and can improve performance.
In fact, the full code matrix C above has a minimum Hamming distance1
</p>
<p>of 7 between any pair of rows. Note that even the indicator response coding
(Section 4.2) is redundant, since 10 classes require only &lceil;log2 10 = 4 bits for
their unique representation. Dietterich and Bakiri (1995) showed impressive
improvements in performance for a variety of multiclass problems when
classification trees were used as the base classifier.
James and Hastie (1998) analyzed the ECOC approach, and showed
</p>
<p>that random code assignment worked as well as the optimally constructed
error-correcting codes. They also argued that the main benefit of the coding
was in variance reduction (as in bagging and random forests), because the
different coded problems resulted in different trees, and the decoding step
(3) above has a similar effect as averaging.
</p>
<p>1The Hamming distance between two vectors is the number of mismatches between
</p>
<p>corresponding entries.</p>
<p/>
</div>
<div class="page"><p/>
<p>16.2 Boosting and Regularization Paths 607
</p>
<p>16.2 Boosting and Regularization Paths
</p>
<p>In Section 10.12.2 of the first edition of this book, we suggested an analogy
between the sequence of models produced by a gradient boosting algorithm
and regularized model fitting in high-dimensional feature spaces. This was
primarily motivated by observing the close connection between a boosted
version of linear regression and the lasso (Section 3.4.2). These connec-
tions have been pursued by us and others, and here we present our current
thinking in this area. We start with the original motivation, which fits more
naturally in this chapter on ensemble learning.
</p>
<p>16.2.1 Penalized Regression
</p>
<p>Intuition for the success of the shrinkage strategy (10.41) of gradient boost-
ing (page 364 in Chapter 10) can be obtained by drawing analogies with
penalized linear regression with a large basis expansion. Consider the dic-
tionary of all possible J-terminal node regression trees T = {Tk} that could
be realized on the training data as basis functions in IRp. The linear model
is
</p>
<p>f(x) =
</p>
<p>K&sum;
</p>
<p>k=1
</p>
<p>αkTk(x), (16.1)
</p>
<p>where K = card(T ). Suppose the coefficients are to be estimated by least
squares. Since the number of such trees is likely to be much larger than
even the largest training data sets, some form of regularization is required.
Let α̂(λ) solve
</p>
<p>min
α
</p>
<p>⎧
⎨
⎩
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>(
yi &minus;
</p>
<p>K&sum;
</p>
<p>k=1
</p>
<p>αkTk(xi)
</p>
<p>)2
+ λ &middot; J(α)
</p>
<p>⎫
⎬
⎭ , (16.2)
</p>
<p>J(α) is a function of the coefficients that generally penalizes larger values.
Examples are
</p>
<p>J(α) =
</p>
<p>K&sum;
</p>
<p>k=1
</p>
<p>|αk|2 ridge regression, (16.3)
</p>
<p>J(α) =
</p>
<p>K&sum;
</p>
<p>k=1
</p>
<p>|αk| lasso, (16.4)
</p>
<p>(16.5)
</p>
<p>both covered in Section 3.4. As discussed there, the solution to the lasso
problem with moderate to large λ tends to be sparse; many of the α̂k(λ) =
0. That is, only a small fraction of all possible trees enter the model (16.1).</p>
<p/>
</div>
<div class="page"><p/>
<p>608 16. Ensemble Learning
</p>
<p>Algorithm 16.1 Forward Stagewise Linear Regression.
</p>
<p>1. Initialize α̌k = 0, k = 1, . . . ,K. Set ε &gt; 0 to some small constant,
and M large.
</p>
<p>2. For m = 1 to M :
</p>
<p>(a) (β&lowast;, k&lowast;) = argminβ,k
&sum;N
</p>
<p>i=1
</p>
<p>(
yi &minus;
</p>
<p>&sum;K
l=1 α̌lTl(xi)&minus; βTk(xi)
</p>
<p>)2
.
</p>
<p>(b) α̌k&lowast; &larr; α̌k&lowast; + ε &middot; sign(β&lowast;).
</p>
<p>3. Output fM (x) =
&sum;K
</p>
<p>k=1 α̌kTk(x).
</p>
<p>This seems reasonable since it is likely that only a small fraction of all pos-
sible trees will be relevant in approximating any particular target function.
However, the relevant subset will be different for different targets. Those
coefficients that are not set to zero are shrunk by the lasso in that their
absolute values are smaller than their corresponding least squares values2:
| α̂k(λ) | &lt; | α̂k(0) |. As λ increases, the coefficients all shrink, each one
ultimately becoming zero.
Owing to the very large number of basis functions Tk, directly solving
</p>
<p>(16.2) with the lasso penalty (16.4) is not possible. However, a feasible
forward stagewise strategy exists that closely approximates the effect of
the lasso, and is very similar to boosting and the forward stagewise Algo-
rithm 10.2. Algorithm 16.1 gives the details. Although phrased in terms
of tree basis functions Tk, the algorithm can be used with any set of ba-
sis functions. Initially all coefficients are zero in line 1; this corresponds
to λ = &infin; in (16.2). At each successive step, the tree Tk&lowast; is selected that
best fits the current residuals in line 2(a). Its corresponding coefficient α̌k&lowast;
</p>
<p>is then incremented or decremented by an infinitesimal amount in 2(b),
while all other coefficients α̌k, k �= k&lowast; are left unchanged. In principle, this
process could be iterated until either all the residuals are zero, or β&lowast; = 0.
The latter case can occur if K &lt; N , and at that point the coefficient values
represent a least squares solution. This corresponds to λ = 0 in (16.2).
After applying Algorithm 16.1 with M &lt; &infin; iterations, many of the coef-
</p>
<p>ficients will be zero, namely, those that have yet to be incremented. The oth-
ers will tend to have absolute values smaller than their corresponding least
squares solution values, | α̌k(M) | &lt; | α̂k(0) |. Therefore this M -iteration
solution qualitatively resembles the lasso, with M inversely related to λ.
</p>
<p>Figure 16.1 shows an example, using the prostate data studied in Chap-
ter 3. Here, instead of using trees Tk(X) as basis functions, we use the origi-
</p>
<p>2If K &gt; N , there is in general no unique &ldquo;least squares value,&rdquo; since infinitely many
solutions will exist that fit the data perfectly. We can pick the minimum L1-norm solution
</p>
<p>amongst these, which is the unique lasso solution.</p>
<p/>
</div>
<div class="page"><p/>
<p>16.2 Boosting and Regularization Paths 609
</p>
<p>&minus;
0
.2
</p>
<p>0
.0
</p>
<p>0
.2
</p>
<p>0
.4
</p>
<p>0
.6
</p>
<p>lcavol
</p>
<p>lweight
</p>
<p>age
</p>
<p>lbph
</p>
<p>svi
</p>
<p>lcp
</p>
<p>gleason
</p>
<p>pgg45
</p>
<p>0.0 0.5 1.0 1.5 2.0
</p>
<p>&minus;
0
.2
</p>
<p>0
.0
</p>
<p>0
.2
</p>
<p>0
.4
</p>
<p>0
.6
</p>
<p>lcavol
</p>
<p>lweight
</p>
<p>age
</p>
<p>lbph
</p>
<p>svi
</p>
<p>lcp
</p>
<p>gleason
</p>
<p>pgg45
</p>
<p>0 50 100 150 200
</p>
<p>t =
&sum;
</p>
<p>k |αk|
</p>
<p>C
o
effi
</p>
<p>ci
en
ts
</p>
<p>C
o
effi
</p>
<p>ci
en
ts
</p>
<p>Lasso Forward Stagewise
</p>
<p>Iteration
</p>
<p>FIGURE 16.1. Profiles of estimated coefficients from linear regression, for the
prostate data studied in Chapter 3. The left panel shows the results from the lasso,
for different values of the bound parameter t =
</p>
<p>&sum;
</p>
<p>k |αk|. The right panel shows
the results of the stagewise linear regression Algorithm 16.1, using M = 220
consecutive steps of size ε = .01.
</p>
<p>nal variables Xk themselves; that is, a multiple linear regression model. The
left panel displays the profiles of estimated coefficients from the lasso, for
different values of the bound parameter t =
</p>
<p>&sum;
k |αk|. The right panel shows
</p>
<p>the results of the stagewise Algorithm 16.1, with M = 250 and ε = 0.01.
[The left and right panels of Figure 16.1 are the same as Figure 3.10 and
the left panel of Figure 3.19, respectively.] The similarity between the two
graphs is striking.
In some situations the resemblance is more than qualitative. For example,
</p>
<p>if all of the basis functions Tk are mutually uncorrelated, then as ε &darr; 0, M &uarr;
such that Mǫ &rarr; t, Algorithm 16.1 yields exactly the same solution as the
lasso for bound parameter t =
</p>
<p>&sum;
k |αk| (and likewise for all solutions along
</p>
<p>the path). Of course, tree-based regressors are not uncorrelated. However,
the solution sets are also identical if the coefficients α̂k(λ) are all monotone
functions of λ. This is often the case when the correlation between the
variables is low. When the α̂k(λ) are not monotone in λ, then the solution
sets are not identical. The solution sets for Algorithm 16.1 tend to change
less rapidly with changing values of the regularization parameter than those
of the lasso.</p>
<p/>
</div>
<div class="page"><p/>
<p>610 16. Ensemble Learning
</p>
<p>Efron et al. (2004) make the connections more precise, by characterizing
the exact solution paths in the ε-limiting case. They show that the coeffi-
cient paths are piece-wise linear functions, both for the lasso and forward
stagewise. This facilitates efficient algorithms which allow the entire paths
to be computed with the same cost as a single least-squares fit. This least
angle regression algorithm is described in more detail in Section 3.8.1.
Hastie et al. (2007) show that this infinitesimal forward stagewise algo-
</p>
<p>rithm (FS0) fits a monotone version of the lasso, which optimally reduces
at each step the loss function for a given increase in the arc length of the
coefficient path (see Sections 16.2.3 and 3.8.1). The arc-length for the ǫ &gt; 0
case is Mǫ, and hence proportional to the number of steps.
</p>
<p>Tree boosting (Algorithm 10.3) with shrinkage (10.41) closely resembles
Algorithm 16.1, with the learning rate parameter ν corresponding to ε. For
squared error loss, the only difference is that the optimal tree to be selected
at each iteration Tk&lowast; is approximated by the standard top-down greedy
tree-induction algorithm. For other loss functions, such as the exponential
loss of AdaBoost and the binomial deviance, Rosset et al. (2004a) show
similar results to what we see here. Thus, one can view tree boosting with
shrinkage as a form of monotone ill-posed regression on all possible (J-
terminal node) trees, with the lasso penalty (16.4) as a regularizer. We
return to this topic in Section 16.2.3.
The choice of no shrinkage [ν = 1 in equation (10.41)] is analogous to
</p>
<p>forward-stepwise regression, and its more aggressive cousin best-subset se-
lection, which penalizes the number of non zero coefficients J(α) =
</p>
<p>&sum;
k |αk|0.
</p>
<p>With a small fraction of dominant variables, best subset approaches often
work well. But with a moderate fraction of strong variables, it is well known
that subset selection can be excessively greedy (Copas, 1983), often yielding
poor results when compared to less aggressive strategies such as the lasso
or ridge regression. The dramatic improvements often seen when shrinkage
is used with boosting are yet another confirmation of this approach.
</p>
<p>16.2.2 The &ldquo;Bet on Sparsity&rdquo; Principle
</p>
<p>As shown in the previous section, boosting&rsquo;s forward stagewise strategy
with shrinkage approximately minimizes the same loss function with a
lasso-style L1 penalty. The model is built up slowly, searching through
&ldquo;model space&rdquo; and adding shrunken basis functions derived from impor-
tant predictors. In contrast, the L2 penalty is computationally much easier
to deal with, as shown in Section 12.3.7. With the basis functions and L2
penalty chosen to match a particular positive-definite kernel, one can solve
the corresponding optimization problem without explicitly searching over
individual basis functions.
However, the sometimes superior performance of boosting over proce-
</p>
<p>dures such as the support vector machine may be largely due to the im-
plicit use of the L1 versus L2 penalty. The shrinkage resulting from the</p>
<p/>
</div>
<div class="page"><p/>
<p>16.2 Boosting and Regularization Paths 611
</p>
<p>L1 penalty is better suited to sparse situations, where there are few basis
functions with nonzero coefficients (among all possible choices).
We can strengthen this argument through a simple example, taken from
</p>
<p>Friedman et al. (2004). Suppose we have 10, 000 data points and our model
is a linear combination of a million trees. If the true population coefficients
of these trees arose from a Gaussian distribution, then we know that in a
Bayesian sense the best predictor is ridge regression (Exercise 3.6). That is,
we should use an L2 rather than an L1 penalty when fitting the coefficients.
On the other hand, if there are only a small number (e.g., 1000) coefficients
that are nonzero, the lasso (L1 penalty) will work better. We think of this
as a sparse scenario, while the first case (Gaussian coefficients) is dense.
Note however that in the dense scenario, although the L2 penalty is best,
neither method does very well since there is too little data from which to
estimate such a large number of nonzero coefficients. This is the curse of
dimensionality taking its toll. In a sparse setting, we can potentially do
well with the L1 penalty, since the number of nonzero coefficients is small.
The L2 penalty fails again.
In other words, use of the L1 penalty follows what we call the &ldquo;bet on
</p>
<p>sparsity&rdquo; principle for high-dimensional problems:
</p>
<p>Use a procedure that does well in sparse problems, since no pro-
cedure does well in dense problems.
</p>
<p>These comments need some qualification:
</p>
<p>&bull; For any given application, the degree of sparseness/denseness depends
on the unknown true target function, and the chosen dictionary T .
</p>
<p>&bull; The notion of sparse versus dense is relative to the size of the train-
ing data set and/or the noise-to-signal ratio (NSR). Larger training
sets allow us to estimate coefficients with smaller standard errors.
Likewise in situations with small NSR, we can identify more nonzero
coefficients with a given sample size than in situations where the NSR
is larger.
</p>
<p>&bull; The size of the dictionary plays a role as well. Increasing the size of the
dictionary may lead to a sparser representation for our function, but
the search problem becomes more difficult leading to higher variance.
</p>
<p>Figure 16.2 illustrates these points in the context of linear models us-
ing simulation. We compare ridge regression and lasso, both for classifi-
cation and regression problems. Each run has 50 observations with 300
independent Gaussian predictors. In the top row all 300 coefficients are
nonzero, generated from a Gaussian distribution. In the middle row, only
10 are nonzero and generated from a Gaussian, and the last row has 30
non zero Gaussian coefficients. For regression, standard Gaussian noise is</p>
<p/>
</div>
<div class="page"><p/>
<p>612 16. Ensemble Learning
</p>
<p>0.1 0.2 0.3 0.4 0.5
</p>
<p>0
.0
</p>
<p>0
.2
</p>
<p>0
.4
</p>
<p>0
.6
</p>
<p>0
.8
</p>
<p>1
.0
</p>
<p>Lasso/Gaussian
</p>
<p>0
.0
</p>
<p>0
.2
</p>
<p>0
.4
</p>
<p>0
.6
</p>
<p>0
.8
</p>
<p>1
.0
</p>
<p>0.1 0.2 0.3 0.4 0.5
</p>
<p>Ridge/Gaussian
</p>
<p>0.1 0.2 0.3 0.4 0.5
</p>
<p>0
.0
</p>
<p>0
.2
</p>
<p>0
.4
</p>
<p>0
.6
</p>
<p>0
.8
</p>
<p>1
.0
</p>
<p>Lasso/Subset 10
</p>
<p>0
.0
</p>
<p>0
.2
</p>
<p>0
.4
</p>
<p>0
.6
</p>
<p>0
.8
</p>
<p>1
.0
</p>
<p>0.1 0.2 0.3 0.4 0.5
</p>
<p>Ridge/Subset 10
</p>
<p>0.1 0.2 0.3 0.4 0.5
</p>
<p>0
.0
</p>
<p>0
.2
</p>
<p>0
.4
</p>
<p>0
.6
</p>
<p>0
.8
</p>
<p>1
.0
</p>
<p>Lasso/Subset 30
</p>
<p>0
.0
</p>
<p>0
.2
</p>
<p>0
.4
</p>
<p>0
.6
</p>
<p>0
.8
</p>
<p>1
.0
</p>
<p>0.1 0.2 0.3 0.4 0.5
</p>
<p>Ridge/Subset 30P
e
rc
</p>
<p>e
n
ta
</p>
<p>g
e
 S
</p>
<p>q
u
a
re
</p>
<p>d
 P
</p>
<p>re
d
ic
</p>
<p>ti
o
n
 E
</p>
<p>rr
o
r 
</p>
<p>E
x
p
la
</p>
<p>in
e
d
</p>
<p>Noise&minus;to&minus;Signal Ratio
</p>
<p>Regression
</p>
<p>0.1 0.2 0.3 0.4 0.5
</p>
<p>0
.0
</p>
<p>0
.2
</p>
<p>0
.4
</p>
<p>0
.6
</p>
<p>0
.8
</p>
<p>1
.0
</p>
<p>Lasso/Gaussian
</p>
<p>0
.0
</p>
<p>0
.2
</p>
<p>0
.4
</p>
<p>0
.6
</p>
<p>0
.8
</p>
<p>1
.0
</p>
<p>0.1 0.2 0.3 0.4 0.5
</p>
<p>Ridge/Gaussian
</p>
<p>0.1 0.2 0.3 0.4 0.5
</p>
<p>0
.0
</p>
<p>0
.2
</p>
<p>0
.4
</p>
<p>0
.6
</p>
<p>0
.8
</p>
<p>1
.0
</p>
<p>Lasso/Subset 10
</p>
<p>0
.0
</p>
<p>0
.2
</p>
<p>0
.4
</p>
<p>0
.6
</p>
<p>0
.8
</p>
<p>1
.0
</p>
<p>0.1 0.2 0.3 0.4 0.5
</p>
<p>Ridge/Subset 10
</p>
<p>0.1 0.2 0.3 0.4 0.5
</p>
<p>0
.0
</p>
<p>0
.2
</p>
<p>0
.4
</p>
<p>0
.6
</p>
<p>0
.8
</p>
<p>1
.0
</p>
<p>Lasso/Subset 30
</p>
<p>0
.0
</p>
<p>0
.2
</p>
<p>0
.4
</p>
<p>0
.6
</p>
<p>0
.8
</p>
<p>1
.0
</p>
<p>0.1 0.2 0.3 0.4 0.5
</p>
<p>Ridge/Subset 30P
e
rc
</p>
<p>e
n
ta
</p>
<p>g
e
 M
</p>
<p>is
c
la
</p>
<p>s
s
if
ic
</p>
<p>a
ti
o
n
 E
</p>
<p>rr
o
r 
</p>
<p>E
x
p
la
</p>
<p>in
e
d
</p>
<p>Noise&minus;to&minus;Signal Ratio
</p>
<p>Classification
</p>
<p>FIGURE 16.2. Simulations that show the superiority of the L1 (lasso) penalty
over L2 (ridge) in regression and classification. Each run has 50 observations
with 300 independent Gaussian predictors. In the top row all 300 coefficients are
nonzero, generated from a Gaussian distribution. In the middle row, only 10 are
nonzero, and the last row has 30 nonzero. Gaussian errors are added to the linear
predictor η(X) for the regression problems, and binary responses generated via the
inverse-logit transform for the classification problems. Scaling of η(X) resulted in
the noise-to-signal ratios shown. Lasso is used in the left sub-columns, ridge in the
right. We report the optimal percentage of error explained on test data (relative
to the error of a constant model), displayed as boxplots over 20 realizations for
each combination. In the only situation where ridge beats lasso (top row), neither
do well.</p>
<p/>
</div>
<div class="page"><p/>
<p>16.2 Boosting and Regularization Paths 613
</p>
<p>added to the linear predictor η(X) = XTβ to produce a continuous re-
sponse. For classification the linear predictor is transformed via the inverse-
logit to a probability, and a binary response is generated. Five differ-
ent noise-to-signal ratios are presented, obtained by scaling η(X) prior
to generating the response. In both cases this is defined to be NSR =
Var(Y |η(X))/Var(η(X)). Both the ridge regression and lasso coefficient
paths were fit using a series of 50 values of λ corresponding to a range of
df from 1 to 50 (see Chapter 3 for details). The models were evaluated on
a large test set (infinite for Gaussian, 5000 for binary), and in each case the
value for λ was chosen to minimize the test-set error. We report percentage
variance explained for the regression problems, and percentage misclassifi-
cation error explained for the classification problems (relative to a baseline
error of 0.5). There are 20 simulation runs for each scenario.
Note that for the classification problems, we are using squared-error loss
</p>
<p>to fit the binary response. Note also that we do not using the training
data to select λ, but rather are reporting the best possible behavior for
each method in the different scenarios. The L2 penalty performs poorly
everywhere. The Lasso performs reasonably well in the only two situations
where it can (sparse coefficients). As expected the performance gets worse
as the NSR increases (less so for classification), and as the model becomes
denser. The differences are less marked for classification than for regression.
These empirical results are supported by a large body of theoretical
</p>
<p>results (Donoho and Johnstone, 1994; Donoho and Elad, 2003; Donoho,
2006b; Candes and Tao, 2007) that support the superiority of L1 estimation
in sparse settings.
</p>
<p>16.2.3 Regularization Paths, Over-fitting and Margins
</p>
<p>It has often been observed that boosting &ldquo;does not overfit,&rdquo; or more as-
tutely is &ldquo;slow to overfit.&rdquo; Part of the explanation for this phenomenon was
made earlier for random forests &mdash; misclassification error is less sensitive to
variance than is mean-squared error, and classification is the major focus
in the boosting community. In this section we show that the regulariza-
tion paths of boosted models are &ldquo;well behaved,&rdquo; and that for certain loss
functions they have an appealing limiting form.
Figure 16.3 shows the coefficient paths for lasso and infinitesimal forward
</p>
<p>stagewise (FS0) in a simulated regression setting. The data consists of a
dictionary of 1000 Gaussian variables, strongly correlated (ρ = 0.95) within
blocks of 20, but uncorrelated between blocks. The generating model has
nonzero coefficients for 50 variables, one drawn from each block, and the
coefficient values are drawn from a standard Gaussian. Finally, Gaussian
noise is added, with a noise-to-signal ratio of 0.72 (Exercise 16.1.) The
FS0 algorithm is a limiting form of algorithm 16.1, where the step size ε
is shrunk to zero (Section 3.8.1). The grouping of the variables is intended
to mimic the correlations of nearby trees, and with the forward-stagewise</p>
<p/>
</div>
<div class="page"><p/>
<p>614 16. Ensemble Learning
</p>
<p>0.0 0.2 0.4 0.6 0.8 1.0
</p>
<p>&minus;
2
</p>
<p>0
&minus;
</p>
<p>1
0
</p>
<p>0
1
</p>
<p>0
2
</p>
<p>0
3
</p>
<p>0
</p>
<p>S
ta
</p>
<p>n
d
</p>
<p>a
rd
</p>
<p>iz
e
</p>
<p>d
 C
</p>
<p>o
e
</p>
<p>ff
ic
</p>
<p>ie
n
</p>
<p>ts
LASSO
</p>
<p>0.0 0.2 0.4 0.6 0.8 1.0
&minus;
</p>
<p>2
0
</p>
<p>&minus;
1
</p>
<p>0
0
</p>
<p>1
0
</p>
<p>2
0
</p>
<p>3
0
</p>
<p>S
ta
</p>
<p>n
d
</p>
<p>a
rd
</p>
<p>iz
e
</p>
<p>d
 C
</p>
<p>o
e
</p>
<p>ff
ic
</p>
<p>ie
n
</p>
<p>ts
</p>
<p>Forward Stagewise
</p>
<p>|α(m)|/|α(&infin;)||α(m)|/|α(&infin;)|
</p>
<p>FIGURE 16.3. Comparison of lasso and infinitesimal forward stagewise paths
on simulated regression data. The number of samples is 60 and the number of
variables is 1000. The forward-stagewise paths fluctuate less than those of lasso
in the final stages of the algorithms.
</p>
<p>algorithm, this setup is intended as an idealized version of gradient boosting
with shrinkage. For both these algorithms, the coefficient paths can be
computed exactly, since they are piecewise linear (see the LARS algorithm
in Section 3.8.1).
Here the coefficient profiles are similar only in the early stages of the
</p>
<p>paths. For the later stages, the forward stagewise paths tend to be mono-
tone and smoother, while those for the lasso fluctuate widely. This is due
to the strong correlations among subsets of the variables &mdash;lasso suffers
somewhat from the multi-collinearity problem (Exercise 3.28).
The performance of the two models is rather similar (Figure 16.4), and
</p>
<p>they achieve about the same minimum. In the later stages forward stagewise
takes longer to overfit, a likely consequence of the smoother paths.
Hastie et al. (2007) show that FS0 solves a monotone version of the lasso
</p>
<p>problem for squared error loss. Let T a = T &cup; {&minus;T } be the augmented
dictionary obtained by including a negative copy of every basis element
in T . We consider models f(x) = &sum;Tk&isin;T a αkTk(x) with non-negative co-
efficients αk &ge; 0. In this expanded space, the lasso coefficient paths are
positive, while those of FS0 are monotone nondecreasing.
</p>
<p>The monotone lasso path is characterized by a differential equation
</p>
<p>&part;α
</p>
<p>&part;ℓ
= ρml(α(ℓ)), (16.6)</p>
<p/>
</div>
<div class="page"><p/>
<p>16.2 Boosting and Regularization Paths 615
</p>
<p>o
oo
o
</p>
<p>o
</p>
<p>oo
</p>
<p>o
o
oo
</p>
<p>oooo
o
oo
</p>
<p>oooooo
oo
</p>
<p>ooooooooo ooooooooooooooooooooooooooo
ooooooo
</p>
<p>ooooooo
oooooooo
</p>
<p>ooooo
ooooooooo
</p>
<p>ooooooo
oooooooo
</p>
<p>oooo
ooo
</p>
<p>oooooo
oooo
</p>
<p>ooo
ooooo
</p>
<p>oooooo
oo
</p>
<p>ooooooooo
oooooooooo
</p>
<p>ooooooooooooo
oooooooooooo
</p>
<p>ooooooooooooooooooooo
ooooooo
</p>
<p>o
oo
o
</p>
<p>oo
</p>
<p>oo
</p>
<p>ooo
ooooo
</p>
<p>oooooo
</p>
<p>ooooooo
ooooooooooooo
</p>
<p>ooooooooooo oooooooooooooooooooooo
ooooooooo
</p>
<p>ooooooooooooooo
ooooooooooooo
</p>
<p>ooooooooo
ooooooo
</p>
<p>ooooooooo
oooooooooooooo
</p>
<p>oooooooo
ooooo
</p>
<p>ooooooo
oooooooooooooooo
</p>
<p>ooooooooo
ooooooooooo
</p>
<p>oooo
ooooooo
</p>
<p>ooooooooo
ooooooooooooooooooo
</p>
<p>ooooooooooooooo
oooooooooooooooooooooooo
</p>
<p>oooooooooooooooooooooooooooooooooooooooooo
oooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooo
</p>
<p>0 10 20 30 40 50 60 70
</p>
<p>2
5
</p>
<p>3
0
</p>
<p>3
5
</p>
<p>4
0
</p>
<p>4
5
</p>
<p>5
0
</p>
<p>5
5
</p>
<p>Lasso
Forward Stagewise
</p>
<p>|α(m)|
</p>
<p>M
e
a
n
S
q
u
a
re
d
E
rr
o
r
</p>
<p>FIGURE 16.4. Mean squared error for lasso and infinitesimal forward stagewise
on the simulated data. Despite the difference in the coefficient paths, the two
models perform similarly over the critical part of the regularization path. In the
right tail, lasso appears to overfit more rapidly.
</p>
<p>with initial condition α(0) = 0, where ℓ is the L1 arc-length of the path
α(ℓ) (Exercise 16.2). The monotone lasso move direction (velocity vector)
ρml(α(ℓ)) decreases the loss at the optimal quadratic rate per unit increase
in the L1 arc-length of the path. Since ρ
</p>
<p>ml
k (α(ℓ)) &ge; 0 &forall;k, ℓ, the solution
</p>
<p>paths are monotone.
The lasso can similarly be characterized as the solution to a differential
</p>
<p>equation as in (16.6), except that the move directions decrease the loss
optimally per unit increase in the L1 norm of the path. As a consequence,
they are not necessarily positive, and hence the lasso paths need not be
monotone.
In this augmented dictionary, restricting the coefficients to be positive is
</p>
<p>natural, since it avoids an obvious ambiguity. It also ties in more naturally
with tree boosting&mdash;we always find trees positively correlated with the
current residual.
There have been suggestions that boosting performs well (for two-class
</p>
<p>classification) because it exhibits maximal-margin properties, much like the
support-vector machines of Chapters 4.5.2 and 12. Schapire et al. (1998)
define the normalized L1 margin of a fitted model f(x) =
</p>
<p>&sum;
k αkTk(x) as
</p>
<p>m(f) = min
i
</p>
<p>yif(xi)&sum;K
k=1 |αk|
</p>
<p>. (16.7)
</p>
<p>Here the minimum is taken over the training sample, and yi &isin; {&minus;1,+1}.
Unlike the L2 margin (4.40) of support vector machines, the L1 margin
m(f) measures the distance to the closest training point in L&infin; units (max-
imum coordinate distance).</p>
<p/>
</div>
<div class="page"><p/>
<p>616 16. Ensemble Learning
</p>
<p>&minus;
0
</p>
<p>.3
&minus;
</p>
<p>0
.2
</p>
<p>&minus;
0
</p>
<p>.1
0
</p>
<p>.0
0
</p>
<p>.1
</p>
<p>Number of Trees
</p>
<p>M
a
</p>
<p>rg
in
</p>
<p>0 2K 4K 6K 8K 10K
</p>
<p>0
.2
</p>
<p>5
0
</p>
<p>.2
6
</p>
<p>0
.2
</p>
<p>7
0
</p>
<p>.2
8
</p>
<p>Number of Trees
</p>
<p>T
e
</p>
<p>s
t 
</p>
<p>E
rr
</p>
<p>o
r
</p>
<p>0 2K 4K 6K 8K 10K
</p>
<p>FIGURE 16.5. The left panel shows the L1 margin m(f) for the Adaboost clas-
sifier on the mixture data, as a function of the number of 4-node trees. The model
was fit using the R package gbm, with a shrinkage factor of 0.02. After 10, 000
trees, m(f) has settled down. Note that when the margin crosses zero, the training
error becomes zero. The right panel shows the test error, which is minimized at
240 trees. In this case, Adaboost overfits dramatically if run to convergence.
</p>
<p>Schapire et al. (1998) prove that with separable data, Adaboost in-
creases m(f) with each iteration, converging to a margin-symmetric so-
lution. Rätsch and Warmuth (2002) prove the asymptotic convergence of
Adaboost with shrinkage to a L1-margin-maximizing solution. Rosset et
al. (2004a) consider regularized models of the form (16.2) for general loss
functions. They show that as λ &darr; 0, for particular loss functions the solution
converges to a margin-maximizing configuration. In particular they show
this to be the case for the exponential loss of Adaboost, as well as binomial
deviance.
Collecting together the results of this section, we reach the following
</p>
<p>summary for boosted classifiers:
</p>
<p>The sequence of boosted classifiers form an L1-regularized mono-
tone path to a margin-maximizing solution.
</p>
<p>Of course the margin-maximizing end of the path can be a very poor, overfit
solution, as it is in the example in Figure 16.5. Early stopping amounts
to picking a point along the path, and should be done with the aid of a
validation dataset.
</p>
<p>16.3 Learning Ensembles
</p>
<p>The insights learned from the previous sections can be harnessed to produce
a more effective and efficient ensemble model. Again we consider functions</p>
<p/>
</div>
<div class="page"><p/>
<p>16.3 Learning Ensembles 617
</p>
<p>of the form
f(x) = α0 +
</p>
<p>&sum;
</p>
<p>Tk&isin;T
</p>
<p>αkTk(x), (16.8)
</p>
<p>where T is a dictionary of basis functions, typically trees. For gradient
boosting and random forests, |T | is very large, and it is quite typical for the
final model to involve many thousands of trees. In the previous section we
argue that gradient boosting with shrinkage fits an L1 regularized monotone
path in this space of trees.
Friedman and Popescu (2003) propose a hybrid approach which breaks
</p>
<p>this process down into two stages:
</p>
<p>&bull; A finite dictionary TL = {T1(x), T2(x), . . . , TM (x)} of basis functions
is induced from the training data;
</p>
<p>&bull; A family of functions fλ(x) is built by fitting a lasso path in this
dictionary:
</p>
<p>α(λ) = argmin
α
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>L[yi, α0 +
</p>
<p>M&sum;
</p>
<p>m=1
</p>
<p>αmTm(xi)] + λ
</p>
<p>M&sum;
</p>
<p>m=1
</p>
<p>|αm|. (16.9)
</p>
<p>In its simplest form this model could be seen as a way of post-processing
boosting or random forests, taking for TL the collection of trees produced
by the gradient boosting or random forest algorithms. By fitting the lasso
path to these trees, we would typically use a much reduced set, which would
save in computations and storage for future predictions. In the next section
we describe modifications of this prescription that reduce the correlations in
the ensemble TL, and improve the performance of the lasso post processor.
As an initial illustration, we apply this procedure to a random forest
</p>
<p>ensemble grown on the spam data.
Figure 16.6 shows that a lasso post-processing offers modest improve-
</p>
<p>ment over the random forest (blue curve), and reduces the forest to about
40 trees, rather than the original 1000. The post-processed performance
matches that of gradient boosting. The orange curves represent a modified
version of random forests, designed to reduce the correlations between trees
even more. Here a random sub-sample (without replacement) of 5% of the
training sample is used to grow each tree, and the trees are restricted to be
shallow (about six terminal nodes). The post-processing offers more dra-
matic improvements here, and the training costs are reduced by a factor
of about 100. However, the performance of the post-processed model falls
somewhat short of the blue curves.
</p>
<p>16.3.1 Learning a Good Ensemble
</p>
<p>Not all ensembles TL will perform well with post-processing. In terms of
basis functions, we want a collection that covers the space well in places</p>
<p/>
</div>
<div class="page"><p/>
<p>618 16. Ensemble Learning
</p>
<p>0 100 200 300 400 500
</p>
<p>0
.0
</p>
<p>4
0
</p>
<p>.0
5
</p>
<p>0
.0
</p>
<p>6
0
</p>
<p>.0
7
</p>
<p>0
.0
</p>
<p>8
0
</p>
<p>.0
9
</p>
<p>Spam Data
</p>
<p>Number of Trees
</p>
<p>T
e
</p>
<p>s
t 
</p>
<p>E
rr
</p>
<p>o
r
</p>
<p>Random Forest
Random Forest (5%, 6)
Gradient Boost (5 node)
</p>
<p>FIGURE 16.6. Application of the lasso post-processing (16.9) to the spam data.
The horizontal blue line is the test error of a random forest fit to the spam data,
using 1000 trees grown to maximum depth (with m = 7; see Algorithm 15.1).
The jagged blue curve is the test error after post-processing the first 500 trees
using the lasso, as a function of the number of trees with nonzero coefficients.
The orange curve/line use a modified form of random forest, where a random
draw of 5% of the data are used to grow each tree, and the trees are forced to
be shallow (typically six terminal nodes). Here the post-processing offers much
greater improvement over the random forest that generated the ensemble.
</p>
<p>where they are needed, and are sufficiently different from each other for
the post-processor to be effective.
Friedman and Popescu (2003) gain insights from numerical quadrature
</p>
<p>and importance sampling. They view the unknown function as an integral
</p>
<p>f(x) =
</p>
<p>&int;
β(γ)b(x; γ)dγ, (16.10)
</p>
<p>where γ &isin; Γ indexes the basis functions b(x; γ). For example, if the basis
functions are trees, then γ indexes the splitting variables, the split-points
and the values in the terminal nodes. Numerical quadrature amounts to
finding a set of M evaluation points γm &isin; Γ and corresponding weights
αm so that fM (x) = α0 +
</p>
<p>&sum;M
m=1 αmb(x; γm) approximates f(x) well over
</p>
<p>the domain of x. Importance sampling amounts to sampling γ at random,
but giving more weight to relevant regions of the space Γ. Friedman and
Popescu (2003) suggest a measure of (lack of) relevance that uses the loss
function (16.9):</p>
<p/>
</div>
<div class="page"><p/>
<p>16.3 Learning Ensembles 619
</p>
<p>Q(γ) = min
c0,c1
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>L(yi, c0 + c1b(xi; γ)), (16.11)
</p>
<p>evaluated on the training data.
If a single basis function were to be selected (e.g., a tree), it would be
</p>
<p>the global minimizer γ&lowast; = argminγ&isin;Γ Q(γ). Introducing randomness in the
selection of γ would necessarily produce less optimal values with Q(γ) &ge;
Q(γ&lowast;). They propose a natural measure of the characteristic width σ of the
sampling scheme S,
</p>
<p>σ = ES [Q(γ)&minus;Q(γ&lowast;)]. (16.12)
</p>
<p>&bull; σ too narrow suggests too many of the b(x; γm) look alike, and similar
to b(x; γ&lowast;);
</p>
<p>&bull; σ too wide implies a large spread in the b(x; γm), but possibly con-
sisting of many irrelevant cases.
</p>
<p>Friedman and Popescu (2003) use sub-sampling as a mechanism for intro-
ducing randomness, leading to their ensemble-generation algorithm 16.2.
</p>
<p>Algorithm 16.2 ISLE Ensemble Generation.
</p>
<p>1. f0(x) = argminc
&sum;N
</p>
<p>i=1 L(yi, c)
</p>
<p>2. For m = 1 to M do
</p>
<p>(a) γm = argminγ
&sum;
</p>
<p>i&isin;Sm(η)
L(yi, fm&minus;1(xi) + b(xi; γ))
</p>
<p>(b) fm(x) = fm&minus;1(x) + νb(x; γm)
</p>
<p>3. TISLE = {b(x; γ1), b(x; γ2), . . . , b(x; γM )}.
</p>
<p>Sm(η) refers to a subsample of N &middot; η (η &isin; (0, 1]) of the training obser-
vations, typically without replacement. Their simulations suggest picking
η &le; 12 , and for large N picking η &sim; 1/
</p>
<p>&radic;
N . Reducing η increases the
</p>
<p>randomness, and hence the width σ. The parameter ν &isin; [0, 1] introduces
memory into the randomization process; the larger ν, the more the pro-
cedure avoids b(x; γ) similar to those found before. A number of familiar
randomization schemes are special cases of Algorithm 16.2:
</p>
<p>Bagging has η = 1, but samples with replacement, and has ν = 0. Fried-
man and Hall (2007) argue that sampling without replacement with
η = 1/2 is equivalent to sampling with replacement with η = 1, and
the former is much more efficient.</p>
<p/>
</div>
<div class="page"><p/>
<p>620 16. Ensemble Learning
</p>
<p>Random forest sampling is similar, with more randomness introduced by
the selection of the splitting variable. Reducing η &lt; 1/2 in algo-
rithm 16.2 has a similar effect to reducing m in random forests, but
does not suffer from the potential biases discussed in Section 15.4.2.
</p>
<p>Gradient boosting with shrinkage (10.41) uses η = 1, but typically does
not produce sufficient width σ.
</p>
<p>Stochastic gradient boosting (Friedman, 1999) follows the recipe exactly.
</p>
<p>The authors recommend values ν = 0.1 and η &le; 12 , and call their combined
procedure (ensemble generation and post processing) Importance sampled
learning ensemble (ISLE).
Figure 16.7 shows the performance of an ISLE on the spam data. It does
</p>
<p>0 500 1000 1500 2000 2500
</p>
<p>0
.0
</p>
<p>4
0
</p>
<p>0
.0
</p>
<p>4
5
</p>
<p>0
.0
</p>
<p>5
0
</p>
<p>0
.0
</p>
<p>5
5
</p>
<p>0
.0
</p>
<p>6
0
</p>
<p>Spam Data
</p>
<p>Number of Trees
</p>
<p>T
e
</p>
<p>s
t 
</p>
<p>E
rr
</p>
<p>o
r
</p>
<p>Gradient Boosting (5 Node)
Lasso Post&minus;processed
</p>
<p>FIGURE 16.7. Importance sampling learning ensemble (ISLE) fit to the spam
data. Here we used η = 1/2, ν = 0.05, and trees with five terminal nodes. The
lasso post-processed ensemble does not improve the prediction error in this case,
but it reduces the number of trees by a factor of five.
</p>
<p>not improve the predictive performance, but is able to produce a more
parsimonious model. Note that in practice the post-processing includes
the selection of the regularization parameter λ in (16.9), which would be</p>
<p/>
</div>
<div class="page"><p/>
<p>16.3 Learning Ensembles 621
</p>
<p>chosen by cross-validation. Here we simply demonstrate the effects of post-
processing by showing the entire path on the test data.
Figure 16.8 shows various ISLEs on a regression example. The generating
</p>
<p>0 500 1000 1500 2000 2500
</p>
<p>1
.0
</p>
<p>1
.5
</p>
<p>2
.0
</p>
<p>2
.5
</p>
<p>3
.0
</p>
<p>3
.5
</p>
<p>Number of Trees
</p>
<p>M
e
</p>
<p>a
n
</p>
<p> S
q
</p>
<p>u
a
</p>
<p>re
d
</p>
<p> E
rr
</p>
<p>o
r
</p>
<p>GBM (1, 0.01)
GBM (0.1, 0.01)
ISLE  GB
ISLE RF
Random Forest
</p>
<p>FIGURE 16.8. Demonstration of ensemble methods on a regression simulation
example. The notation GBM (0.1, 0.01) refers to a gradient boosted model, with
parameters (η, ν). We report mean-squared error from the true (known) function.
Note that the sub-sampled GBM model (green) outperforms the full GBM model
(orange). The lasso post-processed version achieves similar error. The random
forest is outperformed by its post-processed version, but both fall short of the
other models.
</p>
<p>function is
</p>
<p>f(X) = 10 &middot;
5&prod;
</p>
<p>j=1
</p>
<p>e&minus;2X
2
j +
</p>
<p>35&sum;
</p>
<p>j=6
</p>
<p>Xj , (16.13)
</p>
<p>where X &sim; U [0, 1]100 (the last 65 elements are noise variables). The re-
sponse Y = f(X) + ε where ε &sim; N(0, σ2); we chose σ = 1.3 resulting in a
signal-to-noise ratio of approximately 2. We used a training sample of size
1000, and estimated the mean squared error E(f̂(X)&minus;f(X))2 by averaging
over a test set of 500 samples. The sub-sampled GBM curve (light blue)
is an instance of stochastic gradient boosting (Friedman, 1999) discussed in
Section 10.12, and it outperforms gradient boosting on this example.</p>
<p/>
</div>
<div class="page"><p/>
<p>622 16. Ensemble Learning
</p>
<p>16.3.2 Rule Ensembles
</p>
<p>Here we describe a modification of the tree-ensemble method that focuses
on individual rules (Friedman and Popescu, 2003). We encountered rules
in Section 9.3 in the discussion of the PRIM method. The idea is to enlarge
an ensemble of trees by constructing a set of rules from each of the trees
in the collection.
</p>
<p>1 2
</p>
<p>3
</p>
<p>0
</p>
<p>4
</p>
<p>5 6
</p>
<p>X1 &lt; 2.1 X1 &ge; 2.1
</p>
<p>X3 &isin; {M,L}X3 &isin; {S}
</p>
<p>X7 &lt; 4.5 X7 &ge; 4.5
</p>
<p>FIGURE 16.9. A typical tree in an ensemble, from which rules can be derived.
</p>
<p>Figure 16.9 depicts a small tree, with numbered nodes. The following
rules can be derived from this tree:
</p>
<p>R1(X) = I(X1 &lt; 2.1)
R2(X) = I(X1 &ge; 2.1)
R3(X) = I(X1 &ge; 2.1) &middot; I(X3 &isin; {S})
R4(X) = I(X1 &ge; 2.1) &middot; I(X3 &isin; {M,L})
R5(X) = I(X1 &ge; 2.1) &middot; I(X3 &isin; {S}) &middot; I(X7 &lt; 4.5)
R6(X) = I(X1 &ge; 2.1) &middot; I(X3 &isin; {S}) &middot; I(X7 &ge; 4.5)
</p>
<p>(16.14)
</p>
<p>A linear expansion in rules 1, 4, 5 and 6 is equivalent to the tree itself
(Exercise 16.3); hence (16.14) is an over-complete basis for the tree.
For each tree Tm in an ensemble T , we can construct its mini-ensemble
</p>
<p>of rules T m
RULE
</p>
<p>, and then combine them all to form a larger ensemble
</p>
<p>TRULE =
M⋃
</p>
<p>m=1
</p>
<p>T mRULE. (16.15)
</p>
<p>This is then treated like any other ensemble, and post-processed via the
lasso or similar regularized procedure.
There are several advantages to this approach of deriving rules from the
</p>
<p>more complex trees:
</p>
<p>&bull; The space of models is enlarged, and can lead to improved perfor-
mance.</p>
<p/>
</div>
<div class="page"><p/>
<p>16.3 Learning Ensembles 623
</p>
<p>Rules Rules + Linear
</p>
<p>0
.9
</p>
<p>1
.0
</p>
<p>1
.1
</p>
<p>1
.2
</p>
<p>1
.3
</p>
<p>M
e
a
n
 S
</p>
<p>q
u
a
re
</p>
<p>d
 E
</p>
<p>rr
o
r
</p>
<p>FIGURE 16.10. Mean squared error for rule ensembles, using 20 realizations
of the simulation example (16.13).
</p>
<p>&bull; Rules are easier to interpret than trees, so there is the potential for
a simplified model.
</p>
<p>&bull; It is often natural to augment T RULE by including each variable Xj
separately as well, thus allowing the ensemble to model linear func-
tions well.
</p>
<p>Friedman and Popescu (2008) demonstrate the power of this procedure on a
number of illustrative examples, including the simulation example (16.13).
Figure 16.10 shows boxplots of the mean-squared error from the true model
for twenty realizations from this model. The models were all fit using the
Rulefit software, available on the ESL homepage3, which runs in an auto-
matic mode.
On the same training set as used in Figure 16.8, the rule based model
</p>
<p>achieved a mean-squared error of 1.06. Although slightly worse than the
best achieved in that figure, the results are not comparable because cross-
validation was used here to select the final model.
</p>
<p>Bibliographic Notes
</p>
<p>As noted in the introduction, many of the new methods in machine learning
have been dubbed &ldquo;ensemble&rdquo; methods. These include neural networks
boosting, bagging and random forests; Dietterich (2000a) gives a survey of
tree-based ensemble methods. Neural networks (Chapter 11) are perhaps
more deserving of the name, since they simultaneously learn the parameters
</p>
<p>3ESL homepage: www-stat.stanford.edu/ElemStatLearn</p>
<p/>
</div>
<div class="page"><p/>
<p>624 16. Ensemble Learning
</p>
<p>of the hidden units (basis functions), along with how to combine them.
Bishop (2006) discusses neural networks in some detail, along with the
Bayesian perspective (MacKay, 1992; Neal, 1996). Support vector machines
(Chapter 12) can also be regarded as an ensemble method; they perform
L2 regularized model fitting in high-dimensional feature spaces. Boosting
and lasso exploit sparsity through L1 regularization to overcome the high-
dimensionality, while SVMs rely on the &ldquo;kernel trick&rdquo; characteristic of L2
regularization.
C5.0 (Quinlan, 2004) is a commercial tree and rule generation package,
</p>
<p>with some goals in common with Rulefit.
There is a vast and varied literature often referred to as &ldquo;combining clas-
</p>
<p>sifiers&rdquo; which abounds in ad-hoc schemes for mixing methods of different
types to achieve better performance. For a principled approach, see Kittler
et al. (1998).
</p>
<p>Exercises
</p>
<p>Ex. 16.1 Describe exactly how to generate the block correlated data used
in the simulation in Section 16.2.3.
</p>
<p>Ex. 16.2 Let α(t) &isin; IRp be a piecewise-differentiable and continuous coef-
ficient profile, with α(0) = 0. The L1 arc-length of α from time 0 to t is
defined by
</p>
<p>Λ(t) =
</p>
<p>&int; t
</p>
<p>0
</p>
<p>|α̇(t)|1dt. (16.16)
</p>
<p>Show that Λ(t) &ge; |α(t)|1, with equality iff α(t) is monotone.
</p>
<p>Ex. 16.3 Show that fitting a linear regression model using rules 1, 4, 5 and
6 in equation (16.14) gives the same fit as the regression tree corresponding
to this tree. Show the same is true for classification, if a logistic regression
model is fit.
</p>
<p>Ex. 16.4 Program and run the simulation study described in Figure 16.2.</p>
<p/>
</div>
<div class="page"><p/>
<p>17
Undirected Graphical Models
</p>
<p>17.1 Introduction
</p>
<p>A graph consists of a set of vertices (nodes), along with a set of edges join-
ing some pairs of the vertices. In graphical models, each vertex represents
a random variable, and the graph gives a visual way of understanding the
joint distribution of the entire set of random variables. They can be use-
ful for either unsupervised or supervised learning. In an undirected graph,
the edges have no directional arrows. We restrict our discussion to undi-
rected graphical models, also known as Markov random fields or Markov
networks. In these graphs, the absence of an edge between two vertices has
a special meaning: the corresponding random variables are conditionally
independent, given the other variables.
Figure 17.1 shows an example of a graphical model for a flow-cytometry
</p>
<p>dataset with p = 11 proteins measured on N = 7466 cells, from Sachs
et al. (2005). Each vertex in the graph corresponds to the real-valued ex-
pression level of a protein. The network structure was estimated assuming
a multivariate Gaussian distribution, using the graphical lasso procedure
discussed later in this chapter.
Sparse graphs have a relatively small number of edges, and are convenient
</p>
<p>for interpretation. They are useful in a variety of domains, including ge-
nomics and proteomics, where they provide rough models of cell pathways.
Much work has been done in defining and understanding the structure of
graphical models; see the Bibliographic Notes for references.
</p>
<p> &copy; Springer Science+Business Media, LLC 2009 
</p>
<p>T. Hastie et al., The Elements of Statistical Learning, Second Edition, 625
</p>
<p>DOI: 10.1007/b94608_17,</p>
<p/>
</div>
<div class="page"><p/>
<p>626 17. Undirected Graphical Models
</p>
<p>Raf
</p>
<p>Mek
</p>
<p>Plcg
</p>
<p>PIP2
</p>
<p>PIP3
</p>
<p>Erk Akt
</p>
<p>PKA
</p>
<p>PKC
</p>
<p>P38
</p>
<p>Jnk
</p>
<p>FIGURE 17.1. Example of a sparse undirected graph, estimated from a flow
cytometry dataset, with p = 11 proteins measured on N = 7466 cells. The net-
work structure was estimated using the graphical lasso procedure discussed in this
chapter.
</p>
<p>As we will see, the edges in a graph are parametrized by values or po-
tentials that encode the strength of the conditional dependence between
the random variables at the corresponding vertices. The main challenges in
working with graphical models are model selection (choosing the structure
of the graph), estimation of the edge parameters from data, and compu-
tation of marginal vertex probabilities and expectations, from their joint
distribution. The last two tasks are sometimes called learning and inference
in the computer science literature.
We do not attempt a comprehensive treatment of this interesting area.
</p>
<p>Instead, we introduce some basic concepts, and then discuss a few sim-
ple methods for estimation of the parameters and structure of undirected
graphical models; methods that relate to the techniques already discussed
in this book. The estimation approaches that we present for continuous
and discrete-valued vertices are different, so we treat them separately. Sec-
tions 17.3.1 and 17.3.2 may be of particular interest, as they describe new,
regression-based procedures for estimating graphical models.
There is a large and active literature on directed graphical models or
</p>
<p>Bayesian networks; these are graphical models in which the edges have
directional arrows (but no directed cycles). Directed graphical models rep-
resent probability distributions that can be factored into products of condi-
tional distributions, and have the potential for causal interpretations. We
refer the reader to Wasserman (2004) for a brief overview of both undi-
rected and directed graphs; the next section follows closely his Chapter 18.</p>
<p/>
</div>
<div class="page"><p/>
<p>17.2 Markov Graphs and Their Properties 627
</p>
<p>X
</p>
<p>X
</p>
<p>X X
</p>
<p>Y
</p>
<p>Y
</p>
<p>Y
Y
</p>
<p>Z
</p>
<p>Z
</p>
<p>Z
</p>
<p>Z
</p>
<p>W
</p>
<p>W
</p>
<p>W
</p>
<p>(a) (b)
</p>
<p>(c) (d)
</p>
<p>FIGURE 17.2. Examples of undirected graphical models or Markov networks.
Each node or vertex represents a random variable, and the lack of an edge between
two nodes indicates conditional independence. For example, in graph (a), X and
Z are conditionally independent, given Y . In graph (b), Z is independent of each
of X, Y , and W .
</p>
<p>A longer list of useful references is given in the Bibliographic Notes on
page 645.
</p>
<p>17.2 Markov Graphs and Their Properties
</p>
<p>In this section we discuss the basic properties of graphs as models for the
joint distribution of a set of random variables. We defer discussion of (a)
parametrization and estimation of the edge parameters from data, and (b)
estimation of the topology of a graph, to later sections.
Figure 17.2 shows four examples of undirected graphs. A graph G consists
</p>
<p>of a pair (V,E), where V is a set of vertices and E the set of edges (defined
by pairs of vertices). Two vertices X and Y are called adjacent if there
is a edge joining them; this is denoted by X &sim; Y . A path X1, X2, . . . , Xn
is a set of vertices that are joined, that is Xi&minus;1 &sim; Xi for i = 2, . . . , n. A
complete graph is a graph with every pair of vertices joined by an edge.
A subgraph U &isin; V is a subset of vertices together with their edges. For
example, (X,Y, Z) in Figure 17.2(a) form a path but not a complete graph.
</p>
<p>Suppose that we have a graph G whose vertex set V represents a set of
random variables having joint distribution P . In a Markov graph G, the
absence of an edge implies that the corresponding random variables are
conditionally independent given the variables at the other vertices. This is
expressed with the following notation:</p>
<p/>
</div>
<div class="page"><p/>
<p>628 17. Undirected Graphical Models
</p>
<p>No edge joining X and Y &lArr;&rArr; X &perp; Y |rest (17.1)
</p>
<p>where &ldquo;rest&rdquo; refers to all of the other vertices in the graph. For example
in Figure 17.2(a) X &perp; Z|Y . These are known as the pairwise Markov
independencies of G.
</p>
<p>If A,B and C are subgraphs, then C is said to separate A and B if every
path between A and B intersects a node in C. For example, Y separates
X and Z in Figures 17.2(a) and (d), and Z separates Y and W in (d). In
Figure 17.2(b) Z is not connected to X,Y,W so we say that the two sets
are separated by the empty set. In Figure 17.2(c), C = {X,Z} separates
Y and W .
</p>
<p>Separators have the nice property that they break the graph into con-
ditionally independent pieces. Specifically, in a Markov graph G with sub-
graphs A,B and C,
</p>
<p>if C separates A and B then A &perp; B|C. (17.2)
</p>
<p>These are known as the global Markov properties of G. It turns out that the
pairwise and global Markov properties of a graph are equivalent (for graphs
with positive distributions). That is, the set of graphs with associated prob-
ability distributions that satisfy the pairwise Markov independencies and
global Markov assumptions are the same. This result is useful for inferring
global independence relations from simple pairwise properties. For example
in Figure 17.2(d) X &perp; Z|{Y,W} since it is a Markov graph and there is no
link joining X and Z. But Y also separates X from Z and W and hence by
the global Markov assumption we conclude that X &perp; Z|Y and X &perp; W |Y .
Similarly we have Y &perp; W |Z.
The global Markov property allows us to decompose graphs into smaller
</p>
<p>more manageable pieces and thus leads to essential simplifications in com-
putation and interpretation. For this purpose we separate the graph into
cliques. A clique is a complete subgraph&mdash; a set of vertices that are all
adjacent to one another; it is called maximal if it is a clique and no other
vertices can be added to it and still yield a clique. The maximal cliques for
the graphs of Figure 17.2 are
</p>
<p>(a) {X,Y }, {Y, Z},
</p>
<p>(b) {X,Y,W}, {Z},
</p>
<p>(c) {X,Y }, {Y, Z}, {Z,W}, {X,W}, and
</p>
<p>(d) {X,Y }, {Y, Z}, {Z,W}.
</p>
<p>Although the following applies to both continuous and discrete distri-
butions, much of the development has been for the latter. A probability
density function f over a Markov graph G can be can represented as</p>
<p/>
</div>
<div class="page"><p/>
<p>17.2 Markov Graphs and Their Properties 629
</p>
<p>f(x) =
1
</p>
<p>Z
</p>
<p>&prod;
</p>
<p>C&isin;C
</p>
<p>ψC(xC) (17.3)
</p>
<p>where C is the set of maximal cliques, and the positive functions ψC(&middot;) are
called clique potentials. These are not in general density functions1, but
rather are affinities that capture the dependence in XC by scoring certain
instances xC higher than others. The quantity
</p>
<p>Z =
&sum;
</p>
<p>x&isin;X
</p>
<p>&prod;
</p>
<p>C&isin;C
</p>
<p>ψC(xC) (17.4)
</p>
<p>is the normalizing constant, also known as the partition function. Alterna-
tively, the representation (17.3) implies a graph with independence prop-
erties defined by the cliques in the product. This result holds for Markov
networks G with positive distributions, and is known as the Hammersley-
Clifford theorem (Hammersley and Clifford, 1971; Clifford, 1990).
</p>
<p>Many of the methods for estimation and computation on graphs first de-
compose the graph into its maximal cliques. Relevant quantities are com-
puted in the individual cliques and then accumulated across the entire
graph. A prominent example is the join tree or junction tree algorithm for
computing marginal and low order probabilities from the joint distribution
on a graph. Details can be found in Pearl (1986), Lauritzen and Spiegel-
halter (1988), Pearl (1988), Shenoy and Shafer (1988), Jensen et al. (1990),
or Koller and Friedman (2007).
</p>
<p>X
</p>
<p>Y
</p>
<p>Z
</p>
<p>FIGURE 17.3. A complete graph does not uniquely specify the higher-order
dependence structure in the joint distribution of the variables.
</p>
<p>A graphical model does not always uniquely specify the higher-order
dependence structure of a joint probability distribution. Consider the com-
plete three-node graph in Figure 17.3. It could represent the dependence
structure of either of the following distributions:
</p>
<p>f (2)(x, y, z) = 1Zψ(x, y)ψ(x, z)ψ(y, z);
</p>
<p>f (3)(x, y, z) = 1Zψ(x, y, z).
(17.5)
</p>
<p>The first specifies only second order dependence (and can be represented
with fewer parameters). Graphical models for discrete data are a special
</p>
<p>1If the cliques are separated, then the potentials can be densities, but this is in general
</p>
<p>not the case.</p>
<p/>
</div>
<div class="page"><p/>
<p>630 17. Undirected Graphical Models
</p>
<p>case of loglinear models for multiway contingency tables (Bishop et al.,
1975, e.g.); in that language f (2) is referred to as the &ldquo;no second-order
interaction&rdquo; model.
For the remainder of this chapter we focus on pairwise Markov graphs
</p>
<p>(Koller and Friedman, 2007). Here there is a potential function for each
edge (pair of variables as in f (2) above), and at most second&ndash;order interac-
tions are represented. These are more parsimonious in terms of parameters,
easier to work with, and give the minimal complexity implied by the graph
structure. The models for both continuous and discrete data are functions
of only the pairwise marginal distributions of the variables represented in
the edge set.
</p>
<p>17.3 Undirected Graphical Models for Continuous
Variables
</p>
<p>Here we consider Markov networks where all the variables are continuous.
The Gaussian distribution is almost always used for such graphical models,
because of its convenient analytical properties. We assume that the observa-
tions have a multivariate Gaussian distribution with mean μ and covariance
matrix Σ. Since the Gaussian distribution represents at most second-order
relationships, it automatically encodes a pairwise Markov graph. The graph
in Figure 17.1 is an example of a Gaussian graphical model.
The Gaussian distribution has the property that all conditional distri-
</p>
<p>butions are also Gaussian. The inverse covariance matrix Σ&minus;1 contains
information about the partial covariances between the variables; that is,
the covariances between pairs i and j, conditioned on all other variables.
In particular, if the ijth component ofΘ = Σ&minus;1 is zero, then variables i and
j are conditionally independent, given the other variables (Exercise 17.3).
It is instructive to examine the conditional distribution of one variable
</p>
<p>versus the rest, where the role of Θ is explicit. Suppose we partition X =
(Z, Y ) where Z = (X1, . . . , Xp&minus;1) consists of the first p &minus; 1 variables and
Y = Xp is the last. Then we have the conditional distribution of Y give Z
(Mardia et al., 1979, e.g.)
</p>
<p>Y |Z = z &sim; N
(
μY + (z &minus; μZ)TΣ&minus;1ZZσZY , σY Y &minus; σTZY Σ&minus;1ZZσZY
</p>
<p>)
, (17.6)
</p>
<p>where we have partitioned Σ as
</p>
<p>Σ =
</p>
<p>(
ΣZZ σZY
σTZY σY Y
</p>
<p>)
. (17.7)
</p>
<p>The conditional mean in (17.6) has exactly the same form as the pop-
ulation multiple linear regression of Y on Z, with regression coefficient
β = Σ&minus;1ZZσZY [see (2.16) on page 19]. If we partition Θ in the same way,
since ΣΘ = I standard formulas for partitioned inverses give</p>
<p/>
</div>
<div class="page"><p/>
<p>17.3 Undirected Graphical Models for Continuous Variables 631
</p>
<p>θZY = &minus;θY Y &middot;Σ&minus;1ZZσZY , (17.8)
</p>
<p>where 1/θY Y = σY Y &minus; σTZY Σ&minus;1ZZσZY &gt; 0. Hence
</p>
<p>β = Σ&minus;1ZZσZY
= &minus;θZY /θY Y . (17.9)
</p>
<p>We have learned two things here:
</p>
<p>&bull; The dependence of Y on Z in (17.6) is in the mean term alone. Here
we see explicitly that zero elements in β and hence θZY mean that
the corresponding elements of Z are conditionally independent of Y ,
given the rest.
</p>
<p>&bull; We can learn about this dependence structure through multiple linear
regression.
</p>
<p>Thus Θ captures all the second-order information (both structural and
quantitative) needed to describe the conditional distribution of each node
given the rest, and is the so-called &ldquo;natural&rdquo; parameter for the Gaussian
graphical model2.
Another (different) kind of graphical model is the covariance graph or rel-
</p>
<p>evance network, in which vertices are connected by bidirectional edges if the
covariance (rather than the partial covariance) between the corresponding
variables is nonzero. These are popular in genomics, see especially Butte
et al. (2000). The negative log-likelihood from these models is not convex,
making the computations more challenging (Chaudhuri et al., 2007).
</p>
<p>17.3.1 Estimation of the Parameters when the Graph
Structure is Known
</p>
<p>Given some realizations of X, we would like to estimate the parameters
of an undirected graph that approximates their joint distribution. Suppose
first that the graph is complete (fully connected). We assume that we have
N multivariate normal realizations xi, i = 1, . . . , N with population mean
μ and covariance Σ. Let
</p>
<p>S =
1
</p>
<p>N
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>(xi &minus; x̄)(xi &minus; x̄)T (17.10)
</p>
<p>be the empirical covariance matrix, with x̄ the sample mean vector. Ignoring
constants, the log-likelihood of the data can be written as
</p>
<p>2The distribution arising from a Gaussian graphical model is a Wishart distribution.
This is a member of the exponential family, with canonical or &ldquo;natural&rdquo; parameter
Θ = Σ&minus;1. Indeed, the partially maximized log-likelihood (17.11) is (up to constants)
</p>
<p>the Wishart log-likelihood.</p>
<p/>
</div>
<div class="page"><p/>
<p>632 17. Undirected Graphical Models
</p>
<p>ℓ(Θ) = log detΘ&minus; trace(SΘ). (17.11)
In (17.11) we have partially maximized with respect to the mean parameter
μ. The quantity &minus;ℓ(Θ) is a convex function of Θ. It is easy to show that
the maximum likelihood estimate of Σ is simply S.
</p>
<p>Now to make the graph more useful (especially in high-dimensional set-
tings) let&rsquo;s assume that some of the edges are missing; for example, the
edge between PIP3 and Erk is one of several missing in Figure 17.1. As we
have seen, for the Gaussian distribution this implies that the correspond-
ing entries of Θ = Σ&minus;1 are zero. Hence we now would like to maximize
(17.11) under the constraints that some pre-defined subset of the parame-
ters are zero. This is an equality-constrained convex optimization problem,
and a number of methods have been proposed for solving it, in particular
the iterative proportional fitting procedure (Speed and Kiiveri, 1986). This
and other methods are summarized for example in Whittaker (1990) and
Lauritzen (1996). These methods exploit the simplifications that arise from
decomposing the graph into its maximal cliques, as described in the previ-
ous section. Here we outline a simple alternate approach, that exploits the
sparsity in a different way. The fruits of this approach will become apparent
later when we discuss the problem of estimation of the graph structure.
The idea is based on linear regression, as inspired by (17.6) and (17.9).
</p>
<p>In particular, suppose that we want to estimate the edge parameters θij for
the vertices that are joined to a given vertex i, restricting those that are not
joined to be zero. Then it would seem that the linear regression of the node
i values on the other relevant vertices might provide a reasonable estimate.
But this ignores the dependence structure among the predictors in this
regression. It turns out that if instead we use our current (model-based)
estimate of the cross-product matrix of the predictors when we perform
our regressions, this gives the correct solutions and solves the constrained
maximum-likelihood problem exactly. We now give details.
To constrain the log-likelihood (17.11), we add Lagrange constants for
</p>
<p>all missing edges
</p>
<p>ℓC(Θ) = log detΘ&minus; trace(SΘ)&minus;
&sum;
</p>
<p>(j,k) �&isin;E
</p>
<p>γjkθjk. (17.12)
</p>
<p>The gradient equation for maximizing (17.12) can be written as
</p>
<p>Θ&minus;1 &minus; S&minus; Γ = 0, (17.13)
using the fact that the derivative of log detΘ equals Θ&minus;1 (Boyd and Van-
denberghe, 2004, for example, page 641). Γ is a matrix of Lagrange param-
eters with nonzero values for all pairs with edges absent.
We will show how we can use regression to solve for Θ and its inverse
</p>
<p>W = Θ&minus;1 one row and column at a time. For simplicity let&rsquo;s focus on the
last row and column. Then the upper right block of equation (17.13) can
be written as</p>
<p/>
</div>
<div class="page"><p/>
<p>17.3 Undirected Graphical Models for Continuous Variables 633
</p>
<p>w12 &minus; s12 &minus; γ12 = 0. (17.14)
Here we have partitioned the matrices into two parts as in (17.7): part 1
being the first p&minus;1 rows and columns, and part 2 the pth row and column.
With W and its inverse Θ partitioned in a similar fashion, we have
</p>
<p>(
W11 w12
wT12 w22
</p>
<p>)(
Θ11 θ12
θT12 θ22
</p>
<p>)
=
</p>
<p>(
I 0
0T 1
</p>
<p>)
. (17.15)
</p>
<p>This implies
</p>
<p>w12 = &minus;W11θ12/θ22 (17.16)
= W11β (17.17)
</p>
<p>where β = &minus;θ12/θ22 as in (17.9). Now substituting (17.17) into (17.14)
gives
</p>
<p>W11β &minus; s12 &minus; γ12 = 0. (17.18)
These can be interpreted as the p &minus; 1 estimating equations for the con-
</p>
<p>strained regression of Xp on the other predictors, except that the observed
mean cross-products matrix S11 is replaced by W11, the current estimated
covariance matrix from the model.
Now we can solve (17.18) by simple subset regression. Suppose there are
</p>
<p>p&minus;q nonzero elements in γ12&mdash;i.e., p&minus;q edges constrained to be zero. These
p&minus; q rows carry no information and can be removed. Furthermore we can
reduce β to β&lowast; by removing its p &minus; q zero elements, yielding the reduced
q &times; q system of equations
</p>
<p>W&lowast;11β
&lowast; &minus; s&lowast;12 = 0, (17.19)
</p>
<p>with solution β̂&lowast; = W&lowast;11
&minus;1s&lowast;12. This is padded with p&minus; q zeros to give β̂.
</p>
<p>Although it appears from (17.16) that we only recover the elements θ12
up to a scale factor 1/θ22, it is easy to show that
</p>
<p>1
</p>
<p>θ22
= w22 &minus; wT12β (17.20)
</p>
<p>(using partitioned inverse formulas). Also w22 = s22, since the diagonal of
Γ in (17.13) is zero.
This leads to the simple iterative procedure given in Algorithm 17.1 for
</p>
<p>estimating both Ŵ and its inverse Θ̂, subject to the constraints of the
missing edges.
Note that this algorithm makes conceptual sense. The graph estimation
</p>
<p>problem is not p separate regression problems, but rather p coupled prob-
lems. The use of the common W in step (b), in place of the observed
cross-products matrix, couples the problems together in the appropriate
fashion. Surprisingly, we were not able to find this procedure in the lit-
erature. However it is related to the covariance selection procedures of</p>
<p/>
</div>
<div class="page"><p/>
<p>634 17. Undirected Graphical Models
</p>
<p>Algorithm 17.1 A Modified Regression Algorithm for Estimation of an
Undirected Gaussian Graphical Model with Known Structure.
</p>
<p>1. Initialize W = S.
</p>
<p>2. Repeat for j = 1, 2, . . . , p, 1, . . . until convergence:
</p>
<p>(a) Partition the matrix W into part 1: all but the jth row and
column, and part 2: the jth row and column.
</p>
<p>(b) Solve W&lowast;11β
&lowast; &minus; s&lowast;12 = 0 for the unconstrained edge parameters
</p>
<p>β&lowast;, using the reduced system of equations as in (17.19). Obtain
</p>
<p>β̂ by padding β̂&lowast; with zeros in the appropriate positions.
</p>
<p>(c) Update w12 = W11β̂
</p>
<p>3. In the final cycle (for each j) solve for θ̂12 = &minus;β̂ &middot; θ̂22, with 1/θ̂22 =
s22 &minus; wT12β̂.
</p>
<p>X1
</p>
<p>X2X3
</p>
<p>X4
</p>
<p>S =
</p>
<p>⎛
⎜⎜⎝
</p>
<p>10 1 5 4
1 10 2 6
5 2 10 3
4 6 3 10
</p>
<p>⎞
⎟⎟⎠
</p>
<p>FIGURE 17.4. A simple graph for illustration, along with the empirical covari-
ance matrix.
</p>
<p>Dempster (1972), and is similar in flavor to the iterative conditional fitting
procedure for covariance graphs, proposed by Chaudhuri et al. (2007).
Here is a little example, borrowed from Whittaker (1990). Suppose that
</p>
<p>our model is as depicted in Figure 17.4, along with its empirical covariance
matrix S. We apply algorithm (17.1) to this problem; for example, in the
modified regression for variable 1 in step (b), variable 3 is left out. The
procedure quickly converged to the solutions:
</p>
<p>Σ̂=
</p>
<p>⎛
⎜⎝
10.00 1.00 1.31 4.00
1.00 10.00 2.00 0.87
1.31 2.00 10.00 3.00
4.00 0.87 3.00 10.00
</p>
<p>⎞
⎟⎠, Σ̂&minus;1=
</p>
<p>⎛
⎜⎝
</p>
<p>0.12 &minus;0.01 0.00 &minus;0.05
&minus;0.01 0.11 &minus;0.02 0.00
0.00 &minus;0.02 0.11 &minus;0.03
&minus;0.05 0.00 &minus;0.03 0.13
</p>
<p>⎞
⎟⎠ .
</p>
<p>Note the zeroes in Σ̂&minus;1, corresponding to the missing edges (1,3) and (2,4).
</p>
<p>Note also that the corresponding elements in Σ̂ are the only elements dif-
ferent from S. The estimation of Σ̂ is an example of what is sometimes
called the positive definite &ldquo;completion&rdquo; of S.</p>
<p/>
</div>
<div class="page"><p/>
<p>17.3 Undirected Graphical Models for Continuous Variables 635
</p>
<p>17.3.2 Estimation of the Graph Structure
</p>
<p>In most cases we do not know which edges to omit from our graph, and
so would like to try to discover this from the data itself. In recent years a
number of authors have proposed the use of L1 (lasso) regularization for
this purpose.
Meinshausen and Bühlmann (2006) take a simple approach to the prob-
</p>
<p>lem: rather than trying to fully estimate Σ or Θ = Σ&minus;1, they only estimate
which components of θij are nonzero. To do this, they fit a lasso regression
using each variable as the response and the others as predictors. The com-
ponent θij is then estimated to be nonzero if either the estimated coefficient
of variable i on j is nonzero, or the estimated coefficient of variable j on
i is nonzero (alternatively they use an and rule). They show that asymp-
totically this procedure consistently estimates the set of nonzero elements
of Θ.
</p>
<p>We can take a more systematic approach with the lasso penalty, following
the development of the previous section. Consider maximizing the penalized
log-likelihood
</p>
<p>log detΘ&minus; trace(SΘ)&minus; λ||Θ||1, (17.21)
</p>
<p>where ||Θ||1 is the L1 norm&mdash;the sum of the absolute values of the elements
of Σ&minus;1, and we have ignored constants. The negative of this penalized
likelihood is a convex function of Θ.
</p>
<p>It turns out that one can adapt the lasso to give the exact maximizer of
the penalized log-likelihood. In particular, we simply replace the modified
regression step (b) in Algorithm 17.1 by a modified lasso step. Here are the
details.
The analog of the gradient equation (17.13) is now
</p>
<p>Θ&minus;1 &minus; S&minus; λ &middot; Sign(Θ) = 0. (17.22)
</p>
<p>Here we use sub-gradient notation, with Sign(θjk) = sign(θjk) if θjk �= 0,
else Sign(θjk) &isin; [&minus;1, 1] if θjk = 0. Continuing the development in the
previous section, we reach the analog of (17.18)
</p>
<p>W11β &minus; s12 + λ &middot; Sign(β) = 0 (17.23)
</p>
<p>(recall that β and θ12 have opposite signs). We will now see that this system
is exactly equivalent to the estimating equations for a lasso regression.
Consider the usual regression setup with outcome variables y and pre-
</p>
<p>dictor matrix Z. There the lasso minimizes
</p>
<p>1
2 (y &minus; Zβ)T (y &minus; Zβ) + λ &middot; ||β||1 (17.24)
</p>
<p>[see (3.52) on page 68; here we have added a factor 12 for convenience]. The
gradient of this expression is</p>
<p/>
</div>
<div class="page"><p/>
<p>636 17. Undirected Graphical Models
</p>
<p>Algorithm 17.2 Graphical Lasso.
</p>
<p>1. Initialize W = S + λI. The diagonal of W remains unchanged in
what follows.
</p>
<p>2. Repeat for j = 1, 2, . . . p, 1, 2, . . . p, . . . until convergence:
</p>
<p>(a) Partition the matrix W into part 1: all but the jth row and
column, and part 2: the jth row and column.
</p>
<p>(b) Solve the estimating equations W11β &minus; s12 + λ &middot; Sign(β) = 0
using the cyclical coordinate-descent algorithm (17.26) for the
modified lasso.
</p>
<p>(c) Update w12 = W11β̂
</p>
<p>3. In the final cycle (for each j) solve for θ̂12 = &minus;β̂ &middot; θ̂22, with 1/θ̂22 =
w22 &minus; wT12β̂.
</p>
<p>ZTZβ &minus; ZTy + λ &middot; Sign(β) = 0 (17.25)
</p>
<p>So up to a factor 1/N , ZTy is the analog of s12, and we replace Z
TZ by
</p>
<p>W11, the estimated cross-product matrix from our current model.
The resulting procedure is called the graphical lasso, proposed by Fried-
</p>
<p>man et al. (2008b) building on the work of Banerjee et al. (2008). It is
summarized in Algorithm 17.2.
Friedman et al. (2008b) use the pathwise coordinate descent method
</p>
<p>(Section 3.8.6) to solve the modified lasso problem at each stage. Here are
the details of pathwise coordinate descent for the graphical lasso algorithm.
Letting V = W11, the update has the form
</p>
<p>β̂j &larr; S
(
s12j &minus;
</p>
<p>&sum;
</p>
<p>k �=j
</p>
<p>Vkj β̂k, λ
)
/Vjj (17.26)
</p>
<p>for j = 1, 2, . . . , p &minus; 1, 1, 2, . . . , p &minus; 1, . . ., where S is the soft-threshold
operator:
</p>
<p>S(x, t) = sign(x)(|x| &minus; t)+. (17.27)
The procedure cycles through the predictors until convergence.
It is easy to show that the diagonal elements wjj of the solution matrix
</p>
<p>W are simply sjj + λ, and these are fixed in step 1 of Algorithm 17.2
3.
</p>
<p>The graphical lasso algorithm is extremely fast, and can solve a moder-
ately sparse problem with 1000 nodes in less than a minute. It is easy to
modify the algorithm to have edge-specific penalty parameters λjk; since
</p>
<p>3An alternative formulation of the problem (17.21) can be posed, where we don&rsquo;t
penalize the diagonal of Θ. Then the diagonal elements wjj of the solution matrix are
</p>
<p>sjj , and the rest of the algorithm is unchanged.</p>
<p/>
</div>
<div class="page"><p/>
<p>17.3 Undirected Graphical Models for Continuous Variables 637
</p>
<p>λjk = &infin; will force θ̂jk to be zero, this algorithm subsumes Algorithm 17.1.
By casting the sparse inverse-covariance problem as a series of regressions,
one can also quickly compute and examine the solution paths as a function
of the penalty parameter λ. More details can be found in Friedman et al.
(2008b).
</p>
<p>Raf
</p>
<p>Mek
</p>
<p>Plcg
</p>
<p>PIP2
</p>
<p>PIP3
</p>
<p>Erk Akt
</p>
<p>PKA
</p>
<p>PKC
</p>
<p>P38
</p>
<p>Jnk
</p>
<p>Raf
</p>
<p>Mek
</p>
<p>Plcg
</p>
<p>PIP2
</p>
<p>PIP3
</p>
<p>Erk Akt
</p>
<p>PKA
</p>
<p>PKC
</p>
<p>P38
</p>
<p>Jnk
</p>
<p>Raf
</p>
<p>Mek
</p>
<p>Plcg
</p>
<p>PIP2
</p>
<p>PIP3
</p>
<p>Erk Akt
</p>
<p>PKA
</p>
<p>PKC
</p>
<p>P38
</p>
<p>Jnk
</p>
<p>Raf
</p>
<p>Mek
</p>
<p>Plcg
</p>
<p>PIP2
</p>
<p>PIP3
</p>
<p>Erk Akt
</p>
<p>PKA
</p>
<p>PKC
</p>
<p>P38
</p>
<p>Jnk
</p>
<p>λ = 0λ = 7
</p>
<p>λ = 27λ = 36
</p>
<p>FIGURE 17.5. Four different graphical-lasso solutions for the flow-cytometry
data.
</p>
<p>Figure 17.1 shows the result of applying the graphical lasso to the flow-
cytometry dataset. Here the lasso penalty parameter λ was set at 14. In
practice it is informative to examine the different sets of graphs that are
obtained as λ is varied. Figure 17.5 shows four different solutions. The
graph becomes more sparse as the penalty parameter is increased.
Finally note that the values at some of the nodes in a graphical model can
</p>
<p>be unobserved; that is, missing or hidden. If only some values are missing
at a node, the EM algorithm can be used to impute the missing values</p>
<p/>
</div>
<div class="page"><p/>
<p>638 17. Undirected Graphical Models
</p>
<p>(Exercise 17.9). However, sometimes the entire node is hidden or latent.
In the Gaussian model, if a node has all missing values, due to linearity
one can simply average over the missing nodes to yield another Gaussian
model over the observed nodes. Hence the inclusion of hidden nodes does
not enrich the resulting model for the observed nodes; in fact, it imposes
additional structure on its covariance matrix. However in the discrete model
(described next) the inherent nonlinearities make hidden units a powerful
way of expanding the model.
</p>
<p>17.4 Undirected Graphical Models for Discrete
Variables
</p>
<p>Undirected Markov networks with all discrete variables are popular, and
in particular pairwise Markov networks with binary variables being the
most common. They are sometimes called Ising models in the statistical
mechanics literature, and Boltzmann machines in the machine learning lit-
erature, where the vertices are referred to as &ldquo;nodes&rdquo; or &ldquo;units&rdquo; and are
binary-valued.
In addition, the values at each node can be observed (&ldquo;visible&rdquo;) or un-
</p>
<p>observed (&ldquo;hidden&rdquo;). The nodes are often organized in layers, similar to a
neural network. Boltzmann machines are useful both for unsupervised and
supervised learning, especially for structured input data such as images,
but have been hampered by computational difficulties. Figure 17.6 shows
a restricted Boltzmann machine (discussed later), in which some variables
are hidden, and only some pairs of nodes are connected. We first consider
the simpler case in which all p nodes are visible with edge pairs (j, k) enu-
merated in E.
</p>
<p>Denoting the binary valued variable at node j by Xj , the Ising model
for their joint probabilities is given by
</p>
<p>p(X,Θ) = exp
[ &sum;
</p>
<p>(j,k)&isin;E
</p>
<p>θjkXjXk &minus; Φ(Θ)
]
for X &isin; X , (17.28)
</p>
<p>with X = {0, 1}p. As with the Gaussian model of the previous section,
only pairwise interactions are modeled. The Ising model was developed in
statistical mechanics, and is now used more generally to model the joint
effects of pairwise interactions. Φ(Θ) is the log of the partition function,
and is defined by
</p>
<p>Φ(Θ) = log
&sum;
</p>
<p>x&isin;X
</p>
<p>[
exp
</p>
<p>( &sum;
</p>
<p>(j,k)&isin;E
</p>
<p>θjkxjxk
</p>
<p>)]
. (17.29)
</p>
<p>The partition function ensures that the probabilities add to one over the
sample space. The terms θjkXjXk represent a particular parametrization</p>
<p/>
</div>
<div class="page"><p/>
<p>17.4 Undirected Graphical Models for Discrete Variables 639
</p>
<p>of the (log) potential functions (17.5), and for technical reasons requires
a constant node X0 &equiv; 1 to be included (Exercise 17.10), with &ldquo;edges&rdquo; to
all the other nodes. In the statistics literature, this model is equivalent
to a first-order-interaction Poisson log-linear model for multiway tables of
counts (Bishop et al., 1975; McCullagh and Nelder, 1989; Agresti, 2002).
The Ising model implies a logistic form for each node conditional on the
</p>
<p>others (exercise 17.11):
</p>
<p>Pr(Xj = 1|X&minus;j = x&minus;j) =
1
</p>
<p>1 + exp(&minus;θj0 &minus;
&sum;
</p>
<p>(j,k)&isin;E θjkxk)
, (17.30)
</p>
<p>where X&minus;j denotes all of the nodes except j. Hence the parameter θjk
measures the dependence of Xj on Xk, conditional on the other nodes.
</p>
<p>17.4.1 Estimation of the Parameters when the Graph
Structure is Known
</p>
<p>Given some data from this model, how can we estimate the parameters?
Suppose we have observations xi = (xi1, xi2, . . . , xip)&isin; {0, 1}p, i = 1, . . . , N .
The log-likelihood is
</p>
<p>ℓ(Θ) =
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>log PrΘ(Xi = xi)
</p>
<p>=
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>⎡
⎣ &sum;
</p>
<p>(j,k)&isin;E
</p>
<p>θjkxijxik &minus; Φ(Θ)
</p>
<p>⎤
⎦ (17.31)
</p>
<p>The gradient of the log-likelihood is
</p>
<p>&part;ℓ(Θ)
</p>
<p>&part;θjk
=
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>xijxik &minus;N
&part;Φ(Θ)
</p>
<p>&part;θjk
(17.32)
</p>
<p>and
</p>
<p>&part;Φ(Θ)
</p>
<p>&part;θjk
=
</p>
<p>&sum;
</p>
<p>x&isin;X
</p>
<p>xjxk &middot; p(x,Θ)
</p>
<p>= EΘ(XjXk) (17.33)
</p>
<p>Setting the gradient to zero gives
</p>
<p>Ê(XjXk)&minus; EΘ(XjXk) = 0 (17.34)
</p>
<p>where we have defined</p>
<p/>
</div>
<div class="page"><p/>
<p>640 17. Undirected Graphical Models
</p>
<p>Ê(XjXk) =
1
</p>
<p>N
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>xijxik, (17.35)
</p>
<p>the expectation taken with respect to the empirical distribution of the data.
Looking at (17.34), we see that the maximum likelihood estimates simply
match the estimated inner products between the nodes to their observed
inner products. This is a standard form for the score (gradient) equation
for exponential family models, in which sufficient statistics are set equal to
their expectations under the model.
To find the maximum likelihood estimates, we can use gradient search
</p>
<p>or Newton methods. However the computation of EΘ(XjXk) involves enu-
meration of p(X,Θ) over 2p&minus;2 of the |X | = 2p possible values of X, and is
not generally feasible for large p (e.g., larger than about 30). For smaller
p, a number of standard statistical approaches are available:
</p>
<p>Poisson log-linear modeling, where we treat the problem as a large regres-
sion problem (Exercise 17.12). The response vector y is the vector of
2p counts in each of the cells of the multiway tabulation of the data4.
The predictor matrix Z has 2p rows and up to 1+p+p2 columns that
characterize each of the cells, although this number depends on the
sparsity of the graph. The computational cost is essentially that of a
regression problem of this size, which is O(p42p) and is manageable
for p &lt; 20. The Newton updates are typically computed by iteratively
reweighted least squares, and the number of steps is usually in the
single digits. See Agresti (2002) and McCullagh and Nelder (1989) for
details. Standard software (such as the R package glm) can be used
to fit this model.
</p>
<p>Gradient descent requires at most O(p22p&minus;2) computations to compute
the gradient, but may require many more gradient steps than the
second&ndash;order Newton methods. Nevertheless, it can handle slightly
larger problems with p &le; 30. These computations can be reduced
by exploiting the special clique structure in sparse graphs, using the
junction-tree algorithm. Details are not given here.
</p>
<p>Iterative proportional fitting (IPF) performs cyclical coordinate descent on
the gradient equations (17.34). At each step a parameter is updated
so that its gradient equation is exactly zero. This is done in a cyclical
fashion until all the gradients are zero. One complete cycle costs the
same as a gradient evaluation, but may be more efficient. Jirouśek and
Přeučil (1995) implement an efficient version of IPF, using junction
trees.
</p>
<p>4Each of the cell counts is treated as an independent Poisson variable. We get the
multinomial model corresponding to (17.28) by conditioning on the total count N (which
</p>
<p>is also Poisson under this framework).</p>
<p/>
</div>
<div class="page"><p/>
<p>17.4 Undirected Graphical Models for Discrete Variables 641
</p>
<p>When p is large (&gt; 30) other approaches have been used to approximate
the gradient.
</p>
<p>&bull; The mean field approximation (Peterson and Anderson, 1987) esti-
mates EΘ(XjXk) by EΘ(Xj)EΘ(Xj), and replaces the input vari-
ables by their means, leading to a set of nonlinear equations for the
parameters θjk.
</p>
<p>&bull; To obtain near-exact solutions, Gibbs sampling (Section 8.6) is used
to approximate EΘ(XjXk) by successively sampling from the esti-
mated model probabilities PrΘ(Xj |X&minus;j) (see e.g. Ripley (1996)).
</p>
<p>We have not discussed decomposable models, for which the maximum
likelihood estimates can be found in closed form without any iteration
whatsoever. These models arise, for example, in trees: special graphs with
tree-structured topology. When computational tractability is a concern,
trees represent a useful class of models and they sidestep the computational
concerns raised in this section. For details, see for example Chapter 12 of
Whittaker (1990).
</p>
<p>17.4.2 Hidden Nodes
</p>
<p>We can increase the complexity of a discrete Markov network by including
latent or hidden nodes. Suppose that a subset of the variables XH are
unobserved or &ldquo;hidden&rdquo;, and the remainder XV are observed or &ldquo;visible.&rdquo;
Then the log-likelihood of the observed data is
</p>
<p>ℓ(Θ) =
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>log[PrΘ(XV = xiV)]
</p>
<p>=
N&sum;
</p>
<p>i=1
</p>
<p>[
log
</p>
<p>&sum;
</p>
<p>xH&isin;XH
</p>
<p>exp
&sum;
</p>
<p>(j,k)&isin;E
</p>
<p>(θjkxijxik &minus; Φ(Θ))
]
. (17.36)
</p>
<p>The sum over xH means that we are summing over all possible {0, 1} values
for the hidden units. The gradient works out to be
</p>
<p>dℓ(Θ)
</p>
<p>dθjk
= ÊVEΘ(XjXk|XV)&minus; EΘ(XjXk) (17.37)
</p>
<p>The first term is an empirical average of XjXk if both are visible; if one
or both are hidden, they are first imputed given the visible data, and then
averaged over the hidden variables. The second term is the unconditional
expectation of XjXk.
The inner expectation in the first term can be evaluated using basic rules
</p>
<p>of conditional expectation and properties of Bernoulli random variables. In
detail, for observation i</p>
<p/>
</div>
<div class="page"><p/>
<p>642 17. Undirected Graphical Models
</p>
<p>EΘ(XjXk|XV = xiV) =
{ xijxik if j, k &isin; V
</p>
<p>xijPrΘ(Xk = 1|XV = xiV) if j &isin; V, k &isin; H
PrΘ(Xj = 1, Xk = 1|XV = xiV) if j, k &isin; H.
</p>
<p>(17.38)
Now two separate runs of Gibbs sampling are required; the first to estimate
EΘ(XjXk) by sampling from the model as above, and the second to esti-
mate EΘ(XjXk|XV = xiV). In this latter run, the visible units are fixed
(&ldquo;clamped&rdquo;) at their observed values and only the hidden variables are
sampled. Gibbs sampling must be done for each observation in the training
set, at each stage of the gradient search. As a result this procedure can be
very slow, even for moderate-sized models. In Section 17.4.4 we consider
further model restrictions to make these computations manageable.
</p>
<p>17.4.3 Estimation of the Graph Structure
</p>
<p>The use of a lasso penalty with binary pairwise Markov networks has been
suggested by Lee et al. (2007) and Wainwright et al. (2007). The first au-
thors investigate a conjugate gradient procedure for exact maximization of
a penalized log-likelihood. The bottleneck is the computation of EΘ(XjXk)
in the gradient; exact computation via the junction tree algorithm is man-
ageable for sparse graphs but becomes unwieldy for dense graphs.
The second authors propose an approximate solution, analogous to the
</p>
<p>Meinshausen and Bühlmann (2006) approach for the Gaussian graphical
model. They fit an L1-penalized logistic regression model to each node as
a function of the other nodes, and then symmetrize the edge parameter
estimates in some fashion. For example if θ̃jk is the estimate of the j-k
edge parameter from the logistic model for outcome node j, the &ldquo;min&rdquo;
symmetrization sets θ̂jk to either θ̃jk or θ̃kj , whichever is smallest in abso-
lute value. The &ldquo;max&rdquo; criterion is defined similarly. They show that under
certain conditions either approximation estimates the nonzero edges cor-
rectly as the sample size goes to infinity. Hoefling and Tibshirani (2008)
extend the graphical lasso to discrete Markov networks, obtaining a pro-
cedure which is somewhat faster than conjugate gradients, but still must
deal with computation of EΘ(XjXk). They also compare the exact and
approximate solutions in an extensive simulation study and find the &ldquo;min&rdquo;
or &ldquo;max&rdquo; approximations are only slightly less accurate than the exact pro-
cedure, both for estimating the nonzero edges and for estimating the actual
values of the edge parameters, and are much faster. Furthermore, they can
handle denser graphs because they never need to compute the quantities
EΘ(XjXk).
Finally, we point out a key difference between the Gaussian and binary
</p>
<p>models. In the Gaussian case, bothΣ and its inverse will often be of interest,
and the graphical lasso procedure delivers estimates for both of these quan-
tities. However, the approximation of Meinshausen and Bühlmann (2006)
for Gaussian graphical models, analogous to the Wainwright et al. (2007)</p>
<p/>
</div>
<div class="page"><p/>
<p>17.4 Undirected Graphical Models for Discrete Variables 643
</p>
<p>Xj
</p>
<p>Xk
</p>
<p>Xℓ
</p>
<p>Visible V1 Visible V2
</p>
<p>Hidden H
</p>
<p>θjk
</p>
<p>FIGURE 17.6. A restricted Boltzmann machine (RBM) in which there are no
connections between nodes in the same layer. The visible units are subdivided to
allow the RBM to model the joint density of feature V1 and their labels V2.
</p>
<p>approximation for the binary case, only yields an estimate of Σ&minus;1. In con-
trast, in the Markov model for binary data, Θ is the object of interest, and
its inverse is not of interest. The approximate method of Wainwright et al.
(2007) estimates Θ efficiently and hence is an attractive solution for the
binary problem.
</p>
<p>17.4.4 Restricted Boltzmann Machines
</p>
<p>In this section we consider a particular architecture for graphical models
inspired by neural networks, where the units are organized in layers. A
restricted Boltzmann machine (RBM) consists of one layer of visible units
and one layer of hidden units with no connections within each layer. It is
much simpler to compute the conditional expectations (as in (17.37) and
(17.38)) if the connections between hidden units are removed 5. Figure 17.6
shows an example; the visible layer is divided into input variables V1 and
output variables V2, and there is a hidden layer H. We denote such a
network by
</p>
<p>V1 &harr; H &harr; V2. (17.39)
For example, V1 could be the binary pixels of an image of a handwritten
digit, and V2 could have 10 units, one for each of the observed class labels
0-9.
The restricted form of this model simplifies the Gibbs sampling for es-
</p>
<p>timating the expectations in (17.37), since the variables in each layer are
independent of one another, given the variables in the other layers. Hence
they can be sampled together, using the conditional probabilities given by
expression (17.30).
The resulting model is less general than a Boltzmann machine, but is still
</p>
<p>useful; for example it can learn to extract interesting features from images.
</p>
<p>5We thank Geoffrey Hinton for assistance in the preparation of the material on RBMs.</p>
<p/>
</div>
<div class="page"><p/>
<p>644 17. Undirected Graphical Models
</p>
<p>By alternately sampling the variables in each layer of the RBM shown
in Figure 17.6, it is possible to generate samples from the joint density
model. If the V1 part of the visible layer is clamped at a particular feature
vector during the alternating sampling, it is possible to sample from the
distribution over labels given V1. Alternatively classification of test items
can also be achieved by comparing the unnormalized joint densities of each
label category with the observed features. We do not need to compute the
partition function as it is the same for all of these combinations.
As noted the restricted Boltzmann machine has the same generic form
</p>
<p>as a single hidden layer neural network (Section 11.3). The edges in the
latter model are directed, the hidden units are usually real-valued, and the
fitting criterion is different. The neural network minimizes the error (cross-
entropy) between the targets and their model predictions, conditional on
the input features. In contrast, the restricted Boltzmann machine maxi-
mizes the log-likelihood for the joint distribution of all visible units&mdash;that
is, the features and targets. It can extract information from the input fea-
tures that is useful for predicting the labels, but, unlike supervised learning
methods, it may also use some of its hidden units to model structure in the
feature vectors that is not immediately relevant for predicting the labels.
These features may turn out to be useful, however, when combined with
features derived from other hidden layers.
Unfortunately, Gibbs sampling in a restricted Boltzmann machine can
</p>
<p>be very slow, as it can take a long time to reach stationarity. As the net-
work weights get larger, the chain mixes more slowly and we need to run
more steps to get the unconditional estimates. Hinton (2002) noticed em-
pirically that learning still works well if we estimate the second expectation
in (17.37) by starting the Markov chain at the data and only running for a
few steps (instead of to convergence). He calls this contrastive divergence:
we sample H given V1,V2, then V1,V2 given H and finally H given V1,V2
again. The idea is that when the parameters are far from the solution, it
may be wasteful to iterate the Gibbs sampler to stationarity, as just a single
iteration will reveal a good direction for moving the estimates.
We now give an example to illustrate the use of an RBM. Using con-
</p>
<p>trastive divergence, it is possible to train an RBM to recognize hand-written
digits from the MNIST dataset (LeCun et al., 1998). With 2000 hidden
units, 784 visible units for representing binary pixel intensities and one
10-way multinomial visible unit for representing labels, the RBM achieves
an error rate of 1.9% on the test set. This is a little higher than the 1.4%
achieved by a support vector machine and comparable to the error rate
achieved by a neural network trained with backpropagation. The error rate
of the RBM, however, can be reduced to 1.25% by replacing the 784 pixel
intensities by 500 features that are produced from the images without using
any label information. First, an RBM with 784 visible units and 500 hidden
units is trained, using contrastive divergence, to model the set of images.
Then the hidden states of the first RBM are used as data for training a</p>
<p/>
</div>
<div class="page"><p/>
<p>Exercises 645
</p>
<p>FIGURE 17.7. Example of a restricted Boltzmann machine for handwritten
digit classification. The network is depicted in the schematic on the left. Displayed
on the right are some difficult test images that the model classifies correctly.
</p>
<p>second RBM that has 500 visible units and 500 hidden units. Finally, the
hidden states of the second RBM are used as the features for training an
RBM with 2000 hidden units as a joint density model. The details and
justification for learning features in this greedy, layer-by-layer way are de-
scribed in Hinton et al. (2006). Figure 17.7 gives a representation of the
composite model that is learned in this way and also shows some examples
of the types of distortion that it can cope with.
</p>
<p>Bibliographic Notes
</p>
<p>Much work has been done in defining and understanding the structure of
graphical models. Comprehensive treatments of graphical models can be
found in Whittaker (1990), Lauritzen (1996), Cox and Wermuth (1996),
Edwards (2000), Pearl (2000), Anderson (2003), Jordan (2004), and Koller
and Friedman (2007). Wasserman (2004) gives a brief introduction, and
Chapter 8 of Bishop (2006) gives a more detailed overview. Boltzmann
machines were proposed in Ackley et al. (1985). Ripley (1996) has a detailed
chapter on topics in graphical models that relate to machine learning. We
found this particularly useful for its discussion of Boltzmann machines.
</p>
<p>Exercises
</p>
<p>Ex. 17.1 For the Markov graph of Figure 17.8, list all of the implied condi-
tional independence relations and find the maximal cliques.</p>
<p/>
</div>
<div class="page"><p/>
<p>646 17. Undirected Graphical Models
</p>
<p>X1
</p>
<p>X2 X3
</p>
<p>X4
</p>
<p>X5
</p>
<p>X6
</p>
<p>FIGURE 17.8.
</p>
<p>Ex. 17.2 Consider random variables X1, X2, X3, X4. In each of the following
cases draw a graph that has the given independence relations:
</p>
<p>(a) X1 &perp; X3|X2 and X2 &perp; X4|X3.
(b) X1 &perp; X4|X2, X3 and X2 &perp; X4|X1, X3.
(c) X1 &perp; X4|X2, X3, X1 &perp; X3|X2, X4 and X3 &perp; X4|X1, X2.
</p>
<p>Ex. 17.3 Let Σ be the covariance matrix of a set of p variables X. Consider
the partial covariance matrix Σa.b = Σaa &minus;ΣabΣ&minus;1bb Σba between the two
subsets of variables Xa = (X1, X2) consisting of the first two, and Xb
the rest. This is the covariance matrix between these two variables, after
linear adjustment for all the rest. In the Gaussian distribution, this is the
covariance matrix of the conditional distribution of Xa|Xb. The partial
correlation coefficient ρjk|rest between the pair Xa conditional on the rest
</p>
<p>Xb, is simply computed from this partial covariance. Define Θ = Σ
&minus;1.
</p>
<p>1. Show that Σa.b = Θ
&minus;1
aa .
</p>
<p>2. Show that if any off-diagonal element of Θ is zero, then the partial
correlation coefficient between the corresponding variables is zero.
</p>
<p>3. Show that if we treatΘ as if it were a covariance matrix, and compute
the corresponding &ldquo;correlation&rdquo; matrix
</p>
<p>R = diag(Θ)&minus;1/2 &middot;Θ &middot; diag(Θ)&minus;1/2, (17.40)
then rjk = &minus;ρjk|rest
</p>
<p>Ex. 17.4 Denote by
f(X1|X2, X3, . . . , Xp)
</p>
<p>the conditional density of X1 given X2, . . . , Xp. If
</p>
<p>f(X1|X2, X3, . . . , Xp) = f(X1|X3, . . . , Xp),
show that X1 &perp; X2|X3, . . . , Xp.</p>
<p/>
</div>
<div class="page"><p/>
<p>Exercises 647
</p>
<p>Ex. 17.5 Consider the setup in Section 17.3.1 with no missing edges. Show
that
</p>
<p>S11β &minus; s12 = 0
are the estimating equations for the multiple regression coefficients of the
last variable on the rest.
</p>
<p>Ex. 17.6 Recovery of Θ̂ = Σ̂&minus;1 from Algorithm 17.1. Use expression (17.16)
to derive the standard partitioned inverse expressions
</p>
<p>θ12 = &minus;W&minus;111 w12θ22 (17.41)
θ22 = 1/(w22 &minus; wT12W&minus;111 w12). (17.42)
</p>
<p>Since β̂ = W&minus;111 w12, show that θ̂22 = 1/(w22 &minus; wT12β̂) and θ̂12 = &minus;β̂θ̂22.
Thus θ̂12 is a simply rescaling of β̂ by &minus;θ̂22.
</p>
<p>Ex. 17.7 Write a program to implement the modified regression procedure in
Algorithm 17.1 for fitting the Gaussian graphical model with pre-specified
edges missing. Test it on the flow cytometry data from the book website,
using the graph of Figure 17.1.
</p>
<p>Ex. 17.8
</p>
<p>(a) Write a program to fit the lasso using the coordinate descent procedure
(17.26). Compare its results to those from the lars program or some
other convex optimizer, to check that it is working correctly.
</p>
<p>(b) Using the program from (a), write code to implement the graphical
lasso (Algorithm 17.2). Apply it to the flow cytometry data from the
book website. Vary the regularization parameter and examine the
resulting networks.
</p>
<p>Ex. 17.9 Suppose that we have a Gaussian graphical model in which some
or all of the data at some vertices are missing.
</p>
<p>(a) Consider the EM algorithm for a dataset of N i.i.d. multivariate ob-
servations xi &isin; IRp with mean μ and covariance matrix Σ. For each
sample i, let oi and mi index the predictors that are observed and
missing, respectively. Show that in the E step, the observations are
imputed from the current estimates of μ and Σ:
</p>
<p>x̂i,mi = E(xi,mi |xi,oi , θ) = μ̂mi + Σ̂mi,oiΣ̂&minus;1oi,oi(xi,oi &minus; μ̂oi)
(17.43)
</p>
<p>while in the M step, μ and Σ are re-estimated from the empirical
mean and (modified) covariance of the imputed data:
</p>
<p>μ̂j =
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>x̂ij/N</p>
<p/>
</div>
<div class="page"><p/>
<p>648 17. Undirected Graphical Models
</p>
<p>Σ̂jj&prime; =
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>[(x̂ij &minus; μ̂j)(x̂ij&prime; &minus; μ̂j&prime;) + ci,jj&prime; ]/N (17.44)
</p>
<p>where ci,jj&prime; = Σ̂jj&prime; if j, j
&prime; &isin; mi and zero otherwise. Explain the reason
</p>
<p>for the correction term ci,jj&prime; (Little and Rubin, 2002).
</p>
<p>(b) Implement the EM algorithm for the Gaussian graphical model using
the modified regression procedure from Exercise 17.7 for the M-step.
</p>
<p>(c) For the flow cytometry data on the book website, set the data for the
last protein Jnk in the first 1000 observations to missing, fit the model
of Figure 17.1, and compare the predicted values to the actual values
for Jnk. Compare the results to those obtained from a regression of
Jnk on the other vertices with edges to Jnk in Figure 17.1, using only
the non-missing data.
</p>
<p>Ex. 17.10 Using a simple binary graphical model with just two variables,
show why it is essential to include a constant node X0 &equiv; 1 in the model.
</p>
<p>Ex. 17.11 Show that the Ising model (17.28) for the joint probabilities in
a discrete graphical model implies that the conditional distributions have
the logistic form (17.30).
</p>
<p>Ex. 17.12 Consider a Poisson regression problem with p binary variables
xij , j = 1, . . . , p and response variable yi which measures the number of
observations with predictor xi &isin; {0, 1}p. The design is balanced, in that all
n = 2p possible combinations are measured. We assume a log-linear model
for the Poisson mean in each cell
</p>
<p>log μ(X) = θ00 +
&sum;
</p>
<p>(j,k)&isin;E
</p>
<p>xijxikθjk, (17.45)
</p>
<p>using the same notation as in Section 17.4.1 (including the constant variable
xi0 = 1&forall;i). We assume the response is distributed as
</p>
<p>Pr(Y = y|X = x) = e
&minus;μ(x)μ(x)y
</p>
<p>y!
. (17.46)
</p>
<p>Write down the conditional log-likelihood for the observed responses yi,
and compute the gradient.
</p>
<p>(a) Show that the gradient equation for θ00 computes the partition func-
tion (17.29).
</p>
<p>(b) Show that the gradient equations for the remainder of the parameters
are equivalent to the gradient (17.34).</p>
<p/>
</div>
<div class="page"><p/>
<p>18
High-Dimensional Problems: p ≫ N
</p>
<p>18.1 When p is Much Bigger than N
</p>
<p>In this chapter we discuss prediction problems in which the number of
features p is much larger than the number of observations N , often written
p ≫ N . Such problems have become of increasing importance, especially in
genomics and other areas of computational biology. We will see that high
variance and overfitting are a major concern in this setting. As a result,
simple, highly regularized approaches often become the methods of choice.
The first part of the chapter focuses on prediction in both the classification
and regression settings, while the second part discusses the more basic
problem of feature selection and assessment.
To get us started, Figure 18.1 summarizes a small simulation study that
</p>
<p>demonstrates the &ldquo;less fitting is better&rdquo; principle that applies when p ≫ N .
For each of N = 100 samples, we generated p standard Gaussian features
X with pairwise correlation 0.2. The outcome Y was generated according
to a linear model
</p>
<p>Y =
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>Xjβj + σε (18.1)
</p>
<p>where ε was generated from a standard Gaussian distribution. For each
dataset, the set of coefficients βj were also generated from a standard Gaus-
sian distribution. We investigated three cases: p = 20, 100, and 1000. The
standard deviation σ was chosen in each case so that the signal-to-noise
ratio Var[E(Y |X)]/σ2 equaled 2. As a result, the number of significant uni-
</p>
<p> &copy; Springer Science+Business Media, LLC 2009 
</p>
<p>T. Hastie et al., The Elements of Statistical Learning, Second Edition, 649
</p>
<p>DOI: 10.1007/b94608_18,</p>
<p/>
</div>
<div class="page"><p/>
<p>650 18. High-Dimensional Problems: p ≫ N
</p>
<p>1
.0
</p>
<p>1
.5
</p>
<p>2
.0
</p>
<p>2
.5
</p>
<p>3
.0
</p>
<p>20 9 2
</p>
<p>20 features
</p>
<p>T
e
s
t 
E
</p>
<p>rr
o
r
</p>
<p>1
.0
</p>
<p>1
.5
</p>
<p>2
.0
</p>
<p>2
.5
</p>
<p>3
.0
</p>
<p>99 35 7
</p>
<p>100 features
</p>
<p>1
.0
</p>
<p>1
.5
</p>
<p>2
.0
</p>
<p>2
.5
</p>
<p>3
.0
</p>
<p>99 87 43
</p>
<p>1000 features
</p>
<p>Effective Degrees of Freedom
</p>
<p>FIGURE 18.1. Test-error results for simulation experiments. Shown are box-
plots of the relative test errors over 100 simulations, for three different values
of p, the number of features. The relative error is the test error divided by the
Bayes error, σ2. From left to right, results are shown for ridge regression with
three different values of the regularization parameter λ: 0.001, 100 and 1000. The
(average) effective degrees of freedom in the fit is indicated below each plot.
</p>
<p>variate regression coefficients1 was 9, 33 and 331, respectively, averaged
over the 100 simulation runs. The p = 1000 case is designed to mimic the
kind of data that we might see in a high-dimensional genomic or proteomic
dataset, for example.
We fit a ridge regression to the data, with three different values for the
</p>
<p>regularization parameter λ: 0.001, 100, and 1000. When λ = 0.001, this
is nearly the same as least squares regression, with a little regularization
just to ensure that the problem is non-singular when p &gt; N . Figure 18.1
shows boxplots of the relative test error achieved by the different estimators
in each scenario. The corresponding average degrees of freedom used in
each ridge-regression fit is indicated (computed using formula (3.50) on
page 682). The degrees of freedom is a more interpretable parameter than
λ. We see that ridge regression with λ = 0.001 (20 df) wins when p = 20;
λ = 100 (35 df) wins when p = 100, and λ = 1000 (43 df) wins when
p = 1000.
Here is an explanation for these results. When p = 20, we fit all the way
</p>
<p>and we can identify as many of the significant coefficients as possible with
</p>
<p>1We call a regression coefficient significant if |β̂j/ŝej | &ge; 2, where β̂j is the estimated
(univariate) coefficient and ŝej is its estimated standard error.
</p>
<p>2For a fixed value of the regularization parameter λ, the degrees of freedom depends
on the observed predictor values in each simulation. Hence we compute the average
degrees of freedom over simulations.</p>
<p/>
</div>
<div class="page"><p/>
<p>18.2 Nearest Shrunken Centroids 651
</p>
<p>low bias. When p = 100, we can identify some non-zero coefficients using
moderate shrinkage. Finally, when p = 1000, even though there are many
nonzero coefficients, we don&rsquo;t have a hope for finding them and we need
to shrink all the way down. As evidence of this, let tj = β̂j/ŝej , where β̂j
is the ridge regression estimate and ŝej its estimated standard error. Then
using the optimal ridge parameter in each of the three cases, the median
value of |tj | was 2.0, 0.6 and 0.2, and the average number of |tj | values
exceeding 2 was equal to 9.8, 1.2 and 0.0.
Ridge regression with λ = 0.001 successfully exploits the correlation in
</p>
<p>the features when p &lt; N , but cannot do so when p ≫ N . In the latter case
there is not enough information in the relatively small number of samples
to efficiently estimate the high-dimensional covariance matrix. In that case,
more regularization leads to superior prediction performance.
Thus it is not surprising that the analysis of high-dimensional data re-
</p>
<p>quires either modification of procedures designed for the N &gt; p scenario, or
entirely new procedures. In this chapter we discuss examples of both kinds
of approaches for high dimensional classification and regression; these meth-
ods tend to regularize quite heavily, using scientific contextual knowledge
to suggest the appropriate form for this regularization. The chapter ends
with a discussion of feature selection and multiple testing.
</p>
<p>18.2 Diagonal Linear Discriminant Analysis and
Nearest Shrunken Centroids
</p>
<p>Gene expression arrays are an important new technology in biology, and
are discussed in Chapters 1 and 14. The data in our next example form
a matrix of 2308 genes (columns) and 63 samples (rows), from a set of
microarray experiments. Each expression value is a log-ratio log(R/G). R
is the amount of gene-specific RNA in the target sample that hybridizes
to a particular (gene-specific) spot on the microarray, and G is the corre-
sponding amount of RNA from a reference sample. The samples arose from
small, round blue-cell tumors (SRBCT) found in children, and are classified
into four major types: BL (Burkitt lymphoma), EWS (Ewing&rsquo;s sarcoma),
NB (neuroblastoma), and RMS (rhabdomyosarcoma). There is an addi-
tional test data set of 20 observations. We will not go into the scientific
background here.
Since p ≫ N , we cannot fit a full linear discriminant analysis (LDA) to
</p>
<p>the data; some sort of regularization is needed. The method we describe
here is similar to the methods of Section 4.3.1, but with important modifi-
cations that achieve feature selection. The simplest form of regularization
assumes that the features are independent within each class, that is, the
within-class covariance matrix is diagonal. Despite the fact that features
will rarely be independent within a class, when p ≫ N we don&rsquo;t have</p>
<p/>
</div>
<div class="page"><p/>
<p>652 18. High-Dimensional Problems: p ≫ N
</p>
<p>enough data to estimate their dependencies. The assumption of indepen-
dence greatly reduces the number of parameters in the model and often
results in an effective and interpretable classifier.
Thus we consider the diagonal-covariance LDA rule for classifying the
</p>
<p>classes. The discriminant score [see (4.12) on page 110] for class k is
</p>
<p>δk(x
&lowast;) = &minus;
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>(x&lowast;j &minus; x̄kj)2
s2j
</p>
<p>+ 2 log πk. (18.2)
</p>
<p>Here x&lowast; = (x&lowast;1, x
&lowast;
2, . . . , x
</p>
<p>&lowast;
p)
</p>
<p>T is a vector of expression values for a test ob-
servation, sj is the pooled within-class standard deviation of the jth gene,
and x̄kj =
</p>
<p>&sum;
i&isin;Ck
</p>
<p>xij/Nk is the mean of the Nk values for gene j in class
</p>
<p>k, with Ck being the index set for class k. We call x̃k = (x̄k1, x̄k2, . . . x̄kp)
T
</p>
<p>the centroid of class k. The first part of (18.2) is simply the (negative)
standardized squared distance of x&lowast; to the kth centroid. The second part
is a correction based on the class prior probability πk, where
</p>
<p>&sum;K
k=1 πk = 1.
</p>
<p>The classification rule is then
</p>
<p>C(x&lowast;) = ℓ if δℓ(x
&lowast;) = maxk δk(x
</p>
<p>&lowast;). (18.3)
</p>
<p>We see that the diagonal LDA classifier is equivalent to a nearest centroid
classifier after appropriate standardization. It is also a special case of the
naive-Bayes classifier, as described in Section 6.6.3. It assumes that the
features in each class have independent Gaussian distributions with the
same variance.
The diagonal LDA classifier is often effective in high dimensional set-
</p>
<p>tings. It is also called the &ldquo;independence rule&rdquo; in Bickel and Levina (2004),
who demonstrate theoretically that it will often outperform standard lin-
ear discriminant analysis in high-dimensional problems. Here the diagonal
LDA classifier yielded five misclassification errors for the 20 test samples.
One drawback of the diagonal LDA classifier is that it uses all of the fea-
tures (genes), and hence is not convenient for interpretation. With further
regularization we can do better&mdash;both in terms of test error and inter-
pretability.
We would like to regularize in a way that automatically drops out fea-
</p>
<p>tures that are not contributing to the class predictions. We can do this
by shrinking the classwise mean toward the overall mean, for each feature
separately. The result is a regularized version of the nearest centroid clas-
sifier, or equivalently a regularized version of the diagonal-covariance form
of LDA. We call the procedure nearest shrunken centroids (NSC).
The shrinkage procedure is defined as follows. Let
</p>
<p>dkj =
x̄kj &minus; x̄j
</p>
<p>mk(sj + s0)
, (18.4)
</p>
<p>where x̄j is the overall mean for gene j, m
2
k = 1/Nk &minus; 1/N and s0 is a
</p>
<p>small positive constant, typically chosen to be the median of the sj values.</p>
<p/>
</div>
<div class="page"><p/>
<p>18.2 Nearest Shrunken Centroids 653
</p>
<p>(0,0)
</p>
<p>∆
</p>
<p>FIGURE 18.2. Soft thresholding function sign(x)(|x|&minus;∆)+ is shown in orange,
along with the 45◦ line in red.
</p>
<p>This constant guards against large dkj values that arise from expression
values near zero. With constant within-class variance σ2, the variance of
the contrast x̄kj &minus; x̄j in the numerator is m2kσ2, and hence the form of the
standardization in the denominator. We shrink the dkj toward zero using
soft thresholding
</p>
<p>d&prime;kj = sign(dkj)(|dkj | &minus;∆)+; (18.5)
see Figure 18.2. Here ∆ is a parameter to be determined; we used 10-fold
cross-validation in the example (see the top panel of Figure 18.4). Each dkj
is reduced by an amount ∆ in absolute value, and is set to zero if its value
is less than zero. The soft-thresholding function is shown in Figure 18.2;
the same thresholding is applied to wavelet coefficients in Section 5.9. An
alternative is to use hard thresholding
</p>
<p>d&prime;kj = dkj &middot; I(|dkj | &ge; ∆); (18.6)
</p>
<p>we prefer soft-thresholding, as it is a smoother operation and typically
works better. The shrunken versions of x̄kj are then obtained by reversing
the transformation in (18.4):
</p>
<p>x̄&prime;kj = x̄j +mk(sj + s0)d
&prime;
kj . (18.7)
</p>
<p>We then use the shrunken centroids x̄&prime;kj in place of the original x̄kj in the
discriminant score (18.2). The estimator (18.7) can also be viewed as a
lasso-style estimator for the class means (Exercise 18.2).
Notice that only the genes that have a nonzero d&prime;kj for at least one of the
</p>
<p>classes play a role in the classification rule, and hence the vast majority
of genes can often be discarded. In this example, all but 43 genes were
discarded, leaving a small interpretable set of genes that characterize each
class. Figure 18.3 represents the genes in a heatmap.
Figure 18.4 (top panel) demonstrates the effectiveness of the shrinkage.
</p>
<p>With no shrinkage we make 5/20 errors on the test data, and several errors</p>
<p/>
</div>
<div class="page"><p/>
<p>654 18. High-Dimensional Problems: p ≫ N
</p>
<p>on the training and CV data. The shrunken centroids achieve zero test er-
rors for a fairly broad band of values for ∆. The bottom panel of Figure 18.4
shows the four centroids for the SRBCT data (gray), relative to the overall
centroid. The blue bars are shrunken versions of these centroids, obtained
by soft-thresholding the gray bars, using ∆ = 4.3. The discriminant scores
(18.2) can be used to construct class probability estimates:
</p>
<p>p̂k(x
&lowast;) =
</p>
<p>e
1
2 δk(x
</p>
<p>&lowast;)
</p>
<p>&sum;K
ℓ=1 e
</p>
<p>1
2 δℓ(x
</p>
<p>&lowast;)
. (18.8)
</p>
<p>These can be used to rate the classifications, or to decide not to classify a
particular sample at all.
Note that other forms of feature selection can be used in this setting,
</p>
<p>including hard thresholding. Fan and Fan (2008) show theoretically the
importance of carrying out some kind of feature selection with diagonal
linear discriminant analysis in high-dimensional problems.
</p>
<p>18.3 Linear Classifiers with Quadratic
Regularization
</p>
<p>Ramaswamy et al. (2001) present a more difficult microarray classification
problem, involving a training set of 144 patients with 14 different types of
cancer, and a test set of 54 patients. Gene expression measurements were
available for 16, 063 genes.
</p>
<p>Table 18.1 shows the prediction results from eight different classification
methods. The data from each patient was first standardized to have mean
0 and variance 1; this seems to improve prediction accuracy overall this
example, suggesting that the &ldquo;shape&rdquo; of each gene-expression profile is
important, rather than the absolute expression levels. In each case, the
</p>
<p>BL EWS NB RMS
</p>
<p>FIGURE 18.3. Heat-map of the chosen 43 genes. Within each of the horizontal
partitions, we have ordered the genes by hierarchical clustering, and similarly
for the samples within each vertical partition. Yellow represents over- and blue
under-expression.</p>
<p/>
</div>
<div class="page"><p/>
<p>18.3 Linear Classifiers with Quadratic Regularization 655
</p>
<p>0 2 4 6
</p>
<p>M
is
</p>
<p>c
la
</p>
<p>s
s
if
ic
</p>
<p>a
ti
o
n
 E
</p>
<p>rr
o
r
</p>
<p>2308 2059 1223 598 284 159 81 43 23 15 10 5 1
</p>
<p>Number of Genes
</p>
<p>0
.0
</p>
<p>0
.2
</p>
<p>0
.4
</p>
<p>0
.6
</p>
<p>0
.8
</p>
<p>Training
10&minus;fold CV
Test
</p>
<p>Amount of Shrinkage ∆
</p>
<p>&minus;1.0 &minus;0.5 0.0 0.5 1.0
</p>
<p>BL
</p>
<p>0
5
0
0
</p>
<p>1
0
0
0
</p>
<p>1
5
0
0
</p>
<p>2
0
0
0
</p>
<p>&minus;1.0 &minus;0.5 0.0 0.5 1.0
</p>
<p>EWS
</p>
<p>&minus;1.0 &minus;0.5 0.0 0.5 1.0
</p>
<p>NB
</p>
<p>&minus;1.0 &minus;0.5 0.0 0.5 1.0
</p>
<p>RMS
</p>
<p>G
e
n
e
</p>
<p>Centroids: Average Expression Centered at Overall Centroid
</p>
<p>FIGURE 18.4. (Top): Error curves for the SRBCT data. Shown are the train-
ing, 10-fold cross-validation, and test misclassification errors as the threshold
parameter ∆ is varied. The value ∆ = 4.34 is chosen by CV, resulting in a sub-
set of 43 selected genes. (Bottom): Four centroids profiles dkj for the SRBCT
data (gray), relative to the overall centroid. Each centroid has 2308 components,
and we see considerable noise. The blue bars are shrunken versions d&prime;kj of these
centroids, obtained by soft-thresholding the gray bars, using ∆ = 4.3.</p>
<p/>
</div>
<div class="page"><p/>
<p>656 18. High-Dimensional Problems: p ≫ N
</p>
<p>TABLE 18.1. Prediction results for microarray data with 14 cancer classes.
Method 1 is described in Section 18.2. Methods 2, 3 and 6 are discussed in Sec-
tion 18.3, while 4, 7 and 8 are discussed in Section 18.4. Method 5 is described in
Section 13.3. The elastic-net penalized multinomial does the best on the test data,
but the standard error of each test-error estimate is about 3, so such comparisons
are inconclusive.
</p>
<p>Methods CV errors (SE) Test errors Number of
Out of 144 Out of 54 Genes Used
</p>
<p>1. Nearest shrunken centroids 35 (5.0) 17 6,520
2. L2-penalized discriminant 25 (4.1) 12 16,063
</p>
<p>analysis
3. Support vector classifier 26 (4.2) 14 16,063
4. Lasso regression (one vs all) 30.7 (1.8) 12.5 1,429
5. k-nearest neighbors 41 (4.6) 26 16,063
6. L2-penalized multinomial 26 (4.2) 15 16,063
7. L1-penalized multinomial 17 (2.8) 13 269
8. Elastic-net penalized 22 (3.7) 11.8 384
</p>
<p>multinomial
</p>
<p>regularization parameter has been chosen to minimize the cross-validation
error, and the test error at that value of the parameter is shown. When
more than one value of the regularization parameter yields the minimal
cross-validation error, the average test error at these values is reported.
RDA (regularized discriminant analysis), regularized multinomial logistic
</p>
<p>regression, and the support vector machine are more complex methods that
try to exploit multivariate information in the data. We describe each in
turn, as well as a variety of regularization methods, including both L1 and
L2 and some in between.
</p>
<p>18.3.1 Regularized Discriminant Analysis
</p>
<p>Regularized discriminant analysis (RDA) is described in Section 4.3.1. Lin-
ear discriminant analysis involves the inversion of a p&times;p within-covariance
matrix. When p ≫ N , this matrix can be huge, has rank at most N &lt; p,
and hence is singular. RDA overcomes the singularity issues by regulariz-
ing the within-covariance estimate Σ̂. Here we use a version of RDA that
shrinks Σ̂ towards its diagonal:
</p>
<p>Σ̂(γ) = γΣ̂+ (1&minus; γ)diag(Σ̂), with γ &isin; [0, 1]. (18.9)
</p>
<p>Note that γ = 0 corresponds to diagonal LDA, which is the &ldquo;no shrinkage&rdquo;
version of nearest shrunken centroids. The form of shrinkage in (18.9) is</p>
<p/>
</div>
<div class="page"><p/>
<p>18.3 Linear Classifiers with Quadratic Regularization 657
</p>
<p>much like ridge regression (Section 3.4.1), which shrinks the total covariance
matrix of the features towards a diagonal (scalar) matrix. In fact, viewing
linear discriminant analysis as linear regression with optimal scoring of the
categorical response (see (12.57) in Section 12.6), the equivalence becomes
more precise.
The computational burden of inverting this large p&times;pmatrix is overcome
</p>
<p>using the methods discussed in Section 18.3.5. The value of γ was chosen
by cross-validation in line 2 of Table 18.1; all values of γ &isin; (0.002, 0.550)
gave the same CV and test error. Further development of RDA, including
shrinkage of the centroids in addition to the covariance matrix, can be
found in Guo et al. (2006).
</p>
<p>18.3.2 Logistic Regression with Quadratic Regularization
</p>
<p>Logistic regression (Section 4.4) can be modified in a similar way, to deal
with the p ≫ N case. With K classes, we use a symmetric version of the
multiclass logistic model (4.17) on page 119:
</p>
<p>Pr(G = k|X = x) = exp(βk0 + x
Tβk)&sum;K
</p>
<p>ℓ=1 exp(βℓ0 + x
Tβℓ)
</p>
<p>. (18.10)
</p>
<p>This has K coefficient vectors of log-odds parameters β1, β2, . . . , βK . We
regularize the fitting by maximizing the penalized log-likelihood
</p>
<p>max
{β0k,βk}K1
</p>
<p>[
N&sum;
</p>
<p>i=1
</p>
<p>log Pr(gi|xi)&minus;
λ
</p>
<p>2
</p>
<p>K&sum;
</p>
<p>k=1
</p>
<p>||βk||22
</p>
<p>]
. (18.11)
</p>
<p>This regularization automatically resolves the redundancy in the paramet-
rization, and forces
</p>
<p>&sum;K
k=1 β̂kj = 0, j = 1, . . . , p (Exercise 18.3). Note that
</p>
<p>the constant terms βk0 are not regularized (and so one should be set to
zero). The resulting optimization problem is convex, and can be solved by
a Newton algorithm or other numerical techniques. Details are given in Zhu
and Hastie (2004). Friedman et al. (2010) provide software for computing
the regularization path for the two- and multiclass logistic regression mod-
els. Table 18.1, line 6 reports the results for the multiclass logistic regres-
sion model, referred to there as &ldquo;multinomial&rdquo;. It can be shown (Rosset
et al., 2004a) that for separable data, as λ &rarr; 0, the regularized (two-
class) logistic regression estimate (renormalized) converges to the maximal
margin classifier (Section 12.2). This gives an attractive alternative to the
support-vector machine, discussed next, especially in the multiclass case.
</p>
<p>18.3.3 The Support Vector Classifier
</p>
<p>The support vector classifier is described for the two-class case in Sec-
tion 12.2. When p &gt; N , it is especially attractive because in general the</p>
<p/>
</div>
<div class="page"><p/>
<p>658 18. High-Dimensional Problems: p ≫ N
</p>
<p>classes are perfectly separable by a hyperplane unless there are identical
feature vectors in different classes. Without any regularization the support
vector classifier finds the separating hyperplane with the largest margin;
that is, the hyperplane yielding the biggest gap between the classes in
the training data. Somewhat surprisingly, when p ≫ N the unregularized
support vector classifier often works about as well as the best regularized
version. Overfitting often does not seem to be a problem, partly because of
the insensitivity of misclassification loss.
There are many different methods for generalizing the two-class support-
</p>
<p>vector classifier to K &gt; 2 classes. In the &ldquo;one versus one&rdquo; (ovo) approach,
we compute all
</p>
<p>(
K
2
</p>
<p>)
pairwise classifiers. For each test point, the predicted
</p>
<p>class is the one that wins the most pairwise contests. In the &ldquo;one versus all&rdquo;
(ova) approach, each class is compared to all of the others in K two-class
comparisons. To classify a test point, we compute the confidences (signed
distance from the hyperplane) for each of theK classifiers. The winner is the
class with the highest confidence. Finally, Vapnik (1998) and Weston and
Watkins (1999) suggested (somewhat complex) multiclass criteria which
generalize the two-class criterion (12.7).
Tibshirani and Hastie (2007) propose the margin tree classifier, in which
</p>
<p>support-vector classifiers are used in a binary tree, much as in CART
(Chapter 9). The classes are organized in a hierarchical manner, which can
be useful for classifying patients into different cancer types, for example.
Line 3 of Table 18.1 shows the results for the support vector classifier
</p>
<p>using the ova method; Ramaswamy et al. (2001) reported (and we con-
firmed) that this approach worked best for this problem. The errors are
very similar to those in line 6, as we might expect from the comments
at the end of the previous section. The error rates are insensitive to the
choice of C [the regularization parameter in (12.8) on page 420], for values
of C &gt; 0.001. Since p &gt; N , the support vector hyperplane can perfectly
separate the training data by setting C = &infin;.
</p>
<p>18.3.4 Feature Selection
</p>
<p>Feature selection is an important scientific requirement for a classifier when
p is large. Neither discriminant analysis, logistic regression, nor the support-
vector classifier perform feature selection automatically, because all use
quadratic regularization. All features have nonzero weights in both models.
Ad-hoc methods for feature selection have been proposed, for example,
removing genes with small coefficients, and refitting the classifier. This is
done in a backward stepwise manner, starting with the smallest weights and
moving on to larger weights. This is known as recursive feature elimination
(Guyon et al., 2002). It was not successful in this example; Ramaswamy
et al. (2001) report, for example, that the accuracy of the support-vector
classifier starts to degrade as the number of genes is reduced from the full</p>
<p/>
</div>
<div class="page"><p/>
<p>18.3 Linear Classifiers with Quadratic Regularization 659
</p>
<p>set of 16, 063. This is rather remarkable, as the number of training samples
is only 144. We do not have an explanation for this behavior.
All three methods discussed in this section (RDA, LR and SVM) can
</p>
<p>be modified to fit nonlinear decision boundaries using kernels. Usually the
motivation for such an approach is to increase the model complexity. With
p ≫ N the models are already sufficiently complex and overfitting is always
a danger. Yet despite the high dimensionality, radial kernels (Section 12.3.3)
sometimes deliver superior results in these high dimensional problems. The
radial kernel tends to dampen inner products between points far away from
each other, which in turn leads to robustness to outliers. This occurs often
in high dimensions, and may explain the positive results. We tried a radial
kernel with the SVM in Table 18.1, but in this case the performance was
inferior.
</p>
<p>18.3.5 Computational Shortcuts When p ≫ N
The computational techniques discussed in this section apply to any method
that fits a linear model with quadratic regularization on the coefficients.
That includes all the methods discussed in this section, and many more.
When p &gt; N , the computations can be carried out in an N -dimensional
space, rather than p, via the singular value decomposition introduced in
Section 14.5. Here is the geometric intuition: just like two points in three-
dimensional space always lie on a line, N points in p-dimensional space lie
in an (N &minus; 1)-dimensional affine subspace.
Given the N &times; p data matrix X, let
</p>
<p>X = UDVT (18.12)
</p>
<p>= RVT (18.13)
</p>
<p>be the singular-value decomposition (SVD) of X; that is, V is p&times;N with
orthonormal columns, U is N &times; N orthogonal, and D a diagonal matrix
with elements d1 &ge; d2 &ge; dN &ge; 0. The matrix R is N &times;N , with rows rTi .
As a simple example, let&rsquo;s first consider the estimates from a ridge re-
</p>
<p>gression:
β̂ = (XTX+ λI)&minus;1XTy. (18.14)
</p>
<p>Replacing X by RVT and after some further manipulations, this can be
shown to equal
</p>
<p>β̂ = V(RTR+ λI)&minus;1RTy (18.15)
</p>
<p>(Exercise 18.4). Thus β̂ = Vθ̂, where θ̂ is the ridge-regression estimate
using the N observations (ri, yi), i = 1, 2, . . . , N . In other words, we can
simply reduce the data matrix from X to R, and work with the rows of
R. This trick reduces the computational cost from O(p3) to O(pN2) when
p &gt; N .</p>
<p/>
</div>
<div class="page"><p/>
<p>660 18. High-Dimensional Problems: p ≫ N
</p>
<p>These results can be generalized to all models that are linear in the
parameters and have quadratic penalties. Consider any supervised learning
problem where we use a linear function f(X) = β0 + X
</p>
<p>Tβ to model a
parameter in the conditional distribution of Y |X. We fit the parameters β
by minimizing some loss function
</p>
<p>&sum;N
i=1 L(yi, f(xi)) over the data with a
</p>
<p>quadratic penalty on β. Logistic regression is a useful example to have in
mind. Then we have the following simple theorem:
</p>
<p>Let f&lowast;(ri) = θ0 + r
T
i θ with ri defined in (18.13), and consider the pair of
</p>
<p>optimization problems:
</p>
<p>(β̂0, β̂) = arg min
β0,β&isin;IRp
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>L(yi, β0 + x
T
i β) + λβ
</p>
<p>Tβ; (18.16)
</p>
<p>(θ̂0, θ̂) = arg min
θ0,θ&isin;IRN
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>L(yi, θ0 + r
T
i θ) + λθ
</p>
<p>T θ. (18.17)
</p>
<p>Then the β̂0 = θ̂0, and β̂ = Vθ̂.
</p>
<p>The theorem says that we can simply replace the p vectors xi by the
N -vectors ri, and perform our penalized fit as before, but with far fewer
predictors. The N -vector solution θ̂ is then transformed back to the p-
vector solution via a simple matrix multiplication. This result is part of
the statistics folklore, and deserves to be known more widely&mdash;see Hastie
and Tibshirani (2004) for further details.
Geometrically, we are rotating the features to a coordinate system in
</p>
<p>which all but the first N coordinates are zero. Such rotations are allowed
since the quadratic penalty is invariant under rotations, and linear models
are equivariant.
This result can be applied to many of the learning methods discussed
</p>
<p>in this chapter, such as regularized (multiclass) logistic regression, linear
discriminant analysis (Exercise 18.6), and support vector machines. It also
applies to neural networks with quadratic regularization (Section 11.5.2).
Note, however, that it does not apply to methods such as the lasso, which
uses nonquadratic (L1) penalties on the coefficients.
Typically we use cross-validation to select the parameter λ. It can be
</p>
<p>seen (Exercise 18.12) that we only need to construct R once, on the original
data, and use it as the data for each of the CV folds.
The support vector &ldquo;kernel trick&rdquo; of Section 12.3.7 exploits the same re-
</p>
<p>duction used in this section, in a slightly different context. Suppose we have
at our disposal the N &times;N gram (inner-product) matrix K = XXT . From
(18.12) we have K = UD2UT , and so K captures the same information as
R. Exercise 18.13 shows how we can exploit the ideas in this section to fit
a ridged logistic regression with K using its SVD.</p>
<p/>
</div>
<div class="page"><p/>
<p>18.4 Linear Classifiers with L1 Regularization 661
</p>
<p>18.4 Linear Classifiers with L1 Regularization
</p>
<p>The methods of Section 18.3 use an L2 penalty to regularize their pa-
rameters, just as in ridge regression. All of the estimated coefficients are
nonzero, and hence no feature selection is performed. In this section we dis-
cuss methods that use L1 penalties instead, and hence provide automatic
feature selection.
Recall the lasso of Section 3.4.2,
</p>
<p>min
β
</p>
<p>1
</p>
<p>2
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>(
yi &minus; β0 &minus;
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>xijβj
</p>
<p>)2
+ λ
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>|βj |, (18.18)
</p>
<p>which we have written in the Lagrange form (3.52). As discussed there, the
</p>
<p>use of the L1 penalty causes a subset of the solution coefficients β̂j to be
exactly zero, for a sufficiently large value of the tuning parameter λ.
</p>
<p>In Section 3.8.1 we discussed the LARS algorithm, an efficient procedure
for computing the lasso solution for all λ. When p &gt; N (as in this chapter),
as λ approaches zero, the lasso fits the training data exactly. In fact, by
convex duality one can show that when p &gt; N the number of non-zero
coefficients is at most N for all values of λ (Rosset and Zhu, 2007, for
example). Thus the lasso provides a (severe) form of feature selection.
Lasso regression can be applied to a two-class classification problem by
</p>
<p>coding the outcome &plusmn;1, and applying a cutoff (usually 0) to the predictions.
For more than two classes, there are many possible approaches, including
the ova and ovo methods discussed in Section 18.3.3. We tried the ova-
approach on the cancer data in Section 18.3. The results are shown in
line (4) of Table 18.1. Its performance is among the best.
A more natural approach for classification problems is to use the lasso
</p>
<p>penalty to regularize logistic regression. Several implementations have been
proposed in the literature, including path algorithms similar to LARS (Park
and Hastie, 2007). Because the paths are piecewise smooth but nonlinear,
exact methods are slower than the LARS algorithm, and are less feasible
when p is large.
Friedman et al. (2010) provide very fast algorithms for fitting L1-pen-
</p>
<p>alized logistic and multinomial regression models. They use the symmetric
multinomial logistic regression model as in (18.10) in Section 18.3.2, and
maximize the penalized log-likelihood
</p>
<p>max
{β0k,βk&isin;IRp}K1
</p>
<p>⎡
⎣
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>log Pr(gi|xi)&minus; λ
K&sum;
</p>
<p>k=1
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>|βkj |
</p>
<p>⎤
⎦ ; (18.19)
</p>
<p>compare with (18.11). Their algorithm computes the exact solution at a
pre-chosen sequence of values for λ by cyclical coordinate descent (Sec-
tion 3.8.6), and exploits the fact that solutions are sparse when p ≫ N ,</p>
<p/>
</div>
<div class="page"><p/>
<p>662 18. High-Dimensional Problems: p ≫ N
</p>
<p>as well as the fact that solutions for neighboring values of λ tend to be
very similar. This method was used in line (7) of Table 18.1, with the over-
all tuning parameter λ chosen by cross-validation. The performance was
similar to that of the best methods, except here the automatic feature se-
lection chose 269 genes altogether. A similar approach is used in Genkin
et al. (2007); although they present their model from a Bayesian point of
view, they in fact compute the posterior mode, which solves the penalized
maximum-likelihood problem.
</p>
<p>&minus;8 &minus;7 &minus;6 &minus;5 &minus;4 &minus;3 &minus;2 &minus;1
</p>
<p>0
.0
</p>
<p>0
.5
</p>
<p>1
.0
</p>
<p>1
.5
</p>
<p>2
.0
</p>
<p>&minus;8 &minus;7 &minus;6 &minus;5 &minus;4 &minus;3 &minus;2 &minus;1
</p>
<p>0
.0
</p>
<p>0
.5
</p>
<p>1
.0
</p>
<p>1
.5
</p>
<p>2
.0
</p>
<p>C
o
e
ffi
c
ie
n
ts
</p>
<p>C
o
e
ffi
c
ie
n
ts
</p>
<p>log(λ)log(λ)
</p>
<p>Lasso Elastic Net
</p>
<p>FIGURE 18.5. Regularized logistic regression paths for the leukemia data. The
left panel is the lasso path, the right panel the elastic-net path with α = 0.8. At
the ends of the path (extreme left), there are 19 nonzero coefficients for the lasso,
and 39 for the elastic net. The averaging effect of the elastic net results in more
non-zero coefficients than the lasso, but with smaller magnitudes.
</p>
<p>In genomic applications, there are often strong correlations among the
variables; genes tend to operate in molecular pathways. The lasso penalty
is somewhat indifferent to the choice among a set of strong but corre-
lated variables (Exercise 3.28). The ridge penalty, on the other hand, tends
to shrink the coefficients of correlated variables toward each other (Exer-
cise 3.29 on page 99). The elastic net penalty (Zou and Hastie, 2005) is a
compromise, and has the form
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>(
α|βj |+ (1&minus; α)β2j
</p>
<p>)
. (18.20)
</p>
<p>The second term encourages highly correlated features to be averaged, while
the first term encourages a sparse solution in the coefficients of these aver-</p>
<p/>
</div>
<div class="page"><p/>
<p>18.4 Linear Classifiers with L1 Regularization 663
</p>
<p>aged features. The elastic net penalty can be used with any linear model,
in particular for regression or classification.
Hence the multinomial problem above with elastic-net penalty becomes
</p>
<p>max
{β0k,βk&isin;IRp}K1
</p>
<p>⎡
⎣
</p>
<p>N&sum;
</p>
<p>i=1
</p>
<p>log Pr(gi|xi)&minus; λ
K&sum;
</p>
<p>k=1
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>(
α|βkj |+ (1&minus; α)β2kj
</p>
<p>)
⎤
⎦ .
</p>
<p>(18.21)
The parameter α determines the mix of the penalties, and is often pre-
chosen on qualitative grounds. The elastic net can yield more that N non-
zero coefficients when p &gt; N , a potential advantage over the lasso. Line
(8) in Table 18.1 uses this model, with α and λ chosen by cross-validation.
We used a sequence of 20 values of α between 0.05 and 1.0, and a 100
values of λ uniform on the log scale covering the entire range. Values of
α &isin; [0.75, 0.80] gave the minimum CV error, with values of λ &lt; 0.001 for all
tied solutions. Although it has the lowest test error among all methods, the
margin is small and not significant. Interestingly, when CV is performed
separately for each value of α, a minimum test error of 8.8 is achieved at
α = 0.10, but this is not the value chosen in the two-dimensional CV.
</p>
<p>&minus;8 &minus;7 &minus;6 &minus;5 &minus;4 &minus;3 &minus;2 &minus;1
</p>
<p>0
.0
</p>
<p>0
.1
</p>
<p>0
.2
</p>
<p>0
.3
</p>
<p>0
.4
</p>
<p>M
is
</p>
<p>c
la
</p>
<p>s
s
if
ic
</p>
<p>a
ti
o
n
 E
</p>
<p>rr
o
r
</p>
<p>Training
Test
10&minus;fold CV
</p>
<p>&minus;8 &minus;7 &minus;6 &minus;5 &minus;4 &minus;3 &minus;2 &minus;1
</p>
<p>0
5
</p>
<p>1
0
</p>
<p>1
5
</p>
<p>2
0
</p>
<p>2
5
</p>
<p>3
0
</p>
<p>D
e
v
ia
</p>
<p>n
c
e
</p>
<p>log(λ)log(λ)
</p>
<p>FIGURE 18.6. Training, test, and 10-fold cross validation curves for lasso logis-
tic regression on the leukemia data. The left panel shows misclassification errors,
the right panel shows deviance.
</p>
<p>Figure 18.5 shows the lasso and elastic-net coefficient paths on the two-
class leukemia data (Golub et al., 1999). There are 7129 gene-expression
measurements on 38 samples, 27 of them in class ALL (acute lymphocytic
leukemia), and 11 in class AML (acute myelogenous leukemia). There is
also a test set with 34 samples (20, 14). Since the data are linearly separa-
ble, the solution is undefined at λ = 0 (Exercise 18.11), and degrades for
very small values of λ. Hence the paths have been truncated as the fitted
probabilities approach 0 and 1. There are 19 non-zero coefficients in the
left plot, and 39 in the right. Figure 18.6 (left panel) shows the misclas-</p>
<p/>
</div>
<div class="page"><p/>
<p>664 18. High-Dimensional Problems: p ≫ N
</p>
<p>sification errors for the lasso logistic regression on the training and test
data, as well as for 10-fold cross-validation on the training data. The right
panel uses binomial deviance to measure errors, and is much smoother. The
small sample sizes lead to considerable sampling variance in these curves,
even though individual curves are relatively smooth (see, for example, Fig-
ure 7.1 on page 220). Both of these plots suggest that the limiting solution
λ &darr; 0 is adequate, leading to 3/34 misclassifications in the test set. The
corresponding figures for the elastic net are qualitatively similar and are
not shown.
For p ≫ N , the limiting coefficients diverge for all regularized logistic
</p>
<p>regression models, so in practical software implementations a minimum
value for λ &gt; 0 is either explicitly or implicitly set. However, renormalized
versions of the coefficients converge, and these limiting solutions can be
thought of as interesting alternatives to the linear optimal separating hy-
perplane (SVM). With α = 0 the limiting solution coincides with the SVM
(see end of Section 18.3.2), but all the 7129 genes are selected. With α = 1,
the limiting solution coincides with an L1 separating hyperplane (Rosset
et al., 2004a), and includes at most 38 genes. As α decreases from 1, the
elastic-net solutions include more genes in the separating hyperplane.
</p>
<p>18.4.1 Application of Lasso to Protein Mass Spectroscopy
</p>
<p>Protein mass spectrometry has become a popular technology for analyzing
the proteins in blood, and can be used to diagnose a disease or understand
the processes underlying it.
For each blood serum sample i, we observe the intensity xij for many
</p>
<p>time of flight values tj . This intensity is related to the number of particles
observed to take approximately tj time to pass from the emitter to the
detector during a cycle of operation of the machine. The time of flight has
a known relationship to the mass over charge ratio (m/z) of the constituent
proteins in the blood. Hence the identification of a peak in the spectrum
at a certain tj tells us that there is a protein with a corresponding mass
and charge. The identity of this protein can then be determined by other
means.
Figure 18.7 shows an example taken from Adam et al. (2003). It shows
</p>
<p>the average spectra for healthy patients and those with prostate cancer.
There are 16,898 m/z sites in total, ranging in value from 2000 to 40,000.
The full dataset consists of 157 healthy patients and 167 with cancer, and
the goal is to find m/z sites that discriminate between the two groups.
This is an example of functional data; the predictors can be viewed as a
function of m/z. There has been much interest in this problem in the past
few years; see e.g. Petricoin et al. (2002).
The data were first standardized (baseline subtraction and normaliza-
</p>
<p>tion), and we restricted attention to m/z values between 2000 and 40,000
(spectra outside of this range were not of interest). We then applied near-</p>
<p/>
</div>
<div class="page"><p/>
<p>18.4 Linear Classifiers with L1 Regularization 665
</p>
<p>2e+03 5e+03 1e+04 2e+04 5e+04 1e+05 2e+05
</p>
<p>1
0
</p>
<p>2
0
</p>
<p>3
0
</p>
<p>4
0
</p>
<p>m/z
</p>
<p>In
te
</p>
<p>n
s
it
y
</p>
<p>Normal
Cancer
</p>
<p>FIGURE 18.7. Protein mass spectrometry data: average profiles from normal
and prostate cancer patients.
</p>
<p>est shrunken centroids and lasso regression to the data, with the results for
both methods shown in Table 18.2.
By fitting harder to the data, the lasso achieves a considerably lower
</p>
<p>test error rate. However, it may not provide a scientifically useful solu-
tion. Ideally, protein mass spectrometry resolves a biological sample into
its constituent proteins, and these should appear as peaks in the spectra.
The lasso doesn&rsquo;t treat peaks in any special way, so not surprisingly only
some of the non-zero lasso weights were situated near peaks in the spectra.
Furthermore, the same protein may yield a peak at slightly different m/z
values in different spectra. In order to identify common peaks, some kind
of m/z warping is needed from sample to sample.
To address this, we applied a standard peak-extraction algorithm to each
</p>
<p>spectrum, yielding a total of 5178 peaks in the 217 training spectra. Our
idea was to pool the collection of peaks from all patients, and hence con-
struct a set of common peaks. For this purpose, we applied hierarchical
clustering to the positions of these peaks along the log m/z axis. We cut
the resulting dendrogram horizontally at height log(0.005)3, and computed
averages of the peak positions in each resulting cluster. This process yielded
728 common clusters and their corresponding peak centers.
Given these 728 common peaks, we determined which of these were
</p>
<p>present in each individual spectrum, and if present, the height of the peak.
A peak height of zero was assigned if that peak was not found. This pro-
duced a 217&times; 728 matrix of peak heights as features, which was used in a
lasso regression. We scored the test spectra for the same 728 peaks.
</p>
<p>3Use of the value 0.005 means that peaks with positions less than 0.5% apart are
</p>
<p>considered the same peak, a fairly common assumption.</p>
<p/>
</div>
<div class="page"><p/>
<p>666 18. High-Dimensional Problems: p ≫ N
</p>
<p>TABLE 18.2. Results for the prostate data example. The standard deviation for
the test errors is about 4.5.
</p>
<p>Method Test Errors/108 Number of Sites
</p>
<p>1. Nearest shrunken centroids 34 459
2. Lasso 22 113
3. Lasso on peaks 28 35
</p>
<p>The prediction results for this application of the lasso to the peaks are
shown in the last line of Table 18.2: it does fairly well, but not as well
as the lasso on the raw spectra. However, the fitted model may be more
useful to the biologist as it yields 35 peak positions for further study. On
the other hand, the results suggest that there may be useful discriminatory
information between the peaks of the spectra, and the positions of the lasso
sites from line (2) of the table also deserve further examination.
</p>
<p>18.4.2 The Fused Lasso for Functional Data
</p>
<p>In the previous example, the features had a natural order, determined by
the mass-to-charge ratio m/z. More generally, we may have functional fea-
tures xi(t) that are ordered according to some index variable t. We have
already discussed several approaches for exploiting such structure.
We can represent xi(t) by their coefficients in a basis of functions in t,
</p>
<p>such as splines, wavelets or Fourier bases, and then apply a regression using
these coefficients as predictors. Equivalently, one can instead represent the
coefficients of the original features in these bases. These approaches are
described in Section 5.3.
In the classification setting, we discuss the analogous approach of penal-
</p>
<p>ized discriminant analysis in Section 12.6. This uses a penalty that explicitly
controls the resulting smoothness of the coefficient vector.
The above methods tend to smooth the coefficients uniformly. Here we
</p>
<p>present a more adaptive strategy that modifies the lasso penalty to take
into account the ordering of the features. The fused lasso (Tibshirani et
al., 2005) solves
</p>
<p>min
β&isin;IRp
</p>
<p>{
N&sum;
</p>
<p>i=1
</p>
<p>(yi&minus;β0&minus;
p&sum;
</p>
<p>j=1
</p>
<p>xijβj)
2+λ1
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>|βj |+λ2
p&minus;1&sum;
</p>
<p>j=1
</p>
<p>|βj+1&minus;βj |
}
. (18.22)
</p>
<p>This criterion is strictly convex in β, so a unique solution exists. The first
penalty encourages the solution to be sparse, while the second encourages
it to be smooth in the index j.
</p>
<p>The difference penalty in (18.22) assumes an uniformly spaced index j. If
instead the underlying index variable t has nonuniform values tj , a natural
generalization of (18.22) would be based on divided differences</p>
<p/>
</div>
<div class="page"><p/>
<p>18.4 Linear Classifiers with L1 Regularization 667
</p>
<p>0 200 400 600 800 1000
</p>
<p>&minus;
2
</p>
<p>0
2
</p>
<p>4
</p>
<p>Genome order
</p>
<p>lo
g
2
ra
ti
o
</p>
<p>FIGURE 18.8. Fused lasso applied to CGH data. Each point represents the
copy-number of a gene in a tumor sample, relative to that of a control (on the log
base-2 scale).
</p>
<p>λ2
</p>
<p>p&minus;1&sum;
</p>
<p>j=1
</p>
<p>|βj+1 &minus; βj |
|tj+1 &minus; tj |
</p>
<p>. (18.23)
</p>
<p>This amounts to having a penalty modifier for each of the terms in the
series.
A particularly useful special case arises when the predictor matrix X =
</p>
<p>IN , the N &times; N identity matrix. This is a special case of the fused lasso,
used to approximate a sequence {yi}N1 . The fused lasso signal approximator
solves
</p>
<p>min
β&isin;IRN
</p>
<p>{
N&sum;
</p>
<p>i=1
</p>
<p>(yi &minus; β0 &minus; βi)2 + λ1
N&sum;
</p>
<p>i=1
</p>
<p>|βi|+ λ2
N&minus;1&sum;
</p>
<p>i=1
</p>
<p>|βi+1 &minus; βi|
}
. (18.24)
</p>
<p>Figure 18.8 shows an example taken from Tibshirani and Wang (2007). The
data in the panel come from a Comparative Genomic Hybridization (CGH)
array, measuring the approximate log (base-two) ratio of the number of
copies of each gene in a tumor sample, as compared to a normal sample.
The horizontal axis represents the chromosomal location of each gene. The
idea is that in cancer cells, genes are often amplified (duplicated) or deleted,
and it is of interest to detect these events. Furthermore, these events tend
to occur in contiguous regions. The smoothed signal estimate from the
fused lasso signal approximator is shown in dark red (with appropriately
chosen values for λ1 and λ2). The significantly nonzero regions can be used
to detect locations of gains and losses of genes in the tumor.
There is also a two-dimensional version of the fused lasso, in which the
</p>
<p>parameters are laid out in a grid of pixels, and a penalty is applied to the</p>
<p/>
</div>
<div class="page"><p/>
<p>668 18. High-Dimensional Problems: p ≫ N
</p>
<p>first differences to the left, right, above and below the target pixel. This
can be useful for denoising or classifying images. Friedman et al. (2007)
develop fast generalized coordinate descent algorithms for the one- and
two-dimensional fused lasso.
</p>
<p>18.5 Classification When Features are Unavailable
</p>
<p>In some applications the objects under study are more abstract in nature,
and it is not obvious how to define a feature vector. As long as we can fill
in an N&times;N proximity matrix of similarities between pairs of objects in our
database, it turns out we can put to use many of the classifiers in our arsenal
by interpreting the proximities as inner-products. Protein structures fall
into this category, and we explore an example in Section 18.5.1 below.
In other applications, such as document classification, feature vectors are
</p>
<p>available but can be extremely high-dimensional. Here we may not wish
to compute with such high-dimensional data, but rather store the inner-
products between pairs of documents. Often these inner-products can be
approximated by sampling techniques.
Pairwise distances serve a similar purpose, because they can be turned
</p>
<p>into centered inner-products. Proximity matrices are discussed in more de-
tail in Chapter 14.
</p>
<p>18.5.1 Example: String Kernels and Protein Classification
</p>
<p>An important problem in computational biology is to classify proteins into
functional and structural classes based on their sequence similarities. Pro-
tein molecules are strings of amino acids, differing in both length and com-
position. In the example we consider, the lengths vary between 75&ndash;160
amino-acid molecules, each of which can be one of 20 different types, labeled
using letters. Here are two examples, of length 110 and 153, respectively:
</p>
<p>IPTSALVKETLALLSTHRTLLIANETLRIPVPVHKNHQLCTEEIFQGIGTLESQTVQGGTV
</p>
<p>ERLFKNLSLIKKYIDGQKKKCGEERRRVNQFLDYLQEFLGVMNTEWI
</p>
<p>PHRRDLCSRSIWLARKIRSDLTALTESYVKHQGLWSELTEAERLQENLQAYRTFHVLLA
</p>
<p>RLLEDQQVHFTPTEGDFHQAIHTLLLQVAAFAYQIEELMILLEYKIPRNEADGMLFEKK
</p>
<p>LWGLKVLQELSQWTVRSIHDLRFISSHQTGIP
</p>
<p>There have been many proposals for measuring the similarity between a
pair of protein molecules. Here we focus on a measure based on the count
of matching substrings (Leslie et al., 2004), such as the LQE above.
</p>
<p>To construct our features, we count the number of times that a given
sequence of length m occurs in our string, and we compute this number</p>
<p/>
</div>
<div class="page"><p/>
<p>18.5 Classification When Features are Unavailable 669
</p>
<p>for all possible sequences of length m. Formally, for a string x, we define a
feature map
</p>
<p>Φm(x) = {φa(x)}a&isin;Am (18.25)
</p>
<p>where Am is the set of subsequences of length m, and φa(x) is the number
of times that &ldquo;a&rdquo; occurs in our string x. Using this, we define the inner
product
</p>
<p>Km(x1, x2) = 〈Φm(x1),Φm(x2)〉, (18.26)
</p>
<p>which measures the similarity between the two strings x1, x2. This can be
used to drive, for example, a support vector classifier for classifying strings
into different protein classes.
Now the number of possible sequences a is |Am| = 20m, which can be
</p>
<p>very large for moderate m, and the vast majority of the subsequences do
not match the strings in our training set. It turns out that we can compute
the N &times; N inner-product matrix or string kernel Km (18.26) efficiently
using tree-structures, without actually computing the individual vectors.
This methodology, and the data to follow, come from Leslie et al. (2004).4
</p>
<p>The data consist of 1708 proteins in two classes&mdash; negative (1663) and
positive (45). The two examples above, which we will call &ldquo;x1&rdquo; and &ldquo;x2&rdquo;,
are from this set. We have marked the occurrences of subsequence LQE,
which appears in both proteins. There are 203 possible subsequences, so
Φ3(x) will be a vector of length 8000. For this example φLQE(x1) = 1 and
φLQE(x2) = 2.
Using software from Leslie et al. (2004), we computed the string kernel
</p>
<p>for m = 4, which was then used in a support vector classifier to find the
maximal margin solution in this 204 = 160, 000-dimensional feature space.
We used 10-fold cross-validation to compute the SVM predictions on all of
the training data. The orange curve in Figure 18.9 shows the cross-validated
ROC curve for the support vector classifier, computed by varying the cut-
point on the real-valued predictions from the cross-validated support vector
classifier. The area under the curve is 0.84. Leslie et al. (2004) show that
the string kernel method is competitive with, but perhaps not as accurate
as, more specialized methods for protein string matching.
Many other classifiers can be computed using only the information in the
</p>
<p>kernel matrix; some details are given in the next section. The results for
the nearest centroid classifier (green), and distance-weighted one-nearest
neighbors (blue) are shown in Figure 18.9. Their performance is similar to
that of the support vector classifier.
</p>
<p>4We thank Christina Leslie for her help and for providing the data, which is available
</p>
<p>on our book website.</p>
<p/>
</div>
<div class="page"><p/>
<p>670 18. High-Dimensional Problems: p ≫ N
</p>
<p>0.0 0.2 0.4 0.6 0.8 1.0
</p>
<p>0
.0
</p>
<p>0
.2
</p>
<p>0
.4
</p>
<p>0
.6
</p>
<p>0
.8
</p>
<p>1
.0
</p>
<p>ROC Curves for String Kernel
</p>
<p>Specificity
</p>
<p>S
e
n
s
it
iv
</p>
<p>it
y
</p>
<p>SVM  0.84
Nearest Centroid  0.84
One&minus;Nearest Neighbor 0.86
</p>
<p>FIGURE 18.9. Cross-validated ROC curves for protein example using the string
kernel. The numbers next to each method in the legend give the area under the
curve, an overall measure of accuracy. The SVM achieves better sensitivities than
the other two, which achieve better specificities.
</p>
<p>18.5.2 Classification and Other Models Using Inner-Product
Kernels and Pairwise Distances
</p>
<p>There are a number of other classifiers, besides the support-vector ma-
chine, that can be implemented using only inner-product matrices. This
also implies they can be &ldquo;kernelized&rdquo; like the SVM.
An obvious example is nearest-neighbor classification, since we can trans-
</p>
<p>form pairwise inner-products to pairwise distances:
</p>
<p>||xi &minus; xi&prime; ||2 = 〈xi, xi〉+ 〈xi&prime; , xi&prime;〉 &minus; 2〈xi, xi&prime;〉. (18.27)
A variation of 1-NN classification is used in Figure 18.9, which produces
a continuous discriminant score needed to construct a ROC curve. This
distance-weighted 1-NN makes use of the distance of a test points to the
closest member of each class; see Exercise 18.14.
Nearest-centroid classification follows easily as well. For training pairs
</p>
<p>(xi, gi), i = 1, . . . , N , a test point x0, and class centroids x̄k, k = 1, . . . ,K
we can write
</p>
<p>||x0 &minus; x̄k||2 = 〈x0, x0〉 &minus;
2
</p>
<p>Nk
</p>
<p>&sum;
</p>
<p>gi=k
</p>
<p>〈x0, xi〉+
1
</p>
<p>N2k
</p>
<p>&sum;
</p>
<p>gi=k
</p>
<p>&sum;
</p>
<p>gi&prime;=k
</p>
<p>〈xi, xi&prime;〉, (18.28)</p>
<p/>
</div>
<div class="page"><p/>
<p>18.5 Classification When Features are Unavailable 671
</p>
<p>Hence we can compute the distance of the test point to each of the cen-
troids, and perform nearest centroid classification. This also implies that
methods like K-means clustering can also be implemented, using only the
inner products of the data points.
Logistic and multinomial regression with quadratic regularization can
</p>
<p>also be implemented with inner-product kernels; see Section 12.3.3 and
Exercise 18.13. Exercise 12.10 derives linear discriminant analysis using an
inner-product kernel.
Principal components can be computed using inner-product kernels as
</p>
<p>well; since this is frequently useful, we give some details. Suppose first
that we have a centered data matrix X, and let X = UDVT be its SVD
(18.12). Then Z = UD is the matrix of principal component variables (see
Section 14.5.1). But if K = XXT , then it follows that K = UD2UT , and
hence we can compute Z from the eigen decomposition of K. If X is not
centered, then we can center it using X̃ = (I &minus; M)X, where M = 1N 11T
is the mean operator. Thus we compute the eigenvectors of the double-
centered kernel (I &minus; M)K(I &minus; M) for the principal components from an
uncentered inner-product matrix. Exercise 18.15 explores this further, and
Section 14.5.4 discusses in more detail kernel PCA for general kernels, such
as the radial kernel used in SVMs.
If instead we had available only the pairwise (squared) Euclidean dis-
</p>
<p>tances between observations,
</p>
<p>∆2ii&prime; = ||xi &minus; xi&prime; ||2, (18.29)
</p>
<p>it turns out we can do all of the above as well. The trick is to convert the
pairwise distances to centered inner-products, and then proceed as before.
We write
</p>
<p>∆2ii&prime; = ||xi &minus; x̄||2 + ||xi&prime; &minus; x̄||2 &minus; 2〈xi &minus; x̄, xi&prime; &minus; x̄〉. (18.30)
</p>
<p>Defining B = {&minus;∆2ii&prime;/2}, we double center B:
</p>
<p>K̃ = (I&minus;M)B(I&minus;M); (18.31)
</p>
<p>it is easy to check that K̃ii&prime; = 〈xi &minus; x̄, xi&prime; &minus; x̄〉, the centered inner-product
matrix.
Distances and inner-products also allow us to compute the medoid in each
</p>
<p>class&mdash;the observation with smallest average distance to other observations
in that class. This can be used for classification (closest medoids), as well as
to drive k-medoids clustering (Section 14.3.10). With abstract data objects
like proteins, medoids have a practical advantage over means. The medoid is
one of the training examples, and can be displayed. We tried closest medoids
in the example in the next section (see Table 18.3), and its performance is
disappointing.
It is useful to consider what we cannot do with inner-product kernels and
</p>
<p>distances:</p>
<p/>
</div>
<div class="page"><p/>
<p>672 18. High-Dimensional Problems: p ≫ N
</p>
<p>TABLE 18.3. Cross-validated error rates for the abstracts example. The nearest
shrunken centroids ended up using no-shrinkage, but does use a word-by-word
standardization (section 18.2). This standardization gives it a distinct advantage
over the other methods.
</p>
<p>Method CV Error (SE)
</p>
<p>1. Nearest shrunken centroids 0.17 (0.05)
2. SVM 0.23 (0.06)
3. Nearest medoids 0.65 (0.07)
4. 1-NN 0.44 (0.07)
5. Nearest centroids 0.29 (0.07)
</p>
<p>&bull; We cannot standardize the variables; standardization significantly im-
proves performance in the example in the next section.
</p>
<p>&bull; We cannot assess directly the contributions of individual variables.
In particular, we cannot perform individual t-tests, fit the nearest
shrunken centroids model, or fit any model that uses the lasso penalty.
</p>
<p>&bull; We cannot separate the good variables from the noise: all variables get
an equal say. If, as is often the case, the ratio of relevant to irrelevant
variables is small, methods that use kernels are not likely to work as
well as methods that do feature selection.
</p>
<p>18.5.3 Example: Abstracts Classification
</p>
<p>This somewhat whimsical example serves to illustrate a limitation of ker-
nel approaches. We collected the abstracts from 48 papers, 16 each from
Bradley Efron (BE), Trevor Hastie and Rob Tibshirani (HT) (frequent co-
authors), and Jerome Friedman (JF). We extracted all unique words from
these abstracts, and defined features xij to be the number of times word
j appears in abstract i. This is the so-called bag of words representation.
Quotations, parentheses and special characters were first removed from the
abstracts, and all characters were converted to lower case. We also removed
the word &ldquo;we&rdquo;, which could unfairly discriminate HT abstracts from the
others.
There were 4492 total words, of which p = 1310 were unique. We sought
</p>
<p>to classify the documents into BE, HT or JF on the basis of the features
xij . Although it is artificial, this example allows us to assess the possible
degradation in performance if information specific to the raw features is
not used.
We first applied the nearest shrunken centroid classifier to the data, using
</p>
<p>10-fold cross-validation. It essentially chose no shrinkage, and so used all the
features; see the first line of Table 18.3. The error rate is 17%; the number
of features can be reduced to about 500 without much loss in accuracy.</p>
<p/>
</div>
<div class="page"><p/>
<p>18.5 Classification When Features are Unavailable 673
</p>
<p>Note that the nearest shrunken classifier requires the raw feature matrix
X in order to standardize the features individually. Figure 18.10 shows the
</p>
<p>predictive
bayes
using
than
</p>
<p>algorithm
are
</p>
<p>procedure
technology
</p>
<p>values
accuracy
variables
</p>
<p>when
inference
</p>
<p>those
bayesian
</p>
<p>frequentist
propose
</p>
<p>presented
method
</p>
<p>problems
</p>
<p>BE HT JF
</p>
<p>FIGURE 18.10. Abstracts example: top 20 scores from nearest shrunken cen-
troids. Each score is the standardized difference in frequency for the word in the
given class (BE, HT or JF) versus all classes. Thus a positive score (to the right
of the vertical grey zero lines) indicates a higher frequency in that class; a negative
score indicates a lower relative frequency.
</p>
<p>top 20 discriminating words, with a positive score indicating that a word
appears more in that class than in the other classes.
Some of these terms make sense: for example &ldquo;frequentist&rdquo; and &ldquo;Bayesian&rdquo;
</p>
<p>reflect Efron&rsquo;s greater emphasis on statistical inference. However, many oth-
ers are surprising, and reflect personal writing styles: for example, Fried-
man&rsquo;s use of &ldquo;presented&rdquo; and HT&rsquo;s use of &ldquo;propose&rdquo;.
We then applied the support vector classifier with linear kernel and no
</p>
<p>regularization, using the &ldquo;all pairs&rdquo; (ovo) method to handle the three
classes (regularization of the SVM did not improve its performance). The
result is shown in Table 18.3. It does somewhat worse than the nearest
shrunken centroid classifier.
As mentioned, the first line of Table 18.3 represents nearest shrunken cen-
</p>
<p>troids (with no shrinkage). Denote by sj the pooled within-class standard
deviation for feature j, and s0 the median of the sj values. Then line (1)
also corresponds to nearest centroid classification, after first standardizing
each feature by sj + s0 [recall (18.4) on page 652].
Line (3) shows that the performance of nearest medoids is very poor,
</p>
<p>something which surprised us. It is perhaps due to the small sample sizes</p>
<p/>
</div>
<div class="page"><p/>
<p>674 18. High-Dimensional Problems: p ≫ N
</p>
<p>and high dimensions, with medoids having much higher variance than
means. The performance of the one-nearest neighbor classifier is also poor.
The performance of the nearest centroid classifier is also shown in Ta-
</p>
<p>ble 18.3 in line (5): it is better than nearest medoids, but worse than that
of nearest shrunken centroids, even with no shrinkage. The difference seems
to be the standardization of each feature that is done in nearest shrunken
centroids. This standardization is important here, and requires access to
the individual feature values. Nearest centroids uses a spherical metric, and
relies on the fact that the features are in similar units. The support vector
machine estimates a linear combination of the features and can better deal
with unstandardized features.
</p>
<p>18.6 High-Dimensional Regression: Supervised
Principal Components
</p>
<p>In this section we describe a simple approach to regression and generalized
regression that is especially useful when p ≫ N . We illustrate the method
on another microarray data example. The data is taken from Rosenwald
et al. (2002) and consists of 240 samples from patients with diffuse large
B-cell lymphoma (DLBCL), with gene expression measurements for 7399
genes. The outcome is survival time, either observed or right censored. We
randomly divided the lymphoma samples into a training set of size 160 and
a test set of size 80.
Although supervised principal components is useful for linear regression,
</p>
<p>its most interesting applications may be in survival studies, which is the
focus of this example.
We have not yet discussed regression with censored survival data in this
</p>
<p>book; it represents a generalized form of regression in which the outcome
variable (survival time) is only partly observed for some individuals. Sup-
pose for example we carry out a medical study that lasts for 365 days, and
for simplicity all subjects are recruited on day one. We might observe one
individual to die 200 days after the start of the study. Another individ-
ual might still be alive at 365 days when the study ends. This individual
is said to be &ldquo;right censored&rdquo; at 365 days. We know only that he or she
lived at least 365 days. Although we do not know how long past 365 days
the individual actually lived, the censored observation is still informative.
This is illustrated in Figure 18.11. Figure 18.12 shows the survival curve
estimated by the Kaplan&ndash;Meier method for the 80 patients in the test set.
See for example Kalbfleisch and Prentice (1980) for a description of the
Kaplan&ndash;Meier method.
Our objective in this example is to find a set of features (genes) that
</p>
<p>can predict the survival of an independent set of patients. This could be</p>
<p/>
</div>
<div class="page"><p/>
<p>18.6 High-Dimensional Regression: Supervised Principal Components 675
</p>
<p>Time(days)
</p>
<p>P
a
ti
e
n
t
</p>
<p>0 100 200 300 365 
</p>
<p>1
</p>
<p>2
</p>
<p>3
</p>
<p>4
</p>
<p>FIGURE 18.11. Censored survival data. For illustration there are four patients.
The first and third patients die before the study ends. The second patient is alive
at the end of the study (365 days), while the fourth patient is lost to follow-up
before the study ends. For example, this patient might have moved out of the
country. The survival times for patients two and four are said to be &ldquo;censored.&rdquo;
</p>
<p>0 5 10 15 20
</p>
<p>0
.0
</p>
<p>0
.2
</p>
<p>0
.4
</p>
<p>0
.6
</p>
<p>0
.8
</p>
<p>1
.0
</p>
<p>l
</p>
<p>l
ll lll lll l
</p>
<p>ll
ll l l ll l lllll
</p>
<p>l ll l l l l
</p>
<p>P
r(
T
</p>
<p>&ge;
t)
</p>
<p>Months t
</p>
<p>Survival Function
</p>
<p>FIGURE 18.12. Lymphoma data. The Kaplan&ndash;Meier estimate of the survival
function for the 80 patients in the test set, along with one-standard-error curves.
The curve estimates the probability of surviving past t months. The ticks indicate
censored observations.</p>
<p/>
</div>
<div class="page"><p/>
<p>676 18. High-Dimensional Problems: p ≫ N
</p>
<p>Poor Cell Type Good Cell Type
</p>
<p>Survival Time
</p>
<p>FIGURE 18.13. Underlying conceptual model for supervised principal compo-
nents. There are two cell types, and patients with the good cell type live longer on
the average. Supervised principal components estimate the cell type, by averaging
the expression of genes that reflect it.
</p>
<p>useful as a prognostic indicator to aid in choosing treatments, or to help
understand the biological basis for the disease.
The underlying conceptual model for supervised principal components
</p>
<p>is shown in Figure 18.13. We imagine that there are two cell types, and
patients with the good cell type live longer on the average. However there
is considerable overlap in the two sets of survival times. We might think
of survival time as a &ldquo;noisy surrogate&rdquo; for cell type. A fully supervised
approach would give the most weight to those genes having the strongest
relationship with survival. These genes are partially, but not perfectly, re-
lated to cell type. If we could instead discover the underlying cell types of
the patients, often reflected by a sizable signature of genes acting together
in pathways, then we might do a better job of predicting patient survival.
Although the cell type in Figure 18.13 is discrete, it is useful to imagine
</p>
<p>a continuous cell type, define by some linear combination of the features.
We will estimate the cell type as a continuous quantity, and then discretize
it for display and interpretation.
How can we find the linear combination that defines the important under-
</p>
<p>lying cell types? Principal components analysis (Section 14.5) is an effective
method for finding linear combinations of features that exhibit large varia-
tion in a dataset. But what we seek here are linear combinations with both
high variance and significant correlation with the outcome. The lower right
panel of Figure 18.14 shows the result of applying standard principal com-
ponents in this example; the leading component does not correlate strongly
with survival (details are given in the figure caption).
Hence we want to encourage principal component analysis to find linear
</p>
<p>combinations of features that have high correlation with the outcome. To
do this, we restrict attention to features which by themselves have a siz-
able correlation with the outcome. This is summarized in the supervised
principal components Algorithm 18.1, and illustrated in Figure 18.14.
The details in steps (1) and (2b) will depend on the type of outcome
</p>
<p>variable. For a standard regression problem, we use the univariate linear
least squares coefficients in step (1) and a linear least squares model in</p>
<p/>
</div>
<div class="page"><p/>
<p>18.6 High-Dimensional Regression: Supervised Principal Components 677
</p>
<p>7399 
</p>
<p>7350 
</p>
<p>50 
</p>
<p>27 
</p>
<p>1 
</p>
<p>G
e
n
e
s
</p>
<p>Patients
</p>
<p>Supervised PC
</p>
<p>1 80 160
</p>
<p>Absolute Cox Score
</p>
<p>0 2 4
</p>
<p>0 5 10 15 20
</p>
<p>0
.0
</p>
<p>0
.2
</p>
<p>0
.4
</p>
<p>0
.6
</p>
<p>0
.8
</p>
<p>1
.0
</p>
<p>P
ro
</p>
<p>b
a
b
ili
</p>
<p>ty
 o
</p>
<p>f 
S
</p>
<p>u
rv
</p>
<p>iv
a
l
</p>
<p>low score
high score
</p>
<p>Best Single Gene
</p>
<p>P=0.15
</p>
<p>0 5 10 15 20
</p>
<p>0
.0
</p>
<p>0
.2
</p>
<p>0
.4
</p>
<p>0
.6
</p>
<p>0
.8
</p>
<p>1
.0
</p>
<p>P
ro
</p>
<p>b
a
b
ili
</p>
<p>ty
 o
</p>
<p>f 
S
</p>
<p>u
rv
</p>
<p>iv
a
l
</p>
<p>Supervised Principal Component &minus; 27 Genes
</p>
<p>P=0.006
</p>
<p>0 5 10 15 20
</p>
<p>0
.0
</p>
<p>0
.2
</p>
<p>0
.4
</p>
<p>0
.6
</p>
<p>0
.8
</p>
<p>1
.0
</p>
<p>Months
</p>
<p>P
ro
</p>
<p>b
a
b
ili
</p>
<p>ty
 o
</p>
<p>f 
S
</p>
<p>u
rv
</p>
<p>iv
a
l
</p>
<p>Principal Component &minus; 7399 Genes
</p>
<p>P=0.14
</p>
<p>FIGURE 18.14. Supervised principal components on the lymphoma data. The
left panel shows a heatmap of a subset of the gene-expression training data. The
rows are ordered by the magnitude of the univariate Cox-score, shown in the mid-
dle vertical column. The top 50 and bottom 50 genes are shown. The supervised
principal component uses the top 27 genes (chosen by 10-fold CV). It is repre-
sented by the bar at the top of the heatmap, and is used to order the columns
of the expression matrix. In addition, each row is multiplied by the sign of the
Cox-score. The middle panel on the right shows the survival curves on the test
data when we create a low and high group by splitting this supervised PC at zero
(training data mean). The curves are well separated, as indicated by the p-value
for the log-rank test. The top panel does the same, using the top-scoring gene on
the training data. The curves are somewhat separated, but not significantly. The
bottom panel uses the first principal component on all the genes, and the separa-
tion is also poor. Each of the top genes can be interpreted as noisy surrogates for
a latent underlying cell-type characteristic, and supervised principal components
uses them all to estimate this latent factor.</p>
<p/>
</div>
<div class="page"><p/>
<p>678 18. High-Dimensional Problems: p ≫ N
</p>
<p>Algorithm 18.1 Supervised Principal Components.
</p>
<p>1. Compute the standardized univariate regression coefficients for the
outcome as a function of each feature separately.
</p>
<p>2. For each value of the threshold θ from the list 0 &le; θ1 &lt; θ2 &lt; &middot; &middot; &middot; &lt; θK :
</p>
<p>(a) Form a reduced data matrix consisting of only those features
whose univariate coefficient exceeds θ in absolute value, and
compute the first m principal components of this matrix.
</p>
<p>(b) Use these principal components in a regression model to predict
the outcome.
</p>
<p>3. Pick θ (and m) by cross-validation.
</p>
<p>step (2b). For survival problems, Cox&rsquo;s proportional hazards regression
model is widely used; hence we use the score test from this model in step (1)
and the multivariate Cox model in step (2b). The details are not essential
for understanding the basic method; they may be found in Bair et al. (2006).
Figure 18.14 shows the results of supervised principal components in this
</p>
<p>example. We used a Cox-score cutoff of 3.53, yielding 27 genes, where the
value 3.53 was found through 10-fold cross-validation. We then computed
the first principal component (m = 1) using just this subset of the data,
as well as its value for each of the test observations. We included this as
a quantitative predictor in a Cox regression model, and its likelihood-ratio
significance was p = 0.005. When dichotomized (using the mean score on
the training data as a threshold), it clearly separates the patients in the
test set into low and high risk groups (middle-right panel of Figure 18.14,
p = 0.006).
The top-right panel of Figure 18.14 uses the top scoring gene (dichot-
</p>
<p>omized) alone as a predictor of survival. It is not significant on the test set.
Likewise, the lower-right panel shows the dichotomized principal compo-
nent using all the training data, which is also not significant.
Our procedure allows m &gt; 1 principal components in step (2a). However,
</p>
<p>the supervision in step (1) encourages the principal components to align
with the outcome, and thus in most cases only the first or first few com-
ponents tend to be useful for prediction. In the mathematical development
below, we consider only the first component, but extensions to more than
one component can be derived in a similar way.
</p>
<p>18.6.1 Connection to Latent-Variable Modeling
</p>
<p>A formal connection between supervised principal components and the un-
derlying cell type model (Figure 18.13) can be seen through a latent variable
model for the data. Suppose we have a response variable Y which is related</p>
<p/>
</div>
<div class="page"><p/>
<p>18.6 High-Dimensional Regression: Supervised Principal Components 679
</p>
<p>to an underlying latent variable U by a linear model
</p>
<p>Y = β0 + β1U + ε. (18.32)
</p>
<p>In addition, we have measurements on a set of featuresXj indexed by j &isin; P
(for pathway), for which
</p>
<p>Xj = α0j + α1jU + ǫj , j &isin; P. (18.33)
</p>
<p>The errors ε and ǫj are assumed to have mean zero and are independent of
all other random variables in their respective models.
We also have many additional features Xk, k �&isin; P which are independent
</p>
<p>of U . We would like to identify P, estimate U , and hence fit the predic-
tion model (18.32). This is a special case of a latent-structure model, or
single-component factor-analysis model (Mardia et al., 1979, see also Sec-
tion 14.7). The latent factor U is a continuous version of the cell type
conceptualized in Figure 18.13.
The supervised principal component algorithm can be seen as a method
</p>
<p>for fitting this model:
</p>
<p>&bull; The screening step (1) estimates the set P.
</p>
<p>&bull; Given P̂, the largest principal component in step (2a) estimates the
latent factor U .
</p>
<p>&bull; Finally, the regression fit in step (2b) estimates the coefficient in
model (18.32).
</p>
<p>Step (1) is natural, since on average the regression coefficient is nonzero
only if α1j is non-zero. Hence this step should select the features j &isin; P.
Step (2a) is natural if we assume that the errors ǫj have a Gaussian dis-
tribution, with the same variance. In this case the principal component is
the maximum likelihood estimate for the single factor model (Mardia et
al., 1979). The regression in (2b) is an obvious final step.
Suppose there are a total of p features, with p1 features in the relevant set
</p>
<p>P. Then if p and p1 grow but p1 is small relative to p, one can show (under
reasonable conditions) that the leading supervised principal component
is consistent for the underlying latent factor. The usual leading principal
component may not be consistent, since it can be contaminated by the
presence of a large number of &ldquo;noise&rdquo; features.
Finally, suppose that the threshold used in step (1) of the supervised
</p>
<p>principal component procedure yields a large number of features for com-
putation of the principal component. Then for interpretational purposes, as
well as for practical uses, we would like some way of finding a reduced a set
of features that approximates the model. Pre-conditioning (Section 18.6.3)
is one way of doing this.</p>
<p/>
</div>
<div class="page"><p/>
<p>680 18. High-Dimensional Problems: p ≫ N
</p>
<p>18.6.2 Relationship with Partial Least Squares
</p>
<p>Supervised principal components is closely related to partial least squares
regression (Section 3.5.2). Bair et al. (2006) found that the key to the good
performance of supervised principal components was the filtering out of
noisy features in step (2a). Partial least squares (Section 3.5.2) downweights
noisy features, but does not throw them away; as a result a large number
of noisy features can contaminate the predictions. However, a modification
of the partial least squares procedure has been proposed that has a similar
flavor to supervised principal components [Brown et al. (1991),Nadler and
Coifman (2005), for example]. We select the features as in steps (1) and
(2a) of supervised principal components, but then apply PLS (rather than
principal components) to these features. For our current discussion, we call
this &ldquo;thresholded PLS.&rdquo;
Thresholded PLS can be viewed as a noisy version of supervised principal
</p>
<p>components, and hence we might not expect it to work as well in practice.
Assume the variables are all standardized. The first PLS variate has the
form
</p>
<p>z =
&sum;
</p>
<p>j&isin;P
</p>
<p>〈y,xj〉xj , (18.34)
</p>
<p>and can be thought of as an estimate of the latent factor U in model (18.33).
In contrast, the supervised principal components direction û satisfies
</p>
<p>û =
1
</p>
<p>d2
</p>
<p>&sum;
</p>
<p>j&isin;P
</p>
<p>〈û,xj〉xj , (18.35)
</p>
<p>where d is the leading singular value of XP . This follows from the definition
of the leading principal component. Hence thresholded PLS uses weights
which are the inner product of y with each of the features, while supervised
principal components uses the features to derive a &ldquo;self-consistent&rdquo; estimate
û. Since many features contribute to the estimate û, rather than just the
single outcome y, we can expect û to be less noisy than z. In fact, if there
are p1 features in the set P, and N, p and p1 go to infinity with p1/N &rarr; 0,
then it can be shown using the techniques in Bair et al. (2006) that
</p>
<p>z = u+Op(1)
</p>
<p>û = u+Op(
&radic;
p1/N), (18.36)
</p>
<p>where u is the true (unobservable) latent variable in the model (18.32),
(18.33).
We now present a simulation example to compare the methods numeri-
</p>
<p>cally. There are N = 100 samples and p = 5000 genes. We generated the
data as follows:</p>
<p/>
</div>
<div class="page"><p/>
<p>18.6 High-Dimensional Regression: Supervised Principal Components 681
</p>
<p>FIGURE 18.15. Heatmap of the outcome (left column) and first 500 genes from
a realization from model (18.37). The genes are in the columns, and the samples
are in the rows.
</p>
<p>xij =
</p>
<p>{
3 + ǫij if i &le; 50,
4 + ǫij if i &gt; 50
</p>
<p>j = 1, . . . , 50
</p>
<p>xij =
</p>
<p>{
1.5 + ǫij if 1 &le; i &le; 25 or 51 &le; i &le; 75
5.5 + ǫij if 26 &le; i &le; 50 or 76 &le; i &le; 100
</p>
<p>j = 51, . . . , 250
</p>
<p>xij = ǫij j = 251, . . . , 5000
</p>
<p>yi = 2 &middot; 150
&sum;50
</p>
<p>j=1 xij + εi
</p>
<p>(18.37)
</p>
<p>where ǫij and εi are independent normal random variables with mean 0 and
standard deviations 1 and 1.5, respectively. Thus in the first 50 genes, there
is an average difference of 1 unit between samples 1&ndash;50 and 51&ndash;100, and this
difference correlates with the outcome y. The next 200 genes have a large
average difference of 4 units between samples (1&ndash;25, 51&ndash;75) and (26&ndash;50,
76&ndash;100), but this difference is uncorrelated with the outcome. The rest of
the genes are noise. Figure 18.15 shows a heatmap of a typical realization,
with the outcome at the left, and the first 500 genes to the right.
We generated 100 simulations from this model, and summarize the test
</p>
<p>error results in Figure 18.16. The test errors of principal components and
partial least squares are shown at the right of the plot; both are badly
affected by the noisy features in the data. Supervised principal components
and thresholded PLS work best over a wide range of the number of selected
features, with the former showing consistently lower test errors.
While this example seems &ldquo;tailor-made&rdquo; for supervised principal com-
</p>
<p>ponents, its good performance seems to hold in other simulated and real
datasets (Bair et al., 2006).
</p>
<p>18.6.3 Pre-Conditioning for Feature Selection
</p>
<p>Supervised principal components can yield lower test errors than competing
methods, as shown in Figure 18.16. However, it does not always produce a
sparse model involving only a small number of features (genes). Even if the
thresholding in Step (1) of the algorithm yields a relatively small number</p>
<p/>
</div>
<div class="page"><p/>
<p>682 18. High-Dimensional Problems: p ≫ N
</p>
<p>1
.0
</p>
<p>0
1
</p>
<p>.0
5
</p>
<p>1
.1
</p>
<p>0
1
</p>
<p>.1
5
</p>
<p>1
.2
</p>
<p>0
1
</p>
<p>.2
5
</p>
<p>Number of Features
</p>
<p>R
e
</p>
<p>la
ti
v
e
</p>
<p> R
o
</p>
<p>o
t 
</p>
<p>M
e
</p>
<p>a
n
</p>
<p> S
q
</p>
<p>u
a
</p>
<p>re
 T
</p>
<p>e
s
t 
</p>
<p>E
rr
</p>
<p>o
r
</p>
<p>0 50 100 150 200 250 300 ... 5000
</p>
<p>Thresholded PLS
Supervised Principal Components
</p>
<p>FIGURE 18.16. Root mean squared test error (&plusmn; one standard error), for
supervised principal components and thresholded PLS on 100 realizations from
model (18.37). All methods use one component, and the errors are relative to
the noise standard deviation (the Bayes error is 1.0). For both methods, different
values for the filtering threshold were tried and the number of features retained
is shown on the horizontal axis. The extreme right points correspond to regular
principal components and partial least squares, using all the genes.
</p>
<p>of features, it may be that some of the omitted features have sizable inner
products with the supervised principal component (and could act as a good
surrogate). In addition, highly correlated features will tend to be chosen
together, and there may be great deal of redundancy in the set of selected
features.
The lasso (Sections 18.4 and 3.4.2), on the other hand, produces a sparse
</p>
<p>model from the data. How do the test errors of the two methods compare on
the simulated example of the last section? Figure 18.17 shows the test errors
for one realization from model (18.37) for the lasso, supervised principal
components, and the pre-conditioned lasso (described below).
We see that supervised principal components (orange curve) reaches its
</p>
<p>lowest error when about 50 features are included in the model, which is
the correct number for the simulation. Although a linear model in the first
50 features is optimal, the lasso (green) is adversely affected by the large
number of noisy features, and starts overfitting when far fewer are in the
model.
Can we get the low test error of supervised principal components along
</p>
<p>with the sparsity of the lasso? This is the goal of pre-conditioning (Paul
et al., 2008). In this approach, one first computes the supervised principal
component predictor ŷi for each observation in the training set (with the</p>
<p/>
</div>
<div class="page"><p/>
<p>18.7 Feature Assessment and the Multiple-Testing Problem 683
</p>
<p>0 50 100 150 200 250
</p>
<p>3
.0
</p>
<p>3
.2
</p>
<p>3
.4
</p>
<p>3
.6
</p>
<p>3
.8
</p>
<p>4
.0
</p>
<p>4
.2
</p>
<p>4
.4
</p>
<p>Number of Features in Model
</p>
<p>M
e
</p>
<p>a
n
</p>
<p> T
e
</p>
<p>s
t 
</p>
<p>E
rr
</p>
<p>o
r
</p>
<p>Lasso
Supervised Principal Components
Preconditioned Lasso
</p>
<p>FIGURE 18.17. Test errors for the lasso, supervised principal components,
and pre-conditioned lasso, for one realization from model (18.37). Each model is
indexed by the number of non-zero features. The supervised principal component
path is truncated at 250 features. The lasso self-truncates at 100, the sample size
(see Section 18.4). In this case, the pre-conditioned lasso achieves the lowest error
with about 25 features.
</p>
<p>threshold selected by cross-validation). Then we apply the lasso with ŷi as
the outcome variable, in place of the usual outcome yi. All features are used
in the lasso fit, not just those that were retained in the thresholding step
in supervised principal components. The idea is that by first denoising the
outcome variable, the lasso should not be as adversely affected by the large
number of noise features. Figure 18.17 shows that pre-conditioning (purple
curve) has been successful here, yielding much lower test error than the
usual lasso, and as low (in this case) as for supervised principal components.
It also can achieve this using less features. The usual lasso, applied to
the raw outcome, starts to overfit more quickly than the pre-conditioned
version. Overfitting is not a problem, since the outcome variable has been
denoised. We usually select the tuning parameter for the pre-conditioned
lasso on more subjective grounds, like parsimony.
Pre-conditioning can be applied in a variety of settings, using initial
</p>
<p>estimates other than supervised principal components and post-processors
other than the lasso. More details may be found in Paul et al. (2008).
</p>
<p>18.7 Feature Assessment and the Multiple-Testing
Problem
</p>
<p>In the first part of this chapter we discuss prediction models in the p ≫ N
setting. Here we consider the more basic problem of assessing the signif-</p>
<p/>
</div>
<div class="page"><p/>
<p>684 18. High-Dimensional Problems: p ≫ N
</p>
<p>icance of each of the p features. Consider the protein mass spectrometry
example of Section 18.4.1. In that problem, the scientist might not be inter-
ested in predicting whether a given patient has prostate cancer. Rather the
goal might be to identify proteins whose abundance differs between nor-
mal and cancer samples, in order to enhance understanding of the disease
and suggest targets for drug development. Thus our goal is to assess the
significance of individual features. This assessment is usually done without
the use of a multivariate predictive model like those in the first part of this
chapter. The feature assessment problem moves our focus from prediction
to the traditional statistical topic of multiple hypothesis testing. For the
remainder of this chapter we will use M instead of p to denote the number
of features, since we will frequently be referring to p-values.
</p>
<p>TABLE 18.4. Subset of the 12, 625 genes from microarray study of radiation
sensitivity. There are a total of 44 samples in the normal group and 14 in the
radiation sensitive group; we only show three samples from each group.
</p>
<p>Normal Radiation Sensitive
</p>
<p>Gene 1 7.85 29.74 29.50 . . . 17.20 -50.75 -18.89 . . .
Gene 2 15.44 2.70 19.37 . . . 6.57 -7.41 79.18 . . .
Gene 3 -1.79 15.52 -3.13 . . . -8.32 12.64 4.75 . . .
Gene 4 -11.74 22.35 -36.11 . . . -52.17 7.24 -2.32 . . .
...
</p>
<p>...
...
</p>
<p>...
...
</p>
<p>...
...
</p>
<p>...
...
</p>
<p>Gene 12,625 -14.09 32.77 57.78 . . . -32.84 24.09 -101.44 . . .
</p>
<p>Consider, for example, the microarray data in Table 18.4, taken from a
study on the sensitivity of cancer patients to ionizing radiation treatment
(Rieger et al., 2004). Each row consists of the expression of genes in 58
patient samples: 44 samples were from patients with a normal reaction, and
14 from patients who had a severe reaction to radiation. The measurements
were made on oligo-nucleotide microarrays. The object of the experiment
was to find genes whose expression was different in the radiation sensitive
group of patients. There are M = 12, 625 genes altogether; the table shows
the data for some of the genes and samples for illustration.
To identify informative genes, we construct a two-sample t-statistic for
</p>
<p>each gene.
</p>
<p>tj =
x̄2j &minus; x̄1j
</p>
<p>sej
, (18.38)
</p>
<p>where x̄kj =
&sum;
</p>
<p>i&isin;Cℓ
xij/Nℓ. Here Cℓ are the indices of the Nℓ samples in
</p>
<p>group ℓ, where ℓ = 1 is the normal group and ℓ = 2 is the sensitive group.
The quantity sej is the pooled within-group standard error for gene j:</p>
<p/>
</div>
<div class="page"><p/>
<p>18.7 Feature Assessment and the Multiple-Testing Problem 685
</p>
<p>&minus;4 &minus;2 0 2 4
</p>
<p>0
2
0
0
</p>
<p>4
0
0
</p>
<p>6
0
0
</p>
<p>8
0
0
</p>
<p>t&minus;statistics
</p>
<p>FIGURE 18.18. Radiation sensitivity microarray example. A histogram of the
12, 625 t-statistics comparing the radiation-sensitive versus insensitive groups.
Overlaid in blue is the histogram of the t-statistics from 1000 permutations of the
sample labels.
</p>
<p>sej = σ̂j
</p>
<p>&radic;
1
N1
</p>
<p>+ 1N2 ; σ̂
2
j =
</p>
<p>1
N1+N2&minus;2
</p>
<p>(&sum;
</p>
<p>i&isin;C1
</p>
<p>(xij &minus; x̄1j)2 +
&sum;
</p>
<p>i&isin;C2
</p>
<p>(xij &minus; x̄2j)2
)
.
</p>
<p>(18.39)
A histogram of the 12,625 t-statistics is shown in orange in Figure 18.18,
ranging in value from &minus;4.7 to 5.0. If the tj values were normally distributed
we could consider any value greater than two in absolute value to be sig-
nificantly large. This would correspond to a significance level of about 5%.
Here there are 1189 genes with |tj | &ge; 2. However with 12,625 genes we
would expect many large values to occur by chance, even if the group-
ing is unrelated to any gene. For example, if the genes were independent
(which they are surely not), the number of falsely significant genes would
have a binomial distribution with mean 12, 625 &middot; 0.05 = 631.3 and standard
deviation 24.5; the actual 1189 is way out of range.
How do we assess the results for all 12,625 genes? This is called the mul-
</p>
<p>tiple testing problem. We can start as above by computing a p-value for
each gene. This can be done using the theoretical t-distribution probabil-
ities, which assumes the features are normally distributed. An attractive
alternative approach is to use the permutation distribution, since it avoids
assumptions about the distribution of the data. We compute (in principle)
all K =
</p>
<p>(
58
14
</p>
<p>)
permutations of the sample labels, and for each permutation
</p>
<p>k compute the t-statistics tkj . Then the p-value for gene j is</p>
<p/>
</div>
<div class="page"><p/>
<p>686 18. High-Dimensional Problems: p ≫ N
</p>
<p>pj =
1
</p>
<p>K
</p>
<p>K&sum;
</p>
<p>k=1
</p>
<p>I(|tkj | &gt; |tj |). (18.40)
</p>
<p>Of course,
(
58
14
</p>
<p>)
is a large number (around 1013) and so we can&rsquo;t enumer-
</p>
<p>ate all of the possible permutations. Instead we take a random sample of
the possible permutations; here we took a random sample of K = 1000
permutations.
To exploit the fact that the genes are similar (e.g., measured on the
</p>
<p>same scale), we can instead pool the results for all genes in computing the
p-values.
</p>
<p>pj =
1
</p>
<p>MK
</p>
<p>M&sum;
</p>
<p>j&prime;=1
</p>
<p>K&sum;
</p>
<p>k=1
</p>
<p>I(|tkj&prime; | &gt; |tj |). (18.41)
</p>
<p>This also gives more granular p-values than does (18.40), since there many
more values in the pooled null distribution than there are in each individual
null distribution.
Using this set of p-values, we would like to test the hypotheses:
</p>
<p>H0j = treatment has no effect on gene j
</p>
<p>versus (18.42)
</p>
<p>H1j = treatment has an effect on gene j
</p>
<p>for all j = 1, 2, . . . ,M . We reject H0j at level α if pj &lt; α. This test has
type-I error equal to α; that is, the probability of falsely rejecting H0j is α.
Now with many tests to consider, it is not clear what we should use
</p>
<p>as an overall measure of error. Let Aj be the event that H0j is falsely
rejected; by definition Pr(Aj) = α. The family-wise error rate (FWER)
is the probability of at least one false rejection, and is a commonly used
overall measure of error. In detail, if A = &cup;Mj=1Aj is the event of at least
one false rejection, then the FWER is Pr(A). Generally Pr(A) ≫ α for
large M , and depends on the correlation between the tests. If the tests are
independent each with type-I error rate α, then the family-wise error rate
of the collection of tests is (1&minus; (1&minus; α)M ). On the other hand, if the tests
have positive dependence, that is Pr(Aj |Ak) &gt; Pr(Aj), then the FWER
will be less than (1 &minus; (1 &minus; α)M ). Positive dependence between tests often
occurs in practice, in particular in genomic studies.
One of the simplest approaches to multiple testing is the Bonferroni
</p>
<p>method. It makes each individual test more stringent, in order to make the
FWER equal to at most α: we reject H0j if pj &lt; α/M . It is easy to show
that the resulting FWER is &le; α (Exercise 18.16). The Bonferroni method
can be useful if M is relatively small, but for large M it is too conservative,
that is, it calls too few genes significant.
In our example, if we test at level say α = 0.05, then we must use the
</p>
<p>threshold 0.05/12, 625 = 3.9&times;10&minus;6. None of the 12, 625 genes had a p-value
this small.</p>
<p/>
</div>
<div class="page"><p/>
<p>18.7 Feature Assessment and the Multiple-Testing Problem 687
</p>
<p>There are variations to this approach that adjust the individual p-values
to achieve an FWER of at most α, with some approaches avoiding the
assumption of independence; see, e.g., Dudoit et al. (2002b).
</p>
<p>18.7.1 The False Discovery Rate
</p>
<p>A different approach to multiple testing does not try to control the FWER,
but focuses instead on the proportion of falsely significant genes. As we will
see, this approach has a strong practical appeal.
Table 18.5 summarizes the theoretical outcomes of M hypothesis tests.
</p>
<p>Note that the family-wise error rate is Pr(V &ge; 1). Here we instead focus
</p>
<p>TABLE 18.5. Possible outcomes from M hypothesis tests. Note that V is the
number of false-positive tests; the type-I error rate is E(V )/M0. The type-II error
rate is E(T )/M1, and the power is 1&minus; E(T )/M1.
</p>
<p>Called Called
Not Significant Significant Total
</p>
<p>H0 True U V M0
H0 False T S M1
Total M &minus;R R M
</p>
<p>on the false discovery rate
</p>
<p>FDR = E(V/R). (18.43)
</p>
<p>In the microarray setting, this is the expected proportion of genes that
are incorrectly called significant, among the R genes that are called signif-
icant. The expectation is taken over the population from which the data
are generated. Benjamini and Hochberg (1995) first proposed the notion of
false discovery rate, and gave a testing procedure (Algorithm 18.2) whose
FDR is bounded by a user-defined level α. The Benjamini&ndash;Hochberg (BH)
procedure is based on p-values; these can be obtained from an asymptotic
approximation to the test statistic (e.g., Gaussian), or a permutation dis-
tribution, as is done here.
If the hypotheses are independent, Benjamini and Hochberg (1995) show
</p>
<p>that regardless of how many null hypotheses are true and regardless of the
distribution of the p-values when the null hypothesis is false, this procedure
has the property
</p>
<p>FDR &le; M0
M
</p>
<p>α &le; α. (18.45)
</p>
<p>For illustration we chose α = 0.15. Figure 18.19 shows a plot of the or-
dered p-values p(j), and the line with slope 0.15/12625.</p>
<p/>
</div>
<div class="page"><p/>
<p>688 18. High-Dimensional Problems: p ≫ N
</p>
<p>Algorithm 18.2 Benjamini&ndash;Hochberg (BH) Method.
</p>
<p>1. Fix the false discovery rate α and let p(1) &le; p(2) &le; &middot; &middot; &middot; &le; p(M) denote
the ordered p-values
</p>
<p>2. Define
</p>
<p>L = max
{
j : p(j) &lt; α &middot;
</p>
<p>j
</p>
<p>M
</p>
<p>}
. (18.44)
</p>
<p>3. Reject all hypotheses H0j for which pj &le; p(L), the BH rejection
threshold.
</p>
<p>Genes ordered by p&minus;value
</p>
<p>p
&minus;
</p>
<p>v
a
lu
</p>
<p>e
</p>
<p>1 5 10 50 100
</p>
<p>5
*1
</p>
<p>0
^&minus;
</p>
<p>6
5
*1
</p>
<p>0
^&minus;
</p>
<p>5
5
*1
</p>
<p>0
^&minus;
</p>
<p>4
5
*1
</p>
<p>0
^&minus;
</p>
<p>3
</p>
<p>&bull;
</p>
<p>&bull;
&bull; &bull;
</p>
<p>&bull;
&bull; &bull; &bull;
</p>
<p>&bull;&bull;
&bull;
&bull;&bull;&bull;&bull;
</p>
<p>&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;
</p>
<p>&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
</p>
<p>&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
</p>
<p>&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
</p>
<p>&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
</p>
<p>&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
</p>
<p>&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
</p>
<p>FIGURE 18.19. Microarray example continued. Shown is a plot of the ordered
p-values p(j) and the line 0.15 &middot; (j/12, 625), for the Benjamini&ndash;Hochberg method.
The largest j for which the p-value p(j) falls below the line, gives the BH threshold.
Here this occurs at j = 11, indicated by the vertical line. Thus the BH method
calls significant the 11 genes (in red) with smallest p-values.</p>
<p/>
</div>
<div class="page"><p/>
<p>18.7 Feature Assessment and the Multiple-Testing Problem 689
</p>
<p>Algorithm 18.3 The Plug-in Estimate of the False Discovery Rate.
</p>
<p>1. Create K permutations of the data, producing t-statistics tkj for fea-
tures j = 1, 2, . . . ,M and permutations k = 1, 2, . . . ,K.
</p>
<p>2. For a range of values of the cut-point C, let
</p>
<p>Robs =
M&sum;
</p>
<p>j=1
</p>
<p>I(|tj | &gt; C), Ê(V ) =
1
</p>
<p>K
</p>
<p>M&sum;
</p>
<p>j=1
</p>
<p>K&sum;
</p>
<p>k=1
</p>
<p>I(|tkj | &gt; C). (18.46)
</p>
<p>3. Estimate the FDR by F̂DR = Ê(V )/Robs.
</p>
<p>Starting at the left and moving right, the BH method finds the last time
that the p-values fall below the line. This occurs at j = 11, so we reject
the 11 genes with smallest p-values. Note that the cutoff occurs at the 11th
smallest p-value, 0.00012, and the 11th largest of the values |tj | is 4.101
Thus we reject the 11 genes with |tj | &ge; 4.101.
From our brief description, it is not clear how the BH procedure works;
</p>
<p>that is, why the corresponding FDR is at most 0.15, the value used for α.
Indeed, the proof of this fact is quite complicated (Benjamini and Hochberg,
1995).
A more direct way to proceed is a plug-in approach. Rather than starting
</p>
<p>with a value for α, we fix a cut-point for our t-statistics, say the value
4.101 that appeared above. The number of observed values |tj | equal or
greater than 4.101 is 11. The total number of permutation values |tkj | equal
or greater than 4.101 is 1518, for an average of 1518/1000 = 1.518 per
</p>
<p>permutation. Thus a direct estimate of the false discovery rate is F̂DR =
1.518/11 &asymp; 14%. Note that 14% is approximately equal to the value of
α = 0.15 used above (the difference is due to discreteness). This procedure
is summarized in Algorithm 18.3. To recap:
</p>
<p>The plug-in estimate of FDR of Algorithm 18.3 is equivalent to the BH
procedure of Algorithm 18.2, using the permutation p-values (18.40).
</p>
<p>This correspondence between the BH method and the plug-in estimate is
not a coincidence. Exercise 18.17 shows that they are equivalent in general.
Note that this procedure makes no reference to p-values at all, but rather
works directly with the test statistics.
The plug-in estimate is based on the approximation
</p>
<p>E(V/R) &asymp; E(V )
E(R)
</p>
<p>, (18.47)
</p>
<p>and in general F̂DR is a consistent estimate of FDR (Storey, 2002; Storey et
</p>
<p>al., 2004). Note that the numerator Ê(V ) actually estimates (M/M0)E(V ),</p>
<p/>
</div>
<div class="page"><p/>
<p>690 18. High-Dimensional Problems: p ≫ N
</p>
<p>since the permutation distribution uses M rather M0 null hypotheses.
Hence if an estimate of M0 is available, a better estimate of FDR can be
</p>
<p>obtained from (M̂0/M) &middot; F̂DR. Exercise 18.19 shows a way to estimate M0.
The most conservative (upwardly biased) estimate of FDR uses M0 = M .
Equivalently, an estimate of M0 can be used to improve the BH method,
through relation (18.45).
The reader might be surprised that we chose a value as large as 0.15 for
</p>
<p>α, the FDR bound. We must remember that the FDR is not the same as
type-I error, for which 0.05 is the customary choice. For the scientist, the
false discovery rate is the expected proportion of false positive genes among
the list of genes that the statistician tells him are significant. Microarray
experiments with FDRs as high as 0.15 might still be useful, especially if
they are exploratory in nature.
</p>
<p>18.7.2 Asymmetric Cutpoints and the SAM Procedure
</p>
<p>In the testing methods described above, we used the absolute value of the
test statistic tj , and hence applied the same cut-points to both positive and
negative values of the statistic. In some experiments, it might happen that
most or all of the differentially expressed genes change in the positive direc-
tion (or all in the negative direction). For this situation it is advantageous
to derive separate cut-points for the two cases.
The significance analysis of microarrays (SAM) approach offers a way of
</p>
<p>doing this. The basis of the SAM method is shown in Figure 18.20. On the
vertical axis we have plotted the ordered test statistics t(1) &le; t(2) &le; &middot; &middot; &middot; &le;
t(M), while the horizontal axis shows the expected order statistics from the
</p>
<p>permutations of the data: t̃(j) = (1/K)
&sum;K
</p>
<p>k=1 t
k
(j), where t
</p>
<p>k
(1) &le; tk(2) &le; &middot; &middot; &middot; &le;
</p>
<p>tk(M) are the ordered test statistics from permutation k.
Two lines are drawn, parallel to the 45◦ line, ∆ units away. Starting at
</p>
<p>the origin and moving to the right, we find the first place that the genes
leave the band. This defines the upper cutpoint Chi and all genes beyond
that point are called significant (marked red). Similarly we find the lower
cutpoint Clow for genes in the bottom left corner. Thus each value of the
tuning parameter ∆ defines upper and lower cutpoints, and the plug-in
</p>
<p>estimate F̂DR for each of these cutpoints is estimated as before. Typically
</p>
<p>a range of values of ∆ and associated F̂DR values are computed, from which
a particular pair are chosen on subjective grounds.
The advantage of the SAM approach lies in the possible asymmetry of
</p>
<p>the cutpoints. In the example of Figure 18.20, with ∆ = 0.71 we obtain
11 significant genes; they are all in the upper right. The data points in the
bottom left never leave the band, and hence Clow = &minus;&infin;. Hence for this
value of ∆, no genes are called significant on the left (negative) side. We
do not impose symmetry on the cutpoints, as was done in Section 18.7.1,
as there is no reason to assume similar behavior at the two ends.</p>
<p/>
</div>
<div class="page"><p/>
<p>18.7 Feature Assessment and the Multiple-Testing Problem 691
</p>
<p>Expected Order Statistics
</p>
<p>t&minus;
s
ta
</p>
<p>ti
s
ti
c
</p>
<p>&minus;4 &minus;2 0 2 4
</p>
<p>&minus;
4
</p>
<p>&minus;
2
</p>
<p>0
2
</p>
<p>4
</p>
<p>&bull;
&bull;&bull;&bull;
</p>
<p>&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
</p>
<p>&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
</p>
<p>&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>&bull;&bull;&bull;&bull;&bull;
&bull;&bull;&bull;&bull;
</p>
<p>&bull;
&bull;
</p>
<p>&bull;
</p>
<p>∆
</p>
<p>Chi
</p>
<p>FIGURE 18.20. SAM plot for the radiation sensitivity microarray data. On the
vertical axis we have plotted the ordered test statistics, while the horizontal axis
shows the expected order statistics of the test statistics from permutations of the
data. Two lines are drawn, parallel to the 45◦ line, ∆ units away from it. Starting
at the origin and moving to the right, we find the first place that the genes leave
the band. This defines the upper cut-point Chi and all genes beyond that point are
called significant (marked in red). Similarly we define a lower cutpoint Clow. For
the particular value of ∆ = 0.71 in the plot, no genes are called significant in the
bottom left.</p>
<p/>
</div>
<div class="page"><p/>
<p>692 18. High-Dimensional Problems: p ≫ N
</p>
<p>There is some similarity between this approach and the asymmetry possi-
ble with likelihood-ratio tests. Suppose we have a log-likelihood ℓ0(tj) under
the null-hypothesis of no effect, and a log-likelihood ℓ(tj) under the alterna-
tive. Then a likelihood ratio test amounts to rejecting the null-hypothesis
if
</p>
<p>ℓ(tj)&minus; ℓ0(tj) &gt; ∆, (18.48)
for some ∆. Depending on the likelihoods, and particularly their relative
values, this can result in a different threshold for tj than for &minus;tj . The SAM
procedure rejects the null-hypothesis if
</p>
<p>|t(j) &minus; t̃(j)| &gt; ∆ (18.49)
</p>
<p>Again, the threshold for each t(j) depends on the corresponding value of
</p>
<p>the null value t̃(j).
</p>
<p>18.7.3 A Bayesian Interpretation of the FDR
</p>
<p>There is an interesting Bayesian view of the FDR, developed in Storey
(2002) and Efron and Tibshirani (2002). First we need to define the positive
false discovery rate (pFDR) as
</p>
<p>pFDR = E
</p>
<p>[
V
</p>
<p>R
</p>
<p>∣∣∣∣R &gt; 0
]
. (18.50)
</p>
<p>The additional term positive refers to the fact that we are only interested
in estimating an error rate where positive findings have occurred. It is
this slightly modified version of the FDR that has a clean Bayesian inter-
pretation. Note that the usual FDR [expression (18.43)] is not defined if
Pr(R = 0) &gt; 0.
</p>
<p>Let Γ be a rejection region for a single test; in the example above we used
Γ = (&minus;&infin;,&minus;4.10) &cup; (4.10,&infin;). Suppose that M identical simple hypothe-
sis tests are performed with the i.i.d. statistics t1, . . . , tM and rejection
region Γ. We define a random variable Zj which equals 0 if the jth null
hypothesis is true, and 1 otherwise. We assume that each pair (tj , Zj) are
i.i.d random variables with
</p>
<p>tj |Zj &sim; (1&minus; Zj) &middot; F0 + Zj &middot; F1 (18.51)
</p>
<p>for some distributions F0 and F1. This says that each test statistic tj comes
from one of two distributions: F0 if the null hypothesis is true, and F1
otherwise. Letting Pr(Zj = 0) = π0, marginally we have:
</p>
<p>tj &sim; π0 &middot; F0 + (1&minus; π0) &middot; F1. (18.52)
</p>
<p>Then it can be shown (Efron et al., 2001; Storey, 2002) that</p>
<p/>
</div>
<div class="page"><p/>
<p>18.8 Bibliographic Notes 693
</p>
<p>pFDR(Γ) = Pr(Zj = 0|tj &isin; Γ). (18.53)
</p>
<p>Hence under the mixture model (18.51), the pFDR is the posterior proba-
bility that the null hypothesis it true, given that test statistic falls in the
rejection region for the test; that is, given that we reject the null hypothesis
(Exercise 18.20).
The false discovery rate provides a measure of accuracy for tests based
</p>
<p>on an entire rejection region, such as |tj | &ge; 2. But if the FDR of such a test
is say 10%, then a gene with say tj = 5 will be more significant than a gene
with tj = 2. Thus it is of interest to derive a local (gene-specific) version
of the FDR. The q-value (Storey, 2003) of a test statistic tj is defined to
be the smallest FDR over all rejection regions that reject tj . That is, for
symmetric rejection regions, the q-value for tj = 2 is defined to be the
FDR for the rejection region Γ = {&minus;(&infin;,&minus;2) &cup; (2,&infin;)}. Thus the q-value
for tj = 5 will be smaller than that for tj = 2, reflecting the fact that tj = 5
is more significant than tj = 2. The local false discovery rate (Efron and
Tibshirani, 2002) at t = t0 is defined to be
</p>
<p>Pr(Zj = 0|tj = t0). (18.54)
</p>
<p>This is the (positive) FDR for an infinitesimal rejection region surrounding
the value tj = t0.
</p>
<p>18.8 Bibliographic Notes
</p>
<p>Many references were given at specific points in this chapter; we give some
additional ones here. Dudoit et al. (2002a) give an overview and compar-
ison of discrimination methods for gene expression data. Levina (2002)
does some mathematical analysis comparing diagonal LDA to full LDA, as
p,N &rarr; &infin; with p &gt; N . She shows that with reasonable assumptions diago-
nal LDA has a lower asymptotic error rate than full LDA. Tibshirani et al.
(2001a) and Tibshirani et al. (2003) proposed the nearest shrunken-centroid
classifier. Zhu and Hastie (2004) study regularized logistic regression. High-
dimensional regression and the lasso are very active areas of research, and
many references are given in Section 3.8.5. The fused lasso was proposed
by Tibshirani et al. (2005), while Zou and Hastie (2005) introduced the
elastic net. Supervised principal components is discussed in Bair and Tib-
shirani (2004) and Bair et al. (2006). For an introduction to the analysis
of censored survival data, see Kalbfleisch and Prentice (1980).
Microarray technology has led to a flurry of statistical research: see for
</p>
<p>example the books by Speed (2003), Parmigiani et al. (2003), Simon et al.
(2004), and Lee (2004).
The false discovery rate was proposed by Benjamini and Hochberg (1995),
</p>
<p>and studied and generalized in subsequent papers by these authors and</p>
<p/>
</div>
<div class="page"><p/>
<p>694 18. High-Dimensional Problems: p ≫ N
</p>
<p>many others. A partial list of papers on FDR may be found on Yoav Ben-
jamini&rsquo;s homepage. Some more recent papers include Efron and Tibshirani
(2002), Storey (2002), Genovese and Wasserman (2004), Storey and Tib-
shirani (2003) and Benjamini and Yekutieli (2005). Dudoit et al. (2002b)
review methods for identifying differentially expressed genes in microarray
studies.
</p>
<p>Exercises
</p>
<p>Ex. 18.1 For a coefficient estimate β̂j , let β̂j/||β̂j ||2 be the normalized ver-
sion. Show that as λ &rarr; &infin;, the normalized ridge-regression estimates con-
verge to the renormalized partial-least-squares one-component estimates.
</p>
<p>Ex. 18.2 Nearest shrunken centroids and the lasso. Consider a (naive Bayes)
Gaussian model for classification in which the features j = 1, 2, . . . , p are
assumed to be independent within each class k = 1, 2, . . . ,K. With ob-
servations i = 1, 2, . . . , N and Ck equal to the set of indices of the Nk
observations in class k, we observe xij &sim; N(μj + μjk, σ2j ) for i &isin; Ck with&sum;K
</p>
<p>k=1 μjk = 0. Set σ̂
2
j = s
</p>
<p>2
j , the pooled within-class variance for feature j,
</p>
<p>and consider the lasso-style minimization problem
</p>
<p>min
{μj ,μjk}
</p>
<p>⎧
⎨
⎩
</p>
<p>1
</p>
<p>2
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>K&sum;
</p>
<p>k=1
</p>
<p>&sum;
</p>
<p>i&isin;Ck
</p>
<p>(xij &minus; μj &minus; μjk)2
s2j
</p>
<p>+ λ
&radic;
Nk
</p>
<p>p&sum;
</p>
<p>j=1
</p>
<p>K&sum;
</p>
<p>k=1
</p>
<p>|μjk|
sj
</p>
<p>.
</p>
<p>⎫
⎬
⎭(18.55)
</p>
<p>Show that the solution is equivalent to the nearest shrunken centroid es-
timator (18.7), with s0 set to zero, and m
</p>
<p>2
k equal to 1/Nk instead of
</p>
<p>1/Nk &minus; 1/N as before.
</p>
<p>Ex. 18.3 Show that the fitted coefficients for the regularized multiclass
logistic regression problem (18.10) satisfy
</p>
<p>&sum;K
k=1 β̂kj = 0, j = 1, . . . , p.
</p>
<p>What about the β̂k0? Discuss issues with these constant parameters, and
how they can be resolved.
</p>
<p>Ex. 18.4 Derive the computational formula (18.15) for ridge regression.
[Hint: Use the first derivative of the penalized sum-of-squares criterion to
</p>
<p>show that if λ &gt; 0, then β̂ = XT s for some s &isin; IRN .]
</p>
<p>Ex. 18.5 Prove the theorem (18.16)&ndash;(18.17) in Section 18.3.5, by decom-
posing β and the rows of X into their projections into the column space of
V and its complement in IRp.
</p>
<p>Ex. 18.6 Show how the theorem in Section 18.3.5 can be applied to regu-
larized discriminant analysis [Equations 4.14 and (18.9)].</p>
<p/>
</div>
<div class="page"><p/>
<p>Exercises 695
</p>
<p>Ex. 18.7 Consider a linear regression problem where p ≫ N , and assume
the rank of X is N . Let the SVD of X = UDVT = RVT , where R is
N &times;N nonsingular, and V is p&times;N with orthonormal columns.
</p>
<p>(a) Show that there are infinitely many least-squares solutions all with
zero residuals.
</p>
<p>(b) Show that the ridge-regression estimate for β can be written
</p>
<p>β̂λ = V(R
TR+ λI)&minus;1RTy (18.56)
</p>
<p>(c) Show that when λ = 0, the solution β̂0 = VD
&minus;1UTy has residuals
</p>
<p>all equal to zero, and is unique in that it has the smallest Euclidean
norm amongst all zero-residual solutions.
</p>
<p>Ex. 18.8 Data Piling. Exercise 4.2 shows that the two-class LDA solution
can be obtained by a linear regression of a binary response vector y con-
sisting of &minus;1s and +1s. The prediction β̂Tx for any x is (up to a scale and
shift) the LDA score δ(x). Suppose now that p ≫ N .
</p>
<p>(a) Consider the linear regression model f(x) = α + βTx fit to a binary
response Y &isin; {&minus;1,+1}. Using Exercise 18.7, show that there are
infinitely many directions defined by β̂ in IRp onto which the data
project to exactly two points, one for each class. These are known as
data piling directions (Ahn and Marron, 2005).
</p>
<p>(b) Show that the distance between the projected points is 2/||β̂||, and
hence these directions define separating hyperplanes with that mar-
gin.
</p>
<p>(c) Argue that there is a single maximal data piling direction for which
</p>
<p>this distance is largest, and is defined by β̂0 = VD
&minus;1UTy = X&minus;y,
</p>
<p>where X = UDVT is the SVD of X.
</p>
<p>Ex. 18.9 Compare the data piling direction of Exercise 18.8 to the direction
of the optimal separating hyperplane (Section 4.5.2) qualitatively. Which
makes the widest margin, and why? Use a small simulation to demonstrate
the difference.
</p>
<p>Ex. 18.10 When p ≫ N , linear discriminant analysis (see Section 4.3) is
degenerate because the within-class covariance matrix W is singular. One
version of regularized discriminant analysis (4.14) replaces W by a ridged
version W + λI, leading to a regularized discriminant function δλ(x) =
xT (W + λI)&minus;1(x̄1 &minus; x̄&minus;1). Show that δ0(x) = limλ&darr;0 δλ(x) corresponds to
the maximal data piling direction defined in Exercise 18.8.
</p>
<p>Ex. 18.11 Suppose you have a sample of N pairs (xi, yi), with yi binary
and xi &isin; IR1. Suppose also that the two classes are separable; e.g., for each</p>
<p/>
</div>
<div class="page"><p/>
<p>696 18. High-Dimensional Problems: p ≫ N
</p>
<p>pair i, i&prime; with yi = 0 and yi&prime; = 1, xi&prime; &minus; xi &ge; C for some C &gt; 0. You wish
to fit a linear logistic regression model logitPr(Y = 1|X) = α + βX by
maximum-likelihood. Show that β̂ is undefined.
</p>
<p>Ex. 18.12 Suppose we wish to select the ridge parameter λ by 10-fold cross-
validation in a p ≫ N situation (for any linear model). We wish to use the
computational shortcuts described in Section 18.3.5. Show that we need
only to reduce the N &times; p matrix X to the N &times;N matrix R once, and can
use it in all the cross-validation runs.
</p>
<p>Ex. 18.13 Suppose our p &gt; N predictors are presented as an N &times;N inner-
product matrix K = XXT , and we wish to fit the equivalent of a linear
logistic regression model in the original features with quadratic regulariza-
tion. Our predictions are also to be made using inner products; a new x0
is presented as k0 = Xx0. Let K = UD
</p>
<p>2UT be the eigen-decomposition of
K. Show that the predictions are given by f̂0 = k
</p>
<p>T
0 α̂, where
</p>
<p>(a) α̂ = UD&minus;1β̂, and
</p>
<p>(b) β̂ is the ridged logistic regression estimate with input matrix R =
UD.
</p>
<p>Argue that the same approach can be used for any appropriate kernel
matrix K.
</p>
<p>Ex. 18.14 Distance weighted 1-NN classification. Consider the 1-nearest-
neighbor method (Section 13.3) in a two-class classification problem. Let
d+(x0) be the shortest distance to a training observation in class +1, and
likewise d&minus;(x0) the shortest distance for class &minus;1. Let N&minus; be the number
of samples in class &minus;1, N+ the number in class +1, and N = N&minus; +N+.
(a) Show that
</p>
<p>δ(x0) = log
d&minus;(x0)
</p>
<p>d+(x0)
(18.57)
</p>
<p>can be viewed as a nonparametric discriminant function correspond-
ing to 1-NN classification. [Hint: Show that f̂+(x0) =
</p>
<p>1
N+d+(x0)
</p>
<p>can
</p>
<p>be viewed as a nonparametric estimate of the density in class +1 at
x0].
</p>
<p>(b) How would you modify this function to introduce class prior probabil-
ities π+ and π&minus; different from the sample-priors N+/N and N&minus;/N?
</p>
<p>(c) How would you generalize this approach for K-NN classification?
</p>
<p>Ex. 18.15 Kernel PCA. In Section 18.5.2 we show how to compute the
principal component variables Z from an uncentered inner-product matrix
K. We compute the eigen-decomposition (I &minus; M)K(I &minus; M) = UD2UT ,
with M = 11T /N , and then Z = UD. Suppose we have the inner-product</p>
<p/>
</div>
<div class="page"><p/>
<p>Exercises 697
</p>
<p>vector k0, containing the N inner-products between a new point x0 and
each of the xi in our training set. Show that the (centered) projections of
x0 onto the principal-component directions are given by
</p>
<p>z0 = D
&minus;1UT (I&minus;M) [k0 &minus;K1/N ] . (18.58)
</p>
<p>Ex. 18.16 Bonferroni method for multiple comparisons. Suppose we are in
a multiple-testing scenario with null hypotheses H0j , j = 1, 2, . . . ,M , and
corresponding p-values pj , i = 1, 2, . . . ,M . Let A be the event that at least
one null hypothesis is falsely rejected, and let Aj be the event that the
jth null hypothesis is falsely rejected. Suppose that we use the Bonferroni
method, rejecting the jth null hypothesis if pj &lt; α/M .
</p>
<p>(a) Show that Pr(A) &le; α. [Hint: Pr(Aj &cup; Aj&prime;) = Pr(Aj) + Pr(Aj&prime;) &minus;
Pr(Aj &cap;Aj&prime;)]
</p>
<p>(b) If the hypotheses H0j , j = 1, 2, . . . ,M , are independent, then Pr(A) =
</p>
<p>1&minus;Pr(AC) = 1&minus;&prod;Mj=1 Pr(ACj ) = 1&minus; (1&minus;α/M)M . Use this to show
that Pr(A) &asymp; α in this case.
</p>
<p>Ex. 18.17 Equivalence between Benjamini&ndash;Hochberg and plug-in methods.
</p>
<p>(a) In the notation of Algorithm 18.2, show that for rejection threshold
p0 = p(L), a proportion of at most p0 of the permuted values t
</p>
<p>k
j
</p>
<p>exceed |T |(L) where |T |(L) is the Lth largest value among the |tj |.
Hence show that the plug-in FDR estimate F̂DR is less than or equal
to p0 &middot;M/L = α.
</p>
<p>(b) Show that the cut-point |T |(L+1) produces a test with estimated FDR
greater than α.
</p>
<p>Ex. 18.18 Use result (18.53) to show that
</p>
<p>pFDR =
π0 &middot; {Type I error of Γ}
</p>
<p>π0 &middot; {Type I error of Γ}+ π1{Power of Γ}
(18.59)
</p>
<p>(Storey, 2003).
</p>
<p>Ex. 18.19 Consider the data in Table 18.4 of Section (18.7), available from
the book website.
</p>
<p>(a) Using a symmetric two-sided rejection region based on the t-statistic,
compute the plug-in estimate of the FDR for various values of the
cut-point.
</p>
<p>(b) Carry out the BH procedure for various FDR levels α and show the
equivalence of your results, with those from part (a).</p>
<p/>
</div>
<div class="page"><p/>
<p>698 18. High-Dimensional Problems: p ≫ N
</p>
<p>(c) Let (q.25, q.75) be the quartiles of the t-statistics from the permuted
datasets. Let π̂0 = {#tj &isin; (q.25, q.75)}/(.5M), and set π̂0 =min(π̂0, 1).
Multiply the FDR estimates from (a) by π̂0 and examine the results.
</p>
<p>(d) Give a motivation for the estimate in part (c).
</p>
<p>(Storey, 2003)
</p>
<p>Ex. 18.20 Proof of result (18.53). Write
</p>
<p>pFDR = E
</p>
<p>(
V
</p>
<p>R
|R &gt; 0
</p>
<p>)
(18.60)
</p>
<p>=
M&sum;
</p>
<p>k=1
</p>
<p>E
</p>
<p>[
V
</p>
<p>R
|R = k
</p>
<p>]
Pr(R = k|R &gt; 0) (18.61)
</p>
<p>Use the fact that given R = k, V is a binomial random variable, with k
trials and probability of success Pr(H = 0|T &isin; Γ), to complete the proof.</p>
<p/>
</div>
<div class="page"><p/>
<p>References
</p>
<p>Abu-Mostafa, Y. (1995). Hints, Neural Computation 7: 639&ndash;671.
</p>
<p>Ackley, D. H., Hinton, G. and Sejnowski, T. (1985). A learning algorithm
for Boltzmann machines, Trends in Cognitive Sciences 9: 147&ndash;169.
</p>
<p>Adam, B.-L., Qu, Y., Davis, J. W., Ward, M. D., Clements, M. A.,
Cazares, L. H., Semmes, O. J., Schellhammer, P. F., Yasui, Y.,
Feng, Z. and Wright, G. (2003). Serum protein fingerprinting cou-
pled with a pattern-matching algorithm distinguishes prostate cancer
from benign prostate hyperplasia and healthy mean, Cancer Research
63(10): 3609&ndash;3614.
</p>
<p>Agrawal, R., Mannila, H., Srikant, R., Toivonen, H. and Verkamo, A. I.
(1995). Fast discovery of association rules, Advances in Knowledge
Discovery and Data Mining, AAAI/MIT Press, Cambridge, MA.
</p>
<p>Agresti, A. (1996). An Introduction to Categorical Data Analysis, Wiley,
New York.
</p>
<p>Agresti, A. (2002). Categorical Data Analysis (2nd Ed.), Wiley, New York.
</p>
<p>Ahn, J. and Marron, J. (2005). The direction of maximal data piling in high
dimensional space, Technical report, Statistics Department, University
of North Carolina, Chapel Hill.
</p>
<p>Akaike, H. (1973). Information theory and an extension of the maximum
likelihood principle, Second International Symposium on Information
Theory, pp. 267&ndash;281.
</p>
<p> &copy; Springer Science+Business Media, LLC 2009 
</p>
<p>T. Hastie et al., The Elements of Statistical Learning, Second Edition, 699
</p>
<p>DOI: 10.1007/b94608,</p>
<p/>
</div>
<div class="page"><p/>
<p>700 References
</p>
<p>Allen, D. (1974). The relationship between variable selection and data
augmentation and a method of prediction, Technometrics 16: 125&ndash;7.
</p>
<p>Ambroise, C. and McLachlan, G. (2002). Selection bias in gene extraction
on the basis of microarray gene-expression data, Proceedings of the
National Academy of Sciences 99: 6562&ndash;6566.
</p>
<p>Amit, Y. and Geman, D. (1997). Shape quantization and recognition with
randomized trees, Neural Computation 9: 1545&ndash;1588.
</p>
<p>Anderson, J. and Rosenfeld, E. (eds) (1988). Neurocomputing: Foundations
of Research, MIT Press, Cambridge, MA.
</p>
<p>Anderson, T. (2003). An Introduction to Multivariate Statistical Analysis,
3rd ed., Wiley, New York.
</p>
<p>Bach, F. and Jordan, M. (2002). Kernel independent component analysis,
Journal of Machine Learning Research 3: 1&ndash;48.
</p>
<p>Bair, E. and Tibshirani, R. (2004). Semi-supervised methods to predict
patient survival from gene expression data, PLOS Biology 2: 511&ndash;522.
</p>
<p>Bair, E., Hastie, T., Paul, D. and Tibshirani, R. (2006). Prediction by
supervised principal components, Journal of the American Statistical
Association 101: 119&ndash;137.
</p>
<p>Bakin, S. (1999). Adaptive regression and model selection in data mining
problems, Technical report, PhD. thesis, Australian National Univer-
sity, Canberra.
</p>
<p>Banerjee, O., Ghaoui, L. E. and d&rsquo;Aspremont, A. (2008). Model selection
through sparse maximum likelihood estimation for multivariate gaus-
sian or binary data, Journal of Machine Learning Research 9: 485&ndash;516.
</p>
<p>Barron, A. (1993). Universal approximation bounds for superpositions of a
sigmoid function, IEEE Transactions on Information Theory 39: 930&ndash;
945.
</p>
<p>Bartlett, P. and Traskin, M. (2007). Adaboost is consistent, in
B. Schölkopf, J. Platt and T. Hoffman (eds), Advances in Neural Infor-
mation Processing Systems 19, MIT Press, Cambridge, MA, pp. 105&ndash;
112.
</p>
<p>Becker, R., Cleveland, W. and Shyu, M. (1996). The visual design and con-
trol of trellis display, Journal of Computational and Graphical Statis-
tics 5: 123&ndash;155.
</p>
<p>Bell, A. and Sejnowski, T. (1995). An information-maximization approach
to blind separation and blind deconvolution, Neural Computation
7: 1129&ndash;1159.</p>
<p/>
</div>
<div class="page"><p/>
<p>References 701
</p>
<p>Bellman, R. E. (1961). Adaptive Control Processes, Princeton University
Press.
</p>
<p>Benjamini, Y. and Hochberg, Y. (1995). Controlling the false discovery
rate: a practical and powerful approach to multiple testing, Journal of
the Royal Statistical Society Series B. 85: 289&ndash;300.
</p>
<p>Benjamini, Y. and Yekutieli, Y. (2005). False discovery rate controlling
confidence intervals for selected parameters, Journal of the American
Statistical Association 100: 71&ndash;80.
</p>
<p>Bickel, P. and Levina, E. (2004). Some theory for Fisher&rsquo;s linear discrim-
inant function,&ldquo;Naive Bayes&rdquo;, and some alternatives when there are
many more variables than observations, Bernoulli 10: 989&ndash;1010.
</p>
<p>Bickel, P. J., Ritov, Y. and Tsybakov, A. (2008). Simultaneous analysis of
lasso and Dantzig selector, Annals of Statistics. to appear.
</p>
<p>Bishop, C. (1995). Neural Networks for Pattern Recognition, Clarendon
Press, Oxford.
</p>
<p>Bishop, C. (2006). Pattern Recognition and Machine Learning, Springer,
New York.
</p>
<p>Bishop, Y., Fienberg, S. and Holland, P. (1975). Discrete Multivariate
Analysis, MIT Press, Cambridge, MA.
</p>
<p>Boyd, S. and Vandenberghe, L. (2004). Convex Optimization, Cambridge
University Press.
</p>
<p>Breiman, L. (1992). The little bootstrap and other methods for dimension-
ality selection in regression: X-fixed prediction error, Journal of the
American Statistical Association 87: 738&ndash;754.
</p>
<p>Breiman, L. (1996a). Bagging predictors, Machine Learning 26: 123&ndash;140.
</p>
<p>Breiman, L. (1996b). Stacked regressions, Machine Learning 24: 51&ndash;64.
</p>
<p>Breiman, L. (1998). Arcing classifiers (with discussion), Annals of Statistics
26: 801&ndash;849.
</p>
<p>Breiman, L. (1999). Prediction games and arcing algorithms, Neural Com-
putation 11(7): 1493&ndash;1517.
</p>
<p>Breiman, L. (2001). Random forests, Machine Learning 45: 5&ndash;32.
</p>
<p>Breiman, L. and Friedman, J. (1997). Predicting multivariate responses
in multiple linear regression (with discussion), Journal of the Royal
Statistical Society Series B. 59: 3&ndash;37.</p>
<p/>
</div>
<div class="page"><p/>
<p>702 References
</p>
<p>Breiman, L. and Ihaka, R. (1984). Nonlinear discriminant analysis via
scaling and ACE, Technical report, University of California, Berkeley.
</p>
<p>Breiman, L. and Spector, P. (1992). Submodel selection and evaluation
in regression: the X-random case, International Statistical Review
60: 291&ndash;319.
</p>
<p>Breiman, L., Friedman, J., Olshen, R. and Stone, C. (1984). Classification
and Regression Trees, Wadsworth, New York.
</p>
<p>Bremaud, P. (1999). Markov Chains: Gibbs Fields, Monte Carlo Simula-
tion, and Queues, Springer, New York.
</p>
<p>Brown, P., Spiegelman, C. and Denham, M. (1991). Chemometrics and
spectral frequency selection, Transactions of the Royal Society of Lon-
don Series A. 337: 311&ndash;322.
</p>
<p>Bruce, A. and Gao, H. (1996). Applied Wavelet Analysis with S-PLUS,
Springer, New York.
</p>
<p>Bühlmann, P. and Hothorn, T. (2007). Boosting algorithms: regulariza-
tion, prediction and model fitting (with discussion), Statistical Science
22(4): 477&ndash;505.
</p>
<p>Buja, A., Hastie, T. and Tibshirani, R. (1989). Linear smoothers and
additive models (with discussion), Annals of Statistics 17: 453&ndash;555.
</p>
<p>Buja, A., Swayne, D., Littman, M., Hofmann, H. and Chen, L. (2008). Data
vizualization with multidimensional scaling, Journal of Computational
and Graphical Statistics. to appear.
</p>
<p>Bunea, F., Tsybakov, A. and Wegkamp, M. (2007). Sparsity oracle inequal-
ities for the lasso, Electronic Journal of Statistics 1: 169&ndash;194.
</p>
<p>Burges, C. (1998). A tutorial on support vector machines for pattern recog-
nition, Knowledge Discovery and Data Mining 2(2): 121&ndash;167.
</p>
<p>Butte, A., Tamayo, P., Slonim, D., Golub, T. and Kohane, I. (2000).
Discovering functional relationships between RNA expression and
chemotherapeutic susceptibility using relevance networks, Proceedings
of the National Academy of Sciences pp. 12182&ndash;12186.
</p>
<p>Candes, E. (2006). Compressive sampling, Proceedings of the Interna-
tional Congress of Mathematicians, European Mathematical Society,
Madrid, Spain.
</p>
<p>Candes, E. and Tao, T. (2007). The Dantzig selector: Statistical estimation
when p is much larger than n, Annals of Statistics 35(6): 2313&ndash;2351.</p>
<p/>
</div>
<div class="page"><p/>
<p>References 703
</p>
<p>Chambers, J. and Hastie, T. (1991). Statistical Models in S,
Wadsworth/Brooks Cole, Pacific Grove, CA.
</p>
<p>Chaudhuri, S., Drton, M. and Richardson, T. S. (2007). Estimation of a
covariance matrix with zeros, Biometrika 94(1): 1&ndash;18.
</p>
<p>Chen, L. and Buja, A. (2008). Local multidimensional scaling for nonlinear
dimension reduction, graph drawing and proximity analysis, Journal
of the American Statistical Association.
</p>
<p>Chen, S. S., Donoho, D. and Saunders, M. (1998). Atomic decomposition
by basis pursuit, SIAM Journal on Scientific Computing 20(1): 33&ndash;61.
</p>
<p>Cherkassky, V. and Ma, Y. (2003). Comparison of model selection for
regression, Neural computation 15(7): 1691&ndash;1714.
</p>
<p>Cherkassky, V. and Mulier, F. (2007). Learning from Data (2nd Edition),
Wiley, New York.
</p>
<p>Chui, C. (1992). An Introduction to Wavelets, Academic Press, London.
</p>
<p>Clifford, P. (1990). Markov random fields in statistics, in G. R. Grimmett
and D. J. A. Welsh (eds), Disorder in Physical Systems. A Volume in
Honour of John M. Hammersley, Clarendon Press, Oxford, pp. 19&ndash;32.
</p>
<p>Comon, P. (1994). Independent component analysis&mdash;a new concept?, Sig-
nal Processing 36: 287&ndash;314.
</p>
<p>Cook, D. and Swayne, D. (2007). Interactive and Dynamic Graphics for
Data Analysis; with R and GGobi, Springer, New York. With con-
tributions from A. Buja, D. Temple Lang, H. Hofmann, H. Wickham
and M. Lawrence.
</p>
<p>Cook, N. (2007). Use and misuse of the receiver operating characteristic
curve in risk prediction, Circulation 116(6): 928&ndash;35.
</p>
<p>Copas, J. B. (1983). Regression, prediction and shrinkage (with discus-
sion), Journal of the Royal Statistical Society, Series B, Methodological
45: 311&ndash;354.
</p>
<p>Cover, T. and Hart, P. (1967). Nearest neighbor pattern classification,
IEEE Transactions on Information Theory IT-11: 21&ndash;27.
</p>
<p>Cover, T. and Thomas, J. (1991). Elements of Information Theory, Wiley,
New York.
</p>
<p>Cox, D. and Hinkley, D. (1974). Theoretical Statistics, Chapman and Hall,
London.</p>
<p/>
</div>
<div class="page"><p/>
<p>704 References
</p>
<p>Cox, D. and Wermuth, N. (1996). Multivariate Dependencies: Models,
Analysis and Interpretation, Chapman and Hall, London.
</p>
<p>Cressie, N. (1993). Statistics for Spatial Data (Revised Edition), Wiley-
Interscience, New York.
</p>
<p>Csiszar, I. and Tusnády, G. (1984). Information geometry and alternat-
ing minimization procedures, Statistics &amp; Decisions Supplement Issue
1: 205&ndash;237.
</p>
<p>Cutler, A. and Breiman, L. (1994). Archetypal analysis, Technometrics
36(4): 338&ndash;347.
</p>
<p>Dasarathy, B. (1991). Nearest Neighbor Pattern Classification Techniques,
IEEE Computer Society Press, Los Alamitos, CA.
</p>
<p>Daubechies, I. (1992). Ten Lectures on Wavelets, Society for Industrial and
Applied Mathematics, Philadelphia, PA.
</p>
<p>Daubechies, I., Defrise, M. and De Mol, C. (2004). An iterative threshold-
ing algorithm for linear inverse problems with a sparsity constraint,
Communications on Pure and Applied Mathematics 57: 1413&ndash;1457.
</p>
<p>de Boor, C. (1978). A Practical Guide to Splines, Springer, New York.
</p>
<p>Dempster, A. (1972). Covariance selection, Biometrics 28: 157&ndash;175.
</p>
<p>Dempster, A., Laird, N. and Rubin, D. (1977). Maximum likelihood from
incomplete data via the EM algorithm (with discussion), Journal of
the Royal Statistical Society Series B 39: 1&ndash;38.
</p>
<p>Devijver, P. and Kittler, J. (1982). Pattern Recognition: A Statistical Ap-
proach, Prentice-Hall, Englewood Cliffs, N.J.
</p>
<p>Dietterich, T. (2000a). Ensemble methods in machine learning, Lecture
Notes in Computer Science 1857: 1&ndash;15.
</p>
<p>Dietterich, T. (2000b). An experimental comparison of three methods for
constructing ensembles of decision trees: bagging, boosting, and ran-
domization, Machine Learning 40(2): 139&ndash;157.
</p>
<p>Dietterich, T. and Bakiri, G. (1995). Solving multiclass learning problems
via error-correcting output codes, Journal of Artificial Intelligence Re-
search 2: 263&ndash;286.
</p>
<p>Donath, W. E. and Hoffman, A. J. (1973). Lower bounds for the partition-
ing of graphs, IBM Journal of Research and Development pp. 420&ndash;425.
</p>
<p>Donoho, D. (2006a). Compressed sensing, IEEE Transactions on Informa-
tion Theory 52(4): 1289&ndash;1306.</p>
<p/>
</div>
<div class="page"><p/>
<p>References 705
</p>
<p>Donoho, D. (2006b). For most large underdetermined systems of equations,
the minimal ℓ1-norm solution is the sparsest solution, Communications
on Pure and Applied Mathematics 59: 797&ndash;829.
</p>
<p>Donoho, D. and Elad, M. (2003). Optimally sparse representation from
overcomplete dictionaries via ℓ1-norm minimization, Proceedings of
the National Academy of Sciences 100: 2197&ndash;2202.
</p>
<p>Donoho, D. and Johnstone, I. (1994). Ideal spatial adaptation by wavelet
shrinkage, Biometrika 81: 425&ndash;455.
</p>
<p>Donoho, D. and Stodden, V. (2004). When does non-negative matrix
factorization give a correct decomposition into parts?, in S. Thrun,
L. Saul and B. Schölkopf (eds), Advances in Neural Information Pro-
cessing Systems 16, MIT Press, Cambridge, MA.
</p>
<p>Duan, N. and Li, K.-C. (1991). Slicing regression: a link-free regression
method, Annals of Statistics 19: 505&ndash;530.
</p>
<p>Duchamp, T. and Stuetzle, W. (1996). Extremal properties of principal
curves in the plane, Annals of Statistics 24: 1511&ndash;1520.
</p>
<p>Duda, R., Hart, P. and Stork, D. (2000). Pattern Classification (2nd Edi-
tion), Wiley, New York.
</p>
<p>Dudoit, S., Fridlyand, J. and Speed, T. (2002a). Comparison of discrimi-
nation methods for the classification of tumors using gene expression
data, Journal of the American Statistical Association 97(457): 77&ndash;87.
</p>
<p>Dudoit, S., Yang, Y., Callow, M. and Speed, T. (2002b). Statistical meth-
ods for identifying differentially expressed genes in replicated cDNA
microarray experiments, Statistica Sinica pp. 111&ndash;139.
</p>
<p>Edwards, D. (2000). Introduction to Graphical Modelling, 2nd Edition,
Springer, New York.
</p>
<p>Efron, B. (1975). The efficiency of logistic regression compared to normal
discriminant analysis, Journal of the American Statistical Association
70: 892&ndash;898.
</p>
<p>Efron, B. (1979). Bootstrap methods: another look at the jackknife, Annals
of Statistics 7: 1&ndash;26.
</p>
<p>Efron, B. (1983). Estimating the error rate of a prediction rule: some
improvements on cross-validation, Journal of the American Statistical
Association 78: 316&ndash;331.
</p>
<p>Efron, B. (1986). How biased is the apparent error rate of a prediction
rule?, Journal of the American Statistical Association 81: 461&ndash;70.</p>
<p/>
</div>
<div class="page"><p/>
<p>706 References
</p>
<p>Efron, B. and Tibshirani, R. (1991). Statistical analysis in the computer
age, Science 253: 390&ndash;395.
</p>
<p>Efron, B. and Tibshirani, R. (1993). An Introduction to the Bootstrap,
Chapman and Hall, London.
</p>
<p>Efron, B. and Tibshirani, R. (1996). Using specially designed exponential
families for density estimation, Annals of Statistics 24(6): 2431&ndash;2461.
</p>
<p>Efron, B. and Tibshirani, R. (1997). Improvements on cross-validation: the
632+ bootstrap: method, Journal of the American Statistical Associ-
ation 92: 548&ndash;560.
</p>
<p>Efron, B. and Tibshirani, R. (2002). Microarrays, empirical Bayes methods,
and false discovery rates, Genetic Epidemiology 1: 70&ndash;86.
</p>
<p>Efron, B., Hastie, T. and Tibshirani, R. (2007). Discussion of &ldquo;Dantzig
selector&rdquo; by Candes and Tao, Annals of Statistics 35(6): 2358&ndash;2364.
</p>
<p>Efron, B., Hastie, T., Johnstone, I. and Tibshirani, R. (2004). Least angle
regression (with discussion), Annals of Statistics 32(2): 407&ndash;499.
</p>
<p>Efron, B., Tibshirani, R., Storey, J. and Tusher, V. (2001). Empirical
Bayes analysis of a microarray experiment, Journal of the American
Statistical Association 96: 1151&ndash;1160.
</p>
<p>Evgeniou, T., Pontil, M. and Poggio, T. (2000). Regularization networks
and support vector machines, Advances in Computational Mathemat-
ics 13(1): 1&ndash;50.
</p>
<p>Fan, J. and Fan, Y. (2008). High dimensional classification using features
annealed independence rules, Annals of Statistics. to appear.
</p>
<p>Fan, J. and Gijbels, I. (1996). Local Polynomial Modelling and Its Appli-
cations, Chapman and Hall, London.
</p>
<p>Fan, J. and Li, R. (2005). Variable selection via nonconcave penalized
likelihood and its oracle properties, Journal of the American Statistical
Association 96: 1348&ndash;1360.
</p>
<p>Fiedler, M. (1973). Algebraic connectivity of graphs, Czechoslovak Mathe-
matics Journal 23(98): 298&ndash;305.
</p>
<p>Fienberg, S. (1977). The Analysis of Cross-Classified Categorical Data,
MIT Press, Cambridge.
</p>
<p>Fisher, R. A. (1936). The use of multiple measurements in taxonomic
problems, Eugen. 7: 179&ndash;188.</p>
<p/>
</div>
<div class="page"><p/>
<p>References 707
</p>
<p>Fisher, W. (1958). On grouping for maximum homogeniety, Journal of the
American Statistical Association 53(284): 789&ndash;798.
</p>
<p>Fix, E. and Hodges, J. (1951). Discriminatory analysis&mdash;nonparametric
discrimination: Consistency properties, Technical Report 21-49-004,4,
U.S. Air Force, School of Aviation Medicine, Randolph Field, TX.
</p>
<p>Flury, B. (1990). Principal points, Biometrika 77: 33&ndash;41.
</p>
<p>Forgy, E. (1965). Cluster analysis of multivariate data: efficiency vs. inter-
pretability of classifications, Biometrics 21: 768&ndash;769.
</p>
<p>Frank, I. and Friedman, J. (1993). A statistical view of some chemometrics
regression tools (with discussion), Technometrics 35(2): 109&ndash;148.
</p>
<p>Freund, Y. (1995). Boosting a weak learning algorithm by majority, Infor-
mation and Computation 121(2): 256&ndash;285.
</p>
<p>Freund, Y. and Schapire, R. (1996a). Experiments with a new boosting
algorithm, Machine Learning: Proceedings of the Thirteenth Interna-
tional Conference, Morgan Kauffman, San Francisco, pp. 148&ndash;156.
</p>
<p>Freund, Y. and Schapire, R. (1996b). Game theory, on-line prediction and
boosting, Proceedings of the Ninth Annual Conference on Computa-
tional Learning Theory, Desenzano del Garda, Italy, pp. 325&ndash;332.
</p>
<p>Freund, Y. and Schapire, R. (1997). A decision-theoretic generalization of
online learning and an application to boosting, Journal of Computer
and System Sciences 55: 119&ndash;139.
</p>
<p>Friedman, J. (1987). Exploratory projection pursuit, Journal of the Amer-
ican Statistical Association 82: 249&ndash;266.
</p>
<p>Friedman, J. (1989). Regularized discriminant analysis, Journal of the
American Statistical Association 84: 165&ndash;175.
</p>
<p>Friedman, J. (1991). Multivariate adaptive regression splines (with discus-
sion), Annals of Statistics 19(1): 1&ndash;141.
</p>
<p>Friedman, J. (1994a). Flexible metric nearest-neighbor classification, Tech-
nical report, Stanford University.
</p>
<p>Friedman, J. (1994b). An overview of predictive learning and function
approximation, in V. Cherkassky, J. Friedman and H. Wechsler (eds),
From Statistics to Neural Networks, Vol. 136 of NATO ISI Series F,
Springer, New York.
</p>
<p>Friedman, J. (1996). Another approach to polychotomous classification,
Technical report, Stanford University.</p>
<p/>
</div>
<div class="page"><p/>
<p>708 References
</p>
<p>Friedman, J. (1997). On bias, variance, 0-1 loss and the curse of dimen-
sionality, Journal of Data Mining and Knowledge Discovery 1: 55&ndash;77.
</p>
<p>Friedman, J. (1999). Stochastic gradient boosting, Technical report, Stan-
ford University.
</p>
<p>Friedman, J. (2001). Greedy function approximation: A gradient boosting
machine, Annals of Statistics 29(5): 1189&ndash;1232.
</p>
<p>Friedman, J. and Fisher, N. (1999). Bump hunting in high dimensional
data, Statistics and Computing 9: 123&ndash;143.
</p>
<p>Friedman, J. and Hall, P. (2007). On bagging and nonlinear estimation,
Journal of Statistical Planning and Inference 137: 669&ndash;683.
</p>
<p>Friedman, J. and Popescu, B. (2003). Importance sampled learning ensem-
bles, Technical report, Stanford University, Department of Statistics.
</p>
<p>Friedman, J. and Popescu, B. (2008). Predictive learning via rule ensem-
bles, Annals of Applied Statistics, to appear.
</p>
<p>Friedman, J. and Silverman, B. (1989). Flexible parsimonious smoothing
and additive modelling (with discussion), Technometrics 31: 3&ndash;39.
</p>
<p>Friedman, J. and Stuetzle, W. (1981). Projection pursuit regression, Jour-
nal of the American Statistical Association 76: 817&ndash;823.
</p>
<p>Friedman, J. and Tukey, J. (1974). A projection pursuit algorithm for
exploratory data analysis, IEEE Transactions on Computers, Series
C 23: 881&ndash;889.
</p>
<p>Friedman, J., Baskett, F. and Shustek, L. (1975). An algorithm for finding
nearest neighbors, IEEE Transactions on Computers 24: 1000&ndash;1006.
</p>
<p>Friedman, J., Bentley, J. and Finkel, R. (1977). An algorthm for find-
ing best matches in logarithmic expected time, ACM Transactions on
Mathematical Software 3: 209&ndash;226.
</p>
<p>Friedman, J., Hastie, T. and Tibshirani, R. (2000). Additive logistic re-
gression: a statistical view of boosting (with discussion), Annals of
Statistics 28: 337&ndash;307.
</p>
<p>Friedman, J., Hastie, T. and Tibshirani, R. (2008a). Response to &ldquo;Mease
and Wyner: Evidence contrary to the statistical view of boosting&rdquo;,
Journal of Machine Learning Research 9: 175&ndash;180.
</p>
<p>Friedman, J., Hastie, T. and Tibshirani, R. (2008b). Sparse inverse covari-
ance estimation with the graphical lasso, Biostatistics 9: 432&ndash;441.</p>
<p/>
</div>
<div class="page"><p/>
<p>References 709
</p>
<p>Friedman, J., Hastie, T. and Tibshirani, R. (2010). Regularization paths for
generalized linear models via coordinate descent, Journal of Statistical
Software 33(1): 1&ndash;22.
</p>
<p>Friedman, J., Hastie, T., Hoefling, H. and Tibshirani, R. (2007). Pathwise
coordinate optimization, Annals of Applied Statistics 2(1): 302&ndash;332.
</p>
<p>Friedman, J., Hastie, T., Rosset, S., Tibshirani, R. and Zhu, J. (2004).
Discussion of three boosting papers by Jiang, Lugosi and Vayatis, and
Zhang, Annals of Statistics 32: 102&ndash;107.
</p>
<p>Friedman, J., Stuetzle, W. and Schroeder, A. (1984). Projection pursuit
density estimation, Journal of the American Statistical Association
79: 599&ndash;608.
</p>
<p>Fu, W. (1998). Penalized regressions: the bridge vs. the lasso, Journal of
Computational and Graphical Statistics 7(3): 397&ndash;416.
</p>
<p>Furnival, G. and Wilson, R. (1974). Regression by leaps and bounds, Tech-
nometrics 16: 499&ndash;511.
</p>
<p>Gelfand, A. and Smith, A. (1990). Sampling based approaches to calculat-
ing marginal densities, Journal of the American Statistical Association
85: 398&ndash;409.
</p>
<p>Gelman, A., Carlin, J., Stern, H. and Rubin, D. (1995). Bayesian Data
Analysis, CRC Press, Boca Raton, FL.
</p>
<p>Geman, S. and Geman, D. (1984). Stochastic relaxation, Gibbs distribu-
tions and the Bayesian restoration of images, IEEE Transactions on
Pattern Analysis and Machine Intelligence 6: 721&ndash;741.
</p>
<p>Genkin, A., Lewis, D. and Madigan, D. (2007). Large-scale Bayesian logis-
tic regression for text categorization, Technometrics 49(3): 291&ndash;304.
</p>
<p>Genovese, C. and Wasserman, L. (2004). A stochastic process approach to
false discovery rates, Annals of Statistics 32(3): 1035&ndash;1061.
</p>
<p>Gersho, A. and Gray, R. (1992). Vector Quantization and Signal Compres-
sion, Kluwer Academic Publishers, Boston, MA.
</p>
<p>Girosi, F., Jones, M. and Poggio, T. (1995). Regularization theory and
neural network architectures, Neural Computation 7: 219&ndash;269.
</p>
<p>Golub, G. and Van Loan, C. (1983). Matrix Computations, Johns Hopkins
University Press, Baltimore.
</p>
<p>Golub, G., Heath, M. and Wahba, G. (1979). Generalized cross-validation
as a method for choosing a good ridge parameter, Technometrics
21: 215&ndash;224.</p>
<p/>
</div>
<div class="page"><p/>
<p>710 References
</p>
<p>Golub, T., Slonim, D., Tamayo, P., Huard, C., Gaasenbeek, M., Mesirov,
J., Coller, H., Loh, M., Downing, J., Caligiuri, M., Bloomfield, C. and
Lander, E. (1999). Molecular classification of cancer: Class discovery
and class prediction by gene expression monitoring, Science 286: 531&ndash;
536.
</p>
<p>Goodall, C. (1991). Procrustes methods in the statistical analysis of shape,
Journal of the Royal Statistical Society, Series B 53: 285&ndash;321.
</p>
<p>Gordon, A. (1999). Classification (2nd edition), Chapman and Hall/CRC
Press, London.
</p>
<p>Green, P. and Silverman, B. (1994). Nonparametric Regression and Gener-
alized Linear Models: A Roughness Penalty Approach, Chapman and
Hall, London.
</p>
<p>Greenacre, M. (1984). Theory and Applications of Correspondence Analy-
sis, Academic Press, New York.
</p>
<p>Greenshtein, E. and Ritov, Y. (2004). Persistence in high-dimensional lin-
ear predictor selection and the virtue of overparametrization, Bernoulli
10: 971&ndash;988.
</p>
<p>Guo, Y., Hastie, T. and Tibshirani, R. (2006). Regularized linear discrim-
inant analysis and its application in microarrays, Biostatistics 8: 86&ndash;
100.
</p>
<p>Guyon, I., Gunn, S., Nikravesh, M. and Zadeh, L. (eds) (2006). Feature
Extraction, Foundations and Applications, Springer, New York.
</p>
<p>Guyon, I., Weston, J., Barnhill, S. and Vapnik, V. (2002). Gene selection for
cancer classification using support vector machines, Machine Learning
46: 389&ndash;422.
</p>
<p>Hall, P. (1992). The Bootstrap and Edgeworth Expansion, Springer, New
York.
</p>
<p>Hammersley, J. M. and Clifford, P. (1971). Markov field on finite graphs
and lattices, unpublished.
</p>
<p>Hand, D. (1981). Discrimination and Classification, Wiley, Chichester.
</p>
<p>Hanley, J. and McNeil, B. (1982). The meaning and use of the area under
a receiver operating characteristic (roc) curve, Radiology 143: 29&ndash;36.
</p>
<p>Hart, P. (1968). The condensed nearest-neighbor rule, IEEE Transactions
on Information Theory 14: 515&ndash;516.
</p>
<p>Hartigan, J. A. (1975). Clustering Algorithms, Wiley, New York.</p>
<p/>
</div>
<div class="page"><p/>
<p>References 711
</p>
<p>Hartigan, J. A. and Wong, M. A. (1979). [(Algorithm AS 136] A k-means
clustering algorithm (AS R39: 81v30 p355-356), Applied Statistics
28: 100&ndash;108.
</p>
<p>Hastie, T. (1984). Principal Curves and Surfaces, PhD thesis, Stanford
University.
</p>
<p>Hastie, T. and Herman, A. (1990). An analysis of gestational age, neona-
tal size and neonatal death using nonparametric logistic regression,
Journal of Clinical Epidemiology 43: 1179&ndash;90.
</p>
<p>Hastie, T. and Simard, P. (1998). Models and metrics for handwritten digit
recognition, Statistical Science 13: 54&ndash;65.
</p>
<p>Hastie, T. and Stuetzle, W. (1989). Principal curves, Journal of the Amer-
ican Statistical Association 84(406): 502&ndash;516.
</p>
<p>Hastie, T. and Tibshirani, R. (1987). Nonparametric logistic and propor-
tional odds regression, Applied Statistics 36: 260&ndash;276.
</p>
<p>Hastie, T. and Tibshirani, R. (1990). Generalized Additive Models, Chap-
man and Hall, London.
</p>
<p>Hastie, T. and Tibshirani, R. (1996a). Discriminant adaptive nearest-
neighbor classification, IEEE Pattern Recognition and Machine In-
telligence 18: 607&ndash;616.
</p>
<p>Hastie, T. and Tibshirani, R. (1996b). Discriminant analysis by Gaussian
mixtures, Journal of the Royal Statistical Society Series B. 58: 155&ndash;
176.
</p>
<p>Hastie, T. and Tibshirani, R. (1998). Classification by pairwise coupling,
Annals of Statistics 26(2): 451&ndash;471.
</p>
<p>Hastie, T. and Tibshirani, R. (2003). Independent components analysis
through product density estimation, in S. T. S. Becker and K. Ober-
mayer (eds), Advances in Neural Information Processing Systems 15,
MIT Press, Cambridge, MA, pp. 649&ndash;656.
</p>
<p>Hastie, T. and Tibshirani, R. (2004). Efficient quadratic regularization for
expression arrays, Biostatistics 5(3): 329&ndash;340.
</p>
<p>Hastie, T. and Zhu, J. (2006). Discussion of &ldquo;Support vector machines
with applications&rdquo; by Javier Moguerza and Alberto Munoz, Statistical
Science 21(3): 352&ndash;357.
</p>
<p>Hastie, T., Botha, J. and Schnitzler, C. (1989). Regression with an ordered
categorical response, Statistics in Medicine 43: 884&ndash;889.</p>
<p/>
</div>
<div class="page"><p/>
<p>712 References
</p>
<p>Hastie, T., Buja, A. and Tibshirani, R. (1995). Penalized discriminant
analysis, Annals of Statistics 23: 73&ndash;102.
</p>
<p>Hastie, T., Kishon, E., Clark, M. and Fan, J. (1992). A model for
signature verification, Technical report, AT&amp;T Bell Laboratories.
http://www-stat.stanford.edu/&sim;hastie/Papers/signature.pdf.
</p>
<p>Hastie, T., Rosset, S., Tibshirani, R. and Zhu, J. (2004). The entire reg-
ularization path for the support vector machine, Journal of Machine
Learning Research 5: 1391&ndash;1415.
</p>
<p>Hastie, T., Taylor, J., Tibshirani, R. and Walther, G. (2007). Forward
stagewise regression and the monotone lasso, Electronic Journal of
Statistics 1: 1&ndash;29.
</p>
<p>Hastie, T., Tibshirani, R. and Buja, A. (1994). Flexible discriminant analy-
sis by optimal scoring, Journal of the American Statistical Association
89: 1255&ndash;1270.
</p>
<p>Hastie, T., Tibshirani, R. and Buja, A. (2000). Flexible discriminant and
mixture models, in J. Kay and M. Titterington (eds), Statistics and
Artificial Neural Networks, Oxford University Press.
</p>
<p>Hastie, T., Tibshirani, R. and Friedman, J. (2003). A note on &ldquo;Compari-
son of model selection for regression&rdquo; by Cherkassky and Ma, Neural
computation 15(7): 1477&ndash;1480.
</p>
<p>Hathaway, R. J. (1986). Another interpretation of the EM algorithm for
mixture distributions, Statistics &amp; Probability Letters 4: 53&ndash;56.
</p>
<p>Hebb, D. (1949). The Organization of Behavior, Wiley, New York.
</p>
<p>Hertz, J., Krogh, A. and Palmer, R. (1991). Introduction to the Theory of
Neural Computation, Addison Wesley, Redwood City, CA.
</p>
<p>Hinton, G. (1989). Connectionist learning procedures, Artificial Intelli-
gence 40: 185&ndash;234.
</p>
<p>Hinton, G. (2002). Training products of experts by minimizing contrastive
divergence, Neural Computation 14: 1771&ndash;1800.
</p>
<p>Hinton, G., Osindero, S. and Teh, Y.-W. (2006). A fast learning algorithm
for deep belief nets, Neural Computation 18: 1527&ndash;1554.
</p>
<p>Ho, T. K. (1995). Random decision forests, inM. Kavavaugh and P. Storms
(eds), Proc. Third International Conference on Document Analysis
and Recognition, Vol. 1, IEEE Computer Society Press, New York,
pp. 278&ndash;282.</p>
<p/>
</div>
<div class="page"><p/>
<p>References 713
</p>
<p>Hoefling, H. and Tibshirani, R. (2008). Estimation of sparse Markov net-
works using modified logistic regression and the lasso, submitted.
</p>
<p>Hoerl, A. E. and Kennard, R. (1970). Ridge regression: biased estimation
for nonorthogonal problems, Technometrics 12: 55&ndash;67.
</p>
<p>Hothorn, T. and Bühlmann, P. (2006). Model-based boosting in high di-
mensions, Bioinformatics 22(22): 2828&ndash;2829.
</p>
<p>Huber, P. (1964). Robust estimation of a location parameter, Annals of
Mathematical Statistics 53: 73&ndash;101.
</p>
<p>Huber, P. (1985). Projection pursuit, Annals of Statistics 13: 435&ndash;475.
</p>
<p>Hunter, D. and Lange, K. (2004). A tutorial on MM algorithms, The
American Statistician 58(1): 30&ndash;37.
</p>
<p>Hyvärinen, A. and Oja, E. (2000). Independent component analysis: algo-
rithms and applications, Neural Networks 13: 411&ndash;430.
</p>
<p>Hyvärinen, A., Karhunen, J. and Oja, E. (2001). Independent Component
Analysis, Wiley, New York.
</p>
<p>Izenman, A. (1975). Reduced-rank regression for the multivariate linear
model, Journal of Multivariate Analysis 5: 248&ndash;264.
</p>
<p>Jacobs, R., Jordan, M., Nowlan, S. and Hinton, G. (1991). Adaptive mix-
tures of local experts, Neural computation 3: 79&ndash;87.
</p>
<p>Jain, A. and Dubes, R. (1988). Algorithms for Clustering Data, Prentice-
Hall, Englewood Cliffs, N.J.
</p>
<p>James, G. and Hastie, T. (1998). The error coding method and PICTs,
Journal of Computational and Graphical Statistics 7(3): 377&ndash;387.
</p>
<p>Jancey, R. (1966). Multidimensional group analysis, Australian Journal of
Botany 14: 127&ndash;130.
</p>
<p>Jensen, F. V., Lauritzen, S. and Olesen, K. G. (1990). Bayesian updating
in recursive graphical models by local computation, Computational
Statistics Quarterly 4: 269&ndash;282.
</p>
<p>Jiang, W. (2004). Process consistency for Adaboost, Annals of Statistics
32(1): 13&ndash;29.
</p>
<p>Jirouśek, R. and Přeučil, S. (1995). On the effective implementation of the
iterative proportional fitting procedure, Computational Statistics and
Data Analysis 19: 177&ndash;189.
</p>
<p>Johnson, N. (2008). A study of the NIPS feature selection challenge, Sub-
mitted.</p>
<p/>
</div>
<div class="page"><p/>
<p>714 References
</p>
<p>Joliffe, I. T., Trendafilov, N. T. and Uddin, M. (2003). A modified principal
component technique based on the lasso, Journal of Computational
and Graphical Statistics 12: 531&ndash;547.
</p>
<p>Jones, L. (1992). A simple lemma on greedy approximation in Hilbert space
and convergence rates for projection pursuit regression and neural
network training, Annals of Statistics 20: 608&ndash;613.
</p>
<p>Jordan, M. (2004). Graphical models, Statistical Science (Special Issue on
Bayesian Statistics) 19: 140&ndash;155.
</p>
<p>Jordan, M. and Jacobs, R. (1994). Hierachical mixtures of experts and the
EM algorithm, Neural Computation 6: 181&ndash;214.
</p>
<p>Kalbfleisch, J. and Prentice, R. (1980). The Statistical Analysis of Failure
Time Data, Wiley, New York.
</p>
<p>Kaufman, L. and Rousseeuw, P. (1990). Finding Groups in Data: An In-
troduction to Cluster Analysis, Wiley, New York.
</p>
<p>Kearns, M. and Vazirani, U. (1994). An Introduction to Computational
Learning Theory, MIT Press, Cambridge, MA.
</p>
<p>Kittler, J., Hatef, M., Duin, R. and Matas, J. (1998). On combining classi-
fiers, IEEE Transaction on Pattern Analysis and Machine Intelligence
20(3): 226&ndash;239.
</p>
<p>Kleinberg, E. M. (1990). Stochastic discrimination, Annals of Mathematical
Artificial Intelligence 1: 207&ndash;239.
</p>
<p>Kleinberg, E. M. (1996). An overtraining-resistant stochastic modeling
method for pattern recognition, Annals of Statistics 24: 2319&ndash;2349.
</p>
<p>Knight, K. and Fu, W. (2000). Asymptotics for lasso-type estimators,
Annals of Statistics 28(5): 1356&ndash;1378.
</p>
<p>Koh, K., Kim, S.-J. and Boyd, S. (2007). An interior-point method
for large-scale L1-regularized logistic regression, Journal of Machine
Learning Research 8: 1519&ndash;1555.
</p>
<p>Kohavi, R. (1995). A study of cross-validation and bootstrap for accu-
racy estimation and model selection, International Joint Conference
on Artificial Intelligence (IJCAI), Morgan Kaufmann, pp. 1137&ndash;1143.
</p>
<p>Kohonen, T. (1989). Self-Organization and Associative Memory (3rd edi-
tion), Springer, Berlin.
</p>
<p>Kohonen, T. (1990). The self-organizing map, Proceedings of the IEEE
78: 1464&ndash;1479.</p>
<p/>
</div>
<div class="page"><p/>
<p>References 715
</p>
<p>Kohonen, T., Kaski, S., Lagus, K., Salojärvi, J., Paatero, A. and Saarela,
A. (2000). Self-organization of a massive document collection, IEEE
Transactions on Neural Networks 11(3): 574&ndash;585. Special Issue on
Neural Networks for Data Mining and Knowledge Discovery.
</p>
<p>Koller, D. and Friedman, N. (2007). Structured Probabilistic Models, Stan-
ford Bookstore Custom Publishing. (Unpublished Draft).
</p>
<p>Kressel, U. (1999). Pairwise classification and support vector machines,
in B. Schölkopf, C. Burges and A. Smola (eds), Advances in Ker-
nel Methods - Support Vector Learning, MIT Press, Cambridge, MA.,
pp. 255&ndash;268.
</p>
<p>Lambert, D. (1992). Zero-inflated Poisson regression, with an application
to defects in manufacturing, Technometrics 34(1): 1&ndash;14.
</p>
<p>Lange, K. (2004). Optimization, Springer, New York.
</p>
<p>Lauritzen, S. (1996). Graphical Models, Oxford University Press.
</p>
<p>Lauritzen, S. and Spiegelhalter, D. (1988). Local computations with proba-
bilities on graphical structures and their application to expert systems,
J. Royal Statistical Society B. 50: 157&ndash;224.
</p>
<p>Lawson, C. and Hansen, R. (1974). Solving Least Squares Problems,
Prentice-Hall, Englewood Cliffs, NJ.
</p>
<p>Le Cun, Y. (1989). Generalization and network design strategies, Techni-
cal Report CRG-TR-89-4, Department of Computer Science, Univ. of
Toronto.
</p>
<p>Le Cun, Y., Boser, B., Denker, J., Henderson, D., Howard, R., Hubbard,
W. and Jackel, L. (1990). Handwritten digit recognition with a back-
propogation network, in D. Touretzky (ed.), Advances in Neural In-
formation Processing Systems, Vol. 2, Morgan Kaufman, Denver, CO,
pp. 386&ndash;404.
</p>
<p>Le Cun, Y., Bottou, L., Bengio, Y. and Haffner, P. (1998). Gradient-based
learning applied to document recognition, Proceedings of the IEEE
86(11): 2278&ndash;2324.
</p>
<p>Leathwick, J., Elith, J., Francis, M., Hastie, T. and Taylor, P. (2006). Vari-
ation in demersal fish species richness in the oceans surrounding new
zealand: an analysis using boosted regression trees, Marine Ecology
Progress Series 77: 802&ndash;813.
</p>
<p>Leathwick, J., Rowe, D., Richardson, J., Elith, J. and Hastie, T. (2005).
Using multivariate adaptive regression splines to predict the distribu-
tions of New Zealand&rsquo;s freshwater diadromous fish, Freshwater Biology
50: 2034&ndash;2051.</p>
<p/>
</div>
<div class="page"><p/>
<p>716 References
</p>
<p>Leblanc, M. and Tibshirani, R. (1996). Combining estimates in regres-
sion and classification, Journal of the American Statistical Association
91: 1641&ndash;1650.
</p>
<p>LeCun, Y., Bottou, L., Bengio, Y. and Haffner, P. (1998). Gradient-based
learning applied to document recognition, Proceedings of the IEEE
86(11): 2278&ndash;2324.
</p>
<p>Lee, D. and Seung, H. (1999). Learning the parts of objects by non-negative
matrix factorization, Nature 401: 788.
</p>
<p>Lee, D. and Seung, H. (2001). Algorithms for non-negative matrix factor-
ization, Advances in Neural Information Processing Systems, (NIPS
2001), Vol. 13, Morgan Kaufman, Denver., pp. 556&ndash;562.
</p>
<p>Lee, M.-L. (2004). Analysis of Microarray Gene Expression Data, Kluwer
Academic Publishers.
</p>
<p>Lee, S.-I., Ganapathi, V. and Koller, D. (2007). Efficient structure learning
of markov networks using l1-regularization, in B. Schölkopf, J. Platt
and T. Hoffman (eds), Advances in Neural Information Processing
Systems 19, MIT Press, Cambridge, MA, pp. 817&ndash;824.
</p>
<p>Leslie, C., Eskin, E., Cohen, A., Weston, J. and Noble, W. S. (2004). Mis-
match string kernels for discriminative protein classification, Bioinfor-
matics 20(4): 467&ndash;476.
</p>
<p>Levina, E. (2002). Statistical issues in texture analysis, PhD thesis, De-
partment. of Statistics, University of California, Berkeley.
</p>
<p>Lin, H., McCulloch, C., Turnbull, B., Slate, E. and Clark, L. (2000). A
latent class mixed model for analyzing biomarker trajectories in lon-
gitudinal data with irregularly scheduled observations, Statistics in
Medicine 19: 1303&ndash;1318.
</p>
<p>Lin, Y. and Zhang, H. (2006). Component selection and smoothing in
smoothing spline analysis of variance models, Annals of Statistics
34: 2272&ndash;2297.
</p>
<p>Little, R. and Rubin, D. (2002). Statistical Analysis with Missing Data
(2nd Edition), Wiley, New York.
</p>
<p>Lloyd, S. (1957). Least squares quantization in PCM., Technical report, Bell
Laboratories. Published in 1982 in IEEE Transactions on Information
Theory 28 128-137.
</p>
<p>Loader, C. (1999). Local Regression and Likelihood, Springer, New York.</p>
<p/>
</div>
<div class="page"><p/>
<p>References 717
</p>
<p>Loh, W. and Vanichsetakul, N. (1988). Tree structured classification via
generalized discriminant analysis, Journal of the American Statistical
Association 83: 715&ndash;728.
</p>
<p>Lugosi, G. and Vayatis, N. (2004). On the bayes-risk consistency of regu-
larized boosting methods, Annals of Statistics 32(1): 30&ndash;55.
</p>
<p>Macnaughton Smith, P., Williams, W., Dale, M. and Mockett, L. (1965).
Dissimilarity analysis: a new technique of hierarchical subdivision, Na-
ture 202: 1034&ndash;1035.
</p>
<p>MacKay, D. (1992). A practical Bayesian framework for backpropagation
neural networks, Neural Computation 4: 448&ndash;472.
</p>
<p>MacQueen, J. (1967). Some methods for classification and analysis of mul-
tivariate observations, Proceedings of the Fifth Berkeley Symposium
on Mathematical Statistics and Probability, eds. L.M. LeCam and J.
Neyman, University of California Press, pp. 281&ndash;297.
</p>
<p>Madigan, D. and Raftery, A. (1994). Model selection and accounting for
model uncertainty using Occam&rsquo;s window, Journal of the American
Statistical Association 89: 1535&ndash;46.
</p>
<p>Mardia, K., Kent, J. and Bibby, J. (1979). Multivariate Analysis, Academic
Press.
</p>
<p>Mason, L., Baxter, J., Bartlett, P. and Frean, M. (2000). Boosting algo-
rithms as gradient descent, 12: 512&ndash;518.
</p>
<p>Massart, D., Plastria, F. and Kaufman, L. (1983). Non-hierarchical clus-
tering with MASLOC, The Journal of the Pattern Recognition Society
16: 507&ndash;516.
</p>
<p>McCullagh, P. and Nelder, J. (1989). Generalized Linear Models, Chapman
and Hall, London.
</p>
<p>McCulloch, W. and Pitts, W. (1943). A logical calculus of the ideas immi-
nent in nervous activity, Bulletin of Mathematical Biophysics 5: 115&ndash;
133. Reprinted in Anderson and Rosenfeld (1988), pp 96-104.
</p>
<p>McLachlan, G. (1992). Discriminant Analysis and Statistical Pattern
Recognition, Wiley, New York.
</p>
<p>Mease, D. and Wyner, A. (2008). Evidence contrary to the statistical view
of boosting (with discussion), Journal of Machine Learning Research
9: 131&ndash;156.
</p>
<p>Meinshausen, N. (2007). Relaxed lasso, Computational Statistics and Data
Analysis 52(1): 374&ndash;393.</p>
<p/>
</div>
<div class="page"><p/>
<p>718 References
</p>
<p>Meinshausen, N. and Bühlmann, P. (2006). High-dimensional graphs and
variable selection with the lasso, Annals of Statistics 34: 1436&ndash;1462.
</p>
<p>Meir, R. and Rätsch, G. (2003). An introduction to boosting and leverag-
ing, in S. Mendelson and A. Smola (eds), Lecture notes in Computer
Science, Advanced Lectures in Machine Learning, Springer, New York.
</p>
<p>Michie, D., Spiegelhalter, D. and Taylor, C. (eds) (1994). Machine Learn-
ing, Neural and Statistical Classification, Ellis Horwood Series in Ar-
tificial Intelligence, Ellis Horwood.
</p>
<p>Morgan, J. N. and Sonquist, J. A. (1963). Problems in the analysis of survey
data, and a proposal, Journal of the American Statistical Association
58: 415&ndash;434.
</p>
<p>Murray, W., Gill, P. and Wright, M. (1981). Practical Optimization, Aca-
demic Press.
</p>
<p>Myles, J. and Hand, D. (1990). The multiclass metric problem in nearest
neighbor classification, Pattern Recognition 23: 1291&ndash;1297.
</p>
<p>Nadler, B. and Coifman, R. R. (2005). An exact asymptotic formula for the
error in CLS and in PLS: The importance of dimensional reduction in
multivariate calibration, Journal of Chemometrics 102: 107&ndash;118.
</p>
<p>Neal, R. (1996). Bayesian Learning for Neural Networks, Springer, New
York.
</p>
<p>Neal, R. and Hinton, G. (1998). A view of the EM algorithm that justifies
incremental, sparse, and other variants; in Learning in Graphical Mod-
els, M. Jordan (ed.), Dordrecht: Kluwer Academic Publishers, Boston,
MA., pp. 355&ndash;368.
</p>
<p>Neal, R. and Zhang, J. (2006). High dimensional classification with
bayesian neural networks and dirichlet diffusion trees, in I. Guyon,
S. Gunn, M. Nikravesh and L. Zadeh (eds), Feature Extraction, Foun-
dations and Applications, Springer, New York, pp. 265&ndash;296.
</p>
<p>Onton, J. and Makeig, S. (2006). Information-based modeling of event-
related brain dynamics, in Neuper and Klimesch (eds), Progress in
Brain Research, Vol. 159, Elsevier, pp. 99&ndash;120.
</p>
<p>Osborne, M., Presnell, B. and Turlach, B. (2000a). A new approach to
variable selection in least squares problems, IMA Journal of Numerical
Analysis 20: 389&ndash;404.
</p>
<p>Osborne, M., Presnell, B. and Turlach, B. (2000b). On the lasso and its
dual, Journal of Computational and Graphical Statistics 9: 319&ndash;337.</p>
<p/>
</div>
<div class="page"><p/>
<p>References 719
</p>
<p>Pace, R. K. and Barry, R. (1997). Sparse spatial autoregressions, Statistics
and Probability Letters 33: 291&ndash;297.
</p>
<p>Page, L., Brin, S., Motwani, R. and Winograd, T. (1998). The
pagerank citation ranking: bringing order to the web, Tech-
nical report, Stanford Digital Library Technologies Project.
http://citeseer.ist.psu.edu/page98pagerank.html.
</p>
<p>Park, M. Y. and Hastie, T. (2007). l1-regularization path algorithm for gen-
eralized linear models, Journal of the Royal Statistical Society Series
B 69: 659&ndash;677.
</p>
<p>Parker, D. (1985). Learning logic, Technical Report TR-87, Cambridge MA:
MIT Center for Research in Computational Economics and Manage-
ment Science.
</p>
<p>Parmigiani, G., Garett, E. S., Irizarry, R. A. and Zeger, S. L. (eds) (2003).
The Analysis of Gene Expression Data, Springer, New York.
</p>
<p>Paul, D., Bair, E., Hastie, T. and Tibshirani, R. (2008). &ldquo;Pre-conditioning&rdquo;
for feature selection and regression in high-dimensional problems, An-
nals of Statistics 36(4): 1595&ndash;1618.
</p>
<p>Pearl, J. (1986). On evidential reasoning in a hierarchy of hypotheses,
Artificial Intelligence 28: 9&ndash;15.
</p>
<p>Pearl, J. (1988). Probabilistic reasoning in intelligent systems: networks of
plausible inference, Morgan Kaufmann, San Francisco, CA.
</p>
<p>Pearl, J. (2000). Causality: Models, Reasoning and Inference, Cambridge
University Press.
</p>
<p>Peterson and Anderson, J. R. (1987). A mean field theory learning algo-
rithm for neural networks, Complex Systems 1: 995&ndash;1019.
</p>
<p>Petricoin, E. F., Ardekani, A. M., Hitt, B. A., Levine, P. J., Fusaro, V.,
Steinberg, S. M., Mills, G. B., Simone, C., Fishman, D. A., Kohn,
E. and Liotta, L. A. (2002). Use of proteomic patterns in serum to
identify ovarian cancer, Lancet 359: 572&ndash;577.
</p>
<p>Platt, J. (1999). Fast Training of Support Vector Machines using Sequen-
tial Minimal Optimization; in Advances in Kernel Methods&mdash;Support
Vector Learning, B. Schölkopf and C. J. C. Burges and A. J. Smola
(eds), MIT Press, Cambridge, MA., pp. 185&ndash;208.
</p>
<p>Quinlan, R. (1993). C4.5: Programs for Machine Learning, Morgan Kauf-
mann, San Mateo.
</p>
<p>Quinlan, R. (2004). C5.0, www.rulequest.com.</p>
<p/>
</div>
<div class="page"><p/>
<p>720 References
</p>
<p>Ramaswamy, S., Tamayo, P., Rifkin, R., Mukherjee, S., Yeang, C., Angelo,
M., Ladd, C., Reich, M., Latulippe, E., Mesirov, J., Poggio, T., Gerald,
W., Loda, M., Lander, E. and Golub, T. (2001). Multiclass cancer
diagnosis using tumor gene expression signature, PNAS 98: 15149&ndash;
15154.
</p>
<p>Ramsay, J. and Silverman, B. (1997). Functional Data Analysis, Springer,
New York.
</p>
<p>Rao, C. R. (1973). Linear Statistical Inference and Its Applications, Wiley,
New York.
</p>
<p>Rätsch, G. and Warmuth, M. (2002). Maximizing the margin with boost-
ing, Proceedings of the 15th Annual Conference on Computational
Learning Theory, pp. 334&ndash;350.
</p>
<p>Ravikumar, P., Liu, H., Lafferty, J. and Wasserman, L. (2008). Spam:
Sparse additive models, in J. Platt, D. Koller, Y. Singer and S. Roweis
(eds), Advances in Neural Information Processing Systems 20, MIT
Press, Cambridge, MA, pp. 1201&ndash;1208.
</p>
<p>Ridgeway, G. (1999). The state of boosting, Computing Science and Statis-
tics 31: 172&ndash;181.
</p>
<p>Rieger, K., Hong, W., Tusher, V., Tang, J., Tibshirani, R. and Chu, G.
(2004). Toxicity from radiation therapy associated with abnormal
transcriptional responses to DNA damage, Proceedings of the National
Academy of Sciences 101: 6634&ndash;6640.
</p>
<p>Ripley, B. D. (1996). Pattern Recognition and Neural Networks, Cambridge
University Press.
</p>
<p>Rissanen, J. (1983). A universal prior for integers and estimation by mini-
mum description length, Annals of Statistics 11: 416&ndash;431.
</p>
<p>Robbins, H. and Munro, S. (1951). A stochastic approximation method,
Annals of Mathematical Statistics 22: 400&ndash;407.
</p>
<p>Roosen, C. and Hastie, T. (1994). Automatic smoothing spline projection
pursuit, Journal of Computational and Graphical Statistics 3: 235&ndash;248.
</p>
<p>Rosenblatt, F. (1958). The perceptron: a probabilistic model for infor-
mation storage and organization in the brain, Psychological Review
65: 386&ndash;408.
</p>
<p>Rosenblatt, F. (1962). Principles of Neurodynamics: Perceptrons and the
Theory of Brain Mechanisms, Spartan, Washington, D.C.</p>
<p/>
</div>
<div class="page"><p/>
<p>References 721
</p>
<p>Rosenwald, A., Wright, G., Chan, W. C., Connors, J. M., Campo, E.,
Fisher, R. I., Gascoyne, R. D., Muller-Hermelink, H. K., Smeland,
E. B. and Staudt, L. M. (2002). The use of molecular profiling to
predict survival after chemotherapy for diffuse large b-cell lymphoma,
The New England Journal of Medicine 346: 1937&ndash;1947.
</p>
<p>Rosset, S. and Zhu, J. (2007). Piecewise linear regularized solution paths,
Annals of Statistics 35(3): 1012&ndash;1030.
</p>
<p>Rosset, S., Zhu, J. and Hastie, T. (2004a). Boosting as a regularized path to
a maximum margin classifier, Journal of Machine Learning Research
5: 941&ndash;973.
</p>
<p>Rosset, S., Zhu, J. and Hastie, T. (2004b). Margin maximizing loss func-
tions, in S. Thrun, L. Saul and B. Schölkopf (eds), Advances in Neural
Information Processing Systems 16, MIT Press, Cambridge, MA.
</p>
<p>Rousseauw, J., du Plessis, J., Benade, A., Jordaan, P., Kotze, J., Jooste, P.
and Ferreira, J. (1983). Coronary risk factor screening in three rural
communities, South African Medical Journal 64: 430&ndash;436.
</p>
<p>Roweis, S. T. and Saul, L. K. (2000). Locally linear embedding, Science
290: 2323&ndash;2326.
</p>
<p>Rumelhart, D., Hinton, G. and Williams, R. (1986). Learning internal rep-
resentations by error propagation, in D. Rumelhart and J. McClelland
(eds), Parallel Distributed Processing: Explorations in the Microstruc-
ture of Cognition, The MIT Press, Cambridge, MA., pp. 318&ndash;362.
</p>
<p>Sachs, K., Perez, O., Pe&rsquo;er, D., Lauffenburger, D. and Nolan, G. (2005).
Causal protein-signaling networks derived from multiparameter single-
cell data, Science 308: 523&ndash;529.
</p>
<p>Schapire, R. (1990). The strength of weak learnability, Machine Learning
5(2): 197&ndash;227.
</p>
<p>Schapire, R. (2002). The boosting approach to machine learning: an
overview, in D. Denison, M. Hansen, C. Holmes, B. Mallick and B. Yu
(eds), MSRI workshop on Nonlinear Estimation and Classification,
Springer, New York.
</p>
<p>Schapire, R. and Singer, Y. (1999). Improved boosting algorithms using
confidence-rated predictions, Machine Learning 37(3): 297&ndash;336.
</p>
<p>Schapire, R., Freund, Y., Bartlett, P. and Lee, W. (1998). Boosting the
margin: a new explanation for the effectiveness of voting methods,
Annals of Statistics 26(5): 1651&ndash;1686.</p>
<p/>
</div>
<div class="page"><p/>
<p>722 References
</p>
<p>Schölkopf, B., Smola, A. and Müller, K.-R. (1999). Kernel principal compo-
nent analysis, in B. Schölkopf, C. Burges and A. Smola (eds), Advances
in Kernel Methods&mdash;Support Vector Learning, MIT Press, Cambridge,
MA, USA, pp. 327&ndash;352.
</p>
<p>Schwarz, G. (1978). Estimating the dimension of a model, Annals of Statis-
tics 6(2): 461&ndash;464.
</p>
<p>Scott, D. (1992). Multivariate Density Estimation: Theory, Practice, and
Visualization, Wiley, New York.
</p>
<p>Seber, G. (1984). Multivariate Observations, Wiley, New York.
</p>
<p>Segal, M. (2004). Machine learning benchmarks and random forest regres-
sion, Technical report, eScholarship Repository, University of Califor-
nia. http://repositories.edlib.org/cbmb/bench rf regn.
</p>
<p>Shao, J. (1996). Bootstrap model selection, Journal of the American Sta-
tistical Association 91: 655&ndash;665.
</p>
<p>Shenoy, P. and Shafer, G. (1988). An axiomatic framework for Bayesian
and belief-function propagation, AAAI Workshop on Uncertainty in
AI, North-Holland, pp. 307&ndash;314.
</p>
<p>Short, R. and Fukunaga, K. (1981). The optimal distance measure for near-
est neighbor classification, IEEE Transactions on Information Theory
27: 622&ndash;627.
</p>
<p>Silverman, B. (1986). Density Estimation for Statistics and Data Analysis,
Chapman and Hall, London.
</p>
<p>Silvey, S. (1975). Statistical Inference, Chapman and Hall, London.
</p>
<p>Simard, P., Cun, Y. L. and Denker, J. (1993). Efficient pattern recognition
using a new transformation distance, Advances in Neural Information
Processing Systems, Morgan Kaufman, San Mateo, CA, pp. 50&ndash;58.
</p>
<p>Simon, R. M., Korn, E. L., McShane, L. M., Radmacher, M. D., Wright,
G. and Zhao, Y. (2004). Design and Analysis of DNA Microarray
Investigations, Springer, New York.
</p>
<p>Sjöstrand, K., Rostrup, E., Ryberg, C., Larsen, R., Studholme, C., Baezner,
H., Ferro, J., Fazekas, F., Pantoni, L., Inzitari, D. and Waldemar,
G. (2007). Sparse decomposition and modeling of anatomical shape
variation, IEEE Transactions on Medical Imaging 26(12): 1625&ndash;1635.
</p>
<p>Speed, T. and Kiiveri, H. T. (1986). Gaussian Markov distributions over
finite graphs, Annals of Statistics 14: 138&ndash;150.</p>
<p/>
</div>
<div class="page"><p/>
<p>References 723
</p>
<p>Speed, T. (ed.) (2003). Statistical Analysis of Gene Expression Microarray
Data, Chapman and Hall, London.
</p>
<p>Spiegelhalter, D., Best, N., Gilks, W. and Inskip, H. (1996). Hepatitis
B: a case study in MCMC methods, in W. Gilks, S. Richardson and
D. Spegelhalter (eds), Markov Chain Monte Carlo in Practice, Inter-
disciplinary Statistics, Chapman and Hall, London, pp. 21&ndash;43.
</p>
<p>Spielman, D. A. and Teng, S.-H. (1996). Spectral partitioning works: Pla-
nar graphs and finite element meshes, IEEE Symposium on Founda-
tions of Computer Science, pp. 96&ndash;105.
</p>
<p>Stamey, T., Kabalin, J., McNeal, J., Johnstone, I., Freiha, F., Redwine, E.
and Yang, N. (1989). Prostate specific antigen in the diagnosis and
treatment of adenocarcinoma of the prostate II radical prostatectomy
treated patients, Journal of Urology 16: 1076&ndash;1083.
</p>
<p>Stone, C., Hansen, M., Kooperberg, C. and Truong, Y. (1997). Polynomial
splines and their tensor products (with discussion), Annals of Statistics
25(4): 1371&ndash;1470.
</p>
<p>Stone, M. (1974). Cross-validatory choice and assessment of statistical
predictions, Journal of the Royal Statistical Society Series B 36: 111&ndash;
147.
</p>
<p>Stone, M. (1977). An asymptotic equivalence of choice of model by cross-
validation and Akaike&rsquo;s criterion, Journal of the Royal Statistical So-
ciety Series B. 39: 44&ndash;7.
</p>
<p>Stone, M. and Brooks, R. J. (1990). Continuum regression: cross-validated
sequentially constructed prediction embracing ordinary least squares,
partial least squares and principal components regression (Corr: V54
p906-907), Journal of the Royal Statistical Society, Series B 52: 237&ndash;
269.
</p>
<p>Storey, J. (2002). A direct approach to false discovery rates, Journal of the
Royal Statistical Society B. 64(3): 479&ndash;498.
</p>
<p>Storey, J. (2003). The positive false discovery rate: A Bayesian interpreta-
tion and the q-value, Annals of Statistics 31: 2013&ndash;2025.
</p>
<p>Storey, J. and Tibshirani, R. (2003). Statistical significance for genomewide
studies, Proceedings of the National Academy of Sciences 100-: 9440&ndash;
9445.
</p>
<p>Storey, J., Taylor, J. and Siegmund, D. (2004). Strong control, conservative
point estimation, and simultaneous conservative consistency of false
discovery rates: A unified approach., Journal of the Royal Statistical
Society, Series B 66: 187&ndash;205.</p>
<p/>
</div>
<div class="page"><p/>
<p>724 References
</p>
<p>Surowiecki, J. (2004). The Wisdom of Crowds: Why the Many are Smarter
than the Few and How Collective Wisdom Shapes Business, Eco-
nomics, Societies and Nations., Little, Brown.
</p>
<p>Swayne, D., Cook, D. and Buja, A. (1991). Xgobi: Interactive dynamic
graphics in the X window system with a link to S, ASA Proceedings
of Section on Statistical Graphics, pp. 1&ndash;8.
</p>
<p>Tanner, M. and Wong, W. (1987). The calculation of posterior distribu-
tions by data augmentation (with discussion), Journal of the American
Statistical Association 82: 528&ndash;550.
</p>
<p>Tarpey, T. and Flury, B. (1996). Self-consistency: A fundamental concept
in statistics, Statistical Science 11: 229&ndash;243.
</p>
<p>Tenenbaum, J. B., de Silva, V. and Langford, J. C. (2000). A global
geometric framework for nonlinear dimensionality reduction, Science
290: 2319&ndash;2323.
</p>
<p>Tibshirani, R. (1996). Regression shrinkage and selection via the lasso,
Journal of the Royal Statistical Society, Series B 58: 267&ndash;288.
</p>
<p>Tibshirani, R. and Hastie, T. (2007). Margin trees for high-dimensional
classification, Journal of Machine Learning Research 8: 637&ndash;652.
</p>
<p>Tibshirani, R. and Knight, K. (1999). Model search and inference by boot-
strap &ldquo;bumping, Journal of Computational and Graphical Statistics
8: 671&ndash;686.
</p>
<p>Tibshirani, R. and Wang, P. (2007). Spatial smoothing and hot spot de-
tection for CGH data using the fused lasso, Biostatistics 9: 18&ndash;29.
</p>
<p>Tibshirani, R., Hastie, T., Narasimhan, B. and Chu, G. (2001a). Diagnosis
of multiple cancer types by shrunken centroids of gene expression,
Proceedings of the National Academy of Sciences 99: 6567&ndash;6572.
</p>
<p>Tibshirani, R., Hastie, T., Narasimhan, B. and Chu, G. (2003). Class
prediction by nearest shrunken centroids, with applications to DNA
microarrays, Statistical Science 18(1): 104&ndash;117.
</p>
<p>Tibshirani, R., Saunders, M., Rosset, S., Zhu, J. and Knight, K. (2005).
Sparsity and smoothness via the fused lasso, Journal of the Royal
Statistical Society, Series B 67: 91&ndash;108.
</p>
<p>Tibshirani, R., Walther, G. and Hastie, T. (2001b). Estimating the number
of clusters in a dataset via the gap statistic, Journal of the Royal
Statistical Society, Series B. 32(2): 411&ndash;423.
</p>
<p>Tropp, J. (2004). Greed is good: algorithmic results for sparse approxima-
tion, IEEE Transactions on Information Theory 50: 2231&ndash; 2242.</p>
<p/>
</div>
<div class="page"><p/>
<p>References 725
</p>
<p>Tropp, J. (2006). Just relax: convex programming methods for identify-
ing sparse signals in noise, IEEE Transactions on Information Theory
52: 1030&ndash;1051.
</p>
<p>Valiant, L. G. (1984). A theory of the learnable, Communications of the
ACM 27: 1134&ndash;1142.
</p>
<p>van der Merwe, A. and Zidek, J. (1980). Multivariate regression analysis
and canonical variates, The Canadian Journal of Statistics 8: 27&ndash;39.
</p>
<p>Vapnik, V. (1996). The Nature of Statistical Learning Theory, Springer,
New York.
</p>
<p>Vapnik, V. (1998). Statistical Learning Theory, Wiley, New York.
</p>
<p>Vidakovic, B. (1999). Statistical Modeling by Wavelets, Wiley, New York.
</p>
<p>von Luxburg, U. (2007). A tutorial on spectral clustering, Statistics and
Computing 17(4): 395&ndash;416.
</p>
<p>Wahba, G. (1980). Spline bases, regularization, and generalized cross-
validation for solving approximation problems with large quantities
of noisy data, Proceedings of the International Conference on Approx-
imation theory in Honour of George Lorenz, Academic Press, Austin,
Texas, pp. 905&ndash;912.
</p>
<p>Wahba, G. (1990). Spline Models for Observational Data, SIAM, Philadel-
phia.
</p>
<p>Wahba, G., Lin, Y. and Zhang, H. (2000). GACV for support vector ma-
chines, in A. Smola, P. Bartlett, B. Schölkopf and D. Schuurmans
(eds), Advances in Large Margin Classifiers, MIT Press, Cambridge,
MA., pp. 297&ndash;311.
</p>
<p>Wainwright, M. (2006). Sharp thresholds for noisy and high-dimensional
recovery of sparsity using ℓ1-constrained quadratic programming,
Technical report, Department of Statistics, University of California,
Berkeley.
</p>
<p>Wainwright, M. J., Ravikumar, P. and Lafferty, J. D. (2007). High-
dimensional graphical model selection using ℓ1-regularized logistic re-
gression, in B. Schölkopf, J. Platt and T. Hoffman (eds), Advances
in Neural Information Processing Systems 19, MIT Press, Cambridge,
MA, pp. 1465&ndash;1472.
</p>
<p>Wasserman, L. (2004). All of Statistics: a Concise Course in Statistical
Inference, Springer, New York.
</p>
<p>Weisberg, S. (1980). Applied Linear Regression, Wiley, New York.</p>
<p/>
</div>
<div class="page"><p/>
<p>726 References
</p>
<p>Werbos, P. (1974). Beyond Regression, PhD thesis, Harvard University.
</p>
<p>Weston, J. and Watkins, C. (1999). Multiclass support vector machines, in
M. Verleysen (ed.), Proceedings of ESANN99, D. Facto Press, Brussels.
</p>
<p>Whittaker, J. (1990). Graphical Models in Applied Multivariate Statistics,
Wiley, Chichester.
</p>
<p>Wickerhauser, M. (1994). Adapted Wavelet Analysis from Theory to Soft-
ware, A.K. Peters Ltd, Natick, MA.
</p>
<p>Widrow, B. and Hoff, M. (1960). Adaptive switching circuits, IRE
WESCON Convention record, Vol. 4. pp 96-104; Reprinted in An-
dersen and Rosenfeld (1988).
</p>
<p>Wold, H. (1975). Soft modelling by latent variables: the nonlinear iterative
partial least squares (NIPALS) approach, Perspectives in Probability
and Statistics, In Honor of M. S. Bartlett, pp. 117&ndash;144.
</p>
<p>Wolpert, D. (1992). Stacked generalization, Neural Networks 5: 241&ndash;259.
</p>
<p>Wu, T. and Lange, K. (2007). The MM alternative to EM, unpublished.
</p>
<p>Wu, T. and Lange, K. (2008). Coordinate descent procedures for lasso
penalized regression, Annals of Applied Statistics 2(1): 224&ndash;244.
</p>
<p>Yee, T. and Wild, C. (1996). Vector generalized additive models, Journal
of the Royal Statistical Society, Series B. 58: 481&ndash;493.
</p>
<p>Yuan, M. and Lin, Y. (2007). Model selection and estimation in regression
with grouped variables, Journal of the Royal Statistical Society, Series
B 68(1): 49&ndash;67.
</p>
<p>Zhang, P. (1993). Model selection via multifold cross-validation, Annals of
Statistics 21: 299&ndash;311.
</p>
<p>Zhang, T. and Yu, B. (2005). Boosting with early stopping: convergence
and consistency, Annals of Statistics 33: 1538&ndash;1579.
</p>
<p>Zhao, P. and Yu, B. (2006). On model selection consistency of lasso, Jour-
nal of Machine Learning Research 7: 2541&ndash;2563.
</p>
<p>Zhao, P., Rocha, G. and Yu, B. (2008). The composite absolute penalties
for grouped and hierarchichal variable selection, Annals of Statistics.
(to appear).
</p>
<p>Zhu, J. and Hastie, T. (2004). Classification of gene microarrays by penal-
ized logistic regression, Biostatistics 5(2): 427&ndash;443.
</p>
<p>Zhu, J., Zou, H., Rosset, S. and Hastie, T. (2005). Multiclass adaboost,
Unpublished.</p>
<p/>
</div>
<div class="page"><p/>
<p>References 727
</p>
<p>Zou, H. (2006). The adaptive lasso and its oracle properties, Journal of
the American Statistical Association 101: 1418&ndash;1429.
</p>
<p>Zou, H. and Hastie, T. (2005). Regularization and variable selection via
the elastic net, Journal of the Royal Statistical Society Series B.
67(2): 301&ndash;320.
</p>
<p>Zou, H., Hastie, T. and Tibshirani, R. (2006). Sparse principal com-
ponent analysis, Journal of Computational and Graphical Statistics
15(2): 265&ndash;28.
</p>
<p>Zou, H., Hastie, T. and Tibshirani, R. (2007). On the degrees of freedom
of the lasso, Annals of Statistics 35(5): 2173&ndash;2192.</p>
<p/>
</div>
<div class="page"><p/>
<p>Author Index
</p>
<p>Abu-Mostafa, Y. 95, 474
Ackley, D. H. 645
Adam, B.-L. 664
Agrawal, R. 489&ndash;491, 578
Agresti, A. 385, 638, 640
Ahn, J. 695
Akaike, H. 257
Allen, D. 257
Ambroise, C. 247
Amit, Y. 602
Anderson, J. R. 641
Anderson, T. 645
Angelo, M. 654, 658
Ardekani, A. M. 664
</p>
<p>Bach, F. 569
Baezner, H. 551
Bair, E. 676, 679&ndash;683, 693
Bakin, S. 90
Bakiri, G. 605, 606
Banerjee, O. 636
Barnhill, S. 658
Barron, A. 415
Barry, R. 371
Bartlett, P. 384, 615
</p>
<p>Baskett, F. 480
Baxter, J. 384
Becker, R. 369
Bell, A. 578
Bellman, R. E. 22
Benade, A. 122
Bengio, Y. 404, 407, 408, 414, 644
Benjamini, Y. 687, 689, 693
Bentley, J. 480
Best, N. 292
Bibby, J. 94, 135, 441, 539, 559,
</p>
<p>578, 630, 679
Bickel, P. 652
Bickel, P. J. 89
Bishop, C. 38, 233, 414, 623, 645
Bishop, Y. 629, 638
Bloomfield, C. 663
Boser, B. 404, 414
Botha, J. 334
Bottou, L. 404, 407, 408, 414, 644
Boyd, S. 125, 632
Breiman, L. 85, 243, 251, 257, 292,
</p>
<p>308, 310, 334, 339, 367,
384, 451, 453, 455, 554,
587, 602
</p>
<p> &copy; Springer Science+Business Media, LLC 2009 
</p>
<p>T. Hastie et al., The Elements of Statistical Learning, Second Edition, 
</p>
<p>DOI: 10.1007/b94608,
</p>
<p>729</p>
<p/>
</div>
<div class="page"><p/>
<p>730 Author Index
</p>
<p>Bremaud, P. 577
Brin, S. 577
Brooks, R. J. 81
Brown, P. 679
Bruce, A. 181
Bühlmann, P. 87, 361, 384
Buja, A. 110, 297, 441, 446, 451,
</p>
<p>455, 565, 574, 576, 578
Bunea, F. 91
Burges, C. 455
Butte, A. 631
</p>
<p>Caligiuri, M. 663
Callow, M. 686, 693
Campo, E. 674
Candes, E. 86, 89, 613
Carlin, J. 292
Cazares, L. H. 664
Chambers, J. 334
Chan, W. C. 674
Chaudhuri, S. 631, 633
Chen, L. 574, 576, 578
Chen, S. S. 68, 94
Cherkassky, V. 38, 239, 257
Chu, G. 684, 693
Chui, C. 181
Clark, L. 331
Clark, M. 539
Clements, M. A. 664
Cleveland, W. 369
Clifford, P. 629
Cohen, A. 668, 669
Coifman, R. R. 679
Coller, H. 663
Comon, P. 578
Connors, J. M. 674
Cook, D. 565, 578
Cook, N. 317
Copas, J. B. 94, 610
Cover, T. 257, 465, 481
Cox, D. 292, 645
Cressie, N. 171
Csiszar, I. 292
Cun, Y. L. 407, 471, 481
Cutler, A. 554
</p>
<p>Dale, M. 526
Dasarathy, B. 480, 481
d&rsquo;Aspremont, A. 636
Daubechies, I. 92, 181
Davis, J. W. 664
de Boor, C. 181
De Mol, C. 92
de Silva, V. 573
Defrise, M. 92
Dempster, A. 292, 449, 633
Denham, M. 679
Denker, J. 404, 407, 414, 471, 481
Devijver, P. 480
Dietterich, T. 286, 602, 605, 606,
</p>
<p>623
Donath, W. E. 578
Donoho, D. 68, 86, 91, 94, 179,
</p>
<p>181, 554, 613
Downing, J. 663
Drton, M. 631, 633
du Plessis, J. 122
Duan, N. 480
Dubes, R. 508, 522
Duchamp, T. 541
Duda, R. 38, 135
Dudoit, S. 686, 693
Duin, R. 624
</p>
<p>Edwards, D. 645
Efron, B. 73, 86, 90, 94, 97, 98,
</p>
<p>128, 231, 254, 257, 292,
334, 568, 609, 692, 693
</p>
<p>Elad, M. 613
Elith, J. 375, 376, 378
Eskin, E. 668, 669
Evgeniou, T. 168, 181, 455
</p>
<p>Fan, J. 92, 216, 539, 654
Fan, Y. 654
Fazekas, F. 551
Feng, Z. 664
Ferreira, J. 122
Ferro, J. 551
Fiedler, M. 578
Fienberg, S. 585, 629, 638</p>
<p/>
</div>
<div class="page"><p/>
<p>Author Index 731
</p>
<p>Finkel, R. 480
Fisher, N. 334
Fisher, R. A. 136, 455
Fisher, R. I. 674
Fisher, W. 310
Fishman, D. A. 664
Fix, E. 481
Flury, B. 578
Forgy, E. 578
Francis, M. 375, 376, 378
Frank, I. 81, 82, 94
Frean, M. 384
Freiha, F. 3, 49
Freund, Y. 337, 383, 384, 615
Fridlyand, J. 693
Friedman, J. 38, 81, 82, 85, 92&ndash;94,
</p>
<p>111, 121, 126, 251, 257,
258, 308, 310, 334, 339,
345, 365, 367, 384, 391,
414, 437, 451, 453, 475,
480, 565, 578, 602, 611,
617&ndash;621, 623, 636, 657,
661, 667
</p>
<p>Friedman, N. 629, 630, 645
Fu, W. 91, 92
Fukunaga, K. 475
Furnival, G. 57
Fusaro, V. 664
</p>
<p>Gaasenbeek, M. 663
Ganapathi, V. 642
Gao, H. 181
Gascoyne, R. D. 674
Gelfand, A. 292
Gelman, A. 292
Geman, D. 292, 602
Geman, S. 292
Genkin, A. 661
Genovese, C. 693
Gerald, W. 654, 658
Gersho, A. 514, 515, 526, 578
Ghaoui, L. E. 636
Gijbels, I. 216
Gilks, W. 292
Gill, P. 96, 421
</p>
<p>Girosi, F. 168, 174, 181, 415
Golub, G. 257, 335, 535
Golub, T. 631, 654, 658, 663
Goodall, C. 578
Gordon, A. 578
Gray, R. 514, 515, 526, 578
Green, P. 181, 183, 334
Greenacre, M. 455
Greenshtein, E. 91
Guo, Y. 657
Guyon, I. 658
</p>
<p>Haffner, P. 404, 407, 408, 414, 644
Hall, P. 292, 602, 619
Hammersley, J. M. 629
Hand, D. 135, 475
Hanley, J. 317
Hansen, M. 328
Hansen, R. 93
Hart, P. 38, 135, 465, 480, 481
Hartigan, J. A. 510, 578
Hastie, T. 72, 73, 78, 86, 88, 90,
</p>
<p>92&ndash;94, 97, 98, 110, 121,
122, 126, 137, 174, 216,
257, 297, 299, 304, 334,
339, 345, 348, 349, 375,
376, 378, 384, 385, 414,
428, 431, 434, 437, 441,
446, 451, 455, 475, 478,
480, 481, 519, 539, 550,
565, 568, 578, 606, 609&ndash;
611, 614, 615, 636, 657,
658, 660&ndash;662, 664, 667,
676, 679&ndash;683, 693
</p>
<p>Hatef, M. 624
Hathaway, R. J. 292
Heath, M. 257
Hebb, D. 414
Henderson, D. 404, 414
Herman, A. 334
Hertz, J. 414
Hinkley, D. 292
Hinton, G. 292, 334, 408, 414, 644,
</p>
<p>645
Hitt, B. A. 664</p>
<p/>
</div>
<div class="page"><p/>
<p>732 Author Index
</p>
<p>Ho, T. K. 602
Hochberg, Y. 687, 689, 693
Hodges, J. 481
Hoefling, H. 92, 93, 642, 667
Hoerl, A. E. 64, 94
Hoff, M. 396, 414
Hoffman, A. J. 578
Hofmann, H. 578
Holland, P. 629, 638
Hong, W. 684
Hothorn, T. 87, 361, 384
Howard, R. 404, 414
Huard, C. 663
Hubbard, W. 404, 414
Huber, P. 349, 414, 435, 565, 578
Hunter, D. 294
Hyvärinen, A. 560, 562, 578, 583
</p>
<p>Ihaka, R. 455
Inskip, H. 292
Inzitari, D. 551
Izenman, A. 84
</p>
<p>Jackel, L. 404, 414
Jacobs, R. 334
Jain, A. 508, 522
James, G. 606
Jancey, R. 578
Jensen, F. V. 629
Jiang, W. 384
Jirouśek, R. 640
Johnson, N. 412
Johnstone, I. 3, 49, 73, 86, 94, 97,
</p>
<p>98, 179, 181, 609, 613
Joliffe, I. T. 550
Jones, L. 415
Jones, M. 168, 174, 181, 415
Jooste, P. 122
Jordaan, P. 122
Jordan, M. 334, 569, 645
</p>
<p>Kabalin, J. 3, 49
Kalbfleisch, J. 674, 693
Karhunen, J. 583
Kaski, S. 531, 532, 578
</p>
<p>Kaufman, L. 517, 526, 578
Kearns, M. 380
Kennard, R. 64, 94
Kent, J. 94, 135, 441, 539, 559,
</p>
<p>578, 630, 679
Kiiveri, H. T. 632
Kim, S.-J. 125
Kishon, E. 539
Kittler, J. 480, 624
Kleinberg, E. M. 602
Knight, K. 91, 292, 666, 693
Koh, K. 125
Kohane, I. 631
Kohavi, R. 243, 257
Kohn, E. 664
Kohonen, T. 462, 481, 531, 532,
</p>
<p>578
Koller, D. 629, 630, 642, 645
Kooperberg, C. 328
Korn, E. L. 693
Kotze, J. 122
Kressel, U. 437
Krogh, A. 414
</p>
<p>Ladd, C. 654, 658
Lafferty, J. 90, 304
Lafferty, J. D. 642
Lagus, K. 531, 532, 578
Laird, N. 292, 449
Lambert, D. 376
Lander, E. 654, 658, 663
Lange, K. 92, 294, 583, 584
Langford, J. C. 573
Larsen, R. 551
Latulippe, E. 654, 658
Lauffenburger, D. 625
Lauritzen, S. 629, 632, 645
Lawson, C. 93
Le Cun, Y. 404, 406&ndash;408, 414
Leathwick, J. 375, 376, 378
Leblanc, M. 292
LeCun, Y. 644
Lee, D. 552, 553
Lee, M.-L. 693
Lee, S.-I. 642</p>
<p/>
</div>
<div class="page"><p/>
<p>Author Index 733
</p>
<p>Lee, W. 384, 615
Leslie, C. 668, 669
Levina, E. 652, 693
Levine, P. J. 664
Lewis, D. 661
Li, K.-C. 480
Li, R. 92
Lin, H. 331
Lin, Y. 90, 304, 428, 455
Liotta, L. A. 664
Little, R. 332, 647
Littman, M. 578
Liu, H. 90, 304
Lloyd, S. 481, 578
Loader, C. 209, 216
Loda, M. 654, 658
Loh, M. 663
Loh, W. 310
Lugosi, G. 384
</p>
<p>Ma, Y. 257
Macnaughton Smith, P. 526
MacKay, D. 623
MacQueen, J. 481, 578
Madigan, D. 257, 292, 661
Makeig, S. 564, 565
Mannila, H. 489&ndash;491, 578
Mardia, K. 94, 135, 441, 539, 559,
</p>
<p>578, 630, 679
Marron, J. 695
Mason, L. 384
Massart, D. 517
Matas, J. 624
McCullagh, P. 638, 640
McCulloch, C. 331
McCulloch, W. 414
McLachlan, G. 135, 247
McNeal, J. 3, 49
McNeil, B. 317
McShane, L. M. 693
Mease, D. 384, 603
Meinshausen, N. 91, 635, 642
Meir, R. 384
Mesirov, J. 654, 658, 663
Mills, G. B. 664
</p>
<p>Mockett, L. 526
Morgan, J. N. 334
Motwani, R. 577
Mukherjee, S. 654, 658
Mulier, F. 38, 239
Muller-Hermelink, H. K. 674
Müller, K.-R. 547, 548
Munro, S. 397
Murray, W. 96, 421
Myles, J. 475
</p>
<p>Nadler, B. 679
Narasimhan, B. 693
Neal, R. 268, 292, 409&ndash;412, 414,
</p>
<p>605, 623
Nelder, J. 638, 640
Noble, W. S. 668, 669
Nolan, G. 625
Nowlan, S. 334
</p>
<p>Oja, E. 560, 562, 578, 583
Olesen, K. G. 629
Olshen, R. 251, 308, 310, 334, 367,
</p>
<p>451, 453
Onton, J. 564, 565
Osborne, M. 76, 94
Osindero, S. 644
</p>
<p>Paatero, A. 531, 532, 578
Pace, R. K. 371
Page, L. 577
Palmer, R. 414
Pantoni, L. 551
Park, M. Y. 94, 126, 661
Parker, D. 414
Paul, D. 676, 679&ndash;683, 693
Pearl, J. 629, 645
Pe&rsquo;er, D. 625
Perez, O. 625
Peterson 641
Petricoin, E. F. 664
Pitts, W. 414
Plastria, F. 517
Platt, J. 453
Poggio, T. 168, 174, 181, 415, 455,
</p>
<p>654, 658</p>
<p/>
</div>
<div class="page"><p/>
<p>734 Author Index
</p>
<p>Pontil, M. 168, 181, 455
Popescu, B. 617&ndash;619, 621, 623
Prentice, R. 674, 693
Presnell, B. 76, 94
Přeučil, S. 640
</p>
<p>Qu, Y. 664
Quinlan, R. 312, 334, 624
</p>
<p>Radmacher, M. D. 693
Raftery, A. 257, 292
Ramaswamy, S. 654, 658
Ramsay, J. 181, 578
Rao, C. R. 455
Rätsch, G. 384, 615
Ravikumar, P. 90, 304, 642
Redwine, E. 3, 49
Reich, M. 654, 658
Richardson, J. 375
Richardson, T. S. 631, 633
Ridgeway, G. 361
Rieger, K. 684
Rifkin, R. 654, 658
Ripley, B. D. 38, 131, 135, 136,
</p>
<p>234, 308, 310, 400, 414,
415, 455, 468, 480, 481,
641, 645
</p>
<p>Rissanen, J. 257
Ritov, Y. 89, 91
Robbins, H. 397
Rocha, G. 90
Roosen, C. 414
Rosenblatt, F. 102, 129, 414
Rosenwald, A. 674
Rosset, S. 89, 98, 348, 349, 385,
</p>
<p>426, 428, 434, 610, 611,
615, 657, 661, 664, 666,
693
</p>
<p>Rostrup, E. 551
Rousseauw, J. 122
Rousseeuw, P. 517, 526, 578
Rowe, D. 375
Roweis, S. T. 573
Rubin, D. 292, 332, 449, 647
Rumelhart, D. 414
</p>
<p>Ryberg, C. 551
</p>
<p>Saarela, A. 531, 532, 578
Sachs, K. 625
Salojärvi, J. 531, 532, 578
Saul, L. K. 573
Saunders, M. 68, 94, 666, 693
Schapire, R. 337, 380, 383, 384,
</p>
<p>615
Schellhammer, P. F. 664
Schnitzler, C. 334
Schölkopf, B. 547, 548
Schroeder, A. 391
Schwarz, G. 233, 257
Scott, D. 216
Seber, G. 94
Segal, M. 596
Sejnowski, T. 578, 645
Semmes, O. J. 664
Seung, H. 552, 553
Shafer, G. 629
Shao, J. 257
Shenoy, P. 629
Short, R. 475
Shustek, L. 480
Shyu, M. 369
Siegmund, D. 689
Silverman, B. 181, 183, 216, 334,
</p>
<p>486, 567, 578
Silvey, S. 292
Simard, P. 407, 471, 480, 481
Simon, R. M. 693
Simone, C. 664
Singer, Y. 384
Sjöstrand, K. 551
Slate, E. 331
Slonim, D. 631, 663
Smeland, E. B. 674
Smith, A. 292
Smola, A. 547, 548
Sonquist, J. A. 334
Spector, P. 243, 257
Speed, T. 632, 686, 693
Spiegelhalter, D. 292, 629
Spiegelman, C. 679</p>
<p/>
</div>
<div class="page"><p/>
<p>Author Index 735
</p>
<p>Spielman, D. A. 578
Srikant, R. 489&ndash;491, 578
Stamey, T. 3, 49
Staudt, L. M. 674
Steinberg, S. M. 664
Stern, H. 292
Stodden, V. 554
Stone, C. 251, 308, 310, 328, 334,
</p>
<p>367, 451, 453
Stone, M. 81, 257
Storey, J. 689, 692, 693, 697, 698
Stork, D. 38, 135
Studholme, C. 551
Stuetzle, W. 391, 414, 541, 578
Surowiecki, J. 286
Swayne, D. 565, 578
</p>
<p>Tamayo, P. 631, 654, 658, 663
Tang, J. 684
Tanner, M. 292
Tao, T. 89, 613
Tarpey, T. 578
Taylor, J. 88, 94, 610, 614, 689
Taylor, P. 375, 376, 378
Teh, Y.-W. 644
Tenenbaum, J. B. 573
Teng, S.-H. 578
Thomas, J. 257
Tibshirani, R. 73, 78, 86, 88, 90,
</p>
<p>92&ndash;94, 97, 98, 110, 121,
122, 126, 137, 216, 257,
292, 297, 299, 304, 334,
339, 345, 384, 428, 431,
434, 437, 441, 446, 451,
455, 475, 478, 480, 481,
519, 550, 565, 568, 609&ndash;
611, 614, 636, 642, 657,
658, 660, 661, 666, 667,
676, 679&ndash;684, 692, 693
</p>
<p>Toivonen, H. 489&ndash;491, 578
Traskin, M. 384
Trendafilov, N. T. 550
Tropp, J. 91
Truong, Y. 328
Tsybakov, A. 89, 91
</p>
<p>Tukey, J. 414, 565, 578
Turlach, B. 76, 94
Turnbull, B. 331
Tusher, V. 684, 692
Tusnády, G. 292
</p>
<p>Uddin, M. 550
</p>
<p>Valiant, L. G. 380
van der Merwe, A. 84
Van Loan, C. 335, 535
Vandenberghe, L. 632
Vanichsetakul, N. 310
Vapnik, V. 38, 102, 132, 135, 171,
</p>
<p>257, 438, 455, 658
Vayatis, N. 384
Vazirani, U. 380
Verkamo, A. I. 489&ndash;491, 578
Vidakovic, B. 181
von Luxburg, U. 578
</p>
<p>Wahba, G. 168, 169, 181, 257, 268,
428, 429, 455
</p>
<p>Wainwright, M. 91
Wainwright, M. J. 642
Waldemar, G. 551
Walther, G. 88, 94, 519, 610, 614
Wang, P. 667
Ward, M. D. 664
Warmuth, M. 615
Wasserman, L. 90, 304, 626, 645,
</p>
<p>693
Watkins, C. 658
Wegkamp, M. 91
Weisberg, S. 94
Werbos, P. 414
Wermuth, N. 645
Weston, J. 658, 668, 669
Whittaker, J. 632, 633, 641, 645
Wickerhauser, M. 181
Widrow, B. 396, 414
Wild, C. 300
Williams, R. 414
Williams, W. 526
Wilson, R. 57</p>
<p/>
</div>
<div class="page"><p/>
<p>736 Author Index
</p>
<p>Winograd, T. 577
Wold, H. 94
Wolpert, D. 292
Wong, M. A. 510
Wong, W. 292
Wright, G. 664, 674, 693
Wright, M. 96, 421
Wu, T. 92, 294, 583
Wyner, A. 384, 603
</p>
<p>Yang, N. 3, 49
Yang, Y. 686, 693
Yasui, Y. 664
Yeang, C. 654, 658
Yee, T. 300
Yekutieli, Y. 693
Yu, B. 90, 91, 384
Yuan, M. 90
</p>
<p>Zhang, H. 90, 304, 428, 455
Zhang, J. 409&ndash;412, 605
Zhang, P. 257
Zhang, T. 384
Zhao, P. 90, 91
Zhao, Y. 693
Zhu, J. 89, 98, 174, 348, 349, 385,
</p>
<p>426, 428, 434, 610, 611,
615, 657, 661, 664, 666,
693
</p>
<p>Zidek, J. 84
Zou, H. 72, 78, 92, 349, 385, 550,
662, 693</p>
<p/>
</div>
<div class="page"><p/>
<p>Index
</p>
<p>L1 regularization, see Lasso
</p>
<p>Activation function, 392&ndash;395
AdaBoost, 337&ndash;346
Adaptive lasso, 92
Adaptive methods, 429
Adaptive nearest neighbor meth-
</p>
<p>ods, 475&ndash;478
Adaptive wavelet filtering, 181
Additive model, 295&ndash;304
Adjusted response, 297
Affine set, 130
Affine-invariant average, 482, 540
AIC, see Akaike information cri-
</p>
<p>terion
Akaike information criterion (AIC),
</p>
<p>230
Analysis of deviance, 124
Applications
</p>
<p>abstracts, 672
aorta, 204
bone, 152
California housing, 371&ndash;372,
</p>
<p>591
countries, 517
</p>
<p>demographics, 379&ndash;380
document, 532
flow cytometry, 637
galaxy, 201
heart attack, 122, 146, 207
lymphoma, 674
marketing, 488
microarray, 5, 505, 532
nested spheres, 590
New Zealand fish, 375&ndash;379
nuclear magnetic resonance,
</p>
<p>176
ozone, 201
prostate cancer, 3, 49, 61, 608
proteinmass spectrometry, 664
satellite image, 470
skin of the orange, 429&ndash;432
spam, 2, 300&ndash;304, 313, 320,
</p>
<p>328, 352, 593
vowel, 440, 464
waveform, 451
ZIP code, 4, 404, 536&ndash;539
</p>
<p>Archetypal analysis, 554&ndash;557
Association rules, 492&ndash;495, 499&ndash;
</p>
<p>501
</p>
<p> &copy; Springer Science+Business Media, LLC 2009 
</p>
<p>T. Hastie et al., The Elements of Statistical Learning, Second Edition, 
</p>
<p>DOI: 10.1007/b94608,
</p>
<p>737</p>
<p/>
</div>
<div class="page"><p/>
<p>738 Index
</p>
<p>Automatic relevance determination,
411
</p>
<p>Automatic selection of smoothing
parameters , 156
</p>
<p>B-Spline, 186
Back-propagation, 392&ndash;397, 408&ndash;
</p>
<p>409
Backfitting, 297, 391
Backward
</p>
<p>selection, 58
stepwise selection, 59
</p>
<p>Backward pass, 396
Bagging, 282&ndash;288, 409, 587
Basis expansions and regulariza-
</p>
<p>tion, 139&ndash;189
Basis functions, 141, 186, 189, 321,
</p>
<p>328
Batch learning, 397
Baum&ndash;Welch algorithm, 272
Bayes
</p>
<p>classifier, 21
factor, 234
methods, 233&ndash;235, 267&ndash;272
rate, 21
</p>
<p>Bayesian, 409
Bayesian information criterion (BIC),
</p>
<p>233
Benjamini&ndash;Hochberg method, 688
Best-subset selection, 57, 610
Between class covariance matrix,
</p>
<p>114
Bias, 16, 24, 37, 160, 219
Bias-variance decomposition, 24,
</p>
<p>37, 219
Bias-variance tradeoff, 37, 219
BIC, see Bayesian Information Cri-
</p>
<p>terion
Boltzmann machines, 638&ndash;648
Bonferroni method, 686
Boosting, 337&ndash;386, 409
</p>
<p>as lasso regression, 607&ndash;609
exponential loss and AdaBoost,
</p>
<p>343
gradient boosting, 358
</p>
<p>implementations, 360
margin maximization, 613
numerical optimization, 358
partial-dependence plots, 369
regularization path, 607
shrinkage, 364
stochastic gradient boosting,
</p>
<p>365
tree size, 361
variable importance, 367
</p>
<p>Bootstrap, 249, 261&ndash;264, 267,
</p>
<p>relationship to Bayesian
</p>
<p>relationship to maximum like-
lihood method, 267
</p>
<p>Bottom-up clustering, 520&ndash;528
Bump hunting, see Patient rule
</p>
<p>induction method
Bumping, 290&ndash;292
</p>
<p>C5.0, 624
Canonical variates, 441
CART, see Classification and re-
</p>
<p>gression trees
Categorical predictors, 10, 310
Censored data, 674
Classical multidimensional scaling,
</p>
<p>570
Classification, 22, 101&ndash;137, 305&ndash;
</p>
<p>317, 417&ndash;429
Classification and regression trees
</p>
<p>(CART), 305&ndash;317
Clique, 628
Clustering, 501&ndash;528
</p>
<p>k-means, 509&ndash;510
agglomerative, 523&ndash;528
hierarchical, 520&ndash;528
</p>
<p>Codebook, 515
Combinatorial algorithms, 507
Combining models, 288&ndash;290
Committee, 289, 587, 605
Comparison of learning methods,
</p>
<p>350&ndash;352
Complete data, 276
</p>
<p>271&ndash;282, 587
</p>
<p>method, 271</p>
<p/>
</div>
<div class="page"><p/>
<p>Index 739
</p>
<p>Complexity parameter, 37
Computational shortcuts
</p>
<p>quadratic penalty, 659
Condensing procedure, 480
Conditional likelihood, 31
Confusion matrix, 301
Conjugate gradients, 396
Consensus, 285&ndash;286
Convolutional networks, 407
Coordinate descent, 92, 636, 668
COSSO, 304
Cost complexity pruning, 308
Covariance graph, 631
Cp statistic, 230
Cross-entropy, 308&ndash;310
Cross-validation, 241&ndash;245
Cubic smoothing spline, 151&ndash;153
Cubic spline, 151&ndash;153
Curse of dimensionality, 22&ndash;26
</p>
<p>Dantzig selector, 89
Data augmentation, 276
Daubechies symmlet-8 wavelets,
</p>
<p>176
De-correlation, 597
Decision boundary, 13&ndash;15, 21
Decision trees, 305&ndash;317
Decoder, 515, see encoder
Decomposable models, 641
Degrees of freedom
</p>
<p>in an additive model, 302
in ridge regression, 68
of a tree, 336
of smoother matrices, 153&ndash;154,
</p>
<p>158
Delta rule, 397
Demmler-Reinsch basis for splines,
</p>
<p>156
Density estimation, 208&ndash;215
Deviance, 124, 309
Diagonal linear discriminant anal-
</p>
<p>ysis, 651&ndash;654
Dimension reduction, 658
</p>
<p>for nearest neighbors, 479
Discrete variables, 10, 310&ndash;311
</p>
<p>Discriminant
adaptive nearest neighbor
</p>
<p>analysis, 106&ndash;119
coordinates, 108
functions, 109&ndash;110
</p>
<p>Dissimilarity measure, 503&ndash;504
Dummy variables, 10
</p>
<p>Early stopping, 398
Effective degrees of freedom, 17,
</p>
<p>68, 153&ndash;154, 158, 232,
</p>
<p>Effective number of parameters,
15, 68, 153&ndash;154, 158, 232,
302, 336
</p>
<p>Eigenvalues of a smoother matrix,
154
</p>
<p>Elastic net, 662
EM algorithm, 272&ndash;279
</p>
<p>as amaximization-maximization
procedure, 277
</p>
<p>for two component Gaussian
mixture, 272
</p>
<p>Encoder, 514&ndash;515
Ensemble, 616&ndash;623
Ensemble learning, 605&ndash;624
Entropy, 309
Equivalent kernel, 156
Error rate, 219&ndash;230
Error-correcting codes, 606
Estimates of in-sample prediction
</p>
<p>error, 230
Expectation-maximization algo-
</p>
<p>EM algorithm
Extra-sample error, 228
</p>
<p>False discovery rate, 687&ndash;690, 692,
693
</p>
<p>Feature, 1
extraction, 150
selection, 409, 658, 681&ndash;683
</p>
<p>Feed-forward neural networks,
</p>
<p>classifier, 475&ndash;480
</p>
<p>302, 336
</p>
<p>rithm, see
</p>
<p>392&ndash;408</p>
<p/>
</div>
<div class="page"><p/>
<p>740 Index
</p>
<p>Fisher&rsquo;s linear discriminant, 106&ndash;
119, 438
</p>
<p>Flexible discriminant analysis, 440&ndash;
445
</p>
<p>Forward
selection, 58
stagewise, 86, 608
stagewise additive modeling,
</p>
<p>342
stepwise, 73
</p>
<p>Forward pass algorithm, 395
Fourier transform, 168
Frequentist methods, 267
Function approximation, 28&ndash;36
Fused lasso, 666
</p>
<p>Gap statistic, 519
Gating networks, 329
Gauss-Markov theorem, 51&ndash;52
Gauss-Newton method, 391
Gaussian (normal) distribution, 16
Gaussian graphical model, 630
Gaussian mixtures, 273, 463, 492,
</p>
<p>509
Gaussian radial basis functions,
</p>
<p>212
GBM, see Gradient boosting
GBM package, see Gradient boost-
</p>
<p>ing
GCV, see Generalized cross-validation
GEM (generalized EM), 277
Generalization
</p>
<p>error, 220
performance, 220
</p>
<p>Generalized additive model, 295&ndash;
304
</p>
<p>Generalized association rules, 497&ndash;
499
</p>
<p>Generalized cross-validation, 244
Generalized linear discriminant anal-
</p>
<p>ysis, 438
Generalized linear models, 125
Gibbs sampler, 279&ndash;280, 641
</p>
<p>for mixtures, 280
Gini index, 309
</p>
<p>Global Markov property, 628
Gradient Boosting, 359&ndash;361
Gradient descent, 358, 395&ndash;397
Graph Laplacian, 545
Graphical lasso, 636
Grouped lasso, 90
</p>
<p>Haar basis function, 176
Hammersley-Clifford theorem, 629
Hard-thresholding, 653
Hat matrix, 46
Helix, 582
Hessian matrix, 121
Hidden nodes, 641&ndash;642
Hidden units, 393&ndash;394
Hierarchical clustering, 520&ndash;528
Hierarchical mixtures of experts,
</p>
<p>329&ndash;332
High-dimensional problems, 649
Hints, 96
Hyperplane, see Separating Hy-
</p>
<p>perplane
</p>
<p>ICA, see Independent components
analysis
</p>
<p>Importance sampling, 617
In-sample prediction error, 230
Incomplete data, 332
Independent components analysis,
</p>
<p>557&ndash;570
Independent variables, 9
Indicator response matrix, 103
Inference, 261&ndash;294
Information
</p>
<p>Fisher, 266
observed, 274
</p>
<p>Information theory, 236, 561
Inner product, 53, 668, 670
Inputs, 10
Instability of trees, 312
Intercept, 11
Invariance manifold, 471
Invariant metric, 471
Inverse wavelet transform, 179</p>
<p/>
</div>
<div class="page"><p/>
<p>Index 741
</p>
<p>IRLS, see Iteratively reweighted
least squares
</p>
<p>Irreducible error, 224
Ising model, 638
ISOMAP, 572
Isometric feature mapping, 572
Iterative proportional scaling, 585
Iteratively reweighted least squares
</p>
<p>(IRLS), 121
</p>
<p>Jensen&rsquo;s inequality, 293
Join tree, 629
Junction tree, 629
</p>
<p>K-means clustering, 460, 509&ndash;514
K-medoid clustering, 515&ndash;520
K-nearest neighbor classifiers, 463
Karhunen-Loeve transformation (prin-
</p>
<p>cipal components), 66&ndash;
67, 79, 534&ndash;539
</p>
<p>Karush-Kuhn-Tucker conditions,
133, 420
</p>
<p>Kernel
classification, 670
density classification, 210
density estimation, 208&ndash;215
function, 209
logistic regression, 654
principal component, 547&ndash;550
string, 668&ndash;669
trick, 660
</p>
<p>Kernel methods, 167&ndash;176, 208&ndash;215,
423&ndash;438, 659
</p>
<p>Knot, 141, 322
Kriging, 171
Kruskal-Shephard scaling, 570
Kullback-Leibler distance, 561
</p>
<p>Lagrange multipliers, 293
Landmark, 539
Laplacian, 545
Laplacian distribution, 72
LAR, see Least angle regression
Lasso, 68&ndash;69, 86&ndash;90, 609, 635, 636,
</p>
<p>661
</p>
<p>fused, 666
Latent
</p>
<p>factor, 674
variable, 678
</p>
<p>Learning, 1
Learning rate, 396
Learning vector quantization, 462
Least angle regression, 73&ndash;79, 86,
</p>
<p>610
Least squares, 11, 32
Leave-one-out cross-validation, 243
LeNet, 406
Likelihood function, 265, 273
Linear basis expansion, 139&ndash;148
Linear combination splits, 312
Linear discriminant function, 106&ndash;
</p>
<p>119
Linear methods
</p>
<p>for classification, 101&ndash;137
for regression, 43&ndash;99
</p>
<p>Linear models and least squares,
11
</p>
<p>Linear regression of an indicator
matrix, 103
</p>
<p>Linear separability, 129
Linear smoother, 153
Link function, 296
LLE, see Local linear embedding
Local false discovery rate, 693
Local likelihood, 205
Local linear embedding, 572
Local methods in high dimensions,
</p>
<p>22&ndash;27
Local minima, 400
Local polynomial regression, 197
Local regression, 194, 200
Localization in time/frequency, 175
Loess (local regression), 194, 200
Log-linear model, 639
Log-odds ratio (logit), 119
Logistic (sigmoid) function, 393
Logistic regression, 119&ndash;128, 299
Logit (log-odds ratio), 119
Loss function, 18, 21, 219&ndash;223, 346
Loss matrix, 310</p>
<p/>
</div>
<div class="page"><p/>
<p>742 Index
</p>
<p>Lossless compression, 515
Lossy compression, 515
LVQ, see Learning Vector Quan-
</p>
<p>tization
</p>
<p>Mahalanobis distance, 441
Majority vote, 337
Majorization, 294, 553
Majorize-Minimize algorithm, 294,
</p>
<p>584
MAP (maximum aposteriori) es-
</p>
<p>timate, 270
Margin, 134, 418
Market basket analysis, 488, 499
Markov chain Monte Carlo (MCMC)
</p>
<p>methods, 279
Markov graph, 627
Markov networks, 638&ndash;648
MARS, see Multivariate adaptive
</p>
<p>regression splines
MART, see Multiple additive re-
</p>
<p>gression trees
Maximum likelihood estimation,
</p>
<p>31, 261, 265
MCMC, see Markov Chain Monte
</p>
<p>Carlo Methods
MDL, see Minimum description
</p>
<p>length
Mean field approximation, 641
Mean squared error, 24, 285
Memory-based method, 463
Metropolis-Hastings algorithm, 282
Minimum description length (MDL),
</p>
<p>235
Minorization, 294, 553
Minorize-Maximize algorithm, 294,
</p>
<p>584
Misclassification error, 17, 309
Missing data, 276, 332&ndash;333
Missing predictor values, 332&ndash;333
Mixing proportions, 214
Mixture discriminant analysis, 449&ndash;
</p>
<p>455
Mixture modeling, 214&ndash;215, 272&ndash;
</p>
<p>275, 449&ndash;455, 692
</p>
<p>Mixture of experts, 329&ndash;332
Mixtures and the EM algorithm,
</p>
<p>272&ndash;275
MM algorithm, 294, 584
Mode seekers, 507
Model averaging and stacking, 288
Model combination, 289
Model complexity, 221&ndash;222
Model selection, 57, 222&ndash;223, 230&ndash;
</p>
<p>231
Modified regression, 634
Monte Carlo method, 250, 495
Mother wavelet, 178
Multidimensional scaling, 570&ndash;572
Multidimensional splines, 162
Multiedit algorithm, 480
Multilayer perceptron, 400, 401
Multinomial distribution, 120
Multiple additive regression trees
</p>
<p>(MART), 361
Multiple hypothesis testing, 683&ndash;
</p>
<p>693
Multiple minima, 291, 400
Multiple outcome shrinkage and
</p>
<p>selection, 84
Multiple outputs, 56, 84, 103&ndash;106
Multiple regression from simple
</p>
<p>Multiresolution analysis, 178
Multivariate adaptive regression
</p>
<p>splines (MARS), 321&ndash;327
Multivariate nonparametric regres-
</p>
<p>sion, 445
</p>
<p>Nadaraya&ndash;Watson estimate, 193
Naive Bayes classifier, 108, 210&ndash;
</p>
<p>211, 694
Natural cubic splines, 144&ndash;146
Nearest centroids, 670
Nearest neighbor methods, 463&ndash;
</p>
<p>483
Nearest shrunken centroids, 651&ndash;
</p>
<p>654, 694
Network diagram, 392
Neural networks, 389&ndash;416
</p>
<p>univariate regression, 52</p>
<p/>
</div>
<div class="page"><p/>
<p>Index 743
</p>
<p>Newton&rsquo;s method (Newton-Raphson
procedure), 120&ndash;122
</p>
<p>Non-negative matrix factorization,
553&ndash;554
</p>
<p>Nonparametric logistic regression,
299&ndash;304
</p>
<p>Normal (Gaussian) distribution,
16, 31
</p>
<p>Normal equations, 12
Numerical optimization, 395&ndash;396
</p>
<p>Object dissimilarity, 505&ndash;507
Online algorithm, 397
Optimal scoring, 445, 450&ndash;451
Optimal separating hyperplane, 132&ndash;
</p>
<p>135
Optimism of the training error rate,
</p>
<p>228&ndash;230
Ordered categorical (ordinal) pre-
</p>
<p>dictor, 10, 504
Ordered features, 666
Orthogonal predictors, 53
Overfitting, 220, 228&ndash;230, 364
</p>
<p>PageRank, 576
Pairwise distance, 668
Pairwise Markov property, 628
Parametric bootstrap, 264
Partial dependence plots, 369&ndash;370
Partial least squares, 80&ndash;82, 680
Partition function, 638
Parzen window, 208
Pasting, 318
Path algorithm, 73&ndash;79, 86&ndash;89, 432
Patient rule induction method(PRIM),
</p>
<p>317&ndash;321, 499&ndash;501
Peeling, 318
Penalization, 607, see regulariza-
</p>
<p>tion
Penalized discriminant analysis, 446&ndash;
</p>
<p>449
Penalized polynomial regression,
</p>
<p>171
Penalized regression, 34, 61&ndash;69, 171
Penalty matrix, 152, 189
</p>
<p>Perceptron, 392&ndash;416
Piecewise polynomials and splines,
</p>
<p>36, 143
Posterior
</p>
<p>distribution, 268
probability, 233&ndash;235, 268
</p>
<p>Power method, 577
Pre-conditioning, 681&ndash;683
Prediction accuracy, 329
Prediction error, 18
Predictive distribution, 268
PRIM, see Patient rule induction
</p>
<p>method
Principal components, 66&ndash;67, 79&ndash;
</p>
<p>80, 534&ndash;539, 547
regression, 79&ndash;80
sparse, 550
supervised, 674
</p>
<p>Principal curves and surfaces,
</p>
<p>Principal points, 541
Prior distribution, 268&ndash;272
Procrustes
</p>
<p>average, 540
distance, 539
</p>
<p>Projection pursuit, 389&ndash;392, 565
regression, 389&ndash;392
</p>
<p>Prototype classifier, 459&ndash;463
Prototype methods, 459&ndash;463
Proximity matrices, 503
Pruning, 308
</p>
<p>QR decomposition, 55
Quadratic approximations and in-
</p>
<p>ference, 124
Quadratic discriminant function,
</p>
<p>108, 110
</p>
<p>Radial basis function (RBF) net-
work, 392
</p>
<p>Radial basis functions, 212&ndash;214,
275, 393
</p>
<p>Radial kernel, 548
Random forest, 409, 587&ndash;604
</p>
<p>algorithm, 588
</p>
<p>541&ndash;544</p>
<p/>
</div>
<div class="page"><p/>
<p>744 Index
</p>
<p>bias, 596&ndash;601
comparison to boosting, 589
example, 589
out-of-bag (oob), 592
overfit, 596
proximity plot, 595
variable importance, 593
variance, 597&ndash;601
</p>
<p>Rao score test, 125
Rayleigh quotient, 116
Receiver operating characteristic
</p>
<p>(ROC) curve, 317
Reduced-rank linear discriminant
</p>
<p>analysis, 113
Regression, 11&ndash;14, 43&ndash;99, 200&ndash;204
Regression spline, 144
Regularization, 34, 167&ndash;176
Regularized discriminant analysis,
</p>
<p>112&ndash;113, 654
Relevance network, 631
Representer of evaluation, 169
Reproducing kernel Hilbert space,
</p>
<p>167&ndash;176, 428&ndash;429
Reproducing property, 169
Responsibilities, 274&ndash;275
Ridge regression, 61&ndash;68, 650, 659
Risk factor, 122
Robust fitting, 346&ndash;350
Rosenblatt&rsquo;s perceptron learning
</p>
<p>algorithm, 130
Rug plot, 303
Rulefit, 623
</p>
<p>SAM, 690&ndash;693, see Significance Anal-
ysis of Microarrays
</p>
<p>Sammon mapping, 571
SCAD, 92
Scaling of the inputs, 398
Schwarz&rsquo;s criterion, 230&ndash;235
Score equations, 120, 265
Self-consistency property, 541&ndash;543
Self-organizing map (SOM), 528&ndash;
</p>
<p>534
Sensitivity of a test, 314&ndash;317
Separating hyperplane, 132&ndash;135
</p>
<p>Separating hyperplanes, 136, 417&ndash;
419
</p>
<p>Separator, 628
Shape average, 482, 540
Shrinkage methods, 61&ndash;69, 652
Sigmoid, 393
Significance Analysis of Microar-
</p>
<p>rays, 690&ndash;693
Similarity measure, see Dissimi-
</p>
<p>larity measure
Single index model, 390
Singular value decomposition, 64,
</p>
<p>535&ndash;536, 659
singular values, 535
singular vectors, 535
</p>
<p>Sliced inverse regression, 480
Smoother, 139&ndash;156, 192&ndash;199
</p>
<p>matrix, 153
Smoothing parameter, 37, 156&ndash;161,
</p>
<p>198&ndash;199
Smoothing spline, 151&ndash;156
Soft clustering, 512
Soft-thresholding, 653
Softmax function, 393
SOM, see Self-organizing map
Sparse, 175, 304, 610&ndash;613, 636
</p>
<p>additive model, 91
graph, 625, 635
</p>
<p>Specificity of a test, 314&ndash;317
Spectral clustering, 544&ndash;547
Spline, 186
</p>
<p>additive, 297&ndash;299
cubic, 151&ndash;153
cubic smoothing, 151&ndash;153
interaction, 428
regression, 144
smoothing, 151&ndash;156
thin plate, 165
</p>
<p>Squared error loss, 18, 24, 37, 219
SRM, see Structural risk minimiza-
</p>
<p>tion
Stacking (stacked generalization),
</p>
<p>290
Starting values, 397
Statistical decision theory, 18&ndash;22</p>
<p/>
</div>
<div class="page"><p/>
<p>Index 745
</p>
<p>Statistical model, 28&ndash;29
Steepest descent, 358, 395&ndash;397
Stepwise selection, 60
Stochastic approximation, 397
Stochastic search (bumping), 290&ndash;
</p>
<p>292
Stress function, 570&ndash;572
Structural risk minimization (SRM),
</p>
<p>239&ndash;241
Subset selection, 57&ndash;60
Supervised learning, 2
Supervised principal components,
</p>
<p>674&ndash;681
Support vector classifier, 417&ndash;421,
</p>
<p>654
multiclass, 657
</p>
<p>Support vector machine, 423&ndash;437
SURE shrinkage method, 179
Survival analysis, 674
Survival curve, 674
SVD, see Singular value decom-
</p>
<p>position
Symmlet basis, 176
</p>
<p>Tangent distance, 471&ndash;475
Tanh activation function, 424
Target variables, 10
Tensor product basis, 162
Test error, 220&ndash;223
Test set, 220
Thin plate spline, 165
Thinning strategy, 189
Trace of a matrix, 153
Training epoch, 397
Training error, 220&ndash;223
Training set, 219&ndash;223
Tree for regression, 307&ndash;308
Tree-based methods, 305&ndash;317
Trees for classification, 308&ndash;310
Trellis display, 202
</p>
<p>Undirected graph, 625&ndash;648
Universal approximator, 390
Unsupervised learning, 2, 485&ndash;585
Unsupervised learning as super-
</p>
<p>vised learning, 495&ndash;497
</p>
<p>Validation set, 222
Vapnik-Chervonenkis (VC) dimen-
</p>
<p>sion, 237&ndash;239
Variable importance plot, 594
Variable types and terminology, 9
Variance, 16, 25, 37, 158&ndash;161, 219
</p>
<p>between, 114
within, 114, 446
</p>
<p>Variance reduction, 588
Varying coefficient models, 203&ndash;
</p>
<p>204
VCdimension, see Vapnik&ndash;Chervon-
</p>
<p>enkis dimension
Vector quantization, 514&ndash;515
Voronoi regions, 510
</p>
<p>Wald test, 125
Wavelet
</p>
<p>basis functions, 176&ndash;179
smoothing, 174
transform, 176&ndash;179
</p>
<p>Weak learner, 383, 605
Weakest link pruning, 308
Webpages, 576
Website for book, 8
Weight decay, 398
Weight elimination, 398
Weights in a neural network, 395
Within class covariance matrix,
</p>
<p>114, 446</p>
<p/>
</div>
<ul>	<li>Contents</li>
	<li>1 Introduction</li>
<ul>	<li>Who Should Read this Book</li>
	<li>How This Book is Organized</li>
	<li>Book Website</li>
	<li>Note for Instructors</li>
</ul>
	<li>2 Overview of Supervised Learning</li>
<ul>	<li>2.1 Introduction</li>
	<li>2.2 Variable Types and Terminology</li>
	<li>2.3 Two Simple Approaches to Prediction: Least Squares and Nearest Neighbors</li>
<ul>	<li>2.3.1 Linear Models and Least Squares</li>
	<li>2.3.2 Nearest-Neighbor Methods</li>
	<li>2.3.3 From Least Squares to Nearest Neighbors</li>
</ul>
	<li>2.4 Statistical Decision Theory</li>
	<li>2.5 Local Methods in High Dimensions</li>
	<li>2.6 Statistical Models, Supervised Learning and Function Approximation</li>
<ul>	<li>2.6.1 A Statistical Model for the Joint Distribution Pr(X, Y )</li>
	<li>2.6.2 Supervised Learning</li>
	<li>2.6.3 Function Approximation</li>
</ul>
	<li>2.7 Structured Regression Models</li>
<ul>	<li>2.7.1 Difficulty of the Problem</li>
</ul>
	<li>2.8 Classes of Restricted Estimators</li>
<ul>	<li>2.8.1 Roughness Penalty and Bayesian Methods</li>
	<li>2.8.2 Kernel Methods and Local Regression</li>
	<li>2.8.3 Basis Functions and Dictionary Methods</li>
</ul>
	<li>2.9 Model Selection and the Bias&ndash;VarianceTradeoff</li>
	<li>Bibliographic Notes</li>
	<li>Exercises</li>
</ul>
	<li>3 Linear Methods for Regression</li>
<ul>	<li>3.1 Introduction</li>
	<li>3.2 Linear Regression Models and Least Squares</li>
<ul>	<li>3.2.1 Example: Prostate Cancer</li>
	<li>3.2.2 The Gauss&ndash;Markov Theorem</li>
	<li>3.2.3 Multiple Regression from Simple Univariate Regression</li>
	<li>3.2.4 Multiple Outputs</li>
</ul>
	<li>3.3 Subset Selection</li>
<ul>	<li>3.3.1 Best-Subset Selection</li>
	<li>3.3.2 Forward- and Backward-Stepwise Selection</li>
	<li>3.3.3 Forward-Stagewise Regression</li>
	<li>3.3.4 Prostate Cancer Data Example (Continued)</li>
</ul>
	<li>3.4 Shrinkage Methods</li>
<ul>	<li>3.4.1 Ridge Regression</li>
	<li>3.4.2 The Lasso</li>
	<li>3.4.3 Discussion: Subset Selection, Ridge Regression and the Lasso</li>
	<li>3.4.4 Least Angle Regression</li>
<ul>	<li>Degrees-of-Freedom Formula for LAR and Lasso</li>
</ul>
</ul>
	<li>3.5 Methods Using Derived Input Directions</li>
<ul>	<li>3.5.1 Principal Components Regression</li>
	<li>3.5.2 Partial Least Squares</li>
</ul>
	<li>3.6 Discussion: A Comparison of the Selection and Shrinkage Methods</li>
	<li>3.7 Multiple Outcome Shrinkage and Selection</li>
	<li>3.8 More on the Lasso and Related Path Algorithms</li>
<ul>	<li>3.8.1 Incremental Forward Stagewise Regression</li>
	<li>3.8.2 Piecewise-Linear Path Algorithms</li>
	<li>3.8.3 The Dantzig Selector</li>
	<li>3.8.4 The Grouped Lasso</li>
	<li>3.8.5 Further Properties of the Lasso</li>
	<li>3.8.6 Pathwise Coordinate Optimization</li>
</ul>
	<li>3.9 Computational Considerations</li>
	<li>Bibliographic Notes</li>
	<li>Exercises</li>
</ul>
	<li>4 Linear Methods for Classification</li>
<ul>	<li>4.1 Introduction</li>
	<li>4.2 Linear Regression of an Indicator Matrix</li>
	<li>4.3 Linear Discriminant Analysis</li>
<ul>	<li>4.3.1 Regularized Discriminant Analysis</li>
	<li>4.3.2 Computations for LDA</li>
	<li>4.3.3 Reduced-Rank Linear Discriminant Analysis</li>
</ul>
	<li>4.4 Logistic Regression</li>
<ul>	<li>4.4.1 Fitting Logistic Regression Models</li>
	<li>4.4.2 Example: South African Heart Disease</li>
	<li>4.4.3 Quadratic Approximations and Inference</li>
	<li>4.4.4 L1 Regularized Logistic Regression</li>
	<li>4.4.5 Logistic Regression or LDA?</li>
</ul>
	<li>4.5 Separating Hyperplanes</li>
<ul>	<li>4.5.1 Rosenblatt&rsquo;s Perceptron Learning Algorithm</li>
	<li>4.5.2 Optimal Separating Hyperplanes</li>
</ul>
	<li>Bibliographic Notes</li>
	<li>Exercises</li>
</ul>
	<li>5 Basis Expansions and Regularization</li>
<ul>	<li>5.1 Introduction</li>
	<li>5.2 Piecewise Polynomials and Splines</li>
<ul>	<li>5.2.1 Natural Cubic Splines</li>
	<li>5.2.2 Example: South African Heart Disease (Continued)</li>
	<li>5.2.3 Example: Phoneme Recognition</li>
</ul>
	<li>5.3 Filtering and Feature Extraction</li>
	<li>5.4 Smoothing Splines</li>
<ul>	<li>5.4.1 Degrees of Freedom and Smoother Matrices</li>
</ul>
	<li>5.5 Automatic Selection of the SmoothingParameters</li>
<ul>	<li>5.5.1 Fixing the Degrees of Freedom</li>
	<li>5.5.2 The Bias&ndash;Variance Tradeoff</li>
</ul>
	<li>5.6 Nonparametric Logistic Regression</li>
	<li>5.7 Multidimensional Splines</li>
	<li>5.8 Regularization and Reproducing Kernel Hilbert Spaces</li>
<ul>	<li>5.8.1 Spaces of Functions Generated by Kernels</li>
	<li>5.8.2 Examples of RKHS</li>
<ul>	<li>Penalized Polynomial Regression</li>
	<li>Gaussian Radial Basis Functions</li>
	<li>Support Vector Classifiers</li>
</ul>
</ul>
	<li>5.9 Wavelet Smoothing</li>
<ul>	<li>5.9.1 Wavelet Bases and the Wavelet Transform</li>
	<li>5.9.2 Adaptive Wavelet Filtering</li>
</ul>
	<li>Bibliographic Notes</li>
	<li>Exercises</li>
	<li>Appendix: Computations for Splines</li>
<ul>	<li>B-splines</li>
	<li>Computations for Smoothing Splines</li>
</ul>
</ul>
	<li>6 Kernel Smoothing Methods</li>
<ul>	<li>6.1 One-Dimensional Kernel Smoothers</li>
<ul>	<li>6.1.1 Local Linear Regression</li>
	<li>6.1.2 Local Polynomial Regression</li>
</ul>
	<li>6.2 Selecting the Width of the Kernel</li>
	<li>6.3 Local Regression in IRp</li>
	<li>6.4 Structured Local Regression Models in IRp</li>
<ul>	<li>6.4.1 Structured Kernels</li>
	<li>6.4.2 Structured Regression Functions</li>
</ul>
	<li>6.5 Local Likelihood and Other Models</li>
	<li>6.6 Kernel Density Estimation and Classification</li>
<ul>	<li>6.6.1 Kernel Density Estimation</li>
	<li>6.6.2 Kernel Density Classification</li>
	<li>6.6.3 The Naive Bayes Classifier</li>
</ul>
	<li>6.7 Radial Basis Functions and Kernels</li>
	<li>6.8 Mixture Models for Density Estimation and Classification</li>
	<li>6.9 Computational Considerations</li>
	<li>Bibliographic Notes</li>
	<li>Exercises</li>
</ul>
	<li>7 Model Assessment and Selection</li>
<ul>	<li>7.1 Introduction</li>
	<li>7.2 Bias, Variance and Model Complexity</li>
	<li>7.3 The Bias&ndash;Variance Decomposition</li>
<ul>	<li>7.3.1 Example: Bias&ndash;Variance Tradeoff</li>
</ul>
	<li>7.4 Optimism of the Training Error Rate</li>
	<li>7.5 Estimates of In-Sample Prediction Error</li>
	<li>7.6 The Effective Number of Parameters</li>
	<li>7.7 The Bayesian Approach and BIC</li>
	<li>7.8 Minimum Description Length</li>
	<li>7.9 Vapnik&ndash;Chervonenkis Dimension</li>
<ul>	<li>7.9.1 Example (Continued)</li>
</ul>
	<li>7.10 Cross-Validation</li>
<ul>	<li>7.10.1 K-Fold Cross-Validation</li>
	<li>7.10.2 The Wrong and Right Way to Do Cross-validation</li>
	<li>7.10.3 Does Cross-Validation Really Work?</li>
</ul>
	<li>7.11 Bootstrap Methods</li>
<ul>	<li>7.11.1 Example (Continued)</li>
	<li>7.12 Conditional or Expected Test Error?</li>
</ul>
	<li>Bibliographic Notes</li>
	<li>Exercises</li>
</ul>
	<li>8 Model Inference and Averaging</li>
<ul>	<li>8.1 Introduction</li>
	<li>8.2 The Bootstrap and Maximum Likelihood Methods</li>
<ul>	<li>8.2.1 A Smoothing Example</li>
	<li>8.2.2 Maximum Likelihood Inference</li>
	<li>8.2.3 Bootstrap versus Maximum Likelihood</li>
</ul>
	<li>8.3 Bayesian Methods</li>
	<li>8.4 Relationship Between the Bootstrap and Bayesian Inference</li>
	<li>8.5 The EM Algorithm</li>
<ul>	<li>8.5.1 Two-Component Mixture Model</li>
	<li>8.5.2 The EM Algorithm in General</li>
	<li>8.5.3 EM as a Maximization&ndash;Maximization Procedure</li>
</ul>
	<li>8.6 MCMC for Sampling from the Posterior</li>
	<li>8.7 Bagging</li>
<ul>	<li>8.7.1 Example: Trees with Simulated Data</li>
</ul>
	<li>8.8 Model Averaging and Stacking</li>
	<li>8.9 Stochastic Search: Bumping</li>
	<li>Bibliographic Notes</li>
	<li>Exercises</li>
</ul>
	<li>9 Additive Models, Trees, and Related Methods</li>
<ul>	<li>9.1 Generalized Additive Models</li>
<ul>	<li>9.1.1 Fitting Additive Models</li>
	<li>9.1.2 Example: Additive Logistic Regression</li>
	<li>9.1.3 Summary</li>
</ul>
	<li>9.2 Tree-Based Methods</li>
<ul>	<li>9.2.1 Background</li>
	<li>9.2.2 Regression Trees</li>
	<li>9.2.3 Classification Trees</li>
	<li>9.2.4 Other Issues</li>
<ul>	<li>Categorical Predictors</li>
	<li>The Loss Matrix</li>
	<li>Missing Predictor Values</li>
	<li>Why Binary Splits?</li>
	<li>Other Tree-Building Procedures</li>
	<li>Linear Combination Splits</li>
	<li>Instability of Trees</li>
	<li>Lack of Smoothness</li>
	<li>Difficulty in Capturing Additive Structure</li>
</ul>
	<li>9.2.5 Spam Example (Continued)</li>
</ul>
	<li>9.3 PRIM: Bump Hunting</li>
<ul>	<li>9.3.1 Spam Example (Continued)</li>
</ul>
	<li>9.4 MARS: Multivariate Adaptive Regression Splines</li>
<ul>	<li>9.4.1 Spam Example (Continued)</li>
	<li>9.4.2 Example (Simulated Data)</li>
	<li>9.4.3 Other Issues</li>
<ul>	<li>MARS for Classification</li>
	<li>Relationship of MARS to CART</li>
	<li>Mixed Inputs</li>
</ul>
</ul>
	<li>9.5 Hierarchical Mixtures of Experts</li>
	<li>9.6 Missing Data</li>
	<li>9.7 Computational Considerations</li>
	<li>Bibliographic Notes</li>
	<li>Exercises</li>
</ul>
	<li>10 Boosting and Additive Trees</li>
<ul>	<li>10.1 Boosting Methods</li>
<ul>	<li>10.1.1 Outline of This Chapter</li>
</ul>
	<li>10.2 Boosting Fits an Additive Model</li>
	<li>10.3 Forward Stagewise Additive Modeling</li>
	<li>10.4 Exponential Loss and AdaBoost</li>
	<li>10.5 Why Exponential Loss?</li>
	<li>10.6 Loss Functions and Robustness</li>
<ul>	<li>Robust Loss Functions for Classification</li>
	<li>Robust Loss Functions for Regression</li>
</ul>
	<li>10.7 &ldquo;Off-the-Shelf&rdquo; Procedures for Data Mining</li>
	<li>10.8 Example: Spam Data</li>
	<li>10.9 Boosting Trees</li>
	<li>10.10 Numerical Optimization via Gradient Boosting</li>
<ul>	<li>10.10.1 Steepest Descent</li>
	<li>10.10.2 Gradient Boosting</li>
	<li>10.10.3 Implementations of Gradient Boosting</li>
</ul>
	<li>10.11 Right-Sized Trees for Boosting</li>
	<li>10.12 Regularization</li>
<ul>	<li>10.12.1 Shrinkage</li>
	<li>10.12.2 Subsampling</li>
</ul>
	<li>10.13 Interpretation</li>
<ul>	<li>10.13.1 Relative Importance of Predictor Variables</li>
	<li>10.13.2 Partial Dependence Plots</li>
</ul>
	<li>10.14 Illustrations</li>
<ul>	<li>10.14.1 California Housing</li>
	<li>10.14.2 New Zealand Fish</li>
	<li>10.14.3 Demographics Data</li>
</ul>
	<li>Bibliographic Notes</li>
	<li>Exercises</li>
</ul>
	<li>11 Neural Networks</li>
<ul>	<li>11.1 Introduction</li>
	<li>11.2 Projection Pursuit Regression</li>
	<li>11.3 Neural Networks</li>
	<li>11.4 Fitting Neural Networks</li>
	<li>11.5 Some Issues in Training Neural Networks</li>
<ul>	<li>11.5.1 Starting Values</li>
	<li>11.5.2 Overfitting</li>
	<li>11.5.3 Scaling of the Inputs</li>
	<li>11.5.4 Number of Hidden Units and Layers</li>
	<li>11.5.5 Multiple Minima</li>
</ul>
	<li>11.6 Example: Simulated Data</li>
	<li>11.7 Example: ZIP Code Data</li>
	<li>11.8 Discussion</li>
	<li>11.9 Bayesian Neural Nets and the NIPS 2003 Challenge</li>
<ul>	<li>11.9.1 Bayes, Boosting and Bagging</li>
	<li>11.9.2 Performance Comparisons</li>
</ul>
	<li>11.10 Computational Considerations</li>
	<li>Bibliographic Notes</li>
	<li>Exercises</li>
</ul>
	<li>12 Support Vector Machines and Flexible Discriminants</li>
<ul>	<li>12.1 Introduction</li>
	<li>12.2 The Support Vector Classifier</li>
<ul>	<li>12.2.1 Computing the Support Vector Classifier</li>
	<li>12.2.2 Mixture Example (Continued)</li>
</ul>
	<li>12.3 Support Vector Machines and Kernels</li>
<ul>	<li>12.3.1 Computing the SVM for Classification</li>
	<li>12.3.2 The SVM as a Penalization Method</li>
	<li>12.3.3 Function Estimation and Reproducing Kernels</li>
	<li>12.3.4 SVMs and the Curse of Dimensionality</li>
	<li>12.3.5 A Path Algorithm for the SVM Classifier</li>
	<li>12.3.6 Support Vector Machines for Regression</li>
	<li>12.3.7 Regression and Kernels</li>
	<li>12.3.8 Discussion</li>
</ul>
	<li>12.4 Generalizing Linear Discriminant Analysis</li>
	<li>12.5 Flexible Discriminant Analysis</li>
<ul>	<li>12.5.1 Computing the FDA Estimates</li>
</ul>
	<li>12.6 Penalized Discriminant Analysis</li>
	<li>12.7 Mixture Discriminant Analysis</li>
<ul>	<li>12.7.1 Example: Waveform Data</li>
<ul>	<li>Computational Considerations</li>
</ul>
</ul>
	<li>Bibliographic Notes</li>
	<li>Exercises</li>
</ul>
	<li>13 Prototype Methods and Nearest-Neighbors</li>
<ul>	<li>13.1 Introduction</li>
	<li>13.2 Prototype Methods</li>
<ul>	<li>13.2.1 K-means Clustering</li>
	<li>13.2.2 Learning Vector Quantization</li>
	<li>13.2.3 Gaussian Mixtures</li>
</ul>
	<li>13.3 k-Nearest-Neighbor Classifiers</li>
<ul>	<li>13.3.1 Example: A Comparative Study</li>
	<li>13.3.2 Example: k-Nearest-Neighbors and Image Scene Classification</li>
	<li>13.3.3 Invariant Metrics and Tangent Distance</li>
</ul>
	<li>13.4 Adaptive Nearest-Neighbor Methods</li>
<ul>	<li>13.4.1 Example</li>
	<li>13.4.2 Global Dimension Reduction for Nearest-Neighbors</li>
</ul>
	<li>13.5 Computational Considerations</li>
	<li>Bibliographic Notes</li>
	<li>Exercises</li>
</ul>
	<li>14 Unsupervised Learning</li>
<ul>	<li>14.1 Introduction</li>
	<li>14.2 Association Rules</li>
<ul>	<li>14.2.1 Market Basket Analysis</li>
	<li>14.2.2 The Apriori Algorithm</li>
	<li>14.2.3 Example: Market Basket Analysis</li>
	<li>14.2.4 Unsupervised as Supervised Learning</li>
	<li>14.2.5 Generalized Association Rules</li>
	<li>14.2.6 Choice of Supervised Learning Method</li>
	<li>14.2.7 Example: Market Basket Analysis (Continued)</li>
</ul>
	<li>14.3 Cluster Analysis</li>
<ul>	<li>14.3.1 Proximity Matrices</li>
	<li>14.3.2 Dissimilarities Based on Attributes</li>
	<li>14.3.3 Object Dissimilarity</li>
	<li>14.3.4 Clustering Algorithms</li>
	<li>14.3.5 Combinatorial Algorithms</li>
	<li>14.3.6 K-means</li>
	<li>14.3.7 Gaussian Mixtures as Soft K-means Clustering</li>
	<li>14.3.8 Example: Human Tumor Microarray Data</li>
	<li>14.3.9 Vector Quantization</li>
	<li>14.3.10 K-medoids</li>
<ul>	<li>Example: Country Dissimilarities</li>
</ul>
	<li>14.3.11 Practical Issues</li>
	<li>14.3.12 Hierarchical Clustering</li>
<ul>	<li>Agglomerative Clustering</li>
	<li>Example: Human Cancer Microarray Data (Continued)</li>
	<li>Divisive Clustering</li>
</ul>
</ul>
	<li>14.4 Self-Organizing Maps</li>
<ul>	<li>Example: Document Organization and Retrieval</li>
</ul>
	<li>14.5 Principal Components, Curves and Surfaces</li>
<ul>	<li>14.5.1 Principal Components</li>
<ul>	<li>Example: Handwritten Digits</li>
	<li>Example: Procrustes Transformations and Shape Averaging</li>
</ul>
	<li>14.5.2 Principal Curves and Surfaces</li>
	<li>14.5.3 Spectral Clustering</li>
	<li>14.5.4 Kernel Principal Components</li>
	<li>14.5.5 Sparse Principal Components</li>
</ul>
	<li>14.6 Non-negative Matrix Factorization</li>
<ul>	<li>14.6.1 Archetypal Analysis</li>
</ul>
	<li>14.7 Independent Component Analysis and Exploratory Projection Pursuit</li>
<ul>	<li>14.7.1 Latent Variables and Factor Analysis</li>
	<li>14.7.2 Independent Component Analysis</li>
<ul>	<li>Example: Handwritten Digits</li>
	<li>Example: EEG Time Courses</li>
</ul>
	<li>14.7.3 Exploratory Projection Pursuit</li>
	<li>14.7.4 A Direct Approach to ICA</li>
</ul>
	<li>14.8 Multidimensional Scaling</li>
	<li>14.9 Nonlinear Dimension Reduction and Local Multidimensional Scaling</li>
	<li>14.10 The Google PageRank Algorithm</li>
	<li>Bibliographic Notes</li>
	<li>Exercises</li>
</ul>
	<li>15 Random Forests</li>
<ul>	<li>15.1 Introduction</li>
	<li>15.2 Definition of Random Forests</li>
	<li>15.3 Details of Random Forests</li>
<ul>	<li>15.3.1 Out of Bag Samples</li>
	<li>15.3.2 Variable Importance</li>
	<li>15.3.3 Proximity Plots</li>
	<li>15.3.4 Random Forests and Overfitting</li>
</ul>
	<li>15.4 Analysis of Random Forests</li>
<ul>	<li>15.4.1 Variance and the De-Correlation Effect</li>
	<li>15.4.2 Bias</li>
	<li>15.4.3 Adaptive Nearest Neighbors</li>
</ul>
	<li>Bibliographic Notes</li>
	<li>Exercises</li>
</ul>
	<li>16 Ensemble Learning</li>
<ul>	<li>16.1 Introduction</li>
	<li>16.2 Boosting and Regularization Paths</li>
<ul>	<li>16.2.1 Penalized Regression</li>
	<li>16.2.2 The &ldquo;Bet on Sparsity&rdquo; Principle</li>
	<li>16.2.3 Regularization Paths, Over-fitting and Margins</li>
</ul>
	<li>16.3 Learning Ensembles</li>
<ul>	<li>16.3.1 Learning a Good Ensemble</li>
	<li>16.3.2 Rule Ensembles</li>
</ul>
	<li>Bibliographic Notes</li>
	<li>Exercises</li>
</ul>
	<li>17 Undirected Graphical Models</li>
<ul>	<li>17.1 Introduction</li>
	<li>17.2 Markov Graphs and Their Properties</li>
	<li>17.3 Undirected Graphical Models for Continuous Variables</li>
<ul>	<li>17.3.1 Estimation of the Parameters when the Graph Structure is Known</li>
	<li>17.3.2 Estimation of the Graph Structure</li>
</ul>
	<li>17.4 Undirected Graphical Models for Discrete Variables</li>
<ul>	<li>17.4.1 Estimation of the Parameters when the Graph Structure is Known</li>
	<li>17.4.2 Hidden Nodes</li>
	<li>17.4.3 Estimation of the Graph Structure</li>
	<li>17.4.4 Restricted Boltzmann Machines</li>
</ul>
	<li>Bibliographic Notes</li>
	<li>Exercises</li>
</ul>
	<li>18 High-Dimensional Problems: p &gt;&gt; N</li>
<ul>	<li>18.1 When p is Much Bigger than N</li>
	<li>18.2 Diagonal Linear Discriminant Analysis and Nearest Shrunken Centroids</li>
	<li>18.3 Linear Classifiers with Quadratic Regularization</li>
<ul>	<li>18.3.1 Regularized Discriminant Analysis</li>
	<li>18.3.2 Logistic Regression with Quadratic Regularization</li>
	<li>18.3.3 The Support Vector Classifier</li>
	<li>18.3.4 Feature Selection</li>
	<li>18.3.5 Computational Shortcuts When p &gt;&gt; N</li>
</ul>
	<li>18.4 Linear Classifiers with L1 Regularization</li>
<ul>	<li>18.4.1 Application of Lasso to Protein Mass Spectroscopy</li>
	<li>18.4.2 The Fused Lasso for Functional Data</li>
</ul>
	<li>18.5 Classification When Features are Unavailable</li>
<ul>	<li>18.5.1 Example: String Kernels and Protein Classification</li>
	<li>18.5.2 Classification and Other Models Using Inner-Product Kernels and Pairwise Distances</li>
	<li>18.5.3 Example: Abstracts Classification</li>
</ul>
	<li>18.6 High-Dimensional Regression: Supervised Principal Components</li>
<ul>	<li>18.6.1 Connection to Latent-Variable Modeling</li>
	<li>18.6.2 Relationship with Partial Least Squares</li>
	<li>18.6.3 Pre-Conditioning for Feature Selection</li>
</ul>
	<li>18.7 Feature Assessment and the Multiple-Testing Problem</li>
<ul>	<li>18.7.1 The False Discovery Rate</li>
	<li>18.7.2 Asymmetric Cutpoints and the SAM Procedure</li>
	<li>18.7.3 A Bayesian Interpretation of the FDR</li>
</ul>
	<li>18.8 Bibliographic Notes</li>
	<li>Exercises</li>
</ul>
	<li>References</li>
	<li>Author Index</li>
	<li>Index</li>
</ul>
</body></html>